<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>Cytogenetic Bands Dynamic Programming Algorithm</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg"></TH></TR>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
	<table bgcolor="000000" cellpadding="1" cellspacing="1" width="100%" height="27"><tr bgcolor="2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
			<TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp; 
			&nbsp;<a href="/index.html" class="topbar">
				Home</a> &nbsp; - &nbsp;
			<a href="/cgi-bin/hgGateway" class="topbar">
				Genome Browser</a> &nbsp; - &nbsp; 
			<A HREF="/cgi-bin/hgBlat?command=start" class="topbar">
				Blat Search</a> &nbsp; - &nbsp; 
			<A HREF="/FAQ.html" class="topbar">
				FAQ</a> &nbsp; - &nbsp; 
			<A HREF="/goldenPath/help/hgTracksHelp.html" class="topbar">
				User Guide</a> &nbsp;</font></TD>
	</TR></TABLE>
	</TD></TR></TABLE>
</TD></TR>

<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>	
 <table BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><tr><td>
 <table BGCOLOR="D9E4F8"  BACKGROUND="http://genome.ucsc.edu/images/hr.gif" WIDTH=100%><tr><td>
 <font SIZE="4"><a NAME="TOC"></a><b>&nbsp; Cytogenetic Bands</font>
 </td></tr></table>

<!--outer table is for border purposes-->
<table BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0 CELLSPACING=0><th HEIGHT=15></th><tr><td WIDTH=10></td><td>

<H1>A Dynamic Programming Method to Locate Cytogenetic Bands on the
Working Draft Sequence</H1>

<P>Terrence Furey, Wonhee Jang, Arek Kasprzyk, Ewan Birney, Greg Schuler,
Vivian Cheung, Barbara Trask, David Haussler

<P>UCSC, NCBI, EBI, EBI, NCBI, Children's Hospital of Philadelphia, Fred
Hutchinson Cancer Research Center, HHMI/UCSC

<P>Advances in Genome Biology and Technology, February 3-6, 2001, Marco
Island Resort & Golf Club

<P>We have developed a dynamic programming algorithm that predicts the
locations of chromosomal bands and sub-bands on the working draft genome
sequence using FISH-mapped clones. The BAC Resource Consortium [1] has located
positions on the Sept. 5 working draft genome sequence at
<A HREF="http://genome.ucsc.edu">http://genome.ucsc.edu</A> for 7364 clones that have been mapped by FISH to
particular cytogenetic bands or ranges of bands. This is the primary
data
used by the algorithm. The algorithm predicts band locations in the
genome
sequence so as to maximize the number of clones from this data set that
lie in the bands predicted by FISH, subject to the constraint that the
bands must appear in the correct order and their sizes cannot deviate
greatly from their standard fractional sizes. In cases of contradictory
data, higher weight is given to higher resolution FISH data. In the
absence of FISH data, sub-band sizes are optimally fit to the standard
fractional sizes. The resulting band predictions can be used to help
locate genes of clinical and scientific significance in the working
draft,
and also to study properties of the sequence that are associated with
cytogenetic banding.

<P>[1] BAC Resource Consortium, Navigating from chromosomal abberations to genes:
linking the cytogenetic and sequence maps of the human genome,
<i>Nature</i>,
Vol. 409.

    </TD>
    </TR>
  </TABLE>
    <hr>
    <address><a href="mailto:booch@cse.ucsc.edu">Terry Furey</a></address>
<!-- Created: Fri Aug  3 15:16:48 PDT 2001 -->
<!-- hhmts start -->
Last modified: Fri Mar 22 13:31:13 PST 2002
<!-- hhmts end -->
</TD></TR></TABLE>
	</TD></TR></TABLE>
</TD></TR></TABLE>	

</TD></TR></TABLE>
</BODY></HTML>
