<H2>Description</H2>
<P>
This track shows predictions of <A HREF="http://augustus.gobics.de/" TARGET=_blank>AUGUSTUS</A>.
AUGUSTUS is available through the <A HREF="http://gobics.de/department/software" TARGET=_blank>GOBICS</A> web
server.</P>

<H2>Display Conventions and Configuration</H2>
<P>
This annotation follows the display conventions for composite tracks.
To display only selected subtracks, uncheck the boxes next to the tracks
you wish to hide.  This track also follows the display conventions for
<A HREF="../goldenPath/help/hgTracksHelp.html#GeneDisplay">gene prediction
tracks</A>.</P>
<P>
This track contains an optional codon coloring feature that allows users to quickly validate and compare gene predictions.
To display codon colors, select the <em>genomic codons</em> option from the
<em>Color track by codons</em> pull-down menu. Click the
<A HREF="../goldenPath/help/hgCodonColoring.html">Help on codon coloring</A>
link for more information about this feature.</P>

<H2>Methods</H2>
<P>
AUGUSTUS uses a generalized hidden Markov model (GHMM) that models coding and 
non-coding sequence, splice sites, the branch point region, translation start 
and end, and lengths of exons and introns. This version has been trained on a 
set of 1284 human genes. 

<H3>Augustus Gene Predictions Using Hints</H3>
<P>
This subtrack was made using hints from several other tracks:
<ul>
<li>Human mRNAs and Spliced Human ESTs aligned with BLAT</li>
<li>TransMapped RefSeq genes from mouse, rat, zebrafish, chicken, cow, <i>Xenopus tropicalis</i>, dog and chimpanzee
<li>Retroposed Genes</li>
<li>Exoniphy</li>
</ul>
</P>

<H3>Augustus <i>De Novo</i> Gene Predictions</H3>
<P>This subtrack was made using only the target genome sequence and
evolutionary conservation.  The conservation information was extracted
from the Exoniphy track. Further, hints about retroposed genes were used, that are based only
on previous <i>de novo</i> predictions of AUGUSTUS. No transcribed
sequences were used for this track.</P>

<H3>Augustus <i>Ab Initio</i> Gene Predictions</H3>

These predictions were made using only the genomic sequence; no conservation, homology information or 
transcribed sequence was used.</P>

<H2>Credits</H2>
<P>
The Augustus subtracks were created by Mario Stanke. The TransMap track
was created by Mark Diekhans, the Retroposed Genes tracks by Robert
Baertsch, and the Exoniphy track by Adam Siepel's group.</P>

<H2>References</H2>
<P>
Stanke M.
<A HREF="http://webdoc.sub.gwdg.de/diss/2004/stanke/index.html"
TARGET=_blank>Gene prediction with a hidden Markov model</A>.
<em>Ph.D. thesis</em>. Universit&auml;t G&ouml;ttingen, Germany. 2004.</P>

<P>
Stanke M, Steinkamp R, Waack S, Morgenstern B.
<A HREF="http://nar.oxfordjournals.org/cgi/content/abstract/32/suppl_2/W309"
TARGET=_blank>AUGUSTUS: a web server for gene finding in eukaryotes</A>.
<em>Nucl Acids Res</em>. 2004 Jul 1;32(Web Server Issue):W309-12.</P>

<P>
Stanke M, Tzvetkova A, Morgenstern B.
<A HREF="http://genomebiology.com/2006/7/s1/S11/abstract/" TARGET=_blank>
AUGUSTUS at EGASP: using EST, protein and genomic alignments for improved
gene prediction in the human genome</A>.
<em>Genome Biology</em>. 2006;7(Suppl 1):S11.</P>

<P>
Stanke M, Waack S.
<A HREF="http://bioinformatics.oupjournals.org/cgi/content/abstract/19/suppl_2/ii215"
TARGET=_blank>Gene prediction with a hidden Markov model and a new intron
submodel</A>.
<em>Bioinformatics</em>. 2003 Sep;19(Suppl. 2):ii215-25.</P>

