<H2>Description</H2>
<P>
1447 copy number 
variable regions found in the HapMap Phase II data using SNP and BAC microarray analysis.
</UL></P>

<H2>Display Conventions and Configuration</H2>
<P>
CNP and ISV regions are indicated by solid blocks that are color-coded to 
indicated the type of variation detected:
<UL>
<LI>
<B><FONT COLOR="green">Green</FONT>:</B> gain (duplications)
<LI>
<B><FONT COLOR="red">Red</FONT>:</B> loss (deletions)
<LI>
<B><FONT COLOR="blue">Blue</FONT>:</B> gain and loss (both deletion and duplication)
<LI>
<B>Black:</B> inversion
<LI>
<B><FONT COLOR="gray">Gray</FONT>:</B> gain or loss (unknown direction)
</UL></P>

<H2>Methods</H2>

<P>Experiments were performed with the International HapMap DNA and cell-line collection
using two technologies: comparative analysis of hybridization intensities on
Affymetric GeneChip Human Mapping 500K early access arrays (500K EA)
and comparative genomic hybridization with a Whole Genome TilePath (WGTP)
array.

<H2>Validation</H2>


The authors utilized numerous quality meaures, including
repeated experiments on the WGTP for 82 individual and on the 500K EA for 15 individuals.
The average false-positive rate per experiment was held beneath 5%.  Aberrant chromosomes were
removed from the analysis.    Further details are available in the Nature paper cited below.

<H2>References</H2>

<P>
Redon, R., Ishikawa, S., Fitch, K., Feuk, L., Perry, G., Andrews, T., Fiegler, H.,
Lee, C., Jones, K., Scherer, S., Hurles, M. <em>et al</em>.
<A HREF="http://www.nature.com/nature/journal/v444/n7118/full/nature05329.html" TARGET=_blank>
Global variation in copy number in the human genome</A>.
<em>Nature</em> <B>444</B>(7118), 444-454 (2006).
