<html><body>

<H2>Description</H2>
<P>
ENCODE regions were investigated by ChIP-chip, analyzing both histone H3 acetylation (H3ac; H3 acetylated lysines 9 and14) and histone H4 acetylation (H4ac; H4 acetylated lysined 5,8,12,16). This analysis was performed using ChIP material obtained from cells that were either untreated or treated with 5mM Na-Butyrate for 12 hours. Na-Butyrate is a histone deacetylase inhibitor (HDACi) that increases bulk levels of acetylated histones. Four tracks presented in the genome browser represent the ChIP-chip signal obtained for either H3ac or H4ac, using cells that were untreated or treated with butyrate: H3ac 0h, H3ac 12h, H4ac 0h, H4ac 12h. Two additional tracks indicate those spots where H3ac or H4ac levels are significantly changed by butyrate treatment. 
</P>
<H2>Methods</H2>
<P>
Chromatin immunoprecipitation, DNA labelling and array hybridization were exactly as previously described (Rada-Iglesias, et al. 2005). A set of enriched spots was obtained for each of H3ac 0h, H3ac 12h, H4ac 0h and H4ac 12h using the same pre-processing and analysis procedures as in (Rada-Iglesias, et al.). Enriched spots showing different histone acetylation levels between 0h and 12h treatment were then detected through an empirical Bayes method (Smyth). All spots with B-score>0 were either classified as <em>up</em> or <em>down</em> depending on whether the acetylation was increased or decreased. For spots missing all measurements at one of the time points due to filtering, the B-score was instead calculated on un-filtered, print-tip lowess normalized (Yang, et al.) raw data. Enriched spots that were not present in any of the <em>up</em> or <em>down</em> groups were classified as <em>unchanged</em>. 
</P>
<P>
The raw data for this track is available at 
<A TARGET=_BLANK HREF="http://www.ebi.ac.uk/arrayexpress">EBI ArrayExpress</A>, as experiment 
<A TARGET=_BLANK HREF="http://www.ebi.ac.uk/arrayexpress/experiments/E-MEXP-693">E-MEXP-693</A>.

</P>
<H2>Verification</H2>
<P>
New ChIPs were performed for both H3ac and H4ac, both for untreated cells and cells treated with 5mM Na-butyrate for 12 hours. Furthermore, ChIP was performed in cells that were treated with 5mM Na-butyrate for 15 minutes, 2 hours, 6 hours and 12 hours+6 hours without butyrate. All these ChIP DNAs were analyzed by PCR, including 10 regions were loss of acetylation after 12 hours butyrate treatment was observed in ChIP-chip experiments, two regions where a trend towards increase acetylation was observed, one negative region where no acetylation and no change was observed and three control regions not included in the ENCODE array and covering promoter regions of previously known butyrate-responsive genes.
</P>
<H2>Credits</H2>
<P>
These experiments were performed in the 
<A HREF="http://www.genpat.uu.se/Research/genetics_genomics/wadelius/contact/"
TARGET=_blank>Claes Wadelius lab</A>, Department of Genetics and Pathology, Rudbeck Laboratory, Uppsala University. The statistical analysis was done at the 
<A HREF="http://www.lcb.uu.se" TARGET=_blank>Linnaeus Centre for 
Bioinformatics</A> at Uppsala University.  Microarrays were produced at the 
<A HREF="http://www.sanger.ac.uk" TARGET=_blank>Sanger Institute</A>. 
</P>

<h2>References</h2>

<p>
Ameur A, Yankovski V, Enroth S, Spjuth O, Komorowski J.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/16455749" target="_blank">
The LCB Data Warehouse</a>.
<em>Bioinformatics</em>. 2006 Apr 15;22(8):1024-6.
</p>

<p>
Rada-Iglesias A, Wallerman O, Koch C, Ameur A, Enroth S, Clelland G, Wester K, Wilcox S, Dovey OM,
Ellis PD <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/16221759" target="_blank">
Binding sites for metabolic disease related transcription factors inferred at base pair resolution
by chromatin immunoprecipitation and genomic microarrays</a>.
<em>Hum Mol Genet</em>. 2005 Nov 15;14(22):3435-47.
</p>

<p>
Smyth GK.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/16646809" target="_blank">
Linear models and empirical bayes methods for assessing differential expression in microarray
experiments</a>.
<em>Stat Appl Genet Mol Biol</em>. 2004;3:Article3.
</p>

<p>
Yang YH, Dudoit S, Luu P, Lin DM, Peng V, Ngai J, Speed TP.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/11842121" target="_blank">
Normalization for cDNA microarray data: a robust composite method addressing single and multiple
slide systematic variation</a>.
<em>Nucleic Acids Res</em>. 2002 Feb 15;30(4):e15.
</p>
