<H2>Description</H2>
<P>
This track displays the results of ENCODE region-wide localization for three 
transcription factors (HNF-3b, HNF-4a and USF-1) and acetylated 
histone H3 (H3ac). The heights of the peaks in the graphical display indicate 
the ratio of enriched non-amplified DNA to input DNA. </P>
<P>
The data for each of the transcription factors and H3ac are displayed in
individual subtracks. The analysis cut-off threshold is indicated in each 
subtrack by a horizontal line. Tentative binding sites (TBSs) 
in spots passing the cut-off are displayed in a separate subtrack, 
ChIP-chip (HepG2) Sites. These sites are numbered corresponding to the ranking 
of spots based on enrichment ratios. Each TBS is assigned a value indicating 
how often it was found in separate BioProspector software runs for the 
prediction of TBSs (<em>e.g.</em> 1000 indicates that a TBS was found in ten 
out of ten runs).</P>
<P>
The raw data for this track is available at 
<A TARGET=_BLANK HREF="http://www.ebi.ac.uk/arrayexpress">EBI ArrayExpress</A>, as experiment 
<A TARGET=_BLANK HREF="http://www.ebi.ac.uk/arrayexpress/#ae-browse/q=E-MEXP-452">E-MEXP-452</A>.
</P>

<H2>Methods</H2>
<P>
Chromatin from HepG2 cells was cross-linked with formaldehyde and sonicated
to produce DNA fragments of size 0.5-2 kb. Chromatin was precipitated using
antibodies against HNF-4a, HNF-3b, USF-1 or H3ac. DNA from a single ChIP 
reaction was labeled with Cy5, and a fraction of the total input was labeled 
with Cy3. There was no amplification of the ChIP DNA or the input DNA prior to
this step to avoid introducing bias. </P>
<P>
This DNA was combined and hybridized 
to PCR-based tiling path ENCODE arrays. Most array elements were printed 
only once on the slide, but X-chromosomal regions (ENm006 and ENr324) were 
printed in duplicate. There were approximately 19,000 spots/slide. The array 
provided about 75% coverage of the ENCODE regions.</P>
<P>
Spots flagged as bad by the image processing step were removed; those that 
remained were normalized. The average log2 ratio was calculated for spots 
that were replicated on the array. A log odds score for differential 
enrichment with the negative control was calculated using an empirical Bayes
method. There were four log odds scores for each spot, one for each 
antibody. If this score was greater than 0 and the log2 ratio was greater than 
1.25 (indicative of a strong positive signal), based on at least 2 replicates, 
the spots were considered to be enriched. </P>
<P>
Binding sites were identified using the BioProspector software. 
Because the software is non-deterministic, different runs
may produce different results for the same data. Predictions consistent
across many runs are more likely to be correct; therefore, the analysis was
repeated, keeping all binding sites occurring in each top-scoring motif to 
generate a set of candidates. TBSs present in at least
five out of ten runs were selected. Further method details are described in 
Rada-Iglesias <em>et al.</em> (2005). </P>
<P>
In the graphical display, overlapping sequences were 
removed by changing the start position of downstream spots to generate a 
continuous track. To give each track a comparable scale, the values for the 
most enriched spots were lowered to 15. Spots deemed as false positives, when 
compared to a no antibody ChIP-chip experiment, were assigned a value of 0.
</P>
<H2>Verification</H2>
<P>
A negative control was done using no antibody for the ChIP-chip to reduce 
the number of false positives.  Three independent biological replicates 
were performed for each antibody; three negative control ChIPs were also 
analyzed. Semi-quantitative PCR was used to verify enrichment in at 
least ten positive spots for each antibody. </P>

<H2>Credits</H2>
<P>
These experiments were performed in the 
<A HREF="http://www.genpat.uu.se/Research/genetics_genomics/wadelius/contact/"
TARGET=_blank>Claes Wadelius lab</A>. The statistical analysis was done at the 
<A HREF="http://www.lcb.uu.se" TARGET=_blank>Linnaeus Centre for 
Bioinformatics</A> at Uppsala University.  Microarrays were produced at the 
<A HREF="http://www.sanger.ac.uk" TARGET=_blank>Sanger Institute</A>. </P>

<H2>References</H2>
<P>
Rada-Iglesias, A. <em>et al.</em>
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=16221759&query_hl=1"
TARGET=_blank>Binding sites for metabolic disease related transcription factors 
inferred at base pair resolution by chromatin immunoprecipitation and genomic 
microarrays</A>.
<em>Hum Mol Genet.</em> <B>14</B>, 3435-3447 (2005).</P> 

