<H2> Description </H2>
<P>
This track shows RNA secondary structure predictions made with the
EvoFold program, a comparative method that exploits the evolutionary signal
of genomic multiple-sequence alignments for identifying conserved
functional RNA structures. </P>

<H2> Display Conventions and Configuration </H2>
<P>
Track elements are labeled using the convention <em>ID_strand_score</em>.
When zoomed out beyond the base level, secondary structure prediction regions
are indicated by blocks, with the stem-pairing regions shown in a darker shade 
than unpaired regions. Arrows indicate the predicted strand.
When zoomed in to the base level, the specific secondary structure predictions 
are shown in parenthesis format. The confidence score for each position is
indicated in grayscale, with darker shades corresponding to higher scores.
<P>
The details page for each track element shows the predicted secondary structure 
(labeled <em>SS anno</em>), together with details of the multiple species 
alignments at that location.
<P>
Substitutions relative to the human sequence are 
color-coded according to their compatibility with the predicted secondary 
structure (see the color legend on the details page). Each prediction is 
assigned an overall score and a sequence of position-specific scores. The 
overall score measures evidence for any functional RNA structures in the given 
prediction region, while the position-specific scores (0 - 9) measure the
confidence of the base-specific annotations. Base-pairing positions are
annotated with the same pair symbol. The <em>offsets</em> are provided to ease
visual navigation of the alignment in terms of the human sequence. The offset
is calculated (in units of ten) from the start position of the element on 
the positive strand or from the end position when on the negative strand.</P>
<P>
The graphical display may be filtered to show only those track elements 
with scores that meet or exceed a certain threshhold. To set a 
threshhold, type the minimum score into the text box at the top of the 
description page.
<P>


<H2>Methods</H2>
<P>
Evofold makes use of phylogenetic
stochastic context-free grammars (phylo-SCFGs), which are combined
probabilistic models of RNA secondary structure and primary sequence
evolution. The predictions consist of both a specific RNA secondary
structure and an overall score. The overall score is essentially a
log-odd score between a phylo-SCFG modeling the constrained evolution of
stem-pairing regions and one which only models unpaired regions.</P>
<P>
The predictions for this track were based on the conserved elements of
the 28-way threaded blockset aligner (TBA) alignments present in the ENCODE
regions (see the TBA Alignment track for more information).
NOTE: The alignments displayed on the details page reference the
newer ENCODE hg18 TBA alignments, and so differ from those used
to create these predictions.</P>

<H2>Credits</H2> 
<P>
The EvoFold program and browser track were developed by 
<A HREF="mailto:jsp@soe.ucsc.edu">Jakob Skou Pedersen</A> of the UCSC
Genome Bioinformatics Group.</P>
<P>
The 28-way TBA multiple alignments were created by <A
HREF="http://www.genome.gov/19519052" TARGET=_blank>Elliott Margulies</A>
of the <A HREF="http://www.genome.gov/10000452" TARGET=_blank>Green
Lab</A> at NHGRI.</P>
<P>
TBA was provided by Minmei Hou, Scott Schwartz and Webb Miller of the 
<A HREF="http://www.bx.psu.edu/miller_lab/" TARGET=_blank>Penn State Bioinformatics 
Group</A>.</P>

<H2>References</H2>
<H3>EvoFold</H3>
<P>
Pedersen JS, Bejerano G, Siepel A, Rosenbloom K,
Lindblad-Toh K, Lander ES, Kent J, Miller W,
Haussler D. <A HREF="http://dx.doi.org/10.1371/journal.pcbi.0020033"
TARGET=_blank> Identification and classification of conserved RNA
secondary structures in the human genome</A>. <em>PLoS Comput
Biol</em>. 2006 Apr;2(4):e33.</P>

<H3>Phylo-SCFGs</H3>
<P>
Knudsen B, Hein J.  
<A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/15/6/446"
TARGET=_blank>RNA secondary structure prediction using stochastic context-free 
grammars and evolutionary history</A>.
<em>Bioinformatics</em>. 1999 Jun;15(6):446-54.</P>
<P>
Pedersen JS, Meyer IM, Forsberg R, Simmonds P, Hein J. 
<A HREF="http://nar.oxfordjournals.org/cgi/content/full/32/16/4925"
TARGET=_blank>A comparative method for finding and folding RNA
secondary structures within protein-coding regions</A>. 
<em>Nucleic Acids Res</em>. 2004 Sep 24;32(16):4925-36.</P>

<H3>PhastCons</H3>
<P>
Siepel A, Bejerano G, Pedersen JS, Hinrichs AS, Hou M, Rosenbloom
K, Clawson H, Spieth J, Hillier LW, Richards S, Weinstock GM,
Wilson RK, Gibbs RA, Kent WJ, Miller W, Haussler D. 
<A HREF="http://www.genome.org/cgi/doi/10.1101/gr.3715005"
TARGET=_blank>Evolutionarily conserved elements in vertebrate, insect, worm, 
and yeast genomes</A>.
<em>Genome Res</em>. 2005 Aug;15(8):1034-50.</P>
