<H2>Description</H2>
<P>
This track displays DNaseI-hypersensitive sites identified using two methods
(DNase-chip and MPSS sequencing) in seven human cell types:
<UL>
 <LI>primary unactivated and activated CD4+ T cells</LI>
 <LI><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM06990&PgId=166" 
TARGET=_blank>GM06990</A> lymphoblastoid</LI>
 <LI><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=CCL-2.2&Template=cellBiology" 
TARGET=_BLANK>HeLa S3</A>  
cervical carcinoma (Puck <EM>et al.</EM>, 1956)</LI>
 <LI><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=HB-8065&Template=cellBiology" 
TARGET=_blank>HepG2</A> liver carcinoma</LI>
 <LI><A HREF="http://www.nationalstemcellbank.org/" TARGET=_blank>H9</A>
human undifferentiated embryonic stem (ES) (Thomson <EM>et al.</EM>, 1998)</LI>
 <LI><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=CCL-186&Template=cellBiology" 
TARGET=_blank>IMR90</A> human fibroblast</LI>
 <LI><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=CCL-243&Template=cellBiology"
TARGET=_blank>K562</A> myeloid leukemia-derived (Klein <EM>et al.</EM>, 1976)</LI>
</UL>
</P>

<P>
DNaseI-hypersensitive sites are associated with all types of gene 
regulatory regions, including promoters, enhancers, silencers, insulators, and 
locus control regions.</P>

<H2>Display Conventions and Configuration</H2>
<P>
The subtracks within this track are grouped into three sections:
<UL>
<LI>
<B>Raw</B> subtracks display log2 ratio data averaged from
three biological replicates and three DNase concentrations.
<LI>
<B>Pval</B> subtracks show significant regions that likely 
represent valid DNaseI-hypersensitive sites based on the raw data.
The higher the score for the region, the more likely the site
is to be hypersensitive.  
Regions have unique identifiers that are prefixed with
the cell type.
For display purposes, the p value scores were mapped
to integer scores in the range 0-1000.
Regions are displayed in a range of light
gray to black, based on score.
<LI>
<B>MPSS</B> subtracks show hypersensitive sites determined by 
massively parallel signature sequencing (MPSS). 
Each cluster has a unique identifier.
The last digit of each identifier represents the 
number of sequences that map within that particular cluster. The sequence number
is also reflected in the score, <em>e.g.</em> a cluster of two sequences scores 
500, three sequences scores 750 and four or more sequences scores 1000. 
Sites are displayed in a range of light gray to black, 
based on score.
</UL>
The &quot;Raw&quot; and &quot;Pval&quot; subtracks are
displayed by default. Use the checkboxes on the Track Settings page to change
the subtracks displayed.

<H2>Methods</H2>
<H3> DNase-Chip</H3>
<P>
DNaseI hypersensitive sites were isolated using a method called 
DNase-chip (Crawford <EM>et al.</EM>, 2006). 
Briefly, DNaseI digested ends from intact chromatin were captured using
three different DNase concentrations as well as three biological replicates. 
This material was amplified, labeled, and hybridized to NimbleGen 
ENCODE tiled microarrays.
<P>
H9 human ES cells (Thomson <EM>et al.</EM>, 1998) were cultured on a feeder layer of 
mitotically inactivated mouse embryo fibroblasts. For analysis, human ES cell 
colonies were separated away from the feeder layer and processed for 
DNaseI hypersensitive site mapping. Cultures were routinely inspected by
immunohistochemistry, flow cytometry, and microarray to ensure that the
human ES cells were in the undifferentiated state.</P>
<P>
For the DNase-chip experiments, the raw data were averaged from nine
hybridizations per cell type. The Pval scores represent -log10 p values as
determined by the ACME (Algorithm for Capturing Microarray Enrichment) program
(Scacheri <EM>et al.</EM>, 2006). Only regions that had p value &lt;
0.001 were included. For display in the Genome Browser, the p value scores were
mapped to integer scores in the range 0-1000 using the following formula: score
= (pVal * 35) + 100. The -log10 p values can be viewed using the Table Browser.
<P>

<H3> MPSS Sequencing </H3>
<P>
Primary human CD4+ T cells were activated by incubation with anti-CD3
and anti-CD28 antibodies for 24 hours. DNaseI-hypersensitive sites
were cloned from the cells before and after activation,
and sequenced using massively parallel signature sequencing 
(Brenner <em>et al</em>., 2000; Crawford <em>et al</em>., 2006). 
Only those clusters of multiple DNaseI library sequences that map within 500 
bases of each other are displayed.

<H2>Verification</H2>
<H3> DNase-Chip </H3>
<P>
A real-time PCR assay (McArthur <EM>et al.</EM>, 2001; Crawford <EM>et al.</EM>
, 2004) was used to validate a randomly selected subset
of DNase-chip regions.  
For the New DNase-chip, the 
Sensitivity of DNase-chip was determined to be &gt; 86% and 
Specificity to be &gt; 97%.
Approximately 20-30% of regions detected 
in only a single DNase concentration are valid. 
50-80% of regions detected in two out of three DNase concentrations are valid
(the exact percentage depends on which two DNase concentrations had
significant signal).
90% of regions detected in all three DNase concentrations are valid.
<P>
This data set includes elements for all 44 ENCODE regions.</P>

<H3> MPSS Sequencing </H3>
<P>
Real-time PCR was used to verify valid
DNaseI-hypersensitive sites.  Approximately 50% of
clusters of two sequences are valid.  These clusters are shown
in light gray. 80% of clusters of three sequences are valid, indicated by
dark gray. 100% of clusters of four or more
sequences are valid, shown in black.</P>
<P>
This data set includes confirmed elements for 35 of the 44
ENCODE regions.
It is estimated that these data identify
10-20% of all hypersensitive sites within CD4+ T cells. Further
sequencing will be required to identify additional sites.
MPSS data from the whole genome can be found in the Expression and Regulation
track group (NHGRI DNaseI-HS track).</P>

<H2>Credits</H2>
<P>
These data were produced at the 
<A HREF="http://www.genome.duke.edu/people/faculty/crawford/" TARGET=_BLANK>
Crawford Lab</A> 
at Duke University, and at the
<A HREF="http://www.genome.gov/10000351" TARGET=_blank>Collins Lab</A> 
at NHGRI.  Thanks to Gregory E. Crawford and Francis S. Collins for supplying 
the information for this track. H9 cells were grown in collaboration with Ron 
McKay and Paul Tesar at the National Institute of Neurological Disorders and
Stroke (NINDS)&mdash;an institute of the National Institutes of Health (NIH).
</P>

<H2> References </H2>
<P>
Brenner S, Johnson M, Bridgham J, Golda G, Lloyd DH, Johnson D, Luo S,
McCurdy S, Foy M, Ewan M <em>et al</em>.
<A HREF="http://www.nature.com/cgi-taf/DynaPage.taf?file=/nbt/journal/v18/n6/abs/nbt0600_630.html&dynoptions=doi1097883799" 
TARGET=_blank>Gene expression analysis by massively parallel signature 
sequencing (MPSS) on microbead arrays</A>.
<em>Nat Biotechnol</em>. 2000 Jun;18(6):630-4.
<P>
Crawford GE, Davis S, Scacheri PC, Renaud G, Halawi MJ, Erdos MR, Green R, 
Meltzer PS, Wolfsberg TG, Collins FS.
<A HREF="http://www.nature.com/nmeth/journal/v3/n7/abs/nmeth888.html" 
TARGET=_BLANK>
DNase-chip: A high resolution method to identify DNase I hypersensitive sites 
using tiled microarrays</A>. <em>Nature Methods</em>. 2006 Jul;3(7):503-9.
<P>
Crawford GE, Holt IE, Mullikin JC, Tai D, Blakesley R, Bouffard G, Young A, 
Masiello C, Green ED, Wolfsberg TG <em>et al</em>.
<A HREF="http://www.pubmedcentral.nih.gov/articlerender.fcgi?tool=pubmed&pubmedid=14732688" 
TARGET=_blank>Identifying gene regulatory elements by genome-wide recovery of 
DNase hypersensitive sites</A>. <em>Proc Natl Acad Sci USA</em>.
2004 Jan 27;101(4):992-7. 
<P>
Crawford GE, Holt IE, Whittle J, Webb BD, Tai D, Davis S, Margulies EH, Chen Y,
Bernat JA, Ginsburg D <em>et al</em>. 
<A HREF="http://www.genome.org/cgi/content/abstract/16/1/123?etoc" 
TARGET=_BLANK>Genome-wide mapping of DNase hypersensitive sites using 
massively parallel signature sequencing (MPSS)</A>. <em>Genome Res</em>.
2006 Jan;16(1):123-31.
(See also NHGRI's
<A HREF="http://research.nhgri.nih.gov/DNaseHS/May2005/"
TARGET=_blank>data site</A> for the project.)</P>
<P>
Klein E, Ben-Bassat H, Neumann H, Ralph P, Zeuthen J, Polliack A, Vanky F.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=789258&dopt=Abstract"
TARGET=_blank>
Properties of the K562 cell line, derived from a patient with chronic myeloid 
leukemia</A>. <em>Int J Cancer</em>. 1976 Oct 15;18(4):421-31.
<P>
McArthur M, Gerum S, Stamatoyannopoulos G.
<A HREF="http://www.sciencedirect.com/science?_ob=ArticleURL&_udi=B6WK7-457CXYS-4M&_user=4428&_coverDate=10%2F12%2F2001&_rdoc=1&_fmt=high&_orig=gateway&_origin=gateway&_sort=d&_docanchor=&view=c&_acct=C000059601&_version=1&_urlVersion=0&_userid=4428&md5=c7c150e23ce80bf6dc30126b0d66821d&searchtype=a"
TARGET=_blank>Quantification of DNaseI-sensitivity by real-time PCR: 
quantitative analysis of DNaseI-hypersensitivity of the mouse beta-globin LCR
</A>. <em>J Mol Biol</em>. 2001 Oct 12;313(1):27-34.
<P>
Puck TT, Marcus PI, Cieciura SJ. 
<A HREF="http://www.jem.org/cgi/content/abstract/103/2/273" TARGET=_blank>
Clonal growth of mammalian cells in vitro: growth characteristics of 
colonies from single HeLa cells with and without a &quot;feeder&quot; layer</A>. 
<em>J Exp Med</em>. 1956 Feb 1;103(2):273-83.
<P>
Scacheri PC, Crawford GE, Davis S.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=pubmed&cmd=Retrieve&dopt=AbstractPlus&list_uids=16939795&query_hl=3&itool=pubmed_docsum" 
TARGET=_BLANK>Statistics for ChIP-chip and DNase hypersensitivity experiments 
on NimbleGen arrays</A>.  <em>Methods Enzymol</em>. 2006;411:270-82.
<P>
Thomson JA, Itskovitz-Eldor J, Shapirom SS, Waknitz MA, Swiergiel JJ, 
Marshall VS, Jones JM. 
<A HREF="http://www.sciencemag.org/cgi/content/abstract/282/5391/1145" 
TARGET=_blank>Embryonic stem cell lines derived from human 
blastocysts</A>. <em>Science</em>. 1998 Nov 6;282(5391):1145-7.
