<H2>Overview</H2>
<P>
This super-track combines related tracks of ChIP-chip data generated by
the ENCODE group at the Sanger Institute.
ChIP-chip, also known as genome-wide location analysis, is a technique for
isolation and identification of DNA sequences bound by specific proteins in
cells, including histones. Histone methylation and acetylation serves as a
stable genomic imprint
that regulates gene expression and other epigenetic
phenomena.  These histones are found in transcriptionally active domains
called euchromatin.</P>
<P>
These tracks contain ChIP-chip data for H3 and H4 histones
in multiple cell lines, including HeLa (cervical carcinoma), 
GM06990 (lymphoblastoid), K562 (myeloid leukemia), and
HFL-1 (embryonic lung fibroblast).
Experiments were conducted with
antibodies to histones with different post-translational modification marks.
Data are displayed as signals as well as hits and peak
centers identified by hidden Markov model (HMM) analysis.</P>

<H2>Credits</H2>
<P>
The data were generated by the
<A HREF="http://www.sanger.ac.uk/PostGenomics/encode/investigators.shtml"
TARGET=_blank>ENCODE investigators</A> at the
<A HREF="http://www.sanger.ac.uk/" TARGET=_blank>Wellcome Trust Sanger
Institute</a>, Hinxton, UK.
Contacts:
<A HREF="mailto:&#105;&#100;l&#64;&#115;&#97;&#110;&#103;&#101;&#114;.a&#99;.&#117;k">Ian Dunham</A> and
<A HREF="mailto:&#99;&#109;k&#64;&#115;&#97;&#110;g&#101;&#114;.&#97;&#99;.u&#107;">Christoph Koch</A>. </P>
<P>
The HMM analysis was performed at the EBI by
<A HREF="mailto:&#102;l&#105;&#99;&#101;k&#64;&#101;&#98;&#105;.&#97;&#99;.u&#107;">Paul Flicek</A>.</P>
<P>
Raw data may be downloaded from the Sanger Institute website at
<A HREF="ftp://ftp.sanger.ac.uk/pub/encode" TARGET=_blank>ftp://ftp.sanger.ac.uk/pub/encode</A>.</P>
