<H2>Description</H2> 
<P> 
Linkage disequilibrium (LD) is the association of alleles on
chromosomes.  It measures the difference between the observed allele
frequency for a two-locus allele combination as compared to its expected
frequency, which is the product of the two single allele frequencies.
When LD is low, the two loci tend to be inherited in a nearly random
manner.</P>
<P>
This track shows three different measures of linkage disequilibrium
&mdash; D', r<sup>2</sup>, and LOD (log odds) &mdash; between
pairs of SNPs as genotyped by the HapMap consortium.  LD is useful for
understanding the associations between genetic variants throughout the
genome, and can be helpful in selecting SNPs for genotyping. </P>
<P>
By default, the display in full mode shows LOD values.  Each diagonal
represents a different SNP with each diamond representing a pairwise
comparison between two SNPs.  Shades are used to indicate linkage
disequilibrium between the pair of SNPs, with darker shades indicating
stronger LD.  For the LOD values, additional colors are used in some
cases:
<UL>
  <LI><I>White</I> diamonds indicate pairwise D' values less than 1
      with no statistically significant evidence of LD (LOD < 2).</LI>
  <LI><I>Light blue</I> diamonds indicate high D' values (>0.99) with
      low statistical significance (LOD < 2). </LI>
  <LI><I>Light pink</I> diamonds are drawn when the statistical
      significance is high (LOD >= 2) but the D' value is low (less
      than 0.5). </LI>
</UL>
</P>

<H2>Methods</H2>
<P>
Phased genotypes from HapMap Phase II release 22 were used with Haploview to
calculate LD values for all SNP pairs within 250 kb.  
The YRI and CEU tracks each use 30 parents+child trios (90 individuals)
and the combined JPT+CHB track uses 90 unrelated individuals.</P>
<P>
Haploview uses a two marker EM (ignoring missing data) to estimate the
maximum-likelihood values of the four gamete frequencies, from which
the D', LOD, and r<sup>2</sup> calculations derive.</P>

<H2>Display Conventions and Configuration</H2>
<P>
<UL>
 <LI>
  Display Mode<BR>
  <UL>
   <LI><i>Full</i> mode shows the pairwise LD values in a
        Haploview-style mountain plot. </LI>
   <LI><i>Dense</i> mode shows the pairwise LD values in a single line
       for each population, where the intensity at each position is
       the average of all of the LD values between the SNP at that
       position and all other SNPs within 250 kb.</LI>
  </UL>
 </LI>
 <LI>
  LD Values: measures of linkage disequilibrium<BR>
  <UL>
   <LI><i>r<sup>2</sup></i> displays the raw r<sup>2</sup> value, or
       the square of the correlation coefficient for a given marker
       pair.  SNPs that have not been separated by recombination have
       r<sup>2</sup> = 1; in this case, these two markers are said to
       be redundant for genotyping, but may have different functional
       effects.  Lower r<sup>2</sup> values show a lower degree of LD,
       indicating that some recombination has occurred in this
       population.  See Hill and Robertson (1966) for details.
       </LI>
   <LI><i>D'</i> displays the raw D' value, which is the normalized
       covariance for a given marker pair.  A D' value of 1 (complete
       LD) indicates that two SNPs have not been separated by
       recombination, while lower values indicate evidence of
       recombination in the history of the sample.  Only D' values
       near 1 are a reliable measure of LD; lower values are difficult
       to interpret as the magnitude of D' depends strongly on sample
       size.  See Lewontin (1988) for more details.</LI>
   <LI><i>LOD</i> displays the log odds score for linkage
       disequilibrium between a given marker pair, and is shown by
       default.</LI>
  </UL>
 </LI>
 <LI>
  Track Geometry<BR>
  <UL>
   <LI><i>Trim to triangle</i> shows the standard mountain plot
       (default); turning this option off will show LD values with
       SNPs outside the window.</LI>
   <LI><i>Invert</i>ing makes it easier to visually compare two
       adjacent populations.</LI>
  </UL>
 </LI>
 <LI>Colors<BR>
  <UL>
   <LI><i>LD Values</i> can be drawn in a variety of colors, with red
       as default.  The intensity of the color is proportional to the
       strength of the LD measure chosen above.  </LI>
   <LI><i>Outline</i>s can be drawn in contrasting colors or turned
       off.  Outlines are automatically suppressed when the window is
       larger than 100,000 bp.</LI>
  </UL>
 </LI>
 <LI>Population Selection<BR>
     The HapMap populations can be individually displayed or hidden. 
     <UL>
      <LI>YRI: Yoruba people in Ibadan, Nigeria (30 parent-and-adult-child trios)</LI>
      <LI>CEU: European samples from the Centre d'Etude du Polymorphisme Humain (CEPH) (30 trios)</LI>
      <LI>JPT+CHB: Combination of Japanese in Tokyo (45 unrelated individuals) and Han Chinese in Beijing (45 unrelated individuals)</LI>
     </UL>
  </LI>
</UL>
</P>

<H2>Credits</H2>
<P>
This track was created at UCSC using 
<A HREF="ftp://ftp.hapmap.org/pub/hapmap/public/phasing/2007-08_rel22/phased/" 
TARGET=_blank>data</A> 
from the 
<A HREF="http://www.hapmap.org/" TARGET=_blank>International HapMap Project</A>
and LD scores were computed using the 
<A HREF="http://www.broad.mit.edu/mpg/haploview/" TARGET=_blank>Haploview</A> 
program.
The genome browser track display was created by 
<A HREF="mailto:&#100;&#97;r&#121;&#108;&#64;&#115;&#111;e.
&#117;&#99;&#115;&#99;.
&#101;&#100;u">Daryl Thomas</A> following the display style from 
<A HREF="http://www.broad.mit.edu/mpg/haploview/" TARGET=_blank>Haploview</A>.
</P>

<H2>References</H2>

<H3>HapMap Project</H3>
<P>
The International HapMap Consortium.
<A HREF="http://www.nature.com/nature/journal/v449/n7164/abs/nature06258.html" 
TARGET=_blank>A second generation human haplotype map of over 3.1 million SNPs</A>.
<em>Nature.</em> 2007 Oct 18;449(7164):851-61.</P> 
<P>
The International HapMap Consortium.
<A HREF="http://www.nature.com/nature/journal/v437/n7063/abs/nature04226.html" 
TARGET=_blank>A haplotype map of the human genome</A>.
<em>Nature.</em> 2005 Oct 27;437(7063):1299-320.</P> 
<P>
The International HapMap Consortium.
<A HREF="http://www.nature.com/nature/journal/v426/n6968/abs/nature02168.html" 
TARGET=_blank> The International HapMap Project</A>.
<em>Nature.</em> 2003 Dec 18;426(6968):789-96.</P>

<H3>HapMap Data Coordination Center</H3>
<P>
Thorisson GA, Smith AV, Krishnan L, Stein LD. 
<A HREF="http://www.hapmap.org/downloads/presentations/Genome_research_art10.pdf" 
TARGET=_blank> The International HapMap Project Web site</A>.
<em>Genome Res.</em> 2005 Nov;15(11):1592-3.</P> 

<H3>Haploview</H3>
<P>
Barrett JC, Fry B, Maller J, Daly MJ. 
<A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/21/2/263" 
TARGET=_blank> Haploview: analysis and visualization of LD and haplotype 
maps</A>. 
<em>Bioinformatics.</em> 2005 Jan 15;21(2):263-5. Epub 2004 Aug 5.</P>

<H3>General references on Linkage Disequilibrium</H3>
<P>
Lewontin, RC.
<A HREF="http://www.genetics.org/cgi/content/abstract/120/3/849"
TARGET=_blank> On measures of gametic disequilibrium</A>.
<em>Genetics.</em> 1988 Nov;120(3):849-52.</P>
<P>
Hill WG, Robertson A.
<A HREF="http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&Cmd=ShowDetailView&TermToSearch=5980116&ordinalpos=6&itool=EntrezSystem2.PEntrez.Pubmed.Pubmed_ResultsPanel.Pubmed_RVDocSum" 
TARGET=_blank>The effect of linkage on limits to artificial selection</A>.
<em>Genet Res.</em> 1966 Dec;8(3):269-94.</P>
