<H2>Description</H2>
<P>
This track shows measures of evolutionary conservation 
generated using the 
<em>phyloP</em> (Phylogenetic P-Values) program from the  
<A HREF="http://compgen.bscb.cornell.edu/phast" target=_BLANK>
PHAST package</A>. 
Two measurements are provided: 
conservation across 28 species, and an alternative measurement 
restricted to the placental mammal subset (17 species plus human) 
of the multiple alignment.
</P>
<P>
PhyloP differs from phastCons &mdash; which is used to produce the 
scores in the main Conservation track &mdash; in several key ways.  The scores 
produced by phyloP reflect individual alignment columns, and do not take into 
account conservation at neighboring sites.  As a result, the phyloP 
conservation plot has a less smooth appearance, with more &quot;texture&quot; 
at individual bases, than the phastCons plot.  In addition, this property 
makes phyloP more appropriate than phastCons for evaluating signatures of 
selection at particular bases or classes of bases in the genome (e.g., all 
third codon positions).  In addition, phyloP requires fewer assumptions than 
phastCons, by depending only on a model of neutral evolution, rather than on 
models of both neutral evolution and negative selection (conservation).  
Finally, rather than representing probabilities of negative selection and 
ranging between 0 and 1, the phyloP scores represent -log p-values under a 
null hypothesis of neutral evolution, and range from 0 to infinity 
(although in practice there is a maximum achievable value for any particular 
data set).</P>
<P>
See the Conservation track description for information about
the multiple alignments used as the basis of these conservation measurements.
</P>

<H2>Display Conventions and Configuration</H2>
<P>
The track configuration options allow the user to display either
the vertebrate or placental mammal conservation scores, or both
simultaneously.
In full and pack display modes, conservation scores are displayed as a
wiggle track in which the height reflects the size of the score. 
The conservation wiggles can be configured in a variety of ways to 
highlight different aspects of the displayed information. 
For example, the <EM>windowing function</EM> option controls how
scores are combined across sites, with averaging as the default.
This will have a strong effect on how the plot appears when zoomed out.
<P>
Click the <A HREF="../goldenPath/help/hgWiggleTrackHelp.html" 
TARGET=_blank>Graph configuration help</A> link for an explanation 
of the configuration options.</P>

<H2>Methods</H2>
<P> 
Conservation scoring was performed using the <em>phyloP</em> program
from the PHAST package.
PhyloP is a general method for computing p-values of conservation 
by comparing estimated numbers of substitutions along the 
branches of a phylogeny with the distribution expected 
under neutral evolution (Siepel, Pollard, and Haussler, 2006).  
Here it was used to produce separate scores at each 
base (--wig-scores option), considering all branches of
the phylogeny rather than a particular subtree or 
lineage (i.e., --subtree was not used).  
Alignment gaps were treated as missing data. 
</P>
<P>
PhyloP relies on a tree model containing the tree topology,
branch lengths representing evolutionary distance at neutrally
evolving sites, the background distribution of nucleotides, and a substitution
rate matrix.  The 
<A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg18/phastCons28way/28way.mod"
TARGET=_blank>vertebrate tree model</A> for this track was
generated using the phyloFit program from the PHAST package  
(REV model, EM algorithm, medium precision) using multiple alignments of 
4-fold degenerate sites extracted from the 28way alignment
(msa_view).  The 4d sites were derived from the 
<A HREF="http://genome.imim.es/gencode" TARGET=_blank>
Oct 2005 Gencode Reference Gene set</A>,
which was filtered to select single-coverage long transcripts.  A second, 
<A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg18/phastCons28way/placental.mod" 
TARGET=_blank>
mammalian tree model</A> including only placental mammals was used
to generate the placental mammal conservation scoring.
</P>

<H2>Credits</H2>
<P> This track was created using phyloP, phyloFit, and other programs in PHAST by 
<A HREF="http://compgen.bscb.cornell.edu/~acs/software.html" 
TARGET=_blank>Adam Siepel's group</A> at Cornell University (original development
done at the Haussler lab at UCSC).
</P>
<P>The phylogenetic tree is based on Murphy <EM>et al</EM>. (2001) and general 
consensus in the vertebrate phylogeny community as of March 2007.
</P>

<H2>References</H2>
<P>
Siepel A, Pollard KS, Haussler D. 
<A href="http://compgen.bscb.cornell.edu/~acs/dless.pdf" TARGET=_blank>
New methods for detecting lineage-specific selection.</A> 
<em>Proc. 10th Int'l Conf. on Research in Computational Molecular Biolog 
(RECOMB '06)</em>. 2006.</P>
<P>
Murphy WJ, Eizirik E, O'Brien SJ, Madsen O, Scally M, Douady CJ, Teeling E,
Ryder OA, Stanhope MJ, de Jong WW, et al.
<A HREF="http://www.sciencemag.org/cgi/content/abstract/294/5550/2348"
TARGET=_blank>
Resolution of the early placental mammal radiation using Bayesian 
phylogenetics</A>. <EM>Science</EM>. 14 Dec 2001;294(5550):2348-51.
</P>
