<H2>Description</H2>

<P>This track is produced as part of the ENCODE Transcriptome Project. 
Transcription of different 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=rnaExtract">
RNA extracts</A> from different 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=localization">
sub-cellular localizations</A> in different 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=cellType">cell lines</A>
is compared in companion experiments using three different technologies: 
tiling arrays, RNA-seq using Solexa, and RNA-seq using SOLiD.  The
tiling array data are shown in this track. 
The <i>Raw Signal</i> view shows an estimate of abundance of RNA molecules 
and the <i>Transfrags</i> view 
shows the locations of sites corresponding to these molecules. 
</P>

<H2>Display Conventions and Configuration</H2>

<P>To show only selected subtracks, uncheck the boxes next to the tracks that 
you wish to hide.</P>
 
<P>Color differences among the views are arbitrary. They provide a 
visual cue for 
distinguishing between the different cell types and compartments.</P>

<DL>
<DT><i>Transfrags</i>
<DD>The Transfrags view includes all transfrags before filtering.
<DT><i>Filtered Transfrags</i>
<DD>The Filtered Transfrags view excludes repeats and other known annotations 
including: 
tRNAs and rRNAs, mi/snoRNAs, things mapping to the mitochondrial or Y 
chromosomes, 
and many predicted snoRNAs and miRNAs.
<DT><i>Raw Signal</i>
<DD>The Raw Signal view shows the probe intensity based on all probes.
<DT><i>Filtered Signal</i>
<DD>The Filtered Signal view shows the probe intensity based on the 
filtered set of probes.
</DL>
Regions with negative signal value are areas
where the mismatch probe hybridizes better than the match probe.
This could indicate presence of a SNP, a site of RNA editing, or 
sequencing error.

<H2>Methods</H2>

<P>
Cells were grown according to the approved 
<A HREF="/ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture protocols</A>.
RNA molecules longer than 200 nt 
and present in RNA population isolated from different subcellular compartments 
(such as cytosol, nucleus, polysomes and others) were fractionated into polyA+ 
and polyA- fractions  as described in
<A HREF="/ENCODE/protocols/general/rnaExtracts.txt" TARGET=_BLANK>these
protocols</A>.
Each RNA fraction was converted into double-stranded cDNA using 
random hexamers, 
labeled and hybridized to a tiling 91-array set containing probes against the 
non-repetitive 
portion of the human genome tiled on average every 5 bp (center-to-center of 
each consecutive 25-mers). </P>

<P>All arrays were scaled to a median array intensity of 330. Within a sliding 
61 bp window 
centered on each probe, an estimate of RNA abundance (signal; see the 
<i>Raw Signal</i> view) 
was found by calculating the median of all pairwise average PM-MM values, 
where PM is a 
perfect match and MM is a mismatch. Kapranov <EM>et al.</EM> (2002), Cheng 
<EM>et al.</EM> (2005) , Kapranov <EM>et al.</EM> (2007), and Cawley 
<EM>et al.</EM> (2004) 
are good references for the experimental methods. Cawley <EM>et al.</EM> 
also describes the analytical methods.

<H2>Verification</H2>

<P>The reproducibility of the labeling method was assessed separately. Three 
independent 
technical replicates were generated from the same RNA pool for each RNA 
preparation and 
hybridized to duplicate arrays (two technical replicates) that contain the 
ENCODE regions. 
Labeled RNA samples were then pooled and hybridized to the tiling 91-array 
set spanning 
the whole genome. Transcribed regions (transfrags; see the <i>Transfrags</i> 
view) were 
generated from the <i>Raw Signal</i> by merging genomic positions to which 
probes 
are mapped. This merging was based on a 5% false positive rate cutoff in 
negative 
bacterial controls, a maximum gap (MaxGap) of 40 base-pairs and minimum run 
(MinRun) 
of 40 base-pairs. </P>

<H2>Credits</H2>

<P>These data were generated and analyzed by the transcriptome group at 
<A TARGET=_BLANK HREF="http://www.affymetrix.com">Affymetrix</A>
and Cold Spring Harbor Laboratories: 
P. Kapranov, I. Bell, E. Dumais, 
J. Drenkow, J. Dumais, N. Garg, M. Lubinsky,
Carrie A. Davis, Huaien Wang, Kimberly Bell, Jorg Drenkow, Chris Zaleski,
and Thomas R. Gingeras.
<BR>

<P>Contact: 
<A HREF="mailto:&#103;&#105;&#110;&#103;&#101;r&#97;&#115;&#64;&#99;&#115;&#104;&#108;.
&#101;&#100;u">
&#103;&#105;&#110;&#103;&#101;r&#97;&#115;&#64;&#99;&#115;&#104;&#108;.
&#101;&#100;u</A> (Tom Gingeras)</P>
<!-- above address is gingeras at cshl.edu -->

<H2>References</H2>

<P>Cawley S, Bekiranov S, Ng HH, Kapranov P, Sekinger EA, Kampa D, 
Piccolboni A, Sementchenko V, Cheng J, Williams AJ <EM>et al</EM>. 
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=14980218">Unbiased mapping of transcription factor binding sites along human chromosomes 21 and 22 points to widespread regulation of noncoding RNAs</A>. 
<EM>Cell</EM>. 2004 Feb 20;116(4):499-509.</P>

<P>Cheng J, Kapranov P, Drenkow J, Dike S, Brubaker S, Patel S, Long J, 
Stern D, Tammana H, Helt G <EM>et al</EM>.
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/pubmed/15790807?ordinalpos=19&itool=EntrezSystem2.PEntrez.Pubmed.Pubmed_ResultsPanel.Pubmed_RVDocSum">Transcriptional Maps of 10 Human Chromosomes at 5 Nucleotide Resolution</A>. 
<EM>Science</EM>. 2005 May 20;308(5725):1149-1154.</P>

<P>Kapranov P, Cawley SE, Drenkow J, Bekiranov S, Strausberg RL, Fodor SP, 
Gingeras TR.
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=11988577">Large-scale transcriptional activity in chromosomes 21 and 22</A>. 
<EM>Science</EM>. 2002 May 3;296(5569):916-9.</P>

<P>Kapranov P, Cheng J, Dike S, Nix DA, Duttagupta R, Willingham AT, 
Stadler PF, Hertel J, Hackermüller J, Hofacker IL, <EM>et al</EM>.
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&Cmd=ShowDetailView&TermToSearch=17510325&ordinalpos=1&itool=EntrezSystem2.PEntrez.Pubmed.Pubmed_ResultsPanel.Pubmed_RVDocSum">RNA maps reveal new RNA classes and a possible function for pervasive transcription</A>. 
<EM>Science</EM>. 2007 Jun 8;316(5830):1484-8.</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior 
consent, submit publications that use an unpublished ENCODE dataset until 
nine months following the release of the dataset.  This date is listed in 
the <EM>Restricted Until</EM> column, above.  The full data release policy 
for ENCODE is available 
<A HREF="/ENCODE/terms.html" TARGET=_BLANK>here</A>.</P> 
