<h2>Description</h2>
<p>
This track shows average methylation status in CpG islands. In general,
methylation of CpG sites within a promoter causes silencing of the gene
associated with that promoter.

<H2> Release Notes</H2>
This is release 2 of this track.  Release 2 adds tables for several new cell types: GM12891, GM12892, H1-hESC, HeLa-S3, and HepG2.

<h2>Track Conventions</h2>
<p>
Methylation status is color-coded as:
<ul>
<li><font color="#FF7F00">orange</font> = methylated (bed score = 1000)</li>
<li><font color="#0000FF">blue</font> = non-methylated (bed score = 0)</li>
</ul>
</p>

<a name="Methods"></a><h2>Methods</h2>
<p>
CpG regions were assayed via Methyl-seq, a method developed in the Myers
laboratory to measure the methylation status at CpGs throughout the genome. It
combines DNA digestion by a methyl-sensitive enzyme HpaII and its
methyl-insensitive isoschizomer MspI with the Illumina DNA sequencing
platform.  The method was first applied in a collaboration with the laboratory
of Dr. Julie Baker at Stanford University to study methylation and gene
expression changes that occur in human embryonic stem cells before and after
differentiation to definitive endoderm.  A paper describing the results as
well as the method has been submitted for publication [1].
</p>
<p>
This study profiled genomic DNA and mRNA samples derived from two human
embryonic stem cell lines: H9 and BG02.  These cells were differentiated into
definitive endoderm, embryoid bodies, embryoid body-derived cells, and AFP+
(alpha-fetoprotein positive) hepatocytes.  These in vitro samples were profiled 
with Methyl-seq and compared them with normal tissue samples from 11-week and
24-week fetal liver and adult liver.
</p>
<p>
Methyl-seq assays more than 250,000 methyl-sensitive restriction enzyme
cleavage sites, representing more than 90,000 genomic regions.  These regions
include 35,528 annotated CpG islands, while the remaining 55,084 non-CpG
island regions are distributed across the genome in promoters, genes, and
intergenic regions. Sequence tags present in MspI libraries but not in HpaII
libraries are derived from methylated regions.  Conversely, sequence tags that
occur in HpaII libraries come from at least partially unmethylated regions.
</p>
<h3>In vitro differentiation</h3>
<p>
Definitive endoderm precursor cells were generated from H9 hES cells by treating
them with activin A.  Embryoid bodies (EBs) were generated by growing
undifferentitated H9 and BG02 hESCs in suspension. EB-derived
cells were obtained by plating clumps of the cells from the EBs.  
AFP+ fetal
hepatocytes were derived from EBs by plating EB cells with FgF, followed by fluorescence
activated cell sorting (FACS) to isolate cells expressing the green
fluorescent protein (GFP) reporter gene driven from the AFP promoter.
</p>
<h3>Isolation of genomic DNA</h3>
<p>
Genomic DNA is isolated from biological replicates of each cell line by using
the QIAGEN DNeasy Blood & Tissue Kit according to the instructions provided by
the manufacturer. DNA concentrations and a level of quality of each
preparation is determined by UV absorbance.
</p>
<h3>HpaII and MspI digestions</h3>
<p>
Cleavage of DNA by restriction endonuclease HpaII is prevented by the presence
of a 5-methyl group at the internal C residue of its recognition sequence
CCGG.  MspI, an isoschizomer of HpaII, cleaves DNA irrespective of the
presence of a methyl group at this position.
<p>
For the MspI library, 5 &mu;g genomic DNA was digested in a 100 &mu;l reaction
with 1X NEB Buffer2 and 20 units MspI restriction enzyme and incubated for 18 hr
at 37&deg;C.  For the HpaII library, 5 &mu;g genomic DNA was digested in a
100 &mu;l reaction with 1X NEB Buffer1 and 20 units HpaII restriction enzyme
and incubated for 18 hr at 37&deg;C.
<p>
Note that in subsequent versions of the Methyl-seq protocol, which will be
described later, much lower amounts of genomic DNA were used (1 &mu;g and
potentially lower).
</p>
<h3>DNA library construction and sequencing</h3>
<p>
High-throughput sequencing libraries were generated from DNA fragments of the
HpaII or MspI digested genomic DNA according to the protocol posted at the
Myers' lab <a href="http://hudsonalpha.org/myers-lab/protocols"
TARGET="_blank">protocols</a> page.
This approach was recently modified by removing the first PCR amplification
step, just prior to the gel electrophoresis size-selection step, which was
found to reduce a fragment-size bias in the sequencing libraries.  
These libraries were sequenced with an Illumina Genome Analyzer (GA2) according to the
manufacturer's recommendations.
</p>
<h3>Data analysis</h3> 
<p>
For this analyis, reads that align to human genome sequence version hg18 and contain the
5'-CGG-3' HpaII-cut signature on their 5' end were used.  
These aligned sequence reads were mapped to CCGG sites predicted in silico on hg18.  
Sites with four or more Msp1 tags occurring in either the forward or reverse
direction were retained for analysis.  These "assayable" sites were then grouped with
neighboring sites that are within 35-75 bp of each other.  Thus, a "region"
can be comprised of between 2 and 18 digestion sites that are each within
35-75 bp of another site.  Methylated and non-methylated calls were made by
using HpaII tag data from all assayable cut sites.  
For each site across each region, the larger of
either the forward read count or reverse read count was used.
Regions that have an average of 0 or 1 read per cut site are called
methylated, and regions with more than one sequence read per site are called
unmethylated.
</p>
<a name="Credits"></a><h2>Credits</h2>
<p>
Dr. Richard M. Myers<br>
Mr. Yuya Kobayashi: <A HREF="mailto:&#121;&#117;y&#97;&#107;&#64;&#115;t&#97;&#110;&#102;&#111;&#114;&#100;.
ed&#117;">
&#121;&#117;y&#97;&#107;&#64;&#115;t&#97;&#110;&#102;&#111;&#114;&#100;.
ed&#117;</A> <br>
Dr. Devin M. Absher: <A HREF="mailto:&#100;&#97;b&#115;h&#101;&#114;&#64;&#104;u&#100;&#115;&#111;&#110;&#97;&#108;&#112;&#104;&#97;.
&#111;&#114;g">
&#100;&#97;b&#115;h&#101;&#114;&#64;&#104;u&#100;&#115;&#111;&#110;&#97;&#108;&#112;&#104;&#97;.
&#111;&#114;g</A><br />
Dr. Rebekka O. Sprouse:
<A HREF="mailto:&#114;&#115;&#112;&#114;&#111;&#117;&#115;e&#64;&#104;&#117;d&#115;&#111;&#110;&#97;&#108;&#112;&#104;&#97;.
o&#114;g">
&#114;&#115;&#112;&#114;&#111;&#117;&#115;e&#64;&#104;&#117;d&#115;&#111;&#110;&#97;&#108;&#112;&#104;&#97;.
o&#114;g</A>
<p>
Contact:
<A HREF="mailto:&#102;p&#97;&#117;&#108;&#105;&#64;&#104;&#117;&#100;&#115;&#111;&#110;&#97;&#108;&#112;&#104;a.
&#111;r&#103;">Flo Pauli</A>.
<!-- above address is fpauli at hudsonalpha.org (encodeEmail.pl) -->
<br />
</p>
<a name="References"></a><h2>References</h2>
<p>
<p>1.
Brunner AL, Johnson DS, Kim SW, Valouev A, Reddy TE, Neff NF, Anton E, Medina C, Nguyen L, 
Chiao E <i>et al.</i>
<a href="http://genome.cshlp.org/content/early/2009/03/09/gr.088773.108.long" target="_blank">
Distinct DNA methylation patterns characterize differentiated human embryonic
stem cells and developing human fetal liver</a>.
<em>Genome Research</em>. 2009 Jun;19(6):1044-56. Epub 2009 Mar 9.
</p>
<h2> Data Release Policy </h2>

<p>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column on the track configuration page and
the download page.  The full data release policy for ENCODE is available
<a href="../ENCODE/terms.html" TARGET=_BLANK>here</a>.</p>

