<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <title>wgEncodeDNAseSuper.html</title>
</head>
<body>
<h2>Description</h2>
<p>These tracks display evidence of open, accessible chromatin in&nbsp;
<a href="http://www.genome.gov/10005107">ENCODE</a> cell types. Open
chromatin was identified using complementary methods including: DNaseI
hypersensitivity (HS), Formaldehyde-Assisted Isolation of Regulatory
Elements (FAIRE), and chromatin immunoprecipitation (ChIP) for select
regulatory factors. <br>
<br>
DNaseI HS data: DNaseI is an enzyme that has long been used to map
general chromatin accessibility, and DNaseI "hyperaccessibility" or
"hypersensitivity" is a feature of active cis-regulatory sequences. The
use of this method has led to the discovery of functional regulatory
elements that include enhancers, silencers, insulators, promotors,
locus control regions and novel elements. DNaseI hypersensitivity
signifies chromatin accessibility following binding of trans-acting
factors in place of a canonical nucleosome.<br>
<br>
FAIRE data: FAIRE (Formaldehyde Assisted Isolation of Regulatory
Elements) is a method to isolate and identify nucleosome-depleted
regions of the genome. FAIRE was initially discovered in yeast and
subsequently shown to identify active regulatory elements in human
cells (Giresi et al., 2007). Although less well-characterized than
DNase, FAIRE also appears to identify functional regulatory elements
that include enhancers, silencers, insulators, promotors, locus control
regions and novel elements. <br>
<br>
ChIP data: ChIP (Chromatin Immunoprecipitation) is a method to identify
the specific location of proteins that are directly or indirectly bound
to genomic DNA. By identifying the binding location of
sequence-specific transcription factors, general transcription
machinery components, and chromatin factors, ChIP can help in the
functional annotation of the open chromatin regions identified by
DNaseI HS mapping and FAIRE.<br>
</p>
<h2>Display Conventions</h2>
These tracks are multi-view composite tracks that contains multiple
data types (views). For each view, there are multiple subtracks that
display individually on the browser.&nbsp; Each view within each track
has separate display controls, as described <a
href="/goldenPath/help/multiView.html">here</a>.&nbsp;
Specific
aspects
of
the views vary by track.&nbsp;&nbsp; The views
include:<br>
<br>
<div style="margin-left: 40px;"><span style="font-style: italic;">Peaks</span><br>
<div style="margin-left: 40px;">Regions of significant DNaseI
hypersensitivity.<br>
</div>
<span style="font-style: italic;">Signal</span><br>
<div style="margin-left: 40px;">The density of tags mapping within a
sliding window.<br>
</div>
<span style="font-style: italic;">Raw Signal</span><br>
<div style="margin-left: 40px;">Density graph (wiggle) of signal
enrichment based on aligned read density.<br>
</div>
</div>
<h2>Credits<br>
</h2>
These data were generated and analyzed as part of the <a
 href="http://www.genome.gov/10005107">ENCODE</a> project, a
genome-wide consortium project with the aim of cataloging all
functional elements in the human genome.&nbsp; This effort includes
collecting a variety of data over a specific set of <a
href="/ENCODE/cellTypes.html">cell types</a>.&nbsp;
Consequently, data related to these tracks be available under ENCODE
tracks. &nbsp;&nbsp;
<br>
<h2>References</h2>
Ho L, Crabtree GR. <a
 href="http://www.ncbi.nlm.nih.gov/pubmed/20110991"><span
 style="font-style: italic;">Chromatin remodelling during development.</span></a>
Nature. 2010 Jan 28;463(7280):474-84.<br>
<br>
Geiman TM, Robertson KD.&nbsp; <a
 href="http://www.ncbi.nlm.nih.gov/pubmed/12244565"><span
 style="font-style: italic;">Chromatin remodeling, histone
modifications, and DNA methylation-how does it all fit together?</span></a>&nbsp;
J
Cell
Biochem.
2002;87(2):117-25.<br>
<br>
<h2>Data Release Policy</h2>
Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset
until nine months following the release of the dataset. This date is
listed in the Restricted Until column on the track configuration page
and the download page. The full data release policy for ENCODE is
available <a href="/ENCODE/terms.html">here</a>.<br>
<br>
</body>
</html>
