<h2> Description </h2>
<p>These
tracks display the methylation status of specific CpG dinucleotides in the
given <A HREF="http://hgwdev.cse.ucsc.edu/cgi-bin/hgEncodeVocab?type=cellType" 
TARGET=_blank>cell types</A> as identified by the
<a target='_blank' href="http://www.illumina.com/pages.ilmn?ID=243">Illumina 
Infinium HumanMethylation27 BeadArray platform</a>. In general, methylation of 
CpG sites within a promoter causes silencing of the gene associated with that promoter. 
</p>
<p>
This method was used to validate Methyl-seq data generated in collaboration with the
laboratory of Dr. Julie Baker at Stanford University to study methylation and
gene expression changes that occur in human embryonic stem cells before and
after differentiation to definitive endoderm [1].
Based on the results of these experiments, cut-off beta
values to call "methylated" and "unmethylated" are selected.
</p>
<p>Analysis is performed for two replicates for each cell line. Detailed
information for the CpG targets is in an XLS formatted spreadsheet on the Myers' lab  
<a target='_blank' href="http://hudsonalpha.org/myers-lab/protocols">protocols</a> website.
</a>
</p>
<h2>Display Conventions</h2>
<p>
Scores associated with each site are beta value multiplied by 1000. Methylation
status is color-coded as:
<ul>
<li><font color="#FF7f00">orange</font> = methylated (score >= 600) </li>
<li><font color="#800080">purple</font> = partially methylated (200 < score < 600)</li>
<li><font color="#0000CD">bright blue</font> = unmethylated (0 < score <= 200) </li>
<li>black = NA (score = 0)</li>
</ul>
</p>
<p>
</p>

<a name="Methods"></a><h2> Methods </h2>

<P>
Cells were grown according to the approved
<A HREF="../../ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture protocols</A>.</P>

<p>Genomic
DNA was isolated from biological replicates of each cell line by using the
QIAGEN DNeasy Blood &amp; Tissue Kit according to the instructions provided by
the manufacturer. DNA concentrations and a level of quality of each preparation
 was determined by UV absorbance. 
</p>
<p>
The Methyl27K platform uses bisulfite treated genomic DNA to assay the 
methylation status of 27,578 CpG sites within more than 14,000 genes. Genomic 
DNA treated with sodium bisulfite converts unmethylated cytosine of a CpG dinucleotides
into uracil; methylated cytosines do not get converted.
After bisulfite treatment, the methylation status of a site is assayed by 
single base-pair extension with a Cy3 or Cy5 labeled nucleotide on oligo-beads 
specific for the methylated or unmethylated state. A beta value is calculated by
Illumina's Bead Studio software for each CpG target. This value represents the 
intensity value from the methylated bead type divided by the sum of the 
intensity values from the methylated and unmethylated bead types for any given 
CpG target.
</p>
<p>Bisulfite conversion reaction was done using the
<a target='_blank' href="http://www.zymoresearch.com/protocols/dna/%5BD5003%5D%20EZ-96%20DNA%20methylation%20instruction%20v1.2.pdf">Zymo Research EZ-96 DNA Methylation<sup>TM</sup> Kit</a>.
One step of the protocol was modified. During the incubation,
a 30 sec 95<sup>o</sup>C denaturing step every hour was included to increase reaction efficiency as 
recommended by the Illumina Infinium Human Methylation27 protocol. 
</p>
<p>The bead arrays were run according to the 
<a target='_blank' href="http://www.illumina.com/pagesnrn.ilmn?ID=275">protocol</a> provided by Illumina.
</p>
<p>The intensity data from the BeadArray was processed using Illumina's 
BeadStudio software with the Methylation Module v3.2. The data was then 
quality-filtered using p-values.  
</p>
<p>
Any beta value equal to or greater than 0.6 is considered fully methylated. Any
beta value equal to or less than 0.2 is considered to be fully unmethylated. 
Beta values between 0.2 and 0.6 are considered to be partially methylated.
Beta-values are quality filtered and spots that fall below the minimum intensity threshold are displayed as "NA".
</p>
<a name="Credits"></a><h2> Credits </h2>
<p>
Dr. Richard M. Myers<br>
Mr. Yuya Kobayashi: <A HREF="mailto:&#121;&#117;y&#97;&#107;&#64;&#115;t&#97;&#110;&#102;&#111;&#114;&#100;.
ed&#117;">
&#121;&#117;y&#97;&#107;&#64;&#115;t&#97;&#110;&#102;&#111;&#114;&#100;.
ed&#117;</A> <br>
Dr. Devin M. Absher: <A HREF="mailto:&#100;&#97;b&#115;h&#101;&#114;&#64;&#104;u&#100;&#115;&#111;&#110;&#97;&#108;&#112;&#104;&#97;.
&#111;&#114;g">
&#100;&#97;b&#115;h&#101;&#114;&#64;&#104;u&#100;&#115;&#111;&#110;&#97;&#108;&#112;&#104;&#97;.
&#111;&#114;g</A><br />
Dr. Rebekka O. Sprouse:
<A HREF="mailto:&#114;&#115;&#112;&#114;&#111;&#117;&#115;e&#64;&#104;&#117;d&#115;&#111;&#110;&#97;&#108;&#112;&#104;&#97;.
o&#114;g">
&#114;&#115;&#112;&#114;&#111;&#117;&#115;e&#64;&#104;&#117;d&#115;&#111;&#110;&#97;&#108;&#112;&#104;&#97;.
o&#114;g</A>
<P>
Contact:
<A HREF="mailto:&#102;p&#97;&#117;&#108;&#105;&#64;&#104;&#117;&#100;&#115;&#111;&#110;&#97;&#108;&#112;&#104;a.
&#111;r&#103;">Flo Pauli</A>.
<!-- above address is fpauli at hudsonalpha.org (encodeEmail.pl) -->

<br />
</p>
<a name="References"></a><h2>References</h2>
<p>1.
Brunner AL, Johnson DS, Kim SW, Valouev A, Reddy TE, Neff NF, Anton E, Medina C, Nguyen L, 
Chiao E <i>et al.</i>
<a target='_blank' href='http://genome.cshlp.org/content/early/2009/03/09/gr.088773.108.long'>
Distinct DNA methylation patterns characterize differentiated human embryonic
stem cells and developing human fetal liver</a>.
<EM>Genome Research</EM>. 2009 Jun;19(6):1044-56.
</p>
<h2> Data Release Policy </h2>

<p>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column on the track configuration page and
the download page.  The full data release policy for ENCODE is available
<a href="../ENCODE/terms.html" TARGET=_BLANK>here</a>.</p>

