<H2>Description</H2>
<P>
These tracks display evidence of open chromatin in <A TARGET=_BLANK
HREF="/cgi-bin/hgEncodeVocab?type=cellType">multiple cell types</A>
from the Duke/UNC/UT-Austin/EBI
ENCODE group. Open chromatin was identified using two independent and
complementary methods: DNaseI hypersensitivity (HS) and Formaldehyde-Assisted
Isolation of Regulatory Elements (FAIRE), combined with chromatin
immunoprecipitation (ChIP) for select regulatory factors. Each method was
verified by two detection platforms: Illumina (formerly Solexa) sequencing by
synthesis, and high-resolution 1% ENCODE tiled microarrays supplied by
NimbleGen.
</P><P>
<B>DNaseI HS data</B>: DNaseI is an enzyme that has long been used to map
general chromatin accessibility, and DNaseI &quot;hyperaccessibility&quot; or
&quot;hypersensitivity&quot; is a feature of active cis-regulatory sequences. The use of
this method has led to the discovery of functional regulatory elements that
include enhancers, silencers, insulators, promotors, locus control regions and
novel elements. DNaseI hypersensitivity signifies chromatin accessibility
following binding of trans-acting factors in place of a canonical nucleosome.
</P><P>
<B>FAIRE data</B>: FAIRE (Formaldehyde Assisted Isolation of Regulatory
Elements) is a method to isolate and identify nucleosome-depleted regions of
the genome.  FAIRE was initially discovered in yeast and subsequently shown to
identify active regulatory elements in human cells (Giresi <em>et al.</em>,
2007). Although less well-characterized than DNase, FAIRE also appears to
identify functional regulatory elements that include enhancers, silencers,
insulators, promotors, locus control regions and novel elements. DNA fragments
isolated by FAIRE are 100-200 bp in length, with the average length being 140
bp.
</P><P>
<B>ChIP data</B>: ChIP (Chromatin Immunoprecipitation) is a method to identify
the specific location of proteins that are directly or indirectly bound to
genomic DNA. By identifying the binding location of sequence-specific
transcription factors, general transcription machinery components, and chromatin
factors, ChIP can help in the functional annotation of the open chromatin
regions identified by DNaseI HS mapping and FAIRE.
</P>

<H2> Display Conventions and Configuration </H2>
<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>). For each view, there are multiple subtracks that display
individually on the browser. Instructions for configuring multi-view tracks
are <A HREF="/goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
Chromatin data displayed here represents a continuum of signal intensities.
The Crawford lab recommends setting the &quot;Data view scaling:
<EM>auto-scale</EM>&quot; option when viewing signal data in <EM>full</EM>
mode. In general, for each experiment in each of the cell types, the Open
Chromatin tracks contain the following views:
<DL>
<DT><I>Peaks</I></DT><DD> Regions of enriched signal in either
DNaseI HS, FAIRE, or ChIP experiments.  Peaks were called based on signals
created using F-Seq, a software program developed at Duke (Boyle <em>et al.</em>,
2008b).  Significant regions were determined by performing ROC analysis of
sequence data using data from the 1% ENCODE arrays, and determining a cut-off
value at approximately the 95% sensitivity level.  The solid vertical line in
the peak represents the point with highest signal.   ENCODE Peaks tables contain
a p-value for statistical significance.  For these data, this was determined by
fitting the data to a gamma distribution.</DD>
<DT><I>Signal (F-Seq Density)</I></DT><DD> Density graph (wiggle) of signal
enrichment calculated using F-Seq for the combined set of sequences from all
replicates. F-Seq employs Parzen kernel density estimation to create base pair
scores (Boyle <em>et al.</em>, 2008b). This method does not look at fixed-length
windows but rather weights contributions of nearby sequences in proportion to
their distance from that base.  It only considers sequences aligned 4 or less
times in the genome, and uses an alignability background model to try to correct
for regions where sequences cannot be aligned.  For the K562, HepG2 and HelaS3
cell types, where there is an abnormal karyotype, a model to try to correct
for amplifications and deletions was also used. No control data were used in the
creation of these annotations.</DD>
<DT><I>Signal (Base Overlap)</I></DT><DD> An alternative version of the
<I>Signal (F-Seq Density)</I> track annotation that provides a higher resolution
view of the raw sequence data. This track also includes the combined set of
sequences from all replicates.  For each sequence, the aligned read is extended
in the following way: for DNase, the read is extended 5 bp in both directions
from its 5' aligned end where DNase cut the DNA; for FAIRE and ChIP, the
sequence is extend to a fragment length of 134 bp from the 5' aligned end
representing the approximate average fragment length.  The score at each base
pair represents the number of extended fragments that overlap the base pair.
</DD>
<DT><I>Alignments</I></DT><DD> Mappings of short reads to the genome (currently
only available for <A TARGET=_BLANK
HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg18/encodeDCC/wgEncodeChromatinMap/">
download</A>).</DD>
</DL>
Additional data that were used to generate these tracks
are located in the ENCODE <A HREF="hgTrackUi?g=wgEncodeMapability">Mappability</A> track:
<DL>
<DT><I>Uniqueness</I></DT><DD>
The <EM>Duke uniqueness</EM> tracks were used in
identify regions of unique sequence for different tag lengths. The tracks
also identify regions where high-throughput sequence tags cannot be mapped.</DD>
<DT><I>Excluded Regions</I></DT><DD>The <EM>Duke excluded regions</EM> track
was used to identify problematic regions for short sequence tag signal
detection (such as satellites and rRNA genes). These regions of the genome were
excluded from the Open Chromatin tracks.</DD>
</DL>
</P>

<H2>Methods</H2>
<P>
Cells were grown according to the approved <A HREF="/ENCODE/protocols/cell"
TARGET=_BLANK>ENCODE cell culture protocols</A>.
</P>
<P>
<em>DNaseI hypersensitive sites</em> were isolated using methods called DNase-seq or
DNase-chip (Boyle <em>et al.</em>, 2008a, Crawford <em>et al.</em>, 2006).
Briefly, cells were lysed with NP40, and intact nuclei were digested with optimal
levels of DNaseI enzyme. DNaseI digested ends were captured from three different
DNase concentrations, and material was sequenced using Illumina (Solexa)
sequencing. DNase-seq data were verified using material that was hybridized to
NimbleGen Human ENCODE tiling arrays (1% of the genome).  Multiple independent
growths (replicates) were compared to verify the reproducibility of the data.
A more detailed protocol is available
<A HREF="/ENCODE/protocols/general/Duke_DNase_protocol.pdf">here</A>.</P>
</P><P>
<em>FAIRE</em> was performed (Giresi <em>et al.</em>, 2007) by cross-linking proteins
to DNA using 1% formaldehyde solution, and the complex was sheared using
sonication.  Phenol/chloroform extractions were performed to remove DNA
fragments cross-linked to protein. The DNA recovered in the aqueous phase was
hybridized to NimbleGen Human ENCODE tiling arrays (1% of the genome) and sequenced
using a Solexa sequencing system.  The ENCODE array data were used to verify
the accuracy of the sequencing data, and multiple independent growths
(replicates) were compared to assess the reproducibility of the data.
A more detailed protocol is available
<A HREF="/ENCODE/protocols/general/FAIRE_UNC_procedure.pdf">here</A>.
Also see Giresi <em>et al.</em>, 2009.</P>
</P><P>
To perform <em>ChIP</em>, proteins were cross-linked to DNA <em>in vivo</em>
using 1% formaldehyde solution (Bhinge <em>et al.</em>, 2007, ENCODE Project
Consortium., 2007). Cross-linked chromatin was sheared by sonication and
immunoprecipitated using a specific antibody against the protein of interest.
After reversal of the cross-links, the immunoprecipitated DNA was used to
identify the genomic location of transcription factor binding. This was
accomplished by Solexa sequencing of the ends of the immunoprecipitated DNA
(ChIP-seq), as well as labeling and hybridization of the immunoprecipitated
DNA to NimbleGen Human ENCODE tiling arrays (1% of the genome) along with the
input DNA as reference (ChIP-chip). The ENCODE array data were used to verify
the accuracy of the sequencing data, and multiple independent growths
(replicates) were compared to assess the reproducibility of the data. A more
detailed protocol is available
<A HREF="/ENCODE/protocols/general/UTA_ChIPseq_protocol.pdf">here</A>.</P>
</P><P>
ENCODE Array data were normalized using the Tukey biweight normalization, and
peaks were called using ChIPOTle (Buck, <em>et al.</em>, 2005) at multiple
levels of significance.  Regions matched on size to these peaks that were
devoid of any significant signal were also created to allow for ROC analysis.
</P><P>
Sequences from each experiment were aligned to the genome using Maq (Li
<em>et al.</em>, 2008) and those that aligned to 4 or fewer locations were retained.
Other sequences were also filtered based on their alignment to problematic
regions (such as satellites and rRNA genes).  The resulting digital signal was
converted to a continuous wiggle track using F-Seq that employs Parzen kernel
density estimation to create base pair scores (Boyle <em>et al.</em>, 2008b).
Discrete DNase HS, FAIRE, and ChIP sites (peaks) were identified from
DNase/FAIRE/ChIP-seq using F-Seq by setting a Parzen cutoff based on ROC curve
analysis using peaks and non-peaks identified from DNase/FAIRE/ChIP-chip using
NimbleGen Human ENCODE tiling arrays (1% of the genome).
</P><P>
<em>Input</em> data was generated for GM12878, K562, HeLa-S3, HepG2, and HUVEC.
These were used directly to create a control/background model used for
F-Seq when generating signal annotations and subsequenntly peaks for these
cell lines.  These models are meant to correct for sequencing biases,
alignment artifacts, and copy number changes in these cell lines.  Input
data is not being generated directly for other cell lines.  Instead, a
general background model was derived from the five Input data sets.  This
should provide corrections for sequencing biases and alignment artifacts,
but obviously not for cell type specific copy number changes.
</P>

<h2>Release Notes</h2>
<P>
This is Release 2 (Oct 2009) of this track, which includes new experimental
data as well as the changes described below. The affected database tables and
files include 'V2' in the name, and metadata is marked with
"submittedDataVersion=V2", followed by the reason for replacement. Specific
changes are:

</P><P>
<em>Signals</em> were replaced for all previously released experiments, due to
improvements in the background model.
</P><P>
New <em>Peaks</em> were called for all previously released experiments. Not
only had the background model improved, but the method for determining peaks
was changed from relying on ROC AUC analysis to set peak thresholds to fitting
all peaks from each dataset to a gamma distribution, and setting a threshold
based on a p-value.  For DNaseHS and CTCF, this p-value threshold is 0.05.
For c-Myc, this is 0.01 with the exception of GM12878 where it is 0.001.  In
addition, the background models used when generating the signal tracks for each
were modified slightly.
</P><P>
For HepG2, the pseudoautosomal regions (PARs) on ChrX and ChrY were being
incorrectly annotated for this cell line derived from a male individual.
These were corrected.  Only the PARs on chrX are currently annotated.
</P><P>
The following additional changes were made in Release 2:
<TABLE>
<TR valign='top'><TD width='90px'><em>DNaseHS</em><TD width='90px'><em>GM12878</em>
<TD>New sequences were generated for replicates 1 and 3.   </TR>
<TR valign='top'><TD>&nbsp;<TD><em>K562</em>
<TD>Replicate 3 was discarded due to poor quality data.    </TR>
<TR valign='top'><TD><em>FAIRE</em><TD><em>GM12878</em>
<TD>New sequences were generated for replicates 1 and 2.   </TR>
<TR valign='top'><TD>&nbsp;<TD><em>K562</em>
<TD>New sequences were generated for replicates 2 and 3.<br>
    Replicate 1 was discarded due to poor quality data.</TR>
<TR valign='top'><TD>&nbsp;<TD><em>HeLaS3</em>
<TD>New sequences were generated for replicates 1 and 3.<br>
    Replicate 2 was discarded due to poor quality data.</TR>
<TR valign='top'><TD>&nbsp;<TD><em>HepG2 </em>
<TD>New sequences were generated for replicates 1, 2 and 3.</TR>
</TABLE>
</P><P>
Previous versions of these files are available for download from the
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/hg18/encodeDCC/wgEncodeChromatinMap/"
 TARGET=_BLANK>FTP site</A>

<H2>Credits</H2>
<P>
These data and annotations were created by a collaboration of multiple
institutions (contact:
<A HREF="mailto:&#116;&#115;&#102;ur&#101;&#121;&#64;&#100;&#117;&#107;&#101;.e&#100;u">
&#116;&#115;&#102;ur&#101;&#121;&#64;&#100;&#117;&#107;&#101;.e&#100;u</A>):
<!-- above address is tsfurey at duke.edu -->
</P><P>
<UL>
<LI>Duke University's <A HREF="http://www.genome.duke.edu/index.php"
TARGET=_blank>Institute for Genome Sciences & Policy (IGSP)</A>: Alan Boyle,
Lingyun Song, <A HREF="http://www.genome.duke.edu/people/faculty/furey/"
TARGET=_blank>Terry Furey</A>, and
<A HREF="http://www.genome.duke.edu/people/faculty/crawford/"
TARGET=_blank>Greg Crawford</a>
</P><P>
<LI>University of North Carolina at Chapel Hill: Paul Giresi and
<A HREF="http://www.bio.unc.edu/faculty/lieb/labpages/default.shtml"
TARGET=_blank>Jason Lieb</a>
</P><P>
<LI>Universty of Texas at Austin: Zheng Liu, Ryan McDaniell, Bum-Kyu Lee, and
<A HREF="http://microarray.icmb.utexas.edu/research.html"
TARGET=_blank>Vishy Iyer</a>
</P><P>
<LI><A HREF="http://www.ebi.ac.uk/" TARGET=_blank>European Bioinformatics
Insitute</A>: Paul Flicek, Damian Keefe, and
<A HREF="http://www.ebi.ac.uk/~birney/" TARGET=_blank>Ewan Birney</a>
</P><P>
<LI><A HREF="http://www.oncology.cam.ac.uk/" TARGET=_blank>University of
Cambridge, Department of Oncology and CR-UK Cambridge Research Institute (CRI)
</A>: Stefan Graf
</UL>
</P><P>
We thank NHGRI for ENCODE funding support.
</P>

<H2>References</H2>

<P>
Bhinge AA, Kim J, Euskirchen GM, Snyder M, Iyer, VR.
<A HREF="http://genome.cshlp.org/cgi/content/abstract/17/6/910" TARGET=_blank>
Mapping the chromosomal targets of STAT1 by Sequence Tag Analysis of Genomic
Enrichment (STAGE)</A>. <EM>Genome Res.</EM> 2007 Jun;17(6):910-6.
</P><P>

Boyle AP, Davis S, Shulha HP, Meltzer P, Margulies EH, Weng Z, Furey TS,
Crawford GE. <A HREF="http://www.cell.com/abstract/S0092-8674(07)01613-3"
TARGET=_blank>High-resolution mapping and characterization of open chromatin
across the genome</A>. <EM>Cell.</EM> 2008 Jan 25;132(2):311-22.
</P><P>

Boyle AP, Guinney J, Crawford GE, and Furey TS.
<A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btn480"
TARGET=_blank>F-Seq: a feature density estimator for high-throughput sequence
tags</A>. <EM>Bioinformatics.</EM> 2008 Nov 1;24(21):2537-8.
</P><P>

Buck MJ, Nobel AB, Lieb JD.
<A HREF="http://genomebiology.com/2005/6/11/R97" TARGET=_blank>ChIPOTle: a
user-friendly tool for the analysis of ChIP-chip data</A>.
<EM>Genome Biol.</EM> 2005;6(11):R97.
</P><P>

Crawford GE, Davis S, Scacheri PC, Renaud G, Halawi MJ, Erdos MR, Green R,
Meltzer PS, Wolfsberg TG, Collins FS.
<A HREF="http://www.nature.com/nmeth/journal/v3/n7/abs/nmeth888.html"
TARGET=_blank>DNase-chip: a high-resolution method to identify DNase I
hypersensitive sites using tiled microarrays</A>.
<EM>Nat Methods.</EM> 2006 Jul;3(7):503-9.
</P><P>

Crawford GE, Holt IE, Whittle J, Webb BD, Tai D, Davis S, Margulies EH, Chen Y,
Bernat JA, Ginsburg D <EM>et al.</EM>
<A HREF="http://www.genome.org/cgi/content/abstract/16/1/123"
TARGET=_blank>Genome-wide mapping of DNase hypersensitive sites using massively
parallel signature sequencing (MPSS)</A>.
<EM>Genome Res.</EM> 2006 Jan;16(1):123-31.
</P><P>

The ENCODE Project Consortium.
<A HREF="http://www.nature.com/nature/journal/v447/n7146/abs/nature05874.html"
TARGET=_blank>Identification and analysis of functional elements in 1% of the
human genome by the ENCODE pilot project</A>. <EM>Nature.</EM>
2007 Jun 14;447(7146):799-816.
</P><P>

Giresi PG, Kim J, McDaniell RM, Iyer VR, Lieb JD.
<A HREF="http://www.genome.org/cgi/content/abstract/17/6/877" TARGET=_blank>
FAIRE (Formaldehyde-Assisted Isolation of Regulatory Elements) isolated active
regulatory elements in human chromatin</A>.
<EM>Genome Res.</EM> 2007 Jun;17(6):877-85.
</P><P>

Giresi PG, Lieb JD. <A HREF="http://www.ncbi.nlm.nih.gov/pubmed/19303047"
TARGET=blank>Isolation of active regulatory elements from eukaryotic chromatin
using FAIRE (Formaldehyde Assisted Isolation of Regulatory Elements)</A>.
<EM>Methods.</EM> 2009 Jul;48(3):233-9.
</P><P>

Li H, Ruan J, and Durbin R.
<A HREF="http://genome.cshlp.org/content/early/2008/09/25/gr.078212.108.short"
TARGET=_blank>Mapping short DNA sequencing reads and calling variants using
mapping quality scores</A>. <EM>Genome Res.</EM> 2008 Nov;18(11):1851-8.
</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior consent,
submit publications that use an unpublished ENCODE dataset until nine months
following the release of the dataset.  This date is listed in the <EM>Restricted
Until</EM> column on the track configuration page and the download page.  The
full data release policy for ENCODE is available <A HREF="../ENCODE/terms.html"
TARGET=_BLANK>here</A>.</P>
