<H2>Description</H2>
<P>
This track displays human tissue microarray data using Affymetrix Human Exon 1.0 ST
expression arrays. This RNA expression track was produced as part of
the ENCODE Project. RNA was extracted from cells that were also 
analyzed by DNaseI hypersensitivity, FAIRE, and 
ChIP (<a href="../../cgi-bin/hgTrackUi?g=wgEncodeChromatinMap" TARGET=_BLANK>Open Chromatin</A> track). 
<p>
<H2>Display Convention and Configuration</H2>
The display for this track shows probe location and signal value as
grayscale-colored items where higher signal values correspond to
darker-colored blocks.
<P>
Items with scores between 900-1000 are in the highest 10% quantile for
signal value of that particular cell type. Similarly, items scoring
800-900 are the next 10% quantile and at the bottom of scale, items
scoring 100-200 are in the lowest 20% quantile for signal value.
<P>
The subtracks within this composite annotation track correspond to
data from different cell types and tissues. The configuration options
are shown at the top of the track description page, followed by a list
of subtracks. To display only selected subtracks, uncheck the boxes
next to the tracks you wish to hide.
<P>
For information regarding specific microarray probes, turn on the Affy
Exon Probes track, which can be found inside the 
<a href="../../cgi-bin/hgTrackUi?g=affyAllExonSuper" TARGET=_BLANK>Affy Exon</A> 
supertrack in the <I>Expression</I> track group.
<P>
<H2>Methods</H2>
<P>

Cells were grown according to the approved <A HREF="../../ENCODE/protocols/cell"
TARGET=_BLANK>ENCODE cell culture protocols</A>. Total RNA was isolated from these cells using trizol
extraction followed by cleanup on RNEasy column (Qiagen) that included
a DNase step. The RNA was checked for quality using a nanodrop and an
Agilent Bioanalyzer . RNA (1ug) deemed to be of good quality was then
processed according to the standard Affymetrix Whole transcript Sense
Target labeling protocol that included a riboreduction step. The
fragmented biotin-labeled cDNA was hybridized over 16h to Affymetrix
Exon 1.0 ST arrays and scanned on an Affymetrix Scanner 3000 7G using
AGCC software. Exon expression analyses were carried out using
Affymetrix Expression Console 1.1 software tools. Samples were
quantile normalized for background correction and Probe Logarithmic
Intensity Error summarized.  Only values for the <I>CORE</I> probes were
calculated as these seem to be the most robust.
<P>
<H2>Verification</H2>
<P>
Data were verified by sequencing biological replicates displaying
Pearson correlation coefficient >0.9.
<P> 

<H2>Release Notes</H2>
<P>
This is Release 2 (June 2011) of this track, which 
excludes the LHSR cell line (treated and untreated). 
The data has been withdrawn  by the submitting lab for DNase, FAIRE and exon array.
</P>
<P>
Previous version of these files are available for download from the
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/hg18/encodeDCC/wgEncodeDukeAffyExonArray/"
 TARGET=_BLANK>FTP site</A>.
</P>


<H2>Credits</H2>
<P>
RNA was extracted from each cell type by Greg Crawford's group at Duke
University.  RNA was purified and hybridized to Affymetrix Exon arrays
by Sridar Chittur and Scott Tenenbaum at the University of
Albany-SUNY.  Data analyses were performed by Holly Dressman, Darin
London, and Zhancheng Zhang at Duke University.
<P>
<p>Contact: <A HREF="mailto:&#116;&#101;&#114;&#114;&#121;.
&#102;&#117;rey&#64;&#100;&#117;&#107;e.
&#101;&#100;&#117;">
Terry Furey
</A></p>
<P>
<H2>Data Release Policy</H2> Data users may freely use ENCODE data, but
may not, without prior consent, submit publications that use an
unpublished ENCODE dataset until nine months following the release of
the dataset. This date is listed in the Restricted Until column,
above. The full data release policy for ENCODE is available 
<A href="http://genome.ucsc.edu/ENCODE/terms.html">here</A>.

