<H2>Description</H2>

<P>
This track is produced as part of the ENCODE Project.
RNA-Seq is a method for mapping and quantifying the transcriptome of any 
organism that has a genomic DNA sequence assembly.
RNA-Seq is performed by reverse-transcribing an RNA sample into cDNA, followed 
by high throughput DNA sequencing, which 
was done here on an Illumina Genome Analyzer (GA2) (Mortazavi <EM>et al.</EM>, 
2008).  The transcriptome measurements shown on these tracks were performed on 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?term=longPolyA&amp;type=rnaExtract">polyA selected RNA</A>
 from <A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?term=cell&amp;type=localization">total cellular RNA</A>.  
Data have been produced in two formats: single reads, each of which comes from 
one end of a randomly primed cDNA molecule; 
and paired-end reads, which are obtained as pairs from both ends cDNAs 
resulting from random priming. 
The resulting sequence reads are then informatically mapped onto the genome 
sequence (<I>Alignments</I>). Those that don't map to the genome
are mapped to known RNA splice junctions (<I>Splice Sites</I>).  
These mapped reads are then counted to determine their 
frequency of occurrence at known gene models.  Sequence reads that cluster at 
genome locations that lack an existing
transcript model are also identified informatically and they are quantified.  
RNA-Seq is especially suited for giving 
information about RNA splicing patterns and for determining unequivocally the 
presence or absence of lower abundance class RNAs.  
As performed here, internal RNA standards are used to assist in quantification 
and to provide internal process controls.   
This RNA-Seq protocol does not specify the coding strand.  As a result, there 
will be ambiguity at loci where both strands are transcribed.  
The "randomly primed" reverse transcription is, apparently, not fully random.  
This is inferred from a sequence bias in the first residues of the read 
population, and this likely contributes to 
observed unevenness in sequence coverage across transcripts. </P>

<P>These tracks show 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=readType">1x32 n.t. or 2x75 n.t. or 1x75 n.t. directed sequence reads</A>
 of cDNA obtained from biological replicate samples 
(different culture plates) of the ENCODE 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=cellType">cell lines</A>.
 The 32 n.t. sequences were aligned to the human genome (hg18) and UCSC 
known-gene splice junctions using different
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=mapAlgorithm">sequence alignment programs</A>.
The 1x75D n.t. reads are strand-specific reads.
The 2x75 n.t. reads were mapped serially, first with the Bowtie program 
(Langmead <EM>et al.</EM>, 2009) 
against the genome and UCSC known-gene splice junctions (<I>Splice Sites</I>). 
Bowtie-unmapped reads were then mapped using BLAT to find evidence of novel 
splicing, by requiring at least 10 bp 
on the short-side of the splice. </P>


<H2>Display Conventions and Configuration</H2>
<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are <A HREF="/goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
The following views are in this track:
<DL>
<DT><I>Plus Raw Signal</I>
<DD>Density graph (wiggle) of signal enrichment on the positive strand for strand-specific reads based on a normalized aligned 
read density (RPKM).
The RPKM measure assists in visualizing the relative amount of a given transcript across multiple samples. 
<DT><I>Minus Raw Signal</I>
<DD>Density graph (wiggle) of signal enrichment on the negative strand for strand-specific reads based on a normalized aligned 
read density (RPKM).
The RPKM measure assists in visualizing the relative amount of a given transcript across multiple samples. 
<DT><I>Raw Signal</I>
<DD>Density graph (wiggle) of signal enrichment based on a normalized aligned 
read density (RPKM) for non strand-specific reads.
The RPKM measure assists in visualizing the relative amount of a given transcript across multiple samples. 
<DT><I>Splice Sites</I>
<DD>RNA-seq tags aligning to mRNA splice sites.
<DT><I>Alignments</I>
<DD>The Alignments view shows reads mapped to the genome. 
Alignments are colored by cell type.
</DL>
</P>

<H2>Methods</H2>

<P> 
Cells were grown according to the approved
<A HREF="/ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture protocols</A>.
The cells (either 2 X 10<SUP>7</SUP> or 4 X 10<SUP>7</SUP> cells &mdash; 
GM12878 and K562, 
and 8 X 10<SUP>7</SUP> cells HepG2) 
were lysed in either 4mls (GM12878 and K562) or 12 mls (HepG2) of RLT buffer 
(Qiagen RNEasy kit),
and processed on either 2 (GM12878 and K562) or 3 (HepG2) RNEasy midi columns 
according to the 
manufacturer's protocol, with the inclusion of the "on-column" DNAse digestion 
step to remove 
residual genomic DNA.</P>  

<P>75 &micro;gs of total RNA was selected twice with oligodT beads (Dynal) 
according to the manufacturer's 
protocol to isolate mRNA from each of the preparations.  100 ngs of mRNA was 
then processed according 
to the protocol in Mortazavi et al (2008), and prepared for sequencing on the 
Genome Analyzer flow cell
 according to the protocol for the ChIPSeq DNA genomic DNA kit (Illumina).  </P>

<P>Following alignment of the sequence reads to the genome assembly as 
described above, the sequence 
reads were further analyzed using the 
<a href="http://woldlab.caltech.edu/RNA-seq">ERANGE 3.0</a> software package,
 which quantifies the number of reads falling within the mapped boundaries of 
known transcripts from 
the Gencode annotations. ERANGE assigns both genomically unique reads and 
reads that occur in 2-10 
genomic locations for quantification.  ERANGE also contains a subroutine 
(RNAFAR) which allows the 
consolidation of reads that align close to, but outside the mapped borders 
of known transcripts, 
and the identification of novel transcribed regions of the genome using either 
a 20 kb radius for 
the 1x32 datasets or paired-end information for 2x75 datasets.</P>

<P>
For 2x75 datasets, raw Illumina reads (<I>RawData</I> files on the download 
page, fasta format)
are run through bowtie 0.9.8.1 with up to 2 mismatches
and the resulting mappings are stored (<I>RawData2 files</I>, bowtie format)
for up to ten matches per-read to the genome,
spiked controls and UCSC knownGene splice junctions. Reads that were
not mapped by bowtie (<I>RawData3</I> files, fasta format)
are then mapped onto the genome using blat and filtered using pslReps
(<I>RawData4</I> files, psl format).
The bowtie and blat mappings are then analyzed by ERANGE3.0.2 to
generate wiggles (<I>RawSignal</I> view, wiggle format),
bed files of all reads and splices (<I>Alignments</I> and <I>Paired Alignments</I> views, 
bed format),
all bowtie and blat splices (<I>Splice Sites</I> view, bed format)
and blat-only splices (<I>Splice Sites</I> view, bed format),
as well as RPKM expression level measurements at the gene-level
(<I>RawData5</I> files, rpkm format),
exon-level (<I>RawData6</I> files, rpkm format), and candidate
novel exons (<I>RawData7</I> files, rpkm format).
Fasta files for splice sites (hg18splice75.fa.gz) and spikes (spikes.fa.gz) 
can be found on the 
<A TARGET=_BLANK HREF="
http://hgdownload.cse.ucsc.edu/goldenPath/hg18/encodeDCC/wgEncodeCaltechRnaSeq/">downloads page</A>.
</P>

<H2>Verification</H2>

<UL>
<LI>Known exon maps as displayed on the genome browser are confirmed by the 
alignment of sequence reads. </LI>
<LI>Known spliced exons are detected at the expected frequency for transcripts 
of given abundance.   </LI>
<LI>Linear range detection of spiked in RNA transcripts from Arabidopsis and 
phage lambda over 5 orders of magnitude.   </LI>
<LI>Endpoint RTPCR confirms presence of selected RNAFAR 3&prime;UTR extensions.
</LI>
<LI>Correlation to published microarray data r = 0.62 </LI>
</UL>

<H2>Release Notes</H2>
This is release 2 of the Caltech RNA-seq track.
This release adds five new cell types: H1-hESC, HeLa-S3, HepG2, HUVEC, and NHEK. Also, stranded 75 nt reads are now provided for each cell type.

<H2>Credits</H2>

<P>Wold Group: Ali Mortazavi, Brian Williams, Diane Trout, Brandon King, 
Ken McCue, Lorian Schaeffer.</P>

<P>Myers Group: Norma Neff, Florencia Pauli, Fan Zhang, Tim Reddy, Rami Rauch. 
</P> 

<P>Illumina gene expression group: Gary Schroth, Shujun Luo, Eric Vermaas.</P>
<P>
Contacts: 
<A HREF="mailto:&#100;i&#97;&#110;&#101;&#64;&#99;&#97;lt&#101;&#99;&#104;.
e&#100;&#117;">
Diane Trout</A> (informatics) and
<!-- above address is diane at caltech.edu -->
<A HREF="mailto:&#98;&#97;&#119;&#105;&#108;&#108;&#105;&#95;91&#49;2&#53;&#64;&#121;&#97;&#104;&#111;&#111;.
co&#109;">
Brian Williams
</A> (experimental).
<!-- above address is bawilli_91125 at yahoo.com -->

</P>

<H2>References</H2>

<P>Mortazavi A, Williams BA, McCue K, Schaeffer L, and Wold BJ. 
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/pubmed/18516045">Mapping and quantifying mammalian transcriptomes by RNA-Seq</A>
<EM>Nature Methods</EM>. 2008 Jul; 5(7):621-628.</P> 

<P>Langmead B, Trapnell C, Pop M, Salzberg SL.
<A TARGET=_BLANK HREF="http://genomebiology.com/2009/10/3/R25"><U>Ultrafast and memory-efficient alignment of short DNA sequences to the human genome</U></A>
<EM>Genome Biology</EM>. 2009 Mar; 10:R25.</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column, above.  The full data release policy
for ENCODE is available
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>

