<H2>Description</H2>
<P>
This super-track displays related tracks of open chromatin from the Duke/UNC/Texas/EBI 
ENCODE group.  We identified open chromatin using two independent and complementary methods: 
DNaseI hypersensitivity (HS) and Formaldehyde-Assisted Isolation of Regulatory Elements (FAIRE), 
combined with chromatin immunoprecipitation (ChIP) for select regulatory factors.  Each method is 
coupled to two detection platforms: Illumina (formerly Solexa) sequencing by synthesis, and 
high-resolution whole-genome tiled microarrays supplied by NimbleGen.
</P><P>
DNaseI HS data: DNaseI is an enzyme that has long been used to map general chromatin accessibility 
and the DNaseI "hyperaccessibility" or "hypersensitivity" is a feature of active cis-regulatory 
sequences. The use of this method has led to the discovery of functional regulatory elements that 
include enhancers, silencers, insulators, promotors, locus control regions and novel elements. DNaseI 
hypersensitivity signifies chromatin accessibility following binding of trans-acting factors in place 
of a canonical nucleosome. 
</P><P>
FAIRE data (coming soon): FAIRE is method to isolate and identify nucleosome-depleted regions of the 
genome.  FAIRE was initially discovered in yeast, but subsequently was shown to also identify gene 
regulatory elements in human cells (Giresi et al., 2006).  This method operates at the resolution of 
approximately 200 bp, or around one nucleosome.
</P><P>
ChIP data (coming soon):  Chromatin Immunoprecipitation identifies the specific location of transcription 
factors within the genome.  DNA binding and associated Proteins are crosslinked to DNA using formaldehyde.  
Crosslinked chromatin is sheared and is immunoprecipitated using a specific antibody.  After reversal of 
the crosslinks, this material can be used to identify the location of transcription factor binding.
</P><P>
Cell Lines:
<UL>
 <LI><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM12878&PgId=166" 
TARGET=_blank>GM12878</A> lymphoblastoid cell line (ENCODE Tier 1).</LI>
</UL>
</P>

<H2>Methods</H2>
<P>
DNaseI hypersensitive sites were isolated using methods called DNase-seq or DNase-chip 
(Boyle et al., 2008, Crawford et al., 2006). Briefly, DNaseI digested ends from intact 
chromatin were captured using three different DNase concentrations. This material was 
captured and sequenced using Illumina (Solexa) sequencing, or labeled and hybridized to 
NimbleGen ENCODE (1% of the genome) or whole genome tiled microarrays (coming soon).
</P><P>
Discrete DNase HS sites were identified from DNase-seq by setting a parzen cutoff of 10 
standard deviations from the mean (Boyle et al., 2008).  
</P>

<H2>Credits</H2>
<P>
These data and annotations were created by a collaboration of multiple institutions:
</P><P>
Duke University's <A HREF="http://www.genome.duke.edu/index.php" TARGET=_blank>Institute for Genome Sciences & Policy (IGSP)</A>
Alan Boyle, Lingyun Song, <A HREF="http://www.genome.duke.edu/people/faculty/furey/" TARGET=_blank>Terry Furey</A>, and 
<A HREF="http://www.genome.duke.edu/people/faculty/crawford/" TARGET=_blank>Greg Crawford</a>
</P><P>
University of North Carolina at Chapel Hill
Paul Giresi and <A HREF="http://www.bio.unc.edu/faculty/lieb/labpages/default.shtml" TARGET=_blank>Jason Lieb</a>
</P><P>
Universty of Texas at Austin
Zheng Liu, Ryan McDaniell, Bum-Kyu Lee, and <A HREF="http://microarray.icmb.utexas.edu/research.html" TARGET=_blank>Vishy Iyer</a>
</P><P>
<A HREF="http://www.ebi.ac.uk/" TARGET=_blank>European Bioinformatics Insitute</A>
Paul Flicek, Stefan Graf, Damian Keefe, and <A HREF="http://www.ebi.ac.uk/~birney/" TARGET=_blank>Ewan Birney</a>
</P><P>
We thank NHGRI for ENCODE funding support.

</P>
<H2>References</H2>
<P>
Boyle AP, Davis S, Shulha HP, Meltzer P, Margulies EH, Weng Z, Furey TS, Crawford GE. 
<A HREF="http://www.cell.com/content/article/abstract?uid=PIIS0092867407016133" TARGET=_blank>High-resolution mapping and characterization of open chromatin across the genome.</A> <i>Cell.
</i> 2008 Jan 25;<B>132</B>(2):311-22.
</P><P>
Buck MJ, Nobel AB, Lieb JD, <A HREF="http://genomebiology.com/2005/6/11/R97" TARGET=_blank>ChIPOTle: a user-friendly tool for the analysis of ChIP-chip data.</A> <i>Genome Biology.</i> 2005
</P><P>
Crawford GE, Davis S, Scacheri PC, Renaud G, Halawi MJ, Erdos MR, Green R, Meltzer PS, Wolfsberg TG, Collins FS. 
<A HREF="http://www.nature.com/nmeth/journal/v3/n7/abs/nmeth888.html" TARGET=_blank>DNase-chip: a high-resolution method to identify DNase I hypersensitive sites using tiled microarrays.</A>
<i>Nat Methods.</i> 2006 Jul;<B>3</B>(7):503-9.
</P><P>
Crawford GE, Holt IE, Whittle J, Webb BD, Tai D, Davis S, Margulies EH, Chen Y, Bernat JA, Ginsburg D, Zhou D, Luo S, Vasicek TJ, Daly MJ, Wolfsberg TG, Collins FS. 
<A HREF="http://www.genome.org/cgi/content/abstract/16/1/123" TARGET=_blank>Genome-wide mapping of DNase hypersensitive sites using massively parallel signature sequencing (MPSS).</A> 
<i>Genome Res.</i> 2006 Jan;<B>16</B>(1):123-31.
</P><P>
Giresi PG, Kim J, McDaniel RM, Iyer VR, Lieb JD. 
<A HREF="http://www.genome.org/cgi/content/abstract/17/6/877" TARGET=_blank>FAIRE (Formaldehyde-Assisted Isolation of Regulatory Elements) isolated active regulatory elements in human chromatin.</A>
<i>Genome Res.</i> 2007;<B>17</B>:877-885. 
