<H2>Description</H2>
<P>
This super-track displays related tracks of open chromatin from the Duke/UNC/Texas/EBI
ENCODE group. We identified open chromatin using two independent and complementary methods:
DNaseI hypersensitivity (HS) and Formaldehyde-Assisted Isolation of Regulatory Elements (FAIRE),
combined with chromatin immunoprecipitation (ChIP) for select regulatory factors. Each method is
verified by two detection platforms: Illumina (formerly Solexa) sequencing by synthesis, and
high-resolution 1% ENCODE tiled microarrays supplied by NimbleGen.
</P><P>
<B>DNaseI HS data</B>: DNaseI is an enzyme that has long been used to map general chromatin accessibility
and the DNaseI "hyperaccessibility" or "hypersensitivity" is a feature of active cis-regulatory
sequences. The use of this method has led to the discovery of functional regulatory elements that
include enhancers, silencers, insulators, promotors, locus control regions and novel elements. DNaseI
hypersensitivity signifies chromatin accessibility following binding of trans-acting factors in place
of a canonical nucleosome.
</P><P>
<B>FAIRE data</B>: FAIRE (Formaldehyde Assisted Isolation of Regulatory Elements) is a method to isolate and
identify nucleosome-depleted regions of the genome. FAIRE was initially discovered in yeast, but
subsequently was shown to also identify active regulatory elements in human cells (Giresi et al., 2006).
Most DNA fragments isolated by FAIRE are 200 bp in length.
</P><P>
<B>ChIP data</B>: ChIP (Chromatin Immunoprecipitation) is a method to identify the specific location of proteins
that are directly or indirectly bound to genomic DNA. By identifying the binding location of sequence-specific
transcription factors, general transcription machinery components, and chromatin factors, ChIP can help in
the functional annotation of the open chromatin regions identified by DNaseI HS mapping and FAIRE.
</P>

<H2>Methods</H2>
<P>
DNaseI hypersensitive sites were isolated using methods called DNase-seq or DNase-chip
(Boyle et al., 2008, Crawford et al., 2006). Briefly, cells were lysed with NP40 and
intact nuclei were digested with optimal levels of DNaseI enzyme. DNaseI digested
ends were captured from three different DNase concentrations and material was sequenced
using Illumina (Solexa) sequencing. DNase-seq data was verified using material that was
hybridized to NimbleGen Human ENCODE tiled array (1% of the genome).
</P><P>
FAIRE was performed (Giresi et al., 2007) by cross-linking proteins to DNA using 1% formaldehyde
solution, the complex was sheared using sonication, and a phenol/chloroform extraction was
performed to remove DNA fragments crosslinked to protein. The DNA recovered in the aqueous
phase was sequenced using the Illumina sequencing and hybridized to NimbleGen Human ENCODE
arrays (1% of the genome).
</P><P>
To perform ChIP, proteins were cross-linked to DNA in vivo using 1% formaldehyde solution
(Bhinge et al., 2007, ENCODE Project Consortium., 2007). Crosslinked chromatin was sheared
by sonication and immunoprecipitated using a specific antibody against the protein of interest.
After reversal of the crosslinks, the immunoprecipitated DNA was used to identify the genomic
location of transcription factor binding. This was accomplished by Solexa sequencing of the
ends of the immunoprecipitated DNA (ChIP-seq) as well as labeling and hybridization of the
immunoprecipitated DNA to NimbleGen Human ENCODE tiling arrays (1% of the genome) along with
the input DNA as reference (ChIP-chip).

<H3>High-Throughput Sequence Data Analysis:</H3>
ENCODE Array Data Analysis: Discrete DNase HS, FAIRE, and ChIP sites were identified from
DNase/FAIRE/ChIP-seq by setting a parzen cutoff based on ROC curve analysis using peaks and
non-peaks identified from DNase/FAIRE/ChIP-chip using 1% ENCODE NimbleGen arrays. Tiled array
peaks were identified using ChIPOTle (Buck et al., 2005), and non-peaks were identified using
less stringent peak calling cutoffs.
</P>

<H2>Credits</H2>
<P>
These data and annotations were created by a collaboration of multiple institutions:
</P><P>
Duke University's <A HREF="http://www.genome.duke.edu/index.php" TARGET=_blank>Institute for Genome Sciences & Policy (IGSP)</A>
Alan Boyle, Lingyun Song, <A HREF="http://www.genome.duke.edu/people/faculty/furey/" TARGET=_blank>Terry Furey</A>, and
<A HREF="http://www.genome.duke.edu/people/faculty/crawford/" TARGET=_blank>Greg Crawford</a>
</P><P>
University of North Carolina at Chapel Hill
Paul Giresi and <A HREF="http://www.bio.unc.edu/faculty/lieb/labpages/default.shtml" TARGET=_blank>Jason Lieb</a>
</P><P>
Universty of Texas at Austin
Zheng Liu, Ryan McDaniell, Bum-Kyu Lee, and <A HREF="http://microarray.icmb.utexas.edu/research.html" TARGET=_blank>Vishy Iyer</a>
</P><P>
<A HREF="http://www.ebi.ac.uk/" TARGET=_blank>European Bioinformatics Insitute</A>
Paul Flicek, Stefan Graf, Damian Keefe, and <A HREF="http://www.ebi.ac.uk/~birney/" TARGET=_blank>Ewan Birney</a>
</P><P>
We thank NHGRI for ENCODE funding support.

</P>
<H2>References</H2>

The ENCODE Project Consortium. Identification and analysis of functional elements in 1% of the human genome by the ENCODE pilot project. Nature 2007 447: 799-816
Retrieved from "http://encodewiki.ucsc.edu/EncodeDCC/index.php/CrawfordDNaseDoc"

<P>
Bhinge, A.A., Kim, J., Euskirchen, G.M., Snyder, M., & Iyer, V.R.
 <A HREF="http://genome.cshlp.org/cgi/content/abstract/17/6/910" TARGET=_blank>
 Mapping the chromosomal targets of STAT1 by Sequence Tag Analysis of Genomic Enrichment (STAGE).</A>
 <i>Genome Res.</i> 2007 <B>17</B>: 910-916.
</P><P>

Boyle AP, Davis S, Shulha HP, Meltzer P, Margulies EH, Weng Z, Furey TS, Crawford GE.
 <A HREF="http://www.cell.com/content/article/abstract?uid=PIIS0092867407016133" TARGET=_blank>
 High-resolution mapping and characterization of open chromatin across the genome.</A>
 <i>Cell.</i> 2008 Jan 25;<B>132</B>(2):311-22.
</P><P>

Buck MJ, Nobel AB, Lieb JD,
 <A HREF="http://genomebiology.com/2005/6/11/R97" TARGET=_blank>
 ChIPOTle: a user-friendly tool for the analysis of ChIP-chip data.</A>
 <i>Genome Biology.</i> 2005
</P><P>

Crawford GE, Davis S, Scacheri PC, Renaud G, Halawi MJ, Erdos MR, Green R, Meltzer PS, Wolfsberg TG, Collins FS.
 <A HREF="http://www.nature.com/nmeth/journal/v3/n7/abs/nmeth888.html" TARGET=_blank>
 DNase-chip: a high-resolution method to identify DNase I hypersensitive sites using tiled microarrays.</A>
 <i>Nat Methods.</i> 2006 Jul;<B>3</B>(7):503-9.
</P><P>

Crawford GE, Holt IE, Whittle J, Webb BD, Tai D, Davis S, Margulies EH, Chen Y, Bernat JA, Ginsburg D, Zhou D, Luo S, Vasicek TJ, Daly MJ, Wolfsberg TG, Collins FS.
 <A HREF="http://www.genome.org/cgi/content/abstract/16/1/123" TARGET=_blank>
 Genome-wide mapping of DNase hypersensitive sites using massively parallel signature sequencing (MPSS).</A>
 <i>Genome Res.</i> 2006 Jan;<B>16</B>(1):123-31.
</P><P>

Giresi PG, Kim J, McDaniel RM, Iyer VR, Lieb JD.
 <A HREF="http://www.genome.org/cgi/content/abstract/17/6/877" TARGET=_blank>
 FAIRE (Formaldehyde-Assisted Isolation of Regulatory Elements) isolated active regulatory elements in human chromatin.</A>
 <i>Genome Res.</i> 2007;<B>17</B>:877-885.
</P>
