<H2>Description</H2>
<P>
This track displays positions of four different types of RNA in the human 
genome: 
<UL>
<LI>precursor forms of microRNAs (pre-miRNAs) 
<LI>C/D box small nucleolar RNAs (C/D box snoRNAs)
<LI>H/ACA box snoRNAs
<LI>small Cajal body-specific RNAs (scaRNAs) 
</UL></P>
<P>
C/D box and H/ACA box snoRNAs are guides for the 2'O-ribose methylation and 
the pseudouridilation, respectively, of rRNAs and snRNAs, although many of 
them have no documented target RNA.  The scaRNAs guide modifications of the
spliceosomal snRNAs transcribed by RNA polymerase II, and often contain both 
C/D and H/ACA domains.</P>
<P>
The pre-miRNA data are from the
<A HREF="http://microrna.sanger.ac.uk/sequences/" TARGET=_blank>
miRBase Sequence Database</A> at the <A HREF="http://www.sanger.ac.uk/" 
TARGET=_blank>Wellcome Trust Sanger Institute</A>. The snoRNA and scaRNA data 
are from 
<A HREF="http://www-snorna.biotoul.fr/" TARGET=-blank>snoRNABase</A>, which is 
maintained at the <A HREF="http://www-lbme.biotoul.fr/uk/index.html" 
TARGET=_blank>Laboratoire de Biologie Moléculaire Eucaryote</A>.
</P>

<H2>Display Conventions and Configuration</H2>
<P>
This track follows the general display conventions for 
<A HREF="../goldenPath/help/hgTracksHelp.html#GeneDisplay">gene prediction 
tracks</A>. At a zoomed-in resolution, arrows superimposed on the blocks 
indicate the sense orientation of the RNAs. The RNA types are represented by 
blocks of the following colors:
<UL>
<LI><span style="color: red;">red</span> = pre-miRNA</LI>
<LI><span style="color: blue;">blue</span> = C/D box snoRNA</LI>
<LI><span style="color: green;">green</span> = H/ACA box snoRNA</LI>
<LI><span style="color: magenta;">magenta</FONT> = scaRNA</LI></UL>
</P>
<H2>Methods</H2>
<P>Pre-miRNA genomic locations from <A HREF="http://microrna.sanger.ac.uk/" 
TARGET="_blank">miRBase</a> were calculated using wublastn for sequence 
alignment with the requirement of 100% identity. The extents of the precursor 
sequences were not generally known and were predicted based on base-paired 
hairpin structure.</P>  
<P>
The snoRNAs and scaRNAs genomic locations from <A HREF="http://www-snorna.biotoul.fr/" 
TARGET=_blank>snoRNABase</A> were aligned against the human genome using BLAT. 
In a few cases, no exact match was found for the published sequences; these 
likely correspond to sequencing errors. In these cases, the best BLAT hit (which 
differed from the published sno/scaRNA sequence by 1-3 nucleotides) was adopted.
</P>

<H2>Credits</H2>
<P>
The genome coordinates for the pre-miRNAs were obtained from the
<A HREF="ftp://ftp.sanger.ac.uk/pub/mirbase/sequences/CURRENT/genomes/" 
TARGET=_blank>miRBase Sequence Dabtabse FTP site</A>, and the genome 
coordinates for the snoRNA and scaRNA were obtained from the
<A HREF="http://www-snorna.biotoul.fr/coordinates.php" 
TARGET=_blank>snoRNABase coordinates download page</A>.</P> 

<H2>References</H2>
<P>
When making use of these data, please cite the following articles and, if 
applicable, the primary sources of the RNA sequences:</P>
<P>
Griffiths-Jones S. 
<A HREF="http://nar.oxfordjournals.org/cgi/content/abstract/32/suppl_1/D109" 
TARGET=_blank>The microRNA Registry</A>.
<em>Nucleic Acids Res</em>. 2004 Jan 1;32(Database issue):D109-11.</P>
<P>
Griffiths-Jones S, Grocock RJ, van Dongen S, Bateman A, Enright AJ.
<A HREF="http://nar.oxfordjournals.org/cgi/content/full/34/suppl_1/D140"
TARGET=_blank>miRBase: microRNA sequences, targets and gene nomenclature</A>.
<EM>Nucleic Acids Res</EM>. 2006 Jan 1;334(Database issue):D14-4.</P>
<P>
Griffiths-Jones S, Saini HK, van Dongen S, Enright AJ.
<A HREF=http://nar.oxfordjournals.org/cgi/content/full/36/suppl_1/D154"
TARGET=_blank>miRBase: tools for microRNA genomics</A>.
<EM>Nucleic Acids Res</EM>. 2008 Jan;36(Database issue):D154-8.</P>
<P>
Lestrade L, Weber MJ. 
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/16381836?dopt=Abstract" 
TARGET=_blank>snoRNA-LBME-db, a comprehensive database of human H/ACA and C/D 
box snoRNAs</A>. <EM>Nucleic Acids Res</EM>. 2006 Jan 1;34(Database issue):D158-
62.</P>
<P>
Weber MJ. 
<A HREF="http://www.blackwell-synergy.com/doi/abs/10.1111/j.1432-1033.2004.04389.x"
TARGET=_blank>New human and mouse microRNA genes found by homology search</A>.
<em>Febs J</em>. 2005 Jan;272(1):59-73.</P>
<P>
For more information on BLAT, see Kent WJ.
<A HREF="http://www.genome.org/cgi/content/abstract/12/4/656" 
TARGET=_blank>BLAT - the BLAST-like alignment tool</A>.
<em>Genome Res.</em> 2002;12(4):656-664.</P>

<P>
The following publication provides guidelines on miRNA annotation:
Ambros V. <em>et al</em>., 
<A HREF="http://www.rnajournal.org/cgi/content/abstract/9/3/277" 
TARGET=_blank>A uniform system for microRNA annotation</A>.  
<em>RNA.</em> 2003;9(3):277-9.</P>
