<H2>Description</H2>
<P>
This track shows the best $o_organism/$organism chain for 
every part of the $organism genome. It is useful for
finding orthologous regions and for studying genome
rearrangement.  The $o_organism sequence used in this annotation is from
the $o_date ($o_db) assembly.</P>

<H2>Display Conventions and Configuration</H2>
<P>
In full display mode, the top-level (level 1)
chains are the largest, highest-scoring chains that
span this region.  In many cases gaps exist in the
top-level chain.  When possible, these are filled in by
other chains that are displayed at level 2.  The gaps in 
level 2 chains may be filled by level 3 chains and so
forth. </P>
<P>
In the graphical display, the boxes represent ungapped 
alignments; the lines represent gaps.  Click
on a box to view detailed information about the chain
as a whole; click on a line to display information
about the gap.  The detailed information is useful in determining
the cause of the gap or, for lower level chains, the genomic
rearrangement. </P> 
<P> 
Individual items in the display are categorized as one of four types
(other than gap):</P>
<P><UL>
<LI><B>Top</B> - the best, longest match. Displayed on level 1.
<LI><B>Syn</B> - line-ups on the same chromosome as the gap in the level above
it.
<LI><B>Inv</B> - a line-up on the same chromosome as the gap above it, but in 
the opposite orientation.
<LI><B>NonSyn</B> - a match to a chromosome different from the gap in the 
level above.
</UL></P>

<H2>Methods</H2>
<P>
These alignments were generated using blastz and blat alignments of
genomic sequence from the $o_date $o_organism genome assembly
(UCSC $o_db, NCBI Build 34).
The initial alignments were generated using blastz on 
repeatmasked sequence using the following $organism/$o_organism scoring 
matrix:
<PRE>
     A    C    G    T
A   100 -300 -150 -300
C  -300  100 -300 -150
G  -150 -300  100 -300
T  -300 -150 -300  100

K = 4500, L = 3000,  Y = 3400, H = 2000
</PRE>
</P>
<P>
The resulting alignments were processed by the axtChain program.
AxtChain organizes all the alignments between a single $organism scaffold
and a single $o_organism chromosome into a group and makes a kd-tree out
of all the gapless subsections (blocks) of the alignments.
The maximally-scoring chains of these blocks were found by running a
dynamic program over the kd-tree.  Chains scoring below a certain
threshold were discarded.</P>
<P>
To place additional $organism scaffolds that were not initially aligned by 
blastz, a DNA blat of the unmasked sequence was performed. The resulting
blat alignments were also chained, and then merged with the
blastz-based chains produced in the previous step to produce a
merged set of chains, which are displayed 
in the "Human Chain" track.</P>
<P>
These chains were sorted with the 
highest-scoring chains in the genome ranked first. The program
chainNet was then used to place the chains one at a time, trimming them as 
necessary to fit into sections not already covered by a higher-scoring chain. 
During this process, a natural hierarchy emerged in which a chain that filled 
a gap in a higher-scoring chain was placed underneath that chain. The program 
netSyntenic was used to fill in information about the relationship between 
higher- and lower-level chains, such as whether a lower-level
chain was syntenic or inverted relative to the higher-level chain. </P>

<H2>Credits</H2>
<P>
The chainNet, netSyntenic, and netClass programs were
developed at the University of California
Santa Cruz by Jim Kent.</P>
<P>
Blastz was developed at <A HREF="http://bio.cse.psu.edu"
TARGET=_blank>Pennsylvania State University</A> by
Minmei Hou, Scott Schwartz, Zheng Zhang, and Webb Miller with advice from
Ross Hardison.</P>
<P>
The browser display and database storage of the nets were made
by Robert Baertsch and Jim Kent.</P>

<H2>References</H2>
<P>
Kent, W.J., Baertsch, R., Hinrichs, A., Miller, W., and Haussler, D.
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1"
TARGET=_blank>Evolution's cauldron: Duplication, deletion, and rearrangement
in the mouse and human genomes</A>.
<em>Proc Natl Acad Sci USA</em> <B>100</B>(20), 11484-11489 (2003).</P>
<P>
Schwartz, S., Kent, W.J., Smit, A., Zhang, Z., Baertsch, R., Hardison, R.,
Haussler, D., and Miller, W.
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103"
TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>.
<em>Genome Res.</em> <B>13</B>(1), 103-7 (2003).</P>
