<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<HTML>
<HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="UCSC Track Database Definition Changes">
    <META name="keywords" content="tracDb, tracks, UCSC, bioinformatics">
    <TITLE>UCSC Track Database Definition History</TITLE>
    <LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
    <LINK REL="STYLESHEET" HREF="trackDbDoc.css">
    <script type="text/javascript" src="/js/jquery.js"></script>
    <script type="text/javascript" src="/js/staticDoc.js"></script>
    <script type="text/javascript" src="/js/utils.js"></script>
    <script type="text/javascript" src="trackDbDoc.js"></script>
</HEAD>
<BODY>
<A NAME="TOP"></A>

<!-- Format everthing inside this narrower "pageWidth" window -->
<div class="pageWidth">

<!--#include virtual="../../../cgi-bin/hgMenubar"-->

<!--Content Tables-->
    <!--outer table is for border purposes-->

<div class="contentBorder">
<div class="lightBlueBar">
    <div class="lightBlueBarTitle">&nbsp;&nbsp;Track Database Definition Change History</div>
</div>

<!-- Finally to the content -->
<div class="borderGuard">

<DIV class="trackDbDoc_intro">
<H1>Track Database Definition</H1>

<p>This document summarizes the changes to the <a href="trackDbHub.html">UCSC Genome Browser Hub TrackDb specification</a>.<p>

<TABLE class="settingsTable" id="Common_Settings">
<tbody>
<tr valign="TOP">
<th width="100">Date</th>
<th width="15%">Track Types</th>
<th width="15%">Link to Doc</th>
<th>Change</th>
</tr>

<tr>
    <td>2026-01-11</td>
    <td>hic</td>
    <td><a href="trackDbHub.html#hicArcLimit">hicArcLimit</a></td>
    <td>Limit the number of arcs displayed in a Hi-C plot, favoring those with the highest scores</td>
</tr>
<tr>
    <td>2024-06-11</td>
    <td>bigBed, bigGenePred, bigChain, bigPsl</td>
    <td><a href="trackDbHub.html#highlight">highlightValues.*</a></td>
    <td>Allow user to highlight certain features in yellow, if they contain certain values in an extraField</td>
</tr>
<tr>
    <td>2024-01-22</td>
    <td>bigWig</td>
    <td><a href="trackDbHub.html#setColorWith">setColorWith</a></td>
    <td>Color a signal plot from a bigWig file by the ranges in a bigBed file</td>
</tr>
<tr>
    <td>2023-11-08</td>
    <td>bigBed, bigGenePred, bigChain, bigPsl</td>
    <td><a href="../decorator.html">decorator.*</a></td>
    <td>Semi-transparently overlay annotations, e.g. protein domains on transcripts, mutations on exons, or summits on chip-seq peaks</td>
</tr>
<tr>
    <td>2023-10-16</td>
    <td>bigGenePred, bigChain, bigPsl</td>
    <td><a href="../decorator.html">mouseOver, mouseOverField</a></td>
    <td>mouseOver works for all bigBed-based file formats</td>
</tr>
<tr>
    <td>2023-06-27</td>
    <td>all</td>
    <td><a href="trackDbHub.html#maxItems">maxItems</a></td>
    <td>Change: <tt>maxItems</tt>, increased default to 10,000, 10x more items can be shown.</td>
</tr>
<tr>
    <td>2023-04-04</td>
    <td>all</td>
    <td><a href="trackDbHub.html#downloadUrl">downloadUrl</a></td>
    <td>New: <tt>downloadUrl &lt;fileType&gt; &lt;URL&gt;</tt><br>Shows a link where user can download a file. On the UCSC browser, the link is shown above the description page. Right now this is used for GFF files on gene tracks.</td>
</tr>
<tr>
    <td>2022-09-30</td>
    <td>bigBed, BAM</td>
    <td><a href="trackDbHub.html#doWiggle">doWiggle</a></td>
    <td>New: <tt>doWiggle on</tt><br>Instead of showing the features, show coverage of features, as a 'wiggle' (signal) style track. Can be switched off by the user on the track config page.</td>
</tr>
<tr>
    <td>2022-09-22</td>
    <td>bigGenePred</td>
    <td><a href="trackDbHub.html#colorByStrand">colorByStrand</a></td>
    <td>Change: <tt>colorByStrand</tt> can now be used by bigGenePred-type tracks.</td>
</tr>
<tr>
    <td>2022-06-30</td>
    <td>bigPsl, bigChain</td>
    <td><a href="trackDbHub.html#otherTwoBitUrl">otherTwoBitUrl</a></td>
    <td>New: <tt>otherTwoBitUrl &lt;URL&gt;</tt><br>This statement allows to store query sequences in a separate twoBit file rather than in the annotation file itself, which can reduce the size of the annotation file.</td>
</tr>
<tr>
    <td>2022-05-14</td>
    <td>bigWig</td>
    <td><a href="trackDbHub.html#logo">logo</a></td>
    <td>New: <tt>logo on</tt><br>Motif logo of the genome sequence, as a "dynseq" display, see <a href='../bigWig.html#Ex4'>the example</a>.</td>
</tr>
<tr>
    <td>2022-05-05</td>
    <td>bigMaf, maf</td>
    <td><a href="trackDbHub.html#logoMaf">logoMaf</a></td>
    <td>New: <tt>logoMaf on</tt><br>Motif logo on MAF/bigMaf alignment tracks. Zooming in on alignment shows nucleotides scaled by phyloP conservation.</td>
</tr>
<tr>
    <td>2019-05</td>
    <td>all</td>
    <td><a href="../hgTracksHelp.html#UseOneFile">useOneFile</a></td>
    <td><tt>useOneFile on</tt><br>Adding the 'useOneFile on' line to the hub.txt section of a hub allows the contents of all three files to be referenced inside of one file.</td>
</tr>
</tbody>

</table>

</div>
</div>


</BODY></HTML>
