<H2>Description</H2>
<P>
This track shows alignments of $o_organism ($o_db, $o_date) to the
$organism genome using a gap scoring system that allows longer gaps 
than traditional affine gap scoring systems. It can also tolerate gaps in both
$o_organism and $organism simultaneously. These 
&quot;double-sided&quot; gaps can be caused by local inversions and 
overlapping deletions in both species.  The $o_organism genomic sequence is
from the 13 Nov. 2003 Arachne draft assembly.
<P>
The chain track displays boxes joined together by either single or
double lines. The boxes represent aligning regions.
Single lines indicate gaps that are largely due to a deletion in the
$o_organism assembly or an insertion in the $organism 
assembly.  Double lines represent more complex gaps that involve substantial
sequence in both species. This may result from inversions, overlapping
deletions, an abundance of local mutation, or an unsequenced gap in one
species.  In cases where multiple chains align over a particular region of
the $organism genome, the chains with single-lined gaps are often 
due to processed pseudogenes, while chains with double-lined gaps are more 
often due to paralogs and unprocessed pseudogenes.</P> 
<P>
In the "pack" and "full" display
modes, the individual feature names indicate the scaffold, strand, and
location (in thousands) of the match for each matching alignment.</P>


<H2>Display Conventions and Configuration</H2>
<P>By default, the chains to chromosome-based assemblies are colored
based on which chromosome they map to in the aligning organism. To turn
off the coloring, check the &quot;off&quot; button next to: Color
track based on chromosome.</P>
<P>
To display only the chains of one chromosome in the aligning
organism, enter the name of that chromosome (e.g. chr4) in box next to: 
Filter by chromosome.</P>

<H2>Methods</H2>
<P>
The alignments were generated by blastz on repeatmasked sequence using
the following $organism/$o_organism scoring matrix:
<PRE>
          A    C    G    T
     A   100 -300 -150 -300
     C  -300  100 -300 -150
     G  -150 -300  100 -300
     T  -300 -150 -300  100

     K = 4500, L = 3000,  Y = 3400, H = 2000
</PRE></P>
<P>
The resulting alignments were fed into axtChain, which organizes all
alignments between a single $o_organism scaffold and a single
$organism chromosome into a group and creates a kd-tree out
of the gapless subsections (blocks) of the alignments. A dynamic program
was then run over the kd-trees to find the maximally scoring chains of these
blocks. Chains scoring below a threshold were discarded.</P>
<P>
To place additional $o_organism scaffolds that were not initially aligned by 
blastz, a DNA blat of the unmasked sequence was performed. The resulting
blat alignments were also chained and then merged with the blastz-based 
chains produced in the previous step to produce the chains displayed 
in this track.

<H2>Credits</H2>
<P>
The $o_organism sequence used in this track was obtained from the 13 Nov. 
2003 Arachne assembly. We'd like to thank the National Human Genome Research 
Institute (NHGRI), the Broad Institute at MIT/Harvard, and Washington 
University School of Medicine for providing this sequence.</P>
<P>
Blastz was developed at <A HREF="http://bio.cse.psu.edu" 
TARGET=_blank>Pennsylvania State University</A> by 
Minmei Hou, Scott Schwartz, Zheng Zhang, and Webb Miller with advice from
Ross Hardison.</P>
<P>
Lineage-specific repeats were identified by Arian Smit and his 
<A HREF="http://www.repeatmasker.org" TARGET=_blank>RepeatMasker</A>
program.</P>
<P>
The axtChain program was developed at the University of California at 
Santa Cruz by Jim Kent with advice from Webb Miller and David Haussler.</P>
<P>
The browser display and database storage of the chains were generated
by Robert Baertsch and Jim Kent.</P>

<H2>References</H2>
<P>
Chiaromonte, F., Yap, V.B., Miller, W. 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" 
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. 
<em>Pac Symp Biocomput 2002</em>, 115-26 (2002).</P>
<P>
Kent, W.J., Baertsch, R., Hinrichs, A., Miller, W., and Haussler, D.
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" 
TARGET=_blank>Evolution's cauldron: Duplication, deletion, and rearrangement
in the mouse and human genomes</A>.
<em>Proc Natl Acad Sci USA</em> <B>100</B>(20), 11484-11489 (2003).</P>
<P>
Schwartz, S., Kent, W.J., Smit, A., Zhang, Z., Baertsch, R., Hardison, R., 
Haussler, D., and Miller, W.
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103" 
TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>. 
<em>Genome Res.</em> <B>13</B>(1), 103-7 (2003).</P>

