<H2>Description</H2>
<P>
This track displays $organism/$o_organism genomic alignment chains.
These alignments were generated using blastz and blat alignments of $o_organism 
genomic sequence from the 13 Nov. 2003 Arachne $o_organism draft assembly.
<P>
Alignments were performed using a gap scoring system that allows
longer gaps than traditional affine gap scoring systems.  
It can also tolerate gaps in both $o_organism and $organism simultaneously.
These &quot;double-sided&quot; gaps can be caused by local inversions
and overlapping deletions in both species. 
<P>
The graphical display shows boxes joined together by single or 
double lines. The boxes represent aligning regions. 
Single lines indicate gaps that are largely due to a deletion in the genome of 
the non-$organism species or an insertion in the $organism assembly.
Double lines represent more complex gaps that involve substantial
sequence in both species. This may result from inversions, overlapping
deletions, an abundance of local mutation, or an unsequenced gap in one species.
In cases where there are multiple 
chains over a particular portion of the $organism genome, chains with 
single-lined gaps are often due to processed pseudogenes, while chains 
with double-lined gaps are more often due to paralogs and unprocessed 
pseudogenes. In the fuller display
modes, the individual feature names indicate the scaffold name, strand, and 
location (in thousands) of the match for each matching alignment.
<P>
<H2>Methods</H2>
<P>
The alignments were generated by blastz on repeatmasked sequence using
the following $organism/$o_organism scoring matrix:
<PRE>
     A    C    G    T
     A   100 -300 -150 -300
     C  -300  100 -300 -150
     G  -150 -300  100 -300
     T  -300 -150 -300  100

     K = 4500, L = 3000,  Y=3400, H=2000
</PRE>
The resulting alignments were processed by the axtChain program.
AxtChain organizes all the alignments between a single $o_organism scaffold
and a single $organism chromosome into a group and makes a kd-tree out
of all the gapless subsections (blocks) of the alignments.
The maximally-scoring chains of these blocks were found by running a
dynamic program over the kd-tree.  Chains scoring below a certain
threshold were discarded. 
<P>
To place additional $o_organism scaffolds that weren't initially aligned by 
blastz, a DNA blat of the unmasked sequence was performed. The resulting
blat alignments were also chained, and then merged with the 
blastz-based chains produced in the previous step to produce the chains displayed 
in this track.
<P>

<H2>Credits</H2>
The $o_organism sequence used in this track was obtained from the 13 Nov. 2003 
Arachne assembly. We'd like to thank the National Human Genome Research Institute 
(NHGRI), the Eli & Edythe L. Broad Institute at MIT/Harvard, and Washington University 
School of Medicine for providing this sequence.</P>
<P>
Blastz was developed at <A HREF="http://bio.cse.psu.edu">Pennsylvania State University</A> by 
Scott Schwartz, Zheng Zhang, and Webb Miller with advice from
Ross Hardison.</P>
<P>
The chaining and netting programs were developed 
at the University of California at Santa Cruz by Jim Kent.</P>
<P>
The browser display and database storage of the chains were made
by Robert Baertsch and Jim Kent.</P>

<H2>References</H2>
<P>
Chiaromonte F, Yap VB, Miller W (2002). 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. 
<em>Pac Symp Biocomput</em> 2002;:115-26.</P>
<P>
Kent WJ, Baertsch R, Hinrichs A, Miller W, and Haussler D (2003).
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" 
TARGET=_blank>Evolution's cauldron: 
Duplication, deletion, and rearrangement in the mouse and human genomes</A>. 
<I>Proc Natl Acad Sci USA</I> 100(20):11484-11489 Sep 30 2003.
<P>
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison R, Haussler D, and 
Miller W (2003). 
<A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>Human-Mouse 
Alignments with BLASTZ</A>. <em>Genome Res.</em> 13(1):103-7.</P>
