<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=290 HEIGHT=290>
  <TR><TD ALIGN=RIGHT>
    <IMG SRC="/images/human.jpg" WIDTH=275 HEIGHT=275 ALT="human">
  </TD></TR>
  <TR><TD ALIGN=RIGHT>
    <FONT SIZE=-1> <EM>Homo sapiens</EM><BR>
    </FONT>
    <FONT SIZE=-2> (Graphic courtesy of
      <A HREF="http://www.cbse.ucsc.edu" TARGET=_blank>CBSE</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The July 2003 human reference sequence (NCBI Build 34) was produced by the 
International Human Genome Sequencing Consortium. For more information about
this assembly, see
<A HREF="http://www.ncbi.nlm.nih.gov/genome/assembly/111478/" TARGET="_blank">
NCBI34</A> in the NCBI Assembly database.</P>

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic clone, an mRNA or EST or STS marker, or
a cytological band, a chromosomal coordinate range, or keywords from 
the GenBank description of an mRNA. The following list shows 
examples of valid position queries for the human genome.
See the
<A HREF="../goldenPath/help/hgTracksHelp.html" TARGET=_blank>
User's Guide</A> for more information.
<P>
<TABLE  border=0 CELLPADDING=1 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><br></TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chr7</TD>
        <TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays all of chromosome 7</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>20p13</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region for band p13 on chr 20</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr3:1-1000000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays first million bases of chr 3, counting from p-arm telomere</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>scf1:1-1000000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays first million bases of scaffold 1 of an unmapped genome assembly</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr3:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr3 that spans 2000 bases, starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>D16S3046</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region around STS marker D16S3046 from the Genethon/Marshfield maps.
                      Includes 100,000 bases on each side as well.</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>RH18061;RH80175</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region between STS markers RH18061;RH80175.
                      This syntax 
		      may also be used for other range queries, such as between 
		      cytobands and uniquely-determined ESTs, mRNAs, refSeqs, etc.</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>AA205474</TD>
       	<TD WIDTH=14></TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession AA205474 in BRCA1 cancer gene on chr 17</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>AC008101</TD>
    	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of clone with GenBank accession AC008101</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>AF083811</TD>
       	<TD WIDTH=14></TD>
   	<TD VALIGN=Top>Displays region of mRNA with GenBank accession number AF083811</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>PRNP</TD>
       	<TD WIDTH=14></TD>
       	<TD>Displays region of genome with HUGO Gene Nomenclature Committee identifier PRNP</TD>
    </TR>
    <TR>
	<TD VALIGN=Top NOWRAP>NM_017414</TD>
	<TD><BR></TD>
	<TD VALIGN=Top>Displays the region of genome with RefSeq identifier NM_017414</TD>
    </TR>
    <TR>
	<TD VALIGN=Top NOWRAP>NP_059110</TD> 
	<TD><BR></TD>
 	<TD VALIGN=Top NOWRAP> Displays the region of genome with protein accession number NP_059110</TD>
    </TR>
    <TR>
	<TD VALIGN=Top NOWRAP>11274</TD>
	<TD><BR></TD>
	<TD VALIGN=Top NOWRAP>Displays the region of genome with Entrez Gene identifier 11274</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>homeobox caudal</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>kruppel zinc finger</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>huntington</TD>
       	<TD WIDTH=14></TD>
 	<TD VALIGN=Top>Lists candidate genes associated with Huntington's disease</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>zahler</TD>
       	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zahler</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Evans,J.E.</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author J.E.  Evans</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Evans JE</em>, internally it uses 
			the format <em>Evans,J.E.</em>.</TD>
    </TR>
</TABLE>
<P>
<HR>
<H3>Assembly details</H3>
<P>
This sequence covers about 99 percent of 
the gene-containing regions in the genome, and has been sequenced to an 
accuracy of 99.99 percent. 
Of note in this release is the addition of the pseudoautosomal regions
of the Y chromosome. This sequence was taken from the corresponding regions
in the X chromosome and is an exact duplication of that sequence.</P>
<P>
There are 2,843,433,602 finished sequenced bases in the 
ordered and oriented portion of the assembly, which is an increase of 0.4 
percent, or approximately 11 Mb, over the Build 33 assembly.
The reference 
sequence is considered to be &quot;finished&quot;, a technical term indicating that
the sequence is highly accurate (with fewer than one error per 10,000
bases) and highly contiguous (with the only remaining gaps corresponding 
to regions whose sequence cannot be reliably resolved with current
technology). Future work on the reference sequence will focus on improving 
accuracy and reducing gaps in the sequence.</P>
<P>
Some sequence joins between adjacent clones in this assembly could not be 
computationally validated because the clones originated from different 
haplotypes and contained polymorphisms in the overlapping sequence, or the 
overlap was too small to be to be reliable. In these instances, the
sequencing center responsible for the particular chromosome has provided 
data to support the join in the form of an electronic certificate. These
certificates may be reviewed through the link below. </P>
<P>
For further information on NCBI Build 34, see the NCBI 
<A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/human/release_notes.html#b34"
TARGET=_blank>Build 34 release notes</A>.</P>
<P>
Bulk downloads of the sequence and annotation data are available from the
UCSC downloads server via 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/hg16/">ftp</A> or 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#human">http</A>. 
The hg16 annotation tracks were generated by UCSC and 
collaborators worldwide. See the <A HREF="../goldenPath/credits.html#human_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. </P>
</P>
<HR>
<H3>Statistical information</H3>
<UL>
<LI><A HREF="../goldenPath/certificates/">Non-Standard Join Certificates</A>
<LI><A HREF="../goldenPath/stats.html">Summary Statistics</A>
</UL>
<HR>
<H3>Genbank Pipeline Details</H3>
For the purposes of the Genbank alignment pipeline, this assembly is considered to be: <B>well-ordered</B>.
