<H2>Description</H2>
<P>
This track shows the starts and ends of mRNA transcripts 
determined by paired-end ditag (PET) sequencing. PETs are composed of 18 
bases from either end of a cDNA; 36 bp PETs from many clones were 
concatenated together and cloned into pZero-1 for efficient sequencing.  See 
the Methods and References sections below for more details on PET sequencing.
</P>
<P>
The PET sequences in this track are full-length transcripts
derived from two cell lines and mapped on whole genome:
<UL>
<LI>the log phase of MCF7 cells</LI>
<LI>HCT116 cells treated with 5FU (5-fluorouracil) for 6 hours</LI>
</UL>
In total, 584,624 PETs were generated for MCF7 and 280,340 PETs were
generated for HCT116. More than 80% of the PETs in each group were 
mapped to the genome.
The 474,278 MCF7 PETs and 223,261 HCT116 PETs that mapped with single and
multiple (up to ten) matches in the genome are shown in the two 
subtracks. </P>
<P> 
In the graphical display, 
the ends are represented by blocks connected by a horizontal line. In full
and packed display modes, the arrowheads on the horizontal line represent the 
direction of transcription, and an ID of the format <em>XXXXX-N-M</em> is 
shown to the left of each PET, where <em>X</em> is the unique ID for each
PET, <em>N</em> indicates the number of mapping locations in the genome 
(1 for a single mapping location, 2 for two mapping locations, and so forth),
and <em>M</em> is the number of PET sequences at this location. The total 
count of PET sequences mapped to the same locus but with slight nucleotide
differences may reflect the expression level of the transcripts. PETs that 
mapped to multiple locations may represent low complexity or repetitive 
sequences.</P>
<P>
The graphical display also uses color coding to reflect the uniqueness
and expression level of each PET:</P>
<BR>
<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4>
  <TR><TH align=left>Color</TH><TH align=left>Mapping</TH><TH align=left>PETS observed at location</TH></TR>
  <TR><TD>dark blue</TD><TD>unique</TD><TD>2 or more</TD></TR>
  <TR><TD>light blue</TD><TD>unique</TD><TD>1</TD></TR>
  <TR><TD>medium brown</TD><TD>multiple</TD><TD>2 or more</TD></TR>
  <TR><TD>light brown</TD><TD>multiple</TD><TD>1</TD></TR>
</TABLE>

<H2>Methods</H2>
<P>
PolyA+ RNA was isolated from the cells. A full-length cDNA library was
constructed and converted into a PET library for Gene
Identification Signature analysis (Ng <em>et al</em>., 2005). Generation of 
PET sequences involved cloning of cDNA sequences into the plasmid vector, 
pGIS3. pGIS3 contains two <I>MmeI</I> recognition sites that
flank the cloning site, which were used to produce a 36 bp PET. Each 36 bp PET 
sequence contains 18 bp from each of the 5' and 3' ends of the original 
full-length cDNA clone. The 18 bp 3' signature contains 16 bp 3'-specific 
nucleotides and an AA residual of the polyA tail to indicate the sequence 
orientation. PET sequences were mapped to the genome using the following 
specific criteria:
<UL>
<LI>a minimal continuous 16 bp match must exist for the 5' signature; the 
3' signature must have a minimal continuous 14 bp match
<LI>both 5' and 3' signatures must be present on the same chromosome
<LI>their 5' to 3' orientation must be correct
<LI>the maximal genomic span of a PET genomic alignment must be less than 
one million bp
</UL></P>
<P>
Most of the PET sequences (more than 90%) were mapped to specific locations
(single mapping loci). PETs mapping to 2 - 10 locations are
also included and may represent duplicated genes or pseudogenes in
the genome. </P> 

<H2>Verification</H2>
<P>
To assess overall PET quality and mapping specificity, the top ten most 
abundant PET clusters that mapped to well-characterized known genes were
examined. Over 99% of the PETs represented full-length transcripts, and the 
majority fell within ten bp of the known 5' and 3' boundaries of these 
transcripts. The PET mapping was further verified by confirming the existence 
of physical cDNA clones represented by the ditags. PCR primers were designed
based on the PET sequences and amplified the corresponding cDNA inserts from 
the parental GIS flcDNA library for sequencing analysis. In a set of 86
arbitrarily-selected PETs representing a wide range of annotation
categories &mdash; including known genes (38 PETs), predicted genes (2 PETs),
and novel transcripts (46 PETs) &mdash; 84 (97.7%) confirmed
the existence of bona fide transcripts. </P>
 
<H2>Credits</H2>
<P>
The GIS-PET libraries and sequence data for transcriptome analysis were 
produced at the 
<A HREF="http://www.gis.a-star.edu.sg/internet/site"
TARGET=_blank>Genome Institute of Singapore</A>. The data were
mapped and analyzed by scientists from the Genome Institute of
Singapore and the 
<A HREF="http://www.bii.a-star.edu.sg/" TARGET=_blank>Bioinformatics Institute of 
Singapore</A>.</P>
 
<H2>References</H2>
<P>
Ng, P. <em>et al</em>.
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/15782207/" 
TARGET=_blank>Gene identification signature (GIS) analysis for transcriptome 
characterization and genome annotation</A>.
<em>Nat. Methods</em> <B>2</B>(2), 105-11 (2005).</P>
