<H2>Description</H2>
<P>
This track displays different measurements of conservation based on
the <A HREF="http://lagan.stanford.edu/lagan_web/index.shtml" 
TARGET=_blank>MLAGAN</A> multiple sequence alignments of ENCODE regions 
shown in the MLAGAN Alignment track. Three programs &mdash; binCons
(binomial-based conservation method), phastCons (phylogenetic hidden-Markov
model method), and
<A HREF="http://mendel.stanford.edu/sidowlab/downloads/GERP/index.html"
TARGET=_blank>GERP</A> (Genomic Evolutionary Rate Profiling)
&mdash; generated the conservation scoring used to create this track. A related 
track, MLAGAN Elements, shows multi-species conserved sequences (MCSs) based on
the conservation measurements displayed in this track.</P>
<P>
For details on the conservation scores generated by each program, refer to the 
individual Methods subsections.</P>

<H2>Display Conventions and Configuration</H2>
<P>
This annotation follows the display conventions for composite 
tracks. The subtracks within this annotation may be configured in a variety of 
ways to highlight different aspects of the displayed data. The graphical 
configuration options are shown at the top of the track description page, 
followed by a list of subtracks. To display only selected subtracks, uncheck 
the boxes next to the tracks you wish to hide. For more information about the 
graphical configuration options, click the 
<A HREF="../goldenPath/help/hgWiggleTrackHelp.html">Graph configuration 
help</A> link.</P>
<P>
Color differences among the subtracks are arbitrary; they provide a
visual cue for distinguishing the different gene prediction methods. See the
Methods section for display information specific to each subtrack.</P>

<H2>Methods</H2>
<P>
The methods used to create the MLAGAN alignments in the ENCODE
regions are described in the MLAGAN Alignment track description.</P>

<H3>BinCons</H3>
<P>
The binCons score is based on the cumulative binomial probability of
detecting the observed number of identical bases (or greater) in
sliding 25 bp windows (moving one bp at a time) between the
reference sequence and each other species, given the neutral rate
at four-fold degenerate sites. Neutral rates are calculated
separately at each targeted region. For targets with no gene annotations,
the average percent identity across all alignable sequence was instead used
to weight the individual species binomial scores (this latter
weighting scheme was found to closely match 4D weights).</P>
<P>
The negative log of these P-values was then averaged across all
human-referenced pairwise combinations, and the highest scoring overlapping
25 bp window for each base was the resulting score. This track shows the
plotting of a ranked percentile score normalized between 0 and 1 across all
ENCODE regions, such that the top 5% most conserved sequence across all ENCODE
regions have a score of 0.95 or greater (top 10% have a score of 0.9 or
greater, and so on).</P>
<P>
BinCons scores were normalized to represent a percentile to the power of
10. For example, scores representing the top 1 percent most conserved
sequence, 99th percentile, have a score greater than or equal to 0.99^10
= 0.904. Transforming scores to the power of 10 was done for visual
purposes only, in order to accentuate and distinguish the peaks of more
highly conserved regions.</P>
<P>
More details on binCons can be found in Margulies <em>et. al</em>. (2003)
cited below.</P>

<H3>PhastCons</H3>
<P>
The phastCons program predicts conserved elements and produces base-by-base
conservation scores using a two-state phylogenetic hidden Markov model.
The model consists of a state for conserved regions and a
state for nonconserved regions, each of which is associated with a
phylogenetic model.  These two models are identical
except that the branch lengths of the conserved phylogeny are
multiplied by a scaling parameter <em>rho</em> (0 &lt; <em>rho</em> &lt; 1).</P>
<P>
For determining the conservation for the ENCODE MLAGAN
alignments, the nonconserved model was estimated
from four-fold degenerate coding sites within the ENCODE regions using
the program phyloFit. The parameter <em>rho</em> was then estimated by
maximum likelihood, conditional on the nonconserved model, using the EM
algorithm implemented in phastCons.  Parameter estimation was based on
a single large alignment, constructed by concatenating the
alignments for all conserved regions.</P>
<P>
PhastCons was run with the options <TT>--expected-lengths 15</TT> and
<TT>--target-coverage 0.05</TT> to obtain the desired level of
&quot;smoothing&quot; and a final coverage by conserved elements of 5%.</P>
<P>
The conservation score at each base is the posterior probability that the
base was generated by the conserved state of the phylo-HMM.  It can
be interpreted as the probability that the base is in a conserved
element, given the assumptions of the model and the estimated parameters.
Scores range from 0 to 1, with higher scores corresponding to
higher levels of conservation.</P>
<P>
More details on phastCons can be found in Siepel <em>et. al.</em> (2005)
cited below.</P>

<H3>GERP</H3>
<P>
The GERP score is the expected substitution rate divided by
the observed substitution rate at a particular human base.
Scores are estimated on a column-by-column basis using multiple sequence
alignments of mammalian genomic DNA generated by MLAGAN.
The scores range from 0 to 3; those greater than 3 are clipped to 3. 
The expected and observed rates are
both calculated on a phylogenic tree using the same fixed topology.
The branch lengths of the expected tree are based on the average
substitutions at neutral sites.  The branch lengths of the observed
tree, which is calculated separately for each human base, are based on the
substitutions seen at the column of the
multiple alignment at that base.  Species that have gaps at
a particular column are not considered in the scoring for that column.</P>
<P>
Higher scores correspond to human
bases in alignment columns with higher degrees of similarity, <em>i.e.</em>
bases that have evolved slowly, some of which have been under purifying 
selection. The opposite holds true for swiftly evolving (low similarity) 
columns. </P>
<P>
Scores are deterministic, given a maximum-likelihood model of 
nucleotide substitution, species topology, neutral tree, and alignment.</P>

<H2>Credits</H2>
<P>
BinCons was developed by Elliott Margulies of the 
<A HREF="http://www.genome.gov/10000452" TARGET=_blank>Eric Green lab</A> at 
NHGRI.</P>
<P>
PhastCons was developed by Adam Siepel in the 
<A HREF="http://www.cbse.ucsc.edu/staff/hausslerlab.shtml"
TARGET=_blank>Haussler lab</A> at UCSC.</P>
<P>
GERP was developed primarily by Greg Cooper in the lab of 
<A HREF="http://mendel.stanford.edu/sidowlab/" TARGET=_blank>Arend Sidow</A> 
at Stanford University
(Depts of Pathology and Genetics), in close collaboration with 
Eric Stone (Biostatistics, NC State), and George Asimenos and 
Eugene Davydov in the lab of 
<A HREF="http://ai.stanford.edu/~serafim/" TARGET=_blank>Serafim Batzoglou</A> 
(Dept. of Computer Science, Stanford).</P>
<P>
The data for this track were generated by Elliott Margulies, with assistance 
from Adam Siepel.</P>

<H2>References</H2>
<P>
Cooper, G.M., Stone, E.A., Asimenos, G., NISC Comparative Sequencing Program, 
Green, E.D., Batzoglou, S. and Sidow, A.  
<A HREF="http://www.genome.org/cgi/content/abstract/15/7/901" 
TARGET=_blank>Distribution and intensity of constraint in mammalian genomic 
sequence</A>.   
<em>Genome Res.</em>. <B>15</B>(7), 901-13 (2005).</P>
<P>
Margulies, E.H., Blanchette, M., NISC Comparative Sequencing Program,
Haussler, D. and Green, E.D.
<A HREF="http://dx.doi.org/10.1101/gr.1602203"
TARGET=_blank>Identification and characterization of multi-species conserved
sequences</A>.
<em>Genome Res.</em> <B>13</B>, 2507-18 (2003). </P>
<P>
Siepel, A., Bejerano, G., Pedersen, J.S., Hinrichs, A., Hou, M.,
Rosenbloom, K., Clawson, H., Spieth, J., Hillier, L.W. <em>et al.</em>
<A HREF="http://www.genome.org/cgi/content/abstract/15/8/1034"
TARGET=_blank">Evolutionarily conserved elements in vertebrate,
insect, worm, and yeast genomes</A>.
<em>Genome Res.</em> <B>15</B>, 1034-50 (2005).</P>
