<H2>Description</H2>
<P>
ENCODE region-wide location analysis of H3 and H4 histones
was conducted employing ChIP-chip using chromatin extracted from GM06990
(lymphoblastoid) and K562 (myeloid leukemia-derived) cells.
Histone methylation and acetylation serves as a stable genomic imprint
that regulates gene expression and other epigenetic 
phenomena.  These histones are found in transcriptionally active domains 
called euchromatin.</P>
<P>
<TABLE BORDER="0" BGCOLOR="888888" CELLSPACING="0" CELLPADDING="1"><tr><td>
<TABLE BORDER=1 BGCOLOR="FFFEE8">
<TR><TH align=left>Track</TH><TH align=left>Cell<BR>Line/Type</TH><TH align=left>Antibody</TH><TH align=left>Epitope</TH>
    <TH align=left>Data&nbsp;FTP access</TH><TH align=left>Histone</TH><TH align=left>Array ID</TH></TR>
<TR><TD>1:&nbsp;SI&nbsp;H3K4m1&nbsp;GM06990</TD>
    <TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM06990&PgId=166"
	TARGET=_blank>GM06990</A></TD>
    <TD><A HREF="http://www.abcam.com/index.html?datasheet=8895" 
	TARGET=_blank>ab8895</A></TD>
    <TD>H3K4me1</TD>
    <TD><A HREF="ftp://ftp.sanger.ac.uk/pub/encode/H3K4me1_GM06990_1"
	TARGET=_blank>H3K4me1_GM06990_1</A></TD>
    <TD>H3 monomethyl lysine 4</TD>
    <TD><A HREF="ftp://ftp.sanger.ac.uk/pub/encode/microarrays/ENCODE3.1.1.hg16.adf"
	TARGET=_blank>ENCODE3.1.1</A></TD>
    </TR>
<TR><TD>2:&nbsp;SI&nbsp;H3K4m2&nbsp;GM06990</TD>
    <TD><A HREF="http://ccr.coriell.org/nigms/nigms_cgi/display.cgi?GM06990"
	TARGET=_blank>GM06990</A></TD>
    <TD><A HREF="http://www.abcam.com/index.html?datasheet=7766" 
	TARGET=_blank>ab7766</A></TD>
    <TD>H3K4me2</TD>
    <TD><A HREF="ftp://ftp.sanger.ac.uk/pub/encode/H3K4me2_GM06990_1"
	TARGET=_blank>H3K4me2_GM06990_1</A></TD>
    <TD>H3 dimethyl lysine 4</TD>
    <TD><A HREF="ftp://ftp.sanger.ac.uk/pub/encode/microarrays/ENCODE3.1.1.hg16.adf"
	TARGET=_blank>ENCODE3.1.1</A></TD>
    </TR>
<TR><TD>3:&nbsp;SI&nbsp;H3K4m3&nbsp;GM06990</TD>
    <TD><A HREF="http://ccr.coriell.org/nigms/nigms_cgi/display.cgi?GM06990"
	TARGET=_blank>GM06990</A></TD>
    <TD><A HREF="http://www.abcam.com/index.html?datasheet=8580"
	TARGET=_blank>ab8580</A></TD>
    <TD>H3K4me3</TD>
    <TD><A HREF="ftp://ftp.sanger.ac.uk/pub/encode/H3K4me3_GM06990_2"
	TARGET=_blank>H3K4me3_GM06990_2</A></TD>
    <TD>H3 trimethyl lysine 4</TD>
    <TD><A HREF="ftp://ftp.sanger.ac.uk/pub/encode/microarrays/ENCODE3.1.1.hg16.adf"
	TARGET=_blank>ENCODE3.1.1</A></TD>
    </TR>
<TR><TD>4:&nbsp;SI&nbsp;H3ac&nbsp;GM06990</TD>
    <TD><A HREF="http://ccr.coriell.org/nigms/nigms_cgi/display.cgi?GM06990"
	TARGET=_blank>GM06990</A></TD>
    <TD><A HREF="http://www.upstate.com/browse/productdetail.q.productid.e.06-599"
	TARGET=_blank>06-599</A></TD>
    <TD>H3ac</TD>
    <TD><A HREF="ftp://ftp.sanger.ac.uk/pub/encode/H3ac_GM06990_1"
	TARGET=_blank>H3ac_GM06990_1</A></TD>
    <TD>H3 acetylated lysines 9 and 14</TD>
    <TD><A HREF="ftp://ftp.sanger.ac.uk/pub/encode/microarrays/ENCODE3.1.1.hg16.adf"
	TARGET=_blank>ENCODE3.1.1</A></TD>
    </TR>
<TR><TD>5:&nbsp;SI&nbsp;H4ac&nbsp;GM06990</TD>
    <TD><A HREF="http://ccr.coriell.org/nigms/nigms_cgi/display.cgi?GM06990"
	TARGET=_blank>GM06990</A></TD>
    <TD><A HREF="http://www.upstate.com/browse/productdetail.q.productid.e.06-866"
	TARGET=_blank>06-866</A></TD>
    <TD>H4ac</TD>
    <TD><A HREF="ftp://ftp.sanger.ac.uk/pub/encode/H4ac_GM06990_1"
	TARGET=_blank>H4ac_GM06990_1</A></TD>
    <TD>H4 acetylated lysines 5, 8, 12, 16</TD>
    <TD><a href="ftp://ftp.sanger.ac.uk/pub/encode/microarrays/ENCODE3.1.1.hg16.adf"
	TARGET=_blank>ENCODE3.1.1</A></TD>
    </TR>
<TR><TD>6:&nbsp;SI&nbsp;H3K4me2&nbsp;K562</TD>
    <TD><A href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=789258&dopt=Abstract"
	TARGET=_blank>K562</A></TD>
    <TD><A HREF="http://www.abcam.com/index.html?datasheet=7766"
	TARGET=_blank>ab7766</A></TD>
    <TD>H3K4me2</TD>
    <TD><A HREF="ftp://ftp.sanger.ac.uk/pub/encode/H3K4me2_K562_1"
	TARGET=_blank>H3K4me2_K562_1</A></TD>
    <TD>H3 K4 dimethylated</TD>
    <TD><a href="ftp://ftp.sanger.ac.uk/pub/encode/microarrays/ENCODE3.1.1.hg16.adf"
	TARGET=_blank>ENCODE3.1.1</A></TD>
    </TR>
<TR><TD>7:&nbsp;SI&nbsp;H3K4me3&nbsp;K562</TD>
    <TD><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=789258&dopt=Abstract"
	TARGET=_blank>K562</a></TD>
    <TD>
    <a HREF="http://www.abcam.com/index.html?datasheet=8580"
	TARGET=_blank>ab8580</a></TD>
    <TD>H3K4me3</TD>
    <TD><a HREF="ftp://ftp.sanger.ac.uk/pub/encode/H3K4me3_K562_1"
	TARGET=_blank>H3K4me3_K562_1</a></TD>
    <TD>H3 trimethyl lysine 4</TD>
    <TD><a href="ftp://ftp.sanger.ac.uk/pub/encode/microarrays/ENCODE3.1.1.hg16.adf"
	TARGET=_blank>ENCODE3.1.1</a></TD>
    </TR>
<TR><TD>8:&nbsp;SI&nbsp;H3ac&nbsp;K562</TD>
    <TD><A href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=789258&dopt=Abstract"
	TARGET=_blank>K562</A></TD>
    <TD><A HREF="http://www.upstate.com/browse/productdetail.q.productid.e.06-599"
	TARGET=_blank>06-599</A></TD>
    <TD>H3ac</TD>
    <TD><A HREF="ftp://ftp.sanger.ac.uk/pub/encode/H3ac_K562_1"
	TARGET=_blank>H3ac_K562_1</A></TD>
    <TD>H3 acetylated</TD>
    <TD><a href="ftp://ftp.sanger.ac.uk/pub/encode/microarrays/ENCODE3.1.1.hg16.adf"
	TARGET=_blank>ENCODE3.1.1</A></TD>
    </TR>
<TR><TD>9:&nbsp;SI&nbsp;H4ac&nbsp;K562</TD>
    <TD><A href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=789258&dopt=Abstract"
	TARGET=_blank>K562</A></TD>
    <TD><A HREF="http://www.upstate.com/browse/productdetail.q.productid.e.06-866"
	TARGET=_blank>06-866</A></TD>
    <TD>H4ac</TD>
    <TD><A HREF="ftp://ftp.sanger.ac.uk/pub/encode/H4ac_K562_1"
	TARGET=_blank>H4ac_K562_1</A></TD>
    <TD>H4 acetylated</TD>
    <TD><a href="ftp://ftp.sanger.ac.uk/pub/encode/microarrays/ENCODE3.1.1.hg16.adf"
	TARGET=_blank>ENCODE3.1.1</A></TD>
    </TR>
</TABLE>
</TD></TR></TABLE>
</P>

<H2>Display Conventions and Configuration</H2>
<P>
This annotation follows the display conventions for composite 
&quot;wiggle&quot; tracks. The subtracks within this annotation 
may be configured in a variety of ways to highlight different aspects of the 
displayed data. The graphical configuration options are shown at the top of 
the track description page, followed by a list of subtracks.  To display only 
selected subtracks, uncheck the boxes next to the tracks you wish to hide. 
For more information about the graphical configuration options, click the 
<A HREF="../goldenPath/help/hgWiggleTrackHelp.html" TARGET=_blank>Graph
configuration help</A> link.</P>

<H2>Methods</H2>
<P>
Chromatin from the cell line was cross-linked with 1% formaldehyde,
precipitated with antibody binding to the histone, and sheared and hybridized to
a DNA array.  DNA was not amplified prior to hybridization.</P>
<P>
The raw and transformed data files reflect fold enrichment over background,
averaged over six replicates.</P>

<H2>Verification</H2>
<P>                   
There are six replicates: two technical replicates (immunoprecipitations)
for each of the three biological replicates (cell cultures).</P>
<P>
Raw and transformed (averaged) data can be downloaded from the Wellcome
Trust Sanger Institute FTP site as indicated in the table above.</P>

<H2>Credits</H2>
<P>
The data for this track were generated by the 
<A HREF="http://www.sanger.ac.uk/PostGenomics/encode/investigators.shtml"
TARGET=_blank>ENCODE investigators</A> at the 
<A HREF="http://www.sanger.ac.uk/" TARGET=_blank>Wellcome Trust Sanger 
Institute</a>, Hinxton, UK.</P>
