<H2>Description</H2>
<P>
This track displays human-centric multiple sequence alignments in the 
ENCODE regions for the 23 vertebrates included in the May 2005 ENCODE MSA 
freeze, based on comparative sequence data generated for the ENCODE project. 
The alignments in this track were generated using the 
<A HREF="http://lagan.stanford.edu/" TARGET=_blank>LAGAN</A> Alignment Toolkit.
A complete list of the vertebrates included in the May 2005 freeze may be found
at the top of the description page for this track.</P>
<P>
The Genome Browser companion tracks, MLAGAN Cons and MLAGAN Elements, 
display conservation scoring and conserved elements for these alignments based 
on various conservation methods.</P>

<H2>Display Conventions and Configuration</H2>
<P>
In full display mode, this track shows pairwise alignments
of each species aligned to the human genome.
The alignments are shown in dense display mode using a gray-scale
density gradient.  The checkboxes in the track configuration section allow
the exclusion of species from the pairwise display.
<P>
When zoomed-in to the base-display level, the track shows the base
composition of each alignment. The numbers and symbols on the &quot;human
gap&quot; line indicate the lengths of gaps in the human sequence at those
alignment positions relative to the longest non-human sequence. If the gap
size is greater than 9, a non-numeric symbol is shown; if the
gap size is a multiple of 3, a &quot;*&quot; is displayed,
otherwise &quot;+&quot; is shown.
To view detailed information about the
alignments at a specific position, zoom in the display to 30,000 or fewer
bases, then click on the alignment.  </P>

<H2>Methods</H2>
<P>
To create the alignments, the sequence of each non-human species was first 
&quot;rearranged&quot; to be orthologously collinear with respect to the human 
sequence. The rearrangements were generated using a suite of tools and 
algorithms based on Shuffle-LAGAN and SuperMap.
For each pairing of human sequence with that of another species, Shuffle-LAGAN 
was used to find the best-scoring chain of local similarities according to 
a scoring scheme that penalized evolutionary rearrangements. SuperMap was then
used to aggregate parts of the chain into a human-monotonic map of syntenic 
blocks. This mapping was used to undo the genomic rearrangements of the other 
sequence and convert it to a form that was directly alignable to the human 
sequence.
<P>
A multiple global alignment was created for every region using 
<A HREF="http://lagan.stanford.edu/" TARGET=_blank>MLAGAN</A>. The alignments 
were then refined using 
<A HREF="http://www.drive5.com/muscle/" TARGET=_blank>MUSCLE</A>, which 
processes small non-overlapping windows of an alignment and attempts to realign 
them in an iterative fashion, keeping the refined alignment
if it has a better sum-of-pairs score than the original.</P>

<H2>Credits</H2>
<P>
The MLAGAN alignments were generated by George Asimenos from Stanford's 
<A HREF="http://www-shgc.stanford.edu/genetics/encode.html"
TARGET=_blank>ENCODE group</A>.</P>
<P>
Shuffle-LAGAN, SuperMap and MLAGAN were written by Mike Brudno.</P>
<P>
MUSCLE was authored by Bob Edgar.</P>

<H2>References</H2>
<P>
Brudno, M., Do, C., Cooper, G., Kim, M.F., Davydov, E., Green, E.D., Sidow, A.
and Batzoglou, S. 
<A HREF="http://www.genome.org/cgi/content/abstract/13/4/721"
TARGET=_blank>LAGAN and Multi-LAGAN: efficient tools for large-scale multiple 
alignment of genomic DNA</A>.
<em>Genome Res.</em> <B>13</B>(4), 721-31 (2003).</P>
<P>
Brudno, M., Malde, S., Poliakov, A., Do, C., Courone, O., Dubchak, I. and 
Batzoglou, S.
<A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/19/suppl_1/i54"
TARGET=_blank>Glocal alignment: finding rearrangements during alignment</A>.
<em>Bioinformatics</em> <B>19</B>(Suppl. 1), i54-i62 (2003).</P>
<P>
Edgar, R.C. 
<A HREF="http://nar.oupjournals.org/cgi/content/full/32/5/1792?ijkey=48Nmt1tta0fMg&keytype=ref"
TARGET=_blank>MUSCLE: multiple sequence alignment with high
accuracy and high throughput</A>.
<em>Nucl. Acids Res.</em> <B>32</B>(5), 1792-97 (2004).</P>
