<H2>Description</H2>

<P>
This track displays <B>
Methylation Interference
</B> data for the NHGRI ENCODE target regions. Methylation
interference identifies guanines and adenines in the DNA binding site
that, when methylated, interfere with binding of the protein. a single
end-labeled DNA probe that is methylated at an average of one site per
molecule of probe. The labeled probe is a substrate for a
protein-binding reaction. DNA-protein complexes are separated from the
free probe by the mobility shift DNA-binding assay. A DNA probe that
is methylated at a position which interferes with binding will not be
retarded in this assay. Therefore, the specific DNA-protein complex is
depleted for DNA that contains methyl groups on purines important for
binding. After gel purification, DNA is cleaved with
piperidine. Finally, these fragments are electrophoresed on
polyacrylamide sequencing gels and autoradiographed. Guanines and
adenines that interfere with binding are revealed by their absence in
the retarded complex relative to a lane containing piperidine-cleaved
free probe. This procedure offers a rapid and highly analytical means
of characterizing DNA-protein interactions.


<BR>
<P>
The long-term goal of the NHGRI <A
HREF="http://www.genome.gov/10005107" TARGET=_BLANK>ENCODE Project
(ENCyclopedia of DNA Elements)</A> is to identify all functional
elements in the human genome sequence to facilitate a better
understanding of human biology and disease.

<P>
During the pilot phase, 44 regions comprising 30 Mb -- approximately
1% of the human genome -- have been selected for intensive study to
identify, locate and analyze functional elements within the
regions. These targets are being studied by a diverse public research
consortium to test and evaluate the efficacy of various methods,
technologies, and strategies for locating genomic features. The
outcome of this initial phase will form the basis for a larger-scale
effort to analyze the entire human genome.

<P>
<A HREF="http://www.genome.gov/10506161" TARGET=_BLANK>The ENCODE
Project: Target Selection Process</A> provides a description of how
the initial target regions were selected.

<P>
To open a UCSC Genome Browser with a menu for selecting ENCODE
regions, use <A HREF="../ENCODE/index.html" TARGET=_BLANK>ENCODE
Regions in the UCSC Browser
</A>.

<H2>Credits</H2>
Thanks to Laura Elnitski, Ross Hardison, and Webb Miller at Penn State and the NHGRI ENCODE project for providing this initial set of data.

