<H2>Description</H2> 
<P>
This track shows a measure of evolutionary conservation in $organism, chimp, 
mouse and rat based on a phylogenetic hidden Markov model (phastCons).  
Multiz alignments of the following assemblies were used to generate this
annotation: 
<UL>
<LI>$organism $date 003 ($db) 
<LI>chimp Nov. 2003 (panTro1)
<LI>mouse Feb. 2003 (mm3) draft assembly
<LI>rat Jun. 2003 (rn3)
</UL></P>

<H2>Methods</H2>
<P>
Multiz is a multiple alignment program that takes
<A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>blastz</A> 
"Best-in-Genome" alignments (axtBest or axtNet) as input.  For $organism/rodent
alignments, it uses the same scoring matrix as blastz between pairs of sequences:  
<PRE>
        A     C     G     T
  A    91  -114   -31  -123
  C  -114   100  -125   -31
  G   -31  -125   100  -114
  T  -123   -31  -114    91

  O = 400, E = 30, K = 3000, L = 3000, M = 50
</PRE> </P>
<P>
For the mouse/rat alignments the following matrix was used:

<PRE>

        A     C     G     T
  A    86  -135   -68  -157
  C  -135   100  -148   -68
  G   -68  -148   100  -135
  T  -157   -68  -135    86

  O = 600, E = 50
</PRE></P>
<P>
For the $organism/chimpanzee alignments the following matrix was used:

<PRE>

       A    C    G    T
 A   100 -300 -150 -300
 C  -300  100 -300 -150
 G  -150 -300  100 -300
 T  -300 -150 -300  100

 O = 400, E = 30, K = 4500, L = 4500, M = 50
</PRE></P>
<P>
The overall score is the sum of the score over all pairs.  </P>
<P>
The resulting $organism-chimp-mouse-rat multiple alignments were then assigned 
conservation scores by phastCons.
The phastCons program computes conservation scores based on a phylo-HMM, a
type of probabilistic model that describes both the process of DNA
substitution at each site in a genome and the way this process changes from
one site to the next (Felsenstein and Churchill 1996, Yang 1995, Siepel and
Haussler 2005).  PhastCons uses a two-state phylo-HMM, with a state for
conserved regions and a state for non-conserved regions.  The value plotted
at each site is the posterior probability that the corresponding alignment
column was "generated" by the conserved state of the phylo-HMM.  These
scores reflect the phylogeny (including branch lengths) of the species in
question, a continuous-time Markov model of the nucleotide substitution
process, and a tendency for conservation levels to be autocorrelated along
the genome (i.e., to be similar at adjacent sites).  The general reversible
(REV) substitution model was used.  Note that, unlike many
conservation-scoring programs, phastCons does not rely on a sliding window
of fixed size, so short highly-conserved regions and long moderately
conserved regions can both obtain high scores.  More information about
phastCons can be found in Siepel et al. (2005).</P> 
<P> 
PhastCons currently
treats alignment gaps as missing data, which sometimes has the effect of
producing undesirably high conservation scores in gappy regions of the
alignment.  We are looking at several possible ways of improving the
handling of alignment gaps.</P>

<H2>Credits</H2>
<P>
This track was created at UCSC using the following programs:
<UL>
<LI>
Blastz and multiz by Minmei Hou, Scott Schwartz and Webb Miller of the 
<A HREF="http://bio.cse.psu.edu/" TARGET=_blank>Penn State Bioinformatics 
Group</A>. 
<LI>
Mouse sequence data 
are provided by the Mouse Genome Sequencing Consortium. 
<LI>
Rat sequence data are provided by the Rat Sequencing 
Consortium.
<LI>
Chimpanzee sequence data are provided by the Chimpanzee
Sequencing Consortium.
<!--
<LI>
<A HREF="http://www.genome.org/cgi/content/full/13/1/103" TARGET=_blank>AxtBest</A>
, axtChain, chainNet, netSyntenic, and netClass 
developed by Jim Kent at UCSC. --> 
<LI>
Jim Kent wrote
<A HREF="http://www.genome.org/cgi/content/full/13/1/103" TARGET=_blank>axtBest</A>
and the scripts to run multiz genome-wide and to display
the alignments in this browser.  
<LI>PhastCons by Adam Siepel at Cornell University. 
<LI>"Wiggle track" plotting software by Hiram Clawson at UCSC.
</UL>
</P>
</P>

<H2>References</H2>

<H3>Phylo-HMMs and phastCons</H3>
<P>
Felsenstein, J. and Churchill, G.A.
<A HREF="http://mbe.oxfordjournals.org/cgi/content/abstract/13/1/93"
TARGET=_blank>A hidden Markov model approach to
variation among sites in rate of evolution</A>.
<em>Mol Biol Evol</em> <B>13</B>, 93-104 (1996). </P>
<P>
Siepel, A. and Haussler, D. Phylogenetic hidden Markov models.
In R. Nielsen, ed., <em>Statistical Methods in Molecular Evolution</em>,
pp. 325-351, Springer, New York (2005). </P>
<P>
Siepel, A., Bejerano, G., Pedersen, J.S., Hinrichs, A., Hou, M., Rosenbloom,
K., Clawson, H., Spieth, J., Hillier, L.W., Richards, S., Weinstock, G.M.,
Wilson, R. K., Gibbs, R.A., Kent, W.J., Miller, W., and Haussler, D.
<A HREF="http://www.genome.org/cgi/doi/10.1101/gr.3715005"
TARGET=_blank>Evolutionarily conserved elements in vertebrate, insect, worm,
and yeast genomes</A>.
<em>Genome Res.</em> <B>15</B>, 1034-1050 (2005).</P>
<P>
Yang, Z.
<A HREF="http://www.genetics.org/cgi/content/abstract/139/2/993"
TARGET=_blank>A space-time process model for the evolution of DNA
sequences</A>.  <em>Genetics</em>, <B>139</B>, 993-1005 (1995). </P>

<H3>Chain/Net:</H3>
<P>
Kent, W.J., Baertsch, R., Hinrichs, A., Miller, W., and Haussler, D.
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1"
TARGET=_blank>Evolution's cauldron:
Duplication, deletion, and rearrangement in the mouse and human genomes</A>.
<I>Proc Natl Acad Sci USA</I> <B>100</B>(20), 11484-11489 (2003).

<H3>Multiz:</H3>
<P>
Blanchette, M., Kent, W.J., Riemer, C., Elnitski, .L, Smit, A.F.A., Roskin,
K.M., Baertsch, R., Rosenbloom, K., Clawson, H., Green, E.D., Haussler, D.,
Miller, W.
<A HREF="http://www.genome.org/cgi/content/abstract/14/4/708" TARGET=_blank>Aligning Multiple Genomic Sequences with the Threaded Blockset Aligner</A>.
<em>Genome Res.</em> <B>14</B>(4), 708-15 (2004).

<H3>Blastz:</H3>
<P>
Chiaromonte, F., Yap, V.B., and Miller, W.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract"
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>.
<em>Pac Symp Biocomput</em> <B>2002</B>, 115-26 (2002). </P>
<P>
Schwartz, S., Kent, W.J., Smit, A., Zhang, Z., Baertsch, R., Hardison, R.,
Haussler, D., and Miller, W.
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103"
TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>.
<em>Genome Res.</em> <B>13</B>(1), 103-7 (2003).</P>
