<H2>Description</H2>
<P>
This track shows syntenous (corresponding) regions between $organism and rat chromosomes. The June 2003 (rn3) assembly of the rat genome was used to produce this annotation.</P>

<H2>Methods</H2>
<P>
We passed a 100k non-overlapping window over the genome and - using the blastz best in rat 
genome alignments - looked for high-scoring regions with at least 40% of the bases aligning 
with the same region in rat. 100k segments were joined together if they agreed in direction and
were within 500 kb of each other in the $organism genome and within 4 Mb of each other in the rat. 
Gaps were joined between syntenic anchors if the bases between two flanking regions agreed with 
synteny (direction and rat location). Finally, we extended the syntenic block to include those 
areas.</P>

<H2>Credits</H2>
<P>
Contact <A HREF="mailto:&#98;&#97;&#101;&#114;&#116;s&#99;h&#64;&#99;&#115;&#101;.&#117;&#99;s&#99;.e&#100;&#117;">Robert 
Baertsch</A> at UCSC for more information about this track.
Thanks to the Rat Genome Sequencing Consortium for providing the rat sequence 
data. For more information, see the Baylor College Human Genome Sequencing Center
 <A HREF="https://www.hgsc.bcm.edu/content/rat-genome-project" TARGET=_blank>Rat Genome Project page</A>.</P>
