<H2>Description</H2>
<P>
The Twinscan program predicts genes in a manner similar to Genscan, except 
that Twinscan takes advantage of genome comparisons to improve gene prediction
accuracy.  In the version of Twinscan used to generate this track, intronless 
copies of known genes are masked out before gene prediction, reducing 
the number of non-processed pseudogenes in gene models. More information and a
web server can be found at <A HREF="http://mblab.wustl.edu/" 
TARGET=_blank>http://mblab.wustl.edu/</A>.  </P>

<H2>Display Conventions and Configuration</H2>
<P>
This track follows the display conventions for
<A HREF="../goldenPath/help/hgTracksHelp.html#GeneDisplay">gene prediction
tracks</A>. </P>
<P>
The track description page offers the following filter and configuration
options:
<UL>
<LI><B>Color track by codons:</B> Select the <em>genomic codons</em> option
to color and label each codon in a zoomed-in display to facilitate validation
and comparison of gene predictions. Click the 
<A HREF="../goldenPath/help/hgCodonColoring.html">Help on codon coloring</A>
link on the track description page for more information about this feature.
</UL></P>

<H2>Methods</H2>
<P>
The Twinscan algorithm is described in Korf, I. <em>et al.</em> 2001 in the
References section below.</P>

<H2>Credits</H2>
<P>
Thanks to <A HREF="http://mblab.wustl.edu/"
TARGET=_blank>Michael Brent's Computational Genomics Group</A> at Washington
University St. Louis for providing these data.</P>

<H2>References</H2>
<P>
Korf I, Flicek P, Duan D, Brent MR.
<A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/17/suppl_1/S140" TARGET=_blank>Integrating genomic homology into gene structure prediction</A>.
<em>Bioinformatics</em>. 2001 Jun 1;17(90001)S140-8.</P>
