<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=185>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://piggenome.org/"
    TARGET=_blank>
    <IMG SRC="/images/Meleagris_gallopavo.jpg" WIDTH=193 HEIGHT=300 ALT="Turkey"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Meleagris gallopavo</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of USDA
      (<A HREF="http://www.ars.usda.gov/is/graphics/photos/" 
      TARGET=_blank>Agricultural Research Service</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Dec. 2009 <em>Meleagris gallopavo</em> draft assembly
(TGC Turkey_2.01 (NCBI project <A HREF="http://www.ncbi.nlm.nih.gov/genomeprj/10805"
TARGET=_blank>10805</A>, <A HREF="ftp://ftp.ncbi.nlm.nih.gov/genbank/genomes/Eukaryotes/vertebrates_other/Meleagris_gallopavo/Turkey_2.01/"
TARGET=_blank>GCA_000003025.2</A>))
was produced by the  
<A HREF="http://www.vtnews.vt.edu/articles/2010/09/090810-cals-genome.html"
TARGET=_blank>Turkey Genome Consortium</A>. 



<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic supercontig, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
chicken genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>chrZ</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of chromosome Z</TD>
    </TR>
    <TR><TD VALIGN=Top>chr3:1-1000000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of chr 3</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr3:23000+4000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr 3 that spans 4000 bases, starting with position 23,000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>BX950328</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number BX950328</TD>
    </TR>
    <TR><TD VALIGN=Top>BU335953</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		AF070471 on chr 9</TD>
    </TR>
    <TR><TD VALIGN=Top>CWNT-8C</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of genome with gene symbol CWNT-8C</TD>
    </TR>
    <TR>
	<TD VALIGN=Top>NM_204735</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays the region of genome with RefSeq identifier
		NM_204735</TD>
    </TR>
    <TR><TD VALIGN=Top>374180</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays the region of genome with Entrez Gene
		identifier 374180</TD>
    </TR>
    <TR><TD VALIGN=Top>EU177006;AY873804</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays region between genome landmarks, such as the mRNA markers EU177006 and AY873804.
                   This syntax may also be used for other range queries, such as between uniquely determined ESTs,
                   STSs, refSeqs, etc.</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
<!--
    <TR><TD VALIGN=Top>pseudogene mRNA</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
-->
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>jarvis</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Jarvis</TD>
    </TR>
    <TR><TD VALIGN=Top>Taguchi,T.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author T. Taguchi</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Taguchi T</em>, internally it
	uses the format <em>Taguchi,T.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>

<P>
This assembly sequenced DNA from a female turkey "Nici" on
two next-generation sequencing platforms, Roche 454 and Illumina GAII.
The draft assembly has approximately 1.1 GB with 917 MB assigned to 32 specific chromosomes: chr1-30, chrW, and chrZ. 
The mitochondrial sequence chrM was obtained from GenBank accession  NC_010195. 
This assembly covers about 90 percent of the genome and is based on approximate 5X genome coverage in 454 reads 
and 25X coverage in GAII reads.
The total sequence length is approximately 1.1 Gb, of which about 0.12 Gb are 'N' bases in gaps, 
resulting in approximately 1.08 Gb of ACGT bases. 
</P>
<P>
 
For more information about the techniques
and the assembly process used to produce this assembly, see the 
<A HREF="http://www.ncbi.nlm.nih.gov/nuccore/ADDD00000000"
 TARGET=_blank>GenBank accession record</A> and
the paper:  
Dalloul RA et al. 
<A HREF="http://www.plosbiology.org/article/info:doi%2F10.1371%2Fjournal.pbio.1000475"
 TARGET=_blank>Multi-Platform Next-Generation Sequencing of the Domestic Turkey (Meleagris gallopavo): Genome Assembly and Analysis</A>.
</P>



<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/melGal1/">FTP server</A> or 
the 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#turkey">Downloads</A> 
page. These data have 
<A HREF="../goldenPath/credits.html#turkey_use">specific conditions for 
use</A>. 
The turkey browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#turkey_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>

