<h2>Description</h2>

<p>
This track displays the location of RNA polyadenylation (polyA) sites based
on high-throughput RNA sequencing using the PolyA-seq protocol.
</p>

<p>
PolyA-Seq data is strand-specific, therefore two tracks are provided for each tissue. PolyA
site positions correspond to a single base, namely the ends of read alignments immediately
upstream of the polyadenylation site. The data provided in this track consists of filtered
polyA sites (see Methods). When multiple sites occurred within a 30-bp window on the same
strand, the sum of the reads was attributed to the site with the most reads. Units are in
reads per million (RPM) aligned. To obtain read counts, multiply RPM values by the total
number of filtered reads for the corresponding experiment:
</p>

<table class="descTbl">
<thead style="font-weight:bold;background-color:#FCECC0;">
<tr> 
	<th>Sample</th>
	<th>Filtered reads</th>
</tr>
</thead>
<tbody>
<tr>
	<td>Brain</td>
	<td>5549424</td>
</tr>
<tr>
	<td>Testis</td>
	<td>7466688</td>
</tr>
</tbody>
</table>

<h2>Display Conventions and Configuration</h2>

<p>
These tracks may be configured in a variety of ways to highlight different 
aspects of the displayed data. The graphical configuration options 
are shown at the top of the track description page.  For more information, 
see <a href="../goldenPath/help/hgWiggleTrackHelp.html" target="_blank">Configuring 
graph-based tracks</a>.
</p>

<p>
In the full and pack display modes, forward-strand tracks are shown in <span style="color:#993333;">
<b>red</b></span> and reverse-strand tracks are shown in <span
style="color:#000;"><b>black</b></span>.
In the squish and dense display modes, intensity is represented in grayscale (the darker
the shading, the higher the intensity). To show only selected subtracks, uncheck the boxes next to
the tracks that 
you wish to hide.
</p>

<h2>Methods</h2>

<p>
A detailed explanation of the experimental methods is provided at NCBI's Gene 
Expression Omnibus under accession <a
href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE30198" 
target=_blank>GSE30198</a>. Briefly, PolyA+ RNA was reverse-transcribed
using a T(10)VN primer and strand-specific universal adapters, amplified, and sequenced
on an Illumina GAIIx sequencer. Reads were reverse-complemented, aligned to the corresponding
reference genome and splice junctions, and retained only if they aligned uniquely. 3' ends of
alignments were considered polyA sites. Sites were then filtered using downstream base
frequency matrices for true- and false-positive sites determined from a modified experiment 
based on a T(10) primer (i.e., excluding the 3' VN). When multiple filtered sites occurred
within a 30-nt window on the same strand, read counts were summed and attributed to the most
abundant peak. For each tissue, read counts were then divided by the total number of reads,
in millions, from all filtered sites.
</p>

<h2>Credits</h2>

<p>
These data were generated at <a href="http://www.merck.com/mrl/" target="_blank">Merck Research
Laboratories</a> and submitted by Adnan Derti and Tomas Babak.
</p>

<h2> Data Release Policy</h2>

<p>No restrictions.</p>