<H2>Description</H2>
This track depicts gaps in the assembly.
<P>
There are 154,810 gaps of size 100 bases or more for a
total of 98,876,036 bases. The largest gap is 3,000 bases. 
See also:
<A HREF="http://genome.wustl.edu/pub/organism/Primates/Callithrix_jacchus/assembly/Callithrix_jacchus-2.0.2/ASSEMBLY"
TARGET=_blank>WUSTL Callithrix jacchus</A> assembly details.
</P>
<P>
Gaps are represented as black boxes in this track. If the relative order and
orientation of the contigs on either side of the gap is known, it is a bridged
gap and a white line is drawn through the black box representing the gap.
</P>
<P>All gaps in this assembly are bridged fragment gaps:
<UL>
<LI><B>Fragment</B> - gaps between the Whole Genome Shotgun contigs of a 
supercontig. (In this context, a contig is a set of overlapping sequence reads.  
A supercontig is a set of contigs ordered and oriented during the 
Whole Genome Shotgun process using paired-end reads.)
These are represented by varying numbers of <em>N</em>s  in the assembly.  
Fragment gap sizes are usually taken from read pair data.  
</LI>

</UL></P>
