<H2>Description</H2>
<P>
This track shows the draft assembly of the $organism genome.  
Whole-genome shotgun reads were assembled into contigs.  Where possible, 
contigs were grouped into scaffolds (also known as &quot;supercontigs&quot;)
with order, orientation and gap sizes between contigs based on paired-end 
read evidence.  
The assembly was produced by 
<A HREF="http://www.agencourt.com/" TARGET=_BLANK>Agencourt</A> using
the <A HREF="http://www.broad.mit.edu/wga/" TARGET=_BLANK>Arachne</A>
assembler.  Where the assembly indicated a negative gap size, a gap size of 
25 base pairs was substituted in by 
<A HREF="http://rana.lbl.gov/" TARGET=_BLANK>Michael Eisen</A> at Berkeley.
</P>
<P>In dense mode, this track depicts the contigs that make up the 
currently viewed scaffold. 
Contig boundaries are distinguished by the use of alternating gold and brown 
coloration. Where gaps
exist between contigs, spaces are shown between the gold and brown
blocks.  The relative order and orientation of the contigs
is always known for contigs within a scaffold, so a line is drawn to bridge 
the blocks.</P>
<P>
All components within this track are of fragment type &quot;W&quot;: 
Whole Genome Shotgun contig. 

