<P>
The <em>Drosophila simulans</em> Release 1.0 (Apr. 2005) draft genome sequence
was produced by the <A HREF="http://genome.wustl.edu/" TARGET=_blank>Genome 
Sequencing Center</A> at the Washington University in St. Louis (WUSTL) School
of Medicine. </P>

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of an
mRNA or EST, a chromosomal range, or keywords
from the GenBank description of an mRNA. The following list provides
examples of valid position queries for the Drosophila
genome. Note that some position queries (e.g. &quot;huntington&quot;) may return
matches to the mRNA records of other species. In these cases, the
mRNAs are mapped to their homologs in Drosophila.  See the 
<A HREF="../goldenPath/help/hgTracksHelp.html" TARGET=_blank>
User's Guide</A> for more information.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><BR></TD>
    </TR>
	
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chr2L</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays all of chromosome 2L (see the Assembly Details
	section for a list of chromosomes present in this assembly)</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>chrYh_random:1-75000</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays first 75,000 bases of chromosome Yh_random (h = heterochromatin random reads)</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>AY017266</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region aligning with <em>D. simulans</em> mRNA with GenBank accession number AY017266</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>BG642296</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region aligning with <em>D. simulans</em> EST with GenBank accession BG642296</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>p53</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs related to the p53 tumor suppressor</TD>
    </TR> 

    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists zinc finger mRNAs</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>huntington</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs associated with Huntington's disease</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>Panelli,S.</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by co-author S. Panelli</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Panelli S</em>, internally it uses 
			the format <em>Panelli,S.</em>.</TD>
    </TR>
</TABLE>
<P>
<HR>
<H3>Assembly details</H3>
<P>
The Release 1.0 <em>D. simulans</em> assembly represents a mosaic of
several different <em>D. simulans</em> lines. The assembly process
began with an ~4X WGS assembly of the the <em>D. simulans</em> w501 line.  
The w501 contigs were initially anchored, ordered and oriented by
alignment with the <em>D. melanogaster</em> genome. The assembly was then
examined for places where the w501 assembly suggested inversions with
respect to the <em>D. melanogaster</em> assembly. One major inversion was 
found, confirming the already documented inversion found by 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=5729&dopt=Citation"
TARGET=_BLANK>Lemeunier and Ashburner</A> (1976).  Six other
<em>D. simulans</em> lines (c167.4, md106ts, md199s, nc48s, sim4, and sim6)
were assembled with ~1X coverage.  Using the 4X WGS assembly of the
simulans w501 genome as a scaffold, contigs and unplaced reads from
the 1X assemblies of the other individual <em>D. simulans</em> lines were used
to cover gaps in the w501 assembly where possible.  Thus the resulting
assembly is a mosaic containing the w501 contigs as the primary
scaffolding, with contigs and unplaced reads from the other lines
filling gaps in the w501 assembly.</P>
<P>
Mitochondrial sequence was
downloaded separately from Genbank.  Total size, excluding the gapless
14,972 bp mitochondrial sequence, is 142,405,747 bp including gaps and
127,241,461 bp excluding gaps. </P>
<P>
The sequence is distributed among the following chromosomes: 2L, 2R,
2h (random only), 3L, 3R, 3h (random only), 4, X, Xh (random only), Yh
(random only), M (mitochondria) and U.
Sequence that could not be reliably ordered within a chromosome
<em>N</em> has been placed on chr<em>N</em>_random. Heterochromatin
that has been localized to a specific chromosome can be found in
chr<em>N</em>h_random.  For more information about the
<em>D. simulans</em> assembly and statistics, see the WUSTL Genome
Sequencing Center
<A HREF="http://genome.wustl.edu/genomes/view/drosophila_simulans_white_501/"
TARGET=_blank><em>Drosophila simulans</em></A> web page.</P>
<P>
Downloads of the droSim1 data and annotations may be obtained from the 
UCSC Genome Browser 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/droSim1/" TARGET=_blank>FTP server</A>
or <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#droSim">Downloads</A> page. 
The droSim1 annotation tracks were generated by UCSC and 
collaborators worldwide. Thanks to the Genome Sequencing Center, WUSTL School
of Medicine for providing the sequence of this genome.  See the 
<A HREF="../goldenPath/credits.html#droSim_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. </P>
<HR>
<H3>Genbank Pipeline Details</H3>
For the purposes of the Genbank alignment pipeline, this assembly is considered to be: <B>low-coverage</B>.
