<p>
Downloads for data in this track are available:
<ul>
<li>
<a href="http://hgdownload.cse.ucsc.edu/goldenPath/eboVir3/multiz160way/">Multiz alignments</a> (MAF format), and phylogenetic trees
<li>
<a href="http://hgdownload.cse.ucsc.edu/goldenPath/eboVir3/phyloP160way/">PhyloP conservation</a> (WIG format)
<li>
<a href="http://hgdownload.cse.ucsc.edu/goldenPath/eboVir3/phastCons160way/">PhastCons conservation</a> (WIG format)
</ul></p>

<H2>Description</H2>
<p>
This track shows multiple alignments of 160 virus sequences,
composed of 158 $Organism sequences and two Marburg virus sequences
aligned to the $Organism reference sequence G3683/KM034562.1.
It also includes measurements of evolutionary conservation using
two methods (<em>phastCons</em> and <em>phyloP</em>) from the
<a href="http://compgen.bscb.cornell.edu/phast/" target="_blank">
PHAST package</a>, for all 160 virus sequences.
The multiple alignments were generated using multiz and
other tools in the UCSC/<a href="http://www.bx.psu.edu/miller_lab/"
target="_blank">Penn State Bioinformatics</a>
comparative genomics alignment pipeline.
Conserved elements identified by phastCons are also displayed in
this track. </p>
<p>
PhastCons (which has been used in previous Conservation tracks) is a hidden
Markov model-based method that estimates the probability that each
nucleotide belongs to a conserved element, based on the multiple alignment.
It considers not just each individual alignment column, but also its
flanking columns.  By contrast, phyloP separately measures conservation at
individual columns, ignoring the effects of their neighbors.  As a
consequence, the phyloP plots have a less smooth appearance than the
phastCons plots, with more "texture" at individual sites.  The two methods
have different strengths and weaknesses.  PhastCons is sensitive to "runs"
of conserved sites, and is therefore effective for picking out conserved
elements.  PhyloP, on the other hand, is more appropriate for evaluating
signatures of selection at particular nucleotides or classes of nucleotides
(e.g., third codon positions, or first positions of miRNA target sites). </p>
<p>
Another important difference is that phyloP can measure acceleration
(faster evolution than expected under neutral drift) as well as
conservation (slower than expected evolution).  In the phyloP plots, sites
predicted to be conserved are assigned positive scores (and shown in blue),
while sites predicted to be fast-evolving are assigned negative scores (and
shown in red).  The absolute values of the scores represent -log p-values
under a null hypothesis of neutral evolution.  The phastCons scores, by
contrast, represent probabilities of negative selection and range between 0
and 1. </p>
<p>
Both phastCons and phyloP treat alignment gaps and unaligned nucleotides as
missing data. </p>

<p>
The data contained in the <a href="../cgi-bin/hgTrackUi?g=cons160way">160 Accessions</a> and the
<a href="../cgi-bin/hgTrackUi?g=strainCons160way">160 Strains</a> tracks are the same. The only
difference between these two tracks are the identifiers used to label the sequences. In the 160
Accessions track, the sequence is labeled using its NCBI <a href="http://www.ncbi.nlm.nih.gov/nuccore"
target="_blank">Nucleotide</a> accession number. In the 160 Strains track, we used a shortened
version of the strain name from the NCBI Nucleotide entry to label each sequence, and when this
was unavailable, we constructed our own using the <tt>DEFINITION</tt>, <tt>/country</tt>, and
<tt>/collection_date</tt> lines from the NCBI record.
</p>


<H2>Display Conventions and Configuration</H2>
<P>
Pairwise alignments of each species to the $Organism genome are
displayed as a series of colored blocks indicating the functional effect of polymorphisms (in pack
mode), or as a wiggle (in full mode) that indicates alignment quality.
In dense display mode, percent identity of the whole alignments is shown in grayscale using
darker values to indicate higher levels of identity.
<P>
In pack mode, regions that align with 100% identity are not shown.   When there is not 100% percent
identity, blocks of four colors are drawn.
<ul>
<li>Red blocks are drawn when a polymorphism in a coding region results in a change in the amino
acid that is generated.</li>
<li>Green blocks are drawn when a polymorphism in a coding region results in no change to the amino
acid that is generated.</li>
<li>Blue blocks are drawn when a polymorphism is outside a coding region.</li>
<li>Pale yellow blocks are drawn when there is no aligning bases to that region in the reference
genome.</li>
</ul>
<P>
Checkboxes on the track configuration page allow selection of the
species to include in the pairwise display.
Configuration buttons are available to select all of the species
(<EM>Set all</EM>), deselect all of the species (<EM>Clear all</EM>), or
use the default settings (<EM>Set defaults</EM>).
<P>
To view detailed information about the alignments at a specific
position, zoom the display in to 30,000 or fewer bases, then click on
the alignment.</P>

<H3>Base Level</H3>
<P>
When zoomed-in to the base-level display, the track shows the base
composition of each alignment.
The numbers and symbols on the Gaps
line indicate the lengths of gaps in the $Organism sequence at those
alignment positions relative to the longest non-$Organism sequence.
If there is sufficient space in the display, the size of the gap is shown.
If the space is insufficient and the gap size is a multiple of 3, a
&quot;*&quot; is displayed; other gap sizes are indicated by &quot;+&quot;.</P>
<P>
Codon translation is available in base-level display mode if the
displayed region is identified as a coding segment. To display this annotation, select the species
for translation from the pull-down menu in the Codon
Translation configuration section at the top of the page. Then, select one of
the following modes:
<UL>
<LI>
<B>No codon translation:</B> The gene annotation is not used; the bases are
displayed without translation.
<LI>
<B>Use default species reading frames for translation:</B> The annotations from
the genome displayed in the <EM>Default species to establish reading frame</EM>
pull-down menu are used to translate all the aligned species present in the
alignment.
<LI>
<B>Use reading frames for species if available, otherwise no translation:</B>
Codon translation is performed only for those species where the region is
annotated as protein coding.
<LI><B>Use reading frames for species if available, otherwise use default species:</B>
Codon translation is done on those species that are annotated as being protein
coding over the aligned region using species-specific annotation; the remaining
species are translated using the default species annotation.
</UL></P>

<H2>Methods</H2>
<P>
Pairwise alignments with the reference sequence were generated for
each sequence using lastz version 1.03.52.
Parameters used for each lastz alignment:
<pre>
# hsp_threshold      = 2200
# gapped_threshold   = 4000 = L
# x_drop             = 910
# y_drop             = 3400 = Y
# gap_open_penalty   = 400
# gap_extend_penalty = 30
#        A    C    G    T
#   A   91  -90  -25 -100
#   C  -90  100 -100  -25
#   G  -25 -100  100  -90
#   T -100  -25  -90   91
# seed=1110100110010101111 w/transition
# step=1
</pre>
Pairwise alignments were then linked into chains using a dynamic programming
algorithm that finds maximally scoring chains of gapless subsections
of the alignments organized in a kd-tree.  Parameters used in
the chaining (axtChain) step: -minScore=10 -linearGap=loose
</P>
<P>
High-scoring chains were then placed along the genome, with
gaps filled by lower-scoring chains, to produce an alignment net.
</P>
<P>
The multiple alignment was constructed from the resulting best-in-genome
pairwise alignments progressively aligned using multiz/autoMZ,
following a simple binary tree phylogeny:
<pre>
(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((
(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((
(KM034562v1 KJ660346v2) KJ660347v2) KJ660348v2) KM034554v1) KM034555v1) 
KM034557v1) KM034560v1) KM233039v1) KM233043v1) KM233045v1) KM233050v1) 
KM233051v1) KM233053v1) KM233056v1) KM233057v1) KM233063v1) KM233069v1) 
KM233070v1) KM233072v1) KM233089v1) KM233092v1) KM233096v1) KM233097v1) 
KM233098v1) KM233099v1) KM233103v1) KM233104v1) KM233109v1) KM233110v1) 
KM233113v1) AF086833v2) AF272001v1) AY142960v1) EU224440v2) KC242791v1) 
KC242792v1) KC242794v1) KC242796v1) KC242798v1) KC242799v1) KC242801v1) 
KM034551v1) KM034553v1) KM034556v1) KM034558v1) KM034559v1) KM034561v1) 
KM233035v1) KM233036v1) KM233037v1) KM233038v1) KM233040v1) KM233041v1) 
KM233042v1) KM233044v1) KM233046v1) KM233047v1) KM233048v1) KM233049v1) 
KM233052v1) KM233054v1) KM233055v1) KM233058v1) KM233059v1) KM233061v1) 
KM233062v1) KM233064v1) KM233065v1) KM233066v1) KM233067v1) KM233068v1) 
KM233071v1) KM233073v1) KM233074v1) KM233075v1) KM233076v1) KM233077v1) 
KM233078v1) KM233079v1) KM233080v1) KM233081v1) KM233082v1) KM233084v1) 
KM233085v1) KM233086v1) KM233087v1) KM233088v1) KM233093v1) KM233094v1) 
KM233095v1) KM233100v1) KM233101v1) KM233102v1) KM233105v1) KM233106v1) 
KM233107v1) KM233108v1) KM233111v1) KM233112v1) KM233114v1) KM233115v1) 
KM233116v1) KM233091v1) NC_002549v1) KM034552v1) KM233060v1) KM233083v1) 
KM233090v1) KM233117v1) KM233118v1) AY354458v1) KC242784v1) KC242785v1) 
KC242786v1) KC242787v1) KC242788v1) KC242789v1) KC242790v1) KC242793v1) 
KC242795v1) KC242797v1) KC242800v1) AF499101v1) JQ352763v1) HQ613402v1) 
HQ613403v1) KM034549v1) KM034550v1) KM034563v1) FJ217162v1) NC_014372v1) 
FJ217161v1) NC_014373v1) KC545395v1) KC545394v1) KC545393v1) KC545396v1) 
FJ621585v1) FJ621584v1) JX477166v1) AY769362v1) AB050936v1) EU338380v1) 
KC242783v2) JX477165v1) AF522874v1) NC_004161v1) FJ621583v1) KC589025v1) 
FJ968794v1) AY729654v1) NC_006432v1) KC545389v1) KC545390v1) KC545391v1) 
KC545392v1) JN638998v1) NC_024781v1) NC_001608v3)
</pre>
<pre>
(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((
(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((
(G3686v1_2014 Guinea_Kissidougou-C15_2014) Guinea_Gueckedou-C07_2014) 
Guinea_Gueckedou-C05_2014) G3676v1_2014) G3676v2_2014) G3677v2_2014) 
G3682v1_2014) EM112_2014) EM120_2014) EM124v1_2014) G3713v2_2014) G3713v3_2014) 
G3724_2014) G3735v1_2014) G3735v2_2014) G3764_2014) G3770v1_2014) G3770v2_2014) 
G3782_2014) G3814_2014) G3818_2014) G3822_2014) G3823_2014) G3825v1_2014) 
G3825v2_2014) G3831_2014) G3834_2014) G3846_2014) G3848_2014) G3856v1_2014) 
AF086833v2_1976) Mayinga_1976) Mayinga_2002) GuineaPig_Mayinga_2007) 
Bonduni_1977) Gabon_1994) 2Nza_1996) 13625Kikwit_1995) 1Ikot_Gabon_1996) 
13709Kikwit_1995) deRoover_1976) EM096_2014) G3670v1_2014) G3677v1_2014) 
G3679v1_2014) G3680v1_2014) G3683v1_2014) EM104_2014) EM106_2014) EM110_2014) 
EM111_2014) EM113_2014) EM115_2014) EM119_2014) EM121_2014) EM124v2_2014) 
EM124v3_2014) EM124v4_2014) G3707_2014) G3713v4_2014) G3729_2014) G3734v1_2014) 
G3750v1_2014) G3750v2_2014) G3752_2014) G3758_2014) G3765v2_2014) G3769v1_2014) 
G3769v2_2014) G3769v3_2014) G3769v4_2014) G3771_2014) G3786_2014) G3787_2014) 
G3788_2014) G3789v1_2014) G3795_2014) G3796_2014) G3798_2014) G3799_2014) 
G3800_2014) G3805v1_2014) G3807_2014) G3808_2014) G3809_2014) G3810v1_2014) 
G3810v2_2014) G3819_2014) G3820_2014) G3821_2014) G3826_2014) G3827_2014) 
G3829_2014) G3838_2014) G3840_2014) G3841_2014) G3845_2014) G3850_2014) 
G3851_2014) G3856v3_2014) G3857_2014) NM042v1_2014) G3817_2014) 
NC_002549v1_1976) EM098_2014) G3750v3_2014) G3805v2_2014) G3816_2014) 
NM042v2_2014) NM042v3_2014) Zaire_1995) Luebo9_2007) Luebo0_2007) Luebo1_2007) 
Luebo23_2007) Luebo43_2007) Luebo4_2007) Luebo5_2007) 1Eko_1996) 
1Mbie_Gabon_1996) 1Oba_Gabon_1996) Ilembe_2002) Mouse_Mayinga_2002) 
Kikwit_1995) 034-KS_2008) M-M_2007) EM095B_2014) EM095_2014) G3687v1_2014) 
Cote_dIvoire_CIEBOV_1994) Cote_dIvoire_1994) Bundibugyo_Uganda_2007) 
Bundibugyo_2007) EboBund-122_2012) EboBund-120_2012) EboBund-112_2012) 
EboBund-14_2012) Reston08-E_2008) Reston08-C_2008) Alice_TX_USA_MkCQ8167_1996) 
reconstructReston_2008) Reston_1996) Yambio_2004) Maleo_1979) Reston09-A_2009) 
Reston_PA_1990) Pennsylvania_1990) Reston08-A_2008) EboSud-639_2012) 
Boniface_1976) Gulu_Uganda_2000) Gulu_2000) EboSud-602_2012) EboSud-603_2012) 
EboSud-609_2012) EboSud-682_2012) Nakisamata_2011) 
Marburg_KitumCave_Kenya_1987) Marburg_MtElgon_Musoke_Kenya_1980)
</pre>
Framing tables from the genes were constructed to enable
visualization of codons in the multiple alignment display.</P>

<h3> Phylogenetic Tree Model</h3>
<p>
Both <em>phastCons</em> and <em>phyloP</em> are phylogenetic methods that rely
on a tree model containing the tree topology, branch lengths representing
evolutionary distance at neutrally evolving sites, the background distribution
of nucleotides, and a substitution rate matrix.
The
<a href="http://hgdownload.cse.ucsc.edu/goldenPath/eboVir3/phastCons160way/eboVir3.phastCons160way.mod"
target="_blank">all-species tree model</a> for this track was
generated using the <em>phyloFit</em> program from the PHAST package
(REV model, EM algorithm, medium precision) using multiple alignments of
4-fold degenerate sites extracted from the 160-way alignment
(msa_view).  The 4d sites were derived from the NCBI gene set,
filtered to select single-coverage long transcripts. </p>
<p>
This same tree model was used in the phyloP calculations; however, the
background frequencies were modified to maintain reversibility.
The resulting tree model:
<a href="http://hgdownload.cse.ucsc.edu/goldenPath/eboVir3/phyloP160way/eboVir3.phyloP160way.mod"
target="_blank">all species</a>.
</p>

<h3> PhastCons Conservation </h3>
<p>
The phastCons program computes conservation scores based on a phylo-HMM, a
type of probabilistic model that describes both the process of DNA
substitution at each site in a genome and the way this process changes from
one site to the next (Felsenstein and Churchill 1996, Yang 1995, Siepel and
Haussler 2005).  PhastCons uses a two-state phylo-HMM, with a state for
conserved regions and a state for non-conserved regions.  The value plotted
at each site is the posterior probability that the corresponding alignment
column was "generated" by the conserved state of the phylo-HMM.  These
scores reflect the phylogeny (including branch lengths) of the species in
question, a continuous-time Markov model of the nucleotide substitution
process, and a tendency for conservation levels to be autocorrelated along
the genome (i.e., to be similar at adjacent sites).  The general reversible
(REV) substitution model was used.  Unlike many conservation-scoring programs,
phastCons does not rely on a sliding window
of fixed size; therefore, short highly-conserved regions and long moderately
conserved regions can both obtain high scores.
More information about
phastCons can be found in Siepel <em>et al</em>, 2005.</p>
<p>
The phastCons parameters used were: expected-length=45,
target-coverage=0.3, rho=0.3.</p>

<h3> PhyloP Conservation </h3>
<p>
The phyloP program supports several different methods for computing
p-values of conservation or acceleration, for individual nucleotides or
larger elements (<a href="http://compgen.bscb.cornell.edu/phast/"
target="_blank">http://compgen.bscb.cornell.edu/phast/</a>).  Here it was used
to produce separate scores at each base (--wig-scores option), considering
all branches of the phylogeny rather than a particular subtree or lineage
(i.e., the --subtree option was not used).  The scores were computed by
performing a likelihood ratio test at each alignment column (--method LRT),
and scores for both conservation and acceleration were produced (--mode
CONACC). </p>

<h3> Conserved Elements </h3>
<p>
The conserved elements were predicted by running phastCons with the
--most-conserved option.  The predicted elements are segments of the alignment
that are likely to have been "generated" by the conserved state of the
phylo-HMM. Each element is assigned a log-odds score equal to its log
probability under the conserved model minus its log probability under the
non-conserved model. The "score" field associated with this track contains
transformed log-odds scores, taking values between 0 and 1000. (The scores
are transformed using a monotonic function of the form a * log(x) + b.) The
raw log odds scores are retained in the "name" field and can be seen on the
details page or in the browser when the track's display mode is set to
"pack" or "full". </p>

<H2>Credits</H2>
<P> This track was created using the following programs:
<UL>
<LI> Alignment tools: blastz, lastz and multiz by Minmei Hou, Scott Schwartz, Robert Harris, and 
Webb Miller of the <A HREF="http://www.bx.psu.edu/miller_lab/"
TARGET=_blank>Penn State Bioinformatics Group</A>
<LI> Conservation scoring: phastCons, phyloP, phyloFit, tree_doctor, msa_view and
other programs in PHAST by
<a href="http://compgen.bscb.cornell.edu/~acs/software.html"
target="_blank">Adam Siepel</a> at Cornell University (original development
done at the Haussler lab at UCSC).
<LI> Chaining and Netting:  axtChain, chainNet by Jim Kent at UCSC
<LI> MAF Annotation tools: mafAddIRows by Brian Raney, UCSC; mafAddQRows
by Richard Burhans, Penn State; genePredToMafFrames by Mark Diekhans, UCSC
<LI> Tree image generator: phyloGif by Galt Barber, UCSC
<LI> Conservation track display: Kate Rosenbloom, Hiram Clawson (wiggle
display), and Brian Raney (gap annotation and codon framing) at UCSC
</UL>
</P>

<h2>References</h2>

<p>
Gire SK, Goba A, Andersen KG, Sealfon RS, Park DJ, Kanneh L, Jalloh S, Momoh M,
Fullah M, Dudas G <em>et al.</em>
<a href="http://www.sciencemag.org/content/345/6202/1369.abstract"
target="_blank">Genomic surveillance elucidates Ebola virus origin and transmission 
during the 2014 outbreak</a>.
<em>Science</em> 2014 Sep 12;345(6202):1369-72.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/25214632" target="_blank">25214632</a>;
<a href="http://www.sciencemag.org/content/345/6202/1369/suppl/DC1"
target="_blank">Supplemental Materials and Methods</a>
</p>

<h3>Phylo-HMMs, phastCons, and phyloP:</h3>
<p>
Felsenstein J, Churchill GA.
<a href="http://mbe.oxfordjournals.org/content/13/1/93.abstract"
target="_blank">A Hidden Markov Model approach to
variation among sites in rate of evolution</a>.
<em>Mol Biol Evol</em>. 1996 Jan;13(1):93-104.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/8583911" target="_blank">8583911</a>
</p>

<p>
Pollard KS, Hubisz MJ, Rosenbloom KR, Siepel A.
<a href="http://genome.cshlp.org/content/20/1/110.long" target="_blank">
Detection of nonneutral substitution rates on mammalian phylogenies</a>.
<em>Genome Res</em>. 2010 Jan;20(1):110-21.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/19858363" target="_blank">19858363</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2798823/" target="_blank">PMC2798823</a>
</p>

<p>
Siepel A, Bejerano G, Pedersen JS, Hinrichs AS, Hou M, Rosenbloom K,
Clawson H, Spieth J, Hillier LW, Richards S, <em>et al.</em>
<a href="http://genome.cshlp.org/content/15/8/1034"
target="_blank">Evolutionarily conserved elements in vertebrate, insect, worm,
and yeast genomes</a>.
<em>Genome Res</em>. 2005 Aug;15(8):1034-50.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/16024819" target="_blank">16024819</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1182216/" target="_blank">PMC1182216</a>
</p>

<p>
Siepel A, Haussler D.
<a href="http://compgen.bscb.cornell.edu/~acs/phylohmm.pdf"
target="_blank">Phylogenetic Hidden Markov Models</a>.
In: Nielsen R, editor. Statistical Methods in Molecular Evolution.
New York: Springer; 2005. pp. 325-351.
</p>

<p>
Yang Z.
<a href="http://www.genetics.org/content/139/2/993.abstract"
target="_blank">A space-time process model for the evolution of DNA
sequences</a>.
<em>Genetics</em>. 1995 Feb;139(2):993-1005.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/7713447" target="_blank">7713447</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1206396/" target="_blank">PMC1206396</a>
</p>

<h3>Chain/Net:</h3>
<p>
Kent WJ, Baertsch R, Hinrichs A, Miller W, Haussler D.
<a href="http://www.pnas.org/content/100/20/11484.abstract"
target="_blank">Evolution's cauldron:
duplication, deletion, and rearrangement in the mouse and human genomes</a>.
<em>Proc Natl Acad Sci U S A</em>. 2003 Sep 30;100(20):11484-9.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/14500911" target="_blank">14500911</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC208784/" target="_blank">PMC208784</a>
</p>

<h3>Multiz:</h3>
<p>
Blanchette M, Kent WJ, Riemer C, Elnitski L, Smit AF, Roskin KM,
Baertsch R, Rosenbloom K, Clawson H, Green ED, <em>et al.</em>
<a href="http://genome.cshlp.org/content/14/4/708.abstract"
target="_blank">Aligning multiple genomic sequences with the threaded blockset aligner</a>.
<em>Genome Res</em>. 2004 Apr;14(4):708-15.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/15060014" target="_blank">15060014</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC383317/" target="_blank">PMC383317</a>
</p>

<p>
Harris RS.
<a href="http://www.bx.psu.edu/~rsharris/rsharris_phd_thesis_2007.pdf"
target="_blank">Improved pairwise alignment of genomic DNA</a>.
<em>Ph.D. Thesis</em>. Pennsylvania State University, USA. 2007.
</p>

<h3>Blastz:</h3>
<p>
Chiaromonte F, Yap VB, Miller W.
<a href="http://psb.stanford.edu/psb-online/proceedings/psb02/chiaromonte.pdf"
target="_blank">Scoring pairwise genomic sequence alignments</a>.
<em>Pac Symp Biocomput</em>. 2002:115-26.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/11928468" target="_blank">11928468</a>
</p>

<p>
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison RC,
Haussler D, Miller W.
<a href="http://genome.cshlp.org/content/13/1/103.abstract"
target="_blank">Human-mouse alignments with BLASTZ</a>.
<em>Genome Res</em>. 2003 Jan;13(1):103-7.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/12529312" target="_blank">12529312</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC430961/" target="_blank">PMC430961</a>
</p>
