<P>
The HIV genome assembly 
is based on sequence deposited into GenBank as of August 1st, 2006.
To be completed later ... Fan.
<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of an 
mRNA, a coordinate range, a gene identifier, or keywords from 
the GenBank description of an mRNA. The following list shows
examples of valid position queries for the SARS coronavirus TOR2 genome. 
See the
<A HREF="../goldenPath/help/hgTracksHelp.html" TARGET=_blank>
User's Guide</A> for more information.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><BR></TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chr1</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays the entire genome for SARS coronavirus TOR2</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>chr1:15720-186200</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays the region between bases 15720 and 186200 showing the protein NP_828870.1.</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>AF391541</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession AF391541</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>CS000003</TD>
       <TD WIDTH=14></TD>
       <TD VALIGN=Top>Displays region of genome with fgenesv+ gene identifier CS000003</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>bovine coronavirus</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs associated with bovine coronavirus </TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>Chouljenko,V.</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author V. Chouljenko</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Chouljenko V</em>, internally it uses 
			the format <em>Chouljenko,V.</em>.</TD>
    </TR>
</TABLE>
<P>
<HR>
<H3>Assembly details</H3>
This browser &mdash; which represents a departure from UCSC's usual focus on vertebrate 
genomes &mdash; shows gene predictions, locations of putative proteins, and 
viral mRNA and protein alignments. Protein structure analysis and 
predictions were determined by using the  
<A HREF="http://www.cse.ucsc.edu/research/compbio/sam.html" TARGET=_blank> 
Sequence Alignment and Modeling (SAM) T02</A>
tool. 
<P>
UCSC does not plan to provide a comprehensive set of browsers for viruses.  
However, this browser will be maintained as long as there is strong scientific  
and public interest in the SARS coronavirus TOR2.</P>
<P>
The SARS sequence and annotation data may be downloaded from the UCSC Genome 
Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/scApr2003/">FTP server</A> or
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#sars">Downloads</A> web page.
The annotation tracks displayed in the Genome Browser were generated by UCSC and 
collaborators worldwide. See the <A HREF="../goldenPath/credits.html#sars_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 

