<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=315 HEIGHT=138>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://www.fugu-sg.org"
    TARGET=_blank><IMG SRC="/images/Takifugu_rubripes.jpg"
	WIDTH=300 HEIGHT=123 ALT="Fugu"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Takifugu Rubripes</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of
(<A HREF="http://www.imcb.a-star.edu.sg/php/venkatesh.php"
      TARGET=_blank>Byrappa Venkatesh, all rights reserved.</A>)
      <P>
    </FONT>
  </TD></TR>
</TABLE>

<P>
The <em> Takifugu rubripes</em> v5.0 (Oct. 2011) whole genome shotgun assembly
was provided by the
<A HREF="http://www.fugu-sg.org/"
TARGET=_blank>International Fugu Genome Sequencing Consortium,</A>
led by JGI and the 
<A HREF="http://www.imcb.a-star.edu.sg/php/main.php"
TARGET=_blank>Singapore Institute of Molecular and Cell Biology</A> (IMCB). 

<P>
The Oct. 2011 <em>Takifugu Rubripes</em> assembly 
(Fugu Genome Sequencing Consortium - FUGU5 (NCBI project 1434, GCA_000180615.2)) was produced by the <A HREF="http://www.fugu-sg.org/"
TARGET=_blank>International Fugu Genome Sequencing Consortium</A>
See also: <A HREF="http://gbe.oxfordjournals.org/content/3/424.abstract"
TARGET=_blank>Genome Biol Evol (2011) 3 424-442.</A>

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
Fugu genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>chrM</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays the complete mitochondrial chromosome chrM</TD>
    </TR>
    <TR><TD VALIGN=Top>chr1:1-1000000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of chr1</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr1:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr1 that spans 2000 bases, starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>AY617120</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number AY617120</TD>
    </TR>
    <TR><TD VALIGN=Top>CA846300</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		CA846300</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <!--
    <TR><TD VALIGN=Top>pseudogene mRNA</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
    -->
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>zahler</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zahler</TD>
    </TR>
    <TR><TD VALIGN=Top>Begemann,G.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author G. Begemann</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Begemann G</em>, internally it
	uses the format <em>Begemann,G.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
The v5 assembly was generated by filling some gaps in the v4 assembly and by
organizing scaffolds into chromosomes based on a genetic map of fugu.
About 72% of the assembly (281,572,362 bases) is organized into 22
chromosomes.  Approximately 28% of the assembly (109,895,906 bases) has not
been ordered in the assembly and is included in the genome browser
as 6,812 scaffolds with their genbank identifier as their name, for example:
<A HREF="http://www.ncbi.nlm.nih.gov/nuccore/HE591497"
TARGET=_blank>HE591497</A>
The fugu mitochondrial sequence is also available as the virtual chromosome 
&quot;chrM&quot;<A HREF="http://www.ncbi.nlm.nih.gov/nuccore/NC_004299"
TARGET=_blank>(NC_004299).</A>
</P>

<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/fr3/">FTP server</A>
or the
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#fugu">Downloads</A> page.
These data have been freely provided by the JGI for use
in the UCSC Genome Browser.

The Fugu browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#fugu_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
<HR>
<H3>Genbank Pipeline Details</H3>
For the purposes of the Genbank alignment pipeline, this assembly is considered to be: <B>well-ordered</B>.
