<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: Credits</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/topbar.html"--></TD></TR>
<!--Intro Tables------------------------------------------------------->
    <TR><TD COLSPAN=2>	
    	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	    <TR><TD>
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		    <TR><TD>
			<FONT SIZE="4"><B>&nbsp; UCSC Genome Browser Acknowledgments</B></FONT>
		    </TD></TR>
		</TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		    <TR><TD WIDTH=10></TD>
			<TD>
			<P>
			The data and software displayed on this site are the result of
			a large collaborative effort among many individuals at UCSC and
			at research institutions around the world. We'd like to 
			acknowledge the scientists and engineers who have contributed
			to the UCSC Genome Browser and its data, and the organizations 
			who have made this project possible through their generous 
			funding.
			<P>
			<UL>
	        	    <LI><A HREF="#human_credits">Human Genome</A>
	        	    <LI><A HREF="#chimp_credits">Chimpanzee Genome</A>
	        	    <LI><A HREF="#rhesus_credits">Rhesus Genome</A>
	        	    <LI><A HREF="#dog_credits">Dog Genome</A>
	        	    <LI><A HREF="#cow_credits">Cow Genome</A>
	        	    <LI><A HREF="#mouse_credits">Mouse Genome</A>
	        	    <LI><A HREF="#rat_credits">Rat Genome</A>
	        	    <LI><A HREF="#opossum_credits">Opossum Genome</A>
	        	    <LI><A HREF="#chicken_credits">Chicken Genome</A>
	        	    <LI><A HREF="#xenTro_credits"><em>X. tropicalis</em> Genome</A>
	        	    <LI><A HREF="#zebrafish_credits">Zebrafish Genome</A>
	        	    <LI><A HREF="#tetraodon_credits">Tetraodon Genome</A>
	        	    <LI><A HREF="#fugu_credits">Fugu Genome</A>
	        	    <LI><A HREF="#ciona_credits"><em>C. intestinalis</em> Genome</A>
	        	    <LI><A HREF="#urchin_credits"><em>S. purpuratus</em> Genome</A>
	        	    <LI><A HREF="#dm_credits"><em>D. melanogaster</em> Genome</A>
	        	    <LI><A HREF="#droSim_credits"><em>D. simulans</em> Genome</A>
	        	    <LI><A HREF="#droSec_credits"><em>D. sechellia</em> Genome</A>
	        	    <LI><A HREF="#droYak_credits"><em>D. yakuba</em> Genome</A>
	        	    <LI><A HREF="#droEre_credits"><em>D. erecta</em> Genome</A>
	        	    <LI><A HREF="#droAna_credits"><em>D. ananassae</em> Genome</A>
	        	    <LI><A HREF="#droPer_credits"><em>D. persimilis</em> Genome</A>
	        	    <LI><A HREF="#dp_credits"><em>D. pseudoobscura</em> Genome</A>
	        	    <LI><A HREF="#droVir_credits"><em>D. virilis</em> Genome</A>
	        	    <LI><A HREF="#droMoj_credits"><em>D. mojavensis</em> Genome</A>
	        	    <LI><A HREF="#droGri_credits"><em>D. grimshawi</em> Genome</A>
	        	    <LI><A HREF="#apiMel_credits"><em>A. mellifera</em> Genome</A>
	        	    <LI><A HREF="#anoGam_credits"><em>A. gambiae</em> Genome</A>
	        	    <LI><A HREF="#ce_credits"><em>C. elegans</em> Genome</A>
	        	    <LI><A HREF="#cb_credits"><em>C. briggsae</em> Genome</A>
	        	    <LI><A HREF="#yeast_credits">Yeast Genome</A>
	        	    <LI><A HREF="#sars_credits">SARS Genome</A>
	        	    <LI><A HREF="#other_credits">Other scientists featured on this site</A>
	        	    <LI><A HREF="#early_credits">Acknowledgements of early support</A>
			</UL>
			<P>
			Click <A HREF="../staff.html">here</A> to view 
			the contributions of the UCSC Genome Browser project team.
			</P>
		    	</TD>
		    	<TD WIDTH=15></TD>
		    </TR>
		</TABLE>
	    <BR>
	    </TD></TR>
	</TABLE>
	<BR>

<!--Human Credits-------------------------------------------------------->
<A NAME="human_credits"></A>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	    <TR><TD>
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	            <TR><TD>
	    		<FONT SIZE="4"><B>&nbsp; Human Genome: Acknowledgements </B></FONT>
	    	    </TD></TR>
		</TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
 	    	    <TR><TD WIDTH=10></TD>
	     		<TD>
			    <TABLE BORDER=0 HEIGHT=25 WIDTH=435 ALIGN="right" CELLPADDING=2 CELLSPACING=2>
		    		<TR><TD><IMG SRC="../images/DOT.gif" WIDTH=10 HEIGHT=1></TD>
				<TD>
				    <TABLE BORDERCOLOR="999999" BORDER=1 CELLPADDING=0 CELLSPACING=0 HEIGHT=20 ALIGN="right">
			    	        <TR><TD>
			    		    <TABLE BORDERCOLOR="333333" BORDER=1 CELLPADDING=3 CELLSPACING=3 HEIGHT=20>
					        <TR><TD ROWSPAN=2 BGCOLOR="CECE9C" WIDTH="10%">
				    			    <A HREF="http://www.hhmi.org" TARGET=_blank>
				    			    <IMG SRC="../images/HHMI.gif" BORDER=0 ALT="HHMI"></A></TD>
						<TD BGCOLOR="002D59" HEIGHT="38">	
				    			    <A HREF="http://www.ucop.edu/california-institutes" TARGET=_blank>
				    			    <IMG SRC="../images/CISI.gif" BORDER=0 ALT="CISI"></A></TD></TR>
						<TR><TD BGCOLOR="D9E4F8" >			
				    			    <A HREF="http://www.nhgri.nih.gov" TARGET=_blank>
				    			    <IMG SRC="../images/NHGRI.gif" BORDER=0 ALT="NHGRI"></A></TD>
						</TR>
			    		   </TABLE>			
				       </TD></TR>
				    </TABLE>		
				</TD></TR>
		    	    </TABLE>	
			<P>The UCSC Human Genome Browser Project is conducted in collaboration with the 
			<A HREF="http://www.genome.gov/page.cfm?pageID=10001694" TARGET=_blank>International Human 
			Genome Project</A>. We have worked with many collaborators at other institutions
			to produce and annotate this reference sequence of the human genome, as described in our 
			<A HREF="pubs.html">recent publications</A>.
			The UCSC Human Genome Project is made possible by a grant from the 
			<A HREF="http://www.nhgri.nih.gov" TARGET=_blank>National Human Genome Research Institute</A>,
			directed by Francis Collins, and generous support from the 
			<A HREF="http://www.hhmi.org" TARGET=_blank>Howard Hughes Medical Institute</A> and the 
			<A HREF="http://www.ucop.edu/california-institutes" TARGET=_blank>California Institutes for 
			Science and Innovation</A>.</P>
			<P>
			The Human Genome Browser is produced by the 
			<A HREF="../staff.html">UCSC Genome Bioinformatics 
			Group</A>. Annotations are produced by UCSC and 
			collaborators at other institutions. To review 
			the acknowledgements for an annotation, consult the
			&quot;Credits&quot; section on the annotation track's 
			description page in the Genome Browser.
			<P>
		    	The hg18 UCSC Genome Browser was produced by Fan Hsu,
		    	Hiram Clawson, Kate Rosenbloom, Ann Zweig, Kayla Smith, 
			Robert Kuhn, Brooke Rhead, Rachel Harte, Archana 
			Thakkapallayil, and  Donna Karolchik 
		        </TD>
		        <TD WIDTH=15></TD>
	  	    </TR>
	        </TABLE>
	    <BR>
	    </TD></TR>
        </TABLE>
	<BR>

<!--Human Data Sources--------------------------------------------------->
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	    <TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD>
		<FONT SIZE="4"><B>&nbsp; Human Genome: Data Sources</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The human genome sequence data we use have been generated by
		    laboratories belonging to the <A HREF="labs.html">Human Genome 
		    Sequencing Consortium</A>. 
		    Periodic freezes of this data are generated by 
		    <A HREF="http://www.ncbi.nlm.nih.gov/" TARGET=_blank>NCBI</A>.
		    EST, mRNA, BACend and other types of data used for the assembly and annotation are taken
		    from <A HREF="http://www.ncbi.nlm.nih.gov/Ftp/index.html" TARGET=_blank>GenBank</A>
		    /<A HREF=" http://www.ebi.ac.uk/embl/" TARGET=_blank>EMBL</A>
		    /<A HREF="http://www.ddbj.nig.ac.jp/" TARGET=_blank>DDBJ</A>.
		    Clone maps were created by the individual sequencing centers. 
		    <P>
		    The assembly of the finished chromosomes is taken directly from files edited by 
		    Greg Schuler at NCBI. The original files are produced by the institutions and 
		    people listed in the table below.
		    <P>

	<!--finished chromosome table-->
		    <TABLE BORDER=1 BORDERCOLOR="#333333" CELLPADDING=0 WIDTH="100%">
			<TR><TD>
			<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
			    <TR><TD COLSPAN=4><B>Finished Chromosomes</B></TD></TR>
			    <TR><TD nowrap><FONT color="#006666"><B>Chr. #</B></FONT></TD>
				<TD nowrap><FONT color="#006666"><B>Sequencing Centers</B></FONT></TD>
				<TD nowrap><FONT color="#006666"><B>AGP File</B></FONT></TD></TR>
			    <TR valign="top"><TD align="center">1</TD>
				<TD>
				    <UL>
				    	<LI><A HREF="http://www.sanger.ac.uk/" TARGET=_blank> The Sanger Institute</A>, Cambridge, UK.
				    	<LI><A HREF="http://www.genome.washington.edu/UWGC/" TARGET=_blank>University 
						of Washington Genome Center</A>, Seattle, WA, USA
				    </UL>
				</TD>
				<TD><A HREF="mailto:jrh@sanger.ac.uk">Jane Rogers</A></TD>
			    </TR>
			    <TR valign="top">
				<TD align="center">2</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Homo%20sapiens&GROUP=1"
					 	TARGET=_blank>Washington University School of Medicine Genome 
						Sequencing Center</A>, St Louis, MO, USA
				    </UL></TD> 
				<TD nowrap><A HREF="mailto:rwilson@watson.wustl.edu">Rick Wilson</A></TD>
			    </TR>
			    <TR valign="top">
				<TD align="center">3</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://www.hgsc.bcm.tmc.edu/" TARGET=_blank> Baylor 
						College of Medicine Human Genome Sequencing Center</A>, Houston, TX, USA
					<LI><A HREF="http://www.chgb.org.cn/en_index.htm" TARGET=_blank>Beijing 
						Human Genome Center</A>, Beijing, China 
					<LI><A HREF="http://www.genome.washington.edu/UWGC/" TARGET=_blank>University 
						of Washington Genome Center</A>, Seattle, WA, USA
				    </UL>
				</TD> 
				<TD><A HREF="mailto:sscherer@bcm.tmc.edu">Steve Scherer</A></TD>
			    </TR>
			    <TR valign="top">
				<TD align="center">4</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Homo%20sapiens&GROUP=1" TARGET=_blank>Washington 
						University School of Medicine Genome Sequencing Center</A>, St Louis, MO, USA
					<LI><A HREF="http://shgc.stanford.edu/" TARGET=_blank>Stanford 
						Human Genome Center</A>, Palo Alto, CA, USA
				    </UL>
				</TD>
				<TD nowrap><A HREF="mailto:rwilson@watson.wustl.edu">Rick Wilson</A></TD>
			    </TR>
			    <TR valign="top">
				<TD align="center">5</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://genome.jgi-psf.org/chr5/chr5.home.html" TARGET=_blank> Joint Genome Institute</A>, 
						U.S. Department of Energy, Walnut Creek, CA, USA		
					<LI><A HREF="http://shgc.stanford.edu/" TARGET=_blank>Stanford Human Genome 
						Center (Myers)</A>, Palo Alto, CA, USA
				    </UL>
				</TD>
				<TD><A HREF="mailto:jeremy@shgc.stanford.edu">Jeremy Schmutz</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">6</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://www.sanger.ac.uk/" TARGET=_blank> The Sanger Institute</A>, 
						Cambridge, UK.
				    </UL></TD> 
				<TD><A HREF="mailto:jrh@sanger.ac.uk">Jane Rogers</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">7</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Homo%20sapiens&GROUP=1" TARGET=_blank>Washington 
						University School of Medicine Genome Sequencing Center</A>, St Louis, USA
					<LI><A HREF="http://www.genome.washington.edu/UWGC/" TARGET=_blank>University 
						of Washington Genome Center</A>, Seattle, WA, USA
				    </UL>
				</TD> 
				<TD nowrap><A HREF="mailto:rwilson@watson.wustl.edu">Rick Wilson</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">8</TD>
				<TD>
				    <UL>
					<A HREF="http://www.broad.mit.edu/" TARGET=_blank>The Broad 
					Institute</A>, USA
					<LI><A HREF="http://www.dmb.med.keio.ac.jp" TARGET=_blank>Keio University</A>, 
						Tokyo, Japan
					<LI><A HREF="http://genome.imb-jena.de/" TARGET=_blank>Institute of Molecular 
						Biology (IMB)</A>, Jena, Germany
				    </UL>
				</TD> 
				<TD><A HREF="mailto:chad@genome.wi.mit.edu">Chad Nusbaum</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">9</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://www.sanger.ac.uk/" TARGET=_blank> The Sanger Institute</A>, 
						Cambridge, UK.
					<LI><A HREF="http://genome.gbf.de/" TARGET=_blank> Gesellschaft für 
						Biotechnologische Forschung (GBF)</A>, Braunschweig, Germany 
				    </UL>
				</TD>
				<TD><A HREF="mailto:jrh@sanger.ac.uk">Jane Rogers</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">10</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://www.sanger.ac.uk/" TARGET=_blank> The Sanger Institute</A>, 
						Cambridge, UK.
					<LI><A HREF="http://www.genomecorp.com/" TARGET=_blank> Genome Therapeutics 
						Corporation (GTC)</A>, Waltham, MA, USA
				    </UL></TD> 
				<TD><A HREF="mailto:jrh@sanger.ac.uk">Jane Rogers</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">11</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://hgp.gsc.riken.go.jp/" TARGET=_blank>RIKEN Human Genome 	
						Research Group</A>, Saitama, Japan 
					<LI><A HREF="http://www.broad.mit.edu/" TARGET=_blank>The Broad 
						Institute</A>, USA
				    </UL>
				</TD>
				<TD><A HREF="mailto:taylor@gsc.riken.go.jp">Todd Taylor</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">12</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://www.hgsc.bcm.tmc.edu/" TARGET=_blank> Baylor College of 
						Medicine Human Genome Sequencing Center</A>, Houston, TX, USA
				    </UL>
				</TD>
				<TD><A HREF="mailto:sscherer@bcm.tmc.edu">Steve Scherer</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">13</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://www.sanger.ac.uk/" TARGET=_blank> The Sanger Institute</A>, 
						Cambridge, UK.
				    </UL>
				</TD> 
				<TD><A HREF="mailto:jrh@sanger.ac.uk">Jane Rogers</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">14</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://www.genoscope.cns.fr/" TARGET=_blank>Genoscope National 
						Sequencing Centre</A>, Evry, France
					<LI><A HREF="http://www.systemsbiology.org/" TARGET=_blank>Institute for Systems 
						Biology (ISB)</A>, Seattle, WA
					<LI><A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Homo%20sapiens&GROUP=1" TARGET=_blank>Washington 
						University School of Medicine Genome Sequencing Center</A>, St Louis, USA
					<LI><A HREF="http://hgp.gsc.riken.go.jp/" TARGET=_blank>RIKEN Human Genome 
						Research Group</A>, Saitama, Japan 
				    </UL>
				</TD> 
				<TD><A HREF="mailto:jsbach@genoscope.cns.fr">Jean Weissenbach</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">15</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://www.broad.mit.edu/" TARGET=_blank>The Broad 
						Institute</A>, USA
					<LI><A HREF="http://genome.imb-jena.de/" TARGET=_blank>Institute of Molecular 
						Biology (IMB)</A>, Jena, Germany
					<LI><A HREF="http://www.systemsbiology.org/" TARGET=_blank>Institute for Systems 
						Biology (ISB)</A>, Seattle, WA
				    </UL>
				</TD> 
				<TD><A HREF="mailto:chad@genome.wi.mit.edu">Chad Nusbaum</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">16</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://www.jgi.doe.gov/" TARGET=_blank> Joint Genome Institute 
						(JGI)</A>, U.S. Department of Energy, Walnut Creek, CA, USA		
					<LI><A HREF="http://shgc.stanford.edu/" TARGET=_blank>Stanford Human Genome 
						Center</A>, Palo Alto, CA, USA
					<LI><A HREF="http://www.tigr.org/" TARGET=_blank>The Institute for Genomic 
						Research (TIGR)</A>, Rockville, MD, USA
					<LI><A HREF="http://www.sanger.ac.uk/" TARGET=_blank> The Sanger Institute</A>, 
						Cambridge, UK.
				    </UL>
				</TD>
				<TD><A HREF="mailto:jeremy@shgc.stanford.edu">Jeremy Schmutz</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">17</TD>
				<TD>
				    <UL>
					<A HREF="http://www.broad.mit.edu/" 
					TARGET=_blank>The Broad Institute</A> , USA
				    </UL>
				</TD>
				<TD><A HREF="mailto:chad@genome.wi.mit.edu">Chad Nusbaum</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">18</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://www.broad.mit.edu/" 
					TARGET=_blank>The Broad Institute</A>, USA
					<LI><A HREF="http://hgp.gsc.riken.go.jp/" TARGET=_blank>RIKEN Human Genome 
						Research Group</A>, Saitama, Japan 
				    </UL>
				</TD>
				<TD><A HREF="mailto:chad@genome.wi.mit.edu">Chad Nusbaum</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">19</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://genome.jgi-psf.org/Chr19ncbi34/Chr19ncbi34.home.html" TARGET=_blank> Joint Genome Institute 
						(JGI)</A>, 
					U.S. Department of Energy, Walnut Creek, CA, USA		
					<LI><A HREF="http://shgc.stanford.edu/" TARGET=_blank>Stanford Human Genome 
						Center</A>, Palo Alto, CA, USA
				    </UL>
				</TD> 
				<TD><A HREF="mailto:jeremy@shgc.stanford.edu">Jeremy Schmutz</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">20</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://www.sanger.ac.uk/" TARGET=_blank> The Sanger Institute</A>, 
						Cambridge, UK.
				    </UL>
				</TD>
				<TD><A HREF="mailto:jgrg@sanger.ac.uk">James Gilbert</A></TD>
			     </TR>
			     <TR valign="top"><TD align="center">21</TD>
			    	<TD>
				     <UL>
					<LI><A HREF="http://hgp.gsc.riken.go.jp/" TARGET=_blank>RIKEN Human Genome Research Group</A>, 
						Japan
					<LI><A HREF="http://www.mpg.de/" TARGET=_blank>Max Planck Institute for Molecular 
						Genetics</A>, Berlin, Germany  
					<LI><A HREF="http://genome.gbf.de/" TARGET=_blank> Gesellschaft für 
						Biotechnologische Forschung (GBF)</A>, Braunschweig, Germany 
					<LI><A HREF="http://www.dmb.med.keio.ac.jp/" TARGET=_blank>Keio University School 
						of Medicine</A>, Tokyo, Japan
					<LI><A HREF="http://chr21.molgen.mpg.de/Chr21_consortium.html" TARGET=_blank>Chromosome 
						21 Consortium</A>
				    </UL>
				</TD>
				<TD nowrap><A HREF="mailto:taylor@gsc.riken.go.jp">Todd Taylor</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">22</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://www.sanger.ac.uk/" TARGET=_blank> The Sanger Institute</A>, 
						Cambridge, UK.
					<LI><A HREF="http://cheminfo.chem.ou.edu/" TARGET=_blank> The University of 
						Oklahoma</A>, USA.
					<LI><A HREF="http://www.dmb.med.keio.ac.jp/" TARGET=_blank> Keio University School 
						of Medicine</A>, Tokyo, Japan
				    </UL>
				</TD> 
				<TD><A HREF="mailto:jrh@sanger.ac.uk">Jane Rogers</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">X</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://www.sanger.ac.uk/" TARGET=_blank> The Sanger Institute</A>, 
						Cambridge, UK.
					<LI><A HREF="http://www.hgsc.bcm.tmc.edu/" TARGET=_blank> Baylor College of 
						Medicine Human Genome Sequencing Center</A>, Houston, TX, USA
					<LI><A HREF="http://genome.imb-jena.de/" TARGET=_blank>Institute of Molecular 
						Biology (IMB)</A>, Jena, Germany
					<LI><A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Homo%20sapiens&GROUP=1" TARGET=_blank>Washington 
						University School of Medicine Genome Sequencing Center</A>, St Louis, USA
					<LI><A HREF="http://www.mpg.de/" TARGET=_blank>Max Planck Institute for Molecular 
						Genetics</A>, Berlin, Germany  
			    	    </UL>
				</TD>
				<TD><A HREF="mailto:jrh@sanger.ac.uk">Jane Rogers</A></TD>
			    </TR>
			    <TR valign="top"><TD align="center">Y</TD>
				<TD>
				    <UL>
					<LI><A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Homo%20sapiens&GROUP=1" TARGET=_blank>Washington 
						University School of Medicine Genome Sequencing Center</A>, St Louis, USA
				    </UL>
				</TD>
				<TD nowrap><A HREF="mailto:rwilson@watson.wustl.edu">Rick Wilson</A></TD>
			    </TR>
			</TABLE>
		    	</TD></TR>
		    </TABLE>
	        </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
        </TD></TR>
    </TABLE>
    <BR>

<!--Chimp Credits-------------------------------------------------------->
<A NAME="chimp_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Chimpanzee Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC Chimpanzee Genome Browser displays data from the 
		    13 Nov. 2003 chimpanzee Arachne draft assembly (Build 1 
		    Version 1), produced by the Chimpanzee Genome Sequencing 
		    Consortium with the following acknowledgements:
		    <UL>
		    <LI><B>Funding:</B> <A HREF="http://www.nhgri.nih.gov/" 
		    TARGET=_blank>National Human Genome Research Institute</A> 
		    (NHGRI)
		    <BR>
	  	    <LI><B>Sequencing/Assembly:</B>
 		    The sequence was assembled by teams led by Eric 
		    Lander, Ph.D., at 
		    <A HREF="http://www.broad.mit.edu/" 
		    TARGET=_blank>The Broad Institute</A>, 
  		    and Richard K. Wilson, Ph.D., 
		    at the Genome Sequencing Center, 
		    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Pan%20troglodytes&GROUP=1"
		    TARGET=_blank>Washington University School of Medicine</A>, 
		    St. Louis, MO, USA
		    <BR>
		    <LI><B>Alignments:</B> 
		    LaDeana Hillier, <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Pan%20troglodytes&GROUP=1"
		    TARGET=_blank>Washington University School of Medicine</A>, 
		    and 
		    <A HREF="http://www.broad.mit.edu/" TARGET=_blank>The Broad Institute</A>. 
		    <BR>
		    <LI><B>Annotations:</B> 
		    <A HREF="http://genes.cs.wustl.edu/BrentLab/MB-Lab-index.html" 
		    TARGET=_blank>Washington University School of Medicine</A>
		    St. Louis, MO, USA, 
		    Research Group on Biomedical Informatics of IMIM/UPF 
		    (<A HREF="http://www.imim.es/grib/eng/default.htm" 
		    TARGET=_blank>GRIB</A>), 
		    <A HREF="http://www.gsc.riken.go.jp" TARGET=_blank>RIKEN 
		    Genomic Sciences Center</A>. 
		    <LI><B>UCSC Chimp Genome Browser:</B> Kate Rosenbloom, Jim 
		    Kent, Hiram Clawson, Heather Trumbower, Robert Kuhn, Donna 
		    Karolchik, and the Genome Browser sysadmin team.
		    </UL>
		    <P>
		    <HR>
		    <A NAME="chimp_use"></A>
		    The chimpanzee sequence is made freely available before scientific 
		    publication from the Chimpanzee Genome Sequencing Consortium with the 
		    following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers analyzing
			particular genes and regions if the providers of this
			data (the Chimpanzee Genome Sequencing Consortium) are properly 
			acknowledged.
			<LI>
			The centers producing the data reserve the right to publish the
			initial large-scale analyses of the data set, including large-scale
			identification of regions of evolutionary conservation and
			large-scale genomic assembly. Large-scale refers to regions with
			size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--Rhesus Credits---------------------------------------------------->
<A NAME="rhesus_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Rhesus Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC Rhesus Genome Browser (rheMac2) displays data from 
		    the Jan. 2006 draft assembly of the rhesus monkey, version 
		    Mmul_051212 from Baylor College of Medicine Human Genome 
		    Sequencing Center (BCM HGSC). 
		    The rheMac1 Genome Browser displays data from 
		    the Jan. 2005 draft assembly of the rhesus monkey, version 
		    Mmul_0.1.</P>
		    <P>The sequencing and assembly
		    of the <em>Macaca mulatta</em> genome is a project 
		    of the Macaque Genome Sequencing Consortium led by the 
		    Baylor College of Medicine Human Genome Sequencing Center, 
		    in collaboration with the J. Craig Venter Institute Joint 
		    Technology Center, and the Genome Sequencing Center at 
		    Washington University School of Medicine, St. Louis. 
		    This assembly is provided 
		    with the following acknowledgements:
		    <UL>
	   	    <LI>
		    <B>Funding: </B> 
		    <A HREF="http://www.genome.gov" 
		    TARGET=_blank>National Human Genome Research Institute</A>
		    (NHGRI), USA
		    <LI>
		    <B>Sequencing/Assembly: </B>
		    <A HREF="http://www.hgsc.bcm.tmc.edu/" 
		    TARGET=_blank>BCM HGSC</A>, Houston, TX, USA in 
		    collaboration with:
		      <UL>
		      <LI>
		      <A HREF="http://www.genomicresource.org/" 
		      TARGET=_blank>J. Craig Venter Science Foundation Joint 
		      Technology Center</A>, Rockville, MD, USA 
		      <LI>
		      <A HREF="http://genome.wustl.edu/" TARGET=_blank>Genome 
		      Sequencing Center at Washington University School of
		      Medicine</A>, St.  Louis, MO, USA
		      </UL>
		    <LI>
		    <B>BAC resources: </B>
		    <A HREF="http://bacpac.chori.org/rhesus250.htm"
		    TARGET=_blank>Children's Hospital Oakland Research
		    Institue (CHORI)</A>, Oakland, CA, USA
		    <LI>
		    <B>BAC-based fingerprint map: </B>
		    <A HREF="http://www.bcgsc.ca/lab/mapping/data"
		    TARGET=-blank>Genome Sciences Centre</A>, Vancouver, B.C.
		    <LI><B>UCSC Rhesus Genome Browser (rheMac2) and Initial 
		    Annotations:</B> 
		    <A HREF="../staff.html" TARGET=_blank>UCSC Genome 
		    Bioinformatics Group</A>, Santa Cruz, CA, USA - 
		    Robert Baertsch, Kayla Smith, Ann Zweig, Robert Kuhn, and 
		    Donna Karolchik. 
		    <LI><B>UCSC Rhesus Genome Browser (rheMac1) and Initial 
		    Annotations: </B> 
		    <A HREF="../staff.html" TARGET=_blank>UCSC Genome 
		    Bioinformatics Group</A>, Santa Cruz, CA, USA - 
		    Robert Baertsch, Galt Barber, and Donna Karolchik. 
		    </UL>
		    <P>
		    For more information on the rhesus genome project, see
		    the BCM HGSC 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/rmacaque/"
		    TARGET=_blank>Rhesus Monkey Genome Project</A> web page.
		    <A NAME="rhesus_use"></A>
		    <P>
		    <HR>
		    These data are made available before scientific publication 
		    with the following understanding:
		    <UL>
		    <LI>The data may be freely downloaded, used in analyses, 
		    and repackaged in databases.
		    <LI>Users are free to use the data in scientific papers 
		    analyzing particular genes and regions if the providers of 
		    this data (Baylor College of Medicine Human Genome 
		    Sequencing Center and the Rhesus Macaque Genome Sequencing 
		    Consortium) are properly acknowledged. Please cite the 
		    BCM-HGSC web site or publications from BCM-HGSC referring to
		    the genome sequence.
		    <LI>The BCM-HGSC and RMGSC plan to publish the assembly and 
		    genomic annotation of the dataset, including large-scale 
		    identification of regions of evolutionary conservation.
		    <LI>This is in accordance with, and with the understandings 
		    in the 
		    <A HREF="http://www.wellcome.ac.uk/en/1/awtpubrepdat.html"
		    TARGET=_blank>Fort Lauderdale meeting</A> discussing 
		    Community Resource Projects and the resulting
		    <A HREF="http://www.genome.gov/page.cfm?pageID=10506537"
		    TARGET=_blank>NHGRI policy statement</A>.
		    <LI>Any redistribution of the data should carry this notice.
		    </UL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--Dog Credits-------------------------------------------------------->
<A NAME="dog_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Dog Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC Dog Genome Browser displays data from the 
		    July 2004 v1.0 (canFam1) and May 2005 v2.0 (canFam2) draft 
		    assemblies of the domestic dog (<em>Canis familiaris</em>). 
		    These assemblies are provided with the following 
		    acknowledgements:
		    <UL>
			<LI><B>Funding:</B> The 
			<A HREF="http://www.genome.gov/" 
			TARGET=_blank>National Human Genome Research Institute</A>
			<BR>
	  		<LI><B>Project Leadership:</B>
			<A HREF="http://www.broad.mit.edu/" TARGET=_blank>The Broad 
			Institute</A>, Cambridge, MA, USA
	  		<LI><B>Sequencing/Assembly:</B>
			<A HREF="http://www.broad.mit.edu/" TARGET=_blank>The Broad 
			Institute</A>, Cambridge, MA, USA - 
			Team led by by Kerstin Lindblad-Toh, Ph.D.
			<LI><B>Plasmid and fosmid libraries:</B> 
			<A HREF="http://www.broad.mit.edu/" TARGET=_blank>The Broad 
			Institute</A>, Cambridge, MA, USA
			<LI><B>BAC libraries:</B> 
			<A HREF="http://www.chori.org/" TARGET=_blank>Children's
			Hospital Oakland Research Institute</A> (CHORI), Oakland,
			CA, USA
			<LI><B>BAC end sequencing:</B> 
			<A HREF="http://www.agencourt.com/" 
			TARGET=_blank>Agencourt Bioscience</A>, Beverly, MA, USA
			<LI><B>RH map:</B> 
			<A HREF="http://www-recomgen.univ-rennes1.fr/Dogs/maquette.html" 
			TARGET=_blank>Centre National de la Recherche 
			Scientifique</A> (CNRS), France
			<LI><B>FISH mapping:</B> <A HREF="http://www.ncsu.edu/"
			TARGET=_blank>North Carolina State University</A> (NCSU),
			Raleigh, NC, USA
			<BR>
			<LI><B>Dog Selection:</B>
			<A HREF="http://www.fhcrc.org/science/dog_genome/"
			TARGET=_blank>Fred Hutchinson Cancer Research Center</A>
			(FHCRC), Seattle, WA, USA
			<LI><B>UCSC Dog Genome Browser/Initial Annotations 
			(canFam2):</B> 
			<A HREF="../staff.html" 
			TARGET=_blank>UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Angie Hinrichs, Jennifer Jackson, and Donna Karolchik
			<LI><B>UCSC Dog Genome Browser/Initial Annotations 
			(canFam1):</B> 
			<A HREF="../staff.html" 
			TARGET=_blank>UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Angie Hinrichs, Robert Kuhn and Donna Karolchik
		    </UL>
		    <P>
		    <HR>
		    <A NAME="dog_use"></A>
		    The dog sequence is made freely available before scientific
		    publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers 
			of these data (The Broad Institute and
			Agencourt Bioscience) are properly acknowledged.
			<LI>
			The centers producing the data reserve the right to 
			publish the initial large-scale analyses of the data set, 
			including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--Cow Credits------------------------------------------------------->
<A NAME="cow_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Cow Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC Cow Genome Browser displays data obtained 
		    from the Baylor College of Medicine Human Genome Sequencing
		    Center (BCM HGSC). The Mar. 2005 draft assembly is based on 
		    Baylor version Btau_2.0; the Sep. 2004 assembly is based
		    on Baylor version Btau_1.0. These data are provided with the
		    following acknowledgements:
		    <UL>
	   	    <LI><B>Funding</B> - For a list of the many agencies that
		    funded the cow sequencing project, see the BCM HGSC
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/bovine/"
		    TARGET=_blank>web page</A>.
		    <LI><B>Sequencing/Assembly: </B>
		        <A HREF="http://www.hgsc.bcm.tmc.edu/" 
			TARGET=_blank>BCM HGSC</A>, Houston, TX, USA <BR>
		    <LI><B>BAC library DNA: </B>
			Dr. Michael MacNeil's laboratory at the
			<A HREF="http://www.ars.usda.gov/main/main.htm"
			TARGET=_blank>USDA Agricultural Research Service</A>, 
		    	Miles City, MT, USA
		    <LI><B>Whole genome shotgun sequence DNA: </B>
			Dr. Timothy Smith's laboratory at the
			<A HREF="http://www.marc.usda.gov/"
			TARGET=_blank>U.S. Meat Animal Research Center</A>, 
		 	Clay Center, NE, USA
		    <LI><B>BAC-based fingerprint map: </B>
			<A HREF="http://www.bcgsc.ca/lab/mapping/bovine"
			TARGET=-blank>Genome Sciences Centre</A>, Vancouver, 
			B.C.
		    <LI><B>UCSC Cow Genome Browser/Initial Annotations (bosTau2): </B> 
			<A HREF="../staff.html" 
			TARGET=_blank>UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Galt Barber, Brian Raney, Mark Diekhans, Jennifer Jackson,
			and Donna Karolchik. 
		    <LI><B>UCSC Cow Genome Browser/Initial Annotations (bosTau1): </B> 
			<A HREF="../staff.html" 
			TARGET=_blank>UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Heather Trumbower, Jim Kent, Hiram Clawson, Angie
			Hinrichs, Brian Raney, Mark Diekhans, Robert Kuhn,
			Ali Sultan-Qurraie and Donna Karolchik. 
		    </UL>
		    <P>
		    For more information on the bovine genome project, see
		    the BCM HGSC 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/bovine/"
		    TARGET=_blank>Bovine Genome Project</A> web page.
		    <P>
		    <HR>
		    <A NAME="cow_use"></A>
		    To review the conditions for use regarding these 
		    data, see the BCM HGSC 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html"
		    TARGET=_blank>Conditions for Use</A> web page.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>


<!--Mouse Credits-------------------------------------------------------->
<A NAME="mouse_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Mouse Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		<P>The UCSC Mouse Genome Project is conducted in collaboration
	 	with the Mouse Sequencing Consortium and the Mouse Genome 
		Sequencing Consortium. The mouse draft assembly displayed in 
		the Genome Browser is assembled at NCBI.
		<P>
		<HR>
		The initial 3X coverage of the mouse genome was produced by 
		the Mouse Sequencing Consortium, with the following 
		acknowledgements: 
		<UL>
		<LI><B>Project Oversight:</B> Arthur Holden and Francis 
		Collins
		<LI><B>Funding:</B> 
		  <UL>
		  <LI><A HREF="http://www.gsk.com/" 
		      TARGET=_blank>GlaxoSmithKline</A>
		  <LI><A HREF="http://www.mgri.org/home.html" 
		      TARGET=_blank>Merck Genome Research Institute</A> 
		  <LI><A HREF="http://www.affymetrix.com/" 
		      TARGET=_blank>Affymetrix, Inc.</A> 
		  <LI><A HREF="http://www.wellcome.ac.uk/" 
		      TARGET=_blank>The Wellcome Trust</A>
		  <LI><A HREF="http://www.nih.gov/" TARGET=_blank>The National
		      Institutes of Health</A>
	  	  </UL>
		<LI><B>Sequencing:</B> 
		  <UL>
		  <LI><A HREF="http://www.broad.mit.edu/" 
		      TARGET=_blank>The Broad Institute</A>
		  <LI><A HREF="http://medicine.wustl.edu/" 
		      TARGET=_blank>Washington University St. Louis Genome
		      Sequencing Center</A>
		  <LI><A HREF="http://www.sanger.ac.uk"
			TARGET=_blank>The Wellcome Trust Sanger Institute</A>
		  </UL>
	    	</UL></P>
		<P>
		<HR>
		Later releases of the assembly and map have been produced 
		by the Mouse Genome Sequencing Consortium, with these acknowledgements: 
		<UL>  
		    <LI><B>Funding:</B>
		    <UL>
		    <LI><A HREF="http://www.nih.gov/" TARGET=_blank>National 
			Institutes of Health</A>
		    <LI><A HREF="http://www.wellcome.ac.uk/" 
			TARGET=_blank>The Wellcome Trust</A>
		    <LI><A HREF="http://www.gsk.com/index.htm" 
			TARGET=_blank>GlaxoSmithKline</A>
		    <LI><A HREF="http://www.merck.com/" TARGET=_blank>The 
			Merck Genome Research Institute</A>
		    <LI><A HREF="http://www.affymetrix.com/index.affx" 
			TARGET=_blank>Affymetrix</A>
		    </UL>
		   
	    	    <LI><B>Sequencing:</B>
		    <UL>
		    <LI><A HREF="http://www.broad.mit.edu/" 
			TARGET=_blank>The Broad Institute's Arachne Group</A>
		    <LI><A HREF="http://www.sanger.ac.uk"
			TARGET=_blank>The Wellcome Trust Sanger Institute</A> 
		    <LI><A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Mus%20musculus&GROUP=2"
		 	TARGET=_blank>Washington University in St. Louis 
			Genome Sequencing Center</A>
		    </UL>
	     	    <LI><B>Physical Map:</B> 
		    <UL>
		    <LI>Mouse BAC libraries - <A HREF="http://bacpac.chori.org/" 
			TARGET=_blank>Children's Hospital Oakland Research
			Institute</A> (CHORI)
		    <LI>BAC fingerprint map - 
			<A HREF="http://www.bcgsc.ca/lab/mapping/mouse" 
			TARGET=_blank>Genome Sciences Centre</A>
		    <LI>BAC end sequences - 
			<A HREF="http://www.tigr.org/tdb/bac_ends/mouse/bac_end_intro.html"
			TARGET=_blank>The Institute of Genome Research</A> 
			(TIGR)
		    </UL>
		    <LI><B>cDNA Sequencing:</B> 
		    <UL>
		    <LI><A HREF="http://genome.gsc.riken.go.jp/"
		    	TARGET=_blank>Riken Institute</A>
		    <LI><A HREF="http://mgc.nci.nih.gov/"
		    	TARGET=_blank>The Mammalian Gene Collection</A>
		    </UL>
	  	    <LI><B>UCSC Mouse Genome Browser (mm7):</B> Hiram Clawson,
		    Fan Hsu, Ann Zweig, Kayla Smith, Robert Kuhn
		    and Donna Karolchik
	  	    <LI><B>Annotations:</B> 
	  	    <LI><B>UCSC Mouse Genome Browser (mm6):</B> Hiram Clawson,
		    Fan Hsu, Jennifer Jackson, Robert Kuhn, Ali 
		    Sultan-Qurraie, Heather Trumbower and Donna Karolchik
	  	    <LI><B>Annotations:</B> 
		    <UL>
		    <LI><A HREF="../staff.html" TARGET=_blank>UCSC Genome
			Bioinformatics Group</A>
		    <LI><A HREF="http://www.jax.org/" TARGET=_blank>The 
			Jackson Laboratory</A>
		    <LI><A HREF="http://www.ncbi.nlm.nih.gov/" 
			TARGET=_blank>NCBI</A>
		    <LI><A HREF="http://www.ensembl.org/" 
			TARGET=_blank>Ensembl</A>
		    <LI><A HREF="http://bio.cse.psu.edu/" 
			TARGET=_blank>Pennsylvania State University</A>
		    <LI><A HREF="http://www-hgc.lbl.gov/" 
			TARGET=_blank>University of California 
			Berkeley/LBNL</A>
		    </UL>
		</UL></P>
		<P>
		<A NAME="mouse_use"></A>
		<HR>
		Mouse genome sequence data are released weekly into a
		public repository maintained by EBI and NCBI. References to 
		these data should cite the following publication: 
		Mouse Genome Sequencing Consortium. 
		<A HREF="http://www.nature.com/nature/mousegenome/" 
		TARGET=_blank>Initial sequencing and comparative analysis of 
		the mouse genome</A>.
		<I>Nature</I>, <B>420</B>, 520-562 (2002).</P>
		</TD>
		<TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--Rat Credits---------------------------------------------------------->
<A NAME="rat_credits"></A>
   <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; Rat Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The UCSC Rat Genome Project is conducted in collaboration 
		    with the Rat Genome Sequencing Consortium.
		    <P>
		    The Rat genome sequence was produced with the following 
		    acknowledgements:
		    <UL>
			<LI><B>Project Coordination:</B> 
			<A HREF="http://www.hgsc.bcm.tmc.edu/" 
			TARGET=_blank>Baylor College of Medicine Human Genome 
			Sequencing Center</A>, Richard Gibbs, director
			<LI><B>Funding:</B> 
			  <UL>
			  <LI><A HREF="http://www.nhgri.nih.gov/" 
			      TARGET=_blank>National Human Genome Research 
			      Institute</A> (NIH)
			  <LI><A HREF="http://www.nhlbi.nih.gov/" 
			      TARGET=_blank>National Heart, Lung and Blood 
			      Institute</A> (NIH)
			  </UL>
	  		<LI><B>Sequencing:</B> 
			  <UL>
			  <LI>Baylor College of Medicine Human Genome 
			      Sequencing Center
			  <LI><A HREF="http://www.celera.com/" 
			      TARGET=_blank>Celera Genomics Group of 
			      Applera Corporation</A>
			  <LI><A HREF="http://www.genomecorp.com/" 
			      TARGET=_blank>Genome Therapeutics 
			      Corporation</A>
			  <LI><A HREF="http://www.utah.edu/" TARGET=_blank>The
			      University of Utah</A>
			  </UL>
	    		<LI><B>Assembly:</B> Baylor College of Medicine Human 
			Genome Sequencing Center ATLAS group -  Rui Chen, 
			James Durbin, Paul Havlak
	     		<LI><B>Fingerprint Mapping:</B> 
			<A HREF="http://www.bcgsc.bc.ca/" TARGET=_blank>The 
			British Columbia Cancer Agency Genome Sciences 
			Center</A> - Marco Marra, Jackie Schein
	     		<LI><B>BAC End Sequencing:</B> 
			<A HREF="http://www.tigr.org/tdb/bac_ends/rat/bac_end_intro.shtml" 
			TARGET=_blank>The Institute for Genomic Research</A> -
			Shaying Zhao
	     		<LI><B>BAC Library Construction and Distribution:</B>
			<A HREF="http://bacpac.chori.org/" 
			TARGET=_blank>Children's Hospital Oakland Research 
			Institute</A> (CHORI) - Pieter deJong
	     		<LI><B><A HREF="http://rgd.mcw.edu" TARGET=_blank>Rat 
			Genome Database</A>:</B> Medical College of Wisconsin
		    </UL></P>
		    <P>
		    <HR>
		    <A NAME="rat_use"></A>
		    The rat sequence is freely available from the 
		    Rat Genome Project at the 
		    <A HREF="http://hgsc.bcm.tmc.edu/projects/rat/" TARGET=_blank>
		    Baylor College of Medicine Human Genome Sequencing Center</A>.
		    These data are made 
		    available before scientific publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers analyzing
			particular genes and regions if the providers of this
			data (the Rat Genome Sequencing Consortium) are properly acknowledged.
			<LI>
			The Centers producing the data reserve the right to publish the
			initial large-scale analyses of the data set, including large-scale
			identification of regions of evolutionary conservation and
			large-scale genomic assembly. Large-scale refers to regions with
			size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--Opossum Credits---------------------------------------------------->
<A NAME="opossum_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Opossum Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC Opossum Genome Browser displays data from the 
		    Oct. 2004 draft assembly of the opossum (<em>Monodelphis
		    domestica</em>). This assembly is provided with the 
		    following acknowledgements:
		    <UL>
		    <LI><B>Sequencing/Assembly:</B>
		        <A HREF="http://www.broad.mit.edu/" 
			TARGET=_blank>The Broad Institute</A>, 
			Cambridge, MA, USA <BR>
		    <LI><B>UCSC Opossum Genome Browser/Initial Annotations:</B> 
			<A HREF="../staff.html" 
			TARGET=_blank>UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Hiram Clawson, Galt Barber, Ali Sultan-Qurraie, 
			Brian Raney and Donna Karolchik
		    </UL>
		    <P>
		    <HR>
		    <A NAME="opossum_use"></A>
		    The opossum sequence is made freely available before 
		    scientific
		    publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data (The Broad Institute)
			is properly acknowledged.
			<LI>
			The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data set, 
			including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--Chicken Credits-------------------------------------------------------->
<A NAME="chicken_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Chicken Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The April 2006 chicken genome sequence was produced with 
		    the following acknowledgements:
		    <UL>
			<LI><B>Sequencing</B> - 
	  		The <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Gallus%20gallus&GROUP=2"
			TARGET=_blank>Genome Sequencing Center</A> at Washington 
			University School of Medicine, St. Louis
	    		<LI><B>Physical Map</B> - 
	  		The <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Gallus%20gallus&GROUP=2"
			TARGET=_blank>Genome Sequencing Center</A> at Washington 
			University School of Medicine, St. Louis
			<BR>
	     		<LI><B>Genetic Mapping/Linkage Analysis</B> - The Chicken 
			Mapping Consortium:
			<UL>
			<LI>
			Wageningen Map based on Wageningen broilerxbroiler 
			mapping population ("W" linkage maps; 460 F2 animals), 
			coordinated by Martien Groenen, 
			<A HREF="http://www.zod.wau.nl/abg/index.html" 
			TARGET=_blank>Wageningen University</A>
			<LI>
			East Lansing Map based on the RJFxlayer cross 
			("E" linkage maps; 56 BC animals), coordinated by Hans Cheng,
			<A HREF="http://poultry.mph.msu.edu/" TARGET=_blank>Michigan 
			State University</A>
			<LI>
			Compton map based on layerxlayer cross ("C" linkage maps; 
			52 BC animals), coordinated by Nat Bumstead, 
			<A HREF="http://www.iah.bbsrc.ac.uk/primary_index/current_research/group_list.htm"
			TARGET=_blank>Institute for Animal Health</A>
			<LI>
			Consensus linkage map: Martien Groenen, 
			<A HREF="http://www.zod.wau.nl/abg/index.html" 
			TARGET=_blank>Wageningen University</A>
			<LI>
			RH map: <A HREF="http://chickrh.toulouse.inra.fr/"
			TARGET=_blank>Chick RH Server</A>, Alain Vignal, 
			<A HREF="http://www.toulouse.inra.fr/lgc/"
			TARGET=_blank>French National Institute for Agricultural
			Research</A> (INRA)
			<LI>
			Chicken FPC browser: 
			<A HREF="http://www.animalsciences.nl/ChickFPC/"
			TARGET=_blank>ChickFPC</A>, Jan Aerts, 
			<A HREF="http://www.zod.wau.nl/abg/index.html" 
			TARGET=_blank>Wageningen University</A>
			</UL>
			<LI>
			cDNA sequences: <A HREF="http://ncbi.nlm.nih.gov/"
			TARGET=_blank>National Institutes of Health</A> and 
			The University of Manchester 
			<A HREF="http://chick.umist.ac.uk/" 
			TARGET=_blank>BBSRC ChickEST Database</A>
			<LI>
			RJF finished clones: 
			<UL>
			<LI><A HREF="http://www.nisc.nih.gov/"
			TARGET=_blank>NIH Intramural Sequencing Center</A>
			(NISC) (AC091708.2, AC094012.2, AC138565.3)
			<LI>S. Kerje and L. Andersson,
			<A HREF="http://info.uu.se/fakta.nsf/sidor/uppsala.university.id5D.html"
			TARGET=_blank>Uppsala University</A> 
			(AY636124.1)
                        <LI>The Genome Sequencing Center at Washington 
			University School of Medicine, St. Louis (all others)
			</UL>
<!--
		        <LI>
			<B>SNPs and cross-referenced gene annotations:</B> 
			The Beijing Genomics Institute 
			(<A HREF="http://www.genomics.org.cn/bgi_new/index.htm" 
			TARGET=_BLANK>BGI</A>).
-->
			<LI>
			<B>UCSC Chicken Genome Browser:</B> Angie Hinrichs, Heather
		    	Trumbower, Rachel Harte, Donna Karolchik.
			</UL>

		    <P>
		    <HR>
		    <A NAME="chicken_use"></A>
		    The <em>G. gallus</em> sequence is made freely available 
		    to the community by the Genome Sequencing Center,
		    Washington University School of Medicine, with the following
		    understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers analyzing
			particular genes and regions if the providers of this
			data (Genome Sequencing Center, Washington University
			School of Medicine) are properly cited:<BR>
			International Chicken Genome Sequencing Consortium. 
			Sequence and  comparative analysis of the chicken 
			genome provide unique  perspectives on vertebrate 
			evolution. <em>Nature</em>
			2004 Dec 9;432(7018): 695-716.  PMID: 15592404.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
<!--

		    <P>The BGI gene annotations and
		    SNP data are made available with the following terms of use:
		    <OL>
		    <LI>
		    Users are free to use these data in scientific papers 
		    analyzing particular genes, if the Beijing Genomics 
		    Institute (BGI) is acknowledged.
		    <LI>
		    BGI and its collaborators reserve the right to publish the 
	  	    initial analyses of these data, including but not restricted 
		    to the large-scale identification of functional polymorphisms,
		    evolutionary patterns and signs of selection, correlations 
		    with known QTLs, utility of data for genetic mapping 
		    purposes, etc.
		    <LI>
		    Any redistribution of these data should carry this notice. 
		    </OL>
-->
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--X. tropicalis Credits-------------------------------------------------->
<A NAME="xenTro_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>X. tropicalis</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The 3.0  <em>X. tropicalis</em> draft assembly (xenTro1) 
		    was provided by the
		    <A HREF="http://genome.jgi-psf.org/Xentr3/Xentr3.home.html"
		    TARGET=_blank>U.S. DOE Joint Genome Institute</A> (JGI). 
		    This assembly has the following acknowledgements:
		    <P>
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly:</B> 
		    JGI, Walnut Creek, CA, USA
		    <LI>
		    <B>Initial Annotations:</B> 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, Santa 
		    Cruz, CA, USA 
		    <LI>
		    <B>UCSC <em>X. tropicalis</em> Genome Browser:</B> Fan Hsu,
		    Jim Kent, Heather Trumbower, Hiram Clawson, Brian Raney,
		    Galt Barber, Mark Diekhans, Angie Hinrichs, and Donna 
		    Karolchik 
		    </UL>
		    <P>
		    <HR>
		    These sequence data are made freely available by the 
		    JGI. Please refer to the JGI 
		    <A HREF="http://genome.jgi-psf.org/Xentr3/Xentr3.download.html" 
		    TARGET=_blank>data release policy</A> for data use
		    guidelines.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
        </TD></TR>
    </TABLE>
    <BR>

<!--Zebrafish Credits-------------------------------------------------------->
<A NAME="zebrafish_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; Zebrafish Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The zebrafish draft assemblies were provided 
		    by The Wellcome Trust Sanger Institute. The UCSC 
		    Zebrafish Genome Browser has the following 
		    acknowledgements:
		    <P>
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly:</B> 
		    <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/"
		    TARGET=_blank>The Wellcome Trust Sanger Institute</A>, UK. 
		    <LI>
		    <B>FPC map:</B> 
		    <UL>
		    <LI>
		    The Wellcome Trust Sanger Institute, UK
		    <LI>
		    <A HREF="http://www.niob.knaw.nl/" 
		    TARGET=_blank>Hubrecht Laboratory</A>, Netherlands Institute
		    for Developmental Biology, The Netherlands
		    <LI>
		    <A HREF="http://www.eb.tuebingen.mpg.de/" 
		    TARGET=_blank>Max-Planck Institute for Developmental 
		    Biology</A>, Germany
		    </UL>
		    <LI>
		    <B>BAC Clones, BAC End Pairs: </B> 
		    <UL>
		    <LI>
		    The Wellcome Trust Sanger Institute, UK - GenBank 
		    accessions for BAC clones
		    <LI>
		    Max-Planck Institute for Developmental 
		    Biology, Germany - BAC end pairs sequence data
		    <LI>
		    Hubrecht Laboratory</A>, Netherlands Institute
		    for Developmental Biology, The Netherlands - BAC end pairs 
		    sequence data
		    </UL>
		    <LI>
		    <B>Initial Annotations:</B> 
		    <UL>
		    <LI>
		    Yi Zhou, Anthony DiBiase, Leonard Zon and the Zebrafish
		    Genome Initiative at <A HREF="http://zon.tchlab.org/" 
		    TARGET=_blank>Boston Children's Hospital</A>, Boston, MA, US -
		    collaboration on Zebrafish WZ EST clusters, BAC End 
		    Singles/Pairs, Radiation Hybrid, and Human tBLASTn 
	  	    annotation tracks.	
		    <LI>
		    <A HREF="http://www.genetics.wustl.edu/fish_lab/frank/cgi-bin/fish/"
		    TARGET=_blank>Washington University in St. Louis</A>, St. 
		    Louis, MO, US - Zebrafish WZ EST clusters data
		    <LI>
		    <A HREF="http://www.ensembl.org" TARGET=_blank>Ensembl</A>,
		    UK - Ensembl gene predictions
		    <LI>
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, Santa 
		    Cruz, CA, USA - Other annotations
		    </UL>
		    <LI>
		    <B>UCSC Zebrafish Genome Browser (danRer3):</B> Rachel 
		    Harte, Jennifer Jackson, Ann Zweig, Ali Sultan-Qurraie, 
	  	    Donna Karolchik 
		    <LI>
		    <B>UCSC Zebrafish Genome Browser (danRer2):</B> Rachel 
		    Harte, Mark Diekhans, Heather Trumbower, Donna Karolchik, 
		    Jennifer Jackson 
		    </UL>
		    <P>
		    <HR>
		    These sequence data were produced by the Zebrafish 
		    Sequencing Group at the Sanger Institute and can be obtained 
		    directly from <A HREF="ftp://ftp.sanger.ac.uk/pub/zebrafish/" 
		    TARGET=_blank>ftp://ftp.sanger.ac.uk/pub/zebrafish/</A>.
		    Please adhere to the 
		    <A HREF="http://www.sanger.ac.uk/Projects/use-policy.shtml" 
		    TARGET=_blank>data use guidelines</A> for these data.
		    <P>
		    For more information about the assemblies, see the 
		    Sanger Institute's  
		    <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/"
		    TARGET=_blank><em>Danio rerio</em> 
		    Sequencing Project</A> web page.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
        </TD></TR>
    </TABLE>
    <BR>

<!--Tetraodon Credits----------------------------------------------------->
<A NAME="tetraodon_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"></A><B>&nbsp; Tetraodon Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The <em> Tetraodon nigroviridis</em> V7 assembly 
		    (February 2004) was provided by Genoscope in 
		    collaboration with The Broad Institute. 
		    This assembly has the following acknowledgements:
		    <P>
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly:</B> 
		    <UL>
		    <LI>
		    <A HREF="http://www.genoscope.cns.fr/" 
		    TARGET=_blank>Genoscope</A>, Evry, France
		    <LI>
		    <A HREF="http://www.broad.mit.edu/annotation/tetraodon/background.html" 
		    TARGET=_blank>The Broad Institute</A>,MA, USA
		    </UL>
		    <LI>
		    <B>Project Funding:</B> 
		    <UL>
		    <LI>
		    <A HREF="http://www.genopole.org/html/en/connaitre/gip_cnrg.htm" 
		    TARGET=_blank>Consortium National de Recherche en 
		    Genomique</A>, France 
		    <LI>
		    <A HREF="http://www.genome.gov" TARGET=_blank>National Human 
		    Genome Research Institute</A> (NHGRI), USA
		    </UL>
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <UL>
		    <LI>
		    Genoscope - Genscan gene predictions, GAZE gene predictions
		    <LI>
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, Santa 
		    Cruz, CA, USA  - remaining annotations (based on data from 
		    Genoscope)
		    </UL>
		    <LI>
		    <B>UCSC Tetraodon Genome Browser:</B> Rachel Harte,
		    Robert Kuhn, Donna Karolchik 
		    </UL>
<A NAME="tetraodon_use"></A>
		    <P>
		    <HR>
		    These data have been freely provided by Genoscope before 
		    publication for use in the UCSC Genome Browser with the 
		    following understanding
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases. 
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions, provided that the 
		    Genoscope is properly acknowledged.
		    <LI>
		    Genoscope reserves the right to publish the initial 
		    large-scale analyses of the dataset, including large-scale 
		    identification of regions of evolutionary conservation and 
		    large-scale genomic assembly. Large-scale refers to regions
		    with size on the order of magnitude of a Tetraodon 
		    chromosome (that is, 5 Mb or more).
		    </OL>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
        </TD></TR>
    </TABLE>
    <BR>

<!--Fugu Credits-------------------------------------------------------->
<A NAME="fugu_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"></A><B>&nbsp; Fugu Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Fugu whole genome shotgun assembly was provided by the 
		    <A HREF="http://genome.jgi-psf.org/fugu6/fugu6.info.html" TARGET=_blank>U.S. 
		    DOE Joint Genome Institute</A> (JGI) as part of the International Fugu Genome 
		    Consortium, led by JGI and the <A HREF="http://www.fugu-sg.org"
		    TARGET=_blank>Singapore Institute of Molecular and Cell Biology</A> 
		    (IMCB). The assembly was constructed with the JGI assembler, JAZZ, 
		    from paired end sequencing reads produced by JGI and the IMCB
		    at JGI, Myriad Genetics, and Celera Genomics.
		    <P>
		    <HR>
		    The Fugu sequence data have been freely provided by the 
		    JGI for use in the UCSC Genome 
		    Browser. Please see the JGI <A HREF="http://genome.jgi-psf.org/fugu6/fugu6.download.html" 
	 	    TARGET=_blank>data release policy</A> for restrictions
		    regarding the use of these data.
		    <P>
		    We'd like to thank the JGI and the other members of the International Fugu Genome
                    Consortium, including the IMCB (Singapore), the UK Human Genome 
		    Mapping Project (Hinxton), the Molecular Sciences Institute (Berkeley) and 
		    the Institute for Systems Biology (Seattle), for providing the 
		    sequence, assembly, and analysis of this genome.  
		    </P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
        </TD></TR>
    </TABLE>
    <BR>

<!--Ciona Credits-------------------------------------------------------->
<A NAME="ciona_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>C. Intestinalis</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The <em>C. intestinalis</em> v1.0 and v2.0 draft assemblies were provided 
		    by the 
		    <A HREF="http://genome.jgi-psf.org/ciona4/ciona4.home.html"
		    TARGET=_blank>U.S. DOE Joint Genome Institute</A> (JGI).  
		    These assemblies have the following acknowledgements, as 
		    listed on the JGI website:
		    <P>
		    <UL>
		    <LI>
		    <B>Project Funding</B>: Work performed under the auspices of 
		    the US Department of Energy's Office of Science, Biological 
		    and Environmental Research Program and the by the University 
		    of California, Lawrence Livermore National Laboratory under 
		    Contract No. W-7405-Eng-48, Lawrence Berkeley National 
		    Laboratory under contract No. DE-AC03-76SF00098 and Los Alamos
		    National Laboratory under contract No. W-7405-ENG-36.
		    <LI>
		    <B>Project Collaborators</B>:
		    <UL>
		    <LI>
		    JGI, 
		    Walnut Creek, CA, USA
		    <LI>
		    <A HREF="http://smsz.zool.kyoto-u.ac.jp/zool-e.html"
		    TARGET=_blank>Department of Zoology, Graduate School of 
		    Science, Kyoto University</A>, Kyoto, Japan
		    <LI>
		    <A HREF="http://mcb.berkeley.edu/" TARGET=_blank>Department 
		    of Molecular and Cellular Biology, University of 
		    California, Berkeley</A>, CA, USA
		    <LI>
		    <A HREF="http://www.nig.ac.jp/index-e.html" 
		    TARGET=_blank>National Institute of Genetics</A>, Mishima, 
		    Japan
		    <LI>
		    <A HREF="http://www.ornl.gov/" TARGET=_blank>Oak Ridge 
		    National Laboratory</A>, Oak Ridge, TN, USA
		    <LI>
		    <A HREF="http://www.lanl.gov/worldview/" TARGET=_blank>Los 
		    Alamos National Laboratory</A>, Los Alamos, NM, USA
		    </UL>
		    <LI>
		    <B>Annotations:</B> 
		    <UL>
		    <LI>
		    Colin Dewey, University of California, Berkeley, CA, USA -  
		    SNAP Gene Predictions, using the
		    SNAP parser written by Ian Korf
		    <LI>
		    JGI, Walnut Creek, CA, USA - JGI Gene Predictions
		    <LI>
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group<A>, University of California, Santa
		    Cruz, CA, USA - Other annotations
		    </UL>
		    <LI>
		    <B>UCSC <em>C. intestinalis</em> ci1 Genome Browser:</B> Brian
		    Raney, Galt Barber, Heather Trumbower, Robert Kuhn, Donna 
		    Karolchik 
		    <LI>
		    <B>UCSC <em>C. intestinalis</em> ci2 Genome Browser:</B> Brian
		    Raney, Mark Diekhans, Ann Zweig, Kayla Smith, Robert Kuhn,
		    Donna Karolchik 
		    </UL>
		    <P>
		    <HR>
		    The sequence data have been freely provided by the JGI for 
		    use in the UCSC Genome Browser.  
		    The 
		    <A HREF="http://genome.jgi-psf.org/ciona4/ciona4.download.html" 
		    TARGET=_blank>JGI Data Release Policy</A> outlines 
		    restrictions on the use of these data.
		    For more information about the <em>C. intestinalis</em> 
		    assembly, see the 
		    <A HREF="http://genome.jgi-psf.org/ciona4/ciona4.info.html"
		    TARGET=_blank>JGI <em>Ciona</em> web page</A>.
		    </P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
        </TD></TR>
    </TABLE>
    <BR>

<!--S. purpuratus Credits---------------------------------------------->
<A NAME="urchin_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; <em>S. purpuratus</em> Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC <em>S. purpuratus</em> Genome Browser displays data
		    obtained 
		    from the Baylor College of Medicine Human Genome Sequencing
		    Center (BCM HGSC). The Apr. 2005 draft assembly is based on 
		    Baylor version Spur_0.5. These data are provided with the
		    following acknowledgements:
		    <UL>
	   	    <LI><B>Funding</B> - 
		    <A HREF="http://www.genome.gov" TARGET=_blank>National Human
		    Genome Research Institute</A> (NHGRI)
		    <LI><B>Sequencing/Assembly: </B>
		        <A HREF="http://www.hgsc.bcm.tmc.edu/" 
			TARGET=_blank>BCM HGSC</A>, Houston, TX, USA <BR>
		    <LI><B>UCSC <em>S. purpuratus</em> Genome Browser/Initial Annotations (strPur1): </B> 
			<A HREF="../staff.html" 
			TARGET=_blank>UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Andy Pohl, Kayla Smith and Donna Karolchik. 
		    </UL>
		    <P>
		    For more information on the <em>S. purpuratus</em> genome 
		    project, see the BCM HGSC 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/seaurchin/"
		    TARGET=_blank>Sea Urchin Genome Project</A> web page.
		    <P>
		    <HR>
		    <A NAME="urchin_use"></A>
		    To review the conditions for use regarding these 
		    data, see the BCM HGSC 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html"
		    TARGET=_blank>Conditions for Use</A> web page.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--D. melanogaster Credits------------------------------------------------>
<A NAME="dm_credits"></A>
<A NAME="dm_use"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. melanogaster</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The <em>D. melanogaster</em> data displayed in the UCSC
		    Genome Browwser were provided by the Berkeley Drosophila 
		    Genome Project (BDGP) with the following acknowledgements:
		    <UL>
		    <LI><B>Sequencing and Assembly:</B>
		    <UL>
		    <LI>Euchromatic sequence - 
		    <A HREF="http://www.fruitfly.org/" TARGET=_blank>BDGP</A>,
		    Berkeley, CA, USA
		    <LI>Heterochromatic sequence - 
		    <A HREF="http://www.dhgp.org/" TARGET=_blank>Drosophila
		    Heterochromatin Genome Project</A>, Lawrence Berkeley
		    National Laboratory, Berkeley, CA, USA
		    </UL>
		    <LI><B>Annotations: </B>
		    <A HREF="http://flybase.net/annot/" TARGET=_blank>FlyBase 
		    </A> 
		    <LI><B>UCSC <em>D. melanogaster</em> (dm2) Genome Browser 
		    and additional annotations: </B>
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, UCSC, Santa Cruz, CA, USA - 
		    Angie Hinrichs, Brian Raney, Galt Barber,
		    and Donna Karolchik 
		    <LI><B>UCSC <em>D. melanogaster</em> (dm1) Genome Browser 
		    and additional annotations: </B>
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, UCSC, Santa Cruz, CA, USA - 
		    Angie Hinrichs, Brian Raney, Heather Trumbower, Ali 
		    Sultan-Qurraie, and Donna Karolchik
		    </UL>
		    </P>
		    <P>
		    <HR>
		    For additional information about these data, including 
		    citation guidelines, see the 
		    <A HREF="http://www.fruitfly.org" 
		    TARGET=_blank>BDGP web site</A>.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--D. ananassae Credits--------------------------------------------------->
<A NAME="droAna_credits"></A>
<A NAME="droAna_use"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. ananassae</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The droAna2 data were provided by Agencourt Bioscience 
		    Corporation. The droAna1 data were produced by the Institute
		    for Genomic Research (TIGR). The <em>D. ananassae</em> 
		    Genome Browser has the following acknowledgements:
		    <UL>
		    <LI>
		    <B>Sequencing:</B>
		    <A HREF="http://www.agencourt.com/company/experience/nhgri/"
		    TARGET=_blank>Agencourt Bioscience Corporation</A>,
		    Beverly, MA, USA 
		    <LI>
		    <B>Assembly:</B>
		    <UL>
		    <LI>
		    <A HREF="http://www.agencourt.com/company/experience/nhgri/"
		    TARGET=_blank>Agencourt Bioscience Corporation</A>,
		    Beverly, MA, USA, using the 
		    <A HREF="http://www.broad.mit.edu/wga/" 
		    TARGET=_blank>Arachne</A> assembler (droAna2)
		    <LI>
		    <A HREF="http://www.tigr.org/" TARGET=_blank>TIGR</A>,
		    Rockville, MD, USA, using an assembler produced by 
		    <A HREF="http://www.celera.com/" TARGET=_blank>Celera</A>
		    (droAna1)
		    </UL>
		    <LI>
		    <B>UCSC <em>D. ananassae</em> (droAna2) Genome Browser:</B>
		    Angie Hinrichs, Brian Raney, Ann Zweig, Kayla Smith, and 
		    Donna Karolchik
		    <LI>
		    <B>UCSC <em>D. ananassae</em> (droAna1) Genome Browser:</B>
		    Angie Hinrichs, Brian Raney, Heather Trumbower, Ali 
		    Sultan-Qurraie, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, 
		    Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    These data have been freely provided by Agencourt
		    Bioscience Corporation before 
		    publication for use in the UCSC Genome Browser with the 
		    following understanding:
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases. 
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions, provided that the 
		    Agencourt is properly acknowledged.
		    <LI>
		    Agencourt reserves the right to publish the initial 
		    large-scale analyses of the dataset, including large-scale 
		    identification of regions of evolutionary conservation and 
		    large-scale genomic assembly. 
		    <LI>
		    Any redistribution of the data should carry this notice.
		    </OL>
		    <P>
		    Please contact 
	 	    <A HREF="mailto:dsmith@agencourt.com"</A>Douglas R. 
		    Smith</A> at Agencourt with questions concerning these 
		    data.
		    </OL>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--D. persimilis Credits--------------------------------------------------->
<A NAME="droPer_credits"></A>
<A NAME="droPer_use"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. persimilis</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The droPer1 assembly was provided by the Broad Institute
		    with the following acknowledgements:
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly:</B>
		    <A HREF="http://www.broad.mit.edu/" TARGET=_blank>The Broad Institute</A>
		    <LI>
		    <B>UCSC <em>D. persimilis</em> (droPer1) Genome Browser:</B>
		    Angie Hinrichs, Kayla Smith, Robert Kuhn, Jennifer Jackson 
		    and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, 
		    Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="droPer_use"></A>
		    The <em>D. persimilis</em> sequence is made freely available
		    before scientific
		    publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data (The Broad Institute)
			is properly acknowledged.
			<LI>
			The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data set, 
			including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--D. erecta Credits--------------------------------------------------->
<A NAME="droEre_credits"></A>
<A NAME="droEre_use"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. erecta</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    The droEre1 data were provided by
		    Agencourt Bioscience Corporation. The 
		    <em>D.  erecta</em> Genome Browser has the following 
		    acknowledgements:
		    <UL>
		    <LI>
		    <B>Sequencing and Assembly:</B>
		    <A HREF="http://www.agencourt.com/company/experience/nhgri/" 
		    TARGET=_blank>Agencourt Bioscience Corporation</A>,
		    Beverly, MA, USA, using the 
		    <A HREF="http://www.broad.mit.edu/wga/" 
		    TARGET=_blank>Arachne</A> assembler
		    <LI>
		    <B>UCSC <em>D. erecta</em> (droEre1) Genome Browser:</B>
		    Angie Hinrichs, Brian Raney, Jennifer Jackson and Donna 
		    Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, 
		    Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    These data have been freely provided by Agencourt
		    Bioscience Corporation before 
		    publication for use in the UCSC Genome Browser with the 
		    following understanding:
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases. 
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions, provided that the 
		    Agencourt is properly acknowledged.
		    <LI>
		    Agencourt reserves the right to publish the initial 
		    large-scale analyses of the dataset, including large-scale 
		    identification of regions of evolutionary conservation and 
		    large-scale genomic assembly. 
		    <LI>
		    Any redistribution of the data should carry this notice.
		    </OL>
		    <P>
		    Please contact 
	 	    <A HREF="mailto:dsmith@agencourt.com"</A>Douglas R. 
		    Smith</A> at Agencourt with questions concerning these 
		    data.
		    </OL>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--D. simulans Credits--------------------------------------------------->
<A NAME="droSim_credits"></A>
<A NAME="droSim_use"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. simulans</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>These data were provided by The Genome Sequencing Center at
		    Washington University St. Louis (WUSTL) School of 
		    Medicine</A> with the following acknowledgements:
		    <UL>
		    <LI>
		    <B>Sequencing and assembly:</B>
		    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Drosophila%20simulans%20white501&GROUP=6"
		    TARGET=_blank>Genome Sequencing Center, WUSTL School of 
		    Medicine</A> 
		    <LI>
		    <B>UCSC <em>D. simulans</em> Genome Browser</B>:
		    Angie Hinrichs, Jennifer Jackson, Ali Sultan-Qurraie,
		    Brian Raney and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations</B>:
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California,
		    Santa Cruz, CA, USA
		    </UL>
		    <P>
		    <HR>
		    The <em>D. simulans</em> sequence is made freely available 
		    before scientific publication by The Genome Sequencing Center,
		    WUSTL with the following understanding: 
		    <P>
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases. 
		    <LI>
		    Users are free to use the data in scientific papers analyzing 
		    particular genes and regions if the providers of these data 
		    (Genome Sequencing Center, WUSTL School of 
		    Medicine) are properly acknowledged. 
		    <LI>
		    The <em>Drosophila simulans</em> analysis group is aiming to 
		    publish an initial analysis of the <em>D. simulans</em> genome 
		    sequence in 2005 (submitted in early 2005) that will include 
		    descriptions of the assembly, genome landscape, comparative 
		    analysis and initial gene content. People who would like to 
		    coordinate other genome-wide analysis with this work should 
		    contact <A HREF="mailto:rwilson@watson.wustl.edu">Richard K. 
		    Wilson</A>, Genome Sequencing Center Director, Washington 
		    University School of Medicine. We welcome a coordinated 
		    approach to describing this community resource. 
		    <LI>
		    Any redistribution of the data should carry this notice. 
		    </OL>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--D. sechellia Credits--------------------------------------------------->
<A NAME="droSec_credits"></A>
<A NAME="droSec_use"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. sechellia</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The droSec1 assembly was provided by the Broad Institute
		    with the following acknowledgements:
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly:</B>
		    <A HREF="http://www.broad.mit.edu/" TARGET=_blank>The Broad Institute</A>
		    <LI>
		    <B>UCSC <em>D. sechellia</em> (droSec1) Genome Browser:</B>
		    Angie Hinrichs, Kayla Smith and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, 
		    Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="droSec_use"></A>
		    The <em>D. sechellia</em> sequence is made freely available
		    before scientific
		    publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data (The Broad Institute)
			is properly acknowledged.
			<LI>
			The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data set, 
			including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--D. yakuba Credits--------------------------------------------------->
<A NAME="droYak_credits"></A>
<A NAME="droYak_use"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. yakuba</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>These data were provided by The Genome Sequencing Center at
		    Washington University (WUSTL) School of Medicine, St. 
		    Louis</A> with the following acknowledgements:
		    <UL>
		    <LI>
		    <B>Funding:</B>
		    <A HREF="http://www.genome.gov" TARGET=_blank>National Human 
		    Genome Research Institute</A> (NHGRI)
		    <LI>
		    <B>Sequencing, assembly and AGP creation:</B>
		    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Drosophila%20yakuba&GROUP=6"
		    TARGET=_blank>Genome Sequencing Center, WUSTL School of 
		    Medicine</A> 
		    <LI>
		    <B>Initial Sequencing Proposal, Project Collaboration:</B>
		    David J. Begun and Charles H. Langley, 
		    <A HREF="http://www-eve.ucdavis.edu/" 
		    TARGET=_blank>University of California, Davis</A>, CA, USA
		    <LI>
		    <B>Data Downloads:</B>
		    <A HREF="http://species.flybase.net" TARGET=_blank>Flybase</A>
		    <LI>
		    <B>UCSC <em>D. yakuba</em> Genome Browser (droYak2)</B>:
		    Angie Hinrichs, Jennifer Jackson and Donna Karolchik
		    <LI>
		    <B>UCSC <em>D. yakuba</em> Genome Browser (droYak1)</B>:
		    Angie Hinrichs, Michael Chalup and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations</B>:
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California,
		    Santa Cruz, CA, USA
		    </UL>
		    <P>
		    <HR>
		    The <em>D. yakuba</em> sequence is made freely available 
		    before scientific publication by The Genome Sequencing Center,
		    WUSTL with the following understanding: 
		    <P>
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases. 
		    <LI>
		    Users are free to use the data in scientific papers analyzing 
		    particular genes and regions if the providers of these data 
		    (Genome Sequencing Center, WUSTL School of 
		    Medicine) are properly acknowledged. 
		    <LI>
		    The <em>Drosophila yakuba</em> analysis group is aiming to 
		    publish an initial analysis of the <em>D. yakuba</em> genome 
		    sequence in 2005 (submitted in early 2005) that will include 
		    descriptions of the assembly, genome landscape, comparative 
		    analysis and initial gene content. People who would like to 
		    coordinate other genome-wide analysis with this work should 
		    contact <A HREF="mailto:rwilson@watson.wustl.edu">Richard K. 
		    Wilson</A>, Genome Sequencing Center Director, Washington 
		    University School of Medicine. We welcome a coordinated 
		    approach to describing this community resource. 
		    <LI>
		    Any redistribution of the data should carry this notice. 
		    </OL>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--D. virilis Credits--------------------------------------------------->
<A NAME="droVir_credits"></A>
<A NAME="droVir_use"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. virilis</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>These data were produced by Agencourt Bioscience 
		    Corporation. The <em>D. virilis</em> Genome Browser
		    has the following acknowledgements:
		    <UL>
		    <LI>
		    <B>Sequencing and Assembly:</B>
		    <A HREF="http://www.agencourt.com/company/experience/nhgri/" 
		    TARGET=_blank>Agencourt Bioscience Corporation</A>,
		    Beverly, MA, USA, using the 
		    <A HREF="http://www.broad.mit.edu/wga/" 
		    TARGET=_blank>Arachne</A> assembler 
		    <LI>
		    <B>Funding:</B>
		    <A HREF="http://www.genome.gov" TARGET=_blank>National Human 
		    Genome Research Institute</A> (NHGRI), USA
		    <LI>
		    <B>UCSC <em>D. virilis</em> (droVir2) Genome Browser:</B>
		    Angie Hinrichs, Brian Raney, Jennifer Jackson and 
		    Donna Karolchik
		    <LI>
		    <B>UCSC <em>D. virilis</em> (droVir1) Genome Browser:</B>
		    Angie Hinrichs, Brian Raney, Heather Trumbower, 
		    Robert Kuhn, Jennifer Jackson, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, 
		    Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    These data have been freely provided by Agencourt
		    Bioscience Corporation before 
		    publication for use in the UCSC Genome Browser with the 
		    following understanding:
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases. 
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions, provided that the 
		    Agencourt is properly acknowledged.
		    <LI>
		    Agencourt reserves the right to publish the initial 
		    large-scale analyses of the dataset, including large-scale 
		    identification of regions of evolutionary conservation and 
		    large-scale genomic assembly. 
		    <LI>
		    Any redistribution of the data should carry this notice.
		    </OL>
		    <P>
		    Please contact 
	 	    <A HREF="mailto:dsmith@agencourt.com"</A>Douglas R. 
		    Smith</A> at Agencourt with questions concerning these 
		    data.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>



<!--D. mojavensis Credits--------------------------------------------------->
<A NAME="droMoj_credits"></A>
<A NAME="droMoj_use"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. mojavensis</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>These data were produced by Agencourt Bioscience 
		    Corporation. The <em>D. mojavensis</em> Genome Browser
		    has the following acknowledgements:
		    <UL>
		    <LI>
		    <B>Sequencing and Assembly:</B>
		    <A HREF="http://www.agencourt.com/company/experience/nhgri/" 
		    TARGET=_blank>Agencourt Bioscience Corporation</A>,
		    Beverly, MA, USA, using the 
		    <A HREF="http://www.broad.mit.edu/wga/" 
		    TARGET=_blank>Arachne</A> assembler 
		    <LI>
		    <B>Funding:</B>
		    <A HREF="http://www.genome.gov" TARGET=_blank>National Human 
		    Genome Research Institute</A> (NHGRI), USA
		    <LI>
		    <B>UCSC <em>D. mojavensis</em> (droMoj2) Genome Browser:</B>
		    Angie Hinrichs, Brian Raney, Jennifer Jackson and Donna 
		    Karolchik
		    <LI>
		    <B>UCSC <em>D. mojavensis</em> (droMoj1) Genome Browser:</B>
		    Angie Hinrichs, Brian Raney, Galt Barber and Donna 
		    Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, 
		    Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    These data have been freely provided by Agencourt 
		    Bioscience Corporation before 
		    publication for use in the UCSC Genome Browser with the 
		    following understanding:
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases. 
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions, provided that the 
		    Agencourt is properly acknowledged.
		    <LI>
		    Agencourt reserves the right to publish the initial 
		    large-scale analyses of the dataset, including large-scale 
		    identification of regions of evolutionary conservation and 
		    large-scale genomic assembly. 
		    <LI>
		    Any redistribution of the data should carry this notice.
		    </OL>
		    <P>
		    Please contact 
	 	    <A HREF="mailto:dsmith@agencourt.com"</A>Douglas R. 
		    Smith</A> at Agencourt with questions concerning these 
		    data.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>


<!--D. pseudoobscura Credits--------------------------------------------------->
<A NAME="dp_credits"></A>
<A NAME="dp_use"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. pseudoobscura</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>These data were produced by the Human Genome Sequencing
		    Center (HGSC) at Baylor College of Medicine. The <em>D.
		    pseudoobscura</em> Genome Browser has the following 
		    acknowledgements:
		    <UL>
		    <LI>
		    <B>Sequencing and Assembly:</B>
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/drosophila/" 
		    TARGET=_blank>Baylor College of Medicine HGSC</A>
		    <LI>
		    <B>UCSC <em>D. pseudoobscura</em> (dp3) Genome Browser:</B>
		    Angie Hinrichs, Brian Raney, Jennifer Jackson and Donna 
		    Karolchik
		    <LI>
		    <B>UCSC <em>D. pseudoobscura</em> (dp2) Genome Browser:</B>
		    Angie Hinrichs, Heather Trumbower, Robert Kuhn, Donna
		    Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, 
		    Santa Cruz, CA, USA 
		    </UL>
		    <P>The <em>D. pseudoobscura</em> data are made available with 
		    specific <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
		    TARGET=_blank>conditions for use</A>.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--D. grimshawi Credits--------------------------------------------------->
<A NAME="droGri_credits"></A>
<A NAME="droGri_use"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. grimshawi</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    The droGri data were provided by
		    Agencourt Bioscience Corporation. The 
		    <em>D.  grimshawi</em> Genome Browser has the following 
		    acknowledgements:
		    <UL>
		    <LI>
		    <B>Sequencing and Assembly:</B>
		    <A HREF="http://www.agencourt.com/company/experience/nhgri/" 
		    TARGET=_blank>Agencourt Bioscience Corporation</A>,
		    Beverly, MA, USA, using the 
		    <A HREF="http://www.broad.mit.edu/wga/" 
		    TARGET=_blank>Arachne</A> assembler
		    <LI>
		    <B>UCSC <em>D. grimshawi</em> (droGri1) Genome Browser:</B>
		    Angie Hinrichs, Brian Raney, Jennifer Jackson, Kayla Smith
		    and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, 
		    Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    These data have been freely provided by Agencourt
		    Bioscience Corporation before 
		    publication for use in the UCSC Genome Browser with the 
		    following understanding:
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases. 
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions, provided that the 
		    Agencourt is properly acknowledged.
		    <LI>
		    Agencourt reserves the right to publish the initial 
		    large-scale analyses of the dataset, including large-scale 
		    identification of regions of evolutionary conservation and 
		    large-scale genomic assembly. 
		    <LI>
		    Any redistribution of the data should carry this notice.
		    </OL>
		    <P>
		    Please contact 
	 	    <A HREF="mailto:dsmith@agencourt.com"</A>Douglas R. 
		    Smith</A> at Agencourt with questions concerning these 
		    data.
		    </OL>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--A. mellifora Credits------------------------------------------------>
<A NAME="apiMel_credits"></A>
<A NAME="apiMel_use"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>A. mellifera</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The Amel_1.2 data were provided by The Baylor College of
		    Medicine Human Genome Sequencing Center (BCM HGSC)
		    with the following acknowledgements:
		    <UL>
		    <LI>
		    <B>Funding: </B>
		    <A HREF="http://www.genome.gov" TARGET=_blank>National Human 
		    Genome Research Institute</A> (NHGRI) and the 
		    <A HREF="http://www.usda.gov/wps/portal/usdahome"
		    TARGET=_blank>U.S. Department of Agriculture</A> (USDA)
		    <LI>
		    <B>Sequencing and assembly:</B>
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/honeybee/"
		    TARGET=_blank>BCM HGSC</A>, Houston, TX, USA
		    <LI>
		    <B>BAC library:</B>
		    Pieter de Jong and Katzutoyo Osoegawa at 
		    <A HREF="http://bacpac.chori.org/"
		    TARGET=_blank>Children's Hospital Oakland Research 
		    Institute</A> (CHORI), Oakland, CA, USA
		    <LI>
		    <B>UCSC <em>A. mellifera</em> (apiMel2) Genome Browser</B>:
		    Andy Pohl, Angie Hinrichs, Jennifer Jackson, and Donna 
		    Karolchik
		    <LI>
		    <B>UCSC <em>A. mellifera</em> (apiMel1) Genome Browser</B>:
		    Angie Hinrichs, Brian Raney, Robert Kuhn, and Donna 
		    Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations</B>:
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California,
		    Santa Cruz, CA, USA
		    </UL>
		    <P>
		    <HR>
		    The <em>A. mellifera</em> sequence is made freely 
		    available before scientific publication by the BCM HGSC.
		    See the Baylor 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html"
		    TARGET=_blank>conditions of use</A> statement for 
		    guidelines regarding the use of these data.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--A. gambiae Credits--------------------------------------------------->
<A NAME="anoGam_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>A. gambiae</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The MOZ2 <em>A. gambiae</em> data were provided by The 
		    International Anopheles Genome Project and 
		    downloaded from Ensembl. The data have the following 
		    acknowledgements:
		    <UL>
		    <LI>
		    <B>Funding:</B>
 		    <A HREF="http://www.niaid.nih.gov/default.htm"
		    TARGET=_blank>National Institute of Allergy and Infectious 
		    Diseases</A> (NIAID), <A HREF="http://www.recherche.gouv.fr/"
		    TARGET=_blank>French Ministry of Research</A>, and the 
		    <A HREF="http://www.who.int/en/" TARGET=_blank>World Health 
		    Organization</A> (WHO). Celera Genomics and Genoscope
		    were funded respectively by <A HREF="http://www.nih.gov/" 
		    TARGET=_blank>National Institutes of Health</A> (NIH) and 
		    French Ministry of Research.
		    <LI>
		    <B>Project collaboration:</B>
		    The International Anopheles Genome Project - 
		    a collaboration between 
		    <A HREF="http://www.celera.com/" TARGET=_blank>Celera 
		    Genomics</A>, 
		    <A HREF="http://www.genoscope.cns.fr/" 
		    TARGET=_blank>Genoscope</A>, 
		    <A HREF="http://www.nd.edu/" TARGET=_blank>University of 
		    Notre Dame</A>, 
		    <A HREF="http://www.ebi.ac.uk/" 
		    TARGET=_blank>EBI</A>/<A HREF="http://www.sanger.ac.uk/" 
		    TARGET=_blank>Sanger Institute</A>, 
		    <A HREF=" http://www.ebi.ac.uk/embl/" TARGET=_blank>EMBL</A>, 
		    <A HREF="http://www.pasteur.fr/" TARGET=_blank>Institut 
		    Pasteur</A>, <A HREF="http://www.imbb.forth.gr/" 
		    TARGET=_blank>The Institute of Molecular Biology and 
		    Biotechnology</A> (IMBB) and 
		    <A HREF="http://www.tigr.org" TARGET=_blank>The Institute 
		    for Genomic Research</A> (TIGR).
		    <LI>
		    <B>Sequencing and Assembly:</B>
		    <A HREF="http://www.celera.com/" TARGET=_blank>Celera 
		    Genomics</A>, <A HREF="http://www.genoscope.cns.fr/" 
		    TARGET=_blank>Genoscope</A> 
		    <LI>
		    <B>Data Downloads:</B>
		    <A HREF="http://www.ensembl.org/Anopheles_gambiae/" 
		    TARGET=_blank>Ensembl</A>
		    <LI>
		    <B>UCSC <em>A. gambaie</em> Genome Browser</B>:
		    Angie Hinrichs, Galt Barber and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations</B>:
		    <A HREF="http://www.ensembl.org" TARGET=_blank>Ensembl</A> and
		    <A HREF="http://www.celera.com/" TARGET=_blank>Celera 
		    Genomics</A> (Ensembl Genes), the
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California,
		    Santa Cruz, CA, USA (remaining annotations)
		    </UL>
		    <P>
		    <HR>
		    The <em>A. gambaie</em> sequence is made freely available 
		    by The International Anopheles Genome Project. 
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--C. elegans Credits--------------------------------------------------->
<A NAME="ce_credits"></A>
<A NAME="ce_use"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>C. elegans</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The <em>C. elegans</em> data was obtained from 
		    <A HREF="http://www.wormbase.org/" TARGET=_blank>WormBase</A>. We'd like to thank
		     the <A HREF="http://genome.wustl.edu/" TARGET=_blank>Genome Sequencing Center at 
		     Washington University in St. Louis</A> and the <A HREF="http://www.sanger.ac.uk/" 
		     TARGET=_blank>Sanger Institute</A> for their collaborative work
		     on sequencing the <em>C. elegans</em> genome, and the WormBase consortium
		     for providing access to the current <em>C. elegans</em> sequence.
		     </P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--C. briggsae Credits--------------------------------------------------->
<A NAME="cb_credits"></A>
<A NAME="cbr_use"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
  	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>C. briggsae</em> Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The <em>C. briggsae</em> sequence was obtained from publicly available
		    data in <A HREF="http://www.wormbase.org/" TARGET=_blank>WormBase</A>.
		    This data is  made available before scientific publication with 
		    the following understanding: 
		    The <A HREF="http://genome.wustl.edu/" TARGET=_blank>Genome Sequencing Center at 
		    Washington University in St. Louis</A> and the <A HREF="http://www.sanger.ac.uk/" 
		    TARGET=_blank>Sanger Institute</A> reserve the right to publish the results of 
		    any large scale analyses. If you wish to use the results of specific searches in 
		    a publication, please acknowledge these sequencing centers and check that you 
    		    are using the most recent version of the sequence.
		    </P>
		    <P>We'd like to thank
		    the Genome Sequencing Center at Washington University in St. Louis
		    and the Sanger Institute for their collaborative work
		    on sequencing the <em>C. briggsae</em> genome, and the WormBase consortium
		    for providing access to the current <em>C. briggsae</em> sequence. 
		    </P>
		</TD><TD WIDTH=15></TD></TR>
	     </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--Sacchromyces cerevisiae Credits------------------------>
<A NAME="yeast_credits"></A>
<A NAME="yeast_use"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
  	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; Yeast Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Oct. 2003 <I>Sacchromyces cerevisiae</I> genome assembly 
		    is based on sequence dated 1 Oct. 2003 in the 
		    <A HREF="http://www.yeastgenome.org/" TARGET=_blank>Sacchromyces Genome
		    Database</A> (SGD). The sequence, open reading frame (ORF), 
		    and gene annotations were downloaded from the site
		    <A HREF="ftp://genome-ftp.stanford.edu/pub/yeast/data_download"
		    TARGET=_blank>ftp://genome-ftp.stanford.edu/pub/yeast/data_download</A>. 
		    The cellular localization and protein abundance data displayed
		    in the Gene Sorter are taken from the
		    <A HREF="http://yeastgfp.ucsf.edu/" TARGET=_blank>Yeast GFP 
		    Fusion Localization Database</A>.
		    <P>See the SGD 
		    <A HREF="http://www.yeastgenome.org/chromosomes.shtml" 
		    TARGET=_blank>Systematic Sequencing Table</A> for credit and
		    contact information for each of the chromosomes in the 
		    assembly. We'd like to thank Stanford University, the SGD,
		    the University of California San Francisco (UCSF), 
		    Washington University in St. Louis, and The Broad 
		    Institute 
		    for providing the data and annotations for this release.
		</TD><TD WIDTH=15></TD></TR>
	     </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--SARS Credits---------------------------------------------------------->
<A NAME="sars_credits"></A>
<A NAME="sars_use"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
  	<TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		<TR><TD><FONT SIZE="4"><B>&nbsp; SARS Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The SARS draft genome assembly is based on publicly-available sequence 
		    deposited into GenBank as of 14 April 2003. Annotations showing gene predictions, 
		    locations of putative proteins, and viral mRNA and protein alignments were generated
		    at UCSC, with the exception of the Fgenesv+ annotation contributed by Victor Solovyev of
		    Softberry, Inc.  Protein structure analysis and predictions were determined by using the 
		    <A HREF="http://www.cse.ucsc.edu/research/compbio/sam.html" TARGET=_blank>Sequence Alignment 
		    and Modeling</A> (SAM) T02 tool.
		</TD><TD WIDTH=15></TD></TR>
	     </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!--Other Credits-------------------------------------------------------->
<A NAME="other_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	    <TR><TD><FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; External Contributors
		to the UCSC Genome Browser Project</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	    <TH HEIGHT=10></TH>
		<TR><TD COLSPAN=2></TD></TR>
	 	<TR><TD COLSPAN=2>
		    <UL>
			<LI>
			<A HREF="http://www.affymetrix.com" TARGET=_blank>Affymetrix</A> (David Kulp, 
			Alan Williams, Ray Wheeler, Greg Helt, Tom Gingeras, 
			Guoying Liu, Andy Kimbrough, Hajime Matsuzaki, Rui Mei, 
			Keith Jones): Development of genotyping array and genotype 
			data for Affymetrix 120K SNP annotation track, Affymetrix transcriptome data for chr 21 
			and 22, Genie gene prediction track, design and analysis support in 
			the early stages of the UCSC Human Genome project, including software for the synthetic 
			chromosome 22 data we used to test our assembly methods.
			<LI>
			<A HREF="http://www.ncbi.nlm.nih.gov/genome/cyto/hbrc.shtml" TARGET=_blank>BAC Resource 
			Consortium</A>: Cytogenetic markers (FISH-mapped clones annotation track).
			<LI>
			<A HREF="http://www.hgsc.bcm.tmc.edu/" TARGET=_blank>Baylor College of Medicine</A>
			(Richard Gibbs, John Bouck, Kim Worley and James Durbin): Rat Genome sequence data,
			<em>D. pseudoobscura</em> sequence for alignment tracks
			with <em>D. melanogaster</em> genome. 
			<LI>
			<A HREF="http://www.fruitfly.org/" TARGET=_blank>Berkeley Drosophila Genome 
			Project</A> (BDGP): Data for <em>D. melanogaster</em> gene annotations.
			<LI>
			<A HREF="http://www.bioinformaticssolutions.com/" TARGET=_blank>Bioinformatics Solutions, Inc.</A>
			(Ming Li and colleagues): Pattern Hunter mouse homology track.
			<LI>
			<A HREF="http://www.cwru.edu/" TARGET=_blank>Case Western Reserve University</A> (Evan 
			Eichler and Jeff Bailey): Segmental duplications annotation track.
			<LI>
			<A HREF="http://www.csmc.edu/csri/korenberg/int-bac-sts.html" TARGET=_blank>Cedars-Sinai Medical Center</A> (CSMC):
			FISH-mapped clones annotation track. 
			<LI>
			<A HREF="http://www-lbme.biotoul.fr/" TARGET=blank>CNRS 
			Université Paul Sabatier</A>, Laboratoire de Biologie Moléculaire Eucaryote, Toulouse, 
			France (Michel Weber): miRNA Data annotation track (in collaboration with Sam 
			Griffiths-Jones of The Wellcome Trust Sanger Institute).
			<LI>
			<A HREF="http://www.cshl.org/" TARGET=blank>Cold Spring Harbor Laboratory</A>
			<BR>--(<A HREF="http://stein.cshl.org/~lstein/" TARGET=_blank>Lincoln Stein</A>): HapMap data.
			<BR>--(<A HREF=http://rulai.cshl.org/" TARGET=_blank>Ivo Grosse and Michael Q. Zhang</A>): 
			Predictions for <A HREF="http://rulai.cshl.org/tools/FirstEF/" TARGET=_blank>First Exon
			Finder</A> annotation track.
			<LI>
			<A HREF="http://www.ensembl.org/" TARGET=_blank>Ensembl</A> (Ewan Birney, Guy Slater, 
			Imre Vastrik, Arek Kasprzyk, Michele Clamp): Ensembl Gene predictions annotation track, 
			data exchange formats, 
			TPF clone map coordination, 
			exonerate mouse homology track, early help placing FISH-mapped clones. 
		    	<LI>
			<A HREF=" http://www.ebi.ac.uk/embl/" TARGET=_blank>EMBL</A> (Peter Bork): Mouse pseudogene
			predictions used for Bork Pseudogenes custom annotation track.
			<LI>
			<A HREF="http://www.flybase.org/" TARGET=_blank>FlyBase</A>: Curated Drosophila
			gene annotations.
			<LI>
			<A HREF="http://www.fhcrc.org/" TARGET=_blank>Fred Hutchison Cancer Research Center</A> (FHCRC)
			(Barbara Trask): FISH-mapped clones annotation track.
			<LI>
			<A HREF="http://www.geneontology.org/" TARGET=_blank>Gene Ontology (GO) Consortium</A>:
			Structured annotations for Gene Sorter.
			<LI>
			<A HREF="http://www.girinst.org/" TARGET=_blank>Genetic Information Research Institute</A>:
			Repbase, repeatmasking.
			<LI>
			<A HREF="http://www.genoscope.cns.fr/" TARGET=_blank>Genome Informatics Research Lab</A> 
			(GIRL) within Grup de Recerca en Informàtica Mèdica (GRIB) (Francisco Câmara Ferreira, 
			Roderic Guigó): Geneid and SGP gene prediction annotation tracks. 
			<LI>
			<A HREF="http://web.gnf.org/index.shtml" TARGET=_blank>Genomics Institute of the Novartis
			Research Foundation</A> (GNF): Expression data using Affymetrix chips for GNF Expression
			Atlas annotation tracks.
			<LI>
			<A HREF="http://www.genoscope.cns.fr/" TARGET=_blank>Genoscope</A>
			(Jean Weissenbach and Olivier Jaillon): Exofish program, <em>Tetraodon nigroviridis</em> 
			sequence for Tetraodon Blat alignments annotation tracks, data for Anopheles
			Evolutionary Conserved Regions annotation track.
			<LI>
			<A HREF="http://www.tigr.org" TARGET=_blank>The Institute for Genomic Research</A>
			(TIGR): TIGR Gene Index annotation track. 
			<LI>
			<A HREF="http://www.systemsbiology.org" TARGET=_blank>Institute for Systems Biology</A>
			(Arian Smit): <A HREF="http://www.repeatmasker.org" 
			TARGET=_blank>RepeatMasker</A> repeat identification and masking software and libraries.
			<LI>
			<A HREF="http://www.jax.org/" TARGET=_blank>The Jackson Laboratory</A> (Carol Bult): Mouse ortholog links 
			from known genes track. 
			<LI>
			<A HREF="http://www.genome.ad.jp" TARGET=_blank>Kyoto University</A>, GenomeNet: 
			<A HREF="http://www.genome.ad.jp/kegg/kegg2.html" TARGET=_blank>Kyoto Encyclopedia of 
			Genes and Genomes</A> (KEGG) - pathway diagrams for Known Genes annotation track.
			<LI>
			<A HREF="http://www.lanl.gov/" TARGET=_blank>Los Alamos National Laboratory</A> (LANL):
			FISH-mapped clones annotation track. 
			<LI>
			<A HREF="http://mgc.nci.nih.gov" TARGET=_blank>Mammalian Gene Collection (MGC) Project</A>: Data
			used in MGC Genes annotation track.
		 	<LI>
			<A HREF="http://genes.mit.edu/chris" TARGET=_blank>Massachusetts Institute of Technology
			(MIT)</A> (Chris Burge): Genscan gene prediction program. 
		 	<LI>
			<A HREF="http://www2.mrc-lmb.cam.ac.uk/" TARGET=_blank>Medical Research Council Laboratory 
			of Molecular Biology</A> (Julian Gough - now at Stanford School of Medicine): 
			<A HREF="http://supfam.mrc-lmb.cam.ac.uk/SUPERFAMILY/" TARGET=_blank>Superfamily</A> 
			database.
			<LI>
			<A HREF="http://directory.mssm.edu/faculty/facultyInfo.php?id=26162&deptid=2" TARGET=_blank>Mount Sinai School of 
			Medicine</A> (Gary Benson): Tandem Repeats Finder.
			<LI>
			<A HREF="http://cancer.gov/" TARGET=_blank>National Cancer Institute</A> (NCI): 
			<A HREF="http://cgap.nci.nih.gov/" TARGET=_blank>The Cancer
			Genome Anatomy Project</A> (CGAP) - FISH-mapped clones annotation track. SAGE project (in 
			collaboration with NCBI).
		 	<LI>
			<A HREF="http://www.ncbi.nlm.nih.gov/index.html" TARGET=_blank>National Center for Biotechnology
			Information</A> (NCBI) 
			<BR>--(Deanna Church): Mouse 
			synteny track, database of whole genome shotgun mouse reads and assemblies used for mouse tracks.
			<BR>--(Greg Schuler): GenBank freezes of human clones used in the assembly, NT contigs of multiple 
			finished clones, STS maps (in earlier freezes), specially-formatted versions of the finished 
			chromosomes. 
			<BR>--(Steve Sherry): SNP tracks. 
			<BR>--(Wonhee Jang): FISH-mapped clones annotation track.
			<BR>--(Danielle and Jean Thierry-Mieg): Acembly Genes annotation track.
			<BR>--(Lukas Wagner): 3' EST track in the July 2000 browser.
			<BR>--<A HREF="http://www.ncbi.nlm.nih.gov/SNP/" TARGET=_blank>dbSNP</A> database.
			<BR>--(Kim Pruitt, Donna Maglott): <A HREF="http://www.ncbi.nlm.nih.gov/RefSeq/" 
			TARGET=_blank>RefSeq</A> and 
			<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=gene"
			TARGET=_blank>Entrez Gene</A> databases.
			<BR>--<A HREF="http://www.ncbi.nlm.nih.gov/SAGE/" TARGET=_blank>Serial Analysis of
			Gene Expression Tag to Gene Mapping</A> (SAGEmap) data resource.
			<LI>
			<A HREF="http://www.genome.gov" TARGET=_blank>National Human Genome Research Institute</A> 
			(NHGRI)
			<BR>--(Elliott Margulies, Pamela Jacques Thomas, Eric Green): Collaboration and annotations
			for <A HREF="http://www.nisc.nih.gov/index.html" TARGET=_blank>NISC</A> browser, including 
			curated gene annotations, DNAse I Hypersensitive Sites 
			annotation, Multi-species Conserved Sequences (MCS) tracks, Binomial_25 Conservation Scores,
			ZooSeq Blastz alignments tracks. 
			<BR>--Data for ENCyclopedia Of DNA Elements (ENCODE) annotation track.
			<LI>
			<A HREF="http://bioinformatics.med.ohio-state.edu/" TARGET=_blank>Ohio State 
			University Bioinformatics & Computational Biology Group</A> (Ramana V. Davuluri): 
			Predictions for <A HREF="http://rulai.cshl.org/tools/FirstEF/" TARGET=_blank>First Exon
			Finder</A> annotation track.
			<LI>
			<A HREF="http://bio.cse.psu.edu/" TARGET=_blank>Pennsylvania State University Bioinformatics
			Group</A> (Scott Schwartz, Webb Miller, Zheng Zhang, Ross Hardison, Francesca Chiaromonte, 
			Laura Elnitski): Blastz alignment program, multiz alignment protram, Genome Alignment and
			Annotation Database (GALA), collaboration on multiple alignment annotation tracks, ENCODE
			annotation tracks.
			<LI>
			<A HREF="http://www.perlegen.com" TARGET=_blank>Perlegen Sciences</A> (David Hinds, 
			Kelly Frazer, and David Cox): STS markers for the         
			Haplotype Blocks annotation track on chr 21, SNP data for the Affymetrix 120K SNP track, 
			TNG radiation hybrid map.
			<LI>
			<A HREF="http://www.gsc.riken.go.jp" TARGET=_blank>RIKEN
			Genomic Sciences Center</A>: Chimpanzee BAC end sequence
			data, mouse annotations, human genome sequence contributions.
			<LI>
			<A HREF="http://www.roswellpark.org/" TARGET=_blank>Roswell Park Cancer Institute</A> (RPCI):
			FISH-mapped clones annotation track.
			<LI>
			<A HREF="http://snp.cshl.org/" TARGET=_blank>SNP Consortium</A>: Data for SNP annotation tracks.
			<LI>
			<A HREF="http://www.softberry.com/" TARGET=_blank>Softberry Inc.</A> (Victor Solovyev):
			Fgenesh++ and Fgenesv+ gene predictions, TSSW promoters tracks.
			<LI>
			<A HREF="http://www.mad-cow.org/" TARGET=_blank>Sperling Biomedical 
			Foundation</A> (Thomas Pringle): Technical consultant on 
			Proteome Browser, avid Genome Browser user and the 
			source of a myriad of ideas for expanding its functionality, early 
			assistance and support with user's guides, email moderation, web links, 
			and testing.
			<LI>
			<A HREF="http://www-shgc.stanford.edu/myerslab/" TARGET=_blank>Stanford University School 
			of Medicine (Richard Myers Lab)</A>: Data set for Human Promoters custom annotation track.
			<LI>
			<A HREF="http://us.expasy.org/sprot/" TARGET=_blank>Swiss-Prot</A>: Protein data and
			human curated annotation from the Swiss-Prot protein knowledgebase, TrEMBL, and TrEMBL-NEW.
			<LI>
			<A HREF="http://bioinfo2.ugr.es/isochores/" TARGET=_blank>Universidad de Granada</A>, 
			Grupo de Bioinformática, Spain (Jose L. Oliver): Isochore custom annotation track.</A>
			<LI>
			<A HREF="http://bio.math.berkeley.edu/" TARGET=_blank>University of CA, Berkeley</A> (Lior
			Pachter, Simon Cawley, Marina Alexandersson): 
			<A HREF="http://baboon.math.berkeley.edu/~syntenic/about.html" TARGET=_blank>SLAM</A> Gene
			prediction annotation tracks.
			<LI>
			<A HREF="http://cc.ucsf.edu/" TARGET=_blank>University of CA, San Francisco Cancer Center</A>:
			FISH-mapped clones annotation track.
			<LI>
			<A HREF="http://cc.ucsf.edu/" TARGET=_blank>University of Pennsylvania</A> (Y. Thomas Gan):
			Database of Transcribed Sequences (DoTS) Genes custom annotation track.
			<LI>
			<A HREF="http://www.tbi.univie.ac.at/index.html" TARGET=_blank>University of Vienna</A>, Institute 
			for Theoretical Chemistry and Structural Biology, Theoretical Biochemistry Group</A>: 
			Vienna RNA utility <A HREF="http://www.tbi.univie.ac.at/~ivo/RNA/" TARGET=_blank>RNAplot</A> - 
			Known Genes annotation track. 
			<LI>
			<A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Homo%20sapiens&GROUP=1" TARGET=_blank>Washington University in St. 
			Louis</A> 
			<BR>--(Robert Waterston (now at UW), John McPhearson, Asif Chinwalla, Shiaw-Pyng Yang, John Wallis and 
			LaDeana Hillier): Fingerprint-based map of BAC clones from the human genome and a 
			corresponding layout of sequenced clones, plasmid end reads, BACend analysis,
			CpG islands.
			<BR>--(<A HREF="http://www.genetics.wustl.edu/eddy/" TARGET=_blank>Sean Eddy</A>
			and Todd Lowe (now at UCSC)): RNA Genes track.
			<BR>--(<A HREF="http://genes.cs.wustl.edu/BrentLab/MB-Lab-index.html" TARGET=_blank>Michael 
			Brent's Computational Genomics Group</A>): Twinscan Gene prediction annotation track.
			<LI>
			<A HREF="http://www.sanger.ac.uk/HGP/" TARGET=_blank>Wellcome Trust Sanger Institute</A>
			<BR>--(Jennifer Ashurst, James Gilbert, John Sulston, Jane Rogers, Jim Mullikin, Richard 
			Durbin, Steve Searle, Ian Dunham and the annotation teams): Vega Genes and Pseudogenes, 
			Sanger 20 and Sanger 22 Genes and Pseudogenes. Plasmid 
			end reads, processing of whole genome shotgun mouse reads.
			<BR>--(Sam Griffiths-Jones): miRNA Data annotation track (in collaboration with 
			Michel Weber of CNRS Université Paul Sabatier).
			<BR>--<A HREF="http://www.sanger.ac.uk/Software/Pfam/" TARGET=_blank>Protein Families 
			Database of Alignments and HMMs<A> (Pfam).
			<LI>
			<A HREF="http://www.broad.mit.edu/" TARGET=_blank>The Broad Institute</A>
			(Eric Lander, Mike Zody, Chad Nusbaum): Plasmid end reads. Human working
			draft sequence analysis project. Special thanks to Eric 
			for inviting us to join the Hardcore Analysis Group for the draft sequence project and for 
			his direction of that group. 
			<LI>
			<A HREF="http://www.wormbase.org/" TARGET=_blank>WormBase</A> (Lincoln Stein): Wormbase 
			curated gene annotations for <em>C. elegans</em> genome.
		</UL></TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>

<!--Early Credits-------------------------------------------------------->
<A NAME="early_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
	<TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	    <TR><TD>
 	    <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; Acknowledgements of Early Support</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	    <TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD>
		<TD>
		    <P>During the initial stage of this project, from January 2000 to June 2001,
		    we received no direct federal funding to develop the working draft other than travel 
		    support. However, we gratefully acknowledge the support for our personnel provided 
		    by our basic bioinformatics research grants, especially DOE grant
		    DE-FG03-99ER62849 (to develop gene-finding methods), but also NSF grant DBI-9809007, 
		    UC BIOTECH grant 99-11, a grant from the Sloan Foundation, the David and Lucille 
		    Packard Foundation, NIH grant GM-52848, and the Zahler grant that provided support 
		    for Jim Kent in the early stages of this work. 
		    <P>
		    We are particularly thankful for the risks that Patrick Mantey (UCSC Dean of Engineering), 
		    James Gill (Vice Chancellor for Research), John Simpson (UCSC Provost), and 
		    M.R.C. Greenwood (UCSC Chancellor) were willing to take 
		    in advancing us the money in February, 2000 needed to build the original 100-node 
		    LINUX cluster used in this work, and in the case of Dean Mantey, additional 
		    personnel support. Phokion Kolaitis, Chair of Computer Science, contributed as 
		    well. 
		    <P>We thank <A HREF="http://www.compaq.com/" TARGET=_blank>Compaq</A> for providing an 
		    additional 100-CPU computational cluster in Cambridge, MA for use
		    by the public human genome effort. We thank <A HREF="http://www.ilog.com/" TARGET=_blank>ILOG,
		    Inc.</A> for providing discounted licenses to its CPLEX linear programming software, 
		    which was essential for some of our early experiments.</P>
		    <P>
		    Beginning in December, 2000, we started to utilize some support from 
		    <A HREF="http://www.hhmi.org" TARGET=_blank>HHMI</A> for equipment and personnel. In the summer 
		    of 2001, HHMI provided the primary funding for a new cluster of 1000 CPUs running 
		    LINUX purchased from <A HREF="http://www.rackable.com" TARGET=_blank>Rackable</A>, with additional 
		    support from <A HREF="http://www.ucop.edu/california-institutes" TARGET=_blank>CISI</A> and 
		    <A HREF="http://www.nhgri.nih.gov" TARGET=_blank>NHGRI</A>. Dubbed the 
		    UCSC KiloKluster, this equipment now provides the computational backbone
		    for the project.   
		    </P>
		</TD>
		<TD WIDTH=15></TD>
	    </TR>
	</TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    </TD></TR>
</TABLE>
</BODY></HTML>
