<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: Credits</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="#fFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/topbar.html"--></TD></TR>
<!-- Intro Tables====================================================== -->
    <TR><TD COLSPAN=2>	
    	<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	    <TR><TD>
		<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		    <TR><TD>
			<FONT SIZE="4"><B>&nbsp; UCSC Genome Browser Acknowledgments</B></FONT>
		    </TD></TR>
		</TABLE>
		<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		    <TR><TD WIDTH=10></TD>
			<TD>
			<P>
			The data and software displayed on this site are the result of
			a large collaborative effort among many individuals at UCSC and
			at research institutions around the world. We'd like to 
			acknowledge the scientists and engineers who have contributed
			to the UCSC Genome Browser and its data, and the organizations 
			who have made this project possible through their generous 
			funding.
			<P>
			<B>Mammals</B>
			<UL>
	        	    <LI><A HREF="#human_credits">Human Genome</A>
	        	    <LI><A HREF="#alpaca_credits">Alpaca Genome</A>
	        	    <LI><A HREF="#armadillo_credits">Armadillo Genome</A>
	        	    <LI><A HREF="#baboon_credits">Baboon Genome</A>
	        	    <LI><A HREF="#bushbaby_credits">Bushbaby Genome</A>
	        	    <LI><A HREF="#cat_credits">Cat Genome</A>
	        	    <LI><A HREF="#chimp_credits">Chimpanzee Genome</A>
	        	    <LI><A HREF="#cow_credits">Cow Genome</A>
	        	    <LI><A HREF="#dog_credits">Dog Genome</A>
	        	    <LI><A HREF="#dolphin_credits">Dolphin Genome</A>
	        	    <LI><A HREF="#elephant_credits">Elephant Genome</A>
	        	    <LI><A HREF="#gibbon_credits">Gibbon Genome</A>
	        	    <LI><A HREF="#gorilla_credits">Gorilla Genome</A>
	        	    <LI><A HREF="#guinea_pig_credits">Guinea Pig Genome</A>
	        	    <LI><A HREF="#hedgehog_credits">Hedgehog Genome</A>
	        	    <LI><A HREF="#horse_credits">Horse Genome</A>
	        	    <LI><A HREF="#kangaroo_rat_credits">Kangaroo Rat Genome</A>
	        	    <LI><A HREF="#manatee_credits">Manatee Genome</A>
	        	    <LI><A HREF="#marmoset_credits">Marmoset Genome</A>
	        	    <LI><A HREF="#megabat_credits">Megabat Genome</A>
	        	    <LI><A HREF="#microbat_credits">Microbat Genome</A>
	        	    <LI><A HREF="#mouse_credits">Mouse Genome</A>
	        	    <LI><A HREF="#mouse_lemur_credits">Mouse Lemur Genome</A>
      			    <LI><A HREF="#naked_mole-rat_credits">Naked Mole-rat Genome</A>
	        	    <LI><A HREF="#opossum_credits">Opossum Genome</A>
	        	    <LI><A HREF="#orangutan_credits">Orangutan Genome</A>
	        	    <LI><A HREF="#panda_credits">Panda Genome</A>
	        	    <LI><A HREF="#pig_credits">Pig Genome</A>
	        	    <LI><A HREF="#pika_credits">Pika Genome</A>
	        	    <LI><A HREF="#platypus_credits">Platypus Genome</A>
	        	    <LI><A HREF="#rabbit_credits">Rabbit Genome</A>
	        	    <LI><A HREF="#rat_credits">Rat Genome</A>
	        	    <LI><A HREF="#rhesus_credits">Rhesus Genome</A>
	        	    <LI><A HREF="#rock_hyrax_credits">Rock Hyrax Genome</A>
	        	    <LI><A HREF="#sheep_credits">Sheep Genome</A>
	        	    <LI><A HREF="#shrew_credits">Shrew Genome</A>
	        	    <LI><A HREF="#sloth_credits">Sloth Genome</A>
			    <LI><A HREF="#squirrel_credits">Squirrel Genome</A>
			    <LI><A HREF="#squirrel_monkey_credits">Squirrel Monkey Genome</A>
	        	    <LI><A HREF="#tarsier_credits">Tarsier Genome</A>
			    <LI><A HREF="#tasmanian_devil_credits">Tasmanian Devil Genome</A>
                            <LI><A HREF="#tenrec_credits">Tenrec Genome</A>
	        	    <LI><A HREF="#tree_shrew_credits">Tree Shrew Genome</A>
			    <LI><A HREF="#wallaby_credits">Wallaby Genome</A>
			    <LI><A HREF="#rhino_credits">White Rhinoceros Genome</A>
			</UL>
			<P>
			<B>Other Vertebrates</B>
			<UL>
	        	    <LI><A HREF="#atlantic_cod_credits">Atlantic Cod Genome</A>
	        	    <LI><A HREF="#budgerigar_credits">Budgerigar Genome</A>
	        	    <LI><A HREF="#chicken_credits">Chicken Genome</A>
	        	    <LI><A HREF="#coelacanth_credits">Coelacanth Genome</A>
	        	    <LI><A HREF="#fugu_credits">Fugu Genome</A>
	        	    <LI><A HREF="#lamprey_credits">Lamprey Genome</A>
	        	    <LI><A HREF="#lizard_credits">Lizard Genome</A>
	        	    <LI><A HREF="#medaka_credits">Medaka Genome</A>
	        	    <LI><A HREF="#medium_ground_finch_credits">Medium Ground Finch Genome</A>
                    	    <LI><A HREF="#nile_tilapia_credits">Nile Tilapia Genome</A>
                    	    <LI><A HREF="#painted_turtle_credits">Painted Turtle Genome</A>
	        	    <LI><A HREF="#stickleback_credits">Stickleback Genome</A>
	        	    <LI><A HREF="#tetraodon_credits">Tetraodon Genome</A>
	        	    <LI><A HREF="#turkey_credits">Turkey Genome</A>
	        	    <LI><A HREF="#xentro_credits"><em>X. tropicalis</em> Genome</A>
	        	    <LI><A HREF="#zebra_finch_credits">Zebra Finch Genome</A>
	        	    <LI><A HREF="#zebrafish_credits">Zebrafish Genome</A>
			</UL>
			<P>
			<B>Deuterostomes</B>
			<UL>
	        	    <LI><A HREF="#ciona_credits"><em>C. intestinalis</em> Genome</A>
	        	    <LI><A HREF="#lancelet_credits">Lancelet Genome</A>
	        	    <LI><A HREF="#urchin_credits"><em>S. purpuratus</em> Genome</A>
			</UL>
			<P>
			<B>Insects</B>
			<UL>
	        	    <LI><A HREF="#anoGam_credits"><em>A. gambiae</em> Genome</A>
	        	    <LI><A HREF="#apiMel_credits"><em>A. mellifera</em> Genome</A>
	        	    <LI><A HREF="#droAna_credits"><em>D. ananassae</em> Genome</A>
	        	    <LI><A HREF="#droEre_credits"><em>D. erecta</em> Genome</A>
	        	    <LI><A HREF="#droGri_credits"><em>D. grimshawi</em> Genome</A>
	        	    <LI><A HREF="#dm_credits"><em>D. melanogaster</em> Genome</A>
	        	    <LI><A HREF="#droMoj_credits"><em>D. mojavensis</em> Genome</A>
	        	    <LI><A HREF="#droPer_credits"><em>D. persimilis</em> Genome</A>
	        	    <LI><A HREF="#dp_credits"><em>D. pseudoobscura</em> Genome</A>
	        	    <LI><A HREF="#droSec_credits"><em>D. sechellia</em> Genome</A>
	        	    <LI><A HREF="#droSim_credits"><em>D. simulans</em> Genome</A>
	        	    <LI><A HREF="#droVir_credits"><em>D. virilis</em> Genome</A>
	        	    <LI><A HREF="#droYak_credits"><em>D. yakuba</em> Genome</A>
			</UL>
			<P>
			<B>Nematodes</B>
			<UL>
	        	    <LI><A HREF="#worm_credits"><em>C. brenneri</em> Genome</A>
	        	    <LI><A HREF="#worm_credits"><em>C. briggsae</em> Genome</A>
	        	    <LI><A HREF="#ce_credits"><em>C. elegans</em> Genome</A>
	        	    <LI><A HREF="#worm_credits"><em>C. japonica</em> Genome</A>
	        	    <LI><A HREF="#worm_credits"><em>C. remanei</em> Genome</A>
	        	    <LI><A HREF="#worm_credits"><em>P. pacificus</em> Genome</A>
			</UL>
			<P>
			<B>Other</B>
			<UL>
	        	    <LI><A HREF="#yeast_credits">Yeast Genome</A>
	        	    <LI><A HREF="#seahare_credits">California Sea Hare Genome</A>
			</UL>
			<UL>
	        	    <LI><A HREF="#other_credits">Other scientists featured on this site</A>
	        	    <LI><A HREF="#early_credits">Acknowledgment of early support</A>
	        	    <LI><A HREF="../staff.html">UCSC Genome Browser project team</A>
			</UL>
			</P>
		    	</TD>
		    	<TD WIDTH=15></TD>
		    </TR>
		</TABLE>
	    <BR>
	    </TD></TR>
	</TABLE>
	<BR>

<!-- Human Credits ====================================================== -->
<A NAME="human_credits"></A>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	    <TR><TD>
	        <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	            <TR><TD>
	    		<FONT SIZE="4"><B>&nbsp; Human Genome Browser: Acknowledgments </B></FONT>
	    	    </TD></TR>
		</TABLE>
		<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
 	    	    <TR><TD WIDTH=10></TD>
	     		<TD>
			    <TABLE BORDER=0 WIDTH=435 ALIGN="right" CELLPADDING=2 CELLSPACING=2>
		    		<TR><TD><IMG SRC="../images/DOT.gif" WIDTH=10 HEIGHT=1></TD>
				<TD>
				    <TABLE BORDER=1 CELLPADDING=0 CELLSPACING=0 ALIGN="right">
			    	        <TR><TD>
			    		    <TABLE BORDER=1 CELLPADDING=3 CELLSPACING=3>
					        <TR><TD ROWSPAN=2 BGCOLOR="CECE9C" WIDTH="10%">
				    			    <A HREF="http://www.hhmi.org" TARGET="_blank">
				    			    <IMG SRC="../images/HHMI.gif" BORDER=0 ALT="HHMI"></A></TD>
						<TD BGCOLOR="002D59" HEIGHT="38">	
				    			    <A HREF="http://www.ucop.edu/california-institutes" TARGET="_blank">
				    			    <IMG SRC="../images/CISI.gif" BORDER=0 ALT="CISI"></A></TD></TR>
						<TR><TD BGCOLOR="D9E4F8" >			
				    			    <A HREF="http://www.nhgri.nih.gov" TARGET="_blank">
				    			    <IMG SRC="../images/nhgriLogo.gif" BORDER=0 ALT="NHGRI"></A></TD>
						</TR>
			    		   </TABLE>			
				       </TD></TR>
				    </TABLE>		
				</TD></TR>
		    	    </TABLE>	
			<P>The UCSC Human Genome Browser is generated by the 
			<A HREF="../staff.html">UCSC Genome Bioinformatics Group</A>in 
			collaboration with the <A HREF="http://www.genome.gov/10001772" 
			TARGET="_blank">International Human Genome Project</A>. 
			The browser project is funded by grants from the 
			<A HREF="http://www.genome.gov" 
			TARGET="_blank">National Human Genome Research Institute</A>,
			and generous support from the 
			<A HREF="http://www.hhmi.org" 
			TARGET="_blank">Howard Hughes Medical Institute</A> and the 
			<A HREF="http://www.ucop.edu/california-institutes" 
			TARGET="_blank">California Institutes for Science and Innovation</A>.</P>
			We work with collaborators at many other institutions
			to produce and annotate this reference sequence of the human genome.
			To view complete acknowledgments for a specific annotation, see the
			&quot;Credits&quot; section on the track's description page 
			in the Genome Browser. For a list of browser-related publications,
			see our <A HREF="pubs.html">publications page</A>.
			<P>
			UCSC project team acknowledgments: 
			<UL>
		        <LI><B>UCSC Human Genome Browser (hg19):</B> Hiram Clawson, Brooke Rhead, 
			Pauline Fujita, Ann Zweig, Katrina Learned, Donna Karolchik and Robert Kuhn
		        <LI><B>UCSC Human Genome Browser (hg18):</B> Engineering effort led by Fan 
			Hsu; QA effort led by Ann Zweig
		        <LI><B>UCSC Human Genome Browser (hg17):</B> Hiram Clawson, Terry Furey, 
			Heather Trumbower, Robert Kuhn, Donna Karolchik, Kate Rosenbloom, Angie 
			Hinrichs, Rachel Harte, and Jim Kent
		        <LI><B>UCSC Human Genome Browser (hg16):</B> Terry Furey, Hiram Clawson, 
			Heather Trumbower, Mark Diekhans, Robert Baertsch, Donna Karolchik, Jim Kent
		        <LI><B>UCSC Human Genome Browser (hg15):</B> Jim Kent, Terry Furey, Matt Schwartz, 
			Heather Trumbower, Angie Hinrichs, Fan Hsu, Donna Karolchik, Jorge Garcia, 
			Patrick Gavin, Chuck Sugnet, Yontao Lu, Mark Diekhans, Ryan Weber, 
			Robert Baertsch, Krishna Roskin, and the many other students in the UCSC 
			Genome Bioinformatics group
		        <LI><B>Early Human Genome Browsers:</B> UCSC Genome Bioinformatics staff and
			students
			</UL>
		        </TD>
		        <TD WIDTH=15></TD>
	  	    </TR>
	        </TABLE>
	    </TD></TR>
        </TABLE>
	<BR>

<!-- Human Data Sources================================================== -->
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	    <TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		<FONT SIZE="4"><B>&nbsp; Human Genome: Data Sources</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    Starting with the hg19 assembly, the human genome sequence is provided by
		    the <A HREF="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/"
		    TARGET="_blank">Genome Reference Consortium</A> (GRC), whose goal is to correct the 
		    small number of regions in the reference that are currently misrepresented, to 
		    close as many remaining gaps as possible and to produce alternative assemblies of 
		    structurally variant loci when necessary. 
		    <P>
		    Prior to hg19, the human genome sequence data used in the browser were generated 
		    by laboratories belonging to the <A HREF="labs.html">Human Genome 
		    Sequencing Consortium</A>. 
		    Periodic freezes of this data were generated by 
		    <A HREF="http://www.ncbi.nlm.nih.gov/" TARGET="_blank">NCBI</A>, and
		    EST, mRNA, BACend and other types of data used for the assembly and annotation were 
		    obtained from <A HREF="http://www.ncbi.nlm.nih.gov/genbank/"
		    TARGET="_blank">GenBank</A>, <A HREF=" http://www.ebi.ac.uk/embl/" 
		    TARGET="_blank">European Molecular Biology Lab</A> (EMBL), and 
		    <A HREF="http://www.ddbj.nig.ac.jp/" 
		    TARGET="_blank">DNA Data Bank of Japan</A> (DDBJ).
		    Clone maps were created by the individual sequencing centers. 
		    <P>
		    The original chromosome files were produced by the institutions and 
		    people listed in the table below.
		    <P>

	<!-- finished chromosome table-->
		    <TABLE BORDER=1 CELLPADDING=0 WIDTH="100%">
			<TR><TD>
			<TABLE BORDER=1 CELLPADDING=4 WIDTH="100%">
			    <TR><TD COLSPAN=4><B>Finished Chromosomes</B></TD></TR>
			    <TR><TD nowrap><FONT color="#006666"><B>Chr. #</B></FONT></TD>
				<TD nowrap><FONT color="#006666"><B>Sequencing Centers</B></FONT></TD>
				<TD nowrap><FONT color="#006666"><B>AGP File</B></FONT></TD></TR>
			    <TR valign="top"><TD align="center">1</TD>
				<TD>
				    	<A HREF="http://www.sanger.ac.uk/" 
				        TARGET="_blank">The Sanger Institute</A>, Cambridge, UK
					<BR>	
				    	<A HREF="http://www.genome.washington.edu/UWGC/" 
					TARGET="_blank">University of Washington Genome Center</A>, 
					Seattle, WA, USA
				</TD>
				<TD>Jane Rogers</TD>
			    </TR>
			    <TR valign="top">
				<TD align="center">2</TD>
				<TD>
					<A HREF="http://genome.wustl.edu/genomes/view/homo_sapiens"
					TARGET="_blank">Washington University School of Medicine Genome Sequencing Center</A>, 
					St. Louis, MO, USA
				    </TD> 
				<TD nowrap>Rick Wilson</TD>
			    </TR>
			    <TR valign="top">
				<TD align="center">3</TD>
				<TD>
					<A HREF="http://www.hgsc.bcm.tmc.edu/" 
					TARGET="_blank"> Baylor College of Medicine Human Genome Sequencing Center</A>, 
					Houston, TX, USA
					<BR>
					<A HREF="http://www.genomics.cn/en/index"
					TARGET="_blank">Beijing Genomics Institute/Human Genome Center</A>, 
					Beijing, China 
					<BR>
					<A HREF="http://www.gs.washington.edu/" 
					TARGET="_blank">University of Washington Genome Center</A>, 
					Seattle, WA, USA
				</TD> 
				<TD>Steve Scherer</TD>
			    </TR>
			    <TR valign="top">
				<TD align="center">4</TD>
				<TD>
					<A HREF="http://genome.wustl.edu/genomes/view/homo_sapiens"
					TARGET="_blank">Washington University School of Medicine Genome Sequencing Center</A>, 
					St. Louis, MO, USA
					<BR>
					<A HREF="http://shgc.stanford.edu/" 
					TARGET="_blank">Stanford Human Genome Center</A>, Palo Alto, CA, USA
				</TD>
				<TD nowrap>Rick Wilson</TD>
			    </TR>
			    <TR valign="top">
				<TD align="center">5</TD>
				<TD>
					<A HREF="http://genome.jgi-psf.org/chr5/chr5.home.html" 
					TARGET="_blank"> Joint Genome Institute</A>, 
					U.S. Department of Energy, Walnut Creek, CA, USA
					<BR>
					<A HREF="http://shgc.stanford.edu/" 
					TARGET="_blank">Stanford Human Genome Center (Myers)</A>, 
					Palo Alto, CA, USA
				</TD>
				<TD>Jeremy Schmutz</TD>
			    </TR>
			    <TR valign="top"><TD align="center">6</TD>
				<TD>
					<A HREF="http://www.sanger.ac.uk/" 
					TARGET="_blank">The Wellcome Trust Sanger Institute</A>, 
					Cambridge, UK
				</TD> 
				<TD>Jane Rogers</TD>
			    </TR>
			    <TR valign="top"><TD align="center">7</TD>
				<TD>
					<A HREF="http://genome.wustl.edu/genomes/view/homo_sapiens"
					TARGET="_blank">Washington University School of Medicine Genome Sequencing Center</A>, 
					St. Louis, MO USA
					<BR>
					<A HREF="http://www.genome.washington.edu/UWGC/" 
					TARGET="_blank">University of Washington Genome Center</A>, 
					Seattle, WA, USA
				</TD> 
				<TD nowrap>Rick Wilson</TD>
			    </TR>
			    <TR valign="top"><TD align="center">8</TD>
				<TD>
					<A HREF="http://www.broadinstitute.org/"
					TARGET="_blank">The Broad Institute</A>, Cambridge, MA USA
					<BR>
					<A HREF="http://www.keio.ac.jp" 
					TARGET="_blank">Keio University</A>, Tokyo, Japan
					<BR>
					<A HREF="http://genome.imb-jena.de/" 
					TARGET="_blank">Institute of Molecular Biology (IMB)</A>, 
					Jena, Germany
				</TD> 
				<TD>Chad Nusbaum</TD>
			    </TR>
			    <TR valign="top"><TD align="center">9</TD>
				<TD>
					<A HREF="http://www.sanger.ac.uk/" 
					TARGET="_blank">The Wellcome Trust Sanger Institute</A>, 
					Cambridge, UK
					<BR>
					<A HREF="http://www.helmholtz-hzi.de/en/" 
					TARGET="_blank">Gesellschaft fuer Biotechnologische Forschung (GBF)</A>, 
					Braunschweig, Germany 
				</TD>
				<TD>Jane Rogers</TD>
			    </TR>
			    <TR valign="top"><TD align="center">10</TD>
				<TD>
					<A HREF="http://www.sanger.ac.uk/" 
					TARGET="_blank">The Wellcome Trust Sanger Institute</A>, 
					Cambridge, UK
					<BR>
					Genome Therapeutics Corporation (GTC), Waltham, MA, USA
				</TD> 
				<TD>Jane Rogers</TD>
			    </TR>
			    <TR valign="top"><TD align="center">11</TD>
				<TD>
					<A HREF="http://www.gsc.riken.go.jp/eng/index.html" 
					TARGET="_blank">RIKEN Human Genome Research Group</A>, Japan
					<BR>
					<A HREF="http://www.broadinstitute.org/"
					TARGET="_blank">The Broad Institute</A>, Cambridge, MA USA
				</TD>
				<TD>Todd Taylor</TD>
			    </TR>
			    <TR valign="top"><TD align="center">12</TD>
				<TD>
					<A HREF="http://www.hgsc.bcm.edu/" 
					TARGET="_blank"> Baylor College of Medicine Human Genome Sequencing Center</A>, 
					Houston, TX, USA
				</TD>
				<TD>Steve Scherer</TD>
			    </TR>
			    <TR valign="top"><TD align="center">13</TD>
				<TD>
					<A HREF="http://www.sanger.ac.uk/" 
					TARGET="_blank">The Wellcome Trust Sanger Institute</A>, 
					Cambridge, UK
				</TD> 
				<TD>Jane Rogers</TD>
			    </TR>
			    <TR valign="top"><TD align="center">14</TD>
				<TD>
					<A HREF="http://www.genoscope.cns.fr/spip/spip.php?lang=en" 
					TARGET="_blank">Genoscope National Sequencing Centre</A>, 
					Evry, France
					<BR>
					<A HREF="http://www.systemsbiology.org/" 
					TARGET="_blank">Institute for Systems Biology (ISB)</A>, 
					Seattle, WA
					<BR>
					<A HREF="http://genome.wustl.edu/genomes/view/homo_sapiens"
					TARGET="_blank">Washington University School of Medicine Genome Sequencing Center</A>, 
					St. Louis, MO USA
					<BR>
					<A HREF="http://www.gsc.riken.go.jp/eng/index.html" 
					TARGET="_blank">RIKEN Human Genome Research Group</A>, Japan
				</TD> 
				<TD>Jean Weissenbach</TD>
			    </TR>
			    <TR valign="top"><TD align="center">15</TD>
				<TD>
					<A HREF="http://www.broadinstitute.org/"
					TARGET="_blank">The Broad Institute</A>, Cambridge, MA USA
					<BR>
					<A HREF="http://genome.imb-jena.de/" 
					TARGET="_blank">Institute of Molecular Biology (IMB)</A>, 
					<BR>
					<A HREF="http://www.systemsbiology.org/" 
					TARGET="_blank">Institute for Systems Biology (ISB)</A>, 
					Seattle, WA
				</TD> 
				<TD>Chad Nusbaum</TD>
			    </TR>
			    <TR valign="top"><TD align="center">16</TD>
				<TD>
					<A HREF="http://www.jgi.doe.gov/" 
					TARGET="_blank">Joint Genome Institute (JGI)</A>, 
					U.S. Department of Energy, Walnut Creek, CA, USA
					<BR>
					<A HREF="http://shgc.stanford.edu/" 
					TARGET="_blank">Stanford Human Genome Center</A>, 
					Palo Alto, CA, USA
					<BR>
					<A HREF="http://www.jcvi.org/" 
					TARGET="_blank">The Institute for Genomic Research (TIGR)</A>, 
					Rockville, MD, USA
					<BR>
					<A HREF="http://www.sanger.ac.uk/" 
					TARGET="_blank">The Wellcome Trust Sanger Institute</A>, 
					Cambridge, UK
				</TD>
				<TD>Jeremy Schmutz</TD>
			    </TR>
			    <TR valign="top"><TD align="center">17</TD>
				<TD>
					<A HREF="http://www.broadinstitute.org/"
					TARGET="_blank">The Broad Institute</A>, Cambridge, MA USA
				</TD>
				<TD>Chad Nusbaum</TD>
			    </TR>
			    <TR valign="top"><TD align="center">18</TD>
				<TD>
					<A HREF="http://www.broadinstitute.org/"
					TARGET="_blank">The Broad Institute</A>, Cambridge, MA USA
					<BR>
					<A HREF="http://www.gsc.riken.go.jp/eng/index.html" 
					TARGET="_blank">RIKEN Human Genome Research Group</A>, Japan
				</TD>
				<TD>Chad Nusbaum</TD>
			    </TR>
			    <TR valign="top"><TD align="center">19</TD>
				<TD>
					<A HREF="http://genome.jgi-psf.org/Chr19ncbi34/Chr19ncbi34.home.html" 
					TARGET="_blank"> Joint Genome Institute (JGI)</A>, 
					U.S. Department of Energy, Walnut Creek, CA, USA		
					<BR>
					<A HREF="http://shgc.stanford.edu/" 
					TARGET="_blank">Stanford Human Genome Center</A>, 
					Palo Alto, CA, USA
				</TD> 
				<TD>Jeremy Schmutz</TD>
			    </TR>
			    <TR valign="top"><TD align="center">20</TD>
				<TD>
					<A HREF="http://www.sanger.ac.uk/" 
					TARGET="_blank">The Wellcome Trust Sanger Institute</A>, 
					Cambridge, UK
				</TD>
				<TD>James Gilbert</TD>
			     </TR>
			     <TR valign="top"><TD align="center">21</TD>
			    	<TD>
					<A HREF="http://www.gsc.riken.go.jp/eng/index.html" 
					TARGET="_blank">RIKEN Human Genome Research Group</A>, Japan
					<BR>
					<A HREF="http://www.mpg.de/en" 
					TARGET="_blank">Max Planck Institute for Molecular Genetics</A>, 
					Berlin, Germany  
					<BR>
					<A HREF="http://www.helmholtz-hzi.de/en/" 
					TARGET="_blank">Gesellschaft fuer Biotechnologische Forschung (GBF)</A>, 
					Braunschweig, Germany 
					<BR>
					<A HREF="http://www.med.keio.ac.jp/index-e.html" 
					TARGET="_blank">Keio University School of Medicine</A>, 
					Tokyo, Japan
					<BR>
					<A HREF="http://chr21.molgen.mpg.de/index.html" 
                                        TARGET="_blank">Chromosome 21 Consortium</A>
				</TD>
				<TD nowrap>Todd Taylor</TD>
			    </TR>
			    <TR valign="top"><TD align="center">22</TD>
				<TD>
					<A HREF="http://www.sanger.ac.uk/" 
					TARGET="_blank">The Wellcome Trust Sanger Institute</A>, 
					Cambridge, UK
					<BR>
					<A HREF="http://www.genome.ou.edu/" 
					TARGET="_blank">The University of Oklahoma</A>, 
					Norman, OK USA
					<BR>
					<A HREF="http://www.med.keio.ac.jp/index-e.html" 
					TARGET="_blank"> Keio University School of Medicine</A>, 
					Tokyo, Japan
				</TD> 
				<TD>Jane Rogers</TD>
			    </TR>
			    <TR valign="top"><TD align="center">X</TD>
				<TD>
					<A HREF="http://www.sanger.ac.uk/" 
					TARGET="_blank">The Wellcome Trust Sanger Institute</A>, 
					Cambridge, UK
					<BR>
					<A HREF="http://www.hgsc.bcm.edu/" 
					TARGET="_blank"> Baylor College of Medicine Human Genome Sequencing Center</A>, 
					Houston, TX, USA
					<BR>
					<A HREF="http://genome.imb-jena.de/" 
					TARGET="_blank">Institute of Molecular Biology (IMB)</A>, 
					Jena, Germany
					<BR>
					<A HREF="http://genome.wustl.edu/genomes/view/homo_sapiens"
					TARGET="_blank">Washington University School of Medicine Genome Sequencing Center</A>, 
					St. Louis, MO USA
					<BR>
					<A HREF="http://www.mpg.de/en" 
					TARGET="_blank">Max Planck Institute for Molecular Genetics</A>, 
					Berlin, Germany  
				</TD>
				<TD>Jane Rogers</TD>
			    </TR>
			    <TR valign="top"><TD align="center">Y</TD>
				<TD>
					<A HREF="http://genome.wustl.edu/genomes/view/homo_sapiens"
					TARGET="_blank">Washington University School of Medicine Genome Sequencing Center</A>, 
					St. Louis, MO USA
				</TD>
				<TD nowrap>Rick Wilson</TD>
			    </TR>
			</TABLE>
		    	</TD></TR>
		    </TABLE>
	        </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	<BR>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Chimp Credits ====================================================== -->
<A NAME="chimp_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Chimpanzee Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC Chimpanzee Genome Browsers display draft assembly 
		    data produced by the Chimpanzee Genome Sequencing 
		    Consortium: 
		    <UL>
		    <LI><B>Feb. 2011 panTro4 browser:</B> v2.1.4 
		    chromosome-based 6X draft assembly 
		    <LI><B>Oct. 2010 panTro3 browser:</B> v2.1.3 
		    chromosome-based 6X draft assembly 
		    <LI><B>Mar. 2006 panTro2 browser:</B> v2.1 chromosome-based
		    6X draft assembly (Oct. 2005) 
		    <LI><B>Nov. 2003 panTro1 browser:</B> v1.1 Arachne draft 
		    assembly 
		    </UL>
		    The Chimpanzee Genome Browsers have the the following 
		    acknowledgments:
		    <UL>
		    <LI><B>Funding:</B> <A HREF="http://www.nhgri.nih.gov/" 
		    TARGET="_blank">National Human Genome Research Institute</A> 
		    (NHGRI)
		    <BR>
	  	    <LI><B>Sequencing/Assembly (panTro2, panTro3, panTro4):</B>
		    <UL>
		    <LI>
		    Whole genome shotgun (WGS) sequence data assembly and
		    organization: 
		    <A HREF="http://genome.wustl.edu/genomes/view/pan_troglodytes"
		    TARGET="_blank">The Genome Institute at Washington University School of
Medicine</A> (WUSTL), 
		    St. Louis, MO, USA
		    <LI>
		    Underlying WGS data: The Genome Institute at WUSTL and
		    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/chimpanzee/chimpanzee-genome-project" 
		    TARGET =_blank>The Broad Institute</A>, Cambridge, MA USA
		    <LI>
		    5-Mb region of chromosome 7 at chr7:84674857-89461887): finished by The
		    Genome Institute at WUSTL
		    <LI>
		    Chromosome Y sequence: finished by The Genome Institute at WUSTL,
		    with detailed mapping and extensive collaboration with 
		    David Page's group at the <A HREF="http://wi.mit.edu/"
		    TARGET="_blank">Whitehead Institute</A>. For more information, see: 
		    <UL>
		    <LI>
		    Hughes <em>et al</em>. 
		    <A HREF="http://www.nature.com/nature/journal/v437/n7055/full/nature04101.html"
		    TARGET="_blank">Conservation of Y-linked genes during human 
		    evolution revealed by comparative sequencing in chimpanzee</A>. 
		    <em>Nature</em>. 2005 437(7055):100-3.
		    <LI>
		    &quot;The DNA Sequence of Chimpanzee Chromosome Y&quot;, 
		    unpublished.
		    </UL>
		    <LI>
		    Chromosome 21 sequence: Todd Taylor and the 
		    <A HREF="http://www.gsc.riken.go.jp/eng/index.html" 
		    TARGET="_blank">RIKEN Genomic Sciences Center</A>. 
		    See Watanabe <em>et al.</em> 
		    <A HREF="http://www.nature.com/nature/journal/v437/n7055/full/nature04101.html"
		    TARGET="_blank">DNA sequence and comparative analysis of 
		    chimpanzee chromosome 22</A>. 
		    <em>Nature</em>. 2004 May27;429(6990):382-8.
		    </UL>
	  	    <LI><B>Sequencing/Assembly (panTro1):</B>
 		    Sequence assembled by teams led by Eric 
		    Lander, Ph.D. at The Broad Institute and Richard K. Wilson, 
		    Ph.D. at The Genome Institute at WUSTL
		    <BR>
		    <LI><B>Alignments:</B> 
		    LaDeana Hillier, The Genome Institute at WUSTL and The Broad Institute
		    <BR>
		    <LI><B>Annotations:</B> 
		    <UL>
		    <LI>
		    WUSTL, St. Louis, MO, USA
		    <LI>
		    <A HREF="http://www.imim.es/programesrecerca/informaticabiomedica/en_index.html" 
		    TARGET="_blank">Research Group on Biomedical Informatics of IMIM/UPF</A> (GRIB),
		    Barcelona, Spain
		    <LI>
		    Riken Genomic Sciences Center, Yokohama City, Japan 
		    <LI>
		    <A HREF="../staff.html" TARGET="_blank">UCSC Genome Bioinformatics Group</A>, 
		    Santa Cruz, CA, USA
		    </UL>
		    <LI><B>UCSC Chimp Genome Browser (panTro4):</B> 
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI><B>UCSC Chimp Genome Browser (panTro3):</B> Hiram
		    Clawson, Greg Roe, and Donna Karolchik
		    <LI><B>UCSC Chimp Genome Browser (panTro2):</B> Kate 
		    Rosenbloom, Brian Raney, Hiram Clawson, Ann Zweig, Archana 
		    Thakkapallayil, and Donna Karolchik 
		    <LI><B>UCSC Chimp Genome Browser (panTro1):</B> Kate Rosenbloom, Jim 
		    Kent, Hiram Clawson, Heather Trumbower, Robert Kuhn, and
		    Donna Karolchik
		    </UL>
		    <P>
		    <HR>
		    <A NAME="chimp_use"></A>
		    The initial sequencing and analysis of the chimpanzee genome
		    was published by the Chimpanzee Sequencing and Analysis Consortium in 
		    <A HREF="http://www.nature.com/nature/journal/v437/n7055/full/nature04072.html"
		    TARGET="_blank">Initial sequence of the chimpanzee genome and comparison with the human genome</A>. 
		    <em>Nature</em>. 2005 Sep 1;437(7055):69-87.
		    <P>
		    The chimpanzee sequence is made freely available for public use by the 
		    Chimpanzee Genome Sequencing Consortium with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers analyzing
			particular genes and regions if the providers of this
			data (the Chimpanzee Genome Sequencing Consortium) are properly 
			acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Rhesus Credits=================================================== -->
<A NAME="rhesus_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Rhesus macaque Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC rheMac3 Genome Browser displays data from the Oct. 2010 assembly of 
		    the rhesus monkey, version CR_1.0 from <A HREF="http://en.genomics.cn/" 
		    TARGET=_blank>Beijing Genomics Institute-Shenzhen</A> (BGI), Shenzhen, China. 
		    The BGI assembly is provided with the following acknowledgments:
		    <UL>
	   	    <LI>
		    <B>Funding:</B> Southern China Center for Innovative Pharmaceuticals (SCCIP), 
		    the National Natural Science Foundation of China, the Shenzhen Municipal 
		    Government of China, the Major State Basic Research Development Program of 
		    China, the National Basic Research Program of China, and the National Science 
		    and Technology Major Project of Key Drug Innovation and Development
	   	    <LI>
		    <B>Sequencing/Assembly:</B> BGI, Shenzhen, China 
		    <LI><B>UCSC Rhesus Genome Browser/Initial Annotations (rheMac3):</B> 
		    <A HREF="../staff.html" TARGET="_blank">UCSC Genome Bioinformatics Group</A>, 
		    Santa Cruz, CA, USA - Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, 
		    Luvina Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    </UL>
		    </P>
		    <P>
		    The UCSC rheMac2 and rheMac1 Genome Browsers display data from the Macaca 
		    mulatta Genome Sequencing Consortium (MMGCS) led by the 
		    <A HREF="https://www.hgsc.bcm.edu/content/rhesus-monkey-genome-project"
		    TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center</A>
		    (BCM HGSC), in collaboration with the <A HREF="http://www.jcvi.org/" 
		    TARGET="_blank">J. Craig Venter Science Foundation Joint Technology Center</A>
		    and <A HREF="http://genome.wustl.edu/genomes/view/macaca_mulatta"
		    TARGET="_blank">The Genome Institute at Washington University School of 
		    Medicine</A> (WUSTL). The MMGSC assemblies are provided with the following 
		    acknowledgments:
		    <UL>
	   	    <LI>
		    <B>Funding: </B> 
		    <A HREF="http://www.genome.gov/" 
		    TARGET="_blank">National Human Genome Research Institute</A> (NHGRI), USA
		    <LI>
		    <B>Sequencing/Assembly: </B>
		    BCM HGSC, Houston, TX, USA in collaboration with:
		      <UL>
		      <LI>
		      J. Craig Venter Science Foundation Joint Technology Center, Rockville, MD, USA
		      <LI>
		      The Genome Institute at WUSTL, St. Louis, MO, USA
		      </UL>
		    <LI>
		    <B>BAC resources: </B>
		    <A HREF="http://bacpac.chori.org/rhesus250.htm"
		    TARGET="_blank">Children's Hospital Oakland Research Institute</A> (CHORI), 
		    Oakland, CA, USA
		    <LI>
		    <B>BAC-based fingerprint map: </B>
		    <A HREF="http://www.bcgsc.ca/lab/mapping/data"
		    TARGET=-blank>Genome Sciences Centre</A>, Vancouver, B.C.
		    <LI><B>UCSC Rhesus Genome Browser/Initial Annotations (rheMac2, MMGSC Jan. 2006 
		    version Mmul_051212 assembly):</B> 
		    <A HREF="../staff.html" TARGET="_blank">UCSC Genome Bioinformatics Group</A>, 
		    Santa Cruz, CA, USA - Robert Baertsch, Kayla Smith, Ann Zweig, Robert Kuhn, and 
		    Donna Karolchik. 
		    <LI><B>UCSC Rhesus Genome Browser/Initial Annotations (rheMac1, MMGSC Jan. 2005 
	 	    version Mmul_0.1 assembly): </B> 
		    <A HREF="../staff.html" TARGET="_blank">UCSC Genome Bioinformatics Group</A>, 
		    Santa Cruz, CA, USA - Robert Baertsch, Galt Barber, and Donna Karolchik. 
		    </UL>
		    <P>
		    For more information on the MMGSC rhesus macaque genome project, see the BCM 
		    HGSC 
		    <A HREF="https://www.hgsc.bcm.edu/content/rhesus-monkey-genome-project"
		    TARGET="_blank">Rhesus Monkey Genome Project</A> web page.
		    <A NAME="rhesus_use"></A>
		    <P>
		    <HR>
		    The initial sequencing and analysis of the rhesus macaque genome
		    produced by the MMGSC was published in Katze M <em>et al</em>.
		    <A HREF="http://www.sciencemag.org/content/316/5822/222.long"
		    TARGET="_blank">Evolutionary and Biomedical Insights from the Rhesus Macaque 
		    Genome</A>.  <em>Science</em>. 2007 Apr 13;316(5822):222-34.
		    <P>
		    The initial sequencing and analysis of the rhesus macaque genome produced by 
		    BGI was published in Yan G <em>et al</em>. 
		    <A HREF="http://www.nature.com/nbt/journal/v29/n11/full/nbt.1992.html"
		    TARGET=_blank>Genome sequencing and comparison of two nonhuman primate animal 
		    models, the cynomolgus and Chinese rhesus macaques</A>. <em>Nat Biotechnol.</em>
		    2011 Oct 16;29(11):1019-23. 
		    <P>
		    The rhesus macaque sequence is made freely available for public use 
		    with the following understanding:
		    <OL>
		    <LI>The data may be freely downloaded, used in analyses, 
		    and repackaged in databases.
		    <LI>Users are free to use the data in scientific papers 
		    analyzing particular genes and regions if the providers of 
		    this data are properly acknowledged. For the rheMac1 and rheMac2 data, please 
		    cite the BCM HGSC web site or publications from BCM HGSC referring to
		    the genome sequence.
		    <LI>The BCM HGSC and RMGSC plan to publish the assembly and 
		    genomic annotation of the dataset, including large-scale 
		    identification of regions of evolutionary conservation.
		    <LI>This is in accordance with, and with the understandings 
		    in the <A HREF="ftLauderdale.pdf">Fort Lauderdale meeting</A> discussing 
		    Community Resource Projects and the resulting
		    <A HREF="http://www.genome.gov/page.cfm?pageID=10506537"
		    TARGET="_blank">NHGRI policy statement</A>.
		    <LI>Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Rhesus Credits=================================================== -->
<A NAME="rhesus_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Rhesus macaque Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC Genome Browser displays multiple assemblies of the Rhesus macaque
		    genome produced by different institutions:
		    <UL>
		    <LI>
		    <B>rheMac3 Genome Browser:</B> Oct. 2010 draft assembly of the rhesus monkey, 
		    version CR_1.0 from <A HREF="http://en.genomics.cn/" 
		    TARGET=_blank>Beijing Genomics Institute-Shenzhen</A> (BGI), Shenzhen, China 
		    <LI>
		    <B>rheMac2 Genome Browser:</B> Jan. 2006 draft assembly of the rhesus monkey, 
		    version Mmul_051212 from the Macaca mulatta Genome Sequencing Consortium (MMGCS)
		    led by the 
		    <A HREF="https://www.hgsc.bcm.edu/content/rhesus-monkey-genome-project"
		    TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center</A>
		    (BCM HGSC), in collaboration with the <A HREF="http://www.jcvi.org/" 
		    TARGET="_blank">J. Craig Venter Science Foundation Joint Technology Center</A>
		    and <A HREF="http://genome.wustl.edu/genomes/view/macaca_mulatta"
		    TARGET="_blank">The Genome Institute at Washington University School of 
		    Medicine</A> (WUSTL)
		    <LI>
		    <B>rheMac1 Genome Browser:</B> Jan. 2005 draft assembly of the rhesus monkey, 
		    version Mmul_0.1 from MMGSC (see above)
		    </UL></P>
		    <P>
		    The BGI assembly is provided with the following acknowledgments:
		    <UL>
	   	    <LI>
		    <B>Funding:</B> Southern China Center for Innovative Pharmaceuticals (SCCIP), 
		    the National Natural Science Foundation of China, the Shenzhen Municipal 
		    Government of China, the Major State Basic Research Development Program of 
		    China, the National Basic Research Program of China, and the National Science 
		    and Technology Major Project of Key Drug Innovation and Development
	   	    <LI>
		    <B>Sequencing/Assembly:</B> BGI, Shenzhen, China 
		    </UL>
		    <P>
		    The MMGSC assembly is provided with the following acknowledgments:
		    <UL>
	   	    <LI>
		    <B>Funding: </B> 
		    <A HREF="http://www.genome.gov/" 
		    TARGET="_blank">National Human Genome Research Institute</A> (NHGRI), USA
		    <LI>
		    <B>Sequencing/Assembly: </B>
		    BCM HGSC, Houston, TX, USA in collaboration with:
		      <UL>
		      <LI>
		      J. Craig Venter Science Foundation Joint Technology Center, Rockville, MD, USA
		      <LI>
		      The Genome Institute at WUSTL, St. Louis, MO, USA
		      </UL>
		    <LI>
		    <B>BAC resources: </B>
		    <A HREF="http://bacpac.chori.org/rhesus250.htm"
		    TARGET="_blank">Children's Hospital Oakland Research Institute</A> (CHORI), 
		    Oakland, CA, USA
		    <LI>
		    <B>BAC-based fingerprint map: </B>
		    <A HREF="http://www.bcgsc.ca/lab/mapping/data"
		    TARGET=-blank>Genome Sciences Centre</A>, Vancouver, B.C.
		    <LI><B>UCSC Rhesus Genome Browser/Initial Annotations (rheMac3):</B> 
		    <A HREF="../staff.html" TARGET="_blank">UCSC Genome Bioinformatics Group</A>, 
		    Santa Cruz, CA, USA - Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, 
		    Luvina Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI><B>UCSC Rhesus Genome Browser/Initial Annotations (rheMac2):</B> 
		    <A HREF="../staff.html" TARGET="_blank">UCSC Genome Bioinformatics Group</A>, 
		    Santa Cruz, CA, USA - Robert Baertsch, Kayla Smith, Ann Zweig, Robert Kuhn, and 
		    Donna Karolchik. 
		    <LI><B>UCSC Rhesus Genome Browser/Initial Annotations (rheMac1): </B> 
		    <A HREF="../staff.html" TARGET="_blank">UCSC Genome Bioinformatics Group</A>, 
		    Santa Cruz, CA, USA - Robert Baertsch, Galt Barber, and Donna Karolchik. 
		    </UL>
		    <P>
		    For more information on the MMGSC rhesus macaque genome project, see the BCM 
		    HGSC 
		    <A HREF="https://www.hgsc.bcm.edu/content/rhesus-monkey-genome-project"
		    TARGET="_blank">Rhesus Monkey Genome Project</A> web page.
		    <A NAME="rhesus_use"></A>
		    <P>
		    <HR>
		    The initial sequencing and analysis of the rhesus macaque genome
		    produced by the MMGSC was published in Katze M <em>et al</em>.
		    <A HREF="http://www.sciencemag.org/content/316/5822/222.long"
		    TARGET="_blank">Evolutionary and Biomedical Insights from the Rhesus Macaque 
		    Genome</A>.  <em>Science</em>. 2007 Apr 13;316(5822):222-34.
		    <P>
		    The initial sequencing and analysis of the rhesus macaque genome produced by 
		    BGI was published in Yan G <em>et al</em>. 
		    <A HREF="http://www.nature.com/nbt/journal/v29/n11/full/nbt.1992.html"
		    TARGET=_blank>Genome sequencing and comparison of two nonhuman primate animal 
		    models, the cynomolgus and Chinese rhesus macaques</A>. <em>Nat Biotechnol.</em>
		    2011 Oct 16;29(11):1019-23. 
		    <P>
		    The rhesus macaque sequence is made freely available for public use 
		    with the following understanding:
		    <OL>
		    <LI>The data may be freely downloaded, used in analyses, 
		    and repackaged in databases.
		    <LI>Users are free to use the data in scientific papers 
		    analyzing particular genes and regions if the providers of 
		    this data are properly acknowledged. For the rheMac1 and rheMac2 data, please 
		    cite the BCM HGSC web site or publications from BCM HGSC referring to
		    the genome sequence.
		    <LI>The BCM HGSC and RMGSC plan to publish the assembly and 
		    genomic annotation of the dataset, including large-scale 
		    identification of regions of evolutionary conservation.
		    <LI>This is in accordance with, and with the understandings 
		    in the <A HREF="ftLauderdale.pdf">Fort Lauderdale meeting</A> discussing 
		    Community Resource Projects and the resulting
		    <A HREF="http://www.genome.gov/page.cfm?pageID=10506537"
		    TARGET="_blank">NHGRI policy statement</A>.
		    <LI>Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Alpaca Credits ======================================================= -->
<A NAME="alpaca_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Alpaca Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Alpaca Genome Browser is provided with the following acknowledgments:
		    <UL>
		    <LI><B>Sequencing/Assembly: </B>
	  	    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		    <LI>
		    <B>UCSC Alpaca Genome Browser (vicPac1):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="alpaca_use"></A>
		    The alpaca sequence has been made freely available as 
		    part of the Mammalian Genome Project (29 Mammals Project). 
		    The initial analysis of this dataset can be found in 
		    Lindblad-Toh <em>et al.</em>, 
		    <A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		    TARGET=_blank>A high-resolution map of human evolutionary constraint using 29 
		    mammals</A>, <em>Nature</em> 2011 Oct 12;478(7370):476-82. 
		    The alpaca data are provided with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (The Broad Institute) are properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Armadillo Credits ======================================================= -->
<A NAME="armadillo_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Armadillo Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Armadillo Genome Browser is provided with the following acknowledgments:
		    <UL>
		    <LI><B>Sequencing/Assembly: </B>
	  	    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		    <LI>
		    <B>UCSC Armadillo Genome Browser (dasNov3):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="armadillo_use"></A>
		    The armadillo sequence has been made freely available as 
		    part of the Mammalian Genome Project (29 Mammals Project). 
		    The initial analysis of this dataset can be found in 
		    Lindblad-Toh <em>et al.</em>, 
		    <A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		    TARGET=_blank>A high-resolution map of human evolutionary constraint using 29 
		    mammals</A>, <em>Nature</em> 2011 Oct 12;478(7370):476-82. 
		    The armadillo data are provided with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (The Broad Institute) are properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Atlantic Cod Credits ======================================================= -->
<A NAME="atlantic_cod_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Atlantic Cod Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Atlantic Cod Genome Browser is provided with the following acknowledgments:
		    <UL>
		    <LI><B>Funding:</B>
		    <A HREF="http://www.forskningsradet.no/prognett-fuge/Programme_description/1226993493179"
		    TARGET=_blank>Research Council of Norway</A> (FUGE)
		    <LI><B>Sequencing/Assembly: </B>
	  	    <A HREF="http://www.codgenome.no/index.php"
		    TARGET="_blank">GenoFisk Consortium</A>
		    <LI>
		    <B>UCSC Atlantic Cod Genome Browser (gadMor1):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="atlantic_cod_use"></A>
		    The initial analyses of the Atlantic cod dataset can be found in 
		    Johansen SD <em>et al</em>.
		    <A HREF="http://www.sciencedirect.com/science/article/pii/S1871678409000569"
		    TARGET=_blank>Large-scale sequence analyses of Atlantic cod</A>.
		    <em>N Biotechnol</em>. 2009 Jun;25(5):263-71. 
		    The Atlantic cod sequence is made freely available with the following 
		    understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers 
			of these data are properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Baboon Credits ======================================================= -->
<A NAME="baboon_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Baboon Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Baboon Genome Browser is provided by the Baboon Genome Sequencing Consortium
		    with the following acknowledgments:
		    <UL>
		    <LI><B>Funding: </B> 
		    <A HREF="http://www.genome.gov/" 
		    TARGET="_blank">National Human Genome Research Institute</A>
		    (NHGRI), USA
		    <LI>
		    <B>Sequencing/Assembly: </B>
		    <A HREF="https://www.hgsc.bcm.edu/content/baboon-genome-project" 
		    TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center</A>
		    (BCM-HGSC), Houston, TX, USA 
		    <LI>
		    <B>UCSC Baboon Genome Browser (papHam1):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>UCSC Baboon Genome Browser (papAnu2):</B>
		    Chin Li and Luvina Guruvadoo
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="baboon_use"></A>
		    The baboon sequence is made freely available for public use by the 
		    Baboon Genome Sequencing Consortium. Please review the BCM-HGSC 
		    <A HREF="https://www.hgsc.bcm.edu/content/conditions-use"
		    TARGET="_blank">conditions for use</A> guidelines before using this data.
		    </P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Budgerigar Credits ======================================================= -->
<A NAME="budgerigar_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Budgerigar Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Budgerigar Genome Browser is provided with the following acknowledgments:
		    <UL>
		    <LI><B>Sequencing/Assembly: </B>
	  	    <A HREF="http://genome.wustl.edu/genomes/"
		    TARGET="_blank">The Genome Institute at Washington University School of
		    Medicine</A> (WUSTL), St. Louis, MO, USA 
		    <LI>
		    <B>UCSC Budgerigar Genome Browser (melUnd1):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="budgerigar_use"></A>
		    The budgerigar sequence has been made freely available by The Genome Institute 
		    at WUSTL.  Please review the WUSTL 
		    <A HREF="http://genome.wustl.edu/data/data_use_policy"
		    TARGET="_blank">data use policy</A> before using these data.
		    </P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Bushbaby Credits ======================================================= -->
<A NAME="bushbaby_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Bushbaby Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Bushbaby Genome Browser is provided with the following acknowledgments:
		    <UL>
	   	    <LI><B>Funding:</B>
		    <A HREF="http://www.genome.gov" 
		    TARGET="_blank">National Human Genome Research Institute</A> (NHGRI)
		    <LI><B>Sequencing/Assembly: </B>
	  	    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		    <LI>
		    <B>UCSC Bushbaby Genome Browser (otoGar3):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="bushbaby_use"></A>
		    The bushbaby sequence has been made freely available as 
		    part of the Mammalian Genome Project (29 Mammals Project). 
		    The initial analysis of this dataset can be found in 
		    Lindblad-Toh <em>et al.</em>, 
		    <A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		    TARGET=_blank>A high-resolution map of human evolutionary constraint using 29 
		    mammals</A>, <em>Nature</em> 2011 Oct 12;478(7370):476-82. 
		    The bushbaby data are provided with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (The Broad Institute) are properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Cat Credits ====================================================== -->
<A NAME="cat_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Cat Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC felCat5 Genome Browser displays data from the 
		    Sep. 2011 Felis_catus-6.2 assembly of the domestic cat produced by the 
		    International Cat Genome Sequencing Consortium.  This assembly was provided 
		    with the following acknowledgments:
		    <UL>
			<LI><B>Funding:</B> <A HREF="http://www.genome.gov/" 
			TARGET="_blank">National Human Genome Research Institute</A> (NHGRI), 
			Bethesda, MD
			<BR>
	  		<LI><B>454 Production sequencing: </B>
		        <A HREF="http://genome.wustl.edu/genomes/view/felis_catus/"
		    	TARGET="_blank">The Genome Institute at Washington University School of
			Medicine</A> (WUSTL), St. Louis, MO, USA
             		<LI><B>BAC sequencing:</B> WUSTL
			<LI><B>Plasmid and fosmid sequencing:</B> 
			<A HREF="http://www.broad.mit.edu/" TARGET="_blank">The Broad Institute</A>,
			Cambridge, MA, USA and <A HREF="http://www.agencourt.com/" 
			TARGET="_blank">Agencourt Bioscience</A>
             		<LI><B>Sequence assembly and data integration for creation of chromosomal
			AGP files:</B> WUSTL
             		<LI><B>Whole genome alignments:</B> Brian Raney, 
			<A HREF="../index.html">UCSC Genome Browser Group</A>, University of 
		 	California, Santa Cruz, CA, and WUSTL
             		<LI><B>Cat RH map:</B> Bill Murphy, <A HREF="http://genetics.tamu.edu/"
			TARGET=_blank>Texas A&M University</A>, College Station, TX.
 			<LI><B>DNA source:</B> Dr. Kristina Narfstrom, 
			<A HREF="http://www.vmth.missouri.edu/" 
			TARGET=_blank>University of Missouri</A>, Columbia, MO.
			<LI><B>UCSC Cat Genome Browser/Initial Annotations (felCat5):</B> 
			<A HREF="../staff.html" 
			TARGET="_blank">UCSC Genome Bioinformatics Group</A>, Santa Cruz, CA, USA - 
		    	Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina Guruvadoo, 
		 	Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    </UL>
		    <P>
		    The UCSC felCat4 Genome Browser displays data from the
		    Dec. 2008 catChrV17e draft assembly, based on sequence
		    reads from six domestic cats and one wild cat. The
		    browser is provided  with the following acknowledgments:
		    <UL>
	 	      <LI><B>Sequencing/Assembly:</B> 
		        <UL>
		          <LI>
		      	    <A HREF="http://www.genome.gov/10000016"
		      	    TARGET="_blank">NHGRI Genome Technology Branch and NIH
		      	    Intramural Sequencing Center</A>, Bethesda, 
			    MD, USA 
		      	  <LI>
		      	    <A HREF="http://www.beckmangenomics.com/"
	   	      	    TARGET="_blank">Agencourt Bioscience Corporation</A>,
		      	    Danvers, MA, USA
		      	  <LI>
		      	    <A HREF="http://www.cancer.gov/"
		      	    TARGET="_blank">NCI Laboratory of Genomic Diversity</A>, 
			    Frederick, MD, USA 
		      	  <LI>
		      	    <A HREF="http://www.hillspet.com"
		      	    TARGET="_blank">Hill's Pet Nutrition, Inc.</A>, 
		   	    Topeka, KS, USA
		  	</UL>
		       <LI><B>UCSC Cat Genome Browser/Initial Annotations 
		       (felCat4):</B> 
			 <A HREF="../staff.html" 
			 TARGET="_blank">UCSC Genome Bioinformatics Group</A>, Santa Cruz, CA, USA -
			 Chin Li, Hiram Clawson, Antonio Coelho, Mary
			 Goldman, and Donna Karolchik
		    </UL>
		    <P>
		    The UCSC felCat3 Genome Browser displays data from the 
		    March 2006 v3 draft assembly of the domestic cat (<em>Felis catus</em>). 
		    This assembly was provided with the following acknowledgments:
		    <UL>
			<LI><B>Funding:</B> <A HREF="http://www.genome.gov/" 
			TARGET="_blank">National Human Genome Research Institute</A> (NHGRI),
			Bethesda, MD
			<BR>
	  		<LI><B>Project Leadership:</B>
			<A HREF="http://www.broadinstitute.org/science/projects/mammals-models/data-release-mammalian-genome-project"
			TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
	  		<LI><B>Sequencing/Assembly:</B>
			The Broad Institute - team led by by Kerstin Lindblad-Toh, Ph.D.
			<LI><B>Plasmid and fosmid libraries:</B> The Broad Institute
			<LI><B>BAC end sequencing:</B> 
		      	<A HREF="http://www.beckmangenomics.com/"
			TARGET="_blank">Agencourt Bioscience</A>, Beverly, MA, USA
			<BR>
			<LI><B>UCSC Cat Genome Browser/Initial Annotations (felCat3):</B> 
			<A HREF="../staff.html" 
			TARGET="_blank">UCSC Genome Bioinformatics Group</A>, Santa Cruz, CA, USA - 
			Heather Trumbower, Angie Hinrichs, Mark Diekhans,
			Brooke Rhead, and Archana Thakkapallayil
		    </UL>
		    <P>
		    <HR>
		    <A NAME="cat_use"></A>
		    <P>
		    The catChrV17e cat assembly was initially published in  	    
		    Pontius JU <em>et al</em>. 
		    <A HREF="http://genome.cshlp.org/content/17/11/1675.long"
		    TARGET=_blank>Initial sequence and comparative analysis of the cat genome</A>. 
		    <em>Genome Res</em>. 2007 Nov;17(11):1675-89.
		    <P>
		    The Broad Institute released the initial 2X (v3) assembly of the cat genome as 
		    part of the Mammalian Genome Project (29 Mammals Project). 
		    The initial analysis of this dataset can be found in 
		    Lindblad-Toh <em>et al.</em>. 
		    <A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		    TARGET="_blank">A high-resolution map of human evolutionary constraint using 29 
		    mammals</A>. <em>Nature</em> 2011 Oct 12;478(7370):476-82. 
		    <P>
		    The cat sequence is made freely available with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers 
			of these data are properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>


<!-- Dog Credits ====================================================== -->
<A NAME="dog_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Dog Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC Dog Genome Browser displays data from the 
		    July 2004 v1.0 (canFam1) and May 2005 v2.0 (canFam2) draft 
		    assemblies of the domestic dog (<em>Canis familiaris</em>). 
		    These assemblies are provided with the following 
		    acknowledgments:
		    <UL>
			<LI><B>Funding:</B> The 
			<A HREF="http://www.genome.gov/" 
			TARGET="_blank">National Human Genome Research Institute</A>
			<BR>
	  		<LI><B>Project Leadership:</B>
			<A HREF="http://www.broadinstitute.org/"
			TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
	  		<LI><B>Sequencing/Assembly:</B>
			<A HREF="http://www.broadinstitute.org/"
			TARGET="_blank">The Broad Institute</A> - 
			team led by by Kerstin Lindblad-Toh, Ph.D.
			<LI><B>Plasmid and fosmid libraries:</B> 
			<A HREF="http://www.broadinstitute.org/"
			TARGET="_blank">The Broad Institute</A>
			<LI><B>BAC libraries:</B> 
			<A HREF="http://www.chori.org/" 
			TARGET="_blank">Children's Hospital Oakland Research Institute</A> 
			(CHORI), Oakland, CA, USA
			<LI><B>BAC end sequencing:</B> 
			<A HREF="http://www.agencourt.com/" 
			TARGET="_blank">Agencourt Bioscience</A>, Beverly, MA, USA
			<LI><B>RH map:</B> 
			<A HREF="http://www-recomgen.univ-rennes1.fr/Dogs/maquette.html" 
			TARGET="_blank">Centre National de la Recherche Scientifique</A> 
			(CNRS), France
			<LI><B>FISH mapping:</B> <A HREF="http://www.ncsu.edu/"
			TARGET="_blank">North Carolina State University</A> (NCSU),
			Raleigh, NC, USA
			<BR>
			<LI><B>Dog Selection:</B>
			<A HREF="http://www.fhcrc.org/science/en/labs.html"
			TARGET="_blank">Fred Hutchinson Cancer Research Center</A>
			(FHCRC), Seattle, WA, USA
 			<LI><B>UCSC Dog Genome Browser/Initial Annotations 
                        (canFam3):</B>
                        <A HREF="../staff.html" 
                        TARGET="_blank">UCSC Genome Bioinformatics Group</A>,
                        Santa Cruz, CA, USA -
                        Hiram Clawson and Luvina Guruvadoo
			<LI><B>UCSC Dog Genome Browser/Initial Annotations 
			(canFam2):</B> 
			<A HREF="../staff.html" 
			TARGET="_blank">UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Angie Hinrichs, Jennifer Jackson, and Donna Karolchik
			<LI><B>UCSC Dog Genome Browser/Initial Annotations 
			(canFam1):</B> 
			<A HREF="../staff.html" 
			TARGET="_blank">UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Angie Hinrichs, Robert Kuhn and Donna Karolchik
		    </UL>
		    <P>
		    <HR>
		    <A NAME="dog_use"></A>
		    <P>The dog genome sequence is made freely available by the 
		    <A HREF="http://www.broadinstitute.org/mammals/dog/" 
		    TARGET="_blank">Dog Genome Sequencing Project</A>. 
		    Please cite the following publication when using these data:
		    Lindblad-Toh K, <em>et al</em>. 
		    <A HREF="http://www.nature.com/nature/journal/v438/n7069/abs/nature04338.html" 
		    TARGET="_blank">Genome sequence, comparative analysis and haplotype structure of the domestic dog</A>. 
		    <EM>Nature</EM>. 2005 Dec 8;438:803-19.</P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>


<!-- Cow Credits====================================================== -->
<A NAME="cow_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Cow Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC Cow Genome Browser displays two different 
		    assemblies of the bovine genome: one provided by the 
		    University of Maryland Center for Bioinformatics and 
		    Computational Biology (UMD CBCB) and one from the Baylor 
		    College of Medicine Human Genome Sequencing Center (BCM 
		    HGSC). The assemblies available in the browser include:
		    <UL>
		      <LI>Oct. 2011 (bosTau7) assembly - Bovine Genome
Sequencing Consortium (BCM HGSC) version Btau_4.6.1
		      <LI>Nov. 2009 (bosTau6) assembly - UMD CBCB version UMD_3.1
		      <LI>Oct. 2007 (bosTau4) assembly - BCM HGSC version Btau_4.0
		      <LI>Aug. 2006 (bosTau3) assembly - BCM HGSC version Btau_3.1
		      <LI>Mar. 2005 (bosTau2) assembly - BCM HGSC version Btau_2.0
		      <LI>Sep. 2004 (bosTau1) assembly - BCM HGSC version Btau_1.0
		    </UL></P>
		<P>
		The UMD CBCB genome assembly is provided with the following 
		acknowledgments:
		<UL>
		    <LI><B>Genome sequence: </B> 
		        NCBI Trace Archive
		    <LI><B>Genome assembly: </B> 
			<A HREF="http://www.cbcb.umd.edu/" 
			TARGET="_blank">University of Maryland Center for
Bioinformatics and Computational Biology</A>,
			College Park, MD, USA 
		    <LI><B>UCSC Cow Genome Browser/Initial Annotations (bosTau6): </B> 
			<A HREF="../staff.html">UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Hiram Clawson, Chin Li, Brooke Rhead, Greg Roe, Steve 
			Heitner, and Donna Karolchik
		</UL>
		<P>
		<P>The BCM HGSC genome assemblies are provided with the
		following acknowledgments:
		<UL>
		<LI><B>Funding</B> - For a list of the many agencies that
		funded the cow sequencing project, see the BCM HGSC
		<A HREF="http://www.hgsc.bcm.edu/content/bovine-genome-project/"
		TARGET="_blank">Bovine Genome Project</A> page.
		    <LI><B>BAC library DNA: </B>
			Dr. Michael MacNeil's laboratory at the
			<A HREF="http://www.ars.usda.gov/main/main.htm"
			TARGET="_blank">USDA Agricultural Research Service</A>, 
		    	Miles City, MT, USA
		    <LI><B>Whole genome shotgun sequence DNA: </B>
			Dr. Timothy Smith's laboratory at the
			<A HREF="http://www.marc.usda.gov/"
			TARGET="_blank">U.S. Meat Animal Research Center</A>, 
		 	Clay Center, NE, USA
		    <LI><B>BAC-based fingerprint map: </B>
			<A HREF="http://www.bcgsc.ca/lab/mapping/bovine"
			TARGET=-blank>Genome Sciences Centre</A>, Vancouver, 
			B.C.
		    <LI><B>UCSC Cow Genome Browser/Initial Annotations (bosTau7): </B> 
			<A HREF="../staff.html">UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Chin Li, Hiram Clawson, Brian Raney and Steve Heitner. 
		    <LI><B>UCSC Cow Genome Browser/Initial Annotations (bosTau4): </B> 
			<A HREF="../staff.html">UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Hiram Clawson, Brian Raney, and Ann Zweig. 
		    <LI><B>UCSC Cow Genome Browser/Initial Annotations (bosTau3): </B> 
			<A HREF="../staff.html">UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Heather Trumbower, Angie Hinrichs, Kayla Smith and 
		        Donna Karolchik. 
		    <LI><B>UCSC Cow Genome Browser/Initial Annotations (bosTau2): </B> 
			<A HREF="../staff.html">UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Galt Barber, Brian Raney, Mark Diekhans, Jennifer Jackson,
			and Donna Karolchik. 
		    <LI><B>UCSC Cow Genome Browser/Initial Annotations (bosTau1): </B> 
			<A HREF="../staff.html">UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Heather Trumbower, Jim Kent, Hiram Clawson, Angie
			Hinrichs, Brian Raney, Mark Diekhans, Robert Kuhn,
			Ali Sultan-Qurraie and Donna Karolchik. 
		    </UL>
		    <HR>
		    <A NAME="cow_use"></A>
		    Please acknowledge the appropriate institution when using
		    these data sets. Citation information may be obtained from
		    the UMD CBCB 
		    <A HREF="http://www.cbcb.umd.edu/research/bos_taurus_assembly.shtml"
		    TARGET="_blank">Bos taurus assembly</A> page and the BCM HGSC
		    <A HREF="https://www.hgsc.bcm.edu/content/bovine-genome-project"
		    TARGET="_blank">Bovine Genome Project</A> page. 
		    </P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Mouse Credits ====================================================== -->
<A NAME="mouse_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Mouse Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		<P>The UCSC Mouse Genome Project is conducted in collaboration
	 	with the Mouse Sequencing Consortium and the Mouse Genome 
		Sequencing Consortium. The mouse draft assembly displayed in 
		the Genome Browser is assembled at NCBI.
		<P>
		<HR>
		The initial 3X coverage of the mouse genome was produced by 
		the Mouse Sequencing Consortium, with the following 
		acknowledgments: 
		<UL>
		<LI><B>Project Oversight:</B> Arthur Holden and Francis 
		Collins
		<LI><B>Funding:</B> 
		  <UL>
		  <LI><A HREF="http://www.gsk.com/" 
		      TARGET="_blank">GlaxoSmithKline</A>
		  <LI><A HREF="http://www.mgri.org/home.html" 
		      TARGET="_blank">Merck Genome Research Institute</A> 
		  <LI><A HREF="http://www.affymetrix.com/" 
		      TARGET="_blank">Affymetrix, Inc.</A> 
		  <LI><A HREF="http://www.wellcome.ac.uk/" 
		      TARGET="_blank">The Wellcome Trust</A>
		  <LI><A HREF="http://www.nih.gov/" TARGET="_blank">The National
		      Institutes of Health</A>
	  	  </UL>
		<LI><B>Sequencing:</B> 
		  <UL>
		  <LI><A HREF="http://www.broadinstitute.org/"
		      TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		  <LI><A HREF="http://medicine.wustl.edu/" 
		      TARGET="_blank">Washington University St. Louis Genome
		      Sequencing Center</A>
		  <LI><A HREF="http://www.sanger.ac.uk"
			TARGET="_blank">The Wellcome Trust Sanger Institute</A>
		  </UL>
	    	</UL></P>
		<P>
		<HR>
		Later releases of the assembly and map have been produced 
		by the Mouse Genome Sequencing Consortium, with these acknowledgments: 
		<UL>  
		    <LI><B>Funding:</B>
		    <UL>
		    <LI><A HREF="http://www.nih.gov/" TARGET="_blank">National 
			Institutes of Health</A>
		    <LI><A HREF="http://www.wellcome.ac.uk/" 
			TARGET="_blank">The Wellcome Trust</A>
		    <LI><A HREF="http://www.gsk.com/" 
			TARGET="_blank">GlaxoSmithKline</A>
		    <LI><A HREF="http://www.merck.com/" TARGET="_blank">The 
			Merck Genome Research Institute</A>
		    <LI><A HREF="http://www.affymetrix.com/index.affx" 
			TARGET="_blank">Affymetrix</A>
		    </UL>
		   
	    	    <LI><B>Sequencing:</B>
		    <UL>
		    <LI><A HREF="http://www.broadinstitute.org/"
			TARGET="_blank">The Broad Institute's Arachne Group</A>
		    <LI><A HREF="http://www.sanger.ac.uk"
			TARGET="_blank">The Wellcome Trust Sanger Institute</A> 
		    <LI><A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Mus%20musculus&amp;GROUP=2"
		 	TARGET="_blank">Washington University in St. Louis 
			Genome Sequencing Center</A>
		    <LI><A HREF="https://www.hgsc.bcm.edu/content/mouse-genome-project"
			TARGET="_blank">Baylor College of Medicine Human Genome 
			Sequencing Center</A> 
		    <LI><A HREF="http://www.ucl.ac.uk/WIBR/" 
			TARGET="_blank">Wolfson Institute for Biomedical 
			Research</A>
		    <LI><A HREF="http://www.genome.ou.edu/"
			TARGET="_blank">University of Oklahoma, Advanced Center 
			for Genome Technology</A>
		    <LI><A HREF="http://mrcseq.har.mrc.ac.uk/"
			TARGET="_blank">UK MRC Mouse Genome Centre and Mammalian 
			Genetics Unit
		    <LI><A HREF="http://www.nisc.nih.gov/"
			TARGET="_blank">National Institutes of Health Intramural
			Sequencing Center</A>
		    <LI><A HREF="http://www.hpcgg.org/"
			TARGET="_blank">Harvard Partners Genome Center</A>
		    <LI><A HREF="http://www.cshl.edu/"
			TARGET="_blank">Cold Springs Harbor Laboratory</A>
		    </UL>
	     	    <LI><B>Physical Map:</B> 
		    <UL>
		    <LI>Mouse BAC libraries - <A HREF="http://bacpac.chori.org/" 
			TARGET="_blank">Children's Hospital Oakland Research
			Institute</A> (CHORI)
		    <LI>BAC fingerprint map - 
			<A HREF="http://www.bcgsc.ca/lab/mapping/mouse" 
			TARGET="_blank">Genome Sciences Centre</A>
		    <LI>BAC end sequences - 
			<A HREF="http://www.tigr.org/tdb/bac_ends/mouse/bac_end_intro.html"
			TARGET="_blank">The Institute of Genome Research</A> 
			(TIGR)
		    <LI>TPF production - Tina Graves, April Cook, Christine
			Burton, Darren Graham, Shannon Dugan, Donna Muzny
		    </UL>
		    <LI><B>Assembly:</B> Deanna Church, Richa Agarwala, Josh 
			Cherry, Mike DiCuccio, Hsiu-Chuan Chen, Paul Kitts, 
			Victor Sapojnikov
		    
		    <LI><B>cDNA Sequencing:</B> 
		    <UL>
		    <LI><A HREF="http://genome.gsc.riken.go.jp/"
		    	TARGET="_blank">Riken Institute</A>
		    <LI><A HREF="http://mgc.nci.nih.gov/"
		    	TARGET="_blank">The Mammalian Gene Collection</A>
		    </UL>
	  	    <LI><B>UCSC Mouse Genome Browser (mm10):</B> Hiram Clawson and
		    Luvina Guruvadoo
		    <LI><B>UCSC Mouse Genome Browser (mm9):</B> Hiram Clawson,
		    Archana Thakkapallayil, Robert Kuhn and Donna Karolchik
	  	    <LI><B>UCSC Mouse Genome Browser (mm8):</B> Hiram Clawson,
		    Fan Hsu, Kayla Smith, Ann Zweig, Robert Kuhn, Brooke
		    Rhead, Archana Thakkapallayil, and Donna Karolchik
	  	    <LI><B>UCSC Mouse Genome Browser (mm7):</B> Hiram Clawson,
		    Fan Hsu, Ann Zweig, Kayla Smith, Robert Kuhn
		    and Donna Karolchik
	  	    <LI><B>UCSC Mouse Genome Browser (mm6):</B> Hiram Clawson,
		    Fan Hsu, Jennifer Jackson, Robert Kuhn, Ali 
		    Sultan-Qurraie, Heather Trumbower and Donna Karolchik
	  	    <LI><B>Annotations:</B> 
		    <UL>
		    <LI><A HREF="../staff.html" TARGET="_blank">UCSC Genome
			Bioinformatics Group</A>
		    <LI><A HREF="http://www.jax.org/" TARGET="_blank">The 
			Jackson Laboratory</A>
		    <LI><A HREF="http://www.ncbi.nlm.nih.gov/" 
			TARGET="_blank">NCBI</A>
		    <LI><A HREF="http://www.ensembl.org/" 
			TARGET="_blank">Ensembl</A>
		    <LI><A HREF="http://bio.cse.psu.edu/" 
			TARGET="_blank">Pennsylvania State University</A>
		    <LI><A HREF="http://www.lbl.gov/" 
			TARGET="_blank">University of California 
			Berkeley/LBNL</A>
		    </UL>
		</UL></P>
		<P>
		<A NAME="mouse_use"></A>
		<HR>
		Mouse genome sequence data are released weekly into a
		public repository maintained by EBI and NCBI. References to 
		these data should cite the following publication: 
		Mouse Genome Sequencing Consortium. 
		<A HREF="http://www.nature.com/nature/mousegenome/" 
		TARGET="_blank">Initial sequencing and comparative analysis of 
		the mouse genome</A>.
		<I>Nature</I>, <B>420</B>, 520-562 (2002).</P>
		</TD>
		<TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Mouse Lemur Credits =================================================== -->
<A NAME="mouse_lemur_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Mouse Lemur Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Mouse Lemur Genome Browser is provided with the following acknowledgments:
		    <UL>
		    <LI><B>Sequencing/Assembly: </B>
	  	    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A> at MIT 
		    and Harvard, Cambridge, MA, USA 
		    <LI>
		    <B>UCSC Mouse Lemur Genome Browser (micMur1):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="mouse_lemur_use"></A>
		    The mouse lemur sequence has been made freely available as 
		    part of the Mammalian Genome Project (29 Mammals Project). 
		    The initial analysis of this dataset can be found in 
		    Lindblad-Toh <em>et al.</em>, 
		    <A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		    TARGET="_blank">A high-resolution map of human evolutionary constraint using 29 
		    mammals</A>, <em>Nature</em> 2011 Oct 12;478(7370):476-82. 
		    The mouse lemur data are provided with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (The Broad Institute) are properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Rat Credits ====================================================== -->
<A NAME="rat_credits"></A>
   <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Rat Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The UCSC Rat Genome Project is conducted in collaboration 
		    with the Rat Genome Sequencing Consortium.
		    <P>
		    The Rat Genome Browser is provided with the following acknowledgments:
		    <UL>
			<LI><B>Project Coordination:</B> 
			<A HREF="http://www.hgsc.bcm.edu/" 
			TARGET="_blank">Baylor College of Medicine Human Genome 
			Sequencing Center</A>, Richard Gibbs, Director
			<LI><B>Funding:</B> 
			  <UL>
			  <LI><A HREF="http://www.nhgri.nih.gov/" 
			      TARGET="_blank">National Human Genome Research 
			      Institute</A> (NIH)
			  <LI><A HREF="http://www.nhlbi.nih.gov/" 
			      TARGET="_blank">National Heart, Lung and Blood 
			      Institute</A> (NIH)
			  </UL>
	  		<LI><B>Sequencing:</B> 
			  <UL>
			  <LI>Baylor College of Medicine Human Genome 
			      Sequencing Center
			  <LI><A HREF="http://www.celera.com/" 
			      TARGET="_blank">Celera Genomics Group of 
			      Applera Corporation</A>
			  <LI><A HREF="http://www.genomecorp.com/" 
			      TARGET="_blank">Genome Therapeutics 
			      Corporation</A>
			  <LI><A HREF="http://www.utah.edu/" TARGET="_blank">The
			      University of Utah</A>
			  </UL>
	    		<LI><B>Assembly:</B> Baylor College of Medicine Human 
			Genome Sequencing Center ATLAS group -  Rui Chen, 
			James Durbin, Paul Havlak
	     		<LI><B>Fingerprint Mapping:</B> 
			<A HREF="http://www.bcgsc.bc.ca/" TARGET="_blank">The 
			British Columbia Cancer Agency Genome Sciences 
			Center</A> - Marco Marra, Jackie Schein
	     		<LI><B>BAC End Sequencing:</B> 
			<A HREF="http://www.tigr.org/tdb/bac_ends/rat/bac_end_intro.shtml" 
			TARGET="_blank">The Institute for Genomic Research</A> -
			Shaying Zhao
	     		<LI><B>BAC Library Construction and Distribution:</B>
			<A HREF="http://bacpac.chori.org/" 
			TARGET="_blank">Children's Hospital Oakland Research 
			Institute</A> (CHORI) - Pieter deJong
	     		<LI><B><A HREF="http://rgd.mcw.edu" TARGET="_blank">Rat 
			Genome Database</A>:</B> Medical College of Wisconsin
                        <LI><B>UCSC Rat Genome Browser (rn5)</B> - Hiram
                        Clawson, Brian Raney and Steve Heitner
	     		<LI><B>UCSC Rat Genome Browser (rn4)</B> - Angie 
			Hinrichs, Fan Hsu, Brooke Rhead, Archana Thakkapallayil,
		 	Kayla Smith, Ann Zweig, Robert Kuhn, and Donna Karolchik
	     		<LI><B>UCSC Rat Genome Browser (rn3)</B> - Hiram 
			Clawson, Heather Trumbower, Robert Kuhn, Yontao Lu, 
			Terry Furey, Mark Diekhans, Robert Baertsch, Donna 
			Karolchik, and Jim Kent
		    </UL></P>
		    <P>
		    <HR>
		    <A NAME="rat_use"></A>
		    The rat genome sequence is made freely available by the 
		    Rat Genome Project at the 
		    <A HREF="https://www.hgsc.bcm.edu/content/rat-genome-project" TARGET="_blank">
		    Baylor College of Medicine Human Genome Sequencing 
		    Center</A>. Please cite the following publications when
		    using these data:</P>
		    <P>
		    Havlak, P. <em>et al</em>. The Atlas genome assembly system.
		    <em>Genome Res.</em> <B>14</B>(4), 721-32 (2004). </P>
		    <P>
		    Rat Genome Sequencing Project Consortium. Genome sequence 
		    of the Brown Norway rat yields insights into mammalian 
		    evolution. <em>Nature</em> <B>428</B>(6982), 493-521 (2004).
		    </P>
		    <P>
		    See the Baylor 
		    <A HREF="https://www.hgsc.bcm.edu/content/conditions-use"
		    TARGET="_blank">data use limitations</A> web page for 
		    restrictions on the use of these data.</P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Opossum Credits=================================================== -->
<A NAME="opossum_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Opossum Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC Opossum Genome Browser displays data from the 
		    draft assemblies of the opossum (<em>Monodelphis
		    domestica</em>) produced by The Broad Institute. These 
		    assemblies are provided with the following acknowledgments:
		    <UL>
		    <LI><B>Sequencing/Assembly:</B>
		        <A HREF="http://www.broad.mit.edu/mammals/opossum/"
			TARGET="_blank">The Broad Institute</A>, 
			Cambridge, MA, USA 
		    <LI><B>FISH Mapping: </B>
		  	<A HREF="http://www.ncsu.edu/" TARGET="_blank">North 
		 	Carolina State University</A> (NCSU), Raleigh, NC, USA 
		    <LI><B>UCSC Opossum Genome Browser/Initial Annotations
			(monDom5):</B> 
			<A HREF="../staff.html" 
			TARGET="_blank">UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Andy Pohl, Hiram Clawson, Brian Raney, Mark Diekhans,
                        Brooke Rhead and Katrina Learned
		    <LI><B>UCSC Opossum Genome Browser/Initial Annotations
			(monDom4):</B> 
			UCSC Genome Bioinformatics Group,
			Santa Cruz, CA, USA - 
			Hiram Clawson, Archana Thakkapallayil, Ann Zweig, Kayla
		        Smith and Donna Karolchik
		    <LI><B>UCSC Opossum Genome Browser/Initial Annotations
			(monDom1):</B> 
			UCSC Genome Bioinformatics Group,
			Santa Cruz, CA, USA - 
			Hiram Clawson, Galt Barber, Ali Sultan-Qurraie, 
			Brian Raney and Donna Karolchik
		    </UL>
		    <P>
		    <HR>
		    <A NAME="opossum_use"></A>
		    The opossum sequence is made freely available before 
		    scientific
		    publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data (The Broad Institute)
			is properly acknowledged.
			<LI>
			The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data set, 
			including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Orangutan Credits=================================================== -->
<A NAME="orangutan_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Orangutan Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC Orangutan Genome Browser displays data from the 
		    draft assemblies of the Sumatran orangutan (<em>Pongo 
		    pygmaeus abelii</em>), which are provided with the 
		    following acknowledgments:
		    <UL>
		    <LI><B>Sequencing:</B>
		    <UL>
	  	    <LI>The <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Pongo%20abelii"
		    TARGET="_blank">Genome Sequencing Center at Washington 
		    University School of Medicine</A> (WUSTL), St. Louis, MO, 
		    USA 
		    <LI>
		    <A HREF="https://www.hgsc.bcm.edu/content/orangutan-genome-project" 
		    TARGET="_blank">Baylor College of Medicine Human Genome 
		    Sequencing Center</A>, Houston, TX, USA
		    </UL>
		    <LI><B>Sequence Assembly and Chromosomal Sequence/AGP
		    Construction:</B> WUSTL
		    <LI><B>DNA Source:</B>
		    	Dr. Greely Stones - <A HREF="http://www.gpz.org/"
		 	TARGET="_blank">Gladys Porter Zoo</A>, Brownsville, TX, 
			USA
		    <LI><B>BAC Library:</B>
		    Yuko Yoshonaga in Pieter de Jong's laboratory - 
		    <A HREF="http://bacpac.chori.org" TARGET="_blank">Children's 
		    Hospital Oakland Research Institute</A>, Oakland, CA, USA 
		    <LI><B>Fosmid Library:</B> WUSTL
		    <LI><B>Fingerprint Map:</B> WUSTL
		    <LI><B>Cytogenetic Mapping and Human/Orangutan Breakpoint 
		    Analyses:</B> Mariano Rocchi - 
		    <A HREF="http://www.biologia.uniba.it/DIGEMI/" 
		    TARGET="_blank">Department of Genetics and Microbiology</A>, 
		    University of Bari, Bari, Italy 
		    <LI><B>Fosmid End Placement against the Human Genome</B> 
		    (used for breakpoint/inversion analyses during AGP 
		    construction): Lin Chen, Evan Eichler - Department of 
		    Genome Sciences, University of Washington, Seattle, WA, USA
		    <LI><B>Funding: </B> National Human Genome Research 
		    Institute (NHGRI), National Institutes of Health (NIH)
		    <LI><B>UCSC Orangutan Genome Browser/Initial Annotations
			(ponAbe2):</B> 
			<A HREF="../staff.html" 
			TARGET="_blank">UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Hiram Clawson, Kayla Smith, Robert Kuhn, Ann Zweig and 
			Donna Karolchik
		    </UL>
		    <P>
		    <HR>
		    <A NAME="orangutan_use"></A>
		    The orangutan sequence is made freely available before 
		    scientific publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data (Genome Sequencing Center at WUSTL)
			is properly acknowledged.
			<LI>
			The centers producing the data reserve the right to 
			publish the initial large-scale analyses of the data set, 
			including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome. 
			<LI>
			This is in accordance with, and with the understandings
			in the Fort Lauderdale meeting discussing Community 
			Resource Projects and the resulting NHGRI policy 
			<A HREF="http://www.genome.gov/page.cfm?pageID=10506537"
			TARGET="_blank">statement</A>.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Gibbon Credits=================================================== -->
<A NAME="gibbon_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Gibbon Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC Gibbon Genome Browser displays data from the 
		    draft assemblies of the northern white-cheeked gibbon 
		    (<em>Nomascus leucogenys</em>) provided by the Gibbon Genome Sequencing
		    Consortium with the following acknowledgments:
		    <UL>
		    <LI><B>Funding: </B> <A HREF="http://www.genome.gov" 
		    TARGET=_blank>National Human Genome Research Institute</A> (NHGRI)
		    <LI><B>Sequencing/Assembly:</B>
		    <UL>
	  	    <LI>The <A HREF="http://genome.wustl.edu/genomes/view/nomascus_leucogenys"
		    TARGET="_blank">Genome Institute at Washington University School of Medicine</A>
		    (WUSTL), St. Louis, MO, USA 
		    <LI>
		    <A HREF="https://www.hgsc.bcm.edu/content/gibbon-genome-project"
		    TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center</A>
		    (BCM-HGSC), Houston, TX, USA
		    </UL>
		    <LI><B>UCSC Gibbon Genome Browser (nomLeu2):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI><B>UCSC Gibbon Genome Browser (nomLeu1):</B> 
		    Chin Li, Brooke Rhead, Steve Heitner, Lauren Scott and Donna Karolchik
		    <LI><B>Initial Annotations:</B> 
		    <A HREF="../staff.html" TARGET="_blank">UCSC Genome Bioinformatics Group</A>
		    </UL>
		    <P>
		    <HR>
		    <A NAME="gibbon_use"></A>
		    The gibbon sequence is made freely available before scientific publication. 
		    Please review the Genome Institute at WUSTL 
		    <A HREF="http://genome.wustl.edu/data/data_use_policy"
		    TARGET="_blank">data use policy</A> and the BCM-HGSC
		    <A HREF="https://www.hgsc.bcm.edu/content/conditions-use"
		    TARGET="_blank">conditions for use</A> guidelines before using these data.
		    </P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Gorilla Credits=================================================== -->
<A NAME="gorilla_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Gorilla Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC Gorilla Genome Browser displays the 
		    draft assembly of the gorilla genome
		    (<em>Gorilla gorilla gorilla</em>) provided with the 
		    following acknowledgments:
		    <UL>
		    <LI><B>Sequencing:</B>
	  	    The <A HREF="http://www.sanger.ac.uk/resources/downloads/gorilla/"
		    TARGET="_blank">Wellcome Trust Sanger Institute</A>, 
		    Hinxton, Cambridge, UK
		    <LI><B>Assembly:</B>
		    <UL>
	  	    <LI>The <A HREF="http://www.sanger.ac.uk/resources/downloads/gorilla/"
		    TARGET="_blank">Wellcome Trust Sanger Institute</A>, 
		    Hinxton, Cambridge, UK
	  	    <LI>The <A HREF="http://www.ebi.ac.uk/"
		    TARGET="_blank">European Bioinformatics Institute</A> (EBI), 
		    Hinxton, Cambridge, UK
		    </UL>
		    <LI><B>UCSC Gorilla Genome Browser/Initial Annotations
			(gorGor3):</B> 
			<A HREF="../staff.html" 
			TARGET="_blank">UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Hiram Clawson, Mark Diekhans, Brian Raney, Brooke Rhead,
		        Steve Heitner, Lauren Scott and Donna Karolchik
		    </UL>
		    <P>
		    <HR>
		    <A NAME="gorilla_use"></A>
		    The gorilla sequence is made freely available to the 
		    public by the Wellcome Trust Sanger Institute.  Please
		    review their <A HREF="http://www.sanger.ac.uk/datasharing/"
		    TARGET="_blank">data-sharing policy</A> before using these
		    data.
		    </P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Guinea Pig Credits ====================================================== -->
<A NAME="guinea_pig_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Guinea Pig Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC Guinea Pig Genome Browser displays data from the 
		    Feb. 2008 CavPor3 draft assembly of the guinea pig 
		    (<em>Cavia porcellus</em>) from the Broad Institute. 
		    This assembly is provided with the following 
		    acknowledgments:
		    <UL>
			<LI><B>Funding:</B> The 
			<A HREF="http://www.genome.gov/" 
			TARGET="_blank">National Human Genome Research Institute</A>
	  		<LI><B>Sequencing/Assembly:</B>
			<A HREF="http://www.broadinstitute.org/"
			TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
<!-- 
	  		<LI><B>BAC End Reads:</B>
			<A HREF="http://www.tiho-hannover.de/index_e.htm"
		        TARGET="_blank">University of Veterinary Medicine 
		      	Hanover</A>, Germany, and the 
			<A HREF="http://www.helmholtz-hzi.de/en/"
			TARGET="_blank">Helmholtz Centre for Infection 
			Research</A>, Braunschweig, Germany
-->
			<LI><B>UCSC Guinea Pig Genome Browser/Initial Annotations 
			(cavPor3):</B> 
			<A HREF="../staff.html">UCSC Genome Bioinformatics 
			Group</A>, Santa Cruz, CA, USA - 
			Tim Dreszer, Kate Rosenbloom, Hiram Clawson, Kayla 
			Smith, Robert Kuhn, and Donna Karolchik.
		    </UL>
		    <P>
		    <HR>
		    <A NAME="guinea_pig_use"></A>
		    The guinea pig sequence is made freely available before 
		    scientific publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data (The Broad Institute) is properly 
			acknowledged.
			<LI>
			The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data 
			set, including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Hedgehog Credits =================================================== -->
<A NAME="hedgehog_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Hedgehog Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Hedgehog Genome Browser is provided with the 
		    following acknowledgments:
		    <UL>
		    <LI><B>Sequencing/Assembly: </B>
	  	    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A> at MIT 
		    and Harvard, Cambridge, MA, USA 
		    <LI>
		    <B>UCSC Hedgehog Genome Browser (eriEur1):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="hedgehog_use"></A>
		    The hedgehog sequence has been made freely available as 
		    part of the Mammalian Genome Project (29 Mammals Project). 
		    The initial analysis of this dataset can be found in 
		    Lindblad-Toh <em>et al.</em>, 
		    <A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		    TARGET="_blank">A high-resolution map of human evolutionary constraint using 
		    29 mammals</A>, <em>Nature</em> 2011 Oct 12;478(7370):476-82. 
		    The hedgehog data are provided with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (The Broad Institute) are properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>


<!-- Horse Credits ====================================================== -->
<A NAME="horse_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Horse Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC Horse Genome Browser displays the draft assembly 
		    of the horse (<em>Equus caballus</em>) genome from The 
		    Broad Institute. This assembly is provided with the 
		    following acknowledgments:
		    <UL>
			<LI><B>Funding:</B> The 
			<A HREF="http://www.genome.gov/" 
			TARGET="_blank">National Human Genome Research Institute</A>
	  		<LI><B>Project Leadership:</B>
			<A HREF="http://www.broadinstitute.org/"
			TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
	  		<LI><B>Sequencing/Assembly:</B>
			<A HREF="http://www.broadinstitute.org/"
			TARGET="_blank">The Broad Institute</A>
	  		<LI><B>BAC End Reads:</B>
			<A HREF="http://www.tiho-hannover.de/index_e.htm"
		        TARGET="_blank">University of Veterinary Medicine 
		      	Hanover</A>, Germany, and the 
			<A HREF="http://www.helmholtz-hzi.de/en/"
			TARGET="_blank">Helmholtz Centre for Infection 
			Research</A>, Braunschweig, Germany
			<LI><B>UCSC Horse Genome Browser/Initial Annotations 
			(equCab2):</B> 
			<A HREF="../staff.html">UCSC Genome Bioinformatics 
			Group</A>, Santa Cruz, CA, USA - 
			Larry Meyer, Hiram Clawson, Pauline Fujita, Brooke 
		 	Rhead, and Donna Karolchik
			<LI><B>UCSC Horse Genome Browser/Initial Annotations 
			(equCab1):</B> 
			<A HREF="../staff.html">UCSC Genome Bioinformatics 
			Group</A>, Santa Cruz, CA, USA - 
			Fan Hsu, Brooke Rhead, Robert Kuhn, Hiram Clawson, 
			Angie Hinrichs, Kate Rosenbloom, and Donna Karolchik
		    </UL>
		    <P>
		    <HR>
		    <A NAME="horse_use"></A>
		    The horse sequence is made freely available before 
		    scientific publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data (The Broad Institute) is properly 
			acknowledged.
			<LI>
			The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data 
			set, including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Kangaroo Rat Credits ================================================= -->
<A NAME="kangaroo_rat_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Kangaroo Rat Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Kangaroo Rat Genome Browser is provided with the 
		    following acknowledgments:
		    <UL>
		    <LI><B>Sequencing: </B>
	  	    <A HREF="https://www.hgsc.bcm.edu/content/kangaroo-rat-genome-project"
		    TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center</A>, 
		    Houston, TX, USA 
		    <LI><B>Assembly: </B>
	  	    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		    <LI>
		    <B>UCSC Kangaroo Rat Genome Browser (dipOrd1):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="kangaroo_rat_use"></A>
		    The kangaroo rat sequence has been made freely available as 
		    part of the Mammalian Genome Project (29 Mammals Project). 
		    The initial analysis of this dataset can be found in 
		    Lindblad-Toh <em>et al.</em>, 
		    <A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		    TARGET="_blank">A high-resolution map of human evolutionary constraint using 
		    29 mammals</A>, <em>Nature</em> 2011 Oct 12;478(7370):476-82. 
		    The kangaroo rat data are provided with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (Baylor College of Medicine and The Broad Institute) are 
			properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>


<!-- Painted Turtle Credits====================================================== -->

<A NAME="painted_turtle_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
                <TR><TD><FONT SIZE="4"></A><B>&nbsp; Painted Turtle Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>
                    The Painted Turtle Genome Browser is provided with the following 
		    acknowledgments:
                    <UL>
                    <LI>
                    <B>Sequencing/Assembly: </B> International Painted Turtle Genome Sequencing Consortium
                    <LI>
                    <B>UCSC Painted Turtle Genome Browser (chrPic1):</B>
                    Hiram Clawson, Brian Raney and Steve Heitner
                    <LI>
                    <B>Initial Genome Browser Annotations:</B>
                    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>,
                    University of California, Santa Cruz, CA, USA
                    </UL>
                    <P>
                    <A NAME="painted_turtle_use"></A>
                    <HR>
                    The painted turtle genome sequence data are made available to the
                    public with the following understanding:
                    <OL>
                        <LI>
                        The data may be freely downloaded, used in analyses, and
                        repackaged in databases.
                        <LI>
                        Users are free to use the data in scientific papers
                        analyzing particular genes and regions if the provider
                        of these data (International Painted Turtle Genome
                        Sequencing Consortium) is properly acknowledged.
                        <LI>
                        The center producing the data reserves the right to
                        publish the initial large-scale analyses of the data set,
                        including large-scale identification of regions of
                        evolutionary conservation and large-scale genomic
                        assembly. Large-scale refers to regions with size on the
                        order of a chromosome (that is, 30 Mb or more).
                        <LI>
                        Any redistribution of the data should carry this notice.
                    </OL></P>
                    </P>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
   </TABLE>
    <BR>

<!-- Panda Credits ====================================================== -->
<A NAME="panda_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Panda Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
                    The panda assembly was provided by BGI-Shenzhen 
                    with the following acknowledgments:
                    <UL>
                    <LI>
                    <B>Sequencing/Assembly:</B>
                    <A HREF="http://www.genomics.cn/en/" 
                    TARGET="_blank">The Beijing Genomics Institute at Shenzhen</A>
                    <LI>
                    <B>UCSC Panda Genome Browser (ailMel1):</B>
                    Hiram Clawson, Ann Zweig, and Antonio Coelho.
                    <LI>
                    <B>Initial Genome Browser Annotations:</B> 
                    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
                    University of California, Santa Cruz, CA, USA 
                    </UL>
                    <P>
                    <HR>
		    <A NAME="panda_use"></A>
		    The panda sequence is made freely available 
		    with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data (BGI-Shenzhen)
			is properly acknowledged.
			<LI>
			The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data 
			set, including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>


<!-- Pig Credits ====================================================== -->
<A NAME="pig_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Pig Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The pig assembly was provided with the following acknowledgments:
                    <UL>
                    <LI>
                    <B>Sequencing/Assembly:</B>
		    <UL>
                    <LI><A HREF="http://piggenome.org/" 
                    TARGET="_blank">The Swine Genome Sequencing Consortium</A> 
		    <LI><A HREF="http://www.sanger.ac.uk/resources/downloads/othervertebrates/pig.html"
		    TARGET="_blank">Wellcome Trust Sanger Institute</A>
		    </UL>
                    <LI>
                    <B>UCSC Pig Genome Browser (susScr3):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
                    <LI>
                    <B>UCSC Pig Genome Browser (susScr2):</B>
                    Hiram Clawson, Brian Raney, Mark Diekhans, and 
		    Mary Goldman.
                    <LI>
                    <B>Initial Genome Browser Annotations:</B> 
                    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
                    University of California, Santa Cruz, CA, USA 
                    </UL>
                    <P>
                    <HR>
		    <A NAME="pig_use"></A>
		    The pig sequence is made freely available 
		    with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (The Swine Genome Sequencing Consortium and Wellcome
		   	Trust Sanger Institute) are properly acknowledged.
			<LI>
			The centers producing the data reserve the right to 
			publish the initial large-scale analyses of the data 
			set, including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Pika Credits ======================================================= -->
<A NAME="pika_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Pika Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Pika Genome Browser is provided with the following acknowledgments:
		    <UL>
		    <LI><B>Sequencing/Assembly: </B>
	  	    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		    <LI>
		    <B>UCSC Pika Genome Browser (ochPri2):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="pika_use"></A>
		    The pika sequence has been made freely available as 
		    part of the Mammalian Genome Project (29 Mammals Project). 
		    The initial analysis of this dataset can be found in 
		    Lindblad-Toh <em>et al.</em>, 
		    <A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		    TARGET=_blank>A high-resolution map of human evolutionary constraint using 29 
		    mammals</A>, <em>Nature</em> 2011 Oct 12;478(7370):476-82. 
		    The pika data are provided with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (The Broad Institute) are properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Rabbit Credits =================================================== -->
<A NAME="rabbit_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Rabbit Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The rabbit Genome Browser is provided with the 
		    following acknowledgments:
		    <UL>
		    <LI><B>Sequencing/Assembly: </B>
	  	    The <A HREF="http://www.broadinstitute.org/science/projects/mammals-models/rabbit/rabbit-genome-sequencing-project"
		    TARGET="_blank">Broad Institute</A>, Cambridge, MA, USA 
		    <LI>
		    <B>UCSC Rabbit Genome Browser (oryCun2):</B>
		    Hiram Clawson and Antonio Coelho
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="rabbit_use"></A>
		    The rabbit sequence is made freely available before 
		    scientific publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (the Broad Institute) are properly acknowledged.
			<LI>
			The centers producing the data reserve the right to 
			publish the initial large-scale analyses of the data 
			set, including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>


<!-- Platypus Credits =================================================== -->
<A NAME="platypus_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Platypus Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The platypus Genome Browser is provided with the 
		    following acknowledgments:
		    <UL>
		    <LI><B>Sequencing/Assembly: </B>
	  	    The <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Ornithorhynchus%20anatinus"
		    TARGET="_blank">Genome Sequencing Center</A> at Washington 
		    University School of Medicine, St. Louis, MO, USA (WUSTL)
		    <LI>
		    <B>UCSC Platypus Genome Browser (ornAna1):</B>
		    Angie Hinrichs, Kayla Smith, Robert Kuhn, Brian Raney, and 
		    Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="platypus_use"></A>
		    The platypus sequence is made freely available before 
		    scientific publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data are properly acknowledged. See 
		        the <a href="http://genome.wustl.edu/data.cgi">WUSTL website</A>
		   	for credit information.
			<LI>
			The centers producing the data reserve the right to 
			publish the initial large-scale analyses of the data 
			set, including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			This is in accordance with, and with the understandings 
			in the Fort Lauderdale meeting discussing
			<A HREF="ftLauderdale.pdf">Community Resource Projects</A> and the 
			resulting 
			<A HREF="http://www.genome.gov/page.cfm?pageID=10506537"
			TARGET="_blank">NHGRI policy statement</A>.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Chicken Credits ====================================================== -->
<A NAME="chicken_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Chicken Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The Chicken Genome Browser is provided with the following acknowledgments:
		    <UL>
			<LI><B>Sequencing: </B>
	  		The <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Gallus%20gallus&amp;GROUP=2"
			TARGET="_blank">Genome Sequencing Center</A> at Washington 
			University School of Medicine, St. Louis, MO, USA (WUSTL)
	    		<LI><B>Physical Map: </B>WUSTL
			<BR>
	     		<LI><B>Genetic Mapping/Linkage Analysis: </B>The Chicken 
			Mapping Consortium:
			<UL>
			<LI>
			Wageningen Map based on Wageningen broilerxbroiler 
			mapping population ("W" linkage maps; 460 F2 animals), 
			coordinated by Martien Groenen, 
			<A HREF="http://www.abg.wur.nl/UK/"
			TARGET="_blank">Wageningen University</A>, Wageningen,
			The Netherlands 
			<LI>
			East Lansing Map based on the RJFxlayer cross 
			("E" linkage maps; 56 BC animals), coordinated by Hans 
			Cheng,
			<A HREF="http://www.ars.usda.gov/main/site_main.htm?modecode=36-35-00-00"
			TARGET="_blank">USDA-ARS</A>, East Lansing, MI, USA
			<LI>
			Compton map based on layerxlayer cross ("C" linkage maps; 
			52 BC animals), coordinated by Nat Bumstead, 
			<A HREF="http://www.iah.bbsrc.ac.uk/"
			TARGET="_blank">Institute for Animal Health</A>
			<LI>
			Consensus linkage map: Martien Groenen and Richard
			Crooijmans, Wageningen University
			<LI>
			RH map: coordinated by Mireille Morisson and Alain 
			Vignal, 
			<A HREF="http://www.toulouse.inra.fr/lgc/"
			TARGET="_blank">French National Institute for Agricultural
			Research</A> (INRA), Toulouse 
			<LI>
			Linkage Mapping: White Leghorn X Red Junglefowl (800 F2 
			animals), coordinated by Susanne Kerje, Lina Jacobsson 
			and Leif Andersson, 
			<A HREF="http://www.uu.se/en/"
			TARGET="_blank">Uppsala University</A>, Uppsala, Sweden
			<LI>
        		Clone/Marker Pairs: coordinated by Jerry Dodgson, 
			Michigan State University, East Lansing, MI, USA
			<LI>
			Marker Name/Sequence Resolution: coordinated by Hans 
			Cheng, USDA-ARS, East Lansing, MI, USA
			<LI>
			Chicken FPC browser: 
			<A HREF="http://www.bioinformatics.nl/gbrowse/cgi-bin/gbrowse/ChickFPC/"
			TARGET="_blank">ChickFPC</A>, Martien Groenen and Richard 
			Crooijmans (galGal3), Jan Aerts (galGal2), 
			Wageningen University
			<LI>
			Additional Mapping Data Contributions: Winston Bellott, 
			David Page Lab, 
			<A HREF="http://www.whitehead.mit.edu/"
			TARGET="_blank">Whitehead Institute for Biomedical
			Research</A> (WIBR-MIT), Cambridge, MA, USA 
			</UL>
			<LI>
			<B>cDNA sequences:</B> <A HREF="http://ncbi.nlm.nih.gov/"
			TARGET="_blank">National Institutes of Health</A> and 
			The University of Manchester 
			<A HREF="http://chick.umist.ac.uk/" 
			TARGET="_blank">BBSRC ChickEST Database</A>
			<LI>
			<B>RJF finished clones:</B>
			<UL>
			<LI><A HREF="http://www.nisc.nih.gov/"
			TARGET="_blank">NIH Intramural Sequencing Center</A>
			(NISC) (AC091708.2, AC094012.2, AC138565.3)
			<LI>S. Kerje and L. Andersson,
			<A HREF="http://www.uu.se/en/"
			TARGET="_blank">Uppsala University</A> 
			(AY636124.1)
                        <LI>WUSTL (all others)
			</UL>
			<LI>
			<B>Assembly, Assembly/Map Integration, Golden Path 
			Creation:</B> WUSTL
		        <LI>
			<B>SNPs and cross-referenced gene annotations:</B> 
			<A HREF="http://www.genomics.org.cn/"
			TARGET="_blank">The Beijing Genomics Institute (BGI)</A>, 
			Beijing, China
			<LI>
                        <B>UCSC Chicken Genome Browser (galGal4):</B> Hiram
                        Clawson, Brian Raney and Steve Heitner
                        <LI>
			<B>UCSC Chicken Genome Browser (galGal3):</B> Angie 
			Hinrichs, Kayla Smith, Donna Karolchik
			<LI>
			<B>UCSC Chicken Genome Browser (galGal2):</B> Angie 
			Hinrichs, Heather Trumbower, Rachel Harte, Donna 
			Karolchik
			</UL>
		    <P>
		    <HR>
		    <A NAME="chicken_use"></A>
		    The <em>G. gallus</em> sequence is made freely available 
		    to the community by the Genome Sequencing Center,
		    Washington University School of Medicine, with the following
		    understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers analyzing
			particular genes and regions if the providers of this
			data (Genome Sequencing Center, Washington University
			School of Medicine) are properly cited:<BR>
			International Chicken Genome Sequencing Consortium. 
			Sequence and  comparative analysis of the chicken 
			genome provide unique  perspectives on vertebrate 
			evolution. <em>Nature</em>
			2004 Dec 9;432(7018): 695-716.  PMID: 15592404.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		    <P>The BGI gene annotations and
		    SNP data are made available with the following terms of use:
		    <OL>
		    <LI>
		    Users are free to use these data in scientific papers 
		    analyzing particular genes, if the Beijing Genomics 
		    Institute (BGI) is acknowledged.
		    <LI>
		    BGI and its collaborators reserve the right to publish the 
	  	    initial analyses of these data, including but not restricted 
		    to the large-scale identification of functional polymorphisms,
		    evolutionary patterns and signs of selection, correlations 
		    with known QTLs, utility of data for genetic mapping 
		    purposes, etc.
		    <LI>
		    Any redistribution of these data should carry this notice. 
		    </OL>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Coelacanth Credits ======================================================= -->
<A NAME="coelacanth_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Coelacanth Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Coelacanth Genome Browser is provided with the following acknowledgments:
		    <UL>
	   	    <LI><B>Funding:</B>
		    <A HREF="http://www.genome.gov" 
		    TARGET="_blank">National Human Genome Research Institute</A> (NHGRI)
		    <LI><B>Sequencing/Assembly: </B>
	  	    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/vertebrates-invertebrates/coelacanth/coelacanth"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		    <LI>
		    <B>UCSC Coelacanth Genome Browser (latCha1):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="coelacanth_use"></A>
		    The coelacanth sequence is made freely available to the 
		    community by The Broad Institute with the following understanding:
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing these data if the providers of these data are 
		    properly acknowledged.  
		    <LI>
		    The center producing the data (The Broad Institute) reserves the right to 
		    publish the initial large-scale analyses of the data set, including 
		    large-scale identification of regions of evolutionary 
		    conservation and large-scale genomic assembly. Large-scale 
		    refers to regions with size on the order of a chromosome.
		    <LI>
		    This is in accordance with, and with the understandings in 
		    the Fort Lauderdale meeting discussing Community Resource 
		    Projects and the resulting NHGRI 
		    <A HREF="http://www.genome.gov/page.cfm?pageID=10506537"
		    TARGET="_blank">policy statement</A>.
		    <LI>
		    Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Turkey Credits =================================================== -->
<A NAME="turkey_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Turkey Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The turkey Genome Browser was provided with the 
		    following acknowledgments:
		    <UL>
		    <LI><B>Sequencing/Assembly: </B>
	  	    The Turkey Genome Consortium</A> - 
		    <A HREF="http://www.cals.vt.edu/"
		    TARGET="_blank">Virginia Tech College of Agriculture and Life 
		    Sciences</A> and
		    <A HREF="http://www.vbi.vt.edu/"
		    TARGET="_blank">Virginia Bioinformatics Institute</A>, 
		    Blacksburg, VA, USA
		    <LI>
		    <B>UCSC Turkey Genome Browser (melGal1):</B>
		    Chin Li, Mark Diekhans, Brian Raney, Brooke Rhead, Greg Roe,
		    Steve Heitner, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="turkey_use"></A>
		    The turkey sequence is made freely available before 
		    scientific publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (the Turkey Genome Consortium) are 
			properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Wallaby Credits =================================================== -->
<A NAME="wallaby_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; Wallaby Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>
                    The wallaby Genome Browser was provided with the 
                    following acknowledgments:
                    <UL>
                    <LI><B>Sequencing/Assembly: </B>
                    The Tammar Wallaby Genome Sequencing Consortium</A> - Baylor College of Medicine 
                    <A HREF="https://www.hgsc.bcm.edu/content/wallaby-genome-project"
                    TARGET="_blank">Human Genome Sequencing Center</A> and the 
                    <A HREF="http://www.agrf.org.au/"
                    TARGET="_blank">Australian Genome Research Facility</A>
                    <LI>
                    <B>UCSC Wallaby Genome Browser (macEug2):</B>
                    Chin Li and Luvina Guruvadoo
                    <LI>
                    <B>Initial Genome Browser Annotations:</B> 
                    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
                    University of California, Santa Cruz, CA, USA 
                    </UL>
                    <P>
                    <HR>
                    <A NAME="wallaby_use"></A>
                    The wallaby sequence is made freely available before 
                    scientific publication with the following understanding:
                    <OL>
                        <LI>
                        The data may be freely downloaded, used in analyses, and
                        repackaged in databases.
                        <LI>
                        Users are free to use the data in scientific papers 
                        analyzing particular genes and regions if the providers
                        of these data (The Tammar Wallaby Genome Sequencing Consortium) are 
                        properly acknowledged.
                        <LI>
                        Any redistribution of the data should carry this notice.
                    </OL></P>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- White Rhinoceros Credits =================================================== -->
<A NAME="white_rhinoceros_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; White Rhinoceros Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The white rhinoceros Genome Browser is provided with the 
		    following acknowledgments:
		    <UL>
		    <LI><B>Sequencing/Assembly: </B>
	  	    The <A HREF="http://www.broadinstitute.org/science/projects/projects/"
		    TARGET="_blank">Broad Institute</A>, Cambridge, MA, USA 
		    <LI><B>Data Samples: </B>
		    <a href="http://biology.ucsd.edu/faculty/ryder.html" 
		    target ="_blank">Dr. Oliver Ryder</a>, Center for Reproduction of Endangered 
		    Species, San Diego Zoo, San Diego, CA
		    <LI>
		    <B>UCSC White Rhinoceros Genome Browser (cerSim1):</B>
		    Hiram Clawson and Steve Heitner
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="white_rhinoceros_use"></A>
		    The white rhinoceros sequence is made freely available before 
		    scientific publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (the Broad Institute) are properly acknowledged.
			<LI>
			The centers producing the data reserve the right to 
			publish the initial large-scale analyses of the data 
			set, including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- X. tropicalis Credits =============================================== -->
<A NAME="xentro_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>X. tropicalis</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The <em>X. tropicalis</em> draft assemblies 
		    were provided by the
		    <A HREF="http://genome.jgi-psf.org/Xentr4/Xentr4.home.html"
		    TARGET="_blank">U.S. DOE Joint Genome Institute</A> (JGI). 
		    These assemblies have the following acknowledgments:
		    <P>
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly:</B> 
		    JGI, Walnut Creek, CA, USA
		    <LI>
		    <B>UCSC <em>X. tropicalis</em> Genome Browser (xenTro3):</B>
		    Hiram Clawson, Brian Raney, Brooke Rhead, Greg Roe, Steve 
		    Heitner, and Donna Karolchik
		    <LI>
		    <B>UCSC <em>X. tropicalis</em> Genome Browser (xenTro2):</B>
		    Angie Hinrichs, Kayla Smith, Robert Kuhn, and Donna 
		    Karolchik
		    <LI>
		    <B>UCSC <em>X. tropicalis</em> Genome Browser (xenTro1):</B>
		    Fan Hsu,
		    Jim Kent, Heather Trumbower, Hiram Clawson, Brian Raney,
		    Galt Barber, Mark Diekhans, Angie Hinrichs, and Donna 
		    Karolchik 
		    <LI>
		    <B>Initial Annotations:</B> 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, Santa 
		    Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    These sequence data are made freely available by the 
		    JGI. Please refer to the JGI 
		    <A HREF="http://genome.jgi-psf.org/Xentr4/Xentr4.download.html" 
		    TARGET="_blank">data release policy</A> for data use
		    guidelines.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Zebra Finch Credits================================================= -->
<A NAME="zebra_finch_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Zebra Finch Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The zebra finch draft assembly was obtained from the 
		    Genome Sequencing Center at Washington University St. Louis 
		    (WUSTL) School of Medicine with the following 
		    acknowledgments:
		    <P>
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly:</B> 
		    WUSTL, St. Louis, MO, USA
		    <LI>
		    <B>Zebra finch DNA:</B> 
		    Arthur P. Arnold lab, Dept. of Physiological Science, 
		    UCLA, Los Angeles, CA, USA
<!-- Commented out until we verify this info
		    <LI>
		    <B>BAC library:</B> 
		    Barbara Blackmon, Clemson University Genomics Institute, 
		    Clemson, SC, USA 
-->
		    <LI>
		    <B>UCSC Zebra Finch Genome Browser (taeGut1):</B>
		    Brian Raney, Kayla Smith, Pauline Fujita, and Donna 
		    Karolchik
		    <LI>
		    <B>Initial Annotations:</B> 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, Santa 
		    Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="zebra_finch_use"></A>
		    The zebra finch sequence is made freely available to the 
		    community by the Genome Sequencing Center at the WUSTL 
		    School of Medicine with the following understanding:
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing these data if the providers of these data are 
		    properly acknowledged.  See
   		    <A HREF="http://genome.wustl.edu/data.cgi" 
		    TARGET="_blank">http://genome.wustl.edu/data.cgi</A> for 
		    credit information.
		    <LI>
		    The centers producing the data reserve the right to publish 
		    the initial large-scale analyses of the data set, including 
		    large-scale identification of regions of evolutionary 
		    conservation and large-scale genomic assembly. Large-scale 
		    refers to regions with size on the order of a chromosome.
		    <LI>
		    This is in accordance with, and with the understandings in 
		    the Fort Lauderdale meeting discussing Community Resource 
		    Projects and the resulting NHGRI 
		    <A HREF="http://www.genome.gov/page.cfm?pageID=10506537"
		    TARGET="_blank">policy statement</A>.
		    <LI>
		    Any redistribution of the data should carry this notice.
		    </OL>
		    Please refer to the WUSTL 
		    <A HREF="http://genome.wustl.edu/data.cgi"
		    TARGET="_blank">data use policy</A> for additional usage
		    guidelines and citation information.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>


<!-- Zebrafish Credits ====================================================== -->
<A NAME="zebrafish_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Zebrafish Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The zebrafish draft assemblies were provided 
		    by The Wellcome Trust Sanger Institute. The UCSC 
		    Zebrafish Genome Browser has the following 
		    acknowledgments:
		    <P>
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly:</B> 
		    <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/"
		    TARGET="_blank">The Wellcome Trust Sanger Institute</A>, UK. 
		    <LI>
		    <B>FPC map:</B> 
		    <UL>
		    <LI>
		    The Wellcome Trust Sanger Institute, UK
		    <LI>
		    <A HREF="http://www.niob.knaw.nl/" 
		    TARGET="_blank">Hubrecht Laboratory</A>, Netherlands Institute
		    for Developmental Biology, The Netherlands
		    <LI>
		    <A HREF="http://www.eb.tuebingen.mpg.de/" 
		    TARGET="_blank">Max-Planck Institute for Developmental 
		    Biology</A>, Germany
		    </UL>
		    <LI>
		    <B>BAC Clones, BAC End Pairs: </B> 
		    <UL>
		    <LI>
		    The Wellcome Trust Sanger Institute, UK - GenBank 
		    accessions for BAC clones
		    <LI>
		    Max-Planck Institute for Developmental 
		    Biology, Germany - BAC end pairs sequence data
		    <LI>
		    Hubrecht Laboratory</A>, Netherlands Institute
		    for Developmental Biology, The Netherlands - BAC end pairs 
		    sequence data
		    </UL>
		    <LI>
		    <B>Initial Annotations:</B> 
		    <UL>
		    <LI>
		    Yi Zhou, Anthony DiBiase, Leonard Zon and the Zebrafish
		    Genome Initiative at <A HREF="http://zon.tchlab.org/" 
		    TARGET="_blank">Boston Children's Hospital</A>, Boston, MA, US -
		    collaboration on Zebrafish WZ EST clusters, BAC End 
		    Singles/Pairs, Radiation Hybrid, and Human tBLASTn 
	  	    annotation tracks.	
		    <LI>
		    <A HREF="http://www.genetics.wustl.edu/fish_lab/frank/cgi-bin/fish/"
		    TARGET="_blank">Washington University in St. Louis</A>, St. 
		    Louis, MO, US - Zebrafish WZ EST clusters data
		    <LI>
		    <A HREF="http://www.ensembl.org" TARGET="_blank">Ensembl</A>,
		    UK - Ensembl gene predictions
		    <LI>
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, Santa 
		    Cruz, CA, USA - Other annotations
		    </UL>
		    <LI>
		    <B>UCSC Zebrafish Genome Browser (danRer7):</B> Hiram 
		    Clawson, Mary Goldman, Greg Roe, Brian Raney, Donna 
		    Karolchik 
		    <LI>
		    <B>UCSC Zebrafish Genome Browser (danRer6):</B> Galt   
		    Barber, Robert Kuhn, Katrina Learned, Donna Karolchik 
		    <LI>
		    <B>UCSC Zebrafish Genome Browser (danRer5):</B> Rachel 
		    Harte, Ann Zweig, Donna Karolchik 
		    <LI>
		    <B>UCSC Zebrafish Genome Browser (danRer4):</B> Rachel 
		    Harte, Archana Thakkapallayil, Robert Kuhn, Ann Zweig, 
	  	    Donna Karolchik 
		    <LI>
		    <B>UCSC Zebrafish Genome Browser (danRer3):</B> Rachel 
		    Harte, Jennifer Jackson, Ann Zweig, Ali Sultan-Qurraie, 
	  	    Donna Karolchik 
		    <LI>
		    <B>UCSC Zebrafish Genome Browser (danRer2):</B> Rachel 
		    Harte, Mark Diekhans, Heather Trumbower, Donna Karolchik, 
		    Jennifer Jackson 
		    </UL>
		    <P>
		    <HR>
		    These sequence data were produced by the Zebrafish 
		    Sequencing Group at the Sanger Institute and can be obtained 
		    directly from <A HREF="ftp://ftp.sanger.ac.uk/pub/zebrafish/" 
		    TARGET="_blank">ftp://ftp.sanger.ac.uk/pub/zebrafish/</A>.
		    Please adhere to the 
		    <A HREF="http://www.sanger.ac.uk/notices/use-policy.shtml" 
		    TARGET="_blank">data use guidelines</A> for these data.
		    <P>
		    For more information about the assemblies, see the 
		    Sanger Institute's  
		    <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/"
		    TARGET="_blank"><em>Danio rerio</em> 
		    Sequencing Project</A> web page.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Tenrec Credits====================================================== -->

<A NAME="tenrec_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
                <TR><TD><FONT SIZE="4"></A><B>&nbsp; Tenrec Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>
                    The tenrec genome was provided by
                    <a href="http://www.broadinstitute.org/" target="_blank">The Broad Institute</a>
                    with the following acknowledgments:
                    <UL>
                    <LI>
                    <B>Sequencing/Assembly: </B> 
                    <a href="http://www.broadinstitute.org/" 
		    target="_blank">The Broad Institute</a>, Cambridge, MA, USA
                    <LI>
                    <B>UCSC Tenrec Genome Browser (echTel1):</B>
                    Hiram Clawson, Brian Raney and Steve Heitner
                    <LI>
                    <B>Initial Genome Browser Annotations:</B>
                    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>,
                    University of California, Santa Cruz, CA, USA
                    </UL>
                    <P>
                    <A NAME="tenrec_use"></A>
                    <HR>
                    The tenrec genome sequence data are made available to the
                    public with the following understanding:
                    <OL>
                        <LI>
                        The data may be freely downloaded, used in analyses, and
                        repackaged in databases.
                        <LI>
                        Users are free to use the data in scientific papers
                        analyzing particular genes and regions if the provider
                        of these data (The Broad Institute) is properly acknowledged.
                        <LI>
                        The center producing the data reserves the right to
                        publish the initial large-scale analyses of the data set,
                        including large-scale identification of regions of
                        evolutionary conservation and large-scale genomic
                        assembly. Large-scale refers to regions with size on the
                        order of a chromosome (that is, 30 Mb or more).
                        <LI>
                        Any redistribution of the data should carry this notice.
                    </OL></P>
                    </P>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
   </TABLE>
    <BR>

<!-- Tetraodon Credits =================================================== -->
<A NAME="tetraodon_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Tetraodon Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The <em> Tetraodon nigroviridis</em> assemblies 
		    have been provided with the following acknowledgments:
		    <P>
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly:</B> 
		    <UL>
		    <LI>
		    <A HREF="http://www.genoscope.cns.fr/" 
		    TARGET="_blank">Genoscope</A>, Evry, France
		    <LI>
		    <A HREF="http://www.broad.mit.edu/annotation/tetraodon/background.html" 
		    TARGET="_blank">The Broad Institute</A>,MA, USA
		    </UL>
		    <LI>
		    <B>Project Funding:</B> 
		    <UL>
		    <LI>
		    <A HREF="http://www.genopole.org/html/en/connaitre/gip_cnrg.htm" 
		    TARGET="_blank">Consortium National de Recherche en 
		    Genomique</A>, France 
		    <LI>
		    <A HREF="http://www.genome.gov" TARGET="_blank">National Human 
		    Genome Research Institute</A> (NHGRI), USA
		    </UL>
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <UL>
		    <LI>
		    Genoscope - Genscan gene predictions, GAZE gene predictions
		    <LI>
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, Santa 
		    Cruz, CA, USA  - remaining annotations (based on data from 
		    Genoscope)
		    </UL>
		    <LI>
		    <B>UCSC Tetraodon Genome Browser (tetNig2):</B> Hiram 
		    Clawson, Mark Diekhans, Brian Raney, Ann Zweig, Mary
		    Goldman, Vanessa Swing, Antonio Coelho, Robert Kuhn, 
		    and Donna Karolchik 
		    <LI>
		    <B>UCSC Tetraodon Genome Browser (tetNig1):</B> Rachel Harte,
		    Robert Kuhn, Donna Karolchik 
		    </UL>
<A NAME="tetraodon_use"></A>
		    <P>
		    <HR>
		    These data have been freely provided by Genoscope before 
		    publication for use in the UCSC Genome Browser with the 
		    following understanding
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases. 
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions, provided that the 
		    Genoscope is properly acknowledged.
		    <LI>
		    Genoscope reserves the right to publish the initial 
		    large-scale analyses of the dataset, including large-scale 
		    identification of regions of evolutionary conservation and 
		    large-scale genomic assembly. Large-scale refers to regions
		    with size on the order of magnitude of a Tetraodon 
		    chromosome (that is, 5 Mb or more).
		    </OL>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Dolphin Credits ================================================= -->
<A NAME="dolphin_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Dolphin Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Dolphin Genome Browser is provided with the 
		    following acknowledgments:
		    <UL>
		    <LI><B>Sequencing: </B>
	  	    <A HREF="https://www.hgsc.bcm.edu/content/dolphin-genome-project"
		    TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center</A>, 
		    Houston, TX, USA 
		    <LI><B>Assembly: </B>
	  	    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		    <LI>
		    <B>UCSC Dolphin Genome Browser (turTru2):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="dolphin_use"></A>
		    The dolphin sequence has been made freely available as 
		    part of the Mammalian Genome Project (29 Mammals Project). 
		    The initial analysis of this dataset can be found in 
		    Lindblad-Toh <em>et al.</em>, 
		    <A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		    TARGET="_blank">A high-resolution map of human evolutionary constraint using 
		    29 mammals</A>, <em>Nature</em> 2011 Oct 12;478(7370):476-82. 
		    The dolphin data are provided with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (Baylor College of Medicine and The Broad Institute) are 
			properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Elephant Credits ====================================================== -->
<A NAME="elephant_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Elephant Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
                    The elephant assembly was provided by the Broad 
                    Institute with the following acknowledgments:
                    <UL>
                    <LI>
                    <B>Sequencing/Assembly:</B>
                    <A HREF="http://www.broadinstitute.org/"
                    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
                    <LI>
                    <B>UCSC Elephant Genome Browser (loxAfr3):</B>
                    Hiram Clawson, Pauline Fujita, Vanessa Swing, 
                    Antonio Coelho, and Donna Karolchik.
                    <LI>
                    <B>Initial Genome Browser Annotations:</B> 
                    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
                    University of California, Santa Cruz, CA, USA 
                    </UL>
                    <P>
                    <HR>
		    <A NAME="elephant_use"></A>
		    The elephant sequence is made freely available before 
		    scientific publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data (The Broad Institute)
			is properly acknowledged.
			<LI>
			The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data 
			set, including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Fugu Credits ====================================================== -->
<A NAME="fugu_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Fugu Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Fugu whole genome shotgun assembly was provided by the 
		    <A HREF="http://genome.jgi-psf.org/fugu6/fugu6.info.html" 
		    TARGET="_blank">U.S.  DOE Joint Genome Institute</A> (JGI) as 
		    part of the International Fugu Genome 
		    Consortium, led by the JGI and the 
		    <A HREF="http://www.fugu-sg.org"
		    TARGET="_blank">Singapore Institute of Molecular and Cell 
		    Biology</A> (IMCB). The assembly has the following 
		    acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly:</B>
		    Assembly was constructed with the 
		    JGI assembler, JAZZ, from paired end sequencing reads 
		    produced by JGI and the IMCB
		    at JGI, Myriad Genetics, and Celera Genomics
		    <LI>
		    <B>UCSC Fugu Genome Browser (fr3):</B>
		    Hiram Clawson and Greg Roe
		    <LI>
		    <B>UCSC Fugu Genome Browser (fr2):</B>
		    Cory McClean, Hiram Clawson, Ann Zweig, and Donna Karolchik
		    <LI>
		    <B>UCSC Fugu Genome Browser (fr1):</B>
		    Kate Rosenbloom, Heather Trumbower, Robert Kuhn, and Donna 
		    Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    <A NAME="fugu_use"></A>
		    </UL></P>
		    <P>
		    <HR>
		    The Fugu sequence data have been freely provided by the 
		    JGI for use in the UCSC Genome Browser. Please see the JGI 
		    <A HREF="http://genome.jgi-psf.org/Takru4/Takru4.download.html" 
	 	    TARGET="_blank">data release policy</A> for restrictions
		    regarding the use of these data.
		    </P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Lamprey Credits =================================================== -->
<A NAME="lamprey_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Lamprey Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    <P>The lamprey assembly was provided by the 
		    Genome Sequencing Center at
		    Washington University St. Louis (WUSTL) School of 
		    Medicine with the following acknowledgments:
		    <UL>
		    <LI>
	   	    <B>Funding:</B>
		    <A HREF="http://www.genome.gov" TARGET="_blank">National Human
		    Genome Research Institute</A> (NHGRI)
		    <LI>
		    <B>Sequencing/Assembly:</B>
		    <A HREF="http://genome.wustl.edu/genomes/detail/petromyzon-marinus/"
		    TARGET="_blank">WUSTL</A>, St. Louis, MO, USA; Jeramiah Smith
                    (<a href="https://bio.as.uky.edu/users/jjsmit3" target="_blank">University
                     of Kentucky</a>,
                     <a href="http://www.gs.washington.edu/" target="_blank">
                     University of Washington Department of Genome Sciences</a>, and
                     <a href="http://www.benaroyaresearch.org/" target="_blank">
                     Benaroya Research Institute at Virginia Mason</a>)
                    <LI>
                    <B>UCSC Lamprey Genome Browser (petMar2):</B>
                    Hiram Clawson, Brian Raney, and Steve Heitner
		    <LI>
		    <B>UCSC Lamprey Genome Browser (petMar1):</B>
		    Hiram Clawson, Ann Zweig, Pauline Fujita, and 
		    Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="lamprey_use"></A>
		    The lamprey sequence is made freely available before 
		    scientific publication. Please see the WUSTL
		    <A HREF="http://genome.wustl.edu/data.cgi" 
		    TARGET="_blank">data use policy</A> for usage restrictions
		    and citation information.</P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Lancelet Credits =================================================== -->
<A NAME="lancelet_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Lancelet Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    <P>The lancelet assembly was provided by the DOE
		    Joint Genome Institute (JGI) with the following 
		    acknowledgments:
		    <UL>
		    <LI>
		    <B>Funding:</B>
		    U.S. Department of Energy Office of Science, Biological 
		    and Environmental Research Program 
		    <LI>
		    <B>Sequencing/Assembly:</B>
		    <A HREF="http://genome.jgi-psf.org/Brafl1/Brafl1.home.html"
		    TARGET="_blank">JGI</A>
		    <LI>
		    <B>UCSC Lancelet Genome Browser (braFlo1):</B>
		    Hiram Clawson, Ann Zweig, Pauline Fujita, and 
		    Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="lancelet_use"></A>
		    The lancelet sequence is made freely available before 
		    scientific publication by the JGI. Please see the
	            <A HREF="http://genome.jgi-psf.org/Brafl1/Brafl1.download.html"
		    TARGET="_blank">JGI data release policy</A> for usage
		    restrictions and citation information. </P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Lizard Credits =================================================== -->
<A NAME="lizard_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Lizard Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    <P>The lizard assembly was provided by the Broad 
		    Institute with the following acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly:</B>
		    <A HREF="http://www.broadinstitute.org/"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA USA
		    <LI>
		    <B>UCSC Lizard Genome Browser (anoCar2):</B>
		    Hiram Clawson, Brian Raney, Luvina Guruvadoo, and 
		    Donna Karolchik
		    <LI>
		    <B>UCSC Lizard Genome Browser (anoCar1):</B>
		    Hiram Clawson, Archana Thakkapallayil, Robert Kuhn, and 
		    Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="lizard_use"></A>
		    The lizard sequence is made freely available before 
		    scientific publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data (The Broad Institute)
			is properly acknowledged.
			<LI>
			The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data 
			set, including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Rock Hyrax Credits ======================================================= -->
<A NAME="rock_hyrax_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Rock Hyrax Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Rock Hyrax Genome Browser is provided with the following acknowledgments:
		    <UL>
	   	    <LI><B>Funding:</B>
		    <A HREF="http://www.genome.gov" 
		    TARGET="_blank">National Human Genome Research Institute</A> (NHGRI)
		    <LI><B>Sequencing/Assembly: </B>
	  	    <UL>
		    <LI><A HREF="https://www.hgsc.bcm.edu/content/hyrax-genome-project"
		    TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center</A>
		    (BCM-HGSC), Houston, TX, USA 
		    <LI><A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		    </UL>
		    <LI>
		    <B>UCSC Rock Hyrax Genome Browser (proCap1):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="rock_hyrax_use"></A>
		    The rock hyrax sequence has been made freely available as 
		    part of the Mammalian Genome Project (29 Mammals Project). 
		    The initial analysis of this dataset can be found in 
		    Lindblad-Toh <em>et al.</em>, 
		    <A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		    TARGET=_blank>A high-resolution map of human evolutionary constraint using 29 
		    mammals</A>, <em>Nature</em> 2011 Oct 12;478(7370):476-82. 
		    The rock hyrax data are provided with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (The Broad Institute and BCM-HGSC) are properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>


<!--  Sheep Credits ======================================================== -->
<a name="sheep_credits"></a>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<tr><td>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<tr><td><font size="4"><b>&nbsp; Sheep Genome </b></font>
		</td></tr>
	    </table>
	    <table BGCOLOR="#fffee8" width="100%" cellpadding=0>
		<th height=10></th>
		<tr><td width=10></td><td>
		    <p>
		    <p>The sheep (<i>Ovis aries</i>) assembly was provided by the International Sheep Genomics 
		     Consortium (ISGC) sequencing center with the following acknowledgments:
		    <ul>
		    <li><b>Sequencing/Assembly:</b>
		    <a href="http://www.sheephapmap.org/" TARGET="_blank">
		  International Sheep Genomics Consortium (ISGC) sequencing center</a></li>
		  
		  <li><b>Mitochondrial Genome Sequencing:</b>
		    <a href="http://www.uni-giessen.de/cms/" TARGET="_blank">
		    Justus-Liebig-University</a></li>
		    
		   <li><b>UCSC Sheep Genome Browser (oviAri1):</b>
		       Chin Li, Greg Roe, Luvina Guruvadoo</li>
		   
		   <li><b>Initial Genome Browser Annotations:</b> 
		    <a href="../staff">UCSC Genome Bioinformatics Group</a>, 
		    University of California, Santa Cruz, CA, USA </li>
		    </ul>
		    
		    <p>
		    <hr>
		    <a name="sheep_use"></a>
		    The sheep sequence is made freely available before 
		    scientific publication with the following understanding:
		    <ol>
			<li>The data may be freely downloaded, used in analyses, and
			repackaged in databases.</li>
			
			<li>Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data (International Sheep Genomics Consortium)
			is properly acknowledged.</li>
			
			<li>The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data 
			set, including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). </li>
			
			<li>Any redistribution of the data should carry this notice.</li>
		    </ol>
		    </p>
		</td><td width=15></td></tr>
	    </table>
        </td></tr>
    </table>
    <br>

<!-- Shrew Credits ======================================================= -->
<A NAME="shrew_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Shrew Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Common Shrew Genome Browser is provided with the following acknowledgments:
		    <UL>
	   	    <LI><B>Funding:</B>
		    <A HREF="http://www.genome.gov" 
		    TARGET="_blank">National Human Genome Research Institute</A> (NHGRI)
		    <LI><B>Sequencing/Assembly: </B>
	  	    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		    <LI>
		    <B>UCSC Shrew Genome Browser (sorAra1):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="shrew_use"></A>
		    The shrew sequence has been made freely available as 
		    part of the Mammalian Genome Project (29 Mammals Project). 
		    The initial analysis of this dataset can be found in 
		    Lindblad-Toh <em>et al.</em>, 
		    <A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		    TARGET=_blank>A high-resolution map of human evolutionary constraint using 29 
		    mammals</A>, <em>Nature</em> 2011 Oct 12;478(7370):476-82. 
		    The shrew data are provided with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (The Broad Institute) are properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

    
<!-- Sloth Credits ======================================================= -->
<A NAME="sloth_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Sloth Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Sloth Genome Browser is provided with the following acknowledgments:
		    <UL>
		    <LI><B>Sequencing: </B>
	  	    <A HREF="http://genome.wustl.edu/genomes/view/choloepus_hoffmanni"
		    TARGET="_blank">The Genome Institute at Washington University School of
		    Medicine</A> (WUSTL), St. Louis, MO, USA 
		    <LI><B>Assembly: </B>
	  	    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		    <LI>
		    <B>UCSC Sloth Genome Browser (choHof1):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="sloth_use"></A>
		    The sloth sequence has been made freely available as 
		    part of the Mammalian Genome Project (29 Mammals Project). 
		    The initial analysis of this dataset can be found in 
		    Lindblad-Toh <em>et al.</em>, 
		    <A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		    TARGET=_blank>A high-resolution map of human evolutionary constraint using 29 
		    mammals</A>, <em>Nature</em> 2011 Oct 12;478(7370):476-82. 
		    The sloth data are provided with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (WUSTL and Broad Institute) are properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

    
<!-- Stickleback Credits =================================================== -->
<A NAME="stickleback_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Stickleback Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    <P>The stickleback assembly was provided by the Broad 
		    Institute with the following acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly:</B>
		    <A HREF="http://www.broadinstitute.org/"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA USA
		    <LI>
		    <B>UCSC Stickleback Genome Browser (gasAcu1):</B>
		    Angie Hinrichs, Hiram Clawson, Kayla Smith, and Donna 
		    Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="stickleback_use"></A>
		    The stickleback sequence is made freely available before 
		    scientific publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data (The Broad Institute)
			is properly acknowledged.
			<LI>
			The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data 
			set, including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Squirrel Credits ======================================================= -->
<A NAME="squirrel_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Squirrel Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Squirrel Genome Browser is provided with the following acknowledgments:
		    <UL>
		    <LI><B>Sequencing/Assembly: </B>
	  	    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		    <LI>
		    <B>UCSC Squirrel Genome Browser (speTri2):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="squirrel_use"></A>
		    The squirrel sequence has been made freely available as 
		    part of the Mammalian Genome Project (29 Mammals Project). 
		    The initial analysis of this dataset can be found in 
		    Lindblad-Toh <em>et al.</em>, 
		    <A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		    TARGET=_blank>A high-resolution map of human evolutionary constraint using 29 
		    mammals</A>, <em>Nature</em> 2011 Oct 12;478(7370):476-82. 
		    The squirrel data are provided with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (The Broad Institute) are properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Squirrel Monkey Credits ======================================================= -->
<A NAME="squirrel_monkey_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Squirrel Monkey Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Squirrel Monkey Genome Browser is provided with the following 
		    acknowledgments:
		    <UL>
		    <LI><B>Sequencing/Assembly: </B>
	  	    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		    <LI>
		    <B>UCSC Squirrel Monkey Genome Browser (saiBol1):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="squirrel_monkey_use"></A>
                    The squirrel monkey genome sequence data are made available to the
                    public with the following understanding:
                    <OL>
                        <LI>
                        The data may be freely downloaded, used in analyses, and
                        repackaged in databases.
                        <LI>
                        Users are free to use the data in scientific papers
                        analyzing particular genes and regions if the provider
                        of these data (The Broad Institute) is properly acknowledged.
                        <LI>
                        The center producing the data reserves the right to
                        publish the initial large-scale analyses of the data set,
                        including large-scale identification of regions of
                        evolutionary conservation and large-scale genomic
                        assembly. Large-scale refers to regions with size on the
                        order of a chromosome (that is, 30 Mb or more).
                        <LI>
                        Any redistribution of the data should carry this notice.
                    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Tarsier Credits ======================================================= -->
<A NAME="tarsier_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Tarsier Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Tarsier Genome Browser is provided with the following acknowledgments:
		    <UL>
	   	    <LI><B>Funding:</B>
		    <A HREF="http://www.genome.gov" 
		    TARGET="_blank">National Human Genome Research Institute</A> (NHGRI)
		    <LI><B>Sequencing/Assembly: </B>
	  	    <UL>
		    <LI><A HREF="https://www.hgsc.bcm.edu/"
		    TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center</A>
		    (BCM-HGSC), Houston, TX, USA 
		    <LI><A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		    </UL>
		    <LI>
		    <B>UCSC Tarsier Genome Browser (tarSyr1):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="tarsier_use"></A>
		    The tarsier sequence has been made freely available as 
		    part of the Mammalian Genome Project (29 Mammals Project). 
		    The initial analysis of this dataset can be found in 
		    Lindblad-Toh <em>et al.</em>, 
		    <A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		    TARGET=_blank>A high-resolution map of human evolutionary constraint using 29 
		    mammals</A>, <em>Nature</em> 2011 Oct 12;478(7370):476-82. 
		    The tarsier data are provided with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (BCM-HGSC and The Broad Institute) are properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>



<!-- Tasmanian Devil Credits =================================================== -->
<A NAME="tasmanian_devil_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; Tasmanian Devil Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>
                    The Tasmanian Devil assembly was provided with the following 
		    acknowledgments:
                    <UL>
                    <LI>
                    <B>Sequencing/Assembly:</B>
                    <A HREF="http://www.sanger.ac.uk/" 
                    TARGET="_blank">Wellcome Trust Sanger Institute</A>, Cambridge, UK
                    <LI>
                    <B>UCSC Tasmanian Devil Genome Browser (sarHar1):</B>
                    Chin Li and Luvina Guruvadoo
                    <LI>
                    <B>Initial Genome Browser Annotations:</B>
                    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>,
                    University of California, Santa Cruz, CA, USA
                    </UL>
                    <P>
                    <HR>
                    <A NAME="tasmanian_devil_use"></A>
                    The Tasmanian devil sequence is made freely available before
                    scientific publication with the following understanding:
                    <OL>
                        <LI>
                        The data may be freely downloaded, used in analyses, and
                        repackaged in databases.
                        <LI>
                        Users are free to use the data in scientific papers
                        analyzing particular genes and regions if the provider
                        of these data (the Wellcome Trust Sanger Institute)
                        is properly acknowledged.
                        <LI>
                        The center producing the data reserves the right to
                        publish the initial large-scale analyses of the data
                        set, including large-scale identification of regions of
                        evolutionary conservation and large-scale genomic
                        assembly. Large-scale refers to regions with size on the
                        order of a chromosome (that is, 30 Mb or more).
                        <LI>
                        Any redistribution of the data should carry this notice.
                    </OL></P>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Tree Shrew Credits ======================================================= -->
<A NAME="tree_shrew_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Tree Shrew Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Tree Shrew Genome Browser is provided with the following acknowledgments:
		    <UL>
	   	    <LI><B>Funding:</B>
		    <A HREF="http://www.genome.gov" 
		    TARGET="_blank">National Human Genome Research Institute</A> (NHGRI)
		    <LI><B>Sequencing/Assembly: </B>
	  	    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		    <LI>
		    <B>UCSC Tree Shrew Genome Browser (tupBel1):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="tree_shrew_use"></A>
		    The tree shrew sequence has been made freely available as 
		    part of the Mammalian Genome Project (29 Mammals Project). 
		    The initial analysis of this dataset can be found in 
		    Lindblad-Toh <em>et al.</em>, 
		    <A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		    TARGET=_blank>A high-resolution map of human evolutionary constraint using 29 
		    mammals</A>, <em>Nature</em> 2011 Oct 12;478(7370):476-82. 
		    The tree shrew data are provided with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (The Broad Institute) are properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>
<!-- Manatee Credits ======================================================= -->
<A NAME="manatee_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Manatee Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Manatee Genome Browser is provided with the following acknowledgments:
		    <UL>
		    <LI><B>Sequencing/Assembly: </B>
	  	    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		    <LI>
		    <B>UCSC Manatee Genome Browser (triMan1):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="manatee_use"></A>
                    The manatee genome sequence data are made available to the
                    public with the following understanding:
                    <OL>
                        <LI>
                        The data may be freely downloaded, used in analyses, and
                        repackaged in databases.
                        <LI>
                        Users are free to use the data in scientific papers
                        analyzing particular genes and regions if the provider
                        of these data (The Broad Institute) is properly acknowledged.
                        <LI>
                        The center producing the data reserves the right to
                        publish the initial large-scale analyses of the data set,
                        including large-scale identification of regions of
                        evolutionary conservation and large-scale genomic
                        assembly. Large-scale refers to regions with size on the
                        order of a chromosome (that is, 30 Mb or more).
                        <LI>
                        Any redistribution of the data should carry this notice.
                    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Marmoset Credits ====================================================== -->
<A NAME="marmoset_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Marmoset Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The Marmoset Genome Browser is provided with the following acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing and assembly:</B>
		    <A HREF="http://genome.wustl.edu/genomes/view/callithrix_jacchus"
		    TARGET="_blank">The Genome Institute at Washington University School of
		    Medicine</A>, St. Louis, MO, USA and
		    <a href="http://www.hgsc.bcm.edu/" 
		    TARGET="_blank">Human Genome Sequencing Center, Baylor College of Medicine</a>, 
		    Houston, TX, USA
		    <LI>
		    <B>DNA Source:</B>
		    Dr. Suzette Tardif, 
		    <A HREF="http://www.sfbr.org/pages/snprc_index.php"
		    TARGET="_blank">Southwestern National Primate Research Center</A>, 
		    San Antonio, TX, USA
		    <LI>
		    <B>CHORI-259 BAC library:</B>
		    Dr. Pieter DeJong, 
		    <A HREF="http://www.chori.org/" 
		    TARGET="_blank">Children's Hospital Oakland Research Institute</A>, 
		    Oakland, CA, USA
		    <LI>
		    <B>UCSC Marmoset Genome Browser (calJac3):</B>
		    Hiram Clawson, Chin Li, Mark Diekhans, Mary Goldman, Vanessa
		    Swing, and Donna Karolchik
		    <LI>
		    <B>UCSC Marmoset Genome Browser (calJac1):</B>
		    Hiram Clawson, Robert Kuhn, Pauline Fujita, Brooke Rhead, 
		    and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations</B>:
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA
		    </UL>
		    <P>
<A NAME="marmoset_use"></A>
		    <HR>
		    These data were produced by The Genome Center at Washington University
		    School of Medicine and The Human Genome Sequencing Center at the Baylor College 
		    of Medicine, and can be downloaded 
		    <A HREF="http://genome.wustl.edu/pub/organism/Primates/Callithrix_jacchus/assembly/"
		    TARGET="_blank">here</A>.
		    Please review The Genome Center's
		    <A HREF="http://genome.wustl.edu/data/data_use_policy"
		    TARGET="_blank">data use policy</A> before using these
		    data in publications.
		    If you have any questions regarding data use, 
		    please contact The Genome Center at
		    <A HREF="mailto:&#119;&#101;b&#109;a&#115;&#116;&#101;&#114;&#64;&#103;&#101;&#110;&#111;&#109;&#101;.&#119;&#117;&#115;&#116;l.&#101;&#100;u">&#119;&#101;b&#109;a&#115;&#116;&#101;&#114;&#64;&#103;&#101;&#110;&#111;&#109;&#101;.&#119;&#117;&#115;&#116;l.&#101;&#100;u</A>.
		   <!--  above address is webmaster at genome.wustl.edu -->
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>


<!-- Medaka Credits ====================================================== -->
<A NAME="medaka_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Medaka Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Medaka whole genome shotgun assembly was provided by
		    Japan's <A HREF="http://www.nig.ac.jp/english/index.html"
	    	    TARGET="_blank">National Institute of Genetics</A>
		    (NIG) and the <A HREF="http://medaka.utgenome.org/"
		    TARGET="_blank">University of Tokyo</A>. 
		    The assembly was constructed using the RAMEN assembler.</P>
		    <P>
		    NOTE: The UCSC oryLat1 assembly was replaced by oryLat2 in
		    Nov. 2008 to correct a UCSC assembly error with chrUn in 
		    which the gap relationships between contigs within their 
		    ultracontigs were broken. The oryLat2 assembly is based on
		    the same assembly as oryLat1: the v1.0
		    assembly from NIG and the University of Tokyo.
		    </P>
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly: </B> NIG and University of Tokyo,
		    Japan
		    <LI>
		    <B>UCSC Medaka Genome Browser (oryLat2):</B>
		    Hiram Clawson, Pauline Fujita, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
<A NAME="medaka_use"></A>
		    <HR>
		    The Medaka genome sequence data are made available to the
		    public with <A HREF="http://medaka.utgenome.org/#policy"
		    TARGET="_blank">certain restrictions</A>.</P>
		    </P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>
    
    
    
<!-- Medium Ground Finch Credits ====================================================== -->
<A NAME="medium_ground_finch_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Medium Ground Finch Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The initial medium ground finch genome assembly is the product of a 
		    collaboration between the <a href="http://www.genome10k.org/" 
		    target="_blank">Genome 10K Project</a> and <a href="http://www.genomics.cn/en/" 
		    target="_blank">Beijing Genomics Institute (BGI)</a>. The Medium Ground Finch
		    Genome Browser is provided with the following acknowledgments:</P>
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly: </B> Genome 10K Project and Beijing Genomics Institute (BGI)
		    <LI>
		    <B>UCSC Medium Ground Finch Genome Browser (geoFor1):</B>
		    Hiram Clawson and Greg Roe
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
<A NAME="medium_ground_finch_use"></A>
		    <HR>
		    The medium ground finch sequence is made freely available before scientific
		    publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (Genome 10K Project and Beijing Genomics Institute)
			are properly acknowledged.
			<LI>
			The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data set, 
			including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.</P>
		    </P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Megabat Credits ======================================================= -->
<A NAME="megabat_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Megabat Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Megabat (Fruitbat) Genome Browser is provided with the following 
		    acknowledgments:
		    <UL>
	   	    <LI><B>Funding:</B>
		    <A HREF="http://www.genome.gov" 
		    TARGET="_blank">National Human Genome Research Institute</A> (NHGRI)
		    <LI><B>Sequencing/Assembly: </B>
		    <UL>
		    <LI><A HREF="https://www.hgsc.bcm.edu/content/megabat-genome-project"
		    TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center</A> 
		    (BCM-HGSC), Houston, TX, USA 
	  	    <LI><A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		    </UL>
		    <LI>
		    <B>UCSC Megabat Genome Browser (pteVam1):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="megabat_use"></A>
		    The megabat sequence has been made freely available as 
		    part of the Mammalian Genome Project (29 Mammals Project). 
		    The initial analysis of this dataset can be found in 
		    Lindblad-Toh <em>et al.</em>, 
		    <A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		    TARGET=_blank>A high-resolution map of human evolutionary constraint using 29 
		    mammals</A>, <em>Nature</em> 2011 Oct 12;478(7370):476-82. 
		    The megabat data are provided with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the providers
			of these data (BCM-HGSC and The Broad Institute) are properly acknowledged.
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>


<!-- Microbat Credits ====================================================== -->
<A NAME="microbat_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Microbat Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The microbat (little brown bat) genome was provided by
		    the <A HREF="http://www.broadinstitute.org/"
		    TARGET="_blank">The Broad Institute</A> with the following 
		    acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly: </B> <A HREF="http://www.broadinstitute.org/"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		    <LI>
		    <B>UCSC Microbat Genome Browser (myoLuc2):</B>
		    Chin Li, Brian Raney, Robert Kuhn, Steve Heitner, Brooke 
		    Rhead, Greg Roe and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
<A NAME="microbat_use"></A>
		    <HR>
		    The microbat genome sequence data are made available to the
		    public with the following understanding: 
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data (The Broad Institute) is properly 
			acknowledged.
			<LI>
			The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data set, 
			including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		    </P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Naked Mole-rat Credits====================================================== -->

<A NAME="naked_mole-rat_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
                <TR><TD><FONT SIZE="4"></A><B>&nbsp; Naked Mole-rat Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>
                    The naked mole-rat genome was provided with the following acknowledgments:
                    <UL>
                    <LI>
                    <B>Sequencing/Assembly: </B> <A HREF="http://en.genomics.cn/"
		    target="_blank">Beijing Genomics Institute</A>, Shenzhen, China
                    <LI>
                    <B>UCSC Naked Mole-rat Genome Browser (hetGla2):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
                    <LI>
                    <B>UCSC Naked Mole-rat Genome Browser (hetGla1):</B>
                    Chin Li and Luvina Guruvadoo
                    <LI>
                    <B>Initial Genome Browser Annotations:</B>
                    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>,
                    University of California, Santa Cruz, CA, USA
                    </UL>
                    <P>
                    <A NAME="naked_mole-rat_use"></A>
                    <HR>
                    The naked mole-rat genome sequence data are made available to the
                    public with the following understanding:
                    <OL>
                        <LI>
                        The data may be freely downloaded, used in analyses, and
                        repackaged in databases.
                        <LI>
                        Users are free to use the data in scientific papers
                        analyzing particular genes and regions if the provider
                        of these data (Beijing Genomics Institute) is properly
                        acknowledged.
                        <LI>
                        The center producing the data reserves the right to
                        publish the initial large-scale analyses of the data set,
                        including large-scale identification of regions of
                        evolutionary conservation and large-scale genomic
                        assembly. Large-scale refers to regions with size on the
                        order of a chromosome (that is, 30 Mb or more).
                        <LI>
                        Any redistribution of the data should carry this notice.
                    </OL></P>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
   </TABLE>
    <BR>

<!-- Nile Tilapia Credits ======================================================= -->
<A NAME="nile_tilapia_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Nile Tilapia Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The Nile Tilapia Genome Browser is provided with the following acknowledgments:
		    <UL>
		    <LI><B>Sequencing/Assembly: </B>
	  	    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/vertebrates-invertebrates/tilapia/tilapia-genom"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		    <LI>
		    <B>UCSC Nile Tilapia Genome Browser (oreNil2):</B>
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <A NAME="nile_tilapia_use"></A>
                    <HR>
                    The Nile Tilapia genome sequence data are made available to the
                    public with the following understanding:
                    <OL>
                        <LI>
                        The data may be freely downloaded, used in analyses, and
                        repackaged in databases.
                        <LI>
                        Users are free to use the data in scientific papers
                        analyzing particular genes and regions if the provider
                        of these data (Broad Institute) is properly acknowledged.
                        <LI>
                        The center producing the data reserves the right to
                        publish the initial large-scale analyses of the data set,
                        including large-scale identification of regions of
                        evolutionary conservation and large-scale genomic
                        assembly. Large-scale refers to regions with size on the
                        order of a chromosome (that is, 30 Mb or more).
                        <LI>
                        Any redistribution of the data should carry this notice.
                    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Ciona Credits====================================================== -->

<A NAME="ciona_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>C. intestinalis</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The <em>C. intestinalis</em> v1.0 and v2.0 draft assemblies were provided 
		    by the 
		    <A HREF="http://genome.jgi-psf.org/ciona4/ciona4.home.html"
		    TARGET="_blank">U.S. DOE Joint Genome Institute</A> (JGI).  
		    These assemblies have the following acknowledgments, as 
		    listed on the JGI website:
		    <P>
		    <UL>
		    <LI>
		    <B>Project Funding</B>: Work performed under the auspices of 
		    the US Department of Energy's Office of Science, Biological 
		    and Environmental Research Program and the by the University 
		    of California, Lawrence Livermore National Laboratory under 
		    Contract No. W-7405-Eng-48, Lawrence Berkeley National 
		    Laboratory under contract No. DE-AC03-76SF00098 and Los Alamos
		    National Laboratory under contract No. W-7405-ENG-36.
		    <LI>
		    <B>Project Collaborators</B>:
		    <UL>
		    <LI>
		    JGI, 
		    Walnut Creek, CA, USA
		    <LI>
		    <A HREF="http://www.zool.kyoto-u.ac.jp/zool-e.html"
		    TARGET="_blank">Department of Zoology, Graduate School of 
		    Science, Kyoto University</A>, Kyoto, Japan
		    <LI>
		    <A HREF="http://mcb.berkeley.edu/" TARGET="_blank">Department 
		    of Molecular and Cellular Biology, University of 
		    California, Berkeley</A>, CA, USA
		    <LI>
		    <A HREF="http://www.nig.ac.jp/index-e.html" 
		    TARGET="_blank">National Institute of Genetics</A>, Mishima, 
		    Japan
		    <LI>
		    <A HREF="http://www.ornl.gov/" TARGET="_blank">Oak Ridge 
		    National Laboratory</A>, Oak Ridge, TN, USA
		    <LI>
		    <A HREF="http://www.lanl.gov/" TARGET="_blank">Los 
		    Alamos National Laboratory</A>, Los Alamos, NM, USA
		    </UL>
		    <LI>
		    <B>Annotations:</B> 
		    <UL>
		    <LI>
		    Colin Dewey, University of California, Berkeley, CA, USA -  
		    SNAP Gene Predictions, using the
		    SNAP parser written by Ian Korf
		    <LI>
		    JGI, Walnut Creek, CA, USA - JGI Gene Predictions
		    <LI>
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, Santa
		    Cruz, CA, USA - Other annotations
		    </UL>
                    <LI>
                    <B>UCSC <em>C. intestinalis</em> Genome Browser (ci2):</B> Brian
                    Raney, Mark Diekhans, Ann Zweig, Kayla Smith, Robert Kuhn,
                    Donna Karolchik
		    <LI>
		    <B>UCSC <em>C. intestinalis</em> Genome Browser (ci1):</B> Brian
		    Raney, Galt Barber, Heather Trumbower, Robert Kuhn, Donna 
		    Karolchik 
		    </UL>
		    <P>
		    <HR>
		    The sequence data have been freely provided by the JGI for 
		    use in the UCSC Genome Browser.  
		    The 
		    <A HREF="http://genome.jgi-psf.org/ciona4/ciona4.download.html" 
		    TARGET="_blank">JGI Data Release Policy</A> outlines 
		    restrictions on the use of these data.
		    For more information about the <em>C. intestinalis</em> 
		    assembly, see the 
		    <A HREF="http://genome.jgi-psf.org/ciona4/ciona4.info.html"
		    TARGET="_blank">JGI <em>Ciona</em> web page</A>.
		    </P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- S. purpuratus Credits============================================= -->
<A NAME="urchin_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; <em>S. purpuratus</em> Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The UCSC <em>S. purpuratus</em> Genome Browser displays data
		    obtained 
		    from the Baylor College of Medicine Human Genome Sequencing
		    Center (BCM HGSC). The Sep. 2006 draft assembly shows data
		    from Baylor version Spur 2.1; the Apr. 2005 assembly 
		    corresponds to Baylor version Spur_0.5. These data are 
		    provided with the following acknowledgments:
		    <UL>
	   	    <LI><B>Funding</B> - 
		    <A HREF="http://www.genome.gov" TARGET="_blank">National Human
		    Genome Research Institute</A> (NHGRI)
		    <LI><B>Sequencing/Assembly: </B>
		        <A HREF="http://www.hgsc.bcm.edu/" 
			TARGET="_blank">BCM HGSC</A>, Houston, TX, USA <BR>
		    <LI><B>UCSC <em>S. purpuratus</em> Genome Browser/Initial Annotations (strPur2): </B> 
			<A HREF="../staff.html" 
			TARGET="_blank">UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Kord Kober, Heather Trumbower, Rachel Harte, Kayla 
			Smith and Donna Karolchik. 
		     	
		    <LI><B>UCSC <em>S. purpuratus</em> Genome Browser/Initial Annotations (strPur1): </B> 
			<A HREF="../staff.html" 
			TARGET="_blank">UCSC Genome Bioinformatics Group</A>,
			Santa Cruz, CA, USA - 
			Andy Pohl, Kayla Smith and Donna Karolchik. 
		    </UL>
		    <P>
		    For more information on the <em>S. purpuratus</em> genome 
		    project, see the BCM HGSC 
		    <A HREF="https://www.hgsc.bcm.edu/content/sea-urchin-genome-project"
		    TARGET="_blank">Sea Urchin Genome Project</A> web page.
		    <P>
		    <HR>
		    <A NAME="urchin_use"></A>
		    To review the conditions for use regarding these 
		    data, see the BCM HGSC 
		    <A HREF="https://www.hgsc.bcm.edu/content/conditions-use"
		    TARGET="_blank">Conditions for Use</A> web page.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- D. melanogaster Credits=============================================== -->
<A NAME="dm_credits"></A>
<A NAME="dm_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. melanogaster</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The <em>D. melanogaster</em> assemblies displayed in the 
		    UCSC Genome Browser were obtained from the 
		    <A HREF="http://www.fruitfly.org/" TARGET="_blank">Berkeley 
		    Drosophila Genome Project</A> (BDGP) with the following 
		    acknowledgments:
		    <UL>
		    <LI><B>Sequencing and Assembly:</B>
		    <UL>
		    <LI>Euchromatic sequence - 
		    BDGP, Berkeley, CA, USA
		    <LI>Heterochromatic sequence - 
		    <A HREF="http://www.dhgp.org/" TARGET="_blank">Drosophila
		    Heterochromatin Genome Project</A>, Lawrence Berkeley
		    National Laboratory, Berkeley, CA, USA
		    </UL>
		    <LI><B>Annotations: </B>
		    <UL>
		    <LI>Euchromatic regions - 
		    <A HREF="http://flybase.net/annot/" 
		    TARGET="_blank">FlyBase</A> 
		    <LI>Heterochromatic regions - 
		    DHGP, Lawrence Berkeley
		    National Laboratory, Berkeley, CA, USA
		    </UL>
		    <LI><B>UCSC <em>D. melanogaster</em> Genome Browser 
		    and additional annotations (dm3): </B>
		    Angie Hinrichs, Archana Thakkapallayil, Kayla Smith, and
		    Donna Karolchik - 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, UCSC, Santa Cruz, CA, USA 
		    <LI><B>UCSC <em>D. melanogaster</em> Genome Browser 
		    and additional annotations (dm2): </B>
		    Angie Hinrichs, Brian Raney, Galt Barber,
		    and Donna Karolchik - 
		    UCSC Genome Bioinformatics Group, UCSC, Santa Cruz, CA, USA 
		    <LI><B>UCSC <em>D. melanogaster</em> Genome Browser 
		    and additional annotations (dm1): </B>
		    Angie Hinrichs, Brian Raney, Heather Trumbower, Ali 
		    Sultan-Qurraie, and Donna Karolchik - 
		    UCSC Genome Bioinformatics Group, UCSC, Santa Cruz, CA, USA 
		    </UL>
		    </P>
		    <P>
		    <HR>
		    For additional information about these data, including 
		    citation guidelines, see the 
		    <A HREF="http://www.fruitfly.org" 
		    TARGET="_blank">BDGP web site</A>.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- D. ananassae Credits================================================== -->
<A NAME="droAna_credits"></A>
<A NAME="droAna_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. ananassae</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The droAna2 data were provided by Agencourt Bioscience 
		    Corporation. The droAna1 data were produced by the Institute
		    for Genomic Research (TIGR). The <em>D. ananassae</em> 
		    Genome Browser has the following acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing:</B>
		    <A HREF="http://www.agencourt.com/company/experience/nhgri/"
		    TARGET="_blank">Agencourt Bioscience Corporation</A>,
		    Beverly, MA, USA 
		    <LI>
		    <B>Assembly:</B>
		    <UL>
		    <LI>
		    <A HREF="http://www.agencourt.com/company/experience/nhgri/"
		    TARGET="_blank">Agencourt Bioscience Corporation</A>,
		    Beverly, MA, USA, using the 
		    <A HREF="http://www.broad.mit.edu/wga/" 
		    TARGET="_blank">Arachne</A> assembler (droAna2)
		    <LI>
		    <A HREF="http://www.tigr.org/" TARGET="_blank">TIGR</A>,
		    Rockville, MD, USA, using an assembler produced by 
		    <A HREF="http://www.celera.com/" TARGET="_blank">Celera</A>
		    (droAna1)
		    </UL>
		    <LI>
		    <B>UCSC <em>D. ananassae</em> Genome Browser (droAna2):</B>
		    Angie Hinrichs, Brian Raney, Ann Zweig, Kayla Smith, and 
		    Donna Karolchik
		    <LI>
		    <B>UCSC <em>D. ananassae</em> Genome Browser (droAna1):</B>
		    Angie Hinrichs, Brian Raney, Heather Trumbower, Ali 
		    Sultan-Qurraie, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, 
		    Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    These data have been freely provided by Agencourt
		    Bioscience Corporation before 
		    publication for use in the UCSC Genome Browser with the 
		    following understanding:
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases. 
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions, provided that the 
		    Agencourt is properly acknowledged.
		    <LI>
		    Agencourt reserves the right to publish the initial 
		    large-scale analyses of the dataset, including large-scale 
		    identification of regions of evolutionary conservation and 
		    large-scale genomic assembly. 
		    <LI>
		    Any redistribution of the data should carry this notice.
		    </OL>
		    <P>
		    Please contact 
	 	    <A HREF="mailto:dsmith@agencourt.com">Douglas R. 
		    Smith</A> at Agencourt with questions concerning these 
		    data.
		    </OL>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- D. persimilis Credits================================================== -->
<A NAME="droPer_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. persimilis</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The droPer1 assembly was provided with the following acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly:</B>
		    <A HREF="http://www.broadinstitute.org/"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		    <LI>
		    <B>UCSC <em>D. persimilis</em> Genome Browser (droPer1):</B>
		    Angie Hinrichs, Kayla Smith, Robert Kuhn, Jennifer Jackson 
		    and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, 
		    Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="droPer_use"></A>
		    The <em>D. persimilis</em> sequence is made freely available
		    before scientific
		    publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data (The Broad Institute)
			is properly acknowledged.
			<LI>
			The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data set, 
			including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- D. erecta Credits================================================== -->
<A NAME="droEre_credits"></A>
<A NAME="droEre_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. erecta</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    The droEre1 data were provided by
		    Agencourt Bioscience Corporation. The 
		    <em>D.  erecta</em> Genome Browser has the following 
		    acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing and Assembly:</B>
		    <A HREF="http://www.agencourt.com/company/experience/nhgri/" 
		    TARGET="_blank">Agencourt Bioscience Corporation</A>,
		    Beverly, MA, USA, using the 
		    <A HREF="http://www.broad.mit.edu/wga/" 
		    TARGET="_blank">Arachne</A> assembler
		    <LI>
		    <B>UCSC <em>D. erecta</em> Genome Browser (droEre1):</B>
		    Angie Hinrichs, Brian Raney, Jennifer Jackson and Donna 
		    Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, 
		    Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    These data have been freely provided by Agencourt
		    Bioscience Corporation before 
		    publication for use in the UCSC Genome Browser with the 
		    following understanding:
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases. 
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions, provided that the 
		    Agencourt is properly acknowledged.
		    <LI>
		    Agencourt reserves the right to publish the initial 
		    large-scale analyses of the dataset, including large-scale 
		    identification of regions of evolutionary conservation and 
		    large-scale genomic assembly. 
		    <LI>
		    Any redistribution of the data should carry this notice.
		    </OL>
		    <P>
		    Please contact 
	 	    <A HREF="mailto:dsmith@agencourt.com"</A>Douglas R. 
		    Smith</A> at Agencourt with questions concerning these 
		    data.
		    </OL>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- D. simulans Credits================================================== -->
<A NAME="droSim_credits"></A>
<A NAME="droSim_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. simulans</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>These data were provided with the following acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing and assembly:</B>
		    <A HREF="http://genome.wustl.edu/genomes/view/drosophila_simulans_white_501"
		    TARGET="_blank">The Genome Institute at Washington University School of
		    Medicine</A>, St. Louis, MO, USA 
		    <LI>
		    <B>UCSC <em>D. simulans</em> Genome Browser</B>:
		    Angie Hinrichs, Jennifer Jackson, Ali Sultan-Qurraie,
		    Brian Raney and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations</B>:
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA
		    </UL>
		    <P>
		    <HR>
		    The <em>D. simulans</em> sequence is made freely available 
		    by The Genome Institute at Washington University
		    School of Medicine under the restrictions of their 
		    <A HREF="http://genome.wustl.edu/data/data_use_policy"
		    target="_blank">data use policy</A>. 
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- D. sechellia Credits================================================== -->
<A NAME="droSec_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. sechellia</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The droSec1 assembly was provided with the following acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly:</B>
		    <A HREF="http://www.broadinstitute.org/"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA USA
		    <LI>
		    <B>UCSC <em>D. sechellia</em> Genome Browser (droSec1):</B>
		    Angie Hinrichs, Kayla Smith and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="droSec_use"></A>
		    The <em>D. sechellia</em> sequence is made freely available
		    before scientific
		    publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data (The Broad Institute)
			is properly acknowledged.
			<LI>
			The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data set, 
			including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- D. yakuba Credits================================================== -->
<A NAME="droYak_credits"></A>
<A NAME="droYak_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. yakuba</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>These data were provided by the Genome Sequencing Center at
		    Washington University (WUSTL) School of Medicine, St. 
		    Louis</A> with the following acknowledgments:
		    <UL>
		    <LI>
		    <B>Funding:</B>
		    <A HREF="http://www.genome.gov" TARGET="_blank">National Human 
		    Genome Research Institute</A> (NHGRI)
		    <LI>
		    <B>Sequencing, assembly and AGP creation:</B>
		    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Drosophila%20yakuba&GROUP=6"
		    TARGET="_blank">Genome Sequencing Center, WUSTL School of 
		    Medicine</A> 
		    <LI>
		    <B>Initial Sequencing Proposal, Project Collaboration:</B>
		    David J. Begun and Charles H. Langley, 
		    <A HREF="http://www-eve.ucdavis.edu/" 
		    TARGET="_blank">University of California, Davis</A>, CA, USA
		    <LI>
		    <B>UCSC <em>D. yakuba</em> Genome Browser (droYak2)</B>:
		    Angie Hinrichs, Jennifer Jackson and Donna Karolchik
		    <LI>
		    <B>UCSC <em>D. yakuba</em> Genome Browser (droYak1)</B>:
		    Angie Hinrichs, Michael Chalup and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations</B>:
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California,
		    Santa Cruz, CA, USA
		    </UL>
		    <P>
		    <HR>
		    The <em>D. yakuba</em> sequence is made freely available 
		    before scientific publication by the Genome Sequencing Center,
		    WUSTL with the following understanding: 
		    <P>
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases. 
		    <LI>
		    Users are free to use the data in scientific papers analyzing 
		    particular genes and regions if the providers of these data 
		    (Genome Sequencing Center, WUSTL School of 
		    Medicine) are properly acknowledged. 
		    <LI>
		    The <em>Drosophila yakuba</em> analysis group is aiming to 
		    publish an initial analysis of the <em>D. yakuba</em> genome 
		    sequence in 2005 (submitted in early 2005) that will include 
		    descriptions of the assembly, genome landscape, comparative 
		    analysis and initial gene content. People who would like to 
		    coordinate other genome-wide analysis with this work should 
		    contact <A HREF="mailto:rwilson@watson.wustl.edu">Richard K. 
		    Wilson</A>, Genome Sequencing Center Director, Washington 
		    University School of Medicine. We welcome a coordinated 
		    approach to describing this community resource. 
		    <LI>
		    Any redistribution of the data should carry this notice. 
		    </OL>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- D. virilis Credits================================================== -->
<A NAME="droVir_credits"></A>
<A NAME="droVir_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. virilis</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>These data were produced by Agencourt Bioscience 
		    Corporation. The <em>D. virilis</em> Genome Browser
		    has the following acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing and Assembly:</B>
		    <A HREF="http://www.agencourt.com/company/experience/nhgri/" 
		    TARGET="_blank">Agencourt Bioscience Corporation</A>,
		    Beverly, MA, USA, using the 
		    <A HREF="http://www.broad.mit.edu/wga/" 
		    TARGET="_blank">Arachne</A> assembler 
		    <LI>
		    <B>Funding:</B>
		    <A HREF="http://www.genome.gov" TARGET="_blank">National Human 
		    Genome Research Institute</A> (NHGRI), USA
		    <LI>
		    <B>UCSC <em>D. virilis</em> Genome Browser (droVir2):</B>
		    Angie Hinrichs, Brian Raney, Jennifer Jackson and 
		    Donna Karolchik
		    <LI>
		    <B>UCSC <em>D. virilis</em> Genome Browser (droVir1):</B>
		    Angie Hinrichs, Brian Raney, Heather Trumbower, 
		    Robert Kuhn, Jennifer Jackson, and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, 
		    Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    These data have been freely provided by Agencourt
		    Bioscience Corporation before 
		    publication for use in the UCSC Genome Browser with the 
		    following understanding:
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases. 
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions, provided that the 
		    Agencourt is properly acknowledged.
		    <LI>
		    Agencourt reserves the right to publish the initial 
		    large-scale analyses of the dataset, including large-scale 
		    identification of regions of evolutionary conservation and 
		    large-scale genomic assembly. 
		    <LI>
		    Any redistribution of the data should carry this notice.
		    </OL>
		    <P>
		    Please contact 
	 	    <A HREF="mailto:dsmith@agencourt.com"</A>Douglas R. 
		    Smith</A> at Agencourt with questions concerning these 
		    data.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>



<!-- D. mojavensis Credits================================================== -->
<A NAME="droMoj_credits"></A>
<A NAME="droMoj_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. mojavensis</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>These data were produced by Agencourt Bioscience 
		    Corporation. The <em>D. mojavensis</em> Genome Browser
		    has the following acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing and Assembly:</B>
		    <A HREF="http://www.agencourt.com/company/experience/nhgri/" 
		    TARGET="_blank">Agencourt Bioscience Corporation</A>,
		    Beverly, MA, USA, using the 
		    <A HREF="http://www.broad.mit.edu/wga/" 
		    TARGET="_blank">Arachne</A> assembler 
		    <LI>
		    <B>Funding:</B>
		    <A HREF="http://www.genome.gov" TARGET="_blank">National Human 
		    Genome Research Institute</A> (NHGRI), USA
		    <LI>
		    <B>UCSC <em>D. mojavensis</em> Genome Browser (droMoj2):</B>
		    Angie Hinrichs, Brian Raney, Jennifer Jackson and Donna 
		    Karolchik
		    <LI>
		    <B>UCSC <em>D. mojavensis</em> Genome Browser (droMoj1):</B>
		    Angie Hinrichs, Brian Raney, Galt Barber and Donna 
		    Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, 
		    Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    These data have been freely provided by Agencourt 
		    Bioscience Corporation before 
		    publication for use in the UCSC Genome Browser with the 
		    following understanding:
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases. 
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions, provided that the 
		    Agencourt is properly acknowledged.
		    <LI>
		    Agencourt reserves the right to publish the initial 
		    large-scale analyses of the dataset, including large-scale 
		    identification of regions of evolutionary conservation and 
		    large-scale genomic assembly. 
		    <LI>
		    Any redistribution of the data should carry this notice.
		    </OL>
		    <P>
		    Please contact 
	 	    <A HREF="mailto:dsmith@agencourt.com"</A>Douglas R. 
		    Smith</A> at Agencourt with questions concerning these 
		    data.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>


<!-- D. pseudoobscura Credits================================================== -->
<A NAME="dp_credits"></A>
<A NAME="dp_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. pseudoobscura</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>These data were produced by the Human Genome Sequencing
		    Center (HGSC) at Baylor College of Medicine. The <em>D.
		    pseudoobscura</em> Genome Browser has the following 
		    acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing and Assembly:</B>
		    <A HREF="https://www.hgsc.bcm.edu/content/drosophila-genome-project"
		    TARGET="_blank">Baylor College of Medicine HGSC</A>
		    <LI>
		    <B>UCSC <em>D. pseudoobscura</em> Genome Browser (dp3):</B>
		    Angie Hinrichs, Brian Raney, Jennifer Jackson and Donna 
		    Karolchik
		    <LI>
		    <B>UCSC <em>D. pseudoobscura</em> Genome Browser (dp2):</B>
		    Angie Hinrichs, Heather Trumbower, Robert Kuhn, Donna
		    Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, 
		    Santa Cruz, CA, USA 
		    </UL>
		    <P>The <em>D. pseudoobscura</em> data are made available with 
		    specific <A HREF="https://www.hgsc.bcm.edu/content/conditions-use"
		    TARGET="_blank">conditions for use</A>.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- D. grimshawi Credits================================================== -->
<A NAME="droGri_credits"></A>
<A NAME="droGri_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. grimshawi</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    The droGri data were provided by
		    Agencourt Bioscience Corporation. The 
		    <em>D.  grimshawi</em> Genome Browser has the following 
		    acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing and Assembly:</B>
		    <A HREF="http://www.agencourt.com/company/experience/nhgri/" 
		    TARGET="_blank">Agencourt Bioscience Corporation</A>,
		    Beverly, MA, USA, using the 
		    <A HREF="http://www.broad.mit.edu/wga/" 
		    TARGET="_blank">Arachne</A> assembler
		    <LI>
		    <B>UCSC <em>D. grimshawi</em> Genome Browser (droGri1):</B>
		    Angie Hinrichs, Brian Raney, Jennifer Jackson, Kayla Smith
		    and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California, 
		    Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    These data have been freely provided by Agencourt
		    Bioscience Corporation before 
		    publication for use in the UCSC Genome Browser with the 
		    following understanding:
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases. 
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions, provided that the 
		    Agencourt is properly acknowledged.
		    <LI>
		    Agencourt reserves the right to publish the initial 
		    large-scale analyses of the dataset, including large-scale 
		    identification of regions of evolutionary conservation and 
		    large-scale genomic assembly. 
		    <LI>
		    Any redistribution of the data should carry this notice.
		    </OL>
		    <P>
		    Please contact 
	 	    <A HREF="mailto:dsmith@agencourt.com"</A>Douglas R. 
		    Smith</A> at Agencourt with questions concerning these 
		    data.
		    </OL>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR> 
<!-- A. mellifera Credits=============================================== -->
<A NAME="apiMel_credits"></A>
<A NAME="apiMel_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>A. mellifera</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The Amel_1.2 data were provided by The Baylor College of
		    Medicine Human Genome Sequencing Center (BCM HGSC)
		    with the following acknowledgments:
		    <UL>
		    <LI>
		    <B>Funding: </B>
		    <A HREF="http://www.genome.gov" TARGET="_blank">National Human 
		    Genome Research Institute</A> (NHGRI) and the 
		    <A HREF="http://www.usda.gov/wps/portal/usdahome"
		    TARGET="_blank">U.S. Department of Agriculture</A> (USDA)
		    <LI>
		    <B>Sequencing and assembly:</B>
		    <A HREF="https://www.hgsc.bcm.edu/content/honeybee-genome-project"
		    TARGET="_blank">BCM HGSC</A>, Houston, TX, USA
		    <LI>
		    <B>BAC library:</B>
		    Pieter de Jong and Katzutoyo Osoegawa at 
		    <A HREF="http://bacpac.chori.org/"
		    TARGET="_blank">Children's Hospital Oakland Research 
		    Institute</A> (CHORI), Oakland, CA, USA
		    <LI>
		    <B>UCSC <em>A. mellifera</em> Genome Browser (apiMel2):</B>
		    Andy Pohl, Angie Hinrichs, Jennifer Jackson, and Donna 
		    Karolchik
		    <LI>
		    <B>UCSC <em>A. mellifera</em> Genome Browser (apiMel1):</B>
		    Angie Hinrichs, Brian Raney, Robert Kuhn, and Donna 
		    Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations</B>:
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California,
		    Santa Cruz, CA, USA
		    </UL>
		    <P>
		    <HR>
		    The <em>A. mellifera</em> sequence is made freely 
		    available before scientific publication by the BCM HGSC.
		    See the Baylor 
		    <A HREF="https://www.hgsc.bcm.edu/content/conditions-use"
		    TARGET="_blank">conditions of use</A> statement for 
		    guidelines regarding the use of these data.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- A. gambiae Credits================================================== -->
<A NAME="anoGam_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>A. gambiae</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The MOZ2 <em>A. gambiae</em> data were provided by The 
		    International Anopheles Genome Project and 
		    downloaded from Ensembl. The data have the following 
		    acknowledgments:
		    <UL>
		    <LI>
		    <B>Funding:</B>
 		    <A HREF="http://www.niaid.nih.gov/Pages/default.aspx"
		    TARGET="_blank">National Institute of Allergy and Infectious 
		    Diseases</A> (NIAID), <A HREF="http://www.recherche.gouv.fr/"
		    TARGET="_blank">French Ministry of Research</A>, and the 
		    <A HREF="http://www.who.int/en/" TARGET="_blank">World Health 
		    Organization</A> (WHO). Celera Genomics and Genoscope
		    were funded respectively by <A HREF="http://www.nih.gov/" 
		    TARGET="_blank">National Institutes of Health</A> (NIH) and 
		    French Ministry of Research.
		    <LI>
		    <B>Project collaboration:</B>
		    The International Anopheles Genome Project - 
		    a collaboration between 
		    <A HREF="http://www.celera.com/" TARGET="_blank">Celera 
		    Genomics</A>, 
		    <A HREF="http://www.genoscope.cns.fr/" 
		    TARGET="_blank">Genoscope</A>, 
		    <A HREF="http://www.nd.edu/" TARGET="_blank">University of 
		    Notre Dame</A>, 
		    <A HREF="http://www.ebi.ac.uk/" 
		    TARGET="_blank">EBI</A>/<A HREF="http://www.sanger.ac.uk/" 
		    TARGET="_blank">Sanger Institute</A>, 
		    <A HREF=" http://www.ebi.ac.uk/embl/" TARGET="_blank">EMBL</A>, 
		    <A HREF="http://www.pasteur.fr/" TARGET="_blank">Institut 
		    Pasteur</A>, <A HREF="http://www.imbb.forth.gr/" 
		    TARGET="_blank">The Institute of Molecular Biology and 
		    Biotechnology</A> (IMBB) and 
		    <A HREF="http://www.tigr.org" TARGET="_blank">The Institute 
		    for Genomic Research</A> (TIGR).
		    <LI>
		    <B>Sequencing and Assembly:</B>
		    <A HREF="http://www.celera.com/" TARGET="_blank">Celera 
		    Genomics</A>, <A HREF="http://www.genoscope.cns.fr/" 
		    TARGET="_blank">Genoscope</A> 
		    <LI>
		    <B>Data Downloads:</B>
		    <A HREF="http://www.ensembl.org/Anopheles_gambiae/" 
		    TARGET="_blank">Ensembl</A>
		    <LI>
		    <B>UCSC <em>A. gambaie</em> Genome Browser</B>:
		    Angie Hinrichs, Galt Barber and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations</B>:
		    <A HREF="http://www.ensembl.org" TARGET="_blank">Ensembl</A> and
		    <A HREF="http://www.celera.com/" TARGET="_blank">Celera 
		    Genomics</A> (Ensembl Genes), the
		    <A HREF="../staff">UCSC Genome 
		    Bioinformatics Group</A>, University of California,
		    Santa Cruz, CA, USA (remaining annotations)
		    </UL>
		    <P>
		    <HR>
		    The <em>A. gambaie</em> sequence is made freely available 
		    by The International Anopheles Genome Project. 
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- C. elegans Credits================================================== -->
<A NAME="ce_credits"></A>
<A NAME="ce_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>C. elegans</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The <em>C. elegans</em> data was obtained from 
		    <A HREF="http://www.wormbase.org/" 
		    TARGET="_blank">WormBase</A>. We'd like to thank
		    the <A HREF="http://genome.wustl.edu/" 
		    TARGET="_blank">Genome Sequencing Center</A> at 
		    Washington University in St. Louis and the 
		    <A HREF="http://www.sanger.ac.uk/" 
		    TARGET="_blank">Sanger Institute</A> for their collaborative 
		    work on sequencing the <em>C. elegans</em> genome, and the 
		    WormBase consortium for providing access to the current 
		    <em>C. elegans</em> sequence. </P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Misc Worm Credits================================================== -->
<A NAME="worm_credits"></A>
<A NAME="worm_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
  	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Worm Genomes (other than <em>C. elegans</em>)</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The latest versions of the <em>C. brenneri</em>, 
		    <em>C. briggsae</em>, <em>C. japonica</em>, 
		    <em>C. remanei</em>, and <em>P. pacificus</em> sequences 
		    were obtained from the Genome Sequencing Center at 
		    <A HREF="http://genome.wustl.edu/genome_group.cgi?GROUP=6"
		    TARGET="_blank">Washington University in St. Louis</A> (WUSTL)
		    School of Medicine. The cb1 browser data
		    were obtained from <A HREF="http://www.wormbase.org/" 
		    TARGET="_blank">WormBase</A>.
		    Please review the WUSTL 
		    <A HREF="http://genome.wustl.edu/data.cgi"
		    TARGET="_blank">data use policy</A> before using these data
		    in research or a publication. </P>
		    <P>
		    We'd like to thank WUSTL for providing the sequence data
		    for these assemblies, as well as the Sanger Institute for
		    their collaborative work in sequencing the initial 
		    <em>C. briggsae</em> cb1 genome. Thanks also to the 
		    WormBase consortium for providing access to the cb1 
		    sequence. 
		    </P>
		</TD><TD WIDTH=15></TD></TR>
	     </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Sacchromyces cerevisiae Credits======================= -->
<A NAME="yeast_credits"></A>
<A NAME="yeast_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
  	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Yeast Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
			<P><b>UCSC <I>Sacchromyces cerevisiae</I> Genome Browser (sacCer3):</b> Hiram Clawson, 
			Greg Roe, Steve Heitner.<br>
		    The April 2011 <I>Sacchromyces cerevisiae</I> genome assembly 
		    is based on sequence dated April 2011 in the 
		    <A HREF="http://www.yeastgenome.org/" TARGET="_blank">Sacchromyces Genome
		    Database</A> (SGD). This genome sequence was downloaded from
			<a href="ftp://ftp.ncbi.nlm.nih.gov/genbank/genomes/Eukaryotes/fungi/Saccharomyces_cerev
			isiae/SacCer_Apr2011/" TARGET="_blank">NCBI</a> (genbank/genomes/Eukaryotes/fungi/Saccharomy
			ces_cerevisiae/SacCer_Apr2011/). The SGD&#8482; gene annotations were downloaded 
			from the <A HREF="http://downloads.yeastgenome.org/curation/chromosomal_feature/" TARGET="_blank"
			>SGD downloads site</A>. The S288C strain was used in this sequencing project.
		    </P>
		    
		    <P>
		    <b>The June 2008 <I>Sacchromyces cerevisiae</I> genome assembly</b> 
		    is based on sequence dated June 2008 in the 
		    <A HREF="http://www.yeastgenome.org/" TARGET="_blank">Sacchromyces Genome
		    Database</A> (SGD) and was obtained from the site
		    http://downloads.yeastgenome.org/sequence/genomic_sequence/chromosomes/fasta/
		    The S288C strain was used in this sequencing project.
		    </P>
		    <P>
		    <b>The Oct. 2003 <I>Sacchromyces cerevisiae</I> genome assembly</b> 
		    is based on sequence dated 1 Oct. 2003 in the 
		    <A HREF="http://www.yeastgenome.org/" TARGET="_blank">Sacchromyces Genome
		    Database</A> (SGD). The sequence, open reading frame (ORF), 
		    and gene annotations were downloaded from the site
		    <A HREF="ftp://genome-ftp.stanford.edu/pub/yeast/data_download"
		    TARGET="_blank">ftp://genome-ftp.stanford.edu/pub/yeast/data_download</A>. 
		    The cellular localization and protein abundance data displayed
		    in the Gene Sorter are taken from the
		    <A HREF="http://yeastgfp.yeastgenome.org/" TARGET="_blank">Yeast GFP 
		    Fusion Localization Database</A>.
		    <P>See the SGD 
		    <A HREF="http://www.yeastgenome.org/chromosomes.shtml" 
		    TARGET="_blank">Systematic Sequencing Table</A> for credit and
		    contact information for each of the chromosomes in the 
		    assembly. We'd like to thank Stanford University, the SGD,
		    the University of California San Francisco (UCSF), 
		    Washington University in St. Louis, and The Broad 
		    Institute 
		    for providing the data and annotations for this release.
		</TD><TD WIDTH=15></TD></TR>
	     </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- California Sea Hare Credits======================= -->
<A NAME="seahare_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
  	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; California Sea Hare Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The <I>Aplysia californica</I> genome assembly 
		    has the following acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly:</B>
		    <A HREF="http://www.broadinstitute.org/"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		    <LI>
		    <B>UCSC California Sea Hare (aplCal1) Genome Browser:</B>
		    Galt Barber, Brian Raney, Hiram Clawson, Katrina Learned and Donna 
		    Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="seahare_use"></A>
		    The California sea hare sequence is made freely available 
		    before 
		    scientific publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data (The Broad Institute)
			is properly acknowledged.
			<LI>
			The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data 
			set, including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	     </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Other Credits ====================================================== -->
<A NAME="other_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; External Contributors
		to the UCSC Genome Browser Project</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	    <TH HEIGHT=10></TH>
		<TR><TD COLSPAN=2></TD></TR>
	 	<TR><TD COLSPAN=2>
		    <UL>
			<LI>
			<A HREF="http://www.affymetrix.com" 
			TARGET="_blank">Affymetrix</A>: Consensus and exemplar
			sequences for probe selection on several chips, 
			exon array data, transcriptome data, Genie Gene 
			predictions, 
			design and analysis support in the early stages of the 
			UCSC Human Genome project, including software for the 
			synthetic chromosome 22 data we used to test our 
			assembly methods.
			<LI>
			<A HREF="http://www.genomics.agilent.com/"
TARGET="_blank">Agilent Technologies</A> 
			(Shane Giles, Peter Webb, and Dione Bailey): Agilent 
			microarray probe sets.
			<LI>
			<A HREF="http://www.alleninstitute.org/"
TARGET="_blank">Allen Institute 
			for Brain Science</A>: Allen Brain Atlas probes.
			<LI>
			<A HREF="http://www.ncbi.nlm.nih.gov/genome/cyto/hbrc.shtml" TARGET="_blank">BAC Resource 
			Consortium</A>: Cytogenetic markers (FISH-mapped clones annotation track).
			<LI>
			<A HREF="http://www.hgsc.bcm.edu/" TARGET="_blank">Baylor College of Medicine</A>
			(Richard Gibbs, John Bouck, Kim Worley and James Durbin): Rat Genome sequence data,
			<em>D. pseudoobscura</em> sequence for alignment tracks
			with <em>D. melanogaster</em> genome. 
			<LI>
			<A HREF="http://www.fruitfly.org/" TARGET="_blank">Berkeley Drosophila Genome 
			Project</A> (BDGP): Data for <em>D. melanogaster</em> gene annotations.
			<LI>
			<A HREF="http://www.biobase-international.com/"
			TARGET="_blank">BioBase</A>: Transfac database data used
			in HMR Conserved Transcription Factor Binding Sites
			track.
			<LI>
			<A HREF="http://www.bioinformaticssolutions.com/" TARGET="_blank">Bioinformatics Solutions, Inc.</A>
			(Ming Li and colleagues): Pattern Hunter mouse homology track.
			<LI>
			<A HREF="http://www.thebiomarkersconsortium.org/"
			TARGET="_blank">Biomarkers Consortium</A>: I-SPY 
			collaboration on the UCSC Cancer Genome Browser
			project.
			<LI>
			<A HREF="http://h-invitational.jp/hinv/ahg-db/index.jsp"
			TARGET="_blank">Biomedicinal Information Research Center 
			(BIRC)</A>, 
			National Institute of Advanced Industrial Science and 
			Technology (AIST): H-InvDB Gene Predictions.
			<LI>
			<A HREF="http://dna.bu.edu/tullius/"
TARGET="_blank">Boston 
			University</A> (Tullius Lab, in collaboration with MIT,
			NHGRI, and the La Jolla Institute for Allergy and 
			Immunology): ORChID Predicted DNA Cleavage 
			Sites (ENCODE).
			<LI>
			<A HREF="http://www.broadinstitute.org/"
			TARGET="_blank">The Broad Institute</A>
			<BR>--(Eric Lander, Mike Zody, Chad Nusbaum): Plasmid end reads. Human working
			draft sequence analysis project. Special thanks to Eric 
			for inviting us to join the Hardcore Analysis Group 
			for the draft sequence project and for 
			his direction of that group. 
			<BR>--(<A HREF="http://csb.mgh.harvard.edu/bradley_bernstein"
			TARGET="_blank">Bernstein Lab</A>): Histone modifications
			using ChIP-seq (ENCODE).
			<LI>
			<A HREF="http://woldlab.caltech.edu/" 
			TARGET="_blank">California Institute of Technology</A> 
			(Wold Lab): RNA-Seq data (ENCODE).
			<LI>
			<A HREF="http://www.bcgsc.ca/"
			TARGET="_blank">Canada's Michael Smith Genome Sciences
			Centre</A>: ORegAnno data, brain DNA methylation track 
			(ENCODE).
			<LI>
			<A HREF="http://www.csmc.edu/csri/korenberg/int-bac-sts.html" TARGET="_blank">Cedars-Sinai Medical Center</A> (CSMC):
			FISH-mapped clones annotation track. 
			<LI>
			<A HREF="http://projects.tcag.ca/variation/" 
			TARGET="_blank">Centre for Applied Genomics</A>: Database 
			of Genomic Variants (DGV) data.
			<LI>
			<A HREF="http://www.cephb.fr/en/cephdb/#"
			TARGET="_blank">Centre d'Etude du Polymorphisme Humain</A>
			(CEPH): Data from Genotype database for SNP Population
			Allele Frequencies track.
			<LI>
			<A HREF="http://pasteur.crg.es/portal/page/portal/Internet/02_Research/01_Programmes/01_Bioinformatics_Genomics"
			TARGET="_blank">Centre for Genomic Regulation (CRG)</A> 
			(Guigo Lab): CRG mapability (alignability) data,
			RNA-seq data.
			<LI>
			<A HREF="http://www-lbme.biotoul.fr/" TARGET=blank>CNRS 
			Universit Paul Sabatier</A>, Laboratoire de Biologie Molculaire Eucaryote, Toulouse, 
			France (Michel Weber): miRNA Data annotation track (in collaboration with Sam 
			Griffiths-Jones of The Wellcome Trust Sanger Institute).
			<LI>
			<A HREF="http://www.cshl.org/" TARGET=blank>Cold 
			Spring Harbor Laboratory</A> (CSHL)
			<BR>--(<A HREF="http://stein.cshl.org/"
			TARGET="_blank">Stein Lab</A>): HapMap data.
			<BR>--(<A HREF="http://gingeraslab.cshl.edu/" 
			TARGET="_blank">Gingeras Lab</A> and
			<A HREF="http://hannonlab.cshl.edu/"
			TARGET="_blank">Hannon Lab</A>): RNA-seq data
			tracks (ENCODE). 
			<BR>--(<A HREF="http://rulai.cshl.org/" 
			TARGET="_blank">Ivo Grosse and Michael Q. Zhang</A>): 
			Predictions for First Exon Finder annotation track.
			<LI>
			<A HREF="http://www.ncbi.nlm.nih.gov/projects/CCDS/CcdsBrowse.cgi"
			TARGET="_blank">Consensus Coding Sequence Project</A> 
			(EBI, NCBI, WTSI, UCSC): Consensus CDS Genes.
		    	<LI>
			<A HREF="http://compgen.bscb.cornell.edu/" 
			TARGET="_blank">Cornell University (Siepel Lab)</A>: 
			Exoniphy.
			<LI>
			<A HREF="http://www.genome.duke.edu/index.php" 
			TARGET="_blank">Duke University</A>
			(Institute for Genome Sciences and Policy): 
			Open chromatin track, Human tissue Affymetrix Exon 
			expression array data (ENCODE).
			<LI>
			<A HREF=" http://www.ebi.ac.uk/embl/" TARGET="_blank">EMBL</A> (Peter Bork): Mouse pseudogene
			predictions used for Bork Pseudogenes custom annotation track.
			<LI>
			<A HREF="http://genetics.emory.edu/egl/about/index.php/1202"
			TARGET="_blank">Emory University School of Medicine</A> 
			(D.  Ledbetter,
			C. Martin): Collaboration on CGH-ICSA grant. 
			<LI>
			<A HREF="http://www.ensembl.org/" 
			TARGET="_blank">Ensembl</A> (Paul Flicek,  Guy Slater, 
			Imre Vastrik, Arek Kasprzyk, Michele Clamp): Ensembl 
			Gene predictions, 
			data exchange formats, 
			TPF clone map coordination, 
			exonerate mouse homology track, early help placing FISH-mapped clones. 
			<LI>
			<A HREF="http://www.ieo.it/english/Pages/Default.aspx"
			TARGET="_blank">European Institute of Oncology</A>:
			Nuclease accessible sites (ENCODE).
			<LI>
			<A HREF="http://www.flybase.org/" TARGET="_blank">FlyBase</A>: Curated Drosophila
			gene annotations.
			<LI>
			<A HREF="http://www.fhcrc.org/" TARGET="_blank">Fred Hutchison Cancer Research Center</A> (FHCRC)
			(Barbara Trask): FISH-mapped clones annotation track.
			<LI>
			<A HREF="http://www.geneontology.org/" TARGET="_blank">Gene Ontology (GO) Consortium</A>:
			Structured annotations for Gene Sorter.
			<LI>
			<A HREF="http://www.girinst.org/"
			TARGET="_blank">Genetic Information Research Institute</A>
			(GIRI): Repbase repeat library, repeatmasking tools.
			<LI>
			<A HREF="http://www.genoscope.cns.fr/" TARGET="_blank">Genome Informatics Research Lab</A> 
			(GIRL) within Grup de Recerca en Informtica Mdica (GRIB) (Francisco Cmara Ferreira, 
			Roderic Guig): Geneid and SGP gene prediction annotation tracks. 
			<LI>
			<A HREF="http://web.gnf.org/" TARGET="_blank">Genomics Institute of the Novartis
			Research Foundation</A> (GNF): Expression data using Affymetrix chips for GNF Expression
			Atlas annotation tracks.
			<LI>
			<A HREF="http://www.gis.a-star.edu.sg/internet/site/" 
			TARGET="_blank">Genome Institute of Singapore</A>
			(GIS): DNA Paired-end Ditag data, RNA-seq data (ENCODE).
			<LI>
			<A HREF="http://www.genoscope.cns.fr/" TARGET="_blank">Genoscope</A>
			(Jean Weissenbach and Olivier Jaillon): Exofish program, <em>Tetraodon nigroviridis</em> 
			sequence for Tetraodon Blat alignments annotation tracks, data for Anopheles
			Evolutionary Conserved Regions annotation track.
			<LI>
			<A HREF="http://www.helicosbio.com/"
			TARGET="_blank">Helicos BioSciences</A>: RNA-seq data 
			(ENCODE).
			<LI>
			<A HREF="http://www.sickkids.on.ca/"
			TARGET="_blank">Hospital for Sick Children</A> (Sick 
			Kids) (L. Feuk): Structural variation data.
			<LI>
			<A HREF="http://www.hudsonalpha.org/myers-lab"
			TARGET="_blank">HudsonAlpha Institute for
			Biotechnology</A> (Myers Lab): RNA-seq data (ENCODE),
			Methyl-seq data (ENCODE), transcription factor binding 
			sites identified by ChIP-seq (ENCODE), copy number
			variation (CNV) data (ENCODE). 
			<LI>
			<A HREF="http://www.illumina.com/" 
			TARGET="_blank">Illumnia</A>: Probe set alignments.
			<LI>
			Institute for Genomic Research (TIGR): TIGR Gene Index 
			annotation track. 
			<LI>
			<A HREF="http://www.systemsbiology.org" TARGET="_blank">Institute for Systems Biology</A>
			(Arian Smit): <A HREF="http://www.repeatmasker.org" 
			TARGET="_blank">RepeatMasker</A> repeat identification and masking software and libraries.
			<LI>
			<A HREF="http://www.knockoutmouse.org/about"
			TARGET="_blank">International Knockout Mouse 	
			Consortium</A>: Data for IKMC tracks.
			<LI>
			Iowa State University (J.  Murray): Collaboration on 
			<A HREF="https://www.facebase.org/node/68"
			TARGET="_blank">FaceBase</A> project.
			<LI>
			<A HREF="http://www.jcvi.org/"
			TARGET="_blank">J. Craig Venter Institute</A>:
			Nuclease accessible sites (ENCODE).
			<LI>
			<A HREF="http://www.informatics.jax.org/" 
			TARGET="_blank">Jackson Laboratory</A>: 
			Mouse Genome Informatics tracks, 
			Mouse ortholog links from known genes track. 
			<LI>
			<A HREF="http://www.genome.ad.jp" TARGET="_blank">Kyoto University</A>, GenomeNet: 
			<A HREF="http://www.genome.ad.jp/kegg/kegg2.html" TARGET="_blank">Kyoto Encyclopedia of 
			Genes and Genomes</A> (KEGG) - pathway diagrams for Known Genes annotation track.
			<LI>
			<A HREF="http://www.jgi.doe.gov/whoweare/pennacchio.html"
			TARGET="_blank">Lawrence Berkeley National Laboratory</A> 
			(L.A. Pennacchio): Vista HMR-Conserved
			non-coding human enhancers track.
			<LI>
			<A HREF="http://www.lanl.gov/" TARGET="_blank">Los Alamos National Laboratory</A> (LANL):
			FISH-mapped clones annotation track. 
		 	<LI>
			<A HREF="http://genes.mit.edu/burgelab/" 
			TARGET="_blank">Massachusetts Institute of Technology
			(MIT)</A> (Chris Burge): RNA-Seq data, Genscan gene 
			prediction program. 
			<LI>
			<A HREF="http://rgd.mcw.edu/" TARGET="_blank">Medical
			College of Wisconsin</A>: Rat Genome Database (RGD).
		 	<LI>
			<A HREF="http://www2.mrc-lmb.cam.ac.uk/" 
			TARGET="_blank">Medical Research Council Laboratory 
			of Molecular Biology</A>: 
			<A HREF="http://supfam.mrc-lmb.cam.ac.uk/SUPERFAMILY/" 
			TARGET="_blank">Superfamily</A> database.
			<LI>
			<A HREF="http://www.mountsinai.org/" TARGET="_blank">
			Mount Sinai School of Medicine</A> 
			(<A HREF="http://tandem.bu.edu/benson.html" 
			TARGET="_blank">Gary Benson</A> - now at Boston 
			University): Tandem Repeats Finder.
			<LI>
			<A HREF="http://cancer.gov/" 
			TARGET="_blank">National Cancer Institute</A> (NCI), NIH: 
			<BR>--<A HREF="http://cancergenome.nih.gov/"
			TARGET="_blank">The Cancer Genome Atlas</A> (TCGA).
			<BR>--<A HREF="http://mgc.nci.nih.gov" 
			TARGET="_blank">Mammalian Gene Collection (MGC) 
			Project</A>. 
			<BR>--<A HREF="http://cgap.nci.nih.gov/" 	
			TARGET="_blank">Cancer Genome Anatomy Project</A> (CGAP). 
			<BR>--SAGE project (in collaboration with NCBI).
			<BR>--FISH-mapped clones annotation track. 
		 	<LI>
			<A HREF="http://www.ncbi.nlm.nih.gov/index.html" 
			TARGET="_blank">National Center for Biotechnology
			Information</A> (NCBI), NIH 
			<BR>--<A HREF="http://www.ncbi.nlm.nih.gov/RefSeq/" 
			TARGET="_blank">RefSeq</A>, 
			<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=gene"
			TARGET="_blank">Entrez Gene</A>,
			<A HREF="http://www.ncbi.nlm.nih.gov/SNP/" 
			TARGET="_blank">dbSNP</A>, 
			<A HREF="http://hapmap.ncbi.nlm.nih.gov/index.html.en"
			TARGET="_blank">HapMap</A>,
			<A HREF="http://www.ncbi.nlm.nih.gov/sites/entrez?db=omim"
			TARGET="_blank">OMIM</A>, and 
			<A HREF="http://www.ncbi.nlm.nih.gov/pubmed"
			TARGET="_blank">PubMed</A> databases.
			<BR>--<A HREF="http://www.ncbi.nlm.nih.gov/SAGE/" 
			TARGET="_blank">Serial Analysis of
			Gene Expression Tag to Gene Mapping</A> (SAGEmap) data resource.
			<BR>--(Deanna Church): Mouse synteny track, database of 
			whole genome shotgun mouse reads and assemblies used for
			mouse tracks.
			<BR>--(Greg Schuler): GenBank freezes of human clones 
			used in the assembly, NT contigs of multiple 
			finished clones, STS maps (in earlier freezes), 
			specially-formatted versions of the finished 
			chromosomes. 
			<BR>--(Steve Sherry): SNP tracks. 
			<BR>--(Wonhee Jang): FISH-mapped clones annotation 
			track.
			<BR>--(Danielle and Jean Thierry-Mieg): Acembly Genes 
			annotation track.
			<BR>--(Lukas Wagner): 3' EST track in the July 2000 
			browser.
			<LI>
			<A HREF="http://www.genome.gov" TARGET="_blank">National 
			Human Genome Research Institute</A> 
			(NHGRI), NIH
			<BR>--(<A HREF="http://www.genome.gov/12514761"
			TARGET="_blank">Elnitski Lab</A>): Bidirectional 
			promoters, negative regulatory elements (ENCODE).
			<BR>--(ENCODE group): Guidance for ENCODE DCC, 
			collaboration and contributions for various ENCODE 
			annotation tracks.
			<BR>--<A HREF="http://www.genome.gov/gwastudies/"
			TARGET="_blank">GWAS</A>: GWAS data from the Catalog 
			of Published Genome-Wide Association Studies.
			<BR>--(NISC group): Collaboration and annotations
			for <A HREF="http://www.nisc.nih.gov/index.html" 
			TARGET="_blank">NISC</A> browser, including 
			curated gene annotations, DNAse I Hypersensitive Sites 
			annotation, Multi-species Conserved Sequences (MCS) 
			tracks, Binomial_25 Conservation Scores,
			ZooSeq Blastz alignments tracks. 
			<LI>
			<A HREF="http://www.nia.nih.gov/" 
			TARGET="_blank">National Institute on Aging</A>, NIH:
			Data from the 
			<A HREF="http://geneticassociationdb.nih.gov/"
			TARGET="_blank">Genetic Association Database</A>.
			<LI>
			<A HREF="http://research.nki.nl/vansteensellab/"
			TARGET="_blank">Netherlands Cancer Institute</A> (van Steensel Lab):
			Nuclear lamina associated domains tracks.
			<LI>
			<A HREF="http://bmi.osu.edu/" TARGET="_blank">Ohio State
			University Bioinformatics & Computational Biology Group</A> (Ramana V. Davuluri): 
			Predictions for First Exon Finder annotation track.
			<LI>
			<A HREF="http://www.orfeomecollaboration.org/"
			TARGET="_blank">ORFeome Collaboration</A>: ORFeome
			Collaboration human gene clones.
			<LI>
			<A HREF="http://www.bx.psu.edu/" 
			TARGET="_blank">Pennsylvania State University 
			Center for Comparative Genomics and 
			Bioinformatics</A>: BLASTZ, LASTZ, and multiz alignment 
			programs, Genome Alignment and Annotation Database 
			(GALA), collaboration on multiple alignment annotation 
			tracks, ENCODE annotation tracks, Personal Genome 
			Variants track, Galaxy, regulatory potential scores 
			tracks.
			<LI>
			Perlegen Sciences: STS markers for 
			the Haplotype Blocks annotation track on chr 21,
			SNP data for the Affymetrix 120K SNP track, 
			TNG radiation hybrid map.
			<LI>
			<A HREF="http://www.gsc.riken.go.jp" 
			TARGET="_blank">RIKEN
			Genomic Sciences Center</A>: Chimpanzee BAC end sequence
			data, mouse annotations, human genome sequence 
			contributions, CAGE tags RNA subcellular localization
			data (ENCODE).
			<LI>
			<A HREF="http://www.roswellpark.org/" 
			TARGET="_blank">Roswell Park Cancer Institute</A> (RPCI):
			FISH-mapped clones annotation track.
			<LI>
			<A HREF="http://snp.cshl.org/" 
			TARGET="_blank">SNP Consortium</A>: Data for SNP 
			annotation tracks.
			<LI>
			<A HREF="http://www.softberry.com/" 
			TARGET="_blank">Softberry Inc.</A> (V. Solovyev):
			Fgenesh++ and Fgenesv+ gene predictions, TSSW promoters 
			tracks.
			<LI>
			<A HREF="http://www.mad-cow.org/" 
			TARGET="_blank">Sperling Biomedical 
			Foundation</A> (Thomas Pringle): Technical consultant on
			Proteome Browser, avid Genome Browser user and the 
			source of myriad ideas for expanding its functionality, 
			early assistance and support for UCSC Genome Browser.
			<LI>
			<A HREF="http://bejerano.stanford.edu/"
			TARGET="_blank">Stanford University</A> 
			<BR>--(<A HREF="http://bejerano.stanford.edu/"
			TARGET="_blank">Bejerano Lab</A>): GREAT tool.
			<BR>--(<A HREF="http://www.stanford.edu/group/morrinst/abi.html"
			TARGET="_blank">Morrison Institute for Population
			and Resource Studies</A>, Human Genome Diversity 
			Project): Data for SNP Population Allele Frequencies 
			track.
			<BR>--<A HREF="http://source.stanford.edu/"
			TARGET="_blank">SOURCE tool</A>: SOURCE links from UCSC
			Genes tracks.
			<LI>
			Stanford University School of 
			Medicine (Myers Lab - now at HudsonAlpha): Data set for 
			Human Promoters custom annotation track.
			<LI>
			<A HREF="http://switchgeargenomics.com/"
			TARGET="_blank">SwitchGear Genomics</A> (N. Trinklein 
			and S.F. Aldred): Transcription start sites (TSSs).
			<LI>
			<A HREF="http://www.uniprot.org/" 
			TARGET="_blank">UniProt</A>: Protein data from UniProt 
			database.
			<LI>
			<A HREF="http://bioinfo2.ugr.es/isochores/" 
			TARGET="_blank">Universidad de Granada</A>, 
			Grupo de Bioinformtica, Spain (Jose L. Oliver): 
			Isochore custom annotation track.
			<LI>
			<A HREF="http://www.albany.edu/cancergenomics/faculty/stenenbaum/index.shtml"
			TARGET="_blank">University at Albany - SUNY</A> 
			(TenenBaum Lab): Target RNA binding protein 
			expression measured by RIP-chip (ENCODE).
			<LI>
			<A HREF="http://computationalbiology.berkeley.edu/" 
			TARGET="_blank">University of CA, Berkeley</A> (Lior
			Pachter, Simon Cawley, Marina Alexandersson): 
			<A HREF="http://bio.math.berkeley.edu/slam/mouse/" 
			TARGET="_blank">SLAM</A> Gene predictions.
			<LI>
			<A HREF="http://bioinformatics.ucsd.edu/"
			TARGET="_blank">University of CA, San Diego</A> 
			<BR>--(<A HREF="http://www.snl.salk.edu/~geneyeo/"
			TARGET="_blank">G. Yeo</A>, now at Salk Institute): 
			FOX2 adaptor-trimmed CLIP-seq reads track.
			<BR>--(<A HREF="http://bioinformatics-renlab.ucsd.edu/rentrac/"
			TARGET="_blank">Ren Lab</A>, Ludwig Institute for Cancer
			Research): TAF1 binding in fibroblastoid cells assayed
			by ChIP-chip (ENCODE).
			<LI>
			<A HREF="http://www.ucsf.edu/" 
			TARGET="_blank">University of CA, San Francisco</A>:
			Collaboration on UCSC Cancer Browser, 	
			Chimera links from
			SNPs and UCSC Genes, FISH-mapped clones annotation 
			track.
			<BR>--(ENCODE group): Brain DNA methylation track 
			(ENCODE).
			<LI>
			<A HREF="http://www.bme.ucsc.edu/"
			TARGET="_blank">University of CA, Santa Cruz</A> 
			<BR>--(<A HREF="http://www.bme.ucsc.edu/people/ed"
			TARGET="_blank">R.E. Green</A>): Neanderthal annotation 
			tracks.
			<BR>--(<A HREF="http://lowelab.ucsc.edu/"
			TARGET="_blank">Lowe Lab</A>): tRNA Gene Predictions 
			using tRNAscan-SE.  
			<LI>
			<A HREF="http://www.oncology.cam.ac.uk/"
			TARGET="_blank">University of Cambridge, Department of
			Oncology</A>: Open chromatin track (ENCODE).
			<LI>
			<A HREF="http://pritch.bsd.uchicago.edu/"
			TARGET="_blank">University of Chicago</A> (Pritchard Lab):
			Collaboration on Human Genome Diversity Project tracks.
			<LI>
			<A HREF="http://lieblab.bio.unc.edu/"
			TARGET="_blank">University of North Carolina, 
			Chapel Hill</A> (Lieb Lab): Open chromatin track 
			(ENCODE).
			<LI>
			<A HREF="http://www.allgenes.org/" TARGET="_blank">University of Pennsylvania</A> (Y. Thomas Gan):
			Database of Transcribed Sequences (DoTS) Genes custom annotation track.
			<LI>
			University of Pittsburgh (M. Marazita): 
			Collaboration on 
			<A HREF="https://www.facebase.org/node/68"
			TARGET="_blank">FaceBase</A> project.
			<LI>
			<A HREF="http://www.genenetwork.org/"
			TARGET="_blank">University of Tennessee GeneNetwork</A>: GeneNetwork 
			links from UCSC Genes annotations.
			<LI>
			<A HREF="http://www.uthscsa.edu/csb/faculty/penalva.asp"
			TARGET="_blank">University of Texas Health Science
			Center</A>, Greehy Children's Cancer Research Institute 
			(L.O.F. Penalva): 
			<LI>
			<A HREF="http://microarray.icmb.utexas.edu/research.html"
			TARGET="_blank">University of Texas, Austin</A> (Iyer 
			Lab): Open chromatin track (ENCODE).
			<LI>
			<A HREF="http://www.tbi.univie.ac.at/index.html" 
			TARGET="_blank">University of Vienna</A>, Institute 
			for Theoretical Chemistry and Structural Biology, 
			Theoretical Biochemistry Group</A>: 
			Vienna RNA utility 
			<A HREF="http://www.tbi.univie.ac.at/~ivo/RNA/" 
			TARGET="_blank">RNAplot</A> - Known Genes annotation 
			track. 
			<LI>
			<A HREF="http://www.gs.washington.edu/"
			TARGET="_blank">University of Washington</A>
			<BR>--(<A HREF="http://eichlerlab.gs.washington.edu/" 
			TARGET="_blank">Eichler Lab</A>): Segmental duplications 
			data. 
			<BR>--(<A HREF="http://hgsv.washington.edu/"
			TARGET="_blank">Human Genome Structural Variation 
			Project</A>): Variation data for HGSV tracks.
			<BR>--(ENCODE group): Digital DNase genomic footprinting
			track, DNaseI sensitivity using digital DNaseI
			methodology, DNaseI hypersensitive sites, histone
			modifications using ChIP-seq (ENCODE).
			<BR>--(<A HREF="http://noble.gs.washington.edu/"
			TARGET="_blank">Noble Lab</A>): Predicted nucleosome 
			occupancy tracks.
			<BR>--(<A HREF="http://www.gs.washington.edu/faculty/nickerson.htm"
			TARGET="_blank">Nickerson Lab</A>): Tajima's D SNPs track.
			<LI>
			<A HREF="http://www.klinimm.uu.se/research/genetics_genomics/wadelius/"
			TARGET="_blank">Uppsala University</A> 
			(Wadelius Lab): ChIP-chip data (ENCODE).
			<LI>
			<A HREF="http://www.wustl.edu/" 
			TARGET="_blank">Washington University in St. Louis</A> and
			<A HREF="http://medschool.wustl.edu/"
			TARGET="_blank">Washington University in St. Louis
			School of Medicine</A> 
			<BR>--(Robert Waterston (now at UW), John McPhearson, Asif Chinwalla, Shiaw-Pyng Yang, John Wallis and 
			LaDeana Hillier): Fingerprint-based map of BAC clones from the human genome and a 
			corresponding layout of sequenced clones, plasmid end reads, BACend analysis,
			CpG islands.
			<BR>--(<A HREF="http://www.genetics.wustl.edu/cgs/eddy.htm" TARGET="_blank">Sean Eddy</A>
			and Todd Lowe (now at UCSC)): RNA Genes track.
			<BR>--(<A HREF="http://mblab.wustl.edu/" TARGET="_blank">Michael 
			Brent Lab</A>): N-SCAN and Twinscan Genes.
			<BR>--(ENCODE group): Brain DNA methylation track 
			(ENCODE).
			<LI>
			<A HREF="http://www.sanger.ac.uk/HGP/" TARGET="_blank">Wellcome Trust Sanger Institute</A>
			<BR>--(J. Harrow, J. Gilbert, J. Sulston, J. Rogers, 
			J. Mullikin, R. Durbin, S. Searle, I. Dunham and 
			the annotation teams): Vega Genes and Pseudogenes, 
			Sanger 20 and Sanger 22 Genes and Pseudogenes. Plasmid 
			end reads, processing of whole genome shotgun mouse 
			reads. Eponine program.
			<BR>--(S. Griffiths-Jones): miRNA Data annotation track 
			(in collaboration with 
			M. Weber of CNRS Universit Paul Sabatier).
			<BR>--<A HREF="http://decipher.sanger.ac.uk/"
			TARGET="_blank">Decipher</A>
			<BR>--<A HREF="http://www.sanger.ac.uk/Software/Pfam/" 
			TARGET="_blank">Protein Families 
			Database of Alignments and HMMs</A> (Pfam).
			<LI>
			<A HREF="http://jura.wi.mit.edu/bio/"
			TARGET="_blank">Whitehead Institute, Bioinformatics and
			Research Computing</A> (G. Bell): TargetScan miRNA 
			regulatory sites.
			<LI>
			<A HREF="http://www.wormbase.org/" TARGET="_blank">WormBase</A> (Lincoln Stein): Wormbase 
			curated gene annotations for <em>C. elegans</em> genome.
			<LI>
			<A HREF="http://www.cbb.yale.edu/" 
			TARGET="_blank">Yale University</A>: 
			<BR>--(<A HREF="http://www.cbb.yale.edu/" 
			TARGET="_blank">Snyder, Gerstein, and Weissman Labs</A>
			in collaboration with NASA Ames Research Center and 
			Eloret Corporation): 
			Transcriptionally Active Regions (TARs) data (ENCODE).
			<BR>--(<A HREF="http://info.med.yale.edu/neurobio/Sestan/Index.html"
			TARGET="_blank">Sestan Lab</A>): Exon microarray expression
			data from the late mid-fetal human brain.
		</UL></TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>

<!-- Early Credits ====================================================== -->
<A NAME="early_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD>
 	    <FONT SIZE="4"><B>&nbsp; Acknowledgment of Early Support</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	    <TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD>
		<TD>
		    <P>During the initial stage of this project, from January 2000 to June 2001,
		    we received no direct federal funding to develop the working draft other than travel 
		    support. However, we gratefully acknowledge the support for our personnel provided 
		    by our basic bioinformatics research grants, especially DOE grant
		    DE-FG03-99ER62849 (to develop gene-finding methods), but also NSF grant DBI-9809007, 
		    UC BIOTECH grant 99-11, a grant from the Sloan Foundation, the David and Lucille 
		    Packard Foundation, NIH grant GM-52848, and the Zahler grant that provided support 
		    for Jim Kent in the early stages of this work. 
		    <P>
		    We are particularly thankful for the risks that Patrick Mantey (UCSC Dean of Engineering), 
		    James Gill (Vice Chancellor for Research), John Simpson (UCSC Provost), and 
		    M.R.C. Greenwood (UCSC Chancellor) were willing to take 
		    in advancing us the money in February 2000 needed to build the original 100-node 
		    LINUX cluster used in this work, and in the case of Dean Mantey, additional 
		    personnel support. Phokion Kolaitis, Chair of Computer Science, contributed as 
		    well. 
		    <P>We thank <A HREF="http://www.compaq.com/" TARGET="_blank">Compaq</A> for providing an 
		    additional 100-CPU computational cluster in Cambridge, MA for use
		    by the public human genome effort. We thank <A HREF="http://www.ilog.com/" TARGET="_blank">ILOG,
		    Inc.</A> for providing discounted licenses to its CPLEX linear programming software, 
		    which was essential for some of our early experiments.</P>
		    <P>
		    In December 2000, <A HREF="http://www.hhmi.org" TARGET="_blank">HHMI</A> 
		    began providing support for equipment and personnel. In the summer 
		    of 2001, HHMI provided the primary funding for a new cluster of 1000 CPUs running 
		    LINUX purchased from <A HREF="http://www.rackable.com" TARGET="_blank">Rackable</A>, with additional 
		    support from <A HREF="http://www.ucop.edu/california-institutes" TARGET="_blank">CISI</A> and 
		    <A HREF="http://www.nhgri.nih.gov" TARGET="_blank">NHGRI</A>. Dubbed the 
		    UCSC KiloKluster, this equipment became the computational backbone
		    for the initial years of the Genome Browser project.   
		    </P>
		</TD>
		<TD WIDTH=15></TD>
	    </TR>
	</TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    </TD></TR>
</TABLE>
</BODY></HTML>
