<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: Data Organization</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../images/title.jpg"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/topbar.html"--></TD></TR>
<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>	
	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B> &nbsp;Data Organization and Format </B></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=15>&nbsp;&nbsp;</TD><TD>

<P>
The data for the working draft are organized hierarchically by
chromosome and by the sequenced-clone contigs within each chromosome.
At the top level there are 25 folders; 22 of these are for the
numbered chromosomes (autosomes), folders X and Y are for the sex
chromosomes, and Un is for clone contigs that cannot be placed
confidently on a chromosome.  Each of the 25 chromosomal folders
contains a separate clone contig folder for each of the clone contigs
for that chromosome. </P>

<P>
There are two primary files in each clone contig folder; these have
suffixes .fa and .agp respectively. The .fa files gives the working
draft sequence for the clone contig. The format is Fasta format,
e.g.</P>

<PRE>&gt;NT_077768
GAATTCTCTGTAACACTAAGCTCTCTTCCTCAAAACCAGAGGTAGATAGA
ATGTGTAATAATTTACAGAATTTCTAGACTTCAACGATCTGATTTTTTAA
ATTTATTTTTATTTTTTCAGGTTGAGACTGAGCTAAAGTTAATCTGTGGC
...</PRE>

<P>
The .agp file is a kind of index that tells how the .fa file is
built. It looks like</P>

<PRE>
17/NT_077768    1       6538    1       D       AC021317.18     122280  128817  -
17/NT_077768    6539    56206   2       D       AC021317.18     128918  178585  -
17/NT_077768    56207   56306   3       N       100     fragment        yes
17/NT_077768    56307   117971  4       D       AC021317.18     47188   108852  -
17/NT_077768    117972  170563  5       F       AC115992.13     23659   76250   +
17/NT_077768    170564  274979  6       D       AC124789.11     1       104416  -
...</PRE>

<P>Each line represents either an actual sequence record or a gap 
(unless it begins with "#", in which case it is a comment.)  
If the line represents an actual sequence record then it has the form</P>
<P>
&lt;chromosome/ctg&gt;
&lt;start-in-ctg&gt;
&lt;end-in-ctg&gt;
&lt;number&gt;
&lt;type&gt;
&lt;accession&gt;.&lt;version&gt;
&lt;start&gt;
&lt;end&gt;
&lt;orientation&gt;</P>
<P>and if it represents a gap it has the form</P>
<P>&lt;chromosome/ctg&gt;
&lt;start-in-ctg&gt;
&lt;end-in-ctg&gt;
&lt;number&gt;
N
&lt;number-of-Ns&gt;
&lt;kind&gt;
&lt;bridged?&gt;</P>

<P>
The positions &lt;start-in-ctg&gt; and &lt;end-in-ctg&gt; are the
start and end positions for where the sequence is to be put in the .fa
file. For a sequence record, the positions &lt;start&gt; and
&lt;end&gt; are the start and end positions of where the sequence came
from in the GenBank record &lt;accession&gt;.&lt;version&gt;. The
field &lt;orientation&gt; tells whether or not the sequence must be
reverse complemented before it is inserted into its place in the .fa
file. For example, the records above mean that to build the .fa file
for clone contig NT_077768 from chromosome 17 you take</P>

<PRE>
AC021317 version 18, residues 122280 to 128817, reverse complemented, followed by 
AC021317 version 18, residues 128918 to 178585, reverse complemented, followed by 
a gap of 100 Ns, followed by 
AC021317 version 18, residues 47188 to 108852, reverse complemented, followed by 
AC115992 version 13, residues 23659 to 76250, followed by 
AC124789 version 11, residues 1 to 104416, reverse complemented, followed by 
...</PRE>

<P>The joins perfectly abut.  In a sequence record, &lt;type&gt; can be </P>

<P>     F - Finished, 
A - in Active finishing, 
D - Draft, 
P - PreDraft, 
O - Other sequence </P>

<P>and in a gap record it is always N.  
The &lt;number&gt; field just sequentially numbers the records.</P>

<P>In a gap record, &lt;number-of-Ns&gt; is the size of the gap and 
&lt;kind&gt; is</P>
<UL>
<LI>    fragment - a gap between two sequence contigs (also called a 
        "sequence gap") 
<LI>    split_finished - a special sized gap between two finished sequence 
        contigs 
<LI>    clone - a gap between two clones that do not overlap
<LI>    contig - a gap between clone contigs in the genome layout (also called 
        a "layout gap")
<LI>    centromere - a gap inserted for the centromere 
<LI>    short_arm - a gap inserted at the start of an acrocentric chromosome 
<LI>    heterochromatin - a gap inserted for an especially large region of 
        heterochromatin (may include the centromere as well.) 
<LI>    telomere - a gap inserted for a telomere
</UL>

<P>&lt;bridged?&gt; is "yes" if there is a cDNA or BACend pair or 
plasmid end pair that spans the gap, else it is "no".</P>

<P>We provide three ways you can 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html" TARGET=_BLANK>download</A> 
these .fa and .agp files:  
<OL>
<LI> full data set: the entire hierarchy in a zipped format.
<LI> by chromosome: one zipped file for each chromosome containing all
     the sequence ordered along that chromosome.
<LI> by individual clone contig: separate files, not zipped, for each 
     clone contig.
</OL>
<BR></TD><TD WIDTH=15>&nbsp;&nbsp;</TD></TR></TABLE>
</TD></TR></TABLE>
</TD></TR></TABLE>
</BODY></HTML>






