<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
  <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
  <TITLE>UCSC Genome Browser: Annotation Database</TITLE>
  <LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">

<A NAME="TOP"></A>
<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
  <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../images/title.jpg"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/topbar.html"--></TD></TR>
<!--Intro---------------------------------------------------------------->
  <TR><TD COLSPAN=2 CELLPADDING=10>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1>
      <TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%">
	  <TR><TD>
	    <FONT SIZE="4"><B> &nbsp;The Genome Browser Database </B></FONT>
	  </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH>
	  <TR><TD WIDTH=10></TD>
	    <TD>
	      <P>
	      This page describes the format of the genome annotation databases 
	      that underlie the UCSC Genome Browser.  These data were contributed 
	      by many researchers, as described on the Genome Browser 
	      <A HREF="credits.html">Credits</A> page. Please acknowledge the 
	      contributor(s) of the data you use. All tables for an assembly 
	      are freely usable for any purpose except as indicated in the 
	      README.txt file in the downloadable 
	      <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">annotation 
	      database directory</A> for that release. </P>
	      <P>
	      <B>NOTE: This page is no longer maintained. For complete
	      up-to-date table descriptions, use the &quot;describe table
	      schema&quot; button in the <A HREF="../cgi-bin/hgTables">Table
	      Browser</A>.</B></P>
	    </TD><TD WIDTH=15></TD>
	  </TR>
	</TABLE>
	<BR>
      </TD></TR>
    </TABLE>
    <BR>

<!--Table Descriptions--------------------------------------------------->
	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
      <TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	  <TR><TD>
	    <FONT SIZE="4"><b>&nbsp; Table Descriptions</FONT>
	  </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH>
	  <TR><TD WIDTH=10></TD>
	  <TD>
	    <P>
	    The data displayed in the Genome Browser are stored in a 
            <A href="http://www.mysql.com/" TARGET=_blank>MySQL</A> database. 
	    The tables in the database can be grouped into four categories: 
	    <UL>
	    <LI>tables in which the data has been split into a separate table
	    for each chromosome
	    <LI>tables that contain information for all chromosomes
	    <LI>tables that contain information specifically related to mRNA 
 	    sequences
	    <LI>tables intended primarily for internal use
	    </UL>
            <P>
	    All coordinates in these tables are <em>half-open zero-based</em>.
	    This means that the first 100 bases of a chromosome are represented 
	    as [0,100), i.e. 0-99. The second 100 bases are represented as 
	    [100,200), i.e. 100-199, and 
	    so forth. An advantage of half-open coordinate ranges is that the 
	    length can be obtained by simply subtracting the start from the end.
	    </P>
            <P>
	    Data that have been split into a separate table for each chromosome
	    have the naming convention <em>chrN_[table]</em>, where <em>N</em> 
	    is replaced by the chromosome name. 
	    In many cases a table has been added that contains the unordered part 
	    (random) of the chromosome, <em>chrN_random_[table]</em>. Finally, in 
	    addition to the standard chromosome names, some assemblies have
	    two special chromosome names, <em>UL</em> and <em>NA</em>, representing 
	    fingerprint clone contigs that could not be placed confidently on a 
            chromosome, and individual clones that could not be placed, 
	    respectively. The table names are hyperlinked to an 
	    <A HREF="http://www.linuxjournal.com/article.php?sid=5949" 
	    TARGET=_blank>autoSql</A> format description of their fields.</P>
	   </TD><TD WIDTH=15></TD>
  	  </TR>
	</TABLE>
	<BR>
      </TD></TR>
    </TABLE>
    <BR>

<!--Chromomome-based tables---------------------------------------------->

    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
      <TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	  <TR><TD>
	    <FONT SIZE="4"><b>&nbsp; Tables Split by Chromosome </FONT>
	  </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH>
	  <TR><TD WIDTH=10></TD>
	  <TD>
	    <P>
	    This section shows data that has been split into a separate table
	    for each chromosome.
	    <P>
	      <CENTER>
	      <A HREF="#CHR6_GBD">[chr6]</A> 
	      <A HREF="#CHRN_B_GBD">[chrN_B]</A> 
	      <A HREF="#CHRN_C_GBD">[chrN_C]</A> 
	      <A HREF="#CHRN_D_GBD">[chrN_E]</A> 
	      <A HREF="#CHRN_G_GBD">[chrN_G]</A> 
	      <A HREF="#CHRN_H_GBD">[chrN_H]</A> 
	      <A HREF="#CHRN_I_GBD">[chrN_I]</A> 
	      <A HREF="#CHRN_M_GBD">[chrN_M]</A>
	      <A HREF="#CHRN_R_GBD">[chrN_R]</A> 
	      <A HREF="#CHRN_V_GBD">[chrN_V]</A> 
	      <A HREF="#CHRN_W_GBD">[chrN_W]</A> 
	      <A HREF="#CHRN_Z_GBD">[chrN_Z]</A>
	      </CENTER>
<UL>
<A NAME="CHR6_GBD"></A>
  <LI><A 
  href="#Chain">chr6_hla_hap1_chainCanFam1</A>
  - Summary info about chain of dog (canFam1) alignments on human chr6_hla_hap1</LI>
  <LI><A 
  href="#ChainLink">chr6_hla_hap1_chainCanFam1Link</A>
  - Alignment block in dog (canFam1) chain on human chr6_hla_hap1</LI></LI>
  <LI><A 
  href="#Chain">chr6_hla_hap1_chainDanRer1</A>
  - Summary info about chain of zebrafish (danRer1) alignments on human chr6_hla_hap1</LI>
  <LI><A 
  href="#ChainLink">chr6_hla_hap1_chainDanRer1Link</A>
  - Alignment block in zebrafish (danRer1) chain on human chr6_hla_hap1</LI></LI>
  <LI><A 
  href="#Chain">chr6_hla_hap1_chainGalGal2</A>
  - Summary info about chain of chicken (galGal2) alignments on human chr6_hla_hap1</LI>
  <LI><A 
  href="#ChainLink">chr6_hla_hap1_chainGalGal2Link</A>
  - Alignment block in chicken (galGal2) chain on human chr6_hla_hap1</LI></LI>
  <LI><A 
  href="#Chain">chr6_hla_hap1_chainMm5</A>
  - Summary info about chain of mouse (mm5) alignments on human chr6_hla_hap1</LI>
  <LI><A 
  href="#ChainLink">chr6_hla_hap1_chainMm5</A>
  - Alignment block in mouse (mm5) chain on human chr6_hla_hap1</LI></LI>
  <LI><A 
  href="#Chain">chr6_hla_hap1_chainPanTro1</A>
  - Summary info about chain of chimp (panTro1) alignments on human chr6_hla_hap1</LI>
  <LI><A 
  href="#ChainLink">chr6_hla_hap1_chainPanTro1Link</A>
  - Alignment block in chimp (panTro1) chain on human chr6_hla_hap1</LI></LI>
  <LI><A 
  href="#Chain">chr6_hla_hap1_chainRn3</A>
  - Summary info about chain of rat (rn3) alignments on human chr6_hla_hap1</LI>
  <LI><A 
  href="#ChainLink">chr6_hla_hap1_chainRn3</A>
  - Alignment block in rat (rn3) chain on human chr6_hla_hap1</LI></LI>
  <LI><A 
  href="#PslInfo">chr6_hla_hap1_est</A> 
  - EST alignments on human chr6_hla_hap1</LI>
  <LI><A 
  href="#GapInfo">chr6_hla_hap1_gap</A> 
  - Gap locations and types on human chr6_hla_hap1</LI>
  <LI><A 
  href="#FragInfo">chr6_hla_hap1_gl</A> 
  - Golden path positions of all clone fragments on human chr6_hla_hap1</LI>
  <LI><A 
  href="#GpInfo">chr6_hla_hap1_gold</A> 
  - Instructions on creating golden path from fragments on human chr6_hla_hap1</LI>
  <LI><A 
  href="#RepeatInfo">chr6_hla_hap1_rmsk</A> 
  - Repeating elements created using RepeatMasker <on human chr6_hla_hap1</LI>

  <LI><A 
  href="#Chain">chr6_hla_hap2_chainCanFam1</A>
  - Summary info about chain of dog (canFam1) alignments on human chr6_hla_hap2</LI>
  <LI><A 
  href="#ChainLink">chr6_hla_hap2_chainCanFam1Link</A>
  - Alignment block in dog (canFam1) chain on human chr6_hla_hap2</LI></LI>
  <LI><A 
  href="#Chain">chr6_hla_hap2_chainDanRer1</A>
  - Summary info about chain of zebrafish (danRer1) alignments on human chr6_hla_hap2</LI>
  <LI><A 
  href="#ChainLink">chr6_hla_hap2_chainDanRer1Link</A>
  - Alignment block in zebrafish (danRer1) chain on human chr6_hla_hap2</LI></LI>
  <LI><A 
  href="#Chain">chr6_hla_hap2_chainGalGal2</A>
  - Summary info about chain of chicken (galGal2) alignments on human chr6_hla_hap2</LI>
  <LI><A 
  href="#ChainLink">chr6_hla_hap2_chainGalGal2Link</A>
  - Alignment block in chicken (galGal2) chain on human chr6_hla_hap2</LI></LI>
  <LI><A 
  href="#Chain">chr6_hla_hap2_chainMm5</A>
  - Summary info about chain of mouse (mm5) alignments on human chr6_hla_hap2</LI>
  <LI><A 
  href="#ChainLink">chr6_hla_hap2_chainMm5</A>
  - Alignment block in mouse (mm5) chain on human chr6_hla_hap2</LI></LI>
  <LI><A 
  href="#Chain">chr6_hla_hap2_chainPanTro1</A>
  - Summary info about chain of chimp (panTro1) alignments on human chr6_hla_hap2</LI>
  <LI><A 
  href="#ChainLink">chr6_hla_hap2_chainPanTro1Link</A>
  - Alignment block in chimp (panTro1) chain on human chr6_hla_hap2</LI></LI>
  <LI><A 
  href="#Chain">chr6_hla_hap2_chainRn3</A>
  - Summary info about chain of rat (rn3) alignments on human chr6_hla_hap2</LI>
  <LI><A 
  href="#ChainLink">chr6_hla_hap2_chainRn3</A>
  - Alignment block in rat (rn3) chain on human chr6_hla_hap2</LI></LI>
  <LI><A 
  href="#PslInfo">chr6_hla_hap2_est</A> 
  - EST alignments on human chr6_hla_hap2</LI>
  <LI><A 
  href="#GapInfo">chr6_hla_hap2_gap</A> 
  - Gap locations and types on human chr6_hla_hap2</LI>
  <LI><A 
  href="#FragInfo">chr6_hla_hap2_gl</A> 
  - Golden path positions of all clone fragments on human chr6_hla_hap2</LI>
  <LI><A 
  href="#GpInfo">chr6_hla_hap2_gold</A> 
  - Instructions on creating golden path from fragments on human chr6_hla_hap2</LI>
  <LI><A 
  href="#RepeatInfo">chr6_hla_hap2_rmsk</A> 
  - Repeating elements created using RepeatMasker <on human chr6_hla_hap2</LI>

<A NAME="CHRN_B_GBD"></A>
  <LI><A 
  href="#PslInfo">chrN_blastzBestMouse</A> 
  - <A HREF="http://bio.cse.psu.edu/" TARGET=_BLANK>Blastz</A> alignments 
    of mouse vs. human, filtered to retain only the best match for a given 
    region of the human genome  (Dec 2001 and later)
  <LI><A 
  href="#PslInfo">chrN_blastzCb1</A> 
  - Blastz alignments of May 2003 <em>C. elegans</em> (ce1) vs. Jul 2002 <em>C. briggsae</em> (cb1)
  <LI><A 
  href="#PslInfo">chrN_blastzCe1</A> 
  - Blastz alignments of Jul 2002 <em>C. briggsae</em> (cb1) vs. May 2003 <em>C. elegans</em> (ce1)
  <LI><A 
  href="#PslInfo">chrN_blastzCe2</A> 
  - Blastz alignments of <em>C. briggsae</em> (cb1) vs. <em>C. elegans</em> (ce2)
  <LI><A 
  href="#PslInfo">chrN_blastzMm2</A> 
  - Blastz alignments of Feb. 2002 mouse (mm2) vs. human
  <LI><A 
  href="#PslInfo">chrN_blastzMm3</A> 
  - Blastz alignments of Feb. 2003 mouse (mm3) vs. human  (Dec 2001 and 
    later)
  <LI><A 
  href="#PslInfo">chrN_blastzSelf</A> 
  - Blastz alignments of a genome against itself
  <LI><A 
  href="#PslInfo">chrN_blastzTightMm3</A> 
  - Blastz alignments of mouse (mm3) vs. human, filtered to retain only the 
    best match for a given region of the human genome, and further filtered to 
    high-scoring matches
  <LI><A 
  href="#PslInfo">chrN_blastzTightMm4</A> 
  - Blastz alignments of mouse (mm4) vs. human, filtered to retain only the 
    best match for a given region of the human genome, and further filtered to 
    high-scoring matches
  <LI><A 
  href="#PslInfo">chrN_blastzTightMouse</A> 
  - Blastz alignments 
    of mouse vs. human, filtered to retain only the best match for a given 
    region of the human genome, and further filtered to high-scoring matches
    (June 2002)
  <LI><A 
  href="#PslInfo">chrN_blatFish</A> 
  - Translated BLAT alignments of <em>Tetraodon nigrovirisi</em> against the current genome</LI>
  <LI><A 
  href="#PslInfo">chrN_blatFr1</A> 
  - Translated BLAT alignments of <em>Takifugu rubripes</em> assembly fr1 against the current genome</LI>
  <LI><A 
  href="#PslInfo">chrN_blatFugu</A> 
  - Translated BLAT alignments of <em>Takifugu rubripes</em> against the current genome</LI>
  <LI><A 
  href="#PslInfo">chrN_blatHg16</A> 
  - Translated BLAT alignments of human genome (hg16) vs. current genome</LI>
  <LI><A 
  href="#PslInfo">chrN_blatHuman</A> 
  - Translated BLAT alignments of human genome vs. mouse genome</LI>
  <LI><A 
  href="#PslInfo">chrN_blatMouse</A> 
  - Mouse whole genome shotgun alignments (Aug 2001 and earlier)
  <LI><A 
  href="#PslInfo">chrN_blatMus</A> 
  - Translated BLAT mouse alignments.  (Dec 2001 and later)

<A NAME="CHRN_C_GBD"></A>
  <LI><A 
  href="#Chain">chrN_cb1Chain</A> 
  - Summary info about chain of <em>C. briggsae</em> (cb1) alignments </LI>
  <LI><A 
  href="#ChainLink">chrN_cb1ChainLink</A> 
  - Alignment block in <em>C. briggsae</em> (cb1) chain </LI>
  <LI><A 
  href="#Chain">chrN_ce1Chain</A> 
  - Summary info about chain of <em>C. elegans</em> (ce1 assembly) alignments </LI>
  <LI><A 
  href="#ChainLink">chrN_ce1ChainLink</A> 
  - Alignment block in <em>C. elegans</em> (ce1 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_ce2Chain</A> 
  - Summary info about chain of C. elegans</em> (ce2 assembly) alignments </LI>
  <LI><A 
  href="#Chain">chrN_chainAnoGam1</A> 
  - Summary info about chain of <em>A. gambiae</em> (anoGam1 assembly) alignments </LI>
  <LI><A 
  href="#ChainLink">chrN_chainCanFam1Link</A> 
  - Alignment block in <em>A. gambiae</em> (anoGam1 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_chainCanFam1</A> 
  - Summary info about chain of dog (canFam1 assembly) alignments </LI>
  <LI><A 
  href="#ChainLink">chrN_chainCanFam1Link</A> 
  - Alignment block in dog (canFam1 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_chainCb1</A> 
  - Summary info about chain of <em>C. briggsae</em> (cb1 assembly) alignments</LI>
  <LI><A 
  href="#Chain">chrN_chainCb1Link</A> 
  - Alignment block in <em>C. briggsae</em> (cb1 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_chainCe2Link</A> 
  - Alignment block in <em>C. elegans</em> (ce2 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_chainDanRer1</A> 
  - Summary info about chain of zebrafish (danRer1 assembly) alignments </LI>
  <LI><A 
  href="#ChainLink">chrN_chainDanRer1Link</A> 
  - Alignment block in zebrafish (danRer1 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_chainDm1</A> 
  - Summary info about chain of <em>D. melanogaster</em> (dm1 assembly) alignments</LI>
  <LI><A 
  href="#Chain">chrN_chainDm1Link</A> 
  - Alignment block in <em>D. melanogaster</em> (dm1 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_chainDp1</A> 
  - Summary info about chain of <em>D. pseudoobscura</em> (dp1 assembly) alignments</LI>
  <LI><A 
  href="#Chain">chrN_chainDp1Link</A> 
  - Alignment block in <em>D. pseudoobscura</em> (dp1 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_chainDp2</A> 
  - Summary info about chain of <em>D. pseudoobscura</em> (dp2 assembly) alignments</LI>
  <LI><A 
  href="#Chain">chrN_chainDp2Link</A> 
  - Alignment block in <em>D. pseudoobscura</em> (dp2 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_chainDroYak1</A> 
  - Summary info about chain of <em>D. yakuba</em> (droYak1 assembly) alignments</LI>
  <LI><A 
  href="#Chain">chrN_chainDroYak1</A> 
  - Alignment block in <em>D. yakuba</em> (droYak1 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_chainFr1</A> 
  - Summary info about chain of Fugu (fr1 assembly) alignments</LI>
  <LI><A 
  href="#ChainLink">chrN_chainFr1Link</A> 
  - Alignment block in Fugu chain </LI>
  <LI><A 
  href="#Chain">chrN_chainGalGal2</A> 
  - Summary info about chain of chicken (galGal2 assembly) alignments</LI>
  <LI><A 
  href="#ChainLink">chrN_chainGalGal2Link</A> 
  - Alignment block in chicken (galGal2 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_chainHg16</A> 
  - Summary info about chain of human (hg16 assembly) alignments</LI>
  <LI><A 
  href="#ChainLink">chrN_chainHg16Link</A> 
  - Alignment block in human (hg16 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_chainHg17</A> 
  - Summary info about chain of human (hg17 assembly) alignments</LI>
  <LI><A 
  href="#Chain">chrN_chainHg17Link</A> 
  - Alignment block in human (hg17 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_chainMm3</A> 
  - Summary info about chain of mouse (mm3 assembly) alignments</LI>
  <LI><A 
  href="#ChainLink">chrN_chainMm3Link</A> 
  - Alignment block in mouse (mm3 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_chainMm4</A> 
  - Summary info about chain of mouse (mm4 assembly) alignments</LI>
  <LI><A 
  href="#ChainLink">chrN_chainMm4Link</A> 
  - Alignment block in mouse (mm4 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_chainMm5</A> 
  - Summary info about chain of mouse (mm5 assembly) alignments</LI>
  <LI><A 
  href="#ChainLink">chrN_chainMm5Link</A> 
  - Alignment block in mouse (mm5 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_chainMonDom1</A> 
  - Summary info about chain of opossum (monDom1 assembly) alignments</LI>
  <LI><A 
  href="#ChainLink">chrN_chainMonDom1Link</A> 
  - Summary info about chain of opossum (monDom1 assembly) alignments</LI>
  <LI><A 
  href="#Chain">chrN_chainPanTro1</A> 
  - Summary info about chain of chimp (panTro1 assembly) alignments</LI>
  <LI><A 
  href="#ChainLink">chrN_chainPanTro1Link</A> 
  - Summary info about chain of chimp (panTro1 assembly) alignments</LI>
  <LI><A 
  href="#Chain">chrN_chainPt0</A> 
  - Summary info about chain of chimp (13 Nov 2003 assembly) alignments</LI>
  <LI><A 
  href="#ChainLink">chrN_chainPt0Link</A> 
  - Alignment block in chimp (13 Nov 2003 assembly) chain </LI>
  monDommonDom1><A 
  href="#Chain">chrN_chainRn3</A> 
  - Summary info about chain of rat (rn3 assembly) alignments</LI>
  <LI><A 
  href="#ChainLink">chrN_chainRn3Link</A> 
  - Alignment block in Rat (rn3 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_chainSelf</A> 
  - Summary info about chain of self alignments </LI>
  <LI><A 
  href="#ChainLink">chrN_chainSelfLink</A> 
  - Alignment block in self alignment chain </LI>
  <LI><A 
  href="#Chain">chrN_chainTetNig1</A> 
  - Summary info about chain of Tetraodon (tetNig1 assembly) alignments</LI>
  <LI><A 
  href="#ChainLink">chrN_chainTetNig1Link</A> 
  - Alignment block in Tetraodon (tetNig1 assembly) chain </LI>
  <LI><A 
  href="#Chain">chrN_chainXenTro1</A> 
  - Summary info about chain of <em>X. tropicalis</em> (xenTro1 assembly) alignments</LI>
  <LI><A 
  href="#ChainLink">chrN_chainXenTro1Link</A> 
  - Alignment block in <em>X. tropicalis</em> (xenTro1 assembly) chain </LI>
  <LI><A 
  href="#GapInfo">chrN_contigs_gap</A> 
  - Gap locations and types
  <LI><A 
  href="#GpInfo">chrN_contigs_gold</A> 
  - Instructions on creating goldenPath from fragments

<A NAME="CHRN_E_GBD"></A>
  <LI><A 
  href="#PslInfo">chrN_est</A> 
  - EST alignments 

<A NAME="CHRN_G_GBD"></A>
  <LI><A 
  href="#GapInfo">chrN_gap</A> 
  - Gap locations and types 
  <LI><A 
  href="#FragInfo">chrN_gl</A> 
  - Golden path positions of all clone fragments 
  <LI><A 
  href="#GpInfo">chrN_gold</A> 
  - Instructions on creating golden path from fragments 

<A NAME="CHRN_H_GBD"></A>
  <LI><A 
  href="#WiggleTrack">chrN_humMusL</A> 
  - Human/Mouse Evolutionary Conservation Scores 

<A NAME="CHRN_I_GBD"></I>
  <LI><A 
  href="#PslInfo">chrN_intronEst</A> 
  - EST alignments in introns 

<A NAME="CHRN_M_GBD"></A>
  <LI><A 
  href="#PslInfo">chrN_mrna</A> 
  - mRNA alignments 
  <LI><A 
  href="#WiggleTrack">chrN_musHumL</A> 
  - Mouse/Human Evolutionary Conservation Scores 

<A NAME="CHRN_R_GBD"></A>
  <LI><A 
  href="#Chain">chrN_rBestChainPanTro1</A> 
  - Reciprocal best chain alignments of the panTro1 chimp assembly to the human genome</LI>
  <LI><A 
  href="#RepeatInfo">chrN_rmsk</A> 
  - Repeating elements created using RepeatMasker </LI>

<A NAME="CHRN_V_GBD"></A>
<!--
  <LI><A 
  href="#PslInfo">chrN_viralProt</A> 
  - Alignments between SwissProt viral proteins and the SARS assembly </LI>
-->

<A NAME="CHRN_W_GBD"></A>
  <LI><A 
  href="#WabaChromCreate">chrN_wabaCbr</A> 
  - <em>C. briggsae</em> WABA alignments - abbreviated description

<A NAME="CHRN_Z_GBD"></A>
  <LI><A 
  href="#WiggleTrack">chrN_zoom1_humMusL</A> 
  - Human/Mouse Evolutionary Conservation Scores (very large scale)
  <LI><A 
  href="#WiggleTrack">chrN_zoom1_hg15Mm3L</A> 
  - Human/Mouse Evolutionary Conservation Scores (hg15/mm3) (very large scale)
  <LI><A 
  href="#WiggleTrack">chrN_zoom2500_hg15Mm3L</A> 
  - Human/Mouse Evolutionary Conservation Scores (hg15/mm3) (medium scale)
  <LI><A 
  href="#WiggleTrack">chrN_zoom2500_humMusL</A> 
  - Human/Mouse Evolutionary Conservation Scores (medium scale)
  <LI><A 
  href="#WiggleTrack">chrN_zoom50_hg15Mm3L</A> 
  - Human/Mouse Evolutionary Conservation Scores (hg15/mm3) (large scale)
  <LI><A 
  href="#WiggleTrack">chrN_zoom50_humMusL</A> 
  - Human/Mouse Evolutionary Conservation Scores (large scale)
</UL>
	   </TD><TD WIDTH=15></TD>
  	  </TR>
	</TABLE>
	<BR>
      </TD></TR>
    </TABLE>
    <BR>

<!--Individual Tables---------------------------------------------------->
    <TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1>
      <TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%">
	  <TR><TD>
	    <FONT SIZE="4"><B> &nbsp;Data Stored in a Single Table</B></FONT>
	  </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH>
	  <TR><TD WIDTH=10></TD>
	    <TD>
	      <P>
	      The tables in this section contain data stored in a single table 
	      rather than in one table per chromosome.</P>
	      <CENTER>
	      <A HREF="#A_GBD">[A]</A>&nbsp;
	      <A HREF="#B_GBD">[B]</A>&nbsp;
	      <A HREF="#C_GBD">[C]</A>&nbsp;
	      <A HREF="#D_GBD">[D]</A>&nbsp;
	      <A HREF="#E_GBD">[E]</A>&nbsp;
	      <A HREF="#F_GBD">[F]</A>&nbsp;
	      <A HREF="#G_GBD">[G]</A>&nbsp;
	      <A HREF="#H_GBD">[H]</A>&nbsp;
	      <A HREF="#I_GBD">[I]</A>&nbsp;
	      <A HREF="#J_GBD">[J]</A>&nbsp;
	      <A HREF="#K_GBD">[K]</A>&nbsp;
	      <A HREF="#M_GBD">[M]</A>&nbsp;
	      <A HREF="#N_GBD">[N]</A>&nbsp;
	      <A HREF="#O_GBD">[O]</A>&nbsp;
	      <A HREF="#P_GBD">[P]</A>&nbsp;
	      <A HREF="#Q_GBD">[Q]</A>&nbsp;
	      <A HREF="#R_GBD">[R]</A>&nbsp;
	      <A HREF="#S_GBD">[S]</A>&nbsp;
	      <A HREF="#T_GBD">[T]</A>&nbsp;
	      <A HREF="#U_GBD">[U]</A>&nbsp;
	      <A HREF="#V_GBD">[V]</A>&nbsp;
	      <A HREF="#W_GBD">[W]</A>&nbsp;
	      <A HREF="#X_GBD">[X]</A>&nbsp;
	      <A HREF="#Y_GBD">[Y]</A>&nbsp;
	      <A HREF="#Z_GBD">[Z]</A>&nbsp;
	      </CENTER>

<UL>
<A NAME="A_GBD"></A>
  <LI><A 
  href="#ScoredRef">a_gambiae_myp2a</A> 
  - Pairwise MAF alignment of <em>A. gambiae</em> to <em>D. melanogaster</em>
  <LI><A 
  href="#GenePredictions">acembly</A> 
  - Acembly Gene Predictions With Alt-splicing </LI>
  <LI><A 
  href="#PepPredictions">acemblyPep</A> 
  - Translations of Acembly gene predictions into corresponding amino acid
    sequences </LI>
  <LI><A 
  href="#BED">affy10K</A> 
  - SNPs on the Affymetrix 10K SNP Genotyping Array</LI>
  <LI><A 
  href="#Affy10KDetails">affy10KDetails</A> 
  - Affymetrix 10K SNP sequence details</LI>
  <LI><A 
  href="#PslInfo">affyZebrafish</A> 
  - Sequences used for probe selection on Affymetrix zebrafish chip
  <LI><A 
  href="#ExpRecord">affyExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#BED">affyGeno</A> 
  - SNPs on the Affymetrix 120K SNP Genotyping Array</LI>
  <LI><A 
  href="#AffyGenoDetails">affyGenoDetails</A> 
  - Affymetrix 120K SNP array information</LI>
  <LI><A 
  href="#PslInfo">affyGn1h</A> 
  - Sequences used for probe selection on Affy GNF1H chips</LI>
  <LI><A 
  href="#PslInfo">affyGnf1m</A> 
  - Sequences used for probe selection on Affy GNF1M chips</LI>
  <LI><A 
  href="#BED">affyGnfU74A</A> 
  - Sequences used for probe selection on Affy GNFU74A chips</LI>
  <LI><A 
  href="#BED">affyGnfU74B</A> 
  - Sequences used for probe selection on Affy GNFU74B chips</LI>
  <LI><A 
  href="#BED">affyGnfU74C</A> 
  - Sequences used for probe selection on Affy GNFU74C chips</LI>
  <LI><A 
  href="#PslInfo">affyMOE430</A> 
  - Sequences used for probe selection on Affy MOE430 chips</LI>
  <LI><A 
  href="#PslInfo">affyRAE230</A> 
  - Sequences used for probe selection on Affy RAE230 chips</LI>
  <LI><A 
  href="#PslInfo">affyU34A</A> 
  - Sequences used for probe selection on Affy U34A chips</LI>
  <LI><A 
  href="#PslInfo">affyU74</A> 
  - Sequences used for probe selection on Affy U74 chips</LI>
  <LI><A 
  href="#BED">affyRatio</A> 
  - Expression data from GNF using Affymetrix GeneChips</LI>
  <LI><A 
  href="#AffyTranscriptome">affyTranscriptome</A> 
  - Transcriptome data for chr21 and chr22 from Affymetrix </LI>
  <LI><A 
  href="#PslInfo">affyU133</A> 
  - Consensus and exemplar sequences used for probe selection on Affy HG-U133A and HG-U133B chips</LI>
  <LI><A 
  href="#PslInfo">affyU133Plus2</A> 
  - Consensus and exemplar sequences used for probe selection on the Affy HG-U133 Plus 2.0 chip</LI>
  <LI><A 
  href="#PslInfo">affyU95</A> 
  - Consensus and exemplar sequences used for probe selection on Affy HG-U95Av2 chip</LI>
  <LI><A 
  href="#PslInfo">affyZebrafish</A> 
  - Consensus and exemplar sequences used for probe selection on Affy Zebrafish chip</LI>
  <LI><A 
  href="#PslInfo">all_bacends</A> 
  - BLAT alignment information for BAC clone end sequences on 
    the draft assembly sequence </LI>
  <LI><A 
  href="#PslInfo">all_est</A> 
  - EST alignments (concatenation of chr*_est) 
  <LI><A 
  href="#PslInfo">all_fosends</A> 
- Fosmid end alignments 
  <LI><A 
  href="#PslInfo">all_mrna</A> 
  - mRNA alignments (concatenation of chr*_mrna) 
  <LI><A 
  href="#PslInfo">all_sts_primer</A> 
  - BLAT alignment information for STS markers using primer 
    sequences on the draft sequence assembly </LI>
  <LI><A 
  href="#PslInfo">all_sts_seq</A> 
  - BLAT alignment information for full STS markers sequences 
    on the draft sequence assembly </LI>
  <LI><A 
  href="#AltGraphX">altGraphX</A> 
  - Summary of alt-splicing in EST and mRNA tracks
  <LI><A 
  href="#ScoredRef">anoGam1_050201</A> 
  - Scored reference file for anoGam1 alignments
  <LI><A 
  href="#BED">anophelesEcores</A> 
  - <em>D. melanogaster</em> (dm1)/A. gambiae evolutionary conserved regions (ecores) computed by 
  Exofish</LI>
  <LI><A 
  href="#ScoredRef">apiMel1_050201</A> 
  - Scored reference file for apiMel1 alignments
  <LI><A 
  href="#ExpData">arbFlyLifeAll</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpRecord">arbFlyLifeAllExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">arbFlyLifeAllRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpData">arbFlyLifeMedian</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpRecord">arbFlyLifeMedianExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">arbFlyLifeMedianRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#atlasOncoGene">atlasOncoGene</A> 
  - Link to Atlas Oncology site</LI>
  <LI><A 
  href="#ScoredRef">axtNetCb1</A> 
  - Scored reference file for <em>C. elegans</em>/<em>C. briggsae</em> alignments
  <LI><A 
  href="#ScoredRef">axtNetDp1</A> 
  - Scored reference file for <em>D. melanogaster</em>/<em>D. pseudoobscura</em> alignments
  <LI><A 
  href="#ScoredRef">axtNetHg16</A> 
  - Scored reference file for human/chimp alignments

<A NAME="B_GBD"></A>
  <LI><A 
  href="#LFS">bacEndPairs</A> 
  - Locations of BAC clones using placements of its 5' and 3' 
    end sequences.  BAC end sequence information comes from the dbGSS 
    portion of GenBank </LI>
  <LI><A 
  href="#LFS">bacEndSingles</A> 
  - Locations of BAC clones where only one end is mapped (Zebrafish)
  <LI><A 
  href="#BactigPos">bactigPos</A> 
  - Bactig positions in chromosome coordinates
  <LI><A 
  href="#BlastTab">bdgpBlastTab</A> 
  - Blastp results of known genes vs. one another (<em>D. melanogaster</em>)
  <LI><A 
  href="#KnownCanonical">bdgpCanonical</A> 
  - Describes the canonical splice variant of a gene (Drosophila)
  <LI><A 
  href="#GenePredExt">bdgpGene</A> 
  - Berkeley Drosophila Genome Project genes (<em>D. melanogaster</em>)
  <LI><A 
  href="#GenePredExt">bdgpNonCoding</A> 
  - Berkeley Drosophila Genome Project non-coding genes (<em>D. melanogaster</em>)
  <LI><A 
  href="#KnownIsoforms">bdgpIsoforms</A> 
  - Links together various transcripts of a gene into a cluster (Drosophila)
  <LI><A 
  href="#KnownTo">bdgpToLocusLink</A> 
  - Associates known gene and LocusLink accession (Drosophila)
  <LI><A 
  href="#KnownTo">bdgpToPfam</A> 
  - Associates known gene and Pfam ID (Drosophila)
  <LI><A 
  href="#KnownTo">bdgpToRefSeq</A> 
  - Associates known gene and RefSeq accession (Drosophila)
  <LI><A 
  href="#BED">bgiCov</A> 
  - Areas of chicken genomic coverage of 3 alternate chicken strains sequenced by 
  Beijing Genomics Institute
  <LI><A 
  href="#GenePredictions">bgiGene</A> 
  - Alignments of chicken protein-coding genes annotated by Beijing Genomics Institute
  <LI><A 
  href="#BgiSnp">bgiSnp</A> 
  - Chicken SNP information from Beijing Genomics Institute
  <LI><A 
  href="#BioCycMapDesc">bioCycMapDesc</A> 
  - BioCyc pathway maps
  <LI><A 
  href="#BioCycMapDesc">bioCycPathway</A> 
  - Cross reference between BioCyc Pathway and Known Genes 
  <LI><A 
  href="#PslInfo">blastDm1FB</A> 
  - tBLASTn alignments of peptides from <em>D. melanogaster</em> predicted and
  known genes in FlyBase
  <LI><A 
  href="#PslInfo">blastHg16KG</A> 
  - tBLASTn alignments of peptides from hg16 knownGene predicted and known genes
  <LI><A 
  href="#PepPred">blastHg17KG</A> 
  - tBLASTn alignment of peptides from hg17 knownGene predicted and known genes
  <LI><A 
  href="#PslInfo">blastKGPep0*</A> 
  - tBLASTn alignments peptide sequences
  <LI><A 
  href="#BlastRef">blastKGRef0*</A> 
  - Associates a known gene ID with a gene name, position, and SwissProt name
  for tBLASTn protein tracks
  <LI><A 
  href="#PslInfo">blatChimp</A> 
  - Blat alignments of Pan troglodytes against the draft human genome
  <LI><A 
  href="#PslInfo">blatCioSav1</A> 
  - Blat alignments of <em>Ciona savignyi</em> against the current genome
  <LI><A 
  href="#PslInfo">blatFr1</A> 
  - Translated BLAT alignments of <em>Takifugu rubripes</em> assembly fr1 against the current genome</LI>

<A NAME="C_GBD"></A>
  <LI><A 
  href="#BlastTab">ceBlastTab</A> 
  - Blastp results of known genes vs. one another (<em>C. elegans</em>)
  <LI><A 
  href="#CeleraCoverage">celeraCoverage</A> 
  - Summary of large genomic duplications from Celera data</LI>
  <LI><A 
  href="#CeleraDupPositive">celeraDupPositive</A> 
  - Summary of large genomic duplications from Celera data</LI>
  <LI><A 
  href="#CeleraOverlay">celeraOverlay</A> 
  - Celera assembly overlay in the public NCBI assembly</LI>
  <LI><A 
  href="#CgapAlias">cgapAlias</A> 
  - Associates CGAP pathway IDs with gene symbols or mRNA accessions</LI>
  <LI><A 
  href="#CgapBiocDesc">cgapBiocDesc</A> 
  - CGAP/BioCarta pathway description
  <LI><A 
  href="#CgapBiocPathway">cgapBiocPathway</A> 
  - Associates CGAP pathway IDs with BioCarta pathway IDs
  <LI><A 
  href="#ChainLink">chainLink</A> 
  - Alignment block in chain </LI>
  <LI><A 
  href="#Chain">chainRxBestPt0</A> 
  - Summary info about a chain of chimp alignments
  <LI><A 
  href="#ChainLink">chainRxBestPt0Link</A> 
  - Alignment block in chimp chain </LI>
  <LI><A 
  href="#ScoredRef">chicken_hmrg</A> 
  - Chicken scored reference file for human/chimp/mouse/rat/chicken alignments
  <LI><A 
  href="#ScoredRef">chicken_netBlastz</A> 
  - Scored reference file for chicken alignments
  <LI><A 
  href="#PslInfo">chimpBac</A> 
  - Blat alignments of Pan troglodytes BACs against the draft human genome
  <LI><A 
  href="#PslInfo">chimpBlat</A> 
  - Blat alignments of Pan troglodytes against the draft human genome
  <LI><A 
  href="#BED">chimpDels</A> 
  - Human genome assembly regions deleted in chimp
  <LI><A 
  href="#ScoredRef">chimp_hmrg</A> 
  - Chimp scored reference file for human/chimp/mouse/rat/chicken alignments
  <LI><A 
  href="#ScoredRef">chimp_netBlastz</A> 
  - Scored reference file for chimp alignments
  <LI><A 
  href="#ChimpSimpleDiff">chimpSimpleDiff</A> 
  - Simple differences between the chimp and human genome assemblies
  <LI><A 
  href="#ExpDistance">choExpDistance</A> 
  - Distance between 2 genes in expression space (<em>S. cerevisiae</em>)
  <LI><A 
  href="#ChromInfo">chromInfo</A> 
  - Describes name and size of each chromosome as well as an external file 
  containing the chromosome 
  <LI><A 
  href="#CloneInfo">clonePos</A> 
  - Positions of all clones in golden path. Also HTG phase, 
  finished/draft/predraft stage, and sequence size
  <LI><A 
  href="#Contamination">contamination</A> 
  - Contamination information for current assembly
  <LI><A 
  href="#ContigAcc">contigAcc</A> 
  - Maps a contig to its accession
  <LI><A 
  href="#CpGIsland">cpgIsland</A> 
  - CpG islands 
  <LI><A 
  href="#CpGIslandExt">cpgIslandExt</A> 
  - CpG islands including observed/expected ratio
  <LI><A 
  href="#ContigInfo">ctgPos</A> 
  - Name, size, and golden path position of all contigs
  <LI><A 
  href="#ContigInfo2">ctgPos2</A> 
  - Name, size, golden path position and type of all contigs
  <LI><A 
  href="#ChromosomeBand">cytoBand</A> 
  - 800 chromosome bands mapped to golden path by Terry Furey using
  fish data from BAC Resource Consortium
  <LI><A 
  href="#GenePredictions">cytokines</A> 
  - Cytokine gene annotations (Tetraodon) fish data from BAC Resource 
  Consortium
  <LI><A 
  href="#BED">cytoBandIdeo</A> 
  - Information for mapping ideogram
<A NAME="D_GBD"></A>
  <LI><A 
  href="#ScoredRef">d_pseudoobscura_myp2a</A> 
  - Pairwise MAF alignment of <em>D. pseudoobscura</em> to <em>D. melanogaster</em>
  <LI><A 
  href="#ScoredRef">d_yakuba_myp2a</A> 
  - Pairwise MAF alignment of <em>D. yakuba</em> to <em>D. melanogaster</em>
  <LI><A 
  href="#ScoredRef">danRer1_netBlastz</A> 
  - Scored reference file for zebrafish (danRer1) alignments
  <LI><A 
  href="#ScoredRef">danRer2_netBlastz</A> 
  - Scored reference file for zebrafish (danRer2) alignments
  <LI><A 
  href="#DbSnp">dbSnpRS</A> 
  - Information from dbSNP at the reference SNP level </LI>
  <LI><A 
  href="#DbSnp">dbSnpRsHg</A> 
  - Human genome information from dbSNP at the reference SNP level </LI>
  <LI><A 
  href="#DbSnp">dbSnpRsMm</A> 
  - Mouse genome information from dbSNP at the reference SNP level </LI>
  <LI><A 
  href="#DbSnp">dbSnpRsRn</A> 
  - Rat genome information from dbSNP at the reference SNP level </LI>
  <LI><A 
  href="#PslInfo">dbtssAli</A> 
  - RefSeq mRNA extended to the 5' end from DBTSS (Database of Transcriptional Start Sites)</LI>
  <LI><A 
  href="#BlastTab">dmBlastTab</A> 
  - Blastp results of known genes vs. one another (<em>D. melanogastor</em>)
  <LI><A 
  href="#ScoredRef">dog_netBlastz</A> 
  - Scored reference file for dog alignments
  <LI><A 
  href="#ScoredRef">dp2_050201</A> 
  - Scored reference file for dp2 alignments
  <LI><A 
  href="#BlastTab">drBlastTab</A> 
  - Blastp results of known genes vs. one another (<em>Danio rerio</em>)
  <LI><A 
  href="#ScoredRef">droAna1_050201</A> 
  - Scored reference file for droAna1 alignments
  <LI><A 
  href="#ScoredRef">droMoj1_050201</A> 
  - Scored reference file for droMoj1 alignments
  <LI><A 
  href="#ScoredRef">droVir1_050201</A> 
  - Scored reference file for droVir1 alignments
  <LI><A 
  href="#ScoredRef">droYak1_050201</A> 
  - Scored reference file for droYak1 alignments
  <LI><A 
  href="#DupSpMrna">dupSpMrna</A> 
  - Duplicate mRNA/Protein entries that have identical CDS structures</LI>
<A NAME="E_GBD"></A>
  <LI><A 
  href="#GenePredictions">ECgene</A> 
  - ECGene gene predictions </LI>
  <LI><A 
  href="#PepPredictions">ECgenePep</A> 
  - Translations of ECgene genes into corresponding amino acid sequences </LI>
  <LI><A 
  href="#BED">ecoresFr1</A> 
  - Human/Fugu evolutionary conserved regions (ecores) computed by Exofish</LI>
  <LI><A 
  href="#BED">ecoresHg16</A> 
  - Human (hg16)/Fugu evolutionary conserved regions (ecores) computed by Exofish</LI>
  <LI><A 
  href="#BED">ecoresHg17</A> 
  - Human (hg17)/Fugu evolutionary conserved regions (ecores) computed by Exofish</LI>
  <LI><A 
  href="#BED">ecoresMm3</A> 
  - Mouse (mm3)/Fugu evolutionary conserved regions (ecores) computed by Exofish</LI>
  <LI><A 
  href="#BED">ecoresRn3</A> 
  - Rat (rn3)/Fugu evolutionary conserved regions (ecores) computed by Exofish</LI>
  <LI><A 
  href="#BED">ecoresTetNig1</A> 
  - Tetraodon (tetNig1)/Human evolutionary conserved regions (ecores) computed by Exofish</LI>
  <LI><A 
  href="#BED">ecoresTetraodon</A> 
  - Tetraodon/Human evolutionary conserved regions (ecores) computed by Exofish</LI>
  <LI><A 
  href="#WiggleTrack2">encodeAffyChIpRnapHl50PvalHr[00,02,08,32]</A> 
  - Affymetrix ChIP-chip enrichment of Pol2 from retinoic acid stimulated 
  HL60 cells at 0, 2, 8, or 32 hours, in ENCODE regions (4 different tracks)
  <LI><A 
  href="#BED">encodeAffyChIpRnapHl60SitesHr[00,02,08,32]</A> 
  - Affymetrix ChIP-chip of Pol2 from retinoic acid stimulated HL60 cells at
  0, 2, 8, or 32 hours, in ENCODE regions (4 different tracks)
  <LI><A 
  href="#WiggleTrack2">encodeAffyRnaHl60SigHr[00,02,08,32]</A> 
  - Affymetrix PolyA+ RNA from retinoic acid stimulated HL60 cells at 
  0, 2, 8, or 32 hours, in ENCODE regions (4 different tracks)
  <LI><A 
  href="#BED">encodeAffyRnaHl60SitesHr[00,02,08,32]</A> 
  - Affymetrix PolyA+ RNA abundance from retinoic acid stimulated HL60 cells 
  at 0, 2, 8, or 32 hours, in ENCODE regions (4 different tracks)
  <LI><A 
  href="#EncodeRegionInfo">encodeRegionInfo</A> 
  - Descriptive, assembly-independent information about ENCODE regions</LI>
  <LI><A 
  href="#BED">encodeRegions</A> 
  - Names and positions of selected regions for the ENCODE project</LI>
  <LI><A 
  href="#BED">encode_NHGRI_DNAseHS</A> 
  - DNaseI-hypersensitive sites in ENCODE regions</LI>
  <LI><A 
  href="#BED">encode_Stanford_Promoters</A> 
  - Activity levels of predicted promoters in ENCODE regions (Stanford)</LI>
  <LI><A 
  href="#WiggleTrack2">encode_UCSD_ChIP_RNAP_HCT116</A> 
  - ChIP-chip (Pol II HCT116) in ENCODE regions (UCSD)</LI>
  <LI><A 
  href="#WiggleTrack2">encode_UCSD_ChIP_RNAP_HELA</A> 
  - ChIP-chip (Pol II HeLa) in ENCODE regions (UCSD)</LI>
  <LI><A 
  href="#WiggleTrack2">encode_UCSD_ChIP_RNAP_IMR90</A> 
  - ChIP-chip (Pol II IMR90) in ENCODE regions (UCSD)</LI>
  <LI><A 
  href="#WiggleTrack2">encode_UCSD_ChIP_RNAP_THP1</A> 
  - ChIP-chip (Pol II THP1) in ENCODE regions (UCSD)</LI>
  <LI><A 
  href="#WiggleTrack2">encode_UCSD_ChIP_TAF250_HCT116</A> 
  - ChIP-chip (TAF1 HCT116) in ENCODE regions (UCSD)</LI>
  <LI><A 
  href="#WiggleTrack2">encode_UCSD_ChIP_TAF250_HELA</A> 
  - ChIP-chip (TAF1 HeLa) in ENCODE regions (UCSD)</LI>
  <LI><A 
  href="#WiggleTrack2">encode_UCSD_ChIP_TAF250_IMR90</A> 
  - ChIP-chip (TAF1 IMR90) in ENCODE regions (UCSD)</LI>
  <LI><A 
  href="#WiggleTrack2">encode_UCSD_ChIP_TAF250_THP1</A> 
  - ChIP-chip (TAF1 THP1) in ENCODE regions (UCSD)</LI>
  <LI><A 
  href="#BED">encode_UVA_DNARep_0_2</A> 
  - Temporal profiling of DNA replication (0-2 hrs) in ENCODE regions (U. VA)</LI>
  <LI><A 
  href="#BED">encode_UVA_DNARep_2_4</A> 
  - Temporal profiling of DNA replication (2-4 hrs) in ENCODE regions (U. VA)</LI>
  <LI><A 
  href="#BED">encode_UVA_DNARep_4_6</A> 
  - Temporal profiling of DNA replication (4-6 hrs) in ENCODE regions (U. VA)</LI>
  <LI><A 
  href="#BED">encode_UVA_DNARep_6_8</A> 
  - Temporal profiling of DNA replication (6-8 hrs) in ENCODE regions (U. VA)</LI>
  <LI><A 
  href="#BED">encode_UVA_DNARep_8_10</A> 
  - Temporal profiling of DNA replication (8-10 hrs) in ENCODE regions (U. VA)</LI>
  <LI><A 
  href="#GenePredictions">ensEstGene</A> 
  - Cross-reference info between transcript entries and translation entries of Ensembl genes</LI>
  <LI><A 
  href="#EnsemblXref2">ensemblXref2</A> 
  - Cross-reference info between transcript entries and translation entries of Ensembl genes (alternative table used in some assemblies)</LI>
  <LI><A 
  href="#EnsemblXref3">ensemblXref3</A> 
  - Cross-reference info between transcript entries and translation entries of Ensembl genes (alternative table used in some assemblies)</LI>
  <LI><A 
  href="#GenePredictions">ensEst</A> 
  - Ensembl gene predictions based on ESTs</LI>
  <LI><A 
  href="#GenePredictions">ensEstGene</A> 
  - Ensembl gene predictions based on ESTS</LI>
  <LI><A 
  href="#EnsGtp">ensEstGtp</A> 
  - Associates Ensembl gene, transcript, and protein IDs</LI>
  <LI><A 
  href="#PepPredictions">ensEstPep</A> 
  - Translations of Ensembl EST-based gene predictions into corresponding amino acid sequences </LI>
  <LI><A 
  href="#GenePredictions">ensGene</A> 
  - Ensembl Gene Predictions </LI>
  <LI><A 
  href="#EnsGeneXref">ensGeneXref</A> 
  - Table gene_xref downloaded from Ensembl </LI>
  <LI><A 
  href="#EnsGtp">ensGtp</A> 
  - Associates Ensembl gene, transcript, and protein IDs</LI>
  <LI><A 
  href="#PepPredictions">ensPep</A> 
  - Translations of Ensembl gene predictions into corresponding amino acid sequences </LI>
  <LI><A 
  href="#EnsTranscript">ensTranscript</A> 
  - Table transcript downloaded from Ensembl </LI>
  <LI><A 
  href="#EsRegGeneToMotif">esRegGeneToMotif</A> 
  - Eran Segal regulatory module predicted trancription factor binding sites
  <LI><A 
  href="#OrientInfo">estOrientInfo</A> 
  - Extra information on ESTs calculated by polyInfo program
  <LI><A 
  href="#Exofish">exoFish</A> 
  - Genescope Exofish Tetraodon evolutionarily conserved regions (Last used Aug. 2001)
  <LI><A 
  href="#ExoMouse">exoMouse</A> 
  - Exonerate mouse/human evolutionarily conserved regions (Last used Apr. 2001) </LI>
  <LI><A 
  href="#GenePredExt">exoniphy</A> 
  - Evolutionarily conserved protein-coding exons in multiple, aligned sequences 
  identified by the ExoniPhy program

<A NAME="F_GBD"></A>
  <LI><A 
  href="#FbAllele">fbAllele</A> 
  - Alleles of a FlyBase gene
  <LI><A 
  href="#FbGene">fbGene</A> 
  - Links FlyBase IDs, gene symbols and gene names
  <LI><A 
  href="#FbGo">fbGo</A> 
  - Links FlyBase gene IDs and GO IDs/aspects
  <LI><A 
  href="#fbPhenotype">fbPhenotype</A> 
  - FlyBase observed phenotype in mutant 
  <LI><A 
  href="#FbRef">fbRef</A> 
  - FlyBase literature or database reference
  <LI><A 
  href="#FbRole">fbRole</A> 
  - FlyBase role of gene in wildType
  <LI><A 
  href="#FbSynonym">fbSynonym</A> 
  - Links FlyBase ID to gene synonyms in database
  <LI><A 
  href="#FbTranscript">fbTranscript</A> 
  - Links FlyBase gene IDs and BDGP transcript IDs
  <LI><A 
  href="#FlyBase2004Xref">flyBase2004Xref</A> 
  - Cross reference of FlyBase gene ID, symbolic gene name, synonyms, and accessions (circa late 2004)
  <LI><A 
  href="#GenePredictions">flyBaseGene</A> 
  - FlyBase gene predictions
  <LI><A 
  href="#PepPredictions">flyBasePep</A> 
  - FlyBase genes translated proteins
  <LI><A 
  href="#BED">flyBaseNoncoding</A> 
  - FlyBase genes noncoding RNAs
  <LI><A 
  href="#FlyBaseSwissProt">flyBaseSwissProt</A> 
  - FlyBase acc to SwissProt acc, plus some other SwissProt info
  <LI><A 
  href="#FlyReg">flyreg</A> 
  - FlyReg data (<em>D. melanogaster</em> - dm1)
  <LI><A 
  href="#FlyReg2">flyreg2</A> 
  - FlyReg data (<em>D. melanogaster</em> - dm2)
  <LI><A 
  href="#DnaMotif">flyregMotif</A> 
  - FlyReg gapless motif (<em>D. melanogaster</em>)
  <LI><A 
  href="#BED">firstEF</A> 
  - First Exon Finder (FirstEF) exon and promoter predictions</LI>
  <LI><A 
  href="#FishClones">fishClones</A> 
  - Clones placed on Cytogenetic Map using FISH </LI>
  <LI><A 
  href="#RnaFold">foldUtr3</A> 
  - Info about folding of RNA into secondary structure
  <LI><A 
  href="#RnaFold">foldUtr5</A> 
  - Info about folding of RNA into secondary structure
  <LI><A 
  href="#FosEndPairs">fosEndPairs</A> 
  - Positions of end pairs for fosmids
  <LI><A 
  href="#ScoredRef">fugu_netBlastz</A> 
  - Scored reference file for Fugu alignments

<A NAME="G_GBD"></A>
  <LI><A 
  href="#ScoredRef">galGal2_netBlastz</A> 
  - Scored reference file for chicken alignments
  <LI><A 
  href="#GenePredictions">gaze</A> 
  - Genoscope GAZE gene annotations (Tetraodon)
  <LI><A 
  href="#GbProtAnn">gbProtAnn</A> 
  - GenBank protein annotations
  <LI><A 
  href="#WiggleTrack2">gc5Base</A> 
  - Percentage of G or C bases in a 5-base window
  <LI><A 
  href="#GCpercent">gcPercent</A> 
  - Percentage of G or C bases in a 20,000-base window
  <LI><A 
  href="#GeneBands">geneBands</A> 
  - Band locations of known genes
  <LI><A 
  href="#GenePredictions">geneid</A> 
  - Gene predictions from the geneid program </LI>
  <LI><A 
  href="#PepPredictions">geneidPep</A> 
  - Translations of geneid gene predictions into its corresponding amino acid 
  sequence </LI>
  <LI><A 
  href="#GenePredictions">genieAlt</A> 
  - Genie gene predictions - all constrained by one degree or another by EST or 
  mRNA 
  <LI><A 
  href="#PepPredictions">genieAltPep</A> 
  - Translations of Genie gene predictions into its corresponding amino acid 
  sequence 
  <LI><A 
  href="#GenePredictions">genieKnown</A> 
  - Known genes processed by Genie to allow sequencing error (Oct. 
  2000 and earlier known gene track)
  <LI><A 
  href="#PepPredictions">genieKnownPep</A> 
  - Translations of known genes into its corresponding amino acid sequence (Oct. 
  2000 and earlier known gene track)
  <LI><A 
  href="#GenMapDb">genMapDb</A> 
  - BAC clones from GenMapDB placed on the assembly by U Penn (V. Cheung)</LI>
  <LI><A 
  href="#GenomicSuperDups">genomicSuperDups</A> 
  - Duplications of >1000 Bases Sequence </LI>
  <LI><A 
  href="#GenePredictions">genscan</A> 
  - Genscan Gene Predictions </LI>
  <LI><A 
  href="#PepPredictions">genscanPep</A> 
  - Translations of Genscan gene predictions into corresponding amino acid sequences </LI>
  <LI><A 
  href="#LFS">genscanSubopt</A> 
  - Genscan byproducts: good individual exons out of frame with main predictions (Last used Apr. 2001) </LI>
  <LI><A 
  href="#BED">gnfAtlas2</A> 
  - Expression data from the GNF Gene Expression Atlas 2</LI>
  <LI><A 
  href="#ExpDistance">gnfAtlas2Distance</A> 
  - Describes the distance between genes in the expression space (GNF Atlas 2)
  <LI><A 
  href="#ExpData">gnfHumanAtlas2All</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpRecord">gnfHumanAtlas2AllExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">gnfHumanAtlas2AllRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpData">gnfHumanAtlas2Median</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpRecord">gnfHumanAtlas2MedianExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">gnfHumanAtlas2MedianRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpData">gnfHumanU95All</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpRecord">gnfHumanU95AllExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">gnfHumanU95AllRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpData">gnfHumanU95Median</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpRecord">gnfHumanU95MedianExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">gnfHumanU95MedianRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpData">gnfMouseAtlas2All</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpRecord">gnfMouseAtlas2AllExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">gnfMouseAtlas2AllRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpData">gnfMouseAtlas2Median</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpRecord">gnfMouseAtlas2MedianExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">gnfMouseAtlas2MedianRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpData">gnfMouseU74aAll</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpRecord">gnfMouseU74aAllExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">gnfMouseU74aAllRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpData">gnfMouseU74aMedian</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpRecord">gnfMouseU74aMedianExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">gnfMouseU74aMedianRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpData">gnfMouseU74bAll</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpRecord">gnfMouseU74bAllExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">gnfMouseU74bAllRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpData">gnfMouseU74bMedian</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpRecord">gnfMouseU74bMedianExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">gnfMouseU74bMedianRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpData">gnfMouseU74cAll</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpRecord">gnfMouseU74cAllExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">gnfMouseU74cAllRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpData">gnfMouseU74cMedian</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpRecord">gnfMouseU74cMedianExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">gnfMouseU74cMedianRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpData">gnfRatAtlas2All</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpRecord">gnfRatAtlas2AllExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">gnfRatAtlas2AllRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpData">gnfRatAtlas2Median</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpRecord">gnfRatAtlas2MedianExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">gnfRatAtlas2MedianRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpDistance">gnfU95Distance</A> 
  - Describes the distance between genes in the expression space (GNF U95)

<A NAME="H_GBD"></A>
  <LI><A 
  href="#HInv">HInv</A> 
  - Full-length cDNAs from the H-Invitational Human Gene Annotation Database</LI>
  <LI><A 
  href="#PslInfo">HInvGeneMrna</A> 
  - H-Invitational genes mRNA data </LI>
  <LI><A 
  href="#BED">haplotype</A> 
  - Common Haplotype Blocks - U. of Oxford and Sanger Institute (chr22 only) </LI>
  <LI><A 
  href="#ScoredRef">hg17_netBlastz</A> 
  - Scored reference file for human hg17 alignments
  <LI><A 
  href="#HgFindSpec">hgFindSpec</A> 
  - Information that defines a search performed by hgFind binary</LI>
  <LI><A 
  href="#GenePredictions">hoxGenes</A> 
  - HOX gene mRNAs</LI>
  <LI><A 
  href="#Hugo">hugo</A> 
  - Cross reference between hugo and other databases
  <LI><A 
  href="#Chain">humanChain</A> 
  - Summary info about chain of human alignments </LI>
  <LI><A 
  href="#ScoredRef">human_chm</A> 
  - Pairwise multiple alignments between current species and human</LI>
  <LI><A 
  href="#BED">humanDels</A> 
  - Human deletions in chimp genome</LI>
  <LI><A 
  href="#NetAlign">humanNet</A> 
  - Database representation of a net of alignments with human </LI>
  <LI><A 
  href="#ScoredRef">humorMm3Rn3</A> 
  - Human, Mouse, and Rat multiz alignments for hg16, mm3, and rn3 (human only) </LI>

<A NAME="I_GBD"></A>
  <LI><A 
  href="#ImageClone">imageClone</A> 
  - Used with image consortium's cumulative_plate files at: ftp://image.llnl.gov/image/outgoing"</LI>
  <LI><A 
  href="#InterProXref">interProXref</A> 
  - Associates InterPro ID with Swiss-Prot accessions

<A NAME="J_GBD"></A>
  <LI><A 
  href="#JaxOrtholog">jaxOrtholog</A> 
  - Jackson Lab Mouse Orthologs </LI>
  <LI><A 
  href="#JaxQTL">jaxQTL</A> 
  - Jackson Lab Mouse Quantitative Trait Loci </LI>
  <LI><A 
  href="#GenePredExt">jgiGene</A> 
  - Alignments of JGI predicted transcripts to the current genome 

<A NAME="K_GBD"></A>
  <LI><A 
  href="#KeggMapDesc">keggMapDesc</A> 
  - Description of KEGG pathway map
  <LI><A 
  href="#KeggPathway">keggPathway</A> 
  - Associates KEGG pathway IDs with Known Genes and LocusLink IDs
  <LI><A 
  href="#KgAlias">kgAlias</A> 
  - Links together a Known Gene ID and a gene alias
  <LI><A 
  href="#KgProtAlias">kgProtAlias</A> 
  - Links together a Known Gene ID and a protein alias
  <LI><A 
  href="#PslInfo">kgProtMap</A> 
  - Known Genes protein mapping
  <LI><A 
  href="#KgSpAlias">kgSpAlias</A> 
  - Links together a Known Gene ID, Swiss-Prot/TrEMBL accession and an alias
  <LI><A 
  href="#KgXref">kgXref</A> 
  - Links together a Known Gene ID and mRNA, SwissProt, RefSeq, and NCBI accessions/IDs
  <LI><A 
  href="#ExpRecord">kimWormLifeAllExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">kimWormLifeAllRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#ExpRecord">kimWormLifeMedianExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">kimWormLifeMedianRatio</A> 
  - Expression data</LI>
  <LI><A 
  href="#BlastTab">knownBlastTab</A> 
  - Blastp results of known genes vs. one another (human)
  <LI><A 
  href="#KnownCanonical">knownCanonical</A> 
  - Describes the canonical splice variant of a gene
  <LI><A 
  href="#ExpDistance">knownExpDistance</A> 
  - Describes the distance between genes in the expression space (UCLA)
  <LI><A 
  href="#KnownGene">knownGene</A> 
  - Protein coding genes based on proteins from SWISS-PROT, TrEMBL, and 
TrEMBL-NEW and their corresponding mRNAs from GenBank
  <LI><A 
  href="#KnownGeneLink">knownGeneLink</A> 
  - Stores the seqType info for Known Genes track entries.
  <LI><A 
  href="#MrnaPredictions">knownGeneMrna</A> 
  - Known Genes mRNA
  <LI><A 
  href="#PepPredictions">knownGenePep</A> 
  - Known Genes translated proteins
  <LI><A 
  href="#OtherKnown">knownInfo</A> 
  - Other information about known genes (Oct. 2000 and earlier known gene track) 
  <LI><A 
  href="#KnownIsoforms">knownIsoforms</A> 
  - Links together various transcripts of a gene into a cluster
  <LI><A 
  href="#MoreKnown">knownMore</A> 
  - Links to other databases: GenBank, OMIN, HUGO, RefSeq, GBD
  (Oct. 2000 and earlier known (RefSeq) gene track) 
  <LI><A 
  href="#KnownTo">knownToCdsSnp</A> 
  - Associates known gene and coding SNP IDs
  <LI><A 
  href="#KnownTo">knownToEnsembl</A> 
  - Associates known gene and Ensembl transcripts
  <LI><A 
  href="#KnownTo">knownToGnfAtlas2</A> 
  - Associates known gene and Atlas2 expression data
  <LI><A 
  href="#KnownTo">knownToGnf1h</A> 
  - Associates known gene and GNF1H expression data
  <LI><A 
  href="#KnownTo">knownToHInv</A> 
  - Associates known gene and HInv transcripts
  <LI><A 
  href="#KnownTo">knownToLocusLink</A> 
  - Associates known gene and LocusLink ID
  <LI><A 
  href="#KnownTo">knownToMOE430</A> 
  - Associates known gene and Affy MOE430 expression data
  <LI><A 
  href="#KnownTo">knownToMOE430A</A> 
  - Associates known gene and Affy MOE430A expression data
  <LI><A 
  href="#KnownTo">knownToPfam</A> 
  - Associates known gene and Pfam ID
  <LI><A 
  href="#KnownTo">knownToRefSeq</A> 
  - Associates known gene and RefSeq accession
  <LI><A 
  href="#KnownToSuper">knownToSuper</A> 
  - Associates known gene and Superfamily ID
  <LI><A 
  href="#KnownTo">knownToU133</A> 
  - Associates known gene and Affy U133 chip ID
  <LI><A 
  href="#KnownTo">knownToU133Plus2</A> 
  - Associates known gene and Affy U133 Plus 2.0 chip ID
  <LI><A 
  href="#KnownTo">knownToU133</A> 
  - Associates known gene and Affy U133 chip ID
  <LI><A 
  href="#KnownTo">knownToU74</A> 
  - Associates known gene and Affy U74 (mouse) chip ID
  <LI><A 
  href="#KnownTo">knownToU95</A> 
  - Associates known gene and Affy U95 chip ID

<A NAME="M_GBD"></A>
  <LI><A 
  href="#PslInfo">mgcFullMrna</A> 
  - Mammalian Gene Collection (MGC) full ORF mRNAs
  <LI><A 
  href="#MGCFullStatus">mgcFullStatus</A> 
  - Status of MGC clones
  <LI><A 
  href="#GenePredictions">mgcGenes</A> 
  - MGC full ORF mRNAs
  <LI><A 
  href="#MGIid">MGI ID</A> 
  - Links MGI ID to Locus Link ID (Mouse)
  <LI><A 
  href="#BED">miRNA</A> 
  - MicroRNAs from the miRNA Registry at the WellCome Trust Sanger Institute.
  <LI><A 
  href="#ScoredRef">mm5_netBlastz</A> 
  - Scored reference file for mouse mm5 alignments
  <LI><A 
  href="#BlastTab">mmBlastTab</A> 
  - Blastp results of known genes vs. one another (mouse)
  <LI><A 
  href="#BED">mostConserved</A> 
  - BED files of most conserved data.
  <LI><A 
  href="#Chain">mouseChain</A> 
  - Summary info about chain of mouse alignments </LI>
  <LI><A 
  href="#ScoredRef">mouse_chm</A> 
  - Pairwise multiple alignments between current species and mouse</LI>
  <LI><A 
  href="#ChromInfo">mouseChrom</A> 
  - Describes name and size of each mouse chromosome as well as an external file 
  containing the chromosome 
  <LI><A 
  href="#ScoredRef">mouse_hmrg</A> 
  - Mouse scored reference file for human/chimp/mouse/rat/chicken alignments
  <LI><A 
  href="#NetAlign">mouseNet</A> 
  - Database representation of a net of alignments with mouse </LI>
  <LI><A 
  href="#ScoredRef">mouse_netBlastz</A> 
  - Scored reference file for mouse alignments
  <LI><A 
  href="#ExpData">mouseRinnSex</A> 
  - Rinn et al. sex gene expression data on MOE430A chip
  <LI><A 
  href="#ExpRecord">mouseRinnSexExps</A> 
  - Rinn et al. sex gene experiment descriptive data on MOE430A chip
  <LI><A 
  href="#ExpData">mouseRinnSexMedian</A> 
  - Rinn et al. sex gene expression data on MOE430A chip
  <LI><A 
  href="#ExpRecord">mouseRinnSexMedianExps</A> 
  - Rinn et al. sex gene experiment descriptive data on MOE430A chip
  <LI><A 
  href="#ExpData">mouseRinnSexMedianRatio</A> 
  - Rinn et al. sex gene expression data on MOE430A chip
  <LI><A 
  href="#ExpData">mouseRinnSexRatio</A> 
  - Rinn et al. sex gene expression data on MOE430A chip
  <LI><A 
  href="#MouseSyn">mouseSyn</A> 
  - Synteny between mouse and human chromosomes (Last used Apr. 2001) </LI>
  <LI><A 
  href="#MouseSynWhd">mouseSynWhd</A> 
  - Whitehead synteny between mouse and human chromosomes </LI>
  <LI><A 
  href="#OrientInfo">mrnaOrientInfo</A> 
  - Extra information on mRNAs calcluated by polyInfo program
  <LI><A 
  href="#MrnaRefSeq">mrnaRefseq</A> 
  - Associates mRNA IDs with RefSeq IDs
  <LI><A 
  href="#ScoredRef">multiz5way</A> 
  - Index of multiz multiple alignments of 5 organisms.
  NOTE: To obtain the conservation scores associated with this
  assembly, download the data from the assembly's <em>phastCons</em> directory
  on the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/" 
  TARGET=_blank>FTP server</A>.
  <LI><A 
  href="#ScoredRef">multiz8way</A> 
  - Index of multiz multiple alignments of 8 organisms. 
  NOTE: To obtain the conservation scores associated with this
  assembly, download the data from the assembly's <em>phastCons</em> directory
  on the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/" 
  TARGET=_blank>FTP server</A>.
  <LI><A 
  href="#WiggleTrack2">multiz8wayCons</A> 
  - "Wiggle track" display of evolutionary conservation in 8 organisms 
  based on a phylogenetic hidden Markov model, using multiz alignments. 
  NOTE: To obtain the conservation scores associated with this
  assembly, download the data from the assembly's <em>phastCons</em> directory
  on the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/" 
  TARGET=_blank>FTP server</A>.
  <LI><A 
  href="#ScoredRef">mzPt1Mm3Rn3Gg2_pHMM</A> 
  - Index of multiz multiple alignments of human, chimp, mouse, rat and
  chicken.
  NOTE: To obtain the conservation scores associated with this
  assembly, download the data from the assembly's <em>phastCons</em> directory
  on the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/" 
  TARGET=_blank>FTP server</A>.
  <LI><A 
  href="#WiggleTrack2">mzPt1Mm3Rn3Gg2_pHMM_wig</A> 
  - "Wiggle track" display of evolutionary conservation in human, chimp, 
  mouse, rat, and chicken based on a phylogenetic hidden Markov model, using 
  multiz alignments. 
  NOTE: To obtain the conservation scores associated with this
  assembly, download the data from the assembly's <em>phastCons</em> directory
  on the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/" 
  TARGET=_blank>FTP server</A>.
  <LI><A 
  href="#ScoredRef">multizHg15Mm3</A> 
  - <em>Multiz</em> multiple alignments between human (hg15) and mouse (mm3) </LI>
  <LI><A 
  href="#ScoredRef">multizMm3Rn2</A> 
  - <em>Multiz</em> multiple alignments between mouse (mm3) and rat (rn2) </LI>
  <LI><A 
  href="#ScoredRef">multizYeast</A> 
  - <em>Multiz</em> multiple alignments of 7 yeast species against S. cerevisiae (sc1) </LI>
  <LI><A 
  href="#ScoredRef">mzDy1Dp2Ag1_phast</A> 
  - Index of multiz multiple alignments of <em>D. melanogaster</em>, 
  <em>D. pseudoobscura</em>, and <em>A. gambiae</em>
  NOTE: To obtain the conservation scores associated with this
  assembly, download the data from the assembly's <em>phastCons</em> directory
  on the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/" 
  TARGET=_blank>FTP server</A>.
  <LI><A 
  href="#WiggleTrack2">mzDy1Dp2Ag1_phast_wig</A> 
  - "Wiggle track" display of evolutionary conservation in 
  <em>D. melanogaster</em>, <em>D. pseudoobscura</em>, and <em>A. gambiae</em>
  based on a phylogenetic hidden Markov model, using multiz alignments. 
  NOTE: To obtain the conservation scores associated with this
  assembly, download the data from the assembly's <em>phastCons</em> directory
  on the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/" 
  TARGET=_blank>FTP server</A>.
  <LI><A 
  href="#ScoredRef">mzHg17Mm5_phast</A> 
  - Evolutionary conservation in current species, human, and mouse based on a phylogenetic 
  hidden Markov model, using multiz alignments. 
  NOTE: To obtain the conservation scores associated with this
  assembly, download the data from the assembly's <em>phastCons</em> directory
  on the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/" 
  TARGET=_blank>FTP server</A>.
  <LI><A 
  href="#WiggleTrack2">mzPt1Mm3Rn3Gg2_pHMM_wig</A> 
  - "Wiggle track" display of evolutionary conservation in current species, human, and mouse
  based on a phylogenetic hidden Markov model, using multiz alignments. 
  NOTE: To obtain the conservation scores associated with this
  assembly, download the data from the assembly's <em>phastCons</em> directory
  on the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/" 
  TARGET=_blank>FTP server</A>.

<A NAME="N_GBD"></A>
  <LI><A 
  href="#BED">nci60</A> 
  - Microarray Experiments for NCI 60 Cell Lines </LI>
  <LI><A 
  href="#ExpRecord">nci60Exps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#NetAlign">netAnoGam1</A> 
  - Database representation of a net of alignments with <em>A. gambiae</em>
  anoGam1 assembly </LI>
  <LI><A 
  href="#NetAlign">netCanFam1</A> 
  - Database representation of a net of alignments with dog assembly canFam1 </LI>
  <LI><A 
  href="#NetAlign">netCb1</A> 
  - Database representation of a net of alignments with <em>C. briggsae</em> assembly cb1 </LI>
  <LI><A 
  href="#NetAlign">netDanRer1</A> 
  - Database representation of a net of alignments with zebrafish assembly danRer1 </LI>
  <LI><A 
  href="#NetAlign">netDanRer2</A> 
  - Database representation of a net of alignments with zebrafish assembly danRer2 </LI>
  <LI><A 
  href="#NetAlign">netDm1</A> 
  - Database representation of a net of alignments with <em>D. melanogaster</em> assembly dm1 </LI>
  <LI><A 
  href="#NetAlign">netDp1</A> 
  - Database representation of a net of alignments with <em>D. pseudoobscura</em> assembly dp1 </LI>
  <LI><A 
  href="#NetAlign">netDp2</A> 
  - Database representation of a net of alignments with <em>D. pseudoobscura</em> assembly dp2 </LI>
  <LI><A 
  href="#NetAlign">netDroYak1</A> 
  - Database representation of a net of alignments with <em>D. yakuba</em> assembly droYak1 </LI>
  <LI><A 
  href="#NetAlign">netFr1</A> 
  - Database representation of a net of alignments with Fugu assembly fr1 </LI>
  <LI><A 
  href="#NetAlign">netHg15</A> 
  - Database representation of a net of alignments with Huamn assembly hg15 </LI>
  <LI><A 
  href="#NetAlign">netHg16</A> 
  - Database representation of a net of alignments with Huamn assembly hg16 </LI>
  <LI><A 
  href="#NetAlign">netHg17</A> 
  - Database representation of a net of alignments with Huamn assembly hg17 </LI>
  <LI><A 
  href="#NetAlign">netGalGal2</A> 
  - Database representation of a net of alignments with chicken assembly galGal2 </LI>
  <LI><A 
  href="#NetAlign">netMm3</A> 
  - Database representation of a net of alignments with mouse assembly mm3 </LI>
  <LI><A 
  href="#NetAlign">netMm4</A> 
  - Database representation of a net of alignments with mouse assembly mm4 </LI>
  <LI><A 
  href="#NetAlign">netMm5</A> 
  - Database representation of a net of alignments with mouse assembly mm5 </LI>
  <LI><A 
  href="#NetAlign">netMonDom1</A> 
  - Database representation of a net of alignments with opossum assembly monDom1 </LI>
  <LI><A 
  href="#NetAlign">netPanTro1</A> 
  - Database representation of a net of alignments with chimp assembly panTro1 </LI>
  <LI><A 
  href="#NetAlign">netRn3</A> 
  - Database representation of a net of alignments with rat assembly rn3 </LI>
  <LI><A 
  href="#NetAlign">netRxBestPt0</A> 
  - Database representation of a net of alignments with chimp assembly pt0 </LI>
  <LI><A 
  href="#NetAlign">netSelf</A> 
  - Database representation of a net of alignments of assembly with itself</LI>
  <LI><A 
  href="#NetAlign">netSyntenyDm1</A> 
  - Database representation of a net of alignments with <em>D. melanogaster</em> assembly dm1 </LI>
  <LI><A 
  href="#NetAlign">netSyntenyHg16</A> 
  - Database representation of a net of alignments with human assembly hg16 </LI>
  <LI><A 
  href="#NetAlign">netTetNig1</A> 
  - Database representation of a net of alignments with Tetraodon assembly tetNig1 </LI>
  <LI><A 
  href="#NetAlign">netXenTro1</A> 
  - Database representation of a net of alignments with <em>X. tropicalis</em> assembly xenTro1 </LI>
  <LI><A 
  href="#PslInfo">NIAGene</A> 
  - Alignments of the NIA Mouse Gene Index (v.3) with the selected mouse assembly</LI>

<A NAME="O_GBD"></A>
<!--
  <LI><A 
  href="#PslInfo">otherSARS</A> 
  - Other SARS sequences from NCBI aligned to the SARS assembly
-->

<A NAME="P_GBD"></A>
  <LI><A 
  href="#PbAaDistX">pbAaDistX</A> 
  - Proteome Browser distribution for a specific amino acid X
  <LI><A 
  href="#PbAnomLimit">pbAnomLimit</A> 
  - Proteome Browser amino acid anomaly limits for each AA
  <LI><A 
  href="#PbResAvgStd">pbResAvgStd</A> 
  - Proteome Browser residue average and standard deviation
  <LI><A 
  href="#PbStamp">pbStamp</A> 
  - Proteome Browser stamp information
  <LI><A 
  href="#PdbSP">pdbSP</A> 
  - Links a Swiss-Prot accession ID with a PDB ID
  <LI><A 
  href="#PepCCntDist">pepCCntDist</A> 
  - Proteome Browser cysteine count distribution
  <LI><A 
  href="#PepExonCntDist">pepExonCntDist</A> 
  - Proteome Browser exon count distribution
  <LI><A 
  href="#PepHydroDist">pepHydroDist</A> 
  - Proteome Browser hydrophobicity distribution
  <LI><A 
  href="#PepIPCntDist">pepIPCntDist</A> 
  - Proteome Browser InterProt domain count distribution
  <LI><A 
  href="#PepMolWtDist">pepMolWtDist</A> 
  - Proteome Browser molecular weight distribution
  <LI><A 
  href="#PepMwAa">pepMwAa</A> 
  - Proteome Browser molecular weight and AA length of proteins
  <LI><A 
  href="#PepPi">pepPi</A> 
  - Proteome Browser protein pIs 
  <LI><A 
  href="#PepPiDist">pepPiDist</A> 
  - Proteome Browser pI distribution
  <LI><A 
  href="#PepPred">pepPred</A> 
  - Proteome Browser predicted peptide linked to predicted gene
  <LI><A 
  href="#PepResDist">pepResDist</A> 
  - Proteome Browser residue distribution
  <LI><A 
  href="#BED">perlegen</A> 
  - Perlegen Common High-Resolution Haplotype Blocks (chr21 only) </LI>
  <LI><A 
  href="#PfamDesc">pfamDesc</A> 
  - Description of Pfam table
  <LI><A 
  href="#PfamXref">pfamXref</A> 
  - Links Pfam and Swiss-Prot IDs
  <LI><A 
  href="#WiggleTrack">pGC</A> 
  - Percentage of bases that are G or C in small regions across the genome 
  </LI>
  <LI><A 
  href="#WiggleTrack2">phastCons</A> 
  - "Wiggle track" display of evolutionary conservation based on a 
  phylogenetic hidden Markov model, using multiz alignments 
  <LI><A 
  href="#WiggleTrack2">phastCons5</A> 
  - "Wiggle track" display of evolutionary conservation in 4 organisms based 
  on a phylogenetic hidden Markov model, using multiz alignments 
  <LI><A 
  href="#WiggleTrack2">phastCons8way</A> 
  - "Wiggle track" display of evolutionary conservation in 8 organisms based 
  on a phylogenetic hidden Markov model, using multiz alignments 
  <LI><A 
  href="#BED">phastConsElements</A> 
  - Predictions of conserved elements produced by phastCons program
  <LI><A 
  href="#BED">phastConsElements8way</A> 
  - Predictions of conserved elements in 8 organisms produced by phastCons 
  program
  <LI><A 
  href="#LFS">phMouse</A> 
  - Pattern Hunter Mouse/Human Evolutionarily Conserved Regions (Last used Apr. 2001) </LI>
  <LI><A 
  href="#BED">pjt_gene</A> 
  - NISC-generated gene annotations for zoo</LI>
  <LI><A 
  href="#ProtHomolog">protHomolog</A> 
  - SAM homolog data
  <LI><A 
  href="#Pscreen">pscreen</A> 
  - P-screen (BDGP Gene Disruption Project) P element insertion locations 
  and genes
  <LI><A 
  href="#PseudoGeneLink">pseudoGeneLink</A> 
  - Links between predicted gene/pseudogene and ortholog/paralog information</LI>
  <LI><A 
  href="#PslInfo">pseudoMrna</A> 
  - Predicted pseudogene alignment information</LI>
  <LI><A 
  href="#GenePredictions">pseudoYale</A> 
  - Alignments of pseudogenes identified in the Yale Pseudogene Database

<A NAME="Q_GBD"></A>
  <LI><A 
  href="#WiggleTrack2">quality</A> 
  - Genome quality scores</LI>

<A NAME="R_GBD"></A>
  <LI><A 
  href="#RankProp">rankProp</A> 
  - RankProp protein ranking for a pair of proteins
  <LI><A 
  href="#ScoredRef">rat_hmrg</A> 
  - Rat scored reference file for human/chimp/mouse/rat/chicken alignments
  <LI><A 
  href="#ScoredRef">rat_netBlastz</A> 
  - Scored reference file for rat alignments
  <LI><A 
  href="#NetAlign">rBestNetPanTro1</A> 
  - Database representation of a reciprocal best net of alignments with chimp assembly panTro1 </LI>
  <LI><A 
  href="#RecombRate">recombRate</A> 
  - Describes the recombination rate in 1Mb intervals based on deCODE, Genethon and Marshfield maps</LI>
  <LI><A 
  href="#BED">recombRateRat</A> 
  - Describes rat recombination rate  based on SHRSPxBN F2 intercross or FHHxACI F2 intercross genetic maps.</LI>
  <LI><A 
  href="#RefFlat">refFlat</A> 
  - RefSeq genes, with additional name (as it appears in Genome Browser) field  </LI>
  <LI><A 
  href="#GenePredictions">refGene</A> 
  - RefSeq genes </LI>
  <LI><A 
  href="#GenePredictions">refKnown</A> 
  - Known genes as derived from alignments of RefSeq mRNA (Dec 2000 and later 
  known (RefSeq) gene track) 
  <LI><A 
  href="#RefLink">refLink</A> 
  - Relates RefSeq mRNA accession to LocusLink ID, HUGO Gene Nomenclature Committee  symbol, etc. (Dec. 2000 and later) 
  <LI><A 
  href="#MrnaPredictions">refMrna</A> 
  - RefSeq mRNA (Dec. 2000 and later, not present in incrementally-updated assemblies) 
  <LI><A 
  href="#PepPredictions">refPep</A> 
  - RefSeq translated proteins (Dec. 2000 and later, not present in incrementally-updated assemblies) 
  <LI><A 
  href="#PslInfo">refSeqAli</A> 
  - RefSeq translated proteins (Dec. 2001 and later) 
  <LI><A 
  href="#RefSeqStatus">refSeqStatus</A> 
  - Associates RefSeq mRNA accessions with corresponding RefSeq status (Dec. 2001 and later) 
  <LI><A 
  href="#RefSeqSummary">refSeqSummary</A> 
  - Sparse summary table created only for refSeqs that have "Summary:" or "COMPLETENESS:" in comment
  <LI><A 
  href="#WiggleTrack2">regPotential2X</A> 
  - 2-way regulatory potential score computed from human/mouse alignments
  <LI><A 
  href="#WiggleTrack2">regPotential3X</A> 
  - 3-way regulatory potential score computed from human/mouse/rat alignments
  <LI><A 
  href="#PslInfo">rgdEst</A> 
  - Rat Genome Database ESTs </LI>
  <LI><A 
  href="#GenePredictions">rgdGene</A> 
  - Rat Genome Database genes </LI>
  <LI><A 
  href="#BED">rgdQtl</A> 
  - Rat Genome Database QTLs </LI>
  <LI><A 
  href="#BED">rgdSslp</A> 
  - Rat Genome Database simple sequence length polymorphisms (SSLPs) </LI>
  <LI><A 
  href="#PslInfo">rhMap</A> 
  - Alignments of radiation hybrid map sequences to the assembly</LI>
  <LI><A 
  href="#RikenAltid">rikenaltid</A> 
  - Riken altid table (Mouse Feb. 2002 and later)</LI>
  <LI><A 
  href="#RikenAnn">rikenann</A> 
  - Riken annotation table (Mouse Feb. 2002 and later)</LI>
  <LI><A 
  href="#RikenCluster">rikencluster</A> 
  - Riken cluster table (Mouse Feb. 2002 and later)</LI> 
  <LI><A 
  href="#RikenClusterSeq">rikenclusterseq</A> 
  - Riken cluster sequence regulation table (Mouse Feb. 2002 and later)</LI>
  <LI><A 
  href="#RikenSeq">rikenseq</A> 
  - Riken sequence table (Mouse Feb. 2002 and later)</LI>
  <LI><A 
  href="#BED">rinnSex</A> 
  - Rinn et al. sex gene expression data on MOE430A chip
  <LI><A 
  href="#RNAcluster">rnaCluster</A> 
  - Gene boundaries deduced from clustering spliced ESTs and mRNAs against the genome (Apr. 2002 and later) </LI>
  <LI><A 
  href="#RNAgenes">rnaGene</A> 
  - Functional RNA genes (Last used Apr. 2001) </LI>
  <LI><A 
  href="#BED">rosetta</A> 
  - Rosetta Experimental Confirmation of Chr22 Exons </LI>
  <LI><A 
  href="#ExpRecord">rosettaExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#RosettaExps">rosettaExps</A> 
  - Rosetta Experimental Confirmation information</LI>
  <LI><A 
  href="#Sage">sage</A> 
  - Stores SAGE data in terms of UniGene identifiers</LI>

<A NAME="S_GBD"></A>
  <LI><A 
  href="#ScoredRef">sacBay_pwMaf</A> 
  - Scored reference file for <em>S. bayanus</em> alignments
  <LI><A 
  href="#ScoredRef">sacCas_pwMaf</A> 
  - Scored reference file for <em>S. castelli</em> alignments
  <LI><A 
  href="#ScoredRef">sacKlu_pwMaf</A> 
  - Scored reference file for <em>S. kluyveri</em> alignments
  <LI><A 
  href="#ScoredRef">sacKud_pwMaf</A> 
  - Scored reference file for <em>S. kudriavzevii</em> alignments
  <LI><A 
  href="#ScoredRef">sacKud_pwMik</A> 
  - Scored reference file for <em>S. mikatae</em> alignments
  <LI><A 
  href="#ScoredRef">sacKud_pwPar</A> 
  - Scored reference file for <em>S. paradoxus</em> alignments
  <LI><A 
  href="#SageExp">sageExp</A> 
  - Data related to SAGE experiments, tissue descriptions, etc.
  <LI><A 
  href="#SamSubdir">samSubdir</A> 
  - Associates SAM data subdirectory with Known Gene ID
  <LI><A 
  href="#GenePredictions">sanger20</A> 
  - Sanger Institute Chromosome 20 Genes </LI>
  <LI><A 
  href="#SangerExtra">sanger20extra</A> 
  - Additional information about a Sanger 20 or 22 gene </LI>
  <LI><A 
  href="#GenePredictions">sanger22</A> 
  - Sanger Institute Chromosome 22 Genes </LI>
  <LI><A 
  href="#SangerExtra">sanger22extra</A> 
  - Additional information about a Sanger 20 or 22 gene </LI>
  <LI><A 
  href="#PepPredictions">sanger22pep</A> 
  - Translations of Sanger22 gene predictions into corresponding amino acid
    sequences </LI>
  <LI><A 
  href="#GenePredictions">sanger22pseudo</A> 
  - Sanger Institute Chromosome 22 Pseudogenes </LI>
  <LI><A 
  href="#BlastTab">sangerBlastTab</A> 
  - Blastp results of Sanger genes vs. one another (<em>C. elegans</em>)
  <LI><A 
  href="#KnownCanonical">sangerCanonical</A> 
  - Describes the canonical splice variant of a Sanger gene (<em>C. elegans</em>)
  <LI><A 
  href="#SangerGene">sangerGene</A> 
  - Sanger gene predictions, containing protein ID (<em>C. elegans</em>)
  <LI><A 
  href="#GenePredExt">sangerGenefinder</A> 
  - Sanger gene predictions, containing protein ID (<em>C. elegans</em>)
  <LI><A 
  href="#KnownIsoforms">sangerIsoforms</A> 
  - Links together various transcripts of Sanger gene into a cluster (<em>C. elegans</em>)
  <LI><A 
  href="#KnownTo">sangerToKim</A> 
  - Associates Sanger IDs and Kim Lab expression IDs (which are mostly - but not all - the 
  same) (<em>C. elegans</em>)
  <LI><A 
  href="#KnownTo">sangerToPfam</A> 
  - Associates Sanger IDs and Pfam ID (<em>C. elegans</em>)
  <LI><A 
  href="#KnownTo">sangerToRefSeq</A> 
  - Associates Sanger IDs and RefSeq accession (<em>C. elegans</em>)
<!--
  <LI><A 
  href="#BlastTab">scBlastTab</A> 
  - Blastp results of known genes vs. one another (SARS)
-->
  <LI><A 
  href="#ScopDes">scopDes</A> 
  - Structural Classification of Proteins description. See Lo Conte, Brenner et al. NAR 2002"
  <LI><A 
  href="#Seq">seq</A> 
  - Locations of database table sequences referenced in external fasta files </LI>
  <LI><A 
  href="#SfAssign">sfAssign</A> 
  - Superfamily assignment table
  <LI><A 
  href="#SfDes">sfDes</A> 
  - Superfamily description table
  <LI><A 
  href="#SfDescription">sfDescription</A> 
  - Stores domain descriptions for the Superfamily track</LI> 
  <LI><A 
  href="#SgdAbundance">sgdAbundance</A> 
  - Protein abundance data from http://yeastgfp.ucsf.edu via SGD (<em>S. cerevisiae</em>)
  <LI><A 
  href="#BlastTab">sgdBlastTab</A> 
  - Blastp results of SGD genes vs. one another (<em>S. cerevisiae</em>)
  <LI><A 
  href="#KnownCanonical">sgdCanonical</A> 
  - Describes the canonical splice variant of a gene (<em>S. cerevisiae</em>)
  <LI><A 
  href="#SgdClone">sgdClone</A> 
  - Clone information from the Saccharomyces Genome Database (<em>S. cerevisiae</em>)
  <LI><A 
  href="#GenePredictions">sgdGene</A> 
  - Saccharomyces Genome Database gene predictions (<em>S. cerevisiae</em>)
  <LI><A 
  href="#KnownIsoforms">sgdIsoforms</A> 
  - Links together various transcripts of a gene into a cluster (<em>S. cerevisiae</em>)
  <LI><A 
  href="#KnownTo">sgdLocalization</A> 
  - Associates known gene and SGD accession (<em>S. cerevisiae</em>)
  <LI><A 
  href="#SgdOther">sgdOther</A> 
  - Other features from Saccharomyces Genome Database (<em>S. cerevisiae</em>)
  <LI><A 
  href="#SgdOther">sgdOtherDescription</A> 
  - More features from Saccharomyces Genome Database (<em>S. cerevisiae</em>)
  <LI><A 
  href="#PepPredictions">sgdPep</A> 
  - Translations of SGD gene predictions into corresponding amino acid sequences (<em>S. cerevisiae</em>)
  <LI><A 
  href="#KnownTo">sgdToName</A> 
  - Associates SGD gene name and mRNA (<em>S. cerevisiae</em>)
  <LI><A 
  href="#KnownTo">sgdToPfam</A> 
  - Associates SGD ID and Pfam ID (<em>S. cerevisiae</em>)
  <LI><A 
  href="#KnownTo">sgdToSgd</A> 
  - Associates SGD IDs (<em>S. cerevisiae</em>)
  <LI><A 
  href="#KnownTo">sgdToSwissProt</A> 
  - Associates SGD IDs and Swiss-Prot IDs (<em>S. cerevisiae</em>)
  <LI><A 
  href="#GenePredictions">sgpGene</A> 
  - SGP gene predictions (GRIB)
  <LI><A 
  href="#PepPredictions">sgpPep</A> 
  - Translations of SGP gene predictions into corresponding amino acid sequences 
  <LI><A 
  href="#SimpleRepeats">simpleRepeat</A> 
  - Simple tandem repeats generated with the TRF program 
  <LI><A 
  href="#GenePredictions">slamHuman</A> 
  - Slam predictions of coding exons in homologous DNA sequences (human genome)
  <LI><A 
  href="#BED">slamHumanNonCoding</A> 
  - Slam predictions of conserved noncoding regions in homologous DNA sequences (human genome) 
  <LI><A 
  href="#GenePredictions">slamMouse</A> 
  - Slam predictions of coding exons in homologous DNA sequences (mouse genome)
  <LI><A 
  href="#BED">slamMouseNonCoding</A> 
  - Slam predictions of conserved noncoding regions in homologous DNA sequences  (mouse genome)
  <LI><A 
  href="#GenePredictions">slamRat</A> 
  - Slam predictions of coding exons in homologous DNA sequences (rat genome)
  <LI><A 
  href="#BED">slamRatNonCoding</A> 
  - Slam predictions of conserved noncoding regions in homologous DNA sequences (rat genome)
  <LI><A 
  href="#GenePredExt">snapGene</A> 
  - Alignments of SNAP genes (Semi-HMM-based Nucleic Acid Parser) against the current genome
  <LI><A 
  href="#Snp">snp</A> 
  - SNP positions from various sources (hg17 human assemblies and later)</LI>
  <LI><A 
  href="#SnpExceptions">snpExceptions</A> 
  - Contains a set of queries to look for potentially problematic SNPs</LI>
  <LI><A 
  href="#SnpMap">snpMap</A> 
  - SNP positions from various sources (hg13, hg15, and  hg16 human assemblies)</LI>
  <LI><A 
  href="#BED">snpNih</A> 
  - Locations of SNPs detected by aligning reads from random genomic clones against the genome (replaced by snpMap starting with hg13) </LI>
  <LI><A 
  href="#BED">snpTsc</A> 
  - Locations of SNPs detected primarily by looking at overlaps between clones that
cover the same region (replaced by snpMap starting with hg13) </LI>
  <LI><A 
  href="#GenePredictions">softberryGene</A> 
  - Fgenesh++ Gene Predictions </LI>
  <LI><A 
  href="#SoftberryHom">softberryHom</A> 
  - String with GenBank gi and accession </LI>
  <LI><A 
  href="#PepPredictions">softberryPep</A> 
  - Translations of Fgenesh++ gene predictions into corresponding amino acid sequences </LI>
  <LI><A 
  href="#SpDisease">spDisease</A> 
  - Associates SwissProt IDs with a disease description
  <LI><A 
  href="#SpMrna">spMrna</A> 
  - Associates SwissProt IDs with mRNA IDs
  <LI><A 
  href="#SpOrganism">spOrganism</A> 
  - Associates SwissProt IDs with a taxonomy number
  <LI><A 
  href="#SpPsiBlast">spPsiBlast</A> 
  - PSI-BLAST E-value for 2 proteins
  <LI><A 
  href="#SpSecondaryID">spSecondaryID</A> 
  - Associates SwissProt IDs with secondary accession IDs
  <LI><A 
  href="#SpXref2">spXref2</A> 
  - Associates SwissProt IDs with other databases
  <LI><A 
  href="#STSalias">stsAlias</A> 
  - STS marker aliases and associated identification numbers  </LI>
  <LI><A 
  href="#STSinfo">stsInfo</A> 
  - Constant STS marker information </LI>
  <LI><A 
  href="#STSinfo2">stsInfo2</A> 
  - Constant STS marker information - revised version used in later releases</LI>
  <LI><A 
  href="#STSinfoMouse">stsInfoMouse</A> 
  - Constant STS marker information (MGI)</LI>
  <LI><A 
  href="#STSinfoMouseNew">stsInfoMouseNew</A> 
  - Constant STS marker information (NCBI)</LI>
  <LI><A 
  href="#STSinfoRat">stsInfoRat</A> 
  - Constant STS marker information - rat assembly</LI>
  <LI><A 
  href="#STSmap">stsMap</A> 
  - STS markers on genetic (blue) and radiation hybrid (black) maps </LI> 
  <LI><A 
  href="#STSmapMouse">stsMapMouse</A> 
  - STS markers and postions on mouse assembly (MGI)
  <LI><A 
  href="#STSmapMouseNew">stsMapMouseNew</A> 
  - STS markers and postions on mouse assembly (NCBI)
  <LI><A 
  href="#STSmapRat">stsMapRat</A> 
  - STS markers and postions on rat assembly
  <LI><A 
  href="#STSsites">stsMarker</A> 
  - STS marker mappings from various maps and FISH clone mappings from the BAC 
  Resource Consortium (obsolete as of Apr. 2001, replaced by stsMap)</LI>
  <LI><A 
  href="#BED">supercontig</A> 
  - Locations of supercontigs from WGS assembly</LI>
  <LI><A 
  href="#BED">superfamily</A> 
  - Superfamily proteins</LI>
  <LI><A 
  href="#SwInterPro">swInterPro</A> 
  - Associates Swiss-Prot accession IDs with InterPro IDs
  <LI><A 
  href="#BED">syntenyHg16</A> 
  - Human synteny alignnments (hg16)</LI>
  <LI><A 
  href="#SyntenyHuman">syntenyHuman</A> 
  - Human synteny alignnments from blastz single coverage </LI>
  <LI><A 
  href="#BED">syntenyMm3</A> 
  - Synteny between mouse (mm3) and human chromosomes</LI>
  <LI><A 
  href="#BED">syntenyMouse</A> 
  - Synteny between mouse and human chromosomes (First used in hg13) </LI>
  <LI><A 
  href="#SyntenyRat">syntenyRat</A> 
  - Synteny between rat and human chromosomes (First used in hg13) </LI>

<A NAME="T_GBD"></A>
  <LI><A 
  href="#tfbsCons">tfbsCons</A> 
  - Transcription factor binding sites conserved in a multiple species alignment</LI>
  <LI><A 
  href="#tfbsConsMap">tfbsConsMap</A> 
  - Transcription factor binding sites conservation map data</LI>
  <LI><A 
  href="#GenePredictions">tigrGeneIndex</A> 
  - Alignments of the TIGR Gene Index against the selected genome</LI>
  <LI><A 
  href="#TilingPath">tilingPath</A> 
  - A tiling path of clones through a chromosome </LI>
  <LI><A 
  href="#TransRegCode">transRegCode</A> 
  - Transcription factor binding sites from CHIP/CHIP experiments and conservation
  <LI><A 
  href="#TransRegCodeCondition">transRegCodeCondition</A> 
  - Associates a transcription factor with a growth condition
  <LI><A 
  href="#TransRegCodeMotif">transRegCodeMotif</A> 
  - Transcription factor binding motif according to Harbison Gordon et al.
  <LI><A 
  href="#TransRegCodeProbe">transRegCodeProbe</A> 
  - CHIP/CHIP probe and trranscription factor binding info
  <LI><A 
  href="#GenePredictions">twinscan</A> 
  - TwinScan Gene Predictions
  <LI><A 
  href="#PepPredictions">twinscanPep</A> 
  - Translations of Twinscan gene predictions into corresponding amino acid sequences </LI>

<A NAME="U_GBD"></A>
  <LI><A 
  href="#PslInfo">uniGene</A> 
  - UniGene alignments and SAGE Info (Last used Apr. 2001)  </LI>
  <LI><A 
  href="#BED">uniGene_2</A> 
  - UniGene alignments and SAGE Info (new version)  </LI>
  <LI><A 
  href="#PslInfo">uniGene_dr</A> 
  - Zebrafish UniGene alignments </LI>
  <LI><A 
  href="#PslInfo">uniGene_gg</A> 
  - Chicken UniGene alignments </LI>

<A NAME="V_GBD"></A>
  <LI><A 
  href="#GenePredictions">vegaGene</A> 
  - Annotated genes from the <A HREF="http://vega.sanger.ac.uk/" TARGET=_blank>Vertebrate Genome Annotation (VEGA) database</A> (Human chr14, 20, 22 only) </LI>
  <LI><A 
  href="#VegaInfo">vegaInfo</A> 
  - Additional information for Vega genes (Human chr14, 20, 22 only) </LI>
  <LI><A 
  href="#PepPredictions">vegaPep</A> 
  - Translations of Vega genes into corresponding amino acid sequences (Human chr14, 20, 22 only) </LI>
  <LI><A 
  href="#GenePredictions">vegaPseudoGene</A> 
  - Annotated pseudogenes and immunoglobulin segments from the VEGA database</A> (Human chr14, 20, 22 only) </LI>
  <LI><A 
  href="#Vntr">vntr</A> 
  - Microsatellites from Gerome Breen's VNTR program

<A NAME="W_GBD"></A>
  <LI><A 
  href="#WabaFullCreate">wabaCbr</A> 
  - <em>C. briggsae</em> WABA alignments - full description
  <LI><A 
  href="#WgRna">wgRna</A> 
  - CD and H/ACA Box snoRNAs and microRNAs from Weber and Griffiths-Jones
  <LI><A 
  href="#PslInfo">wz_ests</A> 
  - Alignments of WZ ESTs from WUSTL to the zebrafish assembly 

<A NAME="X_GBD"></A>
  <LI><A 
  href="#PslInfo">xenoEst</A> 
  - Nonhuman ESTs from GenBank </LI>
  <LI><A 
  href="#PslInfo">xenoMrna</A> 
  - Nonhuman mRNAs from GenBank </LI>
  <LI><A 
  href="#RefFlat">xenoRefFlat</A> 
  - RefSeq genes from species other than current species, with additional name (as it 
    appears in Genome Browser) field  </LI>
  <LI><A 
  href="#GenePredictions">xenoRefGene</A> 
  - RefSeq genes from species other than current species</LI>
  <LI><A 
  href="#PslInfo">xenoRefSeqAli</A> 
  - RefSeq translated proteins for species other than current species

<A NAME="Y_GBD"></A>
  <LI><A 
  href="#ExpData">yeastChoCellCycle</A> 
  - Expression data </LI>
  <LI><A 
  href="#ExpRecord">yeastChoCellCycleExps</A> 
  - Experiment descriptive data</LI>
  <LI><A 
  href="#ExpData">yeastChoCellCycleRatio</A> 
  - Expression data </LI>

<A NAME="Z_GBD"></A>
  <LI><A 
  href="#ScoredRef">zebrafish_netBlastz</A> 
  - Scored reference file for zebrafish alignments
  <LI><A 
  href="#WiggleTrack">zoom1_affyTranscriptome</A> 
  - Affymetrix Transcriptome scores (large scale)</LI>
  <LI><A 
  href="#WiggleTrack">zoom2_affyTranscriptome</A> 
  - Affymetrix Transcriptome scores (medium scale)</LI>
</UL>
	   </TD><TD WIDTH=15></TD>
  	  </TR>
	</TABLE>
	<BR>
      </TD></TR>
    </TABLE>
    <BR>

<!--mRNA Tables---------------------------------------------------------->
    <TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1>
      <TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%">
	  <TR><TD>
	    <FONT SIZE="4"><B> &nbsp;mRNA Sequence Tables</B></FONT>
	  </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH>
	  <TR><TD WIDTH=10></TD>
	    <TD>
	      <P>
	      The tables in this section contain additional information about 
	      mRNA sequences.
<UL>
  <LI><A 
  href="#MrnaChar">author</A> 
  - List of authors 
  <LI><A 
  href="#MrnaInfo">bacClone</A> 
  - Name of the Bacterial Artificial Chromosome clone that a sequence 
    was grown in, if applicable (Last used Aug. 2001) </LI>
  <LI><A 
  href="#MrnaChar">cds</A> 
  - Coding sequence in GenBank format 
  <LI><A 
  href="#MrnaChar">cell</A> 
  - Cell types 
  <LI><A 
  href="#MrnaInfo">center</A> 
  - Sequencing center (Last used Aug. 2001)
  <LI><A 
  href="#MrnaInfo">chromosome</A> 
  - Chromosome ID and name (Last used Aug. 2001)
  <LI><A 
  href="#MrnaInfo">cytoMap</A> 
  - Position on a cytological map (Last used Aug. 2001)
  <LI><A 
  href="#MrnaChar">description</A> 
  - Short (one line) description 
  <LI><A 
  href="#MrnaChar">development</A> 
  - Stage of development of organism 
  <LI><A 
  href="#MrnaInfo">gbCdnaInfo</A> 
  - Very relational mRNA and EST information table. Consists mostly of the mRNA 
  accession and pointers into other tables that contain additional mRNA information. 
  <LI><A 
  href="#MrnaChar">geneName</A> 
  - Name of gene 
  <LI><A 
  href="#MrnaChar">keyword</A> 
  - Associated keywords in GenBank 
  <LI><A 
  href="#MrnaChar">library</A> 
  - Clone library (mostly for ESTs) 
  <LI><A 
  href="#MrnaChar">moddate</A> 
  - Last modification date
  <LI><A 
  href="#MrnaChar">mrnaClone</A> 
  - Clone name (mostly for ESTs) 
  <LI><A 
  href="#MrnaChar">organism</A> 
  - Name of organism 
  <LI><A 
  href="#MrnaChar">productName</A> 
  - Protein product name 
  <LI><A 
  href="#MrnaChar">sex</A> 
  - Sex of organism 
  <LI><A 
  href="#MrnaChar">source</A> 
  - Source field in GenBank, often fairly redundant with other fields 
  <LI><A 
  href="#MrnaChar">tissue</A> 
  - Tissue isolated from </LI>
  </UL>
	   </TD><TD WIDTH=15></TD>
  	  </TR>
	</TABLE>
	<BR>
      </TD></TR>
    </TABLE>
    <BR>


<!--mRNA Tables---------------------------------------------------------->
    <TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1>
      <TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%">
	  <TR><TD>
	    <FONT SIZE="4"><B> &nbsp;Internal Use Tables</B></FONT>
	  </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH>
	  <TR><TD WIDTH=10></TD>
	    <TD>
<P>The tables in this section are intended primarily for internal use.</P>
<UL>
  <LI><A 
  href="#AxtInfo">axtInfo</A> 
  - Axt alignment names and sizes 
  <LI><A 
  href="#Certificate">certificate</A> 
  - Table for non-standard join certificates
  <LI><A 
  href="#extFileInfo">extFile</A> 
  - References to external files, again only used by RNA. 
  <LI><A 
  href="#GbExtFile">gbExtFile</A> 
  - References to external GenBank files. 
  <LI><A 
  href="#GbLoaded">gbLoaded</A> 
  - GenBank incremental update information. 
  <LI><A 
  href="#TrackGroups">grp</A> 
  - Track group types in browser display window
  <LI><A 
  href="#HistoryInfo">history</A> 
  - Modification history of database, only used by RNA programs </LI>
  <LI><A 
  href="#TableDesc">tableDescriptions</A> 
  - Descriptive information about database tables form autoSQL/gbdDescriptions
  <LI><A 
  href="#TrackDB">trackDb</A> 
  - Describes an annotation track </LI>
</UL>
	   </TD><TD WIDTH=15></TD>
  	  </TR>
	</TABLE>
	<BR>
      </TD></TR>
    </TABLE>
    <BR>

<!--Description Section---------------------------------------->

    <TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1>
      <TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%">
	  <TR><TD>
	    <FONT SIZE="4"><A name=TableFormats></A><B> &nbsp;autoSql Descriptions of Tables</B></FONT>
	  </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH>
	  <TR><TD WIDTH=10></TD>
	    <TD>
	      <P>
	      This section describes the format of each table in 
	      autoSql format.

<H4><A name=Affy10KDetails></A>Affymetrix 10K SNP Details</H4>
<PRE>table affy10KDetails
"Information representing the Affymetrix 10K Mapping Array"
(       
    string     affyId;    "Affymetrix SNP id"
    string     rsId;      "RS identifier (some are null)"
    string     tscId;     "TSC identifier (some are null)"
    char[2]    baseA;     "The first allele (A)"
    char[2]    baseB;     "The second allele (B)"
    char[34]   sequenceA; "The A allele with flanking sequence"
    char[34]   sequenceB; "The B allele with flanking sequence"
    char[8]    enzyme;    "The enzyme that was used to prepare the sample (HindIII
			   or XbaI)"
)
</PRE>

<H4><A name=AffyGenoDetails></A>Affymetrix 120K SNP Array Information</H4>
<PRE>table affyGenoDetails
"Information representing the Affymetrix 120K SNP array"
(       
    uint       affyId;    "Affymetrix SNP id"
    uint       rsId;      "RS identifier (some are null)"
    char[2]    baseA;     "The first allele (A)"
    char[2]    baseB;     "The second allele (B)"
    char[34]   sequenceA; "The A allele with flanking sequence"
    char[34]   sequenceB; "The B allele with flanking sequence"
    char[8]    enzyme;    "The enzyme that was used to prepare the sample (HindIII
			   or XbaI)"
    float      minFreq;   "The minimum allele frequency"
    float      hetzyg;    "The heterozygosity from all observations"
    float      avHetSE;   "The Standard Error for the average heterozygosity (not
			   used)"
    char[2]    NA04477;   "Individual 01"
    char[2]    NA04479;   "Individual 02"
    char[2]    NA04846;   "Individual 03"
    char[2]    NA11036;   "Individual 04"
    char[2]    NA11038;   "Individual 05"
    char[2]    NA13056;   "Individual 06"
    char[2]    NA17011;   "Individual 07"
    char[2]    NA17012;   "Individual 08"
    char[2]    NA17013;   "Individual 09"
    char[2]    NA17014;   "Individual 10"
    char[2]    NA17015;   "Individual 11"
    char[2]    NA17016;   "Individual 12"
    char[2]    NA17101;   "Individual 13"
    char[2]    NA17102;   "Individual 14"
    char[2]    NA17103;   "Individual 15"
    char[2]    NA17104;   "Individual 16"
    char[2]    NA17105;   "Individual 17"
    char[2]    NA17106;   "Individual 18"
    char[2]    NA17201;   "Individual 19"
    char[2]    NA17202;   "Individual 20"
    char[2]    NA17203;   "Individual 21"
    char[2]    NA17204;   "Individual 22"
    char[2]    NA17205;   "Individual 23"
    char[2]    NA17206;   "Individual 24"
    char[2]    NA17207;   "Individual 25"
    char[2]    NA17208;   "Individual 26"
    char[2]    NA17210;   "Individual 27"
    char[2]    NA17211;   "Individual 28"
    char[2]    NA17212;   "Individual 29"
    char[2]    NA17213;   "Individual 30"
    char[2]    PD01;      "Individual 31"
    char[2]    PD02;      "Individual 32"
    char[2]    PD03;      "Individual 33"
    char[2]    PD04;      "Individual 34"
    char[2]    PD05;      "Individual 35"
    char[2]    PD06;      "Individual 36"
    char[2]    PD07;      "Individual 37"
    char[2]    PD08;      "Individual 38"
    char[2]    PD09;      "Individual 39"
    char[2]    PD10;      "Individual 40"
    char[2]    PD11;      "Individual 41"
    char[2]    PD12;      "Individual 42"
    char[2]    PD13;      "Individual 43"
    char[2]    PD14;      "Individual 44"
    char[2]    PD15;      "Individual 45"
    char[2]    PD16;      "Individual 46"
    char[2]    PD17;      "Individual 47"
    char[2]    PD18;      "Individual 48"
    char[2]    PD19;      "Individual 49"
    char[2]    PD20;      "Individual 50"
    char[2]    PD21;      "Individual 51"
    char[2]    PD22;      "Individual 52"
    char[2]    PD23;      "Individual 53"
    char[2]    PD24;      "Individual 54"
)
</PRE>


<H4><A name=AffyTranscriptome></A>Affymetrix Transcriptome</H4>
<PRE>table affyTranscriptome
"Describes (x,y) pairs of samples"
    ushort bin;                         "A field to speed indexing" 
    string  chrom;                      "Chromosome alignment is on"
    uint    chromStart;                 "Start position in chromosome"
    uint    chromEnd;                   "End position in chromosome"
    string  name;                       "Unique index given to each block of samples"
    uint    score;                      "Overall block score from 0 to 1000"
    char[1] strand;                     "Strand: + or -"
    uint    sampleCount;                "Number of samples in this block"
    uint[sampleCount] samplePosition;   "Base position of each sample in this
                                         block, relative to chromStart"
    uint[sampleCount] sampleHeight;     "Score for each sample (y-value) from 0 to
					 1000"
   )
</PRE>

<H4><A name=AltGraphX></A>Alternative Splicing</H4>
<PRE>table altGraphX
"An alternatively-spliced gene graph"
    (
    string tName;                   	"Name of target sequence, often a chrom"
    int tStart;                     	"First bac touched by graph"
    int tEnd;                       	"Start position in first bac"
    string name;                    	"Human readable name"
    uint id;                        	"Unique ID"
    char[2] strand;                 	"+ or - strand"
    uint vertexCount;               	"Number of vertices in graph"
    ubyte[vertexCount] vTypes;      	"Type for each vertex"
    int[vertexCount] vPositions;    	"Position in target for each vertex"
    uint edgeCount;                 	"Number of edges in graph"
    int[edgeCount] edgeStarts;      	"Array with start vertex of edges"
    int[edgeCount] edgeEnds;        	"Array with end vertex of edges"
    table evidence[edgeCount] evidence; "Array of evidence tables containing references 
				  	 to mRNAs that support a particular edge"
    int[edgeCount] edgeTypes;       	"Type for each edge, ggExon, ggIntron, etc"
    int mrnaRefCount;               	"Number of supporting mRNAs"
    string[mrnaRefCount] mrnaRefs;  	"Ids of mrnas supporting this"
    int[mrnaRefCount] mrnaTissues;  	"Ids of tissues that mrnas come from, indexes 
					 into tissue table"
    int[mrnaRefCount] mrnaLibs;     	"Ids of libraries that mrnas come from, indexes 
					 into library table"
   )
</PRE>

<H4><A name=atlasOncoGene></A>Atlas Oncology Site</H4>
<PRE>table atlasOncoGene
"Table used to link into ATLAS Oncology site"
   (
   string locusSymbol;  "LocusLink Symbol"
   string atlasGene;    "ATLAS Gene"
   string otherGene;    "Other gene"
   string url;          "URL for corresonding ATLAS web page"
   )
</PRE>

<H4><A name=BactigPos></A>Bactig Positions</H4>
<PRE>table bactigPos
"Bactig positions in chromosome coordinates (bed 4 +)."
    (
    string chrom;        "Chromosome"
    uint   chromStart;   "Start position in chromosome"
    uint   chromEnd;     "End position in chromosome"
    string name;         "Bactig"
    string startContig;  "First contig in this bactig"
    string endContig;    "Last contig in this bactig"
    )
</PRE>

<H4><A name=BgiSnp></A>BGI (Beijing Genomics Institute) SNPs</H4>
<PRE>table bgiSnp
"Beijing Genomics Institute SNP information as bed 4 +"
    (
    string chrom;       "Chromosome"
    uint   chromStart;  "Start position in chromosome"
    uint   chromEnd;    "End position in chromosome"
    string name;        "BGI SNP name: snp.superctg.ctg.pos.type.strainID"
    char[1] snpType;    "S (substitution), I (insertion), or D (deletion)"
    uint readStart;     "Start position in alternate allele read"
    uint readEnd;       "End position in alternate allele read"
    uint qualChr;       "Quality score in reference assembly"
    uint qualReads;     "Quality score in alternate allele read"
    string snpSeq;      "'X->Y' or indel sequence"
    string readName;    "Name of alternate allele read"
    char[1] readDir;    "Direction of read relative to reference"
    char[4] inBroiler;  "SNP found in Broiler strain? yes, no or n/a if not covered"
    char[4] inLayer;    "SNP found in Layer strain? yes, no or n/a if not covered"
    char[4] inSilkie;   "SNP found in Silkie strain? yes, no or n/a if not covered"
    string primerL;     "Left primer sequence"
    string primerR;     "Right primer sequence"
    char[1] questionM;  "L for dubious indels, H for other indels and SNPs"
    string extra;       "Additional information"
    )
</PRE>

<H4><A name=BioCycMapDesc></A>Bio Cyc Pathway Map</H4>
<PRE>table bioCycMapDesc
"Decription of BioCyc pathway maps"
   (
   string mapID;        "BioCyc pathway map"
   string description;  "BioCyc pathway map description"
   )
</PRE>

<H4><A name=BioCycPathway></A>Bio Cyc Pathway</H4>
<PRE>table bioCycPathway
"BioCyc Pathway to Known Gene cross reference"
   (
   string kgID;         "Known Gene ID" 
   string geneID;       "Gene (RefSeq) ID"
   string mapID;        "BioCyc pathway map ID"
   )
</PRE>

<H4><A name=BlastTab></A>Blast Output</H4>
The following definition is used for several tables that contain blast 
for various organisms.
<PRE>table blastTab
"Tab-delimited blast output file"
    (
    string query;	"Name of query sequence"
    string target;	"Name of target sequence"
    float identity;	"Percent identity"
    uint aliLength;	"Length of alignment"
    uint mismatch;	"Number of mismatches"
    uint gapOpen;	"Number of gap openings"
    uint qStart;	"Start in query (0 based)"
    uint qEnd;		"End in query (non-inclusive)"
    uint tStart;	"Start in target (0 based)"
    uint tEnd;		"End in target (non-inclusive)"
    double eValue;	"Expectation value"
    double bitScore;	"Bit score"
    )
</PRE>


<H4><A name=BED></A>Browser Extensible Data (BED)</H4>
The following definition is used for several tables, which may use 4 or more of 
the fields.<BR>
<PRE>table bed
"Browser extensible data"
    (
    string chrom;      "Chromosome or FPC contig"
    uint   chromStart; "Start position in chromosome"
    uint   chromEnd;   "End position in chromosome"
    string name;       "Name of item"
    uint   score;      "Score from 0-1000"
    char[1] strand;    "+ or -"
    uint thickStart;   "Start of where display should be thick (start codon)"
    uint thickEnd;     "End of where display should be thick (stop codon)"
    uint reserved;     "Always zero for now"
    int blockCount;    "Number of blocks (a block contains no gaps)"
    int[blockCount] blockSizes; "Comma-separated list of block sizes"
    int[blockCount] blockStarts; "Start positions relative to chromStart."
    int expCount;      "Number of experiments."
    int[expCount] expIds; "Experiment IDs."
    int[expCount] expScores; "Experiment scores."
    )
</PRE>

<H4><A name=CeleraCoverage></A>Celera Coverage</H4>
<PRE>table celeraCoverage
"Summary of large genomic Duplications from Celera Data"
    (
    string chrom;   	"Chromosome or FPC contig"
    uint chromStart;    "Start position in chromosome"
    uint chromEnd;      "End position in chromosome"
    string name; 	"Source of information"
    )
</PRE>

<H4><A name=CeleraDupPositive></A>Celera Duplications</H4>
<PRE>table celeraDupPositive
"Summary of large genomic Duplications from Celera Data"
    (
    string chrom;   	"Chromosome or FPC contig"
    uint chromStart;    "Start position in chromosome"
    uint chromEnd;  	"End position in chromosome"
    string name;    	"Celera accession name"
    string fullname;    "Celera accession full name"
    float fracMatch;    "Fraction of matching bases"
    float bpalign;  	"Base pair alignment score"
    )
</PRE>

<H4><A name=CeleraOverlay></A>Celera Overlay</H4>
<PRE>table celeraOverlay
"Celera assembly overlay in the public (NCBI) assembly"
    (
    string chrom;      	"Chromosome"
    uint chromStart;   	"Start position in chromosome"
    uint chromEnd;     	"End position in chromosome"
    string name;       	"Other chromosome and start"
    string otherChrom; 	"Other chromosome"
    uint otherStart;   	"Start in other chromosome"
    uint otherEnd;     	"End in other chromosome"
    char orient;       	"Strand/orientation (F or R)"
    char field1;       	"? (always M)"
    char field2;       	"? (r or u)"
    string localId;    	"?"
    string parentInfo; 	"?"
    string refId;      	"Reference ID"
    uint refStart;     	"Start position in reference"
    uint refLength;     "Length in reference"
    byte refOrient;   	"Strand/orientation in reference (-1 or 1)"
    string queryId;    	"Query ID"
    uint queryStart;    "Start position in query"
    uint queryLength;   "Length in query"
    byte queryOrient; 	"Strand/orientation in query (-1 or 1)"
    )
</PRE>

<H4><A name=CgapAlias></A>CGAP Alias</H4>
<PRE>table cgapAlias
"Associates CGAP pathway IDs with gene symbols or mRNA accessions"
   (
   string cgapID;       "CGAP pathway ID"
   string alias;        "Gene symbol or mRNA"
   )
</PRE>

<H4><A name=CgapBiocDesc></A>CGAP/BioCarta Pathway Descriptions</H4>
<PRE>table cgapBiocDesc
"CGAP/BioCarta pathway description"
  (
  string mapID;         "CGAP/BioCarta pathway map ID"
  string description;   "Description"
  )
</PRE>

<H4><A name=CgapBiocPathway></A>CGAP/BioCarta Pathway Cross-Reference</H4>
<PRE>table cgapBiocPathway
"CGAP BioCarta pathway cross reference"
   (
   string cgapID;       "CGAP pathway ID"
   string mapID;        "BioCarta pathway ID"
   )
</PRE>

<H4><A name=Chain></A>Chain Alignments</H4>
<PRE>table chain
"Summary info about a chain of alignments"
   (
   double score; 	"Score of chain"
   string tName; 	"Target sequence name"
   uint tSize; 		"Target sequence size"
   uint tStart; 	"Alignment start position in target"
   uint tEnd; 		"Alignment end position in target"
   string qName; 	"Query sequence name"
   uint qSize; 		"Query sequence size"
   char qStrand; 	"Query strand"
   uint qStart; 	"Alignment start position in query"
   uint qEnd; 		"Alignment end position in query"
   uint id; 		"Chain ID"
   }
</PRE>

<H4><A name=ChainLink></A>Chain Link</H4>
<PRE>table chainLink
"Alignment block in chain"
   (
   string tName; 	"Target sequence name"
   uint tStart; 	"Alignment start position in target"
   uint tEnd; 		"Alignment end position in target"
   uint qStart; 	"Start in query"
   uint chainId; 	"Chain ID in chain table"
   )
</PRE>

<H4><A name=ChimpSimpleDiff></A>Chimp/Human Simple Differences</H4>
<PRE>table chimpSimpleDiff
"Simple differences between the chimp and human genome assemblies"
   (
   string chrom;        "Chromosome number"
   uint   chromStart;   "Start position in chromosome"
   uint   chromEnd;     "End position in chromosome"
   string   tseq;	"Target sequence (A,T,C,G)"
   string   qseq;	"Query sequence (A,T,C,G)"
   )
</PRE>

<H4><A name=ChromosomeBand></A>Chromosome Band Information</H4>
<PRE>table cytoBand
"Describes the positions of cytogenetic bands within a chromosome"
   (
   string chrom;        "Chromosome number"
   uint   chromStart;   "Start position in genoSeq"
   uint   chromEnd;     "End position in genoSeq"
   string name;         "Name of cytogenetic band"
   string gieStain;     "Giesma stain results"
   )
</PRE>

<H4><A name=ChromInfo></A>Chromosome Information</H4>
<PRE>table chromInfo
"Chromosome names and sizes"
    (
    string chrom;       "Chromosome name"
    uint   size;        "Chromosome size"
    string fileName;    "Chromosome file (raw one byte per base)"
    )
</PRE>

<H4><A name=FragInfo></A>Clone Fragment Positions</H4>
<PRE>table chrN_gl
"Fragment positions in golden path"
    (
    ushort  bin;          "A field to speed indexing added to many 
                           tables August 2001"
    uint    matches;      "Number of bases that match that aren't repeats"
    uint    misMatches;   "Number of bases that don't match"
    string  frag;         "Fragment name"
    uint    start;        "Start position in golden path"
    uint    end;          "End position in golden path"
    char[1] strand;       "+ or - for strand"
    )
</PRE>

<H4><A name=CloneInfo></A>Clone Information</H4>
<PRE>table clonePos
"A clone's position and other info."
    (
    string  name;       "Name of clone including version"
    uint    seqSize;    "base count not including gaps"
    ubyte   phase;      "htg phase"
    string  chrom;      "Chromosome name"
    uint    chromStart; "Start in chromosome"
    uint    chromEnd;   "End in chromosome"
    char[1] stage;      "F/D/P for finished/draft/predraft"
    string  faFile;     "File with sequence."
    )
</PRE>

<H4><A name=Contamination></A>Contamination Information</H4>
<PRE>table contamination
"Contamination in assembly (bed 4 +)"
    (
    string chrom;      "Chromosome"
    uint   chromStart; "Start position in chromosome"
    uint   chromEnd;   "End position in chromosome"
    string name;       "Name (type of contamination: E.coli, vector etc)"
    string acc;        "NCBI accession where contamination was found"
    string ctg;        "Alternate name (e.g. WGS contig name)"
    int    len;        "Length of contaminated portion of sequence"
    )
</PRE>

<H4><A name=ContigAcc></A>Contig/Accession Map</H4>
<PRE>table contigAcc
"Maps a contig to its accession."
    (
    string contig;      "Contig name"
    string acc;         "GenBank accession"
    )
</PRE>

<H4><A name=ContigInfo></A>Contig Information</H4>
<PRE>table ctgPos
"Where a contig is inside of a chromosome."
    (
    string contig;      "Name of contig"
    uint   size;        "Size of contig"
    string chrom;       "Chromosome name"
    uint   chromStart;  "Start in chromosome"
    uint   chromEnd;    "End in chromosome"
    )
</PRE>

<H4><A name=ContigInfo2></A>Contig Information (Expanded)</H4>
<PRE>table ctgPos2
"Where a contig is inside of a chromosome including contig type information."
    (
    string contig;      "Name of contig"
    uint   size;        "Size of contig"
    string chrom;       "Chromosome name"
    uint   chromStart;  "Start in chromosome"
    uint   chromEnd;    "End in chromosome"
    char[1] type;       "(W)GS contig, (F)inished, (P)redraft, (D)raft, (O)ther"
    )
</PRE>

<H4><A name="CpGIsland"></A>CpG Islands</H4>
<PRE>table cpgIsland
"Describes CpG Islands of 50 bases or more that are at least 50% G/C"
   (
   string chrom;        "Chromosome or FPC contig"
   uint   chromStart;   "Start position in chromosome"
   uint   chromEnd;     "End position in chromosome"
   string name;         "CpG Island"
   uint   minLength;    "Minimum Island Length"
   uint   cpgNum;       "Number of CpGs in island"
   uint   gcNum;        "Number of C and G in island"
   uint   perCpg;       "Percentage of island that is CpG"
   uint   perGc;        "Percentage of island that is C or G"
   )
</PRE>

<H4><A name="CpGIslandExt"></A>CpG Islands (Extended)</H4>
<PRE>table cpgIslandExt
"Describes the CpG Islands (includes observed/expected ratio)"
   (
   string chrom;        "Chromosome or FPC contig"
   uint   chromStart;   "Start position in chromosome"
   uint   chromEnd;     "End position in chromosome"
   string name;         "CpG Island"
   uint   length;       "Island Length"
   uint   cpgNum;       "Number of CpGs in island"
   uint   gcNum;        "Number of C and G in island"
   uint   perCpg;       "Percentage of island that is CpG"
   uint   perGc;        "Percentage of island that is C or G"
   float  obsExp;	"Ratio of observed(cpgNum) to expected(numC*numG/length) 
			 CpG in island"
   )
</PRE>

<H4><A name=DbSnp></A>dbSnpRS</H4>
<PRE>table dbSnpRS 
"Information from dbSNP at the reference SNP level"
    (
    uint    rsId;    	"RS identifier"
    float   avHet;   	"Average heterozygosity from all observations"
    float   avHetSE; 	"Standard Error for the average heterozygosity from all 
			 observations"
    string  valid;   	"Validation status of the SNP"
    char[1] base1;   	"Base of the first allele"
    char[1] base2;   	"Base of the second allele"
    string  assembly; 	"Sequence in the ucsc assembly"
    string  alternate; 	"Sequence of the alternate allele"
    )
</PRE>

<H4><A name=DnaMotif></A>DNA Motif</H4>
<PRE>table dnaMotif
"A gapless DNA motif"
    (
    string name;                        "Motif name"
    int columnCount;                    "Count of columns in motif"
    float[columnCount] aProb;           "Probability of A's in each column"
    float[columnCount] cProb;           "Probability of C's in each column"
    float[columnCount] gProb;           "Probability of G's in each column"
    float[columnCount] tProb;           "Probability of T's in each column"
    )
</PRE>

<H4><A name=DupSpMrna></A>DupSpMrna</H4>
<PRE>table dupSpMrna
"Duplicate mRNA/Protein entries that have identical CDS structures"
   (
   string mrnaID;        "mRNA ID name"
   string proteinID;     "Protein ID from SwissProt"
   string dupMrnaID;     "mRNA ID of duplicated entry"
   string dupProteinId;  "Protein ID of duplicated entry"
   )
</PRE>

<H4><A name=EncodeRegionInfo></A>EncodeRegionInfo</H4>
<PRE>table encodeRegionInfo
"Descriptive, assembly-independent information about ENCODE regions"
    (
    string name;        "Name of region"
    string descr;       "Description (gene region, random pick, etc.)"
    )
</PRE>

<H4><A name=EnsGtp></A>Ensembl Gene XRef</H4>
<PRE>table ensGtp
"Associates Ensembl gene, transcript, and protein IDs"
   (
   string gene; 	"Gene ID"
   string transcript; 	"Transcript ID"
   string protein; 	"Protein ID"
   )
</PRE>

<H4><A name=EnsemblXref></A>EnsemblXref</H4>
<PRE>table ensemblXref
"Cross-reference info between transcript entries and translation entries of Ensembl 
genes"  
   (
   string db;			"Ensembl Database"
   int gene_id;			"Gene ID number"
   string gene_name;		"Gene name in database"
   int transcript_id;		"Transcript ID number"
   string transcript_name;	"Transcript name in database"
   uint translation_id;		"Translation ID number"
   string translation_name; 	"Translation name in database"
   string external_db;		"External database"
   string external_name;	"External status"
   )
</PRE>

<H4><A name=EnsemblXref2></A>EnsemblXref2</H4>
<PRE>table ensemblXref2
"Cross-reference info between transcript entries and translation entries of Ensembl 
genes (alternative table used in some assemblies)"  
   (
   string transcript_name; 	"Transcript name"
   string translation_name; 	"Translation name"
   )
</PRE>

<H4><A name=EnsemblXref3></A>EnsemblXref3</H4>
<PRE>table ensemblXref3
"A cross-reference table for Ensembl Genes."
   (
   string gene;            "Ensembl gene ID"
   string geneVer;         "Ensembl gene ID version number"
   string transcript;      "Ensembl transcript ID"
   string transcriptVer;   "Ensembl transcript ID version number"
   string protein;         "Ensembl protein ID"
   string proteinVer;      "Ensembl protein version number"
   string tremblAcc;       "TrEMBL  protein accession number"
   string swissDisplayId;  "Swiss-Prot protein display ID"
   string swissAcc;        "Swiss-Prot protein accession number"
   )
</PRE>

<H4><A name=EnsGeneXref></A>Ensembl gene_xref Table</H4>
Ensemble gene_xref table downloaded directly from the Ensembl ftp site (18.34 
release 11/04). Refer to the Ensembl site for details.  CAUTION: Ensembl sometimes
changes its table definitions and some fields may not contain the data that the 
name indicates, e.g. translation_name.<BR>
<PRE>table ensGeneXref
"Table gene_xref downloaded from Ensembl"
   (
   string db;                   "#"
   string analysis;             "#"
   string type;                 "#"
   int gene_id;                 "#"
   string gene_name;            "#"
   int[5] gene_version;         "#"
   int transcript_id;           "#"
   string transcript_name;      "#"
   int[5] transcript_version;   "#"
   int[5] translation_name;     "#"
   int translation_id;          "#"
   int[5] translation_version;  "#"
   string external_db;          "#"
   string external_name;        "#"
   char[10] external_status;    "#"
   )
</PRE>

<H4><A name=EnsTranscript></A>Ensembl transcript Table</H4>
Ensembl transcript table downloaded directly from the Ensembl ftp site (18.34 
release 11/04). Refer to the Ensembl site for details.  CAUTION: Ensembl 
sometimes changes its table definitions and some fields may not contain the data 
that the name indicates.
<PRE>table ensTranscript
"Table transcript downloaded from Ensembl"
   (
   int id;                      "#"
   string db;                   "#"
   string analysis;             "#"
   string type;                 "#"
   int transcript_id;           "#"
   string transcript_name;      "#"
   int transcript_version;      "#"
   string chr_name;             "#"
   int chr_start;               "#"
   int chr_end;                 "#"
   int chr_strand;              "#"
   int coding_start;            "#"
   int coding_end;              "#"
   int translation_id;          "#"
   string translation_name;     "#"
   string translation_version;  "#"
   int gene_id;                 "#"
   string gene_name;            "#"
   int gene_version;            "#"
   lstring exon_structure;      "#"
   lstring exon_ids;            "#"
   string external_db;          "#"
   string external_name;        "#"
   char[10] external_status;    "#"
   )
</PRE>

<H4><A name=Exofish></A>Exofish Information</H4>
<PRE>table exoFish
"An evolutionarily conserved region (ecore) with Tetroadon"
   (
   string chrom;        "Chromosome or FPC contig"
   uint   chromStart;   "Start position in chromosome"
   uint   chromEnd;     "End position in chromosome"
   string name;         "Just a dummy dot"
   uint   score;        "Score from 0 to 1000"
   )
</PRE>

<H4><A name=ExoMouse></A>Exonerate Mouse Homologies</H4>
Last used Apr. 2001.
<PRE>table exoMouse
"A rough alignment - not detailed"
   (
   string chrom;        "Chromosome or FPC contig"
   uint   chromStart;   "Start position in chromosome"
   uint   chromEnd;     "End position in chromosome"
   string name;         "Name of other sequence"
   uint   score;        "Score from 0 to 1000"
   char[1] strand;      " or -"
   uint otherStart;     "Start in other sequence"
   uint otherEnd;       "End in other sequence"
   )
</PRE>

<H4><A name=ExpRecord></A>Experiment Record</H4>
<PRE>table expRecord
"Minimal descriptive data for an experiment in the browser"
   (
   uint id; 			"Internal id of experiment"
   string name; 		"Name of experiment"
   lstring description; 	"Description of experiment"
   lstring url; 		"URL relevant to experiment"
   lstring ref; 		"Reference for experiment"
   lstring credit; 		"Who to credit with experiment"
   uint numExtras; 		"Number of extra things"
   lstring[numExtras] extras; 	"Extra things of interest, i.e. classifications"
   )
</PRE>

<H4><A name=ExpData></A>Expression Data</H4>
<PRE>table expData
"Expression data (no mapping, just spots)"
   (
   string name; 		"Name of gene/target/probe etc."
   uint expCount; 		"Number of scores"
   float[expCount] expScores; 	"Scores. May be absolute or relative ratio"
   )
</PRE>

<H4><A name=ExpDistance></A>Expression Distance Between 2 Genes</H4>
<P>
<PRE>table expDistance
"Distance between two genes in expression space"
    (
    string query;       "Name of one gene"
    string target;      "Name of other gene"
    float distance;     "Distance in expression space, always >= 0"
    )
</PRE>

<H4><A name=FishClones></A>FISH Clones</H4>
<PRE>table fishClones
"Describes the positions of fishClones in the assembly"
   (
   string chrom;                   "Chromosome number"
   uint   chromStart;              "Start position in chromosome"
   uint   chromEnd;                "End position in chromosome"
   string name;                    "Name of clone"
   uint score;                     "Always 1000"
   uint placeCount;                "Number of times FISH'd"
   string[placeCount] bandStarts;  "Start FISH band"
   string[placeCount] bandEnds;    "End FISH band"
   string[placeCount] labs;        "Lab where clone FISH'd"
   string placeType;               "How clone was placed on the sequence assembly"
   uint accCount;                  "Number of accessions associated with the clone"
   string[accCount] accNames;      "Accession associated with clone"
   uint stsCount;                  "Number of STS markers associated with this clone"
   string[stsCount] stsNames;      "Names of STS  markers"
   uint beCount;                   "Number of BAC end sequences associated with this 
				    clone"
   string[beCount] beNames;        "Accessions of BAC ends"
   )
</PRE>

<H4><A name=FlyBaseSwissProt></A>FlyBase Accessions Linked to SwissProt Accessions</H4>
<PRE>table flyBaseSwissProt
"FlyBase acc to SwissProt acc, plus some other SwissProt info"
    (
    string flyBaseId;	"FlyBase FBgn ID"
    string swissProtId;	"SwissProt ID"
    string spGeneName;	"Gene name from SwissProt (long)"
    string spSymbol;	"Symbolic-looking gene ID from SwissProt"
    )
</PRE>

<H4><A name=FbAllele></A>FlyBase Alleles</H4>
<PRE>table fbAllele
"The alleles of a gene"
    (
    int id;		"Allele ID"
    string geneId;	"FlyBase ID of gene"
    string name;	"Allele name"
    )
</PRE>

<H4><A name=FbSynonym></A>FlyBase Gene Synonyms</H4>
<PRE>table fbSynonym
"Links all the names we call a gene to its FlyBase ID"
    (
    string geneId;	"FlyBase ID"
    string name;	"A name (synonym or real)"
    )
</PRE>

<H4><A name=FbGene></A>FlyBase Genes</H4>
<PRE>table fbGene
"Links FlyBase IDs, gene symbols and gene names"
    (
    string geneId;	"FlyBase ID"
    string geneSym;	"Short gene symbol"
    string geneName;	"Gene name - up to a couple of words"
    )
</PRE>

<H4><A name=FlyBase2004Xref></A>FlyBase Genes Cross-Reference (2004)</H4>
<PRE>table flyBase2004Xref
"FlyBase cross references circa late 2004 (dm2/4.0; dp3/1.0 uses a subset)"
    (
    string name;                "FlyBase annotation gene ID"
    string symbol;              "Symbolic gene name"
    lstring synonyms;           "Comma-separated list of synonyms"
    string fbtr;                "FlyBase FBtr acc"
    string fbgn;                "FlyBase FBgn acc"
    string fbpp;                "FlyBase FBpp acc"
    string fban;                "FlyBase FBan acc"
    string type;                "Annotation type (for noncoding only)"
    )
</PRE>

<H4><A name=FbTranscript></A>FlyBase Genes Linked to BDGP Transcript IDs</H4>
<PRE>table fbTranscript
"Links FlyBase gene IDs and BDGP transcript IDs"
    (
    string geneId;	 "FlyBase gene ID"
    string transcriptId; "BDGP Transcript ID"
    )
</PRE>

<H4><A name=FbGo></A>FlyBase Genes Linked to GO IDs</H4>
<PRE>table fbGo
"Links FlyBase gene IDs and GO IDs/aspects"
    (
    string geneId;      "FlyBase gene ID"
    string goId;        "GO ID"
    string aspect;      "P (process), F (function) or C (cellular component)"
    )
</PRE>

<H4><A name=fbPhenotype></A>FlyBase Observed Phenotype in Mutant</H4>
<PRE>table fbPhenotype
"Observed phenotype in mutant. Sometimes contains gene function info"
    (
    string geneId;	"FlyBase gene ID"
    int fbAllele;	"ID in fbAllele table or 0 if not allele-specific"
    int fbRef;		"ID in fbRef table"
    lstring text;	"Descriptive text"
    )
</PRE>

<H4><A name=FbRef></A>FlyBase Reference</H4>
<PRE>table fbRef
"A literature or sometimes database reference"
    (
    int id;		"Reference ID"
    lstring text;	"Usually begins with FlyBase ref ID, but not always"
    )
</PRE>

<H4><A name=FbRole></A>FlyBase Role of Gene in Wildtype</H4>
<PRE>table fbRole
"Role of gene in wildType"
    (
    string geneId;	"FlyBase gene ID"
    int fbAllele;	"ID in fbAllele table or 0 if not allele-specific"
    int fbRef;		"ID in fbRef table"
    lstring text;	"Descriptive text"
    )
</PRE>

<H4><A name=FlyReg></A>FlyReg Data</H4>
<PRE>table flyreg
"Flyreg data from Casey Bergman"
    (
    string chrom;      "Chromosome"
    uint   chromStart; "Start position in chromosome"
    uint   chromEnd;   "End position in chromosome"
    string name;       "Factor"
    string target;     "Target"
    uint   pmid;       "PubMed ID"
    )
</PRE>

<H4><A name=FlyReg2></A>FlyReg Data (version 2)</H4>
<PRE>table flyreg
"Flyreg data (version 2) from Casey Bergman"
    (
    string chrom;      "Chromosome"
    uint   chromStart; "Start position in chromosome"
    uint   chromEnd;   "End position in chromosome"
    string name;       "Factor"
    string target;     "Target"
    uint   pmid;       "PubMed ID"
    uint   fpid;       "Footprint ID -- stable ID across versions"
    )
</PRE>

<H4><A name=FosEndPairs></A>Fosmid End Pairs</H4>
<PRE>table fosEndPairs
"Positions of end pairs for fosmids"
   (
   short  bin;        	      "Bin number for browser speedup"
   string chrom;      	      "Chromosome"
   uint   chromStart; 	      "Start position of fosmid in chromosome"
   uint   chromEnd;   	      "End position of fosmid in chromosome"
   string name;       	      "Name of fosmid"
   uint   score;      	      "Score = 1000/(# of times fosmid appears in assembly)"
   char[1] strand;    	      "Value should be + or -"
   string pslTable;   	      "Table which contains corresponding PSL records for linked 
			       features"
   uint   lfCount;    	      "Number of linked features in the series"
   uint[lfCount]   lfStarts;  "Comma-separated list of start positions of each linked 
				feature in genomic"
   uint[lfCount]   lfSizes;   "Comma-separated list of sizes of each linked feature 
				in genomic"
   string[lfCount] lfNames;   "Comma-separated list of names of linked features"
   )
</PRE>

<H4><A name=GapInfo></A>Gap Positions</H4>
<PRE>table chrN_gap
"Gaps in golden path"
    (
    ushort bin;         "A field to speed indexing added to many 
                         tables August 2001"
    uint   matches;     "Number of bases that match that aren't repeats"
    uint   misMatches;  "Number of bases that don't match"
    string chrom;       "which chromosome"
    uint   chromStart;  "start position in chromosome"
    uint   chromEnd;    "end position in chromosome"
    int    ix;          "ix of this fragment (useless)"
    char[1] n;          "always 'N'"
    uint   size;        "size of gap"
    string type;        "contig, clone, fragment, etc."
    string bridge;      "yes, no, mrna, bacEndPair, etc."
    )
</PRE>

<H4><A name=GCpercent></A>GC Content</H4>
<PRE>table gcPercent
"Displays GC content in 20Kb non-overlapping blocks for genome"
   (
   string chrom;        "Chromosome number"
   uint   chromStart;   "Start position in chromosome"
   uint   chromEnd;     "End position in chromosome"
   string name;         "Constant string GC"
   uint   gcPpt;        "Average number of G and C per 1000 bases"
   )                                                                            
</PRE>

<H4><A name=GbProtAnn></A>GenBank Protein Annotations</H4>
<PRE>table gbProtAnn
"Protein Annotations from GenPept mat_peptide fields"
    (
    string chrom;      	"Chromosome"
    uint   chromStart; 	"Start position in chromosome"
    uint   chromEnd;   	"End position in chromosome"
    string name;       	"Name of item"
    string product;    	"Protein product name"
    string note;       	"Note (may be empty)"
    string proteinId; 	"GenBank protein accession(.version)"
    uint giId;        	"GenBank db_xref number"
    )
</PRE>

<H4><A name=GeneBands></A>Gene Bands</H4>
<PRE>table geneBands
"Band locations of known genes"
   (
   string name;         "Gene name (HUGO Gene Nomenclature Committee symbol, if possible)"
   string mrnaAcc;      "RefSeq mRNA accession"
   int    count;        "Number of times this accession maps to the genome"
   string[count] bands; "List of chromosome bands to which it maps"
   )                                                                            
</PRE>

<H4><A name=RNAcluster></A>Gene Boundaries as Defined by RNA and Spliced
EST Clusters</H4>
<PRE>table rnaCluster
"Gene boundaries deduced from clustering spliced ESTs and mRNAs against the genome"
    (
    smallint bin;           "Index"
    string chrom;           "Chromosome or FPC contig"
    uint   chromStart;      "Start position in chromosome"
    uint   chromEnd;        "End position in chromosome"
    string name;            "Name of item"
    uint   score;           "Score from 0-1000"
    char[1] strand;         "+ or -"
    uint thickStart;        "Start of where display should be thick (start codon)"
    uint thickEnd;          "End of where display should be thick (stop codon)"
    uint reserved;          "Always zero for now"
    int blockCount;         "Number of blocks (a block contains no gaps)"
    longblob blockSizes;    "Size of each block"
    longblob chromStarts;   "Start position of each block in chromosome"
    ) 
</PRE>    

<H4><A name=GenePredictions></A>Gene Predictions and RefSeq Genes</H4>
The following definition is used for gene prediction tables. In 
alternative-splicing situations, each transcript has a row in this table.
<PRE>table genePred
"A gene prediction."
    (
    string  name;               "Name of gene"
    string  chrom;              "Chromosome name"
    char[1] strand;             "+ or - for strand"
    uint    txStart;            "Transcription start position"
    uint    txEnd;              "Transcription end position"
    uint    cdsStart;           "Coding region start"
    uint    cdsEnd;             "Coding region end"
    uint    exonCount;          "Number of exons"
    uint[exonCount] exonStarts; "Exon start positions"
    uint[exonCount] exonEnds;   "Exon end positions"
    )
</PRE>

<H4><A name=GenePredExt></A>Gene Predictions (Extended)</H4>
The following definition is used for extended gene prediction tables. In 
alternative-splicing situations, each transcript has a row in this table.
<PRE>table genePredExt
"A gene prediction with some additional info."
    (
    string name;        	"Name of gene (usually transcript_id from GTF)"
    string chrom;       	"Chromosome name"
    char[1] strand;     	"+ or - for strand"
    uint txStart;       	"Transcription start position"
    uint txEnd;         	"Transcription end position"
    uint cdsStart;      	"Coding region start"
    uint cdsEnd;        	"Coding region end"
    uint exonCount;     	"Number of exons"
    uint[exonCount] exonStarts; "Exon start positions"
    uint[exonCount] exonEnds;   "Exon end positions"
    uint id;            	"Unique identifier"
    string name2;       	"Alternate name (e.g. gene_id from GTF)"
    string cdsStartStat; 	"enum('none','unk','incmpl','cmpl')"
    string cdsEndStat;   	"enum('none','unk','incmpl','cmpl')"
    lstring exonFrames; 	"Exon frame offsets {0,1,2}"
    )
</PRE>

<H4><A name=RefFlat></A>Gene Predictions and RefSeq Genes with Gene Names</H4>
A version of genePred that associates the gene name with the gene prediction 
information. In alternative splicing situations each transcript has a row in this table.
<PRE>table refFlat
"A gene prediction with additional geneName field."
    (
    string  geneName;           "Name of gene as it appears in Genome Browser."
    string  name;               "Name of gene"
    string  chrom;              "Chromosome name"
    char[1] strand;             "+ or - for strand"
    uint    txStart;            "Transcription start position"
    uint    txEnd;              "Transcription end position"
    uint    cdsStart;           "Coding region start"
    uint    cdsEnd;             "Coding region end"
    uint    exonCount;          "Number of exons"
    uint[exonCount] exonStarts; "Exon start positions"
    uint[exonCount] exonEnds;   "Exon end positions"
    )
</PRE>

<H4><A name=PepPredictions></A>Gene Predictions and RefSeq Genes Amino Acid Translations</H4>
The following definition is used for tables that link a predicted peptide to 
a predicted gene: acemblyPep, ECgenePep, ensPep, genieAltPep, genieKnownPep, 
genscanPep, knownGenePep, refPep, sanger22Pep, softberryPep, twinscanPep, vegaPep.
<PRE>table pepPred
"A predicted peptide linked to a predicted gene."
    (
    string  name;       "Name of gene - same as in genePred"
    lstring seq;        "Peptide sequence"
    )
</PRE>

<H4><A name=GenMapDb></A>GenMapDB BAC Clones</H4>
<PRE>table genMapDb
"BAC clones from GenMapDB placed on the assembly by U Penn (V. Cheung)"
   (
   string  chrom;       "Chromosome number or 'unknown'"
   int     chromStart;  "Start position in chromosome (-1 if unpositioned)"
   int     chromEnd;    "End position in chromosome"
   string  name;        "Clone name"
   uint    score;       "Score - always 1000" 
   char[1] strand;      "+ or -"
   string  accT7;       "Accession number for T7 BAC end sequence"
   uint    startT7;     "T7 start position in chromosome"
   uint    endT7;       "T7 end position in chromosome"
   char[1] strandT7;    "+ or -"
   string  accSP6;      "Accession number for Sp6 BAC end sequence"
   uint    startSP6;    "Sp6 start position in chromosome"
   uint    endSP6;      "Sp6 end position in chromosome"
   char[1] strandSP6;   "+ or -"
   string  stsMarker;   "Name of STS marker found in clone"
   uint    stsStart;    "STS marker start position in chromosome" 
   uint    stsEnd;      "STS marker end position in chromosome"
   )                                                                            
</PRE>

<H4><A name=GpInfo></A>Golden Path Construction</H4>
<PRE>table chrN_gold
"How to get through chromosome based on fragments"
    (
    ushort  bin;        "A field to speed indexing added to many 
                         tables August 2001"
    uint    matches;    "Number of bases that match that aren't repeats"
    uint    misMatches; "Number of bases that don't match"
    string  chrom;      "Which chromosome"
    uint    chromStart; "Start position in chromosome"
    uint    chromEnd;   "End position in chromosome"
    int     ix;         "Ix of this fragment (useless)"
    char[1] type;       "(P)redraft, (D)raft, (F)inished or (O)ther"
    string  frag;       "Which fragment"
    uint    fragStart;  "Start position in frag"
    uint    fragEnd;    "End position in frag"
    char[1] strand;     "+ or - (orientation of fragment)"
    )
</PRE>

<H4><A name=HInv></A>H-Invitational Human Gene Annotation Database Data</H4>
<PRE>table HInv
"H-Invitational gene data"
    (
    string geneId; 	"Gene ID"
    string clusterId;	"Gene cluster ID"
    string mrnaAcc; 	"GenBank mRNA accession"
    )
</PRE>

<H4><A name=HgFindSpec></A>HgFind Search Information</H4>
<PRE>table hgFindSpec
"Defines a search to be performed by hgFind"
    (
    string searchName;          "Unique name for this search. Defaults to searchTable 
				 if not specified in .ra"
    string searchTable;         "(Non-unique!) Table to be searched. (Like trackDb.tableName: 
				 if split, omit chr*_ prefix)"
    string searchMethod;        "Type of search (exact, prefix, fuzzy)"
    string searchType;          "Type of search (bed, genePred, knownGene etc)"
    ubyte shortCircuit;         "If nonzero and there is a result from this search, jump 
				 to the result instead of performing other searches"
    string termRegex;           "Regular expression (see man 7 regex) to eval on search 
				 term: if it matches, perform search query"
    string query;               "sprintf format string for SQL query on a given table 
				 and value"
    string xrefTable;           "If search is xref, perform xrefQuery on search term, 
				 then query with that result"
    string xrefQuery;           "sprintf format string for SQL query on a given (xref) 
				 table and value"
    float searchPriority;       "0-1000 - relative order/importance of this search.  
				 0 is top"
    string searchDescription;   "Description of table/search (default: trackDb 
				 {longLabel,tableName})"
    lstring searchSettings;     "Name/value pairs for searchType-specific stuff"
)
</PRE>

<H4><A name=Hugo></A>Hugo Gene Nomenclature Committee (HGNC) Cross-Reference</H4>
<PRE>table hugo
"A cross-reference table between HGNC and other databases"
   (
   string hgnc;            "HGNC number"
   string symbol;          "HGNC gene symbol"
   string name;            "HGNC gene name"
   string map;             "Gene map location"
   string mim;             "MIM ID"
   string pmid1;           "pmid 1"
   string pmid2;           "pmid 2"
   string refseq;          "RefSeq ID"
   string aliases;         "Aliases"
   string widthdraws;      "Withdrawn gene symbols"
   string locuslink;       "LocusLink ID"
   string gdbID;           "GDB ID"
   string swissprot;       "Swiss-Prot ID"
   )
</PRE>

<H4><A name=SyntenyHuman></A>Human Synteny Alignments</H4>
<PRE>table syntenyHuman
"Human synteny alignnments from blastz single coverage"
    (
    string chrom;       "Mouse Chrom"
    uint chromStart;    "Start on Mouse"
    uint chromEnd;      "End on Mouse"
    string name;      	"Human Chromosome"
    uint score;         "Score (always zero)"
    char[1] strand;     "+ direction matches - opposite"
    uint thickStart;    "Start of where display should be thick"
    uint thickEnd;      "End of where display should be thick"
    )
</PRE>

<H4><A name=ImageClone></A>ImageClone</H4>
<PRE>table imageClone
"Used with image consortium's cumulative_plate files: ftp://image.llnl.gov/image/outgoing"
    (
    uint id; 			"IMAGE clone ID"
    string library; 		"Clone collection (LLAM for amp-resistant libraries,
				 LLCM for chloramphenicol-resistant libraries, LLKM
				 for kanamycin-resistant libraries.  No rearray 
				 locations are given.)"
    uint plateNum; 		"Plate number"
    string row; 		"Row"
    uint column; 		"Column"
    uint libId; 		"IMAGE library ID"
    string organism; 		"Species"
    int numGenbank; 		"Number of GenBank records"
    string[numGenbank] genbankIds; "GenBank accession number(s)"
    )
</PRE>

<H4><A name=InterProXref></A>InterPro Cross Reference</H4>
<PRE>table interProXref
"A cross-reference table for InterPro."
        (
        string accession;       "SWISS-PROT accession number"
        string method;          "Method"
        int start;              "Start position"
        int end;                "End position"
        string interProId;      "InterPro ID"
        string description;     "Description"
        )
</PRE>

<H4><A name=JaxOrtholog></A>Jackson Lab Mouse Orthologs</H4>
<PRE>table jaxOrtholog
"Jackson Lab Mouse Orthologs"
    (
    string humanSymbol;	"Human HUGO Gene Nomenclature Committee symbol"
    string humanBand;   "Human chromosomal location"
    string mgiId;       "Mouse database id"
    string mouseSymbol; "Mouse human symbol"
    string mouseChr;    "Mouse chromosome"
    string mouseCm;     "Mouse genetic map positionin centimorgans"
    string mouseBand;   "Mouse chromosome band if any"
    )
</PRE>

<H4><A name=JaxQTL></A>Jackson Lab Quantitative Trait Loci</H4>
<PRE>table jaxQTL
"Quantitative Trait Loci from Jackson Lab/Mouse Genome Informatics"
    (
    string chrom;       "Human chromosome or FPC contig"
    uint   chromStart;  "Start position in chromosome"
    uint   chromEnd;    "End position in chromosome"
    string name;        "Name of item"
    uint   score;       "Score from 0-1000 (bed6 compat.)"
    char[1] strand;     "+ or - (bed6 compat.)"
    string marker;      "MIT SSLP Marker w/highest correlation"
    string mgiID;       "MGI ID"
    string description; "MGI description"
    float  cMscore;     "cM position of marker associated with peak LOD score"
    )
</PRE>

<H4><A name=KeggMapDesc></A>Kegg Pathway Map Description</H4>
<PRE>table keggMapDesc
"Description of KEGG pathway map"
   (
   string mapID;        "KEGG pathway map ID"
   string description;  "KEGG pathway map description"
   )
</PRE>

<H4><A name=KeggPathway></A>Kegg Pathway Cross Reference</H4>
<PRE>table keggPathway
"Associates KEGG pathway IDs with Known Genes and LocusLink IDs"
   (
   string kgID;         "Known Gene ID"
   string locusID;      "LocusLink ID"
   string mapID;        "KEGG pathway map ID"
   )
</PRE>

<H4><A name=KnownCanonical></A>Known Canonical</H4>
<PRE>table knownCanonical
"Describes the canonical splice variant of a gene"
     (
     string chom;        "Chromosome"
     int chromStart;     "Start position (0 based). Corresponds to txStart"
     int chromEnd;       "End position (non-inclusive). Corresponds to txEnd"
     int clusterId;      "Which cluster of transcripts this belongs to in knownIsoforms"
     string transcript;  "Corresponds to knownGene name field"
     string protein;     "SwissProt ID of associated protein"
     )
</PRE>

<H4><A name=KnownGene></A>Known Genes</H4>
<PRE>table knownGene
"Protein coding genes based on proteins from SWISS-PROT, TrEMBL, and 
TrEMBL-NEW and their corresponding mRNAs from GenBank"
    (
    string  name;               "Name of gene"
    string  chrom;              "Chromosome name"
    char[1] strand;             "+ or - for strand"
    uint    txStart;            "Transcription start position"
    uint    txEnd;              "Transcription end position"
    uint    cdsStart;           "Coding region start"
    uint    cdsEnd;             "Coding region end"
    uint    exonCount;          "Number of exons"
    uint[exonCount] exonStarts; "Exon start positions"
    uint[exonCount] exonEnds;   "Exon end positions"
    string  proteinID;          "SWISS-PROT ID" 
    string  alignID;            "Unique identifier for each (known gene, alignment 
                                 position) pair"
    )
</PRE>

<H4><A name=KgAlias></A>Known Genes Alias</H4>
<PRE>table kgAlias
"Links together a Known Gene ID and a gene alias"
    (
    string kgID;        "Known Gene ID"
    string alias;	"A gene alias"
    )
</PRE>

<H4><A name=KnownTo></A>Known Genes Associations</H4>
<P>
The following definition describes tables that associate Known Genes with other 
IDs and accessions.
<PRE>table knownTo
"Maps Known Gene to another ID"
    (
    string name;        "Same as name field in known gene"
    string value;       "Other id"
    )
</PRE>

<H4><A name=KgXref></A>Known Genes Cross Reference</H4>
<PRE>table kgXref
"Links a Known Gene ID with mRNA, SwissProt, RefSeq, and NCBI accessions/IDs"
    (
    string kgID;        "Known Gene ID"
    string mRNA;        "mRNA ID"
    string spID;        "SWISS-PROT protein Accession number"
    string spDisplayID; "SWISS-PROT display ID"
    string geneSymbol;  "Gene Symbol"
    string refseq;      "RefSeq ID"
    string protAcc;     "NCBI protein Accession number"
    string description; "Description"
    )
</PRE>

<H4><A name=KnownIsoforms></A>Known Genes Isoforms</H4>
<P>
The following definition describes tables that link together various transcripts 
of a gene into a cluster.

<PRE>table knownIsoforms
"Links together various transcripts of a gene into a cluster"
    (
    int clusterId;      "Unique id for transcript cluster (aka gene)"
    string transcript;  "Corresponds to name in knownGene table, transcript in 
			 knownCanonical"
    )
</PRE>

<H4><A name=KnownGeneLink></A>Known Genes Link</H4>
<PRE>table knownGeneLink
"Known Genes link table, currently storing DNA-based entries only"
    (
    string  name;       "Known Genes ID"
    char[1] seqType;    "Known Genes sequence type - mRNA-base genes=m, DNA-based genes=g"
    string  proteinID;  "Corresponding protein ID"
    )
</PRE>

<H4><A name=KgProtAlias></A>Known Genes Protein Alias</H4>
<PRE>table kgProtAlias
"Links together a Known Gene ID and a protein alias"
    (
    string kgID;        "Known Gene ID"
    string displayID;	"protein display ID"
    string alias;	"A protein alias"
    )
</PRE>

<H4><A name=KnownToSuper></A>Known Genes Superfamily Link</H4>
<PRE>table knownToSuper
"Maps protein superfamilies to known genes"
    (
    string gene;	"Known gene ID"
    int superfamily;	"Superfamily ID"
    int start;		"Start of superfamily domain in protein (0 based)"
    int end;		"End (noninclusive) of superfamily domain"
    float eVal;		"E value of superfamily assignment"
    )
</PRE>

<H4><A name=KgSpAlias></A>Known Genes Swiss-Prot Alias</H4>
<PRE>table kgSpAlias
"Links together a Known Gene ID and a protein alias"
    (
    string kgID;        "Known Gene ID"
    string spID;	"Swiss-Prot/TrEMBL protein accession number"
    string alias;	"Alias - could be either a gene alias or protein alias"
    )
</PRE>

<H4><A name=MoreKnown></A>Known (RefSeq) Genes IDs in Other Databases</H4>
Last used Oct. 2000 <BR><BR>
<PRE>table knownMore 
"Lots of auxiliary info about a known gene" 
    (  
    string name;	  "The name displayed in the browser:
                           OMIM, gbGeneName, or transId"   
    string transId;       "Transcript ID. Genie-generated ID" 
    string geneId;        "Gene (not transcript) Genie ID"   
    uint   gbGeneName;    "Connect to geneName table. GenBank gene name"   
    uint   gbProductName; "Connects to productName table. GenBank product name" 
    string gbProteinAcc;  "GenBank accession of protein"  
    string gbNgi;         "GenBank GI of nucleotide sequence" 
    string gbPgi;         "GenBank GI of protein sequence"   
    uint   omimId;        "OMIM ID or 0 if none"  
    string omimName;      "OMIM primary name"  
    uint   hugoId;        "HUGO Gene Nomeclature Committee (HGNC) ID, or 0 if none" 
    string hugoSymbol;    "HGNC short name"  
    string hugoName;      "HGNC descriptive name" 
    string hugoMap;       "HGNC Map position" 
    uint   pmId1;         "Data from HGNC nomeids.txt" 
    uint   pmId2;         "Data from HGNC nomeids.txt"
    string refSeqAcc;     "Accession of RefSeq mRNA" 
    string aliases;       "Aliases, if any.  Comma and space separated list" 
    uint   locusLinkId;   "Locus link ID" 
    string gdbId;         "NCBI GDB database ID" 
    )
</PRE>

<H4><A name=OtherKnown></A>Known (RefSeq) Genes - Other Information</H4>
Last used Oct. 2000
<PRE>table knownInfo
"Auxiliary info about a known gene"
    (
    string name;         "Connects with genieKnown-&gt;name"
    string transId;      "Transcript id. Genie-generated ID"
    string geneId;       "Gene (not transcript) ID"
    uint   geneName;     "Connect to geneName table"
    uint   productName;  "Connects to productName table"
    string proteinId;    "GenBank accession of protein?"
    string ngi;          "GenBank gi of nucleotide seq"
    string pgi;          "GenBank gi of protein seq"
    )
</PRE>

<H4><A name=MGCFullStatus></A>MGC Full Status</H4>
<PRE>table mgcFullStatus
"Status of MGC clone"
    (
    uint imageId;    	"Image ID for clone"
    enum status;	"MGC status code: unpicked, picked, notBack, noDecision,
			 fullLength, incomplete, chimeric, frameShift, contaminated,
			 retainedIntron, mixedWells, noGrowth, noInsert, no5est,
			 microDel, artifact, noPolyATail, or cantSequence. Status will
			 always be fullLength on main databases."
    char acc;		"GenBank accession"
    char organism;    	"Two-letter MGC organism" 
    )
</PRE>

<H4><A name=MGIid></A>MGI ID</H4>
<PRE>table MGIid
"Links MGI ID to Locus Link ID (Mouse)"
    (
    char[8]  LLid;       "Locus Link ID"
    char[12] MGIid;      "MGI ID"
    char[32] symbol;     "Symbol"
    )
</PRE>

<H4><A name=Vntr></A>Microsatellites from Gerome Breen's VNTR Program"</H4>
<PRE>table vntr
"Microsatellites from Gerome Breen's VNTR program"
    (
    string chrom;      		"chrom"
    uint   chromStart; 		"Start position in chromosome"
    uint   chromEnd;   		"End position in chromosome"
    string name;       		"Name of item (Repeat unit)"
    float  repeatCount; 	"Number of perfect repeats"
    int distanceToLast; 	"Distance to previous microsat. repeat"
    int distanceToNext; 	"Distance to next microsat. repeat"
    string forwardPrimer; 	"Forward PCR primer sequence (or Design_Failed)"
    string reversePrimer; 	"Reverse PCR primer sequence (or Design_Failed)"
    string pcrLength;     	"PCR product length (or Design_Failed)"
    )
</PRE>

<H4><A name=MouseSyn></A>Mouse Synteny</H4>
Last used Apr. 2001.
<PRE>table mouseSyn
"Synteny between mouse and human chromosomes."
    (
    string  chrom;       "Name of chromosome"
    uint    chromStart;  "Start in chromosome"
    uint    chromEnd;    "End in chromosome"
    string  name;        "Name of mouse chromosome"
    int     segment;     "Number of segment"
    )
</PRE>

<H4><A name=SyntenyMouse></A>Mouse Synteny Alignments</H4>
<PRE>table syntenyMouse
"Mouse synteny alignnments from blastz single coverage"
    (
    string chrom;       "Human chromosome"
    uint chromStart;    "Start on human chromosome"
    uint chromEnd;      "End on human chromosome"
    string name;  	"Mouse chromosome"
    uint score;         "Score (always zero)"
    char[1] strand;     "+ (direction matches) or - (opposite direction)"
    uint thickStart;    "Start of where display should be thick"
    uint thickEnd;      "End of where display should be thick"
    )
</PRE>

<H4><A name=MouseSynWhd></A>Mouse Synteny (Whitehead)</H4>
<PRE>table mouseSynWhd
"Whitehead synteny between mouse and human chromosomes."
    (
    string  chrom;       "Human chromosome"
    uint    chromStart;  "Start position in human chromosome"
    uint    chromEnd;    "End position in human chromosome"
    string  name;        "Name of mouse chromosome"
    uint    score;       "Unused (bed 6 compatibility)"
    char[1] strand;      "+ or - (orientation of fragment)"
    uint    mouseStart;  "Start position in mouse chromosome"
    unit    mouseEnd;    "End position in mouse chromosome"
    string  segLabel;    "Whitehead segment label"
    )
</PRE>

<H4><A name=MrnaPredictions></A>mRNA Sequences</H4>
<PRE>table mrnaPred
"A mRNA - linked to a gene."
    (
    string  name;       "Name of gene - same as in genePred"
    lstring seq;        "mRNA sequence"
    )
</PRE>

<H4><A name=MrnaRefSeq></A>mrnaRefSeq</H4>
<PRE>table mrnaRefseq
"Associates mRNA IDs with RefSeq IDs"
    (
    string  mrna;       "mRNA ID"
    string  refseq;     "RefSeq ID"
    )
</PRE>

<H4><A name=PslInfo></A>mRNA/EST/Blat Alignments</H4>
The following definition is used for the blat and blastz tables, EST tables, mRNA
tables, dbtssAli, and uniGene.<BR><BR>
<PRE>table psl  
"Summary info about a patSpace alignment" 
    ( 
    ushort  bin;          	  "A field to speed indexing" 
    uint    matches;      	  "Number of bases that match that aren't repeats"
    uint    misMatches;   	  "Number of bases that don't match"  
    uint    repMatches;   	  "Number of bases that match but are part of repeats" 
    uint    nCount;       	  "Number of 'N' bases" 
    uint    qNumInsert;   	  "Number of inserts in query" 
    int     qBaseInsert;  	  "Number of bases inserted in query" 
    uint    tNumInsert;   	  "Number of inserts in target" 
    int     tBaseInsert;  	  "Number of bases inserted in target"
    char[2] strand;       	  "+ or - for query strand. For translated alignments, 
				   second +/- is for genomic strand"  
    string  qName;        	  "Query sequence name"
    uint    qSize;        	  "Query sequence size"
    uint    qStart;       	  "Alignment start position in query"
    uint    qEnd;         	  "Alignment end position in query" 
    string  tName;        	  "Target sequence name" 
    uint    tSize;        	  "Target sequence size" 
    uint    tStart;       	  "Alignment start position in target" 
    uint    tEnd;         	  "Alignment end position in target" 
    uint    blockCount;   	  "Number of blocks in alignment (a block contains no gaps)" 
    uint[blockCount] blockSizes;  "Size of each block" 
    uint[blockCount] qStarts;     "Start of each block in query" 
    uint[blockCount] tStarts;     "Start of each block in target"
    )  
</PRE>
There is a little gotcha in this table format. It has to 
do with how coordinates are handled on the negative strand.
In the qStart/qEnd fields the coordinates are where it
matches from the point of view of the forward strand 
(even when the match is on the reverse strand). However
on the qStarts[] list, the coordinates are reversed. 
Here's an example of a 30-mer that has 2 blocks that
align on the minus strand and 2 blocks on the plus
strand (these types of situations happen in real life in
response to assembly errors sometimes). <BR><BR>
<TABLE CELLPADDING=0 CELLSPACING=0 MARGINS=0 BORDER=0>
<TR><TD><PRE>0         1         2         3 tens position in query   </TD></TR>
<TR><TD><PRE>0123456789012345678901234567890 ones position in query   </TD></TR>
<TR><TD><PRE>            ++++          +++++ plus strand alignment on query   </TD></TR>
<TR><TD><PRE>    --------    ----------      minus strand alignment on query   </TD></TR>
<TR><TD></TD></TD>	
<TR><TD><PRE>Plus strand:   </TD></TR>
<TR><TD><PRE>     qStart 12 qEnd 31 blockSizes 4,5 qStarts 12,26   </TD></TR>
<TR><TD><PRE>Minus strand:   </TD></TR>
<TR><TD><PRE>     qStart 4 qEnd 26 blockSizes 10,8 qStarts 5,19    </TD></TR>
</TABLE><BR>
Essentially the minus strand blockSizes and qStarts are 
what you would get if you reverse complemented the query.
However the qStart and qEnd are non-reversed. To get
from one to the other:
<PRE>
     qStart = qSize - revQEnd
     qEnd = qSize - revQStart
</PRE>

<H4><A name=NetAlign></A>Net Alignments</H4>
<PRE>table netAlign
"Database representation of a net of alignments"
   (
   uint level;       "Level of alignment"
   string tName;     "Target chromosome"
   uint tStart;      "Start on target"
   uint tEnd;        "End on target"
   char[1] strand;   "Orientation of query (+ or -)"
   string qName;     "Query chromosome"
   uint qStart;      "Start on query"
   uint qEnd;        "End on query"
   uint chainId;     "Associated chain ID with alignment details"
   uint ali;         "Bases in gap-free alignments"
   double score;     "Score - a number proportional to 100x matching bases"
   int qOver;        "Overlap with parent gap on query side (-1 for undefined)"
   int qFar;         "Distance from parent gap on query side ( -1 for undefined)"
   int qDup;         "Bases with two or more copies in query ( -1 for undefined)"
   string type;      "Syntenic type: gap/top/syn/nonsyn/inv"
   int tN;           "Unsequenced bases on target (-1 for undefined)"
   int qN;           "Unsequenced bases on query (-1 for undefined)"
   int tR;           "RepeatMasker bases on target (-1 for undefined)"
   int qR;           "RepeatMasker bases on query (-1 for undefined)"
   int tNewR;        "Lineage specific repeats on target (-1 for undefined)"
   int qNewR;        "Lineage specific repeats on query (-1 for undefined)"
   int tOldR;        "Bases of ancient repeats on target (-1 for undefined)"
   int qOldR;        "Bases of ancient repeats on query (-1 for undefined)"
   int tTrf;         "Bases of tandam repeats on target (-1 for undefined)"
   int qTrf;         "Bases of tandam repeats on query (-1 for undefined)"
   )
</PRE>

<H4><A name=Certificate></A>Non-standard Join Certificates</H4>
<PRE>table certificate
"Non-standard Join Certificates"
   (
   string accession1; 	"First accession number"
   string accession2; 	"Second accession number"
   string spanner; 	"Spanner"
   string evaluation; 	"Evaluation"
   string variation; 	"Variation"
   string varEvidence; 	"Variation evidence"
   string contact; 	"Contact info"
   string remark; 	"Remarks"
   string comment; 	"Comments"
   )
</PRE>

<H4><A name="OrientInfo"></A>Orientation Info</H4>
The following definition is used by the mrnaOrientInfo and estOrientInfo 
tables. It contains extra information on mRNAs/ESTs calculated by the 
polyInfo program.
<PRE>table estOrientInfo
"Extra information on mRNAs/ESTs calculated by polyInfo program"
   (
   int    bin;               "A field to speed indexing"
   uint   chromStart;        "Start position in chromosome"
   uint   chromEnd;          "End position in chromosome"
   string name;              "Accession of EST/mRNA"
   short intronOrientation;  "Orientation of introns with respect to EST/mRNA"
   short sizePolyA;          "Number of trailing A's"
   short revSizePolyA;       "Number of trailing A's on reverse strand"
   short signalPos;          "Position of start of polyA signal relative to end 
	                         of EST/mRNA, or 0 if no signal"
   short revSignalPos;       "PolyA signal position on reverse strand if any"
   )
</PRE>

<H4><A name=PfamDesc></A>Pfam Description</H4>
<PRE>table pfamDesc 
"Pfam description table"
   (
   string pfamAC;          "Pfam accession number"
   string pfamID;          "Pfam ID"
   string description;     "Description"
   )
</PRE>

<H4><A name=PfamXref></A>Pfam and Swiss-Prot Cross Reference</H4>
<PRE>table pfamXref
"A cross-reference table between Pfam ID and Swiss-Prot IDs."
   (
   string pfamAC;          "Pfam accession number"
   string swissAC;         "SWISS-PROT accession number"
   string swissDisplayID;  "SWISS-PROT display ID"
   )
</PRE>

<H4><A name=PbAaDistX></A>Proteome Browser Amino Acid Distribution</H4>
<PRE>table pbAaDistX
"Distribution for a specific amino acid X"
   (
   float x; "x value"
   float y; "count"
   )
</PRE>

<H4><A name=PbAnomLimit></A>Proteome Browser Amino Acid Anomaly Limits</H4>
<PRE>table pbAnomLimit
"Protein amino acid anomaly limits for each AA"
   (
   char[1] AA;          "Amino Acid"
   float   pctLow;      "Percentage (100%=1.0) Lower bound"
   float   pctHi;       "Percentage (100%=1.0) Upper bound"
   )
</PRE>

<H4><A name=PepCCntDist></A>Proteome Browser Cysteine Count Distribution</H4>
<PRE>table pepCCntDist
"Cysteine count distribution"
   (
   float x;             "Number of Cysteines"
   float y;             "Count of proteins with x Cysteines"
   )
</PRE>

<H4><A name=PepExonCntDist></A>Proteome Browser Exon Count Distribution</H4>
<PRE>table pepExonCntDist
"Exon count distribution"
   (
   float x;             "Number of exon"
   float y;             "Count of proteins with x exons"
   )
</PRE>

<H4><A name=PepHydroDist></A>Proteome Browser Hydrophobicity Distribution</H4>
<PRE>table pepHydroDist
"Hydrophobicity distribution"
   (
   float x;             "Hydrophobicity value"
   float y;             "Count of proteins with hydrophobicity near x"
   )
</PRE>

<H4><A name=PepIPCntDist></A>Proteome Browser InterProt Domain Count Distribution</H4>
<PRE>table pepIPCntDist
"InterProt domain count distribution"
   (
   float x;             "Number of InterPro domains"
   float y;             "Count of proteins with x InterPro domains" 
   )
</PRE>

<H4><A name=PepMwAa></A>Proteome Browser Molecular Weight and AA Length of Proteins</H4>
<PRE>table pepMwAa 
"Molecular weight and AA length of proteins"
   (
   string accession;    "SwissProt protein accession number"
   float  molWeight;    "Molecular weight" 
   int    aaLen;        "Length of protein sequence"
   )
</PRE>

<H4><A name=PepMolWtDist></A>Proteome Browser Molecular Weight Distribution</H4>
<PRE>table pepMolWtDist
"Molecular weight distribution"
   (
   float x;             "Molecular weight value"
   float y;             "Count of proteins with molecular weight near x"
   )
</PRE>

<H4><A name=PepPiDist></A>Proteome Browser pI Distribution</H4>
<PRE>table pepPiDist
"pI distribution" 
   (
   float x;             "pI  value" 
   float y;             "Count of proteins with pI value near x"
   )
</PRE>

<H4><A name=PepPred></A>Proteome Browser Predicted Peptide/Predicted Gene</H4>
<PRE>table pepPred
"Predicted peptide linked to predicted gene"
    (
    string name;        "Name of gene - same as in genePred"
    lstring seq;        "Peptide sequence" 
    )
</PRE>

<H4><A name=PepPi></A>Proteome Browser Protein pIs</H4>
<PRE>table pepPi
"Protein pIs" 
   (
   string accession;     "Protein accession number" 
   float pI;             "pI value"
   )
</PRE>

<H4><A name=PbResAvgStd></A>Proteome Browser Residue Average/Standard Deviation</H4>
<PRE>table pbResAvgStd
"Residue average and standard deviation"
   (
   char[1] residue;     "Protein residue"
   float   avg;         "Average"
   float   stddev;      "Standard deviation"
   )
</PRE>

<H4><A name=PepResDist></A>Proteome Browser Residue Distribution</H4>
<PRE>table pepResDist
"Residue distribution" 
   (
   float x;             "Residue index (WCMHYNFIDQKRTVPGEASL) "
   float y;             "Count of proteins with residue index of x"
   )
</PRE>

<H4><A name=PbStamp></A>Proteome Browser Stamp Information</H4>
<PRE>table pbStamp
"Info needed for a Proteome Browser stamp"
    (
    char[40] stampName; 	"Short Name of stamp"
    char[40] stampTable;	"Database table name of the stamp (distribution) data"
    char[40] stampTitle;	"Stamp Title to be displayed"
    int len;            	"Number of x-y pairs"
    float  xmin;        	"x minimum"
    float  xmax;        	"x maximum"
    float  ymin;        	"y minimum"
    float  ymax;        	"y maximum"
    string stampDesc;   	"Description of the stamp"
    )
</PRE>

<H4><A name=PdbSP></A>PDB and Swiss-Prot ID links</H4>
<PRE>table pdbSP
"A cross-reference table between Swiss-Prot accession IDs and PDB IDs."
    (
    string pdb;             "PDB ID"
    string sp;              "SWISS-PROT display ID"
    )
</PRE>

<H4><A name=Pscreen></A>P-Screen Data (BDGP Gene Disruption Project)</H4>
<PRE>table pscreen
"P-Screen (BDGP Gene Disruption Project) P el. insertion locations/genes"
    (
    string chrom;      		"Chromosome"
    uint   chromStart; 		"Start position in chromosome"
    uint   chromEnd;   		"End position in chromosome"
    string name;       		"Name of item (mutant strain with P el. insert here)"
    uint   score;      		"Score from 0-1000 (placeholder! for bed 6 compat)"
    char[1] strand;    		"+ or -"
    uint stockNumber;  		"Mutant strain stock number, for ordering"
    uint geneCount;    		"Number of genes disrupted by this insert"
    string[geneCount] geneIds;  "IDs of disrupted genes"
    )
</PRE>

<H4><A name=PseudoGeneLink></A>Pseudogene Link to Ortholog/Paralog</H4>
<PRE>
table pseudoGeneLink
"Links a gene/pseudogene prediction to an ortholog or paralog."
    (
    string chrom;       	 "Chromosome name for pseudogene"
    uint chromStart;    	 "Pseudogene alignment start position"
    uint chromEnd;      	 "Pseudogene alignment end position"
    string name;        	 "Name of pseudogene"
    uint score;         	 "Score of pseudogene with gene"
    char[2] strand;     	 "+ or -"
    uint thickStart;    	 "Start of where display should be thick (start codon)"
    uint thickEnd;      	 "End of where display should be thick (stop codon)"
    uint reserved;      	 "Always zero for now"
    int blockCount;     	 "Number of blocks (a block contains no gaps)"
    int[blockCount] blockSizes;  "Comma-separated list of block sizes"
    int[blockCount] chromStarts; "Start positions relative to chromStart"
    float trfRatio;     	 "Ratio of tandem repeats"
    string type;        	 "Type of evidence"
    int axtScore;       	 "Blastz score, gene mRNA aligned to pseudogene"
    string gChrom;      	 "Chromosome name"
    int gStart;         	 "Gene alignment start position"
    int gEnd;           	 "Gene alignment end position"
    char[2] gStrand;    	 "Strand of gene"
    uint exonCount;     	 "# of exons in gene "
    uint geneOverlap;   	 "Bases overlapping"
    uint polyA;         	 "Length of polyA"
    int polyAstart;     	 "Start of polyA relative to end of pseudogene"
    uint exonCover;     	 "Number of exons in gene covered"
    uint intronCount;   	 "Number of introns in pseudogene"
    uint bestAliCount;  	 "Number of good mRNAs aligning"
    uint matches;       	 "Matches + repMatches"
    uint qSize;         	 "Aligning bases in pseudogene"
    uint qEnd;          	 "End of cDNA alignment"
    uint tReps;         	 "Repeats in gene"
    uint qReps;         	 "Repeats in pseudogene"
    uint overlapDiag;   	 "Bases on the diagonal to mouse"
    uint coverage;      	 "Bases on the diagonal to mouse"
    int label;          	 "1=pseudogene,-1 not pseudogene"
    uint milliBad;      	 "MilliBad score, pseudogene aligned to genome"
    uint oldScore;      	 "Another heuristic"
    int oldIntronCount; 	 "Old simple intron count"
    int conservedIntrons; 	 "Conserved intron count"
    string intronScores; 	 "Intron sizes in gene/pseudogene"
    int maxOverlap ;    	 "Largest overlap with another mRNA"
    string refSeq;      	 "Name of closest RefSeq to gene"
    int rStart;         	 "RefSeq alignment start position"
    int rEnd;           	 "RefSeq alignment end position"
    string mgc;         	 "Name of closest MGC to gene"
    int mStart;         	 "MGC alignment start position"
    int mEnd;           	 "MGC alignment end position"
    string kgName;      	 "Name of closest Known Gene to gene"
    int kStart;         	 "Known Gene alignment start position"
    int kEnd;           	 "Known Gene alignment end position"
    string overName;    	 "Name of overlapping mrna"
    int overStart;      	 "Overlapping mRNA start position"
    int overEnd;       		 "Overlapping mRNA end position"
    char[2] overStrand;  	 "Strand of overlapping mRNA"
    int adaBoost;       	 "AdaBoost label"
    float posConf;      	 "P-value for positive"
    float negConf;      	 "P-value for positive"
    )
</PRE>

<H4><A name=RankProp></A>RankProp Score</H4>
<PRE>table rankProp
"RankProp protein ranking for a pair of proteins"
    (
    string query;       "Known Genes ID of query protein"
    string target;      "Known Genes ID of target protein"
    float score;        "Rankp score"
    )
</PRE>

<H4><A name=SyntenyRat></A>Rat Synteny Alignments </H4>
<PRE>table syntenyRat
"Rat synteny alignnments from blastz single coverage"
    (
    string chrom;       "Human chromosome"
    uint chromStart;    "Start on human chromosome"
    uint chromEnd;      "End on human chromosome"
    string name;    	"Rat chromosome"
    uint score;         "Score (always zero)"
    char[1] strand;     "+ (direction matches) or - (opposite direction)"
    uint thickStart;    "Start of where display should be thick"
    uint thickEnd;      "End of where display should be thick"
    )
</PRE>

<H4><A name=RecombRate></A>Recombination Rate</H4>
<PRE>table recombRate
"Describes the recombination rate in 1Mb intervals based on deCODE, Marshfield and
 Genethon maps"
    (
    string  chrom;            "Chromosome number"
    uint    chromStart;       "Start position in genoSeq"
    uint    chromEnd;         "End position in genoSeq"
    string  name;             "Constant string 'recombRate'"
    float   decodeAvg;        "Calculated deCODE Sex-averaged rate from map"
    float   decodeFemale;     "Calculated deCODE female recombination rate"
    float   decodeMale;       "Calculated deCODE male recombination rate"
    float   marshfieldAvg;    "Calculated Marshfield recombination rate"
    float   marshfieldFemale; "Calculated Marshfield female recombination"
    float   marshfieldMale;   "Calculated Marshfield male recombination rate"
    float   genethonAvg;      "Calculated Genethon recombination rate"
    float   genethonFemale;   "Calculated Genethon female recombination rate"
    float   genethonMale;     "Calculated Genethon male recombination rate"
    )
</PRE>

<H4><A name=RefLink></A>RefSeq Link</H4>
First used Dec. 2000
<PRE>table refLink
"Link between RefSeq mRNAs and HUGO Gene Nomenclature Committee, LocusLink etc."
    (
    string name;        "Name displayed in UI"
    string product;     "Name of protein product"
    string mrnaAcc;     "mRNA accession"
    string protAcc;     "Protein accession"
    uint geneId;        "Pointer to geneName table"
    uint prodId;        "Pointer to prodName table"
    uint locusLinkId;   "Locus Link ID"
    uint omimId;        "OMIM ID"
    )
</PRE>

<H4><A name=RefSeqStatus></A>RefSeq Status</H4>
First used Dec. 2001
<PRE>table refSeqStatus
"Links RefSeq mRNA accessions with status"
    (
    string mrnaAcc;     "RefSeq mRNA accession"
    string status;      "RefSeq status (Reviewed, Provisional, Predicted)"
    )
</PRE>

<H4><A name=RefSeqSummary></A>RefSeq Summary</H4>
First used Oct. 2003
<PRE>table refSeqStatus
"Sparse summary table created for refSeqs with "Summary:" or "COMPLETENESS:" in comment"
    (
    string mrnaAcc; 	"RefSeq mRNA accession"
    enum completeness; 	"Level of completeness: Unknown, Complete5End, 
			Complete3End, FullLength, IncompleteBothEnds, 
			Incomplete5End, Incomplete3End, Partial"
    string summary;	"Summary text"
    )
</PRE>

<H4><A name=EsRegGeneToMotif></A>Regulatory Module TFBS Predictions (Eran Segal)</H4>
<PRE>table esRegGeneToMotif
"Browser extensible data with gene field"
    (
    string    chrom;      "Chromosome"
    uint      chromStart; "Start position in chromosome"
    uint      chromEnd;   "End position in chromosome"
    string    name;       "Motif name"
    uint      score;      "Score from 0-1000"
    char[1]   strand;     "+ or -"
    varchar   gene;    	  "Gene name"
    )
</PRE>

<H4><A name=RepeatInfo></A>Repeating Elements</H4>
Table created from RepeatMasker .out files.
<PRE>table rmsk
"RepeatMasker .out record"
    (
    uint swScore;       "Smith Waterman alignment score"
    uint milliDiv;      "Base mismatches in parts per thousand"
    uint milliDel;      "Bases deleted in parts per thousand"
    uint milliIns;      "Bases inserted in parts per thousand"
    string genoName;    "Genomic sequence name"
    uint genoStart;     "Start in genomic sequence"
    uint genoEnd;       "End in genomic sequence"
    int  genoLeft;      "Size left in genomic sequence"
    char[1] strand;     "Relative orientation + or -"
    string repName;     "Name of repeat"
    string repClass;    "Class of repeat"
    string repFamily;   "Family of repeat"
    int repStart;       "Start in repeat sequence"
    uint repEnd;        "End in repeat sequence"
    int repLeft;        "Size left in repeat sequence"
    char[1] id;         "First digit of id field in RepeatMasker .out file. Best ignored."
    )
</PRE>

<H4><A name=RikenAltid></A>Riken Altid Table</H4>
<PRE>table rikenaltid
"Riken altid table"
    (
    char[64]    ID;             "ID (composite)"
    char[32]    SeqID;          "Sequence ID"
    char[32]    AltIDType;      "Alternate ID type"
    string      AltID;          "Alternate ID"
    )
</PRE>

<H4><A name=RikenAnn></A>Riken Annotation Table</H4>
<PRE>table rikenann
"Riken annotation table"
    (
    string  ID;             "ID"
    string  seqid;          "parent sequence id"
    uint    i;              "sequential number"
    string  Qualifier;      "Qualifier"
    string  Anntext;        "Anntext"
    string  Datasrc;        "Datasrc"
    string  Srckey;         "Srckey"
    string  Href;           "xlink_href"
    string  Evidence;       "Evidence"
    )
</PRE>

<H4><A name=RikenCluster></A>Riken Cluster Table</H4>
<PRE>table rikencluster
"Riken cluster table"
    (
    string  ID;                     "ID"
    string  Fantomid;               "Famtom"
    string  RepresentativeSeqid;    "Representative sequence ID"
    uint    Nsequence;              "Number of Sequencesse"
    )
</PRE>


<H4><A name=RikenClusterSeq></A>Riken Cluster Sequence Regulation Table</H4>
<PRE>table rikenclusterseq
"Riken cluster sequence regulation table"
    (
    string  ClusterID;              "Cluster ID"
    string  SeqID;                  "Sequence ID"
    )
</PRE>


<H4><A name=RikenSeq></A>Riken Sequence Table</H4>
<PRE>table rikenseq
"Riken sequence table"
    (
    string  ID;             "ID"
    uint    Naltid;         "Number of altid"
    string  Seqid;          "sequence ID (no more valid per Riken DTD)"
    string  Fantomid;       "Fantom ID (no more valid per Riken DTD)"
    string  Cloneid;        "Clone ID (no more valid per Riken DTD)"
    string  Rearrayid;      "Rearray ID (no more valid per Riken DTD)"
    string  Accession;      "Accession (no more valid per Riken DTD)"
    string  Annotator;      "Annotator"
    string  version;        "Version"
    string  ModifiedTime;   "Modified Time"
    uint    Nannotation;    "Number of Annotations"
    string  Comment;        "Comment"
    )
</PRE>

<H4><A name=RnaFold></A>RNA Folding Info</H4>
<PRE>table rnaFold
"Info about folding of RNA into secondary structure"
    (
    string name;	"mRNA accession"
    lstring seq;	"mRNA sequence (U's instead of T's)"
    lstring fold;	"Parenthesis and .'s that describe folding"
    float energy;	"Estimated free energy of folding (negative)"
    )
</PRE>

<H4><A name=RNAgenes></A>RNA Gene Positions</H4>
Last used Apr. 2001.
<PRE>table rnaGene
"Describes functional RNA genes."
    (
    string  chrom;        "Chromosome gene is on"
    uint    chromStart;   "Start position in chromosome"
    uint    chromEnd;     "End position in chromosome"
    string  name;         "Name of gene"
    uint    score;        "Score from 0 to 1000"
    char[1] strand;       "Strand: + or -"
    string  source;       "Source as in Sean Eddy's files."
    string  type;         "Type - snRNA, rRNA, tRNA, etc."
    float   fullScore;    "Score as in Sean Eddy's files."
    ubyte   isPsuedo;     "TRUE(1) if psuedo, FALSE(0) otherwise"
    )
</PRE>

<H4><A name=WgRna></A>RNA - Weber and Griffiths-Jones</H4>
<PRE>table wgRna
"CD and H/ACA Box snoRNAs and microRNAs from Weber and Griffiths-Jones"
    (
    string  chrom;        "Chromosome"
    uint    chromStart;   "Start position in chromosome"
    uint    chromEnd;     "End position in chromosome"
    string  name;         "Name of item"
    uint    score;        "Score from 0 to 1000 (bed6 compat.)"
    char[1] strand;       "+ or - (bed6 compat.)"
    uint    thickStart;   "Start of thick region"
    uint    thickEnd;     "End of thick region"
    string  type;         "RNA type"
    }
</PRE>

<H4><A name=RosettaExps></A>Rosetta Information </H4>
<PRE>table rosettaExps
"Rosetta Experimental Confirmation information"
    (
    int	      ID;        	
    string    name;   	
    longblob  description;
    longblob  URL;       
    longblob  ref;      
    longblob  credit;  
    uint      numExtras;
    longblob  extra;   
    )                                                                         
</PRE>

<H4><A name=SgdAbundance></A>Saccharomyces Genome Database Protein Abundance</H4>
<PRE>table sgdAbundance
"Protein abundance data from http://yeastgfp.ucsf.edu via SGD"
    (
    string name;        "ORF name in sgdGene table"
    float abundance;    "Absolute abundance from 41 to 1590000"
    string error;       "Error - either a floating point number or blank"
    )
</PRE>

<H4><A name=SgdClone></A>Saccharomyces Genome Database Clone Information</H4>
<PRE>table sgdClone
"Clone info from yeast genome database"
    (
    string chrom;       "Chromosome in chrN format"
    uint   chromStart;  "Start position in chromosome"
    uint   chromEnd;    "End position in chromosome"
    string name;        "Washington University name"
    string atccName;    "ATCC clone name (optional)"
    )
</PRE>

<H4>Saccharomyces Genome Database Genes Description</H4>
<PRE>table sgdDescription
"Description of SGD Genes and Other Features"
    (
    string  name;        "Name in sgdGene or sgdOther table"
    string type;         "Type of feature from gff3 file"
    lstring description; "Description of feature"
    )
</PRE>

<H4><A name=SgdOther></A>Saccharomyces Genome Database Other Features</H4>
<PRE>table sgdOther
"Features other than coding genes from yeast genome database"
    (
    string chrom;       "Chromosome in chrNN format"
    int chromStart;     "Start (zero based)"
    int chromEnd;       "End (non-inclusive)"
    string name;        "Feature name"
    int score;          "Always 0"
    char[1] strand;     "Strand: +, - or ."
    string type;        "Feature type"
    )
</PRE>

<H4><A name=Sage></A>Sage</H4>
<PRE>table sage
"Stores SAGE data in terms of UniGene identifiers"
    (
    int uni;               "Number portion of UniGene identifier"
    string gb;             "GenBank accession number"
    string gi;             "gi field in UniGene descriptions"
    lstring description;   "Description from UniGene fasta headers"
    int numTags;           "Number of tags"
    string[numTags] tags;  "Tags for this unique sequence"
    int numExps;           "Number of experiments"
    int[numExps] exps;     "Index of experiemtns in order of aves and stdevs"
    float[numExps] meds;   "Meida count of all tags for each experiment"
    float[numExps] aves;   "Average count of all tags for each experiment"
    float[numExps] stdevs  "Standard deviations of all counts for each experiment"
    )
</PRE>

<H4><A name=SageExp></A>Sage Experiments</H4>
<PRE>table sageExp
"Data related to SAGE experiments, tissue descriptions, etc."
    (
    int num;               "Index of the experiment in the sage table"
    string exp;            "Experiment name"
    int totalCount;        "Sum of all the tag counts for this experiment"
    string tissueType;     "Brief description of cells used"
    string tissueDesc;     "Brief description of tissues"
    string tissueSupplier; "Who supplied the tissue"
    string organism;       "Organism source of cells"
    string organ;          "Organ identifier"
    lstring producer;      "Source of tissue"
    lstring desription;    "Description of experiment"
    )
</PRE>

<H4><A name=ProtHomolog></A>SAM Protein Homolog Data</H4>
<PRE>table protHomolog
"table to store SAM homolog results"
    (
    string  proteinID;  	"Protein ID"
    string  homologID;  	"Homolog ID"
    char[1] charin;     	"Chain"
    int     length;     	"Length of protein sequence"
    double  bestEvalue; 	"Best E-value"
    double  evalue;     	"E-value"
    string  FSSP;       	"FSSP ID"
    string  SCOPdomain; 	"SCOP domain ID"
    string  SCOPsuid;   	"SCOP sunid"
    )
</PRE>

<H4><A name=SamSubdir></A>SAM Data Subdirectory Linked to Known Gene ID</H4>
<PRE>table samSubdir
"Link together a Known Gene ID and a gene alias"
    (
    string proteinId;   "Protein ID"
    string subdir;      "Name of the subdirectory where SAM results are"
    )
</PRE>

<H4><A name=WiggleTrack></A>Sample Tracks for Plotting Real-valued Data</H4>
<PRE>table sample9
"Describes (x,y) pairs of samples"
   {
    ushort bin;                         "A field to speed indexing added to many
                                         tables August 2001"
    string  chrom;                      "Chromosome alignment is on."
    uint    chromStart;                 "Start position in chromosome"
    uint    chromEnd;                   "End position in chromosome"
    string  name;                       "Unique index given to each block of samples"
    uint    score;                      "Overall block score from 0 to 1000 (chrN_humMusL 
					 only)"
    char[1] strand;                     "Strand: + or -"
    uint    sampleCount                 "Number of samples in this block"
    uint[sampleCount]   samplePosition  "Base position of each sample in this
                                         block, relative to chromStart"
    uint[sampleCount]   sampleHeight    "Score for each sample (y-value) from 0 to 1000"
    }
</PRE>

<H4><A name=WiggleTrack2></A>Sample Tracks for Plotting Real-valued Data (Rev 2)</H4>
<PRE>table wiggle
"Wiggle track values to display as y-values (first 6 fields are bed6)"
    (
    string chrom;         "Chromosome or FPC contig"
    uint chromStart;      "Start position in chromosome"
    uint chromEnd;        "End position in chromosome"
    string name;          "Name of item"
    uint span;            "Each value spans this many bases"
    uint count;           "Number of values in this block"
    uint offset;          "Offset in file to fetch data"
    string file;          "Path name to data file, one byte per value"
    double lowerLimit;    "Lowest data value in this block"
    double dataRange;     "LowerLimit + dataRange = upperLimit"
    uint validCount;      "Number of valid data values in this block"
    double sumData;       "Sum of the data points, for average and stddev calc"
    double sumSquares;    "Sum of data points squared, for stddev calc"
    )
</PRE>

<H4><A name=SangerExtra></A>Sanger 20 and 22 Genes - Additional Information</H4>
<PRE>table sanger22extra
"Table with additional information about a Sanger 22 gene"
    (
    string name;	  "Transcript name"
    string locus;	  "Possibly biological short name"
    lstring description;  "Description from Sanger gene GFFs"
    string geneType;	  "Type field from Sanger gene GFFs"
    string cdsType;	  "Type field from Sanger CDS GFFs"
    )
</PRE>

<H4><A name=SangerGene></A>Sanger Genes (WormBase)</H4>
<PRE>table sangerGene
"GenePred table with proteinID field for WormBase Genes."
    (
    string name;        	"Name of gene"
    string chrom;       	"Chromosome name"
    char[1] strand;     	"+ or - for strand"
    uint txStart;       	"Transcription start position"
    uint txEnd;         	"Transcription end position"
    uint cdsStart;      	"Coding region start"
    uint cdsEnd;        	"Coding region end"
    uint exonCount;     	"Number of exons"
    uint[exonCount] exonStarts; "Exon start positions"
    uint[exonCount] exonEnds;   "Exon end positions"
    string proteinID;   	"Swiss-Prot protein ID"
    )
</PRE>

<H4><A name=ScopDes></A>SCOP Descriptions</H4>
<PRE>table scopDes
"Structural Classification of Proteins description"
    (
    int sunid;  		"Unique integer"
    char[2] type;       	"Type.  sf=superfamily, fa = family, etc."
    string sccs;        	"Dense hierarchy info."
    string sid;         	"Older ID."
    lstring description; 	"Descriptive text."
   )
</PRE>

<H4><A name=ScoredRef></A>Scored Ref</H4>
<PRE>table scoredRef
"A score, a range of positions in the genome and an extFile offset"
   (
   int bin;          	"Field used to speed indexing"  
   string chrom;      	"Chromosome (this species)"
   uint   chromStart; 	"Start position in chromosome (forward strand)"
   uint   chromEnd;   	"End position in chromosome"
   uint   extFile;	"Pointer to associated external file"
   bigint offset;	"Offset in external file"
   float score;      	"Value between 0.0 and 1.0 that loosely reflects % identity ratio. 
                      	 Derived from multiz score by dividing it by the number of pairwise 
                      	 comparisons in the multiple alignment and by the number of bases 
                      	 in the alignment."
   )
</PRE>

<H4><A name=GenomicSuperDups></A>Segmental Duplications</H4>
<PRE>table genomicSuperDups
"Summary of large genomic Duplications (>1KB >90% similar)"
  (
  string chrom;       	"Chromosome or FPC contig"
  uint chromStart;   	"Start position in chromosome"
  uint chromEnd;     	"End position in chromosome"
  string name;         	"Other chromosome involved"
  uint score;        	"Score from 900-1000.  1000 is best"
  char[1] strand;       "Value should be + or -"
  string otherChrom;   	"Other chromosome or FPC contig"
  uint otherStart;   	"Start in other  sequence"
  uint otherEnd;     	"End in other  sequence"
  uint otherSize;    	"Total size of other sequence"
  uint uid;       	"Unique ID"
  uint posBasesHit;     "HitPositive UnCovered"
  string testResult;    "HitPositive (yes or no) UnCovered (covered=0)"
  string verdict;       "Real or Allele"
  string alignfile;     "Alignment file path"
  uint alignL;      	"Spaces/positions in alignment"
  uint indelN;       	"Number of indels"
  uint indelS;      	"Indel spaces"
  uint alignB;       	"Bases Aligned"
  uint matchB;      	"Aligned bases that match"
  uint mismatchB;    	"Aligned bases that do not match"
  uint transitionsB;    "Number of transitions"
  uint transversionsB;  "Number of transversions"
  float fracMatch;    	"Fraction of matching bases"
  float fracMatchIndel; "Fraction of matching bases with indels"
  float jcK;          	"K-value calculated with Jukes-Cantor"
  float k2K;          	"Kimura K"
  )
</PRE>

<H4><A name=Seq></A>Seq</H4>
<PRE>table seq
"Locations of database table sequences referenced in external fasta files"
   (
   int id; 		  "Sequence ID"
   string acc; 		  "Accession or name of sequence"
   int size;		  "Size of sequence"
   date gb_date;	  "Modification date for genbank mRNAs/ESTs"
   int extFile;		  "ID of file in extFile table"
   int file_offset;	  "Byte offset of sequence start in fasta file"
   int file_size;	  "Size of sequence record in fasta file (includes fasta 
			   sequence header"
   )                                                                           
</PRE>

<H4><A name=SimpleRepeats></A>Simple Repeats</H4>
This table was generated by the trf program.  It does a more thorough job 
of finding simple tandem repeats than RepeatMasker does.
<PRE>table simpleRepeat
"Describes the Simple Tandem Repeats"
   (
   ushort  bin;           "A field to speed indexing"
   uint    matches;       "Number of bases that match that aren't repeats"
   uint    misMatches;    "Number of bases that don't match"
   string chrom;          "Chromosome or FPC contig"
   uint   chromStart;     "Start position in chromosome"
   uint   chromEnd;       "End position in chromosome"
   string name;           "Simple Repeats tag name"
   uint   period;         "Length of repeat unit"
   float  copyNum;        "Mean number of copies of repeat"
   uint   consensusSize;  "Length of consensus sequence"
   uint   perMatch;       "Percentage Match"
   uint   perIndel;       "Percentage Indel"
   uint   score;          "Score between  and .  Best is ."
   uint   A;              "Number of A's in repeat unit"
   uint   C;              "Number of C's in repeat unit"
   uint   G;              "Number of G's in repeat unit"
   uint   T;              "Number of T's in repeat unit"
   float  entropy;        "Entropy"
   string sequence;       "Sequence of repeat unit element"
   )
</PRE>

<H4><A name=Snp></A>SNP Map (combined table of SNPs, hg17 and later)</H4>
<PRE>table snp
"Polymorphism data from dbSnp XML files or genotyping arrays"
    (
    string  chrom;      "Chromosome"
    uint    chromStart; "Start position in chrom"
    uint    chromEnd;   "End position in chrom"
    string  name;       "Reference SNP identifier or Affy SNP name"
    uint    score;      "Not used"
    char[1] strand;     "Which DNA strand contains the observed alleles"
    string  observed;   "The sequences of the observed alleles"
    string  molType;    "Sample type from exemplar ss"
    string  class;      "The class of variant"
    string  valid;      "The validation status of the SNP"
    float   avHet;      "The average heterozygosity from all observations"
    float   avHetSE;    "The Standard Error for the average heterozygosity"
    string  func;       "The functional category of the SNP"
    string  locType;    "How the variant affects the reference sequence"
    string  source;     "Source of the data - dbSnp, Affymetrix, ..."
    string  exception;  "List of exceptionIds for 'invariant' conditions"
    )
</PRE>

<H4><A name=SnpExceptions></A>SNP Exceptions</H4>
<PRE>table snpExceptions
"Set of queries to look for snps that appear problematic"
    (
    uint    exceptionId; "Unique ID for this exception"
    string  query;       "SQL string to retrieve bad records"
    uint    num;         "Count of SNPs that fail this condition"
    string  description; "Text string for readability"
    string  resultPath;  "Path for results file"
    )
</PRE>

<H4><A name=SnpMap></A>SNP Map (combined table of SNPs, hg13-hg16)</H4>
<PRE>table snpMap
"SNP positions from various sources"
    (
    string chrom;       "Chromosome or 'unknown'"
    uint   chromStart;  "Start position in chrom"
    uint   chromEnd;    "End position in chrom"
    string name;        "Name of SNP - rsId or Affy name"
    string source;      "BAC_OVERLAP | MIXED | RANDOM | OTHER | Affy10K | Affy120K
"
    string type;        "SNP | INDEL | SEGMENTAL"
    )
</PRE>

<H4><A name=SoftberryHom></A>Softberry Genes - Protein Homologies</H4>
<PRE>table softberryHom
"Protein homologies behind Softberry genes"
    (
    string name;          "Softberry gene name"
    string giString;      "String with GenBank gi and accession"
    lstring description;  "Freeform (except for no tabs) description"
    )
</PRE>

<H4><A name=LFS></A>Standard Linked Features</H4>
The phMouse and genscanSubopt tables use the first 7 fields of this table.
The bacEndPairs table uses all fields.
<PRE>table lfs
"Standard linked features series table"
   (
   short  bin;                  "Bin number for browser speedup"
   string chrom;                "Chromosome or FPC contig"
   uint   chromStart;           "Start position of clone in chromosome"
   uint   chromEnd;             "End position of clone in chromosome"
   string name;                 "Name of clone"

   uint   score;                "Score = 1000/(# of times clone appears in assembly)"
   char[1] strand;              "Value should be + or -"
   string pslTable;             "Table which contains corresponding PSL records for 
				 linked features"
   uint   lfCount;              "Number of linked features in the series"
   uint[lfCount]   lfStarts;    "Comma-separated list of start positions of each linked 
				 feature in genomic"
   uint[lfCount]   lfSizes;     "Comma-separated list of sizes of each linked feature 
				 in genomic"
   string[lfCount] lfNames;     "Comma-separated list of names of linked features"
   )
</PRE>

<H4><A name=STSsites></A>STS and FISH Clone Positions</H4>
(obsolete as of Apr. 2001, replaced by <A href="#STSmap">stsMap</A>)
<PRE>table stsMarker  
"STS marker and its position on golden path and various maps"
    ( 
    string chrom;            "Chromosome or 'unknown'"
    int    chromStart;       "Start position in chrom (-1 if unpositioned)"
    uint   chromEnd;         "End position in chrom" 
    string name;             "Name of STS marker"  
    uint   score;            "Score of a marker - depends on how many contigs it hits"  
    uint   identNo;          "Identification number of STS" 
    string ctgAcc;           "Contig accession number
    string otherAcc;         "Accession number of other contigs that the marker hits"  
    string genethonChrom;    "Chromosome from Genethon map or 0 if none" 
    float  genethonPos;      "Position on Genethon map"
    string marshfieldChrom;  "Chromosome from Marshfield map or 0 if none"
    float  marshfieldPos;    "Position on Marshfield map" 
    string gm99Gb4Chrom;     "Chromosome from GeneMap99 map or 0 if none"
    float  gm99Gb4Pos;       "Position on GeneMap99 map" 
    string shgcG3Chrom;      "Chromosome from G3 map or 0 if none"
    float  shgcG3Pos;        "Position on G3 map" 
    string wiYacChrom;       "Chromosome from Whitehead YAC map or 0 if none"
    float  wiYacPos;         "Position on Whitehead YAC map" 
    string shgcTngChrom;     "Chromosome from TNG map or 0 if none" 
    float  shgcTngPos;       "Position on TNG map" 
    string fishChrom;        "Chromosome from FISH map or 0 if none" 
    string beginBand;        "Beginning of range of bands on FISH map"
    string endBand;          "End of range of bands on FISH map"
    string lab;              "Laboratory that placed the FISH clone"
    )             
</PRE>

<H4><A name=STSmap></A>STS Clone Positions</H4>
<PRE>table stsMap
"STS marker and its position on golden path and various maps"
    (
    string chrom;	        "Chromosome or 'unknown'"
    int chromStart;             "Start position in chrom - negative 1 if unpositioned"
    uint chromEnd;        	"End position in chrom"
    string name;	        "Name of STS marker"
    uint score;	                "Score of a marker = 1000/(# of placements)"
    uint identNo;        	"Identification number of STS"
    string ctgAcc;	        "Contig accession number"
    string otherAcc;	        "Accession number of other contigs that the marker hits"

    string genethonChrom;	"Chromosome (no chr) from Genethon map or 0 if none"
    float genethonPos;		"Position on Genethon map"
    string marshfieldChrom;	"Chromosome (no chr) from Marshfield map or 0 if none"
    float marshfieldPos;	"Position on Marshfield map"
    string gm99Gb4Chrom;	"Chromosome (no chr) from GeneMap99 map or 0 if none"
    float gm99Gb4Pos;		"Position on gm99_bg4 map"
    string shgcTngChrom;	"Chromosome (no chr) from shgc_tng map or 0 if none"
    float shgcTngPos;		"Position on shgc_tng map"
    string shgcG3Chrom;		"Chromosome (no chr) from Stanford G3 map or 0 if none"
    float shgcG3Pos;		"Position on shgc_g3 map"
    string wiYacChrom;		"Chromosome (no chr) from Whitehead YAC map or 0 if none"
    float wiYacPos;		"Position on wi_yac map"
    string wiRhChrom;		"Chromosome (no chr) from Whitehead RH map or 0 if none"
    float wiRhPos;		"Position on wi_rh map"
    string fishChrom;		"Chromosome (no chr) from FISH map or 0 if none"
    string beginBand;		"Beginning of range of bands on FISH map"
    string endBand;		"End of range of bands on FISH map"
    string lab;			"Laboratory that placed the FISH clone"
    )
</PRE>

<H4><A name=STSmapMouse></A>STS Clone Positions (MGI)</H4>
<PRE>table stsMapMouse
"STS marker and its position on mouse assembly"
    (
    string chrom;       "Chromosome or 'unknown'"
    int chromStart;     "Start position in chromosome (-1 if unpositioned)"
    uint chromEnd;      "End position in chromosome"
    string name;        "Name of STS marker"
    uint score;         "Score of a marker = 1000/(# of placements)"
    uint identNo;       "UCSC ID number"
    uint probeId;       "Probe identification number of STS"
    uint markerId;      "Marker identification number of STS"
    )
</PRE>

<H4><A name=STSmapMouseNew></A>STS Clone Positions (NCBI)</H4>
<PRE>table stsMapMouseNew
"STS marker and its position on mouse assembly - mm3 and higher"
    (
    string chrom;       "Chromosome or 'unknown'"
    int chromStart;     "Start position in chromosome (-1 if unpositioned)"
    uint chromEnd;      "End position in chromosome"
    string name;        "Name of STS marker"
    uint score;         "Score of a marker = 1000/(# of placements)"
    uint identNo;       "Identification number of STS"
    string ctgAcc;      "Contig accession number"
    string otherAcc;    "Accession number of other contigs that the marker hits"
    string rhChrom;     "Chromosome (no chr) from RH map or 0 if none"
    float rhPos;        "Position on RH map"
    float rhLod;        "LOD score of RH map"
    string fhhChr;      "Chromosome (no chr) from FHHxACI genetic or 0 if none"
    float fhhPos;       "Position on FHHxACI map"
    string shrspChrom;  "Chromosome (no chr) from SHRSPxBN geneticmap or 0 if none"
    float shrspPos;     "Position on SHRSPxBN genetic map"
    )
</PRE>

<H4><A name=STSmapRat></A>STS Clone Positions (Rat)</H4>
<PRE>table stsMapRat
"STS marker and its position on rat assembly"
    (
    string chrom;       "Chromosome or 'unknown'"
    int chromStart;     "Start position in chromosome (-1 if unpositioned)"
    uint chromEnd;      "End position in chromosome"
    string name;        "Name of STS marker"
    uint score;         "Score of a marker = 1000/(# of placements)"
    uint identNo;       "Identification number of STS"
    string ctgAcc;      "Contig accession number"
    string otherAcc;    "Accession number of other contigs that the marker hits"
    string rhChrom;     "Chromosome (no chr) from RH map or 0 if none"
    float rhPos;        "Position on RH map"
    float rhLod;        "LOD score of RH map"
    string fhhChr;      "Chromosome (no chr) from FHHxACI genetic or 0 if none"
    float fhhPos;       "Position on FHHxACI map"
    string shrspChrom;  "Chromosome (no chr) from SHRSPxBN geneticmap or 0 if none"
    float shrspPos;     "Position on SHRSPxBN genetic map"
    )
</PRE>

<H4><A name=STSalias></A>STS Marker Aliases and Associated Identification Numbers</H4>
<PRE>table stsAlias
"STS marker aliases and associated identification numbers"
    (
    string alias;       "STS marker name"
    uint identNo;       "Identification number of STS marker"
    string trueName;    "Official UCSC name for marker"
    )
</PRE>

<H4><A name=STSinfo></A>STS Marker Constant Information</H4>
<PRE>table stsInfo
"Constant STS marker information"
    (
    uint identNo;                 "UCSC identification number"
    string name;                  "Official UCSC name"
    uint gbCount;                 "Number of related GenBank accessions"
    string[gbCount] genbank;      "Related GeneBank accessions"
    uint gdbCount;                "Number of related GDB identifiers"
    string[gdbCount] gdb;         "Related GDB identifiers"
    uint nameCount;               "Number of alias names"
    string[nameCount] otherNames; "Alias names"
    uint dbSTSid;                 "ID number in UniSTS or dbSTS"
    uint otherDbstsCount;         "Number of related dbSTS IDs"
    uint[otherDbstsCount] otherDbSTS;  "Related dbSTS IDs"

    string leftPrimer;            "5' primer sequence"
    string rightPrimer;           "3' primer sequence"
    string distance;              "Length of STS sequence"
    string organism;              "Organism for which STS discovered"
    uint sequence;                "Whether the full sequence is available (1) or not
				   (0) for STS"

    uint otherUCSCcount;          "Number of related active UCSC ids"
    uint[otherUCSCcount] otherUCSC;  "Related active UCSC ids"
    uint mergeUCSCcount;          "Number of merged inactive UCSC ids"
    uint[mergeUCSCcount] mergeUCSC;  "Related merged inactive UCSC ids"

    string genethonName;          "Name in Genethon map"
    string genethonChr;           "Chromosome in Genethon map"
    float genethonPos;            "Position in Genethon map"
    float genethonLOD;            "LOD score in Genethon map"
    string marshfieldName;        "Name in Marshfield map"
    string marshfieldChr;         "Chromosome in Marshfield map"
    float marshfieldPos;          "Position in Marshfield map"
    float marshfieldLOD;          "LOD score in Marshfield map"
    string wiyacName;             "Name in WI YAC map"
    string wiyacChr;              "Chromosome in WI YAC map"
    float wiyacPos;               "Position in WI YAC map"
    float wiyacLOD;               "LOD score in WI YAC map"
    string wirhName;              "Name in WI RH map"
    string wirhChr;               "Chromosome in WI RH map"
    float wirhPos;                "Position in WI RH map"
    float wirhLOD;                "LOD score in WI RH map"
    string gm99gb4Name;           "Name in GeneMap99 GB4 map"
    string gm99gb4Chr;            "Chromosome in GeneMap99 GB4 map"
    float gm99gb4Pos;             "Position in GeneMap99 GB4 map"
    float gm99gb4LOD;             "LOD score in GeneMap99 GB4 map"
    string gm99g3Name;            "Name in GeneMap99 G3 map"
    string gm99g3Chr;             "Chromosome in GeneMap99 G3 map"
    float gm99g3Pos;              "Position in GeneMap99 G3 map"
    float gm99g3LOD;              "LOD score in GenMap99 G3 map"
    string tngName;               "Name in Stanford TNG map"
    string tngChr;                "Chromosome in Stanford TNG map"
    float tngPos;                 "Position in Stanford TNG map"
    float tngLOD;                 "LOD score in Stanford TNG map"
    )
</PRE>

<H4><A name=STSinfo2></A>STS Marker Constant Information (Revised version)</H4>
<PRE>table stsInfo2
"Constant STS marker information - revision"
    (
    uint identNo;                 "UCSC identification number"
    string name;                  "Official UCSC name"
    uint gbCount;                 "Number of related GenBank accessions"
    string[gbCount] genbank;      "Related GeneBank accessions"
    uint gdbCount;                "Number of related GDB identifiers"
    string[gdbCount] gdb;         "Related GDB identifies"
    uint nameCount;               "Number of alias names"
    string[nameCount] otherNames; "Alias names"
    uint dbSTSid;                 "ID number in UniSTS or dbSTS"
    uint otherDbstsCount;         "Number of related dbSTS IDs"
    uint[otherDbstsCount] otherDbSTS;  "Related dbSTS IDs"

    string leftPrimer;            "5' primer sequence"
    string rightPrimer;           "3' primer sequence"
    string distance;              "Length of STS sequence"
    string organism;              "Organism for which STS discovered"
    uint sequence;                "Whether the full sequence is available (1) or not 
				   (0) for STS"

    uint otherUCSCcount;          "Number of related active UCSC ids"
    uint[otherUCSCcount] otherUCSC;  "Related active UCSC ids"
    uint mergeUCSCcount;          "Number of merged inactive UCSC ids"
    uint[mergeUCSCcount] mergeUCSC;  "Related merged inactive UCSC ids"

    string genethonName;          "Name in Genethon map"
    string genethonChr;           "Chromosome in Genethon map"
    float genethonPos;            "Position in Genethon map"
    float genethonLOD;            "LOD score in Genethon map"
    string marshfieldName;        "Name in Marshfield map"
    string marshfieldChr;         "Chromosome in Marshfield map"
    float marshfieldPos;          "Position in Marshfield map"
    float marshfieldLOD;          "LOD score in Marshfield map"
    string wiyacName;             "Name in WI YAC map"
    string wiyacChr;              "Chromosome in WI YAC map"
    float wiyacPos;               "Position in WI YAC map"
    float wiyacLOD;               "LOD score in WI YAC map"
    string wirhName;              "Name in WI RH map"
    string wirhChr;               "Chromosome in WI RH map"
    float wirhPos;                "Position in WI RH map"
    float wirhLOD;                "LOD score in WI RH map"
    string gm99gb4Name;           "Name in GeneMap99 GB4 map"
    string gm99gb4Chr;            "Chromosome in GeneMap99 GB4 map"
    float gm99gb4Pos;             "Position in GeneMap99 GB4 map"
    float gm99g3LOD;              "LOD score in GenMap99 G3 map"
    string tngName;               "Name in Stanford TNG map"
    string tngChr;                "Chromosome in Stanford TNG map"
    float tngPos;                 "Position in Stanford TNG map"
    float tngLOD;                 "LOD score in Stanford TNG map"
    string decodeName;            "Name in deCODE map"
    string decodeChr;             "Chromosome in deCODE TNG map"
    float decodePos;              "Position in deCODE TNG map"
    float decodeLOD;              "LOD score in deCODE TNG map"
    )
</PRE>

<H4><A name=STSinfoMouse></A>STS Marker Constant Information (Mouse - mm1, mm2)</H4>
<PRE>table stsInfoMouse
"Constant STS marker information for the mouse genome"
    (
    uint identNo;                "UCSC identification number"
    string name;                 "Official UCSC name"
    uint MGIPrimerID;            "STS primer's MGI ID, or 0 if N/A"
    string primerName;           "STS primer's name"
    string primerSymbol;         "STS primer's symbol"
    string primer1;              "Primer1 sequence"
    string primer2;              "Primer2 sequence"
    string distance;             "Length of STS sequence"
    uint sequence;               "Whether full sequence is available (1) 
                                     or not (0) for STS"
    uint MGIMarkerID;            "STS marker's MGI ID, or 0 if N/A"
    string stsMarkerSymbol;      "Symbol of STS marker"
    string Chr;                  "Chromosome in genetic map"
    float geneticPos;            "Position in genetic map 
                                     (-2 if N/A, -1 if syntenic)"
    string stsMarkerName;        "Name of STS Marker"
    uint LocusLinkID;            "Locuslink ID, or 0 if N/A"
    )
</PRE>

<H4><A name=STSinfoMouseNew></A>STS Marker Constant Information (Mouse - mm3 and higher)</H4>
<PRE>table stsInfoMouseNew
"Constant STS marker information for the mouse genome - mm3 and higher"
    (
    uint identNo;                "UCSC identification number"
    string name;                 "Official UCSC name"
    uint RGDId;		   	 "Marker's RGD ID"
    string RGDName;		 "Marker's RGD name"	
    uint UiStsId;		 "Marker's UiStsId"	   
    uint nameCount;		 "Number of alias"
    string alias;		 "Alias, or N/A"

    
    string primer1;          	 "Primer1 sequence"
    string primer2;          	 "Primer2 sequence"
    string distance;             "Length of STS sequence"
    uint sequence;               "Whether the full sequence is available (1) or not 
				  (0) for STS"
    string organis; 		 "Organism for which STS discovered"
   
    string fhhName;	         "WI_Mouse_Genetic map"
    string fhhChr;               "Chromosome in Genetic map"
    float fhhGeneticPos;         "Position in Genetic map"
    
    string shrspName;	         "MGD map"
    string shrspChr;             "Chromosome in Genetic map"
    float shrspGeneticPos;       "Position in Genetic map"

    string rhName;	    	 "MRC_RH map"
    string rhChr;                "Chromosome in Genetic map"
    float rhGeneticPos;          "Position in Genetic map"
    float RHLOD;		 "LOD score of RH map"


    string GeneName;             "Associated gene name"
    string GeneID;               "Associated gene ID"
    string clone;		 "Clone sequence"
    )
</PRE>

<H4><A name=STSinfoRat></A>STS Marker Constant Information (Rat)</H4>
<PRE>table stsInfoRat
"Constant STS marker information for the rat genome"
    (
    uint identNo;                "UCSC identification number"
    string name;                 "Official UCSC name"
    uint RGDId;		   	 "Marker's RGD ID"
    string RGDName;		 "Marker's RGD name"	
    uint UiStsId;		 "Marker's UiStsId"	   
    uint nameCount;		 "Number of alias"
    string alias;		 "Alias, or N/A"

    
    string primer1;          	 "Primer1 sequence"
    string primer2;          	 "Primer2 sequence"
    string distance;             "Length of STS sequence"
    uint sequence;               "Whether the full sequence is available (1) or not 
				  (0) for STS"
    string organis; 		 "Organism for which STS discovered"
   
    string fhhName;	         "WI_Mouse_Genetic map"
    string fhhChr;               "Chromosome in Genetic map"
    float fhhGeneticPos;         "Position in Genetic map"
    
    string shrspName;	         "MGD map"
    string shrspChr;             "Chromosome in Genetic map"
    float shrspGeneticPos;       "Position in Genetic map"

    string rhName;	    	 "MRC_RH map"
    string rhChr;                "Chromosome in Genetic map"
    float rhGeneticPos;          "Position in Genetic map"
    float RHLOD;		 "LOD score of RH map"


    string GeneName;             "Associated gene name"
    string GeneID;               "Associated gene ID"
    string clone;		 "Clone sequence"
    )
</PRE>

<H4><A name=SfAssign></A>Superfamily Assignments</H4>
<PRE>table sfAssign
"Superfamily assignment table"
  (
  string genomeID;      "Genome ID"
  string seqID;         "Sequence ID"
  string modelID;       "Model ID"
  string region;        "Region"
  string eValue;        "E value"
  string sfID;          "Superfamily entry ID"
  string sfDesc;        "Superfamily entry description"
  )
</PRE>

<H4><A name=SfDes></A>Superfamily Description</H4>
<PRE>table sfDes 
"Superfamily description table"
  (
  int id;                       "ID"
  char[2] level;                "Level"
  string classification;        "Classification"
  string name;                  "Name"
  string description;           "Description"
  )
</PRE>

<H4><A name=SfDescription></A>Superfamily Summary Entry Description</H4>
<PRE>table sfDescription
"Stores domain descriptions for the Superfamily track"
   (
    string name; 	     "Superfamily ID (same as Ensembl transcript name)"
    string proteinID; 	     "ID of corresponding Ensembl translation"
    string description;	     "Domain description"
   )                                                                           
</PRE>

<H4><A name=SpDisease></A>Swiss-Prot IDs Linked to Disease Descriptions</H4>
<PRE>table spDisease
"A cross-reference table between Swiss-Prot IDs and disease description."
   (
   string accession;       "SWISS-PROT accession number"
   string displayID;       "SWISS-PROT display ID"
   lstring diseaseDesc;    "Disease description"
   )
</PRE>

<H4><A name=SwInterPro></A>Swiss-Prot IDs Linked to InterPro IDs</H4>
<PRE>table swInterPro
"A cross-reference table between Swiss-Prot accession IDs and InterPro IDs."
        (
        string accession;       "SWISS-PROT accession number"
        string  interProId;      "InterPro ID"
        )
</PRE>

<H4><A name=SpOrganism></A>Swiss-Prot IDs Linked to Organisms</H4>
<PRE>table spOrganism 
"A cross-reference table between Swiss-Prot display IDs and taxonomy numbers."
   (
   string displayID;       "SWISS-PROT display ID"
   string organism;        "Taxonomy number"
   )
</PRE>

<H4><A name=SpXref2></A>Swiss-Prot IDs Linked to Other Databases</H4>
<PRE>table spXref2
"A xref table between SWISS-PROT ids and other databases."
   (
   string accession;       "SWISS-PROT accession number"
   string displayID;       "SWISS-PROT display ID"
   string division;        "SWISS-PROT division"
   int    bioentryID;      "Biosql bioentry ID"
   int    biodatabaseID;   "Biosql biodatabase ID"
   string description;     "Description"
   string hugoSymbol;      "HUGO Gene Nomenclature Committee gene symbol"
   string hugoDesc;        "HUGO Gene Nomenclature Committee gene description"
   )
</PRE>

<H4><A name=SpSecondaryID></A>Swiss-Prot IDs Linked to Secondary IDs</H4>
<PRE>table spSecondaryID 
"A cross-reference table between Swiss-Prot accession IDs and secondary accessions."
   (
   string displayID;       "SWISS-PROT display ID"
   string accession;       "SWISS-PROT accession number"
   string accession2;      "SWISS-PROT secondary accession number"
   )
</PRE>

<H4><A name=SpMrna></A>Swiss-Prot mRNA</H4>
<PRE>table spMrna
"Associates SwissProt IDs with mRNA IDs"
    (
    string spID; 	"Swiss-Prot ID"
    string mrnaID;	"mRNA ID"
    )
</PRE>

<H4><A name=SpPsiBlast></A>Swiss-Prot PSI-BLAST E-value</H4>
<PRE>table spPsiBlast
"PSI-BLAST all-against-all of Swiss-Prot/TrEMBL proteins. Swiss-Prot accessions 
are mapped to Known Genes IDs before being loaded into the table."
    (
    string kgId1;       "Known Gene ID of one protein"
    string kgId2;       "Known Gene ID of other protein"
    float eValue;       "Best bi-directional E-value"
    )
</PRE>

<H4><A name=BlastRef></A>tBLASTn Protein Track Mappings</H4>
<PRE>table blastRef
"Mapping table for tblastn protein tracks"
    (
    string acc;         "Accession"
    string geneId;      "Gene name"
    string refPos;      "Reference position"
    string extra1;      "Other link 1 (e.g. SwissProt name)
    string extra2;      "Other link 2"
    )
</PRE>

<H4><A name=TilingPath></A>Tiling Path of Clones through a Chromosome</H4>
<PRE>table tilingPath
"A tiling path of clones through a chromosome"
    (
    string chrom;	"Chromosome name: chr1, chr2, etc."
    string accession;   "Clone accession or ? or GAP"
    string clone;       "Clone name in BAC library"
    string contig;      "Contig (or gap size)"
    int chromIx;        "Number of clone in tiling path starting chrom start"
    )
</PRE>

<H4><A name=TransRegCodeMotif></A>Transcription Factor Binding Motif (HarbisonGordon et al.)</H4>
<PRE>table transRegCodeMotif
"A transcription factor binding motif according to Harbison Gordon et al"    
    (
    string name;                        "Motif name."
    int columnCount;                    "Count of columns in motif."
    float[columnCount] aProb;           "Probability of A's in each column."
    float[columnCount] cProb;           "Probability of C's in each column."
    float[columnCount] gProb;           "Probability of G's in each column."
    float[columnCount] tProb;           "Probability of T's in each column."
    )
</PRE>

<H4><A name=tfbsCons></A>Transcription Factor Binding Sites</H4>
<PRE>table tfbsCons
"TFBS data"
    (
    string chrom;      	"Chromosome"
    uint   chromStart; 	"Start position in chromosome"
    uint   chromEnd;   	"End position in chromosome"
    string name;       	"Name of item"
    uint   score;      	"Score from 0-1000"
    char[1] strand;    	"+ or -"
    char[6] species;  	"Common name, scientific name"
    char[64] factor;  	"Factor"
    char[10] id;  	"ID"
    )
</PRE>

<H4><A name=TransRegCodeProbe></A>Transcription Factor Binding Sites and CHIP/CHIP Probe Info</H4>
<PRE>table transRegCodeProbe
"CHIP/CHIP Probe and Transcription Factor Binding Info"    
    (
    string chrom;       	"Chromosome binding site is on"
    uint chromStart;    	"Start position in chromosome"
    uint chromEnd;      	"End position in chromosome" 
    string name;        	"Name of probe" 
    uint tfCount;       	"Count of bound transcription factors_conditions"
    string[tfCount] tfList;    	"List of bound transcription factors_conditions
"   
    float[tfCount] bindVals;   	"E-values for factor binding (lower is better)"
    )
</PRE>

<H4><A name=TransRegCode></A>Transcription Factor Binding Sites - CHIP/CHIP Experiments</H4>
<PRE>table transRegCode
"Transcription factor binding sites from CHIP/CHIP experiments and conservation"
    (
    string chrom;       	"Chromosome binding site is on"
    uint chromStart;    	"Start position in chromosome"
    uint chromEnd;      	"End position in chromosome"
    string name;        	"Name of transcription factor"
    uint score;         	"Score from 0 to 1000"
    string chipEvidence; 	"Evidence strength from CHIP/CHIP assay"
    uint consSpecies;   	"Number of species conserved in"
    )
</PRE>

<H4><A name=tfbsConsMap></A>Transcription Factor Binding Sites Conservation Map Data</H4>
<PRE>table tfbsConsMap
"tfbsConsMap Data"
    (
    string id;   "TRANSFAC id"
    string ac;   "gene-regulation.com AC"
    )
</PRE>

<H4><A name=TransRegCodeCondition></A>Transcription Factor Growth Condition</H4>
<PRE>table transRegCodeCondition
"Growth condition associated with a transcription factor"
    (
    string name;  "Name of transcription factor"
    string growthCondition; "Growth condition seen in"
    )
</PRE>

<H4><A name=VegaInfo></A>Vega Info</H4>
<PRE>table vegaInfo
"Vega Genes track additional information"
    (
    string transcriptId;        "Vega transcript ID"
    string otterId;             "Otter (Ensembl db) transcript ID"
    string geneId;              "Vega gene ID"
    string method;              "GTF method field"
    string geneDesc;            "Vega gene description"
    )
</PRE>

<H4><A name=WabaChromCreate></A>Waba Alignments (Abbreviated version)</H4>
<PRE>table chrN_wabaCbr
"Abbreviated Waba alignment table for quick display"
    (
    string query;	"Name of foreign sequence"
    uint chromStart;	"Start in genomic sequence"
    uint chromEnd; 	"End in genomic sequence"
    char strand; 	"Relative orientation"
    uint milliScore; 	"Identity in parts per thousand"
    longblob squeezed;	"HMM symbols with target inserts squeezed out"
    }
</PRE>

<H4><A name=WabaFullCreate></A>Waba Alignments (Full description)</H4>
<PRE>table wabaCbr
"Information on a Waba alignment"
  (
  string query; 	"Name of foreign sequence"
  uint qStart;		"Start of alignment in query"
  uint qEnd; 		"End of alignment in query"
  char qStrand;		"Which strand"
  string chrom; 	"Which chromosome"
  uint chromStart;	"Start in chromosome sequence"
  uint chromEnd; 	"End in chromosome sequence"
  uint milliScore	"Identity in parts per thousand"
  uint symCount;	"Number of symbols in following blobs"
  longblob qSym; 	"Query sequence and insert chars"
  longblob tSym;	"Target sequence and insert chars"
  longblob hSym;	"HMM symbols"
  )
</PRE>


	<BR></TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>

	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B>&nbsp; mRNA and EST Data</B></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10>&nbsp;</TD><TD>

This is a family of tables. The main table, <TT>gbCdnaInfo</TT>, consists of minor mRNA and EST identification
information, and then ids pointing to the corresponding tables which contain other information.  For example,
to determine the organism, the 'organism' field in the <TT>gbCdnaInfo</TT> table contains an id. Finding this id
in the corresponding <TT>organism</TT> table will give the name of the organism.


<H4><A name=MrnaInfo></A>Main mRNA/EST Information Table</H4>
<PRE>table gbCdnaInfo
"Basic information about mRNA and ESTs with links to other tables"
   (
   uint   id;              "Id, same as seq ID"
   string acc;             "GenBank accession"
   int    version;         "GenBank version"
   date   moddate;         "Last modification date"
   enum   type;            "Full length ('mRNA') or EST ('EST')"
   enum   direction;       "Read direction ('5','3','0')"
   uint   source;          "Ref in source table"
   uint   organism;        "Ref in organism table"
   uint   library;         "Ref in library table"
   uint   mrnaClone;       "Ref in clone table"
   uint   sex;             "Ref in sex table"
   uint   tissue;          "Ref in tissue table"
   uint   development;     "Ref in development table"
   uint   cell;            "Ref in cell table"
   uint   cds;             "Ref in CDS table"
   uint   keyword;         "Ref in key table"
   uint   description;     "Ref in description table"
   uint   geneName;        "Ref in geneName table"
   uint   productName;     "Ref in productName table"
   uint   author;          "Ref in author table"
   )
</PRE>

<H4><A NAME=MrnaChar></A>mRNA/EST Characteristic Tables</H4>
This table is used for the author, bacClone, cds, cell, center, chromosome, 
cytoMap, description, development, geneName, keyword, library, mrnaClone, 
organism, productName, sex, source, and tissue tables.  Each has a 
reference in the above gbCdnaInfo table.
<PRE>table mrnaCharacteristic
"Characteristics of mRNA and EST referenced in the gbCdnaInfo table"
   (
   int    id;              "Unique id of characteristic in each table"
   string name;            "Characteristic"
   int 	  crc;             "Cyclic redundancy check (performance optimization for
			    loading tables)"
   )
</PRE>

	<BR></TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>

	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B>&nbsp; Internal Use Tables</B></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10>&nbsp;</TD><TD>

<H4><A name=TrackDB></A>Annotation Track Description</H4>
<PRE>table trackDb
"This describes an annotation track."
  (
  string tableName;   "Symbolic ID of Track"
  string shortLabel;  "Short label displayed on left"
  string type;        "Track type: bed, psl, genePred, etc."
  string longLabel;   "Long label displayed in middle"
  ubyte visibility;   "0=hide, 1=dense, 2=full"
  float priority;     "0-100 - where to position.  0 is top"
  ubyte colorR;	      "Color red component 0-255"
  ubyte colorG;	      "Color green component 0-255"
  ubyte colorB;	      "Color blue component 0-255"
  ubyte altColorR;    "Light color red component 0-255"
  ubyte altColorG;    "Light color green component 0-255"
  ubyte altColorB;    "Light color blue component 0-255"
  ubyte useScore;     "1 if use score, 0 if not"
  ubyte private;      "1 if only want to show it on test site"
  int restrictCount;  "Number of chromosomes this is on (0=all!)"
  string[restrictCount] restrictList;  "List of chromosomes this is on"
  lstring url;	      "URL to link to when they click on an item"
  lstring html;       "Some html to display when they click on an item"
  )
</PRE>

<H4><A name=TableDesc></A>Database Table Descriptive Info</H4>
<PRE>table tableDescriptions
"Descriptive information about database tables from autoSql / gbdDescriptions"
  (
  string  tableName;	"Name of table (with chr*_ replaced with chrN_)"
  lstring autoSqlDef;	"Contents of autoSql (.as) table definition file"
  string  gbdAnchor;	"Anchor for table description in gbdDescriptions.html"
  )
</PRE>

<H4><A name=HistoryInfo></A>History Table - Tracks Modification History of mRNA Table and Linked Tables</H4>
<PRE>table history
  (
  int       ix        "Update number"
  uint      startId   "Start this session's ids"
  uint      endId     "First id for next session"
  string    who       "User who updated"
  string    what      "What they did"
  timestamp when      "When they did it"
  )
</PRE>

<H4><A name=GbLoaded></A>Reference Information for GenBank Updates</H4>
<PRE>table gbLoaded
"GenBank incremental update information"
  (
  int srcDb;       	"DB name (GenBank)"
  int type;     	"Type of data"
  string loadRelease;   "Release number"
  string loadUpdate;   	"Update number"
  string accPrefix;   	"Accession number prefix"
  timestamp time;    	"Time stamp of update"
  )
</PRE>

<H4><A name=AxtInfo></A>Reference to an Axt File</H4>
<PRE>table axtInfo
"Axt alignment names and sizes"
  (
  string species;   "Long name of species"
  string alignment; "Name of alignment"
  string chrom;     "Chromosome name"
  string fileName;  "Axt file name"
  int sort;	    "Sort order"
  )
</PRE>

<H4><A name=extFileInfo></A>Reference to an External File</H4>
<PRE>table extFile 
"This keeps track of external files and directories"
  (
  uint   id           "Unique ID for each file"
  string name         "Symbolic name of file"
  string path         "Full path. Ends in '/' if a dir"
  uint   size         "Size of file (checked)"
  )
</PRE>

<H4><A name=GbExtFile></A>Reference to an External GenBank File</H4>
<PRE>table gbExtFile 
"This keeps track of external files and directories"
  (
  uint   id           "Unique ID for each file"
  string path         "Full path. Ends in '/' if a dir"
  uint   size         "Size of file (checked)"
  )
</PRE>

<H4><A name=seqInfo></A>Reference to an External Sequence in an FA File</H4>
<PRE>table seq
"This keeps track of a sequence"
  (
  uint   id           "Unique ID for each external sequence"
  string acc          "GenBank accession number"
  uint   size         "Size of sequence in bases"
  date   gb_date      "GenBank last modified date"
  uint   extFile      "File it is in"
  uint   file_offset  "Offset in file"
  uint   file_size    "Size in file."
  )
</PRE>

<H4><A name=TrackGroups></A>Track Groups </H4>
<PRE>table grp
"Track group types in browser display window"
  (
  string name; 		"Track group name"
  string label; 	"Group label on browser display"
  float priority;	"Display order of group"
  )
</PRE>

	<BR></TD><TD WIDTH=7></TD></TR></TABLE>
	</TD></TR></TABLE>
			
</TD></TR></TABLE>
</BODY></HTML>
