<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Bioinformatics: The Database</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg"></TH></TR>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
	<table bgcolor="000000" cellpadding="1" cellspacing="1" width="100%" height="27"><tr bgcolor="2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
			<TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp; 
			&nbsp;<a href="/index.html" class="topbar">
				Home</a> &nbsp; - &nbsp;
			<a href="/cgi-bin/hgGateway" class="topbar">
				Genome Browser</a> &nbsp; - &nbsp; 
			<A HREF="/cgi-bin/hgBlat?command=start" class="topbar">
				Blat Search</a> &nbsp; - &nbsp; 
			<A HREF="/FAQ.html" class="topbar">
				FAQ</a> &nbsp; - &nbsp; 
			<A HREF="/goldenPath/help/hgTracksHelp.html" class="topbar">
				User Guide</a> &nbsp;</font></TD>
	</TR></TABLE>
	</TD></TR></TABLE>
</TD></TR>

<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>	
	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B> &nbsp;The Genome Browser Database </B></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
		This page describes the format of the genome annotation databases that underlie the 
		<A href="hgTracks.html">Human Genome Browser</A>. 
		This data was contributed by many researchers, as described in 
		<A href="credits.html">Credits</A>. 
		Please acknowledge the contributor(s) of the data you use. All
		of the tables for an assembly are 
		freely usable for any purpose except as indicated in the README.txt file in the 
		downloadable <A HREF="../downloads.html">annotation database
		directory</A> for that release. 
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>
	
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><b>&nbsp; Table Descriptions</FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>
<P>The data displayed in the <A href="http://genome.ucsc.edu/cgi-bin/hgGateway">
Human Genome Browser</A> 
is stored in a <A href="http://www.mysql.com/">MySQL</A> database. The tables in 
the database can be grouped into three categories. The first are ones for which 
the data has been split into separate tables for each chromosome. The second 
group has tables that contain information for all chromosomes. The third group 
has tables that contain information specifically related to mRNA sequences.</P>
<P>All of the coordinates in these tables are of "half-open zero based." This 
means that the first 100 bases of a chromosome are represented as [0,100). The 
second 100 bases are represented as [100,200), and so forth. An advantage of
half open coordinate ranges is that the length can be obtained by simply
subtracting the start from the end.</P>
<P>The database currently includes the following tables for each chromosome 
(chrN_[table] where N is replaced by the chromosome name). In many cases there 
is an additional table for the unordered part (random) of the chromosome 
(chrN_random_[table]). Finally, in addition to the standard chromosome names 1, 
..., 22, X, and Y, there are two special chromosome names, UL and NA, 
representing fingerprint clone contigs that could not be placed confidently on a 
chromosome, and individual clones that could not be placed, respectively. The 
table names are hyperlinked to an <a href=" http://www.soe.ucsc.edu/~kent/exe/doc/"> AutoSql</a>    
format description of their fields.</P>
<UL>
  <LI><A 
  href="#PslInfo">chrN_blastzBestMouse</A> 
  - <A HREF="http://bio.cse.psu.edu/" TARGET=_BLANK>Blastz</A> alignments 
    of mouse vs. human, filtered to retain only the best match for a given 
    region of the human genome.  (Dec 2001 and later)
  <LI><A 
  href="#PslInfo">chrN_blastzMm2</A> 
  - <A HREF="http://bio.cse.psu.edu/" TARGET=_BLANK>Blastz</A> alignments 
    of mouse vs. human.  (Dec 2001 and later)
  <LI><A 
  href="#PslInfo">chrN_blastzTightMouse</A> 
  - <A HREF="http://bio.cse.psu.edu/" TARGET=_BLANK>Blastz</A> alignments 
    of mouse vs. human, filtered to retain only the best match for a given 
    region of the human genome, and further filtered to high-scoring matches.
    (June 2002)
  <LI><A 
  href="#PslInfo">chrN_blatFish</A> 
  - Translated BLAT alignments of Tetraodon nigrovirisi against the human or mouse draft genome</LI>
  <LI><A 
  href="#PslInfo">chrN_blatFugu</A> 
  - Translated BLAT alignments of Takifugu rubripes against the human or mouse draft genome</LI>
  <LI><A 
  href="#PslInfo">chrN_blatHuman</A> 
  - Translated BLAT alignments of human genome vs. mouse genome</LI>
  <LI><A 
  href="#PslInfo">chrN_blatMouse</A> 
  - Mouse whole genome shotgun alignments (Aug 2001 and earlier)
  <LI><A 
  href="#PslInfo">chrN_blatMus</A> 
  - Translated BLAT mouse alignments.  (Dec 2001 and later)
  <LI><A 
  href="#PslInfo">chrN_est</A> 
  - EST alignments 
  <LI><A 
  href="#GapInfo">chrN_gap</A> 
  - Gap locations and types 
  <LI><A 
  href="#FragInfo">chrN_gl</A> 
  - Golden path positions of all clone fragments 
  <LI><A 
  href="#GpInfo">chrN_gold</A> 
  - Instructions on creating golden path from fragments 
  <LI><A 
  href="#wiggleTrack">chrN_humMusL</A> 
  - Human/Mouse Evolutionary Conservation Scores 
  <LI><A 
  href="#PslInfo">chrN_intronEst</A> 
  - EST alignments in introns 
  <LI><A 
  href="#PslInfo">chrN_mrna</A> 
  - mRNA alignments 
  <LI><A 
  href="#RepeatInfo">chrN_rmsk</A> 
  - Repeating elements created using RepeatMasker </LI>
  <LI><A 
  href="#wiggleTrack">chrN_zoom1_humMusL</A> 
  - Human/Mouse Evolutionary Conservation Scores (very large scale)
  <LI><A 
  href="#wiggleTrack">chrN_zoom2500_humMusL</A> 
  - Human/Mouse Evolutionary Conservation Scores (medium scale)
  <LI><A 
  href="#wiggleTrack">chrN_zoom50_humMusL</A> 
  - Human/Mouse Evolutionary Conservation Scores (large scale)
</UL>

<P>The following tables contain data that is placed on the golden path, but is 
stored in a single table rather than one table per chromosome:</P>
<UL>
  <LI><A 
  href="#GenePredictions">acembly</A> 
  - Acembly Gene Predictions With Alt-splicing </LI>
  <LI><A 
  href="#PepPredictions">acemblyPep</A> 
  - Translations of Acembly gene predictions into corresponding amino acid
    sequences </LI>
  <LI><A 
  href="#wiggleTrack">affyTranscriptome</A> 
  - Transcriptome data for chr21 and chr22 from Affymetrix </LI>
  <LI><A 
  href="#LFS">affyRatio</A> 
  - Expression data from GNF using Affymetrix GeneChips</LI>
  <LI><A 
  href="#PslInfo">all_bacends</A> 
  - BLAT alignment information for BAC clone end sequences on 
    the draft assembly sequence </LI>
  <LI><A 
  href="#PslInfo">all_est</A> 
  - EST alignments (concatenation of chr*_est) 
  <LI><A 
  href="#PslInfo">all_mrna</A> 
  - mRNA alignments (concatenation of chr*_mrna) 
  <LI><A 
  href="#PslInfo">all_sts_primer</A> 
  - BLAT alignment information for STS markers using primer 
    sequences on the draft sequence assembly </LI>
  <LI><A 
  href="#PslInfo">all_sts_seq</A> 
  - BLAT alignment information for full STS markers sequences 
    on the draft sequence assembly </LI>
  <LI><A 
  href="#LFS">bacEndPairs</A> 
  - Locations of BAC clones using placements of its 5' and 3' 
    end sequences.  BAC end sequence information comes from the dbGSS 
    portion of GenBank </LI>
  <LI><A 
  href="#PslInfo">chimpBac</A> 
  - Blat alignments of Pan troglodytes BACs against the draft human genome
  <LI><A 
  href="#PslInfo">chimpBlat</A> 
  - Blat alignments of Pan troglodytes against the draft human genome
  <LI><A 
  href="#ChromInfo">chromInfo</A> 
  - Describes name and size of each chromosome as well as an external file 
  containing the chromosome 
  <LI><A 
  href="#CloneInfo">clonePos</A> 
  - Positions of all clones in golden path. Also HTG phase, 
  finished/draft/predraft stage, and sequence size
  <LI><A 
  href="#CpG Islands">cpgIsland</A> 
  - CpG islands contributed by Baylor University and Washington University 
  <LI><A 
  href="#ContigInfo">ctgPos</A> 
  - Name, size, and golden path position of all FPC contigs
  <LI><A 
  href="#ChromosomeBand">cytoBand</A> 
  - 800 chromosome bands mapped to golden path by Terry Furey using
  fish data from BAC Resource Consortium
  <LI><A 
  href="#GenePredictions">ensGene</A> 
  - Ensembl Gene Predictions </LI>
  <LI><A 
  href="#PepPredictions">ensPep</A> 
  - Translations of Ensembl gene predictions into corresponding amino acid sequences </LI>
  <LI><A 
  href="#OrientInfo">EstOrientInfo</A> 
  - Extra information on ESTs calculated by polyInfo program
  <LI><A 
  href="#Exofish">exoFish</A> 
  - Genescope Exofish Tetraodon evolutionarily conserved regions (Last used Aug. 2001)
  <LI><A 
  href="#ExoMouse">exoMouse</A> 
  - Exonerate mouse/human evolutionarily conserved regions (Last used Apr. 2001) </LI>
  <LI><A 
  href="#FishClones">fishClones</A> 
  - Clones placed on Cytogenetic Map using FISH </LI>
  <LI><A 
  href="#GCpercent">gcPercent</A> 
  - Average number of G and C per 1000 bases in non-overlapping 20Kb windows 
  <LI><A 
  href="#GeneBands">geneBands</A> 
  - Band locations of known genes
  <LI><A 
  href="#GenMapDb">genMapDb</A> 
  - BAC clones from GenMapDB placed on the assembly by U Penn (V. Cheung)</LI>
  <LI><A 
  href="#GenomicDups">genomicDups</A> 
  - Duplications of >1000 Bases Sequence </LI>
  <LI><A 
  href="#GenePredictions">geneid</A> 
  - Gene predictions from the geneid program </LI>
  <LI><A 
  href="#GenePredictions">genieAlt</A> 
  - Genie gene predictions - all constrained by one degree or another by EST or 
  mRNA 
  <LI><A 
  href="#PepPredictions">genieAltPep</A> 
  - Translations of Genie gene predictions into its corresponding amino acid 
  sequence 
  <LI><A 
  href="#GenePredictions">genieKnown</A> 
  - Known genes processed by Genie to allow sequencing error (Oct. 
  2000 and earlier known gene track)
  <LI><A 
  href="#PepPredictions">genieKnownPep</A> 
  - Translations of known genes into its corresponding amino acid sequence (Oct. 
  2000 and earlier known gene track)
  <LI><A 
  href="#GenePredictions">genScan</A> 
  - Genscan Gene Predictions </LI>
  <LI><A 
  href="#PepPredictions">genscanPep</A> 
  - Translations of Genscan gene predictions into corresponding amino acid sequences </LI>
  <LI><A 
  href="#LFS">genscanSubopt</A> 
  - Genscan byproducts: good individual exons out of frame with main predictions (Last used Apr. 2001) </LI>
  <LI><A 
  href="#JaxOrtholog">jaxOrtholog</A> 
  - Jackson Labs Mouse Orthologs </LI>
  <LI><A 
  href="#JaxQTL">jaxQTL</A> 
  - Jackson Labs Mouse Quantitative Trait Loci </LI>
  <LI><A 
  href="#OtherKnown">knownInfo</A> 
  - Other information about known genes (Oct. 2000 and earlier known gene 
  track) 
  <LI><A 
  href="#MoreKnown">knownMore</A> 
  - Links to other databases: GenBank, OMIN, HUGO, RefSeq, GBD
  (Oct. 2000 and earlier known (RefSeq) gene track) 
  <LI><A 
  href="#MGIid">MGI ID</A> 
  - Links MGI ID to Locus Link ID (Mouse)
  <LI><A 
  href="#ChromInfo">mouseChrom</A> 
  - Describes name and size of each mouse chromosome as well as an external file 
  containing the chromosome 
  <LI><A 
  href="#MouseSyn">mouseSyn</A> 
  - Synteny between mouse and human chromosomes (Last used Apr. 2001) </LI>
  <LI><A 
  href="#MouseSynWhd">mouseSynWhd</A> 
  - Whitehead synteny between mouse and human chromosomes </LI>
  <LI><A 
  href="#OrientInfo">mrnaOrientInfo</A> 
  - Extra information on mRNAs calcluated by polyInfo program
  <LI><A 
  href="#BED">nci60</A> 
  - Microarray Experiments for NCI 60 Cell Lines </LI>
  <LI><A 
  href="#BED">perlegen</A> 
  - Perlegen Common High-Resolution Haplotype Blocks </LI>
  <LI><A 
  href="#LFS">phMouse</A> 
  - Pattern Hunter Mouse/Human Evolutionarily Conserved Regions (Last used Apr. 2001) </LI>
  <LI><A 
  href="#RecombRate">recombRate</A> 
  - Describes the recombination rate in 1Mb intervals based on deCODE, Genethon and Marshfield maps</LI>
  <LI><A 
  href="#RefFlat">refFlat</A> 
  - RefSeq Genes, with additional name (as it appears in genome browser) field  </LI>
  <LI><A 
  href="#GenePredictions">refGene</A> 
  - RefSeq Genes </LI>
  <LI><A 
  href="#GenePredictions">refKnown</A> 
  - Known genes as derived from alignments of RefSeq mRNA (Dec 2000 and later 
  known (RefSeq) gene track) 
  <LI><A 
  href="#RefLink">refLink</A> 
  - Relates RefSeq mRNA accession to LocusLink, HUGO, etc. (Dec. 2000 and later) 
  <LI><A 
  href="#MrnaPredictions">refMrna</A> 
  - RefSeq mRNA (Dec. 2000 and later) 
  <LI><A 
  href="#PepPredictions">refPep</A> 
  - RefSeq translated proteins (Dec. 2000 and later) 
  <LI><A 
  href="#PslInfo">refSeqAli</A> 
  - RefSeq translated proteins (Dec. 2001 and later) 
  <LI><A 
  href="#RefSeqStatus">refSeqStatus</A> 
  - Associates RefSeq mRNA accessions with corresponding RefSeq status (Dec. 2001 and later) 
  <LI><A 
  href="#RikenAltid">rikenaltid</A> 
  - Riken altid table (Mouse Feb. 2002 and later)</LI>
  <LI><A 
  href="#RikenAnn">rikenann</A> 
  - Riken annotation table (Mouse Feb. 2002 and later)</LI>
  <LI><A 
  href="#RikenCluster">rikencluster</A> 
  - Riken cluster table (Mouse Feb. 2002 and later)</LI> 
  <LI><A 
  href="#RikenClusterSeq">rikenclusterseq</A> 
  - Riken cluster sequence regulation table (Mouse Feb. 2002 and later)</LI>
  <LI><A 
  href="#RikenSeq">rikenseq</A> 
  - Riken sequence table (Mouse Feb. 2002 and later)</LI>
  <LI><A 
  href="#RNAcluster">rnaCluster</A> 
  - Gene boundaries deduced from clustering spliced ESTs and mRNAs against the genome (Apr. 2002 and later) </LI>
  <LI><A 
  href="#RNAgenes">rnaGene</A> 
  - Functional RNA genes (Last used Apr. 2001) </LI>
  <LI><A 
  href="#BED">rosetta</A> 
  - Rosetta Experimental Confirmation of Chr22 Exons </LI>
  <LI><A 
  href="#Sage">sage</A> 
  - Stores SAGE data in terms of UniGene identifiers</LI>
  <LI><A 
  href="#SageExp">sageExp</A> 
  - Data related to SAGE experiments, tissue descriptions, etc.
  <LI><A 
  href="#GenePredictions">sanger20</A> 
  - Sanger Institute Chromosome 20 Genes </LI>
  <LI><A 
  href="#SangerExtra">sanger20extra</A> 
  - Additional information about a Sanger 20 or 22 gene </LI>
  <LI><A 
  href="#GenePredictions">sanger22</A> 
  - Sanger Institute Chromosome 22 Genes </LI>
  <LI><A 
  href="#SangerExtra">sanger22extra</A> 
  - Additional information about a Sanger 20 or 22 gene </LI>
  <LI><A 
  href="#PepPredictions">sanger22pep</A> 
  - Translations of Sanger22 gene predictions into corresponding amino acid
    sequences </LI>
  <LI><A 
  href="#GenePredictions">sanger22pseudo</A> 
  - Sanger Institute Chromosome 22 Pseudogenes </LI>
  <LI><A 
  href="#SimpleRepeats">simpleRepeat</A> 
  - Simple tandem repeats generated with the TRF program 
  <LI><A 
  href="#GenePredictions">sgpGene</A> 
  - SGP Gene Predictions (GRIB)
  <LI><A 
  href="#PepPredictions">sgpPep</A> 
  - Translations of SGP gene predictions into corresponding amino acid sequences 
  <LI><A 
  href="#SNPsites">snpNih</A> 
  - Locations of overlap SNPs as determined by the SNP Consortium 
  <LI><A 
  href="#SNPsites">snpTsc</A> 
  - Locations of random SNPs as determined by the SNP Consortium 
  <LI><A 
  href="#GenePredictions">softberryGene</A> 
  - Fgenesh++ Gene Predictions </LI>
  <LI><A 
  href="#SoftberryHom">softberryHom</A> 
  - String with Genbank gi and accession </LI>
  <LI><A 
  href="#PepPredictions">softberryPep</A> 
  - Translations of Fgenesh++ gene predictions into corresponding amino acid sequences </LI>
  <LI><A 
  href="#STSalias">stsAlias</A> 
  - STS marker aliases and associated identification numbers  </LI>
  <LI><A 
  href="#STSinfo">stsInfo</A> 
  - Constant STS marker information </LI>
  <LI><A 
  href="#STSinfoMouse">stsInfoMouse</A> 
  - Constant STS marker information </LI>
  <LI><A 
  href="#STSmap">stsMap</A> 
  - STS markers on genetic (blue) and radiation hybrid (black) maps </LI> 
  <LI><A 
  href="#STSmapMouse">stsMapMouse</A> 
  - STS markers and postions on mouse assembly
  <LI><A 
  href="#STSsites">stsMarker</A> 
  - STS marker mappings from various maps and FISH clone mappings from the BAC 
  Resource Consortium (obsolete as of Apr. 2001, replaced by stsMap)</LI>
  <LI><A 
  href="#GenePredictions">tigrGeneIndex</A> 
  - Fgenesh++ Gene Predictions </LI>
  <LI><A 
  href="#TilingPath">tilingPath</A> 
  - A tiling path of clones through a chromosome </LI>
  <LI><A 
  href="#GenePredictions">twinscan</A> 
  - TwinScan Gene Predictions
  <LI><A 
  href="#PslInfo">uniGene</A> 
  - UniGene Alignments and SAGE Info (Last used Apr. 2001)  </LI>
  <LI><A 
  href="#PslInfo">xenoEst</A> 
  - Nonhuman ESTs from Genbank </LI>
  <LI><A 
  href="#PslInfo">xenoMrna</A> 
  - Nonhuman mRNAs from Genbank </LI>
</UL>

<P>The following tables contain additional information about mRNA sequences: 
<UL>
  <LI><A 
  href="#MrnaInfo">mrna</A> 
  - Very relational mRNA and EST information table. Consists mostly of the mRNA 
  accession and pointers into the tables below. 
  <LI><A 
  href="#MrnaInfo">author</A> 
  - List of authors 
  <LI><A 
  href="#MrnaInfo">bacClone</A> 
  - Name of the Bacterial Artificial Chromosome clone that a sequence 
    was grown in, if applicable </LI>
  <LI><A 
  href="#MrnaInfo">cds</A> 
  - Coding sequence in Genbank format 
  <LI><A 
  href="#MrnaInfo">cell</A> 
  - Cell types 
  <LI><A 
  href="#MrnaInfo">center</A> 
  - Sequencing center 
  <LI><A 
  href="#MrnaInfo">chromosome</A> 
  - Chromosome ID and name
  <LI><A 
  href="#MrnaInfo">cytoMap</A> 
  - Position on a cytological map 
  <LI><A 
  href="#MrnaInfo">description</A> 
  - Short (one line) description 
  <LI><A 
  href="#MrnaInfo">development</A> 
  - Stage of development of organism 
  <LI><A 
  href="#MrnaInfo">geneName</A> 
  - Name of gene 
  <LI><A 
  href="#MrnaInfo">keyword</A> 
  - Associated keywords in Genbank 
  <LI><A 
  href="#MrnaInfo">library</A> 
  - Clone library (mostly for ESTs) 
  <LI><A 
  href="#MrnaInfo">mrnaClone</A> 
  - Clone name (mostly for ESTs) 
  <LI><A 
  href="#MrnaInfo">organism</A> 
  - Name of organism 
  <LI><A 
  href="#MrnaInfo">productName</A> 
  - Protein product name 
  <LI><A 
  href="#MrnaInfo">sex</A> 
  - Sex of organism 
  <LI><A 
  href="#MrnaInfo">source</A> 
  - Source field in Genbank, often fairly redundant with other fields 
  <LI><A 
  href="#MrnaInfo">tissue</A> 
  - Tissue isolated from </LI></UL>
<P>The last set of tables are primarily for internal use:</P>
<UL>
  <LI><A 
  href="#AxtInfo">axtInfo</A> 
  - Axt alignment names and sizes 
  <LI><A 
  href="#extFileInfo">extFile</A> 
  - References to external files, again only used by RNA. 
  <LI><A 
  href="#HistoryInfo">history</A> 
  - Modification history of database, only used by RNA programs </LI>
  <LI><A 
  href="#seqInfo">seq</A> 
  - Describes where mRNA sequence lives. Includes reference to an external file. 
  <LI><A 
  href="#TrackDB">trackDb</A> 
  - Describes an annotation track </LI>
</UL>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>

	<BR>

	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B> &nbsp;<A name=TableFormats></A>Table Formats </B></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10>&nbsp;</TD><TD>

This section describes the format of each table field by field in 
<a href=" http://www.soe.ucsc.edu/~kent/exe/doc/">  AutoSql</a> format.

<H4><A name=BED></A>Browser Extensible Data</H4>
The following definition is used for the 
nci60, perlegen, and rosetta
tables as follows:<BR>
- nci60 uses the first 15 fields of this table. <BR>
- perlegen uses the first 12 fields of this table. <BR>
- rosetta uses the first 15 fields of this table. <BR>
<PRE>table bed
"Browser extensible data"
    (
    string chrom;      "Human chromosome or FPC contig"
    uint   chromStart; "Start position in chromosome"
    uint   chromEnd;   "End position in chromosome"
    string name;       "Name of item"
    uint   score;      "Score from 0-1000"
    char[1] strand;    "+ or -"
    uint thickStart;   "Start of where display should be thick (start codon)"
    uint thickEnd;     "End of where display should be thick (stop codon)"
    uint reserved;     "Always zero for now"
    int blockCount;    "Number of blocks"
    int[blockCount] blockSizes; "Comma separated list of block sizes"
    int[blockCount] blockStarts; "Start positions relative to chromStart."
    int expCount;      "Number of experiments."
    int[expCount] expIds; "Experiment IDs."
    int[expCount] expScores; "Experiment scores."
    )
</PRE>

<H4><A name=ChromosomeBand></A>Chromosome Band Information</H4>
<PRE>table cytoBand
"Describes the positions of cytogenetic bands within a chromosome"
   (
   string chrom;        "Human chromosome number"
   uint   chromStart;   "Start position in genoSeq"
   uint   chromEnd;     "End position in genoSeq"
   string name;         "Name of cytogenetic band"
   string gieStain;     "Giesma stain results"
   )
</PRE>

<H4><A name=ChromInfo></A>Chromosome Information</H4>
<PRE>table chromInfo
"Chromosome names and sizes"
    (
    string chrom;       "Chromosome name"
    uint   size;        "Chromosome size"
    string fileName;    "Chromosome file (raw one byte per base)"
    )
</PRE>

<H4><A name=FragInfo></A>Clone Fragment Positions</H4>
<PRE>table chrN_gl
"Fragment positions in golden path"
    (
    ushort  bin;          "A field to speed indexing added to many 
                           tables August 2001"
    uint    matches;      "Number of bases that match that aren't repeats"
    uint    misMatches;   "Number of bases that don't match"
    string  frag;         "Fragment name"
    uint    start;        "Start position in golden path"
    uint    end;          "End position in golden path"
    char[1] strand;       "+ or - for strand"
    )
</PRE>

<H4><A name=CloneInfo></A>Clone Information</H4>
<PRE>table clonePos
"A clone's position and other info."
    (
    string  name;       "Name of clone including version"
    uint    seqSize;    "base count not including gaps"
    ubyte   phase;      "htg phase"
    string  chrom;      "Chromosome name"
    uint    chromStart; "Start in chromosome"
    uint    chromEnd;   "End in chromosome"
    char[1] stage;      "F/D/P for finished/draft/predraft"
    string  faFile;     "File with sequence."
    )
</PRE>

<H4><A name=ContigInfo></A>Contig Information</H4>
<PRE>table ctgPos
"Where a contig is inside of a chromosome."
    (
    string contig;      "Name of contig"
    uint   size;        "Size of contig"
    string chrom;       "Chromosome name"
    uint   chromStart;  "Start in chromosome"
    uint   chromEnd;    "End in chromosome"
    )
</PRE>

<H4><A name="CpG Islands"></A>CpG Islands</H4>
This table contains CpG islands of 50 bases or more which are at least 50% 
G/C.
<PRE>table cpgIsland
"Describes the CpG Islands"
   (
   string chrom;        "Human chromosome or FPC contig"
   uint   chromStart;   "Start position in chromosome"
   uint   chromEnd;     "End position in chromosome"
   string name;         "CpG Island"
   uint   minLength;    "Minimum Island Length"
   uint   cpgNum;       "Number of CpGs in island"
   uint   gcNum;        "Number of C and G in island"
   uint   perCpg;       "Percentage of island that is CpG"
   uint   perGc;        "Percentage of island that is C or G"
   )
</PRE>

<H4><A name=Exofish></A>Exofish Information</H4>
<PRE>table exoFish
"An evolutionarily conserved region (ecore) with Tetroadon"
   (
   string chrom;        "Human chromosome or FPC contig"
   uint   chromStart;   "Start position in chromosome"
   uint   chromEnd;     "End position in chromosome"
   string name;         "Just a dummy dot"
   uint   score;        "Score from 0 to 1000"
   )
</PRE>

<H4><A name=ExoMouse></A>Exonerate Mouse Homologies</H4>
Last used Apr. 2001.
<PRE>table exoMouse
"A rough alignment - not detailed"
   (
   string chrom;        "Human chromosome or FPC contig"
   uint   chromStart;   "Start position in chromosome"
   uint   chromEnd;     "End position in chromosome"
   string name;         "Name of other sequence"
   uint   score;        "Score from 0 to 1000"
   char[1] strand;      " or -"
   uint otherStart;     "Start in other sequence"
   uint otherEnd;       "End in other sequence"
   )
</PRE>

<H4><A name=FishClones></A>FISH Clones</H4>
<PRE>table fishClones
"Describes the positions of fishClones in the assembly"
   (
   string chrom;                   "Human chromosome number"
   uint   chromStart;              "Start position in chromosome"
   uint   chromEnd;                "End position in chromosome"
   string name;                    "Name of clone"
   uint score;                     "Always 1000"
   uint placeCount;                "Number of times FISH'd"
   string[placeCount] bandStarts;  "Start FISH band"
   string[placeCount] bandEnds;    "End FISH band"
   string[placeCount] labs;        "Lab where clone FISH'd"
   string placeType;               "How clone was placed on the sequence assembly"
   uint accCount;                  "Number of accessions associated with the clone"
   string[accCount] accNames;      "Accession associated with clone"
   uint stsCount;                  "Number of STS markers associated with this clone"
   string[stsCount] stsNames;      "Names of STS  markers"
   uint beCount;                   "Number of BAC end sequences associated with this clone"
   string[beCount] beNames;        "Accessions of BAC ends"
   )
</PRE>

<H4><A name=GapInfo></A>Gap Positions</H4>
<PRE>table chrN_gap
"Gaps in golden path"
    (
    ushort bin;         "A field to speed indexing added to many 
                         tables August 2001"
    uint   matches;     "Number of bases that match that aren't repeats"
    uint   misMatches;  "Number of bases that don't match"
    string chrom;       "which chromosome"
    uint   chromStart;  "start position in chromosome"
    uint   chromEnd;    "end position in chromosome"
    int    ix;          "ix of this fragment (useless)"
    char[1] n;          "always 'N'"
    uint   size;        "size of gap"
    string type;        "contig, clone, fragment, etc."
    string bridge;      "yes, no, mrna, bacEndPair, etc."
    )
</PRE>

<H4><A name=GCpercent></A>GC Content</H4>
<PRE>table gcPercent
"Displays GC content in 20Kb non-overlapping blocks for genome"
   (
   string chrom;        "Human chromosome number"
   uint   chromStart;   "Start position in chromosome"
   uint   chromEnd;     "End position in chromosome"
   string name;         "Constant string GC"
   uint   gcPpt;        "Average number of G and C per 1000 bases"
   )                                                                            
</PRE>

<H4><A name=GeneBands></A>Gene Bands</H4>
<PRE>table geneBands
"Band locations of known genes"
   (
   string name;         "Gene name (HUGO if possible)"
   string mrnaAcc;      "RefSeq mRNA accession"
   int    count;        "Number of times this accession maps to the genome"
   string[count] bands; "List of chromosome bands to which it maps"
   )                                                                            
</PRE>

<H4><A name=RNAcluster></A>Gene Boundaries as Defined by RNA and Spliced
EST Clusters</H4>
<PRE>table rnaCluster
"Gene boundaries deduced from clustering spliced ESTs and mRNAs against the genome"
    (
    smallint bin;           "Index"
    string chrom;           "Human chromosome or FPC contig"
    uint   chromStart;      "Start position in chromosome"
    uint   chromEnd;        "End position in chromosome"
    string name;            "Name of item"
    uint   score;           "Score from 0-1000"
    char[1] strand;         "+ or -"
    uint thickStart;        "Start of where display should be thick (start codon)"
    uint thickEnd;          "End of where display should be thick (stop codon)"
    uint reserved;          "Always zero for now"
    int blockCount;         "Number of blocks"
    longblob blockSizes;    "Size of each block"
    longblob chromStarts;   "Start position of each block in chromosome"
    ) 
</PRE>    

<H4><A name=GenePredictions></A>Gene Predictions and RefSeq Genes</H4>
The following definition is used for the acembly, ensGene, geneid, genieAlt, genScan, genscanSubopt, refGene, sanger20, sanger22, softberryGene, tigrGeneIndex, twinscan, and similar tables. In alternative splicing situations
each transcript has a row in this table.
<PRE>table genePred
"A gene prediction."
    (
    string  name;       "Name of gene"
    string  chrom;      "Chromosome name"
    char[1] strand;     "+ or - for strand"
    uint    txStart;    "Transcription start position"
    uint    txEnd;      "Transcription end position"
    uint    cdsStart;   "Coding region start"
    uint    cdsEnd;     "Coding region end"
    uint    exonCount;  "Number of exons"
    uint[exonCount] exonStarts; "Exon start positions"
    uint[exonCount] exonEnds;   "Exon end positions"
    )
</PRE>

<H4><A name=RefFlat></A>Gene Predictions and RefSeq Genes with Gene Names</H4>
A version of genePred that associates the gene name with the gene prediction 
information. In alternative splicing situations each transcript has a row in this table.
<PRE>table refFlat
"A gene prediction with additional geneName field."
    (
    string  geneName;   "Name of gene as it appears in genome browser."
    string  name;       "Name of gene"
    string  chrom;      "Chromosome name"
    char[1] strand;     "+ or - for strand"
    uint    txStart;    "Transcription start position"
    uint    txEnd;      "Transcription end position"
    uint    cdsStart;   "Coding region start"
    uint    cdsEnd;     "Coding region end"
    uint    exonCount;  "Number of exons"
    uint[exonCount] exonStarts; "Exon start positions"
    uint[exonCount] exonEnds;   "Exon end positions"
    )
</PRE>

<H4><A name=PepPredictions></A>Gene Predictions and RefSeq Genes Amino Acid Translations</H4>
The following definition is used for tables that link a predicted peptide to a predicted gene: acemblyPep, ensPep, genieAltPep, genieKnownPep, genscanPep, refPep, sanger22Pep, softberryPep.
<PRE>table pepPred
"A predicted peptide linked to a predicted gene."
    (
    string  name;       "Name of gene - same as in genePred"
    lstring seq;        "Peptide sequence"
    )
</PRE>

<H4><A name=GenMapDb></A>GenMapDB BAC Clones</H4>
<PRE>table genMapDb
"BAC clones from GenMapDB placed on the assembly by U Penn (V. Cheung)"
   (
   string  chrom;       "Chromosome number or 'unknown'"
   int     chromStart;  "Start position in chromosome (-1 if unpositioned)"
   int     chromEnd;    "End position in chromosome"
   string  name;        "Clone name"
   uint    score;       "Score - always 1000" 
   char[1] strand;      "+ or -"
   string  accT7;       "Accession number for T7 BAC end sequence"
   uint    startT7;     "T7 start position in chromosome"
   uint    endT7;       "T7 end position in chromosome"
   char[1] strandT7;    "+ or -"
   string  accSP6;      "Accession number for Sp6 BAC end sequence"
   uint    startSP6;    "Sp6 start position in chromosome"
   uint    endSP6;      "Sp6 end position in chromosome"
   char[1] strandSP6;   "+ or -"
   string  stsMarker;   "Name of STS marker found in clone"
   uint    stsStart;    "STS marker start position in chromosome" 
   uint    stsEnd;      "STS marker end position in chromosome"
   )                                                                            
</PRE>

<H4><A name=GenomicDups></A>Genomic Duplications</H4>
<PRE>table genomicDups
"Summary of large genomic Duplications (>1KB >90% similar)"
  (
  string   chrom;       "Human chromosome"
  uint    chromStart;   "Start position in chromosome"
  uint    chromEnd;     "End position in chromosome"
  string  name;         "Other FPC contig involved"
  uint    score;        "Score from 900-1000.  1000 is best"
  char[1] strand;       "Value should be + or -"
  string  otherChrom;   "Human chromosome"
  uint    otherStart;   "Start in other sequence"
  uint    otherEnd;     "End in other sequence"
  uint    alignB;       "bases Aligned"
  uint    matchB;       "aligned bases that match"
  uint    mismatchB;    "aligned bases that do not match"
  float   fracMatch;    "fraction of matching bases"
  float   jcK;          "K-value calculated with Jukes-Cantor"
  )
</PRE>

<H4><A name=GpInfo></A>Golden Path Construction</H4>
<PRE>table chrN_gold
"How to get through chromosome based on fragments"
    (
    ushort  bin;        "A field to speed indexing added to many 
                         tables August 2001"
    uint    matches;    "Number of bases that match that aren't repeats"
    uint    misMatches; "Number of bases that don't match"
    string  chrom;      "which chromosome"
    uint    chromStart; "Start position in chromosome"
    uint    chromEnd;   "End position in chromosome"
    int     ix;         "Ix of this fragment (useless)"
    char[1] type;       "(P)redraft, (D)raft, (F)inished or (O)ther"
    string  frag;       "Which fragment"
    uint    fragStart;  "Start position in frag"
    uint    fragEnd;    "End position in frag"
    char[1] strand;     "+ or - (orientation of fragment)"
    )
</PRE>

<H4><A name=JaxOrtholog></A>Jackson Labs Mouse Orthologs</H4>
<PRE>table jaxOrtholog
"Jackson Labs Mouse Orthologs"
    (
    string humanSymbol;	"Human HUGO symbol"
    string humanBand; "Human chromosomal location"
    string mgiId;     "Mouse database id"
    string mouseSymbol; "Mouse human symbol"
    string mouseChr; "Mouse chromosome"
    string mouseCm; "Mouse genetic map positionin centimorgans"
    string mouseBand; "Mouse chromosome band if any"
    )
</PRE>

<H4><A name=JaxQTL></A>Jackson Labs Quantitative Trait Loci</H4>
<PRE>table jaxQTL
"Quantitative Trait Loci from Jackson Labs / Mouse Genome Informatics"
    (
    string chrom;       "Human chromosome or FPC contig"
    uint   chromStart;  "Start position in chromosome"
    uint   chromEnd;    "End position in chromosome"
    string name;        "Name of item"
    uint   score;       "Score from 0-1000 (bed6 compat.)"
    char[1] strand;     "+ or - (bed6 compat.)"
    string marker;      "MIT SSLP Marker w/highest correlation"
    string mgiID;       "MGI ID"
    string description; "MGI description"
    float  cMscore;     "cM position of marker associated with peak LOD score"
    )
</PRE>

<H4><A name=MGIid></A>MGI ID</H4>
<PRE>table MIGIid
"Links MGI ID to Locus Link ID (Mouse)"
    (
    char[8]  LLid;       "Locus Link ID"
    char[12] MGIid;      "MGI ID"
    char[32] symbol;     "Symbol"
    )
</PRE>

<H4><A name=MouseSyn></A>Mouse Synteny</H4>
Last used Apr. 2001.
<PRE>table mouseSyn
"Synteny between mouse and human chromosomes."
    (
    string  chrom;       "Name of chromosome"
    uint    chromStart;  "Start in chromosome"
    uint    chromEnd;    "End in chromosome"
    string  name;        "Name of mouse chromosome"
    int     segment;     "Number of segment"
    )
</PRE>

<H4><A name=MouseSynWhd></A>Mouse Synteny (Whitehead)</H4>
<PRE>table mouseSynWhd
"Whitehead synteny between mouse and human chromosomes."
    (
    string  chrom;       "Human chromosome"
    uint    chromStart;  "Start position in human chromosome"
    uint    chromEnd;    "End position in human chromosome"
    string  name;        "Name of mouse chromosome"
    uint    score;       "Unused (bed 6 compatibility)"
    char[1] strand;      "+ or - (orientation of fragment)"
    uint    mouseStart;  "Start position in mouse chromosome"
    unit    mouseEnd;    "End position in mouse chromosome"
    string  segLabel;    "Whitehead segment label"
    )
</PRE>

<H4><A name=MoreKnown></A>Known (RefSeq) Gene IDs in Other Databases</H4>
Last used Oct. 2000 <BR><BR>
<PRE>table knownMore 
"Lots of auxiliary info about a known gene" 
    (  
    string name;	  "The name displayed in the browser:
                           OMIM, gbGeneName, or transId"   
    string transId;       "Transcript id. Genie generated ID." 
    string geneId;        "Gene (not transcript) Genie ID"   
    uint   gbGeneName;    "Connect to geneName table. Genbank gene name"   
    uint   gbProductName; "Connects to productName table. Genbank product name" 
    string gbProteinAcc;  "Genbank accession of protein"  
    string gbNgi;         "Genbank gi of nucleotide seq." 
    string gbPgi;         "Genbank gi of protein seq."   
    uint   omimId;        "OMIM ID or 0 if none"  
    string omimName;      "OMIM primary name"  
    uint   hugoId;        "HUGO Nomeclature Committee ID or 0 if none" 
    string hugoSymbol;    "HUGO short name"  
    string hugoName;      "HUGO descriptive name" 
    string hugoMap;       "HUGO Map position" 
    uint   pmId1;         "I have no idea - grabbed from a HUGO nomeids.txt" 
    uint   pmId2;         "Likewise, I have no idea" 
    string refSeqAcc;     "Accession of RefSeq mRNA" 
    string aliases;       "Aliases if any.  Comma and space separated list" 
    uint   locusLinkId;   "Locus link ID" 
    string gdbId;         "NCBI GDB database ID" 
    )
</PRE>

<H4><A name=OtherKnown></A>Known (RefSeq) Genes - Other Information</H4>
Last used Oct. 2000
<PRE>table knownInfo
"Auxiliary info about a known gene"
    (
    string name;         "Connects with genieKnown-&gt;name"
    string transId;      "Transcript id. Genie generated ID."
    string geneId;       "Gene (not transcript) id"
    uint   geneName;     "Connect to geneName table"
    uint   productName;  "Connects to productName table"
    string proteinId;    "Genbank accession of protein?"
    string ngi;          "Genbank gi of nucleotide seq."
    string pgi;          "Genbank gi of protein seq."
    )                                                                           
</PRE>

<H4><A name=MrnaPredictions></A>mRNA Sequences</H4>
<PRE>table mrnaPred
"A mRNA - linked to a gene."
    (
    string  name;       "Name of gene - same as in genePred"
    lstring seq;        "mRNA sequence"
    )
</PRE>

<H4><A name=PslInfo></A>mRNA/EST/Blat Alignments</H4>
The following definition is used for the chrN_blatMouse, chrN_blatFish, 
chrN_blatHuman, chrN_mrna, chrN_est, chrN_intronEst, chimpBac, chimpBlat, 
all_mrna, all_est uniGene, xenoEst and xenoMrna tables.<BR><BR>
<PRE>table psl  
"Summary info about a patSpace alignment" 
    ( 
    ushort  bin;          "A field to speed indexing added to many
    uint    matches;      "Number of bases that match that aren't repeats"
    uint    misMatches;   "Number of bases that don't match"  
    uint    repMatches;   "Number of bases that match but are part of repeats" 
    uint    nCount;       "Number of 'N' bases" 
    uint    qNumInsert;   "Number of inserts in query" 
    int     qBaseInsert;  "Number of bases inserted in query" 
    uint    tNumInsert;   "Number of inserts in target" 
    int     tBaseInsert;  "Number of bases inserted in target"
    char[2] strand;       "+ or - for query strand.   
                               For mouse second +/- for genomic strand"  
    string  qName;        "Query sequence name"
    uint    qSize;        "Query sequence size"
    uint    qStart;       "Alignment start position in query"
    uint    qEnd;         "Alignment end position in query" 
    string  tName;        "Target sequence name" 
    uint    tSize;        "Target sequence size" 
    uint    tStart;       "Alignment start position in target" 
    uint    tEnd;         "Alignment end position in target" 
    uint    blockCount;   "Number of blocks in alignment" 
    uint[blockCount] blockSizes;  "Size of each block" 
    uint[blockCount] qStarts;     "Start of each block in query." 
    uint[blockCount] tStarts;     "Start of each block in target."
    )  
</PRE>
There is a little gotcha in this table format. It has to 
do with how coordinates are handled on the negative strand.
In the qStart/qEnd fields the coordinates are where it
matches from the point of view of the forward strand 
(even when the match is on the reverse strand). However
on the qStarts[] list, the coordinates are reversed. 
Here's an example of a 30-mer that has 2 blocks that
align on the minus strand and 2 blocks on the plus
strand (this sort of stuff happens in real life in
response to assembly errors sometimes). <BR><BR>
<TABLE CELLPADDING=0 CELLSPACING=0 MARGINS=0 BORDER=0>
<TR><TD><PRE>0         1         2         3 tens position in query   </TD></TR>
<TR><TD><PRE>0123456789012345678901234567890 ones position in query   </TD></TR>
<TR><TD><PRE>            ++++          +++++ plus strand alignment on query   </TD></TR>
<TR><TD><PRE>    --------    ----------      minus strand alignment on query   </TD></TR>
<TR><TD></TD></TD>	
<TR><TD><PRE>Plus strand:   </TD></TR>
<TR><TD><PRE>     qStart 12 qEnd 31 blockSizes 4,5 qStarts 12,26   </TD></TR>
<TR><TD><PRE>Minus strand:   </TD></TR>
<TR><TD><PRE>     qStart 4 qEnd 26 blockSizes 10,8 qStarts 5,19    </TD></TR>
</TABLE><BR>
Essentially the minus strand blockSizes and qStarts are 
what you would get if you reverse complemented the query.
However the qStart and qEnd are non-reversed. To get
from one to the other:
<PRE>
     qStart = qSize - revQEnd
     qEnd = qSize - revQStart
</PRE>

<H4><A name="OrientInfo"></A>Orientation Info</H4>
The following definition is used by the mrnaOrientInfo and estOrientInfo tables. It contains extra information on mRNAs/ESTs calculated by the polyInfo program.
<PRE>table estOrientInfo
"Extra information on mRNAs/ESTs calculated by polyInfo program"
   (
   int    bin;               "A field to speed indexing"
   uint   chromStart;        "Start position in chromosome"
   uint   chromEnd;          "End position in chromosome"
   string name;              "Accession of EST/mRNA"
   short intronOrientation;  "Orientation of introns with respect to EST/mRNA"
   short sizePolyA;          "Number of trailing A's"
   short revSizePolyA;       "Number of trailing A's on reverse strand"
   short signalPos;          "Position of start of polyA signal relative to end 
	                         of EST/mRNA, or 0 if no signal"
   short revSignalPos;       "PolyA signal position on reverse strand if any"
   )
</PRE>

<H4><A name=RecombRate></A>Recombination Rate</H4>
<PRE>table recombRate
"Describes the recombination rate in 1Mb intervals based on deCODE, Marshfield and Genethon maps"
    (
    string  chrom;            "Human chromosome number"
    uint    chromStart;       "Start position in genoSeq"
    uint    chromEnd;         "End position in genoSeq"
    string  name;             "Constant string 'recombRate'"
    float   decodeAvg;        "Calculated deCODE Sex-averaged rate from map"
    float   decodeFemale;     "Calculated deCODE female recombination rate"
    float   decodeMale;       "Calculated deCODE male recombination rate"
    float   marshfieldAvg;    "Calculated Marshfield recombination rate"
    float   marshfieldFemale; "Calculated Marshfield female recombination"
    float   marshfieldMale;   "Calculated Marshfield male recombination rate"
    float   genethonAvg;      "Calculated Genethon recombination rate"
    float   genethonFemale;   "Calculated Genethon female recombination rate"
    float   genethonMale;     "Calculated Genethon male recombination rate"
    )
</PRE>

<H4><A name=RefLink></A>RefSeq Link</H4>
First used Dec. 2000
<PRE>table refLink
"Link between RefSeq mRNAs and HUGO, LocusLink etc."
    (
    string name;        "Name displayed in UI"
    string product;     "Name of protein product"
    string mrnaAcc;     "mRNA accession"
    string protAcc;     "Protein accession"
    uint geneId;        "Pointer to geneName table"
    uint prodId;        "Pointer to prodName table"
    uint locusLinkId;   "Locus Link ID"
    uint omimId;        "OMIM ID"
    )
</PRE>

<H4><A name=RefSeqStatus></A>RefSeq Status</H4>
First used Dec. 2001
<PRE>table refSeqStatus
"Links RefSeq mRNA accessions with status"
    (
    string mrnaAcc;     "RefSeq mRNA accession"
    string status;      "RefSeq status (Reviewed, Provisional, Predicted)"
    )
</PRE>

<H4><A name=RepeatInfo></A>Repeating Elements</H4>
These tables were created from RepeatMasker .out files.
<PRE>table chrN_rmsk
"RepeatMasker .out record"
    (
    ushort  bin;        "A field to speed indexing added to many 
                         tables August 2001"
    uint    matches;    "Number of bases that match that aren't repeats"
    uint    misMatches; "Number of bases that don't match"
    uint    swScore;    "Smith Waterman alignment score"
    uint    milliDiv;   "Base mismatches in parts per thousand"
    uint    milliDel;   "Bases deleted in parts per thousand"
    uint    milliIns;   "Bases inserted in parts per thousand"
    string  genoName;   "Genomic sequence name"
    uint    genoStart;  "Start in genomic sequence"
    uint    genoEnd;    "End in genomic sequence"
    int     genoLeft;   "Size left in genomic sequence"
    char[1] strand;     "Relative orientation + or -"
    string  repName;    "Name of repeat"
    string  repClass;   "Class of repeat"
    string  repFamily;  "Family of repeat"
    int     repStart;   "Start in repeat sequence"
    uint    repEnd;     "End in repeat sequence"
    int     repLeft;    "Size left in repeat sequence"
    char[1] id;         "'*' or ' '.  I don't know what this means"
    )
</PRE>

<H4><A name=RikenAltid></A>Riken Altid Table</H4>
<PRE>table rikenaltid
"Riken altid table"
    (
    char[64]    ID;             "ID (composite)"
    char[32]    SeqID;          "Sequence ID"
    char[32]    AltIDType;      "Alternate ID type"
    string      AltID;          "Alternate ID"
    )
</PRE>

<H4><A name=RikenAnn></A>Riken Annotation Table</H4>
<PRE>table rikenann
"Riken annotation table"
    (
    string  ID;             "ID"
    string  seqid;          "parent sequence id"
    uint    i;              "sequential number"
    string  Qualifier;      "Qualifier"
    string  Anntext;        "Anntext"
    string  Datasrc;        "Datasrc"
    string  Srckey;         "Srckey"
    string  Href;           "xlink_href"
    string  Evidence;       "Evidence"
    )
</PRE>

<H4><A name=RikenCluster></A>Riken Cluster Table</H4>
<PRE>table rikencluster
"Riken cluster table"
    (
    string  ID;                     "ID"
    string  Fantomid;               "Famtom"
    string  RepresentativeSeqid;    "Representative sequence ID"
    uint    Nsequence;              "Number of Sequencesse"
    )
</PRE>


<H4><A name=RikenClusterSeq></A>Riken Cluster Sequence Regulation Table</H4>
<PRE>table rikenclusterseq
"Riken cluster sequence regulation table"
    (
    string  ClusterID;              "Cluster ID"
    string  SeqID;                  "Sequence ID"
    )
</PRE>


<H4><A name=RikenSeq></A>Riken Sequence Table</H4>
<PRE>table rikenseq
"Riken sequence table"
    (
    string  ID;             "ID"
    uint    Naltid;         "Number of altid"
    string  Seqid;          "sequence ID (no more valid per Riken DTD)"
    string  Fantomid;       "Fantom ID (no more valid per Riken DTD)"
    string  Cloneid;        "Clone ID (no more valid per Riken DTD)"
    string  Rearrayid;      "Rearray ID (no more valid per Riken DTD)"
    string  Accession;      "Accession (no more valid per Riken DTD)"
    string  Annotator;      "Annotator"
    string  version;        "Version"
    string  ModifiedTime;   "Modified Time"
    uint    Nannotation;    "Number of Annotations"
    string  Comment;        "Comment"
    )
</PRE>

<H4><A name=RNAgenes></A>RNA Gene Positions</H4>
Last used Apr. 2001.
<PRE>table rnaGene
"Describes functional RNA genes."
    (
    string  chrom;        "Chromosome gene is on"
    uint    chromStart;   "Start position in chromosome"
    uint    chromEnd;     "End position in chromosome"
    string  name;         "Name of gene"
    uint    score;        "Score from 0 to 1000"
    char[1] strand;       "Strand: + or -"
    string  source;       "Source as in Sean Eddy's files."
    string  type;         "Type - snRNA, rRNA, tRNA, etc."
    float   fullScore;    "Score as in Sean Eddy's files."
    ubyte   isPsuedo;     "TRUE(1) if psuedo, FALSE(0) otherwise"
    )                                                                           
</PRE>

<H4><A name=Sage></A>Sage</H4>
<PRE>table sage
"Stores SAGE data in terms of UniGene identifiers"
    (
    int uni;               "Number portion of UniGene identifier"
    string gb;             "Genbank accession number"
    string gi;             "gi field in UniGene descriptions"
    lstring description;   "Description from UniGene fasta headers"
    int numTags;           "Number of tags"
    string[numTags] tags;  "Tags for this unique sequence"
    int numExps;           "Number of experiments"
    int[numExps] exps;     "Index of experiemtns in order of aves and stdevs"
    float[numExps] meds;   "Meida count of all tags for each experiment"
    float[numExps] aves;   "Average count of all tags for each experiment"
    float[numExps] stdevs  "Standard deviations of all counts for each experiment"
    )
</PRE>

<H4><A name=SageExp></A>Sage Experiments</H4>
<PRE>table sageExp
"Data related to SAGE experiments, tissue descriptions, etc."
    (
    int num;               "Index of the experiment in the sage table"
    string exp;            "Experiment name"
    int totalCount;        "Sum of all the tag counts for this experiment"
    string tissueType;     "Brief description of cells used"
    string tissueDesc;     "Brief description of tissues"
    string tissueSupplier; "Who supplied the tissue"
    string organism;       "Organism source of cells"
    string organ;          "Organ identifier"
    lstring producer;      "Source of tissue"
    lstring desription;    "Description of experiment"
    )
</PRE>

<H4><A name=wiggleTrack></A>Sample Tracks for Plotting Real-valued Data</H4>
The following definition is used for: chrN_humMusL, chrN_zoom2500_humMusL, 
chrN_zoom50_humMusL, chrN_zoom1_humMusL, affyTranscriptome.
Each table entry in the chrN_humMusL is displayed as a "mountain range" on
the browser and corresponds to one blastz alignment. The zoom tables
chrN_zoom2500_humMusL, chrN_zoom50_humMusL, and chrN_zoom1_humMusL are
summarized versions  (taking the MAXIMUM value in each bin) of the chrN_humMusL table. 
The zoom1 table is best for looking at the genome on very large scales and the zoom2500 table
is better when zoomed in further but not close enough for the actual
data. The sampleHeights can be converted into floating point valued 
L-scores with the formula: L = 8.0 * X / 1000.0 for sampleHeight X.

<PRE>table sample9
"Describes (x,y) pairs of samples"
    ushort bin;           "A field to speed indexing added to many
                           tables August 2001"
    string  chrom;        "Chromosome alignment is on."
    uint    chromStart;   "Start position in chromosome"
    uint    chromEnd;     "End position in chromosome"
    string  name;         "Unique index given to each block of samples"
    uint    score;        "Overall block score from 0 to 1000 (chrN_humMusL only)"
    char[1] strand;       "Strand: + or -"
    uint    sampleCount   "Number of samples in this block"
    uint[sampleCount]   samplePosition  "Base position of each sample in this
                                         block, relative to chromStart"
    uint[sampleCount]   sampleHeight    "Score for each sample (y-value) from 0 to 1000"
</PRE>

<H4><A name=SangerExtra></A>Sanger 20 and 22 Genes - Additional Information</H4>
<PRE>table sanger22extra
"Table with additional information about a Sanger 22 gene"
    (
    string name;	"Transcript name"
    string locus;	"Possibly biological short name"
    lstring description; "Description from Sanger gene GFFs"
    string geneType;	"Type field from Sanger gene GFFs"
    string cdsType;	"Type field from Sanger CDS GFFs"
    )
</PRE>

<H4><A name=SimpleRepeats></A>Simple Repeats</H4>
This table was generated by the trf program.  It does a more thorough job 
of finding simple tandem repeats than RepeatMasker does.
<PRE>table simpleRepeat
"Describes the Simple Tandem Repeats"
   (
   ushort  bin;           "A field to speed indexing added to many 
                           tables August 2001"
   uint    matches;       "Number of bases that match that aren't repeats"
   uint    misMatches;    "Number of bases that don't match"
   string chrom;          "Human chromosome or FPC contig"
   uint   chromStart;     "Start position in chromosome"
   uint   chromEnd;       "End position in chromosome"
   string name;           "Simple Repeats tag name"
   uint   period;         "Length of repeat unit"
   float  copyNum;        "Mean number of copies of repeat"
   uint   consensusSize;  "Length of consensus sequence"
   uint   perMatch;       "Percentage Match"
   uint   perIndel;       "Percentage Indel"
   uint   score;          "Score between  and .  Best is ."
   uint   A;              "Number of A's in repeat unit"
   uint   C;              "Number of C's in repeat unit"
   uint   G;              "Number of G's in repeat unit"
   uint   T;              "Number of T's in repeat unit"
   float  entropy;        "Entropy"
   string sequence;       "Sequence of repeat unit element"
   )
</PRE>

<H4><A name=SNPsites></A>SNP Positions</H4>
This table is used for both the snpNih and snpTsc tables.
<PRE>table snp
"Single nucleotide polymorphisms"
   (
   ushort  bin;           "A field to speed indexing added to many 
                           tables August 2001"
   uint    matches;       "Number of bases that match that aren't repeats"
   uint    misMatches;    "Number of bases that don't match"
   string chrom;          "Human chromosome or FPC contig"
   uint   chromStart;     "Start position in chromosome"
   uint   chromEnd;       "End position in chromosome"
   string name;           "Name of SNP"
   )                                                                           
</PRE>

<H4><A name=SoftberryHom></A>Softberry Genes - Protein Homologies</H4>
<PRE>table softberryHom
"Protein homologies behind Softberry genes"
    (
    string name;        "Softberry gene name"
    string giString;    "String with Genbank gi and accession"
    lstring description; "Freeform (except for no tabs) description"
    )
</PRE>

<H4><A name=LFS></A>Standard Linked Features</H4>
The phMouse and genscanSubopt tables use the first 7 fields of this table.
The bacEndPairs table uses all fields.
<PRE>table lfs
"Standard linked features series table"
   (
   short  bin;        "Bin number for browser speedup"
   string chrom;      "Human chromosome or FPC contig"
   uint   chromStart; "Start position of clone in chromosome"
   uint   chromEnd;   "End position of clone in chromosome"
   string name;       "Name of clone"

   uint   score;      "Score = 1000/(# of times clone appears in assembly)"
   char[1] strand;    "Value should be + or -"
   string pslTable;   "Table which contains corresponding PSL records for linked features"
   uint   lfCount;    "Number of linked features in the series"
   uint[lfCount]   lfStarts;  "Comma separated list of start positions of each linked feature in genomic"
   uint[lfCount]   lfSizes;   "Comma separated list of sizes of each linked feature in genomic"
   string[lfCount] lfNames;   "Comma separated list of names of linked features"
   )
</PRE>

<H4><A name=STSalias></A>STS marker aliases and associated identification numbers</H4>
<PRE>table stsAlias
"STS marker aliases and associated identification numbers"
    (
    string alias;       "STS marker name"
    uint identNo;       "Identification number of STS marker"
    string trueName;        "Official UCSC name for marker"
    )
</PRE>

<H4><A name=STSinfo></A>STS Marker Constant Information</H4>
<PRE>table stsInfo
"Constant STS marker information"
    (
    uint identNo;                 "UCSC identification number"
    string name;                  "Official UCSC name"
    uint gbCount;                 "Number of related GenBank accessions"
    string[gbCount] genbank;      "Related GeneBank accessions"
    uint gdbCount;                "Number of related GDB identifiers"
    string[gdbCount] gdb;         "Related GDB identifiers"
    uint nameCount;               "Number of alias names"
    string[nameCount] otherNames; "Alias names"
    uint dbSTSid;                 "ID number in UniSTS or dbSTS"
    uint otherDbstsCount;         "Number of related dbSTS IDs"
    uint[otherDbstsCount] otherDbSTS;  "Related dbSTS IDs"

    string leftPrimer;            "5' primer sequence"
    string rightPrimer;           "3' primer sequence"
    string distance;              "Length of STS sequence"
    string organism;              "Organism for which STS discovered"
    uint sequence;                "Whether the full sequence is available (1) or not (0) for STS"

    uint otherUCSCcount;          "Number of related active UCSC ids"
    uint[otherUCSCcount] otherUCSC;  "Related active UCSC ids"
    uint mergeUCSCcount;          "Number of merged inactive UCSC ids"
    uint[mergeUCSCcount] mergeUCSC;  "Related merged inactive UCSC ids"

    string genethonName;          "Name in Genethon map"
    string genethonChr;           "Chromosome in Genethon map"
    float genethonPos;            "Position in Genethon map"
    float genethonLOD;            "LOD score in Genethon map"
    string marshfieldName;        "Name in Marshfield map"
    string marshfieldChr;         "Chromosome in Marshfield map"
    float marshfieldPos;          "Position in Marshfield map"
    float marshfieldLOD;          "LOD score in Marshfield map"
    string wiyacName;             "Name in WI YAC map"
    string wiyacChr;              "Chromosome in WI YAC map"
    float wiyacPos;               "Position in WI YAC map"
    float wiyacLOD;               "LOD score in WI YAC map"
    string wirhName;              "Name in WI RH map"
    string wirhChr;               "Chromosome in WI RH map"
    float wirhPos;                "Position in WI RH map"
    float wirhLOD;                "LOD score in WI RH map"
    string gm99gb4Name;           "Name in GeneMap99 GB4 map"
    string gm99gb4Chr;            "Chromosome in GeneMap99 GB4 map"
    float gm99gb4Pos;             "Position in GeneMap99 GB4 map"
    float gm99gb4LOD;             "LOD score in GeneMap99 GB4 map"
    string gm99g3Name;            "Name in GeneMap99 G3 map"
    string gm99g3Chr;             "Chromosome in GeneMap99 G3 map"
    float gm99g3Pos;              "Position in GeneMap99 G3 map"
    float gm99g3LOD;              "LOD score in GenMap99 G3 map"
    string tngName;               "Name in Stanford TNG map"
    string tngChr;                "Chromosome in Stanford TNG map"
    float tngPos;                 "Position in Stanford TNG map"
    float tngLOD;                 "LOD score in Stanford TNG map"
    )
</PRE>

<H4><A name=STSinfoMouse></A>STS Marker Constant Information (Mouse)</H4>
<PRE>table stsInfoMouse
"Constant STS marker information for the mouse genome"
    (
    uint identNo;                "UCSC identification number"
    string name;                 "Official UCSC name"
    uint MGIPrimerID;            "STS primer's MGI ID, or 0 if N/A"
    string primerName;           "STS primer's name"
    string primerSymbol;         "STS primer's symbol"
    string primer1;              "Primer1 sequence"
    string primer2;              "Primer2 sequence"
    string distance;             "Length of STS sequence"
    uint sequence;               "Whether full sequence is available (1) 
                                     or not (0) for STS"
    uint MGIMarkerID;            "STS marker's MGI ID, or 0 if N/A"
    string stsMarkerSymbol;      "Symbol of STS marker"
    string Chr;                  "Chromosome in genetic map"
    float geneticPos;            "Position in genetic map 
                                     (-2 if N/A, -1 if syntenic)"
    string stsMarkerName;        "Name of STS Marker"
    uint LocusLinkID;            "Locuslink ID, or 0 if N/A"
    )
</PRE>

<H4><A name=STSmap></A>STS Clone Positions</H4>
<PRE>table stsMap
"STS marker and its position on golden path and various maps"
    (
    string chrom;	        "Chromosome or 'unknown'"
    int chromStart;             "Start position in chrom - negative 1 if unpositioned"
    uint chromEnd;        	"End position in chrom"
    string name;	        "Name of STS marker"
    uint score;	                "Score of a marker = 1000/(# of placements)"
    uint identNo;        	"Identification number of STS"
    string ctgAcc;	        "Contig accession number"
    string otherAcc;	        "Accession number of other contigs that the marker hits"

    string genethonChrom;	"Chromosome (no chr) from Genethon map or 0 if none"
    float genethonPos;		"Position on Genethon map"
    string marshfieldChrom;	"Chromosome (no chr) from Marshfield map or 0 if none"
    float marshfieldPos;	"Position on Marshfield map"
    string gm99Gb4Chrom;	"Chromosome (no chr) from GeneMap99 map or 0 if none"
    float gm99Gb4Pos;		"Position on gm99_bg4 map"
    string shgcTngChrom;	"Chromosome (no chr) from shgc_tng map or 0 if none"
    float shgcTngPos;		"Position on shgc_tng map"
    string shgcG3Chrom;		"Chromosome (no chr) from Stanford G3 map or 0 if none"
    float shgcG3Pos;		"Position on shgc_g3 map"
    string wiYacChrom;		"Chromosome (no chr) from Whitehead YAC map or 0 if none"
    float wiYacPos;		"Position on wi_yac map"
    string wiRhChrom;		"Chromosome (no chr) from Whitehead RH map or 0 if none"
    float wiRhPos;		"Position on wi_rh map"
    string fishChrom;		"Chromosome (no chr) from FISH map or 0 if none"
    string beginBand;		"Beginning of range of bands on FISH map"
    string endBand;		"End of range of bands on FISH map"
    string lab;			"Laboratory that placed the FISH clone"
    )
</PRE>

<H4><A name=STSmapMouse></A>STS Clone Positions (Mouse)</H4>
<PRE>table stsMapMouse
"STS marker and its position on mouse assembly"
    (
    string chrom;       "Chromosome or 'unknown'"
    int chromStart;     "Start position in chromosome (-1 if unpositioned)"
    uint chromEnd;      "End position in chromosome"
    string name;        "Name of STS marker"
    uint score;         "Score of a marker = 1000/(# of placements)"
    uint identNo;       "UCSC ID number"
    uint probeId;       "Probe identification number of STS"
    uint markerId;      "Marker identification number of STS"
    )
</PRE>

<H4><A name=STSsites></A>STS and FISH Clone Positions</H4>
(obsolete as of Apr. 2001, replaced by <A href="#STSmap">stsMap</A>)
<PRE>table stsMarker  
"STS marker and its position on golden path and various maps"
    ( 
    string chrom;            "Chromosome or 'unknown'"
    int    chromStart;       "Start position in chrom (-1 if
                                unpositioned)"
    uint   chromEnd;         "End position in chrom" 
    string name;             "Name of STS marker"  
    uint   score;            "Score of a marker - depends on how 
                                many contigs it hits"  
    uint   identNo;          "Identification number of STS" 
    string ctgAcc;           "Contig accession number
    string otherAcc;         "Accession number of other 
                                contigs that the marker hits"  
    string genethonChrom;    "Chromosome from Genethon map or 0 if none" 
    float  genethonPos;      "Position on Genethon map"
    string marshfieldChrom;  "Chromosome from Marshfield map or 0 if none"
    float  marshfieldPos;    "Position on Marshfield map" 
    string gm99Gb4Chrom;     "Chromosome from GeneMap99 map or 0 if none"
    float  gm99Gb4Pos;       "Position on GeneMap99 map" 
    string shgcG3Chrom;      "Chromosome from G3 map or 0 if none"
    float  shgcG3Pos;        "Position on G3 map" 
    string wiYacChrom;       "Chromosome from Whitehead YAC map or 0 if none"
    float  wiYacPos;         "Position on Whitehead YAC map" 
    string shgcTngChrom;     "Chromosome from TNG map or 0 if none" 
    float  shgcTngPos;       "Position on TNG map" 
    string fishChrom;        "Chromosome from FISH map or 0 if none" 
    string beginBand;        "Beginning of range of bands on FISH map"
    string endBand;          "End of range of bands on FISH map"
    string lab;              "Laboratory that placed the FISH clone"
    )             

<H4><A name=TilingPath></A>Tiling Path of Clones through a Chromosome</H4>
<PRE>table tilingPath
"A tiling path of clones through a chromosome"
    (
    string chrom;	"Chromosome name: chr1, chr2, etc."
    string accession; "Clone accession or ? or GAP"
    string clone;     "Clone name in BAC library"
    string contig;    "Contig (or gap size)"
    int chromIx;      "Number of clone in tiling path starting chrom start"
    )
</PRE>

	<BR></TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>

	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B><A name=MrnaInfo></A>&nbsp; mRNA and EST Data</B></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10>&nbsp;</TD><TD>

This is a family of tables. The main table, <TT>mrna</TT>, consists of minor mRNA and EST identification
information, and then ids pointing to the corresponding tables which contain other information.  For example,
to determine the organism, the 'organism' field in the <TT>mrna</TT> table contains an id. Finding this id
in the corresponding <TT>organism</TT> table will give the name of the organism.


<H4>Main mRNA/EST Information Table</H4>
<PRE>table mrna
"Basic information about mRNA and ESTs with links to other tables"
   (
   uint   id               "Id, same as seq ID"
   string acc              "Genbank accession"
   enum   type             "Full length ('mRNA') or EST ('EST')"
   enum   direction        "Read direction ('5','3','0')"
   uint   source           "Ref in source table"
   uint   organism         "Ref in organism table"
   uint   library          "Ref in library table"
   uint   mrnaClone        "Ref in clone table"
   uint   sex              "Ref in sex table"
   uint   tissue           "Ref in tissue table"
   uint   development      "Ref in development table"
   uint   cell             "Ref in cell table"
   uint   cds              "Ref in CDS table"
   uint   keyword          "Ref in key table"
   uint   description      "Ref in description table"
   uint   geneName         "Ref in geneName table"
   uint   productName      "Ref in productName table"
   uint   author           "Ref in author table"
   )
</PRE>

<H4>mRNA/EST Characteristic Tables</H4>
This table is used for the author, bacClone, cds, cell, center, chromosome, cytoMap, description, development, geneName,
keyword, library, mrnaClone, organism, productName, sex, source, and tissue tables.  Each has a 
reference in the above mrna table.
<PRE>table mrnaCharacteristic
"Characteristics of mRNA and EST referenced in the mrna table"
   (
   int    id               "Unique id of characteristic in each table"
   string name             "Characteristic"
   )
</PRE>

	<BR></TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>

	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B>&nbsp; Internal Use Tables</B></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10>&nbsp;</TD><TD>

<H4><A name=TrackDB></A>Annotation Track Description</H4>
<PRE>table trackDb
"This describes an annotation track."
  (
  string tableName;   "Symbolic ID of Track"
  string shortLabel;  "Short label displayed on left"
  string type;        "Track type: bed, psl, genePred, etc."
  string longLabel;   "Long label displayed in middle"
  ubyte visibility;   "0=hide, 1=dense, 2=full"
  float priority;     "0-100 - where to position.  0 is top"
  ubyte colorR;	      "Color red component 0-255"
  ubyte colorG;	      "Color green component 0-255"
  ubyte colorB;	      "Color blue component 0-255"
  ubyte altColorR;    "Light color red component 0-255"
  ubyte altColorG;    "Light color green component 0-255"
  ubyte altColorB;    "Light color blue component 0-255"
  ubyte useScore;     "1 if use score, 0 if not"
  ubyte private;      "1 if only want to show it on test site"
  int restrictCount;  "Number of chromosomes this is on (0=all!)"
  string[restrictCount] restrictList;  "List of chromosomes this is on"
  lstring url;	      "URL to link to when they click on an item"
  lstring html;       "Some html to display when they click on an item"
  )
</PRE>

<H4><A name=HistoryInfo></A>History Table - Tracks Modification History of mRNA Table and Linked Tables</H4>
<PRE>table history
  (
  int       ix        "Update number"
  uint      startId   "Start this session's ids"
  uint      endId     "First id for next session"
  string    who       "User who updated"
  string    what      "What they did"
  timestamp when      "When they did it"
  )
</PRE>

<H4><A name=AxtInfo></A>Reference to an Axt File</H4>
<PRE>table axtInfo
"Axt alignment names and sizes"
    (
    string species;   "Long name of species"
    string alignment; "Name of alignment"
    string chrom;     "Chromosome name"
    string fileName;  "Axt file name"
    )
</PRE>

<H4><A name=extFileInfo></A>Reference to an External File</H4>
<PRE>table extFile 
"This keeps track of external files and directories"
   (
   uint   id           "Unique ID for each file"
   string name         "Symbolic name of file"
   string path         "Full path. Ends in '/' if a dir"
   uint   size         "Size of file (checked)"
   )
</PRE>

<H4><A name=seqInfo></A>Reference to an External Sequence in an FA File</H4>
<PRE>table seq
"This keeps track of a sequence"
  (
  uint   id           "Unique ID for each external sequence"
  string acc          "GenBank accession number"
  uint   size         "Size of sequence in bases"
  date   gb_date      "GenBank last modified date"
  uint   extFile      "File it is in"
  uint   file_offset  "Offset in file"
  uint   file_size    "Size in file."
  )
</PRE>

	<BR></TD><TD WIDTH=7></TD></TR></TABLE>
	</TD></TR></TABLE>
			
</TD></TR></TABLE>
</BODY></HTML>
