<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: News Archives</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../images/title.jpg"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/topbar.html"--></TD></TR>
<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>	
	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B> &nbsp;News Archives </B></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>

<UL>
<LI><A CLASS="toc" HREF="#2008">2008 News</A></LI>
<LI><A CLASS="toc" HREF="#2007">2007 News</A></LI>
<LI><A CLASS="toc" HREF="#2006">2006 News</A></LI>
<LI><A CLASS="toc" HREF="#2005">2005 News</A></LI>
<LI><A CLASS="toc" HREF="#2004">2004 News</A></LI>
<LI><A CLASS="toc" HREF="#2003">2003 News</A></LI>
<LI><A CLASS="toc" HREF="#2002">2002 News</A></LI>
<LI><A CLASS="toc" HREF="#2001">2001 News</A></LI>
</UL>

<!---------------- 2008 archived news --------------->
<A NAME="2008"></A>

<HR>
<A NAME="042308"></A>
<FONT SIZE=4 COLOR="#006666">23 Apr. 2008 - Marmoset Browser Released</B></FONT>
		    <P>
		    We'd like to announce the release of a Genome Browser and 
		    Blat server for the marmoset genome (<em>Callithrix 
		    jacchus</em>). 
		    The June 2007 assembly -- WUSTL version Callithrix 
		    jacchus-2.0.2, UCSC version calJac1 -- was produced by 
		    Washington University St. Louis (WUSTL) School of Medicine 
		    Genome Sequencing Center in St. Louis, MO, USA. </P> 
		    <P>
		    <em>C. jacchus</em>, a member of the New World 
		    monkey clade, is the most widely studied marmoset. It is a
		    popular non-human primate model due to its small body size
		    and unique biological features, and has contributed to the 
		    study of brain function, immunity, reproductive biology and 
		    drug toxicity. <em>C. jacchus</em> marmosets typically give 
		    birth to twins that are somatic chimeras, i.e. each sibling 
		    is the mixture of sibling genotypes. (Excerpted from the 
		    WUSTL <em>C. jacchus</em>
		    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Callithrix%20jacchus"
		    TARGET=_blank> project</A> page.) </P>
		    <P>
		    The <em>C. jacchus</em> genome was sequenced to 6X coverage 
		    using DNA from a female marmoset provided by the 
		    Southwestern
		    National Primate Research Center in San Antonio, TX, USA.
		    DNA from a full brother of the female was used as the source
		    for the CHORI-259 BAC library. This assembly is composed of 
		    49,724 supercontigs containing a total of approximately 
		    3.02 billion bases. For more statistics and details
		    on the assembly process, refer to the WUSTL 
		    Callithrix_jacchus-2.0.2
		    <A HREF="http://genome.wustl.edu/pub/organism/Primates/Callithrix_jacchus/assembly/Callithrix_jacchus-2.0.2/ASSEMBLY"
		    TARGET=_blank>assembly</A> page. </P>
		    <P>
		    Bulk downloads of the calJac1 sequence and annotations may 
		    be obtained from the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/calJac1/"
		    TARGET=_blank>FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#marmoset">Downloads</A> 
		    page. The sequence data can also be obtained directly from 
		    <A HREF="http://genome.wustl.edu/pub/organism/Primates/Callithrix_jacchus/assembly/Callithrix_jacchus-2.0.2/"
		    TARGET=_blank>WUSTL</A>. See the WUSTL 
		    <A HREF="http://genome.wustl.edu/data.cgi"
		    TARGET=_blank>data use policy</A> for conditions of use.
		    Please acknowledge WUSTL School of Medicine Genome 
		    Sequencing Center in any publications that result from the 
		    use of this sequence assembly. </P>
		    <P>
		    We'd like to thank WUSTL School of Medicine Genome 
		    Sequencing Center for providing this assembly. The 
		    initial set of marmoset browser annotation tracks were 
		    generated by UCSC. The UCSC marmoset Genome Browser team is
		    Hiram Clawson, Robert Kuhn, Pauline Fujita, Brooke Rhead, 
		    and Donna Karolchik. See the 
		    <A HREF="credits.html#marmoset_credits">Credits</A>
		    page for a detailed list of the organizations and
		    individuals who contributed to this release.</P>
		    <P>
		    <HR>

<A NAME="041508"></A>
<FONT SIZE=4 COLOR="#006666">15 Apr. 2008 - Introducing the Archaeal Genome Browser Database</B></FONT>
		    <P>
		    We'd like to introduce the Archaeal Genome Browser Database,
		    a visualization tool and an integrated repository for 
		    archaeal functional genomics data. This resource was
		    developed by the 
		    <A HREF="http://lowelab.ucsc.edu/" TARGET=_blank>Lowe 
		    Lab</A> in the UCSC Biomolecular Engineering Department, 
		    with key assistance from the UCSC Genome Browser team. 
		    The Archaeal Genome Browser Database home page can be
		    accessed via the 
		    <A HREF="http://archaea.cse.ucsc.edu/"
		    TARGET=_blank>Archaeal Genomes</A> menu link on
		    the Genome Browser home page.</P>
		    <P>
		    Currently there are more than 50 completed archaeal genomes,
		    the least studied domain of life. Although archaea and 
		    bacteria are both prokaryotes, often co-existing in the 
		    same environments, many aspects of archaeal cell biology 
		    such as DNA replication, repair, transcription, and 
		    translation are homologous to those found in eukaryotes.    
		    Some members of archaea are also notable for inhabiting 
		    extreme environments, including boiling terrestrial hot 
		    springs, black smoker vents at the bottom of the ocean, 
		    the ultra briny water of the Dead Sea, and highly acidic 
		    drainage water from ore mines, to name a few.</P>
		    <P>
		    The Archaeal Genome Browsers offer a variety of basic tracks
		    derived from Genbank RefSeq annotation, along with published
		    genome analyses from the Lowe Lab and external groups. The
		    annotations include operon predictions, regulatory sequence 
		    motifs (promoters and Shine-Dalgarno), microarray data, 
		    multi-genome alignments, and protein conservation across 
		    major phylogenetic groups. 
		    <P>
		    The goal of the Lowe Lab is to make the Archaeal Genome 
		    Browser Database a forum for ongoing community-based genome 
		    annotation, and they welcome new experimental and 
		    bioinformatic analyses. If you would like to contribute 
		    data, or have questions or feedback about the database, 
		    contact 
		    <A HREF="mailto:&#108;o&#119;&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;&#100;&#117;">Todd Lowe</A>. </P>
		    <P>
		    <HR>

<A NAME="040808"></A>
<FONT SIZE=4 COLOR="#006666">8 Apr. 2008 - Updated Cow Genome Browser Now Available</B></FONT>
		    <P>
		    The UCSC Genome Bioinformatics Group has released a Genome 
		    Browser and Blat server for the Oct. 2007 draft assembly of
 		    the Cow genome <em>Bos taurus</em>. This assembly 
		    (UCSC version bosTau4) was produced by the Baylor College of
		    Medicine Human Genome Sequencing Center (BCM HGSC) as
		    Baylor release Btau_4.0. </P>
		    <P>
		    The Btau_4.0 release was produced using the 
		    <A HREF="http://www.genome.org/cgi/content/abstract/14/4/721">Atlas</A>
		    genome assembly system at BCM HGSC. 
		    The sequencing strategy combined BAC shotgun reads with 
		    whole genome shotgun reads from small insert libraries as 
		    well as BAC end sequences. The assembly contains chromosomes
		    1-29 and X as well as 11869 scaffolds (named chrUn.004.*). 
		    The mitochondrial sequence (available in the browser as 
		    &quot;chrM&quot;) was obtained from Genbank accession 
		    GI:60101824.</P>
		    <P>
		    The Btau_4.0 assembly was tested against available bovine 
		    sequence data sets (EST sequences and finished BAC 
		    sequences) for extent of coverage (completeness). When 
		    assembled contigs were tested, over 95% of the sequences in 
		    these data sets were found to be represented, indicating 
		    that the shotgun libraries 
		    used to sequence the genome were comprehensive. Of the 1.04 
		    million EST sequences 95.0% were contained in the assembled 
		    contigs. Assuming the ESTs are uniformly distributed
		    throughout the genome, the estimated genome size is 
		    2.73Gb/95% = 2.87Gb.
		    For detailed information on the sequencing and assembly 
		    techniques, see the Baylor 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/bovine/"
		    TARGET=_blank>Bovine Genome Project</A> web page.</P>
		    <P>
		    For a list of the chromosomes and scaffolds in this 
		    assembly, click the &quot;Sequences&quot; link on the cow 
		    browser 
		    <A HREF="../../cgi-bin/hgGateway?db=bosTau4">gateway</A> 
		    page. </P> 
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/bosTau4">FTP 
		    server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#cow">Downloads</A> 
		    page. These data have 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html"
		    TARGET=_blank>specific conditions for use</A>. The cow 
		    annotation tracks were 
		    generated by UCSC and collaborators worldwide.</P>
		    <P>
		    We'd like to thank Baylor College of Medicine for providing 
		    this assembly. The UCSC bosTau4 browser and documentation 
		    were produced by Hiram Clawson, Brian Raney, and Ann Zweig. 
		    See the <A HREF="credits.html#cow_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.  </P>
                    <P>
		    <HR>

<A NAME="032408"></A>
<FONT SIZE=4 COLOR="#006666">24 Mar. 2008 - Free Genome Browser Institutional Seminars</B></FONT>
		    <P>
		    The UCSC Bioinformatics Group announces your chance to 
		    bring our hands-on computer workshop on the UCSC Genome 
		    Browser to your institution. The seminar is presented by 
		    our training partner, <A HREF="http://www.openhelix.com"
		    TARGET=_blank>OpenHelix</A>.</P>
		    <P>
		    The 3-1/2-hour introductory tutorial will cover the topics 
		    needed to effectively use our tool set, including: basic 
		    functionality of Genome Browser searching and BLAT use, 
		    Table Browser use, creating and using Custom Tracks, and 
		    an introduction to the Gene Sorter. The workshop material
		    requires knowledge of genomic/biological concepts, but no 
		    programming skills are required.
		    Participants will receive slide hand-outs, exercises, and 
		    UCSC Genome Browser and Table Browser Quick Reference Cards.
		    </P>
		    <P>
		    To bring the seminar to your institution, apply on the
		    OpenHelix <A HREF="http://www.openhelix.com/seminars"
		    TARGET=_blank>website</A>. Participating institutions are
		    required only to:</P>
		    <P>
		    --Provide a computer classroom with internet access. The
		    preferred number of computers is 24 or above. As an 
		    alternative to a computer room, the institution may provide
		    a classroom with wireless access and ask participants to 
		    bring their own laptops. The seminar can be done in the 
		    morning and repeated in the afternoon to reach more users.
		    </P>
		    <P>
		    --Conduct outreach and communication to participants as 
		    necessary.
		    </P>
		    <P>
		    The number of seminars is limited --  apply today!
		    Contact information:  <A HREF="http://www.openhelix.com"
		    TARGET=_blank>www.openhelix.com</A> or 1-888-861-5051.
		    </P>
		    <P>
		    <B>About OpenHelix, LLC:</B> OpenHelix provides the genomics
		    knowledge you need when you need it. OpenHelix offers 
		    online self-run tutorials, web seminars, and on-site 
		    training for institutions and companies on the most 
		    powerful and popular free, web-based, publicly accessible 
		    bioinformatics resources. In addition, OpenHelix is 
		    contracted by resource providers to provide comprehensive, 
		    long-term training and outreach programs.  The company
		    has its headquarters in Seattle, with offices in San 
		    Francisco and Boston. Further information can be found at 
		    www.openhelix.com or by calling 1-888-861-5051.</P>
                    <P>
		    <HR>

<A NAME="031008"></A>
<FONT SIZE=4 COLOR="#006666">10 Mar. 2008 - Orangutan Genome Browser Now Available</B></FONT>
		    <P>
		    The UCSC Genome Bioinformatics Group has released a Genome 
		    Browser and Blat server for the Jul. 2007 draft assembly of
 		    the Sumatran orangutan genome, <em>Pongo pygmaeus 
		    abelii</em>. This assembly (UCSC version 
		    ponAbe2, WUSTL version Pongo_albelii-2.0.2) was provided by 
		    the Genome Sequencing Center at Washington University School
		    of Medicine in St. Louis (WUSTL), MO, USA. </P>
		    <P>
		    The orangutan genome was sequenced to 6X coverage 
		    using a female orangutan known as &quot;Susie&quot; from 
		    the Gladys Park Zoo (Brownsville, TX, USA). The combined 
		    sequence reads were assembled using 
		    <A HREF="http://www.genome.org/cgi/content/abstract/13/9/2164"
		    TARGET=_blank>PCAP</A> and filtered for all 
		    known non-orangutan sequence contaminants. For more details
		    about the assembly, see the orangutan browser
		    <A HREF="../cgi-bin/hgGateway?db=ponAbe2">gateway</A>
		    page and the WUSTL 
		    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Pongo%20abelii"
		    TARGET=_blank><em>Pongo abelii</em></A> web page.</P>
		    <P>
		    Of the 3.09 Gb of total sequence, 3.08 Gb are ordered and 
		    oriented along the chromosomes. Gap sizes between 
		    supercontigs were estimated based on their size in human, 
		    with a maximum gap size of 30 kb allowed. For a list of the
		    chromosomes in this assembly, click the 
		    "Sequences" link on the orangutan browser 
		    <A HREF="../../cgi-bin/hgGateway?db=ponAbe2">gateway</A> 
		    page.
		    The mitochondrial sequence is also available as the virtual 
		    chromosome &quot;chrM&quot;.</P> 
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/ponAbe2">FTP 
		    server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#orangutan">Downloads</A> 
		    page. These data have 
		    <A HREF="credits.html#orangutan_use">specific 
		    conditions for use</A>. The orangutan browser annotation 
		    tracks were generated by UCSC and collaborators worldwide. 
		    <P>
		    We'd like to thank WUSTL for providing this
		    assembly. The UCSC ponAbe2 browser and documentation were
		    produced by Hiram Clawson, Kayla Smith, Robert Kuhn, Ann
		    Zweig and Donna Karolchik.
		    See the <A HREF="credits.html#orangutan_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. 
		    </P>
                    <P>
                    <HR>

<A NAME="030708"></A>
<FONT SIZE=4 COLOR="#006666">7 Mar. 2008 - New Reverse Functionality Released</B></FONT>
		    <P>
		    The Genome Browser team is happy to announce new 
		    functionality in the main genome browser track display.  A 
		    new configuration button, "Reverse," now allows users to 
		    view the entire browser image flipped right-to-left.
                    This is especially useful when a user's gene of interest 
		    aligns on the opposite strand from the reference assembly. 
		    Those genes then appear in the 5' to 3' direction. </P>
                    <P>
		    When the reverse function has been activated, all of the 
		    track labels usually displayed on the left side of the track
		    (including the mini-button providing access to configuration
		    options) are displayed on the right side.  This allows users
		    to see at a glance which direction is being displayed.  All 
		    navigation options operate as expected.</P>
                    <P>
		    Thanks to Mark Diekhans, Ann Zweig, Robert Kuhn and the
                    rest of the engineering team for implementing this feature.
		    </P>
                    <HR>
<A NAME="010808"></A>
<FONT SIZE=4 COLOR="#006666">8 Jan. 2008 - Additional Job Opening with UCSC Genome Browser Project</B></FONT>
                    <P>
                    In addition to the openings listed in the 12 Dec. 2007
                    announcement (see below), the UCSC Genome Browser project is
                    accepting applications for Research Software Architect,
                    a position in the UC Project Scientist academic series.
                    For the job description, qualifications and application
                    information, please
                    see the Center for Biomolecular Science and Engineering
                    <A HREF="http://www.cbse.ucsc.edu/jobs/jobs_staff.shtml#architect"
                    TARGET=_blank>website</A>. To ensure full consideration,
                    applications must be received by 22 Jan. 2008.</P>
		    <HR>


<!---------------- 2007 archived news --------------->
<A NAME="2007"></A>
<A NAME="121207"></A>
<FONT SIZE=4 COLOR="#006666">12 Dec. 2007 - Job Openings with UCSC Genome Browser Project</B></FONT>
                    <P>
                    The UCSC Genome Browser project is currently accepting
                    applications for two positions on our development team:
                    Software Development Engineer (Programmer/Analyst 3) and
                    Biological Database Testing/User Support Technician
                    (Programmer/Analyst 1).
                    We are looking for talented self-motivated individuals who
                    would like to use their skills in computer science,
                    biology, and bioinformatics on a fast-paced project
                    featuring the work of top genomics scientists worldwide.
                    </P>

                    <P>
                    For a summary of the position details and qualifications,
                    see the entries for
                    <A HREF="http://www.cbse.ucsc.edu/jobs/jobs_staff.shtml#sftwrdevel"
                    TARGET=_blank>Software Developer</A> and
                    <A HREF="http://www.cbse.ucsc.edu/jobs/jobs_staff.shtml#bdt"
                    TARGET=_blank>Biological Database Testing/User Support
                    Technician</A> on the Center for Biomolecular Science and
                    Engineering (CBSE) website. For detailed job descriptions
                    and application information,  go to the
                    <A HREF="http://jobs.ucsc.edu/" TARGET=_blank>UCSC Staff
                    Employment website</A>, click the &quot;Search
                    Postings&quot; link on the sidebar, and type in job
                    #0701419 (Software Developer) or job #0701391 (Testing/User
                    Support Technician).
                    </P>
		    <HR>

<A NAME="101507"></A>
<FONT SIZE=4 COLOR="#006666">15 Oct. 2007 - Genome Browser for Zebrafish Zv7 Released</B></FONT>
		    <P>
		    The latest zebrafish assembly -- Zv7 (UCSC version 
		    danRer5, July 2007) -- is now available in the UCSC
		    Genome Browser. The Zv7 assembly was produced by 
		    The Wellcome Trust Sanger Institute in collaboration with 
		    the Max Planck Institute for Developmental Biology in 
		    Tuebingen, Germany, and the Netherlands Institute for 
		    Developmental Biology (Hubrecht Laboratory), Utrecht, 
		    The Netherlands.</P>
		    <P>
		    This assembly consists of 1,440,582,308 bp in 5,036 
		    fragments. It includes sequence anchored to chromosomes 
		    1-25 and chrM (mitochondrial), as well as 5010 unplaced 
		    scaffolds.  The assembly was produced by integrating 
		    finished clone sequence from the physical map with whole 
		    genome shotgun assembly sequence. The N50 size is 
		    1,153,933, n = 277 (i.e. the length such that 50% of the 
		    assembled genome lies in blocks of the N50 size or longer). 
		    For more information about this assembly, see the Sanger 
		    Institute web page for the 
		    <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/Zv7_assembly_information.shtml"
		    TARGET=_blank><em>Danio rerio</em> Sequencing Project</A> 
		    and the track description page for the Assembly track.</P>
		    <P>
		    The danRer5 sequence and annotation data can be 
		    downloaded from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/danRer5/">FTP server</A>
		    or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#zebrafish">downloads</A>
		    page. Please review the 
		    <A HREF="http://www.sanger.ac.uk/notices/use-policy.shtml"
		    TARGET=_blank>guidelines</A> for using these data.
		    <P>
		    We'd like to thank the Wellcome Trust Sanger Institute, 
		    the Max Planck Institute for Developmental Biology,
		    Hubrecht Laboratory and the other
		    institutions who contributed to the sequencing and 
		    mapping effort of this release. Special thanks to the 
		    Zebrafish Genome Initiative at Children's Hospital in 
		    Boston for their collaboration on this release. 
		    The UCSC zebrafish Genome Browser
		    was produced by Rachel Harte, Ann Zweig, and Donna 
		    Karolchik. The 
		    initial set of annotation tracks was generated by the 
		    UCSC Genome Bioinformatics Group.  See the
		    <A HREF="credits.html#zebrafish_credits">credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to the release of
		    this browser.</P>
		    <P>
		    <HR>
<A NAME="101007"></A>
<FONT SIZE=4 COLOR="#006666">10 Oct. 2007 - New Browser for Purple Sea Urchin</B></FONT>
		    <P>
		    The Sep. 2006 release of the purple sea urchin genome 
		    (<em>Strongylocentrotus purpuratus</em>) is now available
		    in the UCSC Genome Browser. This assembly, UCSC version
		    strPur2, was produced by the Baylor College of Medicine 
		    Human Genome Sequencing Center (BCM HGSC) and corresponds 
		    to their Spur_2.1 assembly. </P>
		    <P>
		    The Spur_2.1 release was assembled from BAC sequence reads 
		    (approximately 2x coverage) and whole genome shotgun 
		    reads (6x coverage), and utilizes BAC tiling path 
		    information.  The BCM HGSC Atlas-2.0 genome 
		    assembly system was used to generate this assembly.
		    The total length of all contigs greater than 1kb is 804 
		    Mbp.  When the gaps between contigs in scaffolds are 
		    included, the total span of the assembly is 907 Mbp.  The 
		    estimated size of the genome based on the assembly is 814 
		    Mbp.</P>
		    <P>
		    Compared to previous sea urchin releases, the Spur_2.1 
		    assembly is more continuous and has fewer false 
		    duplications; contaminations identified in the previous
		    Spur_2.0 assembly have been removed.
		    This draft assembly may contain errors; therefore, users 
		    should exercise caution. Typical errors may include 
		    misassemblies of repeat sequences, 
		    collapses of repeat regions, and artificial duplications in 
		    polymorphic regions. However, base accuracy in contigs is 
		    usually very high with most errors near the ends of contigs.
		    </P>
		    <P>
		    More assembly details can be found in the 
		    <A HREF="ftp://ftp.hgsc.bcm.tmc.edu/pub/data/Spurpuratus/fasta/Spur_v2.1/README_Spur_v2.1.txt"
		    TARGET=_blank>Spur_2.1 README file</A> and on the BCM HGSC 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/seaurchin/"
		    TARGET=_blank>Sea Urchin Genome Project</A> web page.</P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/strPur2">FTP server</A>
		    or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#urchin">Downloads</A> 
		    page. These data have 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
		    TARGET=_blank>specific conditions for use</A>. The initial
		    set of strPur2 annotation tracks was generated by UCSC. 
		    See the <A HREF="credits.html#urchin_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. </P>
		    <P>
		    <HR>

<A NAME="100407"></A>
<FONT SIZE=4 COLOR="#006666">4 Oct. 2007 - ENCODE Genome Browser Released for hg18 Assembly</B></FONT>
		    <P>
		    The UCSC ENCODE browser for the human genome assembly hg18 
		    (NCBI Build 36) is now available. You can access the browser
		    directly at 
		    <A HREF="../ENCODE/encode.hg18.html">http://genome.ucsc.edu/ENCODE/encode.hg18.html</A> 
		    or by clicking the ENCODE link in the sidebar menu on this 
		    page, then clicking the Regions (hg18) link in the sidebar 
		    menu on the ENCODE portal page.</P>
		    <P>
		    The hg18 ENCODE browser includes 540 data tables in 59 
		    browser tracks that were migrated from the hg17 browser. 
		    The hg17 data coordinates were converted to hg18 coordinates
		    using the UCSC liftOver process.</P>
		    <P>
		    To improve the accessibility of the data, related ENCODE 
		    tracks have been gathered into new configuration groupings 
		    (&quot;super-tracks&quot;) that can be displayed or hidden
		    using a single visiblity control.  We have also reduced the
 		    number of track groups and have modified some of the group 
		    names for clarity. ENCODE tracks with whole-genome data 
		    have been moved into the standard browser track groups.</P>
		    <P>
		    For more information about the hg18 ENCODE data migration, 
		    see the News section on the UCSC 
		    <A HREF="../ENCODE/">ENCODE portal page</A> and the UCSC 
		    <A HREF="http://genomewiki.cse.ucsc.edu/index.php/ENCODE_Hg18_Migration" TARGET=_blank>genomeWiki</A>.</P>
		    <P>
		    <HR>

<A NAME="082207b"></A>
<FONT SIZE=4 COLOR="#006666">22 Aug. 2007 - New Mouse Browser Released</B></FONT>
		    <P>
		    The latest mouse genome assembly from the Mouse Genome
		    Sequencing Consortium, NCBI Build 37.1, is now available in 
		    the UCSC Genome Browser. This version (UCSC version mm9) is
		    considered to be essentially finished.
		    <P>
		    The Build 37.1 assembly includes approximately 2.6 Gb of
		    sequence on chromosomes 1-19, X, Y, M (mitochondrial DNA) 
		    and Un (unmapped clone contigs). 
		    In-depth information about this assembly will become 
		    available on the 
		    <A HREF="http://www.ncbi.nlm.nih.gov/mapview/stats/BuildStats.cgi?taxid=10090&build=37&ver=1"
                    TARGET=_blank>NCBI website</A>.
		    On chromosome Y in this assembly, only the short arm has 
		    reliable mapping data; therefore, most of the contigs on 
		    the Y chromosome are unplaced.</P>
                    <P>
                    The mm9 sequence and annotation data may be downloaded 
		    from the Genome Browser
                    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/mm9/">FTP
                    server</A> or
                    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/mm9/">Downloads</A>
                    web page. The mm9 annotation tracks were generated by UCSC
                    and collaborators worldwide. 
		    NOTE: To expedite the availability of the mm9 browser on our
		    website, the initial release does not contain the 
		    comparative genomics annotations. These will be added to 
		    our website as they become available. Also, note that
		    the UCSC mm9 database contains only the reference strain 
		    C57BL/6J.</P>
		    <P>
                    We'd like to thank Deanna Church and the Mouse Genome 
		    Sequencing Consortium for this assembly. We'd also like to
                    acknowledge the work of the UCSC mm9 team: Hiram Clawson, 
		    Archana Thakkapallayil, Robert Kuhn, and Donna Karolchik.
		    For a complete list
		    of the individuals and organizations who participated 
		    in this assembly, see the 
		    <A HREF="credits#mouse_credits">Credits</A> 
		    page. </P>
		    <P>
		    <HR>

<A NAME="082207a"></A>
<FONT SIZE=4 COLOR="#006666">22 Aug. 2007 - Latest Bovine Assembly Available</B></FONT>
		    <P>
		    The UCSC Genome Bioinformatics Group has released a 
		    Genome Browser and Blat server for the Aug. 2006 Btau_3.1
		    draft assembly of the cow genome. This assembly (UCSC
		    version bosTau3) was provided by Baylor
		    College of Medicine Human Genome Sequencing Center in
		    Houston, TX.
		    <P>
		    The Btau_3.1 release was produced by the Atlas genome 
		    assembly system at Baylor College of Medicine Human Genome 
		    Sequencing Center. The  sequencing strategy combined BAC 
		    shotgun reads with whole genome shotgun reads from small 
		    insert libraries as well as BAC end sequences. The assembly 
		    contains chromosomes 1-29 and X as well as 13045 scaffolds 
		    (named chrUn.003.*). 
		    More information on the Btau_3.1 assembly can be found on
		    the Baylor 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/bovine/"
		    TARGET=_blank>Bovine Genome Project</A> web page and the
		    <A HREF="ftp://ftp.hgsc.bcm.tmc.edu/pub/data/Btaurus/fasta/Btau20060815-freeze/ReadMeBovine.3.1.txt" 
		    TARGET=_blank>Readme</A> file that accompanies this release.
		    </P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/bosTau3/">FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#cow">Downloads</A> 
		    page. Please refer to the Baylor 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
		    TARGET=_blank>conditions of use</A> regarding these
		    data. The bosTau3 annotation tracks were generated by 
		    UCSC and collaborators worldwide. 
		    <P>
		    We'd like to thank Baylor College of Medicine for 
		    the bovine sequence and assembly. We'd also like to 
		    acknowledge the members of the UCSC Genome Bioinformatics
		    Group who contributed to the Cow Genome Browser:
		    Heather Trumbower, Angie Hinrichs, Kayla Smith, and Donna 
		    Karolchik. See the
		    <A HREF="credits.html#cow_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. 
		    <P>
		    <HR>

<A NAME="080207"></A>
<FONT SIZE=4 COLOR="#006666">2 Aug. 2007 - Five New Worm Assemblies</B></FONT>
		    <P>
		    We've updated our existing nematode browsers--for <em>C. 
		    elegans</em> and <em>C. briggsae</em>--to the latest 
		    publicly available assemblies. In addition, we've added
		    three new worms to our collection: <em>C. brenneri</em> 
		    (Caenorhabditis n. sp. PB2801), <em>C. remanei</em>,
		    and <em>Pristionchus pacificus</em>. The <em>C. elegans</em>
		    sequence was obtained from WormBase; the Genome Sequencing
		    Center at Washington University in St. Louis (WUSTL) 
		    provided sequence data for the other four assemblies.</P>

            	    <TABLE BORDER=1 BORDERCOLOR="#aaaaaa" WIDTH="100%">       
		      <TR>
		        <TD nowrap><FONT color="#006666"><B>SPECIES</B></FONT></TD>
                        <TD nowrap><FONT color="#006666"><B>UCSC VERSION</B></FONT></TD>
                    	<TD nowrap><FONT color="#006666"><B>RELEASE DATE</B></FONT></TD>
                    	<TD nowrap><FONT color="#006666"><B>SOURCE/RELEASE</B></FONT></TD>
                      </TR>
  		      <TR><TD><em>C. elegans</em></TD><TD>ce4</TD><TD>Jan. 2007</TD><TD>WormBase v. WS170</TD></TR>
  		      <TR><TD><em>C. briggsae</em></TD><TD>cb3</TD><TD>Jan. 2007</TD><TD>WUSTL Cb3</TD></TR>
  		      <TR><TD><em>C. brenneri</em></TD><TD>caePb1</TD><TD>Jan. 2007</TD><TD>WUSTL 4.0</TD></TR>
  		      <TR><TD><em>C. remanei</em></TD><TD>caeRem2</TD><TD>Mar. 2007</TD><TD>WUSTL 1.0</TD></TR>
  		      <TR><TD><em>P. pacificus</em></TD><TD>priPac1</TD><TD>Feb. 2007</TD><TD>WUSTL 5.0</TD></TR>
    		    </TABLE>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/">FTP
		    server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">Downloads</A> 
		    page. Please review the 
		    <A HREF="http://genome.wustl.edu/data.cgi"
		    TARGET=_blank>data use policy</A> for the WUSTL-generated 
		    assemblies. </P>
		    <P>
		    We'd like to thank WUSTL, WormBase, and the Sanger Institute
		    for providing the sequence data for these assemblies. 
		    The UCSC worm browsers were produced by Hiram Clawson,
		    Kayla Smith, Brooke Rhead, Ann Zweig, and Donna Karolchik.
		    See the Genome Browser
		    <A HREF="credits.html">Credits</A> 
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. </P>
		    <P>
		    <HR>

<A NAME="071307"></A>
<FONT SIZE=4 COLOR="#006666">13 Jul. 2007 - <em>Drosophila melanogaster</em> Release 5 Now Available in Genome Browser</B></FONT>
		    <P>
		    The latest <em>D. melanogaster</em> assembly can now 
		    be viewed in the UCSC Genome Browser. This version -- 
		    Release 5, dated Apr. 2006 (UCSC version dm3) -- was 
		    provided by the <A HREF="http://www.fruitfly.org/" 
		    TARGET=_blank>Berkeley Drosophila Genome Project</A> (BDGP) 
		    and combines both
		    euchromatic and heterochromatic sequence. The Release 5.1 
		    annotations (Mar. 2007) were provided by 
		    <A HREF="http://flybase.net" TARGET=_blank>FlyBase</A>. 
		    <P>
		    The six euchromatic arms in this assembly were sequenced and
		    assembled by BDGP from a combination of BAC and whole genome
		    shotgun data, and have been finished to high quality.  All 
		    euchromatic sequence have been compared to the restriction 
		    digest fingerprints in multiple enzymes for validity.  The 
		    details of this analysis will be described in a forthcoming 
		    publication.  </P>
		    <P>
		    Heterochromatic sequence from the 
		    <A HREF="http://www.dhgp.org/" TARGET=_blank>Drosophila 
		    Heterochromatin Genome Project</A> (DHGP) are also available		    in this assembly. Scaffolds that could not be unambiguously 
		    mapped to a chromosome arm have been concatenated into 
		    chrUn.  chrUextra contains small scaffolds produced by the 
		    Celera shotgun assembler that could not be consistently
		    joined with larger scaffolds. Because some of the 
		    chrUextra data are of low quality, researchers are 
		    encouraged to contact either BDGP or DHGP for further 
		    details on this resource. For more information on this 
		    assembly, see the Release 5 
		    <A HREF="http://www.fruitfly.org/sequence/README.RELEASE5"
		    TARGET=_blank>assembly release notes</A>.  </P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/dm3/">FTP
		    server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#fruitfly">Downloads</A> 
		    page. 
		    We'd like to thank the BDGP, DHGP, and Flybase for 
		    providing data for this release. The dm3 Genome Browser
		    was produced by Angie Hinrichs, Archana Thakkapallayil,
		    Kayla Smith, and Donna Karolchik.
		    The <em>D. melanogaster</em> browser annotations were 
		    generated by FlyBase, DHGP, and the UCSC Genome 
		    Bioinformatics group. See the 
		    <A HREF="credits.html#dm_credits">Credits</A> 
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. </P>
		    <P>
		    <HR>

<A NAME="062607"></A>
<FONT SIZE=4 COLOR="#006666">26 Jun. 2007 - New 28-Species Conservation Track Released</B></FONT>
		    <P>
		    We are pleased to announce the release of new versions of 
		    the Conservation and Most Conserved annotation tracks for 
		    the Human March 2006 Genome Browser (hg18, NCBI Build 36). 
		    The new Conservation track displays multiple alignments of 
		    27 vertebrate species aligned to the human genome, along 
		    with measurements of evolutionary conservation across all 
		    species in the alignment and a separate measurement of 
		    conservation across the placental mammal subset of species 
		    in the alignment.</P>
		    <P>
		    The new track includes: 
		    <UL>
		    <LI>5 new high-quality assemblies -- horse, platypus, 
		    lizard, and two fish (stickleback and medaka) 
		    <LI>6 new low-coverage mammalian genomes -- bushbaby, 
		    tree shrew, guinea pig, hedgehog, common shrew, and cat  
		    <LI>6 updated assemblies -- chimp, cow, chicken, frog, 
		    fugu, and zebrafish 
		    <LI>10 assemblies included in the previous version of the 
		    track -- rhesus, mouse, rat, rabbit, dog, armadillo, 
		    elephant, tenrec, opossum, and tetraodon  
		    </UL>
		    UCSC hosts browsers for the high-quality assemblies and 
		    the cat. Masked sequence data for the other genomes in the 
		    track are available from our downloads server.</P>
		    <P>
		    In addition to the expanded species list, the new
		    Conservation track features the following improvements:
		    <UL>
		    <LI>
     		    additional filtering of pairwise alignments for each 
		    species to reduce paralogous alignments
		    <LI>
     		    information about the quality of aligning species sequence 
		    included in the multiple alignment downloads
     		    <LI>
		    new track configuration buttons to assist in selecting 
		    which species to display
		    </UL></P>
		    <P>
		    Bulk data downloads are available from the Genome Browser 
		    FTP server 
		    (<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/hg18/multiz28way">alignments</A>, 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/hg18/phastCons28way">conservation</A>) 
		    or from the 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#human">Downloads page</A>.</P>
		    <P>
		    The previous 17-vertebrate versions of the Conservation and
		    Most Conserved tracks remain available on the hg18 Genome 
		    Browser as the &quot;17-Way Cons&quot; and &quot;17-Way Most
		    Cons&quot; tracks.</P>
		    <P>
		    <HR>

<A NAME="061307"></A>
<FONT SIZE=4 COLOR="#006666">13 Jun. 2007 - ENCODE Findings Released to Public</B></FONT>
		    <P>
		    The findings of the ENCODE project have been 
		    released to the public today, the culmination of a 
		    four-year effort to catalog the biologically functional 
		    elements in 1 percent of the human genome. The publications,
		    which include a group paper in the 14 June 2007 issue of 
		    <A HREF="http://www.nature.com/nature/journal/v447/n7146/full/nature05874.html"
		    TARGET=_blank><em>Nature</em></A> and 28 companion papers in
		    the June 2007 issue of 
		    <A HREF="http://www.genome.org/content/vol17/issue6/"
		    TARGET=_blank><em>Genome Research</em></A>, were authored by
		    researchers from academic, governmental, and industry
		    organizations located in 11 countries. The <em>Nature</em> 
		    issue includes a pull-out poster featuring a screenshot of 
		    the UCSC Genome Browser displaying a broad range of the 
		    ENCODE data.</P>
		    <P>
		    In the <A HREF="http://genome.gov/25521554"
		    TARGET=_blank>press release</A> accompanying the publication
		    rollout, NHGRI Director Francis S. Collins is quoted as 
		    saying
		    &quot;This impressive effort has uncovered many exciting 
		    surprises and blazed the way for future efforts to explore 
		    the functional landscape of the entire human genome. Because
		    of the hard work and keen insights of the ENCODE consortium,
		    the scientific community will need to rethink some long-held
		    views about what genes are and what they do, as well as how 
	   	    the genome's functional elements have evolved. This could 
		    have significant implications for efforts to identify the 
		    DNA sequences involved in many human diseases.&quot;</P>
		    <P>
		    The main portal for the ENCODE data is the UCSC ENCODE
		    Genome Browser. The analysis effort has been coordinated
		    from 
		    <A HREF="http://wwww.ensembl.org/Homo_sapiens?encode.html" 
		    TARGET=_blank>Ensembl</A>. Much of the primary data have 
		    been deposited in the 
		    <A HREF="http://www.ncbi.nlm.nih.gov/projects/geo/info/ENCODE.html"
		    TARGET=_blank>NCBI GEO</A> and 
		    <A HREF="http://www.ebi.ac.uk/arrayexpress/"
		    TARGET=_blank>EBI ArrayExpress</A> databases. 
		    To access the UCSC Genome Browser ENCODE portal, click the
		    <A HREF="ENCODE/">ENCODE</A> link in the left sidebar menu 
		    on the Genome Browser home page.</P>
		    <P>
		    For more information on the ENCODE project, 
		    including the consortium's data release and accessibility 
		    policies and a list of NHGRI-funded participants, see the
		    <A HREF="http://www.genome.gov/ENCODE"
		    TARGET=_blank>NHGRI ENCODE website</A>. To read more about
		    UCSC's role on the project, see the 
		    <A HREF="http://www.cbse.ucsc.edu/news/2007/06/13/encode_pilot/index.shtml"
		    TARGET=_blank>news release</A> on the UCSC Center for 
		    Biomolecular Science and Engineering website.</P>
		    <P>
		    <HR>

<A NAME="052507"></A>
<FONT SIZE=4 COLOR="#006666">25 May 2007 - Platypus Assembly Available</B></FONT>
		    <P>
		    We have released a Genome Browser and Blat server for 
		    the Jan. 2007 v5.0.1 draft assembly of <em>Ornithorhynchus 
		    anatinus</em> (UCSC version ornAna1) produced by the 
		    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Ornithorhynchus%20anatinus"
		    TARGET=_blank>Genome Sequencing Center at Washington 
		    University</A>, St. Louis, MO (WUSTL).</P>
		    <P>
		    This assembly, which was sequenced using a combination of
		    whole genome shotgun plasmid, fosmid and BAC end sequences, 
		    has a coverage of approximately 6X. It is comprised of
		    about 1.84 Gb of actual sequence (excluding gap estimates),
		    with 437 Mb anchored and ordered on chromosomes.</P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/ornAna1/">FTP server</A> 
		    or the
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#platypus">Downloads</A>
		    page.  These data have 
		    <A HREF="credits.html#platypus_use">specific
		    conditions for use</A>.</P>
		    <P>
		    We'd like to thank WUSTL for providing this assembly. The
		    platypus Genome Browser was produced by Angie Hinrichs,
		    Kayla Smith, Robert Kuhn, Brian Raney, and Donna Karolchik.
		    The platypus browser annotation tracks were generated by 
		    UCSC and collaborators worldwide. See the
		    <A HREF="credits.html#platypus_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.  </P>
		    <P>
		    <HR>

<A NAME="051707"></A>
<FONT SIZE=4 COLOR="#006666">17 May 2007 - Lizard Assembly Available in Genome Browser</B></FONT>
                    <P>
		    A Genome Browser and Blat server are now available for
		    the Feb 2007 v1.0 draft assembly (UCSC version anoCar1) of 
		    <em>Anolis carolinensis</em> produced by the 
		    <A HREF="http://www.broad.mit.edu/" TARGET=_blank>Broad 
		    Institute</A>.
		    This assembly has been sequenced to 6.8X coverage. The draft
		    sequence contains 7,233 scaffolds comprised of nearly 1.74
		    Gb. </P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/anoCar1/">FTP server</A> 
		    or the
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#lizard">Downloads</A>
		    page.  These data have 
		    <A HREF="credits.html#lizard_use">specific
		    conditions for use</A>.
		    <P>
		    We'd like to thank the Broad Institute for providing this
		    assembly. The lizard Genome Browser was produced by Hiram
		    Clawson, Archana Thakkapallayil, Robert Kuhn, and Donna 
		    Karolchik. The lizard browser annotation tracks were 
		    generated by UCSC and collaborators worldwide. See the
		    <A HREF="credits.html#lizard_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.
		    </P>
		    <HR>

<A NAME="051607"></A>
<FONT SIZE=4 COLOR="#006666">16 May 2007 - New Dates for OpenHelix Seminars in L.A., D.C., Phila.</B></FONT>
		    <P>
		    The dates for the upcoming OpenHelix Genome Browser seminars
		    in Washington, D.C. and Los Angeles, CA have been changed.
		    The following updated announcement includes the new dates:
		    <P>
		    The UCSC Bioinformatics Group announces three regional 
		    seminars and hands-on computer workshops on the UCSC 
		    Genome Browser presented by 
		    <A HREF="http://www.openhelix.com/" 
		    TARGET=_blank>OpenHelix:</A>
		    <UL>
		    <LI>
		    Philadelphia -- Wednesday, 13 June
		    <LI>
		    Washington. D.C./Baltimore -- Wednesday, 20 June
		    <LI>
		    Los Angeles -- Wednesday, 27 June
		    </UL>
		    All sessions will be held 1 p.m. to 4 p.m.
		    </P>
		    <P>
		    These introductory sessions are geared towards anyone with a
		    basic knowledge of genomic and biological concepts who is 
		    interested in learning how to use the UCSC Genome Browser. 
		    No programming experience is required. The seminars will
		    cover the topics necessary to learn how to effectively use 
		    the browser tool set, including basic Genome Browser 
		    functionality, searching and BLAT use, Table Browser use, 
		    creating and using custom annotation tracks, and an 
		    introduction to the Gene Sorter. Lectures will be 
		    accompanied by hands-on computer exercises conducted 
		    directly on the Genome Browser web site. Participants 
		    receive a complete set of slide and exercise handouts and 
		    printed Quick Reference Cards. </P>
		    <P>
		    For further information or to make a reservation, visit the 
		    <A HREF="http://www.openhelix.com/" 
		    TARGET=_blank>OpenHelix</A> website or call 1-888-861-5051. 
		    Register early; seating is limited. Academic and student 
		    discounts are available.</P>
                    <P>
		    <HR>

<A NAME="051407"></A>
<FONT SIZE=4 COLOR="#006666">14 May 2007 - Horse Genome Browser Now Available</B></FONT>
		    <P>
		    The Jan. 2007 EquCab1 release of the horse genome 
		    (<em>Equus caballus</em>) is now available
		    in the UCSC Genome Browser. This assembly, UCSC version
		    equCab1, was produced by the 
		    <A HREF="http://www.broad.mit.edu/" TARGET=_blank>Broad 
		    Institute</A>. </P>
		    <P>
		    The horse draft genome has been sequenced to 6.8X coverage. 
		    Approximately 84% of the sequence has been anchored to 
		    chromosomes, which include autosomes 1-31 and sex 
		    chromosome X. Unanchored contigs that could not be 
		    localized to a chromosome have been concatenated into the 
		    virtual chromosome &quot;chrUn&quot;, separated by gaps of 
		    1,000 bp. The mitochondrial sequence is also available in
		    the Genome Browser as the virtual chromosome 
		    &quot;chrM&quot;. For more details about the assembly, see 
		    the Broad Institute 
		    <A HREF="http://www.broad.mit.edu/mammals/horse/"
		    TARGET=_blank>Horse Genome Project</A> page. </P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/equCab1/">FTP server</A> 
		    or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#horse">Downloads</A> 
		    page. These data have 
		    <A HREF="credits.html#horse_use">specific 
		    conditions for use</A>. </P>
		    <P>
                    The UCSC Horse Genome Browser was produced by Fan Hsu, 
		    Brooke Rhead, Robert Kuhn, Hiram Clawson, Angie Hinrichs, 
		    Kate Rosenbloom, and Donna Karolchik. See the 
		    <A HREF="credits.html#horse_credits">Credits</A> 
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. </P>
                    <P>
		    <HR>

<A NAME="042707"></A>
<FONT SIZE=4 COLOR="#006666">27 April 2007 - Medaka Genome Browser Released</B></FONT>
                    <P>
		    We're happy to announce the release of a Genome Browser and
		    Blat server for the Medaka Version 1.0 draft assembly (Apr. 
		    2006, UCSC version oryLat1). This assembly was produced in 
		    Japan by the <A HREF="http://dolphin.lab.nig.ac.jp/"
		    TARGET=_blank>National Institute of Genetics</A> (NIG) and 
		    the <A HREF="http://medaka.utgenome.org/"
		    TARGET=_blank>University of Tokyo</A>. It is
		    equivalent to Ensembl's Oct. 2005 MEDAKA1 data set.</P> 
		    <P>
		    The v1.0 assembly has been sequenced to 10.6X coverage. It 
		    consists of approximately 700.4 million bp (excluding gaps)
		    on chromosomes 1-24. 7,299 scaffolds comprised of 
		    nearly 36,500 
		    contigs of unplaced sequence are displayed on the 
		    virtual chromosome "chrUn". These contigs are spaced with a 
		    10 bp gap; scaffold gaps are 100 bp in size. The medaka 
		    mitochondrial sequence is also available in the Genome
		    Browser as the virtual chromosome "chrM".</P> 
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/oryLat1/">FTP
		    server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#medaka">Downloads</A> 
		    page. See the University of Tokyo medaka website for the 
		    <A HREF="http://medaka.utgenome.org/#policy"
		    TARGET=_blank>data release policy</A> for this assembly.
		    </P>
		    <P>
		    The Medaka browser annotation tracks were generated by 
		    UCSC and collaborators worldwide. See the Genome Browser
		    <A HREF="credits.html#medaka_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.
		    </P>
		    <HR>

<A NAME="041807"></A>
<FONT SIZE=4 COLOR="#006666">18 April 2007 - Free Genome Browser Seminar at Experimental Biology 2007</B></FONT>
		    <P>
		    OpenHelix will present a free introductory seminar on the 
		    Genome Browser during Experimental Biology 2007, 
		    April 29-May 1, in Washington, D.C. The tutorial will cover 
		    the topics needed to effectively use the Genome Browser, 
		    including: basic functionality 
		    of Genome Browser searching and BLAT use, Table Browser use,
		    creating and using Custom Tracks, and an introduction to the
		    Gene Sorter.</P>
		    <P>
		    The seminar will be held on Monday April 30, 4:30-5:30p.m. 
		    in Room 204C in the Washington, D.C. Convention Center. It 
		    is open to any interested conference attendee with a basic 
		    knowledge of genomic/biological concepts; no programming 
		    skills are needed.</P>
		    <P>
		    <P>The tutorial requires no advanced registration or fee. 
		    Attendees will receive a free download of the training 
		    materials. For more information, see the 
		    <A HREF="http://www.openhelix.com" 
		    TARGET=_blank>OpenHelix</A> 
		    website or call 1-888-861-5051.</P>
		    <P>
		    In addition to the tutorial, OpenHelix will be presenting 
		    brief introductory overviews of the UCSC Genome Browser and 
		    other bioinformatics resources during show hours at Booth 
		    330/332.  Stop by the booth for more information and to 
		    receive your free Genome Browser Quick Reference Cards.</P>
		    <HR>

<A NAME="041107"></A>
<FONT SIZE=4 COLOR="#006666">11 April 2007 - Latest Fugu Assembly Available in Genome Browser</B></FONT>
		    <P>
		    The UCSC Genome Browser now includes the latest 
		    release of the Fugu genome. The v4.0 whole genome shotgun 
		    assembly (Oct. 2004, UCSC fr2) was provided by the 
		    US DOE Joint Genome Institute (JGI) as part of the 
		    International Fugu Genome Consortium led by the JGI and the 
		    Singapore Institute of Molecular and Cell Biology (IMCB). 
		    </P>
		    <P>
		    This version has been sequenced to approximately 8.5X 
		    coverage. The assembly contains 7,213 scaffolds covering 
		    393,312,790 bp. The UCSC browser displays the scaffolds on 
		    the virtual chromosome <em>chrUn</em> with gaps of 1,000 bp 
		    between scaffolds. The scaffolds range in size from 2,223 bp
		    to 7,245,445 bp. Fifty percent of the sequence (196,648,171 
		    bp) is contained within 125 scaffolds of size 858,115 or 
		    greater (N50). The Fugu mitochondrial sequence is also 
		    available as the virtual chromosome <em>chrM</em> 
		    (GenBank accession: NC_004299.1). 
		    </P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/fr2/">FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#fugu">Downloads</A> page. These data have been 
		    freely provided by the JGI for use in the UCSC Genome 
		    Browser. 
		    </P>
		    <P>
                    Many thanks to the JGI, IMCB, and the International Fugu 
		    Genome Consortium for the assembly data.
                    The UCSC Fugu Genome Browser was produced by Cory McClean,
		    Hiram Clawson, Ann Zweig, and Donna Karolchik.
		    The annotation tracks were generated by UCSC 
		    and collaborators worldwide. See the 
		    <A HREF="credits.html#fugu_credits"
		    TARGET=_blank>Credits</A> page for a detailed list of the 
		    organizations and individuals who 
		    contributed to this release. 
		    <P>
		    <HR>

<A NAME="040607"></A>
<FONT SIZE=4 COLOR="#006666">6 April 2007 - New UCSC Gene Prediction Set Released</B></FONT>
                    <P>
		    We are pleased to announce the release of a new gene 
		    prediction set, UCSC Genes, on the latest human 
		    Genome Browser (hg18, NCBI Build 36). This annotation,
		    which includes putative non-coding genes as well as 
		    protein-coding genes and 99.9% of RefSeq genes,
		    is the next generation of the Known Genes set that UCSC 
		    has been providing for several years and supersedes the
		    existing Known Genes annotation on the hg18 assembly.
		    </P>
		    <P>
		    The UCSC Genes is a moderately conservative prediction set 
		    based on data from RefSeq, GenBank, and UniProt. Each
		    entry requires the support of one GenBank RNA sequence plus 
		    at least one additional line of evidence, with the exception
		    of RefSeq RNAs, which require no additional evidence. 
		    Some of the non-coding transcripts in the set may actually 
		    code for protein, but the evidence for the associated 
		    protein is weak at best. Compared to RefSeq, this gene set 
		    generally has about 10% more protein-coding genes, 
		    approximately five times as many putative non-coding genes, 
		    and about twice as many splice variants. 
		    </P>
		    <P>
		    A new companion track to UCSC Genes, Alt Events, shows 
		    various types of alternative splicing, alternative promoter,
		    and other events that result in more than a single 
		    transcript from the same gene. This track is based on an 
		    analysis by the txgAnalyse program of splicing graphs 
		    produced by the txGraph program.  
		    </P>
		    <P>
		    The UCSC Genes set is produced using a computational 
		    pipeline developed at UCSC by Jim Kent, Chuck Sugnet and 
		    Mark Diekhans. The programs used to construct the Alt Events
		    data set were written by Jim Kent. For detailed information 
		    about the process used to construct the genes set, see the 
		    <A HREF="../../cgi-bin/hgTrackUi?db=hg18&g=knownGene">track 
		    description page</A>. In upcoming months, we plan to 
		    release UCSC Genes sets on several organisms in addition
		    to human. The UCSC Genes annotations will be updated 
		    approximately every three months.
		    </P>
		    <P>
		    As part of this change, we are now using our own UCSC 
		    Genes accession numbers as the primary key into the 
		    underlying knownGene table, rather than the GenBank mRNA 
		    accessions 
		    we used in the previous Known Genes prediction set.
		    Note that this may affect external sites with URLs that
		    link into our genes track using the older-style accessions.
		    </P>
		    <P>
		    We will continue to provide the older Known Genes track on 
		    hg18 under the name &quot;Old Known Genes&quot;. You may
		    find the following tables useful in referencing the older 
		    gene set and converting between the two sets:
		    <UL>
		    <LI>
		    knownGeneOld2: new name for table underlying the old Known 
		    Genes (previously called knownGene)
		    <LI>
		    kgXrefOld2: new name for table that contains data for 
		    converting old Known Genes IDs to other IDs (previously 
		    called kgXref)
		    <LI>
		    kg2ToKg3: data for converting old Known Genes IDs 
		    to the newer UCSC Genes IDs
		    </UL></P>
		    <P>
		    We'd like to acknowledge the many people affiliated with 
		    the UCSC Genome
		    Bioinformatics group who worked hard to release this new 
		    annotation: developers Jim Kent,
		    Mark Diekhans, and Fan Hsu (with technical support from 
		    several
		    other engineers in the group); David Haussler; our splendid 
		    QA team -- 
		    Archana Thakkapallayil, Ann Zweig, Robert Kuhn, Kayla Smith,
		    and Brooke Rhead; our build engineer -- Andy
		    Pohl; and our sysadmin group. We'd also like to thank Chuck
		    Sugnet for his input, the people and organizations 
		    maintaining the RefSeq, UniProt, and GenBank databases, and 
		    the scientists worldwide who have contributed to them. If 
		    you have any questions about this new release,
		    feel free to contact us at 
		    <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>
		    (general questions) or 
		    <A HREF="mailto:genome-mirror@soe.ucsc.edu">genome-mirror@soe.ucsc.edu</A> 
		    (mirror-specific questions).
		    </P>
		    <P>
		    <HR>

<A NAME="032707"></A>
<FONT SIZE=4 COLOR="#006666">27 March 2007 - Stickleback Assembly Released in Genome Browser</B></FONT>
                    <P>
		    We have released a Genome Browser and Blat server for 
		    the Feb. 2006 v1.0 draft assembly of <em>Gasterosteus 
		    aculeatus</em> produced by the 
		    <A HREF="http://www.broad.mit.edu/" TARGET=_blank>Broad 
		    Institute</A>.
		    <P>
		    This assembly has been sequenced to approximately 6X 
		    coverage. An estimated  87% of the sequence has been 
		    anchored to chromosomes (chrI - chrXXI). Of the remaining 
		    unanchored scaffolds, those that could be localized to a 
		    chromosome have been concatenated into the virtual 
		    chromosome &quot;chrUn&quot; with 1000bp gaps between 
		    scaffolds. The stickleback mitochondrial sequence is also 
		    available as the virtual chromosome &quot;chrM&quot;.
		    </P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/gasAcu1/">FTP server</A> 
		    or the
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#stickleback">Downloads</A>
		    page.  These data have 
		    <A HREF="credits.html#stickleback_use">specific
		    conditions for use</A>.
		    </P>
		    <P>
		    The stickleback browser annotation tracks were generated by 
		    UCSC and collaborators worldwide. See the
		    <A HREF="credits.html#stickleback_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.
		    </P>
		    <HR>

<A NAME="021607"></A>
<FONT SIZE=4 COLOR="#006666">16 February 2007 - New Browser Session-Sharing Function Available</B></FONT>
		    <P>
                    We are pleased to announce the release of a new session 
                    management functionality 
                    in the Genome Browser, which allows users to save and
                    share browser sessions.</P>
                    <P>
                    Users are now able to configure their browsers with specific
                    track combinations, including custom tracks, and 
                    save the configuraton options.  Multiple sessions may
                    be saved for future reference, for comparison of scenarios
                    or for sharing with colleagues.  Saved sessions persist for
                    one year after the last access, unless deleted.  Custom tracks 
                    persist for at least 48 hours after
                    the last time they are viewed. </P>
                    <P>
                    The new feature may be accessed via the "Sessions" link
                    in the top blue bar in any assembly.  To ensure privacy
                    and security, users must login to the genomewiki site
                    and create a username and password.  Individual sessions
                    may be designated by the user as either "shared" or 
                    "non-shared" to protect the privacy of confidential data.
                    <P>
                    To avoid having a new shared session from someone else
                    override existing Genome Browser settings,
                    users are encouraged to open a new web-browser instance
                    or to save existing settings in a session before loading
                    a new shared session.
                    <P>
                    The Sessions feature was written by Angie Hinrichs of the 
                    UCSC Genome Bioinformatics Group and released with the 
                    assistance of Kayla Smith and Robert Kuhn.</P>
                    <P>
		    <HR>

<A NAME="020207"></A>
<FONT SIZE=4 COLOR="#006666">2 February 2007 - New Genome Graphs Tool Available
 in Genome Browser</B></FONT>
		    <P>
		    We are pleased to announce the release of a new software 
		    tool in the Genome Browser collection, the 
                    <A HREF="../cgi-bin/hgGenome">Genome Graphs</A> 
                    tool. Genome Graphs offers the ability to 
                    upload and display genome-wide data sets such as the 
                    results of genome-wide SNP association studies, linkage 
                    studies and homozygosity mapping. The Genome Graphs tool 
                    may be accessed from the menu on the UCSC Genome 
                    Bioinformatics home page.</P>
                    <P>
		    The initial release of Genome Graphs includes the 
                    following features:</P>
		    <UL>
		    <LI> upload several sets of genome-wide data and display 
                         them simultaneously</LI>
                    <LI> click on an area of interest and go directly to the 
                         genome browser at that position</LI>
                    <LI> set a significance threshold for your data and view 
                         only regions that meet that threshold</LI>
                    <LI> view the genes that exist in areas where your data 
                         meet your significance threshold</LI>
		    </UL>
		    <P>
		    For more information about the Genome Graphs tool, visit 
                    the Gateway page or consult the 
		    <A HREF="help/hgTracksHelp.html#GenomeGraphs">Getting 
                    Started on Genome Graphs</A> 
		    section in the User's Guide.</P>
                    <P>
                    Genome Graphs was written by Jim Kent of the UCSC Genome 
                    Bioinformatics Group and released with the assistance of 
                    Ann Zweig.</P>
		    <P>
		    <HR>


<A NAME="012207"></A>
<FONT SIZE=4 COLOR="#006666">22 January 2007 - Cat Assembly Available in Genome Browser</FONT><BR>
		    <P>
                    The Mar. 2006 release of <em>Felis catus</em> (UCSC version
                    felCat3) is now available in the Genome Browser.  This
                    assembly was produced by <A HREF="http://www.broad.mit.edu/"
                    TARGET=_blank>The Broad Institute</A> of MIT/Harvard and
                    <A HREF="http://www.agencourt.com/"
                    TARGET=_blank>Agencourt Bioscience</A>.</P>
                    <P>
                    The felCat3 genome has been sequenced to 2X coverage and
                    consists of 217,790 scaffolds. The total contig length for
                    this assembly is approximately 1.6 Gb spanning nearly 4.0
                    Gb (with 60.1&#37; in gaps). There are 749,376 contigs, with an
                    N50 length of 2,506 bases. There are 149,283 supercontigs,
                    with an N50 length of 49,769 bases (not including gaps).
                    The N50 size is the length such that 50&#37; of the assembled
                    genome lies in blocks of the N50 size or longer.</P>
                    <P>
                    The felCat3 sequence and annotation data can be downloaded
                    from the Genome Browser
                    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/felCat3/">
                    FTP server</A> or <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#cat">Downloads</A>
                    page. Please review the
                    <A HREF="/goldenPath/credits.html#cat_use">guidelines</A>
                    for using the cat assembly data. </P>                    <P>
                    Many thanks to The Broad Institute for providing these data.
                    The UCSC cat Genome Browser was produced by Heather Trumbower,
                    Angie Hinrichs, Mark Diekhans, Brooke Rhead, and Archana
                    Thakkapallayil. The
                    initial set of annotation tracks was generated by the UCSC
                    Genome Bioinformatics Group. See the Genome Browser
                    <A HREF="/goldenPath/credits.html#cat_credits">Credits</A>
                    page for a
                    detailed list of the organizations and individuals who
                    contributed to the release of this browser. </P>
		    <P>
		    <HR>
<A NAME="010107"></A>
<FONT SIZE=4 COLOR="#006666">1 January 2007 - Upcoming Genome Browser Seminars:  SF, Seattle, NYC, Cleveland</FONT><BR>
		    <P>
		    The UCSC Bioinformatics Group announces four regional 
		    seminars and hands-on computer workshops on the UCSC 
		    Genome Browser, presented by 
		    <A HREF="http://www.openhelix.com/" 
		    TARGET=_blank>OpenHelix:</A>
		    <UL>
		    <LI>
		    San Francisco, CA -- Wednesday, 31 January
		    <LI>
		    Seattle, WA -- Thursday, 1 February
		    <LI>
		    New York City, NY -- Tuesday, 13 February
		    <LI>
		    Cleveland, OH -- Wednesday, 14 February
		    </UL>
		    Two sessions will be offered for the New York and Cleveland
		    seminars: 9:00 a.m. to noon and 1 p.m. to 4 p.m. Only the
		    afternoon session will be offered in San Francisco and 
		    Seattle.</P>
		    <P>
		    These introductory sessions are geared towards anyone with a
		    basic knowledge of genomic and biological concepts who is 
		    interested in learning how to use the UCSC Genome Browser. 
		    No programming experience is required. The seminars will
		    cover the topics necessary to learn how to effectively use 
		    the browser tool set, including basic Genome Browser 
		    functionality, searching and BLAT use, Table Browser use, 
		    creating and using custom annotation tracks, and an 
		    introduction to the Gene Sorter. Lectures will be 
		    accompanied by hands-on computer exercises conducted directly 
		    on the Genome Browser web site. Participants receive a 
		    complete set of slide and exercise handouts and printed 
		    Quick Reference Cards. </P>
		    <P>
		    For further information or to make a reservation, visit the 
		    <A HREF="http://www.openhelix.com/seminars.shtml" 
		    TARGET=_blank>OpenHelix</A> website or call 1-888-861-5051. 
		    Register early; seating is limited. Academic and student 
		    discounts are available.</P>
		    <P>
		    <HR>

<!---------------- 2006 archived news --------------->
<A NAME="2006"></A>

<A NAME="111306"></A>
<FONT SIZE=4 COLOR="#006666">13 November 2006 - UCSC Genome Browser Wiki</FONT><BR>
                    <P>
		    The UCSC Genome Bioinformatics group has launched a wiki 
		    site for sharing information about the UCSC Genome Browser
		    and its data. The wiki -- at 
		    <A HREF="http://genomewiki.ucsc.edu"
		    TARGET=_blank>http://genomewiki.ucsc.edu</A> 
		    -- provides an informal forum for our browser users, mirror 
		    sites, and staff to discuss topics of interest in the
		    genome biology field and exchange usage tips, 
		    scripts/programs, and notes about mirroring the Genome 
		    Browser and working with the Genome Browser source.
		    <P>
		    As with most wiki pages, general users are welcome to edit
		    and add pages (login required). Please note that all content
		    created on the genomewiki site becomes a public resource; 
		    content persists in the history of a page even after it has 
		    been deleted. 
		    <P>
		    <HR>

<A NAME="102406"></A>
<FONT SIZE=4 COLOR="#006666">24 October 2006 - Free Genome Browser Training Sessions at AHA Scientific Sessions</FONT><BR>
                    <P>
		    <A HREF="http://www.openhelix.com/" 
		    TARGET=_blank>OpenHelix</A> will present a free seminar on
		    the UCSC Genome Browser at the American Heart Association's 
		    Scientific Sessions 2006 in Chicago, IL, on 12 Nov. from 
		    7:00 - 8:30 p.m. The seminar will be held in the Hyatt 
		    Regency Conference Center, Room CC21, 2233 South Martin 
		    Luther King Drive, Chicago, IL.</P>  
		    <P>
		    The introductory tutorial will cover the topics needed to 
		    effectively use the Genome Browser including: basic 
		    functionality of Genome Browser searching and BLAT use, 
		    Table Browser use, creating and using Custom Tracks, and an 
		    introduction to the Gene Sorter. The jointly-sponsored 
		    seminar will also include an introduction to VISTA 
		    comparative genomics tools.</P>
		    <P>
		    The session is open to anyone attending the AHA Scientific 
		    Sessions; no registration or fee is required. Participants 
		    should have a basic knowledge of 
		    genomic/biological concepts, but no programming skills 
		    are needed. Attendees will receive a free 
		    download of training materials, and refreshments will be 
		    served. This event is not part of the official Scientific 
		    Sessions 2006 as planned by the AHA Committee on Scientific 
		    Sessions Program.</P>
		    <P>
		    OpenHelix will also be presenting brief introductory 
		    overviews of the Genome Browser and other resources during 
		    show hours at booth 2464. Stop by the booth for more 
		    information and to receive your free Quick Reference Cards
		    for the Genome Browser and Table Browser. </P> 
		    <P>
		    <HR>

<A NAME="100606"></A>
<FONT SIZE=4 COLOR="#006666">6 October 2006 - Announcing Upgrades to the Genome Browser Custom Tracks Functionality</FONT><BR>
                    <P>
                    We have enhanced one of the popular tools in the Genome 
		    Browser collection: the <A HREF="../cgi-bin/hgCustom">custom 
		    tracks</A> utility.
                    The new custom tracks tool provides a more user-friendly 
		    interface and increased flexibility for creating and 
		    managing your custom tracks.</P>
		    <P>
                    The initial release of this upgraded tool includes the 
		    following features:
		    <UL>
                    <LI> Add and display multiple custom tracks simultaneously 
		 	 via URL, file or text
                    <LI> Add to, delete and modify the uploaded custom tracks
			 set using a new track management interface
		    <LI> Load and manage custom tracks from multiple assemblies
                    <LI> Create and upload description pages for custom tracks
                    <LI> Custom tracks will now persist on our server for 
                         48 hours after last access (rather than 8 hours)
		    </UL></P>
		    <P>
                    For more information about the new custom tracks 
		    functionality, see the Genome Browser
		    <A HREF="help/customTrack.html">Users's Guide</A>.
                    The hgCustom CGI was written by Kate Rosenbloom with the 
                    assistance of Archana Thakkapallayil, Ann Zweig and other 
                    members of the UCSC Genome Bioinformatics Group. </P>
		    <P>
		    <HR>

<A NAME="090706"></A>
<FONT SIZE=4 COLOR="#006666">7 September 2006 - Old Rhesus Assembly Archived</FONT><BR>
		    <P>
		    The Jan. 2005 rheMac1 draft assembly has been moved from
		    the main Genome Browser website to our 
		    <A HREF="http://genome-rheMac1.cse.ucsc.edu/cgi-bin/hgGateway">archive
		    server</A>. The data remain available for browsing
		    and downloading, although blat services are no longer 
		    supported.
		    <P>
		    <HR>

<A NAME="080806"></A>
<FONT SIZE=4 COLOR="#006666">8 August 2006 - New Opossum Assembly Available in Genome Browser</FONT><BR>
		    <P>
		    The UCSC Genome Browser now includes the latest draft 
		    assembly of the opossum genome. The Jan. 2006 release of 
		    <em>Monodelphis domestica</em> (UCSC version monDom4) was 
		    sequenced and assembled by 
		    <A HREF="http://www.broad.mit.edu/mammals/opossum/"
		    TARGET=_blank>The Broad Institute</A>, Cambridge, MA, USA. 
		    </P>
		    <P>
		    This draft, which has approximately 6.5X coverage, has an
		    assembly length of nearly 3.61 billion bp including gaps
		    (3.50 billion bp without gaps) contained on chromosomes 1-8,
	 	    X, and Un. The N50 of the genome
		    including gaps is 104,359 bp; the N50 without gaps is 
		    107,990. The N50 size is the length such that 50% of the 
		    assembled genome lies in blocks of the N50 size or longer. 
		    </P>
		    <P>
		    The monDom4 sequence and annotation data can be downloaded 
		    from the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/monDom4/">
		    FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/monDom4/">Downloads</A> 
		    page. Please review the 
		    <A HREF="credits.html#opossum_use">guidelines</A>
		    for using the opposum assembly data. </P>
		    <P>
		    Many thanks to The Broad Institute for providing these data.
		    The UCSC opossum Genome Browser was produced by Hiram 
		    Clawson, Archana Thakkapallayil, Ann Zweig, Kayla Smith and 
		    Donna Karolchik. The 
		    initial set of annotation tracks was generated by the UCSC 
		    Genome Bioinformatics Group. See the Genome Browser
		    <A HREF="credits.html#opossum_credits">Credits</A> 
		    page for a 
		    detailed list of the organizations and individuals who 
		    contributed to the release of this browser. </P>
		    <P>
		    <HR>

<A NAME="080106"></A>
<FONT SIZE=4 COLOR="#006666">1 August 2006 - v2.1 Chicken Assembly Available in Genome Browser</FONT><BR>
		    <P>
	   	    We have updated the Chicken Genome Browser to include the 
		    May 2006 v2.1 assembly (UCSC version galGal3) produced by 
	 	    the Genome Sequencing Center at the Washington University 
		    School of Medicine in St. Louis, MO, USA (WUSTL). The 
		    source of this sequence was a female inbred Red Jungle Fowl 
		    (<em>Gallus gallus</em>), the ancestor of domestic chickens.
		    The chicken genome is the first of the avian genomes to be 
		    sequenced.
		    <P>
		    In this assembly, 198,000 additional reads covering all 
		    contig ends and regions of low quality have been added to 
		    the original assembly's 6.6X coverage. Approximately 95% of 
		    the sequence has been anchored to chromosomes, which include
	 	    autosomes 1-24, 26-28, and 32, and sex chromosomes W and Z. 
		    (In contrast to mammals, the female chicken is heterogametic
		    (ZW) and the male is homogametic (ZZ).) The remaining 
		    unanchored contigs that could be localized to a chromosome 
		    have been concatenated into the virtual chromosomes 
		    &quot;chr*_random&quot;, separated by gaps of 10,000 bp. 
		    Unanchored contigs that could not be localized to a 
		    chromosome have been concatenated into the virtual 
		    chromosome &quot;chrUn_random&quot;, separated by gaps of 
		    100 bp to reduce the total size of chrUn_random. The chicken
		    mitochondrial sequence is also available as the virtual 
		    chromosome &quot;chrM&quot;. </P>
		    <P>
		    Although centromere positions are indicated on this
		    assembly, little is known of their exact sequence. 
		    The centromeres of 18 chromosomes were tentatively localized
		    based on FISH hybridization using BAC clones, genetic 
		    markers flanking the centromeres in coordination with 
		    mapping gaps in the physical map, repetitive sequence 
		    content, and analysis of proximity to the constrictions of 
		    the mitotic metaphase chromosomes. For more information on 
		    the process used to create the chromosomal sequences and 
		    assign centromere locations, see the WUSTL Gallus gallus 
		    <A HREF="http://genome.wustl.edu/pub/organism/Other_Vertebrates/Gallus_gallus/assembly/draft/Gallus_gallus-2.1/ASSEMBLY"
		    TARGET=_blank>assembly release notes</A>.</P>
		    <P>
		    Bulk downloads of the chicken sequence and annotations may 
		    be obtained from the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/galGal3/">FTP 
		    server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#chicken">Downloads</A> 
		    page. These data have
		    <A HREF="credits.html#chicken_use">specific 
		    conditions for use</A>.
		    <P>
                    We'd like to thank WUSTL, who provided the sequence, 
		    physical map, assembly, and assembly/map for this
		    release. The genetic mapping and linkage analysis were 
		    produced through a collaborative effort of labs in The
		    Chicken Mapping Consortium.  The chicken browser annotation 
		    tracks were generated by UCSC and collaborators worldwide. 
		    See the 
		    <A HREF="credits.html#chicken_credits">Credits</A> 
		    page for a detailed list of acknowledgements. The UCSC 
		    Chicken Genome Browser was produced by Angie Hinrichs, 
		    Kayla Smith, and Donna Karolchik.</P>
		    <P></P>
		    <HR>

<A NAME="072006"></A>
<FONT SIZE=4 COLOR="#006666">20 July 2006 - New Chimpanzee Genome Browser Released</FONT><BR>
		    <P>
		    We are happy to announce the release of a Genome Browser for
		    the latest release of the chimpanzee (Pan troglodytes) 
		    genome. The Mar. 2006 assembly -- labeled Chimp Build 2 
		    Version 1 (UCSC version panTro2) -- was produced by the 
		    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Pan%20troglodytes&SECTION=collaborators"
		    TARGET=_BLANK>Chimpanzee Sequencing and Analysis 
		    Consortium</A>.</P>
		    <P>
		    This 6X whole genome assembly contains sequence from the 
		    initial 4X chimpanzee assembly described and 
		    analyzed in <em>Nature</em> 
		    (<A HREF="http://www.nature.com/nature/journal/v437/n7055/full/nature04072.html" 
		    TARGET=_blank>The Chimpanzee Sequencing and Analysis 
		    Consortium, 2005</A>), with additional 2X 
		    sequence generated, assembled, and assigned to chromosomes 
		    by the 
		    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Pan%20troglodytes"
		    TARGET=_BLANK>Genome Sequencing Center of Washington 
		    University School of Medicine</A>, St. Louis, MO, USA.
		    <P> 
		    The whole genome shotgun data were derived primarily from 
		    the donor Clint, a captive-born male chimpanzee from the 
		    Yerkes Primate Research Center in Atlanta, GA, USA. The 
		    reads were assembled using the whole-genome assembly
		    program PCAP. For information about the assembly process,
		    see the panTro2 
		    <A HREF="../cgi-bin/hgGateway?db=panTro2">Gateway</A> page.
		    </P>
		    <P>
		    This assembly covers about 97 percent of the genome and is 
		    based on 6X sequence coverage. It is composed of 265,882 
		    contigs with an N50 length of 29 kb and 44,460 supercontigs 
		    with an N50 length of 9.7 Mb. The total contig length, not 
		    including estimated gap sizes, is 2.97 Gb. Of that total, 
		    2.82 Gb of sequence have been ordered and oriented along 
		    specific chimpanzee chromosomes, 107 Mb have been placed in 
		    chr*_random, and 50 Mb remain in chrUn. </P>
		    <P>
		    A major difference between this assembly and the previous 
		    Nov. 2003 version is the chromosomal numbering scheme, 
		    which  has been changed to reflect a new standard that 
		    preserves orthology with human chromomes. Proposed by 
		    <A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=pubmed&cmd=Retrieve&dopt=Abstract&list_uids=15218271&query_hl=1&itool=pubmed_docsum"
		    TARGET=_blank>E.H. McConkey</A> in 2004, the new numbering 
		    convention was subsequently endorsed by the International 
		    Chimpanzee Sequencing and Analysis Consortium. This 
		    standard assigns the 
		    identifiers &quot;2a&quot; and &quot;2b&quot; to the
		    two chimp chromosomes that fused in the human genome to form
		    chromosome 2. Note that the genome assembly shown in the 
		    Nov. 2003 panTro1 Genome Browser retains the older 
		    numbering scheme, in which these chromosomes are numbered
		    12 and 13.</P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/panTro2/">FTP
		    server</A> or the 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#chimp">Downloads</A> 
		    page. The complete set of sequence reads is available at the

		    <A HREF="http://www.ncbi.nlm.nih.gov/Traces" 
		    TARGET=_blank>NCBI trace archive</A>. These data have 
		    <A HREF="credits.html#chimp_use">specific
		    conditions for use</A>.</P>
		    <P>
		    We'd like to thank the International Chimpanzee 
		    Sequencing and Analysis Consortium, Washington 
		    University at St. Louis School of Medicine Genome Sequencing
		    Center, and the Broad Institute for providing this sequence.
		    We'd also like to acknowledge the UCSC team who worked on 
		    this release: Kate Rosenbloom, Brian Raney, Hiram Clawson, 
		    Ann Zweig, Archana Thakkapallayil, and Donna Karolchik. 
		    The chimpanzee browser annotation tracks were generated by 
		    UCSC and collaborators worldwide. </P>
		    <P>
		    <HR>

<A NAME="062006"></A>
<FONT SIZE=4 COLOR="#006666">20 June 2006 - Genome Browser Released for Baylor v3.4 Rat Assembly</FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics group has released a Genome
		    Browser for the v3.4 rat (<em>Rattus 
		    norvegicus</em>) genome. This assembly--UCSC version rn4,
		    November 2004--was produced by the Atlas group at 
		    <A HREF="http://hgsc.bcm.tmc.edu/" TARGET=_blank>Baylor 
		    Human Genome Sequencing Center</A> (HGSC) as part of the 
		    Rat Genome Sequencing Consortium.</P>
		    <P>
		    The sequence was assembled using a hybrid approach that 
		    combines the clone-by-clone and whole genome shotgun 
		    methods.  The assembly is a minor update to version 3.3 that
		    spliced in 54.6 Mb finished BAC sequences; the overall 
		    statistics are unchanged from releases 3.0 to 3.4.</P>
		    <P>
		    The 3.x assemblies reflect several sequence additions and 
		    software improvements over the previous 2.x assemblies, 
		    including the sequencing of over 1100 new BACs to cover 
		    gaps, an improved marker set from the Medical College of 
		    Wisconsin, a new FPC map from the BC Cancer Agency Genome 
		    Sciences Centre, and improved linking of bactigs. For 
		    detailed information and statistics about the 3.x 
		    assemblies, see the Baylor HGSC 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/rat/" 
		    TARGET=_blank>Rat Genome Project</A> web page. </P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/rn4/">FTP 
		    server</A> or the 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#rat">Downloads</A> 
		    page. These data are made available with 	
		    <A HREF="/goldenPath/credits.html#rat_use">specific 
		    conditions for use</A>. 
		    <P>
		    We'd like to thank the Rat Genome Sequencing Consortium
		    and the Baylor HGSC for providing this assembly. We'd also
		    like to acknowledge the UCSC team who produced the rn4
		    Genome Browser: Angie Hinrichs, Fan Hsu, Brooke Rhead, 
		    Archana Thakkapallayil, Kayla Smith, Ann Zweig, Robert Kuhn,
		    and Donna Karolchik. The rn4 annotation tracks were 
		    generated by UCSC and collaborators worldwide. See the 
		    <A HREF="credits.html#rat_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. </P>
		    <P>
		    <HR>

<A NAME="061206"></A>
<FONT SIZE=4 COLOR="#006666">12 June 2006 - Latest <em>X. tropicalis</em> Assembly Available in Browser</FONT><BR>
		    <P>
		    The v.4.1 <em>Xenopus tropicalis</em> assembly is now
		    available on the UCSC Genome Browser. This whole genome
		    shotgun assembly 
		    (xenTro2, August 2005) was generated by the U.S. DOE 
		    Joint Genome Institute (JGI) using the Jazz assembler.
		    It contains 19,501 scaffolds with an average coverage of 
		    7.65X. Roughly half the genome is contained in 272 
		    scaffolds, each of at least 1.56 Mb in length.</P> 
		    <P>
		    In this release, some scaffolds showing homology to a known
	   	    prokaryotic contaminant as well as non-cellular or vector 
		    contamination have been removed by the JGI and placed in a 
		    separate directory. The <em>X.  tropicalis</em> assembly 
		    will be improved over the coming year by additional 
		    sequencing of large insert clones, targeted gap closure, 
		    and the incorporation of physical and genetic mapping 
		    information as it becomes available.</P>
		    <P>
		    For more information about the 4.1 assembly, see the JGI
		    <A HREF="http://genome.jgi-psf.org/Xentr4/Xentr4.home.html"
		    TARGET=_blank><em>X. tropicalis</em> website</A>.</P>
		    <P>
		    The xenTro2 sequence and annotation data can be downloaded 
		    from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/xenTro2/">FTP server</A>
		    or
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#xenTro">downloads</A>
		    page. These data have
		    <A HREF="http://genome.jgi-psf.org/Xentr4/Xentr4.download.html"
		    TARGET=_blank>specific conditions for use</A>.</P>
		    <P>
		    Many thanks to the JGI and the other
		    institutions who contributed to the sequencing and mapping 
		    effort for this release. The xenTro2 Genome Browser was
		    produced by Angie Hinrichs, Kayla Smith, Robert Kuhn, and
		    Donna Karolchik. The xenTro2 annotation tracks were 
		    generated by UCSC and collaborators worldwide. See the
		    <A HREF="credits.html#xenTro_credits">credits</A> page
		    for a detailed list of the organizations and individuals who
		    contributed to this release. </P>
		    <P>
		    <HR>
<A NAME="052506"></A>
<FONT SIZE=4 COLOR="#006666">25 May 2006 - Zv6 Zebrafish Browser Released</FONT><BR>
		    <P>
		    The latest zebrafish assembly -- Zv6 (UCSC version 
		    danRer4, March 2006) -- is now available in the UCSC
		    Genome Browser. The Zv6 assembly was produced by 
		    The Wellcome Trust Sanger Institute in collaboration with 
		    the Max Planck Institute for Developmental Biology in 
		    Tuebingen, Germany, and the Netherlands Institute for 
		    Developmental Biology (Hubrecht Laboratory), Utrecht, 
		    The Netherlands.</P>
		    <P>
		    This assembly consists of 1,626,077,335 bp in 6.653
		    scaffolds (N50 = 1,247,221 bp) with a sequence coverage of 
		    approximately 6.5-7x. The sequence has been anchored to
		    chromosomes 1-25, chrM (mitochondrial), chrNA_random, and
		    chrUn_random.
		    For more information about this assembly, see the Sanger 
		    Institute web page for the 
		    <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/Zv6_assembly_information.shtml/"
		    TARGET=_blank><em>Danio rerio</em> Sequencing Project</A>.
		    <P>
		    The danRer4 sequence and annotation data can be 
		    downloaded from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/danRer4/">FTP server</A>
		    or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#zebrafish">downloads</A>
		    page. Please review the 
		    <A HREF="http://www.sanger.ac.uk/Projects/use-policy.shtml"
		    TARGET=_blank>guidelines</A> for using these data.
		    <P>
		    We'd like to thank the Wellcome Trust Sanger Institute, 
		    the Max Planck Institute for Developmental Biology,
		    Hubrecht Laboratory and the other
		    institutions who contributed to the sequencing and 
		    mapping effort of this release. Special thanks to the 
		    Zebrafish Genome Initiative at Children's Hospital in 
		    Boston for their collaboration on this release. 
		    The UCSC zebrafish Genome Browser
		    was produced by Rachel Harte, Archana Thakkapallayil, 
		    Robert Kuhn, Ann Zweig, and Donna Karolchik. The 
		    initial set of annotation tracks was generated by the 
		    UCSC Genome Bioinformatics Group.  See the
		    <A HREF="credits.html#zebrafish_credits">credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to the release of
		    this browser.</P>
		    <P>
		    <HR>

<A NAME="051006"></A>
<FONT SIZE=4 COLOR="#006666">10 May 2006 - NCBI Mouse Build 36 Released in Genome Browser</FONT><BR>
		    <P>
		    The latest mouse genome assembly from the Mouse Genome
		    Sequencing Consortium, NCBI Build 36, is now available in 
		    the UCSC Genome Browser. This version (UCSC version mm8) is
		    considered to be essentially finished.
		    <P>
		    The Build 36 assembly includes approximately 2.6 Gb of
		    sequence on chromosomes 1-19, X, Y, M (mitochondrial DNA) 
		    and Un (unmapped clone contigs). 
		    For in-depth
		    information about the process used to assemble this version,
		    see the 
		    <A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/mouse/contig/Build36.html"
                    TARGET=_blank>NCBI website</A>.
		    On chromosome Y in this assembly, only the short arm has 
		    reliable mapping data; therefore, most of the contigs on 
		    the Y chromosome are unplaced.
		    Note that the UCSC mm8 database contains only the 
		    reference strain C57BL/6J.</P>
                    <P>
                    The mm8 sequence and annotation data may be downloaded 
		    from the Genome Browser
                    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/mm8/">FTP
                    server</A> or
                    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/mm8/">Downloads</A>
                    web page. The mm8 annotation tracks were generated by UCSC
                    and collaborators worldwide. </P>
		    <P>
                    We'd like to thank Deanna Church and the Mouse Genome 
		    Sequencing Consortium for this assembly. We'd also like to
                    acknowledge the work of the UCSC mm8 team: Hiram Clawson, 
		    Fan Hsu, Kayla Smith, Ann Zweig, Robert Kuhn, Brooke
		    Rhead, Archana Thakkapallayil, and Donna Karolchik.
		    For a complete list
		    of the individuals and organizations who participated 
		    in this assembly, see the 
		    <A HREF="credits#mouse_credits">Credits</A> 
		    page. </P>
		    <P>
		    <HR>

<A NAME="041406"></A>
<FONT SIZE=4 COLOR="#006666">14 April 2006 - NCBI Build 36.1 Human Reference Sequence Available in Browser</FONT><BR>
		    <P>
		    The latest human genome reference sequence assembly (NCBI 
		    Build 36.1, March 2006) is now available as database hg18 in 
		    the UCSC Genome Browser. This sequence, which was obtained
		    from NCBI,  was produced by the 
		    International Human Genome Sequencing Consortium. </P>
		    <P>
		    The hg18 assembly contains four alternate haplotype regions:
		    <UL>
		    <LI>
		    chr22_h2_hap1 -- an alternate chromosome 22 assembly that 
		    contains the CYP2D6 gene (NT_113959.1). CYP2D6 is deleted 
		    in the reference assembly.
		    <LI>
		    chr5_h2_hap1 -- a chromosome 5 alternate assembly of the 
		    SMN1 gene region (NT_113801.1, NT_113802.1).
		    <LI>
		    chr6_cox_hap1 -- an A1-B8-DR3 alternate haplotype assembly 
		    of the chromosome 6 MHC region based on sequence data from 
		    the COX library (NT_113891.1). 
		    <LI>
		    chr6_qbl_hap2 -- an A26-B18-DR3 alternate haplotype assembly
		    of the chromosome 6 MHC region based on sequence data from 
		    the QBL library (NT_113892.1, NT_113893.1, NT_113894.1, 
		    NT_113895.1, NT_113896.1, NT_113897.1). 
		    </UL></P>
		    <P>
		    See the 
		    <A HREF="http://www.sanger.ac.uk/HGP/Chr6/MHC/"
		    TARGET=_blank>Wellcome Trust Sanger Institute MHC Haplotype 
		    Project</A> web site for additional information on the 
		    chr6 alternate haplotype assemblies.</P>
		    <P>
		    The Y chromosome in this assembly contains two 
		    pseudoautosomal regions (PARs) at chrY:1-2709520 and 
		    chrY:57443438-57772954. These sequences were taken from
		    the corresponding regions in the X chromosome and are exact 
		    duplications of the X chromosome sequences.</P>
		    <P>
		    For further information on NCBI Build 36.1, see the NCBI 
		    <A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/human/release_notes.html"
		    TARGET=_blank>Build 36.1 release notes</A>.</P>
		    <P>
		    Bulk downloads of the data are available from the UCSC
		    downloads server via 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/hg18"
		    TARGET=_blank>ftp</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#human"
		    TARGET=_blank>http</A>. We recommend that you 
		    use ftp or rsync for downloading large or multiple files. 
		    </P>
		    <P>
		    We'd like to thank NCBI and the International Human Genome 
		    Sequencing Consortium for furnishing the data, and the 
		    entire UCSC Genome Browser staff for contributing to this
		    release. Fan Hsu led the UCSC engineering effort; QA was 
		    headed up by Ann Zweig.</P>
		    <P>
		    <HR>

<A NAME="030606"></A>
<FONT SIZE=4 COLOR="#006666">6 March 2006 - Purple Sea Urchin Genome Assembly Available in Genome Browser</FONT><BR>
		    <P>
		    The April 2005 release of the Purple Sea Urchin genome 
		    (<em>Strongylocentrotus purpuratus</em>) is now available
		    in the UCSC Genome Browser. This assembly, UCSC version
		    strPur1, was produced by the Baylor College of Medicine 
		    Human Genome Sequencing Center (BCM HGSC) and corresponds 
		    to their Spur_0.5 whole genome shotgun assembly. </P>
		    <P>
		    This release was assembled from whole genome shotgun reads 
		    using the Atlas genome assembly system at the BCM HGSC. 
		    Several whole genome shotgun libraries, with inserts of 2-6 
		    kb, were used to produce the data. About 7 million reads 
		    were assembled, representing about 800 Mb of sequence and 
		    about 6x coverage of the (clonable) sea urchin genome. 
		    Highly repeated sequences were assembled separately into 
		    reptigs and merged into the genome assembly. Sequences from 
		    BAC clones were omitted from this assembly and will be 
		    placed in a subsequent version of the draft sequence. </P>
		    <P>
		    This is a draft sequence and may contain errors; therefore, 
		    users should exercise caution. Typical errors in draft 
		    genome sequences include misassemblies of repeated 
		    sequences, collapses of repeated regions, and unmerged 
		    overlaps (<em>e.g.</em> due to polymorphisms) creating 
		    artificial duplications. However, base accuracy in contigs 
		    (contiguous blocks of sequence) is usually very high with 
		    most errors near the ends of contigs.</P>
		    <P>
		    More assembly details can be found in the 
		    <A HREF="ftp://ftp.hgsc.bcm.tmc.edu/pub/data/Spurpuratus/fasta/Spur20050415/READMEApril2005.txt" 
		    TARGET=_blank>Spur_0.5 README file</A> and on the BCM HGSC 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/seaurchin/"
		    TARGET=_blank>Sea Urchin Genome Project</A> web page.</P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/strPur1">FTP server</A>
		    or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#urchin">Downloads</A> 
		    page. These data have 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
		    TARGET=_blank>specific conditions for use</A>. The strPur1 
		    annotation tracks were generated by UCSC and collaborators 
		    worldwide. See the 
		    <A HREF="credits.html#urchin_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. </P>
		    <P>
		    <HR>

<A NAME="022706"></A>
<FONT SIZE=4 COLOR="#006666">27 February 2006 - Upcoming Genome Browser Seminars in Texas, Florida and Washington, DC</FONT><BR>
		    <P>
		    The UCSC Bioinformatics Group announces four regional 
		    seminars and hands-on computer workshops on the UCSC 
		    Genome Browser, presented by 
		    <A HREF="http://www.openhelix.com/" 
		    TARGET=_blank>OpenHelix:</A>
		    <UL>
		    <LI>
		    Houston, TX -- Tuesday, 14 March
		    <LI>
		    Austin, TX -- Wednesday, 15 March
		    <LI>
		    Washington, DC -- Tuesday, 18 April
		    <LI>
		    Miami, FL -- Wednesday, 19 April
		    </UL>
		    Two sessions will be offered each day: 9:00 a.m. to noon and
		    1 p.m. to 4 p.m. </P>
		    <P>
		    These introductory sessions are geared towards anyone with a
		    basic knowledge of genomic and biological concepts who is 
		    interested in learning how to use the UCSC Genome Browser. 
		    No programming experience is required. The seminars will
		    cover the topics necessary to learn how to effectively use 
		    the browser tool set, including basic Genome Browser 
		    functionality, searching and BLAT use, Table Browser use, 
		    creating and using custom annotation tracks, and an 
		    introduction to the Gene Sorter. Lectures will be 
		    accompanied by hands-on computer exercises conducted directly 
		    on the Genome Browser web site. Participants receive a 
		    complete set of slide and exercise handouts and printed 
		    Quick Reference Cards. </P>
		    <P>
		    For further information or to make a reservation, visit the 
		    <A HREF="http://www.openhelix.com/noteworthy.shtml" 
		    TARGET=_blank>OpenHelix</A> website or call 1-888-861-5051. 
		    Register early; seating is limited. Academic and student 
		    discounts are available.</P>
		    <P>
		    <HR>

<A NAME="021706"></A>
<FONT SIZE=4 COLOR="#006666">17 February 2006 - Zebrafish Browser Updated</FONT><BR>
		    <P>
		    The latest zebrafish assembly -- Zv5 (UCSC version 
		    danRer3, May 2005) -- is now available in the UCSC
		    Genome Browser and Blat server. The Zv5 assembly was 
		    produced by 
		    The Wellcome Trust Sanger Institute in collaboration with 
		    the Max Planck Institute for Developmental Biology in 
		    Tuebingen, Germany, and the Netherlands Institute for 
		    Developmental Biology (Hubrecht Laboratory), Utrecht, 
		    The Netherlands.</P>
		    <P>
		    This assembly consists of 1,630,306,866 bp in 16,214 
		    scaffolds (N50 = 1,116,981 bp) with a sequence coverage of 
		    approximately 6.5-7x. The assembly has been tied to the 
		    fingerprint contig map (data freeze 15th February, 2005) 
		    and contains 699 Mb from 4,519 sequenced clones. </P>
		    For more information about this assembly, see the Sanger 
		    Institute web page for the 
		    <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/"
		    TARGET=_blank><em>Danio rerio</em> Sequencing Project</A>.
		    <P>
		    The danRer3 sequence and annotation data can be 
		    downloaded from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/danRer3/">FTP server</A>
		    or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#zebrafish">downloads</A>
		    page. Please review the 
		    <A HREF="http://www.sanger.ac.uk/Projects/use-policy.shtml"
		    TARGET=_blank>guidelines</A> for using these data.
		    <P>
		    We'd like to thank the Wellcome Trust Sanger Institute, 
		    the Max Planck Institute for Developmental Biology,
		    Hubrecht Laboratory and the other
		    institutions who contributed to the sequencing and 
		    mapping effort of this release. Special thanks to the 
		    Zebrafish Genome Initiative at Children's Hospital in 
		    Boston for their collaboration on this release. 
		    The UCSC zebrafish Genome Browser
		    was produced by Rachel Harte, Jennifer Jackson, Ann Zweig, 
		    Ali Sultan-Qurraie, and Donna Karolchik. The 
		    initial set of annotation tracks was generated by the 
		    UCSC Genome Bioinformatics Group.  See the
		    <A HREF="credits.html#zebrafish_credits">credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to the release of
		    this browser.</P>
		    <P>
		    <HR>

<A NAME="020906"></A>
<FONT SIZE=4 COLOR="#006666">9 February 2006 - New Rhesus Macaque Browser Released</FONT><BR>
		    <P>
		    The latest rhesus macaque (<em>Macaca mulatta</em>) draft
		    assembly&mdash;v.1.0, Mmul_051212&mdash;is now available 
		    in the UCSC Genome Browser. This version (UCSC rheMac2)
		    was sequenced and assembled by the Macaque Genome Sequencing
		    Consortium led by the Baylor College of Medicine Human 
		    Genome Sequencing Center, in collaboration with the
		    Genome Sequencing Center at Washington University School of 
		    Medicine in St. Louis and the J. Craig Venter Institute 
		    Joint Technology Center. </P> 
		    <P>
		    The rhesus macaque follows the human and chimpanzee as the
		    third primate and first Old World monkey to have its genome
		    sequenced. Overall, the rhesus genome shares approximately
		    92 to 95 precent of its sequence with the human, compared
		    with the chimp at 98 precent. Because of its genetic,
		    physiologic, and metabolic similarities to the human and
		    chimp, the rhesus is an ideal reference point for 
		    comparisons among the three primates.</P>
		    <P>
		    The groups in the Macaque Genome Sequencing Consortium 
		    produced preliminary assemblies of the genome data 
		    using different and complementary approaches. The resulting 
		    data were combined into a single, high-density 	
		    &quot;melded&quot; assembly by a team at J. Craig Venter 
		    Institute. This collaboration made use of published rhesus 
		    maps, the BAC fingerprint map from the Michael Smith Genome 
		    Sciences Centre, and the human reference genome sequence. 
		    The v.1.0 assembly covers about 93 percent of the rhesus 
		    genome. For more information about the rheMac2 assembly, 
		    see the Baylor 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/rmacaque/"
		    TARGET=_blank>Rhesus Monkey Project</A> web page.</P>
		    <P>
		    The rheMac2 sequence and annotation data can be downloaded
		    from the UCSC Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/rheMac2/">FTP
		    server</A> or the
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#rhesus">Downloads</A>
		    page. These data have specific
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
		    TARGET=_blank>conditions for use</A>.</P> 
		    <P>
		    Many thanks to Baylor College of Medicine, the Macaque
		    Genome Sequencing Consortium, and the other 
		    institutions who contributed to the sequencing and mapping
		    effort of the v.1.0 release. The UCSC Rhesus Genome 
		    Browser was produced by Robert Baertsch, Kayla Smith,
		    Ann Zweig, Robert Kuhn, and Donna Karolchik. The initial 
		    set of rheMac2 annotation
	            tracks was generated by the <A HREF="../staff.html">UCSC 
		    Genome Bioinformatics Group</A>. See the 
		    <A HREF="credits.html#rhesus_credits">credits</A> 
		    page for a detailed list of the organizations and 
		    individuals who contributed to the release of this 
		    browser. </P>
		    <P>
		    <HR>

<A NAME="011806"></A>
<FONT SIZE=4 COLOR="#006666">18 January 2006 - <em>D. sechellia</em> Browser Released</FONT><BR>
		    <P>
		    The October 2005 <em>D. sechellia</em> assembly (UCSC 
		    version droSec1) is now available in the UCSC Genome 
		    Browser. This version was sequenced and assembled by the 
		    Broad Institute of MIT and Harvard. 
		    <P>
	  	    Downloads of the droSec1 data and annotations can be 
		    obtained from the UCSC Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/droSec1/">FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#droSec">Downloads</A> page.
		    The initial set of droSec1 annotation tracks were generated 
		    by UCSC. 
		    <P>
		    Many thanks to the Broad Institute for providing the 
		    sequence and assembly of this genome.  The UCSC <em>D. 
		    sechellia</em> Genome Browser was produced by Angie 
		    Hinrichs, Kayla Smith and Donna Karolchik. See 
		    the
		    <A HREF="credits.html#droSec_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.

<A NAME="011106"></A>
<FONT SIZE=4 COLOR="#006666">11 January 2006 - <em>D. yakuba</em> Browser Update</FONT><BR>
		    <P>
		    The latest <em>D. yakuba</em> assembly is now available in
		    the UCSC Genome Browser. This version &mdash;  Release 2.0, 
		    dated Nov. 2005 (UCSC version droYak2) &mdash; was sequenced
		    and assembled by the Genome Sequencing Center, Washington 
		    University (WUSTL) School of Medicine in St. Louis. </P>
		    <P>
		    The whole genome shotgun (WGS) assembly includes both raw 
		    shotgun data and data from two rounds of 
		    automated, directed read selection, which has improved the
		    sequence quality and narrowed or (in some instances) closed 
		    gaps.  For more assembly information and statistics,
		    see the WUSTL Genome Sequencing Center
		    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Drosophila%20yakuba&GROUP=6"
		    TARGET=_blank><em>Drosophila yakuba</em></A> web page.
		    <P>
	  	    Downloads of the droYak2 data and annotations can be 
		    obtained from the UCSC Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/droYak2/">FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#droYak">Downloads</A> page.
		    The initial set of droYak2 annotation tracks was generated
		    by UCSC. </P>
		    <P>
		    Thanks to the Genome Sequencing Center at WUSTL School of 
		    Medicine for providing the sequence and assembly of this 
		    genome.  The droYak2 Genome Browser was produced by Angie 
		    Hinrichs, Jennifer Jackson and Donna Karolchik. See the
		    <A HREF="credits.html#droYak_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.</P>
		    <P>
		    <HR>

<A NAME="010406"></A>
<FONT SIZE=4 COLOR="#006666">4 January 2006 - <em>D. persimilis</em> Genome Browser Released</FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics Group has released a Genome
		    Browser and Blat server for the <em>D. persimilis</em>
		    genome. The Oct. 2005 release 
		    (UCSC version droPer1) was sequenced and assembled by the 
		    Broad Institute of MIT and Harvard.
		    The <em>D. persimilis</em> assembly joins nine other
		    Drosophila species featured in the UCSC Genome Browser.
		    <P>
	  	    Downloads of the droPer1 data and annotations can be obtained
		    from the UCSC Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/droPer1/">FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#droPer">Downloads</A> page.
		    The droPer1 annotation tracks were generated by UCSC and
		    collaborators worldwide. 
		    <P>Thanks to the Broad Institute for providing the sequence 
		    and assembly of this genome.  The UCSC <em>D. 
		    persimilis</em> Genome Browser was produced by Angie 
		    Hinrichs, Kayla Smith, Robert Kuhn, Jennifer Jackson and 
		    Donna Karolchik. See the
		    <A HREF="credits.html#droPer_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.
		    <P>
		    <HR>

<!---------------- 2005 archived news --------------->
<A NAME="2005"></A>

<A NAME="122005"></A>
<FONT SIZE=4 COLOR="#006666">20 December 2005 - Dog Genome Browser Update Released</FONT><BR>
		    <P>
		    UCSC has updated the dog Genome Browser to include the 
		    May 2005 v2.0 assembly (UCSC version canFam2) sequenced and
		    assembled by the Broad Institute of MIT and Harvard and 
		    Agencourt Bioscience. 
		    The whole genome shotgun sequence is 
		    based on 7.6X coverage of the dog genome which includes
		    more than 98% of the euchromatic genome. 
		    <P>
		    The dog genome, which contains approximately 2.5 billion 
		    base pairs, is similar in size to the genomes of humans and
		    other mammals. The boxer breed was selected for the initial 
		    sequencing effort, based on the lower variation rate in its 
		    genome relative to other breeds. In addition to the boxer,
		    samples from several other dog breeds were used to generate
		    a set of single nucleotide polymorphisms (SNPs) to 
		    facilitate disease studies.
		    The SNPs are available from the Broad Institute
		    <A HREF="http://www.broad.mit.edu/ftp/pub/papers/dog_genome/snps_canfam2/"
		    TARGET =_blank>dog SNP</A> web page.
		    For more information about the dog draft assembly, see the 
		    Broad Institute 
		    <A HREF="http://www.broad.mit.edu/mammals/dog/"
		    TARGET=_blank>Dog Genome Sequencing Project</A> web page.
		    <P>
		    The dog sequence and annotation data can be downloaded from
		    the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/canFam2/">FTP server</A>
		    or <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#dog">downloads</A>
		    page. These data have 
		    <A HREF="credits.html#dog_use">specific 
		    conditions for use</A>.
		    <P>
		    Many thanks to the Broad Institute of MIT and Harvard, 
		    Agencourt Bioscience, and the other institutions who 
		    contributed to the sequencing, assembly, and mapping
		    efforts.  The canFam2 Genome Browser team included
		    Angie Hinrichs, Jennifer Jackson, and Donna Karolchik.
		    See the 
		    <A HREF="credits.html#dog_credits">credits</A> 
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release.
		    <P>
		    <HR>

<A NAME="121505"></A>
<FONT SIZE=4 COLOR="#006666">15 December 2005 - New Mouse Assembly Available in Genome Browser</FONT><BR>
		    <P>
		    The latest mouse genome assembly from the Mouse Genome
		    Sequencing Consortium, NCBI Build 35 (UCSC version mm7),
		    is now available in the UCSC Genome Browser. 
		    <P>
		    The Build 35 assembly includes approximately 2.6 Gb of
		    sequence, of which about 2.2 Gb is finished 
		    sequence. Chromosomes 2, 
		    4, 11 and X are finished in this build. To review in-depth
		    statistics on the assembly, see the NCBI
                    <A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/mouse/contig/Build35.html"
                    TARGET=_blank>Build 35 Data</A> web page.
		    Please note that the UCSC mm7 database contains only the 
		    reference strain C57BL/6J.</P>
                    <P>
                    The mm7 sequence and annotation data may be downloaded 
		    from the Genome Browser
                    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/mm7/">FTP
                    server</A> or
                    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/mm7/">Downloads</A>
                    web page. The mm7 annotation tracks were generated by UCSC
                    and collaborators worldwide. </P>
		    <P>
                    We'd like to thank Deanna Church and the Mouse Genome 
		    Sequencing Consortium for this assembly. We'd also like to
                    acknowledge the work of the UCSC mm7 team: Hiram Clawson, 
		    Fan Hsu, Ann Zweig, Kayla Smith, Robert Kuhn
		    and Donna Karolchik. 
		    For a complete list
		    of the individuals and organizations who participated 
		    in this assembly, see the 
		    <A HREF="credits#mouse_credits">Credits</A> 
		    page. </P>
		    <P>
		    <HR>

<A NAME="1202b05"></A>
<FONT SIZE=4 COLOR="#006666">2 December 2005 - Announcing the VisiGene Image Browser</FONT><BR>
		    <P>
		    We are pleased to announce the release of a new software 
		    tool in the Genome Browser collection, the 
		    <A HREF="../../cgi-bin/hgVisiGene">VisiGene Image Browser</A>.
		    VisiGene offers the ability to view <em>in situ</em> images,
		    allowing examination of expression patterns at both
		    the tissue and cellular levels. The browser serves as a
		    virtual microscope that lets viewers retrieve images that
		    meet specific search criteria, then interactively zoom
		    and scroll across the collection.
		    The VisiGene Browser may be accessed from the menu on the
		    UCSC Genome Bioinformatics home page or through a link on 
		    the details pages of Known Genes for which a VisiGene 
		    annotation exists.</P>
		    <P>
		    The initial release of VisiGene includes the following
		    image collections:
		    <UL>
		    <LI>Mouse <em>in situ</em> images from the 
		    <A HREF="http://www.informatics.jax.org/menus/expression_menu.shtml"
		    TARGET=_blank>Jackson Lab Gene Expression Database</A> 
		    (GED) 
		    <LI>Transcription factors in mouse embryos from the 
		    <A HREF="http://mahoney.chip.org/mahoney/" 
		    TARGET=_blank>Mahoney Center for Neuro-Oncology</A>
		    <LI>Mouse head and brain <em>in situ</em> images from NCBI's
		    <A HREF="http://www.ncbi.nlm.nih.gov/projects/gensat/" 
		    TARGET=_blank>Gene Expression Nervous System Atlas</A> 
		    (GENSAT) database
		    <LI><em>Xenopus laevis in situ</em> images from the 
		    <A HREF="http://www.nibb.ac.jp/en/index.php" 
		    TARGET=_blank>National Institute for Basic Biology</A> 
		    (NIBB) XDB project
		    </UL></P>
		    <P>
		    We hope to integrate images from the 
		    <A HREF="http://www.brainatlas.org/" TARGET=_blank>Allen 
		    Brain Atlas</A> into the VisiGene collection within a few 
		    months. </P>
		    <P>
		    For more information about VisiGene, visit the 
		    <A HREF="../../cgi-bin/hgVisiGene">VisiGene Gateway</A> page 
		    or consult the 
		    <A HREF="help/hgTracksHelp.html#VisiGeneHelp">Using
		    the VisiGene Image Browser</A> section in the User's Guide.
		    </P>
		    <P>
		    We'd like to thank the organizations listed above for 
		    permitting us to add their images to the VisiGene database.
		    VisiGene was written by Jim Kent and Galt Barber of the
		    UCSC Genome Bioinformatics Group. Contact
		    <A HREF="mailto:&#107;e&#110;&#116;&#64;&#115;&#111;&#101;.&#117;&#99;&#115;c.&#101;&#100;u">Jim</A>
		    if you have an image set you'd like to contribute for 
		    display.</P> 
		    <P>
		    <HR>

<A NAME="1202a05"></A>
<FONT SIZE=4 COLOR="#006666">2 December 2005 - Allen Brain Atlas Annotation Added to Genome Browser</FONT><BR>
		    <P>
	   	    We have released a new annotation track &mdash; Allen Brain
		    Atlas Probes &mdash; that may be found in the Expression and
		    Regulation section of the latest mouse and human assemblies
		    (mm6 and hg17). The <A HREF="http://www.brainatlas.org/" 
		    TARGET=_blank>Allen Brain Atlas</A> (ABA) is an extensive 
		    database of high resolution <em>in situ</em> hybridization 
		    images of adult male mouse brains covering the majority of 
		    genes. This track provides a link to the ABA images for each
		    probe. For more information about the ABA, see the 
		    description page that accompanies this track.</P>
		    <P>
		    We'd like to thank the Allen Institute for Brain Science, 
		    and Susan Sunkin in particular, for coordinating with UCSC 
		    on this annotation.</P>
		    <P>
		    <HR>

<A NAME="101705"></A>
<FONT SIZE=4 COLOR="#006666">17 October 2005 - UCSC Genome Browser Tutorial at ASHG 2005</FONT><BR>
		    <P>
		    OpenHelix will be presenting a one-hour seminar, 
		    &quot;Introduction to the UCSC Genome Browser&quot;, at the 
		    American Society of Human Genetics (ASHG) 2005 meeting on 
		    26 October, 6:30-7:30 p.m. The introductory tutorial 
		    will cover the topics needed to effectively use the
		    Genome Browser, including basic search functionality and 
		    BLAT use, Table Browser use, creating and using Custom 
		    Tracks, and an introduction to the Gene Sorter. The class,
		    which is open to all registered ASHG attendees,
		    does not require programming skills, although a basic
		    knowledge of genomic and biological concepts is recommended.
		    </P>
		    <P>
		    The free tutorial will be conducted at the Grand America 
		    Hotel, Audubon Room.  Snacks and beverages will be served. 
	            Attendees will receive a free download of training 
		    materials.  For further information, visit 
		    <A HREF="http://www.openhelix.com" 
		    TARGET=_blank>www.openhelix.com</A> or call 1-888-861-5051.
		    </P>
		    <P>
		    <HR>

<A NAME="101105"></A>
<FONT SIZE=4 COLOR="#006666">11 October 2005 - <em>D. grimshawi</em> Browser Now Available</FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics Group has added the
		    <em>Drosophila grimshawi</em> genome to the collection of 
		    fly genomes available in the UCSC Genome Browser. This
		    assembly (UCSC version droGri1, Aug. 2005) was 
		    produced by <A HREF ="http://www.agencourt.com/" 
		    TARGET=_blank>Agencourt Bioscience Corporation</A> in
		    Beverly, MA, USA, using the Arachne assembler.</P>
		    <P>
		    The droGri1 assembly contains 25,052 scaffolds ranging in 
		    size from 196 bases to 14,170,260 bases. 
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/droGri1/">FTP server</A>
		    or <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#droGri">Downloads</A> page. Please review the data use guidelines outlined in
		    the README.txt files that accompany the downloads. The
		    data use restrictions are also available on the Genome 
		    Browser <A HREF="credits.html#droGri_use">Credits</A> 
		    page. 
		    <P>
		    We'd like to thank Agencourt Bioscience Corporation
		    for providing this assembly. The 
		    UCSC droGri1 browser was produced by 
		    Angie Hinrichs, Brian Raney, Jennifer Jackson, Kayla Smith
		    and Donna Karolchik. The UCSC Genome Bioinformatics Group 
		    generated the initial set of annotation tracks. See the
		    <A HREF="credits.html#droGri_credits">Credits</A> page for 
		    a detailed list of the organizations and individuals who 
		    contributed to this release. 
		    <P>
		    <HR>

<A NAME="100705"></A>
<FONT SIZE=4 COLOR="#006666">7 October 2005 - New <em>D. ananassae</em> Browser Released</FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics Group has updated the 
		    <em>Drosophila ananassae</em> Genome Browser to the 
		    1 August, 2005 assembly. This version (UCSC version droAna2)
		    was produced by <A HREF="http://www.agencourt.com/"
		    TARGET=_blank>Agencourt Bioscience Corporation</A> using the
		    <A HREF="http://www.broad.mit.edu/wga/" 
		    TARGET=_blank>Arachne assembler</A>.  
		    <P>
		    The assembly contains 13,772 scaffolds ranging in size from 
		    55 bases to 23,697,768 bases, with a mean size of 16822.3 
		    and median of 1537. 
		    <P>
		    Sequence and annotation data for the droAna2 assembly can be
		    downloaded from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/droAna2">FTP server</A>
		    or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#droAna">downloads</A>
		    page. Please review the data use guidelines outlined in
		    the README.txt files that accompany the downloads. The
		    data use restrictions are also available on the Genome 
		    Browser <A HREF="credits.html#droAna_credits">credits</A> 
		    page. 
		    <P>
		    We'd like to thank Agencourt Bioscience Corporation for this
		    assembly. The UCSC droAna2 browser was produced by 
		    Angie Hinrichs, Brian Raney, Ann Zweig, Kayla Smith and
		    Donna Karolchik. The UCSC Genome Bioinformatics Group 
		    generated the initial set of annotation tracks. See the
		    <A HREF="credits.html#droAna_credits">credits</A> 
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. 
		    <P>
		    <HR>

<A NAME="100405"></A>
<FONT SIZE=4 COLOR="#006666">4 October 2005 - Updated Cow Browser Available</FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics Group has released a 
		    Genome Browser and Blat server for the Mar. 2005 Btau_2.0
		    draft assembly of the cow genome. This assembly (UCSC
		    version bosTau2) was provided by Baylor
		    College of Medicine Human Genome Sequencing Center in
		    Houston, TX.
		    <P>
		    The Btau_2.0 release was assembled from whole genome 
		    shotgun (WGS) reads using the Atlas genome assembly 
		    system. Several WGS libraries, with inserts of 2 - 4 kb 
		    and 4 - 6 kb, were used to produce the data. 
		    Approximately 23 million reads were assembled, representing 
		    about 17.7 Gb of sequence and 6.2x coverage of the 
		    (clonable) bovine genome. Highly repeated sequences and 
		    BAC clones sequences were omitted from this assembly; these 
		    will be placed in a subsequent version of the draft 
		    sequence. The assembly contains chromosomes 1-29, X, M, and 
		    Bin0, as well as 98058 scaffolds. For details about
		    changes UCSC made to the assembly format for display 
		    purposes, please see the bosTau2 Genome Browser 
		    <A HREF="../cgi-bin/hgGateway?db=bosTau2">gateway page</A>. 
		    <P> 
		    More information on the Btau_2.0 assembly can be found on
		    the Baylor 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/bovine/"
		    TARGET=_blank>Bovine Genome Project</A> web page.
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/bosTau2/">FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#cow">Downloads</A> 
		    page. Please refer to the Baylor 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
		    TARGET=_blank>conditions of use</A> regarding these
		    data. The bosTau2 annotation tracks were generated by 
		    UCSC and collaborators worldwide. 
		    <P>
		    We'd like to thank Baylor College of Medicine for 
		    the bovine sequence and assembly. We'd also like to 
		    acknowledge the members of the UCSC Genome Bioinformatics
		    Group who contributed to the Cow Genome Browser:
		    Galt Barber, Brian Raney, Mark Diekhans, Jennifer Jackson
		    and Donna Karolchik. See the 
		    <A HREF="credits.html#cow_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. 
		    <P>
		    <HR>

<A NAME="092005"></A>
<FONT SIZE=4 COLOR="#006666">20 September 2005 - Three Drosophila Assemblies Released in Genome Browser</FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics Group has added three more 
		    Drosophila genomes to the collection of fly genomes in 
		    the UCSC Genome Browser. The new 
		    genomes, which include the initial assembly of <em>D.
		    erecta</em> (droEre1, Aug. 2005) and the updated assemblies
		    of <em>D. virilis</em> (droVir2, Aug. 2005) and <em>D.  
		    mojavensis</em> (droMoj2, Aug. 2005), were
		    produced by <A HREF ="http://www.agencourt.com/" 
		    TARGET=_blank>Agencourt Bioscience Corporation</A> in
		    Beverly, MA, USA. All three genomes were assembled using
		    the Arachne assembler.</P>
		    <P>
		    The droEre1 assembly contains 5,124 scaffolds ranging in 
		    size from 154 bases to 26,647,023 bases, with a mean size 
		    of 29832.7 and median of 1740. 
		    The droVir2 assembly consists of 13,562 scaffolds ranging in
		    size from 57 bases to 25,269,527 bases, with a mean size of 
		    15263.1 and median of 1249.  
		    The droMoj2 assembly is comprised of 6,843 scaffolds ranging
		    in size from 101 bases to 34,172,700 bases, with a mean size
		    of 28389.6 and median of 1671.  </P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available from the UCSC downloads server:
		    <UL>
		    <LI>droEre1 -  <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/droEre1">FTP</A>, <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/droEre1">HTTP</A>
		    <LI>droVir2 - <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/droVir2">FTP</A>, <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/droVir2">HTTP</A>
		    <LI>droMoj2 - <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/droMoj2">FTP</A>, <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/droMoj2">HTTP</A>
		    </UL>
		    <P>
		    Please review the data use guidelines outlined in
		    the README.txt files that accompany the downloads. The
		    data use restrictions are also available on the Genome 
		    Browser <A HREF="credits.html">credits</A> 
		    page. 
		    <P>
		    We'd like to thank Agencourt Bioscience Corporation
		    for providing these assemblies. The 
		    UCSC Drosophila browsers were produced by 
		    Angie Hinrichs, Brian Raney, Jennifer Jackson and Donna
		    Karolchik. The UCSC 
		    Genome Bioinformatics Group generated the initial set 
		    of annotation tracks. See the
		    <A HREF="credits.html">credits</A> page for 
		    a detailed list of the organizations and individuals who 
		    contributed to this release. 
		    <P>
		    <HR>

<A NAME="090905"></A>
<FONT SIZE=4 COLOR="#006666">9 September 2005 - <em>C. intestinalis</em> v2.0 Genome Browser Released</FONT><BR>
		    <P>
		    The v2.0 <em>C. intestinalis</em> draft assembly from the
		    US DOE Joint Genome Institute (JGI) is now available 
		    in the UCSC Genome Browser and Blat server (UCSC database
		    ci2). </P>
		    <P>
		    This whole genome shotgun assembly was constructed with 
		    the JGI assembler, JAZZ, using paired-end sequencing reads. 
		    Starting with a coverage of 11x, additional data -- 
		    including BAC and FISH markers -- were used to map scaffolds
		    to chromosome arms. The size of this assembly, including 
		    unmapped scaffolds, is 173 Mb, with 94 Mb of the sequence 
		    mapped to chromosome arms. </P>
		    <P>
		    For more information about the ci2 assembly, see the JGI
		    <A HREF="http://genome.jgi-psf.org/ciona4/ciona4.info.html"
		    TARGET=_blank><em>C. intestinalis</em> project page</A>.
		    Additional information and an analysis of the euchromatic 
		    regions of this genome may be found in Dehal, P. 
		    <em>et al.</em>
		    <A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=12481130&query_hl=1" 
		    TARGET=_blank>The draft genome of Ciona intestinalis: 
		    insights into chordate and vertebrate origins</A>. 
		    <em>Science</em> <B>298</B>(5601), 2157-67 (2002). 
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/ci2/" 
		    TARGET=_blank>FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#ciona"
		    TARGET=_blank>Downloads page</A>. The ci2 annotation tracks 
		    were generated by UCSC and collaborators worldwide. See the 
		    <A HREF="credits.html#ciona_credits">Credits page</A> for a 
		    detailed list of the organizations and 
		    individuals who contributed to this release. 
		    <P>
		    Many thanks to the JGI and their collaborators for providing
		    the v2.0 sequence and annotations.  The ci2 Genome Browser 
		    was produced by Brian Raney, Mark Diekhans, Ann Zweig, 
		    Kayla Smith, Robert Kuhn and Donna Karolchik.</P>
		    <P>
		    <HR>

<A NAME="081105"></A>
<FONT SIZE=4 COLOR="#006666">11 August 2005 - Upcoming Genome Browser Seminars - San Francisco, Seattle, Chicago, Boston, Philadelphia, Atlanta</FONT><BR>
		    <P>
		    The UCSC Bioinformatics Group announces six regional 
		    seminars and hands-on computer workshops on the UCSC 
		    Genome Browser, presented by 
		    <A HREF="http://www.openhelix.com/" 
		    TARGET=_blank>OpenHelix:</A>
		    <UL>
		    <LI>
		    San Francisco --  Monday, 22 August
		    <LI>
		    Seattle --  Wednesday, 24 August
		    <LI>
		    Chicago --  Tuesday, 27 September
		    <LI>
		    Boston --  Wednesday, 28 September
		    <LI>
		    Philadelphia --  Tuesday, 15 November
		    <LI>
		    Atlanta --  Wednesday, 16 November
		    </UL>
		    <P>
		    These introductory sessions are geared towards anyone with a 
		    basic knowledge of genomic and biological concepts who is 
		    interested in learning how to use the UCSC Genome Browser. 
		    No programming experience is required. The seminars will
		    cover the topics necessary to learn how to effectively use 
		    the browser tool set, including basic Genome Browser 
		    functionality, searching and BLAT use, Table Browser use, 
		    creating and using custom annotation tracks, and an 
		    introduction to the Gene Sorter. Lectures will be 
		    accompanied by hands-on computer exercises conducted directly 
		    on the Genome Browser web site.
		    <P>
		    For location and registration information, visit the 
		    <A HREF="http://www.openhelix.com/noteworthy.shtml" 
		    TARGET=_blank>OpenHelix</A> website or call 1-888-861-5051. 
		    Academic and student discounts are available.
		    <P>
		    <HR>

<A NAME="061305"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 13 June 2005 - <em>Drosophila simulans</em> Genome Browser Released</FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics Group has released a
		    Genome Browser for the <em>Drosophila simulans</em> draft
		    genome sequence. The Release 1.0 assembly (UCSC version
		    droSim1, April 2005) was produced by the 
		    <A HREF="http://genome.wustl.edu/" TARGET=_blank>Genome
		    Sequencing Center</A> at the Washington University in St. 
		    Louis (WUSTL) School of Medicine. Release 1.0 represents
		    a composite of several different <em>D. simulans</em>
		    lines: contigs from the w501 line comprise the primary
		    scaffolding, with contigs and unplaced reads from six 
		    other lines used to fill gaps in the w501 assembly.</P>
		    <P>
		    The total size of this assembly, excluding the gapless
		    14,972 bp mitochondrial sequence, is 142,405,747 bp 
		    including gaps and 127,241,461 bp excluding gaps. For 
		    more information about the <em>D. simulans</em> assembly 
		    and statistics, see the WUSTL Genome Sequencing Center
		    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Drosophila%20simulans%20White%20501"
		    TARGET=_blank><em>Drosophila simulans</em></A> web page.
		    </P>
		    Downloads of the droSim1 data and annotations may be 
		    obtained from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/droSim1/" 
		    TARGET=_blank>FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#droSim" 
		    TARGET=_blank>Downloads</A> page. The droSim1 annotation 
		    tracks were generated by UCSC and collaborators worldwide.
		    </P>
		    Thanks to the Genome Sequencing Center, WUSTL School of 
		    Medicine for providing this assembly. We'd also like to
		    acknowledge the UCSC team who worked on this release:
		    Angie Hinrichs, Jennifer Jackson, Ali Sultan-Qurraie,
		    Brian Raney and Donna Karolchik. See the 
		    <A HREF="credits.html#droSim_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. </P>
		    <P>
		    <HR>

<A NAME="053105"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 31 May 2005 - June Genome Browser Seminars in San Diego, CA and Salt Lake City, UT</FONT><BR>
		    <P>
		    The UCSC Bioinformatics Group announces two seminars and 
		    hands-on workshops on the UCSC Genome Browser, presented by 
		    <A HREF="http://www.openhelix.com/" TARGET=_blank>OpenHelix</A>, 
		    a bioinformatics training, software testing and consulting 
		    company.
		    <P>
		    These introductory sessions are geared towards anyone with a 
		    basic knowledge of genomic and biological concepts who is 
		    interested in learning how to use the UCSC Genome Browser. 
		    No programming experience is required. The seminars will
		    cover the topics necessary to learn how to effectively use 
		    the browser tool set, including basic Genome Browser 
		    functionality, searching and BLAT use, Table Browser use, 
		    creating and using custom annotation tracks, and an 
		    introduction to the Gene Sorter. Lectures will be 
		    accompanied by hands-on computer exercises conducted directly 
		    on the Genome Browser web site.
		    <P>
		    The first three-hour course will be held on Thursday 
		    9 June in the Salt Lake City, UT area. Two sessions will 
		    be offered: 9am-12pm and 1pm-4pm. 
		    <P>
		    The second seminar will be held in San Diego, CA
		    on Friday, 10 June.  Two sessions 
		    will be offered: 9am-12pm and 1pm-4pm. 
		    <P>
		    For registration information, visit the 
		    <A HREF="http://www.openhelix.com/noteworthy.shtml" 
		    TARGET=_blank>OpenHelix</A> website or call 1-888-861-5051. 
		    Academic and student discounts are available.
		    <P>
		    <HR>

<A NAME="051805"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 18 May 2005 - Rhesus Monkey Genome Browser Released</FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics group has released a 
		    Genome Browser for the rhesus monkey draft assembly,
		    Mmul_0.1 (UCSC version rheMac1), produced by the 
                    Baylor College of Medicine Human Genome Sequencing Center 
		    in collaboration with J. Craig Venter Science Foundation 
		    Joint Technology Center and the Genome
                    Sequencing Center at Washington University School of
                    Medicine in St. Louis.
		    This Old World monkey, which is a primate model organism,
		    is important for the study of human biology and disease
		    due to its genetic, physiologic and metabolic 
		    similarities to humans.
		    <P>
		    Mmul_0.1 is a preliminary assembly using whole genome 
		    shotgun (WGS) reads from small and medium insert clones. 
		    Approximately 14.6 million reads were used in the 
		    assembly, representing about 12.6 Gb of sequence and 
		    about 4.6x coverage of the (clonable) genome. 
		    The total length of all contigs is 
		    approximately 2.7 Gb, or 2.8 Gb including gaps between 
		    contigs.
		    Highly-repeated sequences and sequences from BAC clones 
		    were omitted from this assembly and will be placed in a 
		    subsequent version of the draft sequence. 
		    <P>
		    For more information about the rheMac1 assembly, see the 
		    Baylor 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/rmacaque/"
		    TARGET=_blank>Rhesus Monkey Project</A> web page.</P>
		    <P>
		    The rheMac1 sequence and annotation data can be downloaded
		    from the UCSC Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/rheMac1/">FTP
		    server</A> or the
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#rhesus">Downloads</A>
		    page. These data have specific
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
		    TARGET=_blank>conditions for use</A>.</P> 
		    <P>
		    Many thanks to Baylor College of Medicine, the Rhesus
		    Monkey Genome Sequencing Consortium, and the other 
		    institutions who contributed to the sequencing and mapping
		    effort of the Mmul_0.1 release. The UCSC Rhesus Genome 
		    Browser was produced by Robert Baertsch, Galt Barber, 
		    and Donna Karolchik. The initial set of rheMac1 annotation
	            tracks was generated by the <A HREF="../staff.html">UCSC 
		    Genome Bioinformatics Group</A>. See the 
		    <A HREF="/goldenPath/credits.html#rhesus_credits">credits</A> 
		    page for a detailed list of the organizations and 
		    individuals who contributed to the release of this 
		    browser. </P>
		    <P>
<HR>

<A NAME="050405"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 5 May 2005 - New Mouse Assembly Available in Genome Browser</FONT><BR>
		    <P>
		    The latest mouse genome assembly from the Mouse Genome
		    Sequencing Consortium, NCBI Build 34 (UCSC version mm6),
		    is now available in the UCSC Genome Browser. In 
		    conjunction with this release, we have archived mouse
		    assemblies mm3 and mm4. All archived assemblies are 
		    accessible on our 
		    <A HREF="http://genome-archive.cse.ucsc.edu/"
		    TARGET=_blank>archive server</A>.</P>
		    <P>
		    The Build 34 assembly has an assembled length of 
		    approximately 2.6 Gb, of which about 1.9 Gb is finished 
		    sequence. Chromosomes 2, 
		    4, 11 and X are finished in this build. To review in-depth
		    statistics on the assembly, see the NCBI
                    <A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/mouse/contig/Build34.html"
                    TARGET=_blank>Build 34 Data</A> web page.
		    NOTE: In the Genome Browser mm6 assembly,
		    chrY_random erroneously contains a region duplicated from 
		    chrY. For more information about this issue, see the 
		    Genome Browser
		    <A HREF="../FAQ/FAQdownloads#download30">FAQ</A>. Please 
		    also note that the UCSC mm6 database contains only the 
		    reference strain C57BL/6J.</P>
                    <P>
                    The mm6 sequence and annotation data may be downloaded 
		    from the Genome Browser
                    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/mm6/">FTP
                    server</A> or
                    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/mm6/">Downloads</A>
                    web page. The mm6 annotation tracks were generated by UCSC
                    and collaborators worldwide. </P>
		    <P>
		    Mirror sites, please note: due to the size of the 
		    comparative genomics annotations in this assembly, the 
		    data set requires an unusually large amount of disk space.
		    If you regularly download updates of the mouse data to 
		    your site, you may want to examine your available disk 
		    space before adding the annotation database from this 
		    assembly and/or reschedule your download for a period of 
		    low activity on your server. The size of the mm6 
		    annotation database directory 
		    ($WEBROOT/goldenPath/mm6/database/) is 
		    approximately 93 GB. Including the related net and chains,
		    the total size is 124 GB.</P>
		    <P>
                    We'd like to thank Deanna Church and the Mouse Genome 
		    Sequencing Consortium for this assembly. We'd also like to
                    acknowledge the work of the UCSC mm6 team: Hiram Clawson, 
		    Fan Hsu, Jennifer Jackson, Robert Kuhn, Ali 
		    Sultan-Qurraie, Heather Trumbower and Donna Karolchik. 
		    For a complete list
		    of the individuals and organizations who participated 
		    in this assembly, see the 
		    <A HREF="credits#mouse_credits">Credits</A> 
		    page.
                    <P>
<HR>

<A NAME="050205"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 2 May 2005 - Archiving mm3 and mm4 assemblies</FONT><BR>
		    <P>
		    To make room for the NCBI Build 34 mouse assembly (mm6, 
		    March 2005), we are archiving the mm3 and mm4 mouse 
		    assemblies. These assemblies will remain available on our 
		    <A HREF="http://genome-archive.cse.ucsc.edu/">archive
		    server</A>, although blat will no longer be supported. 
		    We plan to release the mm6 assembly in the next day or so.
		    </P>
		    <P>
<HR>

<A NAME="032405"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 24 Mar. 2005 - Genome Browser Tutorial at Experimental Biology 2005 on 5 April</FONT><BR>
		    <P>
		    OpenHelix will be presenting a one-hour introductory 
		    seminar on the UCSC Genome Browser at the 
		    Experimental Biology 2005 meeting on 5 April, 5:00-6:00. 
		    The tutorial will cover the topics needed to effectively 
		    use the Browser, including basic functionality, searching 
		    and BLAT use, Table Browser use, creating and using Custom
		    Tracks, and an introduction to the Gene Sorter.  </P>
		    <P>
		    The free session will be conducted at the San 
		    Diego Convention Center, Room 12. For further information,
		    visit the <A HREF="http://www.openhelix.com" 
		    TARGET=_blank>OpenHelix website</A> or call 
		    1-888-861-5051.</P>
		    <P>
		    OpenHelix will also be presenting brief introductory 
		    overviews of the UCSC Genome Browser in its Experimental 
		    Biology booth #133-135 during show hours. Stop by the 
		    booth for more information and to receive your free UCSC 
		    Quick Reference Card. </P> 
		    <P>
<HR>

<A NAME="030205"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 2 Mar. 2005 - CCDS Data Set Released</FONT><BR>
		    <P>
		    The initial results from the Consensus Coding Sequence 
		    project (CCDS) were released to the public today. 
		    CCDS is a collaborative effort to identify a core set of 
		    human protein-coding regions that are consistently 
		    annotated and of high quality. </P>
		    <P>
		    The initial CCDS data set, 
		    containing nearly 15,000 transcripts, has been posted on three 
		    Internet sites: the 
		    <A HREF="../cgi-bin/hgTracks?db=hg17">UCSC Genome Browser</A>, 
		    the <A HREF="http://www.ensembl.org/" 
		    TARGET=_blank>Ensembl Browser</A> and the 
		    <A HREF="http://www.ncbi.nlm.nih.gov/CCDS/"
		    TARGET=_blank>NCBI CCDS Database</A> website. 
		    The genes may be viewed on the UCSC hg17 (May 2004) 
		    Human Genome Browser in the CCDS annotation track 
		    located in the Genes and Gene Prediction Tracks section.
		    </P>
		    <P>
		    The CCDS gene set is built by consensus among the 
		    following collaborating organizations:
		    <UL>
		    <LI>
		    <A HREF="http://www.ebi.ac.uk/" TARGET=_blank>European 
		    Bioinformatics Institute</A> (EBI)
		    <LI>
		    <A HREF ="http://www.ncbi.nlm.nih.gov/" 
		    TARGET=_blank>National Center for Biotechnology 
		    Information</A> (NCBI)
		    <LI>
		    <A HREF="http://www.cbse.ucsc.edu/" 
		    TARGET=_blank>University of California Santa Cruz</A>
		    (UCSC)
		    <LI>
		    <A HREF="http://www.sanger.ac.uk/"
		    TARGET=_blank>Wellcome Trust Sanger Institute</A> (WTSI)
		    </UL></P>
		    <P>
		    Each gene in the CCDS set is assigned a unique identifier
		    and version number, e.g. CCDS234.1. The version number 
		    will be updated if either the CDS structure or the 
		    underlying genome sequence at that location changes. The 
		    CCDS set will be mapped forward, with identifiers 
		    preserved, when the annotations or sequence update. 
		    Changes to existing CCDS genes will be done by 
		    collaboration agreement; no single group will change the 
		    set unilaterally. CCDS identifiers are included on 
		    relevant NCBI RefSeq and Entrez Gene records.</P>
		    <P>
		    Gene structure information is obtained from both curated
		    and automated genes sources. The primary curation groups 
		    are the <A HREF="http://www.sanger.ac.uk/HGP/havana/" 
		    TARGET=_blank>Havana</A> team at the WTSI and the 
		    <A HREF="http://www.ncbi.nlm.nih.gov/RefSeq/" 
		    TARGET=_blank>RefSeq</A> annotation group at NCBI. In 
		    addition, manually curated information on chr14
		    (Genoscope) and Chr7 (WUSTL) has been included from the 
		    <A HREF="http://vega.sanger.ac.uk/" TARGET=_blank>Vega</A>
		    database. The <A HREF="http://www.ensembl.org/" 
		    TARGET=_blank>Ensembl</A> group and the RefSeq 
		    computational pipeline provide the automated data. Curated
		    information is favored over automated information; the 
		    information must be consistent between the Hinxton 
		    (Vega/Ensembl) and NCBI groups and also pass UCSC's 
		    stringent QC requirements. </P>
		    <P>
		    At a minimum, a gene must meet three quality criteria
		    to be included in the CCDS set:
		    <UL>
		    <LI>
		    The coding region has a full-length annotation with
		    an initiating ATG, a valid stop codon, and no in-frame
		    stop codons. 
		    <LI>
		    Protein can be translated from the genome without 
		    frameshifts. 
		    <LI>
		    Consensus splice-sites are used. 
		    </UL></P>
		    <P>
		    For more information on the process used to identify and 
		    evaluate CCDS genes, refer to the description page for 
		    the Genome Browser CCDS track. The number and type of 
		    quality tests performed may be expanded in the future, 
		    but includes analysis to identify putative pseudogenes, 
		    retrotransposed genes, consensus splice sites, supporting 
		    transcripts, and protein homology.
		    The completeness of the CCDS gene set should improve as 
		    the curation and automated annotation processes mature
		    and additional experimental validation is obtain for
		    weakly-supported genes.</P>
		    <P>
		    UCSC would like to thank EBI, NCBI and WTSI for their
		    collaboration in the release of these data. 
		    We would also like to acknowledge the efforts of the 
		    UCSC staff who worked on this project: Mark Diekhans, 
		    Robert Baertsch, Adam Siepel, Fan Hsu, and Robert Kuhn.
		    <P>
<HR>

<A NAME="022405"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 24 Feb. 2005 - Genome Browser Training - St. Louis and NYC</FONT><BR>
		    <P>
		    We'd like to announce upcoming UCSC Genome Browser 
		    seminars and hands-on computer workshops in St. Louis, MO,
		    and New York, NY,  presented by OpenHelix, the 
		    bioinformatics training, software testing and consulting 
		    company. 
		    <P>
		    These introductory tutorials will cover the topics 
		    needed to effectively use the browser, including basic 
		    functionality of Genome Browser searching and BLAT use, 
		    Table Browser use, creating and using Custom Tracks, and 
		    an introduction to the Gene Sorter. 
		    <P>
		    The St. Louis class will be held on 29 March, 2005 at 
		    the <A HREF="http://www.ociweb.com/about/stl.html"
		    TARGET=_blank>Object Computing, Inc.</A> training 
		    center. Two sessions will be offered: 9 a.m. - 12 p.m. 
		    and 1 - 4 p.m. 
		    <P>
		    The New York City class will be held on 30 March, 2005 at 
		    the 
		    <A HREF="http://www.thinkpathtraining.com/rawdoc.asp?Temp=7240&docID=7087"
		    TARGET=_blank>ThinkPath Training</A> facility. Two
		    sessions will be offered: 9 a.m. - 12 p.m. and 
		    1 - 4 p.m. 
		    <P>
		    For more information about class locations, costs, and
		    registration, visit the 
		    <A HREF="http://www.openhelix.com/noteworthy.shtml" 
		    TARGET=_blank>OpenHelix</A> website or call 
		    1-888-861-5051.  Academic, student, and early registration
		    discounts are available.
		    <P>
<HR>

<A NAME="022305"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 23 Feb. 2005 - New Honeybee Genome Browser</FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics group has released a
		    Genome Browser for a second honeybee assembly, Amel_2.0
		    (UCSC version apiMel2), produced by the Baylor College of 
		    Medicine Human Genome Sequencing Center.
		    <P>
		    The assembly -- which is approximately 229 Mb in size 
		    including contig gaps -- was sequenced using a combined 
		    whole genome shotgun (WGS) and BAC clone approach. 
		    Overall sequence coverage is estimated at 7.5x. 
		    In this assembly, scaffolds have been arranged into 17
		    linkage groups that are somewhat equivalent to 
		    chromosomes.  For more information about the assembly, 
		    see the Baylor
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/honeybee/"
		    TARGET=_blank>honeybee project</A> website.
		    <P>
		    The apiMel2 sequence and annotation data can be 
		    downloaded from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/apiMel2/">FTP server</A>
		    or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#apiMel">downloads</A>
		    page. Please see the Baylor
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html"
		    TARGET=_blank>conditions of use</A> page for guidelines 
		    regarding the use of these data.
		    <P>
		    Many thanks to Baylor College of Medicine and the other
		    institutions who contributed to the sequencing and 
		    mapping effort of the Amel_v2.0 release. The UCSC 
		    <em>A. mellifera</em> Genome Browser
		    was produced by Andy Pohl, Angie Hinrichs, Jennifer 
		    Jackson, and Donna Karolchik. The initial set of apiMel2 
		    annotation tracks was generated by the UCSC Genome 
		    Bioinformatics Group.  See the
		    <A HREF="/goldenPath/credits.html#apiMel_credits">credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to the release of
		    this browser.
		    <P>
<HR>

<A NAME="021705"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 17 Feb. 2005 - Cow Genome Browser Released</FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics Group has released a 
		    Genome Browser and Blat server for the Sep. 2004 Btau_1.0
		    draft assembly of the cow genome. This assembly (UCSC
		    version bosTau1) was provided by Baylor
		    College of Medicine Human Genome Sequencing Center in
		    Houston, TX.
		    <P>
		    The Btau_1.0 release was assembled from whole genome 
		    shotgun (WGS) reads using the Atlas genome assembly 
		    system. Several WGS libraries, with inserts of 2 - 4 kb 
		    and 4 - 6 kb, were used to produce the data. Approximately
		    15 million reads were assembled, representing about 9 Gb 
		    of sequence and 3x coverage of the (clonable) bovine 
		    genome. Highly repeated sequences and BAC clones sequences
		    were omitted from this assembly; these will be placed in a
		    subsequent version of the draft sequence. The assembly 
		    contains approximately 450,000 scaffolds.
		    <P> 
		    For more information on the Btau_1.0 assembly, see the
		    Baylor 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/bovine/"
		    TARGET=_blank>Bovine Genome Project</A> web page.
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/bosTau1/">FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#cow">Downloads</A> 
		    page. Please refer to the Baylor 
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
		    TARGET_blank>conditions of use</A> regarding these
		    data. The bosTau1 annotation tracks were generated by 
		    UCSC and collaborators worldwide. 
		    <P>
		    We'd like to thank Baylor College of Medicine for 
		    the bovine sequence and assembly. We'd also like to 
		    acknowledge the members of the UCSC Genome Bioinformatics
		    Group who contributed to the Cow Genome Browser:
		    Heather Trumbower, Jim Kent, Hiram Clawson, Angie 
		    Hinrichs, Brian Raney, Mark Diekhans, Robert Kuhn, Ali
		    Sultan-Qurraie, and Donna Karolchik. See the 
		    <A HREF="/goldenPath/credits.html#cow_credits">Credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. 
		    <P>
<HR>

<A NAME="021505"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 15 Feb. 2005 - UniProt (Swiss-Prot/TrEMBL) Display Changes</FONT><BR>
		    <P>
		    We have made several adjustments to the Genome
		    Browser databases to accommodate recent changes to display
		    IDs introduced by UniProt (aka Swiss-Prot/TrEMBL). If you 
		    are using the proteinID field in our knownGene table or 
		    the Swiss-Prot/TrEMBL display ID for indexing or 
		    cross-referencing other data, we strongly suggest you 
		    transition to the UniProt accession number. These changes 
		    will also affect anyone who is mirroring our site. 
		    <P>
		    To review a detailed list of the changes to the Genome
		    Browser databases, see our 
		    <A HREF="../FAQ/FAQreleases#release13">FAQ</A>. 
		    <P>
<HR>

<A NAME="021405"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 14 Feb. 2005 -<em>D. melanogaster</em> Release 4 Now Available</FONT><BR>
		    <P>
		    The <em>D. melanogaster</em> Release 4 assembly (UCSC
		    version dm2, Apr. 2004) is now available in the 
		    Genome Browser. This version includes Release 4 
		    euchromatic sequence from the Berkeley Drosophila Genome 
		    Project (<A HREF="http://www.fruitfly.org/" 
		    TARGET=_blank>BDGP</A>), Release 3.2 heterochromatic 
		    genome sequence from the Drosophila Heterochromatin Genome
		    Project (<A HREF="http://www.dhgp.org/" 
		    TARGET=_blank>DHGP</A>), and Release 4.0 annotations 
		    (dated Nov.  2004) from <A HREF="http://flybase.net/"
		    TARGET=_blank>FlyBase</A>. Insects in the UCSC
		    Genome Browser collection include six Drosophila
		    genomes, the honeybee, and <em>Anopheles gambiae</em>.
		    <P>
		    The Release 4 euchromatic sequence, 118.4 Mb in size,
		    contains six euchromatic chromosome arms with a total 
		    of 23 sequence gaps. The assembly was validated in 
		    collaboration with the Genome Sciences Centre at the 
		    British Columbia Cancer Agency in Vancouver, Canada, 
		    using fingerprint analysis of a tiling path of BACs 
		    spanning the genome.
		    <P>
		    The Release 3.2 heterochromatic sequence is based on 
		    20.7 Mb of Release 3 whole genome shotgun scaffolds 
		    from Celera that could not be assembled into the 
		    euchromatin arms, as well as a few BDGP-sequenced 
		    scaffolds.  The scaffolds have been assigned to 
		    chromosomes where possible.
		    <P>     
		    For more information on this assembly, see the release 
		    notes for the 
		    <A HREF="http://www.fruitfly.org/annot/release4.html" 
		    TARGET=_blank>Release 4 euchromatic sequence</A> and
		    <A HREF="http://www.dhgp.org/index_release_notes.html" 
		    TARGET=_blank>Release 3.2 heterochromatic sequence</A>.
		    <P>
		    Downloads of the dm2 data and annotations may be obtained 
		    from the UCSC Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/dm2/" 
		    TARGET=_blank>FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#fruitfly">Downloads</A> 
		    page. The dm2 annotation tracks were provided by 
		    FlyBase or generated by UCSC. 
		    <P>
		    Thanks to BDGP, DHGP, and the Flybase Consortium for
		    providing the sequence, assembly, gene annotations and 
		    analysis of this genome. The UCSC Genome Browser team
		    is Angie Hinrichs, Brian Raney, Galt Barber, and
		    Donna Karolchik. See the 
		    <A HREF="/goldenPath/credits.html#dm_credits">Credits</A>
		    page for a detailed list of the organizations and 
	   	    individuals who contributed to this release.
		    <P>
<HR>

<A NAME="011805"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 18 Jan. 2005 - Opossum Genome Browser</FONT><BR>
		    <P>
		    The opossum (<em>Monodelphis domestica</em>) is the latest
		    vertebrate addition to the UCSC Genome Browser genome
		    collection. The Oct. 2004 preliminary assembly -- UCSC 
		    version monDom1 -- was sequenced and assembled by 
		    <A HREF="http://www.broad.mit.edu/seq/"
		    TARGET=_blank>The Broad Institute</A>, Cambridge, MA, USA.
		    <P>
		    This preliminary draft was assembled from 33,507,069 
		    placed reads, and consists of 109,065 contigs containing 
		    a total of 3,492,108,230 bases.
		    There are 3,563,247,205 total bases in the 19,348 
		    scaffolds in the assembly, with 71,138,975 bases in
		    the gaps between contigs within the scaffolds. The
		    scaffolds range in size from 1,000 bases to 22,286,839 
		    bases.
		    <P>
		    The monDom1 sequence and annotation data can be 
		    downloaded from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/monDom1/">FTP server</A>
		    or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#opossum">downloads</A>
		    page. Please review the 
		    <A HREF="/goldenPath/credits.html#opossum_use"
		    TARGET=_blank>guidelines</A> for using these data.
		    <P>
		    Many thanks to The Broad Institute for providing these
		    data. The UCSC opossum Genome Browser
		    was produced by Hiram Clawson, Galt Barber,
		    Ali Sultan-Qurraie, Brian Raney and Donna Karolchik. The 
		    initial set of annotation tracks was generated by the 
		    UCSC Genome Bioinformatics Group. See the
		    <A HREF="/goldenPath/credits.html#opossum_credits">credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to the release of
		    this browser.
		    <P>
<HR>

<!---------------- 2004 archived news --------------->
<A NAME="2004"></A>
<A NAME="122304"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 23 Dec. 2004 - Updated Zebrafish Assembly (Zv4) Now Available</FONT><BR>
		    <P>
		    The latest zebrafish assembly -- Zv4 (UCSC version 
		    danRer2, June 2004) -- is now available in the UCSC
		    Genome Browser and Blat server. The Zv4 assembly was 
		    produced by 
		    The Wellcome Trust Sanger Institute in collaboration with 
		    the Max Planck Institute for Developmental Biology in 
		    Tuebingen, Germany, and the Netherlands Institute for 
		    Developmental Biology (Hubrecht Laboratory), Utrecht, 
		    The Netherlands.
		    This assembly consists of 1,560,480,686 bp in 21,333 
		    scaffolds with a sequence coverage of approximately 5.7x. 
		    443 Mb of the assembly is from 2,828 finished clones 
		    and 121 Mb is from 1,272 unfinished clones. 
		    <P>
		    75% of the sequence was tied to the fingerprint
		    contig (FPC) map, which provided a template for placing 
		    the unfinished sequence.  The remaining sequence was 
		    filled with whole genome shotgun (WGS) contigs using a 
		    combination of sequence alignment and BAC end positions. 
		    The WGS contigs used in this assembly were identical to 
		    those used for the Zv3 assembly, but the FPC data and 
		    its integration with the WGS data has been considerably 
		    improved. 
		    <P>
		    The 
		    Sanger Institute notes that there is high level of 
		    misassembly present in this release due to the large 
		    amount of 
		    polymorphism in the DNA source. Highly variable regions 
		    within the genome posed assembly difficulties, most 
		    likely because the sequences originated from different 
		    haplotypes. 
		    For more information about this assembly, see the Sanger 
		    Institute web page for the 
		    <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/"
		    TARGET=_blank><em>Danio rerio</em> Sequencing Project</A>.
		    <P>
		    In addition to the assembly improvements noted above, a
		    new zebrafish repeat library was used to mask repeats in
		    the danRer2 Genome Browser, resulting in a cleaner set of 
		    alignments containing considerably less &quot;noise&quot;.
		    <P>
		    The danRer2 sequence and annotation data can be 
		    downloaded from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/danRer2/">FTP server</A>
		    or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#zebrafish">downloads</A>
		    page. Please review the 
		    <A HREF="http://www.sanger.ac.uk/Projects/use-policy.shtml"
		    TARGET=_blank>guidelines</A> for using these data.
		    <P>
		    We'd like to thank the Wellcome Trust Sanger Institute, 
		    the Max Planck Institute for Developmental Biology,
		    Hubrecht Laboratory and the other
		    institutions who contributed to the sequencing and 
		    mapping effort of this release. Special thanks to the 
		    Zebrafish Genome Initiative at Children's Hospital in 
		    Boston for their collaboration on this release. 
		    The UCSC zebrafish Genome Browser
		    was produced by Rachel Harte, Mark Diekhans, Heather 
		    Trumbower, Jennifer Jackson, and Donna Karolchik. The 
		    initial set of annotation tracks was generated by the 
		    UCSC Genome Bioinformatics Group.  See the
		    <A HREF="/goldenPath/credits.html#zebrafish_credits">credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to the release of
		    this browser.
		    <P>
<HR>

<A NAME="122204"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 22 Dec. 2004 - UCSC Releases Honeybee Browser</FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics group has released a
		    Genome Browser for the Amel_1.2 honeybee assembly
		    produced by the Baylor College of Medicine Human Genome
		    Sequencing Center.
		    <P>
		    The assembly -- which is approximately 213 Mb in size 
		    including contig gaps -- 
		    was sequenced using a combined whole genome shotgun (WGS)
		    and BAC clone approach. The BAC clone sequence was
		    produced following a pooled-array strategy. Overall 
		    sequence coverage is estimated at 6X.
		    For more information about this assembly, see the Baylor
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/honeybee/"
		    TARGET=_blank>honeybee project</A> website.
		    <P>
		    The honeybee genome is of interest to the agricultural
		    community, in addition to its role as a model organism in the
		    study of several human health issues, including immunity,
		    allergic reaction, antibiotic resistance, development,
		    mental health, longevity, and diseases of the X chromosome.
		    The honeybee is also interesting for its social and
		    behavioral traits.
		    <P>
		    The apiMel1 sequence and annotation data can be 
		    downloaded from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/apiMel1/">FTP server</A>
		    or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#apiMel">downloads</A>
		    page. Please see the Baylor
		    <A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html"
		    TARGET=_blank>conditions of use</A> page for guidelines 
		    regarding the use of these data.
		    <P>
		    Many thanks to Baylor, Children's Hospital Oakland 
		    Research Institute (CHORI),  and the other
		    institutions who contributed to the sequencing and 
		    mapping effort of the Amel_1.2 release. The UCSC 
		    <em>A. mellifera</em> Genome Browser
		    was produced by Angie Hinrichs, Brian Raney, Robert Kuhn, 
		    and Donna Karolchik. The initial set of apiMel1 annotation
		    tracks was generated by the UCSC Genome Bioinformatics 
		    Group.  See the
		    <A HREF="/goldenPath/credits.html#apiMel_credits">credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to the release of
		    this browser.
		    <P>
<HR>
<A NAME="121504"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 15 Dec. 2004 - Three New Drosophila Assemblies in Genome Browser</FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics Group has added three more 
		    Drosophila genomes to the UCSC Genome Browser. The new 
		    genomes  -- initial assemblies of <em>D.
		    ananassae</em> (droAna1, Jul. 2004), <em>D. virilis</em> 
		    (droVir1, Jul. 2004), and <em>D.  mojavensis</em> 
		    (droMoj, Aug. 2004) -- join the <em>D.  melanogaster</em>,
		    <em>D. yakuba</em>, and <em>D.  pseudoobscura</em> genomes
		    already in the collection.
		    <P>
		    The <em>D. virilis</em> and <em>D. mojavensis</em> genomes
		    were sequenced and assembled by 
		    <A HREF="http://www.agencourt.com/" 
		    TARGET=_blank>Agencourt Bioscience Corporation</A> using
		    the Arachne assembler. <A HREF="http://www.tigr.org/"
		    TARGET=_blank>The Institute of Genomic
		    Research</A> (TIGR) produced the <em>D. ananassae</em>
		    assembly using the Celera assembler. For more details
		    about an individual genome, refer to the Genome Browser
		    <A HREF="../cgi-bin/hgGateway?org=D.+ananassae">gateway</A> 
		    page associated with the assembly.
		    <P>
		    In upcoming months, UCSC plans to provide 
		    several additions and enhancements to the 
		    Drosophila browsers:
		    <P>
		    - additional Drosophila genomes, including
		    <em>D. erecta</em> and <em>D. simulans</em><BR>
		    - updated versions of <em>D. melanogaster</em> and other 
		    genomes<BR>
		    - multiple alignments of all fly assemblies from the 
		    12-Drosophila project to <em>D. melanogaster</em>
		    <P>
		    Sequence and annotation data for all the 
		    Drosophila assemblies can be 
		    downloaded from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/">FTP server</A>
		    or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">downloads</A>
		    page. Please review the data use guidelines outlined in
		    the README.txt files that accompany the downloads. The
		    data use restrictions are also available on the Genome 
		    Browser <A HREF="/goldenPath/credits.html">credits</A> 
		    page. 
		    <P>
		    We'd like to thank Agencourt Bioscience Corporation, 
		    TIGR, and other institutions who contributed to the 
		    sequencing and mapping effort for these assemblies. The 
		    UCSC Drosophila browsers were produced by 
		    Angie Hinrichs, Brian Raney, Heather Trumbower, 
		    Robert Kuhn, Galt 
		    Barber, Jennifer Jackson, Ali Sultan-Qurraie, and Donna 
		    Karolchik. The UCSC 
		    Genome Bioinformatics Group generated the initial set 
		    of annotation tracks. See the
		    <A HREF="/goldenPath/credits.html">credits</A> page for 
		    a detailed list of the organizations and individuals who 
		    contributed to this release. 
		    <P>
<HR>

<A NAME="120304"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 3 Dec. 2004 - SNP Track for Build 35 (hg17) Assembly Released</FONT><BR>
		    <P>
		    UCSC has released the much-awaited SNP annotation track for the 
		    Build 35 (hg17) Human Genome Browser. The SNP data were
	 	    obtained from dbSnp Build 123, the first dbSnp
		    release based on the latest human assembly.
		    <P>
		    In conjunction with the data release, we have made 
		    several improvements to the graphical display of the 
		    track. 
		    Our users now have the ability to filter the SNP data
		    based on validation status,
	 	    functional class, variant class, average heterozygosity
		    and other features. Variants may be drawn in 
		    user-selected colors to highlight additional details
		    in the graphical image. Also, the browser now displays all
		    SNP variants at the appropriate base width, rather than
		    as generic ticks.
		    <P>
<HR>

<A NAME="111704"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 17 Nov. 2004 - <em>X. tropicalis</em> v.3.0 Released in Genome Browser</FONT><BR>
		    <P>
		    The v.3.0 <em>Xenopus tropicalis</em> assembly is now
		    available on the UCSC Genome Browser. This whole genome
		    shotgun assembly 
		    (xenTro1, October 2004) was provided by the U.S. DOE 
		    Joint Genome Institute (JGI).
		    <P>
		    Version 3.0 was constructed with the JGI assembler, 
		    Jazz, using paired end sequencing reads at a coverage of 
		    7.4X.  After trimming for vector and quality, 19.1 
		    million reads assembled into 27,064 scaffolds totaling 
		    1.63 Gbp. Roughly half the genome is contained in 392 
		    scaffolds, each of which is at least 1.2 Mb in length.
		    For more information about this assembly, see the JGI
		    <A HREF="http://genome.jgi-psf.org/Xentr3/Xentr3.home.html"
		    TARGET=_blank><em>X. tropicalis</em> website</A>.
		    <P>
		    The xenTro1 sequence and annotation data can be 
		    downloaded from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/xenTro1/">FTP server</A>
		    or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#xenTro">downloads</A>
		    page. Please see the JGI
		    <A HREF="http://genome.jgi-psf.org/Xentr3/Xentr3.download.html"
		    TARGET=_blank>data release policy</A> for data use 	
		    guidelines.
		    <P>
		    Many thanks to JGI and the other
		    institutions who contributed to the sequencing and 
		    mapping effort of the v. 3.0 release. The UCSC <em>X. 
		    tropicalis</em> Genome Browser
		    was produced by Fan Hsu, Jim Kent, Heather Trumbower, 
		    Hiram Clawson, Brian Raney, Galt Barber, Mark Diekhans, 
		    Angie Hinrichs, and Donna Karolchik. The initial set of 
		    xenTro1 annotation tracks was generated by the
		    UCSC Genome Bioinformatics Group. See the
		    <A HREF="/goldenPath/credits.html#xenTro_credits">credits</A>
		    page for a detailed list of the organizations and 
		    individuals who contributed to the release of
		    this browser.
		    <P>
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 17 Nov. 2004 - Genome Browser Tutorial Offered at ASCB Meeting</FONT><BR>
		    <P>
		    OpenHelix will be presenting the tutorial 
		    &quot;Introduction to the UCSC Genome Browser&quot; 
		    at the American
		    Society for Cell Biology (ASCB) Annual Meeting in 
		    Washington, D.C. on 5 Dec., 8:30 - 10:00 p.m. This
		    introductory session will cover the topics needed to 
		    effectively use the Genome Browser, including basic 
		    functionality of Genome Browser searching and BLAT use, 
		    Table Browser use, creating and using Custom Tracks, and 
		    an introduction to the Gene Sorter.  
		    <P>
		    The free 1-1/2-hour tutorial will be conducted at the 
		    Washington Convention Center, Room 206. OpenHelix will 
		    also be presenting brief introductory overviews of the 
		    UCSC Genome Browser in its ASCB booth #510-512 during 
		    show hours.  Stop by the booth for more information and 
		    to receive your free UCSC Quick Reference Card.  
		    <P>
		    For more information about the tutorial and other
		    services offered by OpenHelix, visit 
		    <A HREF="http://www.openhelix.com" 
		    TARGET=_blank>www.openhelix.com</A> or call 
		    1-888-861-5051.
		    <P>
<HR>

<A NAME="100804"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 8 Oct. 2004 - Updated Proteome Browser Released</FONT><BR>
		    <P>
		    UCSC has released v1.1 of the 
		    <A HREF="../cgi-bin/pbGateway">UCSC 
		    Proteome Browser</A>. This updated version includes the
		    following major enhancements:
		    <UL>
 		    <LI>direct access to a Proteome Browser gateway via the 
		    &quot;Proteome Browser&quot; menu link
		    on the Genome Browser home page. The Proteome Browser gateway 
		    page prompts the user for a protein ID or gene 
		    symbol, then directly starts up the Proteome Browser, 
		    eliminating the multiple Genome Browser steps required by 
		    the previous release. 
 		    <LI>extended protein coverage that includes all proteins in 
		    the Swiss-Prot/TrEMBL databases, rather than just the
		    human, mouse and rat protein sets included in the previous 
		    version.
		    </UL>
		    <P>
		    In addition to providing direct access to the Proteome
		    Browser, the v1.1 release preserves the existing tight
		    coupling between the Proteome Browser and the Genome Browser 
		    for the human, mouse and rat genomes. Users may still 
		    navigate between the Genome Browser Known Genes track and
		    the Proteome Browser for quick, easy access to
		    the details of a gene's genomic and proteomic 
		    sequence structures.
		    <P>
	   	    The UCSC team who worked on the Proteome Browser update
		    includes Fan Hsu, Robert Kuhn, Donna Karolchik and Tom 
		    Pringle. Please send comments or questions to our mailing 
		    list at 
		    <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>.
		    <P>
<HR>

<A NAME="100504"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 5 Oct. 2004 - East Coast Genome Browser Seminars - Nov. 2004</FONT><BR>
		    <P>
		    The UCSC Bioinformatics Group announces two seminars and 
		    hands-on workshops on the UCSC Genome Browser, presented by 
		    <A HREF="http://www.openhelix.com/" TARGET=_blank>OpenHelix</A>, 
		    a bioinformatics training, software testing and consulting 
		    company.
		    <P>
		    These introductory sessions are geared towards anyone with a 
		    basic knowledge of genomic and biological concepts who is 
		    interested in learning how to use the UCSC Genome Browser. 
		    No programming experience is required. The seminars will
		    cover the topics necessary to learn how to effectively use 
		    the browser tool set, including basic Genome Browser 
		    functionality, searching and BLAT use, Table Browser use, 
		    creating and using custom annotation tracks, and an 
		    introduction to the Gene Sorter. Lectures will be 
		    accompanied by hands-on computer exercises conducted directly 
		    on the Genome Browser web site.
		    <P>
		    The first three-hour course will be held on Tuesday 9 November
		    in the Washington, D.C. area.  Two sessions will be 
		    offered: 9am-12pm and 1pm-4pm. 
		    <P>
		    The second seminar will be held in the Raleigh/Durham, N.C.
		    area on Wednesday, 10 November.  Two sessions 
		    will be offered: 9am-12pm and 1pm-4pm. 
		    <P>
		    For registration information, visit the 
		    <A HREF="http://www.openhelix.com/noteworthy.shtml" 
		    TARGET=_blank>OpenHelix</A> website or call 1-888-861-5051. 
		    Academic, student, and early registration discounts are 
		    available.
		    <P>
<HR>

<A NAME="100104"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 1 Oct. 2004 - UCSC Releases New Table Browser</FONT><BR>
		    <P>
		    We are pleased to announce the release of an enhanced version
		    of the <A HREF="../cgi-bin/hgTables">UCSC Table Browser</A>. 
		    This new release sports several improvements and additions
		    over the previous Table Browser, including: 
		    <UL>
		    <LI> a new streamlined user interface 
		    <LI> support for generating filters that include fields from multiple tables, including those from non-positional tables
		    <LI> an enhanced schema-viewing utility that displays all 
		    tables associated with a track, as well as all tables linked 
		    to a selected table
		    <LI> the ability to restrict queries to include only data from ENCODE regions
		    <LI> the addition of a <A HREF="http://gala.cse.psu.edu">GALA</A> output option
		    <LI> an option to save large output results sets directly to a 
		    file rather than displaying them in the Table Browser
		    </UL>
		    <P>
		    The preliminary 
		    <A HREF="/goldenPath/help/hgTablesHelp.html">User's Guide</A>
		    will be enhanced in upcoming weeks to include examples of
		    many common Table Browser queries generated by our users.
	            <P>
	 	    The new version of the Table Browser was produced by
		    Jim Kent, Donna Karolchik, Heather Trumbower, Hiram Clawson,
		    and Robert Kuhn, and incorporates code from the 
		    previous version written by Angie Hinrichs.
		    We'd like to thank Mary Mangan and others on the 
	  	    OpenHelix staff for their feedback on early versions of this
		    software.
		    <P>
		    The older version of the Table Browser will remain available
		    for a limited time at 
		    <A HREF="../cgi-bin/hgText">http://genome.ucsc.edu/cgi-bin/hgText</A>. Please send feedback and questions to our
		    mailing list at <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>.
		    <P>
<HR>

<A NAME="091004"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 10 Sep. 2004 - <em>T. nigroviridis</em> Genome Now Available in Genome Browser</FONT><BR>
		    <P>
		    The Genoscope v7 <em>Tetraodon nigroviridis</em> genome 
		    assembly is now available in the UCSC Genome Browser and
	 	    Blat server.  This assembly, UCSC version tetNig1 dated
		    Feb. 2004, is the result of a collaboration between 
		    <A HREF="http://www.genoscope.cns.fr/" 
		    TARGET=_blank>Genoscope</A> and the 
		    <A HREF="http://www.broad.mit.edu/annotation/tetraodon/background.html" 
		    TARGET=_blank>Broad Institute</A> of MIT and Harvard. 
		    <P>
		    The v7 assembly was constructed using the whole genome shotgun 
		    (WGS) approach, resulting in a sequence coverage of about 
		    7.9X. The assembly contains 45,609 contigs and 25,773 
		    scaffolds generated by the Arachne program and covers more 
		    than 90% of the genome. Additional linking data were used to 
		    build ultracontigs and to organize the assembly into 
		    chromosomes. Genoscope estimates the size of the Tetraodon 
		    genome to be about 385 Mb. 
		    <P>
		    Downloads of the tetNig1 data and annotations may be obtained 
		    from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/tetNig1/" 
		    TARGET=_blank>FTP server</A>  
		    or <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#tetraodon"
		    TARGET=_blank>Downloads</A> 
		    page.  These data have been freely 
		    provided by Genoscope before publication with 
		    <A HREF="/goldenPath/credits.html#tetraodon_credits" 
		    TARGET=_blank>specific conditions for use</A>. 
		    The initial set of annotation tracks were generated by 
		    Genoscope and the UCSC Bioinformatics Group based on 
		    data provided Genoscope. Tetraodon gene predictions
		    generated by Genoscope using 
		    <A HREF="http://www.sanger.ac.uk/Software/analysis/GAZE/"
		    TARGET=_blank>GAZE</A> will be available in the Genome 
		    Browser within a few weeks.
		    <P>
		    Many thanks to Genoscope and the Broad Institute of 
		    MIT and Harvard for this genome assembly.  The UCSC team who 
		    produced this browser are Rachel Harte, Robert Kuhn, 
		    Donna Karolchik, and the Genome Browser sysadmin team. 
		    See the
		    <A HREF="/goldenPath/credits.html#tetraodon_credits">Credits</A>
		    page for a detailed list of the organizations and individuals 
		    who contributed to this release. 
		    <P>
<HR>

<A NAME="090104"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 1 Sep. 2004 - Genome Browser Released for <em>D. pseudoobscura</em> Draft Assembly</FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics Group has released a Genome 
		    Browser and Blat server for the <em>Drosophila 
		    pseudoobscura</em> Freeze 1 draft assembly (Aug. 2003).
		    This assembly, UCSC version dp2, was 
		    produced by the Human Genome Sequencing Center (HGSC) at 
		    Baylor College of Medicine. 
		    <P>
		    Freeze 1 is a whole genome shotgun assembly produced using 
		    Baylor HGSC's assembly engine, Atlas. The assembly, which 
		    provides approximately 7x coverage of the euchromatic portion 
		    of the genome, contains 759 scaffolds. The scaffold N50 size 
		    is 1,018,646 bp. The total scaffold size for this assembly 
		    is 139.3 Mbp, with an average size of 184,465 bp. Due to an 
		    assembly error, four large scaffolds "jumped" chromosomes. 
		    These have been split into "A" and "B" parts in the 
		    downloadable assembly files. See the gateway page for more
		    information.
		    <P>
		    Baylor HGSC has provided a 
		    <A HREF="ftp://ftp.hgsc.bcm.tmc.edu/pub/data/Dpseudoobscura/fasta/D_pseudo_freeze1_scaffold_to_chromosome_assignments_8_28_03.txt" 
		    TARGET=_blank>putative chromosome assignment</A> 
		    for the majority of larger scaffolds (> 90% of unique 
		    sequence), based on conservation between the Muller elements. 
		    <P>     
		    Downloads of the dp2 data and annotations may be obtained 
		    from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/dp2/" 
		    TARGET=_blank>FTP server</A>  
		    or <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#dp">Downloads</A> 
		    page. The initial set of annotation tracks were generated by 
		    UCSC. 
		    <P>
		    Many thanks to the Baylor HGSC for providing the genome
		    assembly data. The UCSC team who produced this browser are 
		    Angie Hinrichs, Heather Trumbower, Robert Kuhn,
		    Donna Karolchik, and the Genome Browser sysadmin team. 
		    See the
		    <A HREF="/goldenPath/credits.html#dp_credits">Credits</A>
		    page for a detailed list of the organizations and individuals 
		    who contributed to this release. 
		    <P>
<HR>

<A NAME="083004"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 30 Aug. 2004 - Genome Browser Seminars - Seattle and San Francisco</FONT><BR>
		    <P>
		    The UCSC Bioinformatics Group announces 2 seminars and 
		    hands-on workshops on the UCSC Genome Browser, presented by 
		    <A HREF="http://www.openhelix.com/" TARGET=_blank>OpenHelix</A>, 
		    a bioinformatics training, software testing and consulting 
		    company.
		    <P>
		    These introductory sessions are geared towards anyone with a 
		    basic knowledge of genomic and biological concepts who is 
		    interested in learning how to use the UCSC Genome Browser. 
		    No programming experience is required. The seminars will
		    cover the topics necessary to learn how to effectively use 
		    the browser tool set, including basic Genome Browser 
		    functionality, searching and BLAT use, Table Browser use, 
		    creating and using custom annotation tracks, and an 
		    introduction to the Gene Sorter. Lectures will be 
		    accompanied by hands-on computer exercises conducted directly 
		    on the Genome Browser web site.
		    <P>
		    The first three-hour course will be held on Monday 4 October
		    in Kirkland, WA (Seattle area).  Two sessions will be 
		    offered: 9:00 a.m. - noon or 1 - 4 p.m. 
		    <P>
		    The second seminar will be held in Cupertino (San 
		    Francisco Bay Area) on Tuesday, 5 October.  Two sessions 
		    will be offered: 1 - 4 p.m. or 6 - 9 p.m. 
		    <P>
		    For registration information, visit the 
		    <A HREF="http://www.openhelix.com/noteworthy.shtml" 
		    TARGET=_blank>OpenHelix</A> website or call 1-888-861-5051. 
		    Academic, student, and early registration discounts are 
		    available.
		    <P>
<HR>

<A NAME="081004"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 10 Aug. 2004 - Genome Browser Released for <em>A. gambiae</em> Draft Genome</FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics Group has released a Genome 
		    Browser and Blat server for the <em>Anopheles gambiae</em> 
		    v. MOZ2 draft genome sequence (Feb. 2003). This assembly -- 
		    UCSC version anoGam1 -- was produced by the International 
		    Anopheles Genome Project and downloaded from Ensembl.
		    <P>
		    The MOZ2 assembly is a 10x whole genome shotgun assembly.  
		    The assembled portion of the genome is about 278 Mbp in 
		    length with a total of 8,987 unique scaffolds, the largest 
		    scaffold being 23.1 Mbp. Approximately 85% of the sequence 
		    has been assigned to chromosomal locations. Chromosome arms 
		    chr2L, chr2R, chr3L, chr3R, and
		    chrX are represented by 13, 49, 42, 28, and 10 large scaffolds
		    respectively. No scaffolds have yet been assigned to the Y 
		    chromosome.  The unassigned scaffolds, concatenated together 
		    in arbitrary order, can be found in the artificial unknown 
		    &quot;chromosome&quot; chrUn.
		    <P>     
		    For more information about the initial <em>A. gambaie</em> 
		    assembly, see Holt et al. (2002),
		    <A HREF="http://www.sciencemag.org/cgi/content/abstract/298/5591/129"
		    TARGET=_blank>The Genome Sequence of the Malaria Mosquito 
		    Anopheles gambiae</A>,
		    <em>Science</em> 2002 298:129-149.
		    <P>     
		    Downloads of the anoGam1 data and annotations may be obtained 
		    from the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/anoGam1/" 
		    TARGET=_blank>FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#anoGam">Downloads</A> 
		    page. The anoGam1 annotation tracks were generated by UCSC and
		    collaborators worldwide. 
		    <P>
		    Many thanks to the International Anopheles Genome 
		    Project and Ensembl for providing the sequence of this genome.
		    The UCSC team who produced this browser are Angie Hinrichs, 
		    Galt Barber, Donna Karolchik, and sysadmins Paul Tatarsky and 
		    Jorge Garcia.  See the
		    <A HREF="/goldenPath/credits.html#anoGam_credits">Credits</A>
		    page for a detailed list of the organizations and individuals 
		    who contributed to this release. 
		    <P>
<HR>

<A NAME="072904"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 29 Jul. 2004 - Zebrafish Genome Assembly Zv3 Now Available in Genome Browser</FONT><BR>
		    <P>
		    The Zv3 Zebrafish genome assembly (UCSC version danRer1) is 
		    now available on the UCSC Genome Browser and Blat server. 
		    This assembly was produced by The Wellcome Trust Sanger 
		    Institute, Hinxton, UK, in collaboration with the Max Planck 
		    Institute for Developmental Biology in Tuebingen, Germany, 
		    and the Netherlands Institute for Developmental Biology 
		    (Hubrecht Laboratory), Utrecht, The Netherlands. 
		    <P>
		    The Zv3 assembly consists of 1,459,115,486 bp in 58,339 
		    supercontigs, with a sequence coverage of approximately 
		    5.7X. This zebrafish assembly is the first to be tied to 
		    the FPC map: 1,083,447,588 bp (74%) of the sequence were 
		    mapped in this way. Please note that this is a preliminary 
		    assembly; a high level of misassembly is present due to 
		    polymorphisms in the DNA source.  
		    <P>
		    For more information about this assembly, see the Sanger 
		    Institute's  
		    <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/"
		    TARGET=_blank><em>Danio rerio</em> Sequencing Project</A> 
		    web page.
		    UCSC plans to release the Zv4 version of the zebrafish 
		    assembly on the Genome Browser in Fall '04.
		    <P>
		    Downloads of the Zebrafish data and annotations can be 
		    obtained from the UCSC 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/danRer1/" 
		    TARGET=_blank>FTP site</A> or
		    <A HREF=http://hgdownload.cse.ucsc.edu/downloads.html#zebrafish>Downloads
		    page</A>. The danRer1 annotation tracks were generated by 
		    UCSC and collaborators worldwide. See the 
		    <A HREF="/goldenPath/credits.html#zebrafish_credits">Credits</A>
		    page for a detailed list of the organizations and individuals 
		    who contributed to the success of this release.
		    <P>
		    We'd like to thank The Wellcome Trust Sanger Institute
		    and their collaborators for providing this assembly. A special
		    thanks to Yi Zhou, Anthony DiBiase and Leonard Zon from the 
		    Children's Hospital in Boston, MA, USA for their collaboration
		    on this release. The UCSC Zebrafish
		    Genome Browser team is Rachel Harte, Heather Trumbower, and
		    Donna Karolchik.
		    <P>
<HR>

<A NAME="072304b"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 23 Jul. 2004 - Latest Human Genome Reference Sequence Now in Browser</FONT><BR>
		    <P>
		    The latest human genome reference sequence (NCBI Build 35, 
		    May 2004) is now available as database hg17 in the UCSC 
		    Genome Browser and Blat server. This sequence was obtained
		    from NCBI and was produced by the International Human
		    Genome Sequencing Consortium.
		    <P>
		    Bulk downloads of the data are available via FTP at 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/hg17">ftp://hgdownload.cse.ucsc.edu/goldenPath/hg17</A> 
		    or through the Downloads link on this page. We recommend 
		    that you use FTP rather than HTML for the download of large 
		    or multiple files. 
		    <P>
		    We'd like to thank NCBI and the International Human Genome 
		    Sequencing Consortium for furnishing the data, and the UCSC 
		    team members who contributed to this release: Hiram Clawson, 
		    Terry Furey, Heather Trumbower, Robert Kuhn, Donna Karolchik, 
		    Kate Rosenbloom, Angie Hinrichs, Rachel Harte, Jim Kent and 
		    our sysadmin team Patrick Gavin, Jorge Garcia, 
		    and Paul Tatarsky. 
		    <P>
<HR>

<A NAME="072304a"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 23 Jul. 2004 - <em>D. yakuba</em> Added to Genome Browser</FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics Group has released a Genome
		    Browser and Blat server on a second species of fruitfly,
		    <em>D. yakuba</em>. The April 2004 Release 1.0 of this genome
		    (UCSC version droYak1) was sequenced and assembled by the 
		    Genome Sequencing Center, Washington University (WUSTL) 
		    School of Medicine in St. Louis. 
		    <P>
		    <em>D. yakuba</em> is closely related to the model organism,
		    <em>D. melanogaster</em>, with which it shared a common 
		    ancestor approximately 10 million years ago. 
		    The <em>D. yakuba</em> genome is largely alignable to the
		    <em>D. melanogaster</em> genome, but differs sufficiently to
		    offer an interesting study of sequence divergence between
		    the two species. <em>D. yakuba</em> occupies a critical 
		    intermediate position among several Drosophila species that 
		    will facilitate evolutionary studies among the fruitflies.
		    For information about the <em>D. yakuba</em> assembly 
		    and statistics, see the WUSTL Genome Sequencing Center
		    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Drosophila%20yakuba&GROUP=6"
		    TARGET=_blank><em>Drosophila yakuba</em></A> web page.
		    <P>
	  	    Downloads of the droYak1 data and annotations can be obtained
		    from the UCSC Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/droYak1/">FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#droYak">Downloads</A> page.
		    The droYak1 annotation tracks were generated by UCSC and
		    collaborators worldwide. 
		    <P>Thanks to the Genome Sequencing Center at WUSTL School of 
		    Medicine for providing the sequence and assembly of this 
		    genome.  The 
		    UCSC <em>D. yakuba</em> Genome Browser was produced by Angie 
		    Hinrichs, Michael Chalup, and Donna Karolchik. See the
		    <A HREF="/goldenPath/credits.html#droYak_credits">Credits</A>
		    page for a detailed list of the organizations and individuals 
		    who contributed to the success of this release.
		    <P>
<HR>

<A NAME="071604"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 16 Jul. 2004 - Genome Browser Released for Mouse Assembly NCBI Build 33</FONT><BR>
		    <P>
		    The latest mouse assembly -- Build 33 from NCBI (UCSC version
		    mm5) -- is now available via the UCSC Genome Browser and 
		    Blat server.  This assembly includes approximately 
		    2.6 gigabases of sequence. 
		    Chromosome 11 is finished in Build 33; the Sanger 
		    Institute has provided a corresponding agp file. 
		    The whole genome N50 for this assembly is 22.3 Mb, in comparison
		    to 17.7 Mb for the previous build. 
		    Please note: the UCSC mm5 assembly contains only the 
		    reference strain C57BL/6J. 
		    <P>
		    This assembly is a composite version in which phase 3 High 
		    Throughput Genome Sequence (HTGS) was merged with the
		    Mouse Genome Sequencing Consortium v3 Whole Genome Shotgun
		    Assembly (MGSCv3). The assembly was performed by NCBI using
		    a &quot;combined&quot; tiling path that was created 
		    automatically for the most part, but was manually curated 
		    in places. This facilitated the placement of finished 
		    sequence in the context of the MGSCv3 assembly. Draft 
		    sequence was not included in this build:  the slight 
		    increase in coverage gained by using this would have been
		    offset by the increase in build errors. 
		    <P>
		    More information about Build 33 will be available soon in 
		    the NCBI
		    <A HREF="http://www.ncbi.nlm.nih.gov/genome/seq/NCBIContigInfo.html"	
	            TARGET=_blank>assembly notes</A> and 
		    <A HREF="http://www.ncbi.nlm.nih.gov/mapview/stats/BuildStats.cgi?taxid=10090&build=33&ver=1"
		    TARGET=_blank>Build 33 statistics</A>. 
		    <P>
		    The mm5 sequence and annotation data may be downloaded from 
		    the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/mm5/">FTP 
		    server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/mm5/">Downloads</A> 
		    web page. The mm5 annotation tracks were generated by UCSC
		    and collaborators worldwide. 
		    <P>
		    We'd like to thank Deanna Church, Richa Agrawala, and
		    the Mouse Genome Sequencing Consortium
		    for this assembly. We'd also like to 
		    acknowledge the work of the UCSC mm5 team: Fan Hsu,
		    Hiram Clawson, Angie Hinrichs, Heather Trumbower, Mark
		    Diekhans, Donna Karolchik and our systems 
		    administrators Jorge Garcia, Patrick Gavin and Paul Tatarsky. 
		    <P>
<HR>

<A NAME="071504"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 15 Jul. 2004 - <em>C. intestinalis</em> v1.0 Released in Genome Browser</FONT><BR>
		    <P>
		    The v1.0 <em>C. intestinalis</em> draft assembly from the
		    US DOE Joint Genome Institute is now available for study 
		    using the UCSC Genome Browser and Blat server (UCSC database
		    ci1). 
		    <P>
		    The whole genome shotgun assembly was constructed with the 
		    JGI assembler (JAZZ) paired-end 
		    sequencing reads at a coverage of 8.2X. The draft contains 
	    	    116.7 million bp of nonrepetitive sequence in 2,501 scaffolds 
		    greater than 3 kb. 60 Mbp of this has been assembled into 
		    117 scaffolds longer than 190 Kbp, and 85% of the assembly 
		    (104.1 Mbp) is found in 905 scaffolds longer than 20 kb. The 
		    assembly, gene modeling and analysis were performed at the 
		    JGI. 
		    <P>
		    For more information about the ci1 assembly, see the JGI
		    <A HREF="http://genome.jgi-psf.org/ciona4/ciona4.info.html"
		    TARGET=_blank><em>C. intestinalis</em> project page</A>.
		    Additional information and an analysis of the euchromatic 
		    regions of this genome may be found in Dehal et al., 
		    <A HREF="http://www.sciencemag.org/cgi/content/full/298/5601/2157" 
		    TARGET=_blank>The Draft Genome of Ciona intestinalis: 
		    Insights into Chordate and Vertebrate Origins</A>. Science. 
		    2002 Dec 13;298(5601):2157-67. 
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/ci1/" 
		    TARGET=_blank>FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#ciona"
		    TARGET=_blank>Downloads page</A>. The ci1 annotation tracks 
		    were generated by UCSC and collaborators worldwide. See the 
		    Credits page for a detailed list of the organizations and 
		    individuals who contributed to this release. 
		    <P>
		    Many thanks to the JGI and their collaborators for providing
		    the v1.0 sequence and annotations.  The ci1 Genome Browser 
		    was produced by
		    Brian Raney, Galt Barber, Heather Trumbower, Robert Kuhn,
		    Donna Karolchik and the Genome Browser sysadmin team - 
		    Patrick Gavin, Jorge Garcia, and Paul Tatarsky. We'd also 
		    like to thank Tom Pringle for his technical input and Mark
		    Diekhans for his work on the incremental updates for this
		    release.
		    <P>
<HR>

<A NAME="071404"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 14 Jul. 2004 - Genome Browser Released for Dog Draft Assembly v1.0</FONT><BR>
		    <P>
		    UCSC has released a Genome Browser and Blat server on the
		    July 2004  v1.0 dog genome sequenced and assembled by the 
		    Broad Institute of MIT and Harvard and Agencourt Bioscience. 
		    The whole genome shotgun (WGS) sequence is 
		    based on 7.6X coverage of the dog genome, assuming a WGS
		    assembly size of 2.4 Gb. The assembly has an N50 contig 
		    length of 123 kb and an N50 supercontig length of 41.6 Mb.
		    <P>
		    The dog genome, which contains approximately 2.5 billion 
		    base pairs, is similar in size to the genomes of humans and
		    other mammals. The boxer breed was selected for the initial 
		    sequencing effort, based on the lower variation rate in its 
		    genome relative to other breeds. In addition to the boxer,
		    samples from nine other dog breeds, four wolves and a coyote 
		    are being used to generate an initial set of single 
		    nucleotide polymorphisms (SNPs) to facilitate disease studies.
		    The SNPs should be available soon from
		    <A HREF="http://www.ncbi.nlm.nih.gov/SNP/" 
		    TARGET =_blank>dbSNP</A>.
		    <P>
		    For more information about the dog draft assembly, see the 
		    <A HREF="http://www.genome.gov/12511476" TARGET=_blank>NHGRI 
		    press release</A>.
		    <P>
		    The dog sequence and annotation data can be downloaded from 
		    the UCSC Genome Browser
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/canFam1/">FTP server</A>
		    or <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#dog">downloads</A>
		    page. These data have 
		    <A HREF="/goldenPath/credits.html#dog_use">specific 
		    conditions for use</A>.
		    <P>
		    Many thanks to the Broad Institute of MIT and Harvard, NHGRI,
		    Agencourt Bioscience, Children's Hospital Oakland Research 
		    Institute, Centre National de la Recherche Scientifique,
		    North Carolina State University, and Fred Hutchinson Cancer 
		    Research Center for their contributions to the sequencing, 
		    assembly, and mapping efforts. The initial canFam1 
		    annotation track set, generated by the UCSC Genome
		    Bioinformatics Group, will soon be
		    supplemented by annotations from collaborators 
		    worldwide. See the 
		    <A HREF="/goldenPath/credits.html#dog_credits">credits</A> page
		    for a detailed list of the organizations and individuals who 
		    contributed to the success of this release.
		    <P>
<HR>

<A NAME="062404"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 24 Jun. 2004 - ENCODE Project Featured in Genome Browser</FONT><BR>
		    <P>
		    We'd like to announce the release of UCSC Genome Browser 
		    features tailored to the ENCODE project community, including 
		    an 
		    <A HREF="/encode/">ENCODE-specific page</A> to 
		    highlight 
		    the ENCODE contributors and their work, guidelines for data 
		    submission, and a list of specific links to ENCODE regions 
		    in the Genome Browser. 
		    <P>
		    The initial resources include sequences 
		    for the current human assemblies (hg16, hg15, hg13, and hg12),
		    sequence of the 
		    comparative species from <A HREF="http://www.nisc.nih.gov" 
		    TARGET=_blank>NISC</A>, tools for coordinate 
		    conversion between human assemblies, format descriptions for 
		    data submission, and contact information for help with 
	 	    submitting annotation data and analyses. Bulk downloads of 
		    the sequence and annotations may be obtained from the ENCODE 
		    Project 
		    <A HREF="../ENCODE/downloads/">Downloads</A> 
		    page. 
		    <P>
		    We'd like to thank NHGRI for their 
		    support of this project and the various contributors of 
		    annotations and analyses.
		    <P>
<HR>

<A NAME="061004"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 10 Jun. 2004 - Genome Browser Training Workshop - Boston, MA</FONT><BR>
		    <P>
		    The UCSC Bioinformatics Group announces a seminar and hands-on
		    workshop on the UCSC Genome Browser, presented by 
		    <A HREF="http://www.openhelix.com/" TARGET=_blank>OpenHelix</A>, 
		    a bioinformatics training, software testing and consulting 
		    company.
		    <P>
		    This introductory session is geared towards anyone with a 
		    basic knowledge of genomic and biological concepts who is 
		    interested in learning how to use the UCSC Genome Browser. 
		    No programming experience is required. The seminar will
		    cover the topics necessary to learn how to effectively use 
		    the browser tool set, including basic Genome Browser 
		    functionality, searching and BLAT use, Table Browser use, 
		    creating and using custom annotation tracks, and an 
		    introduction to the Gene Sorter. The lecture will be 
		    accompanied by hands-on computer exercises conducted directly 
		    on the Genome Browser web site.
		    <P>
		    The three-hour course will be held at Tufts University School 
		    of Medicine, 145 Harrison Street, Boston, MA, on Tuesday, 
		    August 10th.  Two sessions will be offered: 1 - 4 p.m. or 
		    6 - 9 p.m.  For registration information, visit the 
		    <A HREF="http://www.openhelix.com/noteworthy.shtml" 
		    TARGET=_blank>OpenHelix</A> website or call 1-888-861-5051. 
		    Academic, student, and early registration discounts are 
		    available.
		    <P>
<HR>

<A NAME="052804"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 28 May 2004 - UCSC Family Browser Name Changed to UCSC Gene Sorter</FONT><BR>
		    <P>
		    What's in a name? In an effort to clarify the role of the 
		    UCSC Family Browser, we have changed its name to the UCSC 
		    Gene Sorter. We think this name better describes this tool, 
		    which lets the user collect information on groups of genes 
		    that may be related in many different ways. The Gene Sorter
		    provides a wealth of information on gene expression, protein 
		    homology (both within and across species), GO terms, and 
		    Pfam domains, cross links to many other databases, and much 
		    more.
		    <P>
		    If you haven't already tried this tool, we encourage you to 
		    give it a spin.  You'll find it at 
		    <A HREF="../cgi-bin/hgNear">http://genome.ucsc.edu/cgi-bin/hgNear</A>,
		    or click the "Gene Sorter" link on any Genome Browser menu 
		    bar.
		    <P>
<HR>

<A NAME="052704"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 27 May 2004 - New Genome Browser Downloads Server</FONT><BR>
		    <P>
		    As a follow-up to last week's FTP site switch, we are changing
		    the location of the UCSC Genome Browser downloads site to 
		    <A HREF="http://hgdownload.cse.ucsc.edu/">http://hgdownload.cse.ucsc.edu/</A>. 
		    All downloadable files currently located in 
		    http://genome.ucsc.edu/goldenPath
		    will be moved to the new server.
		    Please make a note of the new URL and update any references 
		    to it. Users accessing downloads through the Genome Browser
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">Downloads page</A> 
		    will be redirected automatically to the new location.
		    <P>
<HR>

<A NAME="051904"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 19 May 2004 - New Genome Browser FTP Server</FONT><BR>
		    <P>
		    We have changed the URL for the UCSC Genome Browser ftp site 
		    to <A HREF="ftp://hgdownload.cse.ucsc.edu/">ftp://hgdownload.cse.ucsc.edu/</A>. 
		    This replaces the old URL of ftp://genome.ucsc.edu/. 
		    The old URL will be disabled within a few days.
		    Please make a note of the new URL and update any references 
		    to it.
		    <P>
<HR>

<A NAME="051104"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 11 May 2004 - <em>C. elegans</em> WS120 Added to Genome Browser</FONT><BR>
		    <P>
		    UCSC has released a Genome Browser and Blat server on an 
		    updated version of the <em>C. elegans</em> genome. The
		    March 2004 assembly -- UCSC version ce2 -- is based on 
		    sequence version WS120 deposited into 
		    <A HREF="http://www.wormbase.org/about/about_Celegans.html" TARGET=_blank>WormBase</A> 
		    as of 1 March 2004. This assembly has a finishing error rate 
		    of 1:10,000.  
		    <P>
		    The ce2 sequence and annotation data may be downloaded from 
		    the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/ce2/">FTP server</A>
		    or <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#c_elegans">Downloads</A> web page.
		    The ce2 annotation tracks were generated by UCSC and 
		    collaborators worldwide. 
		    <P>
		    We'd like to thank the 
		    <A HREF="http://genome.wustl.edu/" TARGET=_blank>Genome Sequencing Center</A> 
		    at Washington University in St. Louis and the 
		    <A HREF="http://www.sanger.ac.uk/" TARGET=_blank>Sanger Institute</A> 
		    for their collaborative work in sequencing the 
		    <em>C. elegans</em> genome. Many thanks to the 
		    <A HREF="http://www.wormbase.org/wiki/index.php/WormBase_Consortium" 
		    TARGET=_blank>WormBase consortium</A> for making the worm 
		    sequence publicly available. We'd also like to acknowledge 
		    the UCSC team who contributed to this release: Rachel Harte
		    (lead engineer), Hiram Clawson (WABA and miRNA annotations), 
		    Mike Chalup (QA), Galt Barber (QA), Heather Trumbower (QA), 
		    and Donna Karolchik (documentation). 
		    <P>
<HR>

<A NAME="042204"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 22 Apr. 2004 - Mouse Genome Assembly Added to Proteome Browser</FONT><BR>
		    <P>
		    Proteome Browser functionality is now available on the 
		    Oct. 2003 mouse genome assembly (mm4). Protein
		    information may be viewed for most genes in the Known Genes 
		    track by clicking the Proteome Browser link on the gene's 
		    details page.
		    For more information on the UCSC Proteome Browser, see the
		    news release dated <A HREF="newsarch.html#031004">10 March 2004</A>.
		    <P>
		    In conjunction with this release, the Known Genes and Gene
		    Family Browser protein data have been updated to the
		    Swiss-Prot version dated 15 March 2004.
		    <P>
<HR>

<A NAME="041604"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 16 Apr. 2004 - Genome Browser Training - Silicon Valley Area</FONT><BR>
		    <P>
		    The UCSC Bioinformatics Group announces a seminar and 
		    hands-on workshop on the UCSC Genome Browser, presented by 
		    <A HREF="http://www.openhelix.com/" TARGET=_blank>OpenHelix</A>, 
		    a bioinformatics training, 
		    software testing and consulting company.  
		    <P>
		    This introductory session is geared towards industry and
		    academic biologists engaged in genomics research. No
		    programming experience is required. The seminar will cover 
		    the topics necessary 
		    to learn how to effectively use the browser tool set, 
		    including basic Genome Browser functionality, searching and 
		    BLAT use, Table Browser use, creating and using Custom Tracks,
		    and an introduction to the Family Browser. The lecture will
		    be accompanied by hands-on computer exercises conducted directly on the Genome Browser web site. 
		    <P>
		    The three-hour course will be held at the UCSC 
		    Extension Campus computer lab in Cupertino, CA on Thursday, 
		    May 6th 6-9 p.m. For pricing information or to reserve a
		    seat in the class, visit the 
		    <A HREF="http://www.openhelix.com/noteworthy.shtml"
		    TARGET=_blank>OpenHelix</A> web site or call 1-888-861-5051.
		    Pre-registration is required. Academic, student, and early 
		    registration discounts are available. 
		    <P>
<HR>

<A NAME="041204"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 12 Apr. 2004 - GNF Gene Expression Atlas 2 Data Now Available</FONT><BR>
		    <P>
		    Expression data from the GNF Gene Expression Atlas 2 are now 
		    available on the July 2003 human genome assembly on the UCSC
		    web site. The data may be viewed graphically in the 
		    <A HREF="../cgi-bin/hgNear?db=hg16&near.order=expGnfAtlas2&near.col.gnfHumanAtlas2Id.vis=on">Family Browser</A> or via
		    the GNF Atlas 2 track in the Genome Browser. The track data
		    contain 2 replicates each of 61 mouse tissues and 79 human 
		    tissues run over Affymetrix microarrays. 
		    <P>
		    We'd like to thank the Genomics Institute of the Novartis
		    Research Foundation (GNF) for providing the expression data
		    underlying the browser displays. More information on the data
		    will be available in the paper Su et al.  "A gene 
		    atlas of the mouse and human protein-encoding transcriptomes"
		    (in press - <em>PNAS</em>). 
		    <P>
<HR>

<A NAME="031004"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 10 Mar. 2004 - Announcing the UCSC Proteome Browser</FONT><BR>
		    <P>
		    We are proud to announce a new addition to the
		    UCSC family of genome browsing and analysis tools.  The UCSC Proteome 
		    Browser presents a rich set of useful protein properties as well as 
		    links to several protein and genomic data sources 
		    on the Web. For the first time, Genome Browser users can have 
		    both the genome and proteome worlds at their fingertips 
		    simultaneously. The browser is accessible from the Genome
		    Browser via the &quot;Proteome 
		    Browser&quot; link on the details page of any gene in the 
		    Known Genes track. The initial release is available only on
		    Human Build 34 (hg16); Proteome Browsers for the latest mouse
		    and rat assemblies will follow.
		    <P>
		    For each protein, the browser displays the corresponding 
		    genomic exon structure and its amino acid sequence. 
		    Several protein property tracks are aligned to the sequence 
		    to help a user pinpoint regions of interest. 
		    Additional properties are plotted with histograms against 
		    genome-wide protein data to 
		    highlight significant trends and anomalies. 
		    <P>
		    The Proteome Browser is tightly coupled with the UCSC Genome
		    Browser and UCSC Gene Family Browser, allowing easy navigation 
		    among the tools. For example, clicking on an exon in the
		    Proteome Browser tracks display brings up the Genome Browser tracks
		    page showing 
		    the genomic region of the exon together with a wealth of
		    relevant data. Similarly, clicking on the Proteome Browser's 
		    &quot;Family Browser&quot; link
		    displays related gene family information. 
		    <P>
		    The v1.0 release of the browser offers a variety of 
		    data tracks, including amino acid and DNA sequence, exon 
		    boundaries, hydrophobicity, polarity, cysteine and predicted 
		    glycosylation sites, Superfamily/SCOP domains, and amino acid 
		    anomalies. In addition, the browser includes histograms of 
		    several properties on a genome-wide scale: pI, molecular 
		    weight, exon count, number of cysteines, InterPro domain 
		    counts, hydrophobicity, amino acid frequencies and anomalies.
		    <P>
		    The Proteome Browser also provides links to a variety of
		    external sites containing supplementary information on the protein,
		    including SwissProt, InterPro and 
		    Pfam domains, 3-D structures at PDB and UCSF ModBase, and 
		    pathway maps of KEGG, BioCarta (CGAP), and BioCyc.
		    <P>
		    We'd like to thank SwissProt for sharing their high quality 
		    protein data and the pI calculation algorithm, as well as the other
		    external data sites linked to by the Proteome Browser.
		    We'd also like to acknowledge the hard work of Fan Hsu, lead 
		    engineer on the project, and Jim Kent, Tom Pringle,
		    Donna Karolchik, and Robert Kuhn. The project received
		    technical input, review and support from several other members of 
		    the UCSC Bioinformatics group.</P>

<HR>

<A NAME="030104"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 1 Mar. 2004 - Chicken Draft Genome Added to Genome Browser</FONT><BR>
		    <P>
	   	    We've added the chicken genome to the collection of assemblies available
		    in the UCSC Genome Browser and Blat Server.
		    The Feb. 2004 assembly (UCSC version galGal2) was produced by 
	 	    the Genome Sequencing Center at the Washington University 
		    School of Medicine in St. Louis. The source of
		    this sequence was a female inbred Red Jungle Fowl (<em>Gallus 
		    gallus</em>), the ancestor of domestic chickens. The chicken
		    genome is the first of the avian genomes to be sequenced.
		    <P>
		    The genome has been sequenced 
		    to 6.63X coverage. Approximately 88% of the sequence has been 
		    anchored to chromosomes, which include autosomes 1-24, 26-28,
		    and 32, and sex chromosomes W and Z. (In contrast to mammals,
		    the female chicken is heterogametic (ZW) and the male is 
		    homogametic (ZZ).) The remaining unanchored
	            contigs have been concatenated into the virtual chromosome 
		    &quot;chrUn&quot;, separated by gaps of 10,000 bp. The 
		    chicken mitochondrial sequence is also available as the 
		    virtual chromosome &quot;chrM&quot;. 
		    <P>
		    Washington University School of Medicine in St. Louis
		    created the physical map for this release. Genetic 
		    mapping and linkage analysis were produced through a 
		    collaborative effort led by Martien Groenen at Wageningen 
		    University in the Netherlands.
		    SNP data based on three strains of domestic
		    chickens will soon be available in GenBank from an 
		    international team of scientists led by the Beijing Genomics 
		    Institute in China and supported by the Wellcome Trust in 
		    Britain. 
		    <P>
		    The chicken is considered to be the premier non-mammalian
		    vertebrate model organism. It plays an important role in 
		    the research of viruses and cancer, and is a primary 
		    model for the study of embryology and development. From an 
		    evolutionary standpoint, the chicken's position
		    provides a good intermediate data point between mouse and
		    fugu. Comparative genomics analyses between the chicken and
		    other sequenced organisms should yield valuable
		    information on the evolution of gene order and 
		    arrangement, thus improving our understanding of the
		    structure and function of genes.
		    <P>
		    To facilitate comparative genomics studies, 
		    alignments of the chicken sequence to the human genome 
		    will be available in the Genome Browser later this week. 
		    Downloads of the comparative data are currently 
		    available through the Downloads page (see below).
		    <P>
		    For more information about the release of the chicken genome 
		    assembly, see the NHGRI 
		    <A HREF="http://www.nhgri.nih.gov/11510730" 
		    TARGET=_blank>press release</A>. 
		    Additional background on the rationale behind the chicken genome
		    sequencing effort can be found in the  
		    <A HREF="http://www.genome.wustl.edu/ancillary/data/whitepapers/Gallus_gallus_WP.pdf" 	
		    TARGET=_blank>sequencing proposal</A>.
		    <P>
		    Bulk downloads of the chicken sequence and annotations may be obtained from 
		    the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/galGal2/">FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#chicken">Downloads</A> page. These data have
		    <A HREF="/goldenPath/credits.html#chicken_use">specific conditions for use</A>.
		    <P>
                    We'd like to thank the Genome Sequencing Center at the Washington University
		    School of Medicine in St. Louis, Wageningen University, and 
		    the Chicken Mapping Consortium for providing these data. 
                    The chicken browser annotation tracks were generated by UCSC and 
		    collaborators worldwide. See the 
		    <A HREF="/goldenPath/credits.html#chicken_credits">Credits</A> 
		    page for a detailed list of acknowledgements. The UCSC Chicken Genome Browser
		    was produced by Angie Hinrichs, Heather Trumbower, Rachel Harte, and Donna
	   	    Karolchik.
		    <P></P>
<HR>

<A NAME="022304"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 23 Feb. 2004 - Chimpanzee Genome Browser Released</FONT><BR>
		    <P>
		    We are happy to announce the release of a Genome Browser and Blat server for the
		    chimpanzee (<EM>Pan troglodytes</EM>). 
		    The 13 Nov. 2003 Arachne assembly -- labeled Chimp Build 1
		    Version 1 (UCSC version panTro1) -- was produced by the 
		    Chimpanzee Genome Sequencing 
		    Consortium. 
		    <P>
		    This assembly covers 
		    about 95 percent of the genome and is based on 4X sequence coverage.
	  	    It is composed of 361,782 contigs with an N50 length of 15.7 kb, and 		    37,849 supercontigs having an N50 length of 8.6 Mb (not including 
		    gaps). The total contig length is 2.73 Gb, spanning 3.02 Gb.
		    The DNA donor for this genome assembly, &quot;Clint&quot;, 
		    is a captive-born West African chimpanzee (<em>Pan troglodytes 
		    verus</em>). 
		    <A HREF="http://www.genome.gov/Pages/Research/Sequencing/SeqProposals/ChimpGenome2.pdf" 
		    TARGET=_blank>Background information</A> on the chimp genome
		    sequencing project and the <A HREF="http://www.genome.gov/11509418"
		    TARGET=_blank>initial news release</A> about the chimp 
		    assembly can be found on the NHGRI website.
		    <P>
		    Research has indicated that
		    the human and chimp genomes probably differ by approximately one
		    percent. 
		    Because of this close relationship between chimpanzees and 
		    humans, the assembly should facilitate comparative analyses 
		    of the two genomes that have not been possible with other species that have been sequenced to date. 
		    <P>
		    The initial release of the Chimp Browser provides several 
		    annotation tracks comparing the chimp and human genomes.
		    More comparative annotations will be added in upcoming
		    weeks.
		    <P>
		    Bulk downloads of the chimp sequence and annotations may be obtained from the Genome 
		    Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/panTro1/">FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#chimp">Downloads</A> page. 
		    The complete set of sequence reads is available at the 
		    <A HREF="http://www.ncbi.nlm.nih.gov/Traces" 
		    TARGET=_blank>NCBI trace archive</A>.
		    Blat searches on chrUn_random are not supported in the initial
		    release, but will be available soon.
		    <P>
                    We'd like to thank NHGRI, the Eli & Edythe L. Broad Institute at 
		    MIT/Harvard, and Washington University at St. Louis School of 
		    Medicine for providing this sequence, and LaDeana Hillier, 
		    Washington University School of Medicine, and the
                    Broad Institute for their work on the alignments. The chimpanzee
		    browser annotation tracks were generated by UCSC and collaborators 
		    worldwide. 
                    <P>
                    The UCSC team who worked on this release consisted of
                    Kate Rosenbloom, Jim Kent, Hiram Clawson, Heather Trumbower, Robert
                    Kuhn, Donna Karolchik, and the Genome Browser sysadmin team.
		    <P></P>
		    <HR>

<A NAME="021204"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 12 Feb. 2004 - New UCSC-Supported Mirror Site</FONT><BR>
		    <P>
		    The Genome Browser project now has a UCSC-supported mirror site 
		    that may be used during power or network outages on the UCSC 
		    campus. The mirror -- which can be found at 
		    <A HREF="http://genome.brc.mcw.edu/" TARGET=_blank>http://genome.brc.mcw.edu/</A>
		    -- is located at the Medical College of Wisconsin in Milwaukee. 
		    The site will be updated regularly by UCSC with the latest data and 
		    software to closely replicate the main Genome Browser site at 
		    <A HREF="../index.html">http://genome.ucsc.edu</A>.  
		    Please continue to use the UCSC-based 
		    site for routine Genome Browser and Blat access.
		    <P>
		    We'd like to thank the Department of Physiology at the Medical 
		    College of Wisconsin -- and in particular Jeff Nie and Greg 
		    McQuestion -- for their resources and collaboration on this 
		    project. 
		    We'd also like to acknowledge the hard work of UCSC's Paul 
		    Tatarsky, who invested many hours in arranging the collaboration 
		    and setting up the mirror.
		    <P></P>
		    <HR>

<A NAME="011604"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 16 Jan. 2004 - Eight hg16 Tables Updated</FONT><BR>
		    <P>
		    We've discovered a handful of hg16 chrN_random_gap 
		    and chrN_random_gold tables on our public server that are 
		    out of date. We have replaced the following tables with 
		    updated versions:
		    <UL>
		    <LI>chr4_random_gap
		    <LI>chr4_random_gold
		    <LI>chr8_random_gap
		    <LI>chr8_random_gold
		    <LI>chrX_random_gap
		    <LI>chrX_random_gold
		    <LI>chrUn_random_gap
		    <LI>chrUn_random_gold
		    </UL>
		    <P>
		    Many thanks to Grigoriy Kryukov for discovering this 
		    problem. We apologize for any inconvenience this may have 
		    caused to our users.
		    <P></P>
		    <HR>

<A NAME="011404"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 14 Jan. 2004 - Yeast Genome Browser Released</FONT><BR>
		    <P>
		    We are proud to add yeast
		    (<em>S. cerevisiae</em>) to our growing list
		    of genome assemblies. The study of brewer's yeast, the most
		    basic eukaryotic model system, has led to important discoveries 
		    in a wide variety of areas, including metabolism, centromeres, 
		    recombination, cell division control, meiosis and splicing.
		    <P>
		    This assembly (UCSC version sacCer1) is based on sequence dated 1 Oct. 2003 in the 
		    <A HREF="http://www.yeastgenome.org/" TARGET=_blank>Saccharomyces Genome 
		    Database</A> (SGD). The sequence, open reading frame (ORF), and gene annotations 
		    were downloaded from the site 
		    <A HREF="ftp://genome-ftp.stanford.edu/pub/yeast/data_download" 
		    TARGET=_blank>ftp://genome-ftp.stanford.edu/pub/yeast/data_download</A>. 
		    <P>
		    The S288C strain was used in this sequencing project. Reference information for 
		    each chromosome may be found in the SGD 
		    <A HREF="http://www.yeastgenome.org/chromosomes.shtml" TARGET=_blank>Systematic 
		    Sequencing Table</A>. For more information about the yeast genetic and physical 
		    maps, see the paper Cherry JM et al. 
		    <A HREF="http://db.yeastgenome.org/cgi-bin/SGD/reference/reference.pl?refNo=22828"
		    TARGET=_blank>Genetic and physical maps of Saccharomyces cerevisiae</A>. 
		    <em>Nature</em> 1997 387(6632 Suppl):67-73. 
		    <P>
		    Downloads of the yeast data and annotations may be obtained from the UCSC Genome 
		    Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/sacCer1/">FTP server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads#yeast">Downloads</A> page. 
		    <P>
		    We'd like to thank Stanford University, the SGD, the University of California 
		    San Francisco (UCSF), Washington University in St. Louis, and the Eli & Edythe 
		    L. Broad Institute at MIT/Harvard for providing the data and annotations for 
		    this assembly. We'd also like to acknowledge the UCSC team who worked on this
		    release: Jim Kent, Heather Trumbower, Robert Kuhn, Donna Karolchik, and our
		    sysadmin team.
		    <P></P>
		    <HR>

<!---------------- 2003 archived news --------------->
<A NAME="2003"></A>
<A NAME="121003"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 10 Dec. 2003 - UCSC Releases Alignments of the Chimp Draft Assembly to Human Genome</FONT><BR>
		    <P>
                    UCSC has released alignments of the Nov. 2003 
		    chimpanzee
		    draft assembly to the July 2003 human 
		    assembly in the Genome Browser. These alignments may be 
		    viewed on the 
		    <A HREF="../cgi-bin/hgTracks?org=Human&db=hg16&position=chr4:56214201-56291736">Human July 2003</A> 
		    assembly. This release coincides with today's 
		    <A HREF="http://www.nhgri.nih.gov/11509418" TARGET=_blank>announcement</A>
		    by the National Human Genome Research Institute (NHGRI) of the 
		    first draft assembly of the chimpanzee genome. 
		    <P>
		    The set of human/chimpanzee alignments consists of a 
                    reciprocal best-in-genome net track and a 
                    chimp chain track.
		    These alignments were generated using
		    the blastz program developed at Pennsylvania State 
		    University and the programs blat, axtChain, chainNet, and netSyntenic 
		    developed at UCSC by Jim Kent. 
	 	    Research scientists should find these tracks useful for locating 
		    orthologous regions and studying genome rearrangement in the 
		    two species. 
		    <P> 
		    For more information about the alignment tracks, refer to
		    the track description pages. The tables may be downloaded from the 
		    Genome Browser FTP server's 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/hg16/database/">hg16 database
		    directory</A>. The chimp sequence and alignment data are downloadable 
		    from the 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/hg16/vsPt0/">hg16 human/chimp 
		    alignments directory</A>.  
		    <P>
		    The chimp sequence used in these alignments was obtained 
		    from the 13 Nov. 2003 Arachne assembly. We'd like to thank 
		    NHGRI, the Eli & Edythe L. Broad Institute at MIT/Harvard, 
		    and Washington University School of Medicine for providing 
		    this sequence, and LaDeana Hillier, Washington University School
		    of Medicine, and the Whitehead Institute for their work on the
		    alignments. We'd also like to acknowledge the members
		    of the UCSC team who contributed to the release of these 
		    alignments in the Genome Browser: Jim Kent, Kate Rosenbloom, 
		    Heather Trumbower, and Donna Karolchik.
		    <P></P>
		    <HR>

<A NAME="112403"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 24 Nov. 2003 - Genome Browser Released for Mouse Assembly NCBI Build 32</FONT><BR>
		    <P>
		    We have released a Genome Browser and Blat server for the 
		    latest mouse genome assembly, NCBI Build 32 
		    (UCSC v. mm4).  Build 32 is a composite 
		    assembly in which chromosomes were assembled by two 
		    slightly different algorithms depending on the available 
		    mapping data. Chromosomes 2, 4, 5, 7, 11, 15, 18, 19, X, 
		    and Y were assembled using a clone-based tiling path file 
		    (TPF) provided by the Mouse Genome Sequencing Consortium 
		    (MGSC), with whole genome shotgun sequence used to fill gaps 
		    when necessary. The remaining chromosomes were assembled 
		    using the MGSCv3 whole genome shotgun assembly as the TPF 
		    and merging High Throughput Genomic Sequence (HTGS) as 
		    needed. The UCSC mm4 assembly contains only the reference 
		    strain C57BL/6J. 
		    <P>
		    Build 32 includes 2.6 gigabases of sequence, 1.2 Gb of which is finished. We 
		    estimate that 90-96 percent of the mouse genome is present 
		    in the assembly. For more information about this version, 
		    see the NCBI 
		    <A HREF="http://www.ncbi.nlm.nih.gov/genome/seq/NCBIContigInfo.html" TARGET=_blank>assembly 
		    notes</A> and 
		    <A HREF="http://www.ncbi.nlm.nih.gov/mapview/stats/BuildStats.cgi?taxid=10090&build=32&ver=1" TARGET=_blank>Build 32 statistics</A>. 
		    <P>
		    The mm4 sequence and annotation data may be downloaded from 
		    the UCSC Genome Browser 
		    <A HREF="http://genome-archive.cse.ucsc.edu/downloads.html">downloads
		    page</A>.
		    <P>
		    We'd like to thank the Deanna Church, Richa Agrawala, and
		    the Mouse Genome Sequencing Consortium
		    for this assembly. We'd also like to 
		    acknowledge the work of the UCSC mm4 team: Hiram Clawson 
		    (lead),
		    Terry Furey, Kate Rosenbloom, Heather Trumbower, Bob Kuhn 
		    and Donna Karolchik, and our systems administrators Patrick 
		    Gavin, Jorge Garcia and Paul Tatarsky.
		    <P></P>
<HR>

<A NAME="103103"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 31 Oct. 2003 - <em>D. melanogaster</em> Assembly Added to Genome Browser</FONT><BR>
		    <P>
		    We have added the <EM>Drosophila melanogaster</EM> (fruitfly) 
		    assembly to the growing collection of genomes available in 
		    the UCSC Genome Browser and Blat servers.  Release 3.1 
		    (Jan. 2003) of the <EM>Drosophila</EM> annotated genome 
		    sequence was provided by the
		    <a HREF="http://www.fruitfly.org/" TARGET=_blank>Berkeley 
		    Drosophila Genome Project</a> (BDGP). The 116.8 Mb euchromatic 
		    sequence - which is virtually gap-free and of high accuracy - 
		    contains six euchromatic chromosome arms represented by 13 
		    scaffolds with a total of 37 sequence gaps. The sequence quality 
		    of this release has an estimated error rate of less than one 
		    in 100,000 base pairs in the unique portion of the sequence, 
		    and less than one in 10,000 base pairs in the repetitive portion. 
		    The Release 3.1 sequence was reannotated using the 
		    <a HREF="http://www.fruitfly.org/annot/apollo/" TARGET=_blank>Apollo 
		    Genome Annotation and Curation Tool</a>.   We also provide data
		    comparing the genome of <EM>D. melanogaster</EM> with that of 
		    <EM>D. pseudoobscura</EM>.
		    <P>
		    The fruitfly, one of the first organisms to be used in systematic
		    scientific investigations, has been the subject of intensive study 
		    in genetics for nearly a century and remains a major model organism 
		    in biomedical research, population biology and evolution.  
	 	    We are pleased to add the fruitfly to the roster of assemblies 
		    available on our site.
	            <P>
		    Downloads of the <EM>Drosophila</EM> data and annotations may be
		    obtained from the UCSC Genome Browser 
		    <A HREF ="ftp://hgdownload.cse.ucsc.edu/goldenPath/dm1">ftp site</A>.
	            <P> We'd like to thank BDGP and the 
		    <A HREF = "http://flybase.bio.indiana.edu">Flybase</A> Consortium
		    (<A HREF="http://www.harvard.edu/" TARGET=_blank>Harvard University</A>,
                    <A HREF="http://www.cam.ac.uk/" TARGET=_blank>University of Cambridge</A>,
                    <A HREF="http://www.iub.edu/" TARGET=_blank>Indiana University</A>,
                    the <A HREF="http://www.berkeley.edu/" TARGET=_blank>University of 
		    California, Berkeley</A> and the <A HREF="http://www.ebi.ac.uk/index.html" 
		    TARGET=_blank>European Bioinformatics Institute</A> (EBI))
                    for providing the sequence, assembly, and analysis of this
		    genome. We'd also like to acknowledge the members of the 
		    UCSC Genome Bioinformatics group who contributed to this 
		    release: Angie Hinrichs (lead engineer), Heather Trumbower, 
		    Robert Kuhn, Donna Karolchik, and Jim Kent and the system
		    administrators Jorge Garcia, Patrick Gavin and Paul Tatarsky.
		    <P>
<HR>

<A NAME="101703b"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 17 Oct. 2003 - mRNA, RefSeq and EST Data Now Updated Incrementally for Several Assemblies</FONT><BR>
		    <P>
		    Daily and weekly incremental updates of mRNA, RefSeq, 
		    and EST data are now in place for several of the UCSC Genome 
		    Browser assemblies. Data sets that are updated incrementally 
		    from GenBank include the latest human (hg16), mouse (mm3), 
		    rat (rn3), and Fugu (fr1). Others will soon be added to the 
		    list. 
		    <P>
		    Previously, these tables were updated only when we loaded a 
		    new genome assembly into the Genome Browser or made a major 
		    revision to a table. By updating the data on a nightly basis, 
		    we are able to provide researchers with the most current 
		    versions available in GenBank. All new genome assemblies 
		    released after this date will incorporate the incremental 
		    update technology.
		    <P>
		    Data are updated on the following schedule:<BR>
		    - native and xeno mRNA and refSeq tracks - updated daily at 
		    4:30 p.m. Pacific Time (weekdays), early Saturday morning 
		    (weekends)<BR>
		    - EST data - updated once per week on Saturday morning<BR>
		    - downloadable data files - updated weekly on Sunday morning<BR>
		    - outdated sequences - removed once per quarter<BR>
		    <P>
		    Mirror sites are not required to migrate to an incremental 
		    update process, and should not experience problems as a 
		    result of this upgrade. Mirror site questions should be 
		    addressed to 
		    <A HREF="mailto:&#103;e&#110;&#111;m&#101;&#45;&#109;&#105;&#114;&#114;o&#114;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;&#117;">&#103;e&#110;&#111;m&#101;&#45;&#109;&#105;&#114;&#114;o&#114;&#64;&#115;&#111;e.
&#117;&#99;&#115;&#99;.&#101;&#100;&#117;</A>.
		    <!-- above address is genome-mirror at soe.ucsc.edu -->
		    <P>
		    We'd like to acknowledge the hard work of Mark Diekhans in the
		    implementation of this new feature, and thank the QA and 
	  	    sysadmin teams (particularly Paul Tatarsky) for their 
		    support in this release.
		    <P>
<HR>

<A NAME="101703a"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 17 Oct. 2003 - UCSC Table Browser Enhancements</FONT><BR>
		    <P>
		    The UCSC <A HREF="../cgi-bin/hgText">Table
		    Browser</A> is an excellent tool for retrieving and searching 
		    the data underlying the Genome Browser. We've recently added 
		    some new features to the Table Browser to make it even 
		    easier to query and download data.
		    <P>
		    Many of our users have requested a batch query utility that 
		    will allow them to paste in or upload a list of terms on 
		    which to search. You can now do this by clicking the "Item 
		    name/accession" button, then uploading a list of search 
		    terms by selecting the "Paste in" or "Upload" option. Note 
		    that the Paste option supports wildcards, but the Upload 
		    option does not.
		    <P>
		    We've also added 2 new lists of searchable tables/tracks. 
		    The Browser tracks list contains the names of annotation 
		    tracks in the currently selected assembly as they appear in 
		    the Genome Browser. This list is useful if you don't know 
		    the name of the underlying database table that contains the 
		    data in which you're interested. The Custom tracks list 
		    contains the names of all custom annotation tables currently 
		    loaded into the Genome Browser for the given assembly. This 
		    includes tracks that have been created/loaded by the user as 
		    well as custom annotations created via the Table Browser.
		    <P>
		    If you have feedback or questions about the Table 
		    Browser, please send us email at 
		    <A HREF="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</A>.
		    <!-- above address is genome at soe.ucsc.edu -->
		    <P>
<HR>

<A NAME="092903"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 29 Sep. 2003 - Introducing the UCSC Gene Family Browser</FONT><BR>
		    <P>
		    We are pleased to announce the release of the 
		    <A HREF="../cgi-bin/hgNear/" TARGET=_blank>UCSC 
		    Gene 
		    Family Browser</A>.  This major new addition to our website is 
		    a useful tool for collecting information on groups of genes 
		    that may be related in many different ways. The 
		    Browser provides information on gene expression, 
		    protein homology (both within and across species), GO terms, 
		    and Pfam domains, and cross links to many other databases. 
		    <P>
		    To access the Family Browser, click the link on the top
		    menu on this page. The first time you use the Browser, the
		    application will display a brief overview of the tool and
		    information for starting and configuring it. To read a more
		    detailed description of the Browser, see the Family Browser
		    <A HREF="/goldenPath/help/hgNearHelp" TARGET=_blank>User's 
		    Guide</A>.
		    <P>
		    We are always interested in hearing feedback
		    about the tools on our site. If you have comments or questions
		    about the Family Browser, please email us at 
		    <A HREF="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</A>.
		     <!-- above address is genome at soe.ucsc.edu -->
		    <P>

<HR>
<A NAME="092303"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 23 Sep. 2003 - Fugu Genome Added to Genome Browser</FONT><BR>
		    <P>We have added the <em>Takifugu rubripes</em> (Japanese pufferfish) assembly to 
		    the growing collection of genomes available in the UCSC Genome Browser
		    and Blat servers. The Fugu v.3.0 
		    (Aug. 2002) whole genome shotgun assembly -- 
		    which is the fourth
		    vertebrate assembly to be added to the UCSC Genome Browser -- was 
		    provided by 
		    the <A HREF="http://genome.jgi-psf.org/fugu6/fugu6.info.html" TARGET=_blank>US 
		    DOE Joint Genome Institute</A> (JGI) as part of the 
		    International Fugu Genome Consortium led by JGI and the 
		    <A HREF="http://www.fugu-sg.org" TARGET=_blank>
		    Singapore Institute of Molecular and Cell Biology (IMCB)</A>. 

		    <P>This assembly was 
		    constructed with the JGI assembler, JAZZ, from paired end 
		    sequencing reads produced by JGI and IMCB, at JGI, Myriad Genetics, and 
		    Celera Genomics, resulting in a sequence coverage of 5.7X. 
		    All reads are plasmid, cosmid, or BAC end sequences, with 
		    the predominant coverage derived from 2 Kb insert plasmids. 
		    This assembly contains 20,379 scaffolds totaling 319 million 
		    base pairs. The largest 679 scaffolds total 160 million base 
		    pairs. The Fugu genome was annotated using the Ensembl 
		    system by the Fugu informatics group at IMCB.
		    
		    <P>The Fugu, which was one of the first
		    vertebrate genomes to be draft-sequenced after human, 
		    serves an important role in the exploration of the human 
		    genome. In contrast 
		    to other vertebrates that have been sequenced, the intergenic
		    and intron regions of the Fugu are highly compressed and 
		    uncluttered with repetitive sequence, resulting in a 
		    genome that is unusually compact in size. The Fugu genome
		    has proved useful in gene discovery and the identification and
		    characterization of gene regulatory elements in other genomes.
		   
		    <P>Bulk downloads of the Fugu sequence and annotation data are 
		    available via FTP at
                    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/fr1">ftp://hgdownload.cse.ucsc.edu/goldenPath/fr1</A> 
        	    or through the Downloads link on the Genome Browser home page. We recommend that FTP be used rather than 
		    HTML for the download of large or multiple files. 
		    
		    <P>We'd like to thank JGI and the other members of the 
		    International Fugu Genome Consortium, including IMCB,
		    the UK Human Genome Mapping Project (Hinxton), 
		    the Molecular Sciences Institute (Berkeley) and the
		    Institute for Systems Biology (Seattle),
		    for providing the sequence, assembly, and analysis of this
		    genome. We'd also like to acknowledge the members of the 
		    UCSC Genome Bioinformatics group who contributed to this 
		    release: Kate Rosenbloom (lead engineer), Heather Trumbower, 
		    Robert Kuhn, Donna Karolchik, and Jim Kent.
		    <P>

<HR>

<A NAME="081303"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 13 Aug. 2003 - NISC Comparative Sequencing Program Regions Released in Genome Browser</FONT><BR>
		    <P>
		    The UCSC Genome Bioinformatics group has released a browser 
		    and blat server on the first of more than 100 targeted 
		    genomic regions being sequenced in multiple species and
		    analyzed by the NIH Intramural Sequencing Center (NISC) 
		    Comparative Sequencing Program sponsored by NHGRI. This
		    release coincides with the publication of the results of the
		    study in the 14 Aug 2003 issue of 
		    <A HREF="http://www.nature.com/cgi-taf/DynaPage.taf?file=/nature/journal/v424/n6950/abs/nature01858_fs.html" TARGET=_blank><em>Nature</em></A> 
		    (Thomas, 
		    J.W. et. al. (2003) Comparative analyses of multi-species 
		    sequences from targeted genomic regions. <em>Nature</em> 424:788- 793).
		    The browser displays sequence and annotations on a large 
		    region 
		    containing 10 previously identified genes - including the 
		    gene mutated in cystic fibrosis - in 13 vertebrate species. 
		    Organisms in the study include human, chimpanzee, 
		    baboon, cat, dog, cow, pig, rat, mouse, chicken, zebrafish 
		    and two species of pufferfish (Fugu and Tetraodon). 
		    <P>
		    The NISC Comparative Sequencing Program data may be accessed
		    by clicking the Browser link on the Genome Browser home page 
		    and then selecting the "Zoo" option from the genome list.
		    <P>
		    The research team, led by  NHGRI Scientific Director Eric 
		    D. Green, included scientists from Pennsylvania State 
		    University, University of California Santa Cruz (UCSC),
		    and the University of Washington in Seattle. In the study,
		    the investigators systematically compared the patterns of 
		    transposon insertions among the species' sequences. One
		    key result of the analysis was the confirmation of recently 
		    proposed mammalian evolutionary 
		    trees suggesting that primates are more closely related to 
		    rodents than to carnivores or artiodactyls. Another 
		    significant outcome was the discovery of a 
		    substantial number of previously unidentified non-coding DNA 
		    segments that are conserved across a wide range of species.
		    Many of these regions could be identified only through 
		    comparisons of sequence from multiple species, demonstrating
		    the importance of studying the genomes of a wide range of 
		    organisms as a means for identifying functional elements in
		    the human genome.
		    <P>
		    UCSC built a customized version of the browser to display the
		    target region for this study, allowing scientists to 
		    interactively explore the data and predictions generated 
		    by this project, contribute data of their own, and track 
		    the project as data from additional species are generated.
		    In addition to the browser, the UCSC team also 
		    contributed to the analytical portion of the 
		    study. Mathieu Blanchette
		    identified the regions that are most highly conserved among
		    species. Adam Siepel performed the phylogenetic analysis of
		    rates of substitution. The UCSC team worked with Arian Smit
		    to obtain definitive evidence that rodents branched off from 
		    the common ancestor later than carnivores and artiodactyls.
		    <P>
		    For more information on the NISC study, see the <em>Science 
		    Daily</em>
		    <A HREF="http://www.sciencedaily.com/releases/2003/08/030814072241.htm" TARGET=_blank>press release</A>. 
		    Flat files of the assembled sequence and annotations may be 
		    obtained from <A HREF="http://www.nisc.nih.gov/data/" TARGET=_blank>http://www.nisc.nih.gov/data/</A> or via the 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">Downloads</A> link on the Genome
		    Browser home page.
		    <P>
		    We'd like to thank the NISC Comparative Sequencing Program
		    team for providing the data and comparative analysis 
		    for this Genome Browser release. Special thanks go to
		    Elliott Margulies at NHGRI for serving as the main liaison
		    between NHGRI and UCSC, and for contributing several 
		    annotation tracks to the browser. We'd also like to 
		    acknowledge 
		    the efforts of the many faculty, grad students, and staff 
		    members of the UCSC Genome Bioinformatics group who 
		    contributed to the research effort and browser 
		    development for this project.
		    <P>

<HR>

<A NAME="080803"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 8 Aug. 2003 - New Human Genome Reference Sequence Now Available in Genome Browser</FONT><BR>
		    <P>
		    The latest human genome reference sequence (NCBI Build 34,
		    July 2003) is now available as database hg16 in the UCSC
		    Genome Browser and blat server.
		    <P>
		    There are 
		    2,843,433,602 finished sequenced bases in the ordered and 
		    oriented portion of the assembly, which is an increase of 
		    0.4 percent, or approximately 11 Mb, over the Build 33 
		    assembly. 
		    Of particular note in this release is the addition of the 
		    pseudoautosomal regions of the Y chromosome. This sequence 
		    was taken from the corresponding regions in the X chromosome 
		    and is an exact duplication of that sequence.
		    <P>
		    Some sequence joins between adjacent clones in this assembly
		    could not be computationally 
		    validated because the clones originated from different 
		    haplotypes and contained polymorphisms in the overlapping 
		    sequence, or the overlap was too small to be to be reliable. 
		    In these instances, the sequencing center responsible for 
		    the particular chromosome has provided data to support 
		    the join in the form of an electronic certificate. The 
		    Build 34 certificates may be reviewed 
		    <A HREF="certificates/certificate_hg16.html">here</A>.
		    <P>
		    Bulk downloads of the data are available via FTP at 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/hg16">ftp://hgdownload.cse.ucsc.edu/goldenPath/hg16</A> 
		    or through the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">Downloads</A> 
		    link.  We recommend that FTP be used rather 
		    than HTML for the download of large or multiple files.
		    <P>
		    We'd like to thank NCBI and the International Human Genome 
		    Sequencing Consortium for
		    furnishing the data, and the UCSC team members who 
		    contributed to this release: Terry Furey, Hiram Clawson, 
		    Heather Trumbower, Mark Diekhans, Robert Baertsch, 
		    Donna Karolchik, Jim Kent and our sysadmin team Patrick 
		    Gavin, Jorge Garcia, and Paul Tatarsky.
		    <P>

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 14 Jul. 2003 - Rat Genome v.3.1 Added to Genome Browser</FONT><BR>
		    <P>
                    The UCSC Genome Bioinformatics Group has released a browser
		    and BLAT server on the v. 3.1 rat genome assembly from the 
		    Rat Genome Sequencing Consortium. This
		    assembly (UCSC version rn3, June 2003) was produced by the 
		    Atlas group at <A HREF="http://hgsc.bcm.tmc.edu/projects/rat/" TARGET=_blank>Baylor Human Genome Sequencing Center</A> (HGSC). 
 		    This assembly is a minor update to the 3.0 release.
  		    Sequence changes affect only chromosomes 7 and X. No 
		    additional assembly releases are planned prior to the 
		    publication of the rat genome analysis papers. 
		    <P>
		    The 3.x assemblies reflect several sequence additions and 
		    software improvements over the previous 2.x assemblies, 
		    including the sequencing of over 1100 new BACs to cover gaps, 
 		    an improved marker set from the Medical College of Wisconsin, 
		    a new FPC map from the BC Genome Sciences Centre, and 
		    improved linking of bactigs. For detailed information and 
		    statistics about the 3.x assemblies, see the Baylor HGSC 
		    <A HREF="ftp://ftp.hgsc.bcm.tmc.edu/pub/analysis/rat/README"
		    TARGET=_blank>README</A>. 
		    <P>
		    Downloads of the rat sequence and annotation data are 
		    available at 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/rnJun2003/" TARGET=_blank>
		    ftp://hgdownload.cse.ucsc.edu/goldenPath/rnJun2003/</A> or via 
		    the Downloads link on this page. These data are made 
		    available with 
		    <A HREF="credits.html#rat_use">specific conditions for use</A>.
		    <P>
		    We'd like to thank the Rat Genome Sequencing Consortium and
		    Baylor HGSC for providing this assembly, collaborators
		    from other institutions who have contributed annotations, 
 		    and Arian Smit for updating RepeatMasker for this release.
		    We'd also like
		    to acknowledge the contributions of several individuals at
		    UCSC, including Hiram Clawson, Heather Trumbower, Robert Kuhn,
		    Yontao Lu, Terry Furey, Mark Diekhans, Robert Baertsch, 
		    Donna Karolchik, Jim Kent, and our sysadmin team Jorge 
		    Garcia, Patrick Gavin, and Paul Tatarsky.
		    <P>
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 24 Jun. 2003 - Genome Browsers Released for <em>C. elegans</em> and <em>C. briggsae</em></FONT><BR>
 	<P>
	UCSC has just released browsers and blat servers for 2 worms: 
	C. elegans version WS100 (May 2003) and C. briggsae version 
	cb25.agp8 (July 2002). The browsers are based on sequence 
	obtained from <A HREF="http://www.wormbase.org/" TARGET=_blank>WormBase</A>.
	<P>
	We are pleased to add the nematodes to the roster of genomes 
	available on our site. C. elegans is a major model organism 
        used for biomedical research, and is the first multicellular 
        animal to have a fully sequenced genome. In contrast,  the 
        whole genome shotgun assembly of the C. briggsae genome is 
        estimated to have achieved 98% coverage. Draft chromosome 
        sequences are not available for C. briggsae, due to the lack 
        of dense chromosomal maps that allow assignment of 
        ultracontigs to chromosomal locations.  As a result, all data 
        in the C. briggsae browser maps to chrUn.  
	<P>
	Both worms played a significant role in the early history
	of the UCSC Genome Browser. The browser code originated with 
        a C script that displayed a splicing diagram for a gene 
        prediction from C. elegans.  Tracks for mRNA alignments and 
        for homology with C. briggsae were added, and the tool 
        morphed into the precursor of the Genome Browser, the 
        &quot;Intronerator&quot; (Kent, WJ and Zahler, AM (2000). 
	<A HREF="http://nar.oupjournals.org/cgi/content/abstract/28/1/91?ijkey=155fc1582751ebf7b01072beb40325b1d0682fdc&keytype2=tf_ipsecsha" TARGET=_blank>
	The intronerator: Exploring introns and alternative splicing 
        in C. elegans</A>. <em>Nucleic Acids Res.</em> 28: 91-93). 
	<P>
	Downloads of the C. elegans sequence and annotation data are 
	available at 
	<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/ceMay2003/" TARGET=_blank>
	ftp://hgdownload.cse.ucsc.edu/goldenPath/ceMay2003/</A>; C. briggsae 
	downloads can be found at 
	<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/cbJul2002/" TARGET=_blank>
	ftp://hgdownload.cse.ucsc.edu/goldenPath/cbJul2002/</A>. 
	Both genomes can also be downloaded via the Downloads link on 
	this page.
	<P>
	We'd like to thank the 
	<A HREF="http://genome.wustl.edu/" TARGET=_blank>Genome Sequencing Center 
	at Washington University in St. Louis</A> and the 
	<A HREF="http://www.sanger.ac.uk/" TARGET=_blank>Sanger Institute</A> 
	for their collaborative work in sequencing the C. elegans 
        and C.  briggsae genomes. Many thanks to the 
	<A HREF="http://www.wormbase.org/about/people.html" TARGET=_blank>WormBase consortium</A> 
	for making the worm sequence publicly available.
	We'd also like to acknowledge several UCSC people who 
        contributed to this release: Hiram Clawson (browser and 
        annotation tracks engineering), Jim Kent (WABA and 
        chaining/netting), Al Zahler (WABA), Heather Trumbower (QA 
        and project management), and Donna Karolchik (project 
        management and documentation).
 	<P>

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 23 Jun. 2003 - Genome Browser UI Changes</FONT><BR>
 	<P>
	You may notice that we've removed the Genome pulldown menu 
	and genome assembly information from our home page. 
	Genome-specific information and links, as well as genome 
	selection, are now available on the gateway pages for our 
	tools. To open up a gateway page, simply click the Browser, 
	Blat, or Tables link in the left sidebar.</P>
 	<P>

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 23 May 2003 - SARS Coronavirus TOR2 Draft Assembly Available</FONT><BR>
 	<P>
	Today we'd like to announce the release of a genome browser and BLAT
	server for the SARS coronavirus TOR2 draft assembly. The browser - which is based on
	sequence deposited into GenBank as of 14 April 2003 - provides seven
	annotations showing gene predictions, locations of putative proteins, and
	viral mRNA and protein alignments. Of particular note are the protein 
	structure analysis and predictions, determined by using the 
	<A HREF="http://www.cse.ucsc.edu/research/compbio/sam.html" TARGET=_blank>Sequence Alignment and Modeling (SAM) T02</A> tool. </P>

	<P>
	This browser marks a departure from our usual collection of vertebrate 
	genomes. Its inception was inspired by one of our engineers - Angie 
	Hinrichs - who was vacationing in New Zealand when the SARs draft assembly
	was initially released. Struck by the impact of SARS in that part of the 
	world, she downloaded the sequence and built the initial tracks from a
	terminal at an Internet cafe! The rest of the team joined in on the 
	grassroots effort, generating the additional annotations and SAM 
	T02 protein analyses and predictions. Victor Solovyev chimed in with 
	Fgenesv+ gene predictions from Softberry Inc. UCSC does not intend to 
	provide a comprehensive collection of viral genomes in the future, but 
	will maintain this browser as long as scientific and public interest in 
	SARS persists.</P>

	<P>
	Downloads of the annotation data are available
	at <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/scApr2003/database" TARGET=_blank>
	ftp://hgdownload.cse.ucsc.edu/goldenPath/scApr2003/database</A> or via the Downloads 
	link on this page. </P>

	<P>
	We'd like to thank everyone who worked on this release, including
	Angie Hinrichs, Robert Baertsch, Fan Hsu, Matt Schwartz, Heather Trumbower, Jim
	Kent, Kevin Karplus, Donna Karolchik, Brian Raney, Hiram Clawson, Kate 
	Rosenbloom, Victor Solovyev, and 
	our extremely dedicated systems administrators Paul Tatarsky, Patrick 
	Gavin, and Jorge Garcia.</P>
 	<P>

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 21 Apr. 2003 - Chromosome 8 Data Corrected for April 2003 Human Assembly</FONT><BR>
 	<P>
	The file that we originally used to build the agp files for the 
	April 2003 human release (Build 33) erroneously contained 2 contigs on 
	chromosome 8 that were listed twice: NT_078037, NT_008183.  We've 
	received a corrected version and have updated the following files on our 
	website: contigAgp.zip, chromAgp.zip, liftAll.zip. You can obtain the 
	newer versions of these files from our ftp site at 
	<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/10april2003/bigZips/">
	ftp://hgdownload.cse.ucsc.edu/goldenPath/10april2003/bigZips/</A>.
 	<P>

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 14 Apr. 2003 - Human Genome Project Successfully Completed!</FONT><BR>
 	<P>
	The International Human Genome Sequencing Consortium today announced the
	successful completion of the Human Genome Project. The most
	significant outcome of this project is the reference sequence of
	the human genome. The sequencing of
	the 3 billion letters of DNA in the human genome - which many consider to 
	be one of the most ambitious scientific undertakings in history - was
	completed 2 years ahead of schedule and at substantially less cost than
	original estimates. The reference sequence will serve as a new foundation
	for research in the fields of medicine and human biology.
	<P>
	In conjunction with this announcement, the UCSC Genome Bioinformatics 
	group is proud to release a genome browser and BLAT server on the 
	reference sequence (NCBI Build 33), along with bulk downloads of the 
	sequence and annotation data. The initial browser provides a preliminary 
	set of annotations that will be expanded in coming weeks. Bulk downloads
	of the data are available via FTP at 
	<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/10april2003">ftp://hgdownload.cse.ucsc.edu/goldenPath/10april2003</A> 
	or through the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">Downloads</A> link on this page. 
	We
	recommend that FTP be used rather than HTML for the download
	of large or multiple files.
	<P>
	The reference sequence covers about 99 percent of the human genome's gene-containing
	regions, and has been sequenced to an accuracy of 99.99 percent. The
	missing portions are essentially contained in less than 400 defined gaps
	that represent DNA regions with unusual structures that can't be reliably
	sequenced using current technology. The average DNA letter now lies within
	a stretch of approximately 27,332,000 base pairs of uninterrupted 
	sequence! 
	<P>
	Chromosomal sequences for this release were assembled by the 
	International Human Genome Sequencing Consortium sequencing centers and verified by 
	NCBI and UCSC. In some cases, sequence joins between adjacent clones 
	could not be computationally validated, e.g. due to polymorphisms in the 
	overlapping sequence. In situations like this, supporting evidence for 
	the join has been provided by the sequencing center responsible for that 
	particular chromosome. The <A HREF="certTable.html">
	Non-standard Join Certificates</A> table displays this information. The 
	annotations on the UCSC website have been provided
	by UCSC and collaborators worldwide. See the 
	<A HREF="credits.html">Credits</A> page for a detailed
	list of organizations and individuals who contributed to the success of 
	this release.

	<P>
	We'd like to congratulate the many people worldwide who have worked on 
	the Human Genome Project for this landmark achievement. We'd also like to acknowledge the UCSC
	Genome Browser project team who worked many long hours to ensure that
	the genome browser and sequence data were released on time for this
	announcement: David Haussler, Jim Kent, Terry Furey, Matt Schwartz, Heather Trumbower, 
	Angie Hinrichs, Fan Hsu, 
	Donna Karolchik, Jorge Garcia, Patrick Gavin, Chuck Sugnet, Yontao Lu, Mark Diekhans, Ryan Weber, Robert Baertsch, Krishna Roskin, and the many other students in the UCSC Genome Bioinformatics group.
 	<P>

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 2 Apr. 2003 - Human Assembly hg10 and Mouse Assembly mm1 Archived</FONT><BR>
 	<P>
	The Dec. 2001 Human assembly (hg10) and the Nov. 2001 Mouse assembly
	(mm1) have been moved to the archives. They are no longer accessible
	from the main browser, but instead can be found by clicking the 
	<A HREF="http://genome-archive.cse.ucsc.edu">Archives</A> link on this 
	page.
 	<P>

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 26 Mar. 2003 - New Rat Assembly Released in Genome Browser</FONT><BR>
 	<P>
	We've added an updated rat assembly to our site: Rat Jan 2003 (rn2). This
	corresponds to the Version 2.1 Jan 2003 Update of the rat genome 
	assembly, produced by the Atlas group at Baylor HGSC as part of the Rat
	Genome Sequencing Consortium.
	<P>
	This update corrects duplications that were assembly artifacts in the 
	previous version and improves the linking of bactigs to create larger
	"ultrabactigs". Compared with the previous rat assembly, sequence mapped 
	to specific chromosomal coordinates is reduced by about 1.6 percent. 
	Loosely mapped and unmapped sequence is reduced by 17 percent. For more
	details and statistics on the Jan 2003 assembly, see the Baylor HGSC
	<A HREF="ftp://rat-ftp.hgsc.bcm.tmc.edu/pub/analysis/rat/README">
	README</A> for this release.
	<P>
	UCSC has released a Genome Browser and BLAT server for this assembly 
	update. The initial <A HREF="../cgi-bin/hgTracks?db=rn2">
	browser</A> contains 16 annotation tracks, with more to follow in coming 
	weeks. Sequence downloads are currently available at 
	<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/rnJan2003/" TARGET=_blank>
	ftp://hgdownload.cse.ucsc.edu/goldenPath/rnJan2003/</A> or via the Downloads 
	link on this page. A complete set of database downloads will be available at the beginning 
	of next week. This data contains 
	<A HREF="credits.html#rat_use">specific 
	conditions for use</A>.
	<P>
	Thanks to the Atlas group at Baylor HGSC, the Rat Genome Sequencing Consortium,
	the UCSC Genome Bioinformatics group, and contributors worldwide for 
	making this release available.
 	<P>

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 13 Mar. 2003 - Updated Mouse Genome Assembly Now Available</FONT><BR>
 	<P>
	We're happy to announce an update to the mouse genome sequence. This new 
	version (Mouse Feb.  2003) includes 705 megabases of finished sequence, 
	compared to 96 megabases of finished sequence in the previous assembly. 
	Many people in the Mouse Genome Sequencing Consortium contributed to 
	this update. The Sanger Institute in particular contributed a large 
	amount of finished sequence. Richa Agarwala, Deanna Church, and 
	coworkers at NCBI layered the finished clones on top of the Arachne whole genome shotgun assembly. Arian Smit constructed a new RepeatMasker 
	library.  
	<P>
	UCSC has released a Genome Browser and BLAT server for the Feb. 2003 
	Mouse genome. The initial 
	<A HREF="../cgi-bin/hgTracks?db=mm3">
	browser</A> contains 14 annotation tracks, with more to follow in coming 
	weeks. Sequence downloads are currently available at 
	<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/mmFeb2003/" TARGET=_blank>
	ftp://hgdownload.cse.ucsc.edu/goldenPath/mmFeb2003/</A> or via the Downloads 
	link on this page. Database downloads will be available at the beginning 
	of next week. 
	<P>
	Thanks to everybody at UCSC and around the world that contributed to the 
	success of this release! 
 	<P>
<HR>


<FONT SIZE=4 COLOR="#006666">Posted on 5 Feb. 2003 - Enhanced Version of Genome Browser Released</FONT><BR>
 	<P>
        We're proud to announce the release of version 17 of the UCSC Genome
        Browser. 
        This version contains powerful new features, numerous improvements to 
        the annotation track display, additional annotation 
        tracks, and a number of bug fixes. In this release cycle, we've also 
	introduced   
        an enhanced QA process that formalizes our testing and verification
        of the Genome Browser software and the data displayed in the browser.
        <P>
        <B>New functionality in v.17:</B><BR> 
        -- Numerous enhancements to the table browser that allow the user to
        conduct more complex and specific searches. New features include support
        for intersections of tracks, a new summary statistics output format, and
        the ability to output query results as a custom annotation track that 
        can be viewed in the Genome Browser.
         The new 
        <A HREF="help/hgTextHelp.html"
        TARGET=_blank> Table Browser User's Guide</A> contains a detailed
        description of the new features and provides a wealth of information and
        examples for conducting various types of searches on the database tables.
        <P>
        -- Two new display modes available for most annotation tracks: pack and
        squish modes. In <I>pack mode</I> display, annotation track features are f
ully    
        displayed, but more than one feature may be displayed on the same line.
        This greatly reduces the amount of display space needed by a track
        when a user wishes to view a large number of individual features at one
        time. <I>Squish mode</I> is similar to pack mode, but displays features
        at 50% height and without labels. This mode is particularly
        useful for viewing tracks in which a large number of features align to
        the same section of a chromosome, e.g. EST tracks.
        <P>
        -- Functional groupings of annotation track controls. This
        makes it much easier to find a particular item in the track control list
        and gives a better visual overview
        of the annotations available in a particular category, e.g. comparative
        genomics tracks or gene prediction tracks.
        <P>
        -- A mechanism for saving the annotation tracks image in postscript or
        PDF format. This much requested feature enables Genome Browser users to print an image at
        high resolution, edit it with a drawing program, or display it in a
        postscript or PDF viewer.
        <P>
        -- A <A HREF="customTracks/custTracks.html"
        TARGET=_blank>collection of custom annotation tracks</A> supplied by
        Genome Browser users and members of the UCSC Genome Bioinformatics lab.
        Additional contributions to this collection are welcome! Contact
        <A HREF="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</A>
<!-- above address is genome at soe.ucsc.edu -->
	if you have an annotation you'd like to share.
        <P>
        <B>New annotation tracks in v.17:</B><BR>
        -- a new Known Genes track (Human Builds 30 & 31, Feb 2002 Mouse,
        Nov 2002 Rat) that shows known
        protein coding genes based on proteins from SWISS-PROT, TrEMBL, and
        TrEMBL-NEW and their corresponding mRNAs from Genbank. Features within
        the track are color-coded according to origin and review status.
        <P>
        -- a new Superfamily track (Human Build 30, Feb 2002 Mouse) that shows proteins having
        homologs with known structures or functions. Each entry in the track
        shows the coding region of a gene (based on Ensembl gene prediction).
        The feature label consists of the names of all known protein domains coded 
        by the gene, and usually contains structural and/or function
        descriptions that provide valuable information for getting a quick grasp
        of the biological significance for the gene.
        <P>
        We have also released several additional annotation
        tracks on the latest human and rat assemblies in the past
        month.
        <P>
        <B>Bug fixes in v.17:</B><BR>
        -- Approximately 40 bugs (mostly minor problems) have been fixed in
        this version.
        <P>
        To take full advantage of the new display features in this release, we
        recommend that you reset your browser to the new default settings. NOTE:
        you may not want to reset your browser if you have customized settings
        that you wish to preserve. You can reset your browser by clicking
        the "Click here to reset" link on the Browser Gateway page.
        <P>
        We hope this new release facilitates your work with the UCSC Genome Browser.
        If you have any questions or comments about the new release, send email
        to <A HREF="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</A>.
	<!-- above address is genome at soe.ucsc.edu -->
 	<P>
<HR>

<!---------------- 2002 archived news --------------->
<A NAME="2002"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 23 Dec. 2002 - Human Genome Assembly Build 31 Released in Genome Browser</FONT>
	<P>
        We're pleased to announce the release of the latest human genome
        assembly, Build 31 (UCSC version hg13). This assembly was produced at NCBI
        based on sequence information submitted into GenBank as of Nov. 14, 2002.
        Release notes for this assembly are available from the
        <A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/human/release_notes.html" TARGET=_blank>
        NCBI web site</A>. Because UCSC now obtains its assembly directly from
        NCBI, the UCSC Build 31 data is identical to that of NCBI and Ensembl.
        <P>
        Build 31 is an excellent high-quality assembly that shows a remarkable
        amount of progress toward the milestone of finishing
        the human genome. Greater than 95% of the euchromatic region of the
        genome is now complete, with more than 90% of the sequence in a
        finished state. The number of clone contig gaps has decreased by one
        third from the previous assembly, and the overall number of sequenced
        contigs has been reduced by one half. Seven chromosomes are considered to
        be in a finished state: 6, 7, 13, 20, 21, 22, and Y.
        <P>
        The initial release of the Build 31 Genome Browser contains 25 annotation
        tracks, with several more to follow in the upcoming weeks. Bulk
        downloads of the data are available from our FTP site at
        <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/14nov2002" TARGET=_blank>
        ftp://hgdownload.cse.ucsc.edu/goldenPath/14nov2002</A> or via the
        <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">Downloads</A> link on this page.
        <P>
        UCSC has generated a set of high-level comparisons of the Build 31
        draft sequence against various types of information (STS maps, BAC end
        pairs, and clone overlaps). This information, as well as statistics for
        Build 31, is accessible from the
        <A HREF="http://genome-archive.cse.ucsc.edu/goldenPath/chromReports/14Nov2002/" TARGET=_blank>
        Chromosome Reports</A>,
        <A HREF="http://genome-archive.cse.ucsc.edu/goldenPath/mapPlots/" TARGET=_blank> Genome Map Plots</A>, and
        <A HREF="stats.html" TARGET=_blank> Summary Statistics</A>
        links in the "Technical Information about the Assembled Sequence"
        section below.
        <P>
        We'd like to thank NCBI as well as all the people who collaborated on the
        data and annotations for this release.
 	<P>
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 6 Dec. 2002 - Rat Genome Assembly Now Available in Genome Browser</FONT>
	<P>
        We're pleased to announce the release of a UCSC Genome Browser on the
        Nov. 2002 rat assembly produced by the Baylor College of Medicine Rat
        Genome Sequencing Center and the Rat Genome Sequencing Consortium.
        The sequence was assembled using a hybrid approach that combines the
        clone by clone and whole genome shotgun methods. A new software program -
        ATLAS - was developed for this effort. The assembly process resulted in
        a 6.5-fold coverage of the rat genome, which is estimated to be
        approximately 2.8 Gigabases in size.
        <P>
        Downloads of the rat data and annotations are available through our
        ftp site at <A HREF="ftp://genome-archive.cse.ucsc.edu/goldenPath/rnNov2002" TARGET=_blank>ftp://genome-archive.cse.ucsc.edu/goldenPath/rnNov2002</A>
        or via the <A HREF="http://genome-archive.cse.ucsc.edu/downloads.html">archived downloads</A>
        link. This data contains
        <A HREF="credits.html#rat_use">specific
        conditions for use</A>. The sequence is also available from the
        <A HREF="http://www.hgsc.bcm.tmc.edu/projects/rat" TARGET=_blank> Rat
        Genome Project website</A> for the Human Genome Sequencing Center at
        Baylor College of Medicine or from GenBank.
        <P>
        We'd like to thank the Baylor team and the Rat Genome Sequencing
        Consortium for their collaboration on this project. See the
        <A HREF="credits.html#rat_credits">Credits page</A>
        for a complete list of acknowledgments.. For more
        information on the rat genome, the assembly process, and the
        Rat Genome Sequencing Consortium, refer to the website for the
        <A HREF="http://www.hgsc.bcm.tmc.edu/" TARGET=_blank>
        Human Genome Sequencing Center at Baylor College of Medicine</A>.
 	<P>
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 5 Dec. 2002 - Mouse Genome Browser Released</FONT>
	<P>
	The International Mouse Genome Sequencing Consortium has announced 
	the publication of a high-quality draft sequence of the mouse genome, 
	together with a comparative analysis of the mouse and human genomes. 
	The results from this analysis can be found in the Mouse Genome 
	Browser on this website. The paper appears in the Dec. 5 issue of the 
	journal Nature at 
	<A HREF="http://www.nature.com/nature/mousegenome/" TARGET=_blank>http://www.nature.com/nature/mousegenome/</A>.
	The co-author list includes several members of the UCSC Genome 
	Bioinformatics Group: CBSE Director David Haussler, Research 
	Scientist Jim Kent and research team members 
	Robert Baertsch, Mark Diekhans, Terrence Furey, Angie Hinrichs, Fan Hsu,
	Donna Karolchik, Krishna Roskin, Matt Schwartz, Charles Sugnet and Ryan 
	Weber. 
 	<P>

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 29 Oct. 2002 - Additional Data Downloads Available for Human Assembly June 2002</FONT>
	<P>
	We've added several new directories of downloadable data to the 28 June 
	2002 human genome assembly. These directories contain mouse/human 
	alignments of the June 2002 human assembly vs. the Feb. 2002 mouse assembly. You can access these 
	directories from our <A HREF="http://genome-archive.cse.ucsc.edu/downloads.html" TARGET=_blank>archived downloads</A> 
	link or or via our ftp site at <A HREF="ftp://genome-archive.cse.ucsc.edu/goldenPath/28jun2002" TARGET=_blank>ftp://genome-archive.cse.ucsc.edu/goldenPath/28jun2002/vsMm2/</A>. 
	<P>Within the main directory <I>vsMm2</I> are 3 subdirectories that contain all 
	the alignments (<I>axtAll</I>), alignments filtered to provide only the best alignment 
	for any given region of the human genome (<I>axtBest</I>), and a relatively stringent 
	subset of the axtBest alignments (<I>axtTight</I>). For more information 
	about the format of the alignment files and the methods used to generate 
	the alignments, consult the <I>README.txt</I> file in the vsMm2 directory.
 	<P>

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 18 Oct. 2002 - Enhancements in v16 Genome Browser </FONT>
	<P>
	We've rolled out a new version of the Genome Browser - v.16. In 
	addition to several bug fixes, this release contains some interesting 
	new features. 
	<P>
	The Table Browser has undergone major enhancements. Users 
	can now restrict their queries by specifying a value or range for any of 
	the fields in a table, and by selecting which fields should be displayed in 
	the output. The Table Browser also provides the ability to do a free-form SQL 
	query on a table and supports several new output formats. 
	<P>	
	We've extended the capabilities of the DNA retrieval functionality
	in the Genome Browser and the Table Browser. The new 
	mechanism offers the user several options for configuring the amount and 
	type of sequence region that is retrieved, and options for formatting the 
	sequence output. The retrieval options vary based on the type of table 
	selected.
	<P>	
	The Genome Browser's gene prediction tracks now offer a Comparative 
	Sequence link in addition to the predicted protein, mRNA sequence, and 
	genomic sequence links. The Comparative Sequence feature displays 
	annotated codons and translated protein for the region in alignment to 
	another species. 
	<P>	
	We've recently added a few new tracks/tables to the hg12 and mm2 Browsers. On the 
	latest human assembly (hg12), we now have Chimp Blat and Chimp BAC tracks 
	provided by Ingo Ebersberger, Joshua Bacher, and Svante Pbo at the Max
	Planck Institute for Evolutionary Anthropology. Jim Kent at UCSC has generated a 
	new annotation - Gene Boundaries - that shows the boundaries of genes 
	and the direction of transcription as deduced from clustering spliced 
	ESTs and mRNAs against the genome. Daryl Thomas of UCSC has also added SNP tracks to both 
	hg12 and the latest mouse assembly (mm2), based on data from the SNP
	Consortium and NIH.
	<P>	
	We encourage you to experiment with these new features. Comments, 
	questions, and suggestions are always welcome at 
	<A HREF="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</A>.
	<!-- above address is genome at soe.ucsc.edu -->
 	<P>

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 17 Oct. 2002 - Corrections to Mouse Assembly mm2 Repeat-masking</FONT>
	<P>
	We have found errors with the RepeatMasker track on the 
	Feb. '02 mouse assembly (mm2). This problem affects the RepeatMasker track 
	and RepeatMasked DNA obtained via the browser's DNA links. It does 
	<I>not</I> affect data downloaded from the browser's downloads page or ftp 
	site. We have replaced the erroneous data set with a corrected version. 
	If you have questions about how this change may affect your project, email
	<A HREF="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</A>.
	<!-- above address is genome at soe.ucsc.edu -->
 	<P>

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 4 Oct. 2002 - Chromosome 22 Coordinate Differences Between UCSC and NCBI</FONT>
	<P>
	Please note that the coordinate range for a portion of the Build 30 chr22 in 
	the UCSC Genome 
	Browser differs from that of NCBI and the Sanger Centre. The latter version 
	contains a 100K bp gap 
	inserted into chr22 just after the centromere. This
	modification was made after UCSC released the Build 30 assembly in the
	Genome Browser. All of the chr22 annotations displayed by the UCSC
	browser are correctly positioned relative to one another. However the
	coordinates of all features past the centromere will be 100K less than those 
	of NCBI and Sanger. The Ensembl Genome Browser is consistent with the
	UCSC browser.
 	<P>

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 19 Sep. 2002 - Release of Human/Mouse Evolutionary Conservation Score Annotation</FONT>
	<P>
	We're pleased to announce the release of the Mouse Cons (Human/Mouse
	Evolutionary Conservation Score) annotation track for the <A
	HREF="../cgi-bin/hgTracks?db=hg12">June 2002
	human genome assembly</A>. This track allows a user to interactively explore 
	conservation between the human and mouse genomes and identify highly conserved 
	regions. Highest levels of conservation are typically seen over coding exons. 
	High levels of conservation are also frequently associated with noncoding RNA,
	promoters, other regulatory elements, and pseudogenes.

	<P>The Mouse Cons annotation is displayed using a new type of Genome
	Browser graphical track that plots a continuous function along a chromosome. The
	conservation levels are calculated over 50bp  windows in the human genome
	that have at least 15 bp aligned to mouse. The score for a window reflects
	the probability that the level of observed conservation in that 50bp region
	would occur by chance under neutral evolution. This information is given on
	a logarithmic scale and displays in the track as "mountain ranges". Details
	pages associated with the individual peaks in the track provide access to
	the base level alignments for the whole region and for the individual 50bp
	windows.

	<P>The team that produced this track includes Ryan Weber, Krishna Roskin,
	Mark Diekhans, Jim Kent, Scott Schwartz, and Webb Miller.
 	<P>
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 12 Sep. 2002 - &quot;User's Guide to the Human Genome&quot; Features UCSC Genome Browser</FONT>
	<P>
	<I>Nature Genetics</I> has just published <A
	HREF="http://www.nature.com/cgi-taf/DynaPage.taf?file=/ng/journal/v32/n1s/index.html"
	TARGET=_blank>User's Guide to the Human Genome</A>, a
	hands-on tutorial for using genome browsers as web tools for browsing and
	analyzing data from the Human Genome Project and other sequencing efforts. The 
	3 browsers featured in the tutorial include the UCSC Genome Browser, NCBI's 
	Map Viewer, and the Ensembl Genome Browser. The guide is organized around a 
	collection of step-by-step solutions to 13 typical research questions,
	and serves as a nice supplement to the documentation materials
	available on the UCSC Genome Browser web site. The guide is accessible as a 
	link off the <I>Nature Genetics</I> home page at 
	<A HREF="http://www.nature.com/ng/" TARGET=_blank>http://www.nature.com.ng</A>. 
 	<P>
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 12 Sep. 2002 - Correction to Human Assembly hg12 Chromosome Bands Annotation</FONT>
	<P>
	We've corrected a problem with the Human June 2002 (hg12) cytoBand annotation 
	track that affected chromosome Y. The clones on this chromosome were
	erroneously pushed from the q-arm onto the p-arm, creating some confusion. The
	currently available version of the cytoBand data on this website contains this 
	correction.
 	<P>

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 15 Aug. 2002 - Additional Annotations for Human Draft Assembly hg12 Released</FONT>
	<P>
	We've just released several new annotation tracks/tables for the human genome. 
	On the June 2002 assembly, we've added Gene Bounds, 
	UniGene, CpG Islands, Nonhuman mRNA & EST, SNPs, NC160, and GNF Ratio.
	On the April 2002 assembly we've added a Fgenesh++ Genes track, and have also
	updated the RepeatMasker track. The Dec. 2001 release now includes a Sanger 22 
	track. 

	<P>On the mouse genome, we have 2 new tracks for the Feb. 2002 assembly:
	TIGR Gene Index and RNA Genes.
 	<P>
<HR>
	
<FONT SIZE=4 COLOR="#006666">Posted on 6 Aug. 2002 - Error with Six Chromosome 3 Contigs in Human Assembly hg12</FONT>
	<P>
	We have fixed an error with six of the chr3 contigs in the bigZips/contigAgp.zip 
	file. The following .agp files were corrected: NT_005684.agp, NT_005663.agp, 
	NT_022554.agp, NT_022459.agp, NT_006031.agp, and NT_022419.agp. The
	chr3.agp file in bigZips/chromAgp.zip was also modified. This change does not 
	affect the .gl files, the .fa files, the 
	lift files, or the annotations. Alignments made on the previous version
	of chr3 are still good. Updated versions of the contigAgp.zip and
	chromAgp.zip files were
	uploaded to our site today. You can download the new versions via ftp at <A 
	HREF="ftp://genome-archive.cse.ucsc.edu/goldenPath/28jun2002" TARGET=_blank>
        ftp://genome-archive.cse.ucsc.edu/goldenPath/28jun2002</A> or via the 
	<A HREF="http://genome-archive.cse.ucsc.edu/downloads.html">archived
	downloads</A> link.
 	<P>
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 2 Aug. 2002 - New RepeatMasker Track Available for Human Assembly hg12</FONT>
	<P>The problems with the June 2002 Build 30 (hg12) RepeatMasker track have been 
	resolved. The new RepeatMasker track, along with regenerated Fish Blat
	and Genscan tracks, are now available in the Browser and 
	through our
	Downloads link. We've also added a few new annotation tracks for the
	June 2002 release, and will be adding more over the next 2 weeks.

	<P>The latest Genome Browser has 2 new features. We've added filter
	functionality to the Table Browser, accessible via the Filter Fields
	button on the Table Browser main page. Also, some of the Dec. 2001 human genome tracks (eg. RefSeq Genes) now have a
	Comparative Sequence link from the details page that shows annotated
	codons and translated protein with alignment to the mouse genome.
 	<P>
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 27 Jul. 2002 - Problems with Repeat-masking on Human Assembly hg12</FONT>
	<P>
	We've experienced some RepeatMasker problems on Build 30 and are
	rerunning it. This will directly affect the RepeatMasker track and the
	masking of the fasta files. The Fish Blat and Genscan tracks may also
	change slightly once we're redone this. The EST, mRNA, and RefSeq tracks
	should not be affected. We will also post a new RepeatMasker track for
	Build 29 (see news item below) as soon as the Build 30 tracks are
	completed. We apologize for any rework this may cause.
 	<P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on 26 Jul. 2002 - Bulk Data Downloads Now Available for Human Assembly hg12</FONT>
	<P>
	Bulk downloads of the June 2002 Build 30 human genome assembly (hg12) are now
	available from
	<A HREF="ftp://genome-archive.cse.ucsc.edu/goldenPath/28jun2002" TARGET=_blank>
	ftp://genome-archive.cse.ucsc.edu/goldenPath/28jun2002</A>. You can also 
	access the data via the <A HREF="http://genome-archive.cse.ucsc.edu/downloads.html">archived downloads</A> link. This initial
	release of the annotation database download contains a limited set of
	tables. Additional files will be available for download next week.
 	<P>
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 24 Jul. 2002 - Human Assembly hg12 Blat Server Available</FONT>
	<P>
	The BLAT server and the coordinates conversion feature for human genome 
	assembly Build 30 (hg12) are now functional.
 	<P>
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 23 Jul. 2002 - Pre-release of Human Draft Assembly Build 30 in Genome Browser</FONT>
	<P>We're pleased to announce the pre-release of a browser for 
	human genome assembly Build 30 from NCBI (UCSC version hg12). This assembly was 
	produced at NCBI based on sequence information submitted into GenBank as of
	June 28, 2002. Build 30 release notes and statistics will soon be
	available from the <A
	HREF="http://www.ncbi.nlm.nih.gov/genome/guide/human/" TARGET=_blank> NCBI web 
	site</A>.  
	<P>
	Build 30 is an excellent high-quality assembly. It contains nearly 87%
 	finished sequence, and 94%-97% coverage. The sequence coverage of this build
	is much higher than in previous releases, and there is a high level of
	correspondence between the sequence and the map. Currently, the human genome
	project appears to be on track to achieve the goal of finishing at least 95%
	of the human genome (using Bermuda standards) by April 2003.
	<P>
	UCSC has generated a set of high-level comparisons of the Build 30 draft
	sequence against various types of information (STS maps, BAC end pairs, and
	clone overlaps), accessible from the <A 
	HREF="http://genome-archive.cse.ucsc.edu/goldenPath/chromReports/24Jun2002/"
	TARGET=_blank>Chromosome Reports</A> and <A 
	HREF="http://genome-archive.cse.ucsc.edu/goldenPath/mapPlots/" TARGET=_blank>Genome 
	Map Plots</A> links in the "Technical Information about the Assembled 
	Sequence" section below.
	<P>
	A Blat server for Build 30 is not yet available, but should be
	accessible from this site later this week. Data for the mitochondrial genome 
	and several more annotation tracks will be posted for this release as they 
	become available.  Bulk downloads of the hg12 data should be available from 
	this site in a few days. 
 	<P>
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 9 Jul. 2002 - Repeat-masking Problems on Human Assembly hg11</FONT>
	<P>
	We've found some problems with the repeat-masking of the Build 29 (hg11)
	human sequence. We're in the process of replacing the RepeatMasker
	track, but do not plan to redo the other tracks due to the imminent
	release of Build 30. Because of this, we advise that you do not use the
	cross-species tracks for statistical purposes.

 	<P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on 1 Jul. 2002 - New Look for UCSC Genome Bioinformatics Home Page</FONT>
	<P>
	The UCSC Genome Bioinformatics home page is sporting an updated interface to 
	accommodate the growing number of organisms supported by the UCSC Genome 
	Browser, BLAT, and Table Browser. The list of assembly versions accessible 
	through each of these tools can now be found on the tool's Gateway page. To 
	reach the Gateway page, choose an organism from the dropdown list on the left 
	sidebar of this page, then click the Browser, BLAT, or Tables link. New 
	organisms will be added to the list in the months ahead.

	<P>The UCSC site continues to provide a variety of bulk downloads of a genome 
	assemblies and annotations. The list of downloadable data has been removed 
	from the home page, but is readily available through the Downloads link on 
	the left sidebar.  The downloads list can also be accessed directly at 
	<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">http://hgdownload.cse.ucsc.edu/downloads.html</A> or through 
	our ftp site at 
	<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/" TARGET=_blank>ftp://hgdownload.cse.ucsc.edu/goldenPath/</A>.

	<P>Several new annotation tracks have been added to our site in the past 
	couple weeks.  The Feb. 2002 mouse assembly now has tracks for BAC End pairs, 
	Fgenesh++ gene predictions, and AltGenie gene predictions based on Affymetrix's 
	Genie gene-finding software. New to the Apr. 2002 human assembly is the 
	GenMapDB Clones track, which shows placements of BAC clones from the GenMapDB 
	database based on BAC end sequencing information and confirmed using STS 
	markers by Vivian Cheung's lab at U. Penn.  We've also changed the Known Genes 
	track name to RefSeq Genes in all assemblies.

	<P>This release also includes an updated 
	<A HREF="help/hgTracksHelp.html" TARGET=_blank>User's Guide</A>
	and more detailed documentation on creating & using 
	<A HREF="help/customTrack.html" TARGET=_blank>custom annotation tracks</A>.
<P>

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on 24 May 2002 - Bulk Data Downloads for Human Assembly hg11 Now Available</FONT>
	<P>
	Bulk downloads of the April 2002 hg11 human genome assembly (NCBI Build 29) 
	are now available from <A
	HREF="ftp://genome-archive.cse.ucsc.edu/goldenPath/05apr2002"
	TARGET=_blank>ftp://genome-archive.cse.ucsc.edu/goldenPath/05apr2002</A>. You
	can also access the data from the Downloads link in the left sidebar.

 	<P>
<hr>
<FONT SIZE=4 COLOR="#006666">Posted on 22 May 2002 - Human Draft Assembly Build 29 Released in Genome Browser</FONT>
	<P>
	We've just released a browser and BLAT server on 
	the latest Build 29 human genome assembly from NCBI (UCSC 
	version hg11). This assembly is based 
	on sequence information submitted into GenBank as of Apr. 5 2002. As with the 
	Dec. 2001 (hg10) release, this assembly was produced at 
	NCBI rather than at UCSC.  Consult NCBI's 
	<A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/human/release_notes.html"
	TARGET=_blank>Build 29 release notes</A> and 
	<A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/human/HsStats.html"
	TARGET=_blank>statistics</A> for more information about this release.
	This assembly contains nearly 75% finished sequence. 
	Currently, the human genome project appears to be on track to achieve the goal 
	of finishing at least 95% of the human genome (using Bermuda standards) by April 2003.
	
	<P>Although the NCBI human genome assembly has been steadily
	improving over the past year, mapping problems still exist in
	the current release. Most are small, relatively local rearrangements. 
	Larger scale problems include a rearrangement in the p-arm of Chr16 and
	several discrepancies in Chr17.  Researchers - especially positional 
	cloners - are strongly encouraged to use the tools provided 
	(<A HREF="http://genome-archive.cse.ucsc.edu/goldenPath/mapPlots/"
	TARGET=_blank>comparison plots</A>, <A
	HREF="http://genome-archive.cse.ucsc.edu/goldenPath/chromReports/"
	TARGET=_blank>chromosome reports</A>) to evaluate the
	accuracy of the assembly in specific regions of interest.

	<P>Bulk downloads of the hg11 data should be available from this site
	in approximately one week. New annotation tracks will be posted as soon
	as they become available.
 	<P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on 24 Apr. 2002 - Bulk Data Downloads of Mouse Assembly Feb. 2002 Available</FONT>
	<P>Bulk downloads of the February 2002 mouse genome assembly are now available 
	from <A HREF="ftp://genome-archive.cse.ucsc.edu/goldenPath/mmFeb2002" 
	TARGET=_blank>ftp://genome-archive.cse.ucsc.edu/goldenPath/mmFeb2002</A>.  You can 
	also access the data from the <A HREF="http://genome-archive.cse.ucsc.edu/downloads.html">archived downloads</A> link.<br></P>
 	<P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on 19 Apr. 2002 - New Mouse Genome Assembly Released</FONT>
        <P>The February 2002 mouse genome assembly is now available in the browser and 
	for BLAT searching.  This assembly was produced at the Whitehead Institute 
	using their Arachne software.  We'd like to thank them and the Mouse Genome 
	Sequencing Consortium for providing this assembly, which has 
	<A HREF="credits.html#mouserestrict">specific conditions for use</A>.
	Bulk downloads of the data should be available in approximately one week.  
	Coordination with mouse genome data access at 
	<A HREF="http://www.ensembl.org/Mus_musculus/">Ensembl</A> and 
	<A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/mouse/">NCBI</A> is 
	in progress.  We'd also like to acknowledge the UCSC team that produced this 
	release: Jim Kent, Terry Furey, Matt Schwartz, Fan Hsu, Yontao Lu, 
	Donna Karolchik, Chuck Sugnet, and Ryan Weber.<br></P>
 	<P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on 9 Apr. 2002 - Bulk Data Downloads of Mouse Assembly Nov. 2001 Available</FONT>
        <P>Bulk downloads of the November 2001 mouse genome assembly are now available 
	from <A HREF="ftp://genome-archive.cse.ucsc.edu/goldenPath/mmNov2001"
        TARGET=_blank>ftp://genome-archive.cse.ucsc.edu/goldenPath/mmNov2001</A>.
	You can also access the data from the <A HREF="http://genome-archive.cse.ucsc.edu/downloads.html">archived downloads</A> link.<br></P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on 2 Apr. 2002 - Release of New Genome Browser Version and Mouse/Human Annotations</FONT>
	<P>An updated version of the UCSC Genome Browser (v.11) is now available.
	Along with the v.11 browser, we've released several new annotation tracks on 
	the latest human and mouse assemblies. The new Human Dec. 2001 tracks include:
	Mouse Synteny, Ensembl, Genscan, CpG Islands, Mouse Blat, Fish Blat, 
	Unigene/SAGE, NCI60 Microarray, GNF Affymetrix Microarray, Rosetta
	Microarray, and SNPs. An STS Markers track has been added to the Mouse
	Nov. 2001 browser.<br></P>
 	<P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on 14 Mar. 2002 - Mouse Draft Assembly Nov. 2001 Released in Genome Browser</FONT>
	<P>The November 2001 mouse genome assembly is now available
	for viewing in the browser and for BLAT searching. This assembly
	was produced at the Sanger Center using the Phusion software developed
	by Jim Mullikin and Zemin Ning, and was tied to the 
	mouse fingerprint map by Tim Hubbard. We'd like to
	thank them and the Mouse Genome Sequencing Consortium for providing this
	assembly, which has <A HREF="credits.html#mouserestrict">specific 
	conditions for use</A>. Bulk downloads of the data
	will be available in approximately one week. Coordination with
	mouse genome data access at 
	<A HREF="http://www.ensembl.org/Mus_musculus/">Ensembl</A> and 
	<A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/mouse/">NCBI</A> 
	is in progress.  We'd also like to acknowledge the UCSC team that produced 
	this release: Jim Kent, Terry Furey, Matt Schwartz, Fan Hsu, Yontao Lu, 
	and Donna Karolchik.<br></P>
 	<P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on 16 Feb. 2002 - Human Genome Assembly Build 28 Now Available</FONT>
	<P>A new assembly based on sequence submitted as of Dec. 22 in Genbank (Build 28) is
	now available in the browser and for BLAT search.  
	This assembly was produced at NCBI rather than UCSC, primarily by
	Richa Agarwala, Greg Schuler, and Paul Kitts.  The NCBI assembly has
	been steadily improving over the past year.  Currently it shows slightly
	better local order and orientation compared to the UCSC assembly on the
	same sequence, but somewhat worse tracking of the chromosome level maps.
	The NCBI assembly has the advantage that it can be generated significantly
	faster than the UCSC assembly.  With the human genome sequencing now
	in the end game - over two thirds of the human clones are now finished -
	we feel it more productive to focus worldwide annotation efforts on a
	single assembly rather than continue producing competing assemblies.
	We're working with NCBI to improve their map tracking. <br></P>

 	<P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on 4 Feb. 2002 - Chromosome Reports Released</FONT>
	<P>Chromosome Reports detailing correspondence with STS map, overlap, 
	and BAC end sequence information are available under the "Technical 
	Information About the Assembled Sequence" section below.  This also 
	gives information about the clone map on which the assembled sequence 
	is based.<br></P>		
 	<P>
<hr>

<A NAME="2001"></A>
<FONT SIZE=4 COLOR="#006666">Posted on 18 Dec. 2001 - New Genome Browser Enhancements</FONT>
        <P>There are some major enhancements to the browser. The complete user interface settings
		including track controls, labels, and position are now saved from session to session.
		You can configure the browser once to your liking and it will stay that way. This feature
		will only work if cookies are enabled in your browser. If you
		want to restore the default settings use the reset all button under the main graphic.
		Also under the main graphic are new controls that move just the start or just the end
		of the genome window. These are useful for getting exactly the right view without having
		to do arithmetic on the position.  These controls by default will move two guideline units
		at a time, but you can specify other increments.  There's a new page associated with each
		track. This page is accessible by clicking on the mini-buttons to the left of the track 
		in the main graphic, or by clicking on the new hyperlink associated with the track in the
		track controls section under the graphic.  These pages contain a description of the track
		and in many cases new controls.  The mRNA and EST associated controls let you color or
		filter the display according to tissue, author, organism, and so forth.</P>
		<P>As with any new enhancement there are likely to be a few new bugs too.
		Many of these have been spotted and fixed already.  Please let us know 
		if you find a problem that persists more than a day or two.  It's always helpful to include 
		the freeze and genomic position with a problem report.<br></P>

 	<P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on 30 Nov. 2001 - Link Added to Jackson Lab's MGI Mouse Orthologs</FONT>
        <P>There is now a link from the known genes details page to the Jackson Lab's MGI Mouse Ortholog 
		when the ortholog is known.  Thanks to Carol Bult for her help setting up this link.<br></P>

 	<P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on 29 Nov. 2001 - Duplications Track Released for August Browser</FONT>
        <P>A duplications track is now available in the August browser.  This track 
		shows duplicate blocks of sequence larger than 1000 bases.  The track is hidden
		by default.  To open it look for 'Duplications'  in the third row of track controls 
		under the main graphic window, and change the setting to 'dense'.  Thanks to Evan Eichler 
		and Jeff Bailey for this track.<br></P>

 	<P>
<hr>	

<FONT SIZE=4 COLOR="#006666">Posted on 28 Nov. 2001 - Sanger Curated Gene Annotations Now Available on Chromosome 20</FONT>
        <P>Sanger curated gene annotations are now available on chromosome 20.  Thanks to 
		Jennifer Ashurst, James Gilbert, and all the annotators at the Sanger Institute.<br></P>

 	<P>
<hr>		

<FONT SIZE=4 COLOR="#006666">Posted on 27 Nov. 2001 - Haplotype Blocks Annotation Released for Chromosome 21</FONT>
        <P>A new track has been added to the August freeze browser showing
		haplotype blocks derived from common SNPs on Chromosome 21 by Perlegen,
		as described in "<A HREF="http://www.sciencemag.org/cgi/content/abstract/294/5547/1719">Common 
		High-Resolution Haplotypes</A>." Patil, N. et. al. Science 294:1719-1723 (2001).<br></P>

 	<P>
<hr>		

<FONT SIZE=4 COLOR="#006666">Posted on 19 Nov. 2001 - SNP and Mouse Blat Tracks Available</FONT>
        <P>The SNP and Mouse Blat tracks are now available
		for August.  The Mouse Blat track uses a partial assembly of
		the public whole genome shotgun data courtesy of Whitehead's 
		Arachne program.  <br></P>

 	<P>
<hr>		

<FONT SIZE=4 COLOR="#006666">Posted on 8 Nov. 2001 - Interface of Track Details Pages Updated</FONT>
        <P>The detail web pages for each of the tracks have been updated to reflect
        the overall look and design of this site.  You will now see the familiar
        blue navigation bar with links to the Browser, BLAT, Downloads, and the  
        FAQ page from each of the track detail pages.<br></P>

 	<P>
<hr>	        

<FONT SIZE=4 COLOR="#006666">Posted on 8 Nov. 2001 - STS Markers Track Updated on Apr. and Aug. Browsers</FONT>
        <P>The STS Markers track has been updated on the April and August browsers to
        now include much more information on the detail page including links to
        UniSTS and details on the alignments of the markers to the draft sequence.
        In addition, all known aliases of the markers can be entered in the
        "position" window, and the corresponding merker will be found and
        displayed if its location has been determined.<br></P>

 	<P>
<hr>	

<FONT SIZE=4 COLOR="#006666">Posted on 8 Nov. 2001 - FISH Clones Track Added to Apr. and Aug. Browsers</FONT>
        <P>A new FISH Clones track has been added to the April and August browsers.
        Previously, this information has been included in the STS Markers track.
        Now, this has been broken out into a separate track with additional
        information provided on the detail page not previously shown.<br></P>

 	<P>
<hr>	

<FONT SIZE=4 COLOR="#006666">Posted on 6 Nov. 2001 - Fgenesh++ Gene Predictions and Cross-Species mRNA Tracks Available</FONT>
        <P>The fgenesh++ gene prediction and the cross-species
        mRNA tracks are now available in the August browser.<br></P>

 	<P>
<hr>	

<FONT SIZE=4 COLOR="#006666">Posted on 31 Oct. 2001 - DNA Button Functionality Upgraded</FONT>
	<P>The 'DNA' button at the top of the browser has been significantly 
	upgraded.   By default it now returns DNA that has repeating elements 
	in lower case and other DNA in upper case.  There is also an option to 
	color the DNA output with various tracks.  You can have the case and font 
	features such as underline, bold, and italic follow tracks too.<br></P>

 	<P>
<hr>	

<FONT SIZE=4 COLOR="#006666">Posted on 29 Oct. 2001 - TIGR Gene Index Track Added to April Browser</FONT>
	<P>There is now a TIGR Gene Index track
	in the April 2001 freeze browser.  The TIGR Gene Index is based
	on alignments of assembled ESTs from a number of species.
	Be sure to click into the track and follow the outside links to
	the TIGR site, which contains a wealth of information on the genes.<br></P>

 	<P>
<hr>	

<FONT SIZE=4 COLOR="#006666">Posted on 29 Oct. 2001 - Acembly Track Updated on August Browser</FONT>
	<P>The Acembly track on the August 2001 freeze
	has been updated to include predictions based on human ESTs 
	and Genbank  mRNAs as well as RefSeq human mRNAs.  Protein 
	predictions are now also available in the details page for this
	track.  The outside link for this track is also very informative.<br></P>

 	<P>
<hr>	

<FONT SIZE=4 COLOR="#006666">Posted on 27 Oct. 2001 - Share Your Custom Annotation Tracks!</FONT>
        <P>You can now share your custom tracks with the
        community. The easiest way to do this is to construct a link from your own
        web pages to the browser.  Here is an example of a URL for
        such a link:
        <pre>   http://genome.ucsc.edu/cgi-bin/hgTracks?
        position=chr22:1-20000&db=hg8&
        hgt.customText=http://genome.ucsc.edu/test.bed</pre>

        The position variable tells the browser which part of the genome to
        display.  The db variable refers to the freeze number.
        'hg8' corresponds to the August 2001 freeze.  The customText
        variable should refer to a URL containing plain text in one of the
        formats described in <a href="help/customTrack.html">
        http://genome.ucsc.edu/goldenPath/help/customTrack.html</a>.
        <br><br>
        Note that generally we only keep the last three versions of the
        genome online (hg6, hg7, and hg8).  You'll have to update
        your link and track about every 4 months as a result.
        <br><br>
        Please send in the URLs of tracks you'd like to share to
        <A HREF="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</A>,
	<!-- above address is genome at soe.ucsc.edu -->
        along with a brief description of
        the track and the genome version it is tied to.  We'll create
        an index page of these here.
    <br><br>
        You can also access an external custom track by including
        the URL of the track data (on a separate line starting with
        http://) in the custom track box at the bottom of the browser gateway.</P>

 	<P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on 15 Oct. 2001 - Weekly Updates of Tab-delimited Data Files </FONT>
        <P>Fresh tab-delimited files from the browser database
        are now available.  They will now be updated automatically
        every Sunday evening.   The table browser queries the database
        directly, so it is always up to the minute.</P>

 	<P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on 12 Oct. 2001 - New Gene Prediction Tracks for Apr. and Aug. Browsers</FONT>
	<P>Several new sets of gene predictions came in this week.
	We now have fgenesh++ predictions for the April freeze,
	and Genscan and Acembly predictions for the August freeze.</P>

 	<P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on 11 Oct. 2001 - Table Browser Upgraded</FONT>
	<P>Why struggle with massive genomic file downloads when the UCSC 
	<A HREF="../cgi-bin/hgTables">Table Browser</A> 
	lets you select exactly the track data desired via a convenient web 
	interface? Major new improvements by Krish Roskin have empowered and 
	simplified this feature available now for the three most recent assemblies.</P>

 	<P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on 8 Oct. 2001 - Oct. 2000 Assembly Moved to Archives</FONT> 
	<P>The October 2000 assembly has been moved to the 
	<A HREF="http://genome-archive.cse.ucsc.edu">Archives</A>
	to make room for the August assembly.</P>
	
 	<P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on 5 Oct. 2001 - Revised Aug. 2001 Assembly Released</FONT> 
	<P>A revised August 2001 freeze assembly is now up.  The problems
	with flipped contigs of finished clones and high levels of
	sequence duplication are fixed.   You can now download this
	assembly <A HREF="ftp://genome-archive.cse.ucsc.edu/goldenPath/06aug2001/bigZips">in bulk</A> as 
	well as browse it.   Chromosome by chromosome and annotation
	database files will be following over the next day or two.</P>

 	<P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on 2 Oct. 2001 - Convert Coordinates Between Different Draft Assemblies</FONT>
	<P>You can now convert coordinates between different versions of the draft
	using a new program, hgCoordConv, by Chuck Sugnet.
 	hgCoordConv attempts to cut out sequences of the original 
	draft and align them to the new draft.  When aligning the 
	sequences to the new draft hgCoordConv makes sure that the
	sequences are in the same order, orientation, and have the 
	correct distances between them. </P>
	
	<P>Chuck has also implemented a SAGE/Unigene track in the browser. 
	This track displays data from the <A HREF= http://www.ncbi.nlm.nih.gov/SAGE/>
	SAGEMap project at NCBI </A>.   UniGene cluster sequences are
	displayed in the browser and colored according to their average SAGE
	counts over a series of experiments. Selecting one of the UniGene
	representative sequences displays the SAGE results for UniGene 
	sequences.</P>
	
 	<P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on 21 Sep. 2001 - Clone Order Problems Found in Preliminary Aug. 2001 Assembly</FONT>
	<P>Some systematic problems were found in the clone order on the
	preliminary August 2001 freeze assembly. The sequencing center at
	Washington University, EBI's Ensembl group, and our group here at UCSC
	are currently working together to revise the merged BAC clone maps and
	the assembly process to fix these problems. We hope to update the August
	browser with a revised assembly soon. Then, after further testing via
	the browser, we will release the assembled August freeze genome sequence
	itself. </P>
	
 	<P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on 11 Sep. 2001 - Preliminary Aug. 2001 Assembly Now Available in Genome Browser</FONT>	
	<P>A preliminary assembly of the August 2001 freeze is now
	available in the genome browser. Due to significant progress 
	by the mapping and finishing groups of the international 
	public consortium, this assembly is a major improvement over 
	the April 2001 freeze assembly.	Imre Vastrik, Ewan Birney and 
	colleagues at Ensembl have computed a merge of BAC clone maps
	provided by the individual sequencing centers with
	fingerprint-based maps prepared at Washington University. These
	merged maps were used for the first time in this August 
	assembly.</P>

	<P>The August assembly has successfully passed our internal 
	quality control tests. We will release the sequence and
	annotations in bulk downloadable form in a week or so, after 
	the external testers have had a chance to further verify
	it. Meanwhile if you notice any systematic problems please let 
	us know at 
	<A HREF="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.
&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</A>.
	<!-- above address is genome at soe.ucsc.edu --> 
	Though the state of the working
	draft has improved considerably, remember that where you see 
	solid marks in the 'gap' track, the relative order and 
	orientation of flanking contigs is still uncertain. In some 
	cases of complex repeat structure it is also possible that
	the assembly may be incorrect even in the absence of gaps.
	Also, sometimes ambiguities in the data cause a BAC clone to be 
	split, with parts of it placed at opposite ends of a run of other clones.	
	Localized errors of this type should be corrected by additional 
	finishing efforts at the individual sequencing centers and 
	should not be reported to UCSC. However, please report any
	large-scale or systematic problems you detect with this assembly	
	that could have been caused by our data processing.</P>

	<P>The tracks available on the August 2001 browser are quite
	limited at the moment. More tracks will show up over time.</P>

 	<P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on 11 Sep. 2001 - April 2001 Assembly is New Default in Genome Browser</FONT>	
                <P>The April 2001 assembly is now the default for the browser.
        The SNP and Ensembl gene tracks have come in for this version.
        There is also a new track depicting non-human vertebrate mRNA
        alignments.</P>
                <P>The Sept. 2000 and July 2000 versions of the genome are        
        now only available on our archive site.  Please see the link
        in the blue box to the left for more details.</P>
                <P>The August 6 freeze is progressing through the pipeline.
        We've recently received an updated accession map from Wash U.
        Ensembl will shortly be integrating this with chromosome
        specific maps from the sequencing centers.  We are still
        on track for an early September next release.  </P>

 	<P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on 28 Aug. 2001 - Custom Annotation Track Functionality Added </FONT>	
	<P> Meanwhile we've been continuing work on the genome browser.  
  	It's now possible  to upload your own annotations to be displayed 
  	alongside the  built-in tracks.  Please scroll to the bottom of the 
 	browser gateway pages for further information.  The browser has 
  	also been sped  up, particularly on the larger chromosomes by 
  	using a 'binning'  technique suggested by Lincoln Stein and Richard Durbin.</P>

 	<P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on 23 Aug. 2001 - New Annotation Tracks on April 2001 Browser</FONT>	
		<P>Tracks continue to be added to the 
	<A HREF="http://genome-archive.cse.ucsc.edu/">April 2001 browser</A>.
	Our old friend the Exofish track is back.  The blat mouse homology
	track is now up as well, computed at somewhat more sensitive settings
	than it was in the December 2000 browser.</P>
		<P>We've recently received some significant funding from NHGRI to
	maintain and extend this site.  This has allowed us among other things
	to hire an artist, Jenny Draper, who is responsible for the new look.</P>

   </TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	
<BR>
	
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a NAME="#BROWSER"></a><b>&nbsp; Freeze-Specific News Archives</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=7></TH><TR><TD WIDTH=10></TD><TD>
		
<FONT SIZE=4 COLOR="#006666">1 Apr. 2001 Freeze</FONT>
<P>13 Jul. 2001: fixed bug where some UTRs were
mis-annotated in the known genes on the minus strand.</P>
<P>

<hr><FONT SIZE=4 COLOR="#006666">12 Dec. 2000 Freeze</FONT>
<P>13 Jul. 2001: fixed bug where some UTRs were
mis-annotated in the known genes on the minus strand.</P>
<P>5 Apr. 2001: chromosome level files (but not contig level files)
updated to fix bug where some of the centromeres were misplaced</P>
<P>1 Apr. 2001: chromosome Y updated to fix a bug that put a large
gap between each clone.  This bug was limited to the Y chromosome.</P>
<P>

<hr><FONT SIZE=4 COLOR="#006666">7 Oct. 2000 Freeze</FONT>
<P>7 Apr. 2001: Affymetrix gene predictions updated and available
for bulk download.</P>
<P>9 Jan. 2001: all files were updated after a bug that had caused
some finished clones to be flipped in the assembly was caught and fixed.
Our apologies for any inconvenience this has caused.

<P>10 Nov. 2000: the sequence (.fa) files for two contigs:
X/ctg18523/ctg18523.fa and 7/ctg15082/ctg15082.fa were updated.
These files had null (zero valued) characters that have been
replaced with N characters.  These characters were a result of
a mismatch between clone sizes in the map and in finished NT
contigs.  The sequence for chromosomes X and 7, which contain
these contigs, has also been updated.</P>
<P>

<hr><FONT SIZE=4 COLOR="#006666">5 Sep. 2000 Freeze</FONT>
<P>11 Oct. 2000: chr21.agp and chr22.fa were updated.  chr21.agp
was a version which went with the UCSC draft assembly rather than the
Sanger/NCBI final assembly of this chromosome.  chr21.agp and chr21.fa
are now in sync.   chr22.fa and chr22.agp were also previously out of
sync.  chr22.fa was obtained from Sanger while chr22.agp had been obtained
by NCBI.  With this update they are consistent, both NCBI versions.
Apologies for any rework this causes you. </P>

<P>9 Oct. 2000: chr21_random.* and chr22_random.* were removed from
the zip-files in the September 5th freeze.  These files were relics that
should not have been included in the first place.  The files
chr9_random.agp, chr10_random.agp, chr11_random.agp, chr12_random.agp,
chr13_random.agp and chr14_random.agp were updated.  There was a bug
where the initial field of the initial lines in these files was
"(null)" rather than "chrN_random" as it should have been. </P>
<P>

<hr><FONT SIZE=4 COLOR="#006666">17 Jul. 2000 Freeze</FONT>
<P>22 Sep. 2000: The zip-files chromFa.zip and chromAgp.zip under the
<A HREF="http://genome-archive.cse.ucsc.edu/goldenPath/17july2000/bigZips/">July 17th full data set </A>
were updated to fix some duplications of clone contigs that occurred in the chrN_random.agp and chrN_random.fa
files contained within these zip-files. These "_random" files are files that contain clone contigs that were mapped to a particular chromosome, but
could not be placed at a specific position within that chromosome. They correspond to the "RANDOM" sections of the WashU map. 
None of the regular chrN.agp or chrN.fa files were affected by this update, nor was any of the information in the 
contigAgp.zip or contigFa.zip files changed. For convenience, we include two new files, chromRandAgp.zip and
chromRandFa.zip, for users who would like to download only the data that has changed. These zips consist of
the updated chrN_random.agp and chrN_random.fa files, respectively.

<P>4 Sep. 2000: The files chromFa.zip and contigFa.zip under the
<A HREF="http://genome-archive.cse.ucsc.edu/goldenPath/17july2000/bigZips/">July 17th full data set </A> and the files under 
<A HREF="http://genome-archive.cse.ucsc.edu/goldenPath/17july2000/pieces/">
July 17th data by individual clone contig </A>
were updated to fix some incorrect (null(0))
characters that needed to be replaced by 'n' characters in some
Fasta files. The following contig Fasta files on chromosomes 1,8,11,12,16,
17 and 19 were affected:

<PRE>
1/ctg14250/ctg14250.fa
8/ctg17325/ctg17325.fa
8/ctg16307/ctg16307.fa
8/ctg25150/ctg25150.fa
8/ctg15216/ctg15216.fa
11/ctg15548/ctg15548.fa
12/ctg15285/ctg15285.fa
16/ctg16864/ctg16864.fa
16/ctg28077/ctg28077.fa
16/ctg13970/ctg13970.fa
16/ctg25401/ctg25401.fa
16/ctg13336/ctg13336.fa
16/ctg14937/ctg14937.fa
16/ctg16860/ctg16860.fa
16/ctg17794/ctg17794.fa
17/ctg14047/ctg14047.fa
19/ctg16564/ctg16564.fa
</PRE>

We apologize for any inconvenience this may have caused. 

	<BR><BR></TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
		
</TD></TR></TABLE>
</BODY></HTML>







