<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Bioinformatics: News Archive</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg"></TH></TR>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
	<table bgcolor="000000" cellpadding="1" cellspacing="1" width="100%" height="27"><tr bgcolor="2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
			<TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp;&nbsp;
			<A HREF="/index.html" class="topbar">
				Home</A> &nbsp; - &nbsp;
			<A HREF="/cgi-bin/hgGateway" class="topbar">
				Genome Browser</A> &nbsp; - &nbsp; 
			<A HREF="/cgi-bin/hgNear" class="topbar">
				Family Browser</A> &nbsp; - &nbsp; 
			<A HREF="/cgi-bin/hgBlat?command=start" class="topbar">
				Blat </A> &nbsp; - &nbsp; 
			<A HREF="/cgi-bin/hgText?command=start" class="topbar">
				Table Browser</A> &nbsp; - &nbsp; 
			<A HREF="/FAQ/" class="topbar">
				FAQ</A> &nbsp; - &nbsp; 
			<A HREF="/goldenPath/help/hgTracksHelp.html" class="topbar">
				Help</A> &nbsp;</font></TD>
	</TR></TABLE>
	</TD></TR></TABLE>
</TD></TR>

<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>	
	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
	<FONT SIZE="4"><B> &nbsp;News Archives </B></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0><TH HEIGHT=10></TH><TR><TD WIDTH=10></TD><TD>

<P>
<FONT SIZE=4 COLOR="#006666">Posted on Aug 13, 2003</FONT><BR>
		    The UCSC Genome Bioinformatics group has released a browser 
		    and blat server on the first of more than 100 targeted 
		    genomic regions being sequenced in multiple species and
		    analyzed by the NIH Intramural Sequencing Center (NISC) 
		    Comparative Sequencing Program sponsored by NHGRI. This
		    release coincides with the publication of the results of the
		    study in the 14 Aug 2003 issue of 
		    <A HREF="http://www.nature.com/cgi-taf/DynaPage.taf?file=/nature/journal/v424/n6950/abs/nature01858_fs.html" TARGET=_blank><em>Nature</em></A> 
		    (Thomas, 
		    J.W. et. al. (2003) Comparative analyses of multi-species 
		    sequences from targeted genomic regions. <em>Nature</em> 424:788- 793).
		    The browser displays sequence and annotations on a large 
		    region 
		    containing 10 previously identified genes - including the 
		    gene mutated in cystic fibrosis - in 13 vertebrate species. 
		    Organisms in the study include human, chimpanzee, 
		    baboon, cat, dog, cow, pig, rat, mouse, chicken, zebrafish 
		    and two species of pufferfish (Fugu and Tetraodon). 
		    <P>
		    The NISC Comparative Sequencing Program data may be accessed
		    by clicking the Browser link on the Genome Browser home page 
		    and then selecting the "Zoo" option from the genome list.
		    <P>
		    The research team, led by  NHGRI Scientific Director Eric 
		    D. Green, included scientists from Pennsylvania State 
		    University, University of California Santa Cruz (UCSC),
		    and the University of Washington in Seattle. In the study,
		    the investigators systematically compared the patterns of 
		    transposon insertions among the species' sequences. One
		    key result of the analysis was the confirmation of recently 
		    proposed mammalian evolutionary 
		    trees suggesting that primates are more closely related to 
		    rodents than to carnivores or artiodactyls. Another 
		    significant outcome was the discovery of a 
		    substantial number of previously unidentified non-coding DNA 
		    segments that are conserved across a wide range of species.
		    Many of these regions could be identified only through 
		    comparisons of sequence from multiple species, demonstrating
		    the importance of studying the genomes of a wide range of 
		    organisms as a means for identifying functional elements in
		    the human genome.
		    <P>
		    UCSC built a customized version of the browser to display the
		    target region for this study, allowing scientists to 
		    interactively explore the data and predictions generated 
		    by this project, contribute data of their own, and track 
		    the project as data from additional species are generated.
		    In addition to the browser, the UCSC team also 
		    contributed to the analytical portion of the 
		    study. Mathieu Blanchette
		    identified the regions that are most highly conserved among
		    species. Adam Siepel performed the phylogenetic analysis of
		    rates of substitution. The UCSC team worked with Arian Smit
		    to obtain definitive evidence that rodents branched off from 
		    the common ancestor later than carnivores and artiodactyls.
		    <P>
		    For more information on the NISC study, see the <em>Science 
		    Daily</em>
		    <A HREF="http://www.sciencedaily.com/releases/2003/08/030814072241.htm" TARGET=_blank>press release</A>. 
		    Flat files of the assembled sequence and annotations may be 
		    obtained from <A HREF="http://www.nisc.nih.gov/data/" TARGET=_blank>http://www.nisc.nih.gov/data/</A> or via the 
		    <A HREF="/downloads.html">Downloads</A> link on the Genome
		    Browser home page.
		    <P>
		    We'd like to thank the NISC Comparative Sequencing Program
		    team for providing the data and comparative analysis 
		    for this Genome Browser release. Special thanks go to
		    Elliott Margulies at NHGRI for serving as the main liaison
		    between NHGRI and UCSC, and for contributing several 
		    annotation tracks to the browser. We'd also like to 
		    acknowledge 
		    the efforts of the many faculty, grad students, and staff 
		    members of the UCSC Genome Bioinformatics group who 
		    contributed to the research effort and browser 
		    development for this project.

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Aug 8, 2003</FONT><BR>
		    The latest human genome reference sequence (NCBI Build 34,
		    July 2003) is now available as database hg16 in the UCSC
		    Genome Browser and blat server.
		    <P>
		    There are 
		    2,843,433,602 finished sequenced bases in the ordered and 
		    oriented portion of the assembly, which is an increase of 
		    0.4 percent, or approximately 11 Mb, over the Build 33 
		    assembly. 
		    Of particular note in this release is the addition of the 
		    pseudoautosomal regions of the Y chromosome. This sequence 
		    was taken from the corresponding regions in the X chromosome 
		    and is an exact duplication of that sequence.
		    <P>
		    Some sequence joins between adjacent clones in this assembly
		    could not be computationally 
		    validated because the clones originated from different 
		    haplotypes and contained polymorphisms in the overlapping 
		    sequence, or the overlap was too small to be to be reliable. 
		    In these instances, the sequencing center responsible for 
		    the particular chromosome has provided data to support 
		    the join in the form of an electronic certificate. The 
		    Build 34 certificates may be reviewed 
		    <A HREF="http://genome.cse.ucsc.edu/goldenPath/certificates/certificate_hg16.html">here</A>.
		    <P>
		    Bulk downloads of the data are available via FTP at 
		    <A HREF="ftp://genome.ucsc.edu/goldenPath/hg16">ftp://genome.ucsc.edu/goldenPath/hg16</A> 
		    or through the <A HREF="http://genome.ucsc.edu/downloads.html">Downloads</A> 
		    link.  We recommend that FTP be used rather 
		    than HTML for the download of large or multiple files.
		    <P>
		    We'd like to thank NCBI and the International Human Genome 
		    Sequencing Consortium for
		    furnishing the data, and the UCSC team members who 
		    contributed to this release: Terry Furey, Hiram Clawson, 
		    Heather Trumbower, Mark Diekhans, Robert Baertsch, 
		    Donna Karolchik, Jim Kent and our sysadmin team Patrick 
		    Gavin, Jorge Garcia, and Paul Tatarsky.

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Jul 14, 2003</FONT><BR>
                    The UCSC Genome Bioinformatics Group has released a browser
		    and BLAT server on the v. 3.1 rat genome assembly from the 
		    Rat Genome Sequencing Consortium. This
		    assembly (UCSC version rn3, June 2003) was produced by the 
		    Atlas group at <A HREF="http://hgsc.bcm.tmc.edu/projects/rat/" TARGET=_blank>Baylor Human Genome Sequencing Center</A> (HGSC). 
 		    This assembly is a minor update to the 3.0 release.
  		    Sequence changes affect only chromosomes 7 and X. No 
		    additional assembly releases are planned prior to the 
		    publication of the rat genome analysis papers. 
		    <P>
		    The 3.x assemblies reflect several sequence additions and 
		    software improvements over the previous 2.x assemblies, 
		    including the sequencing of over 1100 new BACs to cover gaps, 
 		    an improved marker set from the Medical College of Wisconsin, 
		    a new FPC map from the BC Genome Sciences Centre, and 
		    improved linking of bactigs. For detailed information and 
		    statistics about the 3.x assemblies, see the Baylor HGSC 
		    <A HREF="ftp://ftp.hgsc.bcm.tmc.edu/pub/analysis/rat/README"
		    TARGET=_blank>README</A>. 
		    <P>
		    Downloads of the rat sequence and annotation data are 
		    available at 
		    <A HREF="ftp://genome.ucsc.edu/goldenPath/rnJun2003/" TARGET=_blank>
		    ftp://genome.ucsc.edu/goldenPath/rnJun2003/</A> or via 
		    the Downloads link on this page. These data are made 
		    available with 
		    <A HREF="http://genome.ucsc.edu/goldenPath/credits.html#rat_use">specific conditions for use</A>.
		    <P>
		    We'd like to thank the Rat Genome Sequencing Consortium and
		    Baylor HGSC for providing this assembly, collaborators
		    from other institutions who have contributed annotations, 
 		    and Arian Smit for updating RepeatMasker for this release.
		    We'd also like
		    to acknowledge the contributions of several individuals at
		    UCSC, including Hiram Clawson, Heather Trumbower, Robert Kuhn,
		    Yontao Lu, Terry Furey, Mark Diekhans, Robert Baertsch, 
		    Donna Karolchik, Jim Kent, and our sysadmin team Jorge 
		    Garcia, Patrick Gavin, and Paul Tatarsky.
<HR>
<FONT SIZE=4 COLOR="#006666">Posted on Jun 24, 2003</FONT><BR>
	UCSC has just released browsers and blat servers for 2 worms: 
	C. elegans version WS100 (May 2003) and C. briggsae version 
	cb25.agp8 (July 2002). The browsers are based on sequence 
	obtained from <A HREF="http://www.wormbase.org/" TARGET=_blank>WormBase</A>.
	<P>
	We are pleased to add the nematodes to the roster of genomes 
	available on our site. C. elegans is a major model organism 
        used for biomedical research, and is the first multicellular 
        animal to have a fully sequenced genome. In contrast,  the 
        whole genome shotgun assembly of the C. briggsae genome is 
        estimated to have achieved 98% coverage. Draft chromosome 
        sequences are not available for C. briggsae, due to the lack 
        of dense chromosomal maps that allow assignment of 
        ultracontigs to chromosomal locations.  As a result, all data 
        in the C. briggsae browser maps to chrUn.  
	<P>
	Both worms played a significant role in the early history
	of the UCSC Genome Browser. The browser code originated with 
        a C script that displayed a splicing diagram for a gene 
        prediction from C. elegans.  Tracks for mRNA alignments and 
        for homology with C. briggsae were added, and the tool 
        morphed into the precursor of the Genome Browser, the 
        &quot;Intronerator&quot; (Kent, WJ and Zahler, AM (2000). 
	<A HREF="http://nar.oupjournals.org/cgi/content/abstract/28/1/91?ijkey=155fc1582751ebf7b01072beb40325b1d0682fdc&keytype2=tf_ipsecsha" TARGET=_blank>
	The intronerator: Exploring introns and alternative splicing 
        in C. elegans</A>. <em>Nucleic Acids Res.</em> 28: 91-93). 
	<P>
	Downloads of the C. elegans sequence and annotation data are 
	available at 
	<A HREF="ftp://genome.ucsc.edu/goldenPath/ceMay2003/" TARGET=_blank>
	ftp://genome.ucsc.edu/goldenPath/ceMay2003/</A>; C. briggsae 
	downloads can be found at 
	<A HREF="ftp://genome.ucsc.edu/goldenPath/cbJul2002/" TARGET=_blank>
	ftp://genome.ucsc.edu/goldenPath/cbJul2002/</A>. 
	Both genomes can also be downloaded via the Downloads link on 
	this page.
	<P>
	We'd like to thank the 
	<A HREF="http://genome.wustl.edu/" TARGET=_blank>Genome Sequencing Center 
	at Washington University in St. Louis</A> and the 
	<A HREF="http://www.sanger.ac.uk/" TARGET=_blank>Sanger Institute</A> 
	for their collaborative work in sequencing the C. elegans 
        and C.  briggsae genomes. Many thanks to the 
	<A HREF="http://www.wormbase.org/about/people.html" TARGET=_blank>WormBase consortium</A> 
	for making the worm sequence publicly available.
	We'd also like to acknowledge several UCSC people who 
        contributed to this release: Hiram Clawson (browser and 
        annotation tracks engineering), Jim Kent (WABA and 
        chaining/netting), Al Zahler (WABA), Heather Trumbower (QA 
        and project management), and Donna Karolchik (project 
        management and documentation).

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Jun 23, 2003</FONT><BR>
	You may notice that we've removed the Genome pulldown menu 
	and genome assembly information from our home page. 
	Genome-specific information and links, as well as genome 
	selection, are now available on the gateway pages for our 
	tools. To open up a gateway page, simply click the Browser, 
	Blat, or Tables link in the left sidebar.</P>

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on May 23, 2003</FONT><BR>
	Today we'd like to announce the release of a genome browser and BLAT
	server for the SARS coronavirus TOR2 draft assembly. The browser - which is based on
	sequence deposited into GenBank as of 14 April 2003 - provides seven
	annotations showing gene predictions, locations of putative proteins, and
	viral mRNA and protein alignments. Of particular note are the protein 
	structure analysis and predictions, determined by using the 
	<A HREF="http://www.cse.ucsc.edu/research/compbio/sam.html" TARGET=_blank>Sequence Alignment and Modeling (SAM) T02</A> tool. </P>

	<P>
	This browser marks a departure from our usual collection of vertebrate 
	genomes. Its inception was inspired by one of our engineers - Angie 
	Hinrichs - who was vacationing in New Zealand when the SARs draft assembly
	was initially released. Struck by the impact of SARS in that part of the 
	world, she downloaded the sequence and built the initial tracks from a
	terminal at an Internet cafe! The rest of the team joined in on the 
	grassroots effort, generating the additional annotations and SAM 
	T02 protein analyses and predictions. Victor Solovyev chimed in with 
	Fgenesv+ gene predictions from Softberry Inc. UCSC does not intend to 
	provide a comprehensive collection of viral genomes in the future, but 
	will maintain this browser as long as scientific and public interest in 
	SARS persists.</P>

	<P>
	Downloads of the annotation data are available
	at <A HREF="ftp://genome.ucsc.edu/goldenPath/scApr2003/database" TARGET=_blank>
	ftp://genome.ucsc.edu/goldenPath/scApr2003/database</A> or via the Downloads 
	link on this page. </P>

	<P>
	We'd like to thank everyone who worked on this release, including
	Angie Hinrichs, Robert Baertsch, Fan Hsu, Matt Schwartz, Heather Trumbower, Jim
	Kent, Kevin Karplus, Donna Karolchik, Brian Raney, Hiram Clawson, Kate 
	Rosenbloom, Victor Solovyev, and 
	our extremely dedicated systems administrators Paul Tatarsky, Patrick 
	Gavin, and Jorge Garcia.</P>

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Apr. 21, 2003</FONT><BR>
	The file that we originally used to build the agp files for the 
	April 2003 human release (Build 33) erroneously contained 2 contigs on 
	chromosome 8 that were listed twice: NT_078037, NT_008183.  We've 
	received a corrected version and have updated the following files on our 
	website: contigAgp.zip, chromAgp.zip, liftAll.zip. You can obtain the 
	newer versions of these files from our ftp site at 
	<A HREF="ftp://genome.ucsc.edu//goldenPath/10april2003/bigZips/">
	ftp://genome.ucsc.edu//goldenPath/10april2003/bigZips/</A>.

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Apr. 14, 2003</FONT><BR>
	The International Human Genome Sequencing Consortium today announced the
	successful completion of the Human Genome Project. The most
	significant outcome of this project is the reference sequence of
	the human genome. The sequencing of
	the 3 billion letters of DNA in the human genome - which many consider to 
	be one of the most ambitious scientific undertakings in history - was
	completed 2 years ahead of schedule and at substantially less cost than
	original estimates. The reference sequence will serve as a new foundation
	for research in the fields of medicine and human biology.
	<P>
	In conjunction with this announcement, the UCSC Genome Bioinformatics 
	group is proud to release a genome browser and BLAT server on the 
	reference sequence (NCBI Build 33), along with bulk downloads of the 
	sequence and annotation data. The initial browser provides a preliminary 
	set of annotations that will be expanded in coming weeks. Bulk downloads
	of the data are available via FTP at 
	<A HREF="ftp://genome.ucsc.edu/goldenPath/10april2003">ftp://genome.ucsc.edu/goldenPath/10april2003</A> 
	or through the <A HREF="../downloads.html">Downloads</A> link on this page. 
	We
	recommend that FTP be used rather than HTML for the download
	of large or multiple files.
	<P>
	The reference sequence covers about 99 percent of the human genome's gene-containing
	regions, and has been sequenced to an accuracy of 99.99 percent. The
	missing portions are essentially contained in less than 400 defined gaps
	that represent DNA regions with unusual structures that can't be reliably
	sequenced using current technology. The average DNA letter now lies within
	a stretch of approximately 27,332,000 base pairs of uninterrupted 
	sequence! 
	<P>
	Chromosomal sequences for this release were assembled by the 
	International Human Genome Sequencing Consortium sequencing centers and verified by 
	NCBI and UCSC. In some cases, sequence joins between adjacent clones 
	could not be computationally validated, e.g. due to polymorphisms in the 
	overlapping sequence. In situations like this, supporting evidence for 
	the join has been provided by the sequencing center responsible for that 
	particular chromosome. The <A HREF="certTable.html">
	Non-standard Join Certificates</A> table displays this information. The 
	annotations on the UCSC website have been provided
	by UCSC and collaborators worldwide. See the 
	<A HREF="credits.html">Credits</A> page for a detailed
	list of organizations and individuals who contributed to the success of 
	this release.

	<P>
	We'd like to congratulate the many people worldwide who have worked on 
	the Human Genome Project for this landmark achievement. We'd also like to acknowledge the UCSC
	Genome Browser project team who worked many long hours to ensure that
	the genome browser and sequence data were released on time for this
	announcement: David Haussler, Jim Kent, Terry Furey, Matt Schwartz, Heather Trumbower, 
	Angie Hinrichs, Fan Hsu, 
	Donna Karolchik, Jorge Garcia, Patrick Gavin, Chuck Sugnet, Yontao Lu, Mark Diekhans, Ryan Weber, Robert Baertsch, Krishna Roskin, and the many other students in the UCSC Genome Bioinformatics group.

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Apr. 2, 2003</FONT><BR>
	The Dec. 2001 Human assembly (hg10) and the Nov. 2001 Mouse assembly
	(mm1) have been moved to the archives. They are no longer accessible
	from the main browser, but instead can be found by clicking the 
	<A HREF="http://genome-archive.cse.ucsc.edu">Archives</A> link on this 
	page.

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Mar. 26, 2003</FONT><BR>
	We've added an updated rat assembly to our site: Rat Jan 2003 (rn2). This
	corresponds to the Version 2.1 Jan 2003 Update of the rat genome 
	assembly, produced by the Atlas group at Baylor HGSC as part of the Rat
	Genome Sequencing Consortium.
	<P>
	This update corrects duplications that were assembly artifacts in the 
	previous version and improves the linking of bactigs to create larger
	"ultrabactigs". Compared with the previous rat assembly, sequence mapped 
	to specific chromosomal coordinates is reduced by about 1.6 percent. 
	Loosely mapped and unmapped sequence is reduced by 17 percent. For more
	details and statistics on the Jan 2003 assembly, see the Baylor HGSC
	<A HREF="ftp://rat-ftp.hgsc.bcm.tmc.edu/pub/analysis/rat/README">
	README</A> for this release.
	<P>
	UCSC has released a Genome Browser and BLAT server for this assembly 
	update. The initial <A HREF="http://genome.ucsc.edu/cgi-bin/hgTracks?db=rn2">
	browser</A> contains 16 annotation tracks, with more to follow in coming 
	weeks. Sequence downloads are currently available at 
	<A HREF="ftp://genome.ucsc.edu/goldenPath/rnJan2003/" TARGET=_blank>
	ftp://genome.ucsc.edu/goldenPath/rnJan2003/</A> or via the Downloads 
	link on this page. A complete set of database downloads will be available at the beginning 
	of next week. This data contains 
	<A HREF="http://genome.ucsc.edu/goldenPath/credits.html#rat_use">specific 
	conditions for use</A>.
	<P>
	Thanks to the Atlas group at Baylor HGSC, the Rat Genome Sequencing Consortium,
	the UCSC Genome Bioinformatics group, and contributors worldwide for 
	making this release available.

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Mar. 13, 2003</FONT><BR>
	We're happy to announce an update to the mouse genome sequence. This new 
	version (Mouse Feb.  2003) includes 705 megabases of finished sequence, 
	compared to 96 megabases of finished sequence in the previous assembly. 
	Many people in the Mouse Genome Sequencing Consortium contributed to 
	this update. The Sanger Institute in particular contributed a large 
	amount of finished sequence. Richa Agarwala, Deanna Church, and 
	coworkers at NCBI layered the finished clones on top of the Arachne whole genome shotgun assembly. Arian Smit constructed a new RepeatMasker 
	library.  
	<P>
	UCSC has released a Genome Browser and BLAT server for the Feb. 2003 
	Mouse genome. The initial 
	<A HREF="http://genome.ucsc.edu/cgi-bin/hgTracks?db=mm3">
	browser</A> contains 14 annotation tracks, with more to follow in coming 
	weeks. Sequence downloads are currently available at 
	<A HREF="ftp://genome.ucsc.edu/goldenPath/mmFeb2003/" TARGET=_blank>
	ftp://genome.ucsc.edu/goldenPath/mmFeb2003/</A> or via the Downloads 
	link on this page. Database downloads will be available at the beginning 
	of next week. 
	<P>
	Thanks to everybody at UCSC and around the world that contributed to the 
	success of this release! 
<HR>


<FONT SIZE=4 COLOR="#006666">Posted on Feb. 5, 2003</FONT><BR>
        We're proud to announce the release of version 17 of the UCSC Genome
        Browser. 
        This version contains powerful new features, numerous improvements to 
        the annotation track display, additional annotation 
        tracks, and a number of bug fixes. In this release cycle, we've also 
	introduced   
        an enhanced QA process that formalizes our testing and verification
        of the Genome Browser software and the data displayed in the browser.
        <P>
        <B>New functionality in v.17:</B><BR> 
        -- Numerous enhancements to the table browser that allow the user to
        conduct more complex and specific searches. New features include support
        for intersections of tracks, a new summary statistics output format, and
        the ability to output query results as a custom annotation track that 
        can be viewed in the Genome Browser.
         The new 
        <A HREF="http://genome.ucsc.edu/goldenPath/help/hgTextHelp.html"
        TARGET=_blank> Table Browser User's Guide</A> contains a detailed
        description of the new features and provides a wealth of information and
        examples for conducting various types of searches on the database tables.
        <P>
        -- Two new display modes available for most annotation tracks: pack and
        squish modes. In <I>pack mode</I> display, annotation track features are f
ully    
        displayed, but more than one feature may be displayed on the same line.
        This greatly reduces the amount of display space needed by a track
        when a user wishes to view a large number of individual features at one
        time. <I>Squish mode</I> is similar to pack mode, but displays features
        at 50% height and without labels. This mode is particularly
        useful for viewing tracks in which a large number of features align to
        the same section of a chromosome, e.g. EST tracks.
        <P>
        -- Functional groupings of annotation track controls. This
        makes it much easier to find a particular item in the track control list
        and gives a better visual overview
        of the annotations available in a particular category, e.g. comparative
        genomics tracks or gene prediction tracks.
        <P>
        -- A mechanism for saving the annotation tracks image in postscript or
        PDF format. This much requested feature enables Genome Browser users to print an image at
        high resolution, edit it with a drawing program, or display it in a
        postscript or PDF viewer.
        <P>
        -- A <A HREF="http://genome.ucsc.edu/goldenPath/customTracks/custTracks.html"
        TARGET=_blank>collection of custom annotation tracks</A> supplied by
        Genome Browser users and members of the UCSC Genome Bioinformatics lab.
        Additional contributions to this collection are welcome! Contact
        <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A> if you have
        an annotation you'd like to share.
        <P>
        <B>New annotation tracks in v.17:</B><BR>
        -- a new Known Genes track (Human Builds 30 & 31, Feb 2002 Mouse,
        Nov 2002 Rat) that shows known
        protein coding genes based on proteins from SWISS-PROT, TrEMBL, and
        TrEMBL-NEW and their corresponding mRNAs from Genbank. Features within
        the track are color-coded according to origin and review status.
        <P>
        -- a new Superfamily track (Human Build 30, Feb 2002 Mouse) that shows proteins having
        homologs with known structures or functions. Each entry in the track
        shows the coding region of a gene (based on Ensembl gene prediction).
        The feature label consists of the names of all known protein domains coded 
        by the gene, and usually contains structural and/or function
        descriptions that provide valuable information for getting a quick grasp
        of the biological significance for the gene.
        <P>
        We have also released several additional annotation
        tracks on the latest human and rat assemblies in the past
        month.
        <P>
        <B>Bug fixes in v.17:</B><BR>
        -- Approximately 40 bugs (mostly minor problems) have been fixed in
        this version.
        <P>
        To take full advantage of the new display features in this release, we
        recommend that you reset your browser to the new default settings. NOTE:
        you may not want to reset your browser if you have customized settings
        that you wish to preserve. You can reset your browser by clicking
        the "Click here to reset" link on the Browser Gateway page.
        <P>
        We hope this new release facilitates your work with the UCSC Genome Browser.
        If you have any questions or comments about the new release, send email
        to <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>.
<HR>


<FONT SIZE=4 COLOR="#006666">Posted on Dec. 23, 2002</FONT>
	<P>
        We're pleased to announce the release of the latest human genome
        assembly, Build 31 (UCSC version hg13). This assembly was produced at NCBI
        based on sequence information submitted into GenBank as of Nov. 14, 2002.
        Release notes for this assembly are available from the
        <A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/human/release_notes.html" TARGET=_blank>
        NCBI web site</A>. Because UCSC now obtains its assembly directly from
        NCBI, the UCSC Build 31 data is identical to that of NCBI and Ensembl.
        <P>
        Build 31 is an excellent high-quality assembly that shows a remarkable
        amount of progress toward the milestone of finishing
        the human genome. Greater than 95% of the euchromatic region of the
        genome is now complete, with more than 90% of the sequence in a
        finished state. The number of clone contig gaps has decreased by one
        third from the previous assembly, and the overall number of sequenced
        contigs has been reduced by one half. Seven chromosomes are considered to
        be in a finished state: 6, 7, 13, 20, 21, 22, and Y.
        <P>
        The initial release of the Build 31 Genome Browser contains 25 annotation
        tracks, with several more to follow in the upcoming weeks. Bulk
        downloads of the data are available from our FTP site at
        <A HREF="ftp://genome.ucsc.edu/goldenPath/14nov2002" TARGET=_blank>
        ftp://genome.ucsc.edu/goldenPath/14nov2002</A> or via the
        <A HREF="../downloads.html">Downloads</A> link on this page.
        <P>
        UCSC has generated a set of high-level comparisons of the Build 31
        draft sequence against various types of information (STS maps, BAC end
        pairs, and clone overlaps). This information, as well as statistics for
        Build 31, is accessible from the
        <A HREF="http://genome.ucsc.edu/goldenPath/chromReports/14Nov2002/" TARGET=_blank>
        Chromosome Reports</A>,
        <A HREF="http://genome.ucsc.edu/goldenPath/mapPlots/" TARGET=_blank> Genome Map Plots</A>, and
        <A HREF="stats.html" TARGET=_blank> Summary Statistics</A>
        links in the "Technical Information about the Assembled Sequence"
        section below.
        <P>
        We'd like to thank NCBI as well as all the people who collaborated on the
        data and annotations for this release.
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Dec. 6, 2002</FONT>
	<P>
        We're pleased to announce the release of a UCSC Genome Browser on the
        Nov. 2002 rat assembly produced by the Baylor College of Medicine Rat
        Genome Sequencing Center and the Rat Genome Sequencing Consortium.
        The sequence was assembled using a hybrid approach that combines the
        clone by clone and whole genome shotgun methods. A new software program -
        ATLAS - was developed for this effort. The assembly process resulted in
        a 6.5-fold coverage of the rat genome, which is estimated to be
        approximately 2.8 Gigabases in size.
        <P>
        Downloads of the rat data and annotations are available through our
        ftp site at <A HREF="ftp://genome.ucsc.edu/goldenPath/rnNov2002" TARGET=_blank>ftp://genome.ucsc.edu/goldenPath/rnNov2002</A>
        or via the <A HREF="../downloads.html">Downloads</A>
        link on this page. This data contains
        <A HREF="credits.html#rat_use">specific
        conditions for use</A>. The sequence is also available from the
        <A HREF="http://www.hgsc.bcm.tmc.edu/projects/rat" TARGET=_blank> Rat
        Genome Project website</A> for the Human Genome Sequencing Center at
        Baylor College of Medicine or from GenBank.
        <P>
        We'd like to thank the Baylor team and the Rat Genome Sequencing
        Consortium for their collaboration on this project. See the
        <A HREF="credits.html#rat_credits">Credits page</A>
        for a complete list of acknowledgments.. For more
        information on the rat genome, the assembly process, and the
        Rat Genome Sequencing Consortium, refer to the website for the
        <A HREF="http://www.hgsc.bcm.tmc.edu/" TARGET=_blank>
        Human Genome Sequencing Center at Baylor College of Medicine</A>.
<HR>


<FONT SIZE=4 COLOR="#006666">Posted on Dec. 5, 2002</FONT>
	<P>
	The International Mouse Genome Sequencing Consortium has announced 
	the publication of a high-quality draft sequence of the mouse genome, 
	together with a comparative analysis of the mouse and human genomes. 
	The results from this analysis can be found in the Mouse Genome 
	Browser on this website. The paper appears in the Dec. 5 issue of the 
	journal Nature at 
	<A HREF="http://www.nature.com/nature/mousegenome/" TARGET=_blank>http://www.nature.com/nature/mousegenome/</A>.
	The co-author list includes several members of the UCSC Genome 
	Bioinformatics Group: CBSE Director David Haussler, Research 
	Scientist Jim Kent and research team members 
	Robert Baertsch, Mark Diekhans, Terrence Furey, Angie Hinrichs, Fan Hsu,
	Donna Karolchik, Krishna Roskin, Matt Schwartz, Charles Sugnet and Ryan 
	Weber. 

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Oct. 29, 2002</FONT>
	<P>
	We've added several new directories of downloadable data to the 28 June 
	2002 human genome assembly. These directories contain mouse/human 
	alignments of the June 2002 human assembly vs. the Feb. 2002 mouse assembly. You can access these 
	directories from our <A HREF="http://genome.ucsc.edu/downloads.html" TARGET=_blank>Downloads</A> 
	link or or via our ftp site at <A HREF="ftp://genome.ucsc.edu/goldenPath/28jun2002" TARGET=_blank>ftp://genome.ucsc.edu/goldenPath/28jun2002/vsMm2/</A>. 
	<P>Within the main directory <I>vsMm2</I> are 3 subdirectories that contain all 
	the alignments (<I>axtAll</I>), alignments filtered to provide only the best alignment 
	for any given region of the human genome (<I>axtBest</I>), and a relatively stringent 
	subset of the axtBest alignments (<I>axtTight</I>). For more information 
	about the format of the alignment files and the methods used to generate 
	the alignments, consult the <I>README.txt</I> file in the vsMm2 directory.

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Oct. 18, 2002</FONT>
	<P>
	We've rolled out a new version of the Genome Browser - v.16. In 
	addition to several bug fixes, this release contains some interesting 
	new features. 
	<P>
	The Table Browser has undergone major enhancements. Users 
	can now restrict their queries by specifying a value or range for any of 
	the fields in a table, and by selecting which fields should be displayed in 
	the output. The Table Browser also provides the ability to do a free-form SQL 
	query on a table and supports several new output formats. 
	<P>	
	We've extended the capabilities of the DNA retrieval functionality
	in the Genome Browser and the Table Browser. The new 
	mechanism offers the user several options for configuring the amount and 
	type of sequence region that is retrieved, and options for formatting the 
	sequence output. The retrieval options vary based on the type of table 
	selected.
	<P>	
	The Genome Browser's gene prediction tracks now offer a Comparative 
	Sequence link in addition to the predicted protein, mRNA sequence, and 
	genomic sequence links. The Comparative Sequence feature displays 
	annotated codons and translated protein for the region in alignment to 
	another species. 
	<P>	
	We've recently added a few new tracks/tables to the hg12 and mm2 Browsers. On the 
	latest human assembly (hg12), we now have Chimp Blat and Chimp BAC tracks 
	provided by Ingo Ebersberger, Joshua Bacher, and Svante P‰‰bo at the Max
	Planck Institute for Evolutionary Anthropology. Jim Kent at UCSC has generated a 
	new annotation - Gene Boundaries - that shows the boundaries of genes 
	and the direction of transcription as deduced from clustering spliced 
	ESTs and mRNAs against the genome. Daryl Thomas of UCSC has also added SNP tracks to both 
	hg12 and the latest mouse assembly (mm2), based on data from the SNP
	Consortium and NIH.
	<P>	
	We encourage you to experiment with these new features. Comments, 
	questions, and suggestions are always welcome at 
	<A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>. 

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Oct. 17, 2002</FONT>
	<P>
	We have found errors with the RepeatMasker track on the 
	Feb. '02 mouse assembly (mm2). This problem affects the RepeatMasker track 
	and RepeatMasked DNA obtained via the browser's DNA links. It does 
	<I>not</I> affect data downloaded from the browser's downloads page or ftp 
	site. We have replaced the erroneous data set with a corrected version. If you have questions about how this change may affect your project, email <A HREF=mailto:genome@soe.ucsc.edu>genome@soe.ucsc.edu</A>.

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Oct. 4, 2002</FONT>
	<P>
	Please note that the coordinate range for a portion of the Build 30 chr22 in 
	the UCSC Genome 
	Browser differs from that of NCBI and the Sanger Centre. The latter version 
	contains a 100K bp gap 
	inserted into chr22 just after the centromere. This
	modification was made after UCSC released the Build 30 assembly in the
	Genome Browser. All of the chr22 annotations displayed by the UCSC
	browser are correctly positioned relative to one another. However the
	coordinates of all features past the centromere will be 100K less than those 
	of NCBI and Sanger. The Ensembl Genome Browser is consistent with the
	UCSC browser.

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Sep. 19, 2002</FONT>
	<P>
	We're pleased to announce the release of the Mouse Cons (Human/Mouse
	Evolutionary Conservation Score) annotation track for the <A
	HREF="http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg12">June 2002
	human genome assembly</A>. This track allows a user to interactively explore 
	conservation between the human and mouse genomes and identify highly conserved 
	regions. Highest levels of conservation are typically seen over coding exons. 
	High levels of conservation are also frequently associated with noncoding RNA,
	promoters, other regulatory elements, and pseudogenes.

	<P>The Mouse Cons annotation is displayed using a new type of Genome
	Browser graphical track that plots a continuous function along a chromosome. The
	conservation levels are calculated over 50bp  windows in the human genome
	that have at least 15 bp aligned to mouse. The score for a window reflects
	the probability that the level of observed conservation in that 50bp region
	would occur by chance under neutral evolution. This information is given on
	a logarithmic scale and displays in the track as "mountain ranges". Details
	pages associated with the individual peaks in the track provide access to
	the base level alignments for the whole region and for the individual 50bp
	windows.

	<P>The team that produced this track includes Ryan Weber, Krishna Roskin,
	Mark Diekhans, Jim Kent, Scott Schwartz, and Webb Miller.
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Sep. 12, 2002</FONT>
	<P>
	<I>Nature Genetics</I> has just published <A
	HREF="http://www.nature.com/cgi-taf/DynaPage.taf?file=/ng/journal/v32/n1s/index.html"
	TARGET=_blank>User's Guide to the Human Genome</A>, a
	hands-on tutorial for using genome browsers as web tools for browsing and
	analyzing data from the Human Genome Project and other sequencing efforts. The 
	3 browsers featured in the tutorial include the UCSC Genome Browser, NCBI's 
	Map Viewer, and the Ensembl Genome Browser. The guide is organized around a 
	collection of step-by-step solutions to 13 typical research questions,
	and serves as a nice supplement to the documentation materials
	available on the UCSC Genome Browser web site. The guide is accessible as a 
	link off the <I>Nature Genetics</I> home page at 
	<A HREF="http://www.nature.com/ng/" TARGET=_blank>http://www.nature.com.ng</A>. 
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Sep. 12, 2002</FONT>
	<P>
	We've corrected a problem with the Human June 2002 (hg12) cytoBand annotation 
	track that affected chromosome Y. The clones on this chromosome were
	erroneously pushed from the q-arm onto the p-arm, creating some confusion. The
	currently available version of the cytoBand data on this website contains this 
	correction.

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Aug. 15, 2002</FONT>
	<P>
	We've just released several new annotation tracks/tables for the human genome. 
	On the June 2002 assembly, we've added Gene Bounds, 
	UniGene, CpG Islands, Nonhuman mRNA & EST, SNPs, NC160, and GNF Ratio.
	On the April 2002 assembly we've added a Fgenesh++ Genes track, and have also
	updated the RepeatMasker track. The Dec. 2001 release now includes a Sanger 22 
	track. 

	<P>On the mouse genome, we have 2 new tracks for the Feb. 2002 assembly:
	TIGR Gene Index and RNA Genes.
<HR>
	
<FONT SIZE=4 COLOR="#006666">Posted on Aug. 6, 2002</FONT>
	<P>
	We have fixed an error with six of the chr3 contigs in the bigZips/contigAgp.zip 
	file. The following .agp files were corrected: NT_005684.agp, NT_005663.agp, 
	NT_022554.agp, NT_022459.agp, NT_006031.agp, and NT_022419.agp. The
	chr3.agp file in bigZips/chromAgp.zip was also modified. This change does not 
	affect the .gl files, the .fa files, the 
	lift files, or the annotations. Alignments made on the previous version
	of chr3 are still good. Updated versions of the contigAgp.zip and
	chromAgp.zip files were
	uploaded to our site today. You can download the new versions via ftp at <A 
	HREF="ftp://genome.ucsc.edu/goldenPath/28jun2002" TARGET=_blank>
        ftp://genome.ucsc.edu/goldenPath/28jun2002</A> or via the Downloads
	link in the left sidebar.
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on Aug. 2, 2002</FONT>
	<P>The problems with the June 2002 Build 30 (hg12) RepeatMasker track have been 
	resolved. The new RepeatMasker track, along with regenerated Fish Blat
	and Genscan tracks, are now available in the Browser and 
	through our
	Downloads link. We've also added a few new annotation tracks for the
	June 2002 release, and will be adding more over the next 2 weeks.

	<P>The latest Genome Browser has 2 new features. We've added filter
	functionality to the Table Browser, accessible via the Filter Fields
	button on the Table Browser main page. Also, some of the Dec. 2001 human genome tracks (eg. RefSeq Genes) now have a
	Comparative Sequence link from the details page that shows annotated
	codons and translated protein with alignment to the mouse genome.
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on July 27, 2002</FONT>
	<P>
	We've experienced some RepeatMasker problems on Build 30 and are
	rerunning it. This will directly affect the RepeatMasker track and the
	masking of the fasta files. The Fish Blat and Genscan tracks may also
	change slightly once we're redone this. The EST, mRNA, and RefSeq tracks
	should not be affected. We will also post a new RepeatMasker track for
	Build 29 (see news item below) as soon as the Build 30 tracks are
	completed. We apologize for any rework this may cause.
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on July 26, 2002</FONT>
	<P>
	Bulk downloads of the June 2002 Build 30 human genome assembly (hg12) are now
	available from
	<A HREF="ftp://genome.ucsc.edu/goldenPath/28jun2002" TARGET=_blank>
	ftp://genome.ucsc.edu/goldenPath/28jun2002</A>. You can also 
	access the data via the Downloads link in the left sidebar. This initial
	release of the annotation database download contains a limited set of
	tables. Additional files will be available for download next week.
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on July 24, 2002</FONT>

	<P>
	The BLAT server and the coordinates conversion feature for human genome 
	assembly Build 30 (hg12) are now functional.
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on July 23, 2002</FONT>
	<P>We're pleased to announce the pre-release of a browser for 
	human genome assembly Build 30 from NCBI (UCSC version hg12). This assembly was 
	produced at NCBI based on sequence information submitted into GenBank as of
	June 28, 2002. Build 30 release notes and statistics will soon be
	available from the <A
	HREF="http://www.ncbi.nlm.nih.gov/genome/guide/human/" TARGET=_blank> NCBI web 
	site</A>.  
	<P>
	Build 30 is an excellent high-quality assembly. It contains nearly 87%
 	finished sequence, and 94%-97% coverage. The sequence coverage of this build
	is much higher than in previous releases, and there is a high level of
	correspondence between the sequence and the map. Currently, the human genome
	project appears to be on track to achieve the goal of finishing at least 95%
	of the human genome (using Bermuda standards) by April 2003.
	<P>
	UCSC has generated a set of high-level comparisons of the Build 30 draft
	sequence against various types of information (STS maps, BAC end pairs, and
	clone overlaps), accessible from the <A 
	HREF="http://genome.ucsc.edu/goldenPath/chromReports/24Jun2002/"
	TARGET=_blank>Chromosome Reports</A> and <A 
	HREF="http://genome.ucsc.edu/goldenPath/mapPlots/" TARGET=_blank>Genome 
	Map Plots</A> links in the "Technical Information about the Assembled 
	Sequence" section below.
	<P>
	A Blat server for Build 30 is not yet available, but should be
	accessible from this site later this week. Data for the mitochondrial genome 
	and several more annotation tracks will be posted for this release as they 
	become available.  Bulk downloads of the hg12 data should be available from 
	this site in a few days. 
<HR>

<FONT SIZE=4 COLOR="#006666">Posted on July 9, 2002</FONT>
	<P>
	We've found some problems with the repeat-masking of the Build 29 (hg11)
	human sequence. We're in the process of replacing the RepeatMasker
	track, but do not plan to redo the other tracks due to the imminent
	release of Build 30. Because of this, we advise that you do not use the
	cross-species tracks for statistical purposes.

<hr>

<FONT SIZE=4 COLOR="#006666">Posted on July 1, 2002</FONT>
	<P>
	The UCSC Genome Bioinformatics home page is sporting an updated interface to 
	accommodate the growing number of organisms supported by the UCSC Genome 
	Browser, BLAT, and Table Browser. The list of assembly versions accessible 
	through each of these tools can now be found on the tool's Gateway page. To 
	reach the Gateway page, choose an organism from the dropdown list on the left 
	sidebar of this page, then click the Browser, BLAT, or Tables link. New 
	organisms will be added to the list in the months ahead.

	<P>The UCSC site continues to provide a variety of bulk downloads of a genome 
	assemblies and annotations. The list of downloadable data has been removed 
	from the home page, but is readily available through the Downloads link on 
	the left sidebar.  The downloads list can also be accessed directly at 
	<A HREF="../downloads.html">http://genome.ucsc.edu/downloads.html</A> or through 
	our ftp site at 
	<A HREF="ftp://genome.ucsc.edu/goldenPath/" TARGET=_blank>ftp://genome.ucsc.edu/goldenPath/</A>.

	<P>Several new annotation tracks have been added to our site in the past 
	couple weeks.  The Feb. 2002 mouse assembly now has tracks for BAC End pairs, 
	Fgenesh++ gene predictions, and AltGenie gene predictions based on Affymetrix's 
	Genie gene-finding software. New to the Apr. 2002 human assembly is the 
	GenMapDB Clones track, which shows placements of BAC clones from the GenMapDB 
	database based on BAC end sequencing information and confirmed using STS 
	markers by Vivian Cheung's lab at U. Penn.  We've also changed the Known Genes 
	track name to RefSeq Genes in all assemblies.

	<P>This release also includes an updated 
	<A HREF="http://genome.ucsc.edu/goldenPath/help/hgTracksHelp.html" TARGET=_blank>User's Guide</A>
	and more detailed documentation on creating & using 
	<A HREF="http://genome.ucsc.edu/goldenPath/help/customTrack.html" TARGET=_blank>custom annotation tracks</A>.

<HR>

<FONT SIZE=4 COLOR="#006666">Posted on May 24, 2002</FONT>
	<P>
	Bulk downloads of the April 2002 hg11 human genome assembly (NCBI Build 29) 
	are now available from <A
	HREF="ftp://genome-archive.cse.ucsc.edu/goldenPath/05apr2002"
	TARGET=_blank>ftp://genome-archive.cse.ucsc.edu/goldenPath/05apr2002</A>. You
	can also access the data from the Downloads link in the left sidebar.

<hr>
<FONT SIZE=4 COLOR="#006666">Posted on May 22, 2002</FONT>
	<P>
	We've just released a browser and BLAT server on 
	the latest Build 29 human genome assembly from NCBI (UCSC 
	version hg11). This assembly is based 
	on sequence information submitted into GenBank as of Apr. 5 2002. As with the 
	Dec. 2001 (hg10) release, this assembly was produced at 
	NCBI rather than at UCSC.  Consult NCBI's 
	<A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/human/release_notes.html"
	TARGET=_blank>Build 29 release notes</A> and 
	<A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/human/HsStats.html"
	TARGET=_blank>statistics</A> for more information about this release.
	This assembly contains nearly 75% finished sequence. 
	Currently, the human genome project appears to be on track to achieve the goal 
	of finishing at least 95% of the human genome (using Bermuda standards) by April 2003.
	
	<P>Although the NCBI human genome assembly has been steadily
	improving over the past year, mapping problems still exist in
	the current release. Most are small, relatively local rearrangements. 
	Larger scale problems include a rearrangement in the p-arm of Chr16 and
	several discrepancies in Chr17.  Researchers - especially positional 
	cloners - are strongly encouraged to use the tools provided 
	(<A HREF="http://genome.ucsc.edu/goldenPath/mapPlots/"
	TARGET=_blank>comparison plots</A>, <A
	HREF="http://genome.ucsc.edu/goldenPath/chromReports/"
	TARGET=_blank>chromosome reports</A>) to evaluate the
	accuracy of the assembly in specific regions of interest.

	<P>Bulk downloads of the hg11 data should be available from this site
	in approximately one week. New annotation tracks will be posted as soon
	as they become available.
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on Apr 24, 2002</FONT>
	<P>Bulk downloads of the February 2002 mouse genome assembly are now available 
	from <A HREF="ftp://genome.ucsc.edu/goldenPath/mmFeb2002" 
	TARGET=_blank>ftp://genome.ucsc.edu/goldenPath/mmFeb2002</A>.  You can 
	also access the data from the Downloading the Mouse Genome section below.<br></P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on Apr 19, 2002</FONT>
        <P>The February 2002 mouse genome assembly is now available in the browser and 
	for BLAT searching.  This assembly was produced at the Whitehead Institute 
	using their Arachne software.  We'd like to thank them and the Mouse Genome 
	Sequencing Consortium for providing this assembly, which has 
	<A HREF="credits.html#mouserestrict">specific conditions for use</A>.
	Bulk downloads of the data should be available in approximately one week.  
	Coordination with mouse genome data access at 
	<A HREF="http://www.ensembl.org/Mus_musculus/">Ensembl</A> and 
	<A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/M_musculus.html">NCBI</A> is 
	in progress.  We'd also like to acknowledge the UCSC team that produced this 
	release: Jim Kent, Terry Furey, Matt Schwartz, Fan Hsu, Yontao Lu, 
	Donna Karolchik, Chuck Sugnet, and Ryan Weber.<br></P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on Apr 9, 2002</FONT>
        <P>Bulk downloads of the November 2001 mouse genome assembly are now available 
	from <A HREF="ftp://genome-archive.cse.ucsc.edu/goldenPath/mmNov2001"
        TARGET=_blank>ftp://genome-archive.cse.ucsc.edu/goldenPath/mmNov2001</A>.
	You can also access the data from the Downloading the Mouse Genome section 
	below.<br></P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on Apr 2, 2002</FONT>
	<P>An updated version of the UCSC Genome Browser (v.11) is now available.
	Along with the v.11 browser, we've released several new annotation tracks on 
	the latest human and mouse assemblies. The new Human Dec. 2001 tracks include:
	Mouse Synteny, Ensembl, Genscan, CpG Islands, Mouse Blat, Fish Blat, 
	Unigene/SAGE, NCI60 Microarray, GNF Affymetrix Microarray, Rosetta
	Microarray, and SNPs. An STS Markers track has been added to the Mouse
	Nov. 2001 browser.<br></P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on Mar 14, 2002</FONT>
	<P>The November 2001 mouse genome assembly is now available
	for viewing in the browser and for BLAT searching. This assembly
	was produced at the Sanger Center using the Phusion software developed
	by Jim Mullikin and Zemin Ning, and was tied to the 
	<A HREF="http://www.ensembl.org/Mus_musculus/assembly.html"> mouse 
	fingerprint map</A> by Tim Hubbard. We'd like to
	thank them and the Mouse Genome Sequencing Consortium for providing this
	assembly, which has <A HREF="credits.html#mouserestrict">specific 
	conditions for use</A>. Bulk downloads of the data
	will be available in approximately one week. Coordination with
	mouse genome data access at 
	<A HREF="http://www.ensembl.org/Mus_musculus/">Ensembl</A> and 
	<A HREF="http://www.ncbi.nlm.nih.gov/genome/guide/M_musculus.html">NCBI</A> 
	is in progress.  We'd also like to acknowledge the UCSC team that produced 
	this release: Jim Kent, Terry Furey, Matt Schwartz, Fan Hsu, Yontao Lu, 
	and Donna Karolchik.<br></P>
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on Feb 16, 2002</FONT>
	<P>A new assembly based on sequence submitted as of Dec. 22 in Genbank (Build 28) is
	now available in the browser and for BLAT search.  
	This assembly was produced at NCBI rather than UCSC, primarily by
	Richa Agarwala, Greg Schuler, and Paul Kitts.  The NCBI assembly has
	been steadily improving over the past year.  Currently it shows slightly
	better local order and orientation compared to the UCSC assembly on the
	same sequence, but somewhat worse tracking of the chromosome level maps.
	The NCBI assembly has the advantage that it can be generated significantly
	faster than the UCSC assembly.  With the human genome sequencing now
	in the end game - over two thirds of the human clones are now finished -
	we feel it more productive to focus worldwide annotation efforts on a
	single assembly rather than continue producing competing assemblies.
	We're working with NCBI to improve their map tracking. <br></P>

<hr>

<FONT SIZE=4 COLOR="#006666">Posted on Feb 4, 2002</FONT>
	<P>Chromosome Reports detailing correspondence with STS map, overlap, 
	and BAC end sequence information are available under the "Technical 
	Information About the Assembled Sequence" section below.  This also 
	gives information about the clone map on which the assembled sequence 
	is based.<br></P>		
<hr>

<FONT SIZE=4 COLOR="#006666">Posted on Dec 18, 2001</FONT>
        <P>There are some major enhancements to the browser. The complete user interface settings
		including track controls, labels, and position are now saved from session to session.
		You can configure the browser once to your liking and it will stay that way. This feature
		will only work if cookies are enabled in your browser. If you
		want to restore the default settings use the reset all button under the main graphic.
		Also under the main graphic are new controls that move just the start or just the end
		of the genome window. These are useful for getting exactly the right view without having
		to do arithmetic on the position.  These controls by default will move two guideline units
		at a time, but you can specify other increments.  There's a new page associated with each
		track. This page is accessible by clicking on the mini-buttons to the left of the track 
		in the main graphic, or by clicking on the new hyperlink associated with the track in the
		track controls section under the graphic.  These pages contain a description of the track
		and in many cases new controls.  The mRNA and EST associated controls let you color or
		filter the display according to tissue, author, organism, and so forth.</P>
		<P>As with any new enhancement there are likely to be a few new bugs too.
		Many of these have been spotted and fixed already.  Please let us know (genome@soe.ucsc.edu) 
		if you find a problem that persists more than a day or two.  It's always helpful to include 
		the freeze and genomic position with a problem report.<br></P>
<hr>
<FONT SIZE=4 COLOR="#006666">Posted on Nov 30, 2001</FONT>
        <P>There is now a link from the known genes details page to the Jackson Lab's MGI Mouse Ortholog 
		when the ortholog is known.  Thanks to Carol Bult for her help setting up this link.<br></P>
<hr>
<FONT SIZE=4 COLOR="#006666">Posted on Nov 29, 2001</FONT>
        <P>A duplications track is now available in the August browser.  This track 
		shows duplicate blocks of sequence larger than 1000 bases.  The track is hidden
		by default.  To open it look for 'Duplications'  in the third row of track controls 
		under the main graphic window, and change the setting to 'dense'.  Thanks to Evan Eichler 
		and Jeff Bailey for this track.<br></P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Nov 28, 2001</FONT>
        <P>Sanger curated gene annotations are now available on chromosome 20.  Thanks to 
		Jennifer Ashurst, James Gilbert, and all the annotators at the Sanger Institute.<br></P>
<hr>		
<FONT SIZE=4 COLOR="#006666">Posted on Nov 27, 2001</FONT>
        <P>A new track has been added to the August freeze browser showing
		haplotype blocks derived from common SNPs on Chromosome 21 by Perlegen,
		as described in "<A HREF="http://www.sciencemag.org/cgi/content/abstract/294/5547/1719">Common 
		High-Resolution Haplotypes</A>." Patil, N. et. al. Science 294:1719-1723 (2001).<br></P>
<hr>		
<FONT SIZE=4 COLOR="#006666">Posted on Nov 19, 2001</FONT>
        <P>The SNP and Mouse Blat tracks are now available
		for August.  The Mouse Blat track uses a partial assembly of
		the public whole genome shotgun data courtesy of Whitehead's 
		Arachne program.  <br></P>
<hr>		
<FONT SIZE=4 COLOR="#006666">Posted on Nov. 8, 2001</FONT>
        <P>The detail web pages for each of the tracks have been updated to reflect
        the overall look and design of this site.  You will now see the familiar
        blue navigation bar with links to the Browser, BLAT, Downloads, and the  
        FAQ page from each of the track detail pages.<br></P>
<hr>	        
<FONT SIZE=4 COLOR="#006666">Posted on Nov. 8, 2001</FONT>
        <P>The STS Markers track has been updated on the April and August browsers to
        now include much more information on the detail page including links to
        UniSTS and details on the alignments of the markers to the draft sequence.
        In addition, all known aliases of the markers can be entered in the
        "position" window, and the corresponding merker will be found and
        displayed if its location has been determined.<br></P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Nov. 8, 2001</FONT>
        <P>A new FISH Clones track has been added to the April and August browsers.
        Previously, this information has been included in the STS Markers track.
        Now, this has been broken out into a separate track with additional
        information provided on the detail page not previously shown.<br></P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Nov. 6, 2001</FONT>
        <P>The fgenesh++ gene prediction and the cross-species
        mRNA tracks are now available in the August browser.<br></P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct 31, 2001</FONT>
	<P>The 'DNA' button at the top of the browser has been significantly 
	upgraded.   By default it now returns DNA that has repeating elements 
	in lower case and other DNA in upper case.  There is also an option to 
	color the DNA output with various tracks.  You can have the case and font 
	features such as underline, bold, and italic follow tracks too.<br></P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct 29, 2001</FONT>
	<P>There is now a TIGR Gene Index track
	in the April 2001 freeze browser.  The TIGR Gene Index is based
	on alignments of assembled ESTs from a number of species.
	Be sure to click into the track and follow the outside links to
	the TIGR site, which contains a wealth of information on the genes.<br></P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct 29, 2001</FONT>
	<P>The Acembly track on the August 2001 freeze
	has been updated to include predictions based on human ESTs 
	and Genbank  mRNAs as well as RefSeq human mRNAs.  Protein 
	predictions are now also available in the details page for this
	track.  The outside link for this track is also very informative.<br></P>
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct 27, 2001</FONT>
        <P>You can now share your custom tracks with the
        community. The easiest way to do this is to construct a link from your own
        web pages to the browser.  Here is an example of a URL for
        such a link:
        <pre>   http://genome.ucsc.edu/cgi-bin/hgTracks?
        position=chr22:1-20000&db=hg8&
        hgt.customText=http://genome.ucsc.edu/test.bed</pre>

        The position variable tells the browser which part of the genome to
        display.  The db variable refers to the freeze number.
        'hg8' corresponds to the August 2001 freeze.  The customText
        variable should refer to a URL containing plain text in one of the
        formats described in <a href="help/customTrack.html">
        http://genome.ucsc.edu/goldenPath/help/customTrack.html</a>.
        <br><br>
        Note that generally we only keep the last three versions of the
        genome online (hg6, hg7, and hg8).  You'll have to update
        your link and track about every 4 months as a result.
        <br><br>
        Please send in the URLs of tracks you'd like to share to
        <a href="mailto:genome@soe.ucsc.edu"> genome@soe.ucsc.edu</a>,
        along with a brief description of
        the track and the genome version it is tied to.  We'll create
        an index page of these here.
    <br><br>
        You can also access an external custom track by including
        the URL of the track data (on a separate line starting with
        http://) in the custom track box at the bottom of the browser gateway.</P>

<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct. 15, 2001</FONT>
        <P>Fresh tab-delimited files from the browser database
        are now available.  They will now be updated automatically
        every Sunday evening.   The table browser queries the database
        directly, so it is always up to the minute.</P>

<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct. 12, 2001</FONT>
	<P>Several new sets of gene predictions came in this week.
	We now have fgenesh++ predictions for the April freeze,
	and Genscan and Acembly predictions for the August freeze.</P>

<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct. 11, 2001</FONT>
	<P>Why struggle with massive genomic file downloads when the UCSC 
	<A HREF="http://genome.ucsc.edu/goldenPath/hgText.html">Table Browser</A> 
	lets you select exactly the track data desired via a convenient web 
	interface? Major new improvements by Krish Roskin have empowered and 
	simplified this feature available now for the three most recent assemblies.</P>

<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct.8, 2001</FONT> 
	<P>The October 2000 assembly has been moved to the 
	<A HREF="http://genome-archive.cse.ucsc.edu">Archives</A>
	to make room for the August assembly.</P>
	
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct.5, 2001</FONT> 
	<P>A revised August 2001 freeze assembly is now up.  The problems
	with flipped contigs of finished clones and high levels of
	sequence duplication are fixed.   You can now download this
	assembly <A HREF="ftp://genome-archive.cse.ucsc.edu/goldenPath/06aug2001/bigZips">in bulk</A> as 
	well as browse it.   Chromosome by chromosome and annotation
	database files will be following over the next day or two.</P>

<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Oct.2, 2001</FONT>
	<P>You can now <A HREF=/cgi-bin/hgCoordConv>convert
	coordinates</A> between different versions of the draft
	using a new program, hgCoordConv, by Chuck Sugnet.
 	hgCoordConv attempts to cut out sequences of the original 
	draft and align them to the new draft.  When aligning the 
	sequences to the new draft hgCoordConv makes sure that the
	sequences are in the same order, orientation, and have the 
	correct distances between them. </P>
	
	<P>Chuck has also implemented a SAGE/Unigene track in the browser. 
	This track displays data from the <A HREF= http://www.ncbi.nlm.nih.gov/SAGE/>
	SAGEMap project at NCBI </A>.   UniGene cluster sequences are
	displayed in the browser and colored according to their average SAGE
	counts over a series of experiments. Selecting one of the UniGene
	representative sequences displays the SAGE results for UniGene 
	sequences.</P>
	
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on Sept. 21, 2001:</FONT>
	<P>Some systematic problems were found in the clone order on the
	preliminary August 2001 freeze assembly. The sequencing center at
	Washington University, EBI's Ensembl group, and our group here at UCSC
	are currently working together to revise the merged BAC clone maps and
	the assembly process to fix these problems. We hope to update the August
	browser with a revised assembly soon. Then, after further testing via
	the browser, we will release the assembled August freeze genome sequence
	itself. </P>
	
<hr>	
<FONT SIZE=4 COLOR="#006666">Posted on 9/11/01:</FONT>	
	<P>A preliminary assembly of the August 2001 freeze is now
	available in the genome browser. Due to significant progress 
	by the mapping and finishing groups of the international 
	public consortium, this assembly is a major improvement over 
	the April 2001 freeze assembly.	Imre Vastrik, Ewan Birney and 
	colleagues at Ensembl have computed a merge of BAC clone maps
	provided by the individual sequencing centers with
	fingerprint-based maps prepared at Washington University. These
	merged maps were used for the first time in this August 
	assembly.</P>

	<P>The August assembly has successfully passed our internal 
	quality control tests. We will release the sequence and
	annotations in bulk downloadable form in a week or so, after 
	the external testers have had a chance to further verify
	it. Meanwhile if you notice any systematic problems please let 
	us know at genome@soe.ucsc.edu. Though the state of the working
	draft has improved considerably, remember that where you see 
	solid marks in the 'gap' track, the relative order and 
	orientation of flanking contigs is still uncertain. In some 
	cases of complex repeat structure it is also possible that
	the assembly may be incorrect even in the absence of gaps.
	Also, sometimes ambiguities in the data cause a BAC clone to be 
	split, with parts of it placed at opposite ends of a run of other clones.	
	Localized errors of this type should be corrected by additional 
	finishing efforts at the individual sequencing centers and 
	should not be reported to UCSC. However, please report any
	large-scale or systematic problems you detect with this assembly	
	that could have been caused by our data processing.</P>

	<P>The tracks available on the August 2001 browser are quite
	limited at the moment. More tracks will show up over time.</P>

<hr>
<FONT SIZE=4 COLOR="#006666">Archived on 9/11/01:</FONT>	
                <P>The April 2001 assembly is now the default for the browser.
        The SNP and Ensembl gene tracks have come in for this version.
        There is also a new track depicting non-human vertebrate mRNA
        alignments.</P>
                <P>The Sept. 2000 and July 2000 versions of the genome are        
        now only available on our archive site.  Please see the link
        in the blue box to the left for more details.</P>
                <P>The August 6 freeze is progressing through the pipeline.
        We've recently received an updated accession map from Wash U.
        Ensembl will shortly be integrating this with chromosome
        specific maps from the sequencing centers.  We are still
        on track for an early September next release.  </P>
<hr>
<FONT SIZE=4 COLOR="#006666">Archived on 8/28/01:</FONT>	
		<P> Meanwhile we've been continuing work on the genome browser.  
  	It's now possible  to upload your own annotations to be displayed 
  	alongside the  built-in tracks.  Please scroll to the bottom of the 
 	browser gateway pages for further information.  The browser has 
  	also been sped  up, particularly on the larger chromosomes by 
  	using a 'binning'  technique suggested by Lincoln Stein and Richard Durbin.</P>

<hr>
<FONT SIZE=4 COLOR="#006666">Archived on 8/23/01:</FONT>	
		<P>Tracks continue to be added to the 
	<A HREF="apr2001Tracks.html">April 2001 browser</A>.
	Our old friend the Exofish track is back.  The blat mouse homology
	track is now up as well, computed at somewhat more sensitive settings
	than it was in the 
	<A HREF="decTracks.html">December 2000 browser.</A></P>
		<P>We've recently received some significant funding from NHGRI to
	maintain and extend this site.  This has allowed us among other things
	to hire an artist, Jenny Draper, who is responsible for the new look.</P>

   </TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	
<BR>
	
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1><TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4" ><a NAME="#BROWSER"></a><b>&nbsp; Freeze-Specific News Archives</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=7></TH><TR><TD WIDTH=10></TD><TD>
		
<FONT SIZE=4 COLOR="#006666">April 1, 2001 Freeze</FONT>
<P>July 13, 2001: fixed bug where some UTRs were
mis-annotated in the known genes on the minus strand.</P>

<hr><FONT SIZE=4 COLOR="#006666">Dec. 12, 2000 Freeze</FONT>
<P>July 13, 2001: fixed bug where some UTRs were
mis-annotated in the known genes on the minus strand.</P>
<P>April 5, 2001: chromosome level files (but not contig level files)
updated to fix bug where some of the centromeres were misplaced</P>
<P>April 1, 2001: chromosome Y updated to fix a bug that put a large
gap between each clone.  This bug was limited to the Y chromosome.</P>

<hr><FONT SIZE=4 COLOR="#006666">Oct. 7, 2000 Freeze</FONT>
<P>April 7, 2001: Affymetrix gene predictions updated and available
for bulk download.</P>
<P>January 9, 2001: all files were updated after a bug that had caused
some finished clones to be flipped in the assembly was caught and fixed.
Our apologies for any inconvenience this has caused.

<P>November 10, 2000: the sequence (.fa) files for two contigs:
X/ctg18523/ctg18523.fa and 7/ctg15082/ctg15082.fa were updated.
These files had null (zero valued) characters that have been
replaced with N characters.  These characters were a result of
a mismatch between clone sizes in the map and in finished NT
contigs.  The sequence for chromosomes X and 7, which contain
these contigs, has also been updated.</P>

<hr><FONT SIZE=4 COLOR="#006666">Sept. 5, 2000 Freeze</FONT>
<P>October 11, 2000: chr21.agp and chr22.fa were updated.  chr21.agp
was a version which went with the UCSC draft assembly rather than the
Sanger/NCBI final assembly of this chromosome.  chr21.agp and chr21.fa
are now in sync.   chr22.fa and chr22.agp were also previously out of
sync.  chr22.fa was obtained from Sanger while chr22.agp had been obtained
by NCBI.  With this update they are consistent, both NCBI versions.
Apologies for any rework this causes you. </P>

<P>October 9, 2000: chr21_random.* and chr22_random.* were removed from
the zip-files in the September 5th freeze.  These files were relics that
should not have been included in the first place.  The files
chr9_random.agp, chr10_random.agp, chr11_random.agp, chr12_random.agp,
chr13_random.agp and chr14_random.agp were updated.  There was a bug
where the initial field of the initial lines in these files was
"(null)" rather than "chrN_random" as it should have been. </P>

<hr><FONT SIZE=4 COLOR="#006666">July 17, 2000 Freeze</FONT>
<P>September 22, 2000: The zip-files chromFa.zip and chromAgp.zip under the
<A HREF="http://genome-archive.cse.ucsc.edu/goldenPath/17july2000/bigZips/">July 17th full data set </A>
were updated to fix some duplications of clone contigs that occurred in the chrN_random.agp and chrN_random.fa
files contained within these zip-files. These "_random" files are files that contain clone contigs that were mapped to a particular chromosome, but
could not be placed at a specific position within that chromosome. They correspond to the "RANDOM" sections of the WashU map. 
None of the regular chrN.agp or chrN.fa files were affected by this update, nor was any of the information in the 
contigAgp.zip or contigFa.zip files changed. For convenience, we include two new files, chromRandAgp.zip and
chromRandFa.zip, for users who would like to download only the data that has changed. These zips consist of
the updated chrN_random.agp and chrN_random.fa files, respectively.

<P>September 4, 2000: The files chromFa.zip and contigFa.zip under the
<A HREF="http://genome-archive.cse.ucsc.edu/goldenPath/17july2000/bigZips/">July 17th full data set </A> and the files under 
<A HREF="http://genome-archive.cse.ucsc.edu/goldenPath/17july2000/pieces/">
July 17th data by individual clone contig </A>
were updated to fix some incorrect (null(0))
characters that needed to be replaced by 'n' characters in some
Fasta files. The following contig Fasta files on chromosomes 1,8,11,12,16,
17 and 19 were affected:

<PRE>
1/ctg14250/ctg14250.fa
8/ctg17325/ctg17325.fa
8/ctg16307/ctg16307.fa
8/ctg25150/ctg25150.fa
8/ctg15216/ctg15216.fa
11/ctg15548/ctg15548.fa
12/ctg15285/ctg15285.fa
16/ctg16864/ctg16864.fa
16/ctg28077/ctg28077.fa
16/ctg13970/ctg13970.fa
16/ctg25401/ctg25401.fa
16/ctg13336/ctg13336.fa
16/ctg14937/ctg14937.fa
16/ctg16860/ctg16860.fa
16/ctg17794/ctg17794.fa
17/ctg14047/ctg14047.fa
19/ctg16564/ctg16564.fa
</PRE>

We apologize for any inconvenience this may have caused. 

	<BR><BR></TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
		
</TD></TR></TABLE>
</BODY></HTML>







