<!DOCTYPE html>
<!--#set var="TITLE" value="UCSC Genome Browser: News Archives" -->
<!--#set var="ROOT" value=".." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->
<!--#include virtual="$ROOT/redmineWidget.html" -->

<h1>News Archives</h1>


<div class="container">
  <div class="row">
    <div class="col-sm-3">
      <ul>
        <li><a href="#2026">2026 News</a></li>
        <li><a href="#2025">2025 News</a></li>
        <li><a href="#2024">2024 News</a></li>
        <li><a href="#2023">2023 News</a></li>
        <li><a href="#2022">2022 News</a></li>
      </ul>
    </div>
    <div class="col-sm-3">
      <ul>
        <li><a href="#2021">2021 News</a></li>
        <li><a href="#2020">2020 News</a></li>
        <li><a href="#2019">2019 News</a></li>
        <li><a href="#2018">2018 News</a></li>
        <li><a href="#2017">2017 News</a></li>
      </ul>
    </div>
    <div class="col-sm-3">
      <ul>
        <li><a href="#2016">2016 News</a></li>
        <li><a href="#2015">2015 News</a></li>
        <li><a href="#2014">2014 News</a></li>
        <li><a href="#2013">2013 News</a></li>
        <li><a href="#2012">2012 News</a></li>
      </ul>
    </div>
    <div class="col-sm-3">
      <ul>
        <li><a href="#2011">2011 News</a></li>
        <li><a href="#2010">2010 News</a></li>
        <li><a href="#2009">2009 News</a></li>
        <li><a href="#2008">2008 News</a></li>
        <li><a href="#2001">2001</a>-<a href="#2007">2007 News</a></li>
      </ul>
    </div>
  </div> 
</div>

<p>You can sign-up to get these announcements via our 
<a target=_blank href="https://groups.google.com/a/soe.ucsc.edu/g/genome-announce?hl=en">Genome-announce</a>
email list. We send around one short announcement email every two weeks.</p>

<p>Smaller software changes are not announced here.  A summary of the three-weekly release changes can be 
found <a target=_blank href="https://genecats.gi.ucsc.edu/builds/versions.html">here</a>. 
For the full list of our daily code changes head to our <a
href="https://github.com/ucscGenomeBrowser/kent/commits/master"
target=_blank>GitHub page</a>. Lastly, see our <a href="credits.html" target="_blank">
credits page</a> for acknowledgments of the data we host.</p>

<!-- ============= 2025 archived news ============= -->
<a name="2026"></a>

<!--
<a name="020326"></a>
<h2>Feb. 03, 2026 &nbsp;&nbsp; New &quot;Visible Tracks&quot; group feature</h2>
<p>
We are excited to announce a new improvement to the UCSC Genome Browser: the
<b>Visible Tracks</b> group. The Genome Browser now includes a dynamic track group that
automatically displays all currently visible tracks in one convenient location. This means you no
longer need to hunt through multiple track categories to adjust tracks you've already turned on.
</p>
<div class="text-center">
<img src="/images/newsArchImages/visibleTracks.png" width='65%'>
</div>
<p>
Previously, managing visible tracks required navigating back to their original categories &mdash;
whether that was Genes and Gene Predictions, Conservation, Variation, or any other group. Now, all
your active tracks appear together in the <b>Visible Tracks</b> group, making it faster and easier
to adjust display modes for visible tracks.
</p>
<p>
We would like to thank Chris Lee and Jairo Navarro Gonzalez for their work on creating and testing
the feature.</p> 
-->

<!--
<a name="020326"></a>
<h2>Feb. 03, 2026 &nbsp;&nbsp; Hub Space: Host Your Track Hubs Directly on the Genome Browser</h2>
<p>
We are excited to introduce <a href="/cgi-bin/hgHubConnect#hubUpload">Hub Space</a>, a new
hosting service that allows users to upload and visualize track hub files directly on the UCSC
Genome Browser without relying on third-party hosting services such as Dropbox, Google Drive, or
AWS.
</p>

<p>
<b>Hub Space Features:</b><br>
Each account is allocated 10 GB of storage space. While we strive to maintain uninterrupted
access, please maintain your own backups. Data persistence over the long term is not
guaranteed. To request additional storage space, please
<a href="/contacts.html">contact us</a>.
</p>

<p>
<b>How to Use Hub Space:</b><br>
There are two primary ways to upload your files:
</p>

<ol>
  <li>
    <b>Single file upload:</b> Upload individual binary-indexed files (bigBed, bigWig, BAM, VCF,
    and other supported formats) for immediate visualization.
  </li>
  <li>
    <b>Complete hub upload:</b> Upload an entire track hub along with all supporting files
    (hub.txt, bigWig, bigBed, etc.) to visualize the hub.
  </li>
</ol>

<p>
If a hub.txt file is not provided when uploading data files, one is automatically generated.
The generated hub.txt file is configured with the
<a href="/goldenPath/help/trackDb/trackDbHub.html#useOneFile">useOneFile</a> trackDb setting, and
as a result, each hub is limited to a single genome assembly. Editing the generated hub.txt file is
not yet supported. To customize the hub configuration or data display, replace the generated hub.txt
file with your own version after uploading the data files.
</p>

<p>
To get started with Hub Space, visit the
<a href="/cgi-bin/hgHubConnect#hubUpload">Track Data Hubs page</a> and
click on the &quot;Hub Space&quot; section. We would love to hear your feedback on this new feature.
Please <a href="/contacts.html">contact us</a> and let us know what you think!
</p>
-->

<a name="011526"></a>
<h2>Jan. 15, 2026 &nbsp;&nbsp; New ENCODE cCREs track available for human (hg38)</h2>
<p>
We are proud to announce a new <a href="/cgi-bin/hgTrackUi?db=hg38&c=chr7&g=cCREs&position=default"
        target="_blank">
ENCODE Registry of cCREs (candidate Cis-Regulatory Elements)</a> 
container track for the hg38/GRCh38 genome as described in <a target="_blank"
href="https://www.nature.com/articles/s41586-025-09909-9">Moore et al., <em>Nature</em> 2026</a>. 
This container contains candidate Cis-Regulatory 
Elements (cCREs) generated by the ENCODE Consortium during Phase 4 (ENCODE4) and Phase 3 (ENCODE3).
The previous "ENCODE3 cCREs" track is retained for archival purposes. The tracks include both integrated 
(biosample-agnostic) and biosample-specific annotations derived from core epigenomic assays.</p>

<div class="text-left">
<a href="https://genome.ucsc.edu/s/Lou/HBBexample" target=_blank>
<img src="/images/encode4cCREsNews.png" width='75%'></a>
</div>
<p>
The new tracks include:</p>
<ul>
<li><b><a href="/cgi-bin/hgTrackUi?c=chr7&g=cCREregistry&position=default&db=hg38" target="_blank">
ENCODE4 cCREs</a>:</b> This track presents the ENCODE Registry of 2,348,854 cCREs identified 
and classified using data from all phases of the ENCODE Project (Phases 1–4). The registry 
integrates chromatin accessibility and ChIP-seq signals across thousands of biosamples. All cCREs 
are consolidated into a single, cell type-agnostic annotation track displayed here.</li>
<li><b><a href="/cgi-bin/hgTrackUi?c=chr7&g=coreCcres&position=default&db=hg38" target="_blank">
ENCODE4 Core Collection</a>:</b> This track displays biosample-specific cCREs alongside 
genome-wide epigenomic signals for the ENCODE4 Core Collection, consisting of 170 biosamples 
comprehensively profiled using four core assays: DNase-seq, ChIP-seq for H3K4me3 and H3K27ac 
(histone modifications), and ChIP-seq for CTCF. These data support detailed analysis of 
regulatory activity in individual biosamples.</li>
</ul>
<p>
The cCREs are colored by their putative functional assignment based on biochemical 
signatures and genomic context:</p>
<div class="text-left">
<p>
<img src="/images/encode4cCREs.png" width='35%'>
</div>
<p>
This results in the following classifications:</p>

<table cellpadding='2'>
  <tr>
    <th style="border-bottom: 2px solid;">Color</th>
    <th style="border-bottom: 2px solid;"></th>
    <th style="border-bottom: 2px solid;">UCSC label</th>
    <th style="border-bottom: 2px solid;">ENCODE classification</th>
    <th style="border-bottom: 2px solid;">Classification criteria</th>
  </tr>
<tr><td style='background-color: #FF0000;'>&nbsp;&nbsp;&nbsp;</td><td>red</td>
        <td>promoter</td>
        <td>promoter-like signature</td>
        <td>within 200 bp of TSS, high chromatin accessibility and H3K4me3</td></tr>
<tr><td style='background-color: #FFA700;'>&nbsp;&nbsp;&nbsp;</td><td>orange</td>
        <td>proximal enhancer</td>
        <td>TSS-proximal enhancer-like signature</td>
        <td>high chromatin accessibility and H3K27ac, within 2 kb of TSS; if within 200 bp of TSS, must have low H3K4me3</td></tr>
<tr><td style='background-color: #FFCD00;'>&nbsp;&nbsp;&nbsp;</td><td>yellow</td>
        <td>distal enhancer</td>
        <td>TSS-distal enhancer-like signature</td>
        <td>high chromatin accessibility and H3K27ac, &gt;2 kb from TSS</td></tr>
<tr><td style='background-color: #FFAAAA;'>&nbsp;&nbsp;&nbsp;</td><td>pink</td>
        <td>CA-H3K4me3</td>
        <td>chromatin accessibility + H3K4me3</td>
        <td>high chromatin accessibility and H3K4me3, low H3K27ac, not within 200 bp of TSS</td></tr>
<tr><td style='background-color: #00B0F0;'>&nbsp;&nbsp;&nbsp;</td><td>blue</td>
        <td>CA-CTCF</td>
        <td>chromatin accessibility + CTCF</td>
        <td>high chromatin accessibility and CTCF, low H3K4me3 and H3K27ac</td></tr>
<tr><td style='background-color: #800080;'>&nbsp;&nbsp;&nbsp;</td><td>dark purple</td>
        <td>CA-TF</td>
        <td>chromatin accessibility + transcription factor</td>
        <td>high chromatin accessibility, low H3K4me3, H3K27ac, and CTCF, bound by transcription factor</td></tr>
<tr><td style='background-color: #06B050;'>&nbsp;&nbsp;&nbsp;</td><td>green</td>
        <td>CA</td>
        <td>chromatin accessibility</td>
        <td>high chromatin accessibility, low H3K4me3, H3K27ac, and CTCF</td></tr>
<tr><td style='background-color: #CC99FF;'>&nbsp;&nbsp;&nbsp;</td><td>light purple</td>
        <td>TF</td>
        <td>transcription factor</td>
        <td>low chromatin accessibility, low H3K4me3, H3K27ac, and CTCF, bound by transcription factor</td></tr>
</table>

<p>
This dataset was produced by the ENCODE Data Analysis Center
(<a href="https://www.umassmed.edu/zlab/" target="_blank">Zlab</a> at UMass Medical Center). Thanks
to Mingshi Gao, Zhiping Weng, and Jill Moore for providing this data and
initial version of the track. Thanks also to the ENCODE Consortium, the ENCODE production
laboratories, and the ENCODE Data Coordination Center for generating and processing the datasets
used here.</p>
<p>
For more details on this track and data, see the <a href="/cgi-bin/hgTrackUi?db=hg38&c=chr7&g=cCREs&position=default" 
target="_blank">track description page</a>, as well as 
Moore <em>et al</em>. <a href="https://www.nature.com/articles/s41586-025-09909-9" target="_blank">
An Expanded Registry of Candidate cis-Regulatory Elements for Studying Transcriptional Regulation</a>
. <em>Nature</em>. 2026 January 7. PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/39763870"
target="_blank">39763870</a>; PMC: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11703161/"
target="_blank">PMC11703161</a>.
</p>

<!-- ============= 2025 archived news ============= -->
<a name="2025"></a>

<a name="121925"></a>
<h2>Dec. 19, 2025 &nbsp;&nbsp; EVA SNP release 8 for 41 assemblies</h2>
<p>
We are pleased to announce the release of the EVA SNP Release 8 tracks for 41 assemblies.
These tracks contain mappings of single nucleotide variants and small insertions and deletions
(indels) &mdash; collectively Simple Nucleotide Variants (SNVs) &mdash; from the European Variation
Archive (<a href="https://www.ebi.ac.uk/eva/" target="_blank">EVA</a>) Release 8. The full list of
assemblies that contain the EVA SNP release 8 track is below:</p>

<p>
<div class="container">
  <div class="row">
    <div class="col-sm-4">
    <ul>
      <li>A. gambiae <a href="../cgi-bin/hgTrackUi?db=anoGam3&g=evaSnp8"
        target="_blank">(anoGam3)</a></li>
      <li>C. intestinalis <a href="../cgi-bin/hgTrackUi?db=ci3&g=evaSnp8"
        target="_blank">(ci3)</a></li>
      <li>Bison <a href="../cgi-bin/hgTrackUi?db=bisBis1&g=evaSnp8"
        target="_blank">(bisBis1)</a></li>
      <li>Cat <a href="../cgi-bin/hgTrackUi?db=felCat8&g=evaSnp8"
        target="_blank">(felCat8)</a>,
        <a href="../cgi-bin/hgTrackUi?db=felCat9&g=evaSnp8"
        target="_blank">(felCat9)</a></li>
      <li>Chicken <a href="../cgi-bin/hgTrackUi?db=galGal4&g=evaSnp8"
        target="_blank">(galGal4)</a>,
        <a href="../cgi-bin/hgTrackUi?db=galGal5&g=evaSnp8"
        target="_blank">(galGal5)</a>,
        <a href="../cgi-bin/hgTrackUi?db=galGal6&g=evaSnp8"
        target="_blank">(galGal6)</a></li>
      <li>Chimp <a href="../cgi-bin/hgTrackUi?db=panTro4&g=evaSnp8"
        target="_blank">(panTro4)</a>,
        <a href="../cgi-bin/hgTrackUi?db=panTro5&g=evaSnp8"
        target="_blank">(panTro5)</a></li></li>
      <li>Cow <a href="../cgi-bin/hgTrackUi?db=bosTau6&g=evaSnp8"
        target="_blank">(bosTau6)</a>,
        <a href="../cgi-bin/hgTrackUi?db=bosTau7&g=evaSnp8"
        target="_blank">(bosTau7)</a>,
        <a href="../cgi-bin/hgTrackUi?db=bosTau9&g=evaSnp8"
        target="_blank">(bosTau9)</a></li>
      <li>Crab-eating macaque <a href="../cgi-bin/hgTrackUi?db=macFas5&g=evaSnp8"
        target="_blank">(macFas5)</a></li>
      <li>D. melanogaster <a href="../cgi-bin/hgTrackUi?db=dm6&g=evaSnp8"
        target="_blank">(dm6)</a></li>
    </ul></div>
   <div class="col-sm-4">
      <ul>
      <li>Dog <a href="../cgi-bin/hgTrackUi?db=canFam3&g=evaSnp8"
        target="_blank">(canFam3)</a></li>
      <li>C. elegans <a href="../cgi-bin/hgTrackUi?db=ce11&g=evaSnp8"
        target="_blank">(ce11)</a></li>
      <li>Gorilla <a href="../cgi-bin/hgTrackUi?db=gorGor3&g=evaSnp8"
        target="_blank">(gorGor3)</a>,
        <a href="../cgi-bin/hgTrackUi?db=gorGor4&g=evaSnp8"
        target="_blank">(gorGor4)</a></li>
      <li>Green monkey <a href="../cgi-bin/hgTrackUi?db=chlSab2&g=evaSnp8"
        target="_blank">(chlSab2)</a></li>
      <li>Horse <a href="../cgi-bin/hgTrackUi?db=equCab3&g=evaSnp8"
        target="_blank">(equCab3)</a></li>
      <li>Marmoset <a href="../cgi-bin/hgTrackUi?db=calJac3&g=evaSnp8"
        target="_blank">(calJac3)</a></li>
      <li>Mouse <a href="../cgi-bin/hgTrackUi?db=mm10&g=evaSnp8"
        target="_blank">(mm10)</a>,
        <a href="../cgi-bin/hgTrackUi?db=mm39&g=evaSnp8"
        target="_blank">(mm39)</a></li>
      <li>Nile tilapia <a href="../cgi-bin/hgTrackUi?db=oreNil2&g=evaSnp8"
        target="_blank">(oreNil2)</a></li>
      <li>Orangutan <a href="../cgi-bin/hgTrackUi?db=ponAbe3&g=evaSnp8"
        target="_blank">(ponAbe3)</a></li>
    </ul></div>
    <div class="col-sm-4">
      <ul>
      <li>Pig <a href="../cgi-bin/hgTrackUi?db=susScr3&g=evaSnp8"
        target="_blank">(susScr3)</a>,
        <a href="../cgi-bin/hgTrackUi?db=susScr11&g=evaSnp8"
        target="_blank">(susScr11)</a></li>
      <li>Rat <a href="../cgi-bin/hgTrackUi?db=rn5&g=evaSnp8"
        target="_blank">(rn5)</a>,
        <a href="../cgi-bin/hgTrackUi?db=rn6&g=evaSnp8"
        target="_blank">(rn6)</a>,
        <a href="../cgi-bin/hgTrackUi?db=rn7&g=evaSnp8"
        target="_blank">(rn7)</a></li>
      <li>Rhesus <a href="../cgi-bin/hgTrackUi?db=rheMac8&g=evaSnp8"
        target="_blank">(rheMac8)</a>,
        <a href="../cgi-bin/hgTrackUi?db=rheMac10&g=evaSnp8"
        target="_blank">(rheMac10)</a></li>
      <li>S. cerevisiae <a href="../cgi-bin/hgTrackUi?db=sacCer3&g=evaSnp8"
        target="_blank">(sacCer3)</a></li>
      <li>Sheep <a href="../cgi-bin/hgTrackUi?db=oviAri3&g=evaSnp8"
        target="_blank">(oviAri3)</a>,
        <a href="../cgi-bin/hgTrackUi?db=oviAri4&g=evaSnp8"
        target="_blank">(oviAri4)</a></li>
      <li>Turkey <a href="../cgi-bin/hgTrackUi?db=melGal5&g=evaSnp8"
        target="_blank">(melGal5)</a></li>
      <li>Zebra finch <a href="../cgi-bin/hgTrackUi?db=taeGut2&g=evaSnp8"
        target="_blank">(taeGut2)</a></li>
      <li>Zebrafish <a href="../cgi-bin/hgTrackUi?db=danRer7&g=evaSnp8"
        target="_blank">(danRer7)</a>,
        <a href="../cgi-bin/hgTrackUi?db=danRer10&g=evaSnp8"
        target="_blank">(danRer10)</a>,
        <a href="../cgi-bin/hgTrackUi?db=danRer11&g=evaSnp8"
        target="_blank">(danRer11)</a></li>
    </ul></div>
  </div>
</div></p>

<p>
These variants are classified by EVA into one of the following <a target="_blank"
href="http://www.sequenceontology.org/">sequence ontology</a> terms:
</p>
<ul>
  <li> <b>substitution</b> &mdash;
       A single nucleotide in the reference is replaced by another, alternate allele.
  <li> <b>deletion</b> &mdash;
       One or more nucleotides are deleted.  The representation in the database is to
       display one additional nucleotide in both the Reference field (Ref) and the
       Alternate Allele field (Alt).  E.g., a variant that is a deletion of an A
       may be represented as Ref = GA and Alt = G.
  <li> <b>insertion</b> &mdash;
       One or more nucleotides are inserted.  The representation in the database is to
       display one additional nucleotide in both the Reference field (Ref) and the
       Alternate Allele field (Alt).  E.g., a variant that is an insertion of a T may
       be represented as Ref = G and Alt = GT.
  <li> <b>delins</b> &mdash;
       Similar to a tandem repeat, in that the runs of Ref and Alt Alleles are of
       different length, except that there is more than one type of nucleotide,
       e.g., Ref = CCAAAAACAAAAACA, Alt = ACAAAAAC.
  <li> <b>multipleNucleotideVariant</b> &mdash;
       More than one nucleotide is substituted by an equal number of different
       nucleotides, e.g.,  Ref = AA, Alt = GC.
  <li> <b>sequence alteration</b> &mdash;
       A parent term meant to signify a deviation from another sequence. Can be
       assigned to variants that have not been characterized yet.
</ul>
<p>
The variants have also been annotated with our <a href="/cgi-bin/hgVai" target="_blank">Variant
Annotation Integrator</a> tool with functional classes such as synonymous variant,
missense variant, stop gained, etc. For additional details on the track colors,
as well as the filters and metadata on each variant, see the
<a href="/cgi-bin/hgTrackUi?db=anoGam3&c=chr2R&g=evaSnpContainer&position=default#TRACK_HTML"
        target="_blank">track description page</a>.</p>

<p>
We would like to thank the <a href="https://www.ebi.ac.uk/eva/" target="_blank">European Variation
Archive</a> for making these data publicly available. We would also like to thank Gerardo Perez, Luis Nassar,
and Angie Hinrichs for the creation and release of these tracks.
</p>

<a name="121525"></a>
<h2>Dec. 15, 2025 &nbsp;&nbsp; Ancient Hominids track for hg38</h2>
<p>
We are happy to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=ancient"
        target="_blank">Ancient Hominids track</a>
featuring data from Archaic Sequence Hub (ArcSeqHub). This track shows variants
identified by ArcSeqHub's remapping of high-quality Altai Neanderthal and
Denisovan genomes onto the hg38/GRCh38 genome. Variants are divided into two subtracks,
one for Denisovian variants and another for Neanderthal variants.
UCSC has removed those positions from the VCF without an alternate allele to show only
variants that are present in the ancient genomes.</p>
<div class="text-center">
<img src="/images/newsArchImages/ancientHominids.png" width='80%'></a>
</div>
<p>
We would like to thank the <a href="http://www.arcseqhub.com/plot/-1/"
target="_blank">ArcSeqHub</a> authors for making the data available.
We would also like to thank Maximilian Haeussler and Matthew Speir for the
creation and release of this track.</p>

<a name="120325"></a>
<h2>Dec. 03, 2025 &nbsp;&nbsp; New gnomAD Missense Deleteriousness Prediction by Constraint (MPC) track for hg19</h2>
<p>
We are happy to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=gnomadMpc"
        target="_blank">gnomAD Missense Deleteriousness Prediction by Constraint (MPC) track</a>
for hg19, now available in the
<a href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=gnomadSuper"
        target="_blank">gnomAD superTrack</a>. This track shows a score that tries to identify
missense-depleted regions using the patterns of rare missense variation in 125,748 gnomAD v2.1.1
exomes, compared to a null mutational model. Missense-depleted regions are enriched for ClinVar pathogenic
variants, de novo missense variants in individuals with neurodevelopmental disorders (NDDs), and
complex trait heritability. This track can be used to aid in interpreting missense variants.</p>
<p>
Transcript regions with constraint predictions are colored using the viridis palette, where yellow 
indicates the lowest OE values and dark blue-purple indicates the highest.</p>
<div class="text-center">
<img src="../../images/newsArchImages/gnomAD_mpc.png" width='80%'></a>
</div>
<p>
A corresponding MPC version for hg38 is planned for release by the gnomAD team in 2026.</p>
<p>
We would like to thank Kaitlin Samocha for suggesting this track and the
<a href="https://gnomad.broadinstitute.org/about"
	target="_blank">Genome Aggregation Database Consortium (gnomAD)</a> for making the data
available. We would also like to thank Maximilian Haeussler, Luis Nassar, and Gerardo Perez for the
creation and release of this track.</p>



<a name="111425"></a>
<h2>Nov. 14, 2025 &nbsp;&nbsp; Updated Varaico Variants track and new Varaico Variants (suppl) track for hg38 and hg19</h2>
<p>
We are pleased to announce the release 2 of the <b>Varaico Variants track</b> and the addition of a new
<b>Varaico Variants (suppl) track</b> for the human assemblies
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=varsInPubs"
        target="_blank">hg38/GRCh38</a> and
<a href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=varsInPubs"
        target="_blank">hg19/GRCh37</a>. The Varaico tracks are generated using literature mining
methods similar to those used for the <a href="http://bejerano.stanford.edu/AVADA/"
        target="_blank">AVADA</a>
resource. With this update, the Varaico Variants track
now contains more than 5.5 million variants.</p>
<p>
The Varaico Variants (suppl) track shows variants extracted from supplementary data files using
similar methods to those used for the primary Varaico track.</p>
<div class="text-center">
<img src="../../images/newsArchImages/varaicoRelease2.png" width='80%'></a>
</div>
<p>
For more information, please visit the <a href="https://varaico.com/"
	target="_blank">Varaico website</a>.</p>
<p>
We would like to thank Johannes Birgmeier for generating and making the Varaico data available.
We would also like to thank Maximilian Haeussler, Jairo Navarro Gonzalez, and Gerardo Perez for
creating and releasing these tracks.</p>

<a name="103125"></a>
<h2>Oct. 31, 2025 &nbsp;&nbsp; New GENCODE &quot;knownGene&quot; V49 for human (hg38/hg19) and VM38
for mouse (mm39)</h2>

<p>
We are happy to announce the new GENCODE gene annotation tracks, corresponding to
<a href="https://www.ensembl.info/2025/09/02/ensembl-115-has-been-released/"
target="_blank">Ensembl release 115</a>, along with GENCODE knownGene V49 for human
(<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=knownGene">hg38/GRCh38</a>
and
<a href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=knownGene">hg19/GRCh37</a>)
and GENCODE knownGene VM38 for mouse
(<a href="/cgi-bin/hgTrackUi?db=mm39&position=default&g=knownGene">mm39/GRCm39</a>).
The GENCODE &quot;knownGene&quot; V49 and VM38 tracks were built using the UCSC knownGene pipeline and the
GENCODE comprehensive gene set to generate high-quality manual annotations merged with
evidence-based automated annotations. The GENCODE &quot;knownGene&quot; tracks are our default
gene tracks, which have extensive associations to external sources. This allows for additional
metadata on every item as well as external links. The track description pages contain options for
configuring the display, such as showing non-coding genes, splice variants, and pseudogenes.</p>
<p>
Below is a summary of the contents found in each release. For more details, visit the
<a target="_blank" href="https://www.gencodegenes.org/">GENCODE site</a>.</p>
<p>
<table class="stdTbl">
<tr><th COLSPAN=4>GENCODE v49 Release Stats</th></tr>
<tr align=left><th>Genes</th><th>Observed</th><th>Transcripts</th><th>Observed</th></tr>
<tr align=left><td>Protein-coding genes</td><td>19,433</td>
               <td>Protein-coding transcripts</td><td>211,446</td></tr>
<tr align=left><td>Long non-coding RNA genes</td><td>35,899</td>
               <td><font size="-1">- full length protein-coding</font></td><td>186,646</td></tr>
<tr align=left><td>Small non-coding RNA genes</td><td>7,563</td>
               <td><font size="-1">- partial length protein-coding</font></td><td>24,800</td></tr>
<tr align=left><td>Pseudogenes</td><td>14,701</td>
               <td>Nonsense mediated decay transcripts</td><td>21,949</td></tr>
<tr align=left><td>Immunoglobulin/T-cell receptor gene segments</td>
               <td>649</td><td>Long non-coding RNA loci transcripts</td><td>191,079</td></tr>
<tr align=left><td>Total No of distinct translations</td><td>129,801</td>
               <td>Genes that have more than one distinct translations</td><td>15,498</td></tr>
</table><BR>
</p>
<p>
<table class="stdTbl">
<tr><th COLSPAN=4>GENCODE VM38 Release Stats</th></tr>
<tr align=left><th>Genes</th><th>Observed</th><th>Transcripts</th><th>Observed</th></tr>
<tr align=left><td>Protein-coding genes</td><td>21,530</td>
               <td>Protein-coding transcripts</td><td>58,647</td></tr>
<tr align=left><td>Long non-coding RNA genes</td><td>36,108</td>
               <td><font size="-1">- full length protein-coding</font></td><td>45,050</td></tr>
<tr align=left><td>Small non-coding RNA genes</td><td>6,105</td>
               <td><font size="-1">- partial length protein-coding</font></td><td>13,597</td></tr>
<tr align=left><td>Pseudogenes</td><td>13,809</td>
               <td>Nonsense mediated decay transcripts</td><td>7,250</td></tr>
<tr align=left><td>Immunoglobulin/T-cell receptor gene segments</td><td>701</td>
               <td>Long non-coding RNA loci transcripts</td><td>155,914</td></tr>
<tr align=left><td>Total No of distinct translations</td><td>44,974</td>
               <td>Genes that have more than one distinct translations</td><td>10,853</td></tr>
</table><BR>
</p>
<p>
We would like to thank the <a target="_blank"
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Jonathan Casper, Mark Diekhans, and Gerardo Perez for the
development and release of these tracks.</p>

<a name="101625"></a>
<h2>Oct. 16, 2025 &nbsp;&nbsp; SpliceAI Wildtype tracks for hg38</h2>
<p>
We are pleased to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=spliceAIWt"
        target="_blank">SpliceAI Wildtype tracks</a> for hg38, available in the
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=spliceImpactSuper"
        target="_blank">Splicing Impact superTrack</a>.
These tracks show the scores for the genome sequence itself, without variants, from predicted
splice donor (5&apos; intron boundaries) and splice acceptor (3&apos; intron boundaries) sites. Predictions
are strand-specific, with separate subtracks for the plus and minus strands.
<ul>
  <li><b>SpliceAI Acceptor Plus</b> &ndash; Splice acceptor sites, plus strand
  <li><b>SpliceAI Acceptor Minus</b> &ndash; Splice acceptor sites, minus strand 
  <li><b>SpliceAI Donor Plus</b> &ndash; Splice donor sites, plus strand 
  <li><b>SpliceAI Donor Minus</b> &ndash; Splice donor sites, minus strand 
</ul>
<p class="text-center">
<img src="../../images/newsArchImages/spliceAIWildtype.png"
alt="SpliceAI Wildtype track display for the CFTR region" width='60%'></p>
<p>
These tracks are useful in combination with the variants track for evaluating new transcript
models. They can be used to assess potential exon boundaries or possible splice acceptor sites.</p>
<p>
We would like to thank Illumina for making SpliceAI available, both the model and the precomputed
data files. Thanks to Francois Lecoquierre from the University of Oxford, Jean-Madeleine de Sainte
Agathe from Institut Pasteur Paris, and Michael Hiller from the Senckenberg Museum Frankfurt for
suggesting and then creating the SpliceAI Wildtype annotations. We would also like to thank Max
Haeussler and Gerardo Perez for their efforts on this release.</p>

<a name="092525"></a>
<h2>Sep. 25, 2025 &nbsp;&nbsp; Panmask Easy 151b Regions track for hg38</h2>
<p>
We are happy to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=panmask151b"
	target="_blank">Panmask Easy 151b Regions</a>
track for hg38. This new track is available in the
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=problematicSuper"
        target="_blank">Problematic Regions superTrack</a>.
The track contains a set of sample-agnostic easy regions where short-read variant calling reaches
high accuracy. Easy regions are derived for variant filtration agnostic to individual samples. They
are genomic intervals where general variant callers achieve high accuracy without sophisticated
filtering.</p>
<p class="text-center">
<img src="../../images/newsArchImages/panmask.png"
alt="Panmask Easy 151b Regions track for the BRCA1 exon 19" width='75%'></p>
<p>
The pm151 regions are used to filter spurious variant calls in centromeres, long repeats, and
other genomic regions where short-read mapping is often problematic. They cover 88.2% of hg38,
92.2% of coding regions, and 96.3% of ClinVar pathogenic variants. The track can be used to filter
variant calls for clinical or research human samples. It shows regions that are easy to sequence,
rather than those that are problematic. The data was derived from the HPRC assemblies, and this
track presents the 151b-easy panmask set.</p>
<p>
We would like to thank the
<a href="https://hlilab.github.io/" target="_blank">HLi lab</a>
at Harvard Medical School for making this data available. We
would also like to thank Max Haeussler and Gerardo Perez for their efforts on this release.</p>

<a name="092425"></a>
<h2>Sep. 24, 2025 &nbsp;&nbsp; CoLoRSdb small and structure variants for hg38 and hs1</h2>
<p>
We are excited to announce the release of the CoLoRSdb Small and Structural Variant tracks for the
human assemblies
<a href="/cgi-bin/hgTrackUi?db=hg38&g=longReadVariants&position=default" target="_blank">GRCh38/hg38</a> and
<a href="/cgi-bin/hgTrackUi?db=hs1&g=hub_3671779_longReadVariants&position=default"
target="_blank">CHM13/hs1</a>. These tracks provide a comprehensive catalog of genetic variation
discovered through long-read whole genome sequencing, contributed by the international
<a href="https://colorsdb.org" target="_blank">Consortium of Long Read Sequencing (CoLoRS)</a>. The
small variant tracks (DeepVariant + GLnexus) contain single nucleotide polymorphisms (SNPs) and
short indels, while the structural variant tracks (pbsv + Jasmine) display larger events including
insertions, deletions, and inversions. Long-read sequencing technology improves sensitivity in
repetitive regions and provides more precise breakpoint resolution than short-read approaches,
enabling accurate visualization of complex loci in the Genome Browser.</p>
<p>
Each track includes allele frequency and sample count annotations, with additional filtering options
for variant size and type. Users can click on individual variants to view detailed metadata, such as
allele counts, homozygous/heterozygous call distributions, and Hardy-Weinberg equilibrium values.
</p>

<div class="text-center">
  <a href="https://genome.ucsc.edu/s/jnavarr5/CoLoRSdb_announcement">
  <img src="../images/newsArchImages/CoLoRSdb.png"
       alt="CoLoRSdb track with the mouseover tooltip on the DEL-159bp item." width='65%'>
  <p class="gbsCaption">
  <em>Genome Browser screenshot of the CoLoRSdb tracks with the mouseover tooltip on the DEL-159bp
      item.</em>
  </p></a>
</div>
<p>
We would like to thank Mike Schatz, Evan Eichler, and all <a href="https://colorsdb.org/membership"
target="_blank">CoLoRSdb investigators</a> for generating and making the data publicly available. We
would also like to thank Karen Wang and Jairo Navarro Gonzalez for the creation and release of these
tracks.
</p>

<a name="091925"></a>
<h2>Sep. 19, 2025 &nbsp;&nbsp; Developmental Disorders Gene2Phenotype (DDG2P) for hg38 and hg19</h2>
<p>
We are pleased to announce the release of the <a href="https://www.deciphergenomics.org/ddd/ddgenes"
target="_blank">Developmental Disorders Genotype-to-Phenotype (DDG2P)</a> track for
<a href="/cgi-bin/hgTrackUi?db=hg19&g=ddg2p&position=default" target="_blank">hg19</a> and
<a href="/cgi-bin/hgTrackUi?db=hg38&g=ddg2p&position=default" target="_blank">hg38</a>. The DDG2P
track displays genes associated with severe developmental disorders. The track can be used to
filter genomic sequencing data of people with genetic disorders to identify likely causative
variants and accelerate diagnosis.
</p>
<p>
Items in this track are colored according to the likelihood that the gene-disease
association is true:</p>
<ul>
    <li> <font style="color: green;"><b>Green</b></font> - Definitive</li>
    <li> <font style="color: blue;"><b>Blue</b></font> - Strong</li>
    <li> <font style="color: orange;"><b>Orange</b></font> - Moderate</li>
    <li> <font style="color: red;"><b>Red</b></font> - Limited</li>
    <li> <font style="color: gray;"><b>Gray</b></font> - Refuted</li>
</ul>

<p>Each <b>mouseover</b> tooltip provides the following information:</p>
<ul>
  <li><strong>G2P ID</strong>: Unique identifier assigned by the Gene2Phenotype (G2P) database.</li>
  <li><strong>Variant Consequence</strong>: Predicted effect each allele of a variant has on a
      transcript.</li>
  <li><strong>Disease Name</strong>: Name of the disease associated with the variant.</li>
  <li><strong>PubMed IDs</strong>: Publications associated with the variant.</li>
  <li><strong>Molecular Mechanism</strong>: Description of the molecular processes and interactions
      causing pathogenic effects.</li>
  <li><strong>Allelic Requirements</strong>: Number of alleles required at a locus to produce a
      pathogenic phenotype (e.g., monoallelic, biallelic).</li>
  <li><strong>Date of Last Review</strong>: Most recent date the entry was manually reviewed.</li>
</ul>

<div class="text-center">
  <a href="https://genome.ucsc.edu/s/jnavarr5/ddg2p_announcement">
  <img src="../images/newsArchImages/ddg2p.png"
       alt="DDG2P track with the mouseover tooltip on the BRCA1 gene." width='75%'>
  <p class="gbsCaption">
  <em>Genome Browser screenshot of the DDG2P track with the mouseover tooltip on the BRCA1 gene.</em>
  </p></a>
</div>
<p>
We would like to thank the <a href="https://www.ebi.ac.uk/gene2phenotype/about/project"
target="_blank">G2P project</a> for making this data publicly available. We would also like to thank
Jaidan Jenkins-Kiefer and Jairo Navarro Gonzalez for the creation and release of the Genome Browser
tracks.</p>

<a name="082125"></a>
<h2>Aug. 21, 2025 &nbsp;&nbsp; MaveDB Experiment Heatmaps and Alignment track for hg38</h2>
<p>
We are excited to announce the release of the <a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=mavedb_maps">MaveDB Experiment Heatmaps and
Alignment</a> track for hg38. This release provides heatmaps of <b>m</b>ultiplexed <b>a</b>ssays of
<b>v</b>ariant <b>e</b>ffects (MAVE) from <a target="_blank" href="https://mavedb.org/">MaveDB</a>.
Each heatmap presents the results of an experiment where many small substitutions were tested within
a gene to examine their functional consequences. Accompanying tracks display alignments of each
experiment sequence to the genome.
</p>
<p>
Please note that only a subset of MaveDB experiments could be displayed as heatmaps; the sequence
alignments in this track only cover those experiments.</p>

<div class="text-center">
  <a href="https://genome.ucsc.edu/s/view/MaveDB_BRCA1">
  <img src="../images/newsArchImages/MaveDB_BRCA1_exon_19.png"
       alt="MaveDB heatmap and alignment tracks for the BRCA1 exon 19" width='75%'>
  <p class="gbsCaption">
  <em>Genome Browser screenshot of BRCA1 exon 19 showing the alignment and heatmap tracks.</em>
  </p></a>
</div>
<p>
Hover over each item in the heatmap to see the consequence of substituting individual amino acids
within the genome with alternatives. Score ranges vary among experiments, but each is presented
with the highest scores in <span style="color: #FF0000; font-weight: bold;">red</span>, the lowest
scores in <span style="color: #0000FF; font-weight: bold;">blue</span>, and scores at the midpoint
between the two in <span style="color: #C0C0C0; font-weight: bold;">silver</span>. Higher scores
correspond to a higher enrichment level for that variant compared to others in the experiment set.
</p>
<p>
We would like to thank Jeremy Arbesfeld and the MaveDB team for making this data publicly
available. We would also like to thank Melissa Cline, Jonathan Casper, and Jairo Navarro for the
creation and release of the Genome Browser tracks.</p>

<a name="081125"></a>
<h2>Aug. 11, 2025 &nbsp;&nbsp; CLS Long-read RNA transcripts track for hg38 and mm10</h2>
<p>
New Capture long-seq (CLS) long-read lncRNAs tracks are available for 
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=clsLongReadRnaTrack" target="_blank">hg38</a> and
<a href="/cgi-bin/hgTrackUi?db=mm10&position=default&g=clsLongReadRnaTrack" target="_blank">mm10</a>. 
These tracks represent the results of targeted long-read RNA sequencing aimed at identifying lowly expressed lncRNAs. 
The data is comprised of the capture target regions, the reads, and transcript models organized in the following way:</p>
<ul>
  <li><b>Targets:</b> Targeted RNA capture regions</li>
  <li><b>Models:</b> Transcript models generated from reads and merging</li>
  <li><b>Sample models:</b> Transcript models by sample in which they were observed</li>
  <li><b>Per-experiment reads:</b> Read mappings per experiment</li>
  <li><b>Per-experiment Models:</b> Transcript models generated from the experiments</li>
</ul>

<div class="text-center">
  <img src="../images/clsLongTranscript.png" alt="CLS Long-read RNA transcripts track options" width='55%'>
</div>

<p>
See the <a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=clsLongReadRnaTrack" target="_blank">track description page</a>
and the <a href="https://github.com/guigolab/CLS3_GENCODE" target="_blank">GENCODE CLS Project site</a> to learn more. Observant users may also
notice that the <b>mRNA and EST</b> track group has now been renamed to <b>RNA and Transcriptome</b> for many species, including
human and mouse. This change updates the track group to be more inclusive of new technology and data types.</p>
<p>
We want to thank the <a href="https://www.gencodegenes.org/" target="_blank">GENCODE consortium</a>
for providing these data. We would also like to thank Mark Diekhans and Lou Nassar for the development and release of these tracks.</p>

</p>
<a name="080825"></a>
<h2>Aug. 08, 2025 &nbsp;&nbsp; PanelApp Australia track for hg38 and hg19</h2>
<p>
We are pleased to announce the addition of the <a href="https://panelapp-aus.org/"
target="_blank">PanelApp Australia</a> tracks for the human assemblies
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=panelApp" target="_blank">hg38</a> and
<a href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=panelApp" target="_blank">hg19</a>,
available in the PanelApp composite track. The PanelApp track shows expert, crowdsourced diagnostic
disease panels among genes, copy-number variants (CNV), and short tandem repeats (STR). PanelApp
Australia was originally launched by
<a href="https://www.australiangenomics.org.au/tools-and-resources/panelapp-australia/"
target="_blank">Australian Genomics</a> in 2019 in collaboration with
<a href="https://panelapp.genomicsengland.co.uk/" target="_blank">Genomics England</a> and is
currently supported by Genomics Australia. The PanelApp Australia track contains data that differs
from the Genomics England PanelApp; more details are available on the 
<a href="/cgi-bin/hgTrackUi?db=hg38&g=panelApp&position=default#TRACK_HTML"
target="_blank">track description page.</a>
<p>
We have also updated the mouse hover for the Genomics England PanelApp track, which now shows the
gene name, associated panel, mode of inheritance (if known), related phenotypes, and confidence
level.</p>
<div class="text-center">
  <img src="../images/panelAppAustralia.jpg"  width='80%'></a>
</div>
<p>
We want to thank Jean-Madeleine for this data request and feedback, as well as Zornitza Stark from
Australia PanelApp for providing guidance. We would also like to thank Beagan Nguy, Lou Nassar, and
Gerardo Perez of the Genome Browser team for the development and release of this track.</p>

<a name="080125"></a>
<h2>Aug. 01, 2025 &nbsp;&nbsp; PubTator Variants track for human, hg38 and hg19</h2>
<p>
We are excited to announce the release of the PubTator Variants track for human assemblies,
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=pubtator">hg38</a> and
<a href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=pubtator">hg19</a>. These tracks were created
using PubTator3 data and are freely accessible to the research community. PubTator3 is a web-based
system that offers a comprehensive set of features and tools that allow researchers to
explore biomedical literature for knowledge discovery. It uses
text mining and AI techniques to annotate and unify bio-entities and their
corresponding relations for semantic and relation searches.</p>
<p>
We would like to thank the PubTator 3.0 authors for generating and making the data publically
available. We would also like to thank Max Haeussler and Johannes Birgmeier for creating the tracks,
and Jairo Navarro the release of the tracks.
</p>


<a name="073125"></a>
<h2>July 31, 2025 &nbsp;&nbsp; New bedMethyl and bigMethyl track type</h2>
<p>
We are excited to announce support for a new track format for visualizing DNA
methylation data:
<a href="https://www.encodeproject.org/data-standards/wgbs/" target="_blank">bedMethyl</a>.
This format, and its binary-indexed counterpart, <b>bigMethyl</b>, is designed to
represent methylation calls from bisulfite sequencing or similar methods at single-base resolution
across the genome.</p>
<p>
The <a href="help/bedMethyl.html" target="_blank">bedMethyl</a> format extends the standard 
<a href="/FAQ/FAQformat.html#format1" target="_blank">BED 9 format</a> to include additional fields
such as coverage, percent methylation, and the number of modified and canonical bases. When viewed
in the Genome Browser, hovering over or clicking an item reveals these additional details. Items are
color-coded from <b><font color="#0000FF">0%</font></b> methylated (blue) to
<b><font color="#FF0000">100%</font></b> (red). Methylation calls are shown separately for CpG sites
(<b>m</b>) and non-CpG (CHG/CHH) sites (<b>h</b>).
</p>
<p class="text-center">
<img "1080" height="148" src="../../images/bedMethylEx.jpg">
</p>
<p>
The <a href="help/bedMethyl.html#bigMethyl" target="_blank">bigMethyl</a>
format allows fast access to large methylation datasets and is ideal for displaying
these tracks at scale in the Genome Browser.</p>
<p>
For more information, see the
<a href="help/bedMethyl.html" target="_blank">bedMethyl and bigMethyl Track Format</a> help page.
</p>
<p>
We would like to thank Brian Joseph Raney, Max Haeussler, and Gerardo Perez for their work
developing and testing this track format.</p>
 
</p>
<a name="072325"></a>
<h2>July 23, 2025 &nbsp;&nbsp; Bionano DLE-1 CTTAAG sites for human, hg38 and hg19</h2>
<p>
We are happy to announce the release of the Bionano DLE-1 track for human assemblies
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=genotypeArrays" target="_blank">hg38</a> and
<a href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=genotypeArrays" target="_blank">hg19</a>.
These tracks show the CTTAAG sites used by the
<a href="https://www.bionano.com/" target="_blank">Bionano Optical Genome Mapping system</a>, an
assay to detect structural variants.
</p>
<p>
OGM (Optical Genome Mapping) begins with the isolation of ultra-high molecular weight (UHMW) DNA
from blood, bone marrow aspirates, cultured cells (including chorionic villi and amniocytes),
tissue, or tumor biopsies. A single enzymatic reaction places fluorescent labels throughout
the genome at a specific sequence motif. The labeled DNA molecules are linearized in nanochannel
arrays on the chip. Changes in the patterning or spacing of the labels are identified by software
solutions to detect all classes of structural variants accurately. The OGM data generated can be
analyzed alone, or in combination with sequencing or array data.</p>
<p>
We would like to thank Bionano Laboratories for providing the BED files for the OGM data. We would
also like to thank Max Haeussler and Jairo Navarro for the creation and release of the UCSC Genome
Browser tracks.
</p>

<a name="071525"></a>
<h2>July 15, 2025 &nbsp;&nbsp; New pathogenicity prediction scores for human: MutScore and M-CAP</h2>
<p>
We have two new pathogenicity prediction score tracks available in our 
<a href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=predictionScoresSuper"
target="_blank">Deleteriousness Predictions</a> super track.</p>
<ul>
<li><a href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=mutScore"
target="_blank">MutScore</a> (hg19/hg38) - MutScore is a pathogenicity predictor that integrates unsupervised 
features of single DNA variants with information derived from such clusters. The predictive model 
for MutScore was trained with a random forest approach on medically-relevant mutations and 
subsequently tested against various genomic databases for both hereditary conditions and 
cancer (ClinVar, HGMD, and DoCM) to achieve high performance.</li>
<li><a href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=mcap"
target="_blank">M-CAP</a> (hg19) - M-CAP is a pathogenicity classifier for
rare missense variants tuned to the high sensitivity required in the clinic. By combining previous
pathogenicity scores (including SIFT, Polyphen-2 and CADD) with novel features and a powerful
model, it yields an effective classifier, reducing a typical exome/genome rare (&lt;1%) missense
variant (VUS) list from 300 to 120, while never mistaking 95% of known pathogenic variants as benign.</li></ul>
<p>
See the track description page
for more information and interpretation guidelines.</p>
<p>
We would like to thank the authors of <a href="https://mutscore.iob.ch/"
target="_blank">MutScore</a> and <a href="http://bejerano.stanford.edu/mcap/"
target="_blank">M-CAP</a> for creating and providing these data. We would also
like to thank Max Haeussler and Lou Nassar for the development and release of these tracks.</p>

<a name="071525a"></a>
<h2>July 15, 2025 &nbsp;&nbsp; ENCODE4 Long-read RNA-seq Transcripts</h2>
<p>
We are pleased to announce the release of the ENCODE4 long-read RNA-seq transcripts track for
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=encode4LongRnaTranscripts"
target="_blank">hg38</a> and
<a href="/cgi-bin/hgTrackUi?db=mm10&position=default&g=encode4LongRnaTranscripts"
target="_blank">mm10</a>.
This track annotates trancripts using numerical triplets representing the
identity of the start site, exon junction chain, and transcript end site of
each transcript. This is presented alongside sample enrichment information to
show how promoter selection, splice pattern, and 3’ processing are deployed
across human tissues.
</p>

<div class="text-center">
  <img src="../images/encode4LongReadTranscriptsHelp.png"
  alt="Screenshot showing ENCODE4 long-read transcripts track">
</div>
<p>
Transcripts are labeled with triplets, e.g. [1,1,1] or [1,1,3] or [2,1,3].
If transcripts share a number in any of the positions that means they share
that feature, e.g. sharing a 8 in the second position but different numbers in
the others means those two transcripts share the same set of exons, but different
start and end sites.
</p>

<p>
This track is part of a &quot;Long-read Transcripts&quot; supertrack that will
consist of other datasets derived from third-generation sequencing technology,
such as PacBio and Oxford Nanopore.
</p>

<a name="070925"></a>
<h2>July 09, 2025 &nbsp;&nbsp; Unusually Conserved Regions track for hg38</h2>
<p>
We are happy to announce the release of the <a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=unusualcons"
target="_blank">Unusually Conserved Regions</a> track for GRCh38/hg38. The supertrack consists of
12 tracks that show regions of unusual conservation in humans relative to other organisms. The full
list of tracks, along with item count and coverage is summarized in the following table:
</p>
<p>
<table class="stdTbl">
    <tbody><tr>
        <th>Track</th>
        <th>Count</th>
        <th>Coverage in bp</th>
    </tr>
    <tr>
        <td>HAQERS: Human Ancestor Quickly Evolved Regions</td>
        <td>1,580</td>
        <td>1,410,669</td>
    </tr>
    <tr>
        <td>HARs: Human Accelerated Regions (HARs) merged from various
            publications by the Pollard Lab</td>
        <td>2,647</td>
        <td>681,420</td>
    </tr>
    <tr>
        <td>Long hConDels: Long Human Conserved Deletions - present in chimp and
            macaque but deleted in humans</td>
        <td>583</td>
        <td>293,809</td>
    </tr>
    <tr>
        <td>Short hConDels: Human Conserved Deletions &lt; 40bp</td>
        <td>10,032</td>
        <td>1,968,123</td>
    </tr>
    <tr>
        <td>UCNEBase Chicken - Chicken-conserved elements</td>
        <td>4,351</td>
        <td>1,415,142</td>
    </tr>
    <tr>
        <td>UCNEBase Paralogs - Paralogous elements</td>
        <td>987</td>
        <td>215,800</td>
    </tr>
    <tr>
        <td>UCNEBase Clusters: Ultra-conserved genomic regulatory blocks</td>
        <td>239</td>
        <td>199,269,634</td>
    </tr>
    <tr>
        <td>Ultracons: Ultraconserved regions -
            100% identical in human, mouse and rat, &gt;200bp</td>
        <td>481</td>
        <td>126,007</td>
    </tr>
    <tr>
        <td>UltraZoos: Ultraconserved regions in Zoonomia alignment -
            100% identical in 235 species, &gt;20bp</td>
        <td>4,552</td>
        <td>131,661</td>
    </tr>
    <tr>
        <td>ZooHARs: Human Accelerated Regions (HARs) from Zoonomia alignments</td>
        <td>312</td>
        <td>49,173</td>
    </tr>
    <tr>
        <td>Zoonomia RoCCs: Runs of contiguous phyloP constraint</td>
        <td>595,536</td>
        <td>26,995,284</td>
    </tr><tr>
        <td>Zoonomia UNICORNs: Unannotated Intergenic Constrained Regions</td>
        <td>423,586</td>
        <td>16,155,520</td>
    </tr>
</tbody></table></p>
<p>
We would like to thank Katie Pollard, Hiram Clawson, James Xue, Matt Christmas, Carol Nguyen,
and Mark Diekhans for providing the data. We would also like to thank Max Haeussler and Jairo
Navarro for the creation and release of the tracks.
</p>

<a name="070725"></a>
<h2>July 7, 2025 &nbsp;&nbsp; UCSC Genome Browser turns 25</h2>
<p>
July 7th marks the 25th anniversary of the human genome going online, and the start of the 
UCSC Genome Browser. Our visualization journey 
began when Jim Kent wrote the <a href = "https://hgwdev-hiram.gi.ucsc.edu/intronerator/index.html"
target="_blank">Intronerator</a>, which displayed regions of C. elegans. This was the 
foundation for the UCSC Genome Browser as we know it now, and decades later, we are 
proud to continue serving our ever-growing users.<br><br>
So, how has the Browser evolved over the years?
</p>
<ul>
<li>We now have <b>165k</b> monthly users and <b>1.5M</b> users per year
<li>The project is now comprised of <b>three million lines of code</b>
<li>We have responded to over <b>15k</b> mailing questions from our users
<li>The Genome Browser’s data exchange formats, like bigBed and track hubs, 
have become industry standards, allowing other software like IGV to build their tooling upon our data
</ul>

<div class="text-center">
  <img src="../images/ucscGenomeBrowser2003Database.png" alt="Example screenshot from 2003 UCSC Genome Browser"
width="50%">
</div>

<p>
With user growth each year, the Genome Browser has continued to become more critically 
important to research with time. A recent <a href="https://www.nature.com/articles/d41586-025-02034-7"
target="_blank">Nature article</a> shared some examples from 
users of the Genome Browser for the 25th year anniversary. See our <a href="/goldenPath/history.html"
target="_blank">history page</a> if you would like to learn more about the project's past.</p>

<a name="062625"></a>
<h2>June 26, 2025 &nbsp;&nbsp; New Genome Browser tutorials</h2>
<p>
We have a new training page, <a href="/docs/tutorials/">Genome Browser Tutorials</a>, that consists of independent 
tutorials covering some of our most popular tools. Each tutorial is comprised of three parts:</p>
<ul>
<li>An annotated screenshot identifying the important elements on the page</li>
<li>A guided walkthrough that leads you through the steps to learn the material</li>
<li>An interactive tutorial that redirects you to the tool and highlights elements on your screen 
while explaining each step</li></ul>
<p>
The page contains three tutorials, with more planned in the future:</p>
<ul>
<li><a href="/docs/tutorials/gb101.html">Genome Browser 101</a>: A basic introduction to the most common 
UCSC Genome Browser usage, the tracks display</li>
<li><a href="/docs/tutorials/gatewayTutorial.html">Gateway Page:</a> Learn how best to use the Gateway page 
to find assemblies and download shortcuts</li>
<li><a href="/docs/tutorials/tableBrowserTutorial.html">Table Browser:</a> Learn to extract Genome Browser 
data sets and configure the outputs</li>
</ul>
<p>
We would also like to remind everyone of our <a href="/training/index.html">training page</a> 
and <a href="/training/education/index.html">education modules</a>, which use 
the Genome Browser to teach various topics on genomics and molecular biology.</p>
<p>
We would like to thank Matthew Speir, Max Haeussler, Jairo Navarro, Gerardo Perez
and Lou Nassar for creating these tutorials.</p>

<a name="062425"></a>
<h2>June 24, 2025 &nbsp;&nbsp; New GENCODE gene tracks: Human V48 (hg19/hg38) - Mouse M37
(mm39)</h2>
<p>
We are happy to announce new GENCODE Gene annotation tracks, which correspond
to <a href="https://www.ensembl.info/2025/05/07/ensembl-114-has-been-released/"
target="_blank">Ensembl 114</a>, for three assemblies: <a target="_blank"
href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV48lift37&position=default">hg19/GRCh37</a>,
<a target="_blank" href="../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeGencodeV48&position=default">hg38/GRCh38</a>,
and <a target="_blank" href="../../cgi-bin/hgTrackUi?db=mm39&g=wgEncodeGencodeVM37&position=default">mm39/GRCm39</a>.
For human, the GENCODE V48 annotations were mapped to hg38/GRCh38 and then back-mapped
to the hg19/GRCh37 assembly. For all three assemblies, the gene sets contain the following
tracks:</p>
<ul>
  <li><b>Basic</b> - a subset of the <em>Comprehensive set</em>.</li>
  <li><b>Comprehensive</b> - all GENCODE coding and non-coding transcript annotations, including
      polymorphic pseudogenes. This includes both manual and automatic annotations.</li>
  <li><b>Pseudogenes</b> - all annotations except polymorphic pseudogenes.</li></ul>
<p>
The hg38 and mm39 assemblies also include the following track that is not available on hg19:
</p>
<ul>
  <li>
      <b>PolyA</b> - polyA signals and sites manually annotated on the genome based on transcribed
      evidence (ESTs and cDNAs) of 3' end of transcripts containing at least 3 A's not matching the
      genome.</li>
</ul>
<p>
Details on each release can be found on the <a target="_blank"
href="https://www.gencodegenes.org/">GENCODE site</a>. This includes <a target="_blank"
href="https://www.gencodegenes.org/human/stats.html">statistics on each release</a>.
<p>We would like to thank the <a target="_blank"
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Mark Diekhans and Gerardo Perez for the development and
release of these tracks.</p>


<a name="061125"></a>
<h2>June 11, 2025 &nbsp;&nbsp; EVA SNP release 7 for 40 assemblies</h2>
<p>
We are happy to announce the release of the EVA SNP Release 7 tracks for 40 assemblies, which now
includes <a href="../cgi-bin/hgTrackUi?db=bosTau9&g=evaSnp7&position=default"
	target="_blank">bosTau9</a>,
<a href="../cgi-bin/hgTrackUi?db=ce11&g=evaSnp7&position=default" target="_blank">ce11</a>, and
<a href="../cgi-bin/hgTrackUi?db=rn6&g=evaSnp7&position=default" target="_blank">rn6</a>.
These tracks contain mappings of single nucleotide variants and
small insertions and deletions (indels) &mdash; collectively Simple Nucleotide Variants (SNVs)
&mdash; from the European Variation Archive
(<a href="https://www.ebi.ac.uk/eva/" target="_blank">EVA</a>) Release 7. The full list of
assemblies that contain the EVA SNP release 7 track is below:</p>
<p>
<div class="container">
  <div class="row">
    <div class="col-sm-4">
    <ul>
      <li>A. gambiae <a href="../cgi-bin/hgTrackUi?db=anoGam3&g=evaSnp7"
        target="_blank">(anoGam3)</a></li>
      <li>C. intestinalis <a href="../cgi-bin/hgTrackUi?db=ci3&g=evaSnp7"
        target="_blank">(ci3)</a></li>
      <li>Bison <a href="../cgi-bin/hgTrackUi?db=bisBis1&g=evaSnp7"
        target="_blank">(bisBis1)</a></li>
      <li>Cat <a href="../cgi-bin/hgTrackUi?db=felCat8&g=evaSnp7"
        target="_blank">(felCat8)</a>,
        <a href="../cgi-bin/hgTrackUi?db=felCat9&g=evaSnp7"
        target="_blank">(felCat9)</a></li>
      <li>Chicken <a href="../cgi-bin/hgTrackUi?db=galGal4&g=evaSnp7"
        target="_blank">(galGal4)</a>,
        <a href="../cgi-bin/hgTrackUi?db=galGal5&g=evaSnp7"
        target="_blank">(galGal5)</a>,
        <a href="../cgi-bin/hgTrackUi?db=galGal6&g=evaSnp7"
        target="_blank">(galGal6)</a></li>
      <li>Chimp <a href="../cgi-bin/hgTrackUi?db=panTro4&g=evaSnp7"
        target="_blank">(panTro4)</a>,
        <a href="../cgi-bin/hgTrackUi?db=panTro5&g=evaSnp7"
        target="_blank">(panTro5)</a></li></li>
      <li>Cow <a href="../cgi-bin/hgTrackUi?db=bosTau7&g=evaSnp7"
        target="_blank">(bosTau7)</a>
        <a href="../cgi-bin/hgTrackUi?db=bosTau9&g=evaSnp7"
        target="_blank">(bosTau9)</a></li>
      <li>Crab-eating macaque <a href="../cgi-bin/hgTrackUi?db=macFas5&g=evaSnp7"
        target="_blank">(macFas5)</a></li>
      <li>D. melanogaster <a href="../cgi-bin/hgTrackUi?db=dm6&g=evaSnp7"
        target="_blank">(dm6)</a></li>
    </ul></div>
   <div class="col-sm-4">
      <ul>
      <li>Dog <a href="../cgi-bin/hgTrackUi?db=canFam3&g=evaSnp7"
        target="_blank">(canFam3)</a></li>
      <li>C. elegans <a href="../cgi-bin/hgTrackUi?db=ce11&g=evaSnp7"
        target="_blank">(ce11)</a></li>
      <li>Gorilla <a href="../cgi-bin/hgTrackUi?db=gorGor3&g=evaSnp7"
        target="_blank">(gorGor3)</a>,
        <a href="../cgi-bin/hgTrackUi?db=gorGor4&g=evaSnp7"
        target="_blank">(gorGor4)</a></li>
      <li>Green monkey <a href="../cgi-bin/hgTrackUi?db=chlSab2&g=evaSnp7"
        target="_blank">(chlSab2)</a></li>
      <li>Horse <a href="../cgi-bin/hgTrackUi?db=equCab3&g=evaSnp7"
        target="_blank">(equCab3)</a></li>
      <li>Marmoset <a href="../cgi-bin/hgTrackUi?db=calJac3&g=evaSnp7"
        target="_blank">(calJac3)</a></li>
      <li>Mouse <a href="../cgi-bin/hgTrackUi?db=mm10&g=evaSnp7"
        target="_blank">(mm10)</a>,
        <a href="../cgi-bin/hgTrackUi?db=mm39&g=evaSnp7"
        target="_blank">(mm39)</a></li>
      <li>Nile tilapia <a href="../cgi-bin/hgTrackUi?db=oreNil2&g=evaSnp7"
        target="_blank">(oreNil2)</a></li>
      <li>Orangutan <a href="../cgi-bin/hgTrackUi?db=ponAbe3&g=evaSnp7"
        target="_blank">(ponAbe3)</a></li>
    </ul></div>
    <div class="col-sm-4">
      <ul>
      <li>Pig <a href="../cgi-bin/hgTrackUi?db=susScr3&g=evaSnp7"
        target="_blank">(susScr3)</a>,
        <a href="../cgi-bin/hgTrackUi?db=susScr11&g=evaSnp7"
        target="_blank">(susScr11)</a></li>
      <li>Rat <a href="../cgi-bin/hgTrackUi?db=rn5&g=evaSnp7"
        target="_blank">(rn5)</a>,
        <a href="../cgi-bin/hgTrackUi?db=rn6&g=evaSnp7"
        target="_blank">(rn6)</a>,
        <a href="../cgi-bin/hgTrackUi?db=rn7&g=evaSnp7"
        target="_blank">(rn7)</a></li>
      <li>Rhesus <a href="../cgi-bin/hgTrackUi?db=rheMac8&g=evaSnp7"
        target="_blank">(rheMac8)</a>,
        <a href="../cgi-bin/hgTrackUi?db=rheMac10&g=evaSnp7"
        target="_blank">(rheMac10)</a></li>
      <li>S. cerevisiae <a href="../cgi-bin/hgTrackUi?db=sacCer3&g=evaSnp7"
        target="_blank">(sacCer3)</a></li>
      <li>Sheep <a href="../cgi-bin/hgTrackUi?db=oviAri3&g=evaSnp7"
        target="_blank">(oviAri3)</a>,
        <a href="../cgi-bin/hgTrackUi?db=oviAri4&g=evaSnp7"
        target="_blank">(oviAri4)</a></li>
      <li>Turkey <a href="../cgi-bin/hgTrackUi?db=melGal5&g=evaSnp7"
        target="_blank">(melGal5)</a></li>
      <li>Zebra finch <a href="../cgi-bin/hgTrackUi?db=taeGut2&g=evaSnp7"
        target="_blank">(taeGut2)</a></li>
      <li>Zebrafish <a href="../cgi-bin/hgTrackUi?db=danRer7&g=evaSnp7"
        target="_blank">(danRer7)</a>,
        <a href="../cgi-bin/hgTrackUi?db=danRer10&g=evaSnp7"
        target="_blank">(danRer10)</a>,
        <a href="../cgi-bin/hgTrackUi?db=danRer11&g=evaSnp7"
        target="_blank">(danRer11)</a></li>
    </ul></div>
  </div>
</div></p>

<p>
These variants are classified by EVA into one of the following <a target="_blank"
href="http://www.sequenceontology.org/">sequence ontology</a> terms:
</p>
<ul>
  <li> <b>substitution</b> &mdash;
       A single nucleotide in the reference is replaced by another, alternate allele.
  <li> <b>deletion</b> &mdash;
       One or more nucleotides are deleted.  The representation in the database is to
       display one additional nucleotide in both the Reference field (Ref) and the
       Alternate Allele field (Alt).  E.g. a variant that is a deletion of an A
       may be represented as Ref = GA and Alt = G.
  <li> <b>insertion</b> &mdash;
       One or more nucleotides are inserted.  The representation in the database is to
       display one additional nucleotide in both the Reference field (Ref) and the
       Alternate Allele field (Alt).  E.g. a variant that is an insertion of a T may
       be represented as Ref = G and Alt = GT.
  <li> <b>delins</b> &mdash;
       Similar to a tandem repeat, in that the runs of Ref and Alt Alleles are of
       different length, except that there is more than one type of nucleotide,
       e.g., Ref = CCAAAAACAAAAACA, Alt = ACAAAAAC.
  <li> <b>multipleNucleotideVariant</b> &mdash;
       More than one nucleotide is substituted by an equal number of different
       nucleotides, e.g.,  Ref = AA, Alt = GC.
  <li> <b>sequence alteration</b> &mdash;
       A parent term meant to signify a deviation from another sequence. Can be
       assigned to variants that have not been characterized yet.
</ul>
<p>
The variants have also been annotated with our <a href="/cgi-bin/hgVai" target="_blank">Variant
Annotation Integrator</a> tool with functional classes such as synonymous variant,
missense variant, stop gained, etc. For additional details on the track colors,
as well as the filters and metadata on each variant, see the
<a href="/cgi-bin/hgTrackUi?db=anoGam3&c=chr2R&g=evaSnpContainer&position=default#TRACK_HTML"
        target="_blank">track description page</a>.</p>

<p>
We would like to thank the <a href="https://www.ebi.ac.uk/eva/" target="_blank">European Variation
Archive</a> for making these data publicly available. We would also like to thank Gerardo Perez, Luis Nassar,
and Angie Hinrichs for the creation and release of these tracks.
</p>

<a name="052125"></a>
<h2>May 21, 2025 &nbsp;&nbsp; Varaico variants track available for human (hg38 &amp; hg19)</h2>
<p>
We are happy to announce the release of the Varaico variants track for human assemblies
<a href="/cgi-bin/hgTrackUi?db=hg38&g=varaico" target="_blank">hg38/GRCh38</a> and
<a href="/cgi-bin/hgTrackUi?db=hg19&g=varaico" target="_blank">hg19/GRCh37</a>. Varaico stands for
&quot;<b>Va</b>riation <b>R</b>esearch <b>A</b>dvancing <b>I</b>nsight in <b>C</b>omplex
<b>O</b>rganisms&quot;. Varaico was created using literature mining, similar to AVADA.
</p>
<p>
Varaico variants are generated by an automated process that extracts purely factual information
about genes from scientific papers (by matching strings against gene names) and HGVS variant
descriptions (using regular expressions). Varaico aims to reduce false-positive gene and variant
mentions and link them together appropriately, but nonetheless, many variants displayed are not
mapped to the genomic position intended by the authors. Visit the
<a href="https://varaico.com/" target="_blank">Varaico website</a> for more details.</p>
<p>
  Mouse over the variants to show the gene, variant, latest author/year/title, number of
  publications mentioning the variant, and variant effect.
  Varaico variants may or may not be disease-causing.</p>
<div class="text-center">
  <img src="../images/newsArchImages/Varaico.png" width="80%">
  <p class="gbsCaption">
  <em>Mouse-over is shown for the orange item (BRCA1 c.4136_4137del).</em>
  </p>
</div>
<p>
We would like to thank Johannes Birgmeier for generating and making the Varaico data available. We
would also like to thank Maximilian Haeussler and Jairo Navarro Gonzalez for creating and releasing
these tracks. 
</p>

<a name="051925"></a>
<h2>May 19, 2025 &nbsp;&nbsp; New GENCODE &quot;knownGene&quot; V48 for human (hg38/hg19) and VM37
for mouse (mm39)</h2>
<p>
We are pleased to announce the release of the GENCODE Genes V48 for human
(<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=knownGene">hg38/GRCh38</a> &amp;
<a href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=knownGene">hg19/GRCh37</a>)
and the GENCODE Genes VM37 for mouse
(<a href="/cgi-bin/hgTrackUi?db=mm39&position=default&g=knownGene">mm39/GRCm39</a>).
The GENCODE &quot;knownGene&quot; V48 and VM37 gene tracks were built using a UCSC
pipeline (knownGene) and the GENCODE comprehensive gene set to generate high-quality manual
annotations merged with evidence-based automated annotations. The GENCODE &quot;knownGene&quot;
tracks are our default gene tracks, which have extensive associations to external sources. This
allows for additional metadata on every item as well as external links. The track description
pages contain options for configuring the display, such as showing non-coding genes, splice
variants, and pseudogenes.</p>
<p>
Below is a summary of the contents found in each release. For more details visit the
<a target="_blank" href="https://www.gencodegenes.org/">GENCODE site</a>.</p>
<p>
<table class="stdTbl">
<tr><th COLSPAN=4>GENCODE v48 Release Stats</th></tr>
<tr align=left><th>Genes</th><th>Observed</th><th>Transcripts</th><th>Observed</th></tr>
<tr align=left><td>Protein-coding genes</td><td>19,435</td>
               <td>Protein-coding transcripts</td><td>89,843</td></tr>
<tr align=left><td>Long non-coding RNA genes</td><td>35,901</td>
               <td><font size="-1">- full length protein-coding</font></td><td>65,024</td></tr>
<tr align=left><td>Small non-coding RNA genes</td><td>7,563</td>
               <td><font size="-1">- partial length protein-coding</font></td><td>24,819</td></tr>
<tr align=left><td>Pseudogenes</td><td>14,695</td>
               <td>Nonsense mediated decay transcripts</td><td>21,902</td></tr>
<tr align=left><td>Immunoglobulin/T-cell receptor gene segments</td>
               <td>649</td><td>Long non-coding RNA loci transcripts</td><td>191,076</td></tr>
<tr align=left><td>Total No of distinct translations</td><td>65,814</td>
               <td>Genes that have more than one distinct translations</td><td>13,646</td></tr>
</table><BR>
</p>
<p>
<table class="stdTbl">
<tr><th COLSPAN=4>GENCODE VM37 Release Stats</th></tr>
<tr align=left><th>Genes</th><th>Observed</th><th>Transcripts</th><th>Observed</th></tr>
<tr align=left><td>Protein-coding genes</td><td>21,529</td>
               <td>Protein-coding transcripts</td><td>58,636</td></tr>
<tr align=left><td>Long non-coding RNA genes</td><td>36,111</td>
               <td><font size="-1">- full length protein-coding</font></td><td>45,038</td></tr>
<tr align=left><td>Small non-coding RNA genes</td><td>6,105</td>
               <td><font size="-1">- partial length protein-coding</font></td><td>13,598</td></tr>
<tr align=left><td>Pseudogenes</td><td>13,790</td>
               <td>Nonsense mediated decay transcripts</td><td>7,247</td></tr>
<tr align=left><td>Immunoglobulin/T-cell receptor gene segments</td><td>701</td>
               <td>Long non-coding RNA loci transcripts</td><td>155,932</td></tr>
<tr align=left><td>Total No of distinct translations</td><td>44,964</td>
               <td>Genes that have more than one distinct translations</td><td>10,850</td></tr>
</table><BR>
</p>
<p>
We would like to thank the <a target="_blank"
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Jonathan Casper and Jairo Navarro Gonzalez for the
development and release of these tracks.</p>

<a name="051425"></a>
<h2>May 14, 2025 &nbsp;&nbsp; Exon Relevance RTS and gnomAD pext track for hg38</h2>
<p>
We are pleased to share our latest <a href="#022222">Recommended Track Set (RTS)</a>,
<b>Exon Relevance</b>, and a new track, 
<a href="/cgi-bin/hgTrackUi?db=hg38&g=gnomadPext" target="_blank">gnomAD Proportion 
Expression Across Transcript Scores (pext)</a> for GRCh38/hg38.</p>

<h3>Exon Relevance RTS</h3>
<p>
The Exon Relevance track set can be found in the blue bar menu (<b>Genome Browser</b> 
&gt; <b>Recommended Track Sets</b>) on both the hg19 and hg38 browsers. This track set
can assist in variant interpretation by evaluating if the variant is present in an exon 
required for function of the expressed gene product.</p>

<div class="text-center">
  <img src="../images/exonRelevanceNewsImage.png" alt="New Exon Relevance track set" width='40%'>
</div>

<h3>gnomAD Proportion Expression Across Transcript Scores (pext) for GRCh38/hg38</h3>

<p>
The hg38 gnomAD super track now includes
the gnomAD pext datasets. This track displays isoform
expression levels across 50 tissues from
the Genotype Tissue Expression (GTEx) v10 dataset, showing a comprehensive
view of the expression of exons across
a gene using the proportion expression across transcripts, or pext metric, a
transcript-level annotation metric which quantifies isoform expression for variants.
</p>
<p>
This is especially useful for those interested in alternative splicing and
clinical assessment of variants. For more
information, see the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=gnomadPext" target="_blank">track description page</a> and 
the <a href="https://www.nature.com/articles/s41586-020-2329-2" target="_blank">associated 
publication</a>.</p>

<div class="text-center">
  <a href="https://genome.ucsc.edu/s/Lou/hg38pextNews" target="_blank"><img src="../images/hg38pextTrackNewsImage.png" 
alt="Image of TRDN gene showing variants primarily expressed in heart tissue." width='80%'></a>
</div>

<p>
We would like to thank the gnomAD team and the UCSC Genome Browser team members
Jeltje van Baren, Max Haeussler, Lou Nassar, and Anna Benet-Pages for developing and
releasing this track, as well as making the Exon Relevance RTS.
</p>

<a name="050525"></a>
<h2>May 5, 2025 &nbsp;&nbsp; VISTA Enhancers track update for Human and Mouse</h2>
<p>
We are happy to announce an update to the VISTA Enhancers tracks for human
(<a target="_blank"
	href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=vistaEnhancersBb">GRCh38/hg38</a>
and <a target="_blank"
        href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=vistaEnhancersBb">GRCh37/hg19</a>)
and mouse (<a target="_blank"
        href="/cgi-bin/hgTrackUi?db=mm39&position=default&g=vistaEnhancersBb">GRCm39/mm39</a>
and <a target="_blank"
        href="/cgi-bin/hgTrackUi?db=mm10&position=default&g=vistaEnhancersBb">GRCm38/mm10</a>)
genome assemblies. The track shows potential enhancers whose activity was experimentally validated
in transgenic mice. With this update, the tracks now include over 1,900 new entries for the human
assemblies and over 2,500 new entries for the mouse assemblies. Hovering over an enhancer annotation
now displays the tissues in which the element is active. Most of these noncoding elements were
selected for testing based on their extreme conservation in other vertebrates or epigenomic
evidence (ChIP-Seq) of putative enhancer marks. More information on this track can be found on the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=vistaEnhancersBb&position=default#TRACK_HTML"
	target="_blank">track description</a> page.</p>
<p>
We would like to thank the Lawrence Berkeley National Laboratory and the
<a target="_blank" href="https://enhancer.lbl.gov/vista/">VISTA Enhancer team</a> for providing
this data. We would also like to thank Max Haeussler and Gerardo Perez for their efforts on this
release.</p>

<a name="033125"></a>
<h2>Mar. 31, 2025 &nbsp;&nbsp; New Pseudogenes track for hg38</h2>
<p>
We are happy to announce the release of the
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&g=pseudogenes&position=default">Pseudogenes</a>
track for the hg38 (GRCh38) genome assembly. The composite track contains pseudogene predictions and
their corresponding parent genes, as identified by <a href="http://pseudogene.org/pseudopipe/"
target="_blank">PseudoPipe</a>, a homology-based computational pipeline that searches a mammalian
genome and identifies pseudogene sequences.</p>
<p>
This composite track consists of two subtracks: the <b>Pseudogene Parents</b> track and the
<b>Pseudogenes</b> track.</p>
<ul>
  <li><b>Pseudogene Parents</b> &ndash; This track displays parent genes labeled with their HUGO
	  IDs, which were derived from Ensembl gene IDs provided by the Gerstein lab after dataset
	  creation. It includes indicators for pseudogenes, each linked to its corresponding entry
	  in the Pseudogenes track. <b><font color="#800080">Parent genes</font></b> are shown with
	  their associated pseudogenes represented as <b><font color="#A9A9A9">grey</font></b>
	  blocks.
  <li><b>Pseudogenes</b> &ndash; This track displays pseudogenes labeled with their parent HUGO ID
	  and the pseudogenes are color-coded by pseudogene type:
	  <b><font color="#FF8C00">unspecified pseudogene</font></b>, <b><font
	  color="#0000FF">unprocessed pseudogene</font></b>, and
	  <b><font color="#556B2F">processed pseudogene</font></b>.
</ul>
<div class="text-center">
  <a href="https://genome.ucsc.edu/s/gperez2/Pseudogenes" target="_blank">
  <img src="../images/newsArchImages/PseudogenesTrack.jpg" alt="hg38 session
  visualizing the Pseudogenes track" width='80%'></a>
</div>
<p>
More information on this composite track can be found on the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=pseudogenes&position=default#TRACK_HTML" target="_blank">track description</a>
page.
</p>
<p>
We would like to thank the Gerstein lab at Yale University for making this data available, and
Cristina Sisu for providing data in GTF format with parent annotations. We would also like to
thank Jeltje van Baren, Mark Diekhans, and Gerardo Perez for their efforts on this release.
</p>

<a name="031425"></a>
<h2>Mar. 14, 2025 &nbsp;&nbsp; denovo-db tracks for hg19 now available</h2>
<p>
We are excited to announce the release of two
<a href="/cgi-bin/hgTrackUi?db=hg19&c=chr7&g=denovoDbComposite" target="_blank">denovo-db V.1.6.1</a>
tracks for the hg19/GRCh37 genome assembly. The
<a href="https://denovo-db.gs.washington.edu/denovo-db/index.jsp" target="_blank">denovo-db
(version 1.6.1)</a> is a curated database of germline de novo variants in the human genome, defined
as variants present in children but absent in their parents. With advancements in whole-exome and
whole-genome sequencing, denovo-db consolidates data from extensive genetic studies, including those
on neurodevelopmental disorders, but is not restricted to specific phenotypes. This database
provides essential resources for researchers investigating variant characteristics, mutability, and
associated phenotypes.</p>
<p>
The denovo-db V.1.6.1 composite track contains <a href="https://www.sfari.org/resource/simons-simplex-collection/"
target="_blank">Simons Simplex Collection (SSC)</a> samples and non-ssc samples. The use of Simons
Simplex Collection (SSC) is limited to projects related to advancing the field of autism and related
developmental disorder research. Questions on SSC consents should be directed to
collections@sfari.org. If you have questions or comments about the denovo-db, please write to
denovo-db@uw.edu.</p>
<p>
We would like to thank denovo-db and the Simons Simplex Collection for making this data available.
We would also like to thank Megna Chalamala, Gerardo Perez, Ana Benet-Pag&#232;s, and Jairo
Navarro for the creation and release of these tracks on the UCSC Genome Browser.</p>


<a name="030625"></a>
<h2>Mar. 06, 2025 &nbsp;&nbsp; New MITOMAP track for hg38 and hg19</h2>
<p>
We are pleased to announce the release of the MITOMAP track for the
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&c=chrM&g=mitoMap">hg38/GRCh38</a>
and <a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg19&c=chrMT&g=mitoMap">hg19/GRCh37</a> human assemblies.
<a target="_blank" href="https://www.mitomap.org/foswiki/bin/view/MITOMAP/WebHome">MITOMAP</a>
is a database of human mitochondrial DNA (mtDNA) information containing a compilation of mtDNA
variation, allowing users to look up human mitochondrial gene loci, search public mitochondrial
sequences, and browse or search for reported nucleotide variants in both the general population and
clinical disease studies.
</p>
<p>
This release includes two tracks within the MITOMAP composite track:
</p>
<ul>
  <li><b>MITOMAP Control and Coding Variants</b> &ndash; Contains variants, including mini
	  insertions and deletions, in the complete mtDNA. The item colors correspond to the variant
	  type: 
	  <b><font color="#53AF01">control region</font></b>
	  vs. <b><font color="#3E89D3">coding region</font></b>.
  <li><b>MITOMAP Disease Mutations</b> &ndash; Contains disease-annotated mutations (variants) in
	  the complete mtDNA. The item colors correspond to the variant type:
          <b><font color="##015587">coding/control</font></b> 
	  vs. <b><font color="#9933FF">rRNA/tRNA</font></b>.
</ul>
<div class="text-center">
  <a href="https://genome.ucsc.edu/s/gperez2/MITOMAP" target="_blank">
  <img src="../images/newsArchImages/mitoMap_MT_TL1.jpg" alt="hg38 session
  visualizing the MITOMAP track" width='60%'></a>
</div>

<p>
More information on this composite track can be found on the
<a href="/cgi-bin/hgTrackUi?db=hg38&c=chrM&g=mitoMap#TRACK_HTML" target="_blank">track description</a>
page.
</p>
<p>
We would like to thank Shiping Zhang and the entire MITOMAP resource for making these annotations
available. We would also like to thank Max Haeussler, Luis Nassar, and Gerardo Perez for their
efforts on this release.
</p>


<a name="030525"></a>
<h2>Mar. 05, 2025 &nbsp;&nbsp; New Splicing Impact Prediction Scores super track (hg19/hg38) and SpliceVarDB track for hg38</h2>
<p>
We are excited to announce the new Splicing Impact Prediction Scores and Databases super track for
the
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&g=spliceImpactSuper&position=default">hg38/GRCh38</a>
and <a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg19&g=spliceImpactSuper&position=default">hg19/GRCh37</a> human assemblies.
This super track includes the 
<a target="_blank" href="newsarch.html#042524">AbSplice Scores</a>,
<a target="_blank" href="newsarch.html#081224">SpliceAI</a>, and, for hg38, the new
<a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg38&g=spliceVarDb&position=default">SpliceVarDB track</a>.
<a target="_blank" href="https://splicevardb.org/">SpliceVarDB</a>
is an online database consolidating over 50,000 variants assayed for their effects on
splicing in over 8,000 human genes. The SpliceVarDB authors evaluated over 500 published data
sources and established a spliceogenicity scale to standardize, harmonize, and consolidate variant
validation data generated by a range of experimental protocols. SpliceVarDB data is displayed as
lollipop plots, color-coded by score value. A score of <b>3</b> indicates a 
<span style="color: rgb(219,61,61);">Splice-altering</span></li> variant,
<b>2</b> represents a <span style="color: rgb(128,82,160);">Low-frequency</span> variant,
<b>1</b> is classified as <span style="color: rgb(57,135,204);">Normal</span>, and
<b>0</b> denotes <span style="color: rgb(140,140,140);">Conflicting</span> data.
</p>

<div class="text-center">
  <a href="https://genome.ucsc.edu/s/gperez2/SpliceVarDB" target="_blank">
  <img src="../images/newsArchImages/SpliceVarDB.jpg" alt="hg38 session
  visualizing the SpliceVarDB track" width='70%'></a>
</div>

<p>
We would like to thank the SpliceVarDB team for providing the data to UCSC. We would also like to
thank Jeltje van Baren, Max Haeussler, Luis Nassar, Jairo Navarro, and Gerardo Perez for their
efforts on this release.
</p>

<a name="030325"></a>
<h2>Mar. 03, 2025 &nbsp;&nbsp; New highlight features for track hubs now available</h2>
<p>
We are happy to announce the release of new highlight features for track hubs. These
settings follow exactly the same syntax and functionality as the <code>filter</code> trackDb
settings, except instead of items being excluded from the display, items are striped with a colored
background to appear &quot;highlighted&quot; compared to the other items in the display.
The full list of highlight trackDb settings is available on the
<a href="/goldenPath/help/trackDb/trackDbHub.html#highlight">trackDb definitions</a> page.
</p>
<p>
Please note that at this time, only one highlight color is available per track, and if multiple
highlight settings are present on the same track, only items that pass ALL highlight settings will
be highlighted.</p>
<p>
<b>Examples:</b>
<pre>
highlight.score 300
</pre>
<p>
In the example above, the <code>highlight.score</code> setting allows you to highlight items based
on the value of the <code>score</code> field. Passing a value of 300 to the setting will highlight
all items with a score of 300 or above.</p>
<pre>
highlightText.name NM*
</pre>
<p>
The example above uses the <code>highlightText</code> setting, which will apply a highlight on the
field <code>name</code>. Using this setting, any items that begin with <i>NM</i> are
highlighted.</p>
<pre>
highlightColor #ff0000
</pre>
<p>
In this final example, the <code>highlightColor</code> is used to set the default highlight color.
With this setting, all highlight stripes will use the color red, <code>#ff0000</code>.</p>
<div class="text-center">
  <img src="../images/newsArchImages/highlightAnnouncement.png"
   alt="Items in the NCBI RefSeq Historical track that begin with NM are highlighted red.">
   <p>Items in the NCBI RefSeq Historical track for hg38 have all items that begin with
      &quot;NM&quot; highlighted in red.</p>
</div>
<p>
We would like to thank Chris Lee and Jairo Navarro for their efforts in creating and testing the
highlight feature for track hubs.<p>

<a name="022425"></a>
<h2>Feb. 24, 2025 &nbsp;&nbsp; enGenome VarChat track for human (hg38 and hg19)</h2>
<p>
We are happy to announce the release of the enGenome VarChat track for the
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&g=varChat&position=default" >hg38/GRCh38</a>
and <a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg19&g=varChat&position=default">hg19/GRCh37</a> human assemblies,
available in the Variants in Papers superTrack.
<a target="_blank" href="https://varchat.engenome.com/">VarChat</a> is an open platform that
leverages the power of generative artificial intelligence to support the genomic variant
interpretation process by searching the available scientific literature for each variant and
condensing it into a brief yet informative text.
</p>

<p>
The track shows how many papers the variant was observed in, its gene, its HGVS nomenclature, and
dbSNP rsID. Variants are color-coded based on the level of literature support, as shown in the
table below:</p>
<table>
  <thead>
    <tr>
      <th style="border-bottom: 2px solid #6678B1;">Color</th>
      <th style="border-bottom: 2px solid #6678B1;">Level of literature support</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="background-color: #012840; width: 50px;"></td>
      <td align="left">High: at least 25 papers mention the variant</td>
    </tr>
    <tr>
      <td style="background-color: #03738C; width: 50px;"></td>
      <td align="left">Medium: between 10 and 24 papers mention the variant</td>
    </tr>
    <tr>
      <td style="background-color: #96D2D9; width: 50px;"></td>
      <td align="left">Low: fewer than 10 papers mention the variant</td>
    </tr>
  </tbody>
</table>
<p>
We would like to thank VarChat for providing the data to UCSC. We would also like to thank Lou
Nassar, Max Haeussler, and Gerardo Perez for their efforts on this release.</p>

<a name="022125"></a>
<h2>Feb. 21, 2025 &nbsp;&nbsp; New AlphaMissense scores track for hg38 and hg19</h2>
<p>
We are pleased to announce the addition of AlphaMissense tracks to the hg38 and
hg19 reference genomes. AlphaMissense scores predict the pathogenicity of
missense variants for all possible single amino acid substitutions in the human
proteome.
</p>

<p>
To access these tracks on the Genome Browser, please visit their description
pages below and change the tracks' visibility:
</p>
<ul>
  <li><a target="_blank"
  href="/cgi-bin/hgTrackUi?db=hg38&g=alphaMissense&position=default">AlphaMissense
  on hg38/GRCh38</a></li>
  <li><a target="_blank"
  href="/cgi-bin/hgTrackUi?db=hg19&g=alphaMissense&position=default">AlphaMissense
  on hg19/GRCh37</a></li>
</ul>
<p>
To learn more about the AlphaMissense dataset, please see the
publication by <a
href="https://www.science.org/doi/full/10.1126/science.adg7492"
target="_blank">Cheng et al. Science. 2023.</a>
</p>

<p>
We would like to thank Google DeepMind for making the data available. We would also like to thank
Jeltje van Baren and Matthew Speir for their efforts on this release.</p>

<a name="021425"></a>
<h2>Feb. 14, 2025 &nbsp;&nbsp; CIViC track for hg19 and hg38 is now available</h2>
<p>
We are excited to announce the release of the CIViC track for <a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg19&g=civic&position=default">hg19/GRCh37</a> and
<a href="/cgi-bin/hgTrackUi?db=hg38&g=civic&position=default" target="_blank">hg38/GRCh38</a>.
The <a target="_blank" href="https://civicdb.org/welcome">
Clinical Interpretation of Variants in Cancer</a> (CIViC) is an online database of clinically
relevant variant interpretations from expert and crowd-sourced peer-reviewed literature, clinical
trials, and some conference abstracts. The details page for a feature will list diseases and
therapies that have been associated with a genomic variant. Visiting the CIViC page for a variant
will allow browsing the Molecular Profiles associated with that variant, and in turn each Molecular
Profile shows the Clinical Evidence and Assertions for various diseases and therapies.
</p>

<p>
This track reflects the monthly data summaries published by CIViC. The
latest information is always available directly on the <a href="https://civicdb.org/welcome"
target="_blank">CIViC website</a>
or by its <a href="https://civicdb.org/api/graphiql" target="_blank">API</a>.
</p>

<p>
We would like to thank the CIViC contributors and organizers for curating the database and for
providing the data to UCSC. We would also like to thank Charlie Vaske and Jairo Navarro for the
creation and release of these tracks.</p>

<a name="013125"></a>
<h2>Jan. 31, 2025 &nbsp;&nbsp; New COSMIC track for hg19 and update for hg38 COSMIC track</h2>
<p>
We are happy to announce the release of the new COSMIC track for <a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg19&g=cosmicMuts&position=default">hg19/GRCh37</a> and
an update for the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=cosmicMuts&position=default" target="_blank">hg38/GRCh38</a>.
The <a target="_blank" href="https://cancer.sanger.ac.uk/cosmic/about"
>Catalogue Of Somatic Mutations In Cancer</a> (COSMIC) is an online database of expert manually
curated somatic mutation information relating to human cancers. The COSMIC track displays data from
the <a target="_blank"
href="https://cancer.sanger.ac.uk/cosmic/release_notes/">COSMIC v101 release</a>, which includes
Whole Genome Sequencing (WGS) data from 383 samples, including 305 individuals exposed to ionizing
radiation as children during the 1986 Chernobyl accident and 78 individuals born after 1986. WGS
detected a total of 318,907 somatic mutations, including 315,022 non-coding variants and 3,885
coding variants.</p>
<p>
We would like to thank COSMIC for providing the data to UCSC. We would also like to thank
Lou Nassar, Max Haeussler, and Gerardo Perez for their efforts on this release.</p>


<!-- ============= 2024 archived news ============= -->
<a name="2024"></a>

<a name="122024"></a>
<h2>Dec. 20, 2024 &nbsp;&nbsp; NCBI Gene Orthologs track available for Human (hg38), Mouse (mm39),
    Zebrafish (danRer11), Dog (canFam6), and Cow (bosTau9)</h2>
<p>
We are excited to announce a new NCBI Orthologs track for the 
<a href="/cgi-bin/hgTrackUi?db=hg38&g=refSeqComposite&position=default"
 target="_blank">Human (hg38)</a>, 
<a href="/cgi-bin/hgTrackUi?db=mm39&g=refSeqComposite&position=default"
 target="_blank">Mouse (mm39)</a>,
<a href="/cgi-bin/hgTrackUi?db=danRer11&g=refSeqComposite&position=default"
 target="_blank">Zebrafish (danRer11)</a>,
<a href="/cgi-bin/hgTrackUi?db=canFam6&g=refSeqComposite&position=default"
 target="_blank">Dog (canFam6)</a>, and
<a href="/cgi-bin/hgTrackUi?db=danRer11&g=refSeqComposite&position=default"
 target="_blank">Cow (bosTau9)</a> assemblies. This track shows orthologous genes across genomes
and allows you to explore gene symbols and orthology information by hovering over or clicking on
the track item, with links to the corresponding genome browsers:</p>

<div class="text-center">
  <a href="https://genome.ucsc.edu/s/gperez2/hg38_NCBI_ortholog_track" target="_blank">
  <img src="../images/newsArchImages/ortholog_track.png" alt="hg38 session
  visualizing the NCBI Gene Orthologs track" width='50%'></a>
</div>

<p>
You can find the NCBI Orthologs track within the NCBI RefSeq composite track:</p>
<div class="text-center">
        <img class='text-center' width="50%" src="../../images/newsArchImages/ortholog_refSeq.png">
</div>

<p>
This track was created using the latest
<a href="https://ftp.ncbi.nih.gov/gene/DATA/" target="_blank">NCBI files</a> (gene2accession and
gene_orthologs). More information on this track can be found on the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=refSeqComposite#TRACK_HTML" target="_blank">track description</a>
page.</p>
<p>
We would like to thank
<a href="https://www.ncbi.nlm.nih.gov/" target="_blank">NCBI</a> for making this data available. We
would also like to thank Jeltje van Baren, Mark Diekhans, Max Haeussler, and Gerardo Perez for the
creation and release of this track.
</p>

<a name="121024"></a>
<h2>Dec. 10, 2024 &nbsp;&nbsp; DECIPHER Population CNVs for Human (hg19/hg38)</h2>
<p>
We are pleased to introduce the DECIPHER Population CNVs track for both the
<a href="/cgi-bin/hgTrackUi?db=hg19&g=decipherPopulation&position=default" target="_blank">GRCh37/hg19</a>
and
<a href="/cgi-bin/hgTrackUi?db=hg38&g=decipherPopulation&position=default" target="_blank">GRCh38/hg38</a>
human genome assemblies. This track shows common copy-number variants (CNVs) and their population
frequencies.</p>

<p>
The Population CNVs track incorporates an intuitive color-coding system alongside detailed
tooltips for an enhanced user experience:</p>

<ul>
  <li><font color="red">Red:</font> Indicates a loss of genetic material (deletions).</li>
  <li><font color="blue">Blue:</font> Signifies a gain of genetic material (duplications).</li>
  <li><font color="grey">Grey:</font> Marks areas with deletions or duplication (indel).</li>
</ul>

<p>The Population CNVs track's <b>mouseover</b> tooltip provides the following information about the data:</p>
<ul>
  <li><b>Position:</b> Chromosomal range of the CNV.</li>
  <li><b>Type of CNV:</b> Loss, gain, or deletions/duplications.</li>
  <li><b>Frequency:</b> Prevalence of the CNV in the population.</li>
  <li><b>Observations:</b> Times the CNV was noted in the dataset.</li>
  <li><b>Sample Size:</b> Number of samples analyzed.</li>
</ul>
<p>
We would like to thank <a href="https://www.deciphergenomics.org/" target="_blank">DECIPHER</a> for
making the data publicly available. We would also like to thank Yesenia Puga, Gerardo Perez, and
Jairo Navarro for the creation and release of the Population CNVs track on the UCSC Genome Browser.</p>

<a name="110824"></a>
<h2>Nov. 8, 2024 &nbsp;&nbsp; New GENCODE gene tracks: Human V47 (hg19/hg38) - Mouse M36
(mm39)</h2>
<p>
We are happy to announce new GENCODE Gene annotation tracks, which correspond
to <a href="https://www.ensembl.info/2024/10/18/ensembl-113-has-been-released/"
target="_blank">Ensembl 113</a>, for three assemblies: <a target="_blank"
href="../../cgi-bin/hgTrackUi?db=hg19&c=chr21&g=wgEncodeGencodeV47lift37">hg19/GRCh37</a>,
<a target="_blank" href="../../cgi-bin/hgTrackUi?db=hg38&c=chr1&g=wgEncodeGencodeV47">hg38/GRCh38</a>,
and <a target="_blank" href="../../cgi-bin/hgTrackUi?db=mm39&c=chr12&g=wgEncodeGencodeVM36">mm39/GRCm39</a>.
For human, the GENCODE V47 annotations were mapped to hg38/GRCh38 and then back-mapped
to the hg19/GRCh37 assembly. For all three assemblies, the gene sets contain the following
tracks:</p>
<ul>
  <li>Basic - a subset of the <em>Comprehensive set</em>.</li>
  <li>Comprehensive - all GENCODE coding and non-coding transcript annotations, including
      polymorphic pseudogenes. This includes both manual and automatic annotations.</li>
  <li>Pseudogenes - all annotations except polymorphic pseudogenes.</li></ul>
<p>
The hg38 and mm39 assemblies also include the following track that is not available on hg19:
</p>
<ul>
  <li>
      PolyA - polyA signals and sites manually annotated on the genome based on transcribed
      evidence (ESTs and cDNAs) of 3' end of transcripts containing at least 3 A's not matching the
      genome.</li>
</ul>
<p>
Details on each release can be found on the <a target="_blank"
href="https://www.gencodegenes.org/">GENCODE site</a>. This includes <a target="_blank"
href="https://www.gencodegenes.org/human/stats.html">statistics on each release</a>.
<p>We would like to thank the <a target="_blank"
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Mark Diekhans and Gerardo Perez for the development and
release of these tracks.</p>



<a name="110424"></a>
<h2>Nov. 4, 2024 &nbsp;&nbsp; GIAB Problematic Regions tracks for human (hg38 and hs1)</h2>
<p>
We are pleased to announce the release of the
<a href="https://www.nist.gov/programs-projects/genome-bottle" target="_blank"
 >Genome in a Bottle (GIAB)</a> Problematic Regions tracks for the
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=problematicGIAB" target="_blank">hg38</a>
and
<a href="/cgi-bin/hgTrackUi?db=hub_3671779_hs1&position=default&g=hub_3671779_problematic"
target="_blank">hs1</a> human assemblies. The tracks provide stratifications of the genome to
evaluate variant calls in complex regions and it is designed for use with Global Alliance for
Genomic Health (GA4GH) benchmarking tools like
<a href="https://github.com/Illumina/hap.py" target="_blank">hap.py</a>. It includes regions with
low complexity, segmental duplications, functional regions, and difficult-to-sequence areas. The
GIAB Problematic Regions composite track consists of four subtracks:</p>
<ul>
  <li><b>All difficult regions</b> &ndash; Genome In a Bottle: all difficult regions
  <li><b>LowMap+SegDup</b> &ndash; Genome In a Bottle: lowMap+SegDup regions
  <li><b>Not difficult regions</b> &ndash; Genome In a Bottle: not difficult regions 
  <li><b>Not lowMap+SegDup</b> &ndash; Genome In a Bottle: not lowMap+SegDup mapping regions
</ul>
<p>
We would like to thank the Genome in a Bottle Consortium, Telomere-to-Telomere Consortium and NIST
for providing this data. We would also like to thank Megna Chalamala, Jairo Navarro, and Gerardo
Perez for the creation and release of these tracks.</p>

<a name="102324"></a>
<h2>Oct. 23, 2024 &nbsp;&nbsp; New GENCODE &quot;KnownGene&quot; V47 for human (hg38) and VM36 for mouse (mm39)</h2>
<p>
We are pleased to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=knownGene">GENCODE V47 (hg38)</a>
and the
<a href="/cgi-bin/hgTrackUi?db=mm39&position=default&g=knownGene">GENCODE VM36 (mm39)</a> gene tracks.
The GENCODE &quot;KnownGene&quot; V47 and VM36 gene tracks were built using a UCSC
pipeline (KnownGene) and the GENCODE comprehensive gene set to generate high-quality manual
annotations merged with evidence-based automated annotations. The GENCODE &quot;KnownGene&quot;
tracks are our default gene tracks, which have extensive associations to external sources. This
allows for additional metadata on every item as well as external links. The track description pages
contain options for configuring the display, such as showing non-coding genes, splice variants, and
pseudogenes.</p>
<p>
Below is a summary of the contents found in each release. For more details visit the <a target="_blank"
href="https://www.gencodegenes.org/">GENCODE site</a>.</p>
<p>
<table class="stdTbl">
<tr><th COLSPAN=4>GENCODE v47 Release Stats</th></tr>
<tr align=left><th>Genes</th><th>Observed</th><th>Transcripts</th><th>Observed</th></tr>
<tr align=left><td>Protein-coding genes</td><td>19,433</td><td>Protein-coding transcripts</td><td>89,832</td></tr>
<tr align=left><td>Long non-coding RNA genes</td><td>35,934</td><td><font size="-1">- full length protein-coding</font></td><td>64,988</td></tr>
<tr align=left><td>Small non-coding RNA genes</td><td>7,565</td><td><font size="-1">- partial length protein-coding</font></td><td>24,844</td></tr>
<tr align=left><td>Pseudogenes</td><td>14,703</td><td>Nonsense mediated decay transcripts</td><td>21,873</td></tr>
<tr align=left><td>Immunoglobulin/T-cell receptor gene segments</td><td>648</td><td>Long non-coding RNA loci transcripts</td><td>191,106</td></tr>
<tr align=left><td>Total No of distinct translations</td><td>65,809</td><td>Genes that have more than one distinct translations</td><td>13,646</td></tr>
</table><BR>
</p>
<p>
<table class="stdTbl">
<tr><th COLSPAN=4>GENCODE VM36 Release Stats</th></tr>
<tr align=left><th>Genes</th><th>Observed</th><th>Transcripts</th><th>Observed</th></tr>
<tr align=left><td>Protein-coding genes</td><td>21,470</td><td>Protein-coding transcripts</td><td>58,547</td></tr>
<tr align=left><td>Long non-coding RNA genes</td><td>36,172</td><td><font size="-1">- full length protein-coding</font></td><td>44,949</td></tr>
<tr align=left><td>Small non-coding RNA genes</td><td>6,105</td><td><font size="-1">- partial length protein-coding</font></td><td>13,598</td></tr>
<tr align=left><td>Pseudogenes</td><td>13,769</td><td>Nonsense mediated decay transcripts</td><td>7,247</td></tr>
<tr align=left><td>Immunoglobulin/T-cell receptor gene segments</td><td>701</td><td>Long non-coding RNA loci transcripts</td><td>156,135</td></tr>
<tr align=left><td>Total No of distinct translations</td><td>44,888</td><td>Genes that have more than one distinct translations</td><td>10,840</td></tr>
</table><BR>
</p>
<p>
We would like to thank the <a target="_blank"
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Jonathan Casper and Gerardo Perez for the development and
release of these tracks.</p>

<a name="100924"></a>
<h2>Oct. 9, 2024 &nbsp;&nbsp; CADD v1.7 and ClinGen CSpec tracks for hg19 and hg38</h2>
<p>
We are excited to announce two new tracks available for human assemblies hg38/GRCh38 and hg19/GRCh37.</p>

<h3>Clingen CSpec Variant Interpretation VCEP Specifications</h3>
<p>
The <a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=clinGenComp">ClinGen VCEP Specifications
track</a> identifies loci that have <a target="_blank"
href="https://cspec.genome.network/cspec/ui/svi/">ClinGen criteria Specification (CSpec)</a>
information. This is used and
applied by ClinGen Variant Curation Expert Panels (VCEPs) and biocurators in the classification 
of variants.</p>

<p>
This track can be found in the existing ClinGen composite track. The data in this track 
updates weekly whenever new data is found in the ClinGen CSpec registry.</p>

<h3>CADD 1.7 Scores for hg19 and hg38</h3>
<p>
The
<a href="../cgi-bin/hgTrackUi?db=hg38&position=defaul&g=caddSuper1_7" target="_blank">Combined Annotation Dependent Depletion
(CADD)</a> v1.7 tracks show scores for deleteriousness of single nucleotide variants as well as
insertion/deletion variants in the GRCh37/hg19 and GRCh38/hg38 genomes. CADD v1.7 integrates
advanced annotation features, including state-of-the-art protein language model scores
(Meta ESM-1v), regulatory variant effect predictions derived from CNN models, and updated sequence
conservation scores from the Zoonomia project.</p> 

<p>
CADD scores are freely available for all non-commercial applications from the
<a href="https://cadd.gs.washington.edu/download" target="_blank">CADD website</a>. For commercial
applications, see the
<a href="https://cadd.gs.washington.edu/contact" target="_blank">license instructions</a> there.</p>

<p>
We would like to thank the CADD development team for providing precomputed data as simple
tab-separated files. We would also like to thank Jeltje van Baren, Lou Nassar, and Gerardo Perez for
the creation and release of these tracks.</p>

<a name="100124"></a>
<h2>Oct. 1, 2024 &nbsp;&nbsp; New interactive variant interpretation tutorial</h2>
<p>
We have a new <a href="/cgi-bin/hgTracks?db=hg38&startClinical=true"
target="_blank">clinical tutorial</a> showcasing resources that could be useful in variant
interpretation. The tutorial is written to educate clinical geneticists with any level of browser
experience. It covers topics such as searching for variants and data, recommended track sets,
and how to save and share browser configurations.</p>

<p>
<div class="text-center">
  <a href="/cgi-bin/hgTracks?db=hg38&startClinical=true" target="_blank">
    <img src="/images/clinicalTutorial.png" width='30%'>
  </a>
</div>
</p>

<p>
We would like to thank Jairo Navarro, Chris Lee, Anna Benet-Pages,
Maximilian Haeussler and Lou Nassar for their work in creating this tutorial.</p>

<a name="093024"></a>
<h2>Sep. 30, 2024  &nbsp;&nbsp; New gnomAD v4.1 tracks for hg38</h2>
<p>
We are excited to release six new
<a target=_blank href="/cgi-bin/hgTrackUi?db=hg38&g=gnomadVariants">
Genome Aggregation Database (gnomAD) v4.1 tracks</a> for human assembly hg38/GRCh38. The new tracks
are found in the gnomAD superTrack:</p>
<ul>
  <li><b>gnomAD v4.1 (composite track)</b>:</li>
   <ul>
    <li><b>Genome variants</b>: Variants from 76,215 genomes</li>
    <li><b>Exome Variants</b>: Variants from 730,947 exomes</li>
   </ul>
  <li><b>gnomAD Constraint Metrics (composite track)</b>:
   <ul>
    <li><b>Transcript LoF v4.1</b>: Predicted Loss of Function Constraint Metrics By Transcript (LOEUF and pLI) v4.1</li>
    <li><b>Transcript Missense v4.1</b>: Predicted Missense Constraint Metrics By Transcript (Z-scores) v4.1</li>
   </ul>
  <li><b>gnomAD Rare CNV Variants</b>: Rare CNV variants (&lt;1% overall site frequency)</li>
  <li><b>gnomAD Structural Variants</b>: Structural Variants v4.1</li>
</ul>
<p>
<b>gnomAD v4.1</b>: The gnomAD v4.1 data release replaces the v4 Pre-Release track and fixes the allele
number <a target="_blank"
href="https://gnomad.broadinstitute.org/news/2024-04-gnomad-v4-1/">
issue</a>. The data contains two subtracks from a total of 807,162 individuals: one for genome
variant calls and one for exome variants.</p>
<p>
<b>gnomAD Constraint Metrics</b>: Contains metrics of pathogenicity per-gene as predicted for
gnomAD v4.1 and identifies genes subject to strong selection against various classes of mutation.
The release of the gnomAD v4.1 constraint metrics consists of two new subtracks in the gnomAD
Constraint Metrics composite: Transcript LoF v4.1 and Transcript Missense v4.1</p>
<p>
<b>gnomAD Rare CNV Variants</b>: Displays rare autosomal coding copy number variants (CNVs) with an
overall site frequency of less than 1%. These variants were identified from exome sequencing (ES)
data of 464,297 individuals.</p>
<p>
<b>gnomAD Structural Variants</b>: Shows structural variants calls (&gt;=50 nucleotides) from
63,046 unrelated genomes. It mostly (but not entirely) overlaps with the genome set used for the
gnomAD short variant release.</p>

<p>
We would like to thank the <a href="https://gnomad.broadinstitute.org/about" target="_blank">Genome
Aggregation Database Consortium</a> for making these data available. We would also like to thank
Chris Lee, Ana Benet-Pag&#232;s, Luis Nassar, and Gerardo Perez for the creation and release of
these tracks.</p>




<a name="092424"></a>
<h2>Sept. 24, 2024  &nbsp;&nbsp; Nuclear mitochondrial DNA segments (NuMTs) for hg38</h2>
<p>
We are happy to announce the release of a new track,
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=nuMtSeq">NuMTs Sequence</a>,
for human assembly (GRCh38/hg38). This track is a collection of Nuclear mitochondrial DNA segments,
provided in BED format with the BED score representing the alignment confidence. The BED score is
calculated by -10log10(E-value) and is reflected in the grayscale value. Scores &gt;=100 are
colored black. It is important to note that when a NuMT is a merged result, the score is taken as
the highest score among all results.</p>

<p>
For more detailed information on the methods used for detecting NuMTs, please visit the
following webpage:<br>
<a href="https://github.com/Koumokuyou/NUMTs">https://github.com/Koumokuyou/NUMTs</a></p>

<p>
We would like to thank Martin Frith and Huang Muyao for generating and building the original
NuMTs track hub. We would also like to thank Megna Chalamala and Jairo Navarro for wrangling and
building the internal UCSC track for hg38.</p>

<a name="091024"></a>
<h2>Sept. 10, 2024  &nbsp;&nbsp; Two new public hubs: ENIGMA VCEP and BRCAExchange</h2>
<p>
We are excited to announce the addition of two new hubs to the
<a href="/cgi-bin/hgHubConnect?#publicHubs" target="_blank">Public Hubs</a> page.
</p>
<p>
The <a href="/cgi-bin/hgTracks?genome=hg38&position=default&hubUrl=https://brcaexchange.org/trackhubs/hub.txt"
target="blank"><b>BRCAExchange</b></a> track hub presents the variants from
<a href ="https://brcaexchange.org/" target="_blank">BRCA Exchange</a> in the
context of the human genome, offering further details on each variant in the variant details pages.
This set of variants is obtained by integrating data from
genomic variation repositories including ClinVar, LOVD and gnomAD, and annotating these variants
according to methods selected and developed in close collaboration with the ENIGMA Consortium, the
international expert consortium on research on HBOC variants.
<ul>
    Data on BRCA variants is available for the hg19 and hg38 genomes assemblies and the tracks
    available are:
    <ul>
        <li><b>BRCA Variants</b> - <br>
        Shows single nucleotide variants (SNVs) and small insertion/deletion
        variants of &lt; 50 nucleotides (indels)</li>
        <li><b>BRCA Structural Variants</b> - <br>
        Shows variants &lt;= 50 nucleotides</li>
        <li><b>ACMG Popfreq Variants</b> - <br>
        Shows the provisionally-assigned ACMG evidence code for the
        Population Frequency group, for variants of &lt; 50 nucleotides</li>
        <li><b>ACMG Popfreq SVs</b> - <br>
        Shows the provisionally-assigned ACMG evidence code for the Population
        Frequency group, for variants of &lt;= 50 nucleotides</li>
    </ul>
</ul>

<p>
The <a href="/cgi-bin/hgTracks?genome=hg38&position=default&hubUrl=https://hgdownload.gi.ucsc.edu/hubs/enigma/hub.txt"
target="blank"><b>ENIGMA BRCA1/BRCA2 specs 1.1.0</b></a> track hub contains data from the
<a href="https://enigmaconsortium.org/" target="_blank">ClinGen ENIGMA BRCA1 and BRCA2 Expert Panel
Specifications</a> to the ACMG/AMP Variant Interpretation
Guidelines for BRCA1/BRCA1 Version 1.1.0. The ENIGMA VCEP has adapted the ACMG-AMP codes for the
BRCA1 and BRCA2 genes. Before using the data, always verify that the ClinGen Criteria Specification
(CSpec) version numbers here match the latest version on the
<a target="_blank" href="https://cspec.genome.network/cspec/ui/svi/affiliation/50087">
CSpec Registry</a>.
</p>
<ul>
    Data is available for the hg19 and hg38 genomes assemblies and the tracks
    available are:
    <ul>
        <li><b>BRCA1/BRCA2 protein domains 1.1.0</b> -<br>
        Shows the (potentially) clinically important functional domains for 
        the genes BRCA1 and BRCA2 as defined by ENIGMA.
        </li>
        <li><b>BRCA1/BRCA2 exon weights 1.1.0</b> - <br>
        This track shows exon-specific weights for PM5_PTC ACMG code with evidence strong used 
        for application of this code for novel protein termination codon (PTC) 
        variants in an exon where a different proven pathogenic PTC variant 
        has been seen before.
        </li>
        <li><b>BRCA1/BRCA2 functional assays 1.1.0</b> - <br>
        Summary of BRCA1 and BRCA2 functional assay results reviewed for 
        application of PS3 and BS3 ACMG codes.
        </li>
        <li><b>BRCA1/BRCA2 splicing 1.1.0</b> - <br>
        Summary of ACMG codes applicable for variants considered against 
        the BRCA1 and BRCA2 PVS1 decision trees. Includes PVS1 and PM5 
        codes recommended for initiation, nonsense/frameshift, deletion, 
        duplication, and splice site (donor/acceptor &plusmn;1,2) variant&mdash;
        organized by exon.
        </li>
        <li><b>BRCA1/BRCA2 likelihood for PP4 and BP5</b> - <br>
        Summary of BRCA1 and BRCA2 multifactorial likelihood analysis 
        scores (displayed as Combined LR score) for ACMG codes PP4 and BP5. 
        </li>
    </ul>
</ul>
<p>
We would like to give thanks to Melissa Cline for the creation of the BRCA Exchange hub.
We would also like to thank Ana Benet Pages and Andreas Laner for making the ENIGMA VCEP hub
available.</p>

<a name="082924"></a>
<h2>Aug. 29, 2024  &nbsp;&nbsp; Download track data in view</h2>
<p>
We are pleased to announce a new feature that allows users to download
all visible data in the current region directly from our tracks display. This facilitates
reproducibility when writing reports or publications as data can update
and change over time.</p>
<p>
This feature can be found in the blue bar menu by going to <b>Downloads</b> &gt; 
<b>Download Current Track Data</b>. The resulting pop-up dialogue box can configure
the exact tracks to download from all visible tracks, as well as the file name and
the output format (JSON, csv, tsv). While most track types are supported, 
please <a href="../contacts.html">contact us</a> if you 
encounter an unsupported format you would like to export.</p>

</p>
<div class="text-center">
        <img class='text-center' src="../images/downloadCurrentRegion.gif" width='60%' alt="Download
track data in view.">
</div>

<p>
We would like to thank Chris Lee, Jairo Navarro, and Lou Nassar for their
work on this release.</p>

<a name="082124"></a>
<h2>Aug. 21, 2024  &nbsp;&nbsp; New groups feature for track hubs</h2>
<p>
We are excited to announce a new grouping feature for track hubs, which allows the structured
organization of tracks into distinct groups. This feature can be applied to a UCSC genome, a
GenArk assembly, or an assembly hub. These track hub groups are kept separate from other track
hubs and the native UCSC Genome Browser track groups, allowing for greater organizational
flexibility. For instance, you can add a "genes" group without causing conflicts or confusion.
You can define groups with names like
&quot;Category 1&quot;, &quot;Category 2&quot;, and &quot;Category 3&quot;.
<div class="text-center">
        <img class='text-center' width="80%" src="../../images/groupsEx.png">
</div>
<p>
For more information, see the
<a href="/goldenPath/help/hgTrackHubHelp.html#Group"
target="_blank">Adding Groups to a Track hub</a> section of the Track Hubs help page.</p>
<p>
We would like to thank Brian Joseph Raney, Mark Diekhans, Luis Nassar, and Gerardo Perez for their
work on this release.</p>



<a name="081424"></a>
<h2>Aug. 14, 2024  &nbsp;&nbsp; New pop-up dialogue box for item details</h2>
<p>
We are excited to introduce the new Item Details feature which simplifies the user experience by
displaying track item details in a pop-up dialogue box. This feature allows the information to be
viewed without the need to navigate away from the current page:</p>
<div class="text-center">
        <img class='text-center' width="60%" src="../../images/ItemDetails_knownGene.png">
</div>
<p>
When a track item is clicked in either pack or full mode, the <b>Item Details</b> dialogue box will
appear. This box can be resized by clicking and dragging the button located in its lower-right
corner. For users who prefer viewing item details on a separate page, clicking the new window
button in the upper-right corner will load the information in a new tab. The dialogue box can be
closed by clicking the <b>x</b> button, the <button type="button">Close</button> button, pressing
the Escape key, or by clicking outside the box.</p>
<p>
If desired, this feature can be disabled by unchecking the &quot;Enable pop-up when clicking items
option&quot; on the Configure page. You can access the Configure page by selecting Configure under
Genome Browser in the blue bar menu, by clicking the <button type="button">Configure</button>
button below the browser graphic, or by using the keyboard shortcut &quot;c f&quot;.</p>
<div class="text-center">
        <img class='text-center' width="30%" src="../../images/configure_ItemDetails.png">
</div>
<p>
We would like to thank Christopher Lee, Lou Nassar, Jairo Navarro, and Gerardo Perez for their work
on this release.</p>

<a name="081224"></a>
<h2>Aug. 12, 2024 &nbsp;&nbsp; Illumina SpliceAI tracks for human (hg38 and hg19)</h2>
<p>
We are pleased to announce the release of the <b>Illumina SpliceAI</b> tracks for the
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=spliceAI" target="_blank">hg38</a> and
<a href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=spliceAI" target="_blank">hg19</a>
human assemblies. SpliceAI is an
<a href="https://github.com/Illumina/SpliceAI" target="_blank">open-source</a> deep
learning splicing prediction algorithm that can predict splicing alterations caused by DNA
variations. Such variants may activate nearby cryptic splice sites, leading to abnormal transcript
isoforms. SpliceAI was developed at Illumina; a <a href="https://spliceailookup.broadinstitute.org"
target="_blank">lookup tool</a> is provided by the Broad institute.</p>
<p>
<b>Important</b>: The SpliceAI data on the UCSC Genome Browser is directly from
Illumina. However, since SpliceAI refers to the algorithm, and not the
computed dataset, the data on the Broad server or other sources may have some differences between
them. The SpliceAI scores are made available by Illumina only for academic or not-for-profit
research only. By accessing the SpliceAI data, you acknowledge and agree that you may only use this
data for your own personal academic or not-for-profit research only, and not for any other purposes.
You may not use this data for any for-profit, clinical, or other commercial purpose without
obtaining a commercial license from Illumina, Inc.
</p>
<p>
We would like to thank Illumina for making this data available on the UCSC Genome Browser. We would
also like to thank Luis Nassar and Jairo Navarro for the creation and release of the tracks.
</p>

<a name="073124"></a>
<h2>Jul. 31, 2024 &nbsp;&nbsp; GENCODE &quot;KnownGene&quot; v45lift37 release for human (hg19)</h2>
<p>
We are excited to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=knownGene" target="_blank">GENCODE
&quot;KnownGene&quot; v45lift37</a> gene track for hg19. With this release, the previous 2013 UCSC
Genes track will be frozen and made available in the
<a href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=knownGeneArchive"
target="_blank">GENCODE/UCSC Genes Archive</a> superTrack for reproducibility.
As new GENCODE tracks are made available, previous versions will also be available in the archive.
Beginning with this update, the &quot;KnownGene&quot; track will use GENCODE v45 gene models
lifted to hg19, which replaces the old UCSC transcript IDs with the official GENCODE IDs.
</p>
<p>
The following is an example of some GENCODE IDs that will replace the UCSC IDs in the update:
</p>
<pre>
oldId      newId
uc003qfo.3 ENST00000341911.10_8
uc003jsk.2 ENST00000462279.5_3
uc003umk.1 ENST00000318238.9_6
uc003gzi.3 ENST00000682860.1_2
uc011dpu.2 ENST00000375023.3_6
uc021raj.2 ENST00000258149.11_6
uc002fxp.3 ENST00000341657.9_12
uc010xhp.1 ENST00000429344.7_6
uc003zze.3 ENST00000242285.11_9</pre>
<p>
For each transcript ID, the _# portion is part of the
<a href="https://github.com/diekhans/gencode-backmap?tab=readme-ov-file#identification"
target="_blank">official hg19 backmap ID</a>, so they are not confused with the gene/transcript they
are derived from in hg38. Between hg38 and hg19, the two IDs are not always in the same sequence and
may not be a one-to-one mapping.
</p>
<p>
The GENCODE
&quot;KnownGene&quot; V45lift37 gene track is built using a UCSC pipeline (KnownGene) and the
GENCODE comprehensive gene set to generate high-quality manual annotations merged with
evidence-based automated annotations. The GENCODE &quot;KnownGene&quot; tracks are our default gene
tracks, which have extensive associations to external sources. This allows for additional metadata
on every item as well as external links. The track description pages contain options for configuring
the display, such as showing non-coding genes, splice variants, and pseudogenes.
</p>
<p>
Below is a summary of the contents found in the GENCODE v45 release.
For more details visit the <a target="_blank"
href="https://www.gencodegenes.org/human/stats_45.html">GENCODE site</a>.
</p>
<table class="stdTbl">
<tbody><tr><th colspan="4">GENCODE v45 Release Stats</th></tr>
<tr align="left"><th>Genes</th><th>Observed</th><th>Transcripts</th><th>Observed</th></tr>
<tr align="left"><td>Protein-coding genes</td><td>19,395</td><td>Protein-coding transcripts</td><td>89,110</td></tr>
<tr align="left"><td>Long non-coding RNA genes</td><td>20,424</td><td><font size="-1">- full length protein-coding</font></td><td>64,028</td></tr>
<tr align="left"><td>Small non-coding RNA genes</td><td>7,565</td><td><font size="-1">- partial length protein-coding</font></td><td>25,082</td></tr>
<tr align="left"><td>Pseudogenes</td><td>14,719</td><td>Nonsense mediated decay transcripts</td><td>21,427</td></tr>
<tr align="left"><td>Immunoglobulin/T-cell receptor gene segments</td><td>648</td><td>Long non-coding RNA loci transcripts</td><td>59,719</td></tr>
<tr align="left"><td>Total No of distinct translations</td><td>65,357</td><td>Genes that have more than one distinct translations</td><td>13,600</td></tr>
</tbody></table>
</p>
<p>
We would like to thank the <a target="_blank"
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Brian Raney, Mark Diekhans, and Jairo Navarro for the
development and release of these tracks.
</p>

<a name="072524"></a>
<h2>Jul. 25, 2024 &nbsp;&nbsp; EVA SNP release 6 for 37 assemblies</h2>
<p>
We are pleased to announce the release of the EVA SNP release 6 track for 37 assemblies. These
tracks contain mappings of single nucleotide variants and small insertions and deletions (indels)
&mdash; collectively Simple Nucleotide Variants (SNVs) &mdash; from the European Variation Archive
(<a href="https://www.ebi.ac.uk/eva/" target="_blank">EVA</a>) Release 6. The full list of
assemblies that contain the EVA SNP release 6 track is below:</p>
<p>
<div class="container">
  <div class="row">
    <div class="col-sm-4">
    <ul>
      <li>A. gambiae <a href="../cgi-bin/hgTrackUi?db=anoGam3&g=evaSnp6"
      target="_blank">(anoGam3)</a></li>
      <li>C. intestinalis <a href="../cgi-bin/hgTrackUi?db=ci3&g=evaSnp6"
      target="_blank">(ci3)</a></li>
      <li>Bison <a href="../cgi-bin/hgTrackUi?db=bisBis1&g=evaSnp6"
      target="_blank">(ci3)</a></li>
      <li>Cat <a href="../cgi-bin/hgTrackUi?db=felCat8&g=evaSnp6"
        target="_blank">(felCat8)</a>,
        <a href="../cgi-bin/hgTrackUi?db=felCat9&g=evaSnp6"
        target="_blank">(felCat9)</a></li>
      <li>Chicken <a href="../cgi-bin/hgTrackUi?db=galGal4&g=evaSnp6"
        target="_blank">(galGal4)</a>,
        <a href="../cgi-bin/hgTrackUi?db=galGal5&g=evaSnp6"
        target="_blank">(galGal5)</a>,
        <a href="../cgi-bin/hgTrackUi?db=galGal6&g=evaSnp6"
        target="_blank">(galGal6)</a></li>
      <li>Chimp <a href="../cgi-bin/hgTrackUi?db=panTro4&g=evaSnp6"
      target="_blank">(panTro4)</a>,
      <a href="../cgi-bin/hgTrackUi?db=panTro5&g=evaSnp6"
      target="_blank">(panTro5)</a></li></li>
      <li>Cow <a href="../cgi-bin/hgTrackUi?db=bosTau7&g=evaSnp6"
        target="_blank">(bosTau7)</a>
      <li>Crab-eating macaque <a href="../cgi-bin/hgTrackUi?db=macFas5&g=evaSnp6"
      target="_blank">(macFas5)</a></li>
      <li>D. melanogaster <a href="../cgi-bin/hgTrackUi?db=dm6&g=evaSnp6"
      target="_blank">(dm6)</a></li>
    </ul></div>
    <div class="col-sm-4">
      <ul>
      <li>Dog <a href="../cgi-bin/hgTrackUi?db=canFam3&g=evaSnp6"
      target="_blank">(canFam3)</a></li>
      <li>Gorilla <a href="../cgi-bin/hgTrackUi?db=gorGor3&g=evaSnp6"
        target="_blank">(gorGor3)</a>,
        <a href="../cgi-bin/hgTrackUi?db=gorGor4&g=evaSnp6"
        target="_blank">(gorGor4)</a></li>
      <li>Green monkey <a href="../cgi-bin/hgTrackUi?db=chlSab2&g=evaSnp6"
      target="_blank">(chlSab2)</a></li>
      <li>Horse <a href="../cgi-bin/hgTrackUi?db=equCab3&g=evaSnp6"
      target="_blank">(equCab3)</a></li>
      <li>Marmoset <a href="../cgi-bin/hgTrackUi?db=calJac3&g=evaSnp6"
      target="_blank">(calJac3)</a></li>
      <li>Mouse <a href="../cgi-bin/hgTrackUi?db=mm10&g=evaSnp6"
      target="_blank">(mm10)</a>,
      <a href="../cgi-bin/hgTrackUi?db=mm39&g=evaSnp6"
      target="_blank">(mm39)</a></li>
      <li>Nile tilapia <a href="../cgi-bin/hgTrackUi?db=oreNil2&g=evaSnp6"
      target="_blank">(oreNil2)</a></li>
      <li>Orangutan <a href="../cgi-bin/hgTrackUi?db=ponAbe3&g=evaSnp6"
      target="_blank">(ponAbe3)</a></li>
    </ul></div>
    <div class="col-sm-4">
      <ul>
      <li>Pig <a href="../cgi-bin/hgTrackUi?db=susScr11&g=evaSnp6"
        target="_blank">(susScr11)</a>,
        <a href="../cgi-bin/hgTrackUi?db=susScr3&g=evaSnp6"
        target="_blank">(susScr3)</a></li>
      <li>Rat <a href="../cgi-bin/hgTrackUi?db=rn5&g=evaSnp6"
        target="_blank">(rn5)</a>,
        <a href="../cgi-bin/hgTrackUi?db=rn7&g=evaSnp6"
        target="_blank">(rn7)</a></li>
      <li>Rhesus <a href="../cgi-bin/hgTrackUi?db=rheMac10&g=evaSnp6"
        target="_blank">(rheMac10)</a>,
        <a href="../cgi-bin/hgTrackUi?db=rheMac8&g=evaSnp6"
        target="_blank">(rheMac8)</a></li>
      <li>S. cerevisiae <a href="../cgi-bin/hgTrackUi?db=sacCer3&g=evaSnp6"
      target="_blank">(ponAbe3)</a></li>
      <li>Sheep <a href="../cgi-bin/hgTrackUi?db=oviAri3&g=evaSnp6"
        target="_blank">(oviAri3)</a>,
        <a href="../cgi-bin/hgTrackUi?db=oviAri4&g=evaSnp6"
        target="_blank">(oviAri4)</a></li>
      <li>Turkey <a href="../cgi-bin/hgTrackUi?db=melGal5&g=evaSnp6"
      target="_blank">(melGal5)</a></li>
      <li>Zebra finch <a href="../cgi-bin/hgTrackUi?db=taeGut2&g=evaSnp6"
      target="_blank">(taeGut2)</a></li>
      <li>Zebrafish <a href="../cgi-bin/hgTrackUi?db=danRer7&g=evaSnp6"
        target="_blank">(danRer7)</a>,
        <a href="../cgi-bin/hgTrackUi?db=danRer10&g=evaSnp6"
        target="_blank">(danRer10)</a>,
        <a href="../cgi-bin/hgTrackUi?db=danRer11&g=evaSnp6"
        target="_blank">(danRer11)</a></li>
    </ul></div>
  </div>
</div></p>

<p>
These variants are classified by EVA into one of the following <a target="_blank"
href="http://www.sequenceontology.org/">sequence ontology</a> terms:
</p>
<ul>
  <li> <b>substitution</b> &mdash;
       A single nucleotide in the reference is replaced by another, alternate allele
  <li> <b>deletion</b> &mdash;
       One or more nucleotides are deleted.  The representation in the database is to
       display one additional nucleotide in both the Reference field (Ref) and the
       Alternate Allele field (Alt).  E.g. a variant that is a deletion of an A
       maybe be represented as Ref = GA and Alt = G.
  <li> <b>insertion</b> &mdash;
       One or more nucleotides are inserted.  The representation in the database is to
       display one additional nucleotide in both the Reference field (Ref) and the
       Alternate Allele field (Alt).  E.g. a variant that is an insertion of a T may
       be represented as Ref = G and Alt = GT
  <li> <b>delins</b> &mdash;
       Similar to a tandem repeat, in that the runs of Ref and Alt Alleles are of
       different length, except that there is more than one type of nucleotide,
       e.g., Ref = CCAAAAACAAAAACA, Alt = ACAAAAAC.
  <li> <b>multipleNucleotideVariant</b> &mdash;
       More than one nucleotide is substituted by an equal number of different
       nucleotides, e.g.,  Ref = AA, Alt = GC.
  <li> <b>sequence alteration</b> &mdash;
       A parent term meant to signify a deviation from another sequence. Can be
       assigned to variants that have not been characterized yet.
</ul>
<p>
The variants have also been annotated with our <a href="/cgi-bin/hgVai" target="_blank">Variant
Annotation Integrator</a> tool with functional classes such as synonymous variant, 
missense variant, stop gained, etc. For additional details on the track colors, 
as well as the filters and metadata on each variant, see the track description page.</p>

<p>
We would like to thank the <a href="https://www.ebi.ac.uk/eva/" target="_blank">European Variation
Archive</a> for making these data publically available. We would also like to thank Luis Nassar, Chris Lee, 
and Angie Hinrichs for the creation and release of these tracks.
</p>

<a name="071224"></a>
<h2>Jul. 12, 2024 &nbsp;&nbsp; First update to hg19's UCSC Genes track since 2013</h2>
<p>
The UCSC Genome Browser is getting ready to update hg19's UCSC Genes dataset for the first time
since 2013. In this update, the UCSC Genes track will now use GENCODE v45 gene models lifted to hg19
and replace the old UCSC transcript IDs with the official GENCODE IDs.
</p>
<p>
The anticipated release date for this update is July 31, 2024.</p>

<p>
As an example of what to expect, here are some GENCODE IDs that will replace the UCSC IDs in the
update:</p>

<pre>
oldId      newId
uc003qfo.3 ENST00000341911.10_8
uc003jsk.2 ENST00000462279.5_3
uc003umk.1 ENST00000318238.9_6
uc003gzi.3 ENST00000682860.1_2
uc011dpu.2 ENST00000375023.3_6
uc021raj.2 ENST00000258149.11_6
uc002fxp.3 ENST00000341657.9_12
uc010xhp.1 ENST00000429344.7_6
uc003zze.3 ENST00000242285.11_9</pre>

<p>
For each transcript ID, the _# portion is part of the
<a href="https://github.com/diekhans/gencode-backmap?tab=readme-ov-file#identification"
target="_blank">official hg19 backmap ID</a>, so they are not confused with the gene/transcript they
are derived from in hg38. Between hg38 and hg19, the two IDs are not always in the same sequence and
may not be a one-to-one mapping.
</p>
<p>
With this update, we will also be freezing the previous 2013 knownGene release in a separate
archive track. Once new GENCODE versions are available, the previous track will then be added to the
archive for future reproducibility. More information about the knownGene archive will be available
once the update is released.</p>

<a name="060724"></a>
<h2>June 7, 2024 &nbsp;&nbsp; New GENCODE gene tracks: Human V46 (hg19/hg38) - Mouse M35
(mm39)</h2>
<p>
We are pleased to announce new GENCODE Gene annotation tracks, which correspond
to <a href="https://www.ensembl.info/2024/05/13/ensembl-112-has-been-released/"
target="_blank">Ensembl 112</a>, for three assemblies: <a target="_blank"
href="../../cgi-bin/hgTrackUi?db=hg19&c=chr21&g=wgEncodeGencodeV46lift37">hg19/GRCh37</a>,
<a target="_blank" href="../../cgi-bin/hgTrackUi?db=hg38&c=chr1&g=wgEncodeGencodeV46">hg38/GRCh38</a>,
and <a target="_blank" href="../../cgi-bin/hgTrackUi?db=mm39&c=chr12&g=wgEncodeGencodeVM35">mm39/GRCm39</a>.
For human, the GENCODE V46 annotations were mapped to hg38/GRCh38 and then back-mapped
to the hg19/GRCh37 assembly. For all three assemblies, the gene sets contain the following
tracks:</p>
<ul>
  <li>Basic - a subset of the <em>Comprehensive set</em>.</li>
  <li>Comprehensive - all GENCODE coding and non-coding transcript annotations, including
      polymorphic pseudogenes. This includes both manual and automatic annotations.</li>
  <li>Pseudogenes - all annotations except polymorphic pseudogenes.</li></ul>
<p>
The hg38 and mm39 assemblies also include the following tracks that are not available on hg19:
</p>
<ul>
  <li>2-way Pseudogenes - pseudogenes predicted by both the Yale Pseudopipe and UCSC
      Retrofinder pipelines.</li>
  <li>PolyA - polyA signals and sites manually annotated on the genome based on transcribed
      evidence (ESTs and cDNAs) of 3' end of transcripts containing at least 3 A's not matching the
      genome.</li></ul>
<p>
Details on each release can be found on the <a target="_blank"
href="https://www.gencodegenes.org/">GENCODE site</a>. This includes <a target="_blank"
href="https://www.gencodegenes.org/human/stats.html">statistics on each release</a>.
<p>We would like to thank the <a target="_blank"
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Mark Diekhans and Lou Nassar for the development and
release of these tracks.</p>

<a name="052224"></a>
<h2>May. 22, 2024 &nbsp;&nbsp; New GENCODE &quot;KnownGene&quot; V46 for human (hg38) and VM35 for mouse (mm39)</h2>
<p>
We are pleased to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=knownGene&c=chrX">GENCODE V46 (hg38)</a>
and the
<a href="/cgi-bin/hgTrackUi?db=mm39&g=knownGene&c=chrX">GENCODE VM35 (mm39)</a> gene tracks.
The GENCODE &quot;KnownGene&quot; V46 and VM35 gene tracks were built using a UCSC
pipeline (KnownGene) and the GENCODE comprehensive gene set to generate high-quality manual
annotations merged with evidence-based automated annotations. The GENCODE &quot;KnownGene&quot;
tracks are our default gene tracks, which have extensive associations to external sources. This
allows for additional metadata on every item as well as external links. The track description pages
contain options for configuring the display, such as showing non-coding genes, splice variants, and
pseudogenes.</p>

<p>
Below is a summary of the contents found in each release. For more details visit the <a target="_blank"
href="https://www.gencodegenes.org/">GENCODE site</a>.</p>
<p>
<table class="stdTbl">
<tr><th COLSPAN=4>GENCODE v46 Release Stats</th></tr>
<tr align=left><th>Genes</th><th>Observed</th><th>Transcripts</th><th>Observed</th></tr>
<tr align=left><td>Protein-coding genes</td><td>19,411</td><td>Protein-coding transcripts</td><td>89,581</td></tr>
<tr align=left><td>Long non-coding RNA genes</td><td>20,310</td><td><font size="-1">- full length protein-coding</font></td><td>64,695</td></tr>
<tr align=left><td>Small non-coding RNA genes</td><td>7,565</td><td><font size="-1">- partial length protein-coding</font></td><td>24,886</td></tr>
<tr align=left><td>Pseudogenes</td><td>14,716</td><td>Nonsense mediated decay transcripts</td><td>21,774</td></tr>
<tr align=left><td>Immunoglobulin/T-cell receptor gene segments</td><td>648</td><td>Long non-coding RNA loci transcripts</td><td>59,927</td></tr>
<tr align=left><td>Total No of distinct translations</td><td>65,650</td><td>Genes that have more than one distinct translations</td><td>13,620</td></tr>
</table><BR>
</p>
<p>
<table class="stdTbl">
<tr><th COLSPAN=4>GENCODE VM35 Release Stats</th></tr>
<tr align=left><th>Genes</th><th>Observed</th><th>Transcripts</th><th>Observed</th></tr>
<tr align=left><td>Protein-coding genes</td><td>21,423</td><td>Protein-coding transcripts</td><td>58,457</td></tr>
<tr align=left><td>Long non-coding RNA genes</td><td>15,126</td><td><font size="-1">- full length protein-coding</font></td><td>44,851</td></tr>
<tr align=left><td>Small non-coding RNA genes</td><td>6,105</td><td><font size="-1">- partial length protein-coding</font></td><td>13,606</td></tr>
<tr align=left><td>Pseudogenes</td><td>13,756</td><td>Nonsense mediated decay transcripts</td><td>7,243</td></tr>
<tr align=left><td>Immunoglobulin/T-cell receptor gene segments</td><td>701</td><td>Long non-coding RNA loci transcripts</td><td>27,096</td></tr>
<tr align=left><td>Total No of distinct translations</td><td>44,819</td><td>Genes that have more than one distinct translations</td><td>10,833</td></tr>
</table><BR>
</p>
<p>
We would like to thank the <a target="_blank"
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Jonathan Casper and Gerardo Perez for the development and
release of these tracks.</p>


<a name="042524"></a>
<h2>Apr. 25, 2024 &nbsp;&nbsp; New AbSplice Prediction Scores track for hg19</h2>
<p>
We are excited to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=abSplice" target="_blank">AbSplice scores track</a>
that was previously only on
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=abSplice" target="_blank">GRCh38/hg38</a>
and is now also available on the human GRCh37/hg19 genome assembly. AbSplice is a method that
predicts aberrant splicing across human tissues, as described in <a href="https://doi.org/10.1038/s41588-023-01373-3"
target="_blank">Wagner, &Ccedil;elik et al., 2023</a>. This track consists of an aberrant splicing
benchmark dataset that spans over 8.8 million rare variants in 49 human tissues from the
<a target="_blank" href="https://www.gtexportal.org/home/samplingSitePage">Genotype-Tissue
Expression (GTEx) dataset</a> and displays precomputed AbSplice scores for all possible
single-nucleotide variants genome-wide. The AbSplice score is a probability estimate of how likely
aberrant splicing of some sort takes place in a given tissue. Aberrant splicing predictions for
tissues that are related to disease may prove to be helpful in identifying the effector gene. The
authors <a href="https://github.com/gagneurlab/absplice?tab=readme-ov-file#output"
target="_blank">suggest</a> three cutoffs which are represented by color in the track.</p>
<ul>
  <li><b><font color="#FF0000">High (red)</font></b> - <b>
    An AbSplice score over 0.2</b> indicates a high likelihood of aberrant splicing in at least one
    tissue.</li>
  <li><b><font color="#FF8000">Medium (orange)</font></b> - <b>
    A score between 0.05 and 0.2 </b> indicates a medium likelihood.</li>
  <li><b><font color="#0000FF">Low (blue)</font></b> - <b>
    A score between 0.01 and 0.05 </b> indicates a low likelihood.</li>
  <li><b>Scores below 0.01 are not displayed.</b></li>
</ul>

<p>
We would like to thank Wagner, &Ccedil;elik et al., 2023 for generating and making the data publicly
available. We would also like to thank Jeltje van Baren, Jairo Navarro, and Gerardo Perez for their
efforts on this release.</p>

<a name="032624"></a>
<h2>Mar. 26, 2024 &nbsp;&nbsp; New gnomAD v4 Constraint Metrics (hg38) and gnomAD Non-cancer filter (hg19/hg38)</h2>
<p>
We are pleased to announce the release of the
<a href="/cgi-bin/hgTrackUi?&db=hg38&g=gnomadPLI">gnomAD v4 constraint metrics</a> for human
assembly hg38/GRCh38 and the addition of a non-cancer filter for the
<a href="/cgi-bin/hgTrackUi?db=hg19&g=gnomadExomes">GRCh37/hg19 Exome Variants v2.1.1</a> and the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=gnomadGenomesVariantsV3_1_1">
hg38/GRCh38 Genome Variants v3.1.1</a> tracks.</p>
<p>
The gnomAD v4 constraint metrics tracks contain metrics of pathogenicity per-gene as predicted for
gnomAD v4.0 and identify genes subject to strong selection against various classes of mutation.
The release of the gnomAD v4 constraint metrics consists of two new subtracks in the gnomAD
Constraint Metrics composite track:</p>
<ol>
  <li><b>Transcript LoF Constraint track</b>: Predicted constraint metrics at the whole transcript
    level for three different types of variation: missense, synonymous, and predicted loss of
    function.</li>
  <li><b>Transcript Missense Constraint track</b>: The missense constraint tracks are built
    similarly to the LoF constraint tracks, however the items displayed are based on missense Z
    scores. All items are colored black, and individual Z scores can be seen on mouseover.
</ol>
<p>
The GRCh37/hg19 Exome Variants v2.1.1 and the hg38/GRCh38 Genome Variants v3.1.1 tracks now have a
non-cancer filter that allows the option to exclude/include variants from samples of individuals
who were not ascertained for having cancer in a cancer study. These non-cancer subsets follow the
new guidelines for clinical variant testing of BRCA1 and BRCA2 genes and are used for the
assessment of breast cancer patients.</p>
<p>
We would like to thank the <a href="https://gnomad.broadinstitute.org/about" target="_blank">Genome
Aggregation Database Consortium</a> for making these data available. We would also like to thank
Chris Lee, Ana Benet-Pag&#232;s, Gerardo Perez, and Jairo Navarro for the creation and release of
these tracks.</p>

<a name="030724"></a>
<h2>Mar. 07, 2024 &nbsp;&nbsp; New Prediction Scores super track and BayesDel track for hg19</h2>
<p>
We are happy to announce the new Human Prediction Scores super track for the
<a href="/cgi-bin/hgTrackUi?db=hg19&g=predictionScoresSuper">GRCh37/hg19</a> assembly. This super
track currently includes the <a href="/cgi-bin/hgTrackUi?db=hg19&g=bayesDel">BayesDel track</a>,
which can be used for clinical variant classification research.
<a href="https://fenglab.chpc.utah.edu/BayesDel/BayesDel.html" target="_blank">BayesDel</a> is a
deleteriousness meta-score for coding and non-coding variants, single nucleotide variants, and small
insertion/deletions. The range of the score is from -1.29334 to 0.75731. The higher the score, the
more likely the variant is pathogenic. There are eight subtracks for the BayesDel track: four
include pre-computed MaxAF-integrated BayesDel scores for missense variants, one for each base.
The other four are of the same format, but scores are not MaxAF-integrated.</p>
<p>
We would like to thank the
<a href="https://fenglab.chpc.utah.edu/BayesDel.html" target="_blank">BayesDel team</a> for
providing precomputed data. We would also like to thank Tiana Pereira, Christopher Lee, Jeltje van
Baren, Gerardo Perez, and Anna Benet-Pages for their efforts on this release.</p>
 
<a name="030524"></a>
<h2>Mar. 05, 2024 &nbsp;&nbsp; New JASPAR tracks: Human (hg19/hg38) - Mouse (mm10/mm39)</h2>
<p>
We are excited to announce the new JASPAR 2024 tracks for human
(<a href="/cgi-bin/hgTrackUi?db=hg19&g=jaspar">GRCh37/hg19</a> and
<a href="/cgi-bin/hgTrackUi?db=hg38&g=jaspar">GRCh38/hg38</a>) and
mouse (<a href="/cgi-bin/hgTrackUi?db=mm39&g=jaspar">GRCm39/mm39</a> and
<a href="/cgi-bin/hgTrackUi?db=mm10&g=jaspar">GRCm38/mm10</a>). These tracks represent genome-wide
predicted binding sites for transcription factors with binding profiles in the
<a href="https://jaspar.genereg.net/about/" target="_blank">JASPAR CORE collection</a>. JASPAR CORE
is an open-source database containing a curated, non-redundant set of binding profiles derived from
collections of experimentally defined transcription factor binding profiles. The JASPAR 2024 update
expanded the JASPAR CORE collection by 20% (329 added and 72 upgraded profiles). JASPAR continues
to uphold its core principles (i) providing high-quality TF binding profiles, (ii) fostering open
access, and (iii) ensuring ease of use, which has been useful for the scientific community in
studying gene transcription regulation.</p>
<p>
The JASPAR database is a joint effort between several labs (please see the latest
<a href="https://pubmed.ncbi.nlm.nih.gov/37962376/" target="_blank">JASPAR paper</a>).
Binding site predictions and UCSC tracks were computed by the
<a href="https://mathelierlab.com/"
target="_blank">Computational Biology & Gene Regulation group</a>. We would like to thank Jairo
Navarro and Gerardo Perez at UCSC for building and testing these tracks.</p>

<a name="030124"></a>
<h2>Mar. 01, 2024 &nbsp;&nbsp; AbSplice Prediction Scores for hg38</h2>
<p>
We are happy to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=abSplice" target="_blank">AbSplice scores</a>
track for the human genome, GRCh38/hg38. AbSplice is a method that predicts aberrant splicing across
human tissues, as described in <a href="https://doi.org/10.1038/s41588-023-01373-3"
target="_blank">Wagner, &Ccedil;elik et al., 2023</a>. This track displays precomputed AbSplice
scores for all possible single-nucleotide variants genome-wide. The scores represent the probability
that a given variant causes aberrant splicing in a given tissue.</p>
<p>
Aberrant splicing is a major cause of genetic disorders but its direct detection in transcriptomes
is limited to clinically accessible tissues such as skin or body fluids. &Ccedil;elik et al.
generated an aberrant splicing benchmark dataset, spanning over 8.8 million rare variants in 49
human tissues from the Genotype-Tissue Expression (GTEx) dataset. The AbSplice score is a
probability estimate of how likely aberrant splicing of some sort takes place in a given tissue.
The authors <a href="https://github.com/gagneurlab/absplice?tab=readme-ov-file#output"
target="_blank">suggest</a> three cutoffs which are represented by color in the track.</p>

<ul>
  <li><b><font color="#FF0000">High (red)</font></b> - <b>
    An AbSplice score over 0.2</b> indicates a high likelihood of aberrant splicing in at least one
    tissue.</li>
  <li><b><font color="#FF8000">Medium (orange)</font></b> - <b>
    A score between 0.05 and 0.2 </b> indicates a medium likelihood.</li>
  <li><b><font color="#0000FF">Low (blue)</font></b> - <b>
    A score between 0.01 and 0.05 </b> indicates a low likelihood.</li>
  <li><b>Scores below 0.01 are not displayed.</b></li>
</ul>

<p>
We would like to thank Wagner, &Ccedil;elik et al., 2023 for generating and making the data publicly
available. We would also like to thank Jeltje van Baren and Jairo Navarro for the creation and
release of these tracks.
</p>

<a name="022124"></a>
<h2>Feb. 21, 2024 &nbsp;&nbsp; New Collins et al 2024 Dosage Sensitivity tracks for Human (hg19/hg38)</h2>
<p>
We are happy to announce the release of dosage sensitivity tracks for human assemblies,
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=dosageSensitivity"
target="_blank">GRCh38/hg38</a> and <a href="/cgi-bin/hgTrackUi?db=hg19&c=chr12&g=dosageSensitivity"
target="_blank">GRCh37/hg19</a>, displaying a cross-disorder dosage sensitivity map of the human
genome. The two tracks correspond to the probability of haploinsufficiency (pHaplo) and the
probability of triplosensitivity (pTriplo).</p>
<p>
Rare copy-number variants (rCNVs) include deletions and duplications that occur infrequently in the
global human population and can confer substantial risk for disease.
<a href="https://europepmc.org/article/MED/35917817" target="_blank">Collins et al</a> aimed to
quantify the properties of haploinsufficiency (i.e., deletion intolerance) and triplosensitivity
(i.e., duplication intolerance) throughout the human genome by analyzing rCNVs from nearly one
million individuals to construct a genome-wide catalog of dosage sensitivity across 54 disorders,
which defined 163 dosage sensitive segments associated with at least one disorder. These segments
were typically gene-dense and often harbored dominant dosage sensitive driver genes. An ensemble
machine learning model was built to predict dosage sensitivity probabilities (pHaplo &amp; pTriplo)
for all autosomal genes, which identified 2,987 haploinsufficient and 1,559 triplosensitive genes,
including 648 that were uniquely triplosensitive.
</p>
<p>
We would like to thank Ryan Collins for generating the data. We would also like to
thank Anna Benet-Pag&egrave;s for suggesting and assisting in track development and interpretation.
Thank you to Luis Nassar and Jairo Navarro for the creation and release of these tracks.</p>

<a name="021424"></a>
<h2>Feb. 14, 2024 &nbsp;&nbsp; New GENCODE gene tracks: Human V45 (hg19/hg38) - Mouse VM34
(mm39)</h2>
<p>
We are happy to announce new GENCODE Gene annotation tracks, which correspond
to <a href="https://www.ensembl.info/2024/01/11/ensembl-111-has-been-released/"
target="_blank">Ensembl 111</a>, for three assemblies: <a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV45lift37">hg19/GRCh37</a>,
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&g=wgEncodeGencodeV45">hg38/GRCh38</a>,
and
<a target="_blank" href="/cgi-bin/hgTrackUi?db=mm39&c=chr12&g=wgEncodeGencodeVM34">mm39/GRCm39</a>.
<a target="_blank" href="https://www.gencodegenes.org/">GENCODE</a> provides gene annotations with
high accuracy based on biological evidence and for the benefit of biomedical research and genome
interpretation. For human, the GENCODE V45 annotations were mapped to hg38/GRCh38 and then back-mapped
to the hg19/GRCh37 assembly. For all three assemblies, the gene sets contain the
following tracks:</p>
<ul>
  <li>
      Basic - a subset of the <em>Comprehensive set</em>.</li>
  <li>
      Comprehensive - all GENCODE coding and non-coding transcript annotations, including
      polymorphic pseudogenes. This includes both manual and automatic annotations.</li>
  <li>
      Pseudogenes - all annotations except polymorphic pseudogenes.</li>
</ul>
<p>
The hg38 and mm39 assemblies also include the following track that is not available on hg19:
</p>
<ul>
  <li>
      PolyA - polyA signals and sites manually annotated on the genome based on transcribed
      evidence (ESTs and cDNAs) of 3' end of transcripts containing at least 3 A's not matching the
      genome.</li>
</ul>
<p>
Details on each release can be found on the <a target="_blank"
href="https://www.gencodegenes.org/">GENCODE site</a>. This includes <a target="_blank"
href="https://www.gencodegenes.org/human/stats.html">statistics on each release</a>.
</p>
<p>
We would like to thank the <a target="_blank"
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Mark Diekhans and Gerardo Perez for the development and
release of these tracks.</p>

<a name="021224"></a>
<h2>Feb. 12, 2024 &nbsp;&nbsp; Variants of Concern SARS-CoV-2 track updated with Omicron
variants</h2>
<p>
We are happy to announce an update to the <a href="/cgi-bin/hgTrackUi?db=wuhCor1&position=default&g=variantMuts"
target="_blank">Variants of Concern</a> track to include the latest WHO-designated variants of
concern (VOC). The evolving SARS-CoV-2 virus has been a challenge for everyone, and we are humbled
to continue providing public data and our high-quality genetics research tools, now updated with the
latest COVID variant information.</p>
<p>
In this update, the following Omicron variants were added to the
Variants of Concern track:</p>
<div class="container">
  <div class="row">
    <div class="col-sm-4">
      <ul>
        <li>BA.2</li>
        <li>BA.2.12.1</li>
        <li>BA.2.75</li>
        <li>BQ.1</li>
        <li>CH.1.1</li>
      </ul>
    </div>
    <div class="col-sm-4">
      <ul>
        <li>EG.5.1</li>
        <li>XBB</li>
        <li>XBB.1.16</li>
        <li>XBB.1.5</li>
        <li>XBB.1.9</li>
      </ul>
    </div>
    <div class="col-sm-4">
      <ul>
        <li>XBB.2.3</li>
        <li>XBB.1.5.70</li>
        <li>HK.3</li>
        <li>BA.2.86</li>
        <li>JN.1</li>
      </ul>
    </div>
  </div>
</div>
<p>
The Variants of Concern track displays amino acid and nucleotide mutations in SARS-CoV-2 variants as
defined in December 2021 by the
<a href="https://www.who.int/en/activities/tracking-SARS-CoV-2-variants/"
target="_blank">World Health Organization (WHO)</a>. Note that the
<a href="https://www.cdc.gov/coronavirus/2019-ncov/variants/variant-classifications.html"
target="_blank">Centers for Disease Control (CDC)</a> classification of SARS-CoV-2 variants is
slightly different than the WHO. Mutations in this track were identified from viral sequences from
<a href="https://gisaid.org/" target="_blank">GISAID</a>. Variant incidence and geographic
distribution information is available from links to the <a href="https://outbreak.info/"
target="_blanK">Outbreak.info</a> web resource on the mutation details pages.</p>
<p>
This work is made possible by the open sharing of genetic data by research groups from all over the
world. We gratefully acknowledge their contributions. We would also like to thank Angie Hinrichs
and Jairo Navarro for creating and releasing these tracks.</p>

<a name="020824"></a>
<h2>Feb. 8, 2024 &nbsp;&nbsp; CRISPR Targets for human (T2T CHM13v2.0/hs1) now available</h2>
<p>
We are pleased to announce the release of the <a href="/cgi-bin/hgTrackUi?db=hub_3671779_hs1&position=default&g=hub_3671779_crisprHs1"
target="_blank">CRISPR Targets</a> track for human (T2T CHM13v2.0/hs1).
</p>
<p>
The CRISPR Targets track shows the DNA sequences targetable by CRISPR RNA guides using the Cas9 enzyme from
S. pyogenes (PAM: NGG) over the entire human (hs1) genome. CRISPR target sites were annotated with
predicted specificity (off-target effects) and predicted efficiency (on-target cleavage) by various
algorithms through the tool CRISPOR. Sp-Cas9 usually cuts double-stranded DNA three or four base
pairs 5' of the PAM site.
</p>
<p>
We would like to thank Maximilian Haeussler, Hiram Clawson, and Jairo Navarro for developing and
releasing these tracks.</p>

<a name="012324"></a>
<h2>Jan. 23, 2024 &nbsp;&nbsp; AVADA variants track available for human (hg38)</h2>
<p>
We are happy to announce the release of the <a href="/cgi-bin/hgTrackUi?db=hg38&g=avada"
target="_blank">AVADA variants</a> track for human (GRCh38/hg38). AVADA is a database of variants
built by machine learning software that analyzes full text research articles to find genes and
variants that appear relevant for monogenic genetic diagnosis.
</p>
<p>
The Avada Variants track shows the genomic positions of variants in the
<a href="http://bejerano.stanford.edu/AVADA/" target="_blank">AVADA database</a>. Genomic locations
of variants are labeled with the variant description in the original text. This is not a
normalized HGVS string, but the original text as the authors of the study described it. The Pubmed
ID, gene, and transcript for each variant are shown on the variant's details page, as well as the
PubMed title, authors, and abstract.
</p>
<p>
Mouse-over the variants to show the gene, variant, first author, year, and title.
</p>
<p>
This data has been lifted from hg19 to hg38. Additional information can be found on the
<a href="https://doi.org/10.1038/s41436-019-0643-6" target="_blank">AVADA publication</a>. We would
like to thank Gill Bejerano and Johannes Birgmeier for making the AVADA data available. We would
also like to thank Maximilian Haeussler and Jairo Navarro for creating and releasing the tracks. 
</p>

<a name="012224"></a>
<h2>Jan. 22, 2024 &nbsp;&nbsp; New hg38 HPRC track group and data</h2>
<p>
We are proud to announce the release of four new tracks and a new track
group on hg38 dedicated to the NIH's <a target="_blank"
href="https://humanpangenome.org/">Human Pangenome Reference Consortium
(HPRC)</a> data.</p>

<div class="text-center">
	<img class='text-center' src="../images/HPRCtrackGroup.png" width='80%' alt="New 
hg38 HPRC track group.">
</div>

<p>
No single reference genome such as hg19 or hg38 can accurately represent
human genetic diversity. The <a target="_blank" 
href="https://humanpangenome.org/">HPRC's</a> goal is to improve this by sequencing
thousands of human genomes at high quality and building new tools to
improve working with them. The <a target="_blank" 
href="https://www.nature.com/articles/s41586-023-05896-x">first data release</a> 
from this project consists of 47
phased, diploid assemblies, more than 99% accurate at the structural and
base pair levels. We obtained alignments of these new genomes to hg38 from
the HPRC analysis groups and have created new Genome Browser annotation
tracks that visualize the differences between the established hg38
reference and the new 94 pan-genome assemblies. The new tracks are grouped
into short and structural variants, with the latter further split by type
(insertion, deletion, inversion, duplication, etc). We plan to update these
and add other tracks as soon as more HPRC data is released.<br><br></p>

<figure class="text-center">
<a target="_blank" href="https://genome.ucsc.edu/s/Lou/HPRC">
<img class='text-center' src="../images/HPRCsession.png" width='80%' alt="hg38 session
visualizing the new HPRC tracks."></a>
<figcaption style="font-size:13px">
Session on hg38 near the MHC region where the reference sequence upstream of HLA-F contains
an insertion present in only a few of the HPRC populations. Few HPRC and no common dbSNP155
variants are annotated, further evidencing this is a region on the reference that is likely
an uncommon insertion in the global population. Click on the image to explore the session
further.</figcaption>
</figure>

<p>
In this first HPRC data release, we are adding four new tracks to this new track group. 
Details on each of the tracks are as follows:</p>

<h3>Feature and Variation Tracks</h3>
<p>
The <a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&g=hprcVCF">Short Variants 
container track</a> shows tracks of short nucleotide variants of a few base 
pairs when aligning HPRC genomes to the hg38 reference assembly using the 
Minigraph-cactus approach. Short variants have been used in population genetics 
to investigate population-specific allele frequencies and genetic diversity, and 
have been used in the association of diseases. The track consists of three subtracks:</p>
<ul>
<li><b>HPRC All Variants</b>: HPRC variants decomposed from 
hprc-v1.0-mc.grch38.vcfbub.a100k.wave.vcf.gz (<a target="_blank" 
href="https://pubmed.ncbi.nlm.nih.gov/37165242/">Liao et al 2023</a>), no size filtering</li>
<li><b>HPRC Variants &ge; 3bp</b>: HPRC VCF variants filtered for items size &ge; 3bp</li>
<li><b>HPRC Variants &gt; 3bp</b>: HPRC VCF variants filtered for items size &gt; 3bp</li></ul>

<p>
The <a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&c=chr6&g=hprcArrV1">Rearrangements 
container track</a> shows various rearrangements in the HPRC assemblies with respect 
to hg38. The types include indels, duplications, inversions, and other more 
complicated rearrangements.</p>
<p>
There are five tracks in the Rearrangement composite track:</p>
<ul>
<li><b>Insertions</b>: Deletions in hg38 = Insertion in the HPRC assemblies</li>
<li><b>Deletions</b>: Insertions in hg38 = Deletion in the HPRC assemblies</li>
<li><b>Inversions</b>: Inversions with respect to hg38 in HPRC assemblies</li>
<li><b>Duplications</b>: Duplications with respect to hg38 in HPRC assemblies</li>
<li><b>Other Rearrangements</b>: Other Rearrangements: Unalignable sequences in both 
assemblies (inversions, partial transpositions)</li></ul>

<p>
Many of these features are unique to this dataset, although overlap can be 
found with other structural variant databases such as DGV. Potential applications of 
these rearrangements could be data validation for new and existing data and a 
better understanding of the prevalence of rearrangements in diverse populations, 
many of which are underrepresented in current clinical and genomic databases.</p>

<h3>Alignment and Conservation tracks</h3>
<p>
The <a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&c=chr6&g=hprcChainNet">Chain/Net 
track</a> shows regions of the human genome that are alignable between the HPRC genomes 
as well as hg38 and T2T-CHM13. A total of 176 maternal and paternal haplotypes 
were used in this analysis. The configuration page for this track sorts 
the haplotypes into 14 subpopulations as follows:</p>

<div class="container">
  <div class="row">
    <div class="col-sm-4">
    <ul>
      <li>T2T</li>
      <li>HAPMAP</li>
      <li>Yoruba Nigeria</li>
      <li>Esan Nigeria</li>
      <li>Gambian</li>
    </ul></div>
    <div class="col-sm-4">
    <ul>
      <li>Mende Sierra Leone</li>
      <li>Afr Carib Barabdos</li>
      <li>African SW USA</li>
      <li>Puerto Rico</li>
      <li>Peru Lima</li>
    </ul></div>
    <div class="col-sm-4">
    <ul>
      <li>Columbia Medellin</li>
      <li>Han SoChina</li>
      <li>Vietnam Kinh</li>
      <li>Punjabo Pakis</li>
    </ul></div>
  </div>
</div>

<p>
The <a target="_blank>" href="/cgi-bin/hgTrackUi?db=hg38&c=chr6&g=consHprc90way">90-way Multiple Alignment track</a> contains multiple alignments of 90 human genomes generated by the Minigraph-Cactus pangenome pipeline, which creates pangenomes directly from whole-genome alignments. This method builds graphs containing all forms of genetic variation while allowing the use of current mapping and genotyping tools. The confirmation page sorts the Maternal and Paternal haplotypes by the same 14 subpopulations described above.
</p>

<h3>Acknowledgments</h3>
<p>
We are always looking for feedback, if you would like to see other HPRC data, or 
the data presented differently, please contact us at <a 
href="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</a>. 
Likewise, if you find this data useful and see potential improvements, we would be interested in hearing from you.
</p>
<p>
We would like to thank the Human Pangenome Reference Consortium for taking on this genomics
challenge and providing these data. In particular, we would like to thank Benedict Paten, 
Heng Li and Glenn Hickey
for their help in putting these Browser tracks together. From our team, we would like to thank
Max Haeussler, Brian Raney, Mark Diekhans, Hiram Clawson, Gerardo Perez, Jairo Navarro, and
Lou Nassar for their work in developing, testing, and releasing these tracks.</p>

<!-- ============= 2023 archived news ============= -->
<a name="2023"></a>
<a name="120723"></a>
<h2>Dec. 7, 2023 &nbsp;&nbsp; VISTA Enhancers for human and mouse</h2>
<p>
We are excited to announce the release of the VISTA Enhancers tracks for human and mouse genome
assemblies. These datasets were previously part of the VISTA Enhancers public hub and have now
been incorporated natively into the Genome Browser to better serve the research community.
</p>
<p>
View the VISTA Enhancer track on the following assemblies:
</p>
<ul>
  <li><a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=vistaEnhancersBb"
       target="_blank">hg38</a>,
      <a href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=vistaEnhancersBb"
       target="_blank">hg19</a></li>
  <li><a href="/cgi-bin/hgTrackUi?db=mm39&position=default&g=vistaEnhancersBb"
       target="_blank">mm39</a>,
      <a href="/cgi-bin/hgTrackUi?db=mm10&position=default&g=vistaEnhancersBb"
       target="_blank">mm10</a>,
      <a href="/cgi-bin/hgTrackUi?db=mm9&position=default&g=vistaEnhancersBb"
       target="_blank">mm9</a></li>
</ul>
<p>
The VISTA Enhancers track contains potential enhancers whose activity was experimentally validated
in transgenic mice. Most of these non-coding elements were selected for testing based on their
extreme conservation in other vertebrates or epigenomic evidence (ChIP-Seq) of putative enhancer
marks. The goal of VISTA Enhancers project is to identify distant-acting transcriptional enhancers
in the human and mouse genomes. More information about can be found on the
<a href="https://enhancer.lbl.gov/" target="_blank">VISTA Enhancer Browser</a> website.
</p>
<p>
We would like to thank the Lawrence Berkeley National Laboratory and the VISTA Enhancer team for
providing this data. We would also like to thank Gerardo Perez and Jairo Navarro for the creation
and release of these tracks.
</p>

<a name="113023"></a>
<h2>Nov. 30, 2023 &nbsp;&nbsp; Support for previous RefSeq transcripts while searching on hg38</h2>
<p>
Have you ever found a variant in a paper and searched for it on the Genome Browser only 
to receive an error that the sequence cannot be found? Or perhaps looked up a familiar 
NM_ identifier and suddenly found no results?</p>
<p>
We are pleased to share that we now have support for searching previous RefSeq 
transcript versions on hg38. This support works for both NM_ accessions 
and <a target="_blank" href="https://hgvs-nomenclature.org/en/latest/">HGVS</a> 
searching as demonstrated below:</p>

<p>
<b>Searching for the latest transcript which always worked:</b>
<ul>
<li>
Sequence search: <a target="_blank" 
href="../cgi-bin/hgSearch?search=NM_198056.3&db=hg38">NM_198056.3</a></li>
<li>
HGVS search: <a target="_blank"
href="../cgi-bin/hgSearch?search=NM_198056.3%3Ac.1A%3EC&db=hg38">NM_198056.3:c.1A>C</a></li></ul></p>
<p>
<b>Searching for a previous version that now works::</b>
<ul>
<li>
Sequence search: <a target="_blank"
href="../cgi-bin/hgSearch?search=NM_198056.3&db=hg38">NM_198056.2</a></li>
<li>
HGVS search: <a target="_blank"
href="../cgi-bin/hgSearch?search=NM_198056.3%3Ac.1A%3EC&db=hg38">NM_198056.2:c.1A>C</a></li></ul></p>

<p>
We thank <a target="_blank" href="https://www.ncbi.nlm.nih.gov/">NCBI</a> and Terence 
Murphy for creating the archive of deprecated transcripts that allows this feature 
to work. We would also like to thank the users who wrote requesting the feature 
allowing us to prioritize it effectively. Finally, we would like to thank Chris Lee, 
Max Haeussler, Gerardo Perez, and Lou Nassar for developing and testing this feature.</p>

<a name="112223"></a>
<h2>Nov. 22, 2023 &nbsp;&nbsp; CRISPR Targets for Zebrafish (danRer10/danRer11) now available</h2>
<p>
We are happy to announce the release of the CRISPR Targets track for the Zebrafish
<a href="../cgi-bin/hgTrackUi?db=danRer10&g=crisprAllTargets" target="_blank">danRer10</a> and
<a href="../cgi-bin/hgTrackUi?db=danRer11&g=crisprAllTargets" target="_blank">danRer11</a>
assemblies. CRISPR-Cas9 has been applied in Zebrafish for gene knockout, gene screening
and other gene editing studies.</p>
<p>
The CRISPR Targets track shows the DNA sequences targetable by CRISPR RNA guides using the Cas9
enzyme from S. pyogenes (PAM: NGG) over the entire zebrafish genome. CRISPR target sites were
annotated with predicted specificity (off-target effects) and predicted efficiency (on-target
cleavage) by various algorithms through the tool
<a href="http://crispor.tefor.net/" target="_blank">CRISPOR</a>. The target sequence of the guide
is shown with a thick (exon) bar. The PAM motif match (NGG) is shown with a thinner bar. Guides are
colored to reflect both predicted specificity and efficiency. Specificity reflects the "uniqueness"
of a 20mer sequence in the genome; the less unique a sequence is, the more likely it is to cleave
other locations of the genome (off-target effects). Efficiency is the frequency of cleavage at the
target site (on-target efficiency).</p>
<p>
We would like to thank Maximilian Haeussler, Hiram Clawson, and Gerardo Perez for developing and
releasing these tracks.</p>


<a name="110823"></a>
<h2>Nov. 08, 2023 &nbsp;&nbsp; New track decorators feature</h2>
<p>
We are excited to introduce the new track decorators feature which allows highlighting parts of
features with colors and/or symbols (glyphs/shapes) within a single track.
</p>
<div class="text-center">
  <a href="http://genome.ucsc.edu/s/gperez2/RM_32467" target="_blank">
  <img src="../images/newsArchImages/feature_decorators.png" style="width:80%;max-width:1083px"></a>
</div>
<p>
The genome browser&lsquo;s primary way to annotate the genome uses colored rectangles
(&ldquo;exons&rdquo; for gene tracks) linked by thin lines (&ldquo;introns&rdquo;), often stored
as a bigBed. These were originally used for genes but then evolved to cover other types of
annotations, e.g. enhancers, chromatin modifications, or single nucleotide variants. We usually
call these annotations &ldquo;features&rdquo;. Each rectangle (&ldquo;exon&rdquo;) of a feature has
the same color and individual parts cannot be highlighted. If you wanted to highlight parts of the
features, traditionally this required a second track.
</p>
<div class="text-center">
        <img src="../../images/runx2.png" style="width:80%;max-width:1083px">
</div>
<p>
Track decorators change this: in your custom track and track hubs, you can now highlight parts of a
genome annotation with colors or symbols. Track decorators can be shown in two styles,
&ldquo;block&rdquo; and &ldquo;glyph&rdquo; style, and can either be overlaid onto the feature or
shown directly underneath. The &ldquo;block&rdquo; style option can be used to color exons and
introns and can display a label for them. For example, the &ldquo;block&rdquo; track decorator
could be used to overlay protein domain boundaries on transcripts where usually one would use an
entirely different track for the domains.
</p>
<div class="text-center">
        <img src="../../images/runx2_decorator.png" style="width:80%;max-width:1083px">
</div>
<p>
The &ldquo;glyph&rdquo; style option offers 8 different types of glyphs and the color of choice.
</p>
<div class="text-center">
        <img src="../../images/glyphsEx.png" style="width:40%;max-width:1083px">
</div>
<p>
The &ldquo;glyph&rdquo; style option can be used to draw entirely new symbols, for example, to
indicate insertion positions on the genome with small triangles. For more information, see the
<a href="/goldenPath/help/decorator.html" target="_blank">Track Decorators help page</a>.</p>
<p>
We would like to thank Jonathan Casper, Max Haeussler, Mark Diekhans, and Gerardo Perez for their
work on this release. We appreciate user feedback. If you have questions, feedback, suggestions,
glyph style requests, or have found new glyph applications, please contact our
<a href="mailto:genome@soe.ucsc.edu">mailing list</a>.
We would appreciate responses on the creative ways of using the new track decorators feature.
</p>

<a name="102323"></a>
<h2>Oct. 23, 2023 &nbsp;&nbsp; eMERGE polygenic risk scores for human (hg19)</h2>
<p>
We are pleased to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg19&position=default&g=prsEmerge">eMerge polygenic risk scores</a>
for the human assembly GRCh37/hg19. Polygenic risk scores (PRS) have clinical utility and are the
result of many years of GWAS studies. A score is given for a combination of SNPs to calculate the
risk of getting a disease in a healthy population. These scores work very well for cancer,
cardiomyopathies, and is extending to other diseases.
</p>
<p>
The Polygenic Risk Scores eMERGE track shows variants that are part of selected polygenic risk
scores for 8 common diseases. The risk scores were selected by the
<a href="https://www.genome.gov/Funded-Programs-Projects/Electronic-Medical-Records-and-Genomics-Network-eMERGE"
target="_blank">NHGRI eMERGE project</a>, and the selection process is described in
<a href="https://doi.org/10.1101/2023.05.25.23290535" target="_blank">Lennon et al. 2023</a>. Many
PRS models were evaluated, and the 8 models in this track were selected based on quality:
</p>
<ul>
  <li>Asthma</li>
  <li>Atrial Fibrillation</li>
  <li>Breast Cancer</li>
  <li>Coronary Heart Disease</li>
  <li>Chronic Kidney Disease</li>
  <li>Hypercholesterolemia</li>
  <li>Prostate Cancer</li>
  <li>T1 Diabetes</li>
  <li>T2 Diabetes</li>
</ul>
<p>
We would like to thank Elisabeth McNally for providing advice and Niall Lennon for sharing the data
with the UCSC Genome Browser team. We would also like to thank Zia Truong and Max Haeussler for
building these tracks, and Jairo Navarro for releasing the tracks.
</p>

<a name="091923"></a>
<h2>Sep. 19, 2023 &nbsp;&nbsp; EVA SNP release 5 for 36 assemblies</h2>
<p>
We are happy to announce the release of the EVA SNP release 5 track for many assemblies. These
tracks contain mappings of single nucleotide variants and small insertions and deletions (indels)
&mdash; collectively Simple Nucleotide Variants (SNVs) &mdash; from the European Variation Archive
(<a href="https://www.ebi.ac.uk/eva/" target="_blank">EVA</a>) Release 5. The full list of
assemblies that contain the EVA SNP release 5 track is below:
<div class="container">
  <div class="row">
    <div class="col-sm-4">
    <ul>
      <li>Crab-eating macaque <a href="../cgi-bin/hgTrackUi?db=macFas5&g=evaSnp5"
          target="_blank">(macFas5)</a></li>
      <li>D. melanogaster <a href="../cgi-bin/hgTrackUi?db=dm6&g=evaSnp5"
          target="_blank">(dm6)</a></li>
    </ul></div>
    <div class="col-sm-4">
    <ul>
      <li>Dog <a href="../cgi-bin/hgTrackUi?db=canFam3&g=evaSnp5"
          target="_blank">(canFam3)</a></li>
      <li>Nile tilapia <a href="../cgi-bin/hgTrackUi?db=oreNil2&g=evaSnp5"
          target="_blank">(oreNil2)</a></li>
      <li>Orangutan <a href="../cgi-bin/hgTrackUi?db=ponAbe3&g=evaSnp5"
          target="_blank">(ponAbe3)</a></li>
    </ul></div>
    <div class="col-sm-4">
    <ul>
      <li>Zebra finch <a href="../cgi-bin/hgTrackUi?db=taeGut2&g=evaSnp5"
          target="_blank">(taeGut2)</a></li>
      <li>Zebrafish <a href="../cgi-bin/hgTrackUi?db=danRer7&g=evaSnp5"
          target="_blank">(danRer7)</a>, <a href="../cgi-bin/hgTrackUi?db=danRer10&g=evaSnp5"
          target="_blank">(danRer10)</a>, <a href="../cgi-bin/hgTrackUi?db=danRer11&g=evaSnp5"
          target="_blank">(danRer11)</a></li>
    </ul></div>
  </div>
</div>
<p>
Variants are classified by EVA into one of the following <a target="_blank"
href="http://www.sequenceontology.org/">sequence ontology</a> terms:
</p>
<ul>
  <li> <b>substitution</b> &mdash;
       A single nucleotide in the reference is replaced by another, alternate allele
  <li> <b>deletion</b> &mdash;
       One or more nucleotides is deleted.  The representation in the database is to
       display one additional nucleotide in both the Reference field (Ref) and the
       Alternate Allele field (Alt).  E.g. a variant that is a deletion of an A
       maybe be represented as Ref = GA and Alt = G.
  <li> <b>insertion</b> &mdash;
       One or more nucleotides is inserted.  The representation in the database is to
       display one additional nucleotide in both the Reference field (Ref) and the
       Alternate Allele field (Alt).  E.g. a variant that is an insertion of a T maybe
       be represented as Ref = G and Alt = GT
  <li> <b>delins</b> &mdash;
       Similar to tandemRepeat, in that the runs of Ref and Alt Alleles are of
       different length, except that there is more than one type of nucleotide,
       e.g., Ref = CCAAAAACAAAAACA, Alt = ACAAAAAC.
  <li> <b>multipleNucleotideVariant</b> &mdash;
       More than one nucleotide is substituted by an equal number of different
       nucleotides, e.g.,  Ref = AA, Alt = GC.
  <li> <b>sequence alteration</b> &mdash;
       A parent term meant to signify a deviation from another sequence. Can be
       assigned to variants that have not been characterized yet.
</ul>
<p>
We would like to thank the <a href="https://www.ebi.ac.uk/eva/" target="_blank">European Variation
Archive</a> making this data publically available. We would also like to thank Luis Nassar and Jairo
Navarro for the creation and release of these tracks.
</p>

<a name="091523"></a>
<h2>Sep. 15, 2023 &nbsp;&nbsp; New COSMIC Track for hg38</h2>
<p>
We are pleased to announce the release of the new
<a href="/cgi-bin/hgTrackUi?db=hg38&c=chr17&g=cosmicMuts">COSMIC track</a>
for hg38. The
<a target="_blank"
href="https://cancer.sanger.ac.uk/cosmic/about">Catalogue Of Somatic Mutations In Cancer</a>
(COSMIC) is an online database of expert manually curated somatic mutation information relating to
human cancers. This new track displays data from the
<a target="_blank" href="https://cancer.sanger.ac.uk/cosmic/release_notes/">COSMIC v98 release</a>,
which consists of 410,000 new genomic variants, 585,000 new coding mutations, 290,000 non-coding
mutations, 4,300 new samples, and 1,358 new whole genomes. More information on this track can be
found on the
<a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg38&g=cosmicMuts#TRACK_HTML">track description page</a>.</p>
<p>
This track will receive yearly updates and is meant as a replacement for the previous
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&c=chr17&g=cosmicRegions">COSMIC Regions track</a>.
The COSMIC Regions track will no longer be updated</p>
<p>
We would like to thank COSMIC for providing the data to UCSC. We would also like to thank
Lou Nassar, Max Haeussler, and Gerardo Perez for their efforts on this release.</p>

<a name="090723"></a>
<h2>Sep. 07, 2023 &nbsp;&nbsp; New GENCODE &quot;KnownGene&quot; V44 for human (hg38) and VM33 for mouse (mm39)</h2>
<p>
We are pleased to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=knownGene&c=chrX">GENCODE V44 (hg38)</a>
and the
<a href="/cgi-bin/hgTrackUi?db=mm39&g=knownGene&c=chrX">GENCODE VM33 (mm39)</a> gene tracks.
The GENCODE &quot;KnownGene&quot; V44 for hg38 and VM33 gene tracks were built using a UCSC 
pipeline (KnownGene) and the GENCODE comprehensive gene set to generate high-quality manual
annotations merged with evidence-based automated annotations. The GENCODE &quot;KnownGene&quot;
tracks are our default gene tracks, which have extensive associations to external sources. This
allows for additional metadata on every item as well as external links. The track description pages
contain options for configuring the display, such as showing non-coding genes, splice variants, and
pseudogenes.</p>

<p>
Below is a summary of the contents found in each release. For more details visit the <a target="_blank"
href="https://www.gencodegenes.org/">GENCODE site</a>.</p>
<p>
<table class="stdTbl">
<tr><th COLSPAN=4>GENCODE v44 Release Stats</th></tr>
<tr align=left><th>Genes</th><th>Observed</th><th>Transcripts</th><th>Observed</th></tr>
<tr align=left><td>Protein-coding genes</td><td>19,396</td><td>Protein-coding transcripts</td><td>89,067</td></tr>
<tr align=left><td>Long non-coding RNA genes</td><td>19,922</td><td><font size="-1">- full length protein-coding</font></td><td>63,968</td></tr>
<tr align=left><td>Small non-coding RNA genes</td><td>7,566</td><td><font size="-1">- partial length protein-coding</font></td><td>25,099</td></tr>
<tr align=left><td>Pseudogenes</td><td>14,735</td><td>Nonsense mediated decay transcripts</td><td>21,384</td></tr>
<tr align=left><td>Immunoglobulin/T-cell receptor gene segments</td><td>647</td><td>Long non-coding RNA loci transcripts</td><td>58,246</td></tr>
<tr align=left><td>Total No of distinct translations</td><td>65,342</td><td>Genes that have more than one distinct translations</td><td>13,594</td></tr>
</table><BR>
</p>
<p>
<table class="stdTbl">
<tr><th COLSPAN=4>GENCODE VM33 Release Stats</th></tr>
<tr align=left><th>Genes</th><th>Observed</th><th>Transcripts</th><th>Observed</th></tr>
<tr align=left><td>Protein-coding genes</td><td>21,403</td><td>Protein-coding transcripts</td><td>58,750</td></tr>
<tr align=left><td>Long non-coding RNA genes</td><td>14,842</td><td><font size="-1">- full length protein-coding</font></td><td>45,112</td></tr>
<tr align=left><td>Small non-coding RNA genes</td><td>6,105</td><td><font size="-1">- partial length protein-coding</font></td><td>13,638</td></tr>
<tr align=left><td>Pseudogenes</td><td>13,809</td><td>Nonsense mediated decay transcripts</td><td>7,218</td></tr>
<tr align=left><td>Immunoglobulin/T-cell receptor gene segments</td><td>701</td><td>Long non-coding RNA loci transcripts</td><td>26,564</td></tr>
<tr align=left><td>Total No of distinct translations</td><td>44,993</td><td>Genes that have more than one distinct translations</td><td>10,893</td></tr>
</table><BR>
</p>

<p>
We would like to thank the <a target="_blank"
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Jonathan Casper and Gerardo Perez for the development and
release of these tracks.</p>


<a name="081823"></a>
<h2>Aug. 18, 2023 &nbsp;&nbsp; New GENCODE gene tracks: Human V44 (hg19/hg38) - Mouse VM33
(mm39)</h2>
<p>
We are pleased to announce new GENCODE Gene annotation tracks, which correspond
to <a href="https://www.ensembl.info/2023/07/17/ensembl-110-has-been-released/"
target="_blank">Ensembl 110</a>, for three assemblies: <a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg19&c=chrX&g=wgEncodeGencodeV44lift37">hg19/GRCh37</a>,
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=wgEncodeGencodeV44">hg38/GRCh38</a>,
and
<a target="_blank" href="/cgi-bin/hgTrackUi?db=mm39&c=chr12&g=wgEncodeGencodeVM33">mm39/GRCm39</a>.
For human, the GENCODE V44 annotations were mapped to hg38/GRCh38 and then back-mapped
to the hg19/GRCh37 assembly. For all three assemblies, the gene sets contain the
following tracks:</p>
<ul>
  <li>
      Basic - a subset of the <em>Comprehensive set</em>.</li>
  <li>
      Comprehensive - all GENCODE coding and non-coding transcript annotations, including
      polymorphic pseudogenes. This includes both manual and automatic annotations.</li>
  <li>
      Pseudogenes - all annotations except polymorphic pseudogenes.</li>
</ul>
<p>
The hg38 and mm39 assemblies also include the following track that is not available on hg19:
</p>
<ul>
  <li>
      PolyA - polyA signals and sites manually annotated on the genome based on transcribed
      evidence (ESTs and cDNAs) of 3' end of transcripts containing at least 3 A's not matching the
      genome.</li>
</ul>
<p>
Details on each release can be found on the <a target="_blank"
href="https://www.gencodegenes.org/">GENCODE site</a>. This includes <a target="_blank"
href="https://www.gencodegenes.org/human/stats.html">statistics on each release</a>.
</p>
<p>
We would like to thank the <a target="_blank"
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Mark Diekhans and Gerardo Perez for the development and
release of these tracks.</p>


<a name="080723"></a>
<h2>Aug. 07, 2023 &nbsp;&nbsp; Introducing an interactive tutorial for the UCSC Genome Browser</h2>
<p>
We are happy to announce the release of a new interactive tutorial for the UCSC Genome Browser. The
tutorial is designed to help new users navigate the UCSC Genome Browser. Topics such as navigating
around the Genome Browser display, configuring track display settings, searching for tracks, and
viewing the negative strand (3' to 5') are covered in the tutorial.</p>
<p>
To begin the tutorial, select the <em>Interactive tutorial</em> from the &quot;Help&quot; drop-down
menu.
<div class="text-center">
  <a href="/cgi-bin/hgTracks?db=hg38&position=default">
    <img src="/images/newsArchImages/hgTracksTutorial.png" width="900">
  </a>
</div>
</p>
<p>
We would like to thank Chris Lee and Jairo Navarro for their work in creating this tutorial.</p>

<a name="080123"></a>
<h2>Aug. 01, 2023 &nbsp;&nbsp; New ability to create duplicate tracks</h2>
<p>
We are pleased to announce the new Duplicate track feature, which allows a copy of a track to
have its own independent settings and be used for multiple display views.</p>
<div class="text-center">
  <a href="http://genome.ucsc.edu/s/gperez2/RM_30365" target="_blank">
  <img src="../images/duplicateDisplay.png"  width="300"></a>
</div>
<p>
You can have multiple duplicates of the same track where the number will increment by one for
each additional duplicate, and each duplicated track is independent of each other.
Only tracks that are not inside of composite or supertracks can be duplicated and this feature does
not work in hubs.</p>
<p>
Duplicate tracks can be helpful when comparing different settings and visibility
types on the same data; for instance, displaying all GENCODE genes, including pseudogenes, in dense
mode, alongside a more concise gene set, such as the default, in pack mode to see gene symbols and
accessions. This is shown in the image below, where duplicated GENCODE track has the additional
text &quot;(duplicate #1)&quot; on the label above the track display and &quot;#1&quot; text on
the left label of the track display.</p>
<div class="text-center">
  <a href="http://genome.ucsc.edu/s/gperez2/RM_30365_v2" target="_blank">
  <img src="../images/duplicationHgTracks2_GENCODEV43.png" alt="GENCODE V43 duplicate" width="1000"></a>
</div>
<p>
For more information, see the "Duplicating a track" section of the
<a target="_blank" href="help/hgTracksHelp.html#duplicate">Genome Browser User Guide</a>.</p>
<p>
We would like to thank Brian Joseph Raney, Max Haeussler, Jairo Navarro, and Gerardo Perez for
their work on this release.</p>


<a name="071923"></a>
<h2>Jul. 19, 2023 &nbsp;&nbsp; FANTOM5 transcription start sites for Human, Mouse, Dog, Rhesus, Rat,
and Chicken</h2>
<p>
We are happy to announce the release of FANTOM5 promoter level expression data for
<a href="/cgi-bin/hgTrackUi?db=hg19&g=fantom5" target="_blank">hg19</a>,
<a href="/cgi-bin/hgTrackUi?db=hg38&g=fantom5" target="_blank">hg38</a>,
<a href="/cgi-bin/hgTrackUi?db=mm10&g=fantom5" target="_blank">mm10</a>,
<a href="/cgi-bin/hgTrackUi?db=canFam3&g=fantom5" target="_blank">canFam3</a>,
<a href="/cgi-bin/hgTrackUi?db=rheMac8&g=fantom5" target="_blank">rheMac8</a>,
<a href="/cgi-bin/hgTrackUi?db=rn6&g=fantom5" target="_blank">rn6</a>, and
<a href="/cgi-bin/hgTrackUi?db=galGal5&g=fantom5" target="_blank">galGal5</a>. The FANTOM5 tracks
shows mapped transcription start sites (TSS) and their usage in primary cells, cell lines, and
tissues to produce a comprehensive overview of gene expression across the human body by using
single molecule sequencing.
</p>
<p>
We would like to thank to the <a href="https://fantom.gsc.riken.jp/5/" target="_blank">FANTOM5
consortium</a>, the Large Scale Data Managing Unit and Preventive Medicine and
Applied Genomics Unit, the <a href="http://www.riken.jp/en/research/labs/ims/" 
target="_blank">Center for Integrative Medical Sciences (IMS)</a>, and
<a href="http://www.riken.jp/" target="_blank">RIKEN</a> for providing this data and analysis. We
would also like to thank Gerardo Perez and Jairo Navarro for the creation and release of these
tracks.</p>


<a name="062123"></a>
<h2>Jun. 21, 2023 &nbsp;&nbsp; Highly Reproducible genomic regions for sequencing for human</h2>
<p>
We are happy to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=highlyReproducible" target="_blank">Highly Reproducible
genomic regions for sequencing</a> for the human <a href="../cgi-bin/hgGateway?db=hg38"
target="_blank">(GRCh38/hg38)</a> genome assembly, which is part of the Problematic Regions
supertrack. The Highly Reproducible Regions composite track highlights regions and variants from
eight samples that can be used to assess variant detection pipelines.</p>
<p>
The &quot;Highly Reproducible Regions&quot; subtrack comprises the intersection of the
reproducible regions across all eight samples, while the &quot;Variants&quot; subtracks contain the
reproducible variants from each assayed sample. Both tracks contain data from the following samples:
</p>
<ul>
  <li>a Chinese Quartet, samples <b>CQ-5</b>, <b>CQ-6</b>, <b>CQ-7</b>, <b>CQ-8</b></li>
  <li>a HapMap Trio, samples <b>NA10385</b>, <b>NA12248</b>, <b>NA12249</b></li>
  <li>a Genome in a Bottle sample, <b>NA12878s</b></li>
</ul>
<p>
We would like to thank the authors of the <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02569-8"
target="_blank">Assessing reproducibility of inherited variants detected with short-read whole
genome sequencing</a> paper for making the data publicly accessible. We would also like to thank
Christopher Lee and Jairo Navarro for the creating and releasing these tracks.</p>

<a name="050423"></a>
<h2>May 04, 2023 &nbsp;&nbsp; EVA SNP release 4 for 36 assemblies</h2>
<p>
We are happy to announce the release of the EVA SNP release 4 tracks for many assemblies. These
tracks contain mappings of single nucleotide variants and small insertions and deletions (indels)
&mdash; collectively Simple Nucleotide Variants (SNVs) &mdash; from the European Variation Archive
(<a href="https://www.ebi.ac.uk/eva/" target="_blank">EVA</a>) Release 4. The full list of
assemblies that contain the EVA SNP release 4 track is below:
<div class="container">
  <div class="row">
    <div class="col-sm-6">
    <ul>
      <li>A. gambiae <a href="../cgi-bin/hgTrackUi?db=anoGam3&g=evaSnp4"
      target="_blank">(anoGam3)</a></li>
      <li>C. intestinalis <a href="../cgi-bin/hgTrackUi?db=ci3&g=evaSnp4"
      target="_blank">(ci3)</a></li>
      <li>Cat <a href="../cgi-bin/hgTrackUi?db=felCat8&g=evaSnp4"
        target="_blank">(felCat8)</a>,
        <a href="../cgi-bin/hgTrackUi?db=felCat9&g=evaSnp4"
        target="_blank">(felCat9)</a></li>
      <li>Chicken <a href="../cgi-bin/hgTrackUi?db=galGal4&g=evaSnp4"
        target="_blank">(galGal4)</a>,
        <a href="../cgi-bin/hgTrackUi?db=galGal5&g=evaSnp4"
        target="_blank">(galGal5)</a>,
        <a href="../cgi-bin/hgTrackUi?db=galGal6&g=evaSnp4"
        target="_blank">(galGal6)</a></li>
      <li>Chimp <a href="../cgi-bin/hgTrackUi?db=panTro5&g=evaSnp4"
      target="_blank">(panTro5)</a></li>
      <li>Cow <a href="../cgi-bin/hgTrackUi?db=bosTau6&g=evaSnp4"
        target="_blank">(bosTau6)</a>,
        <a href="../cgi-bin/hgTrackUi?db=bosTau7&g=evaSnp4"
        target="_blank">(bosTau7)</a>,
        <a href="../cgi-bin/hgTrackUi?db=bosTau9&g=evaSnp4"
        target="_blank">(bosTau9)</a></li>
      <li>Crab-eating macaque <a href="../cgi-bin/hgTrackUi?db=macFas5&g=evaSnp4"
      target="_blank">(macFas5)</a></li>
      <li>D. melanogaster <a href="../cgi-bin/hgTrackUi?db=dm6&g=evaSnp4"
      target="_blank">(dm6)</a></li>
      <li>Dog <a href="../cgi-bin/hgTrackUi?db=canFam3&g=evaSnp4"
      target="_blank">(canFam3)</a></li>
      <li>Gorilla <a href="../cgi-bin/hgTrackUi?db=gorGor3&g=evaSnp4"
        target="_blank">(gorGor3)</a>,
        <a href="../cgi-bin/hgTrackUi?db=gorGor4&g=evaSnp4"
        target="_blank">(gorGor4)</a></li>
      <li>Green monkey <a href="../cgi-bin/hgTrackUi?db=chlSab2&g=evaSnp4"
      target="_blank">(chlSab2)</a></li>
      <li>Horse <a href="../cgi-bin/hgTrackUi?db=equCab3&g=evaSnp4"
      target="_blank">(equCab3)</a></li>
    </ul></div>
    <div class="col-sm-6">
      <ul>
      <li>Marmoset <a href="../cgi-bin/hgTrackUi?db=calJac3&g=evaSnp4"
      target="_blank">(calJac3)</a></li>
      <li>Mouse <a href="../cgi-bin/hgTrackUi?db=mm10&g=evaSnp4"
      target="_blank">(mm10)</a>,
      <a href="../cgi-bin/hgTrackUi?db=mm39&g=evaSnp4"
      target="_blank">(mm39)</a></li>
      <li>Nile tilapia <a href="../cgi-bin/hgTrackUi?db=oreNil2&g=evaSnp4"
      target="_blank">(oreNil2)</a></li>
      <li>Orangutan <a href="../cgi-bin/hgTrackUi?db=ponAbe3&g=evaSnp4"
      target="_blank">(ponAbe3)</a></li>
      <li>Pig <a href="../cgi-bin/hgTrackUi?db=susScr11&g=evaSnp4"
        target="_blank">(susScr11)</a>,
        <a href="../cgi-bin/hgTrackUi?db=susScr3&g=evaSnp4"
        target="_blank">(susScr3)</a></li>
      <li>Rat <a href="../cgi-bin/hgTrackUi?db=rn5&g=evaSnp4"
        target="_blank">(rn5)</a>,
        <a href="../cgi-bin/hgTrackUi?db=rn7&g=evaSnp4"
        target="_blank">(rn7)</a></li>
      <li>Rhesus <a href="../cgi-bin/hgTrackUi?db=rheMac10&g=evaSnp4"
        target="_blank">(rheMac10)</a>,
        <a href="../cgi-bin/hgTrackUi?db=rheMac8&g=evaSnp4"
        target="_blank">(rheMac8)</a></li>
      <li>Sheep <a href="../cgi-bin/hgTrackUi?db=oviAri3&g=evaSnp4"
        target="_blank">(oviAri3)</a>,
        <a href="../cgi-bin/hgTrackUi?db=oviAri4&g=evaSnp4"
        target="_blank">(oviAri4)</a></li>
      <li>Turkey <a href="../cgi-bin/hgTrackUi?db=melGal5&g=evaSnp4"
      target="_blank">(melGal5)</a></li>
      <li>Zebra finch <a href="../cgi-bin/hgTrackUi?db=taeGut2&g=evaSnp4"
      target="_blank">(taeGut2)</a></li>
      <li>Zebrafish <a href="../cgi-bin/hgTrackUi?db=danRer7&g=evaSnp4"
        target="_blank">(danRer7)</a>,
        <a href="../cgi-bin/hgTrackUi?db=danRer10&g=evaSnp4"
        target="_blank">(danRer10)</a>,
        <a href="../cgi-bin/hgTrackUi?db=danRer11&g=evaSnp4"
        target="_blank">(danRer11)</a></li>
    </ul></div>
  </div>
</div>
<p>
Variants are classified by EVA into one of the following <a target="_blank"
href="http://www.sequenceontology.org/">sequence ontology</a> terms:
</p>
<ul>
  <li> <b>substitution</b> &mdash;
       A single nucleotide in the reference is replaced by another, alternate allele
  <li> <b>deletion</b> &mdash;
       One or more nucleotides is deleted.  The representation in the database is to
       display one additional nucleotide in both the Reference field (Ref) and the
       Alternate Allele field (Alt).  E.g. a variant that is a deletion of an A
       maybe be represented as Ref = GA and Alt = G.
  <li> <b>insertion</b> &mdash;
       One or more nucleotides is inserted.  The representation in the database is to
       display one additional nucleotide in both the Reference field (Ref) and the
       Alternate Allele field (Alt).  E.g. a variant that is an insertion of a T maybe
       be represented as Ref = G and Alt = GT
  <li> <b>delins</b> &mdash;
       Similar to tandemRepeat, in that the runs of Ref and Alt Alleles are of
       different length, except that there is more than one type of nucleotide,
       e.g., Ref = CCAAAAACAAAAACA, Alt = ACAAAAAC.
  <li> <b>multipleNucleotideVariant</b> &mdash;
       More than one nucleotide is substituted by an equal number of different
       nucleotides, e.g.,  Ref = AA, Alt = GC.
  <li> <b>sequence alteration</b> &mdash;
       A parent term meant to signify a deviation from another sequence. Can be
       assigned to variants that have not been characterized yet.
</ul>
<p>
We would like to thank the <a href="https://www.ebi.ac.uk/eva/" target="_blank">European Variation
Archive</a> making this data publically available. We would also like to thank Luis Nassar and Jairo
Navarro for the creation and release of these tracks.
</p>

<a name="042423"></a>
<h2>Apr. 24, 2023 &nbsp;&nbsp; New DGV Gold Standard track for hg38</h2>
<p>
We are pleased to announce the addition of the new
<a href="/cgi-bin/hgTrackUi?db=hg38&g=dgvPlus" target="_blank">DGV Gold Standard track</a>
for hg38. The track displays curated variants from a selected number of studies in the
<a href="http://dgv.tcag.ca/dgv/app/home" target="_blank">Database of Genomic Variants</a> (DGV)
with a criterion that requires a variant to be found in at least two different studies and found in
at least two different samples. More information on this track can be found on the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=dgvPlus#TRACK_HTML"
target="_blank">track description page</a>.</p>
<p>
We would like to thank the Database of Genomic Variants for making these data available. We would
also like to thank Lou Nassar, Anna Benet-Pages, and Gerardo Perez for their efforts on this
release.</p>

<a name="041423"></a>
<h2>Apr. 14, 2023 &nbsp;&nbsp; Problematic Regions for hg38 and a new public hub</h2>
<p>
We would like to announce the release of the <a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=problematic">Problematic Regions</a> composite
track to show problematic/special genomic regions for sequencing or very variable regions for
GRCh38/hg38. This container track helps call out sections of the genome that often cause problems or
confusion when working with the genome. There are three subtracks for now, Anshul Kundaje's ENCODE
Blacklist, GRC (Genome Reference Consortium) Exclusions, and the UCSC Unusual Regions track. 
</p>
<p>
We are interested in adding to the Problematic Regions container, so if you know of any annotations
of hg38 that highlight regions where artifactual alignments are common, regions that are hard to
sequence, please let us know. In addition to these problematic regions tracks, we also have
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=grcIncidentDb">GRC Incidents</a> and the
<a href="/cgi-bin/hgTrackUi?db=hg38&position=default&g=mappability">mappability</a> track available
for hg38 as well.
</p>
<p>
Thanks to Anna Benet-Pag&egrave;s, Max Haeussler, Angie Hinrichs, Daniel Schmelter, and Jairo
Navarro at the UCSC Genome Browser for planning, building, and testing these tracks.
</p>
<hr>
<p>
We would also like to announce our latest public hub: <b>PsychENCODE MPRA</b>! This hub contains
lentiMPRA testing of ~103k sequences for cis-regulatory activity in both primary human cells from
mid-gestation cortex and cerebral organoids, identifying ~47k enhancers and 164 disorder-associated
QTLs that significantly alter enhancer activity.</p>
<p>
See the hub here: <a href="https://bit.ly/PsychENCODEucscGB"
target="_blank">https://bit.ly/PsychENCODEucscGB</a>. We would like to thank Sean Whalen for
creating this hub, as well as Alex Pollen, the Nowakowski lab, Nadav Ahituv, Chengyu Deng, Katie
Pollard, and the PsychENCODE consortium for making this hub possible.</p>

<a name="041023"></a>
<h2>Apr. 10, 2023 &nbsp;&nbsp; New Cross Tissue Nuclei track for hg38</h2>
<p>
We are happy to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=crossTissueMaps" target="_blank"
>Single-Nuclei Cross-Tissue Map</a> track for the human assembly
<a href="/cgi-bin/hgGateway?db=hg38" target="_blank">GRCh38/hg38</a>. This
track collection contains three bar chart tracks of RNA expression. The first track,
<a href="/cgi-bin/hgTrackUi?db=hg38&g=crossTissueMapsTissueCellType" target="_blank"
>Cross Tissue Nuclei</a>,
allows cells to be grouped together and faceted on up to 4 categories: tissue, cell class, cell
subclass, and cell type. The second track,
<a href="/cgi-bin/hgTrackUi?db=hg38&g=crossTissueMapsFullDetails" target="_blank"
>Cross Tissue Details</a>,
allows cells to be grouped together and faceted on up to 7 categories: tissue, cell class, cell
subclass, cell type, granular cell type, sex, and donor. The third track,
<a href="/cgi-bin/hgTrackUi?db=hg38&g=gtexImmuneAtlasFullDetails" target="_blank"
>GTEx Immune Atlas</a>,
allows cells to be grouped together and faceted on up to 5 categories: tissue, cell type, cell
class, sex, and donor.
<p>
This track collection shows data from
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9383269/" target="_blank">
Single-nucleus cross-tissue molecular reference maps toward understanding disease gene function</a>.
The dataset covers ~200,000 single nuclei from a total of 16 human donors across 25 samples,
using 4 different sample preparation protocols followed by droplet based single-cell RNA-seq. The
samples were obtained from frozen tissue as part of the
<a href="https://commonfund.nih.gov/GTEx/index" target="_blank">
Genotype-Tissue Expression (GTEx) project</a>. Samples were taken from the esophagus, skeletal
muscle, heart, lung, prostate, breast, and skin. The dataset includes 43 broad cell classes, some
specific to certain tissues and some shared across all tissue types. Please see the
<a href="https://gtexportal.org/home/singleCellOverviewPage" target="_blank">GTEx portal</a>
for further interactive displays and additional data.</p>
<p>
We would like to thank the GTEx investigators, analysts, and portal team for providing this data.
We would also like to thank Jim Kent, Brittney Wick, Chris Lee, and Gerardo Perez for integrating
the data into the UCSC Genome Browser.
</p>


<a name="032923"></a>
<h2>Mar. 29, 2023 &nbsp;&nbsp; Updated DGV tracks for hg19/hg38</h2>
<p>
We are pleased to announce an update to the DGV tracks for the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=dgvPlus" target="_blank">hg38</a> and
<a href="/cgi-bin/hgTrackUi?db=hg19&g=dgvPlus" target="_blank">hg19</a> human assemblies. The
tracks display copy number variants (CNVs), insertions/deletions (InDels), inversions and inversion
breakpoints annotated by the
<a href="http://dgv.tcag.ca/dgv/app/home" target="_blank">Database of Genomic Variants</a> (DGV),
which contains genomic variations observed in healthy individuals. For the hg38 update, the
Structural Variant Regions track now has 825,343 total variants and the Supporting Structural
Variants track now has 7,270,185 total variants. For the hg19 update, Structural Variant Regions track now has
826,661 total variants and the Supporting Structural Variants track now has 7,346,512 total variants. More
information on these tracks can be found on the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=dgvPlus#TRACK_HTML" target="_blank">track description page</a>.
</p>
<p>
We would like to thank the Database of Genomic Variants for making these data available. We would also
like to thank Jairo Navarro, Max Haeussler, Anna Benet-Pages, and Gerardo Perez for their efforts
on this release.
</p>


<a name="030823"></a>
<h2>Mar. 8, 2023 &nbsp;&nbsp; TOGA Gene Prediction tracks for 41 assemblies</h2>
<p>
We are happy to announce the release of the <b>TOGA</b> (<b>T</b>ool to infer <b>O</b>rthologs from
<b>G</b>enome <b>A</b>lignments) gene prediction tracks for 41 assemblies. TOGA is a homology-based
method that integrates gene annotation, inferring orthologs and classifying genes as intact or lost.
The full list of assemblies containing the TOGA gene prediction tracks is displayed below:
</p>
<p>
<div class="container">
  <div class="row">
  <div class="col-sm-4">
    <ul>
    <li>Brown kiwi <a href="/cgi-bin/hgTrackUi?db=aptMan1&g=HLTOGAannotvGalGal6v1&position=default"
    target="_blank">(aptMan1)</a></li>
    <li>Minke whale <a href="/cgi-bin/hgTrackUi?db=balAcu1&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(balAcu1)</a></li>
    <li>Bison <a href="/cgi-bin/hgTrackUi?db=bisBis1&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(bisBis1)</a></li>
    <li>Cow <a href="/cgi-bin/hgTrackUi?db=bosTau9&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(bosTau9)</a></li>
    <li>Dog <a href="/cgi-bin/hgTrackUi?db=canFam4&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(canFam4)</a></li>
    <li>Dog <a href="/cgi-bin/hgTrackUi?db=canFam5&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(canFam5)</a></li>
    <li>Guinea pig <a href="/cgi-bin/hgTrackUi?db=cavPor3&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(cavPor3)</a></li>
    <li>White rhinoceros <a href="/cgi-bin/hgTrackUi?db=cerSim1&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(cerSim1)</a></li>
    <li>Green monkey <a href="/cgi-bin/hgTrackUi?db=chlSab2&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(chlSab2)</a></li>
    <li>Armadillo <a href="/cgi-bin/hgTrackUi?db=dasNov3&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(dasNov3)</a></li>
    <li>Tenrec <a href="/cgi-bin/hgTrackUi?db=echTel2&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(echTel2)</a></li>
    <li>Southern sea otter <a href="/cgi-bin/hgTrackUi?db=enhLutNer1&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(enhLutNer1)</a></li>
    <li>Horse <a href="/cgi-bin/hgTrackUi?db=equCab3&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(equCab3)</a></li>
    <li>Hedgehog <a href="/cgi-bin/hgTrackUi?db=eriEur2&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(eriEur2)</a></li>
    </ul></div>
  <div class="col-sm-4">
    <ul>
    <li>Cat <a href="/cgi-bin/hgTrackUi?db=felCat9&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(felCat9)</a></li>
    <li>Medium ground finch <a href="/cgi-bin/hgTrackUi?db=geoFor1&g=HLTOGAannotvGalGal6v1&position=default"
    target="_blank">(geoFor1)</a></li>
    <li>Gorilla <a href="/cgi-bin/hgTrackUi?db=gorGor6&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(gorGor6)</a></li>
    <li>Naked mole-rat <a href="/cgi-bin/hgTrackUi?db=hetGla2&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(hetGla2)</a></li>
    <li>Chinese pangolin <a href="/cgi-bin/hgTrackUi?db=manPen1&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(manPen1)</a></li>
    <li>Turkey <a href="/cgi-bin/hgTrackUi?db=melGal5&g=HLTOGAannotvGalGal6v1&position=default"
    target="_blank">(melGal5)</a></li>
    <li>Mouse <a href="/cgi-bin/hgTrackUi?db=mm10&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(mm10)</a></li>
    <li>Mouse <a href="/cgi-bin/hgTrackUi?db=mm39&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(mm39)</a></li>
    <li>Opossum <a href="/cgi-bin/hgTrackUi?db=monDom5&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(monDom5)</a></li>
    <li>Proboscis monkey <a href="/cgi-bin/hgTrackUi?db=nasLar1&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(nasLar1)</a></li>
    <li>Hawaiian monk seal <a href="/cgi-bin/hgTrackUi?db=neoSch1&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(neoSch1)</a></li>
    <li>Pika <a href="/cgi-bin/hgTrackUi?db=ochPri3&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(ochPri3)</a></li>
    <li>Rabbit <a href="/cgi-bin/hgTrackUi?db=oryCun2&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(oryCun2)</a></li>
    <li>Bushbaby <a href="/cgi-bin/hgTrackUi?db=otoGar3&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(otoGar3)</a></li>
    </ul></div>
  <div class="col-sm-4">
    <ul>
    <li>Bonobo <a href="/cgi-bin/hgTrackUi?db=panPan3&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(panPan3)</a></li>
    <li>Chimp <a href="/cgi-bin/hgTrackUi?db=panTro6&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(panTro6)</a></li>
    <li>Orangutan <a href="/cgi-bin/hgTrackUi?db=ponAbe3&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(ponAbe3)</a></li>
    <li>Rhesus <a href="/cgi-bin/hgTrackUi?db=rheMac10&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(rheMac10)</a></li>
    <li>Rat <a href="/cgi-bin/hgTrackUi?db=rn6&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(rn6)</a></li>
    <li>Squirrel monkey <a href="/cgi-bin/hgTrackUi?db=saiBol1&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(saiBol1)</a></li>
    <li>Shrew <a href="/cgi-bin/hgTrackUi?db=sorAra2&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(sorAra2)</a></li>
    <li>Squirrel <a href="/cgi-bin/hgTrackUi?db=speTri2&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(speTri2)</a></li>
    <li>Pig <a href="/cgi-bin/hgTrackUi?db=susScr11&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(susScr11)</a></li>
    <li>Tarsier <a href="/cgi-bin/hgTrackUi?db=tarSyr2&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(tarSyr2)</a></li>
    <li>Manatee <a href="/cgi-bin/hgTrackUi?db=triMan1&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(triMan1)</a></li>
    <li>Tree shrew <a href="/cgi-bin/hgTrackUi?db=tupBel1&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(tupBel1)</a></li>
    <li>Alpaca <a href="/cgi-bin/hgTrackUi?db=vicPac2&g=HLTOGAannotvHg38v1&position=default"
    target="_blank">(vicPac2)</a></li>
    </ul></div>
  </div>
</div>
</p>
<p>
Each annotated transcript is shown in a color-coded classification as:
<ul>
<li><span style='display:inline-block; width:40px; height:15px; background-color:blue;'>&nbsp;</span>
    <span style='color:blue'>"intact"</span>: middle 80% of the CDS
    (coding sequence) is present and exhibits no gene-inactivating mutation.
    These transcripts likely encode functional proteins.</li>
<li><span style='display:inline-block; width:40px; height:15px; background-color:lightblue;'>&nbsp;</span>
    <span style='color:#7193a0'>"partially intact"</span>: 50% of the CDS
     is present in the query and the middle 80% of the CDS exhibits no
     inactivating mutation. These transcripts may also encode functional
     proteins, but the evidence is weaker as parts of the CDS are missing,
     often due to assembly gaps.</li>
<li><span style='display:inline-block; width:40px; height:15px; background-color:grey;'>&nbsp;</span>
    <span style='color:grey'>"missing"</span>: &lt;50% of the CDS is present
     in the query and the middle 80% of the CDS exhibits no inactivating
     mutation.</li>
<li><span style='display:inline-block; width:40px; height:15px; background-color:orange;'>&nbsp;</span>
    <span style='color:orange'>"uncertain loss"</span>: there is 1
     inactivating mutation in the middle 80% of the CDS, but evidence is not
     strong enough to classify the transcript as lost. These transcripts may
     or may not encode a functional protein.</li>
<li><span style='display:inline-block; width:40px; height:15px; background-color:red;'>&nbsp;</span>
    <span style='color:red'>"lost"</span>: typically several inactivating
     mutations are present, thus there is strong evidence that the transcript
     is unlikely to encode a functional protein.</li>
</ul>
</p>
<p>
The <b>TOGA</b> software is available from <a href="https://github.com/hillerlab/TOGA"
target="_blank">github.com/hillerlab/TOGA</a>.
</p>
<p>
We would like to thank the <a href="https://tbg.senckenberg.de/hillerlab/"
target="_blank">Michael Hiller Lab</a> for preparing the data. We would also like to thank Hiram
Clawson, Brian Raney, and Jairo Navarro for the creation and release of these tracks.</p>

<a name="022423"></a>
<h2>Feb. 22, 2023 &nbsp;&nbsp; GENCODE Genes V43 for human (hg38/hg19) and VM32 for mouse (mm39)</h2>
<p>
We are pleased to announce the release of five new <a href="https://www.gencodegenes.org/" target="_blank">
GENCODE Gene</a> tracks corresponding to GENCODE release V43 for human and VM32 for mouse. While all of the
tracks are built from the GENCODE release, they fall into two categories. Two of these tracks, 
<a href="/cgi-bin/hgTrackUi?db=hg38&g=knownGene&c=chrX">GENCODE V43 (hg38)</a> and
<a href="/cgi-bin/hgTrackUi?db=hg38&g=knownGene&c=chrX">GENCODE VM32 (mm39)</a> were built with our
<b>knownGene</b> pipeline and are now the default gene tracks for those assemblies. The knownGene pipeline
builds extensive associations from the annotations and allows us to show additional metadata for each
item as well as link to external resources. The track description pages for these tracks contain options
for configuring the display such as also showing non-coding genes, splice variants, and pseudogenes. 
Different tags and labels may also be toggled.</p>
<p>
The remaining three tracks were each nested within our GENCODE Versions superTrack for each of the three 
assemblies: <a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg19&c=chrX&g=wgEncodeGencodeV43lift37">hg19</a>,
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=wgEncodeGencodeV43">hg38</a>,
and
<a target="_blank" href="/cgi-bin/hgTrackUi?db=mm39&c=chr12&g=wgEncodeGencodeVM32">mm39</a>.
For human, the GENCODE V43 annotations were mapped to hg38 and then back-mapped
to the hg19 assembly. New GENCODE releases now have an
assigned rank for transcripts within the gene. The transcript rank may be used to filter the number
of transcripts displayed in a principled manner. More details about transcript ranking can be found
on the <a href="../cgi-bin/hgTrackUi?db=hg38&position=default&g=wgEncodeGencodeV43#Methods"
target="_blank">track description page</a>. For all three assemblies, the gene sets contain the
following tracks:</p> 

<ul>
  <li>
      <b>Basic</b> - a subset of the <em>Comprehensive set</em>.</li>
  <li>
      <b>Comprehensive</b> - all GENCODE coding and non-coding transcript annotations, including
      polymorphic pseudogenes. This includes both manual and automatic annotations.</li>
  <li>
      <b>Pseudogenes</b> - all annotations except polymorphic pseudogenes.</li>
</ul>
<p>
The hg38 and mm39 assemblies also include the following track:
</p>
<ul>
  <li>
      <b>PolyA</b> - polyA signals and sites manually annotated on the genome based on transcribed
      evidence (ESTs and cDNAs) of 3' end of transcripts containing at least 3 A's not matching the
      genome.</li>
</ul>
<p>
Below is a summary of the contents found in each release. For more details visit the <a target="_blank"
href="https://www.gencodegenes.org/">GENCODE site</a>.</p>
<p>
<table class="stdTbl">
<tr><th COLSPAN=4>GENCODE v43 Release Stats</th></tr>
<tr align=left><th>Genes</th><th>Observed</th><th>Transcripts</th><th>Observed</th></tr>
<tr align=left><td>Protein-coding genes</td><td>19,393</td><td>Protein-coding transcripts</td><td>89,411</td></tr>
<tr align=left><td>Long non-coding RNA genes</td><td>19,928</td><td><font size="-1">- full length protein-coding</font></td><td>64,004</td></tr>
<tr align=left><td>Small non-coding RNA genes</td><td>7,566</td><td><font size="-1">- partial length protein-coding</font></td><td>25,407</td></tr>
<tr align=left><td>Pseudogenes</td><td>14,737</td><td>Nonsense mediated decay transcripts</td><td>21,354</td></tr>
<tr align=left><td>Immunoglobulin/T-cell receptor gene segments</td><td>410</td><td>Long non-coding RNA loci transcripts</td><td>58,023</td></tr>
<tr align=left><td>Total No of distinct translations</td><td>65,519</td><td>Genes that have more than one distinct translations</td><td>13,618</td></tr>
</table><BR>
</p>
<p>
<table class="stdTbl">
<tr><th COLSPAN=4>GENCODE VM32 Release Stats</th></tr>
<tr align=left><th>Genes</th><th>Observed</th><th>Transcripts</th><th>Observed</th></tr>
<tr align=left><td>Protein-coding genes</td><td>21,565</td><td>Protein-coding transcripts</td><td>58,913</td></tr>
<tr align=left><td>Long non-coding RNA genes</td><td>14,834</td><td><font size="-1">- full length protein-coding</font></td><td>45,219</td></tr>
<tr align=left><td>Small non-coding RNA genes</td><td>6,105</td><td><font size="-1">- partial length protein-coding</font></td><td>13,694</td></tr>
<tr align=left><td>Pseudogenes</td><td>13,722</td><td>Nonsense mediated decay transcripts</td><td>7,211</td></tr>
<tr align=left><td>Immunoglobulin/T-cell receptor gene segments</td><td>701</td><td>Long non-coding RNA loci transcripts</td><td>26,421</td></tr>
<tr align=left><td>Total No of distinct translations</td><td>45,163</td><td>Genes that have more than one distinct translations</td><td>10,914</td></tr>
</table><BR>
</p>

<p>
We would like to thank the <a target="_blank"
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Mark Diekhans, Brian Raney, and Lou Nassar for the development and
release of these tracks.</p>

<a name="021323"></a>
<h2>Feb. 13, 2023 &nbsp;&nbsp; New recombination rate tracks for hg38</h2>
<p>
We are pleased to announce the new
<a href="/cgi-bin/hgTrackUi?db=hg38&g=recombRate2" target="_blank">recombination rate</a> tracks
for the <a href="/cgi-bin/hgGateway?db=hg38" target="_blank">GRCh38/hg38</a> genome browser. This track represents
calculated rates of recombination based on the genetic maps from
<a href="https://www.decode.com/" target="_blank">deCODE</a> and
<a href="https://www.internationalgenome.org/about" target="_blank">1000 Genomes</a>. These tracks
are organized in a super track that includes three subtracks with the deCODE recombination rates
(paternal, maternal, and average) and one subtrack with the 1000 Genomes recombination rate, which
was lifted from hg19 and can be used as a drop-in replacement for the
<a href="/cgi-bin/hgTrackUi?db=hg19&g=recombRate" target="_blank">GRCh37/hg19  track</a>. Note that
the deCODE recombination rate data is newer and has a higher resolution. Also, two subtracks that
show additional details provided by deCODE: one with the raw data of all cross-overs tagged with
their proband ID and another one with around 8000 human de-novo mutation variants that are linked to
cross-over changes.
<ul>
  <li>deCODE Genetics, average from paternal and maternal</li>
  <li>deCODE Genetics, paternal</li>
  <li>deCODE Genetics, maternal</li>
  <li>Recombination events in deCODE Genetic Map</li>
  <li>De-novo mutations found in deCODE samples</li>
  <li>1000 Genomes, lifted from hg19</li>
</ul></p>
<p>
More information on these tracks can be found on the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=recombRate2#TRACK_HTML" target="_blank">track description page</a>.
We would like to thank deCODE genetics, the 1000 Genomes Project, and contributors for doing the
research and making their data accessible. Thanks to Po-Ru Loh at the Broad Institute for lifting
and helping import the 1000 Genomes data to hg38. We would also like to thank Maximilian Haeussler
and Gerardo Perez for their work in creating and releasing these tracks. If you have questions,
feedback, or suggestions on this new track, please contact one of our
<a href="../contacts.html" target="_blank">Genome Browser support emails</a>.</p>


<a name="013023"></a>
<h2>Jan. 30, 2023 &nbsp;&nbsp; Addition of GRCh38 patch 14 sequences to hg38</h2>
<p>
We are pleased to announce the addition of GRCh38 patch release 14 to the
<a href="/cgi-bin/hgGateway?db=hg38">hg38</a> assembly. hg38 has been updated with patches
since its release in 2013. The GRC patch releases do not change any previously existing
sequences; they simply add new sequences for fix patches or alternate haplotypes that correspond
to specific regions of the main chromosome sequences. For most users, the patches are unlikely
to make a difference and may complicate the analysis as they introduce more duplication.</p>

<p>
More information on how patch sequences are incorporated can be found on the
<a href="https://genome-blog.gi.ucsc.edu/blog/patches/" target="_blank">Patching up the Genome</a> blog post.
The blog post contains details about the new
<a href="https://hgdownload.gi.ucsc.edu/goldenPath/hg38/bigZips/latest/"
target="_blank">/latest</a>,
<a href="https://hgdownload.gi.ucsc.edu/goldenPath/hg38/bigZips/initial/"
target="_blank">/initial</a>, and
<a href="https://hgdownload.gi.ucsc.edu/goldenPath/hg38/bigZips/p14/" target="_blank">/p14</a>
download directories on the downloads server. With the addition of new sequences to hg38, we can
expect to see BLAT return more matches to the genome.</p>

<ul>
  <li>
    The <b>initial/</b> subdirectory contains files for the initial release of GRCh38, which has 455
    sequences, 261 alternate sequences, and no fix sequences. It is the same as the
    parent download directory. This is probably the best genome file for aligners and most analysis
    tasks, a version called &quot;analysisSet&quot; for the human genome.
  </li>
  <li>
    The <b>p14/</b> subdirectory contains files for GRCh38.p14 (patch release 14), which has 711
    sequences, 351 alternate sequences, and 166  fix sequences.
  </li>
  <li>
    The <b>latest/</b> symbolic link points to the subdirectory for the most recent patch version.
  </li>
</ul>

<p>
We would like to thank the <a href="https://www.ncbi.nlm.nih.gov/grc" target="_blank">Genome
Reference Consortium</a> for creating the patches for hg38. We would also like to thank Galt Barber,
Jairo Navarro, and Gerardo Perez at UCSC for implementing and testing the latest patch to the
hg38 genome.</p>

<!-- ============= 2022 archived news ============= -->
<a name="2022"></a>

<a name="122022"></a>
<h2>Dec 20, 2022 &nbsp;&nbsp; Multiz Alignment &amp; Conservation (470 mammals) for hg38</h2>
<p>
A new <a href="/cgi-bin/hgTrackUi?db=hg38&g=cons470way" target="_blank">470-way Multiz Alignment
&amp; Conservation</a> track has been added to the human <a href="/cgi-bin/hgGateway?db=hg38"
target="_blank">(GRCh38/hg38)</a> genome browser. The composite track displays multiple alignments
(Multiz) and measurements of evolutionary conservation (phastCons and phyloP) for 470 mammals.
</p>
<h3>Tracks available:</h3>
<ul>
    <li>Multiz Alignments of 470 mammals</li> 
    <li>470 mammals Basewise Conservation by PhyloP</li>
    <li>470 mammals Element Conservation by PhastCons</li>
    <li>470 mammals Conserved Elements</li> 
</ul>
<h3>Species</h3>
<p>
Please see the <a href="/cgi-bin/hgTrackUi?db=hg38&g=cons30way" target="_blank">track description
page</a> for assembly details of participating assemblies. The following table describes the gene
tracks used for codon translation:
<table class="stdTbl">
<tr align=left><td><b>Gene Track</b></td><td><b>Species</b></td></tr>
<tr align=left><td>RefSeq Genes</td><td>aardvark, American pika, Amur tiger, Angolan colobus, big brown bat, black flying fox, black snub-nosed monkey, Bolivian squirrel monkey, Brandt's bat, Cape elephant shrew, Cape golden mole, cattle, chimpanzee, Chinese tree shrew, Coquerel's sifaka, degu, dog, domestic cat, domestic guinea pig, drill, European shrew, Florida manatee, golden hamster, gray mouse lemur, green monkey, Hawaiian monk seal, horse, house mouse, house mouse, human, killer whale, lesser Egyptian jerboa, little brown bat, long-tailed chinchilla, Ma's night monkey, minke whale, naked mole-rat, nine-banded armadillo, Northern sea otter, Norway rat, Ord's kangaroo rat, Pacific walrus, Panamanian white-faced capuchin, Philippine tarsier, pig, pig-tailed macaque, polar bear, prairie vole, Przewalski's horse, pygmy chimpanzee, rabbit, Rhesus monkey, small Madagascar hedgehog, small-eared galago, sooty mangabey, southern white rhinoceros, star-nosed mole, Sumatran orangutan, thirteen-lined ground squirrel, Upper Galilee mountains blind mole rat, Vespertilio Davidii, Weddell seal, western European hedgehog, western lowland gorilla, Yangtze River dolphin</td></tr>
<tr align=left><td>Ensembl Genes</td><td>Bos bison bison, Brazilian guinea pig, dog, gray short-tailed opossum, northern tree shrew</td></tr>
<tr align=left><td>Xeno RefGene</td><td>alpaca, black lemur, Chinese pangolin, common bottlenose dolphin, proboscis monkey, Sclater's lemur, Southern sea otter, tammar wallaby</td></tr>
<tr align=left><td>no annotation</td><td>African buffalo, African grass rat, African hunting dog, African hunting dog, African savanna elephant, African woodland thicket rat, Agile Gracile Mouse Opossum, Allen's swamp monkey, Alpine ibex, Alpine marmot, alpine musk deer, American beaver, American black bear, American black bear, American mink, Amur leopard cat, antarctic fur seal, Antarctic minke whale, Antillean ghost-faced bat, aoudad, Arabian camel, Arctic fox, Arctic ground squirrel, argali, Asian black bear, Asian palm civet, Asiatic elephant, Asiatic mouflon, Asiatic tapir, Asiatic tapir, ass, Australian echidna, aye-aye, babakoto, Bactrian camel, banded mongoose, Bank vole, bearded seal, beluga whale, bighorn sheep, bighorn sheep, black muntjac, black rat, black rhinoceros, black-footed cat, black-handed spider monkey, Blue whale, Bohar reedbuck, Bolivian squirrel monkey, Bolivian titi, Bonin flying fox, boutu, bowhead whale, Brazilian free-tailed bat, Brazilian porcupine, Brazilian tapir, brindled gnu, brown lemur, brush rabbit, bush duiker, bushbuck, Cacomistle, cactus mouse, California big-eared bat, California sea lion, Canada lynx, Cantor's roundleaf bat, Cape rock hyrax, capybara, Central European red deer, Chacoan peccary, cheetah, Chinese forest musk deer, Chinese hamster, Chinese pangolin, Chinese rufous horseshoe bat, Chinese water deer, chiru, Clouded leopard, Cobus hunteri, common bottlenose dolphin, common bottlenose dolphin, common brushtail, common pipistrelle, common pipistrelle, common vampire bat, Common vole, common wombat, coppery ringtail possum, Coquerel's mouse lemur, crab-eating macaque, crested porcupine, Cuvier's beaked whale, Damara mole-rat, dassie-rat, Daurian ground squirrel, De Brazza's monkey, desert woodrat, dingo, domestic ferret, domestic yak, donkey, dugong, dwarf mongoose, eastern gray kangaroo, eastern mole, Eastern roe deer, Egyptian rousette, Egyptian spiny mouse, Equus burchelli boehmi, ermine, Eurasian elk, Eurasian red squirrel, Eurasian river otter, Eurasian water vole, European polecat, European rabbit, European woodmouse, evening bat, Fat dormouse, fat sand rat, Fin whale, fossa, franciscana, Francois's langur, Gambian giant pouched rat, gaur, gayal, gelada, gemsbok, gerenuk, giant anteater, giant otter, giant otter, giant panda, giraffe, giraffe, goat, Gobi jerboa, golden ringtail possum, golden snub-nosed monkey, golden spiny mouse, gracile shrew mole, Grant's gazelle, gray seal, gray squirrel, great gerbil, great roundleaf bat, greater bamboo lemur, greater bulldog bat, Greater cane rat, greater horseshoe bat, greater Indian rhinoceros, greater kudu, greater mouse-eared bat, grey whale, grizzly bear, ground cuscus, guanaco, Gunnison's prairie dog, Hanuman langur, harbor porpoise, harbor porpoise, harbor seal, Harvey's duiker, hazel dormouse, Hesperomys crinitus, Himalayan marmot, hippopotamus, hippopotamus, Hispaniolan solenodon, hispid cotton rat, hoary bamboo rat, hoary bat, Hoffmann's two-fingered sloth, Hog deer, hog-nosed bat, Honduran yellow-shouldered bat, humpback whale, Iberian mole, impala, Indian false vampire, Indian flying fox, Indo-pacific bottlenose dolphin, Indo-pacific bottlenose dolphin, Indo-pacific humpbacked dolphin, Indus River dolphin, jaguar, jaguar, jaguarundi, Jamaican fruit-eating bat, Jamaican fruit-eating bat, Japanese macaque, Java mouse-deer, kinkajou, Kirk's dik-dik, klipspringer, koala, Kuhl's pipistrelle, Lama pacos huacaya, large flying fox, Leadbeater's possum, lechwe, leopard, Leschenault's rousette, lesser dawn bat, Lesser dwarf lemur, lesser kudu, Lesser long-nosed bat, lesser mouse-deer, lesser panda, lesser short-nosed fruit bat, lion, little brown bat, llama, llama, long-finned pilot whale, long-tongued fruit bat, Madagascan rousette, Malagasy flying fox, Malagasy straw-colored fruit bat, Malayan pangolin, Malayan pangolin, mandrill, mantled howler monkey, Masai giraffe, Maxwell's duiker, meadow jumping mouse, meerkat, meerkat, melon-headed whale, Miniopterus schreibersii natalensis, Mona monkey, Mongolian gerbil, mongoose lemur, Montane guinea pig, mountain beaver, mountain goat, Mountain hare, mouse lemur, mule deer, muntjak, Murina feae, muskrat, narwhal, Nilgiri tahr, North American badger, North American opossum, North American porcupine, North Atlantic right whale, North Pacific right whale, Northern American river otter, Northern elephant seal, northern fur seal, Northern giant mouse lemur, northern gundi, Northern long-eared myotis, Northern mole vole, northern rock mouse, Northern rufous mouse lemur, northern white rhinoceros, northern white-cheeked gibbon, Norway rat, nutria, okapi, oldfield mouse, olive baboon, pacarana, Pacific pocket mouse, Pacific white-sided dolphin, pale spear-nosed bat, Pallas's mastiff bat, pallid bat, Parnell's mustached bat, Patagonian cavy, Pere David's deer, Peromyscus californicus subsp. insignis, platypus, porcupine caribou, prairie deer mouse, pronghorn, Przewalski's gazelle, puma, punctate agouti, pygmy Bryde's whale, pygmy marmoset, pygmy sperm whale, rabbit, raccoon, ratel, red bat, red fox, red guenon, red kangaroo, Red shanked douc langur, reed vole, Reeves' muntjac, reindeer, Ring-tailed lemur, roan antelope, root vole, royal antelope, Ryukyu mouse, sable, sable antelope, saiga antelope, Schizostoma hirsutum, Schreibers' long-fingered bat, scimitar-horned oryx, Sclater's lemur, Seba's short-tailed bat, sheep, short-tailed field vole, shrew mouse, Siberian ibex, Siberian musk deer, silvery gibbon, slow loris, snow sheep, snowshoe hare, social tuco-tuco, South African ground squirrel, Southern elephant seal, southern grasshopper mouse, southern multimammate mouse, southern tamandua, Southern three-banded armadillo, southern two-toed sloth, southern two-toed sloth, Sowerby's beaked whale, Spanish lynx, sperm whale, sperm whale, spotted hyena, springbok, springhare, steenbok, Steller sea lion, Steller's sea cow, Stephens's kangaroo rat, steppe mouse, straw-colored fruit bat, stripe-headed round-eared bat, striped hyena, Sumatran rhinoceros, Sunda flying lemur, suni, tailed tailless bat, Talazac's shrew tenrec, tamarin, tammar wallaby, Tasmanian devil, Tasmanian wolf, Thomson's gazelle, topi, Transcaucasian mole vole, Tree pangolin, Tree pangolin, tufted capuchin, Ugandan red Colobus, Vancouver Island marmot, vaquita, Vicugna mensalis, walrus, water buffalo, waterbuck, western gray kangaroo, Western ringtail oppossum, western spotted skunk, western wild mouse, white-faced saki, white-footed mouse, white-fronted capuchin, white-lipped deer, White-nosed coati, white-tailed deer, white-tailed deer, white-tailed deer, white-tufted-ear marmoset, Wild Bactrian camel, wild goat, wild yak, wolverine, woodchuck, woodchuck, woodland dormouse, Yangtze finless porpoise, Yarkand deer, yellow-bellied marmot, yellow-footed antechinus, yellow-spotted hyrax, zebu cattle,
</td></tr>
</table>
</p>

<p>
We would like to thank the <a href="https://tbg.senckenberg.de/hillerlab/" target="_blank">Michael
Hiller lab</a> for providing the data. We would also like to thank Hiram Clawson and Jairo Navarro
for the creation and release of these tracks.</p>

<a name="121322"></a>
<h2>Dec. 13, 2022 &nbsp;&nbsp; New GENCODE gene tracks: Human V42 (hg19/hg38) - Mouse VM31
(mm39)</h2>
<p>
We are pleased to announce new GENCODE Gene annotation tracks, which correspond
to <a href="https://www.ensembl.info/2022/10/20/ensembl-108-has-been-released/"
target="_blank">Ensembl 108</a>, for three assemblies: <a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg19&c=chrX&g=wgEncodeGencodeV42lift37">hg19/GRCh37</a>,
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=wgEncodeGencodeV42">hg38/GRCh38</a>,
and
<a target="_blank" href="/cgi-bin/hgTrackUi?db=mm39&c=chr12&g=wgEncodeGencodeVM31">mm39/GRCm39</a>.
For human, the GENCODE V42 annotations were mapped to hg38/GRCh38 and then back-mapped
to the hg19/GRCh37 assembly. The hg38 GENCODE V42 and mm39 GENCODE VM31 releases now have an
assigned rank for transcripts within the gene. The transcript rank may be used to filter the number
of transcripts displayed in a principled manner. More details about transcript ranking can be found
on the <a href="../cgi-bin/hgTrackUi?db=hg38&position=default&g=wgEncodeGencodeV42#Methods"
target="_blank">track description page</a>. For all three assemblies, the gene sets contain the
following tracks:</p>
<ul>
  <li>
      Basic - a subset of the <em>Comprehensive set</em>.</li>
  <li>
      Comprehensive - all GENCODE coding and non-coding transcript annotations, including
      polymorphic pseudogenes. This includes both manual and automatic annotations.</li>
  <li>
      Pseudogenes - all annotations except polymorphic pseudogenes.</li>
</ul>
<p>
The hg38 and mm39 assemblies also include the following track that is not available on hg19:
</p>
<ul>
  <li>
      PolyA - polyA signals and sites manually annotated on the genome based on transcribed
      evidence (ESTs and cDNAs) of 3' end of transcripts containing at least 3 A's not matching the
      genome.</li>
</ul>
<p>
The 2-way Pseudogenes track has been dropped in these releases. Details on each release can be found on the <a target="_blank"
href="https://www.gencodegenes.org/">GENCODE site</a>. This includes <a target="_blank"
href="https://www.gencodegenes.org/human/stats.html">statistics on each release</a>.
</p>
<p>
We would like to thank the <a target="_blank"
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Mark Diekhans and Gerardo Perez for the development and
release of these tracks.</p>

<a name="111822"></a>
<h2>Nov. 18, 2022 &nbsp;&nbsp; New enhanced Genome Browser search</h2>
<p>
We are pleased to announce our enhanced Genome Browser search, which now includes results in our
help docs and public hubs. This new search can be used in the search bar of our
<a href="../cgi-bin/hgGateway" target="_blank">gateway page</a> or the
<a href="../cgi-bin/hgTracks" target="_blank">Genome Browser tracks display page</a>. The search
now finds matches to words in track data, track names, track descriptions, public hub track names,
and public hub descriptions within the current assembly. In addition, it will also find matches in
any of our documentation. The results page also includes a collapsible tree of the search results,
which shows the types of categories for the given search results and the number of matches
available in each category. The search results will belong to one of 5 categories:</p>
<ul>
  <li>UCSC Genes/GENCODE Genes</li>
  <li>Assembly Track Data Items</li>
  <li>Assembly Track Names/Descriptions</li>
  <li>Public Hubs</li>
  <li>Help Pages</li>
</ul>
<p>
Specifically, the matches to knownGene terms will have a short description with the knownGene term
in bold. All categories will show the first 10 hits and can expand up to 500 hits per category. The
checkboxes in the tree can be used to hide or show all the matches belonging to a category.</p>
<div class="text-center">
  <a href="../cgi-bin/hgSearch?search=TP53&db=hg38" target="_blank">
  <img src="../images/newsArchImages/hgSearch_TP53.png" alt="TP53 search" width="1000"></a>
</div>
<p>
Entering positional coordinates, such as chr2:135,820,191-135,823,087, 2:135820191-135823087,
NC_000002.12:135820191-135823087, or CM000664.2:135820191-135823087, will directly navigate to the
tracks display page at that position. Search terms that have a single result, such as the HGVS
expressions listed in our <a href="/goldenPath/help/query.html"
target="_blank">Querying the Genome Browser page</a>, will automatically redirect you from the
results page to the Genome Browser tracks display page. For more information, see the
<a href="../cgi-bin/hgSearch" target="_blank">Using the Genome Browser Search section</a> at the
bottom of the search results page.</p>
<p>
We would like to thank Christopher Lee, Max Haeussler, Daniel Schmelter, Lou Nassar, and Gerardo
Perez for their work on this release.</p>

<a name="103122"></a>
<h2>Oct. 31, 2022 &nbsp;&nbsp; UK Biobank Depletion rank score for human</h2>
<p>
We are happy to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=ukbDepletion">UK Biobank Depletion rank score</a> track
for the human assembly (<a href="/cgi-bin/hgGateway?db=hg38">GRCh38/hg38</a>). This track is part of
the Constraint score container track that includes several subtracks showing the results of
constraint prediction algorithms.</p>
<p>
A variant depletion score was computed for every overlapping set of 500bp windows in the genome
with a 50bp step size, and assigned a rank (depletion rank (DR)) from 0 (most depletion) to 100
(least depletion) for each 500bp window. Since the windows are overlapping, only the value in the
central 50bp of the 500bp window is plotted. You can learn more about how the data was generated
from the <a href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=ukbDepletion">track description</a> page.</p>
<p>
We would like to thank the UK Biobank / deCODE Genetics team for providing the data for the track.
We would also like to thank Max Haeussler and Jairo Navarro at UCSC for the creation and release of
this track.</p>

<a name="102422"></a>
<h2>Oct. 24, 2022 &nbsp;&nbsp; Two new data tracks on human: HGNC and PanelApp</h2>
<p>
The UCSC Genome Browser team is happy to announce two new fantastic data tracks 
on the human reference genomes, hg19 and hg38. We hope these additions will 
improve disease diagnostic capacity and harmonize different gene naming systems.
</p><p>
The PanelApp track is a Phenotype and Literature track from <a 
href="https://panelapp.genomicsengland.co.uk" target="_blank">Genomics England</a>
which shows expert, crowdsourced diagnostic disease panels among genes, 
copy-number variants (CNV), and short tandem repeats (STR). This collection 
of nearly 50,000 associations includes a confidence level color-score,
detailed mouseover, inheritance patterns, and links to the primary data 
source. We are proud to say that this evolving resource will be kept 
up-to-date with automatic weekly data updates. This is a much-requested 
data track and we hope it sees some attention!
</p><p>
The HGNC track is a Genes track from the <a href="https://www.genenames.org/" 
target="_blank">Hugo Gene Nomenclature Committee</a>, 
the official source for gene names. This searchable track is a unifier 
of different identifiers for the same gene, with up to 24 aliases 
linked together under one annotation. Clicking into an entry will show data 
on current names, former names, and full names; you name it! Actually don't 
name it, because it's already been named. We have also added this thesaurus 
archive to our search engine to help researchers find the gene they're looking for, 
regardless of information source and version.
</p><p>
Try PanelApp and HGNC yourself here:</p>
<p><a href="https://genome.ucsc.edu/s/view/PanelApp.HGNC">
https://genome.ucsc.edu/s/view/PanelApp.HGNC</a><br>
<a href="https://genome.ucsc.edu/s/view/PanelApp.HGNC">
<img src="../../images/panelApp.HGNC.png" alt="PanelApp and HGNC tracks image" width="700">
</a></p>

<a name="101922"></a>
<h2>Oct. 19, 2022 &nbsp;&nbsp; Two new curated assemblies: hs1 and mpxvRivers</h2>
<p>
We are happy to announce the release of two new curated assembly hubs for human
<a href="/cgi-bin/hgGateway?db=hs1">(T2T/hs1)</a> and the Monkeypox virus
<a href="/cgi-bin/hgGateway?db=mpxvRivers">(MT903340.1/mpxvRivers)</a>. These two assemblies
showcase the new method of how we will be releasing new assemblies on the UCSC Genome Browser.
Curated assembly hubs will be reviewed by our quality assurance team like previous native assemblies
which will allow for quicker and streamlined releases. For the majority of users, there will be no
obvious difference between the curated hubs and native assemblies. 
</p>
<p>
Please note, the hs1 assembly is identical to the T2T Genark assembly hub. Unfortunately, with the
database name change and the change in sequence names, users will need to reattach any
custom tracks or hubs that were previously on the T2T assembly hub.
</p>
<p>
<b>hs1 statistics</b>:</p>
<ul>
  <li>Common name: human</li>
  <li>Taxonomic name: Homo sapiens, taxonomy ID:
      <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=9606"
      target="_blank">9606</a></li>
  <li>Sequencing/Assembly provider ID: T2T Consortium</li>
  <li>Assembly date: 24 Jan 2022</li>
  <li>Assembly type: haploid</li>
  <li>Assembly level: Complete Genome</li>
  <li>Biosample: (n/a)</li>
  <li>Assembly accession ID:
      <a href="https://www.ncbi.nlm.nih.gov/assembly/GCA_009914755.4"
      target="_blank">GCA_009914755.4</a></li>
  <li>Assembly FTP location:
      <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/009/914/755/GCA_009914755.4_T2T-CHM13v2.0"
      target="_blank">GCA/009/914/755/GCA_009914755.4_T2T-CHM13v2.0</a></li>
  <li>Total assembly nucleotides: 3,117,292,070</li>
  <li>Assembly contig count: 25</li>
  <li>N50 size: 150,617,247</li>
</ul>
<p>
<b>mpxvRivers statistics</b>:</p>
<ul>
  <li>Common name: monkeypox
  <li>Taxonomic name: Monkeypox virus, taxonomy ID:
      <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=10244"
      target="_blank">10244</a></li>
  <li>Sequencing/Assembly provider ID: na</li>
  <li>Assembly date: 30 May 2022</li>
  <li>Assembly type: na</li>
  <li>Assembly level: Complete Genome</li>
  <li>Biosample: (n/a)</li>
  <li>Assembly accession ID: <a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_014621545.1"
      target="_blank">GCF_014621545.1</a></li>
  <li>Assembly FTP location:
      <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/014/621/545/GCF_014621545.1_ASM1462154v1"
      target="_blank">GCF/014/621/545/GCF_014621545.1_ASM1462154v1</a></li>
  <li>Total assembly nucleotides: 197,209</li>
  <li>Assembly contig count: 1</li>
  <li>N50 size: 197,209</li>
</ul>
<p>
We would like to thank Brian Raney and Jairo Navarro for the creation and release of these
assemblies.</p>

<a name="090522"></a>
<h2>Sep. 5, 2022 &nbsp;&nbsp; Using the Genome Browser in your Teaching</h2>
<p>
Today we are announcing a new education module in the UCSC Genome Browser training pages.
</p>

<p>
In response to requests from Browser users, the new module is designed to assist 
teachers and students in understanding selected topics in Molecular Biology, 
Genetics, Medicine, Population Biology and Evolution.  Using the Genome Browser 
as a display device, the new module is organized as short stories written by 
undergraduates for undergraduates.  Supported by numerous links to live Browser 
sessions, the stories tell the tales and show some of the capabilities of the 
Genome Browser.
</p>

<p>
Find the stories in the new
</p>

<p>
&nbsp; &nbsp; &nbsp;
   <a href = "../training/education/">
   education pages</a>
</p>

<p>
We thank Zo&#235; Shmidt, Mateo Etchevesta, Cally Lin, Arturo Marquez, Ana Gallego 
and Robert Kuhn for their contributions.
</p>

<a name="083122"></a>
<h2>Aug. 31, 2022 &nbsp;&nbsp; New Illumina Array tracks for hg19 and hg38</h2>
<p>
We are pleased to announce the addition of new
<a href="https://www.illumina.com/" target="_blank">Illumina</a> Array tracks for the
<a href="../cgi-bin/hgTrackUi?db=hg38&g=genotypeArrays" target="_blank">GRCh38/hg38</a> and
<a href="../cgi-bin/hgTrackUi?db=hg19&g=genotypeArrays" target="_blank">GRCh37/hg19</a>  human
assemblies. These tracks are contained in the Array Probsets composite track.
<ul>
  <li><b>Illumina 450k</b> - Illumina 450k Methylation Array</li>
  <li><b>Illumina 850k</b> - Illumina 850k EPIC Methylation Array</li>
  <li><b>CytoSNP 850k</b> - Illumina 850k CytoSNP Array</li>
</ul></p>
<p>
The Illumina 450k and 850k tracks quantitatively interrogate methylation sites across the genome at
single-nucleotide resolution. The Illumina 850k CytoSNP track contains approximately 850,000
empirically selected SNPs spanning the entire genome with enriched coverage for 3,262 genes of known
cytogenetics relevance in both constitutional and cancer applications.</p>
<p>
We would like to thank Illumina for providing the data. We would also like to thank Jairo Navarro,
Brooke Rhead, and Gerardo Perez for their efforts on this release.</p>

<a name="082222"></a>
<h2>Aug. 22, 2022 &nbsp;&nbsp; dbSNP build 155 tracks for hg19 and hg38</h2>
<p>
We are pleased to announce the release of
<a href="https://www.ncbi.nlm.nih.gov/SNP/" target="_blank">NCBI's dbSNP</a> build 155 data for
<a href="../cgi-bin/hgTrackUi?db=hg38&g=dbSnp155Composite" target="_blank">GRCh38/hg38</a> and
<a href="../cgi-bin/hgTrackUi?db=hg19&g=dbSnp155Composite" target="_blank">GRCh37/hg19</a>  human
assemblies. The release reaches a new milestone of over 1 billion RefSNP (rs) records. For hg38
(GRCh38), approximately 998 million distinct variants (RefSNP clusters with rs# ids) have been
mapped to more than 1.06 billion genomic locations including alternate haplotype and fix patch
sequences. dbSNP remapped variants from hg38 to hg19 (GRCh37); approximately 981 million distinct
variants were mapped to more than 1.02 billion genomic locations including alternate haplotype and
fix patch sequences (not all of which are included in UCSC's hg19).</p>

<p>
dbSNP build 155 is composed of 5 subtracks:
<ul>
  <li><b>Common dbSNP(155)</b> - Common (1000 Genomes Phase 3 MAF >= 1%) Short Genetic Variants</li>
  <li><b> ClinVar dbSNP(155)</b> - Short Genetic Variants in ClinVar</li>
  <li><b>Mult. dbSNP(155)</b> - Short Genetic Variants that map to multiple genomic loci</li>
  <li><b>All dbSNP(155)</b> - All Short Genetic Variants</li>
  <li><b>Map Err dbSnp(155)</b> - Mappings with Inconsistent Coordinates</li>
</ul></p>

<h3>Data Access</h3>
<p>
The complete data can be found across two separate directories in our download server, a bigBed file
(bigDbSnp) for <a target="_blank" href="http://hgdownload.gi.ucsc.edu/gbdb/hg19/snp/">hg19</a>
and <a target="_blank" href="http://hgdownload.gi.ucsc.edu/gbdb/hg38/snp/">hg38</a>, and a
<a target="_blank" href="http://hgdownload.gi.ucsc.edu/gbdb/hgFixed/dbSnp/">shared secondary
details file</a> which holds additional variant details.<p>
</p>

<p>
We would like to thank the dbSNP group at NCBI for providing access to these data. We would also
like to thank Galt Barber, Angie Hinrichs, and Gerardo Perez for their efforts on this release.
</p>


<a name="080222"></a>
<h2>Aug. 08, 2022 &nbsp;&nbsp; New GENCODE gene tracks: Human V41 (hg19/hg38) - Mouse VM30
(mm39)</h2>
<p>
We are pleased to announce new GENCODE Gene annotation tracks, which correspond
to <a href="https://www.ensembl.info/2022/05/31/whats-coming-in-ensembl-release-107-ensembl-genomes-54/"
target="_blank">Ensembl 107</a>, for three assemblies: <a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg19&c=chrX&g=wgEncodeGencodeV41lift37">hg19/GRCh37</a>,
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=wgEncodeGencodeV41">hg38/GRCh38</a>,
and
<a target="_blank" href="/cgi-bin/hgTrackUi?db=mm39&c=chr12&g=wgEncodeGencodeVM30">mm39/GRCm39</a>.
For human, the GENCODE V41 annotations were mapped to hg38/GRCh38 and then back-mapped
to the hg19/GRCh37 assembly. For all three assemblies, the gene sets contain the following
tracks:</p>
<ul>
  <li>
      Basic - a subset of the <em>Comprehensive set</em>.</li>
  <li>
      Comprehensive - all GENCODE coding and non-coding transcript annotations, including
      polymorphic pseudogenes. This includes both manual and automatic annotations.</li>
  <li>
      Pseudogenes - all annotations except polymorphic pseudogenes.</li>
</ul>
<p>
The hg38 and mm39 assemblies also include the following tracks that are not available on hg19:
</p>
<ul>
  <li>
      2-way Pseudogenes - pseudogenes predicted by both the Yale Pseudopipe and UCSC
      Retrofinder pipelines.</li>
  <li>
      PolyA - polyA signals and sites manually annotated on the genome based on transcribed
      evidence (ESTs and cDNAs) of 3' end of transcripts containing at least 3 A's not matching the
      genome.</li>
</ul>
<p>
Details on each release can be found on the <a target="_blank"
href="https://www.gencodegenes.org/">GENCODE site</a>. This includes <a target="_blank"
href="https://www.gencodegenes.org/human/stats.html">statistics on each release</a>.
</p>
<p>
We would like to thank the <a target="_blank"
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Mark Diekhans and Jairo Navarro for the development and
release of these tracks.</p>

<a name="071222"></a>
<h2>July 12, 2022 &nbsp;&nbsp; GENCODE V41 and VM30 now available for human and mouse</h2>
<p>
We are happy to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=knownGene">GENCODE V41</a> track for human
(GRCh38/hg38) and the
<a href="/cgi-bin/hgTrackUi?db=mm39&c=chr12&g=knownGene">GENCODE VM30</a> track for mouse
(GRCm39/mm39). These tracks includes protein-coding genes, non-coding RNA genes, and pseudo-genes,
though pseudo-genes are not displayed by default. Gene annotations are available on the reference
chromosomes, assembly patches, and alternative loci (haplotypes).
</p>
<p>
Details on each release can be found on the <a target="_blank"
href="https://www.gencodegenes.org/">GENCODE website</a>. This includes statistics on the
<a target="_blank" href="https://www.gencodegenes.org/human/stats_41.html">human</a> and
<a target="_blank" href="https://www.gencodegenes.org/mouse/stats_M30.html">mouse</a> release.
<p>
We would like to thank the <a target="_blank"
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Mark Diekhans, Brian Raney, and Jairo Navarro for the
development and release of these tracks.</p>

<a name="063022"></a>
<h2>June 30, 2022 &nbsp;&nbsp; New gnomAD Constraint Metrics track on GRCh38/hg38</h2>
<p>
We are pleased to announce the <a href="/cgi-bin/hgTrackUi?db=hg38&g=gnomadPLI">Genome
Aggregation Database (gnomAD) Constraint Metrics</a> track, previously only on hg19,
is now also available on the human GRCh38/hg38 genome assembly.</p>
<p>
These data show various metrics of pathogenicity per-gene as predicted for gnomAD v2.1.1 and
identifies genes subject to strong selection against various classes of mutation. It is comprised
of four subtracks:</p>
<ul>
  <li><b>Gene LoF</b> &ndash; gnomAD Predicted Loss of Function Constraint Metrics By Gene (pLI)
  <li><b>Gene Missense</b> &ndash; gnomAD Predicted Missense Constraint Metrics By Gene (Z-scores)
  <li><b>Transcript LoF</b> &ndash; gnomAD Predicted Loss of Function Constraint Metrics By Transcript (pLI)
  <li><b>Transcript Missense</b> &ndash; gnomAD Predicted Missense Constraint Metrics By Transcript (Z-scores)
</ul>
<p>
More information on this track can be found in the <a href="/cgi-bin/hgTrackUi?db=hg38&g=gnomadPLI#TRACK_HTML"
>track description page</a>, as well as the <a href="https://gnomad.broadinstitute.org/faq" target="_blank">
gnomAD</a> site. We would like to thank the <a href="https://gnomad.broadinstitute.org/" target="_blank">
Genome Aggregation Database Consortium</a> for making these data available. We would also like to thank Lou
Nassar, Maximilian Haeussler, Anna Benet-Pages, and Gerardo Perez for their work in creating and releasing these
tracks.</p>

<a name="062822"></a>
<h2>June 28, 2022 &nbsp;&nbsp; Update to GTEx cis-eQTL for hg38</h2>
<p>
We are happy to announce an update to the
<a href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=gtexEqtlHighConf">GTEx cis-eQTL</a> track for the human
assembly (GRCh38/hg38). The track shows genetic variants likely affecting proximal gene expression
in 49 human tissues from the <a href="https://gtexportal.org/home/"
target="_blank">Genotype-Tissue Expression (GTEx)</a> V8 data release. With this update, we are
introducing the <a href="https://github.com/xqwen/dap" target="_blank">DAP-G</a> 95% credible set
alongside the <a href="http://genetics.cs.ucla.edu/caviar/" target="_blank">CAVIAR</a>
high-confidence set.</p>
<p>
Hovering over items in the track display will show the variant ID (often a dbSNP rsID), the target
gene, tissue, and posterior probablity (Causal Posterior Probability (CPP) for CAVIAR; SNP Posterior
Inclusion Probability (PIP) for DAP-G). Clicking an item will show the details of that interaction
with link outs to view more details on the GTEx website.</p>
<p>
Track configuration supports filtering by tissue, gene, or posterior probability.</p>
<p>
We would like to thank the GTEx investigators, analysts, and portal team for providing this data.
We also want to thank Matt Speir, Max Haeussler, and Jairo Navarro for the creation and release of
these tracks.</p>

<a name="062722"></a>
<h2>June 27, 2022 &nbsp;&nbsp; New dynseq &quot;logo on&quot; display for bigWigs</h2>
<p>
There is a new <a href="help/bigWig.html#Ex4" target="_blank"><code>logo=on</code></a>
setting for bigWig custom tracks that allows dynamically displaying
sequence at the base level. This <a href="https://kundajelab.github.io/dynseq-pages/"
target="_blank">dynseq</a> display scales nucleotide characters by user-specified,
base-resolution scores. The dynseq feature was developed by the <a target="_blank"
href="https://kundajelab.github.io/about/">Kundaje Lab</a>.</p>
<div class="text-center">
 <a href="help/bigWig.html#Ex4">
 <img src="../../images/dynseq.png" alt="Example4 of dynseq logo=on setting" width="700">
 </a>
</div>
<p>
See <a href="help/bigWig.html#Ex4" target="_blank">an example</a> on the bigWig help page
or read about the <a href="help/trackDb/trackDbHub.html#logo"
target="_blank"><code>logo on</code></a> feature for Track Hubs.</p>

<a name="060322"></a>
<h2>June 3, 2022 &nbsp;&nbsp; New clinical data track on hg19/hg38: GenCC</h2>
<p>
We are happy to add the GenCC clinical track to the human genome assemblies hg19(GRCh37) 
and hg38(GRCh38). This data set is from the <a target="_blank" href="https://thegencc.org/">
Gene Curation Coalition</a> and contains a list of gene-disease relationships and data.
This global coalition harmonizes gene-level resources with disease associations.  
Visually, the browser tracks are filterable and color classified by evidence validity, from red to green.
You can find this track listed under the Phenotype and Literature group. Click on the session
below to see GenCC alongside select clinical tracks. </p>
<ul>
<li><a href="https://genome.ucsc.edu/s/view/GenCC.selectedClinical">
https://genome.ucsc.edu/s/view/GenCC.selectedClinical</a></li>
</ul>
<p>
You can read more and visualize this data youself on the Genome Browser from the link below:</p>
<ul>
<li><a href="../../cgi-bin/hgTrackUi?db=hg38&g=genCC">GenCC on hg38</a></li>
<li><a href="../../cgi-bin/hgTrackUi?db=hg19&g=genCC">GenCC on hg19</a></li>
</ul>



<a name="052422"></a>
<h2>May 24, 2022 &nbsp;&nbsp; New Assembly Request Page</h2>
<p>
Find the genome of your favorite organism on our new <a href="/assemblyRequest.html"
target="_blank">assembly request</a> page. Click &quot;view&quot; to launch a browser,
or the &quot;request&quot; button to request that we add yours.</p>
<div class="text-center">
  <a href="/assemblyRequest.html">
   <img src="../images/newsArchImages/requestPage.png" alt="New Assembly Request Page" width="700">
  </a>
</div>
<p>Our most recent blog post,
<a href="https://genome-blog.gi.ucsc.edu/blog/2022/04/12/genark-hubs-part-4/"
target="_blank">GenArk Hubs Part 4 - New assembly request page</a> describes how to use this new
feature.</p>
<p>
With the realization that we could not possibly host every single genomic assembly available, we created
<a href="help/hgTrackHubHelp.html#Assembly" target="_blank">Assembly Hubs</a>. Assembly Hubs allow
anyone to host and view the genome of any species in the context of the UCSC Genome Browser.
Last summer we <a href="newsarch.html#060121">announced</a> the creation of the Genome Repository
(GenArk) which included Assembly Hubs for the genomic assemblies of more than 1,300 species.
Since then, we have taken this a step further by creating this new <a href="/assemblyRequest.html"
target="_blank">Assembly Request</a> page that displays about 14,000 assemblies: those that
are available for viewing and those that can be requested. We currently have nearly 1,700 assembly
hubs available for viewing at the click of a button. Upon first visiting the page shows only
<b>the first 500 assemblies</b>. Use the <em>select assembly type to display</em> button to
&quot;show all&quot; assemblies. If you don't see the organism or assembly you
are interested in then please use the page to request it. We will build an Assembly Hub for you
and let you know when it is available for viewing.</p>

<a name="051122"></a>
<h2>May 11, 2022 &nbsp;&nbsp; New GENCODE gene tracks: Human V40 (hg19/hg38) - Mouse M29
(mm39)</h2>
<p>
We are pleased to announce new GENCODE Gene annotation tracks, which correspond
to <a href="https://www.ensembl.info/2022/04/12/ensembl-106-has-been-released/"
target="_blank">Ensembl 106</a>, for three assemblies: <a target="_blank"
href="../../cgi-bin/hgTrackUi?db=hg19&c=chr21&g=wgEncodeGencodeV40lift37">hg19/GRCh37</a>,
<a target="_blank" href="../../cgi-bin/hgTrackUi?db=hg38&c=chr1&g=wgEncodeGencodeV40">hg38/GRCh38</a>,
and <a target="_blank" href="../../cgi-bin/hgTrackUi?db=mm39&c=chr12&g=wgEncodeGencodeVM29">mm39/GRCm39</a>.
For human, the GENCODE V40 annotations were mapped to hg38/GRCh38 and then back-mapped
to the hg19/GRCh37 assembly. For all three assemblies, the gene sets contain the following
tracks:</p>
<ul>
  <li>Basic - a subset of the <em>Comprehensive set</em>.</li>
  <li>Comprehensive - all GENCODE coding and non-coding transcript annotations, including
      polymorphic pseudogenes. This includes both manual and automatic annotations.</li>
  <li>Pseudogenes - all annotations except polymorphic pseudogenes.</li></ul>
<p>
The hg38 and mm39 assemblies also include the following tracks that are not available on hg19:
</p>
<ul>
  <li>2-way Pseudogenes - pseudogenes predicted by both the Yale Pseudopipe and UCSC
      Retrofinder pipelines.</li>
  <li>PolyA - polyA signals and sites manually annotated on the genome based on transcribed
      evidence (ESTs and cDNAs) of 3' end of transcripts containing at least 3 A's not matching the
      genome.</li></ul>
<p>
Details on each release can be found on the <a target="_blank"
href="https://www.gencodegenes.org/">GENCODE site</a>. This includes <a target="_blank"
href="https://www.gencodegenes.org/human/stats.html">statistics on each release</a>.
<p>We would like to thank the <a target="_blank"
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Mark Diekhans and Lou Nassar for the development and
release of these tracks.</p>

<a name="050522"></a>
<h2>May 5, 2022 &nbsp;&nbsp; Merged Cell Expression on hg38</h2>
<p>
The Genome Browser already provided <a href="#111721">single-cell RNA-seq datasets</a> for the
human GRCh38/hg38 assembly, but those data have so far been split among a collection of tracks
depending on the organ and publication source. We are happy to announce that data from 12 of
those papers (and 14 organs) are now available in a combined
<a href="../cgi-bin/hgTrackUi?db=hg38&g=singleCellMerged">Merged Cells</a> track that provides
normalized RNA-seq values for every cell type in those sets. All components were normalized
to show expression in parts per million.</p>
<figure class="text-center">
<img src="../images/singleCell/singleCellMerged.png" style="width: 80%; max-width: 1318px;"
alt="Example of the Merged Cells track display for the ACE2 gene">
</figure>
<p>
The following tracks were incorporated into this Merged Cells track:
<ul>
  <li>
    <a target="_blank"
    href="/cgi-bin/hgTrackUi?db=hg38&g=bloodHao&position=default">Blood (PBMC) Hao</a>
  </li>
  <li>
    <a target="_blank"
    href="/cgi-bin/hgTrackUi?db=hg38&g=colonWang&position=default">Colon Wang</a> 
  </li>
  <li>
    <a target="_blank"
    href="/cgi-bin/hgTrackUi?db=hg38&g=cortexVelmeshev&position=default">Cortex Velmeshev</a> 
  </li>
  <li>
    <a target="_blank"
    href="/cgi-bin/hgTrackUi?db=hg38&g=fetalGeneAtlas&position=default">Fetal Gene Atlas</a> 
  </li>
  <li>
    <a target="_blank"
    href="/cgi-bin/hgTrackUi?db=hg38&g=heartCellAtlas&position=default">Heart Cell Atlas</a> 
  </li>
  <li>
    <a target="_blank"
    href="/cgi-bin/hgTrackUi?db=hg38&g=ileumWang&position=default">Ileum Wang</a>
  </li>
  <li>
    <a target="_blank"
    href="/cgi-bin/hgTrackUi?db=hg38&g=kidneyStewart&position=default">Kidney Stewart</a>
  </li>
  <li>
    <a target="_blank"
    href="/cgi-bin/hgTrackUi?db=hg38&g=liverMacParland&position=default">Liver MacParland</a>
  </li>
  <li>
    <a target="_blank"
    href="/cgi-bin/hgTrackUi?db=hg38&g=lungTravaglini2020&position=default">Lung Travaglini</a>
  </li>
  <li>
    <a target="_blank"
    href="/cgi-bin/hgTrackUi?db=hg38&g=muscleDeMicheli&position=default">Muscle De Micheli</a>
  </li>
  <li>
    <a target="_blank"
    href="/cgi-bin/hgTrackUi?db=hg38&g=pancreasBaron&position=default">Pancreas Baron</a>
  </li>
  <li>
    <a target="_blank"
    href="/cgi-bin/hgTrackUi?db=hg38&g=placentaVentoTormo&position=default">Placenta Vento-Tormo</a>
  </li>
  <li>
    <a target="_blank"
    href="/cgi-bin/hgTrackUi?db=hg38&g=rectumWang&position=default">Rectum Wang</a>
  </li>
  <li>
    <a target="_blank"
    href="/cgi-bin/hgTrackUi?db=hg38&g=skinSoleBoldo&position=default">Skin Sole-Boldo</a>
  </li>
</ul>
<p>
We hope that this combined format will facilitate research and comparison with the recently
released <a href="#033122">Tabula Sapiens</a> expression tracks. We would like to thank Jim Kent,
Brittney Wick, Jonathan Casper, and Jairo Navarro for their work in creating and releasing these
tracks.
</p>

<a name="050322"></a>
<h2>May 3, 2022 &nbsp;&nbsp; New GnomAD Mutation Constraint track on GRCh38/hg38</h2>
<p>
We have just released the <b>GnomAD Genomes Mutation Constraint</b> track on the human GRCh38/hg38
genome assembly. This track's data is based on gnomAD v3.1.2 and shows
relative frequency of variation in one kilobase windows across the entire genome. 
This quantifies population occurrence of disruptive variation caused by purifying natural 
selection, similar to negative selection on loss-of-function (LoF) for genes, but 
calculated for non-coding regions, too. View the complete set of gnomAD tracks and read more on the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=gnomadVariants">gnomAD description page</a>.

<a name="050222"></a>
<h2>May 2, 2022 &nbsp;&nbsp; New Track Hub bigBarChart display and selection features</h2>
<p>
We are pleased to announce new settings to support the display of bigBarChart data in Track Hubs.
<a href="help/hgTrackHubHelp.html#Intro">Track Hubs</a> are web-accessible directories of
genomic data that can be viewed on the UCSC Genome Browser. Track Hubs allow a lab to share and
host all their genomic data (<a href="help/bam.html">BAMs</a>, <a href="help/vcf.html">VCFs</a>,
etc.) via a single Genome Browser link. The <a href="help/barChart.html">bigBarChart</a>
track format displays a bar graph of category-specific values over genomic regions and is
useful for displaying gene expression across a spectrum of tissues or other datasets where
it is desirable to compare a set of related variables over specific genomic regions.</p>
<p>
Two new settings <a href="help/trackDb/trackDbHub.html#barChartBarMinWidth">barChartBarMinWidth</a>
and <a href="help/trackDb/trackDbHub.html#barChartBarMinPadding">barChartBarMinPadding</a>
allow customizing the width and padding between bars as can be seen in these two example
tracks using these settings on the same source data.</p>
<div class="text-center">
  <img src="../images/barChartPadWidth_1.png" alt="barChartBarMinPadding/MinWidth ex" width="700">
</div>
<p>
See this example with more details on the <a target="_blank"
href="help/barChart.html#example5">bigBarChart help page</a>.</p>
<p>
Another set of new <a href="help/barChart.html#example6">bigBarChart</a> track
settings aid in the selection of bar charts on the details page:
<a href="help/trackDb/trackDbHub.html#barChartFacets" target="_blank">barChartFacets</a>,
<a href="help/trackDb/trackDbHub.html#barChartStatsUrl" target="_blank">barChartStatsUrl</a>,
and <a href="help/trackDb/trackDbHub.html#barChartMerge" target="_blank">barChartMerge</a>.</p>
<div class="text-center">
  <img src="../images/barChartFacets1.png" alt="barChartBarMinPadding/MinWidth ex" width="700">
</div>
<p>
These new settings allow users to explore the bigBarChart data on the individual details page more
closely. By presenting a button to &quot;merge&quot; or &quot;unmerge&quot; categories of data
organized by groups such as &quot;tissue,&quot; &quot;cell_class,&quot; and &quot;cell_type&quot;
the data can be compacted. Also, the column headers can be clicked (cluster/count/val) to
reorder the displayed data by numerical value or alphabetical name. By using the facet checkboxes
a more narrow set of specific cell types can be selected, such as just &quot;macrophage&quot;
where the &quot;unmerge&quot; button can then expand the details on tissues, as seen in the below image.</p>
<div class="text-center">
  <img src="../images/barChartFacets3.png" alt="barChartBarMinPadding/MinWidth ex" width="700">
</div>
<p>
After making selections, a &quot;Return to Genome Browser&quot; link on the details page allows
a user to return to the main view. Only the bars that were just selected by the facet selections
will then be displayed.</p>
<div class="text-center">
  <img src="../images/barChartFacets4.png" alt="barChartBarMinPadding/MinWidth ex" width="700">
</div>
<p>
Track Hub developers wishing to use these new bigBarChart facet selection options can see
this example illustrating these new settings on the <a target="_blank"
href="help/barChart.html#example6">bigBarChart help page</a>.</p>

<a name="042622"></a>
<h2>Apr. 26, 2022 &nbsp;&nbsp; ReMap Atlas of Regulatory Regions for human, mouse, and drosophila</h2>
<p>
We are happy to announce the release of the ReMap Atlas of Regulatory Regions track for
<a href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=ReMap">hg38</a>,
<a href="/cgi-bin/hgTrackUi?db=hg19&c=chrX&g=ReMap">hg19</a>,
<a href="/cgi-bin/hgTrackUi?db=mm39&c=chrX&g=ReMap">mm39</a>,
<a href="/cgi-bin/hgTrackUi?db=mm10&c=chrX&g=ReMap">mm10</a>, and
<a href="/cgi-bin/hgTrackUi?db=dm6&c=chr4&g=ReMap">dm6</a>. These tracks represent the
<a href="https://remap.univ-amu.fr/" target="_blank">ReMap Atlas</a> of regulatory regions, which
consists of a large-scale integrative analysis of all Public ChIP-seq data for transcriptional
regulators from GEO, ArrayExpress, and ENCODE. Each transcription factor follows a specific RGB
color. Individual BED files for specific TFs, cells/biotypes, or data sets can be found and
downloaded on the <a href="https://remap.univ-amu.fr/download_page"
target="_blank">ReMap website</a>.
</p>
<p>
The image below shows the ReMap ChIP-seq Track, where a right-click can bring up the configuration
option.  Then there are filters available using either regular expression (regexp) or wildcards.
In the image <code>.*A\|B\|F.*</code> is using regexp, which will select all Transcriptional
regulators that include either the letter A, B, or F, while a separate wildcard filter on the
Biotypes field, <code>MC*</code>, selects all tissue names starting with MC. Drop-down menus
also exist to perform the same filters, as well as the ability to customize the label, for instance,
to only show the Biotype.</p>
<div class="text-center">
  <img src="../images/newsArchImages/ReMap.png" alt="ReMap Filter" width="1000">
</div>
<p>
We would like to thank Benoit Ballester and the ReMap team at
<a href="https://tagc.univ-amu.fr/" target="_blank">TAGC INSERM</a> (Inserm National Institute of
Health and Medical Research) for making this data available. We also want to thank Max Haeussler,
Gerardo Perez and Jairo Navarro for the creation and release of these tracks.
</p>

<a name="042222"></a>
<h2>Apr. 22, 2022 &nbsp;&nbsp;European Variant Archive (EVA) data now available on 16 assemblies</h2>
<p>
The <a href="https://www.ebi.ac.uk/eva/" target="_blank">European Variant Archive</a>
has extensive data on variation across many different species and assemblies. We have 
updated our 16 most popular non-human genome assemblies with their latest data:
<a href="https://www.ebi.ac.uk/eva/?RS-Release&releaseVersion=3" target="_blank">Release 3</a>.
Adding more to this dataset, we processed the raw data to include two extra fields, describing
variant class and amino acid change if present. You can read more about these datasets on
the <a href="/cgi-bin/hgTrackUi?db=mm39&g=evaSnp">description page</a>
of any of the listed organisms. UCSC is proud to support research into ever-increasing species
diversity and variant display. Happy Earth Day!</p>
<ul>
<li>Mouse (<a href="/cgi-bin/hgTracks?db=mm39&evaSnp=pack">mm39</a>,
   <a href="/cgi-bin/hgTracks?db=mm10&evaSnp=pack">mm10</a>)
<li>Zebrafish (<a href="/cgi-bin/hgTracks?db=danRer11&evaSnp=pack">danRer11</a>,
   <a href="/cgi-bin/hgTracks?db=danRer10&evaSnp=pack">danRer10</a>)
<li>Cow (<a href="/cgi-bin/hgTracks?db=bosTau9&evaSnp=pack">bosTau9</a>)
<li>Fruit Fly (<a href="/cgi-bin/hgTracks?db=dm6&evaSnp=pack">dm6</a>)
<li>Chicken (<a href="/cgi-bin/hgTracks?db=galGal6&evaSnp=pack">galGal6</a>,
   <a href="/cgi-bin/hgTracks?db=galGal5&evaSnp=pack">galGal5</a>)
<li>Pig (<a href="/cgi-bin/hgTracks?db=susScr11&evaSnp=pack">susScr11</a>)
<li>Rat (<a href="/cgi-bin/hgTracks?db=rn7&evaSnp=pack">rn7</a>,
   <a href="/cgi-bin/hgTracks?db=rn6&evaSnp=pack">rn6</a>)
<li>Horse (<a href="/cgi-bin/hgTracks?db=equCab3&evaSnp=pack">equCab3</a>)
<li>Rhesus (<a href="/cgi-bin/hgTracks?db=rheMac10&evaSnp=pack">rheMac10</a>)
<li>Macaque (<a href="/cgi-bin/hgTracks?db=macFas5&evaSnp=pack">macFas5</a>)
<li>Sheep (<a href="/cgi-bin/hgTracks?db=oviAri4&evaSnp=pack">oviAri4</a>)
<li>Cat (<a href="/cgi-bin/hgTracks?db=felCat9&evaSnp=pack">felCat9</a>)
</ul>

<a name="041522"></a>
<h2>Apr. 15, 2022 &nbsp;&nbsp; DECIPHER variants track for human (hg38)</h2>
<p>
We are pleased to announce the release of two DECIPHER variant tracks,
<a href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=decipher">DECIPHER CNVs</a> and
<a href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=decipherSnvs">DECIPHER SNVs</a>, available for human
(GRCh38/hg38)</a>. The genomic locations of DECIPHER variants are labeled with the DECIPHER variant
descriptions. <b>Mouseover</b> on items shows variant details, clinical interpretation, and
associated conditions. Further information on each variant is displayed on the details page by a
click onto any variant.
</p>

<p>
For the <b>CNVs track</b>, the entries are colored by the <b>type of variant</b>:
<ul>
 <li><b><font color="red">red</font></b> for loss</li>
 <li><b><font color="blue">blue</font></b> for gain</li>
 <li><b><font color="grey">grey</font></b> for amplification</li>
</ul>
</p>

<p>
A light-to-dark color gradient indicates the <b>clinical significance</b> of each variant, with 
the lightest shade being benign, to the darkest shade being pathogenic. Detailed information on the 
CNV color code is described on the following
<a href="/goldenPath/help/hgCnvColoring.html">help page</a>. Items can be filtered according to the
size of the variant, variant type, and clinical significance using the track <b>Configure</b>
options.
</p>

<P>
For the <b>SNVs track</b>, the entries are colored according to the estimated
<b>clinical significance</b> of the variant:
<ul>
 <li><b><font color="black">black</font></b> for likely or definitely pathogenic</li>
 <li><b><font color="#888">dark grey</font></b> for uncertain or unknown</li>
 <li><b><font color="#c8c8c8">light grey</font></b> for likely or definitely benign</li>
 </ul>
</P>

<p>
For more information on DECIPHER, please contact
<A HREF="mailto:&#99;&#111;n&#116;&#97;c&#116;&#64;&#100;&#101;&#99;&#105;p&#104;&#101;&#114;&#103;&#101;&#110;&#111;&#109;&#105;c&#115;.
&#111;&#114;g">
&#99;&#111;n&#116;&#97;c&#116;&#64;&#100;&#101;&#99;&#105;p&#104;&#101;&#114;&#103;&#101;&#110;&#111;&#109;&#105;c&#115;.
&#111;&#114;g</A>.
We would like to thank Jonathan Casper and Jairo Navarro of the UCSC Genome Browser group for their
work in creating and releasing these tracks.
</p>

<a name="041222"></a>
<h2>Apr. 12, 2022 &nbsp;&nbsp; T2T CHM13 v2.0 now available in the Genome Browser</h2>
<p>
The Genome Browser has a <a href="/goldenPath/history.html">rich history</a> intricately connected
to human genomic research. We have provided display to almost two dozen human genomes beginning 
with the first drafts in the year 2000. Nearly 22 years later, the <a 
href="https://sites.google.com/ucsc.edu/t2tworkinggroup" target="_blank">T2T consortium</a> has 
published the most complete human haploid genome sequence to date, having added just about all of the 200 
million bases (8%) missing from the current reference. We are proud of all the scientists 
involved, including our colleagues in the <a href="https://genomics.ucsc.edu/" 
target="_blank">UCSC Genomics Institute</a>, who played a role in this release. We strive
to facilitate omics research and thus would like to announce our expanded support for 
the <a 
href="/cgi-bin/hgTracks?hubUrl=https://hgdownload.gi.ucsc.edu/hubs/GCA/009/914/755/GCA_009914755.4/hub.txt&genome=GCA_009914755.4&position=lastDbPos" 
target="_blank">T2T-CHM13 v2.0 browser</a>.</p>

<a name="CHM13"></a><h3>What is T2T-CHM13 v2.0?</h3>
<p>
<a href="https://www.science.org/doi/10.1126/science.abj6987" target="_blank">T2T-CHM13 v2.0</a> 
was produced by sequencing the CHM13hTERT human cell line from a hydatiform mole, which is
haploid, meaning it contains nearly uniform homozygosity. 
It also employed recent technologies such as <a target="_blank"
href="https://www.pacb.com/technology/hifi-sequencing/">HiFi</a> and <a target="_blank" 
href="https://nanoporetech.com/">nanopore</a> sequencing. The result is a 
3.055 billion base pair genome that includes gapless assemblies for all main chromosomes 
and introduces nearly 200 Mbp of novel sequence containing 1956 gene predictions, 99 of
which are predicted to be protein coding. The completed regions include all centromeric satellite 
arrays, recent segmental duplications, and the short arms of all five acrocentric chromosomes. A Y
chromosome was added from Genome in a Bottle's HG002 sample.</p>

<figure class="text-center">
<img class='text-center' src="../images/scienceFillingTheGaps.jpg" width='55%' alt="Representation
of novel regions added to current reference.">
<figcaption style="font-size:13px">Each bar is a linear visualization of a chromosome, with the chromosome number shown at 
the left. Red segments denote previously missing sequences that the T2T Consortium resolved.
Source: <a target="_blank" 
href="https://www.science.org/doi/10.1126/science.abp8653">V. ALTOUNIAN/SCIENCE</a></figcaption>
</figure>

<p>
CHM13 removes 1.2 Mbp of duplicated sequence in hg38, and 263 GENCODE genes from hg38
are absent in CHM13 as well as 3604 genes in CHM13 are absent in hg38, mostly in the 
centromeres. Variant calling using CHM13 <a target="_blank" 
href="https://www.science.org/doi/10.1126/science.abl3533">reduces the numbers of false 
positives</a> in certain medically relevant genes, and CHM13 also resolves duplications 
collapsed in hg38 that affect 48 protein-coding genes (e.g. KCNJ18, KCNJ12, KMT2C, 
MAP2K3), so it is more representative of human copy-number variation than hg38.</p>
<p>
It is also important to recognize, however, that while this assembly's chromosome 
sequences are more complete than the main chromosomes of the hg38 reference genome, 
it is not &quot;hg39&quot; as it is an alternate or 
companion assembly, not a primary reference assembly for the Genome Reference Consortium 
and NCBI. It does not contain any alternative haplotypes, and most genome annotation tracks 
now are based on the hg19 and hg38 coordinates. 
Hundreds of human genomes at a similar accuracy as CHM13 are expected to be released over 
the next 1-2 years, and therefore T2T CHM13 is the foundation of the future <a target="_blank"
href="https://humanpangenome.org/">human pangenome reference genome</a>.</p>

<h3>How to access this assembly in the Genome Browser?</h3>
<p>
As with many of our assemblies, there are a few different ways to gain access. We have 
added CHM13 to our Genomes drop-down menu, which provides direct access from almost 
anywhere on our site. Also, like most of our other genomes, it can be found by searching 
our <a target="_blank" href="/cgi-bin/hgGateway">Gateway page</a>.</p>

<figure class="text-center">
<img class='text-center' src="../images/t2tGenomesMenu.png" width='20%' alt="Finding CHM13
in the Genomes menu dropdown.">
<img class='text-center' src="../images/chm13Gateway.png" width='25%' alt="Searching CHM13
on the Gateway page.">
</figure>

<p>
CHM13 is a part of our <a href="/goldenPath/newsarch.html#060121" target="_blank">Genome 
Repository (GenArk)</a> system, and thus exists as an <a target="_blank"
href="/goldenPath/help/hgTrackHubHelp.html#Assembly">assembly hub</a>. GenArk assemblies can 
always be reached directly via their shortlink URL corresponding to their GCA accession, 
e.g. CHM13: <a 
href="https://genome.ucsc.edu/h/GCA_009914755.4">https://genome.ucsc.edu/h/GCA_009914755.4</a></p>

<h3>What annotations are currently available on the CHM13 browser?</h3>

<p>
Some notable annotations currently available on the CHM13 are listed below. Additional 
annotations will continue to be added as they become available.</p>
<p>
<b>Gene and mRNA annotations:</b>
<ul>
<li><a href="https://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hub_3267197_GCA_009914755.4&c=CP068277.2&g=hub_3267197_catLiftOffGenesV1"
target="_blank">CAT/Liftoff Genes</a> - Gene models generated using the CAT software filling in 
from the LiftOff mappings when needed. The reference annotations are from GENCODE V35.</li>
<li><a href="https://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hub_3267197_GCA_009914755.4&c=CP068277.2&g=hub_3267197_proseq"
target="_blank">CHM13 PROseq</a> - CHM13 cell line PRO-seq Bowtie2 alignments to CHM13v2.0 
(minus chrY) and unique genome-wide 21mer filtering (stranded).</li>
<li><a href="https://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hub_3267197_GCA_009914755.4&c=CP068277.2&g=hub_3267197_rnaseq"
target="_blank">CHM13 RNA-Seq</a> - CHM13 cell line RNA-seq Bowtie2 alignments to CHM13v2.0 
(minus chrY) and unique genome-wide 21mer filtering (unstranded).</li></ul></p>

<p>
<b>Clinical annotations:</b>
<ul>
<li><a href="https://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hub_3267197_GCA_009914755.4&c=CP068277.2&g=hub_3267197_clinVar20220313"
target="_blank">ClinVar Variants</a> - Lifted ClinVar data from the hg38 March 13th, 2022 release.</li>
<li><a href="https://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hub_3267197_GCA_009914755.4&c=CP068277.2&g=hub_3267197_dbSNP155"
target="_blank">dbSNP 155</a> - Lifted dbSNP 155 variants from the hg38 release.</li>
<li><a href="https://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hub_3267197_GCA_009914755.4&c=CP068277.2&g=hub_3267197_gwasSNPs2022-03-08"
target="_blank">GWAS Variants</a> - GWAS catalog variants lifted from hg38.</li></ul></p>

<p>
<b>Comparative genomics:</b>
<ul>
<li><a href="https://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hub_3267197_GCA_009914755.4&c=CP068277.2&g=hub_3267197_hgUnique"
target="_blank">CHM13 unique</a> - Regions unique to the T2T-CHM13 v2.0 assembly as compared 
to the hg38 and hg19 reference assemblies.</li>
<li><a href="https://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hub_3267197_GCA_009914755.4&c=CP068277.2&g=hub_3267197_hgLiftOver"
target="_blank">Human liftOver</a> - Contains one to one Nextflow LiftOver pipeline 
alignments between CHM13 and hg19/hg38.</li>
<li><a href="https://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hub_3267197_GCA_009914755.4&c=CP068277.2&g=hub_3267197_GCA_009914755_4_T2T-CHM13v2_0ChainNet"
target="_blank">Chain/Net Track</a> - Alignment track between CHM13 and four other human
genomes that shows rearrangements in our usual chains (=alignable) and net (=synteny)
display formats. Other genomes are hg19, hg38, HG002pat, and HG002mat.</li></ul></p>

<h3>How to display my data in CHM13?</h3>
<p>
We have added support for CHM13 to our <a href="/cgi-bin/hgConvert?db=hg38&position=lastDbPos"
target="_blank">hgConvert tool</a>. This allows region conversion of the current viewing 
window between hg19/hg38 to CHM13 and vice versa. We will also be adding support for 
conversion of data using our <a href="/cgi-bin/hgLiftOver" target="_blank">hgLiftOver tool</a> 
at our next version release on May 3rd. In the meantime, the command line version of 
<a href="https://hgdownload.gi.ucsc.edu/downloads.html#utilities_downloads"
target="_blank">liftOver</a> in combination with the proper <a target="_blank"
href="https://hgdownload.gi.ucsc.edu/hubs/GCA/009/914/755/GCA_009914755.4/liftOver/">chain file</a>
can be used to lift annotations.</p>

<figure class="text-center">
<img class='text-center' src="../images/hgConvert.png" width='80%' alt="Using hgConvert tool
to see coordinates between hg38 and CHM13.">
</figure>

<p>
<a href="/cgi-bin/hgCustom" target="_blank">Custom tracks</a> and <a href="/cgi-bin/hgHubConnect"
target="_blank">track hubs</a> can also be used to display annotations on CHM13. In the case of 
track hubs, using <code>genome GCA_009914755.4</code> is sufficient to declare the assembly. 
We have also expanded our support of variable chromosome names, so data can be loaded using either 
UCSC (&quot;chr1&quot;), NCBI (&quot;CP068277.2&quot;) or Ensembl (&quot;1&quot;) sequence 
identifiers. <b>There should no longer be a need to convert sequence names</b>.</p>
<p> 
It is worth noting that GenArk assemblies are functionally hubs, which means all data is 
stored in binary files, not MySQL databases. If your existing data pipelines do not work 
because our data formats have changed compared to hg19/hg38, please do not hesitate to 
contact us. Most formats are very similar to the MySQL tables and we have command 
line tools that can perform the conversions.</p>

<h3>Where to download CHM13 data?</h3>
<p>
All GenArk hubs are hosted on our download server. This means that all settings information 
and data for displaying this browser can be found there: 
<a href="https://hgdownload.gi.ucsc.edu/hubs/GCA/009/914/755/GCA_009914755.4/"
target="_blank">https://hgdownload.gi.ucsc.edu/hubs/GCA/009/914/755/GCA_009914755.4/</a></p>
<p>
We also provide FASTA files there with two different sequence identifiers (the 
&quot;chr1&quot; format and Genbank accessions), gene annotations in GFF and other 
formats and assembly indexes with either Genbank or &quot;chr1&quot; sequence names for the 
aligners bwa-mem2, bowtie2, hisat2 and minimap2. Detailed download instructions can 
be found in the README and on our 
<a href="https://genome.ucsc.edu/cgi-bin/hgGateway?db=hub_3267197_GCA_009914755.4"
target="_blank">assembly description page</a></p>
<p>
All liftOver files, including files to/from hg19/hg38 and CHM13 can also be found 
on our download server: 
<a href="https://hgdownload.gi.ucsc.edu/hubs/GCA/009/914/755/GCA_009914755.4/liftOver/"
target="_blank">https://hgdownload.gi.ucsc.edu/hubs/GCA/009/914/755/GCA_009914755.4/liftOver/</a></p>

<h3>Acknowledgments</h3>
<p>
We would like to thank the <a href="https://sites.google.com/ucsc.edu/t2tworkinggroup"
target="_blank">T2T Consortium</a> for this landmark accomplishment. 
We would like to extend an additional kudos to our fellow <a href="https://genomics.ucsc.edu/"
target="_blank">UCSC Genomics Institute</a> members who are part of the consortium,
Karen Miga, Benedict Paten, Kishwar Shafin, Mark Diekhans, and Miten Jain. 
Lastly, to the engineers and QA members of the Genome Browser for the rapid 
development and release of CHM13 data and features.</p>


<a name="041122"></a>
<h2>Apr. 11, 2022 &nbsp;&nbsp; CRISPR Targets for Cow (bosTau9) now available</h2>
<p>
We are happy to announce the release of the the
<a href="/cgi-bin/hgTrackUi?db=bosTau9&c=chr6&g=crisprAllTargets">CRISPR Targets</a> track for the
cow assembly (<a href="/cgi-bin/hgGateway?db=bosTau9">ARS-UCD1.2/bosTau9</a>).
</p>
<p>
The track &quot;CRISPR Targets&quot; shows all potential -NGG target sites across the genome. CRISPR
target sites were annotated with predicted specificity (off-target effects) and predicted efficiency
(on-target cleavage) by various algorithms through the tool
<a href="http://crispor.tefor.net/" target="_blank">CRISPOR</a>.
The target sequence of the guide is shown with a thick (exon) bar. The PAM motif match (NGG) is
shown with a thinner bar. Guides are colored to reflect both predicted specificity and efficiency.
Specificity reflects the &quot;uniqueness&quot; of a 20mer sequence in the genome; the less unique a
sequence is, the more likely it is to cleave other locations of the genome (off-target effects).
Efficiency is the frequency of cleavage at the target site (on-target efficiency).
</p>
<p>
This track was created by Maximilian Haeussler, with helpful input from Jean-Paul Concordet (MNHN
Paris) and Alberto Stolfi (NYU). We would also like to thank Hiram Clawson and Jairo Navarro for
developing and releasing the track.
</p>

<a name="033122"></a>
<h2>Mar. 31, 2022 &nbsp;&nbsp; Tabula Sapiens now available on hg38</h2>
<p>
We are happy to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=tabulaSapiens">Tabula Sapiens</a>
single-cell track for the human assembly GRCh38/hg38. This track collection contains two bar chart
tracks of RNA expression. The first track,
<a href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=tabulaSapiensTissueCellType">Tabula Tissue Cell</a>
allows cells to be grouped together and faceted on up to 3 categories: tissue, cell class, and cell
type. The second track,
<a href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=tabulaSapiensFullDetails">Tabula Details</a>
allows cells to be grouped together and faceted on up to 7 categories: tissue, cell class, cell
type, subtissue, sex, donor, and assay.
</p>
<p>
This track shows data from
<a href="https://www.biorxiv.org/content/10.1101/2021.07.19.452956v2" target="_blank">
The Tabula Sapiens: a multiple organ single cell transcriptomic atlas of humans</a>. The dataset
covers ~500,000 cells from a total of 24 human tissues and organs from all regions of the body using
both droplet-based and plate-based single-cell RNA-sequencing (scRNA-seq). Samples were taken from
the human bladder, blood, bone marrow, eye, fat, heart, kidney, large intestine, liver, lung, lymph
node, mammary, muscle, pancreas, prostate, salivary gland, skin, small intestine, spleen, thymus,
tongue, trachea, uterus, and vasculature. The dataset includes 264,009 immune cells, 102,580
epithelial cells, 32,701 endothelial cells, and 81,529 stromal cells. A total of 475 distinct cell
types were identified. Please see the
<a href="https://tabula-sapiens-portal.ds.czbiohub.org/" target="_blank">Tabula Sapiens Portal</a>
for further interactive displays and additional data.
</p>
<p>
We would like to thank the <a href="https://tabula-sapiens-portal.ds.czbiohub.org/about#consortium"
target="_blank">Tabula Sapiens Consortium</a> who worked on producing and publishing this data set.
We would also like to thank Jim Kent, Brittney Wick, Rachel Schwartz, and Jairo Navarro for
integrating the data into the UCSC Genome Browser.
</p>

<a name="030422"></a>
<h2>Mar. 4, 2022 &nbsp;&nbsp; New COVID track on Predicted Drug Resistance Variants</h2>
<p>
We have just added the <a href="/cgi-bin/hgTrackUi?db=wuhCor1&g=resistPred">Predicted Drug Resistance</a>
track to the SARS-CoV-2 (COVID-19) genome. This track shows 1931 amino acid variants
predicted to be relevant to anti-viral drug resistance. This data was created using
3D protein structures bound to an inhibitor drug, highlighting the adjacent amino acids.
Try it yourself:</p>
<ul>
<li><a href=" https://genome.ucsc.edu/s/view/DrugResistance">
https://genome.ucsc.edu/s/view/DrugResistance</a></li>
</ul>
<p>We would like to thank the team at 
<a target=_blank href="https://coronavirus3d.org/#/drug">Coronavirus3d.org</a>
for creating and sharing this research.</p>

<a name="022522a"></a>
<h2>Feb. 25, 2022 &nbsp;&nbsp; MANE select v1.0 track available for hg38</h2>
<p>
We are happy to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=mane">MANE select v1.0</a> track for the human
(<a href="/cgi-bin/hgGateway?db=hg38">GRCh38/hg38</a>) assembly. The
<a href="https://www.ncbi.nlm.nih.gov/refseq/MANE/" target="_blank">Matched Annotation from NCBI and
EMBL-EBI (MANE)</a> project aims to produce a matched set of high-confidence transcripts that are
identically annotated between RefSeq (NCBI) and Ensembl/GENCODE (led by EMBL-EBI). Transcripts for
MANE are chosen by a combination of automated and manual methods based on conservation, expression
levels, clinical significance, and other factors.</p>
<p>
We would like to thank the NCBI and Ensembl-GENCODE collaboration for the MANE project. We would
also like to thank Brian Raney and Jairo Navarro for the development and release of this track.</p>

<a name="022522"></a>
<h2>Feb. 25, 2022 &nbsp;&nbsp; New COVID track on Drug Resistance Mutations</h2>
<p>
We have just added the <a href="/cgi-bin/hgTrackUi?db=wuhCor1&g=resist">Drug Resistance</a>
track to the SARS-CoV-2 (COVID-19) genome. This track shows 5 amino acid variants 
documented to confer drug treatment resistance to Remdesivir. Try it yourself:</p>
<p>
<a href=" https://genome.ucsc.edu/s/view/DrugResistance">https://genome.ucsc.edu/s/view/DrugResistance</a>
</p>
<p>We would like to thank data contributor Paul Gordon for making the data
accessible on his <a href="https://people.ucalgary.ca/~gordonp/remdesivir_failure-RdRp.html">
University of Calgary website</a>.</p>

<p>Note, this track was updated on May, 3rd 2022. New data on Sotrovimab drug resistance
mutations were added via collaboration with Anna Niewiadomska from JCVI.</p>

<a name="022222"></a>
<h2>Feb. 22, 2022 &nbsp;&nbsp; Recommended Track Sets paper</h2>
<p>
We are pleased to announce a new paper about our Recommended Tracks Set feature, available
on GRCh37/hg19, which collects related clinical tracks to help investigate variants.
The paper, <a href="https://doi.org/10.1002/humu.24335"
target="_blank">Variant interpretation: UCSC Genome Browser Recommended Track Sets</a>,
covers how this new feature can facilitate the interpretation of variants for clinicians.</p>
<p>
Track Sets allow a user to quickly swap out the on-screen annotations they may be looking at
for a different set of tracks relevant to specific medical scenarios: investigating single
nucleotide variants in coding regions (Clinical SNVs), structural copy number variants
(Clinical CNVs), and functional aspects of non-coding variants (Non-coding SNVs).</p>
<p class ="text-center">
  <a href="https://doi.org/10.1002/humu.24335">
    <img class="text-center" src="../images/newsArchImages/track_sets1.png" width="900px">
  </a>
</p>
<p>
To access Recommended Track Sets, currently  available only on the hg19 assembly, go to the top blue bar
and under the  &quot;Genome Browser&quot; menu click the &quot;Recommended Track Sets&quot;
option. This will launch a dialog box offering pre-configured track sets, enabling swapping from
one view to another view without changing the current position. Please note this figure does not
include a fourth Track Set, Problematic Regions, we added since the release of the paper. Problematic
Regions help users evaluate if annotations in the Browser are located in areas of the genome of low
confidence due to high homology or other reported concerns. Once displayed, Recommended Track Sets can
then be customized, where one can configure tracks in the browser by clicking the grey bar on the
left of a track. Or to see more information about displayed items, one can mouseover for an
informative pop-up summary, or click on the item to access a details page.</p>
<p>
The paper gives an example of using the experimental data in the Clinical SNV Track Set to examine a variant.</p>
<p class ="text-center">
  <a href="https://doi.org/10.1002/humu.24335">
    <img class="text-center" src="../images/newsArchImages/track_sets2.png" width="900px">
  </a>
</p>
<p>
This figure from the paper shows a vertical blue highlight for NM_172107.4(KCNQ2):c.635A>G (p.Asp212Gly),
the Human Genome Variation Society (HGVS) nomenclature to describe this variant.
Located in the KCNQ2 gene, this variant is in a codon for Aspartic acid (D212), and the mutation
has been associated with early-onset epileptic encephalopathy, an autosomal dominant inherited
disease. The ClinVar Short Variants and SNVs tracks show that all three possible single nucleotide
substitutions have been described at this position with pathogenic or likely pathogenic
classifications with red blocks (A>C, A>T, A>G). Each described variant leads to a missense
change of the Asp residue. The SNV Track Set also includes (discussed in another figure in the paper)
computational and predictive components, such as the Rare Exome Variant Ensemble Learner (REVEL)
and Combined Annotation Dependent Depletion (CADD) tracks, which calculate impacts of nucleotide changes.</p>
<p>
The paper also discusses many of the other features developed with the release of Track Sets,
including the ability to merge items that span a region to declutter the view of CNVs and the
addition of multiple configuration filters to allow users to dynamically generate a subset
view from dense data on specified categories of interest.  Lastly, a far more informative mouseover
was created when viewing clinical data that could combine multiple fields. For instance,
in the ClinVar track, it is now possible to view a variant's HGVS annotation, molecular
consequence, clinical significance, population frequency, and associated phenotypes without
having to click into an item's details page.</p>
<p>
We would like to thank Ana Benet-Pag&#232;s and Robert Kuhn for their work publishing the paper,
and Kate Rosenbloom, Ana Benet-Pag&#232;s and Lou Nassar for implementation of the feature.</p>

<a name="021022"></a>
<h2>Feb. 10, 2022 &nbsp;&nbsp; GENCODE Genes v39 for human (hg38)</h2>
<p>
We are happy to announce the release of the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=knownGene&c=chrX">GENCODE Genes v39</a> track for the human
(<a href="/cgi-bin/hgGateway?db=hg38">GRCh38/hg38</a>) genome assembly. The track includes
protein-coding genes, non-coding RNA genes, and pseudo-genes, though pseudo-genes are not displayed
by default. It contains annotations on the reference chromosomes as well as assembly patches and
alternative loci (haplotypes).
</p>

<p>
<table class="stdTbl">
    <tr><th COLSPAN=4 style="text-align:center">GENCODE v39 Release Stats</th></tr>
    <tr align=left>
        <th>Genes</th><th>Observed</th><th>Transcripts</th><th>Observed</th></tr>
    <tr align=left>
        <td>Protein-coding genes</td><td>19,982</td>
        <td>Protein-coding transcripts</td><td>87,151</td></tr>
    <tr align=left>
        <td>Long non-coding RNA genes</td><td>18,811</td>
        <td><font size="1">- full length protein-coding</font></td><td>61,516</td></tr>
    <tr align=left>
        <td>Small non-coding RNA genes</td><td>7,567</td>
        <td><font size="1">- partial length protein-coding</font></td><td>25,635</td></tr>
    <tr align=left>
        <td>Pseudogenes</td><td>14,763</td>
        <td>Nonsense mediated decay transcripts</td><td>19,762</td></tr>
    <tr align=left>
        <td>Immunoglobulin/T-cell receptor gene segments</td><td>409</td>
        <td>Long non-coding RNA loci transcripts</td><td>53,009</td></tr>
</table>
</p>

<p>
More details about the GENCODE Genes v39 track can be found on the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=knownGene&c=chrX">track description</a> page.
</p>
<p>
We would like to thank the <a href="https://www.gencodegenes.org/" target="_blank">GENCODE
Project</a> for providing the data. The creation of this track was made possible thanks to Brian
Raney, Mark Diekhans, and Jairo Navarro.</p>

<a name="020922"></a>
<h2>Feb. 9, 2022 &nbsp;&nbsp; New GENCODE gene tracks: Human V39 (hg19/hg38) - Mouse M28
(mm39)</h2>
<p>
We are pleased to announce new GENCODE Gene annotation tracks, which correspond
to <a href="https://www.ensembl.info/2021/12/10/ensembl-105-has-been-released/"
target="_blank">Ensembl 105</a>, for three assemblies: <a target="_blank"
href="../../cgi-bin/hgTrackUi?db=hg19&c=chr21&g=wgEncodeGencodeV39lift37&position=default">hg19/GRCh37</a>,
<a target="_blank" href="../../cgi-bin/hgTrackUi?db=hg38&c=chr1&g=wgEncodeGencodeV39&position=default">hg38/GRCh38</a>,
and <a target="_blank" href="../../cgi-bin/hgTrackUi?db=mm39&c=chr12&g=wgEncodeGencodeVM28&position=default">mm39/GRCm39</a>.
For human, the GENCODE V39 annotations were mapped to hg38/GRCh38 and then back-mapped
to the hg19/GRCh37 assembly. For all three assemblies, the gene sets contain the following
tracks:</p>
<ul>
  <li>Basic - a subset of the <em>Comprehensive set</em>.</li>
  <li>Comprehensive - all GENCODE coding and non-coding transcript annotations, including
      polymorphic pseudogenes. This includes both manual and automatic annotations.</li>
  <li>Pseudogenes - all annotations except polymorphic pseudogenes.</li></ul>
<p>
The hg38 and mm39 assemblies also include the following tracks that are not available on hg19:
</p>
<ul>
  <li>2-way Pseudogenes - pseudogenes predicted by both the Yale Pseudopipe and UCSC
      Retrofinder pipelines.</li>
  <li>PolyA - polyA signals and sites manually annotated on the genome based on transcribed
      evidence (ESTs and cDNAs) of 3' end of transcripts containing at least 3 A's not matching the
      genome.</li></ul>
<p>
Details on each release can be found on the <a target="_blank"
href="https://www.gencodegenes.org/">GENCODE site</a>. This includes <a target="_blank"
href="https://www.gencodegenes.org/human/stats.html">statistics on each release</a>.
<p>We would like to thank the <a target="_blank"
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Mark Diekhans and Lou Nassar for the development and
release of these tracks.</p>

<a name="010622"></a>
<h2>Jan. 6, 2022 &nbsp;&nbsp; JASPAR Transcription Factor Binding Site track available for human
and mouse</h2>
<p>
We are happy to announce the new JASPAR 2022 track for human
(<a href="/cgi-bin/hgTrackUi?db=hg19&g=jaspar">GRCh37/hg19</a> and
<a href="/cgi-bin/hgTrackUi?db=hg38&g=jaspar">GRCh38/hg38</a>) and
mouse (<a href="/cgi-bin/hgTrackUi?db=mm39&g=jaspar">GRCm39/mm39</a> and
<a href="/cgi-bin/hgTrackUi?db=mm10&g=jaspar">GRCm38/mm10</a>). With this release, we are also
introducing the JASPAR 2020 track for GRCm38/mm10.
</p>
<p>
These tracks represent genome-wide predicted binding sites for transcription factor binding profiles
in the <a href="https://jaspar.genereg.net/about/" target="_blank">JASPAR CORE collection</a>.
JASPAR CORE is an open-source database containing a curated, non-redundant set of binding profiles
derived from published collections of experimentally defined transcription factor binding sites for
eukaryotes.
</p>
<p>
The JASPAR database is a joint effort between several labs (please see
the latest <a href="https://www.ncbi.nlm.nih.gov/pubmed/34850907" target="_blank">JASPAR paper</a>).
Binding site predictions and UCSC tracks were computed by the <a href="http://cisreg.ca/"
target="_blank">Wasserman Lab</a>. We would like to thank Daniel Schmelter and Jairo Navarro at UCSC
for building and testing these tracks.</p>

<!-- ============= 2021 archived news ============= -->
<a name="2021"></a>

<a name="122021"></a>
<h2>Dec. 20, 2021 &nbsp;&nbsp; Last of three blog posts about new GenArk hubs</h2>
<p>
We are pleased to announce the third blog post in a three-part series about
our new GenArk Assembly Hubs.
This <a href="https://genome-blog.gi.ucsc.edu/blog/2021/12/20/genark-hubs-part-3/"
target="_blank">final installment</a> provides some explanation of the technical
details behind the process. The <a href="https://bit.ly/UCSC_blog_GenArk1">first post</a>
was about accessing the data, and the <a href="https://bit.ly/UCSC_blog_GenArk2">second post</a>
shared examples of using the hub's data with BLAT, PCR, and External Tools.</p>

<a name="121621"></a>
<h2>Dec. 16, 2021 &nbsp;&nbsp; Track Hub Filters Quick Start Guide</h2>
<p>
We are pleased to announce a help page specific to building filters on Track Hubs.
<a href="help/hgTrackHubHelp.html#Intro">Track Hubs</a> are web-accessible directories of
genomic data that can be viewed on the UCSC Genome Browser. Track Hubs allow a lab to share and
host all their genomic data (<a href="help/bam.html">BAMs</a>, <a href="help/vcf.html">VCFs</a>,
ect.) via a single Genome Browser link. Such links can load multiple hubs and even be combined with
a position parameter to easily reference laboratory data around a gene of interest, see examples in
a recent <a href="https://bit.ly/UCSC_blog_sharing">blog bost</a> about sharing data with URLs.</p>
<p>
The <a href="help/hubQuickStartFilter.html">Track Hub Filters Quick Start Guide</a> provides example
hubs employing new extensive <a href="help/trackDb/trackDbHub.html#filter">filter</a> settings
available for Track Hubs.</p>
<p>
The help page is divided into three sections:
<ul>
<li>filter.fieldName - Used for numerical data</li>
<li>filterText.fieldName - Used for text filtering</li>
<li>filterValues.fieldName - Used for filtering by pre-specified values or categories in data</li>
</ul></p>
<p>
The <a href="help/hubQuickStartFilter.html#filterText">filterText.fieldName</a> section discusses
settings that allow to filter items in a hub with names as seen with this screenshot of only
displaying items starting with BRCA using the wildcard match &quot;BRCA*&quot;.</p>
<p class ="text-center">
  <a href="https://genome.ucsc.edu/s/Lou/filterTextExample1">
    <img class="text-center" src="help/examples/hubQuickStartFilter/filterTextExample1.1.png" width="800px">
    <img class="text-center" src="help/examples/hubQuickStartFilter/filterTextExample1.2.png" width="500px">
  </a>
</p>
<p>
The <a href="help/hubQuickStartFilter.html#filterValues">filterValues.fieldName</a> section discusses
how to display a drop-down list of options to select items as seen with this screenshot of filtering
items that are labeled as either a &quot;DNA-binding region&quot; or &quot;alpha helix&quot; or
&quot;beta strand&quot;.</p>
<p class ="text-center">
  <a href="http://genome.ucsc.edu/s/Lou/filterValuesExample2">
    <img class="text-center" src="help/examples/hubQuickStartFilter/filterValuesExample2.1.png" width="800px">
    <img class="text-center" src="help/examples/hubQuickStartFilter/filterValuesExample2.2.png" width="500px">
  </a>
</p>
<p>
Each section of the <a href="help/hubQuickStartFilter.html">Track Hub Filters Quick Start Guide</a>
comes with examples and sessions allowing to quickly click into real interactive
demonstration hubs to experiment with how the settings work. The example hubs also provide a
starter template for building similar hubs.</p>
<p>
Many thanks to Lou Nassar for building this Filters Quick Start Guide and Brian Raney for
his work implementing the new filter software.</p>

<a name="120621"></a>
<h2>Dec. 6, 2021 &nbsp;&nbsp; Second of three blog posts about new GenArk hubs</h2>
<p>
We are pleased to announce the second blog post in a three-part series about
our new GenArk Assembly Hubs.
This <a href="https://genome-blog.gi.ucsc.edu/blog/2021/12/06/genark-hubs-part-2/"
target="_blank">second installment</a> provides examples of using the GenArk
hubs' data with BLAT, PCR, and External Tools.</p>

<a name="120321"></a>
<h2>Dec. 3, 2021 &nbsp;&nbsp; Updated Variants of Concern SARS-CoV-2 track with Omicron</h2>
<p>
The Genome Browser has updated our <a href="/cgi-bin/hgTrackUi?db=wuhCor1&g=variantMuts">
Variants of Concern</a> track to include the 
latest WHO-designated variant of concern (VOC), Omicron B.1.1.529. This evolving virus has 
been a challenge for everyone and we are humbled to continue providing public data
and our high-quality genetics research tools, now updated with the latest COVID 
variant information.</p>
<p>
This update comes as two separate subtracks to the Variants of Concern (VOC) track, one with amino 
acid mutations sourced from <a href="https://cov-lineages.org/" target="_blank">cov-lineages.org</a> 
and another with nucleotide mutations identified from 
<a href="https://www.gisaid.org/" target="_blank">GISAID</a> sequences. These tracks
easily interface with our data filter and query tools including 
<a href="/cgi-bin/hgTables">Table Browser</a> and our 
<a href="../goldenPath/help/api.html">JSON API</a>. You can also
cross-reference independent genetics data using our Custom Track and Track Hub options.
Feel free to contact 
one of the <a href="../contacts.html">Genome Browser support emails</a> for any 
browser-related questions. View these new Omicron tracks on the 
<a href="/cgi-bin/hgTracks?db=wuhCor1">SARS-CoV-2 Genome Browser</a> or with the following
Saved Session:</p>

<p style="text-align:center;">
<a href="https://genome.ucsc.edu/s/dschmelt/omicronAnnouncement">Omicron VOC Session</a></p>

<a name="112421"></a>
<h2>Nov. 24, 2021 &nbsp;&nbsp; New Genome Browser: Dog, canFam6 (<em>Canis lupus familiaris</em>)</h2>
<p>
A genome browser is now available for the <em>Canis lupus familiaris</em> assembly released in Oct. 2020 by the
<a href="https://www.broadinstitute.org/scientific-community/science/projects/mammals-models/dog/dog-genome-links"
target="_blank">Dog Genome Sequencing Consortium</a> (Dog10K_Boxer_Tasha).
<h3>About the assembly:</h3>
<ul>
  <li><b>UCSC Genome Browser assembly ID:</b> <a href="../../cgi-bin/hgGateway?db=canFam6" 
  target="_blank">canFam6</a></li>
  <li><b>Sequencing/Assembly provider ID:</b> <a href="https://www.broadinstitute.org/scientific-community/science/projects/mammals-models/dog/dog-genome-links">
  Dog Genome Sequencing Consortium</a></li>
  <li><b>Assembly date:</b> Oct. 2020</li>
  <li><b>Accession ID:</b> GCF_000002285.5</li>
  <li><b>NCBI Genome ID:</b> <a href="https://www.ncbi.nlm.nih.gov/genome/85"
  target="_blank">85</A> (<em>Canis lupus familiaris</em>)</li>
  <li><b>NCBI Assembly ID:</b> <a href="https://www.ncbi.nlm.nih.gov/assembly/8227741"
  target="_blank">8227741</a></li>
  <li><b>NCBI BioProject ID:</b> <a href="https://www.ncbi.nlm.nih.gov/bioproject/13179"
  target="_blank">13179</A></li>
  <li><b>NCBI BioSample ID:</b> <a href="https://www.ncbi.nlm.nih.gov/biosample/SAMN02953603"
  target="_blank">SAMN02953603</a></li>
  <li><b>Scaffold count: </b>147</li>
  <li><b>Total size: </b>2,312,802,198</li>
  <li><a href="../goldenPath/credits.html#dog_credits" target="_blank">Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
  <li><a href="../goldenPath/credits.html#dog_use" target="_blank">Data use conditions and
      restrictions</a></li>
  <li><a href="../goldenPath/help/ftp.html" target="_blank">Rsync</a> and
      <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/canFam6/" target="_blank">FTP</a></li>
  <li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#dog" target="_blank">Downloads page</a></li>
</ul>
</p>

<a name="112321"></a>
<h2>Nov. 23, 2021 &nbsp;&nbsp; First of three blog posts about new GenArk hubs</h2>
<p>
We are pleased to announce the first blog post in a three-part series about
our new GenArk Assembly Hubs.
This <a href="https://genome-blog.gi.ucsc.edu/blog/2021/11/23/genark-hubs-part-1/"
target="_blank">first installment</a> explains what the GenArk hubs are, and
shares a number of different ways to access their over 1500 genomes.</p>

<a name="111821"></a>
<h2>Nov. 18, 2021 &nbsp;&nbsp; New Clinical Rare Disease track - Orphadata</h2>
<p>
We are happy to share another clinical resource for genetic disease correlation, 
<a href="/cgi-bin/hgTrackUi?db=hg38&g=orphadata">the Orphadata track</a> from 
<a target="_blank" href="https://www.orpha.net/">the Orphanet consortium</a>. This 
track shows nearly 8000 genes, on hg19 and hg38, 
annotated with human disorders and epidemiological information 
including Human Phenotype Ontology (HPO) disorder name, association type, 
modes of inheritance, age of first symptoms, and disease prevalence (if available). 
This data is gathered by a consortium of more than 40 countries, focusing 
on rare diseases. This track includes gene-disease display filters
based on association type, inheritance mode, and age of symptom onset.
</p>
<p>
Orphadata can be found on the Browser in the Phenotype and Literature 
track group, natively interfacing with other UCSC Genome Browser 
tools such as the JSON API, Table Browser, and Data Integrator.
We would like to thank the Orphanet team for providing this data. We
would also like to thank the UCSC affiliates Chris Lee, Tiana Pereira, 
and Daniel Schmelter for the creation and release of these tracks.
</p>

<a name="111721"></a>
<h2>Nov. 17, 2021 &nbsp;&nbsp; Releasing A New Genome Browser Track Group: Single-Cell RNA-seq</h2>
<p>
We are excited to release a new Genome Browser track group with single-cell
RNA-seq (scRNA-seq) datasets for the hg38 assembly. Data generated by scRNA-seq
allows us to study the heterogeneity of cells in organs, explore gene expression at
a cellular level, and track cellular states in both development and disease. 

<p class ="text-center">
    <img class="text-center" src="../images/newsArchImages/scRNAseq.png" width="70%"></p>
<p>
We are starting with 14 scRNA-seq tracks covering different major organs of the body.
Each new scRNA-seq track contains anywhere from 2-19 individual mRNA expression tracks in 
<a target="_blank" href="/goldenPath/help/barChart.html">barChart</a>
format. By default, tracks display gene expression per individual cell type
annotation and are colored according to cell class:</p>
<table cellpadding="2">
  <thead>
  <tr>
    <th style="border-bottom: 2px solid;">Color</th>
    <th style="border-bottom: 2px solid;">Cell Classification</th>
  </tr>
  </thead>
  <tbody>
    <tr>
      <td style="background-color: #f0c000; padding-left: 20px"></td>
      <td>neural</td>
    </tr>
    <tr>
      <td style="background-color: #707070; padding-left: 20px"></td>
      <td>stem cell</td>
    </tr>
    <tr>
      <td style="background-color: #ff8838; padding-left: 20px"></td>
      <td>adipose</td>
    </tr>
    <tr>
      <td style="background-color: #b05020; padding-left: 20px"></td>
      <td>fibroblast</td>
    </tr>
        <tr>
      <td style="background-color: #e02900; padding-left: 20px"></td>
      <td>myeloid</td>
    </tr>
    </tr>
        <tr>
      <td style="background-color: #f01111; padding-left: 20px"></td>
      <td>immune</td>
    </tr>
    </tr>
        <tr>
      <td style="background-color: #ff334a; padding-left: 20px"></td>
      <td>lymphoid</td>
    </tr>
    </tr>
        <tr>
      <td style="background-color: #c000c0; padding-left: 20px"></td>
      <td>muscle</td>
    </tr>
    </tr>
        <tr>
      <td style="background-color: #6000b0; padding-left: 20px"></td>
      <td>hepatocyte</td>
    </tr>
    </tr>
        <tr>
      <td style="background-color: #0000ff; padding-left: 20px"></td>
      <td>kidney tubule</td>
    </tr>
    </tr>
        <tr>
      <td style="background-color: #0050ff; padding-left: 20px"></td>
      <td>secretory</td>
    </tr>
    </tr>
        <tr>
      <td style="background-color: #0072d6; padding-left: 20px"></td>
      <td>ciliated</td>
    </tr>
    </tr>
        <tr>
      <td style="background-color: #0099c0; padding-left: 20px"></td>
      <td>epithelial</td>
    </tr>
    </tr>
        <tr>
      <td style="background-color: #00c000; padding-left: 20px"></td>
      <td>endothelial</td>
    </tr>
    </tr>
        <tr>
      <td style="background-color: #80d000; padding-left: 20px"></td>
      <td>glia</td>
    </tr>
  </tbody>
</table>
<p>
When available, expression tracks organized by other categories such as sex,
donor, sample, detailed cell type, etc. are also included in these container
tracks. In all of the tracks, the gene expression is averaged for each item
in a category.</p>
<p>
Clicking on any of the track titles below will lead to the track description
page, which includes additional information and allows for configuration of
various display options. <br><br>This release includes the following
tracks:<br></p>
<ul>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&g=bloodHao&position=default">Blood (PBMC) Hao</a>
- This track displays peripheral blood mononuclear cell expression data from 
<a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/34062119/">Hao <em>et al.</em>, 2020</a> 
for 3 levels of cell type annotations, donor, phase, and time.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&g=colonWang&position=default">Colon Wang</a> 
- This track shows colon expression data from 
<a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/31753849/">Wang <em>et al.</em>, 2020</a> 
grouped by cell type and donor.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&g=cortexVelmeshev&position=default">Cortex Velmeshev </a> 
- This track shows cortex expression data from 
<a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/31097668/">Velmeshev <em>et al.</em>, 2019</a> 
grouped by cell type, sex, sample, donor, and diagnosis.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&g=fetalGeneAtlas&position=default">Fetal Gene Atlas</a> 
- This track shows expression data from 
<a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/33184181/">Cao <em>et al.</em>, 2020</a> 
binned by cell type and other categories including sex, organ, experiment, donor, etc.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&g=heartCellAtlas&position=default">Heart Cell Atlas</a> 
- This track shows heart expression data from <a target="_blank"
href="https://pubmed.ncbi.nlm.nih.gov/32971526/">Litvi&#328;ukov&#225 <em>et
al.</em>, 2020</a> binned by cell type and various categories including cell
state, sample, region, donor, age, etc.</li>
<li>
<a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg38&g=ileumWang&position=default">Ileum Wang</a> -
This track shows ileum expression data from <a target="_blank"
href="https://pubmed.ncbi.nlm.nih.gov/31753849/">Wang <em>et al.</em>, 2020</a>
grouped by cell type and donor.</li>
<li>
<a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg38&g=kidneyStewart&position=default">Kidney
Stewart</a> - This track shows kidney expression data from <a target="_blank"
href="https://pubmed.ncbi.nlm.nih.gov/31604275/">Stewart <em>et al.</em>,
2019</a> grouped by cell type, detailed cell type, project, experiment, etc.
</li>
<li>
<a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg38&g=liverMacParland&position=default">Liver
MacParland</a> - This track shows liver expression data from <a target="_blank"
href="https://pubmed.ncbi.nlm.nih.gov/30348985/">MacParland <em>et al.</em>,
2018</a> grouped by cell type, broad cell type, and donor.</li>
<li>
<a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg38&g=lungTravaglini2020&position=default">Lung
Travaglini</a> - This track shows lung expression data from <a target="_blank"
href="https://pubmed.ncbi.nlm.nih.gov/33208946/"> Travaglini <em>et al.</em>,
2020</a> binned by categories such as cell type, sample, donor, compartment,
etc. using both 10x and Smart-seq2 library preparation methods.</li>
<li>
<a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg38&g=muscleDeMicheli&position=default">Muscle De
Micheli</a> - This track shows muscle expression data from <a target="_blank"
href="https://pubmed.ncbi.nlm.nih.gov/32624006/">De Micheli <em>et al.</em>,
2020</a> grouped by cell type and sample.</li>
<li>
<a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg38&g=pancreasBaron&position=default">Pancreas
Baron</a> - This track shows pancreas expression data from <a target="_blank"
href="https://pubmed.ncbi.nlm.nih.gov/27667365/">Baron <em>et al.</em>, 2016
</a> grouped by cell type, detailed cell type, donor, and batch.
</li>
<li>
<a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg38&g=placentaVentoTormo&position=default">Placenta
Vento-Tormo</a> - This track shows placenta and matched decidua and maternal
PBMCs expression data from <a target="_blank"
href="https://pubmed.ncbi.nlm.nih.gov/30429548/"> Vento-Tormo <em>et al.</em>,
2018</a> grouped by cell type, detailed cell type, stage, etc. using both 10x
and Smart-seq2 library preparation methods.</li>
<li>
<a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg38&g=rectumWang&position=default">Rectum Wang</a>
- This track shows rectum expression data from <a target="_blank"
href="https://pubmed.ncbi.nlm.nih.gov/31753849/"> Wang <em>et al.</em>,
2020</a> grouped by cell type and donor.</li>
<li>
<a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg38&g=skinSoleBoldo&position=default">Skin
Sole-Boldo</a> - This track shows skin expression data from <a target="_blank"
href="https://pubmed.ncbi.nlm.nih.gov/32327715/"> Sol&#233-Boldo <em>et
al.</em>, 2020</a> grouped by cell type, cell type with donor's age, donor, and
age.</li>
</ul>
<p>
Many of these datasets can also be explored interactively using the 
<a target="_blank" href="http://cells.ucsc.edu/">UCSC Cell Browser</a>. We are
interested in bringing more single-cell datasets into the Genome Browser and
the UCSC Cell Browser as well as the 
<a target="_blank" href="http://cells.ucsc.edu/">Human Cell Atlas</a> project. 
If you have data you would like to contribute, check out our 
<a target="_blank" href="https://docs.google.com/forms/d/e/1FAIpQLSdbr65jBIHwWn3GecJll2RrvxIvzxLikOTIdG17Qzsvs2seig/viewform">
questionnaire</a>.</p>
<p>
We would like to thank the publication authors for making these data available.
We would also like to thank Jim Kent, Brittney Wick, Luis Nassar, Daniel
Schmelter, Jairo Navarro, and Gerardo Perez for building and testing these
tracks.</p>

<a name="102821"></a>
<h2>
Oct. 28, 2021 &nbsp;&nbsp; New GTEx cis-eQTL and RNA-seq read coverage tracks for human (hg38)</h2>
<p>
We are happy to announce the release of two new tracks for the human assembly (GRCh38/hg38): the
<a href="../../cgi-bin/hgTrackUi?db=hg38&c=chrX&g=gtexEqtlHighConf">GTEx High-Confidence cis-eQTLs
from CAVIAR</a> and <a href="../../cgi-bin/hgTrackUi?db=hg38&c=chrX&g=gtexCov">GTEx RNA-seq read
coverage</a> tracks.
</p>
<dl>
  <dt>GTEx High-Confidence cis-eQTLs from CAVIAR</dt>
    <dd>
      <ul>
        <li>
        Shows genetic variants likely affecting proximal gene expression in 49 human tissues from
        the <a href="https://gtexportal.org/home/" target="_blank">Genotype-Tissue Expression
        (GTEx)</a> V8 data release.
        </li>
        <li>
        Items in the track are colored according to their tissue, with the color matching those in
        the <a href="../../cgi-bin/hgGtexTrackSettings?g=gtexGeneV8">GTEx Gene V8 Track</a>.
        </li>
        <li>
        The data items displayed are gene expression quantitative trait loci within 1MB of gene
        transcription start sites (cis-eQTLs), significantly associated with gene expression and in
        the credible set of variants for the gene at a high confidence level, with a causal
        posterior probability of &gt;0.1.
        </li>
        <li>
        Raw data for these analyses are available from the
        <a href="https://gtexportal.org/home/protectedDataAccess" target="_blank">GTEx Portal</a>.
        </li>
    </dd>
  <dt>GTEx V8 RNA-Seq Read Coverage by Tissue</dt>
    <dd>
      <ul>
        <li>
        The <a href="https://commonfund.nih.gov/GTEx/index" target="_blank">NIH Genotype-Tissue
        Expression</a> (GTEx) project determined genetic variation and gene expression in 52 tissues
        and 2 cell lines using RNA-seq data (V8, August 2019), on 17,382 samples from 948 adults.
        </li>
        <li>
        This track focuses on gene expression and shows read coverage, from one single sample
        per tissue, selected for high-quality and high read depth. The data is summarized to one
        number per base pair, the number of sequencing reads that cover this position.
        </li>
        <li>
        The plot shows if a given exon is transcribed primarily in certain tissues and whether
        transcription is uniform over the length of a single exon.
        </li>
        <li>
        Data can also be obtained directly from the
        <a href="https://gtexportal.org/home/datasets" target="_blank">GTEx Portal</a>.
        </li>
    </dd>
</dl>
<p class ="text-center">
  <a href="http://genome.ucsc.edu/s/Lou/GTEx">
    <img class="text-center" src="../images/newsArchImages/GTExV8_eQTL_RNA-seq.jpg" width="800px">
  </a>
</p>
<p>
We would like to thank the GTEx investigators, analysts, and portal team for providing this data. We
would also like to thank Max Haeussler, Matt Speir, and Jairo Navarro for the creation and release
of these tracks.
</p>

<a name="102121"></a>
<h2>Oct. 21, 2021 &nbsp;&nbsp; GENCODE Genes V38 for human (hg38)</h2>
<p>
We are pleased to announce the release of the
<a href="../../cgi-bin/hgTrackUi?db=hg38&c=chrX&g=knownGene">GENCODE Genes V38</a> track for the
human (<a href="../../cgi-bin/hgGateway?db=hg38">GRCh38/hg38</a>) Genome Browser.</p>
<p>
The GENCODE Genes track (version 38, May 2021) shows high-quality manual annotations merged with
evidence-based automated annotations across the entire human genome generated by the
<a href="https://www.gencodegenes.org/" target="_blank">GENCODE project</a>. By default, this track
displays only the basic GENCODE set, splice variants, and non-coding genes. This track contains
annotations on the reference chromosomes as well as assembly patches and alternative loci
(haplotypes).
</p>
<p>
More details about the new GENCODE Genes track can be found on the
<a href="../../cgi-bin/hgTrackUi?db=hg38&c=chrX&g=knownGene">track description page</a>.
</p>
<p>
We would like to thank the <a href="https://www.gencodegenes.org/"
target="_blank">GENCODE Project</a> for providing the data. We would also like to thank Brian Raney,
Mark Diekhans, and Jairo Navarro for the creation and release of this track.
</p>

<a name="102021"></a>
<h2>Oct. 20, 2021 &nbsp;&nbsp; SARS-CoV-2 Variants of Concern (VOC) track updated</h2>
<p>The UCSC Genom Browser is proud to share our latest update to the 
SARS-CoV-2 browser, the updated 
<a href="/cgi-bin/hgTrackUi?db=wuhCor1&g=variantMuts">
Variants of Concern (VOC) track</a>. This data track
includes amino acid and nucleotide annotations for 10 different COVID variants,
including the Delta and Mu variants, mapped along the SARS-CoV-2 reference genome.
These variants are
<a href="https://www.who.int/en/activities/tracking-SARS-CoV-2-variants/"
 target="_blank">classified by the WHO</a> into several categories: Variants of Concern (VOC),
Variants of Interest (VOI), and Variants under Investigation (VUM). These tracks help
provide a more clear understanding of the mutations that comprise each named variant. 
This track's items also include links to <a target="_blank" href="https://outbreak.info">
Outbreak.info</a>, providing geographic distibutions for each variant.
</p>

<p>
The underlying data is publically accessible and compatible with many analysis tools, including 
our <a href="/cgi-bin/hgTables?db=wuhCor1">Table Browser</a>, 
 <a href="/cgi-bin/hgIntegrator">Data Integrator</a>, 
and <a href="/goldenPath/help/api.html">JSON API</a>. More information on this track
can be found on the <a href="/cgi-bin/hgTrackUi?db=wuhCor1&g=variantMuts">
Variants of Concern (VOC) track description page</a>.</p>

<a name="101221"></a>
<h2>Oct. 18, 2021 &nbsp;&nbsp; Addition of GRCh38 patch 13 sequences to hg38</h2>
<p>
We are pleased to announce the addition of GRCh38 patch release 13 to the
<a href="../../cgi-bin/hgGateway?db=hg38">hg38</a> assembly. hg38 has been updated with patches
since its release in 2013. The GRC patch releases do not change any previously existing
sequences; they simply add new sequences for fix patches or alternate haplotypes that correspond
to specific regions of the main chromosome sequences. For most users, the patches are unlikely
to make a difference and may complicate the analysis as they introduce more duplication.</p>

<p>
More information on how patch sequences are incorporated can be found on the
<a href="https://genome-blog.gi.ucsc.edu/blog/patches/" target="_blank">Patching up the Genome</a> blog post.
The blog post contains details about the new
<a href="https://hgdownload.gi.ucsc.edu/goldenPath/hg38/bigZips/latest/"
target="_blank">/latest</a>,
<a href="https://hgdownload.gi.ucsc.edu/goldenPath/hg38/bigZips/initial/"
target="_blank">/initial</a>, and
<a href="https://hgdownload.gi.ucsc.edu/goldenPath/hg38/bigZips/p13/" target="_blank">/p13</a>
download directories on the downloads server. With the addition of new sequences to hg38, we can
expect to see BLAT return more matches to the genome.</p>

<ul>
  <li>
    The <b>initial/</b> subdirectory contains files for the initial release of GRCh38, which has 455
    sequences, 261 alternate sequences, and no fix sequences. It is the same as the
    parent download directory. This is probably the best genome file for aligners and most analysis
    tasks, a version called &quot;analysisSet&quot; for the human genome.
  </li>
  <li>
    The <b>p13/</b> subdirectory contains files for GRCh38.p13 (patch release 13). It has 640 sequences
    including alternate and fix sequences.
  </li>
  <li>
    The <b>latest/</b> symbolic link points to the subdirectory for the most recent patch version.
  </li>
</ul>

<p>
We would like to thank the <a href="https://www.ncbi.nlm.nih.gov/grc" target="_blank">Genome
Reference Consortium</a> for creating the patches for hg38. We would also like to thank Galt Barber,
Jairo Navarro, and Gerardo Perez at UCSC for implementing and testing the latest patch to the
hg38 genome.</p>

<a name="092721"></a>
<h2>Sep. 27, 2021 &nbsp;&nbsp; JASPAR tracks for human (hg19/hg38)</h2>
<p>
We are pleased to announce the release of the <a target="_blank" 
href="/cgi-bin/hgTrackUi?db=hg38&g=jaspar&position=default">
JASPAR Transcription Factors tracks</a> for human assemblies hg19/GRCh37 and hg38/GRCh38.</p> 
<p>
These data represent genome-wide predicted binding sites for TF (transcription factor) 
binding profiles in the <a target="_blank" href="http://jaspar.genereg.net/about/">
JASPAR CORE collection</a>. This collection is an open-source database containing a curated, 
non-redundant set of binding profiles derived from published collections of experimentally 
defined transcription factor binding sites for eukaryotes.</a>
<p>
JASPAR has been a popular <a target="_blank" href="/cgi-bin/hgHubConnect">Public Hub</a> 
for many years and in collaboration with their group, the data can now be found as native 
tracks. The hub will continue to exist uninterrupted and includes data on additional 
assemblies mm10, danRer11, dm6, ce10, araTha1, and sacCer3.</a></p>
<p>
See the <a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&g=jaspar&position=default">
track description page</a> for more information including data access, available filters, 
display conventions, and direct links to the other assemblies.</a></p>
<p>
We would like to thank Anthony Mathelier and Oriol Fornes as well as the rest of the researchers 
that develop and maintain the
<a target="_blank" href="http://jaspar.genereg.net/about/"</a>JASPAR database</a>. We
would also like to thank Daniel Schmelter and Lou Nassar for the development and release
of these tracks.</p>

<a name="091321"></a>
<h2>Sep. 13, 2021 &nbsp;&nbsp; Cactus 241-way comparative genomics track for hg38</h2>
<p>
A new <a href="../../cgi-bin/hgTrackUi?db=hg38&g=cons241way&position=default"> 241-way
comparative genomics track</a> has been added to the human
(<a href="../../cgi-bin/hgGateway?db=hg38">GRCh38/hg38</a>) genome browser. This track
shows multiple alignments of 241 vertebrate species and measurements of evolutionary
conservation from the <a href="https://zoonomiaproject.org/the-project/"
target="_blank">Zoonomia Project</a>.</p>
<p>
The multiple alignments were generated using
the <a href="https://github.com/ComparativeGenomicsToolkit/cactus"
target="_blank">Cactus comparative genomics alignment system</a> and the base-wise
conservation scores were computed using phyloP from the PHAST package. For genome
assemblies not available in the genome browser, there are alternative assembly hub
genome browsers. Missing sequence in any assembly is highlighted in the track display
by regions of yellow when zoomed out and by Ns when displayed at base level.</p>
<p>
We would like to thank the Zoonomia Project for providing the data. We would also like to thank Hiram
Clawson, Brian Joseph Raney, Mark Diekhans, Lou Nassar, and Gerardo Perez for their efforts in
creating this track.</p>

<a name="083121"></a>
<h2>Aug. 31, 2021 &nbsp;&nbsp; GENCODE Genes VM27 for mouse (mm39)</h2>
<p>
We are pleased to announce the release of the
<a href="../../cgi-bin/hgTrackUi?db=mm39&c=chr12&g=knownGene">GENCODE Genes VM27</a> track for the
mouse (<a href="../../cgi-bin/hgGateway?db=mm39">GRCm39/mm39</a>) Genome Browser.</p>
<p>
The GENCODE Genes track (version M27, Dec 2020) shows high-quality manual annotations merged with
evidence-based automated annotations across the entire mouse genome generated by the
<a href="https://www.gencodegenes.org/" target="_blank">GENCODE project</a>. By default, this track
displays only the basic GENCODE set, splice variants, and non-coding genes. This track contains
annotations on the reference chromosomes as well as assembly patches and alternative loci
(haplotypes).
</p>
<p>
More details about the new GENCODE Genes track can be found on the
<a href="../../cgi-bin/hgTrackUi?db=mm39&c=chr12&g=knownGene">track description page</a>.
</p>
<p>
We would like to thank the <a href="https://www.gencodegenes.org/"
target="_blank">GENCODE Project</a> for providing the data. We would also like to thank Brian Raney,
Mark Diekhans, and Jairo Navarro for the creation and release of this track.
</p>

<a name="080421"></a>
<h2>Aug. 04, 2021 &nbsp;&nbsp; Addition of GRCm38 patch 6 sequences to mm10</h2>
<p>
We are pleased to announce the addition of GRCm38 patch release 6 to the mm10 assembly. mm10 has
been updated with patches since its release in 2012. The GRC patch releases do not change any
previously existing sequences; they simply add new sequences for fix patches or alternate haplotypes
that correspond to specific regions of the main chromosome sequences. For most users, the patches
are unlikely to make a difference and may complicate the analysis as they introduce more
duplication.</p>

<p>
More information on how patch sequences are incorporated can be found on the
<a href="https://genome-blog.gi.ucsc.edu/blog/patches/" target="_blank">Patching up the Genome</a> blog post.
The blog post contains details about the new
<a href="https://hgdownload.gi.ucsc.edu/goldenPath/mm10/bigZips/latest/"
target="_blank">/latest</a>,
<a href="https://hgdownload.gi.ucsc.edu/goldenPath/mm10/bigZips/initial/"
target="_blank">/initial</a>, and
<a href="https://hgdownload.gi.ucsc.edu/goldenPath/mm10/bigZips/p6/" target="_blank">/p6</a>
download directories on the downloads server. With the addition of new sequences to mm10, we can
expect to see BLAT return more matches to the genome.</p>

<ul>
  <li>
    The <b>initial/</b> subdirectory contains files for the initial release of GRCm38, which has 66
    sequences, no original alternate sequences, and no fix sequences. It is the same as the
    parent download directory. This is probably the best genome file for aligners and most analysis
    tasks, a version called &quot;analysisSet&quot; for the human genome.
  </li>
  <li>
    The <b>p6/</b> subdirectory contains files for GRCm38.p6 (patch release 6). It has 239 sequences
    including alternate and fix sequences. Note that these patches include
    &quot;strain-specific&quot; sequences. You may want to check with the authors of your aligner if
    the software can recognize these sequences.
  </li>
  <li>
    The <b>latest/</b> symbolic link points to the subdirectory for the most recent patch version.
  </li>
</ul>

<p>
We would like to thank the <a href="https://www.ncbi.nlm.nih.gov/grc" target="_blank">Genome
Reference Consortium</a> for creating the patches for mm10. We would also like to thank Galt Barber
and Jairo Navarro at UCSC for implementing and testing the latest patch to the mm10 genome.</p>

<a name="080221"></a>
<h2>Aug. 02, 2021 &nbsp;&nbsp; New Genome Browser: X. tropicalis, xenTro10 (<em>Xenopus tropicalis</em>)</h2>
<p>
A genome browser is now available for the <em>Xenopus tropicalis</em> assembly released in Nov. 2019 by the
<a href="https://www.berkeley.edu/" target="_blank"> University of California, Berkeley</a> (UCB_Xtro_10.0).
<h3>About the assembly:</h3>
<ul>
  <li><b>UCSC Genome Browser assembly ID:</b>  <a href="../../cgi-bin/hgGateway?db=xenTro10" 
  target="_blank">xenTro10</a></li>
  <li><b>Sequencing/Assembly provider ID:</b> <a href="https://www.berkeley.edu/">
University of California, Berkeley</a></li>
  <li><b>Assembly date:</b> Nov. 2019</li>
  <li><B>Accession ID:</B> GCF_000004195.4</li>
  <li><B>NCBI Genome ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/genome/80'
TARGET='_blank'>80</A> (<em>Xenopus tropicalis</em>)</li>
  <li><B>NCBI Assembly ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/assembly/5323661'
TARGET='_blank'>5323661</A></li>
  <li><B>NCBI BioProject ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/bioproject/577946'
TARGET='_blank'>577946</A></li>
  <li><B>NCBI BioSample ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/biosample/SAMN13041969'
TARGET='_blank'>SAMN16261960</A></li>
  <li><b>Scaffold count: </b>167</li>
  <li><b>Total size: </b>1,451,301,209</li>
  <li><a href="../goldenPath/credits.html#xentro_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
  <li><a href="../goldenPath/credits.html#xentro_use" 
target="_blank" >Data use conditions and restrictions</a></li>
  <li><a href="../goldenPath/help/ftp.html" 
target="_blank" >Rsync</a> and
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/xenTro10/" 
target="_blank" >FTP</a></li>
  <li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#xentro"
target="_blank" >Downloads page</a></li>
</ul>
</p>
<p>
We would like to thank the University of California, Berkeley for creating this new X. tropicalis assembly.
We would also like to thank Hiram Clawson, Lou Nassar, and Gerardo Perez for the development and release of
the xenTro10 genome browser.
</p>

<a name="072021"></a>
<h2>July 20, 2021 &nbsp;&nbsp; Updated GENCODE tracks released for hg38, hg19, and mm39</h2>
<p>
We have updated our GENCODE gene datasets to the latest releases; v38 
for human (hg38 &amp; hg19) and v27 for the mouse genome assembly (mm39). 
These new annotation data were released by GENCODE in May 2021 
and correspond to the Ensembl Genes 104 dataset. 
</p>
<p>
These tracks can be found in our All GENCODE super-track folder, 
with subset tracks and abundant filter options. This super-track
contains many previous versions of the GENCODE genes for easy 
comparison. From the 
<a target="_blank" href="https://www.gencodegenes.org/human/stats.html">
GENCODE stats page</a>,
this updated dataset has approximately 60.6k genes and 237k transcripts on 
the human hg38 assembly.</p>
<p>
For more information or to visualize the data, 
you can visit the following track description pages:</p>
<ul>
<li><a href="../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeGencodeV38">
All GENCODE v38 for hg38</a></li>
<li><a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV38lift37">
GENCODE lifted Annotations V38lift37 for hg19</a></li>
<li><a href="../../cgi-bin/hgTrackUi?db=mm39&g=wgEncodeGencodeVM27">
All GENCODE annotations VM27 for mm39</a></li>
</ul>

<a name="071421"></a>
<h2>July 14, 2021 &nbsp;&nbsp; gnomAD v3.1.1 track available for hg38</h2>
<p>
We are happy to announce an update to the Genome Aggregation Database (gnomAD) variants track for
the <a href="/cgi-bin/hgGateway?db=hg38">hg38/GRCh38</a> assembly. The
<a href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=gnomadGenomesVariantsV3_1_1">gnomAD v3.1.1 variants</a>
track contains the same underlying data as v3.1, but with minor corrections to the VEP annotations
and dbSNP rsIDs. In this update, we have now included the mitochondrial chromosome data that was
released as part of gnomAD v3.1 but after the UCSC version of the track was created and released.
For more information about gnomAD v3.1.1, please see the related
<a href="https://gnomad.broadinstitute.org/news/2021-03-gnomad-v3-1-1/"
target="_blank">changelog</a>.
</p>
<p>
We would like to thank the <a href="http://gnomad.broadinstitute.org/" target="_blank">Genome
Aggregation Database</a> team at the BROAD institute for releasing the underlying data. We would
also like to thank Chris Lee and Jairo Navarro for the creation and release of the UCSC Genome
Browser track.</p>

<a name="070821"></a>
<h2>July 08, 2021 &nbsp;&nbsp; Updated data for NCBI RefSeq tracks for human assembly hg38</h2>
<p>
We are pleased to announce the updated  <a href="https://www.ncbi.nlm.nih.gov/genome/annotation_euk/
Homo_sapiens/109.20210514/?utm_source=blog&utm_medium=referral&utm_campaign=RefSeq-release-206&
utm_term=homo-sapiens&utm_content=20210527link1"
target="_blank">NCBI RefSeq Annotation Release 109.20210514</a> tracks for the
<a href="/cgi-bin/hgTrackUi?db=hg38&g=refSeqComposite&position=default">hg38/GRCh38</a>
assembly. To keep up with the periodic version releases, we will update the NCBI RefSeq tracks
automatically for both hg19/GRCh37 and hg38/GRCh38. For all of these tracks, the alignments
and coordinates are provided by RefSeq. These tracks are organized in a composite track that
includes:</p>
<ul>
  <li><b>RefSeq All</b> &ndash; all annotations from the curated and predicted sets
  <li><b>RefSeq Curated</b> &ndash; curated annotations beginning with NM, NR, or NP
  <li><b>RefSeq Other</b> &ndash; all other RefSeq annotations not included in RefSeq All
  <li><b>RefSeq Alignments</b> &ndash; alignments of transcripts to the genome provided by RefSeq
  <li><b>RefSeq Diffs</b> &ndash; alignment differences between the human reference genome(s) and RefSeq transcripts
  <li><b>RefSeq Select and MANE</b> &ndash; a single Select transcript representative for each protein-coding gene
  <li><b>RefSeq HGMD</b> &ndash; transcripts with clinical variants in Human Gene Mutation Database
</ul>
<p>
We would like to thank NCBI and the RefSeq Annotation database for collecting and curating these data.
We would also like to thank Hiram Clawson, Lou Nassar, and Gerardo Perez for developing and releasing these tracks.
</p>


<a name="063021"></a>
<h2>June 30, 2021 &nbsp;&nbsp; Older dbSNP tracks archived into a super track folder: dbSNP Archive</h2>
<p>
The UCSC Genome Browser's list of genetic variation datasets is now more organized, 
with a new folder supertrack for previous dbSNP versions. We have listened to requests 
and have simplified our tracks listing. This change adds a new track link and 
clears up older versions of dbSNP links (for mm10, hg19, and hg38) while retaining 
all functionality of filters and saved sessions.
The previous versions of dbSNP can be accessed and visualized by the new dbSNP Archive supertrack.</p>
<ul>
<li><a href="../../cgi-bin/hgTrackUi?db=hg38&g=dbSnpArchive">dbSNP Archive for hg38</a></li>
<li><a href="../../cgi-bin/hgTrackUi?db=hg19&g=dbSnpArchive">dbSNP Archive for hg19</a></li>
<li><a href="../../cgi-bin/hgTrackUi?db=mm10&g=dbSnpArchive">dbSNP Archive for mm10</a></li>
</ul>

<p class="text-center">
  <img class='text-center' src="../images/dbSnp.Archive.png"  width='100%' alt="The 
variation section now has the dbSNP Archive track.">
</p>

<a name="062821"></a>
<h2>June 28, 2021 &nbsp;&nbsp; Version 3 of Track Database Definition Document Released</h2>
<p>
The <a href="help/trackDb/trackDbHub.html" target="_blank">Track Database Definition Document</a>
defining hub settings has been updated from version two to version three. This document primarily
serves as a library of settings that users can refer to when they are setting up and configuring
their own track hubs. The update in the version is to account for two new sections that highlight
the new <a href="help/bigLolly.html">bigLolly</a> type specialized for variant data and the new
<a href="help/vcf.html#trio" >vcfPhasedTrio</a> type specialized for phased VCF data.</p>
<p>
The new specifications for these types are included in their relevant
sections such as <code>vcfChildSample</code> and
<code>vcfUseAltSampleNames</code> for the new <a href="help/trackDb/trackDbHub.html#vcfPhasedTrio"
target="_blank">vcfPhasedTrio settings</a>, or <code>lollySizeField</code> and <code>lollyField</code>
for the new  <a href="help/trackDb/trackDbHub.html#bigLolly" target="_blank">bigLolly settings</a>
with full descriptions in the document.</p>
<p>
The Track Database Definition Document also serves a second function beyond defining settings as
providing a versioned specification for those sites that support and display data from track hubs
such as Ensembl and NCBI. See this <a href="https://genome-blog.gi.ucsc.edu/blog/how-portable-is-your-track-hub-use-hubcheck-to-find-out/"
target="_blank">blog post</a> to learn more about this second element of the document and using the
hubCheck utility to gain an idea of how your hub's settings might work on different sites.</p>

<a name="062421"></a>
<h2>June 24, 2021 &nbsp;&nbsp; New REVEL missense scores track for hg38 and hg19</h2>
<p>
The UCSC Genome Browser has just added the REVEL tracks to the hg38 and hg19 reference genomes. 
This much requested clinical dataset is now visualized and interactive on the Genome Browser. 
REVEL scores predict the pathogenicity of missense variants for every possible 
basepair change across the entire genome's coding sequences. 
This dataset bases its scores on 13 individual tools
to produce high predictive performance of common and rare mutations. To learn more about the
REVEL dataset, please visit their site below:</p>
<ul>
  <li><a href="https://sites.google.com/site/revelgenomics/" target="_blank">
  https://sites.google.com/site/revelgenomics</a></li>
</ul>
<p>
To access these tracks on the Genome Browser, please visit their description pages below 
and change the tracks' visibility.</p>
<ul>
  <li><a href="/cgi-bin/hgTrackUi?db=hg38&g=revel">REVEL track on hg38</a></li>
  <li><a href="/cgi-bin/hgTrackUi?db=hg19&g=revel">REVEL track on hg19</a></li>
</ul>
<p>
Thanks to REVEL for providing these annotations, as well as
Max Haeussler and Dan Schmelter of the UCSC Genome Browser group for their
work in creating and releasing these tracks.</p>

<a name="061421"></a>
<h2>June 14, 2021 &nbsp;&nbsp; Updated Ensembl gene tracks for 48 assemblies (v104)</h2>
<p>
We are pleased to announce updated
<a href="http://uswest.ensembl.org/info/genome/genebuild/index.html" target="_blank">
Ensembl gene annotation</a> tracks corresponding to <strong>v104</strong> for the
following assemblies:</p><br>
<div class="container">
  <div class="row">
  <div class="col-sm-4">
  <ul>
        <li>Armadillo <a href="../cgi-bin/hgTrackUi?db=dasNov3&g=ensGene&position=default"
         target=_blank>(dasNov3)</a></li>
        <li>Baboon <a href="../cgi-bin/hgTrackUi?db=papAnu4&g=ensGene&position=default"
         target=_blank>(papAnu4)</a></li>
        <li>Bison <a href="../cgi-bin/hgTrackUi?db=bisBis1&g=ensGene&position=default"
         target=_blank>(bisBis1)</a></li>
        <li>Bonobo <a href="../cgi-bin/hgTrackUi?db=panPan2&g=ensGene&position=default"
         target=_blank>(panPan2)</a></li>
        <li>Budgerigar <a href="../cgi-bin/hgTrackUi?db=melUnd1&g=ensGene&position=default"
         target=_blank>(melUnd1)</a></li>
        <li>Bushbaby <a href="../cgi-bin/hgTrackUi?db=otoGar3&g=ensGene&position=default"
         target=_blank>(otoGar3)</a></li>
        <li>C. elegans <a href="../cgi-bin/hgTrackUi?db=ce11&g=ensGene&position=default"
         target=_blank>(ce11)</a></li>
        <li>C. intestinalis <a href="../cgi-bin/hgTrackUi?db=ci3&g=ensGene&position=default"
         target=_blank>(ci3)</a></li>
        <li>Cat <a href="../cgi-bin/hgTrackUi?db=felCat9&g=ensGene&position=default"
         target=_blank>(felCat9)</a></li>
        <li>Chicken <a href="../cgi-bin/hgTrackUi?db=galGal6&g=ensGene&position=default"
         target=_blank>(galGal6)</a></li>
        <li>Chimp <a href="../cgi-bin/hgTrackUi?db=panTro5&g=ensGene&position=default"
         target=_blank>(panTro5)</a></li>
        <li>Chinese hamster <a href="../cgi-bin/hgTrackUi?db=criGriChoV1&g=ensGene&position=default"
         target=_blank>(criGriChoV1)</a></li>
        <li>Chinese hamster <a href="../cgi-bin/hgTrackUi?db=criGriChoV2&g=ensGene&position=default"
         target=_blank>(criGriChoV2)</a></li>
        <li>Coelacanth <a href="../cgi-bin/hgTrackUi?db=latCha1&g=ensGene&position=default"
         target=_blank>(latCha1)</a></li>
        <li>Cow <a href="../cgi-bin/hgTrackUi?db=bosTau9&g=ensGene&position=default"
         target=_blank>(bosTau9)</a></li>
        <li>D. melanogaster <a href="../cgi-bin/hgTrackUi?db=dm6&g=ensGene&position=default"
         target=_blank>(dm6)</a></li>
  </ul></div>
  <div class="col-sm-4">
  <ul>
        <li>Dog <a href="../cgi-bin/hgTrackUi?db=canFam3&g=ensGene&position=default"
         target=_blank>(canFam3)</a></li>
        <li>Dog <a href="../cgi-bin/hgTrackUi?db=canFam5&g=ensGene&position=default"
         target=_blank>(canFam5)</a></li>
        <li>Elephant <a href="../cgi-bin/hgTrackUi?db=loxAfr3&g=ensGene&position=default"
         target=_blank>(loxAfr3)</a></li>
        <li>Elephant shark <a href="../cgi-bin/hgTrackUi?db=calMil1&g=ensGene&position=default"
         target=_blank>(calMil1)</a></li>
        <li>Ferret <a href="../cgi-bin/hgTrackUi?db=musFur1&g=ensGene&position=default"
         target=_blank>(musFur1)</a></li>
        <li>Gibbon <a href="../cgi-bin/hgTrackUi?db=nomLeu3&g=ensGene&position=default"
         target=_blank>(nomLeu3)</a></li>
        <li>Golden snub-nosed monkey <a href="../cgi-bin/hgTrackUi?db=rhiRox1&g=ensGene&position=default"
         target=_blank>(rhiRox1)</a></li>
        <li>Gorilla <a href="../cgi-bin/hgTrackUi?db=gorGor4&g=ensGene&position=default"
         target=_blank>(gorGor4)</a></li>
        <li>Green monkey <a href="../cgi-bin/hgTrackUi?db=chlSab2&g=ensGene&position=default"
         target=_blank>(chlSab2)</a></li>
        <li>Guinea pig <a href="../cgi-bin/hgTrackUi?db=cavPor3&g=ensGene&position=default"
         target=_blank>(cavPor3)</a></li>
        <li>Horse <a href="../cgi-bin/hgTrackUi?db=equCab3&g=ensGene&position=default"
         target=_blank>(equCab3)</a></li>
        <li>Lamprey <a href="../cgi-bin/hgTrackUi?db=petMar2&g=ensGene&position=default"
         target=_blank>(petMar2)</a></li>
        <li>Lizard <a href="../cgi-bin/hgTrackUi?db=anoCar2&g=ensGene&position=default"
         target=_blank>(anoCar2)</a></li>
        <li>Medium ground finch <a href="../cgi-bin/hgTrackUi?db=geoFor1&g=ensGene&position=default"
         target=_blank>(geoFor1)</a></li>
        <li>Microbat <a href="../cgi-bin/hgTrackUi?db=myoLuc2&g=ensGene&position=default"
         target=_blank>(myoLuc2)</a></li>
        <li>Mouse <a href="../cgi-bin/hgTrackUi?db=mm39&g=ensGene&position=default"
         target=_blank>(mm39)</a></li>
  </ul></div>
  <div class="col-sm-4">
  <ul>
        <li>Naked mole-rat <a href="../cgi-bin/hgTrackUi?db=hetGla1&g=ensGene&position=default"
         target=_blank>(hetGla1)</a></li>
        <li>Naked mole-rat <a href="../cgi-bin/hgTrackUi?db=hetGla2&g=ensGene&position=default"
         target=_blank>(hetGla2)</a></li>
        <li>Orangutan <a href="../cgi-bin/hgTrackUi?db=ponAbe2&g=ensGene&position=default"
         target=_blank>(ponAbe2)</a></li>
        <li>Panda <a href="../cgi-bin/hgTrackUi?db=ailMel1&g=ensGene&position=default"
         target=_blank>(ailMel1)</a></li>
        <li>Pig <a href="../cgi-bin/hgTrackUi?db=susScr11&g=ensGene&position=default"
         target=_blank>(susScr11)</a></li>
        <li>Rabbit <a href="../cgi-bin/hgTrackUi?db=oryCun2&g=ensGene&position=default"
         target=_blank>(oryCun2)</a></li>
        <li>Rat <a href="../cgi-bin/hgTrackUi?db=rn6&g=ensGene&position=default"
         target=_blank>(rn6)</a></li>
        <li>Rhesus <a href="../cgi-bin/hgTrackUi?db=rheMac10&g=ensGene&position=default"
         target=_blank>(rheMac10)</a></li>
        <li>S. cerevisiae <a href="../cgi-bin/hgTrackUi?db=sacCer3&g=ensGene&position=default"
         target=_blank>(sacCer3)</a></li>
        <li>Sheep <a href="../cgi-bin/hgTrackUi?db=oviAri3&g=ensGene&position=default"
         target=_blank>(oviAri3)</a></li>
        <li>Squirrel <a href="../cgi-bin/hgTrackUi?db=speTri2&g=ensGene&position=default"
         target=_blank>(speTri2)</a></li>
        <li>Squirrel monkey <a href="../cgi-bin/hgTrackUi?db=saiBol1&g=ensGene&position=default"
         target=_blank>(saiBol1)</a></li>
        <li>Tarsier <a href="../cgi-bin/hgTrackUi?db=tarSyr2&g=ensGene&position=default"
         target=_blank>(tarSyr2)</a></li>
        <li>Tetraodon <a href="../cgi-bin/hgTrackUi?db=tetNig2&g=ensGene&position=default"
         target=_blank>(tetNig2)</a></li>
        <li>Turkey <a href="../cgi-bin/hgTrackUi?db=melGal5&g=ensGene&position=default"
         target=_blank>(melGal5)</a></li>
        <li>X. tropicalis <a href="../cgi-bin/hgTrackUi?db=xenTro9&g=ensGene&position=default"
         target=_blank>(xenTro9)</a></li>
  </ul></div>
</div></div>
<p>
Thanks to Ensembl for providing these annotations, as well as
Hiram Clawson and Lou Nassar of the UCSC Genome Browser group for their
work in creating and releasing these tracks.</p>

<a name="061021"></a>
<h2>June 10, 2021 &nbsp;&nbsp; Updated data for NCBI RefSeq tracks for human assembly hg19</h2>
<p>
We are pleased to announce the updated  <a href="https://www.ncbi.nlm.nih.gov/genome/annotation_euk/
Homo_sapiens/109.20190905/#:~:text=NCBI%20Homo%20sapiens%20Updated%20Annotation%20Release%
20105.20201022&text=Updated%20Annotation%20Release%20105.20201022%20is,to%20update%20the%20annotated%20features"
target="_blank">NCBI RefSeq Annotation Release 105.20201022</a> tracks for the
<a href="/cgi-bin/hgTrackUi?db=hg19&g=refSeqComposite&position=default">hg19/GRCh37</a>
assembly. For all of these tracks, the alignments and coordinates are provided by RefSeq.
These tracks are organized in a composite track that includes:</p>
<ul>
  <li><b>RefSeq All</b> &ndash; all annotations from the curated and predicted sets
  <li><b>RefSeq Curated</b> &ndash; curated annotations beginning with NM, NR, or NP
  <li><b>RefSeq Other</b> &ndash; all other RefSeq annotations not included in RefSeq All
  <li><b>RefSeq Alignments</b> &ndash; alignments of transcripts to the genome provided by RefSeq
  <li><b>RefSeq Diffs</b> &ndash; alignment differences between the human reference genome(s) and RefSeq transcripts
  <li><b>RefSeq Select</b> &ndash; a single Select transcript representative for each protein-coding gene
  <li><b>RefSeq HGMD</b> &ndash; transcripts with clinical variants in Human Gene Mutation Database
</ul>
<p>
We would like to thank NCBI and the RefSeq Annotation database for collecting and curating these data.
We would also like to thank Hiram Clawson, Lou Nassar, and Gerardo Perez for developing and releasing these tracks.
</p>

<a name="060121"></a>
<h2>June 1, 2021 &nbsp;&nbsp; Genome Repository (GenArk) hubs and Dynamic BLAT</h2>
<p>
Do you have a favorite species or a research organism for which you'd like to visualize
the genomic sequence? With the recent addition of genomic assemblies for more than 1,300
species, it's possible that your favorite is now available in the UCSC Genome Browser.
Check <a  href="https://hgdownload.gi.ucsc.edu/hubs/">this list</a> of new assemblies 
to see if yours is a click away. Each of these Genome Browser assembly hubs comes pre-loaded 
with several annotation tracks, gene models, and the ability to align genomic sequence to 
the reference assembly using the <a  href="/FAQ/FAQblat.html">BLAT</a> 
alignment tool. You can add your own data to your view of the assembly hub by uploading a 
<a  href="help/customTrack.html">custom track</a> 
or creating a <a href="help/hgTrackHubHelp.html">track hub</a>.</p>
<p>
We call this group of assembly hubs, the <a href=https://hgdownload.gi.ucsc.edu/hubs/>
Genome Repository (GenArk)</a>. The genomes in the GenArk are sourced from 
<a href=https://www.ncbi.nlm.nih.gov/refseq/ target="_blank">NCBI RefSeq</a>, 
the  <a href="https://vertebrategenomesproject.org/" target="_blank">Vertebrate Genomes Project (VGP)</a> 
and other projects. See the complete list of assembly hubs <a target="_blank"
href="https://hgdownload.gi.ucsc.edu/hubs/UCSC_GI.assemblyHubList.txt">here</a>.</p>
<p>
GenArk assemblies can be found by searching the common name (e.g.
 <a href="../h/GCF_001984765.1">American Beaver</a>) or GCA/GCF accession
(e.g. <a href="../h/GCF_001984765.1">GCF_001984765.1</a>) directly on the
<a href="/cgi-bin/hgGateway">Genome Browser Gateway page</a>.
A full list can also be seen on the <a 
href="https://hgdownload.gi.ucsc.edu/hubs">GenArk portal</a> which
is organized by phylogenetic groups and projects:<br>

<div align="left">
<table class="stdTbl">
 <tr><td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/primates/index.html">Primates</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/primates/asmStats.html">Assembly statistics</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/primates/trackData.html">Track statistics</a></td></tr>

 <tr><td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/mammals/index.html">Mammals</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/mammals/asmStats.html">Assembly statistics</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/mammals/trackData.html">Track statistics</a></td></tr>

 <tr><td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/birds/index.html">Birds</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/birds/asmStats.html">Assembly statistics</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/birds/trackData.html">Track statistics</a></td></tr>

 <tr><td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/fish/index.html">Fishes</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/fish/asmStats.html">Assembly statistics</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/fish/trackData.html">Track statistics</a></td></tr>

 <tr><td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/vertebrate/index.html">Vertebrates</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/vertebrate/asmStats.html">Assembly statistics</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/vertebrate/trackData.html">Track statistics</a></td></tr>

 <tr><td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/invertebrate/index.html">Invertebrates</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/invertebrate/asmStats.html">Assembly statistics</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/invertebrate/trackData.html">Track statistics</a></td></tr>

 <tr><td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/fungi/index.html">Fungi</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/fungi/asmStats.html">Assembly statistics</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/fungi/trackData.html">Track statistics</a></td></tr>

 <tr><td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/plants/index.html">Plants</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/plants/asmStats.html">Assembly statistics</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/plants/trackData.html">Track statistics</a></td></tr>

 <tr><td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/VGP/index.html">VGP</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/VGP/asmStats.html">Assembly statistics</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/VGP/trackData.html">Track statistics</a></td></tr>

 <tr><td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/legacy/index.html">Legacy</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/legacy/asmStats.html">Assembly statistics</a></td>
 <td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/legacy/trackData.html">Track statistics</a></td></tr>

 <tr><td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/globalReference/index.html">Global Reference</a></td>
 <td><a></a></td>
 <td><a></a></td></tr>

 <tr><td><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/mouseStrains/hubIndex.html">Mouse Strains</a></td>
 <td><a></a></td>
 <td><a></a></td></tr>

</table></div></p>
<p>
The number of NCBI RefSeq assemblies has been increasing each year, and we expect the trend to continue as assemblies 
from VGP, and other projects are released. Our next step is to add additional clades to GenArk including archaea,
viruses, and bacteria.</p>

<p>
As mentioned above, assembly hubs in the GenArk allow for sequence alignment using the BLAT tool. We extended our traditional
<a href="../cgi-bin/hgBlat?command=start">BLAT tool</a> to accommodate these assembly hubs and we call it Dynamic BLAT. Dynamic BLAT
pre-computes index genomes files, which allows assemblies to have a BLAT server without needing it to be resident in the 
computer's memory. Dynamic BLAT allows us to offer BLAT services on nearly all of our GenArk assemblies, with a few exceptions
due to excessive genome size.</p>

<p>
We would like to thank NCBI RefSeq and Vertebrate Genome Project (VGP) for making these assemblies available. We would also 
like to thank Hiram Clawson, Mark Diekhans, Galt Barber, Lou Nassar, and Gerardo Perez, and the rest of the Browser team 
for their work on building the assembly hubs, creating GenArk, and extending BLAT to include these new assembly hubs.</p>


<a name="051121"></a>
<h2>May 11, 2021 &nbsp;&nbsp; Vaccines track available for the SARS-CoV-2 Genome Browser</h2>
<p>
We are happy to announce the release of the
<a href="../cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=vaccines">Vaccines</a> track for the UCSC
SARS-CoV-2 Genome Browser. This track shows the alignment of three different mRNA vaccine sequences
to the coronavirus genome:
</p>
<ol>
  <li>
    The BioNTech/Pfizer BNT-162b2 sequence as published by the World Health Organization (WHO)</li>
  <li>
    The reconstructed BioNTech/Pfizer BNT-162b2 RNA as sequenced by the
    <a href="https://firelab.sites.stanford.edu/" target="_blank">Andrew Fire lab</a>, Stanford
    University School of Medicine</li>
  <li>
    The Moderna mRNA-1273 sequence as sequenced by the
    <a href="https://firelab.sites.stanford.edu/" target="_blank">Andrew Fire lab</a>, Stanford
    University School of Medicine</li>
</ol>
<p>
The protein encoded by the three sequences has two amino acid substitutions compared to the
SARS-CoV-2 S glycoprotein. These vaccines include two variations compared to the reference Spike
sequence to stabilize the shape into a pre-infection structure that delivers a better immune
response, S:K986P and S:V987P. See also:
<a href="https://cen.acs.org/pharmaceuticals/vaccines/tiny-tweak-behind-COVID-19/98/i38"
target="_blank">The tiny tweak behind COVID-19 vaccines</a>. More information about this track can
be found on the <a href="../cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=vaccines">description
page</a>.
<p>
We want to thank the <a href="https://twitter.com/Fire__Lab" target="_blank">Andrew Fire lab</a>,
Stanford University School of Medicine for providing the sequencing data of these vaccines. We also
want to thank Hiram Clawson and Jairo Navarro for the creation and release of this track.</p>

<a name="050421"></a>
<h2>May 04, 2021 &nbsp;&nbsp; New Genome Browser: Rat, rn7 (<em>Rattus norvegicus</em>)</h2>
<p>
A genome browser is now available for the <em>Rattus norvegicus</em> assembly released in Nov. 2020 by the
<a href="https://www.darwintreeoflife.org/" target="_blank"> Darwin Tree of Life Project</a> at the
<a href="https://www.sanger.ac.uk/" target="_blank"> Wellcome Sanger Institute</a> (mRatBN7.2).
<h3>About the assembly:</h3>
<ul>
  <li><b>UCSC Genome Browser assembly ID:</b>  <a href="../../cgi-bin/hgGateway?db=rn7" 
  target="_blank">rn7</a></li>
  <li><b>Sequencing/Assembly provider ID:</b> <a href="https://www.sanger.ac.uk/"
  target="_blank">Wellcome Sanger Institute</a></li>
  <li><b>Assembly date:</b> Nov. 2020</li>
  <li><B>Accession ID:</B> GCF_015227675.2</li>
  <li><B>NCBI Genome ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/genome/73'
TARGET='_blank'>73</A> (<em>Rattus norvegicus</em>)</li>
  <li><B>NCBI Assembly ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/assembly/9079111'
TARGET='_blank'>9079111</A></li>
  <li><B>NCBI BioProject ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/bioproject/662791'
TARGET='_blank'>662791</A></li>
  <li><B>NCBI BioSample ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/biosample/SAMN16261960'
TARGET='_blank'>SAMN16261960</A></li>
  <li><b>Scaffold count: </b>176</li>
  <li><b>Total size: </b>2,647,915,728</li>
  <li><a href="../goldenPath/credits.html#rat_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
  <li><a href="../goldenPath/credits.html#rat_use" 
target="_blank" >Data use conditions and restrictions</a></li>
  <li><a href="../goldenPath/help/ftp.html" 
target="_blank" >Rsync</a> and
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/rn7/" 
target="_blank" >FTP</a></li>
  <li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#rat"
target="_blank" >Downloads page</a></li>
</ul>
</p>
<p>
We would like to thank the Darwin Tree of Life Project at the Wellcome Sanger Institute for 
creating this new rat assembly. We would also like to thank Hiram Clawson, Lou Nassar, and 
Gerardo Perez for the development and release of the rn7 genome browser.
</p>

<a name="041321"></a>
<h2>Apr. 13, 2021 &nbsp;&nbsp; New Haploinsufficiency predictions for human (hg19)</h2>
<p>
We are happy to announce a new haploinsufficiency predictions track for human assembly
<a href="/cgi-bin/hgTrackUi?db=hg19&c=chrX&g=decipherHaploIns">GRCh37/hg19</a>. This track displays
haploinsufficiency predictions for human genes from
<a href="https://www.deciphergenomics.org/" target="_blank">DECIPHER</a>. Human cells have two
copies of most genes, one from each parent. If a mutation alters one allele, or copy, the other is
usually still sufficient to maintain proper gene function. For haploinsufficient genes, however,
both copies must be functioning for the organism to have a normal life. Haploinsufficiency is
implicated in a number of health disorders.
</p>
<p>
Moving the mouse cursor over any gene in the track will display a pop-up box with the gene name
and &alpha;-upper quantile for the prediction. The genes in this track are also color-coded
according to quantile:
</p>
<ul>
  <li>
    <b><span style='color: #AA00AA'>Magenta</span></b> shades indicate a higher expectation of being
    haploinsufficient</li>
  <li>
    <b><span style='color: #00AA00'>Green</span></b> shades indicate a lower expectation of being
    haploinsufficient</li>
</ul>
<p>
We would like to thank the DECIPHER team for making the haploinsufficiency prediction data
available. We also want to thank Jonathan Casper and Jairo Navarro for the creation and release of
this track.
</p>

<a name="040121"></a>
<h2>Apr. 01, 2021 &nbsp;&nbsp; Update to sno/miRNA track for human (hg19)</h2>
<p>
We are happy to announce an update to the sno/miRNA track for human assembly
<a href="/cgi-bin/hgTrackUi?db=hg19&c=chrX&g=wgRna">GRCh37/hg19</a>. With this update, the miRBase
database was updated to version 20. The four different types of RNA and their counts in the human
genome are listed below.
</p>
<ul>
  <li>
  C/D box small nucleolar RNAs (C/D box snoRNAs): 269</li>
  <li>
  H/ACA box snoRNAs: 112</li>
  <li>
  precursor forms of microRNAs (pre-miRNAs): 1871</li>
  <li>
  small Cajal body-specific RNAs (scaRNAs): 21</li>
</ul>
<p>
We would like to thank Hiram Clawson and Jairo Navarro for the creation and release of this update.
</p>

<a name="033021"></a>
<h2>Mar. 30, 2021 &nbsp;&nbsp; Updated Chain/Net tracks for Mouse (mm39/GRCm39)</h2>
<p>We are pleased to announce the update to the <a target="_blank"
href="../../cgi-bin/hgTrackUi?db=mm39&g=mm39ChainNet&position=default">Chain/Net</a> tracks
for the recently released mouse assembly
<a target="_blank" href="../../cgi-bin/hgTracks?db=mm39" >mm39/GRCm39</a>.
The chain track shows alignments of the mouse assembly to 29 other
organisms using a gap scoring system that allows longer gaps than traditional affine gap scoring
systems. The net track shows the best mouse/other chain for every part of the other genome.</p><br>
<div class="container">
 <div class="row">
  <div class="col-sm-4">
  <ul>
     <li>Bonobo <a href="../cgi-bin/hgTrackUi?db=panPan3&g=chainNetMm39&position=default"
     target=_blank>(panPan3)</a></li>
     <li>Bushbaby <a href="../cgi-bin/hgTrackUi?db=otoGar3&g=chainNetMm39&position=default"
     target=_blank>(otoGar3)</a></li>
     <li>Chicken <a href="../cgi-bin/hgTrackUi?db=galGal6&g=chainNetMm39&position=default"
     target=_blank>(galGal6)</a></li>
     <li>Chimp <a href="../cgi-bin/hgTrackUi?db=panTro6&g=chainNetMm39&position=default"
     target=_blank>(panTro6)</a></li>
     <li>Chinese pangolin <a href="../cgi-bin/hgTrackUi?db=manPen1&g=chainNetMm39&position=default"
     target=_blank>(manPen1)</a></li>
     <li>Cow <a href="../cgi-bin/hgTrackUi?db=bosTau9&g=chainNetMm39&position=default"
     target=_blank>(bosTau9)</a></li>
     <li>Dolphin <a href="../cgi-bin/hgTrackUi?db=turTru2&g=chainNetMm39&position=default"
     target=_blank>(turTru2)</a></li>
     <li>Elephant <a href="../cgi-bin/hgTrackUi?db=loxAfr3&g=chainNetMm39&position=default"
     target=_blank>(loxAfr3)</a></li>
     <li>Gorilla <a href="../cgi-bin/hgTrackUi?db=gorGor6&g=chainNetMm39&position=default"
     target=_blank>(gorGor6)</a></li>
     <li>Guinea pig <a href="../cgi-bin/hgTrackUi?db=cavPor3&g=chainNetMm39&position=default"
     target=_blank>(cavPor3)</a></li>
  </ul>
  </div>
  <div class="col-sm-4">
  <ul>
     <li>Hawaiian monk seal <a href="../cgi-bin/hgTrackUi?db=neoSch1&g=chainNetMm39&position=default"
     target=_blank>(neoSch1)</a></li>
     <li>Hedgehog <a href="../cgi-bin/hgTrackUi?db=eriEur2&g=chainNetMm39&position=default"
     target=_blank>(eriEur2)</a></li>
     <li>Horse <a href="../cgi-bin/hgTrackUi?db=equCab3&g=chainNetMm39&position=default"
     target=_blank>(equCab3)</a></li>
     <li>Lamprey <a href="../cgi-bin/hgTrackUi?db=petMar3&g=chainNetMm39&position=default"
     target=_blank>(petMar3)</a></li>
     <li>Malayan flying lemur <a href="../cgi-bin/hgTrackUi?db=galVar1&g=chainNetMm39&position=default"
     target=_blank>(galVar1)</a></li>
     <li>Opossum <a href="../cgi-bin/hgTrackUi?db=monDom5&g=chainNetMm39&position=default"
     target=_blank>(monDom5)</a></li>
     <li>Pig <a href="../cgi-bin/hgTrackUi?db=susScr3&g=chainNetMm39&position=default"
     target=_blank>(susScr3)</a></li>
     <li>Pika <a href="../cgi-bin/hgTrackUi?db=ochPri3&g=chainNetMm39&position=default"
     target=_blank>(ochPri3)</a></li>
     <li>Rabbit <a href="../cgi-bin/hgTrackUi?db=oryCun2&g=chainNetMm39&position=default"
     target=_blank>(oryCun2)</a></li>
     <li>Rat <a href="../cgi-bin/hgTrackUi?db=rn6&g=chainNetMm39&position=default"
     target=_blank>(rn6)</a></li>
  </ul>
  </div>
  <div class="col-sm-4">
  <ul>
     <li>Rhesus <a href="../cgi-bin/hgTrackUi?db=rheMac10&g=chainNetMm39&position=default"
     target=_blank>(rheMac10)</a></li>
     <li>Sheep <a href="../cgi-bin/hgTrackUi?db=oviAri4&g=chainNetMm39&position=default"
     target=_blank>(oviAri4)</a></li>
     <li>Shrew <a href="../cgi-bin/hgTrackUi?db=sorAra2&g=chainNetMm39&position=default"
     target=_blank>(sorAra2)</a></li>
     <li>Squirrel <a href="../cgi-bin/hgTrackUi?db=speTri2&g=chainNetMm39&position=default"
     target=_blank>(speTri2)</a></li>
     <li>Tarsier <a href="../cgi-bin/hgTrackUi?db=tarSyr2&g=chainNetMm39&position=default"
     target=_blank>(tarSyr2)</a></li>
     <li>Tenrec <a href="../cgi-bin/hgTrackUi?db=echTel2&g=chainNetMm39&position=default"
     target=_blank>(echTel2)</a></li>
     <li>Tree shrew <a href="../cgi-bin/hgTrackUi?db=tupBel1&g=chainNetMm39&position=default"
     target=_blank>(tupBel1)</a></li>
     <li>X. tropicalis <a href="../cgi-bin/hgTrackUi?db=xenTro9&g=chainNetMm39&position=default"
     target=_blank>(xenTro9)</a></li>
     <li>Zebrafish <a href="../cgi-bin/hgTrackUi?db=danRer11&g=chainNetMm39&position=default"
     target=_blank>(danRer11)</a></li>
  </ul>
  </div>
 </div>
</div>

<p>
We would like to thank Angie Hinrichs, Hiram Clawson, Lou Nassar, and Gerardo Perez for the
development and release of these tracks.</p>

<a name="032321"></a>
<h2>Mar. 23, 2021 &nbsp;&nbsp; New GENCODE gene tracks: Human V37 (hg19/hg38) - Mouse M26 
(mm39)</h2>
<p>
We are pleased to announce new GENCODE Gene annotation tracks, which correspond
to <a href="https://www.ensembl.info/2021/02/15/ensembl-103-has-been-released/"
target="_blank">Ensembl 103</a>, for three assemblies: <a target="_blank" 
href="../../cgi-bin/hgTrackUi?db=hg19&c=chr21&g=wgEncodeGencodeV37lift37">hg19/GRCh37</a>,
<a target="_blank" href="../../cgi-bin/hgTrackUi?db=hg38&c=chr1&g=wgEncodeGencodeV37">hg38/GRCh38</a>,
and <a target="_blank" href="../../cgi-bin/hgTrackUi?db=mm39&c=chr12&g=wgEncodeGencodeVM26">mm39/GRCm39</a>.
For human, the GENCODE V37 annotations were mapped to hg38/GRCh38 and then back-mapped
to the hg19/GRCh37 assembly. For all three assemblies, the gene sets contain the following
tracks:</p>
<ul>
  <li>Basic - a subset of the <em>Comprehensive set</em>.</li>
  <li>Comprehensive - all GENCODE coding and non-coding transcript annotations, including
      polymorphic pseudogenes. This includes both manual and automatic annotations.</li>
  <li>Pseudogenes - all annotations except polymorphic pseudogenes.</li></ul>
<p>
The hg38 and mm39 assemblies also include the following tracks that are not available on hg19:
</p>
<ul>
  <li>2-way Pseudogenes - pseudogenes predicted by both the Yale Pseudopipe and UCSC
      Retrofinder pipelines.</li>
  <li>PolyA - polyA signals and sites manually annotated on the genome based on transcribed
      evidence (ESTs and cDNAs) of 3' end of transcripts containing at least 3 A's not matching the
      genome.</li></ul>
<p>
Details on each release can be found on the <a target="_blank" 
href="https://www.gencodegenes.org/">GENCODE site</a>. This includes <a target="_blank" 
href="https://www.gencodegenes.org/human/stats.html">statistics on each release</a>.
<p>We would like to thank the <a target="_blank" 
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Mark Diekhans and Lou Nassar for the development and
release of these tracks.</p>

<a name="031821"></a>
<h2>Mar. 18, 2021 &nbsp;&nbsp; Updated transcription start miRNA sites for human</h2>
<p>
We are happy to announce an update to the TS miRNA Targets track for human,
<a href="/cgi-bin/hgTrackUi?db=hg19&c=chrX&g=targetScanMiRnaSitesV72" >GRCh37/hg19</a>.
This track shows conserved mammalian microRNA regulatory target sites for conserved microRNA
families in the 3' UTR regions of Refseq Genes, as predicted by TargetScanHuman 7.2.</p>

<p>
The track items are colored based on the four classes of target sites identified in Agarwal
<em>et al.</em>, 2015.</p>
<table cellpadding="2">
  <tbody>
    <tr>
      <td>Purple</td>
      <td style="background-color: #800080; padding-left: 20px"></td>
      <td>8mer</td>
    </tr>
    <tr>
      <td>Red</td>
      <td style="background-color: #ff0000; padding-left: 20px"></td>
      <td>7mer-m8</td>
    </tr>
    <tr>
      <td>Blue</td>
      <td style="background-color: #1e90ff; padding-left: 20px"></td>
      <td>7mer-A1</td>
    </tr>
    <tr>
      <td>Green</td>
      <td style="background-color: #00ff00; padding-left: 20px"></td>
      <td>Non-canonical</td>
    </tr>
  </tbody>
</table>
<p>
Thanks to George Bell of
<a href="http://barc.wi.mit.edu/" target="_blank">Bioinformatics and Research Computing</a> at the
Whitehead Institute for providing this annotation, which was generated in collaboration with the
labs of David Bartel and Chris Burge. Additional information on microRNA target prediction is
available on the <a href="http://www.targetscan.org/" target="_blank">TargetScan</A> website. We
would also like to thank UCSC Genome Browser team members Brittney Wick, Kate Rosenbloom, and Jairo
Navarro for the development and release of this track.</p>

<a name="031721"></a>
<h2>Mar. 17, 2021 &nbsp;&nbsp; Updated hg38 default gene track to GENCODE v36</h2>
<p>
We have updated the most recent human genome, GRCh38/hg38, with an updated default gene set,
<a href="../cgi-bin/hgTrackUi?db=hg38&g=knownGene">GENCODE v36</a>. This is part of our commitment
to keep our annotation data as up-to-date as possible with the most 
useful information, all versioned for stability and reproducibility. This geneset 
replaces our previous default, GENCODE v32, with a more recent dataset while retaining all 
of the filters, viewing options, and data access abilities. GENCODE v32 can still be 
viewed in the <a href="../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeGencodeSuper">
All GENCODE</a> track. You can view the new default geneset at the following link:
<a href="../cgi-bin/hgTracks?db=hg38&hideTracks=1&knownGene=pack">GENCODE v36</a></p>

<p>
The default gene dataset is called &quot;knownGene&quot; internally at UCSC and is 
used in many bioinformatics pipelines as a reference geneset. 
The GRCh37/hg19 default geneset will stay the same.
The UCSC Genome Browser would like to thank the <a href="https://www.gencodegenes.org/"
 target="_blank">GENCODE consortium</a>
and the internal development team at UCSC, including Daniel Schmelter and Brian Raney.</p>

<a name="030421"></a>
<h2>Mar. 4, 2021 &nbsp;&nbsp; Updated NCBI RefSeq Gene Tracks for 13 organisms</h2>
<p>
We are pleased to announce a combination of both updated and new
<a href="https://www.ncbi.nlm.nih.gov/refseq/" target="_blank">
NCBI RefSeq gene annotation</a> tracks for the
following 13 assemblies. Most notably, we have updated both human assemblies and the new 
mouse assembly.</p><br>
<div class="container">
 <div class="row">
  <div class="col-sm-3">
  <ul>
     <li>Human <a href="../cgi-bin/hgTrackUi?g=refSeqComposite&db=hg19">(hg19)</a></li>
     <li>Human <a href="../cgi-bin/hgTrackUi?g=refSeqComposite&db=hg38">(hg38)</a></li>
     <li>Mouse <a href="../cgi-bin/hgTrackUi?g=refSeqComposite&db=mm39">(mm39)</a></li>
     <li>Worm <a href="../cgi-bin/hgTrackUi?g=refSeqComposite&db=ce11">(ce11)</a></li>
  </ul>
  </div>
  <div class="col-sm-3">
  <ul>
     <li>Fruit fly <a href="../cgi-bin/hgTrackUi?g=refSeqComposite&db=dm6">(dm6)</a></li>
     <li>Turkey <a href="../cgi-bin/hgTrackUi?g=refSeqComposite&db=melGal5">(melGal5)</a></li>
     <li>Sheep <a href="../cgi-bin/hgTrackUi?g=refSeqComposite&db=oviAri4">(oviAri4)</a></li>
     <li>Yeast <a href="../cgi-bin/hgTrackUi?g=refSeqComposite&db=sacCer3">(sacCer3)</a></li>
  </ul>
  </div>
  <div class="col-sm-3">
  <ul>
     <li>Frog <a href="../cgi-bin/hgTrackUi?g=refSeqComposite&db=xenTro9">(xenTro9)</a></li>
     <li>Pig <a href="../cgi-bin/hgTrackUi?g=refSeqComposite&db=susScr11">(susScr11)</a></li>
     <li>Chimp <a href="../cgi-bin/hgTrackUi?g=refSeqComposite&db=panTro5">(panTro5)</a></li>
     <li>Green Monkey <a href="../cgi-bin/hgTrackUi?g=refSeqComposite&db=chlSab2">(chlSab2)</a></li>
  </ul>
  </div>
  <div class="col-sm-3">     
  <ul>
     <li>Chinese Hamster Ovary <a href="../cgi-bin/hgTrackUi?g=refSeqComposite&db=criGriChoV1">(criGriChoV1)</a></li>
  </ul>
  </div>
 </div>
</div>

<p>
Thanks to <a target="_blank" href="https://www.ncbi.nlm.nih.gov/refseq/">
NCBI RefSeq</a> for providing these annotations. We would also like to thank Hiram Clawson
and Daniel Schmelter of the UCSC Genome Browser group for their
work in creating and releasing these tracks.</p>

<a name="030221"></a>
<h2>Mar. 02, 2021 &nbsp;&nbsp; CADD Scores now available for hg19 and hg38</h2>
<p>
We are happy to announce the release of a new track for human assemblies, 
<a href="../cgi-bin/hgTrackUi?db=hg19&chr=chrX&g=caddSuper">hg19</a> and
<a href="../cgi-bin/hgTrackUi?db=hg38&chr=chrX&g=caddSuper">hg38</a>. The new track shows
<a href="https://cadd.gs.washington.edu/" target="_blank">Combined Annotation Dependent Depletion
(CADD)</a> scores for each base in the genome and corresponds to the v1.6 developmental release.
CADD is a tool for scoring the deleteriousness of single nucleotide variants as well as
insertion/deletion variants in the human genome. CADD scores strongly correlate with allelic
diversity, pathogenicity of both coding and non-coding variants, experimentally measured regulatory
effects, and also highly rank causal variants within individual genome sequences. Finally, CADD
scores of complex trait-associated variants from genome-wide association studies (GWAS) are
significantly higher than matched controls and correlate with study sample size, likely reflecting
the increased accuracy of larger GWAS.</p>

<p>The supertrack consists of the following tracks:</p>
<ul>
  <li><strong>CADD</strong> (composite track)
    <ul>
      <li>Mutation: A</li>
      <li>Mutation: C</li>
      <li>Mutation: G</li>
      <li>Mutation: T</li>
    </ul>
  </li>
  <li><strong>Deletions</strong></li>
  <li><strong>Insertions</strong></li>  
</ul>

<p>
CADD scores are freely available for all non-commercial applications from the
<a href="https://cadd.gs.washington.edu/download" target="_blank">CADD website</a>. For commercial
applications, see the
<a href="https://cadd.gs.washington.edu/contact" target="_blank">license instructions</a> there.</p>

<p>
We would like to thank the CADD development team for providing precomputed data as simple
tab-separated files. We would also like to thank Max Haeussler and Jairo Navarro for the creation
and release of these tracks.</p>

<a name="021821"></a>
<h2>Feb. 18, 2021 &nbsp;&nbsp; New Genome Browser: Marmoset, calJac4</h2>
<p>
A genome browser is now available for the marmoset (<em>Callithrix jacchus</em>) assembly released
in May 2020 by the <a href="https://www.genome.wustl.edu/" target="_blank">McDonnell Genome
Institute at Washington University</a> (Callithrix_jacchus_cj1700_1.1).</p>

<h3>About the assembly:</h3>
<ul>
  <li><b>UCSC Genome Browser assembly ID:</b>
    <a href="../../cgi-bin/hgGateway?db=calJac4" target="_blank">calJac4</a>
  </li>
  <li><b>Sequencing/Assembly provider ID:</b>
    <a href="https://www.genome.wustl.edu/" target="_blank">McDonnell Genome Institute at
    Washington University</a>
  </li>
  <li><b>Assembly date:</b> May 2020</li>
  <li><b>Accession ID:</b> GCF_009663435.1</li>
  <li><b>NCBI Genome ID:</b>
    <a href="https://www.ncbi.nlm.nih.gov/genome/442" target="_blank">442</a>
    (<em>Callithrix jacchus</em>)
  </li>
  <li><b>NCBI Assembly ID:</b>
    <a href="https://www.ncbi.nlm.nih.gov/assembly/7157801" target="_blank">7157801</a>
  </li>
  <li><b>NCBI BioProject ID:</b>
    <a href="https://www.ncbi.nlm.nih.gov/bioproject/566173" target="_blank">566173</a>
  </li>
  <li><b>NCBI BioSample ID:</b>
    <a href="https://www.ncbi.nlm.nih.gov/biosample/SAMN12783337" target="_blank">SAMN12783337</a>
  </li>
  <li><b>Scaffold count:</b> 994</li>
  <li><b>Total size:</b> 2,897,824,427</li>
  <li>
    <a href="../goldenPath/credits.html#marmoset_credits" target="_blank">Acknowledgments</a>
  </li>
</ul>

<h3>Data and Downloads:</h3>
<ul>
  <li>
    <a href="../goldenPath/credits.html#marmoset_use" target="_blank">Data use conditions and
    restrictions</a>
  </li>
  <li>
    <a href="../goldenPath/help/ftp.html" target="_blank">Rsync</a> and
    <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/calJac4/" target="_blank">FTP</a>
  </li>
  <li>
    <a href="http://hgdownload.gi.ucsc.edu/downloads.html#marmoset" target="_blank">Downloads
    page</a>
  </li>
</ul>

<a name="020321"></a>
<h2>Feb. 3, 2021 &nbsp;&nbsp; New Exome Sequencing Probesets for human (hg38/hg19)</h2>
<p>
In our effort to better support personalized medicine and clinical genetics researchers,
we are happy to announce the addition of 78 new whole exome sequencing (WES) probes and targets 
tracks from several major companies for human genomes hg38 and hg19. Our new tracks 
include data on WES kits from Illumina, Agilent, Roche, IDT, Twist, and MGI (BGI).
Understanding exome sequencing probes and targets helps bring awareness to the different
gene isoforms and alternative splicing that may be important for disease research.
You can find these tracks under the &quot;Mapping and Sequencing&quot; group or go directly
to the description pages of the &quot;Exome Probesets&quot; tracks below:</p>
<ul>
<li><a href="/cgi-bin/hgTrackUi?db=hg38&g=exomeProbesets">Exome Probesets for hg38</a></li>
<li><a href="/cgi-bin/hgTrackUi?db=hg19&g=exomeProbesets">Exome Probesets for hg19</a></li>
</ul>
<p>
Thank you to those who contributed to these tracks, including Ana Benet, Max Hauessler, 
Daniel Schmelter, Beagan Nguy, Tiana Pereira, and Pranav Muthuraman.</p>

<a name="020221"></a>
<h2>Feb. 2, 2021 &nbsp;&nbsp; New Conservation track for Mouse (mm39/GRCm39)</h2>
<p>
We are pleased to announce new conservation tracks for
the recently released mouse assembly (mm39/GRCm39). The tracks display multiple 
alignments (multiZ) for 35 different organisms and evolutionary conservation 
scores (phastCons and phyloP).
For more details, visit the track description page below.</p>
<ul>
<li><a href="/cgi-bin/hgTrackUi?db=mm39&g=cons35way">Conservation (35-species on mm39)</a> - 
This track shows pairwise genome alignment scores for 35 vertebrates at each position 
on the mouse genome, aggregate conservation scores, and regions predicted to be 
Conserved Elements.</li>
</ul>

<a name="012921"></a>
<h2>Jan. 29, 2021 &nbsp;&nbsp; January 29th SARS-CoV-2 Update</h2>
<p>
The UCSC Genome Browser is proud to share our sixth round of genetics data for the 
SARS-CoV-2 virus (see our <a
href="#040320">first</a>, <a href="#050420">second</a>, <a href="#080720">third</a>,
<a href="#102120">fourth</a>, and <a href="010921">fifth</a> releases). 
These data focus on the virus's mutations, evolution, and antibody 
responses. As usual, these data are all open-access and interactive. For more information,
you can also visit our <a href="../covid19.html">COVID-19 home page</a>.</p>
<p>
Clicking on any of the track titles below will lead to the track description page, 
which includes additional information and allows for configuration of various 
display options. This release includes the following tracks:</p>
<ul>
<li><a href="/cgi-bin/hgTrackUi?db=wuhCor1&g=sarsCov2PhyloPub">
Phylogeny: Public</a> - Updated 1/18/21 to contain more than 227k SARS-CoV-2 
genome sequences from <a 
href="https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/virus?SeqType_s=Nucleotide&VirusLineage_ss=SARS-CoV-2,%20taxid:2697049" 
target="_blank">GenBank</a>, <a href="https://www.cogconsortium.uk/data/" 
target="_blank">COG-UK</a>, the <a href="https://bigd.big.ac.cn/ncov/release_genome" 
target="_blank">China National Center for Bioinformation</a>, and an inferred phylogenetic 
tree. Since this track 
is sourced from public databases, we offer 
<a href="https://en.wikipedia.org/wiki/Variant_Call_Format">VCF</a> files with mutations for 
download.</li>
<li><a href="/cgi-bin/hgTrackUi?db=wuhCor1&g=spikeMuts">Spike protein mutations</a> - 
Detailed annotations on spike protein mutations, based on community annotation from 
mid-January 2021 at <a href="https://covariants.org/" target="_blank">
CoVariants.org</a>, 
supplemented by the Variants of SARS-CoV-2 page at Wikipedia. 
Linked information includes date first sequenced, notes of clinical significance and 
geographic incidence, and links to Nextstrain build, incidence charts and relevant publications. 
Mutations in this track include: H69-, D80Y, S98F, A222V, N439K, Y453F, S477N, E484, 
N501, D614G, A626S, P681H, A701B, V1122.</li>
<li><a href="/cgi-bin/hgTrackUi?db=wuhCor1&g=abEscape">Antibody escape mutation</a> - 
Antibody binding scores from multiple 
labs across all possible mutations of the Spike protein's receptor-binding domain (RBD). These 
data come from experiments using patient serum antibodies or monoclonal antibodies and a 
yeast-surface display system.</li>
</ul>

<a name="012621"></a>
<h2>Jan. 26, 2021 &nbsp;&nbsp; New Genome Browser: Dog, canFam5 (<em>Canis lupus familiaris</em>)</h2>
<p>
A genome browser is now available for the <em>Canis lupus familiaris</em> assembly released in May 2019 by the
<a href="https://umich.edu/" target="_blank"> University of Michigan</a> (UMICH_Zoey_3.1).
<h3>About the assembly:</h3>
<ul>
  <li><b>UCSC Genome Browser assembly ID:</b>  <a href="../../cgi-bin/hgGateway?db=canFam5" 
  target="_blank">canFam5</a></li>
  <li><b>Sequencing/Assembly provider ID:</b> <a href="https://umich.edu/">
University of Michigan</a></li>
  <li><b>Assembly date:</b> May 2019</li>
  <li><B>Accession ID:</B> GCA_005444595.1</li>
  <li><B>NCBI Genome ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/genome/85'
TARGET='_blank'>85</A> (<em>Canis lupus familiaris</em>)</li>
  <li><B>NCBI Assembly ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/assembly/3218611'
TARGET='_blank'>3218611</A></li>
  <li><B>NCBI BioProject ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/bioproject/318403'
TARGET='_blank'>318403</A></li>
  <li><B>NCBI BioSample ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/biosample/SAMN04851098'
TARGET='_blank'>SAMN04851098</A></li>
  <li><b>Scaffold count: </b>794</li>
  <li><b>Total size: </b>2,343,218,756</li>
  <li><a href="../goldenPath/credits.html#dog_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
  <li><a href="../goldenPath/credits.html#dog_use" 
target="_blank" >Data use conditions and restrictions</a></li>
  <li><a href="../goldenPath/help/ftp.html" 
target="_blank" >Rsync</a> and
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/canFam5/" 
target="_blank" >FTP</a></li>
  <li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#dog"
target="_blank" >Downloads page</a></li>
</ul>
</p>

<a name="012221"></a>
<h2>Jan. 22, 2021 &nbsp;&nbsp; Updates to gnomAD variation track (GRCh37/hg19)(GRCh38/hg38)</h2>
<p>
We are pleased to announce updates to the gnomAD Variants super-track for the <a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg19&c=chrX&g=gnomadSuper">hg19/GRCh37</a>
and <a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=gnomadVariants">hg38/GRCh38</a>
human assemblies.</p>
<ul>
  <li>gnomAD v3.1 is now available and contains variants from 76,156 whole genomes (and no exomes),
      all mapped to the GRCh38/hg38 reference sequence. 4,454 genomes were added to the number of
      genomes in the previous v3 release. For more detailed information on gnomAD v3.1, see the related
      <a href="https://gnomad.broadinstitute.org/blog/2020-10-gnomad-v3-1/"target="_blank">blog post</a>.</li>
  <li>The latest release (gnomAD 2.1.1) is also available for GRCh37/hg19 genome assembly. This release
      contains variant data for 125,748 exomes and 15,708 whole genomes.</li>
  <li>Three filters are now available for our updated v2.1.1 (GRCh37/hg19) and gnomAD v3.1 (GRCh38/hg38) tracks,
      by minor allele frequency (MAF), quality tags, annotation type, and variant type.</li>
  <li>Mouse hover over now displays details about each variant, including the affected gene(s), the variant type,
      and annotation.</li>
  <li>Following the conventions on the gnomAD browser, items are shaded according to their annotation type: pLoF
      is red, missense is orange, synonymous is green, and others are black.</li></ul>
<p class="text-center">
  <img class='text-center' src="../images/gnomadV3_1.jpg"  width='100%' alt="The gnomAD v3.1 track display.">
</p>

<p>
We would like to thank the <a target="_blank" href="http://gnomad.broadinstitute.org/">Genome Aggregation Database
(gnomAD)</a> team at the BROAD institute and the UCSC Genome Browser team members Christopher Lee, Anna Benet-Pages,
Daniel Schmelter, and Gerardo Perez for developing and releasing these tracks.</p>


<a name="011321"></a>
<h2>Jan. 13, 2021 &nbsp;&nbsp; New GENCODE gene tracks: Human V35/V36 (hg19/hg38)</h2>
<p>
We are pleased to announce new GENCODE Gene annotation tracks, which correspond
to <a href="http://www.ensembl.info/2020/08/20/ensembl-101-has-been-released/"
target="_blank">Ensembl 101</a> and <a target="_blank"
href="https://www.ensembl.info/2020/11/30/ensembl-102-has-been-released/">Ensembl 102</a>,
for the <a target="_blank" 
href="../../cgi-bin/hgTrackUi?db=hg19&c=chr21&g=wgEncodeGencodeV36lift37">hg19/GRCh37</a> and
<a target="_blank" href="../../cgi-bin/hgTrackUi?db=hg38c=chr1&g=wgEncodeGencodeSuper">hg38/GRCh38</a>
human assemblies. The annotations were mapped to hg38 and then back-mapped
to the hg19 assembly. All gene sets contain the following tracks:</p>
<ul>
  <li>Basic - a subset of the <em>Comprehensive set</em>.</li>
  <li>Comprehensive - all GENCODE coding and non-coding transcript annotations, including
      polymorphic pseudogenes. This includes both manual and automatic annotations.</li>
  <li>Pseudogenes - all annotations except polymorphic pseudogenes.</li></ul>
<p>
The hg38 annotations also include the following tracks not available on hg19:
</p>
<ul>
  <li>2-way Pseudogenes - pseudogenes predicted by both the Yale Pseudopipe and UCSC
      Retrofinder pipelines.</li>
  <li>PolyA - polyA signals and sites manually annotated on the genome based on transcribed
      evidence (ESTs and cDNAs) of 3' end of transcripts containing at least 3 A's not matching the
      genome.</li></ul>
<p>
Details on each release can be found on the <a target="_blank" 
href="https://www.gencodegenes.org/">GENCODE site</a>. This includes <a target="_blank" 
href="https://www.gencodegenes.org/human/stats.html">statistics on each release</a>.
<p>We would like to thank the <a target="_blank" 
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Mark Diekhans and Lou Nassar for the development and
release of these data.</p>

<a name="010821"></a>
<h2>Jan. 8, 2021 &nbsp;&nbsp; January 8th SARS-CoV-2 update</h2>
<p>
We are pleased to start the year with our fifth data release for the <a href="/cgi-bin/hgTracks?db=wuhCor1" 
target="_blank">coronavirus genome browser</a>. (See also our <a
href="#040320">first</a>, <a href="#050420">second</a>, <a href="#080720">third</a>,
and <a href="#102120">fourth</a> releases)</p>

<p>
This update includes an updated track as well as three new tracks for 
the <a href="/cgi-bin/hgTracks?db=wuhCor1" 
target="_blank">coronavirus genome browser</a>, and a track for the <a 
href="/cgi-bin/hgTracks?db=hg19" target="_blank">hg19/GRCh37</a> and <a
href="/cgi-bin/hgTracks?db=hg38" target="_blank">hg38/GRCh38</a> human assemblies. Worth note is
the <a href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=lineageB_1_1_7_US" 
target="_blank">B.1.1.7 in USA</a>
track which shows substitution and deletion variants in the <b>B.1.1.7 lineage</b>, known for its 
rapid spread in England in November and December 2020, alongside the first nine genome 
sequences in the United States that were assigned to the lineage.</p>

<p>
We have also updated our <a target="_blank" href="/cgi-bin/hgPhyloPlace">web interface</a> 
to <a target="_blank" href="https://github.com/yatisht/usher">UShER</a>, a tool for 
placing new SARS-CoV-2 sequences in a global phylogenetic tree,
to include <a target="_blank" href="https://www.cogconsortium.uk/data/">COG-UK</a> 
sequences available as of Dec. 30, 2020 including thousands of new sequences in 
the B.1.1.7 lineage.</p>

<p>
Clicking on any of the track titles below will lead to the track description page, 
which includes additional information and allows for configuration of various display options.</p>
<p>
This release includes the following tracks:<br><br></p>

<ul><li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=lineageB_1_1_7_US">
B.1.1.7 in USA</a> - This track shows single-nucleotide variants 
and deletions from the SARS-CoV-2 reference genome in the B.1.1.7 consensus sequence 
and the first nine genome sequences in the United States that were assigned to B.1.1.7.
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=escape">
Antibody escape</a> - This track contains mutations 
in the Spike protein that attenuate monoclonal and serum antibody neutralization.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=pbmShanghai">
S Antib Pept Array</a> - This track shows intensities of a microarray spotted 
with short peptides derived from the S protein. Sera from both convalescent COVID-19 
and control patients were screened on the peptide microarray for both IgG and IgM responses.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&c=chr1&g=covidHgiGwasR4Pval">
COVID GWAS v4 (hg38</a>, <a target="_blank" href="/cgi-bin/hgTrackUi?db=hg19&c=chr1&g=covidHgiGwasR4Pval">
hg19</a>) - This track set shows the results of the <a target="_blank" 
href="https://www.covid19hg.org/blog/2020-11-24-covid-19-hgi-results-for-data-freeze-4-october-2020/">
GWAS Data Release 4 (October 2020)</a> from the <a target="_blank" href="https://www.covid19hg.org/">
COVID-19 Host Genetics Initiative (HGI)</a> which aims to identify the genetic determinants of 
SARS-CoV-2 infection susceptibility, disease severity and outcomes.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=sarsCov2PhyloPub">
Updated - Phylogenetic Tree and Variants from High-coverage Sequences in Public Databases</a>
- Updated from release <b>28-08-20</b> to <b>28-10-20</b>. This track displays a phylogenetic tree
inferred from SARS-CoV-2 genome sequences from <a 
href="https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/virus?SeqType_s=Nucleotide&VirusLineage_ss=SARS-CoV-2,%20taxid:2697049" 
target="_blank">GenBank</a>, <a href="https://www.cogconsortium.uk/data/" 
target="_blank">COG-UK</a> and the <a href="https://bigd.big.ac.cn/ncov/release_genome" 
target="_blank">China National Center for Bioinformation</a>, and variants found
in the sequences. Since this track is sourced from public databases, we can offer VCF files with
sequence variants for <a href="https://hgdownload.gi.ucsc.edu/gbdb/wuhCor1/sarsCov2PhyloPub/" 
target="_blank">download</a>.</li>
</ul>

<p><br>
We would like to thank the publication authors <a target="_blank" 
href="https://www.ncbi.nlm.nih.gov/pubmed/32895485">Li <em>et al</em></a> and
<a target="_blank" 
href="https://www.biorxiv.org/content/10.1101/2020.11.06.372037v1">Liu <em>et al</em></a> for
making these data available. We would also like to thank
<a href="http://public.gi.ucsc.edu/~jferna10/" target="_blank">Jason Fernandes</a>,
<a href="https://www.cogconsortium.uk/" target="_blank">COG-UK</a>,
<a href="https://www.ncbi.nlm.nih.gov/genbank/" target="_blank">GenBank</a>,
the <a href="https://gisaid.org/" target="_blank">GISAID Initiative</a>,
the <a href="https://www.covid19hg.org/" target="_blank">COVID-19 Host Genetics Initiative (HGI)</a>
and their <a target="_blank" href="https://www.covid19hg.org/partners/">partners</a>, as well as
Rachel Liao, Juha Karjalainen, and Kumar Veerapen from the <a target="_blank" 
href="https://www.broadinstitute.org/">Broad Institute</a>.</p>

<p>
These tracks are made possible due to the worldwide efforts of scientists, including the
<a href="/staff.html" target="_blank">Genome Browser team</a>. We will continue to
provide SARS-COVID-2 resources as they become available. For the latest data, see our
<a href="http://genome-preview.gi.ucsc.edu/cgi-bin/hgTracks?db=wuhCor1&position=lastDbPos" 
target="_blank">preview site</a>. Note that content on our preview server has not
undergone our QA process, and is subject to change at any time.</p>

<!-- ============= 2020 archived news ============= -->
<a name="2020"></a>

<a name="122320"></a>
<h2>Dec. 23, 2020 &nbsp;&nbsp; New ClinVar Interpretations track for human (hg19/hg38)</h2>
<p>
We are pleased to release a new track, <b>ClinVar Interpretations</b>, for the hg19/GRCh37 and 
hg38/GRCh38 human assemblies. This track can be found as part of the <a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg38&c=chr10&g=clinvar">ClinVar Composite</a>. It is the first
track to use our bead graph display, which is a variation of our existing <a target="_blank"
href="#062019">lollipop display</a>.</p>
<p>
The <b>ClinVar Interpretations track</b> displays the genomic positions of individual variant 
submissions and interpretations of the clinical significance, as well as their relationship 
to disease in the ClinVar database. As seen on the image below, the variants are classified into
six categories each on a separate horizontal line:</p>

<p class="text-center">
  <img class='text-center' src="../images/clinVarInterp.png"  width='100%' 
alt="ClinVar Interpretations track with bead graph display.">
</p>

<ul>
<li><font color="d20000"><b>P</b> - Pathogenic</font></li>
<li><font color="d20000"><b>LP</b> - Likely Pathogenic</font></li>
<li><font color="000088"><b>VUS</b> - Variant of Unknown Significance</font></li>
<li><font color="#00d200"><b>LB</b> - Likely Benign</font></li>
<li><font color="#00d200"><b>B</b> - Benign</font></li>
<li><font color="#888"><b>OTH</b> - Others</li></font></ul>

<p>
The size of the bead on the line represents the number of submissions at that genomic position.
The color of the beads aids to distinguish the categories further. Hovering on the track 
items shows the genomic variations which start at that position and the 
number of individual submissions with that classification. Additional information on the variants
can be found by clicking into the item.</p>
<p>
For more information see the <a target="_blank"
href="/cgi-bin/hgTrackUi?db=hg38&c=chr10&g=clinvar">track description page</a>.</p>
<p>
We would link to thank <a target="_blank" href="https://www.ncbi.nlm.nih.gov/clinvar/">ClinVar</a>
for making these data available, as well as Brian Joseph Raney, Ana Benet Pages, Gerardo Perez, 
and Lou Nassar for the development and release of this track.</p>

<a name="121620"></a>
<h2>Dec. 16, 2020 &nbsp;&nbsp; New In-Silico PCR feature for assembly hubs</h2>
<p>
We are excited to announce assembly hubs can now perform In-Silico PCR. Assembly hub developers 
have long been able to activate DNA and amino acid BLAT searches on their hubs by running gfServers 
from their institution, and now with the new isPcr setting, they can enable In-Silico PCR searches too 
using the same running gfServers. In-Silico PCR searches a sequence database with a pair of PCR primers, 
using an indexing strategy for fast performance to find target regions across the genome. Browser users without an 
institution to host gfServers can still benefit from this new feature by downloading and installing a virtual machine 
<a href="help/gbib.html" target="_blank">Genome Browser in a Box (GBiB)</a>, and alternatively running gfServers on the
 GBiB, all locally from a laptop. For more information see the <a href="http://genomewiki.ucsc.edu/index.php/Assembly_Hubs"
target="_blank">Assembly Hub Wiki</a> and the <a href="help/hubQuickStartAssembly.html" 
target="_blank">Quick Start Guide to Assembly Hubs</a>.</p>

<p>
We would like to thank Brian Raney and Gerardo Perez for their work on this release.</p>

<a name="121520"></a>
<h2>Dec. 15, 2020 &nbsp;&nbsp; New V35 GENCODE genes track for Human (hg19/GRCh37)</h2>
<p>
We are pleased to announce a new GENCODE Genes annotation track, which corresponds
to <a href="http://www.ensembl.info/2020/08/20/ensembl-101-has-been-released/"
target="_blank">Ensembl 101</a>, for the: <a target="_blank" 
href="../../cgi-bin/hgTrackUi?db=hg19&c=chr21&g=wgEncodeGencodeV35lift37">hg19/GRCh37</a> assembly.
The GENCODE V35 annotations were mapped to hg38/GRCh38 and then back-mapped
to the hg19/GRCh37 assembly. The gene set contains the following
tracks:</p>
<ul>
  <li>Basic - a subset of the <em>Comprehensive set</em>.</li>
  <li>Comprehensive - all GENCODE coding and non-coding transcript annotations, including
      polymorphic pseudogenes. This includes both manual and automatic annotations.</li>
  <li>Pseudogenes - all annotations except polymorphic pseudogenes.</li></ul>
<p>
Details on each release can be found on the <a target="_blank" 
href="https://www.gencodegenes.org/">GENCODE site</a>. This includes <a target="_blank" 
href="https://www.gencodegenes.org/human/stats.html">statistics on each release</a>.
<p>We would like to thank the <a target="_blank" 
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Mark Diekhans and Lou Nassar for the development and
release of this track.</p>

<a name="120920"></a>
<h2>Dec. 9, 2020 &nbsp;&nbsp; New Genome Browser: Hawaiian monk seal, neoSch1 (<em>Neomonachus schauinslandi</em>)</h2>
<p>
A genome browser is now available for the Hawaiian monk seal  (<em>Neomonachus schauinslandi</em>) assembly
released in Jun. 2017 by the
<a href="https://www.hopkinsmedicine.org/profiles/results/directory/profile/5985613/alan-scott" 
target="_blank">Johns Hopkins University</a> (ASM220157v1).
<h3>About the assembly:</h3>
<ul>
  <li><b>UCSC Genome Browser assembly ID:</b>  <a href="../../cgi-bin/hgGateway?db=neoSch1" 
  target="_blank">neoSch1</a></li>
  <li><b>Sequencing/Assembly provider ID:</b> <a href="https://www.jhu.edu/">
Johns Hopkins University</a></li>
  <li><b>Assembly date:</b> Jun. 2017</li>
  <li><B>Accession ID:</B> GCF_002201575.1</li>
  <li><B>NCBI Genome ID:</B> <A HREF="https://www.ncbi.nlm.nih.gov/genome/8042"
TARGET="_blank">8042</A> (<em>Neomonachus schauinslandi</em>)</li>
  <li><B>NCBI Assembly ID:</B> <A HREF="https://www.ncbi.nlm.nih.gov/assembly/GCF_002201575.1/"
TARGET="_blank">1137021</A></li>
  <li><B>NCBI BioProject ID:</B> <A HREF="https://www.ncbi.nlm.nih.gov/bioproject/384558"
TARGET="_blank">384558</A></li>
  <li><B>NCBI BioSample ID:</B> <A HREF="https://www.ncbi.nlm.nih.gov/biosample/SAMN06841387"
TARGET="_blank">SAMN06841387</A></li>
  <li><b>Scaffold count: </b>7,873</li>
  <li><b>Total size: </b>2,400,934,290</li>
  <li><a href="../goldenPath/credits.html#hawaiian_monk_seal_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
  <li><a href="../goldenPath/credits.html#hawaiian_monk_seal_use" 
target="_blank" >Data use conditions and restrictions</a></li>
  <li><a href="../goldenPath/help/ftp.html" 
target="_blank" >Rsync</a> and
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/neoSch1/" 
target="_blank" >FTP</a></li>
  <li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#hawaiian_monk_seal"
target="_blank" >Downloads page</a></li>
</ul>
</p>
<p>
We would like to thank Alan F. Scott at Johns Hopkins University and NCBI for creating this new Hawaiian 
monk seal assembly. We would also like to thank Hiram Clawson, Robert Kuhn, and Gerardo 
Perez for the development and release of the neoSch1 genome browser.
</p>

<a name="120220"></a>
<h2>Dec. 2, 2020 &nbsp;&nbsp; New and updated UniProt tracks for 117 assemblies</h2>
<p>
We are pleased to announce new and updated
<a href="https://www.uniprot.org/" target="_blank"</a>UniProt tracks</a> for the
following assemblies:</p><br>
<div class="container">
  <div class="row">
  <div class="col-sm-4">
  <ul>
        <li>A. gambiae <a href="../cgi-bin/hgTrackUi?db=anoGam1&g=uniprot&position=default"
         target=_blank>(anoGam1)</a></li>
        <li>A. gambiae <a href="../cgi-bin/hgTrackUi?db=anoGam3&g=uniprot&position=default"
         target=_blank>(anoGam3)</a></li>
        <li>A. mellifera <a href="../cgi-bin/hgTrackUi?db=apiMel2&g=uniprot&position=default"
         target=_blank>(apiMel2)</a></li>
        <li>African clawed frog <a href="../cgi-bin/hgTrackUi?db=xenLae2&g=uniprot&position=default"
         target=_blank>(xenLae2)</a></li>
        <li>Alpaca <a href="../cgi-bin/hgTrackUi?db=vicPac2&g=uniprot&position=default"
         target=_blank>(vicPac2)</a></li>
        <li>American alligator <a href="../cgi-bin/hgTrackUi?db=allMis1&g=uniprot&position=default"
         target=_blank>(allMis1)</a></li>
        <li>Armadillo <a href="../cgi-bin/hgTrackUi?db=dasNov3&g=uniprot&position=default"
         target=_blank>(dasNov3)</a></li>
        <li>Atlantic cod <a href="../cgi-bin/hgTrackUi?db=gadMor1&g=uniprot&position=default"
         target=_blank>(gadMor1)</a></li>
        <li>Baboon <a href="../cgi-bin/hgTrackUi?db=papAnu4&g=uniprot&position=default"
         target=_blank>(papAnu4)</a></li>
        <li>Baboon <a href="../cgi-bin/hgTrackUi?db=papHam1&g=uniprot&position=default"
         target=_blank>(papHam1)</a></li>
        <li>Bison <a href="../cgi-bin/hgTrackUi?db=bisBis1&g=uniprot&position=default"
         target=_blank>(bisBis1)</a></li>
        <li>Bonobo <a href="../cgi-bin/hgTrackUi?db=panPan3&g=uniprot&position=default"
         target=_blank>(panPan3)</a></li>
        <li>Brown kiwi <a href="../cgi-bin/hgTrackUi?db=aptMan1&g=uniprot&position=default"
         target=_blank>(aptMan1)</a></li>
        <li>Budgerigar <a href="../cgi-bin/hgTrackUi?db=melUnd1&g=uniprot&position=default"
         target=_blank>(melUnd1)</a></li>
        <li>Bushbaby <a href="../cgi-bin/hgTrackUi?db=otoGar3&g=uniprot&position=default"
         target=_blank>(otoGar3)</a></li>
        <li>C. brenneri <a href="../cgi-bin/hgTrackUi?db=caePb2&g=uniprot&position=default"
         target=_blank>(caePb2)</a></li>
        <li>C. briggsae <a href="../cgi-bin/hgTrackUi?db=cb3&g=uniprot&position=default"
         target=_blank>(cb3)</a></li>
        <li>C. elegans <a href="../cgi-bin/hgTrackUi?db=ce10&g=uniprot&position=default"
         target=_blank>(ce10)</a></li>
        <li>C. elegans <a href="../cgi-bin/hgTrackUi?db=ce11&g=uniprot&position=default"
         target=_blank>(ce11)</a></li>
        <li>C. elegans <a href="../cgi-bin/hgTrackUi?db=ce6&g=uniprot&position=default"
         target=_blank>(ce6)</a></li>
        <li>C. intestinalis <a href="../cgi-bin/hgTrackUi?db=ci3&g=uniprot&position=default"
         target=_blank>(ci3)</a></li>
        <li>C. japonica <a href="../cgi-bin/hgTrackUi?db=caeJap1&g=uniprot&position=default"
         target=_blank>(caeJap1)</a></li>
        <li>C. remanei <a href="../cgi-bin/hgTrackUi?db=caeRem3&g=uniprot&position=default"
         target=_blank>(caeRem3)</a></li>
        <li>Cat <a href="../cgi-bin/hgTrackUi?db=felCat9&g=uniprot&position=default"
         target=_blank>(felCat9)</a></li>
        <li>Chicken <a href="../cgi-bin/hgTrackUi?db=galGal6&g=uniprot&position=default"
         target=_blank>(galGal6)</a></li>
        <li>Chimp <a href="../cgi-bin/hgTrackUi?db=panTro6&g=uniprot&position=default"
         target=_blank>(panTro6)</a></li>
        <li>Chinese hamster <a href="../cgi-bin/hgTrackUi?db=criGriChoV2&g=uniprot&position=default"
         target=_blank>(criGriChoV2)</a></li>
        <li>Chinese pangolin <a href="../cgi-bin/hgTrackUi?db=manPen1&g=uniprot&position=default"
         target=_blank>(manPen1)</a></li>
        <li>Coelacanth <a href="../cgi-bin/hgTrackUi?db=latCha1&g=uniprot&position=default"
         target=_blank>(latCha1)</a></li>
        <li>Cow <a href="../cgi-bin/hgTrackUi?db=bosTau8&g=uniprot&position=default"
         target=_blank>(bosTau8)</a></li>
        <li>Cow <a href="../cgi-bin/hgTrackUi?db=bosTau9&g=uniprot&position=default"
         target=_blank>(bosTau9)</a></li>
        <li>Crab-eating macaque <a href="../cgi-bin/hgTrackUi?db=macFas5&g=uniprot&position=default"
         target=_blank>(macFas5)</a></li>
        <li>D. ananassae <a href="../cgi-bin/hgTrackUi?db=droAna2&g=uniprot&position=default"
         target=_blank>(droAna2)</a></li>
        <li>D. erecta <a href="../cgi-bin/hgTrackUi?db=droEre1&g=uniprot&position=default"
         target=_blank>(droEre1)</a></li>
        <li>D. grimshawi <a href="../cgi-bin/hgTrackUi?db=droGri1&g=uniprot&position=default"
         target=_blank>(droGri1)</a></li>
        <li>D. melanogaster <a href="../cgi-bin/hgTrackUi?db=dm6&g=uniprot&position=default"
         target=_blank>(dm6)</a></li>
        <li>D. mojavensis <a href="../cgi-bin/hgTrackUi?db=droMoj2&g=uniprot&position=default"
         target=_blank>(droMoj2)</a></li>
        <li>D. persimilis <a href="../cgi-bin/hgTrackUi?db=droPer1&g=uniprot&position=default"
         target=_blank>(droPer1)</a></li>
        <li>D. pseudoobscura <a href="../cgi-bin/hgTrackUi?db=dp3&g=uniprot&position=default"
         target=_blank>(dp3)</a></li>
   </ul></div>
   <div class="col-sm-4">
   <ul>
        <li>D. sechellia <a href="../cgi-bin/hgTrackUi?db=droSec1&g=uniprot&position=default"
         target=_blank>(droSec1)</a></li>
        <li>D. simulans <a href="../cgi-bin/hgTrackUi?db=droSim1&g=uniprot&position=default"
         target=_blank>(droSim1)</a></li>
        <li>D. virilis <a href="../cgi-bin/hgTrackUi?db=droVir2&g=uniprot&position=default"
         target=_blank>(droVir2)</a></li>
        <li>D. yakuba <a href="../cgi-bin/hgTrackUi?db=droYak2&g=uniprot&position=default"
         target=_blank>(droYak2)</a></li>
        <li>Dog <a href="../cgi-bin/hgTrackUi?db=canFam4&g=uniprot&position=default"
         target=_blank>(canFam4)</a></li>
        <li>Dolphin <a href="../cgi-bin/hgTrackUi?db=turTru2&g=uniprot&position=default"
         target=_blank>(turTru2)</a></li>
        <li>Ebola virus <a href="../cgi-bin/hgTrackUi?db=eboVir3&g=uniprot&position=default"
         target=_blank>(eboVir3)</a></li>
        <li>Elephant <a href="../cgi-bin/hgTrackUi?db=loxAfr3&g=uniprot&position=default"
         target=_blank>(loxAfr3)</a></li>
        <li>Elephant shark <a href="../cgi-bin/hgTrackUi?db=calMil1&g=uniprot&position=default"
         target=_blank>(calMil1)</a></li>
        <li>Ferret <a href="../cgi-bin/hgTrackUi?db=musFur1&g=uniprot&position=default"
         target=_blank>(musFur1)</a></li>
        <li>Fugu <a href="../cgi-bin/hgTrackUi?db=fr3&g=uniprot&position=default"
         target=_blank>(fr3)</a></li>
        <li>Garter snake <a href="../cgi-bin/hgTrackUi?db=thaSir1&g=uniprot&position=default"
         target=_blank>(thaSir1)</a></li>
        <li>Gibbon <a href="../cgi-bin/hgTrackUi?db=nomLeu3&g=uniprot&position=default"
         target=_blank>(nomLeu3)</a></li>
        <li>Golden eagle <a href="../cgi-bin/hgTrackUi?db=aquChr2&g=uniprot&position=default"
         target=_blank>(aquChr2)</a></li>
        <li>Golden snub-nosed monkey <a href="../cgi-bin/hgTrackUi?db=rhiRox1&g=uniprot&position=default"
         target=_blank>(rhiRox1)</a></li>
        <li>Gorilla <a href="../cgi-bin/hgTrackUi?db=gorGor6&g=uniprot&position=default"
         target=_blank>(gorGor6)</a></li>
        <li>Green monkey <a href="../cgi-bin/hgTrackUi?db=chlSab2&g=uniprot&position=default"
         target=_blank>(chlSab2)</a></li>
        <li>Guinea pig <a href="../cgi-bin/hgTrackUi?db=cavPor3&g=uniprot&position=default"
         target=_blank>(cavPor3)</a></li>
        <li>Hedgehog <a href="../cgi-bin/hgTrackUi?db=eriEur2&g=uniprot&position=default"
         target=_blank>(eriEur2)</a></li>
        <li>Horse <a href="../cgi-bin/hgTrackUi?db=equCab3&g=uniprot&position=default"
         target=_blank>(equCab3)</a></li>
        <li>Human <a href="../cgi-bin/hgTrackUi?db=hg19&g=uniprot&position=default"
         target=_blank>(hg19)</a></li>
        <li>Human <a href="../cgi-bin/hgTrackUi?db=hg38&g=uniprot&position=default"
         target=_blank>(hg38)</a></li>
        <li>Kangaroo rat <a href="../cgi-bin/hgTrackUi?db=dipOrd1&g=uniprot&position=default"
         target=_blank>(dipOrd1)</a></li>
        <li>Lamprey <a href="../cgi-bin/hgTrackUi?db=petMar3&g=uniprot&position=default"
         target=_blank>(petMar3)</a></li>
        <li>Lancelet <a href="../cgi-bin/hgTrackUi?db=braFlo1&g=uniprot&position=default"
         target=_blank>(braFlo1)</a></li>
        <li>Lizard <a href="../cgi-bin/hgTrackUi?db=anoCar2&g=uniprot&position=default"
         target=_blank>(anoCar2)</a></li>
        <li>Malayan flying lemur <a href="../cgi-bin/hgTrackUi?db=galVar1&g=uniprot&position=default"
         target=_blank>(galVar1)</a></li>
        <li>Manatee <a href="../cgi-bin/hgTrackUi?db=triMan1&g=uniprot&position=default"
         target=_blank>(triMan1)</a></li>
        <li>Marmoset <a href="../cgi-bin/hgTrackUi?db=calJac3&g=uniprot&position=default"
         target=_blank>(calJac3)</a></li>
        <li>Medaka <a href="../cgi-bin/hgTrackUi?db=oryLat2&g=uniprot&position=default"
         target=_blank>(oryLat2)</a></li>
        <li>Medium ground finch <a href="../cgi-bin/hgTrackUi?db=geoFor1&g=uniprot&position=default"
         target=_blank>(geoFor1)</a></li>
        <li>Megabat <a href="../cgi-bin/hgTrackUi?db=pteVam1&g=uniprot&position=default"
         target=_blank>(pteVam1)</a></li>
        <li>Microbat <a href="../cgi-bin/hgTrackUi?db=myoLuc2&g=uniprot&position=default"
         target=_blank>(myoLuc2)</a></li>
        <li>Minke whale <a href="../cgi-bin/hgTrackUi?db=balAcu1&g=uniprot&position=default"
         target=_blank>(balAcu1)</a></li>
        <li>Mouse <a href="../cgi-bin/hgTrackUi?db=mm10&g=uniprot&position=default"
         target=_blank>(mm10)</a></li>
        <li>Mouse <a href="../cgi-bin/hgTrackUi?db=mm9&g=uniprot&position=default"
         target=_blank>(mm9)</a></li>
        <li>Mouse lemur <a href="../cgi-bin/hgTrackUi?db=micMur2&g=uniprot&position=default"
         target=_blank>(micMur2)</a></li>
        <li>Naked mole-rat <a href="../cgi-bin/hgTrackUi?db=hetGla2&g=uniprot&position=default"
         target=_blank>(hetGla2)</a></li>
        <li>Nile tilapia <a href="../cgi-bin/hgTrackUi?db=oreNil2&g=uniprot&position=default"
         target=_blank>(oreNil2)</a></li>
   </ul></div>
   <div class="col-sm-4">
   <ul>
        <li>Opossum <a href="../cgi-bin/hgTrackUi?db=monDom5&g=uniprot&position=default"
         target=_blank>(monDom5)</a></li>
        <li>Orangutan <a href="../cgi-bin/hgTrackUi?db=ponAbe3&g=uniprot&position=default"
         target=_blank>(ponAbe3)</a></li>
        <li>P. pacificus <a href="../cgi-bin/hgTrackUi?db=priPac1&g=uniprot&position=default"
         target=_blank>(priPac1)</a></li>
        <li>Painted turtle <a href="../cgi-bin/hgTrackUi?db=chrPic1&g=uniprot&position=default"
         target=_blank>(chrPic1)</a></li>
        <li>Panda <a href="../cgi-bin/hgTrackUi?db=ailMel1&g=uniprot&position=default"
         target=_blank>(ailMel1)</a></li>
        <li>Pig <a href="../cgi-bin/hgTrackUi?db=susScr3&g=uniprot&position=default"
         target=_blank>(susScr3)</a></li>
        <li>Pika <a href="../cgi-bin/hgTrackUi?db=ochPri3&g=uniprot&position=default"
         target=_blank>(ochPri3)</a></li>
        <li>Platypus <a href="../cgi-bin/hgTrackUi?db=ornAna2&g=uniprot&position=default"
         target=_blank>(ornAna2)</a></li>
        <li>Proboscis monkey <a href="../cgi-bin/hgTrackUi?db=nasLar1&g=uniprot&position=default"
         target=_blank>(nasLar1)</a></li>
        <li>Rabbit <a href="../cgi-bin/hgTrackUi?db=oryCun2&g=uniprot&position=default"
         target=_blank>(oryCun2)</a></li>
        <li>Rat <a href="../cgi-bin/hgTrackUi?db=rn6&g=uniprot&position=default"
         target=_blank>(rn6)</a></li>
        <li>Rhesus <a href="../cgi-bin/hgTrackUi?db=rheMac8&g=uniprot&position=default"
         target=_blank>(rheMac8)</a></li>
        <li>Rock hyrax <a href="../cgi-bin/hgTrackUi?db=proCap1&g=uniprot&position=default"
         target=_blank>(proCap1)</a></li>
        <li>S. cerevisiae <a href="../cgi-bin/hgTrackUi?db=sacCer2&g=uniprot&position=default"
         target=_blank>(sacCer2)</a></li>
        <li>S. cerevisiae <a href="../cgi-bin/hgTrackUi?db=sacCer3&g=uniprot&position=default"
         target=_blank>(sacCer3)</a></li>
        <li>S. purpuratus <a href="../cgi-bin/hgTrackUi?db=strPur2&g=uniprot&position=default"
         target=_blank>(strPur2)</a></li>
        <li>SARS-CoV-2 <a href="../cgi-bin/hgTrackUi?db=wuhCor1&g=uniprot&position=default"
         target=_blank>(wuhCor1)</a></li>
        <li>Sea hare <a href="../cgi-bin/hgTrackUi?db=aplCal1&g=uniprot&position=default"
         target=_blank>(aplCal1)</a></li>
        <li>Sheep <a href="../cgi-bin/hgTrackUi?db=oviAri3&g=uniprot&position=default"
         target=_blank>(oviAri3)</a></li>
        <li>Sheep <a href="../cgi-bin/hgTrackUi?db=oviAri4&g=uniprot&position=default"
         target=_blank>(oviAri4)</a></li>
        <li>Shrew <a href="../cgi-bin/hgTrackUi?db=sorAra2&g=uniprot&position=default"
         target=_blank>(sorAra2)</a></li>
        <li>Sloth <a href="../cgi-bin/hgTrackUi?db=choHof1&g=uniprot&position=default"
         target=_blank>(choHof1)</a></li>
        <li>Southern sea otter <a href="../cgi-bin/hgTrackUi?db=enhLutNer1&g=uniprot&position=default"
         target=_blank>(enhLutNer1)</a></li>
        <li>Squirrel <a href="../cgi-bin/hgTrackUi?db=speTri2&g=uniprot&position=default"
         target=_blank>(speTri2)</a></li>
        <li>Squirrel monkey <a href="../cgi-bin/hgTrackUi?db=saiBol1&g=uniprot&position=default"
         target=_blank>(saiBol1)</a></li>
        <li>Stickleback <a href="../cgi-bin/hgTrackUi?db=gasAcu1&g=uniprot&position=default"
         target=_blank>(gasAcu1)</a></li>
        <li>Tarsier <a href="../cgi-bin/hgTrackUi?db=tarSyr2&g=uniprot&position=default"
         target=_blank>(tarSyr2)</a></li>
        <li>Tasmanian devil <a href="../cgi-bin/hgTrackUi?db=sarHar1&g=uniprot&position=default"
         target=_blank>(sarHar1)</a></li>
        <li>Tenrec <a href="../cgi-bin/hgTrackUi?db=echTel2&g=uniprot&position=default"
         target=_blank>(echTel2)</a></li>
        <li>Tetraodon <a href="../cgi-bin/hgTrackUi?db=tetNig2&g=uniprot&position=default"
         target=_blank>(tetNig2)</a></li>
        <li>Tibetan frog <a href="../cgi-bin/hgTrackUi?db=nanPar1&g=uniprot&position=default"
         target=_blank>(nanPar1)</a></li>
        <li>Tree shrew <a href="../cgi-bin/hgTrackUi?db=tupBel1&g=uniprot&position=default"
         target=_blank>(tupBel1)</a></li>
        <li>Turkey <a href="../cgi-bin/hgTrackUi?db=melGal5&g=uniprot&position=default"
         target=_blank>(melGal5)</a></li>
        <li>Wallaby <a href="../cgi-bin/hgTrackUi?db=macEug2&g=uniprot&position=default"
         target=_blank>(macEug2)</a></li>
        <li>White rhinoceros <a href="../cgi-bin/hgTrackUi?db=cerSim1&g=uniprot&position=default"
         target=_blank>(cerSim1)</a></li>
        <li>X. tropicalis <a href="../cgi-bin/hgTrackUi?db=xenTro9&g=uniprot&position=default"
         target=_blank>(xenTro9)</a></li>
        <li>Zebra finch <a href="../cgi-bin/hgTrackUi?db=taeGut2&g=uniprot&position=default"
         target=_blank>(taeGut2)</a></li>
        <li>Zebrafish <a href="../cgi-bin/hgTrackUi?db=danRer10&g=uniprot&position=default"
         target=_blank>(danRer10)</a></li>
        <li>Zebrafish <a href="../cgi-bin/hgTrackUi?db=danRer7&g=uniprot&position=default"
         target=_blank>(danRer7)</a></li>
    </ul></div>
</div></div>
<p>
These tracks will continue to be updated automatically as part of a quarterly pipeline.</p>
<p>
Thanks to UniProt for providing these annotations, as well as
Maximilian Haeussler and Lou Nassar of the UCSC Genome Browser group for their
work in creating and releasing these tracks.</p>

<a name="111620"></a>
<h2>Nov. 16, 2020 &nbsp;&nbsp; Genome in a Bottle tracks now available for human</h2>
<p>
We are pleased to announce the release of three new tracks from the
<a href="https://www.nist.gov/programs-projects/genome-bottle" target="_blank">Genome in a
Bottle Consortium (GIAB)</a> available for human assemblies,
<a href="../cgi-bin/hgTrackUi?db=hg19&c=chrX&g=giab" target="_blank">GRCh37/hg19</a> and
<a href="../cgi-bin/hgTrackUi?db=hg38&c=chrX&g=giab" target="_blank">GRCh38/hg38</a>. GIAB is an
open, public consortium hosted by <a href="https://www.nist.gov/" target="_blank">NIST</a>. The
priority of GIAB is to develop reference standards, reference methods, and reference data by
authoritative characterization of human genomes for use in benchmarking, including analytical
validation and technology development that will support translation of whole human genome sequencing
to clinical practice. The sole purpose of this work is to provide validated variants and regions to
enable technology and bioinformatics developers to benchmark and optimize their detection methods.
</p>
<p>
There are two phased trio tracks and a structural variants track available:
</p>
<ul>
  <li>
    <strong>Ashkenazim and the Chinese Trio tracks</strong> - shows benchmark SNV calls from two
    son/father/mother trios of Ashkenazi Jewish and Han Chinese ancestry from the Personal Genome
    Project, consented for commercial redistribution.
  </li>
  <li>
    <strong>Structural Variants track</strong> - shows benchmark SV calls (nssv) and variant regions
    (nsv) (5,262 insertions and 4,095 deletions, &gt; 50 bp, in 2.51 Gb of the genome) from the son
    (HG002/NA24385) from the Ashkenazi Jewish trio.
  </li>
</ul>
<p>
We would like to thank NCBI, the Genome in a Bottle Consortium, and NIST for providing the
benchmark VCF and BED files. We would also like to thank Chris Lee and Jairo Navarro for the
creation and release of the Genome Browser tracks.
</p>

<a name="110320"></a>
<h2>Nov. 3, 2020 &nbsp;&nbsp; New gnomAD pext track for Tissue specific transcript expression</h2>
<p>
We are happy to share that we have updated our gnomAD super track to include
the gnomAD pext datasets on the GRCh37/hg19 genome. This track displays isoform 
expression levels across 53 tissues based on 11,706 tissue samples from 
the Genotype Tissue Expression (GTEx) v7 dataset, showing a comprehensive 
view of the expression of exons across 
a gene using the proportion expression across transcripts, or pext metric, a 
transcript-level annotation metric which quantifies isoform expression for variants.
</p>
<p>
This may be especially useful to those interested in alternative splicing and 
clinicial assessment of variants. For more
information, please see the track description page and the associated publication:</p>
<ul>
<li><a href="/cgi-bin/hgTrackUi?db=hg19&g=gnomadPext">Gnomad pext track for hg19 </a></li>
<li><a href="https://www.nature.com/articles/s41586-020-2329-2" target="_blank">Transcript 
expression-aware annotation improves rare variant interpretation</a></li>
</ul>
<p>
We would like to thank the gnomAD team at the BROAD institute and the UCSC Genome Browser team members
Chris Lee, Anna Benet-Pages, and Daniel Schmelter for developing, and 
releasing this track.
</p>


<a name="102320"></a>
<h2>Oct. 23, 2020 &nbsp;&nbsp; The new GRCm39 Mouse Genome Browser is here</h2>
<p>
We are pleased to announce the release of a new mouse assembly, mm39, which corresponds to the
GRCm39 assembly released by the <a href="https://www.ncbi.nlm.nih.gov/grc/mouse"
target="_blank">Genome Reference Consortium</a> (GRC). Starting with this release, the UCSC Genome
Browser version numbers for mouse assemblies will match those of the GRC to minimize version
confusion. Hence, the GRCm39 assembly is referred to as mm39 in Genome Browser datasets and
documentation.</p>
<h3>About the assembly (mm39):</h3>
<ul>
  <li>
    UCSC Genome Browser assembly ID: mm39</li>
  <li>
    Sequencing/Assembly provider ID: Genome Reference Consortium Mouse Build 39 (GCA_000001635.9)
    GRCm39<li>
  <li>
    Assembly date: Jun. 2020</li>
  <li>
    Accession ID: GCA_000001635.9</li>
  <li>
    NCBI Genome ID: <a href="https://www.ncbi.nlm.nih.gov/genome/52" target="_blank">52</a>
    (Mus musculus)</li>
  <li>
    NCBI Assembly ID: <a href="https://www.ncbi.nlm.nih.gov/assembly/7358741"
    target="_blank">7358741</a></li>
  <li>
    NCBI BioProject ID: <a href="https://www.ncbi.nlm.nih.gov/bioproject/20689"
    target="_blank">20689</a></li>
</ul>
<p>
For more information about the files included in the GRCm39 GenBank submission, see the
<a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/635/GCA_000001635.9_GRCm39/README.txt"
target="_blank">GRCm39 README</a>. The GRCm39 GenBank record provides a detailed array of
<a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000001635.27/" target="_blank">statistics about
this assembly</a>. Bulk downloads of the sequence and annotation data may be obtained from the
Genome Browser <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/mm39/">FTP server</a> or
the <a href="http://hgdownload.gi.ucsc.edu/downloads.html#mouse" target="_blank">Downloads
page</a>. The annotation tracks for this browser were generated by UCSC and collaborators
worldwide.</p>
<p>
We would like to thank the GRC and NCBI for creating this new mouse assembly. We would also like to
thank Hiram Clawson and Jairo Navarro for the development and release of the mm39 genome browser.
</p>

<a name="102120"></a>
<h2>Oct. 21, 2020 &nbsp;&nbsp; October 21st release for SARS-CoV-2 genome browser</h2>
<p>
We are pleased to announce our fourth data release for the <a href="/cgi-bin/hgTracks?db=wuhCor1" 
target="_blank">coronavirus genome browser</a>. (See also our <a 
href="#040320">first</a>, <a href="#050420">second</a>, and <a href="#080720">third</a> 
releases)</p>
<p>
In line with our previous releases, these tracks include diverse data such as gene 
models, immunology, pathogenicity, and conservation. We would also like to bring 
attention to our <a href="#092520">recently released</a> COVID-19 GWAS 
tracks for the <a href="/cgi-bin/hgTrackUi?db=hg19&g=covidHgiGwas&position=lastDbPos" 
target="_blank">GRCh37/hg19</a> and <a 
href="/cgi-bin/hgTrackUi?db=hg38&g=covidHgiGwas&position=lastDbPos" 
target="_blank">GRCh38/hg38</a> human assemblies. The COVID-19 GWAS track displays data 
which aims to identify genetic determinants of SARS-CoV-2 infection susceptibility 
and disease severity.</p>
<p>
Clicking on any of the track titles below will lead to the track description 
page, which includes additional information and allows for configuration of 
various display options.<br><br>This release includes the following tracks:</p><br>

<ul>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=phyloGenes">
PhyloCSF Genes - Curated conserved genes</a> - This composite includes two tracks 
displaying curated SARS-CoV-2 protein-coding genes conserved within the 
<em>Sarbecovirus</em> subgenus as determined using PhyloCSF, FRESCo, and other 
comparative genomics methods, consistent with experimental evidence in SARS-CoV-2.
<ul>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=phyloGenes">
PhyloCSF Genes</a> - This track shows the conserved protein-coding genes, namely 
ORF1a, ORF1ab, S, ORF3a, ORF3c (a.k.a. ORF3h, ORF3a*, and 3a.iORF1), E, M, 
ORF6, ORF7a, ORF7b, ORF8, N, and ORF9b (a.k.a. ORF9a).</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=phyloGenes">
PhyloCSF Rejected Genes</a> - This track shows other genes that have been proposed 
that do not show the signature of conserved protein-coding genes or persuasive 
experimental evidence, and are thus unlikely to be actual protein-coding genes, 
namely ORF3d, ORF3b, ORF14 (a.k.a. ORF9b, ORF9c), and ORF10.</li></ul>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=weizmanOrfs">
New ORFs based on RNA-seq and Ribo-seq by the Weizman Institute tracks</a> - The 
Weizman ORFs (Open Reading Frames) track shows previously unannotated ORF predictions 
based on Ribo-Seq and RNA-seq data. It contains four tracks comprised of the 
predicted gene models, and data supporting them.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=icshape">
icSHAPE RNA Structure</a> - This track shows normalized icSHAPE reactivity data of 
in vivo and in vitro SARS-CoV-2 experiments.</li>	
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=iedb">
Validated epitopes from IEDB</a> - This track shows epitope sequences displayed by 
various class I MHC alleles as annotated by National Institute for Allergy and 
Infectious Diseases (NIAID) <a href="https://www.iedb.org/" target="_blank">
Immune Epitope Database (IEDB)</a>.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=potPathoIndel">
Potential pathogenic insertions and deletions from Gussow <em>et al</em>, PNAS 2020</a>
- This track shows genomic features that differentiate SARS-CoV-2 and the viruses 
behind the two previous deadly coronavirus outbreaks, SARS-CoV and Middle East 
respiratory syndrome coronavirus (MERS-CoV), from less pathogenic coronaviruses.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=pond">
Natural selection analysis from Sergei Pond's research group</a> - This track shows 
data from Pond <em>et al</em>, 2020 &quot;Natural selection analysis of 
SARS-CoV-2/COVID-19&quot; where authors used several statistical techniques to 
identify selection sites of interest in SARS-CoV-2 data from <a 
href="https://www.gisaid.org/about-us/mission/" target="_blank">GISAID</a>.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=sarsCov2PhyloPub">
Phylogenetic Tree and Variants from High-coverage Sequences in Public Databases</a>
- This track displays a phylogenetic tree inferred from SARS-CoV-2 genome sequences 
from <a 
href="https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/virus?SeqType_s=Nucleotide&VirusLineage_ss=SARS-CoV-2,%20taxid:2697049" 
target="_blank">GenBank</a>, <a href="https://www.cogconsortium.uk/data/" 
target="_blank">COG-UK</a> and the <a href="https://bigd.big.ac.cn/ncov/release_genome" 
target="_blank">China National Center for Bioinformation</a>, and variants found 
in the sequences. It uses the phylogenetic tree from the <a 
href="https://github.com/roblanf/sarscov2phylo/releases/tag/28-08-20" 
target="_blank">sarscov2phylo 28-08-20</a> release, pruned to include only public 
sequences. Since the public sequences are unrestricted, we can offer VCF files with 
sequence variants for <a href="https://hgdownload.gi.ucsc.edu/gbdb/wuhCor1/sarsCov2PhyloPub/" 
target="_blank">download</a>.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=Starr_Bloom_antibody">
Spike protein receptor-binding domain (S RBD) Deep Mutational Scanning</a> - Tracks created 
from this data include:<ul>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=Starr_Bloom_antibody">
S RBD Deep Mutational Scanning: Antibody Escape</a> - This track shows deep mutational 
scanning data measuring the effect of the mutations to the Spike RBD to binding of antibodies 
using a yeast surface display system.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=Starr_Bloom_bind">
S RBD Deep Mutational Scanning: ACE2 Binding</a> - This track shows deep mutational scanning 
data measuring effect of all possible point (amino acid) RBD mutations on ACE2 binding 
affinity using a yeast surface display system.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=Starr_Bloom">
S RBD Deep Mutational Scanning: Expression</a> - This track shows deep mutational scanning 
data measuring effect of all possible point (amino acid) RBD mutations on protein expression 
using a yeast surface display system.</li></ul></li></ul>
<ul>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=problematicSites">
Updated - Problematic sites where masking or caution are recommended for analysis</a>
- This track shows locations in the SARS-CoV-2 genome that have been identified as 
problematic for analysis for various reasons. The data was updated to the most recent 
release from July 29th.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=sarsCov2Phylo">
Updated - Phylogenetic Tree and Variants from High-coverage Sequences in GISAID EpiCoV TM</a>
- This track displays a phylogenetic tree inferred from SARS-CoV-2 genome sequences 
collected by GISAID, and variants found in the sequences. It has now been updated to 
sarscov2phylo 28-08-20 release.</li></ul><br>

<p>
We would like to thank the publication authors <a 
href="https://virological.org/t/issues-with-sars-cov-2-sequencing-data/473" 
target="_blank">De Maio <em>et al</em></a>, <a href="https://www.pnas.org/content/117/26/15193" 
target="_blank">Gussow <em>et al</em></a>, <a 
href="https://observablehq.com/@spond/natural-selection-analysis-of-sars-cov-2-covid-19" 
target="_blank">Pond <em>et al</em></a>, <a href="https://pubmed.ncbi.nlm.nih.gov/32841599/" 
target="_blank">Starr <em>et al</em></a>, <a href="https://www.biorxiv.org/content/10.1101/2020.07.07.192732v1" 
target="_blank">Sun <em>et al</em></a>, and <a href="https://www.biorxiv.org/content/10.1101/2020.06.08.141127v1" 
target="_blank">Turakhia <em>et al</em></a> for making these data available. 
We would also like to thank <a href="http://www.robertlanfear.com/" target="_blank">
Rob Lanfear</a>, <a href="https://www.biorxiv.org/content/10.1101/2020.06.02.130955v1" 
target="_blank">Irwin Jungreis</a>, <a href="http://zhanglab.net/" target="_blank">
Qiangfeng Cliff Zhang</a>, <a href="http://public.gi.ucsc.edu/~jferna10/" target="_blank">
Jason Fernandes</a>, <a href="https://snerli.wordpress.com/about/" target="_blank">
Santrupti Nerli</a>, <a href="https://www.lji.org/faculty-research/labs/peters/#overview" 
target="_blank">Bjoern Peters</a>,
the <a href="https://research.fhcrc.org/bloom/en.html" 
target="_blank">Bloom Lab</a>, the <a href="https://www.weizmann.ac.il/pages/" 
target="_blank">Weizmann Institute of Science</a> and the <a href="https://gisaid.org/" 
target="_blank">GISAID Initiative</a>.</p>
<p>
These tracks are made possible due to the worldwide efforts of scientists, including the 
<a href="/staff.html" target="_blank">Genome Browser team</a>. We will continue to 
provide SARS-COVID-2 resources as they become available. For the latest data, see our 
<a href="http://genome-preview.gi.ucsc.edu/cgi-bin/hgTracks?db=wuhCor1&position=lastDbPos" 
target="_blank">development site</a>. Note that content on our preview server has not 
undergone our QA process, and is subject to change at any time.</p>

<a name="101520"></a>
<h2>Oct. 15, 2020 &nbsp;&nbsp; Updated Ensembl gene tracks for 47 assemblies (v101)</h2>
<p>
We are pleased to announce an update to the
<a href="http://uswest.ensembl.org/info/genome/genebuild/index.html" target="_blank">
Ensembl gene annotation</a> tracks to <strong>v101</strong> for the
following assemblies:</p><br>
<div class="container">
  <div class="row">
  <div class="col-sm-4">
  <ul>
        <li>Armadillo <a href="../cgi-bin/hgTrackUi?db=dasNov3&g=ensGene"
         target=_blank>(dasNov3)</a></li>
        <li>Baboon <a href="../cgi-bin/hgTrackUi?db=papAnu4&g=ensGene"
         target=_blank>(papAnu4)</a></li>
        <li>Bison <a href="../cgi-bin/hgTrackUi?db=bisBis1&g=ensGene"
         target=_blank>(bisBis1)</a></li>
        <li>Bonobo <a href="../cgi-bin/hgTrackUi?db=panPan2&g=ensGene"
         target=_blank>(panPan2)</a></li>
        <li>Budgerigar <a href="../cgi-bin/hgTrackUi?db=melUnd1&g=ensGene"
         target=_blank>(melUnd1)</a></li>
        <li>Bushbaby <a href="../cgi-bin/hgTrackUi?db=otoGar3&g=ensGene"
         target=_blank>(otoGar3)</a></li>
        <li>C. elegans <a href="../cgi-bin/hgTrackUi?db=ce11&g=ensGene"
         target=_blank>(ce11)</a></li>
        <li>C. intestinalis <a href="../cgi-bin/hgTrackUi?db=ci3&g=ensGene"
         target=_blank>(ci3)</a></li>
        <li>Cat <a href="../cgi-bin/hgTrackUi?db=felCat9&g=ensGene"
         target=_blank>(felCat9)</a></li>
        <li>Chicken <a href="../cgi-bin/hgTrackUi?db=galGal6&g=ensGene"
         target=_blank>(galGal6)</a></li>
        <li>Chimp <a href="../cgi-bin/hgTrackUi?db=panTro5&g=ensGene"
         target=_blank>(panTro5)</a></li>
        <li>Chinese hamster <a href="../cgi-bin/hgTrackUi?db=criGriChoV1&g=ensGene"
         target=_blank>(criGriChoV1)</a></li>
        <li>Chinese hamster <a href="../cgi-bin/hgTrackUi?db=criGriChoV2&g=ensGene"
         target=_blank>(criGriChoV2)</a></li>
        <li>Coelacanth <a href="../cgi-bin/hgTrackUi?db=latCha1&g=ensGene"
         target=_blank>(latCha1)</a></li>
        <li>Cow <a href="../cgi-bin/hgTrackUi?db=bosTau9&g=ensGene"
         target=_blank>(bosTau9)</a></li>
        <li>Crab-eating macaque <a href="../cgi-bin/hgTrackUi?db=macFas5&g=ensGene"
         target=_blank>(macFas5)</a></li>
   </ul></div>
   <div class="col-sm-4">
   <ul>
        <li>D. melanogaster <a href="../cgi-bin/hgTrackUi?db=dm6&g=ensGene"
         target=_blank>(dm6)</a></li>
        <li>Dog <a href="../cgi-bin/hgTrackUi?db=canFam3&g=ensGene"
         target=_blank>(canFam3)</a></li>
        <li>Elephant <a href="../cgi-bin/hgTrackUi?db=loxAfr3&g=ensGene"
         target=_blank>(loxAfr3)</a></li>
        <li>Elephant shark <a href="../cgi-bin/hgTrackUi?db=calMil1&g=ensGene"
         target=_blank>(calMil1)</a></li>
        <li>Ferret <a href="../cgi-bin/hgTrackUi?db=musFur1&g=ensGene"
         target=_blank>(musFur1)</a></li>
        <li>Gibbon <a href="../cgi-bin/hgTrackUi?db=nomLeu3&g=ensGene"
         target=_blank>(nomLeu3)</a></li>
        <li>Golden snub-nosed monkey <a href="../cgi-bin/hgTrackUi?db=rhiRox1&g=ensGene"
         target=_blank>(rhiRox1)</a></li>
        <li>Gorilla <a href="../cgi-bin/hgTrackUi?db=gorGor4&g=ensGene"
         target=_blank>(gorGor4)</a></li>
        <li>Green monkey <a href="../cgi-bin/hgTrackUi?db=chlSab2&g=ensGene"
         target=_blank>(chlSab2)</a></li>
        <li>Guinea pig <a href="../cgi-bin/hgTrackUi?db=cavPor3&g=ensGene"
         target=_blank>(cavPor3)</a></li>
        <li>Horse <a href="../cgi-bin/hgTrackUi?db=equCab3&g=ensGene"
         target=_blank>(equCab3)</a></li>
        <li>Lamprey <a href="../cgi-bin/hgTrackUi?db=petMar2&g=ensGene"
         target=_blank>(petMar2)</a></li>
        <li>Lizard <a href="../cgi-bin/hgTrackUi?db=anoCar2&g=ensGene"
         target=_blank>(anoCar2)</a></li>
        <li>Medium ground finch <a href="../cgi-bin/hgTrackUi?db=geoFor1&g=ensGene"
         target=_blank>(geoFor1)</a></li>
        <li>Microbat <a href="../cgi-bin/hgTrackUi?db=myoLuc2&g=ensGene"
         target=_blank>(myoLuc2)</a></li>
        <li>Naked mole-rat <a href="../cgi-bin/hgTrackUi?db=hetGla1&g=ensGene"
         target=_blank>(hetGla1)</a></li>
   </ul></div>
   <div class="col-sm-4">
   <ul>
        <li>Naked mole-rat <a href="../cgi-bin/hgTrackUi?db=hetGla2&g=ensGene"
         target=_blank>(hetGla2)</a></li>
        <li>Orangutan <a href="../cgi-bin/hgTrackUi?db=ponAbe2&g=ensGene"
         target=_blank>(ponAbe2)</a></li>
        <li>Panda <a href="../cgi-bin/hgTrackUi?db=ailMel1&g=ensGene"
         target=_blank>(ailMel1)</a></li>
        <li>Pig <a href="../cgi-bin/hgTrackUi?db=susScr11&g=ensGene"
         target=_blank>(susScr11)</a></li>
        <li>Rabbit <a href="../cgi-bin/hgTrackUi?db=oryCun2&g=ensGene"
         target=_blank>(oryCun2)</a></li>
        <li>Rhesus <a href="../cgi-bin/hgTrackUi?db=rheMac10&g=ensGene"
         target=_blank>(rheMac10)</a></li>
        <li>S. cerevisiae <a href="../cgi-bin/hgTrackUi?db=sacCer3&g=ensGene"
         target=_blank>(sacCer3)</a></li>
        <li>Sheep <a href="../cgi-bin/hgTrackUi?db=oviAri3&g=ensGene"
         target=_blank>(oviAri3)</a></li>
        <li>Squirrel <a href="../cgi-bin/hgTrackUi?db=speTri2&g=ensGene"
         target=_blank>(speTri2)</a></li>
        <li>Squirrel monkey <a href="../cgi-bin/hgTrackUi?db=saiBol1&g=ensGene"
         target=_blank>(saiBol1)</a></li>
        <li>Tarsier <a href="../cgi-bin/hgTrackUi?db=tarSyr2&g=ensGene"
         target=_blank>(tarSyr2)</a></li>
        <li>Tasmanian devil <a href="../cgi-bin/hgTrackUi?db=sarHar1&g=ensGene"
         target=_blank>(sarHar1)</a></li>
        <li>Tetraodon <a href="../cgi-bin/hgTrackUi?db=tetNig2&g=ensGene"
         target=_blank>(tetNig2)</a></li>
        <li>Turkey <a href="../cgi-bin/hgTrackUi?db=melGal1&g=ensGene"
         target=_blank>(melGal1)</a></li>
        <li>X. tropicalis <a href="../cgi-bin/hgTrackUi?db=xenTro9&g=ensGene"
         target=_blank>(xenTro9)</a></li>
    </ul></div>
</div></div>
<p>
Thanks to Ensembl for providing these annotations, as well as 
Hiram Clawson and Lou Nassar of the UCSC Genome Browser group for their
work in creating and releasing these tracks.</p>

<a name="101320"></a>
<h2>Oct 13. 2020 &nbsp;&nbsp; Updated clinical tracks and features: LOVD, Decipher, and GeneReviews</h2>
<p>
We have updated three clinical tracks to add three ease-of-use features.
Decipher and LOVD Copy Number Variant (CNV) tracks now have the option to 
&quot;Merge items that span the current region&quot;, collapsing items that
are annotated across the entire Genome Browser visible region. Decipher now
includes options to filter by Pathogenicity and Variant Class. GeneReviews now
has a simple mouse-over which shows gene symbol and disease associations.<p>

<ul>
<li><a href="/cgi-bin/hgTrackUi?db=hg38&g=lovdComp">LOVD</a> 
(<a href="/cgi-bin/hgTrackUi?db=hg38&g=lovdComp">hg38</a>/
<a href="/cgi-bin/hgTrackUi?db=hg19&g=lovdComp">hg19</a>)</li>
<li><a href="/cgi-bin/hgTrackUi?db=hg19&g=decipher">Decipher</a> 
(<a href="/cgi-bin/hgTrackUi?db=hg19&g=decipher">hg19</a>)</li>
<li><a href="/cgi-bin/hgTrackUi?db=hg38&g=geneReviews">GeneReviews</a> (<a href="/cgi-bin/hgTrackUi?db=hg38&g=geneReviews">hg38</a>/
<a href="/cgi-bin/hgTrackUi?db=hg19&g=geneReviews">hg19</a>/<a href="/cgi-bin/hgTrackUi?db=hg18&g=geneReviews">hg18</a>)</li>
</ul>
<p>
Thank you to the LOVD, Decipher, and GeneReviews teams for their curation of those 
high-quality datasets. Thank you Genome Browser staff Kate Rosenbloom, Chris Lee, 
Anna Benet-Pages, and Daniel Schmelter for developing and releasing these 
feature updates.</p>

<a name="100520"></a>
<h2>Oct. 5, 2020 &nbsp;&nbsp; New dbVar Common Structural Variants track</h2>
<p>
We are happy to release the dbVar composite tracks, providing a curated resource for structural 
variant datasets on human GRCh37/hg19 and GRCh38/hg38 genome assemblies. 
The track aggregates data from more than one hundred sources 
including Gnomad, 1000 Genomes Consortium, and Decipher from the 
<a href="https://www.ncbi.nlm.nih.gov/dbvar/content/overview/" 
target="_blank">dbVar, the database of Structural Variation from NCBI</a>.
The Common Structural Variants track shows variants that meet the following criteria:
more than 50 basepairs, occur in at least 100 samples, and have an allele frequency of 
&gt;=0.01 in at least one population. For more information, please see the
track description page:</p>
<ul>
<li><a href="/cgi-bin/hgTrackUi?db=hg38&g=dbVarSv">dbVar Structural Variants </a> (<a href="/cgi-bin/hgTrackUi?db=hg38&g=dbVarSv">hg38</a>/<a href="/cgi-bin/hgTrackUi?db=hg19&g=dbVarSv">hg19</a>)</li>
</ul>
<p>
We thank the dbVar team at NCBI, especially John Lopez and Tim Hefferon for technical 
coordination and consultation, and to Chris Lee, Anna Benet-Pages, and Daniel Schmelter 
of the Genome Browser team for developing, and releasing this track.
</p>
<a name="100120"></a>

<h2>Oct. 1, 2020 &nbsp;&nbsp; New video: Making links to the Genome Browser. 
Part 3: Composite tracks, custom tracks and spreadsheets
</h2>

<p>
We are happy to announce another release to the UCSC Genome Browser 
<a href = "https://bit.ly/ucscVideos"
target = _blank>YouTube channel</a>.
</p>

<p>
This new video, <a href = 
"https://bit.ly/ucscVid23" target = _blank>Making links to the UCSC Genome Browser.  
Part 3: Composite tracks, custom tracks and spreadsheets</a> is the final installment 
of a three-part series showing how to make links to the Browser.  This release features 
how to link to the tracks grouped together in a composite track, such as dbSNP or
gnomAD, how to load your remotely hosted custom tracks and hubs via URL and how
to make columns of links in a spreadsheet so each link opens a Genome Browser to 
to a unique configuration determined by data in the sheet.  Please see our
<a href = "../training/">training page</a> for more resources.
</p>

<p>
As is our usual practice, the video comes with a <a href = 
"../training/vids/transcript23.pdf" target = _blank>transcript</a> with internal 
links to make it easier to find specific topics.
</p>

<p> 
<ul>
 <li>Part 1:  Dissecting / Understanding the URL </li>
 <li>Part 2:  Links inside genes and more tricks for links </li>
 <li>Part 3:  Composite tracks, hosted custom tracks and spreadsheets </li>
</ul>
<p>

<p>
Thanks to Robert Kuhn for production and to the Browser staff for all the features.
<p>

</p>

<a name="093020"></a>
<h2>Sept. 30, 2020 &nbsp;&nbsp; Updates to ClinVar and ClinGen (GRCh37/hg19)(GRCh38/hg38)</h2>

<p>
We have updated the <b>ClinVar Variants track (hg19/hg38)</b>, and made changes to the current
<b>ClinGen CNVs track (hg19/hg38)</b>. We have also added three new tracks in support of a new
ClinGen composite. These changes have been made in an effort to support our clinical users,
with an emphasis on facilitating Genome Browser use in variant interpretation.</p>

<a name="093020a"></a>
<h3>Updates to ClinVar Variants track</h3>
<p>
The <a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&c=chrX&g=clinvar">ClinVar Variants
track</a> has been reorganized based on variant length.
The previous tracks, ClinVar Short Variants (&lt;= 100bp) and
ClinVar Long Variants (&gt; 100bp), are now ClinVar SNVs (&lt; 50bp) and
ClinVar CNVs (&gt;= 50bp). We have also added a mouse-over from the track display which
shows phenotype and clinical significance information. In addition,
the CNV track now has the following improvements:</p>
<ul>
<li>The <a target="_blank" href="/goldenPath/help/trackDb/trackDbHub.html#mergeSpannedItems">
mergeSpannedItems</a> feature has been enabled which collapses all CNVs that span a larger
genomic region than the browser's window display. Merged items can be shown by right-click.</li>
<li>CNVs now have a color gradient according to the clinical significance for easy
identification of pathogenic variants.</li></ul>
<p>
Lastly, multiple feature options have been added to both tracks independently:</p>
<ul>
<li>Filtering by variant length is available on both tracks.</li>
<li>Filter by variation (INS, DEL, etc.) now available on both tracks.</li>
<li>Filter by clinical significance (benign, conflicting, etc.) now available on both tracks.</li>
<li>Filter on allele origin (somatic, germ line, de novo, etc.) now available on both tracks.</li>
<li>Filter by molecular consequence (stop lost, nonsense, intron variant, etc.) now available
on short variants track.</li>
</ul>
<p>
Below is an example of the filter options available for the <b>ClinVar SNVs track</b>. For
additional details on the updated display, see the <a target="_blank" 
href="/cgi-bin/hgTrackUi?db=hg38&c=chr1&g=clinvar">track description page</a>.</p>

<p class="text-center">
  <img class='text-center' src="../images/clinVarFilters.png"  width='60%' 
alt="Example of filter options in ClinVar SNVs track">
</p>

<a name="093020b"></a>
<h3>Changes to ClinGen and new tracks</h3>
<p>
We have created a new composite track, <a target="_blank" 
href="/cgi-bin/hgTrackUi?db=hg38&c=chr1&g=clinGenComp">ClinGen</a>, and deprecated the previous
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&c=chr1&g=iscaComposite">
ClinGen CNVs track</a>. The ClinGen CNVs track will continue to be available, however,
the data will no longer be updated. This was done by request of ClinGen, as all the
data, as well as further updates, can be found in the <a target="_blank" 
href="/cgi-bin/hgTrackUi?db=hg38&g=clinvar">ClinVar Copy Number Variants (ClinVar CNVs) track</a>.</p>
<p>
The new <a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&g=clinGenComp">ClinGen composite 
track</a> includes three new tracks described below:</p>
<ul>
<li><b>ClinGen Dosage Sensitivity Map - Haploinsufficiency</b> - Shows evidence supporting or
refuting haploinsufficiency (loss) as mechanisms for disease at gene-level and larger genomic
regions.</li>
<li><b>ClinGen Dosage Sensitivity Map - Triplosensitivity</b> - Shows evidence supporting or
refuting triplosensitivity (gain) as mechanisms for disease at gene-level and larger genomic
regions.</li>
<li><b>ClinGen Gene-Disease Validity Classification (ClinGen Validity)</b> - Provides a
semi-qualitative measurement for the strength of evidence of a gene-disease relationship.</li>
</ul>

<p class="text-center">
  <img class='text-center' src="../images/clinGenComp.png"  width='90%' 
alt="Example of three new ClinGen Composite tracks">
</p>

<p>
For more information on these tracks, including display conventions, scores, and classifications,
see the <a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&g=clinGenComp">track description
page</a>.</p><br>
<p>
We would like to thank Erin Riggs and May Flowers as well as the rest of the ClinGen team. We
would also like to thank ClinVar for making these data available. Track development and release
was made possible by Anna Benet-Pages, Christopher Lee, Max Haeussler, and Lou Nassar.</p>

<a name="092520"></a>
<h2>Sept. 25, 2020 &nbsp;&nbsp New data and visualization types: Covid GWAS (Lollypop) and Family Trios (VCF Trios)</h2>
<h3>Covid GWAS meta-analysis</h3>
<p>
We are happy to announce the first COVID-19 tracks in our human genome browsers
<a href="../../cgi-bin/hgGateway?db=hg19">GRCh37/hg19</a> and
<a href="../../cgi-bin/hgGateway?db=hg38">GRCh38/hg38</a>, the COVID-19 GWAS meta-analysis.
This track brings together data from 17 international GWAS studies and aims to identify
genetic determinants of SARS-CoV-2 infection susceptibility and disease severity.
These data are from the <a href="https://www.covid19hg.org/results/" target="_blank">
COVID-19 Host Genetics Initiative (HGI)</a>, a collaborative effort to analyze and share viral host
genetics research. More resources 
are available on our page: <a href="../../covid19.html">COVID-19 Resources at UCSC</a>.
</p>

<p>
These data are shown in our new lollipop visualization in order to show the dimensions of 
effect size and p-value in addition to location. Filters are also
available for these variables and the number of studies. Custom tracks using the 
lollipop format are not yet available. For more information on these
datasets, please read the track description page and try the interactive visualization
for yourself:</p>
<ul>
<li><a href="../../cgi-bin/hgTrackUi?db=hg19&g=covidHgiGwas">COVID-19 GWAS on GRCh37/hg19</a></li>
<li><a href="../../cgi-bin/hgTrackUi?db=hg38&g=covidHgiGwas">COVID-19 GWAS on GRCh38/hg38</a></li>
</ul>

<div class="text-center">
<a href="../../cgi-bin/hgTracks?hgS_doOtherUser=submit&hgS_otherUserName=dschmelt&hgS_otherUserSessionName=covidGwas">
<img width="80%" height="80%" 
src="../../images/CovidGWAS.png" alt="The COVID-19 GWAS track shows variants as lollipops,
with the colored head showing the p-value and the height of the stick showing effect size."></a>
</div>

<p>Our gratitude goes to the COVID-19 Host Genetics Initiative contributors and project leads for
making this data available, Kumar Veerapen, Rachel Liao, and Juha Karjalainen
at the Broad Institute for their input during track development. 
We would also like to thank the UCSC Genome Browser staff Kate Rosenbloom, 
Daniel Schmelter, Brain Raney, and Ana Benet-Pages for
developing, staging, and documenting this track.</p>

 
<h3>1000 Genomes Family Trios (Phased VCFs)</h3>
<p>
We are happy to share a new data set, 1000 Genomes Trios on hg38, and a visualization tool for 
displaying chromosome phased VCFs and family inheritance patterns. This track shows phased 
variants from seven trios of parents and child across different regions. 
You can infer which parent passed on which variant and identify
possible de-Novo variants. The data was produced by <a
href="https://www.internationalgenome.org/" target="_blank">International Genome Sample
Resource (IGSR)</a> from sequence data generated by the 
<a href="https://www.internationalgenome.org/about" target="_blank">1000 Genomes Project</a>.
For more information on the 1000 Genomes Trios track and the VCF Trio format, please visit 
the track description page and VCF help page:
</p>
<ul>
<li><a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg38&g=tgpTrios">
1000 Genomes Trios on GRCh38/hg38</a></li>
<li><a href="help/vcf.html#trio">VCF track format</a></li>
</ul>

<div class="text-center">
<a href="../../cgi-bin/hgTracks?hgS_doOtherUser=submit&hgS_otherUserName=dschmelt&hgS_otherUserSessionName=tgpTrios">
<img width="80%" height="80%" 
src="../../images/trioExample.png" alt="3 VCF Phased Trio tracks along with the GENCODE 
v32 genes from the Human/GRCh38 assembly. Each of two diploid haplotypes for each individual 
in a trio is drawn as a black lane, with snps as vertical ticks on the haplotype they fall on. 
Ticks are shaded blue,red,green or black according to their predicted functional effect."></a>
</div>

<p>
We would like to thank the IGSR and 1000 Genomes Project teams and contributors for
doing the research and making their data accessible. We would also like to thank Chris Lee,
Daniel Schmelter, and Angie Hinrichs for designing, developing, and releasing this track.</p>

<a name="091120"></a>
<h2>Sept. 11, 2020 &nbsp;&nbsp; UCSC LiftOver and NCBI Remap alignments for human (hg38/hg19)</h2>
<p>
We are proud to announce the release of UCSC LiftOver and NCBI Remap alignments for human
(<a href="../../cgi-bin/hgTrackUi?db=hg19&g=liftHg38">GRCh37/hg19</a> and 
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=liftHg19">GRCh38/hg38</a>). These alignments are
contained in a composite track with three subtracks, one for UCSC LiftOver and two for NCBI
alignments. 
</p>
<ul>
  <li>UCSC liftOver alignments to hg19/hg38</li>
  <li>NCBI ReMap alignments to hg19/hg38</li>
  <li>NCBI ReMap alignments to hg19/hg38, joined by axtChain</li>
</ul>
<p>
We would like to thank NCBI for making the ReMap data available and to Angie Hinrichs for the file
conversion. We would also like to thank Max Haeussler and Jairo Navarro for the development and
release of these tracks. 
</p>

<a name="082820"></a>
<h2>Aug. 28, 2020 &nbsp;&nbsp; New, chromosome-level Gorilla genome browser available (gorGor6) </h2>
<p>
We are happy to release a <a href="../../cgi-bin/hgTracks?db=gorGor6">
Genome Browser data visualization</a> for the much anticipated
2019 Gorilla genome. Thanks to the <a target="_blank" href="https://eichlerlab.gs.washington.edu/">
University of Washington</a> for releasing the high-quality assembly, 
to <a target="_blank" href="https://www.ncbi.nlm.nih.gov/refseq/">
NCBI RefSeq</a> for providing the RefSeq annotations, as well as Hiram Clawson
and Daniel Schmelter of the UCSC Genome Browser group for their
work in creating and releasing this Genome Browser.</p>

<h3>About the assembly (gorGor6):</h3>
<ul>
  <li><b>UCSC Genome Browser assembly ID:</b>  <a href="../../cgi-bin/hgGateway?db=gorGor6"> 
  gorGor6</a></li>
  <li><b>Sequencing/Assembly provider ID:</b> <a href="https://eichlerlab.gs.washington.edu/">
University of Washington Kamilah_GGO_v0</a></li>
  <li><b>Assembly date:</b> Aug. 2019</li>
  <li><B>Accession ID:</B> GCF_008122165.1</li>
  <li><B>NCBI Genome ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/genome/2156'
TARGET='_blank'>2156</A> (<em>Gorilla gorilla gorilla</em>)</li>
  <li><B>NCBI Assembly ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/assembly/4439481'
TARGET='_blank'>4438481</A></li>
  <li><B>NCBI BioProject ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/bioproject/369439'
TARGET='_blank'>369439</A></li>
  <li><B>NCBI BioSample ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/biosample/SAMN11078986'
TARGET='_blank'>SAMN11078986</A></li>
  <li><b>Scaffold count: </b>5486</li>
  <li><b>Total size: </b>3,044,872,214</li>
  <li><a href="../goldenPath/credits.html#gorilla_credits">Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
  <li><a href="../goldenPath/credits.html#gorilla_use" 
target="_blank" >Data use conditions and restrictions</a></li>
  <li><a href="../goldenPath/help/ftp.html" 
target="_blank" >Rsync</a></li>
  <li>FTP: <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/gorGor6/" >Gorilla</a></li>
  <li>Downloads page: <a href="http://hgdownload.gi.ucsc.edu/downloads.html#gorilla">Gorilla</a>
</li>
</ul>
</p>



<a name="082120"></a>
<h2>Aug. 21, 2020 &nbsp;&nbsp; Wildfires in Santa Cruz County</h2>
<p>
Northern California and the county of Santa Cruz are experiencing  
serious wildfires, including in the Santa Cruz mountains just north 
of UCSC campus.  
Campus has been evacuated and shutdowns are occurring.  We believe 
that the Genome Browser will not be affected, but it is possible 
that if power is lost, backup power will not last indefinitely.  
We do not know how long before the fire is contained or how far
it will progress before it is.
</p>
<p>
If the Genome Browser loses power, we will redirect traffic to our
mirror server in Bielefeld, <a href = "https://genome-euro.ucsc.edu" 
target = _blank>Germany</a>.  


That server and our other in Yokahama, 
<a href = "https://genome-asia.ucsc.edu" 
target = _blank>Japan</a>, are maintained to be identical to 
the server in Santa Cruz,  but you should know that your custom tracks 
and <a href = "../cgi-bin/hgSession" target =_blank>saved sessions</a>
are local to the machine where they were saved.
</p>
<p>
We do not expect to lose any data, as we have backups.  Nevertheless,
we have always advised that you should keep local copies of your data.
This would be a good time to make local backups of custom tracks,
if you do not already have them, and save your sessions to a local file.  To do
so, simply load a saved session and in the "Save Settings" section of
the page, name a local file.  This small file contains the parameters to
recreate your sessions and can be uploaded to the European or Asian server.  
</p>
<p>
Any custom tracks associated with sessions will have to be loaded 
separately on the other servers.
You can back up custom tracks from a session by clicking the Submit button 
next to "Save Custom Tracks," in that same section. Read more in this 
<a href = "#031819" target = _blank>news announcement</a> from last year. 
</p>
<p>
We should add that the Browser team wishes to express our sadness 
and concern for the loss of property and displacement experienced by 
many people in the UCSC community.  We are doing everything we can to 
help those who need it.  It is possible that responses on the mailing 
list will be delayed.
</p>

<a name="080720"></a>
<h2>Aug. 7, 2020 &nbsp;&nbsp; Eight new data tracks for SARS-CoV-2 (COVID) </h2>
<p>
We are working hard to support the scientific effort against the COVID pandemic and are happy
to announce eight new annotation tracks on the SARS-CoV-2 genome. These datasets reveal information
about duplication regions, sequencing, gene identities, plasmid constructs,
protein contact sites, and variant distribution. Like nearly all of our tracks, the data are open-access
and downloadable through our <a href="http://hgdownload.gi.ucsc.edu/downloads.html#SARS-CoV-2">
Downloads page</a> or <a href="../../cgi-bin/hgTables">Table Browser</a> page. 
More information on each of these datasets can be obtained by reading the data 
track descriptions linked below.</p>
<ul>
<li><a href="../../cgi-bin/hgTrackUi?db=wuhCor1&g=problematicSites">Problematic sites</a> shows 
regions that are ambiguous and may warrant caution or masking for analysis.</li>
<li><a href="../../cgi-bin/hgTrackUi?db=wuhCor1&g=rapid">RAPID Primers</a>
marks regions that were successfully used for Nanopore sequencing. </li>
<li><a href="../../cgi-bin/hgTrackUi?db=wuhCor1&g=unipCov2AliSwissprot">UniProt proteins</a>
was updated and displays protein sequences mapped to the SARS-CoV-2 genome.</li>
<li><a href="../../cgi-bin/hgTrackUi?db=wuhCor1&g=phyloGenes">PhyloCSF Genes</a>
shows curated SARS-CoV-2 protein-coding genes conserved within the Sarbecovirus subgenus 
as calculated with comparative genomics algorithms.</li>
<li><a href="../../cgi-bin/hgTrackUi?db=wuhCor1&g=addgene">AddGene Plasmid Alignments</a>
describes sequences contained in Addgene plasmids that align to the SARS-CoV-2 genomes.</li>
<li><a href="../../cgi-bin/hgTrackUi?db=wuhCor1&g=contacts">PDB Ligand Contact residues</a>
shows potential contact amino acids for ligands, inferred from structures in the Protein 
Database (PDB).</li>
<li><a href="../../cgi-bin/hgTrackUi?db=wuhCor1&g=nextstrainParsimony">Nextstrain Parsimony Scores</a>
shows scores of how variants are distributed in a phylogenetic tree.</li>
<li><a href="../../cgi-bin/hgTrackUi?db=wuhCor1&g=nextstrainFreq">Nextstrain variants frequency</a>
shows occurrence counts of each SARS-CoV-2 variant in GSAID sequences.</li>
</ul>
<p>
We would like to acknowledge the work of the UCSC Genome Browser team and the data collaborators.
Thank you to the research community for creating, sharing, and using this important 
data to propel science in light of this pandemic.</p>


<a name="080320"></a>
<h2>Aug. 3, 2020 &nbsp;&nbsp; Annotated SNVs track for Rhesus Macaque (rheMac10)</h2>
<p>
We are pleased to announce an annotated SNVs track for the Rhesus macaque (rheMac10) assembly.
SNVs were called using the Genome Analysis Toolkit (GATK), then the Variant Effect Predictor 
software was used to predict the functional consequence of SNVs. Ensembl release 98 gene models 
were used in the predictions, which are based on both Ensembl and RefSeq gene predictions as 
well as PacBio iso-seq data.</p>
<p>
For more information including a credits and methods section, see the <a target="_blank"
href="/cgi-bin/hgTrackUi?db=rheMac10&g=rhesusSNVs">track description page</a>.</p>
<p>
We would like to thank the Rhesus Macaque Genome Consortium and <a target="_blank"
href="https://www.hgsc.bcm.edu/non-human-primates/rhesus-monkey-genome-project">
Jeff Rogers' lab at BCM-HGSC</a> for supplying the data for this track. We would 
also like to thank Mark Diekhans and Lou Nassar for the development and release of this track.</p>

<a name="072720"></a>
<h2>Jul. 27, 2020 &nbsp;&nbsp; New video: Making links to the Genome Browser. 
Part 2:  Links inside genes and more tricks</h2>

<p>
We are happy to announce another release to the UCSC Genome Browser 
<a href = "https://bit.ly/ucscVideos"
target = _blank>YouTube channel</a>.
</p>

<p>
This new video, <a href = 
"https://bit.ly/ucscVid22" target = _blank>Making links to the UCSC Genome Browser.  
Part 2:  Links inside genes and more tricks</a>, is Part Two of a three-part series 
showing how to make links to the Browser.  This release features how to fashion a 
URL that gets to a gene or a location within a gene using nucleotide or amino-acid 
numbers instead of genomic coordinates, how to set highlights, how to find 
variables for setting Browser parameters and how to make a URL that retrieves
just the Brower graphic.
</p>

<p>
The third installment will show how to link to composite tracks, how to access your 
remotely-hosted custom tracks and how to make useful links in your spreadsheets.
</p>

<p>
As is our usual practice, the video comes with a <a href = 
"../training/vids/transcript22.pdf" target = _blank>transcript</a> with internal 
links to make it easier to find specific topics.
</p>

<p> 
<ul>
 <li>Part 1:  Dissecting / Understanding the URL </li>
 <li>Part 2:  Links inside genes and more tricks for links </li>
 <li>Part 3:  Composite tracks and hosted custom tracks </li>
</ul>
<p>

<p>
Thanks to Robert Kuhn for production and to the Browser staff for all the features.
<p>

</p>


<a name="072420"></a>
<h2>Jul. 24, 2020 &nbsp;&nbsp; New RefSeq Diffs tracks on 51 assemblies</h2>
<p>
We have generated RefSeq Diffs tracks for 51 new assemblies. These tracks
show alignment differences between the reference genome and RefSeq transcripts.
This track can be found as part of the NCBI RefSeq composite track. Click the 
assembly links below to go to the respective assembly RefSeq composite page:</p><br>
<div class="container">
  <div class="row">
  <div class="col-sm-4">
  <ul>
        <li>African clawed frog <a href="../cgi-bin/hgTrackUi?db=xenLae2&g=refSeqComposite"
         target=_blank>(xenLae2)</a></li>
        <li>Armadillo <a href="../cgi-bin/hgTrackUi?db=dasNov3&g=refSeqComposite"
         target=_blank>(dasNov3)</a></li>
        <li>Baboon <a href="../cgi-bin/hgTrackUi?db=papAnu4&g=refSeqComposite"
         target=_blank>(papAnu4)</a></li>
        <li>Bonobo <a href="../cgi-bin/hgTrackUi?db=panPan2&g=refSeqComposite"
         target=_blank>(panPan2)</a></li>
        <li>Budgerigar <a href="../cgi-bin/hgTrackUi?db=melUnd1&g=refSeqComposite"
         target=_blank>(melUnd1)</a></li>
        <li>Bushbaby <a href="../cgi-bin/hgTrackUi?db=otoGar3&g=refSeqComposite"
         target=_blank>(otoGar3)</a></li>
        <li>C. elegans <a href="../cgi-bin/hgTrackUi?db=ce11&g=refSeqComposite"
         target=_blank>(ce11)</a></li>
        <li>Cat <a href="../cgi-bin/hgTrackUi?db=felCat9&g=refSeqComposite"
         target=_blank>(felCat9)</a></li>
        <li>Chicken <a href="../cgi-bin/hgTrackUi?db=galGal6&g=refSeqComposite"
         target=_blank>(galGal6)</a></li>
        <li>Chimp <a href="../cgi-bin/hgTrackUi?db=panTro6&g=refSeqComposite"
         target=_blank>(panTro6)</a></li>
        <li>Chinese hamster <a href="../cgi-bin/hgTrackUi?db=criGriChoV1&g=refSeqComposite"
         target=_blank>(criGriChoV1)</a></li>
        <li>Coelacanth <a href="../cgi-bin/hgTrackUi?db=latCha1&g=refSeqComposite"
         target=_blank>(latCha1)</a></li>
        <li>Cow <a href="../cgi-bin/hgTrackUi?db=bosTau9&g=refSeqComposite"
         target=_blank>(bosTau9)</a></li>
        <li>Crab-eating macaque <a href="../cgi-bin/hgTrackUi?db=macFas5&g=refSeqComposite"
         target=_blank>(macFas5)</a></li>
        <li>D. melanogaster <a href="../cgi-bin/hgTrackUi?db=dm6&g=refSeqComposite"
         target=_blank>(dm6)</a></li>
        <li>Dog <a href="../cgi-bin/hgTrackUi?db=canFam3&g=refSeqComposite"
         target=_blank>(canFam3)</a></li>
        <li>Elephant <a href="../cgi-bin/hgTrackUi?db=loxAfr3&g=refSeqComposite"
         target=_blank>(loxAfr3)</a></li>
  </ul></div>
  <div class="col-sm-4">
  <ul>
        <li>Elephant shark <a href="../cgi-bin/hgTrackUi?db=calMil1&g=refSeqComposite"
         target=_blank>(calMil1)</a></li>
        <li>Ferret <a href="../cgi-bin/hgTrackUi?db=musFur1&g=refSeqComposite"
         target=_blank>(musFur1)</a></li>
        <li>Garter snake <a href="../cgi-bin/hgTrackUi?db=thaSir1&g=refSeqComposite"
         target=_blank>(thaSir1)</a></li>
        <li>Guinea pig <a href="../cgi-bin/hgTrackUi?db=cavPor3&g=refSeqComposite"
         target=_blank>(cavPor3)</a></li>
        <li>Hedgehog <a href="../cgi-bin/hgTrackUi?db=eriEur2&g=refSeqComposite"
         target=_blank>(eriEur2)</a></li>
        <li>Horse <a href="../cgi-bin/hgTrackUi?db=equCab3&g=refSeqComposite"
         target=_blank>(equCab3)</a></li>
        <li>Human <a href="../cgi-bin/hgTrackUi?db=hg19&g=refSeqComposite"
         target=_blank>(hg19)</a></li>
        <li>Lizard <a href="../cgi-bin/hgTrackUi?db=anoCar2&g=refSeqComposite"
         target=_blank>(anoCar2)</a></li>
        <li>Malayan flying lemur <a href="../cgi-bin/hgTrackUi?db=galVar1&g=refSeqComposite"
         target=_blank>(galVar1)</a></li>
        <li>Manatee <a href="../cgi-bin/hgTrackUi?db=triMan1&g=refSeqComposite"
         target=_blank>(triMan1)</a></li>
        <li>Marmoset <a href="../cgi-bin/hgTrackUi?db=calJac3&g=refSeqComposite"
         target=_blank>(calJac3)</a></li>
        <li>Medium ground finch <a href="../cgi-bin/hgTrackUi?db=geoFor1&g=refSeqComposite"
         target=_blank>(geoFor1)</a></li>
        <li>Microbat <a href="../cgi-bin/hgTrackUi?db=myoLuc2&g=refSeqComposite"
         target=_blank>(myoLuc2)</a></li>
        <li>Minke whale <a href="../cgi-bin/hgTrackUi?db=balAcu1&g=refSeqComposite"
         target=_blank>(balAcu1)</a></li>
        <li>Mouse <a href="../cgi-bin/hgTrackUi?db=mm10&g=refSeqComposite"
         target=_blank>(mm10)</a></li>
        <li>Naked mole-rat <a href="../cgi-bin/hgTrackUi?db=hetGla2&g=refSeqComposite"
         target=_blank>(hetGla2)</a></li>
        <li>Orangutan <a href="../cgi-bin/hgTrackUi?db=ponAbe3&g=refSeqComposite"
         target=_blank>(ponAbe3)</a></li>
  </ul></div>
  <div class="col-sm-4">
  <ul>
        <li>Panda <a href="../cgi-bin/hgTrackUi?db=ailMel1&g=refSeqComposite"
         target=_blank>(ailMel1)</a></li>
        <li>Pika <a href="../cgi-bin/hgTrackUi?db=ochPri3&g=refSeqComposite"
         target=_blank>(ochPri3)</a></li>
        <li>Rabbit <a href="../cgi-bin/hgTrackUi?db=oryCun2&g=refSeqComposite"
         target=_blank>(oryCun2)</a></li>
        <li>Rat <a href="../cgi-bin/hgTrackUi?db=rn6&g=refSeqComposite"
         target=_blank>(rn6)</a></li>
        <li>S. cerevisiae <a href="../cgi-bin/hgTrackUi?db=sacCer3&g=refSeqComposite"
         target=_blank>(sacCer3)</a></li>
        <li>Sheep <a href="../cgi-bin/hgTrackUi?db=oviAri4&g=refSeqComposite"
         target=_blank>(oviAri4)</a></li>
        <li>Shrew <a href="../cgi-bin/hgTrackUi?db=sorAra2&g=refSeqComposite"
         target=_blank>(sorAra2)</a></li>
        <li>Squirrel <a href="../cgi-bin/hgTrackUi?db=speTri2&g=refSeqComposite"
         target=_blank>(speTri2)</a></li>
        <li>Squirrel monkey <a href="../cgi-bin/hgTrackUi?db=saiBol1&g=refSeqComposite"
         target=_blank>(saiBol1)</a></li>
        <li>Tarsier <a href="../cgi-bin/hgTrackUi?db=tarSyr2&g=refSeqComposite"
         target=_blank>(tarSyr2)</a></li>
        <li>Tasmanian devil <a href="../cgi-bin/hgTrackUi?db=sarHar1&g=refSeqComposite"
         target=_blank>(sarHar1)</a></li>
        <li>Tibetan frog <a href="../cgi-bin/hgTrackUi?db=nanPar1&g=refSeqComposite"
         target=_blank>(nanPar1)</a></li>
        <li>White rhinoceros <a href="../cgi-bin/hgTrackUi?db=cerSim1&g=refSeqComposite"
         target=_blank>(cerSim1)</a></li>
        <li>X. tropicalis <a href="../cgi-bin/hgTrackUi?db=xenTro7&g=refSeqComposite"
         target=_blank>(xenTro7)</a></li>
        <li>X. tropicalis <a href="../cgi-bin/hgTrackUi?db=xenTro9&g=refSeqComposite"
         target=_blank>(xenTro9)</a></li>
        <li>Zebrafish <a href="../cgi-bin/hgTrackUi?db=danRer10&g=refSeqComposite"
         target=_blank>(danRer10)</a></li>
        <li>Zebrafish <a href="../cgi-bin/hgTrackUi?db=danRer11&g=refSeqComposite"
         target=_blank>(danRer11)</a></li>
  </ul></div>
</div></div>

<p>
Thanks to <a target="_blank" href="https://www.ncbi.nlm.nih.gov/refseq/">
NCBI RefSeq</a> for providing the RefSeq annotations, as well as Hiram Clawson
and Lou Nassar of the UCSC Genome Browser group for their
work in creating and releasing these tracks.</p>

<a name="071020"></a>
<h2>Jul. 10, 2020 &nbsp;&nbsp; New Genome Browsers: Dog (canFam4) and Bonobo (panPan3)</h2>
<p>
Genome browsers are now available for the <em>Canis lupus familiaris</em> assembly released 
in March 2020 by <a href="https://www.uu.se/en" target="_blank">Uppsala University</a> 
(UU_Cfam_GSD_1.0), and the <em>Pan paniscus</em> assembly released in May 2020 by the 
<a href="https://www.washington.edu/" target="_blank">University of Washington</a> 
(Mhudiblu_PPA_v0).</p>
<h3>About the assembly (canFam4):</h3>
<ul>
  <li><b>UCSC Genome Browser assembly ID:</b>  <a href="../../cgi-bin/hgGateway?db=canFam4" 
  target="_blank">canFam4</a></li>
  <li><b>Sequencing/Assembly provider ID:</b> <a href="https://www.uu.se/en">
Uppsala University</a></li>
  <li><b>Assembly date:</b> Mar. 2020</li>
  <li><B>Accession ID:</B> GCA_011100685.1</li>
  <li><B>NCBI Genome ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/genome/85'
TARGET='_blank'>85</A> (<em>Canis lupus familiaris</em>)</li>
  <li><B>NCBI Assembly ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/assembly/6119491'
TARGET='_blank'>6119491</A></li>
  <li><B>NCBI BioProject ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/bioproject/587469'
TARGET='_blank'>587469</A></li>
  <li><B>NCBI BioSample ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/biosample/SAMN13230619'
TARGET='_blank'>SAMN13230619</A></li>
  <li><b>Scaffold count: </b>2,198</li>
  <li><b>Total size: </b>2,482,000,080</li>
  <li><a href="../goldenPath/credits.html#dog_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>About the assembly (panPan3):</h3>
<ul>
  <li><b>UCSC Genome Browser assembly ID:</b>  <a href="../../cgi-bin/hgGateway?db=panPan3" 
  target="_blank">panPan3</a></li>
  <li><b>Sequencing/Assembly provider ID:</b> <a href="https://www.washington.edu/">
University of Washington</a></li>
  <li><b>Assembly date:</b> May 2020</li>
  <li><B>Accession ID:</B> GCF_013052645.1</li>
  <li><B>NCBI Genome ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/genome/10729'
TARGET='_blank'>10729</A> (<em>Pan paniscus</em>)</li>
  <li><B>NCBI Assembly ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/assembly/7133581'
TARGET='_blank'>7133581</A></li>
  <li><B>NCBI BioProject ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/bioproject/526933'
TARGET='_blank'>526933</A></li>
  <li><B>NCBI BioSample ID:</B> <A HREF='https://www.ncbi.nlm.nih.gov/biosample/SAMN11123633'
TARGET='_blank'>SAMN11123633</A></li>
  <li><b>Scaffold count: </b>4,357</li>
  <li><b>Total size: </b>3,051,901,337</li>
  <li><a href="../goldenPath/credits.html#bonobo_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
  <li><a href="../goldenPath/credits.html#dog_use" 
target="_blank" >Data use conditions and restrictions</a></li>
  <li><a href="../goldenPath/help/ftp.html" 
target="_blank" >Rsync</a></li>
  <li>FTP: <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/canFam4/" 
target="_blank" >Dog</a> and
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/panPan3/" 
target="_blank" >Bonobo</a></li>
  <li>Downloads page: <a href="http://hgdownload.gi.ucsc.edu/downloads.html#dog"
target="_blank" >Dog</a> and
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#bonobo"
target="_blank" >Bonobo</li>
</ul>
</p>

<a name="070720"></a>
<h2>Jul. 07 2020 &nbsp;&nbsp; Three new SARS-CoV-2 (COVID) variation and 
comparative Genomics tracks published; 20th Anniversary of the Genome Browser</h2>
<h3>New SARS-CoV-2 (COVID) Tracks</h3>
<p>
We are pleased to announce three new SARS-CoV-2 (COVID) data track about coronavirus variation 
and conservation. Like nearly all of our tracks, the data are open-access
and downloadable through our <a href="http://hgdownload.gi.ucsc.edu/downloads.html#SARS-CoV-2">
Downloads page</a> or <a href="../../cgi-bin/hgTables">Table Browser</a> page. </p>
<ul>
<li><a href="../../cgi-bin/hgTrackUi?db=wuhCor1&g=PhyloCSF_smooth">PhyloCSF (Codon Substitution 
Frequency)</a> shows protein-coding potential score for each nucleotide in the genome in each
of coding frame based on an aligning the SARS-CoV-2 genome with 44 Sarbecovirus genomes.</li> 
<li><a href="../../cgi-bin/hgTrackUi?db=wuhCor1&g=microdel">Microdeletions</a> 
shows sequence deletions found among thousands of SARS-CoV-2 genomes deposited in the 
<a href="https://www.gisaid.org/epiflu-applications/next-hcov-19-app/" target="_blank">GISAID 
database</a> (Global Initiative on Sharing All Influenza Data). </li>
<li><a href="../../cgi-bin/hgTrackUi?db=wuhCor1&g=multiz7way">Human CoV</a> shows an alignment
between SARS-CoV-2 and six other human-infecting coronaviruses, including SARS-CoV-1 and MERS.</li>
</ul>
<p>
We would also like to thank the many people involved in this research and publication including 
the researchers Irwin Jungreis at MIT, Brianna Sierra Chrisman at Stanford, and
UCSC staff members Max Haeussler, Hiram Clawson, and Daniel Schmelter.</p>  

<h3>The 20th Anniversary of UCSC Publishing the first Human Genome</h3>
<p>
July 7, 2020 marks the 20th anniversary of UCSC publishing the first-ever human Genome 
on the Internet in 2000. This was the first time that the code for human life was
replicated and spread through computers instead of through biological means. 
We are lucky to have some of the original publishers still with us at the Genome Browser.
Thank you, David Haussler and Jim Kent for your incredible effort pushing forward the field of human
genomics and open-source data sharing.</p>
<p>
For a timeline and pictures, please visit the 
<a href="https://ucscgenomics.soe.ucsc.edu/event/
20th-anniversary-of-publishing-the-human-genome-to-the-internet/">
Genomics Institute's article on the anniversary</a> as well as this short video
commemorating the occasion:</p>
<p><iframe width="560" height="315" src="https://www.youtube.com/embed/_6caCfvMIiY?rel=0" 
frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" 
allowfullscreen></iframe></p>

<a name="070220"></a>
<h2>Jul. 02, 2020 &nbsp;&nbsp; New video: Making links to the Genome Browser. 
Part 1:  Dissecting / Understanding the URL</h2>

<p>
We are happy to announce another release to the UCSC Genome Browser 
<a href = "https://bit.ly/ucscVideos"
target = _blank>YouTube channel</a>.
</p>

<p>
This new video, <a href = 
"https://bit.ly/ucscVid21" target = _blank>Making links to the UCSC Genome Browser.  
Part 1: Dissecting / 
Understanding the URL</a>, is Part One of a three-part series showing how to make 
links to the Browser, featuring how to fashion a URL that turns on datasets of 
choice at specific locations using coordinates, gene names and other annotation 
identifiers.
</p>

<p>
Future installments will show how to navigate to specific locations <em>within genes</em> 
without knowing their genomic coordinates, how to set highlights, loading composite 
tracks, accessing your remotely hosted custom tracks and how to make useful links 
in your spreadsheets.
</p>

<p>
As is our usual practice, the video comes with a <a href = 
"../training/vids/transcript21.pdf" target = _blank>transcript</a> with internal 
links to make it easier to find specific topics.
</p>

<p> 
<ul>
 <li>Part 1:  Dissecting / Understanding the URL </li>
 <li>Part 2:  Links inside genes and more tricks for links </li>
 <li>Part 3:  Composite tracks and hosted custom tracks </li>
</ul>
<p>

<p>
Thanks to Robert Kuhn for production and to the Browser staff for all the features.
<p>

</p>

<a name="062520"></a>
<h2>Jun. 25, 2020 &nbsp;&nbsp; Vertebrate Genomes Project assembly hub update</h2>
<p>
We are proud to announce our first major update to the <a href="https://hgdownload.gi.ucsc.edu/hubs/VGP/" 
target="_blank">Vertebrate Genomes Project assembly hub</a>. This is a follow up to our 
<a href="#082819">initial release</a>, and expands the hub to 168 assemblies.</p>
<p>
With this update, we have created three additional hubs in addition to the main hub:</p>

<ul>
<li><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/VGP/index.html">
<b>Primary assembly (72)</b></a> - Main hub which contains the primary
assemblies for each of the organisms.</li>
<li><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/VGP/vgpAlt.html">
<b>Alternate/haplotype (68)</b></a> - Secondary hub which contains alternate 
haplotype assemblies for a subset of the organisms in the primary hub.</li>
<li><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/VGP/vgpTrio.html">
<b>Trio mat/pat (17)</b></a> - Secondary hub which contains trio assemblies (child, maternal,
paternal) for a subset of the organisms in the primary hub.</li>
<li><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/VGP/vgpLegacy.html">
<b>Legacy/superseded (11)</b></a> - Hub containing previous assemblies that have since been updated.</li>
</ul>

<p>
As with the previous release, all assemblies are built with a minimum set of data tracks including
gene annotations, CpG islands, GC content, and repeat masking. <a target="_blank" 
href="https://hgdownload.gi.ucsc.edu/hubs/VGP/asmStats.html">Assembly statistics</a> and 
<a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/VGP/trackData.html">
track statistics</a> are also available for all hubs. Links to the different pages are included
below:</p>

<p>
<table border='1'><thead>
<tr><th colspan=5 style='text-align:center;'>Available VGP assemblies</th></tr>
<tr><th>Index pages:&nbsp;</th>
<th><a href='https://hgdownload.gi.ucsc.edu/hubs/VGP/index.html'>primary assembly</a></th>
<th><a href='https://hgdownload.gi.ucsc.edu/hubs/VGP/vgpAlt.html'>alternate/haplotype</a></th>
<th><a href='https://hgdownload.gi.ucsc.edu/hubs/VGP/vgpTrio.html'>trio mat/pat</a></th>
<th><a href='https://hgdownload.gi.ucsc.edu/hubs/VGP/vgpLegacy.html'>legacy/superseded</a></th>
</tr><tr>
<th>Assembly statistics:&nbsp;</th>
<th><a href='https://hgdownload.gi.ucsc.edu/hubs/VGP/asmStats.html'>primary assembly</a></th>
<th><a href='https://hgdownload.gi.ucsc.edu/hubs/VGP/vgpAltStats.html'>alternate/haplotype</a></th>
<th><a href='https://hgdownload.gi.ucsc.edu/hubs/VGP/vgpTrioStats.html'>trio mat/pat</a></th>
<th><a href='https://hgdownload.gi.ucsc.edu/hubs/VGP/vgpLegacyStats.html'>legacy/superseded</a></th>
</tr><tr>
<th>Track statistics:&nbsp;</th>
<th><a href='https://hgdownload.gi.ucsc.edu/hubs/VGP/trackData.html'>primary assembly</a></th>
<th><a href='https://hgdownload.gi.ucsc.edu/hubs/VGP/vgpAltData.html'>alternate/haplotype</a></th>
<th><a href='https://hgdownload.gi.ucsc.edu/hubs/VGP/vgpTrioData.html'>trio mat/pat</a></th>
<th><a href='https://hgdownload.gi.ucsc.edu/hubs/VGP/vgpLegacyData.html'>legacy/superseded</a></th>
</tr>
</thead>
</table></p>

<p>
All data are available from the UCSC download servers. For information on how to download
specific assembly data, refer to the assembly gateway page once the hub is connected, e.g.
<a href="../../cgi-bin/hgGateway?hubUrl=https://hgdownload.gi.ucsc.edu/hubs/VGP/hub.txt&genome=GCA_004115265.2_mRhiFer1_v1.p" 
target="_blank">greater horseshoe bat</a>. This page also includes information on starting
a local BLAT server for the assembly.</p>
<p>
We would like to thank the <a href="https://vertebrategenomesproject.org/" 
target="_blank">Vertebrate Genomes Project</a> and the <a href="https://genome10k.soe.ucsc.edu/" 
target="_blank">Genome 10k Project</a> for their sequencing efforts and making these genomes
available. We would also like to thank Hiram Clawson and Luis Nassar from the Genome Browser team
for the development and release of these assembly hubs.</p>

<a name="061920"></a>
<h2>Jun. 19, 2020 &nbsp;&nbsp; Updated Self Chain track for human (GRCh38/hg38)</h2>
<p>
We have updated the <a target="_blank" href="/cgi-bin/hgTrackUi?db=hg38&c=chr16&g=chainSelf">
Self Chain</a> track for the human GRCh38/hg38 assembly. The Self Chain track shows
alignments of the human genome with itself, using a gap scoring system that allows longer
gaps than traditional affine gap scoring systems. After filtering out the alignments produced
when identical locations of the genome map to one another (e.g. chrN mapping to chrN), the
remaining alignments point out areas of duplication within the human genome.</p>
<p>
The updated
track now includes 33,039,051 chains, and should provide improved sensitivity for pseudogenes
and other homologous regions. For more information, see the track <a target="_blank" 
href="/cgi-bin/hgTrackUi?db=hg38&c=chr16&g=chainSelf">description page</a>.</p>
<p>
We would like to thank Angie Hinrichs, Anna Benet-Pages, and Lou Nassar for the development
and release of this track.</p>

<a name="061820"></a>
<h2>Jun. 18, 2020 &nbsp;&nbsp; New ENCODE cCREs track available for human and mouse (hg38 and mm10)
</h2>
<p>
We are pleased to announce the <em>ENCODE Registry of candidate cis-Regulatory Elements</em> (cCREs)
track for the human and mouse genomes
(<a href="../../cgi-bin/hgTrackUi?db=hg38&c=chrX&g=encodeCcreCombined"
target="_blank">GRCh38/hg38</a> and
<a href="../../cgi-bin/hgTrackUi?db=mm10&c=chrX&g=encodeCcreCombined"
target="_blank">GRCm38/mm10</a>).
cCREs are the subset of representative DNase hypersensitive sites across ENCODE and Roadmap
Epigenomics samples that are supported by either histone modifications (H3K4me3 and H3K27ac) or
CTCF-binding data. The Registry of cCREs is one of the core components of the integrative level of
the <a href="https://www.encodeproject.org/" target="_blank">ENCODE Encyclopedia of DNA
Elements</a>. A total of 926,535 elements for human and 339,815 elements for mouse were identified
and classified by the ENCODE Data Analysis Center according to biochemical signatures.</p>

<a href="http://genome.ucsc.edu/s/Kate/hg38%20ENCODE%20cCRE%20announcement%20largeprint"
target="_blank">
<img style="margin-left" width="950px"  src="../images/ccreAnnouncementScreenshot.png"></a>

<p>
CCREs are colored and labeled according to classification by regulatory signature:
<p>
<table cellpadding='2'>
  
  <tr>
    <th style="border-bottom: 2px solid;">Color</th>
    <th style="border-bottom: 2px solid;"></th>
    <th style="border-bottom: 2px solid;">UCSC label</th>
    <th style="border-bottom: 2px solid;">ENCODE classification</th>
    <th style="border-bottom: 2px solid;">ENCODE label</th>
  </tr>
  </thead>

<tr><td style='background-color: #FF0000;'></td><td>red</td>
        <td>prom</td>
        <td>promoter-like signature</td>
        <td>PLS</td></tr>
<tr><td style='background-color: #FFA700;'></td><td>orange</td>
        <td>enhP</td>
        <td>proximal enhancer-like signature</td>
        <td>pELS</td></tr>
<tr><td style='background-color: #FFCD00;'></td><td>yellow</td>
        <td>enhD</td>
        <td>distal enhancer-like signature</td>
        <td>dELS</td></tr>
<tr><td style='background-color: #FFAAAA;'></td><td>pink</td>
        <td>K4m3</td>
        <td>DNase-H3K4me3</td>
        <td>DNase-H3K4me3</td></tr>
<tr><td style='background-color: #00B0F0;'</td><td>blue</td>
        <td>CTCF</td>
        <td>CTCF-only</td>
        <td>CTCF-only</td></tr>
</table>
</p>

<p>
This dataset was produced by the ENCODE Data Analysis Center
(<a href="https://www.umassmed.edu/zlab/" target="_blank">Zlab</a> at UMass Medical Center). Thanks
to Henry Pratt, Jill Moore, Michael Purcaro, and Zhiping Weng, PI for providing this data and
initial version of the track. Thanks also to the ENCODE Consortium, the ENCODE production
laboratories, and the ENCODE Data Coordination Center for generating and processing the datasets
used here. Thanks to Kate Rosenbloom and Jairo Navarro for the development and release of these
tracks.</p>



<a name="052720"></a>
<h2>May 26, 2020 &nbsp;&nbsp; New video on the Browser's YouTube channel</h2>

<p>
We have released a new video to the Browser's
<a href = "https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos"
target = _blank>YouTube channel</a>.
</p>

<p>
   <b> &nbsp; &nbsp; Browser Basics, Part Three &mdash; Configuring the Browser
   + DNA navigation </b>
</p>

<p>
This <a href =
"https://bit.ly/ucscVid19" target = _blank>video</a>
is part of a three-part series that goes back to the Basics,
designed to introduce new users to the Browser.

Our previous videos have been directed
either at experienced users who may have questions about how to do certain
tasks, or serve as introductions to new features in the Browser.
In this three-part series, we fill in the background with more basic
information, though we expect experienced users to find some new things
as well.
</p>

<p>
Today's release, Part Three, shows further configuration options
in the Browser and also how to use DNA sequence to find locations
in the genome.  Examples include BLAT, Short Match, Drag-N-Highlight,
removing highlights, Drag-N-Zoom, configuring the scale bar/base position
track, adding a title to the image and more.
</p>

<p>
As is our practice, a <a href ="../training/vids/transcript19.pdf"
target = _blank>transcript</a>
is provided for those who wish to review the content before watching.
</p>

<ul>
<li>
<a href = "https://bit.ly/ucscVid17" target = _blank>
Part One</a>:  Getting around in the Browser</li>
<li>
<a href = "https://bit.ly/ucscVid18" target = _blank>
Part Two</a>:  Configuring the Browser </li>
<li>
<a href = "https://bit.ly/ucscVid19" target = _blank>
Part Three</a>:  Configuration + DNA Navigation </li>
</ul>
</p>
<p>
Thanks to Robert Kuhn for the video production.
</p>


<a name="052620"></a>
<h2>May 26, 2020 &nbsp;&nbsp; NCBI RefSeq assembly hubs </h2>
<p>
We are pleased to announce new NCBI RefSeq assembly hubs for vertebrate genomes. Currently these
hubs contain 295 assemblies, encompassing most RefSeq vertebrate genomes.</p> 
<p>
All assemblies contain multiple gene models including NCBI RefSeq annotations, and Ensembl gene
predictions where available. Additional tracks include repeat masking and simple repeat data.
A complete list of available tracks for each assembly can be found in the respective track
statistics pages.</p>
<p>
The assemblies are divided into five categories. See the assembly statistics link below for a 
complete list of available genomes in each hub. Assembly statistics such as genome size, 
gaps, and masking are also available for each genome. Lastly, track statistics are also available
displaying all available tracks for each assembly, as well as their genomic coverage.</p><br>
 
<ul>
<li><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/primates/index.html">
</b>Primates</b></a> - <a target="_blank" 
href="https://hgdownload.gi.ucsc.edu/hubs/primates/asmStats.html">Assembly statistics</a>
- <a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/primates/trackData.html">
Track statistics</a></li>
<li><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/mammals/index.html">
</b>Mammals</b></a> - <a target="_blank" 
href="https://hgdownload.gi.ucsc.edu/hubs/mammals/asmStats.html">Assembly statistics</a>
- <a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/mammals/trackData.html">
Track statistics</a></li>
<li><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/birds/index.html">
</b>Birds</b></a> - <a target="_blank" 
href="https://hgdownload.gi.ucsc.edu/hubs/birds/asmStats.html">Assembly statistics</a>
- <a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/birds/trackData.html">
Track statistics</a></li>
<li><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/fish/index.html">
</b>Fish</b></a> - <a target="_blank" 
href="https://hgdownload.gi.ucsc.edu/hubs/fish/asmStats.html">Assembly statistics</a>
- <a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/fish/trackData.html">
Track statistics</a></li>
<li><a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/vertebrate/index.html">
</b>Other vertebrates</b></a> - <a target="_blank" 
href="https://hgdownload.gi.ucsc.edu/hubs/vertebrate/asmStats.html">Assembly statistics</a>
- <a target="_blank" href="https://hgdownload.gi.ucsc.edu/hubs/vertebrate/trackData.html">
Track statistics</a></li>
</ul>

<p>
These assemblies can be found on our list of <a target="_blank" href="/cgi-bin/hgHubConnect">
Public Hubs</a>, or by searching for a species by NCBI accession or name directly from the 
<a target="_blank" href="/cgi-bin/hgGateway">Gateway page</a>.</p>
<p>
We would like to thank NCBI RefSeq for making these data available. 
We would also like to thank Hiram Clawson and Lou Nassar for the 
development and release of these hubs.</p>

<a name="052220"></a>
<h2>May. 22, 2020 &nbsp;&nbsp; Updated GENCODE gene tracks: Human V34 (hg19/hg38) - Mouse M25
(mm10) and updated GTEx V8 gene expression track in hg38</h2>
<h3>GENCODE Gene Set Updates</h3>
<p>
We are pleased to announce new GENCODE Gene annotation tracks, which correspond
to <a href="http://www.ensembl.info/2020/04/29/ensembl-100-has-been-released/"
target="_blank">Ensembl 100</a>, for three assemblies: <a target="_blank" 
href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV34lift37">hg19/GRCh37</a>,
<a target="_blank" 
href="../../cgi-bin/hgTrackUi?db=hg38c=chr1&g=wgEncodeGencodeV34">hg38/GRCh38</a>,
and <a target="_blank" 
href="../../cgi-bin/hgTrackUi?db=mm10&c=chr12&g=wgEncodeGencodeVM25">mm10/GRCm38</a>.
For human, the GENCODE V34 annotations were mapped to hg38/GRCh38 and then back-mapped
to the hg19/GRCh37 assembly. For all three assemblies, the gene sets contain the following
tracks:</p>
<ul>
  <li>Basic - a subset of the <em>Comprehensive set</em>.</li>
  <li>Comprehensive - all GENCODE coding and non-coding transcript annotations, including
      polymorphic pseudogenes. This includes both manual and automatic annotations.</li>
  <li>Pseudogenes - all annotations except polymorphic pseudogenes.</li></ul>
<p>
The hg38 and mm10 assemblies also include the following tracks that are not available on hg19:
</p>
<ul>
  <li>2-way Pseudogenes - pseudogenes predicted by both the Yale Pseudopipe and UCSC
      Retrofinder pipelines.</li>
  <li>PolyA - polyA signals and sites manually annotated on the genome based on transcribed
      evidence (ESTs and cDNAs) of 3' end of transcripts containing at least 3 A's not matching the
      genome.</li></ul>
<p>
Details on each release can be found on the <a target="_blank" 
href="https://www.gencodegenes.org/">GENCODE site</a>. This includes <a target="_blank" 
href="https://www.gencodegenes.org/human/stats.html">statistics on each release</a>.
<p>We would like to thank the <a target="_blank" 
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these
annotations. We would also like to thank Mark Diekhans and Daniel Schmelter for the development and
release of these tracks.</p>


<h3>GTEx V8 gene transcript expression from RNA-seq of 54 tissues</h3>
<a name="052220a"></a>
<p>We are excited to announce an update to the GTEx V8 for the gene expression track
based on data from the <a target="_blank" href="http://commonfund.nih.gov/GTEx/index">
NIH Genotype-Tissue Expression (GTEx) project</a>. This track displays 
tissue-specific gene expression based on RNA-seq in 54 tissues from 948 donors
released in August 2019. 
The original data for this track can be found at the <a target="_blank"
 href="https://www.gtexportal.org/">GTEx Portal</a> hosted by the Broad Institute.</p>

<p>
This track features a horizontal bar chart for each GENCODE gene, resulting in 
colored bars which show median tissue expression values assayed by the GTEx project in RPKM.
Mouse over the bar in the graph shows a tissue specific expression value, while clicking on a chart
shows a much larger box-and-whiskers graph for that transcript. The complete tissue color legend 
and filters are shown on the <a target="_blanK" 
a href="/cgi-bin/hgGtexTrackSettings?db=hg38&g=gtexGeneV8">GTEx track configuration page</a>. 
Below the bar graph, a line is shown indicating the gene extent that was used to 
generate the annotation, colored by gene class using GENCODE conventions 
(e.g. blue for protein-coding, green for non-coding).
</p>
<p>
<a href = 
"/cgi-bin/hgGtexTrackSettings?db=hg38&g=gtexGeneV8" 
target = _blank>
<img src = "../images/GTExV8.png" WIDTH = 750 HEIGHT = 432>
</a>
</p>

<p>
Thank you Kate Rosenbloom and Daniel Schmelter for developing and releasing these tracks</p>


<a name="051220"></a>
<h2>May 12, 2020 &nbsp;&nbsp; New video:  Coronavirus Browser SARS CoV-2  </h2>
<p>
We are pleased to announce the release of a 
<a href = 
"https://www.youtube.com/watch?v=Ee6h0xyZDOM&feature=youtu.be&list=UUQnUJepyNOw0p8s2otX4RYQ" 
target = _blank>new video</a> on our 
<a href = "https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos" 
target = _blank>YouTube channel</a>, designed to help make the Browser more 
accessible to those not usually using the Genome Browser, especially virologists 
and molecular biologists developing assays and vaccines.  The video highlights 
many features that regular Browser users may already know, but in the 
context of SARS-CoV-2 genome assembly.  The Browser coronavirus tour highlights 
RT-PCR data, UniProt , crowd-sourced data, T-cell Reactive epitopes and 
comparative genomics data for viral isolates from around the world and those 
infecting other vertebrates.
</p>
<p>
As is our practice, a <a href= "../training/vids/transcript20.pdf"
  target = _blank>transcript</a> of the voiceover is available.
</p>
<p> 
<a href = 
"https://www.youtube.com/watch?v=Ee6h0xyZDOM&feature=youtu.be&list=UUQnUJepyNOw0p8s2otX4RYQ" 
target = _blank>
 <img src = "../images/titleVid20.jpg" WIDTH = 458 HEIGHT = 258>
</a>
</p>

<p> 
Thanks to Robert Kuhn for production and Max Haeussler and Jason Fernandes
for input.  
</p>

<a name="050620"></a>
<h2>May 6, 2020 &nbsp;&nbsp; Problematic Regions for NGS or Sanger sequencing or very variable regions</h2>
<p>
We are happy to announce a new data track which describes
<a target="_blank" href="../../cgi-bin/hgTrackUi?db=hg19&g=problematic">Problematic Regions for NGS or Sanger sequencing or very variable regions</a> in the human assembly (hg19/GRCh37).</p>

<p>
This track shows genomic regions known to cause artefacts for common sequencing downstream analyses,
like alignment, variant, or peak calling. This track is composed of 12 subtracks with underlying
data imported from these projects:</p>

<ul>
<li>
<a target="_blank" href="../../cgi-bin/hgGateway">UCSC</a> - manually curated annotations of fixed sequences,
alternate haplotypes, unplaced contigs, pseudo-autosomal regions, and mitochondria that can yield
alignments with low quality mapping scores and discordant read pairs. </li>
<li>
<a target="_blank" href="https://personal.broadinstitute.org/anshul/projects/encode/rawdata/blacklists/hg19-
blacklist-README.pdf">ENCODE Blacklist</a> - comprehensive set of regions that have anomalous,
unstructured or high signal in next-generation sequencing experiments.</li>
<li>
<a target="_blank" href="https://www.ncbi.nlm.nih.gov/variation/tools/get-rm/">NCBI GeT-RM</a> - highly homologous
gene- and exon-level regions that are difficult or impossible to analyze with standard Sanger or NGS
approaches and are relevant to current clinical testing.</li>
<li>
<a target="_blank" href="https://www.nist.gov/programs-projects/genome-bottle">Genome-in-a-Bottle</a> - defined
regions where it is difficult to make a confident call, due to low coverage, systematic sequencing
errors, local alignment problems, etc.</li>
</ul>

<p>
We would like to thank Anna Benet-Pages, Max Haeussler, Angie Hinrichs, and Daniel Schmelter at the
UCSC Genome Browser for planning, building, and testing these tracks.
</p>

<a name="050420"></a>
<h2>May 4, 2020 &nbsp;&nbsp; May 4th data release for SARS-CoV-2 genome browser</h2>
<p>
We are pleased to announce our second data release for the <a target="_blank" 
href="/cgi-bin/hgTracks?db=wuhCor1">coronavirus genome browser</a>. Much like our 
<a href="#040320">first release</a>, this includes a variety of annotation types such as CRISPR 
tracks, protein structure and interaction tracks, immunology tracks, and a 119-way multiple 
sequence alignment.</p>
<p>
One new track we would like to highlight is the <a target="_blank" 
href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=publicAnnots">Crowd-Sourced Data</a> track. 
This track displays annotations made via a public spreadsheet available at 
<a target="_blank" href="http://bit.ly/cov2annots">http://bit.ly/cov2annots</a>. Any user 
can freely contribute to the spreadsheet, after which the annotations will display in 
the public coronavirus genome browser.</a>
<p>
Clicking on any of the track titles below will lead to the track description page, 
which includes additional information and allows for configuration of various display 
options. <br><br>This release includes the following tracks:</p><br>

<ul>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=publicAnnots">
Crowd-Sourced Data</a> - This track shows varied annotations submitted by users via a 
public spreadsheet available at <a target="_blank" href="http://bit.ly/cov2annots">
http://bit.ly/cov2annots</a>.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=cas13Crispr">
Cas13 CRISPR</a> - This track shows the in silico design of crRNAs for Cas13 using 
the tool <a target="_blank" href="https://github.com/QilabGitHub/nCov2019_Guide_Design">
nCov2019_Guide_Design</a>, as described in Abbott et al, 2020.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=crisprDet">
CRISPR Detection Guides</a> - This track shows the locations of CRISPR detection 
guides and flanking primers, containing data aggregated from three institutions.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=artic">
ARTIC Primers</a> - This track shows the primers for the <a target="_blank" 
href="https://artic.network/">ARTIC network</a> SARS-CoV-2 sequencing protocol, Version 3.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=gordon2">
Human Interacting Proteins</a> - This track shows human protein-protein interactions 
as described by <a target="_blank" 
href="https://www.biorxiv.org/content/10.1101/2020.03.22.002386v1.full.pdf+html">Gordon et al, 2020</a>.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=pdb">
Protein Data Bank (PDB) Structures</a> - This track shows alignments of sequences 
with known protein structures in the <a target="_blank" href="https://www.rcsb.org/">
Protein Data Bank.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=rnaStructRangan">
Rangan et al. RNA predictions</a> - This track shows the locations of RNA structure 
predictions reported in <a target="_blank" href="https://doi.org/10.1101/2020.03.27.012906">
Rangan et al, 2020</a>.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=rfam">
Rfam</a> - This track shows coronavirus-specific structured RNA families from the 
<a target="_blank" href="https://rfam.org/">Rfam database</a>.</li>
<li>
<a target="_blank" href="https://www.cell.com/pb-assets/products/coronavirus/CELL_CELL-D-20-00765.pdf">
Kim et al, Cell 2020.</a> - Tracks created from this publication include:
  <ul>
  <li>
  <a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=kimTranscripts">
  Kim Transcripts</a> - This track shows predicted and experimental representations of the 
  SARS-CoV-2 transcriptome based on long-read Nanopore sequencing.</li>
  <li>
  <a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=kimNp">
  Kim RNA Breakpoints</a> - This track shows the locations of RNA transcript breakpoints 
  as determined by Nanopore and DNA Nanoball MGISEQ sequencing.</li>
  <li>
  <a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=kimRnaMod">
  Kim RNA Modifications</a> - This track shows the locations of RNA-modifications as 
  determined by Nanopore sequencing.</li>
  </ul></li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=rosettaMhc">
CD8 RosettaMHC</a> - This track includes 718 CD8 epitopes restricted to HLA-A*02:01 
as predicted by NetMHCpan4.0 and RosettaMHC.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=poranHla1">
Poran HLA</a> - This track shows putative epitopes for CD4+ and CD8+ T cells whose HLA 
binding properties cover over 99% for US, European, and Asian populations, for both 
HLA-I and HLA-II.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=targets">
T-Reactive Epitopes</a> - This track shows T-cell reactive epitopes in patients and 
donors as described by <a target="_blank" 
href="https://www.medrxiv.org/content/10.1101/2020.04.17.20061440v1.full.pdf+html">
Braun et al, 2020</a>.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=strainCons119way">
119-Way Coronavirus Alignment</a> - This track shows multiple alignments of 119 coronavirus 
sequences, aligned to the SARS-CoV-2 reference. It includes both 2019/2020 outbreak sequences 
and more distantly related vertebrate host sequences.</li>
</ul>
<p>
We would like to thank the publication authors Abbott et al, Gordon et al, Rangan et al, 
Poran et al, and Braun et al for making these data available. We would also like to thank 
<a target="_blank" href="https://rfam.xfam.org/">Rfam</a>, <a target="_blank"
href="https://www.rcsb.org/">PDB</a>, <a target="_blank" href="https://mammoth.bio/">
Mammoth Biosciences</a>, the <a target="_blank" href="https://www.sabetilab.org/">
Sabeti Lab</a> at the Broad Institute, and the <a target="_blank" href="http://sanjanalab.org/">
Sanjana Lab</a> at New York University.</p>
<p>
Lastly, we would like to thank Kord Kober and Kiley Charbonneau from the <a target="_blank"
href="https://kober.ucsf.edu/">Kober lab</a> at UCSF for their contributions to the 
crowd-sourced data track, as well as David Haussler, Jason Fernandes, Nikolaos Sgourakis, 
Santrupti Nerli, Manny Ares, Justin Sim, Alinne Gonzalez Armenta, and the rest of the 
scientists at UCSC (and the Genome Browser) for making these tracks possible.</p>

<a name="042920"></a>
<h2>Apr. 29, 2020 &nbsp;&nbsp; Updated NCBI RefSeq gene tracks for 43 assemblies</h2>
<p>
We are pleased to announce a combination of both updated and new
<a href="https://www.ncbi.nlm.nih.gov/refseq/" target="_blank">
NCBI RefSeq gene annotation</a> tracks for the
following 43 assemblies:</p><br>
<div class="container">
  <div class="row">
  <div class="col-sm-4">
  <ul>
        <li>African clawed frog <a href="../cgi-bin/hgTrackUi?db=xenLae2&g=refSeqComposite"
         target=_blank>(xenLae2)</a></li>
        <li>Armadillo <a href="../cgi-bin/hgTrackUi?db=dasNov3&g=refSeqComposite"
         target=_blank>(dasNov3)</a></li>
        <li>Baboon <a href="../cgi-bin/hgTrackUi?db=papAnu4&g=refSeqComposite"
         target=_blank>(papAnu4)</a></li>
        <li>Bonobo <a href="../cgi-bin/hgTrackUi?db=panPan2&g=refSeqComposite"
         target=_blank>(panPan2)</a></li>
        <li>Budgerigar <a href="../cgi-bin/hgTrackUi?db=melUnd1&g=refSeqComposite"
         target=_blank>(melUnd1)</a></li>
        <li>Bushbaby <a href="../cgi-bin/hgTrackUi?db=otoGar3&g=refSeqComposite"
         target=_blank>(otoGar3)</a></li>
        <li>Cat <a href="../cgi-bin/hgTrackUi?db=felCat9&g=refSeqComposite"
         target=_blank>(felCat9)</a></li>
        <li>Chicken <a href="../cgi-bin/hgTrackUi?db=galGal6&g=refSeqComposite"
         target=_blank>(galGal6)</a></li>
        <li>Chimp <a href="../cgi-bin/hgTrackUi?db=panTro6&g=refSeqComposite"
         target=_blank>(panTro6)</a></li>
        <li>Chinese hamster <a href="../cgi-bin/hgTrackUi?db=criGriChoV1&g=refSeqComposite"
         target=_blank>(criGriChoV1)</a></li>
        <li>Coelacanth <a href="../cgi-bin/hgTrackUi?db=latCha1&g=refSeqComposite"
         target=_blank>(latCha1)</a></li>
        <li>Cow <a href="../cgi-bin/hgTrackUi?db=bosTau9&g=refSeqComposite"
         target=_blank>(bosTau9)</a></li>
        <li>Crab-eating macaque <a href="../cgi-bin/hgTrackUi?db=macFas5&g=refSeqComposite"
         target=_blank>(macFas5)</a></li>
        <li>Dog <a href="../cgi-bin/hgTrackUi?db=canFam3&g=refSeqComposite"
         target=_blank>(canFam3)</a></li>
        <li>Elephant <a href="../cgi-bin/hgTrackUi?db=loxAfr3&g=refSeqComposite"
         target=_blank>(loxAfr3)</a></li>
   </ul></div>
   <div class="col-sm-4">
   <ul>
        <li>Elephant shark <a href="../cgi-bin/hgTrackUi?db=calMil1&g=refSeqComposite"
         target=_blank>(calMil1)</a></li>
        <li>Ferret <a href="../cgi-bin/hgTrackUi?db=musFur1&g=refSeqComposite"
         target=_blank>(musFur1)</a></li>
        <li>Garter snake <a href="../cgi-bin/hgTrackUi?db=thaSir1&g=refSeqComposite"
         target=_blank>(thaSir1)</a></li>
        <li>Green monkey <a href="../cgi-bin/hgTrackUi?db=chlSab2&g=refSeqComposite"
         target=_blank>(chlSab2)</a></li>
        <li>Guinea pig <a href="../cgi-bin/hgTrackUi?db=cavPor3&g=refSeqComposite"
         target=_blank>(cavPor3)</a></li>
        <li>Hedgehog <a href="../cgi-bin/hgTrackUi?db=eriEur2&g=refSeqComposite"
         target=_blank>(eriEur2)</a></li>
        <li>Horse <a href="../cgi-bin/hgTrackUi?db=equCab3&g=refSeqComposite"
         target=_blank>(equCab3)</a></li>
        <li>Lizard <a href="../cgi-bin/hgTrackUi?db=anoCar2&g=refSeqComposite"
         target=_blank>(anoCar2)</a></li>
        <li>Malayan flying lemur <a href="../cgi-bin/hgTrackUi?db=galVar1&g=refSeqComposite"
         target=_blank>(galVar1)</a></li>
        <li>Manatee <a href="../cgi-bin/hgTrackUi?db=triMan1&g=refSeqComposite"
         target=_blank>(triMan1)</a></li>
        <li>Marmoset <a href="../cgi-bin/hgTrackUi?db=calJac3&g=refSeqComposite"
         target=_blank>(calJac3)</a></li>
        <li>Medium ground finch <a href="../cgi-bin/hgTrackUi?db=geoFor1&g=refSeqComposite"
         target=_blank>(geoFor1)</a></li>
        <li>Microbat <a href="../cgi-bin/hgTrackUi?db=myoLuc2&g=refSeqComposite"
         target=_blank>(myoLuc2)</a></li>
        <li>Minke whale <a href="../cgi-bin/hgTrackUi?db=balAcu1&g=refSeqComposite"
         target=_blank>(balAcu1)</a></li>
   </ul></div>
   <div class="col-sm-4">
   <ul>
        <li>Naked mole-rat <a href="../cgi-bin/hgTrackUi?db=hetGla2&g=refSeqComposite"
         target=_blank>(hetGla2)</a></li>
        <li>Orangutan <a href="../cgi-bin/hgTrackUi?db=ponAbe3&g=refSeqComposite"
         target=_blank>(ponAbe3)</a></li>
        <li>Panda <a href="../cgi-bin/hgTrackUi?db=ailMel1&g=refSeqComposite"
         target=_blank>(ailMel1)</a></li>
        <li>Pika <a href="../cgi-bin/hgTrackUi?db=ochPri3&g=refSeqComposite"
         target=_blank>(ochPri3)</a></li>
        <li>Rabbit <a href="../cgi-bin/hgTrackUi?db=oryCun2&g=refSeqComposite"
         target=_blank>(oryCun2)</a></li>
        <li>Rat <a href="../cgi-bin/hgTrackUi?db=rn6&g=refSeqComposite"
         target=_blank>(rn6)</a></li>
        <li>Shrew <a href="../cgi-bin/hgTrackUi?db=sorAra2&g=refSeqComposite"
         target=_blank>(sorAra2)</a></li>
        <li>Squirrel <a href="../cgi-bin/hgTrackUi?db=speTri2&g=refSeqComposite"
         target=_blank>(speTri2)</a></li>
        <li>Squirrel monkey <a href="../cgi-bin/hgTrackUi?db=saiBol1&g=refSeqComposite"
         target=_blank>(saiBol1)</a></li>
        <li>Tarsier <a href="../cgi-bin/hgTrackUi?db=tarSyr2&g=refSeqComposite"
         target=_blank>(tarSyr2)</a></li>
        <li>Tasmanian devil <a href="../cgi-bin/hgTrackUi?db=sarHar1&g=refSeqComposite"
         target=_blank>(sarHar1)</a></li>
        <li>Tibetan frog <a href="../cgi-bin/hgTrackUi?db=nanPar1&g=refSeqComposite"
         target=_blank>(nanPar1)</a></li>
        <li>White rhinoceros <a href="../cgi-bin/hgTrackUi?db=cerSim1&g=refSeqComposite"
         target=_blank>(cerSim1)</a></li>
        <li>Zebrafish <a href="../cgi-bin/hgTrackUi?db=danRer11&g=refSeqComposite"
         target=_blank>(danRer11)</a></li>
    </ul></div>
</div></div>

<p>
Thanks to <a target="_blank" href="https://www.ncbi.nlm.nih.gov/refseq/">
NCBI RefSeq</a> for providing these annotations. We would also like to thank Hiram Clawson 
and Lou Nassar of the UCSC Genome Browser group for their
work in creating and releasing these tracks.</p>

<a name="042820"></a>
<h2>Apr. 28, 2020 &nbsp;&nbsp; gnomAD Constraint Metrics and Structural Variants data now available
on human (GRCh37/hg19) assembly</h2>

<p>
We are pleased to announce two new Genome Aggregation Database (gnomAD) tracks,
<a target="_blank"href="../../cgi-bin/hgTrackUi?db=hg19&c=chrX&g=gnomadPLI">Predicted Constraint Metrics</a> and
<a target="_blank"href="../../cgi-bin/hgTrackUi?db=hg19&c=chrX&g=gnomadStructuralVariants">Structural Variants</a>,
for the human (GRCh37/hg19) assembly.
</p>

<h3>Predicted Constraint Metrics</h3>
<p>
These data show various metrics of pathogenicity per-gene as predicted for gnomAD v2.1.1 and
identifies genes subject to strong selection against various classes of mutation. It is comprised
of three subtracks:

<ul>
<li>
<b>gnomAD Gene Constraint</b> - gnomAD Predicted Constraint Metrics By Gene
</li>
<li>
<b>gnomAD Transcript Constraint</b> - gnomAD Predicted Constraint Metrics By Transcript
<li>
<b>gnomAD Regional Constraint</b> - gnomAD Predicted Constraint Metrics By Transcript Regions
</li>
</ul>
</p>

<h3>Structural Variants</h3>
<p>
These data show structural variants calls (&gt;=50 nucleotides) from the gnomAD v2.1 release on
10,847 unrelated genomes. It is comprised of three subtracks:

<ul>
<li>
<b>All SV's</b> - gnomAD Structural Variants All
</li>
<li>
<b>Control Only SV's</b> - gnomAD Structural Variants Controls Only
<li>
<b>Non-neuro SV's</b> - gnomAD Structural Variants Non-neuro Only
</li>
</ul>
</p>

<p>
These data can be found as part of the <a target="_blank"
href="../../cgi-bin/hgTrackUi?db=hg19&g=gnomadSuper">gnomAD</a> super-track. More information on
this track can be found in the track description pages, as well as the <a target="_blank"
href="https://gnomad.broadinstitute.org/faq">gnomAD</a> site. 
</p>

<p class="text-center">
  <img class='text-center' src="../images/GnomadConstraintStructruralVarRelease.PNG"  width='80%' 
alt="Example of Constraint Metrics and Structural Variants tracks">
</p>

<p>
We would like to thank the <a target="_blank"
href="https://gnomad.broadinstitute.org">Genome Aggregation Database Consortium</a> for making
these data available. We would also like to thank Christopher Lee, Maximilian Haeussler, Lou Nassar,
Jairo Navarro, Robert Kuhn and Anna Benet-Pages for their effort in the creation of these tracks.
</p>

<a name="042020"></a>
<h2>Apr. 20, 2020 &nbsp;&nbsp; New video on the Browser's YouTube channel</h2>

<p>
We have released a new video to the Browser's
<a href = "https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos"
target = _blank>YouTube channel</a>.
</p>

<p>
   <b> &nbsp; &nbsp; Browser Basics, Part Two &mdash; Configuring the Browser </b>
</p>

<p>
This <a href =
"https://bit.ly/ucscVid18" target = _blank>video</a>
is part of a three-part series that goes back to the Basics,
designed to introduce new users to the Browser.

Our previous videos have been directed
either at experienced users who may have questions about how to do certain
tasks, or serve as introductions to new features in the Browser.
In this three-part series, we fill in the background with more basic
information, though we expect experienced users to find some new things
as well.
</p>

<p>
Today's release, Part Two, shows how to configure the Browser with
special attention to how you might prepare a Browser session for
export to a poster or paper. Examples include changing font size
and screen width, removing blue guidelines, navigating quickly
to nearby annotations
and more.
</p>

<p>
The final installment will focus on more configuration features
of the Browser and on navigating using DNA sequence.
As with all our videos, a transcript is provided for those who
wish to review the content before watching.
</p>

<ul>
<li>
Part One:  Getting around in the Browser</li>
<li>
Part Two:  Configuring the Browser </li>
<li>
Part Three:  Configuration + DNA Navigation </li>
</ul>
</p>

<a name="041420"></a>
<h2>Apr. 14, 2020 &nbsp;&nbsp; Updated Ensembl gene tracks for 43 assemblies</h2>
<p>
We are pleased to announce an update to the
<a href="http://uswest.ensembl.org/info/genome/genebuild/index.html" target="_blank">
Ensembl gene annotation</a> tracks to <strong>v99</strong> for the
following assemblies:</p><br>
<div class="container">
  <div class="row">
  <div class="col-sm-4">
  <ul>
        <li>Armadillo <a href="../cgi-bin/hgTrackUi?db=dasNov3&g=ensGene"
         target=_blank>(dasNov3)</a></li>
        <li>Bison <a href="../cgi-bin/hgTrackUi?db=bisBis1&g=ensGene"
         target=_blank>(bisBis1)</a></li>
        <li>Bonobo <a href="../cgi-bin/hgTrackUi?db=panPan2&g=ensGene"
         target=_blank>(panPan2)</a></li>
        <li>Budgerigar <a href="../cgi-bin/hgTrackUi?db=melUnd1&g=ensGene"
         target=_blank>(melUnd1)</a></li>
        <li>C. elegans <a href="../cgi-bin/hgTrackUi?db=ce11&g=ensGene"
         target=_blank>(ce11)</a></li>
        <li>C. intestinalis <a href="../cgi-bin/hgTrackUi?db=ci3&g=ensGene"
         target=_blank>(ci3)</a></li>
        <li>Cat <a href="../cgi-bin/hgTrackUi?db=felCat9&g=ensGene"
         target=_blank>(felCat9)</a></li>
        <li>Chicken <a href="../cgi-bin/hgTrackUi?db=galGal6&g=ensGene"
         target=_blank>(galGal6)</a></li>
        <li>Chimp <a href="../cgi-bin/hgTrackUi?db=panTro5&g=ensGene"
         target=_blank>(panTro5)</a></li>
        <li>Chinese hamster <a href="../cgi-bin/hgTrackUi?db=criGriChoV1&g=ensGene"
         target=_blank>(criGriChoV1)</a></li>
        <li>Chinese hamster <a href="../cgi-bin/hgTrackUi?db=criGriChoV2&g=ensGene"
         target=_blank>(criGriChoV2)</a></li>
        <li>Coelacanth <a href="../cgi-bin/hgTrackUi?db=latCha1&g=ensGene"
         target=_blank>(latCha1)</a></li>
        <li>Cow <a href="../cgi-bin/hgTrackUi?db=bosTau9&g=ensGene"
         target=_blank>(bosTau9)</a></li>
        <li>Crab-eating macaque <a href="../cgi-bin/hgTrackUi?db=macFas5&g=ensGene"
         target=_blank>(macFas5)</a></li>
        <li>D. melanogaster <a href="../cgi-bin/hgTrackUi?db=dm6&g=ensGene"
         target=_blank>(dm6)</a></li>
   </ul></div>
   <div class="col-sm-4">
   <ul>
        <li>Dog <a href="../cgi-bin/hgTrackUi?db=canFam3&g=ensGene"
         target=_blank>(canFam3)</a></li>
        <li>Elephant <a href="../cgi-bin/hgTrackUi?db=loxAfr3&g=ensGene"
         target=_blank>(loxAfr3)</a></li>
        <li>Elephant shark <a href="../cgi-bin/hgTrackUi?db=calMil1&g=ensGene"
         target=_blank>(calMil1)</a></li>
        <li>Ferret <a href="../cgi-bin/hgTrackUi?db=musFur1&g=ensGene"
         target=_blank>(musFur1)</a></li>
        <li>Gibbon <a href="../cgi-bin/hgTrackUi?db=nomLeu3&g=ensGene"
         target=_blank>(nomLeu3)</a></li>
        <li>Golden snub-nosed monkey <a href="../cgi-bin/hgTrackUi?db=rhiRox1&g=ensGene"
         target=_blank>(rhiRox1)</a></li>
        <li>Green monkey <a href="../cgi-bin/hgTrackUi?db=chlSab2&g=ensGene"
         target=_blank>(chlSab2)</a></li>
        <li>Guinea pig <a href="../cgi-bin/hgTrackUi?db=cavPor3&g=ensGene"
         target=_blank>(cavPor3)</a></li>
        <li>Horse <a href="../cgi-bin/hgTrackUi?db=equCab3&g=ensGene"
         target=_blank>(equCab3)</a></li>
        <li>Lamprey <a href="../cgi-bin/hgTrackUi?db=petMar2&g=ensGene"
         target=_blank>(petMar2)</a></li>
        <li>Lizard <a href="../cgi-bin/hgTrackUi?db=anoCar2&g=ensGene"
         target=_blank>(anoCar2)</a></li>
        <li>Microbat <a href="../cgi-bin/hgTrackUi?db=myoLuc2&g=ensGene"
         target=_blank>(myoLuc2)</a></li>
        <li>Naked mole-rat <a href="../cgi-bin/hgTrackUi?db=hetGla1&g=ensGene"
         target=_blank>(hetGla1)</a></li>
        <li>Naked mole-rat <a href="../cgi-bin/hgTrackUi?db=hetGla2&g=ensGene"
         target=_blank>(hetGla2)</a></li>
        <li>Orangutan <a href="../cgi-bin/hgTrackUi?db=ponAbe2&g=ensGene"
         target=_blank>(ponAbe2)</a></li>
    </ul></div>
<div class="col-sm-4">
   <ul>
        <li>Panda <a href="../cgi-bin/hgTrackUi?db=ailMel1&g=ensGene"
         target=_blank>(ailMel1)</a></li>
        <li>Pig <a href="../cgi-bin/hgTrackUi?db=susScr11&g=ensGene"
         target=_blank>(susScr11)</a></li>
        <li>Rabbit <a href="../cgi-bin/hgTrackUi?db=oryCun2&g=ensGene"
         target=_blank>(oryCun2)</a></li>
        <li>Rhesus <a href="../cgi-bin/hgTrackUi?db=rheMac10&g=ensGene"
         target=_blank>(rheMac10)</a></li>
        <li>S. cerevisiae <a href="../cgi-bin/hgTrackUi?db=sacCer3&g=ensGene"
         target=_blank>(sacCer3)</a></li>
        <li>Sheep <a href="../cgi-bin/hgTrackUi?db=oviAri3&g=ensGene"
         target=_blank>(oviAri3)</a></li>
        <li>Squirrel <a href="../cgi-bin/hgTrackUi?db=speTri2&g=ensGene"
         target=_blank>(speTri2)</a></li>
        <li>Squirrel monkey <a href="../cgi-bin/hgTrackUi?db=saiBol1&g=ensGene"
         target=_blank>(saiBol1)</a></li>
        <li>Tarsier <a href="../cgi-bin/hgTrackUi?db=tarSyr2&g=ensGene"
         target=_blank>(tarSyr2)</a></li>
        <li>Tasmanian devil <a href="../cgi-bin/hgTrackUi?db=sarHar1&g=ensGene"
         target=_blank>(sarHar1)</a></li>
        <li>Tetraodon <a href="../cgi-bin/hgTrackUi?db=tetNig2&g=ensGene"
         target=_blank>(tetNig2)</a></li>
        <li>Turkey <a href="../cgi-bin/hgTrackUi?db=melGal1&g=ensGene"
         target=_blank>(melGal1)</a></li>
        <li>X. tropicalis <a href="../cgi-bin/hgTrackUi?db=xenTro9&g=ensGene"
         target=_blank>(xenTro9)</a></li>
    </ul></div>
</div></div>
<p>
Thanks to Hiram Clawson and Lou Nassar of the UCSC Genome Browser group for their
work in creating and releasing these tracks.</p>


<a name="040620"></a>
<h2>Apr. 6, 2020 &nbsp;&nbsp; New video on the Browser's YouTube channel</h2>

<p>
We have released a new video to the Browser's
<a href = "https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos"
target = _blank>YouTube channel</a>.
</p>

<p>
   <b> &nbsp; &nbsp; Browser Basics, Part One &mdash; Getting around in the Browser </b>
</p>

<p>
This <a href =
"https://bit.ly/ucscVid17" target = _blank>video</a>
is part of a three-part series that goes back to the Basics,
designed to introduce new users to the Browser.

Our previous videos have been directed
either at experienced users who may have questions about how to do certain
tasks, or serve as introductions to new features in the Browser.
Here we fill in the background with more basic information, though we expect
experienced users to find some gems as well.
</p>

<p>
Today's release, Part One, shows many ways to get to a chosen location in
the Browser using accession names in a variety of formats:  gene names,
genomic coordinates, codon numbers, SNP rsIDs, RefSeq NM_ identifiers and
more.
As with all our videos, a transcript is provided for those who
wish to review the content before watching.
</p>

<p>
Future installments will focus on configuration features of the Browser.
</p>

<ul>
<li>
Part One:  Getting around in the Browser</li>
<li>
Part Two:  Configuring the Browser </li>
<li>
Part Three:  Configuration + DNA Navigation </li>
</ul>
</p>

<p>
Rest in Peace, James Taylor of Galaxy.  You will be missed.
</p>


<a name="040320"></a>
<h2>Apr. 2, 2020 &nbsp;&nbsp; First data release for SARS-CoV-2 genome browser</h2>
<p>
In recent months, we have seen the beginning of the global effort against the 
coronavirus. Here at the UCSC Genome Browser, we have also been directing 
some work towards that front and will continue to do so.</p>
<p>
This has brought new users to our site who may have previously been unfamiliar 
with this resource. The Genome Browser aims to facilitate genome 
research by offering data visualization, genome annotations, and other tools. 
We encourage anyone who would like to learn more to see our 
<a target="_blank" href="help/hgTracksHelp.html#GetStarted">user guide</a>.</p>
<p>
With this in mind, we would like to announce a new
<a target="_blank" href="/covid19.html">COVID-19 Resources at UCSC</a> 
landing page as well as the first release of novel coronavirus annotation data for the 
<a target="_blank" href="/cgi-bin/hgGateway?db=wuhCor1">SARS-CoV-2 genome assembly browser</a> 
released in early February. The SARS-CoV-2 genome browser includes displays of the virus's 
molecular evolution in other species and its further evolution during this human 
pandemic. We have also added multiple lung datasets to the <a target="_blank" 
href="https://cells.ucsc.edu/?bp=lung">UCSC Single Cell Browser</a>.

This information is made freely available to researchers everywhere, 
with the goal of advancing our knowledge of SARS-CoV-2 (COVID-19).</p>
<p>
These latest data were primarily sourced from outside groups and include different 
kinds of information such as gene annotations, variant data, and locally produced 
multiple genome alignments.</p>
<p>
<small><b>Note:</b> Genome Browser data is often referred to as 'tracks', and the 
term 'track' and 'data annotation track' can be used interchangeably.</small></p><br>

<ul>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=ncbiGeneBGP">
NCBI Genes</a> - This track shows genes annotated on the SARS-CoV-2 genome released 
by the National Center for Biotechnology Information (NCBI) on 2/16/20.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=nextstrainGene">
Nextstrain Genes</a> - This track shows genes annotated by Nextstrain.org in relation to 
their collection and processing of SARS-CoV-2 variant data from the Global Initiative 
on Sharing All Influenza Data (GSAID).</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=unipCov2AliSwissprot">
UniProt Protein Annotations</a> - A collection of tracks that show protein sequence annotations 
from the UniProt/SwissProt database, mapped to genomic coordinates. All data has been 
curated from scientific publications by the UniProt/SwissProt staff. These data are comprised
of 11 data tracks, some of these include:
  <ul>
  <li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=unipCov2AliSwissprot">
UniProt full-length proteins</a> - The protein sequences from SwissProt mapped onto the 
genome, genomic coordinates for other UniProt feature tracks are based on these data.</li>
  <li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=unipCov2Interest">
UniProt highlighted 'Regions of Interest'</a> - This track shows protein sequence annotations 
defined as "regions of interest" from the UniProt/SwissProt database, these data have been
curated from scientific publications by the UniProt/SwissProt staff.</li>
  <li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=unipCov2Chain">
UniProt Mature, Processed Protein Products (Polypeptide Chains)</a> - Polypeptide chain in 
mature protein after post-processing.</li>
  <li>
For a full list of available UniProt tracks, see <b>UniProt Protein Annotations</b> in 
the <a target="_blank" href="/cgi-bin/hgTracks?db=wuhCor1">SARS-CoV-2 browser</a>.</li></ul></li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=epitopes">
Immune Epitope Database and Analysis Resource (IEDB) Epitopes</a> - This track indicates the 
immune epitope predictions for B cells, CD4 T-cells and CD8 T-cells, using varying 
software packages.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=primers">
RT-PCR Primers</a> - This track shows RT-PCR Primers in viral detection kits aligned to 
the SARS-CoV-2 genome from six different sources, including government agencies from 
the US, China, Japan, and Thailand.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=nextstrainSamples">
Nextstrain Variants</a> - This track displays all single-nucleotide variants in the 
thousands of SARS-CoV-2 genome sequences from GISAID collected and processed 
by Nextstrain.org. This track can be used to examine variation, protein 
changes, and sequence conservation among SARS-CoV-2 sequences.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=nextstrainClade">
Nextstrain Clades</a> - This track shows the location of variants that distinguish each 
of the branches of interest defined by Nextstrain.org and can be used in conjunction 
with their tree and map diagrams to examine viral lineage.</li>
<li>
<a target="_blank" href="/cgi-bin/hgTrackUi?db=wuhCor1&c=NC_045512v2&g=strainCons44way">
Multiz Alignment & Conservation (44 Strains with bats as hosts)</a> - This track shows 
multiple alignments of 44 virus sequences, aligned to the SARS-CoV-2 reference 
sequence SARS-CoV-2/NC_045512.2. It also includes measurements of evolutionary 
conservation using two methods (phastCons and phyloP) from the PHAST package, 
for all 44 virus sequences.</li></ul>

<p>
More information on any of these data annotation tracks can be found by clicking 
on the names. This will lead to the track description page, which also allows 
for configuration of various display options. Additional information on how to use 
the UCSC Genome Browser can be found on our <a target="_blank" 
href="/training/index.html">training page</a>.</p>
<p>
We would like to thank NCBI, UniProt, Nextstrain, the Sgourakis Research Group, 
Tomer Altman, and Jason Fernandes as well as the scientists from UCSC for providing these data.</p>
<p>
The UCSC Genome Browser is funded by the NIH National Human Genome Research Institute. In addition 
to this funding, the SARS-CoV-2 assembly and data were made possible by generous individual donors 
including local donors Pat and Roland Rebele.</p>

<a name="031720"></a>
<h2>Mar. 17, 2020 &nbsp;&nbsp; New mitochondrial sequence for human (hg19)</h2>
<p>
We are pleased to announce the release of a patch to the hg19 assembly that will introduce a new
mitochondrial sequence, <a href="../../cgi-bin/hgTracks?db=hg19&position=chrMT:4277-5648"
target="_blank">chrMT</a>, to the assembly. We used GenBank sequence NC_001807 for chrM in hg19 and
earlier, but the sequence preferred by the community is the revised Cambridge Reference Sequence
(rCRS), <a href="https://www.ncbi.nlm.nih.gov/nuccore/251831106" target="_blank">NC_012920</a>. The
new chrMT is the rCRS, NC_012920. <a href="https://www.ncbi.nlm.nih.gov/grc/help/patches/"
target="_blank">Patch sequences</a> from
<a href="https://www.ncbi.nlm.nih.gov/assembly/GCA_000001405.14" target="_blank">GRCh37.p13</a> have
also been added to hg19.</p>
<p>
More information on how patch sequences are incorporated can be found on the
<a href="https://genome-blog.gi.ucsc.edu/blog/patches/" target="_blank">Patching up the Genome</a> blog post.
The blog post contains details about the new
<a href="https://hgdownload.gi.ucsc.edu/goldenPath/hg19/bigZips/latest/"
target="_blank">/latest</a> download directory on the downloads server. With the addition of new
sequences to hg19, we can expect to see BLAT return more matches to the genome.</p>
<p>
Also, with these patches, the hg19 genome is not optimal anymore for aligners. So we added an
&quot;<a href="https://hgdownload.gi.ucsc.edu/goldenPath/hg19/bigZips/analysisSet/"
target="_blank">analysis set</a>&quot; version of the hg19 genome fasta file to our
<a href="https://hgdownload.gi.ucsc.edu/goldenPath/hg19/bigZips/" target="_blank">bigZips</a>
directory, and indexes for BWA, Bowtie2, and Hisat2.</p>
<p>
We would like to thank the Genome Reference Consortium for creating the patches to hg19. We would
also like to thank Angie Hinrichs and Jairo Navarro at UCSC for implementing and testing the latest
patch to hg19.</a>

<a name="022720"></a>
<h2>Feb. 27, 2020 &nbsp;&nbsp; New NCBI RefSeq Select + MANE track for Human (hg38)</h2>
<p>
We are pleased to announce a new track, RefSeq Select+MANE, for the <a 
target="_blank" href="../../cgi-bin/hgTrackUi?db=hg38&c=chr1&g=refSeqComposite">GRCh38/hg38</a> 
human assembly. This track is a combination of NCBI transcripts with the <em>RefSeq Select</em> 
tag, as well as transcripts with the <em>MANE Select</em> tag. The result is a track with a 
single representative transcript for every protein-coding gene. The track can be 
found as part of the NCBI RefSeq composite track.</p>
<p>
<a target="_blank" href="https://www.ncbi.nlm.nih.gov/refseq/refseq_select/">RefSeq Select</a> 
transcripts are chosen by an NCBI pipeline to be representative of every protein-coding 
gene. As part of the <a target="_blank" 
href="https://www.ncbi.nlm.nih.gov/refseq/MANE/">MANE project</a>, however, RefSeq 
Select transcripts that have a 100% identical match to a transcript in the 
Ensembl annotation are given the MANE Select designation. In this way, MANE Select 
is a subset of RefSeq Select. It should be noted, however, that there are 
cases where RefSeq Select transcripts are replaced when a different MANE Select 
transcript is chosen. Once the MANE project is complete, RefSeq Select and MANE 
Select for human should be nearly the same.</p>
<p>
In summary, this track provides a functional canonical gene set until the MANE 
set is complete. More information on this track and data can be found in the <a target="_blank" 
href="../../cgi-bin/hgTrackUi?db=hg38&c=chr1&g=refSeqComposite">track description page</a>, 
NCBI's <a target="_blank" href="https://www.ncbi.nlm.nih.gov/refseq/refseq_select/">RefSeq Select 
page</a>, and the <a target="_blank" href="https://www.ncbi.nlm.nih.gov/refseq/MANE/">MANE page</a>.</p>
<p>
We would like to thank NCBI for releasing the RefSeq Select data, as well as the NCBI and
Ensembl-GENCODE collaboration for MANE. We would also like to thank Max Haussler and Lou 
Nassar for the development and release of these tracks.</p>

<a name="021120"></a>
<h2>Feb. 11, 2020 &nbsp;&nbsp; New GENCODE gene tracks: Human V33 (hg19/hg38) - Mouse M24 
(mm10)</h2>
<p>
We are pleased to announce new GENCODE Gene annotation tracks, which correspond
to <a href="http://www.ensembl.info/2019/10/17/whats-coming-up-in-ensembl-release-99/"
target="_blank">Ensembl 99</a>, for three assemblies: <a target="_blank" 
href="../../cgi-bin/hgTrackUi?db=hg19&c=chr21&g=wgEncodeGencodeV33lift37">hg19/GRCh37</a>, 
<a target="_blank" href="../../cgi-bin/hgTrackUi?db=hg38c=chr1&g=wgEncodeGencodeV33">hg38/GRCh38</a>, 
and <a target="_blank" href="../../cgi-bin/hgTrackUi?db=mm10&c=chr12&g=wgEncodeGencodeVM24">mm10/GRCm38</a>.
For human, the GENCODE V33 annotations were mapped to hg38/GRCh38 and then back-mapped 
to the hg19/GRCh37 assembly. For all three assemblies, the gene sets contain the following 
tracks:</p>
<ul>
  <li>Basic - a subset of the <em>Comprehensive set</em>.</li>
  <li>Comprehensive - all GENCODE coding and non-coding transcript annotations, including
      polymorphic pseudogenes. This includes both manual and automatic annotations.</li>
  <li>Pseudogenes - all annotations except polymorphic pseudogenes.</li></ul>
<p>
The hg38 and mm10 assemblies also include the following tracks that are not available on hg19:
</p>
<ul>
  <li>2-way Pseudogenes - pseudogenes predicted by both the Yale Pseudopipe and UCSC
      Retrofinder pipelines.</li>
  <li>PolyA - polyA signals and sites manually annotated on the genome based on transcribed
      evidence (ESTs and cDNAs) of 3' end of transcripts containing at least 3 A's not matching the
      genome.</li></ul>
<p>
Details on each release can be found on the <a target="_blank" 
href="https://www.gencodegenes.org/">GENCODE site</a>. This includes <a target="_blank" 
href="https://www.gencodegenes.org/human/stats.html">statistics on each release</a>.
<p>We would like to thank the <a target="_blank" 
href="https://www.gencodegenes.org/pages/gencode.html">GENCODE project</a> for providing these 
annotations. We would also like to thank Mark Diekhans and Lou Nassar for the development and 
release of these tracks.</p>

<a name="021020"></a>
<h2>Feb. 10 2020 &nbsp;&nbsp; Updated data for NCBI RefSeq tracks for human assemblies hg19 
and hg38</h2>
<p>
We are happy to release the updated NCBI RefSeq Annotation Release 109.20190905 track on the
Genome Browser. The NCBI RefSeq Genes composite track shows human protein-coding and 
non-protein-coding genes taken from the NCBI RNA reference sequences collection (RefSeq). 
This update adds more than 500 new gene entries to the collection of nearly 167,000 annotations.
More information can be found on the track description pages for these composite
tracks for 
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=refSeqComposite">hg19</a> and 
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=refSeqComposite">hg38</a>.</p>

<p>
We would like to thank NCBI and the RefSeq Annotation database for collecting and curating
these data. We would also like to thank Hiram Clawson and Daniel Schmelter for their role
creating, documenting, and reviewing these tracks.</p>

<a name="020720"></a>
<h2>Feb. 7, 2020 &nbsp;&nbsp; New and updated Variants in Papers tracks: Avada variants 
(hg19) & Mastermind variants (hg19, hg38)</h2>
<p>
We are pleased to announce a new track, Avada Variants, now available on hg19. Additionally, 
we have updated the Mastermind Variants track and expanded it to hg38.</p>

<p class="text-center">
  <img class='text-center' src="../images/variantsInPapersNewsArch2020.png"  width='80%' 
alt="Example of AVADA and Mastermind tracks">
</p>

<h3>Avada Variants</h3>
<p>
The Avada Variants track shows the genomic positions of variants in the <a target="_blank"
href="http://bejerano.stanford.edu/AVADA/">AVADA database</a>. AVADA is a database of variants 
built by machine learning software that analyzes full text research articles in PDF format 
to find genes and variants that look most relevant for genetic diagnosis.</p>
<p>
Additional information can be found on the <a target="_blank" 
<a target="_blank" href="https://doi.org/10.1038/s41436-019-0643-6">AVADA publication</a>.</p>

<h3>Mastermind Variants</h3>
<p>
The <a href="#032619">Mastermind Variants</a> track is now available for the hg38 assembly 
alongside hg19. Both tracks will now be updated on a quarterly cycle. The track is also 
now filtered to make the data more specific by removing longer non-indels that are 
not directly mentioned in the text.</p>
<p>
Additional information on Mastermind Variants can be found in the <a target="_blank" 
href="../../cgi-bin/hgTrackUi?db=hg19&g=mastermind">track description page</a>.</p>
<p>
We would like to thank Gill Bejerano and Johannes Birgmeier for making the AVADA data
available, as well as Genomenon for the Mastermind track. We would also like to thank
Maximilian Haeussler, Lou Nassar, and the rest of the Genome Browser team for the 
creation of these tracks.</p>

<a name="012020"></a>
<h2>Jan. 20, 2020 &nbsp;&nbsp; 1000 Genomes Project Phase 3 variants now available for human
  (GRCh38/hg38)</h2>
<p>
We are happy to announce the release of the 1000 Genomes Phase 3 Integrated Variant Calls from the
<a href="https://www.internationalgenome.org/" target="_blank">International Genome Sample Resource
(IGSR)</a> for human
(<a href="../../cgi-bin/hgTrackUi?db=hg38&c=chrX&g=tgpPhase3" target="_blank">GRCh38/hg38</a>). This
track shows approximately 73 million single nucleotide variants (SNVs) and 5 million short insertions/deletions (indels)
produced by the IGSR from sequence data generated by the 1000 Genomes Project in its Phase 3
sequencing of 2,504 genomes from 16 populations worldwide. The genomes of 2,504 individuals were
sequenced using both whole-genome sequencing (mean depth = 7.4x) and targeted exome sequencing
(mean depth = 65.7x). Sequence reads were aligned to the reference genome using alt-aware BWA-MEM
(<a href="https://academic.oup.com/gigascience/article/6/7/gix038/3836916"
target="_blank">Zheng-Bradley et al.</a>). Variant discovery and quality control were performed as
described in (<a href="https://wellcomeopenresearch.org/articles/4-50" target="_blank">Lowy-Gallego
et al.</a>).</p>
<p>
Variants were called on the autosomes (chromosomes 1 through 22) and on the Pseudo-Autosomal Regions
(PARs) of chromosome X. Therefore this track has no annotations on alternate haplotype sequences,
fix patches, chromosome Y, or the non-PAR portion (the majority) of chromosome X.</p>
<p>
We would like to thank the IGSR for making these variant calls freely available. We would also like
to thank Angie Hinrichs and Jairo Navarro for their efforts in creating and reviewing this track.
</p>

<!-- ============= 2019 archived news ============= -->
<a name="2019"></a>

<a name="121319"></a>
<h2>Dec. 13, 2019 &nbsp;&nbsp; New dbSNP pipeline: dbSNP b153 release, bigDbSnp track type</h2>
<p>
We are pleased to announce a new dbSNP pipeline, along with the first new dataset: dbSNP 
b153 for <a target="_blank" href="../../cgi-bin/hgTrackUi?db=hg19&g=dbSnp153Composite">hg19</a> 
and <a target="_blank" href="../../cgi-bin/hgTrackUi?db=hg38&g=dbSnp153Composite">hg38</a>.</p>
<p> 
dbSNP has seen an <a target="_blank" 
href="https://ncbiinsights.ncbi.nlm.nih.gov/2018/07/02/dbsnp-database-doubles-size-twice-13-months/">
explosive growth</a> in recent releases, from roughly 324 million variants in <a target="_blank"
href="https://ncbiinsights.ncbi.nlm.nih.gov/2017/05/08/dbsnps-human-build-150-has-doubled-the-amount-of-refsnp-records/">
build 150</a>, to over 700 million variants in the latest build b153. In an effort to continue 
providing efficient access to these data, dbSNP has redesigned their <a target="_blank"
href="https://ncbiinsights.ncbi.nlm.nih.gov/2017/07/07/dbsnp-redesign-supports-future-data-expansion/">
architecture and data flow</a>. We have also taken this opportunity to redesign our dbSNP 
ingestion pipeline. Below is a short summary of the UCSC changes brought forth by both 
dbSNP's redesign, as well as UCSC's new pipeline.</p>
<ul>
  <li>New track type bigDbSnp</li>
  <li>Expanded dbSNP track composition</li>
  <li>New UCSC annotation of dbSNP data termed <b>UCSC Notes</b></li>
  <li>dbSNP data will now be a bigBed file download (see Data Access below)</li>
</ul>

<h3>bigDbSnp and dbSNP v153</h3>
<p>
&quot;SNPs&quot; tracks were previously based on related mysql database tables, but the new 
bigDbSnp format is a bigBed file with extra columns that contains all necessary information 
to display the variant. An accompanying dbSnpDetails file contains additional data displayed 
in the item details page. Schemas are available for both <a target="_blank" 
href="https://github.com/ucscGenomeBrowser/kent/blob/master/src/hg/lib/bigDbSnp.as">
bigDbSnp</a> and <a target="_blank" 
href="https://github.com/ucscGenomeBrowser/kent/blob/master/src/hg/lib/dbSnpDetails.as">
dbSnpDetails</a>.</p>
<p>
dbSNP's redesign includes an important change to the representation of insertion/deletion 
variants (indels) in repetitive regions.  Rather than annotating the minimal representation 
of the indel on the genome, dbSNP now expands the reference and alternate alleles to cover 
the entire repetitive region on the genome.  Accordingly, we display the newly expanded regions, 
but use thin and thick lines to indicate the region of uncertain placement and the minimal size.  
For example, when there is a deletion of one base in a range of three identical bases, we draw 
a thin rectangle over the first two bases to show that there is uncertain placement, and a thick 
rectangle over the last base to show that one base is deleted from the range. Some indel variants 
have multiple alternate alleles. You may notice some thick but gray rectangles when there are 
deletion alleles of different sizes.</p>
<p>
Below is an image from an <a target="_blank" href="https://genome.ucsc.edu/s/Lou/hg38DbSnpNews">
example session</a> which displays an instance of this new visualization. In a repetitive region 
with 11 As, the previous dbSNP track, dbSNP 151, displays three variants (blue). Two of them 
include a single base insertion, but are arbitrarily placed one base apart from each other. 
There is also a single-base deletion. In release 153, dbSNP has clustered them into a single 
variant spanning the 11 As on the genome. We indicate the uncertainty of placement within the 
11 As with a thin line across the entire repeat, and a thick gray rectangle over the final 
base to indicate that one base may be deleted (orange). At the right end there is a tall 
line to indicate an insertion(s).</p>

<p class="text-center">
  <img class='text-center' src="../images/bigDbSnpNewsArch.png"  width='80%'alt="Example of 
new bigDbSnp display">
</p>

<h3>Track Composition</h3>
<p>
dbSNP b153 is composed of 5 subtracks. Four of these closely correlate to our previous 
SNP releases; a new subtrack displays mappings with inconsistent coordinates in dbSNP 
download files:</p>
<ul>
  <li>Common dbSNP(153) - Common (1000 Genomes Phase 3 MAF >= 1%) variants</li>
  <li>ClinVar dbSNP(153) - Variants included in ClinVar</li>
  <li>Mult. dbSNP(153) - Variants that that map to multiple genomic loci</li>
  <li>All dbSNP(153) - All variants in dbSNP release</li>
  <li>Map Err dbSnp(153) - Mappings with inconsistent coordinates</li>
</ul>

<h3>UCSC Notes</h3>
<p>
While processing the information downloaded from dbSNP, UCSC annotates some properties of 
interest. These are noted on the item details page, and may be used to include or exclude 
affected variants. These UCSC notes (currently 26) can be divided into three categories:</p>
<ul>
  <li>Information about ClinVar status, allele frequencies reported by twelve projects, and 
the presence of other variants at the same genomic position</li>
  <li>Notes about rare variants or ambiguous nucleotides in the reference genome</li>
  <li>Indicators that allele frequency data might be incomplete and/or mapping variants across 
different assemblies had issues with indel differences between assemblies</li>
</ul>

<h3>Data Access</h3>
<p>
With the bigDbSnp format, these data will no longer be available as a database table dump. The
complete data can be found across two separate files in our download server, a bigBed file 
(bigDbSnp) for <a target="_blank" href="http://hgdownload.gi.ucsc.edu/gbdb/hg19/snp/">hg19</a> 
and <a target="_blank" href="http://hgdownload.gi.ucsc.edu/gbdb/hg38/snp/">hg38</a>, and a 
<a target="_blank" href="http://hgdownload.gi.ucsc.edu/gbdb/hgFixed/dbSnp/">shared secondary 
details file</a> which holds additional variant details.<p>
<p>
Additional information including visibility display, a complete list of UCSC notes, and a methods 
section can be found in the <a target="_blank" 
href="../../cgi-bin/hgTrackUi?db=hg38&g=dbSnp153Composite">track description 
page</a>.</p>
<p>
We would like to thanks the dbSNP group at NCBI for providing access to these data. We would 
also like to thank Angie Hinrichs and the UCSC Genome Browser team for their efforts on this
release.</p>

<a name="112719a"></a>
<h2>Nov. 27, 2019 &nbsp;&nbsp; New EPD TSS track for human and mouse</h2>
<p>
We are pleased to announce the release of the new EPDnew Promoters track for human
(<a href="../../cgi-bin/hgTrackUi?db=hg38&c=chrX&g=epdNew" target="_blank">hg38</a> and
<a href="../../cgi-bin/hgTrackUi?db=hg19&c=chrX&g=epdNew" target="_blank">hg19</a>)
and mouse (<a href="../../cgi-bin/hgTrackUi?db=mm10&c=chrX&g=epdNew" target="_blank">mm10</a>)
assemblies. These tracks represent the experimentally validated promoters generated by the
<a href="https://epd.epfl.ch/" target="_blank">Eukaryotic Promoter Database</a>, based on gene
transcript models obtained from multiple sources (HGNC, GENCODE, Ensembl, RefSeq), then validated
using data from CAGE and RAMPAGE experimental studies obtained from FANTOM 5, UCSC, and ENCODE. Peak
calling, clustering and filtering based on relative expression were applied to identify the most
expressed promoters and those present in the largest number of samples.</p>
<p>
We would like to thanks Philipp Bucher and the EPD team at the Swiss Bioinformatics Institute for
providing these data, and Kate Rosenbloom and Jairo Navarro at UCSC for creating and reviewing these
tracks.</p>

<a name="112719b"></a>
<h2>Nov. 27, 2019 &nbsp;&nbsp; New ENCODE Enhancer-Gene Interactions for mouse</h2>
<p>
We are pleased to announce a new
<a href="../../cgi-bin/hgTrackUi?db=mm10&c=chrX&g=encode3RenEnhancerEpdNewPromoter"
target="_blank">Enhancer-Gene Map</a> for the mouse mm10 assembly. This track set presents
enhancer-promoter interactions predicted from correlation of enhancer-associated chromatin signals
and gene expression across tissue stages, based on histone modification (ChIP-seq) and transcription
(RNA-seq) assays and analysis performed in the ENCODE project as part of the Mouse Developmental
Series (part of ENCODE phase 3). Data underlying this track are presented in the Histone
Modifications and Chromatin State tracks, part of the ENCODE Regulation supertrack. The promoters
in this track were derived from experimentally validated promoters provided by the Eukaryotic
Promoter Database (EPDNew). A more complete presentation of this annotation can be found in the
EPDnew Promoters track.</p>
<p>
We would like to thank Iros Barozzi and colleagues in the Environmental Genomics and Systems Biology
Division at the Lawrence Berkeley Laboratory, and David Gorkin and Yanxiao Zhang at the Ren lab
(UCSD/Ludwig Institute for Cancer Research) and the ENCODE Consortium for providing these data
and assisting with track development at UCSC. We also thank Kate Rosenbloom and Jairo Navarro at
UCSC for creating and reviewing these tracks.</p>

<a name="112519"></a>
<h2>
Nov. 25, 2019 &nbsp;&nbsp; LOVD track split to separate long and short variants for human</h2>
<p>
We are glad to announce an update to the <a href="../../cgi-bin/hgTrackUi?db=hg19&c=chrX&g=lovdComp"
target="_blank">LOVD track</a> for human (GRCh37/hg19). The LOVD track is now a composite track
with the following two subtracks:</p>
<ul>
  <li>LOVD Variants &lt;= 100bp</li>
  <li>LOVD Variants &gt; 100bp</li>
</ul>
<p>
With this new update, the option is now available to only show the short variants and exclude
the long variants, or vice versa.</p>
<p>
We would like to than Maximillian Haussler and Jairo Navarro for creating and testing these changes.
We would also like to thank the LOVD team, Ivo Fokkema, Peter Taschner, Johan den Dunnen, and all
LOVD curators who gave permission to show their data.</p>

<a name="111319"></a>
<h2>Nov. 13, 2019 &nbsp;&nbsp; Hi-C Heatmaps now supported for Custom Tracks and Track Hubs</h2>
<p>
We are happy to announce that Hi-C heatmaps can now be used to visualize 
custom Hi-C data as Custom Tracks and Track Hubs. This heatmap visualization shows 
interaction scores for contact sequencing methods which may indicate 
enhancer-promoter interactions. Supporting users' Hi-C data is the next 
step in our support for 3C sequencing data, following the 
<a href="#102519b">popular Hi-C tracks</a> on 
<a href="../cgi-bin/hgTracks?db=hg19&hideTracks=1&knownGene=pack&rao2014Hic=full" 
target="_blank">hg19</a> and <a 
href="../cgi-bin/hgTracks?db=hg38&hideTracks=1&knownGene=pack&hicAndMicroC=full" 
target="_blank">hg38</a> as well as the
<a href=#081018>Interact format used in GeneHancer</a> for low-density 
interactions.</p>

<p>
Information and examples of displaying your own Hi-C data can be found 
on the <a href="help/hic.html">hic track format help page</a>.

<a name="110519"></a>
<h2>Nov. 5, 2019 &nbsp;&nbsp; Expanded CRISPR track released for mouse (GRCm38/mm10)</h2>
<p>
We are pleased to announce the release of an expanded
 <a href="../cgi-bin/hgTracks?db=mm10&hideTracks=1&knownGene=pack&crisprAllTargets=pack&position=chr1:11186551-11186838" target="_blank">CRISPR Targets track</a> for the mouse (GRCm38/mm10)
assembly. This track shows the DNA sequences targetable by CRISPR RNA guides
using the Cas9 enzyme from <i>S. pyogenes</i> (PAM: NGG) over the entire mouse genome. CRISPR target
sites were annotated with predicted specificity (off-target effects) and predicted efficiency
(on-target cleavage) by various algorithms through the tool
<a href="http://crispor.tefor.net/" target="_blank">CRISPOR</a>. The target sequence of the guide
is shown with a thick (exon) bar. The PAM motif match (NGG) is shown with a thinner bar.
</p>
<p class="text-center">
  <img class='text-center' src="../images/mm10CrisprAll.png" width='800' alt="CrisprAllTargets track">
</p>

<p>
We would like to thank Maximilian Haeussler, Hiram Clawson, and Lou Nassar for their effort
creating and releasing this data track.</p>

<a name="110119"></a>
<h2>Nov. 1, 2019 &nbsp;&nbsp; New ENCODE 3 Regulation super-track set available on mouse (GRCm38/mm10) assembly</h2>
<p>
We are pleased to announce the new <a target="_blank" 
href="../cgi-bin/hgTrackUi?db=mm10&g=encode3Reg">ENCODE 3 Regulation super-track</a>
for the mouse (GRCm38/mm10) assembly. The <a target="_blank" href="https://www.genome.gov/Funded-Programs-Projects/ENCODE-Project-ENCyclopedia-Of-DNA-Elements">ENCODE project</a> 
has established an epigenomic resource for mammalian development, profiling a diverse panel of
mouse tissues at eight developmental stages from 10.5 days post conception until birth. The ENCODE
3 Regulation super-track is composed of the three following composite tracks:
<ul>
  <li> <a target="_blank" href="../cgi-bin/hgTrackUi?db=mm10&g=encode3RenChromHmm">Chromatin State</a> - 
  presents chromatin state annotations derived from ChIP-seq of histone modifications.</li>
  <li> <a target="_blank" href="../cgi-bin/hgTrackUi?db=mm10&g=encode3RenHistone">Histone Modifications</a> -
  presents the results of a comprehensive study of chromatin state across these developmental
  stages, based on 1,128 ChiP-seq assays of 8 histone modifications in 12 tissues.</li>
  <li> <a target="_blank" href="../cgi-bin/hgTrackUi?db=mm10&g=encode3RenAtac">Open Chromatin</a> -
  presents the results of a comprehensive study of chromatin accessibility across these
  developmental stages, based on 122 ATAC-seq assays in 12 tissues.</li>
</ul></p>
<p class="text-center">
  <img class='text-center' src="../images/enR.png" width='700' alt="Example of the ENCODE 3 Regulation supertrack">
</p>

<p>
Thanks to David Gorkin and Yanxiao Zhang at the <a target="_blank"
href="http://renlab.sdsc.edu/renlab_website//">Ren lab (UCSD/Ludwig Institute for Cancer Research)</a>
and Iros Barozzi of the <a target="_blank" 
href="https://biosciences.lbl.gov/divisions/egsb/">Environmental Genomics and Systems Biology Division</a>
at the Lawrence Berkeley National Laboratory for providing these data and assisting with track
development at UCSC. We would also like to thank Kate Rosenbloom, Conner Powell and the UCSC Genome
Browser team for their efforts on this release.
</p>

<a name="103119"></a>
<h2>Oct. 31, 2019 &nbsp;&nbsp; Locus Reference Genomic (LRG) update for human</h2>
<p>
We are pleased to announce an update to the Locus Reference Genomic (LRG) regions track for human,
(<a href="../../cgi-bin/hgTrackUi?db=hg19&c=chrX&g=lrg" target="_blank">GRCh37/hg19</a>) and
<a href="../../cgi-bin/hgTrackUi?db=hg38&c=chrX&g=lrg" target="_blank">(GRCh38/hg38)</a>. Each LRG
record also includes at least one stable transcript on which variants may be reported. These
transcripts appear in the LRG Transcripts track in the Gene and Gene Predictions track section.</p>
<p>
This track was produced at UCSC using <a href="ftp://ftp.ebi.ac.uk/pub/databases/lrgex/"
target="_blank">LRG XML files</a>. Thanks to
<a href="http://www.lrg-sequence.org/documentation/lrg-collaborators/"
target="_blank">LRG collaborators</a> for making these data available. We would also like to thank
Angie Hinrichs and Jairo Navarro for creating and releasing these tracks.</p>

<a name="102819"></a>
<h2>Oct. 28, 2019 &nbsp;&nbsp; gnomAD data now available on human (GRCh38/hg38) assembly</h2>
<p>
We are pleased to announce the new <a target="_blank" 
href="../cgi-bin/hgTrackUi?db=hg38&c=chr22&g=gnomadVariants">gnomAD Variants super-track</a>
for the Human (GRCh38/hg38) assembly. This super-track currently includes <strong>gnomAD v3</strong>
and <strong>gnomAD v2</strong>.</p>
<p>
<a target="_blank" 
href="../cgi-bin/hgTrackUi?db=hg38&c=chr22&g=gnomadGenomesVariantsV3"><strong>gnomAD v3</strong></a>
contains variants from 71,702 whole genomes (and no exomes), all mapped to the GRCh38/hg38
reference sequence. For more information on gnomAD v3, see the following
<a target="_blank" href="https://macarthurlab.org/2019/10/16/gnomad-v3-0/">blog post</a>.</p>
<p>
<a target="_blank"  
href="../cgi-bin/hgTrackUi?db=hg38&c=chr22&g=gnomadVariantsV2"><strong>gnomAD v2</strong></a>
is a composite of gnomAD Genome and Exome Variants v2.1. These two tracks contain variants from
125,748 exomes and 15,708 whole genomes, all mapped to the GRCh37/hg19 reference sequence and
lifted to the GRCh38/hg38 assembly. For more information on gnomAD v2.1, see the following
<a target="_blank" href="https://macarthurlab.org/2018/10/17/gnomad-v2-1/">blog post</a>.</p> 
<p>
Additional information, display conventions, and data access can be found on the
<a target="_blank" 
href="../cgi-bin/hgTrackUi?db=hg38&c=chr22&g=gnomadVariants">track description page</a>.<p>
<p>
We would like to thank the <a target="_blank" 
href="https://gnomad.broadinstitute.org/about">Genome Aggregation Database Consortium</a>
for making these data available. We would also like to thank Kate Rosenbloom,
Angie Hinrichs, and Lou Nassar for their work on this release.</p>

<a name="102519a"></a>
<h2>Oct. 25, 2019 &nbsp;&nbsp; New Illumina Global Diversity Array for human (GRCh37/hg19)</h2>
<p>
We are pleased to announce the release of the
<a href="../cgi-bin/hgTrackUi?db=hg19&c=chrX&g=genotypeArrays" target="_blank">Illumina Global
Diversity Array</a> (GDA) for human (GRCh37/hg19). This track is included in the SNP/CNV composite
track under the Variation group.</p>
<p>
The
<a href="https://www.illumina.com/products/by-type/microarray-kits/infinium-global-diversity.html"
target="_blank">Global Diversity Array-8</a> v1.0 BeadChip includes coverage of the ACMG 59-gene
clinical research variants and multi-ethnic, genome-wide content. The GDA is the commercial version
of the array chosen by the <a href="https://allofus.nih.gov/" target="_blank">All of Us</a> Research
Program and is designed to capture a wider range of the world's populationsthan traditional
microarrays.</p>
<p>
We would like to thank Jay Kaufman and John Picuri at Illumina for providing the data. We would also
like to thank Brian Raney and Jairo Navarro at UCSC for creating and reviewing the tracks for hg19.
</p>

<a name="102519b"></a>
<h2>Oct. 25, 2019 &nbsp;&nbsp; New Hi-C heatmap tracks</h2>
<p>
We are pleased to announce two new Hi-C heatmap tracks now available on the Genome Browser for hg19
and hg38. They can be configured as square, triangular, or arc interactions. These heatmaps will 
soon be available to visualize custom data for Chromosome Conformation Capture (3C), Hi-C, and other 
high-density interaction based sequencing methods. This feature will be the next evolution of our
support for 3C sequencing data, following the popular <a href=#081018>Interact format 
used in GeneHancer</a>, which was made for low-density interactions. 
<ul>
<li>
<a href="../cgi-bin/hgTracks?db=hg19&hideTracks=1&knownGene=pack&rao2014Hic=full" target="_blank">Hi-C Heatmap Track on hg19</a></li>
<li>
<a href="../cgi-bin/hgTracks?db=hg38&hideTracks=1&knownGene=pack&hicAndMicroC=full" target="_blank">Hi-C Heatmap Track on hg38</a></li></ul>

<p class="text-center">
  <img class='text-center' src="../images/hicExample.png" width='600' alt="Example of a Hi-C Track">
</p>

<a name="102419"></a>
<h2>Oct. 24, 2019 &nbsp;&nbsp; New Platinum Genomes VCF Track</h2>
<p>
We are pleased to announce the new Platinum Genomes VCF Tracks. These variant tracks 
offer deeply sequenced and validated VCFs on both hg19 and hg38. These data were derived from
consensus analysis of a 17-member family pedigree to reduce potential error.
You can view the tracks below:</p>
<ul>
<li>
<a href="../cgi-bin/hgTracks?db=hg19&hideTracks=1&knownGene=pack&platinumGenomes=pack" target="_blank">Platinum Genomes Track on hg19</a></li>
<li>
<a href="../cgi-bin/hgTracks?db=hg38&hideTracks=1&knownGene=pack&platinumGenomes=pack" target="_blank">Platinum Genomes Track on hg38</a></li></ul> 


<a name="101619"></a>
<h2>Oct. 16, 2019 &nbsp;&nbsp; New GENCODE Genes tracks for human (V32) and mouse (VM23)</h2>
<p>
We are pleased to announce the release of updated GENCODE Genes annotation tracks for both the
<a href="../cgi-bin/hgTracks?db=hg38&g=knownGene" target="_blank">hg38/GRCh38</a> and 
<a href="../cgi-bin/hgTracks?db=mm10&g=knownGene" target="_blank">mm10/GRCm38</a> assemblies
corresponding to the
<a href="http://www.ensembl.info/2019/09/26/ensembl-98-has-been-released/" target="_blank">
Ensembl 98</a> release.</p>

<p>
The following table provides statistics for the 
<a href="../cgi-bin/hgTrackUi?db=hg38&g=knownGene" target="_blank">human V32</a> and 
<a href="../cgi-bin/hgTrackUi?db=mm10&g=knownGene" target="_blank">mouse VM23</a> releases derived
from the GTF files that contain annotations only on the main chromosomes. More information on how
these statistics were generated can be found on the
<a href="https://www.gencodegenes.org/" target="_blank">GENCODE site</a>.

<div align="center">
<table class="stdTbl">
 <tr><th>Genes &amp; Transcripts</th><th>human V32 Release Stats</th><th>mouse V23 Release Stats</th></tr>
 <tr><th>Protein-coding Genes</th><td style="text-align:center">19,965</td><td style="text-align:center">21,849</td></tr>
 <tr><th>Protein-coding transcripts</th><td style="text-align:center">83,986</td><td style="text-align:center">59,188</td></tr>
 <tr><th>Long non-coding RNA genes</th><td style="text-align:center">17,910</td><td style="text-align:center">13,201</td></tr>
 <tr><th>Small non-coding RNA genes</th><td style="text-align:center">7,576</td><td style="text-align:center">6,108</td></tr>
 <tr><th>Psuedogenes</th><td style="text-align:center">14,749</td><td style="text-align:center">13,681</td></tr>
</table></div>

<p>
More details about these new GENCODE Genes tracks can be found on the respective GENCODE description pages.
We would like to thank the GENCODE team and thank Brian Raney, Mark Diekhans, Jairo Navarro, and
Conner Powell at UCSC for their work on this track.</p>

<a name="101119"></a>
<h2>Oct. 11, 2019 &nbsp;&nbsp; New "group auto-scale" option</h2>
<p>
A new &quot;group auto-scale&quot; option is now available for signal tracks
collected together in composites. The original auto-scale setting, which is still
available, acts to auto-scale each track individually inside a composite group.
With the new &quot;group auto-scale&quot; setting, all of the tracks within the
composite will scale to the track with the highest auto-scale value viewed in the region.</p>
<p>
For example, below is a side-by-side image of two views of the same data from
a selection of cell lines within a composite of related RNA-seq experiments. On
the left is the original &quot;auto-scale to data view&quot; setting, where each track is
auto-scaled to appear at each track's highest value. And on the right is
the new &quot;group auto-scale&quot; setting for the same RNA-seq data where all tracks
are scaled against the one track in the region that has the highest value
(67215 for IMR9 cell TAP + 1).</p>
<p class="text-center">
  <img class='text-center' src="../images/groupAutoScale.png" width='600' alt="Auto-scale comparison">
</p>
<p>
While the original auto-scale setting on the left has the advantage of showing any
available data calibrated to each track's maximum, the new &quot;group auto-scale&quot;
setting on the right quickly draws attention to the track with the highest signal
within the composite group.</p>
<p>
The new setting can be found for signal tracks within a composite group
under the &quot;Data view scaling:&quot; options.
<p class="text-center">
  <img class='text-center' src="../images/groupAutoSetting.png" width='400' alt="Scaling options">
</p>

<a name="100919"></a>
<h2>Oct. 9, 2019 &nbsp;&nbsp; Expanded CRISPR track released for human (hg38/GRCh38)</h2>
<p>
We are pleased to announce the release of an expanded
 <a href="../cgi-bin/hgTracks?db=hg38&hideTracks=1&knownGene=pack&crisprAllTargets=pack&position=chr7:116397500-116398000" target="_blank">CRISPR Targets track</a> for the human
hg38/GRCh38 assembly. This track shows the DNA sequences targetable by CRISPR RNA guides
using the Cas9 enzyme from <i>S. pyogenes</i> (PAM: NGG) over the entire human genome. CRISPR target
sites were annotated with predicted specificity (off-target effects) and predicted efficiency
(on-target cleavage) by various algorithms through the tool
<a href="http://crispor.tefor.net/" target="_blank">CRISPOR</a>. The target sequence of the guide
is shown with a thick (exon) bar. The PAM motif match (NGG) is shown with a thinner bar. 
</p>
<p class="text-center">
  <img class='text-center' src="../images/crisperAllTargetsHg38.png" width='800' alt="CrisprAllTargets track">
</p>

<p>
We would like to thank Maximilian Haeussler, Hiram Clawson, and Conner Powell for their effort
creating and releasing this data track.</p>

<a name="091719"></a>
<h2>Sep. 17, 2019 &nbsp;&nbsp; New GRC Incident track for Chicken (galGal6) and Zebrafish (danRer11)
</h2>
<p>
We are pleased to announce the release of a GRC Incident track for both the Chicken
(<a href="../cgi-bin/hgTrackUi?&db=galGal6&g=grcIncidentDb" target="_blank">galGal6</a>) and
Zebrafish (<a href="../cgi-bin/hgTrackUi?&db=danRer11&g=grcIncidentDb" target="_blank">danRer11</a>)
assemblies.
These tracks show locations in the Chicken and Zebrafish assemblies where assembly problems
have been noted or resolved, as reported by the
<a href="https://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/index.shtml" target="_blank">
Genome Reference Consortium (GRC)</a>.
<ul>
  <li> <a href="../cgi-bin/hgTracks?db=galGal6&hideTracks=1&grcIncidentDb=pack&refSeqComposite=squish&knownGene_sel=1&position=chr1%3A81414-19700000"
  target="_blank">Chicken (galGal6) GRC Incident Database track</a></li>
  <li> <a href="../cgi-bin/hgTracks?db=danRer11&hideTracks=1&grcIncidentDb=pack&refSeqComposite=squish&knownGene_sel=1"
  target="_blank">Zebrafish (danRer11) GRC Incident Database track</a></li>
</ul>
</p>
<p>
Data for these tracks are extracted from the <a href="ftp://ftp.ncbi.nlm.nih.gov/pub/grc/"
target="_blank">GRC incident database</a> from the specific species *_issues.gff3 file. The tracks are
updated daily from the GRC server to incorporate new updates.
</p>
<p>
Thanks to Hiram Clawson and Conner Powell for the development and release of these tracks.
</p>

<a name="090419"></a>
<h2>Sep. 4, 2019 &nbsp;&nbsp; New Genome Browser: Rhesus, rheMac10 (<em>Macaca mulatta</em>)</h2>
<p>
A genome browser is now available for the rhesus monkey (<em>Macaca mulatta</em>) assembly 
released in Feb. 2019 by the
<a href="https://www.genome.wustl.edu/" 
target="_blank">Genome Institute at Washington University School of Medicine</a> (Mmul_10).
<h3>About the assembly:</h3>
<ul>
  <li><b>UCSC Genome Browser assembly ID:</b>  <a href="../../cgi-bin/hgGateway?db=rheMac10" 
  target="_blank">rheMac10</a></li>
  <li><b>Sequencing/Assembly provider ID:</b> <a href="https://www.genome.wustl.edu/">
The Genome Institute at Washington University School of Medicine</a></li>
  <li><b>Assembly date:</b> Feb. 2019</li>
  <li><B>Accession ID:</B> GCF_003339765.1</li>
  <li><B>NCBI Genome ID:</B> <A HREF="https://www.ncbi.nlm.nih.gov/genome/215"
TARGET="_blank">215</A> (<em>Macaca mulatta</em>)</li>
  <li><B>NCBI Assembly ID:</B> <A HREF="https://www.ncbi.nlm.nih.gov/assembly/2345051"
TARGET="_blank">2345051</A></li>
  <li><B>NCBI BioProject ID:</B> <A HREF="https://www.ncbi.nlm.nih.gov/bioproject/476474"
TARGET="_blank">476474</A></li>
  <li><B>NCBI BioSample ID:</B> <A HREF="https://www.ncbi.nlm.nih.gov/biosample/SAMN09435472"
TARGET="_blank">SAMN09435472</A></li>
  <li><b>Scaffold count: </b>2,979</li>
  <li><b>Total size: </b>2,971,331,530</li>
  <li><a href="../goldenPath/credits.html#rhesus_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
  <li><a href="../goldenPath/credits.html#rhesus_use" 
target="_blank" >Data use conditions and restrictions</a></li>
  <li><a href="../goldenPath/help/ftp.html" 
target="_blank" >Rsync</a> and
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/rheMac10/" 
target="_blank" >FTP</a></li>
  <li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#rhesus"
target="_blank" >Downloads page</a></li>
</ul>
</p>

<a name="082819"></a>
<h2>Aug. 28, 2019 &nbsp;&nbsp; Vertebrate Genomes Project assembly hub</h2>

<p>
The UCSC Genome Browser is proud to announce its <a href="https://hgdownload.gi.ucsc.edu/hubs/VGP/" 
target="_blank">Vertebrate Genomes Project assembly hub</a>. The <a 
href="https://vertebrategenomesproject.org/" target="_blank">Vertebrate Genomes Project (VGP)</a> 
aims to generate near error-free reference genome assemblies of all 70,000 extant 
vertebrate species.</p>

<p>
As of now, the <a href="https://hgdownload.gi.ucsc.edu/hubs/VGP/" 
target="_blank">VGP assembly hub</a> includes 24 vertebrate assemblies. See our <a 
href="https://hgdownload.gi.ucsc.edu/hubs/VGP/asmStatsVGP.html" 
target="_blank">assembly statistics page</a> for specific information (gap counts, 
masking percent, genome size, etc.) on each individual assembly. All assemblies have a 
combination of Genome Browser generated data tracks, as well as external data. These include:</p>
<ul>
  <li>Augustus gene predictions on all assemblies, as well as imported NCBI gene predictions 
when available.</li>
  <li>A RefSeq mRNAs track which is generated in-house, mapping RefSeq mRNAs from various other 
organisms to each individual assembly. This track includes origin mRNA species and gene names, 
which may help to identify genes on the less annotated assemblies.</li>
  <li>CpG island and GC percent tracks.</li>
  <li>Repeat masking tracks by both RepeatMasker and WindowMasker + SDust.</li></ul>
<p>
These tracks also include individual track description pages with a summary of their contents 
or results where applicable. Such as a summary of all RepeatMasker elements found in the 
<a href="https://genome.ucsc.edu/cgi-bin/hgTrackUi?hubUrl=https://hgdownload.gi.ucsc.edu/hubs/VGP/hub.txt&genome=hub_1489433_GCA_007474595.1_mLynCan4_v1.p&g=hub_1489433_repeatMasker" 
target="_blank">Canada lynx</a> or gene count and genome coverage of NCBI genes for the 
<a href="https://genome.ucsc.edu/cgi-bin/hgTrackUi?hubUrl=https://hgdownload.gi.ucsc.edu/hubs/VGP/hub.txt&genome=hub_1489433_GCF_004115215.1_mOrnAna1.p.v1&g=hub_1489433_ncbiGene" 
target="_blank">platypus</a>.</p>

<p>
All data are available from the UCSC download servers. For information on how to download 
specific assembly data, refer to the assembly gateway page once the hub is connected. Here 
is an example of the gateway page for the 
<a href="../../cgi-bin/hgGateway?hubUrl=https://hgdownload.gi.ucsc.edu/hubs/VGP/hub.txt&genome=GCA_004115265.2_mRhiFer1_v1.p" 
target="_blank">greater horseshoe bat</a>. This page also includes information on starting 
a local BLAT server for the assembly.</p>

<p>
We would like to thank the <a href="https://vertebrategenomesproject.org/" 
target="_blank">Vertebrate Genomes Project</a> and the <a href="https://genome10k.soe.ucsc.edu/" 
target="_blank">Genome 10k Project</a> for their sequencing efforts and making these genomes 
available. We would also like to thank Hiram Clawson and Luis Nassar from the Genome Browser team 
for the development and release of this assembly hub.</p>

<a name="073119"></a>
<h2>Jul. 31, 2019 &nbsp;&nbsp; New Genome Browser: Horse, equCab3 (<em>Equus caballus</em>)</h2>
<p>
A genome browser is now available for the <em>Equus caballus</em> assembly released in Jan. 2018 by the
<a href="https://louisville.edu/medicine/" target="_blank"> University of Louisville</a> (EquCab3.0).
<h3>About the assembly:</h3>
<ul>
  <li><b>UCSC Genome Browser assembly ID:</b>  <a href="../../cgi-bin/hgGateway?db=equCab3" 
  target="_blank">equCab3</a></li>
  <li><b>Sequencing/Assembly provider ID:</b> <a href="https://louisville.edu/medicine/">
University of Louisville</a></li>
  <li><b>Assembly date:</b> Jan. 2018</li>
  <li><b>Accession ID:</b> GCF_002863925.1</li>
  <li><b>NCBI Genome ID:</b> <a href="https://www.ncbi.nlm.nih.gov/genome/145"
target="_blank">145</a> (<em>Equus caballus</em>)</li>
  <li><b>NCBI Assembly ID:</b> <a href="https://www.ncbi.nlm.nih.gov/assembly/1512011"
target="_blank">1512011</a></li>
  <li><b>NCBI BioProject ID:</b> <a href="https://www.ncbi.nlm.nih.gov/bioproject/421018"
target="_blank">421018</a></li>
  <li><b>NCBI BioSample ID:</b> <a href="https://www.ncbi.nlm.nih.gov/biosample/SAMN02953672"
target="_blank">SAMN02953672</a></li>
  <li><b>Scaffold count: </b>4,701</li>
  <li><b>Total size: </b>2,506,966,135</li>
  <li><a href="../goldenPath/credits.html#horse_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
  <li><a href="../goldenPath/credits.html#horse_use" 
target="_blank" >Data use conditions and restrictions</a></li>
  <li><a href="../goldenPath/help/ftp.html" 
target="_blank" >Rsync</a> and
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/equCab3/" 
target="_blank" >FTP</a></li>
  <li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#horse"
target="_blank" >Downloads page</a></li>
</ul>
</p>

<a name="071919"></a>
<h2>Jul. 19, 2019 &nbsp;&nbsp; New GENCODE V31 gene tracks for human, hg19 and hg38</h2>
<p>
We are pleased to announce the release of the new GENCODE Gene annotation tracks, which corresponds
to the <a href="http://www.ensembl.info/2019/05/20/whats-coming-in-ensembl-97-ensembl-genomes-44/"
target="_blank">Ensembl 97 release</a> for human assemblies, <a target="_blank"
href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV31lift37">hg19/GRCh37</a> and
<a target="_blank" href="../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeGencodeV31">hg38/GRCh38</a>.
All GENCODE V31 annotations are available for hg38/GRCh38, and the annotation release was
back-mapped to the hg19/GRCh37 assembly. For both hg19 and hg38, the GENCODE v31 gene set contains
the following tracks:</p>
<ul>
  <li>Basic - a subset of the <em>Comprehensive set</em>.</li>
  <li>Comprehensive - all GENCODE coding and non-coding transcript annotations, including
      polymorphic pseudogenes. This includes both manual and automatic annotations.</li>
  <li>Pseudogenes - all annotations except polymorphic pseudogenes.</li></ul>
<p>
The hg38 assembly also includes the following tracks that are not available on hg19:
</p>
<ul>
  <li>2-way Pseudogenes - pseudogenes predicted by both the Yale Pseudopipe and UCSC
      Retrofinder pipelines.</li>
  <li>PolyA - polyA signals and sites manually annotated on the genome based on transcribed
      evidence (ESTs and cDNAs) of 3' end of transcripts containing at least 3 A's not matching the
      genome.</li></ul>
<p>Thanks to Mark Diekhans and Jairo Navarro for the development and release of these tracks.</p>

<a name="071719"></a>
<h2>Jul. 17, 2019 &nbsp;&nbsp; New GENCODE gene set tracks available for mouse (VM22)</h2>
<p>We are happy to announce the release of the most up-to-date GENCODE M22 gene set for the
mouse genome (mm10/GRCm38).</p>
<p>
The GENCODE Genes track (version M22, June 2019) shows high-quality manual annotations merged with
evidence-based automated annotations across the entire mouse genome.
This track includes 5 subtracks: a basic gene set, a comprehensive
gene set, a pseudogene set, a high confidence pseudogene set, and a set of polyA site 
locations. You can read more about the VM22 by navigating there from the 
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=wgEncodeGencodeSuper">All GENCODE supertrack
</a> or by going directly to the 
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=wgEncodeGencodeVM22">track description page</a>.

<p>Thanks to Mark Diekhans and Daniel Schmelter for their work developing and releasing this track.
</p>

<a name="071519"></a>
<h2>Jul. 15, 2019 &nbsp;&nbsp; Updated TransMap tracks (V5)</h2>
<p>
We are pleased to announce the release of a new set of TransMap (V5) tracks for 94 different 
assemblies.</p>

<p> 
The TransMap tracks are an attempt to map features, such as mRNAs, ESTs, and gene annotations
from one organism to another through pairwise, whole-genome alignments. These new TransMap tracks
are:</p>
<ul>
  <li>Ensembl Mappings</li>
  <li>RefSeq Gene Mappings</li>
  <li>GenBank RNA Mappings</li>
  <li>EST Mappings</li>
</ul>

<p>
You can read more about how these tracks were created on the
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=transMapV5" target="_blank">track description page</a> 
for the hg38/GRCh38 version of these tracks. The TransMapV4 data can still be found in our 
<a href="http://hgdownload.gi.ucsc.edu/gbdb/" target="_blank">download server</a>.</p>

<p>
Thanks to Mark Diekhans and Lou Nassar for their work in creating and releasing these tracks.</p>

<a name="070919"></a>
<h2>Jul. 9, 2019 &nbsp;&nbsp; Expanded CRISPR track released for human (hg19/GRCh37)</h2> 
<p>
We are pleased to announce the release of an expanded
 <a href="../cgi-bin/hgTracks?db=hg19&hideTracks=1&knownGene=pack&crisprAllTargets=pack&position=chr7:116397500-116398000" target="_blank">CRISPR Targets track</a> for the human 
hg19/GRCh37 assembly. This track shows the DNA sequences targetable by CRISPR RNA guides 
using the Cas9 enzyme from <i>S. pyogenes</i> (PAM: NGG) over the entire human genome. CRISPR target
sites were annotated with predicted specificity (off-target effects) and predicted efficiency
(on-target cleavage) by various algorithms through the tool 
<a href="http://crispor.tefor.net/" target="_blank">CRISPOR</a>. 
</p>
<p class="text-center">
  <img class='text-center' src="../images/crisprAll.png" width='850' alt="CrisprAllTargets track">
</p>
<p>
We would like to thank Maximilian Haeussler, Hiram Clawson, and Daniel Schmelter for their effort
creating, developing, and releasing this data track.</p> 

<a name="070119"></a>
<h2>Jul. 1, 2019 &nbsp;&nbsp; New DGV Gold Standard track for human (hg19/GRCh37)</h2>
<p>
We are pleased to announce the DGV Gold Standard track for human (hg19/GRCh37). This track displays
copy number variants (CNVs), insertions/deletions (InDels), inversions and inversion breakpoints
annotated by the <a href="http://dgv.tcag.ca/dgv/app/home" target="_blank">Database of Genomic Variants</a>
(DGV). The initial criteria for the Gold Standard set require that a variant be found in at least
two different studies and found in at least two different samples. After filtering out low-quality
variants, the remaining variants are clustered according to 50% minimum overlap, and then merged
into a single record. Gains and losses are merged separately.
</p>

<p>
The <a href="../cgi-bin/hgTrackUi?db=hg19&g=dgvPlus" target="_blank">DGV Gold Standard track</a>
utilizes a boxplot-like display to represent the merging of records. The highest ranking variant in
the cluster defines the inner &quot;box&quot;, while the outer lines define the outermost start and stop
coordinates of the CNV. In this way, the inner box forms a high-confidence CNV location and the
thin connecting lines indicate confidence intervals for the location of the CNV.
</p>
<p class="text-center">
  <img class='text-center' src="../images/dgvGold.png" width='850' alt="DGV Gold Standard track">
</p>

<p>
We would like to thank Jeff MacDonald and the <a href="http://dgv.tcag.ca/dgv/app/home" target="_blank">Database of Genomic Variants</a>
group for providing guidance and these data. We would also like to thank Christopher Lee and Conner
Powell as well as the entire UCSC Genome Browser team for creating and releasing this track.
</p>

<a name="062819"></a>
<h2>Jun. 28, 2019 &nbsp;&nbsp; New Genome Browsers: Cow, bosTau9 (<em>Bos taurus</em>) and 
Baboon, papAnu4 (<em>Papio anubis</em>)</h2>
<p>
Genome browsers are now available for the <em>Bos taurus</em> assembly released in Apr. 2018 by the
<a href="https://www.ars.usda.gov/" target="_blank"> USDA Agricultural Research Service</a> (ARS-UCD1.2), 
and the <em>Papio anubis</em> assembly released in Apr. 2017 by the
<a href="https://www.hgsc.bcm.edu/" target="_blank"> Human Genome Sequencing Center</a> (Panu_3.0).
<h3>About the assembly (bosTau9):</h3>
<ul>
  <li><b>UCSC Genome Browser assembly ID:</b>  <a href="../../cgi-bin/hgGateway?db=bosTau9" 
  target="_blank">bosTau9</a></li>
  <li><b>Sequencing/Assembly provider ID:</b> <a href="https://www.ars.usda.gov/">
USDA Agricultural Research Service</a></li>
  <li><b>Assembly date:</b> Apr. 2018</li>
  <li><B>Accession ID:</B> GCF_002263795.1</li>
  <li><B>NCBI Genome ID:</B> <A HREF="https://www.ncbi.nlm.nih.gov/genome/82"
TARGET="_blank">82</A> (<em>Bos taurus</em>)</li>
  <li><B>NCBI Assembly ID:</B> <A HREF="https://www.ncbi.nlm.nih.gov/assembly/1677391"
TARGET="_blank">1677391</A></li>
  <li><B>NCBI BioProject ID:</B> <A HREF="https://www.ncbi.nlm.nih.gov/bioproject/391427"
TARGET="_blank">391427</A></li>
  <li><B>NCBI BioSample ID:</B> <A HREF="https://www.ncbi.nlm.nih.gov/biosample/SAMN03145444"
TARGET="_blank">SAMN03145444</A></li>
  <li><b>Scaffold count: </b>2,211</li>
  <li><b>Total size: </b>2,358,151,106</li>
  <li><a href="../goldenPath/credits.html#Cow_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>About the assembly (papAnu4):</h3>
<ul>
  <li><b>UCSC Genome Browser assembly ID:</b>  <a href="../../cgi-bin/hgGateway?db=papAnu4" 
  target="_blank">papAnu4</a></li>
  <li><b>Sequencing/Assembly provider ID:</b> <a href="https://www.hgsc.bcm.edu/">
Human Genome Sequencing Center</a></li>
  <li><b>Assembly date:</b> Apr. 2017</li>
  <li><B>Accession ID:</B> GCF_000264685.3</li>
  <li><B>NCBI Genome ID:</B> <A HREF="https://www.ncbi.nlm.nih.gov/genome/394"
TARGET="_blank">394</A> (<em>Papio anubis</em>)</li>
  <li><B>NCBI Assembly ID:</B> <A HREF="https://www.ncbi.nlm.nih.gov/assembly/1082401"
TARGET="_blank">1082401</A></li>
  <li><B>NCBI BioProject ID:</B> <A HREF="https://www.ncbi.nlm.nih.gov/bioproject/54005"
TARGET="_blank">54005</A></li>
  <li><B>NCBI BioSample ID:</B> <A HREF="https://www.ncbi.nlm.nih.gov/biosample/SAMN02981400"
TARGET="_blank">SAMN02981400</A></li>
  <li><b>Scaffold count: </b>71,135</li>
  <li><b>Total size: </b>2,959,373,024</li>
  <li><a href="../goldenPath/credits.html#Baboon_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
  <li><a href="../goldenPath/credits.html" 
target="_blank" >Data use conditions and restrictions</a></li>
  <li><a href="../goldenPath/help/ftp.html" 
target="_blank" >Rsync</a></li>
  <li>FTP: <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/bosTau9/" 
target="_blank" >Cow</a> and 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/papAnu4/" 
target="_blank" >Baboon</a></li>
  <li>Downloads page: <a href="http://hgdownload.gi.ucsc.edu/downloads.html#cow"
target="_blank" >Cow</a> and 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#baboon"
target="_blank" >Baboon</li>
</ul>
</p>

<a name="062519"></a>
<h2> Jun. 25, 2019 &nbsp;&nbsp; JSON API now available</h2>
<p>
The UCSC Genome Browser is pleased to announce its new <a href="help/api.html" 
target="_blank">JavaScript Object Notation (JSON) Application Programming 
Interface (API)</a>. The API provides direct access to different 
data including annotations and sequence data for both native Genome Browser assemblies as 
well as hubs.</p>

<p>
Some functions possible with the API include:
<ul>
  <li>List of all available UCSC Genome Browser assemblies</li>
  <li>List of chromosomes contained in a UCSC Genome Browser assembly or assembly hub</li>
  <li>List of all data tracks in a specific assembly</li>
  <li>Extraction of any track data from a UCSC Genome Browser assembly or hub</li>
</ul></p>

<p>
See our <a href="help/api.html#Return" target="_blank">help page</a> for 
a complete list of what data can be accessed, as well as as 
<a href="help/api.html#list_examples" 
target="_blank">examples of how to access the data</a>.</p>

<p>
The UCSC Genome Browser would like to thank Hiram Clawson and Lou Nassar for the development and 
testing of this new feature.</p>

<a name="062019"></a>
<h2> Jun. 20, 2019 &nbsp;&nbsp; New GDC Cancer variants track and lollipop display for human
(GRCh38/hg38)</h2>
<p>
We are pleased to announce the
<a href="../cgi-bin/hgTrackUi?db=hg38&position=chr1%3A11102837-11267747&g=gdcCancer" target="_blank">
GDC Cancer mutations</a> track for human (<em>GRCh38/hg38</em>). This
track shows the genomic positions of somatic variants found through whole genome sequencing of
tumors as part of
<a href="https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga"
target="_blank">The Cancer Genome Atlas (TCGA)</a> by the National Cancer Institute, made available
through the <a href="https://portal.gdc.cancer.gov/" taret="_blank">Genomic Data Commons Portal</a>.
The data shown is sometimes called the &quot;Pan-Cancer dataset&quot;, a collection of
thirty-three TCGA projects processed in a uniform way.
</p>
<p>
With the release of the GDC Cancer track, we are introducing a new lollipop display mode, available
when this track is in 'full' or 'squish' mode. These variants can be shown in two display
modes:
<ul>
  <li>Lollipop Mode: variants are shown with circles, the y-axis represents the number of samples
    within the cohort. (<em>Available in 'full' or 'squish' modes</em>.)</li>
  <li>BED Mode: variants are shown as normal genomic features.
    (<em>Available in 'pack' or 'dense' modes</em>.)</li>
</ul></p>
<p class="text-center">
  <img class='text-center' src="../images/gdcCancerLollipop.jpg" width='699' alt="GDC Cancer lollipop display">
</p>
<p>
We would like to thank the <a href="https://portal.gdc.cancer.gov/" target="_blank">Genomic Data
Commons Portal</a> for making the TCGA data available on their web site. We would also like to thank
Brian Raney and Jairo Navarro at the UCSC Genome Browser for creating and testing this track and
the new lollipop display mode.</p>


<a name="061719"></a>

<h2> Jun. 17, 2019 &nbsp;&nbsp; New ENCODE 3 transcription factor binding site track sets for human (<em>GRCh37/hg19, GRCh38/hg38</em>)</h2>
<p>
We are pleased to announce two new ENCODE 3 transcription factor binding site track sets for human 
(GRCh37/hg19, GRCh38/hg38). These track sets show regions of transcription factor binding 
derived from a large collection of ChIP-seq experiments performed by the ENCODE project between 
February 2011 and November 2018.
</p>
<p>
The ENCODE TF ChIP-seq data were processed using the
<a href="https://www.encodeproject.org/chip-seq/transcription_factor/" target="_blank"
>ENCODE Transcription Factor ChIP-seq Processing Pipeline</a> to generate peaks of TF binding.
Peaks from 1264 experiments (1256 in hg38) representing 338 transcription factors (340 in hg38) in
130 cell types (129 in hg38) were used to generate these two track sets.<br> 
<ul>
  <li>The <b>ENCODE 3 TF ChIP Clusters tracks</b> are collections of ChIP-seq peaks consolidated into a
   summary display or &lsquo;cluster&rsquo; showing occupancy regions for each factor. The cluster
   labels now include the count of cell types contributing to the cluster (those with ChIP-seq peaks
   in the region) and the total count of cell types tested for the factor. The new label feature
   can be disabled via a user option on the track configuration page. You can read more about the
   TF ChIP Clusters tracks here:
   <a href="../cgi-bin/hgTrackUi?db=hg19&g=encRegTfbsClustered" target="_blank">hg19</a>, 
   <a href="../cgi-bin/hgTrackUi?db=hg38&g=encRegTfbsClustered" target="_blank">hg38</a>.</li>
  <li>The <b>ENCODE 3 TF ChIP Peaks track sets</b> are collections of the underlying ChIP-seq peak tracks
      which can be viewed individually or collectively by cell type or transcription factor. You can read more about the TF ChIP Peaks track sets here:
      <a href="../cgi-bin/hgTrackUi?db=hg19&g=encTfChipPk" target="_blank">hg19</a>,                                    
      <a href="../cgi-bin/hgTrackUi?db=hg38&g=encTfChipPk" target="_blank">hg38</a>.</li>
</ul>
</p>
<p>
An example of these two track sets is shown in the following sessions where CTCF binds to the MYC Proto-Oncogene
across a few selected cell types. CTCF is a transcription factor identified as a repressor of MYC
shown to inhibit cell growth (<a href="https://www.pnas.org/content/100/2/633" 
target="_blank">Qi et al., 2002</a>).<br>
<ul>
 <li><a href="../cgi-bin/hgTracks?db=hg19&hideTracks=1&wgEncodeReg=show&wgEncodeReg_hideKids=1&encRegTfbsClustered=dense&encRegTfbsClustered=full&encRegTfbsClustered.filterBy.name=CTCF&wgEncodeTfBindingSuper=show&wgEncodeTfBindingSuper_hideKids=1&encTfChipPk=dense&encTfChipPk_hideKids=1&encTfChipPkENCFF003YLJ=dense&encTfChipPkENCFF003YLJ_sel=1&position=chr8%3A128748315-128753680&knownGene=full&encTfChipPkENCFF016EUX=dense&encTfChipPkENCFF016EUX_sel=1&encTfChipPkENCFF030BPR=dense&encTfChipPkENCFF030BPR_sel=1&encTfChipPkENCFF032DEW=dense&encTfChipPkENCFF032DEW_sel=1&encTfChipPkENCFF777TZZ=dense&encTfChipPkENCFF777TZZ_sel=1&pix=900"
target="_blank"
     >CTCF transcription factor binding to MYC on hg19</a></li>
 <li><a href="../cgi-bin/hgTracks?db=hg38&hideTracks=1&wgEncodeReg=show&wgEncodeReg_hideKids=1&encTfChipPk=pack&encTfChipPk_hideKids=1&encTfChipPkENCFF535MZG=dense&encTfChipPkENCFF535MZG_sel=1&encTfChipPkENCFF579XTC=dense&encTfChipPkENCFF579XTC_sel=1&encTfChipPkENCFF574FIL=dense&encTfChipPkENCFF574FIL_sel=1&encTfChipPkENCFF560GGY=dense&encTfChipPkENCFF560GGY_sel=1&encTfChipPkENCFF552XDP=dense&encTfChipPkENCFF552XDP_sel=1&encTfChipPkENCFF540DVR=dense&encTfChipPkENCFF540DVR_sel=1&encRegTfbsClustered_sel=1&encRegTfbsClustered=full&encRegTfbsClustered.filterBy.name=CTCF&position=chr8%3A127736594-127740958&knownGene=full&pix=900"
target="_blank"
     >CTCF transcription factor binding to MYC on hg38</a></li>
</ul>
</p>
<p>
Thanks to the <a href="https://www.encodeproject.org/" target="_blank">ENCODE Consortium</a>,
the ENCODE ChIP-seq production laboratories, and the ENCODE Data Coordination Center for generating
and processing the TF ChIP-seq datasets. The ENCODE accession numbers of the constituent
datasets are available on the peak details page as links in the Clusters tracks and in the track
title for the Peaks tracks. Special thanks to Henry Pratt, Jill Moore, Michael Purcaro, and Zhiping
Weng, PI, at the 
<a href="https://www.umassmed.edu/zlab/" target="_blank">ENCODE Data Analysis Center (ZLab at UMass Medical Center)</a>
for providing the peak datasets, metadata, and guidance developing these track sets. Thanks to Kate
Rosenbloom and Conner Powell of the UCSC Genome Browser group for their work in creating and
releasing these track sets.
</p>


<a name="060619"></a>
<h2>Jun. 6, 2019 &nbsp;&nbsp; New Genome Browser: Chinese Hamster Ovary (CHO) cell line, criGriChoV2 (<em>Cricetulus griseus</em>)</h2>
<p>
A genome browser is now available for the <em>Chinese Hamster Ovary cell line</em> assembly released in June 2017 by the
<a href="https://www.eaglegenomics.com/eagle-case-studies/horizon-case-study/" target="_blank">Eagle Genomics Ltd</a> (CHOK1S_HZDv1).
<h3>About the assembly:</h3>
<ul>
  <li><b>UCSC Genome Browser assembly ID:</b> <a href=../cgi-bin/hgTracks?db=criGriChoV2>criGriChoV2</a></li>
  <li><b>Sequencing/Assembly provider ID:</b> <a href="https://www.eaglegenomics.com/" target="_blank">
  Eagle Genomics Ltd</a></li>
  <li><b>Assembly date:</b> June 2017</li>
  <li><b>Accession ID:</b> GCA_900186095.1</li>
  <li><b>NCBI Genome information:</b><a href="https://www.ncbi.nlm.nih.gov/genome/2791" 
target="_blank" > 2791</a> <em>(Cricetulus griseus)</em></li>
  <li><b>NCBI Assembly information:</b><a href="https://www.ncbi.nlm.nih.gov/assembly/1422381" 
target="_blank" > 1422381</a></li>
  <li><b>NCBI BioProject information:</b><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJEB21211" 
target="_blank" > PRJEB21211</a></li>
  <li><b>NCBI BioSample information:</b><a href="https://www.ncbi.nlm.nih.gov/biosample/SAMEA104116709"
target="_blank" > SAMEA104116709</a></li>
  <li><b>Contig+scaffold count:</b> 71,598</li>
  <li><b>Total size:</b> 2,358,908,658</li>
  <li><a href="../goldenPath/credits.html#chinese_hamster_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
  <li><a href="../goldenPath/credits.html#chinese_hamster_use" 
target="_blank" >Data use conditions and restrictions</a></li>
  <li><a href="../goldenPath/help/ftp.html" 
target="_blank" >Rsync</a> and
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/criGriChoV2/" 
target="_blank" >FTP</a></li>
  <li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#chinese_hamster"
target="_blank" >Downloads page</a></li>
</ul>
</p>

<a name="052119"></a>
<h2>May 21, 2019 &nbsp;&nbsp; New Genome Browser: Cat, felCat9 (<em>Felis catus</em>)</h2>
<p>
A genome browser is now available for the <em>Felis catus</em> assembly released in November 2017 by the
<a href="https://www.genome.wustl.edu/" target="_blank">Genome Sequencing Center (GSC) at Washington University (WashU) School of Medicine</a> (Felis_catus_9.0).
<h3>About the assembly:</h3>
<ul>
  <li><b>UCSC Genome Browser assembly ID:</b> felCat9</li>
  <li><b>Sequencing/Assembly provider ID:</b> <a href="https://www.genome.wustl.edu/" target="_blank">
  Genome Sequencing Center (GSC) at Washington University (WashU) School of Medicine</a></li>
  <li><b>Assembly date:</b> November 2017</li>
  <li><b>Accession ID:</b> GCF_000181335.3</li>
  <li><b>NCBI Genome information:</b><a href="https://www.ncbi.nlm.nih.gov/genome/78" 
target="_blank" > 78</a> <em>(Felis catus)</em></li>
  <li><b>NCBI Assembly information:</b><a href="https://www.ncbi.nlm.nih.gov/assembly/1448961" 
target="_blank" > 1448961 </a></li>
  <li><b>NCBI BioProject information:</b><a href="https://www.ncbi.nlm.nih.gov/bioproject/16726" 
target="_blank" > 16726</a></li>
  <li><b>NCBI BioSample information:</b><a href="https://www.ncbi.nlm.nih.gov/biosample/SAMN02953640"
target="_blank" > SAMN02953640</a></li>
  <li><b>Contig+scaffold count:</b> 9434</li>
  <li><b>Total size:</b> 2,521,863,845</li>
  <li><a href="../goldenPath/credits.html#cat_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
  <li><a href="../goldenPath/credits.html#cat_use" 
target="_blank" >Data use conditions and restrictions</a></li>
  <li><a href="../goldenPath/help/ftp.html" 
target="_blank" >Rsync</a> and
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/felCat9/" 
target="_blank" >FTP</a></li>
  <li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#cat"
target="_blank" >Downloads page</a></li>
</ul>
</p>



<a name="041019"></a>
<h2>Apr. 10, 2019 &nbsp;&nbsp; GENCODE Genes VM20 track released for mouse (GRCm38/mm10)</h2>
<p>
We are  pleased to announce the release of the GENCODE Genes VM20 track for the Mouse assembly
(<a href="../cgi-bin/hgTracks?db=mm10&position=chr12%3A56694976-56714605&hideTracks=1&knownGene=full&ignoreCookie=1"
target="_blank">GRCm38/mm10</a>). 
In this GENCODE Genes update, we have made the ENSMUST* IDs the default IDs and have added
the GENCODE pseudogenes to the knownGene tables.
</p>
<p>
More details about the GENCODE Genes VM20 track can be found on the
<a href="../cgi-bin/hgTrackUi?db=mm10&g=knownGene" target="_blank">GENCODE Genes track description
page</a>.</p>
<p>
We would like to thank the <a href="https://www.gencodegenes.org/" target="_blank">GENCODE
Project</a> for providing the data. The creation of this track was made possible thanks to Brian
Raney, Mark Diekhans, and Conner Powell.</p>


<a name="032619"></a>
<h2>
Mar. 26, 2019 &nbsp;&nbsp; New Genomenon Mastermind Variants track for Human (hg19)
</h2>
<p>
We are pleased to announce the release of the Genomenon Mastermind Variants track for the 
<a href="../cgi-bin/hgTracks?db=hg19&position=chr21:33032078-33032099&hideTracks=1&varsInPubs=show&mastermind=pack&knownGene=pack&ignoreCookie=1"
target="_blank">hg19/GRCh37 assembly</a>. This track is
composed of variants found in full-text scientific publications collected by <a 
href="https://www.genomenon.com/mastermind" target="_blank">Genomenon Mastermind</a>. Mastermind
uses software that searches for disease-gene-variant associations in the scientific literature to
generate a commercial database.
</p>
<p>
Genomic locations of variants are labeled with the nucleotide change (C&gt;A) where one can hover
over the features to see the gene impacted, the specific amino acid change, and the related support
scores (SOD1:A2E - 3/3/3). By clicking an item a link titled &quot;Protein change and link to
details&quot; provides access to the respective Mastermind variant database page for further
information. More information about this track and its display can be found on the
<a href="../cgi-bin/hgTrackUi?db=hg19&g=mastermind" target="_blank">track description</a> page.
<p>
We would like to thank Mark Kiel, Steve Schwartz and Clayton Wheeler from
Genomenon for making this data available as well as Max Haeussler, Conner Powell and the rest of the
UCSC Genome Browser team.
</p>
<a name="031819"></a>
<h2>
Mar. 18, 2019 &nbsp;&nbsp; New feature: Backup Sessions and Custom Track Data 
</h2>
<p>
Genome Browser users can now download custom track data from the 
<a href="../cgi-bin/hgSession">Session Management page</a>. You can backup Custom Tracks from a 
particular session by clicking into a named session with uploaded custom tracks, returning to the
Session Management page, then clicking the <button>Submit</button> button next to "Save Custom 
Tracks". You will be directed to a review page where you will be able download your custom track 
data. Read more in the <a href="help/hgSessionHelp.html#Create">"How to backup text-based Custom 
Track data to a file"</a> section of the Session User Guide.</p>

<a name="030519"></a>
<h2>
Mar. 11, 2019 &nbsp;&nbsp; Addition of GRCh38 patch sequences to hg38
</h2>
<p>
We are pleased to announce the addition of GRCh38 patches to the GRCh38/hg38 database. <a 
href="https://www.ncbi.nlm.nih.gov/grc/human" target="_blank">Without a release date</a> for the 
next coordinate-changing assembly, incorporating these sequences is an important step towards 
providing the most up to date human assembly.</p>

<p>
This release includes new tracks visualizing patch sequences; the <a 
href="../cgi-bin/hgTrackUi?db=hg38&g=fixSeqLiftOverPsl" target="_blank">Fix Patches</a> track
represents error corrections, and the <a href="../cgi-bin/hgTrackUi?db=hg38&g=altSeqLiftOverPsl" 
target="_blank">Alt Haplotypes</a> track represents novel patches (alt loci).</p>
<p>
Additional information can be found on our <a href="https://genome-blog.gi.ucsc.edu/blog/patches/" 
target="_blank">Genome Browser blog</a> including a specific section detailing <a 
href="https://genome-blog.gi.ucsc.edu/blog/patches/#changes" target="_blank">what changes to expect</a>.</p>

<p class="text-center">
  <img class='text-center' src="../images/patches.jpg" alt="Patches increase">
  <p class="gbsCaption text-center">The number of human assembly patch sequences is quickly
        growing. This is primarily due to alternative haplotypes (_alt) sequences, though fix
        sequences (_fix) are also being introduced. The fix patches reset from GRCh37.p13 to
        GRCh38 as they were integrated into the assembly.</p>
<p>
We would like to thank Angie Hinrichs and Lou Nassar for their work on this release.</p>

<a name="030519"></a>
<h2>
Mar. 05, 2019 &nbsp;&nbsp; Single cell data from Tabula Muris available for mouse (mm10)
</h2>
<p>
We are pleased to announce the release of the
<a href="../cgi-bin/hgTrackUi?db=mm10&g=tabulaMuris" target="_blank">Tabula Muris</a> superTrack
for mouse, <a href="../cgi-bin/hgGateway?db=mm10" target="_blank">mm10/GRCm38</a>. These tracks
incorporate single cell FACS RNA-seq data where 44,779 cells were assigned to single cell
clusters, representing a cell or tissue type. The Tabula Muris superTrack contains the following
three composite tracks:</p>
<dl>
    <dt>Cell expression</dt>
        <dd>
        Bar charts displaying the median expression level per cell type on the genome. A
        click-through on the bar chart will display a Tukey boxplot of expression level quartiles
        with outliers, per cluster.</dd>
    <dt>Genome Coverage</dt>
        <dd>
        Bar graphs indicating the number of RNA-seq reads at each base pair.</dd>
    <dt>Splice Sites</dt>
        <dd>
        Thick rectangles display exons around a splice site, connected by a line representing the
        intron. These introns are annotated with the number of RNA-seq reads that support the
        gap, in the 'score' field.</dd>
</dl>
<p>
More information about these tracks and their display can be found on the
<a href="../cgi-bin/hgTrackUi?db=mm10&g=tabulaMuris" target="_blank">track description</a> page, and
more information about the <a href="https://www.czbiohub.org/tabula-muris-introduction/"
target="_blank">Tabula Muris dataset</a> can be found at the CZI Biohub. We would like to
acknowledge our CZI contacts that spent quite a bit of time to get the data to us: Angela Pisco,
James Webber, and Joshua Batson. Track creation was done by Max Haeussler and tested by Jairo
Navarro at the UCSC Genome Browser. WiggleTools, written by Daniel Zerbino, and IntronProspector,
written by Mark Diekhans, were used in the creation of these tracks. Track hubs were written to a
large extent by Brian Raney and colleages at the UCSC Genome Browser.</p>

<a name="022519"></a>
<h2>Feb. 25, 2019 &nbsp;&nbsp; New NCBI RefSeq Human Gene Mutation Database (HGMD) track for hg38
and hg19</h2>
<p>
We are pleased to announce the release of a new gene annotation track for the GRCh38/hg38 and 
GRCh37/hg19 assemblies based on data from The Human Gene Mutation Database (HGMD) and NCBI's 
Reference Sequence Database (RefSeq). The alignments and coordinates for these tracks are provided
by RefSeq.

The NCBI RefSeq HGMD tracks show only RefSeq Curated transcripts mentioned in the public version of
the Human Gene Mutation Database. More details about the tracks, their construction, and how to 
access their underlying data can be found on the track description pages by clicking into an item 
after loading the tracks with the following links:


</p>
<ul>
<li><a href = "../cgi-bin/hgTracks?db=hg38&hideTracks=1&refSeqComposite=full&ncbiRefSeqHgmd_sel=1&ncbiRefSeqCurated_sel=0&ncbiRefSeqHgmd=pack&hgmd=dense" 
target="_blank">GRCh38/hg38</a></li>
<li><a href = "../cgi-bin/hgTracks?db=hg19&hideTracks=1&refSeqComposite=full&ncbiRefSeqHgmd_sel=1&ncbiRefSeqCurated_sel=0&ncbiRefSeqHgmd=pack&hgmd=dense"
target="_blank">GRCh37/hg19</a></li>
</ul>
<p>
A huge thank you to the HGMD group at Cardiff University, Frank Schacherer and Rupert Yip from 
QIAGEN, Anna Benet-Pages and Andreas Laner from The Medizinisch Genetisches Zentrum (MGZ), Max 
Haeussler, Conner Powell, and many others from the UCSC Genome Browser staff for bringing this 
track to life.
</p>


<a name="021119"></a>
<h2>Feb. 11, 2019 &nbsp;&nbsp; New Ensembl gene tracks for 56 assemblies</h2>
<p>
The UCSC Genome Browser is proud to announce an update to the
<a href="http://www.ensembl.org/info/genome/genebuild/genome_annotation.html" target="_blank">
Ensembl gene annotation</a> tracks, from <strong>v92 to v95</strong>, for the
following assemblies:</p>
<div class="container">
  <div class="row">
  <div class="col-sm-4">
  <ul>
	<li>Alpaca <a href="../cgi-bin/hgTrackUi?db=vicPac1&g=ensGene"
	 target=_blank>(vicPac1)</a></li>
	<li>Armadillo <a href="../cgi-bin/hgTrackUi?db=dasNov3&g=ensGene"
	 target=_blank>(dasNov3)</a></li>
	<li>Atlantic cod <a href="../cgi-bin/hgTrackUi?db=gadMor1&g=ensGene"
	 target=_blank>(gadMor1)</a></li>
	<li>Bonobo <a href="../cgi-bin/hgTrackUi?db=panPan2&g=ensGene" 
	 target=_blank>(panPan2)</a></li>
	<li>Bushbaby <a href="../cgi-bin/hgTrackUi?db=otoGar3&g=ensGene" 
	 target=_blank>(otoGar3)</a></li>
	<li>C. elegans <a href="../cgi-bin/hgTrackUi?db=ce11&g=ensGene"
	 target=_blank>(ce11)</a></li>
	<li>C. intestinalis <a href="../cgi-bin/hgTrackUi?db=ci3&g=ensGene"
	 target=_blank>(ci3)</a></li>
	<li>Chicken <a href="../cgi-bin/hgTrackUi?db=galGal6&g=ensGene"
	 target=_blank>(galGal6)</a></li>
	<li>Chimp <a href="../cgi-bin/hgTrackUi?db=panTro5&g=ensGene"
	 target=_blank>(panTro5)</a></li>
	<li>Chinese hamster <a href="../cgi-bin/hgTrackUi?db=criGriChoV1&g=ensGene"
	 target=_blank>(criGriChoV1)</a></li>
	<li>Coelacanth <a href="../cgi-bin/hgTrackUi?db=latCha1&g=ensGene"
	 target=_blank>(latCha1)</a></li>
	<li>Crab-eating macaque <a href="../cgi-bin/hgTrackUi?db=macFas5&g=ensGene"
	 target=_blank>(macFas5)</a></li>
	<li>D. melanogaster <a href="../cgi-bin/hgTrackUi?db=dm6&g=ensGene"
	 target=_blank>(dm6)</a></li>
	<li>Dog <a href="../cgi-bin/hgTrackUi?db=canFam3&g=ensGene"
	 target=_blank>(canFam3)</a></li>
	<li>Elephant <a href="../cgi-bin/hgTrackUi?db=loxAfr3&g=ensGene"
	 target=_blank>(loxAfr3)</a></li>
	<li>Ferret <a href="../cgi-bin/hgTrackUi?db=musFur1&g=ensGene"
	 target=_blank>(musFur1)</a></li>
	<li>Fugu <a href="../cgi-bin/hgTrackUi?db=fr3&g=ensGene"
	 target=_blank>(fr3)</a></li>
	<li>Gibbon <a href="../cgi-bin/hgTrackUi?db=nomLeu3&g=ensGene"
	 target=_blank>(nomLeu3)</a></li>
	<li>Golden snub-nosed monkey <a href="../cgi-bin/hgTrackUi?db=rhiRox1&g=ensGene"
	 target=_blank>(rhiRox1)</a></li>
   </ul></div>
   <div class="col-sm-4">
   <ul>
	<li>Gorilla <a href="../cgi-bin/hgTrackUi?db=gorGor4&g=ensGene"
	 target=_blank>(gorGor4)</a></li>
	<li>Green monkey <a href="../cgi-bin/hgTrackUi?db=chlSab2&g=ensGene"
	 target=_blank>(chlSab2)</a></li>
	<li>Guinea pig <a href="../cgi-bin/hgTrackUi?db=cavPor3&g=ensGene"
	 target=_blank>(cavPor3)</a></li>
	<li>Hedgehog <a href="../cgi-bin/hgTrackUi?db=eriEur1&g=ensGene"
	 target=_blank>(eriEur1)</a></li>
	<li>Lamprey <a href="../cgi-bin/hgTrackUi?db=petMar2&g=ensGene"
	 target=_blank>(petMar2)</a></li>
	<li>Lizard <a href="../cgi-bin/hgTrackUi?db=anoCar2&g=ensGene"
	 target=_blank>(anoCar2)</a></li>
	<li>Megabat <a href="../cgi-bin/hgTrackUi?db=pteVam1&g=ensGene"
	 target=_blank>(pteVam1)</a></li>
	<li>Microbat <a href="../cgi-bin/hgTrackUi?db=myoLuc2&g=ensGene"
	 target=_blank>(myoLuc2)</a></li>
	<li>Naked mole-rat <a href="../cgi-bin/hgTrackUi?db=hetGla1&g=ensGene"
	 target=_blank>(hetGla1)</a></li>
	<li>Naked mole-rat <a href="../cgi-bin/hgTrackUi?db=hetGla2&g=ensGene"
	 target=_blank>(hetGla2)</a></li>
	<li>Opossum <a href="../cgi-bin/hgTrackUi?db=monDom5&g=ensGene"
	 target=_blank>(monDom5)</a></li>
	<li>Orangutan <a href="../cgi-bin/hgTrackUi?db=ponAbe2&g=ensGene"
	 target=_blank>(ponAbe2)</a></li>
	<li>Panda <a href="../cgi-bin/hgTrackUi?db=ailMel1&g=ensGene"
	 target=_blank>(ailMel1)</a></li>
	<li>Pig <a href="../cgi-bin/hgTrackUi?db=susScr11&g=ensGene"
	 target=_blank>(susScr11)</a></li>
	<li>Pika <a href="../cgi-bin/hgTrackUi?db=ochPri2&g=ensGene"
	 target=_blank>(ochPri2)</a></li>
	<li>Platypus <a href="../cgi-bin/hgTrackUi?db=ornAna1&g=ensGene"
	 target=_blank>(ornAna1)</a></li>
	<li>Platypus <a href="../cgi-bin/hgTrackUi?db=ornAna2&g=ensGene"
	 target=_blank>(ornAna2)</a></li>
	<li>Rabbit <a href="../cgi-bin/hgTrackUi?db=oryCun2&g=ensGene"
	 target=_blank>(oryCun2)</a></li>
	<li>Rat <a href="../cgi-bin/hgTrackUi?db=rn6&g=ensGene"
	 target=_blank>(rn6)</a></li>
    </ul></div>
    <div class="col-sm-4">
    <ul>
	<li>Rhesus <a href="../cgi-bin/hgTrackUi?db=rheMac8&g=ensGene"
	 target=_blank>(rheMac8)</a></li>
	<li>Rock hyrax <a href="../cgi-bin/hgTrackUi?db=proCap1&g=ensGene"
	 target=_blank>(proCap1)</a></li>
	<li>S. cerevisiae <a href="../cgi-bin/hgTrackUi?db=sacCer3&g=ensGene"
	 target=_blank>(sacCer3)</a></li>
	<li>Sheep <a href="../cgi-bin/hgTrackUi?db=oviAri3&g=ensGene"
	 target=_blank>(oviAri3)</a></li>
	<li>Shrew <a href="../cgi-bin/hgTrackUi?db=sorAra1&g=ensGene"
	 target=_blank>(sorAra1)</a></li>
	<li>Sloth <a href="../cgi-bin/hgTrackUi?db=choHof1&g=ensGene"
	 target=_blank>(choHof1)</a></li>
	<li>Squirrel <a href="../cgi-bin/hgTrackUi?db=speTri2&g=ensGene"
	 target=_blank>(speTri2)</a></li>
	<li>Squirrel monkey <a href="../cgi-bin/hgTrackUi?db=saiBol1&g=ensGene"
	 target=_blank>(saiBol1)</a></li>
	<li>Stickleback <a href="../cgi-bin/hgTrackUi?db=gasAcu1&g=ensGene"
	 target=_blank>(gasAcu1)</a></li>
	<li>Tarsier <a href="../cgi-bin/hgTrackUi?db=tarSyr2&g=ensGene"
	 target=_blank>(tarSyr2)</a></li>
	<li>Tasmanian devil <a href="../cgi-bin/hgTrackUi?db=sarHar1&g=ensGene"
	 target=_blank>(sarHar1)</a></li>
	<li>Tenrec <a href="../cgi-bin/hgTrackUi?db=echTel1&g=ensGene"
	 target=_blank>(echTel1)</a></li>
	<li>Tetraodon <a href="../cgi-bin/hgTrackUi?db=tetNig2&g=ensGene"
	 target=_blank>(tetNig2)</a></li>
	<li>Tree shrew <a href="../cgi-bin/hgTrackUi?db=tupBel1&g=ensGene"
	 target=_blank>(tupBel1)</a></li>
	<li>Turkey <a href="../cgi-bin/hgTrackUi?db=melGal1&g=ensGene"
	 target=_blank>(melGal1)</a></li>
	<li>X. tropicalis <a href="../cgi-bin/hgTrackUi?db=xenTro3&g=ensGene"
	 target=_blank>(xenTro3)</a></li>
	<li>Zebra finch <a href="../cgi-bin/hgTrackUi?db=taeGut1&g=ensGene"
	 target=_blank>(taeGut1)</a></li>
	<li>Zebrafish <a href="../cgi-bin/hgTrackUi?db=danRer11&g=ensGene"
	 target=_blank>(danRer11)</a></li>
    </ul></div>
</div></div>
<p>
Thanks to Hiram Clawson and Conner Powell of the UCSC Genome Browser group for their
work in creating and releasing these tracks.</p>


<a name="012219"></a>
<h2>Jan. 22, 2019 &nbsp;&nbsp;Three new conservation tracks available for D. melanogaster, C. elegans, and G. gallus
</h2>
<p>
We are pleased to announce new comparative genomics tracks for each of the following organisms:
<a href="../cgi-bin/hgTracks?db=dm6">D. melanogaster</a> (fruit fly), <a href=
"../cgi-bin/hgTracks?db=ce11">C. elegans</a> (roundworm), and <a href=
"../cgi-bin/hgTracks?db=galGal6">G. gallus</a> (chicken). These composite tracks display multiple
alignments (multiZ) and measurements of evolutionary conservation (phastCons and phyloP) across
different species. For more details, visit the track description pages below.
</p>
<ul>
<li><em>D. melanogaster</em> alignment with 124 other insects:
<a href = "../cgi-bin/hgTrackUi?db=dm6&g=cons124way" target="_blank"> dm6</a></li>
<li><em>C. elegans</em> alignment with 135 other worms:
<a href = "../cgi-bin/hgTrackUi?db=ce11&g=cons135way" target="_blank">ce11</a></li>
<li><em>G. gallus</em> alignment with 77 other vertebrates:
<a href = "../cgi-bin/hgTrackUi?db=galGal6&g=cons77way" target="_blank">galGal6</a></li>
</ul>
<p>
The UCSC Genome Browser would like to thank Hiram Clawson, Conner Powell, Lou Nassar, and Daniel Schmelter for their efforts in creating these tracks.
</p>

<!-- ============= 2018 archived news ============= -->
<a name="2018"></a>

<a name="120418"></a>
<h2>Dec. 04, 2018 &nbsp;&nbsp; New dbSNP build 151 tracks now available for hg19 and hg38</h2>
<p>
We are pleased to announce the release of our &quot;SNP&quot; tracks derived from
<a href="https://www.ncbi.nlm.nih.gov/SNP/snp_summary.cgi?view+summary=view+summary&build_id=151"
target="_blank">NCBI&#39;s dbSNP Human Build 151 data</a>, available on the two most recent human
assemblies, 
<a href="../cgi-bin/hgTrackUi?db=hg38&g=snp151" target="_blank">GRCh38/hg38</a> and 
<a href="../cgi-bin/hgTrackUi?db=hg19&g=snp151" target="_blank">GRCh37/hg19</a>.
This release 
<a href="https://ncbiinsights.ncbi.nlm.nih.gov/2018/07/02/dbsnp-database-doubles-size-twice-13-
months/" target="_blank">will again double in size</a> to about 650 million variants compared to
the previous 150 release (300 million for hg38, and over 200 million for hg19).</p>

<p>
There are four SNP tracks available as part of this release:
<ul>
  <li>
  All SNPs (151): containing all mappings of reference SNPs to the human assembly. The tracks below
  are subsets of <i>All SNPs (151)</i>.</li>
  <li>
  Common SNPs (151): uniquely mapped variants that appear in at least 1% of the global population or
  are 100% non-reference.</li>
  <li>
  Flagged SNPs (151): uniquely mapped variants, excluding Common SNPs, that have been flagged by
  dbSNP as &quot;clinically associated.&quot;</li>
  <li>
  Mult. SNPs (151): variants that have been mapped to more than one genomic location. This track
  has been shrinking over the course of the last few releases as dbSNP now excludes most SNPs whose
  flanking sequences map to multiple locations in the genome.</li>
</ul></p>

<p class='text-center'>
  <img class='text-center' src="../images/SNPs151Announce.jpg" alt="SNPs151 growth" width="699"
  height="464">
  <p class='gbsCaption text-center'>The graph above shows the increasing growth of variants
  in the All SNPs track, by dbSNP release number, and the stagnant growth of the Common SNPs track, 
  essentially stuck at
  14.8 million variants.</p>
<p>
By default, only the <i>Common SNPs (151)</i> are visible in the browser; other tracks must be made
visible using the track controls. All four SNP tracks reside in the &quot;Variation&quot; group
category for the hg19 and hg38 genome browsers.</p>
<p>
We would like to thank Angie Hinrichs, Robert Kuhn, and Jairo Navarro for their work on creating
and analyzing the data. We'd like to thank the dbSNP group at NCBI for providing access to the data.
</p> 

<a name="112118"></a>
<h2>Nov. 21, 2018 &nbsp;&nbsp; New Genome Browser: Chicken, galGal6 (<em>Gallus gallus</em>)</h2>
<p>
A genome browser is now available for the <em>Gallus gallus</em> assembly released in March 2018 by the
<a href="https://www.ncbi.nlm.nih.gov/grc" target="_blank"> Genome Reference Consortium</a> (GRCg6a).
<h3>About the assembly:</h3>
<ul>
  <li><b>UCSC Genome Browser assembly ID:</b> galGal6</li>
  <li><b>Sequencing/Assembly provider ID:</b> <a href="https://www.ncbi.nlm.nih.gov/grc">
Genome Reference Consortium</a></li>
  <li><b>Assembly date:</b> March 2018</li>
  <li><b>Accession ID:</b> GCF_000002315.5</li>
  <li><b>NCBI Genome information:</b><a href="https://www.ncbi.nlm.nih.gov/genome/111" 
target="_blank" > 111</a> <em>(Gallus gallus)</em></li>
  <li><b>NCBI Assembly information:</b><a href="https://www.ncbi.nlm.nih.gov/assembly/1668981" 
target="_blank" > 1668981 </a></li>
  <li><b>NCBI BioProject information:</b><a href="https://www.ncbi.nlm.nih.gov/bioproject/13342" 
target="_blank" > 13342</a></li>
  <li><b>NCBI BioSample information:</b><a href="https://www.ncbi.nlm.nih.gov/biosample/SAMN02981218"
target="_blank" > SAMN02981218</a></li>
  <li><b>Contig+scaffold count:</b> 1928</li>
  <li><b>Total size:</b> 1,065,365,425</li>
  <li><a href="../goldenPath/credits.html#chicken_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
  <li><a href="../goldenPath/credits.html#chicken_use" 
target="_blank" >Data use conditions and restrictions</a></li>
  <li><a href="../goldenPath/help/ftp.html" 
target="_blank" >Rsync</a> and
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/galGal6/" 
target="_blank" >FTP</a></li>
  <li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#chicken"
target="_blank" >Downloads page</a></li>
</ul>
</p>

<a name="111918"></a>
<h2>Nov. 19, 2018 &nbsp;&nbsp; New Genome Browser: Orangutan, ponAbe3 (<em>Pongo pygmaeus abelii
</em>)</h2>
<p>
A genome browser is now available for the <em>Pongo pygmaeus abelii</em> assembly released in January 2018 by the
<a href="http://www.gs.washington.edu/" target="_blank"> University of Washington</a> (Susie_PABv2).
<h3>About the assembly:</h3>
<ul>
  <li><b>UCSC Genome Browser assembly ID:</b> ponAbe3</li>
  <li><b>Sequencing/Assembly provider ID:</b> <a href="http://eichlerlab.gs.washington.edu/index.html">
University of Washington</a></li>
  <li><b>Assembly date:</b> January 2018</li>
  <li><b>Accession ID:</b> GCF_002880775.1</li>
  <li><b>NCBI Genome information:</b><a href="https://www.ncbi.nlm.nih.gov/genome/325" 
target="_blank" > 325</a> <em>(Pongo pygmaeus abelii)</em></li>
  <li><b>NCBI Assembly information:</b><a href="https://www.ncbi.nlm.nih.gov/assembly/1529631" 
target="_blank" > 1529631 </a></li>
  <li><b>NCBI BioProject information:</b><a href="https://www.ncbi.nlm.nih.gov/bioproject/369439" 
target="_blank" > 369439</a></li>
  <li><b>NCBI BioSample information:</b><a href="https://www.ncbi.nlm.nih.gov/biosample/SAMN06275555" 
target="_blank" > SAMN06275555</a></li>
  <li><b>Contig+scaffold count:</b> 5,300</li>
  <li><b>Total size:</b> 3,065,052,215</li>
  <li><a href="../goldenPath/credits.html#orangutan_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
  <li><a href="../goldenPath/credits.html#orangutan_use" 
target="_blank" >Data use conditions and restrictions</a></li>
  <li><a href="../goldenPath/help/ftp.html" 
target="_blank" >Rsync</a> and
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/ponAbe3/" 
target="_blank" >FTP</a></li>
  <li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#orangutan"
target="_blank" >Downloads page</a></li>
</ul>
</p>

<a name="111618"></a>
<h2>Nov. 16, 2018 &nbsp;&nbsp; GENCODE Genes v29 track released for human (GRCh38/hg38)</h2>
<p>
We are  pleased to announce the release of the GENCODE Genes v29 track for the Human
(<a href="../cgi-bin/hgGateway?db=hg38" target="_blank">GRCh38/hg38</a>) 
assembly. In this GENCODE Genes update, we have made the ENST* IDs the default IDs and have added
the GENCODE pseudogenes to the knownGene tables. 
</p>
<p>
More details about the GENCODE Genes v29 track can be found on the
<a href="../cgi-bin/hgTrackUi?db=hg38&g=knownGene" target="_blank">GENCODE Genes track description
page</a>.</p>
<p>
We would like to thank the <a href="https://www.gencodegenes.org/" target="_blank">GENCODE
Project</a> for providing the data. The creation of this track was made possible thanks to Brian
Raney, Mark Diekhans, and Jairo Navarro.</p>

<a name="111318"></a>
<h2>Nov. 13, 2018 &nbsp;&nbsp; New video:  Saving and Sharing Sessions   </h2>
<p>
We are pleased to announce the release of another video tutorial, 
<a href =
"https://www.youtube.com/watch?v=d5rHBLXwraM&feature=youtu.be&feature=player_detailpage&v=8ATcoDTOc0g&list=UUQnUJepyNOw0p8s2otX4RYQ"
target = _blank>Saving and sharing sessions</a>.  

The Saved Sessions feature of the Browser has been around
for quite some time, but many of our users have not made full use of it.  It
offers a great way to keep track of your thinking on a particular topic and
has also been discussed in our <a href = "https://genome-blog.gi.ucsc.edu/blog/2015/08/"
target = blank>blog</a>.
</p>

<p>
This
<a href =
"https://www.youtube.com/watch?v=d5rHBLXwraM&feature=youtu.be&feature=player_detailpage&v=8ATcoDTOc0g&list=UUQnUJepyNOw0p8s2otX4RYQ"
target = _ blank>video</a>
shows how to create an
account, how to save sessions for easy future access, how to share sessions with
colleagues and how to make a public session that you can share with all Genome Browser
users.
The video comes with a full <a href = "../training/vids/transcript16.pdf" target =
blank>transcript</a>
</p>

<p>
Links to all of our tutorials are available on our <a href = "../training/"
target = _blank>training page</a> and on our <a href =
"https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos" target = _blank>YouTube
channel</a>.  At the YouTube channel and in the transcript, quick links offer access
to the video at key points.
</p>

<p>
Produced at UCSC by Robert  Kuhn and David Gibson.
</p>

<a name="110918"></a>
<h2>Nov. 09, 2018 &nbsp;&nbsp; BLAT ALL genomes feature</h2>
<p>
The UCSC Genome Browser is proud to announce a new <a href="../../cgi-bin/hgBlat?command=start" 
target="_blank">BLAT</a> feature: <em>Search ALL</em>. Generally, BLAT is used to find locations
of sequence similarity in a single target genome or to determine the exon structure of a mRNA. The
<em>Search ALL</em>  feature allows users to search the genomes of the default assemblies for all
of our organisms at once. The results page displays an ordered list of all our organisms and their
similarity to your query sequence. The results are ordered so that the organism with the most hits
is at the top, indicating which region(s) of that organism has the greatest similarity with your query
sequence. One example of how a user might employ the new <em>Search ALL</em> feature is to determine
what organism an unknown sequence comes from.
</p>
<p>
See the following link for examples and more information on the
<a href="../../goldenPath/help/hgTracksHelp.html#BLATAlign" target="_blank">Using BLAT alignments</a>
and
<a href="../../FAQ/FAQblat.html" target="_blank">BLAT FAQs</a>. <br><br>

Thanks to Galt Barber and Conner Powell for development and testing of this new feature.
</p>



<a name="110718"></a>
<h2>Nov. 7, 2018 &nbsp;&nbsp; Interact enhancements and GeneHancer track</h2>
<p>
The UCSC Genome Browser is pleased to announce enhancements to our 
<a href="../goldenPath/help/interact.html">interact track format</a> as well as 
our first native track using this format: 
<a href="../cgi-bin/hgTrackUi?db=hg19&g=geneHancer">GeneHancer</a>. GeneHancer is a 
database of human regulatory elements (enhancers and promoters) and their inferred target genes. 
These elements were created from over 1 million regulatory elements obtained from seven 
genome-wide databases--more information is available in the 
<a href="../cgi-bin/hgTrackUi?db=hg19&g=geneHancer#TRACK_HTML">methods</a>. 
This track is available on the hg19 and hg38 assemblies.</p>

<p>
Interact format now supports an inverted view (hills instead of valleys), pack and squish visibility 
modes, and a cluster mode which combines directional interactions with the same target or source 
into a single item. For additional information see our 
<a href="../goldenpath/help/trackDb/trackDbDoc.html#bigInteract">help docs</a>.</p>

<p>
Thanks to Simon Fishilevich, Marilyn Safran, Naomi Rosen, and Tsippi Iny Stein of the 
<a href="https://www.genecards.org/" target="_blank">GeneCards group</a> and Shifra Ben-Dor of the 
<a href="https://www.weizmann.ac.il/LS_CoreFacilities/bioinformatics/about" target="_blank">
Bioinformatics Core group at the Weizmann Institute</a> for providing this data and 
assisting with track development. Thanks to Kate Rosenbloom, Max Haeussler, and 
Luis Nassar of the UCSC Genome Browser for development and testing.</p>

<p>
<img src="../../images/GeneHancer_Announce.png" alt="GeneHancer_Release.png"></p>

<a name="101118"></a>
<h2>Oct. 11, 2018 &nbsp;&nbsp; Four new genome assemblies available </h2>

<p>
We are pleased to announce the release of four new assemblies now available in the Genome Browser:
chimpanzee, sheep, African malaria mosquito, and garter snake.
</p>

<h3>Chimpanzee <a href="../cgi-bin/hgGateway?db=panTro6">(Clint_PTRv2/panTro6)</a></h3>
<ul>
  <li>Assembly released Jan. 2018 by the <a href="https://www.washington.edu/" target="_blank">
  University of Washington</a>.</li>
  <li>There are 4,432 scaffolds with a total sequence length of 3,050,398,082 bases.</li>
  <li>For more information about this assembly, see the NCBI assembly record for
  <a href="https://www.ncbi.nlm.nih.gov/assembly/1642151" target="_blank">panTro6</a>.</li>
</ul>

<h3>Sheep <a href="../cgi-bin/hgGateway?db=oviAri4">(Oar_v4.0/oviAri4)</a></h3>
<ul>
  <li>Assembly released Dec. 2015 by the <a href="http://www.sheephapmap.org/" target="_blank">
  International Sheep Genomics Consortium</a>.</li>
  <li>There are 5,466 scaffolds with a total sequence length of 2,615,516,299 bases.</li>
  <li>For more information about this assembly, see the NCBI assembly record for
  <a href="https://www.ncbi.nlm.nih.gov/assembly/585171" target="_blank" >oviAri4</a>.</li>
</ul>

<h3>African malaria mosquito <a href="../cgi-bin/hgGateway?db=anoGam3">(AgamP3/anoGam3)</a></h3>
<ul>
  <li>Assembly released Oct. 2006 by <a href="https://www.ncbi.nlm.nih.gov/pubmed/17210077/"
  target="_blank">The International Consortium for the Sequencing of Anopheles Genome</a>. </li>
  <li>There are 8,145 scaffolds with a total sequence length of 265,027,044 bases.</li>
  <li>For more information about this assembly, see the NCBI assembly record for
  <a href="https://www.ncbi.nlm.nih.gov/assembly/305108" target="_blank">anoGam3</a>. </li>
</ul>

<h3>Garter snake <a href="../cgi-bin/hgGateway?db=thaSir1">(Thamnophis_sirtalis-6.0/thaSir1)</a>
</h3>
<ul>
  <li>Assembly released Jun. 2015 by <a href="https://www.genome.wustl.edu/"
  target="_blank">The Genome Institute at Washington University School of Medicine</a>.</li>
  <li>There are 7,930 scaffolds with a total size of 1,424,897,867 bases.</li>
  <li>For more information about this assembly, see the NCBI assembly record for
  <a href="https://www.ncbi.nlm.nih.gov/assembly/472161" target="_blank">thaSir1</a>.</li>
</ul>

<p>
Bulk downloads of the sequence and annotation data may be obtained from the
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/">Genome Browser FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html">Downloads page</a>. Please observe the
<a href="https://genome.ucsc.edu/FAQ/FAQdownloads.html#download2">conditions of use</a> when
accessing and using these data sets. The annotation tracks for this browser were generated by UCSC
and collaborators worldwide. See the
<a href="https://genome.ucsc.edu/goldenPath/credits.html">Credits</a> for a detailed list of the
organizations and individuals who contributed to these releases.</p>

<a name="081018"></a>
<h2>Aug. 10, 2018 &nbsp;&nbsp; New interact track type</h2>
<p>
The UCSC Genome Browser is proud to announce a new method to visualize interactions between regions
of the genome. The interact track format (and the binary-indexed bigInteract version) displays
pairwise interactions as arcs or half-rectangles connecting two genomic regions on the same
chromosome. Cross-chromosomal interactions can also be represented in this format. This format is
useful for displaying functional element interactions such as SNP/gene interactions, and is also
suitable for low-density chromatin interactions, such as ChIA-PET, and other use cases with a
limited number of interactions on the genome. It is not suitable for high-density chromatin data
such as Hi-C, but we are working on another new visualization to enable such display. Stay tuned...
<p>
See the following link for examples and more information on the
<a href="http://genome.ucsc.edu/goldenPath/help/interact.html">interact and bigInteract formats.</a>

<p>
Thanks to Kate Rosenbloom, Brian Raney, and Jairo Navarro for the
development and testing of this new track format.</p>

<p> 
<img src="../images/GeneHancer_Example.png" alt="GeneHancer_Example.png"></p>

<a name="073118"></a>
<h2>Jul. 31, 2018 &nbsp;&nbsp; New Ensembl gene tracks for 60 assemblies</h2>
<p>
The UCSC Genome Browser is proud to announce an update to the 
<a href="http://www.ensembl.org/info/genome/genebuild/genome_annotation.html" target="_blank">
Ensembl gene annotation</a> tracks, from <strong>v91 to v92</strong>, for the
following assemblies:</p>
<div class="container">
  <div class="row">
  <div class="col-sm-4">
  <ul>
      <li>Alpaca <a href="../cgi-bin/hgTrackUi?db=vicPac1&g=ensGenen/hgGateway?db=petMar3"
      target="_blank">(vicPac1)</a></li>
      <li>Armadillo <a href="../cgi-bin/hgTrackUi?db=dasNov3&g=ensGene"
      target="_blank">(dasNov3)</a></li>
      <li>Atlantic cod <a href="../cgi-bin/hgTrackUi?db=gadMor1&g=ensGene"
      target="_blank">(gadMor1)</a></li>
      <li>Bonobo <a href="../cgi-bin/hgTrackUi?db=panPan2&g=ensGene"
      target="_blank">(panPan2)</a></li>
      <li>Bushbaby <a href="../cgi-bin/hgTrackUi?db=otoGar3&g=ensGene"
      target="_blank">(otoGar3)</a></li>
      <li>C. elegans <a href="../cgi-bin/hgTrackUi?db=ce11&g=ensGene"
      target="_blank">(ce11)</a></li>
      <li>C. intestinalis <a href="../cgi-bin/hgTrackUi?db=ci3&g=ensGene"
      target="_blank">(ci3)</a></li>
      <li>Cat <a href="../cgi-bin/hgTrackUi?db=felCat8&g=ensGene"
      target="_blank">(felCat8)</a></li>
      <li>Chicken <a href="../cgi-bin/hgTrackUi?db=galGal5&g=ensGene"
      target="_blank">(galGal5)</a></li>
      <li>Chimp <a href="../cgi-bin/hgTrackUi?db=panTro5&g=ensGene"
      target="_blank">(panTro5)</a></li>
      <li>Chinese hamster <a href="../cgi-bin/hgTrackUi?db=criGriChoV1&g=ensGene"
      target="_blank">(criGriChoV1)</a></li>
      <li>Coelacanth <a href="../cgi-bin/hgTrackUi?db=latCha1&g=ensGene"
      target="_blank">(latCha1)</a></li>
      <li>Cow <a href="../cgi-bin/hgTrackUi?db=bosTau6&g=ensGene"
      target="_blank">(bosTau6)</a></li>
      <li>Crab-eating macaque <a href="../cgi-bin/hgTrackUi?db=macFas5&g=ensGene"
      target="_blank">(macFas5)</a></li>
      <li>D. melanogaster <a href="../cgi-bin/hgTrackUi?db=dm6&g=ensGene"
      target="_blank">(dm6)</a></li>
      <li>Dog <a href="../cgi-bin/hgTrackUi?db=canFam3&g=ensGene"
      target="_blank">(canFam3)</a></li>
      <li>Elephant <a href="../cgi-bin/hgTrackUi?db=loxAfr3&g=ensGene"
      target="_blank">(loxAfr3)</a></li>
      <li>Ferret <a href="../cgi-bin/hgTrackUi?db=musFur1&g=ensGene"
      target="_blank">(musFur1)</a></li>
      <li>Fugu <a href="../cgi-bin/hgTrackUi?db=fr2&g=ensGene"
      target="_blank">(fr2)</a></li>
      <li>Gibbon <a href="../cgi-bin/hgTrackUi?db=nomLeu3&g=ensGene"
      target="_blank">(nomLeu3)</a></li>
      <li>Golden snub-nosed monkey <a href="../cgi-bin/hgTrackUi?db=rhiRox1&g=ensGene"
      target="_blank">(rhiRox1)</a></li>
    </ul></div>
  <div class="col-sm-4">
    <ul>
      <li>Gorilla <a href="../cgi-bin/hgTrackUi?db=gorGor4&g=ensGene"
      target="_blank">(gorGor4)</a></li>
      <li>Green monkey <a href="../cgi-bin/hgTrackUi?db=chlSab2&g=ensGene"
      target="_blank">(chlSab2)</a></li>
      <li>Guinea pig <a href="../cgi-bin/hgTrackUi?db=cavPor3&g=ensGene"
      target="_blank">(cavPor3)</a></li>
      <li>Hedgehog <a href="../cgi-bin/hgTrackUi?db=eriEur1&g=ensGene"
      target="_blank">(eriEur1)</a></li>
      <li>Horse <a href="../cgi-bin/hgTrackUi?db=equCab2&g=ensGene"
      target="_blank">(equCab2)</a></li>
      <li>Lamprey <a href="../cgi-bin/hgTrackUi?db=petMar2&g=ensGene"
      target="_blank">(petMar2)</a></li>
      <li>Lizard <a href="../cgi-bin/hgTrackUi?db=anoCar2&g=ensGene"
      target="_blank">(anoCar2)</a></li>
      <li>Medaka <a href="../cgi-bin/hgTrackUi?db=oryLat2&g=ensGene"
      target="_blank">(oryLat2)</a></li>
      <li>Megabat <a href="../cgi-bin/hgTrackUi?db=pteVam1&g=ensGene"
      target="_blank">(pteVam1)</a></li>
      <li>Microbat <a href="../cgi-bin/hgTrackUi?db=myoLuc2&g=ensGene"
      target="_blank">(myoLuc2)</a></li>
      <li>Naked mole-rat <a href="../cgi-bin/hgTrackUi?db=hetGla1&g=ensGene"
      target="_blank">(hetGla1)</a></li>
      <li>Naked mole-rat <a href="../cgi-bin/hgTrackUi?db=hetGla2&g=ensGene"
      target="_blank">(hetGla2)</a></li>
      <li>Opossum <a href="../cgi-bin/hgTrackUi?db=monDom5&g=ensGene"
      target="_blank">(monDom5)</a></li>
      <li>Orangutan <a href="../cgi-bin/hgTrackUi?db=ponAbe2&g=ensGene"
      target="_blank">(ponAbe2)</a></li>
      <li>Panda <a href="../cgi-bin/hgTrackUi?db=ailMel1&g=ensGene"
      target="_blank">(ailMel1)</a></li>
      <li>Pig <a href="../cgi-bin/hgTrackUi?db=susScr11&g=ensGene"
      target="_blank">(susScr11)</a></li>
      <li>Pika <a href="../cgi-bin/hgTrackUi?db=ochPri2&g=ensGene"
      target="_blank">(ochPri2)</a></li>
      <li>Platypus <a href="../cgi-bin/hgTrackUi?db=ornAna1&g=ensGene"
      target="_blank">(ornAna1)</a></li>
      <li>Platypus <a href="../cgi-bin/hgTrackUi?db=ornAna2&g=ensGene"
      target="_blank">(ornAna2)</a></li>
      <li>Rabbit <a href="../cgi-bin/hgTrackUi?db=oryCun2&g=ensGene"
      target="_blank">(oryCun2)</a></li>
      <li>Rat <a href="../cgi-bin/hgTrackUi?db=rn6&g=ensGene"
      target="_blank">(rn6)</a></li>
    </ul></div>
  <div class="col-sm-4">
    <ul>
      <li>Rhesus <a href="../cgi-bin/hgTrackUi?db=rheMac8&g=ensGene"
      target="_blank">(rheMac8)</a></li>
      <li>Rock hyrax <a href="../cgi-bin/hgTrackUi?db=proCap1&g=ensGene"
      target="_blank">(proCap1)</a></li>
      <li>S. cerevisiae <a href="../cgi-bin/hgTrackUi?db=sacCer3&g=ensGene"
      target="_blank">(sacCer3)</a></li>
      <li>Sheep <a href="../cgi-bin/hgTrackUi?db=oviAri3&g=ensGene"
      target="_blank">(oviAri3)</a></li>
      <li>Shrew <a href="../cgi-bin/hgTrackUi?db=sorAra1&g=ensGene"
      target="_blank">(sorAra1)</a></li>
      <li>Sloth <a href="../cgi-bin/hgTrackUi?db=choHof1&g=ensGene"
      target="_blank">(choHof1)</a></li>
      <li>Squirrel <a href="../cgi-bin/hgTrackUi?db=speTri2&g=ensGene"
      target="_blank">(speTri2)</a></li>
      <li>Squirrel monkey <a href="../cgi-bin/hgTrackUi?db=saiBol1&g=ensGene"
      target="_blank">(saiBol1)</a></li>
      <li>Stickleback <a href="../cgi-bin/hgTrackUi?db=gasAcu1&g=ensGene"
      target="_blank">(gasAcu1)</a></li>
      <li>Tarsier <a href="../cgi-bin/hgTrackUi?db=tarSyr2&g=ensGene"
      target="_blank">(tarSyr2)</a></li>
      <li>Tasmanian devil <a href="../cgi-bin/hgTrackUi?db=sarHar1&g=ensGene"
      target="_blank">(sarHar1)</a></li>
      <li>Tenrec <a href="../cgi-bin/hgTrackUi?db=echTel1&g=ensGene"
      target="_blank">(echTel1)</a></li>
      <li>Tetraodon <a href="../cgi-bin/hgTrackUi?db=tetNig2&g=ensGene"
      target="_blank">(tetNig2)</a></li>
      <li>Tree shrew <a href="../cgi-bin/hgTrackUi?db=tupBel1&g=ensGene"
      target="_blank">(tupBel1)</a></li>
      <li>Turkey <a href="../cgi-bin/hgTrackUi?db=melGal1&g=ensGene"
      target="_blank">(melGal1)</a></li>
      <li>X. tropicalis <a href="../cgi-bin/hgTrackUi?db=xenTro3&g=ensGene"
      target="_blank">(xenTro3)</a></li>
      <li>Zebra finch <a href="../cgi-bin/hgTrackUi?db=taeGut1&g=ensGene"
      target="_blank">(taeGut1)</a></li>
      <li>Zebrafish <a href="../cgi-bin/hgTrackUi?db=danRer11&g=ensGene"
      target="_blank">(danRer11)</a></li>
    </ul></div>
</div></div>
<p>
Thanks to Hiram Clawson, Chris Lee, and Jairo Navarro of the UCSC Genome Browser group for their
work in creating and releasing these tracks.</p>

<a name="071718"></a>
<h2>Jul. 17, 2018 &nbsp;&nbsp; DECIPHER variants track available for human (GRCh37/hg19)</h2>
<p>
We are pleased to announce the release of a new DECIPHER variants track available for human,
<a href="../cgi-bin/hgGateway?db=hg19" target="_blank">GRCh37/hg19</a>. The entries in the
<a href="../cgi-bin/hgTrackUi?db=hg19&g=decipherSnvs" target="_blank">DECIPHER SNVs</a> track are
colored according to the estimated pathogenicity of the variant. Entries are colored:</p>
<ul>
  <li><b><font color="black">black</font></b> for likely or definitely pathogenic</li>
  <li><b><font color="#888">dark grey</font></b> for uncertain or unknown</li>
  <li><b><font color="#c8c8c8">light grey</font></b> for likely or definitely benign</li>
</ul>
For more information on DECIPHER, please contact
<A HREF="mailto:&#100;&#101;c&#105;&#112;h&#101;&#114;&#64;&#115;&#97;&#110;&#103;&#101;&#114;.
a&#99;.&#117;&#107;">
&#100;&#101;c&#105;&#112;h&#101;&#114;&#64;&#115;&#97;&#110;&#103;&#101;&#114;.a&#99;.&#117;&#107;</a>.
Thanks to Jonathan Casper and Jairo Navarro of the UCSC Genome Browser group for their work in
creating and releasing this track.
</p>

<a name="070618"></a>
<h2>Jul. 06, 2018 &nbsp;&nbsp; GRCh37 patch release 13 available for human (GRCh37/hg19)</h2>
<p>
We are happy to announce the <a href="../cgi-bin/hgTrackUi?db=hg19&g=hg19Patch13" target="_blank">
patch release 13</a> is available for the <a href="../cgi-bin/hgGateway?db=hg19" target="_blank">
GRCh37/hg19</a> assembly. This is a minor release of GRCh37 that does not disrupt the coordinate
system in the reference sequence GRCh37. All but four patch scaffolds from patch releases 1-12 were
retained.</p>

<p>Between patch release 12 and patch release 13:</p>
<ul>
  <li>10 patch scaffolds of type FIX were added.</li>
  <li>14 patch scaffolds of type FIX were updated.</li>
  <li>1 patch scaffold of type NOVEL was updated.</li>
</ul>
<p>Total patch scaffolds in this patch release: 204</p>
<ul>
  <li>Patch scaffolds of type FIX: 131</li>
  <li>Patch scaffolds of type NOVEL: 73</li>
</ul>
<p>
Thanks to Hiram Clawson and Jairo Navarro of the UCSC Genome Browser group for their work in
creating and releasing these tracks.</p>

<a name="062118"></a>
<h2>Jun. 21, 2018 &nbsp;&nbsp;
New exome and whole genome sequencing data for human (GRCh37/hg19) - gnomAD</h2>
<p>
We are happy to announce the release of the <a href="http://gnomad.broadinstitute.org/"
target="_blank">Genome Aggregation Database (gnomAD)</a> for the human (<a target="_blank"
href="../cgi-bin/hgGateway?db=hg19">GRCh37/hg19</a>) assembly. This release consists of two
callsets: variants from 123,136 exomes and 15,496 whole genomes of unrelated individuals sequenced
as part of various population-genetic and disease-specific
<a href="http://gnomad.broadinstitute.org/about" target="_blank">studies</a> collected by the
Genome Aggregation Database (gnomAD), release 2.0.2.</p>
<p>
For more information on the processing pipeline and population annotations, see the following <a
href="https://macarthurlab.org/2017/02/27/the-genome-aggregation-database-gnomad" target="_blank">
blog post</a> and the 2.0.2 <a target="_blank"
href="https://storage.googleapis.com/gnomad-public/release/2.0.2/README.txt" target="_blank">
README</a>. Thanks to the <a href="http://gnomad.broadinstitute.org/about" target="_blank">Genome
Aggregation Database Consortium</a> for making this data available. The data is released under the
<a href="http://opendatacommons.org/licenses/odbl/1.0/" target="_blank">ODC Open Database License
(OBdL)</a> as described <a href="http://opendatacommons.org/licenses/odbl/1.0/"
target="_blank">here</a>. Special thanks to Jonathan Casper and Jairo Navarro for their efforts in
creating and releasing these tracks.</p>

<a name="052318"></a>
<h2>May 23, 2018 &nbsp;&nbsp; Updated UCSC Genes track released for mouse (GRCm38/mm10)</h2>
<p>
We are  pleased to announce the release of an updated UCSC Genes track for the Mouse
(<a href="../cgi-bin/hgGateway?db=mm10" target="_blank">GRCm38/mm10</a>) 
assembly. The new release has 63,814 total transcripts, compared with 63,759 in the previous 
version. The total number of canonical genes has decreased from 33,079 to 32,989. Comparing the new 
gene set with the previous version:</p>
<ul>
  <li>
  57,239 transcripts did not change between versions.</li>
  <li>
  453 transcripts were not carried forward to the new version.</li>
  <li>
  5,273 transcripts are &quot;compatible&quot; with those in the previous set, meaning
  that the two transcripts show consistent splicing.  In most cases, the old and new
  transcripts differ in the lengths of their UTRs.</li>
  <li>
  794 transcripts overlap with those in the previous set but do not show consistent
  splicing (i.e., they contain overlapping introns with differing splice sites).</li>
</ul>
<p>
More details about the new UCSC Genes track can be found on the
<a href="../cgi-bin/hgTrackUi?db=mm10&g=knownGene" target="_blank">UCSC Genes track description
page</a>.</p>
<p>
Many thanks to Brian Raney and Jairo Navarro for their work on this track.</p>

<a name="050318"></a>
<h2>May 03, 2018 &nbsp;&nbsp; Updated GENCODE gene tracks for human assemblies, hg19 and hg38</h2>
<p>
We are pleased to announce the release of the updated GENCODE Gene annotation tracks, which
corresponds to the <a href="http://www.ensembl.info/2018/04/05/ensembl-92-has-been-released/"
target="_blank">Ensembl 92 release</a> for human assemblies, <a target="_blank"
href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV28lift37">hg19/GRCh37</a> and
<a target="_blank" href="../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeGencodeV28">hg38/GRCh38</a>.
All GENCODE V28 annotations are available for hg38/GRCh38, and the annotation release was
back-mapped to the hg19/GRCh37 assembly. For both hg19 and hg38, the GENCODE v28 gene set contains
the following tracks:</p>
<ul>
  <li>Basic - a subset of the <em>Comprehensive set</em>.</li>
  <li>Comprehensive - all GENCODE coding and non-coding transcript annotations, including
      polymorphic pseudogenes. This includes both manual and automatic annotations.</li>
  <li>Pseudogenes - all annotations except polymorphic pseudogenes.</li></ul>
<p>
The hg38 assembly also includes the following tracks that are not available on hg19:
</p>
<ul>
  <li>2-way Pseudogenes - pseudogenes predicted by both the Yale Pseudopipe and UCSC
      Retrofinder pipelines.</li>
  <li>PolyA - polyA signals and sites manually annotated on the genome based on transcribed
      evidence (ESTs and cDNAs) of 3' end of transcripts containing at least 3 A's not matching the
      genome.</li></ul>
<p>Thanks to Mark Diekhans and Jairo Navarro for the development and release of these tracks.</p>

<a name="040418"></a>
<h2>Apr. 04, 2018 &nbsp;&nbsp; New NCBI RefSeq tracks for human, rat, yeast, C. elegans, zebrafish, X. tropicalis and fly</h2>
<p>
We are pleased to announce the release of a new set of gene annotation tracks for the hg19/GRCh37, 
hg38/GRCh38, rn6/Rnor_6.0, sacCer3/R64, ce11/WBcel235, danRer10/GRCz10, danRer11/GRCz11, 
xenTro7/Xtropicalis_v7, xenTro9/Xenopus_tropicalis_v9.1 and dm6/Release 6 plus ISO1 MT
assemblies based on data from NCBI's Reference Sequence Database (RefSeq). For all of these tracks,
the alignments and coordinates are provided by RefSeq. These tracks are organized in a composite
track that includes:</p>
<ul>
  <li>RefSeq All &ndash; all annotations from the curated and predicted sets
  <li>RefSeq Curated &ndash; curated annotations beginning with NM, NR, or NP
  <li>RefSeq Predicted &ndash; predicted annotations beginning with XM or XR
  <li>RefSeq Other &ndash; all other RefSeq annotations not included in RefSeq All
  <li>RefSeq Alignments &ndash; alignments of transcripts to the genome provided by RefSeq
</ul>
<p>
The new composite track also includes a &quot;UCSC RefSeq&quot; track that is based on our original
&quot;RefSeq Genes&quot; track. As with before, this UCSC track is a result of our realignments
of RefSeq RNAs to the genome, which means that there may be some cases where the annotations
differ from those in the new NCBI RefSeq tracks. Also note that the <em>Predictions</em> subtrack is 
unavailable for the following assemblies: hg19, dm6, ce11, and sacCer3.</p>
<p>
More details about these tracks, their construction, and how to access their underlying data
can be found on
<a href="https://genome-blog.gi.ucsc.edu/blog/the-new-ncbi-refseq-tracks-and-you/"
target="_blank">this blog post</a>.</p>
<p>
A huge thank you to Terence Murphy from the RefSeq group and to Hiram Clawson, Angie Hinrichs, 
Christopher Lee and many others from the UCSC Genome Browser staff for
bringing this track to life.</p>

<a name="040318"></a>
<h2>Apr. 03, 2018 &nbsp;&nbsp; New genome browser tool: Track Collection Builder</h2>
<p>
We are pleased to announce a new tool in the UCSC Genome Browser:</p>
<p>
The <a href="../cgi-bin/hgCollection" target="_blank">Track Collection Builder</a> (My Data &gt;
Track Collection Builder) is a tool that allows multiple tracks to be copied and grouped together
into one composite track or &quot;collection&quot;. A 
<a href="../goldenPath/help/trackDb/trackDbHub.html#compositeTrack" target="_blank">
composite track</a> can be thought of as a container for grouping a set of similar-type annotation 
tracks. Once grouped, settings (e.g., track height, graph color, overlay views) can be applied to 
all sub-tracks at once. Currently, <a href="../cgi-bin/hgCollection#SupportedData" target="_blank">
supported data types</a> are limited to continuous graphing data (wiggle, bigWig, and bedGraph).</p>
<p>
The Track Collection Builder is pictured below - note the left and right panes. The left pane, 
&quot;Available Tracks&quot; displays tracks that are available to add to a new collection. The 
first folder, &quot;Visible Tracks&quot; displays all supported tracks that are currently seen in 
the browser (i.e., not set to "hide" visibility). Below that are expandable folders which show 
selectable tracks from all other browser categories, regardless of browser visibility. To add 
custom tracks or tracks from hubs, first load them into the browser and then add them to a 
collection.</p>
<p>
The right pane, &quot;Collected Tracks&quot; displays current collections. In the image below, two 
tracks (which are currently visible in the browser) have been added to a new collection. In the 
lower part of the image, the browser graphic is shown displaying this collection (highlighted 
in green). Below the collection are the two original tracks (&quot;My Custom Track&quot; and the 
native GC Percent track). Note that the two original tracks have been copied into the new 
collection, where they are now grouped sub-tracks.</p>
<p>
Below are several new features which are detailed in the 
<a href="../cgi-bin/hgCollection#Configure" target="_blank">detailed in the help guide</a>:</p>
<ul>
  <li>Sort sub-tracks in a collection by similarity</li> 
  <li>Sort sub-tracks in a collection by magnitude</li> 
  <li>New collection &quot;overlay&quot; to add all sub-tracks together</li> 
  <li>New collection &quot;overlay&quot; to subtract the first (top) track from the sum of all 
  other sub-tracks</li> 
</ul>

<p>
See also: <a href="https://www.youtube.com/watch?v=2_RiIY07omY" target="_blank">
Quick Demo Video of the Track Collection Builder</a></p>

<p> 
<img src="../../images/hgCollectionNewsDemo.jpg" alt="hgCollectionNewsDemo.jpg"></p>

<a name="032118"></a>
<h2>Mar. 21, 2018 &nbsp;&nbsp; New genome browser: Sea lamprey, petMar3
(<em>Petromyzon marinus</em>)</h2>
<p>A genome browser is now available for the <em>Petromyzon marinus</em> assembly released in
December 2017 by the <a href="https://ukhealthcare.uky.edu/genomics-core-laboratory"
target="_blank">University of Kentucky</a> (Pmar_germline 1.0, UCSC version petMar3).</p>
<h3>About the assembly:</h3>
<ul>
   <li><b>UCSC Genome Browser assembly ID:</b>
      <a href="../cgi-bin/hgGateway?db=petMar3" target="_blank">petMar3</a></li>
   <li><b>Sequencing/Assembly provider ID:</b>
      <a href="https://ukhealthcare.uky.edu/genomics-core-laboratory" target="_blank">
      University of Kentucky</a></li>
   <li><b>Assembly date:</b> December 2017</li>
   <li><b>Accession ID:</b> GCA_002833325.1</li>
   <li><b>NCBI Genome information:</b><a href="https://www.ncbi.nlm.nih.gov/genome/287" 
      target="_blank"> 287</a> <em>(Petromyzon marinus)</em></li>
   <li><b>NCBI Assembly information:</b><a href="https://www.ncbi.nlm.nih.gov/assembly/1474531" 
      target="_blank"> 1474531</a> <em> (Pmar_germline 1.0) </em></li>
   <li><b>BioProject information:</b><a href="https://www.ncbi.nlm.nih.gov/bioproject/357048" 
      target="_blank"> 357048</a></li>
   <li><b>BioSample information:</b><a href="https://www.ncbi.nlm.nih.gov/biosample/SAMN06131802" 
      target="_blank"> SAMN06131802</a></li>
   <li><b>Contig+scaffold count:</b> 36,453</li>
   <li><b>Total size:</b> 1,130,417,696</li>
   <li><a href="../goldenPath/credits.html#lamprey_credits" target="_blank">Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
   <li><a href="../goldenPath/credits.html#lamprey_use" target="_blank">
       Data use conditions and restrictions</a></li>
   <li><a href="../goldenPath/help/ftp.html" target="_blank">Rsync</a> and 
       <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/petMar3/" target="_blank">FTP</a></li>
   <li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#lamprey" target="_blank">
       Downloads page</a></li>
</ul>

<a name="022018"></a>
<h2>Feb. 20, 2018 &nbsp;&nbsp; New video:  Visibility control in the Browser </h2>
<p>
We have released a new video demonstrating the
<a href="https://www.youtube.com/watch?v=jKix2B3hwnw&list=UUQnUJepyNOw0p8s2otX4RYQ"
   target=_blank>visibility controls</a> in the Browser.
 In our Browser <a href = "http://bit.ly/ucscTraining" target = _blank>training
 workshops</a> we are often asked about the meaning of the track controls:
 squish, dense, pack and full.  In this tutorial we describe the controls for
 BED, wiggle and conservation tracks.  There is also a discussion of visibility 
 control in composite tracks (2-level nesting) and supertracks (3-level).

 This brings to 16 the number of video tutorials available on our 
 <a href="https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos" 
   target="_blank"> YouTube channel</a>.  Most are accompanied by a transcript,
 via a link on the video itself and from our <a href = "../../training/">training 
 page</a>.
</p>
<p>
 We thank Robert Kuhn and David Gibson for video production.
</p>


<a name="021618"></a>
<h2>Feb. 16, 2018 &nbsp;&nbsp; New search support for chromosome aliases</h2>
<p>
Have you ever wanted to use 1 instead of chr1 to search for a chromosome position?</p>
<p>
The following assemblies now support a new feature for matching different chromosome aliases.
Equivalent entries only exist in the new table, chromAlias, when an exact sequence match has been
verified. This feature is limited to searching coordinates, it is not yet supported for custom
tracks.</p>
<p>
<div class="container">
  <div class="row">
  <div class="col-sm-4">
    <ul>
      <li>American alligator <a href="../cgi-bin/hgGateway?db=allMis1"
      target="_blank">(allMis1)</a></li>
      <li>Armadillo <a href="../cgi-bin/hgGateway?db=dasNov3"
      target="_blank">(dasNov3)</a></li>
      <li>Baboon <a href="../cgi-bin/hgGateway?db=papAnu2"
      target="_blank">(papAnu2)</a></li>
      <li>Bonobo <a href="../cgi-bin/hgGateway?db=panPan1"
      target="_blank">(panPan1)</a></li>
      <li>Budgerigar <a href="../cgi-bin/hgGateway?db=melUnd1"
      target="_blank">(melUnd1)</a></li>
      <li>Bushbaby <a href="../cgi-bin/hgGateway?db=otoGar3"
      target="_blank">(otoGar3)</a></li>
      <li>C. elegans <a href="../cgi-bin/hgGateway?db=ce11"
      target="_blank">(ce11)</a></li>
      <li>Cat <a href="../cgi-bin/hgGateway?db=felCat5"
      target="_blank">(felCat5)</a></li>
      <li>Cat <a href="../cgi-bin/hgGateway?db=felCat8"
      target="_blank">(felCat8)</a></li>
      <li>Chicken <a href="../cgi-bin/hgGateway?db=galGal4"
      target="_blank">(galGal4)</a></li>
      <li>Chimp <a href="../cgi-bin/hgGateway?db=panTro4"
      target="_blank">(panTro4)</a></li>
      <li>Chimp <a href="../cgi-bin/hgGateway?db=panTro5"
      target="_blank">(panTro5)</a></li>
      <li>Coelacanth <a href="../cgi-bin/hgGateway?db=latCha1"
      target="_blank">(latCha1)</a></li>
      <li>Cow <a href="../cgi-bin/hgGateway?db=bosTau8"
      target="_blank">(bosTau8)</a></li>
      <li>Crab-eating macaque <a href="../cgi-bin/hgGateway?db=macFas5"
      target="_blank">(macFas5)</a></li>
      <li>D. melanogaster <a href="../cgi-bin/hgGateway?db=dm6"
      target="_blank">(dm6)</a></li>
    </ul></div>
  <div class="col-sm-4">
    <ul>
      <li>D. persimilis <a href="../cgi-bin/hgGateway?db=droPer1"
      target="_blank">(droPer1)</a></li>
      <li>Dog <a href="../cgi-bin/hgGateway?db=canFam3"
      target="_blank">(canFam3)</a></li>
      <li>Elephant shark <a href="../cgi-bin/hgGateway?db=calMil1"
      target="_blank">(calMil1)</a></li>
      <li>Fugu <a href="../cgi-bin/hgGateway?db=fr3"
      target="_blank">(fr3)</a></li>
      <li>Green monkey <a href="../cgi-bin/hgGateway?db=chlSab2"
      target="_blank">(chlSab2)</a></li>
      <li>Guinea pig <a href="../cgi-bin/hgGateway?db=cavPor3"
      target="_blank">(cavPor3)</a></li>
      <li>Hedgehog <a href="../cgi-bin/hgGateway?db=eriEur2"
      target="_blank">(eriEur2)</a></li>
      <li>Lizard <a href="../cgi-bin/hgGateway?db=anoCar2"
      target="_blank">(anoCar2)</a></li>
      <li>Malayan flying lemur <a href="../cgi-bin/hgGateway?db=galVar1"
      target="_blank">(galVar1)</a></li>
      <li>Manatee <a href="../cgi-bin/hgGateway?db=triMan1"
      target="_blank">(triMan1)</a></li>
      <li>Marmoset <a href="../cgi-bin/hgGateway?db=calJac3"
      target="_blank">(calJac3)</a></li>
      <li>Medium ground finch <a href="../cgi-bin/hgGateway?db=geoFor1"
      target="_blank">(geoFor1)</a></li>
      <li>Microbat <a href="../cgi-bin/hgGateway?db=myoLuc2"
      target="_blank">(myoLuc2)</a></li>
      <li>Minke whale <a href="../cgi-bin/hgGateway?db=balAcu1"
      target="_blank">(balAcu1)</a></li>
      <li>Naked mole-rat <a href="../cgi-bin/hgGateway?db=hetGla2"
      target="_blank">(hetGla2)</a></li>
      <li>Nile tilapia <a href="../cgi-bin/hgGateway?db=oreNil2"
      target="_blank">(oreNil2)</a></li>
    </ul></div>
  <div class="col-sm-4">
    <ul>
      <li>Panda <a href="../cgi-bin/hgGateway?db=ailMel1"
      target="_blank">(ailMel1)</a></li>
      <li>Pig <a href="../cgi-bin/hgGateway?db=susScr3"
      target="_blank">(susScr3)</a></li>
      <li>Rhesus <a href="../cgi-bin/hgGateway?db=rheMac8"
      target="_blank">(rheMac8)</a></li>
      <li>Sheep <a href="../cgi-bin/hgGateway?db=oviAri3"
      target="_blank">(oviAri3)</a></li>
      <li>Shrew <a href="../cgi-bin/hgGateway?db=sorAra2"
      target="_blank">(sorAra2)</a></li>
      <li>Squirrel <a href="../cgi-bin/hgGateway?db=speTri2"
      target="_blank">(speTri2)</a></li>
      <li>Tarsier <a href="../cgi-bin/hgGateway?db=tarSyr2"
      target="_blank">(tarSyr2)</a></li>
      <li>Tasmanian devil <a href="../cgi-bin/hgGateway?db=sarHar1"
      target="_blank">(sarHar1)</a></li>
      <li>Tenrec <a href="../cgi-bin/hgGateway?db=echTel2"
      target="_blank">(echTel2)</a></li>
      <li>Turkey <a href="../cgi-bin/hgGateway?db=melGal5"
      target="_blank">(melGal5)</a></li>
      <li>White rhinoceros <a href="../cgi-bin/hgGateway?db=cerSim1"
      target="_blank">(cerSim1)</a></li>
      <li>X. tropicalis <a href="../cgi-bin/hgGateway?db=xenTro7"
      target="_blank">(xenTro7)</a></li>
      <li>Zebra finch <a href="../cgi-bin/hgGateway?db=taeGut2"
      target="_blank">(taeGut2)</a></li>
      <li>Zebrafish <a href="../cgi-bin/hgGateway?db=danRer7"
      target="_blank">(danRer7)</a></li>
    </ul></div>
</div></div></p>
<p>
For example, for <a href="../cgi-bin/hgTracks?db=felCat8" target="_blank">felCat8</a>, aliases that
now map to chrA2 are NC_018724.2, CM001379.2, and A2. Search terms such as
<code>CM001379.2 133,978,000 134,105,980</code> will be equivalent to
<code>chrA2 133,978,000 134,105,980</code>.
</p>

<a name="020818"></a>
<h2>Feb. 08, 2018 &nbsp;&nbsp; New video:  in-silico PCR (isPCR)</h2>
<p>
We have released a new video demonstrating the
<a href="https://youtu.be/U8_QYwmdGYU?feature=player_detailpage&v=8ATcoDTOc0g&list=UUQnUJepyNOw0p8s2otX4RYQ"
   target=_blank>in-silico PCR tool</a> (isPCR).
 This tutorial shows how to use a pair of primers to amplify either DNA or RNA 
 templates for human or mouse genomes and to visualize the results on the Genome 
 Browser.
 For other genomes, only DNA may serve as a template.  This brings to 15 the
 number of video tutorials available on our 
 <a href="https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos" 
   target="_blank"> YouTube channel</a>.  Most are accompanied by a transcript,
 via a link on the video itself and from our <a href = "../../training/">training page</a>.
</p>
<p>
 We thank Robert Kuhn and David Gibson for video production.
</p>

<a name="020118"></a>
<h2>Feb. 01, 2018 &nbsp;&nbsp; New Ensembl gene tracks for 66 assemblies</h2>
<p>
The UCSC Genome Browser has released updated
<a href="http://www.ensembl.org/info/genome/genebuild/genome_annotation.html" target="_blank">
Ensembl gene annotation</a> tracks, from <strong>v89 to v90</strong>, for the
following assemblies:
<ul>
    <li>Baboon <a href="../cgi-bin/hgTrackUi?db=papAnu2&g=ensGene"
    target="_blank">(papAnu2)</a></li>
    <li>Cat <a href="../cgi-bin/hgTrackUi?db=felCat5&g=ensGene"
    target="_blank">(felCat5)</a></li>
    <li>Chimp <a href="../cgi-bin/hgTrackUi?db=panTro4&g=ensGene"
    target="_blank">(panTro4)</a></li>
    <li>Gibbon <a href="../cgi-bin/hgTrackUi?db=nomLeu1&g=ensGene"
    target="_blank">(nomLeu1)</a></li>
    <li>Gorilla <a href="../cgi-bin/hgTrackUi?db=gorGor3&g=ensGene"
    target="_blank">(gorGor3)</a></li>
    <li>Mouse lemur <a href="../cgi-bin/hgTrackUi?db=micMur2&g=ensGene"
    target="_blank">(micMur2)</a></li>
    <li>Tarsier <a href="../cgi-bin/hgTrackUi?db=tarSyr1&g=ensGene"
    target="_blank">(tarSyr1)</a></li>
</ul></p>

<p>
The UCSC Genome Browser has also released updated
<a href="http://www.ensembl.org/info/genome/genebuild/genome_annotation.html" target="_blank">
Ensembl gene annotation tracks</a>, from <strong>v89 to v91</strong>, for the
following assemblies:
<ul>
    <li>Alpaca <a href="../cgi-bin/hgTrackUi?db=vicPac1&g=ensGene"
    target="_blank">(vicPac1)</a></li>
    <li>Armadillo <a href="../cgi-bin/hgTrackUi?db=dasNov3&g=ensGene"
    target="_blank">(dasNov3)</a></li>
    <li>Atlantic cod <a href="../cgi-bin/hgTrackUi?db=gadMor1&g=ensGene"
    target="_blank">(gadMor1)</a></li>
    <li>Bushbaby <a href="../cgi-bin/hgTrackUi?db=otoGar3&g=ensGene"
    target="_blank">(otoGar3)</a></li>
    <li>C. elegans <a href="../cgi-bin/hgTrackUi?db=ce11&g=ensGene"
    target="_blank">(ce11)</a></li>
    <li>Cat <a href="../cgi-bin/hgTrackUi?db=felCat8&g=ensGene"
    target="_blank">(felCat8)</a></li>
    <li>Chicken <a href="../cgi-bin/hgTrackUi?db=galGal5&g=ensGene"
    target="_blank">(galGal5)</a></li>
    <li>Chimp <a href="../cgi-bin/hgTrackUi?db=panTro5&g=ensGene"
    target="_blank">(panTro5)</a></li>
    <li>Chinese hamster <a href="../cgi-bin/hgTrackUi?db=criGriChoV1&g=ensGene"
    target="_blank">(criGriChoV1)</a></li>
    <li>Coelacanth <a href="../cgi-bin/hgTrackUi?db=latCha1&g=ensGene"
    target="_blank">(latCha1)</a></li>
    <li>Cow <a href="../cgi-bin/hgTrackUi?db=bosTau6&g=ensGene"
    target="_blank">(bosTau6)</a></li>
    <li>Crab-eating macaque <a href="../cgi-bin/hgTrackUi?db=macFas5&g=ensGene"
    target="_blank">(macFas5)</a></li>
    <li>D. melanogaster <a href="../cgi-bin/hgTrackUi?db=dm6&g=ensGene"
    target="_blank">(dm6)</a></li>
    <li>Dog <a href="../cgi-bin/hgTrackUi?db=canFam3&g=ensGene"
    target="_blank">(canFam3)</a></li>
    <li>Elephant <a href="../cgi-bin/hgTrackUi?db=loxAfr3&g=ensGene"
    target="_blank">(loxAfr3)</a></li>
    <li>Ferret <a href="../cgi-bin/hgTrackUi?db=musFur1&g=ensGene"
    target="_blank">(musFur1)</a></li>
    <li>Fugu <a href="../cgi-bin/hgTrackUi?db=fr2&g=ensGene"
    target="_blank">(fr2)</a></li>
    <li>Gibbon <a href="../cgi-bin/hgTrackUi?db=nomLeu3&g=ensGene"
    target="_blank">(nomLeu3)</a></li>
    <li>Golden snub-nosed monkey <a href="../cgi-bin/hgTrackUi?db=rhiRox1&g=ensGene"
    target="_blank">(rhiRox1)</a></li>
    <li>Gorilla <a href="../cgi-bin/hgTrackUi?db=gorGor4&g=ensGene"
    target="_blank">(gorGor4)</a></li>
    <li>Green monkey <a href="../cgi-bin/hgTrackUi?db=chlSab2&g=ensGene"
    target="_blank">(chlSab2)</a></li>
    <li>Guinea pig <a href="../cgi-bin/hgTrackUi?db=cavPor3&g=ensGene"
    target="_blank">(cavPor3)</a></li>
    <li>Hedgehog <a href="../cgi-bin/hgTrackUi?db=eriEur1&g=ensGene"
    target="_blank">(eriEur1)</a></li>
    <li>Horse <a href="../cgi-bin/hgTrackUi?db=equCab2&g=ensGene"
    target="_blank">(equCab2)</a></li>
    <li>Lamprey <a href="../cgi-bin/hgTrackUi?db=petMar2&g=ensGene"
    target="_blank">(petMar2)</a></li>
    <li>Lizard <a href="../cgi-bin/hgTrackUi?db=anoCar2&g=ensGene"
    target="_blank">(anoCar2)</a></li>
    <li>Marmoset <a href="../cgi-bin/hgTrackUi?db=calJac3&g=ensGene"
    target="_blank">(calJac3)</a></li>
    <li>Medaka <a href="../cgi-bin/hgTrackUi?db=oryLat2&g=ensGene"
    target="_blank">(oryLat2)</a></li>
    <li>Megabat <a href="../cgi-bin/hgTrackUi?db=pteVam1&g=ensGene"
    target="_blank">(pteVam1)</a></li>
    <li>Microbat <a href="../cgi-bin/hgTrackUi?db=myoLuc2&g=ensGene"
    target="_blank">(myoLuc2)</a></li>
    <li>Naked mole-rat <a href="../cgi-bin/hgTrackUi?db=hetGla1&g=ensGene"
    target="_blank">(hetGla1)</a></li>
    <li>Naked mole-rat <a href="../cgi-bin/hgTrackUi?db=hetGla2&g=ensGene"
    target="_blank">(hetGla2)</a></li>
    <li>Opossum <a href="../cgi-bin/hgTrackUi?db=monDom5&g=ensGene"
    target="_blank">(monDom5)</a></li>
    <li>Orangutan <a href="../cgi-bin/hgTrackUi?db=ponAbe2&g=ensGene"
    target="_blank">(ponAbe2)</a></li>
    <li>Panda <a href="../cgi-bin/hgTrackUi?db=ailMel1&g=ensGene"
    target="_blank">(ailMel1)</a></li>
    <li>Pig <a href="../cgi-bin/hgTrackUi?db=susScr11&g=ensGene"
    target="_blank">(susScr11)</a></li>
    <li>Pika <a href="../cgi-bin/hgTrackUi?db=ochPri2&g=ensGene"
    target="_blank">(ochPri2)</a></li>
    <li>Platypus <a href="../cgi-bin/hgTrackUi?db=ornAna1&g=ensGene"
    target="_blank">(ornAna1)</a></li>
    <li>Platypus <a href="../cgi-bin/hgTrackUi?db=ornAna2&g=ensGene"
    target="_blank">(ornAna2)</a></li>
    <li>Rabbit <a href="../cgi-bin/hgTrackUi?db=oryCun2&g=ensGene"
    target="_blank">(oryCun2)</a></li>
    <li>Rat <a href="../cgi-bin/hgTrackUi?db=rn6&g=ensGene"
    target="_blank">(rn6)</a></li>
    <li>Rhesus <a href="../cgi-bin/hgTrackUi?db=rheMac8&g=ensGene"
    target="_blank">(rheMac8)</a></li>
    <li>Rock hyrax <a href="../cgi-bin/hgTrackUi?db=proCap1&g=ensGene"
    target="_blank">(proCap1)</a></li>
    <li>S. cerevisiae <a href="../cgi-bin/hgTrackUi?db=sacCer3&g=ensGene"
    target="_blank">(sacCer3)</a></li>
    <li>Sheep <a href="../cgi-bin/hgTrackUi?db=oviAri3&g=ensGene"
    target="_blank">(oviAri3)</a></li>
    <li>Shrew <a href="../cgi-bin/hgTrackUi?db=sorAra1&g=ensGene"
    target="_blank">(sorAra1)</a></li>
    <li>Sloth <a href="../cgi-bin/hgTrackUi?db=choHof1&g=ensGene"
    target="_blank">(choHof1)</a></li>
    <li>Squirrel monkey <a href="../cgi-bin/hgTrackUi?db=saiBol1&g=ensGene"
    target="_blank">(saiBol1)</a></li>
    <li>Squirrel <a href="../cgi-bin/hgTrackUi?db=speTri2&g=ensGene"
    target="_blank">(speTri2)</a></li>
    <li>Stickleback <a href="../cgi-bin/hgTrackUi?db=gasAcu1&g=ensGene"
    target="_blank">(gasAcu1)</a></li>
    <li>Tarsier <a href="../cgi-bin/hgTrackUi?db=tarSyr2&g=ensGene"
    target="_blank">(tarSyr2)</a></li>
    <li>Tasmanian devil <a href="../cgi-bin/hgTrackUi?db=sarHar1&g=ensGene"
    target="_blank">(sarHar1)</a></li>
    <li>Tenrec <a href="../cgi-bin/hgTrackUi?db=echTel1&g=ensGene"
    target="_blank">(echTel1)</a></li>
    <li>Tetraodon <a href="../cgi-bin/hgTrackUi?db=tetNig2&g=ensGene"
    target="_blank">(tetNig2)</a></li>
    <li>Tree shrew <a href="../cgi-bin/hgTrackUi?db=tupBel1&g=ensGene"
    target="_blank">(tupBel1)</a></li>
    <li>Turkey <a href="../cgi-bin/hgTrackUi?db=melGal1&g=ensGene"
    target="_blank">(melGal1)</a></li>
    <li>X. tropicalis <a href="../cgi-bin/hgTrackUi?db=xenTro3&g=ensGene"
    target="_blank">(xenTro3)</a></li>
    <li>Zebra finch <a href="../cgi-bin/hgTrackUi?db=taeGut1&g=ensGene"
    target="_blank">(taeGut1)</a></li>
    <li>Zebrafish <a href="../cgi-bin/hgTrackUi?db=danRer10&g=ensGene"
    target="_blank">(danRer10)</a></li>
</ul></p>

<a name="011518"></a>
<h2>Jan. 15, 2018 &nbsp;&nbsp; New hg38 comparative genomics track: 30 mammals (27 primates) multiz 
alignment &amp; conservation </h2>
<p>
A new <a href="../../cgi-bin/hgTrackUi?db=hg38&g=cons30way" target="_blank">30-way 
comparative genomics track</a> has been added to the 
<a href="../../cgi-bin/hgGateway?db=hg38" target="_blank">human GRCh38/hg38 browser</a>. This 
composite track displays multiple alignments (Multiz) and measurements of evolutionary conservation 
(phastCons and phyloP) for 30 species (27 primates). Developer: Hiram Clawson, UCSC Genome Browser.
</p>
<h3>Tracks</h3>
<ul>
<li>Multiz Alignments</li> 
<li>Basewise Conservation by PhyloP</li>
<li>Element Conservation by PhastCons</li>
<li>Conserved Elements</li> 

</ul>
<h3>Species</h3>
<p>
Please see the <a href="../../cgi-bin/hgTrackUi?db=hg38&g=cons30way" target="_blank">
30-way Multiz &amp; Conservation track description page</a> for assembly details of participating 
assemblies: </p>
<p>
Human, chimp, bonobo, gorilla, orangutan, gibbon, rhesus, crab-eating macaque, pig-tailed macaque,
 sooty mangabey, baboon, green monkey, drill, proboscis monkey, angolan colobus, 
 golden snub-nosed monkey, black snub-nosed monkey, marmoset, squirrel monkey, white-faced sapajou,
 ma's night monkey, tarsier, mouse lemur, coquerel's sifaka, black lemur, sclater's lemur, bushbaby,
  mouse, dog, and armadillo.</p>

<p>
<a name="010518"></a>
<h2>Jan. 05, 2018 &nbsp;&nbsp; New mappability data for human, hg38</h2>
<p>
We are pleased to announce the release of a new set of annotations from the
<a href="https://hoffmanlab.org/" target="_blank">Hoffman Lab</a> that contains mappability data for
both the reference genome and bisulfite-converted genome of
<a href="../../cgi-bin/hgGateway?db=hg38" target="_blank">hg38/GRCh38</a>. These tracks indicate
regions with uniquely mappable reads of particular lengths before and after bisulfite conversion.
</p>

<p>
The <a href="../../cgi-bin/hgTrackUi?db=hg38&g=mappability" target="_blank">Mappability</a>
supertrack contains two composite tracks, <a href="../../cgi-bin/hgTrackUi?db=hg38&g=mappability"
target="_blank">Bismap</a> and <a href="../../cgi-bin/hgTrackUi?db=hg38&g=mappability"
target="_blank">Umap</a>. Both Umap and Bismap tracks contain single-read and multi-read mappability
tracks for four different read lengths: 24 bp, 36 bp, 50 bp, and 100 bp. You can use these tracks
for many purposes, including filtering unreliable signal from sequencing assays. The Bismap track
can help filter unreliable signal from sequencing assays involving bisulfite conversion, such as
whole-genome bisulfite sequencing or reduced representation bisulfite sequencing.</p>

<p>
For greater detail and explanatory diagrams, see the
<a href="https://www.biorxiv.org/content/early/2016/12/20/095463" target="_blank">preprint</a>,
the <a href="http://bismap.hoffmanlab.org/" target="_blank">Umap and Bismap project website</a>,
or the <a href="https://github.com/hoffmangroup/umap" target="_blank">Umap and Bismap software
documentation</a>.</p>
<p>
A huge thank you to <a href="https://sites.google.com/site/anshulkundaje/"
target="_blank">Anshul Kundaje</a> (Stanford University) who created the original Umap software in
MATLAB and Mehran Karimzadeh (<a href="https://hoffmanlab.org/" target="_blank">Michael Hoffman
lab</a>, Princess Margaret Cancer Centre) who implemented the Python version of Umap and added
features, including Bismap. Thanks to Hiram Clawson, Chris Eisenhart, and Jairo Navarro of the UCSC
Genome Browser.

<!-- ============= 2017 archived news ============= -->
<a name="2017"></a>

<a name="121917"></a>
<h2>Dec. 19, 2017 &nbsp;&nbsp; New genome browser: Bonobo, panPan2
(<em>Pan paniscus</em>)</h2>
<p>A genome browser is now available for the <em>Pan paniscus</em> assembly released in August
2015 by the <a href="http://www.eva.mpg.de/index.html">Max-Planck Institute for Evoluntionary
Anthropology</a> (MPI-EVA panpan1.1, UCSC version panPan2).</p>
<h3>About the assembly:</h3>
<ul>
<li><b>UCSC Genome Browser assembly ID:</b> panPan2</li>
<li><b>Sequencing/Assembly provider ID:</b> <a href="http://www.eva.mpg.de/index.html">Max-Planck
Institute for Evolutionary Anthropology</a></li>
<li><b>Assembly date:</b> August 2015</li>
<li><b>Accession ID:</b> GCF_000258655.2</li>
<li><b>NCBI Genome information:</b><a href="https://www.ncbi.nlm.nih.gov/genome/10729" 
target="_blank" > NCBI genome/10729</a> <em>(Pan paniscus)</em></li>
<li><b>NCBI Assembly information:</b><a href="https://www.ncbi.nlm.nih.gov/assembly/474211" 
target="_blank" >  NCBI assembly/474211 (panpan1.1) </a></li>
<li><b>BioProject information:</b><a href="https://www.ncbi.nlm.nih.gov/bioproject/169343" 
target="_blank" > NCBI Bioproject: 169343</a></li>
<li><b>BioSample information:</b><a href="https://www.ncbi.nlm.nih.gov/biosample/SAMEA1029457" 
target="_blank" > NCBI Biosample: SAMEA1029457</a></li>
<li><b>Contig+scaffold count:</b> 132,340</li>
<li><b>Total size:</b> 3,286,643,896</li>
<li><a href="../goldenPath/credits.html#bonobo_credits" 
target="_blank">Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
<li><a href="../goldenPath/credits.html#bonobo_use" 
target="_blank">Data use conditions and restrictions</a></li>
<li><a href="../goldenPath/help/ftp.html" 
target="_blank">Rsync</a> and 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/panPan2/" 
target="_blank">FTP</a></li>
<li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#bonobo"
target="_blank">Downloads page</a></li>
</ul>

<a name="121317"></a>
<h2>Dec. 13, 2017 &nbsp;&nbsp; New NCBI RefSeq Functional Elements tracks for human and mouse</h2>
<p>
We are pleased to announce the release of a new set of 
<a href="https://www.ncbi.nlm.nih.gov/refseq/functionalelements/" target="_blank">functional
regulatory elements</a> for the <a href="../cgi-bin/hgTrackUi?db=hg38&g=refSeqFuncElems"
target="_blank">hg38/GRCh38</a> and <a href="../cgi-bin/hgTrackUi?db=mm10&g=refSeqFuncElems"
target="_blank">mm10/GRCm38</a> assemblies. NCBI is now providing
<a href="https://www.ncbi.nlm.nih.gov/refseq/" target="_blank">RefSeq</a> and
<a href="https://www.ncbi.nlm.nih.gov/gene/" target="_blank">Gene</a> records for non-genic
functional elements that have been described in the literature and are experimentally validated.
Elements in scope include experimentally-verified gene regulatory regions, known structural
elements, well-characterized DNA replication origins, and clinically-significant sites of DNA
recombination and genomic instability.</p>
<p>
Functional elements are colored by <a href="http://www.sequenceontology.org/"
                                      target=_blank>Sequence Ontology (SO)</a> term
using the same scheme as NCBI's Genome Data Viewer:
  <ul>
    <li><span style="color: #008080"><b>Regulatory elements</b></span>
      (items labeled by <a href="http://www.insdc.org/controlled-vocabulary-regulatoryclass"
                           target=_blank>INSDC regulatory class</a>)
    <li><span style="color: #b00000"><b>Protein binding sites</b></span>
      (items labeled by bound moiety)
    <li><span style="color: #0000b0"><b>Mobile elements</b></span>
    <li><span style="color: #a0522d"><b>Recombination features</b></span>
    <li><span style="color: #b000b0"><b>Sequence features</b></span>
    <li><span style="color: #000000"><b>Other</b></span>
  </ul>
</p>
<p>
A huge thank you to Terence Murphy from the RefSeq group, Angie Hinrichs, and many others from the
UCSC Genome Browser staff for bringing this track to life.</p>

<a name="121117"></a>
<h2>Dec. 11, 2017 &nbsp;&nbsp;New bigNarrowPeak track format</h2>
<p>
We're happy to announce support for a new binary indexed track type called
<a href="help/bigNarrowPeak.html">bigNarrowPeak</a>. This new track type joins
the ranks of other recent &quot;big&quot; track types: bigMaf, bigChain, bigPsl,
bigBarChart and bigGenePred</p>
<p>
With bigNarrowPeak, any narrowPeak files that were previously already made into bigBeds
in the past with the bedToBigBed utility and the proper narrowPeak.as definitions should
now display just as though they were narrowPeak custom tracks. For example here is a file
from the ENCODE 2012 AWG Hub that can be pasted into the
<a href="../../cgi-bin/hgCustom?db=hg19" target="_blank">Custom Tracks page</a> with
<code>type=bigNarrowPeak</code> instead of <code>type=bigBed</code> used previously:</p>
<pre><code>track type=bigNarrowPeak visibility=full db=hg19 name="nPk" description="AWG ENCODE narrowPeak Example" bigDataUrl=http://ftp.ebi.ac.uk/pub/databases/ensembl/encode/integration_data_jan2011/byDataType/peaks/jan2011/spp/optimal/hub/spp.optimal.wgEncodeOpenChromChipGm12878Pol2AlnRep0_VS_wgEncodeOpenChromChipGm12878InputAln.bb
</pre></code>
<p>
These tracks also support additional filter options for their additional fields,
see more with this illustrated <a href="help/bigNarrowPeak.html#example3">example</a>.</p>

<a name="120817"></a>
<h2>Dec. 8, 2017 &nbsp;&nbsp;BLAT directly from the search box</h2>
<p>
Did you know you could BLAT from the search box?</p>
<p>
If you have short sequences (less than 2000 characters) you can directly search
for them when Browsing without first visiting the BLAT tool to input the sequence.
You can <a href ="#050417">also output your BLAT results as a custom track</a>,
giving the results a name and description, and then save special results into Sessions
allowing you to return to those findings when you wish to see them at a later date.</p>

<a name="120617"></a>
<h2>Dec. 6, 2017 &nbsp;&nbsp; New genome browser: Western clawed frog, xenTro9
(<em>Xenopus tropicalis</em></h2>
<p>A genome browser is now available for the <em>X. tropicalis</em> assembly released in July
2016 by the <a href="http://genome.jgi.doe.gov/">US DOE Joint Genome Institute</a> (JGI v9.1,
UCSC version xenTro9).</p>
<h3>About the assembly:</h3>
<ul>
<li><b>UCSC Genome Browser assembly ID:</b> xenTro9</li>
<li><b>Sequencing/Assembly provider ID:</b> <a href="http://genome.jgi.doe.gov/">US DOE Joint 
Genome Institute</a></li>
<li><b>Assembly date:</b> July 2016</li>
<li><b>Accession ID:</b> GCF_000004195.3</li>
<li><b>NCBI Genome information:</b><a href="https://www.ncbi.nlm.nih.gov/genome/80" 
target="_blank" > NCBI genome/80</a> <em>(Xenopus tropicalis)</em></li>
<li><b>NCBI Assembly information:</b><a href="https://www.ncbi.nlm.nih.gov/assembly/768701" 
target="_blank" >  NCBI assembly/768701 (US DOE Joint Genome Institute) </a></li>
<li><b>BioProject information:</b><a href="https://www.ncbi.nlm.nih.gov/bioproject/12348" 
target="_blank" > NCBI Bioproject: 12348</a></li>
<li><b>BioSample information:</b><a href="https://www.ncbi.nlm.nih.gov/biosample/SAMN00000117" 
target="_blank" > NCBI Biosample: SAMN00000117</a></li>
<li><b>Contig+scaffold count:</b> 68,152</li>
<li><b>Total size:</b> 1,440,398,454</li>
<li><a href="../goldenPath/credits.html#xentro_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
<li><a href="../goldenPath/credits.html#xentro_use" 
target="_blank" >Data use conditions and restrictions</a></li>
<li><a href="../goldenPath/help/ftp.html" 
target="_blank" >Rsync</a> and 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/xenTro9/" 
target="_blank" >FTP</a></li>
<li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#xentro"
target="_blank" >Downloads page</a></li>
</ul>

<a name="112117"></a>
<h2>Nov. 21, 2017 &nbsp;&nbsp; New NCBI RefSeq tracks for mouse, mm10</h2>
<p>
We are pleased to announce the release of a new set of gene annotation tracks for the mm10/GRCm38
assembly based on data from NCBI's Reference Sequence Database (RefSeq). For all of these tracks,
the alignments and coordinates are provided by RefSeq. These tracks are organized in a composite
track that includes:</p>
<ul>
  <li>RefSeq All &ndash; all annotations from the curated and predicted sets
  <li>RefSeq Curated &ndash; curated annotations beginning with NM, NR, or NP
  <li>RefSeq Predicted &ndash; predicted annotations beginning with XM or XR
  <li>RefSeq Other &ndash; all other RefSeq annotations not included in RefSeq All
  <li>RefSeq Alignments &ndash; alignments of transcripts to the genome provided by RefSeq
</ul>
<p>
The new composite track also includes a &quot;UCSC RefSeq&quot; track that is based on our original
&quot;RefSeq Genes&quot; track. As with before, this UCSC track is a result of our realignments
of RefSeq RNAs to the genome, which means that there may be some cases where the annotations
differ from those in the new NCBI RefSeq tracks.</p>
<p>
More details about these tracks, their construction, and how to access their underlying data
can be found on
<a href="https://genome-blog.gi.ucsc.edu/blog/the-new-ncbi-refseq-tracks-and-you/"
target="_blank">this blog</a>
or the <a href="../../cgi-bin/hgTrackUi?db=mm10&g=refSeqComposite">track description page</a>.</p>
<p>
A huge thank you to Terence Murphy from the RefSeq group and to Hiram Clawson, Brian Raney,
Mark Diekhans, Angie Hinrichs, Jairo Navarro and many others from the UCSC Genome Browser staff for
bringing this track to life.</p>

<a name="111517"></a>
<h2>Nov. 15, 2017 &nbsp;&nbsp; New genome browser: African clawed frog, xenLae2 <em>
(Xenopus laevis)</em> </h2>
<p>Please welcome <a href="../cgi-bin/hgGateway?db=xenLae2" target="_blank" 
>xenLae2</a>, the 192nd genome assembly in the UCSC Genome Browser. This is the first assembly for
the African clawed frog <em>(Xenopus laevis)</em>.</p>
<h3>About the assembly:</h3>
<ul>
<li><b>UCSC Genome Browser assembly ID:</b> xenLae2</li>
<li><b>Sequencing/Assembly provider ID:</b> International Xenopus Sequencing Consortium</li>
<li><b>Assembly date:</b> Aug. 2016</li>
<li><b>Accession ID:</b> GCF_001663975.1</li>
<li><b>NCBI Genome information:</b><a href="https://www.ncbi.nlm.nih.gov/genome/81" 
target="_blank" > NCBI genome/81</a> <em>(Xenopus laevis)</em></li>
<li><b>NCBI Assembly information:</b><a href="https://www.ncbi.nlm.nih.gov/assembly/779791" 
target="_blank" >  NCBI assembly/779791 (International Xenopus Sequencing Consortium) </a></li>
<li><b>BioProject information:</b><a href="https://www.ncbi.nlm.nih.gov/bioproject/338693" 
target="_blank" > NCBI Bioproject: 338693</a></li>
<li><b>BioSample information:</b><a href="https://www.ncbi.nlm.nih.gov/biosample/SAMN04518361" 
target="_blank" > NCBI Biosample: SAMN04518361</a></li>
<li><b>Contig/scaffold count: </b>108,033</li>
<li><b>Total size:</b> 2,718,433,805</li>
<li><a href="../goldenPath/credits.html#african_clawed_frog_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
<li><a href="../goldenPath/credits.html#african_clawed_frog_use" 
target="_blank" >Data use conditions and restrictions</a></li>
<li><a href="../goldenPath/help/ftp.html" 
target="_blank" >Rsync</a> and 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/xenLae2/" 
target="_blank" >FTP</a></li>
<li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#african_clawed_frog" 
target="_blank" >Downloads page</a></li>
</ul>


<a name="111317"></a>
<h2>Nov. 13, 2017 &nbsp;&nbsp; New genome browser: Pig, susScr11 <em>(Sus scrofa)</em> </h2>
<p>Please welcome <a href="../cgi-bin/hgGateway?db=susScr11" target="_blank" 
>susScr11</a>, the 191st genome assembly in the UCSC Genome Browser. This is the third assembly in
the browser for pig <em>(Sus scrofa)</em> .</p>
<h3>About the assembly:</h3>
<ul>
<li><b>UCSC Genome Browser assembly ID:</b> susScr11</li>
<li><b>Sequencing/Assembly provider ID:</b> The Swine Genome Sequencing Consortium (SGSC) 
Sscrofa11.1</li>
<li><b>Assembly date:</b> Feb. 2017</li>
<li><b>Accession ID:</b>GCF_000003025.6</li>
<li><b>NCBI Genome ID:</b><a href="https://www.ncbi.nlm.nih.gov/genome/84" 
target="_blank" > 84</a> <em>(Sus scrofa)</em></li>
<li><b>NCBI Assembly ID:</b><a href="https://www.ncbi.nlm.nih.gov/assembly/1004191" 
target="_blank" > 1004191 </a><em>(Sus scrofa)</em></li>
<li><b>BioProject ID:</b><a href="https://www.ncbi.nlm.nih.gov/bioproject/13421" 
target="_blank" > 13421</a></li>
<li><b>BioSample ID:</b><a href="https://www.ncbi.nlm.nih.gov/biosample/SAMN02953785" 
target="_blank" > SAMN02953785</a></li>
<li><b>Contig/scaffold count: </b>613</li>
<li><b>Total size:</b> 2,501,912,388</li>
<li><a href="../goldenPath/credits.html#pig_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
<li><a href="../goldenPath/credits.html#pig_use" 
target="_blank" >Data use conditions and restrictions</a></li>
<li><a href="../goldenPath/help/ftp.html" 
target="_blank" >Rsync</a> and 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/susScr11/" 
target="_blank" >FTP</a></li>
<li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#pig" 
target="_blank" >Downloads page</a></li>
</ul>

<a name="110717"></a>
<h2>Nov. 07, 2017 &nbsp;&nbsp; New genome browser: Chinese hamster <em>(Cricetulus griseus)</em> 
ovary cell line, CHO-K1</h2>
<p>Please welcome <a href="../cgi-bin/hgGateway?db=criGriChoV1" target="_blank" 
>criGriChoV1</a>, the 190th genome assembly in the UCSC Genome Browser. This assembly marks the 
first cell line to reside in the browser; the Chinese hamster <em>(Cricetulus griseus)</em> ovary 
cell line, CHO-K1.</p>
<h3>About the assembly:</h3>
<ul>
<li><b>UCSC Genome Browser assembly ID:</b> criGriChoV1</li>
<li><b>Sequencing/Assembly provider ID:</b> Beijing Genomics Institute CriGri_1.0</li>
<li><b>Assembly date:</b> Aug. 2011</li>
<li><b>Accession ID:</b> GCF_000223135.1</li>
<li><b>NCBI Genome ID:</b><a href="https://www.ncbi.nlm.nih.gov/genome/2791" 
target="_blank" > 2791</a> (Cricetulus griseus)</li>
<li><b>NCBI Assembly ID:</b><a href="https://www.ncbi.nlm.nih.gov/assembly/309608" 
target="_blank" > 309608</a> (CriGri_1.0)</li>
<li><b>BioProject ID:</b><a href="https://www.ncbi.nlm.nih.gov/bioproject/69991" 
target="_blank" > 69991</a></li>
<li><b>BioSample ID:</b><a href="https://www.ncbi.nlm.nih.gov/biosample/SAMN02981352" 
target="_blank" > SAMN02981352</a></li>
<li><b>Contig/scaffold count: </b>109,152</li>
<li><b>Total size:</b> 2,399,786,748</li>
<li><a href="../goldenPath/credits.html#chinese_hamster_credits" 
target="_blank" >Acknowledgments</a></li>
</ul>
<h3>Data and Downloads:</h3>
<ul>
<li><a href="../goldenPath/credits.html#chinese_hamster_use" 
target="_blank" >Data use conditions and restrictions</a></li>
<li><a href="../goldenPath/help/ftp.html" 
target="_blank" >Rsync</a> and 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/criGriChoV1/" 
target="_blank" >FTP</a></li>
<li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#chinese_hamster" 
target="_blank" >Downloads page</a></li>
</ul>

<a name='110217'></a>
<h2>Nov. 2, 2017 &nbsp;&nbsp; New Zebrafish assembly danRer11 available!</h2>
<p>
A genome browser is now available for the Zebrafish (<em>Danio rerio</em>) assembly released in
May 2017 by The Genome Reference Consortium. For more information and statistics
about this assembly, see the <a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000002035.6/">
NCBI assembly record</a> for GRCz11. There are 20,680 contigs and 2,848 scaffolds in this assembly 
with a total size of 1,679,203,469 bases.
</p>

<p>
Bulk downloads of the sequence and annotation data may be obtained from the <a
href="ftp://hgdownload.gi.ucsc.edu/goldenPath/danRer11">Genome Browser FTP
server</a> or the <a href="http://hgdownload.gi.ucsc.edu/downloads.html#zebrafish">Downloads
page</a>. Please observe the <a href="https://genome.ucsc.edu/FAQ/FAQdownloads.html#download2">
conditions for use</a> when accessing and using these data sets. The annotation tracks for this 
browser were generated by UCSC and collaborators worldwide. See the 
<a href="credits.html#zebrafish_credits">Credits page</a> for a detailed list of the 
organizations and individuals who contributed to this release.
</p>

<a name="102717"></a>
<h2>Oct. 27, 2017 &nbsp;&nbsp; Public European UCSC MariaDB server now available!</h2>
<p>
We are pleased to announce that a public MariaDB server of UCSC Genome Browser
data is now available in Europe! This should speed up programmatic access to
the underlying MariaDB databases for users in Europe. Connect to this new MariaDB
server using the command:</p>
<pre><code> mysql --user=genome --host=genome-euro-mysql.soe.ucsc.edu -A</code></pre>

<p>
More information about connecting to our public MariaDB servers can be found on our
<a href="help/mysql.html">MariaDB access</a> page.</p>

<p>
Additionally, we have configured both the <a href="help/gbib.html">Genome Browser in a Box</a>
(GBiB) and <a href="help/gbic.html"> Genome Browser in the Cloud</a> (GBiC) to
take advantage of this new server. If you are in Europe and have auto-updates
enabled for your GBiB, it should automatically switch to using this new server
the next time it updates. The GBiC will only auto-configure this
connection when first created, otherwise, you will need to manually configure
which MariaDB server you want your mirror to use.</p>

<a name="101817"></a>
<h2>Oct. 18, 2017 &nbsp;&nbsp; New GTEx eQTL 44 tissues and summary tracks on hg19</h2>
<p>
We are pleased to announce two new GTEx eQTL tracks in the GRCh37/hg19 browser within 
the &quot;Regulation&quot; track category. These tracks show genetic variants associated with and
likely causal for differences in gene expression in 44 human tissues from the 
<a href="https://www.genome.gov/27543767/" target="_blank">Genotype-Tissue Expression (GTEx)</a> 
V6 data release. The data items displayed are gene expression quantitative trait loci within 1MB 
of gene transcription start sites (cis-eQTLs), significantly associated with gene expression and 
in the credible set of variants for the gene at a high confidence level (95%).</p>

<p>
<b><a href="../cgi-bin/hgGtexTrackSettings?db=hg19&g=gtexEqtlCluster"
target="_blank">hg19: GTEx Combined eQTL</a></b></br>
This "all in one" track displays all cis-eQTL's from the 44 tissues, collapsed into a single track
for a summary view in the browser display. Gene/variant pairs occurring in multiple tissues are
combined into a single item in the display. The item label shows the number of tissues where the
eQTL was identified, or the tissue name and the GTEx-convention tissue color if the eQTL was
identified solely in one tissue. Mouseover lists all tissues affected and the effect size. The item
color reflects the largest effect size in any tissue (red for positive, blue for negative).
Click-through on a combined eQTL shows for each tissue the significance of the association, effect
size on gene expression, and the probability the eQTL belongs to the the 95% credible set. Track
configuration supports filtering by gene, effect size, or probability. Tissues can be selected via
checkboxes or from the UCSC GTEx Body Map graphic. </p>

<p>
<b><a href="../cgi-bin/hgTrackUi?db=hg19&g=gtexEqtlTissue" target="_blank">hg19: GTEx Tissue eQTL
</a></b></br>This track is a composite track containing 44 subtracks representing the GTEx eQTL 
tissues. Each subtrack contains all eQTLs identified for that tissue. Filtering is available for 
all 44 subtracks.</p>

<p>
These two tracks join the previously released GTEx data tracks and track hubs in the UCSC Genome
Browser:
<ul>
<li><a href="../cgi-bin/hgGtexTrackSettings?db=hg19&g=gtexGene"
target="_blank">GTEx Gene</a></li>
<li><a href="../cgi-bin/hgTrackUi?db=hg19&g=gtexTranscExpr" target="_blank">
GTEx Transcript</a></li>
<li><a href="http://hgdownload.gi.ucsc.edu/hubs/gtex/gtexHub.html" target="_blank">
GTEx RNA-Seq Signal Hub</a> on <a href="
../cgi-bin/hgTracks?db=hg38&hubUrl=http://hgdownload.gi.ucsc.edu/hubs/gtex/hub.txt" 
target="_blank">hg19</a> and 
<a href="../cgi-bin/hgTracks?db=hg38&hubUrl=http://hgdownload.gi.ucsc.edu/hubs/gtex/hub.txt" 
target="_blank">hg38</a>.</li>
<li><a href="http://hgdownload.gi.ucsc.edu/hubs/gtexAnalysis/gtexAse.html" target="_blank">
GTEx Analysis Hub: Allele-Specific Expression</a> on <a href="
../cgi-bin/hgTracks?db=hg38&hubUrl=
http://hgdownload.gi.ucsc.edu/hubs/gtexAnalysis/gtexAnalysisHub.html" target="_blank">hg19</a> and 
<a href="../cgi-bin/hgTracks?db=hg38&hubUrl=http://hgdownload.gi.ucsc.edu/hubs/gtexAnalysis/gtexAnalysisHub.html" 
target="_blank">hg38</a>.</li>
</ul>

<p>
Please see the related October 11, 2017 publication in <i>Nature</i>,
&quot;<a href="https://www.nature.com/nature/journal/v550/n7675/full/nature24277.html" 
target="_blank">Genetic effects on gene expression across human tissues</a>&quot; for an in-depth 
article about these data.</p>

<p>
Acknowledgments to UCSC Genome Browser engineer, Kate Rosenbloom, for creating these tracks.
</p>

<a name="092917"></a>
<h2>Sep. 29, 2017 &nbsp;&nbsp; VAI tool can now output HGVS terms</h2>
<p>
The UCSC Genome Browser has <a href="help/query.html#HGVS" target="_blank">continued to expand</a>
the support for Human Genome Variation Society (HGVS) nomenclature. We have recently enhanced our
<a href="/cgi-bin/hgVai" target="_blank">Variant Annotation Integrator (VAI) tool</a> to allow 
output of HGVS terms from inputed variants when the gene prediction source is set to annotate on 
RefSeq genes.</p>
<p>
For a quick experience click the &quot;Get results&quot; button in this
<a href="http://genome.ucsc.edu/cgi-bin/hgVai?hgS_doOtherUser=submit&hgS_otherUserName=brianlee&hgS_otherUserSessionName=hg38.exampleVAI"
target="_blank">example session</a>. This session is loaded with an artifical input of variants
on the human hg38 assembly with NCBI RefSeq Genes track (curated NM_*, NR_*, and YP_* subset)
filtered for coding exons and splice site roles and on DNase regulatory elements with output
in HTML Variant Effect Predictor format where the &quot;Extra&quot; column will include
HGVS notation. Please note that a semi-colon &quot;;&quot; will separate results in
that field (HGVSG=NC_000010.11:g.27005592_27005594delAGA;
HGVSCN=NM_014915.2:c.5129_5131delTCT; EXON=34/34).</p>
<p>
Credit goes to Angie Hinrichs and Christopher Lee for the implementation and testing of
this feature.
</p>

<a name="092617"></a>
<h2>Sep. 26, 2017 &nbsp;&nbsp; dbSNP video released</h2>
<p>
Announcing the release of a new addition to our <a href =
"https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos">
YouTube</a> channel.</p>
<p>
Our new tutorial describes the <a href =
"https://youtu.be/d4XrkjnydN8?feature=player_detailpage&v=8ATcoDTOc0g&list=UUQnUJepyNOw0p8s2otX4RYQ">dbSNP
resources </a> in the Genome Browser, including display conventions and the
subdivision of the data into several useful subset tracks, especially the
Common SNPs. There is also a discussion about changes to the genome assemblies
from one version to another and of two ways to navigate between different
assemblies of the human genome in the Browser.

<p>Principal engineering on the dbSNP integration by Angie Hinrichs, quality
assurance by Cath Tyner and Christopher Lee, with other contributions from
the rest of the Genome Browser staff.  Video production by Robert Kuhn and
David Gibson.</p>
</p>

<a name="091917"></a>
<h2>Sep. 19, 2017 &nbsp;&nbsp; Data Integrator video released</h2>
<p>
We are pleased to announce the release of a new <a href =
"https://www.youtube.com/watch?feature=player_detailpage&v=8ATcoDTOc0g&list=UUQnUJepyNOw0p8s2otX4RYQ"
target = _blank>video</a> to the UCSC Genome Browser <a href =
"https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos"
target = _blank>YouTube channel</a>.  This video demonstrates the Data Integrator,
a tool that allows data from up to five primary table to be exported.  The tables
do not have to share a common data field, but are intersected using the
coordinate footprint of the annotations in the first selected dataset.</p>

<p>For example, even though the variants in the Common SNPs table share no
identifiers with the genes in the UCSC Genes track of the human genome assembly
hg19, it is possible to export a list of genes and all the Common SNPs that map
to the chosen set of genes.  This functionality extends the reach of data mining
beyond what is possible with the Table Browser.</p>

<p>Principal engineering on the Data Integrator by Angie Hinrichs, quality 
assurance by Matt Speir, with other contributions from the rest of the Genome 
Browser staff.  Video production by Robert Kuhn and David Gibson.</p>


<a name="091217"></a>
<h2>Sep. 12, 2017 &nbsp;&nbsp; GTEx Transcript Expression tracks now available for human</h2>
<p>
We are pleased to announce the release of the GTEx Transcript Expression track available on the two
most recent human assemblies, <a href="../cgi-bin/hgTrackUi?db=hg38&g=gtexTranscExpr"
target="_blank">GRCh38/hg38</a> and <a href="../cgi-bin/hgTrackUi?db=hg19&g=gtexTranscExpr"
target="_blank">GRCh37/hg19</a>. This track provides additional analysis of the same data as the
existing GTEx Gene track and displays median transcript expression levels in 53 tissues, based on
RNA-seq data from the GTEx midpoint milestone data release (V6, October 2015).
Data for this track were computed at UCSC from GTEx RNA-seq sequence data using the
<a href="http://toil.ucsc-cgl.org/" target="_blank">Toil</a> pipeline running the kallisto
transcript-level quantification tool.</p>
<p>
The data in the <a href="../cgi-bin/hgTrackUi?db=hg19&g=gtexTranscExpr"
target="_blank">GRCh37/hg19</a> version of this track was generated by coordinate converting the
<a href="../cgi-bin/hgTrackUi?db=hg38&g=gtexTranscExpr" target="_blank">GRCh38/hg38</a>
track data. Of the 189,615 BED entries from the original hg38 track, 176,220 were mapped
over by transcript name to hg19 using wgEncodeGencodeCompV24lift37 (~93% coverage).
</p>
<p>
Samples were collected by the GTEx Consortium. RNA-seq was performed by the GTEx Laboratory, Data
Analysis and Coordinating Center (LDACC) at the Broad Institute. John Vivian, Melissa Cline, and
Benedict Paten of the UCSC Computational Genomics lab were responsible for the sequence read
quantification used to produce this track. Thanks to Kate Rosenbloom, Chris Eisenhart, and Jairo
Navarro of the UCSC Genome Browser group for their work in creating and releasing these tracks.</p>

<a name="090717"></a>
<h2>Sep. 07, 2017 &nbsp;&nbsp; New Ensembl v89 gene tracks for 7 assemblies</h2>
<p>
The UCSC Genome Browser has released updated
<a href="http://www.ensembl.org/info/genome/genebuild/genome_annotation.html"
target="_blank">Ensembl gene annotation</a> tracks, from v86 to v89, for the 
following assemblies:</p>  
<ul>
<li>Zebrafish <a href="../cgi-bin/hgTrackUi?db=danRer10&g=ensGene" 
target="_blank">(danRer10)</a></li>
<li>Chicken <a href="../cgi-bin/hgTrackUi?db=galGal5&g=ensGene" 
target="_blank">(galGal5)</a></li>
<li>Mouse Lemur <a href="../cgi-bin/hgTrackUi?db=micMur2&g=ensGene" 
target="_blank">(micMur2)</a></li>
<li>Opossum <a href="../cgi-bin/hgTrackUi?db=monDom5&g=ensGene" 
target="_blank">(monDom5)</a></li>
<li>Platypus <a href="../cgi-bin/hgTrackUi?db=ornAna1&g=ensGene" 
target="_blank">(ornAna1</a> 
and <a href="../cgi-bin/hgTrackUi?db=ornAna2&g=ensGene" 
target="_blank">ornAna2)</a></li>
<li> Rat <a href="../cgi-bin/hgTrackUi?db=rn6&g=ensGene" 
target="_blank">(rn6)</a></li>
</ul>
</p>

<a name="082517"></a>
<h2>Aug. 23, 2017 &nbsp;&nbsp; New dbSNP build 150 tracks now available for hg19 and hg38</h2>
<p>
We are pleased to announce the release of our &quot;SNP&quot; tracks derived from
<a href="https://www.ncbi.nlm.nih.gov/SNP/" target="_blank"> NCBI&#39;s dbSNP Human Build 150 
data</a>, available on the two most recent human assemblies, 
<a href="../cgi-bin/hgTrackUi?db=hg38&g=snp150" target="_blank">GRCh38/hg38</a> and 
<a href="../cgi-bin/hgTrackUi?db=hg19&g=snp150" target="_blank">GRCh37/hg19</a>.
NCBI's dbSNP database is a collection which includes a range of molecular variation, such as single 
nucleotide polymorphisms and small insertions/deletions (indels). This release 
<a href="https://ncbiinsights.ncbi.nlm.nih.gov/2017/05/08/dbsnps-human-build-150-has-
doubled-the-amount-of-refsnp-records/" target="_blank">
nearly doubles</a> the number of SNP records compared to the previous 147 release (now over 300
million for hg38, and over 200 million for hg19). </p>

<p>
There are four SNP tracks available as part of this release:
<ul>
 <li>
All SNPs (150): containing all mappings of reference SNPs to the human assembly. The tracks below
are subsets of <i>All SNPs (150)</i>.</li>
<li>
Common SNPs (150): uniquely mapped variants that appear in at least 1% of the population or are
100% non-reference.</li>
<li>
Flagged SNPs (150): uniquely mapped variants, excluding Common SNPs, that have been flagged by dbSNP
as &quot;clinically associated.&quot;</li>
<li>
Mult. SNPs (150): variants that have been mapped to more than one genomic location. This track has
been shrinking over the course of the last few releases as dbSNP now excludes most SNPs whose
flanking sequences map to multiple locations in the genome.</li>
</ul>
<p>
By default, only the <i>Common SNPs (150)</i> are visible in the browser; other tracks must be made
visible using the track controls. <i>All SNPs (150)</i> tracks reside in the &quot;Variation&quot;
group category for the hg19 and hg38 browsers.</p>
<p>
These tracks were produced at UCSC by Angie Hinrichs and checked for quality by Cath Tyner. 
We'd like to thank the dbSNP group at NCBI for providing access to these data.</p>

<a name="081717"></a>
<h2>Aug. 17, 2017 &nbsp;&nbsp; New CRISPR track for Chinese hamster</h2>
<p>
A new <a href="../cgi-bin/hgTrackUi?db=criGri1&g=crispr" target="_blank">CRISPR/Cas9</a> track is 
now available for the Chinese hamster (C_griseus_v1.0/<a href="../cgi-bin/hgGateway?db=criGri1"
target="_blank">criGri1</a>) assembly.</p>
<p>
This track shows regions of the genome within 200 bp of transcribed regions and DNA sequences 
targetable by CRISPR RNA guides using the Cas9 enzyme from <i>S. pyogenes</i> (PAM: NGG). CRISPR target 
sites were annotated with predicted specificity (off-target effects) and predicted efficiency 
(on-target cleavage) by various algorithms using the <a href="http://crispor.tefor.net/"
target="_blank">CRISPOR</a> tool.</p>
<p>
The details page for each target contains a wealth of information related to the target, such as the
<a href="http://dx.doi.org/10.1038/nbt.2647" target="_blank">MIT Specificity score</a>, 
<a href="http://dx.doi.org/10.1038/nmeth.3543"
target="_blank">Moreno-Mateos Efficiency Score</a>, 
<a href="http://dx.doi.org/10.1038/nbt.3437"
target="_blank">Doench/Fusi Efficiency Score</a>, and any off-targets.</p> 
<p>
If you would like us to generate this track for another assembly, please notify us at
<a
href="mailto:&#103;e&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;&#115;c.&#101;d&#117;">&#103;e&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;&#115;c.&#101;d&#117;</a>.</p>
<!-- above address is genome at soe.ucsc.edu --> 
<p>
We would like to thank Hiram Clawson, Maximilian Haeussler, Christopher Lee, and Jairo Navarro for
the development and release of these tracks.</p>

<a name="071417"></a>
<h2>Jul. 14, 2017 &nbsp;&nbsp; New alternative mouse strain hub and strain-specific annotations</h2>
<p>
A mouse strain assembly hub is now available and shows an alignment of 16 assembled laboratory mice
strains, including the reference mouse (mm10) as well as the reference rat (rn6). This hub contains
some of the most common mice strains used in disease models such as NOD, BALB/c, and C3H. Each
mouse strain in the assembly hub is displayed as the primary sequence along with strain-specific
annotations. You can load this hub from our <a href="../cgi-bin/hgHubConnect"
target="_blank">Public Hubs</a> or the following
<a href="http://hgdownload.gi.ucsc.edu/hubs/mouseStrains/hubIndex.html" target="_blank">page</a>.
</p>
<p>
Along with the release of this assembly hub, a new track is available for mm10. The
<a href="../cgi-bin/hgTrackUi?db=mm10&g=strainSNPs" target="_blank">Mouse SNPs</a> track shows
single nucleotide variants between the different strains of mice and all SNP and indel calls are
relative to the reference mouse genome (GRCm38/mm10).</p>
<p>
Special thanks to Joel Armstrong, Ian Fiddles, and Benedict Paten for the alignment generation, as
well as Thomas Keane and the <a href="http://www.sanger.ac.uk/science/data/mouse-genomes-project"
target="_blank">Mouse Genomes Project</a> for providing the genome assemblies. Thanks to Hiram 
Clawson and Jairo Navarro for the development and release of this track and hub.</p>

<a name="2017-07-14"></a>
<h2>Jul. 14, 2017 &nbsp;&nbsp; New Video:  Multi-Region mode for viewing 
discontinuous regions in the Browser</h2>
<p>
Today we release a new <a href = "https://goo.gl/gQTcyR" target = _blank>video</a> 
showing how to use the enhanced Multi-Region display mode.  This feature allows 
the display of any regions of the genome together on the same screen.  A recent 
enhancement of the feature allows users to load their coordinates directly into 
a text box, bypassing the previous requirement to use a URL to upload coordinates.
</p>
<p>
This completes a trilogy of videos demonstrating the Multi-Region mode.  Previous 
episodes showed how to use <a href = "https://goo.gl/n8szK2" target = _blank>Exon-Only 
mode</a> and how to view <a href = "https://goo.gl/99tX3Y" target = _blank>alternate 
haplotypes</a> in their chromosomal context.
</p>
<p>
This video, like the other content in this series, is annotated in
our <a href = "../training/vids/" target = _blank>training pages</a> with an 
index showing the steps, with links directly to the corresponding places in the 
video.  There is also a <a 
href = "http://genome.ucsc.edu/training/vids/transcript11.pdf" 
target = _blank>transcript</a>.
</p>
<p>
The videos were produced at UCSC by Robert Kuhn and David Gibson.
The Multi-Region engineering was by Galt Barber.
</p>

<a name="062217"></a>
<h2>Jun. 22, 2017 &nbsp;&nbsp; Visualize gene pathways and interactions</h2>
<p>
Individual genes do not exist in a vacuum and their expression, activation, and function
are often dependent on their interactions with other genes, frequently as part of pathways or 
complexes. The new &quot;Gene Interactions&quot; tool combines data from a number of 
<a href="help/hgGeneGraph.html#methods" target="_blank">curated interaction and
pathway databases</a> as well as interactions mined from over 20 million PubMed abstracts
through the <a href="http://literome.azurewebsites.net/" target="_blank">Literome</a> project.
<p><img class='text-center' src="../../images/hgGeneGraph.png" alt="Gene Interaction Graph"
width="35%" height="35%"></p>

<p>
This new tool displays a detailed pathway and interaction graph for a selected gene, marked
in yellow. This primary gene is connected by lines to other genes, with those displayed
representing the interactions with the most support from both text mining and databases.
You can hover over any gene in the image to see more details about its
product; you can hover over the connecting lines to see more detailed information about the data
supporting a particular interaction. Clicking on one of these interaction lines brings you to a
page that provides links to supporting information. Interactions can be filtered based on
various support levels (both text-mined and databases, all databases, curated-only databases);
and genes can be annotated and colored according to a variety of properties (GNF2 average
expression, DrugBank and more.) You can read more about this tool and its features on the 
<a href="help/hgGeneGraph.html" target="_blank">help page</a>.</p>

<p>
This new interaction graph can be accessed from the &quot;Tools&quot; menu or through the
&quot;Gene Interactions&quot; track for the human assemblies
<a href="../cgi-bin/hgTrackUi?db=hg38&g=interactions" target="_blank">hg38</a> or
<a href="../cgi-bin/hgTrackUi?db=hg19&g=interactions" target="_blank">hg19</a>.
Note that while this tool can be accessed from the &quot;Tools&quot; menu while browsing any
genome assembly, the data will always be based on human sources.</p>

<p>
A huge thank you to the many
<a href="help/hgGeneGraph.html#credits" target="_blank">collaborators and contributors</a>
who helped by providing their databases and expertise, notably Hoifung Poon and Chris Quirk at 
Microsoft Research who provided text mining data. Thank you to Max Haeussler and Matthew Speir at 
the UCSC Genome Browser for all of their efforts in the creation and release of these new resources.
</p>

<a name="061317"></a>
<h2>Jun. 13, 2017 &nbsp;&nbsp; New barChart track format</h2>
<p>
Are you loving the 
<a href="../../cgi-bin/hgTracks?db=hg38&hideTracks=1&gtexGene=pack&position=chr9%3A133255602-133275214">
GTEx track</a> and want to make a similar barChart display track for yourself? Are you wondering
how you can get boxplots of data points to show up on track item 
<a href="../../cgi-bin/hgc?db=hg38&c=chr9&l=133255601&r=133275214&o=133257408&t=133275201&g=gtexGene&i=ABO">
details pages</a>? Well, look no
further as we have just released support for a GTEx style 
<a href="help/barChart.html">barChart track format</a>. This new track format is
available for use with custom tracks, and in binary format (bigBarChart) for use in track hubs. 
Both barChart and bigBarChart formats support external matrix and sample files specifying data 
points for a boxplot display on details pages, while the barChart/bigBarChart file typically 
specifies summary values (mean or median) over the matrix, which determines the bar heights. See 
the <a href="help/barChart.html">documentation</a> for further explanation and examples. 
</p>

<p>
Along with the release of this new track format are some tools that help to create the tracks,
<code>expMatrixToBarchartBed</code> and <code>bedJoinTabOffset</code>. Both utilities are free for
all use cases, and can be downloaded from our <a href="http://hgdownload.gi.ucsc.edu/admin/exe">
directory of utilities</a>. For more information about this new track format and helper scripts, 
please see the <a href="help/barChart.html">help documentation</a>.
</p>

<p>
Thanks to Kate Rosenbloom, Christopher Eisenhart and Christopher Lee for the development
and testing of this new track format. 
</p>

<a name="060817"></a>
<h2>Jun. 8, 2017 &nbsp;&nbsp; New command line tool for annotating variants, vai.pl</h2>
<p>
We are pleased to announce the release of a new command-line tool for annotating variants, 
<code>vai.pl</code>. This program is similar to our web-based 
<a href="../../cgi-bin/hgVai">Variant Annotation Integrator (VAI)</a>, except it allows for bypassing 
the variant limit imposed by web-based VAI, and does not require connections to UCSC, allowing 
annotation of private clinical data.</p>

<p>
Along with the release of this program, we have published a 
<a href="https://genome-blog.gi.ucsc.edu/blog/annotating-millions-of-private-variants-with-vai-pl/">
new blog post</a> with some background on both web-based and command-line VAI, as well 
as some example usage to serve users. <code>vai.pl</code> is available for download from the
<a href="https://genome-store.ucsc.edu">genome-browser store</a>, via purchase of the GBiB, GBiC,
or full source.
</p>

<p>
Thanks to Angie Hinrichs and Christopher Lee for the development and testing of the program.</p>

<a name="060617"></a>
<h2>Jun. 6, 2017 &nbsp;&nbsp; New genome browser available for ciona intestinalis</h2>
<p>
A genome browser is now available for the ciona intestinalis(<em>vase tunicate</em>)
assembly released in April 2011 by The Graduate School of Science, Kyoto University. 
For more information and statistics
about this assembly, see the NCBI assembly record for
<a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000224145.1/" target="_blank">
Ciona intestinalis</a>.
There are 6,381 contigs and 1,272 scaffolds in this assembly with a total size of
115,227,500 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/ci3/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#ciona">Downloads</a> page.
Please observe the <a href="credits.html#ciona">conditions for use</a> when
accessing and using these data sets. The annotation tracks for this browser were
generated by UCSC and collaborators worldwide. See the
<a href="credits.html#ciona_credits">Credits</a> page for a detailed list of the
organizations and individuals who contributed to this release.</p>
<a name="060217"></a>
<h2>Jun. 2, 2017 &nbsp;&nbsp; New default tracks for human and mouse assemblies</h2>
<p>
We have made several changes to the sets of default tracks for some of our human and
mouse assemblies. The
<a href="https://genome-blog.gi.ucsc.edu/blog/gtex-resources-in-the-browser/" target="_blank">GTEx
Gene Expression</a> track is now part of the set of default tracks for both human genome assemblies
hg38 and hg19. In addition, we have added the
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=dgvPlus">DGV Struct Var</a> track, which displays large
structural variant such as CNVs, inversions and in/dels, to the default track set for
the human genome assembly hg19. Lastly, for the mouse genome assembly mm10, we have added the
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=cons60way">60-way Vertebrate Conservation</a> track to
its set of default tracks.</p>

<a name="051617"></a>
<h2>May 16, 2017 &nbsp;&nbsp; New genome browser available for golden eagle</h2>
<p>
A genome browser is now available for the golden eagle (<em>Aquila chrysaetos canadensis</em>)
assembly released in October 2014 by Washington University. For more information and statistics
about this assembly, see the NCBI assembly record for
<a href="https://www.ncbi.nlm.nih.gov/assembly/GCA_000766835.1/" target="_blank">Aquila_chrysaetos-1.0.2</a>.
There are 17,032 contigs and 1,142 scaffolds in this assembly with a total size of
1,192,743,076 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/aquChr2/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#golden_eagle">Downloads</a> page.
Please observe the <a href="credits.html#golden_eagle_use">conditions for use</a> when
accessing and using these data sets. The annotation tracks for this browser were
generated by UCSC and collaborators worldwide. See the
<a href="credits.html#golden_eagle_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>

<a name="051517"></a>
<h2>May 15, 2017 &nbsp;&nbsp; New TransMap annotation tracks released</h2>
<p>We are pleased to announce the release of a new set of TransMap V4 tracks for nearly 70
different assemblies.</p>

<p> The TransMap tracks are an attempt to map features, such as mRNAs, ESTs,and gene predictions
from one organism to another through pairwise, whole-genome alignments. These new TransMap tracks
are:
<ul>
  <li>Ensembl Mappings</li>
  <li>RefSeq Gene Mappings</li>
  <li>GenBank RNA Mappings</li>
  <li>EST Mappings</li>
</ul>

<p>You can read more about how these tracks were created on the
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=transMapV4">track description page</a> for the
hg38/GRCh38 version of these tracks.</p>

<p>Thanks to Mark Diekhans and Matthew Speir for their work in creating and releasing these tracks.</p>

<a name="050417"></a>
<h2>May 4, 2017 &nbsp;&nbsp; New BLAT custom track feature</h2>
<p>We are excited to announce the release of a long awaited feature: the ability to save  BLAT 
search results as a Genome Browser custom track.</p>

<p>Now, after running a BLAT search, you can save your results as a 
<a href="help/bigPsl.html">bigPsl</a> custom track with the click of a button. You can even 
configure the name and description of the track to your liking!</p>

<p>These custom tracks display and function exactly the same as BLAT results (with a slight 
coloring difference for amino acid searches), with the advantage that they will persist 
when saved in a session, just like any other custom track. That means you can now send your 
colleagues a session of your BLAT searches and they will still be able to see them weeks later!</p>

<p>Thanks to Brian Raney and Christopher Lee for the implementation and release of this feature.</p>

<a name="050317"></a>
<h2>May 3, 2017 &nbsp;&nbsp; Highlighting feature enhanced for genome browsers</h2>
<p>
In <a href="../goldenPath/newsarch.html#030414" target="_blank">
March of 2014, a new highlight feature was released</a>, allowing an option to 
drag-and-select a single highlight color in genome browsers. This highlighting feature 
has been enhanced with new additions, as displayed on the &quot;Drag-and-select&quot; menu that    
appears when conducting &quot;shift+drag&quot; or &quot;click-drag&quot; within the Genome Browser 
(for example, when zooming into a region). </p>
<p>
Main features in drag-and-select menu:</p>
<ul>
  <li>
  Use shift+drag or click-drag to enable the new &quot;Drag-and-select&quot; dialog box menu.</li>
  <li>
  In the menu, a checkbox controls behavior for drag-and-select; you can hide the menu and always 
  zoom with shift+select. If selected, re-enable via 'View - Configure Browser' (keyboard shortcut
  : c then f).</li>
  <li>
  A &quot;color picker&quot; option allows for easy color selection of each highlight; you can also 
  create multiple highlights (each with various colors if desired).</li>
  <li>
  Hold Shift+drag to show the menu (or click+drag).</li>
  <li>
  Hold Alt+drag to add a highlight (without displaying the menu).</li>
  <li>
  Hold Ctrl+drag (Windows) or Cmd+drag (Mac) to zoom (without displaying the menu).</li>
  <li>
  To cancel, press Esc anytime or drag mouse outside image.</li>
  <li>
  Highlight the current position with the keyboard shortcut &quot;h then m.&quot;</li>
  <li>
  Clear all highlights with View - Clear Highlights (or keyboard shortcut h then c), or simply 
  right-click on a highlight to remove all highlights.</li>
</ul>
<p>
In the Genome Browser, there are also options for right-clicking:</p>
<ul>
  <li>
  Remove highlighting</li>
  <li>
  Zoom in to a highlighted region</li>
  </li>
  <li>
  Highlight a gene - right-click on the gene (e.g., SOD1) and select
  &quot;Highlight SOD1&quot;</li>
</ul>

<img src="../images/highlightScreenshot2017.png" alt="Highlight screenshot" width="899"
height="602"></p>

<a name="042117"></a>
<h2>Apr. 21, 2017 &nbsp;&nbsp; New Genome Browser available for Chinese pangolin</h2>
<p>
A Genome Browser is now available for the Chinese pangolin (<em>Manis pentadactyla</em>) assembly 
released in August 2014 by Washington University. For more information and statistics about this
assembly, see the NCBI assembly record for
<a href="https://www.ncbi.nlm.nih.gov/assembly/GCA_000738955.1/" target="_blank">M_pentadactyla-1.1.1</a>.
There are 230,930 contigs and 92,772 scaffolds in this assembly with a total size of
2,204,741,241 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/manPen1/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#chinese_pangolin">Downloads</a> page.
Please observe the <a href="credits.html#chinese_pangolin_credits">conditions for use</a> when
accessing and using these data sets. The annotation tracks for this browser were
generated by UCSC and collaborators worldwide. See the
<a href="credits.html#chinese_pangolin_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>


<a name="042017"></a>
<h2>Apr. 20, 2017 &nbsp;&nbsp; New Genome Browsers available for Tibetan frog and American Bison</h2>
<p>
A Genome Browser is now available for the Tibetan frog (<em>Nanorana parkeri</em>) assembly released
in March 2015 by the Bejing Genomics Institute. For more information and statistics about this
assembly, see the NCBI assembly record for
<a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000935625.1/" target="_blank">BGI_ZX_2015</a>.
There are 147,513 contigs and 25,188 scaffolds in this assembly with a total size of
2,053,867,363 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/nanPar1/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#tibetan_frog">Downloads</a> page.
Please observe the <a href="credits.html#tibetan_frog_use">conditions for use</a> when
accessing and using these data sets. The annotation tracks for this browser were
generated by UCSC and collaborators worldwide. See the
<a href="credits.html#tibetan_frog_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>
<hr>
<p>
A Genome Browser is now available for the American bison (<em>Bison bison bison</em>) assembly
released in October 2014 by U.Maryland.
For more information and statistics about this assembly, see the NCBI assembly record
for <a href="https://www.ncbi.nlm.nih.gov/genome/8907" target="_blank">Bison_UMD1.0</a>.
There are 470,415 contigs and 128,431 scaffolds in this assembly
with a total size of 2,828,031,685 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/bisBis1/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#bison">Downloads</a> page.
Please observe the <a href="credits.html#bison_credits">conditions for use</a> when
accessing and using these data sets. The annotation tracks for this browser were
generated by UCSC and collaborators worldwide. See the
<a href="credits.html#bison_credits">Credits</a> page for a detailed list of the
organizations and individuals who contributed to this release.</p>

<a name="032017"></a>
<h2>Mar. 20, 2017 &nbsp;&nbsp; 20 Species Conservation Track now available on RGSC 6.0/rn6</h2>
<p>
We are pleased to announce the release of a new Conservation track based on the rat
(RGSC 6.0/rn6) assembly. This track shows multiple alignments of 20 vertebrate species and
measurements of evolutionary conservation using two methods (phastCons and phyloP) from the
PHAST package for all species. The multiple alignments were generated using multiz and other
tools in the UCSC/Penn State Bioinformatics comparative genomics alignment pipeline. Conserved 
elements identified by phastCons are also displayed in this track. For more details, visit the
<a href="../cgi-bin/hgTrackUi?db=rn6&g=cons20way">track description page</a>. </p>
<p>
We would like to thank Hiram Clawson, Chris Lee and Jairo Navarro for their efforts in creating this
track. </p>

<a name="031517b"></a>
<h2>Mar. 15, 2017 &nbsp;&nbsp; Version 2 of Track Database Definitions released</h2>
<p>
We're happy to announce the recent release of Version 2 of our
<a href="help/trackDb/trackDbHub.html">Track Database Definition Document</a>. This document
includes all of the currently supported track configuration options (trackDb) for
track and assembly hubs in the UCSC Genome Browser.</p>
<p>
This latest version includes settings for our latest &quot;big&quot; track types:
bigMaf, bigChain, bigPsl, and bigGenePred, as well as the &quot;CRAM&quot; track type.
This newest version also marks the movement of some trackDb settings (html, priority,
colorByStrand, autoScale, spectrum) from the &quot;full&quot; to the &quot;base&quot; support
level to indicate that they are supported by other genome browsers. At this time, it is
primarily Ensembl that supports the use of these new &quot;base&quot; settings in track hubs
on their site.</p>
<p>
Additionally, we have published a
<a href="https://genome-blog.gi.ucsc.edu/blog/how-portable-is-your-track-hub-use-hubcheck-to-find-out/">new 
blog post</a> that discusses the versioning of these trackDb settings and how you can use
hubCheck to see if your hubs are compatible with other genome browsers. We have also added
examples to our <a href="help/hgTrackHubHelp.html#Compatibility">help documentation</a> that show 
how to use hubCheck to test track hub compatibility.</p> 

<a name="031517a"></a>
<h2>Mar. 15, 2017 &nbsp;&nbsp; New Genome Browsers available for
Golden Snub-Nosed Monkey, Proboscis Monkey, and Turkey</h2>
<p>
A Genome Browser is now available for the Golden Snub-Nosed Monkey (<em>Rhinopithecus 
roxellana</em>) assembly released in October 2014 by Novogene.
For more information and statistics about this assembly, see the NCBI assembly record
for <a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000769185.1/" target="_blank">Rrox_v1</a>.
There are 196,797 contigs and 135,512 scaffolds in this assembly
with a total size of 2,899,552,139 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/rhiRox1/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#golden_snub-nosed_monkey">Downloads</a> page.
Please observe the <a href="credits.html#golden_snub-nosed_monkey_use">conditions for use</a> when
accessing and using these data sets. The annotation tracks for this browser were
generated by UCSC and collaborators worldwide. See the
<a href="credits.html#golden_snub-nosed_monkey_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>
<hr>
<p>
A Genome Browser is now available for the Proboscis Monkey (<em>Nasalis larvatus</em>)
assembly released in November 2014 by the Proboscis Monkey Functional Genome Consortium.
For more information and statistics about this assembly, see the NCBI assembly record
for <a href="https://www.ncbi.nlm.nih.gov/assembly/GCA_000772465.1/" target="_blank">Charlie1.0</a>.
There are 553,412 contigs and 319,549 scaffolds in this assembly
with a total size of 3,011,966,170 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/nasLar1/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#proboscis_monkey">Downloads</a> page.
Please observe the <a href="credits.html#proboscis_monkey_use">conditions for use</a> when
accessing and using these data sets. The annotation tracks for this browser were
generated by UCSC and collaborators worldwide. See the
<a href="credits.html#proboscis_monkey_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>
<hr>
<p>
A Genome Browser is now available for the Turkey (<em>Meleagris gallopavo</em>)
assembly released in November 2014 by the Turkey Genome Consortium.
For more information and statistics about this assembly, see the NCBI assembly record
for <a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000146605.2/" target="_blank">Turkey_5.0</a>.
There are 296,331 contigs and 233,806 scaffolds in this assembly
with a total size of 1,128,339,136 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/melGal5/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#Turkey">Downloads</a> page.
Please observe the <a href="credits.html#Turkey">conditions for use</a> when
accessing and using these data sets.  The annotation tracks for this browser were
generated by UCSC and collaborators worldwide. See the
<a href="credits.html#Turkey">Credits</a> page for a detailed list of the organizations and
individuals who contributed to this release.</p>

<a name="030317"></a>
<h2>Mar. 3, 2017 &nbsp;&nbsp; New NCBI RefSeq tracks released</h2>
<p>
We are pleased to announce the release of a new set of gene annotation tracks
for the hg38/GRCh38 assembly based on data from NCBI's Reference Sequence Database (RefSeq).
For all of these tracks, the alignments and coordinates are provided by RefSeq.
These tracks are organized in a composite track that includes:</p>
<ul>
  <li>RefSeq All &ndash; all annotations from the curated and predicted sets
  <li>RefSeq Curated &ndash; curated annotations beginning with NM, NR, or YP
  <li>RefSeq Predicted &ndash; predicted annotations beginning with XM or XR
  <li>RefSeq Other &ndash; all other RefSeq annotations not included in RefSeq All
  <li>RefSeq Alignments &ndash; alignments of transcripts to the genome provided by RefSeq
</ul>
<p>
The new composite track also includes a &quot;UCSC RefSeq&quot; track that is based on our original
&quot;RefSeq Genes&quot; track. As with before, this UCSC track is a result of our realignments
of RefSeq RNAs to the genome, which means that there may be some cases where the annotations
differ from those in the new NCBI RefSeq tracks.</p>
<p>
More details about these tracks, their construction, and how to access their underlying data
can be found in 
<a href="https://genome-blog.gi.ucsc.edu/blog/the-new-ncbi-refseq-tracks-and-you/"
target="_blank">this blog</a>
or on the <a href="../../cgi-bin/hgTrackUi?db=hg38&g=refSeqComposite">track description page</a>.
</p>
<p>
A huge thank you to Terence Murphy from the RefSeq group and to Hiram Clawson, Brian Raney,
Mark Diekhans, Matthew Speir and many others from the UCSC Genome Browser staff for
bringing this track to life.</p>

<a name="021617"></a>
<h2>Feb. 13, 2017 &nbsp;&nbsp; New keyboard shortcuts for browser zoom views</h2>
<p>
Please welcome the newest additions to the UCSC Genome Browser keyboard shortcuts menu, 
now providing over 40 shortcuts. To view a menu of shortcuts, simply press the &quot;?&quot; 
key on your keyboard while on an assembly&apos;s browser display page (e.g., 
<a href="../cgi-bin/hgTracks?db=hg38" target="_blank">GRCh38/hg38 assembly display</a>). 
The new shortcut set utilizes the keyboard keys 1 - 6 to provide six levels of zoom views 
for tracks in the browser display, as listed below:</p>
<ul>
  <li>Press &quot;1&quot; to zoom to 50 bp (1 zero)</li>
  <li>Press &quot;2&quot; to zoom to 500 bp (2 zeros)</li>
  <li>Press &quot;3&quot; to zoom to 5000 bp (3 zeros)</li>
  <li>Press &quot;4&quot; to zoom to 50 kbp (4 zeros)</li>
  <li>Press &quot;5&quot; to zoom to 500 kbp (5 zeros)</li>
  <li>Press &quot;6&quot; to zoom to 5 Mbp (6 zeros)</li>
</ul>
<p>
Credit: Michael Hiller for initial development, Max Haeussler for browser implementation.</p>

<a name="012317"></a>
<h2>Jan. 23, 2017 &nbsp;&nbsp; New Genome Browser product gives freedom to easily collaborate in the
cloud</h2>
<p>
Until now, genomics research groups working with sensitive medical data were largely limited to 
using local Genome Browser installations to maintain confidentiality, complicating data-sharing 
among collaborators. Today, the Genome Browser group of the UC Santa Cruz Genomics Institute 
announced they have changed that by launching a new product, Genome Browser in the Cloud (GBiC). 
GBiC introduces new freedom to collaborate by allowing rapid Browser installation, in any 
UNIX-based cloud.</p>
<p>
Users provide the cloud instance, then install the Genome Browser image and grant access to 
whomever needs it. GBiC functions the same and is as secure as the public version of the Genome 
Browser, Genome Browser in a Box (GBiB), or a Genome Browser mirror site. Another GBiC innovation 
is significantly reduced installation time as compared to earlier Genome Browser versions.</p>
<p>
&quot;We are very pleased with how this product facilitates remote collaboration &mdash; for 
example, between a hospital physician, an off-site lab technician and a third-party genomic 
researcher,&quot; said Genome Browser author and Principal Investigator Jim Kent. 
&quot;Thanks to the efforts of GBiC Engineer Max Haeussler, users also benefit from significantly 
faster installation time,&quot; Kent continued. &quot;What historically took at least a week, now 
typically is less than an hour,&quot; he said.</p>
<p>
While the GBiC is intended specifically for cloud-based installations, its functionality is 
versatile. For most purposes, the GBiC essentially replaces the manual installation process for 
mirroring the UCSC Genome Browser in multiple environments (cloud servers, dedicated servers, or 
even a laptop).</p>
<p>
GBiC is free for non-profit academic research and for personal use. Corporate use requires a 
license, a one-time setup fee of $2,000 and annual payment of $1,000 per instance. GBiC is 
available exclusively at the UCSC Genome Browser 
<a href="https://genome-store.ucsc.edu" target="_blank">store</a>.</p>
<p>
For an installation demonstration, see the <a href="https://www.youtube.com/watch?v=dcJERBVnjio"
target="_blank">Genome Browser in the Cloud (GBiC) Introduction</a> video.</p>

<!-- ============= 2016 archived news ============= -->
<a name="2016"></a>

<a name="121316"></a>
<h2>Dec. 13, 2016 &nbsp;&nbsp; New Tissue Body Map track settings page for GTEx tracks</h2>
<p>
We are pleased to announce the release of a new graphic-based 
<a href="../cgi-bin/hgGtexTrackSettings?&amp;g=gtexGene" 
target="_blank">&quot;Tissue Body Map&quot; configuration page</a> for the GTEx Gene 
Expression track. This new page allows users to view GTEx-sampled tissues in an 
anatomical context and select tissues from the anatomy graphic as an alternative to 
using the alphabetical tissue list.<br /><br />This new track settings page contains 
its original functionality, but uses a new interface for track configuration. Next to 
a color-coded checkbox list of the 53 tissues is an interactive illustration of a 
human anatomical "body map." Hovering over a tissue in the list will highlight the 
corresponding anatomical region in the body map image. To show or hide tissues in the 
GTEx Gene Expression track, tissue selections can be made by either selecting tissues 
in the list or by clicking on the tissue labels in the body map.</p>
<p>
To quickly see which tissues are selected, right-click the bar graph displayed in 
the browser for the GTEx track and then click the wrench icon to go to the 
&quot;Configure GTEx&quot; page. To navigate to the 
<a href="../cgi-bin/hgGtexTrackSettings?&g=gtexGene" target="_blank">new track settings page</a> 
from the <a href="../" target="_blank">UCSC Genome Browser</a>, go to the 
<a href="../cgi-bin/hgGateway?db=hg38" target="_blank">hg38</a> 
or <a href="../cgi-bin/hgGateway?db=hg19" target="_blank">hg19</a> 
browser and click on the track label "GTEx" in the Expression group.<br /><br />In the 
coming months, we plan to incorporate the Tissue Body Map into other GTEx resources.</p>
<!-- <img src="../images/gtexBodyMap.svg" svg width="500" svg height="650"> -->

<a name="120716"></a>
<h2>Dec. 7, 2016 &nbsp;&nbsp;New video: Substitute alternate haplotypes into hg38 chromosome</h2>
<p>
The Genome Reference Consortium has released many alternate haplotypes as part of the most recent 
human genome assembly, hg38 (GRCh38). Our new <a href="http://bit.ly/ucscAlts" 
target="_blank">video tutorial</a>, shows how to use the Genome Browser's multi-region viewing mode 
to substitute these sequences, along with their annotations, into the main chromosomes of
this assembly, as well as other genome assemblies with alternate haplotypes. The video was produced 
by Robert Kuhn and David Gibson.</p> 
<p> 
See our <a href="../training/vids">video page</a> for a full list of our video offerings.</p>

<a name="110716"></a>
<h2>Nov. 7, 2016 &nbsp;&nbsp; New CRISPR track for many assemblies</h2>
<p>
A new CRISPR/Cas9 track has been released for the human (hg38, hg19), mouse (mm10), rat (rn5), 
zebrafish (danRer7), C. elegans (ce10), D. melanogaster (dm6), yeast (sacCer3) and C. intestinalis 
(ci2) assemblies.</p>
<p>
This track shows regions of the genome within 200bp of transcribed regions and DNA sequences 
targetable by CRISPR RNA guides using the Cas9 enzyme from <i>S. pyogenes</i> (PAM: NGG). CRISPR target 
sites were annotated with predicted specificity (off-target effects) and predicted efficiency 
(on-target cleavage) by various algorithms using the tool 
<a href="http://crispor.org">crispor.org</a>.</p>
<p>
The details page for each target contains a wealth of information related to the target, 
such as the <a href="http://dx.doi.org/10.1038/nbt.2647" target="_blank">MIT Specificity score</a>, 
<a href="http://www.nature.com/nmeth/journal/v12/n10/full/nmeth.3543.html"
target="_blank">Moreno-Mateos Efficiency Score</a>, 
<a href="http://portals.broadinstitute.org/gpp/public/analysis-tools/sgrna-design"
target="_blank">Doench/Fusi Efficiency Score</a>, and any off-targets.</p> 
<p>
If you would like us to generate this track for another assembly, please notify us at
<a
href="mailto:&#103;e&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;&#115;c.&#101;d&#117;">&#103;e&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;&#115;c.&#101;d&#117;</a>.</p>
<!-- above address is genome at soe.ucsc.edu --> 
<p>
The tracks were produced at UCSC by Maximilian Haeussler and Christopher Lee.</p>

<a name="110316"></a>
<h2>Nov. 3, 2016 &nbsp;&nbsp; New chromosome aliases search support</h2>
<p> 
Have you ever wanted to use 1 instead of chr1 to search for a chromosome position?</p>
<p>
The hg38 assembly now supports a new feature of matching different chromosome aliases. Equivalent 
entries only exist in the new table, chromAlias, when an exact sequence match has been verified. 
This feature is limited to searching coordinates, it is not yet supported for custom tracks.</p>
<p>
For example, aliases that now map to chr1 are NC_000001.11, CM000663.2, and 1, so that a hg38 search
such as CM000663.2 16,560,000 16,615,000 will be equivalent to chr1 16,560,000 16,615,000.</p>

<a name="102416"></a>
<h2>Oct. 24, 2016 &nbsp;&nbsp; Genome Browsers for chimpanzee, green monkey and gorilla now available</h2>
<p>
A Genome Browser is now available for the Chimp (<em>Pan troglodytes</em>)
assembly released in May 2016 by the Chimpanzee Sequencing and Analysis Consortium.
For more information and statistics about this assembly, see the NCBI assembly record
for <a href="https://www.ncbi.nlm.nih.gov/assembly/733711" target="_blank">Pan_tro 3.0</a>.
There are 26 different chromosomes along with 72,326 contigs and 45,511 scaffolds in this assembly
with a total size of 3,231,170,666 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/panTro5/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#chimp">Downloads</a> page.
Please observe the <a href="credits.html#chimp_use">conditions for use</a> when
accessing and using these data sets.  The annotation tracks for this browser were
generated by UCSC and collaborators worldwide. See the 
<a href="credits.html#chimp_credits">Credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p>
<p>
A Genome Browser is now available for the Green Monkey (<em>Chlorocebus sabaeus</em>)
assembly released in March 2014 by the Vervet Genomics Consortium.
For more information and statistics about this assembly, see the NCBI assembly record for 
<a href="https://www.ncbi.nlm.nih.gov/assembly/132581" target="_blank">Chlorocebus_sabeus 1.1</a>.
There are 32 different chromosomes along with 162,724 contigs and 2,022 scaffolds in this assembly
with a total size of 2,789,656,328 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/chlSab2/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#green_monkey">Downloads</a> page.
Please observe the <a href="credits.html#green_monkey_use">conditions for use</a> when
accessing and using these data sets.  The annotation tracks for this browser were
generated by UCSC and collaborators worldwide.  See the 
<a href="credits.html#green_monkey_credits">Credits</a> page for a detailed list of the
organizations and individuals who contributed to this release.</p>
<p>
A Genome Browser is now available for the Gorilla (<em>Gorilla gorilla gorilla</em>)
assembly released in March 2016 by the University of Washington.
For more information and statistics about this assembly, see the NCBI assembly record
for <a href="https://www.ncbi.nlm.nih.gov/assembly/705391" target="_blank">GSMRT3</a>.
There are 15,997 contigs, Contig N50 of 10,016,017 in this assembly
with a total size of 3,080,414,926 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/gorGor5/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#gorilla">Downloads</a> page.
Please observe the <a href="credits.html#gorilla_use">conditions for use</a> when accessing and 
using these data sets. The annotation tracks for this browser were generated by UCSC and 
collaborators worldwide. See the <a href="credits.html#gorilla_credits">Credits</a> page for a 
detailed list of the organizations and individuals who contributed to this release.</p>

<a name="101716"></a>
<h2>Oct. 17, 2016 &nbsp;&nbsp; HGVS nomenclature now supported in the Genome Browser</h2>
<p>
We are pleased to announce the first release of our support for 
<a href="http://www.hgvs.org/" target="blank">Human Genome Variation Society</a> (HGVS) 
nomenclature in the Genome Browser. Navigate directly to individual nucleotides or codons using
coordinates in the following formats:</p>
<ul>
  <li>NM_198576.3(AGRN):c.1057C>T</li>
  <li>NM_198056.2:c.1654G>T</li>
  <li>NP_002993.1:p.Asp92Glu</li>
  <li>NP_002993.1:p.D92E</li>
  <li>BRCA1 Ala744Cys</li>
</ul>
<p>
The gene-name format uses RefSeq genes to identify the codon. In the next 
release we will add support for nomenclature describing coding (c.) 
intron and UTR coordinates as well as insertions, deletions and duplications.  </p>

<a name="083116"></a>
<h2>Aug. 31, 2016 &nbsp;&nbsp; Genome Browser for chicken now available</h2>
<p>
A Genome Browser is now available for the Chicken (<em>Gallus gallus</em>)
assembly released December 2015 by the International Chicken Genome Consortium (ICGC).
For more information and statistics about this assembly, see the NCBI assembly record
for <a href="https://www.ncbi.nlm.nih.gov/assembly/595851" target="_blank">Gallus_gallus-5.0</a>.
There are 36 different chromosomes along with 23,439 contigs/scaffolds in this assembly
with a total size of 1,230,258,557 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/galGal5/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#chicken">Downloads</a> page.
Please observe the <a href="credits.html#chicken_use">conditions for use</a> when
accessing and using these data sets.  The annotation tracks for this browser were
generated by UCSC and collaborators worldwide.  See the <a
href="credits.html#chicken_credits">Credits</a> page for a detailed list of the
organizations and individuals who contributed to this release.</p>

<a name="083016"></a>
<p>
<h2>Aug. 30, 2016 &nbsp;&nbsp; Allele-specific expression in 53 tissues from GTEx Analysis Hub now
available</h2>
<p>
We are pleased to announce the release of Allele Specific Expression (ASE) data for the human hg19 
and hg38 assemblies. ASE data from the <a href="http://tllab.org" target="_blank">Lappalainen 
Lab</a> at the New York Genome Center is now available in the UCSC Genome Browser as a public track 
hub.  This track hub contains ASE data identified from transcriptome and genotype
data in 53 tissues collected by the <a href="https://commonfund.nih.gov/GTEx/index" 
target="blank">Genotype-Tissue Expression (GTEx) project</a> and analyzed by the Lappalainen Lab. 
The hub contains 3 tracks, a cross tissue summary via density graph of median allelic imbalance, 
a summary track of all SNPs with evidence of ASE in any tissue, and a composite track showing ASE 
on a tissue by tissue basis. In the composite track each subtrack is colored based on median ASE for
the site across all samples of the tissue, shaded on a spectrum of gray (low) to GTEx convention
tissue color (high).</p>
<p>
To access and view this hub, navigate to the <a href="../cgi-bin/hgHubConnect" 
target="_blank">Track Hub gateway page</a> and select &quot;GTEx Analysis Hub&quot; from the
Public Hubs list. Please direct any queries to <a href="mailto:scastel@nygenome.org">Stephane 
Castel</a>.</p>
<p>
Many thanks to Stephane Castel and the Lappalainen Lab at the NY Genome Center for providing this 
data.</p>

<a name="081616"></a>
<h2>Aug. 16, 2016 &nbsp;&nbsp; dbSNP 147 available for hg19 and hg38</h2>
<p>
We are pleased to announce the release of four tracks derived from NCBI 
<a href="https://www.ncbi.nlm.nih.gov/SNP/"> dbSNP</a> Build 147 data, available on the two most 
recent human assemblies GRCh37/hg19 and GRCh38/hg38. NCBI's dbSNP database is a collection of 
&quot;simple nucleotide polymorphisms&quot; (SNPs), which are a class of genetic variations
that include single nucleotide polymorphisms and small insertions/deletions (indels). This immense 
database contains over 150 million such SNPs that cover the human genome.</p>
<p>
There are four SNP tracks available as part of this release. One is a track containing all mappings 
of reference SNPs to the human assembly, labeled &quot;All SNPs (147)&quot;. The other three tracks 
are subsets of this track and show interesting and easily defined subsets of dbSNP:</p>
<ul>
<li>
Common SNPs (147): uniquely mapped variants that appear in at least 1% of the population or are 
100% non-reference.</li>
<li>
Flagged SNPs (147): uniquely mapped variants, excluding Common SNPs, that have been flagged by dbSNP
as &quot;clinically associated&quot;.</li>
<li>
Mult. SNPs (147): variants that have been mapped to more than one genomic location. This track has 
been shrinking over the course of the last few releases as dbSNP now excludes most SNPs whose 
flanking sequences map to multiple locations in the genome.</li>
</ul>
<p>
By default, only the Common SNPs (147) are visible; other tracks must be made visible using the 
track controls. You will find the other SNPs (147) tracks on both of GRCh37/hg19 and GRCh38/hg38 
browsers in the &quot;Variation&quot; group.</p>
<p>
The tracks were produced at UCSC by Angie Hinrichs and Christopher Lee. We'd like to thank the dbSNP
group at NCBI for providing access to these data.</p>

<a name="081216"></a>
<h2>Aug. 12, 2016 &nbsp;&nbsp; Genome Browser for Malayan flying lemur now available</h2>
<p>
A Genome Browser is now available for the Malayan flying lemur (<em>Galeopterus variegatus</em>)
assembly released June 2014 by 
<a href="http://genome.wustl.edu/genomes/detail/galeopterus-variegatus/"
target="_blank">The Genome Institute at Washington University</a>. For more information and 
statistics about this assembly, see the NCBI assembly record for 
<a href="https://www.ncbi.nlm.nih.gov/assembly/182621" target="_blank"> G_variegatus-3.0.2</a>. There
are 179,514 scaffolds with a total size of 3,187,660,572 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/galVar1/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#malayan_flying_lemur">Downloads</a> page.
Please observe the <a href="credits.html#malayan_flying_lemur_use">conditions for use</a> when
accessing and using these data sets. The annotation tracks for this browser were generated by UCSC
and collaborators worldwide. See the <a href="credits.html#malayan_flying_lemur_credits">Credits</a>
page for a detailed list of the organizations and individuals who contributed to this release.</p>

<a name="063016_2"></a>
<h2>Jun. 30, 2016 &nbsp;&nbsp; CRAM files now supported</h2>
<p>
We are excited to announce the support of <a href="help/cram.html" target="_blank">CRAM</a> files 
on the Browser!</p>
<p>
CRAM files, which are more dense than BAM files, use an additional external &quot;reference 
sequence&quot; file to both compress and decompress the read information. For CRAM tracks to load 
there is an expectation that the checksum of the reference sequence used to create the CRAM file 
will be in the CRAM header and be accessible from the 
<a href="http://www.ebi.ac.uk/ena/software/cram-reference-registry" target="_blank">EBI CRAM 
reference registry</a> or in a designated &quot;<tt>refUrl</tt>&quot; location.</p>
<p>
<a href="/cgi-bin/hgTracks?db=hg19&position=chr21%3A33031597-33041570&hgct_customText=track%20type=bam%20name=%22CRAM%20Example%22%20description=%22A%201000%20Genomes%20CRAM%20file%22%20visibility=full%20bigDataUrl=ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/data/HG00139/exome_alignment/HG00139.mapped.ILLUMINA.bwa.GBR.exome.20121211.bam.cram%20doWiggle=1"
target="_blank">Click here</a> to load an example CRAM file from the 1000 Genomes Project where 
reads are displayed as a bar graph using the &quot;Display data as a density graph&quot; option. 
Loading CRAM tracks is identical to loading BAM files, the only difference is that instead of 
providing the URL to a BAM file, the URL instead points to a CRAM file. Both use <code>type
bam</code> in hub and custom track lines, and both expect an associated index file (CRAM.crai for 
CRAMs).</p>
<p>
See complete details on the <a href="help/cram.html" target="_blank">CRAM Track Format</a> page.</p>
<p>
Thanks to Jonathan Casper and Brian Raney for the implementation of this feature.</p>

<a name="063016"></a>
<h2>Jun. 30, 2016 &nbsp;&nbsp; Updated gene tracks available for the human (hg38) and mouse (mm10)
assemblies</h2>
<p>
We are pleased to announce the release of an updated GENCODE Genes track for the Human (GRCh38/hg38)
assembly. The GENCODE v24 gene set has 197,782 total transcripts, compared with 195,178 in the 
previous version.  The total number of canonical genes has increased from 49,534 to 49,948.
Comparing the new gene set with the previous version:</p>
<ul>
  <li>
  193,647 transcripts did not change.</li>
  <li>
  128 transcripts were not carried forward to the new version.</li>
  <li>
  997 transcripts are "compatible" with those in the previous set, meaning that the two transcripts 
  show consistent splicing. In most cases, the old and new transcripts differ in the lengths of 
  their UTRs.</li>
  <li>
  406 transcripts overlap with those in the previous set, but do not show consistent splicing (i.e.,
  they contain overlapping introns with differing splice sites)</li>
</ul>
<p>
More details about the GENCODE Genes track can be found on the
<a href="../cgi-bin/hgTrackUi?db=hg38&g=knownGene">
GENCODE v24 track description page</a>.</p>
<p>
Many thanks to Brian Raney and Luvina Guruvadoo for their work on this track!</p>
<hr>
<p>
We are  pleased to announce the release of an updated UCSC Genes track for the Mouse (GRCm38/mm10) 
assembly. The new release has 63,759 total transcripts, compared with 63,244 in the previous 
version. The total number of canonical genes has increased from 32,958 to 33,079. Comparing the new 
gene set with the previous version:</p>
<ul>
  <li>
  60,827 transcripts did not change between versions.</li>
  <li>
  148 transcripts were not carried forward to the new version.</li>
  <li>
  1,929 transcripts are &quot;compatible&quot; with those in the previous set, meaning
  that the two transcripts show consistent splicing.  In most cases, the old and new
  transcripts differ in the lengths of their UTRs.</li>
  <li>
  340 transcripts overlap with those in the previous set, but do not show consistent
  splicing (i.e., they contain overlapping introns with differing splice sites).</li>
</ul>
<p>
More details about the new UCSC Genes track can be found on the
<a href="../cgi-bin/hgTrackUi?db=mm10&g=knownGene">UCSC Genes track description page</a>.</p>
<p>
Many thanks to Jonathan Casper and Christopher Lee for their work on this track.</p>

<a name="062816"></a>
<h2>Jun. 28, 2016 &nbsp;&nbsp; New way to share sessions - public sessions</h2>
<p>
We are excited to announce the release of the <a href="../cgi-bin/hgPublicSessions">Public 
Sessions</a> page, the newest way to share your UCSC Genome Browser snapshots with your colleagues
and the world. The Public Sessions page collects sessions that users have elected to share publicly.
Sessions on this page can be filtered based on assembly, name, or a phrase from the description. 
Sessions can also be sorted based on their popularity or creation date. To add your own sessions to 
this Public Sessions page, just check the box under "post in public listing" on the session 
management page.</p>
<p>
The Genome Browser <a href="../cgi-bin/hgSession">session management</a> page has also received an
update with the release of the new Public Sessions page. Sessions are now arranged in a table,
where varying numbers of sessions can be displayed on a page of this table. Sessions can now be
filtered and ordered based on session name, primary assembly, and creation date. Clicking on the
session name now displays that session in the Genome Browser. Session details, such as the name
or the description, can be changed using the "details" button in the new table layout.</p>
<p>
The release of the new Public Sessions page also marks a change in our session expiration policy.
We no longer expire sessions and associated custom tracks four months after their last use.
Please note, however, that the UCSC Genome Browser is not a data storage service and that we
encourage you to keep local copies of your session contents and custom track data.</p>
<p>
For more information about using sessions to share snapshots of the UCSC Genome Browser,
please see the <a href="help/hgSessionHelp.html">User Guide</a> or our blog post,
<a href="https://genome-blog.gi.ucsc.edu/blog/how-to-share-your-ucsc-screenthoughts/"
target="_blank">How to share your UCSC screenthoughts</a>.</p>
<p>
Thanks to Jonathan Casper and Matthew Speir for their work on this new tool.</p>

<a name="061516"></a>
<h2>Jun. 15, 2016 &nbsp;&nbsp; New supported Asian mirror site released</h2>
<p>
We are pleased to announce the introduction of a new supported Genome Browser mirror to serve our 
users in Asia. The genome-asia server <a href="http://genome-asia.ucsc.edu" 
target="_blank">(http://genome-asia.ucsc.edu)</a>, which is administered by UCSC, is physically 
located at the <a href="http://www.yokohama.riken.jp/english/" target="_blank">RIKEN Yokohama 
Campus</a> in Japan.</p>
<p>
Genome-asia is intended as an alternate, faster access point for those Browser users
who are geographically closer to Asia than to the western United States. The mirror
is functionally identical to the U.S.-based server, with one exception. Custom tracks
residing on the U.S. server are not automatically accessible from the genome-asia mirror;
therefore, the use of these custom tracks and saved sessions containing them will require
some manual intervention.</p>
<p>
Genome Browser users located in Asia will be automatically redirected to genome-asia
when they click the "Genomes" menu item on the Genome Browser home page, but will have
the option to remain on the U.S.-based server. In addition to the
genome-asia mirror, we continue to support a Europe-based Genome Browser mirror
<a href="http://genome-euro.ucsc.edu" target="_blank">(http://genome-euro.ucsc.edu)</a>
in Bielefeld, Germany. A third-party mirror in <a href="http://genome-mirror.moma.ki.au.dk/" 
target="_blank">Aarhus, Denmark</a>, is available as an emergency backup in the rare event
that our three supported sites are offline.</p>
<p>
We wish to express our special thanks to Dr. Hideya Kawaji at the
<a href="http://www.riken.jp/en/research/labs/accc/" target="_blank">RIKEN Advanced Center for 
Computing and Communication</a> for making this possible. We would also like to acknowledge the 
following members of the UCSC Genome Browser team for their efforts: Galt Barber, Robert Kuhn, 
Jorge Garcia, and Luvina Guruvadoo.</p>

<a name="060616"></a>
<h2>Jun. 6, 2016 &nbsp;&nbsp; New video: Exon-only mode</h2>
<p>
We are pleased to announce the release to our YouTube channel of a new
training <a href = "https://www.youtube.com/watch?v=mNEJBOkf_Gk" target = _BLANK>video</a>,
in which we demonstrate the recently released Multi-Region feature in exon-only mode. This feature
is useful for those studying gene expression or undertaking whole-exome sequencing. By removing 
intronic and intergenic regions, exon-only mode shows the portion (sometimes as little as a few 
percent) of the genome most useful for these analyses.</p>
<p>
This video is <a href ="../training/vids/#vid09" TAGET = _BLANK>indexed</a> to allow easy access 
to relevant sections and comes with a transcript for the hard-of-hearing.</p>
<p>
Thanks to Robert Kuhn and David Gibson for production.</p>

<a name="060116"></a>
<h2>Jun. 1, 2016 &nbsp;&nbsp; New density graph feature</h2>
<p>
The Genome Browser is proud to announce a new &quot;density graph&quot; feature for
a variety of tracks, including BAM custom tracks, that converts data into a summary display.</p>
<p>
The density graph feature enables certain track types to be displayed as a bar graph where
the height is proportional to the number of reads mapped to each genomic position. Through dynamic 
calculation of items in the current window, this feature plots a line similar to a wiggle graph that
can be customized with a number of <a href="help/hgWiggleTrackHelp.html"
target="_blank">graph-based configuration options</a> such as drawing indicator lines,
smoothing plots, adjusting graph height and vertical range, and switching from bars to
points. Please note that the feature is best displayed with Display mode set to full
and that the default is auto-scale data to view. Please note that using this
feature on a <a href="help/gbib.html#GbibMirroring"
target="_blank">virtual machine Genome Browser in a Box (GBiB)</a> may benefit from increasing 
memory.</p>
<p>
To turn on the feature click into the track controls for certain track types, such as a BAM custom 
track, and check the box next to &quot;Display data as a density graph&quot; to see graph 
configurations and then click &quot;Submit&quot;.  To see the new density graph feature in use, 
please see this example
<a href="../../cgi-bin/hgTracks?hgS_doLoadUrl=submit&hgS_loadUrlName=http://genome.ucsc.edu/goldenPath/help/examples/sessions/session.hg19.densityGraph&hgct_customText=track%20type=bam%20name=bamExample%20visibility=full%20bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bamExample.bam%20doWiggle=1"
target="_blank">session</a>.</p>  
<p>
Thanks to Brian Raney for the implementation of this feature.</p>

<a name="051016"></a>
<h2>May 10, 2016 &nbsp;&nbsp; New Gateway page</h2>
<p>
The UCSC Genome Browser team is proud to announce a newly redesigned 
<a href="../cgi-bin/hgGateway" target="_blank">Genome Browser Gateway</a> page.
The Gateway retains its original functionality as a central access point for all genome assemblies
available on our site, while sporting several helpful new features and updates:</p>
<ul>
  <li>
  Autocomplete searching for any genome browser, genome version, or public hub</li>
  <li>
  Species browsing through a scrollable visual
  &quot;tree&quot; menu based on phylogenetic order</li>
  <li>
  Quick access shortcuts to popular browsers</li>
  <li>
  New style and color scheme</li>
</ul>
<p>
Please see our <a href="https://youtu.be/5zc9sqd4SD4"
target="_blank">Gateway video</a> for an introduction to these new features.</p>
<p>
The new Gateway addresses the need for streamlined access to the rapidly 
growing number of genome assemblies available on our public site. 
As an alternative to scrolling through long drop-down menus of genome assemblies, you can
now simply type in assembly search terms for autocomplete recognition, visually scroll
through our collection, or click a button to access our most popular species.</p>
<p>
Note that a few browser utilities that were previously accessed through links and buttons on the 
Gateway page have been moved to the top menu bar:</p>
<ul>
  <li>
  Browser reset: Genome Browser > Reset All User Settings</li>
  <li>
  Track search: Genome Browser > Track Search</li>
  <li>
  Add custom tracks: My Data > Custom Tracks</li>
  <li>
  Track hubs: My Data > Track Hubs</li>
  <li>
  Configure tracks and display: Genome Browser > Configure</li>
</ul>
<p>
We would like to acknowledge the following people for their contributions
to the new Gateway: original art by Jeff West Design, concept by Robert Kuhn,
software design by Angie Hinrichs, and quality assurance by Cath Tyner.</p>

<a name="042216"></a>
<h2>Apr. 22, 2016 &nbsp;&nbsp; Data from the Lens PatSeq Database now available</h2>
<p>
We are pleased to announce the release of a set of tracks showing the genomic mapping of
biomedical sequences submitted as part of patent application documents worldwide. The sequence 
data, mappings and associated patent information were obtained from the 
<a href="https://www.lens.org/lens/bio" target="_blank">PatSeq</a> database provided by 
<a href="https://www.lens.org/" target="_blank">The Lens</a>. The PatSeq data are 
mapped to the genome, and the individual sequence features within the tracks are then
color-coded to indicate their status within the associated patent documents. 
Track details pages show information about the patent documents in which the sequences
are referenced, and provide a link to The Lens PatSeq Analyzer tool for the given chromosome 
range. The PatSeq data are divided into two tracks: a bulk patent track for sequences affiliated 
with patents in which more than 100 sequences were submitted, and a non-bulk patent track
for all other sequences. These tracks are currently present on the human (hg19), mouse (mm10),
and Ebola virus (eboVir3) genomes.</p>
<p>
Thanks to our collaborators at The Lens, Osmat Jefferson and Deniz Koellhofer, for providing the
data and feedback on the visualization. Thank you to Max Haeussler and Matthew Speir, members of
the UCSC Genome Browser team, for their efforts in creating these tracks.</p>

<a name="042016"></a>
<h2>Apr. 20, 2016 &nbsp;&nbsp; New GTEx gene expression track for hg19 and hg38</h2>
<p>
We are excited to announce the release of a new gene expression track based on data from the NIH
<a href="http://commonfund.nih.gov/GTEx/index" target = _BLANK> Genotype-Tissue Expression</a> 
(GTEx) project. This track displays tissue-specific gene expression based on RNA-seq in 53 tissues 
from 570 donors obtained from the GTEx 'midpoint' data release (V6, October 2015). The original data
for this track can be found at the <a href="http://www.gtexportal.org/home" 
target="_blank">GTEX Portal</a> hosted by the Broad Institute.</p>
<p>
This track also features a new gene expression display method that extends the traditional Genome 
Browser display &mdash; a horizontal bar graph. Every gene is annotated by a graph with colored 
bars, each of which corresponds to a specific tissue assayed by the GTEx project. Within
a graph, the bar color indicates the tissue type, using GTEx conventions, and the bar height 
depicts the median expression level (in RPKM).To quickly view the tissue and expression level 
represented by a bar in the tracks display, mouse over the bar in the graph. The complete tissue 
color legend is shown on the <a href="../cgi-bin/hgTrackUi?db=hg19&g=gtexGene" target="_blank">track
configuration page</a>, and can also be popped up for viewing alongside the track using the 
right-click menu. Below the bar graph, a line is shown indicating the gene extent that was used to 
generate the annotation, colored by gene class using GENCODE conventions (e.g. <span style='color: 
#0c0c78'>blue</span> for protein-coding, <span style='color: #006400'>green</span> for
non-coding).</p>
<p>
<a href="../cgi-bin/hgTracks?hgS_doLoadUrl=submit&hgS_loadUrlName=http://genome.ucsc.edu/goldenPath/help/examples/sessions/gtexAnnouncementSession.txt" 
target="_blank"><img src="../images/gtexAnnounceExample.png" alt="GTEx announcement
session" width="822" height="254"></a></p>
<p>
User alert: In the figure above, do you notice how the bar graph sometimes extends past the 
righthand end of the associated gene annotation? This is because all bar graphs in the
display are the same width (and have the same tissue ordering) to facilitate comparison.
For example, in this figure (which may also be viewed 
<a href="../cgi-bin/hgTracks?hgS_doLoadUrl=submit&hgS_loadUrlName=http://genome.ucsc.edu/goldenPath/help/examples/sessions/gtexAnnouncementSession.txt" 
target="_blank">here</a>), the three tall bars in the annotation of the TCAP gene indicate this 
<em>entire</em> gene is highly expressed in cardiac and skeletal muscle. It does <em>not</em> 
indicate that there are expression peaks in the intergenic region beyond the gene!</p>
<p>
Similar to other Genome Browser track displays, clicking on the graph will bring up a description 
page that shows a detailed box-and-whiskers plot of the RPKM data, as well as a description of the 
methods used to generate the data.</p>
<p>
Credit goes to Kate Rosenbloom and Christopher Lee for the implementation and testing of this 
feature.</p>

<a name="041516_2"></a>
<h2>Apr. 15, 2016 &nbsp;&nbsp; dbSNP 146 available for hg19 and hg38</h2>
<p>
We are pleased to announce the release of four tracks derived from NCBI 
<a href="https://www.ncbi.nlm.nih.gov/SNP/"> dbSNP</a> Build 146 data, available on the two most
recent human assemblies GRCh37/hg19 and GRCh38/hg38. NCBI's dbSNP database is a collection of 
&quot;simple nucleotide polymorphisms&quot; (SNPs), which are a class of genetic variations
that include single nucleotide polymorphisms and small insertions/deletions (indels). This immense 
database contains over 150 million such SNPs that cover the human genome.</p>
<p>
There are four SNP tracks available as part of this release. One is a track containing all mappings 
of reference SNPs to the human assembly, labeled &quot;All SNPs (146)&quot;. The other three tracks 
are subsets of this track and show interesting and easily defined subsets of dbSNP:</p>
<ul>
  <li>
  Common SNPs (146): uniquely mapped variants that appear in at least 1% of the population or are 
  100% non-reference.</li>
  <li>
  Flagged SNPs (146): uniquely mapped variants, excluding Common SNPs, that have been flagged by 
  dbSNP as &quot;clinically associated&quot;.</li>
  <li>
  Mult. SNPs (146): variants that have been mapped to more than one genomic location. This track has
  been shrinking over the course of the last few releases as dbSNP now excludes most SNPs whose 
  flanking sequences map to multiple locations in the genome.</li>
</ul>
<p>
By default, only the Common SNPs (146) are visible; other tracks must be made visible using the 
track controls.  You will find the other SNPs (146) tracks on both of GRCh37/hg19 and GRCh38/hg38 
browsers in the &quot;Variation&quot; group.</p>
<p>
The tracks were produced at UCSC by Angie Hinrichs and Matthew Speir. We'd like to thank the dbSNP
group at NCBI for providing access to these data.</p>

<a name="041516_1"></a>
<p>
<h2>Apr. 15, 2016 &nbsp;&nbsp; Three new assemblies now available in the Genome Browser</h2>
<p>
A Genome Browser is now available for the gorilla (<em>Gorilla gorilla gorilla</em>) assembly 
released December 2014 by the Wellcome Trust Sanger Institute. For more information and statistics
about this assembly, see the NCBI assembly record for 
<a href="https://www.ncbi.nlm.nih.gov/assembly/503571" target="_blank">gorGor4</a>.  
There are 25 complete chromosomes alongside 40,667 scaffolds with a total size of 3,063,362,754 
bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/gorGor4/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#gorilla">Downloads</a> page.
Please observe the <a href="credits.html#gorilla_use">conditions for use</a> when
accessing and using these data sets. The annotation tracks for this browser were
generated by UCSC and collaborators worldwide. See the 
<a href="credits.html#gorilla_credits">Credits</a> page for a detailed list of the
organizations and individuals who contributed to this release.</p>
<hr>
<p>
A Genome Browser is now available for the rhesus monkey (<em>Macaca mulatta</em>) assembly 
released November 2015 by the
<a href="https://www.hgsc.bcm.edu/non-human-primates/rhesus-monkey-genome-project"
target="_blank">Baylor College of Medicine Human Genome Sequencing Center</a>. 
For more information and statistics about this assembly, see the NCBI assembly record for 
<a href="https://www.ncbi.nlm.nih.gov/assembly/2715158" target="_blank"> Mmul_8.0.1</a>. There are 
23 complete chromosomes alongside 284,705 scaffolds with a total size of 3,236,224,332 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/rheMac8/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#rhesus">Downloads</a> page. Please observe 
the <a href="credits.html#rhesus_use">conditions for use</a> when accessing and using these data 
sets. The annotation tracks for this browser were generated by UCSC and collaborators worldwide.  
See the <a href="credits.html#rhesus_credits">Credits</a> page for a detailed list of the
organizations and individuals who contributed to this release.</p>
<hr>
<p>
A Genome Browser is now available for the crab-eating macaque (<em>Macaca fascicularis</em>)
assembly released June 2013 by the
<a href="http://genome.wustl.edu/genomes/detail/macaca-fascicularis/"
target="_blank">Genome Sequencing Center at Washington University, St. Louis</a>. For more 
information about this assembly, see <a href="https://www.ncbi.nlm.nih.gov/assembly/704988"
target="_blank">Macaca_fascicularis_5.0</a> in the NCBI Assembly database. There are 22 complete 
chromosomes alongside 7,579 scaffolds with a total size of 2,946,843,737 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/macFas5/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#crab-eating_macaque">Downloads</a> page.
Please observe the <a href="credits.html#crab-eating_macaque_use">conditions for use</a> when
accessing and using these data sets. The annotation tracks for this browser were
generated by UCSC and collaborators worldwide. See the 
<a href="credits.html#crab-eating_macaque_credits">Credits</a> page for a detailed list of the
organizations and individuals who contributed to this release.</p>

<a name="040416"></a>
<h2>Apr. 4, 2016 &nbsp;&nbsp; New video shows how to obtain sequence or coordinates for exons. </h2>
<p>
We have released a new <a href="http://genome.ucsc.edu/training/vids/index.html#vid08" 
target="_blank">video</a> to our YouTube channel. This newest installment in our effort to share the
capabilities of the Genome Browser shows how to use the Table Browser to obtain either the 
sequences or the coordinates for exons in the genes from a chosen region of the genome.</p>
<p>
We also want to take this opportuinty to remind you that we offer onsite 
<a href="http://genome.ucsc.edu/training/index.html" target="_blank">workshops</a> to help you and 
your colleagues make the most out of the Genome Browser.</p>

<a name="040116"></a>
<h2>Apr. 1, 2016 &nbsp;&nbsp; Going back to our roots</h2>
<p>
For many years, the UCSC Genome Browser has provided a dynamic, cutting-edge interface for
viewing genomic data.  As time has gone by, however, it has become clear to us that we
cannot keep this up.  It is not fair to you, our users, that our interface keeps changing,
and it has increasingly been a burden on us to keep up with emerging web technologies.  Today
that all changes, as we return you to the roots of the web: ASCII.</p>
<p>
<img src="../images/joke04012016.png" alt="April Fools joke" width="712" height="279"></p>
<p>
We apologize again for subjecting you to so many changes over the years, and hope that this
new, more consistent interface will find your approval.</p>
<p>
In related news, we have also revised the interface for uploading custom tracks. Rather than
forcing you to wrestle with inconvenient file formats for your data or attempt to set up a web 
server, we now provide the option of loading your data from a deck of ISO 6586:1980 punch cards.  
Simply punch your data onto a deck and mail the deck to us. Our staff of highly trained 
specialists will take care of loading track, and will inform you by mail when the process is 
complete. Please note that our current policy of expiring unused custom tracks remains in effect; 
you must visit your track within 72 hours of the postmark date. Regrettably, decks cannot be 
returned.</p>
<p>Card decks may be addressed to:<br>
UCSC Genome Informatics Group<br>
Center for Biomolecular Science & Engineering<br>
CBSE, 501D Engineering II Building<br>
University of California, Santa Cruz<br>
1156 High Street<br>
Santa Cruz, CA 95064</p>
<p>
This new file format is also an option for data output from the <a href="../cgi-bin/hgTables">UCSC
Table Browser</a>. Simply select &quot;mail card deck&quot; from the &quot;output format&quot; 
menu, and then enter your name and address on the subsequent page. Please allow 4-6 weeks for 
delivery.</p>

<a name="032116"></a>
<h2>Mar. 21, 2016 &nbsp;&nbsp; Genome Browsers for <em>C. elegans</em> and brown kiwi now 
available</h2>
<p>
A Genome Browser is now available for the <em>Caenorhabditis elegans</em> assembly released February
2013 by the C. elegans Sequencing Consortium (C. elegans Sequencing Consortium version WBcel235, 
UCSC version ce11). For more information and statistics about this assembly, see the NCBI assembly 
record for <a href="https://www.ncbi.nlm.nih.gov/assembly/554278" target="_blank">WBcel235</a>.
There are 7 complete chromosomes with a total size of 100,286,401 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/ce11/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#c_elegans">Downloads</a> page. Please observe
the <a href="credits.html#ce_credits">conditions for use</a> when accessing and using these data 
sets. The annotation tracks for this browser were generated by UCSC and collaborators worldwide. See
the <a href="credits.html#ce_credits">Credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p>
<hr>
<p>
A Genome Browser is now available for the brown kiwi (<em>Apteryx australis</em>) assembly released 
June 2015 by the <a href="http://www.eva.mpg.de/index.html"
target="_blank">Max-Plank Institute for Evolutionary Anthropology</a> (MPI-EVA version AptMan0,
UCSC version aptMan1). For more information and statistics about this assembly, see
<a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_001039765.1"
target="_blank">AptMant0</a> in the NCBI Assembly database. There are
24,719 scaffolds with a total size of 1,523,986,457 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/aptMan1/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#brown_kiwi">Downloads</a> page.
Please observe the <a href="credits.html#brown_kiwi_use">conditions for use</a> when accessing and 
using these data sets. The annotation tracks for this browser were generated by UCSC and 
collaborators worldwide. See the <a href="credits.html#brown_kiwi_credits">Credits</a> page for a 
detailed list of the organizations and individuals who contributed to this release.</p>

<a name="030816"></a>
<h2>Mar. 8, 2016 &nbsp;&nbsp; Combine multiple regions of the Genome Browser into a single 
visualization</h2>
<p>
Have you ever wished you could remove all of the intronic or intergenic regions from the
Genome Browser display? Have you ever dreamed of being able to visualize multiple
far-flung regions of a genome? Well, now you can with the new &quot;multi-region&quot; option
in the Genome Browser!</p>
<p>
The multi-region modes provide four different ways to manipulate the display.
First, we have an &quot;exon-only&quot; mode that allows you to
remove all of the intergenic and intronic sequences from the display, leaving you with
only the exonic regions. Next, there is the &quot;gene-only&quot; mode allowing you to remove only
the intergenic regions from the display, leaving the genes with exons and introns intact.
Then there is the &quot;custom regions&quot; mode that allows you to use a remote BED file to
specify the regions you are interested in viewing alongside each other. Lastly, for human
assemblies hg17 and newer, there is the &quot;alternative haplotype&quot; mode that allows you to
view a haplotype sequence inserted into its position in the reference genome.</p>
<p>
To access these exciting, new multi-region modes, first select your organism and assembly
of interest and navigate to the Genome Browser visualization. Below the tracks display,
there is a button labeled &quot;multi-region&quot;. Clicking this button will bring up
a configuration box for the different multi-region modes. For example:</p>
<p>
<img src="../images/MultiRegionButton.png" alt="MultiRegion button" width="799" height="201"></p>
<p>
From this pop-up, you will be able to switch between the different modes or exit these multi-region 
modes if desired. For instance: </p>
<p>
<img src="../images/MultiRegionConfigurationWindow.png" alt="MultiRegion configuation 
window" width="757" height="312"></p>
<p>
Additionally, you can enter the &quot;Exon-only&quot; mode by typing the keyboard shortcut
&quot;e&quot; then &quot;v&quot;. You can exit any of the multi-region modes and return to the
default Genome Browser display by typing the keyboard shortcut &quot;d&quot; then &quot;v&quot;.</p>
<p>
For more information about each individual multi-region mode, please see the
<a href="help/multiRegionHelp.html">User Guide</a>.</p>
<p>
Thank you to Galt Barber, Matthew Speir, and the entire UCSC Genome Browser quality assurance team 
for all of their efforts in creating these exciting new display modes.</p>

<a name="021016"></a>
<h2>Feb. 10, 2016 &nbsp;&nbsp; Two new assemblies now available in the Genome Browser</h2>
<p>
A Genome Browser is now available for the mouse lemur (<em>Microcebus murinus</em>) assembly 
released May 2015 by the 
<a href="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project" 
target="_blank">The Broad Institute</a> and the
<a href="https://www.hgsc.bcm.edu/non-human-primates/mouse-lemur-genome-project"
target="_blank">Baylor College of Medicine Human Genome Sequencing Center</a>. 
For more information and statistics about this assembly, see the NCBI assembly record
for <a href="https://www.ncbi.nlm.nih.gov/assembly/329621" target="_blank"> Mmur_2.0</a>.  
There are 10,311 scaffolds with a total size of 2,438,804,424 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/micMur2/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#mouse_lemur">Downloads</a> page. Please 
observe the <a href="credits.html#mouse_lemur_use">conditions for use</a> when accessing and using 
these data sets. The annotation tracks for this browser were generated by UCSC and collaborators 
worldwide. See the <a href="credits.html#mouse_lemur_credits">Credits</a> page for a detailed list 
of the organizations and individuals who contributed to this release.</p>
<hr>
<p>
A Genome Browser is now available for the platypus (<em>Ornithorhynchus anatinus</em>) assembly 
released February 2007 by the 
<a href="http://genome.wustl.edu/genomes/detail/ornithorhynchus-anatinus/"
target="_blank">Genome Sequencing Center at Washington University, St. Louis</a>. For more information
about this assembly, see <a href="https://www.ncbi.nlm.nih.gov/assembly/237598/" target="_blank">
Ornithorhynchus_anatinus-5.0.1</a> in the NCBI Assembly database. There are 201,525 scaffolds with 
a total size of 1,996,826,513 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/ornAna2/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#platypus">Downloads</a> page.
Please observe the <a href="credits.html#platypus_use">conditions for use</a> when accessing and 
using these data sets. The annotation tracks for this browser were generated by UCSC and 
collaborators worldwide. See the <a href="credits.html#platypus_credits">Credits</a> page for a 
detailed list of the organizations and individuals who contributed to this release.</p>

<a name="012016"></a>
<h2>Jan. 20, 2016 &nbsp;&nbsp; dbSNP 142 available for mm10</h2>
<p>
Data from dbSNP build 142 is now available for the most recent mouse assembly (mm10/GRCm38). As was 
the case for previous annotations based on dbSNP data, there are three tracks in this release. One 
is a track containing all mappings of reference SNPs to the mouse assembly, labeled &quot;All SNPs 
(142)&quot; The other two tracks are subsets of this track and show different interesting and easily
defined subsets of dbSNP: </p>
<ul>
  <li>
  Common SNPs (142): uniquely mapped variants that appear in at least 1% of the population</li>
  <li>
  Mult. SNPs (142): variants that have been mapped to more than one genomic location</li>
</ul>
<p>
By default, only the Common SNPs (142) are visible. The other tracks can be made visible using the 
track controls. These three SNPs (142) tracks can be found on the Mouse Dec. 2011 (mm10/GRCm38) 
browser in the Variation and Repeats group.</p>
<p>
Thank you to the <a href="https://www.ncbi.nlm.nih.gov/SNP/" target="_blank">dbSNP</a> group at NCBI 
for making these data publicly available. The tracks were produced at UCSC by Brian Raney, Angie 
Hinrichs and Matthew Speir.</p>

<a name="010816"></a>
<p>
<h2>Jan. 8, 2016 &nbsp;&nbsp; dbSNP 144 available for hg19 and hg38</h2>
<p>
We are pleased to announce the release of four tracks derived from NCBI 
<a href="https://www.ncbi.nlm.nih.gov/SNP/" target ="_blank">dbSNP</a> Build 144 data, available on 
the two most recent human assemblies GRCh37/hg19 and GRCh38/hg38.</p>
<p>
There are four SNP tracks available as part of this release. One is a track containing all mappings 
of reference SNPs to the human assembly, labeled &quot;All SNPs (144)&quot; The other three tracks 
are subsets of this track and show interesting and easily defined subsets of dbSNP:</p>
<ul>
  <li>
  Common SNPs (144): uniquely mapped variants that appear in at least 1% of the population or 
  are 100% non-reference</li>
  <li>
  Flagged SNPs (144): uniquely mapped variants, excluding Common SNPs, that have been flagged by 
  dbSNP as &quot;clinically associated&quot;li>
  <li>
  Mult. SNPs (144): variants that have been mapped to more than one genomic location</li> 
</ul>
<p>
By default, only the Common SNPs (144) are visible; other tracks must be made visible using the 
track controls. You will find the other SNPs (144) tracks on both of GRCh37/hg19 and GRCh38/hg38 
browsers in the Variation group.</p>
<p>
The tracks were produced at UCSC by Angie Hinrichs and Luvina Guruvadoo. We'd like to thank the 
dbSNP group at NCBI for providing access to these data.</p>

<!-- ============= 2015 archived news ============= -->
<a name="2015"></a>
<a name="122115"></a>
<p>
<h2>Dec. 21, 2015 &nbsp;&nbsp; Two new assemblies now available in the Genome Browser</h2>
<p>
A Genome Browser is now available for the frog (<em>Xenopus tropicalis</em>) assembly released 
Sept. 2012 by the <a href="http://genome.jgi.doe.gov/" target="_blank">US DOE Joint Genome Institute
(JGI-PGF)</a> (JGI v7.0, UCSC version xenTro7). For more information and statistics about this 
assembly, see the NCBI assembly record for <a href="https://www.ncbi.nlm.nih.gov/assembly/515038" 
target="_blank">Xtropicalis_v7</a>. There are 7,728 scaffolds with a total size of 1,437,530,879
bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/xenTro7/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#xentro">Downloads</a> page. Please observe 
the <a href="credits.html#xentro_use">conditions for use</a> when accessing and using these data 
sets. The annotation tracks for this browser were generated by UCSC and collaborators woldwide. 
See the <a href="credits.html#xentro_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>
<hr>
<p>
A Genome Browser is now available for the cat (<em>Felis catus</em>) assembly released Nov. 2014 by 
the International Cat Genome Sequencing Consortium (ICGSC version Felis_catus_8.0, UCSC version 
felCat8). For more information and statistics about this assembly, see the NCBI assembly record
for <a href="https://www.ncbi.nlm.nih.gov/assembly/1373248" target="_blank">Felis_catus_8.0</a>. 
There are 267,625 scaffolds with a total size of 2,641,342,258 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/felCat8/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#cat">Downloads</a> page.
Please observe the <a href="credits.html#cat_use">conditions for use</a> when accessing and using 
these data sets. The annotation tracks for this browser were generated by UCSC and collaborators 
woldwide. See the <a href="credits.html#cat_credits">Credits</a> page for a detailed list of the
organizations and individuals who contributed to this release.</p>

<a name="121615"></a>
<p>
<h2>Dec. 16, 2015 &nbsp;&nbsp; Output from related tables now available in the Data
Integrator</h2>
<p>
In the UCSC Genome Browser database, detailed information on the annotations for many tracks
is stored in extra tables. This information can include things such as identifiers
in other databases, transcript status, or other descriptive information. You can now obtain
data from these related tables in your output from the
<a href="../cgi-bin/hgIntegrator">Data Integrator</a>.</p>
<p>
To add related tables to your Data Integrator output, first, select and add at least one
track to your data sources. Then, scroll down to the "Output Options" section and click the
<button type="button">Choose fields</button> button.
Lastly, at the bottom of the section for each selected track, you will see a drop-down menu
labeled "Related tables" which allows you to select and add various tables to your output.
<p>
<img src="../images/hgIntegratorRelatedTablesAnnouncement.png" alt="hgIntegrator related
tables" width="684" height="418"></p>
<p>
For more information about using the Data Integrator, please see the 
<a href="help/hgIntegratorHelp.html">User Guide</a>.</p>
<p>
Thank you to Angie Hinrichs and Matthew Speir for their work on this feature.</p>

<a name="120915"></a>
<h2>Dec. 9, 2015 &nbsp;&nbsp; Data from 1000 Genomes Project Phase 3 now available</h2>
<p>
We have made data from Phase 3 of the <a href="http://www.1000genomes.org/" target="_blank">1000 
Genomes Project</a> available for the hg19 version of the human assembly. The data includes almost 
90 million variants in the form of single nucleotide variants (SNVs), insertions/deletions (InDels),
and structural variants (SVs). Each variant includes a detail page that incorporates links to the 
variant in the dbSNP database, quality scores, and allele frequency information for various 
populations. More information is available on the 
<a href="../cgi-bin/hgTrackUi?db=hg19&g=tgpPhase3">track description page</a>.</p>
<p>
In addition to variant calls, the data includes two tracks that highlight regions of the
genome that are accessible to next generation sequencing technologies that utilize paired-end reads.
One track contains regions that meet the  1000 Genomes Project &quot;pilot&quot; criteria while the 
other contains regions that meet a more &quot;strict&quot; set of guidelines. More information about
how these accessible regions were identified can be found on the
<a href="../cgi-bin/hgTrackUi?db=hg19&g=tgpPhase3Accessibility">track description page</a>.</p>
<p>
Thank you to the 1000 Genomes Project and Tom Blackwell and Mary Kate Wing at the
University of Michigan for making these data available. Thank you to Angie Hinrichs
and Matthew Speir at UCSC for their work on these tracks.</p>

<a name="112515"></a>
<h2>Nov. 25, 2015 &nbsp;&nbsp; New &quot;View - In External Tools&quot; menu sends data to 
prediction websites</h2>
<p>
Do you often get the current sequence with &quot;View - DNA&quot; just to copy and paste it into 
another website? With the new menu &quot;View - In External Tools&quot; you can now send the current
region to remote tools with two clicks.</p>
<p>
For example, when browsing hg38 and zoomed in on the start of a gene like ABO at
<a href="/cgi-bin/hgTracks?db=hg38&position=chr9%3A133255000-133255555"
target="_blank">chr9:133,255,000-133,255,555</a>, click the View menu and select &quot;In External 
Tools&quot;, or simply type the keyboard shortcut &quot;s&quot; then &quot;t&quot;, to send the 
region to a list of different prediction tools.</p>
<p>
Some of the available tools (not available for all assemblies or all zoom levels)
are primer design and restriction enzyme predictions (Primer3Plus, Primer-BLAST,
NEBCutter), mRNA and protein predictions (RegRNA 2, Riboswitches, RNAfold, PFAM),
cis-regulatory predictions (Promoter 2, RSAT Metazoa), and CRISPR/Cas9 predictions
(CRISPOR, CHOPCHOP), as well as links to view the region at NCBI or Ensembl.
You can suggest other tools via our <a href="/cgi-bin/hgUserSuggestion">Suggestion Box</a>.</p>
<p>
Credit goes to Max Haeussler for implementing this new feature.</p>

<a name="110315"></a>
<h2>Nov. 3, 2015 &nbsp;&nbsp; New quick links for the Browser's top menu bar</h2>
<p>
The &quot;Genomes&quot; menu now features direct links to jump to browsing the human (hg38 and hg19)
and mouse (mm10 and mm9) assemblies! These links also preserve location information, such 
as a favorite gene's location, allowing users to quickly compare available annotation 
tracks between two assemblies.</p>
<p>
The &quot;Genome Browser&quot; menu now features a "Configure" link to quickly set visibilities for 
all tracks on the currently browsed assembly and a "Reset All User Settings" option that will 
remove all external hubs and custom tracks data while resetting the entire browser to default 
visibilities and the hg38 assembly.</p>

<a name="092315"></a>
<h2>Sep. 23, 2015 &nbsp;&nbsp; New keyboard shortcuts on the Genome Browser</h2>
<p>
We are excited to announce the addition of keyboard shortcuts to navigate the Genome Browser 
display. It's now possible to zoom in and out, navigate left and right, hide all tracks, configure 
the browser display and much more using only your keyboard. Use the "?" key to view the entire list 
of shortcuts. Happy browsing!</p>
<p>
Credit goes to Max Haeussler for implementing this new feature. </p>

<a name="091415"></a>
<h2>Sep. 14, 2015 &nbsp;&nbsp; Human Genome Browser default changed to GRCh38/hg38</h2>
<p>
In conjunction with the release of the new 100-species Conservation track on the
hg38/GRCh38 human assembly, we have now changed the default human browser on our
website from hg19 to hg38. This should not affect your current browsing sessions;
if you were last looking at the hg19 (or older) browser, the Genome Browser will
continue to display that assembly for you when you start it up. There are circumstances,
however, in which the selected assembly can switch to the newer version. For instance,
the assembly will switch to hg38 if you reset your browser defaults. If you find
yourself in a situation where some of your favorite browser tracks have &quot;disappeared&quot;,
you may want to check that you're viewing the right assembly.</p>
<p>
We will continue our efforts to expand the annotation track set on the hg38 browser
to include many of the tracks present on previous human assemblies. In cases where
it makes sense, data may be simply &quot;lifted&quot; from hg19 using migration tools.
In many instances, however, we must rely on our data providers to generate new versions
of their data on the latest assembly. We will publish these data sets as they become available.</p>
<p>
For a summary of the new features in the GRCh38 assembly, see the 
<a href="newsarch.html#030614">overview</a> we published in March 2014.</p>

<a name="091015"></a>
<h2>Sep. 10, 2015 &nbsp;&nbsp; 100 Species Conservation Track now available on GRCh38/hg38</h2>
<p>
We are pleased to announce the release of a new Conservation track based on the human
(GRCh38/hg38) assembly. This track shows multiple alignments of 100 vertebrate species and
measurements of evolutionary conservation using two methods (phastCons and phyloP) from the
PHAST package for all species. The multiple alignments were generated using multiz and other
tools in the UCSC/Penn State Bioinformatics comparative genomics alignment pipeline. Conserved 
elements identified by phastCons are also displayed in this track. For more details, visit the
<a href="../cgi-bin/hgTrackUi?db=hg38&g=cons100way">track description page</a>.</p>
<p>
We would like to thank Hiram Clawson and Jonathan Casper for their efforts in creating this 
track.</p>

<a name="081215"></a> 
<h2>Aug. 12, 2015 &nbsp;&nbsp; How to share your UCSC screenthoughts</h2>
<p>
Do you want to save your UCSC Browser screenthoughts, or share them with a colleague,
or make a slide for a talk, or publish them in a paper?  Saving the state of your screen
can take two forms: static or dynamic. You can snap and save a picture of the screen,
or you can share a link to an active Genome Browser. We talk about both approaches
<a href="https://genome-blog.gi.ucsc.edu/blog/how-to-share-your-ucsc-screenthoughts/" 
target="_blank">here</a> and discuss some of the advantages and pitfalls of each.</p>

<a name="062915"></a> 
<h2>Jun. 29, 2015 &nbsp;&nbsp; GENCODE Genes now the default gene set on the Human (GRCh38/hg38) 
assembly</h2>
<p>
In a move towards standardizing on a common gene set within the bioinformatics community, UCSC has 
made the decision to adopt the GENCODE set of gene models as our default gene set on the human 
genome assembly. Today we have released the GENCODE v22 comprehensive gene set as our default gene 
set on human genome assembly GRCh38 (hg38), replacing the previous default UCSC Genes set generated 
by UCSC. To facilitate this transition, the new gene set employs the same familiar UCSC Genes 
schema, using nearly all the same table names and fields that have appeared in earlier versions of 
the UCSC set.  </p>
<p>
By default, the browser displays only the transcripts tagged as &quot;basic&quot; by the GENCODE 
Consortium. These may be found in the track labeled "GENCODE Basic" in the Genes and Gene 
Predictions track group. However, all the transcripts in the GENCODE comprehensive set are present 
in the tables, and may be viewed by adjusting the track configuration settings for the All GENCODE 
super-track. The most recent version of the UCSC-generated genes can still be accessed in the track 
&quot;Old UCSC Genes&quot;.</p>
<p>
The new release has 195,178 total transcripts, compared with 104,178 in the previous version. The 
total number of canonical genes has increased from 48,424 to 49,534. Comparing the new gene set with
the previous version: </p>
<ul>
  <li>
  9,459 transcripts did not change.</li>
  <li>
  22,088 transcripts were not carried forward to the new version.</li>
  <li>
  43,681 transcripts are "compatible" with those in the previous set, meaning that the two 
  transcripts show consistent splicing. In most cases, the old and new transcripts differ in the 
  lengths of their UTRs.</li>
  <li>
  28,950 transcripts overlap with those in the previous set, but do not show consistent splicing 
  (i.e., they contain overlapping introns with differing splice sites)</li>
</ul>
<p>
More details about the new GENCODE Basic track can be found on the
<a href="../cgi-bin/hgTrackUi?db=hg38&g=knownGene">
GENCODE Basic track description page</a>.</p>

<a name="062615"></a>
<h2>Jun. 26, 2015 &nbsp;&nbsp; New bonobo (panPan1) assembly now available in the Genome
Browser</h2>
<p>
We are pleased to announce the release of a Genome Browser for the May 2012 assembly of bonobo, 
<em>Pan paniscus</em> (Max-Planck Institute panpan1, UCSC version panPan1). The assembly was 
provided by the <a href="http://www.eva.mpg.de/" target="_blank">Max-Planck Institute for 
Evolutionary Anthropology</a>. There are 10,867 scaffolds with a total size of 2,869,190,071 
bases.</p>
<p>
Bulk downloads of the sequence and annotation data are available via the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/panPan1/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#bonobo">Downloads</a> page. These data have
<a href="credits.html#bonobo_credits">specific conditions for use</a>. The bonobo (panPan1) browser 
annotation tracks were generated by UCSC and collaborators worldwide. See the 
<a href="credits.html#bonobo_credits">Credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p>

<a name="061215"></a>
<h2>Jun. 12, 2015 &nbsp;&nbsp; Data Integrator</h2>
<p>
Have you ever wished that the Table Browser could associate your custom track items with some other 
track, while retaining the item names from both? We have released a new tool that can do just that, 
and more: the Data Integrator.</p>
<p>
Using the Data Integrator you can select up to five tracks, including custom tracks
and hub tracks, and output tab-separated text that contains all the fields from both
the first track and the overlapping items from the other track(s). If you need only
a subset of the fields from your selected tracks, the Data Integrator allows you to
choose which fields to include in the output. You can also query your current Genome
Browser position, search for positions of items such as genes, SNPs, etc. by name,
query over a set of up to 1000 positions, and even retrieve genome-wide data
(subject to data size). The Data Integrator output can be viewed in your web browser
window or downloaded to an optionally gzip-compressed file. Look for &quot;Data Integrator&quot;
in the Genome Browser Tools menu (top blue bar) if you'd like to try it out. </p>
<p>
For more information about the Data Integrator's capabilities and usage, see the
user's guide: <a href="http://genome.ucsc.edu/goldenPath/help/hgIntegratorHelp.html">
http://genome.ucsc.edu/goldenPath/help/hgIntegratorHelp.html</a>.
As always, we welcome questions and feedback on our public mailing list:
<a href="mailto:&#103;&#101;&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.uc&#115;c.&#101;&#100;&#117;">&#103;&#101;&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.uc&#115;c.&#101;&#100;&#117;</a>
<!-- above address is genome at soe.ucsc.edu --> </p>
<p>
Thanks to Angie Hinrichs and Matthew Speir for all of their work on this feature.</p>

<a name="052815"></a>
<p>
<h2>May 28, 2015 &nbsp;&nbsp; New UCSC Genes track released for GRCm38/mm10</h2>
<p>
We're happy to announce the release of an updated UCSC Genes track for the GRCm38/mm10
mouse Genome Browser. The new release has 63,244 total transcripts, compared with 61,642
in the previous version. The total number of canonical genes has increased from 32,408
to 32,958. Comparing the new gene set with the previous version:</p>
<ul>
  <li>
  59,119 transcripts did not change between versions.</li>
  <li>
  55 transcripts were not carried forward to the new version.</li>
  <li>
  2,308 transcripts are &quot;compatible&quot; with those in the previous set, meaning
  that the two transcripts show consistent splicing.  In most cases, the old and new
  transcripts differ in the lengths of their UTRs.</li>
  <li>
  160 transcripts overlap with those in the previous set, but do not show consistent
  splicing (i.e., they contain overlapping introns with differing splice sites).</li>
</ul>
<p>
More details about the new UCSC Genes track can be found on the 
<a href="../cgi-bin/hgTrackUi?db=mm10&g=knownGene">UCSC Genes track description page</a>.</p>
<p>
Many thanks to Brian Raney and Matthew Speir for their work on this track.</p>

<a name="051515"></a>
<h2>May 15, 2015 &nbsp;&nbsp; Work for us! New job openings on the UCSC Genome Browser project</h2>
<p>
The UCSC Genome Browser project team is looking for two talented people to join our
engineering staff based in Santa Cruz, CA.</p>
<p>
The Software Engineer will work with a small engineering team to support and extend the UCSC Genome 
Browser database and software, while interfacing with Genome Browser collaborators and users 
worldwide. Our engineering work requires solid experience with C, UNIX and SQL. For application 
information and more details, see <a href="https://jobs.ucsc.edu/applicants/Central?quickFind=68100"
target="_blank">Job #150568</a> on the UCSC staff employment page. </p> 
<p>
The Quality and User Support Analyst will work with a small QA group that interacts closely 
with browser software developers and genomics researchers throughout the world. As
a member of the QA team, you will participate in all aspects of quality assurance for 
genomics projects staffed by our group and provide technical support for the affiliated 
user communities. For application information and more details about the position,
see <a href="https://jobs.ucsc.edu/applicants/Central?quickFind=68101"
target="_blank">Job #1505685</a> on the UCSC staff employment page. </p>
<p>
If you'd like to apply for one of these positions, be sure to submit your application materials by 
May 25, 2015, for maximum consideration.</p>

<a name="040815"></a> 
<h2>Apr. 8, 2015 &nbsp;&nbsp; bigGenePred format now supported in the Genome Browser</h2>
<p>
We are pleased to announce that the UCSC Genome Browser now supports data in bigGenePred format. 
bigGenePred format stores annotation items that are a linked collection of exons, much as BED files 
do, but bigGenePred has additional information about the coding frames and other gene specific 
information. bigGenePred files are created initially from BED-type files with some extra fields, 
using the program bedToBigBed with a special AutoSql file that defines the fields of the 
bigGenePred. The resulting bigBed files are in xed binary format. The advantage of these bigBed 
files is that only portions of the files needed to display a particular region are transferred to 
UCSC. So for large data sets, bigBed is considerably faster than regular BED files.</p>
<p>
This new format is available for use in custom tracks and data hubs. For more information about 
bigGenePred, please see our 
<a href="help/bigGenePred.html">bigGenePred Track Format help page</a>.</p>

<a name="040215b"></a>
<h2>Apr. 2, 2015 &nbsp;&nbsp; New tarsier (tarSyr2) assembly now available in the Genome
Browser</h2>
<p>
We are pleased to announce the release of a Genome Browser for the Sep. 2013 assembly of tarsier, 
<em>Tarsius syrichta</em> (WashU Tarsius_syrichta-2.0.1, UCSC version tarSyr2). The whole genome 
shotgun assembly was provided by <a href="http://genome.wustl.edu/genomes/detail/tarsius-syrichta/"
target="_blank">Washington University</a>. There are 337,189 scaffolds with a total size of 
3,453,864,774 bases.</p>   
<p>
Bulk downloads of the sequence and annotation data are available via the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/tarSyr2/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#tarsier">Downloads</a> page. These data have
<a href="credits.html#tarsier_credits">specific conditions for use</a>. The tarsier (tarSyr2) 
browser annotation tracks were generated by UCSC and collaborators worldwide. See the 
<a href="credits.html#tarsier_credits">Credits</a> page for a detailed list of the organizations and
individuals who contributed to this release.</p>

<a name="040215a"></a>
<h2>Apr. 2, 2015 &nbsp;&nbsp; New training video</h2>
<p>
Going crazy counting amino acids? Learn an easier way to find the codon number in our new video. 
See our <a href="../training/index.html" target="_blank">training page</a> or our
<a href="http://bit.ly/genomebrowserYoutube" target=_"blank">YouTube channel</a> for details.</p>

<a name="040115"></a>
<h2>Apr. 1, 2015 &nbsp;&nbsp; New GRCz10 zebrafish assembly now available</h2>
<p>
The most recent zebrafish assembly -- GRCz10 (UCSC version danRer10, Sep. 2014) -- is now
available in the UCSC Genome Browser. This is the first zebrafish release by the
<a href="https://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/zebrafish/"
target="_blank">Genome Reference Consortium</a>, who took over the improvement and
maintenance of the zebrafish genome assembly.</p>
<p>
The GRCz10 assembly improves upon Zv9 by utilizing optical mapping and additional meiotic
mapping to fill gaps with new clone sequence and improve the positioning of assembly
components.  This has also been supplemented with a new whole genome shotgun assembly
WGS31. The result is an assembly with a sequence length of 1.3 Gb in 26 chromosomes and
1,035 scaffolds.  For more details about the GRCz10 assembly, see the
<a href="https://www.ncbi.nlm.nih.gov/assembly/210611"
target="_blank">GRCz10 assembly page</a> or the official
<a href="https://www.ncbi.nlm.nih.gov/news/10-08-2014-zebrafish-grcz10-annotated/"
target="_blank">GRCz10 announcement</a>.</p>
<p>
The danRer10 sequence and annotation data can be downloaded from the UCSC Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/danRer10/">FTP server</a> or
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#zebrafish">downloads</a> page.</p>
<p>
We'd like to thank the Genome Reference Consortium for providing this assembly. The UCSC
zebrafish Genome Browser was produced by Hiram Clawson, Brian Raney and Steve Heitner. See 
<a href="../goldenPath/credits.html#zebrafish_credits">credits</a> page for a detailed list
of the organizations and individuals who contributed to this release.</p>

<a name="032015"></a>
<p>
<h2>Mar. 20, 2015 &nbsp;&nbsp; New visualization option for RepeatMasker</h2>
<p>
We are excited to announce the release of the new RepeatMasker Visualization track for the
hg38 and mm10 assemblies. This new RepeatMasker track displays detailed information that
includes the amount of unaligned repeat sequence and lines joining repeat fragments.
Additionally, repeats are now colored based on their repeat class. This new RepeatMasker
track incorporates a new details page that includes the detailed RepeatMasker output and
the raw alignment used to generate the final repeat annotation. You can read more about
this track on the description page for either
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=rmskJoinedBaseline">hg38</a> or
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=rmskJoinedBaseline">mm10</a>.</p>
<p>
This track is the result of a collaboration between the <a href="http://www.systemsbiology.org/" 
target="_blank">Institute for Systems Biology</a> and the UCSC Genome Browser. We would like to 
thank Robert Hubley, Arian Smit, Hiram Clawson, and Matthew Speir for their efforts in creating this
track.</p>

<a name="030315"></a>
<h2>Mar. 3, 2015 &nbsp;&nbsp; Use UDR for quick sizable downloads</h2>
<p>
The UCSC Genome Browser is pleased to share a download protocol to use when downloading large sets 
of files from our download servers: <strong>UDR (UDT Enabled Rsync)</strong>. UDR utilizes rsync
as the transport mechanism, but sends the data over the UDT protocol, which enables huge amounts of 
data to be downloaded efficiently over long distances.</p>
<p>
We now have two identical download servers to better serve your needs. You can use either one:
<ul>
  <li>
  <a href="http://hgdownload.gi.ucsc.edu" target="_blank">http://hgdownload.gi.ucsc.edu</a></li>
  <li>
  <a href="http://hgdownload-sd.sdsc.edu" target="_blank">http://hgdownload-sd.sdsc.edu</a></li>
</ul>
<p>
<strong>The background</strong></p>
<p>
Typical TCP-based protocols like http, ftp and rsync have a problem in that the further
away the download source is from you, the slower the speed becomes. Protocols like UDT/UDR
allow for many UDP packets to be sent in batch, thus allowing for much higher transmit speeds
over long distances. UDR will be especially useful for users who are downloading from
locations distant to California. <em>The US East Coast and the international community will
likely see much higher download speeds by using UDR rather than rsync, http or ftp.</em></p>
<p>
<strong>Getting UDR & setting it up on your system</strong></p>
<p>
It should be noted that UDR is not written or managed by UCSC, it was written by the
Laboratory for Advanced Computing at the University of Chicago. It has been tested to work
under Linux, FreeBSD and Mac OSX, but may work under other UNIX variants. The source code can
be obtained here, through GitHub: <a href="https://github.com/LabAdvComp/UDR" 
target="_blank">https://github.com/LabAdvComp/UDR</a></p>
<p>
If you need help building the UDR binaries or have questions about how UDR functions,
please read the documentation on the GitHub page, and if necessary, contact the UDR authors
via the GitHub page. We recommend reading the documentation on the UDR GitHub page to better
understand how UDR works. UDR is written in C++. UDR is Open Source and is released under the
Apache 2.0 License. You must first have rsync installed on your system.</p> 
<p>
For your convenience, we are offering a binary distribution of UDR for Red Hat Enterprise Linux 6.x 
and 7.x (or variants such as CentOS 6/7 or Scientific Linux 6/7). You'll find the rpm packages 
here: <a href="http://hgdownload.gi.ucsc.edu/admin/udr" 
target="_blank">http://hgdownload.gi.ucsc.edu/admin/udr</a>.</p>
<p>
For RHEL 6 we have 32-bit and 64-bit packages, but for RHEL 7 we have dropped the 32-bit and offer 
onlya 64-bit package. Those who need a 32-bit package for RHEL 7 can download the source and compile
it locally.</p>
<p>
<strong>Example of using UDR to download ENCODE data from the UCSC Genome Browser Download
Server</strong></p>
<p>
Once you have a working UDR binary, either by building from source or by installing the rpm
(if you are using RHEL 6.x or other variant), you can download files from either of our 
download servers in a fashion very similar to rsync. For example, using rsync, you may want
to download all of the ENCODE information for the mm9 database using the command:</p>
<pre><code>$ rsync -avP rsync://hgdownload.gi.ucsc.edu/goldenPath/mm9/encodeDCC/ /my/local/mm9/</code></pre>
<p>
Using UDR is very similar. The UDR syntax for downloading the same data would be:</p>
<pre><code>$ udr rsync -avP hgdownload.gi.ucsc.edu::goldenPath/mm9/encodeDCC/ /my/local/mm9/</code></pre>
<p>
If you installed the rpm, use the 'man udr' command for more information via the man page;
if you installed from source please refer to the UDR GitHub page for more details on the
capabilities of UDR and how to use it.</p>
<p>
<strong>Firewall Considerations</strong></p>
<p>
UDR establishes connections on TCP/9000, then transmits the data stream over UDP/9000-9100.
Your institution may need to modify its firewall rules to allow inbound and outbound ports
TCP/9000 and UDP/9000-9100 from either of the two download machines.</p>
<p>
If you decide to install and use UDR, we hope that you experience greatly increased download
speeds. If you have difficulties installing UDR on your system, please contact the Laboratory
for Advanced Computing through their github page: <a href="https://github.com/LabAdvComp/UDR"
target="_blank">https://github.com/LabAdvComp/UDR</a>.</p>

<a name="022615"></a>
<h2>Feb. 26, 2015 &nbsp;&nbsp; New training video</h2>
<p>
Ever wondered how to find the tables associated with your favorite Browser track? Find out how in 
our new training video. See our <a href="../training/index.html" target="_blank">training page</a>, 
our <a href="http://bit.ly/genomebrowserYoutube" target=_"blank">YouTube channel</a>, and our
<a href="https://genome-blog.gi.ucsc.edu/blog/introducing-the-genome-browser-youtube-channel/"
target="_blank">blog</a> for details.</p>

<a name="021215"></a>
<h2>Feb. 12, 2015 &nbsp;&nbsp; Blat your assembly hub</h2>
<p>
Assembly data hubs, track hubs that allow researchers to annotate genomes that are not in the UCSC 
Genome Browser, can now use blat to quickly find DNA and protein sequences in their unique 
assemblies.</p>
<p>
While running remote blat servers with the gfServer utility, you can now add lines to your assembly 
hub's genomes.txt file to inform the Browser where to send blat searches. Or by downloading and 
installing a virtual machine Genome Browser In a Box <a href="help/gbib.html" 
target="_blank">(GBiB)</a>, you can alternatively locally run gfServers, preinstalled on GBiB, all 
from your laptop. For more information see the 
<a href="http://genomewiki.ucsc.edu/index.php/Assembly_Hubs" target="_blank">Assembly Hub Wiki</a> 
and the new <a href="help/hubQuickStartAssembly.html" target="_blank">Quick Start Guide to Assembly 
Hubs</a>.</p>

<a name="021115"></a>
<p>
<h2>Feb. 11, 2015 &nbsp;&nbsp; dbSNP 142 available for hg19 and hg38</h2>
<p>
We are pleased to announce the release of four tracks derived from NCBI 
<a href="https://www.ncbi.nlm.nih.gov/SNP/" target="_blank">dbSNP</a> Build 142 data, available on 
the two most recent human assemblies GRCh37/hg19 and GRCh38/hg38. The new tracks contain a 
substantial amount of additional annotation data not included in previous dbSNP tracks, with 
corresponding coloring and filtering options in the Genome Browser.</p>
<p>
There are four SNP tracks available as part of this release. One is a track containing all mappings 
of reference SNPs to the human assembly, labeled &quot;All SNPs (142)&quot; The other three tracks 
are subsets of this track and show interesting and easily defined subsets of dbSNP:</p>
<ul>
  <li>
  Common SNPs (142): uniquely mapped variants that appear in at least 1% of the population or are 
  100% non-reference</li>
  <li>
  Flagged SNPs (142): uniquely mapped variants, excluding Common SNPs, that have been flagged by
  dbSNP as &quot;clinically associated&quot;</li>
  <li>
  Mult. SNPs (142): variants that have been mapped to more than one genomic location</li>
</ul>
<p>
By default, only the Common SNPs (142) are visible; other tracks must be made visible using the 
track controls. You will find the other SNPs (142) tracks on both of GRCh37/hg19 and GRCh38/hg38 
browsers in the Variation group.</p>
<p>
The tracks were produced at UCSC by Angie Hinrichs and Jonathan Casper. We'd like to thank the 
dbSNP group at NCBI for providing access to these data.</p>
<p>
As part of this release, we are also retiring the older dbSNP Build 135 and 137 data from display 
on the GRCh37/hg19 human assembly. Those tracks will still be available for viewing on our
<a href="http://genome-preview.gi.ucsc.edu" target="_blank">preview server</a>, and the associated 
masked FASTA files will continue to be available on our <a href="http://hgdownload.gi.ucsc.edu" 
target="_blank">download server</a>. </p>

<a name="020215"></a>
<h2>Feb. 2, 2015 &nbsp;&nbsp; Host a Genome Browser workshop</h2>
<p>
New timeslots are now available to host a Genome Browser workshop at your institution. Thanks to the
funding support of NHGRI, we offer hands-on Genome Browser training onsite at your institution, 
tailored to your audience's level of expertise.</p>
<p>
For more information or to submit a request to host a workshop, please visit 
<a href="http://bit.ly/ucscTraining" target="_blank">our sign-up page</a>.</p>

<a name="012315"></a>
<h2>Jan. 23, 2015 &nbsp;&nbsp; Genome Browser YouTube channel</h2>
<p>
We are pleased to announce the release of the UCSC Genome Browser
<a href="http://bit.ly/genomebrowserYoutube" target=_"blank">YouTube channel</a>. The channel 
contains short videos showing how to use the Genome Browser and associated tools to solve selected 
problems, supplementing the other materials on our 
<a href="http://bit.ly/genomebrowserTraining">training page</a>. The videos were produced by Robert 
Kuhn and Pauline Fujita, with assistance from Luvina Guruvadoo, Jonathan Casper, Matt Speir and the 
rest of the Browser staff.</p>
<p>
If you have any comments or questions, including requests for future video topics, please send them 
to our training group:
&#117;&#99;&#115;&#99;g&#98;&#116;&#114;&#97;&#105;n&#105;n&#103;&#64;&#115;&#111;&#101;.&#117;c&#115;c.&#101;&#100;&#117;.</p>
<!-- above address is ucscgbtraining at soe.ucsc.edu -->
        
<a name="011515"></a>
<h2>Jan. 15, 2015 &nbsp;&nbsp; Proteomics data now available in Genome Browser</h2>
<p>
We are pleased to announce the release of proteomics data for the human hg19 assembly. Data from the
National Cancer Institute's (NCI) <a href="http://proteomics.cancer.gov/programs/cptacnetwork" 
target="_blank">Clinical Proteomic Tumor Analysis Consortium (CPTAC)</a> is now available in the 
UCSC Genome Browser as a public track hub. This track hub contains peptides that were identified by 
CPTAC in their deep mass spectrometry based characterization of the proteome content of breast, 
colorectal and ovarian cancer biospecimens that were initially sequenced by 
<a href="http://cancergenome.nih.gov/" target="_blank">The Cancer Genome Atlas</a>. This effort 
extends the accessibility of CPTAC data to more researchers and provides an additional level of 
analysis to assist the cancer biology community.</p>
<p>
To access and view this hub, navigate to the <a href="../cgi-bin/hgHubConnect" 
target="_blank">Track Hub gateway page</a> and select &quot;CPTAC Hub v1&quot; from the Public Hubs 
list.  Please direct any queries to the Fenyo Lab at 
<a href="mailto:info@fenyolab.org">info@fenyolab.org</a>.</p>
<p>
In addition, we have also released a <a href="http://www.peptideatlas.org/" 
target="_blank">PeptideAtlas</a> track which displays peptide identifications from the
<a href="https://db.systemsbiology.net/sbeams/cgi/PeptideAtlas/buildDetails?atlas_build_id=433" 
target="_blank">PeptideAtlas August 2014 (Build 433) Human build</a>. This build, based on 971 
samples containing more than 420 million spectra, identified over a million distinct peptides 
covering more than 15,000 canonical proteins. To read more about this track, please see the 
<a href="../cgi-bin/hgTrackUi?db=hg19&g=peptideAtlas" 
target="_blank">track description page</a>. Many thanks to Eric Deutsch, Zhi Sun, and the 
PeptideAtlas team at the Institute for Systems Biology, Seattle for providing this data.</p>

<!-- ============= 2014 archived news ============= -->
<a name="2014"></a>
<a name="121914"></a>
<h2>Dec. 19, 2014 &nbsp;&nbsp; New cow (bosTau8) assembly now available in the Genome Browser</h2>
<p> 
We are pleased to announce the release of a Genome Browser for the June 2014 assembly
of cow, <i>Bos taurus</i> (Bos_taurus_UMD 3.1.1, UCSC version bosTau8). This updated cow 
assembly was provided by the  <a href="http://www.cbcb.umd.edu/"
target="_blank">UMD Center for Bioinformatics and Computational Biology</a> (CBCB). This assembly is
an update to the previous UMD 3.1 (bosTau6) assembly.  UMD 3.1 contained 138 unlocalized contigs 
that were found to be contaminants.  These have been suppressed in UMD 3.1.1.</p>
<p>
Bulk downloads of the sequence and annotation data are available via the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/bosTau8/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#cow">Downloads</a> page. These data have
<a href="/goldenPath/credits.html#cow_credits">specific conditions for use</a>. The cow browser 
annotation tracks were generated by UCSC and collaborators worldwide. See the
<a href="/goldenPath/credits.html#cow_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>
 
<a name="121214"></a>
<h2>Dec. 12, 2014 &nbsp;&nbsp; New <em>D. melanogaster</em> (dm6) assembly now available</h2>
<p>
We are pleased to announce the release of a Genome Browser for the August 2014 assembly of 
<em>Drosophila melanogaster</em> (BDGP Release 6, UCSC version dm6). This updated <em>D.
melanogaster</em> assembly was provided by the  <a href="http://www.fruitfly.org/"
target="_blank">Berkeley Drosophila Genome Project</a> (BDGP). Please see
<a href="http://flybase.org/static_pages/feature/previous/articles/2014_07/FB2014_04.html"
target="_blank">this announcement</a> on the FlyBase website for more details about this
updated <em>D. melanogaster</em> assembly.</p>
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/dm6/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#fruitfly">Downloads</a> page. These data have
<a href="/goldenPath/credits.html#dm_credits">specific conditions for use</a>.
The <i>D. melanogaster</i> (dm6) browser annotation tracks were generated by UCSC and collaborators 
worldwide. See the
<a href="/goldenPath/credits.html#dm_credits">Credits</a> page for a detailed
list of the organizations and individuals who contributed to this release.</p>

<a name="102914"></a>
<h2>Oct. 9, 2014 &nbsp;&nbsp; Genome Browser in a Box (GBiB)</h2>
<p>
Sometimes you just want to keep your genomics data to yourself. Have you ever hesitated when 
uploading your data set into the UCSC Genome Browser? If so, you'll be happy to know that we have 
created a stand-alone personal version: Genome Browser in a Box (GBiB). If you have sensitive 
genomics data that you would like to view securely on your own laptop in the context of the UCSC
Genome Browser, GBiB is for you.</p>
<p>
GBiB is an easy-to-install personal copy of the Genome Browser that comes preloaded with the most 
popular annotation tracks for human. It is highly configurable &mdash; you can access or download 
other annotation tracks of interest, or view any of the other 90+ organisms featured in the public
Genome Browser. GBiB runs inside of Oracle's free VirtualBox virtual machine. It has the same core 
functionality as the UCSC Genome Browser, but keeps your data private and local to your own 
computer.</p>
<p>
GBiB is free for non-profit academic research and for personal use. Commercial use requires 
purchase of a license with setup fee and annual payment. Download or purchase GBiB in our secure 
online <a href="https://genome-store.ucsc.edu/">store</a>.</p>
<p>
You can read more about GBiB on our 
<a href="https://genome-blog.gi.ucsc.edu/blog/genome-browser-in-a-box-gbib-origins/">blog</a> or in
our <a href="help/gbib.html">documentation</a>.</p>

<a name="102014"></a>
<h2>Oct. 20, 2014 &nbsp;&nbsp; dbSNP 141 available for hg19 and hg38</h2>
<p>
We are pleased to announce the release of four tracks derived from NCBI
<a href="https://www.ncbi.nlm.nih.gov/SNP/" target="_blank">dbSNP</a> Build 141 data, available on 
the two most recent human assemblies GRCh37/hg19 and GRCh38/hg38. The new tracks contain additional 
annotation data not included in previous dbSNP tracks, with corresponding coloring and filtering 
options in the Genome Browser.</p>
<p>
There are three SNP tracks available for the GRCh37/hg19 assembly. One is a track containing all 
mappings of reference SNPs to the human assembly, labeled &quot;All SNPs (141)&quot;. The other two 
tracks are subsets of this track and show interesting and easily defined subsets of dbSNP:</p>
<ul>
  <li>
  Common SNPs (141): uniquely mapped variants that appear in at least 1% of the population or are 
  100% non-reference</li>
  <li>
  Flagged SNPs (141): uniquely mapped variants, excluding Common SNPs, that have been flagged by 
  dbSNP as &quot;clinically associated&quot;li>
</ul>
<p>
Unlike previous dbSNP releases for GRCh37/hg19, Build 141 does not include mappings to alternate 
haplotype sequences. In addition, variants that map to multiple genomic locations (e.g. to different
chromosomes, or more than one position on the same chromosome) have been removed.</p>
<p>
This release also marks the first set of dbSNP data available for the newest human assembly, 
GRCh38/hg38. In addition to the three tracks described for hg19, there is one additional track 
available for hg38:</p>
<ul>
  <li>
  Mult. SNPs (141): variants that have been mapped to more than one genomic location</li>
</ul>
<p>
By default, only the Common SNPs (141) are visible; other tracks must be made visible using the 
track controls. You will find the other SNPs (141) tracks on both of GRCh37/hg19 and GRCh38/hg38 
browsers in the Variation group.</p>
<p>
The tracks were produced at UCSC by Angie Hinrichs and Matthew Speir. We'd like to thank the dbSNP 
group at NCBI for providing access to these data.</p>

<a name="100714"></a>
<h2>Oct. 7, 2014 &nbsp;&nbsp; Introducing the UCSC Genome Browser blog</h2>
<p>
We're excited to introduce a new <a href="http://genome.ucsc.edu/blog" target="_blank">blog</a> 
that will feature posts by Genome Browser staff and guests. On the blog we'll be publishing 
in-depth information about UCSC Genome Browser features, tools, projects and related topics that we 
hope people will find both useful and interesting. We just published our inaugural post written by 
Jim Kent, the UCSC Genome Browser Architect and PI, on the
<a href="https://genome-blog.gi.ucsc.edu/blog/2014-ebola-epidemic/" target="_blank">2014 Ebola 
Epidemic</a>.</p>
<p>
If you have a suggestion for a blog post topic, please let us know via our
<a href="/cgi-bin/hgUserSuggestion">suggestion box</a>.</p>

<a name="093014"></a>
<h2>Sep. 30, 2014 &nbsp;&nbsp; Genome Browser and portal released for the Ebola Virus</h2>
<p>
We are excited to announce the release of a Genome Browser and information portal for the 
Jun. 2014 assembly of the Ebola virus (UCSC version eboVir3, GenBank accession 
<a href="https://www.ncbi.nlm.nih.gov/nuccore/KM034562" target="_blank">KM034562</a>) 
submitted by the Broad Institute. We have worked closely with the Pardis Sabeti lab at 
the Broad Institute and other Ebola experts throughout the world to incorporate
annotations that will be useful to those studying Ebola. Annotation tracks included in this
initial release include genes from <a href="https://www.ncbi.nlm.nih.gov/"
target="_blank">NCBI</a>, B- and T-cell epitopes from the <a href="http://www.iedb.org/"
target="_blank">IEDB</a>, structural annotations from <a href="http://www.uniprot.org/"
target="_blank">UniProt</a> and a wealth of SNP data from the
<a href="https://www.ncbi.nlm.nih.gov/pubmed/25214632" target="_blank">2014 publication</a>
by the Sabeti lab. This initial release also contains a 160-way alignment
comprising 158 Ebola virus sequences from various African outbreaks and 2 Marburg virus sequences.
You can find links to the Ebola virus Genome Browser and more information on the Ebola virus
itself on our <a href="../ebolaPortal/">Ebola Portal</a> page.</p>
<p>
Bulk downloads of the sequence and annotation data are available via the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/eboVir3/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#ebola_virus">Downloads</a> page. The Ebola 
virus (eboVir3) browser annotation tracks were generated by UCSC and collaborators worldwide. See 
the <a href="credits.html#ebola_credits">Credits</a> page for a detailed list of the organizations 
and individuals who contributed to this release and conditions for use of these data.</p>

<a name="092314"></a>
<h2>Sep. 23, 2014 &nbsp;&nbsp; Ensembl Regulatory Build now available in the Genome Browser</h2>
<p>
We are pleased to announce that data from the 
<a href="http://www.ensembl.org/info/genome/funcgen/regulatory_build.html"
target="_blank">Ensembl Regulatory Build</a> is now available in the UCSC Genome Browser
as a public track hub for both hg19 and hg38. This track hub contains promoters and their
flanking regions, enhancers, and many other regulatory features predicted across a number
of cell lines using annotated segmentation states. Click
<a href="/cgi-bin/hgTracks?db=hg19&position=chr21%3A33031597-33041570&hubUrl=http://ngs.sanger.ac.uk/production/ensembl/regulation/hub.txt">here</a>
to view this hub on the hg19 assembly, or click
<a href="/cgi-bin/hgTracks?db=hg38&position=chr9%3A133252000-133280861&hubUrl=http://ngs.sanger.ac.uk/production/ensembl/regulation/hub.txt">here</a>
to view it on the hg38 assembly.</p>
<p>
We would like to thank our collaborators Daniel Zerbino, Steve Wilder, and Nathan Johnson
from Ensembl and Kate Rosenbloom and Matthew Speir from UCSC for their efforts in
bringing these resources to the UCSC Genome Browser.</p>

<a name="090414"></a>
<h2>Sep. 4, 2014 &nbsp;&nbsp; New ENCODE portal launched</h2>
<p>
The website for up-to-date information about the ENCODE project is now hosted at 
<a href="https://www.encodeproject.org" target="_blank">www.encodeproject.org</a>. The
former portal at <a href="../ENCODE/index.html">genome.ucsc.edu/ENCODE</a> is no longer maintained. 
Those UCSC ENCODE pages now archive information and tools from the ENCODE production and pilot 
phases (2003 to 2012) including live links to visualize and download data. Please navigate to the 
new ENCODE portal for recent data releases. Along with the ability to use faceted searching to 
explore all ENCODE data, the ENCODE portal provides visualization in the UCSC Genome Browser via a 
&quot;Visualize Data&quot; link on assay pages when processed data files are available.</p>

<a name="090314"></a>
<h2>Sep. 3, 2014 &nbsp;&nbsp; Data from the FaceBase Consortium now available in the Genome 
Browser</h2>
<p>
We are pleased to announce that much of the data from the <a href="https://www.facebase.org/"
target="_blank">FaceBase Consortium</a> is now available in the UCSC Genome Browser as a
public track hub. This track hub contains data from numerous experiments carried out by
the Potter and Visel labs to study craniofacial development in mice and humans, with
tracks for the mm9, mm10, hg18, and hg19 assemblies. You can check out the hub on the mm9 assembly
<a href="../cgi-bin/hgTracks?db=mm9&position=chr12%3A57795963-57815592&hubUrl=http://trackhub.facebase.org/hub.txt"
target="_blank">here</a>. In addition to this public track hub, we have a set of
microarray expression data hosted as a native track for both the mm9 and mm10 assemblies.
This expression data compares expression across the mouse genome for a number of embryonic
days and craniofacial cell types. For more information on this track, please see the track
<a href="../cgi-bin/hgTrackUi?org=mouse&db=mm9&g=FaceBase24SampleTypesAvg"
target="_blank">description page</a>.</p>
<p>
We would like to thank Robert Kuhn, Pauline Fujita, and Matthew Speir from UCSC and Thomas
Maher from the FaceBase Consortium for their efforts in bringing these resources to the
UCSC Genome Browser.</p>

<a name="073014"></a>
<h2>Jul. 30, 2014 &nbsp;&nbsp; New rat (rn6) assembly now available</h2> 
<p>
We are excited to announce the release of a Genome Browser for the July 2014 assembly
of rat, <em>Rattus norvegicus</em> (RGSC Rnor_6.0, UCSC version rn6)! This
assembly is provided by the Rat Genome Sequencing Consortium, which is comprised of
eight research organizations across the United States and Canada and led by the
<a href="https://www.hgsc.bcm.edu/" target="_blank">Baylor College of Medicine</a>.
The new RGSC Rnor_6.0 assembly contains a new, partially assembled Y chromosome
as well as improvements to other regions of the genome. You can find more information
on the RGSC's efforts to sequence rat genome on the Baylor College of Medicine's
<a href="https://www.hgsc.bcm.edu/other-mammals/rat-genome-project" target="_blank">project 
page</a>.</p>
<p>
Bulk downloads of the sequence and annotation data are available via the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/rn6/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#rat">Downloads</a> page. These data have
<a href="../goldenPath/credits.html#rat_use">specific conditions for use</a>. The rat (rn6) browser 
annotation tracks were generated by UCSC and collaborators worldwide. See the
<a href="../goldenPath/credits.html#rat_credits">Credits</a> page for a detailed
list of the organizations and individuals who contributed to this release. </p>

<a name="071514b"></a>
<p>
<h2>Jul. 15, 2014 &nbsp;&nbsp; New shrew (sorAra2) assembly available in the Genome Browser</h2>
<p>
We are pleased to announce the release of a Genome Browser for the August 2008 assembly of shrew, 
<em>Sorex araneus</em> (Broad SorAra2.0, UCSC version sorAra2). The whole genome shotgun assembly 
was provided by <a href="http://www.broadinstitute.org/" target="_blank">The Broad Institute</a>.
There are 12,845 scaffolds with a total size of 2,423,158,183 bases.</p>
<p>
Bulk downloads of the sequence and annotation data are available via the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/sorAra2/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#shrew">Downloads</a> page. These data have
<a href="/goldenPath/credits.html#shrew_credits">specific conditions for use</a>. The shrew 
(sorAra2) browser annotation tracks were generated by UCSC and collaborators worldwide. See the
<a href="/goldenPath/credits.html#shrew_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>

<a name="071514a"></a>
<h2>Jul. 15 2014 &nbsp;&nbsp; New pika (ochPri3) assembly released in the Genome Browser</h2>
<p>
We are pleased to announce the release of a Genome Browser for the May 2012 assembly of pika, 
<em>Ochotona princeps</em> (Broad OchPri3.0, UCSC version ochPri3). The whole genome shotgun 
assembly was provided by <a href="http://www.broadinstitute.org/" target="_blank">The Broad 
Institute</a>.  There are 10,420 scaffolds with a total size of 2,229,835,716 bases.</p>
<p>
Bulk downloads of the sequence and annotation data are available via the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/ochPri3/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#pika">Downloads</a> page. These data have
<a href="/goldenPath/credits.html#pika_credits">specific conditions for use</a>. The pika (ochPri3) 
browser annotation tracks were generated by UCSC and collaborators worldwide. See the
<a href="/goldenPath/credits.html#pika_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>

<a name="061914"></a>
<h2>Jun. 19, 2014 &nbsp;&nbsp; Conservation track released for rat</h2>
<p>
We are pleased to announce the release of a new Conservation track based on the rat 
(RGSC 5.0/rn5) assembly. This track shows multiple alignments of 13 vertebrate species and 
measurements of evolutionary conservation using two methods (phastCons and phyloP) from the 
PHAST package for all species. The multiple alignments were generated using multiz and other
tools in the UCSC/Penn State Bioinformatics comparative genomics alignment pipeline. Conserved 
elements identified by phastCons are also displayed in this track. For more details, visit the 
<a href="../cgi-bin/hgTrackUi?db=rn5&g=cons13way"> track description page</a>.</p>
<p>
With the release of this new conservation track, we have changed the default rat browser on 
site from rn4 to rn5. In general, the Genome Browser will continue to display
whichever rat assembly you have already been viewing, and will default to the rn5 assembly 
only for those users who haven't recently viewed a rat assembly. However, there are 
circumstances in which the assembly may update the view to the newer version, for instance 
if you reset your browser settings to the default values. If you find yourself
in a situation where some of your favorite browser tracks have &quot;disappeared&quot;, you 
may want to check that you're viewing the expected assembly.</p>
<p>
We would like to thank Pauline Fujita, Matthew Speir, and Hiram Clawson for their efforts in
creating this track.</p>

<a name="053014"></a>
<h2>May 30, 2014 &nbsp;&nbsp; New Human Variation track now available</h2>
<p>
We would like to announce the release of the new Exome Variant Server (EVS) Variants track. 
The <a href="http://evs.gs.washington.edu/EVS/" target="_blank">EVS</a>, maintained by the
<a href="http://chum.gs.washington.edu/" target="_blank">Nickerson lab</a> at the University
of Washington, contains variant annotations for over 6,500 exomes that were sequenced as 
part of the <a href="https://esp.gs.washington.edu/drupal/" 
target="_blank">NHLBI GO Exome Sequencing Project</a> (ESP). This annotation includes 
information such as function, clinical association, dbSNP rsID, average sample read depth, 
and much more. You can find more information on how the data for this track was generated on
the <a href="../cgi-bin/hgTrackUi?db=hg19&g=evsEsp6500">track description page</a>.</p>
<p>
We would like to acknowledge Tim Dreszer and Matthew Speir for their hard work in creating 
this track.</p>

<a name="052714"></a>
<h2>May 27, 2014 &nbsp;&nbsp; New Roadmap Epigenomics Integrative Analysis Hub now available</h2>
<p>
We are pleased to announce the addition of the
<a href="../cgi-bin/hgTracks?db=hg19&hubUrl=http://vizhub.wustl.edu/VizHub/RoadmapIntegrative.txt">
Roadmap Epigenomics Integrative Analysis Hub</a> to our list of publicly available track hubs.
This new hub, produced by <a href="http://vizhub.wustl.edu/" target="blank">VizHub</a> at 
Washington University in St. Louis (WUSTL), contains hundreds of tracks that cover the wide 
range of epigenomic data available from the <a href="http://www.roadmapepigenomics.org/" 
target="blank">Roadmap Epigenomics Project</a>. This hub contains data from over 40 
different assays carried out on over 250 different cell and sample types. These assays 
include RNA-seq, MeDIP-seq, MRE-seq, methylC-seq, DNase hypersensitivity, chromHMM 
segmentation, ChIP-seq, and more than 30 histone modifications. The data contained in this 
hub is complementary to the ENCODE data available at UCSC and in the ENCODE Integrative 
Analysis public hub.</p>
<p>
You can view more data produced by the Roadmap Epigenomics Project in their other publicly 
le track hub, the Roadmap Epigenomics Data Complete Collection. You can use these two
gether to access all the available Roadmap Epigenomics Project data using the UCSC 
Browser. To access and view these hubs, navigate to the <a href="../cgi-bin/hgHubConnect">Track Hub 
gateway page</a> and select both the Roadmap Epigenomics Integrative Analysis Hub and Roadmap 
Epigenomics Data Complete Collection Hub from the Public Hubs list.</p>
<p>
Any queries about the data should be directed to
<a href="mailto:&#116;w&#97;&#110;&#103;&#64;&#103;&#101;&#110;&#101;t&#105;&#99;&#115;.&#119;&#117;&#115;&#116;&#108;.&#101;d&#117;">
Ting Wang</a> from VizHub.
<!-- above address is twang at genetics.wustl.edu -->
</p>

<a name="051514"></a>
<h2>May 15, 2014 &nbsp;&nbsp; New UCSC Genes track released for GRCh38/hg38</h2>
<p>
We're happy to announce the release of a new UCSC Genes track for the GRCh38/hg38 human Genome 
Browser.  The new release has 104,178 transcripts, compared with 82,960 in the previous
version on hg19. The total number of canonical genes has increased from 31,848 to 48,424. Comparing 
the new gene set with the previous version:</p>
<ul>
  <li>
  69,691 transcripts did not change between versions.</li>
  <li>
  619 transcripts were not carried forward to the new version.</li>
  <li>
  8,879 transcripts are "compatible" with those in the previous set, meaning that the two 
  transcripts show consistent splicing.</li>
  <li>
  3,393 transcripts overlap with those in the previous set.</li>
</ul>
<p>
More details about the new UCSC Genes track can be found on the
<a href="../cgi-bin/hgTrackUi?db=hg38&g=knownGene" 
>UCSC Genes track description page</a>.</p>
<p>
Many thanks to Brian Raney, Jim Kent, and Luvina Guruvadoo for their work on this track.</p>

<a name="050914"></a>
<h2>May 9, 2014 &nbsp;&nbsp; Three new assemblies now available in the Genome Browser</h2>
<p>
A Genome Browser is now available for the Chinese hamster (<em>Cricetulus griseus</em>) assembly 
released July 2013 by the <a href="http://www.genomics.cn/en/index" target="_blank">Beijing Genomics
Institution-Shenzhen</a> (BGI version C_griseus_v1.0, UCSC version criGri1). For more information 
and statistics about this assembly, see the NCBI assembly record for 
<a href="https://www.ncbi.nlm.nih.gov/assembly/40911" target="_blank">C_griseus_v1.0</a>. There are 
52,711 scaffolds with a total size of 2,360,146,428 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/criGri1/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#chinese_hamster">Downloads</a> page.
Please observe the <a href="credits.html#chinese_hamster_use">conditions for use</a> when accessing 
and using these data sets. The annotation tracks for this browser were generated by UCSC and 
collaborators woldwide. See the <a href="credits.html#chinese_hamster_credits">Credits</a> page for 
a detailed list of the organizations and individuals who contributed to this release.</p>
<hr>
<p>
A Genome Browser is now available for the hedgehog (<em>Erinaceus europaeus</em>) assembly released 
May 2012 by the <a href="http://www.broadinstitute.org/" target="_blank">Broad Institute</a>
(Broad version EriEur2.0, UCSC version eriEur2). For more information and statistics about this 
assembly, see the NCBI assembly record for <a href="https://www.ncbi.nlm.nih.gov/assembly/426148" 
target="_blank">EriEur2.0</a>. There are 5,802 scaffolds with a total size of 2,715,720,925 
bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/eriEur2/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#hedgehog">Downloads</a> page. Please observe 
the <a href="credits.html#hedgehog_use">conditions for use</a> when accessing and using these data 
sets.  The annotation tracks for this browser were generated by UCSC and collaborators woldwide. 
See the <a href="credits.html#hedgehog_credits">Credits</a> page for a detailed list of the
organizations and individuals who contributed to this release.</p>
<hr>
<p>
A Genome Browser is now available for the zebra finch (<em>Taeniopygia guttata</em>) assembly 
released February 2013 by <a href="http://genome.wustl.edu/genomes/detail/taeniopygia-guttata/"
target="_blank">The Genome Institute at Washington University</a> (WUSTL version 
Taeniopygia_guttata-3.2.4, UCSC version taeGut2). For more information and statistics about this 
assembly, see the NCBI assembly record for <a href="https://www.ncbi.nlm.nih.gov/assembly/524908" 
target="_blank">Taeniopygia_guttata-3.2.4</a>. There are 37,096 scaffolds with a total size of 
1,232,135,591 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/taeGut2/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#zebra_finch">Downloads</a> page. Please 
observe the <a href="credits.html#zebra_finch_use">conditions for use</a> when accessing and using 
these data sets.  The annotation tracks for this browser were generated by UCSC and collaborators 
woldwide. See the <a href="credits.html#zebra_finch_credits">Credits</a> page for a detailed list of
the organizations and individuals who contributed to this release.</p>

<a name="050814"></a>
<h2>May 8, 2014 &nbsp;&nbsp; New elephant shark (calMil1) assembly now available</h2>
<p>
A Genome Browser is now available for the elephant shark (<em>Callorhinchus milii</em>)
assembly released December 2013 by the <a href="http://www.imcb.a-star.edu.sg/"
target="_blank">Singapore Institute of Molecular and Cell Biology</a> (IMCB version 
Callorhinchus_milii-6.1.3, UCSC version calMil1). For more information and statistics about this 
assembly, see the NCBI assembly record for <a href="https://www.ncbi.nlm.nih.gov/assembly/85971" 
target="_blank"> Callorhinchus_milii-6.1.3</a>. There are 21,204 scaffolds with a total size of 
974,498,586 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/calMil1/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#elephant_shark">Downloads</a> page.
Please observe the <a href="credits.html#elephant_shark_use">conditions for use</a> when accessing 
and using these data sets. The annotation tracks for this browser were generated by UCSC and 
collaborators woldwide. See the <a href="credits.html#elephant_shark_credits">Credits</a> page for 
a detailed list of the organizations and individuals who contributed to this release.</p>

<a name="050614"></a>
<h2>May 6, 2014 &nbsp;&nbsp; Introducing the new stacked overlay for multi-wiggle tracks</h2>
<p>
We are excited to announce the release of the new stacked overlay method for multi-wiggle tracks. 
This new stacked overlay method displays each graph stacked on top of each other.</p>
<p>
<img src="../images/stackedMultiWig_screenshot.png" alt="Stacked screenshot" width="819" 
height="313"></p>
<p>
Navigate to the track settings for 
<a href="../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRegMarkH3k27ac">H3K27Ac Mark</a> on the human hg19 
assembly to view this new stacked overlay method on a multi-wiggle track. On the track settings 
page, from the &quot;Overlay method&quot; drop-down menu select &quot;stacked&quot;. Make sure
that the display mode is not set to &quot;hide&quot;, then click Submit.</p>
<p>
Credit for the implementation and testing of this new feature goes to Brian Raney and Matthew 
Speir.</p>

<a name="042914"></a>
<h2>Apr. 29, 2014 &nbsp;&nbsp; 100-species Conservation track re-release</h2>
<p>
We released the initial version of the 100-species Conservation track for the hg19 human assembly 
in Nov.2013. Over the past few months, we found a couple of inconsistencies and decided, for the 
integrity of the data, that we should rerun the computation pipeline and re-release the data. The 
fixes required minimal changes, and resulted in a more precise data set. We published the new 
version of the hg19 100-species Conservation track on 15 April 2014.</p>
<p>
Here are more details about the two fixes that we made:</p>
<ol>
  <li>
  We tracked down the cause of some unusual gaps in the alignments to a bug in the multiz program 
  used in our alignment pipeline. The Penn State Bioinformatics Group provided us with a fixed 
  version of multiz that we used to rerun the pipeline. </li>
  <li>
  We inadvertently used the pre-release assembly for one of the species in the 100-way alignment: 
  the panda. We replaced this assembly with the correct version and verified that the data sets for 
  all other species were correct before rerunning the pipeline.</li>
</ol>

<a name="041814"></a>
<h2>Apr. 18, 2014 &nbsp;&nbsp; New motif displays for transcription factor ChIP-seq track and new 
genome segmentations from ENCODE</h2>
<p>
The Browser's default displayed Transcription Factor ChIP-seq track is updating to the
<a href="../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRegTfbsClusteredV3">latest data 
release</a>, which has been enhanced with the display of Factorbook motifs. Within a cluster, a 
green highlight indicates the highest scoring site of a Factorbook-identified canonical motif for 
the corresponding factor. Upon clicking a transcription factor's cluster with a motif, the 
details page now displays the motif's sequence logo, alignment and underlying Positional 
Weight Matrix. Also, the track configuration page now enables the filtering of factors.</p>
<p>
The newly added 
<a href="../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeAwgSegmentation">
Genome Segmentations from ENCODE</a> tracks display multivariate genome-segmentation performed on
<a target=ucscVocab href="../cgi-bin/hgEncodeVocab?term=GM12878,H1-hESC,K562,HeLa-S3,HepG2,HUVEC">six human cell types</a>
(GM12878, K562, H1-hESC, HeLa-S3, HepG2, and HUVEC), integrating ChIP-seq data for
<a target=ucscVocab href="../cgi-bin/hgEncodeVocab?term=CTCF,H3K4me1,H3K4me2,H3K4me3,H3K27ac,H3K9ac,H3K36me3,H4K20me1,H3K27me3,Pol2,Input">
eight chromatin marks, RNA Polymerase II, the CTCF transcription factor and input data</a>. 
In total, twenty-five states were used to segment the genome, and these states were then grouped and
colored to highlight predicted functional elements. These Genome Segmentations are the same data as
found in the Analysis Working Group Hub, but are now hosted natively in the Browser with 
enhanced filtering capability where desired segmented states can be selected using the
Filter by Segment Type' control on the track configuration page.</p>

<a name="041514"></a>
<h2>Apr. 15, 2014 &nbsp;&nbsp; New minke whale (balAcu1) assembly now available</h2>
<p>
A Genome Browser is now available for the minke whale (Balaenoptera acutorostrata scammoni)
assembly released October 2013 by the <a href="http://eng.kiost.ac/kordi_eng/main/" target="_BLANK">
Korea Ocean Research &amp; Development Institute</a> (KORDI version BalAcu1.0, UCSC version 
balAcu1). For more information and statistics about this assembly, see the NCBI assembly record for
<a href="https://www.ncbi.nlm.nih.gov/assembly/78761" target="_blank">BalAcu1.0</a>. There are
10,776 scaffolds with a total size of 2,431,687,698 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/balAcu1/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#minke_whale">Downloads</a> page. Please
observe the <a href="credits.html#minke_whale_use">conditions for use</a> when accessing and
using these data sets. The annotation tracks for this browser were generated by UCSC and
collaborators worldwide. See the <a href="credits.html#minke_whale_credits">Credits</a>
page for a detailed list of the organizations and individuals who contributed to this release.</p>

<a name="041114"></a>
<h2>Apr. 11, 2014 &nbsp;&nbsp; New UCSC Genes track released for GRCm38/mm10</h2>
<p>
We're happy to announce the release of an updated UCSC Genes track for the GRCm38/mm10
mouse Genome Browser. The new release has 61,642 total transcripts, compared with 59,121
in the previous version. The total number of canonical genes has increased from 31,469
to 32,408. Comparing the new gene set with the previous version:</p>
<ul>
  <li>
  54,714 transcripts did not change between versions.</li>
  <li>
  19 transcripts were not carried forward to the new version.</li>
  <li>
  3,866 transcripts are &quot;compatible&quot; with those in the previous set, meaning that the two 
  transcripts show consistent splicing. In most cases, the old and new transcripts differ in the 
  lengths of their UTRs.</li>
  <li>
  522 transcripts overlap with those in the previous set, but do not show consistent splicing 
  (i.e., they contain overlapping introns with differing splice sites).</li>
</ul>
<p>
More details about the new UCSC Genes track can be found on the 
<a href="../cgi-bin/hgTrackUi?db=mm10&g=knownGene">UCSC Genes track description page</a>.</p>
<p>
Many thanks to Brian Raney and Steve Heitner for their work on this track.</p>

<a name="030614"></a>
<h2>Mar. 6, 2014 &nbsp;&nbsp; The new GRCh38 Human Genome Browser is here</h2>
<p>
In the final days of 2013, the <a href="https://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/" 
target="_blank">Genome Reference Consortium</a> (GRC) released the eagerly awaited GRCh38 
human genome assembly, the first major revision of the human genome in more than four years. During 
the past two months, the UCSC team has been hard at work building a browser that will let our users 
explore the new assembly using their favorite Genome Browser features and tools. Today we're 
announcing the release of a preliminary browser on the GRCh38 assembly. Although we still have 
plenty of work ahead of us in constructing the rich feature set that our users have come to expect, 
this early release will allow you to take a peek at what's new.</p> 
<p>
Starting with this release, the UCSC Genome Browser version numbers for human assemblies will match 
those of the GRC to minimize version confusion. Hence, the GRCh38 assembly is referred to as hg38 in
Genome Browser datasets and documentation. We've also made some slight changes to our chromosome 
naming scheme that affect primarily the names of haplotype chromosomes, unplaced contigs and 
unlocalized contigs. For more details about this, see the 
<a href="../cgi-bin/hgGateway?db=hg38">hg38 gateway page</a>.</p>
<p>
<strong>What's new in GRCh38?</strong></p> 
<ul>
  <li>
  <strong>Alternate sequences</strong> - Several human chromosomal regions exhibit sufficient 
  variability to prevent adequate representation by a single sequence. To address this, the 
  GRCh38 assembly provides alternate sequence for selected variant regions through the 
  inclusion of  <em>alternate loci scaffolds</em> (or <em>alt loci</em>). Alt loci are 
  separate accessioned sequences that are aligned to reference chromosomes. This assembly 
  contains 261 alt loci, many of which are associated with the LRC/KIR area of chr19 and the 
  MHC region on chr6. (See the 
  <a href="../cgi-bin/hgTracks?chromInfoPage=">sequences</a> page for a complete list of 
  the reference chromosomes and alternate sequences in GRCh38.)</li>
  <li>
  <strong>Centromere representation</strong> - Debuting in this release, the large megabase-sized 
  gaps that were previously used to represent centromeric regions in human assemblies have been 
  replaced by sequences from centromere models created by 
  <a href="http://genome.cshlp.org/content/24/4/697.abstract" target="_blank">Karen Miga et al.</a> 
  of UCSC's <a href="http://cbse.soe.ucsc.edu/people/kent" target="_blank">Kent lab</a>, under the 
  guidance and mentorship of the <a href="http://www.genome.duke.edu/labs/WillardLab/"
  target="_blank">Willard lab</a> at Duke University. The models, which provide the approximate 
  repeat number and order for each centromere, will be useful for read mapping and variation 
  studies. </li>
  <li>
  <strong>Mitochondrial genome</strong> - The mitochondrial reference sequence included in the 
  GRCh38 assembly and hg38 Genome Browser (termed &quot;chrM&quot; in the browser) is the 
  <a href="http://mitomap.org/bin/view.pl/MITOMAP/HumanMitoSeq" 
  target="_blank">Revised Cambridge Reference Sequence</a> (rCRS) from 
  <a href="http://mitomap.org/MITOMAP" target="_blank">MITOMAP</a> with GenBank accession 
  number J01415.2 and RefSeq accession number NC_012920.1. This differs from the chrM 
  sequence (RefSeq accession number NC_001907) used by the previous hg19 Genome Browser, 
  which was not updated when the GRCh37 assembly later transitioned to the new version.</li>
  <li>
  <strong>Sequence updates</strong> - Several erroneous bases and misassembled regions in GRCh37 
  have been corrected in the GRCh38 assembly, and more than 100 gaps have been filled or reduced. 
  Much of the data used to improve the reference sequence was obtained from other genome 
  sequencing and analysis projects, such as the <a href="http://www.1000genomes.org/" 
  target="_blank">1000 Genomes Project</a>.</li>  
  <li>
  <strong>Analysis set</strong> - The GRCh38 assembly offers an &quot;analysis set&quot; that was 
  created to accommodate next generation sequencing read alignment pipelines. Several
  GRCh38 regions have been eliminated from this set to improve read mapping. 
  The analysis set may be downloaded from the Genome Browser 
  <a href="http://hgdownload.gi.ucsc.edu/goldenPath/hg38/bigZips/analysisSet/"
  target="_blank">downloads page</a>.</li>
</ul>
<p>
For more information about the files included in the GRCh38 GenBank submission, see the
<a href="ftp://ftp.ncbi.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Homo_sapiens/GRCh38/README"
target="_blank">GRCh38 README</a>. The GRCh38 GenBank record provides a detailed array of 
<a href="https://www.ncbi.nlm.nih.gov/assembly/883148/#/st_Primary-Assembly"
target="_blank">statistics about this assembly</a>. Bulk downloads of the sequence and annotation 
data may be obtained from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg38/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#human">Downloads</a> page. The annotation 
tracks for this browser were generated by UCSC and collaborators worldwide.</p>
<p>
<strong>Much more to come!</strong> This initial release of the hg38 Genome Browser provides a 
rudimentary set of annotations.  Many of our annotations rely on data sets from external 
contributors (such as our popular SNPs tracks) or require massive computational effort 
(our comparative genomics tracks). In the upcoming months/years, we will release many more 
annotation tracks as they become available. To stay abreast of new datasets, join our 
<a href="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
target="_blank">genome-announce</a> mailing list or follow us on
<a href="http://www.twitter.com/GenomeBrowser" target="_blank">twitter</a>.</p>
<p>
We'd like to thank our GRC and NCBI collaborators who worked closely with
us in producing the hg38 browser. Their quick responses and helpful feedback were a
key factor in expediting this release. The production of the hg38 Genome Browser was
a team effort, but in particular we'd like to acknowledge the engineering efforts of Hiram 
Clawson and Brian Raney, the QA work done by Steve Heitner, project guidance
provided by Ann Zweig, Robert Kuhn, and Jim Kent, and documentation work by Donna
Karolchik. See the <a href="credits.html#human_credits">Credits</a> page for a detailed
list of the organizations and individuals who contributed to this release.</p>

<a name="030414"></a>
<h2>Mar. 4, 2014 &nbsp;&nbsp; Introducing the Genome Browser highlight feature</h2>
<p>
We are excited to announce the new highlight feature in the UCSC Genome Browser. Using 
drag-and-select, you can now highlight a region or gene of interest.</p>
<p>
<img src="../images/highlight_screenshot.png" alt="Highlight screenshot" width="721"
height="267"></p>
<p>
To highlight a region: Click and hold the mouse button on one edge of the desired area to be 
highlighted in the Base Position track, drag the mouse right or left to highlight the selection 
area, then release the mouse button. Click the &quot;Highlight&quot; button on the
&quot;drag-and-select&quot; popup. More details about this new feature can be found on our 
<a href="http://genome.ucsc.edu/goldenPath/help/hgTracksHelp.html#Highlight"
target="_blank">help</a> page.</p> 
<p>
Credit goes to Tim Dreszer, Larry Meyer, Robert Kuhn and Luvina Guruvadoo for the design, 
development and testing of this feature. Additional testing was also provided by several members of 
the QA team.</p>

<a name="022814"></a>
<h2>Feb. 28, 2014 &nbsp;&nbsp; New expanded onsite workshop program</h2>
<p>
Explore the full power of the UCSC Genome Browser! Thanks to the funding support of NHGRI, we can 
now offer hands-on Genome Browser training onsite at your institution, tailored to your audience's 
level of expertise. Through this program, the services of our expert trainer are provided for free, 
and participating institutions cover only the transportation from UCSC, lodging, and a flat fee to 
cover other costs.</p>
<p>
For more information or to submit a request for a workshop, please visit
<a href="http://bit.ly/ucscTraining" target="_blank">our signup</a>.</p>

<a name="012214"></a>
<h2>Jan. 22, 2014 &nbsp;&nbsp; All the DNA on the internet now at your fingertips</h2>
<p>
We're pleased to announce the release of the Web Sequences track on the UCSC Genome Browser.
This track, produced in collaboration with Microsoft Research, contains the results of a 30-day 
scan for DNA sequences from over 40 billion different webpages. The sequences were then mapped with 
Blat to the human genome (hg19) and numerous other species including mouse (mm9), rat (rn4), and 
zebrafish (danRer7). The data were extracted from a variety of sources including patents, online 
textbooks, help forums, and any other webpages that contain DNA sequence. In essence, this track 
displays the Blat alignments of nearly every DNA sequence on the internet! The
<a href="../cgi-bin/hgTrackUi?db=hg19&g=pubsBingBlat">Web Sequences</a>
track description page contains more details on how the track was generated.</p>
<p>
We would like to acknowledge Max Haeussler and Matt Speir from the UCSC Genome Browser staff
and Bob Davidson from Microsoft Research for their hard work in creating this track.</p>

<a name="011414"></a>
<h2>Jan. 14, 2014 &nbsp;&nbsp; Two E. coli comparative assembly hubs now available</h2>
<p>
We're pleased to add two new assembly hubs produced by the UCSC
<a href="https://ucscgenomics.soe.ucsc.edu/team/" target="_blank">David Haussler lab</a> to our
collection of publicly available hubs. The new hubs feature over 60 bacterial assemblies,
including more than 55 different E. coli strains. The assembly annotations include genes, 
pathogenic genes, conservation, GC percent, repetitive elements and much more.</p>
<p>
These hubs focus on comparative genomics and showcase the new &quot;snake&quot; track type. Snakes, 
which visualize alignments from
<a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format20">Hierarchical Alignment</a>
(HAL) files, provide a way to view sets of pairwise gapless alignments that may overlap on 
both the chosen genome (reference) and the query genome, and show various types of genomic 
variations such as insertions, substitutions, and duplications. More details about the new snake 
track display and its configuration options can be found on our Genome Browser
<a href="http://genome.ucsc.edu/goldenPath/help/hgTracksHelp.html" 
target="_blank">help</a> page.</p>
<p>
To access and view these hubs, navigate to the
<a href="../cgi-bin/hgHubConnect">Track Hub gateway page</a>
and select one of the two E. coli comparative assembly hubs from the Public Hubs list.</p>
<p>
We would like to acknowledge Ngan Nguyen, Glenn Hickey, Brian Raney, Joel Armstrong, 
Benedict Paten, Matt Speir, and Luvina Guruvadoo for their hard work in creating these hubs.</p>

<!-- ============= 2013 archived news ============= -->

<a name="2013"></a>
<a name="112713"></a>
<h2>Nov. 27, 2013 &nbsp;&nbsp; 100-Species Conservation track now available on hg19</h2>
<p>
After 15.4 years of CPU run-time in 9,905,594 individual jobs and 99 cluster runs for lastz
pair-wise alignment...we are excited to announce the release of a 100 species alignment on 
the hg19/GRCh37 human Genome Browser.</p>
<p>
This new Conservation track shows multiple alignments of 100 species and measurements of
evolutionary conservation using two methods (phastCons and phyloP) from the PHAST package.
This adds 40 more species to the existing 60-way on the mm10 mouse browser. For more information 
about the 100 species Conservation track, see its 
<a href="../cgi-bin/hgTrackUi?db=hg19&g=cons100way" 
target="_blank">description page</a>.</p>
<p>
We'd also like to acknowledge the hard work of the UCSC Genome Browser staff who pulled 
together the information for this track: Hiram Clawson and Pauline Fujita.</p>

<a name="102413"></a>
<h2>Oct. 24, 2013 &nbsp;&nbsp; Job opening: UCSC Genome Browser Trainer</h2>
<p>
The <a href="http://cbse.ucsc.edu/" target="_blank">Center for Biomolecular Science
and Engineering</a> (CBSE) at UCSC seeks an articulate, self-motivated educator for the two-year 
position of UCSC Genome Browser trainer. The trainer develops curriculum and provides in-person 
training on the UCSC Genome Browser at universities, hospitals, institutes, and professional
meetings in the United States and internationally. Typical audiences include graduate 
and post-graduate biologists and doctors, with Genome Browser experience ranging from novice
users to bioinformatics specialists. Presentations include formal talks, problem-solving 
sessions, and two-day workshops. </p>
<p>
This position requires a Master's degree in a biological science, depth in molecular
biology, experience in a research environment, working knowledge of the UCSC Genome Browser,
understanding of its role in research methodology, and experience teaching or training in a
scientific environment. Preferred qualifications include a PhD in a relevant field,
experience with video production, and experience with HTML or web content management systems.</p>
<p>
For more information and to apply for this position, see <a href="http://bit.ly/browserJob" 
target="_blank">Job #1304619</a> on the UCSC Staff Employment website.</p>
	
<a name="102313"></a>
<h2>Oct. 23, 2013 &nbsp;&nbsp; dbSNP 138 available for hg19</h2>
<p>
We are pleased to announce the release of four tracks derived from NCBI
<a href="https://www.ncbi.nlm.nih.gov/SNP/" target="_blank">dbSNP</a> Build 138 data, available on 
the human assembly (GRCh37/hg19). The new tracks contain additional annotation data not included in 
previous dbSNP tracks, with corresponding coloring and filtering options in the Genome Browser.</p>
<p>
As was the case for the annotations based on the previous dbSNP build 137, there are four tracks in 
this release. One is a track containing all mappings of reference SNPs to the human assembly, 
labeled &quot;All SNPs (138)&quot;. The other three tracks are subsets of this track and show 
interesting and easily defined subsets of dbSNP:</p>
<ul>
  <li>
  Common SNPs (138): uniquely mapped variants that appear in at least 1% of the population or are 
  100% non-reference</li>
  <li>
  Flagged SNPs (138): uniquely mapped variants, excluding Common SNPs, that have been flagged by 
  dbSNP as &quot;clinically associated&quot;li>
  <li>
  Mult. SNPs (138): variants that have been mapped to more than one genomic location</li>
</ul>
<p>
By default, only the Common SNPs (138) are visible; other tracks must be made visible using the 
track controls.</p>
<p>
You will find the four SNPs (138) tracks on the Human Feb. 2009 (GRCh37/hg19) browser in the 
Variation and Repeats group.</p>
<p>
The tracks were produced at UCSC by Angie Hinrichs and Luvina Guruvadoo. We'd like to thank the 
dbSNP group at NCBI for providing access to these data.</p>

<a name="101113"></a>
<h2>Oct. 11, 2013 &nbsp;&nbsp; Updated DGV Structural Variation track</h2>
<p>
The UCSC Genome Browser is pleased to announce an updated DGV Structural Variation track to reflect 
the official launch of the new  <a href="http://dgv.tcag.ca/dgv/app/home"
target="_blank">Database of Genomic Variants</a>. The new release includes a number of updates and 
corrections and completes DGV's transition from their beta version as they retire their original 
site for the new and improved site. For more background on DGV's changes, please see DGV's 
<a href="http://dgv.tcag.ca/dgv/app/news" target="_blank">June 2013 newsletter</a>.</p>

<a name="092613"></a>
<h2>Sep. 26, 2013 &nbsp;&nbsp; New American alligator (allMis1) assembly now available</h2>
<p>
A Genome Browser is now available for the American alligator (<em>Alligator mississippiensis</em>) 
assembly released August 2012 by the International Crocodilian Genomes Working Group (version 
allMis0.2, UCSC version allMis1). Please note the 
<a href="https://www.ncbi.nlm.nih.gov/Traces/wgs/?val=AKHW01" target="_blank">assembly construction 
details</a>. For more information and statistics about this assembly, see the NCBI assembly record 
for <a href="https://www.ncbi.nlm.nih.gov/assembly/406428/" target="_blank">allMis0.2</a>. There are 
14,645 scaffolds with a total size of 2,174,259,888 bases.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/allMis1/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#american_alligator">Downloads</a> page. 
Please observe the <a href="credits.html#american_alligator_use">conditions for use</a>
when accessing and using these data sets. The annotation tracks for this browser were generated by 
UCSC and collaborators worldwide. See the 
<a href="credits.html#american_alligator_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release. </p>

<a name="091113"></a>
<h2>Sep. 11, 2013 &nbsp;&nbsp; New alpaca (vicPac2) assembly released in the Genome Browser</h2>
<p>
We are pleased to announce the release of a Genome Browser for the March 2013 assembly of alpaca, 
<em>Vicugna pacos</em> (Broad Vicugna_pacos-2.0.1, UCSC version vicPac2). The whole genome shotgun 
assembly was provided by <a href="http://www.broadinstitute.org/" target="_blank">The Broad 
Institute</a>.  There are 276,725 scaffolds with a total size of 2,172,191,320 bases.</p>
<p>
Bulk downloads of the sequence and annotation data are available via the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/vicPac2/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#alpaca">Downloads</a> page. These data have
<a href="/goldenPath/credits.html#alpaca_credits">specific conditions for use</a>. The alpaca 
browser annotation tracks were generated by UCSC and collaborators worldwide. See the
<a href="/goldenPath/credits.html#alpaca_credits">Credits</a> page for a detailed
list of the organizations and individuals who contributed to this release. </p>

<a name="090413b"></a>
<h2>Sep. 4, 2013 &nbsp;&nbsp; New tenrec (echTel2) assembly added to the Genome Browser</h2>
<p>
We are pleased to announce the release of a Genome Browser for the Nov. 2012 assembly of tenrec, 
<em>Echinops telfairi</em> (Broad EchTel2.0, UCSC version echTel2). The whole genome shotgun 
assembly was provided by <a href="http://www.broadinstitute.org/" target="_blank">The Broad 
Institute</a>.  There are 8,402 scaffolds with a total size of 2,947,024,286 bases.</p>
<p>
Bulk downloads of the sequence and annotation data are available via the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/echTel2/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#tenrec">Downloads</a> page. These data have
<a href="/goldenPath/credits.html#tenrec_credits">specific conditions for use</a>.
The tenrec browser annotation tracks were generated by UCSC and collaborators worldwide. See the 
<a href="/goldenPath/credits.html#tenrec_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>

<a name="090413a"></a>
<h2>Sep. 4, 2013 &nbsp;&nbsp; New sheep (oviAri3) assembly now available</h2>
<p>
We are pleased to announce the release of a Genome Browser for the August 2012 assembly of sheep, 
<em>Ovis aries</em> (ISGC Oar_v3.1, UCSC version oviAri3). The whole genome shotgun assembly was 
provided by the <a href="http://www.sheephapmap.org" target="_blank">International Sheep Genomics 
Consortium</a>.  There are 5,698 scaffolds with a total size of 2,619,054,388 bases.</p>
<p>
Bulk downloads of the sequence and annotation data are available via the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/oviAri3/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#sheep">Downloads</a> page. These data have
<a href="/goldenPath/credits.html#sheep_credits">specific conditions for use</a>. The sheep browser 
annotation tracks were generated by UCSC and collaborators worldwide. See the
<a href="/goldenPath/credits.html#sheep_credits">Credits</a> page for a detailed 
list of the organizations and individuals who contributed to this release.</p>

<a name="082213"></a>
<h2>Aug. 22, 2013 &nbsp;&nbsp; Three new human variation tracks released</h2>
<p>
We are pleased to announce the release of three tracks that describe human genetic variation. They 
are based on the <a href="http://lovd.nl/2.0/index_list.php" target="_blank">Leiden Open Variation 
Database (LOVD)</a>, the <a href="http://www.hgmd.cf.ac.uk/ac/index.php" target="_blank">Human Gene
Mutation Database Public (HGMD)</a>, and amino acid mutations in the 
<a href="http://www.uniprot.org/" target="_blank">UniProt database</a>. The majority of the data 
were collected from publications by curators in the three databases.</p>
<p>
The three databases have varying license restrictions. UniProt provides complete information about 
the mutation amino acid change, the disease and a link to the publications that mention it. LOVD 
provides the mutation and a link to a website with disease and publication details. HGMD provides 
some coordinates and a link to its own website which requires registration.</p>
<p>
These tracks are intended for use primarily by physicians and other professionals concerned with 
genetic disorders, by genetics researchers, and by advanced students in science and medicine. While 
the Genome Browser database is open to the public, users seeking information about a personal 
medical or genetic condition are urged to consult with a qualified physician for diagnosis and for 
answers to personal questions.</p>
<p>
We'd like to thank all three databases for their help in putting the data together. The tracks were 
produced at UCSC by Maximilian Haeussler and Steve Heitner.</p>

<a name="072213"></a>
<h2>Jul. 22, 2013 &nbsp;&nbsp; Variant Annotation Integrator</h2>
<p>
In order to assist researchers in annotating and prioritizing thousands of variant calls from 
sequencing projects, we have developed the Variant Annotation Integrator (VAI). Given a set of 
variants uploaded as a custom track (in either <a href="../FAQ/FAQformat.html#format10">pgSnp</a>
or <a href="help/vcf.html">VCF</a> format), the VAI will return the predicted functional effect 
(e.g., synonymous, missense, frameshift, intronic) for each variant. The VAI can optionally add
several other types of relevant information, including: the 
<a href="https://www.ncbi.nlm.nih.gov/SNP/" target="_blank">dbSNP</a> identifier if the variant is 
found in dbSNP, protein damage scores for missense variants from the 
<a href="https://sites.google.com/site/jpopgen/dbNSFP" target="_blank">Database of Non-synonymous 
Functional Predictions (dbNSFP)</a>, and conservation scores computed from multi-species alignments.
The VAI also offers filters to help narrow down results to the most interesting variants.</p>
<p>
Future releases of the VAI will include more input/upload options, output formats, and annotation 
options, and a way to add information from any track in the Genome Browser, including custom 
tracks.</p>
<p>
There are two ways to navigate to the VAI:</p>
<ol>
  <li>
  From the &quot;Tools&quot; menu, follow the &quot;Variant Annotation Integrator&quot; link.</li> 
  <li>
  After uploading a custom track, press the &quot;go to variant annotation integrator&quot; button. 
  The user's guide is at the bottom of the page, under &quot;Using the Variant Annotation 
  Integrator.&quot;</li>
</ol>
<p>
Thank you to Angie Hinrichs, Bob Kuhn, Brian Raney, Brooke Rhead, and Jonathan Casper for their 
input and work on this feature.</p>

<a name="070213"></a>
<h2>Jul. 2, 2013 &nbsp;&nbsp; Assembly Data Hubs</h2>
<p>
We are pleased to announce a new feature in the UCSC Genome Browser: Assembly Data Hubs.</p>
<p>
Assembly Data Hubs extend the functionality of Track Data Hubs to assemblies that are not hosted 
natively on the Browser. Assembly Data Hubs were developed to address the increasing need for 
researchers to annotate sequence for which UCSC does not provide an annotation database. They allow 
researchers to include the underlying reference sequence, as well as data tracks that annotate that 
sequence. Sequence is stored in the UCSC twoBit format, and the annotation tracks are stored in the 
same manner as Track Data Hubs.</p>
<p>
To read more about using Assembly Data Hubs, or about creating your own, see these guides:</p>
<ul>
  <li>
  <a href="http://genome.ucsc.edu/goldenPath/help/hgTrackHubHelp.html">Using UCSC Genome Browser 
  Track Hubs</a> </li>
  <li>
  <a href="http://genomewiki.ucsc.edu/index.php/Assembly_Hubs">Assembly_Hubs wiki page</a> </li>
</ul>
<p>
Many thanks to Hiram Clawson, Brian Raney, and Pauline Fujita for their work on this feature.</p>

<a name="070213b"></a>
<h2>Jul. 2, 2013 &nbsp;&nbsp; New gene alleles feature on the details pages of UCSC genes</h2>
<p>
We are excited to announce the release the new Gene Alleles feature on the details pages of UCSC 
genes. For each protein-coding gene in the UCSC Genes track, variant data from the 2,184 (per 
autosome) phased chromosomes from Phase 1 of the 1000 Genomes Project have been distilled into 
&quot;gene haplotype alleles&quot;. Each haplotype allele is a distinct set of variants found on at 
least one of the 1000 Genomes subject chromosomes. For each haplotype, we display its frequency in
the 1000 Genomes populations and indicate the frequency with which it occurs homozygously in each 
population. Unexpected frequencies of occurrence may thus be used to identify alleles that merit 
further study. Predicted protein sequence for common haplotypes can also be displayed, allowing 
differences among alleles to be used to identify differences at the amino acid level.</p>
<p>
To see the Gene Alleles feature, go to the details page for any protein-coding gene in the UCSC 
Genes track and click the &quot;Gene Alleles&quot; link in the &quot;Page Index&quot; matrix. This
will take you to the &quot;Common Gene Haplotype Alleles&quot; section. For more information on 
this feature see: <a href="http://genome.ucsc.edu/goldenPath/help/haplotypes.html">
http://genome.ucsc.edu/goldenPath/help/haplotypes.html</a>. </p>
<p>
Many thanks to Tim Dreszer, Brooke Rhead and Pauline Fujita for their work on this feature.</p>

<a name="062813"></a>
<h2>Jun. 28, 2013 &nbsp;&nbsp;UCSC Genes track released for GRCh37/hg19</h2>
<p>
We're happy to announce the release of an updated UCSC Genes track for the GRCh37/hg19 human Genome 
Browser.  The new release has 82,960 total transcripts, compared with 80,922 in the previous 
version. The total number of canonical genes has increased from 31,227 to 31,848. Comparing the new 
gene set with the previous version:</p>
<ul>
  <li>
  74,657 transcripts did not change between versions.</li>
  <li>
  88 transcripts were not carried forward to the new version.</li>
  <li>
  5,556 transcripts are "compatible" with those in the previous set, meaning that the two 
  transcripts show consistent splicing. In most cases, the old and new transcripts differ in the 
  lengths of their UTRs.</li>
  <li>
  621 transcripts overlap with those in the previous set but do not show consistent splicing, 
  i.e., they contain overlapping introns with differing splice sites.</li>
</ul>
<p>
More details about the new UCSC Genes track can be found on the 
<a href="../cgi-bin/hgTrackUi?db=hg19&g=knownGene" >UCSC Genes track description
page</a>.</p>
<p>
Many thanks to Brian Raney, Jim Kent, and Luvina Guruvadoo for their work on this track.</p>

<a name="062713"></a> 
<h2>Jun. 27, 2013 &nbsp;&nbsp; New supported European mirror site established</h2>
<p>
The UCSC Genome Browser is pleased to announce the introduction of a new mirror site to serve our 
users in Europe. Genome-euro is an official European mirror site of the UCSC Genome Browser at 
<a href="http://genome-euro.ucsc.edu" target="_blank">http://genome-euro.ucsc.edu</a>. The server 
is physically located at the Universit&auml;t Bielefeld Center for Biotechnology in Bielefeld, 
Germany, and is administered by UCSC. Genome-euro is meant to be an alternate, faster access point 
for those Browser users who are geographically closer to central Europe than to the western United 
States.</p>
<p>
All functionality will be the same as on the US server, although Custom Tracks will not be 
transfered. Saved Sessions containing Custom Tracks will require some manual 
<a href= "http://genome.ucsc.edu/goldenPath/help/genomeEuro.html#sessions" 
target="_blank">intervention</a>.</p>
<p>
When European users navigate to the US server home page and click the &quot;Genomes&quot; menu item,
they will receive a notification that they have been redirected to the more geographically 
appropriate server. They will have the option to remain on the US server, as described in our
<a href="http://genome.ucsc.edu/goldenPath/help/genomeEuro.html" 
target="_blank">documentation</a>.</p>
<p>
The backup mirror in <a href="http://genome-mirror.moma.ki.au.dk/" target="_blank">Aarhus, 
Denmark</a> will continue to serve as an emergency site in the event of the official sites in 
California and Germany malfunctioning.</p>
<p>
We wish to express our special thanks to our colleagues at the Universit&auml;t Bielefeld 
Bioinformatics, especially Jens Stoye, for making this possible.</p>

<a name="062613"></a>
<h2>Jun. 26, 2013 &nbsp;&nbsp; Suggestion box</h2>
<p>
Due to popular demand, we have created a suggestion box for track, assembly, tool, and other 
requests. All suggestions will be assigned a reference number so that you may follow up on their 
progress with our staff. Please limit the use of this form to requests for new features; questions 
and reports of problems about existing features and data should continue to be directed to
our mailing lists. You may reach the suggestion box from the 
<a href="../contacts.html" target="_blank">Contact Us</a> page or by 
<a href="../cgi-bin/hgUserSuggestion" target="_blank">clicking here</a>. </p>

<a name="061213"></a>
<h2>Jun. 12, 2013 &nbsp;&nbsp; Ferret Genome Browser released</h2>
<p>
A Genome Browser is now available for the ferret (<em>Mustela putorius furo</em>) assembly released 
by the Ferret Genome Sequencing Consortium April 2011 (version MusPutFur1.0, UCSC version musFur1).
Please note the <a href="https://www.ncbi.nlm.nih.gov/Traces/wgs/?val=AEYP01"
target="_blank">assembly construction details</a>. For more information and statistics about this 
assembly, see the NCBI assembly record for <a href="https://www.ncbi.nlm.nih.gov/assembly/286418/"
target="_blank">MusPutFur1.0</a>.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/musFur1/" target="_blank">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#ferret" target="_blank">Downloads</a> page. 
Please observe the <a href="credits.html#ferret_use">conditions for use</a> when accessing and using
these data sets. The annotation tracks for this browser were generated by UCSC and collaborators 
worldwide. See the <a href="credits.html#ferret_credits">Credits</a> page for a detailed list of the
organizations and individuals who contributed to this release.</p>

<a name="052213"></a>
<h2>May 22, 2013 &nbsp;&nbsp; Gibbon Genome Browser released</h2>
<p>
A Genome Browser is now available for the Gibbon (<em>Nomascus leucogenys</em>) assembly released by
the Gibbon Genome Sequencing Consortium Oct. 2012 (version Nleu_3.0, UCSC version nomLeu3). Please 
note the <a href="https://www.ncbi.nlm.nih.gov/Traces/wgs/?val=ADFV01" target="_blank">assembly 
construction details</a>. For more information and statistics about this assembly, see the NCBI 
assembly record for <a href="https://www.ncbi.nlm.nih.gov/assembly/506498/"
target="_blank">Nleu_3.0</a>.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/nomLeu3/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#gibbon">Downloads</a> page. Please observe 
the <a href="credits.html#gibbon_use">conditions for use</a> when accessing and using these data 
sets. The annotation tracks for this browser were generated by UCSC and collaborators worldwide. 
See the <a href="credits.html#gibbon_credits">Credits</a> page for a detailed the organizations and 
individuals who contributed to this release.</p>

<a name="040913"></a>
<h2>Apr. 9, 2013 &nbsp;&nbsp; The UCSC Cancer Genomics Browser has a new look</h2>
<p>
The UCSC Cancer Genomics group has recently remodeled the interface of the
<a href="https://genome-cancer.ucsc.edu/" target="blank">Cancer Genomics Browser</a> to make it 
easier to navigate and more intuitive to display, investigate, and analyze cancer genomics data and 
associated clinical information. This tool provides access to many types of 
information&mdash;biological pathways, collections of genes, genomic and clinical 
information&mdash;that can be used to sort, aggregate, and perform statistical tests on a group of 
samples. The Cancer Browser currently displays 473 datasets of 25 cancers from 
<a href="http://cancergenome.nih.gov/" target=_"blank">The Cancer Genome Atlas</a> (TCGA) as well 
as data from the <a href="http://www.broadinstitute.org/ccle/home" target="_blank">Cancer Cell 
Line Encyclopedia</a> (CCLE) and <a href="http://www.standup2cancer.org/" target="_blank">Stand Up 
To Cancer</a>.</p>
<p>
You can find more information about how to use this tool in the online 
<a href="https://genome-cancer.soe.ucsc.edu/proj/site/hgHeatmap/#?tutorial=true" 
target="_blank">tutorial</a>, <a href="https://genome-cancer.ucsc.edu/proj/site/help/" 
target="_blank">user's guide</a> and <a href="https://genome-cancer.ucsc.edu/proj/site/faq/" 
target="_blank">FAQ</a>. Any questions or comments should be directed to 
<a href="mailto:genome-cancer@soe.ucsc.edu">genome-cancer@soe.ucsc.edu</a>.</p>

<a name="030513"></a>
<h2>Mar. 5, 2013 &nbsp;&nbsp; dbSNP 137 available for mm10</h2>
<p>
We are pleased to announce the release of three tracks derived from dbSNP build 137, available on 
the mouse assembly (GRCm38/mm10). dbSNP build 137 is available at NCBI. The new tracks contain
additional annotation data not included in previous dbSNP tracks, with corresponding coloring and 
filtering options in the Genome Browser.</p>
<p>
As was the case for previous annotations based on dbSNP data, there are three tracks in this 
release.  One is a track containing all mappings of reference SNPs to the mouse assembly, labeled 
&quot;All SNPs (137)&quot;. The other two tracks are subsets of this track and show interesting and 
easily defined subsets of dbSNP:</p>
<ul>
  <li>
  Common SNPs (137): uniquely mapped variants that appear in at least 1% of the population</li>
  <li>
  Mult. SNPs (137): variants that have been mapped to more than one genomic location</li>
</ul>
<p>
By default, only the Common SNPs (137) are visible; other tracks must be made visible using the 
track controls.</p>
<p>
You will find the three SNPs (137) tracks on the Mouse Dec. 2011 (GRCm38/mm10) browser in the 
Variation and Repeats group.</p>
<p>
The tracks were produced at UCSC by Angie Hinrichs and Luvina Guruvadoo.</p>  

<a name="021113"></a>
<h2>Feb. 11, 2013 &nbsp;&nbsp; Denisova tracks released on hg19 browser</h2>
<p>
In conjunction with the publication of the paper by Meyer <em>et al.</em>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/22936568">A High-Coverage Genome Sequence from an 
Archaic Denisovan Individual</a> the UCSC Genome Browser is hosting a set of new tracks. Briefly, 
Meyer <em>et al.</em> applied a novel single-stranded DNA library preparation method to DNA 
previously extracted from 40mg of a phalanx bone excavated from Denisova Cave in the Altai Mountains
in southern Siberia.</p>
<p>	
Two supporting datasets and one analysis dataset are now available in the Genome Browser:</p>
<ul>
  <li>
  Modern Human Variants: variant calls made from sequence reads of eleven individuals 
  mapped to the human genome, used in statistical analyses to put the divergence of the 
  Denisova genome into perspective with regard to present-day humans.</li>
  <li>
  Denisova Sequence: high-coverage (30X) sequence reads from an archaic Denisovan individual mapped 
  to the human genome reference assembly.</li>
  <li>
  Modern Human Derived, Denisova Ancestral: mutations in the modern human lineage that rose to 
  fixation or near fixation since the split from the last common ancestor with Denisovans, along 
  with predicted functional effects from Ensembl's Variant Effect Predictor (VEP).</li>
</ul>
<p>
We'd like to thank Richard Green and the many authors of the Science paper for providing the UCSC 
Genome Browser with these data. We'd also like to acknowledge the hard work of the UCSC Genome 
Browser staff who pulled together the annotations and information for these tracks: Angie Hinrichs 
and Pauline Fujita.</p>

<a name="012513"></a>
<h2>Jan. 25, 2013 &nbsp;&nbsp; Southern white rhinoceros Genome Browser released</h2>
<p>
A Genome Browser is now available for the Southern White Rhinoceros 
(<em>Ceratotherium simum simum</em>) assembly released by the Broad Institute in May 2012 (Broad 
version cerSimSim1.0, UCSC version cerSim1). This genome was sequenced and assembled at the Broad 
Institute using samples provided by Dr. Oliver Ryder at the San Diego Zoo Institute for Conservation
Research. For more information and statistics about this assembly, see the NCBI assembly record for 
<a href="https://www.ncbi.nlm.nih.gov/assembly/406328/" target="_blank">CerSimSim1.0</a>.</p>
<p>
Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/cerSim1/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#white_rhinoceros">Downloads</a> page. Please 
observe the <a href="credits.html#white_rhinoceros_use">conditions for use</a> when accessing and 
using these data sets. The annotation tracks for this browser were generated by UCSC and 
collaborators worldwide. See the <a href="credits.html#white_rhinoceros_credits">Credits</a> page 
for a detailed list of the organizations and individuals who contributed to this release.</p>

<a name="012213"></a>
<h2>Jan. 22, 2013 &nbsp;&nbsp; New baboon (papAnu2) assembly now available in the Genome
Browser</h2>
<p>
We are pleased to announce the release of a Genome Browser for the March 2012 assembly of the olive 
baboon, <em>Papio anubis</em> (Baylor Panu_2.0, UCSC version papAnu2). The whole genome shotgun 
assembly was provided by <a href="https://www.hgsc.bcm.edu/content/baboon-genome-project"
target="_blank">The Baylor College of Medicine Human Genome Sequencing Center</a>. There are 
72,500 scaffolds with a total size of 2,948,380,710 bases.</p>
<p>
Bulk downloads of the sequence and annotation data are available via the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/papAnu2/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#baboon">Downloads</a> page. These data have
<a href="/goldenPath/credits.html#baboon_credits">specific conditions for use</a>. The baboon 
browser annotation tracks were generated by UCSC and collaborators worldwide. See the
<a href="/goldenPath/credits.html#baboon_credits">Credits</a> page for a detailed 
list of the organizations and individuals who contributed to this release.</p>

<a name="011513"></a> 
<h2>Jan. 15, 2013 &nbsp;&nbsp; New lamprey (petMar2) assembly released in Genome Browser</h2>
<p>
We are pleased to announce the release of a Genome Browser for the Sep. 2010 assembly of the 
lamprey, <em>Petromyzon marinus</em> (WUGSC 7.0, UCSC version petMar2). The whole genome shotgun 
assembly was provided by <a href="http://genome.wustl.edu/genomes/detail/petromyzon-marinus/"
target="_blank">The Genome Institute at Washington University</a>. There are 25,006 scaffolds with 
a total size of 885,550,958 bases.</p>
<p>
Bulk downloads of the sequence and annotation data are available via the Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/petMar2/">FTP server</a> or the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#lamprey">Downloads</a> page. These data have
<a href="/goldenPath/credits.html#lamprey_credits">specific conditions for use</a>.  The lamprey 
browser annotation tracks were generated by UCSC and collaborators worldwide. See the
<a href="/goldenPath/credits.html#lamprey_credits">Credits</a> page for a detailed 
list of the organizations and individuals who contributed to this release.</p>

<!-- ============= 2012 archived news ============= -->
<a name="2012"></a>
<a name="122012"></a>
<h2>Dec. 20, 2012 &nbsp;&nbsp; 28 new vertebrate assemblies</h2>
<p>
Over the past several weeks, we have released 28 new vertebrate assemblies on the public Genome 
Browser website, featuring 22 new species and 6 assembly updates. These assemblies were added to 
support the 60-species Conservation track on the latest mouse assembly (mm10/GRCm38). Several of 
these species were originally sequenced and assembled for the Mammalian Genome Project 
(Lindblad-Toh <em>et al</em>., <em>Nature</em> 2011)*.</p>
<p>
<strong><em>Primates:</em></strong></p>
<ul>
  <li> 
  <strong>Baboon</strong> (<em>Papio hamadryas</em>) <strong>papHam1</strong> &ndash; Pham_1.0 
  (Nov. 2008) from the Baylor College of Medicine HGSC<li>
  <li> 
  <strong>Bushbaby</strong> (<em>Otolemur garnettii</em>) <strong>otoGar3</strong> &ndash; 
  OtoGar3 (Mar. 2011) from the Broad Institute</li> 
  <li> 
  <strong>Chimpanzee</strong> (<em>Pan troglodytes</em>) <strong>panTro4</strong> &ndash; Build 
  2.1.4 (Feb. 2011) from the Chimpanzee Sequencing and Analysis Consortium</li>
  <li> 
  <strong>Gibbon</strong> (<em>Nomascus leucogenys</em>) <strong> nomLeu2</strong> &ndash; Nleu1.1 
  (Jun. 2011) from the Gibbon Genome Sequencing Consortium</li> 
  <li> 
  <strong>Mouse lemur</strong> (<em>Microcebus murinus</em>) <strong>micMur1</strong> &ndash;
  MicMur1.0 (Jul. 2007) from the Broad Institute</li> 
  <li> 
  <strong>Rhesus macaque</strong> (<em>Macaca mulatta</em>) <strong>rheMac3</strong> &ndash; CR_1.0 
  (Oct. 2010) from the Beijing Genomics Institute</li> 
  <li> 
  <strong>Squirrel monkey</strong> (<em>Saimiri boliviensis</em>) <strong>saiBol1</strong> &ndash; 
  SaiBol1.0 (Oct. 2011) from the Broad Institute</li> 
  <li> 
  <strong>Tarsier</strong> (<em>Tarsius syrichta</em>) <strong>tarSyr1</strong> &ndash; Tarsyr1.0 
  (Aug. 2008) from WUSTL and the Broad Institute</li> 
</ul>
<p>
<strong><em>Other mammals:</em></strong></p>
<ul>
  <li> 
  <strong>Alpaca</strong> (<em>Vicugna pacos</em>) <strong>vicPac1</strong> &ndash; VicPac1.0 
  (Jul. 2008) from the Broad Institute</li> 
  <li> 
  <strong>Armadillo</strong> (<em>Dasypus novemcinctus</em>) <strong>dasNov3</strong> &ndash; 
  DasNov3 (Dec. 2011) from the Baylor College of Medicine HGSC</li> 
  <li> 
  <strong>Cat</strong> (<em>Felis catus</em>) <strong>felCat5</strong> &ndash; Felis_catus-6.2 
  (Sep. 2011) from the International Cat Genome Sequencing Consortium</li> 
  <li> 
  <strong>Dolphin</strong> (<em>Tursiops truncatus</em>) <strong>turTru2</strong> &ndash; Ttru_1.4 
  (Oct. 2011) from the  Baylor College of Medicine HGSC</li> 
  <li> 
  <strong>Hedgehog</strong> (<em>Erinaceus europaeus</em>) <strong>eriEur1</strong> &ndash; 
  Draft_v1 (Jun. 2006) from the Broad Institute</li> 
  <li> 
  <strong>Kangaroo rat</strong> (<em>Dipodomys ordii</em>) <strong>dipOrd1</strong> &ndash; 
  DipOrd1.0 (Jul. 2008) from Baylor College of Medicine HGSC and the Broad Institute</li> 
  <li> 
  <strong>Manatee</strong> (<em>Trichechus manatus latirostris</em>) <strong>triMan1</strong> 
  &ndash; TriManLat1.0 (Oct. 2011) from the Broad Institute</li> 
  <li> 
  <strong>Megabat</strong> (<em>Pteropus vampyrus</em>) <strong>pteVam1</strong> &ndash; PteVap1.0 
  (Jul. 2008) from the Broad Institute</li> 
  <li> 
  <strong>Naked mole rat</strong> (<em>Heterocephalus glaber</em>) <strong>hetGla2</strong> &ndash; 
  HetGla_female_1.0 (Jan. 2012) from the Broad Institute</li> 
  <li> 
  <strong>Pig</strong> (<em>Sus scrofa</em>) <strong>susScr3</strong> &ndash; Sscrofa10.2 (Aug. 
  2011) from the Swine Genome Sequencing Consortium</li> 
  <li> 
  <strong>Pika</strong> (<em>Ochotona princeps</em>) <strong>ochPri2</strong> &ndash; OchPri2 (Jul. 
  2008) from the Broad Institute</li> 
  <li> 
  <strong>Rock hyrax</strong> (<em>Procavia capensis</em>) <strong>proCap1</strong> &ndash; 
  Procap1.0 (Jul. 2008) from the Baylor College of Medicine HGSC</li> 
  <li> 
  <strong>Shrew</strong> (<em>Sorex araneus</em>) <strong>sorAra1</strong> &ndash; SorAra1.0 (Jun. 
  2006) from the Broad Institute</li> 
  <li> 
  <strong>Sloth</strong> (<em>Choloepus hoffmanni</em>) <strong>choHof1</strong> &ndash; ChoHof1.0 
  (Jul. 2008) from the Broad Institute</li> 
  <li> 
  <strong>Squirrel</strong> (<em>Spermophilus tridecemlineatus</em>) <strong>speTri2</strong> 
  &ndash; SpeTri2.0 (Nov. 2011) from the Broad Institute</li> 
  <li> 
  <strong>Tree shrew</strong> (<em>Tupaia belangeri</em>) <strong>tupBel1</strong> &ndash; Tupbel1.0
  (Dec. 2006) from the Broad Institute</li> 
</ul>
<p>
<strong><em>Other vertebrates:</em></strong></p>
<ul>
  <li> 
  <strong>Atlantic cod</strong> (<em>Gadus morhua</em>) <strong>gadMor1</strong> &ndash; 
  GadMor_May2010 (May 2010) from Genofisk</li> 
  <li> 
  <strong>Budgerigar</strong> (<em>Melopsittacus undulatus</em>) <strong>melUnd1</strong> &ndash; 
  v6.3 (Sep. 2011) from WUSTL</li> 
  <li> 
  <strong>Coelacanth</strong> (<em>Latimeria chalumnae</em>) <strong>latCha1</strong> &ndash; 
  LatCha1 (Aug. 2011) from the Broad Institute</li> 
  <li> 
  <strong>Nile tilapia</strong> (<em>Oreochromis niloticus</em>) <strong>oreNil2</strong> &ndash; 
  OreNil1.1 (Jan. 2011) from the Broad Institute</li> 
</ul>
<p>
Our sincere thanks to the many organizations who contributed to the release of these datasets. 
We'd also like to acknowledge the Genome Browser staff who worked on this release: Hiram Clawson, 
Chin Li, Brian Raney, Brooke Rhead, Steve Heitner, Pauline Fujita, Luvina Guruvadoo, Greg Roe, and
Donna Karolchik. Please see the Genome Browser <a href="credits.html">credits page</a> for complete 
acknowledgment information and data use restrictions. All datasets may be downloaded from our
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/">ftp server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html">downloads</a> page.</p>
<p>
We regret that we don't have the resources to support the many other genomics research communities 
who would like us to visualize their data in our browser. However, we can provide you with 
information for setting up your own browser or track hub to display your data, as well as links to 
other visualization resources on the web that are specific to your genome of interest. Feel free to 
contact our <a href="mailto:genome@soe.ucsc.edu">mail list</a> for more information.</p>
<p>
*Lindblad-Toh K <em>et al</em>. 
<a href="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
target="_blank">A high-resolution map of human evolutionary constraint using 29 mammals</a>. 
<em>Nature</em>. 2011 Oct 12;478(7370):476-82.</p>

<a name="120712"></a>
<h2>Dec. 7, 2012 &nbsp;&nbsp; dbSNP 137 available for hg19</h2>
<p>
We are pleased to announce the release of four tracks derived from dbSNP build 137, available on 
the human assembly (GRCh37/hg19). dbSNP build 137 is available at NCBI. The new tracks contain 
additional annotation data not included in previous dbSNP tracks, with corresponding coloring and 
filtering options in the Genome Browser.</p>
<p>
As was the case for previous annotations based on dbSNP data, there are four tracks in this release.
One is a track containing all mappings of reference SNPs to the uman assembly, labeled &quot;All 
SNPs (137)&quot;. The other three tracks are subsets of this track and show interesting and easily 
defined subsets of dbSNP:</p>
<ul>
  <li>
  Common SNPs (137): uniquely mapped variants that appear in at least 1% of the population</li>
  <li>
  Flagged SNPs (137): uniquely mapped variants, excluding Common SNPs, that have been flagged by 
  dbSNP as &quot;clinically associated&quot;</li>
  <li>
  Mult. SNPs (137): variants that have been mapped to more than one genomic location</li>
</ul>
<p>
By default, only the Common SNPs (137) are visible; other tracks must be made visible using the 
track controls.</p>
<p>
You will find the four SNPs (137) tracks on the Human Feb. 2009 (GRCh37/hg19) browser in the 
Variation and Repeats group. The tracks were produced at UCSC by Angie Hinrichs and Steve 
Heitner.</p>

<a name="112712"></a>
<h2>Nov. 27, 2012  &nbsp;&nbsp; New UCSC Genes track released for mouse</h2>
<p>
We're happy to announce the release of a new UCSC Genes track for the GRCm38/mm10 mouse Genome 
Browser.</p>
<p>
The new release has 59,121 transcripts, compared with 55,419 in the previous version on mm9. The 
total number of canonical genes has increased from 28,661 to 31,227. Comparing the new gene set 
with the previous version:</p>
<ul>
  <li>
  48,739 transcripts did not change between versions.</li>
  <li>
  423 transcripts were not carried forward to the new version.</li>
  <li>
  5,108 transcripts are "compatible" with those in the previous set, meaning that the two 
  transcripts show consistent splicing.</li>
  <li>
  1,037 transcripts overlap with those in the previous set.</li>
</ul>
<p>
More details about the new UCSC Genes track can be found on the 
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=mm10&g=knownGene">UCSC Genes track description 
page</a>. Also, please note that GRCm38/mm10 is now the default mouse assembly on the Genome 
Browser Gateway.</p>
<p>
Many thanks to Brian Raney, Jim Kent, and Luvina Guruvadoo for their work on this track.</p>

<a name="111412"></a>
<h2>Nov. 14, 2012  &nbsp;&nbsp; ENCODE tutorials available from OpenHelix</h2>
<p>
OpenHelix and the UCSC Bioinformatics group have collaborated to provide free access to the newly 
updated ENCODE tutorial until the end of February 2013.</p>
<p>
The online tutorial suite, which runs on most Internet browsers, teaches users how to access the 
ENCODE data in the UCSC Genome Browser. The approximately 60-minute tutorial was created by 
OpenHelix in conjunction with the UCSC Bioinformatics Group, and can be viewed at no cost at 
<a href="http://www.openhelix.com/ENCODE2" target="_blank">http://www.openhelix.com/ENCODE2</a>. 
The material has been updated to reflect recent changes to the UCSC Genome Browser, and can be 
viewed sequentially or navigated using chapters and forward/backward sliders. The original, older 
tutorial is still available at <a href="http://www.openhelix.com/ENCODE" 
target="_blank">http://www.openhelix.com/ENCODE</a>.</p>
<p>
In addition to the tutorial, users can access other useful training materials including the 
animated PowerPoint slides used as a basis for the tutorial, a suggested script for the slides, 
slide handouts, and exercises.</p>

<a name="091712"></a>
<h2>Sep. 17, 2012  &nbsp;&nbsp; Retiring the Proteome Browser </h2>
<p>
We are announcing the imminent retirement of the Proteome Browser. The Proteome Browser was 
introduced in 2003 to provide access to protein-specific information independent of the genomic 
details presented in the Genome Browser. Since then we've incorporated much of this information 
into the UCSC Genes details page accessible from the Genome Browser, and into the columns of the 
Gene Sorter. Our decision to retire this tool was based on its low usage and our desire to focus 
our work on the more frequently used UCSC Genes track and the Gene Sorter. Access to the databases 
that supported the Proteome Browser (uniProt and proteome) will still be available through the Table
Browser and our public MariaDB server.</p>

<a name="081612"></a>
<h2>Aug. 16, 2012  &nbsp;&nbsp; Announcing a Genome Browser for the medium ground finch</h2>
<p> 
We have released a browser for the Medium ground finch, <em>Geospiza fortis</em>, renowned as one of
naturalist Charles Darwin's Galapagos finches. This species, which has been the subject of many 
evolutionary studies, is one of a group of birds that evolved over a few million years from a single
ancestral species into multiple species whose beak sizes and shapes are specialized for using 
different food resources. The phenotypic diversity of these birds contributed to Darwin's theory of 
evolution. The significance of this genome assembly is described in the Aug. 16, 2012, 
<a href="https://cbse.soe.ucsc.edu/news/article/1939" target="_blank">press release</a> issued by 
the UCSC Center for Biomolecular Science and Engineering (CBSE).</p>
<p>
The initial Medium ground finch genome assembly (GeoFor_1.0, UCSC version geoFor1) is the product 
of a collaboration between the Genome 10K project and Beijing Genomics Institute (BGI) to sequence 
100 vertebrate species, and is the first to be released in the UCSC Genome Browser. For more 
information about the <em>G. fortis</em> genome, see the 
<a href="https://www.ncbi.nlm.nih.gov/genome/13302" target="_blank">NCBI website</a>.</p>
<p>
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/geoFor1/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#mediumgroundfinch">Downloads</a> page. 
The browser annotation tracks were generated by UCSC and collaborators worldwide. See the 
<a href="credits.html#mediumgroundfinch_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release. We'd like to thank BGI for 
contributing the data for this assembly and acknowledge the UCSC staff members who released this 
browser: Hiram Clawson and Greg Roe.</p>

<a name="081412"></a>
<h2>Aug. 14,  2012  &nbsp;&nbsp; Changes to our website</h2>
<p>
We are pleased to announce a few changes to the look and feel of the Genome Browser website. In 
addition to a new font throughout the site, the first thing you might notice is a new menu bar 
across the top of many of the pages. We have reorganized the navigation on our web applications with
you in mind; we hope you find it more intuitive! </p>
<p> 
Additionally, we greatly simplified the search mechanism by collapsing all search methods into a 
single box.</p>
<p>
<a href="http://genome.ucsc.edu/images/autoselect.jpg">
<img src="http://genome.ucsc.edu/images/autoselect.jpg" alt="Autoselect menu" width="700"
height="129"></a></p>
<p>
In this box, you can search for chromosomal coordinate ranges, gene names, accession numbers, 
keywords, etc. For a complete list of ways to query the browser data, see 
<a href="http://genome.ucsc.edu/goldenPath/help/query.html">Querying the Genome Browser</a>.</p>

<a name="071612"></a>
<h2>Jul. 16,  2012  &nbsp;&nbsp; Tasmanian devil assembly now available in Genome Browser</h2>
<p> 
We are pleased to announce the release of a Genome Browser for the Feb. 2011 Tasmanian devil 
assembly, <em>Sarcophilus harrisii</em> (WTSI version Devil_refv7.0, UCSC version sarHar1). The 
Tasmanian devil, which is native to the island of Tasmania, is at risk of extinction due to a 
transmissible facial cancer. The devil population in Tasmania has declined by more than 80% since 
the first time the disease was observed 14 years ago. This draft assembly, created by the 
<a href="http://www.sanger.ac.uk/" target="_blank">Wellcome Trust Sanger Institute</a>, was 
sequenced using Illumina reads. This completed genome is the first step towards finding genetic 
mutations in the transmissible cancer, which will help researchers understand the disease and its 
spread.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/sarHar1/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#tasmanian_devil">Downloads</a> page. The 
Tasmanian devil browser annotation tracks were generated by UCSC and collaborators worldwide. See 
the <a href="credits.html#tasmanian_devil_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>

<a name="070612"></a>
<h2>Jul. 6, 2012  &nbsp;&nbsp; Retirement of hgText CGI</h2> 
<p>	
We are announcing the imminent retirement of the hgText CGI.  hgText was the original version of the
Table Browser, and was replaced by the new CGI, hgTables, in <a href="#100104">2004</a>. A very 
small number of our users still access the database using the old hgText CGI. We plan to remove 
hgText from our site on Jul. 24, 2012, after which all users must use hgTables.</p>

<a name="061912"></a>
<h2>Jun. 19, 2012  &nbsp;&nbsp; Many new assemblies available in the browser</h2>
<p> 
With the abundance of new vertebrate assemblies available in GenBank, the UCSC Genome Browser team 
has streamlined its browser release pipeline in the effort to keep pace. In the past week we have 
released a new browser on the Western painted turtle and updated assemblies for the cow, rat, and 
chicken. Browsers for the lesser hedgehog tenrec and the latest dog assembly should be available 
later this week, for a total of 18 assemblies released in the past half year. You can expect many 
new assemblies to show up on our website in upcoming months, along with broader multiple alignment 
annotations on selected organisms that include many of the new assemblies.</p> 
<p> 
Keep in mind that some of these new assemblies may not yet be fully populated with annotation 
tracks. Some data, such as the data underlying genes and SNPs tracks, may not be immediately 
available, and other annotations (e.g., comparative genomics tracks) may take a while to generate. 
As always, our mandated focus is on vertebrates, with a particular emphasis on richly annotating the
human genome, although we do provide occasional updated assemblies for other selected model 
organisms (fly, worm, yeast).</p> 
<p> 
We regret that we don't have the resources to support the many other genomics research communities 
who would like us to visualize their data in our browser, but we can provide you with information 
for setting up your own browser or track hub to display your data, as well as links to other 
visualization resources on the web that are specific to your genome of interest. Feel free to 
<a href="../contacts.html">contact us</a> for more information.</p> 
<p> 
Newly released assemblies:</p>
<ul> 
  <li>
  <strong>Cow</strong> (<em>Bos taurus</em>) <strong>bosTau7</strong> &ndash; Btau_4.6.1 (Oct. 2011)
  from the Bovine Genome Sequencing Consortium and Baylor College of Medicine</li>
  <li>
  <strong>Rat</strong> (<em>Rattus norvegicus</em>) <strong>rn5</strong> &ndash; RGSC Rnor_5.0 
  (Oct. 2011) from the Rat Genome Sequencing Consortium</li>
  <li>
  <strong>Chicken</strong> (<em>Gallus gallus</em>) <strong>galGal4</strong> &ndash; 
  Gallus_gallus-4.0 (Nov. 2011) from the International Chicken Genome Consortium</li>
  <li>
  <strong>Western painted turtle</strong> (<em>Chrysemys picta bellii</em>) <strong>chrPic1</strong>
  &ndash; Chrysemys_picta_bellii-3.0.1 (Dec. 2011) from the International Painted Turtle Genome 
  Sequencing Consortium</li>
  <li>
  <strong>Lesser hedgehog tenrec</strong> (<em>Echinops telfairi</em>) <strong>echTel1</strong> 
  &ndash; echTel1 (Jul. 2005) from the Broad Institute</li>
  <li>
  <strong>Dog</strong> (<em>Canis familiaris</em>) <strong>canFam3</strong> &ndash; v3.1 (Sep. 2011)
  from the Broad Institute</li>
</ul>
<p>
Released in the past half year:</p>
<ul>
  <li>
  <strong>Mouse</strong> (<em>Mus musculus</em>) <strong>mm10</strong> &ndash; GRCm38 (Dec. 2011) 
  from the Mouse Genome Reference Consortium</li>
  <li>
  <strong>Cow</strong> (<em>Bos taurus</em>) <strong>bosTau6</strong> &ndash; Bos_taurus_UMD_3.1 
  (Nov. 2009) from the University of Maryland</li>
  <li>
  <strong>Fugu</strong> (<em>Takifugu rubripes</em>) <strong>fr3</strong> &ndash; FUGU5 (Oct. 2011) 
  from the International Fugu Genome Sequencing Consortium</li>
  <li>
  <strong>Gibbon</strong> (<em>Nomascus leucogenys</em>) <strong>nomLeu1</strong> &ndash; Nleu1.0 
 (Jan. 2010) from the Gibbon Genome Sequencing Consortium</li>
  <li>
  <strong>Gorilla</strong> (<em>Gorilla gorilla gorilla</em>) <strong>gorGor3</strong> &ndash; 
  gorGor3.1 (May 2011) from the Wellcome Trust Sanger Institute</li>
  <li>
  <strong>Microbat</strong> (<em>Myotis lucifugus</em>) <strong>myoLuc2</strong> &ndash; Myoluc2.0 
  (Jul. 2010) from the Broad Institute</li>
  <li>
  <strong>Naked mole-rat</strong> (<em>Heterocephalus glaber</em>) <strong>hetGla1</strong> &ndash; 
  HetGla_1.0 (Jul. 2011) from the Beijing Genomics Institute</li>
  <li>
  <strong>Tammar wallaby</strong> (<em>Macropus eugenii</em>) <strong>macEug2</strong> &ndash; 
  Meug_1.1 (Sep. 2009) from the Tammar Wallaby Genome Sequencing Consortium</li>
  <li>
  <strong>Turkey</strong> (<em>Meleagris gallopavo</em>) <strong>melGal1</strong> &ndash; 
  Turkey_2.01 (Dec. 2009) from the Turkey Genome Consortium</li>
  <li>
  <strong>Western clawed frog</strong> (<em>X. tropicalis</em>) <strong>xenTro3</strong> &ndash; 
  v4.2 (Nov.  2009) from the DOE Joint Genome Institute</li>
  <li>
  <strong><em>C. elegans</em></strong> <strong>ce10</strong> &ndash; WS220 (Oct. 2010) from 
  WormBase</li>
  <li>
  <strong><em>S. cerevisiae</em></strong> <strong>sacCer3</strong> &ndash; SacCer_Apr2011 (Apr. 
  2011) from the Saccharomyces Genome Database</li>
</ul>
<p>
Bulk downloads of the sequence and annotation data for these assemblies are available via the Genome
Browser <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/" target="_blank">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html" target="_blank">Downloads</a> page. Be sure 
to check the <a href="credits.html">credits</a> page for data use restrictions and 
acknowledgments.</p>
<p>
Many thanks to the sequencing and assembly groups, as well as NCBI's GenBank, for making these 
genomes available for public use. We'd also like to acknowledge Hiram Clawson, Chin Li, and our 
hard-working QA team for putting together these browsers.</p>

<a name="051512b"></a>
<h2>May 15, 2012  &nbsp;&nbsp; New Fugu (fr3) assembly now available in the Genome Browser</h2>
<p> 
We are pleased to announce the release of a Genome Browser for the October 2011 assembly of the 
Fugu, <em>Takifugu rubripes</em> (JGI v5.0, UCSC version fr3). The whole genome shotgun assembly was
provided by the <a target="_blank" href="http://www.fugu-sg.org/">International Fugu Genome 
Sequencing Consortium</a> led by JGI and the <a target="_blank" 
href="http://www.imcb.a-star.edu.sg/">Singapore Institute of Molecular and Cell Biology</a> 
(IMCB).</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/fr3/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#fugu">Downloads</a> page. These data have 
<a href="http://genome.jgi-psf.org/Takru4/Takru4.download.html" target="_blank">specific conditions 
for use</a>. The Fugu (fr3) browser annotation tracks were generated by UCSC and collaborators 
worldwide. See the <a href="credits.html#fugu_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>

<a name="051512a"></a>
<h2>May 15, 2012  &nbsp;&nbsp; Naked mole-rat assembly now available in Genome Browser</h2> 
<p> 
We are pleased to announce the release of a Genome Browser for the July 2011 Naked mole-rat, 
<em>Heterocephalus glaber</em> (BGI version HetGla1.0, UCSC version hetGla1). This draft assembly,  
produced by the <a href="http://www.genomics.cn/en/index" target="_blank">Beijing Genomics 
Institute</a>, has been sequenced using Illumina reads with a sequencing depth of >20x. There are 
39,266 scaffolds with a total size of 2,643,961,837 bases. The N50 size of the scaffolds is 
1,603,177 bases with 502 scaffolds larger, and 38,764 scaffolds smaller. The mitochondrial sequence 
is also available as the virtual chromosome &quot;chrM&quot;.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/hetGla1/" target="_blank">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#naked_mole-rat" target="_blank">Downloads</a>
page. These data have <a href="credits.html#naked_mole-rat_use">specific conditions for use</a>. 
The naked mole-rat browser annotation tracks were generated by UCSC and collaborators worldwide. See
the <a href="credits.html#naked_mole-rat_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>

<a name="050812"></a>
<h2>May 8, 2012  &nbsp;&nbsp; OpenHelix to conduct free Genome Browser webinars in May</h2>
<p> 
<a href="http://www.openhelix.com" target="_blank">OpenHelix</a> is conducting two free webinars on 
the UCSC Genome Browser:</p> 
<ul>
   <li>
   UCSC Genome Browser: An Introduction (Thursday, May 17, 1pm EDT) -- designed for new users of
   the UCSC Genome Browser and those who want to improve their skills at basic navigation and 
   display.</li> 
   <li>
   UCSC Table Browser and Custom Tracks (Thursday, May 24, at 1pm EDT) -- advanced Genome Browser 
   topics, including creating Custom Tracks and using the Table Browser.</li> 
</ul> 
<p> 
The webinars will last approximately 1hr 15min.</p> 
<p> 
Seating is limited and registration is required. Slides may be downloaded from the 
<a href="http://www.openhelix.com/ucsc" target="_blank">OpenHelix website</a> prior to the
webinar.</p> 
<p> 
OpenHelix provides training materials and programs on hundreds of free, publicly accessible 
bioinformatics and genomics resources. To browse their entire collection of UCSC genomics tutorials,
go to the OpenHelix <a href="http://www.openhelix.com/ucsc" target="_blank">UCSC training
page</a>.</p> 

<a name="050212"></a>
<h2>May 2, 2012  &nbsp;&nbsp; OpenHelix releases updated Genome Browser tutorials</h2> 
<p> 
<a href="http://www.openhelix.com" target="_blank">OpenHelix</a> has released updated versions of 
two Genome Browser tutorial suites:</p> 
<ul> 
  <li>
  <a href="http://www.openhelix.com/ucsc" target="_blank">UCSC Genome Browser: An 
  Introduction</a></li>
  <li>
  <a href="http://www.openhelix.com/cgi/tutorialInfo.cgi?id=28" target="_blank">UCSC Genome Browser:
  Custom Tracks and Table Browser</a></li>
</ul> 
<p>
The OpenHelix website also provides UCSC tutorials on the Gene Sorter and VisiGene 
(<a href="http://www.openhelix.com/cgi/tutorialInfo.cgi?id=76" target="_blank">UCSC Genome Browser: 
The Additional Tools</a>), <a href="http://www.openhelix.com/cgi/tutorialInfo.cgi?id=116" 
target="_blank">ENCODE Foundations</a>, and the 
<a href="http://www.openhelix.com/cgi/tutorialInfo.cgi?id=112" target="_blank">UCSC Archaeal Genome 
Browser</a>.</p> 
<p> 
Each suite contains an online, narrated tutorial that highlights and explains the features and 
functionality needed to use the UCSC tools effectively, and a collection of training materials that 
includes the PowerPoint slides used as a basis for the tutorial, a suggested script for the slides, 
slide handouts, and exercises. These materials, all of which are free, can be used as a reference 
for users or as a ready-made presentation for training others on the use of the Genome Browser.</p> 
<p>
For more information on the tutorials and OpenHelix, see the OpenHelix 
<a href="http://www.digitaljournal.com/pr/691373#ixzz1tkDsoGSS" target="_blank">press 
release</a>.</p>

<a name="041812"></a>
<h2>Apr. 18, 2012  &nbsp;&nbsp; GRC planning phase for GRCh38 human reference assembly</h2> 
<p> 
The <a href="https://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/" target="_blank">Genome 
Reference Consortium</a> (GRC) plans to update the human reference assembly to GRCh38 in the summer 
of 2013. This revision is aimed at addressing issues found with the current model for representing 
genome assemblies, which uses a single, preferred tiling path to produce a single consensus 
representation of the genome. Subsequent analysis has shown that for most mammalian genomes a single
tiling path is insufficient to represent a genome in regions with complex allelic diversity. The 
GRC is working to create assemblies that better represent this diversity and provide more robust 
substrates for genome analysis.</p> 
<p>
If you have questions, concerns or feedback about the plans for the GRCh38 assembly, the GRC would 
<a href="https://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/contact.shtml" 
target="_blank">like your input</a>.</p>

<a name="030712"></a>
<h2>Mar. 7, 2012  &nbsp;&nbsp; New mouse browser available</h2> 
<p> 
We have released the latest Genome Browser for the Dec. 2011 mouse genome assembly produced by the 
<a href="https://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/mouse/" target="_blank">Mouse 
Genome Reference Consortium</a> (Genome Reference Consortium GRCm38, UCSC version mm10). This 
version, which includes approximately 2.6 Gb of sequence, is considered to be &quot;essentially 
complete&quot;. The assembly includes chromosomes 1-19, X, Y, M (mitochondrial DNA) and chr*_random 
(unlocalized) and chrUn_* (unplaced clone contigs). Note that the UCSC mm10 database contains only 
the reference strain C57BL/6J.</p> 
<p>					
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/mm10/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#mouse">Downloads</a> page. The mouse browser 
annotation tracks were generated by UCSC and collaborators worldwide. See the 
<a href="credits.html#mouse_credits" target="_blank">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>

<a name="022912"></a>
<h2>Feb. 29, 2012  &nbsp;&nbsp; RGD Genes available for rn4</h2> 
<p> 
We are pleased to announce the release of RGD Genes for rat (Baylor 3.4/rn4). This track shows 
protein-coding gene annotations curated by <a href="http://rgd.mcw.edu/" target="_blank">RGD</a>. 
This track replaces UCSC Known Genes as the main gene track for this assembly.</p> 
<p> 
More details about this track can be found on the <a href="../cgi-bin/hgTrackUi?db=rn4&g=rgdGene2" 
target="_blank">RGD Genes track description page</a>.</p> 
<p> 
We'd like to thank RGD for providing the base annotations for this track. The RGD Genes track was 
produced by Fan Hsu, Mary Goldman and Hiram Clawson.</p>

<a name="021012"></a>
<h2>Feb.10, 2012  &nbsp;&nbsp; dbSNP 135 available for hg19</h2> 
<p> 
We are pleased to announce the release of four tracks derived from dbSNP build 135, available on the
human assembly (GRCh37/hg19). dbSNP build 135 is available at NCBI. The new tracks contain 
additional annotation data not included in previous dbSNP tracks, with corresponding coloring and 
filtering options in the Genome Browser.</p> 
<p> 
As was the case with dbSNP build 132, there are four tracks in this release. One track, labeled 
&quot;All SNPs (135)&quot;, contains all mappings of reference SNPs to the human assembly. The other
three tracks are subsets of this track and show interesting and easily defined subsets of dbSNP:</p>
<ul> 
  <li>
  Common SNPs (135): uniquely mapped variants that appear in at least 1% of the population</li> 
  <li>
  Flagged SNPs (135): uniquely mapped variants, excluding Common SNPs, that have been flagged by 
  dbSNP as &quot;clinically associated&quot;</li> 
  <li>
  Mult. SNPs (135): variants that have been mapped to more than one genomic location</li> 
</ul> 
<p> 
By default, only the Common SNPs (135) are visible; other tracks must be made visible using the 
track controls.</p> 
<p> 
You will find the four SNPs (135) tracks on the Human Feb.  2009 (GRCh37/hg19) browser in the 
Variation and Repeats group.</p> 
<p> 
Many thanks to dbSNP at NCBI for the data, and to Mary-Claire King at the University of Washington 
for helpful comments that motivated us to enhance our dbSNP annotations. The tracks were produced at
UCSC by Angie Hinrichs and Brooke Rhead.</p>

<a name="020212"></a>
<h2>Feb. 2, 2012  &nbsp;&nbsp; New UCSC Genes track released for GRCh37/hg19</h2>
<p> 
We're happy to announce the release of an updated UCSC Genes track for the GRCh37/hg19 human Genome 
Browser. This release includes more noncoding transcripts based on data from 
<a href="http://rfam.sanger.ac.uk/" target="_blank">Rfam</a> and from the 
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=tRNAs">tRNA Genes track</a> contributed 
by the <a href="http://lowelab.ucsc.edu/" target="_blank">Todd Lowe lab</a> at UCSC.</p> 
<p> 
The new release has 80,922 total transcripts, compared with 77,614 in the previous version. The 
total number of canonical genes has increased from 27,297 to 31,227. Comparing the new gene set with
the previous version:</p> 
<ul> 
  <li>
  24,804 transcripts did not change between versions.</li>
  <li>
  483 transcripts were not carried forward to the new version.</li> 
  <li>
  46,367 transcripts are &quot;compatible&quot; with those in the previous set, meaning that the two
  transcripts show consistent splicing. In most cases, the old and new transcripts differ in the 
  lengths of their UTRs.</li> 
  <li>
  5,960 transcripts overlap with those in the previous set but do not show consistent splicing, 
  i.e., they contain overlapping introns with differing splice sites.</li> 
</ul> 
<p> 
Additionally, several improvements where made to the UCSC Genes build pipeline:</p> 
<ul> 
  <li>
  We developed new genomic alignment protocols for aligning short sequences, facilitating the 
  expansion of the noncoding RNA content.</li> 
  <li>
  We changed the way that gene symbols are assigned to transcripts so that names from curated 
  sources are favored over names coming directly from GenBank mRNA records. This change resolved 
  several confusing naming issues reported to us by users.</li> 
  <li>
  We tracked down and corrected a bug in the pipeline that was causing a number of poor quality 
  protein mappings to displace high quality protein mappings.</li> 
  <li>
  We have started producing two new tables, knownGeneTxMrna and knownGeneTxPep, that contain 
  sequence derived from the genome rather than from the mRNA used for the transcript.</li>  
</ul> 
<p> 
More details about the new UCSC Genes track can be found on the 
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=knownGene">UCSC Genes track description 
page</a>.</p> 
<p> 
Many thanks to Melissa Cline, Jim Kent, Brian Raney, Mark Diekhans, and Brooke Rhead for their work 
on this track.</p>

<a name="230112"></a>
<h2>Jan. 23, 2012  &nbsp;&nbsp; Wallaby browser available</h2> 
<p> 
We have released a Genome Browser for the September 2009 genome assembly of the Tammar wallaby, 
<em>Macropus eugenii</em> (TWGS version Meug_1.1, UCSC version macEug2). This assembly, which has 
been sequenced to 2X coverage, was produced by the Tammar Wallaby Genome Sequencing Consortium, a 
collaboration between the Baylor College of Medicine 
<a href="http://www.hgsc.bcm.tmc.edu/project-species-m-Wallaby.hgsc" target="_blank">Human Genome 
Sequencing Center</a> and the <a href="http://www.agrf.org.au/" target="_blank">Australian Genome 
Research Facility</a>.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/macEug2/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#wallaby">Downloads</a> page. These data have 
<a href="credits.html#wallaby_use">specific conditions for use</a>.</p>  
<p> 
We'd like to thank the Tammar Wallaby Genome Sequencing Consortium for providing this assembly. The 
wallaby browser and annotation tracks were produced by Chin Li and Luvina Guruvadoo. See the 
<a href="credits.html#wallaby_credits">Credits</a> page for a detailed list of the organizations and
 individuals who contributed to this release.</p> 

<a name="030112"></a>
<h2>Jan. 3, 2012  &nbsp;&nbsp; Roadmap Epigenomics now available through data hub at Washington 
University</h2> 
<p> 
We are pleased to announce the release of the Roadmap Epigenomics data on the UCSC Genome Browser 
through our data hub utility. The Roadmap Epigenomics Project is part of The NIH Common Fund's 
<a href="https://commonfund.nih.gov/epigenomics/" target="_blank">Epigenomics Program</a>. 
It was launched with the goal of producing a public resource of human epigenomic data to catalyze 
basic biology and disease-oriented research. The Consortium leverages experimental pipelines built 
around next-generation sequencing technologies to map DNA methylation, histone modifications, 
chromatin accessibility and small RNA transcripts in stem cells and primary ex vivo tissues 
selected to represent the normal counterparts of tissues and organ systems frequently involved in 
human disease. The Consortium expects to deliver a collection of normal epigenomes that will provide
a framework or reference for comparison and integration within a broad array of future studies.</p> 
<p> 
All data were produced and processed by the <a href="http://www.roadmapepigenomics.org/" 
target="_blank">Roadmap Epigenomics Mapping Consortium</a>, and will be periodically updated. 
Genome Browser tracks were constructed and hosted by <a href="http://vizhub.wustl.edu/" 
target="_blank">VizHub</a> at Washington University in St.  Louis. Tracks are available at UCSC 
Genome Browser via the data hub utility, or follow this 
<a href="http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&hubUrl=http://vizhub.wustl.edu/VizHub/RoadmapRelease4.txt">link</a>.  
The Roadmap Epigenomics Mapping Consortium is responsible for the quality of the data.</p>

<!-- ============= 2011 archived news ============= -->
<a name="2011"></a>

<a name="191211"></a>
<h2>Dec. 19, 2011  &nbsp;&nbsp; Variant Call Format (VCF) now supported in Genome Browser</h2> 
<p> 
We are pleased to announce that the UCSC Genome Browser now supports Variant Call Format (VCF). 
VCF is a flexible and extendable line-oriented text format developed by the 1000 Genomes Project for
releases of single nucleotide variants, indels, copy number variants and structural variants 
discovered by the project. Similar to bigBed, bigWig and BAM, the Browser transfers only the 
portions of VCF files necessary to display viewed regions, making VCF a fast and attractive option 
for large data sets. VCF files will need to be compressed and indexed using the tabix package 
available from <a href="http://samtools.sourceforge.net/" target="_blank">SAMtools</a>. This new 
format is available for use in custom tracks and data hubs. For more information about VCF and 
tabix, please see our <a href="help/vcf.html">VCF Track Format help page</a>.</p>

<a name="071111"></a>
<h2>Nov. 7, 2011  &nbsp;&nbsp; UMD cow assembly now available in Genome Browser</h2> 
<p> 
In response to the many requests from the bovine research community, we now offer the University of 
Maryland (UMD) <em>Bos taurus</em> assembly in the Genome Browser, in addition to the assemblies 
produced by the Baylor College of Medicine Human Genome Sequencing Center. The UMD v3.1 assembly 
(Nov. 2009, UCSC bosTau6) was produced by the <a href="http://www.cbcb.umd.edu/" 
target="_blank">UMD Center for Bioinformatics and Computational Biology</a> (CBCB) in College Park, 
MD.</p> 
<p> 
The UMD 3.1 assembly is based on reads downloaded from the NCBI Trace Archive, most of which were 
sequenced by the Baylor College of Medicine. UMD reports that they were able to apply several 
assembly improvement techniques to produce an assembly that is substantially more accurate and 
complete than previous alternative versions, as evaluated using independent metrics. Based on UMD 
statistics, this assembly provides more genome coverage, closes thousands of gaps, corrects many 
erroneous inversions, deletions, and translocations, and fixes thousands of single-nucleotide 
errors. For more information, see Zimin, AV <em>et al.</em>, 
<a href="http://genomebiology.com/2009/10/4/R42" target="_blank">A whole-genome assembly of the 
domestic cow, <em>Bos taurus</em></a>, <em>Genome Biol.</em> 2009;10(4). </p> 
<p> 
Downloads of the UMD bovine data and annotations may be obtained from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/bosTau6/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#cow">Downloads</a> page.</p> 
<p>
We'd like to thank the UMD CBCB for providing this genome assembly. The UMD Cow Genome Browser 
and annotation tracks were produced by Hiram Clawson, Chin Li, Brooke Rhead, Greg Roe, Steve 
Heitner, and Donna Karolchik. See the <a href="credits.html#cow_credits">Credits</a> page for a 
detailed list of the organizations and individuals who contributed to this release.<p>

<a name="101011"></a> 
<h2>Oct. 10, 2011 &nbsp;&nbsp; Updated yeast browser released</h2> 
<p> 
We are happy to announce the release of an updated browser for the <em>Saccharomyces cerevisiae</em>
yeast genome. The April 2011 sacCer3 assembly corresponds to the S288c assembly produced by the 
<a href="http://www.yeastgenome.org/" target="_blank">Saccharomyces Genome Database</a> (SGD) 
project. For more information on this genome, see the 
<a href="http://www.yeastgenome.org/cache/genomeSnapshot.html" target="_blank">SGD Genome 
Overview</a>.</p> 
<p> 
Downloads of the yeast data and annotations may be obtained from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/sacCer3/" target="_blank">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#yeast" target="_blank">Downloads</a> 
page.</p> 
<p> 
We'd like to thank the SGD for providing this genome assembly. The <em>S. cerevisiae</em> Genome 
Browser and annotation tracks were produced by Hiram Clawson, Greg Roe, and Steve Heitner. See the 
<a href="credits.html#yeast_credits">Credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p> 

<a name="090811"></a>
<h2>Sep. 8, 2011  &nbsp;&nbsp; New navigation and display features</h2> 
<p> 
We've added several new features to the Genome Browser that make it easier to quickly configure and 
navigate around in the browser's annotation tracks window.</p> 
<p> 
<strong>Automatic image resizing:</strong> The first time the annotation track window is displayed, 
or after the Genome Browser has been reset, the size of the track window is now set by default to 
the width that best fits your Internet browser window. If you subsequently resize your browser 
window, you can automatically adjust the annotation track image size to the new width by clicking 
the <em>resize</em> button under the track image. The default width can still be manually overridden
on the Track Configuration page.</p> 
<p> 
<strong>Scrolling left or right in the track window:</strong> You can now scroll (pan) horizontally 
through the tracks image by clicking on the image, dragging the cursor to the left or right, then 
releasing the mouse button. The view may be scrolled by up to one image width.</p> 
<p> 
<strong>Improved drag-zoom navigation:</strong>  The browser's &quot;drag-and-zoom&quot; feature 
lets you quickly zoom to a specific region of interest on the annotation tracks image. To define the
region you wish to zoom to, click-and-hold the mouse button on one edge of the desired zoom area 
(which can be anywhere in the tracks window), depress the shift key, drag the mouse right or left 
to highlight the selection area, then release the mouse button. The annotation tracks image will 
automatically zoom to the new region. The Genome Browser still supports the earlier implementation 
of this feature, which restricted the click-drag to the Base Position track area of the image, but 
did not require the shift key to be pressed.</p> 
<p>  
<strong>Reordering groups of tracks:</strong> You can now vertically reposition an entire group of 
associated tracks in the tracks image (such as all the displayed subtracks in a composite track) by 
clicking and holding the gray bar to the left of the tracks, dragging the group to the new position,
then releasing the mouse button. To move a single track up or down, click and hold the mouse button 
on the side label, drag the highlighted track to the new position, then release the mouse
button.</p> 
<p>
If you haven't yet tried the browser's right-click menu for quick access to frequently used track 
configuration features and functionality, read more <a href="newsarch.html#011011">here</a>.</p>

<a name="081811"></a>
<h2>Aug. 18, 2011  &nbsp;&nbsp; New feature: Track Data Hubs</h2>
<p>
We are pleased to announce a new feature in the UCSC Genome Browser: track data hubs. Track hubs 
are web-accessible directories of genomic data that can be viewed on the UCSC Genome Browser 
alongside native annotation tracks. The Track Hub utility allows efficient access to data sets from 
around the world through the familiar Genome Browser interface. Browser users can display tracks 
from any public track hub that has been registered with UCSC. Additionally, users can import data 
from unlisted hubs or can set up, display, and share their own track hubs.</p> 
<p>
The data underlying the tracks in a hub reside on the remote server of the data provider rather than
at UCSC. The data are stored in compressed binary indexed files in bigBed, bigWig or BAM format that
contain the data at several resolutions. When a hub track is displayed in the Genome Browser, only 
the relevant data needed to support the view of the current genomic region are transmitted rather 
than the entire file. The transmitted data are cached on the UCSC server to expedite future access. 
This on-demand transfer mechanism eliminates the need to transmit large data sets across the 
Internet, thereby minimizing upload time into the browser.</p> 
<p>
To check out some track data hubs that are already available, most notably those from the NIH 
Roadmap Epigenomics Mapping Consortium, press the &quot;Track Hubs&quot; button in the genome 
Browser or follow this link: 
<a href="/cgi-bin/hgHubConnect">http://genome.ucsc.edu/cgi-bin/hgHubConnect</a>.</p> 
<p>
To read about using hubs or creating your own, see:</p>
<ul> 
<li><a href="help/hgTrackHubHelp.html">Using UCSC Genome Browser Track Hubs</a></li> 
<li>Raney et al.  <a href="http://bioinformatics.oxfordjournals.org/content/30/7/1003" 
target="_blank">Track Data Hubs enable visualization of user-defined genome-wide annotations on the 
UCSC Genome Browser</a>. <em>Bioinformatics</em>. 2014 Apr 1;30(7):1003-5. Epub 2013 Nov 13.</li> 
</ul>

<a name="061611"></a> 
<h2>Jun. 16, 2011  &nbsp;&nbsp; Re-engineered OMIM tracks released</h2> 
<p> 
We announce today the release of our newly re-engineered OMIM (Online Mendelian Inheritance in Man) 
tracks for both hg18 and hg19. With the kind assistance of Ada Hamosh (director), Joanna Amberger 
and Francois Schiettecatte of the OMIM project, we have divided the OMIM records into three separate
tracks:</p> 
<ul> 
  <li><strong>OMIM Allelic Variant SNPs</strong> -- Variants in the OMIM database that have 
  associated dbSNP identifiers.</li>
  <li>
  <strong>OMIM Genes</strong> -- Genomic positions of gene entries in the OMIM database. The 
  coloring indicates the associated OMIM phenotype class.</li>
  <li> 
  <strong>OMIM Phenotypes - Gene Unknown</strong> -- Regions known to be associated with a 
  phenotype, but for which no specific gene is known to be causative. This track also includes 
  known multi-gene syndromes.</li>
</ul>
<p> 
The new tracks can be found in the Phenotype and Disease Associations track group below the browser 
graphic and are searchable by OMIM number. In most cases, simply typing the 6-digit MIM number into 
the position/search box on the Browser will take you to the record.</p>
<p> 
The OMIM data are the property of Johns Hopkins University and will not be available for download 
from UCSC. Please contact the OMIM project at <a href="http://omim.org/downloads" 
target="_blank">omim.org</a> for download information.</p> 
<p> 
UCSC thanks engineers Fan Hsu, Brooke Rhead and Robert Kuhn for this release.</p>

<a name="060911"></a>
<h2>Jun. 9, 2011  &nbsp;&nbsp; UCSC Preview Browser available</h2> 
<p> 
Early access to ENCODE and other UCSC browser data tracks under construction is now available from 
the new UCSC Preview Browser site: 
<a href="http://genome-preview.ucsc.edu">http://genome-preview.ucsc.edu</a>. This site is a 
weekly mirror of our internal development server for public access. Data and tools there are under 
construction, have not been quality reviewed, and are subject to change at any time. We provide this
site for early access, with the warning that it is less available and stable than our public site. 
For high-quality reviewed annotations on our production server, visit our public website: 
<a href="../index.html">http://genome.ucsc.edu</a>.</p>

<a name="060711"></a> 
<h2>Jun. 7, 2011  &nbsp;&nbsp; Updated lizard browser available</h2> 
<p> 
We have released a Genome Browser for the May 2010 genome assembly of the green anole lizard, 
<em>Anolis carolinensis</em> (Broad version AnoCar2.0, UCSC version anoCar2). This assembly, which 
has been sequenced to 7.1X coverage, was produced by the <a href="http://www.broad.mit.edu/" 
target="_blank">Broad Institute</a>, Cambridge, MA.</p>
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/anoCar2/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#lizard">Downloads</a> page. These data have 
<a href="credits.html#lizard_use">specific conditions for use</a>.</p>  
<p> 
We'd like to thank the Broad Institute for providing this assembly. The lizard Genome Browser and 
annotation tracks were produced by Hiram Clawson, Brian Raney, and Luvina Guruvadoo. See the 
<a href="credits.html#lizard_credits">Credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p>

<a name="052611"></a>
<h2>May 26, 2011  &nbsp;&nbsp; New release of UCSC Genes for mouse</h2> 
<p> 
We've released an updated set of UCSC Genes for the mm9 (NCBI Build 37) mouse browser. This version 
of the gene set was generated using the same computational pipeline as the previous mm9 UCSC Genes, 
but is based on more recent GenBank data.</p>
<p> 
The new release has 55,419 total transcripts, compared with 49,409 in the previous version. The 
total number of canonical genes has increased from 27,389 to 28,661. Comparing the new gene set with
the previous version:</p>
<ul> 
  <li>
  33,977 transcripts did not change between versions</li>
  <li>
  69 transcripts were not carried forward to the new version</li>
  <li>
  13,701 transcripts are &quot;compatible&quot; with those in the previous set</li>
  <li>
  1,662 transcripts overlap with those in the previous set</li>
</ul>

<a name="051311"></a>
<h2>May 13, 2011  &nbsp;&nbsp; Free browser training webinar from OpenHelix</h2> 
<p>
The UCSC Bioinformatics Group announces two free webinars on the UCSC Genome Browser. The webinars 
will be conducted by OpenHelix, a provider of training on many free, publicly accessible 
bioinformatics and genomics resources.</p> 
<p> 
The webinars (one hour and 15 minutes long) will cover the topics needed to effectively use the 
Genome Browser. The first webinar, to be held Tuesday, May 24, 11:00-12:15 PM PDT, 
<strong>Introduction to the UCSC Genome Browser</strong>, is designed for new users of the UCSC 
Genome Browser, and those who want to improve their skills at basic navigation and display.</p> 
<p> 
The second webinar, <strong>UCSC Genome Browser: Custom Tracks and Table Browser</strong>, will be 
held Thursday, May 26, 1:00-2:15 PM PDT.</p> 
<p>Sign-up required. For details see <a href = "http://blog.openhelix.eu/?p=8359" 
target="_blank">OpenHelix</a>.</p>

<a name="050111"></a>
<h2>May 1, 2011  &nbsp;&nbsp; ENCODE User's Guide and usability survey</h2> 
<p> 
The ENCODE project has just published an overview of their ongoing large-scale efforts to interpret 
the human genome sequence in the journal PLoS Biology. <a target=_blank 
href="http://www.plosbiology.org/article/info:doi/10.1371/journal.pbio.1001046">A Users Guide to the
Encyclopedia of DNA Elements</a>, highlights the scope of data production, and provides guidance for
locating and using the data. With the vast amount of data now available, the ENCODE Data 
Coordination Center at UCSC continues to focus on improving the accessiblity and usability of this 
valuable resource. We invite all current and prospective users of ENCODE data to participate in the 
<a target=_blank href="http://www.surveymonkey.com/s/5YYTRWQ">2011 ENCODE Usability Survey</a>. Your
input will help us to make this data more accessible to the scientific community. Thank you!</p>

<a name="041811.2"></a>
<h2>Apr. 18, 2011  &nbsp;&nbsp; dbSNP 132 available for hg19</h2> 
<p> 
We are pleased to announce the release of four tracks derived from 
<a href="https://www.ncbi.nlm.nih.gov/projects/SNP/" target="_blank">dbSNP</a> build 132, available 
on the human assembly (GRCh37/hg19). dbSNP build 132 is available at NCBI. The new tracks contain 
additional annotation data not included in previous dbSNP tracks, with corresponding coloring and 
filtering options in the Genome Browser.</p> 
<p> 
As was the case for previous annotations based on dbSNP data, there is a track that contains all 
mappings of reference SNPs to the human assembly; it is now labeled &quot;All SNPs (132)&quot;. 
Three new tracks have been added to show interesting and easily defined subsets of dbSNP:</p> 
<ul> 
  <li>
  Common SNPs (132): uniquely mapped variants that appear in at least 1% of the population</li> 
  <li>
  Flagged SNPs (132): uniquely mapped variants, excluding Common SNPs, that have been flagged by 
  dbSNP as &quot;clinically associated&quot;</li> 
  <li>
  Mult. SNPs (132): variants that have been mapped to more than one genomic location</li> 
</ul> 
<p> 
By default, only the Common SNPs (132) are visible; other tracks must be made visible using the 
track controls.</p> 
<p> 
The counts of distinct reference SNPs and their mappings to hg19 differ greatly among the 
tracks:</p> 
<table> 
  <tr>
    <th>Track</th> 
    <th>Reference SNPs</th> 
    <th>Mappings to hg19</th></tr> 
  <tr> 
    <td>Common SNPs (132)</td> 
    <td>13,842,381</td> 
    <td>14,024,295 *</td></tr> 
  <tr> 
    <td>Flagged SNPs (132)</td> 
    <td>17,899</td> 
    <td>18,084 *</td></tr> 
  <tr> 
    <td>Mult. SNPs (132)</td> 
    <td>1,114,599</td> 
    <td>3,568,988</td></tr> 
  <tr> 
    <td>All SNPs (132)</td> 
    <td>30,030,754</td> 
    <td>33,026,121</td></tr> 
</table> 
<p> 
*The Common and Flagged tracks have more mappings than SNPs due to the alternate haplotype sequences
and the pseudoautosomal regions on X and Y. SNPs are considered uniquely mapped if they map only 
once to a haploid reference genome. These regions add non-haploid sequence to the reference genome; 
therefore, multiple mappings involving these regions are still considered unique.</p> 
<p> 
You will find the four SNPs (132) tracks on the Human Feb. 2009 (GRCh37/hg19) browser in the 
Variation and Repeats group.</p> 
<p> 
Many thanks to dbSNP at NCBI for the data, and to 
<a href="http://www.gs.washington.edu/faculty/king.htm" target="_blank">Mary-Claire King</a> at the 
University of Washington for helpful comments that motivated us to enhance our dbSNP annotations. 
The tracks were produced at UCSC by Angie Hinrichs, Brooke Rhead, Bob Kuhn, and Jim Kent.</p>

<a name="041811.1"></a>
<h2>Apr. 18, 2011  &nbsp;&nbsp; Sheep browser released</h2> 
<p>
We are happy to announce a Genome Browser for the initial release of sheep (<em>Ovis aries</em>) 
genome. The Feb. 2010 <em>Ovis aries</em> draft assembly (NCBI project 
<a href="https://www.ncbi.nlm.nih.gov/sites/entrez?Db=genomeprj&cmd=ShowDetailView&TermToSearch=10709" 
target="_blank">10709</a>, 
<a href="ftp://ftp.ncbi.nlm.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Ovis_aries/Ovis_aries_1.0/" 
target="_blank">GCA_000005525.1</a>) was produced by the <a href="http://www.sheephapmap.org/" 
target="_blank">International Sheep Genomics Consortium (ISGC) sequencing center</a>. The 
mitochondrial genome was sequenced by <a href="http://www.uni-giessen.de/cms/" 
target="_blank">Justus-Liebig-University</a>.</p> 
<p>
There are 27 chromosomes: chr1 - chr26, chrX, plus the mitochondrial sequence chrM from GenBank 
accession NC_001941, for a total sequence length of 2,860,512,983 bases. There are 1,659,241,706 
&quot;N&quot; bases in gaps, leaving 1,201,271,277 ACGT bases.</p> 
<p>The genome has been sequenced using 454 FLX sequencing to 3X coverage (DNA derived from six 
female sheep each sequenced with 0.5X coverage). Contigs were assembled based on alignment to the 
bovine genome and ordered into ovine chromosomes using BAC-end sequences and the virtual sheep 
genome.</p> 
<p>
We'd like to thank the International Sheep Genomics Consortium (ISGC) sequencing center and 
Justus-Liebig-University for providing this assembly. We'd also like to acknowledge the UCSC team 
who worked on this release: Chin Li, Greg Roe, and Luvina Guruvadoo. The sheep browser annotation 
tracks were generated by UCSC and collaborators worldwide.</p>
		    
<a name="030411"></a>
<h2>Mar. 4, 2011  &nbsp;&nbsp; The meaning of red: consensus on color scheme for CNVs</h2> 
<p> 
Consensus was reached on color standards to represent CNV loss (red) and gain (blue) in genomics 
databases by DGV, NCBI, DECIPHER, UCSC and ISCA during the 1st annual ISCA Scientific 
Conference.</p> 
<p> 
At the recent ISCA Scientific Conference in Atlanta, GA, Nigel Carter, representing DECIPHER, made 
a plea for a standardized color scheme for representation of CNV loss (deletion) or gain 
(duplication). Discussions in person and by email with leaders of major CNV/genomic databases 
(including DGV, dbVar, ISCA, and the UCSC Genome Browser) resulted in the recommendation to utilize 
red to represent loss/deletion (consistent with the original convention for CGH on metaphase 
chromosomes) and blue to represent gain/duplication (avoiding green which is difficult for 
color-blind individuals to discriminate from red).</p> 
<p> 
Announcememnt reprinted from the website for the International Standards for Cytogenomic Arrays 
(<a href = "http://iscaconsortium.org" target="_blank">ISCA</a>) Consortium. The UCSC Genome Browser
is now conforming to this standard for the DECIPHER track and the DGV track on human assemblies 
(which was released today) and on any future tracks derived from dbVar.</p> 
<p> 
Thanks to Nigel Carter, Steve Scherer, Deanna Church, Angie Hinrichs, Fan Hsu, Pauline Fujita and 
Robert Kuhn.</p>

<a name="030111"></a>
<h2>Mar. 1, 2011  &nbsp;&nbsp; Updated chimpanzee browser released</h2> 
<p> 
We are happy to announce the release of a Genome Browser for the latest release of the chimpanzee 
(<em>Pan troglodytes</em>) genome. The Oct. 2010 assembly -- CGSC v2.1.3, UCSC version panTro3 -- 
was produced by the 
<a href="http://genome.wustl.edu/genome.cgi?GENOME=Pan%20troglodytes&SECTION=collaborators" 
target=_BLANK>Chimpanzee Sequencing and Analysis Consortium</a>.</p> 
<p> 
This assembly covers about 97 percent of the genome and is based on 6X sequence coverage. It 
comprises 192,898 contigs with an N50 length of 44 kb and 33,990 supercontigs with an N50 length of 
8.4 Mb. Improvements introduced with the 2.1.3 assembly include the addition of over 300,000 
finishing reads and the merging in of 640 finished BACS. Approximately 49,000 additional merges were
made in this assembly as compared with the 2.1 assembly.</p> 
<p> 
The whole genome shotgun data were derived primarily from the donor Clint, a captive-born male 
chimpanzee from the Yerkes Primate Research Center in Atlanta, GA, USA. The sequence data were 
assembled and organized by the Washington University Genome Center. The underlying whole genome 
shotgun data were generated at the Washington University School of Medicine and the Broad Institute.
For detailed information about the assembly process, see the sequencing and analysis discussion in 
the <a href="https://www.ncbi.nlm.nih.gov/nuccore/AACZ00000000" target="_blank">GenBank accession 
record</a>.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/panTro3/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#chimp">Downloads</a> page. The complete set 
of sequence reads is available at the <a href="https://www.ncbi.nlm.nih.gov/Traces" 
target="_blank">NCBI trace archive</a>. These data have <a href="credits.html#chimp_use">specific 
conditions for use</a>.</p> 
<p> 
We'd like to thank the International Chimpanzee Sequencing and Analysis Consortium, Washington 
University at St. Louis School of Medicine Genome Sequencing Center, and the Broad Institute for 
providing this assembly.  We'd also like to acknowledge the UCSC team who worked on this release: 
Hiram Clawson, Greg Roe, and Donna Karolchik. The chimpanzee browser annotation tracks were 
generated by UCSC and collaborators worldwide.</p>

<a name="011911"></a>
<h2>Jan. 19, 2011  &nbsp;&nbsp; Browser released for zebrafish Zv9 assembly</h2> 
<p> 
The most recent zebrafish assembly -- Zv9 (UCSC version danRer7, Jul. 2010) -- is now available in 
the UCSC Genome Browser. This assembly was produced by <a href="http://www.sanger.ac.uk/" 
target="_blank">The Wellcome Trust Sanger Institute</a>, UK.</p> 
<p>
The Zv9 assembly comprises a sequence length of 1.4 Gb in 26 chromosomes and 1,107 scaffolds. This 
assembly is based on a clone path sorted with the high-density meiotic map SATMAP (Clark et al., in 
preparation). The data freeze was taken on Apr. 1, 2010. The remaining gaps were filled with 
sequence from WGS31, a combined Illumina and capillary assembly. The assembly integration process 
involved sequence alignemnts as well as cDNA, marker and BAC/Fosmid end sequence placements. The 
zebrafish mitochondrial sequence is also available in the Genome Browser as the virtual chromosome, 
chrM. For more details about the Zv9 assembly, see the Sanger Institute page for the 
<a href="http://www.sanger.ac.uk/Projects/D_rerio/Zv9_assembly_information.shtml" 
target="_blank"><em>Danio rerio</em> Sequencing Project</a>.</p> 
<p> 
The danRer7 sequence and annotation data can be downloaded from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/danRer7/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#zebrafish">downloads</a> page. Please review 
the Sanger Institute <a href="http://www.sanger.ac.uk/legal/" target="_blank">guidelines</a> for 
using these data.</p>
<p> 
We'd like to thank the Wellcome Trust Sanger Institute for providing this assembly. The UCSC 
zebrafish Genome Browser was produced by Hiram Clawson, Greg Roe, Mary Goldman, Brian Raney and 
Donna Karolchik. See the <a href="credits.html#zebrafish_credits">credits</a> page for a detailed 
list of the organizations and individuals who contributed to this release.</p>

<a name="011011"></a>
<h2>Jan. 10, 2011  &nbsp;&nbsp; Introducing Genome Browser right-click navigation</h2> 
<p> 
Several of the common display and navigation operations offered on the Genome Browser tracks page 
may now be quickly accessed by right-clicking on a feature on the tracks image and selecting an 
option from the displayed menu. Depending on context, the right-click feature will let you change 
the track display mode, zoom in or out to the exact position coordinates of the feature, open the 
&quot;Get DNA&quot; window at the feature's coordinates, display details about the feature, open a 
popup window to configure the track's display, or display the entire tracks image in a separate 
window for inclusion in spreadsheets or other documents. (Note that the &quot;PDF/PS&quot; option in
the Genome Browser top navigation bar can also be used to generate a high-quality annotation tracks 
image suitable for printing.)</p>
<p> 
<img src="../images/rightclick.png" alt="Right-click screenshot" width="576" height="306"></p>
<p> 
To use the right-click feature, make sure the &quot;Enable advanced javascript features&quot; 
option on the 
<a hreF="../cgi-bin/hgTracks?hgTracksConfigPage=configure+tracks+and+display">tracks configuration 
page</a> is checked, and configure your internet browser to allow the display of popup windows from 
<em>genome.ucsc.edu</em>. When enabled, the right-click navigation feature replaces the default 
contextual menu typically displayed by your internet browser when you right-click on the tracks 
image. A few combinations of the Mozilla Firefox browser on Mac OS do not support the right-click 
menu functionality using secondary click; in these instances, you must use the explicit ctrl-click 
action to display the menu.</p> 
<p> 
Credit goes to Larry Meyer and Brooke Rhead for doing the lion's share of the design, development 
and testing of this feature, with engineering support from Tim Dreszer and additional testing by 
several others on the QA team.</p>

<!-- ============= 2010 archived news ============= -->

<a name="2010"></a>
<a name="120210"></a>
<h2>Dec. 2, 2010  &nbsp;&nbsp; Searching for tracks just got easier</h2> 
<p> 
The number of genome assemblies and annotation tracks in the UCSC Genome Browser has increased 
exponentially over the past few years. Along with this growth, it has become more difficult for 
browser users to sort through the many tracks to find the data they're interested in.</p> 
<p> 
With our latest Genome Browser release, we've introduced a new track search feature that will help 
you to quickly and easily find and display specific annotation data sets. The search feature is 
available via the &quot;track search&quot; button on the browser gateway and tracks display 
pages.</p> 
<p> 
The default search takes one or more terms as input, and returns a list of all the browser tracks in
which the name, description, group or associated metadata contains the terms. The advanced search 
option lets the user fine-tune the search based on the track name, description, and group. 
Individuals interested in the ENCODE data available on selected human genome assemblies can use the 
advanced search to specify particular ENCODE metadata terms, such as cell line, experiment type, 
etc.</p>
<p>
<img src="/images/tracksearch.gif" alt="Track search screenshot" width="648" height="305"></p>
<p> 
Once the list of search results is shown, you may choose which tracks you'd like to view in the 
browser, and can adjust the display modes. The search results list is sortable by multiple criteria,
and provides direct links to detailed track descriptions and additional information about the 
tracks.</p> 
<p> 
Special thanks to Tim Dreszer and Larry Meyer for implementing track search, and to the QA team for 
their work in testing and finetuning it. We hope our users, particularly those exploring ENCODE 
data, will take advantage of this new feature. As always, we appreciate your feedback. Please send 
comments or questions to 
<a
href="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</a>.</p> 
<!-- above address is genome at soe.ucsc.edu -->

<a name="082410"></a>
<h2>Aug. 24, 2010  &nbsp;&nbsp; New drag and reorder functionality released</h2> 
<p> 
We are pleased to announce new functionality in the Genome Browser. It is now possible to rearrange 
the order that tracks appear in the browser image directly from the browser image itself. To reorder
tracks, click-and-hold the side label or gray mini-button of a single track and drag the highlighted
track to a new position within the image.</p> 
<p> 
Because reordering is now available in the browser image itself, we have removed the &quot;enable 
track reordering&quot; option from the configuration page.</p> 
<p> 
Thanks to Tim Dreszer and the rest of the team for implementing and testing this feature.</p>

<a name="082010"></a>
<h2>Aug. 20, 2010  &nbsp;&nbsp; New ENCODE Integrated Regulation super-track released</h2> 
<p> 
We are pleased to announce the release of the ENCODE Integrated Regulation super-track, a collection
of regulatory tracks containing state-of-the-art information about the mechanisms that turn genes on
and off at the transcription level. Individual tracks within the set show enrichment of histone 
modifications suggestive of enhancer and promoter activity, DNAse clusters indicating open 
chromatin, regions of transcription factor binding, and transcription levels. When viewed in 
combination, the complementary nature of the data within these tracks has the potential to greatly 
facilitate our understanding of regulatory DNA.</p> 
<p> 
The data comprising these tracks were generated from hundreds of experiments on multiple cell lines 
conducted by labs participating in the Encyclopedia of DNA Elements (ENCODE) project, and were 
submitted to the UCSC ENCODE Data Coordination Center for display on the Genome Browser.</p> 
<p> 
Faced with the problem of how to display such a large amount of data in a manner facilitating 
analysis, UCSC has developed new visualization methods that cluster and overlay the data, and then 
display the resulting tracks on a single screen. Each of the cell lines in a track is associated 
with a particular color. Light, saturated colors are used to produce the best transparent 
overlay.</p> 
<p> 
<a href="http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&hgS_otherUserName=Example1&hgS_otherUserSessionName=hg18EncReg"><img src="/images/encodeRegShot.gif" 
alt="ENCODE Regulatory track screenshot" width="622" height="167"></a></p> 
<p> 
The data in the ENCODE Regulation super-track, as with all data from the production phase of the 
ENCODE project, have genome-wide coverage. In general, Genome Browser tracks that show 
ENCODE-generated data can be identified by the double-helix icon preceding the name in the track 
list. Currently, the ENCODE Regulation data are available only on the Mar. 2006 (NCBI Build 36, UCSC
version hg18) assembly of the human genome.</p> 
<p> 
For a detailed description of the datasets contained in this super-track and a discussion of how the
tracks can be used synergistically to examine regions of regulatory functionality within the genome,
see the <a href="../cgi-bin/hgTrackUi?db=hg18&c=chr1&g=wgEncodeReg">track description</a> page.</p>

<a name="081810"></a>
<h2>Aug. 18, 2010  &nbsp;&nbsp; Cat Genome Browser available</h2> 
<p> 
We have released a Genome Browser for the latest assembly of Cat (<em>Felis catus</em>). The GTB 
V17E assembly (Dec. 2008, UCSC version felCat4) was produced by the NHGRI Genome Technology Branch 
and NIH Intramural Sequencing Center, Maryland; Agencourt Bioscience Corporation, Massachusetts; 
NCI Laboratory of Genomic Diversity, Maryland; and Hill's Pet Nutrition, Inc, Kansas. The 
mitochondrial genome was sequenced by NCI Laboratory of Genomic Diversity. For detailed sequencing 
and assembly information, see the NCBI Felis catus  
<a href="https://www.ncbi.nlm.nih.gov/sites/entrez?Db=genomeprj&cmd=ShowDetailView&TermToSearch=10703" 
target="_blank">assembly page</a>.</p> 
<p> 
Downloads of the Cat data and annotations may be obtained from the 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/felCat4/">UCSC FTP site</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#cat">Downloads</a> page. These data have been
freely provided for use in the UCSC Genome Browser with specific conditions for use.</p> 
<p> 
Many thanks to NHGRI, Agencourt Bioscience Corporation, NCI and Hill's Pet Nutrition for providing 
these data. The felCat4 annotation tracks were generated by UCSC and its collaborators. We'd also 
like to acknowledge the UCSC Cat browser team: Chin Li, Antonio Coelho, Mary Goldman and Donna 
Karolchik. See the <a href="credits.html#cat_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>

<a name="072310"></a>
<h2>Jul. 23, 2010  &nbsp;&nbsp; BigBed/BigWig paper published</h2> 
<p>
Last summer, we announced the release of new data formats for very large data sets: BigBed and 
BigWig. Read more about these data formats 
<a href="https://lists.soe.ucsc.edu/pipermail/genome-announce/2009-July/000165.html" 
target="_blank">here</a>. As a follow up, we are pleased to announce that we have published a paper 
on these new file formats:  Kent WJ, Zweig AS, Barber G, Hinrichs AS, Karolchik D. 
<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btq351" 
target="_blank">BigWig and BigBed: enabling browsing of large distributed data sets.</a> 
<em>Bioinformatics</em>. 2010 July 17. Published online in advance of print.</p> 
<p>
From the abstract: &quot;BigWig and BigBed files are compressed binary indexed files containing data
at several resolutions that allow the high-performance display of next-generation sequencing 
experiment results in the UCSC Genome Browser. The visualization is implemented using a 
multi-layered software approach that takes advantage of specific capabilities of web-based protocols
and Linux and UNIX operating systems files, R trees, and various indexing and compression tricks. 
As a result, only the data needed to support the current browser view is transmitted rather than the
entire file, enabling fast remote access to large distributed data sets.&quot;</p> 
<p>
The supplemental materials (available online) contain the technical details about the 
implementation.</p>

<a name="072110"></a>
<h2>Jul. 21, 2010  &nbsp;&nbsp; Free Genome Browser workshops available</h2> 
<p>
In the years since its introduction, the UCSC Genome Browser has become an important tool for 
research and teaching. As it has grown in capacity and therefore complexity, however, even 
experienced users sometimes have difficulty leaning how to make the most of it.</p> 
<p>
We plan to provide a limited number free browser training workshops at selected institutions during 
the coming year. We invite you to apply. Please indicate your interest by completing our survey: 
<a href="http://www.surveymonkey.com/s/XJF93F5" target="_blank">Apply for free workshop</a>.</p>

<a name="071510"></a>
<h2>Jul. 15, 2010  &nbsp;&nbsp; Conservation track available for zebrafish (danRer6)</h2> 
<p> 
We are pleased to announce the release of a new Conservation track based on the zebrafish (danRer6) 
assembly. This track shows multiple alignments of 6 vertebrate species and measurements of 
evolutionary conservation using phastCons from the PHAST package. The multiple alignments were 
generated using multiz and other tools in the UCSC/Penn State Bioinformatics comparative genomics 
alignment pipeline. Conserved elements identified by phastCons are displayed in the companion 
&quot;Most Conserved&quot; track.</p>
<p> 
For more details, please visit the <a href="../cgi-bin/hgTrackUi?db=danRer6&g=multiz6way">track 
description page</a>.</p>

<a name="070710"></a>
<h2>Jul. 7, 2010  &nbsp;&nbsp; Happy 10th birthday, Human Genome</h2> 
<p>
<table align=right>
  <tr>
    <td align=right width = 519 height=344><img src = "../../images/10yrGraph4.gif" 
    alt="10-year usage graph" width="504" height="329"</td>
  </tr> 
  <tr>
    <td align=right><font size=-1>Top graph: total traffic on the UCSC domain during June-July, 
    2000. Bottom graph: page hit statistics on genome.ucsc.edu in the ensuing years since the Genome
    Browser was released.<br></font></td>
  </tr>
</table>
<p> 
UCSC is pleased to celebrate the 10-year anniversary of the debut of the first assembled human 
genome sequence and its then-fledgling visualization tool, the UCSC Genome Browser. Released on 
July 7, 2000, the genome sequence instantly created unprecedented web traffic on the ucsc.edu domain
as researchers around the world scrambled to download the data: 0.5 terabytes per day, a
record that stood for many years.</p>  
<p> 
David Haussler recounts that day: &quot;Seeing the waterfall of As, Gs, Cs, and Ts pouring off our 
server was an emotional moment. We were witnessing the product of more than three billion years of 
evolution, sequences passed down from the beginning of life to present-day humans.&quot; 
<sup>1</sup></p>
<p> 
The UCSC Genome Bioinformatics group is proud of the role we have played in bringing the human 
genome, as well as the genomes of many other species, to the world. We'd like to thank NHGRI, HHMI,
and our other funding agencies for making this service possible, and also our many users and 
collaborators for their suggestions, contributions, and support throughout the years. We look 
forward to serving as a reliable source of bioinformatics tools and genomic annotations in the years
to come while facing the exciting challenges presented by rapidly advancing technology.</p> 
<p> 
To read more about UCSC's role in the race to assemble the first working draft of the human genome, 
including Jim Kent's remarkable programming feat that pulled it all together in the final weeks, see
the <a href="http://www.cbse.ucsc.edu/research/hgp_race" target="_blank">Center for Biomolecular 
Science and Engineering</a> (CBSE) website.</p>
<p> 
<font size=-1>(1) Sansom C, Morrison McKay BJ (2008) ISCB Honors David Haussler and Aviv Regev. 
PLoS Comput Biol 4(7): e1000101.</font></p>

<a name="060410"></a>
<h2>Jun. 4, 2010  &nbsp;&nbsp; dbSNP 131 available for hg19</h2> 
<p> 
We are pleased to announce the release of dbSNP build 131, available from on the human assembly 
(GRCh37/hg19). The dbSNP build 131 is available at 
<a href="https://www.ncbi.nlm.nih.gov/projects/SNP" target="_blank">NCBI</a>. The SNPs (131) track 
contains 26,033,053 mappings of 23,445,889 reference SNPs that have been mapped to the reference 
genome by dbSNP. This is a significant increase from the provisional hg19 mappings of build 130 
(18,404,149 mappings of 17,512,501 SNPs).</p>
<p> 
You will find the SNPs (131) track on the Human Feb. 2009 (GRCh37/hg19) browser in the Variation and
Repeats group.</p> 
<p> 
Many thanks to dbSNP at NCBI for the data. The track was produced at UCSC by Angie Hinrichs and Mary
Goldman.</p>

<a name="060310"></a>
<h2>Jun. 3, 2010  &nbsp;&nbsp; Updated marmoset browser available</h2> 
<p> 
We have released an updated marmoset (<em>Callithrix jacchus</em>) assembly, UCSC version calJac3, 
corresponding to the Mar. 2009 v3.2 assembly from the Genome Center at Washington University St. 
Louis School of Medicine (WUSTL).</p>  
<p> 
This draft of the marmoset genome has approximately 6X coverage. For statistics and details on the 
assembly process, see the WUSTL Callithrix jacchus 
<a href="http://genome.wustl.edu/genomes/detail/callithrix-jacchus/" target="_blank">assembly 
information</a> page.</p>  
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/calJac3/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#marmoset">Downloads</a> page. See the WUSTL 
Genome Center's <a href="http://genome.wustl.edu/data/data_use_policy" target="_blank">data use 
policy</a> for conditions of use.</p> 
<p> 
Many thanks to the Genome Center at WUSTL School of Medicine for providing the assembly data. The 
calJac3 annotation tracks were generated by UCSC and collaborators. See the 
<a href="credits.html#marmoset_credits">Credits</a> page for a detailed list of the organizations 
and individuals who contributed to this release.</p> 

<a name="051210"></a>
<h2>May 12, 2010  &nbsp;&nbsp; Pig Genome Browser released</h2> 
<p> 
We have released a Genome Browser for the pig, Sus scrofa.  This assembly (SGSC Sscrofa9.2 
(<a href="https://www.ncbi.nlm.nih.gov/sites/entrez?Db=genomeprj&cmd=ShowDetailView&TermToSearch=10718" 
target="_blank">NCBI project 10718</a>, 
<a href="ftp://ftp.ncbi.nlm.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Sus_scrofa/Sscrofa9.2/" 
target="_blank">GCA_000003025.2</a>)) was produced by the <a href="http://piggenome.org/" 
target="_blank">Swine Genome Sequencing Consortium</a> and the 
<a href="http://www.sanger.ac.uk/resources/" targetargetnk>Wellcome Trust Sanger Institute</a>. 
The pig is a member of the artiodactyls (cloven-hoofed mammals), which is an evolutionary clade 
distinct from the primates and rodents. It is an important model for human health, particularly for 
understanding complex traits such as obesity and cardiovascular disease. There is extensive 
conserved homology with the human genome.</p>
<p> 
The pig genome assembly is a hybrid of BAC end-sequencing to anchor contigs combined with 
whole-genome shotgun (WGS) libraries. There are 19 chromosomes: chr1 - chr18, chrX and the 
mitochondrial sequence chrM (GenBank accession number NC_012095) for a total sequence length of 
2,262,501,571 bases. There are 31,203,023 &quot;N&quot; bases in gaps, leaving 2,231,298,548 ACGT 
bases. For more information, see the Sanger Institute's 
<a href="http://www.sanger.ac.uk/Projects/S_scrofa/" target="_blank">Porcine Genome Sequencing 
Project</a> web page.</p>
<p> 
Many thanks to the Swine Genome Sequencing Consortium and the Wellcome Trust Sanger Institute for 
the pig assembly data.  Bulk downloads of the sequence and annotation data are available via the 
Genome Browser <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/susScr2/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#pig">Downloads</a> page. These data have 
<a href="credits.html#pig_use">specific conditions for use</a>. The pig browser annotation tracks 
were generated by UCSC and collaborators worldwide. The UCSC Pig browser was produced by: Hiram 
Clawson, Brian Raney, Mark Diekhans, and Mary Goldman. See the 
<a href="credits.html#pig_credits">Credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p>

<a name="050610"></a>
<h2>May 6, 2010  &nbsp;&nbsp; UCSC hosts portal for Neandertal sequence, alignments, and
analyses</h2> 
<p> 
In conjunction with the publication of the paper Green <em>et al.</em> 
<a href="http://www.sciencemag.org/cgi/content/full/328/5979/710" target="_blank">A Draft Sequence 
of the Neandertal Genome</a> in the May 7 issue of <em>Science</em>, the UCSC Genome Browser project
has released a public <a href="../Neandertal">Neandertal portal</a> that may be used to access the 
Neandertal sequence, alignments to the UCSC hg18 (NCBI Build 36) human reference assembly and the 
UCSC panTro2 (Chimpanzee Sequencing and Analysis Consortium v2.1) chimpanzee reference assembly, and
several associated analyses.</p>  
<p> 
Neandertals are the closest extinct relatives of humans. They lived in much of Europe and western 
Asia before disappearing from the fossil records approximately 30,000 years ago. The Neandertal 
genome sequence consists of short sequence fragments, usually about 50 base pairs long, mapped to 
the human reference genome. The sequence was derived primarily from DNA extracted from three 
Neandertal bones, each about 40,000 years old, found in the Vindija Cave in Croatia; smaller amounts
of data were also obtained from three bones from other sites. The bulk sequencing was carried out on
the Illumina GAII platform.  Neandertal DNA was differentiated from the background of microbial 
sequences in the bone by similarity to the human or chimpanzee genomes.</p> 
<p> 
The draft sequence of the Neandertal genome yields important new insights into the evolution of 
modern humans. Among the findings discussed in the <em>Science</em> publication is evidence that 
some early modern humans, after their migration out of Africa, interbred with Neandertals, resulting
in traces of Neandertal DNA sequences in the genomes of present-day non-Africans. The authors also 
cataloged genetic features unique to modern humans by comparing the Neandertal, human, and 
chimpanzee genomes. Among the genes highlighted in the study as likely to have undergone important 
changes in recent human evolution are those involved in cognitive development, skull structure, 
energy metabolism, skin morphology and physiology.</p>  
<p> 
We'd like to thank Richard Green (formerly of the Max-Planck Institute, and now an assistant 
professor of biomedical engineering at UCSC) and the many authors of the <em>Science</em> paper for 
providing the UCSC Genome Browser with these data. We'd also like to acknowledge the hard work of 
the UCSC Genome Browser staff who pulled together the annotations and information for the UCSC 
Neandertal portal: Angie Hinrichs, Katrina Learned, and Pauline Fujita.</p>

<a name="042610"></a>
<h2>Apr. 26, 2010  &nbsp;&nbsp; European Rabbit Genome Browser released</h2> 
<p> 
We have released a Genome Browser for the European rabbit, <em>Oryctolagus cuniculus</em>. This 
assembly (UCSC version oryCun2, Broad oryCun2) was produced by the 
<a href="http://www.broadinstitute.org/" target="_blank">Broad Institute of MIT and Harvard</a>, 
Cambridge, MA. The European rabbit is the precursor of all domestic rabbits. It is a valuable model 
in both immunology and evolutionary biology.</p> 
<p> 
The genome has been sequenced to nearly 7.5X coverage. The sequence is 2.7 Gb in length and includes
21 autosomes, the X chromosome, and the mitochondrial DNA (chrM), as well as 3,219 unplaced 
segments. For more information on the assembly, see the 
<a href="http://www.broadinstitute.org/science/projects/mammals-models/rabbit/rabbit-genome-sequencing-project" 
target="_blank">Rabbit Genome Project</a> page.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/oryCun2/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#rabbit">Downloads</a> page. These data have 
<a href="credits.html#rabbit_use">specific conditions for use</a>.</p>
<p> 
Many thanks to the Broad Institute for the rabbit assembly data. The oryCun2 annotation tracks were 
generated by UCSC and collaborators. We'd also like to acknowledge the UCSC Rabbit browser team: 
Hiram Clawson and  Antonio Coelho. See the <a href="credits.html#rabbit_credits">Credits</a> page 
for a detailed list of the organizations and individuals who contributed to this release.</p> 

<a name="041610"></a>
<h2>Apr. 16, 2010  &nbsp;&nbsp; Giant Panda Genome Browser released</h2> 
<p> 
We have released a Genome Browser for the giant panda, <em>Ailuropoda melanoleuca</em>. This 
assembly (UCSC version ailMel1, BGI AilMel1) was produced by the 
<a href="http://www.genomics.cn/en/index" target="_blank">Beijing Genomics Institute at 
Shenzhen</a>, Shenzhen, China. The panda is a member of Ursidae, despite earlier taxonomy that 
marked it as a relative of the raccoon. It has become the most recognized symbol for the fight for 
animal conservation.</p> 
<p> 
This draft of the panda genome has a size of approximately 2.3 Gb at 94% coverage. The assembly 
comprises 81,467 scaffolds and chrM (mitochondrial DNA). For more information on the assembly, see 
the BGI <a href="http://panda.genomics.org.cn/page/panda/index.jsp" target="_blank">Giant Panda 
Database</a> page.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/ailMel1/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#panda">Downloads</a> page. These data have 
<a href="credits.html#panda_use">specific conditions for use</a>.</p> 
<p> 
Many thanks to the Beijing Genomics Institute at Shenzhen for the panda assembly data. The ailMel1 
annotation tracks were generated by UCSC and collaborators. We'd also like to acknowledge the UCSC 
Panda browser team: Hiram Clawson, Ann Zweig, and  Antonio Coelho. See the 
<a href="credits.html#panda_credits">Credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p>

<a name="032410"></a>
<h2>Mar. 24, 2010  &nbsp;&nbsp; African Savannah Elephant Genome Browser released</h2> 
<p> 
We have released a Genome Browser for the African savannah elephant, <em>Loxodonta africana</em>. 
This assembly (UCSC version loxAfr3, Broad loxAfr3) was produced by the 
<a href="http://www.broadinstitute.org/" target="_blank">Broad Institute</a>, Cambridge, MA, USA. The 
elephant was the first member of Afrotheria to be sequenced. Afrotheria is the deepest node of 
Eutheria, and the elephant sequence should be useful in reconstructing the ancestral eutherian 
genome.</p> 
<p> 
This draft of the elephant genome has a size of approximately 3 Gb with 7X coverage. The assembly 
comprises 2352 scaffolds and chrM (mitochondrial DNA). For more information on the assembly, see the
Broad Institute 
<a href="http://www.broadinstitute.org/science/projects/mammals-models/elephant/elephant" 
target="_blank">Elephant Genome Project</a> page.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/loxAfr3/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#elephant">Downloads</a> page. These data have
<a href="credits.html#elephant_use">specific conditions for use</a>.</p>
<p> 
Many thanks to the Broad Institute for the elephant assembly data. The loxAfr3 annotation tracks 
were generated by UCSC and collaborators. We'd also like to acknowledge the UCSC Elephant browser 
team: Hiram Clawson, Pauline Fujita, Vanessa Swing,  Antonio Coelho, and Donna Karolchik. See the 
<a href="credits.html#elephant_credits">Credits</a> page for a detailed list of the organizations 
and individuals who contributed to this release.</p>					

<a name="031910"></a>
<h2>Mar. 19, 2010  &nbsp;&nbsp; California Sea Hare Genome Browser released</h2> 
<p> 
We have released a Genome Browser for the California sea hare, <em>Aplysia californica</em>. This 
assembly (UCSC version aplCal1, Broad version Aplcal2.0) was produced by the 
<a href="http://www.broadinstitute.org/" target="_blank">Broad Institute</a>, Cambridge, MA, USA. The 
sea hare is the first mollusc to be sequenced, and is particularly noted for its nervous system, 
which is quite useful for neurobiological experimentation.</p> 
<p> 
This draft of the sea hare genome has a size of approximately 712 Mb with nearly 10X coverage. The 
assembly comprises 8766 scaffolds and chrM (mitochondrial DNA). For more information on the 
assembly, see the Broad Institute 
<a href="http://www.broadinstitute.org/science/projects/mammals-models/vertebrates-invertebrates/aplysia/aplysia-genome-sequencing-project" 
target="_blank">Aplysia Genome Project</a> page.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/aplCal1/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#seahare">Downloads</a> page. These data have 
<a href="credits.html#seahare_use">specific conditions for use</a>.</p> 
<p> 
Many thanks to the Broad Institute for the sea hare assembly data. The aplcal1 annotation tracks 
were generated by UCSC and collaborators. We'd also like to acknowledge the UCSC Sea Hare browser 
team: Galt Barber, Brian Raney, Hiram Clawson, Katrina Learned, Antonio Coelho, and Donna Karolchik.
See the <a href="credits.html#seahare_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>

<a name="031210"></a>
<h2>Mar. 12, 2010  &nbsp;&nbsp; Updated Tetraodon Genome Browser available</h2> 
<p> 
We have released a Genome Browser for the latest assembly of <em>Tetraodon nigroviridis</em>. The 
V8 assembly (March 2007, UCSC version tetNig2) was provided by 
<a href="http://www.genoscope.cns.fr/" target="_blank">Genoscope</a>, Evry, France, in collaboration
with the <a href="http://www.broad.mit.edu/annotation/tetraodon/background.html" 
target="_blank">Broad Institute</a>, Cambridge, MA, USA. For detailed sequencing and assembly 
information, see the Genoscope <em>Tetraodon nigroviridis</em> 
<a href="http://www.genoscope.cns.fr/spip/Tetraodon-nigroviridis-whole.html" 
target="_blank">assembly</a> page.</p> 
<p> 
Downloads of the Tetraodon data and annotations may be obtained from the UCSC 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/tetNig2/" target="_blank">FTP site</a> or 
<a href=http://hgdownload.gi.ucsc.edu/downloads.html#tetraodon>Downloads</a> page. These data have 
been freely provided by Genoscope for use in the UCSC Genome Browser with 
<a href="credits.html#tetraodon_use" target="_blank">specific conditions for use</a>. For more 
information about the <em>Tetraodon nigroviridis</em> project, see the Genoscope 
<a href="http://www.genoscope.cns.fr/spip/-Tetraodon-nigroviridis,387-.html" 
target="_blank">website</a>.</p> 
<p> 
Many thanks to Genoscope and the Broad Institute for providing these data. The tetNig2 annotation 
tracks were generated by UCSC and its collaborators. We'd also like to acknowledge the UCSC 
Tetraodon browser team, which includes several new faces: Hiram Clawson, Mark Diekhans, Brian Raney,
Ann Zweig, Mary Goldman, Vanessa Swing, Antonio Coelho, Robert Kuhn, and Donna Karolchik. See the 
<a href="credits.html#tetraodon_credits">credits</a> page for a detailed list of the organizations 
and individuals who contributed to this release.</p>

<a name="022410"></a>
<h2>Feb. 24, 2010  &nbsp;&nbsp; New gene name search functionality released</h2> 
<p> 
UCSC has introduced a new gene search functionality in the Genome Browser, available through a 
&quot;gene&quot; text box on the gateway and annotation tracks pages. If you are searching for a 
particular gene, but you don't remember the exact coordinates, simply type the gene name into the 
box. The software will suggest gene names as you type; you can either choose a gene from the list, 
or finish typing your full gene name and then press &quot;jump&quot; or press enter. This 
functionality is available now&mdash;there's nothing to configure. Start using it 
<a href="../cgi-bin/hgGateway">here</a>.</p>
<p> 
<img src="../images/autoselect.jpg" alt="Autoselect menu" width="800" height="147"></p> 
<p> 
Thanks to Larry Meyer and the rest of the team for implementing and testing this feature.</p>

<a name="012510"></a>
<h2>Jan. 25, 2010  &nbsp;&nbsp; New custom track type: Binary Alignment/Map (BAM)</h2> 
<p> 
We are pleased to announce the availability of another new track type for Custom Tracks: the Binary 
Alignment/Map (<a href="help/bam.html">BAM</a>).</p> 
<p> 
BAM is the compressed binary version of the <a href="http://samtools.sourceforge.net/" 
target="_blank">Sequence Alignment/Map</a> (SAM) format, a compact and index-able representation of 
nucleotide sequence alignments. Many next-generation sequencing and analysis tools work with 
SAM/BAM. For custom track display, the main advantage of indexed BAM over PSL and other 
human-readable alignment formats is that only the portions of the files needed to display a 
particular region are transferred to UCSC. This makes it possible to display alignments from files 
that are so large that the connection to UCSC would time out when attempting to upload the whole 
file to UCSC. Both the i BAM file and its associated index file remain on your web-accessible server
(http or ftp), not on the UCSC server. UCSC temporarily caches the accessed portions of the files 
to speed up interactive display.</p>

<a name="011310"></a>
<h2>Jan. 13, 2010  &nbsp;&nbsp; Lifespan of custom tracks within sessions</h2> 
<p> 
Due to the popularity of UCSC <a href="help/hgTracksHelp.html#CustomTracks">custom tracks</a> and 
<a href="help/hgSessionHelp.html">sessions</a>, we are running out of disk space for storing custom 
tracks accessed within sessions. Although sessions themselves are saved for one year, user-generated
tracks within sessions (including custom tracks) are saved for 48 hours, the typical period for 
other custom tracks. Until now, because we have had space available on the server, we have 
attempted to keep custom tracks alive for the lifetime of the session.</p> 
<p> 
While we provide short-term custom track storage as a service to users who want to view their own 
data in the UCSC Genome Browser, we unfortunately do not have the resources to guarantee long-term 
data storage. Over the next few days, we will begin to actively delete custom tracks within sessions
that have not been accessed within the past four months. If you actively use your custom tracks 
within saved sessions, they will not be deleted.</p> 
<p> 
Instead of uploading your data to our servers, please consider using alternate data types such as 
<a href="help/bigBed.html">bigBed</a>, <a href="help/bigWig.html">bigWig</a>, or other URL-based 
tracks that enable you to store your data locally.</p>

<a name="011110"></a>
<h2>Jan. 11, 2010  &nbsp;&nbsp; Zebrafish release zv8 now available in Genome Browser</h2> 
<p> 
The most recent zebrafish assembly -- zv8 (UCSC version danRer6, Dec. 2008) -- is now available in 
the UCSC Genome Browser. This assembly was produced by the Wellcome Trust Sanger Institute, UK.</p> 
<p> 
The danRer6 assembly consists of nearly 1.5 billion bp in 11,623 scaffolds with a coverage of 
6.5-7x. It includes sequence anchored to chromosomes 1-25 and chrM (mitochondrial), as well as 
11,418 unplaced scaffolds. Significant problems existing in previous assemblies have been rectified 
in this assembly through the use of a whole-genome shotgun assembly with more coverage and a 
reorganization of the fingerprint contig order and orientation through more careful use of maps. 
For more information about this assembly, see the Sanger Institute web page for the 
<a href="http://www.sanger.ac.uk/Projects/D_rerio/Zv8_assembly_information.shtml" 
target="_blank"><em>Danio rerio</em> Sequencing Project</a> and the track description page for the 
danRer6 Assembly track.</p> 
<p> 
The danRer6 sequence and annotation data can be downloaded from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/danRer6/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#zebrafish">downloads</a> page. Please review 
the <a href="http://www.sanger.ac.uk/notices/use-policy.shtml" target="_blank">guidelines</a> for 
using these data.</p> 
<p>
We'd like to thank the Wellcome Trust Sanger Institute for providing this assembly. The UCSC 
zebrafish Genome Browser was produced by Galt Barber, Robert Kuhn, Katrina Learned, and Donna 
Karolchik. The initial set of annotation tracks was generated by the UCSC Genome Bioinformatics 
Group. See the <a href="credits.html#zebrafish_credits">credits</a> page for a detailed list of the 
organizations and individuals who contributed to the release of this browser.</p>

<!-- ============= 2009 archived news ============= -->
<a name="2009"></a>

<a name="121409"></a>
<h2>Dec. 14, 2009  &nbsp;&nbsp; New job posting: Biological Data Technician</h2> 
<p> 
The UCSC Genome Browser project is looking for a bioinformatician, biologist, or software engineer 
with a strong biology background to collect and import data  into the UCSC Genome Browser database 
and website. This person will work closely with external research laboratories to capture their 
experimental results and methods and with internal software developers and database testing staff to
make the data accessible to the worldwide scientific community.</p> 
<p> 
Candidates must have a bachelor's degree in bioinformatics or a biological science (or equivalent 
experience), be proficient in UNIX/Linux command-line use, competent in UNIX shell scripting and 
Perl programming, and familiar with relational database concepts and SQL. Besides having the ability
to quickly learn and interpret biological and technical information, the ideal candidate is an 
effective communicator, resourceful, and a diplomatic team player who is both quality-oriented and 
able to work effectively under deadline.</p> 
<p> 
To find more information and application instructions for this job as well as other open positions 
with the UCSC Genome Browser project and the UCSC Center for Biomolecular Science and Engineering, 
see the CBSE <a href="http://www.cbse.ucsc.edu/jobs/staff" target="_blank">staff positions</a> web 
page.</p>

<a name="120709"></a>
<h2>Dec. 7, 2009  &nbsp;&nbsp; Human Genome Browser default changing to hg19</h2> 
<p> 
In conjunction with the release of the UCSC Genes and Conservation tracks on the hg19 (GRCh37) 
human assembly, we have changed the default human browser on our website from hg18 to hg19. In 
general, if you are currently using the hg18 (or older) browser, the Genome Browser will continue 
to display that assembly for you when you start it up. However, there are circumstances in which the
assembly may switch to the newer version (for instance, if you reset your browser defaults). If you 
find yourself in a situation where some of your favorite browser tracks have 
&quot;disappeared&quot;, you may want to check that you're viewing the right assembly.</p> 

<a name="120109"></a>
<h2>Dec. 1, 2009  &nbsp;&nbsp; New UCSC Genes and Conservation tracks released on hg19 browser</h2> 
<p> 
We're happy to announce the release of two of our most popular data sets on the hg19/GRCh37 human 
Genome Browser. The <strong>UCSC Genes</strong> track is a moderately conservative set of gene 
predictions based on data from RefSeq, Genbank, CCDS and UniProt. The <strong>Conservation</strong> 
track shows multiple alignments of 46 vertebrate species and measurements of evolutionary 
conservation using two methods (phastCons and phyloP) from the PHAST package for all vertebrate 
species as well as primate and placental mammal subsets. For more information about these 
annotations, read the description pages that accompany these tracks.</p> 

<a name="102009"></a>
<h2>Oct. 20, 2009  &nbsp;&nbsp; Job opening with UCSC Genome Browser project</h2> 
<p> 
The UCSC Genome Browser project is currently accepting applications for a Biological Database 
Testing/User Support Technician (Programmer/Analyst I). We are looking for a talented self-motivated
individual who would like to use their skills in computer science, biology, and bioinformatics on a 
fast-paced project featuring the work of top genomics scientists worldwide.</p> 
<p>
For a summary of the position details and qualifications, see 
<a href="http://www.cbse.ucsc.edu/jobs/staff#bdtust" target="_blank">Biological Database 
Testing/User Support Technician</a> on the Center for Biomolecular Science and Engineering (CBSE) 
website. For detailed job descriptions and application information, go to the 
<a href="http://jobs.ucsc.edu/" target="_blank">UCSC Staff Employment website</a>, click the 
&quot;Search Postings&quot; link on the sidebar, and type in job #0902365 (Testing/User Support 
Technician).</p> 

<a name="090909"></a>
<h2>Sep. 9, 2009  &nbsp;&nbsp; Changes to the bigBed/bigWig data formats</h2> 
<p>
If you have been taking advantage of the new bigBed format (for very large data sets), you'll be 
happy to hear that we have considerably slimmed down the memory footprint of the program that 
converts BED files into bigBed files: bedToBigBed. Because it now uses a multi-pass approach, it now
takes only 1/4 the amount of RAM as the size of the uncompressed BED input file (instead of the 5x 
RAM it needed previously!). Read more <a href="help/bigBed.html">here</a>. Pick up the new 
bedToBigBed executable 
<a href="http://hgdownload.gi.ucsc.edu/admin/exe/linux.x86_64/">here</a>.</p> 
<p> 
In conjunction with this change, there is also a change to the way you must specify your bigBed or 
bigWig custom track. When you specify the location of your local bigBed/bigWig file (on your 
web-accessible http, https, or ftp server), use the designation &quot;bigDataUrl&quot; instead of 
the old &quot;dataUrl&quot;:</p> 
<pre><code>track type=bigBed name="My Big Bed" description="Some Data from My Lab"
<strong>bigDataUrl</strong>=http://myorg.edu/mylab/myBigBed.bb</code></pre> 
<p> 
Additionally, we would like to announce a companion program to the previously-announced wigToBigWig 
program: bedGraphToBigWig. This program converts bedGraph files into bigWig files. The bedGraph 
format allows display of sparse or varying-size data.  Read more 
<a href="help/bigWig.html">here</a>. You can download the new bedGraphToBigWig utility 
<a href="http://hgdownload.gi.ucsc.edu/admin/exe/linux.x86_64/">here</a>.</p> 
<p> 
The main advantage of the bigBed and bigWig formats is that only the portions of the files needed to
display a particular region are transferred to UCSC, so for large data sets, displaying 
bigBed/bigWig data is considerably faster than regular BED/wig data. The bigBed/bigWig file remains 
on your web accessible server (http, https, or ftp), not on the UCSC server. Consequently, creating 
your custom track is very fast. Only the portion that is needed for the chromosomal position you are
currently viewing is locally cached at UCSC as a &quot;sparse file&quot;.</p>

<a name="081009"></a>
<h2>Aug. 10, 2009  &nbsp;&nbsp; Updated Yeast Genome Browser released</h2> 
<p> 
We have added the June 2008 release of the yeast genome (<em>Saccharomyces cerevisiae</em>) based on
sequence from the <a href="http://www.yeastgenome.org/" target="_blank">Saccharomyces Genome 
Database</a> (SGD). The sequence, open reading frame (ORF), and gene annotations were downloaded 
from the site <a href="ftp://genome-ftp.stanford.edu/pub/yeast/data_download" 
target="_blank">ftp://genome-ftp.stanford.edu/pub/yeast/data_download</a>.</p>
<p> 
The S288C strain was used in this sequencing project. Reference information for each chromosome may 
be found in the SGD <a href="http://www.yeastgenome.org/chromosomes.shtml" 
target="_blank">Systematic Sequencing Table</a>. Bulk downloads of the sequence and annotation data 
are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/sacCer2/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#yeast">Downloads</a> page.</p>
<p> 
See the <a href="credits.html#yeast_credits">Credits</a> page for details on contributions to this 
release.</p>

<a name="072209"></a>
<h2>Jul. 22, 2009  &nbsp;&nbsp; Updated Opossum Genome Browser released</h2> 
<p> 
We have released an updated opossum (<em>Monodelphis domestica</em>) genome browser, UCSC version 
monDom5, corresponding to the <a href="http://www.broad.mit.edu/mammals/opossum/" 
target="_blank">Broad Institute's</a> October 2006 assembly.</p>
<p> 
This draft of the opossum genome has approximately 6.8X coverage. The assembly length is nearly 
3.61 billion bp including gaps (3.50 billion bp without gaps) contained on chromosomes 1-8, X, M 
(mitochondrial DNA), and Un (unmapped clone contigs). The N50 of the scaffolds including gaps is 
60,456 kb; the N50 without gaps is 110 kb.</p>
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/monDom5/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#opossum">Downloads</a> page. These data have 
<a href="../goldenPath/credits.html#opossum_use">specific conditions for use</a>.</p>
<p> 
Many thanks to The Broad Institute for the opossum assembly data. The monDom5 annotation tracks were
generated by UCSC and collaborators worldwide. See the 
<a href="../goldenPath/credits.html#opossum_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>

<a name="070709"></a>
<h2>Jul. 7, 2009  &nbsp;&nbsp; Engineering job opening with UCSC Genome Browser project</h2> 
<p> 
The UCSC Center for Biomolecular Science and Engineering is currently accepting applications for a 
full-time, experienced engineer to develop software and databases for the UCSC Genome Browser and 
related projects. The successful candidate must be independent, articulate, self-motivated, able to 
learn and integrate technical information quickly, and have a strong commitment to quality and best 
engineering practices.</p>
<p> 
For more information about this position, see the CBSE 
<a href="http://www.cbse.ucsc.edu/jobs/staff#sftwr" target="_blank">employment page</a>. To apply 
for this job, go to the <a href="https://jobs.ucsc.edu/" target="_blank">UCSC Staff Employment 
Opportunities page</a>, click the &quot;Search Postings&quot; link, and then enter job #0902109.</p>

<a name="052709"></a>
<h2>May 27, 2009  &nbsp;&nbsp; Australasian Genome Browser mirror available</h2> 
<p> 
We are pleased to announce the availability of a public Genome Browser mirror for the Australasian 
research community: <a href="https://ucsc.genome.edu.au/" 
target="_blank">https://ucsc.genome.edu.au/</a>.</p>
<p> 
The Queensland Facility for Advanced Bioinformatics (QFAB), in conjunction with the University of 
Queensland, Australia, maintains this full mirror of the UCSC Genome Browser, which is sponsored by 
the Australian Research Council. The QFAB mirror offers alternative access to the Genome Browser 
data when the UCSC site is unavailable, and may provide better performance for our Australasian 
users.</p>  
<p> 
We'd like to extend our thanks to QFAB for providing and maintaining this mirror, and in particular 
to Jeremy Barker and Michael Pheasant for their collaborative work with UCSC in setting up this 
site.</p>

<a name="042709"></a>
<h2>Apr. 27, 2009  &nbsp;&nbsp; New Human Genome Browser released</h2> 
<p> 
We are pleased to announce the release of the February 2009 human genome browser, UCSC version 
hg19.</p> 
<p> 
Starting with this assembly, the human genome sequence is now provided by the 
<a href="https://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/" target="_blank">Genome Reference 
Consortium</a>, whose goal is to correct the small number of regions in the reference that are 
currently misrepresented, to close as many remaining gaps as possible and to produce alternative 
assemblies of structurally variant loci when necessary. The hg19 browser corresponds to GRCh37.</p>
<p> 
Statistics for the GRCh37 build assembly can be found on the NCBI 
<a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.13/"
target="_blank">Build 37.1 Statistics</a> web page.</p>  
<p> 
The hg19 browser contains 9 haplotypes. See the <a href="http://www.sanger.ac.uk/HGP/Chr6/MHC/" 
target="_blank">Wellcome Trust Sanger Institute MHC Haplotype Project</a> web site for additional 
information on the chr6 alternate haplotype assemblies.</p> 
<p> 
The UCSC staff responsible for producing the initial hg19 browser include Hiram Clawson, Brooke 
Rhead, Pauline Fujita, Ann Zweig, Katrina Learned, and Robert Kuhn. See the 
<a href="credits.html#human_credits">Credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p>

<a name="032009"></a>
<h2>Mar. 20, 2009  &nbsp;&nbsp; Two research scientist positions open</h2> 
<p> 
The Center for Biomolecular Science &amp; Engineering (CBSE) invites applications from outstanding 
PhD- or MD-level scientists for two distinct positions. Position is open until filled, but 
applications received after March 27 may not be considered.</p>
<p> 
<em>Cancer Genomics Director:</em> We seek an independent, creative scientist with cancer research 
experience to guide the group in building our cancer genomics data analysis platform into an 
essential tool for cancer research and an engine for personalized, genome-based cancer treatment. 
The successful candidate will extend our existing cancer research collaborations, conceptualize new 
research directions, oversee their implementation by our outstanding postdoctoral students and 
scientific programming staff, write papers and present results at scientific meetings, and apply for
additional grants. Our exceptional collegiality and interdisciplinary collaborations make UCSC an 
excellent environment for an innovative scientist who can benefit from and contribute to the rapid 
growth in the sciences and engineering at UCSC. 
<a href="http://www.cbse.ucsc.edu/jobs/staff#cancer">Read more</a>.</p>
<p> 
<em>Medical Genomics Director:</em> We seek an independent, creative scientist who can guide the 
group in the direction of clinical diagnostics, pediatric/prenatal diagnostics, personalized 
medicine, pharmacogenomics, or related fields, building on the exceptional genome analysis resources
at UCSC. The successful candidate will interface with the medical community, conceptualize genomics 
projects, oversee implementation by the scientific programming staff, write papers describing the 
projects, and apply for grants to fund future projects. Our exceptional collegiality and 
interdisciplinary collaborations make UCSC an excellent environment for an innovative scientist who 
can benefit from and contribute to the rapid growth in the sciences and engineering at UCSC. 
<a href="http://www.cbse.ucsc.edu/jobs/staff#medical">Read more</a>.</p>

<a name="031309"></a>
<h2>Mar. 13, 2009  &nbsp;&nbsp; ENCODE DCC data available</h2> 
<p> 
We'd like to invite everyone to explore UCSC's <a href="../ENCODE/">ENCODE Data Coordination 
Center</a> website, as well as the ENCODE data displayed as annotation tracks in the UCSC Genome 
Browser. Today's <a href="../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeYaleChIPseq">featured dataset</a> 
is a track showing transcription factor binding sites determined by ChIP-seq, produced through a 
collaboration of Yale University, University of CA Davis,  and Harvard University. Many more tracks 
from the Feb. 2009 and Dec. 2008 ENCODE data freezes will become available on the UCSC Genome 
Browser in upcoming weeks. To monitor the ENCODE data releases in the Genome Browser, see the UCSC 
ENCODE <a href="../ENCODE/releaseLog.html">data release</a> page.</p> 

<a name="021709"></a>
<h2>Feb. 17, 2009  &nbsp;&nbsp; Navigate with drag-and-zoom</h2> 
<p> 
We would like to announce the release of a new navigation feature, called drag-and-zoom, in the 
Genome Browser. It is now possible to position your mouse over a location in the Base Position 
track, at the top of the browser image and define a zoom range by moving the mouse to a second place
within the image. The browser will redraw at the new coordinates when the mouse button is 
released.</p>

<a name="021209"></a>
<h2>Feb 12,  2009  &nbsp;&nbsp; v200 code release on Darwin's 200th birthday</h2> 
<p>
<img src="../../images/darwin.jpg" alt="Darwin" width="250" height="330"></p>
<p>
As coincidence would have it, we are releasing version 200 of the browser code base this week. 
Though we typically release code  on a Friday, we couldn't resist the impulse to release a day 
early, on the 200th birthday of Charles Darwin, whose ground-breaking work set the stage for modern 
biology.</p>

<a name="020509"></a>
<h2>Feb. 5, 2009  &nbsp;&nbsp; NHGRI solicits genome sequencing input</h2> 
<p> 
The National Human Genome Research Institute (NHGRI) seeks community comments on the future of its 
large-scale genome sequencing program. To stimulate discussion, NHGRI has produced a white paper, 
&quot;The Future of Genome Sequencing&quot;. If you'd like to review the paper or would like more 
information on NHGRI's long-range planning process, please visit their 
<a href="http://www.genome.gov/About/Planning/" target="_blank">planning website</a>.</p>

<a name="013009"></a>
<h2>Jan. 30, 2009  &nbsp;&nbsp; 44-vertebrate Conservation track available</h2> 
<p>	
We are pleased to announce the release of a new Conservation track based on the human (hg18) 
assembly. This track shows multiple alignments of 44 vertebrate species and measurements of 
evolutionary conservation using two methods (phastCons and phyloP) from the PHAST package, for all 
species (vertebrate) and two subsets (primate and placental mammal). The multiple alignments were 
generated using multiz and other tools in the UCSC/Penn State Bioinformatics comparative genomics 
alignment pipeline. Conserved elements identified by phastCons are also displayed in this track. 
For more details, visit the <a href="/cgi-bin/hgTrackUi?db=hg18&g=cons44way">track description 
page</a>.</p> 
<p> 
This track is now the default Conservation track for the hg18 human assembly. It replaces the 
previous 28-vertebrate Conservation track, which is now available as the &quot;28-Way Cons&quot; 
track.</p> 
<p> 
Many people contributed to the creation of this track including, but not limited to:</p> 
<ul> 
  <li>Adam Siepel (Cornell University): phastCons, phyloP, and other programs in the PHAST
  package</li> 
  <li>Webb Miller, Bob Harris and Minmei Hou (Penn State University Bioinformatics Group): blastz 
  and multiz</li> 
  <li>Hiram Clawson, Tim Dreszer, Brian Raney, Kate Rosenbloom and Ann Zweig (UCSC Genome Browser 
  team)</li> 
</ul>

<!-- ============= 2008 archived news ============= -->
<a name="2008"></a>
<a name="120308"></a>
<h2>Dec. 3, 2008 &nbsp;&nbsp; Updated Horse Genome Browser available</h2> 
<p>
The Sep. 2007 EquCab2 release of the horse genome (Equus caballus) is now available in the UCSC 
Genome Browser. This assembly, UCSC version equCab2, was produced by the 
<a href="http://www.broad.mit.edu/" target="_blank">Broad Institute</a>. </p> 
<p> 
The horse draft genome has been sequenced to 6.8X coverage. Approximately 84% of the sequence has 
been anchored to chromosomes, which include autosomes 1-31 and sex chromosome X. Unanchored contigs 
that could not be localized to a chromosome have been concatenated into the virtual chromosome 
&quot;chrUn&quot;, separated by gaps of 1,000 bp. The mitochondrial sequence is also available in 
the Genome Browser as the virtual chromosome &quot;chrM&quot;.
<p>For more details about the current assembly, see the Broad Institute 
<a href="http://www.broad.mit.edu/mammals/horse/" target="_blank">Horse Genome Project</a> page.</p> 
<p>
Bulk downloads of the sequence and annotation data are available from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/equCab2/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#horse">Downloads</a> page. These data have 
<a href="credits.html#horse_use">specific conditions for use</a>.</p> 
<p> 
The UCSC Horse Genome Browser was produced by Larry Meyer, Hiram Clawson, Pauline Fujita, Brooke 
Rhead, and Donna Karolchik. See the <a href="credits.html#horse_credits">Credits</a> page for a 
detailed list of the organizations and individuals who contributed to this release.</p>

<a name="110708"></a>
<h2>Nov. 7, 2008 &nbsp;&nbsp; Medaka Genome Browser updated</h2> 
<p> 
We have updated the initial UCSC Medaka Genome Browser (oryLat1) to correct an error with chrUn in 
which the gap relationships between the contigs within their ultracontigs were incorrect. This error
was introduced by UCSC during the browser assembly process. The corrected replacement browser, 
oryLat2 (Oct. 2005), is based on the same v1.0 assembly produced by the National Institute of 
Genetics (NIG) and the University of Tokyo, Japan. We have removed the original oryLat1 assembly 
from the browser.</p>

<a name="091708"></a>
<h2>Sep. 17, 2008 &nbsp;&nbsp; Seven assemblies archived</h2> 
<p>
We have archived seven older assemblies of the human (hg15), mouse (mm5, mm6), rat (rn2), 
<em>C. elegans</em> (ce1), zebrafish (danRer2), and cow (bosTau1) genomes. You can continue to 
download these assemblies from the Genome Browser 
<a href ="http://hgdownload.gi.ucsc.edu/downloads.html">downloads server</a>. At least two later 
assemblies for each of these organisms remain available on the main Genome Browser site.</p> 
<p> 
We do not provide blat servers or updated GenBank data for archived assemblies. However, it is 
still possible to lift coordinates between these assemblies and those on the main browser site.</p>

<a name="091508"></a>
<h2>Sep. 15, 2008 &nbsp;&nbsp; Request for input on ENCODE data release plan</h2> 
<p> 
The National Human Genome Research Institute (NHGRI) has updated the data release plan for the 
<a href="http://www.genome.gov/ENCODE" target="_blank">ENCODE</a> and model organism ENCODE 
(<a href="http://www.genome.gov/modENCODE" target="_blank">modENCODE</a>) projects. Because the plan
is intended to take into account the needs and responsibilities of the resource users, NHGRI is 
soliciting feedback on the proposal from the research community before the plan is finalized. To 
facilitate this review, UCSC has made the plan available to the community 
<a href="../encodeDCC/ENCODE_Consortia_Data_Release_Policy_2008-09-11.doc">here</a> on our ENCODE 
website.</p> 
<p>
Please send comments directly to 
<a href="mailto:&#69;&#110;&#99;o&#100;e&#64;&#109;&#97;i&#108;.&#110;&#105;&#104;.g&#111;&#118;">&#69;&#110;&#99;o&#100;e&#64;&#109;&#97;i&#108;.&#110;&#105;&#104;.g&#111;&#118;</a>
<!-- above address is Encode at mail.nih.gov -->
by Wednesday, Oct. 15, 2008.</p>

<a name="090308"></a>
<h2>Sep. 3, 2008 - Updated UCSC Genes set released for hg18</h2> 
<p> 
We have released an updated UCSC Genes data set on the latest human assembly (hg18, Mar. 2006). 
The new annotation can be found on the hg18 Genome Browser in the &quot;Genes and Gene
Predictions&quot; track group. The previous set of UCSC Genes on the hg18 assembly has been renamed 
as Old UCSC Genes in the same track group.</p> 
<p> 
The UCSC Genes track shows gene predictions based on data from RefSeq, GenBank, CCDS and UniProt. 
This is a moderately conservative set of predictions, requiring the support of one GenBank RNA 
sequence plus at least one additional line of evidence. The RefSeq RNAs are an exception to this, 
requiring no additional evidence. The track includes both protein-coding and putative non-coding 
transcripts. Some of these non-coding transcripts may actually code for protein, but the evidence 
for the associated protein is weak. Compared to RefSeq, this gene set has generally about 10% more 
protein-coding genes, approximately five times as many putative non-coding genes, and about twice 
as many splice variants.</p> 
<p> 
The latest version of UCSC Genes explicitly incorporates CCDS proteins in addition to the evidence 
included in previous versions, as well as additional data from GenBank, RefSeq, and UniProt. 
Although we have made every effort to preserve UCSC Gene IDs across versions, some gene IDs have 
changed in the new data set. In general, the ID number has been incremented if the new version of a 
gene is larger than the previous version, and a completely new ID has been assigned in those 
instances where a splice site changed or the protein-coding region shrank. However, searching on an 
old discontinued ID from the previous gene set will take you to the corresponding gene in the 
updated track.  Alternatively, you can check the kg3ToKg4 table directly to find corresponding gene 
IDs.</p>
<p> 
The UCSC Genes set was produced by Jim Kent. Thanks to Ann Zweig for heading up the testing of the 
new data set, and to the entire Genome Browser QA team for coordinating its release on our public
site.</p>

<a name="082708"></a>
<h2>Aug. 27, 2008 &nbsp;&nbsp; Zebra Finch Genome Browser released</h2> 
<p> 
We've added the Jul. 2008 release of the zebra finch genome (<em>Taeniopygia guttata</em>) to our 
collection of vertebrate genome browsers. The v3.2.4 draft assembly (UCSC version taeGut1) was 
produced by the Genome Sequencing Center at the Washington University in St. Louis (WUSTL) School 
of Medicine in St. Louis, MO, USA.</p> 
<p> 
The zebra finch genome is the second bird genome to be sequenced (following the chicken, 
<em>Gallus gallus</em>). The DNA used for the shotgun sequencing and the BAC and cosmid libraries 
was derived from a single male zebra finch. The initial assembly was generated using PCAP with 
approximately 6X coverage. About 1.0 Gb of the 1.2-Gb genome has been ordered and oriented along 33 
chromosomes and one linkage group. The chromosome names are based on their homologous chromosomes in
the chicken (<em>Gallus gallus</em>). For more assembly details, as well as a cross reference table 
of zebra finch and chicken chromosome names, see the WUSTL 
<a href="http://genome.wustl.edu/pub/organism/Other_Vertebrates/Taeniopygia_guttata/assembly/Taeniopygia_guttata-3.2.4/ASSEMBLY" 
target="_blank">assembly notes</a>.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/taeGut1/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#zebrafinch">Downloads</a> page. These data 
have <a href="credits.html#zebrafinch_use">specific conditions for use</a>. Please refer to the 
WUSTL <a href="http://genome.wustl.edu/data.cgi" target="_blank">data use policy</a> for additional 
usage guidelines and citation information.</p> 
<p> 
The UCSC Zebra Finch Genome Browser and initial annotations were produced by Brian Raney, Kayla 
Smith, Pauline Fujita, and Donna Karolchik. See the 
<a href="credits.html#zebrafinch_credits">Credits</a> page for a detailed list of the organizations 
and individuals who contributed to this release. </p>

<a name="072208"></a>
<h2>Jul. 22, 2008 &nbsp;&nbsp; Updated <em>C. elegans</em> Browser released</h2> 
<p> 
Culminating our recent updates of the nematode browsers on our site, we've released a Genome Browser
for the latest <em>C. elegans</em> assembly based on sequence version WS190 deposited into WormBase 
in May 2008. This new assembly (UCSC version ce6) was produced jointly by the Wellcome Trust Sanger 
Institute in Hinxton, England, and the Genome Sequencing Center at Washington University in St. 
Louis (WUSTL) School of Medicine, MO, USA. The ce6 browser includes a Conservation annotation and 
pairwise alignment tracks comparing <em>C. elegans</em> sequence with that of five other worms.</p>
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/ce6">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#c_elegans">Downloads</a> page. Please
review the WUSTL <a href="http://genome.wustl.edu/data.cgi" target="_blank">data use policy</a> for 
usage restrictions and citation information.</p> 
<p> 
We'd like to thank WUSTL and the Sanger Institue for providing the sequence data for this assembly. 
The UCSC ce6 browser was produced by Hiram Clawson, Kayla Smith, Pauline Fujita, Brooke Rhead, and 
Donna Karolchik. See the Genome Browser <a href="credits.html#worm_credits">Credits</a> page for a 
detailed list of the organizations and individuals who contributed to this release. </p>

<a name="062608"></a>
<h2>Jun. 26, 2008 &nbsp;&nbsp; New Worm Genome available</h2> 
<p> 
Along with the set of worm browser updates that we're currently releasing, we've added a new 
nematode to the collection: <em>Caenorhabditis japonica</em>. This genome assembly (UCSC version 
caeJap1, Mar. 2008) corresponds to the v. 3.0.2 assembly produced by the Genome Sequencing Center 
at the Washington University St. Louis (WUSTL) School of Medicine.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/caeJap1">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#caeJap">Downloads</a> page. Please review the
WUSTL <a href="http://genome.wustl.edu/data.cgi" target="_blank">data use policy</a> for usage 
restrictions and citation information.</p> 
<p>
We'd like to thank WUSTL for providing the sequence data for this assembly. The UCSC caeJap1 browser
was produced by Hiram Clawson, Ann Zweig, and Donna Karolchik. See the Genome Browser
<a href="credits.html#worm_credits">Credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p> 

<a name="062008"></a>
<h2>Jun. 20, 2008 &nbsp;&nbsp; Two worm updates released</h2> 
<p> 
We've updated our browsers for the <em>C.  remanei</em> and <em>C. brenneri</em> nematode genomes. 
Both sets of sequence were obtained from the Genome Sequencing Center at Washington University in 
St. Louis (WUSTL) School of Medicine. The  <em>C. remanei</em> assembly (UCSC version caeRem3) 
corresponds to WUSTL version 15.0.1 dated May 2007. The <em>C. brenneri</em> assembly (UCSC version 
caePb2) is based on WUSTL version 6.0.1 dated Feb. 2008.</p>
<p>Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html">Downloads</a> page. Please review the WUSTL 
<a href="http://genome.wustl.edu/data.cgi" target="_blank">data use policy</a> for usage 
restrictions and citation information.</p> 
<p>
We'd like to thank WUSTL for providing the sequence data for these assemblies. The UCSC worm 
browsers were produced by Hiram Clawson, Brooke Rhead, Pauline Fujita, and Donna Karolchik. See the 
Genome Browser <a href="credits.html#worm_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>

<a name="061008b"></a>
<h2>Jun. 10, 2008 &nbsp;&nbsp; Lamprey Browser released</h2> 
<p> 
We have released a Genome Browser for the Mar. 2007 assembly of the lamprey genome, 
<em>Petromyzon marinus</em>. This assembly, UCSC version petMar1, was produced by the Genome 
Sequencing Center at the Washington University in St. Louis School of Medicine (WUSTL), St. Louis, 
MO, USA.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/petMar1/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#lamprey">Downloads</a> page. The lamprey 
sequence is made freely available before scientific publication. Please see the WUSTL 
<a href="http://genome.wustl.edu/data.cgi" target="_blank">data use policy</a> for usage 
restrictions and citation information.</p> 
<p> 
The UCSC Lamprey Genome Browser was produced by Hiram Clawson, Ann Zweig, Pauline Fujita, and Donna 
Karolchik. See the <a href="credits.html#lamprey_credits">Credits</a> page for a detailed list of 
the organizations and individuals who contributed to this release. </p>

<a name="061008a"></a>
<h2>Jun. 10, 2008 &nbsp;&nbsp; Lancelet genome available in browser</h2> 
<p> 
The Mar. 2006 release of the lancelet genome (<em>Branchiostoma floridae</em>) is now available in 
the UCSC Genome Browser. This assembly, UCSC version braFlo1, was produced by the 
<a href="http://genome.jgi-psf.org/Brafl1/Brafl1.info.html" target="_blank">DOE Joint Genome 
Institute</a> (JGI), Walnut Creek, CA, USA.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/braFlo1/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#lancelet">Downloads</a> page. The lancelet 
sequence is made freely available before scientific publication. Please see the JGI 
<a href="http://genome.jgi-psf.org/Brafl1/Brafl1.download.html" target="_blank">data release 
policy</a> for usage restrictions and citation information.</p> 
<p> 
The UCSC Lancelet Genome Browser was produced by Hiram Clawson, Ann Zweig, Pauline Fujita, and Donna
Karolchik. See the <a href="credits.html#lancelet_credits">Credits</a> page for a detailed list of 
the organizations and individuals who contributed to this release. </p>

<a name="060508"></a>
<h2>Jun. 5, 2008 &nbsp;&nbsp; Guinea Pig Browser released</h2> 
<p> 
The Feb. 2008 CavPor3 release of the guinea pig genome (<em>Cavia porcellus</em>) is now available 
in the UCSC Genome Browser. This assembly, UCSC version cavPor3, was produced by the 
<a href="http://www.broad.mit.edu/" target="_blank">Broad Institute</a>.</p> 
<p> 
The guinea pig is one of 24 mammals whose genomes are being sequenced as part of the 
<a href="http://www.broad.mit.edu/mammals/" target="_blank">Mammalian Genome Project</a>, funded by 
the National Institutes of Health. While most of these genomes are slated for low-coverage (2X), a 
limited subset (including the guinea pig genome) are being sequenced to a higher quality of 
6-7X.</p> 
<p> 
The guinea pig genome has been sequenced to 6.76X coverage with 95.55% of bases assembled. A total 
of 3143 scaffolds cover 2,722,377,657 bases (2.17% in gaps), with 50% of the scaffolds having a 
(N50) length of at least 27,408,292 bases (not including gaps). For more details about the assembly,
see the Broad Institute <a href="http://www.broad.mit.edu/mammals/" target="_blank">Mammmalian 
Genome Project</a> page.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/cavPor3/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#guineapig">Downloads</a> page. These data 
have <a href="credits.html#guineapig_use">specific conditions for use</a>. </p> 
<p> 
The UCSC Guinea Pig Genome Browser was produced by Tim Dreszer, Kate Rosenbloom, Hiram Clawson, 
Kayla Smith, Robert Kuhn, and Donna Karolchik. See the 
<a href="credits.html#guineapig_credits">Credits</a> page for a detailed list of the organizations 
and individuals who contributed to this release.</p>

<a name="050508"></a>
<h2>May 5, 2008 &nbsp;&nbsp; GSID HIV data browser now available</h2> 
<p> 
Global Solutions for Infectious Diseases (GSID) has announced the launch of an HIV Data Browser with
clinical and viral sequence data from infected subjects in the VAX004 (North American/European) 
Phase III clinical trial of the AIDSVAX B/B vaccine. The browser, which is a customized version of 
the UCSC Genome Browser developed by the UCSC Genome Bioinformatics group and hosted by GSID, 
provides researchers with searchable demographic and clinical data from volunteers who became HIV 
infected during the VAX004 trial. Using the browser, viral sequences may be aligned with one another
or with reference or consensus sequences.</p> 
<p> 
GSID is making these AIDSVAX data and serological samples available to the HIV research community 
through an agreement with VaxGen and with funding provided by the Bill and Melinda Gates Foundation.
Future releases will include the addition of clinical and viral sequence data from infected subjects
in the VAX003 (Thai) Phase III clinical trial of AIDSVAX B/E, and immunogenicity data from infected 
subjects in both the VAX004 and VAX003 trials. The browser may be expanded to include data from 
uninfected subjects in both trials as well.</p> 
<p> 
For information on accessing the GSID HIV Data Browser and background on the AIDSVAX clinical 
trials, visit <a href="http://gsid.org/gsid_hiv_data_browser.html" 
target="_blank">http://gsid.org/gsid_hiv_data_browser.html</a>.</p> 

<a name="042308"></a>
<h2>Apr. 23, 2008 &nbsp;&nbsp; Marmoset Browser released</h2> 
<p> 
We'd like to announce the release of a Genome Browser and Blat server for the marmoset genome 
(<em>Callithrix jacchus</em>). The Jun. 2007 assembly -- WUSTL version Callithrix jacchus-2.0.2, 
UCSC version calJac1 -- was produced by Washington University St. Louis (WUSTL) School of Medicine 
Genome Sequencing Center in St. Louis, MO, USA.</p> 
<p> 
<em>C. jacchus</em>, a member of the New World monkey clade, is the most widely studied marmoset. It
is a popular non-human primate model due to its small body size and unique biological features, and 
has contributed to the study of brain function, immunity, reproductive biology and drug toxicity. 
<em>C. jacchus</em> marmosets typically give birth to twins that are somatic chimeras, i.e. each 
sibling is the mixture of sibling genotypes. (Excerpted from the WUSTL <em>C. jacchus</em> 
<a href="http://genome.wustl.edu/genome.cgi?GENOME=Callithrix%20jacchus" 
target="_blank">project</a> page.)</p> 
<p> 
The <em>C. jacchus</em> genome was sequenced to 6X coverage using DNA from a female marmoset 
provided by the Southwestern National Primate Research Center in San Antonio, TX, USA. DNA from a 
full brother of the female was used as the source for the CHORI-259 BAC library. This assembly is 
composed of 49,724 supercontigs containing a total of approximately 3.02 billion bases. For more 
statistics and details on the assembly process, refer to the WUSTL Callithrix_jacchus-2.0.2 
<a href="http://genome.wustl.edu/pub/organism/Primates/Callithrix_jacchus/assembly/Callithrix_jacchus-2.0.2/ASSEMBLY" 
target="_blank">assembly</a> page.</p> 
<p> 
Bulk downloads of the calJac1 sequence and annotations may be obtained from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/calJac1/" target="_blank">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#marmoset">Downloads</a> page. The sequence 
data can also be obtained directly from 
<a href="http://genome.wustl.edu/pub/organism/Primates/Callithrix_jacchus/assembly/Callithrix_jacchus-2.0.2/" 
target="_blank">WUSTL</a>. See the WUSTL <a href="http://genome.wustl.edu/data.cgi" 
target="_blank">data use policy</a> for conditions of use. Please acknowledge WUSTL School of Medicine
Genome Sequencing Center in any publications that result from the use of this sequence assembly.</p>
<p> 
We'd like to thank WUSTL School of Medicine Genome Sequencing Center for providing this assembly. 
The initial set of marmoset browser annotation tracks were generated by UCSC. The UCSC marmoset 
Genome Browser team is Hiram Clawson, Robert Kuhn, Pauline Fujita, Brooke Rhead, and Donna 
Karolchik. See the <a href="credits.html#marmoset_credits">Credits</a> page for a detailed list of 
the organizations and individuals who contributed to this release.</p>

<a name="041508"></a>
<h2>Apr. 15, 2008 &nbsp;&nbsp; Introducing the Archaeal Genome Browser Database</h2> 
<p> 
We'd like to introduce the Archaeal Genome Browser Database, a visualization tool and an integrated 
repository for archaeal functional genomics data. This resource was developed by the 
<a href="http://lowelab.ucsc.edu/" target="_blank">Lowe Lab</a> in the UCSC Biomolecular Engineering
Department, with key assistance from the UCSC Genome Browser team. The Archaeal Genome Browser 
Database home page can be accessed via the <a href="http://archaea.soe.ucsc.edu/" 
target="_blank">Archaeal Genomes</a> menu link on the Genome Browser home page.</p> 
<p> 
Currently there are more than 50 completed archaeal genomes, the least studied domain of life. 
Although archaea and bacteria are both prokaryotes, often co-existing in the same environments, many
aspects of archaeal cell biology such as DNA replication, repair, transcription, and translation are
homologous to those found in eukaryotes. Some members of archaea are also notable for inhabiting 
extreme environments, including boiling terrestrial hot springs, black smoker vents at the bottom of
the ocean, the ultra briny water of the Dead Sea, and highly acidic drainage water from ore mines, 
to name a few.</p> 
<p> 
The Archaeal Genome Browsers offer a variety of basic tracks derived from Genbank RefSeq annotation,
along with published genome analyses from the Lowe Lab and external groups. The annotations include 
operon predictions, regulatory sequence motifs (promoters and Shine-Dalgarno), microarray data, 
multi-genome alignments, and protein conservation across major phylogenetic groups.</p> 
<p> 
The goal of the Lowe Lab is to make the Archaeal Genome Browser Database a forum for ongoing 
community-based genome annotation, and they welcome new experimental and bioinformatic analyses. If 
you would like to contribute data, or have questions or feedback about the database, contact 
<a href="mailto:&#108;o&#119;&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;&#100;&#117;">Todd Lowe</a>.</p>

<a name="040808"></a>
<h2>Apr. 8, 2008 &nbsp;&nbsp; Updated Cow Genome Browser now available</h2> 
<p> 
The UCSC Genome Bioinformatics Group has released a Genome Browser and Blat server for the Oct. 2007
draft assembly of the Cow genome <em>Bos taurus</em>. This assembly (UCSC version bosTau4) was 
produced by the Baylor College of Medicine Human Genome Sequencing Center (BCM HGSC) as Baylor 
release Btau_4.0.</p> 
<p> 
The Btau_4.0 release was produced using the 
<a href="http://www.genome.org/cgi/content/abstract/14/4/721">Atlas</a> genome assembly system at 
BCM HGSC. The sequencing strategy combined BAC shotgun reads with whole genome shotgun reads from 
small insert libraries as well as BAC end sequences. The assembly contains chromosomes 1-29 and X 
as well as 11869 scaffolds (named chrUn.004.*). The mitochondrial sequence (available in the browser
as &quot;chrM&quot;) was obtained from GenBank accession GI:60101824.</p> 
<p> 
The Btau_4.0 assembly was tested against available bovine sequence data sets (EST sequences and 
finished BAC sequences) for extent of coverage (completeness). When assembled contigs were tested, 
over 95% of the sequences in these data sets were found to be represented, indicating that the 
shotgun libraries used to sequence the genome were comprehensive. Of the 1.04 million EST sequences 
95.0% were contained in the assembled contigs. Assuming the ESTs are uniformly distributed 
throughout the genome, the estimated genome size is 2.73Gb/95% = 2.87Gb. For detailed information 
on the sequencing and assembly techniques, see the Baylor 
<a href="http://www.hgsc.bcm.tmc.edu/projects/bovine/" target="_blank">Bovine Genome Project</a> web
page.</p> 
<p> 
For a list of the chromosomes and scaffolds in this assembly, click the &quot;Sequences&quot; link 
on the cow browser <a href="../cgi-bin/hgGateway?db=bosTau4">gateway</a> page.</p> 
<p>Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/bosTau4">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#cow">Downloads</a> page. These data have 
<a href="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
target="_blank">specific conditions for use</a>. The cow annotation tracks were generated by UCSC 
and collaborators worldwide.</p> 
<p> 
We'd like to thank Baylor College of Medicine for providing this assembly. The UCSC bosTau4 browser 
and documentation were produced by Hiram Clawson, Brian Raney, and Ann Zweig. See the 
<a href="credits.html#cow_credits">Credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p>

<a name="032408"></a>
<h2>Mar. 24, 2008 &nbsp;&nbsp; Free Genome Browser institutional seminars</h2> 
<p> 
The UCSC Bioinformatics Group announces your chance to bring our hands-on computer workshop on the 
UCSC Genome Browser to your institution. The seminar is presented by our training partner, 
<a href="http://www.openhelix.com" target="_blank">OpenHelix</a>.</p> 
<p> 
The 3-1/2-hour introductory tutorial will cover the topics needed to effectively use our tool set, 
including: basic functionality of Genome Browser searching and BLAT use, Table Browser use, creating
and using Custom Tracks, and an introduction to the Gene Sorter. The workshop material requires 
knowledge of genomic/biological concepts, but no programming skills are required. Participants will 
receive slide hand-outs, exercises, and UCSC Genome Browser and Table Browser Quick Reference 
Cards.</p> 
<p> 
To bring the seminar to your institution, apply on the OpenHelix <a href="http://www.openhelix.com" 
target="_blank">website</a>. Participating institutions are required only to:</p> 
<ul> 
  <li>
  Provide a computer classroom with internet access. The preferred number of computers is 24 or 
  above. As an alternative to a computer room, the institution may provide a classroom with wireless
  access and ask participants to bring their own laptops. The seminar can be done in the morning and
  repeated in the afternoon to reach more users.</li> 
  <li>
  Conduct outreach and communication to participants as necessary.</li> 
</ul>
<p> 
The number of seminars is limited --  apply today! Contact information:  
<a href="http://www.openhelix.com" target="_blank">www.openhelix.com</a> or 1-888-861-5051.</p> 
<p> 
<strong>About OpenHelix, LLC:</strong> OpenHelix provides the genomics knowledge you need when you 
need it. OpenHelix offers online self-run tutorials, web seminars, and on-site training for 
institutions and companies on the most powerful and popular free, web-based, publicly accessible 
bioinformatics resources. In addition, OpenHelix is contracted by resource providers to provide 
comprehensive, long-term training and outreach programs. The company has its headquarters in 
Seattle, with offices in San Francisco and Boston. Further information can be found at 
www.openhelix.com or by calling 1-888-861-5051.</p>

<a name="031008"></a>
<h2>Mar. 10, 2008 &nbsp;&nbsp; Orangutan Genome Browser now available</h2> 
<p> 
The UCSC Genome Bioinformatics Group has released a Genome Browser and Blat server for the Jul. 2007
draft assembly of the Sumatran orangutan genome, <em>Pongo pygmaeus abelii</em>. This assembly 
(UCSC version ponAbe2, WUSTL version Pongo_albelii-2.0.2) was provided by the Genome Sequencing 
Center at Washington University School of Medicine in St. Louis (WUSTL), MO, USA.</p> 
<p> 
The orangutan genome was sequenced to 6X coverage using a female orangutan known as 
&quot;Susie&quot; from the Gladys Park Zoo (Brownsville, TX, USA). The combined sequence reads were 
assembled using <a href="http://www.genome.org/cgi/content/abstract/13/9/2164" 
target="_blank">PCAP</a> and filtered for all known non-orangutan sequence contaminants. For more 
details about the assembly, see the orangutan browser 
<a href="../cgi-bin/hgGateway?db=ponAbe2">gateway</a> page and the WUSTL 
<a href="http://genome.wustl.edu/genome.cgi?GENOME=Pongo%20abelii" target="_blank"><em>Pongo 
abelii</em></a> web page.</p> 
<p> 
Of the 3.09 Gb of total sequence, 3.08 Gb are ordered and oriented along the chromosomes. Gap sizes 
between supercontigs were estimated based on their size in human, with a maximum gap size of 30 kb 
allowed. For a list of the chromosomes in this assembly, click the &quot;Sequences&quot; link on the
orangutan browser <a href="../cgi-bin/hgGateway?db=ponAbe2">gateway</a> page. The mitochondrial 
sequence is also available as the virtual chromosome &quot;chrM&quot;.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/ponAbe2">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#orangutan">Downloads</a> page. These data 
have <a href="credits.html#orangutan_use">specific conditions for use</a>. The orangutan browser 
annotation tracks were generated by UCSC and collaborators worldwide.</p> 
<p> 
We'd like to thank WUSTL for providing this assembly. The UCSC ponAbe2 browser and documentation 
were produced by Hiram Clawson, Kayla Smith, Robert Kuhn, Ann Zweig and Donna Karolchik. See the 
<a href="credits.html#orangutan_credits">Credits</a> page for a detailed list of the organizations 
and individuals who contributed to this release.</p>

<a name="030708"></a>
<h2>Mar. 7, 2008 &nbsp;&nbsp; New reverse functionality released</h2> 
<p>
The Genome Browser team is happy to announce new functionality in the main Genome Browser track 
display. A new configuration button, &quot;Reverse,&quot; now allows users to view the entire 
browser image flipped right-to-left. This is especially useful when a user's gene of interest aligns
on the opposite strand from the reference assembly. Those genes then appear in the 5' to 3' 
direction.</p>
<p> 
When the reverse function has been activated, all of the track labels usually displayed on the left 
side of the track (including the mini-button providing access to configuration options) are 
displayed on the right side. This allows users to see at a glance which direction is being 
displayed. All navigation options operate as expected.</p> 
<p> 
Thanks to Mark Diekhans, Ann Zweig, Robert Kuhn and the rest of the engineering team for 
implementing this feature.</p>

<a name="010808"></a>
<h2>Jan. 8, 2008 &nbsp;&nbsp; Additional job opening with UCSC Genome Browser project</h2> 
<p> 
In addition to the openings listed in the Dec. 12, 2007 announcement (see below), the UCSC Genome 
Browser project is accepting applications for Research Software Architect, a position in the UC 
Project Scientist academic series. For the job description, qualifications and application 
information, please see the Center for Biomolecular Science and Engineering 
<a href="http://www.cbse.ucsc.edu/jobs/jobs_staff.shtml#architect" target="_blank">website</a>. To 
ensure full consideration, applications must be received by Jan. 22, 2008.</p>

<!---------------- 2007 archived news --------------->
<a name="2007"></a>
<a name="121207"></a>
<h2>Dec. 12, 2007 &nbsp;&nbsp; Job openings with UCSC Genome Browser project</h2> 
<p> 
The UCSC Genome Browser project is currently accepting applications for two positions on our 
development team: Software Development Engineer (Programmer/Analyst 3) and Biological Database 
Testing/User Support Technician (Programmer/Analyst 1). We are looking for talented self-motivated 
individuals who would like to use their skills in computer science, biology, and bioinformatics on 
a fast-paced project featuring the work of top genomics scientists worldwide.</p> 
<p> 
For a summary of the position details and qualifications, see the entries for 
<a href="http://www.cbse.ucsc.edu/jobs/jobs_staff.shtml#sftwrdevel" target="_blank">Software 
Developer</a> and <a href="http://www.cbse.ucsc.edu/jobs/jobs_staff.shtml#bdt" 
target="_blank">Biological Database Testing/User Support Technician</a> on the Center for 
Biomolecular Science and Engineering (CBSE) website. For detailed job descriptions and application 
information,  go to the <a href="http://jobs.ucsc.edu/" target="_blank">UCSC Staff Employment 
website</a>, click the &quot;Search Postings&quot; link on the sidebar, and type in job #0701419 
(Software Developer) or job #0701391 (Testing/User Support Technician).</p>

<a name="101507"></a>
<h2>Oct. 15, 2007 &nbsp;&nbsp; Genome Browser for Zebrafish Zv7 released</h2> 
<p> 
The latest zebrafish assembly -- Zv7 (UCSC version danRer5, July 2007) -- is now available in the 
UCSC Genome Browser. The Zv7 assembly was produced by The Wellcome Trust Sanger Institute in 
collaboration with the Max Planck Institute for Developmental Biology in Tuebingen, Germany, and the
Netherlands Institute for Developmental Biology (Hubrecht Laboratory), Utrecht, The Netherlands.</p>
<p> 
This assembly consists of 1,440,582,308 bp in 5,036 fragments. It includes sequence anchored to 
chromosomes 1-25 and chrM (mitochondrial), as well as 5010 unplaced scaffolds. The assembly was 
produced by integrating finished clone sequence from the physical map with whole genome shotgun 
assembly sequence. The N50 size is 1,153,933, n = 277 (i.e. the length such that 50% of the 
assembled genome lies in blocks of the N50 size or longer). For more information about this 
assembly, see the Sanger Institute web page for the 
<a href="http://www.sanger.ac.uk/Projects/D_rerio/Zv7_assembly_information.shtml" 
target="_blank"><em>Danio rerio</em> Sequencing Project</a> and the track description page for the 
Assembly track.</p> 
<p> 
The danRer5 sequence and annotation data can be downloaded from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/danRer5/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#zebrafish">downloads</a> page. Please review 
the <a href="http://www.sanger.ac.uk/notices/use-policy.shtml" target="_blank">guidelines</a> for 
using these data.</p>
<p> 
We'd like to thank the Wellcome Trust Sanger Institute, the Max Planck Institute for Developmental 
Biology, Hubrecht Laboratory and the other institutions who contributed to the sequencing and 
mapping effort of this release. Special thanks to the Zebrafish Genome Initiative at Children's 
Hospital in Boston for their collaboration on this release. The UCSC zebrafish Genome Browser was 
produced by Rachel Harte, Ann Zweig, and Donna Karolchik. The initial set of annotation tracks was 
generated by the UCSC Genome Bioinformatics Group. See the 
<a href="credits.html#zebrafish_credits">credits</a> page for a detailed list of the organizations 
and individuals who contributed to the release of this browser.</p>

<a name="101007"></a>
<h2>10 Oct. 10, 2007 &nbsp;&nbsp; New Browser for Purple Sea Urchin</h2> 
<p> 
The Sep. 2006 release of the purple sea urchin genome (<em>Strongylocentrotus purpuratus</em>) is 
now available in the UCSC Genome Browser. This assembly, UCSC version strPur2, was produced by the 
Baylor College of Medicine Human Genome Sequencing Center (BCM HGSC) and corresponds to their 
Spur_2.1 assembly.</p>  
<p> 
The Spur_2.1 release was assembled from BAC sequence reads (approximately 2x coverage) and whole 
genome shotgun reads (6x coverage), and utilizes BAC tiling path information. The BCM HGSC 
Atlas-2.0 genome assembly system was used to generate this assembly. The total length of all contigs
greater than 1kb is 804 Mbp.  When the gaps between contigs in scaffolds are included, the total 
span of the assembly is 907 Mbp. The estimated size of the genome based on the assembly is 814 
Mbp.</p> 
<p> 
Compared to previous sea urchin releases, the Spur_2.1 assembly is more continuous and has fewer 
false duplications; contaminations identified in the previous Spur_2.0 assembly have been removed.  
This draft assembly may contain errors; therefore, users should exercise caution. Typical errors may
include misassemblies of repeat sequences, collapses of repeat regions, and artificial duplications 
in polymorphic regions. However, base accuracy in contigs is usually very high with most errors near
the ends of contigs.</p> 
<p> 
More assembly details can be found in the 
<a href="ftp://ftp.hgsc.bcm.tmc.edu/pub/data/Spurpuratus/fasta/Spur_v2.1/README_Spur_v2.1.txt" 
target="_blank">Spur_2.1 README file</a> and on the BCM HGSC 
<a href="http://www.hgsc.bcm.tmc.edu/projects/seaurchin/" target="_blank">Sea Urchin Genome 
Project</a> web page.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/strPur2">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#urchin">Downloads</a> page. These data have 
<a href="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" target="_blank">specific 
conditions for use</a>. The initial set of strPur2 annotation tracks was generated by UCSC. See the 
<a href="credits.html#urchin_credits">Credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release. </p>

<a name="100407"></a>
<h2>Oct. 4, 2007 &nbsp;&nbsp; ENCODE Genome Browser Released for hg18 Assembly</h2> 
<p> 
The UCSC ENCODE browser for the human genome assembly hg18 (NCBI Build 36) is now available. You can
access the browser directly at
<a href="../ENCODE/encode.hg18.html">http://genome.ucsc.edu/ENCODE/encode.hg18.html</a> or by 
clicking the ENCODE link in the sidebar menu on this page, then clicking the Regions (hg18) link in 
the sidebar menu on the ENCODE portal page.</p> 
<p> 
The hg18 ENCODE browser includes 540 data tables in 59 browser tracks that were migrated from the 
hg17 browser. The hg17 data coordinates were converted to hg18 coordinates using the UCSC liftOver 
process.</p> 
<p> 
To improve the accessibility of the data, related ENCODE tracks have been gathered into new 
configuration groupings (&quot;super-tracks&quot;) that can be displayed or hidden using a single 
visiblity control. We have also reduced the number of track groups and have modified some of the 
group names for clarity. ENCODE tracks with whole-genome data have been moved into the standard 
browser track groups.</p> 
<p> 
For more information about the hg18 ENCODE data migration, see the News section on the UCSC 
<a href="../ENCODE/">ENCODE portal page</a> and the UCSC 
<a href="http://genomewiki.gi.ucsc.edu/index.php/ENCODE_Hg18_Migration" 
target="_blank">genomeWiki</a>.</p>

<a name="082207b"></a>
<h2>Aug. 22, 2007 &nbsp;&nbsp; New Mouse Browser released</h2> 
<p> 
The latest mouse genome assembly from the Mouse Genome Sequencing Consortium, NCBI Build 37.1, is 
now available in the UCSC Genome Browser. This version (UCSC version mm9) is considered to be 
essentially finished.</p>
<p> 
The Build 37.1 assembly includes approximately 2.6 Gb of sequence on chromosomes 1-19, X, Y, M 
(mitochondrial DNA) and Un (unmapped clone contigs). In-depth information about this assembly will 
become available on the 
<a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000001635.16/"
target="_blank">NCBI website</a>. On chromosome Y in this assembly, only the short arm has reliable 
mapping data; therefore, most of the contigs on the Y chromosome are unplaced.</p> 
<p> 
The mm9 sequence and annotation data may be downloaded from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/mm9/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/goldenPath/mm9/">Downloads</a> web page. The mm9 annotation 
tracks were generated by UCSC and collaborators worldwide. NOTE: To expedite the availability of the
mm9 browser on our website, the initial release does not contain the comparative genomics 
annotations. These will be added to our website as they become available. Also, note that the UCSC 
mm9 database contains only the reference strain C57BL/6J.</p>
<p> 
We'd like to thank Deanna Church and the Mouse Genome Sequencing Consortium for this assembly. 
We'd also like to acknowledge the work of the UCSC mm9 team: Hiram Clawson, Archana Thakkapallayil, 
Robert Kuhn, and Donna Karolchik. For a complete list of the individuals and organizations who 
participated in this assembly, see the <a href="credits.html#mouse_credits">Credits</a> page.</p>

<a name="082207a"></a>
<h2>Aug. 22, 2007 &nbsp;&nbsp; Latest Bovine assembly available</h2> 
<p> 
The UCSC Genome Bioinformatics Group has released a Genome Browser and Blat server for the Aug. 2006
Btau_3.1 draft assembly of the cow genome. This assembly (UCSC version bosTau3) was provided by 
Baylor College of Medicine Human Genome Sequencing Center in Houston, TX.</p> 
<p> 
The Btau_3.1 release was produced by the Atlas genome assembly system at Baylor College of Medicine 
Human Genome Sequencing Center. The  sequencing strategy combined BAC shotgun reads with whole 
genome shotgun reads from small insert libraries as well as BAC end sequences. The assembly contains
chromosomes 1-29 and X as well as 13045 scaffolds (named chrUn.003.*). More information on the 
Btau_3.1 assembly can be found on the Baylor <a href="http://www.hgsc.bcm.tmc.edu/projects/bovine/" 
target="_blank">Bovine Genome Project</a> web page and the 
<a href="ftp://ftp.hgsc.bcm.tmc.edu/pub/data/Btaurus/fasta/Btau20060815-freeze/ReadMeBovine.3.1.txt"
target="_blank">Readme</a> file that accompanies this release.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/bosTau3/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#cow">Downloads</a> page. Please refer to the 
Baylor <a href="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
target="_blank">conditions of use</a> regarding these data. The bosTau3 annotation tracks were 
generated by UCSC and collaborators worldwide.</p> 
<p>
We'd like to thank Baylor College of Medicine for the bovine sequence and assembly. We'd also like 
to acknowledge the members of the UCSC Genome Bioinformatics Group who contributed to the Cow Genome
Browser: Heather Trumbower, Angie Hinrichs, Kayla Smith, and Donna Karolchik. See the 
<a href="credits.html#cow_credits">Credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p>

<a name="080207"></a>
<Aug. 2, 2007 &nbsp;&nbsp; Five new worm assemblies</h2> 
<p> 
We've updated our existing nematode browsers--for <em>C.  elegans</em> and <em>C. briggsae</em>--to 
the latest publicly available assemblies. In addition, we've added three new worms to our 
collection: <em>C. brenneri</em> (Caenorhabditis n. sp. PB2801), <em>C. remanei</em>, and 
<em>Pristionchus pacificus</em>. The <em>C. elegans</em> sequence was obtained from WormBase; the 
Genome Sequencing Center at Washington University in St. Louis (WUSTL) provided sequence data for 
the other four assemblies.</p> 
<table>       
  <tr><th nowrap>SPECIES</th><th nowrap>UCSC VERSION</th><th nowrap>RELEASE DATE</th><th nowrap>SOURCE/RELEASE</th></tr> 
  <tr><td><em>C. elegans</em></td><td>ce4</td><td>Jan. 2007</td><td>WormBase v. WS170</td></tr> 
  <tr><td><em>C. briggsae</em></td><td>cb3</td><td>Jan. 2007</td><td>WUSTL Cb3</td></tr> 
  <tr><td><em>C. brenneri</em></td><td>caePb1</td><td>Jan. 2007</td><td>WUSTL 4.0</td></tr> 
  <tr><td><em>C. remanei</em></td><td>caeRem2</td><td>Mar. 2007</td><td>WUSTL 1.0</td></tr> 
  <tr><td><em>P. pacificus</em></td><td>priPac1</td><td>Feb. 2007</td><td>WUSTL 5.0</td></tr> 
</table> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html">Downloads</a> page. Please review the 
<a href="http://genome.wustl.edu/data.cgi" target="_blank">data use policy</a> for the 
WUSTL-generated assemblies. </p> 
<p> 
We'd like to thank WUSTL, WormBase, and the Sanger Institute for providing the sequence data for 
these assemblies. The UCSC worm browsers were produced by Hiram Clawson, Kayla Smith, Brooke Rhead, 
Ann Zweig, and Donna Karolchik. See the Genome Browser <a href="credits.html">Credits</a> page for 
a detailed list of the organizations and individuals who contributed to this release.</p>

<a name="071307"></a>
<h2>Jul. 13, 2007 &nbsp;&nbsp; <em>Drosophila melanogaster</em> Release 5 now available in Genome Browser</h2> 
<p> 
The latest <em>D. melanogaster</em> assembly can now be viewed in the UCSC Genome Browser. This 
version -- Release 5, dated Apr. 2006 (UCSC version dm3) -- was provided by the 
<a href="http://www.fruitfly.org/" target="_blank">Berkeley Drosophila Genome Project</a> (BDGP) and
combines both euchromatic and heterochromatic sequence. The Release 5.1 annotations (Mar. 2007) were
provided by <a href="http://flybase.net" target="_blank">FlyBase</a>.</p>  
<p> 
The six euchromatic arms in this assembly were sequenced and assembled by BDGP from a combination of
BAC and whole genome shotgun data, and have been finished to high quality.  All euchromatic sequence
have been compared to the restriction digest fingerprints in multiple enzymes for validity. The 
details of this analysis will be described in a forthcoming publication.</p> 
<p> 
Heterochromatic sequence from the <a href="http://www.fruitfly.org/" target="_blank">Drosophila 
Heterochromatin Genome Project</a> (DHGP) are also available in this assembly. Scaffolds that could 
not be unambiguously mapped to a chromosome arm have been concatenated into chrUn. chrUextra 
contains small scaffolds produced by the Celera shotgun assembler that could not be consistently 
joined with larger scaffolds. Because some of the chrUextra data are of low quality, researchers are
encouraged to contact either BDGP or DHGP for further details on this resource. For more information
on this assembly, see the Release 5 <a href="http://www.fruitfly.org/sequence/README.RELEASE5" 
target="_blank">assembly release notes</a>.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/dm3/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#fruitfly">Downloads</a> page. We'd like to 
thank the BDGP, DHGP, and Flybase for providing data for this release. The dm3 Genome Browser was 
produced by Angie Hinrichs, Archana Thakkapallayil, Kayla Smith, and Donna Karolchik. The 
<em>D. melanogaster</em> browser annotations were generated by FlyBase, DHGP, and the UCSC Genome 
Bioinformatics group. See the <a href="credits.html#dm_credits">Credits</a> page for a detailed list
of the organizations and individuals who contributed to this release.</p>

<a name="062607"></a>
<h2>Jun. 26, 2007 &nbsp;&nbsp; New 28-Species Conservation track released</h2> 
<p> 
We are pleased to announce the release of new versions of the Conservation and Most Conserved 
annotation tracks for the Human March 2006 Genome Browser (hg18, NCBI Build 36). The new 
Conservation track displays multiple alignments of 27 vertebrate species aligned to the human 
genome, along with measurements of evolutionary conservation across all species in the alignment and
a separate measurement of conservation across the placental mammal subset of species in the 
alignment.</p> 
<p> 
The new track includes: 
<ul> 
  <li>
  5 new high-quality assemblies -- horse, platypus, lizard, and two fish (stickleback and
  medaka)</li> 
  <li>
  6 new low-coverage mammalian genomes -- bushbaby, tree shrew, guinea pig, hedgehog, common shrew, 
  and cat</li>  
  <li>
  6 updated assemblies -- chimp, cow, chicken, frog, fugu, and zebrafish</li> 
  <li>
  10 assemblies included in the previous version of the track -- rhesus, mouse, rat, rabbit, dog, 
  armadillo, elephant, tenrec, opossum, and tetraodon</li> 
</ul>
<p> 
UCSC hosts browsers for the high-quality assemblies and the cat. Masked sequence data for the other 
genomes in the track are available from our downloads server.</p> 
<p> 
In addition to the expanded species list, the new Conservation track features the following 
improvements: 
<ul> 
  <li> 
  additional filtering of pairwise alignments for each species to reduce paralogous alignments</li> 
  <li> 
  information about the quality of aligning species sequence included in the multiple alignment 
  downloads</li> 
  <li> 
  new track configuration buttons to assist in selecting which species to display</li> 
</ul> 
<p> 
Bulk data downloads are available from the Genome Browser FTP server 
(<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/multiz28way">alignments</a>, 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/phastCons28way">conservation</a>) or from the
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#human">Downloads page</a>.</p> 
<p> 
The previous 17-vertebrate versions of the Conservation and Most Conserved tracks remain available 
on the hg18 Genome Browser as the &quot;17-Way Cons&quot; and &quot;17-Way Most Cons&quot; 
tracks.</p>

<a name="061307"></a>
<h2>Jun. 13, 2007 &nbsp;&nbsp; ENCODE findings released to public</h2> 
<p> 
The findings of the ENCODE project have been released to the public today, the culmination of a 
four-year effort to catalog the biologically functional elements in 1 percent of the human genome. 
The publications, which include a group paper in the 14 June 2007 issue of 
<a href="http://www.nature.com/nature/journal/v447/n7146/full/nature05874.html" 
target="_blank"><em>Nature</em></a> and 28 companion papers in the June 2007 issue of 
<a href="http://www.genome.org/content/vol17/issue6/" target="_blank"><em>Genome Research</em></a>, 
were authored by researchers from academic, governmental, and industry organizations located in 11 
countries. The <em>Nature</em> issue includes a pull-out poster featuring a screenshot of the UCSC 
Genome Browser displaying a broad range of the ENCODE data.</p> 
<p> 
In the <a href="http://genome.gov/25521554" target="_blank">press release</a> accompanying the 
publication rollout, NHGRI Director Francis S. Collins is quoted as saying &quot;This impressive 
effort has uncovered many exciting surprises and blazed the way for future efforts to explore the 
functional landscape of the entire human genome. Because of the hard work and keen insights of the 
ENCODE consortium, the scientific community will need to rethink some long-held views about what 
genes are and what they do, as well as how the genome's functional elements have evolved. This could
have significant implications for efforts to identify the DNA sequences involved in many human 
diseases.&quot;</p> 
<p> 
The main portal for the ENCODE data is the UCSC ENCODE Genome Browser. The analysis effort has been 
coordinated by <a href="http://wwww.ensembl.org/" target="_blank">Ensembl</a>. Much of the primary 
data have been deposited in the <a href="https://www.ncbi.nlm.nih.gov/projects/geo/info/ENCODE.html" 
target="_blank">NCBI GEO</a> and <a href="http://www.ebi.ac.uk/arrayexpress/" target="_blank">EBI 
ArrayExpress</a> databases. To access the UCSC Genome Browser ENCODE portal, click the 
<a href="../ENCODE/">ENCODE</a> link in the left sidebar menu on the Genome Browser home page.</p> 
<p> 
For more information on the ENCODE project, including the consortium's data release and 
accessibility policies and a list of NHGRI-funded participants, see the 
<a href="http://www.genome.gov/ENCODE" target="_blank">NHGRI ENCODE website</a>. To read more about 
UCSC's role on the project, see the 
<a href="http://www.cbse.ucsc.edu/news/2007/06/13/encode_pilot/index.shtml" target="_blank">news 
release</a> on the UCSC Center for Biomolecular Science and Engineering website.</p>

<a name="052507"></a>
<h2>May 25, 2007 &nbsp;&nbsp; Platypus Assembly available</h2> 
<p> 
We have released a Genome Browser and Blat server for the Jan. 2007 v5.0.1 draft assembly of 
<em>Ornithorhynchus anatinus</em> (UCSC version ornAna1) produced by the 
<a href="http://genome.wustl.edu/genome.cgi?GENOME=Ornithorhynchus%20anatinus" 
target="_blank">Genome Sequencing Center at Washington University</a>, St. Louis, MO (WUSTL).</p> 
<p> 
This assembly, which was sequenced using a combination of whole genome shotgun plasmid, fosmid and 
BAC end sequences, has a coverage of approximately 6X. It is comprised of about 1.84 Gb of actual 
sequence (excluding gap estimates), with 437 Mb anchored and ordered on chromosomes.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/ornAna1/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#platypus">Downloads</a> page. These data have
<a href="credits.html#platypus_use">specific conditions for use</a>.</p> 
<p> 
We'd like to thank WUSTL for providing this assembly. The platypus Genome Browser was produced by 
Angie Hinrichs, Kayla Smith, Robert Kuhn, Brian Raney, and Donna Karolchik. The platypus browser 
annotation tracks were generated by UCSC and collaborators worldwide. See the 
<a href="credits.html#platypus_credits">Credits</a> page for a detailed list of the organizations 
and individuals who contributed to this release.</p> 

<a name="051707"></a>
<h2>May 17, 2007 &nbsp;&nbsp; Lizard Assembly available in Genome Browser</h2> 
<p> 
A Genome Browser and Blat server are now available for the Feb. 2007 v1.0 draft assembly (UCSC 
version anoCar1) of <em>Anolis carolinensis</em> produced by the <a href="http://www.broad.mit.edu/"
target="_blank">Broad Institute</a>. This assembly has been sequenced to 6.8X coverage. The draft 
sequence contains 7,233 scaffolds comprised of nearly 1.74 Gb.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/anoCar1/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#lizard">Downloads</a> page. These data have 
<a href="credits.html#lizard_use">specific conditions for use</a>.</p>  
<p> 
We'd like to thank the Broad Institute for providing this assembly. The lizard Genome Browser was 
produced by Hiram Clawson, Archana Thakkapallayil, Robert Kuhn, and Donna Karolchik. The lizard 
browser annotation tracks were generated by UCSC and collaborators worldwide. See the 
<a href="credits.html#lizard_credits">Credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p>

<a name="051607"></a>
<h2>May 16, 2007 &nbsp;&nbsp; New dates for OpenHelix seminars in L.A., D.C., Phila.</h2> 
<p> 
The dates for the upcoming OpenHelix Genome Browser seminars in Washington, D.C. and Los Angeles, CA
have been changed. The following updated announcement includes the new dates:</p> 
<p> 
The UCSC Bioinformatics Group announces three regional seminars and hands-on computer workshops on 
the UCSC Genome Browser presented by <a href="http://www.openhelix.com/" 
target="_blank">OpenHelix:</a></p> 
<ul> 
  <li> 
  Philadelphia -- Wednesday, Jun. 13</li> 
  <li> 
  Washington. D.C./Baltimore -- Wednesday, Jun. 20</li> 
  <li> 
  Los Angeles -- Wednesday, Jun. 27</li> 
</ul>
<p> 
All sessions will be held 1 p.m. to 4 p.m.</p> 
<p> 
These introductory sessions are geared towards anyone with a basic knowledge of genomic and 
biological concepts who is interested in learning how to use the UCSC Genome Browser. No programming
experience is required. The seminars will cover the topics necessary to learn how to effectively use
the browser tool set, including basic Genome Browser functionality, searching and BLAT use, Table 
Browser use, creating and using custom annotation tracks, and an introduction to the Gene Sorter. 
Lectures will be accompanied by hands-on computer exercises conducted directly on the Genome Browser
web site. Participants receive a complete set of slide and exercise handouts and printed Quick 
Reference Cards.</p> 
<p> 
For further information or to make a reservation, visit the <a href="http://www.openhelix.com/" 
target="_blank">OpenHelix</a> website or call 1-888-861-5051. Register early; seating is limited. 
Academic and student discounts are available.</p> 

<a name="051407"></a>
<h2>May 14, 2007 &nbsp;&nbsp; Horse Genome Browser now available</h2> 
<p> 
The Jan. 2007 EquCab1 release of the horse genome (<em>Equus caballus</em>) is now available in the 
UCSC Genome Browser. This assembly, UCSC version equCab1, was produced by the 
<a href="http://www.broad.mit.edu/" target="_blank">Broad Institute</a>.</p> 
<p> 
The horse draft genome has been sequenced to 6.8X coverage. Approximately 84% of the sequence has 
been anchored to chromosomes, which include autosomes 1-31 and sex chromosome X. Unanchored contigs 
that could not be localized to a chromosome have been concatenated into the virtual chromosome 
&quot;chrUn&quot;, separated by gaps of 1,000 bp. The mitochondrial sequence is also available in 
the Genome Browser as the virtual chromosome &quot;chrM&quot;. For more details about the assembly, 
see the Broad Institute <a href="http://www.broad.mit.edu/mammals/horse/" target="_blank">Horse 
Genome Project</a> page.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/equCab1/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#horse">Downloads</a> page. These data have 
<a href="credits.html#horse_use">specific conditions for use</a>.</p> 
<p> 
The UCSC Horse Genome Browser was produced by Fan Hsu, Brooke Rhead, Robert Kuhn, Hiram Clawson, 
Angie Hinrichs, Kate Rosenbloom, and Donna Karolchik. See the 
<a href="credits.html#horse_credits">Credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p>

<a name="042707"></a>
<h2>Apr. 27, 2007 &nbsp;&nbsp; Medaka Genome Browser released</h2> 
<p> 
We're happy to announce the release of a Genome Browser and Blat server for the Medaka Version 1.0 
draft assembly (Apr. 2006, UCSC version oryLat1). This assembly was produced in Japan by the 
<a href="http://www.nig.ac.jp/english/index.html" target="_blank">National Institute of Genetics</a>
(NIG) and the <a href="http://medaka.utgenome.org/" target="_blank">University of Tokyo</a>. It is 
equivalent to Ensembl's Oct. 2005 MEDAKA1 data set.</p> 
<p>
The v1.0 assembly has been sequenced to 10.6X coverage. It consists of approximately 700.4 million 
bp (excluding gaps) on chromosomes 1-24. 7,299 scaffolds comprised of nearly 36,500 contigs of 
unplaced sequence are displayed on the virtual chromosome &quot;chrUn&quot; These contigs are spaced
with a 10 bp gap; scaffold gaps are 100 bp in size. The medaka mitochondrial sequence is also 
available in the Genome Browser as the virtual chromosome &quot;chrM&quot;.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/oryLat1/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#medaka">Downloads</a> page. See the 
University of Tokyo Medaka website for the <a href="http://medaka.utgenome.org/#policy" 
target="_blank">data release policy</a> for this assembly.</p> 
<p> 
The Medaka browser annotation tracks were generated by UCSC and collaborators worldwide. See the 
Genome Browser <a href="credits.html#medaka_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>

<a name="041807"></a>
<h2>Apr. 18, 2007 &nbsp;&nbsp; Free Genome Browser seminar at Experimental Biology 2007</h2> 
<p> 
OpenHelix will present a free introductory seminar on the Genome Browser during Experimental 
Biology 2007, April 29-May 1, in Washington, D.C. The tutorial will cover the topics needed to 
effectively use the Genome Browser, including: basic functionality of Genome Browser searching and 
BLAT use, Table Browser use, creating and using Custom Tracks, and an introduction to the Gene 
Sorter.</p> 
<p> 
The seminar will be held on Monday April 30, 4:30-5:30p.m. in Room 204C in the Washington, D.C. 
Convention Center. It is open to any interested conference attendee with a basic knowledge of 
genomic/biological concepts; no programming skills are needed.</p> 
<p>
The tutorial requires no advanced registration or fee. Attendees will receive a free download of the
training materials. For more information, see the <a href="http://www.openhelix.com" 
target="_blank">OpenHelix</a> website or call 1-888-861-5051.</p> 
<p> 
In addition to the tutorial, OpenHelix will be presenting brief introductory overviews of the UCSC 
Genome Browser and other bioinformatics resources during show hours at Booth 330/332. Stop by the 
booth for more information and to receive your free Genome Browser Quick Reference Cards.</p>

<a name="041107"></a>
<h2>Apr. 11, 2007 &nbsp;&nbsp; Latest Fugu assembly available in Genome Browser</h2> 
<p> 
The UCSC Genome Browser now includes the latest release of the Fugu genome. The v4.0 whole genome 
shotgun assembly (Oct. 2004, UCSC fr2) was provided by the US DOE Joint Genome Institute (JGI) as 
part of the International Fugu Genome Consortium led by the JGI and the Singapore Institute of 
Molecular and Cell Biology (IMCB).</p> 
<p> 
This version has been sequenced to approximately 8.5X coverage. The assembly contains 7,213 
scaffolds covering 393,312,790 bp. The UCSC browser displays the scaffolds on the virtual chromosome
<em>chrUn</em> with gaps of 1,000 bp between scaffolds. The scaffolds range in size from 2,223 bp to
7,245,445 bp. Fifty percent of the sequence (196,648,171 bp) is contained within 125 scaffolds of 
size 858,115 or greater (N50). The Fugu mitochondrial sequence is also available as the virtual 
chromosome <em>chrM</em> (GenBank accession: NC_004299.1).</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/fr2/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#fugu">Downloads</a> page. These data have 
been freely provided by the JGI for use in the UCSC Genome Browser.</p> 
<p> 
Many thanks to the JGI, IMCB, and the International Fugu Genome Consortium for the assembly data. 
The UCSC Fugu Genome Browser was produced by Cory McClean, Hiram Clawson, Ann Zweig, and Donna 
Karolchik. The annotation tracks were generated by UCSC and collaborators worldwide. See the 
<a href="credits.html#fugu_credits" target="_blank">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p> 

<a name="040607"></a>
<h2>Apr. 6, 2007 &nbsp;&nbsp; New UCSC Gene Prediction set released</h2> 
<p> 
We are pleased to announce the release of a new gene prediction set, UCSC Genes, on the latest human
Genome Browser (hg18, NCBI Build 36). This annotation, which includes putative non-coding genes as 
well as protein-coding genes and 99.9% of RefSeq genes, is the next generation of the Known Genes 
set that UCSC has been providing for several years and supersedes the existing Known Genes 
annotation on the hg18 assembly.</p> 
<p> 
The UCSC Genes is a moderately conservative prediction set based on data from RefSeq, GenBank, and 
UniProt. Each entry requires the support of one GenBank RNA sequence plus at least one additional 
line of evidence, with the exception of RefSeq RNAs, which require no additional evidence. Some of 
the non-coding transcripts in the set may actually code for protein, but the evidence for the 
associated protein is weak at best. Compared to RefSeq, this gene set generally has about 10% more 
protein-coding genes, approximately five times as many putative non-coding genes, and about twice as
many splice variants.</p> 
<p> 
A new companion track to UCSC Genes, Alt Events, shows various types of alternative splicing, 
alternative promoter, and other events that result in more than a single transcript from the same 
gene. This track is based on an analysis by the txgAnalyse program of splicing graphs produced by 
the txGraph program.</p> 
<p> 
The UCSC Genes set is produced using a computational pipeline developed at UCSC by Jim Kent, Chuck 
Sugnet and Mark Diekhans. The programs used to construct the Alt Events data set were written by 
Jim Kent. For detailed information about the process used to construct the genes set, see the 
<a href="../cgi-bin/hgTrackUi?db=hg18&g=knownGene">track description page</a>. In upcoming months, 
we plan to release UCSC Genes sets on several organisms in addition to human. The UCSC Genes 
annotations will be updated approximately every three months.</p> 
<p> 
As part of this change, we are now using our own UCSC Genes accession numbers as the primary key 
into the underlying knownGene table, rather than the GenBank mRNA accessions we used in the previous
Known Genes prediction set.  Note that this may affect external sites with URLs that link into our 
genes track using the older-style accessions.</p> 
<p> 
We will continue to provide the older Known Genes track on hg18 under the name &quot;Old Known 
Genes&quot;. You may find the following tables useful in referencing the older gene set and 
converting between the two sets: 
<ul> 
  <li> 
  knownGeneOld2: new name for table underlying the old Known Genes (previously called 
  knownGene)</li> 
  <li> 
  kgXrefOld2: new name for table that contains data for converting old Known Genes IDs to other IDs 
  (previously called kgXref)</li> 
  <li> 
  kg2ToKg3: data for converting old Known Genes IDs to the newer UCSC Genes IDs</li> 
</ul> 
<p> 
We'd like to acknowledge the many people affiliated with the UCSC Genome Bioinformatics group who 
worked hard to release this new annotation: developers Jim Kent, Mark Diekhans, and Fan Hsu (with 
technical support from several other engineers in the group); David Haussler; our splendid QA 
team -- Archana Thakkapallayil, Ann Zweig, Robert Kuhn, Kayla Smith, and Brooke Rhead; our build 
engineer -- Andy Pohl; and our sysadmin group. We'd also like to thank Chuck Sugnet for his input, 
the people and organizations maintaining the RefSeq, UniProt, and GenBank databases, and the 
scientists worldwide who have contributed to them. If you have any questions about this new release,
feel free to contact us at 
<a href="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</a>
<!-- above address is genome at soe.ucsc.edu -->
(general questions) or 
<a href="mailto:&#103;e&#110;&#111;m&#101;&#45;&#109;&#105;&#114;&#114;o&#114;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;&#117;">&#103;e&#110;&#111;m&#101;&#45;&#109;&#105;&#114;&#114;o&#114;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;&#117;</a> 
<!-- above address is genome-mirror at soe.ucsc.edu --> 
(mirror-specific questions).</p>

<a name="032707"></a>
<h2>Mar. 27, 2007 &nbsp;&nbsp; Stickleback assembly released in Genome Browser</h2> 
<p> 
We have released a Genome Browser and Blat server for the Feb. 2006 v1.0 draft assembly of 
<em>Gasterosteus aculeatus</em> produced by the <a href="http://www.broad.mit.edu/" 
target="_blank">Broad Institute</a>.</p> 
<p> 
This assembly has been sequenced to approximately 6X coverage. An estimated  87% of the sequence has
been anchored to chromosomes (chrI - chrXXI). Of the remaining unanchored scaffolds, those that 
could be localized to a chromosome have been concatenated into the virtual chromosome 
&quot;chrUn&quot; with 1000bp gaps between scaffolds. The stickleback mitochondrial sequence is also
available as the virtual chromosome &quot;chrM&quot;.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/gasAcu1/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#stickleback">Downloads</a> page.These data 
have <a href="credits.html#stickleback_use">specific conditions for use</a>.</p> 
<p> 
The stickleback browser annotation tracks were generated by UCSC and collaborators worldwide. See 
the <a href="credits.html#stickleback_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>

<a name="021607"></a>
<h2>Feb. 16, 2007 &nbsp;&nbsp; New browser session-sharing function available</h2> 
<p> 
We are pleased to announce the release of a new session management functionality in the Genome 
Browser, which allows users to save and share browser sessions.</p> 
<p> 
Users are now able to configure their browsers with specific track combinations, including custom 
tracks, and save the configuraton options. Multiple sessions may be saved for future reference, for 
comparison of scenarios or for sharing with colleagues. Saved sessions persist for one year after 
the last access, unless deleted. Custom tracks persist for at least 48 hours after the last time 
they are viewed.</p> 
<p> 
The new feature may be accessed via the &quot;Sessions&quot; link in the top blue bar in any 
assembly. To ensure privacy and security, users must login to the genomewiki site and create a 
username and password. Individual sessions may be designated by the user as either 
&quot;shared&quot; or &quot;non-shared&quot; to protect the privacy of confidential data.</p>
<p> 
To avoid having a new shared session from someone else override existing Genome Browser settings, 
users are encouraged to open a new web-browser instance or to save existing settings in a session 
before loading a new shared session.</p> 
<p> 
The Sessions feature was written by Angie Hinrichs of the UCSC Genome Bioinformatics Group and 
released with the assistance of Kayla Smith and Robert Kuhn.</p>

<a name="020207"></a>
<h2>Feb. 2, 2007 &nbsp;&nbsp; New Genome Graphs tool available in Genome Browser</h2> 
<p> 
We are pleased to announce the release of a new software tool in the Genome Browser collection, the 
<a href="../cgi-bin/hgGenome">Genome Graphs</a> tool. Genome Graphs offers the ability to upload and
display genome-wide data sets such as the results of genome-wide SNP association studies, linkage 
studies and homozygosity mapping. The Genome Graphs tool may be accessed from the menu on the UCSC 
Genome Bioinformatics home page.</p> 
<p> 
The initial release of Genome Graphs includes the following features:</p> 
<ul> 
  <li> 
  upload several sets of genome-wide data and display them simultaneously</li> 
  <li> 
  click on an area of interest and go directly to the genome browser at that position</li> 
  <li> 
  set a significance threshold for your data and view only regions that meet that threshold</li> 
  <li> view the genes that exist in areas where your data meet your significance threshold</li> 
</ul> 
<p> 
For more information about the Genome Graphs tool, visit the Gateway page or consult the 
<a href="help/hgTracksHelp.html#GenomeGraphs">Getting Started on Genome Graphs</a> section in the 
User's Guide.</p> 
<p> 
Genome Graphs was written by Jim Kent of the UCSC Genome Bioinformatics Group and released with the 
assistance of Ann Zweig.</p>

<a name="012207"></a>
<h2>Jan. 22, 2007 &nbsp;&nbsp; Cat assembly available in Genome Browser</h2> 
<p> 
The Mar. 2006 release of <em>Felis catus</em> (UCSC version felCat3) is now available in the Genome 
Browser. This assembly was produced by <a href="http://www.broad.mit.edu/" target="_blank">The Broad
Institute</a> of MIT/Harvard and <a href="http://www.agencourt.com/" target="_blank">Agencourt 
Bioscience</a>.</p> 
<p> 
The felCat3 genome has been sequenced to 2X coverage and consists of 217,790 scaffolds. The total 
contig length for this assembly is approximately 1.6 Gb spanning nearly 4.0 Gb (with 60.1&#37; in 
gaps). There are 749,376 contigs, with an N50 length of 2,506 bases. There are 149,283 supercontigs,
with an N50 length of 49,769 bases (not including gaps). The N50 size is the length such that 
50&#37; of the assembled genome lies in blocks of the N50 size or longer.</p> 
<p> 
The felCat3 sequence and annotation data can be downloaded from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/felCat3/"> FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#cat">Downloads</a> page. Please review the 
<a href="/goldenPath/credits.html#cat_use">guidelines</a> for using the cat assembly data.</p>      
<p> 
Many thanks to The Broad Institute for providing these data. The UCSC cat Genome Browser was 
produced by Heather Trumbower, Angie Hinrichs, Mark Diekhans, Brooke Rhead, and Archana 
Thakkapallayil. The initial set of annotation tracks was generated by the UCSC Genome Bioinformatics
Group. See the Genome Browser <a href="/goldenPath/credits.html#cat_credits">Credits</a> page for a 
detailed list of the organizations and individuals who contributed to the release of this 
browser.</p>

<a name="010107"></a>
<h2>Jan. 1, 2007 &nbsp;&nbsp; Upcoming Genome Browser seminars: SF, Seattle, NYC, Cleveland</h2> 
<p> 
The UCSC Bioinformatics Group announces four regional seminars and hands-on computer workshops on 
the UCSC Genome Browser, presented by <a href="http://www.openhelix.com/" 
target="_blank">OpenHelix:</a></p>
<ul> 
  <li> 
  San Francisco, CA -- Wednesday, Jan. 31</li> 
  <li> 
  Seattle, WA -- Thursday, Feb. 1</li> 
  <li> 
  New York City, NY -- Tuesday, Feb. 13</li> 
  <li> 
  Cleveland, OH -- Wednesday, Feb. 14</li> 
</ul>
<p> 
Two sessions will be offered for the New York and Cleveland seminars: 9:00 a.m. to noon and 1 p.m. 
to 4 p.m. Only the afternoon session will be offered in San Francisco and Seattle.</p> 
<p> 
These introductory sessions are geared towards anyone with a basic knowledge of genomic and 
biological concepts who is interested in learning how to use the UCSC Genome Browser. No programming
experience is required. The seminars will cover the topics necessary to learn how to effectively use
the browser tool set, including basic Genome Browser functionality, searching and BLAT use, Table 
Browser use, creating and using custom annotation tracks, and an introduction to the Gene Sorter. 
Lectures will be accompanied by hands-on computer exercises conducted directly on the Genome Browser
web site. Participants receive a complete set of slide and exercise handouts and printed Quick 
Reference Cards.</p> 
<p> 
For further information or to make a reservation, visit the <a href="http://www.openhelix.com/" 
target="_blank">OpenHelix</a> website or call 1-888-861-5051. Register early; seating is limited. 
Academic and student discounts are available.</p> 

<!---------------- 2006 archived news --------------->
<a name="2006"></a>

<a name="111306"></a>
<h2>Nov. 13, 2006 &nbsp;&nbsp; UCSC Genome Browser wiki</h2> 
<p> 
The UCSC Genome Bioinformatics group has launched a wiki site for sharing information about the UCSC
Genome Browser and its data. The wiki -- at <a href="http://genomewiki.ucsc.edu" 
target="_blank">http://genomewiki.ucsc.edu</a> -- provides an informal forum for our browser users, 
mirror sites, and staff to discuss topics of interest in the genome biology field and exchange usage
tips, scripts/programs, and notes about mirroring the Genome Browser and working with the Genome 
Browser source.</p>
<p> 
As with most wiki pages, general users are welcome to edit and add pages (login required). Please 
note that all content created on the genomewiki site becomes a public resource; content persists in 
the history of a page even after it has been deleted.</p>

<a name="102406"></a>
<h2>Oct. 24, 2006 &nbsp;&nbsp; Free Genome Browser training sessions at AHA Scientific Sessions</h2>
<p> 
<a href="http://www.openhelix.com/" target="_blank">OpenHelix</a> will present a free seminar on the
UCSC Genome Browser at the American Heart Association's Scientific Sessions 2006 in Chicago,
IL, on Nov. 12 from 7:00 - 8:30 p.m. The seminar will be held in the Hyatt Regency Conference Center
, Room CC21, 2233 South Martin Luther King Drive, Chicago, IL.</p>  
<p> 
The introductory tutorial will cover the topics needed to effectively use the Genome Browser 
including: basic functionality of Genome Browser searching and BLAT use, Table Browser use, creating
and using Custom Tracks, and an introduction to the Gene Sorter. The jointly-sponsored seminar will 
also include an introduction to VISTA comparative genomics tools.</p> 
<p> 
The session is open to anyone attending the AHA Scientific Sessions; no registration or fee is 
required. Participants should have a basic knowledge of genomic/biological concepts, but no 
programming skills are needed. Attendees will receive a free download of training materials, and 
refreshments will be served. This event is not part of the official Scientific Sessions 2006 as 
planned by the AHA Committee on Scientific Sessions Program.</p> 
<p> 
OpenHelix will also be presenting brief introductory overviews of the Genome Browser and other 
resources during show hours at booth 2464. Stop by the booth for more information and to receive 
your free Quick Reference Cards for the Genome Browser and Table Browser.</p> 

<a name="100606"></a>
<h2>Oct. 6, 2006 &nbsp;&nbsp; Announcing upgrades to the Genome Browser custom tracks functionality</h2> 
<p> 
We have enhanced one of the popular tools in the Genome Browser collection: 
the <a href="../cgi-bin/hgCustom">custom tracks</a> utility. The new custom tracks tool provides a 
more user-friendly interface and increased flexibility for creating and managing your custom 
tracks.</p> 
<p> 
The initial release of this upgraded tool includes the following features:</p> 
<ul> 
  <li> 
  Add and display multiple custom tracks simultaneously via URL, file or text</li>
  <li> 
  Add to, delete and modify the uploaded custom tracks set using a new track management
  interface</li> 
  <li> 
  Load and manage custom tracks from multiple assemblies</li>
  <li> 
  Create and upload description pages for custom tracks</li>
  <li> 
  Custom tracks will now persist on our server for 48 hours after last access (rather than 8
  hours)</li>
</ul> 
<p> 
For more information about the new custom tracks functionality, see the Genome Browser 
<a href="help/customTrack.html">Users's Guide</a>. The hgCustom CGI was written by Kate Rosenbloom 
with the assistance of Archana Thakkapallayil, Ann Zweig and other members of the UCSC Genome 
Bioinformatics Group.</p>

<a name="090706"></a>
<h2>Sep. 7, 2006 &nbsp;&nbsp; Old Rhesus assembly archived</h2> 
<p> 
The Jan. 2005 rheMac1 draft assembly has been archived. The data remain available on our 
<a href="http://hgdownload.gi.ucsc.edu/goldenPath/rheMac1/">downloads server</a>, but blat 
services are no longer supported. </p>

<a name="080806"></a>
<h2>Aug. 8, 2006 &nbsp;&nbsp; New Opossum assembly available in Genome Browser</h2> 
<p> 
The UCSC Genome Browser now includes the latest draft assembly of the opossum genome. The Jan. 2006 
release of <em>Monodelphis domestica</em> (UCSC version monDom4) was sequenced and assembled by 
<a href="http://www.broad.mit.edu/mammals/opossum/" target="_blank">The Broad Institute</a>, 
Cambridge, MA, USA.</p> 
<p> 
This draft, which has approximately 6.5X coverage, has an assembly length of nearly 3.61 billion bp 
including gaps (3.50 billion bp without gaps) contained on chromosomes 1-8, X, and Un. The N50 of 
the genome including gaps is 104,359 bp; the N50 without gaps is 107,990. The N50 size is the length
such that 50% of the assembled genome lies in blocks of the N50 size or longer.</p> 
<p> 
The monDom4 sequence and annotation data can be downloaded from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/monDom4/"> FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/goldenPath/monDom4/">Downloads</a> page. Please review the 
<a href="credits.html#opossum_use">guidelines</a> for using the opposum assembly data.</p> 
<p> 
Many thanks to The Broad Institute for providing these data. The UCSC opossum Genome Browser was 
produced by Hiram Clawson, Archana Thakkapallayil, Ann Zweig, Kayla Smith and Donna Karolchik. The 
initial set of annotation tracks was generated by the UCSC Genome Bioinformatics Group. See the 
Genome Browser <a href="credits.html#opossum_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to the release of this browser.</p>

<a name="080106"></a>
<h2>Aug. 1, 2006 &nbsp;&nbsp; v2.1 Chicken assembly available in Genome Browser</h2> 
<p> 
We have updated the Chicken Genome Browser to include the May 2006 v2.1 assembly (UCSC version 
galGal3) produced by the Genome Sequencing Center at the Washington University School of Medicine 
in St. Louis, MO, USA (WUSTL). The source of this sequence was a female inbred Red Jungle Fowl 
(<em>Gallus gallus</em>), the ancestor of domestic chickens. The chicken genome is the first of the 
avian genomes to be sequenced.</p>
<p> 
In this assembly, 198,000 additional reads covering all contig ends and regions of low quality have 
been added to the original assembly's 6.6X coverage. Approximately 95% of the sequence has been 
anchored to chromosomes, which include autosomes 1-24, 26-28, and 32, and sex chromosomes W and Z. 
(In contrast to mammals, the female chicken is heterogametic (ZW) and the male is homogametic (ZZ).)
The remaining unanchored contigs that could be localized to a chromosome have been concatenated into
the virtual chromosomes &quot;chr*_random&quot;, separated by gaps of 10,000 bp.  Unanchored contigs
that could not be localized to a chromosome have been concatenated into the virtual chromosome 
&quot;chrUn_random&quot;, separated by gaps of 100 bp to reduce the total size of chrUn_random. The 
chicken mitochondrial sequence is also available as the virtual chromosome &quot;chrM&quot;.</p> 
<p> 
Although centromere positions are indicated on this assembly, little is known of their exact 
sequence. The centromeres of 18 chromosomes were tentatively localized based on FISH hybridization 
using BAC clones, genetic markers flanking the centromeres in coordination with mapping gaps in the 
physical map, repetitive sequence content, and analysis of proximity to the constrictions of the 
mitotic metaphase chromosomes. For more information on the process used to create the chromosomal 
sequences and assign centromere locations, see the WUSTL 
<a href="http://genome.wustl.edu/genomes/detail/gallus-gallus/" target="_blank">Gallus gallus 
genome</a> page.</p> 
<p> 
Bulk downloads of the chicken sequence and annotations may be obtained from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/galGal3/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#chicken">Downloads</a> page. These data have 
<a href="credits.html#chicken_use">specific conditions for use</a>.</p>
<p>
We'd like to thank WUSTL, who provided the sequence, physical map, assembly, and assembly/map for 
this release. The genetic mapping and linkage analysis were produced through a collaborative effort 
of labs in The Chicken Mapping Consortium. The chicken browser annotation tracks were generated by 
UCSC and collaborators worldwide. See the <a href="credits.html#chicken_credits">Credits</a> page 
for a detailed list of acknowledgements. The UCSC Chicken Genome Browser was produced by Angie 
Hinrichs, Kayla Smith, and Donna Karolchik.</p> 

<a name="072006"></a>
<h2>Jul. 20, 2006 &nbsp;&nbsp; New Chimpanzee Genome Browser released</h2> 
<p> 
We are happy to announce the release of a Genome Browser for the latest release of the chimpanzee 
(Pan troglodytes) genome. The Mar. 2006 assembly -- labeled Chimp Build 2 Version 1 (UCSC version 
panTro2) -- was produced by the 
<a href="http://genome.wustl.edu/genome.cgi?GENOME=Pan%20troglodytes&SECTION=collaborators" 
target=_BLANK>Chimpanzee Sequencing and Analysis Consortium</a>.</p> 
<p>
This 6X whole genome assembly contains sequence from the initial 4X chimpanzee assembly described 
and analyzed in <em>Nature</em> 
(<a href="http://www.nature.com/nature/journal/v437/n7055/full/nature04072.html" target="_blank">The
Chimpanzee Sequencing and Analysis Consortium, 2005</a>), with additional 2X sequence generated, 
assembled, and assigned to chromosomes by the 
<a href="http://genome.wustl.edu/genome.cgi?GENOME=Pan%20troglodytes" 
target=_BLANK>Genome Sequencing Center of Washington University School of Medicine</a>, St. Louis, 
MO, USA.</p> 
<p> 
The whole genome shotgun data were derived primarily from the donor Clint, a captive-born male 
chimpanzee from the Yerkes Primate Research Center in Atlanta, GA, USA. The reads were assembled 
using the whole-genome assembly program PCAP. For information about the assembly process, see the 
panTro2 <a href="../cgi-bin/hgGateway?db=panTro2">Gateway</a> page. </p> 
<p> 
This assembly covers about 97 percent of the genome and is based on 6X sequence coverage. It is 
composed of 265,882 contigs with an N50 length of 29 kb and 44,460 supercontigs with an N50 length 
of 9.7 Mb. The total contig length, not including estimated gap sizes, is 2.97 Gb. Of that total, 
2.82 Gb of sequence have been ordered and oriented along specific chimpanzee chromosomes, 107 Mb 
have been placed in chr*_random, and 50 Mb remain in chrUn.</p> 
<p> 
A major difference between this assembly and the previous Nov. 2003 version is the chromosomal 
numbering scheme, which  has been changed to reflect a new standard that preserves orthology with 
human chromomes. Proposed by 
<a href="https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=pubmed&cmd=Retrieve&dopt=Abstract&list_uids=15218271&query_hl=1&itool=pubmed_docsum" 
target="_blank">E.H. McConkey</a> in 2004, the new numbering convention was subsequently endorsed by
the International Chimpanzee Sequencing and Analysis Consortium. This standard assigns the 
identifiers &quot;2a&quot; and &quot;2b&quot; to the two chimp chromosomes that fused in the human 
genome to form chromosome 2. Note that the genome assembly shown in the Nov. 2003 panTro1 Genome 
Browser retains the older numbering scheme, in which these chromosomes are numbered 12 and 13.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/panTro2/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#chimp">Downloads</a> page. The complete set 
of sequence reads is available at the <a href="https://www.ncbi.nlm.nih.gov/Traces" 
target="_blank">NCBI trace archive</a>. These data have <a href="credits.html#chimp_use">specific 
conditions for use</a>.</p> 
<p> 
We'd like to thank the International Chimpanzee Sequencing and Analysis Consortium, Washington 
University at St. Louis School of Medicine Genome Sequencing Center, and the Broad Institute for 
providing this sequence. We'd also like to acknowledge the UCSC team who worked on this release: 
Kate Rosenbloom, Brian Raney, Hiram Clawson, Ann Zweig, Archana Thakkapallayil, and Donna Karolchik.
The chimpanzee browser annotation tracks were generated by UCSC and collaborators worldwide.</p>

<a name="062006"></a>
<h2>Jun. 20, 2006 &nbsp;&nbsp; Genome Browser released for Baylor v3.4 Rat assembly</h2> 
<p> 
The UCSC Genome Bioinformatics group has released a Genome Browser for the v3.4 rat (<em>Rattus 
norvegicus</em>) genome. This assembly--UCSC version rn4, November 2004--was produced by the Atlas 
group at <a href="http://www.hgsc.bcm.edu/" target="_blank">Baylor Human Genome Sequencing 
Center</a> (HGSC) as part of the Rat Genome Sequencing Consortium.</p> 
<p> 
The sequence was assembled using a hybrid approach that combines the clone-by-clone and whole genome
shotgun methods. The assembly is a minor update to version 3.3 that spliced in 54.6 Mb finished BAC 
sequences; the overall statistics are unchanged from releases 3.0 to 3.4.</p> 
<p> 
The 3.x assemblies reflect several sequence additions and software improvements over the previous 
2.x assemblies, including the sequencing of over 1100 new BACs to cover gaps, an improved marker 
set from the Medical College of Wisconsin, a new FPC map from the BC Cancer Agency Genome Sciences 
Centre, and improved linking of bactigs. For detailed information and statistics about the 3.x 
assemblies, see the Baylor HGSC <a href="http://www.hgsc.bcm.tmc.edu/projects/rat/" 
target="_blank">Rat Genome Project</a> web page. </p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/rn4/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#rat">Downloads</a> page. These data are made 
available with 	<a href="/goldenPath/credits.html#rat_use">specific conditions for use</a>. 
<p> 
We'd like to thank the Rat Genome Sequencing Consortium and the Baylor HGSC for providing this 
assembly. We'd also like to acknowledge the UCSC team who produced the rn4 Genome Browser: Angie 
Hinrichs, Fan Hsu, Brooke Rhead, Archana Thakkapallayil, Kayla Smith, Ann Zweig, Robert Kuhn, and 
Donna Karolchik. The rn4 annotation tracks were generated by UCSC and collaborators worldwide. See 
the <a href="credits.html#rat_credits">Credits</a> page for a detailed list of the organizations and
individuals who contributed to this release.</p> 

<a name="061206"></a>
<h2>Jun. 12, 2006 &nbsp;&nbsp; Latest <em>X. tropicalis</em> assembly available in Browser</h2> 
<p> The v.4.1 <em>Xenopus tropicalis</em> assembly is now available on the UCSC Genome Browser. This
whole genome shotgun assembly (xenTro2, Aug. 2005) was generated by the U.S. DOE Joint Genome 
Institute (JGI) using the Jazz assembler. It contains 19,501 scaffolds with an average coverage of 
7.65X. Roughly half the genome is contained in 272 scaffolds, each of at least 1.56 Mb in 
length.</p> 
<p> 
In this release, some scaffolds showing homology to a known prokaryotic contaminant as well as 
non-cellular or vector contamination have been removed by the JGI and placed in a separate 
directory. The <em>X.  tropicalis</em> assembly will be improved over the coming year by additional 
sequencing of large insert clones, targeted gap closure, and the incorporation of physical and 
genetic mapping information as it becomes available.</p> 
<p> 
For more information about the 4.1 assembly, see the JGI 
<a href="http://genome.jgi-psf.org/Xentr4/Xentr4.home.html" target="_blank"><em>X. tropicalis</em> 
website</a>.</p> 
<p> 
The xenTro2 sequence and annotation data can be downloaded from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/xenTro2/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#xenTro">downloads</a> page. These data have 
<a href="http://genome.jgi-psf.org/Xentr4/Xentr4.download.html" target="_blank">specific conditions 
for use</a>.</p>
<p> 
Many thanks to the JGI and the other institutions who contributed to the sequencing and mapping 
effort for this release. The xenTro2 Genome Browser was produced by Angie Hinrichs, Kayla Smith, 
Robert Kuhn, and Donna Karolchik. The xenTro2 annotation tracks were generated by UCSC and 
collaborators worldwide. See the <a href="credits.html#xenTro_credits">credits</a> page for a 
detailed list of the organizations and individuals who contributed to this release.</p>

<a name="052506"></a>
<h2>May 25, 2006 - Zv6 Zebrafish Browser Released</h2> 
<p>
The latest zebrafish assembly -- Zv6 (UCSC version danRer4, March 2006) -- is now available in the 
UCSC Genome Browser. The Zv6 assembly was produced by The Wellcome Trust Sanger Institute in 
collaboration with the Max Planck Institute for Developmental Biology in Tuebingen, Germany, and the
Netherlands Institute for Developmental Biology (Hubrecht Laboratory), Utrecht, The Netherlands.</p>
<p> 
This assembly consists of 1,626,077,335 bp in 6.653 scaffolds (N50 = 1,247,221 bp) with a sequence 
coverage of approximately 6.5-7x. The sequence has been anchored to chromosomes 1-25, chrM 
(mitochondrial), chrNA_random, and chrUn_random. For more information about this assembly, see the 
Sanger Institute web page for the 
<a href="http://www.sanger.ac.uk/Projects/D_rerio/Zv6_assembly_information.shtml/" 
target="_blank"><em>Danio rerio</em> Sequencing Project</a>.</p>  
<p> 
The danRer4 sequence and annotation data can be downloaded from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/danRer4/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#zebrafish">downloads</a> page. Please review 
the <a href="http://www.sanger.ac.uk/Projects/use-policy.shtml" target="_blank">guidelines</a> for 
using these data.</p>  
<p> 
We'd like to thank the Wellcome Trust Sanger Institute, the Max Planck Institute for Developmental 
Biology, Hubrecht Laboratory and the other institutions who contributed to the sequencing and 
mapping effort of this release. Special thanks to the Zebrafish Genome Initiative at Children's 
Hospital in Boston for their collaboration on this release. The UCSC zebrafish Genome Browser was 
produced by Rachel Harte, Archana Thakkapallayil, Robert Kuhn, Ann Zweig, and Donna Karolchik. The 
initial set of annotation tracks was generated by the UCSC Genome Bioinformatics Group.  See the 
<a href="credits.html#zebrafish_credits">credits</a> page for a detailed list of the organizations 
and individuals who contributed to the release of this browser.</p>

<a name="051006"></a>
<h2>May 10, 2006 &nbsp;&nbsp; NCBI Mouse Build 36 released in Genome Browser</h2> 
<p> 
The latest mouse genome assembly from the Mouse Genome Sequencing Consortium, NCBI Build 36, is now 
available in the UCSC Genome Browser. This version (UCSC version mm8) is considered to be 
essentially finished.</p>  
<p> 
The Build 36 assembly includes approximately 2.6 Gb of sequence on chromosomes 1-19, X, Y, M 
(mitochondrial DNA) and Un (unmapped clone contigs). For in-depth information about the process used
to assemble this version, see the NCBI website.  On chromosome Y in this assembly, only the short 
arm has reliable mapping data; therefore, most of the contigs on the Y chromosome are unplaced. 
Note that the UCSC mm8 database contains only the reference strain C57BL/6J.</p> 
<p> 
The mm8 sequence and annotation data may be downloaded from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/mm8/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/goldenPath/mm8/">Downloads</a> web page. The mm8 annotation 
tracks were generated by UCSC and collaborators worldwide.</p> 
<p> 
We'd like to thank Deanna Church and the Mouse Genome Sequencing Consortium for this assembly. We'd 
also like to acknowledge the work of the UCSC mm8 team: Hiram Clawson, Fan Hsu, Kayla Smith, Ann 
Zweig, Robert Kuhn, Brooke Rhead, Archana Thakkapallayil, and Donna Karolchik. For a complete list 
of the individuals and organizations who participated in this assembly, see the 
<a href="credits.html#mouse_credits">Credits</a> page.</p>

<a name="041406"></a>
<h2>Apr. 14, 2006 &nbsp;&nbsp; NCBI Build 36.1 human reference sequence available in browser</h2> 
<p> 
The latest human genome reference sequence assembly (NCBI Build 36.1, March 2006) is now available 
as database hg18 in the UCSC Genome Browser. This sequence, which was obtained from NCBI, was 
produced by the International Human Genome Sequencing Consortium.</p> 
<p> 
The hg18 assembly contains four alternate haplotype regions: 
<ul> 
  <li> 
  chr22_h2_hap1 -- an alternate chromosome 22 assembly that contains the CYP2D6 gene (NT_113959.1). 
  CYP2D6 is deleted in the reference assembly.</li> 
  <li> 
  chr5_h2_hap1 -- a chromosome 5 alternate assembly of the SMN1 gene region (NT_113801.1, 
  NT_113802.1).</li> 
  <li> 
  chr6_cox_hap1 -- an A1-B8-DR3 alternate haplotype assembly of the chromosome 6 MHC region based on
  sequence data from the COX library (NT_113891.1).</li>  
  <li> 
  chr6_qbl_hap2 -- an A26-B18-DR3 alternate haplotype assembly of the chromosome 6 MHC region based 
  on sequence data from the QBL library (NT_113892.1, NT_113893.1, NT_113894.1, NT_113895.1, 
  NT_113896.1, NT_113897.1).</li> 
</ul> 
<p> 
See the <a href="http://www.sanger.ac.uk/HGP/Chr6/MHC/" target="_blank">Wellcome Trust Sanger 
Institute MHC Haplotype Project</a> web site for additional information on the chr6 alternate 
haplotype assemblies.</p> 
<p> 
The Y chromosome in this assembly contains two pseudoautosomal regions (PARs) at chrY:1-2709520 and 
chrY:57443438-57772954. These sequences were taken from the corresponding regions in the X 
chromosome and are exact duplications of the X chromosome sequences.</p> 
<p> 
For further information on NCBI Build 36.1, see the NCBI 
<a href="https://www.ncbi.nlm.nih.gov/mapview/stats/BuildStats.cgi?taxid=9606&build=36&ver=1" 
target="_blank">Build 36.1 release notes</a>.</p> 
<p> 
Bulk downloads of the data are available from the UCSC downloads server via 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18" target="_blank">ftp</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#human" target="_blank">http</a>. We recommend
that you use ftp or rsync for downloading large or multiple files.</p> 
<p> 
We'd like to thank NCBI and the International Human Genome Sequencing Consortium for furnishing the 
data, and the entire UCSC Genome Browser staff for contributing to this release. Fan Hsu led the 
UCSC engineering effort; QA was headed up by Ann Zweig.</p>

<a name="030606"></a>
<h2>Mar. 6, 2006 &nbsp;&nbsp; Purple Sea Urchin genome assembly available in Genome Browser</h2> 
<p> 
The Apr. 2005 release of the Purple Sea Urchin genome (<em>Strongylocentrotus purpuratus</em>) is 
now available in the UCSC Genome Browser. This assembly, UCSC version strPur1, was produced by the 
Baylor College of Medicine Human Genome Sequencing Center (BCM HGSC) and corresponds to their 
Spur_0.5 whole genome shotgun assembly.</p> 
<p> 
This release was assembled from whole genome shotgun reads using the Atlas genome assembly system 
at the BCM HGSC. Several whole genome shotgun libraries, with inserts of 2-6 kb, were used to 
produce the data. About 7 million reads were assembled, representing about 800 Mb of sequence and 
about 6x coverage of the (clonable) sea urchin genome. Highly repeated sequences were assembled 
separately into reptigs and merged into the genome assembly. Sequences from BAC clones were omitted 
from this assembly and will be placed in a subsequent version of the draft sequence.</p> 
<p> 
This is a draft sequence and may contain errors; therefore, users should exercise caution. Typical 
errors in draft genome sequences include misassemblies of repeated sequences, collapses of repeated 
regions, and unmerged overlaps (<em>e.g.</em> due to polymorphisms) creating artificial 
duplications. However, base accuracy in contigs (contiguous blocks of sequence) is usually very high
with most errors near the ends of contigs.</p> 
<p> 
More assembly details can be found in the 
<a href="ftp://ftp.hgsc.bcm.tmc.edu/pub/data/Spurpuratus/fasta/Spur20050415/READMEApril2005.txt" 
target="_blank">Spur_0.5 README file</a> and on the BCM HGSC 
<a href="http://www.hgsc.bcm.tmc.edu/projects/seaurchin/" target="_blank">Sea Urchin Genome 
Project</a> web page.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/strPur1">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#urchin">Downloads</a> page. These data have 
<a href="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
target="_blank">specific conditions for use</a>. The strPur1 annotation tracks were generated by UCSC 
and collaborators worldwide. See the <a href="credits.html#urchin_credits">Credits</a> page for a 
detailed list of the organizations and individuals who contributed to this release.</p> 

<a name="022706"></a>
<h2>Feb. 27, 2006 &nbsp;&nbsp; Upcoming Genome Browser seminars in Texas, Florida and Washington, 
DC</h2> 
<p> 
The UCSC Bioinformatics Group announces four regional seminars and hands-on computer workshops on 
the UCSC Genome Browser, presented by <a href="http://www.openhelix.com/" 
target="_blank">OpenHelix:</a></p>
<ul> 
  <li> 
  Houston, TX -- Tuesday, Mar. 14</li>
  <li> 
  Austin, TX -- Wednesday, Mar. 15</li>
  <li> 
  Washington, DC -- Tuesday, Apr. 18</li>
  <li> 
  Miami, FL -- Wednesday, Apr. 19</li>
</ul> 
<p>
Two sessions will be offered each day: 9:00 a.m. to noon and 1 p.m. to 4 p.m.</p> 
<p> 
These introductory sessions are geared towards anyone with a basic knowledge of genomic and 
biological concepts who is interested in learning how to use the UCSC Genome Browser. No 
programming experience is required. The seminars will cover the topics necessary to learn how to 
effectively use the browser tool set, including basic Genome Browser functionality, searching and 
BLAT use, Table Browser use, creating and using custom annotation tracks, and an introduction to the
Gene Sorter. Lectures will be accompanied by hands-on computer exercises conducted directly on the 
Genome Browser web site. Participants receive a complete set of slide and exercise handouts and 
printed Quick Reference Cards.</p> 
<p> 
For further information or to make a reservation, visit the 
<a href="http://www.openhelix.com/noteworthy.shtml" target="_blank">OpenHelix</a> website or call 
1-888-861-5051. Register early; seating is limited. Academic and student discounts are 
available.</p> 

<a name="021706"></a>
<h2>Feb. 17, 2006 &nbsp;&nbsp; Zebrafish Browser updated</h2> 
<p> 
The latest zebrafish assembly -- Zv5 (UCSC version danRer3, May 2005) -- is now available in the 
UCSC Genome Browser and Blat server. The Zv5 assembly was produced by The Wellcome Trust Sanger 
Institute in collaboration with the Max Planck Institute for Developmental Biology in Tuebingen, 
Germany, and the Netherlands Institute for Developmental Biology (Hubrecht Laboratory), Utrecht, 
The Netherlands.</p> 
<p> 
This assembly consists of 1,630,306,866 bp in 16,214 scaffolds (N50 = 1,116,981 bp) with a sequence 
coverage of approximately 6.5-7x. The assembly has been tied to the fingerprint contig map (data 
freeze 15th February, 2005) and contains 699 Mb from 4,519 sequenced clones.</p> 
<p>
For more information about this assembly, see the Sanger Institute web page for the 
<a href="http://www.sanger.ac.uk/Projects/D_rerio/" target="_blank"><em>Danio rerio</em> Sequencing 
Project</a>.</p>
<p> 
The danRer3 sequence and annotation data can be downloaded from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/danRer3/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#zebrafish">downloads</a> page. Please review 
the <a href="http://www.sanger.ac.uk/Projects/use-policy.shtml" target="_blank">guidelines</a> for 
using these data.</p>  
<p> 
We'd like to thank the Wellcome Trust Sanger Institute, the Max Planck Institute for Developmental 
Biology, Hubrecht Laboratory and the other institutions who contributed to the sequencing and 
mapping effort of this release. Special thanks to the Zebrafish Genome Initiative at Children's 
Hospital in Boston for their collaboration on this release. The UCSC zebrafish Genome Browser was 
produced by Rachel Harte, Jennifer Jackson, Ann Zweig, Ali Sultan-Qurraie, and Donna Karolchik. The 
initial set of annotation tracks was generated by the UCSC Genome Bioinformatics Group. See the 
<a href="credits.html#zebrafish_credits">credits</a> page for a detailed list of the organizations 
and individuals who contributed to the release of this browser.</p>

<a name="020906"></a>
<h2>Feb. 9, 2006 &nbsp;&nbsp; New Rhesus Macaque Browser released</h2> 
<p> 
The latest rhesus macaque (<em>Macaca mulatta</em>) draft assembly&mdash;v.1.0, Mmul_051212&mdash;is
now available in the UCSC Genome Browser. This version (UCSC rheMac2) was sequenced and assembled by
the Macaque Genome Sequencing Consortium led by the Baylor College of Medicine Human Genome 
Sequencing Center, in collaboration with the Genome Sequencing Center at Washington University 
School of Medicine in St. Louis and the J. Craig Venter Institute Joint Technology Center.</p> 
<p> 
The rhesus macaque follows the human and chimpanzee as the third primate and first Old World monkey 
to have its genome sequenced. Overall, the rhesus genome shares approximately 92 to 95 precent of 
its sequence with the human, compared with the chimp at 98 precent. Because of its genetic, 
physiologic, and metabolic similarities to the human and chimp, the rhesus is an ideal reference 
point for comparisons among the three primates.</p> 
<p> 
The groups in the Macaque Genome Sequencing Consortium produced preliminary assemblies of the 
genome data using different and complementary approaches. The resulting data were combined into a 
single, high-density 	&quot;melded&quot; assembly by a team at J. Craig Venter Institute. This 
collaboration made use of published rhesus maps, the BAC fingerprint map from the Michael Smith 
Genome Sciences Centre, and the human reference genome sequence. The v.1.0 assembly covers about 93 
percent of the rhesus genome. For more information about the rheMac2 assembly, see the Baylor 
<a href="http://www.hgsc.bcm.tmc.edu/projects/rmacaque/" target="_blank">Rhesus Monkey Project</a> 
web page.</p> 
<p> 
The rheMac2 sequence and annotation data can be downloaded from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/rheMac2/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#rhesus">Downloads</a> page. These data have 
specific <a href="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
target="_blank">conditions for use</a>.</p> 
<p> 
Many thanks to Baylor College of Medicine, the Macaque Genome Sequencing Consortium, and the other 
institutions who contributed to the sequencing and mapping effort of the v.1.0 release. The UCSC 
Rhesus Genome Browser was produced by Robert Baertsch, Kayla Smith, Ann Zweig, Robert Kuhn, and 
Donna Karolchik. The initial set of rheMac2 annotation tracks was generated by the 
<a href="../staff.html">UCSC Genome Bioinformatics Group</a>. See the 
<a href="credits.html#rhesus_credits">credits</a> page for a detailed list of the organizations and 
individuals who contributed to the release of this browser.</p>

<a name="011806"></a>
<h2>Jan. 18, 2006 &nbsp;&nbsp; <em>D. sechellia</em> Browser released</h2>
<p>
The Oct. 2005 <em>D. sechellia</em> assembly (UCSC version droSec1) is now available in the UCSC 
Genome Browser. This version was sequenced and assembled by the Broad Institute of MIT and 
Harvard.</p> 
<p> 
Downloads of the droSec1 data and annotations can be obtained from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/droSec1/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#droSec">Downloads</a> page. The initial set 
of droSec1 annotation tracks were generated by UCSC.</p> 
<p> 
Many thanks to the Broad Institute for providing the sequence and assembly of this genome. The UCSC 
<em>D.  sechellia</em> Genome Browser was produced by Angie Hinrichs, Kayla Smith and 
Donna Karolchik. See the <a href="credits.html#droSec_credits">Credits</a> page for a detailed list 
of the organizations and individuals who contributed to this release.</p>

<a name="011106"></a>
<h2>Jan. 11, 2006 &nbsp;&nbsp; <em>D. yakuba</em> Browser update</h2> 
<p> 
The latest <em>D. yakuba</em> assembly is now available in the UCSC Genome Browser. This version 
&mdash;  Release 2.0, dated Nov. 2005 (UCSC version droYak2) &mdash; was sequenced and assembled by 
the Genome Sequencing Center, Washington University (WUSTL) School of Medicine in St. Louis. </p> 
<p> 
The whole genome shotgun (WGS) assembly includes both raw shotgun data and data from two rounds of 
automated, directed read selection, which has improved the sequence quality and narrowed or (in some
instances) closed gaps.  For more assembly information and statistics, see the WUSTL Genome 
Sequencing Center <a href="http://genome.wustl.edu/genome.cgi?GENOME=Drosophila%20yakuba&GROUP=6" 
target="_blank"><em>Drosophila yakuba</em></a> web page.</p> 
<p> 
Downloads of the droYak2 data and annotations can be obtained from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/droYak2/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#droYak">Downloads</a> page. The initial set 
of droYak2 annotation tracks was generated by UCSC.</p> 
<p> 
Thanks to the Genome Sequencing Center at WUSTL School of Medicine for providing the sequence and 
assembly of this genome. The droYak2 Genome Browser was produced by Angie Hinrichs, Jennifer Jackson
and Donna Karolchik. See the <a href="credits.html#droYak_credits">Credits</a> page for a detailed 
list of the organizations and individuals who contributed to this release.</p>

<a name="010406"></a>
<h2>Jan. 4, 2006 &nbsp;&nbsp; <em>D. persimilis</em> Genome Browser released</h2> 
<p> 
The UCSC Genome Bioinformatics Group has released a Genome Browser and Blat server for the 
<em>D. persimilis</em> genome. The Oct. 2005 release (UCSC version droPer1) was sequenced and 
assembled by the Broad Institute of MIT and Harvard. The <em>D. persimilis</em> assembly joins nine 
other Drosophila species featured in the UCSC Genome Browser.</p> 
<p> 
Downloads of the droPer1 data and annotations can be obtained from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/droPer1/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#droPer">Downloads</a> page. The droPer1 
annotation tracks were generated by UCSC and collaborators worldwide.</p>  
<p>
Thanks to the Broad Institute for providing the sequence and assembly of this genome. The UCSC 
<em>D.  persimilis</em> Genome Browser was produced by Angie Hinrichs, Kayla Smith, Robert Kuhn, 
Jennifer Jackson and Donna Karolchik. See the <a href="credits.html#droPer_credits">Credits</a> 
page for a detailed list of the organizations and individuals who contributed to this release.</p>

<!---------------- 2005 archived news --------------->
<a name="2005"></a>

<a name="122005"></a>
<h2>Dec. 20, 2005 &nbsp;&nbsp; Dog Genome Browser update released</h2> 
<p> 
UCSC has updated the dog Genome Browser to include the May 2005 v2.0 assembly (UCSC version canFam2)
sequenced and assembled by the Broad Institute of MIT and Harvard and Agencourt Bioscience. The 
whole genome shotgun sequence is based on 7.6X coverage of the dog genome which includes more than 
98% of the euchromatic genome.</p>  
<p> 
The dog genome, which contains approximately 2.5 billion base pairs, is similar in size to the 
genomes of humans and other mammals. The boxer breed was selected for the initial sequencing effort,
based on the lower variation rate in its genome relative to other breeds. In addition to the boxer, 
samples from several other dog breeds were used to generate a set of single nucleotide polymorphisms
(SNPs) to facilitate disease studies. The SNPs are available from the Broad Institute 
<a href="http://www.broad.mit.edu/ftp/pub/papers/dog_genome/snps_canfam2/" target="_blank">dog 
SNP</a> web page. For more information about the dog draft assembly, see the Broad Institute 
<a href="http://www.broad.mit.edu/mammals/dog/" target="_blank">Dog Genome Sequencing Project</a> 
web page.</p>  
<p> 
The dog sequence and annotation data can be downloaded from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/canFam2/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#dog">downloads</a> page. These data have 
<a href="credits.html#dog_use">specific conditions for use</a>.</p>  
<p> 
Many thanks to the Broad Institute of MIT and Harvard, Agencourt Bioscience, and the other 
institutions who contributed to the sequencing, assembly, and mapping efforts. The canFam2 Genome 
Browser team included Angie Hinrichs, Jennifer Jackson, and Donna Karolchik. See the 
<a href="credits.html#dog_credits">credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p>  

<a name="121505"></a>
<h2>Dec. 15, 2005 &nbsp;&nbsp; New Mouse assembly available in Genome Browser</h2> 
<p> 
The latest mouse genome assembly from the Mouse Genome Sequencing Consortium, NCBI Build 35 (UCSC 
version mm7), is now available in the UCSC Genome Browser.</p>  
<p> 
The Build 35 assembly includes approximately 2.6 Gb of sequence, of which about 2.2 Gb is finished 
sequence. Chromosomes 2, 4, 11 and X are finished in this build. To review in-depth statistics on 
the assembly, see the NCBI 
<a href="https://www.ncbi.nlm.nih.gov/genome/guide/mouse/release_notes.html" 
target="_blank">Build 35 Data</a> web page. Please note that the UCSC mm7 database contains only the
reference strain C57BL/6J.</p> 
<p> 
The mm7 sequence and annotation data may be downloaded from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/mm7/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/goldenPath/mm7/">Downloads</a> web page. The mm7 annotation 
tracks were generated by UCSC and collaborators worldwide.</p> 
<p> 
We'd like to thank Deanna Church and the Mouse Genome Sequencing Consortium for this assembly. We'd 
also like to acknowledge the work of the UCSC mm7 team: Hiram Clawson, Fan Hsu, Ann Zweig, Kayla 
Smith, Robert Kuhn and Donna Karolchik. For a complete list of the individuals and organizations who
participated in this assembly, see the <a href="credits.html#mouse_credits">Credits</a> page.</p>

<a name="1202b05"></a>
<h2Dec. 2, 2005 &nbsp;&nbsp; Announcing the VisiGene Image Browser</h2> 
<p> 
We are pleased to announce the release of a new software tool in the Genome Browser collection, the 
<a href="../cgi-bin/hgVisiGene">VisiGene Image Browser</a>. VisiGene offers the ability to view 
<em>in situ</em> images, allowing examination of expression patterns at both the tissue and cellular
levels. The browser serves as a virtual microscope that lets viewers retrieve images that meet 
specific search criteria, then interactively zoom and scroll across the collection. The VisiGene 
Browser may be accessed from the menu on the UCSC Genome Bioinformatics home page or through a link 
on the details pages of Known Genes for which a VisiGene annotation exists.</p> 
<p> 
The initial release of VisiGene includes the following image collections:</p> 
<ul> 
  <li>
  Mouse <em>in situ</em> images from the <a href="http://www.informatics.jax.org/expression.shtml" 
  target="_blank">Mouse Genome Informatics Gene Expression Database</a> (GED)</li> 
  <li>
  Transcription factors in mouse embryos from the Mahoney Center for Neuro-Oncology</li> 
  <li>
  Mouse head and brain <em>in situ</em> images from NCBI's 
  <a href="https://www.ncbi.nlm.nih.gov/projects/gensat/" target="_blank">Gene Expression Nervous 
  System Atlas</a> (GENSAT) database</li>
  <li>
  <em>Xenopus laevis in situ</em> images from the <a href="http://www.nibb.ac.jp/en/" 
  target="_blank">National Institute for Basic Biology</a> (NIBB) XDB project</li>
</ul> 
<p> 
We hope to integrate images from the <a href="http://www.brainatlas.org/" target="_blank">Allen 
Brain Atlas</a> into the VisiGene collection within a few months.</p> 
<p> 
For more information about VisiGene, visit the 
<a href="../cgi-bin/hgVisiGene">VisiGene Gateway</a> page or consult the 
<a href="help/hgTracksHelp.html#VisiGeneHelp">Using the VisiGene Image Browser</a> section in the 
User's Guide.</p> 
<p> 
We'd like to thank the organizations listed above for permitting us to add their images to the 
VisiGene database. VisiGene was written by Jim Kent and Galt Barber of the UCSC Genome 
Bioinformatics Group. Contact 
<a href="mailto:&#107;e&#110;&#116;&#64;&#115;&#111;&#101;.&#117;&#99;&#115;c.&#101;&#100;u">Jim</a>
if you have an image set you'd like to contribute for display.</p> 

<a name="1202a05"></a>
<h2>Dec. 2, 2005 &nbsp;&nbsp; Allen Brain Atlas annotation added to Genome Browser</h2> 
<p> 
We have released a new annotation track &mdash; Allen Brain Atlas Probes &mdash; that may be found 
in the Expression and Regulation section of the latest mouse and human assemblies (mm6 and hg17). 
The <a href="http://www.brainatlas.org/" target="_blank">Allen Brain Atlas</a> (ABA) is an extensive
database of high resolution <em>in situ</em> hybridization images of adult male mouse brains 
covering the majority of genes. This track provides a link to the ABA images for each probe. For 
more information about the ABA, see the description page that accompanies this track.</p> 
<p> 
We'd like to thank the Allen Institute for Brain Science, and Susan Sunkin in particular, for 
coordinating with UCSC on this annotation.</p>

<a name="101705"></a>
<h2>Oct. 17, 2005 &nbsp;&nbsp; UCSC Genome Browser tutorial at ASHG 2005</h2> 
<p> 
OpenHelix will be presenting a one-hour seminar, &quot;Introduction to the UCSC Genome 
Browser&quot;, at the American Society of Human Genetics (ASHG) 2005 meeting on Oct. 26, 6:30-7:30 
p.m. The introductory tutorial will cover the topics needed to effectively use the Genome Browser, 
including basic search functionality and BLAT use, Table Browser use, creating and using Custom 
Tracks, and an introduction to the Gene Sorter. The class, which is open to all registered ASHG 
attendees, does not require programming skills, although a basic knowledge of genomic and biological
concepts is recommended.</p> 
<p> 
The free tutorial will be conducted at the Grand America Hotel, Audubon Room. Snacks and beverages 
will be served. Attendees will receive a free download of training materials. For further 
information, visit <a href="http://www.openhelix.com" target="_blank">www.openhelix.com</a> or call 
1-888-861-5051.</p>

<a name="101105"></a>
<h2>Oct. 11, 2005 &nbsp;&nbsp; <em>D. grimshawi</em> Browser now available</h2> 
<p> 
The UCSC Genome Bioinformatics Group has added the <em>Drosophila grimshawi</em> genome to the 
collection of fly genomes available in the UCSC Genome Browser. This assembly (UCSC version droGri1,
Aug. 2005) was produced by <a href ="http://www.agencourt.com/" target="_blank">Agencourt Bioscience
Corporation</a> in Beverly, MA, USA, using the Arachne assembler.</p> 
<p> 
The droGri1 assembly contains 25,052 scaffolds ranging in size from 196 bases to 14,170,260
bases.</p>  
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/droGri1/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#droGri">Downloads</a> page. Please review the data use guidelines outlined in the README.txt files that accompany the downloads. The data use 
restrictions are also available on the Genome Browser <a href="credits.html#droGri_use">Credits</a> 
page.</p>  
<p> 
We'd like to thank Agencourt Bioscience Corporation for providing this assembly. The UCSC droGri1 
browser was produced by Angie Hinrichs, Brian Raney, Jennifer Jackson, Kayla Smith and Donna 
Karolchik. The UCSC Genome Bioinformatics Group generated the initial set of annotation tracks. See 
the <a href="credits.html#droGri_credits">Credits</a> page for a detailed list of the organizations 
and individuals who contributed to this release.</p>

<a name="100705"></a>
<h2>Oct. 7, 2005 &nbsp;&nbsp; New <em>D. ananassae</em> Browser released</h2> 
<p> 
The UCSC Genome Bioinformatics Group has updated the <em>Drosophila ananassae</em> Genome Browser to
the 1 August, 2005 assembly. This version (UCSC version droAna2) was produced by 
<a href="http://www.agencourt.com/" target="_blank">Agencourt Bioscience Corporation</a> using the 
<a href="http://www.broad.mit.edu/wga/" target="_blank">Arachne assembler</a>.</p>  
<p> 
The assembly contains 13,772 scaffolds ranging in size from 55 bases to 23,697,768 bases, with a 
mean size of 16822.3 and median of 1537.</p> 
<p> 
Sequence and annotation data for the droAna2 assembly can be downloaded from the UCSC Genome Browser
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/droAna2">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#droAna">downloads</a> page. Please review the
data use guidelines outlined in the README.txt files that accompany the downloads. The data use 
restrictions are also available on the Genome Browser 
<a href="credits.html#droAna_credits">credits</a> page.</p> 
<p> 
We'd like to thank Agencourt Bioscience Corporation for this assembly. The UCSC droAna2 browser was 
produced by Angie Hinrichs, Brian Raney, Ann Zweig, Kayla Smith and Donna Karolchik. The UCSC Genome
Bioinformatics Group generated the initial set of annotation tracks. See the 
<a href="credits.html#droAna_credits">credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p> 

<a name="100405"></a>
<h2>Oct. 4, 2005 &nbsp;&nbsp; Updated Cow Browser available</h2> 
<p> 
The UCSC Genome Bioinformatics Group has released a Genome Browser and Blat server for the Mar. 
2005 Btau_2.0 draft assembly of the cow genome. This assembly (UCSC version bosTau2) was provided by
Baylor College of Medicine Human Genome Sequencing Center in Houston, TX.</p> 
<p> 
The Btau_2.0 release was assembled from whole genome shotgun (WGS) reads using the Atlas genome 
assembly system. Several WGS libraries, with inserts of 2 - 4 kb and 4 - 6 kb, were used to produce 
the data. Approximately 23 million reads were assembled, representing about 17.7 Gb of sequence and 
6.2x coverage of the (clonable) bovine genome. Highly repeated sequences and BAC clones sequences 
were omitted from this assembly; these will be placed in a subsequent version of the draft sequence.
The assembly contains chromosomes 1-29, X, M, and Bin0, as well as 98058 scaffolds. For details 
about changes UCSC made to the assembly format for display purposes, please see the bosTau2 Genome 
Browser <a href="../cgi-bin/hgGateway?db=bosTau2">gateway page</a>.</p>  
<p> 
More information on the Btau_2.0 assembly can be found on the Baylor 
<a href="http://www.hgsc.bcm.tmc.edu/projects/bovine/" target="_blank">Bovine Genome Project</a> web
page.</p>  
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/bosTau2/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#cow">Downloads</a> page. Please refer to the 
Baylor <a href="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
target="_blank">conditions of use</a> regarding these data. The bosTau2 annotation tracks were 
generated by UCSC and collaborators worldwide.</p>  
<p> 
We'd like to thank Baylor College of Medicine for the bovine sequence and assembly. We'd also like 
to acknowledge the members of the UCSC Genome Bioinformatics Group who contributed to the Cow Genome
Browser: Galt Barber, Brian Raney, Mark Diekhans, Jennifer Jackson and Donna Karolchik. See the 
<a href="credits.html#cow_credits">Credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p>

<a name="092005"></a>
<h2>Sep. 20, 2005 &nbsp;&nbsp; Three Drosophila assemblies released in Genome Browser</h2> 
<p> 
The UCSC Genome Bioinformatics Group has added three more Drosophila genomes to the collection of 
fly genomes in the UCSC Genome Browser. The new genomes, which include the initial assembly of 
<em>D.  erecta</em> (droEre1, Aug. 2005) and the updated assemblies of <em>D. virilis</em> (droVir2,
Aug. 2005) and <em>D.  mojavensis</em> (droMoj2, Aug. 2005), were produced by 
<a href ="http://www.agencourt.com/" target="_blank">Agencourt Bioscience Corporation</a> in 
Beverly, MA, USA. All three genomes were assembled using the Arachne assembler.</p> 
<p> 
The droEre1 assembly contains 5,124 scaffolds ranging in size from 154 bases to 26,647,023 bases, 
with a mean size of 29832.7 and median of 1740. The droVir2 assembly consists of 13,562 scaffolds 
ranging in size from 57 bases to 25,269,527 bases, with a mean size of 15263.1 and median of 1249.  
The droMoj2 assembly is comprised of 6,843 scaffolds ranging in size from 101 bases to 34,172,700 
bases, with a mean size of 28389.6 and median of 1671.</p> 
<p> Bulk downloads of the sequence and annotation data are available from the UCSC downloads
server:</p> 
<ul> 
  <li>
  droEre1 -  <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/droEre1">FTP</a>, 
  <a href="http://hgdownload.gi.ucsc.edu/goldenPath/droEre1">HTTP</a></li> 
  <li>
  droVir2 - <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/droVir2">FTP</a>, 
  <a href="http://hgdownload.gi.ucsc.edu/goldenPath/droVir2">HTTP</a</li>
  <li>
  droMoj2 - <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/droMoj2">FTP</a>, 
  <a href="http://hgdownload.gi.ucsc.edu/goldenPath/droMoj2">HTTP</a></li>
</ul> 
<p> 
Please review the data use guidelines outlined in the README.txt files that accompany the downloads.
The data use restrictions are also available on the Genome Browser 
<a href="credits.html">credits</a> page.</p> 
<p> 
We'd like to thank Agencourt Bioscience Corporation for providing these assemblies. The UCSC 
Drosophila browsers were produced by Angie Hinrichs, Brian Raney, Jennifer Jackson and Donna 
Karolchik. The UCSC Genome Bioinformatics Group generated the initial set of annotation tracks. 
See the <a href="credits.html">credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p>

<a name="090905"></a>
<h2>Sep. 9, 2005 &nbsp;&nbsp; <em>C. intestinalis</em> v2.0 Genome Browser released</h2>
<p> 
The v2.0 <em>C. intestinalis</em> draft assembly from the US DOE Joint Genome Institute (JGI) is now
available in the UCSC Genome Browser and Blat server (UCSC database ci2).</p> 
<p> 
This whole genome shotgun assembly was constructed with the JGI assembler, JAZZ, using paired-end 
sequencing reads. Starting with a coverage of 11x, additional data -- including BAC and FISH markers
-- were used to map scaffolds to chromosome arms. The size of this assembly, including unmapped 
scaffolds, is 173 Mb, with 94 Mb of the sequence mapped to chromosome arms.</p> 
<p> 
For more information about the ci2 assembly, see the JGI 
<a href="http://genome.jgi-psf.org/ciona4/ciona4.info.html" target="_blank"><em>C. intestinalis</em>
project page</a>. Additional information and an analysis of the euchromatic regions of this genome 
may be found in Dehal, P.  <em>et al.</em> 
<a href="https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=12481130&query_hl=1" 
target="_blank">The draft genome of Ciona intestinalis: insights into chordate and vertebrate 
origins</a>. <em>Science</em> <B>298</B>(5601), 2157-67 (2002).<p> 
<p>
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/ci2/" target="_blank">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#ciona" target="_blank">Downloads page</a>. 
The ci2 annotation tracks were generated by UCSC and collaborators worldwide. See the 
<a href="credits.html#ciona_credits">Credits page</a> for a detailed list of the organizations and 
individuals who contributed to this release.</p> 
<p> 
Many thanks to the JGI and their collaborators for providing the v2.0 sequence and annotations. 
The ci2 Genome Browser was produced by Brian Raney, Mark Diekhans, Ann Zweig, Kayla Smith, Robert 
Kuhn and Donna Karolchik.</p>

<a name="081105"></a>
<h2Aug. 11, 2005 &nbsp;&nbsp; Upcoming Genome Browser seminars - San Francisco, Seattle, Chicago, 
Boston, Philadelphia, Atlanta</h2> 
<p> 
The UCSC Bioinformatics Group announces six regional seminars and hands-on computer workshops on the
UCSC Genome Browser, presented by <a href="http://www.openhelix.com/" 
target="_blank">OpenHelix:</a></p> 
<ul> 
  <li> 
  San Francisco --  Monday, Aug. 22 
  <li> 
  Seattle --  Wednesday, 24 August 
  <li> 
  Chicago --  Tuesday, 27 September 
  <li> 
  Boston --  Wednesday, 28 September 
  <li> 
  Philadelphia --  Tuesday, 15 November 
  <li> Atlanta --  Wednesday, 16 November 
</ul> 
<p> 
These introductory sessions are geared towards anyone with a basic knowledge of genomic and 
biological concepts who is interested in learning how to use the UCSC Genome Browser. No programming
experience is required. The seminars will cover the topics necessary to learn how to effectively 
use the browser tool set, including basic Genome Browser functionality, searching and BLAT use, 
Table Browser use, creating and using custom annotation tracks, and an introduction to the Gene 
Sorter. Lectures will be accompanied by hands-on computer exercises conducted directly on the Genome
Browser web site.</p>  
<p>
For location and registration information, visit the 
<a href="http://www.openhelix.com/noteworthy.shtml" target="_blank">OpenHelix</a> website or call 
1-888-861-5051.  Academic and student discounts are available.</p>

<a name="061305"></a>
<h2>Jun. 13, 2005 &nbsp;&nbsp; <em>Drosophila simulans</em> Genome Browser released</h2> 
<p> 
The UCSC Genome Bioinformatics Group has released a Genome Browser for the <em>Drosophila 
simulans</em> draft genome sequence. The Release 1.0 assembly (UCSC version droSim1, Apr. 2005) was 
produced by the <a href="http://genome.wustl.edu/" target="_blank">Genome Sequencing Center</a> at 
the Washington University in St.  Louis (WUSTL) School of Medicine. Release 1.0 represents a 
composite of several different <em>D. simulans</em> lines: contigs from the w501 line comprise the 
primary scaffolding, with contigs and unplaced reads from six other lines used to fill gaps in the 
w501 assembly.</p> 
<p> 
The total size of this assembly, excluding the gapless 14,972 bp mitochondrial sequence, is 
142,405,747 bp including gaps and 127,241,461 bp excluding gaps. For more information about the 
<em>D. simulans</em> assembly and statistics, see the WUSTL Genome Sequencing Center 
<a href="http://genome.wustl.edu/genome.cgi?GENOME=Drosophila%20simulans%20White%20501" 
target="_blank"><em>Drosophila simulans</em></a> web page.</p> 
<p>
Downloads of the droSim1 data and annotations may be obtained from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/droSim1/" target="_blank">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#droSim" target="_blank">Downloads</a> page. 
The droSim1 annotation tracks were generated by UCSC and collaborators worldwide.</p> 
<p>
Thanks to the Genome Sequencing Center, WUSTL School of Medicine for providing this assembly. We'd 
also like to acknowledge the UCSC team who worked on this release: Angie Hinrichs, Jennifer Jackson,
Ali Sultan-Qurraie, Brian Raney and Donna Karolchik. See the 
<a href="credits.html#droSim_credits">Credits</a> page for a detailed list of the organizations and 
individuals who contributed to this release.</p>

<a name="053105"></a>
<h2>May 31, 2005 &nbsp;&nbsp; June Genome Browser seminars in San Diego, CA and Salt Lake
City, UT</h2> 
<p> 
The UCSC Bioinformatics Group announces two seminars and hands-on workshops on the UCSC Genome 
Browser, presented by <a href="http://www.openhelix.com/" target="_blank">OpenHelix</a>, a 
bioinformatics training, software testing and consulting company.</p> 
<p> 
These introductory sessions are geared towards anyone with a basic knowledge of genomic and 
biological concepts who is interested in learning how to use the UCSC Genome Browser. No programming
experience is required. The seminars will cover the topics necessary to learn how to effectively use
the browser tool set, including basic Genome Browser functionality, searching and BLAT use, Table 
Browser use, creating and using custom annotation tracks, and an introduction to the Gene Sorter. 
Lectures will be accompanied by hands-on computer exercises conducted directly on the Genome Browser
web site.</p>  
<p> 
The first three-hour course will be held on Thursday Jun. 9 in the Salt Lake City, UT area. Two 
sessions will be offered: 9am-12pm and 1pm-4pm.</p> 
<p> 
The second seminar will be held in San Diego, CA on Friday, Jun. 10. Two sessions will be offered: 
9am-12pm and 1pm-4pm.</p>  
<p> 
For registration information, visit the <a href="http://www.openhelix.com/noteworthy.shtml" 
target="_blank">OpenHelix</a> website or call 1-888-861-5051. Academic and student discounts are 
available.</p>

<a name="051805"></a>
<h2>May 18, 2005 &nbsp;&nbsp; Rhesus Monkey Genome Browser released</h2> 
<p> 
The UCSC Genome Bioinformatics group has released a Genome Browser for the rhesus monkey draft 
assembly, Mmul_0.1 (UCSC version rheMac1), produced by the Baylor College of Medicine Human Genome 
Sequencing Center in collaboration with J. Craig Venter Science Foundation Joint Technology Center 
and the Genome Sequencing Center at Washington University School of Medicine in St. Louis. This Old 
World monkey, which is a primate model organism, is important for the study of human biology and 
disease due to its genetic, physiologic and metabolic similarities to humans.</p> 
<p> 
Mmul_0.1 is a preliminary assembly using whole genome shotgun (WGS) reads from small and medium 
insert clones. Approximately 14.6 million reads were used in the assembly, representing about 
12.6 Gb of sequence and about 4.6x coverage of the (clonable) genome. The total length of all 
contigs is approximately 2.7 Gb, or 2.8 Gb including gaps between contigs. Highly-repeated sequences
and sequences from BAC clones were omitted from this assembly and will be placed in a subsequent 
version of the draft sequence.</p> 
<p> 
For more information about the rheMac1 assembly, see the Baylor 
<a href="http://www.hgsc.bcm.tmc.edu/projects/rmacaque/" target="_blank">Rhesus Monkey Project</a> 
web page.</p> 
<p> 
The rheMac1 sequence and annotation data can be downloaded from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/rheMac1/">FTP server</a> or the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#rhesus">Downloads</a> page. These data have 
specific <a href="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
target="_blank">conditions for use</a>.</p> 
<p> 
Many thanks to Baylor College of Medicine, the Rhesus Monkey Genome Sequencing Consortium, and the 
other institutions who contributed to the sequencing and mapping effort of the Mmul_0.1 release. The
UCSC Rhesus Genome Browser was produced by Robert Baertsch, Galt Barber, and Donna Karolchik. The 
initial set of rheMac1 annotation tracks was generated by the <a href="../staff.html">UCSC Genome 
Bioinformatics Group</a>. See the <a href="/goldenPath/credits.html#rhesus_credits">credits</a> 
page for a detailed list of the organizations and individuals who contributed to the release of this
browser.</p>

<a name="050405"></a>
<h2>May 5, 2005 &nbsp;&nbsp; New Mouse assembly available in Genome Browser</h2> 
<p> 
The latest mouse genome assembly from the Mouse Genome Sequencing Consortium, NCBI Build 34 (UCSC 
version mm6), is now available in the UCSC Genome Browser. In conjunction with this release, we have
archived mouse assemblies mm3 and mm4. All archived assemblies are available for download via our 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#mouse" target="_blank">downloads 
server</a>.</p> 
<p> 
The Build 34 assembly has an assembled length of approximately 2.6 Gb, of which about 1.9 Gb is 
finished sequence. Chromosomes 2, 4, 11 and X are finished in this build. To review in-depth 
statistics on the assembly, see the NCBI 
<a href="https://www.ncbi.nlm.nih.gov/genome/guide/mouse/release_notes.html"
target="_blank">Build 34 Data</a> web page. NOTE: In the Genome Browser mm6 assembly, chrY_random 
erroneously contains a region duplicated from chrY. For more information about this issue, see the 
Genome Browser <a href="../FAQ/FAQdownloads.html#download30">FAQ</a>. Please also note that the 
UCSC mm6 database contains only the reference strain C57BL/6J.</p> 
<p> 
The mm6 sequence and annotation data may be downloaded from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/mm6/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/goldenPath/mm6/">Downloads</a> web page. The mm6 annotation 
tracks were generated by UCSC and collaborators worldwide.</p> 
<p> 
Mirror sites, please note: due to the size of the comparative genomics annotations in this assembly,
the data set requires an unusually large amount of disk space. If you regularly download updates of 
the mouse data to your site, you may want to examine your available disk space before adding the 
annotation database from this assembly and/or reschedule your download for a period of low activity 
on your server. The size of the mm6 annotation database directory 
($WEBROOT/goldenPath/mm6/database/) is approximately 93 GB. Including the related net and chains, 
the total size is 124 GB.</p> 
<p> 
We'd like to thank Deanna Church and the Mouse Genome Sequencing Consortium for this assembly. We'd 
also like to acknowledge the work of the UCSC mm6 team: Hiram Clawson, Fan Hsu, Jennifer Jackson, 
Robert Kuhn, Ali Sultan-Qurraie, Heather Trumbower and Donna Karolchik. For a complete list of the 
individuals and organizations who participated in this assembly, see the 
<a href="credits.html#mouse_credits">Credits</a> page.</p>

<a name="050205"></a>
<h2>May 2, 2005 &nbsp;&nbsp; Archiving mm3 and mm4 assemblies</h2> 
<p> 
To make room for the NCBI Build 34 mouse assembly (mm6, Mar. 2005), we are archiving the mm3 and 
mm4 mouse assemblies. These assemblies will remain available for download on our 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#mouse">downloads server</a>, although blat 
will no longer be supported. We plan to release the mm6 assembly in the next day or so.</p>

<a name="032405"></a>
<h2>Mar. 24, 2005 &nbsp;&nbsp; Genome Browser tutorial at Experimental Biology 2005 on Apr. 5</h2> 
<p> 
OpenHelix will be presenting a one-hour introductory seminar on the UCSC Genome Browser at the 
Experimental Biology 2005 meeting on 5 April, 5:00-6:00. The tutorial will cover the topics needed 
to effectively use the Browser, including basic functionality, searching and BLAT use, Table Browser
use, creating and using Custom Tracks, and an introduction to the Gene Sorter.</p> 
<p> 
The free session will be conducted at the San Diego Convention Center, Room 12. For further 
information, visit the <a href="http://www.openhelix.com" target="_blank">OpenHelix website</a> or 
call 1-888-861-5051.</p> 
<p> 
OpenHelix will also be presenting brief introductory overviews of the UCSC Genome Browser in its 
Experimental Biology booth #133-135 during show hours. Stop by the booth for more information and 
to receive your free UCSC Quick Reference Card.</p> 

<a name="030205"></a>
<h2>Mar. 2, 2005 &nbsp;&nbsp; CCDS Data set released</h2> 
<p> 
The initial results from the Consensus Coding Sequence project (CCDS) were released to the public 
today. CCDS is a collaborative effort to identify a core set of human protein-coding regions that 
are consistently annotated and of high quality.</p> 
<p> 
The initial CCDS data set, containing nearly 15,000 transcripts, has been posted on three Internet 
sites: the <a href="../cgi-bin/hgTracks?db=hg17">UCSC Genome Browser</a>, the 
<a href="http://www.ensembl.org/" target="_blank">Ensembl Browser</a> and the 
<a href="https://www.ncbi.nlm.nih.gov/CCDS/" target="_blank">NCBI CCDS Database</a> website. The 
genes may be viewed on the UCSC hg17 (May 2004) Human Genome Browser in the CCDS annotation track 
located in the Genes and Gene Prediction Tracks section.</p> 
<p> 
The CCDS gene set is built by consensus among the following collaborating organizations:</p> 
<ul> 
  <li> 
  <a href="http://www.ebi.ac.uk/" target="_blank">European Bioinformatics Institute</a> (EBI)</li> 
  <li> 
  <a href ="https://www.ncbi.nlm.nih.gov/" target="_blank">National Center for Biotechnology 
  Information</a> (NCBI)</li> 
  <li> 
  <a href="http://www.cbse.ucsc.edu/" target="_blank">University of California Santa Cruz</a>
  (UCSC)</li> 
  <li> 
  <a href="http://www.sanger.ac.uk/" target="_blank">Wellcome Trust Sanger Institute</a> (WTSI)</li>
</ul> 
<p> 
Each gene in the CCDS set is assigned a unique identifier and version number, e.g. CCDS234.1. The 
version number will be updated if either the CDS structure or the underlying genome sequence at that
location changes. The CCDS set will be mapped forward, with identifiers preserved, when the 
annotations or sequence update. Changes to existing CCDS genes will be done by collaboration 
agreement; no single group will change the set unilaterally. CCDS identifiers are included on 
relevant NCBI RefSeq and Entrez Gene records.</p> 
<p> 
Gene structure information is obtained from both curated and automated genes sources. The primary 
curation groups are the <a href="http://www.sanger.ac.uk/HGP/havana/" target="_blank">Havana</a> 
team at the WTSI and the <a href="https://www.ncbi.nlm.nih.gov/RefSeq/" target="_blank">RefSeq</a> 
annotation group at NCBI. In addition, manually curated information on chr14 (Genoscope) and Chr7 
(WUSTL) has been included from the <a href="http://vega.sanger.ac.uk/" target="_blank">Vega</a> 
database. The <a href="http://www.ensembl.org/" target="_blank">Ensembl</a> group and the RefSeq 
computational pipeline provide the automated data. Curated information is favored over automated 
information; the information must be consistent between the Hinxton (Vega/Ensembl) and NCBI groups 
and also pass UCSC's stringent QC requirements.</p> 
<p> 
At a minimum, a gene must meet three quality criteria to be included in the CCDS set:</p> 
<ul> 
  <li>
  The coding region has a full-length annotation with an initiating ATG, a valid stop codon, and no 
  in-frame stop codons.</li> 
  <li> 
  Protein can be translated from the genome without frameshifts.</li> 
  <li> 
  Consensus splice-sites are used.</li>
</ul> 
<p> 
For more information on the process used to identify and evaluate CCDS genes, refer to the 
description page for the Genome Browser CCDS track. The number and type of quality tests performed 
may be expanded in the future, but includes analysis to identify putative pseudogenes, 
retrotransposed genes, consensus splice sites, supporting transcripts, and protein homology. The 
completeness of the CCDS gene set should improve as the curation and automated annotation processes 
mature and additional experimental validation is obtain for weakly-supported genes.</p> 
<p> 
UCSC would like to thank EBI, NCBI and WTSI for their collaboration in the release of these data. 
We would also like to acknowledge the efforts of the UCSC staff who worked on this project: Mark 
Diekhans, Robert Baertsch, Adam Siepel, Fan Hsu, and Robert Kuhn.</p>

<a name="022405"></a>
<h2>Feb. 24, 2005 &nbsp;&nbsp; Genome Browser training - St. Louis and NYC</h2> 
<p> 
We'd like to announce upcoming UCSC Genome Browser seminars and hands-on computer workshops in St. 
Louis, MO, and New York, NY,  presented by OpenHelix, the bioinformatics training, software testing 
and consulting company.</p> 
<p> 
These introductory tutorials will cover the topics needed to effectively use the browser, including 
basic functionality of Genome Browser searching and BLAT use, Table Browser use, creating and using 
Custom Tracks, and an introduction to the Gene Sorter.</p> 
<p> 
The St. Louis class will be held on Mar. 29, 2005 at the Object Computing, Inc. training center. 
Two sessions will be offered: 9 a.m. - 12 p.m.  and 1 - 4 p.m.</p> 
<p> 
The New York City class will be held on 30 March, 2005 at the 
<a href="http://www.thinkpathtraining.com/rawdoc.asp?Temp=7240&docID=7087" 
target="_blank">ThinkPath Training</a> facility. Two sessions will be offered: 9 a.m. - 12 p.m. and 
1 - 4 p.m.</p>  
<p> 
For more information about class locations, costs, and registration, visit the 
<a href="http://www.openhelix.com/noteworthy.shtml" target="_blank">OpenHelix</a> website or call 
1-888-861-5051. Academic, student, and early registration discounts are available.</p>

<a name="022305"></a>
<h2>Feb. 23, 2005 &nbsp;&nbsp; New Honeybee Genome Browser</h2> 
<p> 
The UCSC Genome Bioinformatics group has released a Genome Browser for a second honeybee assembly, 
Amel_2.0 (UCSC version apiMel2), produced by the Baylor College of Medicine Human Genome Sequencing 
Center.</p> 
<p> 
The assembly -- which is approximately 229 Mb in size including contig gaps -- was sequenced using 
a combined whole genome shotgun (WGS) and BAC clone approach. Overall sequence coverage is estimated
at 7.5x. In this assembly, scaffolds have been arranged into 17 linkage groups that are somewhat 
equivalent to chromosomes. For more information about the assembly, see the Baylor 
<a href="http://www.hgsc.bcm.tmc.edu/projects/honeybee/" target="_blank">honeybee project</a> 
website.</p>  
<p> 
The apiMel2 sequence and annotation data can be downloaded from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/apiMel2/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#apiMel">downloads</a> page. Please see the 
Baylor <a href="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
target="_blank">conditions of use</a> page for guidelines regarding the use of these data.</p> 
<p> 
Many thanks to Baylor College of Medicine and the other institutions who contributed to the 
sequencing and mapping effort of the Amel_v2.0 release. The UCSC <em>A. mellifera</em> Genome 
Browser was produced by Andy Pohl, Angie Hinrichs, Jennifer Jackson, and Donna Karolchik. The 
initial set of apiMel2 annotation tracks was generated by the UCSC Genome Bioinformatics Group. See 
the <a href="/goldenPath/credits.html#apiMel_credits">credits</a> page for a detailed list of the 
organizations and individuals who contributed to the release of this browser.</p>

<a name="021705"></a>
<h2>Feb. 17, 2005 &nbsp;&nbsp; Cow Genome Browser released</h2> 
<p> 
The UCSC Genome Bioinformatics Group has released a Genome Browser and Blat server for the Sep. 2004
Btau_1.0 draft assembly of the cow genome. This assembly (UCSC version bosTau1) was provided by 
Baylor College of Medicine Human Genome Sequencing Center in Houston, TX.</p> 
<p> 
The Btau_1.0 release was assembled from whole genome shotgun (WGS) reads using the Atlas genome 
assembly system. Several WGS libraries, with inserts of 2 - 4 kb and 4 - 6 kb, were used to produce 
the data. Approximately 15 million reads were assembled, representing about 9 Gb of sequence and 3x 
coverage of the (clonable) bovine genome. Highly repeated sequences and BAC clones sequences were 
omitted from this assembly; these will be placed in a subsequent version of the draft sequence. The 
assembly contains approximately 450,000 scaffolds.</p> 
<p> 
For more information on the Btau_1.0 assembly, see the Baylor 
<a href="http://www.hgsc.bcm.tmc.edu/projects/bovine/" target="_blank">Bovine Genome Project</a> web
page.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/bosTau1/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#cow">Downloads</a> page. Please refer to the 
Baylor <a href="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
target"_blank">conditions of use</a> regarding these data. The bosTau1 annotation tracks were 
generated by UCSC and collaborators worldwide.</p> 
<p> 
We'd like to thank Baylor College of Medicine for the bovine sequence and assembly. We'd also like 
to acknowledge the members of the UCSC Genome Bioinformatics Group who contributed to the Cow Genome
Browser: Heather Trumbower, Jim Kent, Hiram Clawson, Angie Hinrichs, Brian Raney, Mark Diekhans, 
Robert Kuhn, Ali Sultan-Qurraie, and Donna Karolchik. See the 
<a href="/goldenPath/credits.html#cow_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>

<a name="021505"></a>
<h2>Feb. 15, 2005 &nbsp;&nbsp; UniProt (Swiss-Prot/TrEMBL) display changes</h2>
<p> 
We have made several adjustments to the Genome Browser databases to accommodate recent changes to 
display IDs introduced by UniProt (aka Swiss-Prot/TrEMBL). If you are using the proteinID field in 
our knownGene table or the Swiss-Prot/TrEMBL display ID for indexing or cross-referencing other 
data, we strongly suggest you transition to the UniProt accession number. These changes will also 
affect anyone who is mirroring our site.</p>  
<p> 
To review a detailed list of the changes to the Genome Browser databases, see our 
<a href="../FAQ/FAQreleases.html#release13">FAQ</a>.</p>

<a name="021405"></a>
<h2>Feb. 14, 2005 &nbsp;&nbsp; <em>D. melanogaster</em> release 4 now available</h2> 
<p> 
The <em>D. melanogaster</em> Release 4 assembly (UCSC version dm2, Apr. 2004) is now available in 
the Genome Browser. This version includes Release 4 euchromatic sequence from the Berkeley 
Drosophila Genome Project (<a href="http://www.fruitfly.org/" target="_blank">BDGP</a>), Release 3.2
heterochromatic genome sequence from the Drosophila Heterochromatin Genome Project 
(<a href="http://www.fruitfly.org/" target="_blank">DHGP</a>), and Release 4.0 annotations (dated 
Nov. 2004) from <a href="http://flybase.net/" target="_blank">FlyBase</a>. Insects in the UCSC 
Genome Browser collection include six Drosophila genomes, the honeybee, and <em>Anopheles 
gambiae</em>.</p>
<p> 
The Release 4 euchromatic sequence, 118.4 Mb in size, contains six euchromatic chromosome arms with 
a total of 23 sequence gaps. The assembly was validated in collaboration with the Genome Sciences 
Centre at the British Columbia Cancer Agency in Vancouver, Canada, using fingerprint analysis of a 
tiling path of BACs spanning the genome.</p> 
<p> 
The Release 3.2 heterochromatic sequence is based on 20.7 Mb of Release 3 whole genome shotgun 
scaffolds from Celera that could not be assembled into the euchromatin arms, as well as a few 
BDGP-sequenced scaffolds. The scaffolds have been assigned to chromosomes where possible.</p> 
<p>     
For more information on this assembly, see the release notes for the 
<a href="http://www.fruitfly.org/annot/release4.html" target="_blank">Release 4 euchromatic 
sequence</a>.</p>  
<p> 
Downloads of the dm2 data and annotations may be obtained from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/dm2/" target="_blank">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#fruitfly">Downloads</a> page. The dm2 
annotation tracks were provided by FlyBase or generated by UCSC.</p> 
<p> 
Thanks to BDGP, DHGP, and the Flybase Consortium for providing the sequence, assembly, gene 
annotations and analysis of this genome. The UCSC Genome Browser team is Angie Hinrichs, Brian 
Raney, Galt Barber, and Donna Karolchik. See the 
<a href="/goldenPath/credits.html#dm_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p>

<a name="011805"></a>
<h2>Jan. 18, 2005 &nbsp;&nbsp; Opossum Genome Browser</h2> 
<p> 
The opossum (<em>Monodelphis domestica</em>) is the latest vertebrate addition to the UCSC Genome 
Browser genome collection. The Oct. 2004 preliminary assembly -- UCSC version monDom1 -- was 
sequenced and assembled by <a href="http://www.broad.mit.edu/seq/" target="_blank">The Broad 
Institute</a>, Cambridge, MA, USA.</p>  
<p> 
This preliminary draft was assembled from 33,507,069 placed reads, and consists of 109,065 contigs 
containing a total of 3,492,108,230 bases.  There are 3,563,247,205 total bases in the 19,348 
scaffolds in the assembly, with 71,138,975 bases in the gaps between contigs within the scaffolds. 
The scaffolds range in size from 1,000 bases to 22,286,839 bases.</p> 
<p> 
The monDom1 sequence and annotation data can be downloaded from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/monDom1/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#opossum">downloads</a> page. Please review 
the <a href="/goldenPath/credits.html#opossum_use" target="_blank">guidelines</a> for using these 
data.</p> 
<p> 
Many thanks to The Broad Institute for providing these data. The UCSC opossum Genome Browser was 
produced by Hiram Clawson, Galt Barber, Ali Sultan-Qurraie, Brian Raney and Donna Karolchik. The 
initial set of annotation tracks was generated by the UCSC Genome Bioinformatics Group. See the 
<a href="/goldenPath/credits.html#opossum_credits">credits</a> page for a detailed list of the 
organizations and individuals who contributed to the release of this browser.</p>

<!---------------- 2004 archived news --------------->
<a name="2004"></a>

<a name="122304"></a>
<h2>Dec. 23, 2004 &nbsp;&nbsp; Updated Zebrafish assembly (Zv4) now available</h2> 
<p> 
The latest zebrafish assembly -- Zv4 (UCSC version danRer2, June 2004) -- is now available in the 
UCSC Genome Browser and Blat server. The Zv4 assembly was produced by The Wellcome Trust Sanger 
Institute in collaboration with the Max Planck Institute for Developmental Biology in Tuebingen, 
Germany, and the Netherlands Institute for Developmental Biology (Hubrecht Laboratory), Utrecht, 
The Netherlands. This assembly consists of 1,560,480,686 bp in 21,333 scaffolds with a sequence 
coverage of approximately 5.7x.  443 Mb of the assembly is from 2,828 finished clones and 121 Mb 
is from 1,272 unfinished clones.</p>  
<p> 
75% of the sequence was tied to the fingerprint contig (FPC) map, which provided a template for 
placing the unfinished sequence.  The remaining sequence was filled with whole genome shotgun (WGS) 
contigs using a combination of sequence alignment and BAC end positions.  The WGS contigs used in 
this assembly were identical to those used for the Zv3 assembly, but the FPC data and its 
integration with the WGS data has been considerably improved.</p> 
<p> 
The Sanger Institute notes that there is high level of misassembly present in this release due to 
the large amount of polymorphism in the DNA source. Highly variable regions within the genome posed 
assembly difficulties, most likely because the sequences originated from different haplotypes. For 
more information about this assembly, see the Sanger Institute web page for the 
<a href="http://www.sanger.ac.uk/Projects/D_rerio/" target="_blank"><em>Danio rerio</em> Sequencing 
Project</a>.</p>  
<p> 
In addition to the assembly improvements noted above, a new zebrafish repeat library was used to 
mask repeats in the danRer2 Genome Browser, resulting in a cleaner set of alignments containing 
considerably less &quot;noise&quot;.</p>  
<p> 
The danRer2 sequence and annotation data can be downloaded from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/danRer2/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#zebrafish">downloads</a> page. Please review 
the <a href="http://www.sanger.ac.uk/Projects/use-policy.shtml" target="_blank">guidelines</a> for 
using these data.</p>  
<p> 
We'd like to thank the Wellcome Trust Sanger Institute, the Max Planck Institute for Developmental 
Biology, Hubrecht Laboratory and the other institutions who contributed to the sequencing and 
mapping effort of this release. Special thanks to the Zebrafish Genome Initiative at Children's 
Hospital in Boston for their collaboration on this release. The UCSC zebrafish Genome Browser was 
produced by Rachel Harte, Mark Diekhans, Heather Trumbower, Jennifer Jackson, and Donna Karolchik. 
The initial set of annotation tracks was generated by the UCSC Genome Bioinformatics Group. See the 
<a href="/goldenPath/credits.html#zebrafish_credits">credits</a> page for a detailed list of the 
organizations and individuals who contributed to the release of this browser.</p>

<a name="122204"></a>
<h2>Dec. 22, 2004 &nbsp;&nbsp;  UCSC releases Honeybee Browser</h2> 
<p> 
The UCSC Genome Bioinformatics group has released a Genome Browser for the Amel_1.2 honeybee 
assembly produced by the Baylor College of Medicine Human Genome Sequencing Center.</p> 
<p> 
The assembly -- which is approximately 213 Mb in size including contig gaps -- was sequenced using 
a combined whole genome shotgun (WGS) and BAC clone approach. The BAC clone sequence was produced 
following a pooled-array strategy. Overall sequence coverage is estimated at 6X. For more 
information about this assembly, see the Baylor 
<a href="http://www.hgsc.bcm.tmc.edu/projects/honeybee/" target="_blank">honeybee project</a> 
website.</p> 
<p> 
The honeybee genome is of interest to the agricultural community, in addition to its role as a model
organism in the study of several human health issues, including immunity, allergic reaction, 
antibiotic resistance, development, mental health, longevity, and diseases of the X chromosome. The 
honeybee is also interesting for its social and behavioral traits.</p> 
<p> 
The apiMel1 sequence and annotation data can be downloaded from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/apiMel1/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#apiMel">downloads</a> page. Please see the 
Baylor <a href="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
target="_blank">conditions of use</a> page for guidelines regarding the use of these data.</p> 
<p> 
Many thanks to Baylor, Children's Hospital Oakland Research Institute (CHORI), and the other 
institutions who contributed to the sequencing and mapping effort of the Amel_1.2 release. The UCSC 
<em>A. mellifera</em> Genome Browser was produced by Angie Hinrichs, Brian Raney, Robert Kuhn, and 
Donna Karolchik. The initial set of apiMel1 annotation tracks was generated by the UCSC Genome 
Bioinformatics Group. See the <a href="/goldenPath/credits.html#apiMel_credits">credits</a> page for
a detailed list of the organizations and individuals who contributed to the release of this
browser.</p>

<a name="121504"></a>
<h2>Dec. 15,2004 &nbsp;&nbsp; Three new Drosophila assemblies in Genome Browser</h2> 
<p> 
The UCSC Genome Bioinformatics Group has added three more Drosophila genomes to the UCSC Genome 
Browser. The new genomes  -- initial assemblies of <em>D. ananassae</em> (droAna1, Jul. 2004), 
<em>D. virilis</em> (droVir1, Jul. 2004), and <em>D.  mojavensis</em> (droMoj, Aug. 2004) -- join 
the <em>D.  melanogaster</em>, <em>D. yakuba</em>, and <em>D.  pseudoobscura</em> genomes already in
the collection.</p>  
<p>
The <em>D. virilis</em> and <em>D. mojavensis</em> genomes were sequenced and assembled by 
<a href="http://www.agencourt.com/" target="_blank">Agencourt Bioscience Corporation</a> using the 
Arachne assembler. <a href="http://www.tigr.org/" target="_blank">The Institute of Genomic 
Research</a> (TIGR) produced the <em>D. ananassae</em> assembly using the Celera assembler. For more
details about an individual genome, refer to the Genome Browser 
<a href="../cgi-bin/hgGateway?org=D.+ananassae">gateway</a> page associated with the assembly.</p> 
<p> 
In upcoming months, UCSC plans to provide several additions and enhancements to the Drosophila 
browsers: </p>
<ul> 
  <li> 
  additional Drosophila genomes, including <em>D. erecta</em> and <em>D. simulans</em></li> 
  updated versions of <em>D. melanogaster</em> and other genomes</li> 
  <li>
  multiple alignments of all fly assemblies from the 12-Drosophila project to 
  <em>D. melanogaster</em>
</ul> 
<p> 
Sequence and annotation data for all the Drosophila assemblies can be downloaded from the UCSC 
Genome Browser <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html">downloads</a> page. Please review the data 
use guidelines outlined in the README.txt files that accompany the downloads. The data use 
restrictions are also available on the Genome Browser <a href="/goldenPath/credits.html">credits</a>
page.</p> 
<p>
We'd like to thank Agencourt Bioscience Corporation, TIGR, and other institutions who contributed
to the sequencing and mapping effort for these assemblies. The UCSC Drosophila browsers were 
produced by Angie Hinrichs, Brian Raney, Heather Trumbower, Robert Kuhn, Galt Barber, Jennifer 
Jackson, Ali Sultan-Qurraie, and Donna Karolchik. The UCSC Genome Bioinformatics Group generated 
the initial set of annotation tracks. See the <a href="/goldenPath/credits.html">credits</a> page 
for a detailed list of the organizations and individuals who contributed to this release.</p>

<a name="120304"></a>
<h2>Dec. 3, 2004 &nbsp;&nbsp; SNP track for Build 35 (hg17) assembly released</h2> 
<p> 
UCSC has released the much-awaited SNP annotation track for the Build 35 (hg17) Human Genome 
Browser. The SNP data were obtained from dbSnp Build 123, the first dbSnp release based on the 
latest human assembly.</p>  
<p> 
In conjunction with the data release, we have made several improvements to the graphical display of 
the track. Our users now have the ability to filter the SNP data based on validation status, 
functional class, variant class, average heterozygosity and other features. Variants may be drawn in
user-selected colors to highlight additional details in the graphical image. Also, the browser now 
displays all SNP variants at the appropriate base width, rather than as generic ticks.</p>

<a name="111704"></a>
<h2>Nov. 17, 2004 &nbsp;&nbsp; <em>X. tropicalis</em> v.3.0 released in Genome Browser</h2> 
<p> 
The v.3.0 <em>Xenopus tropicalis</em> assembly is now available on the UCSC Genome Browser. This 
whole genome shotgun assembly (xenTro1, Oct. 2004) was provided by the U.S. DOE Joint Genome 
Institute (JGI).</p>
<p> 
Version 3.0 was constructed with the JGI assembler, Jazz, using paired end sequencing reads at a 
coverage of 7.4X.  After trimming for vector and quality, 19.1 million reads assembled into 27,064 
scaffolds totaling 1.63 Gbp. Roughly half the genome is contained in 392 scaffolds, each of which is
at least 1.2 Mb in length. For more information about this assembly, see the JGI 
<a href="http://genome.jgi-psf.org/Xentr3/Xentr3.home.html" target="_blank"><em>X. tropicalis</em> 
website</a>.</p>  
<p> 
The xenTro1 sequence and annotation data can be downloaded from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/xenTro1/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#xenTro">downloads</a> page. Please see the 
JGI <a href="http://genome.jgi-psf.org/Xentr3/Xentr3.download.html" target="_blank">data release 
policy</a> for data use	guidelines.</p>
<p> 
Many thanks to JGI and the other institutions who contributed to the sequencing and mapping effort 
of the v. 3.0 release. The UCSC <em>X.  tropicalis</em> Genome Browser was produced by Fan Hsu, Jim 
Kent, Heather Trumbower, Hiram Clawson, Brian Raney, Galt Barber, Mark Diekhans, Angie Hinrichs, and
Donna Karolchik. The initial set of xenTro1 annotation tracks was generated by the UCSC Genome 
Bioinformatics Group. See the <a href="/goldenPath/credits.html#xenTro_credits">credits</a> page for
a detailed list of the organizations and individuals who contributed to the release of this 
browser.</p>

<h2>Nov. 17,2004 &nbsp;&nbsp;  Genome Browser tutorial offered at ASCB meeting</h2> 
<p> 
OpenHelix will be presenting the tutorial &quot;Introduction to the UCSC Genome Browser&quot; at the
American Society for Cell Biology (ASCB) Annual Meeting in Washington, D.C. on Dec. 5, 8:30 - 10:00 
p.m. This introductory session will cover the topics needed to effectively use the Genome Browser, 
including basic functionality of Genome Browser searching and BLAT use, Table Browser use, creating 
and using Custom Tracks, and an introduction to the Gene Sorter.</p>  
<p> 
The free 1-1/2-hour tutorial will be conducted at the Washington Convention Center, Room 206. 
OpenHelix will also be presenting brief introductory overviews of the UCSC Genome Browser in its 
ASCB booth #510-512 during show hours.  Stop by the booth for more information and to receive your 
free UCSC Quick Reference Card.</p>  
<p> 
For more information about the tutorial and other services offered by OpenHelix, visit 
<a href="http://www.openhelix.com" target="_blank">www.openhelix.com</a> or call 1-888-861-5051.</p>

<a name="100804"></a>
<h2>Oct. 8, 2004 &nbsp;&nbsp; Updated Proteome Browser released</h2> 
<p> 
UCSC has released v1.1 of the UCSC Proteome Browser. This updated version includes the following 
major enhancements:</p> 
<ul> 
  <li>
  direct access to a Proteome Browser gateway via the &quot;Proteome Browser&quot; menu link on the 
  Genome Browser home page. The Proteome Browser gateway page prompts the user for a protein ID or 
  gene symbol, then directly starts up the Proteome Browser, eliminating the multiple Genome Browser
  steps required by the previous release.</li>
  <li>
  extended protein coverage that includes all proteins in the Swiss-Prot/TrEMBL databases, rather 
  than just the human, mouse and rat protein sets included in the previous version.</li> 
</ul> 
<p> 
In addition to providing direct access to the Proteome Browser, the v1.1 release preserves the 
existing tight coupling between the Proteome Browser and the Genome Browser for the human, mouse and
rat genomes. Users may still navigate between the Genome Browser Known Genes track and the Proteome 
Browser for quick, easy access to the details of a gene's genomic and proteomic sequence 
structures.</p> 
<p> 
The UCSC team who worked on the Proteome Browser update includes Fan Hsu, Robert Kuhn, Donna 
Karolchik and Tom Pringle. Please send comments or questions to our mailing list at 
<a href="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</a>. 
<!-- above address is genome at soe.ucsc.edu --> </p>

<a name="100504"></a>
<h2>Oct. 5, 2004 &nbsp;&nbsp; East Coast Genome Browser seminars - Nov. 2004</h2> 
<p> 
The UCSC Bioinformatics Group announces two seminars and hands-on workshops on the UCSC Genome 
Browser, presented by <a href="http://www.openhelix.com/" target="_blank">OpenHelix</a>, a 
bioinformatics training, software testing and consulting company.</p>  
<p> 
These introductory sessions are geared towards anyone with a basic knowledge of genomic and 
biological concepts who is interested in learning how to use the UCSC Genome Browser. No programming
experience is required. The seminars will cover the topics necessary to learn how to effectively use
the browser tool set, including basic Genome Browser functionality, searching and BLAT use, Table 
Browser use, creating and using custom annotation tracks, and an introduction to the Gene Sorter. 
Lectures will be accompanied by hands-on computer exercises conducted directly on the Genome Browser
web site.  <p> The first three-hour course will be held on Tuesday Nov. 9 in the Washington, 
D.C. area. Two sessions will be offered: 9am-12pm and 1pm-4pm.</p> 
<p> 
The second seminar will be held in the Raleigh/Durham, N.C. area on Wednesday, Nov. 10. Two 
sessions will be offered: 9am-12pm and 1pm-4pm.</p> 
<p> 
For registration information, visit the <a href="http://www.openhelix.com/noteworthy.shtml" 
target="_blank">OpenHelix</a> website or call 1-888-861-5051. Academic, student, and early 
registration discounts are available.</p>

<a name="100104"></a>
<h2>Oct. 1, 2004 &nbsp;&nbsp; UCSC releases new Table Browser</h2> 
<p> 
We are pleased to announce the release of an enhanced version of the 
<a href="../cgi-bin/hgTables">UCSC Table Browser</a>. This new release sports several improvements 
and additions over the previous Table Browser, including:</p> 
<ul> 
  <li> 
  a new streamlined user interface <li> support for generating filters that include fields from
  multiple tables, including those from non-positional tables</li> 
  <li> 
  an enhanced schema-viewing utility that displays all tables associated with a track, as well as 
  all tables linked to a selected table</li> 
  <li> 
  the ability to restrict queries to include only data from ENCODE regions</li> 
  <li> 
  the addition of a GALA output option</li> 
  <li> 
  an option to save large output results sets directly to a file rather than displaying them in the 
  Table Browser</li> 
</ul> 
<p> 
The preliminary <a href="/goldenPath/help/hgTablesHelp.html">User's Guide</a> will be enhanced in 
upcoming weeks to include examples of many common Table Browser queries generated by our users.</p> 
<p> 
The new version of the Table Browser was produced by Jim Kent, Donna Karolchik, Heather Trumbower, 
Hiram Clawson, and Robert Kuhn, and incorporates code from the previous version written by Angie 
Hinrichs. We'd like to thank Mary Mangan and others on the OpenHelix staff for their feedback on 
early versions of this software.  <p>
		    The older version of the Table Browser will remain available
		    for a limited time at 
		    http://genome.ucsc.edu/cgi-bin/hgText. Please send feedback and questions to our
		    mailing list at
                    <a href="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</a>.
                    <!-- above address is genome at soe.ucsc.edu -->
		    <p>
<HR>

<a name="091004"></a>
<h2>Sep. 10, 2004 &nbsp;&nbsp; <em>T. nigroviridis</em> genome now available in Genome Browser</h2> 
<p> 
The Genoscope v7 <em>Tetraodon nigroviridis</em> genome assembly is now available in the UCSC Genome
Browser and Blat server. This assembly, UCSC version tetNig1 dated Feb. 2004, is the result of a 
collaboration between <a href="http://www.genoscope.cns.fr/" target="_blank">Genoscope</a> and the 
<a href="http://www.broad.mit.edu/annotation/tetraodon/background.html" target="_blank">Broad 
Institute</a> of MIT and Harvard.</p>  
<p> 
The v7 assembly was constructed using the whole genome shotgun (WGS) approach, resulting in a 
sequence coverage of about 7.9X. The assembly contains 45,609 contigs and 25,773 scaffolds 
generated by the Arachne program and covers more than 90% of the genome. Additional linking data 
were used to build ultracontigs and to organize the assembly into chromosomes. Genoscope estimates 
the size of the Tetraodon genome to be about 385 Mb.</p> 
<p> 
Downloads of the tetNig1 data and annotations may be obtained from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/tetNig1/" target="_blank">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#tetraodon" target="_blank">Downloads</a> 
page.  These data have been freely provided by Genoscope before publication with 
<a href="/goldenPath/credits.html#tetraodon_credits" target="_blank">specific conditions for 
use</a>.  The initial set of annotation tracks were generated by Genoscope and the UCSC 
Bioinformatics Group based on data provided Genoscope. Tetraodon gene predictions generated by 
Genoscope using <a href="http://www.sanger.ac.uk/Software/analysis/GAZE/" target="_blank">GAZE</a> 
will be available in the Genome Browser within a few weeks.</p> 
<p> 
Many thanks to Genoscope and the Broad Institute of MIT and Harvard for this genome assembly. The 
UCSC team who produced this browser are Rachel Harte, Robert Kuhn, Donna Karolchik, and the Genome 
Browser sysadmin team. See the <a href="/goldenPath/credits.html#tetraodon_credits">Credits</a> page
for a detailed list of the organizations and individuals who contributed to this
release.</p> 

<a name="090104"></a>
<h2>Sep. 1,2004 &nbsp;&nbsp; Genome Browser released for <em>D. pseudoobscura</em> draft assembly</h2> 
<p> 
The UCSC Genome Bioinformatics Group has released a Genome Browser and Blat server for the 
<em>Drosophila pseudoobscura</em> Freeze 1 draft assembly (Aug. 2003). This assembly, UCSC version 
dp2, was produced by the Human Genome Sequencing Center (HGSC) at Baylor College of Medicine.</p> 
<p> 
Freeze 1 is a whole genome shotgun assembly produced using Baylor HGSC's assembly engine, Atlas. 
The assembly, which provides approximately 7x coverage of the euchromatic portion of the genome, 
contains 759 scaffolds. The scaffold N50 size is 1,018,646 bp. The total scaffold size for this 
assembly is 139.3 Mbp, with an average size of 184,465 bp. Due to an assembly error, four large 
scaffolds &quot;jumped&quot; chromosomes. These have been split into &quot;A&quot; and
&quot;B&quot; parts in the downloadable assembly files. See the gateway page for more 
information.</p> 
<p> 
Baylor HGSC has provided a 
<a href="ftp://ftp.hgsc.bcm.tmc.edu/pub/data/Dpseudoobscura/fasta/D_pseudo_freeze1_scaffold_to_chromosome_assignments_8_28_03.txt" 
target="_blank">putative chromosome assignment</a> for the majority of larger scaffolds (> 90% of 
unique sequence), based on conservation between the Muller elements.</p> 
<p>     
Downloads of the dp2 data and annotations may be obtained from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/dp2/" target="_blank">FTP server</a>  or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#dp">Downloads</a> page. The initial set of 
annotation tracks were generated by UCSC.</p> 
<p> 
Many thanks to the Baylor HGSC for providing the genome assembly data. The UCSC team who produced 
this browser are Angie Hinrichs, Heather Trumbower, Robert Kuhn, Donna Karolchik, and the Genome 
Browser sysadmin team. See the <a href="/goldenPath/credits.html#dp_credits">Credits</a> page for a 
detailed list of the organizations and individuals who contributed to this release.</p> 

<a name="083004"></a>
<h2>Aug. 30, 2004 &nbsp;&nbsp; Genome Browser seminars - Seattle and San Francisco</h2> 
<p> 
The UCSC Bioinformatics Group announces 2 seminars and hands-on workshops on the UCSC Genome 
Browser, presented by <a href="http://www.openhelix.com/" target="_blank">OpenHelix</a>, a 
bioinformatics training, software testing and consulting company.</p> 
<p> 
These introductory sessions are geared towards anyone with a basic knowledge of genomic and 
biological concepts who is interested in learning how to use the UCSC Genome Browser. No 
programming experience is required. The seminars will cover the topics necessary to learn how to 
effectively use the browser tool set, including basic Genome Browser functionality, searching and 
BLAT use, Table Browser use, creating and using custom annotation tracks, and an introduction to the
Gene Sorter. Lectures will be accompanied by hands-on computer exercises conducted directly on the 
Genome Browser web site.</p>  
<p> 
The first three-hour course will be held on Monday Oct. 4 in Kirkland, WA (Seattle area). Two 
sessions will be offered: 9:00 a.m. - noon or 1 - 4 p.m.</p>  
<p> 
The second seminar will be held in Cupertino (San Francisco Bay Area) on Tuesday, Oct. 5. Two 
sessions will be offered: 1 - 4 p.m. or 6 - 9 p.m.</p>  
<p> 
For registration information, visit the <a href="http://www.openhelix.com/noteworthy.shtml" 
target="_blank">OpenHelix</a> website or call 1-888-861-5051. Academic, student, and early 
registration discounts are available.</p> 

<a name="081004"></a>
<h2>Aug. 10, 2004 &nbsp;&nbsp; Genome Browser released for <em>A. gambiae</em> draft genome</h2> 
<p> 
The UCSC Genome Bioinformatics Group has released a Genome Browser and Blat server for the 
<em>Anopheles gambiae</em> v. MOZ2 draft genome sequence (Feb. 2003). This assembly -- UCSC version 
anoGam1 -- was produced by the International Anopheles Genome Project and downloaded from
Ensembl.</p>  
<p> 
The MOZ2 assembly is a 10x whole genome shotgun assembly. The assembled portion of the genome is 
about 278 Mbp in length with a total of 8,987 unique scaffolds, the largest scaffold being 23.1 Mbp.
Approximately 85% of the sequence has been assigned to chromosomal locations. Chromosome arms chr2L,
chr2R, chr3L, chr3R, and chrX are represented by 13, 49, 42, 28, and 10 large scaffolds 
respectively. No scaffolds have yet been assigned to the Y chromosome. The unassigned scaffolds, 
concatenated together in arbitrary order, can be found in the artificial unknown 
&quot;chromosome&quot; chrUn.</p>  
<p>     
For more information about the initial <em>A. gambaie</em> assembly, see Holt et al. (2002), 
<a href="http://www.sciencemag.org/cgi/content/abstract/298/5591/129" target="_blank">The Genome 
Sequence of the Malaria Mosquito Anopheles gambiae</a>, <em>Science</em> 2002 298:129-149.</p>  
<p>     
Downloads of the anoGam1 data and annotations may be obtained from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/anoGam1/" target="_blank">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#anoGam">Downloads</a> page. The anoGam1 
annotation tracks were generated by UCSC and collaborators worldwide.</p> 
<p> 
Many thanks to the International Anopheles Genome Project and Ensembl for providing the sequence of 
this genome. The UCSC team who produced this browser are Angie Hinrichs, Galt Barber, Donna 
Karolchik, and sysadmins Paul Tatarsky and Jorge Garcia. See the 
<a href="/goldenPath/credits.html#anoGam_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to this release.</p> 

<a name="072904"></a>
<h2>Jul. 29, 2004 &nbsp;&nbsp; Zebrafish Genome assembly Zv3 now available in Genome Browser</h2> 
<p> 
The Zv3 Zebrafish genome assembly (UCSC version danRer1) is now available on the UCSC Genome Browser
and Blat server. This assembly was produced by The Wellcome Trust Sanger Institute, Hinxton, UK, in 
collaboration with the Max Planck Institute for Developmental Biology in Tuebingen, Germany, and the
Netherlands Institute for Developmental Biology (Hubrecht Laboratory), Utrecht, The
Netherlands.</p> 
<p> 
The Zv3 assembly consists of 1,459,115,486 bp in 58,339 supercontigs, with a sequence coverage of 
approximately 5.7X. This zebrafish assembly is the first to be tied to the FPC map: 1,083,447,588 
bp (74%) of the sequence were mapped in this way. Please note that this is a preliminary assembly; 
a high level of misassembly is present due to polymorphisms in the DNA source.</p>  
<p> 
For more information about this assembly, see the Sanger Institute's  
<a href="http://www.sanger.ac.uk/Projects/D_rerio/" target="_blank"><em>Danio rerio</em> Sequencing 
Project</a> web page. UCSC plans to release the Zv4 version of the zebrafish assembly on the Genome 
Browser in Fall '04.</p> 
<p> 
Downloads of the Zebrafish data and annotations can be obtained from the UCSC 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/danRer1/" target="_blank">FTP site</a> or 
<a href=http://hgdownload.gi.ucsc.edu/downloads.html#zebrafish>Downloads page</a>. The danRer1 
annotation tracks were generated by UCSC and collaborators worldwide. See the 
<a href="/goldenPath/credits.html#zebrafish_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to the success of this release.</p> 
<p> 
We'd like to thank The Wellcome Trust Sanger Institute and their collaborators for providing this 
assembly. A special thanks to Yi Zhou, Anthony DiBiase and Leonard Zon from the Children's Hospital 
in Boston, MA, USA for their collaboration on this release. The UCSC Zebrafish Genome Browser team 
is Rachel Harte, Heather Trumbower, and Donna Karolchik.</p>

<a name="072304b"></a>
<h2>Jul. 23, 2004 &nbsp;&nbsp; Latest Human Genome reference sequence now in Browser</h2> 
<p> 
The latest human genome reference sequence (NCBI Build 35, May 2004) is now available as database 
hg17 in the UCSC Genome Browser and Blat server. This sequence was obtained from NCBI and was 
produced by the International Human Genome Sequencing Consortium.</p> 
<p> 
Bulk downloads of the data are available via FTP at 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg17">ftp://hgdownload.gi.ucsc.edu/goldenPath/hg17</a> 
or through the Downloads link on this page. We recommend that you use FTP rather than HTML for the 
download of large or multiple files.</p> 
<p> 
We'd like to thank NCBI and the International Human Genome Sequencing Consortium for furnishing the 
data, and the UCSC team members who contributed to this release: Hiram Clawson, Terry Furey, Heather
Trumbower, Robert Kuhn, Donna Karolchik, Kate Rosenbloom, Angie Hinrichs, Rachel Harte, Jim Kent and
our sysadmin team Patrick Gavin, Jorge Garcia, and Paul Tatarsky.</p>

<a name="072304a"></a>
<h2>Jul. 23, 2004 &nbsp;&nbsp; <em>D. yakuba</em> added to Genome Browser</h2> 
<p> 
The UCSC Genome Bioinformatics Group has released a Genome Browser and Blat server on a second 
species of fruitfly, <em>D. yakuba</em>. The April 2004 Release 1.0 of this genome (UCSC version 
droYak1) was sequenced and assembled by the Genome Sequencing Center, Washington University (WUSTL) 
School of Medicine in St. Louis.</p>  
<p> 
<em>D. yakuba</em> is closely related to the model organism, <em>D. melanogaster</em>, with which it
shared a common ancestor approximately 10 million years ago. The <em>D. yakuba</em> genome is 
largely alignable to the <em>D. melanogaster</em> genome, but differs sufficiently to offer an 
interesting study of sequence divergence between the two species. <em>D. yakuba</em> occupies a 
critical intermediate position among several Drosophila species that will facilitate evolutionary 
studies among the fruitflies. For information about the <em>D. yakuba</em> assembly and statistics, 
see the WUSTL Genome Sequencing Center 
<a href="http://genome.wustl.edu/genome.cgi?GENOME=Drosophila%20yakuba&GROUP=6" 
target="_blank"><em>Drosophila yakuba</em></a> web page.</p>  
<p> 
Downloads of the droYak1 data and annotations can be obtained from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/droYak1/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#droYak">Downloads</a> page. The droYak1 
annotation tracks were generated by UCSC and collaborators worldwide.</p> 
<p>
Thanks to the Genome Sequencing Center at WUSTL School of Medicine for providing the sequence and 
assembly of this genome. The UCSC <em>D. yakuba</em> Genome Browser was produced by Angie Hinrichs, 
Michael Chalup, and Donna Karolchik. See the 
<a href="/goldenPath/credits.html#droYak_credits">Credits</a> page for a detailed list of the 
organizations and individuals who contributed to the success of this release.</p>

<a name="071604"></a>
<h2>Jul. 16, 2004 &nbsp;&nbsp; Genome Browser released for mouse assembly NCBI Build 33</h2> 
<p> 
The latest mouse assembly -- Build 33 from NCBI (UCSC version mm5) -- is now available via the UCSC 
Genome Browser and Blat server. This assembly includes approximately 2.6 gigabases of sequence. 
Chromosome 11 is finished in Build 33; the Sanger Institute has provided a corresponding agp file. 
The whole genome N50 for this assembly is 22.3 Mb, in comparison to 17.7 Mb for the previous build. 
Please note: the UCSC mm5 assembly contains only the reference strain C57BL/6J.</p> 
<p> 
This assembly is a composite version in which phase 3 High Throughput Genome Sequence (HTGS) was 
merged with the Mouse Genome Sequencing Consortium v3 Whole Genome Shotgun Assembly (MGSCv3). The 
assembly was performed by NCBI using a &quot;combined&quot; tiling path that was created 
automatically for the most part, but was manually curated in places. This facilitated the placement 
of finished sequence in the context of the MGSCv3 assembly. Draft sequence was not included in this 
build: the slight increase in coverage gained by using this would have been offset by the increase 
in build errors.</p> 
<p> 
More information about Build 33 will be available soon in the NCBI 
<a href="https://www.ncbi.nlm.nih.gov/genome/seq/NCBIContigInfo.html" target="_blank">assembly 
notes</a> and 
<a href="https://www.ncbi.nlm.nih.gov/mapview/stats/BuildStats.cgi?taxid=10090&build=33&ver=1" 
target="_blank">Build 33 statistics</a>.</p>  
<p> 
The mm5 sequence and annotation data may be downloaded from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/mm5/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/goldenPath/mm5/">Downloads</a> web page. The mm5 annotation 
tracks were generated by UCSC and collaborators worldwide.</p> 
<p> 
We'd like to thank Deanna Church, Richa Agrawala, and the Mouse Genome Sequencing Consortium for 
this assembly. We'd also like to acknowledge the work of the UCSC mm5 team: Fan Hsu, Hiram Clawson, 
Angie Hinrichs, Heather Trumbower, Mark Diekhans, Donna Karolchik and our systems administrators 
Jorge Garcia, Patrick Gavin and Paul Tatarsky.</p>

<a name="071504"></a>
<h2>Jul. 15, 2004 &nbsp;&nbsp; <em>C. intestinalis</em> v1.0 released in Genome Browser</h2> 
<p> 
The v1.0 <em>C. intestinalis</em> draft assembly from the US DOE Joint Genome Institute is now 
available for study using the UCSC Genome Browser and Blat server (UCSC database ci1).</p> 
<p> 
The whole genome shotgun assembly was constructed with the JGI assembler (JAZZ) paired-end 
sequencing reads at a coverage of 8.2X. The draft contains 116.7 million bp of nonrepetitive 
sequence in 2,501 scaffolds greater than 3 kb. 60 Mbp of this has been assembled into 117 scaffolds 
longer than 190 Kbp, and 85% of the assembly (104.1 Mbp) is found in 905 scaffolds longer than 20 
kb. The assembly, gene modeling and analysis were performed at the JGI.</p> 
<p> 
For more information about the ci1 assembly, see the JGI 
<a href="http://genome.jgi-psf.org/ciona4/ciona4.info.html" target="_blank"><em>C. intestinalis</em>
project page</a>. Additional information and an analysis of the euchromatic regions of this genome 
may be found in Dehal et al., <a href="http://www.sciencemag.org/cgi/content/full/298/5601/2157" 
target="_blank">The Draft Genome of Ciona intestinalis: Insights into Chordate and Vertebrate 
Origins</a>. Science.  2002 Dec 13;298(5601):2157-67.</p> 
<p> 
Bulk downloads of the sequence and annotation data are available via the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/ci1/" target="_blank">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#ciona" target="_blank">Downloads page</a>. 
The ci1 annotation tracks were generated by UCSC and collaborators worldwide. See the Credits page 
for a detailed list of the organizations and individuals who contributed to this release.</p> 
<p> 
Many thanks to the JGI and their collaborators for providing the v1.0 sequence and annotations. The 
ci1 Genome Browser was produced by Brian Raney, Galt Barber, Heather Trumbower, Robert Kuhn, Donna 
Karolchik and the Genome Browser sysadmin team - Patrick Gavin, Jorge Garcia, and Paul Tatarsky. 
We'd also like to thank Tom Pringle for his technical input and Mark Diekhans for his work on the 
incremental updates for this release.</p>

<a name="071404"></a>
<h2>Jul. 14, 2004 &nbsp;&nbsp; Genome Browser released for dog draft assembly v1.0</h2> 
<p> 
UCSC has released a Genome Browser and Blat server on the July 2004  v1.0 dog genome sequenced and 
assembled by the Broad Institute of MIT and Harvard and Agencourt Bioscience. The whole genome 
shotgun (WGS) sequence is based on 7.6X coverage of the dog genome, assuming a WGS assembly size of 
2.4 Gb. The assembly has an N50 contig length of 123 kb and an N50 supercontig length of
41.6 Mb.</p>  
<p> 
The dog genome, which contains approximately 2.5 billion base pairs, is similar in size to the 
genomes of humans and other mammals. The boxer breed was selected for the initial sequencing effort,
based on the lower variation rate in its genome relative to other breeds. In addition to the boxer, 
samples from nine other dog breeds, four wolves and a coyote are being used to generate an initial 
set of single nucleotide polymorphisms (SNPs) to facilitate disease studies. The SNPs should be 
available soon from <a href="https://www.ncbi.nlm.nih.gov/SNP/" target="_blank">dbSNP</a>.</p> 
<p> 
For more information about the dog draft assembly, see the 
<a href="http://www.genome.gov/12511476" target="_blank">NHGRI press release</a>.</p> 
<p> 
The dog sequence and annotation data can be downloaded from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/canFam1/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#dog">downloads</a> page. These data have 
<a href="/goldenPath/credits.html#dog_use">specific conditions for use</a>.</p>  
<p> 
Many thanks to the Broad Institute of MIT and Harvard, NHGRI, Agencourt Bioscience, Children's 
Hospital Oakland Research Institute, Centre National de la Recherche Scientifique, North Carolina 
State University, and Fred Hutchinson Cancer Research Center for their contributions to the 
sequencing, assembly, and mapping efforts. The initial canFam1 annotation track set, generated by 
the UCSC Genome Bioinformatics Group, will soon be supplemented by annotations from collaborators 
worldwide. See the <a href="/goldenPath/credits.html#dog_credits">credits</a> page for a detailed 
list of the organizations and individuals who contributed to the success of this release.</p>

<a name="062404"></a>
<h2>Jun. 24, 2004 &nbsp;&nbsp; ENCODE project featured in Genome Browser</h2> 
<p> 
We'd like to announce the release of UCSC Genome Browser features tailored to the ENCODE project 
community, including an <a href="/ENCODE/">ENCODE-specific page</a> to highlight the ENCODE 
contributors and their work, guidelines for data submission, and a list of specific links to ENCODE 
regions in the Genome Browser.</p> 
<p> 
The initial resources include sequences for the current human assemblies (hg16, hg15, hg13, and 
hg12), sequence of the comparative species from <a href="http://www.nisc.nih.gov" 
target="_blank">NISC</a>, tools for coordinate conversion between human assemblies, format 
descriptions for data submission, and contact information for help with submitting annotation data 
and analyses. Bulk downloads of the sequence and annotations may be obtained from the ENCODE Project
<a href="../ENCODE/downloads.html">Downloads</a> page.</p> 
<p> 
We'd like to thank NHGRI for their support of this project and the various contributors of 
annotations and analyses.</p>

<a name="061004"></a>
<h2>Jun. 10, 2004 &nbsp;&nbsp; Genome Browser training workshop - Boston, MA</h2> 
<p> 
The UCSC Bioinformatics Group announces a seminar and hands-on workshop on the UCSC Genome Browser, 
presented by <a href="http://www.openhelix.com/" target="_blank">OpenHelix</a>, a bioinformatics 
training, software testing and consulting company.</p> 
<p> 
This introductory session is geared towards anyone with a basic knowledge of genomic and biological 
concepts who is interested in learning how to use the UCSC Genome Browser. No programming experience
is required. The seminar will cover the topics necessary to learn how to effectively use the browser
tool set, including basic Genome Browser functionality, searching and BLAT use, Table Browser use, 
creating and using custom annotation tracks, and an introduction to the Gene Sorter. The lecture 
will be accompanied by hands-on computer exercises conducted directly on the Genome Browser web 
site.</p> 
<p> 
The three-hour course will be held at Tufts University School of Medicine, 145 Harrison Street, 
Boston, MA, on Tuesday, Aug. 10.  Two sessions will be offered: 1 - 4 p.m. or 6 - 9 p.m. For 
registration information, visit the <a href="http://www.openhelix.com/noteworthy.shtml" 
target="_blank">OpenHelix</a> website or call 1-888-861-5051. Academic, student, and early 
registration discounts are available.</p>

<a name="052804"></a>
<h2>May 28, 2004 &nbsp;&nbsp; UCSC Family Browser name changed to UCSC Gene Sorter</h2> 
<p> 
What's in a name? In an effort to clarify the role of the UCSC Family Browser, we have changed its 
name to the UCSC Gene Sorter. We think this name better describes this tool, which lets the user 
collect information on groups of genes that may be related in many different ways. The Gene Sorter 
provides a wealth of information on gene expression, protein homology (both within and across 
species), GO terms, and Pfam domains, cross links to many other databases, and much more.</p> 
<p> 
If you haven't already tried this tool, we encourage you to give it a spin. You'll find it at 
<a href="../cgi-bin/hgNear">http://genome.ucsc.edu/cgi-bin/hgNear</a>, or click the
&quot;Gene Sorter&quot; link on any Genome Browser menu bar.</p>

<a name="052704"></a>
<h2>May 27, 2004 &nbsp;&nbsp; New Genome Browser downloads server</h2> 
<p> 
As a follow-up to last week's FTP site switch, we are changing the location of the UCSC Genome 
Browser downloads site to 
<a href="http://hgdownload.gi.ucsc.edu/">http://hgdownload.gi.ucsc.edu/</a>. All downloadable 
files currently located in http://genome.ucsc.edu/goldenPath will be moved to the new server. 
Please make a note of the new URL and update any references to it. Users accessing downloads through
the Genome Browser <a href="http://hgdownload.gi.ucsc.edu/downloads.html">Downloads page</a> will 
be redirected automatically to the new location.</p>

<a name="051904"></a>
<h2>May 19, 2004 &nbsp;&nbsp; New Genome Browser FTP server</h2> 
<p> 
We have changed the URL for the UCSC Genome Browser ftp site to 
<a href="ftp://hgdownload.gi.ucsc.edu/">ftp://hgdownload.gi.ucsc.edu/</a>. This replaces the old 
URL of ftp://genome.ucsc.edu/. The old URL will be disabled within a few days. Please make a note of
the new URL and update any references to it.</p>

<a name="051104"></a>
<h2>May 11, 2004 &nbsp;&nbsp; <em>C. elegans</em> WS120 added to Genome Browser</h2> 
<p> 
UCSC has released a Genome Browser and Blat server on an updated version of the <em>C. elegans</em> 
genome. The March 2004 assembly -- UCSC version ce2 -- is based on sequence version WS120 deposited 
into <a href="http://www.wormbase.org/species/c_elegans" target="_blank">WormBase</a> as of Mar. 1, 
2004. This assembly has a finishing error rate of 1:10,000.</p> 
<p> 
The ce2 sequence and annotation data may be downloaded from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/ce2/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#c_elegans">Downloads</a> web page. The ce2 
annotation tracks were generated by UCSC and collaborators worldwide.</p> 
<p> 
We'd like to thank the <a href="http://genome.wustl.edu/" target="_blank">Genome Sequencing 
Center</a> at Washington University in St. Louis and the 
<a href="http://www.sanger.ac.uk/" target="_blank">Sanger Institute</a> for their collaborative work
in sequencing the <em>C. elegans</em> genome. Many thanks to the 
<a href="http://www.wormbase.org/wiki/index.php/WormBase_Consortium" 
target="_blank">WormBase consortium</a> for making the worm sequence publicly available. We'd also 
like to acknowledge the UCSC team who contributed to this release: Rachel Harte (lead engineer), 
Hiram Clawson (WABA and miRNA annotations), Mike Chalup (QA), Galt Barber (QA), Heather Trumbower 
(QA), and Donna Karolchik (documentation).</p> 

<a name="042204"></a>
<h2>Apr. 22, 2004 &nbsp;&nbsp; Mouse Genome assembly added to Proteome Browser</h2> 
<p> 
Proteome Browser functionality is now available on the Oct. 2003 mouse genome assembly (mm4). 
Protein information may be viewed for most genes in the Known Genes track by clicking the Proteome 
Browser link on the gene's details page. For more information on the UCSC Proteome Browser, see the 
news release dated <a href="newsarch.html#031004">10 March 2004</a>. 
<p> 
In conjunction with this release, the Known Genes and Gene Family Browser protein data have been 
updated to the Swiss-Prot version dated 15 March 2004.</p>

<a name="041604"></a>
<h2>Apr. 16, 2004 &nbsp;&nbsp; Genome Browser training - Silicon Valley area</h2> 
<p> 
The UCSC Bioinformatics Group announces a seminar and hands-on workshop on the UCSC Genome Browser, 
presented by <a href="http://www.openhelix.com/" target="_blank">OpenHelix</a>, a bioinformatics 
training, software testing and consulting company.</p>  
<p> 
This introductory session is geared towards industry and academic biologists engaged in genomics 
research. No programming experience is required. The seminar will cover the topics necessary to 
learn how to effectively use the browser tool set, including basic Genome Browser functionality, 
searching and BLAT use, Table Browser use, creating and using Custom Tracks, and an introduction to 
the Family Browser. The lecture will be accompanied by hands-on computer exercises conducted 
directly on the Genome Browser web site.</p>  
<p> 
The three-hour course will be held at the UCSC Extension Campus computer lab in Cupertino,
CA on Thursday, May 6 6-9 p.m. For pricing information or to reserve a seat in the class, visit the 
<a href="http://www.openhelix.com/noteworthy.shtml" target="_blank">OpenHelix</a> web site or call 
1-888-861-5051. Pre-registration is required. Academic, student, and early registration discounts 
are available.</p>

<a name="041204"></a>
<h2>Apr. 12, 2004 &nbsp;&nbsp; GNF Gene Expression Atlas 2 data now available</h2> 
<p> 
Expression data from the GNF Gene Expression Atlas 2 are now available on the July 2003 human genome
assembly on the UCSC web site. The data may be viewed graphically in the 
<a href="../cgi-bin/hgNear?db=hg16&near.order=expGnfAtlas2&near.col.gnfHumanAtlas2Id.vis=on">Family 
Browser</a> or via the GNF Atlas 2 track in the Genome Browser. The track data contain 2 replicates 
each of 61 mouse tissues and 79 human tissues run over Affymetrix microarrays.</p> 
<p> 
We'd like to thank the Genomics Institute of the Novartis Research Foundation (GNF) for providing 
the expression data underlying the browser displays. More information on the data will be available 
in the paper Su et al.  &quot;A gene atlas of the mouse and human protein-encoding transcriptomes&quot; 
(in press - <em>PNAS</em>).</p>

<a name="031004"></a>
<h2>Mar. 10, 2004 &nbsp;&nbsp;  Announcing the UCSC Proteome Browser</h2> 
<p> 
We are proud to announce a new addition to the UCSC family of genome browsing and analysis tools.
The UCSC Proteome Browser presents a rich set of useful protein properties as well as links to 
several protein and genomic data sources on the Web. For the first time, Genome Browser users can 
have both the genome and proteome worlds at their fingertips simultaneously. The browser is 
accessible from the Genome Browser via the &quot;Proteome Browser&quot; link on the details page of 
any gene in the Known Genes track. The initial release is available only on Human Build 34 (hg16); 
Proteome Browsers for the latest mouse and rat assemblies will follow.</p> 
<p> 
For each protein, the browser displays the corresponding genomic exon structure and its amino acid 
sequence. Several protein property tracks are aligned to the sequence to help a user pinpoint regions
of interest.  Additional properties are plotted with histograms against genome-wide protein data to 
highlight significant trends and anomalies.</p> 
<p> 
The Proteome Browser is tightly coupled with the UCSC Genome Browser and UCSC Gene Family Browser, 
allowing easy navigation among the tools. For example, clicking on an exon in the Proteome Browser 
tracks display brings up the Genome Browser tracks page showing the genomic region of the exon 
together with a wealth of relevant data. Similarly, clicking on the Proteome Browser's 
&quot;Family Browser&quot; link displays related gene family information.</p> 
<p> 
The v1.0 release of the browser offers a variety of data tracks, including amino acid and DNA 
sequence, exon boundaries, hydrophobicity, polarity, cysteine and predicted glycosylation sites, 
Superfamily/SCOP domains, and amino acid anomalies. In addition, the browser includes histograms of 
several properties on a genome-wide scale: pI, molecular weight, exon count, number of cysteines, 
InterPro domain counts, hydrophobicity, amino acid frequencies and anomalies.</p> 
<p> 
The Proteome Browser also provides links to a variety of external sites containing supplementary 
information on the protein, including SwissProt, InterPro and Pfam domains, 3-D structures at PDB 
and UCSF ModBase, and pathway maps of KEGG, BioCarta (CGAP), and BioCyc.</p> 
<p> 
We'd like to thank SwissProt for sharing their high quality protein data and the pI calculation 
algorithm, as well as the other external data sites linked to by the Proteome Browser. We'd also 
like to acknowledge the hard work of Fan Hsu, lead engineer on the project, and Jim Kent, Tom 
Pringle, Donna Karolchik, and Robert Kuhn. The project received technical input, review and support 
from several other members of the UCSC Bioinformatics group.</p>

<a name="030104"></a>
<h2>Mar. 1, 2004 - Chicken draft genome added to Genome Browser</h2> 
<p> 
We've added the chicken genome to the collection of assemblies available in the UCSC Genome Browser 
and Blat Server. The Feb. 2004 assembly (UCSC version galGal2) was produced by the Genome Sequencing
Center at the Washington University School of Medicine in St. Louis. The source of this sequence 
was a female inbred Red Jungle Fowl (<em>Gallus gallus</em>), the ancestor of domestic chickens. 
The chicken genome is the first of the avian genomes to be sequenced.</p>  
<p> 
The genome has been sequenced to 6.63X coverage. Approximately 88% of the sequence has been 
anchored to chromosomes, which include autosomes 1-24, 26-28, and 32, and sex chromosomes W and Z. 
(In contrast to mammals, the female chicken is heterogametic (ZW) and the male is homogametic (ZZ).)
The remaining unanchored contigs have been concatenated into the virtual chromosome 
&quot;chrUn&quot;, separated by gaps of 10,000 bp. The chicken mitochondrial sequence is
also available as the virtual chromosome &quot;chrM&quot;.</p>  
<p> 
Washington University School of Medicine in St. Louis created the physical map for this release. 
Genetic mapping and linkage analysis were produced through a collaborative effort led by Martien 
Groenen at Wageningen University in the Netherlands.  SNP data based on three strains of domestic 
chickens will soon be available in GenBank from an international team of scientists led by the 
Beijing Genomics Institute in China and supported by the Wellcome Trust in Britain.</p> 
<p> 
The chicken is considered to be the premier non-mammalian vertebrate model organism. It plays an 
important role in the research of viruses and cancer, and is a primary model for the study of 
embryology and development. From an evolutionary standpoint, the chicken's position provides a good 
intermediate data point between mouse and fugu. Comparative genomics analyses between the chicken 
and other sequenced organisms should yield valuable information on the evolution of gene order and 
arrangement, thus improving our understanding of the structure and function of genes. /<p> 
<p> 
To facilitate comparative genomics studies, alignments of the chicken sequence to the human genome 
will be available in the Genome Browser later this week. Downloads of the comparative data are 
currently available through the Downloads page (see below).</p> 
<p> 
For more information about the release of the chicken genome assembly, see the NHGRI 
<a href="http://www.nhgri.nih.gov/11510730" target="_blank">press release</a>. Additional background
on the rationale behind the chicken genome sequencing effort can be found in the  
<a href="http://www.genome.wustl.edu/ancillary/data/whitepapers/Gallus_gallus_WP.pdf"
target="_blank">sequencing proposal</a>.</p>  
<p> 
Bulk downloads of the chicken sequence and annotations may be obtained from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/galGal2/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#chicken">Downloads</a> page. These data 
have <a href="/goldenPath/credits.html#chicken_use">specific conditions for use</a>.</p> 
<p> 
We'd like to thank the Genome Sequencing Center at the Washington University School of Medicine in 
St. Louis, Wageningen University, and the Chicken Mapping Consortium for providing these data. 
The chicken browser annotation tracks were generated by UCSC and collaborators worldwide. See the 
<a href="/goldenPath/credits.html#chicken_credits">Credits</a> page for a detailed list of 
acknowledgements. The UCSC Chicken Genome Browser was produced by Angie Hinrichs, Heather Trumbower,
Rachel Harte, and Donna Karolchik.</p>

<a name="022304"></a>
<h2>Feb. 23, 2004 &nbsp;&nbsp; Chimpanzee genome browser released</h2> 
<p> 
We are happy to announce the release of a Genome Browser and Blat server for the chimpanzee 
(<em>Pan troglodytes</em>). The 13 Nov. 2003 Arachne assembly -- labeled Chimp Build 1 Version 1 
(UCSC version panTro1) -- was produced by the Chimpanzee Genome Sequencing Consortium.</p> 
<p> 
This assembly covers about 95 percent of the genome and is based on 4X sequence coverage. It is 
composed of 361,782 contigs with an N50 length of 15.7 kb, and 37,849 supercontigs having an N50 
length of 8.6 Mb (not including gaps). The total contig length is 2.73 Gb, spanning 3.02 Gb. The DNA
donor for this genome assembly, &quot;Clint&quot;, is a captive-born West African chimpanzee 
(<em>Pan troglodytes verus</em>). 
<a href="http://www.genome.gov/Pages/Research/Sequencing/SeqProposals/ChimpGenome2.pdf" 
target="_blank">Background information</a> on the chimp genome sequencing project and the 
<a href="http://www.genome.gov/11509418" target="_blank">initial news release</a> about the chimp 
assembly can be found on the NHGRI website.</p>  
<p> 
Research has indicated that the human and chimp genomes probably differ by approximately one
percent. Because of this close relationship between chimpanzees and humans, the assembly should 
facilitate comparative analyses of the two genomes that have not been possible with other species 
that have been sequenced to date.</p>  
<p> 
The initial release of the Chimp Browser provides several annotation tracks comparing the chimp and 
human genomes.  More comparative annotations will be added in upcoming weeks.</p> 
<p> 
Bulk downloads of the chimp sequence and annotations may be obtained from the Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/panTro1/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#chimp">Downloads</a> page. The complete set 
of sequence reads is available at the <a href="https://www.ncbi.nlm.nih.gov/Traces" 
target="_blank">NCBI trace archive</a>. Blat searches on chrUn_random are not supported in the 
initial release, but will be available soon.</p>  
<p> 
We'd like to thank NHGRI, the Broad Institute at MIT/Harvard, and Washington University at St. Louis
School of Medicine for providing this sequence, and LaDeana Hillier, Washington University School of
Medicine, and the Broad Institute for their work on the alignments. The chimpanzee browser 
annotation tracks were generated by UCSC and collaborators worldwide.</p> 
<p> 
The UCSC team who worked on this release consisted of Kate Rosenbloom, Jim Kent, Hiram Clawson, 
Heather Trumbower, Robert Kuhn, Donna Karolchik, and the Genome Browser sysadmin team.</p>

<a name="021204"></a>
<h2>Feb. 12, 2004 &nbsp;&nbsp; New UCSC supported mirror site</h2> 
<p> 
The Genome Browser project now has a UCSC-supported mirror site that may be used during power or 
network outages on the UCSC campus. The mirror is located at the Medical College of Wisconsin in 
Milwaukee. The site will be updated regularly by UCSC with the latest data and software to closely 
replicate the main Genome Browser site at <a href="../index.html">http://genome.ucsc.edu</a>. 
Please continue to use the UCSC-based site for routine Genome Browser and Blat access.</p> 
<p> 
We'd like to thank the Department of Physiology at the Medical College of Wisconsin -- and in 
particular Jeff Nie and Greg McQuestion -- for their resources and collaboration on this project. 
We'd also like to acknowledge the hard work of UCSC's Paul Tatarsky, who invested many hours in 
arranging the collaboration and setting up the mirror.</p>

<a name="011604"></a>
<h2>Jan. 16, 2004 &nbsp;&nbsp; Eight hg16 tables updated</h2> 
<p> 
We've discovered a handful of hg16 chrN_random_gap and chrN_random_gold tables on our public server 
that are out of date. We have replaced the following tables with updated versions:</p> 
<ul> 
  <li>
  chr4_random_gap</li> 
  <li>
  chr4_random_gold</li> 
  <li>
  chr8_random_gap</li> 
  <li>
  chr8_random_gold</li> 
  <li>
  chrX_random_gap</li> 
  <li>
  chrX_random_gold</li> 
  <li>
  chrUn_random_gap</li> 
  <li>
  chrUn_random_gold</li> 
</ul> 
<p> 
Many thanks to Grigoriy Kryukov for discovering this problem. We apologize for any inconvenience 
this may have caused to our users.</p>

<a name="011404"></a>
<h2>Jan. 14, 2004 &nbsp;&nbsp; Yeast Genome Browser released</h2> 
<p> 
We are proud to add yeast (<em>S. cerevisiae</em>) to our growing list of genome assemblies. The 
study of brewer's yeast, the most basic eukaryotic model system, has led to important discoveries 
in a wide variety of areas, including metabolism, centromeres, recombination, cell division control,
meiosis and splicing.</p>  
<p> 
This assembly (UCSC version sacCer1) is based on sequence dated 1 Oct. 2003 in the 
<a href="http://www.yeastgenome.org/" target="_blank">Saccharomyces Genome Database</a> (SGD). The 
sequence, open reading frame (ORF), and gene annotations were downloaded from the site 
<a href="ftp://genome-ftp.stanford.edu/pub/yeast/data_download" 
target="_blank">ftp://genome-ftp.stanford.edu/pub/yeast/data_download</a>.</p>  
<p> 
The S288C strain was used in this sequencing project. Reference information for each chromosome may 
be found in the SGD <a href="http://www.yeastgenome.org/chromosomes.shtml" 
target="_blank">Systematic Sequencing Table</a>. For more information about the yeast genetic and 
physical maps, see the paper Cherry JM et al. 
<a href="http://db.yeastgenome.org/cgi-bin/SGD/reference/reference.pl?refNo=22828" 
target="_blank">Genetic and physical maps of Saccharomyces cerevisiae</a>. <em>Nature</em> 1997 
387(6632 Suppl):67-73.</p>  
<p> 
Downloads of the yeast data and annotations may be obtained from the UCSC Genome Browser 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/sacCer1/">FTP server</a> or 
<a href="http://hgdownload.gi.ucsc.edu/downloads#yeast">Downloads</a> page.</p> 
<p> 
We'd like to thank Stanford University, the SGD, the University of California San Francisco (UCSF), 
Washington University in St. Louis, and the Broad Institute at MIT/Harvard for providing the data 
and annotations for this assembly. We'd also like to acknowledge the UCSC team who worked on this 
release: Jim Kent, Heather Trumbower, Robert Kuhn, Donna Karolchik, and our sysadmin team.</p>

<!---------------- 2003 archived news --------------->
<a name="2003"></a>

<a name="121003"></a>
<h2>Dec. 10, 2003 &nbsp;&nbsp; UCSC Releases alignments of the chimp draft assembly to human genome</h2> 
<p> 
UCSC has released alignments of the Nov. 2003 chimpanzee draft assembly to the Jul. 2003 human 
assembly in the Genome Browser. These alignments may be viewed on the 
<a href="../cgi-bin/hgTracks?org=Human&db=hg16&position=chr4:56214201-56291736">Human Jul. 2003</a> 
assembly. This release coincides with today's <a href="http://www.nhgri.nih.gov/11509418" 
target="_blank">announcement</a> by the National Human Genome Research Institute (NHGRI) of the 
first draft assembly of the chimpanzee genome.</p> 
<p> 
The set of human/chimpanzee alignments consists of a reciprocal best-in-genome net track and a 
chimp chain track. These alignments were generated using the blastz program developed at 
Pennsylvania State University and the programs blat, axtChain, chainNet, and netSyntenic developed 
at UCSC by Jim Kent. Research scientists should find these tracks useful for locating orthologous 
regions and studying genome rearrangement in the two species.</p> 
<p> 
For more information about the alignment tracks, refer to the track description pages. The tables 
may be downloaded from the Genome Browser FTP server's 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg16/database/">hg16 database directory</a>. The 
chimp sequence and alignment data are downloadable from the 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg16/vsPt0/">hg16 human/chimp alignments 
directory</a>.</p>  
<p> 
The chimp sequence used in these alignments was obtained from the Nov. 13, 2003 Arachne assembly. 
We'd like to thank NHGRI, the Eli & Edythe L. Broad Institute at MIT/Harvard, and Washington 
University School of Medicine for providing this sequence, and LaDeana Hillier, Washington 
University School of Medicine, and the Whitehead Institute for their work on the alignments. We'd 
also like to acknowledge the members of the UCSC team who contributed to the release of these 
alignments in the Genome Browser: Jim Kent, Kate Rosenbloom, Heather Trumbower, and Donna 
Karolchik.</p>

<a name="112403"></a>
<h2>Nov. 24, 2003 &nbsp;&nbsp; Genome Browser released for mouse assembly NCBI Build 32</h2> 
<p> 
We have released a Genome Browser and Blat server for the latest mouse genome assembly, NCBI Build 
32 (UCSC v. mm4).  Build 32 is a composite assembly in which chromosomes were assembled by two 
slightly different algorithms depending on the available mapping data. Chromosomes 2, 4, 5, 7, 11, 
15, 18, 19, X, and Y were assembled using a clone-based tiling path file (TPF) provided by the Mouse
Genome Sequencing Consortium (MGSC), with whole genome shotgun sequence used to fill gaps when 
necessary. The remaining chromosomes were assembled using the MGSCv3 whole genome shotgun assembly 
as the TPF and merging High Throughput Genomic Sequence (HTGS) as needed. The UCSC mm4 assembly 
contains only the reference strain C57BL/6J.</p> 
<p> 
Build 32 includes 2.6 gigabases of sequence, 1.2 Gb of which is finished. We estimate that 90-96 
percent of the mouse genome is present in the assembly. For more information about this version, 
see the NCBI <a href="https://www.ncbi.nlm.nih.gov/genome/seq/NCBIContigInfo.html" 
target="_blank">assembly notes</a> and 
<a href="https://www.ncbi.nlm.nih.gov/mapview/stats/BuildStats.cgi?taxid=10090&build=32&ver=1" 
target="_blank">Build 32 statistics</a>.</p>  
<p> 
The mm4 sequence and annotation data may be downloaded from the UCSC Genome Browser 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html#mouse">downloads server</a>.</p>  
<p>
We'd like to thank the Deanna Church, Richa Agrawala, and the Mouse Genome Sequencing Consortium for
this assembly. We'd also like to acknowledge the work of the UCSC mm4 team: Hiram Clawson (lead), 
Terry Furey, Kate Rosenbloom, Heather Trumbower, Bob Kuhn and Donna Karolchik, and our systems 
administrators Patrick Gavin, Jorge Garcia and Paul Tatarsky.</p>

<a name="103103"></a>
<h2>Oct. 31, 2003 &nbsp;&nbsp; <em>D. melanogaster</em> assembly added to Genome Browser</h2> 
<p> 
We have added the <em>Drosophila melanogaster</em> (fruitfly) assembly to the growing collection of 
genomes available in the UCSC Genome Browser and Blat servers. Release 3.1 (Jan. 2003) of the 
<em>Drosophila</em> annotated genome sequence was provided by the 
<a href="http://www.fruitfly.org/" target="_blank">Berkeley Drosophila Genome Project</a> (BDGP). 
The 116.8 Mb euchromatic sequence - which is virtually gap-free and of high accuracy - contains six 
euchromatic chromosome arms represented by 13 scaffolds with a total of 37 sequence gaps. The 
sequence quality of this release has an estimated error rate of less than one in 100,000 base pairs 
in the unique portion of the sequence, and less than one in 10,000 base pairs in the repetitive 
portion. The Release 3.1 sequence was reannotated using the 
<a href="http://www.fruitfly.org/annot/apollo/" target="_blank">Apollo Genome Annotation and 
Curation Tool</a>. We also provide data comparing the genome of <em>D. melanogaster</em> with that 
of <em>D. pseudoobscura</em>.</p> 
<p> 
The fruitfly, one of the first organisms to be used in systematic scientific investigations, has 
been the subject of intensive study in genetics for nearly a century and remains a major model 
organism in biomedical research, population biology and evolution. We are pleased to add the 
fruitfly to the roster of assemblies available on our site.</p> 
<p>
Downloads of the <em>Drosophila</em> data and annotations may be obtained from the UCSC Genome 
Browser <a href ="ftp://hgdownload.gi.ucsc.edu/goldenPath/dm1">ftp site</a>.</p> 
<p> 
We'd like to thank BDGP and the <a href = "http://flybase.bio.indiana.edu">Flybase</a> Consortium 
(<a href="http://www.harvard.edu/" target="_blank">Harvard University</a>, 
<a href="http://www.cam.ac.uk/" target="_blank">University of Cambridge</a>, 
<a href="http://www.iub.edu/" target="_blank">Indiana University</a>, the 
<a href="http://www.berkeley.edu/" target="_blank">University of California, Berkeley</a> and the 
<a href="http://www.ebi.ac.uk/index.html" target="_blank">European Bioinformatics Institute</a> 
(EBI)) for providing the sequence, assembly, and analysis of this genome. We'd also like to 
acknowledge the members of the UCSC Genome Bioinformatics group who contributed to this release: 
Angie Hinrichs (lead engineer), Heather Trumbower, Robert Kuhn, Donna Karolchik, and Jim Kent and 
the system administrators Jorge Garcia, Patrick Gavin and Paul Tatarsky.</p>

<a name="101703b"></a>
<h2>Oct. 17, 2003 &nbsp;&nbsp; mRNA, RefSeq and EST data now updated incrementally for several 
assemblies</h2> 
<p> 
Daily and weekly incremental updates of mRNA, RefSeq, and EST data are now in place for several of 
the UCSC Genome Browser assemblies. Data sets that are updated incrementally from GenBank include 
the latest human (hg16), mouse (mm3), rat (rn3), and Fugu (fr1). Others will soon be added to the 
list.</p> 
<p> 
Previously, these tables were updated only when we loaded a new genome assembly into the Genome 
Browser or made a major revision to a table. By updating the data on a nightly basis, we are able to
provide researchers with the most current versions available in GenBank. All new genome assemblies 
released after this date will incorporate the incremental update technology.</p> 
<p> 
Data are updated on the following schedule:<br> 
- native and xeno mRNA and refSeq tracks - updated daily at 4:30 p.m. Pacific Time (weekdays), early
Saturday morning (weekends)<br> 
- EST data - updated once per week on Saturday morning<br> 
- downloadable data files - updated weekly on Sunday morning<br> 
- outdated sequences - removed once per quarter<br></p> 
<p> 
Mirror sites are not required to migrate to an incremental update process, and should not experience
problems as a result of this upgrade. Mirror site questions should be addressed to 
<a href="mailto:&#103;e&#110;&#111;m&#101;&#45;&#109;&#105;&#114;&#114;o&#114;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;&#117;">&#103;e&#110;&#111;m&#101;&#45;&#109;&#105;&#114;&#114;o&#114;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;&#117;</a>. 
<!-- above address is genome-mirror at soe.ucsc.edu --> 
<p> 
We'd like to acknowledge the hard work of Mark Diekhans in the implementation of this new feature, 
and thank the QA and sysadmin teams (particularly Paul Tatarsky) for their support in this
release.</p>

<a name="101703a"></a>
<h2>Oct. 17, 2003 &nbsp;&nbsp; UCSC Table Browser enhancements</h2> 
<p> 
The UCSC <a href="../cgi-bin/hgTables">Table Browser</a> is an excellent tool for retrieving and 
searching the data underlying the Genome Browser. We've recently added some new features to the 
Table Browser to make it even easier to query and download data.</p> 
<p> 
Many of our users have requested a batch query utility that will allow them to paste in or upload a 
list of terms on which to search. You can now do this by clicking the &quot;Item
name/accession&quot; button, then uploading a list of search terms by selecting the
&quot;Paste in&quot; or &quot;Upload&quot; option. Note that the Paste option supports wildcards, 
but the Upload option does not.</p> 
<p> 
We've also added 2 new lists of searchable tables/tracks. The Browser tracks list contains the 
names of annotation tracks in the currently selected assembly as they appear in the Genome Browser. 
This list is useful if you don't know the name of the underlying database table that contains the 
data in which you're interested. The Custom tracks list contains the names of all custom annotation 
tables currently loaded into the Genome Browser for the given assembly. This includes tracks that 
have been created/loaded by the user as well as custom annotations created via the Table 
Browser.</p> 
<p> 
If you have feedback or questions about the Table Browser, please send us email at 
<a href="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</a>.  
<!-- above address is genome at soe.ucsc.edu --></p>

<a name="092903"></a>
<h2>Sep. 29, 2003 &nbsp;&nbsp; Introducing the UCSC Gene Family Browser</h2> 
<p> 
We are pleased to announce the release of the <a href="../cgi-bin/hgNear" target="_blank">UCSC Gene 
Family Browser</a>. This major new addition to our website is a useful tool for collecting 
information on groups of genes that may be related in many different ways. The Browser provides 
information on gene expression, protein homology (both within and across species), GO terms, and 
Pfam domains, and cross links to many other databases.</p>  
<p> 
To access the Family Browser, click the link on the top menu on this page. The first time you use 
the Browser, the application will display a brief overview of the tool and information for starting 
and configuring it. To read a more detailed description of the Browser, see the Family Browser 
<a href="/goldenPath/help/hgNearHelp.html" target="_blank">User's Guide</a>.</p>  
<p> 
We are always interested in hearing feedback about the tools on our site. If you have comments or 
questions about the Family Browser, please email us at 
<a href="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</a>. 
<!-- above address is genome at soe.ucsc.edu --></p>

<a name="092303"></a>
<h2>Sep. 23, 2003 &nbsp;&nbsp; Fugu genome added to Genome Browser</h2> 
<p>
We have added the <em>Takifugu rubripes</em> (Japanese pufferfish) assembly to the growing 
collection of genomes available in the UCSC Genome Browser and Blat servers. The Fugu v.3.0 (Aug. 
2002) whole genome shotgun assembly -- which is the fourth vertebrate assembly to be added to the 
UCSC Genome Browser -- was provided by the <a href="http://genome.jgi-psf.org/fugu6/fugu6.info.html"
target="_blank">US DOE Joint Genome Institute</a> (JGI) as part of the International Fugu Genome 
Consortium led by JGI and the <a href="http://www.fugu-sg.org" target="_blank">Singapore Institute 
of Molecular and Cell Biology (IMCB)</a>.</p>  
<p>
This assembly was constructed with the JGI assembler, JAZZ, from paired end sequencing reads 
produced by JGI and IMCB, at JGI, Myriad Genetics, and Celera Genomics, resulting in a sequence 
coverage of 5.7X. All reads are plasmid, cosmid, or BAC end sequences, with the predominant coverage
derived from 2 Kb insert plasmids. This assembly contains 20,379 scaffolds totaling 319 million 
base pairs. The largest 679 scaffolds total 160 million base pairs. The Fugu genome was annotated 
using the Ensembl system by the Fugu informatics group at IMCB.</p> 
<p>
The Fugu, which was one of the first vertebrate genomes to be draft-sequenced after human, serves 
an important role in the exploration of the human genome. In contrast to other vertebrates that have
been sequenced, the intergenic and intron regions of the Fugu are highly compressed and uncluttered 
with repetitive sequence, resulting in a genome that is unusually compact in size. The Fugu genome 
has proved useful in gene discovery and the identification and characterization of gene regulatory 
elements in other genomes.</p> 
<p>
Bulk downloads of the Fugu sequence and annotation data are available via FTP at 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/fr1">ftp://hgdownload.gi.ucsc.edu/goldenPath/fr1</a> 
or through the Downloads link on the Genome Browser home page. We recommend that FTP be used rather 
than HTML for the download of large or multiple files.</p> 
<p>
We'd like to thank JGI and the other members of the International Fugu Genome Consortium, including 
IMCB, the UK Human Genome Mapping Project (Hinxton), the Molecular Sciences Institute (Berkeley) and
the Institute for Systems Biology (Seattle), for providing the sequence, assembly, and analysis of 
this genome. We'd also like to acknowledge the members of the UCSC Genome Bioinformatics group who 
contributed to this release: Kate Rosenbloom (lead engineer), Heather Trumbower, Robert Kuhn, Donna 
Karolchik, and Jim Kent.</p>

<a name="081303"></a>
<h2>Aug. 13, 2003 &nbsp;&nbsp; NISC Comparative Sequencing Program regions released in
Genome Browser</h2> 
<p> 
The UCSC Genome Bioinformatics group has released a browser and blat server on the first of more 
than 100 targeted genomic regions being sequenced in multiple species and analyzed by the NIH 
Intramural Sequencing Center (NISC) Comparative Sequencing Program sponsored by NHGRI. This release 
coincides with the publication of the results of the study in the 14 Aug 2003 issue of 
<a href="http://www.nature.com/cgi-taf/DynaPage.taf?file=/nature/journal/v424/n6950/abs/nature01858_fs.html" 
target="_blank"><em>Nature</em></a> (Thomas, J.W. et. al. (2003) Comparative analyses of 
multi-species sequences from targeted genomic regions. <em>Nature</em> 424:788- 793). The browser 
displays sequence and annotations on a large region containing 10 previously identified genes - 
including the gene mutated in cystic fibrosis - in 13 vertebrate species. Organisms in the study 
include human, chimpanzee, baboon, cat, dog, cow, pig, rat, mouse, chicken, zebrafish and two 
species of pufferfish (Fugu and Tetraodon).</p>  
<p> 
The NISC Comparative Sequencing Program data may be accessed by clicking the Browser link on the 
Genome Browser home page and then selecting the "Zoo" option from the genome list.</p> 
<p> 
The research team, led by  NHGRI Scientific Director Eric D. Green, included scientists from 
Pennsylvania State University, University of California Santa Cruz (UCSC), and the University of 
Washington in Seattle. In the study, the investigators systematically compared the patterns of 
transposon insertions among the species' sequences. One key result of the analysis was the 
confirmation of recently proposed mammalian evolutionary trees suggesting that primates are more 
closely related to rodents than to carnivores or artiodactyls. Another significant outcome was the 
discovery of a substantial number of previously unidentified non-coding DNA segments that are 
conserved across a wide range of species. Many of these regions could be identified only through 
comparisons of sequence from multiple species, demonstrating the importance of studying the genomes 
of a wide range of organisms as a means for identifying functional elements in the human genome.</p>
<p> 
UCSC built a customized version of the browser to display the target region for this study, allowing
scientists to interactively explore the data and predictions generated by this project, contribute 
data of their own, and track the project as data from additional species are generated. In addition 
to the browser, the UCSC team also contributed to the analytical portion of the study. Mathieu 
Blanchette identified the regions that are most highly conserved among species. Adam Siepel 
performed the phylogenetic analysis of rates of substitution. The UCSC team worked with Arian Smit 
to obtain definitive evidence that rodents branched off from the common ancestor later than 
carnivores and artiodactyls.</p> 
<p> 
For more information on the NISC study, see the <em>Science Daily</em> 
<a href="http://www.sciencedaily.com/releases/2003/08/030814072241.htm" 
target="_blank">press release</a>. Flat files of the assembled sequence and annotations may be 
obtained from <a href="http://www.nisc.nih.gov/data/" 
target="_blank">http://www.nisc.nih.gov/data/</a> or via the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html">Downloads</a> link on the Genome Browser 
home page.</p> 
<p> 
We'd like to thank the NISC Comparative Sequencing Program team for providing the data and 
comparative analysis for this Genome Browser release. Special thanks go to Elliott Margulies at 
NHGRI for serving as the main liaison between NHGRI and UCSC, and for contributing several 
annotation tracks to the browser. We'd also like to acknowledge the efforts of the many faculty, 
grad students, and staff members of the UCSC Genome Bioinformatics group who contributed to the 
research effort and browser development for this project.</p>

<a name="080803"></a>
<h2>Aug. 8, 2003 &nbsp;&nbsp; New human genome reference sequence now available in Genome
Browser</h2> 
<p> 
The latest human genome reference sequence (NCBI Build 34, Jul. 2003) is now available as database 
hg16 in the UCSC Genome Browser and blat server.</p> 
<p> 
There are 2,843,433,602 finished sequenced bases in the ordered and oriented portion of the 
assembly, which is an increase of 0.4 percent, or approximately 11 Mb, over the Build 33 assembly. 
Of particular note in this release is the addition of the pseudoautosomal regions of the Y 
chromosome. This sequence was taken from the corresponding regions in the X chromosome and is an 
exact duplication of that sequence.</p>  
<p> 
Some sequence joins between adjacent clones in this assembly could not be computationally validated 
because the clones originated from different haplotypes and contained polymorphisms in the 
overlapping sequence, or the overlap was too small to be to be reliable.  In these instances, the 
sequencing center responsible for the particular chromosome has provided data to support the join 
in the form of an electronic certificate. The Build 34 certificates may be reviewed 
<a href="certificates/certificate_hg16.html">here</a>.</p> 
<p> 
Bulk downloads of the data are available via FTP at 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg16">ftp://hgdownload.gi.ucsc.edu/goldenPath/hg16</a> 
or through the <a href="http://hgdownload.gi.ucsc.edu/downloads.html">Downloads</a> link. We 
recommend that FTP be used rather than HTML for the download of large or multiple files.</p> 
<p> 
We'd like to thank NCBI and the International Human Genome Sequencing Consortium for furnishing the 
data, and the UCSC team members who contributed to this release: Terry Furey, Hiram Clawson, 
Heather Trumbower, Mark Diekhans, Robert Baertsch, Donna Karolchik, Jim Kent and our sysadmin team 
Patrick Gavin, Jorge Garcia, and Paul Tatarsky.</p>

<h2>Jul. 14, 2003 &nbsp;&nbsp; Rat genome v.3.1 added to Genome Browser</h2> 
<p> 
The UCSC Genome Bioinformatics Group has released a browser and BLAT server on the v. 3.1 rat genome
assembly from the Rat Genome Sequencing Consortium. This assembly (UCSC version rn3, June 2003) was 
produced by the Atlas group at <a href="http://hgsc.bcm.tmc.edu/projects/rat/"
target="_blank">Baylor Human Genome Sequencing Center</a> (HGSC). This assembly is a minor update to
the 3.0 release. Sequence changes affect only chromosomes 7 and X. No additional assembly releases 
are planned prior to the publication of the rat genome analysis papers.</p> 
<p> 
The 3.x assemblies reflect several sequence additions and software improvements over the previous 
2.x assemblies, including the sequencing of over 1100 new BACs to cover gaps, an improved marker set
from the Medical College of Wisconsin, a new FPC map from the BC Genome Sciences Centre, and 
improved linking of bactigs. For detailed information and statistics about the 3.x assemblies, see 
the Baylor HGSC <a href="ftp://ftp.hgsc.bcm.tmc.edu/pub/analysis/rat/README" 
target="_blank">README</a>.</p> 
<p> 
Downloads of the rat sequence and annotation data are available at 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/rnJun2003/" 
target="_blank"> ftp://hgdownload.gi.ucsc.edu/goldenPath/rnJun2003/</a> or via the Downloads link 
on this page. These data are made available with <a href="credits.html#rat_use">specific conditions 
for use</a>.</p> 
<p> 
We'd like to thank the Rat Genome Sequencing Consortium and Baylor HGSC for providing this assembly,
collaborators from other institutions who have contributed annotations, and Arian Smit for updating 
RepeatMasker for this release.  We'd also like to acknowledge the contributions of several 
individuals at UCSC, including Hiram Clawson, Heather Trumbower, Robert Kuhn, Yontao Lu, Terry 
Furey, Mark Diekhans, Robert Baertsch, Donna Karolchik, Jim Kent, and our sysadmin team Jorge 
Garcia, Patrick Gavin, and Paul Tatarsky.</p>

<h2>Jun. 24, 2003 &nbsp;&nbsp; Genome Browsers released for <em>C. elegans</em> and <em>C.
briggsae</em></h2> 
<p> 
UCSC has just released browsers and blat servers for 2 worms: C. elegans version WS100 (May 2003) 
and C. briggsae version cb25.agp8 (Jul. 2002). The browsers are based on sequence obtained from 
<a href="http://www.wormbase.org/" target="_blank">WormBase</a>.</p>  
<p> 
We are pleased to add the nematodes to the roster of genomes available on our site. C. elegans is a 
major model organism used for biomedical research, and is the first multicellular animal to have a 
fully sequenced genome. In contrast,  the whole genome shotgun assembly of the C. briggsae genome is
estimated to have achieved 98% coverage. Draft chromosome sequences are not available for C. 
briggsae, due to the lack of dense chromosomal maps that allow assignment of ultracontigs to 
chromosomal locations. As a result, all data in the C. briggsae browser maps to chrUn.</p>  
<p> 
Both worms played a significant role in the early history of the UCSC Genome Browser. The browser 
code originated with a C script that displayed a splicing diagram for a gene prediction from C. 
elegans. Tracks for mRNA alignments and for homology with C. briggsae were added, and the tool 
morphed into the precursor of the Genome Browser, the &quot;Intronerator&quot; (Kent, WJ and Zahler,
AM (2000). 
<a href="http://nar.oupjournals.org/cgi/content/abstract/28/1/91?ijkey=155fc1582751ebf7b01072beb40325b1d0682fdc&keytype2=tf_ipsecsha" 
target="_blank">The intronerator: Exploring introns and alternative splicing in C. elegans</a>. 
<em>Nucleic Acids Res.</em> 28: 91-93).</p>  
<p> 
Downloads of the C. elegans sequence and annotation data are available at 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/ceMay2003/" 
target="_blank"> ftp://hgdownload.gi.ucsc.edu/goldenPath/ceMay2003/</a>; C. briggsae downloads can 
be found at <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/cbJul2002/" 
target="_blank">ftp://hgdownload.gi.ucsc.edu/goldenPath/cbJul2002/</a>. Both genomes can also be 
downloaded via the Downloads link on this page.</p> 
<p> 
We'd like to thank the <a href="http://genome.wustl.edu/" target="_blank">Genome Sequencing Center 
at Washington University in St. Louis</a> and the <a href="http://www.sanger.ac.uk/" 
target="_blank">Sanger Institute</a> for their collaborative work in sequencing the C. elegans and 
C.  briggsae genomes. Many thanks to the <a href="http://www.wormbase.org/about/staff" 
target="_blank">WormBase consortium</a> for making the worm sequence publicly available. We'd also 
like to acknowledge several UCSC people who contributed to this release: Hiram Clawson (browser and 
annotation tracks engineering), Jim Kent (WABA and chaining/netting), Al Zahler (WABA), Heather 
Trumbower (QA and project management), and Donna Karolchik (project management and
documentation).</p>

<h2>Jun. 23, 2003 &nbsp;&nbsp; Genome Browser UI changes</h2> 
<p> 
You may notice that we've removed the Genome pulldown menu and genome assembly information from our 
home page. Genome-specific information and links, as well as genome selection, are now available on 
the gateway pages for our tools. To open up a gateway page, simply click the Browser, Blat, or 
Tables link in the left sidebar.</p>

<h2>May 23, 2003 &nbsp;&nbsp; SARS Coronavirus TOR2 draft assembly available</h2> 
<p> 
[NOTE: The SARS assembly is no longer available on our public site, but may be accessed on our test 
server, <a href="http://genome-test.soe.ucsc.edu">genome-test.soe.ucsc.edu</a>.]</p> 
<p> 
Today we'd like to announce the release of a genome browser and BLAT server for the SARS coronavirus
TOR2 draft assembly. The browser - which is based on sequence deposited into GenBank as of Apr. 14, 
2003 - provides seven annotations showing gene predictions, locations of putative proteins, and 
viral mRNA and protein alignments. Of particular note are the protein structure analysis and 
predictions, determined by using the <a href="http://www.soe.ucsc.edu/research/compbio/sam.html" 
target="_blank">Sequence Alignment and Modeling (SAM) T02</a> tool. </p> 
<p> 
This browser marks a departure from our usual collection of vertebrate genomes. Its inception was 
inspired by one of our engineers - Angie Hinrichs - who was vacationing in New Zealand when the SARs
draft assembly was initially released. Struck by the impact of SARS in that part of the world, she 
downloaded the sequence and built the initial tracks from a terminal at an Internet cafe! The rest 
of the team joined in on the grassroots effort, generating the additional annotations and SAM T02 
protein analyses and predictions. Victor Solovyev chimed in with Fgenesv+ gene predictions from 
Softberry Inc. UCSC does not intend to provide a comprehensive collection of viral genomes in the 
future, but will maintain this browser as long as scientific and public interest in SARS 
persists.</p> 
<p> 
Downloads of the annotation data are available at 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/scApr2003/database" 
target="_blank">ftp://hgdownload.gi.ucsc.edu/goldenPath/scApr2003/database</a> or via the Downloads
link on this page.</p> 
<p> 
We'd like to thank everyone who worked on this release, including Angie Hinrichs, Robert Baertsch, 
Fan Hsu, Matt Schwartz, Heather Trumbower, Jim Kent, Kevin Karplus, Donna Karolchik, Brian Raney, 
Hiram Clawson, Kate Rosenbloom, Victor Solovyev, and our extremely dedicated systems administrators 
Paul Tatarsky, Patrick Gavin, and Jorge Garcia.</p>

<h2>Apr. 21, 2003 &nbsp;&nbsp; Chromosome 8 data corrected for April 2003 human assembly</h2> 
<p> 
The file that we originally used to build the agp files for the Apr. 2003 human release (Build 33) 
erroneously contained 2 contigs on chromosome 8 that were listed twice: NT_078037, NT_008183. We've 
received a corrected version and have updated the following files on our website: contigAgp.zip, 
chromAgp.zip, liftAll.zip. You can obtain the newer versions of these files from our ftp site at 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/10april2003/bigZips/"> 
ftp://hgdownload.gi.ucsc.edu/goldenPath/10april2003/bigZips/</a>.</p>

<h2>Apr. 14, 2003 &nbsp;&nbsp; Human Genome Project successfully completed!</h2> 
<p> 
The International Human Genome Sequencing Consortium today announced the successful completion of 
the Human Genome Project. The most significant outcome of this project is the reference sequence of 
the human genome. The sequencing of the 3 billion letters of DNA in the human genome - which many 
consider to be one of the most ambitious scientific undertakings in history - was completed 2 years 
ahead of schedule and at substantially less cost than original estimates. The reference sequence 
will serve as a new foundation for research in the fields of medicine and human biology.</p>  
<p> 
In conjunction with this announcement, the UCSC Genome Bioinformatics group is proud to release a 
genome browser and BLAT server on the reference sequence (NCBI Build 33), along with bulk downloads 
of the sequence and annotation data. The initial browser provides a preliminary set of annotations 
that will be expanded in coming weeks. Bulk downloads of the data are available via FTP at 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/10april2003">ftp://hgdownload.gi.ucsc.edu/goldenPath/10april2003</a> 
or through the <a href="http://hgdownload.gi.ucsc.edu/downloads.html">Downloads</a> link on this 
page. We recommend that FTP be used rather than HTML for the download of large or multiple
files.</p> 
<p> 
The reference sequence covers about 99 percent of the human genome's gene-containing regions, and 
has been sequenced to an accuracy of 99.99 percent. The missing portions are essentially contained 
in less than 400 defined gaps that represent DNA regions with unusual structures that can't be 
reliably sequenced using current technology. The average DNA letter now lies within a stretch of 
approximately 27,332,000 base pairs of uninterrupted sequence!</p> 
<p> 
Chromosomal sequences for this release were assembled by the International Human Genome Sequencing 
Consortium sequencing centers and verified by NCBI and UCSC. In some cases, sequence joins between 
adjacent clones could not be computationally validated, e.g. due to polymorphisms in the overlapping
sequence. In situations like this, supporting evidence for the join has been provided by the 
sequencing center responsible for that particular chromosome. The <a href="certificate.html">Non-standard Join Certificates</a> table displays this information. The annotations on the UCSC website 
have been provided by UCSC and collaborators worldwide. See the <a href="credits.html">Credits</a> 
page for a detailed list of organizations and individuals who contributed to this release.</p> 
<p>
We'd like to congratulate the many people worldwide who have worked on the Human Genome Project for 
this landmark achievement. We'd also like to acknowledge the UCSC Genome Browser project team who 
worked many long hours to ensure that the genome browser and sequence data were released on time for
this announcement: David Haussler, Jim Kent, Terry Furey, Matt Schwartz, Heather Trumbower, Angie 
Hinrichs, Fan Hsu, Donna Karolchik, Jorge Garcia, Patrick Gavin, Chuck Sugnet, Yontao Lu, Mark 
Diekhans, Ryan Weber, Robert Baertsch, Krishna Roskin, and the many other students in the UCSC 
Genome Bioinformatics group.</p>

<h2>Apr. 2, 2003 &nbsp;&nbsp; Human assembly hg10 and mouse assembly mm1 archived</h2> 
<p> 
The Dec. 2001 Human assembly (hg10) and the Nov. 2001 Mouse assembly (mm1) have been moved to the 
archives. They are no longer viewable in the main browser, but can still be downloaded from the 
<a href="http://hgdownload.gi.edu">downloads server</a>.</p>

<h2>Mar. 26, 2003 &nbsp;&nbsp; New rat assembly released in Genome Browser</h2> 
<p> 
We've added an updated rat assembly to our site: Rat Jan 2003 (rn2). This corresponds to the 
Version 2.1 Jan 2003 Update of the rat genome assembly, produced by the Atlas group at Baylor HGSC 
as part of the Rat Genome Sequencing Consortium.</p> 
<p> 
This update corrects duplications that were assembly artifacts in the previous version and improves 
the linking of bactigs to create larger "ultrabactigs". Compared with the previous rat assembly, 
sequence mapped to specific chromosomal coordinates is reduced by about 1.6 percent. Loosely mapped 
and unmapped sequence is reduced by 17 percent. For more details and statistics on the Jan. 2003 
assembly, see the Baylor HGSC 
<a href="ftp://rat-ftp.hgsc.bcm.tmc.edu/pub/analysis/rat/README">README</a> for this release.</p>  
<p> 
UCSC has released a Genome Browser and BLAT server for this assembly update. The initial 
<a href="../cgi-bin/hgTracks?db=rn2"> browser</a> contains 16 annotation tracks, with more to follow
in coming weeks. Sequence downloads are currently available at 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/rnJan2003/" 
target="_blank">ftp://hgdownload.gi.ucsc.edu/goldenPath/rnJan2003/</a> or via the Downloads link on
this page. A complete set of database downloads will be available at the beginning of next week. 
This data contains <a href="credits.html#rat_use">specific conditions for use</a>.</p> 
<p> 
Thanks to the Atlas group at Baylor HGSC, the Rat Genome Sequencing Consortium, the UCSC Genome 
Bioinformatics group, and contributors worldwide for making this release available.</p>

<h2>Mar. 13, 2003 &nbsp;&nbsp; Updated mouse genome assembly now available</h2> 
<p> 
We're happy to announce an update to the mouse genome sequence. This new version (Mouse Feb.  2003) 
includes 705 megabases of finished sequence, compared to 96 megabases of finished sequence in the 
previous assembly. Many people in the Mouse Genome Sequencing Consortium contributed to this update.
The Sanger Institute in particular contributed a large amount of finished sequence. Richa Agarwala, 
Deanna Church, and coworkers at NCBI layered the finished clones on top of the Arachne whole genome 
shotgun assembly. Arian Smit constructed a new RepeatMasker library.</p>  
<p> 
UCSC has released a Genome Browser and BLAT server for the Feb. 2003 Mouse genome. The initial 
<a href="../cgi-bin/hgTracks?db=mm3"> browser</a> contains 14 annotation tracks, with more to follow
in coming weeks. Sequence downloads are currently available at 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/mmFeb2003/" 
target="_blank">ftp://hgdownload.gi.ucsc.edu/goldenPath/mmFeb2003/</a> or via the Downloads link on
this page. Database downloads will be available at the beginning of next week.</p> 
<p> 
Thanks to everybody at UCSC and around the world that contributed to this release!</p> 

<h2>Feb. 5, 2003 &nbsp;&nbsp;  Enhanced version of Genome Browser released</h2> 
<p> 
We're proud to announce the release of version 17 of the UCSC Genome Browser. This version contains 
powerful new features, numerous improvements to the annotation track display, additional annotation 
tracks, and a number of bug fixes. In this release cycle, we've also introduced an enhanced QA 
process that formalizes our testing and verification of the Genome Browser software and the data 
displayed in the browser.</p>  
<p> 
<strong>New functionality in v.17:</strong><br> 
-- Numerous enhancements to the table browser that allow the user to conduct more complex and 
specific searches. New features include support for intersections of tracks, a new summary 
statistics output format, and the ability to output query results as a custom annotation track that 
can be viewed in the Genome Browser. The new <a href="help/hgTablesHelp.html" target="_blank">Table 
Browser User's Guide</a> contains a detailed description of the new features and provides a wealth 
of information and examples for conducting various types of searches on the database tables.</p> 
<p>
-- Two new display modes available for most annotation tracks: pack and squish modes. In <em>pack 
mode</em> display, annotation track features are fully displayed, but more than one feature may be 
displayed on the same line. This greatly reduces the amount of display space needed by a track when 
a user wishes to view a large number of individual features at one time. <em>Squish mode</em> is 
similar to pack mode, but displays features at 50% height and without labels. This mode is 
particularly useful for viewing tracks in which a large number of features align to the same section
of a chromosome, e.g. EST tracks.</p>  
<p> 
-- Functional groupings of annotation track controls. This makes it much easier to find a particular
item in the track control list and gives a better visual overview of the annotations available in a 
particular category, e.g. comparative genomics tracks or gene prediction tracks.</p> 
<p> 
-- A mechanism for saving the annotation tracks image in postscript or PDF format. This much 
requested feature enables Genome Browser users to print an image at high resolution, edit it with a 
drawing program, or display it in a postscript or PDF viewer.</p> 
<p> 
-- A <a href="help/customTrack.html" target="_blank">collection of custom annotation tracks</a> 
supplied by Genome Browser users and members of the UCSC Genome Bioinformatics lab. Additional 
contributions to this collection are welcome! Contact 
<a href="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</a> 
<!-- above address is genome at soe.ucsc.edu --> 
if you have an annotation you'd like to share.</p> 
<p> 
<strong>New annotation tracks in v.17:</strong><br> 
-- a new Known Genes track (Human Builds 30 & 31, Feb 2002 Mouse, Nov 2002 Rat) that shows known 
protein-coding genes based on proteins from SWISS-PROT, TrEMBL, and TrEMBL-NEW and their 
corresponding mRNAs from GenBank. Features within the track are color-coded according to origin and 
review status.</p> 
<p>
-- a new Superfamily track (Human Build 30, Feb 2002 Mouse) that shows proteins having homologs with
known structures or functions. Each entry in the track shows the coding region of a gene (based on 
Ensembl gene prediction). The feature label consists of the names of all known protein domains coded
by the gene, and usually contains structural and/or function descriptions that provide valuable 
information for getting a quick grasp of the biological significance for the gene.</p> 
<p> 
We have also released several additional annotation tracks on the latest human and rat assemblies in
the past month.</p> 
<p> 
<strong>Bug fixes in v.17:</strong><br> 
-- Approximately 40 bugs (mostly minor problems) have been fixed in this version.</p> 
<p> 
To take full advantage of the new display features in this release, we recommend that you reset your
browser to the new default settings. NOTE: you may not want to reset your browser if you have 
customized settings that you wish to preserve. You can reset your browser by clicking the
&quot;Click here to reset&quot; link on the Browser Gateway page.</p>  
<p> 
We hope this new release facilitates your work with the UCSC Genome Browser. If you have any 
questions or comments about the new release, send email to 
<a href="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</a>.  
<!-- above address is genome at soe.ucsc.edu --></p>

<!---------------- 2002 archived news --------------->
<a name="2002"></a>

<h2>Dec. 23, 2002  &nbsp;&nbsp; Human Genome assembly Build 31 released in Genome Browser</h2> 
<p> 
We're pleased to announce the release of the latest human genome assembly, Build 31 (UCSC version 
hg13). This assembly was produced at NCBI based on sequence information submitted into GenBank as of
Nov. 14, 2002.  Release notes for this assembly are available from the 
<a href="https://www.ncbi.nlm.nih.gov/genome/guide/human/release_notes.html" target="_blank">NCBI web
site</a>. Because UCSC now obtains its assembly directly from NCBI, the UCSC Build 31 data is 
identical to that of NCBI and Ensembl.</p>  
<p> 
Build 31 is an excellent high-quality assembly that shows a remarkable amount of progress toward the
milestone of finishing the human genome. Greater than 95% of the euchromatic region of the genome 
is now complete, with more than 90% of the sequence in a finished state. The number of clone contig 
gaps has decreased by one third from the previous assembly, and the overall number of sequenced 
contigs has been reduced by one half. Seven chromosomes are considered to be in a finished state: 6,
7, 13, 20, 21, 22, and Y.</p>  
<p> 
The initial release of the Build 31 Genome Browser contains 25 annotation tracks, with several more 
to follow in the upcoming weeks. Bulk downloads of the data are available from our FTP site at
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/14nov2002" 
target="_blank">ftp://hgdownload.gi.ucsc.edu/goldenPath/14nov2002</a> or via the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html">Downloads</a> link on this page.</p> 
<p> 
UCSC has generated a set of high-level comparisons of the Build 31 draft sequence against various 
types of information (STS maps, BAC end pairs, and clone overlaps). This information, as well as 
statistics for Build 31, is accessible from the 
<a href="http://hgdownload.gi.ucsc.edu/goldenPath/human/chromReports/14Nov2002/" 
target="_blank">Chromosome Reports</a>, 
<a href="http://hgdownload.gi.ucsc.edu/goldenPath/human/mapPlots/" 
target="_blank">Genome Map Plots</a>, and <a href="stats.html" target="_blank">Summary 
Statistics</a> links in the &quot;Technical Information about the Assembled Sequence&quot; section 
below.</p> 
<p> 
We'd like to thank NCBI as well as all the people who collaborated on the data and annotations for 
this release.</p>

<h2>Dec. 6, 2002 &nbsp;&nbsp; Rat Genome assembly now available in Genome Browser</h2> 
<p> 
We're pleased to announce the release of a UCSC Genome Browser on the Nov. 2002 rat assembly 
produced by the Baylor College of Medicine Rat Genome Sequencing Center and the Rat Genome 
Sequencing Consortium. The sequence was assembled using a hybrid approach that combines the clone by
clone and whole genome shotgun methods. A new software program - ATLAS - was developed for this 
effort. The assembly process resulted in a 6.5-fold coverage of the rat genome, which is estimated 
to be approximately 2.8 Gigabases in size.</p> 
<p> 
Downloads of the rat data and annotations are available through our 
ftp site at <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/rn1" target="_blank">downloads 
server</a>. This data contains <a href="credits.html#rat_use">specific conditions for use</a>. The 
sequence is also available from the <a href="http://www.hgsc.bcm.tmc.edu/projects/rat" 
target="_blank">Rat Genome Project website</a> for the Human Genome Sequencing Center at Baylor 
College of Medicine or from GenBank.</p> 
<p> 
We'd like to thank the Baylor team and the Rat Genome Sequencing Consortium for their collaboration 
on this project. See the <a href="credits.html#rat_credits">Credits page</a> for a complete list of 
acknowledgments. For more information on the rat genome, the assembly process, and the Rat Genome 
Sequencing Consortium, refer to the website for the <a href="http://www.hgsc.bcm.tmc.edu/" 
target="_blank">Human Genome Sequencing Center at Baylor College of Medicine</a>.</p>

<h2>Dec. 5, 2002 &nbsp;&nbsp; Mouse Genome Browser released</h2> 
<p> 
The International Mouse Genome Sequencing Consortium has announced the publication of a high-quality
draft sequence of the mouse genome, together with a comparative analysis of the mouse and human 
genomes. The results from this analysis can be found in the Mouse Genome Browser on this website. 
The paper appears in the Dec. 5 issue of the journal Nature at 
<a href="http://www.nature.com/nature/mousegenome/" 
target="_blank">http://www.nature.com/nature/mousegenome/</a>. The co-author list includes several 
members of the UCSC Genome Bioinformatics Group: CBSE Director David Haussler, Research Scientist 
Jim Kent and research team members Robert Baertsch, Mark Diekhans, Terrence Furey, Angie Hinrichs, 
Fan Hsu, Donna Karolchik, Krishna Roskin, Matt Schwartz, Charles Sugnet and Ryan Weber.</p>

<h2>Oct. 29, 2002 &nbsp;&nbsp; Additional data downloads available for human assembly Jun. 2002</h2>
<p> 
We've added several new directories of downloadable data to the Jun. 28, 2002 human genome assembly.
These directories contain mouse/human alignments of the Jun. 2002 human assembly vs. the Feb. 2002 
mouse assembly. You can access these directories from our 
<a href="http://hgdownload.gi.ucsc.edu/goldenPath/hg12/" target="_blank">downloads server</a>.</p> 
<p>
Within the main directory <em>vsMm2</em> are 3 subdirectories that contain all the alignments 
(<em>axtAll</em>), alignments filtered to provide only the best alignment for any given region of 
the human genome (<em>axtBest</em>), and a relatively stringent subset of the axtBest alignments 
(<em>axtTight</em>). For more information about the format of the alignment files and the methods 
used to generate the alignments, consult the <em>README.txt</em> file in the vsMm2 directory. </p>

<h2>Oct. 18, 2002 &nbsp;&nbsp; Enhancements in v16 Genome Browser</h2> 
<p> 
We've rolled out a new version of the Genome Browser - v.16. In addition to several bug fixes, this 
release contains some interesting new features.</p> 
<p> 
The Table Browser has undergone major enhancements. Users can now restrict their queries by 
specifying a value or range for any of the fields in a table, and by selecting which fields should 
be displayed in the output. The Table Browser also provides the ability to do a free-form SQL query 
on a table and supports several new output formats.</p>  
<p>	
We've extended the capabilities of the DNA retrieval functionality in the Genome Browser and the 
Table Browser. The new mechanism offers the user several options for configuring the amount and 
type of sequence region that is retrieved, and options for formatting the sequence output. The 
retrieval options vary based on the type of table selected.</p> 
<p>	
The Genome Browser's gene prediction tracks now offer a Comparative Sequence link in addition to the
predicted protein, mRNA sequence, and genomic sequence links. The Comparative Sequence feature 
displays annotated codons and translated protein for the region in alignment to another species.</p>
<p>	
A logarithmic scale and displays in the track as &quot;mountain ranges&quot; Details pages 
associated with the individual peaks in the track provide access to the base level alignments for 
the whole region and for the individual 50bp windows.</p> 
<p>
The team that produced this track includes Ryan Weber, Krishna Roskin, Mark Diekhans, Jim Kent, 
Scott Schwartz, and Webb Miller.</p>

<h2>Sep. 12, 2002 &nbsp;&nbsp; &quot; User's Guide to the Human Genome&quot; features UCSC
Genome Browser</h2> 
<p> 
<em>Nature Genetics</em> has just published 
<a href="http://www.nature.com/cgi-taf/DynaPage.taf?file=/ng/journal/v32/n1s/index.html" 
target="_blank">User's Guide to the Human Genome</a>, a hands-on tutorial for using genome browsers 
as web tools for browsing and analyzing data from the Human Genome Project and other sequencing 
efforts. The 3 browsers featured in the tutorial include the UCSC Genome Browser, NCBI's Map Viewer,
and the Ensembl Genome Browser. The guide is organized around a collection of step-by-step solutions
to 13 typical research questions, and serves as a nice supplement to the documentation materials 
available on the UCSC Genome Browser web site. The guide is accessible as a link off the <em>Nature 
Genetics</em> home page at <a href="http://www.nature.com/ng/" 
target="_blank">http://www.nature.com.ng</a>.</p>

<h2>Sep. 12, 2002 &nbsp;&nbsp; Correction to Human Assembly hg12 Chromosome Bands annotation</h2> 
<p> 
We've corrected a problem with the Human Jun. 2002 (hg12) cytoBand annotation track that affected 
chromosome Y. The clones on this chromosome were erroneously pushed from the q-arm onto the p-arm, 
creating some confusion. The currently available version of the cytoBand data on this website 
contains this correction.</p>

<h2>Aug. 15, 2002 &nbsp;&nbsp; Additional annotations for human draft assembly hg12 released</h2> 
<p> 
We've just released several new annotation tracks/tables for the human genome. On the Jun. 2002 
assembly, we've added Gene Bounds, UniGene, CpG Islands, Nonhuman mRNA & EST, SNPs, NC160, and GNF 
Ratio. On the Apr. 2002 assembly we've added a Fgenesh++ Genes track, and have also updated the 
RepeatMasker track. The Dec. 2001 release now includes a Sanger 22 track.</p> 
<p>
On the mouse genome, we have 2 new tracks for the Feb. 2002 assembly: TIGR Gene Index and
RNA Genes.</p>
	
<h2>Aug. 6, 2002 &nbsp;&nbsp; Error with six chromosome 3 contigs in human assembly hg12</h2> 
<p> 
We have fixed an error with six of the chr3 contigs in the bigZips/contigAgp.zip file. The following
.agp files were corrected: NT_005684.agp, NT_005663.agp, NT_022554.agp, NT_022459.agp, 
NT_006031.agp, and NT_022419.agp. The chr3.agp file in bigZips/chromAgp.zip was also modified. This 
change does not affect the .gl files, the .fa files, the lift files, or the annotations. Alignments 
made on the previous version of chr3 are still good. Updated versions of the contigAgp.zip and 
chromAgp.zip files were uploaded to our site today. You can download the new versions via ftp from 
our <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg12" target="_blank">downloads 
server</a>.</p>

<h2>Aug. 2, 2002 &nbsp;&nbsp; New RepeatMasker track available for human assembly hg12</h2> 
<p>
The problems with the Jun. 2002 Build 30 (hg12) RepeatMasker track have been resolved. The new 
RepeatMasker track, along with regenerated Fish Blat and Genscan tracks, are now available in the 
Browser and through our Downloads link. We've also added a few new annotation tracks for the Jun. 
2002 release, and will be adding more over the next 2 weeks.</p> 
<p>
The latest Genome Browser has 2 new features. We've added filter functionality to the Table Browser,
accessible via the Filter Fields button on the Table Browser main page. Also, some of the Dec. 2001 
human genome tracks (eg. RefSeq Genes) now have a Comparative Sequence link from the details page 
that shows annotated codons and translated protein with alignment to the mouse genome.</p>

<h2>Jul. 27, 2002 &nbsp;&nbsp; Problems with repeat-masking on human assembly hg12</h2> 
<p> 
We've experienced some RepeatMasker problems on Build 30 and are rerunning it. This will directly 
affect the RepeatMasker track and the masking of the fasta files. The Fish Blat and Genscan tracks 
may also change slightly once we're redone this. The EST, mRNA, and RefSeq tracks should not be 
affected. We will also post a new RepeatMasker track for Build 29 (see news item below) as soon as 
the Build 30 tracks are completed. We apologize for any rework this may cause.</p>

<h2>Jul. 26, 2002 &nbsp;&nbsp; Bulk data downloads now available for human assembly hg12</h2> 
<p> 
Bulk downloads of the Jun. 2002 Build 30 human genome assembly (hg12) are now available from the 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg12/" target="_blank">downloads server</a>. This 
initial release of the annotation database download contains a limited set of tables. Additional 
files will be available for download next week.</p>

<h2>Jul. 24, 2002 &nbsp;&nbsp; Human assembly hg12 blat server available</h2> 
<p> 
The BLAT server and the coordinates conversion feature for human genome assembly Build 30 (hg12) are
now functional.</p>

<h2>Jul. 23, 2002 &nbsp;&nbsp; Pre-release of human draft assembly Build 30 in Genome Browser</h2> 
<p>
We're pleased to announce the pre-release of a browser for human genome assembly Build 30 from NCBI 
(UCSC version hg12). This assembly was produced at NCBI based on sequence information submitted into
GenBank as of Jun. 28, 2002. Build 30 release notes and statistics will soon be available from the 
<a href="https://www.ncbi.nlm.nih.gov/" target="_blank">NCBI web site</a>.</p>  
<p> 
Build 30 is an excellent high-quality assembly. It contains nearly 87% finished sequence, and 
94%-97% coverage. The sequence coverage of this build is much higher than in previous releases, and 
there is a high level of correspondence between the sequence and the map. Currently, the human 
genome project appears to be on track to achieve the goal of finishing at least 95% of the human 
genome (using Bermuda standards) by Apr. 2003.</p>  
<p> 
UCSC has generated a set of high-level comparisons of the Build 30 draft sequence against various 
types of information (STS maps, BAC end pairs, and clone overlaps), accessible from the 
<a href="http://hgdownload.gi.ucsc.edu/goldenPath/human/chromReports/24Jun2002/" 
target="_blank">Chromosome Reports</a> and 
<a href="http://hgdownload.gi.ucsc.edu/goldenPath/human/mapPlots/" target="_blank">Genome Map 
Plots</a> links in the &quot;Technical Information about the Assembled Sequence&quot; section 
below.</p> 
<p> 
A Blat server for Build 30 is not yet available, but should be accessible from this site later this 
week. Data for the mitochondrial genome and several more annotation tracks will be posted for this 
release as they become available. Bulk downloads of the hg12 data should be available from this site
in a few days.</p>

<h2>Jul. 9, 2002 &nbsp;&nbsp; Repeat-masking problems on human assembly hg11</h2> 
<p>
We've found some problems with the repeat-masking of the Build 29 (hg11) human sequence. We're in 
the process of replacing the RepeatMasker track, but do not plan to redo the other tracks due to the
imminent release of Build 30. Because of this, we advise that you do not use the cross-species 
tracks for statistical purposes.</p>

<h2>Jul. 1, 2002 &nbsp;&nbsp; New look for UCSC Genome Bioinformatics home page</h2> 
<p> 
The UCSC Genome Bioinformatics home page is sporting an updated interface to accommodate the 
growing number of organisms supported by the UCSC Genome Browser, BLAT, and Table Browser. The list 
of assembly versions accessible through each of these tools can now be found on the tool's Gateway 
page. To reach the Gateway page, choose an organism from the dropdown list on the left sidebar of 
this page, then click the Browser, BLAT, or Tables link. New organisms will be added to the list in 
the months ahead.</p>  
<p>
The UCSC site continues to provide a variety of bulk downloads of a genome assemblies and 
annotations. The list of downloadable data has been removed from the home page, but is readily 
available through the Downloads link on the left sidebar. The downloads list can also be accessed 
directly at 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html">http://hgdownload.gi.ucsc.edu/downloads.html</a> 
or through our ftp site at <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/" 
target="_blank">ftp://hgdownload.gi.ucsc.edu/goldenPath/</a>.</p>  
<p>
Several new annotation tracks have been added to our site in the past couple weeks. The Feb. 2002 
mouse assembly now has tracks for BAC End pairs, Fgenesh++ gene predictions, and AltGenie gene 
predictions based on Affymetrix's Genie gene-finding software. New to the Apr. 2002 human assembly 
is the GenMapDB Clones track, which shows placements of BAC clones from the GenMapDB database based 
on BAC end sequencing information and confirmed using STS markers by Vivian Cheung's lab at U. Penn.
We've also changed the Known Genes track name to RefSeq Genes in all assemblies.</p>  
<p>
This release also includes an updated <a href="help/hgTracksHelp.html" target="_blank">User's 
Guide</a> and more detailed documentation on creating & using <a href="help/customTrack.html" 
target="_blank">custom annotation tracks</a>.</p>

<h2>May 24, 2002 &nbsp;&nbsp; Bulk data downloads for human assembly hg11 now available</h2> 
<p> 
Bulk downloads of the April 2002 hg11 human genome assembly (NCBI Build 29) are now available from 
the <a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg11/" target="_blank">downloads
server</a>.</p>

<h2>May 22, 2002 &nbsp;&nbsp; Human draft assembly Build 29 released in Genome Browser</h2> 
<p> 
We've just released a browser and BLAT server on the latest Build 29 human genome assembly from NCBI
(UCSC version hg11). This assembly is based on sequence information submitted into GenBank as of 
Apr. 5 2002. As with the Dec. 2001 (hg10) release, this assembly was produced at NCBI rather than at
UCSC. Consult NCBI's Build 29 release notes and statistics
for more information about this release. This assembly contains nearly 75% finished sequence. 
Currently, the human genome project appears to be on track to achieve the goal of finishing at least
95% of the human genome (using Bermuda standards) by Apr. 2003.</p> 
<p>
Although the NCBI human genome assembly has been steadily improving over the past year, mapping 
problems still exist in the current release. Most are small, relatively local rearrangements. Larger
scale problems include a rearrangement in the p-arm of Chr16 and several discrepancies in Chr17. 
Researchers - especially positional cloners - are strongly encouraged to use the tools provided 
(<a href="http://hgdownload.gi.ucsc.edu/goldenPath/human/mapPlots/" target="_blank">comparison 
plots</a>, <a href="http://hgdownload.gi.ucsc.edu/goldenPath/human/chromReports/" 
target="_blank">chromosome reports</a>) to evaluate the accuracy of the assembly in specific regions
of interest.</p> 
<p>
Bulk downloads of the hg11 data should be available from this site in approximately one week. New 
annotation tracks will be posted as soon as they become available.</p>

<h2>Apr. 24, 2002 &nbsp;&nbsp; Bulk data downloads of mouse assembly Feb. 2002 available</h2> 
<p>
Bulk downloads of the February 2002 mouse genome assembly are now available from the 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/mm2/" target="_blank">downloads server</a>.</p>

<h2>Apr. 19, 2002 &nbsp;&nbsp; New mouse genome assembly released</h2> 
<p>
The Feb. 2002 mouse genome assembly is now available in the browser and for BLAT searching. This 
assembly was produced at the Whitehead Institute using their Arachne software. We'd like to thank 
them and the Mouse Genome Sequencing Consortium for providing this assembly, which has 
<a href="credits.html#mouserestrict">specific conditions for use</a>. Bulk downloads of the data 
should be available in approximately one week. Coordination with mouse genome data access at 
<a href="http://www.ensembl.org/Mus_musculus/">Ensembl</a> and 
<a href="https://www.ncbi.nlm.nih.gov/genome/guide/mouse/">NCBI</a> is in progress. We'd also like to
acknowledge the UCSC team that produced this release: Jim Kent, Terry Furey, Matt Schwartz, Fan Hsu,
Yontao Lu, Donna Karolchik, Chuck Sugnet, and Ryan Weber.</p>

<h2>Apr. 9, 2002 &nbsp;&nbsp; Bulk data downloads of mouse assembly Nov. 2001 available</h2> 
<p>
Bulk downloads of the November 2001 mouse genome assembly are now available from the 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/mm1/" target="_blank">downloads server</a>.</p>

<h2>Apr. 2, 2002 &nbsp;&nbsp; Release of new Genome Browser version and mouse/human annotations</h2>
<p>
An updated version of the UCSC Genome Browser (v.11) is now available. Along with the v.11 browser, 
we've released several new annotation tracks on the latest human and mouse assemblies. The new Human
Dec. 2001 tracks include: Mouse Synteny, Ensembl, Genscan, CpG Islands, Mouse Blat, Fish Blat, 
Unigene/SAGE, NCI60 Microarray, GNF Affymetrix Microarray, Rosetta Microarray, and SNPs. An STS 
Markers track has been added to the Mouse Nov. 2001 browser.</p>

<h2> Mar. 14, 2002 &nbsp;&nbsp; Mouse draft assembly Nov. 2001 released in Genome Browser</h2> 
<p>
The November 2001 mouse genome assembly is now available for viewing in the browser and for BLAT 
searching. This assembly was produced at the Sanger Center using the Phusion software developed by 
Jim Mullikin and Zemin Ning, and was tied to the mouse fingerprint map by Tim Hubbard. We'd like to 
thank them and the Mouse Genome Sequencing Consortium for providing this assembly, which has 
<a href="credits.html#mouserestrict">specific conditions for use</a>. Bulk downloads of the data 
will be available in approximately one week. Coordination with mouse genome data access at 
<a href="http://www.ensembl.org/Mus_musculus/">Ensembl</a> and 
<a href="https://www.ncbi.nlm.nih.gov/genome/guide/mouse/">NCBI</a> is in progress. We'd also like to
acknowledge the UCSC team that produced this release: Jim Kent, Terry Furey, Matt Schwartz, Fan Hsu,
Yontao Lu, and Donna Karolchik.</p>

<h2>Feb. 16, 2002 &nbsp;&nbsp; Human Genome assembly Build 28 now available</h2> 
<p>
A new assembly based on sequence submitted as of Dec. 22 in GenBank (Build 28) is now available in 
the browser and for BLAT search. This assembly was produced at NCBI rather than UCSC, primarily by 
Richa Agarwala, Greg Schuler, and Paul Kitts. The NCBI assembly has been steadily improving over the
past year. Currently it shows slightly better local order and orientation compared to the UCSC 
assembly on the same sequence, but somewhat worse tracking of the chromosome level maps. The NCBI 
assembly has the advantage that it can be generated significantly faster than the UCSC assembly. 
With the human genome sequencing now in the end game - over two thirds of the human clones are now 
finished - we feel it more productive to focus worldwide annotation efforts on a single assembly 
rather than continue producing competing assemblies. We're working with NCBI to improve their map 
tracking.</p> 

<h2>Feb. 4, 2002 &nbsp;&nbsp; Chromosome Reports released</h2> 
<p>
Chromosome Reports detailing correspondence with STS map, overlap, and BAC end sequence information 
are available under the &quot;Technical Information About the Assembled Sequence&quot; section 
below. This also gives information about the clone map on which the assembled sequence is based.</p>

<a name="2001"></a>
<h2>Dec. 18, 2001 &nbsp;&nbsp; New Genome Browser enhancements</h2> 
<p>
There are some major enhancements to the browser. The complete user interface settings including 
track controls, labels, and position are now saved from session to session. You can configure the 
browser once to your liking and it will stay that way. This feature will only work if cookies are 
enabled in your browser. If you want to restore the default settings use the reset all button under 
the main graphic. Also under the main graphic are new controls that move just the start or just the 
end of the genome window. These are useful for getting exactly the right view without having to do 
arithmetic on the position. These controls by default will move two guideline units at a time, but 
you can specify other increments.  There's a new page associated with each track. This page is 
accessible by clicking on the mini-buttons to the left of the track in the main graphic, or by 
clicking on the new hyperlink associated with the track in the track controls section under the 
graphic. These pages contain a description of the track and in many cases new controls. The mRNA and
EST associated controls let you color or filter the display according to tissue, author, organism, 
and so forth.</p> 
<p>
As with any new enhancement there are likely to be a few new bugs too. Many of these have been 
spotted and fixed already. Please let us know if you find a problem that persists more than a day 
or two. It's always helpful to include the freeze and genomic position with a problem report.</p>

<h2>Nov. 39, 2001 &nbsp;&nbsp; Link added to Jackson Lab's MGI Mouse Orthologs</h2> 
<p>
There is now a link from the known genes details page to the Jackson Lab's MGI Mouse Ortholog when 
the ortholog is known. Thanks to Carol Bult for her help setting up this link.</p> 

<h2>Nov. 28, 2001 &nbsp;&nbsp; Duplications track released for August browser</h2> 
<p>
A duplications track is now available in the August browser. This track shows duplicate blocks of 
sequence larger than 1000 bases. The track is hidden by default. To open it look for
&quot;Duplications&quot; in the third row of track controls under the main graphic window, and 
change the setting to &quot;dense&quot;. Thanks to Evan Eichler and Jeff Bailey for </p>
<a href="noPageHere.html">this track</a>.

<h2>Nov. 28, 2001 &nbsp;&nbsp; Sanger Curated Gene annotations now available on chromosome 20</h2> 
<p>
Sanger curated gene annotations are now available on chromosome 20. Thanks to Jennifer Ashurst, 
James Gilbert, and all the annotators at the Sanger Institute.</p>

<h2>Nov. 27, 2001 &nbsp;&nbsp; Haplotype blocks annotation released for chromosome 21</h2> 
<p>
A new track has been added to the August freeze browser showing haplotype blocks derived from 
common SNPs on Chromosome 21 by Perlegen, as described in 
<a href="http://www.sciencemag.org/cgi/content/abstract/294/5547/1719">Common High-Resolution 
Haplotypes</a>. Patil, N. et. al. Science 294:1719-1723 (2001).</p>

<h2>Nov. 19, 2001 &nbsp;&nbsp; SNP and mouse blat tracks available</h2> 
<p>
The SNP and Mouse Blat tracks are now available for August. The Mouse Blat track uses a partial 
assembly of the public whole genome shotgun data courtesy of Whitehead's Arachne program.</p>

<h2>Nov. 8, 2001 &nbsp;&nbsp; Interface of track details pages updated</h2> 
<p>
The detail web pages for each of the tracks have been updated to reflect the overall look and design
of this site.  You will now see the familiar blue navigation bar with links to the Browser, BLAT, 
Downloads, and the  FAQ page from each of the track detail pages.</p>

<h2>Nov. 8, 2001 &nbsp;&nbsp; STS Markers track updated on Apr. and Aug. browsers</h2> 
<p>
The STS Markers track has been updated on the April and August browsers to now include much more 
information on the detail page including links to UniSTS and details on the alignments of the 
markers to the draft sequence. In addition, all known aliases of the markers can be entered in the 
&quot;position&quot; window, and the corresponding merker will be found and displayed if its 
location has been determined.</p>

<h2>Nov. 8, 2001 &nbsp;&nbsp; FISH Clones track added to Apr. and Aug. browsers</h2> 
<p>
A new FISH Clones track has been added to the April and August browsers. Previously, this 
information has been included in the STS Markers track. Now, this has been broken out into a 
separate track with additional information provided on the detail page not previously shown.</p>

<h2>Nov. 6, 2001 &nbsp;&nbsp; Fgenesh++ gene predictions and cross-species mRNA tracks available</h2> 
<p>
The fgenesh++ gene prediction and the cross-species mRNA tracks are now available in the August 
browser.</p> 

<h2>Oct. 31, 2001 &nbsp;&nbsp; DNA button functionality upgraded</h2> 
<p>
The &quot;DNA&quot; button at the top of the browser has been significantly upgraded. By default it 
now returns DNA that has repeating elements in lower case and other DNA in upper case. There is also
an option to color the DNA output with various tracks.  You can have the case and font features such
as underline, bold, and italic follow tracks too.</p> 

<h2>Oct. 28, 2001 &nbsp;&nbsp; TIGR Gene Index track added to April browser</h2> 
<p>
There is now a TIGR Gene Index track in the Apr. 2001 freeze browser. The TIGR Gene Index is based 
on alignments of assembled ESTs from a number of species. Be sure to click into the track and follow
the outside links to the TIGR site, which contains a wealth of information on the genes.</p>

<h2>Oct. 29, 2001 &nbsp;&nbsp; Acembly track updated on August browser</h2> 
<p>
The Acembly track on the Aug. 2001 freeze has been updated to include predictions based on human 
ESTs and GenBank mRNAs as well as RefSeq human mRNAs. Protein predictions are now also available in 
the details page for this track. The outside link for this track is also very informative.</p>

<h2>Oct. 27, 2001 &nbsp;&nbsp; Share your custom annotation tracks!</h2> 
<p>
You can now share your custom tracks with the community. The easiest way to do this is to construct 
a link from your own web pages to the browser. Here is an example of a URL for such a link:
        <pre>   http://genome.ucsc.edu/cgi-bin/hgTracks?
        position=chr22:1-20000&db=hg8&
        hgt.customText=http://genome.ucsc.edu/test.bed</pre></p>
<p> 
The position variable tells the browser which part of the genome to display. The db variable refers 
to the freeze number. hg8 corresponds to the August 2001 freeze. The customText variable should 
refer to a URL containing plain text in one of the formats described in 
<a href="help/customTrack.html">http://genome.ucsc.edu/goldenPath/help/customTrack.html</a>.</p>
<p> 
Note that generally we only keep the last three versions of the genome online (hg6, hg7, and hg8).  
You'll have to update your link and track about every 4 months as a result.</p> 
<p>
Please send in the URLs of tracks you'd like to share to 
<a href="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;">&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</a>, 
<!-- above address is genome at soe.ucsc.edu --> 
along with a brief description of the track and the genome version it is tied to. We'll create an 
index page of these here.</p> 
<p>
You can also access an external custom track by including the URL of the track data (on a separate 
line starting with http://) in the custom track box at the bottom of the browser gateway.</p>

<h2>Oct. 15, 2001 &nbsp;&nbsp; Weekly updates of tab-delimited data files</h2> 
<p>
Fresh tab-delimited files from the browser database are now available. They will now be updated 
automatically every Sunday evening. The table browser queries the database directly, so it is always
up to the minute.</p>

<h2>Oct. 12, 2001 &nbsp;&nbsp; New gene prediction tracks for Apr. and Aug. browsers</h2> 
<p>
Several new sets of gene predictions came in this week. We now have fgenesh++ predictions for the 
April freeze, and Genscan and Acembly predictions for the August freeze.</p>

<h2>Oct. 11, 2001 &nbsp;&nbsp; Table Browser upgraded</h2> 
<p>
Why struggle with massive genomic file downloads when the UCSC 
<a href="../cgi-bin/hgTables">Table Browser</a> lets you select exactly the track data desired via 
a convenient web interface? Major new improvements by Krish Roskin have empowered and simplified 
this feature available now for the three most recent assemblies.</p>

<h2>Oct. 8, 2001 &nbsp;&nbsp; Oct. 2000 assembly moved to archives</h2> 
<p>
The October 2000 assembly has been moved to the 
<a href="http://hgdownload.gi.ucsc.edu/goldenPath/hg5/">archives</a> to make room for the August 
assembly.</p>
	
<h2>Oct. 5, 2001 &nbsp;&nbsp; Revised Aug. 2001 assembly released</h2> 
<p>
A revised Aug. 2001 freeze assembly is now up. The problems with flipped contigs of finished clones 
and high levels of sequence duplication are fixed. You can now download this assembly 
<a href="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg8/">in bulk</a> as well as browse it. 
Chromosome by chromosome and annotation database files will be following over the next day or 
two.</p>

<h2>Oct. 2, 2001 &nbsp;&nbsp; Convert coordinates between different draft assemblies</h2> 
<p>
You can now convert coordinates between different versions of the draft using a new program, 
hgCoordConv, written by Chuck Sugnet. hgCoordConv attempts to cut out sequences of the original 
draft and align them to the new draft. When aligning the sequences to the new draft hgCoordConv 
makes sure that the sequences are in the same order, orientation, and have the correct distances 
between them.</p> 
<p>
Chuck has also implemented a SAGE/Unigene track in the browser. This track displays data from the 
<a href="https://www.ncbi.nlm.nih.gov/SAGE/" target="_blank">SAGEMap project at NCBI</a>. UniGene 
cluster sequences are displayed in the browser and colored according to their average SAGE counts 
over a series of experiments. Selecting one of the UniGene representative sequences displays the 
SAGE results for UniGene sequences.</p>

<h2>Sep. 21, 2001 &nbsp;&nbsp; Clone order problems found in preliminary Aug. 2001 assembly</h2> 
<p>
Some systematic problems were found in the clone order on the preliminary Aug. 2001 freeze assembly.
The sequencing center at Washington University, EBI's Ensembl group, and our group here at UCSC 
are currently working together to revise the merged BAC clone maps and the assembly process to fix 
these problems. We hope to update the August browser with a revised assembly soon. Then, after 
further testing via the browser, we will release the assembled August freeze genome sequence 
itself.</p>

<h2>Sep. 11, 2001 &nbsp;&nbsp; Preliminary Aug. 2001 assembly now available in Genome Browser</h2> 
<p>
A preliminary assembly of the Aug. 2001 freeze is now available in the genome browser. Due to 
significant progress by the mapping and finishing groups of the international public consortium, 
this assembly is a major improvement over the Apr. 2001 freeze assembly. Imre Vastrik, Ewan Birney 
and colleagues at Ensembl have computed a merge of BAC clone maps provided by the individual 
sequencing centers with fingerprint-based maps prepared at Washington University. These merged maps 
were used for the first time in this Aug. assembly.</p> 
<p>
The Aug. assembly has successfully passed our internal quality control tests. We will release the 
sequence and annotations in bulk downloadable form in a week or so, after the external testers have 
had a chance to further verify it. Meanwhile if you notice any systematic problems please let us 
know at 
<a href="mailto:&#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;"> &#103;&#101;n&#111;m&#101;&#64;&#115;&#111;&#101;.&#117;&#99;s&#99;.&#101;d&#117;</a>.  
<!-- above address is genome at soe.ucsc.edu --> 
Though the state of the working draft has improved considerably, remember that where you see solid 
marks in the Gap track, the relative order and orientation of flanking contigs is still uncertain. 
In some cases of complex repeat structure it is also possible that the assembly may be incorrect 
even in the absence of gaps. Also, sometimes ambiguities in the data cause a BAC clone to be split, 
with parts of it placed at opposite ends of a run of other clones. Localized errors of this type 
should be corrected by additional finishing efforts at the individual sequencing centers and should 
not be reported to UCSC. However, please report any large-scale or systematic problems you detect 
with this assembly that could have been caused by our data processing.</p> 
<p>
The tracks available on the August 2001 browser are quite limited at the moment. More tracks will 
show up over time.</p>

<h2>Sep. 11, 2001 &nbsp;&nbsp; Apr. 2001 assembly is new default in Genome Browser</h2> 
<p>
The Apr. 2001 assembly is now the default for the browser. The SNP and Ensembl gene tracks have come
in for this version. There is also a new track depicting non-human vertebrate mRNA alignments.</p> 
<p>
The Sept. 2000 and Jul. 2000 versions of the genome are now only available on our archive site.
Please see the link in the blue box to the left for more details.</p> 
<p>
The Aug. 6 freeze is progressing through the pipeline. We've recently received an updated accession 
map from Wash U.  Ensembl will shortly be integrating this with chromosome specific maps from the 
sequencing centers. We are still on track for an early September next release.</p>

<h2>Aug. 28, 2001 &nbsp;&nbsp; Custom annotation track functionality added</h2> 
<p> 
Meanwhile we've been continuing work on the Genome Browser. It's now possible  to upload your own 
annotations to be displayed alongside the  built-in tracks. Please scroll to the bottom of the 
browser gateway pages for further information. The browser has also been sped up, particularly on 
the larger chromosomes by using a &quot;binning&quot; technique suggested by Lincoln Stein and 
Richard Durbin.</p>

<h2>Aug. 23, 2001 &nbsp;&nbsp; New annotation tracks on Apr. 2001 browser</h2> 
<p>
map. None of the regular chrN.agp or chrN.fa files were affected by this update, nor was any of the 
information in the contigAgp.zip or contigFa.zip files changed. For convenience, we include two new 
files, chromRandAgp.zip and chromRandFa.zip, for users who would like to download only the data that
has changed. These zips consist of the updated chrN_random.agp and chrN_random.fa files, 
respectively.</p>
<p>
Sep. 4, 2000: The files chromFa.zip and contigFa.zip under the July 17th full data set and the files
under July 17th data by individual clone contig were updated to fix some incorrect (null(0)) 
characters that needed to be replaced by 'n' characters in some Fasta files. The following contig 
Fasta files on chromosomes 1,8,11,12,16, 17 and 19 were affected:</p> <pre>
    1/ctg14250/ctg14250.fa
    8/ctg17325/ctg17325.fa
    8/ctg16307/ctg16307.fa
    8/ctg25150/ctg25150.fa
    8/ctg15216/ctg15216.fa
    11/ctg15548/ctg15548.fa
    12/ctg15285/ctg15285.fa
    16/ctg16864/ctg16864.fa
    16/ctg28077/ctg28077.fa
    16/ctg13970/ctg13970.fa
    16/ctg25401/ctg25401.fa
    16/ctg13336/ctg13336.fa
    16/ctg14937/ctg14937.fa
    16/ctg16860/ctg16860.fa
    16/ctg17794/ctg17794.fa
    17/ctg14047/ctg14047.fa
    19/ctg16564/ctg16564.fa </pre>

<p> 
We apologize for any inconvenience this may have caused.</p>

<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->
