<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>

	
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META http-equiv="Content-Script-Type" content="text/javascript">
	<TITLE>
Track and Table Releases	</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">

</HEAD>
<BODY BGCOLOR="#FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">

<TR><TH COLSPAN=1 ALIGN="left">
<IMG SRC="/images/title.jpg">
</TH></TR>


<!-- +++++++++++++++++++++ HOTLINKS BAR +++++++++++++++++++ -->
<TR><TD COLSPAN=3 HEIGHT=40 >
<table bgcolor="#000000" cellpadding="1" cellspacing="1" width="100%%" height="27">
<tr bgcolor="#2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="#2636D1" height="24"><TR>
	 	<TD VALIGN="middle"><font color="#89A1DE">&nbsp;

&nbsp;<A HREF="/index.html" class="topbar">
           Home</A> &nbsp;&nbsp;&nbsp;
       <A HREF="/cgi-bin/hgGateway" class="topbar">
           Genomes</A> &nbsp;&nbsp;&nbsp;
       <A HREF="/cgi-bin/hgBlat?command=start&" class="topbar">           Blat</A> &nbsp;&nbsp;&nbsp;
       <A HREF="/cgi-bin/hgTables" class="topbar">
           Tables</A> &nbsp;&nbsp;&nbsp;
       <A HREF="/cgi-bin/hgNear" class="topbar">
           Gene Sorter</A> &nbsp;&nbsp;&nbsp;
       <A HREF="/cgi-bin/hgPcr" class="topbar">
           PCR</A> &nbsp;&nbsp;&nbsp;
       <A HREF="/FAQ/" class="topbar">
           FAQ</A> &nbsp;&nbsp;&nbsp;

       <A HREF="/goldenPath/help/hgTracksHelp.html"
       class="topbar">
           Help</A> 
&nbsp;</font></TD>
       </TR></TABLE>
</TD></TR></TABLE>
</TD></TR>	


<!-- +++++++++++++++++++++ CONTENT TABLES +++++++++++++++++++ -->
<TR><TD COLSPAN=3>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp;
Track and Table Releases</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
	

 This page contains track and table release information for the following genome assemblies:<br>
<ul>
<li><a CLASS="toc" HREF="#recent" ><b>10 Latest Changes (all assemblies)</b></a></li><li><a CLASS="toc" HREF="#hg17">Human May 2004 (hg17)</a></li>
<li><a CLASS="toc" HREF="#hg16">Human July 2003 (hg16)</a></li>
<li><a CLASS="toc" HREF="#hg15">Human Apr. 2003 (hg15)</a></li>
<li><a CLASS="toc" HREF="#panTro1">Chimp Nov. 2003 (panTro1)</a></li>
<li><a CLASS="toc" HREF="#rheMac2">Rhesus Jan. 2006 (rheMac2)</a></li>
<li><a CLASS="toc" HREF="#rheMac1">Rhesus Jan. 2005 (rheMac1)</a></li>
<li><a CLASS="toc" HREF="#canFam2">Dog May 2005 (canFam2)</a></li>
<li><a CLASS="toc" HREF="#canFam1">Dog July 2004 (canFam1)</a></li>
<li><a CLASS="toc" HREF="#bosTau2">Cow Mar. 2005 (bosTau2)</a></li>
<li><a CLASS="toc" HREF="#bosTau1">Cow Sep. 2004 (bosTau1)</a></li>
<li><a CLASS="toc" HREF="#mm7">Mouse Aug. 2005 (mm7)</a></li>
<li><a CLASS="toc" HREF="#mm6">Mouse Mar. 2005 (mm6)</a></li>
<li><a CLASS="toc" HREF="#mm5">Mouse May 2004 (mm5)</a></li>
<li><a CLASS="toc" HREF="#rn3">Rat June 2003 (rn3)</a></li>
<li><a CLASS="toc" HREF="#rn2">Rat Jan. 2003 (rn2)</a></li>
<li><a CLASS="toc" HREF="#monDom1">Opossum Oct. 2004 (monDom1)</a></li>
<li><a CLASS="toc" HREF="#galGal2">Chicken Feb. 2004 (galGal2)</a></li>
<li><a CLASS="toc" HREF="#xenTro1">X. tropicalis Oct. 2004 (xenTro1)</a></li>
<li><a CLASS="toc" HREF="#danRer3">Zebrafish May 2005 (danRer3)</a></li>
<li><a CLASS="toc" HREF="#danRer2">Zebrafish June 2004 (danRer2)</a></li>
<li><a CLASS="toc" HREF="#danRer1">Zebrafish Nov. 2003 (danRer1)</a></li>
<li><a CLASS="toc" HREF="#tetNig1">Tetraodon Feb. 2004 (tetNig1)</a></li>
<li><a CLASS="toc" HREF="#fr1">Fugu Aug. 2002 (fr1)</a></li>
<li><a CLASS="toc" HREF="#ci2">C. intestinalis Mar. 2005 (ci2)</a></li>
<li><a CLASS="toc" HREF="#ci1">C. intestinalis Dec. 2002 (ci1)</a></li>
<li><a CLASS="toc" HREF="#dm2">D. melanogaster Apr. 2004 (dm2)</a></li>
<li><a CLASS="toc" HREF="#droSim1">D. simulans Apr. 2005 (droSim1)</a></li>
<li><a CLASS="toc" HREF="#droSec1">D. sechellia Oct. 2005 (droSec1)</a></li>
<li><a CLASS="toc" HREF="#dm1">D. melanogaster Jan. 2003 (dm1)</a></li>
<li><a CLASS="toc" HREF="#droYak2">D. yakuba Nov. 2005 (droYak2)</a></li>
<li><a CLASS="toc" HREF="#droYak1">D. yakuba Apr. 2004 (droYak1)</a></li>
<li><a CLASS="toc" HREF="#droEre1">D. erecta Aug. 2005 (droEre1)</a></li>
<li><a CLASS="toc" HREF="#droAna2">D. ananassae Aug. 2005 (droAna2)</a></li>
<li><a CLASS="toc" HREF="#droPer1">D. persimilis Oct. 2005 (droPer1)</a></li>
<li><a CLASS="toc" HREF="#dp3">D. pseudoobscura Nov. 2004 (dp3)</a></li>
<li><a CLASS="toc" HREF="#dp2">D. pseudoobscura Aug. 2003 (dp2)</a></li>
<li><a CLASS="toc" HREF="#droVir2">D. virilis Aug. 2005 (droVir2)</a></li>
<li><a CLASS="toc" HREF="#droVir1">D. virilis July 2004 (droVir1)</a></li>
<li><a CLASS="toc" HREF="#droMoj2">D. mojavensis Aug. 2005 (droMoj2)</a></li>
<li><a CLASS="toc" HREF="#droMoj1">D. mojavensis Aug. 2004 (droMoj1)</a></li>
<li><a CLASS="toc" HREF="#droGri1">D. grimshawi Aug. 2005 (droGri1)</a></li>
<li><a CLASS="toc" HREF="#droAna1">D. ananassae July 2004 (droAna1)</a></li>
<li><a CLASS="toc" HREF="#apiMel2">A. mellifera Jan. 2005 (apiMel2)</a></li>
<li><a CLASS="toc" HREF="#apiMel1">A. mellifera July 2004 (apiMel1)</a></li>
<li><a CLASS="toc" HREF="#anoGam1">A. gambiae Feb. 2003 (anoGam1)</a></li>
<li><a CLASS="toc" HREF="#ce2">C. elegans Mar. 2004 (ce2)</a></li>
<li><a CLASS="toc" HREF="#ce1">C. elegans May 2003 (ce1)</a></li>
<li><a CLASS="toc" HREF="#cb1">C. briggsae July 2002 (cb1)</a></li>
<li><a CLASS="toc" HREF="#strPur1">S. purpuratus April 2005 (strPur1)</a></li>
<li><a CLASS="toc" HREF="#sacCer1">S. cerevisiae Oct. 2003 (sacCer1)</a></li>
<li><a CLASS="toc" HREF="#sc1">SARS Apr. 2003 (sc1)</a></li>
<li><a CLASS="toc" HREF="#zoo1">NISC (Zoo) Jun. 2002 (zoo1)</a></li>
</ul>
<p>
 For more information about the tracks and tables listed on this page, refer to the <a href=/goldenPath/gbdDescriptions.html>Description of the annotation database</a> and the <a href=/goldenPath/help/hgTracksHelp.html#IndivTracks>User's Guide</a>.<p>
<em>Last updated 05 Apr 2006. <a HREF="mailto:genome@soe.ucsc.edu">Inquiries and feedback welcome</a>.</em>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=recent></A> 10 Latest Changes (all assemblies)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR>
<TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
<TD nowrap><FONT color="#006666"><B>Assembly</B></FONT></TD>
<TD nowrap><FONT color="#006666"><B>Release Date</B></FONT></TD>
</TR>
<TR valign=top><TD align=left>
Most Conserved (correction to make rhesus non-informative)</td>
<td>hg17</td>
<td>28 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
Conservation - Data Update (rhesus made non-informative)</td>
<td>hg17</td>
<td>22 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
quality scores for rheMac2</td>
<td>rheMac2</td>
<td>20 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
VisiGene (Allen Brain Atlas)</td>
<td>mm7, mm6, hg17</td>
<td>17 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
BosTau2, DanRer3 chains/net/liftOver</td>
<td>hg17</td>
<td>15 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
Most Conserved (17-way alignment phastCons conserved elements)</td>
<td>hg17</td>
<td>10 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
Conservation (17-way vertebrate multiple alignment)</td>
<td>hg17</td>
<td>10 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>canFam1</td>
<td>07 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
Initial strPur1 release (using Baylor Spur_0.5 whole genome shotgun assembly) Assembly, gap, RepeatMasker, SimpleRepeats, GC Percent, S. purpuratus mRNAs, Other mRNAs, Genscan, Mapped Human Proteins, Quality, RefSeq.</td>
<td>strPur1</td>
<td>07 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
RheMac2 chain/net/liftOver</td>
<td>hg17</td>
<td>06 Mar 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=hg17></A>Human May 2004 (hg17, NCBI Build 35)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Most Conserved (correction to make rhesus non-informative)</td>
<td>28 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
Conservation - Data Update (rhesus made non-informative)</td>
<td>22 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
VisiGene (Allen Brain Atlas)</td>
<td>17 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
BosTau2, DanRer3 chains/net/liftOver</td>
<td>15 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
Most Conserved (17-way alignment phastCons conserved elements)</td>
<td>10 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
Conservation (17-way vertebrate multiple alignment)</td>
<td>10 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
RheMac2 chain/net/liftOver</td>
<td>06 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
sno/miRNA (wgRna)</td>
<td>21 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
HapMap LD (release 20)</td>
<td>21 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Improved searching capability for VisiGene</td>
<td>16 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Improved searching capability for Known Genes</td>
<td>03 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
gladHumESDistance (data update)</td>
<td>01 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Superfamily (data update)</td>
<td>30 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Comparatives downloads (replacements for overlapping axtNet alignments)</td>
<td>18 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP Sites (data replaced with 5 new transcription factors). Previous track Yale ChIP Sites renamed Yale STAT1 Sites.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP Signal (data replaced with 5 new transcription factors). Previous track Yale ChIP Signal renamed Yale STAT1 Signal.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP pVal (data replaced with 5 new transcription factors). Previous track Yale ChIP pVal renamed Yale STAT1 pVal.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE EGASP Full (Correction)</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Hgdownloads update: goldenPath/hg17/vs{PanTro1,CanFam1,Fr1,DanRer1,Mm5}/axtNet/{chr5.axt.gz,md5sum.txt}</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE EGASP Partial (data correction)</td>
<td>11 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Recomb Hotspots: ENCODE recombination hotspots and rates</td>
<td>06 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated hg17 -> mm6 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>06 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Recomb Rates and Hotspots: from HapMap and Perlegen data</td>
<td>04 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: fr1, tetNig1, xenTro1, galGal2, bosTau2, rheMac1, panTro1, hg17, rn3 -> mm7; fr1, tetNig1, xenTro1, galGal2, panTro1 -> mm6; galGal2 -> danRer2, mm5, hg17, monDom1, tetNig1, xenTro1; rheMac1 -> hg17; panTro1 -> hg16, hg17</td>
<td>03 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Uppsala Chip (University of Uppsala, Sweden ChIP/chip coordinate corrections)</td>
<td>03 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale TAR (Replace June data with new methods)</td>
<td>02 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UVa DNA Rep Ori (University of Virginia DNA Replication Origins, Oct 2005 Freeze)</td>
<td>29 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UT-Austin STAGE (Sequence Tag Analysis of Genomic Enrichment), c-Myc in HeLa cells</td>
<td>28 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Dog Chain/Net/LiftOver (canFam2)</td>
<td>22 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy Transfrags (updated HeLa data)</td>
<td>21 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy RNA Signal (updated HeLa data)</td>
<td>21 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Cow Synteny</td>
<td>20 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
mRNA, RefSeq mRNA, and native ESTs: Realigned and filtered
-- improvements in process.</td>
<td>16 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UVa DNA Rep TR50 (University of Virginia DNA Smoothed Timing at 50% Replication, Oct 2005 Freeze)</td>
<td>16 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UVa DNA Rep Seg (University of Virginia DNA Replication Temporal Segmentation, Oct 2005 Freeze)</td>
<td>02 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Sanger ChIP Hits (Peaks and Centers, Oct 2005 Freeze)</td>
<td>22 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Sanger ChIP (H3, H4 histones in GM, K562 and HeLa cells, Oct 2005 Freeze)</td>
<td>22 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
 ENCODE UNC FAIRE (Formaldehyde Assisted Isolation of Regulatory Elements, Oct 2005 Freeze)
</td>
<td>22 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UCD Ng ChIP (UC Davis ChIP/Chip of E2F1, Myc in HeLa, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Uppsala Chip (University of Uppsala, Sweden ChIP/chip of H3ac,HNF3b,HNF4a,USF1 in HepG2 cells, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy Strict Sites (Affymetrix ChIP/chip of 5 factors in HL60 cells, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy Strict (Affymetrix ChIP/chip of 5 factors in HL60 cells, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy Strict Pval (Affymetrix ChIP/chip on 5 factors, in HL60 cells, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
Allen Brain Atlas Images</td>
<td>10 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy RNA (RNA abundance and transfrags, Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UT-Austin ChIP (c-Myc and EF24 in HeLa and 20901 fibroblasts, Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI DnaseI-HS (MPSS and Chip methods, in CD4 and GM09660, Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE GIS PET RNA (update Oct 2005 Freeze and new estrogen-treated MCF7)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENODE Genocde Introns (Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gencode Genes (Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
Gladstone arrays -- track and Gene Sorter columns</td>
<td>04 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
sno/miRNA (data update)</td>
<td>27 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE LI Ng gIF ChIP (Addition of 5 antibodies, in HeLa cells)</td>
<td>27 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
Oxford Recombination Hotspots (from HapMap phase I, release 16a)</td>
<td>25 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
Oxford Recombination Rates (from HapMap phase I, release 16a)</td>
<td>25 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE EGASP Update (Add Fgenesh++ corrected predictions)</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
Affy Transcription Phase 2</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
Exon Walk</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
Exoniphy</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
AffyHuEx1.0 (Probe Set locations for Affymetrix Human Exon Genechip)</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
NHGRI DNaseI-HS</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE GIS STAT1 +/- gIF HeLa</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN Gene Predictions</td>
<td>04 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UVa DNA Rep Seg, UVa DNA Rep Ori (DNA replication timing segmentation and origins)</td>
<td>04 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
EvoFold (removed non-syntenic predictions and predictions with only 4 pairing bases)
</td>
<td>03 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
targetScanS miRNA</td>
<td>30 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE YALE MAS TAR (additional data)</td>
<td>27 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE YALE MAS RNA (additional data from June freeze)</td>
<td>27 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
Tajima's D</td>
<td>23 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
5x Reg Potential</td>
<td>22 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE BU Orchid</td>
<td>21 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI DIPs: Whole Genome Shotgun insertion/deletion polymorphisms from Jim Mullikin/SSAHA</td>
<td>21 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: LI Ng gIF ChIP (HeLa)</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: LI Ng Validation (Pol2, Taf1, H3K4me, H3ac)</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Sanger ChIP/chip</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Ludwig Institute/UCSD ChIP/Chip - Pol II, TAF1, AcH3, H3K4me, SUZ12, and H3K27me antibodies - repackage three existing tracks into one</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP/Chip (STAT1 ab, Hela cells, P-Values)</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: LI gIF ChIP (HeLa)
</td>
<td>19 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE HapMap Allele Frequencies</td>
<td>19 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: GIS P53 5FU HCT116</td>
<td>19 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENENCODE: Affy Transfrags, Affy pVal {0,2,8,32}h, Affy Sites {0,2,8,32}h</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Affy RNA Signal </td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale MAS array, RNA Transcript Map, Placenta, Nimblegen Protocol</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale MAS array, RNA Transcriptionally Active Regions, Placenta, Nimblegen Protocol</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP/Chip (STAT1 ab, Hela cells, Signal)</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP/Chip (STAT1 ab, Hela cells, Binding Sites)</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale TAR </td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
HapMap Coverage: Depth of resequencing for 10 encode regions (work done by Broad and Baylor for the HapMap project)</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Stanford Endogenous Transcript Levels in HCT116 Cells; 
Stanford ChIP/chip (HCT116, Jurkat, K562 cells; Sp1, Sp3 ChIP); 
Stanford ChIP/chip Smoothed Score; 
Stanford Methylation Digest: Be2C, CRL1690, HCT116, HT1080, HepG2, JEG3, Snu182, U87; 
Stanford Methylation Digest Smoothed Score</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Stanf Promoter</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: BU First Exon</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale RNA </td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Sanger Assoc</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: UC Davis ChIP/Chip Nimblegen - E2F1 ab, HeLa Cells</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: GIS RNA PET</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: TBA23 EvoFold (Predictions of RNA Secondary Structure Using TBA23, ENCODE Regions only)</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Known+Pred RNA (Known and Predicted RNA Transcription)</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: UW/Regulome DNase I Hypersensitivity Probability</td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Regulome DNase I Hypersensitivity</td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
UW/Regulome Plate Quality</td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
Regulome DNAse I Hypersensitivity Amplicons</td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: NHGRI DNaseI-HS </td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Riken CAGE - predicted gene start sites</td>
<td>13 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
Updated AceView (acembly) track from NCBI.</td>
<td>12 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gene Prediction Workshop (EGASP) Updates</td>
<td>12 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
EGASP Partial</td>
<td>31 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Sangamo/EIO CD34 DNase Hypersensitivity</td>
<td>30 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
EGASP Full</td>
<td>29 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Gencode Introns</td>
<td>29 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Gencode Genes</td>
<td>29 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Aternative Splicing</td>
<td>29 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
sno/miRNA (data update)</td>
<td>26 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter (updated to pick up Pfam Domains for splice variants in new Known Genes)</td>
<td>12 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Reloaded table for new known genes</td>
<td>03 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter: RankProp Score (table reloaded for new Known Genes)</td>
<td>01 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter support</td>
<td>29 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Old Known Genes (Table Browser only)</td>
<td>28 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Rhesus macaque (rheMac1) Chain and Net</td>
<td>25 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
bosTau1ToHg17 and hg17ToBosTau1 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>22 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
rheMac1ToHg17 and hg17ToRheMac1 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>22 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
EvoFold (Predictions of RNA Secondary Structure)</td>
<td>15 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
sno/miRNA (data update)</td>
<td>13 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
augustus gene predictions</td>
<td>07 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
hg17ToXenTro1, hg17ToGalGal2, hg17ToMonDom1, hg17ToTetNig1, hg17ToDanRer2 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>06 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
hg17ToCanFam1 and canFam1ToHg17 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>05 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Known Genes, Gene Sorter, Proteome Browser (data update, new  filtering process)</td>
<td>15 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter (humanNormal microarray data)</td>
<td>02 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
hg17ToMm6 liftOver (Downloads and browser Batch Coordinate Conversion utility)</td>
<td>01 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
hg17ToRn3 liftOver (Downloads and browser Batch Coordinate Conversion utility)</td>
<td>01 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Restriction Enzyme mapping (REbase release 505).</td>
<td>24 May 2005</td>
</tr>
<TR valign=top><TD align=left>
vega and vegaPseudogene</td>
<td>10 May 2005</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
wgRna (data update)</td>
<td>26 Apr 2005</td>
</tr>
<TR valign=top><TD align=left>
Retroposed Genes</td>
<td>21 Apr 2005</td>
</tr>
<TR valign=top><TD align=left>
add table ccdsKgMap to link ccdsGene to knownGenes</td>
<td>20 Apr 2005</td>
</tr>
<TR valign=top><TD align=left>
Cow BAC Ends</td>
<td>31 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Consensus CDS (X/Y PAR region corrected)</td>
<td>22 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
SNPs (data update: Affy genotyping SNPs)</td>
<td>11 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
CCDS genes (enabled robust details page)</td>
<td>04 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Non-Human RefSeqs</td>
<td>04 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
STS Markers (updated mapping)</td>
<td>28 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Yale Pseudo (Processed Pseudogenes from yale)</td>
<td>25 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Cow Chain/Net</td>
<td>18 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Ecores</td>
<td>15 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
RGD QTL (data update)</td>
<td>14 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Bounds
(Gene Boundaries as Defined by RNA and Spliced EST Clusters)</td>
<td>10 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Acembly (AceView Gene Models)</td>
<td>07 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
TFBS Conserved (transcr. factor binding sites)</td>
<td>04 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
SNPs (dbSNP build 124)</td>
<td>01 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
SNP track enhancement to support exceptions from dbSnp. (dbSnp build 123)</td>
<td>20 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
sno/miRNA - C/D and H/ACA Box snoRNAs, scaRNAs, and microRNAs from Weber and Griffiths-Jones (replaces the miRNA track)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish (danRer2) Chain and Net</td>
<td>06 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter extension (rankProp, psi-blast)</td>
<td>23 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>21 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Proteome Browser (data update)</td>
<td>18 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (data update)</td>
<td>18 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter (data update)</td>
<td>18 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA data correction for hsa-mir-30a</td>
<td>15 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
SNPs (track based on 'snp' table, not 'snpMap' table)</td>
<td>05 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>24 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
X. tropicalis Net</td>
<td>22 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
X. tropicalis Chain</td>
<td>22 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene new track, EST clustering gene prediction</td>
<td>22 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
WSSD Duplication (from Xinwei She and Evan Eichler)</td>
<td>29 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Segmental Dups (from Xinwei She and Evan Eichler)</td>
<td>29 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
UniGene version 171 added to Human browser</td>
<td>20 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Affy U133Plus2</td>
<td>13 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
RDG QTL</td>
<td>13 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (add 3', 5' UTR mRNA secondary structure predictions)</td>
<td>11 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (add BioCyc pathways links)</td>
<td>11 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>01 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (data update), Most Conserved added</td>
<td>21 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Self Chain</td>
<td>21 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Expression Distance (expanded data set)</td>
<td>16 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Regions</td>
<td>15 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Chain, Net</td>
<td>13 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
MicroRNAs from the miRNA Registry</td>
<td>09 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan gene predictions</td>
<td>08 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Retroposed Genes</td>
<td>01 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
fishClones</td>
<td>13 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
H-Inv (Human gene database, with full-length cDNA clones)</td>
<td>13 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Expression Distance</td>
<td>11 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Gene Maps</td>
<td>11 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Fundamentals</td>
<td>11 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid</td>
<td>10 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial hg17 release (using NCBI Build 35): Chromosome Band, STS Markers, FISH Clones, Recomb Rate, Map Contigs, Assembly, Gap, Coverage, BAC End Pairs, Fosmid End Pairs, GC Percent, Known Genes, Proteome Browser, RefSeq Genes, MGC Genes, Genscan Genes, Human mRNAs, Spliced ESTs, Human ESTs, NonHuman mRNAs, NonHuman ESTs, CpG Islands, Fugu Blat, 
Conservation, Fugu Chain, Fugu Net, Zebrafish Chain, Zebrafish Net, Chicken Chain, Chicken Net, Rat Chain, Rat Net, Mouse Chain, Mouse Net, Dog Chain, Dog Net, Chimp Chain, Chimp Net, RepeatMasker, Simple Repeats.

 Downloads: bigZips, chromosomes, database</td>
<td>23 Jul 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=hg16></A>Human July 2003 (hg16, NCBI Build 34)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
HapMap LD (Phase II; October 2005 data)</td>
<td>08 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Improved searching capability for Known Genes</td>
<td>03 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE EGASP Full (Correction)</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Recomb Hotspots: ENCODE recombination hotspots and rates</td>
<td>06 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: hg16 -> galgal2, mm4, mm5, mm6, panTro1, pt0, rn3</td>
<td>04 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Recomb Rates and Hotspots: from HapMap and Perlegen data</td>
<td>04 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE GIS PET RNA (update)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
Oxford Recombination Hotspots (from HapMap phase I, release 16a)</td>
<td>25 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
Oxford Recombination Rates (from HapMap phase I, release 16a)</td>
<td>25 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
AffyHuEx1.0 (Probe Set locations for Affymetrix Human Exon Genechip)</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
NHGRI DNaseI-HS</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
EvoFold (removed non-syntenic predictions and predictions with only 4 pairing bases)
</td>
<td>03 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE YALE MAS TAR (additional data)</td>
<td>27 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE YALE MAS RNA (additional data from June freeze)</td>
<td>27 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
Tajima's D</td>
<td>23 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
Gencode Unannotated Region Classification</td>
<td>30 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Consensus Unannotated Transfrags</td>
<td>30 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Yale and Affymetrix Unannotated TransFrags</td>
<td>30 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
EGASP Full</td>
<td>26 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix 25 bp Probe Locations</td>
<td>18 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Yale RNA reloaded to fix off-by-one coordinates</td>
<td>08 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Yale RNA reloaded to fix off-by-one coordinates</td>
<td>08 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Yale TAR reloaded to fix off-by-one coordinates</td>
<td>08 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Affy RNA Signal reloaded to fix off-by-one coordinates</td>
<td>08 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Affy pVal reloaded to fix off-by-one coordinates</td>
<td>08 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
CNPs; Copy Number Polymorphisms</td>
<td>29 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Mlagan Alignment (data update)</td>
<td>26 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE All Introns Proximal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE All Introns Distal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE All Intergenic Proximal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE All Intergenic Distal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Introns Proximal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Introns Distal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Intergenic Proximal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Intergenic Distal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
TBA23 EvoFold (Predictions of RNA Secondary Structure Using TBA23, ENCODE Regions only)</td>
<td>15 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
EvoFold (Predictions of RNA Secondary Structure)</td>
<td>15 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
HapMap Coverage: Depth of resequencing for 10 encode regions (work done by Broad and Baylor for the HapMap project)</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Regulome DNAse I Hypersensitivity Amplicons</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
UW/Regulome Plate Quality</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
UW/Regulome DNase I Hypersensitivity Probability</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Regulome DNase I Hypersensitivity</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Consensus Elements: TBA and MLAGAN Union/Intersection</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford LAGAN Alignments</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: NHGRI/PSU TBA Alignments</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Recombination in 10 ENCODE regions (cM/Mb)</td>
<td>13 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford LAGAN Conserved Elements </td>
<td>11 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford LAGAN Conservation</td>
<td>11 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Primate Alleles</td>
<td>08 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Prediction Workshop (EGASP) for Partial ENCODE Regions</td>
<td>06 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Prediction Workshop (EGASP) All ENCODE Regions</td>
<td>06 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Stanford Endogenous Transcript Levels in HCT116 Cells; 
Stanford ChIP/chip (HCT116, Jurkat, K562 cells; Sp1, Sp3 ChIP); 
Stanford ChIP/chip Smoothed Score; 
Stanford Methylation Digest: Be2C, CRL1690, HCT116, HT1080, HepG2, JEG3, Snu182, U87; 
Stanford Methylation Digest Smoothed Score</td>
<td>06 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Riken CAGE - predicted gene start sites</td>
<td>06 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gene Prediction Workshop (EGASP) Updates</td>
<td>05 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE HapMap Allele Frequencies</td>
<td>05 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
H3K4me3/K562 new subtrack added to Sanger ChIP</td>
<td>01 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Sanger Assoc</td>
<td>01 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Whole Genome Shotgun insertion/deletion polymorphisms from Jim Mullikin/SSAHA</td>
<td>01 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale MAS array, RNA Transcriptionally Active Regions, Placenta, Nimblegen Protocol</td>
<td>27 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale MAS array, RNA Transcript Map, Placenta, Nimblegen Protocol</td>
<td>27 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Affy RNA Signal, Affy Transfrags, Affy pVal {0,2,8,32}h, Affy Sites {0,2,8,32}h</td>
<td>17 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale RNA Transcriptionally Active Regions (TARs)</td>
<td>17 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale RNA Transcript Map (Neutrophil, Placenta and NB4 cells)</td>
<td>17 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP/Chip (STAT1 ab, Hela cells, P-Values)</td>
<td>15 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP/Chip (STAT1 ab, Hela cells, Binding Sites)</td>
<td>15 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Boston University ORChID</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: UC Davis ChIP/Chip Nimblegen - E2F1 ab, HeLa Cells</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP/Chip (STAT1 ab, Hela cells, Signal)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Gencode Introns (Intron validation for selected Gencode Genes)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Gencode Genes (Manual annotations in the ENCODE regions from the Gencode project)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Retroposed Genes (data update)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: ChIP-PET/GIS (data update, has cluster counts)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Ludwig Institute/UCSD ChIP/Chip - Pol II, TAF1, AcH3, H3K4me, SUZ12, and H3K27me antibodies - repackage three existing tracks into one</td>
<td>13 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Ludwig Institute/UCSD ChIP/Chip - gamma interferon experiments</td>
<td>13 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: GIS RNA PET</td>
<td>10 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: NHGRI DNaseI-HS (data update; now includes sites from activated as well as unactivated cells)</td>
<td>10 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Sanger ChIP/Chip (histones H3,H4 GM06990)</td>
<td>10 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix ChIP/Chip (Pol2 RA-stim HL60) Signal in ENCODE Regions (changed from wig to bedGraph)</td>
<td>08 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix ChIP/Chip (Pol2 RA-stim HL60) PValue in ENCODE Regions (changed from wig to bedGraph)</td>
<td>08 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Stanf. Promoter in ENCODE regions (data update; changed to composite track -  old name: Promoter/Stanford) </td>
<td>06 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
DNA Rep/UVA in ENCODE regions (reloaded with new data)</td>
<td>03 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP-PET/GIS (P53 TFBS in ENCODE regions)</td>
<td>24 May 2005</td>
</tr>
<TR valign=top><TD align=left>
vega and vegaPseudogene</td>
<td>10 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Promoters/BU (Boston Univ. Promoters in ENCODE regions)</td>
<td>07 May 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UCSD/Ludwig Chip (acetylated and methylated histone binding)</td>
<td>06 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Mouse (March 2005/mm6) Alignment Net</td>
<td>04 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Mouse (March 2005/mm6) Chained Alignments</td>
<td>04 May 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UCSD Ludwig RNAP, ENCODE UCSD Ludwig TAF1 (tracks combined into composite format)</td>
<td>29 Apr 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIp/Sanger (Lysine 4 methylation of histone in ENCODE regions)</td>
<td>07 Apr 2005</td>
</tr>
<TR valign=top><TD align=left>
Updates hg16 SNPs to dbSnp build 122, including Affy genotyping SNPs.  </td>
<td>14 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Yale Pseudo (Processed Pseudogenes from yale)</td>
<td>25 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
RNA/Affy Sites (Affymetrix PolyA+ RNA from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
RNA/Affy Signal (Affymetrix PolyA+ RNA abundance from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP/Affy Pol2 Sites (Affymetrix ChIP/Chip of Pol2 from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP/Affy Pol2 Pval (Affymetrix ChIP/Chip enrichment of Pol2 from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP/chip data in ENCODE regions</td>
<td>05 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
DNA Rep/UVA (ENCODE DNA Replication data from Dutta Lab, University of VIrginia)</td>
<td>03 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
Promoter/Stanford (Promoters in ENCODE regions, from Myers lab at Stanford)</td>
<td>02 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
DNaseI-HS/NHGRI (ENCODE DNaseI Hypersensitive sites, from Collins lab, NHGRI)</td>
<td>01 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
WSSD Overlay (from Xinwei She and Evan Eichler)</td>
<td>29 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
WSSD Coverage (from Xinwei She and Evan Eichler)</td>
<td>29 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
WSSD Duplication (from Xinwei She and Evan Eichler)</td>
<td>29 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Segmental Dups (from Xinwei She and Evan Eichler)</td>
<td>29 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Perlegen Haplotypes (8 records changed to correct block ordering)</td>
<td>11 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl EST Genes</td>
<td>10 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Vega Gene: data update (extending 3' UTR for 398 annotations).</td>
<td>23 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan: data update (additional exons added for 27 annotations).</td>
<td>20 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes: data update (64 annotations changed by 1 or 2 bases).</td>
<td>20 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Exoniphy: Conserved Exon Predictions</td>
<td>14 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Retroposed Genes</td>
<td>19 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
H-Inv (Full-length cDNA from the H-Invitational Human Gene Annotation Database)</td>
<td>08 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
ensTranscript and ensGeneXref tables dropped (replaced by
ensemblXref3)  </td>
<td>03 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Gene (updated to 34d), Superfamily (updated)</td>
<td>03 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
GC% Win20K (dropped - replaced by 5-base window GC Percent track)</td>
<td>01 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Alt-Splicing (data update)</td>
<td>27 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Chain, Fugu Net (using fr1)</td>
<td>24 May 2004</td>
</tr>
<TR valign=top><TD align=left>
STS Markers (updated)</td>
<td>24 May 2004</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>21 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Chimp Recip Chain, Chimp Recip Net (updated to panTro1 assembly)
</td>
<td>21 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Chimp Chain (updated to panTro1 assembly)</td>
<td>21 May 2004</td>
</tr>
<TR valign=top><TD align=left>
RNA Genes</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
GC Percent (updated to 5-base window track)</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
TFBS Conserved</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Ecores</td>
<td>03 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Microsatellites</td>
<td>03 May 2004</td>
</tr>
<TR valign=top><TD align=left>
jaxOrtholog table</td>
<td>30 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Chicken Chain, Chicken Net (using galGal2)</td>
<td>29 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Ecores</td>
<td>29 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Acembly</td>
<td>20 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
SNPs (updated to dbSNP Build 120)</td>
<td>20 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
RNA Genes</td>
<td>13 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Hu/Mouse/Rat/Chick removed (replaced by Hu/Chp/Mo/Ra/Chk)</td>
<td>12 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Hu/Chp/Mo/Ra/Chk</td>
<td>12 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (updated to use proteins040315)</td>
<td>09 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix U133 (updated)</td>
<td>09 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF Atlas 2</td>
<td>09 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix GNF1H</td>
<td>09 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes (updated to include new peptide table)</td>
<td>05 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse/Rat removed (replaced by Hu/Mouse/Rat/Chick)</td>
<td>29 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Hu/Mouse/Rat/Chick (using mm3, rn3, galGal2) - replaces Human/Mouse/Rat track</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes (updated)</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP (updated to use mm4)</td>
<td>05 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Map Contigs (updated)</td>
<td>02 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Perlegen (updated)</td>
<td>02 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>13 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
affyU133, affyU95 tables</td>
<td>10 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (updated)</td>
<td>06 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
2X Reg Potential, 3X Reg Potential</td>
<td>05 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes (update)</td>
<td>04 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
Mm3 Chain, Mm3 Net removed (replaced by mm4 versions)</td>
<td>20 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
chr{4,8,X,Un}_random_gap, chr{4,8,X,Un}_random_gold tables (updated)</td>
<td>16 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Chimp Diff</td>
<td>16 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Affy10K</td>
<td>14 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Mm4 Net (updates Mouse Net using mm4)</td>
<td>12 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Mm4 Chain (updates Mouse Chain using mm4)</td>
<td>09 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Mm4 Tight (updates Mouse Tight using mm4)</td>
<td>08 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes</td>
<td>22 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
vsPt0/axtBest downloads (chimp reciprocal best)</td>
<td>22 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
Alt-Splicing</td>
<td>18 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan Genes</td>
<td>18 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
Chimp All Chains, Chimp Net, Chimp sequence and chain downloads</td>
<td>10 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs (updated)</td>
<td>19 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>18 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>18 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Chain Self</td>
<td>15 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
UniGene_2, sage</td>
<td>12 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Bounds</td>
<td>12 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Chain Self</td>
<td>11 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Vega Genes, Vega Pseudogenes</td>
<td>07 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
NCI60</td>
<td>24 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (add UTR 3' and 5' folding data)</td>
<td>24 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Haplotype Blocks, Perlegen (chr 21 and 22 only)</td>
<td>24 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse/Rat downloads</td>
<td>24 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Synteny</td>
<td>21 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Genbank daily updates enabled</td>
<td>14 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Affy 120K SNPs</td>
<td>13 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Random SNPs, Overlap SNPs (dbSnp 117)</td>
<td>10 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse/Rat (using mm3 and rn3)</td>
<td>09 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (update)</td>
<td>08 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Fugu Blat (v. 3.0)</td>
<td>02 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
affyRatio table</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
vsMm3, vsRn3 data downloads</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Geneid</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain (using mm3)</td>
<td>10 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Net (using mm3) (contains fixes to chr1 data)</td>
<td>10 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Tight (using mm3) (contains fixes to chr1 data)</td>
<td>10 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
First EF (contains fixes to chrX and chrY)</td>
<td>09 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Net (using mm3)</td>
<td>02 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat Net (using rn3)</td>
<td>02 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat Chain</td>
<td>29 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain</td>
<td>29 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Tight Mouse</td>
<td>29 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Self Chain</td>
<td>29 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes</td>
<td>26 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
STS Markers (updated)</td>
<td>11 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial hg16 release (using NCBI Build 34): Chromosome Band, FISH Clones, Assembly, Gap, STS Markers, Map Contigs, Recomb Rate, GC Percent, Coverage, BAC End Pairs, Fosmid End Pairs, RefSeq Genes, MGC Genes, Genscan Genes, Human mRNAs, Spliced ESTs, Human ESTs, NonHuman mRNAs, NonHuman ESTs, CpG Islands, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosomes, database</td>
<td>08 Aug 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=hg15></A>Human Apr. 2003 (hg15, NCBI Build 33)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Improved searching capability for Known Genes</td>
<td>03 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Cow Synteny</td>
<td>20 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Affy Txn Phase2</td>
<td>03 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Cow Synteny</td>
<td>31 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Cow BAC Ends</td>
<td>31 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Transcription and Transfrags</td>
<td>13 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Retroposed Genes</td>
<td>22 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
RNA Genes</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
TFBS Conserved</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF Ratio (updated)</td>
<td>14 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF Ratio (updated)</td>
<td>05 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes (updated)</td>
<td>05 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (updated to use proteins040315)</td>
<td>05 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes (update)</td>
<td>30 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Affy10K</td>
<td>14 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Random SNPs, Overlap SNPs (dbSnp 112)</td>
<td>31 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Affy 120K SNPs</td>
<td>13 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
KnownGenes (extended)</td>
<td>08 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
affyRatio table</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
ensPep table</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Segmental Dups</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
First EF (contains fixes to chrX and chrY)</td>
<td>09 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
First EF</td>
<td>27 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Downloads: Mouse (mm3)/Rat (rn2)/Human multiple alignments</td>
<td>22 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
NonHuman ESTs</td>
<td>25 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>15 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Cons (using mm3)</td>
<td>09 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Best Mouse (using mm3)</td>
<td>09 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
bigZips (upstream files updated)</td>
<td>09 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Blastz axtTight mouse alignments (using mm3)</td>
<td>25 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Acembly</td>
<td>05 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
AxtInfo (updated to use axtNetMm3)</td>
<td>30 May 2003</td>
</tr>
<TR valign=top><TD align=left>
UniGene (v. Hs 160)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Tight Rat (using rn2)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Bounds</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat Synteny (using rn2)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse/Rat (using mm3, rn2)</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
MGC Genes</td>
<td>27 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Fosmid End Pairs</td>
<td>20 May 2003</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes (using mm3)</td>
<td>20 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Haplotype Blocks (chr 21 and 22 only)</td>
<td>20 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Overlap and Random SNPs (dbSnp 114)</td>
<td>15 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Blastz axtBest mouse alignments (using mm3)</td>
<td>13 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse Chain and Net (using mm3)</td>
<td>09 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>08 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan (using mm2)</td>
<td>08 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Recomb Rate</td>
<td>01 May 2003</td>
</tr>
<TR valign=top><TD align=left>
geneBands (table only)</td>
<td>30 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
NCI60</td>
<td>30 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
GNF Ratio</td>
<td>30 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
GC Percent (data reloaded)</td>
<td>29 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Chromosome Band</td>
<td>28 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>28 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
FISH Clones</td>
<td>28 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>25 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Coverage</td>
<td>25 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
liftAll.zip (correction for NT_008183 - table only)</td>
<td>16 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial hg15 release (using NCBI Build 33): Assembly, Gap, STS Markers, Map Contigs, GC Percent, Known Genes, RefSeq Genes, Genscan Genes, Human mRNAs, Spliced ESTs, Human ESTs, NonHuman mRNAs, CpG Islands, Fugu Blat (v 3.0), Tight Mouse (mm3), RepeatMasker, Simple Repeats, Mouse Synteny (mm3)</td>
<td>14 Apr 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=panTro1></A>Chimp Nov. 2003 (panTro1, CGSC Build 1 Version 1 Arachne assembly)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
LiftOvers: fr1, tetNig1, xenTro1, galGal2, bosTau2, rheMac1, panTro1, hg17, rn3 -> mm7; fr1, tetNig1, xenTro1, galGal2, panTro1 -> mm6; galGal2 -> danRer2, mm5, hg17, monDom1, tetNig1, xenTro1; rheMac1 -> hg17; panTro1 -> hg16, hg17</td>
<td>03 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>22 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Blat</td>
<td>14 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>10 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Contigs</td>
<td>28 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>27 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Deletions (updated)</td>
<td>14 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes (chr1 updated)</td>
<td>07 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>24 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Best, vsHg15 downloads</td>
<td>02 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial panTro1 Release (using the Build 1 Version 1 Arachne assembly dated 13 Nov. 2003): Scaffolds, Gap, GC Percent, Quality Scores, Human RefSeq, Chimp mRNAs, Spliced ESTs, Chimp ESTs, NonChimp mRNAs, NonChimp ESTs, CpG Islands, Human Chain, Human Net, Human Deletions, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosome, database.</td>
<td>23 Feb 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=rheMac2></A>Rhesus Jan. 2006 (rheMac2, Venter Inst./Baylor/WashU Merged assembly Jan 2006 v1 edit4)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
quality scores for rheMac2</td>
<td>20 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
Contigs. Table update.  Added missing records.</td>
<td>16 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Initial rhesus macaque browser release (Macaca mulatta preliminary assembly, Baylor, rheMac2) Assembly, Gap, RepeatMasker, Simple Repeats, Other RefSeq, Other mRNAs, Spliced ESTs, Rhesus mRNAs, Rhesus ESTs</td>
<td>13 Feb 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=rheMac1></A>Rhesus Jan. 2005 (rheMac1, Baylor v0.1)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
LiftOvers: fr1, tetNig1, xenTro1, galGal2, bosTau2, rheMac1, panTro1, hg17, rn3 -> mm7; fr1, tetNig1, xenTro1, galGal2, panTro1 -> mm6; galGal2 -> danRer2, mm5, hg17, monDom1, tetNig1, xenTro1; rheMac1 -> hg17; panTro1 -> hg16, hg17</td>
<td>03 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
rheMac1ToHg17 and hg17ToRheMac1 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>22 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Quality Scores</td>
<td>15 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Initial rhesus macaque browser release 
(Macaca mulatta preliminary assembly, Baylor Mmul_0.1, rheMac1) 
Assembly, Gap, RepeatMasker, Simple Repeats, GC Percent, non-Rhesus RefSeq, 
Rhesus mRrnas, Rhesus ESTs, Spliced ESTs, non-Rhesus mRrnas, 
Human Chain/Net. Downloads: bigZips, database.</td>
<td>18 May 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=canFam2></A>Dog May 2005 (canFam2, Broad Institute v. 2.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
NCBI accessions for assembly contigs (used for Assembly track details)</td>
<td>05 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain/Net/LiftOver (mm7)</td>
<td>22 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Dog (canFam2, May 2005, Broad Institute v2.0) Initial Release: Assembly, Gap, Quality Scores, GC Percent, Uncertified, BAC End Pairs, Short Match, Restr Enzymes, Human Proteins, Genscan Genes, Other RefSeq, Dog mRNAs, Other mRNAs, Dog ESTs, Spliced Ests, CpG Islands, Conservation, Most Conserved, Mouse Chain/Net (mm7), Rat Chain/Net (rn3), Human Chain/Net (hg17), Self Chain, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosomes, database, comparatives, liftOvers.
 </td>
<td>20 Dec 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=canFam1></A>Dog July 2004 (canFam1, Broad Institute v. 1.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>07 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
Keyword search by NCBI contig accession ID added.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>02 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
canFam1ToMm6 and mm6ToCanFam1 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>05 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
hg17ToCanFam1 and canFam1ToHg17 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>05 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>15 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (data update)</td>
<td>06 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Chain, Human Net (chained and netted blastz alignments to May 2004 Human)</td>
<td>30 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (dog-human-mouse multiz alignments, scored by phastCons)</td>
<td>29 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain, Mouse Net (chained and netted blastz alignments to May 2004 Mouse)</td>
<td>29 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu BLAT (translated BLAT of Fugu scaffolds to Dog genome)</td>
<td>26 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial canFam1 release (using Broad Institute WGS v1.0): Assembly, Gap, GC Percent, Genscan Genes, Dog mRNAs, Spliced ESTs, Dog ESTs, NonDog mRNAs, RepeatMasker, Simple Repeats

 Downloads: bigZips, chromosomes, database</td>
<td>13 Jul 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=bosTau2></A>Cow Mar. 2005 (bosTau2, Baylor Btau_2.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
LiftOvers: fr1, tetNig1, xenTro1, galGal2, bosTau2, rheMac1, panTro1, hg17, rn3 -> mm7; fr1, tetNig1, xenTro1, galGal2, panTro1 -> mm6; galGal2 -> danRer2, mm5, hg17, monDom1, tetNig1, xenTro1; rheMac1 -> hg17; panTro1 -> hg16, hg17</td>
<td>03 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Bos taurus (bosTau2, March 2005) Update Release: Assembly, Gap, Contig Orient, GC Percent, Quality Scores, Short Match, Restr. enzymes, Human Proteins, Other RefSeq, Genscan Genes, Cow mRNAs, Spliced ESTs, Cow ESTs, CpG Islands, Repeat Masker, Simple Repeats. Downloads: bigZips, chromosomes, database.</td>
<td>04 Oct 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=bosTau1></A>Cow Sep. 2004 (bosTau1, Baylor v1.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Keyword search by NCBI contig accession ID added.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
mm5ToBosTau1 and bosTau1ToMm5 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>22 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
mm6ToBosTau1 and bosTau1ToMm6 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>22 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
bosTau1ToHg17 and hg17ToBosTau1 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>22 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Fixes for 8 scaffolds, which were truncated (SCAFFOLD36273, SCAFFOLD120295, SCAFFOLD130877, SCAFFOLD165124, SCAFFOLD195025, SCAFFOLD215637, SCAFFOLD225287, SCAFFOLD300287)</td>
<td>05 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Quality Scores</td>
<td>05 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Non-Cow Refseq</td>
<td>16 May 2005</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>22 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Initial bosTau1 release (Baylor ): Assembly, Gap, GC Percent, Genscan, Cow mRNAs, Spliced ESTs, Cow ESTs, CpG Islands, Human Proteins, Human Chain/Net, RepeatMasker, Simple Repeats.
Downloads: bigZips, database, comparatives.</td>
<td>22 Feb 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=mm7></A>Mouse Aug. 2005 (mm7, NCBI Build 35)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
VisiGene (Allen Brain Atlas)</td>
<td>17 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish Chain/Net/LiftOver (danRer3)</td>
<td>21 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Link to Gene Network Id from Known Gene details page</td>
<td>16 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Improved searching capability for VisiGene</td>
<td>16 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Improved searching capability for Known Genes</td>
<td>03 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Geneid gene predictions from IMIM at GRIB</td>
<td>31 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Allen Brain Atlas Images</td>
<td>30 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: mm7 -> fr1/tetNig1/xenTro1/galGal2/bosTau2/rheMac1/panTro1/hg17/rn3</td>
<td>22 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Dog Chain/Net/LiftOver (canFam2)</td>
<td>22 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Mouse liftOvers: mm3, mm4, mm5, mm6 -> mm7; mm5-> galGal2; mm6 -> galGal2, monDom1, xenTro1, canFam2, hg16, panTro1, danRer2, fr1, tetNig1</td>
<td>21 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Initial mm7 Release (using NCBI build 35): Chromosome Band;  STS Markers; Map Contigs; Gap; BAC End Pairs; GC Percent; Short Match; Restriction Enzymes; CpG Islands; Known Genes; MGC Genes; RefSeq Genes; Other RefSeq; Genscan Genes; ESTs; miRNAs; mRNAs; Non-Mouse mRNAs; Repeat Masker; Simple Repeats; Riken CAGE; Riken CAGE Tc; Allen Brain Atlas; Rinn Sex Expression; GNF Expression; Conservation; Most Conserved; Chains/Nets: Fugu, Tetraodon, X. tropicalis, Chicken, Cow, Rhesus, Chimp, Human, Rat; Gene Sorter; Proteome Browser. Downloads: bigZips, chromosomes, database, comparatives.</td>
<td>15 Dec 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=mm6></A>Mouse Mar. 2005 (mm6, NCBI Build 34)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
VisiGene (Allen Brain Atlas)</td>
<td>17 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
Jackson Labs/MGI Representative Transcript, Phenotype and Allele data</td>
<td>06 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Improved searching capability for Known Genes</td>
<td>03 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
mouseLandscapeDistance (data update)</td>
<td>01 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Update to 10-way Conservation Track and Most Conserved Track</td>
<td>09 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated galGal2 -> mm6 chain and net alignments; Updated galGal2 -> mm6 liftOver file (Downloads, Batch Coordinate Conversion utility)
</td>
<td>09 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated rn3 -> mm6 chain and net alignments; Updated rn3 -> mm6 liftOver file (Downloads, Batch Coordinate Conversion utility)
</td>
<td>09 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated hg17 -> mm6 chain and net alignments; Updated hg17 -> mm6 liftOver file (Downloads, Batch Coordinate Conversion utility)
</td>
<td>09 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
mRNA, RefSeq mRNA, and native ESTs: Realigned and filtered
-- improvements in process.</td>
<td>27 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Mouse liftOvers: mm3, mm4, mm5, mm6 -> mm7; mm5-> galGal2; mm6 -> galGal2, monDom1, xenTro1, canFam2, hg16, panTro1, danRer2, fr1, tetNig1</td>
<td>21 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
VisiGene in-situ image viewer (Known Genes details)</td>
<td>09 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Riken CAGE</td>
<td>06 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Riken CAGE TC</td>
<td>06 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Allen Brain Atlas Images</td>
<td>10 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
STS Markers (filtered)</td>
<td>02 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
miRNA (data update)</td>
<td>27 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
NIAGene (update)</td>
<td>19 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN Gene Predictions</td>
<td>04 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
link in KG details page to GeneNetwork</td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
MicroRNAs from the miRNA Registry</td>
<td>26 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>24 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>24 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter (updated to pick up Pfam Domains for splice variants in new Known Genes)</td>
<td>12 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
SCDb Clones - Princeton Stem Cell Database (Mouse & Human; freeze date 7/15/2005)</td>
<td>04 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>27 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
NIA (National Institute on Aging) Mouse Gene Index Version 5</td>
<td>22 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
mm6ToBosTau1 and bosTau1ToMm6 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>22 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (Exon Primer Link)</td>
<td>20 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
STS Markers on Genetic and Radiation Hybrid Maps</td>
<td>12 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
canFam1ToMm6 and mm6ToCanFam1 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>05 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Known Gene rebuild using new filters plus Gene Sorter, Proteome Browser initial release.</td>
<td>30 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
NIA Mouse Gene Index 5.0</td>
<td>24 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
SGP Gene Predictions Using Mouse/Human Homology (provided by GRIB)</td>
<td>15 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
mouseLandscape microarray columns for Gene Sorter</td>
<td>07 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
mm6ToRn3 liftOver (Downloads and browser Batch Coordinate Conversion utility)</td>
<td>06 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
mm6ToHg17 liftOver (Downloads and browser Batch Coordinate Conversion utility)</td>
<td>01 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>27 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Restriction Enzyme mapping (REbase release 505).</td>
<td>24 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Expression and Regulation tracks: GNF Atlas 2, Rinn Sex Exp, GNF U74A, GNF U74B, GNF U74C, Affy GNF1M, Affy U74, Affy MOE430</td>
<td>11 May 2005</td>
</tr>
<TR valign=top><TD align=left>
10-Way Vertebrate Multiz Alignment & Conservation
</td>
<td>10 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Initial mm6 Release (using NCBI build 34): Chromosome Band, Assembly, Gap, Map Contigs, GC Percent, Known Genes, Genscan Genes, MGC Genes, RefSeq Genes, Mouse mRNAs, Non-Mouse mRNAs, Mouse ESTs, Mouse Spliced ESTs, Non-Mouse ESTs, CpG Islands, Chicken Chain/Net, Chimp Chain/Net, Cow Chain/Net, Dog Chain/Net, Fugu Chain/Net, Human Chain/Net, Opossum Chain/Net, Rat Chain/Net, Tetraodon Chain/Net, X. tropicalis Chain/Net, Zebrafish Chain/Net, RepeatMasker, Simple Repeats, Self Chain. Downloads: bigZips, chromosomes, database, comparatives.</td>
<td>04 May 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=mm5></A>Mouse May 2004 (mm5, NCBI Build 33)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Improved searching capability for Known Genes</td>
<td>03 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Mouse liftOvers: mm3, mm4, mm5, mm6 -> mm7; mm5-> galGal2; mm6 -> galGal2, monDom1, xenTro1, canFam2, hg16, panTro1, danRer2, fr1, tetNig1</td>
<td>21 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
VisiGene in-situ image viewer (Known Genes details)</td>
<td>09 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Riken CAGE</td>
<td>08 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Riken CAGE TC</td>
<td>08 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
hgBlastTab (synch with updated Known Genes in hg17)</td>
<td>01 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
mm5ToBosTau1 and bosTau1ToMm5 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>22 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
rn3ToMm5 and mm5ToRn3 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>05 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes (reloaded; original set was missing chrX)</td>
<td>27 May 2005</td>
</tr>
<TR valign=top><TD align=left>
SNPs (dbSNP build 124)</td>
<td>10 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Map Contigs (non-C57BL contigs removed)</td>
<td>04 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes (update ensGene table to correct chrMT -> chrM)</td>
<td>01 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish (danRer2) Chain and Net</td>
<td>06 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>22 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>21 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene new track, EST clustering gene prediction</td>
<td>22 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
NIA Gene</td>
<td>17 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
MicroRNAs - miRNA Registry</td>
<td>04 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>04 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Most Conserved (data update)</td>
<td>23 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Cytoband (updated data from NCBI)</td>
<td>16 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Chain, Net</td>
<td>13 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Blat</td>
<td>09 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
STS Markers</td>
<td>26 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
	</td>
<td>20 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Genscan</td>
<td>11 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid</td>
<td>11 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (Rat/Human/Dog/Chicken Multiz Alignments and PhyloHMM Conservation)</td>
<td>08 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Dog Chain and Net</td>
<td>08 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Rat Chain and Net</td>
<td>08 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Chicken Chain and Net</td>
<td>08 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Chain and Net</td>
<td>08 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Expression Distance</td>
<td>04 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Gene Maps</td>
<td>04 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Orthologs</td>
<td>04 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Fundamentals</td>
<td>04 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>30 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Expression Tracks (mm5)</td>
<td>29 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial mm5 release (using NCBI Build 33): Assembly, Gap, GC Percent, Known Genes, Proteome Browser, RefSeq Genes, MGC Genes, Mouse mRNAs, Spliced ESTs, Mouse ESTs, NonMouse mRNAs, NonMouse ESTs, CpG Islands, RepeatMasker, Simple Repeats.  Downloads: bigZips, chromosomes, database.</td>
<td>16 Jul 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=rn3></A>Rat June 2003 (rn3, Baylor HGSC v. 3.1)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Improved searching capability for Known Genes</td>
<td>03 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Keyword search by NCBI contig accession ID added.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated rn3 -> mm6 liftOver (Downloads, Batch Coordinate Conversion utility)
</td>
<td>06 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: fr1, tetNig1, xenTro1, galGal2, bosTau2, rheMac1, panTro1, hg17, rn3 -> mm7; fr1, tetNig1, xenTro1, galGal2, panTro1 -> mm6; galGal2 -> danRer2, mm5, hg17, monDom1, tetNig1, xenTro1; rheMac1 -> hg17; panTro1 -> hg16, hg17</td>
<td>03 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Dog Chain/Net/LiftOver (canFam2)</td>
<td>22 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
link in KG details page to GeneNetwork</td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
rn3ToMm5 and mm5ToRn3 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>05 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
rn3ToMm6 liftOver (Downloads and browser Batch Coordinate Conversion utility)</td>
<td>06 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
rn3ToHg17 liftOver (Downloads and browser Batch Coordinate Conversion utility)</td>
<td>01 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Restriction Enzyme mapping (REbase release 505).</td>
<td>24 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Non-Rat Refseq</td>
<td>18 May 2005</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
vsHg17 and vsMm5 pairwise alignments</td>
<td>01 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Cytoband Ideo</td>
<td>30 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
MGC Genes</td>
<td>24 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene track update, EST clustering gene predictions</td>
<td>22 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
Assembly/gold track update to fix fragStart off by 1</td>
<td>19 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain and Net (update to mm5)</td>
<td>13 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Chain, Human Net (update  hg15 --> hg17)</td>
<td>30 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Rat/Mouse/Human (updated)</td>
<td>20 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter for rn3</td>
<td>24 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
RGD SSLP</td>
<td>01 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
RGD ESTs</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
RGD Genes</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
RGD QTL</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF Atlas 2</td>
<td>27 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix RG-U34A</td>
<td>24 May 2004</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>21 May 2004</td>
</tr>
<TR valign=top><TD align=left>
GC Percent (updated to 5-base window track)</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Proteome Browser</td>
<td>07 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>06 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Ecores</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Ecores</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Ecores, Tetraodon Ecores</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>03 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix RAE230</td>
<td>15 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (updated)</td>
<td>02 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Overlap SNPs (dbSnp 118)</td>
<td>25 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
Chromosome Band</td>
<td>23 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes</td>
<td>22 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>18 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Genbank daily updates enabled</td>
<td>14 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain, Mouse Net (using mm3)</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (extended)</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat/Mouse/Human (using mm3 and hg16)</td>
<td>05 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Human Chain, Human Net (using hg16)</td>
<td>04 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Human Tight (using hg16)</td>
<td>03 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>05 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes</td>
<td>30 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>18 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial rn3 Release (using Baylor HGSC v. 3.1): Assembly, Gap, Bactigs, GC Percent, Bac End Pairs, RefSeq Genes, STS Markers, Geneid Genes, Genscan Genes, Rat mRNAs, Spliced ESTs, Rat ESTs, NonRat mRNAs, NonRat ESTs, CpG Islands, RepeatMasker, Simple Repeats, bigZips downloads, chromosome downloads</td>
<td>14 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan (using hg15)</td>
<td>14 Jul 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=rn2></A>Rat Jan. 2003 (rn2, Baylor HGSC v. 2.1)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Improved searching capability for Known Genes</td>
<td>03 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Geneid</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter: rn2 added</td>
<td>01 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes (update)</td>
<td>28 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes</td>
<td>30 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Fugu Blat (v. 3.0)</td>
<td>29 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>28 May 2003</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>20 May 2003</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>17 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>13 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Bounds</td>
<td>13 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan (using hg13)</td>
<td>02 Apr 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial rn2 Release (using Baylor HGSC v. 2.1): Assembly, Gap, Bactigs, GC Percent, RefSeq Genes, STS Markers, Genscan Genes, Rat mRNAs, Spliced ESTs, Rat ESTs, NonRat mRNAs, CpG Islands, RepeatMasker, Simple Repeats, bigZips downloads, chromosome downloads</td>
<td>14 Mar 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=monDom1></A>Opossum Oct. 2004 (monDom1, Broad Inst. Prelim Oct04)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Keyword search by NCBI contig accession ID added.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Initial monDom1 release (using Broad Inst. Prelim Oct04): Assembly, Gap, GC Percent, Quality Scores, Oppossum mRNAs, Non-Opposum mRNAs, Conservation, Most Conserved, Human Proteins, Human Chain/Net, Mouse Chain/Net, Chicken Chain/Net, Zebrafish Chain/Net, CpG Islands, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosomes, database.</td>
<td>18 Jan 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=galGal2></A>Chicken Feb. 2004 (galGal2, Chicken Genome Sequencing Consortium Feb. 2004 release)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Keyword search by NCBI contig accession ID added.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes (update)</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated with new predicitions from IMIM, using cow as well as human.</td>
<td>11 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated galGal2 -> mm6 liftOver (Downloads, Batch Coordinate Conversion utility)

</td>
<td>06 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: fr1, tetNig1, xenTro1, galGal2, bosTau2, rheMac1, panTro1, hg17, rn3 -> mm7; fr1, tetNig1, xenTro1, galGal2, panTro1 -> mm6; galGal2 -> danRer2, mm5, hg17, monDom1, tetNig1, xenTro1; rheMac1 -> hg17; panTro1 -> hg16, hg17</td>
<td>03 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Chain/Net Alignments (Feb. 2004 tetNig1)</td>
<td>12 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish Chain/Net Alignments (June 2004 danRer2)</td>
<td>12 May 2005</td>
</tr>
<TR valign=top><TD align=left>
X. tropicalis Chain/Net Alignments (Oct. 2004 xenTro1)</td>
<td>12 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Isochores</td>
<td>06 May 2005</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
conservation tracks (Conservation, Most Conserved)</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Human Chain, Human Net (updated to May 2004 assembly)</td>
<td>24 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Updated SGP Genes.</td>
<td>29 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
cpgIslands (data update) replace WUSTL cpgIslands with calculations by Andy Law, Roslin Institute</td>
<td>29 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>14 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Ecores</td>
<td>20 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Contamination (alignments of Maize mitochondrion, E. Coli and vector sequence to WGS assembly)</td>
<td>03 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
BGI SNPs, BGI Genes, BGI Coverage (Beijing Genomics Institute SNPs, gene annotations and read coverage locations) </td>
<td>02 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>21 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>18 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>17 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
RNA Genes</td>
<td>30 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
CpG Island (updated)</td>
<td>06 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes (updated)</td>
<td>17 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs, Genscan Genes, Geneid Genes, UniGene, CpG Island, Fugu Blat, GC Percent, Quality Scores, Human Chain, Human Net, Human Synteny, Self Chain, Self Net</td>
<td>10 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial galGal2 Release (using the Feb. 2004 assembly): Supercontigs, Assembly, Gap, Chicken mRNAs, Spliced ESTs, Chicken ESTs, NonChicken mRNAs, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosome, database.</td>
<td>01 Mar 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=xenTro1></A>X. tropicalis Oct. 2004 (xenTro1, JGI v3.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
LiftOvers: fr1, tetNig1, xenTro1, galGal2, bosTau2, rheMac1, panTro1, hg17, rn3 -> mm7; fr1, tetNig1, xenTro1, galGal2, panTro1 -> mm6; galGal2 -> danRer2, mm5, hg17, monDom1, tetNig1, xenTro1; rheMac1 -> hg17; panTro1 -> hg16, hg17</td>
<td>03 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
JGI Filtered Genes</td>
<td>18 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins (data update)</td>
<td>03 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
MGC Genes</td>
<td>14 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Frog Browser Initial Release (using JGI version 3.0): Assembly, Gap, GC Percent, Mapped Human, Genscan, Frog mRNA, Spliced ESTs, Frog ESTs, CpG Islands, Conservation, Most Conserved, Fugu Blat, Zebrafish Chain/Net, Mouse Chain/Net, Human Chain/Net, Chicken Chain/Net, RepeatMasker, Simple Repeats</td>
<td>17 Nov 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=danRer3></A>Zebrafish May 2005 (danRer3, Sanger Centre, Danio rerio Sequencing Project Zv5)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Zebrafish (Danio rerio) Initial Release (danRer3, May 2005, Zv5): Scaffolds, Contigs, Gap, BAC End Pairs/Singles, GC Percent, Short Match, Restr Enzymes, RefSeq Genes, ZGC Genes, Ensembl Genes, Zebrafish mRNAs, Spliced ESTs, Zebrafish ESTs, Other mRNAs, TIGR Gene Index, Affy Zebrafish, 5-way Conservation, Most Conserved, Human Proteins (hg17), Human Chain/Net (hg18), Mouse Chain/Net/LiftOver (mm7), Fugu Chain/Net/LiftOver (fr1), Tetraodone Chain/Net/LiftOver (tetNig1), Repeat Masker, Simple Repeats, Self Chain. Downloads: bigZips, database, comparatives, chromosomes, liftOvers.</td>
<td>17 Feb 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=danRer2></A>Zebrafish June 2004 (danRer2, Sanger Centre, Danio rerio Sequencing Project Zv4)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Keyword search by NCBI contig accession ID added.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: danRer1 -> danRer2; danRer2 -> mm6</td>
<td>04 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
		</td>
<td>05 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
New version of chr1.fa in downloads that has corrected masking.</td>
<td>16 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Contigs (data update)</td>
<td>15 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
PhastCons Conserved Elements</td>
<td>07 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
6-Way Vertebrate Multiz Alignment & Conservation</td>
<td>07 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Radiation Hybrid Map (data update with tighter filtering)</td>
<td>24 May 2005</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>11 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Zfish WZ EST Clusters</td>
<td>06 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Fugu (fr1) Chain and Net</td>
<td>06 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Mouse (mm5) Chain and Net</td>
<td>06 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Human (hg17) Chain and Net</td>
<td>06 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Initial danRer2 release (using Zv4): Contigs, Scaffolds, Gap, GC Percent, Radiation Hybrid Map, RefSeq Genes, ZGC Genes, Ensembl Genes, Zebrafish mRNAs, Spliced ESTs, Zebrafish ESTs, Non-Zebrafish mRNAs, Affy Zebrafish, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosomes, database.</td>
<td>23 Dec 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=danRer1></A>Zebrafish Nov. 2003 (danRer1, Sanger Centre, Danio rerio Sequencing Project Zv3)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
LiftOvers: danRer1 -> danRer2; danRer2 -> mm6</td>
<td>04 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
		</td>
<td>05 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Radiation Hybrid Map (data update with tighter filtering)</td>
<td>12 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ZGC Genes</td>
<td>11 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
UniGene</td>
<td>04 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins (data update)</td>
<td>17 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Chain, Human Net</td>
<td>19 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs and BAC End Singles</td>
<td>13 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial danRer1 release (using Zv3): Contigs, Scaffolds, Gap, GC Percent, RefSeq Genes, Ensembl Genes, Zebrafish mRNAs, Spliced ESTs, Zebrafish ESTs, Zebrafish WZ EST clusters, NonZebrafish mRNAs, Affy Zebrafish, Fugu Chain, Fugu Net, Human Proteins, RepeatMasker, Simple Repeats. 

 Downloads: bigZips, chromosomes, database, vsFr1, vsHg17</td>
<td>28 Jul 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=tetNig1></A>Tetraodon Feb. 2004 (tetNig1, Genoscope and Broad Institute, V7)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Zebrafish Chain/Net/LiftOver (danRer3)</td>
<td>21 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Keyword search by NCBI contig accession ID added.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: fr1, tetNig1, xenTro1, galGal2, bosTau2, rheMac1, panTro1, hg17, rn3 -> mm7; fr1, tetNig1, xenTro1, galGal2, panTro1 -> mm6; galGal2 -> danRer2, mm5, hg17, monDom1, tetNig1, xenTro1; rheMac1 -> hg17; panTro1 -> hg16, hg17</td>
<td>03 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Human Ecores</td>
<td>15 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Self Chain, Self Net</td>
<td>26 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Cytokine Genes (removed extensions)</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
HOX Genes (removed extensions)</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Genoscope GAZE (removed extensions)</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>29 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Cytokine Genes</td>
<td>17 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
HOX Genes</td>
<td>15 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Genoscope GAZE gene predictions</td>
<td>15 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain, Net (mm5)</td>
<td>13 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial tetNig1 release (Genoscope v6): Scaffolds, Gap, GC Percent, Short Match, Genscan, mRNAs, Spliced ESTs, ESTs, Xeno ESTs, Human Chain/Net, Fugu Chain/Net, RepeatMasker, Simple Repeats.
Downloads: bigZips, chromosomes, database, comparatives.</td>
<td>10 Sep 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=fr1></A>Fugu Aug. 2002 (fr1, JGI V3.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Zebrafish Chain/Net/LiftOver (danRer3)</td>
<td>21 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: fr1, tetNig1, xenTro1, galGal2, bosTau2, rheMac1, panTro1, hg17, rn3 -> mm7; fr1, tetNig1, xenTro1, galGal2, panTro1 -> mm6; galGal2 -> danRer2, mm5, hg17, monDom1, tetNig1, xenTro1; rheMac1 -> hg17; panTro1 -> hg16, hg17</td>
<td>03 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish (danRer2) Chain and Net</td>
<td>06 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>14 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Ecores, Mouse Ecores, Rat Ecores</td>
<td>07 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>29 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial fr1 Release (using JGI V3.0): Scaffolds, Gap, GC Percent, Ensembl Genes, Genscan Genes, Fugu mRNAs, Fugu ESTs, Non-Fugu mRNAs, CpG Islands, Human Blat, RepeatMasker, Simple Repeats. Downloads: bigZips, database</td>
<td>23 Sep 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=ci2></A>C. intestinalis Mar. 2005 (ci2, JGI v2.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Initial ci2 Release (using JGI v2.0, March 2005): Gap, GC Percent, Short Match, Restr. enzymes, RefSeq, Non-Ciona RefSeq, mRNAs, Spliced ESTs, C. intestinalis ESTs, Non-Ciona mRNAs, Repeat Masker, Simple Repeats. Downloads: bigZips, database.</td>
<td>09 Sep 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=ci1></A>C. intestinalis Dec. 2002 (ci1, JGI v1.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
RefSeq Genes</td>
<td>21 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins (data update)</td>
<td>17 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial ci1 release (using JGI v1.0): Gap, GC Percent, Non-Ciona RefSeq Genes, SNAP Genes, JGI Genes, Ciona mRNAs, Spliced ESTs, Ciona ESTs, NonCiona mRNAs, C. savignyi Blat, Human Proteins, RepeatMasker, Simple Repeats

Downloads: bigZips, database</td>
<td>15 Jul 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=dm2></A>D. melanogaster Apr. 2004 (dm2, BDGP v. 4 / DHGP v. 3.2)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
D. sechellis (droSec1) Chain/Net and Lift Over (Downloads, Batch Coordinate Conversion utility)</td>
<td>18 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
D. persimilis (droPer1) Chain/Net and Lift Over (Downloads, Batch Coordinate Conversion utility)</td>
<td>18 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
D. pseudo. (dp3) Chain/Net and Lift Over (Downloads, Batch Coordinate Conversion utility)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
A. mellifera (apiMel2) Chain/Net and Lift Over (Downloads, Batch Coordinate Conversion utility)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: droGri1, droMoj2, droAna2, droEre1, droSim1, droVir2, dp3 -> dm2</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
DroYak1 Chain/Net - regenerated with more sensitive blastz parameters</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
DroMoj1 Chain/Net - regenerated with more sensitive blastz parameters</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
DroVir1 Chain/Net - regenerated with more sensitive blastz parameters</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
AnoGam1 Chain/Net - regenerated with more sensitive blastz parameters</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: dm2 -> anoGam1, droGri1, droMoj1, droMoj2, droAna1, droAna2, droEre1, droSim1, driVir1, droVir2, droYak1, apiMel2, dp3</td>
<td>04 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>10 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN Gene Predictions</td>
<td>07 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>21 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
D. simulans (droSim1) Chain and Net</td>
<td>08 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
augustus gene predictions</td>
<td>07 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Conservation, Most Conserved (Rebuilt from new pairwise alignments with more sensitive blastz parameters; now includes D. simulans)</td>
<td>28 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Conservation, Most Conserved (re-release to correct omission of D. yakuba from multiple alignments)</td>
<td>11 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
FlyReg (Drosophila DNase I Footprint Database)</td>
<td>04 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
FlyBase Genes, FB Noncoding (updated FlyBase annotations from version 4.0 to 4.1)</td>
<td>28 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Initial D. melanogaster 2 release (BDGP Rel.4 euchromatic, DHGP Rel. 3.2 heterochromatic): Gap, GC Percent, Genscan, RefSeq Genes, Flybase, FB-Noncoding, Conservation, Most-Conserved,  mRNAs, xenoMrnas, Spliced ESTs, ESTs, {anoGam1, apiMel1, dp2, droAna1, droMoj1, droVir1, droYak1} Chain/Net, RepeatMasker, Simple Repeats.
Downloads: bigZips, database, comparatives.</td>
<td>14 Feb 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=droSim1></A>D. simulans Apr. 2005 (droSim1, WUSTL version 1.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
LiftOvers: droGri1, droMoj2, droAna2, droEre1, droSim1, droVir2, dp3 -> dm2</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>21 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
D.mel. Proteins (data update to dm2)</td>
<td>27 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Initial Drosophila simulans release (droSim1; Release 1.0 Apr. 2005 from WUSTL): Assembly, Gap, GC Percent, Short Match, Restr. enzymes, RepeatMasker, Simple Repeats, Non D. sim. RefSeq, Genscan Genes, D. sim mRnas, D. sim Spliced ESTs, D. sim ESTs, Non D. sim. mRnas, D. mel Proteins, D. mel Chain, D. mel Net.</td>
<td>10 Jun 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=droSec1></A>D. sechellia Oct. 2005 (droSec1, Broad 28 Oct. 2005)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Initial droSec1 Release (using Broad Institute):  supporting tables, Assembly, GC Percent, Gap, Simple Repeats, RepeatMasker, D. melanogaster (dm2) chain/net, Genscan, Mapped Fly Proteins, support for Genbank, Other mRNAs, Other RefSeq
Downloads: bigZips, database, liftOver, vsDm2</td>
<td>18 Jan 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=dm1></A>D. melanogaster Jan. 2003 (dm1, BDGP v. 3)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Gene Sorter P2P Protein-to-Protein Interaction Network (worm, fly, yeast)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Restriction Enzyme mapping (REbase release 505).</td>
<td>24 May 2005</td>
</tr>
<TR valign=top><TD align=left>
FlyBase Non-coding (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
FlyBase Genes (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
FlyReg</td>
<td>23 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>21 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (chrUn-based dp1 replaced with scaffold-based dp2 for D. pseudoobscura sequence; rescored with latest phastCons)</td>
<td>01 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Incremental update to BDGP Insertions.</td>
<td>27 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>14 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
D. pseudoobscura Chain, Net (updated to dp2)</td>
<td>01 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
BDGP Insertions (P element insertion locations from the BDGP Gene Disruption Project)</td>
<td>27 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
D. yakuba Chain/Net</td>
<td>12 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Anopheles Chain and Net</td>
<td>12 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (D. yakuba, D. pseudoob., A. gambiae)</td>
<td>11 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter</td>
<td>01 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
BDGP Non-coding</td>
<td>30 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Chromosome Bands (Cytology) from Flybase</td>
<td>30 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
BDGP Genes</td>
<td>30 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Anopheles Ecores</td>
<td>17 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial dm1 Release (using BDGP V3.1): Gap, GC Percent, RefSeq Genes, BDGP Genes, BDGP Non-coding, Genscan Genes, Fruitfly mRNAs, Spliced ESTs, Fruitfly ESTs, Fugu Blat, D. pseud. Chain, D. pseud. Net, D. pseud Aligns, RepeatMasker, Simple Repeats. Downloads: bigZips, database</td>
<td>30 Oct 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=droYak2></A>D. yakuba Nov. 2005 (droYak2, WUSTL version 2.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Initial Drosophila yakuba Release (droYak2, Nov. 2005 using WUSTL):  Assembly, Gap, GC Percent, Contamination, Short Match, Restr Enzymes, Other RefSeq, Genscan Genes, D. yakuba mRNAs, D. yakuba ESTs, Spliced Ests, Other mRNAs, D. melanogaster Proteins/Chain/Net, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosomes, database, liftOver, vsDm2.</td>
<td>11 Jan 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=droYak1></A>D. yakuba Apr. 2004 (droYak1, WUSTL version 1.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
D. melanogaster Chain/Net/LiftOver (dm2 update)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>21 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
D.mel. Proteins (data update to dm2)</td>
<td>27 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Fly (dm1) Proteins (data update)</td>
<td>16 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
D. mel. Proteins (chained tblastn alignments of D. melanogaster proteins)</td>
<td>21 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
GC Percent</td>
<td>26 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial D. yakuba Release (version 1) - Assembly, Gap, Genscan Genes, D. yakuba mRNAs, Spliced ESTs, D. yakuba ESTs, Non-D. yakuba mRNAs, D. mel. Chain, D. mel. Net, RepeatMasker, Simple Repeats</td>
<td>23 Jul 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=droEre1></A>D. erecta Aug. 2005 (droEre1, Agencourt 1 Aug 2005)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Reloaded Gap and loaded Assembly from assembly.links files included with the assemblies.  Previously, Gap was deduced from runs of N's in the sequence.</td>
<td>25 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: droGri1, droMoj2, droAna2, droEre1, droSim1, droVir2, dp3 -> dm2</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Drosophila erecta (droEre1, August 2005) Initial Release: Gap, Short Match, Restr. enzymes, RepeatMasker, Simple Repeats, GC Percent, D. ere. mRNAs, Non-D. ere. mRNAs, Non-D. ere. RefSeq, Genscan Genes, D. mel. (dm2) Chain/Net, D.mel. (dm2) Proteins. Downloads: bigZips, database, vsDm2.</td>
<td>20 Sep 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=droAna2></A>D. ananassae Aug. 2005 (droAna2, Agencourt 1 August 2005)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Reloaded Gap and loaded Assembly from assembly.links files included with the assemblies.  Previously, Gap was deduced from runs of N's in the sequence.</td>
<td>25 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: droGri1, droMoj2, droAna2, droEre1, droSim1, droVir2, dp3 -> dm2</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Drosophila ananassae (Agencourt Bioscience Corporation, August 2005) Initial Release -- Gap, GC Percent, Other Refseq, Genscan, D. anan. mRNA, Other mRNAs, D. mel. proteins, D. mel. Chain/Net, RepeatMasker, Simple Repeats. Downloads: bigZips, database, vsDm2.</td>
<td>07 Oct 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=droPer1></A>D. persimilis Oct. 2005 (droPer1, Broad 28 Oct. 2005)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Initial droPer1 Release (using Broad Institute):  supporting tables, Assembly, GC Percent, Gap, Simple Repeats, RepeatMasker, D. melanogaster (dm2) chain/net, Genscan, Mapped Fly Proteins, support for Genbank, Other mRNAs, Other RefSeq.  Downloads: bigZips, database, liftOver, vsDm2.</td>
<td>06 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Initial droPer1 Release (using Broad Institute):  supporting tables, Assembly, GC Percent, Gap, Simple Repeats, RepeatMasker, D. melanogaster (dm2) chain/net, Genscan, Mapped Fly Proteins, support for Genbank, Other mRNAs, Other RefSeq.  Downloads: bigZips, database, liftOver, vsDm2.</td>
<td>06 Jan 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=dp3></A>D. pseudoobscura Nov. 2004 (dp3, FlyBase Release 1.03)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
LiftOvers: droGri1, droMoj2, droAna2, droEre1, droSim1, droVir2, dp3 -> dm2</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>21 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
D.mel. Proteins (data update to dm2)</td>
<td>27 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
D. pseudoobscura Initial Release (dp3, based on FlyBase Release 1.03 November 2004): Gap, GC Percent, Short Match, Restr. enzymes, FlyBase, Non-D.pseudo. RefSeq, Twinscan, Genewise, Genscan Genes, D.pseudo. mRNAs, Spliced ESTs, D.pseudo. ESTs, Non-D.pseudo. mRNAs, D.mel. Proteins (dm1), D.mel Chain/Net (dm2), Repeat Masker, Simple Repeats, Downloads: bigZips, database, vsDm2.</td>
<td>24 Jun 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=dp2></A>D. pseudoobscura Aug. 2003 (dp2, Baylor HGSC Freeze 1)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Initial dp2 release (Baylor freeze 1 August 2003): GC Percent, Xeno RefSeq, Twinscan Genes, Genscan Genes, mRNAs, Xeno mRNAs, D. melanogaster, proteins, D. mel. Chain, D. mel. Net, RepeatMasker, Simple Repeats.
Downloads: bigZips, chromosomes, database</td>
<td>01 Sep 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=droVir2></A>D. virilis Aug. 2005 (droVir2, Agencourt 1 August 2005)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Reloaded Gap and loaded Assembly from assembly.links files included with the assemblies.  Previously, Gap was deduced from runs of N's in the sequence.</td>
<td>25 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: droGri1, droMoj2, droAna2, droEre1, droSim1, droVir2, dp3 -> dm2</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Drosophila virilis (droVir2, Month 2005) Initial Release: Gap, Short Match, Restr. enzymes, RepeatMasker, Simple Repeats, GC Percent, D. vir. ESTs, Spliced Ests, D. vir. mRNAs, Non-D. vir. mRNAs, Non-D. vir. RefSeq, Genscan Genes, D. mel. (dm2) Chain/Net, D.mel. (dm2) Proteins. Downloads: bigZips, database, vsDm2.</td>
<td>20 Sep 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=droVir1></A>D. virilis July 2004 (droVir1, Agencourt 12 July 2004)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
D. melanogaster Chain/Net/LiftOver (dm2 update)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>21 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
D.mel. Proteins (data update to dm2)</td>
<td>27 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
D. virilis Browser Initial Release (Agencourt 12 July 2004): Gap, GC Percent, Non-D. vir. Refseq, Genscan, D. vir. mRNA, Non-D. vir. mRNA, D. mel. proteins, D. mel. Chain/Net, RepeatMasker, Simple Repeats</td>
<td>15 Dec 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=droMoj2></A>D. mojavensis Aug. 2005 (droMoj2, Agencourt 1 August 2005)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Reloaded Gap and loaded Assembly from assembly.links files included with the assemblies.  Previously, Gap was deduced from runs of N's in the sequence.</td>
<td>25 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: droGri1, droMoj2, droAna2, droEre1, droSim1, droVir2, dp3 -> dm2</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Drosophila mojavensis (droMoj2, August 2005) Initial Release: Gap, Short Match, Restr. enzymes, RepeatMasker, Simple Repeats, GC Percent, D. moj. mRNAs, Non-D. moj. mRNAs, Non-D. moj. RefSeq, Genscan Genes, D. mel. (dm2) Chain/Net, D.mel. (dm2) Proteins. Downloads: bigZips, database, vsDm2.</td>
<td>20 Sep 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=droMoj1></A>D. mojavensis Aug. 2004 (droMoj1, Agencourt 11 Aug 2004)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
D. melanogaster Chain/Net/LiftOver (dm2 update)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>21 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
D.mel. Proteins (data update to dm2)</td>
<td>27 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
D. mojavensis Browser Initial Release (Agencourt 11 Aug 2004): Gap, GC Percent, Non-D.moj. Refseq, Genscan, D.moj. mRNA, Non D.moj. mRNAs, D. mel. proteins, D. mel. Chain/Net, RepeatMasker, Simple Repeats</td>
<td>15 Dec 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=droGri1></A>D. grimshawi Aug. 2005 (droGri1, Agencourt 1 Aug 2005)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Reloaded Gap and loaded Assembly from assembly.links files included with the assemblies.  Previously, Gap was deduced from runs of N's in the sequence.</td>
<td>25 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: droGri1, droMoj2, droAna2, droEre1, droSim1, droVir2, dp3 -> dm2</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Drosophila grimshawi (droGri1, August 2005) Initial Release: Gap, GC Percent, Short Match, Restr. enzymes, Other RefSeq, Genscan Genes, Other mRNAs, D. melanogaster (dm2) Proteins, D. melanogaster (dm2) Chain/Net, RepeatMasker, Simple Repeats. Downloads: bigZips, database, vsDm2.</td>
<td>11 Oct 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=droAna1></A>D. ananassae July 2004 (droAna1, TIGR 15 July 2004)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
D. melanogaster Chain/Net/LiftOver (dm2 update)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>21 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
D.mel. Proteins (data update to dm2)</td>
<td>27 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
D. ananassae Browser Initial Release (TIGR 15 July 2004): Gap, GC Percent, Non-D.anan. Refseq, Genscan, D.anan. mRNA, Non D.anan. mRNAs, D. mel. proteins, D. mel. Chain/Net, RepeatMasker, Simple Repeats.  Downloads bigZips, database, vsDm1.</td>
<td>15 Dec 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=apiMel2></A>A. mellifera Jan. 2005 (apiMel2, Baylor HGSC Amel_2.0)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
D. melanogaster LiftOver (dm2)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
D.mel. Proteins (data update to dm2)</td>
<td>27 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
NCBI Genes (update from NCBI: alternate gene names)</td>
<td>01 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
EMBL Homologs to D. melanogaster</td>
<td>31 May 2005</td>
</tr>
<TR valign=top><TD align=left>
NCBI Genes</td>
<td>19 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>21 Apr 2005</td>
</tr>
<TR valign=top><TD align=left>
Human(hg17) proteins mapped by chained tBLASTn</td>
<td>28 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
D. melanogaster Proteins</td>
<td>28 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Assembly Scaffolds</td>
<td>28 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
A. mellifera (Honeybee) Browser Jan. 2005 Release (apiMel2, Baylor HGSC Amel_2.0): Assembly, Gap, GC Percent, Genscan,  A. mellifera mRNAs, Non-A. mellifera mRNAs, A. mellifera Spliced ESTs, A. mellifera ESTs, D. melanogaser Chain/Net, RepeatMasker, Simple Repeats. Downloads: bigZips, database, vsDm2.</td>
<td>23 Feb 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=apiMel1></A>A. mellifera July 2004 (apiMel1, Baylor HGSC Amel_1.2)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
D. melanogaster Chain/Net/LiftOver (dm2 update)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Honeybee Browser Initial Release (Baylor HGSC Amel_1.2): Assembly, Gap, GC Percent, Genscan, Honeybee mRNA, Spliced ESTs, Honeybee ESTs, Mapped D.melanogaster Proteins, D. melanogaser Chain/Net, RepeatMasker, Simple Repeats</td>
<td>22 Dec 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=anoGam1></A>A. gambiae Feb. 2003 (anoGam1, IAGEC v.MOZ2)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
D. melanogaster Chain/Net/LiftOver (dm2 update)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>21 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
D.mel. Proteins (data update to dm2)</td>
<td>27 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
AnoEST Unique and Non-Unique (TCLAG*: Clustered A. gambiae ESTs)</td>
<td>08 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
GC Percent</td>
<td>22 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Fly (dm1) Proteins (data update)</td>
<td>16 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
D. mel. Proteins (chained tblastn alignments of D. melanogaster proteins)</td>
<td>21 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial anoGam1 release (MOZ2 draft genome sequence Feb. 2003): Assembly, Gap, Ensembl Genes, Genscan Genes, A. gambiae mRNAs, Spliced ESTs, A. gambiae ESTs, Non-A. gambiae mRNAs, Non-A.gambiae ESTs, D. mel Chain, D. mel. Net, RepeatMasker, Simple Repeats

 Downloads: bigZips, chromosomes, database</td>
<td>10 Aug 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=ce2></A>C. elegans Mar. 2004 (ce2, WormBase v. WS120)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Gene Sorter P2P Protein-to-Protein Interaction Network (worm, fly, yeast)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>24 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Conservation, Most Conserved</td>
<td>17 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>14 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter</td>
<td>04 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
WormBase Genes</td>
<td>26 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Self Chain</td>
<td>20 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial ce2 Release (using WormBase v. WS120): Assembly, Gap, GC Percent, WormBase Genes, WormBase Genefinder, RefSeq Genes, Twinscan, miRNA, Worm mRNAs, Spliced ESTs, Worm ESTs, Gene Bounds, Fugu Blat, Briggsae Waba, Briggsae Blastz, Briggsae Chain, Briggsae Net, Briggsae Aligns, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosome, annotations.</td>
<td>11 May 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=ce1></A>C. elegans May 2003 (ce1, WormBase v. WS100)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
RefSeq Genes</td>
<td>19 May 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>23 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Sanger Links (data update)</td>
<td>28 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter</td>
<td>03 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
sangerPep table</td>
<td>23 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>27 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial ce1 Release (using WormBase v. WS100): Assembly, Gap, GC Percent, RefSeq Genes, Sanger Genes, Worm mRNAs, Worm ESTs, Fugu Blat, Briggsae Waba, Briggsae Chain, Briggsae Blastz, RepeatMasker, Simple Repeats, Self Chain, chromosome downloads.</td>
<td>24 Jun 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=cb1></A>C. briggsae July 2002 (cb1, WormBase v. cb25.agp8)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Twinscan</td>
<td>03 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Elegans Chain (updated to ce2)</td>
<td>25 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Elegans Blastz (updated to ce2)</td>
<td>25 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial cb1 Release (using WormBase v. cb25.agp8): Assembly, Gap, GC Percent, Worm mRNAs, Worm ESTs, Elegans Chain, Elegans Blastz, RepeatMasker, Simple Repeats, data set downloads.</td>
<td>24 Jun 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=strPur1></A>S. purpuratus April 2005 (strPur1, Baylor)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Initial strPur1 release (using Baylor Spur_0.5 whole genome shotgun assembly) Assembly, gap, RepeatMasker, SimpleRepeats, GC Percent, S. purpuratus mRNAs, Other mRNAs, Genscan, Mapped Human Proteins, Quality, RefSeq.</td>
<td>07 Mar 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=sacCer1></A>S. cerevisiae Oct. 2003 (sacCer1, SGD sequence data as of 1 Oct. 2003)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Gene Sorter P2P Protein-to-Protein Interaction Network (worm, fly, yeast)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Known Genes -- UCSC SAM-T02 protein structure analysis and prediction.</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Regulatory Code and Reg. CHIP/CHIP</td>
<td>17 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (data update), Most Conserved (data update)</td>
<td>28 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter: Regulatory Module column</td>
<td>24 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Regulatory Module (first public release of Eran Segal's regulatory module track.)</td>
<td>24 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>14 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation, Most Conserved</td>
<td>10 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial sacCer1 Release (based on sequence in SGD as of 1 Oct 2003): WashU Clones, GC Percent, SGD Genes, SGD Other, Yeast ESTs, Spliced ESTs, Yeast ESTs, Multiz Yeast, Simple Repeats. Downloads: bigZips, database.</td>
<td>14 Jan 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=sc1></A>SARS Apr. 2003 (sc1, GenBank sequence as of 14 Apr 2003)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
bigZips downloads</td>
<td>05 Jun 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial sc1 Release (based on sequence deposited into GenBank as of 14 Apr 2003): GC Percent, GenBank Protein Annotations, Fgenesv+ Genes, Viral mRNAs, Viral mRNAs Chained, Other SARS, Coronavirus Proteins</td>
<td>23 May 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=zoo1></A>NISC (Zoo) Jun. 2002 (zoo1, Comparative Sequencing Program Target 1)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Downloads: Annotation database (by organism)</td>
<td>23 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial NISC Target 1 Release: Assembly, Gap, GC Samples, Gene Annotations, Genscan Genes, mRNAs, NonOrganism mRNAs, NonOrganism ESTs, Tetraodon Blat, ZooSeq, RepeatMasker, Simple Repeats. Human only: RefSeq Genes, ESTs, Spliced ESTs, DNase HyperSens, MCS (Binomial), Binomial Cons Score, MCS (ParsPVal), Parsimony P-Values, MCS InterMerged, MCS Inter</td>
<td>15 Aug 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
</TD></TR></TABLE>

</BODY></HTML>
