<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<HTML>
<HEAD>

	
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META http-equiv="Content-Script-Type" content="text/javascript">
	<META HTTP-EQUIV="Pragma" CONTENT="no-cache">
	<META HTTP-EQUIV="Expires" CONTENT="-1">
	<TITLE>Track and Table Releases	</TITLE>
    <LINK rel='STYLESHEET' href='../style/HGStyle.css' TYPE='text/css' />
</HEAD>
<BODY CLASS="qaPushQ cgi">
<script type='text/javascript'>
document.write("<center><div id='warnBox' style='display:none;'><CENTER><B id='warnHead'></B></CENTER><UL id='warnList'></UL><CENTER><button id='warnOK' onclick='hideWarnBox();return false;'></button></CENTER></div></center>");
function showWarnBox() {document.getElementById('warnOK').innerHTML='&nbsp;OK&nbsp;';var warnBox=document.getElementById('warnBox');warnBox.style.display=''; warnBox.style.width='65%';document.getElementById('warnHead').innerHTML='Warning/Error(s):';window.scrollTo(0, 0);}
function hideWarnBox() {var warnBox=document.getElementById('warnBox');warnBox.style.display='none';warnBox.innerHTML='';var endOfPage = document.body.innerHTML.substr(document.body.innerHTML.length-20);if(endOfPage.lastIndexOf('-- ERROR --') > 0) { history.back(); }}
window.onunload = function(){}; // Trick to avoid FF back button issue.
</script>

<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">

<TR><TH COLSPAN=1 ALIGN="left">
<IMG SRC="../images/title.jpg">
</TH></TR>


<noscript><div class='noscript'><div class='noscript-inner'><p><b>JavaScript is disabled in your web browser</b></p><p>You must have JavaScript enabled in your web browser to use the Genome Browser</p></div></div></noscript>
<script type='text/javascript' SRC='../js/jquery.js'></script>
<script type='text/javascript' SRC='../js/jquery.plugins.js'></script>
<LINK rel='STYLESHEET' href='../style/nice_menu.css' TYPE='text/css' />
<div id="main-menu-whole">

		<div id="main-menu"><div id="block-nice_menus-1" class="clear-block block block-nice_menus">

	<div class="upcorner"></div>
	<div class="blockbg">
		<div class="blockcontent">
						<div class="content">

<ul class="nice-menu nice-menu-down sf-js-enabled" id="nice-menu-1">

<li class="first even" id="home"><a href="../index.html" ><span id="home-link">Home</span></a></li> 

<li class="menuparent" id="tools1"><a href="../cgi-bin/hgGateway">Genomes</a>
	<ul style="display: none; visibility: hidden;">
	<li class="first"><a href="../cgi-bin/hgTracks?db=hg38&amp;position=lastDbPos">Human GRCh38/hg38</a></li>
	<li><a href="../cgi-bin/hgTracks?db=hg19&amp;position=lastDbPos">Human GRCh37/hg19</a></li>
	<li><a href="../cgi-bin/hgTracks?db=mm10&amp;position=lastDbPos">Mouse GRCm38/mm10</a></li>
	<li><a href="../cgi-bin/hgTracks?db=mm9&amp;position=lastDbPos">Mouse NCBI37/mm9</a></li>
	<li class="last"><a href="../cgi-bin/hgGateway">Other</a></li>
        </ul>
</li>

<li class="menuparent" id="tools2"><a href="../cgi-bin/hgTracks">Genome Browser</a>
	<ul style="display: none; visibility: hidden;">
	<li class="first"><a id="configureMenuLink" href="../cgi-bin/hgTracks?hgTracksConfigPage=configure">Configure</a></li>
	<li><a id="trackSearchMenuLink" href="../cgi-bin/hgTracks?hgt_tSearch=track+search">Track Search</a></li>
	<li><a id="cartResetMenuLink" href="../cgi-bin/cartReset">Reset All User Settings</a></li>
        </ul>
</li>


<li class="menuparent" id="tools3"><span>Tools</span>
	<ul style="display: none; visibility: hidden;">
	<li class="first"><a id="blatMenuLink" href="../cgi-bin/hgBlat?command=start">Blat</a></li>
	<li><a id="tableBrowserMenuLink" href="../cgi-bin/hgTables">Table Browser</a></li>
	<li><a id="vaiMenuLink" href="../cgi-bin/hgVai">Variant Annotation Integrator</a></li>
	<li><a id="diMenuLink" href="../cgi-bin/hgIntegrator">Data Integrator</a></li>
	<li><a id="gsMenuLink" href="../cgi-bin/hgNear">Gene Sorter</a></li>
	<li><a id="ggMenuLink" href="../cgi-bin/hgGenome">Genome Graphs</a></li>
	<li><a id="ispMenuLink" href="../cgi-bin/hgPcr">In-Silico PCR</a></li>
	<li><a id="liftOverMenuLink" href="../cgi-bin/hgLiftOver">LiftOver</a></li>
	<li><a id="visiGeneMenuLink" href="../cgi-bin/hgVisiGene">VisiGene</a></li>
	<li class="last"><a href="../util.html">Other Utilities</a></li>
	</ul>
</li>
<li class="menuparent" id="mirrors"><span>Mirrors</span>
	<ul style="display: none; visibility: hidden;">
	<li class="first"><a href="../mirror.html">Third Party Mirrors</a></li>
	<li><a href="../goldenPath/help/mirror.html">Mirroring Instructions</a></li>
	
	</ul>

</li>
<li class="menuparent" id="downloads"><span>Downloads</span>
	<ul style="display: none; visibility: hidden;">
	<li class="first"><a href="http://hgdownload.gi.ucsc.edu/downloads.html">Genome Data</a></li>
	<li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#source_downloads">Source Code</a></li>
  <li><a href="https://genome-store.ucsc.edu">Genome Browser Store</a></li>
	<li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#utilities_downloads">Utilities</a></li>
	<li class="last"><a href="../goldenPath/help/ftp.html">FTP</a></li>
	<li class="last"><a href="../goldenPath/help/mysql.html">MySQL Access</a></li>
	</ul>
</li>
<li class="menuparent" id="myData"><span>My Data</span>
	<ul style="display: none; visibility: hidden;">
	<li><a id="sessionsMenuLink" href="../cgi-bin/hgSession?hgS_doMainPage=1">My Sessions</a></li>
	<li><a id="publicSessionsMenuLink" href="../cgi-bin/hgPublicSessions">Public Sessions</a></li>
	<li><a id="trackHubsMenuLink" href="../cgi-bin/hgHubConnect">Track Hubs</a></li>
	<li class="last"><a id="customTracksMenuLink" href="../cgi-bin/hgCustom">Custom Tracks</a></li>
	</ul>
</li>

<!-- OPTIONAL_VIEW_MENU -->

<li class="menuparent" id="help"><span>Help</span><ul style="display: none; visibility: hidden;">
<!-- CONTEXT_SPECIFIC_HELP -->
<li><a href="../FAQ/">FAQs & Search</a></li>
<li><a href="../goldenPath/help/hgTracksHelp.html">Browser Documentation</a></li>
<li><a href="../training/index.html">Training</a></li>
<li><a href="../contacts.html">Mailing Lists</a></li>
<li class="last"><a href="http://genomewiki.ucsc.edu/" target="_blank">GenomeWiki</a></li>
</ul>
</li>

<li class="menuparent last" id="aboutUs"><span>About Us</span>
	<ul style="display: none; visibility: hidden;">
	<li class="first odd"><a href="../goldenPath/newsarch.html">News</a></li>
	<li class="even"><a href="../goldenPath/pubs.html">Publications</a></li>
	<li class="even"><a href="http://genome.ucsc.edu/blog/">Blog</a></li>
	<li class="odd"><a href="../cite.html">Cite Us</a></li>
	<li class="even"><a href="../goldenPath/credits.html">Credits</a></li>
	<li class="odd"><a href="../goldenPath/releaseLog.html">Release Log</a></li>
	<li class="even"><a href="../staff.html">Staff</a></li>
	<li class="odd"><a href="../contacts.html">Contact Us</a></li>
	<li class="even"><a href="../conditions.html">Conditions of Use</a></li>
	<li class="odd"><a href="https://ucscgenomics.soe.ucsc.edu/careers/" TARGET="_blank">Jobs</a></li>
	<li class="even last"><a href="../license/">Licenses</a></li>
</ul>
</li>

</ul>
</div>
		</div>
	</div>
	<div class="downcorner"></div>

</div>
</div><!-- end main-menu -->
</div><!-- end main-menu-whole -->

<!-- +++++++++++++++++++++ CONTENT TABLES +++++++++++++++++++ -->
<TR><TD COLSPAN=3>
<div id=firstSection>      <!--outer table is for border purposes-->
      <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>
     <div class='subheadingBar'><div class='windowSize' id='sectTtl'>
Track and Table Releases Prior to 2017</div></div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


This release log contains information about data released prior to 2017. <b>For more recent log 
entries, please see <a href="releaseLog.html">this page</a>.</b><br><br>

This page contains track and table release information for the following genome assemblies:<br>
<ul>
<li><a CLASS="toc" HREF="#hg38">Human Dec. 2013 (GRCh38/hg38) (hg38)</a></li>
<li><a CLASS="toc" HREF="#hg19">Human Feb. 2009 (GRCh37/hg19) (hg19)</a></li>
<li><a CLASS="toc" HREF="#hg18">Human Mar. 2006 (NCBI36/hg18) (hg18)</a></li>
<li><a CLASS="toc" HREF="#hg17">Human May 2004 (NCBI35/hg17) (hg17)</a></li>
<li><a CLASS="toc" HREF="#hg16">Human July 2003 (NCBI34/hg16) (hg16)</a></li>
<li><a CLASS="toc" HREF="#mm10">Mouse Dec. 2011 (GRCm38/mm10) (mm10)</a></li>
<li><a CLASS="toc" HREF="#mm9">Mouse July 2007 (NCBI37/mm9) (mm9)</a></li>
<li><a CLASS="toc" HREF="#mm8">Mouse Feb. 2006 (NCBI36/mm8) (mm8)</a></li>
<li><a CLASS="toc" HREF="#mm7">Mouse Aug. 2005 (NCBI35/mm7) (mm7)</a></li>
<li><a CLASS="toc" HREF="#anoGam1">A. gambiae Feb. 2003 (IAGEC MOZ2/anoGam1) (anoGam1)</a></li>
<li><a CLASS="toc" HREF="#apiMel2">A. mellifera Jan. 2005 (Baylor 2.0/apiMel2) (apiMel2)</a></li>
<li><a CLASS="toc" HREF="#apiMel1">A. mellifera July 2004 (Baylor 1.2/apiMel1) (apiMel1)</a></li>
<li><a CLASS="toc" HREF="#vicPac2">Alpaca Mar. 2013 (Vicugna_pacos-2.0.1/vicPac2) (vicPac2)</a></li>
<li><a CLASS="toc" HREF="#vicPac1">Alpaca Jul. 2008 (Broad/vicPac1) (vicPac1)</a></li>
<li><a CLASS="toc" HREF="#allMis1">American alligator Aug. 2012 (allMis0.2/allMis1) (allMis1)</a></li>
<li><a CLASS="toc" HREF="#dasNov3">Armadillo Dec. 2011 (Baylor/dasNov3) (dasNov3)</a></li>
<li><a CLASS="toc" HREF="#gadMor1">Atlantic cod May 2010 (Genofisk GadMor_May2010/gadMor1) (gadMor1)</a></li>
<li><a CLASS="toc" HREF="#papAnu2">Baboon Mar. 2012 (Baylor Panu_2.0/papAnu2) (papAnu2)</a></li>
<li><a CLASS="toc" HREF="#papHam1">Baboon Nov. 2008 (Baylor Pham_1.0/papHam1) (papHam1)</a></li>
<li><a CLASS="toc" HREF="#panPan1">Bonobo May 2012 (Max-Planck/panPan1) (panPan1)</a></li>
<li><a CLASS="toc" HREF="#aptMan1">Brown kiwi Jun. 2015 (MPI-EVA AptMant0/aptMan1) (aptMan1)</a></li>
<li><a CLASS="toc" HREF="#melUnd1">Budgerigar Sep. 2011 (WUSTL v6.3/melUnd1) (melUnd1)</a></li>
<li><a CLASS="toc" HREF="#otoGar3">Bushbaby Mar. 2011 (Broad/otoGar3) (otoGar3)</a></li>
<li><a CLASS="toc" HREF="#caePb2">C. brenneri Feb. 2008 (WUGSC 6.0.1/caePb2) (caePb2)</a></li>
<li><a CLASS="toc" HREF="#caePb1">C. brenneri Jan. 2007 (WUGSC 4.0/caePb1) (caePb1)</a></li>
<li><a CLASS="toc" HREF="#cb3">C. briggsae Jan. 2007 (WUGSC 1.0/cb3) (cb3)</a></li>
<li><a CLASS="toc" HREF="#cb1">C. briggsae July 2002 (WormBase cb25.agp8/cb1) (cb1)</a></li>
<li><a CLASS="toc" HREF="#ce11">C. elegans Feb. 2013 (WBcel235/ce11) (ce11)</a></li>
<li><a CLASS="toc" HREF="#ce10">C. elegans Oct. 2010 (WS220/ce10) (ce10)</a></li>
<li><a CLASS="toc" HREF="#ce6">C. elegans May 2008 (WS190/ce6) (ce6)</a></li>
<li><a CLASS="toc" HREF="#ce4">C. elegans Jan. 2007 (WS170/ce4) (ce4)</a></li>
<li><a CLASS="toc" HREF="#ce2">C. elegans Mar. 2004 (WS120/ce2) (ce2)</a></li>
<li><a CLASS="toc" HREF="#ci2">C. intestinalis Mar. 2005 (JGI 2.1/ci2) (ci2)</a></li>
<li><a CLASS="toc" HREF="#ci1">C. intestinalis Dec. 2002 (JGI 1.0/ci1) (ci1)</a></li>
<li><a CLASS="toc" HREF="#caeJap1">C. japonica Mar. 2008 (WUGSC 3.0.2/caeJap1) (caeJap1)</a></li>
<li><a CLASS="toc" HREF="#caeRem3">C. remanei May 2007 (WUGSC 15.0.1/caeRem3) (caeRem3)</a></li>
<li><a CLASS="toc" HREF="#caeRem2">C. remanei Mar. 2006 (WUGSC 1.0/caeRem2) (caeRem2)</a></li>
<li><a CLASS="toc" HREF="#felCat8">Cat Nov. 2014 (ICGSC Felis_catus_8.0/felCat8) (felCat8)</a></li>
<li><a CLASS="toc" HREF="#felCat5">Cat Sep. 2011 (ICGSC Felis_catus 6.2/felCat5) (felCat5)</a></li>
<li><a CLASS="toc" HREF="#felCat4">Cat Dec. 2008 (NHGRI/GTB V17e/felCat4) (felCat4)</a></li>
<li><a CLASS="toc" HREF="#felCat3">Cat Mar. 2006 (Broad/felCat3) (felCat3)</a></li>
<li><a CLASS="toc" HREF="#galGal5">Chicken Dec 2015 (Gallus_gallus-5.0/galGal5) (galGal5)</a></li>
<li><a CLASS="toc" HREF="#galGal4">Chicken Nov. 2011 (ICGSC Gallus_gallus-4.0/galGal4) (galGal4)</a></li>
<li><a CLASS="toc" HREF="#galGal3">Chicken May 2006 (WUGSC 2.1/galGal3) (galGal3)</a></li>
<li><a CLASS="toc" HREF="#galGal2">Chicken Feb. 2004 (WUGSC 1.0/galGal2) (galGal2)</a></li>
<li><a CLASS="toc" HREF="#panTro5">Chimp May 2016 (Pan_tro 3.0/panTro5) (panTro5)</a></li>
<li><a CLASS="toc" HREF="#panTro4">Chimp Feb. 2011 (CSAC 2.1.4/panTro4) (panTro4)</a></li>
<li><a CLASS="toc" HREF="#panTro3">Chimp Oct. 2010 (CGSC 2.1.3/panTro3) (panTro3)</a></li>
<li><a CLASS="toc" HREF="#panTro2">Chimp Mar. 2006 (CGSC 2.1/panTro2) (panTro2)</a></li>
<li><a CLASS="toc" HREF="#panTro1">Chimp Nov. 2003 (CGSC 1.1/panTro1) (panTro1)</a></li>
<li><a CLASS="toc" HREF="#criGri1">Chinese hamster Jul. 2013 (C_griseus_v1.0/criGri1) (criGri1)</a></li>
<li><a CLASS="toc" HREF="#latCha1">Coelacanth Aug. 2011 (Broad/latCha1) (latCha1)</a></li>
<li><a CLASS="toc" HREF="#bosTau8">Cow Jun. 2014 (Bos_taurus_UMD_3.1.1/bosTau8) (bosTau8)</a></li>
<li><a CLASS="toc" HREF="#bosTau7">Cow Oct. 2011 (Baylor Btau_4.6.1/bosTau7) (bosTau7)</a></li>
<li><a CLASS="toc" HREF="#bosTau6">Cow Nov. 2009 (Bos_taurus_UMD_3.1/bosTau6) (bosTau6)</a></li>
<li><a CLASS="toc" HREF="#bosTau4">Cow Oct. 2007 (Baylor 4.0/bosTau4) (bosTau4)</a></li>
<li><a CLASS="toc" HREF="#bosTau3">Cow Aug. 2006 (Baylor 3.1/bosTau3) (bosTau3)</a></li>
<li><a CLASS="toc" HREF="#bosTau2">Cow Mar. 2005 (Baylor 2.0/bosTau2) (bosTau2)</a></li>
<li><a CLASS="toc" HREF="#macFas5">Crab-eating macaque Jun. 2013 (Macaca_fascicularis_5.0/macFas5) (macFas5)</a></li>
<li><a CLASS="toc" HREF="#droAna2">D. ananassae Aug. 2005 (Agencourt prelim/droAna2) (droAna2)</a></li>
<li><a CLASS="toc" HREF="#droAna1">D. ananassae July 2004 (TIGR/droAna1) (droAna1)</a></li>
<li><a CLASS="toc" HREF="#droEre1">D. erecta Aug. 2005 (Agencourt prelim/droEre1) (droEre1)</a></li>
<li><a CLASS="toc" HREF="#droGri1">D. grimshawi Aug. 2005 (Agencourt prelim/droGri1) (droGri1)</a></li>
<li><a CLASS="toc" HREF="#dm6">D. melanogaster Aug. 2014 (BDGP Release 6 + ISO1 MT/dm6) (dm6)</a></li>
<li><a CLASS="toc" HREF="#dm3">D. melanogaster Apr. 2006 (BDGP R5/dm3) (dm3)</a></li>
<li><a CLASS="toc" HREF="#dm2">D. melanogaster Apr. 2004 (BDGP R4/dm2) (dm2)</a></li>
<li><a CLASS="toc" HREF="#dm1">D. melanogaster Jan. 2003 (BDGP R3/dm1) (dm1)</a></li>
<li><a CLASS="toc" HREF="#droMoj2">D. mojavensis Aug. 2005 (Agencourt prelim/droMoj2) (droMoj2)</a></li>
<li><a CLASS="toc" HREF="#droMoj1">D. mojavensis Aug. 2004 (Agencourt prelim/droMoj1) (droMoj1)</a></li>
<li><a CLASS="toc" HREF="#droPer1">D. persimilis Oct. 2005 (Broad/droPer1) (droPer1)</a></li>
<li><a CLASS="toc" HREF="#dp3">D. pseudoobscura Nov. 2004 (FlyBase 1.03/dp3) (dp3)</a></li>
<li><a CLASS="toc" HREF="#dp2">D. pseudoobscura Aug. 2003 (Baylor freeze1/dp2) (dp2)</a></li>
<li><a CLASS="toc" HREF="#droSec1">D. sechellia Oct. 2005 (Broad/droSec1) (droSec1)</a></li>
<li><a CLASS="toc" HREF="#droSim1">D. simulans Apr. 2005 (WUGSC mosaic 1.0/droSim1) (droSim1)</a></li>
<li><a CLASS="toc" HREF="#droVir2">D. virilis Aug. 2005 (Agencourt prelim/droVir2) (droVir2)</a></li>
<li><a CLASS="toc" HREF="#droVir1">D. virilis July 2004 (Agencourt prelim/droVir1) (droVir1)</a></li>
<li><a CLASS="toc" HREF="#droYak2">D. yakuba Nov. 2005 (WUGSC 7.1/droYak2) (droYak2)</a></li>
<li><a CLASS="toc" HREF="#droYak1">D. yakuba Apr. 2004 (WUGSC 1.0/droYak1) (droYak1)</a></li>
<li><a CLASS="toc" HREF="#canFam3">Dog Sep. 2011 (Broad CanFam3.1/canFam3) (canFam3)</a></li>
<li><a CLASS="toc" HREF="#canFam2">Dog May 2005 (Broad/canFam2) (canFam2)</a></li>
<li><a CLASS="toc" HREF="#canFam1">Dog July 2004 (Broad/canFam1) (canFam1)</a></li>
<li><a CLASS="toc" HREF="#turTru2">Dolphin Oct. 2011 (Baylor Ttru_1.4/turTru2) (turTru2)</a></li>
<li><a CLASS="toc" HREF="#eboVir3">Ebola virus Sierra Leone 2014 (G3683/KM034562.1/eboVir3) (eboVir3)</a></li>
<li><a CLASS="toc" HREF="#loxAfr3">Elephant Jul. 2009 (Broad/loxAfr3) (loxAfr3)</a></li>
<li><a CLASS="toc" HREF="#calMil1">Elephant shark Dec. 2013 (Callorhinchus_milii-6.1.3/calMil1) (calMil1)</a></li>
<li><a CLASS="toc" HREF="#musFur1">Ferret  Apr. 2011 (MusPutFur1.0/musFur1) (musFur1)</a></li>
<li><a CLASS="toc" HREF="#fr3">Fugu Oct. 2011 (FUGU5/fr3) (fr3)</a></li>
<li><a CLASS="toc" HREF="#fr2">Fugu Oct. 2004 (JGI 4.0/fr2) (fr2)</a></li>
<li><a CLASS="toc" HREF="#fr1">Fugu Aug. 2002 (JGI 3.0/fr1) (fr1)</a></li>
<li><a CLASS="toc" HREF="#nomLeu3">Gibbon Oct. 2012 (GGSC Nleu3.0/nomLeu3) (nomLeu3)</a></li>
<li><a CLASS="toc" HREF="#nomLeu2">Gibbon Jun. 2011 (GGSC Nleu1.1/nomLeu2) (nomLeu2)</a></li>
<li><a CLASS="toc" HREF="#nomLeu1">Gibbon Jan. 2010 (GGSC Nleu1.0/nomLeu1) (nomLeu1)</a></li>
<li><a CLASS="toc" HREF="#gorGor5">Gorilla Mar. 2016 (GSMRT3/gorGor5) (gorGor5)</a></li>
<li><a CLASS="toc" HREF="#gorGor4">Gorilla Dec 2014 (gorGor4.1/gorGor4) (gorGor4)</a></li>
<li><a CLASS="toc" HREF="#gorGor3">Gorilla May 2011 (gorGor3.1/gorGor3) (gorGor3)</a></li>
<li><a CLASS="toc" HREF="#chlSab2">Green monkey Mar. 2014 (Chlorocebus_sabeus 1.1/chlSab2) (chlSab2)</a></li>
<li><a CLASS="toc" HREF="#cavPor3">Guinea pig Feb. 2008 (Broad/cavPor3) (cavPor3)</a></li>
<li><a CLASS="toc" HREF="#eriEur2">Hedgehog May 2012 (EriEur2.0/eriEur2) (eriEur2)</a></li>
<li><a CLASS="toc" HREF="#eriEur1">Hedgehog June 2006 (Broad/eriEur1) (eriEur1)</a></li>
<li><a CLASS="toc" HREF="#equCab2">Horse Sep. 2007 (Broad/equCab2) (equCab2)</a></li>
<li><a CLASS="toc" HREF="#equCab1">Horse Jan. 2007 (Broad/equCab1) (equCab1)</a></li>
<li><a CLASS="toc" HREF="#dipOrd1">Kangaroo rat Jul. 2008 (Broad/dipOrd1) (dipOrd1)</a></li>
<li><a CLASS="toc" HREF="#petMar2">Lamprey Sep. 2010 (WUGSC 7.0/petMar2) (petMar2)</a></li>
<li><a CLASS="toc" HREF="#petMar1">Lamprey Mar. 2007 (WUGSC 3.0/petMar1) (petMar1)</a></li>
<li><a CLASS="toc" HREF="#braFlo1">Lancelet Mar. 2006 (JGI 1.0/braFlo1) (braFlo1)</a></li>
<li><a CLASS="toc" HREF="#anoCar2">Lizard May 2010 (Broad AnoCar2.0/anoCar2) (anoCar2)</a></li>
<li><a CLASS="toc" HREF="#anoCar1">Lizard Feb. 2007 (Broad/anoCar1) (anoCar1)</a></li>
<li><a CLASS="toc" HREF="#galVar1">Malayan flying lemur Jun. 2014 (G_variegatus-3.0.2/galVar1) (galVar1)</a></li>
<li><a CLASS="toc" HREF="#triMan1">Manatee Oct. 2011 (Broad v1.0/triMan1) (triMan1)</a></li>
<li><a CLASS="toc" HREF="#calJac3">Marmoset March 2009 (WUGSC 3.2/calJac3) (calJac3)</a></li>
<li><a CLASS="toc" HREF="#calJac1">Marmoset June 2007 (WUGSC 2.0.2/calJac1) (calJac1)</a></li>
<li><a CLASS="toc" HREF="#oryLat2">Medaka Oct. 2005 (NIG/UT MEDAKA1/oryLat2) (oryLat2)</a></li>
<li><a CLASS="toc" HREF="#geoFor1">Medium ground finch Apr. 2012 (GeoFor_1.0/geoFor1) (geoFor1)</a></li>
<li><a CLASS="toc" HREF="#pteVam1">Megabat Jul. 2008 (Broad/pteVam1) (pteVam1)</a></li>
<li><a CLASS="toc" HREF="#myoLuc2">Microbat Jul. 2010 (Broad Institute Myoluc2.0/myoLuc2) (myoLuc2)</a></li>
<li><a CLASS="toc" HREF="#balAcu1">Minke whale Oct. 2013 (BalAcu1.0/balAcu1) (balAcu1)</a></li>
<li><a CLASS="toc" HREF="#micMur2">Mouse lemur May 2015 (Mouse lemur/micMur2) (micMur2)</a></li>
<li><a CLASS="toc" HREF="#micMur1">Mouse lemur Jul. 2007 (Broad/micMur1) (micMur1)</a></li>
<li><a CLASS="toc" HREF="#hetGla2">Naked mole-rat Jan. 2012 (Broad HetGla_female_1.0/hetGla2) (hetGla2)</a></li>
<li><a CLASS="toc" HREF="#hetGla1">Naked mole-rat Jul. 2011 (BGI HetGla_1.0/hetGla1) (hetGla1)</a></li>
<li><a CLASS="toc" HREF="#oreNil2">Nile tilapia Jan. 2011 (Broad oreNil1.1/oreNil2) (oreNil2)</a></li>
<li><a CLASS="toc" HREF="#monDom5">Opossum Oct. 2006 (Broad/monDom5) (monDom5)</a></li>
<li><a CLASS="toc" HREF="#monDom4">Opossum Jan. 2006 (Broad/monDom4) (monDom4)</a></li>
<li><a CLASS="toc" HREF="#monDom1">Opossum Oct. 2004 (Broad prelim/monDom1) (monDom1)</a></li>
<li><a CLASS="toc" HREF="#ponAbe2">Orangutan July 2007 (WUGSC 2.0.2/ponAbe2) (ponAbe2)</a></li>
<li><a CLASS="toc" HREF="#priPac1">P. pacificus Feb. 2007 (WUGSC 5.0/priPac1) (priPac1)</a></li>
<li><a CLASS="toc" HREF="#chrPic1">Painted turtle Dec. 2011 (v3.0.1/chrPic1) (chrPic1)</a></li>
<li><a CLASS="toc" HREF="#ailMel1">Panda Dec. 2009 (BGI-Shenzhen 1.0/ailMel1) (ailMel1)</a></li>
<li><a CLASS="toc" HREF="#susScr3">Pig Aug. 2011 (SGSC Sscrofa10.2/susScr3) (susScr3)</a></li>
<li><a CLASS="toc" HREF="#susScr2">Pig Nov. 2009 (SGSC Sscrofa9.2/susScr2) (susScr2)</a></li>
<li><a CLASS="toc" HREF="#ochPri3">Pika May 2012 (OchPri3.0/ochPri3) (ochPri3)</a></li>
<li><a CLASS="toc" HREF="#ochPri2">Pika Jul. 2008 (Broad/ochPri2) (ochPri2)</a></li>
<li><a CLASS="toc" HREF="#ornAna2">Platypus Feb. 2007 (ASM227v2/ornAna2) (ornAna2)</a></li>
<li><a CLASS="toc" HREF="#ornAna1">Platypus Mar. 2007 (WUGSC 5.0.1/ornAna1) (ornAna1)</a></li>
<li><a CLASS="toc" HREF="#oryCun2">Rabbit Apr. 2009 (Broad/oryCun2) (oryCun2)</a></li>
<li><a CLASS="toc" HREF="#rn6">Rat Jul. 2014 (RGSC 6.0/rn6) (rn6)</a></li>
<li><a CLASS="toc" HREF="#rn5">Rat Mar. 2012 (RGSC 5.0/rn5) (rn5)</a></li>
<li><a CLASS="toc" HREF="#rn4">Rat Nov. 2004 (Baylor 3.4/rn4) (rn4)</a></li>
<li><a CLASS="toc" HREF="#rn3">Rat June 2003 (Baylor 3.1/rn3) (rn3)</a></li>
<li><a CLASS="toc" HREF="#rheMac8">Rhesus Nov. 2015 (BCM Mmul_8.0.1/rheMac8) (rheMac8)</a></li>
<li><a CLASS="toc" HREF="#rheMac3">Rhesus Oct. 2010 (BGI CR_1.0/rheMac3) (rheMac3)</a></li>
<li><a CLASS="toc" HREF="#rheMac2">Rhesus Jan. 2006 (MGSC Merged 1.0/rheMac2) (rheMac2)</a></li>
<li><a CLASS="toc" HREF="#proCap1">Rock hyrax Jul. 2008 (Broad/proCap1) (proCap1)</a></li>
<li><a CLASS="toc" HREF="#sacCer3">S. cerevisiae Apr. 2011 (SacCer_Apr2011/sacCer3) (sacCer3)</a></li>
<li><a CLASS="toc" HREF="#sacCer2">S. cerevisiae June 2008 (SGD/sacCer2) (sacCer2)</a></li>
<li><a CLASS="toc" HREF="#sacCer1">S. cerevisiae Oct. 2003 (SGD/sacCer1) (sacCer1)</a></li>
<li><a CLASS="toc" HREF="#strPur2">S. purpuratus Sep. 2006 (Baylor 2.1/strPur2) (strPur2)</a></li>
<li><a CLASS="toc" HREF="#strPur1">S. purpuratus Apr. 2005 (Baylor 1.1/strPur1) (strPur1)</a></li>
<li><a CLASS="toc" HREF="#aplCal1">Sea hare Sept. 2008 (Broad 2.0/aplCal1) (aplCal1)</a></li>
<li><a CLASS="toc" HREF="#oviAri3">Sheep Aug. 2012 (ISGC Oar_v3.1/oviAri3) (oviAri3)</a></li>
<li><a CLASS="toc" HREF="#oviAri1">Sheep Feb. 2010 (ISGC Ovis_aries_1.0/oviAri1) (oviAri1)</a></li>
<li><a CLASS="toc" HREF="#sorAra2">Shrew Aug. 2008 (Broad/sorAra2) (sorAra2)</a></li>
<li><a CLASS="toc" HREF="#sorAra1">Shrew June 2006 (Broad/sorAra1) (sorAra1)</a></li>
<li><a CLASS="toc" HREF="#choHof1">Sloth Jul. 2008 (Broad/choHof1) (choHof1)</a></li>
<li><a CLASS="toc" HREF="#speTri2">Squirrel Nov. 2011 (Broad/speTri2) (speTri2)</a></li>
<li><a CLASS="toc" HREF="#saiBol1">Squirrel monkey Oct. 2011 (Broad/saiBol1) (saiBol1)</a></li>
<li><a CLASS="toc" HREF="#gasAcu1">Stickleback Feb. 2006 (Broad/gasAcu1) (gasAcu1)</a></li>
<li><a CLASS="toc" HREF="#tarSyr2">Tarsier Sep. 2013 (Tarsius_syrichta-2.0.1/tarSyr2) (tarSyr2)</a></li>
<li><a CLASS="toc" HREF="#tarSyr1">Tarsier Aug. 2008 (Broad/tarSyr1) (tarSyr1)</a></li>
<li><a CLASS="toc" HREF="#sarHar1">Tasmanian devil Feb. 2011 (WTSI Devil_ref v7.0/sarHar1) (sarHar1)</a></li>
<li><a CLASS="toc" HREF="#echTel2">Tenrec Nov. 2012 (Broad/echTel2) (echTel2)</a></li>
<li><a CLASS="toc" HREF="#echTel1">Tenrec July 2005 (Broad/echTel1) (echTel1)</a></li>
<li><a CLASS="toc" HREF="#tetNig2">Tetraodon Mar. 2007 (Genoscope 8.0/tetNig2) (tetNig2)</a></li>
<li><a CLASS="toc" HREF="#tetNig1">Tetraodon Feb. 2004 (Genoscope 7/tetNig1) (tetNig1)</a></li>
<li><a CLASS="toc" HREF="#tupBel1">Tree shrew Dec. 2006 (Broad/tupBel1) (tupBel1)</a></li>
<li><a CLASS="toc" HREF="#melGal1">Turkey Dec. 2009 (TGC Turkey_2.01/melGal1) (melGal1)</a></li>
<li><a CLASS="toc" HREF="#macEug2">Wallaby Sep. 2009 (TWGS Meug_1.1/macEug2) (macEug2)</a></li>
<li><a CLASS="toc" HREF="#cerSim1">White rhinoceros May 2012 (CerSimSim1.0/cerSim1) (cerSim1)</a></li>
<li><a CLASS="toc" HREF="#xenTro7">X. tropicalis Sep. 2012 (JGI 7.0/xenTro7) (xenTro7)</a></li>
<li><a CLASS="toc" HREF="#xenTro3">X. tropicalis Nov. 2009 (JGI 4.2/xenTro3) (xenTro3)</a></li>
<li><a CLASS="toc" HREF="#xenTro2">X. tropicalis Aug. 2005 (JGI 4.1/xenTro2) (xenTro2)</a></li>
<li><a CLASS="toc" HREF="#xenTro1">X. tropicalis Oct. 2004 (JGI 3.0/xenTro1) (xenTro1)</a></li>
<li><a CLASS="toc" HREF="#taeGut2">Zebra finch Feb. 2013 (WashU taeGut324/taeGut2) (taeGut2)</a></li>
<li><a CLASS="toc" HREF="#taeGut1">Zebra finch Jul. 2008 (WUGSC 3.2.4/taeGut1) (taeGut1)</a></li>
<li><a CLASS="toc" HREF="#danRer10">Zebrafish Sep. 2014 (GRCz10/danRer10) (danRer10)</a></li>
<li><a CLASS="toc" HREF="#danRer7">Zebrafish Jul. 2010 (Zv9/danRer7) (danRer7)</a></li>
<li><a CLASS="toc" HREF="#danRer6">Zebrafish Dec. 2008 (Zv8/danRer6) (danRer6)</a></li>
<li><a CLASS="toc" HREF="#danRer5">Zebrafish July 2007 (Zv7/danRer5) (danRer5)</a></li>
<li><a CLASS="toc" HREF="#danRer4">Zebrafish Mar. 2006 (Zv6/danRer4) (danRer4)</a></li>
<li><a CLASS="toc" HREF="#danRer3">Zebrafish May 2005 (Zv5/danRer3) (danRer3)</a></li>
</ul>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=hg38></A>Human Dec. 2013 (GRCh38/hg38) (hg38, GRCh38 Genome Reference Consortium Human Reference 38 (GCA_000001405.2))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
hg38 uniprot track bug fixes</td>
<td>09 Dec 2016</td>
</tr>
<TR valign=top><TD align=left>
CRISPR</td>
<td>07 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Chimp Chain/Net</td>
<td>31 Oct 2016</td>
</tr>
<TR valign=top><TD align=left>
Gorilla Chain/Net</td>
<td>28 Oct 2016</td>
</tr>
<TR valign=top><TD align=left>
Green monkey Chain/Net</td>
<td>28 Oct 2016</td>
</tr>
<TR valign=top><TD align=left>
Chicken Chain/Net</td>
<td>26 Sep 2016</td>
</tr>
<TR valign=top><TD align=left>
Locus Reference Genome (LRG) Regions and Transcripts</td>
<td>23 Sep 2016</td>
</tr>
<TR valign=top><TD align=left>
DGV - data update (2016-05-15) on hg18, hg19 and hg38</td>
<td>20 Sep 2016</td>
</tr>
<TR valign=top><TD align=left>
Malayan flying lemur Chain/Net</td>
<td>16 Sep 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../cgi-bin/hgTrackUi?db=hg38&g=spUniprot" target=_blank>UniProt Annotations on hg38</a></td>
<td>24 Aug 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeGencodeV24" target=_blank>GENCODE V24 release</a></td>
<td>14 Jul 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=snp147Common" target=_blank>Common SNPs(147), Flagged SNPs(147), Mult. SNPs(147), All SNPS(147) initial release</a></td>
<td>07 Jul 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg38&g=hg38Patch7" target=_blank>GRCh38 patch release 7, alternate sequences and reference sequence patches</a></td>
<td>23 Jun 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=gtexGene" target=_blank>Gene Expression in 53 Tissue from GTEx RNA-seq</a></td>
<td>22 Apr 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=gwipsvizRiboseq" target=_blank>Ribosome Profiling from GWIPS-viz</a></td>
<td>20 Apr 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=primateChainNet" target=_blank>Gorilla Chain/Net</a></td>
<td>19 Apr 2016</td>
</tr>
<TR valign=top><TD align=left>
Rhesus Chain/Net</td>
<td>18 Apr 2016</td>
</tr>
<TR valign=top><TD align=left>
Common SNPs(146), Flagged SNPs(146), Mult. SNPs(146), All SNPS(146) initial release</td>
<td>15 Apr 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=primateChainNet" target=_blank>Crab-eating macaque Chain/Net</a></td>
<td>07 Apr 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=fishClones" target=_blank>FISH Clones (lifted from hg18) </a></td>
<td>09 Mar 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg38&g=hg38Patch6" target=_blank>GRCh38 patch release 6, alternate sequences and reference sequence patches</a></td>
<td>07 Mar 2016</td>
</tr>
<TR valign=top><TD align=left>
Mouse lemur Chain/Net</td>
<td>12 Feb 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg38&g=knownGene" target=_blank>UCSC Genes (update to Lynx linkout)</a></td>
<td>08 Feb 2016</td>
</tr>
<TR valign=top><TD align=left>
dbNSFP release 3.1a (data files only, for use by hgVai)</td>
<td>25 Jan 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=snp144" target=_blank>Common SNPs(144), Flagged SNPs(144), Mult. SNPs(144), All SNPS(144) initial release</a></td>
<td>11 Jan 2016</td>
</tr>
<TR valign=top><TD align=left>
X. tropicalis Chain/Net</td>
<td>22 Dec 2015</td>
</tr>
<TR valign=top><TD align=left>
Cat Chain/Net</td>
<td>22 Dec 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=snp142" target=_blank>Common SNPs(142), Flagged SNPs(142), Mult. SNPs(142), All SNPS(142) Removal of incorrect allele frequency data for ~1M SNPs submitted by 1000 Genomes assigned to the '-' strand</a></td>
<td>23 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>20 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeGencodeV23" target=_blank>GENCODE V23 release</a></td>
<td>17 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
Copy Number Variation Morbidity Map of Developmental Delay</td>
<td>17 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Vertebrate Multiz Alignment & Conservation (100 Species)</td>
<td>14 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=stsMap" target=_blank>STS Markers on Genetic and Radiation Hybrid Maps</a></td>
<td>02 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=hgIkmc" target=_blank>IKMC Genes Mapped (Lifted from hg19)</a></td>
<td>28 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
SGP Gene Predictions Using homology</td>
<td>27 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../cgi-bin/hgTrackUi?db=hg38&g=coriellDelDup" target=_blank>Coriell CNVs (lift from hg19 to hg38) </a></td>
<td>24 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
DGV - data update (2015-07-23) on hg18, hg19 and hg38</td>
<td>21 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=cloneEndSuper" target=_blank>Mapping of clone libraries end placements </a></td>
<td>21 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../cgi-bin/hgTrackUi?db=hg38&g=nonCodingRNAs" target=_blank>Non-coding RNA super track</a></td>
<td>19 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=iscaComposite" target=_blank>Clinical Genome Resource (ClinGen) CNVs</a></td>
<td>30 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeRegDnaseClustered" target=_blank>ENCODE Regulation: DNase I Hypersensitivity Peak Clusters</a></td>
<td>19 Jun 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeRegDnaseWig" target=_blank>ENCODE Regulation:  DNase I Hypersensitivity Signal Colored by Similarity</a></td>
<td>19 Jun 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeRegDnase" target=_blank>ENCODE Regulation:  DNase I Hypersensitivity in 95 cell types</a></td>
<td>19 Jun 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeReg" target=_blank>ENCODE Regulation (Lifted from hg19)</a></td>
<td>19 Jun 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg38&g=hg38Patch3" target=_blank>GRCh38 patch release 3, alternate sequences and reference sequence patches</a></td>
<td>27 May 2015</td>
</tr>
<TR valign=top><TD align=left>
Tarsier Chain/Net</td>
<td>22 May 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg38&g=cons20way" target=_blank>Primate 20-way Conservation</a></td>
<td>18 May 2015</td>
</tr>
<TR valign=top><TD align=left>
Dog (Sep. 2011 (Broad CanFam3.1/canFam3)) Chained Alignments
Rhesus (Oct. 2010 (BGI CR_1.0/rheMac3)) Chained Alignments
Mouse (Dec. 2011 (GRCm38/mm10)) Chained Alignments</td>
<td>14 May 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeGencodeV22" target=_blank>GENCODE Genes V22</a></td>
<td>30 Apr 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg38&g=vertebrateChainNet" target=_blank>Zebrafish Chain/Net</a></td>
<td>01 Apr 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg38&g=hg38Patch2" target=_blank>GRCh38 patch release 2, alternate sequences and reference sequence patches</a></td>
<td>04 Mar 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=ucscRetroAli9" target=_blank>Retroposed Genes V9, Including Pseudogenes</a></td>
<td>24 Feb 2015</td>
</tr>
<TR valign=top><TD align=left>
DGV - data update (2014-10-16) on hg18 and hg19; first release on hg38</td>
<td>23 Feb 2015</td>
</tr>
<TR valign=top><TD align=left>
Cow Chain/Net</td>
<td>17 Feb 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=snp142" target=_blank>Common SNPs(142), Flagged SNPs(142), Mult. SNPs(142), All SNPS(142) initial release</a></td>
<td>11 Feb 2015</td>
</tr>
<TR valign=top><TD align=left>
ClinVar Variants </td>
<td>05 Feb 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg38&g=sibTxGraph" target=_blank>SIB Alt-Splicing</a></td>
<td>03 Nov 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg38&g=sibGene" target=_blank>SIB Genes</a></td>
<td>30 Oct 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=snp141" target=_blank>Common SNPs(141), Flagged SNPs(141), Mult. SNPs(141), All SNPS(141) initial release</a></td>
<td>20 Oct 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=genomicSuperDups" target=_blank>Duplications of >1000 Bases of Non-RepeatMasked Sequence </a></td>
<td>15 Oct 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeGencodeV20" target=_blank>GENCODE V20 / Ensembl 76 release for GRCh38</a></td>
<td>13 Oct 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=centromeres" target=_blank>Centromere Locations</a></td>
<td>18 Aug 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=cons7way" target=_blank>Human (hg38) 7-way conservation</a></td>
<td>11 Aug 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=cytoBand" target=_blank>Chromosome Band</a></td>
<td>07 Aug 2014</td>
</tr>
<TR valign=top><TD align=left>
Locus Reference Genome (LRG) Regions and Transcripts</td>
<td>21 Jul 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg38&g=grcIncidentDb" target=_blank>GRC Incident Database</a></td>
<td>01 Jul 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=omimGene2" target=_blank>OMIM Genes</a></td>
<td>23 Jun 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=geneReviews" target=_blank>GeneReviews</a></td>
<td>19 May 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg38&g=knownGene" target=_blank>UCSC Genes</a></td>
<td>16 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Initial hg38 release (using GRCh38 Genome Reference Consortium Human Reference 38 (GCA_000001405.2)): Assembly, Hg19 Diff, Alt Map, Gap, GC Percent, GRC Contigs, INSDC, Scaffolds, RefSeq Genes, GENCODE Genes V19, Genscan Genes, MGC Genes, ORFeome Clones, Other RefSeq, Human mRNAs, Spliced ESTs, Human ESTs, Other ESTs, Other mRNAs, CpG Islands, Dog Chain/Net, Mouse Chain/Net, Primate Chain/Net, Self Chain, RepeatMasker, Interrupted Repeats, Simple Repeats, WM + SDust</td>
<td>06 Mar 2014</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=hg19></A>Human Feb. 2009 (GRCh37/hg19) (hg19, GRCh37 Genome Reference Consortium Human Reference 37 (GCA_000001405.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
CRISPR</td>
<td>07 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
DGV - data update (2016-05-15) on hg18, hg19 and hg38</td>
<td>20 Sep 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=snp147Common" target=_blank>Common SNPs(147), Flagged SNPs(147), Mult. SNPs(147), All SNPS(147) initial release</a></td>
<td>07 Jul 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV24lift37" target=_blank>GENCODE V24 mapped from GRCh38/hg38 to GRCh37/hg19</a></td>
<td>02 May 2016</td>
</tr>
<TR valign=top><TD align=left>
Common SNPs(146), Flagged SNPs(146), Mult. SNPs(146), All SNPS(146) initial release</td>
<td>15 Apr 2016</td>
</tr>
<TR valign=top><TD align=left>
Sequences from the Lens PatSeq Patent Document Database</td>
<td>11 Apr 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=primateChainNet" target=_blank>Crab-eating macaque Chain/Net</a></td>
<td>07 Apr 2016</td>
</tr>
<TR valign=top><TD align=left>
1000 Genomes Phase 3 Variants -- corrected filenames for chrX and chrY</td>
<td>03 Mar 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=snp144" target=_blank>Common SNPs(144), Flagged SNPs(144), Mult. SNPs(144), All SNPS(144) initial release</a></td>
<td>11 Jan 2016</td>
</tr>
<TR valign=top><TD align=left>
1000 Genomes Phase 3 Paired-end Accessible Regions and Integrated Variant Calls: SNVs, InDels, SVs</td>
<td>14 Dec 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=snp142" target=_blank>Common SNPs(142), Flagged SNPs(142), Mult. SNPs(142), All SNPS(142) Removal of incorrect allele frequency data for ~1M SNPs submitted by 1000 Genomes assigned to the '-' strand</a></td>
<td>23 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../cgi-bin/hgTrackUi?hdb=hg19&g=tRNAs" target=_blank>tRNAs (Update from the Lowe lab)</a></td>
<td>12 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
Copy Number Variation Morbidity Map of Developmental Delay</td>
<td>17 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
DGV - data update (2015-07-23) on hg18, hg19 and hg38</td>
<td>21 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=gwipsvizRiboseq" target=_blank>Ribosome Profiling from GWIPS-viz</a></td>
<td>19 Jun 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=exac" target=_blank>Exome Aggregation Consortium (ExAC) Variants and Calling Regions</a></td>
<td>17 Apr 2015</td>
</tr>
<TR valign=top><TD align=left>
DGV - data update (2014-10-16) on hg18 and hg19; first release on hg38</td>
<td>23 Feb 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=snp142" target=_blank>Common SNPs(142), Flagged SNPs(142), Mult. SNPs(142), All SNPS(142) initial release</a></td>
<td>11 Feb 2015</td>
</tr>
<TR valign=top><TD align=left>
ClinVar Variants </td>
<td>05 Feb 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=hgmd" target=_blank>Human Gene Mutation Database (update)</a></td>
<td>12 Jan 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=peptideAtlas" target=_blank>Peptide sequences identified from MS spectra of 971 samples by PeptideAtlas</a></td>
<td>06 Jan 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeAwgDnaseMasterSites" target=_blank>DNaseI Hypersensitive Site Master List (125 cell types) from ENCODE/Analysis </a></td>
<td>24 Nov 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRegDnaseClustered" target=_blank>ENCODE Regulation: DNase Clusters in 125 cell types from ENCODE (update to include clusters from 1 cell type, and to extend schema to include sources)</a></td>
<td>03 Nov 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=snp141" target=_blank>Common SNPs(141), Flagged SNPs(141), All SNPS(141) initial release</a></td>
<td>20 Oct 2014</td>
</tr>
<TR valign=top><TD align=left>
Denisova and Neandertal Reconstructed DNA Methylation Maps</td>
<td>18 Sep 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=placentalChainNet" target=_blank>Shrew Chain/Net</a></td>
<td>14 Jul 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=placentalChainNet" target=_blank>Pika Chain/Net</a></td>
<td>14 Jul 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=vertebrateChainNet" target=_blank>Zebra finch Chain/Net</a></td>
<td>08 May 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=placentalChainNet" target=_blank>Hedgehog Chain/Net</a></td>
<td>08 May 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=placentalChainNet" target=_blank>Chinese hamster Chain/Net</a></td>
<td>08 May 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=evsEsp6500" target=_blank>EVS Variants (initial push)</a></td>
<td>07 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Vertebrate Multiz Alignment & Conservation (100 Species)</td>
<td>01 May 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=snp138" target=_blank>Common SNPs(138), Flagged SNPs(138), Mult. SNPs(138), All SNPS(138) rebuilt with updated files from dbSNP</a></td>
<td>24 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeAwgSegmentation" target=_blank>Genome State Segmentations from ENCODE</a></td>
<td>17 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=chainNetFelCat5" target=_blank>Cat Chain/Net</a></td>
<td>03 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=gad" target=_blank>GAD: Genetic Association Studies of Complex Diseases and Disorders (update)</a></td>
<td>18 Feb 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRegTfbsClusteredV3" target=_blank>ENCODE Regulation: Txn Factor ChIP Clusters (update to 161 TF targets, from uniform processing of June 2012 freeze)</a></td>
<td>24 Jan 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://text.soe.ucsc.edu" target=_blank>pubsBingBlat (a run of hg19-BLAT against all webpages on the internet indexed by bing.com)</a></td>
<td>23 Jan 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV19" target=_blank>GENCODE V19 release</a></td>
<td>15 Jan 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes v74</td>
<td>23 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Locus Reference Genome (LRG) Regions and Transcripts</td>
<td>17 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=ucsfBrainMethyl" target=_blank>UCSF Brain DNA Methylation </a></td>
<td>10 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Vertebrate Multiz Alignment & Conservation (100 Species)</td>
<td>21 Nov 2013</td>
</tr>
<TR valign=top><TD align=left>
phastBias gBGC - regions predicted to be influenced by GC-biased gene conversion </td>
<td>08 Nov 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=snp138" target=_blank>Common SNPs(138), Flagged SNPs(138), Mult. SNPs(138), All SNPS(138)</a></td>
<td>24 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
American alligator Chain/Net</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=clinvar" target=_blank>ClinVar Variants</a></td>
<td>14 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
DGV - data update (v11)</td>
<td>11 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Alpaca Chain/Net</td>
<td>13 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
SNPs(135/137): Corrected some allele frequencies that had been tallied incorrectly due to strand error.</td>
<td>05 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Sheep Chain/Net</td>
<td>04 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=placentalChainNet" target=_blank>Tenrec Chain/Net</a></td>
<td>29 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Retroposed Genes for hg19, Build 5.0
Retroposed Genes for mm10, Build 2.0
</td>
<td>22 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Uniprot amino acid variants</td>
<td>16 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
hgmd</td>
<td>15 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV17" target=_blank>GENCODE Version 17 release</a></td>
<td>05 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=pgSnp" target=_blank>Genome Variants: 1000 Genomes tracks updated; several new tracks added</a></td>
<td>01 Jul 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=primateChainNet" target=_blank>Chimp Chain/Net</a></td>
<td>21 Jun 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=ensGene" target=_blank>Ensembl Genes v71</a></td>
<td>11 Jun 2013</td>
</tr>
<TR valign=top><TD align=left>
Wikipedia links on UCSC Genes</td>
<td>05 Jun 2013</td>
</tr>
<TR valign=top><TD align=left>
Gibbon Chain/Net</td>
<td>29 May 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=gwasCatalog" target=_blank>GWAS Catalog (update)</a></td>
<td>17 May 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeAwgTfbsUniform" target=_blank>Transcription Factor ChIP-seq Uniform Peaks from ENCODE/Analysis</a></td>
<td>10 May 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=ensGene" target=_blank>Ensembl Genes v70</a></td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=dgv" target=_blank>Database of Genomic Variants: Structural Variation (CNV, Inversion, In/del) (data update)</a></td>
<td>14 Feb 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=dhcHumDerDenAnc" target=_blank>Modern Derived
dhcHumDerDenAnc</a></td>
<td>11 Feb 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=dhcVcfModern" target=_blank>Variants called from high-coverage sequence of 11 modern humans</a></td>
<td>11 Feb 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=dhcVcfDenisovaPinky" target=_blank>Variants called from high-coverage Denisova Sequence</a></td>
<td>11 Feb 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=dhcBamDenisova" target=_blank>Denisova High-coverage Sequence Reads</a></td>
<td>11 Feb 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>07 Feb 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=placentalChainNet" target=_blank>Dog Chain/Net</a></td>
<td>31 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
qPCR Primers</td>
<td>29 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=ucscGenePfam" target=_blank>Pfam Domains in UCSC Genes</a></td>
<td>29 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRegDnaseClusteredV2" target=_blank>ENCODE Regulation: DNase Clusters (update to 125 cell types using ENCODE uniform peaks from Jan 2011 freeze)</a></td>
<td>25 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=placentalChainNet" target=_blank>White rhinoceros Chain/Net</a></td>
<td>24 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeAwgDnaseUniform" target=_blank>DNaseI Hypersensitivity Uniform Peaks from ENCODE/Analysis</a></td>
<td>22 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=iscaComposite" target=_blank>ISCA</a></td>
<td>18 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=coriellDelDup" target=_blank>Coriell Cell Line Copy Number Variants</a></td>
<td>17 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV14" target=_blank>GENCODE Version 14 release</a></td>
<td>17 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix CytoScan HD Array track in SNP/CNV Arrays composite track</td>
<td>16 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>14 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=decipher" target=_blank>DECIPHER: Chromosomal Imbalance and Phenotype in Humans</a></td>
<td>11 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=altSeqComposite10" target=_blank>GRCh37 patch10 alternate sequences, haplotypes and reference sequence patches </a></td>
<td>10 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=vertebrateChainNet" target=_blank>Lamprey Chain/Net</a></td>
<td>08 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=snp137" target=_blank>Common SNPs(137), Flagged SNPs(137), Mult. SNPs(137), All SNPS(137)</a></td>
<td>07 Dec 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=placentalChainNet" target=_blank>Tenrec Chain/Net</a></td>
<td>06 Dec 2012</td>
</tr>
<TR valign=top><TD align=left>
Rhesus Chain/Net</td>
<td>03 Dec 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=tgpPhase1Accessibility" target=_blank>1000 Genomes Phase 1 Paired-end Accessible Regions</a></td>
<td>30 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=tgpPhase1" target=_blank>1000 Genomes Phase 1 Integrated Variant Calls: SNVs, InDels, SVs</a></td>
<td>30 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>28 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlLongRnaSeq" target=_blank>Long RNA-seq from ENCODE/Cold Spring Harbor Lab (Release 3)</a></td>
<td>26 Sep 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromFaire" target=_blank>Open Chromatin by FAIRE from ENCODE/OpenChrom(UNC Chapel Hill) (Release 2)</a></td>
<td>19 Sep 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUncBsuProtGenc" target=_blank>Proteogenomics Hg19 and GENCODE Mapping from ENCODE/Univ. North Carolina/Boise State Univ.</a></td>
<td>11 Sep 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/HAIB (Release 3)</a></td>
<td>10 Sep 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromDnase" target=_blank>Open Chromatin by DNaseI HS from ENCODE/OpenChrom(Duke University) (Release 3)</a></td>
<td>28 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromChip" target=_blank>Open Chromatin TFBS by ChIP-seq from ENCODE/Open Chrom(UT Austin) (Release 2)</a></td>
<td>27 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>24 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/Stanford/Yale/USC/Harvard (Release 3)</a></td>
<td>20 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
Medium ground finch Chain/Net</td>
<td>17 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisRnaPet" target=_blank>RNA Sub-cellular Localization by Paired-end diTag Sequencing from ENCODE/GIS (Release 2)</a></td>
<td>14 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCaltechRnaSeq" target=_blank>RNA-seq from ENCODE/Caltech (Release 4)</a></td>
<td>08 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeBroadHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/Broad Institute (Release 3)</a></td>
<td>06 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDgf" target=_blank>DNaseI Digital Genomic Footprinting from ENCODE/University of Washington (Release 4)</a></td>
<td>06 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=altSeqComposite9" target=_blank>GRCh37 patch9 alternate sequences, haplotypes and reference sequence patches </a></td>
<td>01 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwAffyExonArray" target=_blank>Affymetrix Exon Array from ENCODE/University of Washington (Release 4)</a></td>
<td>31 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeFsuRepliChip" target=_blank>Replication Timing by Repli-chip from ENCODE/FSU (Release 2)</a></td>
<td>31 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibMethylRrbs" target=_blank>DNA Methylation by Reduced Representation Bisulfite Seq from ENCODE/HudsonAlpha (Release 3)</a></td>
<td>30 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRikenCage" target=_blank>RNA Subcellular CAGE Localization from ENCODE/RIKEN (Release 4)</a></td>
<td>30 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/University of Washington (Release 5)</a></td>
<td>24 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase" target=_blank>DNaseI Hypersensitivity by Digital DNaseI from ENCODE/University of Washington (Release 6)</a></td>
<td>23 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUncBsuProt" target=_blank>Proteogenomics Hg19 Mapping from ENCODE/Univ. North Carolina/Boise State Univ. (Release 2)</a></td>
<td>18 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwAffyExonArray" target=_blank>Affymetrix Exon Array from ENCODE/University of Washington (Release 3)</a></td>
<td>12 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRegTxn" target=_blank>Transcription Levels Assayed by RNA-seq on 9 Cell Lines from ENCODE</a></td>
<td>10 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlShortRnaSeq" target=_blank>Small RNA-seq from ENCODE/Cold Spring Harbor Lab (Release 3)</a></td>
<td>09 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUmassDekker5C" target=_blank>Chromatin Interactions by 5C from ENCODE/Dekker Univ. Mass. (Release 2)</a></td>
<td>09 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>27 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=hgIkmc" target=_blank>IKMC Genes Mapped (update)</a></td>
<td>22 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlLongRnaSeq" target=_blank>Long RNA-seq from ENCODE/Cold Spring Harbor Lab (Release 2)</a></td>
<td>21 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV12" target=_blank>Gene Annotations from ENCODE/GENCODE Version 12</a></td>
<td>21 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/Stanford/Yale/USC/Harvard (Release 3)</a></td>
<td>15 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase" target=_blank>DNaseI Hypersensitivity by Digital DNaseI from ENCODE/University of Washington (Release 5)</a></td>
<td>12 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=chainNetBosTau7" target=_blank>Cow Chain/Net</a></td>
<td>11 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
ISCA update</td>
<td>05 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisChiaPet" target=_blank>Chromatin Interaction Analysis Paired-End Tags (ChIA-PET) from ENCODE/GIS-Ruan</a></td>
<td>24 May 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/Stanford/Yale/USC/Harvard (Release 2)</a></td>
<td>14 May 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=decodeRmap" target=_blank>deCODE Recombination maps, 10Kb bin size, October 2010
</a></td>
<td>09 May 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRikenCage" target=_blank>RNA Subcellular CAGE Localization from ENCODE/RIKEN (Release 3)</a></td>
<td>02 May 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>01 May 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCaltechRnaSeq" target=_blank>RNA-seq from ENCODE/Caltech (Release 3)</a></td>
<td>19 Apr 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeFsuRepliChip" target=_blank>Replication Timing by Repli-chip from ENCODE/FSU</a></td>
<td>18 Apr 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=grcIncidentDb" target=_blank>GRC Incident</a></td>
<td>17 Apr 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/University of Washington (Release 4)</a></td>
<td>11 Apr 2012</td>
</tr>
<TR valign=top><TD align=left>
ISCA update</td>
<td>10 Apr 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeDukeAffyExon" target=_blank>Affymetrix Exon Array from ENCODE/Duke (Release 3)</a></td>
<td>04 Apr 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromDnase" target=_blank>Open Chromatin by DNaseI HS from ENCODE/OpenChrom(Duke University) (Release 2)</a></td>
<td>28 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=encodeRegions" target=_blank>ENCODE Pilot (regions lifted from hg18)</a></td>
<td>26 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUmassDekker5C" target=_blank>Chromatin Interactions by 5C from ENCODE/Dekker Univ. Mass.</a></td>
<td>22 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV11" target=_blank>Gene Annotations from ENCODE/GENCODE Version 11</a></td>
<td>08 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=geneReviews" target=_blank>GeneReviews (update)</a></td>
<td>07 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeBroadHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/Broad Institute (Release 2)</a></td>
<td>06 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase" target=_blank>DNaseI Hypersensitivity by Digital DNaseI from ENCODE/University of Washington (Release 4)</a></td>
<td>06 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/HAIB (Release 2)</a></td>
<td>29 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwTfbs" target=_blank>CTCF Binding Sites by ChIP-seq from ENCODE/University of Washington (Release 3)</a></td>
<td>29 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=iscaComposite" target=_blank>ISCA Composite Pathogenic Cumulative subtracks (Pathogenic Gain/Loss)</a></td>
<td>28 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibRnaSeq" target=_blank>RNA-seq from ENCODE/HAIB</a></td>
<td>23 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV10" target=_blank>Gene Annotations from ENCODE/GENCODE Version 10</a></td>
<td>22 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlShortRnaSeq" target=_blank>Small RNA-seq from ENCODE/Cold Spring Harbor Lab (Release 2)</a></td>
<td>22 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/University of Washington (Release 3)</a></td>
<td>16 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwRepliSeq" target=_blank>Replication Timing by Repli-seq from ENCODE/University of Washington</a></td>
<td>14 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDgf" target=_blank>DNaseI Digital Genomic Footprinting from ENCODE/University of Washington (Release 3)</a></td>
<td>13 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromSynth" target=_blank>DNaseI/FAIRE/ChIP Synthesis from ENCODE/OpenChrom(Duke/UNC/UTA) (Release 2)</a></td>
<td>10 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update; hg19 now uses dbSNP build 135 mappings)</a></td>
<td>07 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=knownGene" target=_blank>UCSC Genes (data update: adding data from Rfam and tRNAs)</a></td>
<td>02 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibGenotype" target=_blank>Genotype (CNV and SNP) by Illumina 1MDuo and CBS from ENCODE/HudsonAlpha (Release 2)</a></td>
<td>20 Jan 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCaltechRnaSeq" target=_blank>RNA-seq from ENCODE/Caltech (Release 2)</a></td>
<td>17 Jan 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwTfbs" target=_blank>CTCF Binding Sites by ChIP-seq from ENCODE/University of Washington (Release 2)</a></td>
<td>17 Jan 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwAffyExonArray" target=_blank>Affymetrix Exon Array from ENCODE/University of Washington (Release 2)</a></td>
<td>04 Jan 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRikenCage" target=_blank>RNA Subcellular CAGE Localization from ENCODE/RIKEN</a></td>
<td>22 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
SIB Genes</td>
<td>20 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=ensGene" target=_blank>Ensembl Genes v65</a></td>
<td>20 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=snp135" target=_blank>Common SNPs(135), Flagged SNPs(135), Mult. SNPs(135), All SNPs(135)</a></td>
<td>16 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>12 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibMethyl450" target=_blank>CpG Methylation by Methyl 450K  Bead Arrays from ENCODE/HAIB</a></td>
<td>01 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=iscaComposite" target=_blank>International Standards for Cytogenomic Arrays Consortium (ISCA)</a></td>
<td>29 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=chainNomLeu1" target=_blank>Gibbon Chain/Net</a></td>
<td>28 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=chainGorGor3" target=_blank>Gorilla Chain/Net</a></td>
<td>28 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=lincRNAs" target=_blank>lincRNA RNA-Seq reads</a></td>
<td>28 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=geneReviews" target=_blank>GeneReviews</a></td>
<td>21 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=agilentCgh" target=_blank>Agilent Cancer 4x180k Arrays</a></td>
<td>18 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/University of Washington</a></td>
<td>17 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=chainNetXenTro3" target=_blank>X. tropicalis Chain/Net</a></td>
<td>15 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=chainNetMelGal1" target=_blank>Turkey Chain/Net</a></td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=chainNetBosTau6" target=_blank>Cow Chain/Net</a></td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/Stanford/Yale/USC/Harvard</a></td>
<td>29 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeMapability" target=_blank>ENCODE Mapability (hg19 release 3)</a></td>
<td>28 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=cosmic" target=_blank>COSMIC: Catalogue Of Somatic Mutations In Cancer</a></td>
<td>28 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase" target=_blank>DNaseI Hypersensitivity by Digital DNaseI from ENCODE/University of Washington (Release 3)</a></td>
<td>19 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisRnaPet" target=_blank>RNA Subcellular Localization by Paired End diTag Sequencing from GIS/ENCODE</a></td>
<td>17 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlLongRnaSeq" target=_blank>Long RNA-seq from ENCODE/Cold Spring Harbor Lab</a></td>
<td>17 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=hiSeqDepth" target=_blank>Hi Seq Depth</a></td>
<td>14 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeDukeAffyExon" target=_blank>Affymetrix Exon Array from ENCODE/Duke (Release 2)</a></td>
<td>13 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUw5C" target=_blank>Chromatin Interactions by 5C from ENCODE/University of Washington</a></td>
<td>10 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/SYDH (Release 2)</a></td>
<td>07 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSunyAlbanyGeneSt" target=_blank>RNA Binding Protein Associated RNA by RIP-chip GeneST from ENCODE/SUNY Albany (Release 2 - fixed p & q values and scores)</a></td>
<td>04 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibMethylRrbs" target=_blank>DNA Methylation by Reduced Representation Bisulfite Seq from ENCODE/HudsonAlpha (Release 2)</a></td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUncBsuProt" target=_blank>Proteogenomics from ENCODE/Univ. North Carolina/Boise State Univ.</a></td>
<td>29 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>28 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=hgIkmc" target=_blank>IKMC Genes Mapped (update)</a></td>
<td>09 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhRnaSeq" target=_blank>RNA-seq from ENCODE/Stanford/Yale/USC/Harvard</a></td>
<td>08 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=decipher" target=_blank>DECIPHER: Chromosomal Imbalance and Phenotype in Humans (update)</a></td>
<td>31 Aug 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?g=wgEncodeSunyRipSeq" target=_blank>RIP-seq from ENCODE/SUNY Albany</a></td>
<td>29 Aug 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=altSeqComposite" target=_blank>GRCh37 patch5 alternate sequences, haplotypes and reference sequence patches </a></td>
<td>26 Aug 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibGenotype" target=_blank>Genotype (CNV and SNP) by Illumina 1MDuo and CBS from ENCODE/HudsonAlpha </a></td>
<td>16 Aug 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSunySwitchgear" target=_blank>RNA Binding Protein Associated RNA by SwitchGear from ENCODE/SUNY Albany</a></td>
<td>15 Aug 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=numtSeq" target=_blank>NumtS mitochondrial sequence</a></td>
<td>05 Aug 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhNsome" target=_blank>Nucleosome Position by MNase-seq from ENCODE/Stanford/BYU</a></td>
<td>29 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/HAIB</a></td>
<td>18 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
Lizard Chain/Net</td>
<td>14 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV7" target=_blank>Gene Annotations from ENCODE/GENCODE Version 7</a></td>
<td>11 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeBroadHmm" target=_blank>Chromatin State Segmentation by HMM from ENCODE/Broad</a></td>
<td>24 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeDukeAffyExon" target=_blank>Affymetrix Exon Array from ENCODE/Duke</a></td>
<td>14 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>13 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromSynth" target=_blank>DNaseI/FAIRE/ChIP Synthesis from ENCODE/OpenChrom(Duke/UNC/UTA)</a></td>
<td>10 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUchicagoTfbs" target=_blank>Transcription Factor Binding Sites by Epitope-Tag ChIP-seq from ENCODE/University of Chicago </a></td>
<td>03 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeReg" target=_blank>Integrated Regulation from ENCODE Tracks (Transcription, Layered H3K4Me1, Layered H3K4Me3, Layered H3K27Ac, DNase Clusters, Txn Factor ChIP)</a></td>
<td>19 May 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/Stanford/Yale/Davis/Harvard</a></td>
<td>19 May 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeBuOrchid" target=_blank>ORChID Predicted DNA Cleavage Sites from ENCODE/Boston Univ (Tullius Lab)</a></td>
<td>18 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase" target=_blank>DNaseI Hypersensitivity by Digital DNaseI from ENCODE/University of Washington</a></td>
<td>04 May 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=acembly" target=_blank>AceView Genes</a></td>
<td>04 May 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSunyAlbanyGeneSt" target=_blank>RNA Binding Protein Associated RNA by RIP-chip GeneST from ENCODE/SUNY Albany</a></td>
<td>28 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromDnase" target=_blank>Open Chromatin by DNaseI HS from ENCODE/Duke University</a></td>
<td>19 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
Sheep Chain/Net</td>
<td>19 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
Common SNPs(132), Flagged SNPs(132), Mult. SNPs(132), All SNPs(132)</td>
<td>11 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=affyExonArray" target=_blank>hg19 Version 2: featuring probe/probeset stratification and multiple views 
hg19 Version 1 (BED files dated Nov 4 2009)</a></td>
<td>08 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDgf" target=_blank>DNaseI Digital Genomic Footprinting from ENCODE/University of Washington (Release 2)</a></td>
<td>07 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>01 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwAffyExonArray" target=_blank>Affymetrix Exon Array from ENCODE/University of Washington</a></td>
<td>01 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRikenCage" target=_blank>RNA CAGE Subcellular Localization from ENCODE/RIKEN</a></td>
<td>23 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromFaire" target=_blank>Open Chromatin by FAIRE-seq from ENCODE/UNC</a></td>
<td>21 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=hgIkmc" target=_blank>IKMC Genes Mapped (update)</a></td>
<td>21 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwTfbs" target=_blank>CTCF Binding Sites by ChIP-seq from ENCODE/University of Washington</a></td>
<td>17 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromChip" target=_blank>Open Chromatin TFBS by ChIP-seq from ENCODE/Open Chrom(UT Austin)</a></td>
<td>17 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
DGV - data update (v10)</td>
<td>04 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=decipher" target=_blank>DECIPHER: Chromosomal Imbalance and Phenotype in Humans</a></td>
<td>22 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="./../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSunyAlbanyTiling" target=_blank>RNA Binding Protein Associated RNA by Tiling Array from ENCODE/SUNY Albany</a></td>
<td>14 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/University of Washington</a></td>
<td>11 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeBroadHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/Broad Institute</a></td>
<td>10 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?c=chr21&g=wgEncodeCaltechRnaSeq" target=_blank>ENCODE Caltech (Wold Lab) RNA-seq</a></td>
<td>08 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=agilentCgh" target=_blank>Agilent CGH</a></td>
<td>04 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=chainNetDanRer7" target=_blank>Zebrafish Chain/Net</a></td>
<td>03 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=hapmapSnps" target=_blank>HapMap SNPs (lift from hg18 to hg19)</a></td>
<td>01 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?g=uMassBrainHistone" target=_blank>UMass Med School Brain Histone H3K4me3 ChIP-seq</a></td>
<td>26 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=snp131" target=_blank>Simple Nucleotide Polymorphisms (dbSNP build 131): corrected missing functional annotation value</a></td>
<td>25 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=polya" target=_blank>Poly(A) Sites, hg18 liftover</a></td>
<td>21 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=ucsfBrainMethyl" target=_blank>UCSF Brain DNA Methylation (lift from hg18)</a></td>
<td>21 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=gad" target=_blank>Genetic Association Studies of Complex Diseases and Disorders (lift from hg18)</a></td>
<td>21 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>21 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=HInvGeneMrna" target=_blank>H-Invitational Genes mRNA Alignments (lifted from hg18)</a></td>
<td>19 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=fosEndPairs" target=_blank>Fosmid End Pairs (lift from hg18)</a></td>
<td>19 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=illuminaProbes" target=_blank>Alignments of Illumina WG-6 3.0 Probe set (lifted from hg18)</a></td>
<td>18 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=jaxQtlMapped" target=_blank>MGI Mouse QTL (lifted from hg18)</a></td>
<td>18 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=rgdRatQtl" target=_blank>RGD Rat QTL (lift from hg18)</a></td>
<td>18 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=switchDbTss" target=_blank>SwitchGear Genomics Transcription Start Sites (lifted from hg17)</a></td>
<td>14 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=consIndelsHgMmCanFam" target=_blank>Cons Indels MmCf (lifted from hg18)</a></td>
<td>14 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=snpArray" target=_blank>SNP Arrays (lift from hg18)</a></td>
<td>14 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=sibTxGraph" target=_blank>Alternative Splicing Graph from Swiss Institute of Bioinformatics (lift from hg18)</a></td>
<td>14 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=eioJcviNAS" target=_blank>Eur. Inst. Oncology/J. C. Venter Inst. Nuclease Accessible Sites (lifted from hg18)</a></td>
<td>13 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisDnaPet" target=_blank>ENCODE GIS DNA PET (update: added fastq files)</a></td>
<td>12 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=ntMito" target=_blank>Neandertal Mitochondrial Sequence (Vi33.16, 2008)(lifted from hg18)</a></td>
<td>12 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=bamSLFeld1" target=_blank>Neandertal Sequence Reads </a></td>
<td>12 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=ntOoaHaplo" target=_blank>Candidate Regions for Gene Flow from Neandertal to Non-African Modern Humans (lifted from hg18)</a></td>
<td>12 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=ntSssSnps" target=_blank>SNPS Used for Selective Sweep Scan (S) (lifted from hg18)</a></td>
<td>12 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=ntSssTop5p" target=_blank>Selective Sweep Scan (S): 5% Smallest S scores (lifted from hg18)</a></td>
<td>12 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=ntSssZScorePMVar" target=_blank>Selective Sweep Scan (S) on Neandertal vs. Human Polymorphisms (Z-Score +- Variance) (lifted from hg18)</a></td>
<td>12 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=ntHumChimpCodingDiff" target=_blank>Neandertal Alleles in Human/Chimp Coding Non-synonymous Differences in Human Lineage (lifted from hg18)</a></td>
<td>12 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=laminB1Super" target=_blank>NKI Nuclear Lamina Associated Domains (LaminB1 DamID) (lift from hg18)</a></td>
<td>11 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=altSeqComposite" target=_blank>GRCh37 patch2 alternate sequences, haplotypes and reference sequence patches </a></td>
<td>11 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin//hgTrackUi?g=oreganno&db=hg19" target=_blank>ORegAnno (lifted from hg18)</a></td>
<td>06 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=fishClones" target=_blank>FISH Clones (lifted from hg18)</a></td>
<td>05 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin//hgTrackUi?g=recombRate&db=hg19" target=_blank>Recombination Rate</a></td>
<td>05 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=rgdQtl" target=_blank>RGD Human QTL (lifted from hg18)</a></td>
<td>27 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=targetScanS" target=_blank>TS miRNA sites</a></td>
<td>17 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase" target=_blank>ENCODE UW DNaseI HS</a></td>
<td>16 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibMethylRrbs" target=_blank>ENCODE HAIB Methyl RRBS</a></td>
<td>15 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
Add links to LS-SNP for hg19</td>
<td>15 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=darned" target=_blank>Human RNA Editing (from the DAtabase of RNa EDiting)</a></td>
<td>15 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDgf" target=_blank>ENCODE UW DNaseI DGF</a></td>
<td>14 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=evofold" target=_blank>EvoFold Predictions of RNA Secondary Structure</a></td>
<td>14 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=vistaEnhancers" target=_blank>Vista Enhancers</a></td>
<td>13 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=affyGnf1h" target=_blank>Alignments of Affymetrix Consensus/Exemplars from GNF1H</a></td>
<td>13 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=affyU95" target=_blank>Affymetrix Consensus/Exemplars from HG-U95</a></td>
<td>08 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=affyU133Plus2" target=_blank>Affymetrix Consensus/Exemplars from HG-U133 Plus 2.0</a></td>
<td>08 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hgs19&g=wgEncodeAffyRnaChip" target=_blank>ENCODE Affymetrix/CSHL Subcellular RNA Localization by Tiling Array</a></td>
<td>03 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=microsat" target=_blank>Microsatellites</a></td>
<td>22 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=hg19ContigDiff" target=_blank>Contigs new to GRCh37/(hg19), not carried forward from NCBI build 36(hg18) plus the GRC Map Contigs track</a></td>
<td>22 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlShortRnaSeq" target=_blank>ENCODE CSHL Sm RNA-seq</a></td>
<td>17 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisRnaSeq" target=_blank>ENCODE Genome Institute of Singapore RNA-seq
</a></td>
<td>17 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=snp131CodingDbSnp" target=_blank>Corrected coordinates of NCBI's coding functional annotations used in item details pages in hg18's and hg19's SNPs (130) and in hg19's SNPs (131).  Also added a few missing coding functional annotations to hg19's SNPs (131).</a></td>
<td>15 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=sestanBrainAtlas" target=_blank>Sestan Brain Atlas</a></td>
<td>15 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>09 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=rnaCluster" target=_blank>Gene Bounds</a></td>
<td>09 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=affyU133" target=_blank>Affy U133</a></td>
<td>08 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=hgdpGeo" target=_blank>HGDP Alleles</a></td>
<td>05 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
Vega Annotations, version 39</td>
<td>28 Oct 2010</td>
</tr>
<TR valign=top><TD align=left>
Genscan</td>
<td>26 Oct 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies (data update)</a></td>
<td>12 Oct 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgRna" target=_blank>sno/miRNA track</a></td>
<td>28 Sep 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=hgIkmc" target=_blank>IKMC Genes Mapped</a></td>
<td>23 Sep 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencode" target=_blank>ENCODE Gencode Gene Annotations (V4)</a></td>
<td>15 Sep 2010</td>
</tr>
<TR valign=top><TD align=left>
DGV - data update (v9)</td>
<td>09 Sep 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisDnaPet" target=_blank>ENCODE Genome Institute of Singapore DNA Paired-End Ditags</a></td>
<td>08 Sep 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
BlastTab tables rebuild</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Cat Chain/Net</td>
<td>17 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=snp131" target=_blank>Simple Nucleotide Polymorphisms (dbSNP build 131)</a></td>
<td>04 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies (data update)</a></td>
<td>04 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeMapability" target=_blank>ENCODE Mapability or Uniqueness of Reference Genome</a></td>
<td>03 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Marmoset Chain/Net</td>
<td>03 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Burge RNA-seq</td>
<td>27 May 2010</td>
</tr>
<TR valign=top><TD align=left>
Pig Chain/Net</td>
<td>12 May 2010</td>
</tr>
<TR valign=top><TD align=left>
Rabbit Chain/Net</td>
<td>26 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Panda Chain/Net</td>
<td>16 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies (data update)</a></td>
<td>06 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Elephant Chain and Net</td>
<td>24 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
Sea Hare Chain/Net and liftOver file</td>
<td>15 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Chain/Net and liftOver file</td>
<td>12 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies (data update)</a></td>
<td>03 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgPcr?org=Human&db=hg19" target=_blank>UCSC In-Silico PCR (update: PCR can be run against UCSC Genes for hg19)</a></td>
<td>01 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=genomicSuperDups" target=_blank>Segmental Dups</a></td>
<td>22 Feb 2010</td>
</tr>
<TR valign=top><TD align=left>
GWAS Catalog</td>
<td>18 Feb 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=exoniphy" target=_blank>Exoniphy</a></td>
<td>13 Jan 2010</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish Chain and Net</td>
<td>12 Jan 2010</td>
</tr>
<TR valign=top><TD align=left>
multiz46way spp tree update</td>
<td>11 Jan 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
Added link to The GeneNetwork from the UCSC Genes details page.</td>
<td>03 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=knownAlt" target=_blank>Alt Events</a></td>
<td>01 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=cons46way" target=_blank>Vertebrate Multiz Alignment & Conservation (46 Species)</a></td>
<td>01 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=omim" target=_blank>OMIM Genes</a></td>
<td>01 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=knownGene" target=_blank>Initial release of UCSC Genes</a></td>
<td>01 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=allenBrainAli" target=_blank>Allen Brain Atlas Probes</a></td>
<td>18 Nov 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=gnfAtlas2" target=_blank>GNF Atlas 2</a></td>
<td>10 Nov 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Vertebrate genomes, Chain and Net alignment tracks (secondary release)</a></td>
<td>09 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=snp130" target=_blank>Simple Nucleotide Polymorphisms (dbSNP build 130): dbSNP's provisional mapping of SNP coordinates from hg18 (NCBI human build 36) to hg19 (GRCh37).</a></td>
<td>30 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Vertebrate genomes, Chain and Net alignment tracks (primary release)</a></td>
<td>29 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Chains and Nets for non-primate mammal species vs Human</a></td>
<td>16 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Initial CCDS release on GRCh37</td>
<td>14 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene predictions v55</td>
<td>11 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Primate chains and nets from/to human hg19</a></td>
<td>21 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN gene predictions from Brent Lab at WUSTL </td>
<td>17 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
Chromosome Bands</td>
<td>12 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
STS Markers on Genetic and Radiation Hybrid Maps</td>
<td>22 May 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=ctgPos" target=_blank>Physical Map Contigs</a></td>
<td>21 May 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=bacEndPairs" target=_blank>BAC End Pairs</a></td>
<td>21 May 2009</td>
</tr>
<TR valign=top><TD align=left>
UCSC to Ensembl chrom name conversion and haplotype lift</td>
<td>19 May 2009</td>
</tr>
<TR valign=top><TD align=left>
Chimp Chain/Net/LiftOver (panTro2)</td>
<td>06 May 2009</td>
</tr>
<TR valign=top><TD align=left>
Initial hg19 release (using GRCh37 Genome Reference Consortium Human Reference 37 (GCA_000001405.1)): Assembly, Gap, GC Percent, RefSeq Genes, Non-Human RefSeq, MGC Genes, ORFeome Clones, Human mRNAs, Spliced ESTs, Human ESTs, Non-Human mRNAs, Non-Human ESTs, CpG Islands, RepeatMasker, Interrupted Rpts, Simple Repeats, Self Chain.  Downloads: bigZips, chromosomes, database.</td>
<td>27 Apr 2009</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=hg18></A>Human Mar. 2006 (NCBI36/hg18) (hg18, NCBI Build 36.1)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
DGV - data update (2016-05-15) on hg18, hg19 and hg38</td>
<td>20 Sep 2016</td>
</tr>
<TR valign=top><TD align=left>
DGV - data update (2015-07-23) on hg18, hg19 and hg38</td>
<td>21 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
DGV - data update (2014-10-16) on hg18 and hg19; first release on hg38</td>
<td>23 Feb 2015</td>
</tr>
<TR valign=top><TD align=left>
phastBias gBGC - regions predicted to be influenced by GC-biased gene conversion </td>
<td>08 Nov 2013</td>
</tr>
<TR valign=top><TD align=left>
DGV - data update (v11)</td>
<td>11 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=dgv" target=_blank>Database of Genomic Variants: Structural Variation (CNV, Inversion, In/del) (data update)</a></td>
<td>14 Feb 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>07 Feb 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>14 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>28 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>24 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg18&g=chainNetEchTel1" target=_blank>Tenrec Chain/Net</a></td>
<td>29 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>27 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
ISCA update</td>
<td>05 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=decodeRmap" target=_blank>deCODE Recombination maps, 10Kb bin size, October 2010
</a></td>
<td>09 May 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>01 May 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=geneReviews" target=_blank>GeneReviews (update)</a></td>
<td>07 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update; hg19 now uses dbSNP build 135 mappings)</a></td>
<td>07 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>12 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=iscaComposite" target=_blank>International Standards for Cytogenomic Arrays Consortium (ISCA)</a></td>
<td>29 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=geneReviews" target=_blank>GeneReviews</a></td>
<td>21 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=hiSeqDepth" target=_blank>Hi Seq Depth</a></td>
<td>14 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeUwDnaseSeq" target=_blank>ENCODE Univ. Washington DNaseI Hypersensitivity by Digital DNaseI - Validation data (Release 5)</a></td>
<td>11 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>28 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
EvoFold V2</td>
<td>21 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBroadHmm" target=_blank>ENCODE Chromatin State Segmentation by HMM from Broad Institute, MIT and MGH</a></td>
<td>24 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg18&g=ceuBcellComposite" target=_blank>B-Cell Transcriptome (RNA-Seq) </a></td>
<td>14 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>13 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeYaleChIPseq" target=_blank>ENCODE Transcription Factor Binding Sites by ChIP-seq from Yale/UC-Davis/Harvard</a></td>
<td>03 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaCnv" target=_blank>ENCODE Common Cell CNV (Release 2): colors in Regions view subtracks updated to new standards</a></td>
<td>08 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>01 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeUwDnaseSeq" target=_blank>ENCODE UW DNaseI HS (Release 4): Jurkat (G1 Phase) and HIPEpic replicates have been revoked</a></td>
<td>24 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>21 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=bamSLDenisova" target=_blank>Denisova Sequence Reads and Alignments of Sequence Reads from 7 Modern Humans</a></td>
<td>23 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg18&g=rdmr" target=_blank>Reprogrammed Differentially Methylated Regions (R-DMR)</a></td>
<td>01 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=hg18ContigDiff" target=_blank>Contigs dropped or changed from NCBI build 36(hg18) to GRCh37(hg19)</a></td>
<td>19 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg18&g=NumtSeq" target=_blank>NumtSeq</a></td>
<td>17 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=snp131CodingDbSnp" target=_blank>Corrected coordinates of NCBI's coding functional annotations used in item details pages in hg18's and hg19's SNPs (130) and in hg19's SNPs (131).  Also added a few missing coding functional annotations to hg19's SNPs (131).</a></td>
<td>15 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies, GWAS (data update)</a></td>
<td>09 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=hgdpGeo" target=_blank>HGDP Alleles</a></td>
<td>05 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies (data update)</a></td>
<td>12 Oct 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=darned" target=_blank>Human RNA Editing</a></td>
<td>07 Sep 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=ucsfBrainMethyl" target=_blank>UCSF Brain DNA methylation</a></td>
<td>27 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeReg" target=_blank>ENCODE Regulation</a></td>
<td>12 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaRnaSeq" target=_blank>ENCODE HudsonAlpha RNA-seq</a></td>
<td>30 Jul 2010</td>
</tr>
<TR valign=top><TD align=left>
DECIPHER data</td>
<td>13 Jul 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeYaleRNAseq" target=_blank>ENCODE RNA-seq from Yale/UC-Davis/Harvard</a></td>
<td>09 Jul 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBuOrchid" target=_blank>ENCODE Boston Univ (Tullius Lab) ORChID Predicted DNA Cleavage Sites</a></td>
<td>29 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBroadChipSeq" target=_blank>ENCODE Broad Histone (update to Release 3) - improved display of Signal subtracks when zoomed in to a range spanning less than 16,500 base pairs.</a></td>
<td>16 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaChipSeq" target=_blank>ENCODE HAIB TFBS (release 2)</a></td>
<td>11 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeCaltechRnaSeq" target=_blank>ENCODE Caltech RNA-seq (Release 2)</a></td>
<td>10 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies (data update)</a></td>
<td>04 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeMapability" target=_blank>ENCODE Mapability or Uniqueness of Reference Genome (update: adding CRG Alignability data)</a></td>
<td>03 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeGisPet" target=_blank>ENCODE GIS Subcellular RNA Localization by Paired End diTag Sequencing</a></td>
<td>01 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeGisRnaSeq" target=_blank>ENCODE Genome Institute of Singapore RNA-seq</a></td>
<td>26 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeYaleChIPseq" target=_blank>ENCODE Yale TFBS (release 3)</a></td>
<td>26 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeRikenCage" target=_blank>ENCODE RIKEN RNA Subcellular Localization by CAGE Tags (Release 2)</a></td>
<td>24 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeDukeAffyExonArray" target=_blank>ENCODE Duke Affy All-Exon Arrays</a></td>
<td>24 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeUwDGF" target=_blank>ENCODE Univ. Washington Digital DNase Genomic Footprinting</a></td>
<td>17 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?g=ntModernHumans&db=hg18" target=_blank>Human Seq</a></td>
<td>07 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?g=ntSssSnps&db=hg18" target=_blank>S SNPs</a></td>
<td>07 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=ntOoaHaplo" target=_blank>Cand. Gene Flow</a></td>
<td>07 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?g=ntSeqReads&db=hg18" target=_blank>Neandertal Seq</a></td>
<td>07 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?g=ntSeqContigs&db=hg18" target=_blank>Neandertal Cntgs</a></td>
<td>07 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?g=ntSssTop5p&db=hg18" target=_blank>5% Lowest S</a></td>
<td>07 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=ntSssZScorePMVar" target=_blank>Sel Swp Scan (S)</a></td>
<td>07 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?g=ntMito&db=hg18" target=_blank>Neandertal Mito</a></td>
<td>07 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=ntHumChimpCodingDiff" target=_blank>H-C Coding Diffs</a></td>
<td>07 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeGisDnaPet" target=_blank>ENCODE Genome Institute of Singapore DNA Paired-End Ditags</a></td>
<td>19 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Fixed corrupt table from ENCODE pilot project: Gencode Gene Annotations (March 2007)</td>
<td>09 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies (data update)</a></td>
<td>06 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeChromatinMap" target=_blank>ENCODE Open Chromatin (release 3)</a></td>
<td>31 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBroadChipSeq" target=_blank>ENCODE Histone Modifications by Broad Institute ChIP-seq (release 3)</a></td>
<td>23 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeUwAffyExonArray" target=_blank>ENCODE UW Affy All-Exon Arrays (release 2)</a></td>
<td>18 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeUwDnaseSeq" target=_blank>UW DNaseI HS (release 3)</a></td>
<td>11 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeUwChIPSeq" target=_blank>ENCODE Histone Modifications by Univ. Washington ChIP-seq</a></td>
<td>11 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaMethylSeq" target=_blank>ENCODE HudsonAlpha Methyl-seq</a></td>
<td>05 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=gwasCatalog" target=_blank>NHGRI Catalog of Published Genome-Wide Association Studies (data update)</a></td>
<td>03 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeSangerGencode" target=_blank>ENCODE Gencode Gene Annotations (Version 3c)</a></td>
<td>26 Feb 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeCshlShortRnaSeq" target=_blank>ENCODE Cold Spring Harbor Labs Small RNA-seq</a></td>
<td>26 Feb 2010</td>
</tr>
<TR valign=top><TD align=left>
GWAS Catalog</td>
<td>18 Feb 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeUwDnaseSeq" target=_blank>ENCODE Univ. Washington DNaseI Hypersensitivity by Digital DNaseI (Release 2)</a></td>
<td>03 Feb 2010</td>
</tr>
<TR valign=top><TD align=left>
SNP Array (update: Illumina CytoSNP-12, Illumina Human660W-Quad, & Illumina HumanOmni1-Quad)</td>
<td>13 Jan 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBroadChipSeq" target=_blank>ENCODE Broad Histone Release 2 (release 2)</a></td>
<td>18 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeCaltechRnaSeq" target=_blank>ENCODE Caltech RNA-seq Map Algorithm</a></td>
<td>18 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeCshlLongRnaSeq" target=_blank>ENCODE Cold Spring Harbor Labs Long RNA-seq (Release 2: GM12878, K562 whole cell)</a></td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeChromatinMap" target=_blank>ENCODE Open Chromatin, Duke/UNC/UT (Release 2)</a></td>
<td>04 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
lsSnpPdb update: updates Mappings to PDB Protein Structures on SNPs(130) details page and the LS-SNP links from the UCSC Genes details page.</td>
<td>04 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeRikenCage" target=_blank>ENCODE RIKEN CAGE Loc: updated download files</a></td>
<td>30 Nov 2009</td>
</tr>
<TR valign=top><TD align=left>
DGV - changed color on inversions.  no change in data from 2009-08-12 release.</td>
<td>05 Nov 2009</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter and UCSC Genes otherOrgs links to yeast sacCer2</td>
<td>02 Nov 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=omimGene" target=_blank>OMIM Genes</a></td>
<td>27 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=pgSnp" target=_blank>Genome Variants (Added Korean (SJK) and 1000 Genomes high-coverage pilot individuals (NA12878, NA12891, NA12892, NA19240))</a></td>
<td>15 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeYaleChIPseq" target=_blank>ENCODE Yale TFBS (Update several subtracks; add new subtracks)</a></td>
<td>14 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeAffyRnaChip" target=_blank>ENCODE Affymetrix/CSHL Subcellular RNA Localization by Tiling Array (Update:  GM12878 Nucleolus Total, HepG2 Nucleus PolyA+ and PolyA-, HepG2 Nucleolus Total)</a></td>
<td>02 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
DGV: Updated to version 8.</td>
<td>02 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeNhgriNre" target=_blank>ENCODE NHGRI Elnitski Negative Regulatory Elements</a></td>
<td>24 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Agilent Array (two new subtracks: HRD 1x1m and CNV 2x105k)</td>
<td>17 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
sno/miRNA(wgRna)-data update </td>
<td>15 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeUwAffyExonArray" target=_blank>ENCODE UW Affy All Exon Arrays</a></td>
<td>15 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=snp130" target=_blank>dbSNP build 130: corrected functional annotations from dbSNP (- strand mRNA, swapped near-gene-3 and near-gene-5)</a></td>
<td>14 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaCnv" target=_blank>ENCODE Common Cell-Type Copy Number Variation</a></td>
<td>11 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaMethylSeq" target=_blank>ENCODE HudsonAlpha Methyl-seq</a></td>
<td>11 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaMethyl27" target=_blank>ENCODE HudsonAlpha CpG Methylation by Illumina Methyl27</a></td>
<td>11 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?g=wgEncodeUwDnaseSeq" target=_blank>ENCODE Univ. Washington DNaseI Hypersensitivity by Digital DNaseI</a></td>
<td>11 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaChipseq" target=_blank>ENCODE HudsonAlpha CHIP-seq (Correction to labels and metadata - TAF1)</a></td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeRikenCage" target=_blank>ENCODE RIKEN RNA Subcellular Localization by CAGE Tags: PolyA- in K562 nucleus and cytosol, PolyA- in prostate whole cell</a></td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeCshlLoRnaSeq" target=_blank>ENCODE Cold Spring Harbor Labs Long RNA-seq (K562)</a></td>
<td>11 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 54</td>
<td>07 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHelicosRnaSeq" target=_blank>ENCODE Helicos RNA-seq (K562)</a></td>
<td>28 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeMapability" target=_blank>ENCODE Mapability or Uniqueness of Reference Genome</a></td>
<td>24 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=snp130" target=_blank>Simple Nucleotide Polymorphisms (dbSNP build 130)</a></td>
<td>16 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeAffyRnaChip" target=_blank>ENCODE Affymetrix/CSHL Subcellular RNA Localization by Tiling Array</a></td>
<td>06 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeChromatinMap" target=_blank>ENCODE Open Chromatin, Duke/UNC/UT</a></td>
<td>26 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeRikenCage" target=_blank>ENCODE RIKEN RNA Subcellular Localization by CAGE Tags</a></td>
<td>09 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=uwNucOcc" target=_blank>UW Predicted Nucleosome Occupancy. Fix data truncation on chromosome 3</a></td>
<td>05 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaChipSeq" target=_blank>ENCODE HudsonAlpha Transcription Factor Binding Sites by ChIP-seq</a></td>
<td>05 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
UCSC to Ensembl chrom name conversion and haplotype lift</td>
<td>19 May 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=fox2ClipSeqComp" target=_blank>FOX2 CLIP-seq (new clusters subtrack)</a></td>
<td>12 May 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeSangerGencode" target=_blank>ENCODE Gencode Gene Annotations</a></td>
<td>08 May 2009</td>
</tr>
<TR valign=top><TD align=left>
UCSC Genes: update to some secondary tables</td>
<td>07 May 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=hgdpXpehh" target=_blank>HGDP XP-EHH</a></td>
<td>30 Apr 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=hgdpHzy" target=_blank>HGDP Hetrzygsty</a></td>
<td>30 Apr 2009</td>
</tr>
<TR valign=top><TD align=left>
LiftOver to hg19</td>
<td>27 Apr 2009</td>
</tr>
<TR valign=top><TD align=left>
Vega Genes (build 33 update)</td>
<td>21 Apr 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeSunyalbanyRnaGeneChip" target=_blank>ENCODE SUNY Albany RNA Binding Proteins by RIP-chip</a></td>
<td>08 Apr 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBroadChipSeq" target=_blank>ENCODE Histone Modifications by Broad Institute ChIP-seq (Raw Signal tracks)</a></td>
<td>03 Apr 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBroadChipSeq" target=_blank>ENCODE Histone Modifications by Broad Institute ChIP-seq</a></td>
<td>03 Apr 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=omimGene" target=_blank>OMIM Genes</a></td>
<td>03 Apr 2009</td>
</tr>
<TR valign=top><TD align=left>
DGV: Updated to version 7.</td>
<td>26 Mar 2009</td>
</tr>
<TR valign=top><TD align=left>
affyExonTissues (initial release)</td>
<td>16 Mar 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeYaleChIPseq" target=_blank>ENCODE Transcription Factor Binding Sites by ChIP-seq from Yale/UC-Davis/Harvard</a></td>
<td>13 Mar 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../ENCODE/protocols/cell" target=_blank>ENCODE cell culture protocols</a></td>
<td>09 Mar 2009</td>
</tr>
<TR valign=top><TD align=left>
DGV: Removed extraneous thin bars that implied only that there is overlap with another item.</td>
<td>24 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Illumina's Human1M-Duo Genotyping BeadChip</td>
<td>24 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://ls-snp.icm.jhu.edu/ls-snp-pdb/" target=_blank>Links to LS-SNP/PDB site and links to launch the Chimera 3D protein structure viewer (from Gene details and SNP details pages)</a></td>
<td>20 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>18 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Repeating Elements (by RepeatMasker version 3.2.7) & Fragments of Interrupted Repeats Joined by RepeatMasker ID (RM version 3.2.7)</td>
<td>05 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=cons44way " target=_blank>Vertebrate Multiz Alignment & Conservation (44 Species)</a></td>
<td>29 Jan 2009</td>
</tr>
<TR valign=top><TD align=left>
agilentProbes (formerly agilentCgh; 5 new probesets)</td>
<td>27 Jan 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../ENCODE" target=_blank>ENCODE portal update for production phase</a></td>
<td>23 Jan 2009</td>
</tr>
<TR valign=top><TD align=left>
FOX2 adaptor-trimmed CLIPseq reads and density plots</td>
<td>12 Jan 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=encodeMsaTbaDec07&" target=_blank>ENCODE 36-Way TBA Alignments and Conservation</a></td>
<td>19 Dec 2008</td>
</tr>
<TR valign=top><TD align=left>
Eur. Inst. Oncology/J. C. Venter Inst. Nuclease Accessible Sites</td>
<td>26 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Basewise Conservation by PhyloP for 28-Species Vertebrate Multiz Alignment</td>
<td>26 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Enable isPCR on UCSC Gene set to amplify exons only.</td>
<td>26 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Medaka Chain and Net</td>
<td>19 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
DGV: Updated to version 6.</td>
<td>19 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Alignments of Illumina WG-6 3.0 Probe set</td>
<td>30 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
HGSV Discordant</td>
<td>01 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Removed 12 non-coding genes</td>
<td>23 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Converted MGC track to PSL (to allow for alignment visualization)</td>
<td>22 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Converted ORFeome tracks to PSL (to allow for alignment visualization)</td>
<td>16 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
UCSC Genes (Update of UCSC Genes set. Brings in CCDS coding regions, and also RefSeq, GenBank, SwissProt, and trEMBL data from the past year.)</td>
<td>03 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Dropped "NIMH Bipolar" and "Wellcome Trust Case Control Consortium" per new NIH policy on genome-wide association studies.</td>
<td>29 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update (version 50)</td>
<td>25 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
DGV: Updated to version 5.</td>
<td>21 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
Predicted Nucleosome Occupancy from University of Washington</td>
<td>18 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
UniGene Alignments update</td>
<td>11 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
SNPs (dbSNP build 129)</td>
<td>01 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
NKI Nuclear Lamina tracks</td>
<td>23 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Positively Selected Genes (6 species)</td>
<td>18 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Chains & Nets to Chimp (panTro2) updated with new blastz matrix</td>
<td>15 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Guinea Pig Chain and Net</td>
<td>11 Jun 2008</td>
</tr>
<TR valign=top><TD align=left>
Lamprey Chain/Net/LiftOver (petMar1)</td>
<td>10 Jun 2008</td>
</tr>
<TR valign=top><TD align=left>
Lancelet Chain/Net/LiftOver (braFlo1)</td>
<td>10 Jun 2008</td>
</tr>
<TR valign=top><TD align=left>
C/D and H/ACA Box snoRNAs, scaRNAs, and microRNAs from snoRNABase and miRBase (track update)</td>
<td>05 Jun 2008</td>
</tr>
<TR valign=top><TD align=left>
Update of CCDS annotations</td>
<td>04 Jun 2008</td>
</tr>
<TR valign=top><TD align=left>
DGV</td>
<td>23 May 2008</td>
</tr>
<TR valign=top><TD align=left>
Marmoset Chain/Net/LiftOver (calJac1)</td>
<td>24 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix ENCODE Extension Transcription Sites & Signal</td>
<td>16 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl v49 gene predictions</td>
<td>15 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Chain track, Net track, and liftOver file to bosTau4</td>
<td>08 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
UCSC Genes link to Allen Institute for Brain Science - Gene Expression in the Human Cortex website</td>
<td>20 Mar 2008</td>
</tr>
<TR valign=top><TD align=left>
SNPs (dbSNP build 128)</td>
<td>11 Mar 2008</td>
</tr>
<TR valign=top><TD align=left>
Adds a link to HuGE database from Known Gene details page.</td>
<td>06 Mar 2008</td>
</tr>
<TR valign=top><TD align=left>
HapMap Linkage Disequilibrium - Phase II - from phased genotypes (replaces the HapMap LD Unphased track)</td>
<td>29 Feb 2008</td>
</tr>
<TR valign=top><TD align=left>
Updated with CCDS genes (as of 2008-02-01) </td>
<td>12 Feb 2008</td>
</tr>
<TR valign=top><TD align=left>
Genetic Association Studies of Complex Diseases and Disorders (data update)</td>
<td>16 Jan 2008</td>
</tr>
<TR valign=top><TD align=left>
Link to SeattleSNPs PGA database from UCSC Genes page</td>
<td>08 Jan 2008</td>
</tr>
<TR valign=top><TD align=left>
Link to geneTests database from UCSC Genes page</td>
<td>08 Jan 2008</td>
</tr>
<TR valign=top><TD align=left>
CCDS update (as of 2007-12-12)</td>
<td>20 Dec 2007</td>
</tr>
<TR valign=top><TD align=left>
encodeAffyEcSignal</td>
<td>13 Dec 2007</td>
</tr>
<TR valign=top><TD align=left>
encodeAffyEcSites </td>
<td>13 Dec 2007</td>
</tr>
<TR valign=top><TD align=left>
Affy Tx sRNA Sig (lifted from hg17)</td>
<td>12 Dec 2007</td>
</tr>
<TR valign=top><TD align=left>
Affy Tx sRNA Reg (lifted from hg17)</td>
<td>12 Dec 2007</td>
</tr>
<TR valign=top><TD align=left>
Affy Tx lRNA Sig (lifted from hg17)</td>
<td>12 Dec 2007</td>
</tr>
<TR valign=top><TD align=left>
Affy Tx lRNA Reg (lifted from hg17)</td>
<td>12 Dec 2007</td>
</tr>
<TR valign=top><TD align=left>
HapMap LD Unph.</td>
<td>07 Dec 2007</td>
</tr>
<TR valign=top><TD align=left>
NIMH Bipolar Disease track (and Genome Graphs built-in data)</td>
<td>20 Nov 2007</td>
</tr>
<TR valign=top><TD align=left>
Case Control Consortium track (and Genome Graphs built-in data)</td>
<td>20 Nov 2007</td>
</tr>
<TR valign=top><TD align=left>
Uppsala ChIP</td>
<td>01 Nov 2007</td>
</tr>
<TR valign=top><TD align=left>
TargetScan miRNA Regulatory Sites</td>
<td>01 Nov 2007</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish Chain/Net/LiftOver (danRer5)</td>
<td>29 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Vista HMR-Conserved Non-coding Human Enhancers from LBNL updated.</td>
<td>26 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
lift rnaGene from hg17</td>
<td>24 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI DIPs (lift from hg17)</td>
<td>23 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Updated mouse column in Gene Sorter and in Orthologous Genes section of gene details pages to mm9 UCSC Genes</td>
<td>19 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
SNP Array Affy 6 (data update)</td>
<td>17 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Added CONTRAST gene predictions</td>
<td>17 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Whole-Genome ENCODE LI/UCSD TAF1 Super Track</td>
<td>17 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN PASA-EST subtrack (corrected naming scheme)</td>
<td>16 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Human Quantitative Trait Loci from RGD</td>
<td>15 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
wgRna (new miRnas added)</td>
<td>15 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
SGP Gene Predictions Using Mouse/Human Homology (data update)</td>
<td>15 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Gencode RACEfrags (strand information added)</td>
<td>12 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
BU OrChID</td>
<td>12 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE data migration from hg17</td>
<td>08 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
GIS PET RNA and GIS ChIP-PET tracks: lifted from hg17 and moved out of Encode to group Expression and Regulation. 
Added 3 new subtracks:
wgEncodeGisChipPetHes3H3K27me3 (belongs to wgEncodeGisChipPetAll)
wgEncodeGisChipPetHes3H3K4me3 (belongs to wgEncodeGisChipPetAll)
wgEncodeGisRnaPetHes3 (belongs to wgEncodeGisRnaPet)</td>
<td>28 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Intron Size column for Gene Sorter</td>
<td>21 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
CDS Score column for Gene Sorter</td>
<td>17 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Gene Category column for Gene Sorter</td>
<td>17 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Adding Coding SNP column to gene sorter</td>
<td>13 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
update to 7X Reg Potential to fix missing data for chroms 3,4,9,10,20,21</td>
<td>07 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
FISH Clones (data update)</td>
<td>06 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Initial dm3 Regulatory elements from Open Regulatory Annotation (ORegAnno) release; updated hg18 ORegAnno track</td>
<td>06 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
AceView Genes (replacing the initial data which was liftOver'd from hg17)</td>
<td>05 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Gene Boundaries as Defined by RNA and Spliced EST Clusters</td>
<td>30 Aug 2007</td>
</tr>
<TR valign=top><TD align=left>
Mouse (mm9) Chain and Net tracks and liftOver file</td>
<td>28 Aug 2007</td>
</tr>
<TR valign=top><TD align=left>
Updated Drosophila column in Gene Sorter from dm2 to dm3 FlyBase genes</td>
<td>28 Aug 2007</td>
</tr>
<TR valign=top><TD align=left>
SNP Arrays</td>
<td>02 Aug 2007</td>
</tr>
<TR valign=top><TD align=left>
HMR Conserved Transcription Factor Binding Sites (data update)</td>
<td>20 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
Augustus</td>
<td>09 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
HapMap SNPs</td>
<td>09 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN composite track: ab-inito, PASA cDNA clusters</td>
<td>02 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
SwitchGear Genomics Transcription Start Sites</td>
<td>27 Jun 2007</td>
</tr>
<TR valign=top><TD align=left>
Most Conserved</td>
<td>25 Jun 2007</td>
</tr>
<TR valign=top><TD align=left>
Conservation (28 vertebrate alignment)</td>
<td>25 Jun 2007</td>
</tr>
<TR valign=top><TD align=left>
sno/miRNA(wgRna)-data update</td>
<td>31 May 2007</td>
</tr>
<TR valign=top><TD align=left>
Structural Variation (9 subtracks)</td>
<td>08 May 2007</td>
</tr>
<TR valign=top><TD align=left>
Eponine Predicted Transcription Start Sites</td>
<td>03 May 2007</td>
</tr>
<TR valign=top><TD align=left>
Clones from genbank ORFeome collaboration.</td>
<td>02 May 2007</td>
</tr>
<TR valign=top><TD align=left>
Exapted Repeats (Conserved non-exonic regions that were deposited by mobile elements)</td>
<td>30 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
ACEScan Alternate exon finder</td>
<td>12 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Chains Nets and liftOvers to fr2</td>
<td>11 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Update of hg18 xxBlastTab tables to coincide with UCSC Genes</td>
<td>06 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Alt Events</td>
<td>06 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Old Known Genes</td>
<td>06 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
UCSC Genes</td>
<td>06 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Added initial hg18 CCDS release, updated hg17 CCDS</td>
<td>12 Mar 2007</td>
</tr>
<TR valign=top><TD align=left>
sno/miRNA(wgRna)-data update </td>
<td>13 Feb 2007</td>
</tr>
<TR valign=top><TD align=left>
Added NCBI Clone Validation information to details of MGC genes</td>
<td>12 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
Exoniphy</td>
<td>13 Dec 2006</td>
</tr>
<TR valign=top><TD align=left>
AceView Genes (lifted from hg17)</td>
<td>08 Dec 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated HMR Conserved Transcription Factor Binding Sites</td>
<td>20 Nov 2006</td>
</tr>
<TR valign=top><TD align=left>
Coverage track reworked to be more accurate</td>
<td>20 Nov 2006</td>
</tr>
<TR valign=top><TD align=left>
Yale Transcriptionally Active Regions (TARs) (Bertone data)</td>
<td>07 Nov 2006</td>
</tr>
<TR valign=top><TD align=left>
HMR Conserved Transcription Factor Binding Sites</td>
<td>03 Nov 2006</td>
</tr>
<TR valign=top><TD align=left>
AffyHuEx1.0 (Probe Set locations for Affymetrix Human Exon Genechip)</td>
<td>24 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
SNPs (new function data from dbSNP)</td>
<td>09 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
Update to UCSC Known Genes color coding scheme</td>
<td>03 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
poly(A) DB</td>
<td>03 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
Protein to Protein Interaction Network by Erich Wanker (Gene Sorter column)</td>
<td>18 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
Protein to Protein Interaction Network by Marc Vidal (Gene Sorter column)</td>
<td>18 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
Exon Primer links in Known Genes</td>
<td>11 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
Self Net downloads</td>
<td>02 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
Segmental Dups</td>
<td>01 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
Update to BAC End Pairs track, recovering previously filtered BAC Ends</td>
<td>28 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
Rat Ortholog column of Gene Sorter and Known Gene details updated to rn4</td>
<td>20 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
SNP build 126</td>
<td>30 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
EvoFold (changed documentation on description page)</td>
<td>29 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Superfamily track</td>
<td>27 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Human Chained Self Alignments</td>
<td>27 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
ESPERR Regulatory Potential (7 species)</td>
<td>22 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
EvoFold (lifted from hg17, weak predictions filtered out)</td>
<td>21 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated frame annotation for Conservation track to fix error with spliced codons
</td>
<td>15 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Known Genes: added link to Ensembl Transcripts</td>
<td>05 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins </td>
<td>31 May 2006</td>
</tr>
<TR valign=top><TD align=left>
wgRna -data update May 2006</td>
<td>24 May 2006</td>
</tr>
<TR valign=top><TD align=left>
First Exon Finder</td>
<td>18 May 2006</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>11 May 2006</td>
</tr>
<TR valign=top><TD align=left>
H-Inv DB link from Known Gene page</td>
<td>09 May 2006</td>
</tr>
<TR valign=top><TD align=left>
Initial hg18 release (using NCBI Build 36.1): Chromosome Band; STS Markers; FISH Clones; Recomb Rate; Map Contigs; Assembly; Gap; Coverage; BAC End Pairs; Fosmid End Pairs; GC Percent; Known Genes; Proteome Browser; Gene Sorter; Human and NonHuman RefSeq Genes; MGC Genes; N-SCAN Genes; Genscan Genes; sno/miRNA, Human and NonHuman mRNAs; Spliced ESTs; Human and NonHuman ESTs; H-Inv Genes; Repeat Masker; Simple Repeats; Allen Brain; GNF Atlas2; GNF Ratio; CpG Islands; Affy: U133, U133Plus2, U95, GNF1H.

Conservation; Most Conserved; Chains & Nets: Fugu, Tetraodon, Zebrafish, X. tropicalis, Chicken, Cow, Dog, Rat, Mouse, Rhesus, Chimp; RepeatMasker;  Simple Repeats.

Downloads: bigZips, chromosomes, database, liftOvers, multiz, phastCons.</td>
<td>14 Apr 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=hg17></A>Human May 2004 (NCBI35/hg17) (hg17, NCBI Build 35)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?org=human&g=dgv" target=_blank>Database of Genomic Variants: Structural Variation (CNV, Inversion, In/del) (data update)</a></td>
<td>14 Feb 2013</td>
</tr>
<TR valign=top><TD align=left>
liftOver to hg19</td>
<td>28 Apr 2009</td>
</tr>
<TR valign=top><TD align=left>
Enable isPCR on UCSC Gene set to amplify exons only.</td>
<td>26 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
HGSV Validated</td>
<td>22 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
HGSV Discordant</td>
<td>01 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Converted MGC track to PSL (to allow for alignment visualization)</td>
<td>22 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Converted ORFeome tracks to PSL (to allow for alignment visualization)</td>
<td>16 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Dropped "NIMH Bipolar" and "Wellcome Trust Case Control Consortium" per new NIH policy on genome-wide association studies.</td>
<td>29 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
Updated with CCDS genes (as of 2008-02-01) </td>
<td>12 Feb 2008</td>
</tr>
<TR valign=top><TD align=left>
HapMap SNPs.  Data correction of chimp orthologous alleles.</td>
<td>06 Feb 2008</td>
</tr>
<TR valign=top><TD align=left>
CCDS update (as of 2007-12-12)</td>
<td>17 Dec 2007</td>
</tr>
<TR valign=top><TD align=left>
 Affy Tx sRNA Sig ( renamed tables and files to fix the missing track )</td>
<td>11 Dec 2007</td>
</tr>
<TR valign=top><TD align=left>
Vista HMR-Conserved Non-coding Human Enhancers from LBNL updated.</td>
<td>26 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE TBA SCONE Conservation (Update)</td>
<td>19 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
update to 7X Reg Potential to fix missing data for chroms 4,8,9</td>
<td>07 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UC Davis ChIP-chip NimbleGen (E2F1, c-Myc, TAF, POLII)</td>
<td>24 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sites - Jun07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sig - Jun07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Pval - Jun07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sites - Apr07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sig - Apr07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Pval - Apr07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sites - Update to Jan07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sig - Jan07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Pval - Jan07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sites Update to Nov06 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sig - Nov06 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCOE Yale ChIP-chip Pval - Nov06 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
HapMap SNPs. Data correction.</td>
<td>25 Jun 2007</td>
</tr>
<TR valign=top><TD align=left>
Affy Tx sRNA Sig</td>
<td>19 Jun 2007</td>
</tr>
<TR valign=top><TD align=left>
Affy Tx lRNA Sig</td>
<td>19 Jun 2007</td>
</tr>
<TR valign=top><TD align=left>
Affy Tx sRNA Reg</td>
<td>19 Jun 2007</td>
</tr>
<TR valign=top><TD align=left>
Affy Tx lRNA Reg</td>
<td>19 Jun 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE University of Virginia DNA Replication Origins (data update)</td>
<td>30 May 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Duke Dnase/chip - Feb07 batch (data addition)</td>
<td>21 May 2007</td>
</tr>
<TR valign=top><TD align=left>
HapMap SNPs</td>
<td>19 May 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gencode RACEfrags</td>
<td>18 May 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gencode Gene Annotations (March 2007)</td>
<td>04 May 2007</td>
</tr>
<TR valign=top><TD align=left>
Structural Variation: new subtrack for Redon data</td>
<td>02 May 2007</td>
</tr>
<TR valign=top><TD align=left>
Structural Variation: data correction for Conrad subtrack</td>
<td>02 May 2007</td>
</tr>
<TR valign=top><TD align=left>
Clones from genbank ORFeome collaboration.</td>
<td>02 May 2007</td>
</tr>
<TR valign=top><TD align=left>
Locus Variants update (new LSDBs added, old ones updated)</td>
<td>27 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Update 3 subtracks with missing data</td>
<td>17 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UCD Ng ChIP Hits (update of cMyc and E2F1 data)</td>
<td>17 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Vista HMR-Conserved Non-coding Human Enhancers from LBNL</td>
<td>12 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Augustus Gene Predictions (two subtracks: Hints, and de novo).  This composite track replaces the Augustus ab initio track.</td>
<td>12 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Added initial hg18 CCDS release, updated hg17 CCDS</td>
<td>12 Mar 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale RFBR (Regulatory Factor Binding Regions)</td>
<td>21 Feb 2007</td>
</tr>
<TR valign=top><TD align=left>
Locus Variants (Jan 2007 snapshot)</td>
<td>20 Feb 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: UCSC EvoFold (Encode predictions analyzed in Washietl, Pedersen, et al., 2007).  Replaces TBA23 EvoFold data. </td>
<td>13 Feb 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE - University of Vienna, RNA secondary structure predicted by RNAz</td>
<td>16 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
Added NCBI Clone Validation information to details of MGC genes</td>
<td>12 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale RNA (update with October 2005 freeze data)</td>
<td>10 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UW/Reg QCP DNaseI Sens (replaces the following three tracks: UW/Reg DNaseI HSs,  UW/Reg Q/A,  UW DNase GM)</td>
<td>07 Dec 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Duke/NHGRI DNase (New Chip method in HepG2, Hela, GM, and CD4+)</td>
<td>05 Dec 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UT-Austin STAGE (update to add STAT1 tags)</td>
<td>29 Nov 2006</td>
</tr>
<TR valign=top><TD align=left>
Markers greater than 5,000 nt were removed.</td>
<td>14 Nov 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE GIS Chip-PET (P53 in HCT116, STAT1 in HeLa, c-Myc in P493)</td>
<td>07 Nov 2006</td>
</tr>
<TR valign=top><TD align=left>
HMR Conserved Transcription Factor Binding Sites (data reloaded)</td>
<td>06 Nov 2006</td>
</tr>
<TR valign=top><TD align=left>
Locus Variants (Sept 2006 snapshot)</td>
<td>24 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI/PSU/UCSC TBA Conserved Elements</td>
<td>05 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI/PSU/UCSC TBA Conservation</td>
<td>05 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI/PSU TBA Alignments</td>
<td>05 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE MSA Consensus Constrained Elements</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UC Berkeley  MAVID Alignments</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UC Berkeley MAVID Conservation</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UC Berkeley MAVID Conserved Elements</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford MLAGAN Conserved Elements</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford MLAGAN Conservation</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford MLAGAN Alignments</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE MSA Alignment Gaps</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE MSA Alignment Agreement</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
SNP Arrays</td>
<td>03 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
Update to UCSC Known Genes color coding scheme</td>
<td>03 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
Update CCDS to 2006-09-20 snapshot</td>
<td>22 Sep 2006</td>
</tr>
<TR valign=top><TD align=left>
CCDS (annotation update)</td>
<td>29 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
Retrotransposon Insertion Polymorphisms in Humans</td>
<td>23 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UNC FAIRE track (data update)</td>
<td>21 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
update CCDS to 2006-08-15 snapshot</td>
<td>17 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
8 new subtracks for ENCODE Sanger ChIP-chip track</td>
<td>17 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
Structural Variation</td>
<td>28 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
Rat Chain/Net(rn4)</td>
<td>13 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE University of Uppsala, Sweden ChIP-chip Na-butyrate time series</td>
<td>26 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
ESPERR Regulatory Potential (7 species)</td>
<td>22 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UNC FAIRE Peaks (ChIPOTle) Apr. 2006 Update, new subtrack added to UNC FAIRE</td>
<td>15 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated frame annotation for Conservation track to fix error with spliced codons</td>
<td>15 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Dog Chain/Net/LiftOver(canFam2), regenerated with better parameters</td>
<td>15 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Human Disease Variants and Other Mutations</td>
<td>14 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UW DNase/Array GM06990</td>
<td>13 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Sanger ChIP (additional data in HFL-1 cell line)</td>
<td>24 May 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Detection of LinEage Specific Selection (DLESS)</td>
<td>19 May 2006</td>
</tr>
<TR valign=top><TD align=left>
SNP build 125</td>
<td>01 May 2006</td>
</tr>
<TR valign=top><TD align=left>
MicroRNA target sites in 3' UTRs as predicted by PicTar</td>
<td>27 Apr 2006</td>
</tr>
<TR valign=top><TD align=left>
Most Conserved - data update</td>
<td>27 Apr 2006</td>
</tr>
<TR valign=top><TD align=left>
Conservation - Data Update (fix overlaps)</td>
<td>27 Apr 2006</td>
</tr>
<TR valign=top><TD align=left>
Most Conserved (correction to make rhesus non-informative)</td>
<td>28 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
Conservation - Data Update (rhesus made non-informative)</td>
<td>22 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
VisiGene (Allen Brain Atlas)</td>
<td>17 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
BosTau2, DanRer3 chains/net/liftOver</td>
<td>15 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
Most Conserved (17-way alignment phastCons conserved elements)</td>
<td>10 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
Conservation (17-way vertebrate multiple alignment)</td>
<td>10 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
RheMac2 chain/net/liftOver</td>
<td>06 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
sno/miRNA (wgRna)</td>
<td>21 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
HapMap LD (release 20)</td>
<td>21 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Improved searching capability for VisiGene</td>
<td>16 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Improved searching capability for Known Genes</td>
<td>03 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
gladHumESDistance (data update)</td>
<td>01 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Superfamily (data update)</td>
<td>30 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Comparatives downloads (replacements for overlapping axtNet alignments)</td>
<td>18 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP Sites (data replaced with 5 new transcription factors). Previous track Yale ChIP Sites renamed Yale STAT1 Sites.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP Signal (data replaced with 5 new transcription factors). Previous track Yale ChIP Signal renamed Yale STAT1 Signal.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP pVal (data replaced with 5 new transcription factors). Previous track Yale ChIP pVal renamed Yale STAT1 pVal.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Hgdownloads update: goldenPath/hg17/vs{PanTro1,CanFam1,Fr1,DanRer1,Mm5}/axtNet/{chr5.axt.gz,md5sum.txt}</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE EGASP Partial (data correction)</td>
<td>11 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Recomb Hotspots: ENCODE recombination hotspots and rates</td>
<td>06 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated hg17 -> mm6 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>06 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Recomb Rates and Hotspots: from HapMap and Perlegen data</td>
<td>04 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Uppsala Chip (University of Uppsala, Sweden ChIP-chip coordinate corrections)</td>
<td>03 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale TAR (Replace June data with new methods)</td>
<td>02 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UVa DNA Rep Ori (University of Virginia DNA Replication Origins, Oct 2005 Freeze)</td>
<td>29 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UT-Austin STAGE (Sequence Tag Analysis of Genomic Enrichment), c-Myc in HeLa cells</td>
<td>28 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Dog Chain/Net/LiftOver (canFam2)</td>
<td>22 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy Transfrags (updated HeLa data)</td>
<td>21 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy RNA Signal (updated HeLa data)</td>
<td>21 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Cow Synteny</td>
<td>20 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
mRNA, RefSeq mRNA, and native ESTs: Realigned and filtered
-- improvements in process.</td>
<td>16 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UVa DNA Rep TR50 (University of Virginia DNA Smoothed Timing at 50% Replication, Oct 2005 Freeze)</td>
<td>16 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UVa DNA Rep Seg (University of Virginia DNA Replication Temporal Segmentation, Oct 2005 Freeze)</td>
<td>02 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Sanger ChIP Hits (Peaks and Centers, Oct 2005 Freeze)</td>
<td>22 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Sanger ChIP (H3, H4 histones in GM, K562 and HeLa cells, Oct 2005 Freeze)</td>
<td>22 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
 ENCODE UNC FAIRE (Formaldehyde Assisted Isolation of Regulatory Elements, Oct 2005 Freeze)
</td>
<td>22 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UCD Ng ChIP (UC Davis ChIP-chip of E2F1, Myc in HeLa, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Uppsala Chip (University of Uppsala, Sweden ChIP-chip of H3ac,HNF3b,HNF4a,USF1 in HepG2 cells, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy Strict Sites (Affymetrix ChIP-chip of 5 factors in HL60 cells, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy Strict (Affymetrix ChIP-chip of 5 factors in HL60 cells, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy Strict Pval (Affymetrix ChIP-chip on 5 factors, in HL60 cells, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
Allen Brain Atlas Images</td>
<td>10 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy RNA (RNA abundance and transfrags, Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UT-Austin ChIP (c-Myc and EF24 in HeLa and 20901 fibroblasts, Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI DnaseI-HS (MPSS and Chip methods, in CD4 and GM09660, Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE GIS PET RNA (update Oct 2005 Freeze and new estrogen-treated MCF7)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENODE Genocde Introns (Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gencode Genes (Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
sno/miRNA (data update)</td>
<td>27 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE LI Ng gIF ChIP (Addition of 5 antibodies, in HeLa cells)</td>
<td>27 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
Oxford Recombination Hotspots (from HapMap phase I, release 16a)</td>
<td>25 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
Oxford Recombination Rates (from HapMap phase I, release 16a)</td>
<td>25 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE EGASP Update (Add Fgenesh++ corrected predictions)</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
Affy Transcription Phase 2</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
Exon Walk</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
Exoniphy</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE GIS STAT1 +/- gIF HeLa</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN Gene Predictions</td>
<td>04 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UVa DNA Rep Seg, UVa DNA Rep Ori (DNA replication timing segmentation and origins)</td>
<td>04 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
EvoFold (removed non-syntenic predictions and predictions with only 4 pairing bases)
</td>
<td>03 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
targetScanS miRNA</td>
<td>30 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE YALE MAS TAR (additional data)</td>
<td>27 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE YALE MAS RNA (additional data from June freeze)</td>
<td>27 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
Tajima's D, Tajima's D SNPs</td>
<td>23 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
5x Reg Potential</td>
<td>22 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE BU Orchid</td>
<td>21 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI DIPs: Whole Genome Shotgun insertion/deletion polymorphisms from Jim Mullikin/SSAHA</td>
<td>21 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: LI Ng gIF ChIP (HeLa)</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: LI Ng Validation (Pol2, Taf1, H3K4me, H3ac)</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Sanger ChIP-chip</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Ludwig Institute/UCSD ChIP-chip - Pol II, TAF1, AcH3, H3K4me, SUZ12, and H3K27me antibodies - repackage three existing tracks into one</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, P-Values)</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: LI gIF ChIP (HeLa)
</td>
<td>19 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE HapMap Allele Frequencies</td>
<td>19 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: GIS P53 5FU HCT116</td>
<td>19 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENENCODE: Affy Transfrags, Affy pVal {0,2,8,32}h, Affy Sites {0,2,8,32}h</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Affy RNA Signal </td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale MAS array, RNA Transcript Map, Placenta, Nimblegen Protocol</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale MAS array, RNA Transcriptionally Active Regions, Placenta, Nimblegen Protocol</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, Signal)</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, Binding Sites)</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale TAR </td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
HapMap Coverage: Depth of resequencing for 10 encode regions (work done by Broad and Baylor for the HapMap project)</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Stanford Endogenous Transcript Levels in HCT116 Cells; 
Stanford ChIP-chip (HCT116, Jurkat, K562 cells; Sp1, Sp3 ChIP); 
Stanford ChIP-chip Smoothed Score; 
Stanford Methylation Digest: Be2C, CRL1690, HCT116, HT1080, HepG2, JEG3, Snu182, U87; 
Stanford Methylation Digest Smoothed Score</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Stanf Promoter</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: BU First Exon</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale RNA </td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Sanger Assoc</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: UC Davis ChIP-chip Nimblegen - E2F1 ab, HeLa Cells</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: GIS RNA PET</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: TBA23 EvoFold (Predictions of RNA Secondary Structure Using TBA23, ENCODE Regions only)</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Known+Pred RNA (Known and Predicted RNA Transcription)</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: UW/Regulome DNase I Hypersensitivity Probability</td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Regulome DNase I Hypersensitivity</td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
UW/Regulome Plate Quality</td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
Regulome DNAse I Hypersensitivity Amplicons</td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: NHGRI DNaseI-HS </td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Riken CAGE - predicted gene start sites</td>
<td>13 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
Updated AceView (acembly) track from NCBI.</td>
<td>12 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gene Prediction Workshop (EGASP) Updates</td>
<td>12 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
EGASP Partial</td>
<td>31 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Sangamo/EIO CD34 DNase Hypersensitivity</td>
<td>30 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
EGASP Full</td>
<td>29 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Gencode Introns</td>
<td>29 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Gencode Genes</td>
<td>29 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Aternative Splicing</td>
<td>29 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
sno/miRNA (data update)</td>
<td>26 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Reloaded table for new known genes</td>
<td>03 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter: RankProp Score (table reloaded for new Known Genes)</td>
<td>01 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter support</td>
<td>29 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Old Known Genes (Table Browser only)</td>
<td>28 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Rhesus macaque (rheMac1) Chain and Net</td>
<td>25 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
bosTau1ToHg17 and hg17ToBosTau1 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>22 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
rheMac1ToHg17 and hg17ToRheMac1 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>22 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
EvoFold (Predictions of RNA Secondary Structure)</td>
<td>15 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
sno/miRNA (data update)</td>
<td>13 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
hg17ToXenTro1, hg17ToGalGal2, hg17ToMonDom1, hg17ToTetNig1, hg17ToDanRer2 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>06 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Known Genes, Gene Sorter, Proteome Browser (data update, new  filtering process)</td>
<td>15 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter (humanNormal microarray data)</td>
<td>02 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
hg17ToMm6 liftOver (Downloads and browser Batch Coordinate Conversion utility)</td>
<td>01 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
hg17ToRn3 liftOver (Downloads and browser Batch Coordinate Conversion utility)</td>
<td>01 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
vega and vegaPseudogene</td>
<td>10 May 2005</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
wgRna (data update)</td>
<td>26 Apr 2005</td>
</tr>
<TR valign=top><TD align=left>
Retroposed Genes</td>
<td>21 Apr 2005</td>
</tr>
<TR valign=top><TD align=left>
add table ccdsKgMap to link ccdsGene to knownGenes</td>
<td>20 Apr 2005</td>
</tr>
<TR valign=top><TD align=left>
Consensus CDS (X/Y PAR region corrected)</td>
<td>22 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
SNPs (data update: Affy genotyping SNPs)</td>
<td>11 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
CCDS genes (enabled robust details page)</td>
<td>04 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Non-Human RefSeqs</td>
<td>04 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
STS Markers (updated mapping)</td>
<td>28 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Yale Pseudo (Processed Pseudogenes from yale)</td>
<td>25 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Cow Chain/Net</td>
<td>18 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Ecores</td>
<td>15 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
RGD QTL (data update)</td>
<td>14 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Bounds
(Gene Boundaries as Defined by RNA and Spliced EST Clusters)</td>
<td>10 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Acembly (AceView Gene Models)</td>
<td>07 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
TFBS Conserved (transcr. factor binding sites)</td>
<td>04 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
SNPs (dbSNP build 124)</td>
<td>01 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
SNP track enhancement to support exceptions from dbSnp. (dbSnp build 123)</td>
<td>20 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
sno/miRNA - C/D and H/ACA Box snoRNAs, scaRNAs, and microRNAs from Weber and Griffiths-Jones (replaces the miRNA track)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish (danRer2) Chain and Net</td>
<td>06 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter extension (rankProp, psi-blast)</td>
<td>23 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>21 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Proteome Browser (data update)</td>
<td>18 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (data update)</td>
<td>18 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter (data update)</td>
<td>18 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA data correction for hsa-mir-30a</td>
<td>15 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
SNPs (track based on 'snp' table, not 'snpMap' table)</td>
<td>05 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>24 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
X. tropicalis Net</td>
<td>22 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
X. tropicalis Chain</td>
<td>22 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene new track, EST clustering gene prediction</td>
<td>22 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
WSSD Duplication (from Xinwei She and Evan Eichler)</td>
<td>29 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Segmental Dups (from Xinwei She and Evan Eichler)</td>
<td>29 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
UniGene version 171 added to Human browser</td>
<td>20 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Affy U133Plus2</td>
<td>13 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
RDG QTL</td>
<td>13 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (add 3', 5' UTR mRNA secondary structure predictions)</td>
<td>11 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (add BioCyc pathways links)</td>
<td>11 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>01 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (data update), Most Conserved added</td>
<td>21 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Self Chain</td>
<td>21 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Expression Distance (expanded data set)</td>
<td>16 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Regions</td>
<td>15 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Chain, Net</td>
<td>13 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
MicroRNAs from the miRNA Registry</td>
<td>09 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan gene predictions</td>
<td>08 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Retroposed Genes</td>
<td>01 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
fishClones</td>
<td>13 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
H-Inv (Human gene database, with full-length cDNA clones)</td>
<td>13 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Expression Distance</td>
<td>11 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Gene Maps</td>
<td>11 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter Fundamentals</td>
<td>11 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid</td>
<td>10 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial hg17 release (using NCBI Build 35): Chromosome Band, STS Markers, FISH Clones, Recomb Rate, Map Contigs, Assembly, Gap, Coverage, BAC End Pairs, Fosmid End Pairs, GC Percent, Known Genes, Proteome Browser, RefSeq Genes, MGC Genes, Genscan Genes, Human mRNAs, Spliced ESTs, Human ESTs, NonHuman mRNAs, NonHuman ESTs, CpG Islands, Fugu Blat, 
Conservation, Fugu Chain, Fugu Net, Zebrafish Chain, Zebrafish Net, Chicken Chain, Chicken Net, Rat Chain, Rat Net, Mouse Chain, Mouse Net, Dog Chain, Dog Net, Chimp Chain, Chimp Net, RepeatMasker, Simple Repeats.

 Downloads: bigZips, chromosomes, database</td>
<td>23 Jul 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=hg16></A>Human July 2003 (NCBI34/hg16) (hg16, NCBI Build 34)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
repair Known Genes Other Orgs links to mouse and elegans</td>
<td>27 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
Enable isPCR on UCSC Gene set to amplify exons only.</td>
<td>26 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Converted MGC track to PSL (to allow for alignment visualization)</td>
<td>22 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Added NCBI Clone Validation information to details of MGC genes</td>
<td>12 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
Add Conrad, McCarroll and Hinds deletions.</td>
<td>30 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
HapMap LD (Phase II; October 2005 data)</td>
<td>08 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Improved searching capability for Known Genes</td>
<td>03 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE EGASP Full (Correction)</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: hg16 -> galgal2, mm4, mm5, mm6, panTro1, pt0, rn3</td>
<td>04 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE GIS PET RNA (update)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
NHGRI DNaseI-HS</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE YALE MAS TAR (additional data)</td>
<td>27 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE YALE MAS RNA (additional data from June freeze)</td>
<td>27 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
Gencode Unannotated Region Classification</td>
<td>30 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Consensus Unannotated Transfrags</td>
<td>30 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Yale and Affymetrix Unannotated TransFrags</td>
<td>30 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
EGASP Full</td>
<td>26 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix 25 bp Probe Locations</td>
<td>18 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Yale RNA reloaded to fix off-by-one coordinates</td>
<td>08 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Yale RNA reloaded to fix off-by-one coordinates</td>
<td>08 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Yale TAR reloaded to fix off-by-one coordinates</td>
<td>08 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Affy RNA Signal reloaded to fix off-by-one coordinates</td>
<td>08 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Affy pVal reloaded to fix off-by-one coordinates</td>
<td>08 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
Mlagan Alignment (data update)</td>
<td>26 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE All Introns Proximal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE All Introns Distal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE All Intergenic Proximal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE All Intergenic Distal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Introns Proximal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Introns Distal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Intergenic Proximal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Intergenic Distal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
TBA23 EvoFold (Predictions of RNA Secondary Structure Using TBA23, ENCODE Regions only)</td>
<td>15 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
EvoFold (Predictions of RNA Secondary Structure)</td>
<td>15 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
HapMap Coverage: Depth of resequencing for 10 encode regions (work done by Broad and Baylor for the HapMap project)</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Regulome DNAse I Hypersensitivity Amplicons</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
UW/Regulome Plate Quality</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
UW/Regulome DNase I Hypersensitivity Probability</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Regulome DNase I Hypersensitivity</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Consensus Elements: TBA and MLAGAN Union/Intersection</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford LAGAN Alignments</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: NHGRI/PSU TBA Alignments</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Recombination in 10 ENCODE regions (cM/Mb)</td>
<td>13 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford LAGAN Conserved Elements </td>
<td>11 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford LAGAN Conservation</td>
<td>11 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Primate Alleles</td>
<td>08 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Prediction Workshop (EGASP) for Partial ENCODE Regions</td>
<td>06 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Prediction Workshop (EGASP) All ENCODE Regions</td>
<td>06 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Stanford Endogenous Transcript Levels in HCT116 Cells; 
Stanford ChIP-chip (HCT116, Jurkat, K562 cells; Sp1, Sp3 ChIP); 
Stanford ChIP-chip Smoothed Score; 
Stanford Methylation Digest: Be2C, CRL1690, HCT116, HT1080, HepG2, JEG3, Snu182, U87; 
Stanford Methylation Digest Smoothed Score</td>
<td>06 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Riken CAGE - predicted gene start sites</td>
<td>06 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gene Prediction Workshop (EGASP) Updates</td>
<td>05 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE HapMap Allele Frequencies</td>
<td>05 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
H3K4me3/K562 new subtrack added to Sanger ChIP</td>
<td>01 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Sanger Assoc</td>
<td>01 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Whole Genome Shotgun insertion/deletion polymorphisms from Jim Mullikin/SSAHA</td>
<td>01 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale MAS array, RNA Transcriptionally Active Regions, Placenta, Nimblegen Protocol</td>
<td>27 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale MAS array, RNA Transcript Map, Placenta, Nimblegen Protocol</td>
<td>27 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Affy RNA Signal, Affy Transfrags, Affy pVal {0,2,8,32}h, Affy Sites {0,2,8,32}h</td>
<td>17 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale RNA Transcriptionally Active Regions (TARs)</td>
<td>17 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale RNA Transcript Map (Neutrophil, Placenta and NB4 cells)</td>
<td>17 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, P-Values)</td>
<td>15 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, Binding Sites)</td>
<td>15 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Boston University ORChID</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: UC Davis ChIP-chip Nimblegen - E2F1 ab, HeLa Cells</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, Signal)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Gencode Introns (Intron validation for selected Gencode Genes)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Gencode Genes (Manual annotations in the ENCODE regions from the Gencode project)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Retroposed Genes (data update)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: ChIP-PET/GIS (data update, has cluster counts)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Ludwig Institute/UCSD ChIP-chip - Pol II, TAF1, AcH3, H3K4me, SUZ12, and H3K27me antibodies - repackage three existing tracks into one</td>
<td>13 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Ludwig Institute/UCSD ChIP-chip - gamma interferon experiments</td>
<td>13 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: GIS RNA PET</td>
<td>10 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: NHGRI DNaseI-HS (data update; now includes sites from activated as well as unactivated cells)</td>
<td>10 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Sanger ChIP-chip (histones H3,H4 GM06990)</td>
<td>10 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix ChIP-chip (Pol2 RA-stim HL60) Signal in ENCODE Regions (changed from wig to bedGraph)</td>
<td>08 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix ChIP-chip (Pol2 RA-stim HL60) PValue in ENCODE Regions (changed from wig to bedGraph)</td>
<td>08 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Stanf. Promoter in ENCODE regions (data update; changed to composite track -  old name: Promoter/Stanford) </td>
<td>06 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
DNA Rep/UVA in ENCODE regions (reloaded with new data)</td>
<td>03 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP-PET/GIS (P53 TFBS in ENCODE regions)</td>
<td>24 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Promoters/BU (Boston Univ. Promoters in ENCODE regions)</td>
<td>07 May 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UCSD/Ludwig Chip (acetylated and methylated histone binding)</td>
<td>06 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Mouse (March 2005/mm6) Alignment Net</td>
<td>04 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Mouse (March 2005/mm6) Chained Alignments</td>
<td>04 May 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UCSD Ludwig RNAP, ENCODE UCSD Ludwig TAF1 (tracks combined into composite format)</td>
<td>29 Apr 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIp/Sanger (Lysine 4 methylation of histone in ENCODE regions)</td>
<td>07 Apr 2005</td>
</tr>
<TR valign=top><TD align=left>
Updates hg16 SNPs to dbSnp build 122, including Affy genotyping SNPs.  </td>
<td>14 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
RNA/Affy Sites (Affymetrix PolyA+ RNA from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
RNA/Affy Signal (Affymetrix PolyA+ RNA abundance from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP Affy Pol2 Sites (Affymetrix ChIP-chip of Pol2 from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP Affy Pol2 Pval (Affymetrix ChIP-chip enrichment of Pol2 from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP-chip data in ENCODE regions</td>
<td>05 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
DNA Rep/UVA (ENCODE DNA Replication data from Dutta Lab, University of VIrginia)</td>
<td>03 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
DNaseI-HS/NHGRI (ENCODE DNaseI Hypersensitive sites, from Collins lab, NHGRI)</td>
<td>01 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
WSSD Overlay (from Xinwei She and Evan Eichler)</td>
<td>29 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
WSSD Coverage (from Xinwei She and Evan Eichler)</td>
<td>29 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Perlegen Haplotypes (8 records changed to correct block ordering)</td>
<td>11 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl EST Genes</td>
<td>10 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Vega Gene: data update (extending 3' UTR for 398 annotations).</td>
<td>23 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan: data update (additional exons added for 27 annotations).</td>
<td>20 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes: data update (64 annotations changed by 1 or 2 bases).</td>
<td>20 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Exoniphy: Conserved Exon Predictions</td>
<td>14 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Retroposed Genes</td>
<td>19 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
H-Inv (Full-length cDNA from the H-Invitational Human Gene Annotation Database)</td>
<td>08 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
ensTranscript and ensGeneXref tables dropped (replaced by
ensemblXref3)  </td>
<td>03 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Gene (updated to 34d), Superfamily (updated)</td>
<td>03 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
GC% Win20K (dropped - replaced by 5-base window GC Percent track)</td>
<td>01 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Alt-Splicing (data update)</td>
<td>27 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Chain, Fugu Net (using fr1)</td>
<td>24 May 2004</td>
</tr>
<TR valign=top><TD align=left>
STS Markers (updated)</td>
<td>24 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Chimp Recip Chain, Chimp Recip Net (updated to panTro1 assembly)
</td>
<td>21 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Chimp Chain (updated to panTro1 assembly)</td>
<td>21 May 2004</td>
</tr>
<TR valign=top><TD align=left>
TFBS Conserved</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Ecores</td>
<td>03 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Microsatellites</td>
<td>03 May 2004</td>
</tr>
<TR valign=top><TD align=left>
jaxOrtholog table</td>
<td>30 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Chicken Chain, Chicken Net (using galGal2)</td>
<td>29 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Ecores</td>
<td>29 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Acembly</td>
<td>20 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
SNPs (updated to dbSNP Build 120)</td>
<td>20 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
RNA Genes</td>
<td>13 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Hu/Mouse/Rat/Chick removed (replaced by Hu/Chp/Mo/Ra/Chk)</td>
<td>12 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Hu/Chp/Mo/Ra/Chk</td>
<td>12 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (updated to use proteins040315)</td>
<td>09 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix U133 (updated)</td>
<td>09 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF Atlas 2</td>
<td>09 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix GNF1H</td>
<td>09 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes (updated to include new peptide table)</td>
<td>05 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse/Rat removed (replaced by Hu/Mouse/Rat/Chick)</td>
<td>29 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Hu/Mouse/Rat/Chick (using mm3, rn3, galGal2) - replaces Human/Mouse/Rat track</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes (updated)</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA</td>
<td>19 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP (updated to use mm4)</td>
<td>05 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Map Contigs (updated)</td>
<td>02 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Perlegen (updated)</td>
<td>02 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>13 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
affyU133, affyU95 tables</td>
<td>10 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (updated)</td>
<td>06 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
2X Reg Potential, 3X Reg Potential</td>
<td>05 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes (update)</td>
<td>04 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
Mm3 Chain, Mm3 Net removed (replaced by mm4 versions)</td>
<td>20 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
chr{4,8,X,Un}_random_gap, chr{4,8,X,Un}_random_gold tables (updated)</td>
<td>16 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Chimp Diff</td>
<td>16 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Affy10K</td>
<td>14 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Mm4 Net (updates Mouse Net using mm4)</td>
<td>12 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Mm4 Chain (updates Mouse Chain using mm4)</td>
<td>09 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Mm4 Tight (updates Mouse Tight using mm4)</td>
<td>08 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes</td>
<td>22 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
vsPt0/axtBest downloads (chimp reciprocal best)</td>
<td>22 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
Alt-Splicing</td>
<td>18 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan Genes</td>
<td>18 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
Chimp All Chains, Chimp Net, Chimp sequence and chain downloads</td>
<td>10 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs (updated)</td>
<td>19 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>18 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>18 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Chain Self</td>
<td>15 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
UniGene_2, sage</td>
<td>12 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Bounds</td>
<td>12 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Chain Self</td>
<td>11 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Vega Genes, Vega Pseudogenes</td>
<td>07 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
NCI60</td>
<td>24 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (add UTR 3' and 5' folding data)</td>
<td>24 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Haplotype Blocks, Perlegen (chr 21 and 22 only)</td>
<td>24 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse/Rat downloads</td>
<td>24 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Synteny</td>
<td>21 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Genbank daily updates enabled</td>
<td>14 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Affy 120K SNPs</td>
<td>13 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Random SNPs, Overlap SNPs (dbSnp 117)</td>
<td>10 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Human/Mouse/Rat (using mm3 and rn3)</td>
<td>09 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (update)</td>
<td>08 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Fugu Blat (v. 3.0)</td>
<td>02 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
affyRatio table</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
vsMm3, vsRn3 data downloads</td>
<td>29 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Geneid</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain (using mm3)</td>
<td>10 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Net (using mm3) (contains fixes to chr1 data)</td>
<td>10 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Tight (using mm3) (contains fixes to chr1 data)</td>
<td>10 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
First EF (contains fixes to chrX and chrY)</td>
<td>09 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Net (using mm3)</td>
<td>02 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat Net (using rn3)</td>
<td>02 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat Chain</td>
<td>29 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain</td>
<td>29 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Tight Mouse</td>
<td>29 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Self Chain</td>
<td>29 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes</td>
<td>26 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
STS Markers (updated)</td>
<td>11 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial hg16 release (using NCBI Build 34): Chromosome Band, FISH Clones, Assembly, Gap, STS Markers, Map Contigs, Recomb Rate, GC Percent, Coverage, BAC End Pairs, Fosmid End Pairs, RefSeq Genes, MGC Genes, Genscan Genes, Human mRNAs, Spliced ESTs, Human ESTs, NonHuman mRNAs, NonHuman ESTs, CpG Islands, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosomes, database</td>
<td>08 Aug 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=mm10></A>Mouse Dec. 2011 (GRCm38/mm10) (mm10, Genome Reference Consortium Mouse Build 38 (GCA_000001635.2))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=ucscGenePfam" target=_blank>Pfam Domains in UCSC Genes - update</a></td>
<td>31 Jan 2017</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=wgEncodeGencodeVM11" target=_blank>GENCODE VM11 release</a></td>
<td>09 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
CRISPR</td>
<td>07 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Chimp Chain/Net</td>
<td>31 Oct 2016</td>
</tr>
<TR valign=top><TD align=left>
Chicken Chain/Net</td>
<td>26 Sep 2016</td>
</tr>
<TR valign=top><TD align=left>
Malayan flying lemur Chain/Net</td>
<td>16 Sep 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm10&g=mm10Patch4" target=_blank>GRCm38 patch release 4, alternate sequences and reference sequence patches</a></td>
<td>08 Jul 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../cgi-bin/hgTrackUi?db=mm10&g=knownGene" target=_blank>UCSC Genes</a></td>
<td>01 Jul 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=cloneEndSuper" target=_blank>Mapping of clone libraries end placements </a></td>
<td>03 May 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=wgEncodeGencodeVM7" target=_blank>GENCODE Version M9</a></td>
<td>22 Apr 2016</td>
</tr>
<TR valign=top><TD align=left>
Rhesus Chain/Net</td>
<td>18 Apr 2016</td>
</tr>
<TR valign=top><TD align=left>
Sequences from the Lens PatSeq Patent Document Database</td>
<td>11 Apr 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=snp142" target=_blank>Common SNPs(142), Flagged SNPs(142), Mult. SNPs(142), All SNPS(142) initial release

</a></td>
<td>20 Jan 2016</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>20 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=wgEncodeGencodeVM7" target=_blank>GENCODE Version M7</a></td>
<td>12 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
SGP Gene Predictions Using homology</td>
<td>27 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=knownGene" target=_blank>UCSC Genes</a></td>
<td>19 Jun 2015</td>
</tr>
<TR valign=top><TD align=left>
Human (Dec. 2013 (GRCh38/hg38)) Chained Alignments</td>
<td>14 May 2015</td>
</tr>
<TR valign=top><TD align=left>
Detailed Visualization of RepeatMasker Annotations</td>
<td>17 Mar 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=ucscRetroAli6" target=_blank>Retroposed Genes V6, Including Pseudogenes</a></td>
<td>24 Feb 2015</td>
</tr>
<TR valign=top><TD align=left>
Cow Chain/Net</td>
<td>17 Feb 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/db=mm10&g=wgEncodeGencodeVM4" target=_blank>GENCODE Version M4</a></td>
<td>22 Jan 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm10&g=wgEncodeGencodeVM3" target=_blank>GENCODE VM3 import from http://www.gencodegenes.org/</a></td>
<td>28 Oct 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm10&g=placentalChainNet" target=_blank>Shrew Chain/Net</a></td>
<td>14 Jul 2014</td>
</tr>
<TR valign=top><TD align=left>
FaceBase 24 Sample Types Averaged</td>
<td>02 Jul 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm10&g=cons60way" target=_blank>Vertebrate Multiz Alignment & Conservation (60 Species)</a></td>
<td>23 Jun 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=knownGene" target=_blank>UCSC Genes</a></td>
<td>11 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm10&g=wgEncodeGencodeVM2" target=_blank>GENCODE Gene annotations for mouse, Version M2</a></td>
<td>11 Mar 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm10&g=euarchontoglireChainNet" target=_blank>Human Chain/Net</a></td>
<td>06 Mar 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm10&g=snp138" target=_blank>Common SNPs(138), Mult. SNPs(138), All SNPS(138)</a></td>
<td>21 Feb 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes v74</td>
<td>23 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=jaxQtl" target=_blank>Mouse Genome Informatics Quantitative Trait Loci (lifted from mm9/NCBI 37)</a></td>
<td>03 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Alpaca Chain/Net</td>
<td>13 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=placentalChainNet" target=_blank>Tenrec Chain/Net</a></td>
<td>29 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Retroposed Genes for hg19, Build 5.0
Retroposed Genes for mm10, Build 2.0
</td>
<td>22 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
cytoBandIdeo tables added for navigation/orientation</td>
<td>24 Apr 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm10&g=snp137Common" target=_blank>SNPs (137) for mouse</a></td>
<td>06 Mar 2013</td>
</tr>
<TR valign=top><TD align=left>
qPCR Primers</td>
<td>29 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=placentalChainNet" target=_blank>White rhinoceros Chain/Net</a></td>
<td>24 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=mm10Strains1" target=_blank>Alternate mouse strain sequences</a></td>
<td>18 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=altSeqComposite1" target=_blank>GRCm38 patch1 alternate sequences and reference sequence patches </a></td>
<td>09 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=vertebrateChainNet" target=_blank>Lamprey Chain/Net</a></td>
<td>08 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
Contig fragments used in the assembly</td>
<td>19 Dec 2012</td>
</tr>
<TR valign=top><TD align=left>
Rhesus Chain/Net</td>
<td>03 Dec 2012</td>
</tr>
<TR valign=top><TD align=left>
Dolphin Chain/Net</td>
<td>03 Dec 2012</td>
</tr>
<TR valign=top><TD align=left>
Naked mole-rat Chain/Net</td>
<td>03 Dec 2012</td>
</tr>
<TR valign=top><TD align=left>
Cat Chain/Net</td>
<td>03 Dec 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=euarchontoglireChainNet" target=_blank>Chimp Chain/Net</a></td>
<td>21 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=euarchontoglireChainNet" target=_blank>Bushbaby Chain/Net</a></td>
<td>21 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=vertebrateChainNet" target=_blank>Nile tilapia Chain/Net</a></td>
<td>21 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=gliresChainNet" target=_blank>Pika Chain/Net</a></td>
<td>21 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=placentalChainNet" target=_blank>Armadillo Chain/Net</a></td>
<td>21 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=grcIncidentDb" target=_blank>GRC Incident</a></td>
<td>21 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
Budgerigar Chain/Net</td>
<td>20 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
Coelacanth Chain/Net</td>
<td>20 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
Atlantic cod Chain/Net</td>
<td>20 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=placentalChainNet" target=_blank>Alpaca Chain/Net</a></td>
<td>16 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=euarchontoglireChainNet" target=_blank>Tree shrew Chain/Net</a></td>
<td>16 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=euarchontoglireChainNet" target=_blank>Tarsier Chain/Net</a></td>
<td>16 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=placentalChainNet" target=_blank>Pig Chain/Net</a></td>
<td>16 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=placentalChainNet" target=_blank>Shrew Chain/Net</a></td>
<td>16 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=placentalChainNet" target=_blank>Megabat Chain/Net</a></td>
<td>16 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=placentalChainNet" target=_blank>Rock hyrax Chain/Net</a></td>
<td>16 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=euarchontoglireChainNet" target=_blank>Baboon Chain/Net</a></td>
<td>16 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=euarchontoglireChainNet" target=_blank>Gibbon Chain/Net</a></td>
<td>16 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=placentalChainNet" target=_blank>Manatee Chain/Net</a></td>
<td>13 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=gliresChainNet" target=_blank>Squirrel Chain/Net</a></td>
<td>13 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=euarchontogliresChainNet" target=_blank>Squirrel monkey Chain/Net</a></td>
<td>13 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=euarchontogliresChainNet" target=_blank>Mouse lemur Chain/Net</a></td>
<td>08 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=placentalChainNet" target=_blank>Hedgehog Chain/Net</a></td>
<td>08 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=gliresChainNet" target=_blank>Kangaroo rat Chain/Net</a></td>
<td>08 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=placentalChainNet" target=_blank>Sloth Chain/Net</a></td>
<td>08 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=cons60way" target=_blank>Vertebrate Multiz Alignment & Conservation (60 Species)</a></td>
<td>04 Oct 2012</td>
</tr>
<TR valign=top><TD align=left>
Medium ground finch Chain/Net</td>
<td>17 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
Tasmanian devil Chain/Net</td>
<td>20 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm10&g=placentalChainNet" target=_blank>Tenrec Chain/Net</a></td>
<td>29 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
Dog Chain/Net</td>
<td>25 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=chainNetRn5" target=_blank>Rat Chain/Net</a></td>
<td>11 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=chainNetGalGal4" target=_blank>Chicken Chain/Net</a></td>
<td>11 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=chainNetChrPic1" target=_blank>Painted turtle Chain/Net</a></td>
<td>11 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=chainNetBosTau7" target=_blank>Cow Chain/Net</a></td>
<td>11 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
Naked mole-rat Chain/Net</td>
<td>16 May 2012</td>
</tr>
<TR valign=top><TD align=left>
Initial mm10 release (using Genome Reference Consortium Mouse Build 38 (GCA_000001635.2)): Assembly, Gap, GC Percent, Short Match, Restr Enzymes, RefSeq Genes, Other RefSeq, MGC Genes, Ensembl Genes, Genscan Genes, Mouse mRNAs, Spliced ESTs, Mouse ESTs, Other mRNAs, CpG Islands, Repeat Masker, Interrupted Rpts, WM+SDust, Simple Repeats, Microsatellite</td>
<td>08 Mar 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=mm9></A>Mouse July 2007 (NCBI37/mm9) (mm9, NCBI Build 37)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Pfam in UCSC Gene</td>
<td>08 Jan 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://text.soe.ucsc.edu" target=_blank>pubsBingBlat (a run of hg19-BLAT against all webpages on the internet indexed by bing.com)</a></td>
<td>23 Jan 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeUwDnase" target=_blank>DNaseI Hypersensitivity by Digital DNaseI from ENCODE/University of Washington (Release 2)</a></td>
<td>25 Sep 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeUwDgf" target=_blank>DNaseI Digital Genomic Footprinting from ENCODE/University of Washington</a></td>
<td>07 Sep 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/LICR (Release 3)</a></td>
<td>05 Sep 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeCshlLongRnaSeq" target=_blank>Long RNA-seq from ENCODE/Cold Spring Harbor Lab (Release 3)</a></td>
<td>05 Sep 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhHist" target=_blank>Histone Modifications by ChIP-seq from ENCODE/SYDH (Release 2)</a></td>
<td>20 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/Stanford/Yale (Release 4)</a></td>
<td>16 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodePsuDnase" target=_blank>DNaseI Hypersensitivity by Digital DNaseI from ENCODE/PSU</a></td>
<td>16 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/LICR (Release 3)</a></td>
<td>13 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodePsuTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/PSU (Release 2)</a></td>
<td>07 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodePsuHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/PSU (Release 2)</a></td>
<td>06 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhRnaSeq" target=_blank>RNA-seq from ENCODE/Stanford/Yale (Release 2)</a></td>
<td>01 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeUwRnaSeq" target=_blank>RNA-seq from ENCODE/UW</a></td>
<td>10 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeFsuRepliChip" target=_blank>Replication Timing by Repli-chip from ENCODE/FSU</a></td>
<td>03 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeCshlLongRnaSeq" target=_blank>Long RNA-seq from ENCODE/Cold Spring Harbor Lab (Release 2)</a></td>
<td>29 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhRnaSeq" target=_blank>RNA-seq from ENCODE/Stanford</a></td>
<td>25 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm9&g=ikmc" target=_blank>IKMC Genes (update)</a></td>
<td>22 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeMapability" target=_blank>Mapability or Uniqueness of Reference Genome from ENCODE</a></td>
<td>19 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/Stanford/Yale (Release 3)</a></td>
<td>15 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeCaltechRnaSeq" target=_blank>RNA-seq from ENCODE/Caltech</a></td>
<td>07 May 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrRnaSeq" target=_blank>RNA-seq from ENCODE/LICR (Release 2)</a></td>
<td>04 May 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodePsuHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/PSU</a></td>
<td>30 Apr 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=grcIncidentDb" target=_blank>GRC Incident</a></td>
<td>17 Apr 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/LICR (Release 2)</a></td>
<td>13 Apr 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/LICR (Release 2)</a></td>
<td>28 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeCshlLongRnaSeq" target=_blank>Long RNA-seq from ENCODE/Cold Spring Harbor Lab</a></td>
<td>29 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodePsuTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/PSU</a></td>
<td>13 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
Yale Pseudogenes based on Ensembl Release 60</td>
<td>17 Jan 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeCaltechHist" target=_blank>Histone Modifications by ChIP-seq by ChIP-seq from ENCODE/Caltech</a></td>
<td>28 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeCaltechTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/Caltech</a></td>
<td>23 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=chainNetXenTro3" target=_blank>X. tropicalis Chain/Net</a></td>
<td>15 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=chainNetMelGal1" target=_blank>Turkey Chain/Net</a></td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=chainNetBosTau6" target=_blank>Cow Chain/Net</a></td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm9&g=chainNetAnoCar2" target=_blank>Lizard Chain/Net</a></td>
<td>04 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/Stanford/Yale (Release 2)</a></td>
<td>19 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm9&g=ikmc" target=_blank>IKMC Genes (update)</a></td>
<td>09 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm9&g=numtSeq" target=_blank>NumtS mitochondrial sequence</a></td>
<td>05 Aug 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrRnaSeq" target=_blank> 	
RNA-seq from ENCODE/LICR</a></td>
<td>04 Aug 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/LICR</a></td>
<td>22 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhHist" target=_blank>Histone Modifications by ChIP-seq from ENCODE/Stanford/Yale</a></td>
<td>22 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=allHg19RS_BW" target=_blank>Genomic Evolutionary Rate Profiling (GERP) scores for mammalian alignments</a></td>
<td>07 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeUwDnase" target=_blank>DNaseI Hypersensitivity by Digital DNaseI from ENCODE/University of Washington</a></td>
<td>01 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/LICR</a></td>
<td>24 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm9&g=jaxAllele" target=_blank>Data update for MGI annotations: MGI QTL, MGI Rep Transcript, MGI Phenotype, MGI Allele, MGI Gene Trap</a></td>
<td>22 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm9&g=laminB1Super" target=_blank>NKI Nuclear Lamina Associated Domains (LaminB1 DamID)</a></td>
<td>02 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=chainNetPanTro3" target=_blank>Chimp (Oct. 2010 (CGSC 2.1.3/panTro3)), Chain and Net Alignments</a></td>
<td>20 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Sheep Chain/Net</td>
<td>19 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=ikmc" target=_blank>IKMC Genes (update)</a></td>
<td>25 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm9&g=chainNetDanRer7" target=_blank>Zebrafish Chain/Net</a></td>
<td>04 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/Stanford/Yale </a></td>
<td>07 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=genomicSuperDups" target=_blank>Segmental Dups</a></td>
<td>26 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm9&g=targetScanS" target=_blank>TS miRNA sites</a></td>
<td>20 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=cons30way" target=_blank>PhyloP conservation tracks for 30-way alignment</a></td>
<td>17 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm9&g=jaxGeneTrap" target=_blank>MGI Gene Trap</a></td>
<td>05 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm9&g=jaxAllele" target=_blank>Data update for MGI annotations: MGI QTL, MGI Rep Transcript, MGI Phenotype, MGI Allele</a></td>
<td>05 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Cat Chain/Net</td>
<td>17 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Marmoset Chain/Net</td>
<td>03 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Pig Chain/Net</td>
<td>12 May 2010</td>
</tr>
<TR valign=top><TD align=left>
Mapability or Uniqueness of Reference Genome</td>
<td>04 May 2010</td>
</tr>
<TR valign=top><TD align=left>
Rabbit Chain/Net</td>
<td>26 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Panda Chain/Net</td>
<td>16 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=ikmc" target=_blank>IKMC Genes: database table name change</a></td>
<td>13 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Elephant Chain and Net</td>
<td>24 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=komp" target=_blank>IKMC Genes: data update</a></td>
<td>16 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Chain/Net and liftOver file</td>
<td>12 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish Chain/Net  composite tracks</td>
<td>08 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=NIAGene" target=_blank>NIA Gene Index</a></td>
<td>23 Feb 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=komp" target=_blank>IKMC Genes: data update</a></td>
<td>13 Jan 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
Chain and Net for Horse/equCab2 on Dog and Mouse</td>
<td>03 Nov 2009</td>
</tr>
<TR valign=top><TD align=left>
rn4 <--> mm9 Chains & Nets and LiftOvers (update)</td>
<td>30 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
Opossum Chain and Net</td>
<td>24 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=komp" target=_blank>IKMC Genes: data update</a></td>
<td>23 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=rest" target=_blank>Repressor Element 1-Silencing Transcription Factor (REST) binding sites</a></td>
<td>21 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Data update for MGI annotations: MGI QTL, MGI Rep Transcript, MGI Phenotype, MGI Allele</td>
<td>18 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Chains and Nets for non-primate mammal species vs Human</a></td>
<td>16 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Vega Genes (build 35 update)</td>
<td>09 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 55</td>
<td>11 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=igtc" target=_blank>Gene Trap (data update 07-01-2009)</a></td>
<td>21 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
Data update for MGI annotations: MGI QTL, MGI Rep Transcript, MGI Phenotype, MGI Allele</td>
<td>07 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=igtc" target=_blank>Gene Trap (data update 06-01-2009)</a></td>
<td>15 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=komp" target=_blank>IKMC Genes: data update</a></td>
<td>10 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=igtc" target=_blank>Gene Trap (data update 05-01-2009)</a></td>
<td>20 May 2009</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>14 Apr 2009</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 03-01-2009)</td>
<td>12 Mar 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>18 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 01-01-2009)</td>
<td>21 Jan 2009</td>
</tr>
<TR valign=top><TD align=left>
Data update for MGI annotations: MGI QTL, MGI Rep Transcript, MGI Phenotype, MGI Allele</td>
<td>15 Dec 2008</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 12-01-2008)</td>
<td>08 Dec 2008</td>
</tr>
<TR valign=top><TD align=left>
Enable isPCR on UCSC Gene set to amplify exons only.</td>
<td>26 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Medaka Chain and Net</td>
<td>19 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 11-01-2008)</td>
<td>05 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
KOMP Genes</td>
<td>24 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 10-01-2008)</td>
<td>13 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
ORegAnno track update</td>
<td>07 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Converted MGC track to PSL (to allow for alignment visualization)</td>
<td>22 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 09-01-2008)</td>
<td>04 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update (version 50)</td>
<td>22 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 08-01-2008)</td>
<td>11 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
MGI annotations on NCBI Build 37 (mm9): MGI QTL, MGI Rep Transcript, MGI Phenotype, MGI Allele</td>
<td>29 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 07-01-2008)</td>
<td>09 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 06-01-2008)</td>
<td>13 Jun 2008</td>
</tr>
<TR valign=top><TD align=left>
Guinea Pig Chain and Net</td>
<td>11 Jun 2008</td>
</tr>
<TR valign=top><TD align=left>
Lamprey Chain/Net/LiftOver (petMar1)</td>
<td>10 Jun 2008</td>
</tr>
<TR valign=top><TD align=left>
Lancelet Chain/Net/LiftOver (braFlo1)</td>
<td>10 Jun 2008</td>
</tr>
<TR valign=top><TD align=left>
MicroRNAs from miRBase (track update)</td>
<td>05 Jun 2008</td>
</tr>
<TR valign=top><TD align=left>
Vega Genes - Vega Annotations, Vega Annotated Pseudogenes and Immunoglobulin Segments</td>
<td>04 Jun 2008</td>
</tr>
<TR valign=top><TD align=left>
TROMER Transcriptome database, SIB Alt-Splicing</td>
<td>29 May 2008</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 05-01-2008)</td>
<td>07 May 2008</td>
</tr>
<TR valign=top><TD align=left>
Marmoset Chain/Net/LiftOver (calJac1)</td>
<td>24 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 04-01-2008)</td>
<td>10 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Chain track, Net track, and liftOver file to bosTau4</td>
<td>08 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 03-01-2008)</td>
<td>06 Mar 2008</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 02-01-2008)</td>
<td>14 Feb 2008</td>
</tr>
<TR valign=top><TD align=left>
Updated with CCDS genes (as of 2008-02-01) </td>
<td>12 Feb 2008</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 01-01-2008)</td>
<td>11 Jan 2008</td>
</tr>
<TR valign=top><TD align=left>
Adding BAC End: RP23-473N24</td>
<td>09 Jan 2008</td>
</tr>
<TR valign=top><TD align=left>
CCDS update (as of 2007-12-12)</td>
<td>14 Dec 2007</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 12-01-2007)</td>
<td>04 Dec 2007</td>
</tr>
<TR valign=top><TD align=left>
Conservation (30-vertebrate alignment)</td>
<td>30 Nov 2007</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 11-01-2007)</td>
<td>16 Nov 2007</td>
</tr>
<TR valign=top><TD align=left>
X. tropicalis (Aug. 2005/xenTro2) Chained Alignments
X. tropicalis (Aug. 2005/xenTro2) Alignment Net</td>
<td>05 Nov 2007</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands</td>
<td>25 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN gene predictions from WUStL</td>
<td>25 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Micro-RNAs</td>
<td>19 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Platypus (Mar. 2007/ornAna1) Chained Alignments
Platypus (Mar. 2007/ornAna1) Alignment Net</td>
<td>19 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Chimp (Mar 2006/panTro2) Chained Alignments
Chimp (Mar 2006/panTro2) Alignment Net</td>
<td>19 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Chicken (May 2006/galGal3) Chained Alignments
Chicken (May 2006/galGal3) Alignment Net</td>
<td>19 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Cow (Aug. 2006/bosTau3) Chained Alignments
Cow (Aug. 2006/bosTau3) Alignment Net</td>
<td>16 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Horse (Jan. 2007/equCab1) Chained Alignments
Horse (Jan. 2007/equCab1) Alignment Net</td>
<td>16 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Lizard (Feb. 2007/anoCar1) Chained Alignments
Lizard (Feb. 2007/anoCar1) Alignment Net</td>
<td>16 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
SGP Gene Predictions Using Mouse/Human Homology</td>
<td>15 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>12 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 10-01-2007)</td>
<td>12 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
UCSC Genes</td>
<td>12 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Opossum (Jan 2006/monDom4) Chained Alignments
Opossum (Jan 2006/monDom4) Alignment Net</td>
<td>04 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
AceView Genes</td>
<td>19 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Fugu (Oct. 2004/fr2) Chained Alignments
Fugu (Oct. 2004/fr2) Chained Alignments</td>
<td>14 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Stickleback (Feb. 2006/gasAcu1) Chained Alignments
Stickleback (Feb. 2006/gasAcu1) Chained Alignments</td>
<td>12 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Rhesus (Jan. 2006/rheMac2) Chained Alignments
Rhesus (Jan. 2006/rheMac2) Alignment Net</td>
<td>11 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon (Feb. 2004/tetNig1) Alignment Net
Tetraodon (Feb. 2004/tetNig1) Chained Alignments</td>
<td>10 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Dog (May 2005/canFam2) Chained Alignments
Dog (May 2005/canFam2) Alignment Net</td>
<td>07 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Medaka (Apr. 2006/oryLat1) Chained Alignments
Medaka (Apr. 2006/oryLat1) Alignment Net</td>
<td>06 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Rat (rn4) Chain and Net tracks and liftOver file</td>
<td>28 Aug 2007</td>
</tr>
<TR valign=top><TD align=left>
Human (hg18) Chain and Net tracks and liftOver file</td>
<td>28 Aug 2007</td>
</tr>
<TR valign=top><TD align=left>
Initial mm9 release (using NCBI Build 37): Chromosome Band; STS Markers; Map Contigs; Assembly; Gap; BAC End Pairs; GC Percent; RefSeq Genes; Other RefSeq; MGC Genes; Genscan Gene;Mouse mRNAs; Spliced ESTs; Mouse ESTs; Non Mouse mRNAs; RepeatMasker; Simple Repeats; Microsatellite; Downloads: bigZips, chromosomes, database.</td>
<td>22 Aug 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=mm8></A>Mouse Feb. 2006 (NCBI36/mm8) (mm8, NCBI Build 36)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Enable isPCR on UCSC Gene set to amplify exons only.</td>
<td>26 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Broad H3 ChIPseq (Initial submission of mouse chromatin mapping tracks)</td>
<td>23 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Converted MGC track to PSL (to allow for alignment visualization)</td>
<td>22 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Updated with CCDS genes (as of 2008-02-01) </td>
<td>12 Feb 2008</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix Mouse Exon array version 1.0 probes and transcripts</td>
<td>18 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Added CONTRAST gene predictions</td>
<td>17 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Swiss Institute of Bioinformatics, SIB Genes initial release</td>
<td>25 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Swiss Institute of Bioinformatics Alt-Splicing initial release. </td>
<td>25 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 09-01-2007)</td>
<td>12 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Updated Drosophila column in Gene Sorter from dm2 to dm3 FlyBase genes</td>
<td>28 Aug 2007</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 08-01-2007)</td>
<td>03 Aug 2007</td>
</tr>
<TR valign=top><TD align=left>
Data update and added phenotype-allele relationships.</td>
<td>01 Aug 2007</td>
</tr>
<TR valign=top><TD align=left>
Jackson Lab/MGI QTL</td>
<td>01 Aug 2007</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 07-01-2007)</td>
<td>09 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 06-01-2007)</td>
<td>15 Jun 2007</td>
</tr>
<TR valign=top><TD align=left>
CGAP SAGE</td>
<td>15 Jun 2007</td>
</tr>
<TR valign=top><TD align=left>
miRNA (data update)</td>
<td>01 Jun 2007</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 05-01-2007)</td>
<td>07 May 2007</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 04-01-2007)</td>
<td>19 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Updated annotated maf's to include gap information from other species</td>
<td>12 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Update of hgBlastTab tables (to hg18) to coincide with UCSC Genes</td>
<td>06 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 3-1-2007)</td>
<td>14 Mar 2007</td>
</tr>
<TR valign=top><TD align=left>
Allen Brain Atlas Probes</td>
<td>07 Mar 2007</td>
</tr>
<TR valign=top><TD align=left>
Added NCBI Clone Validation information to details of MGC genes</td>
<td>12 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
miRNA data update</td>
<td>12 Dec 2006</td>
</tr>
<TR valign=top><TD align=left>
Jackson Lab/MGI Representative Transcript, Phenotype and Allele data</td>
<td>20 Nov 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated STS Markers on Genetic and Radiation Hybrid Maps to correct map distances</td>
<td>03 Nov 2006</td>
</tr>
<TR valign=top><TD align=left>
SGP Gene Predictions</td>
<td>18 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>17 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
NIA Mouse Gene Index</td>
<td>15 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
Consensus CDS (CCDS)</td>
<td>10 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
Update to UCSC Known Genes color coding scheme</td>
<td>03 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
MicroRNAs from miRBase</td>
<td>07 Sep 2006</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN gene predictions from WUStL</td>
<td>31 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
SNP build 126</td>
<td>14 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
Rat Ortholog column of Gene Sorter and Known Gene details updated to rn4</td>
<td>20 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Gene Predictions</td>
<td>10 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
ESPERR Regulatory Potential (7 species)</td>
<td>22 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap</td>
<td>19 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Physical Map Contigs</td>
<td>16 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated frame annotation for Conservation track to fix error with spliced codons</td>
<td>15 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
chains and nets to Zebrafish (danRer4)</td>
<td>31 May 2006</td>
</tr>
<TR valign=top><TD align=left>
affy U74</td>
<td>22 May 2006</td>
</tr>
<TR valign=top><TD align=left>
affy MOE430</td>
<td>22 May 2006</td>
</tr>
<TR valign=top><TD align=left>
affy GNF1m</td>
<td>22 May 2006</td>
</tr>
<TR valign=top><TD align=left>
Initial mm8 release (using NCBI Build 36): Chromosome Band; STS Markers; Assembly; Gap; BAC End Pairs; GC Percent; Known Genes; Proteome Browser; RefSeq Genes; MGC Genes; Genscan Gene;Mouse mRNAs; Spliced ESTs; Mouse ESTs; NonMouse mRNAs; Allen Brain; Rinn Sex Exp; GNF U74A, GNF U74B, GNF U74C, CpG Islands; Conservation; Most Conserved; Chains & Nets: Fugu, Tetraodon, Zebrafish, X. tropicalis, Chicken, Cow, Dog,  Rhesus, Chimp, Human; RepeatMasker;  Simple Repeats.

Downloads: bigZips, chromosomes, database, liftOvers, multiz17way, phastCons17Scores</td>
<td>10 May 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=mm7></A>Mouse Aug. 2005 (NCBI35/mm7) (mm7, NCBI Build 35)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Converted MGC track to PSL (to allow for alignment visualization)</td>
<td>22 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Gene Trap (data update 3-1-2007)</td>
<td>14 Mar 2007</td>
</tr>
<TR valign=top><TD align=left>
Allen Brain Atlas Probes</td>
<td>07 Mar 2007</td>
</tr>
<TR valign=top><TD align=left>
Added NCBI Clone Validation information to details of MGC genes</td>
<td>12 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
Update to UCSC Known Genes color coding scheme</td>
<td>03 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
regPotential7X data download files for hg18, hg17, mm8, mm7</td>
<td>26 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
ESPERR Regulatory Potential (7 species)</td>
<td>22 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
MicroRNA target sites in 3' UTRs as predicted by PicTar</td>
<td>24 May 2006</td>
</tr>
<TR valign=top><TD align=left>
sgpGene</td>
<td>24 May 2006</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish Chain/Net/LiftOver (danRer3)</td>
<td>21 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Link to Gene Network Id from Known Gene details page</td>
<td>16 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Improved searching capability for VisiGene</td>
<td>16 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Improved searching capability for Known Genes</td>
<td>03 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Geneid gene predictions from IMIM at GRIB</td>
<td>31 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Allen Brain Atlas Images</td>
<td>30 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: mm7 -> fr1/tetNig1/xenTro1/galGal2/bosTau2/rheMac1/panTro1/hg17/rn3</td>
<td>22 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Dog Chain/Net/LiftOver (canFam2)</td>
<td>22 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Mouse liftOvers: mm3, mm4, mm5, mm6 -> mm7; mm5-> galGal2; mm6 -> galGal2, monDom1, xenTro1, canFam2, hg16, panTro1, danRer2, fr1, tetNig1</td>
<td>21 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Initial mm7 Release (using NCBI build 35): Chromosome Band;  STS Markers; Map Contigs; Gap; BAC End Pairs; GC Percent; Short Match; Restriction Enzymes; CpG Islands; Known Genes; MGC Genes; RefSeq Genes; Other RefSeq; Genscan Genes; ESTs; miRNAs; mRNAs; Non-Mouse mRNAs; Repeat Masker; Simple Repeats; Riken CAGE; Riken CAGE Tc; Allen Brain Atlas; Rinn Sex Expression; GNF Expression; Conservation; Most Conserved; Chains/Nets: Fugu, Tetraodon, X. tropicalis, Chicken, Cow, Rhesus, Chimp, Human, Rat; Gene Sorter; Proteome Browser. Downloads: bigZips, chromosomes, database, comparatives.</td>
<td>15 Dec 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=anoGam1></A>A. gambiae Feb. 2003 (IAGEC MOZ2/anoGam1) (anoGam1, IAGEC v.MOZ2)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
D. melanogaster Chain/Net</td>
<td>18 Dec 2014</td>
</tr>
<TR valign=top><TD align=left>
D. melanogaster Chain/Net/LiftOver (dm2 update)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>21 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
AnoEST Unique and Non-Unique (TCLAG*: Clustered A. gambiae ESTs)</td>
<td>08 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
GC Percent</td>
<td>22 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial anoGam1 release (MOZ2 draft genome sequence Feb. 2003): Assembly, Gap, Ensembl Genes, Genscan Genes, A. gambiae mRNAs, Spliced ESTs, A. gambiae ESTs, Non-A. gambiae mRNAs, Non-A.gambiae ESTs, D. mel Chain, D. mel. Net, RepeatMasker, Simple Repeats

 Downloads: bigZips, chromosomes, database</td>
<td>10 Aug 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=apiMel2></A>A. mellifera Jan. 2005 (Baylor 2.0/apiMel2) (apiMel2, Baylor HGSC Amel_2.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
update to existing dm2 chains and nets</td>
<td>02 Nov 2006</td>
</tr>
<TR valign=top><TD align=left>
D. melanogaster LiftOver (dm2)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
NCBI Genes (update from NCBI: alternate gene names)</td>
<td>01 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
EMBL Homologs to D. melanogaster</td>
<td>31 May 2005</td>
</tr>
<TR valign=top><TD align=left>
NCBI Genes</td>
<td>19 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>21 Apr 2005</td>
</tr>
<TR valign=top><TD align=left>
Human(hg17) proteins mapped by chained tBLASTn</td>
<td>28 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
D. melanogaster Proteins</td>
<td>28 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Assembly Scaffolds</td>
<td>28 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
A. mellifera (Honeybee) Browser Jan. 2005 Release (apiMel2, Baylor HGSC Amel_2.0): Assembly, Gap, GC Percent, Genscan,  A. mellifera mRNAs, Non-A. mellifera mRNAs, A. mellifera Spliced ESTs, A. mellifera ESTs, D. melanogaser Chain/Net, RepeatMasker, Simple Repeats. Downloads: bigZips, database, vsDm2.</td>
<td>23 Feb 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=apiMel1></A>A. mellifera July 2004 (Baylor 1.2/apiMel1) (apiMel1, Baylor HGSC Amel_1.2)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
D. melanogaster Chain/Net/LiftOver (dm2 update)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Honeybee Browser Initial Release (Baylor HGSC Amel_1.2): Assembly, Gap, GC Percent, Genscan, Honeybee mRNA, Spliced ESTs, Honeybee ESTs, Mapped D.melanogaster Proteins, D. melanogaser Chain/Net, RepeatMasker, Simple Repeats</td>
<td>22 Dec 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=vicPac2></A>Alpaca Mar. 2013 (Vicugna_pacos-2.0.1/vicPac2) (vicPac2, Broad Institute)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Initial vicPac2 release (using Broad Institute): Assembly, Gap, GC Percent, Short Match, Restr Enzymes, Other RefSeq, Genscan Genes, Aplaca mRNAs, Spliced ESTs, Alpaca ESTs, CpG Islands, Mouse Chain/Net, Human Chain/Net, RepeatMasker, Interrupted Rpts, WM + SDust, Simple Repeats</td>
<td>11 Sep 2013</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=vicPac1></A>Alpaca Jul. 2008 (Broad/vicPac1) (vicPac1, Broad Institute vicPac1)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl gene predictions - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial vicPac1 release (using Broad Institute vicPac1): Assembly, Gap, GC Percent, Quality Scores, Ensembl Assembly, Other RefSeq, Ensembl Genes, Genscan Genes, Spliced ESTs, Alpaca ESTs, CpG Islands, Cow Chain/Net, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>16 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=allMis1></A>American alligator Aug. 2012 (allMis0.2/allMis1) (allMis1, International Crocodilian Genomes Working Group)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Initital allMis1 release (using Inter. Crocodilian Genomes Working Group allMis0.2 (NCBI project 159843, WGS AKHW01)): Assembly, Gap, GC Percent, Short Match, Restriction Enzymes, Other RefSeq, Genscan Genes, American alligator mRNAs, Spliced ESTs, American alligator ESTs, Other mRNAs, CpG Islands, Human Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>02 Oct 2013</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=dasNov3></A>Armadillo Dec. 2011 (Baylor/dasNov3) (dasNov3, Baylor College of Medicine Dasnov3.0 (GCA_000208655.2))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v74</td>
<td>23 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial dasNov3 release (using Baylor College of Medicine Dasnov3.0 (GCA_000208655.2)): Assembly, Gap, GC Percent, Other RefSeq, Genscan Genes, Armadillo mRNAs, Other mRNAs, CpG Islands, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>21 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=gadMor1></A>Atlantic cod May 2010 (Genofisk GadMor_May2010/gadMor1) (gadMor1, Genofisk GadMor_May2010 (GCA_000231765.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl gene predictions - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial gadMor1 release (using Genofisk GadMor_May2010 (GCA_000231765.1)): Assembly, Gap, GC Percent, Short Match, Restr Enzymes, Other RefSeq, Genscan Genes, Atlantic cod mRNAs, Spliced ESTs, Atlantic cod ESTs, CpG Islands, Fugu Chain/Net, Mouse Chain/Net, RepeatMasker, Interrupted Rpts, WM + SDust, Simple Repeats</td>
<td>20 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=papAnu2></A>Baboon Mar. 2012 (Baylor Panu_2.0/papAnu2) (papAnu2, Baylor College of Medicine)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>20 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
Tarsier Chain/Net</td>
<td>22 May 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Gene Predictions - Ensembl 79</td>
<td>08 Apr 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Gene Predictions - Ensembl 78</td>
<td>06 Feb 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=papAnu2&g=ensGene" target=_blank>Ensembl Gene Predictions - Ensembl 76</a></td>
<td>21 Aug 2014</td>
</tr>
<TR valign=top><TD align=left>
Initial papAnu2 release (using Baylor College of Medicine): Assembly, Gap, GC Percent, Short Match, Restr Enzymes, RefSeq Genes, Other RefSeq, Genscan Genes, Baboon mRNAs, Spliced ESTs, Baboon ESTs, Other mRNAs, CpG Islands,  RepeatMasker, Interrupted Rpts, Simple Repeats.</td>
<td>23 Jan 2013</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=papHam1></A>Baboon Nov. 2008 (Baylor Pham_1.0/papHam1) (papHam1, Baylor BCM HGSC Pham_1.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=papHam1&g=chainNetMacFas5" target=_blank>Crab-eating macaque Chain/Net</a></td>
<td>07 Apr 2016</td>
</tr>
<TR valign=top><TD align=left>
Initial papHam1 release (using Baylor BCM HGSC Pham_1.0): Assembly, Gap, GC Percent, Quality Scores, Other RefSeq, Genscan Genes, Baboon mRNAs, Other mRNAS, CpG Islands, Gorilla Chain/Net, Marmoset Chain/Net, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>16 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=panPan1></A>Bonobo May 2012 (Max-Planck/panPan1) (panPan1, Max-Planck Institute panpan1 (GCA_000258665.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>20 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
Initial panPan1 release (using Max-Planck Institute panpan1 (GCA_000258665.1)): Assembly, Gap, GC Percent, INSDC, Genscan Genes, Augustus, RefSeq Genes, Other RefSeq, Bonobo mRNAs, CpG Islands, Tarsier Chain/Net, Human Chain/Net, Simple Repeats, Interrupted Rpts, WM + SDust, RepeatMasker, Microsatellite</td>
<td>02 Jul 2015</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=aptMan1></A>Brown kiwi Jun. 2015 (MPI-EVA AptMant0/aptMan1) (aptMan1, Max-Plank Institute for Evolutionary Anthropology AptMant0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Initial aptMan1 release (using Max-Plank Institute for Evolutionary Anthropology AptMant0): Assembly, Gap, GC Percent, Restr Ensymes, Short Match, Augustus Genes, Genscan Genes, RefSeq Genes, Other RefSeq, Brown kiwi mRNAs, CpG Islands, Microsatellite, RepeatMasker, Simple Repeats,  WM + SDust</td>
<td>22 Mar 2016</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=melUnd1></A>Budgerigar Sep. 2011 (WUSTL v6.3/melUnd1) (melUnd1, WUSTL Melopsittacus undulatus Budgerigar Version 6.3 (GCA_000238935.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial melUnd1 release (using WUSTL Melopsittacus undulatus Budgerigar Version 6.3 (GCA_000238935.1)): Assembly, Gap, GC Percent, Short Match, Restr Enzymes, Other RefSeq, Genscan Genes, Budgerigar mRNAs, CpG Islands, Medium ground finch Chain/Net, Mouse Chain/Net, RepeatMasker, Interrupted Rpts, WM + SDust, Simple Repeats</td>
<td>20 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=otoGar3></A>Bushbaby Mar. 2011 (Broad/otoGar3) (otoGar3, Broad Institute (GCA_000181295.3))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Tarsier Chain/Net</td>
<td>22 May 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v74</td>
<td>23 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial otoGar3 release (using Broad Institute (GCA_000181295.3)): see separate push queue otoGar3.</td>
<td>21 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=caePb2></A>C. brenneri Feb. 2008 (WUGSC 6.0.1/caePb2) (caePb2, Washington University School of Medicine GSC Caenorhabditis brenneri (PB2801) 6.0.1)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
C. elegans Chain/Net</td>
<td>06 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
C. elegans Proteins WS190 Mapped by Chained tBLASTn</td>
<td>18 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial caePb2 release (using WUSTL Caenorhabditis brenneri (PB2801) 6.0.1):  Scaffolds, Contigs, Gap, GC Percent, Short Match, Restr Enzymes, Non-C. brenneri RefSeq, C. brenneri mRNAs, Spliced ESTs, C. brenneri ESTs, Non-C. brenneri mRNAs, WM + SDust, Simple Repeats.  Downloads: bigZips, chromsomes, database.</td>
<td>20 Jun 2008</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=caePb1></A>C. brenneri Jan. 2007 (WUGSC 4.0/caePb1) (caePb1, Washington University School of Medicine GSC Caenorhabditis n. sp. PB2801)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
tRNA Genes</td>
<td>15 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial caePb1 release (using WUSTL Caenorhabditis n. sp. PB2801): Scaffolds; Contigs; Gap; GC Percent; Quality Scores; Short Match; Restr Enzymes; Non-C. brenneri RefSeq; C. elegans Proteins; Non-C. brenneri mRNAs; Chains and Nets: P. pacificus, C. briggsae, C. remanei, C. elegans; RepeatMasker; WM + SDust; Simple Repeats.  Downloads: bigZips, chromsomes, database, liftOvers, comparatives.</td>
<td>03 Aug 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=cb3></A>C. briggsae Jan. 2007 (WUGSC 1.0/cb3) (cb3, Washington University School of Medicine GSC and Sanger Institute cb3)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
P. pacificus Chain/Net</td>
<td>06 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
C. elegans Chain/Net</td>
<td>06 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
C. elegans Proteins WS190 Mapped by Chained tBLASTn</td>
<td>18 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial cb3 release (using WUSTL cb3): Assembly; Gap; GC Percent; Short Match; Restr Enzymes; Non-C. briggsae RefSeq; C. elegans Proteins; C. briggsae mRNAs; Spliced ESTs; C. briggsae ESTs; Non-C. briggsae mRNAs; Chains and Nets: P. pacificus, C. elegans, C. brenneri, C. remanei; RepeatMasker; WM + SDust; Simple Repeats. Downloads: bigZips, chromsomes, database, liftOvers, comparatives.</td>
<td>03 Aug 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=cb1></A>C. briggsae July 2002 (WormBase cb25.agp8/cb1) (cb1, WormBase v. cb25.agp8)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
genbank updates enabled</td>
<td>29 Nov 2007</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>03 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Elegans Chain (updated to ce2)</td>
<td>25 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Elegans Blastz (updated to ce2)</td>
<td>25 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial cb1 Release (using WormBase v. cb25.agp8): Assembly, Gap, GC Percent, Worm mRNAs, Worm ESTs, Elegans Chain, Elegans Blastz, RepeatMasker, Simple Repeats, data set downloads.</td>
<td>24 Jun 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=ce11></A>C. elegans Feb. 2013 (WBcel235/ce11) (ce11, C. elegans Sequencing Consortium WBcel235)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Initial ce11 release (using C. elegans Sequencing Consortium WBcel235 (GCA_000002985.3)): Assembly, Gap, GC Percent, INSDC, Augustus Genes, Genscan Genes, RefSeq Genes, WS245 Genes, C. elegans ESTs, C. elegans mRNAs, Spliced ESTs, Other mRNAs, CpG Islands, Nematode 26-way Conservation, Interrupted Repeats, Microsatellites, RepeatMasker, Self Chain, Simple Repeats, WM + SDust</td>
<td>21 Mar 2016</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=ce10></A>C. elegans Oct. 2010 (WS220/ce10) (ce10, Washington University School of Medicine GSC and Sanger Institute WS220)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
CRISPR</td>
<td>07 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://text.soe.ucsc.edu" target=_blank>pubsBingBlat (a run of hg19-BLAT against all webpages on the internet indexed by bing.com)</a></td>
<td>23 Jan 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=ce10&g=blastHg18KG" target=_blank>Human Proteins</a></td>
<td>12 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
Initial ce10 release (using Washington University School of Medicine GSC and Sanger Institute WS220): Assembly, Gap, GC Percent, Ensembl Genes v65, RefSeq Genes, Other RefSeq, C. elegans mRNAs, Spliced ESTs, C. elegans ESTs, Other mRNAs, Conservation, RepeatMasker, Interrupted Repeats, Simple Repeats</td>
<td>05 Jan 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=ce6></A>C. elegans May 2008 (WS190/ce6) (ce6, Washington University School of Medicine GSC and Sanger Institute WS190)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
BlastTab tables rebuild</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
liftOver file: ce6 to ce4</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=ce6&g=algBinding" target=_blank>ALG-1 Argonaute binding sites</a></td>
<td>19 Jan 2010</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter and UCSC Genes otherOrgs links to yeast sacCer2</td>
<td>02 Nov 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>29 Jan 2009</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Nucleosome, MNase Coverage, NSome Coverage, Adj NSome Covrg, NSome Stringency, 25mers (lifted from ce4/WS170)</td>
<td>10 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes (version 50)</td>
<td>22 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial ce6 release (using Washington University School of Medicine GSC and Sanger Institute WS190): Assembly; Gap; GC Percent; Short Match; Restr Enzymes; WormBase Genes; WormBase RNAs; WormBase PseudoGenes; RefSeq Genes; non-C. elegans RefSeq Genes; C. elegans mRNAs; Spliced ESTs; C. elegans ESTs; non-C. elegans mRNAs; Chains & Nets: C. remanei, C. briggsae, C. brenneri, C. japonica, P. pacificus; RepeatMasker; Interrupted Repeats; Simple Repeats; Self Chain. Downloads: bigZips, chromosomes, database, liftOvers, comparatives.</td>
<td>30 Jul 2008</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=ce4></A>C. elegans Jan. 2007 (WS170/ce4) (ce4, Washington University School of Medicine GSC and Sanger Institute WS170)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
TROMER Transcriptome database, SIB Alt-Splicing</td>
<td>29 May 2008</td>
</tr>
<TR valign=top><TD align=left>
25mers repeat annotation</td>
<td>06 May 2008</td>
</tr>
<TR valign=top><TD align=left>
Adjusted nucleosome coverage, Nucleosome Positioning Stringency from SOLiD core alignments, Coverage of nucleosome predictions from SOLiD core alignments, Micrococcal nuclease control coverage, Nucleosome predictions from SOLiD core alignments</td>
<td>18 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Updated mouse column in Gene Sorter and in Orthologous Genes section of gene details pages to mm9 UCSC Genes</td>
<td>19 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Initial ce4 Release (using WormBase WS170): Assembly; Gap; GC Percent; Short Match;  Restr Enzymes; WormBase Genes; WormBase RNAs; WormBase PseudoGenes; RefSeq Genes; Non-C. elegans RefSeq; Human Proteins; C. elegans mRNAs; Spliced ESTs; C. elegans ESTs; Non-C. elegans mRNAs; Conservation; Most Conserved; Chains and Nets: P. pacificus, C. briggsae, C. remanei, C. brenneri; RepeatMasker; WM + SDust; Simple Repeats; Self Chain.  Downloads: bigZips, chromosomes, database, liftOvers, multiz, phastCons, comparatives.</td>
<td>03 Aug 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=ce2></A>C. elegans Mar. 2004 (WS120/ce2) (ce2, WormBase v. WS120)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Updated Drosophila column in Gene Sorter from dm2 to dm3 FlyBase genes</td>
<td>28 Aug 2007</td>
</tr>
<TR valign=top><TD align=left>
Update of hgBlastTab tables (to hg18) to coincide with UCSC Genes</td>
<td>06 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Rat Ortholog column of Gene Sorter and Known Gene details updated to rn4</td>
<td>20 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
MicroRNA target sites in 3' UTRs as predicted by PicTar</td>
<td>26 Apr 2006</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>24 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Conservation, Most Conserved</td>
<td>17 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter</td>
<td>04 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
WormBase Genes</td>
<td>26 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Self Chain</td>
<td>20 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial ce2 Release (using WormBase v. WS120): Assembly, Gap, GC Percent, WormBase Genes, WormBase Genefinder, RefSeq Genes, Twinscan, miRNA, Worm mRNAs, Spliced ESTs, Worm ESTs, Gene Bounds, Fugu Blat, Briggsae Waba, Briggsae Blastz, Briggsae Chain, Briggsae Net, Briggsae Aligns, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosome, annotations.</td>
<td>11 May 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=ci2></A>C. intestinalis Mar. 2005 (JGI 2.1/ci2) (ci2, JGI v2.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
CRISPR</td>
<td>07 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://text.soe.ucsc.edu" target=_blank>pubsBingBlat (a run of hg19-BLAT against all webpages on the internet indexed by bing.com)</a></td>
<td>23 Jan 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 55</td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>02 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update (version 50)</td>
<td>18 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial ci2 Release (using JGI v2.0, March 2005): Gap, GC Percent, Short Match, Restr. enzymes, RefSeq, Non-Ciona RefSeq, mRNAs, Spliced ESTs, C. intestinalis ESTs, Non-Ciona mRNAs, Repeat Masker, Simple Repeats. Downloads: bigZips, database.</td>
<td>09 Sep 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=ci1></A>C. intestinalis Dec. 2002 (JGI 1.0/ci1) (ci1, JGI v1.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
RefSeq Genes</td>
<td>21 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins (data update)</td>
<td>17 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial ci1 release (using JGI v1.0): Gap, GC Percent, Non-Ciona RefSeq Genes, SNAP Genes, JGI Genes, Ciona mRNAs, Spliced ESTs, Ciona ESTs, NonCiona mRNAs, C. savignyi Blat, Human Proteins, RepeatMasker, Simple Repeats

Downloads: bigZips, database</td>
<td>15 Jul 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=caeJap1></A>C. japonica Mar. 2008 (WUGSC 3.0.2/caeJap1) (caeJap1, Washington University School of Medicine GSC C. remanei 3.0.2)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
C. elegans Proteins WS190 Mapped by Chained tBLASTn</td>
<td>18 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial caeJap1 release (using Washington University School of Medicine GSC C. remanei 3.0.2): Assembly, Contigs, Gap, GC Percent, Quality Scores, Short Match, Restr Enzymes, Non-C. japonica RefSeq, Spliced ESTs, Non-C. japonica mRNAs, WM + SDust, Simple Repeats, Chains/Nets to C. elegans. Downloads: bigZips, chromsomes, database.</td>
<td>27 Jun 2008</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=caeRem3></A>C. remanei May 2007 (WUGSC 15.0.1/caeRem3) (caeRem3, Washington University School of Medicine GSC C. remanei 15.0.1)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
C. elegans Chain/Net</td>
<td>06 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
C. elegans Proteins WS190 Mapped by Chained tBLASTn</td>
<td>18 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial caeRem3 release (using WUSTL C. remanei 15.0.1): Assembly, Contigs, Gap, GC Percent, Quality Scores, Short Match, Restr Enzymes, Non-C. remanei RefSeq, C. remanei mRNAs, Spliced ESTs, C. remanei ESTs, Non-C. ramenei mRNAs, WM + SDust, Simple Repeats. Downloads: bigZips, chromsomes, database.</td>
<td>20 Jun 2008</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=caeRem2></A>C. remanei Mar. 2006 (WUGSC 1.0/caeRem2) (caeRem2, Washington University School of Medicine GSC C. remanei 1.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial caeRem2 release (using WUSTL C. remanei 1.0): Assembly; Contigs; Gap; GC Percent; Short Match; Restr Enzymes; Non-C. ramenei RefSeq; C. elegans Proteins; C. remanei mRNAs; Spliced ESTs; C. remanei ESTs; Non-C. ramenei mRNAs; Chains and Nets: P. pacificus, C. elegans, C. brenneri, C. briggsae; RepeatMasker; WM + SDust; Simple Repeats.  Downloads: bigZips, chromosomes, database, liftOvers, comparatives.</td>
<td>03 Aug 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=felCat8></A>Cat Nov. 2014 (ICGSC Felis_catus_8.0/felCat8) (felCat8, International Cat Genome Sequencing Consortium)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Initial felCat8 release (using International Cat Genome Sequencing Consortium (GCA_000181335.3)): Assembly, Gap, GC Percent, INSDC, Genscan Genes, Augustus, RefSeq Genes, Other RefSeq, Cat mRNAs, Cat ESTs, Spliced ESTs, CpG Islands, Dog Chain/Net, Human Chain/Net, Simple Repeats, Interrupted Rpts, WM + SDust, RepeatMasker, Microsatellite</td>
<td>21 Dec 2015</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=felCat5></A>Cat Sep. 2011 (ICGSC Felis_catus 6.2/felCat5) (felCat5, International Cat Genome Sequencing Consortium Felis_catus-6.2 (NCBI project 16726, accession AANG00000000.2, GCA_000181335.2))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>20 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=felCat5&g=chainNetHg19" target=_blank>Human Chain/Net</a></td>
<td>03 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial felCat5 release (using International Cat Genome Sequencing Consortium Felis_catus-6.2 (NCBI project 16726, accession AANG00000000.2, GCA_000181335.2)): Assembly, Gap, GC Percent, Short Match, Restr Enzymes, RefSeq Genes, Other RefSeq, Genscan Genes, Cat mRNAs, Spliced ESTs, Cat ESTs, Other mRNAs, CpG Islands, Mouse Chain/Net, RepeatMasker, Interrupted Rpts, WM + SDust, Simple Repeats.</td>
<td>03 Dec 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=felCat4></A>Cat Dec. 2008 (NHGRI/GTB V17e/felCat4) (felCat4, NHGRI/Genome Technology Branch (NCBI project 10703, accession ACBE0100000))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
N-SCAN gene predictions from WUStL</td>
<td>05 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
Initial felCat4 release (using NHGRI/Genome Technology Branch (NCBI project 10703, accession ACBE0100000)): Assembly, Scaffolds, Gap, GC Percent, Short Match, Restr Enzymes, RefSeq Genes, Other RefSeq, Cat mRNAs, Spliced ESTs, Cat ESTs, Other mRNAs, CpG Islands, Opossum Chain/Net, Dog Chain/Net, Mouse Chain/Net, Panda Chain/Net, Human Chain/Net, RepeatMasker, Interrupted Rpts, Simple Repeats. Downloads: bigZips, chromosomes, database.
</td>
<td>18 Aug 2010</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=felCat3></A>Cat Mar. 2006 (Broad/felCat3) (felCat3, Broad Release 3)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=felCat3&g=ensGene" target=_blank>Ensembl gene update, version 63</a></td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=felCat3&g=ensGene" target=_blank>Ensembl gene update, version 62</a></td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=felCat3&g=ensGene" target=_blank>Ensembl gene update, version 61</a></td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Chains and Nets for non-primate mammal species vs Human</a></td>
<td>16 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 55</td>
<td>07 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>28 Jan 2009</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update (version 50) and Ensembl Gene Scaffolds track update</td>
<td>18 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes, version 49; Ensembl Gene Scaffold</td>
<td>18 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Most Conserved</td>
<td>20 Jun 2007</td>
</tr>
<TR valign=top><TD align=left>
Conservation (cat, dog, mouse, human)</td>
<td>20 Jun 2007</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN gene predictions</td>
<td>13 Mar 2007</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>23 Feb 2007</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>16 Feb 2007</td>
</tr>
<TR valign=top><TD align=left>
Initial felCat3 release (using Broad Release 3): Assembly; Gap; GC Percent; Quality Scores; Short Match; Restr Enzymes; RefSeq Genes; Other RefSeq; Genscan Genes; Cat mRNAs; Spliced ESTs; Cat ESTs; Other mRNAs; Cpg Islands; Chains & Nets: Human, Mouse, Dog; RepeatMasker; Simple Repeats; Microsatellite. Downloads: bigZips, database, liftOvers, comparatives.</td>
<td>22 Jan 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=galGal5></A>Chicken Dec 2015 (Gallus_gallus-5.0/galGal5) (galGal5, International Chicken Genome Consortium)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Initial galGal5 release (using International Chicken Genome Consortium (ICGC)(GCF_000002315.4)): Assembly, Gap, GC Percent, GRC Incident, INSDC, RefSeq Acc, Agustus Genes, Genscan Genes, Other RefSeq, RefSeq Genes, Chicken ESTs, Chicken mRNAs, Other mRNAs, Spliced ESTs, CpG Islands, Mouse Chain/Net, Rat Chain/Net, Human Chain/Net, Zebrafish Chain/Net, Interrupted Repeats, Microsatellite, Repeat Masker, Simple Repeats, WM + SDust.</td>
<td>31 Aug 2016</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=galGal4></A>Chicken Nov. 2011 (ICGSC Gallus_gallus-4.0/galGal4) (galGal4, ICGSC Gallus_gallus-4.0 (GCA_000002315.2))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl genes - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Animal Quantitative Trait Loci from animalQTLdb</td>
<td>27 Jun 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=galGal4&g=chainNetTaeGut2" target=_blank>Zebra finch Chain/Net</a></td>
<td>08 May 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=galGal4&g=snp138" target=_blank>Common SNPs(138), Mult. SNPs(138), All SNPs(138)</a></td>
<td>06 Mar 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://text.soe.ucsc.edu" target=_blank>pubsBingBlat (a run of hg19-BLAT against all webpages on the internet indexed by bing.com)</a></td>
<td>23 Jan 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v74</td>
<td>23 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=galGal4&g=ensGene" target=_blank>Ensembl Genes v71</a></td>
<td>11 Jun 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=galGal4&g=chainNetPetMar2" target=_blank>Lamprey Chain/Net</a></td>
<td>08 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
Medium ground finch Chain/Net</td>
<td>17 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
Initial galGal4 release (using ICGSC Gallus_gallus-4.0 (GCA_000002315.2)): Assembly, Gap, GC Percent, Genscan Genes, CpG Islands, RepeatMasker, Interrupted Repeats, Simple Repeats, WM + SDust, RefSeq Genes, Other RefSeq, Chicken mRNAs, Spliced ESTs, Chicken ESTs, Other mRNAs, Mouse Chain/Net</td>
<td>11 Jun 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=galGal3></A>Chicken May 2006 (WUGSC 2.1/galGal3) (galGal3, Chicken Genome Sequencing Consortium May 2006 release)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=galGal3&g=chainNetXenTro3" target=_blank>X. tropicalis Chain/Net</a></td>
<td>15 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=galGal3&g=chainNetMelGal1" target=_blank>Turkey Chain/Net</a></td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=galGal3&g=chainNetAnoCar2" target=_blank>Lizard Chain/Net</a></td>
<td>12 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Chimp (Oct. 2010 (CGSC 2.1.3/panTro3)), Chain and Net Alignments  </td>
<td>27 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Alignments of California Condor mRNA 454 sequences to chicken</td>
<td>12 Jan 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
Qauality Scores (update fixes omission of chrUn_random scores)</td>
<td>10 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
Chain and Net for Horse/equCab2 on Chicken</td>
<td>09 Nov 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Vertebrate genomes, Chain and Net alignment tracks (primary release)</a></td>
<td>29 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Opossum Chain and Net</td>
<td>25 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 55</td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
Zebra finch Chain/Net/LiftOver (taeGut1)</td>
<td>15 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN gene predictions using Zebra Finch informant from Brent Lab, WUStL</td>
<td>06 Apr 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>02 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Quality Scores</td>
<td>26 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update (version 50)</td>
<td>18 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Guinea Pig Chain and Net</td>
<td>11 Jun 2008</td>
</tr>
<TR valign=top><TD align=left>
Lamprey Chain/Net/LiftOver (petMar1)</td>
<td>10 Jun 2008</td>
</tr>
<TR valign=top><TD align=left>
Lancelet Chain/Net/LiftOver (braFlo1)</td>
<td>10 Jun 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl v49 gene predictions</td>
<td>15 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Mouse (Jul. 2007/mm9) Chained Alignments
Mouse (Jul. 2007/mm9) Alignment Net</td>
<td>19 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Chains Nets and liftOvers to fr2</td>
<td>11 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Fugu (Aug. 2002/fr1) Chained Alignments</td>
<td>17 Mar 2007</td>
</tr>
<TR valign=top><TD align=left>
Fugu (Aug. 2002/fr1) Alignment Net</td>
<td>17 Mar 2007</td>
</tr>
<TR valign=top><TD align=left>
Initial galGal3 release (using Chicken Genome Sequencing Consortium May 2006 release):
Assembly; Gap; GC Percent; BAC End Pairs; RefSeq Genes; Non-Chicken RefSeq Genes, GenScan Genes; Human Proteins; Chicken mRNAs; Spliced ESTs, Chicken ESTs, Non-Chicken mRNAs; CpG Islands; Conservation; Chains & Nets: Human, Mouse, Rat, Opossum, x. Tropicalis, Zebrafish; Repeat Masker; Simple Repeats; Microsatellite.

Downloads: bigZips, chromosomes, database, liftover files.  </td>
<td>04 Aug 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=galGal2></A>Chicken Feb. 2004 (WUGSC 1.0/galGal2) (galGal2, Chicken Genome Sequencing Consortium Feb. 2004 release)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
liftOver files for galGal2 to galGal3</td>
<td>24 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
Chimp (panTro2) Chains, Net and LiftOvers</td>
<td>31 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes (update)</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated with new predicitions from IMIM, using cow as well as human.</td>
<td>11 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated galGal2 -> mm6 liftOver (Downloads, Batch Coordinate Conversion utility)

</td>
<td>06 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Chain/Net Alignments (Feb. 2004 tetNig1)</td>
<td>12 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish Chain/Net Alignments (June 2004 danRer2)</td>
<td>12 May 2005</td>
</tr>
<TR valign=top><TD align=left>
X. tropicalis Chain/Net Alignments (Oct. 2004 xenTro1)</td>
<td>12 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Isochores</td>
<td>06 May 2005</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
conservation tracks (Conservation, Most Conserved)</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Human Chain, Human Net (updated to May 2004 assembly)</td>
<td>24 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Updated SGP Genes.</td>
<td>29 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
cpgIslands (data update) replace WUSTL cpgIslands with calculations by Andy Law, Roslin Institute</td>
<td>29 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>14 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Ecores</td>
<td>20 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Contamination (alignments of Maize mitochondrion, E. Coli and vector sequence to WGS assembly)</td>
<td>03 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
BGI SNPs, BGI Genes, BGI Coverage (Beijing Genomics Institute SNPs, gene annotations and read coverage locations) </td>
<td>02 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>21 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>18 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>17 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
RNA Genes</td>
<td>30 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
CpG Island (updated)</td>
<td>06 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes (updated)</td>
<td>17 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs, Genscan Genes, Geneid Genes, UniGene, CpG Island, Fugu Blat, GC Percent, Quality Scores, Human Chain, Human Net, Human Synteny, Self Chain, Self Net</td>
<td>10 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial galGal2 Release (using the Feb. 2004 assembly): Supercontigs, Assembly, Gap, Chicken mRNAs, Spliced ESTs, Chicken ESTs, NonChicken mRNAs, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosome, database.</td>
<td>01 Mar 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=panTro5></A>Chimp May 2016 (Pan_tro 3.0/panTro5) (panTro5, Chimpanzee Sequencing and Analysis Consortium)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href="https://genome.ucsc.edu/cgi-bin/hgGateway?db=panTro5" target=_blank>Initial panTro5 release (using Chimpanzee Sequencing and Analysis Consortium):  panTro5Initial panTro5 release (using Chimpanzee Sequencing and Analysis Consortium GCF_000001515.7: Assembly, Gap, GC Percent, Augustus Genes, Genscan Genes, RefSeq Genes, Other RefSeq, Chimp mRNAs, CpG Islands, Human Chain/Net, Mouse Chain/Net, Gorilla Chain/NetInterrupted Repeats, Microsatellite, RepeatMasker, Simple Repeats, WM + SDust.</a></td>
<td>31 Oct 2016</td>
</tr>
<TR valign=top><TD align=left>
Gorilla Chain/Net</td>
<td>28 Oct 2016</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=panTro4></A>Chimp Feb. 2011 (CSAC 2.1.4/panTro4) (panTro4, CSAC Pan_troglodytes-2.1.4 (GCA_000001515.4))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Gorilla Chain/Net</td>
<td>28 Oct 2016</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>20 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
Tarsier Chain/Net</td>
<td>22 May 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=panTro4&g=chainNetHg19" target=_blank>Human Chain/Net</a></td>
<td>21 Jun 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
Rhesus Chain/Net</td>
<td>03 Dec 2012</td>
</tr>
<TR valign=top><TD align=left>
Initial panTro4 release (using CSAC Pan_troglodytes-2.1.4 (GCA_000001515.4)): Assembly, Gap, GC Percent, RefSeq Genes, Other RefSeq, Ensembl Genes, Genscan Genes, Chimp mRNAs, Spliced ESTs, Chimp ESTs, Other mRNAS, CpG Islands, Rhesus Chain/Net, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>21 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=panTro3></A>Chimp Oct. 2010 (CGSC 2.1.3/panTro3) (panTro3, CGSC 2.1.3 (GCA_000001515.3))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=panTro3&g=chainGorGor3" target=_blank>Gorilla Chain/Net</a></td>
<td>28 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=panTro3&g=chainNetMonDom5" target=_blank>Opossum Chain/Net</a></td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=panTro3&g=cons12way" target=_blank>Multiz Alignment & Conservation (12 Species)</a></td>
<td>22 Aug 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=panTro3&g=nscanGene" target=_blank>N-SCAN gene predictions</a></td>
<td>20 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
panTro3 to panTro2 liftOver</td>
<td>15 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=panTro3&g=chainNetCanFam2" target=_blank>Dog (May 2005 (Broad/canFam2)), Chain and Net Alignments</a></td>
<td>24 May 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=panTro3&g=chainNetMm9" target=_blank>Mouse (July 2007 (NCBI37/mm9)), Chain and Net Alignments</a></td>
<td>20 May 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=panTro3&g=chainNetCalJac3" target=_blank>Marmoset (March 2009 (WUGSC 3.2/calJac3)), Chain and Net Alignments</a></td>
<td>19 May 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=panTro3&g=chainNetRheMac2" target=_blank>Rhesus (Jan. 2006 (MGSC Merged 1.0/rheMac2)), Chain and Net Alignments</a></td>
<td>19 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Horse (Sep. 2007 (Broad/equCab2)), Chain and Net Alignments</td>
<td>29 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
Orangutan (July 2007 (WUGSC 2.0.2/ponAbe2)), Chain and Net Alignments</td>
<td>28 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
Chicken (May 2006 (WUGSC 2.1/galGal3)), Chain and Net Alignments</td>
<td>27 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=panTro3&g=chainNetDanRer7" target=_blank>Zebrafish (Jul. 2010 (Zv9/danRer7)), Chain and Net Alignments</a></td>
<td>11 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Initial panTro3 release (using CGSC 2.1.3 (GCA_000001515.3)): Assembly, Scaffolds, Gap, GC Percent, Short Match, Restr Enzymes, PCR Results, RefSeq Genes, Other RefSeq, Chimp mRNAs, Spliced ESTs, Chimp ESTs, Other mRNAs, CpG Islands, Human Chain/Net, Repeat Masker, Interrupted Rpts, Simple Repeats.</td>
<td>01 Mar 2011</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=panTro2></A>Chimp Mar. 2006 (CGSC 2.1/panTro2) (panTro2, Washington University Build 2 Version 1)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=panro2&g=bamSLDenisova" target=_blank>Denisova Sequence Reads</a></td>
<td>30 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Marmoset Chain/Net</td>
<td>03 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?g=ntSeqReads&db=panTro2" target=_blank>Neandertal Seq</a></td>
<td>07 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?g=ntSeqContigs&db=panTro2" target=_blank>Neandertal Cntgs</a></td>
<td>07 May 2010</td>
</tr>
<TR valign=top><TD align=left>
Remove special-case handling human mRNAs, ESTs and RefSeqs to treat them as native chimp transcripts.</td>
<td>28 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
Opossum Chain and Net</td>
<td>20 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 55</td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Primate chains and nets from/to human hg19</a></td>
<td>21 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
Human Chain/Net/LiftOver (hg19)</td>
<td>06 May 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>09 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update (version 50)</td>
<td>18 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Marmoset Chain/Net/LiftOver (calJac1)</td>
<td>24 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Mouse (Jul. 2007/mm9) Chained Alignments
Mouse (Jul. 2007/mm9) Alignment Net</td>
<td>19 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN chimp predictions using mouse (mm8) as the informant</td>
<td>01 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
Chicken (galGal2) Chains, Net and LiftOvers</td>
<td>31 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish (danRer4) Chains, Net and LiftOvers</td>
<td>31 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
Dog (canFam2) Chains, Net and LiftOvers</td>
<td>26 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>21 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
Initial panTro2 release (Build 2, Version 1, October 2005):  
Assembly; Gap; GC Percent; Quality Scores; RefSeq Genes; Non-Chimp/Human RefSeq Genes; Chimp/Human ESTs; Spliced ESTs; Chimp/Human mRNAs; Non-Chimp/Human mRNAs; CpG Islands; Chains & Nets: Rat, Mouse, Rhesus, Human; RepeatMasker;  Simple Repeats.

Downloads: bigZips, chromosomes, database, lift over files.</td>
<td>20 Jul 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=panTro1></A>Chimp Nov. 2003 (CGSC 1.1/panTro1) (panTro1, CGSC Build 1 Version 1 Arachne assembly)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Remove special-case handling human mRNAs, ESTs and RefSeqs to treat them as native chimp transcripts.</td>
<td>28 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
SNP build 125</td>
<td>29 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>22 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Blat</td>
<td>14 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>10 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Contigs</td>
<td>28 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>27 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Deletions (updated)</td>
<td>14 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes (chr1 updated)</td>
<td>07 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Geneid Genes</td>
<td>24 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Best, vsHg15 downloads</td>
<td>02 Mar 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial panTro1 Release (using the Build 1 Version 1 Arachne assembly dated 13 Nov. 2003): Scaffolds, Gap, GC Percent, Quality Scores, Human RefSeq, Chimp mRNAs, Spliced ESTs, Chimp ESTs, NonChimp mRNAs, NonChimp ESTs, CpG Islands, Human Chain, Human Net, Human Deletions, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosome, database.</td>
<td>23 Feb 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=criGri1></A>Chinese hamster Jul. 2013 (C_griseus_v1.0/criGri1) (criGri1, Beijing Genomics Institution-Shenzhen)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Initial criGri1 release (using Beijing Genomics Institution-Shenzhen): Assembly, Gap, GC Percent, INSDC, RefSeq Genes, Other RefSeq, Genscan Genes, Chinese hamster mRNAs, Spliced ESTs, Chinese hamster ESTs, CpG Islands, Human Chain/Net, RepeatMasker, Interrupted Repeats, Simple Repeats, WM + SDust</td>
<td>08 May 2014</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=latCha1></A>Coelacanth Aug. 2011 (Broad/latCha1) (latCha1, Broad LatCha1 (GCA_000225785.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl genes - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial latCha1 release (using Broad LatCha1 (GCA_000225785.1)): Assembly, Gap, GC Percent, Short Match, Restr Enzymes, Other RefSeq, Ensembl Genes, Genscan Genes, Coelacanth mRNAs, Other mRNAs, CpG Islands, Fugu Chain/Net, Mouse Chain/Net, RepeatMasker, Interrupted Rpts, WM + SDust, Simple Repeats</td>
<td>20 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=bosTau8></A>Cow Jun. 2014 (Bos_taurus_UMD_3.1.1/bosTau8) (bosTau8, University of Maryland Bos_taurus_UMD_3.1.1)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href="../cgi-bin/hgTrackUi?db=bosTau8&g=snp148" target=_blank>All SNPs(148), Mult SNPs(148) Initial Release</a></td>
<td>23 Sep 2016</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>20 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
SGP Gene Predictions Using homology</td>
<td>27 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
Initial bosTau8 release (using University of Maryland Bos_taurus_UMD_3.1.1): Assembly, Gap, GC Percent, INSDC, Restr Enzymes, Short Match, Genscan Genes, RefSeq Genes, Other RefSeq, Cow mRNAs, Spliced ESTs, Cow ESTs, CpG Islands, Mouse Chain/Net, Human Chain/Net, RepeatMasker, Interrupted Repeats, Simple Repeats, WM + SDust</td>
<td>02 Jan 2015</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=bosTau7></A>Cow Oct. 2011 (Baylor Btau_4.6.1/bosTau7) (bosTau7, Btau_4.6.1 (NCBI project 12555, accession GCA_000003205.4))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
Animal Quantitative Trait Loci from animalQTLdb</td>
<td>27 Jun 2014</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=bosTau7&g=snp138" target=_blank>Common SNPs(138), Mult. SNPs(138), All SNPs(138)</a></td>
<td>28 Feb 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://text.soe.ucsc.edu" target=_blank>pubsBingBlat (a run of hg19-BLAT against all webpages on the internet indexed by bing.com)</a></td>
<td>23 Jan 2014</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=bosTau7&g=chainNetCerSim1" target=_blank>White rhinoceros Chain/Net</a></td>
<td>24 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
Dolphin Chain/Net</td>
<td>03 Dec 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=bosTau7&g=chainNetVicPac1" target=_blank>Alpaca Chain/Net</a></td>
<td>16 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
Initial bosTau7 release (using Btau_4.6.1 (NCBI project 12555, accession GCA_000003205.4)): Assembly, Gap, GC Percent, Scaffolds, Genscan Genes, CpG Islands, RepeatMasker, Interrupted Repeats, Simple Repeats, WM + SDust, RefSeq Genes, Other RefSeq, Cow mRNAs, Spliced ESTs, Cow ESTs, Other mRNAs, Dog Chain/Net, Human Chain/Net, Mouse Chain/Net, Opossum Chain/Net, Platypus Chain/Net, Rat Chain/Net, Pig Chain/Net</td>
<td>11 Jun 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=bosTau6></A>Cow Nov. 2009 (Bos_taurus_UMD_3.1/bosTau6) (bosTau6, UMD_3.1 (NCBI project 10708, accession GCA_000003055.3))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=bosTau6&g=numtSeq2012" target=_blank>NumtS Sequence</a></td>
<td>19 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
cytoBandIdeo (for navigation)</td>
<td>19 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial bosTau6 release (using UMD_3.1 (NCBI project 10708, accession GCA_000003055.3)): Assembly, Gap, GC Percent, CpG Islands. Repeat Masker, Interrupted Repeats, Simple Repeats, Scaffolds, RefSeq Genes, Other RefSeq, Cow mRNAs, Spliced ESTs, Cow ESTs, Other mRNAs, Ensembl Genes, Pig Chain/Net, Platypus Chain/Net, Opossum Chain/Net, Dog Chain/Net, Mouse Chain/Net, Human Chain/Net, Rat Chain/Net.</td>
<td>02 Nov 2011</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=bosTau4></A>Cow Oct. 2007 (Baylor 4.0/bosTau4) (bosTau4, Baylor Release 4.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Sheep Chain/Net</td>
<td>19 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
LiftOver files for bosTau4 (Baylor 4.0) to/from bosTau5 (Baylor Btau_4.2)</td>
<td>11 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Pig Chain/Net</td>
<td>12 May 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Chains and Nets for non-primate mammal species vs Human</a></td>
<td>16 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 55</td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI Elnitski Bi-directional Promoters</td>
<td>30 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>09 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Converted MGC track to PSL (to allow for alignment visualization)</td>
<td>22 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes (version 50)</td>
<td>15 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN gene predictions from WUStL</td>
<td>23 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial Bos taurus release (bosTau4; using Baylor Release 4.0). Assembly, Gap, GC Percent, Quality Scores, Short Match, Restr. Enzymes, RefSeq Genes, MGC Genes, Genscan Genes, Human Proteins, Cow mRNAs, Spliced ESTs, Cow ESTs, Non-Cow mRNAs, CpG Islands, Repeat Masker, Interrupted Repeats, Simple Repeats, 5-way Conservation and Most Conserved, Nets & Chains: Platypus, Mouse, Dog, Human. Downloads: bigZips, chromosomes, database.</td>
<td>07 Apr 2008</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=bosTau3></A>Cow Aug. 2006 (Baylor 3.1/bosTau3) (bosTau3, Baylor Release 3.1)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Converted MGC track to PSL (to allow for alignment visualization)</td>
<td>22 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v49</td>
<td>23 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
liftOver file to bosTau4</td>
<td>08 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions (data reload)</td>
<td>01 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Mouse (Jul. 2007/mm9) Chained Alignments
Mouse (Jul. 2007/mm9) Alignment Net</td>
<td>16 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Initial Bos taurus release (bosTau3; using Baylor Release 3.1). Assembly, Gap, GC Percent, Quality Scores, Short Match, Restr. Enzymes, RefSeq Genes, MGC Genes, Genscan Genes, Human Proteins, Cow mRNAs, Spliced ESTs, Cow ESTs, Non-Cow mRNAs, CpG Islands, Repeat Masker, Interrupted Repeats, Simple Repeats, SNPs, Nets & Chains: Rat, Mouse, Human. Downloads: bigZips, chromosomes, database</td>
<td>27 Aug 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=bosTau2></A>Cow Mar. 2005 (Baylor 2.0/bosTau2) (bosTau2, Baylor Btau_2.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Converted MGC track to PSL (to allow for alignment visualization)</td>
<td>22 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins track (updated from hg17 to hg18)</td>
<td>27 Feb 2007</td>
</tr>
<TR valign=top><TD align=left>
Added NCBI Clone Validation information to details of MGC genes</td>
<td>12 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
MGC Genes</td>
<td>05 Sep 2006</td>
</tr>
<TR valign=top><TD align=left>
Bos taurus (bosTau2, March 2005) Update Release: Assembly, Gap, Contig Orient, GC Percent, Quality Scores, Short Match, Restr. enzymes, Human Proteins, Other RefSeq, Genscan Genes, Cow mRNAs, Spliced ESTs, Cow ESTs, CpG Islands, Repeat Masker, Simple Repeats. Downloads: bigZips, chromosomes, database.</td>
<td>04 Oct 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=macFas5></A>Crab-eating macaque Jun. 2013 (Macaca_fascicularis_5.0/macFas5) (macFas5, Washington University (WashU) GCA_000364345.1)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Initial macFas5 release (using Washington University (WashU) GCA_000364345.1): Assembly, Gap, GC Percent, INSDC, Augustus, Genscan Genes, NCBI RefSeq, Other RefSeq, Crab-eating macaque ESTs, Crab-eating macaque mRNAs, Spliced ESTs, CpG Islands, Baboon Chain/Net, Tarsier Chain/Net, Human Chain/Net (hg19), Human Chain/Net (hg38), Microsatellite, RepeatMasker, Simple Repeats, WM + SDust</td>
<td>07 Apr 2016</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=droAna2></A>D. ananassae Aug. 2005 (Agencourt prelim/droAna2) (droAna2, Agencourt 1 August 2005)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Drosophila ananassae (Agencourt Bioscience Corporation, August 2005) Initial Release -- Gap, GC Percent, Other Refseq, Genscan, D. anan. mRNA, Other mRNAs, D. mel. proteins, D. mel. Chain/Net, RepeatMasker, Simple Repeats. Downloads: bigZips, database, vsDm2.</td>
<td>07 Oct 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=droAna1></A>D. ananassae July 2004 (TIGR/droAna1) (droAna1, TIGR 15 July 2004)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
D. melanogaster Chain/Net/LiftOver (dm2 update)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
D. ananassae Browser Initial Release (TIGR 15 July 2004): Gap, GC Percent, Non-D.anan. Refseq, Genscan, D.anan. mRNA, Non D.anan. mRNAs, D. mel. proteins, D. mel. Chain/Net, RepeatMasker, Simple Repeats.  Downloads bigZips, database, vsDm1.</td>
<td>15 Dec 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=droEre1></A>D. erecta Aug. 2005 (Agencourt prelim/droEre1) (droEre1, Agencourt 1 Aug 2005)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Drosophila erecta (droEre1, August 2005) Initial Release: Gap, Short Match, Restr. enzymes, RepeatMasker, Simple Repeats, GC Percent, D. ere. mRNAs, Non-D. ere. mRNAs, Non-D. ere. RefSeq, Genscan Genes, D. mel. (dm2) Chain/Net, D.mel. (dm2) Proteins. Downloads: bigZips, database, vsDm2.</td>
<td>20 Sep 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=droGri1></A>D. grimshawi Aug. 2005 (Agencourt prelim/droGri1) (droGri1, Agencourt 1 Aug 2005)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Drosophila grimshawi (droGri1, August 2005) Initial Release: Gap, GC Percent, Short Match, Restr. enzymes, Other RefSeq, Genscan Genes, Other mRNAs, D. melanogaster (dm2) Proteins, D. melanogaster (dm2) Chain/Net, RepeatMasker, Simple Repeats. Downloads: bigZips, database, vsDm2.</td>
<td>11 Oct 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=dm6></A>D. melanogaster Aug. 2014 (BDGP Release 6 + ISO1 MT/dm6) (dm6, The FlyBase Consortium/Berkeley Drosophila Genome Project/Celera Genomics)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
CRISPR</td>
<td>07 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../cgi-bin/hgTrackUi?db=dm6&g=refGenePfam" target=_blank>Pfam Domains in Refseq Genes</a></td>
<td>21 Sep 2016</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Initial dm6 release (using The FlyBase Consortium/Berkeley Drosophila Genome Project/Celera Genomics (GCA_000001215.4)): Chromosome Band, Assembly, Gap, GC Percent, INSDC, RefSeq Genes, Genscan Genes, Other RefSeq, Spliced ESTs, D. melanogaster ESTs, D. melanogaster mRNAs, Other mRNAs, CpG Islands, 27-way Conservation, D. simulans Chain/Net, D. sechellia Chain/Net, D. persimilis Chain/Net, A. gambiae Chain/Net, Interrupted Repeats, Simple Repeats, WM + SDust, RepeatMasker</td>
<td>11 Dec 2014</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=dm3></A>D. melanogaster Apr. 2006 (BDGP R5/dm3) (dm3, BDGP Release 5)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://text.soe.ucsc.edu" target=_blank>pubsBingBlat (a run of hg19-BLAT against all webpages on the internet indexed by bing.com)</a></td>
<td>23 Jan 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=dm3&g=blastHg18KG" target=_blank>Human Proteins</a></td>
<td>12 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=dm3&g=bdtnpChipper" target=_blank>BDTNP ChIP/chip</a></td>
<td>16 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=dm3&g=bdtnpDnase" target=_blank>BDTNP DNase Accs</a></td>
<td>16 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
BlastTab tables rebuild</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter and UCSC Genes otherOrgs links to yeast sacCer2</td>
<td>02 Nov 2009</td>
</tr>
<TR valign=top><TD align=left>
Updated to FlyBase version 5.12 annotations.</td>
<td>30 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Transcription start upstream MAF files based on FlyBase 5.12 Genes.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
multiz15way upstream*.maf.gz files</td>
<td>27 May 2008</td>
</tr>
<TR valign=top><TD align=left>
Gene Disruption Project P-element and Minos Insertion Locations</td>
<td>27 May 2008</td>
</tr>
<TR valign=top><TD align=left>
ORegAnno track update</td>
<td>07 Nov 2007</td>
</tr>
<TR valign=top><TD align=left>
EvoFold predictions of RNA secondary structure</td>
<td>06 Nov 2007</td>
</tr>
<TR valign=top><TD align=left>
Updated mouse column in Gene Sorter and in Orthologous Genes section of gene details pages to mm9 UCSC Genes</td>
<td>19 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Added CONTRAST gene predictions</td>
<td>17 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN PASA-EST gene predictions from WUStL</td>
<td>16 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Updated to FlyBase version 5.3 annotations.</td>
<td>10 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Initial dm3 release (using BDGP Release 5): Chromosome Bands, Chromosome Ideogram, Assembly, Gap, BAC Endpairs, GC Percent; RefSeq Genes, Flybase Genes, FB Noncoding; D.melanogaster mRNAs, xenoMrnas, Spliced ESTs, ESTs; Conservation, MostConserved, Chains & Nets: A.gambiae, D.pseudoobscura, D.persimilis, D.ananassae, D.erecta, D.yakuba, D.sechellia, D.simulans;  RepeatMasker, Simple Repeats; Downloads: bigZips, chromosomes, database, liftOvers, comparatives.</td>
<td>16 Jul 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=dm2></A>D. melanogaster Apr. 2004 (BDGP R4/dm2) (dm2, BDGP v. 4 / DHGP v. 3.2)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
LiftOver to: D. pseudoobscura (dp4), D. ananassae (droAna3), D. erecta (droEre2), D. grimshawi (droGri2), D. mojavensis (droMoj3), D. virilis (droVir3), D. willistoni (droWil1), D. yakuba (droYak2)</td>
<td>10 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
EvoFold predictions of RNA secondary structure</td>
<td>06 Nov 2007</td>
</tr>
<TR valign=top><TD align=left>
Chromosome Bands, Chromosome Ideogram</td>
<td>02 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ORegAnno track update</td>
<td>30 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Updated to FlyBase release 4.3.</td>
<td>20 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Update of hgBlastTab tables (to hg18) to coincide with UCSC Genes</td>
<td>06 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>26 Feb 2007</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix Drosophila Development Signal & Transfrags</td>
<td>11 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
Regulatory elements from Open Regulatory Annotation (ORegAnno)</td>
<td>05 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
Conservation, Most Conserved (15-way alignment)</td>
<td>12 Dec 2006</td>
</tr>
<TR valign=top><TD align=left>
Correction to CDS annotation of 36 transcripts in Fly Base Gene Track with start codon whose first base is the last base in an exon.</td>
<td>26 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
Rat Ortholog column of Gene Sorter and Known Gene details updated to rn4</td>
<td>20 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins </td>
<td>20 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Update to 4.2 FlyBase Noncoding Genes track to correct the omission of many miRNAs.  </td>
<td>15 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Added to Gene Sorter.</td>
<td>22 May 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated to FlyBase release 4.2 annotations.</td>
<td>22 May 2006</td>
</tr>
<TR valign=top><TD align=left>
MicroRNA target sites in 3' UTRs as predicted by PicTar</td>
<td>26 Apr 2006</td>
</tr>
<TR valign=top><TD align=left>
D. sechellis (droSec1) Chain/Net and Lift Over (Downloads, Batch Coordinate Conversion utility)</td>
<td>18 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
D. persimilis (droPer1) Chain/Net and Lift Over (Downloads, Batch Coordinate Conversion utility)</td>
<td>18 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
D. pseudo. (dp3) Chain/Net and Lift Over (Downloads, Batch Coordinate Conversion utility)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
A. mellifera (apiMel2) Chain/Net and Lift Over (Downloads, Batch Coordinate Conversion utility)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: droGri1, droMoj2, droAna2, droEre1, droSim1, droVir2, dp3 -> dm2</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
DroYak1 Chain/Net - regenerated with more sensitive blastz parameters</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
DroMoj1 Chain/Net - regenerated with more sensitive blastz parameters</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
DroVir1 Chain/Net - regenerated with more sensitive blastz parameters</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
AnoGam1 Chain/Net - regenerated with more sensitive blastz parameters</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: dm2 -> anoGam1, droGri1, droMoj1, droMoj2, droAna1, droAna2, droEre1, droSim1, driVir1, droVir2, droYak1, apiMel2, dp3</td>
<td>04 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>10 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN Gene Predictions</td>
<td>07 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
D. simulans (droSim1) Chain and Net</td>
<td>08 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
augustus gene predictions</td>
<td>07 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Conservation, Most Conserved (Rebuilt from new pairwise alignments with more sensitive blastz parameters; now includes D. simulans)</td>
<td>28 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Conservation, Most Conserved (re-release to correct omission of D. yakuba from multiple alignments)</td>
<td>11 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
FlyReg (Drosophila DNase I Footprint Database)</td>
<td>04 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
FlyBase Genes, FB Noncoding (updated FlyBase annotations from version 4.0 to 4.1)</td>
<td>28 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Initial D. melanogaster 2 release (BDGP Rel.4 euchromatic, DHGP Rel. 3.2 heterochromatic): Gap, GC Percent, Genscan, RefSeq Genes, Flybase, FB-Noncoding, Conservation, Most-Conserved,  mRNAs, xenoMrnas, Spliced ESTs, ESTs, {anoGam1, apiMel1, dp2, droAna1, droMoj1, droVir1, droYak1} Chain/Net, RepeatMasker, Simple Repeats.
Downloads: bigZips, database, comparatives.</td>
<td>14 Feb 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=dm1></A>D. melanogaster Jan. 2003 (BDGP R3/dm1) (dm1, BDGP v. 3)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
FlyBase Non-coding (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
FlyBase Genes (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
FlyReg</td>
<td>23 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>21 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (chrUn-based dp1 replaced with scaffold-based dp2 for D. pseudoobscura sequence; rescored with latest phastCons)</td>
<td>01 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Incremental update to BDGP Insertions.</td>
<td>27 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
D. pseudoobscura Chain, Net (updated to dp2)</td>
<td>01 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
BDGP Insertions (P element insertion locations from the BDGP Gene Disruption Project)</td>
<td>27 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
D. yakuba Chain/Net</td>
<td>12 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Anopheles Chain and Net</td>
<td>12 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (D. yakuba, D. pseudoob., A. gambiae)</td>
<td>11 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter</td>
<td>01 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
BDGP Non-coding</td>
<td>30 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Chromosome Bands (Cytology) from Flybase</td>
<td>30 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
BDGP Genes</td>
<td>30 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Anopheles Ecores</td>
<td>17 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial dm1 Release (using BDGP V3.1): Gap, GC Percent, RefSeq Genes, BDGP Genes, BDGP Non-coding, Genscan Genes, Fruitfly mRNAs, Spliced ESTs, Fruitfly ESTs, Fugu Blat, D. pseud. Chain, D. pseud. Net, D. pseud Aligns, RepeatMasker, Simple Repeats. Downloads: bigZips, database</td>
<td>30 Oct 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=droMoj2></A>D. mojavensis Aug. 2005 (Agencourt prelim/droMoj2) (droMoj2, Agencourt 1 August 2005)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Drosophila mojavensis (droMoj2, August 2005) Initial Release: Gap, Short Match, Restr. enzymes, RepeatMasker, Simple Repeats, GC Percent, D. moj. mRNAs, Non-D. moj. mRNAs, Non-D. moj. RefSeq, Genscan Genes, D. mel. (dm2) Chain/Net, D.mel. (dm2) Proteins. Downloads: bigZips, database, vsDm2.</td>
<td>20 Sep 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=droMoj1></A>D. mojavensis Aug. 2004 (Agencourt prelim/droMoj1) (droMoj1, Agencourt 11 Aug 2004)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
D. melanogaster Chain/Net/LiftOver (dm2 update)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
D.mel. Proteins (data update to dm2)</td>
<td>27 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
D. mojavensis Browser Initial Release (Agencourt 11 Aug 2004): Gap, GC Percent, Non-D.moj. Refseq, Genscan, D.moj. mRNA, Non D.moj. mRNAs, D. mel. proteins, D. mel. Chain/Net, RepeatMasker, Simple Repeats</td>
<td>15 Dec 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=droPer1></A>D. persimilis Oct. 2005 (Broad/droPer1) (droPer1, Broad 28 Oct. 2005)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
D. melanogaster Chain/Net</td>
<td>18 Dec 2014</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial droPer1 Release (using Broad Institute):  supporting tables, Assembly, GC Percent, Gap, Simple Repeats, RepeatMasker, D. melanogaster (dm2) chain/net, Genscan, Mapped Fly Proteins, support for Genbank, Other mRNAs, Other RefSeq.  Downloads: bigZips, database, liftOver, vsDm2.</td>
<td>06 Jan 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=dp3></A>D. pseudoobscura Nov. 2004 (FlyBase 1.03/dp3) (dp3, FlyBase Release 1.03)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
D. pseudoobscura Initial Release (dp3, based on FlyBase Release 1.03 November 2004): Gap, GC Percent, Short Match, Restr. enzymes, FlyBase, Non-D.pseudo. RefSeq, Twinscan, Genewise, Genscan Genes, D.pseudo. mRNAs, Spliced ESTs, D.pseudo. ESTs, Non-D.pseudo. mRNAs, D.mel. Proteins (dm1), D.mel Chain/Net (dm2), Repeat Masker, Simple Repeats, Downloads: bigZips, database, vsDm2.</td>
<td>24 Jun 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=dp2></A>D. pseudoobscura Aug. 2003 (Baylor freeze1/dp2) (dp2, Baylor HGSC Freeze 1)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial dp2 release (Baylor freeze 1 August 2003): GC Percent, Xeno RefSeq, Twinscan Genes, Genscan Genes, mRNAs, Xeno mRNAs, D. melanogaster, proteins, D. mel. Chain, D. mel. Net, RepeatMasker, Simple Repeats.
Downloads: bigZips, chromosomes, database</td>
<td>01 Sep 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=droSec1></A>D. sechellia Oct. 2005 (Broad/droSec1) (droSec1, Broad 28 Oct. 2005)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
D. melanogaster Chain/Net</td>
<td>18 Dec 2014</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial droSec1 Release (using Broad Institute):  supporting tables, Assembly, GC Percent, Gap, Simple Repeats, RepeatMasker, D. melanogaster (dm2) chain/net, Genscan, Mapped Fly Proteins, support for Genbank, Other mRNAs, Other RefSeq
Downloads: bigZips, database, liftOver, vsDm2</td>
<td>18 Jan 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=droSim1></A>D. simulans Apr. 2005 (WUGSC mosaic 1.0/droSim1) (droSim1, WUSTL version 1.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
D. melanogaster Chain/Net</td>
<td>18 Dec 2014</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial Drosophila simulans release (droSim1; Release 1.0 Apr. 2005 from WUSTL): Assembly, Gap, GC Percent, Short Match, Restr. enzymes, RepeatMasker, Simple Repeats, Non D. sim. RefSeq, Genscan Genes, D. sim mRnas, D. sim Spliced ESTs, D. sim ESTs, Non D. sim. mRnas, D. mel Proteins, D. mel Chain, D. mel Net.</td>
<td>10 Jun 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=droVir2></A>D. virilis Aug. 2005 (Agencourt prelim/droVir2) (droVir2, Agencourt 1 August 2005)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Reloaded Gap and loaded Assembly from assembly.links files included with the assemblies.  Previously, Gap was deduced from runs of N's in the sequence.</td>
<td>25 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Drosophila virilis (droVir2, Month 2005) Initial Release: Gap, Short Match, Restr. enzymes, RepeatMasker, Simple Repeats, GC Percent, D. vir. ESTs, Spliced Ests, D. vir. mRNAs, Non-D. vir. mRNAs, Non-D. vir. RefSeq, Genscan Genes, D. mel. (dm2) Chain/Net, D.mel. (dm2) Proteins. Downloads: bigZips, database, vsDm2.</td>
<td>20 Sep 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=droVir1></A>D. virilis July 2004 (Agencourt prelim/droVir1) (droVir1, Agencourt 12 July 2004)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
D. melanogaster Chain/Net/LiftOver (dm2 update)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
D. virilis Browser Initial Release (Agencourt 12 July 2004): Gap, GC Percent, Non-D. vir. Refseq, Genscan, D. vir. mRNA, Non-D. vir. mRNA, D. mel. proteins, D. mel. Chain/Net, RepeatMasker, Simple Repeats</td>
<td>15 Dec 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=droYak2></A>D. yakuba Nov. 2005 (WUGSC 7.1/droYak2) (droYak2, WUSTL version 2.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial Drosophila yakuba Release (droYak2, Nov. 2005 using WUSTL):  Assembly, Gap, GC Percent, Contamination, Short Match, Restr Enzymes, Other RefSeq, Genscan Genes, D. yakuba mRNAs, D. yakuba ESTs, Spliced Ests, Other mRNAs, D. melanogaster Proteins/Chain/Net, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosomes, database, liftOver, vsDm2.</td>
<td>11 Jan 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=droYak1></A>D. yakuba Apr. 2004 (WUGSC 1.0/droYak1) (droYak1, WUSTL version 1.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
D. melanogaster Chain/Net/LiftOver (dm2 update)</td>
<td>17 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Fly (dm1) Proteins (data update)</td>
<td>16 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
D. mel. Proteins (chained tblastn alignments of D. melanogaster proteins)</td>
<td>21 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
GC Percent</td>
<td>26 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial D. yakuba Release (version 1) - Assembly, Gap, Genscan Genes, D. yakuba mRNAs, Spliced ESTs, D. yakuba ESTs, Non-D. yakuba mRNAs, D. mel. Chain, D. mel. Net, RepeatMasker, Simple Repeats</td>
<td>23 Jul 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=canFam3></A>Dog Sep. 2011 (Broad CanFam3.1/canFam3) (canFam3, Broad CanFam3.1 (GCA_000002285.2))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
ORegAnno update</td>
<td>12 May 2016</td>
</tr>
<TR valign=top><TD align=left>
Cat Chain/Net</td>
<td>22 Dec 2015</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>20 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
Human (Dec. 2013 (GRCh38/hg38)) Chained Alignments</td>
<td>14 May 2015</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=canFam3&g=chainNetHg38" target=_blank>Human Chain/Net</a></td>
<td>06 Mar 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=canFam3&g=chainNetHg19" target=_blank>Human Chain/Net</a></td>
<td>31 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial canFam3 release (using Broad CanFam3.1 (GCA_000002285.2)): Assembly, Gap, GC Percent, Short Match, Restr Enzymes, Genscan Genes, CpG Islands, RepeatMasker, Interrupted Repeats, Simple Repeats, WM + SDust, RefSeq Genes, Other RefSeq, Dog mRNAs, Spliced ESTs, Dog ESTs, Other mRNAs, Mouse Chain/Net</td>
<td>25 Jun 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=canFam2></A>Dog May 2005 (Broad/canFam2) (canFam2, Broad Institute v. 2.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=canFam2&g=snp131" target=_blank>SNPs (131)</a></td>
<td>16 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=canFam2&g=chainNetBosTau7" target=_blank>Cow Chain/Net</a></td>
<td>11 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=canFam2&g=chainNetBosTau6" target=_blank>Cow Chain/Net</a></td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=canFam2&g=chainNetPanTro3" target=_blank>Chimp (Oct. 2010 (CGSC 2.1.3/panTro3)), Chain and Net Alignments</a></td>
<td>24 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Cat Chain/Net</td>
<td>17 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Marmoset Chain/Net</td>
<td>03 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Panda Chain/Net</td>
<td>16 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
Opossum Chain/Net and liftOver</td>
<td>24 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Chains and Nets for non-primate mammal species vs Human</a></td>
<td>16 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 55</td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>09 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update (version 50)</td>
<td>19 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Marmoset Chain/Net/LiftOver (calJac1)</td>
<td>24 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Chain track, Net track, and liftOver file to bosTau4</td>
<td>08 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Mouse (Jul. 2007/mm9) Chained Alignments
Mouse (Jul. 2007/mm9) Alignment Net</td>
<td>07 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
RefSeq Genes, Other RefSeq, Dog mRNAs, Spliced ESTs, Dog ESTs, Other mRNAs: reloaded tables to address problems caused by dog organism name change.</td>
<td>16 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN informant was incorrectly documented as mouse, human was actually used.</td>
<td>18 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>17 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
Chimp (panTro2) Chains, Net and LiftOvers</td>
<td>26 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
gene prediction peptide tables</td>
<td>24 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN gene predictions from WUStL</td>
<td>08 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
NCBI accessions for assembly contigs (used for Assembly track details)</td>
<td>05 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain/Net/LiftOver (mm7)</td>
<td>22 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
Dog (canFam2, May 2005, Broad Institute v2.0) Initial Release: Assembly, Gap, Quality Scores, GC Percent, Uncertified, BAC End Pairs, Short Match, Restr Enzymes, Human Proteins, Genscan Genes, Other RefSeq, Dog mRNAs, Other mRNAs, Dog ESTs, Spliced Ests, CpG Islands, Conservation, Most Conserved, Mouse Chain/Net (mm7), Rat Chain/Net (rn3), Human Chain/Net (hg17), Self Chain, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosomes, database, comparatives, liftOvers.
 </td>
<td>20 Dec 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=canFam1></A>Dog July 2004 (Broad/canFam1) (canFam1, Broad Institute v. 1.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Microsatellites</td>
<td>25 Feb 2008</td>
</tr>
<TR valign=top><TD align=left>
SNP build 125</td>
<td>24 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN gene predictions from Brent Lab at WUStL</td>
<td>21 Apr 2006</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>07 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs</td>
<td>02 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
hg17ToCanFam1 and canFam1ToHg17 liftOver (Downloads, Batch Coordinate Conversion utility)</td>
<td>05 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>15 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (data update)</td>
<td>06 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Chain, Human Net (chained and netted blastz alignments to May 2004 Human)</td>
<td>30 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (dog-human-mouse multiz alignments, scored by phastCons)</td>
<td>29 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain, Mouse Net (chained and netted blastz alignments to May 2004 Mouse)</td>
<td>29 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu BLAT (translated BLAT of Fugu scaffolds to Dog genome)</td>
<td>26 Jul 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial canFam1 release (using Broad Institute WGS v1.0): Assembly, Gap, GC Percent, Genscan Genes, Dog mRNAs, Spliced ESTs, Dog ESTs, NonDog mRNAs, RepeatMasker, Simple Repeats

 Downloads: bigZips, chromosomes, database</td>
<td>13 Jul 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=turTru2></A>Dolphin Oct. 2011 (Baylor Ttru_1.4/turTru2) (turTru2, Baylor College of Medicine Ttru_1.4 (NCBI project 20365, GCA_000151865.2, WGS ABRN02))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial turTru2 release (using Baylor College of Medicine Ttru_1.4 (NCBI project 20365, GCA_000151865.2, WGS ABRN02)): Assembly, Gap, GC Percent, Short Match, Restr Enzymes, Other RefSeq, Genscan Genes, CpG Islands, Cow Chain/Net, Mouse Chain/Net, RepeatMasker, Interrupted Rpts, WM + SDust, Simple Repeats.</td>
<td>03 Dec 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=eboVir3></A>Ebola virus Sierra Leone 2014 (G3683/KM034562.1/eboVir3) (eboVir3, West Africa 01 June 2014 EBOV/G3683/KM034562.1)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Sequences from the Lens PatSeq Patent Document Database</td>
<td>11 Apr 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=eboVir3&g=newSeqs" target=_blank>Recent related sequences from GenBank</a></td>
<td>18 Feb 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=eboVir3&g=iedbPred" target=_blank>IEDB Ebola HLA predictions</a></td>
<td>19 Dec 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=eboVir3&g=muPIT" target=_blank>muPIT - Mapping Genomic Positions on Protein Structures</a></td>
<td>30 Oct 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=eboVir3&g=strainCons160way" target=_blank>Ebola Virus PhastCons and PhyloP Conservation Scores</a></td>
<td>28 Oct 2014</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=loxAfr3></A>Elephant Jul. 2009 (Broad/loxAfr3) (loxAfr3, Broad Institute loxAfr3 (NCBI project 12569, AAGU03000000))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>20 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v74</td>
<td>23 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>02 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v65</td>
<td>22 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=loxAfr3&g=chainNetMonDom5" target=_blank>Opossum Chain/Net</a></td>
<td>21 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Initial loxAfr3 release (using Broad Institute loxAfr3 (NCBI project 12569, AAGU03000000)): Assembly, Gap, GC Percent, Quality Scores, RepeatMasker, Interrupted Rpts, Simple Repeats, Human Proteins, Other RefSeq, Elephant mRNAs, Other mRNAs, Human Chain and Net, Elephant Chain and Net, Mouse Chain and Net</td>
<td>24 Mar 2010</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=calMil1></A>Elephant shark Dec. 2013 (Callorhinchus_milii-6.1.3/calMil1) (calMil1, Institute of Molecular and Cell Biology, Singapore)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
Initial calMil1 release (using Institute of Molecular and Cell Biology, Singapore):  Assembly, Gap, GC Percent, INSDC, Restr Enzymes, Short Match, Genscan Genes, Other RefSeq, Elephant shark ESTs, Elephant shark mRNAs, Spliced ESTs, CpG Islands, RepeatMasker, Interrupted Rpts, Simple Repeats, WM + SDust</td>
<td>09 May 2014</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=musFur1></A>Ferret  Apr. 2011 (MusPutFur1.0/musFur1) (musFur1, Ferret Genome Sequencing Consortium)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial musFur1 release (using Ferret Genome Sequencing Consortium MusPutFur1.0 (GCA_000215625.1)): Assembly, Gap, GC Percent, Short Match, Restr Enzymes, Other RefSeq, Ensembl Genes, Genscan Genes, Ferret mRNAs, Spliced ESTs, Ferret ESTs, CpG Islands, RepeatMasker, Interrupted Rpts, WM + SDust, Simple Repeats.</td>
<td>13 Jun 2013</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=fr3></A>Fugu Oct. 2011 (FUGU5/fr3) (fr3, Fugu Genome Sequencing Consortium - FUGU5 (NCBI project 1434, GCA_000180615.2))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>02 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Coelacanth Chain/Net</td>
<td>20 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
Atlantic cod Chain/Net</td>
<td>20 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
Initial fr3/Fugu release (using Fugu Genome Sequencing Consortium - FUGU5 (NCBI project 1434, GCA_000180615.2)): Scaffolds, Gap, GC Percent, Short Match, Restr Enzymes, RefSeq Genes, Ensembl Genes (liftover from fr2), Fugu mRNAs, Spliced ESTs, Fugu ESTs, Other mRNAs, Conservation, oreNil1 Chain/Net, Tetraodon Chain/Net, Medaka Chain/Net, Stickleback Chain/Net, Zebradish Chain/Net, RepeatMasker, Interrupted Rpts, WM + SDust, Simple Repeats.</td>
<td>15 May 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=fr2></A>Fugu Oct. 2004 (JGI 4.0/fr2) (fr2, JGI V4.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl genes - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v74</td>
<td>23 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=fr2&g=chainNetDanRer7" target=_blank>Zebrafish Chain/Net</a></td>
<td>24 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Chain/Net and liftOver file</td>
<td>12 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Vertebrate genomes, Chain and Net alignment tracks (secondary release)</a></td>
<td>09 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 55</td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>18 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Medaka Chain and Net</td>
<td>19 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update (version 50)</td>
<td>18 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl v49 gene predictions</td>
<td>15 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Mouse (Jul. 2007/mm9) Chained Alignments
Mouse (Jul. 2007/mm9) Alignment Net</td>
<td>14 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Initial fr2 release (using JGI Oct 2004 build): Assembly; Gap; GC Percent; Ensembl Genes; Human Proteins; Repeat Masker; Simple Repeats; Window Masker; Fugu & non-Fugu mRNAs; Fugu ESTs; Spliced ESTs. Conservation; Most Conserved; Chains & Nets: Human, Chicken, Zebrafish, Stickleback, Medaka, Tetraodon.  Downloads: bigZips, chromosomes, database, liftOvers, multiz, phastCons.</td>
<td>11 Apr 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=fr1></A>Fugu Aug. 2002 (JGI 3.0/fr1) (fr1, JGI V3.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Chicken (May 2006/galGal3) Alignment Net</td>
<td>17 Mar 2007</td>
</tr>
<TR valign=top><TD align=left>
Chicken (May 2006/galGal3) Chained Alignments</td>
<td>17 Mar 2007</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish Chain/Net/LiftOver (danRer3)</td>
<td>21 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish (danRer2) Chain and Net</td>
<td>06 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Human Ecores, Mouse Ecores, Rat Ecores</td>
<td>07 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
Twinscan</td>
<td>29 Jan 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial fr1 Release (using JGI V3.0): Scaffolds, Gap, GC Percent, Ensembl Genes, Genscan Genes, Fugu mRNAs, Fugu ESTs, Non-Fugu mRNAs, CpG Islands, Human Blat, RepeatMasker, Simple Repeats. Downloads: bigZips, database</td>
<td>23 Sep 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=nomLeu3></A>Gibbon Oct. 2012 (GGSC Nleu3.0/nomLeu3) (nomLeu3, Gibbon Genome Sequencing Consortium )     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>20 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
Tarsier Chain/Net</td>
<td>22 May 2015</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial nomLeu3 release (using Gibbon Genome Sequencing Consortium ): see separate push queue nomLeu3: Human Chain, Human Net, Assembly, Gap, GC Percent, Other RefSeq, Genscan Genes, Other mRNAs, CpG Islands, RepeatMasker, Interrupted Rpts, WM + SDust, Simple Repeats, Human Chain/Net, Chain, Net
</td>
<td>25 May 2013</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=nomLeu2></A>Gibbon Jun. 2011 (GGSC Nleu1.1/nomLeu2) (nomLeu2, Gibbon Genome Sequencing Consortium Nleu1.1 (NCBI project 13975, GCA_000146795.2, WGS ADFV01))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Initial nomLeu2 release (using Gibbon Genome Sequencing Consortium Nleu1.1 (NCBI project 13975, GCA_000146795.2, WGS ADFV01)): Assembly, Gap, GC Percent, Other RefSeq, Genscan Genes, Other mRNAs, CpG Islands, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>16 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=nomLeu1></A>Gibbon Jan. 2010 (GGSC Nleu1.0/nomLeu1) (nomLeu1, GGSC (NCBI project 13975, accession GCA_000146795.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v74</td>
<td>23 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial nomLeu1 release (using GGSC (NCBI project 13975, accession GCA_000146795.1)): Assembly, Gap, GC Percent, RepeatMasker, Interrupted Repeats, Simple Repeats, Other RefSeq, Gibbon mRNAs, Other mRNAs, Ensembl Genes, Gorilla Chain/Net, Human Chain/Net</td>
<td>23 Nov 2011</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=gorGor5></A>Gorilla Mar. 2016 (GSMRT3/gorGor5) (gorGor5, UNIVERSITY OF WASHINGTON)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Chimp Chain/Net</td>
<td>31 Oct 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgGateway?db=gorGor5" target=_blank>Initial gorGor5 release (using UNIVERSITY OF WASHINGTON GCA_900006654.1): Assembly, Gap, GC Percent, Augustus Genes, Genscan Genes, RefSeq Genes, Other RefSeq, Gorilla mRNAs, CpG Islands, Human Chain/Net, Chimp Chain/Net, Interrupted Repeats, Microsatellite, RepeatMasker, Simple Repeats, WM + SDust</a></td>
<td>28 Oct 2016</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=gorGor4></A>Gorilla Dec 2014 (gorGor4.1/gorGor4) (gorGor4, Wellcome Trust Sanger Institute Dec 2014 (NCBI project 31265, GCA_000151905.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Initial gorGor4 release (using Wellcome Trust Sanger Institute Dec 2014 (NCBI project 31265, GCA_000151905.1)): Assembly, Gap, GC Percent, INSDC, Augustus Genes, Genscan Genes, RefSeq Genes, Other RefSeq, Gorilla mRNAs, CpG Islands, Human Chain/Net, Interrupted Repeats, Microsatellite, RepeatMasker, Simple Repeats, WM + SDust</td>
<td>18 Apr 2016</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=gorGor3></A>Gorilla May 2011 (gorGor3.1/gorGor3) (gorGor3, Wellcome Trust Sanger Institute May 2011 (NCBI project 31265, GCA_000151905.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>20 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Tarsier Chain/Net</td>
<td>22 May 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Rhesus Chain/Net</td>
<td>03 Dec 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=gorGor3&g=chainNetTarSyr1" target=_blank>Tarsier Chain/Net</a></td>
<td>16 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=gorGor3&g=chainPapHam1" target=_blank>Baboon Chain/Net</a></td>
<td>16 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=gorGor3&g=chainNetMicMur1" target=_blank>Mouse lemur Chain/Net</a></td>
<td>08 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=gorGor3&g=cons11way" target=_blank>Multiz Alignment & Conservation (11 Species)</a></td>
<td>09 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
Initial gorGor3 release (using Wellcome Trust Sanger Institute May 2011 (NCBI project 31265, GCA_000151905.1)): Assembly, Gap, GC Percent, RepeatMasker, Interrupted Repeats, Simple Repeats, Other RefSeq, Gorilla mRNAs, Other mRNAs, Ensembl Genes, Chimp Chain/Net, Gibbon Chain/Net, Human Chain/Net, Marmoset Chain/Net, Orangutan Chain/Net, Rhesus Chain/Net</td>
<td>23 Nov 2011</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=chlSab2></A>Green monkey Mar. 2014 (Chlorocebus_sabeus 1.1/chlSab2) (chlSab2, Vervet Genomics Consortium GCA_000409795.2)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Initial chlSab2 release (using Vervet Genomics Consortium GCA_000409795.2): Assembly, Gap, GC Percent, INSDC, Augustus Genes, Genscan Genes, Other RefSeq, Green Monkey mRNAs, Green Monkey ESTs, Spliced ESTs, CpG Islands, Human Chain/Net, Rhesus Chain/Net, Tarsier Chain/net, Interrupted Repeats, Microsatellite, RepeatMasker, Simple Repeats, WM + SDust</td>
<td>28 Oct 2016</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=cavPor3></A>Guinea pig Feb. 2008 (Broad/cavPor3) (cavPor3, Broad Institute cavPor3)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Malayan flying lemur Chain/Net</td>
<td>16 Sep 2016</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Added Guinea pig native RefSeq mRNA alignments</td>
<td>22 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Rabbit Chain/Net</td>
<td>26 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 55</td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>02 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial cavPor3 release (using Broad Institute cavPor3)</td>
<td>11 Jun 2008</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=eriEur2></A>Hedgehog May 2012 (EriEur2.0/eriEur2) (eriEur2, Broad Institute EriEur2.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>20 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
SGP Gene Predictions Using homology</td>
<td>27 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Initial eriEur2 release (using Broad Institute EriEur2.0): Assembly, Gap, GC Percent, INSDC, Other RefSeq, Genscan Genes, Hedgehog mRNAs, CpG Islands, Human Chain/Net, RepeatMasker, Interrupted Repeats, Simple Repeats, WM + SDust</td>
<td>08 May 2014</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=eriEur1></A>Hedgehog June 2006 (Broad/eriEur1) (eriEur1, Broad Institute eriEur1 (Draft_v1, NCBI project 12575))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>29 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene predictions - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial eriEur1 release (using Broad Institute eriEur1 (Draft_v1, NCBI project 12575)): Assembly, Gap, GC Percent, Quality Scores, Ensembl Assembly, Other RefSeq, Ensembl Genes, Genscan Genes, Hedgehog mRNAs, CpG Islands, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>08 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=equCab2></A>Horse Sep. 2007 (Broad/equCab2) (equCab2, Broad Institute EquCab2)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>20 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Cow Chain/Net</td>
<td>17 Feb 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=equCab2&g=animalQtl" target=_blank>Horse QTL</a></td>
<td>13 Oct 2014</td>
</tr>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>02 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Chimp (Oct. 2010 (CGSC 2.1.3/panTro3)), Chain and Net Alignments  </td>
<td>29 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
Sheep Chain/Net</td>
<td>19 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Add alignments of non-horse refseq genes</td>
<td>05 Feb 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
Chain and Net tracks on Horse for: platypus and opossum</td>
<td>08 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
Chain and Net tracks on Horse for chicken</td>
<td>09 Nov 2009</td>
</tr>
<TR valign=top><TD align=left>
Chain and Net tracks on Horse for: mouse, dog</td>
<td>03 Nov 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Chains and Nets for non-primate mammal species vs Human</a></td>
<td>16 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 55</td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>09 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Initial equCab2 release (using Broad Institute EquCab2): Assembly; Gap; GC Percent; Quality Scores; Short Match; Restr Enzymes; RefSeq Genes; Trans Map; Ensembl Genes; N-SCAN; Human Proteins; Horse mRNAs; Spliced ESTs; Horse ESTs; Non-Horse mRNAs; RepeatMasker; Interrupted Repeats; Simple Repeats. Downloads: bigZips, chromosomes, database.</td>
<td>04 Dec 2008</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=equCab1></A>Horse Jan. 2007 (Broad/equCab1) (equCab1, Broad Institute equCab1 (1.0))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
Horse LiftOver (equCab2)</td>
<td>04 Dec 2008</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Mouse (Jul. 2007/mm9) Chained Alignments
Mouse (Jul. 2007/mm9) Alignment Net</td>
<td>16 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Initial equCab1 release (using Broad Institute equCab1 (1.0)): Assembly; Gap; GC Percent; Quality Scores; Short Match; Restr Enzymes; RefSeq Genes; N-SCAN; Human Proteins; Horse mRNAs; Spliced ESTs; Horse ESTs; Non-Horse mRNAs; Chains & Nets: Chicken, Rat, Mouse, Chimp, Human, Dog; RepeatMasker; WM + SDust; Simple Repeats.  Downloads: bigZips, chromosomes, database, liftOvers, comparatives.</td>
<td>14 May 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=dipOrd1></A>Kangaroo rat Jul. 2008 (Broad/dipOrd1) (dipOrd1, Broad Institute dipOrd1 (NCBI project 20385, ABRO01000000))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl gene predictions - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial dipOrd1 release (using Broad Institute dipOrd1 (NCBI project 20385, ABRO01000000)): Assembly, Gap, GC Percent, Quality Scores, Ensembl Assembly, Other RefSeq, Ensembl Genes, Genscan Genes, Kangaroo rat mRNAs, Other mRNAs, CpG Islands, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>08 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=petMar2></A>Lamprey Sep. 2010 (WUGSC 7.0/petMar2) (petMar2, GSC Washington University School of Medicine)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=petMar2&g=lampreyGene" target=_blank>Lamprey Genes</a></td>
<td>13 Jun 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial petMar2 release (using GSC Washington University School of Medicine): Assembly, Gap, GC Percent, Other RefSeq, Genscan Genes, Lamprey mRNAs, Spliced ESTs, Lamprey ESTs, CpG Islands, Conservation, Medaka Chain/Net, Chicken Chain/Net, Opossum Chain/Net, Mouse Chain/Net, Human Chain/Net, Lancelet Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>08 Jan 2013</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=petMar1></A>Lamprey Mar. 2007 (WUGSC 3.0/petMar1) (petMar1, WUSTL v.3.0 (March 2007))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Vertebrate genomes, Chain and Net alignment tracks (secondary release)</a></td>
<td>09 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
Quality Scores</td>
<td>26 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Medaka Chain and Net</td>
<td>19 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial petMar1 release (using WUSTL v.3.0 (March 2007)): Assembly, Gap, GC Percent, Short Match, Restriction Enzymes, Other RefSeq Genes, Human Proteins, mRNAs, Other mRNAs, Spliced ESTs, Interrupted Repeats, Window Masker, Simple Repeats, 6-way Conservation, Chains & Nets: Lancelet, Medaka, Chicken, Mouse, Human.  Downloads: BigZips, Database, LiftOver, Conservation, Pairwise Alignments.</td>
<td>09 Jun 2008</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=braFlo1></A>Lancelet Mar. 2006 (JGI 1.0/braFlo1) (braFlo1, JGI v.1.0 (March 2006))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial braFlo1 release (using JGI v.1.0 (March 2006)): Scaffolds, Gap, GC Percent, Short Match, Restriction Enzymes, Other RefSeq Genes, Human Proteins, mRNAs, Other mRNAs, Spliced ESTs, Window Masker, Simple Repeats, 5-way Conservation, Chains & Nets: Lamprey, Chicken, Mouse, Human.  Downloads: BigZips, Chromosomes, Database, LiftOver, Conservation, Pairwise Alignments.</td>
<td>09 Jun 2008</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=anoCar2></A>Lizard May 2010 (Broad AnoCar2.0/anoCar2) (anoCar2, Broad Institute of MIT and Harvard AnoCar 2.0 (GCA_000090745.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>02 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v71</td>
<td>11 Jun 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=anoCar2&g=qualityBw" target=_blank>Quality Scores</a></td>
<td>15 Mar 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v65</td>
<td>22 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=anoCar2&g=chainNetXenTro3" target=_blank>X. tropicalis Chain/Net</a></td>
<td>15 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=anoCar2&g=cpgIslandExt" target=_blank>CpG Islands (Islands < 300 Bases are Light Green)</a></td>
<td>26 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=anoCar2&g=cons7way" target=_blank>Multiz Alignment & Conservation (7 Species)</a></td>
<td>01 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
Initial anoCar2 release (using Broad Institute of MIT and Harvard AnoCar 2.0 (GCA_000090745.1)): Assembly, Gap, GC Percent, Short Match, Restr Enzymes, Other RefSeq, Lizard mRNAs, Spliced ESTs, Lizard ESTs, Other mRNAs, RepeatMasker, Interrupted Rpts, WM + SDust, Simple Repeats, 
Ensembl Gene v62, Stickleback Chain and Net, Platypus Chain and Net, X. Tropicalis Chain and Net, Chicken Chain and Net, Human Chain and Net, Human Proteins</td>
<td>09 Jun 2011</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=anoCar1></A>Lizard Feb. 2007 (Broad/anoCar1) (anoCar1, Broad Institute AnoCar (1.0))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Vertebrate genomes, Chain and Net alignment tracks (secondary release)</a></td>
<td>09 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes version 55</td>
<td>12 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Mouse (Jul. 2007/mm9) Chained Alignments
Mouse (Jul. 2007/mm9) Alignment Net</td>
<td>16 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Initial anoCar1 release (using Broad Institute AnoCar (1.0)): Assembly; Gap; GC Percent; Quality Scores; Genscan Genes; Human Proteins; Repeat Masker; Simple Repeats; Window Masker; Lizard & non-Lizard mRNAs; non-Lizard RefSeq. Chains & Nets: Human, Stickleback, X.tropicalis, Mouse, Chicken. Downloads: bigZips, database, liftOvers, comparatives.
</td>
<td>23 May 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=galVar1></A>Malayan flying lemur Jun. 2014 (G_variegatus-3.0.2/galVar1) (galVar1, Washington University (WashU))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Initial galVar1 release (using Washington University (WashU)): Assembly, Gap, GC Percent, Short Match, Restr Enzymes, Other RefSeq, Genscan Genes, Augustus Genes, Geneid Genes, Other mRNAs, CpG Islands, Human, Chain/Net, Tarsier Chain/Net, RepeatMasker, Interrupted Rpts, WM + SDust, Microsatellite, Simple Repeats.</td>
<td>16 Sep 2016</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=triMan1></A>Manatee Oct. 2011 (Broad v1.0/triMan1) (triMan1, Broad Institute of MIT and Harvard TriManLat1.0 (GCA_000243295.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial triMan1 release (using Broad Institute of MIT and Harvard TriManLat1.0 (GCA_000243295.1)): Assembly, Gap, GC Percent, Other RefSeq, Genscan Genes, Manatee mRNAs, Other mRNAs, CpG Islands, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>13 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=calJac3></A>Marmoset March 2009 (WUGSC 3.2/calJac3) (calJac3, WUSTL 3.2 (GCA_000004665.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
Tarsier Chain/Net</td>
<td>22 May 2015</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v74</td>
<td>23 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>02 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
Rhesus Chain/Net</td>
<td>03 Dec 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=calJac3&g=chainNetPapHam1" target=_blank>Baboon Chain/Net</a></td>
<td>16 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=calJac3&g=chainNetMicMur1" target=_blank>Mouse lemur Chain/Net</a></td>
<td>08 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=calJac3&g=chainGorGor3" target=_blank>Gorilla Chain/Net</a></td>
<td>28 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=calJac3&g=cons13way" target=_blank>Vertebrate Multiz Alignment & Conservation (13 Species)</a></td>
<td>21 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=calJac3&g=chainNetPanTro3" target=_blank>Chimp (Oct. 2010 (CGSC 2.1.3/panTro3)), Chain and Net Alignments</a></td>
<td>19 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=calJac3&g=qualityBw" target=_blank>Marmoset Quality Scores</a></td>
<td>02 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=calJac3&g=bacEnds" target=_blank>Marmoset BAC Ends and BAC End Singletons </a></td>
<td>29 Oct 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN gene predictions from WUStL</td>
<td>03 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Initial calJac3 release (using WUSTL 3.2 (GCA_000004665.1)):
Assembly; Scaffolds; Gap; GC Percent; Short Match; Restr Enzymes; Other RefSeq; Marmoset mRNAs; Spliced ESTs; Marmoset ESTs; Non-Marmoset mRNAs; CpG Islands; Chain & Nets: Rhesus, Organgutan, Chimp, Human, Mouse, Dog, Opossum; Marmoset Repeat Masker; Interrupted Rpts; Simple Repeats.
Downloads: bigZips, chromosomes, database, liftOvers, comparatives. </td>
<td>03 Jun 2010</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=calJac1></A>Marmoset June 2007 (WUGSC 2.0.2/calJac1) (calJac1, WUSTL 2.0.2)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
Marmoset Chain/Net</td>
<td>03 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Primate chains and nets from/to human hg19</a></td>
<td>21 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
Quality Scores</td>
<td>26 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial calJac1 release (using WUSTL 2.0.2):  Assembly, Gap, GC Percent, Other RefSeq, N-SCAN, Genscan Genes, Marmoset mRNAs, Spliced ESTs, Marmoset ESTs, Non-marmoset mRNAs, Conservation, Most Conserved, RepeatMasker, Simple Repeats, Net/Chains: Rhesus, Orangutan, Chimp, Human, Mouse, Dog, Opossum, Platypus. Downloads: bigZips, database, liftOvers, comparitaves, multiz, phastCons.</td>
<td>24 Apr 2008</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=oryLat2></A>Medaka Oct. 2005 (NIG/UT MEDAKA1/oryLat2) (oryLat2, NIG and Tokyo Univ MEDAKA1 (NCBI project 16702, BAAF04000000))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://text.soe.ucsc.edu" target=_blank>pubsBingBlat (a run of hg19-BLAT against all webpages on the internet indexed by bing.com)</a></td>
<td>23 Jan 2014</td>
</tr>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>02 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=oryLat2&g=chainNetPetMar2" target=_blank>Lamprey Chain/Net</a></td>
<td>08 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=oryLat2&g=chainNetFr3" target=_blank>Fugu Chain/Net</a></td>
<td>19 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=oryLat2&g=chainNetDanRer7" target=_blank>Zebrafish Chain/Net</a></td>
<td>13 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Chain/Net and liftOver file</td>
<td>12 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish Chain and Net</td>
<td>12 Jan 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Vertebrate genomes, Chain and Net alignment tracks (secondary release)</a></td>
<td>09 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 55</td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>18 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=oryLat2&g=blastHg18KG" target=_blank>Human Proteins</a></td>
<td>18 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Initial oryLat2 release (using NIG and Tokyo Univ MEDAKA1) : Assembly, Scaffolds, Gap, GC Percent, Quality Scores, RefSeq Genes, Non-Medaka RefSeq, Ensembl Genes, Medaka mRNAs, Spliced ESTs, Medaka ESTs, Non-Medaka mRNAs, Conservation, Most Conserved, Chains & Nets: Human, Mouse, Zebrafish, Fugu, Tetraodon, Stickleback, Lamprey, RepeatMasker, Interrupted Rpts ,WM + SDust, Simple Repeats, Microsatellite. Downloads: bigZips, database, liftOver, multiz, phastCons, vsDanRer5, vsFr2, vsGasAcu1, vsHg18, vsMm9, vsPetMar1, vsTetNig1.</td>
<td>13 Nov 2008</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=geoFor1></A>Medium ground finch Apr. 2012 (GeoFor_1.0/geoFor1) (geoFor1, Beijing Genomics Institute)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=geoFor1&g=chainNetMelUnd1" target=_blank>Budgerigar Chain/Net</a></td>
<td>20 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
Initial geoFor1/Medium ground finch release (using Beijing Genomics Institute (NCBI project 13302, GCA_000277835.1)): Base Position, Assembly, Gap, GC Percent, Short Match, Restr Enzymes, Other RefSeq, Genscan Genes, Medium ground finch mRNAs, CpG Islands, Conservation (7-way), Zebra finch Chain/Net, Chicken Chain/Net, Turkey Chain/Net, Mouse Chain/Net, Human Chain/Net, Repeat Masker, Interrupted Repeats, WM + SDust, Simple Repeats.</td>
<td>16 Aug 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=pteVam1></A>Megabat Jul. 2008 (Broad/pteVam1) (pteVam1, Broad Institute pteVam1 (NCBI project 20325, ABRP01000000))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl gene predictions - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial pteVam1 release (using Broad Institute pteVam1 (NCBI project 20325, ABRP01000000)): Assembly, Gap, GC Percent, Quality Scores, Ensembl Assembly, Other RefSeq, Ensembl Genes, Genscan Genes, Megabat mRNAs, CpG Islands, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>16 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=myoLuc2></A>Microbat Jul. 2010 (Broad Institute Myoluc2.0/myoLuc2) (myoLuc2, Broad Institute (NCBI Project ID: 16951, Accession: GCA_000147115.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v74</td>
<td>23 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>02 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial microbat release (using Broad Institute (NCBI Project ID: 16951, Accession: GCA_000147115.1)): Assembly, Gap, Repeat Masker, Interrupted Repeats, Simple Repeats, CpG Islands, GC Percent, Human Proteins, Other RefSeq, Microbat mRNAs, Other mRNAs.</td>
<td>02 Nov 2011</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=balAcu1></A>Minke whale Oct. 2013 (BalAcu1.0/balAcu1) (balAcu1, Korea Ocean Research & Development Institute)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
Initial balAcu1 release (using Korea Ocean Research & Development Institute BalAcu1.0 (GCA_000493695.1)): Assembly, Gap, GC Percent, Genscan Genes, Other RefSeq, INSDC, CpG Islands, RepeatMasker, Interrupted Rpts, Simple Repeats, WindowMasker + SDust</td>
<td>15 Apr 2014</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=micMur2></A>Mouse lemur May 2015 (Mouse lemur/micMur2) (micMur2, Broad Institute and Baylor College of Medicine Mmur_2.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Initial micMur2 release (using Broad Institute and Baylor College of Medicine Mmur_2.0): Assembly, Gap, GC Percent, INSDC, Genscan Genes, Augustus, Other RefSeq, Mouse Lemur mRNAs, CpG Islands, Tarsier Chain/Net, Human Chain/Net, Simple Repeats, Interrupted Rpts, WM + SDust, RepeatMasker, Microsatellite</td>
<td>11 Feb 2016</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=micMur1></A>Mouse lemur Jul. 2007 (Broad/micMur1) (micMur1, Broad Institute micMur1 (NCBI project 19967, ABDC00000000))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl gene predictions - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Tarsier Chain/Net</td>
<td>22 May 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial micMur1 release (using Broad Institute micMur1 (NCBI project 19967, ABDC00000000)): Assembly, Gap, GC Percent, Quality Scores, Ensembl Assembly, Other RefSeq, Ensembl Genes, Genscan Genes, Mouse lemur mRNAs, CpG Islands, Gorilla Chain/Net, Marmoset Chain/Net, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>08 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=hetGla2></A>Naked mole-rat Jan. 2012 (Broad HetGla_female_1.0/hetGla2) (hetGla2, Broad Institute HetGla_female_1.0 (NCBI project 72441, GCA_000247695.1, WGS AHKG01))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial hetGla2 release (using Broad Institute HetGla_female_1.0 (NCBI project 72441, GCA_000247695.1, WGS AHKG01)): Assembly, Gap, GC Percent, Short Match, Restr Enzymes, Other RefSeq, Genscan Genes, Other mRNAs, CpG Islands, Mouse Chain/Net, RepeatMasker, Interrupted Rpts, WM + SDust, Simple Repeats.</td>
<td>03 Dec 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=hetGla1></A>Naked mole-rat Jul. 2011 (BGI HetGla_1.0/hetGla1) (hetGla1, HetGla_1.0 (NCBI project 68323, accession GCA_000230455.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Initial hetGla1 release (using HetGla_1.0 (NCBI project 68323, accession GCA_000230455.1)): Assembly, Gap, GC Percent, PCR Results, Short Match, Restr Enzymes, Other RefSeq, Genscan Genes, Naked mole-rat mRNAs, Other mRNAs, CpG Islands, mm10 Chain/Net, Repeat Masker, Interrupted Rpts, WM+SDust, Simple Repeats</td>
<td>16 May 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=oreNil2></A>Nile tilapia Jan. 2011 (Broad oreNil1.1/oreNil2) (oreNil2, Broad Institute of MIT and Harvard Orenil1.1 (GCA_000188235.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial oreNil2 release (using Broad Institute of MIT and Harvard Orenil1.1 (GCA_000188235.1)): Assembly, Gap, GC Percent, Other RefSeq, Genscan Genes, Nile tilapia mRNAs, Spliced ESTs, Nile tilapia ESTs, CpG Islands, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>21 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=monDom5></A>Opossum Oct. 2006 (Broad/monDom5) (monDom5, Broad Institute monDom5 (NCBI project 12561, accession AAFR03000000))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=monDom5&g=chainNetCerSim1" target=_blank>White rhinoceros Chain/Net</a></td>
<td>24 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=monDom5&g=chainNetPetMar2" target=_blank>Lamprey Chain/Net</a></td>
<td>08 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
Rhesus Chain/Net</td>
<td>03 Dec 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=monDom5&g=chainNetBosTau7" target=_blank>Cow Chain/Net</a></td>
<td>11 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=monDom5&g=chainNetLoxAfr3" target=_blank>Elephant Chain/Net</a></td>
<td>21 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=monDom5&g=chainNetXenTro3" target=_blank>X. tropicalis Chain/Net</a></td>
<td>15 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=monDom5&g=chainNetMelGal1" target=_blank>Turkey Chain/Net</a></td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=monDom5&g=chainNetBosTau6" target=_blank>Cow Chain/Net</a></td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=monDom5&g=chainNetPanTro3" target=_blank>Chimp (Oct. 2010 (CGSC 2.1.3/panTro3)), Chain and Net Alignments</a></td>
<td>09 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Sheep Chain/Net</td>
<td>19 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Cat Chain/Net</td>
<td>17 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Marmoset Chain/Net</td>
<td>03 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Pig Chain/Net</td>
<td>12 May 2010</td>
</tr>
<TR valign=top><TD align=left>
Rabbit Chain/Net</td>
<td>26 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Panda Chain/Net</td>
<td>16 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
LiftOver chain to/from monDom5 <--> xenTro2</td>
<td>05 Feb 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
Dog Chain & Net, and liftOver</td>
<td>12 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Vertebrate genomes, Chain and Net alignment tracks (primary release)</a></td>
<td>29 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Most Conserved</td>
<td>29 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Conservation (9 vertebrate alignment)</td>
<td>29 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Chicken Chain and Net </td>
<td>25 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish Chain and Net</td>
<td>25 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Platypus Chain/Net/LiftOver (ornAna1)</td>
<td>24 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Orangutan Chain and Net</td>
<td>22 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl update</td>
<td>21 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Chimp Chain and Net</td>
<td>20 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Initial monDom5 release (using Broad Institute monDom5 (NCBI project 12561, accession AAFR03000000)): Assembly, Gap, GC Percent, Quality Scores, RefSeq Genes, Non-Opposum RefSeq, Ensembl Genes, N-SCAN, Genscan Genes, Human Proteins, Opossum mRNAs, Spliced ESTs, Opossum ESTs, Non-Opposum mRNAs, CpG Islands, RepeatMasker, Interrupted Rpts, WM + SDust, Simple Repeats.</td>
<td>22 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=monDom4></A>Opossum Jan. 2006 (Broad/monDom4) (monDom4, Broad Inst. monDom4 Jan06)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
LiftOver to monDom5</td>
<td>20 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Guinea Pig Chain and Net</td>
<td>11 Jun 2008</td>
</tr>
<TR valign=top><TD align=left>
Marmoset Chain/Net/LiftOver (calJac1)</td>
<td>24 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Mouse (Jul. 2007/mm9) Chained Alignments
Mouse (Jul. 2007/mm9) Alignment Net</td>
<td>04 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN gene predictions from WUStL</td>
<td>26 Sep 2006</td>
</tr>
<TR valign=top><TD align=left>
Initial monDom4 release (using Broad Inst. Jan 2006 Draft Assembly): Assembly, Gap, GC Percent, Quality Scores, Oppossum mRNAs, Other mRNAs, Conservation, Most Conserved, Human Proteins, CpG Islands, RepeatMasker, Simple Repeats, Genscan Genes, Ensembl Genes, Opossum EST's, Spliced EST's, RefSeq Genes, Other RefSeq; Chains & Nets: Human, Chimp, Mouse, Rat, Chicken, X.tropicalis, Zebrafish; Downloads: bigZips, chromosomes, database.
</td>
<td>08 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
hgdownload files for monDom4</td>
<td>24 May 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=monDom1></A>Opossum Oct. 2004 (Broad prelim/monDom1) (monDom1, Broad Inst. Prelim Oct04)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Keyword search by NCBI contig accession ID added.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Initial monDom1 release (using Broad Inst. Prelim Oct04): Assembly, Gap, GC Percent, Quality Scores, Oppossum mRNAs, Non-Opposum mRNAs, Conservation, Most Conserved, Human Proteins, Human Chain/Net, Mouse Chain/Net, Chicken Chain/Net, Zebrafish Chain/Net, CpG Islands, RepeatMasker, Simple Repeats. Downloads: bigZips, chromosomes, database.</td>
<td>18 Jan 2005</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=ponAbe2></A>Orangutan July 2007 (WUGSC 2.0.2/ponAbe2) (ponAbe2, WUSTL 3.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>20 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
Tarsier Chain/Net</td>
<td>22 May 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
Rhesus Chain/Net</td>
<td>03 Dec 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=ponAbe2&g=chainGorGor3" target=_blank>Gorilla Chain/Net</a></td>
<td>28 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Chimp (Oct. 2010 (CGSC 2.1.3/panTro3)), Chain and Net Alignments  </td>
<td>28 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Marmoset Chain/Net</td>
<td>03 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
Opossum Chain and Net</td>
<td>22 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 55</td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Primate chains and nets from/to human hg19</a></td>
<td>21 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>09 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update (version 50)</td>
<td>19 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes, version 49</td>
<td>02 May 2008</td>
</tr>
<TR valign=top><TD align=left>
Marmoset Chain/Net/LiftOver (calJac1)</td>
<td>24 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN gene predictions</td>
<td>17 Mar 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial ponAbe2 release (using WUSTL 3.0): Assembly, Gap, GC Percent, Quality Scores, Non-Orangutan RefSeq, Genscan Genes, mRNAs, Spliced ESTs, ESTs, Other mRNAs,Conservation, Most Conserved, RepeatMasker, Interrupted Rpts, Simple Repeats,
Opossum Chain & Net, Platypus Chain & Net, Mouse Chain & Net, Rhesus Chain & Net, Chimp Chain & Net, Human Chain & Net</td>
<td>11 Mar 2008</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=priPac1></A>P. pacificus Feb. 2007 (WUGSC 5.0/priPac1) (priPac1, Washington University School of Medicine GSC Pristionchus pacificus 5.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
C. elegans Chain/Net</td>
<td>06 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
C. elegans Proteins WS190 Mapped by Chained tBLASTn</td>
<td>18 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial priPac1 release (using WUSTL Pristionchus pacificus 5.0): Assembly; Contigs; Gap; GC Percent; Quality Scores; Short Match; Restr Enzymes; Non-P. pacificus RefSeq; C. elegans Proteins; P. pacificus mRNAs; Spliced ESTs; P. pacificus ESTs; Non-P. pacificus mRNAs; Chains and Nets: C. briggsae, C. remanei, C. brenneri, C. elegans; RepeatMasker; Simple Repeats.  Downloads: bigZips, chromsomes, database, liftOvers, comparatives.</td>
<td>03 Aug 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=chrPic1></A>Painted turtle Dec. 2011 (v3.0.1/chrPic1) (chrPic1, International Painted Turtle Genome Sequencing Consortium (GCA_000241765.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial chrPic1 release (using International Painted Turtle Genome Sequencing Consortium (GCA_000241765.1)): Assembly, Gap, GC Percent, Genscan Genes, CpG Islands, RepeatMasker, Interrupted Repeats, Simple Repeats, WM + SDust, Other RefSeq, Painted turtle mRNAs, Other mRNAs, Mouse Chain/Net</td>
<td>11 Jun 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=ailMel1></A>Panda Dec. 2009 (BGI-Shenzhen 1.0/ailMel1) (ailMel1, BGI-Shenzhen AilMel 1.0 Dec. 2009)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl genes - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Augustus gene predictions v3.1</td>
<td>23 Jul 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v74</td>
<td>23 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>02 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
tRNA Genes</td>
<td>15 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v65</td>
<td>22 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Cat Chain/Net</td>
<td>17 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Initial ailMel1 release (using BGI-Shenzhen AilMel 1.0 Dec. 2009): 

Assembly, Gap, GC Percent, Other RefSeq, Panda mRNAs, Other mRNAs, Panda ESTs, RepeatMasker, Interrupted Rpts, Simple Repeats, Opossum Chain and Net, Dog Chain and Net, Mouse Chain and Net, Human Chain and Net, Rat Chain and Net. Downloads: bigZips, chromosomes, database.</td>
<td>16 Apr 2010</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=susScr3></A>Pig Aug. 2011 (SGSC Sscrofa10.2/susScr3) (susScr3, SGSC Sscrofa10.2 (NCBI project 13421, GCA_000003025.4, WGS AEMK01))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Animal Quantitative Trait Loci from animalQTLdb</td>
<td>27 Jun 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=susScr3&g=snp138" target=_blank>Mult. SNPs(138), All SNPs(138)</a></td>
<td>28 Feb 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://text.soe.ucsc.edu" target=_blank>pubsBingBlat (a run of hg19-BLAT against all webpages on the internet indexed by bing.com)</a></td>
<td>23 Jan 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v74</td>
<td>23 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial susScr3 release (using SGSC Sscrofa10.2 (NCBI project 13421, GCA_000003025.4, WGS AEMK01)): Assembly, Gap, GC Percent, RefSeq Genes, Other RefSeq, Ensembl Genes, Genscan Genes, Pig mRNAs, Spliced ESTs, Pig ESTs, Other mRNAs, CpG Islands, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>16 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=susScr2></A>Pig Nov. 2009 (SGSC Sscrofa9.2/susScr2) (susScr2, SGSC Sscrofa9.2 (NCBI project 10718, GCA_000003025.2))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=susScr2&g=numtSeq2012" target=_blank>NumtS Sequence</a></td>
<td>23 Jul 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=susScr2&g=chainNetBosTau7" target=_blank>Cow Chain/Net</a></td>
<td>11 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=susScr2&g=chainNetBosTau6" target=_blank>Cow Chain/Net</a></td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>12 May 2010</td>
</tr>
<TR valign=top><TD align=left>
Initial susScr2 release (using SGSC Sscrofa9.2 (NCBI project 10718, GCA_000003025.2)): 
Assembly; Scaffolds; Gap; GC Percent; Short Match; Restr Enzymes; Ensembl Genes; Human Proteins; Pig mRNAs; Spliced ESTs; Pig ESTs; Non-Pig mRNAs; CpG Islands; Chain & Nets: Cow, Human, Mouse, Opossum; Pig Repeat Masker; Interrupted Rpts; Simple Repeats.
Downloads: bigZips, chromosomes, database, liftOvers, comparatives. </td>
<td>12 May 2010</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=ochPri3></A>Pika May 2012 (OchPri3.0/ochPri3) (ochPri3, Broad Institute)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Initial ochPri3 release (using Broad Institute): Assembly, Gap, GC Percent, INSDC, Genscan Genes, Other RefSeq, Pika mRNAs, CpG Islands, Human Chain/Net, Interrupted Repeats, RepeatMasker, Simple Repeats, WM + SDust</td>
<td>14 Jul 2014</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=ochPri2></A>Pika Jul. 2008 (Broad/ochPri2) (ochPri2, Broad Institute ochPri2)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl gene predictions - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial ochPri2 release (using Broad Institute ochPri2): Assembly, Gap, GC Percent, Quality Scores, Ensembl Assembly, Other RefSeq, Ensembl Genes, Genscan Genes, Pika mRNAs, CpG Islands, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>21 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=ornAna2></A>Platypus Feb. 2007 (ASM227v2/ornAna2) (ornAna2, WUGSC ASM227v2)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Initial ornAna2 release (using WUGSC ASM227v2): Assembly, Gap, GC Percent, INSDC, Augustus Genes, Genscan Genes, RefSeq Genes, Other RefSeq, Spliced ESTs, Platypus ESTs, Platypus mRNAs, CpG Islands, Interrupted Repeats, Simple Repeats, Microsatellites, WM + SDust, RepeatMasker</td>
<td>19 Feb 2016</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=ornAna1></A>Platypus Mar. 2007 (WUGSC 5.0.1/ornAna1) (ornAna1, WUSTL version 5.0.1)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=ornAna1&g=chainNetBosTau7" target=_blank>Cow Chain/Net</a></td>
<td>11 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=anoCar1&g=cpgIslandExt" target=_blank>CpG Islands</a></td>
<td>29 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=ornAna1&g=chainNetBosTau6" target=_blank>Cow Chain/Net</a></td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=ornAna1&g=chainNetAnoCar2" target=_blank>Lizard Chain/Net</a></td>
<td>12 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
Chain and Net to Horse (equCab2)</td>
<td>08 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
Opossum Chain and Net</td>
<td>24 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 55</td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>02 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Quality Scores</td>
<td>26 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update (version 50)</td>
<td>18 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Marmoset Chain/Net/LiftOver (calJac1)</td>
<td>24 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl v49 gene predictions</td>
<td>15 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Chain track, Net track, and liftOver file to bosTau4</td>
<td>08 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Mouse (Jul. 2007/mm9) Chained Alignments
Mouse (Jul. 2007/mm9) Alignment Net</td>
<td>19 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Interrupted Rpts</td>
<td>01 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
WSSD Coverage</td>
<td>20 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Regenerated with improved platypus library.</td>
<td>24 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
Initial ornAna1 release (using WUSTL version 5.0): gold, gap, gc5Base, ensGene, 6-way Conservation and phastCons, repeatMasker, simple repeats, Human Proteins,  Platypus mRNAs, Spliced ESTs, Platypus ESTs, Other mRNAs, refSeq Genes, Chains and Nets:  Lizard, Chicken, Opossum, Mouse, Human</td>
<td>01 Jun 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=oryCun2></A>Rabbit Apr. 2009 (Broad/oryCun2) (oryCun2, Broad Institute oryCun2 (NCBI project 12819, AAGW00000000))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>20 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>02 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v65</td>
<td>22 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins for Rabbit</td>
<td>29 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Initial oryCun2 release (using Broad Institute oryCun2 (NCBI project 12819, AAGW00000000)): Scaffolds, Assembly, Gap, GC Percent, Quality Scores, RefSeq Genes, Other RefSeq, Rabbit mRNAs, Spliced ESTs, Rabbit ESTs, Other mRNAs, TransMap UCSC, TransMap RefGene, TransMap mRNA, TransMap ESTs, Ensembl Genes, RepeatMasker, Interrupted Rpts, Simple Repeats, Opossum Chain and Net, Mouse Chain and Net, Human Chain and Net, Guinea pig Chain and Net. Downloads: bigZips, chromosomes, database</td>
<td>26 Apr 2010</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=rn6></A>Rat Jul. 2014 (RGSC 6.0/rn6) (rn6, RGSC Rnor_6.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Chicken Chain/Net</td>
<td>26 Sep 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../cgi-bin/hgTrackUi?db=rn6&g=cloneEndSuper" target=_blank>Rat: Mapping of clone libraries end placements</a></td>
<td>11 Jul 2016</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>20 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
SGP Gene Predictions Using homology</td>
<td>27 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
Initial rn6 release (using RGSC Rnor_6.0 (GCA_000001895.4)): Assembly, Gap, GC Percent, INSDC, RefSeq Genes, MGC Genes, Other RefSeq, Genscan Genes, Rat mRNAs, Rat ESTs, Spliced ESTs, Other mRNAs, CpG Islands, RepeatMasker, Interrupted Repeats, Simple Repeats, WindowMasker + SDust.</td>
<td>01 Aug 2014</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=rn5></A>Rat Mar. 2012 (RGSC 5.0/rn5) (rn5, RGSC Rnor_5.0 (GCA_000001895.3))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
CRISPR</td>
<td>07 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Gene Predictions - Ensembl 78</td>
<td>06 Feb 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=rn5&g=cons13way" target=_blank>Multiz Alignment & Conservation (13 Species)</a></td>
<td>20 Jun 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v74</td>
<td>23 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=rn5&g=cytoBand" target=_blank>Chromosome Band Track</a></td>
<td>26 Jul 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
Rhesus Chain/Net</td>
<td>03 Dec 2012</td>
</tr>
<TR valign=top><TD align=left>
Initial rn5 release (using RGSC Rnor_5.0 (GCA_000001895.3)): Assembly, Gap, GC Percent, Genscan Genes, CpG Islands, RepeatMasker, Interrupted Repeats, Simple Repeats, WM + SDust, RefSeq Genes, Other RefSeq, MGC Genes, Rat mRNAs, Spliced ESTs, Rat ESTs, Other mRNAs, Mouse Chain/Net</td>
<td>11 Jun 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=rn4></A>Rat Nov. 2004 (Baylor 3.4/rn4) (rn4, Baylor HGSC v. 3.4)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href="http://text.soe.ucsc.edu" target=_blank>pubsBingBlat (a run of hg19-BLAT against all webpages on the internet indexed by bing.com)</a></td>
<td>23 Jan 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=rn4&g=chainNetBosTau7" target=_blank>Cow Chain/Net</a></td>
<td>11 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=rn4&g=rgdGene2" target=_blank>RGD Genes for rat (rn4)</a></td>
<td>29 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=rn4&g=chainNetXenTro3" target=_blank>X. tropicalis Chain/Net</a></td>
<td>15 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=rn4&g=chainNetMelGal1" target=_blank>Turkey Chain/Net</a></td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=rn4&g=chainNetBosTau6" target=_blank>Cow Chain/Net</a></td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
BlastTab tables rebuild</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
tRNA Genes</td>
<td>18 May 2010</td>
</tr>
<TR valign=top><TD align=left>
Panda Chain/Net</td>
<td>16 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
Added link to The GeneNetwork from the UCSC Genes details page.</td>
<td>02 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter and UCSC Genes otherOrgs links to yeast sacCer2</td>
<td>02 Nov 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 56</td>
<td>30 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Chains and Nets for non-primate mammal species vs Human</a></td>
<td>16 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 55</td>
<td>11 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>18 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
agilentProbes update (formerly agilentCgh)
</td>
<td>23 Jan 2009</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Converted MGC track to PSL (to allow for alignment visualization)</td>
<td>22 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update (version 50)</td>
<td>22 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Guinea Pig Chain and Net</td>
<td>11 Jun 2008</td>
</tr>
<TR valign=top><TD align=left>
Agilent Catalog Oligonucleotide CGH Microarray probesets</td>
<td>04 Jun 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl v49 gene predictions</td>
<td>15 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
SGP and Geneid tracks</td>
<td>02 Jan 2008</td>
</tr>
<TR valign=top><TD align=left>
Updated mouse column in Gene Sorter and in Orthologous Genes section of gene details pages to mm9 UCSC Genes</td>
<td>19 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix Rat Exon array version 1.0 probes and transcripts</td>
<td>18 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Added CONTRAST gene predictions</td>
<td>17 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Rat Quantitative Trait Locus from RGD (data correction)</td>
<td>03 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Quantitative Trait Locus from RGD (data update)</td>
<td>25 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Mouse (mm9) Chain and Net tracks and liftOver file</td>
<td>28 Aug 2007</td>
</tr>
<TR valign=top><TD align=left>
Updated Drosophila column in Gene Sorter from dm2 to dm3 FlyBase genes</td>
<td>28 Aug 2007</td>
</tr>
<TR valign=top><TD align=left>
removed "e-rows" from the upstream MAF files</td>
<td>09 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
Update of hgBlastTab tables (to hg18) to coincide with UCSC Genes</td>
<td>06 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
synteny-filter update for xxBlastTab for HMR (hg18, hg17, mm8, mm7, rn4)</td>
<td>27 Feb 2007</td>
</tr>
<TR valign=top><TD align=left>
Lift Over File from rn4 to rn3</td>
<td>12 Feb 2007</td>
</tr>
<TR valign=top><TD align=left>
Added NCBI Clone Validation information to details of MGC genes</td>
<td>12 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
SNP Alignment Display</td>
<td>01 Dec 2006</td>
</tr>
<TR valign=top><TD align=left>
SNPs</td>
<td>23 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
Update to UCSC Known Genes color coding scheme</td>
<td>03 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN gene predictions from WUStL, using human as informant.</td>
<td>03 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
Superfamily track, Gene Sorter column</td>
<td>28 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Proteome Browser release</td>
<td>28 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Known Genes</td>
<td>22 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish (danRer4) Chain/Net/liftOver</td>
<td>21 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Initial rn4 release (using Baylor HGSC v3.4): Chromosome Band; RGD QTL; STS Markers; Assembly; Gap; BAC End Pairs; GC Percent; Short Match; Restr Enzymes; Known Genes; Rat and NonRat RefSeq Genes; MGC Genes; Ensembl Genes; Genscan Genes; miRNA, Rat mRNAs; Spliced ESTs; Rat ESTs; NonRat mRNAs; GNF Atlas 2; Affy RG-U34A; Affy RAE230; CpG Islands; CpG Islands (AL); Conservation; Most Conserved; Chains & Nets: X. tropicalis, Chicken, Cow, Dog, Human, Rhesus, Mouse; RepeatMasker; Simple Repeats.  Downloads: bigZips, chromosomes, database, liftOvers, comparatives.</td>
<td>20 Jun 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=rn3></A>Rat June 2003 (Baylor 3.1/rn3) (rn3, Baylor HGSC v. 3.1)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Converted MGC track to PSL (to allow for alignment visualization)</td>
<td>22 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Added NCBI Clone Validation information to details of MGC genes</td>
<td>12 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
Update to UCSC Known Genes color coding scheme</td>
<td>03 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
Replaced twinscan track with N-SCAN (with hg17 informant)</td>
<td>09 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Improved searching capability for Known Genes</td>
<td>03 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated rn3 -> mm6 liftOver (Downloads, Batch Coordinate Conversion utility)
</td>
<td>06 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
LiftOvers: fr1, tetNig1, xenTro1, galGal2, bosTau2, rheMac1, panTro1, hg17, rn3 -> mm7; fr1, tetNig1, xenTro1, galGal2, panTro1 -> mm6; galGal2 -> danRer2, mm5, hg17, monDom1, tetNig1, xenTro1; rheMac1 -> hg17; panTro1 -> hg16, hg17</td>
<td>03 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Dog Chain/Net/LiftOver (canFam2)</td>
<td>22 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
link in KG details page to GeneNetwork</td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
rn3ToMm6 liftOver (Downloads and browser Batch Coordinate Conversion utility)</td>
<td>06 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
rn3ToHg17 liftOver (Downloads and browser Batch Coordinate Conversion utility)</td>
<td>01 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Restriction Enzyme mapping (REbase release 505).</td>
<td>24 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Non-Rat Refseq</td>
<td>18 May 2005</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
vsHg17 and vsMm5 pairwise alignments</td>
<td>01 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Cytoband Ideo</td>
<td>30 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
MGC Genes</td>
<td>24 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
ECgene track update, EST clustering gene predictions</td>
<td>22 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
Assembly/gold track update to fix fragStart off by 1</td>
<td>19 Oct 2004</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain and Net (update to mm5)</td>
<td>13 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Human Chain, Human Net (update  hg15 --> hg17)</td>
<td>30 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Rat/Mouse/Human (updated)</td>
<td>20 Aug 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter for rn3</td>
<td>24 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
RGD SSLP</td>
<td>01 Jun 2004</td>
</tr>
<TR valign=top><TD align=left>
RGD ESTs</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
RGD Genes</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
RGD QTL</td>
<td>28 May 2004</td>
</tr>
<TR valign=top><TD align=left>
GNF Atlas 2</td>
<td>27 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix RG-U34A</td>
<td>24 May 2004</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>21 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Proteome Browser</td>
<td>07 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>06 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Ecores</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Ecores</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
miRNA (updated)</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Superfamily</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Fugu Ecores, Tetraodon Ecores</td>
<td>05 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>03 May 2004</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix RAE230</td>
<td>15 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (updated)</td>
<td>02 Apr 2004</td>
</tr>
<TR valign=top><TD align=left>
Overlap SNPs (dbSnp 118)</td>
<td>25 Feb 2004</td>
</tr>
<TR valign=top><TD align=left>
Chromosome Band</td>
<td>23 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
ECgene Genes</td>
<td>22 Dec 2003</td>
</tr>
<TR valign=top><TD align=left>
TIGR Gene Index</td>
<td>18 Nov 2003</td>
</tr>
<TR valign=top><TD align=left>
Genbank daily updates enabled</td>
<td>14 Oct 2003</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain, Mouse Net (using mm3)</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes (extended)</td>
<td>18 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Rat/Mouse/Human (using mm3 and hg16)</td>
<td>05 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Human Chain, Human Net (using hg16)</td>
<td>04 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
Human Tight (using hg16)</td>
<td>03 Sep 2003</td>
</tr>
<TR valign=top><TD align=left>
SGP Genes</td>
<td>05 Aug 2003</td>
</tr>
<TR valign=top><TD align=left>
Known Genes</td>
<td>30 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Fgenesh++ Genes</td>
<td>18 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Initial rn3 Release (using Baylor HGSC v. 3.1): Assembly, Gap, Bactigs, GC Percent, Bac End Pairs, RefSeq Genes, STS Markers, Geneid Genes, Genscan Genes, Rat mRNAs, Spliced ESTs, Rat ESTs, NonRat mRNAs, NonRat ESTs, CpG Islands, RepeatMasker, Simple Repeats, bigZips downloads, chromosome downloads</td>
<td>14 Jul 2003</td>
</tr>
<TR valign=top><TD align=left>
Twinscan (using hg15)</td>
<td>14 Jul 2003</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=rheMac8></A>Rhesus Nov. 2015 (BCM Mmul_8.0.1/rheMac8) (rheMac8, Baylor College of Medicine Genome Sequencing Center)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Green monkey Chain/Net</td>
<td>28 Oct 2016</td>
</tr>
<TR valign=top><TD align=left>
Initial rheMac8 release (using Baylor College of Medicine Genome Sequencing Center): Assembly, Gap, GC Percent, INSDC, Restr Enzymes, Short Match, Augustus, Genscan Genes, Other RefSeq, RefSeq Genes, Rhesus ESTs, Rhesus mRNAs, Spliced ESTs, CpG Islands, Human Chain/Net, Mouse Chain/Net, Interrupted Rpts, Microsatellite, RepeatMasker, Simple Repeats, WM + SDust.</td>
<td>18 Apr 2016</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=rheMac3></A>Rhesus Oct. 2010 (BGI CR_1.0/rheMac3) (rheMac3, Beijing Genomics Institute, Shenzhen (GCA_000230795.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>20 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
Tarsier Chain/Net</td>
<td>22 May 2015</td>
</tr>
<TR valign=top><TD align=left>
Human (Dec. 2013 (GRCh38/hg38)) Chained Alignments</td>
<td>14 May 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=rheMac3&g=chainNetHg38" target=_blank>Human Chain/Net</a></td>
<td>06 Mar 2014</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial rheMac3 release (using Beijing Genomics Institute, Shenzhen (GCA_000230795.1)): Assembly, Scaffolds, Gap, GC Percent, Short Match, Restr Enzymes, RefSeq Genes, Other RefSeq, Genscan Genes, Rhesus mRNAs, Spliced ESTs, Rhesus ESTs, Other mRNAs, CpG Islands, Orangutan Chain/Net, Gorilla Chain/Net, Chimp Chain/Net, Human Chain/Net, Marmoset Chain/Net, Rat Chain/Net, Mouse Chain/Net, Opossum Chain/Net, RepeatMasker, Interrupted Rpts, WM + SDust, Simple Repeats.</td>
<td>03 Dec 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=rheMac2></A>Rhesus Jan. 2006 (MGSC Merged 1.0/rheMac2) (rheMac2, Venter Inst./Baylor/WashU Merged assembly Jan 2006 v1 edit4)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=rheMac2&g=chainGorGor3" target=_blank>Gorilla Chain/Net</a></td>
<td>28 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=rheMac2&g=chainNetPanTro3" target=_blank>Chimp (Oct. 2010 (CGSC 2.1.3/panTro3)), Chain and Net Alignments</a></td>
<td>19 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=rheMac2&g=refSeqAnno" target=_blank>RefSeq 1.2 genomic annotations from NCBI</a></td>
<td>03 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Marmoset Chain/Net</td>
<td>03 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
Opossum Chain and Net</td>
<td>23 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 55</td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Primate chains and nets from/to human hg19</a></td>
<td>21 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>09 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update (version 50)</td>
<td>19 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Marmoset Chain/Net/LiftOver (calJac1)</td>
<td>24 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Mouse (Jul. 2007/mm9) Chained Alignments
Mouse (Jul. 2007/mm9) Alignment Net</td>
<td>11 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Native RefSeqs</td>
<td>13 Jun 2007</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands</td>
<td>12 Sep 2006</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN gene prediction for Rhesus rheMac2 from Michael Brent's Computational Genomics Group at Washington University St. Louis 
</td>
<td>11 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
quality scores for rheMac2</td>
<td>20 Mar 2006</td>
</tr>
<TR valign=top><TD align=left>
Contigs. Table update.  Added missing records.</td>
<td>16 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
Initial rhesus macaque browser release (Macaca mulatta preliminary assembly, Baylor, rheMac2) Assembly, Gap, RepeatMasker, Simple Repeats, Other RefSeq, Other mRNAs, Spliced ESTs, Rhesus mRNAs, Rhesus ESTs</td>
<td>13 Feb 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=proCap1></A>Rock hyrax Jul. 2008 (Broad/proCap1) (proCap1, Broad Institute proCap1 (NCBI project 13972, ABRQ01000000))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl gene predictions - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial proCap1 release (using Broad Institute proCap1 (NCBI project 13972, ABRQ01000000)): Assembly, Gap, GC Percent, Quality Scores, Ensembl Assembly, Other RefSeq, Ensembl Genes, Genscan Genes, CpG Islands, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>16 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=sacCer3></A>S. cerevisiae Apr. 2011 (SacCer_Apr2011/sacCer3) (sacCer3, Saccharomyces cerevisiae S288c assembly from Saccharomyces Genome Database (GCA_000146055.2))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
CRISPR</td>
<td>07 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
LiftOver from sacCer3 to sacCer2 and sacCer1</td>
<td>09 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Initial sacCer3 release (using Saccharomyces cerevisiae S288c assembly from Saccharomyces Genome Database (GCA_000146055.2)): Base Position, Assembly, Gap, WashU Clones, GC Percent, Short Match, Restr Enzymes,SGD Genes, SGD Other, Ensembl Genes, Human Proteins, S. cer. mRNAs, Spliced ESTs, S. cer. ESTs, Regulatory Code, Reg. ChIP-chip, Reg. Module, ORegAnno, Conservation, Most Conserved, Simple Repeats.</td>
<td>07 Oct 2011</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=sacCer2></A>S. cerevisiae June 2008 (SGD/sacCer2) (sacCer2, SGD June 2008)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href="http://text.soe.ucsc.edu" target=_blank>pubsBingBlat (a run of hg19-BLAT against all webpages on the internet indexed by bing.com)</a></td>
<td>23 Jan 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
BlastTab tables rebuild</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
UW Footprints (updated to a composite track)</td>
<td>07 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://hgdownload.gi.ucsc.edu/goldenPath/sacCer2/bigZips/" target=_blank>sgdGene.upstream1000.fa.gz  sgdGene.upstream5000.fa.gz
sgdGene.upstream2000.fa.gz
</a></td>
<td>03 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgGateway?org=S.+cerevisiae&db=sacCer2" target=_blank>Initial S. cerevisiae Release (based on sequence in SGD as of June 2008): Assembly, Gap, WashU Clones, GC Percent, Ensembl Genes, SGD Genes, SGD Other, Human Proteins, Yeast mRNAs, Spliced ESTs, Yeast ESTs, Regulatory Code, Reg. CHIP/CHIP, Regulatory Module, ORegAnno, Simple Repeats, Conservation (S. paradoxus; S. mikatae; S. kudriavzevii; S. bayanus; S. castelli; S. kluyveri), Most Conserved. Downloads: bigZips, chromosomes, liftOver, conservation, database.</a></td>
<td>11 Aug 2009</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=sacCer1></A>S. cerevisiae Oct. 2003 (SGD/sacCer1) (sacCer1, SGD sequence data as of 1 Oct. 2003)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
UW Footprints (updated to a composite track)</td>
<td>07 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>29 Jan 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update (version 50)</td>
<td>19 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
Regulatory elements from Open Regulatory Annotation (ORegAnno) </td>
<td>16 Nov 2007</td>
</tr>
<TR valign=top><TD align=left>
Updated mouse column in Gene Sorter and in Orthologous Genes section of gene details pages to mm9 UCSC Genes</td>
<td>19 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Updated Drosophila column in Gene Sorter from dm2 to dm3 FlyBase genes</td>
<td>28 Aug 2007</td>
</tr>
<TR valign=top><TD align=left>
Update of dmBlastTab tables (to dm2) to coincide with Fly Base v4.3 Genes</td>
<td>20 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Update of hgBlastTab tables (to hg18) to coincide with UCSC Genes</td>
<td>06 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Rat Ortholog column of Gene Sorter and Known Gene details updated to rn4</td>
<td>20 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter P2P Protein-to-Protein Interaction Network (worm, fly, yeast)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Known Genes -- UCSC SAM-T02 protein structure analysis and prediction.</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Regulatory Code and Reg. CHIP/CHIP</td>
<td>17 Dec 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation (data update), Most Conserved (data update)</td>
<td>28 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Gene Sorter: Regulatory Module column</td>
<td>24 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Regulatory Module (first public release of Eran Segal's regulatory module track.)</td>
<td>24 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Conservation, Most Conserved</td>
<td>10 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial sacCer1 Release (based on sequence in SGD as of 1 Oct 2003): WashU Clones, GC Percent, SGD Genes, SGD Other, Yeast ESTs, Spliced ESTs, Yeast ESTs, Multiz Yeast, Simple Repeats. Downloads: bigZips, database.</td>
<td>14 Jan 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=strPur2></A>S. purpuratus Sep. 2006 (Baylor 2.1/strPur2) (strPur2, Baylor release 3 Spur 2.1)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial strPur2 release (using Baylor release 3 Spur 2.1): gap, gold, gc5Base, quality, genscan, Human Proteins, refGene, S. purpuratus mRNAs, Spliced ESTs, Other mRNAs, Human nets and chains, Repeat Masker, Simple Repeats</td>
<td>16 Oct 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=strPur1></A>S. purpuratus Apr. 2005 (Baylor 1.1/strPur1) (strPur1, Baylor)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins track (updated from hg17 to hg18)</td>
<td>27 Feb 2007</td>
</tr>
<TR valign=top><TD align=left>
Initial strPur1 release (using Baylor Spur_0.5 whole genome shotgun assembly) Assembly, gap, RepeatMasker, SimpleRepeats, GC Percent, S. purpuratus mRNAs, Other mRNAs, Genscan, Mapped Human Proteins, Quality, RefSeq.</td>
<td>07 Mar 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=aplCal1></A>Sea hare Sept. 2008 (Broad 2.0/aplCal1) (aplCal1, Broad v. Aplcal2.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Initial aplCal1 release (using Broad v. Aplcal2.0): Assembly, Gap, GC Percent, Quality Scores, Short Match, Restr Enzymes, Other RefSeq, Human Proteins, Sea Hare mRNAs, Spliced ESTs, Sea Hare ESTs, Other mRNAs, CpG Islands, Human Chain/Net, RepeatMasker, Interrupted Rpts, Simple Repeats. Downloads: bzips, database, liftOver files, comparatives.</td>
<td>15 Mar 2010</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=oviAri3></A>Sheep Aug. 2012 (ISGC Oar_v3.1/oviAri3) (oviAri3, International Sheep Genome Consortium)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Chromosome Band (Ideogram)</td>
<td>08 Dec 2014</td>
</tr>
<TR valign=top><TD align=left>
Animal Quantitative Trait Loci from animalQTLdb</td>
<td>27 Jun 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v74</td>
<td>23 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial oviAri3 release (using International Sheep Genome Consortium): Assembly, Gap, GC Percent, RefSeq Genes, Other RefSeq, Genscan Genes, Sheep mRNAs, Spliced ESTs, Sheep ESTs, Other mRNAs, CpG Islands, Human Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats.</td>
<td>04 Sep 2013</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=oviAri1></A>Sheep Feb. 2010 (ISGC Ovis_aries_1.0/oviAri1) (oviAri1, ISGC (NCBI project 10709, accession GCA_000005525.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
oviAri1 Initial release (using ISGC (NCBI project 10709, accession GCA_000005525.1)): Assembly, Scaffolds, Gap, GC Percent, Short Match, Restr Enzymes, RefSeq Genes, Other RefSeq, Sheep mRNA, Spliced ESTs, Sheep ESTs, Other mRNAs, CpG Islands, Cow Chain/Net, Horse Chain/Net, Mouse Chain/Net, Human Chain/Net, Opossum Chain/Net, RepeatMasker, Interrupted Rpts, Simple Repeats.</td>
<td>18 Apr 2011</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=sorAra2></A>Shrew Aug. 2008 (Broad/sorAra2) (sorAra2, Broad Institute SorAra2.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Initial sorAra2 release (using Broad Institute SorAra2.0): Assembly, Gap, GC Percent, INSDC, Genscan Genes, Other RefSeq, Shrew mRNAs, CpG Islands, Mouse Chain/Net, Human Chain/Net, Interrupted Repeats, RepeatMasker, Simple Repeats, WM + SDust</td>
<td>14 Jul 2014</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=sorAra1></A>Shrew June 2006 (Broad/sorAra1) (sorAra1, Broad Institute sorAra1 (Draft_v2))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl gene predictions - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial sorAra1 release (using Broad Institute sorAra1 (Draft_v2)): Assembly, Gap, GC Percent, Quality Scores, Ensembl Assembly, Other RefSeq, Ensembl Genes, Genscan Genes, Shrew mRNAs, CpG Islands, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>16 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=choHof1></A>Sloth Jul. 2008 (Broad/choHof1) (choHof1, Broad Institute choHof1)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl gene predictions - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial choHof1 release (using Broad Institute choHof1): Assembly, Gap, GC Percent, Quality Scores, Ensembl Assembly, Other RefSeq, Ensembl Genes, Genscan Genes, Sloth mRNAs, Other mRNAs, CpG Islands, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>08 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=speTri2></A>Squirrel Nov. 2011 (Broad/speTri2) (speTri2, Broad Institute (GCA_000236235.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes v74</td>
<td>23 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial speTri2 release (using Broad Institute (GCA_000236235.1)): Assembly, Gap, GC Percent, Other RefSeq, Ensembl Genes, Genscan Genes, Squirrel mRNAs, Spliced ESTs, Squirrel ESTs, CpG Islands, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>13 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=saiBol1></A>Squirrel monkey Oct. 2011 (Broad/saiBol1) (saiBol1, Broad SaiBol1.0 (GCA_000235385.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Tarsier Chain/Net</td>
<td>22 May 2015</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial saiBol1 release (using Broad SaiBol1.0 (GCA_000235385.1)): Assembly, Gap, GC Percent, Other RefSeq, Genscan Genes, Squirrel monkey mRNAs, Other mRNAs, CpG Islands, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>13 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=gasAcu1></A>Stickleback Feb. 2006 (Broad/gasAcu1) (gasAcu1, Broad Institute gasAcu1 (1.0) (NCBI project 13579, AANH01000000))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl genes - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v74</td>
<td>23 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>02 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=gasAcu1&g=ensGene" target=_blank>Ensembl Genes v71</a></td>
<td>11 Jun 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=gasAcu1&g=chainNetFr3" target=_blank>Fugu Chain/Net</a></td>
<td>19 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v65</td>
<td>22 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=gasAcu1&g=chainNetAnoCar2" target=_blank>Lizard Chain/Net</a></td>
<td>12 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=gasAcu1&g=chainNetDanRer7" target=_blank>Zebrafish Chain/Net</a></td>
<td>01 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Chain/Net and liftOver file</td>
<td>12 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish Chain/Net  composite tracks</td>
<td>08 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Vertebrate genomes, Chain and Net alignment tracks (secondary release)</a></td>
<td>09 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 55</td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>18 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Medaka Chain and Net</td>
<td>19 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update (version 50)</td>
<td>18 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl v49 gene predictions</td>
<td>15 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Mouse (Jul. 2007/mm9) Chained Alignments
Mouse (Jul. 2007/mm9) Alignment Net</td>
<td>12 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
Chains Nets and liftOvers to fr2</td>
<td>11 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Initial gasAcu1 release (using Broad Institute gasAcu1 (1.0)):  Assembly, Gap, GC Percent, Quality, Short Match, Restr Enzymes, Ensembl Genes, Human Proteins, Stickleback mRNAs, Stickleback ESTs, Spliced ESTs, Non-Stickleback mRNAs, Conservation, RepeatMasker, WM+SDust Simple Repeats. Downloads: bigZips, database, liftOver, comparatives</td>
<td>28 Mar 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=tarSyr2></A>Tarsier Sep. 2013 (Tarsius_syrichta-2.0.1/tarSyr2) (tarSyr2, Washington University)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Green monkey Chain/Net</td>
<td>28 Oct 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=tarSyr2&g=chainNetMacFas5" target=_blank>Crab-eating macaque Chain/Net</a></td>
<td>07 Apr 2016</td>
</tr>
<TR valign=top><TD align=left>
Mouse lemur Chain/Net</td>
<td>12 Feb 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=tarSyr2&g=cons20way" target=_blank>Primate 20-way conservation track</a></td>
<td>22 May 2015</td>
</tr>
<TR valign=top><TD align=left>
Initial tarSyr2 release (using Washington University (GCA_000164805.2)): Assembly, Gap, GC Percent, INSDC, Genscan Genes, Other RefSeq, Tarsier mRNAs, Other mRNAs, CpG Islands, Interrupted Repeats, Simple Repeats, WM + SDust, RepeatMasker</td>
<td>02 Apr 2015</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=tarSyr1></A>Tarsier Aug. 2008 (Broad/tarSyr1) (tarSyr1, Broad Institute tarSyr1 (NCBI project 20335, ABRT000000000))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl gene predictions - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial tarSyr1 release (using Broad Institute tarSyr1 (NCBI project 20335, ABRT000000000)): Assembly, Gap, GC Percent, Quality Scores, Ensembl Assembly, Other RefSeq, Ensembl Genes, Genscan Genes, Tarsier mRNAs, CpG Islands, Gorilla Chain/Net, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>16 Nov 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=sarHar1></A>Tasmanian devil Feb. 2011 (WTSI Devil_ref v7.0/sarHar1) (sarHar1, Wellcome Trust Sanger Institute Devil_ref v7.0 (GCA_000189315.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial sarHar1 release (using Wellcome Trust Sanger Institute Devil_ref v7.0 (GCA_000189315.1)): Assembly, Gap, GC Percent, Short Match, Restr Enzymes, Other RefSeq, Genscan Genese, Tasmanian devil mRNAs, Other mRNAs, 
CpG Islands, Mouse Chain/Net, Repeat Masker, Interrupted Rpts, WM + SDust, Simple Repeats.</td>
<td>17 Jul 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=echTel2></A>Tenrec Nov. 2012 (Broad/echTel2) (echTel2, Broad/EchTel2.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Initial echTel2 release (using Broad/EchTel2.0): Assembly, Gap, GC Percent, Other RefSeq, Genscan Genes, CpG Islands, Mouse Chain/Net, Human Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>29 Aug 2013</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=echTel1></A>Tenrec July 2005 (Broad/echTel1) (echTel1, Broad July 2005)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl gene predictions - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=echTel1&g=chainNetHg19" target=_blank>Human Chain/Net</a></td>
<td>25 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
Initial echTel1 release (using Broad July 2005): Assembly, Gap, GC Percent, Quality Scores, Ensembl Assembly, Genscan Genes, CpG Islands, RepeatMasker, Interrupted Repeats, Simple Repeats, WM + SDust, Other RefSeq, Ensembl Genes, TransMap Alignments, Human Chain/Net, Mouse Chain/Net</td>
<td>29 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=tetNig2></A>Tetraodon Mar. 2007 (Genoscope 8.0/tetNig2) (tetNig2, Genoscope Tetraodon v8.0 (NCBI project 12350, CAAE01000000))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>02 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=tetNig2&g=chainNetFr3" target=_blank>Fugu Chain/Net</a></td>
<td>19 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=tetNig2&g=chainNetDanRer7" target=_blank>Zebrafish Chain/Net</a></td>
<td>17 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Initial tetNig2 release (using Genoscope Tetraodon v8.0 (NCBI project 12350, CAAE01000000)): Scaffolds, Gap, GC Percent, TransMap Genes, Ensembl Genes, Genoscope GAZE Genes, Human Proteins, mRNAs, other mRNAs, CpG Islands, Interrupted Rpts, Simple Repeats, Window Masker, Chains & Nets: Human, Mouse, Fugu, Stickleback, Medaka, Zebrafish</td>
<td>11 Mar 2010</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=tetNig1></A>Tetraodon Feb. 2004 (Genoscope 7/tetNig1) (tetNig1, Genoscope and Broad Institute V7 (NCBI project 12350, CAAE01000000))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Vertebrate genomes, Chain and Net alignment tracks (secondary release)</a></td>
<td>09 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
Medaka Chain and Net</td>
<td>19 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins Mapped by Chained tBLASTn</td>
<td>28 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Mouse (Jul. 2007/mm9) Chained Alignments
Mouse (Jul. 2007/mm9) Alignment Net</td>
<td>10 Sep 2007</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN gene predictions from WUStL</td>
<td>27 Aug 2007</td>
</tr>
<TR valign=top><TD align=left>
Chains Nets and liftOvers to fr2</td>
<td>11 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish Chain/Net/LiftOver (danRer3)</td>
<td>21 Feb 2006</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Human Ecores</td>
<td>15 Feb 2005</td>
</tr>
<TR valign=top><TD align=left>
Self Chain, Self Net</td>
<td>26 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Cytokine Genes (removed extensions)</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
HOX Genes (removed extensions)</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Genoscope GAZE (removed extensions)</td>
<td>10 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Geneid Gene Predictions</td>
<td>29 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Cytokine Genes</td>
<td>17 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
HOX Genes</td>
<td>15 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Genoscope GAZE gene predictions</td>
<td>15 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain, Net (mm5)</td>
<td>13 Sep 2004</td>
</tr>
<TR valign=top><TD align=left>
Initial tetNig1 release (Genoscope v6): Scaffolds, Gap, GC Percent, Short Match, Genscan, mRNAs, Spliced ESTs, ESTs, Xeno ESTs, Human Chain/Net, Fugu Chain/Net, RepeatMasker, Simple Repeats.
Downloads: bigZips, chromosomes, database, comparatives.</td>
<td>10 Sep 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=tupBel1></A>Tree shrew Dec. 2006 (Broad/tupBel1) (tupBel1, Broad Institute tupBel1 (NCBI project 13971, AAPY01000000))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Malayan flying lemur Chain/Net</td>
<td>16 Sep 2016</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene predictions - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Tarsier Chain/Net</td>
<td>22 May 2015</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial tupBel1 release (using Broad Institute tupBel1 (NCBI project 13971, AAPY01000000)): Assembly, Gap, GC Percent, Quality Scores, Ensembl Assembly, Other RefSeq, Ensembl Genes, Genscan Genes, Tree shrew mRNAs, Spliced ESTs, Tree shrew ESTs, CpG Islands, Mouse Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>16 Nov 2012</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=melGal1></A>Turkey Dec. 2009 (TGC Turkey_2.01/melGal1) (melGal1, TGC (NCBI Project ID: 10805, Accession: GCA_000146605.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=melGal1&g=chainNetTaeGut2" target=_blank>Zebra finch Chain/Net</a></td>
<td>08 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>02 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
Medium ground finch Chain/Net</td>
<td>17 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v65</td>
<td>22 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
Initial melGal1 release (using TGC (NCBI Project ID: 10805, Accession: GCA_000146605.1)): Assembly, Gap, GC Percent, CpG Islands, Repeat Masker, Interrupted Repeats, Simple Repeats, Human Proteins, N-SCAN, Other RefSeq, Turkey mRNAs, Spliced ESTs, Turkey ESTs, Other mRNAs, X. tropicalis Chain/Net, Zebrafish Chain/Net, Chicken Chain/Net, Human Chain/Net, Mouse Chain/Net, Opossum Chain/Net, Rat Chain/Net, Zebra finch Chain/Net, xenTro2 Chain/Net.</td>
<td>02 Nov 2011</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=macEug2></A>Wallaby Sep. 2009 (TWGS Meug_1.1/macEug2) (macEug2, TWGS (NCBI Project ID: 12586, Accession: GCA_000004035.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>02 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial macEug2 release (using TWGS (NCBI Project ID: 12586, Accession: GCA_000004035.1)): Assembly, Gap, GC Percent, Short Match, Restr Enzymes, Other RefSeq, Wallaby mRNAs, Spliced ESTs, Wallaby ESTs, Other mRNAs, RepeatMasker, Interrupted Repeats, WindowMasker + SDust, Simple Repeats</td>
<td>23 Jan 2012</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=cerSim1></A>White rhinoceros May 2012 (CerSimSim1.0/cerSim1) (cerSim1, Broad Institute)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Microsatellites - Di-nucleotide and Tri-nucleotide Repeats</td>
<td>18 Aug 2015</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Initial cerSim1 release (using Broad Institute): Assembly, Gap, GC Percent, Other RefSeq, Genscan Genes, White rhinoceros mRNAs, CpG Islands, Opossum Chain/Net, Cow Chain/Net, Mouse Chain/Net, Human Chain/Net, RepeatMasker, Interrupted Repeats, WM + SDust, Simple Repeats</td>
<td>24 Jan 2013</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=xenTro7></A>X. tropicalis Sep. 2012 (JGI 7.0/xenTro7) (xenTro7, US DOE Joint Genome Institute (JGI-PGF))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Initial xenTro7 release (using US DOE Joint Genome Institute (JGI-PGF)(GCA_000004195.2)): Assembly, Gap, GC Percent, INSDC, Augustus Genes, Genscan Genes, RefSeq Genes, XGC Genes, Other RefSeq, Spliced ESTs, X. tropicalis ESTs, X. tropicalis mRNAs, CpG Islands, Interrupted Repeats, Simple Repeats, Microsatellites, WM + SDust, RepeatMasker</td>
<td>21 Dec 2015</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=xenTro3></A>X. tropicalis Nov. 2009 (JGI 4.2/xenTro3) (xenTro3, US DOE Joint Genome Institute (JGI-PGF) (GCA_000004195.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=xenTro3&g=cons9way" target=_blank>Multiz Alignment & Conservation (9 Species)</a></td>
<td>09 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
Initial xenTro3 release (using US DOE Joint Genome Institute (JGI-PGF) (GCA_000004195.1)): Assembly, Gap, GC Percent, CpG Islands, Repeat Masker, Interrupted Repeats, Simple Repeats, Human Proteins, Microsatellite, WM + SDust, Genscan Genes, Scaffolds, RefSeq Genes, XGC Genes, X. tropicalis mRNAs, Spliced ESTs, X. tropicalis ESTs, Ensembl Genes, Lizard Chain/Net, Zebrafish Chain/Net, Chicken Chain/Net, Human Chain/Net, Turkey Chain/Net, Mouse Chain/Net, Opossum Chain/Net, Rat Chain/Net.</td>
<td>02 Nov 2011</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=xenTro2></A>X. tropicalis Aug. 2005 (JGI 4.1/xenTro2) (xenTro2, JGI v4.1)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href="http://text.soe.ucsc.edu" target=_blank>pubsBingBlat (a run of hg19-BLAT against all webpages on the internet indexed by bing.com)</a></td>
<td>23 Jan 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=xenTro2&g=chainNetAnoCar2" target=_blank>Lizard Chain/Net</a></td>
<td>12 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=xenTro2&g=chainNetDanRer7" target=_blank>Zebrafish Chain/Net</a></td>
<td>18 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish Chain/Net  composite tracks</td>
<td>09 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
LiftOver chain to/from monDom5 <--> xenTro2</td>
<td>05 Feb 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Vertebrate genomes, Chain and Net alignment tracks (primary release)</a></td>
<td>29 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update to version 55</td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>02 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Converted MGC track to PSL (to allow for alignment visualization)</td>
<td>22 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update (version 50)</td>
<td>19 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes, version 49</td>
<td>17 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Mouse (Jul. 2007/mm9) Chained Alignments
Mouse (Jul. 2007/mm9) Alignment Net</td>
<td>05 Nov 2007</td>
</tr>
<TR valign=top><TD align=left>
Added NCBI Clone Validation information to details of MGC genes</td>
<td>12 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
Microsatellite Track: Di-nucleotide and Tri-nucleotide Repeats</td>
<td>01 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>20 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Initial xenTro2 release (version 4.1 using JGI Jazzer Assembler): Assembly; Gap; GC Percent; Known Genes; RefSeq Genes; MGC Genes; Genscan Genes; X. tropicalis mRNAs; cpg Islands; Spliced ESTs; X. tropicalis ESTs;  
Chains & Nets: Zebrafish, Rat, Human, Mouse, Chicken; RepeatMasker;  Simple Repeats.

Downloads: bigZips, chromosomes, database, vsDanRer4, vsHg18, vsGalGal2, vsMm8, vsRn4</td>
<td>13 Jun 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=xenTro1></A>X. tropicalis Oct. 2004 (JGI 3.0/xenTro1) (xenTro1, JGI v3.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Converted MGC track to PSL (to allow for alignment visualization)</td>
<td>22 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Lift over to xenTro2</td>
<td>28 Feb 2007</td>
</tr>
<TR valign=top><TD align=left>
Added NCBI Clone Validation information to details of MGC genes</td>
<td>12 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
chain/net tracks to mm6</td>
<td>05 May 2005</td>
</tr>
<TR valign=top><TD align=left>
JGI Filtered Genes</td>
<td>18 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins (data update)</td>
<td>03 Mar 2005</td>
</tr>
<TR valign=top><TD align=left>
MGC Genes</td>
<td>14 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Genscan (removed extensions)</td>
<td>07 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
Frog Browser Initial Release (using JGI version 3.0): Assembly, Gap, GC Percent, Mapped Human, Genscan, Frog mRNA, Spliced ESTs, Frog ESTs, CpG Islands, Conservation, Most Conserved, Fugu Blat, Zebrafish Chain/Net, Mouse Chain/Net, Human Chain/Net, Chicken Chain/Net, RepeatMasker, Simple Repeats</td>
<td>17 Nov 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=taeGut2></A>Zebra finch Feb. 2013 (WashU taeGut324/taeGut2) (taeGut2, Washington University School of Medicine)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Initial taeGut2 release (using Washington University School of Medicine): Assembly, Gap, GC Percent, INSDC, RefSeq Genes, Other RefSeq, Genscan Genes, Zebra finch mRNAs, Spliced ESTs, Zebra finch ESTs, CpG Islands, Turkey Chain/Net, Chicken Chain/Net, Human Chain/Net, RepeatMasker, Interrupted Repeats, Simple Repeats, WM + SDust</td>
<td>08 May 2014</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=taeGut1></A>Zebra finch Jul. 2008 (WUGSC 3.2.4/taeGut1) (taeGut1, Washington University taeGut1)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Medium ground finch Chain/Net</td>
<td>17 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=taeGut1&g=chainNetMelGal1" target=_blank>Turkey Chain/Net</a></td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
TransMap Alignments
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Vertebrate genomes, Chain and Net alignment tracks (secondary release)</a></td>
<td>09 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes version 55</td>
<td>12 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
quality score fasta files</td>
<td>16 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
Chicken Chain/Net/LiftOver (galGal3)</td>
<td>15 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN gene predictions from Brent Lab at WUStL</td>
<td>28 Apr 2009</td>
</tr>
<TR valign=top><TD align=left>
Updated Quality Scores</td>
<td>04 Dec 2008</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins</td>
<td>15 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Initial taeGut1 release (using Washington University taeGut1):  Assembly, Gap, GC Percent, Quality Scores, RefSeq Genes, Non-zebra finch Refseq Genes, Zebra Finch ESTs, spliced ESTs, Zebra Finch mRNAs, non-Zebra Finch mRNAs, repeatMasker, windowMaskerSDust</td>
<td>28 Aug 2008</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=danRer10></A>Zebrafish Sep. 2014 (GRCz10/danRer10) (danRer10, Genome Reference Consortium Zebrafish Build 10)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Ensembl Genes version 86, October 2016</td>
<td>21 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
Chicken Chain/Net</td>
<td>26 Sep 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../cgi-bin/hgTrackUi?db=danRer10&g=grcIncidentDb" target=_blank>GRC Incident Database</a></td>
<td>11 Jul 2016</td>
</tr>
<TR valign=top><TD align=left>
Ensembl genes - 81</td>
<td>10 Sep 2015</td>
</tr>
<TR valign=top><TD align=left>
Initial danRer10 release (using Genome Reference Consortium Zebrafish Build 10): Assembly, Gap, GC Percent, INSDC, Genscan Genes, Other RefSeq, RefSeq Genes, ZGC Genes, Other mRNAs, Spliced ESTs, Zebrafish ESTs, Zebrafish mRNAs, CpG Islands, Human Chain/Net, Interrupted Repeats, RepeatMasker, Simple Repeats, WM + SDust</td>
<td>01 Apr 2015</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=danRer7></A>Zebrafish Jul. 2010 (Zv9/danRer7) (danRer7, Wellcome Trust Sanger Institute Zv9 (GCA_000002035.2))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
CRISPR</td>
<td>07 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v75</td>
<td>01 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://text.soe.ucsc.edu" target=_blank>pubsBingBlat (a run of hg19-BLAT against all webpages on the internet indexed by bing.com)</a></td>
<td>23 Jan 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v74</td>
<td>23 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>02 Dec 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v73</td>
<td>22 Oct 2013</td>
</tr>
<TR valign=top><TD align=left>
Accession name at the INSDC - International Nucleotide Sequence Database Collaboration</td>
<td>10 Sep 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v72</td>
<td>12 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v71</td>
<td>11 Jun 2013</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v70</td>
<td>01 May 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=danRer7&g=chainNetFr3 " target=_blank>Fugu Chain/Net</a></td>
<td>19 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=grcIncidentDb" target=_blank>GRC Incident</a></td>
<td>17 Apr 2012</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v65</td>
<td>22 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=danRer7&g=chainNetXenTro3" target=_blank>X. tropicalis Chain/Net</a></td>
<td>15 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=danRer7&g=chainNetMelGal1" target=_blank>Turkey Chain/Net</a></td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v64</td>
<td>07 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=danRer7&g=UMassChIP-SeqComposite" target=_blank>UMass ChIP-Seq</a></td>
<td>03 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=danRer7&g=cons8way" target=_blank>Multiz Alignment & Conservation; PhyloP; PhastCons (8 Species)</a></td>
<td>05 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 63</td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=danRer7&g=cpgIslandExt" target=_blank>CpG Islands (Islands < 300 Bases are Light Green)</a></td>
<td>04 Aug 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 62</td>
<td>12 May 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=danRer7&g=chainNetPanTro3" target=_blank>Chimp (Oct. 2010 (CGSC 2.1.3/panTro3)), Chain and Net Alignments</a></td>
<td>11 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=ncbiIncidentDb" target=_blank>NCBI Incident database</a></td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 61</td>
<td>10 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=danRer7&g=fishChainNet" target=_blank>Fish Chain and Net:  Stickleback, Medaka, Tetraodon, Fugu</a></td>
<td>04 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=danRer7&g=vertebrateChainNet" target=_blank>Vertebrate Chain and Net:  X. tropicalis, Mouse, Human</a></td>
<td>04 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
Initial danRer7 release (using Wellcome Trust Sanger Institute Zv9 (GCA_000002035.2)): Assembly, Contigs, Gap, GC Percent, Ensembl Genes, RepeatMasker, Interrupted Rpts, Simple Repeats, Human Proteins, Short Match, 
Restriction Enzymes, RefSeq Genes, Other RefSeq, Zebrafish mRNAs, Spliced ESTs, Zebrafish ESTs, Other mRNAs.</td>
<td>20 Jan 2011</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=danRer6></A>Zebrafish Dec. 2008 (Zv8/danRer6) (danRer6, Wellcome Trust Sanger Institute, Zv8 assembly (CAAK00000000.5))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
CpG Islands on All Sequence (unmasked sequence CpG island track)</td>
<td>30 May 2014</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 59</td>
<td>23 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
Conservation, Most Conserved</td>
<td>19 Jul 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 58</td>
<td>21 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update, version 57</td>
<td>20 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
liftOver file: danRer6 to danRer5</td>
<td>06 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Tetraodon Chain/Net and liftOver file</td>
<td>12 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
Human Chain/Net  composite tracks</td>
<td>09 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
Mouse Chain/Net  composite tracks</td>
<td>08 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
Stickleback Chain/Net  composite tracks</td>
<td>08 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
Initial danRer6 release (using Wellcome Trust Sanger Institute, Zv8 assembly (CAAK00000000.5)): Assembly, Gap, GC Percent, Ensembl Genes, CpG Islands, RepeatMasker, Interrupted Rpts, Simple Repeats, Human Proteins, Self Chain, Medaka Chain and Net, Human Chain and Net, TransMap, liftover danRer5 to danRer6</td>
<td>14 Jan 2010</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=danRer5></A>Zebrafish July 2007 (Zv7/danRer5) (danRer5, Wellcome Trust Sanger Institute, Zv7 assembly)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
TransMap Alignments update
</td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genomewiki.ucsc.edu/index.php/Hg19_Genome_size_statistics" target=_blank>Vertebrate genomes, Chain and Net alignment tracks (primary release)</a></td>
<td>29 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Opossum Chain and Net</td>
<td>25 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes v52</td>
<td>02 Feb 2009</td>
</tr>
<TR valign=top><TD align=left>
Medaka Chain and Net</td>
<td>19 Nov 2008</td>
</tr>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Converted MGC track to PSL (to allow for alignment visualization)</td>
<td>22 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl gene update (version 50)</td>
<td>18 Aug 2008</td>
</tr>
<TR valign=top><TD align=left>
TransMap</td>
<td>17 Jul 2008</td>
</tr>
<TR valign=top><TD align=left>
Ensembl v49 gene predictions</td>
<td>15 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
Human Chain/Net/LiftOver (hg18)</td>
<td>29 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Human Proteins Mapped by Chained tBLASTn</td>
<td>18 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Initial danRer5 release (using Wellcome Trust Sanger Institute, Zv7 assembly): Map Contigs; Assembly; Gap; GC Percent; Ensembl Genes; RefSeq Genes; ZGC Genes; Vega Genes; Vega Pseudogenes; Zebrafish mRNAs; Spliced ESTs; Zebrafish ESTs; NonZfish mRNAs; Wild Type Array; Affy Zebrafish; Chains & Nets: Fugu, Tetraodon, Mouse, Medaka; RepeatMasker; Interrupted Repeats; Simple Repeats. Downloads: bigZips, chromosomes, database.</td>
<td>15 Oct 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=danRer4></A>Zebrafish Mar. 2006 (Zv6/danRer4) (danRer4, Sanger Centre, Danio rerio Sequencing Project Zv6)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Converted MGC track to PSL (to allow for alignment visualization)</td>
<td>22 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Updated mouse column in Gene Sorter and in Orthologous Genes section of gene details pages to mm9 UCSC Genes</td>
<td>19 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Added CONTRAST gene predictions</td>
<td>17 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Radiation Hybrid Map</td>
<td>07 May 2007</td>
</tr>
<TR valign=top><TD align=left>
Chains Nets and liftOvers to fr2</td>
<td>11 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Update of hgBlastTab tables (to hg18) to coincide with UCSC Genes</td>
<td>06 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
Ensembl Genes</td>
<td>12 Feb 2007</td>
</tr>
<TR valign=top><TD align=left>
Added NCBI Clone Validation information to details of MGC genes</td>
<td>12 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
N-SCAN gene predictions from WUStL</td>
<td>01 Sep 2006</td>
</tr>
<TR valign=top><TD align=left>
Chimp (panTro2) Chains, Net and LiftOvers</td>
<td>31 Jul 2006</td>
</tr>
<TR valign=top><TD align=left>
Rat (rn4) Chain/Net/liftOver</td>
<td>21 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated frame annotation for Conservation track to fix error with spliced codons
</td>
<td>15 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
Genscan Genes</td>
<td>01 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
chains and nets to Mouse (mm8)</td>
<td>31 May 2006</td>
</tr>
<TR valign=top><TD align=left>
Initial danRer4 release (using Sanger Centre Build Zv6): Map Contigs; Assembly; Gap; GC Percent; RefSeq Genes; ZGC Genes; Zebrafish mRNAs; Spliced ESTs; Zebrafish ESTs; NonZfish mRNAs; Affy Zebrafish; Chains & Nets: Fugu, Tetraodon, X. tropicalis, Opossum, Human; RepeatMasker;  Simple Repeats.

Downloads: bigZips, chromosomes, database</td>
<td>25 May 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=danRer3></A>Zebrafish May 2005 (Zv5/danRer3) (danRer3, Sanger Centre, Danio rerio Sequencing Project Zv5)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
Added alignments of non-coding RNAs (NR_*) to RefSeq and Xeno-RefSeq tracks.</td>
<td>27 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
automated rebuilding of transcription start upstream fasta and MAF files based on genbank or refseq tracks.</td>
<td>16 Oct 2008</td>
</tr>
<TR valign=top><TD align=left>
Converted MGC track to PSL (to allow for alignment visualization)</td>
<td>22 Sep 2008</td>
</tr>
<TR valign=top><TD align=left>
Added NCBI Clone Validation information to details of MGC genes</td>
<td>12 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
BAC End Pairs/Singles (data update)</td>
<td>08 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
Updated frame annotation for Conservation track to fix error with spliced codons</td>
<td>15 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
liftOver file: danRer3 to danRer2</td>
<td>11 May 2006</td>
</tr>
<TR valign=top><TD align=left>
Zebrafish (Danio rerio) Initial Release (danRer3, May 2005, Zv5): Scaffolds, Contigs, Gap, BAC End Pairs/Singles, GC Percent, Short Match, Restr Enzymes, RefSeq Genes, ZGC Genes, Ensembl Genes, Zebrafish mRNAs, Spliced ESTs, Zebrafish ESTs, Other mRNAs, TIGR Gene Index, Affy Zebrafish, 5-way Conservation, Most Conserved, Human Proteins (hg17), Human Chain/Net (hg18), Mouse Chain/Net/LiftOver (mm7), Fugu Chain/Net/LiftOver (fr1), Tetraodone Chain/Net/LiftOver (tetNig1), Repeat Masker, Simple Repeats, Self Chain. Downloads: bigZips, database, comparatives, chromosomes, liftOvers.</td>
<td>17 Feb 2006</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
</TD></TR></TABLE>


<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('require', 'displayfeatures');
  ga('create', 'UA-4289047-6', 'auto');
  ga('send', 'pageview');

</script></BODY></HTML>
