<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=223 HEIGHT=185>
  <TR><TD ALIGN=RIGHT><A HREF="http://www.sanger.ac.uk/Projects/M_musculus/" 
    TARGET=_blank>
    <IMG SRC="/images/Mus_musculus.jpg" WIDTH=208 HEIGHT=160 ALT="Mouse"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1> <em>Mus musculus</em><BR>
    </FONT>
    <FONT SIZE=-2> (Photo courtesy of
      <A HREF="http://www.jax.org/" TARGET=_blank>The Jackson Laboratory</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The October 2003 mouse (<em>Mus musculus</em>) genome assembly 
is based on NCBI Build 32.

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a 
sequenced genomic clone, an mRNA or EST or STS marker, or 
a cytological band, a chromosomal coordinate range, or keywords 
from the GenBank description of an mRNA. The following list 
provides examples of various types of position queries for the 
mouse genome. See the 
<A HREF="../goldenPath/help/hgTracksHelp.html" 
TARGET=_blank>User's Guide</A> for more information. 
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><BR></TD>
    </TR>
	
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chr16</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays all of chromosome 16</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>chr16:1-5000000</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays first 5 million bases of chr 16</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <!--
    <TR><TD VALIGN=Top NOWRAP>D16Mit120</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays region around STS marker DMit16120 from the MGI consensus genetic map.
 			Includes 100,000 bases on each side as well.</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>D16Mit203;D16Mit70</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays region between STS markers D16Mit203 and D16Mit70.
			Includes 100,000 bases on each side as well.</TD>
    </TR>
-->

    <TR><TD VALIGN=Top NOWRAP>AW045217</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays region of EST with GenBank accession AW045217</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>Ncam2</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays region of genome with official MGI mouse genetic nomenclature Ncam2</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>

    <TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists transcribed pseudogenes but not cDNAs</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>kruppel zinc finger</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>huntington</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists candidate genes associated with Huntington's disease</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>Xu,C.S.</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by co-author C.S. Xu</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Xu CS</em>, internally it uses 
			the format <em>Xu,C.S.</em>.</TD>
    </TR>

</TABLE>
<BR>
<HR>
<H3>Assembly details</H3>
This assembly is a composite version in which chromosomes were 
assembled by two slightly different algorithms depending on the available mapping data. 
Chromosomes 2, 4, 5, 7, 11, 15, 18, 19, X, and Y were assembled using a 
clone-based tiling path file (TPF) provided by the Mouse Genome Sequencing
Consortium (MGSC), with whole genome shotgun sequence used 
to fill gaps when necessary. The remaining chromosomes were assembled using 
the MGSCv3 whole genome shotgun assembly as the TPF
and merging High Throughput Genomic Sequence (HTGS) as needed. 
The UCSC mm4 assembly contains only the reference strain C57BL/6J. 
<P>
Build 32 includes 2.6 gigabases of sequence &mdash; 1.2 Gb of which is 
finished &mdash; contained in chromosomes 1-19, X, Y, M (mitochondrial DNA) and 
Un (unmapped clone contigs). 
We estimate that 90-96 percent of the mouse genome is present in the assembly.
For more information about this version, see the NCBI 
<A HREF="http://www.ncbi.nlm.nih.gov/genome/assembly/grc/mouse/"
TARGET=_blank>assembly notes</A> and 
<A HREF="http://www.ncbi.nlm.nih.gov/mapview/stats/BuildStats.cgi?taxid=10090&build=32&ver=1" 
TARGET=_blank>Build 32 statistics</A>. For an analysis of some problems
found with this assembly, see the Ensembl 
<A HREF="http://www.ensembl.org/Mus_musculus/" TARGET=_blank><em>Mus musculus</em></A>
web page.
<P>
The mm4 sequence and annotation data may be downloaded from the UCSC Genome 
Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/mm4/">FTP server</A> or
<A HREF="/downloads.html#mouse">Downloads</A> web page.
The mm4 annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="../goldenPath/credits.html#mouse_credits">credits</A> page
for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>

