<P><H3>About the Mus musculus assembly</P></H3>

<P>The October 2003 mouse genome assembly (UCSC version mm4) 
is based on NCBI Build 32, a composite version in which chromosomes were 
assembled by two slightly different algorithms depending on the available mapping data. 
Chromosomes 2, 4, 5, 7, 11, 15, 18, 19, X, and Y were assembled using a 
clone-based tiling path file (TPF) provided by the Mouse Genome Sequencing
Consortium (MGSC), with whole genome shotgun sequence used 
to fill gaps when necessary. The remaining chromosomes were assembled using 
the MGSCv3 whole genome shotgun assembly as the TPF
and merging High Throughput Genomic Sequence (HTGS) as needed. 
The UCSC mm4 assembly contains only the reference strain C57BL/6J. 
<P>
Build 32 includes 705 megabases of finished sequence. 
We estimate that 90-96 percent of the mouse genome is present in the assembly.
For more information about this version, see the NCBI 
<A HREF="http://www.ncbi.nlm.nih.gov/genome/seq/NCBIContigInfo.html"
TARGET=_blank>assembly notes</A> and 
<A HREF="http://www.ncbi.nlm.nih.gov/mapview/stats/BuildStats.cgi?taxid=10090&build=32&ver=1" 
TARGET=_blank>Build 32 statistics</A>.
<P>
The mm4 sequence and annotation data may be downloaded from the UCSC Genome 
Browser <A HREF="ftp://genome.ucsc.edu/goldenPath/mm4/">FTP server</A>.
</P>
<P><H3>Sample position queries</P></H3>

<P>A genome position can be specified by the accession number of a sequenced genomic clone, an mRNA or EST or STS marker, or 
a cytological band, a chromosomal coordinate range, or keywords from the Genbank description of an mRNA. The following list provides examples of various types of position queries for the mouse genome. See the <A HREF="http://genome.cse.ucsc.edu/goldenPath/help/hgTracksHelp.html" TARGET=_blank>User Guide</A> for more help. 
<P>

<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
<TR><TD VALIGN=Top NOWRAP><B>Request:</B><br></TD>
	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><br></TD></TR>
	
<TR><TD VALIGN=Top><br></TD></TR>
	
<TR><TD VALIGN=Top NOWRAP>chr16</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays all of chromosome 16</TD></TR>
<TR><TD VALIGN=Top NOWRAP>chr16:1-5000000</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays first 5 million bases of chr 16</TD></TR>

<TR><TD VALIGN=Top><br></TD></TR>

<TR><TD VALIGN=Top NOWRAP>D16Mit120</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region around STS marker DMit16120
 from the MGI consensus genetic map.
 Includes 100,000 bases on each side as well.</TD></TR>
<TR><TD VALIGN=Top NOWRAP>D16Mit203;D16Mit70</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region between STS markers D16Mit203 and D16Mit70.
Includes 100,000 bases on each side as well.</TD></TR>
<TR><TD VALIGN=Top NOWRAP>AW045217</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region of EST with GenBank accession AW045217</TD></TR>
<TR><TD VALIGN=Top NOWRAP>Ncam2</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays region of genome with official MGI mouse genetic nomenclature Ncam2</TD></TR>
<TR><TD VALIGN=Top><br></TD></TR>

<TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists transcribed pseudogenes but not cDNAs</TD></TR>
<TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD></TR>
<TR><TD VALIGN=Top NOWRAP>kruppel zinc finger</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD></TR>
	<TD VALIGN=Top>Lists candidate genes associated with Huntington's disease</TD></TR>
<TR><TD VALIGN=Top NOWRAP>Evans,J.E.</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by co-author J.E.  Evans</TD></TR>

<TR><TD VALIGN=Top><br></TD></TR>
	
<TR><TD COLSPAN="3" > Use this last format for entry authors -- even though Genbank searches require Evans JE format, GenBank entries themselves use Evans,J.E.  internally.
</TABLE>


