<!-- Display image in righthand corner -->
<!--
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=185>
  <TR><TD ALIGN=RIGHT>
    <A HREF="*** link to sequencing center's project page"
    TARGET=_blank>
    <IMG SRC="/images/Haemonchus_contortus.jpg" WIDTH=*** HEIGHT=*** ALT="H. contortus"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Haemonchus contortus</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of ***
      (<A HREF="*** link to source" 
      TARGET=_blank>***source</A>)
    </FONT>
  </TD></TR>
</TABLE>
-->

<P>
The Dec. 2009 <em>Haemonchus contortus</em> (Barber Pole worm) draft assembly 
(Sanger Institute) was produced by the
<A HREF="http://www.sanger.ac.uk/Projects/H_contortus/"
TARGET=_blank>Wellcome Trust Sanger Institute</A> sequencing center.

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
H. contortus genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>ABKE01000001:1-100,000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first hundred thousand bases of contig ABKE01000001</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>ABKE01000001:100,000+2,000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of contig ABKE01000001 that spans 2000 bases, starting with position 100,000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>DQ270213</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number DQ270213</TD>
    </TR>
    <TR><TD VALIGN=Top>BQ088015</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		BQ088015</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>pseudogene</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes</TD>
    </TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>zhang</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zhang</TD>
    </TR>
    <TR><TD VALIGN=Top>Taguchi,T.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author T. Taguchi</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Taguchi T</em>, internally it
	uses the format <em>Taguchi,T.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
See also: Sanger Institute
<A HREF="http://www.sanger.ac.uk/Projects/H_contortus/"
TARGET=_blank>project information.</A>
This draft assembly is composed of 59,708 contigs for a total size of
297,989,404 nucleotides.  The largest contig, Hcon_Contig0000157, is
152,663 nucleotides.  One half of the sequence, 148,994,702 bases,
is contained in 5116 contigs, all of at least 13,340 bases (N50).
There are 19,144,420 bases of gap, indicated by "N", in the sequence 
</P>
<P>
The included mitochondrial sequence is from the genbank accession
<A HREF="http://www.ncbi.nlm.nih.gov/nuccore/195972419"
TARGET=_blank>NC_010383.2</A>.
</P>


<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/haeCon1/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#h_contortus">Downloads</A> page. 

The H. contortus browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#h_contortus_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
