<!DOCTYPE html>
<!--#set var="TITLE" value="Genome Browser bedMethyl Track Format" -->
<!--#set var="ROOT" value="../.." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<h1>BedMethyl Track Format</h1>
<p> 
The bedMethyl format allows display of continuous-valued data in track format. This display type is 
useful for probability scores and transcriptome data.  This track type is similar to the wiggle 
(<a href="/goldenPath/help/wiggle.html">WIG</a>) format, but unlike the wiggle format, data exported
in the bedMethyl format are preserved in their original state. This can be seen on export using the
table browser. For more details on data compression in wiggle tracks see the notes section of the
wiggle <a href="wiggle.html" target="_blank">track description page</a>. If you have a very large
data set and you would like to keep it on your own server, you should use the <a href="bigWig.html">
bigWig</a> data format. In fact, an attempt to load a bedMethyl custom track over 50,000,000 lines
will result in an error message, but can be addressed by turning the bedMethyl into a bigWig (see
<a href="bigWig.html#Ex3">Example 3</a>). Note that bedMethyl files cannot easily be converted to
wiggle files; converting bedMethyl to bigWig and using <code>bigWigToWig</code> will return the
original bedMethyl file.</p>

<h2>General Structure</h2>
<p>
The bedMethyl format is line-oriented. BedMethyl data are preceded by a 
<a href="customTrack.html#TRACK">track definition line</a>, which adds a number of options for 
controlling the default display of this track.</p> 
<p> 
Following the track definition line are the track data in four column BED format:</p>

<pre><code><em>chromA</em>  <em>chromStartA</em>  <em>chromEndA</em>  <em>dataValueA</em>
<em>chromB</em>  <em>chromStartB</em>  <em>chromEndB</em>  <em>dataValueB</em></code></pre>

<h3>Parameters for bedMethyl track definition lines</h3>
<p> 
All options are placed in a single line separated by spaces:</p>
<pre><code><strong>track type=</strong>bedMethyl <strong>name=</strong><em>track_label</em> <strong>description=</strong><em>center_label</em>
    <strong>visibility=</strong><em>display_mode</em> <strong>color=</strong><em>r,g,b</em> <strong>altColor=</strong><em>r,g,b</em>
    <strong>priority=</strong><em>priority</em> <strong>autoScale=</strong><em>on|off</em> <strong>alwaysZero=</strong><em>on|off</em> <strong>gridDefault=</strong><em>on|off</em>
    <strong>maxHeightPixels=</strong><em>max:default:min</em> <strong>graphType=</strong><em>bar|points</em> <strong>viewLimits=</strong><em>lower:upper</em>
    <strong>yLineMark=</strong><em>real-value</em> <strong>yLineOnOff=</strong><em>on|off</em>
    <strong>windowingFunction=</strong><em>maximum|mean|minimum</em> <strong>smoothingWindow=</strong>off|2-16</em></code></pre>
<p>
<strong>Note:</strong> if you copy/paste the above example, you must remove the line breaks.</p>
<p> 
The track type is REQUIRED, and must be <em>bedMethyl</em>:</p>
<pre><code><strong>type=</strong>bedMethyl</code></pre>
<p> 
The remaining values are OPTIONAL. The <a href="wiggle.html" target="_blank">wiggle</a> 
documentation contains details on these options. A functional description of these options can be 
seen in the <a href="hgWiggleTrackHelp.html" target="_blank">track configuration</a> description. 
(Custom tracks do not have interactive configuration options.)

<h3>Data Values</h3>
<p>
BedMethyl track data values can be integer or real, positive or negative values. The
chromosome coordinates are <a href="../../FAQ/FAQtracks.html#tracks1">zero-based, half-open</a>. 
This means that the first chromosome position is 0, and the last position in a chromosome 
of length <em>N</em> would be <em>N - 1</em>. The positions listed in the input data must be in 
numerical order, and only the specified positions will be graphed. bedMethyl format has four
columns of data: <pre><code><em>chrom chromStart chromEnd dataValue</em></code></pre></p> 
<p>
<h2>Example</h2>
<p>
This example specifies 9 separate data points in three tracks on chr19 in the region 49,302,001 to 
49,304,701. To view this example as a custom track in the Genome Browser, copy the text and paste 
it into the browser annotation track text box.</p> 
<pre><code>browser position chr19:49302001-49304701
browser hide all
browser pack refGene encodeRegions
browser full altGraph
#	300 base wide bar graph, autoScale is on by default == graphing
#	limits will dynamically change to always show full range of data
#	in viewing window, priority = 20 positions this as the second graph
#	Note, zero-relative, half-open coordinate system in use for bedMethyl format
track type=bedMethyl name="BedMethyl Format" description="BedMethyl format" visibility=full color=200,100,0 altColor=0,100,200 priority=20
chr19 49302000 49302300 -1.0
chr19 49302300 49302600 -0.75
chr19 49302600 49302900 -0.50
chr19 49302900 49303200 -0.25
chr19 49303200 49303500 0.0
chr19 49303500 49303800 0.25
chr19 49303800 49304100 0.50
chr19 49304100 49304400 0.75
chr19 49304400 49304700 1.00</code></pre>
<p>
<strong>Note:</strong>
The above example is a custom track that includes a <code>track type=</code> line that is 
specific for loading the data in the browser. This line will cause a raw bedMethyl data file to fail 
validation by other tools, such as <code>validateFiles</code>, outside of the browser.</p>

<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->
