<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: bigGenePred Track Format</TITLE>
	<LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../cgi-bin/hgMenubar"--></TD></TR>
<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp;  bigGenePred Track Format</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
		    <P>
The bigGenePred format stores annotation items that are a
linked collection of exons, much as 
<A HREF="../../FAQ/FAQformat.html#format1">BED</A> files indexed as bigBeds do,
but bigGenePred has additional information about the coding frames and other
gene specific information in eight additional fields.</p>
<p>
bigGenePred files are created using the program <TT>bedToBigBed</TT> with a
special AutoSQL file that defines the fields of the bigGenePred.   The
resulting bigBed files are in an indexed binary format.  The main advantage of
the bigBed files is that only portions of the files needed to display a
particular region are transferred to UCSC. So for large data sets, bigBed is
considerably faster than regular BED files.  The bigBed file remains on 
your web accessible server (http, https, or ftp), not on the UCSC server. 
Only the portion that is needed
for the chromosomal position you are currently viewing is locally cached as a 
&quot;sparse file&quot;.</P>

<H4><A name=bigGenePred></A>Big Gene Predictions</H4>
The following AutoSQL definition is used for bigGenePred gene prediction files. 
This is the <A HREF="examples/bigGenePred.as"><tt>bigGenePred.as</tt></A>
file defined by the <tt>-as=</tt> option when using <tt>bedToBigBed</tt>.
Click this <A HREF="examples/bigGenePred.txt"><tt>bed12+8</tt></A> file for
an example of bigGenePred input. In alternative-splicing situations, each
transcript has its own row.
<PRE>

table bigGenePred
"bigGenePred gene models"
   (
   string chrom;       "Reference sequence chromosome or scaffold"
   uint   chromStart;  "Start position in chromosome"
   uint   chromEnd;    "End position in chromosome"
   string name;        "Name or ID of item, ideally both human readable and unique"
   uint score;         "Score (0-1000)"
   char[1] strand;     "+ or - for strand"
   uint thickStart;    "Start of where display should be thick (start codon)"
   uint thickEnd;      "End of where display should be thick (stop codon)"
   uint reserved;       "RGB value (use R,G,B string in input file)"
   int blockCount;     "Number of blocks"
   int[blockCount] blockSizes; "Comma separated list of block sizes"
   int[blockCount] chromStarts; "Start positions relative to chromStart"
   string name2;       "Alternative/human readable name"
   string cdsStartStat; "enum('none','unk','incmpl','cmpl')"
   string cdsEndStat;   "enum('none','unk','incmpl','cmpl')"
   int[blockCount] exonFrames; "Exon frame {0,1,2}, or -1 if no frame for exon"
   string type;        "Transcript type"
   string geneName;    "Primary identifier for gene"
   string geneName2;   "Alternative/human readable gene name"
   string geneType;    "Gene type"
   )

</PRE>

<P>Note that the <TT>bedToBigBed</TT> utility uses a substantial amount of 
memory; somewhere on the order of 1.25 times more RAM than the 
uncompressed BED input file.</P>

<P>
To create a bigGenePred track, follow these steps:<BR>

<OL>
 <LI> Create a <tt>bed12+8</tt> bigGenePred format file that has the first twelve fields
  described by a normal BED file as described <A HREF="../../FAQ/FAQformat.html#format1">here</A>.
  (You can also read about genePred <A HREF="../../FAQ/FAQformat.html#format9">here</A>.)  
   <UL>
    <LI> Your bigGenePred file must have the extra eight fields described in the AutoSQL file above:
    <tt>name2, cdsStartStat, cdsEndStat, exonFrames, type, geneName, geneName2, geneType.</tt> 
    <LI> Your bigGenePred file must be sorted by chrom then chromStart. You can use
    the UNIX sort command to do this: 
    <TT>sort -k1,1 -k2,2n unsorted.bed &gt; input.bed</TT>
   </UL>
 <LI> Download the <TT>bedToBigBed</TT> program from the 
  <A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</A> 
  of binary utilities.
 <LI> Use the <TT>fetchChromSizes</TT> script from the same 
  <A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</A> 
  to create a chrom.sizes file for the UCSC database you are working with 
  (e.g. hg38). Alternatively, you can download the chrom.sizes file for
  any assembly hosted at UCSC from our <a href="http://hgdownload.soe.ucsc.edu/downloads.html">
  downloads</a> page (click on "Full data set" for any assembly). For example, for the hg38
  database, the hg38.chrom.sizes are located at
  <a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.chrom.sizes"
  target="_blank">http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.chrom.sizes</a>.
 <LI> Create the bigBed file from your sorted bigGenePred input file using the <TT>bedToBigBed</TT>
  utility like so: 
  <TT><B>bedToBigBed</B> -as=bigGenePred.as -type=bed12+8 bigGenePred.txt chrom.sizes myBigGenePred.bb</TT><BR>
 <LI> Move the newly created bigBed file (<TT>myBigGenePred.bb</TT>) to an http, 
  https, or ftp location.
 <LI> Construct a <A HREF="hgTracksHelp.html#CustomTracks">custom track</A> 
  using a single 
  <A HREF="hgTracksHelp.html#TRACK">track line</A>. 
  Note that any of the track attributes listed
  <A HREF="customTrack.html#TRACK">here</A> are applicable 
  to tracks of type bigBed. 
  The most basic version of the &quot;track&quot; line will look something 
  like this:<BR>
  <TT>track type=bigGenePred name="My Big GenePred" description="A Gene Set
  Built from Data from My Lab" bigDataUrl=http://myorg.edu/mylab/myBigGenePred.bb</TT>
 <LI> Paste this custom track line into the text box in the 
  <A HREF="../../cgi-bin/hgCustom">custom track</A> management page.
</OL>

The <tt>bedToBigBed</tt> program can also be run with several additional options.
A full list of the available options can be seen by running
<tt>bedToBigBed</tt> with no arguments to display the usage message.

<p>
<H3>Example One</H3>
</p>
<P>In this example, you will use an existing bigGenePred file to create a bigGenePred 
custom track.  A bigGenePred file that contains data on the hg38 
assembly has been placed on our http server.  
You can create a custom track using this bigGenePred file by constructing a 
&quot;track&quot; line that references this file like so:</P>

<TT>track type=bigGenePred name=&quot;bigGenePred Example One&quot; 
description=&quot;A bigGenePred file&quot; 
bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigGenePred.bb</TT>

<P>Paste the above &quot;track&quot; line into the 
<A HREF="../../cgi-bin/hgCustom?db=hg38">custom track</A> management page for the
human assembly hg38 (Dec. 2013), then press the <EM>submit</EM> button.
<p>
Custom tracks can also be loaded via one URL line.  The below link loads the same
bigGenePred track, but includes parameters on the URL line:
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38&hgct_customText=track%20type=bigGenePred%20name=Example%20bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigGenePred.bb"
target="_blank"><tt>http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38&hgct_customText=track%20type=bigGenePred%20name=Example%20bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigGenePred.bb</tt></a>
</p>
<p>
With this example bigGenePred loaded, click into a gene from the track.  Note
that the details page has a &quot;Links to sequence:&quot; section that includes
&quot;Translated Protein&quot;, &quot;Predicted mRNA&quot;, and &quot;Genomic
Sequence&quot; links. Click the &quot;Go to ... track controls&quot; link.
There change the &quot;Color track by codons:&quot; option from &quot;OFF&quot;
too &quot;genomic codons&quot; and be sure &quot;Display mode:&quot is &quot;full&quot;
then click &quot;Submit&quot;.  Then zoom to a region where amino acids display,
such as <tt>chr9:133,255,650-133,255,700</tt> and see how bigGenePred allows
the display of codons. Click back into the track controls page and click the box next
to &quot;Show codon numbering&quot;. Return to the browser to see amino acid
numbering.</p>
<p>
You can also add a parameter in the custom track line,
<tt>baseColorDefault=genomicCodons</tt>, to set the display of codons:
<p>
<tt>browser position chr10:67,884,600-67,884,900<br>
track type=bigGenePred baseColorDefault=genomicCodons name="bigGenePred Example Two" description="A bigGenePred file" visibility=pack bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigGenePred.bb
</tt>
<p>
Paste the above into the hg38 <A HREF="../../cgi-bin/hgCustom?db=hg38">custom track</A>
page to see an example of bigGenePred amino acid display around the beginning of the gene SIRT1
on chromosome 10.</p>
</p>

<H3>Example Two</H3>
<P>In this example, you will create your own bigGenePred file from an existing 
bigGenePred input file.

<UL>
 <LI> Save this <tt>bed12+8</tt> <A HREF="examples/bigGenePred.txt">bigGenePred.txt</A>
  example input file to your machine (<EM>satisfies above step 1</EM>).
 <LI> Download the <TT>bedToBigBed</TT> <A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/">utility</A>
 (<EM>step 2</EM>).
 <LI> Save this <tt>hg38.chrom.sizes</tt> <A HREF="hg38.chrom.sizes">text file</A> to your machine.
  It contains the chrom.sizes for the human (hg38) assembly 
  (<EM>step 3</EM>).
 <LI> Save this <tt>bigGenePred.as</tt> <A HREF="examples/bigGenePred.as"> text file</A> to your machine.
 <LI> Run the utility to create the bigBed output file 
  (<EM>step 4</EM>):<BR>
  <TT><B>bedToBigBed</B> -type=bed12+8 -tab -as=bigGenePred.as bigGenePred.txt hg38.chrom.sizes bigGenePred.bb</TT>
 <LI> Place the bigBed file you just created (<TT>bigGenePred.bb</TT>) on a 
  web-accessible server (<EM>step 5</EM>).
 <LI> Construct a &quot;track&quot; line that points to your bigGenePred file
  (see <EM>step 6</EM>).
 <LI> Create the custom track on the human assembly hg38 (Dec. 2013), and 
  view it in the genome browser (see <EM>step 7</EM>).
</UL>
Note the above description in <b>Example One</b> on how to view genomic codons,
including numbering. 

<H3>Sharing Your Data with Others</H3>
<P>
If you would like to share your bigGenePred data track with a colleague, learn
how to create a URL by looking at <B><EM>Example 11</EM></B> on
<A HREF="customTrack.html#SHARE">this</A> page.
</P>

<H3>Extracting Data from bigBed Format</H3>
<P>
Since the bigGenePred files are an extension of bigBed files, which are indexed binary files, 
they can be difficult to
extract data from.  We have developed the following 
programs, all of which are available from the 
<A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</A> of binary
utilities.<BR>
<UL>
 <LI><TT>bigBedToBed</TT> &mdash; this program converts a bigBed file 
      to ASCII BED format.
 <LI><TT>bigBedSummary</TT> &mdash; this program extracts summary information
      from a bigBed file.
 <LI><TT>bigBedInfo</TT> &mdash; this program prints out information about a 
      bigBed file.
</UL>
As with all UCSC Genome Browser programs, simply type the program name
at the command line with no parameters to see the usage statement.
</P>

<H3>Troubleshooting</H3>
<P>If you encounter an error when you run the <TT>bedToBigBed</TT> program,
it may be because your input bigGenePred file has data off the end of a chromosome.
In this case, use the <TT>bedClip</TT> program 
<A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/">here</A> before the
<TT>bedToBigBed</TT> program.  It will remove the row(s) in your input BED
file that are off the end of a chromosome.
</P>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	</TD></TR></TABLE>
</TD></TR></TABLE>
</BODY></HTML>
