<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: User's Guide</TITLE>
	<LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../inc/topbar.html"--></TD></TR>
<!-----------mRNA Codon Coloring Help Page------------------------------>
<A NAME="Topic5"></A><BR>

	<!--outer table is for border purposes-->
	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    		<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>	
			<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
			    <TR><TD><FONT SIZE="4"><A
                            NAME="TOC"></A><B>&nbsp; Codon and Base Coloring for Alignment Tracks</B></FONT>
			    </TD></TR>
			</TABLE>
			<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
			    <TR><TD WIDTH=10>&nbsp;</TD><TD>
				<P>
                                The Genome Browser's codon coloring feature allows
			 	users to quickly validate and compare alignment tracks, including 
				mRNAs. To turn on codon coloring, select the
				desired option from the <em>Color 
				track by codons</em> pull-down menu. 
				<p>
				<B>Color Options</B><BR>
				<ul>
                                <li><i>genomic codons</i> - codons are labeled and
			        colored according to the genomic sequence	
				</li>
                                <li><i>mRNA codons</i> - codons are labeled and colored
			        according to the aligned sequence	
				</li>
                                <li><i>nonsynonymous mRNA codons</i> - codons are 
				labeled and colored according to the aligned sequence. 
				Only nonsynonymous codons are labeled.
				Conservative and non-conservative substitutions are defined
				using the BLOSUM62 score matrix. Conservative 
				substitutions (yellow) are defined as positive
				BLOSUM scores, while non-conservative substitutions (red) are 
				defined as negative scores.
				</li>
                                <li><i>mRNA bases</i> - bases are labeled.	
				</li>
                                <li><i>different mRNA bases</i> - only those bases 
				that differ from the genomic sequence are labeled
				and colored.
				</li>
                                </ul></p><p> 
				<b>Color Legend</b><br><br>
				<table border="1" border-width="1" cellpadding="10">
				  <tr>
				    <th>&nbsp;</th>
				    <th>genomic/<br>mRNA codons</th>
				    <th>nonsynonymous<br>mRNA codons</th>
				    <th>mRNA bases/<br>different mRNA bases</th>
				  </tr>
				  <tr>
				    <td><font color="FF0000"><b>red</b></font></td>
				    <td>"*" stop codons</td>
				    <td>non-conservative<br>codon substitutions</td>
				    <td>base substitutions</td>
				  </tr>
				  <tr>
				    <td><font color="E0E000"><b>yellow</b></font></td>
				    <td>&nbsp;</td>
				    <td>conservative<br>codon substitutions</td>
				    <td>&nbsp;</td>
				  </tr>
				  <tr>
				    <td><font color="00DBDB"><b>cyan</b></font></td>
				    <td>spliced or truncated<br>partial codons</td>
				    <td>spliced or truncated<br>partial codons</td>
				    <td>&nbsp;</td>
				  </tr>
				  <tr>
				    <td><font color="00E000"><b>green</b></font></td>
				    <td>start codons<br>incl. methionine</td>
				    <td>&nbsp;</td>
				    <td>&nbsp;</td>
				  </tr>
				  <tr>
				    <td><b>black</b></td>
				    <td>"X" truncated<br>codons</td>
				    <td>&nbsp;</td>
				    <td>&nbsp;</td>
				  </tr>
				</table>
                                </p>
				<p>
				<B>Details</B><BR>
				The codon reading frame is defined by the genbank CDS
				annotation of the aligned sequence.
				Each codon is colored and labeled in the direction of 
				transcription. A sequence without a CDS will not be 
				colored; these can be non-protein coding RNAs or an
				mRNA where a CDS annotation was not provided in the
				original data.
				</p><p>
				The mRNA bases/different mRNA bases display options are  
				colored and labeled in the direction of the genomic sequence. Note 
				that it is possible to show the base complement, and
				thus change the base labeling, by clicking the arrow
				to the left of the base display or by clicking the
				"reverse" button.
				</p><p>
				When zoomed out past the base level, the browser will 
				choose one color to represent many bases. The priority of 
				display, from most important to least important, is: 
				different mRNA base/nonsynonmous codon coloring (if enabled), 
				and then <a href="hgIndelDisplay.html">alignment coloring</a>
				(if enabled).
				The browser will not display genomic/mRNA codon coloring when 
				viewing large regions of the genome. To view 
				labeling, the track must be zoomed to within 3 
				times the base level.
				</p><p>
				<b>For information about alignment insertion/deletion display options click 
				<a href="hgIndelDisplay.html">here</a>.</b>
				</p>
				<h2>References</h2>
				<p>Henikoff S and Henikoff JG.
				<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC50453/" target=_blank>
				Amino acid substitution matrices from protein blocks</a>.
				<em>PNAS</em>. 1992;89(22):10915-10919.<p>
			        </TD>
			        <TD WIDTH=15></TD>
			    </TR>
			</TABLE>
		    <BR></TD></TR>
		</TABLE>
	    </TD></TR>
	</TABLE>
    </TABLE>
</BODY></HTML>
