<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Gene Sorter: User's Guide</TITLE>
	<LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../inc/topbar.html"--></TD></TR>
<!-----Content Tables------------------------------------>  
<!--
    <TD CELLPADDING=10 ROWSPAN=2 VALIGN="top">
-->
    <TR><TD CELLPADDING=10 ROWSPAN=2 VALIGN="top">	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    		<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>	
			<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
			    <TR><TD>
			    <FONT SIZE="4"><B>&nbsp;UCSC Gene Sorter User's Guide </B></FONT> 
			    </TD></TR>
			</TABLE>
			<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
			    <TR><TD WIDTH=10>&nbsp;</TD><TD> 
				<B><FONT COLOR="#006666" SIZE=4> Table of Contents: </FONT></B>
				<TABLE BORDER=0 NOWRAP CELLPADDING=0 CELLSPACING=0 MARGINS=0>
			 	    <TR><TH HEIGHT=10></TH></TR>
				    <TR><TD>
				        <UL><B>
				  	    <LI><A HREF="#Intro">Introduction</A>
  					    <LI><A HREF="#Start">Getting Started</A>
					    <LI><A HREF="#Table">Understanding the Gene Sorter display</A>
					    <LI><A HREF="#Configure">Configuring the Gene Sorter display</A>
					    <LI><A HREF="#Filter">Filtering the gene display</A>
					    <LI><A HREF="#Download">Displaying sequence and text-based output</A>
				  	</B></UL></TD>
				    </TR>
				</TABLE>
				<P>
				This user's guide was last updated on 14 Dec. 2004. 
				<A HREF="mailto:&#103;&#101;&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;sc.e&#100;&#117;">Questions and feedback 
				are welcome</A>.
				</TD>
				<TD WIDTH=15></TD>
			    </TR>
			</TABLE>
	 	    <BR></TD></TR>
		</TABLE>
	    </TD></TR>
	</TABLE>

<!-----------Introduction-------------------------------------------->
<A NAME="Intro"></A><BR>

	<!--outer table is for border purposes-->
	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    		<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>	
			<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
			    <TR><TD><FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; Introduction</B></FONT>
			    </TD></TR>
			</TABLE>
			<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
			    <TR><TD WIDTH=10>&nbsp;</TD><TD>
				<P>
				Genes function and evolve together. To understand a gene,
				you often need to understand an entire gene family.
				Many such families are already known and described, such as the 
				HOX family that mediates many aspects of limb and brain development 
				and the Cytochrome P450 family that is central to the metabolism of many 
				medications.
				<P>
				One easy way to identify well-known relatives of a gene 
				is by looking for genes with name similarity, because biologists 
				tend to use similar names for similar genes. However, scientists
				only partly understand the function of perhaps one-third of
				the genes of the genome; therefore, other techniques for grouping genes
				into families are necessary.  
				<P>
				The UCSC Gene Sorter is an excellent resource for 
				exploring gene families and the relationships among genes. This 
				tool displays a table of genes within 
				a selected genome that are related to one another. Several 
				different relationships may be explored: protein-level homology, 
				similarity of gene expression profiles, or genomic proximity. The 
				Gene Sorter supports searches on a variety of terms and phrases, 
				including the gene name, the SwissProt protein name, a GenBank 
				accession, or a word or phrase present in a gene's description. The 
				gene family display is highly configurable, allowing the user to 
				control the order and number of columns, the number of rows, and 
				the genes displayed. The tool provides several output formats, 
				including a simple tab-delimited format that may be imported into a 
				spreadsheet or a relational database.
				<P>
				An important use of the Gene Sorter is to gather 
				together a collection of genes that share similar properties for
				statistical analysis.  For instance, one might want to examine
				promoter regions of genes that share a similar expression
				pattern  or look for protein sequence motifs in genes that
				share similar GO annotations.  
				<P>
				One of the most powerful features of the Gene Sorter is its 
				filtering capabilities. The filter enables the user to 
				quickly select an interesting subset of the 25,000 genes in 
				the genome based on a variety of detailed and flexible selection criteria. 
				For example, the filter 
				may be used to select all human genes over-expressed in the cerebellum 
				that have GO-annotated G-protein coupled receptor activity. 
				<P>
				The Gene Sorter was designed and implemented by Jim Kent, 
				Fan Hsu, David Haussler, and the UCSC Genome Bioinformatics Group. 
				This work is supported by a grant from the National Human Genome 
				Research Institute and by the Howard Hughes Medical Institute.
			        </TD>
			        <TD WIDTH=15></TD>
			    </TR>
			</TABLE>
		    <BR></TD></TR>
		</TABLE>
	    </TD></TR>
	</TABLE>

<!-----------Getting Started----------------------------------------->
<A NAME="Start"></A><BR>

	<!--outer table is for border purposes-->
	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    		<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>	
			<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
			    <TR><TD><FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; Getting Started</B></FONT>
			    </TD></TR>
			</TABLE>
			<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
			    <TR><TD WIDTH=10>&nbsp;</TD><TD>
				<P>
				To begin using the Gene Sorter, you will first
				have to select a genomic region and the type of gene relationship
				you wish to display.  
				You may also want to change some of the Gene Sorter's configuration
				settings to tailor the display to your research needs. These
				configuration options are described in <A HREF="#Configure">Configuring
				the Gene Sorter display</A>.
				<P>
				<B>Starting the Gene Sorter</B>
				<OL>
				<LI>Open the Gene Sorter
				<A HREF="../../cgi-bin/hgNear" TARGET=_BLANK">home 
				page</A>. 
				<LI>Specify the genome and assembly you wish to view by selecting 
				the appropriate options from the <em>genome</em> and 
				<em>assembly</em> pull-down menus.
				<LI>Type a term or phrase into the <em>search</em> text box to
				determine which genes will be displayed in the browser. Valid search
				terms (with human genome examples) include:
				    <UL>
				    <LI>a gene name (HOXA9)
				    <LI>a SwissProt protein name (HXA9) 
				    <LI>a word or phrase that occurs in the description of a gene (MAP kinase)
				    <LI>a GenBank mRNA accession (U14680)
				    </UL>
				<LI>Choose the gene relationship that you would like to examine by 
				selecting an option from the <em>sort by</em> pull-down menu. Genes will
				be sorted in order of proximity to the chosen gene, based on one of the
				following criteria:
				    <UL>
<!--Commented out til implemented
				    <LI>Expression (GNF Atlas2) -- similarity in gene expression, based on GNF 
					Atlas 2 data
				    <LI>Expression (UCLA) -- similarity in gene expression, based on UCLA data
-->
				    <LI>Expression (GNF Atlas1) -- similarity in gene expression, based on GNF 
					Atlas 1 data
				    <LI>Protein Homology - BLASTP -- similarity in protein homology, based on the 
					BLASTP E-value
				    <LI>Protein Homology - Rankprop -- similarity in protein homology, based
					on the Rankprop algorithm
				    <LI>Protein Homology - PSI-BLAST -- similarity in protein homology, based on 
					the PSI-BLAST E-value
				    <LI>Pfam Similarity -- similarity based on number of shared domains
				    <LI>Gene Distance -- absolute distance (left or right) on the chromosome 
					from the selected gene
				    <LI>Chromosome -- list sorted by chromosomal location
				    <LI>Name Similarity -- similarity to the name of selected gene, based on 
					the first several characters of the name
				    <LI>Alphabetical -- list sorted by gene name
				    <LI>GO Similarity -- number of Gene Ontology (GO) terms shared with 
					selected gene 
				    </UL>
				<LI>Choose the number of items to display from the <em>display</em> 
				pull-down menu (the default is 50).
				<LI>Press the <em>Go!</em> button to display your search results.
				</OL>
			        </TD>
			        <TD WIDTH=15></TD>
			    </TR>
			</TABLE>
		    <BR></TD></TR>
		</TABLE>
	    </TD></TR>
	</TABLE>

<!-----------Understanding the Gene Sorter display----------------------->
<A NAME="Table"></A><BR>       
                                
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>            
                <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                    <TR><TD>    
                        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%> 
                            <TR><TD><FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; Understanding 
                                the Gene Sorter display</B></FONT>
                            </TD></TR>
                        </TABLE>
                        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                            <TR><TH HEIGHT=10></TH></TR>
                            <TR><TD WIDTH=10>&nbsp;</TD><TD>
                                <P>
                                The main page of the Gene Sorter displays a table
				containing rows of genes and associated attributes. In most 
				cases, the currently-selected gene is shown at the top of 
				the list, highlighted in light green. The remaining genes
				are ordered relative to the selected gene based on the sort
				criteria specified in the <em>sort by</em> menu. For example,
				in a table sorted by gene distance, the genes are listed in
				order of greater to lesser chromosomal proximity to the 
				selected gene.
				<P>				
				The initial Gene Sorter display shows only a default subset of
				the columns available. The set of columns may be expanded,
				reduced, and rearranged by using the Gene Sorter's 
				<A HREF="#Configure">configuration utility</A>. To view 
				information about the data shown in the column, click on
				the column's label.
				<P>
				To select a different gene in the table, click on the name of
				the gene. The Gene Sorter will move the gene entry to the top of the 
				list and highlight it. The remaining genes will be reordered
				relative to the new selection.
                              	<P>  
				<B>Column descriptions</B> (listed in alphabetical order)
				<UL>
				<LI>
				<B># column:</B> Displays the position of each gene in the table,
				which is useful when examining data in tables with
				many rows. Clicking on the gene number selects it and moves
				it to the top of the list.<BR>
				<LI>	
				<B>3' UTR Fold column:</B> Shows the estimated 
				energy in kcal/mol of folding the 3' UTR of a 
				gene into the best predicted secondary structure. 
				The energy calculations and secondary structure 
				predictions were obtained from the RNAfold 
				program, which is part of the Vienna RNA 
				Package.<BR> 
				<LI>	
				<B>5' UTR Fold column:</B> Shows the estimated 
				energy in kcal/mol of folding the 5' UTR of a 
				gene into the best predicted secondary structure. 
				The energy calculations and secondary structure 
				predictions were obtained from the RNAfold 
				program, which is part of the Vienna RNA 
				Package.<BR> 
                              	<LI>  
				<B>Abundance column (Yeast):</B> Displays protein 
				abundance information, as reported in Ghaemmaghami
				et al. <A HREF="http://www.nature.com/cgi-taf/DynaPage.taf?file=/nature/journal/v425/n6959/abs/nature02046_fs.html" 
				TARGET=_blank>Global analysis of protein 
				expression in yeast</A>, <em>Nature</em> 
				<B>425</B>(6959), 737-741 (2003). For more 
				information, see the 
				<A HREF="http://yeastgfp.ucsf.edu/" 
				TARGET=_blank>Yeast GFP Fusion Localization 
				Database</A>.<BR> 
                              	<LI>  
				<B>Arbeitman et al. 2002 Life-Cycle Expression Data column (Fruitfly):</B>
				Shows the median ratio of gene expression in various phases of the fly life cycle
				relative to the expression of the gene in mixed egg-to-adult cultures.
				The level of detail may be
				increased or decreased with the <A HREF="#Configure">configuration
				utility</A>. See the <A HREF="#Expression">Expression columns</A> 
				description for more information about the display and configuration 
				of this column. For more details about the experiments and methods used to 
				create this data, click on the column's label.<BR>
				<LI>	
				<B>BLASTP Bits column:</B> Shows the bit score measure of protein similarity
				between a gene and the selected gene. The greater the 
				similarity between two proteins, the higher the bit score. For
				more information about how the bit score was calculated, click
				on the Bits column label.<BR>
                              	<LI>  
				<B>BLASTP E-Value column:</B> Shows the Blastp E-value (expectation value) 
				between each gene and the selected gene. The greater the 
				similarity of two proteins, the lower the E-value is. 
				Identical long proteins have an E-value of zero. 
				Formally, an E-value is the number of other known genes that 
				are expected to have at least this level of homology by chance.
				An E-value of less than 0.1 can be safely interpreted as the
				probability that a match this good would occur merely by
				chance.  For more information about how the E-value was 
				calculated, click on the E-Value column label.
				Clicking on a gene's E-value displays an alignment
				between the gene and the selected (highlighted)
				gene.<BR>
                              	<LI>  
				<B><em>C. elegans</em> column:</B> Shows the best Blastp match to the WormBase 
				protein set. Clicking on the ID number displays the 
				corresponding WormBase database record.<BR> 
                              	<LI>  
				<B>Coding SNPs column:</B> Shows the Simple Nucleotide Polymorphisms (SNPs) 
				located in the coding region of each gene. Clicking on a SNP 
				ID displays the SNP record associated with the gene in 
				NCBI's dbSNP.<BR> 
                              	<LI>  
				<B>Description column:</B> Shows a brief description of each gene taken from 
				its mRNA record. Clicking on a description displays a page 
				showing additional information and links for the gene.<BR>
                              	<LI>  
				<B>Drosophila column:</B> Shows the FlyBase ID of the best Blastp match to 
				the FlyBase protein set. Clicking on the FlyBase ID displays
				the corresponding FlyBase report.<BR> 
                              	<LI>  
				<B>Ensembl column:</B> Shows the Ensembl transcript ID associated with 
				the gene. Ensembl is an automatic gene prediction pipeline and
				a major genome database and web site run by the Sanger
				Institute and the European Bioinformatics Institute (EMBL-EBI).
				It is especially effective at mapping genes that are known 
				in one organism to another organism. Compared to other gene
				predictions, those of Ensembl tend to have high specificity 
				but low sensitivity to genes not already associated with
				characterized mRNA or protein sequence. Clicking on an 
				Ensembl ID displays the Ensembl GeneView page for the gene.<BR> 
                              	<LI>  
				<B>Entrez Gene (Human):</B> Formerly LocusLink. Shows 
				the NCBI Entrez Gene
				ID associated with the gene. Clicking on the entry
				displays the Entrez Gene record. If the record shows a 
				link to the Online Mendelian Inheritance in Man 
				(OMIM) -- indicated by an orange square with an 
			 	"O" inside it -- an OMIM record is available for 
				the gene.<BR>
                              	<LI>  
				<B>Exon Count column:</B> Displays the number of
				exons in the gene (coding and non-coding).<BR>
                              	<LI>  
				<B>Exp Delta column:</B> Shows the similarity of the expression of 
				each gene to the selected gene. Genes with identical expression 
				profiles have a value of 0. This column shows data only for the 
				1000 genes (including splicing variants) that have the most 
				similar expression profiles.  For more information about how 
				expression distance was calculated, click on the Exp Delta column 
				label.<BR>
                              	<LI>  
				<A NAME="Expression"></A>
				<B>Expression columns:</B> Show the ratio of 		
				expression of the gene in selected tissues or life cycle stages
				to the expression of the gene overall. 
				A gene that is more highly expressed is colored red, and a less 
				expressed gene is shown in green. The values are colored on a 
				logarithmic scale. This coloring is standard, but is the opposite 
				of what an inexperienced user might expect: in this case, red means go 
				and green means stop! Black indicates that a gene is neither 
				over nor under expressed in the tissue. Uncolored boxes 
				(white on most browsers) represent missing data. 
				Various attributes of the expression column displays may be configured 
				using the <A HREF="#Configure">configuration utility</A>. Depending on the 
				organism, the user
				may adjust the color scheme and brightness, toggle between expression
				ratio and absolute expression values, and increase or decrease the 
				level of detail shown. In particular, color-blind users 
				may wish to switch the coloring from red/green to yellow/blue.
				For more information about the selection criteria used for expression	
				columns, click on the column's label.<BR>
                              	<LI>  
				<B>GenBank column:</B> Shows the GenBank RefSeq or mRNA accession number
				associated with the gene. Clicking on the accession number 
				displays the GenBank record associated with it.<BR>
                              	<LI>  
				<B>Gene Ontology column:</B> Shows the Gene Ontology (GO) terms associated with 
				the gene. GO terms are words from a controlled vocabulary 
				assigned to a gene by human curators. Clicking on a GO term 
				displays the associated Gene Ontology Consortium database 
				record.<BR> 
                              	<LI>  
				<B>Genome Position column:</B> Shows the chromosomal location of each gene in 
				the genome. Clicking on a chromosomal position displays the 
				gene at that location in the UCSC Genome Browser.<BR> 
                              	<LI>  
				<B>GNF1M ID column (Mouse):</B> Shows the 
				Affymetrix ID from the GNF1M chip that best 
				corresponds to each gene. For more details about 
				the experiments and methods used to 
				create this data, click on the column's label.<BR>
                              	<LI>  
				<B>GNF Atlas2 column (Human):</B> Shows the ID of 
				the probe in the GNF Atlas 2 that overlaps most 
				with the selected gene. The GNF Altas 2 is based 
				on two Affymetrix chips: the U133A and a 
				custom-designed GNF1H chip. <BR>
				<LI>
				<B>GNF Delta column:</B> Shows the similarity of the 
				expression of each gene to the selected gene. 
				Genes with identical expression profiles have a 
				value of 0. This column shows data only for the 
				1000 genes (including splicing variants) that 
				have the most similar expression profiles.<BR> 
                              	<LI>  
				<B>GNF U74a, GNF U74b, GNF U74c columns (Mouse):</B> Shows data
				from the Mouse Gene Expression Atlas from the Genomics Institute of 
				the Novartis Research Foundation (GNF) on the Affymetrix U74a, U74b,
				and U74c chips. By default, the columns display the median ratio of 
				expression in a specific set of tissues relative to the expression 
				of the gene overall. The level of detail may be
				increased or decreased with the <A HREF="#Configure">configuration
				utility</A>. Currently, the full spectrum of tissues is available only 
				on the U74a chip. 
				See the <A HREF="#Expression">Expression columns</A> 
				description for more information about the display and configuration 
				of this column. For more details about the experiments and methods used to 
				create this data, click on the column's label.<BR>
                              	<LI>  
				<B>GNF U95 column (Human):</B> Shows data from from the GNF Expression 
				Atlas on the Affymetrix U95 chip. By default, the column
				displays the median ratio of expression in a specific set of tissues
				relative to the expression of the gene overall. The level of detail may be
				increased or decreased with the <A HREF="#Configure">configuration
				utility</A>. See the <A HREF="#Expression">Expression columns</A> 
				description for more information about the display and configuration 
				of this column. For more details about the experiments and methods used to 
				create this data, click on the GNF U95 column's label.<BR>
                              	<LI>  
				<B>Human column (Mouse):</B> Shows that best Blastp match to the known genes
				protein set from the UCSC Human Genome Browser database. 
				Clicking the accession number displays the gene in the UCSC 
				Human Genome Browser.<BR> 
				<LI>
				<B>Kim-Lab Life-Cycle Expression Data (All) column (<em>C. elegans</em>):</B>
				Shows the ratio of gene in all phases of the worm life cycle relative to 
				the expression of the gene in mixed wild-type adult cultures.
				See the <A HREF="#Expression">Expression columns</A> 
				description for more information about the display and configuration 
				of this column. For more details about the experiments and methods used to 
				create this data, click on the column's label.<BR>
				<LI>
				<B>Kim-Lab Life-Cycle Expression Data (Median) column (<em>C. elegans</em>):</B>
				Shows the median ratio of gene expression in selected phases of the worm life cycle. 
				See the <A HREF="#Expression">Expression columns</A> 
				description for more information about the display and configuration 
				of this column. For more details about the experiments and methods used to 
				create this data, click on the column's label.<BR>
                              	<LI>  
				<B>Max GNF Atlas 2 column:</B> Shows the 
				maximum absolute expression level for any tissue 
				in the GNF Gene Expression Atlas 2. Most of the 
				values fall in the zero to 50,000 range, but a 
				few outliers may range as high as 200,000. A 
				value of less than 20 indicates the expression 
				could not be detected in any tissue at levels 
				significantly above the cross-hybridization 
				controls.<BR>
                              	<LI>  
				<B>Max GNF U95 column (Human):</B> Shows the 
				maximum absolute expression level for any tissue. 
				Most values range between
				zero and 30,000, but a few outliers may range as
				high as 52,000. A value of less than 20 indicates
				the expression could no be detected in any 
				tissue at levels significantly above the 
				cross-hybridization controls. 
                              	<LI>  
				<B>Max Rinn Sex column (Mouse):</B> Shows the 
				maximum expression value in adult male and female 
				mouse tissue as described in (Rinn et al., 
				Developmental Cell, 2004). For more information
				about the methods used to generate these data,
				click on the Max Rinn Sex column header.
				<LI>	
				<B>Module column (Yeast):</B> Shows the predicted 
				regulatory module (a combination of transcription 
				factor binding sites) that regulates the gene. 
				The approach underlying this annotation is 
				described in Segal, E. et al., 
				<A HREF="http://robotics.stanford.edu/~erans/publications/segal_promoters.pdf"
				TARGET=_blank>Genome-wide discovery of 
				transcriptional modules from DNA sequence and gene
				expression</A>, <em>Bioinformatics</em> 
				<B>19</B>(Suppl 1), i273-i282 (2003). For more
				information about the methods used, click the
				Module column header. To view genes that share
				a regulatory module, select the <em>Regulatory 
				Module</em> option from the <em>sort by</em> 
				menu.<BR>
				<LI>	
				<B>MOE430 ID column (Mouse):</B> Shows the 
				Affymetrix ID from the MOE430 series of chips 
				(A & B) that best corresponds to each gene.
				For more details about the experiments and methods
				used to create this data, click on the column's 
				label.<BR>
				<LI>	
				<B>Mouse column (Human):</B> Shows the accession 
				number of the best Blastp match to the known genes 
				protein set in the UCSC Mouse Genome Browser database. 
				Clicking on the accession number displays the gene in 
				the UCSC Mouse Genome Browser.<BR> 
				<LI>
    				<B>Name column:</B> Displays the name of the gene. When possible, 
				the HUGO Gene Nomenclature Committee (HGNC) name is shown. If the gene does not yet have an HGNC 
				name, the GenBank accession number of the associated RefSeq 
				or mRNA record is shown instead. Clicking on the gene name
				selects it and moves it to the top of the list.<BR>
                             	<LI>  
				<B>% ID column:</B> Shows the percentage identity at the protein 
				level between the gene and the selected gene. For more 
				information about how the % ID was 
				calculated, click on the % ID column label.<BR>
                              	<LI>  
				<B>PDB column:</B> Displays all Protein Data 
				Bank (PDB) IDs associated with the gene. PDB is a
				database of proteins with known 3-D structures. 
				In some cases these records will correspond 
				to only a fragment of the gene. In other cases 
				the PDB record may include other molecules with 
				which the protein interacts. Clicking on a PDB
				entry displays the associated PDB Structure
				Explorer page.<BR>
                              	<LI>  
				<B>Pfam Domains column:</B> Shows a list of 
				Protein Family (Pfam) domains contained in the gene 
				product. Clicking on a domain displays the associated 
				Pfam record.<BR> 
                              	<LI>  
				<B>PSI-BLAST E-Value column:</B> Shows the 
				PSI-BLAST E-value (expectation value) between the
				Swiss-Prot or TrEMBL protein associated with the 
				gene and the protein associated with the selected 
				(highlighted) gene. The greater the similarity of 
				two proteins, the lower the E-value is. Identical 
				long proteins have an E-value of zero. 
				For more information about how the E-value was 
				calculated, click on the E-Value column label.
				Clicking on a gene's E-value displays an alignment
				between the gene and the selected (highlighted)
				gene.<BR>
                              	<LI>  
				<B>Rankprop column:</B> Displays protein 
				similarity scores assigned by the Rankprop 
				algorithm. The scores reported in this column 
				range from zero to one, with one being the most 
				significant. Currently, Rankprop does not report 
				an E-value statistic. For more information on this
				algorithm, click on the Rankprop column
				label.<BR>
                              	<LI>  
				<B>RefSeq column:</B> Displays the NCBI RefSeq 
				accession associated with the gene, if available. 
				RefSeq genes are a non-redundant set of
				high-quality mRNA sequences. Clicking on the accession number
				displays the NCBI Entrez Gene record for a RefSeq accession. If
				the record shows a link to the Online Mendelian Inheritance in
				Man (OMIM) -- represented by an orange box with an "O"
				on it -- an OMIM record is available for the gene. OMIM is 
				often an excellent source of human-curated information about
				human genes.<BR>
				<LI>
				<B>Regulatory Motif column (Yeast):</B> Shows the 
				predicted transcription factor binding sites that 
				correlate with expression of the gene. 
				regulatory module (a combination of transcription 
				factor binding sites) that regulates the gene. 
				The approach underlying this annotation is 
				described in Segal, E. et al., 
				<A HREF="http://robotics.stanford.edu/~erans/publications/segal_promoters.pdf"
				TARGET=_blank>Genome-wide discovery of 
				transcriptional modules from DNA sequence and gene
				expression</A>, <em>Bioinformatics</em> 
				<B>19</B>(Suppl 1), i273-i282 (2003). For more
				information about the methods used, click the
				Module column header.<BR>
				<LI>	
				<B>SGD ORF:</B> Shows the Saccharomyces Genome 
				Database (SGD) ID associated with the gene.<BR>
                              	<LI>  
				<B>SP Acc column:</B> Shows the Swiss-Prot 
				protein accession of a gene. Clicking on an entry
				displays the corresponding Swiss-Prot NiceProt
				view of the protein.
                              	<LI>  
				<B>Superfamily column:</B> Displays a list of 
				Structural Classification of Protein (SCOP)
				superfamilies associated with a protein. The gene 
				set was mapped to SCOP superfamilies using the 
				Superfamily HMM library. Clicking on an entry
				displays the associated Superfamily record.<BR>
                              	<LI>  
				<B>Swiss-Prot column:</B> Displays the Swiss-Prot 
				protein name of each gene, 
				if it is available. Otherwise, it shows the primary 
				accession number. Clicking on a protein name or accession 
				displays the corresponding Swiss-Prot NiceProt view of the
				protein.<BR>
                              	<LI>  
				<B>U133 ID column (Human):</B> Shows the Affymetrix ID 
				from the HG-U133 chip that best 
				corresponds to each gene. For more information about 
				the selection criteria used for this column, click on 
				the U133 ID column's label.<BR>
                              	<LI>  
				<B>U133Plus2 ID column (Human):</B> Shows the 
				Affymetrix ID from the HG-U133 Plus 2.0 chip that 
				best corresponds to each gene. For more 
				information about the selection
				criteria used for this column, click on the 
				U133Plus2 ID column's label.<BR>
                              	<LI>  
				<B>U74 ID column (Mouse):</B> Shows the Affymetrix ID 
				from the U74 series of chips (a, b, c) that best 
				corresponds to each gene. For more information about 
				the selection criteria used for this 
				column, click on the U74 ID column's label.<BR>
                              	<LI>  
				<B>U95 ID column (Human)</B>: Shows the Affymetrix ID 
				from the HG-U95 chip that best corresponds to each gene. 
				For more information about the selection criteria used 
				for this column, click on the U95 ID column's label.<BR>
                              	<LI>  
				<B>UCLA Long Expression column (Human):</B> Shows UCLA 
				expression data from normal tissues on the U133 chip. 
				This column shows the ratio of expression of a gene in the entire 
				tissue set relative to the expression of the gene overall.
				See the <A HREF="#Expression">Expression columns</A> 
				description for more information about the display and configuration 
				of this column. For more information about the methods used to 
				generate this data, click on the UCLA Long Expression column's label.<BR>
                              	<LI>  
				<B>UCLA Short Expression column (Human):</B> Shows UCLA 
				expression data from normal tissues on the U133 chip. 
				This column shows the ratio of expression of a gene in a particular
				subset of tissues relative to the expression of the gene overall.
				See the <A HREF="#Expression">Expression columns</A> 
				description for more information about the display and configuration 
				of this column. For more information about the methods used to 
				generate this data, click on the UCLA Short Expression column's label.<BR>
                              	<LI>  
				<B>WormBase column (<em>C. elegans</em>):</B> Shows the ORF name 
				associated with each gene.  Clicking on an 
				ORF name displays the associated WormBase record.<BR> 
                              	<LI>  
				<B>Yeast column:</B> Shows the best Blastp match to the Saccharomyces 
				Genome Database (SGD) protein set. Clicking on the yeast ID 
				displays the corresponding SGD record.<BR> 
                              	<LI>  
				<B>Zebrafish column:</B> Shows the Ensembl peptide ID of the best Blastp 
				match to the Ensembl gene predictions on the zebrafish genome. 
				Clicking on the Ensembl peptide ID displays it in Ensembl 
				Zebrafish Protein View.
				</UL>
				</TD>
                                <TD WIDTH=15></TD>
                            </TR>
                        </TABLE>
                    <BR></TD></TR>
                </TABLE>        
            </TD></TR>          
        </TABLE>           

<!-----------Configuring the Gene Sorter display------------------------->
<A NAME="Configure"></A><BR>

	<!--outer table is for border purposes-->
	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    		<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>	
			<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
			    <TR><TD><FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; Configuring the 
				Gene Sorter display</B></FONT>
			    </TD></TR>
			</TABLE>
			<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
			    <TR><TD WIDTH=10>&nbsp;</TD><TD>
				<P>
				The Gene Sorter is highly configurable, allowing you
				to fine-tune the display to show just the genes and data
				columns in which you're interested in an order that best suits your research
				needs. Most of the configuration is controlled through settings on
				the Configuration page, accessed via the <em>configure</em> button at
				the top of the Gene Sorter page.
				<P>
				<B>Changing the number of rows displayed</B><BR>
				To increase or decrease the number of rows shown in the table, pick a
				new value from the <em>display</em> pull-down menu, then click the <em>Go!</em> 
				button.
				<P>
                                <B>Changing the number of columns displayed</B><BR>
                                By default, the Gene Sorter shows only a small subset of the
				table columns available for the genome. You can view the full list of columns,
				or add or remove columns from your display, on the Configuration page.
				<P>The configuration table shows all the
				columns available for the currently-selected genome, listed in left-to-right
				display order. To add or remove a column from the Gene Sorter display, click the
				<em>On</em> checkbox to toggle the setting (a check indicates that the column
				is displayed). To quickly change the <em>On</em> settings of all columns,
				click the <em>Hide All</em> or <em>Show All</em> button at the top of the page. Click the 
				<em>Submit</em> button to display the changes in the Gene Sorter.
				<P>
				<B>Changing the column positions</B><BR>
				In addition to adding or removing columns, it is 
				also possible to move the columns to the left or right within the Gene Sorter
				table. The order of the column names in the configuration table indicates the
				current relative position of the columns in the Gene Sorter display from left to
				right. To shift a column one position to the left, click the <em>up</em> arrow in the
				item's <em>Position</em> column. Similarly, click the <em>down</em> arrow to shift a 
				column to the right. When you have finished making changes, click the <em>Submit</em>
				button.
				<P>
				<B>Changing the expression colors</B><BR>
				By default, the gene expression ratios are shown using a red/green color 
				scheme,
				where red indicates a gene that is more highly expressed and green corresponds
				to less expression. Color-blind users may find it helpful to switch the 
				coloring from red/green to yellow/blue. To do so, select the &quot;yellow high/blue
				low&quot; option from the <em>Expression ratio colors</em> pull-down menu on the 
				Configuration page, then click the <em>Submit</em> button.
				<P>
				<B>Changing the brightness of expression colors</B><BR>
				To increase or decrease the brightness of the colors in an expression 
				column, edit the <em>brightness</em> value for the corresponding entry in the 
				configuration table. Values greater than 1.0 increase the brightness, while 
				those less than 1.0 dim the color. Click the <em>Submit</em> button to display the 
				new values.
				<P>
				<B>Changing the type of tissue data shown in expression columns</B><BR>
				By default, the expression columns show the median ratio of expression
				of a gene in a small selected set of tissues. Use the the <em>tissues</em> 
				pull-down menu to configure the tissue display for the column. 
				The &quot;all replicas&quot; option will show the value
				of each individual experimental replica of each tissue. The &quot;median 
				of replicas&quot; option displays a single value for each tissue that 
				represents the median of all replicas for that tissue.
				<P>
				<B>Toggling between ratio and absolute expression values </B><BR>
				By default, expression columns show the ratio of expression of a gene relative
				to expression of the gene overall. To view absolute expression values instead,
				select the &quot;absolute&quot; option from the <em>values</em> pull-down menu. 
				<P>
				<B>Displaying splicing variants</B><BR>
				By default, the Gene Sorter shows only one splicing variant: the one that produces
				the largest protein. To show all splicing variants, click the <em>Show all 
				splicing variants</em> checkbox. Note that in most cases, the column values
				(and sometimes the names) will be identical across variants.
				<P>
				<B>Restoring the default settings</B><BR>
				At any time during your Gene Sorter session, you can restore the Gene Sorter table
				to its default layout by clicking the <em>Default</em> button on the Configuration page,
				then clicking <em>Submit</em>.
				<P>
				<B>Saving a configuration for future use</B><BR>
				The Gene Sorter configuration utility allows you to store multiple configurations
				for use in future sessions. This feature is particularly useful if you require
				different layouts for different research uses.
				To save the current configuration of the Gene Sorter layout, click the <em>Save</em> button
				on the Configuration page. Type in a name for the configuration in the text
				box at the top of the page, then click <em>Save</em>. 
				<P>
				<B>Loading a previously-saved configuration</B><BR>
				Once you have saved a configuration, you can load it back into your Gene Sorter
				in a future session. To load a configuration, click the <em>Load</em> button on the
				Configuration page. The Gene Sorter will display a list of the names of your saved 
				configurations. Click on a name to highlight it, then click <em>Load</em> to 
				reconfigure your Gene Sorter based on the saved settings.
				<P>
				<B>Viewing a list of saved configurations</B><BR>
				To display a list of configurations that you have saved, click the <em>Save</em> button
				on the Configuration page. If you have any saved configurations, the Gene Sorter 
				will display an <em>Existing Setups</em> list that shows the configuration names. To
				permanently remove a configuration from the list, click on the name to 
				highlight it, then click the <em>Delete Existing Setup</em> button.
			        </TD>
			        <TD WIDTH=15></TD>
			    </TR>
			</TABLE>
		    <BR></TD></TR>
		</TABLE>
	    </TD></TR>
	</TABLE>

<!-----------Filtering the gene display------------------------------>
<A NAME="Filter"></A><BR>

	<!--outer table is for border purposes-->
	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    		<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>	
			<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
			    <TR><TD><FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; Filtering the gene display</B></FONT>
			    </TD></TR>
			</TABLE>
			<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
			    <TR><TD WIDTH=10>&nbsp;</TD><TD>
				<P>
				The Gene Sorter's gene filtering capabilities provide a versatile
				way to fine-tune the display to show just the genes in which you are
				interested. Filters are applied to individual gene fields, and may be
				combined to increase the specificity of the search. To access the Filter page, 
				click the <em>filter</em> button at the top of the Gene Sorter page. 
				<P>
				At any time during the filter setup process, you can click the <em>List Names</em>
				button on the Filter page to view a list of genes
				that will be returned when the current filter settings are applied to the
				genome. You may find this list helpful in fine-tuning the filter. 
				<P>
				<B>Filtering based on matching one or more terms</B><BR>
				Filters based on names, IDs, or other words restrict the display 
				to only those genes that match one or more terms typed into the 
				<em>search</em> text box. 
				Examples of values that can be filtered on this basis include the gene
				name, RefSeq accession number, gene description, coding SNPs, and GO terms.
			 	<P>
				This search supports wildcard matching on &quot;*&quot; and &quot;?&quot;.
				Multiple terms must be separated by a space or tab. For example,
				the search criteria "HOXA9 FOX*" on the gene name field returns the gene 
				named HOXA9 and any gene whose name begins with the letters &quot;FOX&quot;. 
				When searching on fields that consist of values containing more than one 
				word (GO terms, coding SNPs, Pfam domains, and gene descriptions), the 
				multi-word
				elements must be enclosed in single quotes. For instance, a search on the
				description phrase &quot;forkhead box protein&quot; should be entered as
				'forkhead box protein'.
				Use the 
				&quot;any&quot; and &quot;all&quot; options to determine whether the search should 
				return any gene that matches any term (&quot;any&quot;) or only those genes that 
				match all terms (&quot;all&quot;).
				<P>
				To facilitate searching on multiple terms, the Gene Sorter provides
				the option to paste in or upload a list of search terms. To paste in a list
				of terms, click the filter's <em>Paste List</em> button, then paste or type the 
				terms into the text box. Terms must be separated by a space, a tab, or be 
				entered on separate lines, and may not include wildcards. When you have 
				completed the list, click the Submit button to return to the main Filter page.
				The file upload utility - accessed via the <em>Upload List</em> button - has a similar
				functionality.
				<P>
				<B>Filtering based on numerical ranges</B><BR>
				Several of the gene fields can be filtered by specifying a numerical range
				within which the value must fall. Examples of
				fields in this category include expression ratios, Blastp data, and genome
				position. To use this type of filter, enter the minimum and maximum
				values delimiting the range in which you are interested. In some cases, the
				range of valid values is indicated in the filter box. 
				<P>
				The genome position filter requires the name of a chromosome (in the 
				format <em>chrN</em>) in addition
				to the chromosomal start and end positions. To list all genes on a chromosome,
				enter only the chromosome name.
				<P>
				Expression filters include &quot;any&quot; and &quot;all&quot; options to 
				determine whether the search should 
				return a gene if any of the tissue expression values meet the minimum and
				maximum criteria (&quot;any&quot;) or only if all tissue expression values meet 
				the search criteria (&quot;all&quot;).
				<P>
				<B>Saving filter settings</B><BR>
				The Gene Sorter provides a mechanism for saving filter settings for use in future
				sessions. To preserve the current filter configuration, click the <em>Save Filter</em>
				button on the Filter page. Type in a name for the filter, then click <em>Save</em>
				to save the filter and return to the Filter page.
				<P>
				<B>Loading a saved filter</B><BR>
				Once a filter configuration has been saved, you can retrieve it in later
				sessions by loading it back into your Gene Sorter. To load the saved filter
				settings, click the <em>Load Filter</em> button on the Filter page. Click on the
				name of the filter you wish to load, then click the <em>Load</em> button. Click
				the <em>Submit</em> button on the Filter page to apply the filter settings to the
				Gene Sorter.
				<P>
				<B>Viewing a list of saved filters</B><BR>
				To display a list of filter settings that you have saved, click the <em>Save</em> button
				on the Filter page. If you have any saved filters, the Gene Sorter 
				will display an <em>Existing Setups</em> list that shows the filter names. To
				permanently remove a filter from the list, click on the name to 
				highlight it, then click the <em>Delete Existing Setup</em> button.
			        </TD>
			        <TD WIDTH=15></TD>
			    </TR>
			</TABLE>
		    <BR></TD></TR>
		</TABLE>
	    </TD></TR>
	</TABLE>

<!-----------Displaying sequence and text-based output--------------->
<A NAME="Download"></A><BR>

	<!--outer table is for border purposes-->
	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    		<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>	
			<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
			    <TR><TD><FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; Displaying sequence and 
				text-based output</B></FONT>
			    </TD></TR>
			</TABLE>
			<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
			    <TR><TD WIDTH=10>&nbsp;</TD><TD>
				<P>
				The Gene Sorter's graphical presentation of data facilitates the visual
				observation of relationships and patterns among the genes in the
				display. However, it is often useful to convert the data to a text-based
				format that can be easily saved to a file or loaded into another program,
				database, or spreadsheet for further analysis. The Gene Sorter provides
				a mechanism for saving the current display in a tab-delimited text file or
				showing a text-based view of the sequence underlying the current display.
				<P>
				<B>Creating text-based output</B><BR>
				To output the current Gene Sorter table as text, click the text button at the
				top of the page. The Gene Sorter will display each row of table data 
				on a separate tab-delimited line.
				<P>
				<B>Viewing the underlying sequence</B><BR>
				To display the protein, mRNA, or genomic sequence underlying the current
				Gene Sorter table, click the sequence button at the top of the 
				page. On the Get Sequence page, select the desired sequence configuration 
				settings that you'd like, then click the <em>Get Sequence</em> button. The Gene Sorter will 
				display a text-based list of FASTA format records for each gene displayed
				in the table. The FASTA records may be cut and pasted into
				Blat for further study.
			        </TD>
			        <TD WIDTH=15></TD>
			    </TR>
			</TABLE>
		    <BR></TD></TR>
		</TABLE>
	    </TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>

</BODY></HTML>
