<!DOCTYPE html>
<!--#set var="TITLE" value="Table Browser Help" -->
<!--#set var="ROOT" value="../.." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<h1>Table Browser User's Guide </h1>

<h2>Contents</h2> 

<h6><a href="#Introduction">Introduction</a></h6> 
<h6><a href="#Tables">About the Table Browser databases and tables</a></h6> 
<ul> 
  <li><a href="#Positional"><strong>Position-oriented tables</strong></a></li> 
  <li><a href="#NonPositional"><strong>Non-positional tables</strong></a></li> 
</ul> 
<h6><a href="#GettingStarted">Getting started - simple queries</a></h6> 
<ul> 
  <li><a href="#PositionQuery"><strong>Simple position-based query</strong></a></li> 
  <li><a href="#BatchQuery"><strong>Batch query using identifiers</strong></a></li>
  <li><a href="#GetInfoFromPositions"><strong>Batch query using positions</strong></a></li>
  <li><a href="#GetGeneSymbols"><strong>Query to get gene symbols</strong></a></li> 
</ul> 
<h6><a href="#Filter">Filtering output by constraining field values</a></h6> 
<ul> 
  <li><a href="#FilterSingle"><strong>Filtering on fields from a single table</strong></a></li> 
  <li><a href="#FilterMultiple"><strong>Filtering on fields from multiple tables</strong></a></li> 
  <li><a href="#FilterConstraints"><strong>Filter constraints</strong></a></li> 
</ul> 
<h6><a href="#Intersection">Intersecting data from multiple tables</a></h6> 
<ul> 
  <li><a href="#SimpleIntersection"><strong>Intersecting data from two tables</strong></a></li> 
  <li><a href="#MultiIntersection"><strong>Intersecting data from multiple tables</strong></a></li> 
  <li><a href="#IntersectionOptions"><strong>Intersection options</strong></a></li> 
</ul> 
<h6><a href="#Correlation">Correlating data from two tables</a></h6> 
<h6><a href="#OutputFormats">Output formats</a></h6> 
<ul> 
  <li><a href="#AllFields"><strong>Displaying all fields in a table</strong></a></li> 
  <li><a href="#SelectedFields"><strong>Displaying selected fields from one or more 
  tables</strong></a></li>
  <li><a href="#Sequence"><strong>Displaying sequence (FASTA) data</strong></a></li> 
  <li><a href="#FASTA"><strong>Displaying CDS FASTA alignments</strong></a></li> 
  <li><a href="#GTF_BED"><strong>Saving query results in GTF or BED format</strong></a></li> 
  <li><a href="#SaveFile"><strong>Saving data to a file</strong></a></li> 
  <li><a href="#CustomTrack"><strong>Saving data as a custom track</strong></a></li> 
  <li><a href="#Hyperlink"><strong>Displaying query results as Genome Browser
  hyperlinks</strong></a></li>
  <li><a href="#SummaryStats"><strong>Displaying a statistical summary of query
  data</strong></a></li>
</ul>
<h6><a href="#Videos">Video examples of Table Browser queries</a></h6>
<ul>
  <li><a href="#listOfGenes"><strong>Find list of genes in a region
  </strong></a></li>
  <li><a href="#codonSeq"><strong>Obtaining coordinates and sequences of gene exons
  </strong></a></li>
  <li><a href="#snpsInGene"><strong>Find SNPs in a gene
  </strong></a></li>
  <li><a href="#snpsUpstream"><strong>Find SNPs upstream of Genes
  </strong></a></li>
</ul>
<hr>

<form name="googleForm1" method="get" action="https://www.google.com/search" onSubmit="document.googleForm1.q.value=document.googleForm1.qq.value+'   site:genome.ucsc.edu/goldenPath/help';"> 
  <p>
  Search the Genome Browser help pages: &nbsp; 
  <input type="hidden" name="q" value=""> 
  <input type="hidden" name="num" value="10"> 
  <input type="hidden" name="filter" value="0"> 
  <Input type=text name=qq size=30 maxlength=255 value=""> 
  <input type="submit" value="Submit"> 
</p>
</form> 
<p> 
<a href="../../contacts.html">Questions and feedback are welcome</a>.</p>

<!-- ====Introduction======================== -->
<a name="Introduction"></a> 
<h2>Introduction</h2>
<p> 
The Table Browser provides a powerful and flexible graphical interface for querying and manipulating
the Genome Browser annotation tables. Because the Table Browser uses the same database as the Genome
Browser, the two views are always consistent.</p>  
<p> 
Using the Table Browser, you can: 
<ul> 
  <li> 
  retrieve the DNA sequence data or annotation data underlying Genome Browser tracks for the entire 
  genome, a specified coordinate range, or a set of accessions</li> 
  <li> 
  apply a filter to set constraints on field values included in the output</li> 
  <li> generate a <a href="/goldenPath/help/hgTracksHelp.html#CustomTracks">custom track</a> and 
  automatically add it to your session so that it can be graphically displayed in the Genome
  Browser</li>
  <li> 
  conduct both structured and free-from SQL queries on the data</li> 
  <li> combine queries on multiple tables or custom tracks through an intersection or union and 
  generate a single set of output data</li> 
  <li> 
  display basic statistics calculated over a selected data set</li> 
  <li> 
  display the schema for table and list all other tables in the database connected to the table</li>
  <li> 
  organize the output data into several different formats for use in other applications, 
  spreadsheets, or databases</li> 
</ul> 
<p> 
This User's Guide is aimed at both the novice Table Browser user as well the advanced user. If you 
are new to the Table Browser, read the <a href="#GettingStarted">Getting started</a> section to 
learn about browser basics and try some simple queries. Advanced users may want to proceed directly 
to the section that addresses a particular area of functionality in detail.</p> 
<p> 
Although the Table Browser provides sufficient flexibility to satisfy the needs of most users, some 
advanced users may require the ability to run SQL commands directly on the Genome Browser database.
UCSC provides two public MariaDB servers: (1) genome-mysql.gi.ucsc.edu (US West Coast),
(2) genome-euro-mysql.soe.ucsc.edu (Europe). More information can be found on our
<a href="mysql.html">MariaDB Access</a> page. Alternatively, the database may be
downloaded to a local computer for MariaDB access. See the <a href="mirror.html">mirror 
site</a> documentation for information on setting up a local copy of the database.</p>

<!-- ====Tables======================== -->
<a name="Tables"></a>
<h2>About the Table Browser databases and tables</h2> 
<p> 
The Table Browser is built on top of the Genome Browser database</a>, which actually consists of 
several separate databases, one for each genome assembly.</p>  
<p> 
Tables within the databases may be differentiated by whether the data are based on genome start-stop
coordinates (positional tables) or are independent of position (non-positional tables).Some output 
formats and query options are applicable only to positional tables, hence the distinction.</p> 
<!-- ====Positional========================== -->
<a name="Positional"></a>
<h3>Positional tables</h3>
<p>
Positional tables contain data associated with specific locations in the genome, such as mRNA
alignments, gene predictions, cross-species alignments, and other annotations. Each of the
annotation tracks displayed in the Genome Browser is based on a positional table. In some instances,
data from other positional and non-positional tables may also be incorporated into the track. Data
associated with custom annotation tracks active within the user's Table Browser session are also
available as positional tables.</p>
<p>
Positional tables can be further subdivided into several categories based on the type of data they
describe. Alignment data can be best described by using a block structure to represent each element.
Other tables require only start and end coordinate data for each element. Some tables specify a
translation start and end in addition to the transcription start and end. Some tables contain strand
information, others don't. Most tables, but not all, specify a name for each element. Based on the
format of the data described by a table, different query and output formatting options may be
offered.</p>
<!-- ====Non-positional======================= -->
<a name="NonPositional"></a> 

<h3>Non-positional tables</h3> 
<p> 
Non-positional tables contain data not tied to genomic location, for example a table that correlates
a Known Gene ID with a RefSeq accession ID. Some non-positional tables relate internal numeric mRNA 
IDs to extended information such as author, tissue, or keyword.  Some &quot;meta&quot; tables in 
this category contain information about the structure of the database itself or describe external 
files containing sequence data.</p>  

<!-- REPLACE OR REMOVE
<p> 
For descriptions of the Genome Browser database tables, see the 
<a href="/goldenPath/gbdDescriptions.html">annotation 
		database</a> documentation.
-->

<!-- ====Getting Started===================== -->
<a name="GettingStarted"></a>
<h2>Getting started - simple queries</h2> 
<p> 
In its most basic form, the Table Browser can be used to retrieve a specific subset of records from 
a track or positional table in a selected genome assembly. The query may be based on a specific 
position or a set of one or more identifiers.</p> 
<p> 
This section describes the steps required to conduct basic simple data queries using the Table 
Browser. Once you have mastered the basic Table Browser functionality, refer to subsequent sections 
for information about generating more complex queries that use filters, intersections, and 
alternative data output formats.</p>

<!-- ====Position-based query================ -->
<a name="PositionQuery"></a>
<h3>Simple position-based query</h3> 
<p> 
Follow these steps to display a list of records that lie within a specific position in a table:</p> 
<p> 
<strong>Step 1. Pick a genome assembly</strong><br> 
Specify the genome assembly from which you'd like to retrieve the data by choosing the appropriate 
organism in the <code>genome</code> list, then selecting the assembly version from the 
<code>assembly</code> list. Note that the <code>assembly</code> list refreshes each time a different
option is selected in the <code>genome</code> list. Assemblies are typically named after the first 
three characters of an organism's genus and species names.</p> 
<p> 
<strong>Step 2. Pick an annotation track</strong><br> 
The <code>group</code> list shows all the annotation track groups available in the selected genome 
assembly. The names correspond to the groupings displayed at the bottom of the Genome Browser 
annotation tracks page. When a group is selected from the list, the <code>track</code> list 
automatically updates to show all the annotation tracks available within that group.</p> 
<ul> 
  <li> 
  If you already know the name of the annotation track in which you're interested, select the 
  <em>All Tracks</em> option in the <code>group</code> list, then select the track from the 
  <code>track</code> list. Similarly, you can directly select a table by choosing the 
  <em>All Tables</em> option in the <code>group</code> list, selecting a database from the 
  <code>database</code> list, then selecting the table from the <code>table</code> list.</li>  
  <li> 
  To examine all the tracks available within a certain group (e.g., all gene prediction tracks), 
  select the group name from the <code>group</code> list, then browse the entries in the 
  <code>track</code> list.</li>  
  <li> 
  Custom annotation tracks created during the current session are listed under the <em>Custom 
  Tracks</em> group.</li>  
  <li> 
  If no selections are made from the <code>group</code> or <code>track</code> lists, the track 
  selection defaults to the <em>Known Genes</em> track in the <em>Genes and Gene Prediction 
  Tracks</em> group.  
</ul> 
<p> 
<strong>Step 3. Pick a table</strong><br> 
The <code>table</code> list shows all tables (both positional and non-positional) associated with 
the currently-selected track. By default, it displays the primary table for the track, i.e. the 
table containing the data shown in the Genome Browser annotation track. Other tables in the list are
linked to the primary table by a common field and may provide supporting data used in constructing 
the annotation.</p>  
<ul> 
  <li> 
  If the <code>group</code> list is set to the <em>All Tables</em> option, the tables list will show
  all tables present in the database currently selected in the <code>database</code> list, rather 
  than those associated with a particular track.</li>  
</ul> 
<p> 
<strong>Step 4. Pick a genomic region</strong> (positional tables only)<br> 
By default, the Table Browser region is set to <code>genome</code>, which will display all the data 
records in the selected table.</p>  
<ul> 
  <li> 
  To restrict the data to a specific position range, type the position into the 
  <code>position</code> box. Some examples of specific positions include a chromosome name 
  (<em>chrX</em>), a coordinate range within a chromosome (<em>chrX:100000-400000</em>), or a 
  scaffold name.</li> 
  <li> 
  You can select multiple genomic regions by clicking the &quot;define regions&quot; button and 
  entering up to 1,000 regions in a 3- or 4-field 
  <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED</a> file format.</li> 
  <li> 
  To look up the position range of a genomic element -- such as a gene name, an accession ID, an 
  STS marker, etc. -- or keywords from the GenBank description of an mRNA, type the string into the 
  <code>position</code> box, then click the <code>Lookup</code> button.</li>  
  <li> 
  The data in non-positional tables are not tied to genomic coordinates; therefore, the 
  <code>region</code> option is unavailable when a non-positional table is selected. A basic query 
  on a non-positional table will show all the data in the table.</li> 
</ul> 
<p> 
<strong>Step 5. Display the output</strong><br> 
Click the <code>Get Output</code> button to display the results of the query. By default, the Table 
Browser outputs the data from all fields in the selected table as tab-separated text on the screen. 
See the <a href="#OutputFormats">Output formats</a> section for information on configuring the query
output.</p>
<p> 
<strong>Example:</strong><br> 
Here is an example of a simple query that retrieves all the RefSeq Genes records in the position 
range chr7:26906938-26940301 on the May 2004 human genome assembly.</p> 
<ol> 
  <li> 
  Select the <em>Human</em> option in the <code>genome</code> list.</li> 
  <li> 
  Select the <em>May 2004</em> option in the <code>assembly</code> list.</li> 
  <li> 
  Select the <em>Genes and Gene Prediction Tracks</em> option in the <code>group</code> list.</li>  
  <li> 
  Select the <em>RefSeq Genes</em> option in the <code>track</code> list.</li>  
  <li> 
  Type <em>chr7:26906938-26940301</em> in the <code>position</code> box (the Table Browser will 
  automatically select the <code>position</code> option button).</li>  
  <li> 
  Click the <code>Get Output</code> button.</li> 
</ol> 
<p> 
The Table Browser will display the records for the RefSeq accessions NM_005522, NM_153620, 
NM_006735, NM_153632, NM_030661, and NM_153631.</p> 

<!-- ====Batch Query========================= -->
<a name="BatchQuery"></a> 
<h3>Batch query using identifiers</h3> 
<p> 
In many cases, you may want to retrieve data based on a list of one or more accessions, IDs, or names, 
rather than querying by genomic position. Many tracks in the Table Browser, such as those in the 
<em>Genes and Gene Prediction</em> or <em>Variation</em>track groups, support identifier queries. The identifier type used
in the query must match the kind of identifiers present in the track data, e.g., mRNA accession IDs 
must be used to query the mRNA table and rsIDs must match those in the dbSNP table.</p>  
<p>
Follow these steps to display a list of records that correspond to a set of accessions or names 
entered as query input.</p> 
<p> 
<strong>Step 1. Pick the genome assembly, track, and table</strong></p> 
<p> 
<strong>Step 2. Select the <em>genome</em> <code>region</code> setting</strong></p> 
<p> 
<strong>Step 3. Load the identifiers into the browser</strong><br> 
Click the <code>Paste List</code> button to type or paste in the identifiers or the <code>Upload 
List</code> button to load the data from a file existing on your local computer.</p>  
<ul> 
  <li>
  If you are loading multiple identifiers, entries must be separated by a space, tab, or line.</li> 
  <li>
  Wildcards may not be used in the list (see the <a href="#Filter">Filter</a> section for 
  information about conducting queries that include wildcards).</li> 
  <li>
  The Table Browser will retain the identifier list until you delete the information by clicking the
  <code>Clear List</code> button.</li> 
</ul> 
<p> 
<strong>Step 4. Click the <code>Get Output</code> button</strong><br> 
See the <a href="#OutputFormats">Output formats</a> section for information about configuring the 
query output. 
 <!-- 
<p> 
<em><strong>Example:</strong></em><br> 
[FIXME - add example]
-->
<!-- ====Query to get gene symbols========================== -->

<!-- ====Batch Query with positions========================== -->
<a name="GetInfoFromPositions"></a>
<h3>Batch query from positions</h3>
<p>
If you have a list of genomic positions and want to retrieve information
about their properties, you can use the <code>Define Regions</code> button to input
multiple positions to query a chosen table. <b>Please note</b>, any items in the table that overlap
with the defined regions will be included in the Table Browser output. In this example, you want to
determine the dbSNP rsID names for your list of positions.
</p>

<p><strong>Step 1. Select genome assembly and track</strong></br>
To determine dbSNP rsIDs we will be using Human genome hg38 and dbSNP153.</p>

<p><strong>Step 2. Select the <code>define regions</code> button, enter regions</strong></br>
You can find the <code>define regions</code> button under the <code>Define region of
interest</code> section. Upload, type, or paste in your regions of interest, making sure they are
in the desired 0/1 base notation. They will only be accepted in BED or positional format.</p>

<p><strong>Step 3. Select output format and <code>get output</code></strong></br>
If you want all data from a table, you need not change the output format from the default.
If you want only particular columns from the table, you can change it to <code>selected fields
from primary and related tables</code>. Once you hit the <code>get output</code> button,
you will be redirected to a column selection page or if you did not change the output format,
your output data itself.</p>
<a name="GetGeneSymbols"></a>
<h3>Get gene symbols in a query</h3>

<p>
Follow the example below to obtain gene symbols in your query:
<p> 
<ul>
  <li>1. Select the clade, genome, assembly, group, table, and region as desired.</li> 
  <li>2. Change the <code>output format</code> to <em>selected fields from primary
and related tables</em>.</li>
  <li>3. Click <code> get output</code> to go to the next step of selecting fields from 
related tables.</li>
  <li>4. Select the fields you would like from your primary table.</li>
  <li>5. On the same <em>Select Fields</em> form, find the table for the related 
<em>kgXref</em> table. For example, look for the <em>hg38.kgXref</em> table, and then
check the checkbox next to <em>Gene Symbol</em> to add gene symbols to your query
results.</li>
<li>6. Click <code> get output</code> again to get the final query output.</li>
</ul>
</p>

<!-- ====Filtering Output================== -->
<a name="Filter"></a>
<h2>Filtering output by constraining field values</h2>
<p> 
The Table Browser <code>filter</code> option can be used to:</p> 
<ul> 
  <li> 
  apply constraints on table field values to restrict which records should appear in the query 
  output</li> 
  <li> 
  conduct batch queries using wildcards</li> 
  <li> 
  include fields from multiple tables in the query output</li> 
</ul> 

<!-- ====Filtering Output - Single Table=== -->
<a name="FilterSingle"></a>
<h3>Filtering on fields from a single table</h3>
<p> 
Follow these steps to create a filter on one or more fields in a single table:</p> 
<p> 
<strong>Step 1. Select the assembly, track, and region</strong></p> 
<p> 
<strong>Step 2. Click the <code>Create</code> button on the <code>filter</code> line</strong></p> 
<p> 
<strong>Step 3. Add the filter constraints</strong><br> 
One or more of the fields in the currently selected table may be filtered by typing constraints into
the corresponding text boxes.</p> 
<ul> 
  <li>
  By default, the initial values set up in the filter match all records in the table.</li> 
  <li>
  Constraints must match the data type of the field to be applied successfully. For example, the 
  <em>geneName</em> field in the hg17 <em>refFlat</em> table is a string; therefore, constraining 
  values must also be strings. See the <a href="#FilterConstraints">Filter constraints</a> sections 
  for more information on valid filter values.</li>  
  <li>
  Multiple filter values may be applied against one field by separating the values with spaces.</li>
  <li>
  Individual field constraints are combined with <em>AND</em>, i.e. a record must meet the 
  constraints on all fields to be retrieved.</li>  
</ul> 
<p> 
<strong>Step 4. Click the Submit button to apply the filter</strong></p> 
<p> 
Once a filter has been created on a table, it will persist for the duration of the Table Browser 
session or until it has been cleared. Only one filter can exist for a table at a time, but multiple 
filters may exist in one session if they are applied on different tables. To modify an existing 
filter, click the <code>Edit</code> button on the <code>filter</code> line. To remove a filter, 
click the <code>Clear</code> button.</p>  

<!-- ====Filtering Output - Multiple Tables -->
<a name="FilterMultiple"></a>
<h3>Filtering on fields from multiple tables</h3> 
<p> 
A Table Browser filter may include constraints on fields from tables related to the primary table. 
To create a filter composed of fields from multiple tables:</p> 
<p> 
<strong>Step 1. Select the assembly, track, and region</strong></p> 
<p> 
<strong>Step 2. Click the <code>Create</code> button on the <code>filter</code> line</strong><br> 
<strong>Note:</strong> If a filter already exists on the table, click the <code>Edit</code> button 
to modify it or the <code>Clear</code> button to remove it.</p> 
<p> 
<strong>Step 3. Select the tables to include in the filter</strong><br> 
Scroll down to the <em>Linked Tables</em> section of the page. The tables listed in this section are
linked to the selected table by one or more common fields (typically a name, accession, or ID 
field). Click the boxes in front of the table(s) whose fields you wish to include in the filter, 
then click the <code>Allow Filtering Using Field in Checked Tables</code> button. The fields of the 
selected tables will be displayed in the top portion of the page.</p> 
<p> 
<strong>Step 4. Add the filter constraints</strong></p> 
<p> 
<strong>Step 5. Click the Submit button to apply the filter</strong></p> 
<p> 
<strong>Note:</strong> In the current implementation of the Table Browser, the <em>selected fields 
from primary and related tables</em> output format option must be used when including fields from 
multiple tables in a filter. Check the boxes for all tables in the <code>Linked Tables</code> list 
on which filter constraints have been applied, then click the <code>Allow Selection From Checked 
Tables</code> button to include them in the output.</p>  

<!-- ====Filter Constraints================ -->
<a name="FilterConstraints"></a>
<h3>Filter constraints</h3> 
<p>
<strong>Strings</strong><br> 
Text fields are compared to words or patterns containing wildcard characters. Valid wildcards are 
i




&quot;*&quot; (matches 0 or more characters) and &quot;?&quot; (matches a single character). Each 
space-separated word or pattern in a text field box is matched against the value of that field in 
each record. If any word or pattern matches the value, then the record meets the constraint on that 
field.</p> 
<p> 
<strong>Numbers</strong><br> 
Numeric fields are compared to table data using an operator such as <, >, != (not equals) followed 
by a number. To specify a range, enter two numbers (start and end) separated by white space and/or a
comma.</p> 
<p> 
<strong>Free-form queries</strong><br> 
When the filters on individual fields aren't sufficiently flexible, the <code>free-form query</code>
text box allows the application of more complex constraints that typically relate two or more field 
names of the selected table. Valid free-form queries use the syntax of the SQL 
<em><a href="http://www.w3schools.com/sql/sql_where.asp" target="_blank">where</a></em> clause 
(using wildcards as defined above).</p> 
<p> 
Free-form queries combine simple constraints with <em>AND</em>, <em>OR</em>, and <em>NOT</em> using 
parentheses as needed for clarity.  A simple constraint consists of a table field name, a comparison
operator (see below), and a value: a number, string, wildcard value (see below), or another field 
name. In place of a field name, you may use an arithmetic expression of numeric field names.</p> 
<ul> 
  <li> 
  String or wildcard values for text comparisons must be quoted. Single or double quotes may be 
  used. If comparing to a literal string value, use the &quot;=&quot; or &quot;!=&quot; operator. 
  If comparing to a wildcard value, use the &quot;LIKE&quot; or &quot;NOT LIKE&quot; operator.</li> 
  <li> 
  Numeric comparison operators include <, <=, =, != (not equals), >=, and >.</li>  
  <li>
  Arithmetic operators include +, -, *, and /.</li>  
  <li>
  Other SQL comparison keywords may also be used.</li>  
</ul> 
<p> 
<strong><em>Example:</em></strong><br> 
The following examples show free-form queries applied to the human <em>refGene</em> table).</p>  
<ul> 
  <li> 
  <code>txStart = cdsStart</code>  - searches for gene models missing expected 5' UTR upstream 
  sequence (if strand is &quot;+&quot;; 3' UTR downstream if strand is &quot;-&quot;)</li> 
  <li> 
  <code>chrom NOT LIKE &quot;chr??&quot;</code> - restricts search to chromosomes 1 - 9,  X and 
  Y</li> 
  <li> 
  <code>cdsEnd - cdsStart) > 10000</code> - selects genes with coding spanning more than 10 kbp</li>
  <li> 
  <code>txStart != cdsStart) AND (txEnd != cdsEnd) AND exonCount = 1</code> - finds single exon 
  genes with both 3' and 5' flanking UTR</li>  
  <li> 
  <code>cdsEnd - cdsStart) > 30000) AND (exonCount=2 OR exonCount=3)</code> - finds genes with long 
  spans but only 2 - 3 exons</li> 
</ul

<!-- ====Intersecting Tables=============== -->
<a name="Intersection"></a>
<h2>Intersecting data from multiple tables</h2>
<p> 
It is often interesting to compare the positions of features in different annotation tracks to 
identify points of overlap. The Table Browser <code>intersection</code> utility can be used to 
generate various position-based comparisons of track features. Using the <code>intersection</code> 
utility, you can:</p> 
<ul> 
  <li> 
  examine all genomic positions where the feature data from the two tracks overlap</li> 
  <li> 
  identify genomic locations where there is no overlap between track features</li> 
  <li> 
  establish thresholds for the amount of overlap that must exist between the two feature sets</li> 
  <li> 
  conduct feature-by-feature comparisons as well as base-by-base comparisons of tracks</li> 
  <li> 
  complement (invert) a position set before comparing the tracks</li> 
</ul> 
<p> 
An intersection may be expanded to include additional tables by using the Table Browser custom track
feature.</p> 
<p> 
<strong>Note:</strong> The <code>intersection</code> utility can be used only on 
positional tables. To generate intersections incorporating data in non-positional tables, use the 
Table Browser <code>filter</code> utility. See the <a href="#FilterMultiple">Filtering on fields 
from multiple tables</a> section for more information.</p>

<!-- ====Intersecting 2 Tables============= -->
<a name="SimpleIntersection"></a>
<h3>Intersecting data from two tables</h3> 
<p> 
Follow these steps to configure and generate an intersection between two positional tables:</p> 
<p> 
<strong>Step 1. Select the assembly, track, table, and region for the primary table</strong><br> 
<strong>Note:</strong> Only positional tables may be used in an intersection.</p>  
<p> 
<strong>Step 2. Click the <code>Create</code> button on the <code>intersection</code> 
line</strong><br> 
<strong>Note:</strong> If an intersection already exists on the table, click the <code>Edit</code> 
button to modify it or the <code>Clear</code> button to remove it.</p> 
<p> 
<strong>Step 3. Select the secondary track to include in the filter</strong><br> 
Select a group in the <code><strong>group</code></strong> list, then select a track from the 
<code>track</code> list. To view all the tracks available, regardless of group, select the 
<em>All Tracks</em> option in the <code><strong>group</code></strong> list.</p> 
<p> 
<strong>Step 4. Select a combination method</strong><br> 
The Table Browser provides two major types of comparisons:</p> 
<ul> 
  <li> 
  <em>feature-by-feature</em> comparisons preserve the structure of the primary table. For example, 
  if the primary table describes exon structure and the features are compared with a second table, 
  the results will describe exon structure (unless you choose an output format in which the 
  structure is lost).</li> 
  <li> 
  <em>base-by-base</em> comparisons examine the primary table and the table underlying the secondary
  track one base at a time. The structure of the primary table is not preserved in this comparison. 
  For example, even if the primary table describes exon structure, the intersection results will 
  contain only position ranges; no information about exon/block structure, strand, or translation 
  region will be retained.</li>  
</ul> 
<p>
Click the circle in front of a combination method to select it. Only one method may be selected from
the two sets of methods. For more information about the individual combination options, see the 
<a href="#IntersectionOptions">Intersection Options</a> section.</p> 
<p> 
<strong>Step 5. (optional) Select the complement options</strong><br> 
<!-- FIX ME -->
Check the box in front of one or both tables to complement the feature data. The complement options
allow you to invert the set of positions covered by one or both tables. For example, if you
choose to complement the primary track, any position covered by the that track's features will be
considered <em>not</em> covered, and vice versa. This option provides more flexibility in comparing
track positions.</p>
<p> 
<strong>Step 6. Click the <code>Submit</code> button to apply the intersection</strong><br> 
Once an intersection has been created on a table, it will persist for the duration of the Table 
Browser session or until it has been cleared. Only one intersection may exist at a time. To modify 
an existing intersection, click the <code>Edit</code> button on the <code>intersection</code> 
line</strong>. To remove an intersection, click the <code>Clear</code> button.</p>

<!-- ====Intersecting Multiple Tables===== -->
<a name="MultiIntersection"></a>
<h3>Intersecting data from more than two tables</h3> 
<p> 
The Table Browser <code><strong>intersection</strong></code> utility limits combinations to only two
tables. An existing intersection may be expanded to include additional tables by using the Table 
Browser custom track utility. To create an intersection on multiple tables:</p> 
<p> 
<strong>Step 1. Set up an intersection between two tables</strong><br> 
See the <a href="#SimpleIntersection">Intersecting data from two tables</a> section for more 
information.</p> 
<p> 
<strong>Step 2. Save the intersection data in a custom track</strong><BR> See the 
<a href="#CustomTrack">Saving data as a custom track</a> section for information on generating a 
custom track. <strong>Note:</strong> In the current implementation of the Table Browser, you must 
use the <code>Get Custom Track</code> button on the custom track page to add the custom track to the
Table Browser <code>track</code> list.</p> 
<p> 
<strong>Step 3. Select the newly-generated custom track</strong><br> 
Select the <em>Custom Tracks</em> option in the <code>group</code> list, then select the newly 
created custom track from the <code>track</code> list.</p>  
<p> 
<strong>Step 4. Create an intersection with another track</strong><br> 
Follow the steps in the <a href="#SimpleIntersection">Intersecting data from two tables</a> section 
to intersect the custom track with another track.</p>

<!-- ====Intersecting 2 Tables============= -->
<a name="IntersectionOptions"></a>
<h3>Intersection options</h3> 
<p> 
<strong>Feature-by-feature comparisons</strong><br> 
Some comparisons preserve the primary table's gene and alignment structure, if it exists. For 
example, if the <em>refGene</em> table (human <em>RefSeq Genes</em> track) is combined with another 
table  using one of these comparisons, the resulting output data will describe exon structure 
(unless you choose an output format in which the structure is lost). Primary table features are kept
or discarded based on the amount of positional overlap with the features in the table underlying the
secondary track. The Table Browser offers the following options in this category:</p> 
<ul> 
  <li> 
  <strong>Any overlap:</strong> A primary table record will appear in the output if any of its base 
  positions are covered by any feature in the secondary table.</li>  
  <li> 
  <strong>No overlap:</strong> A primary table record will appear in the output only if none of its 
  base positions are covered by any feature in the secondary table.</li>  
  <li> 
  <strong>Overlap greater than a specified threshold:</strong> A primary table record will appear in
  the output if the percentage of its base positions covered by secondary table features is greater 
  than the user-specified threshold.</li> 
  <li> 
  <strong>Overlap less a specified threshold:</strong> A primary table record will appear in the 
  output if the percentage of its base positions covered by secondary table features is less than 
  the user-specified threshold.</li>  
</ul> 
<p>
<strong>Note:</strong> If the primary table has an exon/block structure, only those bases located in
exons and/or blocks will be counted.</p>  
<p> 
<strong>Base-by-base comparisons</strong><br> 
In these combination options, the positions of the primary and secondary table features are compared
one base position at a time. When applying base-by-base comparisons, the structure of the primary 
table is not preserved. For example, if the <em>refGene</em> table (from the human <em>RefSeq 
Genes</em> track) is compared with a secondary table using these comparisons, the resulting output 
data will not describe exon structure. Instead, only position ranges will be returned; the 
exon/block structure, strand, and translation region information will be discarded. The Table 
Browser provides the following base-by-base combination options:</p> 
<ul> 
  <li> 
  <strong>Base-by-base intersection (AND):</strong> A nucleotide position is included in the output 
  if it is covered by at least one feature of both the primary table <em>and</em> the secondary 
  table.</li>  
  <li> 
  <strong>Base-by-base union (OR):</strong> A nucleotide position is included in the output if it is
  covered by at least one feature of either the primary table <em>or</em> the secondary table.</li> 
</ul> 
<p>
<strong>Note:</strong> If the primary table has an exon/block structure, only base positions located
in exons and/or blocks will be counted.</p> 
<p>
<strong>Base-by-base complement (NOT)</strong><br> 
Before the Table Browser applies a feature-by-feature or base-by-base comparison to the table data, 
the set of positions covered by one or both tables can be inverted (complemented). When the data set
of a table is complemented, any position covered by the table's features in the original data will 
be considered not covered in the inverted data, and vice versa.  This option gives the user more 
flexibility in comparing table positions.</p> 

<!-- ====Correlation==================== -->
<a name="Correlation"></a>
<h2>Correlating data from two tables</h2>
<p> 
The Table Browser Correlation function creates a scatter plot of the data points of two tables as 
well as provides individual histograms of the data points from both tables. Additionally, it will 
also show a plot of the Residuals vs. Fitted which can be used to detect non-linearity, unequal 
error variances and outliers.</p> 
<p> 
The correlation function uses Pearson's correlation, which is optimized to work with continuous data
such as <a href="wiggle.html" target="_blank">wiggle tracks</a>. For tracks that do not have data 
values such as gene-structured tracks, the data value used in the calculation is 1.0 for bases 
covered by exons and 0.0 at all other positions in the region.</p> 
<p> 
Due to memory and processing limitations, the number of data points that can be plotted is limited 
to 300,000,000. The &quot;Window data to&quot; function allows you to smooth out your plot by taking
the average of the number of data points specified (defaults to 1). The total number of bases 
analyzed is independent of the data window. There is currently no way to output the results of the 
Correlation function.</p>

<!------Output Formats---------------------->
<a name="OutputFormats"></a>
<h2>Output formats</h2>
<p> 
The data resulting from a Table Browser query may be configured in a number of different ways:</p> 
<ul> 
  <li> 
  The output can be displayed on the screen, saved to a file, or saved to an annotation track table 
  that can be displayed in the Genome Browser or used in a subsequent Table Browser query.</li> 
  <li> 
  The data can include all fields from the primary or selected table, or can be restricted to 
  selected fields from the primary table and related tables.</li> 
  <li> 
  The data can be organized in one of several formats: tab-separated, sequence (FASTA), Browser 
  Extensible Data format (BED), Gene Transfer Format (GTF), or a statistical summary of the data in 
  the query.</li>  
</ul> 
<p>
The output options available for a specific query may vary depending on the table(s) selected. For 
example, non-positional table data cannot be organized in a position-based format, but instead may 
be displayed only in tab-separated format. The Table Browser will automatically update the options 
on the <code>output format</code> list to show only those available for the current query.</p>

<!-- ===Display All Fields================= -->
<a name="AllFields"></a> 
<h3>Displaying all fields in a table</h3> 
<p> 
To display all the fields of the records in the query output in tab-separated format, select the 
<em>all fields from primary table</em> option.</p>

<!-- ===Display Selected Fields=========== -->
<a name="SelectedFields"></a> 
<h3>Displaying selected fields from one or more tables</h3> 
<p> 
To restrict the query output to a subset of the fields in a table, choose the <em>selected fields 
from primary and related tables</em> option. You will be prompted to pick the table fields to 
display. Click the box in front of the fields you would like to see in the query output (or click 
the <code>Check All</code> button to select all the fields), then click the <code>Get Fields</code> 
button.</p> 
<p> 
To include data fields from other tables linked to the selected table, choose the <em>selected 
fields from primary and related tables</em> option, then scroll down to the <em>Linked Tables</em> 
section of the page. The tables listed in this section are linked to the selected table by one or 
more common fields (typically a name, accession, or ID field). Click the boxes in front of the 
table(s) whose fields you wish to include in the query output, then click the <code>Allow Selection 
From Checked Tables</code>. The fields of the selected tables will be displayed in the top portion 
of the page. Click the boxes in front of the fields that you wish to include in the query output, 
then click the <code>Get Fields</code> button underneath any of the field lists to generate 
tab-separated output that includes data from all the selected fields. Note that the <code>Get 
Fields</code> and <code>Cancel</code> buttons apply globally to all the selected tables, but the 
<code>Check All</code> and <code>Clear All</code> buttons apply only to the fields listed directly 
above the buttons.</p>

<!-- ==== Display Sequence ================ -->
<a name="Sequence"></a>
<h3>Displaying sequence (FASTA) data (positional tables only)</h3> 
<p> 
To display the genomic sequence underlying the query results, select the <em>sequence</em> option in
the <code>output format</code> list. The Table Browser will present you with several options to 
configure the output display. When you have completed the configuration, click the <code>Get 
Sequence</code> button. When displaying sequence data for gene prediction tracks, you will also be 
offered the option to view the protein and mRNA sequence as extracted from the data source in 
addition to the genomic sequence.</p>

<!-- ==== CDS FASTA alignments ================== -->
<a name="FASTA"></a> 
<h3>Displaying CDS FASTA alignments</strong> (genePred tables only)</h3>
<p> 
The CDS FASTA alignments are created from a Multiple Alignment File 
(<a href="../../FAQ/FAQformat.html#format5">MAF</a>) in combination with a 
<a href="../../FAQ/FAQformat.html#format9">genePred</a> table. The UCSC MAF format stores multiple 
alignments at the DNA level between entire genomes. You can use the Table Browser to return FASTA 
alignments of coding regions in nucleotide-space or translated into amino acid-space. However, it is
worth noting that the initial MAF files are all created by aligning genomes at the DNA level.</p> 
<p> 
<strong>Genome-wide CDS FASTA alignments</strong><br> 
Note that when using the Table Browser to fetch CDS FASTA output, it is best to restrict your query 
to a reasonable-sized position range rather than requesting output from the entire genome. A 
genome-wide query will take a substantial amount of compute time, and it is likely that your 
Internet browser will time out and disconnect. If you would like to download genome-wide CDS FASTA 
output for any of several model organisms, you can do so from the 
<a href="http://hgdownload.gi.ucsc.edu">download server</a>.</p> 
<p> 
<strong>Creating CDS FASTA alignments using the Table Browser</strong><br> 
To display FASTA multiple alignments for the CDS regions of genes, select the <em>CDS FASTA 
alignment from multiple alignment</em> option in the <code>output format</code> list. In order to 
see this output format option, you must have a genePred table selected. If you limit your search to 
a certain position range within the genome (rather than searching the entire genome), the tool will 
return FASTA alignments for all genes that overlap the position for which you are searching. The 
Table Browser will present you with a configuration page. On this page, you can select options for 
your output.</p> 
<p> 
First, select your MAF table.  This is the table from which the multiple alignments will be 
extracted for the CDS regions of your gene track. If you do not know the name of the MAF table that 
corresponds to the Conservation track, you can find it in the Genome Browser by following these 
<a href="../../FAQ/FAQtracks.html#tracks21">instructions</a>.</P> 
<p> 
Then select any of the following choices:</p>
<ul> 
  <li>
  <strong>Separate into exons</strong> - The default behavior is for the coding exons of each gene 
  to be concatenated into one sequence in the output FASTA multiple alignment. In this case each 
  output row header has the format listed below under &quot;Whole gene format&quot;. If the separate
  into exons option is chosen then each exon will be listed with a separate header in the format 
  listed below under &quot;Exon format&quot;.</li>  
  <li>
  <strong>Show nucleotides</strong> - The default behavior is for the nucleotides in the alignment 
  to be translated into amino acids according to the strand and exon frames defined in the selected 
  genePred table. If this option is chosen, then the nucleotides in the alignment will not be 
  translated into amino acids.</li>        
  <li>
  <strong>Output lines with just dashes</strong> - The default behavior is for the alignment rows 
  that contain only dashes to not be printed. If this option is chosen, then these dashes-only 
  rows are printed.</li>  
  <li>
  <strong>Format output as table</strong> - If this option is chosen, the header and sequence for 
  each organism will appear on the same line.</li> 
  <li>
  <strong>Truncate headers as __ characters (enter zero for no headers)</strong> - This option 
  works in conjunction with the &quot;Format output as table&quot; option. If you want to see only
  a portion of the headers, choose this option, and enter the number of characters at which you 
  would like the headers truncated.</li> 
</ul> 
<p>
Finally, from the list of species, select those that you would like included in the FASTA multiple 
alignment output. Press the &quot;get output&quot; button to view the output.</p> 
<p> 
<strong>Explanation of CDS FASTA header format</strong><br> 
Whole gene format: <code>geneName_assemblyName peptideLength location</code></p>
<p> 
Exon format: <code>geneName_assemblyName_exonNum_totalExons exonLength inFrame outFrame 
location</code></p>
<p> 
Here are the descriptions for each field name: 
<ul> 
  <li>
  <strong>geneName</strong>- the name field from the genePred table.</li>
  <li>
  <strong>assemblyName</strong>- the UCSC assembly 
  <a href="../../FAQ/FAQreleases.html#release1">name</a> for the species.</li> 
  <li>
  <strong>peptideLength</strong>- the length of the entire coding region. If the &quot;Show 
  nucleotides&quot; option is chosen, this will be in nucleotides, otherwise it will be the number 
  of amino acids in the peptide.</li> 
  <li>
  <strong>location</strong>- this is the chromosome position within the assembly that is aligned in 
  the multiple alignment. The format of this string is chrom:start-end followed by the strand where 
  the alignment occurs. If more than one region is aligned then all the regions are listed with a 
  semi-colon (;) between each position. This address is in genome browser coordinates (i.e. the 
  start address is <a href="../../FAQ/FAQtracks.html#tracks1">one-based</a>).</li> 
  <li>
  <strong>exonNum</strong>- the ordinal of the exon. Exons are counted starting at one and begin at 
  the transcription start site and progress along the strand of transcription.</li> 
  <li>
  <strong>totalExons</strong>- the number of coding exons in the gene.</li> 
  <li>
  <strong>exonLength</strong>- the length of the current exon. If the &quot;Show nucleotides&quot; 
  option is chosen, this will be the number of nucleotides in the exon, otherwise it will be the 
  number of amino acids in the exon (with amino acids translated from split codons placed in the 
  exon where two of the three nucleotides lie).</li> 
  <li>
  <strong>inFrame</strong>- the frame number of the first nucleotide in the exon. Frame numbers can 
  be 0, 1, or 2 depending on what position that nucleotide takes in the codon which contains 
  it.</li> 
  <li>
  <strong>outFrame</strong>- the frame number of the nucleotide after the last nucleotide in this 
  exon. Frame numbers can be 0, 1, or 2 depending on what position that nucleotide takes in the 
  codon which contains it.</li> 
</ul> 
<p> 
<strong>Explanation of CDS FASTA sequence format</strong><br> 
As noted above, the CDS FASTA output files can be in either DNA-space or protein-space.</p> 
<p>
In some instances, there is a dash (&quot;&ndash;&quot;) in the sequence portion of the CDS FASTA 
file. Dashes are used in several circumstances. They indicate missing sequence for the aligning 
genome, as well as deletions in the aligning genome or insertions in the base genome.</p> 
<p> 
Because the CDS FASTA alignments are based on one reference genome, any amino acids or nucleotides 
that are not in the reference genome are not displayed. Consequently the peptides shown for aligning
genomes are not necessarily the peptide that the gene of the other organism would generate. Any 
sequence inserted in an aligning genome or deleted in the base genome will not be present in the 
alignment. We represent this condition with an orange bar in the Genome Browser display, but the 
CDS FASTA alignments silently ignore this issue.</p> 
<p> 
<strong><em>Nucleotide CDS FASTA sequence:</em></strong><br> 
Consider the example below that shows the FASTA sequence for four species aligned with the first 
exon of the human gene PLEKHO1 (UCSC Gene: uc001ett.1). Note that the rat (rn4) row is missing the 
first three nucleotides. This could be due to a lineage-specific insertion between the rat and human
genomes, or a lineage-specific deletion between the human and rat genomes. Note also that the 
Zebrafish (danRer4) row contains only dashes. This could be due to excessive evolutionary distance 
between the zebrafish and human, missing data in the zebrafish, or independent indels in the region 
in both species. Sometimes it is helpful to view the Conservation track in the Genome Browser in 
this area to clarify the exact meaning of the dashes.</p>
<pre><code>&gt;uc001ett.1_hg18_1_6 30 0 0 chr1:148389072-148389101+
ATGATGAAGAAGAACAAcode
&gt;uc001ett.1_panTro2_1_6 30 0 0 chr1:129156502-129156531+
ATGATGAAGAAGAACAAcode
&gt;uc001ett.1_rn4_1_6 30 0 0 chr2:190795892-190795918-
---ATGAAGAAGAGCGGCTCCGGCAAGCGG
&gt;uc001ett.1_danRer4_1_6 30 0 0
------------------------------
&gt;uc001ett.1_oryLat2_1_6 30 0 0 chr11:3404940-3404969-
AGGATGAAGAAAAGCAACCAGAGCAGGCGG </code></pre>
<p> 
<strong><em>Amino Acid CDS FASTA sequence:</em></strong><br> 
<ul> 
  <li>
  Codons that have a dash in any of the three nucleotides are represented by a dash in the amino 
  acid.</li>  
  <li>
  Codons with an N in any position are represented with an X.</li>  
  <li>
  Stop codons are represented with a Z.</li>  
  <li>
  All other amino acids follow the IUPAC amino acid codes.</li> 
  <li>
  In exon format, when the codon triplet is split between two exons, the amino acid will be 
  displayed as part of the exon containing two of the three nucleotides like so:</li> 
  <pre><code>|exon1|  |exon2|
nucleotide: AAACCCT  code
protein:     K  P    F  G  K </code></pre>
</ul>

<!-- ====GTF/BED Format=================== -->
<a name="GTF_BED"></a> 
<h3>Saving query results in GTF or BED format (positional tables only)</h3> 
<p> 
To format the query results using 
<a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format4">GTF</a> or 
<a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED</a> conventions, select the 
corresponding option in the <code>output format</code> list. Note that when you select GTF, the 
table browser translates the output into this format. For tables that lack feature designations, all
records are arbitrarily assigned the feature &quot;exon&quot; to conform to GTF specifications. If 
you select BED format, you will be presented with the option to include and configure a custom track
header and options for organizing the data. When you have finished the configuration -- or to accept
the default options -- click the <code>Get BED</code> button at the bottom of the window. To understand
the name column in the BED format, see this <a href="/FAQ/FAQdownloads.html#download34">FAQ</a>.</p>

<!-- === Save to File ==================== -->
<a name="SaveFile"></a> 
<h3>Saving data to a file</h3> 
<p> 
By default, the Table Browser displays query results directly in your internet browser window. To 
redirect the data to a file, type a file name into the <code>output file</code> box before starting 
the query. The Table Browser will prompt you for the location of this file on your local disk while 
processing the query.</p>

<!-- === Save as Custom Track ============ --> 
<a name="CustomTrack"></a>
<h3>Saving data as a custom track (positional tables only)</h3> 
<p> 
Query output may be saved in a format that can be displayed as a custom annotation track in the 
Genome Browser. Custom tracks created during a Table Browser session may also be used for 
subsequent queries and intersections in the same session. For more information on custom tracks, see
the Genome Browser <a href="hgTracksHelp.html#CustomTracks">User's Guide</a>.</p>  
<p> 
To save query data in custom track format, select the <em>custom track</em> option in the 
<code>output format</code> list. When the query is executed, the Table Browser will prompt you to 
customize the track header and configure the record layout of the data. The configuration is 
optional; the Table Browser automatically sets up a default track configuration. Click the 
<em>Custom track</em> link for more information on custom track syntax and format.</p> 
<p> 
When you have finished configuring the custom track -- or to accept the default configuration -- 
click one of the buttons at the bottom of the window to create the custom annotation track.</p>  
<ul> 
  <li> 
  To display the query results as text on the screen, click the <code>Get Custom Track File</code> 
  button.  
  <li> 
  To save the query results to a file on your local disk for future use, specify a file name in the 
  <code>output file</code> box before executing the query, then click the <code>Get Custom Track    
  File</code> button.</li>  
  <li> 
  To load the query results into a table accessible from the Table Browser <code>table</code> list, 
  click the <code>Get Custom Track in Table Browser</code> button.</li>  
  <li> 
  To view the query results as a custom track in the Genome Browser, click the <code>>Get Custom 
  Track in Genome Browser</code> button. Your browser display will be redirected automatically to 
  the Genome Browser, with your custom track positioned near the top of the annotation tracks 
  window.</li>  
  <li> 
  To access your custom track data in a subsequent query in the same Table Browser session, select 
  the <em>Custom Tracks</em> option from the <code>group</code> list to display the custom tracks 
  available.</li>  
</ul>

<!-- === Display as Hyperlink =========== -->
<a name="Hyperlink"></a>
<h3>Displaying query results as Genome Browser hyperlinks (positional tables only)</h3> 
<p> 
To examine the records in the query output individually in the Genome Browser, select the 
<em>hyperlinks to Genome Browser</em> output option. The Table Browser will display a list of one or
more hyperlinks corresponding to the individual records in the output data. Click a link to open up 
the Genome Browser display to the item and position shown on the hyperlink.</p>

<!-- == Display Summary/Statistics ===== -->
<a name="SummaryStats"></a> 
<h3>Displaying a statistical summary of query data (positional tables only)</h3> 
<p> 
To generate a statistical summary of the query output data, the region covered by the query, and the
CPU time required to process the query, click the <code>Summary/Statistics</code> button.</p> 

<!-- ===== Table Schema ================== -->
<!--FIXME - ADD
<a name="TableSchema"></a>
<h3>Table Schema</h3>
-->

<!-- === Video using Table Browser  =========== -->
<a name="Videos"></a>
</p><p>
<h2>Video examples of Table Browser queries</h2>
<p></p> 

<!-- === list of genes =========== -->
<a name="listOfGenes"></a>
<h3>Finding a list of genes in a genomic region</h3>
<p>
<!--#if expr="${SERVER_NAME} = /-china/" -->
    <a href='../../../videos/RjB_N1GpT0U.mp4'>
    <img src=../../images/videoIcon.png></a>
    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
    Visit our <a href="/videos/" target="_blank">Video Page</a>.
<!--#else -->
    <iframe width="560" height="315" src="https://www.youtube.com/embed/RjB_N1GpT0U?rel=0"
    frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
    allowfullscreen></iframe>
    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
    Visit our
    <a href="https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos"
    target="_blank">YouTube channel</a>.
<!--#endif -->
</p>

<!-- Getting codon sequences -->
<a name="codonSeq"></a>
<h3>Obtaining coordinate sequences for a gene exon</h3>
<p>
<!--#if expr="${SERVER_NAME} = /-china/" -->
    <a href='../../../videos/6JoUqM1iKxQ.mp4'>
    <img src=../../images/videoIcon.png></a>
    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
    Visit our <a href="/videos/" target="_blank">Video Page</a>.
<!--#else -->
    <iframe width="560" height="315" src="https://www.youtube.com/embed/6JoUqM1iKxQ?rel=0"
    frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
    allowfullscreen></iframe>
    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
    Visit our
    <a href="https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos"
    target="_blank">YouTube channel</a>.
<!--#endif -->
</p>

<!-- === SNPs in a gene =========== -->
<a name="snpsInGene"></a>
<h3>Finding all the SNPs in a gene</h3>
<a name="Videos"></a>
<p>
<!--#if expr="${SERVER_NAME} = /-china/" -->
    <a href='../../../videos/Y_cQ3JkhsUQ.mp4'>
    <img src=../../images/videoIcon.png></a>
    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
    Visit our <a href="/videos/" target="_blank">Video Page</a>.
<!--#else -->
    <iframe width="560" height="315" src="https://www.youtube.com/embed/Y_cQ3JkhsUQ?rel=0"
    frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
    allowfullscreen></iframe>
    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
    Visit our
    <a href="https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos"
    target="_blank">YouTube channel</a>.
<!--#endif -->
</p>

<!-- === SNPS upstream of genes =========== -->
<a name="snpsUpstream"></a>
<h3>Finding SNPs upstream of a gene</h3>
<p>
<!--#if expr="${SERVER_NAME} = /-china/" -->
    <a href='../../../videos/ZqxlBF3vqhY.mp4'>
    <img src=../../images/videoIcon.png></a>
    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
    Visit our <a href="/videos/" target="_blank">Video Page</a>.
<!--#else -->
    <iframe width="560" height="315" src="https://www.youtube.com/embed/ZqxlBF3vqhY?rel=0"
    frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
    allowfullscreen></iframe>
    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
    Visit our
    <a href="https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos"
    target="_blank">YouTube channel</a>.
<!--#endif -->
</p>

<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->
