<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>

<HEAD>
  <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
  <META name="description" content="UCSC Table Browser">
  <META name="keywords" content="genome, genome data, table browser, genome browser, genome assembly, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA">
  <TITLE>UCSC Genome Browser: Table Browser User's Guide</TITLE>
  <LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
  <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../inc/topbar.html"--></TD></TR>
<!--Content Tables-->  
  <TR><TD CELLPADDING=10 ROWSPAN=2 VALIGN="top">
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>	
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>	
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	        <FONT SIZE="4"><B>&nbsp;Table Browser User's Guide </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
   	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
	        <H3> Table of Contents: </H3>
	 	<UL><B>
  		<LI><A CLASS="toc" HREF="#Introduction">Introduction</A>
        	<LI><A CLASS="toc" HREF="#Tables">About the Table Browser databases and tables</A>
	   	  <UL>
                  <LI><A CLASS="toc" HREF="#NonPositional">Non-positional tables</A>
                  <LI><A CLASS="toc" HREF="#Positional">Position-oriented tables</A>
	  	  </UL
                <LI><A CLASS="toc" HREF="#GettingStarted">Getting started - simple queries</A>
	   	  <UL>
                  <LI><A CLASS="toc" HREF="#PositionQuery">Simple position-based query</A>
                  <LI><A CLASS="toc" HREF="#BatchQuery">Batch query using identifiers</A>
		</UL>
  		<LI><A CLASS="toc" HREF="#Filter">Filtering output by constraining field values</A>
	 	  <UL>
  		  <LI><A CLASS="toc" HREF="#FilterSingle">Filtering on fields from a single table</A>
  		  <LI><A CLASS="toc" HREF="#FilterMultiple">Filtering on fields from multiple tables</A>
  		  <LI><A CLASS="toc" HREF="#FilterConstraints">Filter constraints</A>
		  </UL>
  		<LI><A CLASS="toc" HREF="#Intersection">Intersecting data from multiple tables</A>
  		  <UL>
		  <LI><A CLASS="toc" HREF="#SimpleIntersection">Intersecting data from two tables</A>
		  <LI><A CLASS="toc" HREF="#MultiIntersection">Intersecting data from multiple tables</A>
		  <LI><A CLASS="toc" HREF="#IntersectionOptions">Intersection options</A>
		  </UL>
  		<LI><A CLASS="toc" HREF="#OutputFormats">Output Formats</A>
      	          <UL>
        	  <LI><A CLASS="toc" HREF="#AllFields">Displaying all fields in a table</A>
        	  <LI><A CLASS="toc" HREF="#SelectedFields">Displaying selected fields from one or more tables</A>
        	  <LI><A CLASS="toc" HREF="#Sequence">Displaying sequence (FASTA) data</A>
        	  <LI><A CLASS="toc" HREF="#FASTA">Displaying CDS FASTA alignments</A>
        	  <LI><A CLASS="toc" HREF="#GTF_BED">Saving query results in GTF or BED format</A>
        	  <LI><A CLASS="toc" HREF="#SaveFile">Saving data to a file</A>
        	  <LI><A CLASS="toc" HREF="#CustomTrack">Saving data as a custom track</A>
        	  <LI><A CLASS="toc" HREF="#Hyperlink">Displaying query results as Genome Browser hyperlinks</A>
        	  <LI><A CLASS="toc" HREF="#SummaryStats">Displaying a statistical summary of query data</A>
      		  </UL>
		</B></UL>
		<P>
		The UCSC Table Browser User's Guide was last updated on 1 
		October 2004. 
		<A HREF="../../contacts.html">Questions and feedback are welcome</A>.
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
  <P>

<!------Introduction-------------------------->
<A NAME="Introduction"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>
 	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	    	<FONT SIZE="4"><B>&nbsp; Introduction </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		The Table Browser provides a powerful and flexible graphical
		interface for querying and manipulating the Genome Browser 
		annotation tables. Because the Table Browser uses the same 
		database as the Genome Browser, the two views are always 
		consistent.  
		<P>
		Using the Table Browser, you can: 
		<UL>
		<LI>
		retrieve the DNA sequence data or annotation data underlying 
		Genome Browser tracks for the entire genome, a specified 
		coordinate range, or a set of accessions
		<LI>
		apply a filter to set constraints on field values included in 
		the output
		<LI>
		generate a 
		<A HREF="/goldenPath/help/hgTracksHelp.html#CustomTracks">custom
		track</A> and automatically add it to your session so that it 
		can be graphically displayed in the Genome Browser
		<LI>
		conduct both structured and free-from SQL queries on the data
		<LI>
		combine queries on multiple tables or custom tracks through an 
		intersection or union and generate a single set of output data 
		<LI>
		display basic statistics calculated over a selected data set 
		<LI>
		display the schema for table and list all other tables in 
		the database connected to the table
		<LI>
		organize the output data into several different formats for use
		in other applications, spreadsheets, or databases
		</UL>
		<P>
		This User's Guide is aimed at both the novice Table Browser user
		as well the advanced user. If you are new to the Table Browser,
		read the <A HREF="#GettingStarted">Getting started</A> section to learn about browser basics
		and try some simple queries. Advanced users may want to proceed
		directly to the section that addresses a particular 
		area of functionality in detail.
	        <P>
		Although the Table Browser provides sufficient flexibility to 
		satisfy the needs of most users, some advanced users may require
		the ability to run MySQL directly on the Genome Browser database.
		UCSC provides a public MySQL server at 
		<FONT color="#0000FF">genome-mysql.cse.ucsc.edu</FONT>.
		Alternatively, the database may be downloaded to a local computer
		for MySQL access. See the 
		<A HREF="../../admin/mirror.html">mirror site</A>
		documentation for information on setting up a local copy of the 
		database.
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>

<!------Tables-------------------------->
<A NAME="Tables"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
          <TR><TD>
 	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	    	<FONT SIZE="4"><B>&nbsp; About the Table Browser databases and tables </B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		The Table Browser is built on top of the Genome Browser 
		database</A>, which actually consists of several separate 
		databases, one for each genome assembly.  
		<P>
		Tables within the databases may be differentiated by whether the
		data are based on genome start-stop coordinates 
		(positional tables) or are independent of position
		(non-positional tables). 
		Some output formats and query options are applicable only to 
		positional tables, hence the distinction.
		<P>

<!------Non-positional------------------------->
<A NAME="NonPositional"></A>
	        <HR>
	        <B><FONT SIZE="3" COLOR="#0000FF">Non-positional tables</FONT></B>
		<P>
		Non-positional tables contain data not tied to genomic 
		location, for example a table that correlates a Known Gene 
		ID with a RefSeq accession ID. Some non-positional tables relate 
		internal numeric mRNA IDs to extended information such as author, 
		tissue, or keyword.  Some &quot;meta&quot; tables in this category
		contain information about the structure of the database itself 
		or describe external files containing sequence data.  
	        <P>
<!------Positional---------------------------->
<A NAME="Positional"></A>
	        <HR>
	        <B><FONT SIZE="3" COLOR="#0000FF">Positional tables</FONT></B>
	        <P>
		Positional tables contain data associated with specific 
		locations in the genome, such as mRNA alignments, gene 
		predictions, cross-species alignments, and other annotations.
     		Each of the annotation tracks displayed in the Genome Browser
		is based on a positional table. In some instances, data from
		other positional and non-positional tables may also be 
		incorporated into the track. Data associated with custom 
		annotation tracks active within the user's Table Browser session
		are also available as positional tables.
		<P>
		Positional tables can be further subdivided into several 
		categories based on the type of data they describe. Alignment
		data can be best described by using a block structure to represent
		each element. Other tables require only start and end 
		coordinate data for each element. Some tables specify a 
		translation start and end in addition to the transcription start 
		and end. Some tables contain strand information, others don't.  
		Most tables, but not all, specify a name for each element.  
		Based on the format of the data described by a table, different 
		query and output formatting options may be offered.  
		<P>
		For descriptions of the Genome Browser database tables, 
		see the <A HREF="/goldenPath/gbdDescriptions.html">annotation 
		database</A> documentation.
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>

<!------Getting Started----------------------->
<A NAME="GettingStarted"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	  <TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	       <FONT SIZE="4"><B>&nbsp; Getting started - simple queries</B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
	        <P>
		In its most basic form, the Table Browser can be used to 
		retrieve a specific subset of records from a track or positional
		table in a selected genome assembly. The query may be based on
	        a specific position or a set of one or more identifiers.
	        <P>
	        This section describes the steps required to conduct basic
	        simple data queries using the Table Browser. Once you have
	        mastered the basic Table Browser functionality, refer to 
	        subsequent sections for information about generating more 
	        complex queries that use filters, intersections, and alternative 
	        data output formats.
	 	<P>

<!------Position-based query------------------>
<A NAME="PositionQuery"></A>
	 	<HR>
	        <B><FONT SIZE="3" COLOR="#0000FF">Simple position-based query</FONT></B>
		<P>
		Follow these steps to display a list of records that lie
		within a specific position in a table:
		<P>
		<B>Step 1. Pick a genome assembly</B><BR>
		Specify the genome assembly from which you'd like to retrieve the
		data by choosing the appropriate organism in the <TT><B>genome</B></TT>
		list, then selecting the assembly version from the 
		<TT><B>assembly</B></TT> list. Note that the <TT><B>assembly</B></TT> list 
		refreshes each time a different option is selected in the 
		<TT><B>genome</B></TT> list. 
		<P>
		<B>Step 2. Pick an annotation track</B><BR>
		The <TT><B>group</B></TT> list shows all the annotation track groups 
		available in the selected genome assembly. The names correspond
		to the groupings displayed at the bottom of the Genome Browser
		annotation tracks page. When a group is selected from the list,
		the <TT><B>track</B></TT> list automatically updates to show all the 
		annotation tracks available within that group. 
		  <UL>
		  <LI>
		  If you already know the name of the annotation track in which 
		  you're interested, select the <em>All Tracks</em> option in the 
		  <TT><B>group</B></TT> list, then select the track from the 
		  <TT><B>track</B></TT> list.
		  Similarly, you can directly select a table by 
		  choosing the <em>All Tables</em> option in the 
		  <TT><B>group</B></TT> list, selecting a database from the 
		  <TT><B>database</B></TT> list, then selecting the table from 
		  the <TT><B>table</B></TT> list.
		  <LI>
		  To examine all the tracks available within a certain group (e.g.
		  all gene prediction tracks), select the group name from the
		  <TT><B>group</B></TT> list, then browse the entries in the 
		  <TT><B>track</B></TT> list.
		  <LI>
		  Custom annotation tracks created during the current session
		  are listed under the <em>Custom Tracks</em> group.
		  <LI>
		  If no selections are made from the <TT><B>group</B></TT> or 
		  <TT><B>track</B></TT> lists, the track selection defaults to the
		  <em>Known Genes</em> track in the <em>Genes and Gene Prediction Tracks</em> group.
		  </UL>
		<P>
		<B>Step 3. Pick a table</B><BR>
		The <TT><B>table</B></TT> list shows all tables (both positional
		and non-positional) associated
		with the currently-selected track. By default, it 
		displays the primary table for the track, i.e. the table
		containing the data shown in the Genome Browser
		annotation track. Other tables in the list are linked to the
		primary table by a common field and may provide supporting
		data used in constructing the annotation. 
		<UL>
		<LI>
		If the <TT><B>group</B></TT> list is set to the 
		<em>All Tables</em> option, the tables list will show all tables
		present in the database currently selected in the 
		<TT><B>database</B></TT> list, rather than those associated with
		a particular track.  
		</UL>
		<P>
		<B>Step 4. Pick a genomic region</B> (positional tables only)<BR>
		By default, the Table Browser region is set to <TT><B>genome</B></TT>,
		which will display all the data records in the selected table. 
		  <UL>
		  <LI>
		  To restrict the data to a specific position
		  range, type the position into the <TT><B>position</B></TT> box. Some
		  examples of specific positions include a chromosome name 
		  (<em>chrX</em>), a coordinate range within a chromosome
		  (<em>chrX:100000-400000</em>), or a scaffold name.
		  <LI>
		  To view data only in those regions covered by the ENCODE 
		  project, click the button preceding the <TT><B>ENCODE</B></TT> option.
		  <LI>
		  To look up the position range of a genomic element -- such
		  as a gene name, an accession ID, an STS marker, etc. -- or 
		  keywords from the GenBank description of an mRNA,
		  type the string into the <TT><B>position</B></TT> box, then 
		  click the <TT><B>Lookup</B></TT> button.
		  <LI>
		  The data in non-positional tables are not tied to genomic
		  coordinates; therefore, the <TT><B>region</B></TT> option is
		  unavailable when a non-positional table is selected. A basic 
		  query on a non-positional table will show all the data in the 
		  table.
		  </UL>
		<P>
		<B>Step 5. Display the output</B><BR>
		Click the <TT><B>Get Output</B></TT> button to display the 
		results of the query. By default, the Table Browser outputs the 
		data from all fields in the selected table as tab-separated text
		on the screen. See the <A HREF="#OutputFormats">Output Formats</A>
		section for information on configuring the query output.
		</OL>
		<P>
		<em><B>Example:</B></em><BR>
		Here is an example of a simple query that retrieves all the
		RefSeq Genes records in the position range 
		chr7:26906938-26940301 on the May 2004 human genome assembly.
		<OL>
		<LI>
		Select the <em>Human</em> option in the <TT><B>genome</B></TT> list
		<LI>
		Select the <em>May 2004</em> option in the <TT><B>assembly</B></TT> list
		<LI>
		Select the <em>Genes and Gene Prediction Tracks</em> option in the
		<TT><B>group</B></TT> list.
		<LI>
		Select the <em>RefSeq Genes</em> option in the
		<TT><B>track</B></TT> list.
		<LI>
		Type <em>chr7:26906938-26940301</em> in the 
		<TT><B>position</B></TT> box (the Table Browser will 
		automatically select the <TT><B>position</B></TT> option button).
		<LI>
		Click the <TT><B>Get Output</B></TT> button.
		</OL>
		<P>
		The Table Browser will display the records for the RefSeq
		accessions NM_005522, NM_153620, NM_006735, NM_153632, NM_030661,
		and NM_153631.
	        <P>

<!------Batch Query--------------------------->
<A NAME="BatchQuery"></A>
	        <HR>
	        <B><FONT SIZE="3" COLOR="#0000FF">Batch query using identifiers</FONT></B>
		<P>
		In many cases, you may want to retrieve data based on a list of 
		one or more accessions or names, rather than querying by
		genomic position. Many tracks in the Table Browser, such as
		those in the <em>Genes and Gene Prediction</em> track group, 
		support identifier queries. The identifier type used in the
		query must match the kind of identifiers present in the track data, e.g.
		mRNA accession IDs must be used to query the mRNA table.
		<P>
		Follow these steps to display a list of 
		records that correspond to a set of accessions or names entered 
		as query input.
		<P>
		<B>Step 1. Pick the genome assembly, track, and table</B>
		<P>
		<B>Step 2. Select the <em>genome</em> <TT>region</TT> setting</B>
		<P>
		<B>Step 3. Load the identifiers into the browser</B><BR>
		Click the <TT><B>Paste List</B></TT> 
		button to type or paste in the identifiers or the 
		<TT><B>Upload List</B></TT> button to load the data from a
		file existing on your local computer. 
		<UL>
		<LI>If you are loading 
		multiple identifiers, entries must be separated by a space, tab,
		or line. 
		<LI>Wildcards may not be used in the list (see the
		<A HREF="#Filter">Filter</A> section for information about
		conducting queries that include wildcards). 
		<LI>The Table Browser
		will retain the identifier list until you delete the
		information by clicking the <TT><B>Clear List</B></TT> button.
		</UL>
		<P>
		<B>Step 4. Click the <TT>Get Output</TT> button</B><BR>
		See the <A HREF="#OutputFormats">Output Formats</A>
		section for information about configuring the query output.
<!--
		<P>
		<em><B>Example:</B></em><BR>
		[FIXME - add example]
-->
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>

<!------Filtering Output-------------------->
<A NAME="Filter"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	  <TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	       <FONT SIZE="4"><B>&nbsp; Filtering output by constraining field values</B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
	      The Table Browser <TT><B>filter</B></TT> option can be used to: 
	      <UL>
	      <LI>
	      apply constraints on table field values to restrict which records
	      should appear in the query output
	      <LI>
	      conduct batch queries using wildcards
	      <LI>
	      include fields from multiple tables in the query output
	      </UL>
	      <P>

<!------Filtering Output - Single Table----->
<A NAME="FilterSingle"></A>
	      <HR>
	      <B><FONT SIZE="3" COLOR="#0000FF">Filtering on fields from a single table</FONT></B>
	      <P>
	      Follow these steps to create a filter on one or more fields in a 
	      single table:
	      <P>
	      <B>Step 1. Select the assembly, track, and region</B>
	      <P>
	      <B>Step 2. Click the <TT><B>Create</B></TT> button on the 
	      <TT><B>filter</B></TT> line</B>
	      <P>
	      <B>Step 3. Add the filter constraints</B><BR>
	      One or more of the fields in the currently selected table may
	      be filtered by typing constraints into the corresponding text
	      boxes. 
	      <UL>
	      <LI>By default, the initial values set up in the filter match all 
	      records in the table. 
	      <LI>Constraints must match the data type of the field to be applied
	      successfully. For example, the <em>geneName</em> field in the hg17 
	      <em>refFlat</em> table is a string; therefore, constraining values
	      must also be strings. See the <A HREF="#FilterConstraints">Filter
	      constraints</A> sections for more information on valid filter 
	      values.
	      <LI>Multiple filter values may be applied against one field by
	      separating the values with spaces.
	      <LI>Individual field constraints are combined with <em>AND</em>, 
	      i.e. a record must meet the constraints on all fields to be 
	      retrieved. 
	      </UL>
	      <P>
	      <B>Step 4. Click the Submit button to apply the filter</B><BR>
	      <P>
	      Once a filter has been created on a table, it will persist for the 
	      duration of the Table Browser session or until it has been 
	      cleared. Only one filter can exist for a table at a time, but multiple filters may exist in one session if they are applied on different tables.
	      To modify an existing filter, click the <TT><B>Edit</B></TT> 
	      button on the <TT><B>filter</B></TT> line</B>. To remove a filter,
	      click the <TT><B>Clear</B></TT> button.
	      <P>
<!------Filtering Output - Multiple Tables-->
<A NAME="FilterMultiple"></A>
              <HR>
	      <B><FONT SIZE="3" COLOR="#0000FF">Filtering on fields from multiple
	      tables </FONT></B>
	      <P>
	      A Table Browser filter may include constraints on fields from
	      tables related to the primary table. To create a filter composed
	      of fields from multiple tables:
	      <P>
	      <B>Step 1. Select the assembly, track, and region</B>
	      <P>
	      <B>Step 2. Click the <TT><B>Create</B></TT> button on the 
	      <TT><B>filter</B></TT> line</B><BR>
	      <B>Note:</B> If a filter already exists on the table, 
	      click the <TT><B>Edit</B></TT> 
	      button to modify it or the <TT><B>Clear</B></TT> button to
	      remove it.
	      <P>
	      <B>Step 3. Select the tables to include in the filter</B><BR>
	      Scroll down to the <em>Linked Tables</em>
	      section of the page. The tables listed in this section are
	      linked to the selected table by one or more common fields (typically
	      a name, accession, or ID field). Click the boxes in front of the
	      table(s) whose fields you wish to include in the filter,
	      then click the <TT><B>Allow Filtering Using Field in Checked 
	      Tables</B></TT> button. 
	      The fields of the selected tables will be displayed in the top
	      portion of the page. 
	      <P>
	      <B>Step 4. Add the filter constraints</B><BR>
	      <P>
	      <B>Step 5. Click the Submit button to apply the filter</B><BR>
	      <P>
	      <B>Note:</B> In the current implementation of the Table Browser,
	      the <em>selected fields from primary and related tables</em> output
	      format option must be used when including fields from multiple
	      tables in a filter. Check the boxes for all tables in the 
	      <TT><B>Linked Tables</B></TT> list on which filter constraints 
	      have been applied, then click the <TT><B>Allow Selection From
	      Checked Tables</B></TT> button to include them in the output.
	      <P>

<!------Filter Constraints------------------>
<A NAME="FilterConstraints"></A>
	      <HR>
	      <B><FONT SIZE="3" COLOR=#0000FF>Filter constraints</FONT></B>
	      <P>
	      <B>Strings</B><BR>
	      Text fields are compared to words or patterns containing wildcard 
	      characters. Valid wildcards are "*" (matches 0 or more characters) 
	      and "?" (matches a single character). Each space-separated 
	      word or pattern in a text field box is matched against the 
	      value of that field in each record. If any word or pattern matches 
	      the value, then the record meets the constraint on that field. 
	      <P>
	      <B>Numbers</B><BR> 
	      Numeric fields are compared to table data using an operator such 
	      as <, >, != (not equals) followed by a number. To specify a range, 
	      enter two numbers (start and end) separated by white space and/or 
	      a comma. 
	      <P>
	      <B>Free-form queries</B><BR> 
	      When the filters on individual fields aren't sufficiently flexible,
 	      the <TT><B>free-form query</B></TT> text box allows the application
	      of more complex constraints that typically relate two or more 
	      field names of the selected table. Valid free-form queries use 
	      the syntax of the SQL 
	      <em><A HREF="http://www.w3schools.com/sql/sql_where.asp" 
	      TARGET=_blank>where</A></em> clause (using
	      wildcards as defined above). 
	      <P>
	      Free-form queries combine simple 
	      constraints with <em>AND</em>, <em>OR</em>, and <em>NOT</em> 
	      using parentheses as needed for clarity.  A simple constraint 
	      consists of a table field name, a comparison operator (see below), 	         and a value: a number, string, wildcard value (see below), or 
	      another field name. In place of a field name, you may use an 
	      arithmetic expression of numeric field names. 
	      <P>
	      <UL>
	      <LI>
	      String or wildcard values for text comparisons must be quoted. 
	      Single or double quotes may be used. If comparing to a literal 
	      string value, use the "=" or "!=" operator. If comparing 
	      to a wildcard value, use the "LIKE" or "NOT LIKE" 
	      operator. 
	      <LI>
	      Numeric comparison operators include <, <=, =, != (not equals), 
	      >=, and >. 
	      <LI>Arithmetic operators include +, -, *, and /. 
	      <LI>Other SQL comparison keywords may also be used. 
	      </UL>
	      <P>
	      <B><em>Example:</em></B><BR>
	      The following examples show free-form queries applied to the 
	      human <em>refGene</em> table).
	      <UL>
	      <LI>
	      <TT><B>txStart = cdsStart</B></TT>  - 
	      searches for gene models 
	      missing expected 5' UTR upstream sequence (if strand is '+'; 3' UTR
	      downstream if strand is '-')
	      <LI>
	      <TT><B>chrom NOT LIKE "chr??"</B></TT>
	      - restricts search to chromosomes 1 - 9,  X and Y
	      <LI>
	      <TT><B>cdsEnd - cdsStart) > 10000</B>
	      </TT> - selects genes with coding spanning more than 10 kbp
	      <LI>
	      <TT><B>txStart != cdsStart) AND 
	      (txEnd != cdsEnd) AND exonCount = 1</B></TT> -  
	      finds single exon genes with both 3' and 5' flanking UTR.
	      <LI>
	      <TT><B>cdsEnd - cdsStart) > 30000) AND 
	      (exonCount=2 OR exonCount=3)</B></TT> - finds genes with 
	      long spans but only 2 - 3 exons
	      </UL
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>

<!------Intersecting Tables----------------->
<A NAME="Intersection"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	  <TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	       <FONT SIZE="4"><B>&nbsp; Intersecting data from multiple tables</B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
	      It is often interesting to compare the positions of features
	      in different annotation tracks to identify points of overlap. The 
	      Table Browser <TT><B>intersection</B></TT> 
	      utility can be used to generate various position-based 
	      comparisons of track features. Using the 
	      <TT><B>intersection</B></TT> utility, you can:
	      <UL>
	      <LI>
	      examine all genomic positions where the feature data from the two 
	      tracks overlap
	      <LI>
	      identify genomic locations where there is no overlap between 
	      track features
	      <LI>
	      establish thresholds for the amount of overlap that must exist 
	      between the two feature sets
	      <LI>
	      conduct feature-by-feature comparisons as well as
	      base-by-base comparisons of tracks
	      <LI>
	      complement (invert) a position set before comparing the tracks 
	      </UL>
	      <P>
	      An 
	      intersection may be expanded to include additional tables by using 
	      the Table Browser custom track feature.
	      <P>
	      <B>Note:</B> The <TT><B>intersection</B></TT> utility can be used 
	      only on positional tables. To 
	      generate intersections incorporating data in non-positional tables, 
	      use the Table Browser <TT><B>filter</B></TT> utility. See the 
	      <A HREF="#FilterMultiple">Filtering on fields from multiple 
	      tables</A> section for more information.
	      <P>

<!------Intersecting 2 Tables--------------->
<A NAME="SimpleIntersection"></A>
	      <HR>
	      <B><FONT SIZE="3" COLOR="#0000FF">Intersecting data from two tables</FONT></B>
	      <P>
	      Follow these steps to configure and generate an intersection between
	      two positional tables:
	      <P>
	      <B>Step 1. Select the assembly, track, table, and region for the 
	      primary table</B><BR>
	      <B>Note:</B> Only positional tables may be used in an intersection.
	      <P>
	      <B>Step 2. Click the <TT><B>Create</B></TT> button on the 
	      <TT><B>intersection</B></TT> line</B><BR>
	      <B>Note:</B> If an intersection already exists on the table, 
	      click the <TT><B>Edit</B></TT> 
	      button to modify it or the <TT><B>Clear</B></TT> button to
	      remove it.
	      <P>
	      <B>Step 3. Select the secondary track to include in the filter</B><BR>
	      Select a group in the <TT><B>group</TT></B> list, then select a 
	      track from the <TT><B>track</TT></B> list. To view all the
	      tracks available, regardless of group, select the 
	      <em>All Tracks</em> option in the <TT><B>group</TT></B> list.
	      <P>
	      <B>Step 4. Select a combination method</B><BR>
	      The Table Browser provides two major types of comparisons:
	      <UL>
	      <LI>
	      <em>feature-by-feature</em> comparisons preserve the structure of 
	      the primary table. For example, if the primary 
	      table describes exon structure and the features are compared
	      with a second table, the results will describe exon structure 
	      (unless you choose an output format in which the structure is 
	      lost). 
	      <LI>
	      <em>base-by-base</em> comparisons examine 
	      the primary table and the table underlying the secondary track one 
	      base at a time. The structure 
	      of the primary table is not preserved in this comparison. For 
	      example, even if the primary table describes 
	      exon structure, the intersection results will contain only position 
	      ranges; no information about exon/block structure, strand, or 
	      translation region will be retained. 
	      </UL>
	      Click the circle in front of a combination method to select it. 
	      Only one method may be selected from the two sets of methods. For
	      more information about the individual combination options, see the 
	      <A HREF="#IntersectionOptions">Intersection Options</A> section.
	      <P>
	      <B>Step 5. (optional) Select the complement options</B><BR>
	      Check the box in front of one or both tables to complement the
	      feature data in the The complement options allow you to invert the 
	      set of positions
	      covered by one or both tables. For example, if you choose to 
	      complement the primary track, any position covered by the that
	      track's features will be considered <em>not</em> covered, and vice 
	      versa. This option provides more flexibility in comparing track 
	      positions.
	      <P>
	      <B>Step 6. Click the <TT><B>Submit</B></TT> button to apply the 
	      intersection</B><BR>
	      Once an intersection has been created on a table, it will persist 
	      for the duration of the Table Browser session or until it has been 
	      cleared. Only one intersection may exist at a time. 
	      To modify an existing intersection, click the <TT><B>Edit</B></TT> 
	      button on the <TT><B>intersection</B></TT> line</B>. To remove an 
	      intersection, click the <TT><B>Clear</B></TT> button.
	      <P>

<!------Intersecting Multiple Tables------->
<A NAME="MultiIntersection"></A>
	      <HR>
	      <B><FONT SIZE="3" COLOR="#0000FF">Intersecting data from more than two tables</FONT></B>
	      <P>
	      The Table Browser <TT><B>intersection</B></TT> utility limits
	      combinations to only two tables. An existing intersection may be
	      expanded to include additional tables by using the Table Browser
	      custom track utility. To create an intersection on multiple tables:
	      <P>
	      <B>Step 1. Set up an intersection between two tables</B><BR>
	      See the <A HREF="#SimpleIntersection">Intersecting data 
	      from two tables</A> section for more information.
	      <P>
	      <B>Step 2. Save the intersection data in a custom track</B><BR>
	      See the <A HREF="#CustomTrack">Saving data as a custom track</A>
	      section for information on generating a custom track. 
	      <B>Note:</B> In the 
	      current implementation of the Table Browser, you must use the
	      <TT><B>Get Custom Track</B></TT> button on the custom track page
	      to add the custom track to the Table Browser <TT><B>track</B></TT>
	      list.
	      <P>
	      <B>Step 3. Select the newly-generated custom track</B><BR>
	      Select the <em>Custom Tracks</em> option in the 
	      <TT><B>group</B></TT> list, then select the newly-created custom
	      track from the <TT><B>track</B></TT> list.
	      <P>
	      <B>Step 4. Create an intersection with another track</B><BR>
	      Follow the steps in the 
	      <A HREF="#SimpleIntersection">Intersecting data from two tables</A> 
	      section to intersect the custom track with another track.
	      <P>

<!------Intersecting 2 Tables--------------->
<A NAME="IntersectionOptions"></A>
	      <HR>
	      <B><FONT SIZE="3" COLOR="#0000FF">Intersection options</FONT></B>
	      <P>
	      <B>Feature-by-feature comparisons</B><BR> 
	      Some comparisons preserve the primary table's gene and alignment 
	      structure, if it exists. For example, if the <em>refGene</em>
	      table (human <em>RefSeq Genes</em> track) is combined with another
	      table  using one of these comparisons, the resulting output data
	      will describe exon structure (unless you choose an output format 
	      in which the structure is lost). Primary table features are kept 
	      or discarded based on the amount of positional overlap with 
	      the features in the table underlying the secondary track. The 
	      Table Browser offers the following options in this category:
	      <UL>
	      <LI>
	      <B>Any overlap:</B> A primary table record will appear in the 
	      output if any of its base positions are covered by any 
	      feature in the secondary table. 
	      <LI>
	      <B>No overlap:</B> A primary table record will appear in the 
	      output only if none of its base positions are covered by any 
	      feature in the secondary table. 
	      <LI>
	      <B>Overlap greater than a specified threshold:</B> A primary table 
	      record will appear in the output if the percentage of its 
	      base positions covered by secondary table features is greater than 
	      the user-specified threshold. 
	      <P>
	      <LI>
	      <B>Overlap less a specified threshold:</B> A primary table record 
	      will appear in the output if the percentage of its base positions 
	      covered by secondary table features is less than the user-specified
	      threshold. 
	      </UL>
	      <P><B>Note:</B> If the primary table has an exon/block structure, 
	      only those bases located in exons and/or blocks will be counted. 
	      <P>
	      <B>Base-by-base comparisons</B><BR>
	      In these combination options, the positions of the primary 
	      and secondary table features are compared one base position at a 
	      time. When applying base-by-base comparisons, the structure of the 
	      primary table is not preserved. For example, if the <em>refGene</em>
	      table (from the human <em>RefSeq Genes</em> track) is compared with
	      a secondary table using these comparisons, the resulting output
	      data will not describe exon structure. Instead, only position 
	      ranges will be returned; the exon/block structure, strand, and
	      translation region information will be discarded. The Table Browser
	      provides the following base-by-base combination options:
	      <UL>
	      <LI>
	      <B>Base-by-base intersection (AND):</B> A nucleotide position 
	      is included in the output if it is covered by at least one feature 
	      of both the primary table <em>and</em> the secondary table.
	      <LI>
	      <B>Base-by-base union (OR):</B> A nucleotide position is 
	      included in the output if it is covered by at least one feature of 
	      either the primary table <em>or</em> the secondary table. 
	      </UL>
	      <P>
	      <P><B>Note:</B> If the primary table has an exon/block structure, 
	      only base positions located in exons and/or blocks will be counted. 
	      <P>
	      <B>Base-by-base complement (NOT)</B><BR>
              Before the Table Browser applies a feature-by-feature or 
	      base-by-base comparison to the table data, the set of positions 
	      covered by one or both tables can be inverted (complemented). When 
	      the data set of a table is complemented, any position 
	      covered by the table's features in the original data will be 
	      considered not covered in the inverted data, and vice versa. 
	      This option gives the user more flexibility in comparing table 
	      positions. 
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>

<!------Output Formats---------------------->
<A NAME="OutputFormats"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	  <TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	       <FONT SIZE="4"><B>&nbsp; Output Formats</B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
	      <P>
	      The data resulting from a Table Browser query may be configured in
	      a number of different ways: 
	      <UL>
	      <LI>
	      The output can be displayed on the
	      screen, saved to a file, or saved to an annotation track table 
	      that can be displayed in the Genome Browser or used in a subsequent
	      Table Browser query.
	      <LI>
	      The data can include all fields from the primary or selected table, 
	      or can be restricted to selected fields from the
	      primary table and related tables.
	      <LI>
	      The data can be organized in one of several formats: tab-separated, 
	      sequence (FASTA), Browser Extensible Data format (BED), 
	      Gene Transfer Format (GTF), or a statistical summary of the data
	      in the query.
	      </UL>
	      The output options available for a specific query may vary
	      depending on the table(s) selected. For example, non-positional
	      table data cannot be organized in a position-based format, but 
	      instead may be displayed only in tab-separated format. The Table 
	      Browser will automatically update the options on the 
	      <TT><B>output format</B></TT> list to show only those available
	      for the current query.
	      <P>

<!------Display All Fields------------------>
<A NAME="AllFields"></A>
	      <HR>
	      <B><FONT SIZE="3" COLOR="#0000FF">Displaying all fields in a table</FONT></B>
	      <P>
	      To display all the fields of the records in the query output
	      in tab-separated format, select the <em>all fields from primary
	      table</em> option.
	      <P>

<!------Display Selected Fields------------->
<A NAME="SelectedFields"></A>
	      <HR>
	      <B><FONT SIZE="3" COLOR="#0000FF">Displaying selected fields from one or more tables</FONT></B>
	      <P>
	      To restrict the query output to a subset of the fields in a table, 
	      choose
	      the <em>selected fields from primary and related tables</em> option.
	      You will be prompted to pick the table fields to display. Click the 
	      box in front of the fields you would like to see in the query 
	      output (or
	      click the <TT><B>Check All</B></TT> button to select all the 
	      fields), then click the <TT><B>Get Fields</B></TT> button.
	      <P>
	      To include data fields from other tables linked to the selected
	      table, choose the <em>selected fields from primary and related 
	      tables</em> option, then scroll down to the <em>Linked Tables</em>
	      section of the page. The tables listed in this section are
	      linked to the selected table by one or more common fields (typically
	      a name, accession, or ID field). Click the boxes in front of the
	      table(s) whose fields you wish to include in the query output,
	      then click the <TT><B>Allow Selection From Checked Tables</B></TT>. 
	      The fields of the selected tables will be displayed in the top
	      portion of the page. Click the boxes in front of the fields
	      that you wish to include in the query output, then click the 
	      <TT><B>Get Fields</B></TT> button underneath any of the field
	      lists to generate tab-separated output that includes data from
	      all the selected fields. Note that the <TT><B>Get Fields</B></TT>
	      and <TT><B>Cancel</B></TT> buttons apply globally to all the 
	      selected tables, but the <TT><B>Check All</B></TT> and 
	      <TT><B>Clear All</B></TT> buttons apply only to the fields listed
	      directly above the buttons.
	      <P>

<!------Display Sequence-------------------->
<A NAME="Sequence"></A>
	      <HR>
	      <FONT SIZE="3" COLOR="#0000FF"><B>Displaying sequence (FASTA) data</B> (positional tables only)</FONT>
	      <P>
	      To display the genomic sequence underlying the query results, 
	      select the <em>sequence</em> option in the <TT><B>output 
	      format</B></TT> list. The Table Browser will present you with 
	      several options to configure the output display. When you have
	      completed the configuration, click the <TT><B>Get Sequence</B></TT>
	      button.
	      When displaying sequence data for tracks in the <em>Genes and 
	      Gene Predictions</em> group, you will be offered the option to
	      view the protein sequence associated with the track rather than
	      the genomic sequence.
	      <P>

<!------ CDS FASTA alignments -------------------->
<A NAME="FASTA"></A>
	  <HR>
	  <FONT SIZE="3" COLOR="#0000FF"><B>Displaying CDS FASTA alignments</B> (genePred tables only)</FONT>
          <P>
          The CDS FASTA alignments are created from a Multiple Alignment File
          (<A HREF="../../FAQ/FAQformat#format5">MAF</A>) in
          combination with a 
          <A HREF="../../FAQ/FAQformat#format9">genePred</A> table.  
          The UCSC MAF format stores 
          multiple alignments at the DNA level between entire genomes. You can
          use the Table Browser to return FASTA alignments of coding regions in
          nucleotide-space or translated into
          amino acid-space.  However, it is worth noting that the initial
          MAF files are all created by aligning genomes at the DNA level.
          </P>
          <P>
	  <B>Genome-wide CDS FASTA alignments</B><BR>
          Note that when using
          the Table Browser to fetch CDS FASTA output, it is best to 
          restrict your query to a reasonable-sized position range rather than 
          requesting output from the entire genome.  A genome-wide
          query will take a substantial amount of compute time, and it is
          likely that your Internet browser will time out and disconnect.
          If you would like to download genome-wide CDS FASTA output 
          for any of several model organisms, you can do so from the 
          <A HREF="http://hgdownload.cse.ucsc.edu">download server</A>.
          </P>
	  <P>
          <B>Creating CDS FASTA alignments using the Table Browser</B><BR>
          To display FASTA multiple alignments for the CDS regions of genes,
          select the <em>CDS FASTA alignment from multiple alignment</em> 
          option in the <TT><B>output format</B></TT> list. In order to see
          this output format option, you must have a genePred table 
          selected. 
          If you limit your search to a certain position range within the
          genome (rather than searching the entire genome), the tool
          will return FASTA alignments for all genes that overlap
          the position for which you are searching. 
          The Table Browser will present you with a configuration page.
          On this page, you can select options for your ouput.
          </P>
          <P>
          First, select your MAF table.  This is the table from which 
          the multiple alignments will be extracted for the CDS regions
          of your gene track. If you do not know the name of the MAF table
          that corresponds to the Conservation track, you can find it in the
          Genome Browser by following these 
          <A HREF="../../FAQ/FAQtracks#tracks21">instructions</A>. 
          </P>
	  <P>
          Then select any of the following choices:
	    <UL>
	    <LI><B>Separate into exons</B> -
	    The default behavior is for the coding exons of each gene to be
            concatenated into
	    one sequence in the output FASTA multiple alignment. In this
	    case each output row header has the format listed
	    below under &quot;Whole gene format&quot;.
	    If the separate into exons option is chosen then each exon will be
	    listed with a separate header in the format
	    listed below under &quot;Exon format&quot;.<br>
            <LI><B>Show nucleotides</B> -
	    The default behavior is for the nucleotides in the alignment
            to be translated
	    into amino acids according to the strand and exon frames
	    defined in the selected genePred table.
            If this option is chosen, then the nucleotides
	    in the alignment will not be translated into amino
	    acids.        
            <LI><B>Output lines with just dashes</B> - 
	    The default behavior is for the alignment rows that contain
            only dashes to not be printed.  If this option is chosen, then
	    these dashes-only rows are printed.
            <LI><B>Format output as table</B> - 
            If this option is chosen, the header and sequence for each organism
            will appear on the same line.
              <UL>
              <LI><B>Truncate headers as &mdash; characters (enter zero for
              no headers)</B> -
              This option works in conjunction with the &quot;Format output
              as table&quot; option. 
              If you do not wish to see the entire header, choose this option,
              and enter the number of characters at which you would like the 
              header truncated.
              </UL>
            </UL>
         Finally, from the list of species, select those that you would
         like included in the FASTA multiple alignment output.  Press
         the &quot;get output&quot; button to view the output. 
         <P>
            <B>Explanation of CDS FASTA header format</B><BR>
 
            &mdash; Whole gene format:
            <tt>geneName_assemblyName peptideLength location</tt><br>
	    &mdash; Exon format:
	    <tt>geneName_assemblyName_exonNum_totalExons exonLength inFrame 
            outFrame location</tt><br><br>

	    Following are the descriptions for each field name:
	      <UL>
	      <LI><B>geneName</B>- the name field from the genePred table.
	      <LI><B>assemblyName</B>- the UCSC assembly 
              <A HREF="../../FAQ/FAQreleases#release1">name</A>
              for the species.
	      <LI><B>peptideLength</B>- the length of the entire coding region.
	      If the &quot;Show nucleotides&quot; option is chosen, this will 
              be in nucleotides, otherwise it will be the number of amino 
              acids in the peptide.
	      <LI><B>location</B>- this is the chromosome position within the 
	      assembly that is aligned in the multiple alignment.  The format
	      of this string is chrom:start-end followed by the strand
	      where the alignment occurs.  If more than one region is
	      aligned then all the regions are listed with a semi-colon (;)
	      between each position.  This address is in genome browser
	      coordinates (i.e. the start address is 
              <A HREF="../../FAQ/FAQtracks#tracks1">one-based</A>).
	      <LI><B>exonNum</B>- the ordinal of the exon.  Exons are
	      counted starting at one and begin at the transcription start site
	      and progress along the strand of transcription.
	      <LI><B>totalExons</B>- the number of coding exons in the gene.
	      <LI><B>exonLength</B>- the length of the current exon.
	      If the &quot;Show nucleotides&quot; option is chosen, this will 
              be the 
	      number of nucleotides in the exon, otherwise it will be 
	      the number of amino acids in 
	      the exon (with amino acids translated from split codons 
	      placed in the exon where two of the three nucleotides lie).
	      <LI><B>inFrame</B>- the frame number of the first nucleotide
	      in the exon. Frame numbers can be 0, 1, or 2 depending on
	      what position that nucleotide takes in the codon which
	      contains it.
	      <LI><B>outFrame</B>- the frame number of the nucleotide
	      after the last nucleotide in this exon.
	      Frame numbers can be 0, 1, or 2 depending on
	      what position that nucleotide takes in the codon which
	      contains it.
	      </UL>
         <P>
         <B>Explanation of CDS FASTA sequence format</B><BR>
         As noted above, the CDS FASTA output files can be in either DNA-space 
         or protein-space.
         </P>
         <P>
         In some instances, there is a dash (&quot;&ndash;&quot;) in the 
         sequence portion of the CDS FASTA file. Dashes are used in several 
         circumstances.  They indicate missing sequence for the aligning 
         genome, as well as deletions in the aligning genome or insertions
         in the base genome.
         </P>
         <P>
         Because the CDS FASTA alignments are based on one reference
         genome, any amino acids or nucleotides that are not
         in the reference genome are not displayed.  Consequently the peptides
         shown for aligning genomes are not necessarily the peptide that
         the other organisms' gene would generate.  Any sequence inserted
         in an aligning genome or deleted in the base genome will not be 
         present in the alignment.  We represent this condition
         with an orange bar in the Genome Browser display, but
         the CDS FASTA alignments silently ignore this issue. 
         </P>
         <P>
         &mdash; Nucleotide CDS FASTA sequence:<BR>
         Consider the example below that shows the FASTA sequence for four
         species aligned with the first exon of the human gene
         PLEKHO1 (UCSC Gene: uc001ett.1). Note that the rat (rn4) row 
         is missing 
         the first three nucleotides.  This could be due to a 
         lineage-specific insertion between the rat and  
         human genomes, or a lineage-specific deletion between the
         human and rat genomes. Note also that the Zebrafish 
         (danRer4) row contains only dashes.  This could be due to 
         excessive evolutionary distance between the zebrafish and human, 
         missing data in the zebrafish, or independent indels in the region in
         both species. Sometimes it is helpful to view the  
         Conservation track in the Genome Browser in this area to
         clarify the exact meaning of the dashes.
<PRE>
&gt;uc001ett.1_hg18_1_6 30 0 0 chr1:148389072-148389101+
ATGATGAAGAAGAACAATTCCGCCAAGCGG
&gt;uc001ett.1_panTro2_1_6 30 0 0 chr1:129156502-129156531+
ATGATGAAGAAGAACAATTCCGCCAAGCGG
&gt;uc001ett.1_rn4_1_6 30 0 0 chr2:190795892-190795918-
---ATGAAGAAGAGCGGCTCCGGCAAGCGG
&gt;uc001ett.1_danRer4_1_6 30 0 0
------------------------------
&gt;uc001ett.1_oryLat1_1_6 30 0 0 chr11:3404940-3404969-
AGGATGAAGAAAAGCAACCAGAGCAGGCGG
</PRE>

         &mdash; Amino Acid CDS FASTA sequence:<BR>
         <UL>
          <LI>Codons that have a dash in any of the three nucleotides are 
          represented by a dash in the amino acid.  
          <LI>Codons with an N in any position are represented with an X.  
          <LI>Stop codons are represented with a Z.  
          <LI>All other amino acids follow the IUPAC amino acid codes.
          <LI>In exon format, when the codon triplet is split between two 
          exons, the amino acid will be displayed as part of the exon
          containing two of the three nucleotides like so:
         </UL>

<PRE><TT>
            |exon1|  |exon2|
nucleotide: AAACCCT  TTGGGAAA
protein:     K  P    F  G  K 
</TT></PRE>

         </P>


<!------GTF/BED Format---------------------->
<A NAME="GTF_BED"></A>
	      <HR>
	      <FONT SIZE="3" COLOR="#0000FF"><B>Saving query results in GTF or BED format</B> (positional tables only)</FONT></B>
	      <P>
	      To format the query results using 
	      <A HREF="http://genome.ucsc.edu/FAQ/FAQformat#format4">GTF</A> or 
	      <A HREF="http://genome.ucsc.edu/FAQ/FAQformat#format1">BED</A> conventions,
	      select the corresponding option in the <TT><B>output 
	      format</B></TT> list. Note that when you select GTF, the table browser
	      translates the output into this format. For tables that lack feature
	      designations, all records are arbitrarily assigned the feature "exon" to
	      conform to GTF specifications.  
	      If you select BED format, you will be
	      presented with the option to include and configure a custom track
	      header and options for organizing the data. When you have finished
	      the configuration -- or to accept the default options -- click
	      the <TT><B>Get BED</B></TT> button at the bottom of the window.
	      <P> 

<!------Save to File------------------------>
<A NAME="SaveFile"></A>
	      <HR>
	      <B><FONT SIZE="3" COLOR="#0000FF">Saving data to a file</FONT></B>
	      <P>
	      By default, the Table Browser displays query results directly in
	      your internet browser window. To redirect the data to a file,
	      type a file name into the <TT><B>output file</B></TT> box before
	      starting the query. The Table Browser will prompt you for the
	      location of this file on your local disk while processing the
	      query.
	      <P>

<!------Save as Custom Track---------------->
<A NAME="CustomTrack"></A>
	      <HR>
	      <FONT SIZE="3" COLOR="#0000FF"><B>Saving data as a custom track</B> (positional tables only)</FONT>
	      <P>
	      Query output may be saved in a format that can be displayed as
	      a custom annotation track in the Genome Browser. Custom tracks
	      created during a Table Browser session may also be used for 
	      subsequent queries and intersections in the same session. For 
	      more information on custom tracks, see the Genome Browser
	      <A HREF="hgTracksHelp.html#CustomTracks">User's Guide</A>.
	      <P>
	      To save query data in custom track format, select the <em>custom
  	      track</em> option in the <TT><B>output format</B></TT> list. When
	      the query is executed, the Table Browser will prompt you to
	      customize the track header and configure the record layout of
	      the data. The configuration is optional; the Table Browser
	      automatically sets up a default track configuration. Click the 
	      <em>Custom track</em> link for more information on custom track 
	      syntax and format.
	      <P>
	      When you have finished configuring the custom track -- or to accept
	      the default configuration -- click one of the buttons at the bottom
	      of the window to create the custom annotation track. 
	      <UL>
	      <LI>
	      To display the query results as text on the screen, click the 
	      <TT><B>Get Custom Track File</B></TT> button. 
	      <LI>
	      To save the query results to a file on your local disk for future 
	      use, specify a file name in the <TT><B>output file</B></TT> box
	      before executing the query, then click the <TT><B>Get Custom Track 
	      File</B></TT> button. 
	      <LI>
	      To load the query results into a table accessible from the 
	      Table Browser <TT><B>table</B></TT> list, click the 
	      <TT><B>Get Custom Track in Table Browser</B></TT> button. 
	      <LI>
	      To view the query results as a custom track in the Genome 
	      Browser, click the <TT><B>Get Custom Track in Genome 
	      Browser</B></TT> button. Your 
	      browser display will be redirected automatically to the Genome 
	      Browser, with your custom track positioned near the top of the 
	      annotation tracks window. 
	      <LI>
	      To access your custom track data in a subsequent query in the same
	      Table Browser session, select the <em>Custom Tracks</em> option
	      from the <TT><B>group</B></TT> list to display the custom tracks
	      available. 
	      </UL>
	      <P>

<!------Display as Hyperlink--------------->
<A NAME="Hyperlink"></A>
	      <HR>
	      <FONT SIZE="3" COLOR="#0000FF"><B>Displaying query results as Genome Browser hyperlinks</B> (positional tables only)</FONT>
	      <P>
	      To examine the records in the query output individually in the
	      Genome Browser, select the <em>hyperlinks to Genome Browser</em>
	      output option. The Table Browser will display a list of one or
	      more hyperlinks corresponding to the individual records in the 
	      output data. Click a link to open up the Genome Browser display
	      to the item and position shown on the hyperlink.
	      <P>

<!------Display Summary/Statistics--------->
<A NAME="SummaryStats"></A>
	      <HR>
	      <FONT SIZE="3" COLOR="#0000FF"><B>Displaying a statistical summary
	      of query data</B> (positional tables only)</FONT>
	      <P>
	      To generate a statistical summary of the query output data, the
	      region covered by the query, and the CPU time required to process 
	      the query, click the <TT><B>Summary/Statistics</B></TT> button.

	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>

<!------Table Schema------------------------>
<!--FIXME - ADD
<A NAME="TableSchema"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>
	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	  <TR><TD>
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
	       <FONT SIZE="4"><B>&nbsp; Table Schema</B></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	  <BR></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
    <P>
-->




<!------End of Document--------------------->
  </TD></TR>
</TABLE>
</BODY></HTML>

