<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Bioinformatics: User Guide</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg"></TH></TR>

<!--HOTLINKS BAR-->
<TR><TD COLSPAN=2 HEIGHT=40>
	<table bgcolor="000000" cellpadding="1" cellspacing="1" width="100%" height="27"><tr bgcolor="2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
			<TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp; 
			&nbsp;<a href="/index.html" class="topbar">
				Home</a> &nbsp; - &nbsp;
			<a href="/cgi-bin/hgGateway" class="topbar">
				Genome Browser</a> &nbsp; - &nbsp; 
			<A HREF="/cgi-bin/hgBlat?command=start" class="topbar">
				Blat Search</a> &nbsp; - &nbsp; 
			<A HREF="/FAQ.html" class="topbar">
				FAQ</a> &nbsp; - &nbsp; 
			<A HREF="/goldenPath/help/hgTracksHelp.html" class="topbar">
				User Guide</a> &nbsp;</font></TD>
	</TR></TABLE>
	</TD></TR></TABLE>
</TD></TR>

<!--Content Tables-->  
<TD CELLPADDING=10 ROWSPAN=2 VALIGN="top">
<TR><TD CELLPADDING=10>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp;Genome Browser User Guide </B></FONT> 
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
			
<CENTER><TABLE BGCOLOR="ffffdd" BORDERCOLOR="cccc99" BORDER=1 CELLPADDING=3><TR><TD>	
	<CENTER><FONT SIZE="2">Web tool created by <A
	HREF="http://www.cse.ucsc.edu/~kent">Jim Kent</A>, <A
	HREF="http://www.cse.ucsc.edu/~sugnet">Charles Sugnet</A>, <A
	HREF="http://www.cse.ucsc.edu/~booch">Terry
	Furey</A>, <BR>and <A HREF="http://www.cse.ucsc.edu/~haussler">David
	Haussler</A> of UCSC,<BR> in collaboration with the <BR>
	<A HREF="http://www.nature.com/genomics/human/papers/409860a0_fs_1.html">genome analysis group of the International Human Genome Project</A><BR>
	and the <A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/">UCSC Human Genome Project</A>.<BR>
	Individual tracks contributed by various <A HREF="http://genome.cse.ucsc.edu/goldenPath/credits.html">individuals</A> and <A HREF="http://genome.cse.ucsc.edu/goldenPath/other.resources.html">institutions</A>.<BR>
	Last updated 24 Jun 02; questions and suggestions welcome.</FONT></CENTER>
</TD></TR></TABLE></CENTER>
<BR>
<B><FONT COLOR="#006666" SIZE=4> Table of Contents: </FONT></B>
<table border=0 nowrap cellpadding=0 cellspacing=0 margins=0><tr><th height=10></th></tr>
<tr><td><ul><b>
  		<li><A CLASS="toc" HREF="#What">
			What does the Genome Browser do?</A>
  		<li><A CLASS="toc" HREF="#GetStarted">
			Getting started: the Genome Browser Gateway</A>
  		<li><A CLASS="toc" HREF="#FineTuning">
			Fine-tuning the Genome Browser display</A>
  		<li><A CLASS="toc" HREF="#BLATAlign">
			Using BLAT alignments</A>
  		<li><A CLASS="toc" HREF="#TableBrowser">
			Getting Started on the Table Browser</A>
	</b></ul></td><td><ul><b>
		<li><A CLASS="toc" HREF="#TrackFormatDNA">
			DNA text formatting</A>
		<li><A CLASS="toc" HREF="#Convert">
			Converting data between assemblies</A>
		<li><A CLASS="toc" HREF="#Download">
			Downloading genome data</A>
		<li><A CLASS="toc" HREF="#CustomTracks">
			Creating custom annotation tracks</A>
  		<li><A CLASS="toc" HREF="#IndivTracks">
			Annotation track descriptions</A>
	</b></ul></td></tr></table>
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="What"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b><a NAME="TOC"></a><b>&nbsp; What does
			the Genome Browser do?  </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
As vertebrate genome sequences near completion and research re-focuses on 
their analysis, the issue of effective sequence display becomes critical: it is not 
helpful to have 3 billion letters of genomic DNA shown as plain text. As an 
alternative, the UCSC Genome Browser provides a rapid and reliable display of any 
requested portion of genomes at any scale, together with dozens of 
aligned annotation tracks (known genes, predicted genes, ESTs, mRNAs, CpG islands, 
assembly gaps and coverage, chromosomal bands, mouse homologies, and more). Half of 
the annotation tracks are computed at UCSC from publicly available
sequence data. The remaining tracks are provided by collaborators worldwide. Users can 
also add their own custom tracks to the browser for educational or research
purposes.
<P>
The Genome Browser stacks annotation tracks beneath genome 
coordinate positions, allowing rapid visual correlation of different types of 
information. The user can look at a whole chromosome to get a feel for gene 
density, open a specific cytogenetic band to see a positionally mapped disease gene 
candidate, or zoom in to a particular gene to view its spliced ESTs and possible 
alternative splicing. The Genome Browser itself does not draw conclusions; rather, it 
collates all relevant information in one location, leaving the exploration and 
interpretation to the user.
<P>
The Genome Browser supports text and sequence based searches that provide quick, precise 
access to any region of specific interest. Secondary links from individual
entries within annotation tracks lead to sequence details and supplementary off-site 
databases.  To control information overload, tracks need not be displayed in full. Tracks
can be hidden, collapsed into one dense line, or filtered according to the
user's criteria.  Zooming and scrolling controls help to narrow or broaden the 
displayed chromosomal range to focus on the exact region of interest. 
Clicking on an individual item within a track opens a details page containing a 
summary of properties and links to off-site repositories such as 
PubMed, GenBank, LocusLink, and OMIM. The page provides item-specific information on 
position, cytoband, strand, data source, and encoded protein, mRNA, genomic sequence 
and alignment, as appropriate to the nature of the track.
<P>
A blue navigation bar at the top of the browser provides links to several other tools 
and data sources. For instance, the DNA link
enables the user to view the raw genomic DNA sequence for the coordinate range 
displayed in the browser window. This DNA can encode track features via elaborate 
text formatting options. Other links tie the Genome Browser to the BLAT 
alignment tool, provide access to the underlying relational database via the
Table Browser, convert coordinates across different assembly dates, and open the 
window at the complementary <A HREF="http://www.ensembl.org/" TARGET=_blank>Ensembl</A>
or <A HREF="http://www.ncbi.nlm.nih.gov/cgi-bin/Entrez/maps.cgi" TARGET=_blank>NCBI Map 
Viewer</A> annotation.
<P>
The browser data represents an immense 
<A HREF="http://genome.cse.ucsc.edu/goldenPath/credits.html">collaborative effort</A> 
involving thousands of people from the international biomedical research community. 
UCSC itself does no sequencing and creates only about half of the tracks in-house. Its 
other major roles include building genome assemblies, creating the Genome Browser work 
environment, and serving it online. The majority of the sequence data, annotation tracks, and even 
software are in the public domain and are available for anyone
to <A HREF="#Download">download</A>. 
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="GetStarted"></A><BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Getting Started: the
		Genome Browser Gateway</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
The UCSC Genome Bioinformatics <A HREF="http://genome.ucsc.edu" TARGET=_blank>home
page</A>
provides access to Genome Browsers on several different genome assemblies. 
To get started, select the organism type from the list in the left sidebar, then
click the Browser link directly below it. This will take you
to a Gateway page for the selected organism. 
<P>
<B>Opening the Genome Browser at a specific position</B><BR> 
To get oriented in using the Genome Browser, try viewing a gene or region of the genome with 
which you are already familiar, or use the default position. To open the Genome Browser
window:
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top"><B>1.</B></TD><TD>
	Select the assembly that you want to display from the <I>assembly</I> menu.
	To access older assembly versions that are no longer available from the menu,
	look in the Genome Browser <A HREF="http://genome-archive.cse.ucsc.edu/"
	TARGET=_blank>archives</A>.
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top"><B>2.</B></TD><TD>
	Specify the genome location that you'd like the Genome Browser to open to. To 
	select a location, you must enter a valid position query in the <I>position</I> 
	text box at the top of the Gateway page, or accept the default position already 
	displayed. The position search supports several different types of queries: gene 
	symbols, mRNA or EST accession numbers, chromosome 
	bands, descriptive terms likely to occur in GenBank text, or specific chromosomal 
	ranges. The Gateway page shows examples of some of the position requests supported by
	the Genome Browser.
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top"><B>3.</B></TD><TD>
	Click the <I>Submit</I> button to open up the Genome Browser window to that location. 
	</TD></TR>
</TABLE>
<P>
Occasionally the Gateway page returns a list of several matches in response to a 
position search, rather than immediately displaying the Genome Browser window. When
this occurs, click on the item in which you're interested and the Genome Browser
will open to that location.
<P>
The position search mechanism is not a site-wide search engine. Instead, it primarily 
searches GenBank mRNA records whose text annotations can include gene names, gene 
symbols, journal title words, author names, and RefSeq mRNAs. Searches on NP and
NM accession numbers, OMIM identifiers, and LocusLink identifiers are supported. 
However, some types of position queries will return an error, e.g. post-freeze GenBank 
entries, withdrawn gene names, and abandoned synonyms. 
<P>
<B>Finding a genome location using BLAT</B><BR>
If you have genomic, mRNA, or protein sequence, but don't know the name or
the location to which it maps in the genome, the <A HREF="http://genome.cse.ucsc.edu/cgi-bin/hgBlat">
BLAT</A> tool will rapidly locate the position by homology alignment, provided that the 
region has been sequenced. This search will find close members of the gene
family, as well as assembly duplication artifacts. An entire set of query sequences can 
be looked up simultaneously when provided in fasta format. 
<P>
A successful BLAT search 
returns a list of one or more genome locations that match the input sequence. To
view one of the alignments in the Genome Browser, click the <I>browser</I> link for the 
match. The <I>details</I> link can be used to preview the alignment to determine if
it is of sufficient match quality to merit viewing in the Genome Browser. If too many BLAT hits 
occur, try narrowing the search by filtering the sequence in slow mode with
<A HREF="http://ftp.genome.washington.edu/cgi-bin/RepeatMasker" TARGET=_blank>RepeatMasker</A>, then rerunning the BLAT search. 
<P>
For more information
on conducting and fine-tuning BLAT searches, refer to the <A HREF="#BLATAlign">
BLAT section</A> of this document.
<P>
<B>Opening the Genome Browser with a custom annotation track</B><BR>
You can open the Genome Browser window with a custom annotation track displayed
by using the Add Your Own Tracks section at the bottom of the Gateway page. For more information on creating and using custom annotation tracks, refer to the <A
HREF="customTrack.html" TARGET=_blank>Custom 
Annotation Track</A> documentation.
<P>
Annotation track data can be entered in one of three ways:
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Enter the file name for an annotation track source file in the
	Annotation File text box.
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Type or paste the annotation track data into the large text box.
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	If the annotation data is accessible through a URL, enter the URL name in the 
	large text box.
	</TD></TR>
</TABLE>
<P>
Once you've entered the annotation information, click the <I>Submit</I> button at the top of 
the Gateway page to open up the Genome Browser with the annotation track
displayed.
<P><B>NOTE:</B> If an annotation track does not display correctly when
you attempt to upload it, you may need to reset the Genome Browser to 
its default settings, then reload the track. For information on
troubleshooting display problems with custom annotation tracks, refer to the
section on <A
HREF="customTrack.html#TROUBLE" 
TARGET=_blank>Troubleshooting Annotation Display Problems</A>.
<P>
<B>Viewing genome data as text</B><BR> 
The 
<A HREF="http://genome.cse.ucsc.edu/goldenPath/hgText.html" TARGET=_blank>Table Browser</A>,
a portal to the underlying open source <A
HREF="http://www.MySQL.com" TARGET=_blank>MySQL</A> relational database driving the Genome
Browser, displays genomic data as columns of text rather than as
graphical tracks. For more information on using the Table Browser, see the
section <A HREF="#TableBrowser"> Getting Started: on the Table Browser</A>.
<P>
<B>Opening the Genome Browser from external gateways</B><BR> 
Several external gateways provide direct links into the Genome Browser. Examples
include: <A HREF="http://www.ncbi.nlm.nih.gov/LocusLink/" TARGET=_blank>LocusLink</A>, 
<A HREF="http://www.ncbi.nlm.nih.gov/IEB/Research/Acembly/index.html"
TARGET=_blank>AceView</A>,
<A HREF="http://www.ensembl.org/" TARGET=_blank>Ensembl</A>, <A
HREF="http://supfam.mrc-lmb.cam.ac.uk/SUPERFAMILY/" TARGET=_blank>SuperFamily</A>, <A
HREF="http://www.genelynx.org/" TARGET=_blank>GeneLynx</A>, and <A
HREF="http://bioinformatics.weizmann.ac.il/cards/" TARGET=_blank>GeneCards</A>. Journal
articles can also link to the browser and provide custom tracks. Be sure to use
the assembly date appropriate to the provided coordinates when using data from
a journal source.
<P>
<B>Tips for Use</B><BR>
To facilitate your return to regions of interest within the Genome
Browser, save the coordinate range or bookmark the page of displays that you plan to 
revisit or wish to share with others. 
<P>
It is usually best to work with the most recently released assembly even though a full 
set of tracks might not yet be ready. Be aware that the coordinates of a given feature 
on an unfinished chromosome may change from one assembly to the next as
gaps are filled, artifactual duplications are reduced, and strand orientations
are corrected.  The Genome Browser offers a tool that can correctly 
convert coordinates between different assembly releases. To access the <A
HREF="http://genome.cse.ucsc.edu/cgi-bin/hgCoordConv" TARGET=_blank>conversion
tool</A>, click the Convert link on the top menu bar
on the Genome Browser window.
<P>
To ensure uninterrupted browser services for your research during UCSC server 
maintenance and power outages, bookmark a geographically convenient 
<A HREF="http://genome.cse.ucsc.edu/mirror.html" TARGET=_blank>mirror</A> site that replicates the 
UCSC genome browser.
<P>
Bear in mind that the Genome Browser cannot outperform the underlying quality of 
the draft genome. Assembly errors and sequence gaps may still occur well into
the sequencing process due to regions that are intrinsically difficult to sequence. 
Artifactual duplications arise as unavoidable compromises during a build, causing 
misleading matches in genome coordinates found by alignment. Post-freeze GenBank 
submissions lead to gradual track obsolescence, which can be addressed by supplemental 
<A HREF="http://www.ncbi.nlm.nih.gov/blast/index.nojs.cgi" TARGET=_blank>BLAST</A> searches. The 
Genome Browser does the best it can to operate under this environment.
<BR>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<A NAME="FineTuning"></A><BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Fine-tuning the
		Genome Browser display</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
<p>
The Genome Browser page displays a genome location specified
through a Gateway position search, a BLAT search, or an uploaded custom annotation track.
There are three main features on this page: the annotations tracks
window, navigation controls, and display controls. The first time you open the
Genome Browser, it will use application default values to configure the
annotation track display. By using the navigation and display configuration
controls, you can customize the annotation tracks display to suit your needs.
For a complete description of the annotation tracks available in all assembly
versions supported by the Genome Browser, see the <A HREF="#IndivTracks">
Annotation Track Descriptions</A> section.
<P>
The Genome Browser retains user preferences from session to session within the
same web browser, although it never monitors or records user activities or submitted 
data. To restore the
default settings, click the <I>reset all</I> button.
<P>
<B>Display conventions</B><BR>
The annotation tracks displayed in the Genome Browser use a common set of
display conventions:<BR>
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Annotation track
	descriptions:</B> 
	Each annotation track has an associated description page that contains a
	discussion of the track, the methods used to create the annotation, the data 
	sources and credits for the track, and (in some cases) filters to
	fine-tune the information displayed in the track. To display the
	description page, click on the gray mini-button to the left of a displayed
	track or on the label for the track in the Track Controls section.</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Annotation track
	details pages:</B> 
	When an annotation track is displayed in full mode, each line item
	within the track has an associated details page that can be displayed by 
	clicking on the item's label. The
	information contained in the details page varies by annotation track, but
	may include basic position information about the item, related links to outside sites 
	and databases, links to genomic alignments, or links to corresponding
	mRNA, genomic, and protein sequences.
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Gene prediction tracks:</B> 
	Coding exons are represented by blocks connected by horizontal lines
	representing introns. The 5' and 3' untranslated regions (UTRs) are displayed 
	as thinner blocks on the leading and trailing ends of the aligning
	regions. In full display mode, arrowheads on
	the connecting intron lines indicate the direction of transcription.</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>mRNA 
	tracks:</B> 
	Aligning regions (usually exons) are shown as black 
	blocks connected by lines representing gaps (usually spliced-out introns). 
	In full display mode, arrowheads on the lines indicate the direction of 
	transcription.  Only 300 track entries can be displayed in the
	annotation track window at a time in full display mode; the track 
	automatically defaults to dense display mode if the number exceeds this limit.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>EST and BLAT alignment 
	tracks:</B> 
	Aligning regions (usually exons) are shown as black blocks. In dense display 
	mode, the degree of darkness corresponds to the number of features aligning to 
	the region. In full display mode, the aligning regions are connected by lines 
	representing gaps (usually spliced-out introns), with arrowheads indicating 
	the direction of transcription. Display limitations similar to the mRNA
	tracks apply to this track.</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Cross-species synteny and
	orthology tracks:</B> 
	Sequences from an organism that match regions of the displayed genome 
	are represented by blocks of various colors. The block color indicates the 
	chromosomal location of the sequence in the organism's genome, as shown by the
	Chromosome Color Key below the annotation track window.</TD></TR>

</TABLE> 
<P>
<B>Changing the width of the annotation track window</B><BR>
By default, the width of the annotation track window is set to 610 pixels. To modify the width to 
best suit the display capabilities of your monitor, enter a new value in the 
<I>image width</I> text box, then click the <I>Jump</I> button. For example, setting the 
display to 1100 pixels on a 19" monitor will increase the visible portion of the genome
and reduce the need for redraws. The maximum supported width is 5000 pixels,
although it is recommended that you don't exceed your monitor width initially. 
<P>
<B>Changing the display mode of an individual annotation track</B><BR>
Each annotation track within the window has three display modes: 
<P>
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Hide:</B> the
	track is not displayed at all. To hide all the annotation tracks,
	click the <I>hide all</I> button. This mode is useful for restricting
	the display to only those tracks in which you are interested. For
	example, someone who is not interested in SNPs or mouse synteny may want 
	to hide these tracks to reduce track clutter and improve speed. There
	are a few annotation tracks that pertain only to one specific
	chromosome, e.g. Sanger22, Rosetta. In these cases, the track and its 
	associated controller will be hidden automatically when the track window is 
	not open to the relevant chromosome.</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Dense:</B> the 
	track is displayed with all features collapsed into a single 
	line. This mode is useful for reducing the amount of space used by a
	track when you don't need individual line item details or when you just
	want to get an overall view of an annotation. For example, by opening an
	entire chromosome and setting the RefSeq Genes track to dense, you can get 
	a feel for the known gene density of the chromosome without displaying
	excessive detail. </TD></TR></TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Full:</B> the
	track is displayed with each annotation feature on a separate line. It is
	recommended that you use this option sparingly, due to the large number
	of individual track items that may potentially align at the selected position. For
	example, hundreds of ESTs might align with a specified gene. When
	the number of lines within a requested track location exceeds 300, the
	track automatically defaults to dense mode. In this situation,
	you can restore the track display to full mode by narrowing the chromosomal 
	range displayed or by using a track filter to reduce the number of items 
	displayed. To toggle between full and dense display modes, click on the
	track's center label.</TD></TR>
</TABLE> 
<P>
To change the display mode for a track, find the track's controller in the
Track Controls section at the bottom of the Genome Browser page, select the desired
mode from the control's display menu, and then click the <I>refresh</I> button. Alternatively, you can toggle between dense and full modes
for a displayed track by clicking on the optional center label for the track. 
<P>
<B>Changing the display of a track by using filters</B><BR>
Some tracks have additional
filter capabilities, e.g. EST tracks, mRNA tracks, NC160, etc. These filters
let the user modify the color or restrict the data displayed within an annotation track.
Filters are useful for focusing attention on items relevant to the current task
in tracks that contain large amounts of data. For example, to highlight ESTs
expressed in the liver, set the EST track filter to display items in a
different color when the associated tissue keyword is "liver".
<P>
To access the filter for a specific annotation track, click on the label for the track's
control under the Track Controls section or on the gray mini-button to the left of the
displayed track. The Genome Browser will display a description page for the
track that contains a filter at the top of the page and a section entitled
<I>Using the Filter</I> that contains instructions for using the filter. Filter
settings are persistent from session to session on the same web browser. To
turn off filtering for a track, delete any text in the filter text boxes and
return the buttons to their original state on the track's description page, or simply 
click the <I>reset all</I> button on the Genome Browser page.
<P>
<B>Zooming and scrolling the tracks display</B><BR>
At times you may want to adjust the amount of flanking region displayed in the
annotation tracks window or adjust the scale of the display. At a scale of 1 pixel per 
base pair, the window accurately displays the width of exons and introns, and 
indicates the direction of transcription (using arrowheads) for multi-exon features. At a 
grosser scale, certain features - such as thin exons - may disappear. Also, some
exons may falsely appear to fall within RepeatMasker features at some scales. 
<P>
Click the <I>zoom in</I> and <I>zoom out</I> buttons at the top of the 
Genome Browser page to zoom in or out on the center of the annotation tracks window by 1.5, 3 or
10-fold. Alternatively, you can zoom in 3-fold on the display by clicking
anywhere on the
Base Position track. In this case, the zoom is centered on the coordinate of the
mouse click. 
<P>
To scroll the annotation tracks sideways to the left or right by 10%, 50%, or 95% of the displayed
size (as given in base pairs), click the corresponding <I>move</I> arrow. It is also 
possible to scroll the left or right side of the tracks while keeping the
position of the opposite side fixed. To do this, click the appropriate <I>move 
start</I> or <I>move end</I> arrow, located under the annotation tracks window.
For example, to keep the left-hand display coordinate fixed but increase the
right-hand coordinate, you would click the right-hand <I>move end</I> arrow. To increase
or decrease the scroll interval, edit the number in the <I>move start</I> or
<I>move end</I> text box. 
<P>
<B>Changing the displayed track position</B><BR>
To display a completely different position in the genome, enter the new position query in
the <I>position</I> text box, then click the <I>Jump</I> button. For more information on
valid entries for this text box, refer to the <A HREF="#GetStarted">Getting
Started</A> section.
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<A NAME="BLATAlign"></A><BR>	

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Using BLAT Alignments </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
BLAT (BLAST-Like Alignment Tool) is a very fast sequence alignment tool similar to 
BLAST. 
For more information on BLAT's internal scoring schemes and its overall n-mer 
alignment seed strategy, refer to the paper <A
HREF="http://www.genome.org/cgi/content/abstract/12/4/656" TARGET=_blank>BLAT -
The BLAST-Like Alignment Tool</A>, W. James Kent, Genome Research, 12:4 656-664. 
<P>
On DNA queries, BLAT is designed to quickly find sequences with 95% or greater 
similarity of length 40 bases or more. It may miss genomic sequences that
are more divergent or shorter than these minimums, although it will find perfect 
sequence matches of 33 bases and sometimes as few as 22 bases. The tool is
capable of aligning sequences that contain large introns. 
On protein queries, BLAT rapidly locates genomic sequences with 80% or greater 
similarity of length 20 amino acids or more. In general, gene family members that 
arose within the last 350 million years can generally be detected. More
divergent sequences can be aligned to the human genome by using NCBI's BLAST and
psi-BLAST, then by using BLAT to align the resulting match onto the UCSC genome 
assembly. In practice DNA BLAT works well on primates, and protein BLAT works
well on land vertebrates. 
<P>
Some common uses of BLAT include: 
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	finding the genomic coordinates of mRNA or protein within a given assembly
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	determining the exon structure of a gene
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	displaying a coding region within a full-length gene
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	isolating an EST of special interest as its own track
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	searching for gene family members
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	finding human homologs of a query from another species. 
</TABLE>	
	
<P><B>Making a BLAT query</B><BR>
To locate a nucleotide or protein within a genome using BLAT: 
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top"><B>1.</B></TD><TD>
	Open the <A HREF="http://genome.cse.ucsc.edu/cgi-bin/hgBlat?command=start" 
	TARGET=_blank>BLAT Search Genome page</A> by clicking the BLAT link on the top 
	blue menu bar of any of the Genome Browser pages.</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top"><B>2.</B></TD><TD>
	Select the assembly freeze date, query type, output sort order, and output
	type. To order the search results based on the closeness of the sequence
	match, choose one of the score options in the <I>Sort output</I> menu.
	The score is determined by the number of matches vs. mismatches in the
	final alignment of the query to the genome.  
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top"><B>3.</B></TD><TD>
	If the sequence to be uploaded is in an unformatted plain text file, enter 
	the file name in the <I>Upload sequence</I> text box, then click the
	<I>Submit File</I> 
	button.  Otherwise, paste the sequence or fasta-formatted list into the large 
	edit box, and then click the <I>Submit</I> button. Input sequence can be
	obtained from the Genome Browser as well as from a custom annotation
	track.</TD></TR> 
</TABLE> 
<P>
Header lines may be included in the input text if they are preceded by > and
contain unique names. 
Multiple sequences may be submitted at the same time if they are of the
same type and are preceded by unique header lines. Numbers, spaces, and extraneous 
characters are ignored. 
<P>
DNA input sequences are limited to a maximum length of 20,000 bases. Protein or
translated input sequences must not exceed 5000 letters. If multiple sequences are 
submitted at the same time, the maximum combined length of DNA input is 50,000
bases (with a 20,000 base limit per individual sequence). For protein or
translated input, the maximum combined input length is 12,500 letters (with a 5000 
letter limit per individual sequence). 
<P><B>BLAT query search results</B><BR>
If a query returns successfully, BLAT will display a flat database file that 
summarizes the alignments found. A BLAT query often generates multiple 
hits. This can happen when the genome contains 
multiple copies of a sequence, paralogs, pseudogenes, statistical coincidences, 
artifactual assembly duplications, or when the query itself contains repeats or common 
retrotransposons. When too many hits occur, try resubmitting the query sequence
after filtering in slow mode with RepeatMasker. 
<P>
Items in the search results list are ordered by the criteria specified in the 
<I>Sort output</I> menu. Each line item provides links to view the details of the sequence 
alignment or to open the corresponding view in the Genome Browser. The
<I>details</I> link gives the letter-by-letter alignment of the sequence to the 
genome. It is recommended 
that you first examine the details of the alignment for match quality before viewing
the sequence in the Genome Browser. 
<P>
When several nearby BLAT matches occur on a single chromosome, a 
simple trick can be used to quickly adjust the Genome Browser track window to display 
all of them: open the Genome Browser with the match that has the lowest chromosome 
start coordinate, paste in the highest chromosome end coordinate from the list of 
matches, then click the <I>Jump</I> button. 
<P><B>Creating a custom annotation track from BLAT output</B><BR>
To make a custom track directly from BLAT, select the 
<A HREF="http://genome.cse.ucsc.edu/goldenPath/help/customTrack.html#PSL">PSL format</A>
output option. The resulting PSL track can be uploaded into the Genome Browser by 
pasting it into the <I>Add Your Own Tracks</I> section at the bottom of the Browser 
Gateway page.
<P><B>Using BLAT for large jobs or commercial use</B><BR>
The BLAT servers are <A HREF="http://genome.cse.ucsc.edu/mirror.html"
TARGET=_blank>mirrored</A> at several institutions to ensure continuous availability. 
For large batch jobs and internal parameter changes, it is best to install
command line BLAT on your own Linux server. Sources and executables are free for
academic, personal, and non-profit purposes. Non-exclusive commercial licenses
are also available. Contact <A HREF="http://www.cse.ucsc.edu/~kent"
TARGET=_blank>Jim Kent</A>
for more information. 
<P>
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="TableBrowser"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Getting started on the Table Browser  </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
The <A HREF="http://genome.ucsc.edu/goldenPath/hgText.html" TARGET=_blank>Table
Browser</A> provides a tabular alternative to the graphical display of the Genome
Browser. This tool allows the user to view and download the primary database tables containing 
a genome sequence and its associated annotation tracks. The tables are presented in
simple tab-delimited text format. The Table Browser provides a convenient
alternative to downloading and manipulating the entire genome and its massive data tracks.
(For more information on data downloads, see the <A
HREF="#Download">Downloading Genome Data</A> section.) 
<P>
When a new genome assembly is released, a set of annotations tracks is generated by
aligning features such as mRNAs and ESTs against the genome using BLAT. The alignment
process generates <I>positional</I> stop-start coordinates and other descriptive data, which are 
stored within MySQL relational database tables. Other data is inherently 
<I>non-positional</I>, i.e. not tied to genomic coordinates. Non-positional data includes such 
things as names of authors who have submitted data into GenBank, types of tissue from
which sequences originate, and links to RefSeq. 
<P>
Because the same positional and non-positional tables drive both the graphical annotation tracks 
displayed in the Genome Browser and the text-based data displayed in the Table Browser, the data 
is always in synch.
<P><B>Using the table browser</B><BR>
The Table Browser can be accessed in two different ways: 
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Clicking the Tables link on the UCSC Genome Bioinformatics home page opens up 
	the Table Browser Gateway page. On this page, select a date from the
	<I>Freeze</I> menu and
	enter the position of the genome that you would like to view, then click the
	<I>Submit</I> button. The Gateway page contains several examples of valid position
	queries.  <TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Clicking the Tables link in the blue menu bar on the Genome Browser page will open
	up the Table Browser to the same assembly and position displayed in the annotation
	tracks window. </TD></TR>
</TABLE> 
<P>
To display a table, follow these steps:
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top"><B>1.</B></TD><TD>
	If the default position is acceptable, skip to the next step. Otherwise, enter the
	position in the genome that you would like to display, then click the
	<I>Look up</I> button. 
	The same query criteria used for position searches on the Gateway pages apply to this 
	position search as well. As with position searches in other parts of the Genome
	Browser, a query may return a list of several options, requiring you to select the
	appropriate one.
	<TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top"><B>2.</B></TD><TD>
	Choose one of the options from either the <I>Positional tables</I> or
	<I>Non-positional tables</I> lists. Positional tables are listed by their symbolic names. For a
	complete description of the tables, refer to the documentation on the 
	<A HREF="http://genome.cse.ucsc.edu/goldenPath/gbdDescriptions.html"
	TARGET=_blank>Genome Browser database</A>.
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top"><B>3.</B></TD><TD>
	Click one of the buttons on the <I>Choose an action</I> line to select the type of output
	and display it. Click the <I>Choose fields</I> button if you want to hand-select
	the table fields that are displayed in the text output. The <I>Get all fields</I> button
	defaults to a display containing all the fields in the table. If you have selected
	a positional table, click the <I>Get DNA</I> button to display a list of DNA sequences in
	fasta format for all the table features in the specified position range. If a
	Table Browser query fails to return data, it may be necessary to broaden the range
	of the position query.
	</TD></TR>
</TABLE> 
<P>
If you are not familiar with a particular table, it is recommended that you select the <I>Choose 
fields</I> button to configure the output of the table display. Some tables offer many columns 
of data, of which only a few may be relevant. Reducing the number of fields to the minimum 
necessary simplifies viewing and further processing of the data. 
<P>
The tab-delimited data returned by a Table Browser search can be post-processed in a
spreadsheet or database to produce output in the format required for a <A
HREF="customTrack.html" TARGET=_blank>custom annotation track</A>. 
<P>
In the future, the Table Browser may offer the user the ability to filter records, which will 
reduce the size of large tables by allowing both row and column selection. Limited boolean logic 
processing, such as joins of two tracks, may also be supported at a later time.
<P>
As with the graphical Genome Browser, note that chromosomal coordinates within tables often 
change from one assembly to the next, as gap are filled and the assembly procedure is refined. 
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="TrackFormatDNA"></A><BR>	

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; DNA text formatting</b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
The Genome Browser provides a feature to configure the formatting
and coloring of the text used to depict the DNA sequence underlying the features in the displayed 
annotation tracks window. Formatting can range from simply displaying exons in upper case to 
elaborately marking up a sequence according to multiple track data. The DNA sequence covered by 
various tracks can be highlighted by case, underlining, bold or italic fonts, and color.
<P>
The DNA display configuration feature can be useful to highlight features within a
genomic sequence, point out overlaps between two types of features (for example, known
genes vs. gene predictions), or mask out unwanted features. 
<P><B>Using the DNA text formatting feature</B><BR>
To access the feature, click on the DNA link on the top blue menu
bar on the Genome Browser page. The Get DNA in Window page that appears lists several options for
adjusting the case of all or part of the DNA sequence. To choose one of these formats,
click the corresponding option button, then click the <I>Submit</I> button. To access a table 
of extended formatting options, click the <I>extended case/color options</I> button, then
click <I>Submit</I>.
<P>
The Extended DNA Case/Color page presents a table with many more format options. The page
provides instructions for using the formatting table, as well as examples of its use. The
list of tracks in the Track Name column is automatically generated from the list of tracks
available on the current genome.
<P><B>Tips for Use</B><BR>
A few caveats mentioned on the Extended DNA Case/Color page bear repeating. Keep the formatting simple at first: it is easy to 
make a display that is pretty to look at but is also completely cryptic. Also, be careful when 
requesting complex formatting for a large chromosomal region: when all the HTML tags have been 
added to the output page, the file size may exceed the size limits that your internet browser, 
clipboard, and other software can safely display. The maximum size of genome that can be 
formatted by the tool is approximately 10 Mbp.
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="Convert"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Converting data between assemblies</b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>Coordinates of features frequently
change from one assembly to the next as gaps are closed, strand orientations are
corrected, and duplications are reduced. The Genome Browser Convert tool is useful for 
locating the position of a feature of interest in a newer release of a genome.
During the conversion process, portions of the genome in the coordinate range of
the original assembly are aligned to the new assembly while preserving their order and 
orientation. In general, it is easier to achieve successful conversions with shorter 
sequences.
<P>
<P><B>Using the convert tool</B><BR>
To access the conversion tool, click the Convert link in the top menu bar on the
Genome Browser page. On the page that displays, select the assembly
version in which the feature is located in the <I>Original Draft</I> list, then
pick the assembly version you'd like to convert to in the <I>New Draft</I> list.
Modify the position coordinates if necessary, then click the <I>submit</I>
button. Note that archived assembly versions aren't accessible from the
conversion tool. Currently the tool supports only the human genome.
<P>
If the match is successful, the Genome Browser will announce success and display
the coordinate ranges for both the original and new assemblies. Clicking on the
<I>browser</I> link to the right of the coordinate range will start up the
browser at the given position on the assembly. The Alignment Details section
shows which sequences of the original draft were aligned by BLAT to determine
the new set of coordinates. Note that the conversion is a best guess: it is
recommended that you check with local landmarks and use 
common sense when evaluating the results.
<P>
If the conversion is unsuccessful, the Genome Browser returns a failure message
and a possible explanation for the failure.

		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="Download"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Downloading genome data</b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>Most of the underlying tables containing the genomic sequence and annotation data 
displayed in the Genome Browser can be downloaded. All of the tables are
freely usable for any purpose except as indicated in the README.txt file
in the download directories. This data was contributed by many researchers, as 
listed on the Genome Browser 
<A HREF="../credits.html" TARGET=_blank>Credits</A> page. Please
acknowledge the contributor(s) of the data you use. 
<P><B>Downloading the data</B><BR>
Genome data can be downloaded in two different ways:
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Via the Downloads link:</B> Click the
	<A HREF="http://hgwdev-learithe.cse.ucsc.edu/downloads.html"
	TARGET=_blank>Downloads</A> link on the left side bar on the UCSC 
	Genome Bioinformatics home page to display a list of all database directories 
	available for 
	download. If the data you wish to download pre-dates the assembly 
	versions listed, look in the archives accessible from the
	<A HREF="http://genome-archive.cse.ucsc.edu/"
	TARGET=_blank>Archive</A> link on the home page.
	</TD></TR>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Via
	ftp:</B>The UCSC Genome Bioinformatics ftp site contains download
	directories for all genome versions currently accessible in the
	Genome Browser. The ftp command
	<tt>ftp://genome.cse.ucsc.edu/goldenPath/</tt> will take you to a
	directory that contains the genome download directories. 
	</TD></TR>

</TABLE> 
<P><B>Types of data available</B><BR>
There are three download directories associated with each version of a genome
assembly: the full data set <I>(bigZips)</I>, the full data set by 
chromosome <I>(chromosome)</I>, and the annotation database tables
<I>(database)</I>. 
<P>
<I>BigZips</I> contains the entire draft of the 
genome in chromosome and/or contig form. Depending on the genome, this 
directory may contain some or all of the following files:<BR> 
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>chromFa.zip:</B>
	the working draft sequence chromosomes, in one file per chromosome. 
	Repeats are
	shown in lower case and non-repeating sequence is shown in upper case.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>chromFaMasked.zip:</B>
	the working draft sequence chromosomes, in one file per chromosome. 
	Repeats are masked by capital N's and non-repeating sequence is shown in upper 
	case.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>contigFa.zip:</B>
	the working draft sequence contigs, in one file per contig. All 
	contigs are in forward orientation relative to the chromosome. In some 
	cases of
	the human genome, this means that some contigs will be reversed 
	relative to their orientation in the NCBI data. Repeats are
	shown in lower case non-repeating sequence is shown in upper case.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>contigAgp.zip:</B>
	description of how the working draft was generated from fragments at a 
	contig layout level. 
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>chromAgp.zip:</B>
    	similar to contigAGP.zip, but unpacks to one file per chromosome.  
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>contigOut.zip:</B>
	RepeatMasker .out file for contigs, generated with RepeatMasker at the 
	<I>-s</I> sensitive setting.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>chromOut.zip:</B>
	RepeatMasker .out file for chromosomes, generated with RepeatMasker at 
	the <I>-s</I> sensitive setting.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>database.zip:</B>
	the Genome Browser database as tab-delimited files and associated 
	MySQL table-creation tiles.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>liftAll.zip:</B>
	the offsets of contigs within chromosomes.
	</TD></TR> 
</TABLE> 
<P><I>Chromosomes</I> contains 
the assembled sequence for the genome in separate files for each
chromosome in a zipped fasta format.
<P>
<I>Database</I> contains all of the 
positional and non-positional tables in the genome annotation database. Each 
table is represented by 2 files:
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>.sql file:</B>
	the MySQL commands used to create the table.
	</TD></TR>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>.txt.gz file:</B>
	the MySQL database table data in tab-delimited format and 
	compressed with
	gzip.
	</TD></TR>
</TABLE> 
<P>
The <A HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html"
TARGET=_blank>Description of the annotation database</A> link on the Downloads page 
points to a document containing a full
description of all the tables in the annotation database.

		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="CustomTracks"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Creating custom annotation tracks </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
The Genome Browser provides dozens of aligned annotation tracks that have been
computed at UCSC or have been provided by outside collaborators. In addition to these 
standard tracks, it is also possible for users to upload their own annotations for
temporary display in the browser. These custom annotation tracks are viewable
only on the machine from which they were uploaded and are only
kept for 8 hours after the last time they were accessed. Users can optionally
make custom annotations viewable by others as well. 
<P>
Custom tracks are an important research feature of the Genome Browser. Because
space is limited in the Genome Browser track window, many excellent genome-wide tracks 
cannot be included in the set of tracks packaged with the Genome Browser. Other
tracks of interest may be excluded from distribution because the annotation
track data is too specific to be of general interest or can't be shared until
journal publication. 
<P>
Custom annotation tracks are similar to standard tracks, but never become part of the 
MySQL genome database. Each track has its own controller and persists even when not 
displayed in the Genome Browser window, e.g. if the position changes to a range
that no longer includes the track. Typically custom annotation tracks are aligned 
under corresponding genomic sequence, but they can also be completely unrelated to 
the data. For example, a track could be displayed under a long sequence
consisting of millions of N's.
<P>
This section presents a summary of the steps involved in constructing and
displaying a custom annotation track. For a complete discussion of custom
annotation tracks, file formats, custom URLs, and a troubleshooting guide, refer
to the document <A HREF="customTrack.html" TARGET=_blank>Displaying Your Own Annotations in the Genome Browser</A>.
<P>
<B>Creating a custom annotation track</B><BR>
Genome Browser annotation tracks are based on files in line-oriented format.
Each line in the file defines a display characteristic for the annotation track
or defines a data item within the track.
Annotation files contain 3 types of lines: browser lines, track lines, and data
lines. Empty lines and lines starting with # in the
annotation file are ignored.
<P>
The easiest way to create a correctly formatted annotation track is by
collecting PSL output from BLAT. Advanced users can make custom tracks from the Table 
Browser and track-formatted DNA. To create a custom annotation track from scratch, 
it's best to begin with the examples and experiment with new lines and altered values 
in a spreadsheet or text editor. 
<P>
To construct an annotation file and display it in the Genome Browser, follow
these steps:

<P><B>Step 1. Format the data set</B><BR>
Formulate your data set as a tab-separated file using one of the formats supported by the Genome Browser. Annotation data can be in 
standard <A HREF="customTrack.html#GFF">GFF</A> format or in a format designed specifically for the Human Genome Project, including 
<A HREF="customTrack.html#GTF">GTF</A>, <A
HREF="customTrack.html#PSL">PSL</A>, or <A HREF="customTrack.html#BED">BED</A>. GFF and GTF files <I>must</I> be tab-delimited rather than
space-delimited to display correctly. You may include more than one data
set in your annotation file. However, all of the data lines for a given annotation track must be in the same format.

<P><B>Step 2. Define the Genome Browser display characteristics</B><BR>
Add one or more optional <A HREF="customTrack.html#BROWSER">browser lines</A> to the beginning of your formatted data file to configure the overall 
display of the Genome Browser when it initially displays your annotation data. Browser lines allow you to configure such things 
as the genome position that the browser will initially open to, the width of the display, and the configuration of the other
annotation tracks that are shown (or hidden) in the initial display. NOTE: If
the browser position is not explicitly set in the annotation file, the initial display will default
to the position setting most recently used by the user, which may not be an
appropriate position for viewing the annotation track.

<P><B>Step 3. Define the annotation track display characteristics</B><BR>
Following the browser lines - and immediately preceding the formatted data - add
a <A HREF="customTrack.html#TRACK">track line</A> to 
define the display attributes for your annotation data set. Track lines allow you to define annotation track characteristics 
such as the name, description, colors, initial display mode, use score,
etc. If you have included more than one data set in your annotation
file, insert a track line at the beginning of each new set of data.

<P><B><I>Example:</I></B><BR>
Here is an example of an annotation file that defines 2 separate
annotation tracks in BED format. The first track displays blue one-base
tick marks every 10000 bases at the beginning of chr 22. 
The second track displays red 100-base features alternating with 
blank space in the same region of chr 22.
<PRE><TT>browser position chr22:1-20001
track name=spacer description="My blue ticks" color=0,0,255
chr22   0        1
chr22   10000 10001
chr22   20000 20001
track name=even description="My red 100 bases tracks" color=255,0,0
chr22   0       100 first
chr22   200     300 second
chr22   400     500 third</TT></PRE>

<P><B><I>Example:</I></B><BR>
This example shows an annotation file containing one data set in BED
format. The track displays paired features with a thick end and thin
end, and hatch marks indicating the direction of transcription. The track
labels display in green (0,128,0), and the gray level of the each feature
reflects the score value of that line. NOTE: The track line in this example
has been
split over 2 lines for documentation purposes. If you paste this example into
the browser, you must remove the line break to display the track successfully.
<PRE><TT>browser position chr22:1000-10000
browser hide all
track name=pairedReads description="Clone Paired Reads" visibility=2
color=0,128,0 useScore=1
chr22 1000 5000 cloneA 960 + 1000 5000 0 2 567,488, 0,4512
chr22 2000 6000 cloneB 200 - 2000 6000 0 2 433,399, 0,5601
</TT></PRE>

<P><B>Step 4. View your annotation track in the Genome Browser</B><BR>
To view your annotation data in the Genome Browser, open up the UCSC Genome
Bioinformatics home page <A HREF="http://genome.ucsc.edu/"
TARGET=_blank>(http://genome.ucsc.edu/)</a>, select the type of genome on which your annotation data is based, and then click on the Genome 
Browser link in the top menu bar. On the Genome Browser Gateway page that
displays, select the assembly on which your
annotation data is based, then scroll down to the <I>Add Your Own Tracks</I> section at the bottom of the page. Upload your annotation
file by entering the name of your file in the <I>Annotation File</I> box or by pasting the contents of your file into the large edit
box. Scroll back to the top of the page and click the <I>Submit</I> button to
display the Genome Browser track window with your annotation. If you
encounter difficulties in displaying your annotation, read the section <A
HREF="customTrack.html#TROUBLE">Troubleshooting Annotation Display Problems</A>.</P> 
<P>
To upload a custom annotation track from another machine or web site, paste the
URL of the track into the large edit box. Custom tracks can be displayed in
conjunction with ordinary BLAT tracks.</P>

<P><B>Step 5. (Optional) Add details pages for individual track features</B><BR>
After you've constructed your track and have successfully displayed it in the
Genome Browser, you may wish to customize the details pages for individual track
features.  The Genome Browser automatically creates a default details page for each
feature in the track containing the feature's name, position information, and a
link to the corresponding DNA sequence. To view the
details page for a feature in your custom annotation track (in full display
mode), click on the item's label in the annotation track window. 
<P>
You can add a link from a details page to an external web page containing
additional information about the feature by using the track line <I>url</I> attribute. 
In the annotation file, set the <I>url</I> attribute in the track line to point
to a publicly available page on a web server. The <I>url</I> attribute 
substitutes each occurrence of '$$' in the URL string with the name defined by the 
<I>name</I> attribute. You can take advantage of this feature to provide 
individualized information for each feature in your track by creating HTML anchors 
that correspond to the feature names in your web page.

<P><B><I>Example:</I></B><BR>
Here is an example of a file in which the <I>url</I> attribute has been set to
point to the file <A HREF="http://genome-test.cse.ucsc.edu/clones.html"
TARGET=_blank>http://genome-test.cse.ucsc.edu/clones.html</A>. The '#$$'
appended to the end of the file name in the example points to the HTML NAME tag within the file that 
matches the name of the feature (cloneA, cloneB, etc.). 
NOTE: The track line in this example has been split over 2 lines for 
documentation purposes. If you paste this example into
the browser, you must remove the line break to display the track successfully.
<BR><BR>
<PRE><TT>browser position chr22:1000-10000
browser hide all
track name=clones description="Clones" visibility=2
color=0,128,0 useScore=1 url="http://genome-test.cse.ucsc.edu/clones.html#$$"
chr22 1000 5000 cloneA 960 
chr22 2000 6000 cloneB 200 
chr22 5000 9000 cloneC 700 
chr22 6000 10000 cloneD 600
chr22 11000 15000 cloneE 300
chr22 13000 17000 cloneF 100 
</TT></PRE>
<P>
<B>Sharing your annotation track with others</B><BR>
The previous steps showed you how to upload annotation data for your own use on your own machine. However, many users would like to 
share their annotation data with members of their research group on different machines 
or with colleagues at other sites.  To make your Genome Browser annotation track 
viewable by others, follow the steps below. (Note that 
some of the URL examples in this section have been broken up into 2
lines for documentation display purposes).<BR><BR>

<P><B>Step 1.</B>
Put your formatted annotation file on your web site. Be sure that the file permissions allow it to be read by others.</P>

<P><B>Step 2.</B>
Construct a URL that will link this annotation file to the Genome Browser. The URL must contain 3 pieces of information specific
to your annotation data: 
<UL>
<LI>The genome freeze on which your annotation data is based. This
information is of the form <tt>db=database_name</tt>, where
<I>database_name</I> is to the UCSC code for the genome freeze. For a list
of these codes, see the Genome Browser <A
HREF="http://genome.ucsc.edu/FAQ.html#124" TARGET=_blank>FAQ</A>.
Examples of this include: db=hg11 (Human April 2002 freeze), db=mm2 (Mouse Feb 2002 freeze).
<LI>The genome position that the Genome Browser should initially open to. This information is of the form
<tt>position=chr_position</tt>, where <I>chr_position</I> is a chromosome number, with or without a set of 
coordinates. Examples of this include: position=chr22, position=chr22:15916196-31832390.
<LI>The URL of the annotation file on your web site. This information is
of the form <tt>hgt.customText=URL</tt>, where
<I>URL</I> points to the annotation file on your website. An example of an annotation file URL is <font color="000099"><A HREF="http://genome-test.cse.ucsc.edu/test.bed"
TARGET=_blank>http://genome-test.cse.ucsc.edu/test.bed</A></font>.  
</UL>

<P>Combine the above pieces of information into a URL of the following
format (the information specific to your annotation file is highlighted):
<BR><BR>
<tt>http://genome.ucsc.edu/cgi-bin/hgTracks?db=<font color="000099"><I>database_name</I></font>&amp;<BR>position=<font
color="000099"><I>chr_position</I></font>&amp;hgt.customText=<font color="000099"><I>URL</I></font></tt>.

<P><B><I>Example:</I></B><BR>
The following URL will open up the Genome Browser window to display chr 22 of the April 2002 freeze of the 
human genome (hg11), and will display the annotation track pointed to by the URL http://genome.ucsc.edu/test.bed:<BR><BR>

<font color="000099"><A HREF="
http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg11&amp;position=chr22&amp;hgt.customText=http://genome-test.cse.ucsc.edu/test.bed"><tt>http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg11&amp;position=chr22&amp;<br>hgt.customText=http://genome-test.cse.ucsc.edu/test.bed</tt></A></font><BR><BR>

<P>If you'd like to share your annotation track with a broader audience, send the URL for your track - along with a description of 
the format, methods, and data used - to the UCSC Genome mailing list 
<A HREF="mailto:genome@cse.ucsc.edu">genome@cse.ucsc.edu</A>.</P>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	

	<A NAME="IndivTracks"></A><BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp; Annotation track descriptions</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
This section contains an alphabetical listing of the annotation tracks available
on various versions of the Genome Browser. For a more information on configuring
and using the tracks displayed in the Genome Browser track window, see the section 
<A HREF="#FineTuning">Fine-tuning the Genome Browser display</A>.<BR>

<P><B>Tips for viewing annotation track data</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	To display a description page with more information about the track,
	click on the grey mini-button to the left of a track.</TD></TR>   
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	To display a details page with additional information about a specific line 
	item within a track in full display mode, click on the item's label. </TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	A track does not appear in the browser if its display mode is set to 
	<I>hide</I>. To restrict the browser's display to only those tracks in
	which you're interested, set the display mode of the unwanted tracks to
	<I>hide</I>.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	A track set to <I>full</I> display mode will display in <I>dense</I> mode if 
	the total number of lines in the track exceeds 300.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	To quickly toggle between <I>full</I> and <I>dense</I> display modes, click on 
	the track's center label.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Only the most recent assemblies are fully active. Older assemblies are <A HREF="http://genome-archive.cse.ucsc.edu/">archived</A>.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Not all tracks appear in all assemblies. Only basic tracks appear
	initially in a new assembly.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Track data can be viewed equivalently in columns as text tables using
	the <A HREF="#TableBrowser">Table Browser</A>. </TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<A HREF="http://genome.cse.ucsc.edu/goldenPath/credits.html">Credit</A>
	goes to many individuals and institutions for generously contributing
	the tracks. For specific information about the contributors to a given track,
	look at the track's description page.</TD></TR>  
</TABLE>

<P><B>Acembly Genes</B><BR>
This track shows gene models reconstructed solely from mRNA and EST evidence 
by Danielle and Jean Thierry-Mieg and Vahan Simonyan using the Acembly program. 
Acembly attempts to find the best alignment of each mRNA against the genome, 
and considers alternative splice models. If more than one gene model is 
produced that has statistical significance, all of these models are
displayed.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track uses the display conventions for gene prediction
	tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show 
	position information, links to translated protein, predicted mRNA, and genomic
	sequence, and a link to an extensive text and graphical annotation of 
	supporting data at NCBI's <A
	HREF="http://www.ncbi.nlm.nih.gov/IEB/Research/Acembly/index.html">AceView</A>.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>acembly</I> position table in the Table Browser. To view the table
	definition, click 
	<A HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE>

<P><B>Assembly</B><BR>
This track shows the draft assembly of the human genome. This assembly merges
contigs from overlapping draft and finished clones into longer sequence 
contigs. The sequence contigs are ordered and oriented when possible by mRNA, 
EST, paired plasmid reads (from the SNP Consortium) and BAC end sequence pairs.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	In dense display mode, this track depicts in alternating gold and brown the 
	path through the draft and finished clones (aka the golden path) used to 
	create the assembled sequence. Individual boxes show the assembly extractable 
	from a single clone fragment. Where gaps exist in the path, spaces are shown 
	between the gold and brown blocks. If the relative order and orientation of 
	the contigs between the two blocks is known, a line is drawn to bridge the 
	blocks. The display must be sufficiently zoomed in to view the gap and bridge 
	features.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode will show information about the
	clone fragment and bases, position information, and the DNA sequence
	corresponding to the fragment.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>chrN_gold</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GpInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>BAC End Pairs</B><BR>
This track shows the genome sequence of bacterial artificial chromosomes 
(BACs) in which a small amount of the both ends have been sequenced to place 
the full insert on the assembly. A valid pair of BAC end sequences must be 
separated by at least 50Kb but no more than 600Kb, and must have opposite 
orientations.  Strand assignment follows 
<a href="http://www.ncbi.nlm.nih.gov/genome/clone/">Clone Registry</a> 
storage conventions.  The BAC end sequences are placed on the assembled 
sequence using BLAT.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	The sequenced BAC ends are represented by black boxes connected by a
	line with hatch marks indicating the direction of transcription.
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show hyperlinks to the
	NCBI clone registry and GenBank, position information, and genomic
	alignments. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>bacEndPairs</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#LFS"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Base Position</B><BR>
Shows the chromosomal coordinates as numbered starting from the telomere of 
the short p arm. The coordinates differ between assemblies.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Clicking on a position in this track will zoom in the display by 3x,
	centered around the click. This track has only on and off modes, and no 
	dense mode.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	None.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	None.</TD></TR>
</TABLE></P>

<P><B>Chromosome Band</B><BR>
The light and dark blocks indicate the approximate location of
traditional cytological bands seen on Giemsa-stained chromosomes at an 850
band resolution. The bands were <A
HREF="http://www.ncbi.nlm.nih.gov:80/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11237021&dopt=Abstract">mapped</A>
by fluorescent in situ hybridization (FISH) of large-insert clones
with sequence tags, STS markers, and BAC end sequence locations.  
resolution of 862 bands genome-wide. The uncertainty of 
mapping to band staining is still a megabase or more.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	The light and dark blocks indicate the approximate location of the bands on the
	assembly sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on a band in full display mode will show more details about
	the track, a discussion of mapping methods used, and lab
	credits.</TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>cytoBand</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#ChromosomeBand"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Coverage</B><BR>
Shows the coverage level and positions of all clones in the genome.
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	In dense display mode, this track shows the coverage level of the
	genome. Finished regions are shown in black. Draft regions are
	shown in various shades of gray that correspond to the level of
	coverage:<BR>

	<BLOCKQUOTE>
		o White: no coverage (gap) <BR>
		o Light Gray: predraft (less than 4x shotgun) <BR>
		o Medium Gray: draft (at least 4x shotgun) <BR>
		o Dark Gray: covered by multiple draft clones <BR>
		o Black: covered by a finished clone
	</BLOCKQUOTE>

		<TR><TD WIDTH="20" VALIGN="top"></TD><TD>
	In full display mode, coverage for each unfinished clone is represented as a 
	two-tiered gray box.  The fragments inside the clone are shown as numbered 
	black boxes. Their positions correspond to their relative assembly placement, 
	including strand assignment. For some assemblies, clones in the sequencing 
	center tiling path are shown with blue rather than gray backgrounds. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode will show the clone hyperlink to 
	Genbank record and the fasta format sequence (one per fragment).</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>clonePos</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#CloneInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>CpG Islands  </B><BR>
This track shows CpG islands, areas rich in the dinucleotide CG. A CpG island
is a region where C's followed by G's (CpGs) are present at significantly 
higher levels than is typical for the genome as a whole. The CpG islands 
displayed in this browser are all at least 200 bases long, and have a GC 
content of at least 50%.
CpG islands are associated with genes, particularly housekeeping genes, in 
vertebrates. They are particularly common near transcription start sites, and 
may be associated with promoter regions. Normally a CpG is rare in vertebrate 
DNA because the C's in such an arrangement tend to be methylated. This 
methylation helps distinguish the newly synthesized DNA strand from the parent 
strand, which aids in the final stages of DNA proofreading after duplication. 
However, over evolutionary time, methylated C's tend to turn into T's due to 
spontaneous deamination. The result is that CpGs are relatively rare unless 
there is selective pressure to keep them or a region is not methylated for 
some reason, perhaps having to do with the regulation of gene expression. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	CpG islands are represented by green boxes. Darker green items indicate CpG 
	islands that exceed 400 occurrences. 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Clicking on a CpG entry in full display mode will show basic track information, 
	detailed information about the CpG island, and sequence information. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>cpgIsland</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#CpG Islands"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Duplications </B><BR>
This track shows genomic 
<a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11381028&dopt=Abstract">
duplications</a> greater than 1,000 bp if not overly repeat-masked. Duplications are 
more common in unordered and unassigned contigs (i.e., difficult to assign to their 
proper position). 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Duplications are show as red if 99%+ identity (likely unmerged assembly 
	overlap artifact), as yellow for 98-99% identity (origin uncertain), and as 
	gray for 90-98% identity (below draft sequencing error, often genuine 
	pericentromeric or subtelomeric duplication). </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	In the left panel of the browser, duplications are named by the chromosome 
	and start of the other member of the duplication pair. Clicking on an 
	individual duplication gives details of both locations and match quality 
	parameters. </TD></TR>
</TABLE></P>

<P><B>Ensembl Genes</B><BR>
This track shows gene predictions from <A
HREF="http://www.ensembl.org">Project Ensembl</A>. For a description of
the methods used in Ensembl gene prediction, refer to
<A HREF="http://nar.oupjournals.org/cgi/content/full/30/1/38" TARGET=_blank>The Ensembl genome database project</A>, Nucleic Acids Research, 2002, 30(1) 38-41.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode brings up basic track
	information, links to the translated protein, predicted mRNA, and
	genomic sequences, plus a link to detailed supporting evidence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>ensGene</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Exofish ecores</B><BR>
The track displays conserved regions, generally protein-coding homologs, 
from a genomics project of long random reads in the 
<A HREF="http://www.genoscope.cns.fr/exofish">pufferfish</A>, Tetraodon 
nigroviridis.  Ecore is an acronym for <B>e</B>volutionary <B>co</B>nserved 
<b>re</b>gion. The data was 
<A HREF="http://www.ncbi.nlm.nih.gov:80/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=10835645&dopt=Abstract">
used</A> to estimate the number of human genes. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Regions of greater homology are displayed in a darker blue. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full mode brings up information on a particular 
	ecore, access the relevant Genoscope page, and the relevant DNA.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>exoFish</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#ExoFish"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Fgenesh++ Genes</B><BR>
This tracks shows Fgenesh++ gene predictions based on <A
HREF="http://www.softberry.com">Softberry Inc.'s</A> gene-finding
software. Fgenesh++ uses both hidden Markov models (HMMs) and protein 
similarity to find genes in a completely automated manner, with less
emphasis on cDNA/EST data. See the paper 
<A HREF="http://www.genome.org/cgi/content/full/10/4/516" TARGET=_BLANK>"Ab 
initio gene finding in Drosophila genomic DNA"</A>, Genome Research 10(4) 516-522 
for more information.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode brings up basic track
	information, links to the translated protein, predicted mRNA, and
	genomic sequences, plus a link to the best matching protein at GenBank.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the human data set, select the 
	<I>softberryGene</I> position table in the Table Browser. 
	To view a text version of the mouse data set, choose the 
	<I>softberryMouse</I> position table in the Table Browser. 
	To view the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Fish Blat </B><BR>
This track displays translated alignments of 728 million bases
of Tetraodon nigrovirisi whole genome shotgun reads vs. the
draft human genome. Areas painted by this track are quite likely to be coding
regions. The alignments were done with BLAT in translated protein mode
requiring 2 nearby 4-mer matches to trigger a detailed alignment. The human 
genome was masked with RepeatMasker and Tandem Repeat Finder before running 
BLAT. The Tetraodon sequence was provided by <A
HREF="http://www.genoscope.cns.fr" TARGET=_blank">Genoscope</A>. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for BLAT tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode brings up links to the DNA
	sequence and the alignment. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>chrN_blatFish</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>FISH Clones</B><BR>
Shows the location of fluorescent in situ hybridization (FISH) mapped clones along 
the draft assembly sequence. 
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	The green blocks depict the clone placements on the assembly sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B>
	Clicking on a specific marker in full display mode will show information on 
	clone placement, clone registry, its STS markers, and BAC end sequences.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>fishClones</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#FishClones"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Gap</B><BR>
This track depicts gaps in the assembly. These gaps - with the
exception of intractable heterochromatic gaps - will be closed
during the finishing process.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Gaps are represented as black boxes. If the relative order and orientation of 
	the contigs on either side of the gap is known from mRNA, ESTs, or
	paired BAC end reads, it is a "bridged" gap and a white line is drawn through 
	the black box representing the gap. The display must be sufficiently zoomed in 
	to view these features: at a coarser scale, the gaps can appear to coalesce in 
	the the graphic. In full display mode, the gap label indicates the type
	of gap (see below) and whether the gap is bridged.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top"></TD><TD>
	Strand misorientation errors can occur adjacent to unbridged gaps.
	Unbridged gaps are assigned somewhat arbitrary sequence sizes (runs of N's in 
	genomic DNA); uncertainty is less with bridged gaps. A gap within contig 
	fragments of a clone is typically smaller than a gap between draft clones. </TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top"></TD><TD>
	There are four principle types of gaps:<BR>
	<BLOCKQUOTE>
		o Fragment: gaps between the contigs of a draft clone. These are represented 
	by 100 N's in the assembly, whether or not there is bridging
	information.<BR>
		o Clone: gaps between clones in the same map contig. These are represented 
	by 50,000 N's in the assembly if the gap is bridged or 100,000 N's if it is not.<BR>
		o Contig - gaps between map contigs. These are represented by 200,000 N's in 
	the assembly.<BR>
		o Heterochromatin - gaps from centromeres (3,000,000 N's) or other large 
	blocks of heterochromatin (size varies).<BR>
	</BLOCKQUOTE></TD></TR>

		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode will show the gap type,
	bridging information and position information about the gap. Unfinished 
	clones may have gaps of uncertain size between fragments.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>chrN_gap</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GapInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>GC Percent</B><BR>
This shows the percentage of bases that are G or
C in a 20,000 base window. High GC content is
associated with gene rich areas. The average GC percent across
the human genome is 41%. Both Celera and public consortium reported 
isochore concept in conflict with human genome sequence data. 
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	Windows with high GC content are drawn more darkly than those with low GC 
	content. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full mode provides position information and the GC 
	percent for the item.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>gcPercent</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GCpercent"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Geneid Gene Predictions</B><BR>
This track shows gene predictions from the geneid program, which is
being developed at the Grup de Recerca en Informatica Biomedica at IMIM
in Barcelona. Geneid uses information from sequence signals involved in
genes specification, as well as coding coding statistics, to define exons
and genes. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode brings up basic track
	information, plus links to the translated protein, predicted mRNA, and
	genomic sequences.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>geneid</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Genscan Genes</B><BR>
This track shows gene predictions from <A HREF="http://genes.mit.edu/GENSCAN.html">
GenScan</A>. These predictions are based on transcriptional, translational, 
and donor and acceptor splicing signals, plus length and compositional 
distributions of exons, introns and intergenic regions. The program does not 
use similarity to known proteins like 
<A HREF="http://www.ncbi.nlm.nih.gov:80/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11337476&dopt=Abstract">
GenomeScan</A>.  For a description of the Genscan program and the model that
underlies it, refer to Burge, C. and Karlin, S. (1997) 
<A HREF="http://linkage.rockefeller.edu/wli/gene/burge97.pdf"
TARGET=_blank>Prediction of complete gene structures in human genomic
DNA</A>. J. Mol. Biol. 268, 78-94.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show basic track
	information, plus links to the translated protein, predicted mRNA, and
	genomic sequences.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>genscan</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>GNF Ratio</B><BR>
This track shows expression data from GNF (The Genomics Institute of the
Novartis Research Foundation) using Affymetrix GeneChips. For more
information about the experiments, refer to the description page for this track.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	In dense display mode, the track color denotes the average signal over
	all experiments on a log base 2 scale. Lighter colors correspond
	to lower signals and darker colors correspond to higher signals. 
	In full display mode, the color of each item represents the log base 2 ratio 
	of the signal of that particular experiment to the median signal of all 
	experiments for that probe. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track filter:</I></B>	
	Clicking on the mini-button to the left of the track will display the track
	description page containing a filter to fine-tune the track display.
	The filter allows the user to change the display mode, group the displayed
	results, and change the display colors. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>affyRatio</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#LFS"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Human Blat </B><BR>
This track displays translated BLAT alignments of the human genome
vs. the mouse genome. 
The alignments were done with BLAT in translated protein mode using the
parameters -q=dnax -t=dnax and otherwise the default settings. Both
genomes were masked with RepeatMasker and Tandem Repeat Finder before
running BLAT. Places where more than 250 alignments occurred
over the same place were filtered out. Beware of alignments of greater
than 97% identity: these may reflect mouse contamination in the human
genome or human contamination in the mouse genome.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for a BLAT alignment track.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show basic track
	information and a link to sequence information for the alignment. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>chrN_blatHuman</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Human ESTs </B><BR>
This track shows alignments between Expressed Sequence Tags (ESTs) -
spliced and unspliced - in Genbank and the genome. ESTs are single read 
(typically approximately 500 base) sequences 
that usually represent fragments of transcribed genes. 
To view only spliced ESTs, see the Spliced EST track.
This track was generated by aligning human ESTs from Genbank against the
genome using the BLAT program. Note that the maximum intron length
allowed by BLAT is 500,000 bases, which may eliminate some ESTs with
very long introns that might otherwise align. When a single EST aligns
in multiple places, the alignment having the highest base identity is
found. Only alignments that have a base identity level within 1% of the
best are kept. Alignments must also have at least 93% base identity to be 
kept.
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for EST tracks.  Hatch marks on the 
	introns indicate the direction of transcription. In the December 2001 
	assembly and later, this direction is taken by looking at the splice sites. 
	In previous assemblies, the direction of transcription was taken from the
	Genbank annotations, which frequently were inaccurate; these should be 
	verified in the details page alignment on the basis of the GT-AG splice 
	donor/acceptor rule.  As with otherEST tracks, it may be necessary to zoom 
	in to display entries in full display mode due to the 300 line display 
	limit.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track filter:</I></B> 
	Clicking on the mini-button to the left of the track will display the track
	description page which contains a filter to fine-tune the track display.
	The filter allows the user to restrict the display to only those entries that 
	match certain keywords, and to change the color of the selected entries.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an individual EST entry will show basic track information,
	the accession number, GenBank highlights, and links to EST/Genomic
	Alignments.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>chrN_est</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Human mRNAs</B><BR>
The Human mRNA track shows alignments between human mRNAs in Genbank and the 
genome.  Genbank human mRNAs are aligned against the genome using the BLAT 
program. When a single mRNA aligns 
in multiple places, the alignment having the highest base identity is found. 
Only alignments that have a base identity level within 1% of the best
are kept. Alignments must also have at least 95% base identity
to be kept.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Aligning regions (usually exons) are shown as black boxes connected by 
	lines for gaps (usually spliced out introns). In dense display mode,
	darker gray items correspond to better quality matches. In full display mode,
	hatch marks on the introns indicate the direction of transcription. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on the mini-button to the left of the track will display the track
	description page which contains a filter to fine-tune the track display.
	The filter allows the user to restrict the display to only those entries that 
	match certain keywords, and to change the color of the selected entries.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Clicking on an entry in full display mode will show a summary of the Genbank 
	entry on that mRNA, convenient hyperlinks to major databases, and base-by-base 
	mRNA/genomic alignments. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>chrN_mrna</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>


<P><B>Map Contigs</B><BR>
This track shows the locations of contigs of clones on the physical map. 
In assembly versions prior to the August 6 2001 freeze, this track was
based on the Washington University accession map, which in turn was
based on a fingerprint contig (FPC) map described in 
<A
HREF="http://www.nature.com/cgi-taf/DynaPage.taf?file=/nature/journal/v409/n6822/full/409934a0_fs.html"
TARGET=_blank">'A physical map of the human genome'</A> in Nature volume 409 pages 934-941.
Starting with the August 6 2001 freeze, this track is based on tiling path 
fingerprint (TPF) maps curated by the sequencing centers responsible for each 
chromosome.  The TPF maps are merged with the FPC map,
favoring the TPF map where conflicts occur. This step increases the
clone coverage substantially over that in the TPF maps. The clone contigs in
this merged map are shown in this track.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	The clone contigs are represented by blocks that indicate their location
	on the assembly sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode will show a brief description of the 
	extent and number of clones in contig, and the DNA sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>ctgPos</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#ContigInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Mouse Blat </B><BR>
This track displays alignments of mouse genome to human. The mouse genomic 
DNA is from the Arachne assembly of mouse whole genome shotgun reads available 
at the <A HREF="http://www.ncbi.nlm.nih.gov/Traces/">Trace Repository</A>. 
The alignments were done with BLAT in translated protein mode requiring 2 nearby 
4-mer matches to trigger a detailed alignment. The human genome was masked with 
<A HREF="http://ftp.genome.washington.edu/cgi-bin/RepeatMasker">RepeatMasker</A> 
and <A HREF="http://c3.biomath.mssm.edu/trf.html">Tandem Repeat Finder</A> prior 
to running BLAT. Regions with more than 300 aligned reads were assumed novel repeat 
elements and filtered out. Alignments of greater than 97% identity may represent 
mouse contamination of human or vice versa.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows the display conventions for BLAT tracks.</TD></TR>	
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item will show details on mouse sequence fragment and 
	alignment details. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>chrN_blatMouse</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Mouse ESTs </B><BR>
This track shows alignments between mouse Expressed Sequence Tags (ESTs) -
spliced and unspliced - in Genbank and the genome. This track was generated by 
aligning mouse ESTs from Genbank against the
genome using the BLAT program. Note that the maximum intron length
allowed by BLAT is 500,000 bases, which may eliminate some ESTs with
very long introns that might otherwise align. When a single EST aligns
in multiple places, the alignment having the highest base identity is
found. Only alignments that have a base identity level within 1% of the
best are kept. Alignments must also have at least 93% base identity to be 
kept.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for EST tracks. As with other
	EST tracks, it may be necessary to zoom in to display entries in full
	display mode due to the 300 line display limit.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an individual EST entry will show basic track information,
	the accession number, GenBank highlights, and links to EST/Genomic
	Alignments.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>chrN_est</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Mouse mRNA </B><BR>
This track displays translated BLAT alignments of mouse mRNA 
from Genbank against the genome. The alignments were passed through a 
near-best-in-genome filter.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for mRNA tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track filter:</I></B> 
	Clicking on the mini-button to the left of the track will display the track
	description page which contains a filter to fine-tune the track display.
	The filter allows the user to restrict the display to only those entries that 
	match certain keywords, and to change the color of the selected entries.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show a summary of the Genbank 
	entry on that mRNA, convenient hyperlinks to major databases, and base-by-base 
	mRNA/genomic alignments. Some mRNAs align to multiple places but only the 
	near-best are retained. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>xenoMrna</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Mouse Synteny </B><BR>
This track shows syntenous (corresponding) regions between human and mouse 
chromosomes. This track was created by looking for homology to known mouse
genes in the draft assembly. The mouse data is provided at the chromosome 
(not cytoband) level. The data for this track was provided by NCBI. Refer to the 
<A HREF="http://www.ncbi.nlm.nih.gov/Homology/">NCBI Homology site</A> 
for more details.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for synteny tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode will show details of the
	mouse-human correspondence. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>mouseSyn</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#MouseSyn"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>NCI60</B><BR>
 This track displays cDNA expression data for a micro-array of 8,000 genes for 
60 cell lines used in an <A HREF="http://genome-www.stanford.edu/nci60/">NCI 
cancer drug screen</A>.  

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	To understand the track, experiment with it set to 
	full mode but other tracks (except known genes) hidden, while reading its
	<A HREF="http://genome.cse.ucsc.edu/cgi-bin/hgTrackUi?hgsid=1017057&c=chr3&g=nci60"> 
	filter page</A>. Each column of colored boxes represents variation in 
	transcript levels for a given cDNA across array experiments; each row 
	represents the measured transcript levels for all genes in a single sample. 
	The saturation of a color corresponds to the magnitude of transcript 
	variation. Black  indicates an undetectable change in expression; gray shows 
	missing data. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an individual track item in full mode open a rich details page. 
	The exon probe and experiment selected are highlighted in blue under item 
	name. Use filter page options to limit tissue types displayed and change the 
	color scheme of the display. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>nci60</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Nonhuman EST </B><BR>
This track displays translated BLAT alignments of non-human vertebrate
ESTs from Genbank. The alignments were passed through a piecewise
near-best-in-genome filter. In cases where ESTs align to multiple places in the 
genome, only the near-best are retained. Matches help validate human exon 
identification.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for EST tracks. As with other
	EST tracks, it may be necessary to zoom in to 
	display entries in full display mode due to the 300 line display 
	limit.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display will show a summary of the Genbank 
	entry, a hyperlink to the full record, PubMed links by gene name, 
	product, and author, and mRNA/genomic alignment information. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>xenoEst</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Nonhuman mRNA </B><BR>
This track displays translated BLAT alignments of non-human vertebrate mRNA 
from Genbank. The alignments were passed through a near-best-in-genome
filter. Some mRNAs align to multiple places but only the near-best are retained. 
For a distant vertebrate, alignment requires a fairly slowly evolving gene. 
Matches are often plausible ortholog candidates.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for mRNA tracks. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track filter:</I></B> 
	Clicking on the mini-button to the left of the track will display the track
	description page which contains a filter to fine-tune the track display.
	The filter allows the user to restrict the display to only those entries that 
	match certain keywords, and to change the color of the selected entries.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show a summary of the Genbank 
	entry on that mRNA, convenient hyperlinks to major databases, and 
	base-by-base mRNA/genomic alignments. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>xenoMrna</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Nonmouse mRNA </B><BR>
This track displays translated BLAT alignments of non-mouse vertebrate mRNA 
from Genbank. The alignments were passed through a near-best-in-genome
filter. Some mRNAs align to multiple places but only the near-best are retained. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for mRNA tracks. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track filter:</I></B> 
	Clicking on the mini-button to the left of the track will display the track
	description page which contains a filter to fine-tune the track display.
	The filter allows the user to restrict the display to only those entries that 
	match certain keywords, and to change the color of the selected entries.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show a summary of the Genbank 
	entry on that mRNA, convenient hyperlinks to major databases, and base-by-base 
	mRNA/genomic alignments. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>xenoMrna</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Overlap SNPs</B><BR>
This track shows single nucleotide polymorphisms found on overlapping contigs. 
Overlap SNPs are useful markers in mapping genes and diseases. This data was provided 
by the <A HREF="http://snp.cshl.org/">SNP Consortium</A>.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	SNP locations are indicated by thin grey vertical lines.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking in full mode brings up a location description and 
	<A HREF="http://www.ncbi.nlm.nih.gov/SNP/">dbSNP</A> link.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>snpNih</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#SNPsites"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Random SNPs</B><BR> 
This track displays single nucleotide polymorphisms (SNPs)found by random 
sequencing. Random SNPs are useful markers in mapping genes and complex traits.
Because these SNPs are detected from random reads, they are useful for
evolutionary studies. This data was provided by the <A HREF="http://snp.cshl.org/">SNP 
Consortium</A>.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	SNP locations are indicated by thin grey vertical lines.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
 	Clicking in full mode brings up a location description 
 	and <A HREF="http://www.ncbi.nlm.nih.gov/SNP/">dbSNP</A> link.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>snpTsc</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#SNPsites"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>


<P><B>RefSeq Genes </B><BR>
The RefSeq Gene track shows known protein coding genes taken from mRNA
reference sequences compiled at LocusLink. Refseq mRNAs are aligned against the 
genome using the BLAT program. When a single mRNA aligns in multiple places, only 
the best alignments are kept. Alignments must also have at least 98% sequence 
identity to be kept. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for known genes and gene predictions 
	tracks.
	Non-coding RNA genes have their own track in some assemblies. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display provides provides links to the 
	corresponding protein, mRNA, and genomic sequences, and related links to 
	outside databases. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>refGene</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>RepeatMasker</B><BR>
The track shows dispersed repeats as determined by 
<A HREF="http://ftp.genome.washington.edu/cgi-bin/RepeatMasker">RepeatMasker</A>. 
These elements include SINE, LINE, LTR, DNA, simple, low complexity, 
micro-satellite, tRNA, and other repeat families. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Darker gray boxes indicates a more recent age (as dated by alignment to a 
	canonical repeat). In dense mode for large chromosomal regions, the black 
	boxes representing repeats may merge into misleading single blocks in the 
	display. Full display mode shows 9 tracks representing the major repeat 
	classes.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>chrN_rmsk</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#RepeatInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>RNA Genes</B><BR>
This track shows human non-protein-coding RNA genes and RNA pseudogenes. This 
includes tRNAs, rRNAs, SRPs, methylation guide snoRNAs, U2-U6 RNA-like, and 
similar elements both known and predicted (by tRNAscanSE etc). The density of 
these elements can range from 1-4 per million bp in some chromosomes. Protein 
coding genes are displayed elsewhere in the RefSeq Genes track.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	RNA genes are represented by light-colored blocks. Pseudogenes are
	represented by darker blocks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will display feature 
	classification, method of prediction, location, reliability score, and
	a link to DNA sequence for the feature. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>rnaGene</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#RNAgenes"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Rosetta</B><BR>
This track displays microarray tissue 
<A HREF="http://www.ncbi.nlm.nih.gov:80/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11237012&dopt=Abstract">
expression data</A> provided by <A HREF="http://www.rii.com">Rosetta Inpharmatics</A>. 
It helps catalogue genes and transcriptome, complements computational gene 
predictions, and seeks to define gene boundaries via co-regulated expression 
of nearby exons. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	The data is stratified by various tissue and disease-specific 
	conditions of gene expression.  Confirmed and predicted exons are placed in 
	separate tracks. Click on the mini-button to the left of the track to
	display a track description page containing more information.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track filter:</I></B> 
	The track description page contains a filter to fine-tune the track display.
	The following display options are configurable:
	<BLOCKQUOTE>
		o Reference Sample: Determines which of the 69 different
		experiments are displayed, based on the reference sample
		used.<BR> 
		o Exons Shown: Determines whether data is shown for probes 
		corresponding to confirmed, predicted, or all exons.<BR> 
		o Color Scheme: Data are presented using two color false display. 
		The default colors can be changed for those who are colorblind. Gray 
		values indicate missing data.<BR> 
	</BLOCKQUOTE></TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on a Rosetta exon in full display mode will show an averaged display 
	that can be expanded via checkboxes to a graph of actual intensities for an 
	individual experiment.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>rosetta</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Sanger 22</B><BR>
This track contains the Sanger Centre annotations of chromosome 22. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on the track while in full display mode will show details of
	the methods used to construct this track as well as the contributors.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>sanger22</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Simple Repeats</B><BR>
This track displays simple tandem repeats (possibly imperfect) located by 
<A HREF="http://c3.biomath.mssm.edu/trf.html">Tandem Repeat Finder</A>, which 
is specialized to this purpose. These repeats can occur within coding regions of 
genes and may be quite polymorphic. Repeat expansions are sometimes associated with 
specific diseases.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	Repeats are represented by block boxes that indicate location in the
	assembly.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode will show the simple repeat 
	sequence, copy number, and quality of match statistics.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>simpleRepeats</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#SimpleRepeats"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Spliced ESTs</B><BR>
The Spliced EST track displays the subset of Expressed Sequence Tags (ESTs)
from Genbank that show signs of splicing when aligned against
the genome. ESTs are single read (typically approximately 500 base) sequences 
that usually represent fragments of transcribed genes. By requiring splicing, 
the level of contamination in the EST databases is drastically reduced at the 
expense of eliminating many genuine 3' ESTs. This track may also suggest 
alternate splices. The GT-AG splice donor/acceptor rule has not been applied.  
For a display of all ESTs (including unspliced), see the Human EST 
track.
This track was generated by aligning human ESTs from Genbank against the
genome using the BLAT program. Note that the maximum intron length
allowed by BLAT is 500,000 bases, which may eliminate some ESTs with
very long introns that might otherwise align. When a single EST aligns
in multiple places, the alignment having the highest base identity is
found. Only alignments that have a base identity level within 1% of the
best are kept. Alignments must also have at least 93% base identity to be 
kept.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for EST tracks 
	As with other EST tracks, it may be necessary to zoom in to 
	display entries in full display mode due to the 300 line display 
	limit.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an individual EST entry will show basic track information,
	the accession number, GenBank highlights, and links to translated
	protein, predicted mRNA, and genomic sequences.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>chrN_intronEst</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>STS Markers (human)</B><BR>
This track shows the locations of Sequence Tagged Site (STS) markers along
the draft assembly. The STS markers have been mapped using either
genetic (Genethon and Marshfield maps), radiation hybridization (the Stanford,
Whitehead RH, and GeneMap99 maps) or YAC mapping (the Whitehead YAC map)
techniques. In releases prior to August 2001, this track also shows the 
approximate position of FISH mapped clones. Starting with the August 2001 assembly, 
the FISH clones have been moved into a separate track. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Genetic map markers are shown in blue, and radiation hybrid
	map markers are shown in black. When a marker maps to multiple
	positions in the genome, it is shown in a lighter color.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on a specific marker in full display mode will show synonyms, 
	position information, primers, YAC map position, and uniSTS and GDB links.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>stsMap</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#STSMap"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>STS Markers (mouse)</B><BR>
This track shows the locations of Sequence Tagged Site (STS) markers
along the draft assembly. These STS markers appear on the MGI consensus
mouse genetic map. Information about the genetic map and STS marker
primer sequences are provided by the <A HREF="http://www.informatics.jax.org/"
TARGET=_blank> Mouse Genome Informatics</A> database group at The Jackson 
Laboratory.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Genetic map markers are show as thin black boxes. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on a specific marker in full display mode will show position 
	information, primers, genetic map position, genomic alignments, and MGI
	links.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>stsMapMouse</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#STSMap"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Tigr Gene Index</B><BR>
This track displays alignments of the 
<A HREF="http://www.tigr.org/tdb/hgi/index.html ">TIGR Gene Index</A> (TGI) 
against the human genome. The TGI is based largely on assemblies of EST
sequences from the public databases and TIGR.  Entries are 
actual assemblies providing a consensus sequence, not simply clusters of 
overlapping sequences, but generally don't predict whole genes. Gene
indices for mouse, rat, cow, and pig also contribute to this track. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show basic track
	information, links to the  predicted mRNA and
	genomic sequences, and a link to supporting information in the TIGR
	Human Gene Index.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>tigrGeneIndex</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>UniGene</B><BR>
This track shows <A HREF="http://www.ncbi.nlm.nih.gov/UniGene/">UniGene</A> 
clusters of ESTs aligned to the genome using BLAT. This data imperfectly 
indicates exon/intron boundaries and to a lesser extent genes. The Serial 
Analysis of Gene Expression <A HREF="http://www.ncbi.nlm.nih.gov/SAGE/ ">
(SAGE)</A> project at NCBI - a quantitative measurement of gene expression in 
various tissues and disease states - mapped small 10bp bar-coded expression 
markers onto these UniGene clusters in the SageMap project. The Genome Browser 
uses this mapping to provide access to the SAGE data associated with the 
UniGene track in the assembled and annotated genome. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on a UniGene feature will show an elaborate display of tabular and 
	graphical data from the SAGE project relevant to that cluster.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>uniGene</I> position table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

	</TD><TD WIDTH=15></TD></TR></TABLE>
    <br></TD></TR></TABLE>
</TD></TR></TABLE>

</BODY></HTML>

