<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: User's Guide</TITLE>
	<LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../inc/help.topbar.html"--></TD></TR>

<!--Content Tables----------------------------------------------------->  
<TR><TD ROWSPAN=2 VALIGN="top"></TD></TR>
<TR><TD>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp;Genome Browser User's Guide </B></FONT> 
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

<B><FONT SIZE=4> Table of Contents: </FONT></B>
<TABLE BORDER=0 cellpadding=0 cellspacing=0><TR><TH HEIGHT=10></TH></TR>
<TR><TD><UL><B>
  		<LI><A CLASS="toc" HREF="#What">
			What does the Genome Browser do?</A>
  		<LI><A CLASS="toc" HREF="#GetStarted">
			Getting started: Genome Browser gateways</A>
  		<LI><A CLASS="toc" HREF="#FineTuning">
			Fine-tuning the Genome Browser display</A>
  		<LI><A CLASS="toc" HREF="#BLATAlign">
			Using BLAT alignments</A>
  		<LI><A CLASS="toc" HREF="#TableBrowser">
			Getting Started on the Table Browser</A>
	</B></UL></TD><TD><UL><B>
		<LI><A CLASS="toc" HREF="#TrackFormatDNA">
			DNA text formatting</A>
		<LI><A CLASS="toc" HREF="#Convert">
			Converting data between assemblies</A>
		<LI><A CLASS="toc" HREF="#Download">
			Downloading genome data</A>
		<LI><A CLASS="toc" HREF="#CustomTracks">
			Creating custom annotation tracks</A>
  		<LI><A CLASS="toc" HREF="#IndivTracks">
			Annotation track descriptions</A>
	</B></UL></TD></TR></TABLE>
<em>Last updated on 3 Mar. 2005. 
<A HREF="mailto:&#103;&#101;&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;sc.e&#100;&#117;">Questions and feedback welcome</A></em>.
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="What"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; What does
			the Genome Browser do?  </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
As vertebrate genome sequences near completion and research re-focuses on 
their analysis, the issue of effective sequence display becomes critical: it is not 
helpful to have 3 billion letters of genomic DNA shown as plain text! As an 
alternative, the UCSC Genome Browser provides a rapid and reliable display of any 
requested portion of genomes at any scale, together with dozens of 
aligned annotation tracks (known genes, predicted genes, ESTs, mRNAs, CpG islands, 
assembly gaps and coverage, chromosomal bands, mouse homologies, and more). Half of 
the annotation tracks are computed at UCSC from publicly available
sequence data. The remaining tracks are provided by collaborators worldwide. Users can 
also add their own custom tracks to the browser for educational or research
purposes.
<P>
The Genome Browser stacks annotation tracks beneath genome 
coordinate positions, allowing rapid visual correlation of different types of 
information. The user can look at a whole chromosome to get a feel for gene 
density, open a specific cytogenetic band to see a positionally mapped disease gene 
candidate, or zoom in to a particular gene to view its spliced ESTs and possible 
alternative splicing. The Genome Browser itself does not draw conclusions; rather, it 
collates all relevant information in one location, leaving the exploration and 
interpretation to the user.
<P>
The Genome Browser supports text and sequence based searches that provide quick, precise 
access to any region of specific interest. Secondary links from individual
entries within annotation tracks lead to sequence details and supplementary off-site 
databases.  To control information overload, tracks need not be displayed in full. Tracks
can be hidden, collapsed into a condensed or single-line display, or filtered according 
to the user's criteria.  Zooming and scrolling controls help to narrow or broaden the 
displayed chromosomal range to focus on the exact region of interest. 
Clicking on an individual item within a track opens a details page containing a 
summary of properties and links to off-site repositories such as 
PubMed, GenBank, Entrez, and OMIM. The page provides item-specific information on 
position, cytoband, strand, data source, and encoded protein, mRNA, genomic sequence 
and alignment, as appropriate to the nature of the track.
<P>
A blue navigation bar at the top of the browser provides links to several other tools 
and data sources. For instance, the DNA link
enables the user to view the raw genomic DNA sequence for the coordinate range 
displayed in the browser window. This DNA can encode track features via elaborate 
text formatting options. Other links tie the Genome Browser to the BLAT 
alignment tool, provide access to the underlying relational database via the
Table Browser, convert coordinates across different assembly dates, and open the 
window at the complementary <A HREF="http://www.ensembl.org/" TARGET=_blank>Ensembl</A>
or <A HREF="http://www.ncbi.nlm.nih.gov/mapview/" TARGET=_blank>NCBI Map 
Viewer</A> annotation.
<P>
The browser data represents an immense 
<A HREF="../credits.html">collaborative effort</A> 
involving thousands of people from the international biomedical research community. 
The UCSC Bioinformatics Group itself does no sequencing. Although it creates 
the majority of the annotation tracks in-house, the annotations are based on
publicly available data contributed by many labs and research groups throughout
the world. Several of the Genome Browser annotations are generated in 
collaboration with outside individuals or are contributed wholly by external
research groups. UCSC's
other major roles include building genome assemblies, creating the Genome Browser work 
environment, and serving it online. The majority of the sequence data, annotation tracks, and even 
software are in the public domain and are available for anyone
to <A HREF="#Download">download</A>. 
<P>
In addition to the Genome Browser, the UCSC Genome Bioinformatics group provides 
several other tools for viewing and interpreting genome data: 
<UL>
<LI><A HREF="../../cgi-bin/hgBlat?command=start">BLAT<A> - a 
fast sequence-alignment tool similar to BLAST. <A HREF="#BLATAlign">Read more</A>.
<LI><A HREF="../../cgi-bin/hgTables">Table Browser</A> - 
convenient text-based access to the database underlying the Genome Browser. 
<A HREF="hgTablesHelp.html">Read
more</A>.
<LI><A HREF="../../cgi-bin/hgNear">Gene Sorter</A> - 
expression, homology, and other information on groups of genes that can
be related in many ways. 
<A HREF="hgNearHelp.html">Read more</A>.
<LI>Proteome Browser (accessible from Known Genes details pages) - 
protein property data and links to a wealth of related information. 
<A HREF="pbTracksHelpFiles/pbTracksHelp.shtml">Read more</A>.
</UL>
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="GetStarted"></A><BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Getting Started: 
		Genome Browser gateways</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
The UCSC Genome Bioinformatics <A HREF="../../index.html" 
TARGET=_blank>home page</A>
provides access to Genome Browsers on several different genome assemblies. 
To get started, click the Browser link on the blue sidebar. This will take 
you to a Gateway page where you can select which genome to display.
<P>
<B>Opening the Genome Browser at a specific position</B><BR> 
To get oriented in using the Genome Browser, try viewing a gene or region of 
the genome with which you are already familiar, or use the default position. 
To open the Genome Browser window:
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top"><B>1.</B></TD><TD>
	Select the clade, genome and assembly that you wish to display from 
	the corresponding pull-down menus. To access older assembly versions 
	that are no longer available from the menu, look in the Genome 
	Browser <A HREF="http://genome-archive.cse.ucsc.edu/"
	TARGET=_blank>archives</A>.
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top"><B>2.</B></TD><TD>
	Specify the genome location you'd like the Genome Browser to open to.
	To select a location, enter a valid position query in the 
	<I>position or search term</I> text box at the top of the Gateway 
	page or accept the default position already 
	displayed. The search supports several different types of queries: gene 
	symbols, mRNA or EST accession numbers, chromosome 
	bands, descriptive terms likely to occur in GenBank text, or specific chromosomal 
	ranges. To display a region encompassed by two features on the same 
	chromosome, use a semi-colon, e.g. CRYBB3;CRYBB1. The Gateway page shows examples of some of the search requests supported by
	the Genome Browser.
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top"><B>3.</B></TD><TD>
	Click the <I>submit</I> button to open up the Genome Browser window to the
	requested location. In cases where a specific term (accession, gene name,
	etc.) was queried, the item will be highlighted in the display.
	</TD></TR>
</TABLE>
<P>
Occasionally the Gateway page returns a list of several matches in response 
to a search, rather than immediately displaying the Genome Browser 
window. When this occurs, click on the item in which you're interested and 
the Genome Browser will open to that location.
<P>
The search mechanism is not a site-wide search engine. Instead, it 
primarily searches GenBank mRNA records whose text annotations can include 
gene names, gene symbols, journal title words, author names, and RefSeq 
mRNAs. Searches on other selected identifiers, such as NP and NM accession 
numbers, OMIM identifiers, and Entrez Gene IDs are supported.  However, some 
types of queries will return an error, e.g. post-assembly GenBank 
entries, withdrawn gene names, and abandoned synonyms. If your initial
query is unsuccessful, try entering a different related term that
may produce the same location. For example, if a query on a gene symbol
produces no results, try entering an mRNA accession, gene ID number, or
descriptive words associated with the gene.
<P>
<B>Finding a genome location using BLAT</B><BR>
If you have genomic, mRNA, or protein sequence, but don't know the name or
the location to which it maps in the genome, the <A HREF="../../cgi-bin/hgBlat">
BLAT</A> tool will rapidly locate the position by homology alignment, provided that the 
region has been sequenced. This search will find close members of the gene
family, as well as assembly duplication artifacts. An entire set of query sequences can 
be looked up simultaneously when provided in fasta format. 
<P>
A successful BLAT search 
returns a list of one or more genome locations that match the input sequence. To
view one of the alignments in the Genome Browser, click the <I>browser</I> link for the 
match. The <I>details</I> link can be used to preview the alignment to determine if
it is of sufficient match quality to merit viewing in the Genome Browser. If too many BLAT hits 
occur, try narrowing the search by filtering the sequence in slow mode with
<A HREF="http://www.repeatmasker.org" TARGET=_blank>RepeatMasker</A>, then rerunning the BLAT search. 
<P>
For more information
on conducting and fine-tuning BLAT searches, refer to the <A HREF="#BLATAlign">
BLAT section</A> of this document.
<P>
<B>Opening the Genome Browser with a custom annotation track</B><BR>
You can open the Genome Browser window with a custom annotation track displayed
by using the Add Your Own Tracks feature, accessed from the Gateway page. For more information on creating and using custom annotation tracks, refer to the <A
HREF="customTrack.html" TARGET=_blank>Custom 
Annotation Track</A> documentation.
<P>
Annotation track data can be entered in one of three ways:
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Enter the file name for an annotation track source file in the
	Annotation File text box.
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Type or paste the annotation track data into the large text box.
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	If the annotation data is accessible through a URL, enter the URL name in the 
	large text box.
	</TD></TR>
</TABLE>
<P>
Once you've entered the annotation information, click the <I>submit</I> button at the top of 
the Gateway page to open up the Genome Browser with the annotation track
displayed.
<P>
The Genome Browser also provides a <A HREF="../customTracks/custTracks.html" TARGET=_blank>collection of custom annotation tracks</A> contributed by the UCSC Genome Bioinformatics group and the research community. 

<P><B>NOTE:</B> If an annotation track does not display correctly when
you attempt to upload it, you may need to reset the Genome Browser to 
its default settings, then reload the track. For information on
troubleshooting display problems with custom annotation tracks, refer to the
section on <A
HREF="customTrack.html#TROUBLE" 
TARGET=_blank>Troubleshooting Annotation Display Problems</A>.
<P>
<B>Viewing genome data as text</B><BR> 
The 
<A HREF="../../cgi-bin/hgTables" TARGET=_blank>Table Browser</A>,
a portal to the underlying open source <A
HREF="http://www.MySQL.com" TARGET=_blank>MySQL</A> relational database driving the Genome
Browser, displays genomic data as columns of text rather than as
graphical tracks. For more information on using the Table Browser, see the
section <A HREF="#TableBrowser"> Getting Started: on the Table Browser</A>.
<P>
<B>Opening the Genome Browser from external gateways</B><BR> 
Several external gateways provide direct links into the Genome Browser. Examples
include: <A HREF="http://www.ncbi.nlm.nih.gov/entrez/" TARGET=_blank>Entrez Gene</A>, 
<A HREF="http://www.ncbi.nlm.nih.gov/IEB/Research/Acembly/index.html"
TARGET=_blank>AceView</A>,
<A HREF="http://www.ensembl.org/" TARGET=_blank>Ensembl</A>, <A
HREF="http://supfam.mrc-lmb.cam.ac.uk/SUPERFAMILY/" TARGET=_blank>SuperFamily</A>, <A
HREF="http://www.genelynx.org/" TARGET=_blank>GeneLynx</A>, and <A
HREF="http://bioinformatics.weizmann.ac.il/cards/" TARGET=_blank>GeneCards</A>. Journal
articles can also link to the browser and provide custom tracks. Be sure to use
the assembly date appropriate to the provided coordinates when using data from
a journal source.
<P>
<B>Tips for Use</B><BR>
To facilitate your return to regions of interest within the Genome
Browser, save the coordinate range or bookmark the page of displays that you plan to 
revisit or wish to share with others. 
<P>
It is usually best to work with the most recent assembly even though a full 
set of tracks might not yet be ready. Be aware that the coordinates of a given feature 
on an unfinished chromosome may change from one assembly to the next as
gaps are filled, artifactual duplications are reduced, and strand orientations
are corrected.  The Genome Browser offers multiple tools that can correctly 
convert coordinates between different assembly releases. For more information
on conversion tools,
see the section <A HREF="#Convert">Converting data between assemblies</A>.
<P>
To ensure uninterrupted browser services for your research during UCSC server 
maintenance and power outages, bookmark a
<A HREF="../../mirror.html" TARGET=_blank>mirror</A> site that 
replicates the UCSC genome browser. 
<P>
Bear in mind that the Genome Browser cannot outperform the underlying quality of 
the draft genome. Assembly errors and sequence gaps may still occur well into
the sequencing process due to regions that are intrinsically difficult to sequence. 
Artifactual duplications arise as unavoidable compromises during a build, causing 
misleading matches in genome coordinates found by alignment. 
<BR>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<A NAME="FineTuning"></A><BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; Interpreting and fine-tuning the
		Genome Browser display</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
<p>
The Genome Browser annotation tracks page displays a genome location 
specified through a Gateway search, a BLAT search, or an uploaded 
custom annotation track. There are five main features on this page: a set of 
<A HREF="#NAV">navigation controls</A>, a 
chromosome ideogram, the annotations tracks image, 
<A HREF=#TRACK_CONFIG>display configuration buttons</A>, and a set of track 
<A HREF="#TRACK_CONT">display controls</A>.
<P>
The first time you open the
Genome Browser, it will use the application default values to configure the
annotation tracks display. By manipulating the navigation, configuration 
and display controls, you can customize the annotation tracks display to 
suit your needs. 
For a complete description of the annotation tracks available in all assembly
versions supported by the Genome Browser, see the <A HREF="#IndivTracks">
Annotation Track Descriptions</A> section.
<P>
The Genome Browser retains user preferences from session to session within the
same web browser, although it never monitors or records user activities or submitted 
data. 
To restore the default settings, click the &quot;Click here to reset&quot; 
link on the Genome Browser Gateway page. To return the display to the default 
set of tracks (but retain other configured Genome Browser settings), click 
the <I>default tracks</I> button on the Genome Browser page.
<P>
<B>Display conventions</B><BR>
The annotation tracks displayed in the Genome Browser use a common set of
display conventions:<BR>
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Annotation track
	descriptions:</B> 
	Each annotation track has an associated description page that contains a
	discussion of the track, the methods used to create the annotation, the data 
	sources and credits for the track, and (in some cases) filter and
	configuration options to
	fine-tune the information displayed in the track. To view the
	description page, click on the mini-button to the left of a displayed
	track or on the label for the track in the Track Controls section.</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Annotation track
	details pages:</B> 
	When an annotation track is displayed in full, pack, or squish mode, 
	each line item
	within the track has an associated details page that can be displayed by 
	clicking on the item or its label. The
	information contained in the details page varies by annotation track, but
	may include basic position information about the item, related links 
	to outside sites 
	and databases, links to genomic alignments, or links to corresponding
	mRNA, genomic, and protein sequences.
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><A NAME="GeneDisplay"></A><B>Gene prediction tracks:</B> 
	Coding exons are represented by blocks connected by horizontal lines
	representing introns. The 5' and 3' untranslated regions (UTRs) are displayed 
	as thinner blocks on the leading and trailing ends of the aligning
	regions. In full display mode, arrowheads on
	the connecting intron lines indicate the direction of transcription.
	In situations where no intron is visible (e.g. single-exon genes,
	extremely zoomed-in displays), the arrowheads are displayed on the exon 
	block itself.</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><A NAME="PSLDisplay"></A><B>Pat space layout (PSL) alignment 
	tracks:</B> 
	Aligning regions (usually exons) are shown as black blocks. In dense display 
	mode, the degree of darkness corresponds to the number of features aligning to 
	the region or the degree of quality of the match. In full display mode, the aligning regions are connected by lines 
	representing gaps in the alignment (typically spliced-out introns), with arrowheads indicating 
	the direction of transcription, which is determined by looking at the
	splice sites. In situations where no gap lines are 
	visible, the arrowheads are displayed on the block itself. 
	To prevent display problems, the Genome Browser imposes an upper 
	limit on the number of alignments that can be viewed simultaneously 
	within the tracks image. When this limit is exceeded, the Browser 
	displays the best several hundred alignments in a condensed display
	mode, then lists the number of undisplayed alignments in the last 
	row of the track. In this situation, try zooming in to display 
	more entries or to return the track to full display mode. </TD></TR>

<A NAME="ChainDisplay"></A>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>&quot;Chain&quot; tracks 
	(2-species alignment):</B> 
	Chain tracks display boxes joined together by either single or double 
	lines. The boxes represent aligning regions. Single lines indicate gaps 
	that are largely due to a deletion in the genome of the first species or 
	an insertion in the genome of the second species. Double lines represent 
	more complex gaps that involve substantial sequence in both species. 
	This may result from inversions, overlapping deletions, an abundance of 
	local mutation, or an unsequenced gap in one species. In cases where 
	there are multiple chains over a particular portion of the genome, 
	chains with single-lined gaps are often due to processed pseudogenes, 
	while chains with double-lined gaps are more often due to paralogs and 
	unprocessed pseudogenes. In the fuller display modes, the individual 
	feature names indicate the chromosome, strand, and location 
	(in thousands) of the match for each matching alignment. </TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><A NAME="NetDisplay"></A><B>&quot;Net&quot; tracks 
	(2-species alignment):</B> 
	Boxes represent ungapped alignments, while lines represent gaps. Clicking 
	on a box displays detailed information about the chain as a whole, while 
	clicking on a line shows information on the gap. The detailed information 	  
	is useful in determining the cause of the gap or, for lower level chains, 	    
	the genomic rearrangement.  Individual items in the display are 
	categorized as one of four types (other than gap):
	<UL>
	<LI>
	<em>Top</em> - The best, longest match. Displayed on level 1. 
	<LI>
	<em>Syn</em> - Lineups on the same chromosome as the gap in the level 
	above it. 
	<LI>
	<em>Inv</em> - A lineup on the same chromosome as the gap above it, but 
	in the opposite orientation. 
	<LI>
	<em>NonSyn</em> - A match to a chromosome different from the gap in the 
	level above. 
	</UL></TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><A NAME="XSpeciesDisplay"></A><B>Cross-species synteny and
	orthology tracks:</B> 
	Sequences from an organism that match regions of the displayed genome 
	are represented by blocks of various colors. The block color indicates the 
	chromosomal location of the sequence in the organism's genome, as shown by the
	Chromosome Color Key below the annotation track window.</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>&quot;Wiggle&quot; tracks 
	(Cross-species conservation):</B> 
	These tracks plot a continuous function along a chromosome. Data is 
	displayed in windows of a set number of base pairs in width. 
	The score for each window is given on a logarithmic scale, which 
	displays as "mountain ranges". The display characteristics vary among
	the tracks in this group. See the individual track descriptions for more
	information on interpreting the display.</TD></TR>
</TABLE> 
<P>
<A NAME="TRACK_CONT"></A>
<B>Changing the display mode of an individual annotation track</B><BR>
Each annotation track within the window may have up to five display modes: 
<P>
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Hide:</B> the
	track is not displayed at all. To hide all the annotation tracks,
	click the <I>hide all</I> button. This mode is useful for restricting
	the display to only those tracks in which you are interested. For
	example, someone who is not interested in SNPs or mouse synteny may want 
	to hide these tracks to reduce track clutter and improve speed. There
	are a few annotation tracks that pertain only to one specific
	chromosome, e.g. Sanger22, Rosetta. In these cases, the track and its 
	associated controller will be hidden automatically when the track window is 
	not open to the relevant chromosome.</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Dense:</B> the 
	track is displayed with all features collapsed into a single 
	line. This mode is useful for reducing the amount of space used by a
	track when you don't need individual line item details or when you just
	want to get an overall view of an annotation. For example, by opening an
	entire chromosome and setting the RefSeq Genes track to dense, you can get 
	a feel for the known gene density of the chromosome without displaying
	excessive detail. </TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Full:</B> the
	track is displayed with each annotation feature on a separate line. It is
	recommended that you use this option sparingly, due to the large number
	of individual track items that may potentially align at the selected position. For
	example, hundreds of ESTs might align with a specified gene. When
	the number of lines within a requested track location exceeds 250, the
	track automatically defaults to a more tightly-packed display mode. In 
	this situation,
	you can restore the track display to full mode by narrowing the chromosomal 
	range displayed or by using a track filter to reduce the number of items 
	displayed. On tracks that contain only hide, dense, and full modes, you 
	can toggle between full and dense display modes by clicking on the
	track's center label.</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Squish:</B> the 
	track is displayed with each annotation feature shown separately, but
	at 50% the height of full mode. Features are unlabeled, and more than one
	may be drawn on the same line. This mode is useful for reducing the 
	amount of space used by a track when you want to view a large number of
	individual features and get an overall view of an annotation. It is
	particularly good for displaying tracks in which a large number of 
	features align to a particular section of a chromosome, e.g. EST tracks. 
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Pack:</B> the 
	track is displayed with each annotation feature shown separately and
	labeled, but not necessarily displayed on a separate line. 
	This mode is useful for reducing the 
	amount of space used by a track when you want to view the large number of
	individual features allowed by squish mode, but need the labeling and
	display size provided by full mode. When the number of lines within the
	requested track location exceeds 250, the track automatically defaults
	to squish display mode. In this situation, you can restore the track 
	display to pack mode by narrowing the chromosomal range displayed or by 
	using a track filter to reduce the number of items 
	displayed.  To toggle between pack and full display modes, click on 
	the track's center label.
	</TD></TR>
</TABLE> 
<P>
The track display controls are grouped into categories that reflect the type 
of data in the track, e.g. Gene Prediction Tracks, mRNA and EST tracks, etc. 
To change the display mode for a track, find the track's controller in the
Track Controls section at the bottom of the Genome Browser page, select the desired
mode from the control's display menu, and then click the <I>refresh</I> button. 
Alternatively, you can toggle between dense and full modes 
for a displayed track (or pack mode when available) by clicking on the optional 
center label for the track. 
<P>
<B>Changing the display mode for a group of tracks</B><BR>
Track display modes may be set individually or as a group on the Genome
Browser Track Configuration page. To access the configuration page, click the
<em>configure</em> button on the annotation tracks page or the 
<em>configure tracks and display</em> button on the Gateway page. Exercise
caution when using the <em>show all</em> buttons on track groups or 
assemblies that contain a large number tracks; this may seriously impact the
display performance of the Genome Browser or cause your Internet browser to
time out.
<P>
<B>Hiding the track display controls</B><BR>
The entire set of track display controls at the bottom of the annotation 
tracks page may be hidden from view by checking the <em>Show track controls 
under main graphic</em> option in the Configure Image section of the Track 
Configuration page.
<P>
<B>Changing the display of a track by using filters and configuration options</B><BR>
Some tracks have additional filter and configuration capabilities, 
e.g. EST tracks, mRNA tracks, NC160, etc. These options
let the user modify the color or restrict the data displayed within an annotation track.
Filters are useful for focusing attention on items relevant to the current task
in tracks that contain large amounts of data. For example, to highlight ESTs
expressed in the liver, set the EST track filter to display items in a
different color when the associated tissue keyword is "liver". Configuration
options let the user adjust the display to best show the data of interest.
For example, the min vertical viewing range value on wiggle tracks
can be used to establish a data threshold. By setting the min value to &quot;50&quot;, only data values greater than 50 percent will display.
<P>
To access filter and configuration options for a specific annotation track, open
the tracks' description page by clicking the label for the track's control menu 
under the Track Controls section or the mini-button to the left of the displayed 
track. The filter and configration section is located at the top of the 
description page. In most instance, more information about the configuration 
options is available within the description text or through a special help link 
located in the configuration section. 
<P>
Filter and configuration settings are persistent from session to session on the 
same web browser. To return the Genome Browser display to the default set of 
tracks (but retain other configured Genome Browser settings), click the 
<I>default tracks</I> button on the Genome Browser tracks page.
To remove all user configuration settings and completely restore the defaults,
click the &quot;Click here to reset&quot; link on the Genome Browser Gateway 
page. 
<P>
<A NAME="NAV"></A>
<B>Zooming and scrolling the tracks display</B><BR>
At times you may want to adjust the amount of flanking region displayed in the
annotation tracks window or adjust the scale of the display. At a scale of 1 pixel per 
base pair, the window accurately displays the width of exons and introns, and 
indicates the direction of transcription (using arrowheads) for multi-exon features. At a 
grosser scale, certain features - such as thin exons - may disappear. Also, some
exons may falsely appear to fall within RepeatMasker features at some scales. 
<P>
Click the <I>zoom in</I> and <I>zoom out</I> buttons at the top of the 
Genome Browser page to zoom in or out on the center of the annotation tracks window by 1.5, 3 or
10-fold. Alternatively, you can zoom in 3-fold on the display by clicking
anywhere on the
Base Position track. In this case, the zoom is centered on the coordinate of the
mouse click.  To view the base composition of the sequence underlying the
current annotation track display, click the <em>base</em> button. 
<P>
To scroll the annotation tracks sideways to the left or right by 10%, 50%, or 95% of the displayed
size (as given in base pairs), click the corresponding <I>move</I> arrow. It is also 
possible to scroll the left or right side of the tracks by a specified number
of vertical gridlines while keeping the
position of the opposite side fixed. To do this, click the appropriate <I>move 
start</I> or <I>move end</I> arrow, located under the annotation tracks window.
For example, to keep the left-hand display coordinate fixed but increase the
right-hand coordinate, you would click the right-hand <I>move end</I> arrow. To increase
or decrease the gridline scroll interval, edit the value in the <I>move start</I> or
<I>move end</I> text box. 
<P>
<B>Changing the displayed track position</B><BR>
To display a completely different position in the genome, enter the new query in
the <I>position/search</I> text box, then click the <I>jump</I> button. For more information on
valid entries for this text box, refer to the <A HREF="#GetStarted">Getting
Started</A> section.
<P>
<A NAME="TRACK_CONFIG"></A>
<B>Changing the width of the annotation track window</B><BR>
By default, the width of the annotation track window is set to 620 pixels. To modify the width to 
best suit the display capabilities of your monitor, enter a new value in the 
<I>image width</I> text box on the Track Configuration page, then click the 
<I>submit</I> button. For example, setting the 
display to 1100 pixels on a 19" monitor will increase the visible portion of the genome
and reduce the need for redraws. The maximum supported width is 5000 pixels.
<P>
<B>Changing the text size in the annotation track image</B><BR>
The annotation track image may be adjusted to display text in a range of
fonts from &quot;tiny&quot; to &quot;huge&quot;. To change the size of the
text, select an option from the <em>text size</em> pull-down menu on the 
Track Configuration page, then click <em>Submit</em>. The text size is set to
&quot;small&quot; by default.
<P>
<B>Hiding the annotation track labels</B><BR>
The track and element labels displayed above and to the left of the tracks
in the annotation tracks image may be hidden from view by unchecking the
<em>Display track descriptions above each track</em> and <em>Display labels 
to the left of items in tracks</em> boxes, respectively, on the Track 
Configuration page.
<P>
<B>Hiding the display grid on the annotation tracks image</B><BR>
The light blue vertical guidelines on the annotation tracks image may be
removed by unchecking the <em>Show light blue vertical guidelines</em> box
on the Track Configuration page.
<P>
<B>Hiding the chromosome ideogram</B><BR>
The chromosome ideogram, located just above the annotation tracks image,
provides a graphical overview of the features on the selected chromosome, 
including its bands, the position of the centromere, and an indication of
the region currently displayed in the annotation tracks image. 
To hide the ideogram, uncheck the <em>Display chromosome ideogram above 
main graphic</em> box on the Tracks Configuraiton page.
<P>
<B>Printing a copy of the annotation track window</B><BR>
The Genome Browser provides a mechanism for saving a copy of 
the currently displayed annotation tracks image to a file that can be printed 
or edited. Images saved in PostScript format can be printed at high resolution
and edited by drawing programs such as Adobe Illustrator. This is useful for
generating figures intended for publication. Images can also be saved in PDF
format for viewing by Adobe Acrobat Reader.
<P>
To print or save the image to a file:
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top"><B>1.</B></TD><TD>
	Click the PDF/PS link in the menu on the annotation tracks page.</TD></TR>
	<TR><TD WIDTH="20" VALIGN="top"><B>2.</B></TD><TD>
	Click the PostScript or PDF link.</TD></TR> 
</TABLE>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<A NAME="BLATAlign"></A><BR>	

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Using BLAT alignments </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
BLAT (BLAST-Like Alignment Tool) is a very fast sequence alignment tool similar to 
BLAST. 
For more information on BLAT's internal scoring schemes and its overall n-mer 
alignment seed strategy, refer to W. James Kent (2002) <A
HREF="http://www.genome.org/cgi/content/abstract/12/4/656" TARGET=_blank>BLAT -
The BLAST-Like Alignment Tool</A>, <em>Genome Res</em> 12:4 656-664. 
<P>
On DNA queries, BLAT is designed to quickly find sequences with 95% or greater 
similarity of length 40 bases or more. It may miss genomic alignments that
are more divergent or shorter than these minimums, although it will find perfect 
sequence matches of 33 bases and sometimes as few as 22 bases. The tool is
capable of aligning sequences that contain large introns. 
On protein queries, BLAT rapidly locates genomic sequences with 80% or greater 
similarity of length 20 amino acids or more. In general, gene family members that 
arose within the last 350 million years can generally be detected. More
divergent sequences can be aligned to the human genome by using NCBI's BLAST and
psi-BLAST, then using BLAT to align the resulting match onto the UCSC genome 
assembly. In practice DNA BLAT works well on primates, and protein BLAT works
well on land vertebrates. 
<P>
Some common uses of BLAT include: 
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	finding the genomic coordinates of mRNA or protein within a given assembly
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	determining the exon structure of a gene
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	displaying a coding region within a full-length gene
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	isolating an EST of special interest as its own track
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	searching for gene family members
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	finding human homologs of a query from another species. 
	</TD></TR> 
</TABLE>	
	
<P><B>Making a BLAT query</B><BR>
To locate a nucleotide or protein within a genome using BLAT: 
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top"><B>1.</B></TD><TD>
	Open the <A HREF="../../cgi-bin/hgBlat?command=start" 
	TARGET=_blank>BLAT Search Genome page</A> by clicking the BLAT link on the top 
	blue menu bar of any of the Genome Browser pages.</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top"><B>2.</B></TD><TD>
	Select the genome, assembly, query type, output sort order, and output
	type. To order the search results based on the closeness of the sequence
	match, choose one of the score options in the <I>Sort output</I> menu.
	The score is determined by the number of matches vs. mismatches in the
	final alignment of the query to the genome.  
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top"><B>3.</B></TD><TD>
	If the sequence to be uploaded is in an unformatted plain text file, enter 
	the file name in the <I>Upload sequence</I> text box, then click the
	<I>submit file</I> 
	button.  Otherwise, paste the sequence or fasta-formatted list into the large 
	edit box, and then click the <I>submit</I> button. Input sequence can be
	obtained from the Genome Browser as well as from a custom annotation
	track.</TD></TR> 
</TABLE> 
<P>
Header lines may be included in the input text if they are preceded by > and
contain unique names. 
Multiple sequences may be submitted at the same time if they are of the
same type and are preceded by unique header lines. Numbers, spaces, and extraneous 
characters are ignored: 
<P>
<PRE>
>sequence_1
ATGCAGAGCAAGGTGCTGCTGGCCGTCGCCCTGTGGCTCTGCGTGGAGAC
CCGGGCCGCCTCTGTGGGTTTGCCTAGTGTTTCTCTTGATCTGCCCAGGC
>sequence_2
ATGTTGTTTACCGTAAGCTGTAGTAAAATGAGCTCGATTGTTGACAGAGA
TGACAGTAGTATTTTTGATGGGTTGGTGGAAGAAGATGACAAGGACAAAG
>sequence_3
ATGCTGCGAACAGAGAGCTGCCGCCCCAGGTCGCCCGCCGGACAGGTGGC
CGCGGCGTCCCCGCTCCTGCTGCTGCTGCTGCTGCTCGCCTGGTGCGCGG
</PRE>

<P><B>BLAT limitations</B><BR>
DNA input sequences are limited to a maximum length of 25,000 bases. Protein or
translated input sequences must not exceed 5000 letters. As many as 25 multiple 
sequences may be submitted at the same time. The maximum combined length of DNA 
input for multiple sequence submissions is 50,000 bases (with a 25,000 base 
limit per individual sequence). For 
protein or translated input, the maximum combined input length is 12,500 letters 
(with a 5000 letter limit per individual sequence). 
<P>
NOTE: Program-driven BLAT use is
limited to a maximum of one hit every 15 seconds and no more than 5000 hits per
day.
<P><B>BLAT query search results</B><BR>
If a query returns successfully, BLAT will display a flat database file that 
summarizes the alignments found. A BLAT query often generates multiple 
hits. This can happen when the genome contains 
multiple copies of a sequence, paralogs, pseudogenes, statistical coincidences, 
artifactual assembly duplications, or when the query itself contains repeats or common 
retrotransposons. When too many hits occur, try resubmitting the query sequence
after filtering in slow mode with RepeatMasker. 
<P>
Items in the search results list are ordered by the criteria specified in the 
<I>Sort output</I> menu. Each line item provides links to view the details of the sequence 
alignment or to open the corresponding view in the Genome Browser. The
<I>details</I> link gives the letter-by-letter alignment of the sequence to the 
genome. It is recommended 
that you first examine the details of the alignment for match quality before viewing
the sequence in the Genome Browser. 
<P>
When several nearby BLAT matches occur on a single chromosome, a 
simple trick can be used to quickly adjust the Genome Browser track window to display 
all of them: open the Genome Browser with the match that has the lowest chromosome 
start coordinate, paste in the highest chromosome end coordinate from the list of 
matches, then click the <I>jump</I> button. 
<P><B>Creating a custom annotation track from BLAT output</B><BR>
To make a custom track directly from BLAT, select the 
<A HREF="customTrack.html#PSL">PSL format</A>
output option. The resulting PSL track can be uploaded into the Genome Browser by 
pasting it into the <I>Add Your Own Tracks</I> text box, accessed from the Browser 
Gateway page.
<P><B>Using BLAT for large batch jobs or commercial use</B><BR>
For large batch jobs or internal parameter changes, it is best to install
command line BLAT on your own Linux server. Sources and executables are free for
academic, personal, and non-profit purposes. BLAT source may be downloaded from 
<A HREF="http://www.soe.ucsc.edu/~kent/src" 
TARGET=_blank>http://www.soe.ucsc.edu/~kent</A> (look for the <em>blatSrc*.zip</em> file 
with the most recent date). For BLAT executables, go to 
<A HREF="http://www.soe.ucsc.edu/~kent/exe/" 
TARGET=_blank>http://www.soe.ucsc.edu/~kent/exe/</A>; binaries are sorted by 
platform. 
Non-exclusive commercial licenses are available from the 
<A HREF="http://www.genomeblat.com/genomeblat/index.asp" TARGET=_blank>Genome
BLAT</A> website. 
<P><B>BLAT documentation</B><BR>
For more information on the BLAT suite of programs, see the 
<A HREF="blatSpec.html">BLAT
Program Specifications</A> and the 
<A HREF="../../FAQ/FAQblat">Blat section</A> of the Genome 
Browser FAQ.
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="TableBrowser"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Getting started on the Table Browser  </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
The <A HREF="../../cgi-bin/hgTables" TARGET=_blank>Table
Browser</A> provides text-based access to the genome assemblies and annotation
data stored in the Genome Browser database. As a flexible alternative to the 
graphical-based Genome Browser, this tool offers an enhanced level of 
query support that includes restrictions based on field values, free-form 
SQL queries, and combined queries on multiple tables. Output can be filtered 
to restrict the fields and lines returned, and may be organized into one of 
several formats, including a simple tab-delimited file that can be loaded into 
a spreadsheet or database as well as advanced formats that may be uploaded into 
the Genome Browser as custom annotation tracks. 
The Table Browser provides a convenient
alternative to downloading and manipulating the entire genome and its massive data tracks.
(See the <A
HREF="#Download">Downloading Genome Data</A> section.) 
<P>
For information on using the Table Browser features, refer to the <A HREF="hgTablesHelp.html">Table Browser User Guide</A>.
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="TrackFormatDNA"></A><BR>	

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; DNA text formatting</b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
The Genome Browser provides a feature to configure the retrieval, formatting,
and coloring of the text used to depict the DNA sequence underlying the features in the displayed 
annotation tracks window. Retrieval options allow the user to add a padding of 
extra bases to the upstream or downstream end of the sequence. Formatting options range from simply displaying exons in upper case to 
elaborately marking up a sequence according to multiple track data. The DNA sequence covered by 
various tracks can be highlighted by case, underlining, bold or italic fonts, and color.
<P>
The DNA display configuration feature can be useful to highlight features within a
genomic sequence, point out overlaps between two types of features (for example, known
genes vs. gene predictions), or mask out unwanted features. 
<P><B>Using the DNA text formatting feature</B><BR>
To access the feature, click on the DNA link on the top blue menu
bar on the Genome Browser page. The Get DNA in Window page that appears contains 
sections for configuring the retrieval and output format. 
<P>
To display extra bases upstream of the 5' end of your sequence or downstream
of the 3' end of the sequence, enter the number of bases in the corresponding
text box. This option is useful in looking for regulatory regions.
<P>
The Sequence Formatting section lists several options for
adjusting the case of all or part of the DNA sequence. To choose one of these formats,
click the corresponding option button, then click the <I>get DNA</I> button. To access a table 
of extended formatting options, click the <I>Extended case/color options</I> button. 
<P>
The Extended DNA Case/Color page presents a table with many more format options. The page
provides instructions for using the formatting table, as well as examples of its use. The
list of tracks in the Track Name column is automatically generated from the list of tracks
available on the current genome.
<P><B>Tips for Use</B><BR>
A few caveats mentioned on the Extended DNA Case/Color page bear repeating. Keep the formatting simple at first: it is easy to 
make a display that is pretty to look at but is also completely cryptic. Also, be careful when 
requesting complex formatting for a large chromosomal region: when all the HTML tags have been 
added to the output page, the file size may exceed the size limits that your internet browser, 
clipboard, and other software can safely display. The maximum size of genome that can be 
formatted by the tool is approximately 10 Mbp.
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="Convert"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Converting data between assemblies</b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>Coordinates of features frequently
change from one assembly to the next as gaps are closed, strand orientations are
corrected, and duplications are reduced. Occasionally, a chunk of sequence may
be moved to an entirely different chromosome as the map is
refined. There are 3 different methods available for migrating data from one 
assembly to another: BLAT alignment, coordinate conversion, and lifting of
coordinates. The BLAT alignment tool is described in the section 
<A HREF="#BLATAlign">Using BLAT alignments</A>.
<P><B>Coordinate conversion</B><BR>
The Genome Browser Convert tool is useful for 
locating the position of a feature of interest in a newer release of a genome.
During the conversion process, portions of the genome in the coordinate range of
the original assembly are aligned to the new assembly while preserving their order and 
orientation. In general, it is easier to achieve successful conversions with shorter 
sequences. NOTE: At the present time, this tool may be used only on human
genome assemblies.
<P>
The conversion tool works by performing a BLAT search on the first 1000, last 1000, and middle
1000 bases
in the current window. If all three searches land uniquely in the same order on
the other version, the program announces a successful
conversion. If the search results are not so straightforward, the user is given
various options to find the corresponding sequence.
Frequently, if the feature the user is looking for is tied to an mRNA, it is
simplest just to BLAT the mRNA. 
<P><B>Using the convert tool</B><BR>
To access the conversion tool, click the Convert link in the top menu bar on the
Genome Browser page. On the page that displays, select the assembly
version in which the feature is located in the <I>Original Draft</I> list, then
pick the assembly version you'd like to convert to in the <I>New Draft</I> list.
Modify the position coordinates if necessary, then click the <I>submit</I>
button. Note that archived assembly versions are not accessible from the
conversion tool. Currently the tool supports only the human genome.
<P>
If the match is successful, the Genome Browser will announce success and display
the coordinate ranges for both the original and new assemblies. Clicking the
<I>browser</I> link to the right of the coordinate range will start up the
browser at the given position on the assembly. The Alignment Details section
shows which sequences of the original draft were aligned by BLAT to determine
the new set of coordinates. Note that the conversion is a best guess: it is
recommended that you check with local landmarks and use 
common sense when evaluating the results.
<P>
If the conversion is unsuccessful, the Genome Browser returns a failure message
and a possible explanation for the failure.

<A NAME="Liftover"></A><BR>

<P><B>Lifting coordinates</B><BR>
The liftOver tool is useful for converting a large number of coordinates to a 
different assembly. This command-line utility requires access to a Linux platform.
The executable file can be downloaded from 
<A HREF="http://www.soe.ucsc.edu/~kent/exe/linux/liftOver.gz">www.soe.ucsc.edu/~kent/exe/linux/liftOver.gz</A>.
LiftOver requires a UCSC-generated <em>over.chain</em> file as input. Pre-generated files 
for a given assembly are located in the liftOver subdirectory of the assembly's 
downloads directory. If the desired conversion file is not available, one can be obtained by sending a 
request to the <A HREF="mailto:genome@cse.ucsc.edu">genome mailing list</A>. 

		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="Download"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Downloading genome data</b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>Most of the underlying tables containing the genomic sequence and annotation data 
displayed in the Genome Browser can be downloaded. All of the tables are
freely usable for any purpose except as indicated in the README.txt file
in the download directories. This data was contributed by many researchers, as 
listed on the Genome Browser 
<A HREF="../credits.html" TARGET=_blank>Credits</A> page. Please
acknowledge the contributor(s) of the data you use. 
<P><B>Downloading the data</B><BR>
Genome data can be downloaded in two different ways:
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Via
	ftp:</B>The UCSC Genome Bioinformatics ftp site contains download
	directories for all genome versions currently accessible in the
	Genome Browser. The ftp command
	<tt>ftp://hgdownload.cse.ucsc.edu/goldenPath/</tt> will take you to a
	directory that contains the genome download directories. This 
	download method is recommended if you plan to download a large file
	or multiple files from a single directory. Use the <I>mget</I> command
	to download multiple files: mget <I>filename1 filename2</I>, or
	mget -a (to download all the files in the directory).</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Via the Downloads link:</B> Click the
	<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html"
	TARGET=_blank>Downloads</A> link on the left side bar on the UCSC 
	Genome Bioinformatics home page to display a list of all database directories 
	available for 
	download. If the data you wish to download pre-dates the assembly 
	versions listed, look in the archives accessible from the
	<A HREF="http://genome-archive.cse.ucsc.edu/"
	TARGET=_blank>Archive</A> link on the home page.
	</TD></TR>
</TABLE> 
<P><B>Types of data available</B><BR>
There may be several download directories associated with each version of a genome
assembly: the full data set <I>(bigZips)</I>, the full data set by 
chromosome <I>(chromosome)</I>, the annotation database tables
<I>(database)</I>, and one or more sets of comparative cross-species alignments. 
<P>
<I>BigZips</I> contains the entire draft of the 
genome in chromosome and/or contig form. Depending on the genome, this 
directory may contain some or all of the following files:<BR> 
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>chromAgp.zip:</B>
    	Description of how the assembly was generated, unpacking to one file
	per chromosome. 
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>chromFa.zip:</B>
	The assembly sequence chromosomes, in one file per chromosome. 
	Repeats from RepeatMasker and Tandem Repeats Finder are shown in 
	lower case; non-repeating sequence is in upper case.
	The main assembly is contained in the <em>chrN.fa</em> files, where 
	<em>chrN</em> is the name of the chromosome.  The <em>chrN_random.fa</em> 
	files contain clones that are not yet finished or cannot be placed
	with certainty at a specific place on the chromosome. In some
	cases, including the human HLA region on chromosome 6, the 
	<em>chrN_random.fa</em> files also contain haplotypes that differ from
	the main assembly.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>chromFaMasked.zip:</B>
	The assembly sequence chromosomes, in one file per chromosome. 
	Repeats are masked by capital Ns; non-repeating sequence is shown in upper
	case.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>chromOut.zip:</B>
	RepeatMasker <em>.out</em> file for chromosomes, generated by RepeatMasker
	at the <I>-s</I> sensitive setting.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>chromTrf.zip:</B>
	Tandem Repeats Finder locations, filtered to keep repeats with period 
	less than or equal to 12, translated into one <em>.bed</em> file per 
	chromosome.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>contigAgp.zip:</B>
	Description of how the assembly was generated from fragments at a contig 
	layout level. 
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>contigFa.zip:</B>
	The assembly sequence contigs, in one file per contig. All 
	contigs are in forward orientation relative to the chromosome. 
        In some cases, this means that contigs will be reversed 
	relative to their orientation in the NCBI assembly. Repeats are
	shown in lower case; non-repeating sequence is shown in upper case.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>contigFaMasked.zip:</B>
	The assembly sequence contigs, in one file per contig. 
	Repeats are masked by capital <em>N</em>s; non-repeating sequence is 
	shown in upper case.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>contigOut.zip:</B>
	RepeatMasker <em>.out</em> file for contigs, generated by RepeatMasker 
	at the <I>-s</I> sensitive setting.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>contigTrf.zip:</B>
	Tandem Repeats Finder locations, filtered to keep repeats with period 
	less than or equal to 12, and translated into one <em>.bed</em> file per 
	contig.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>database.zip:</B>
	The Genome Browser database as tab-delimited files and associated 
	MySQL table-creation tiles (eliminated in later assemblies due to
	size restrictions).
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>est.fa.zip:</B>
	Sequences of all GenBank ESTs for the selected species. 
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>liftAll.zip:</B>
	The offsets of contigs within chromosomes.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>mrna.zip:</B>
	mRNAs in GenBank from the selected species. 
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>refmrna.zip:</B>
	RefSeq mRNAs from the selected species.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>upstream1000.zip:</B>
	Sequences 1000 bases upstream of annotated transcription start of RefSeq 
	genes. This includes only cases where the transcription start is 
	annotated separately from the coding region start.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>upstream2000.zip:</B>
	Same as upstream1000, but with 2000 bases. 
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>upstream5000.zip:</B>
	Same as upstream1000, but with 5000 bases. 
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>xenoMrna.zip:</B>
	All GenBank mRNAs from species other than that of the selected one.
	</TD></TR> 
</TABLE> 
<P>
<I>Chromosomes</I> contains the assembled sequence for the 
genome in separate files for each chromosome in a zipped fasta format.
The main assembly can be found in the <em>chrN.fa</em> files, where <em>N</em> 
is the name of the chromosome.  The <em>chrN_random.fa</em> files contain clones
that are not yet finished or cannot be placed with certainty
at a specific place on the chromosome. In some cases, the <em>chrN_random.fa</em>
files also contain haplotypes that differ from the main assembly.
<P>
<I>Database</I> contains all of the 
positional and non-positional tables in the genome annotation database. Each 
table is represented by 2 files:
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>.sql file:</B>
	the MySQL commands used to create the table.
	</TD></TR>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>.txt.gz file:</B>
	the MySQL database table data in tab-delimited format and 
	compressed with gzip.
	</TD></TR>
</TABLE> 
<P>
The <A HREF="../gbdDescriptions.html"
TARGET=_blank>Description of the annotation database</A> link on the Downloads page 
points to a document containing a full
description of all the tables in the annotation database.
<P>
<I>Cross-species alignments</I> directories, such as the <I>vsMm4</I> 
and <em>humorMm3Rn3</em> directories in the hg16 assembly, contain pairwise
and multiple species alignments and filtered alignment files 
used to produce cross-species annotations. For more information, refer to the 
READMEs in these directories and the description of the
<A HREF="maf.html">Multiple Alignment Format</A> (MAF).
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="CustomTracks"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Creating custom annotation tracks </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
The Genome Browser provides dozens of aligned annotation tracks that have been
computed at UCSC or have been provided by outside collaborators. In addition to these 
standard tracks, it is also possible for users to upload their own annotations for
temporary display in the browser. These custom annotation tracks are viewable
only on the machine from which they were uploaded and are only
kept for 8 hours after the last time they were accessed. Optionally, users can 
make custom annotations viewable by others as well. 
<P>
Custom tracks are an important research feature of the Genome Browser. Because
space is limited in the Genome Browser track window, many excellent genome-wide tracks 
cannot be included in the set of tracks packaged with the Genome Browser. Other
tracks of interest may be excluded from distribution because the annotation
track data is too specific to be of general interest or can't be shared until
journal publication. To view a list of custom annotation tracks submitted by
Genome Browser users, click the <A HREF="/goldenPath/customTracks/custTracks.html">Custom 
Tracks</A> link on the Genome Browser home page.
<P>
Custom annotation tracks are similar to standard tracks, but never become part of the 
MySQL genome database. Each track has its own controller and persists even when not 
displayed in the Genome Browser window, e.g. if the position changes to a range
that no longer includes the track. Typically, custom annotation tracks are aligned 
under corresponding genomic sequence, but they can also be completely unrelated to 
the data. For example, a track can be displayed under a long sequence
consisting of millions of <em>N</em>s.
<P>
This section presents a summary of the steps involved in constructing and
displaying a custom annotation track. For a complete discussion of custom
annotation tracks, file formats, custom URLs, and a troubleshooting guide, refer
to the document <A HREF="customTrack.html" TARGET=_blank>Displaying Your Own Annotations in the Genome Browser</A>.
<P>
<B>Creating a custom annotation track</B><BR>
Genome Browser annotation tracks are based on files in line-oriented format.
Each line in the file defines a display characteristic for the annotation track
or defines a data item within the track.
Annotation files contain 3 types of lines: browser lines, track lines, and data
lines. Empty lines and lines starting with # in the
annotation file are ignored.
<P>
The easiest way to create a correctly formatted annotation track is by
collecting PSL output from BLAT. Advanced users can make custom tracks from the Table 
Browser and track-formatted DNA. To create a custom annotation track from scratch, 
it's best to begin with the examples and experiment with new lines and altered values 
in a spreadsheet or text editor. 
<P>
To construct an annotation file and display it in the Genome Browser, follow
these steps:

<P><B>Step 1. Format the data set</B><BR>
Formulate your data set as a tab-separated file using one of the formats supported by the Genome Browser. Annotation data can be in 
standard <A HREF="customTrack.html#GFF">GFF</A> format or in a format designed specifically for the Human Genome Project, including 
<A HREF="customTrack.html#GTF">GTF</A>, <A
HREF="customTrack.html#PSL">PSL</A>, or <A HREF="customTrack.html#BED">BED</A>. GFF and GTF files <I>must</I> be tab-delimited rather than
space-delimited to display correctly. You may include more than one data
set in your annotation file. However, all of the data lines for a given annotation track must be in the same format.

<P><B>Step 2. Define the Genome Browser display characteristics</B><BR>
Add one or more optional <A HREF="customTrack.html#BROWSER">browser lines</A> to the beginning of your formatted data file to configure the overall 
display of the Genome Browser when it initially displays your annotation data. Browser lines allow you to configure such things 
as the genome position that the browser will initially open to, the width of the display, and the configuration of the other
annotation tracks that are shown (or hidden) in the initial display. NOTE: If
the browser position is not explicitly set in the annotation file, the initial display will default
to the position setting most recently used by the user, which may not be an
appropriate position for viewing the annotation track.

<P><B>Step 3. Define the annotation track display characteristics</B><BR>
Following the browser lines - and immediately preceding the formatted data - add
a <A HREF="customTrack.html#TRACK">track line</A> to 
define the display attributes for your annotation data set. Track lines allow you to define annotation track characteristics 
such as the name, description, colors, initial display mode, use score,
etc. If you have included more than one data set in your annotation
file, insert a track line at the beginning of each new set of data.

<P><B><I>Example:</I></B><BR>
Here is an example of an annotation file that defines 2 separate
annotation tracks in BED format. The first track displays blue one-base
tick marks every 10000 bases at the beginning of chr 22. 
The second track displays red 100-base features alternating with 
blank space in the same region of chr 22.
<PRE><TT>browser position chr22:20100000-20140000
track name=spacer description="Blue ticks every 10000 bases" color=0,0,255,
chr22   20100000 20100001
chr22   20110000 20110001
chr22   20120000 20120001
track name=even description="Red ticks every 100 bases, skip 100" color=255,0,0
chr22   20100000 20100100	first
chr22   20100200 20100300	second
chr22   20100400 20100500	third </TT></PRE>

<P><B><I>Example:</I></B><BR>
This example shows an annotation file containing one data set in BED
format. The track displays paired features with a thick end and thin
end, and hatch marks indicating the direction of transcription. The track
labels display in green (0,128,0), and the gray level of the each feature
reflects the score value of that line. NOTE: The track line in this example
has been
split over 2 lines for documentation purposes. If you paste this example into
the browser, you must remove the line break to display the track successfully.
<PRE><TT>browser position chr22:1000-10000
browser hide all
track name=pairedReads description="Clone Paired Reads" visibility=2
color=0,128,0 useScore=1
chr22 1000 5000 cloneA 960 + 1000 5000 0 2 567,488, 0,4512
chr22 2000 6000 cloneB 200 - 2000 6000 0 2 433,399, 0,5601
</TT></PRE>

<P><B>Step 4. View your annotation track in the Genome Browser</B><BR>
To view your annotation data in the Genome Browser, open up the UCSC Genome
Bioinformatics home page <A HREF="../../index.html"
TARGET=_blank>(http://genome.ucsc.edu/)</a> and click on the Genome 
Browser link in the top menu bar. On the Genome Browser Gateway page that
displays, select the genome and assembly on which your
annotation data is based, then click the <em>add your own tracks</em> button. Upload your annotation
file by entering the name of your file in the <I>Annotation File</I> box or by pasting the contents of your file into the large edit
box. Scroll back to the top of the page and click the <I>submit</I> button to
display the Genome Browser track window with your annotation. If you
encounter difficulties in displaying your annotation, read the section <A
HREF="customTrack.html#TROUBLE">Troubleshooting Annotation Display Problems</A>.</P> 
<P>
To upload a custom annotation track from another machine or web site, paste the
URL of the track into the large edit box. Custom tracks can be displayed in
conjunction with ordinary BLAT tracks.</P>

<P><B>Step 5. (Optional) Add details pages for individual track features</B><BR>
After you've constructed your track and have successfully displayed it in the
Genome Browser, you may wish to customize the details pages for individual track
features.  The Genome Browser automatically creates a default details page for each
feature in the track containing the feature's name, position information, and a
link to the corresponding DNA sequence. To view the
details page for a feature in your custom annotation track (in full display
mode), click on the item's label in the annotation track window. 
<P>
You can add a link from a details page to an external web page containing
additional information about the feature by using the track line <I>url</I> attribute. 
In the annotation file, set the <I>url</I> attribute in the track line to point
to a publicly available page on a web server. The <I>url</I> attribute 
substitutes each occurrence of '$$' in the URL string with the name defined by the 
<I>name</I> attribute. You can take advantage of this feature to provide 
individualized information for each feature in your track by creating HTML anchors 
that correspond to the feature names in your web page.

<P><B><I>Example:</I></B><BR>
Here is an example of a file in which the <I>url</I> attribute has been set to
point to the file <A HREF="clones.html"
TARGET=_blank>http://genome.ucsc.edu/goldenPath/help/clones.html</A>. The '#$$'
appended to the end of the file name in the example points to the HTML NAME tag within the file that 
matches the name of the feature (cloneA, cloneB, etc.). 
NOTE: The track line in this example has been split over 3 lines for 
display purposes. If you paste this example into
the browser, you must remove the line breaks to display the track successfully.
<BR><BR>
<PRE><TT>browser position chr22:1000-10000
browser hide all
track name=clones description="Clones" visibility=2
color=0,128,0 useScore=1 
url="http://genome.ucsc.edu/goldenPath/help/clones.html#$$"
chr22 1000 5000 cloneA 960 
chr22 2000 6000 cloneB 200 
chr22 5000 9000 cloneC 700 
chr22 6000 10000 cloneD 600
chr22 11000 15000 cloneE 300
chr22 13000 17000 cloneF 100 
</TT></PRE>
<P>
<B>Sharing your annotation track with others</B><BR>
The previous steps showed you how to upload annotation data for your own use on your own machine. However, many users would like to 
share their annotation data with members of their research group on different machines 
or with colleagues at other sites.  To make your Genome Browser annotation track 
viewable by others, follow the steps below. (Note that 
some of the URL examples in this section have been broken up into 2
lines for documentation display purposes).<BR><BR>

<P><B>Step 1.</B>
Put your formatted annotation file on your web site. Be sure that the file permissions allow it to be read by others.</P>

<P><B>Step 2.</B>
Construct a URL that will link this annotation file to the Genome Browser. The URL must contain 3 pieces of information specific
to your annotation data: 
<UL>
<LI>The species or genome assembly on which your annotation data is based. 
To automatically display the most recent assembly for a given organism, set 
the <em>org</em> parameter: <em>e.g.</em> <tt>org=human</tt>. To specify a 
particular genome assembly for an organism, use the <em>db</em> parameter, 
<tt>db=database_name</tt>, where <I>database_name</I> is to the UCSC code for 
the genome assembly. For a list of these codes, see the Genome Browser <A
HREF="/FAQ/FAQreleases#release1" TARGET=_blank>FAQ</A>.
Examples of this include: db=hg16 (Human July 2003 assembly), db=mm6 
(Mouse Mar. 2005 assembly). 
<LI>The genome position that the Genome Browser should initially open to. This information is of the form
<tt>position=chr_position</tt>, where <I>chr_position</I> is a chromosome number, with or without a set of 
coordinates. Examples of this include: position=chr22, position=chr22:15916196-31832390.
<LI>The URL of the annotation file on your web site. This information is
of the form <tt>hgt.customText=URL</tt>, where
<I>URL</I> points to the annotation file on your website. An example of an
annotation file URL is <font color="000099"><A
HREF="test.bed"
TARGET=_blank>http://genome.ucsc.edu/goldenPath/help/test.bed</A></font>.  
</UL>

<P>Combine the above pieces of information into a URL of the following
format (the information specific to your annotation file is highlighted):
<BR><BR>
<tt>http://genome.ucsc.edu/cgi-bin/hgTracks?org=<font color="000099"><I>species_name</I></font>&amp;<BR>position=<font
color="000099"><I>chr_position</I></font>&amp;hgt.customText=<font color="000099"><I>URL</I></font></tt>

<P><B><I>Example:</I></B><BR>
The following URL will open up the Genome Browser window to display chr 22 of 
the latest human genome assembly and will display the annotation track pointed 
to by the URL
http://genome.ucsc.edu/goldenPath/help/test.bed:<BR><BR>

<font color="000099"><A HREF="../../cgi-bin/hgTracks?db=hg11&amp;position=chr22&amp;hgt.customText=http://genome.ucsc.edu/goldenPath/help/test.bed"
TARGET=_blank><tt>http://genome.ucsc.edu/cgi-bin/hgTracks?org=human&amp;position=chr22&amp;<br>hgt.customText=http://genome.ucsc.edu/goldenPath/help/test.bed</tt></A></font><BR><BR>

<P>If you'd like to share your annotation track with a broader audience, send the URL for your track - along with a description of 
the format, methods, and data used - to the UCSC Genome mailing list 
<A HREF="mailto:&#103;&#101;&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;sc.e&#100;&#117;">&#103;&#101;&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;sc.e&#100;&#117;</A>.
<!-- above address is genome at soe.ucsc.edu -->
</P>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	

	<A NAME="IndivTracks"></A><BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp; Annotation track descriptions</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
This section contains an alphabetical listing of the annotation tracks available
on various versions of the Genome Browser. For a more information on configuring
and using the tracks displayed in the Genome Browser track window, see the section 
<A HREF="#FineTuning">Fine-tuning the Genome Browser display</A>.<BR>

<P><B>Tips for viewing annotation track data</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	To display a description page with more information about the track,
	click on the mini-button to the left of a track.</TD></TR>   
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	To display a details page with additional information about a specific line 
	item within a track in full display mode, click on the item or its label. </TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	A track does not appear in the browser if its display mode is set to 
	<I>hide</I>. To restrict the browser's display to only those tracks in
	which you're interested, set the display mode of the unwanted tracks to
	<I>hide</I>.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	A track set to <I>full</I> display mode will default to a more tightly
	packed display mode if 
	the total number of lines in the track exceeds 250.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	To quickly toggle between <I>full</I> and <I>dense</I> or <I>pack</I> display modes, click on 
	the track's center label.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Only the most recent assemblies are fully active. Older assemblies are <A HREF="http://genome-archive.cse.ucsc.edu/" TARGET=_blank>archived</A>.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Not all tracks appear in all assemblies. Only a basic set of tracks 
	appears
	initially in a new assembly.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Track data can be viewed equivalently in columns as text tables using
	the <A HREF="#TableBrowser">Table Browser</A>. </TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<A HREF="../credits.html" TARGET=_blank>Credit</A>
	goes to many individuals and institutions for generously contributing
	the tracks. For specific information about the contributors to a given track,
	look at the track's description page.</TD></TR>  
</TABLE> 
<P>
<HR>
	      <CENTER>
	      <A HREF="#A_tracks">[A]</A>&nbsp;
	      <A HREF="#B_tracks">[B]</A>&nbsp;
	      <A HREF="#C_tracks">[C]</A>&nbsp;
	      <A HREF="#D_tracks">[D]</A>&nbsp;
	      <A HREF="#E_tracks">[E]</A>&nbsp;
	      <A HREF="#F_tracks">[F]</A>&nbsp;
	      <A HREF="#G_tracks">[G]</A>&nbsp;
	      <A HREF="#H_tracks">[H]</A>&nbsp;
	      <A HREF="#J_tracks">[J]</A>&nbsp;
	      <A HREF="#K_tracks">[K]</A>&nbsp;
	      <A HREF="#M_tracks">[M]</A>&nbsp;
	      <A HREF="#N_tracks">[N]</A>&nbsp;
	      <A HREF="#O_tracks">[O]</A>&nbsp;
	      <A HREF="#P_tracks">[P]</A>&nbsp;
	      <A HREF="#Q_tracks">[Q]</A>&nbsp;
	      <A HREF="#R_tracks">[R]</A>&nbsp;
	      <A HREF="#S_tracks">[S]</A>&nbsp;
	      <A HREF="#T_tracks">[T]</A>&nbsp;
	      <A HREF="#U_tracks">[U]</A>&nbsp;
	      <A HREF="#V_tracks">[V]</A>&nbsp;
	      <A HREF="#W_tracks">[W]</A>&nbsp;
	      <A HREF="#X_tracks">[X]</A>&nbsp;
	      <A HREF="#Y_tracks">[Y]</A>&nbsp;
	      <A HREF="#Z_tracks">[Z]</A>&nbsp;
	      </CENTER>

<P><B>2X Reg Potential</B><BR>
This track displays the 2-way regulatory potential (RP) score computed from 
alignments of the human and mouse assemblies. RP scores compare frequencies of 
short alignment patterns between regulatory elements and neutral DNA. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track uses the display conventions for wiggle tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>regPotential2X</I> positional table in the Table Browser. To view the table
	definition, click 
	<A HREF="../gbdDescriptions.html#WiggleTrack2"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE>

<A NAME="A_tracks"></A>
<P><B>3X Reg Potential</B><BR>
This track displays the 3-way regulatory potential (RP) score computed from 
alignments of the human, mouse, and rat assemblies. RP scores compare frequencies of 
short alignment patterns between regulatory elements and neutral DNA. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track uses the display conventions for wiggle tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>regPotential3X</I> positional table in the Table Browser. To view the table
	definition, click 
	<A HREF="../gbdDescriptions.html#WiggleTrack2"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE>

<P><B>Acembly Genes</B><BR>
This track shows gene models reconstructed solely from mRNA and EST evidence 
by Danielle and Jean Thierry-Mieg and Vahan Simonyan using the Acembly program. 
Acembly attempts to find the best alignment of each mRNA against the genome, 
and considers alternative splice models. If more than one gene model is 
produced that has statistical significance, all of these models are
displayed.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track uses the display conventions for gene prediction
	tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show 
	position information, links to translated protein, predicted mRNA, and genomic
	sequence, and a link to an extensive text and graphical annotation of 
	supporting data at NCBI's <A
	HREF="http://www.ncbi.nlm.nih.gov/IEB/Research/Acembly/index.html">AceView</A>.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>acembly</I> positional table in the Table Browser. To view the table
	definition, click 
	<A HREF="../gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE>

<P><B>Affy GNF1H</B><BR>
This track shows the location of the sequences used for the selection of probes 
on Affymetrix GNF1H chips. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows display conventions similar to PSL alignment tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	In dense or packed display mode, the track shows averages of related
	tissues. In full display mode, all tissues are displayed. Click on an 
	individual tissue entry to show a gene expression map for all tissues in 
	the experiment. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>affyGnf1h</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="../gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Affy GNF1M</B><BR>
This track shows the location of the sequences used for the selection of probes 
on Affymetrix GNF1M chips. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows display conventions similar to PSL alignment tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	In dense or packed display mode, the track shows averages of related
	tissues. In full display mode, all tissues are displayed. Click on an 
	individual tissue entry to show a gene expression map for all tissues in 
	the experiment. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>affyGnf1m</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="../gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Affy GNF1H</B><BR>
This track shows the location of the sequences used for the selection of probes 
on Affymetrix GNF1M chips. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows display conventions similar to PSL alignment tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	In dense or packed display mode, the track shows averages of related
	tissues. In full display mode, all tissues are displayed. Click on an 
	individual tissue entry to show a gene expression map for all tissues in 
	the experiment. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>affyGnf1m</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="../gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Affy MOE430</B><BR>
This track shows the location of the sequences used for 
the selection of probes on the Affymetrix Mouse MOE430 set of chips (A and B).

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows display conventions similar to PSL alignment tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in one of the fuller modes displays positional
	and % identity information. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>affyMOE430</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="../gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Affy RAE230</B><BR>
This track shows the location of the sequences used for the selection of probes 
on Affymetrix RAE230 chips. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows display conventions similar to PSL alignment tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	In dense or packed display mode, the track shows averages of related
	tissues. In full display mode, all tissues are displayed. Click on an 
	individual tissue entry to show a gene expression map for all tissues in 
	the experiment. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>affyRAE230</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="../gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Affy RG-U34A</B><BR>
This track shows the location of the sequences used for the selection of probes 
on Affymetrix RG-U34A chips. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows display conventions similar to PSL alignment tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	In dense or packed display mode, the track shows averages of related
	tissues. In full display mode, all tissues are displayed. Click on an 
	individual tissue entry to show a gene expression map for all tissues in 
	the experiment. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>affyU34A</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Affy U74</B><BR>
This track shows the location of sequences used for 
the selection of probes on the Affymetrix MG-U74v2 set of chips (A, B, and C).

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows display conventions similar to PSL alignment tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in one of the fuller modes displays positional
	and % identity information. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>affyU74</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Affy U95</B><BR>
This track shows the location of the consensus and exemplar sequences used for 
the selection of probes on the Affymetrix HG-U95Av2 chip.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows display conventions similar to PSL alignment tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in one of the fuller modes displays positional
	and % identity information. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>affyU95</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Affy U133</B><BR>
This track shows the location of the consensus and exemplar sequences used for 
the selection of probes on the Affymetrix HG-U133A and HG-U133B chips.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows display conventions similar to PSL alignment tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in one of the fuller modes displays positional
	and % identity information. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>affyU133</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>


<P><B>Affy U133Plus2</B><BR>
This track shows the location of the consensus and exemplar sequences used for 
the selection of probes on the Affymetrix HG-U133 Plus 2.0 chip.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows display conventions similar to PSL alignment tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in one of the fuller modes displays positional
	and % identity information. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>affyU133Plus2</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Affy Zebrafish</B> (Zebrafish)<BR>
This track shows the location of the sequences used for the selection of probes 
on the Affymetrix Zebrafish chip. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows display conventions similar to PSL alignment tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	In dense or packed display mode, the track shows averages of related
	tissues. In full display mode, all tissues are displayed. Click on an 
	individual tissue entry to show a gene expression map for all tissues in 
	the experiment. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>affyZebrafish</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>All Ciona ESTs</B><BR>
See the description for Human ESTs.

<P><B>Alt-Splicing</B><BR>
This track summarizes the alternative splicing shown in the mRNA and EST tracks. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	Exons are represented by blocks; possible splice junctions by lines. 
	In full mode with the resolution set to approximately gene-level, 
	alternative exons are laid out with minimal overlap and are therefore
	easier to view. The level of shading of exons and splice junctions 
	reflects the quantity of ESTs and mRNAs that contain that exon or splice 
	junction.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in one of the fuller modes displays positional 
	information and alt-splicing plots, as well as a link to a zoomed-in
	browser view of that feature.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>altGraphX</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#AltGraphX"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>A. gambiae Chain </B><BR>
This track shows <em>A. gambiae</em> genomic alignments to the selected genome using 
blastz and axtChain. See the Human Chain track description for more information.

<P><B>A. gambiae ESTs</B><BR>
See the description for Human ESTs.

<P><B>A. gambiae mRNAs</B><BR>
See the description for Human mRNAs.

<P><B>A. gambiae Net </B><BR>
This track shows the best <em>A. gambiae</em> chain for every part of the selected genome.
See the Human Net track description for more information.

<P><B>A. mellifera Chain </B><BR>
This track shows <em>A. mellifera</em> genomic alignments to the selected genome using 
blastz and axtChain. See the Human Chain track description for more information.

<P><B>A. mel. ESTs</B><BR>
<em>A. mellifera</em> ESTs. See the description for Human ESTs.

<P><B>A. mel. mRNAs</B><BR>
<em>A. mellifera</em> mRNAs. See the description for Human mRNAs.

<P><B>A. mellifera Net </B><BR>
This track shows the best <em>A. mellifera</em> chain for every part of the selected 
genome.  See the Human Net track description for more information.


<P><B>Anopheles Ecores</B><BR>
This track shows Anopheles evolutionary conserved regions (ecores) computed 
against the selected assembly genome by the Exofish program at Genoscope. For 
information on this track, see the Human Ecores track description.

<P><B>Assembly</B><BR>
This track shows the draft assembly of the human genome. This assembly merges
contigs from overlapping draft and finished clones into longer sequence 
contigs. The sequence contigs are ordered and oriented when possible by mRNA, 
EST, paired plasmid reads (from the SNP Consortium) and BAC end sequence pairs.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	In dense display mode, this track depicts in alternating gold and brown the 
	path through the draft and finished clones (aka the golden path) used to 
	create the assembled sequence. Individual boxes show the assembly extractable 
	from a single clone fragment. Where gaps exist in the path, spaces are shown 
	between the gold and brown blocks. If the relative order and orientation of 
	the contigs between the two blocks is known, a line is drawn to bridge the 
	blocks. The display must be sufficiently zoomed in to view the gap and bridge 
	features.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode will show information about the
	clone fragment and bases, position information, and the DNA sequence
	corresponding to the fragment.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>gold</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GpInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="B_tracks"></A>
<P><B>BAC End Pairs</B><BR>
This track shows the genome sequence of bacterial artificial chromosomes 
(BACs) in which a small amount of the both ends have been sequenced to place 
the full insert on the assembly. A valid pair of BAC end sequences must be 
separated by at least 50Kb but no more than 600Kb, and must have opposite 
orientations.  Strand assignment follows 
<a href="http://www.ncbi.nlm.nih.gov/genome/clone/">Clone Registry</a> 
storage conventions.  The BAC end sequences are placed on the assembled 
sequence using BLAT.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	The sequenced BAC ends are represented by black boxes connected by a
	line with hatch marks indicating the direction of transcription.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show hyperlinks to the
	NCBI clone registry and GenBank, position information, and genomic
	alignments. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>bacEndPairs</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#LFS"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Bactigs</B><BR>
The rat genome assembly is pieced together from bactig assemblies rather than clone-by-clone assemblies. Bactigs are reassemblies of the reads
from sets of overlapping skimmed BAC clones, including mapped whole shotgun
reads. The assembly process reworks the contigs (splitting some) and
selects them under more stringent criteria than is used for the BAC submissions.  

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Bactig positions are indicated by rectangular blocks. Bactig names are 
	of the form <I>gaab_gnmm</I>, the internal names of the two bracketing 
	projects. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show position information
	and the first and last contig in the bactig.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>bactigPos</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BactigPos"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Base Position</B><BR>
Shows the chromosomal coordinates as numbered starting from the telomere of 
the short p arm. The coordinates differ between assemblies.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Clicking on a position in this track will zoom in the display by 3x,
	centered around the click. This track has only on and off modes, and no 
	dense mode.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	None.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	None.</TD></TR>
</TABLE></P>

<P><B>BDGP Genes</B> (<em>D. melanogaster</em>)<BR>
This track shows protein-coding genes annotated by the Berkeley Drosophila 
Genome Project (<A HREF="http://www.fruitfly.org/" TARGET=_blank>BDGP</A>). 
For a description of the methods used to generate the 
data, see the BDGP <A HREF="http://www.fruitfly.org/about/methods/index.html" 
TARGET=_blank>Methods</A> web page.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for known genes and gene 
	predictions tracks.
	</TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display provides links to the 
	corresponding protein, mRNA, and genomic sequences, and related links to 
	outside databases. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter that can be used to turn
	on codon coloring for the track. Click the &quot;Codon coloring 
	help&quot; link on the page for more information about this feature.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>bdgpGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>BDGP Insertions</B> (<em>D. melanogaster</em>)<BR>
This track shows the locations of P transposable element insertions from 
P-Screen, the online database of the 
<A HREF="http://www.fruitfly.org/p_disrupt/index.html" TARGET=_blank>BDGP 
Gene Disruption Project</A>. When a stock order number is available 
for the strain, a link is provided to the Bloomington stock center. The 
project's strain library contains more than 7140 strains disrupting at least 
5362 different genes, corresponding to 39% of the 13,666 currently annotated 
Drosophila genes. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Triangular arrows on the reference sequence indicate the 
	approximate locations where P elements have inserted themselves
	in the mutant strain. Left-pointing arrows indicate insertions
	that are oriented in a backwards direction; right-pointing arrows 
	indicated forwards-oriented insertions.
	The item name indicates the strain having a P element disruption
	at that location.
	</TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode provides position and strand
	information, strain name, a link to the Bloomington stock center (when
	applicable), and a link to the underlying DNA for the feature.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page allows the user to set the minimum 
	unnormalized score of the data displayed in the track.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>pscreen</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#Pscreen"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>BDGP Non-Coding Genes</B> (<em>D. melanogaster</em>)<BR>
This track shows non-coding genes annotated by the Berkeley Drosophila 
Genome Project (<A HREF="http://www.fruitfly.org/" TARGET=_blank>BDGP</A>). 
For a description of the methods used to generate the 
data, see the BDGP <A HREF="http://www.fruitfly.org/about/methods/index.html" 
TARGET=_blank>Methods</A> web page.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for known genes and gene 
	predictions tracks.
	</TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display provides links to the 
	corresponding protein, mRNA, and genomic sequences, and related links to 
	outside databases. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter that can be used to turn
	on codon coloring for the track. Click the &quot;Codon coloring 
	help&quot; link on the page for more information about this feature.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>bdgpNonCoding</I> positional table in the Table Browser. To view 
	the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Best Human</B><BR>
This track displays <I>Blastz</I> alignments of the Nov. 2002 human draft
assembly to the mouse genome filtered to display only the best alignment for any
given region of the mouse genome.  See the track's description page for more 
details. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for cross-species synteny 
	tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show mouse and 
	human position information, alignment details, and a link to a detailed
	view of parts of the alignment.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter that can be used to turn 
	on the chromosome color track or to filter the display output by 
	chromosome.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>blastzBestHuman</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Best Mouse</B><BR>
This track displays <I>Blastz</I> alignments of the specified mouse draft
assembly to the human genome filtered to display only the best alignment for any
given region of the human genome.  See the track's description page for more 
details. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for cross-species synteny 
	tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show human and 
	mouse position information, alignment details, and a link to a detailed
	view of parts of the alignment.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter that can be used to turn 
	on the chromosome color track or to filter the display output by 
	chromosome.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>blastzBestMouse</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Best Rat</B><BR>
This track displays <I>Blastz</I> alignments of the Nov. 2002 rat draft
assembly to the mouse genome filtered to display only the best alignment for any
given region of the mouse genome.  See the track's description page for more 
details. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for cross-species synteny 
	tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show rat and 
	mouse position information, alignment details, and a link to a detailed
	view of parts of the alignment. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter that can be used to turn 
	on the chromosome color track or to filter the display output by 
	chromosome.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>blastzBestRat</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>BGI Coverage</B><BR>
This track shows areas of coverage for genomic regions of 3 alternate strains of 
chicken sequenced by the Beijing Genomics Institute (BGI): Broiler, Layer, and Silkie. 
Single reads from those three strains were mapped to the reference assembly for the Red 
Jungle Fowl (RJF). 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Coverage regions are represented by solid blocks.  </TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode shows position information and a link 
	to the DNA underlying this feature.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>bgiCov</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>BGI Genes</B><BR>
This track shows protein-coding genes annotated by the Beijing Genomics Institute (BGI). 
If Single Nucleotide Polymorphisms (SNPs) and/or insertions/deletions from BGI have been 
associated with a gene, those SNPs are listed with links to their details pages. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for known genes and gene 
	predictions tracks. </TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode shows position information, GO and 
	InterPro terms from BGI, BGI SNPs associated with the gene, and links to the
	predicted protein, mRNA, and genomic sequences. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>bgiGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>BGI SNPs</B><BR>
This track shows the single nucleotide polymorphisms (SNPs) and/or insertions/deletions 
(indels) determined by the Beijing Genomics Institute (BGI). Genomic sequence reads from 
3 alternate strains of chicken (Broiler, Layer and Silkie) were used to determine the SNPs 
relative to the reference assembly for the Red Jungle Fowl (RJF). SNPs and indels are 
associated with genes (annotations by BGI until Ensembl annotations are ready). 
Information on the association can be displayed along with a link to the details page 
for the gene. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	SNP locations are indicated by thin vertical lines. SNP names are of the format
	[supercontig].[contig].[position].[type].[strain]. See the track description
	page for more information.<TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking in full display mode shows information position and SNP information, 
	primer sequences, and a link to the DNA underlying the feature.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>bgiSnp</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BgiSnp"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Binomial Cons</B><BR>
This track displays the conservation scores resulting from an analysis of Zoo 
blastz alignments. Conservation scores were generated by weighing the relative 
contribution of each species sequence to the overall conservation score, based 
on how diverged each species was from the reference sequence. This weighting 
scheme was determined by calculating the neutral substitution
rates at 4-fold degenerate positions. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for wiggle tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains options to configure 
	several of the graph characteristics. To read more information
	about each option, click the &quot;Graph configuration help&quot; link. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on one of the peaks in the track when in full display mode will 
	show genome position information for the window and a link to the
	underlying DNA sequence. In full mode in 
	a zoomed-out display, detailed information may not be accessible for an 
	individual feature. In this case, zoom in and try again. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>binomialCons2</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#WiggleTrack"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Blastz Mouse</B><BR>
This track displays <EM>blastz</EM> alignments of the specified mouse draft
assembly to the selected genome.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for cross-species synteny 
	tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will position information 
	for the mouse and the selected genome, alignment details, and a link 
	to a detailed
	view of parts of the alignment.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter that can be used to turn 
	on the chromosome color track or to filter the display output by 
	chromosome.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>blastzX</I> positional table in the Table Browser, where X represents
	the assembly version in which you are interested, e.g. Mm4). To view the
	table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Briggsae Aligns </B><BR>
This track shows best-in-genome <em>C. elegans/C. briggsae</em> alignments 
based on the <em>C. elegans/C. briggsae</em> alignment net, i.e. the best 
chain for every part of the <em>C. elegans</em> genome. See the D. pseud Aligns 
track description for more information.

<P><B>Briggsae Blastz </B><BR>
This track displays Blastz alignments of the <em>C. briggsae</em> assembly to 
selected assembly. See the Blastz Mouse track description for more information.

<P><B>Briggsae Chain </B><BR>
This track shows <em>C. briggsae</em> genomic alignment to the selected genome 
using blastz and axtChain. See the Human Chain track description for more 
information.

<P><B>Briggsae Net </B><BR>
This track shows the best <em>C. briggsae</em> chain for every part of the 
selected genome. 
See the Human Net track description for more information.

<P><B>Briggsae Waba </B><BR>
This track shows alignments of <em>C. briggsae</em> to <em>C. elegans</em> 
using the Wobble Aware Bulk Aligner (WABA), an alignment tool developed by 
Jim Kent for doing large-scale alignments between genomic DNA of different 
species.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	Aligning regions are represented by blocks. The shading indicates the
	hidden Markov model (HMM) states of the bases.</TD></TR>	
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode will show alignment details
	and the HMM state for each alignment pair`. Markov model . 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>wabaCbr</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#WabaChromeCreate"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="C_tracks"></A>

<P><B>C. briggsae ESTs</B><BR>
See the description for Human ESTs.

<P><B>C. briggsae mRNAs</B><BR>
See the description for Human mRNAs.

<P><B>C. elegans ESTs</B><BR>
See the description for Human ESTs.

<P><B>C. elegans mRNAs</B><BR>
See the description for Human mRNAs.

<P><B>Chicken Chain </B><BR>
This track shows chicken genomic alignment to the selected genome using blastz 
and axtChain. See the Human Chain track description for more information.

<P><B>Chicken ESTs</B><BR>
See the description for Human ESTs.

<P><B>Chicken mRNAs</B><BR>
See the description for Human mRNAs.

<P><B>Chicken Net </B><BR>
This track shows the best chicken chain for every part of the selected genome. 
See the Human Net track description for more information.

<P><B>Chimp BAC Ends </B><BR>
This track displays BAC alignments of the chimp genome (Pan troglodytes) to human, based on BLAT alignments from Ingo Ebersberger, Joshua Bacher, and Svante Pbo at the <A HREF="http://www.eva.mpg.de" TARGET=_blank>MPI for Evolutionary Anthropology</A>. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows the display conventions for BLAT tracks.</TD></TR>	
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item will show details on the chimp sequence fragment and 
	alignment details, as well as a link to more information on the track data. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>chimpBac</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Chimp Blat </B><BR>
This track displays alignments of the chimp genome (Pan troglodytes) to human, based on Blat alignments from Ingo Ebersberger, Joshua Bacher, and Svante Pbo at the <A HREF="http://www.eva.mpg.de" TARGET=_blank>MPI for Evolutionary Anthropology</A>. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows the display conventions for BLAT tracks.</TD></TR>	
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item will show details on the chimp sequence fragment and 
	alignment details, as well as a link to more information on the track data. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>chimpBlat</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Chimp Chain </B><BR>
This track shows chimp genomic alignment to the selected genome using blastz and
axtChain. See the Human Chain track description for more information.

<P><B>Chimp Deletions</B><BR>
This track displays regions of the human assembly that are deleted in the 
chimp assembly. Only regions of between 80 and 12000 bases are included. 
The name of each deletion is a unique pointer to that deletion followed by an 
underscore and then its length. A similar track, showing human deletions in the 
chimp assembly, appears in the chimp Genome Browser.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Deletions are indicated by thin vertical lines. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on a deletion in full display mode provides the item name,
	position, genomic size, and a link to the DNA underlying the feature.
	</TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>chimpDels</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Chimp Diff</B><BR>
This track shows simple differences between chimp alignments and the human 
assembly within regions of high quality chimp sequence. For a difference to be 
included in this track, it had to meet the following criteria: 
<UL>
<LI>the difference must occur at a base of quality 30 or better 
<LI>all bases within an 11-base window around this base must have a quality of 
25 or better 
<LI>the 11-base window must contain no more than two base differences 
<LI>no insertions or deletions may be present within the window 
</UL>
Only reciprocal best chimp alignments were considered for this track (see the 
Chimp Net track description page for more information about this alignment 
strategy). 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Differences are indicated by thin vertical lines. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on a deletion in full display mode provides the item name,
	position, genomic size, and a link to the DNA underlying the feature.
	</TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>chimpSimpleDiff</I> positional table in the Table Browser. To view 
	the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#ChimpSimpleDiff"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Chimp ESTs</B><BR>
See the description for Human ESTs.

<P><B>Chimp mRNAs</B><BR>
See the description for Human mRNAs.

<P><B>Chimp Recip Chain </B><BR>
This track displays &quot;reciprocal best&quot; human/chimpanzee genomic 
alignment chains. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for chain tracks.</TD></TR>
	<B><I>Track details:</I></B> 
	In full display mode, clicking on a box shows details information about 
	the chain as a whole, while clicking on a line shows information
	about the gap. The detailed information is useful in determining the 
	cause of the gap or, for lower level chains, the genomic 
	rearrangement.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>rBestChainPanTro1</I> positional table in the Table Browser. 
	To view the table definition, click <A
	HREF="../gbdDescriptions.html#Chain"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Chimp Recip Net </B><BR>
This track shows the reciprocal best alignment net between the chimpanzee and the
selected genome assembly. Due to the draft nature of the
initial chimp assembly, this track uses a "reciprocal best" strategy that
attempts to minimize paralog fill-in for missing orthologous chimp sequence by
filtering out of the human net all sequences not in the chimp side of the net.
See the track description page for more information about the methods used to
generate this track.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for net tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	In full display mode, clicking on a box shows details information about 
	the chain as a whole, while clicking on a line shows information
	about the gap. The detailed information is useful in determining the 
	cause of the gap or, for lower level chains, the genomic 
	rearrangement.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>rBestNetPanTro1</I> positional table in the Table Browser, where <I>X</I>
	is the assembly version in which you're interested. To view the table
	definition, click <A
	HREF="../gbdDescriptions.html#NetAlign"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Chromosome Band</B><BR>
The light and dark blocks indicate the approximate location of
traditional cytological bands seen on Giemsa-stained chromosomes at an 850
band resolution. The bands were <A
HREF="http://www.ncbi.nlm.nih.gov:80/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11237021&dopt=Abstract">mapped</A>
by fluorescent in situ hybridization (FISH) of large-insert clones
with sequence tags, STS markers, and BAC end sequence locations.  
resolution of 862 bands genome-wide. The uncertainty of 
mapping to band staining is still a megabase or more.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	The light and dark blocks indicate the approximate location of the bands on the
	assembly sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on a band in full display mode will show more details about
	the track, a discussion of mapping methods used, and lab
	credits.</TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>cytoBand</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#ChromosomeBand"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B><em>C. intestinalis</em> ESTs</B><BR>
See the description for Human mRNAs.

<P><B><em>C. intestinalis</em> mRNAs</B><BR>
See the description for Human mRNAs.

<P><B>Ciona mRNAs</B><BR>
See the description for Human mRNAs.

<P><B>Conservation</B><BR>
This track shows a measure of evolutionary conservation among several species, 
based on a phylogenetic hidden Markov model (phylo-HMM). The multiz 
alignments of the various assemblies were used to construct the annotation.
The raw conservation scores used to generate this track can be downloaded
from the <em>[assembly_name]/phastCons</em> directory on the Genome Browser 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/" TARGET=_blank>FTP 
server</A>. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	In full display mode, the track displays pairwise alignments of several
	species aligned to the human genome. The pairwise alignments 
	are displayed in standard UCSC browser "dense" mode using a grey-scale 
	density gradient.  When zoomed-in to the base-display level, the track 
	shows the base composition of each alignment. The numbers and symbols on 
	the "human gap" line indicate the lengths of gaps in the human sequence 
	at those alignment positions. If the gap size is greater than 9, the 
	"+" symbol is displayed. The track must be zoomed-in to 30,000
	or fewer bases to access detailed information about the 
	alignments at a specific position.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show detailed alignment 
	information for each of the organisms. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> <BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	To view an index of the multiz multiple alignment data used to 
	generate
 	this track, select one of the following positional tables from the 
	Table Browser: <I>multiNway</I> (most recent assemblies), 
 	<I>mz*_pHMM</I> (hg16), <I>mz*_phast</I> (canFam1, dm1), 
        <I>c_briggsae_pwMaf</I> (<em>C. elegans</em>).<BR>
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	To view the wiggle track data used to generate the conservation 
	display, select onf of the following positional tables: <I>phastCons</I>
	(most assemblies), <I>mz*_wig</I> (hg16), <I>mz*phast_wig</I> 
	(canFam1, dm1).<BR>
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	To download the raw conservation scores displayed in the 
	Conservation track, download the data files from the 
	<em>[assembly_name]/phastCons</em> directory on the Genome Browser 
	<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/" TARGET=_blank>FTP 
	server</A>. <BR>
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	To view the table definition for the multiz multiple alignment
	tables, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#ScoredRef"
	TARGET=_blank>here</A>.<BR>
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	To view the table definition for the phastCons tables, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#WiggleTrack2"
	TARGET=_blank>here</A>.<BR>
	</TD></TR>
</TABLE>
</TABLE>

<P><B>Contamination</B> (Chicken)<BR>
This track shows locations of likely contamination in the chicken (galGal2) assembly, i.e. 
high-scoring alignments of <em>E.coli</em>, vector and maize mitochondrion sequence. The 
name of each feature indicates the type of contamination found there. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Area of contamination are represented by solid blocks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show the type of contamination,
	position and size information, and a link to display the DNA for the feature.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>contamination</I> positional table in the Table Browser. 
	To view the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#Contamination"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Contigs</B> (Zebrafish)<BR>
This track shows the whole genome shotgun (WGS) contigs of the zebrafish assembly from 
The Wellcome Trust Sanger Institute. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	Contigs are represented by blocks that indicate their location
	on the assembly sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode will show contig and positional 
	information.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>ctgPos2</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#ContigInfo2"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Coverage</B><BR>
Shows the coverage level and positions of all clones in the genome.
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	In dense display mode, this track shows the coverage level of the
	genome. Finished regions are shown in black. Draft regions are
	shown in various shades of gray that correspond to the level of
	coverage:<BR>

	<BLOCKQUOTE>
		o White: no coverage (gap) <BR>
		o Light Gray: predraft (less than 4x shotgun) <BR>
		o Medium Gray: draft (at least 4x shotgun) <BR>
		o Dark Gray: covered by multiple draft clones <BR>
		o Black: covered by a finished clone
	</BLOCKQUOTE>
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top"></TD><TD>
	In full display mode, this track shows the position of each clone that
	aligns to the genome sequence. Finished clones are shown in black and
	unfinished clones are shown in gray. In assemblies prior to the April 
	2003 human release,
	coverage for each unfinished clone is represented as a 
	two-tiered gray box.  The fragments inside the clone are shown as numbered 
	black boxes. Their positions correspond to their relative assembly placement, 
	including strand assignment. For some assemblies, clones in the sequencing 
	center tiling path are shown with blue rather than gray backgrounds. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode will show the clone hyperlink to 
	GenBank record and the fasta format sequence (one per fragment).</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>clonePos</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#CloneInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Cow Chain </B><BR>
This track shows cow genomic alignments to the selected genome using 
blastz and axtChain. See the Human Chain track description for more information.

<P><B>Cow ESTs</B><BR>
See the description for Human ESTs.

<P><B>Cow mRNAs</B><BR>
the description for Human mRNAs.

<P><B>Cow Net </B><BR>
This track shows the best cow chain for every part of the selected 
genome.  See the Human Net track description for more information.

<P><B>CpG Islands  </B><BR>
This track shows CpG islands, areas rich in the dinucleotide CG. A CpG island
is a region where C's followed by G's (CpGs) are present at significantly 
higher levels than is typical for the genome as a whole. The CpG islands 
displayed in this browser are all at least 200 bases long, and have a GC 
content of at least 50%.
<P>
CpG islands are associated with genes, particularly housekeeping genes, in 
vertebrates. They are particularly common near transcription start sites, and 
may be associated with promoter regions. Normally a CpG is rare in vertebrate 
DNA because the C's in such an arrangement tend to be methylated. This 
methylation helps distinguish the newly synthesized DNA strand from the parent 
strand, which aids in the final stages of DNA proofreading after duplication. 
However, over evolutionary time, methylated C's tend to turn into T's due to 
spontaneous deamination. The result is that CpGs are relatively rare unless 
there is selective pressure to keep them or a region is not methylated for 
some reason, perhaps having to do with the regulation of gene expression. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	CpG islands are represented by green boxes. Darker green items indicate CpG 
	islands that contain  300 or more bases.</TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Clicking on a CpG entry in full display mode will show basic track information, 
	detailed information about the CpG island, and sequence information. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>cpgIsland</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#CpGIslands"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>C. savignyi Blat</B><BR>
This track displays translated BLAT alignments of the <em>C. savignyi</em> genome 
against the <em>C. intestinalis</em> genome assembly. The alignments 
were done with Blat in translated protein mode
requiring 2 nearby 4-mer matches to trigger a detailed alignment. The 
<em>C. intestinalis</em> genome was masked with RepeatMasker and Tandem Repeats Finder 
before running Blat. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for BLAT tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode displays links to the DNA
	sequence and the alignment. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>blatCioSav1</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Cytokine Genes</B> (Tetraodon)<BR>
This track shows cytokine genes annotated by human experts using 
sequence similarity information between Tetraodon and human, mouse and 
zebrafish cytokine protein and nucleotide sequences. These annotations 
are available only on the following chromosomes: 1, 2, 3, 4, 7, 8, 10, 12, 13, 
15, 16, 1_random, 3_random, and Un_random. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track uses the display conventions for gene prediction tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a control that can be used to turn 
	on codon coloring.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on a track feature in full display mode shows position 
	information and provides links to the translated protein, predicted mRNA,
	and genome sequences.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>cytokines</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="D_tracks"></A>

<P><B>DBTSS mRNA</B><BR>
This track shows RefSeq mRNA sequences extended to the 5' end, based on sequence
information in the <A HREF="http://dbtss.hgc.jp/" TARGET=_blank>DataBase of 
Transcriptional Start Sites</A> (DBTSS). For more information on the process used
to construct this track, refer to the track's description page. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for PSL alignment tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show PSL output information
	about the item.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>dbtssAli</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>D. mel. Chain</B><BR>
This track shows <em>D. melanogaster</em> genomic alignment to the selected genome using 
blastz and axtChain. See the Human Chain track description for more information.

<P><B>D. mel. Net</B><BR>
This track shows the best <em>D. melanogaster</em> chain for every part of the selected 
genome. See the Human Net track description for more information.

<P><B>D. mel. Proteins</B> (<em>A. gambiae</em>, <em>A. mellifera</em><BR>
This track contains tBLASTn alignments of the peptides from the predicted and 
known genes identified in the specified version of FlyBase. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for PSL alignment tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will provide positional
	information, a link to the related FlyBase record, protein length, and
	links to the alignment and peptide sequences.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>blastDm1FB</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>D. melanogaster ESTs</B><BR>
See the description for Human ESTs.

<P><B>D. melanogaster mRNAs</B><BR>
See the description for Human mRNAs.

<P><B>DNAse HyperSens</B> (Zoo)<BR>
This track depicts DNAse I hypersensitive sites (DHS) mapped to the human genome 
sequence. Each track element represents the surrounding 500 bp from the position 
of the estimated DHS, as determined from Smith et al. (2000) &quot;Multiple 
Potential Intragenic Regulatory Elements in the CFTR Gene&quot;, <em>Genomics</em>
64(1):90-6.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Each DHS is depicted by a block representing the 500 bp surrounding the
	position of the estimated DHS. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show position information
	and a link to the underlying DNA sequence. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>dhs</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Dog Chain </B><BR>
This track shows dog genomic alignment to the selected genome using blastz 
and axtChain. See the Human Chain track description for more information.

<P><B>Dog ESTs</B><BR>
See the description for Human ESTs.

<P><B>Dog mRNAs</B><BR>
See the description for Human mRNAs.

<P><B>Dog Net </B><BR>
This track shows the best dog chain for every part of the selected genome. 
See the Human Net track description for more information.

<P><B>D. ananassae Chain </B><BR>
This track shows <em>D. ananassae</em> genomic alignments to the selected genome using 
blastz and axtChain. See the Human Chain track description for more information.

<P><B>D. anan. mRNAs</B><BR>
<em>D. ananassae</em> mRNAs. See the description for Human mRNAs.

<P><B>D. ananassae Net </B><BR>
This track shows the best <em>D. ananassae</em> chain for every part of the selected 
genome.  See the Human Net track description for more information.

<P><B>D. mojavensis Chain </B><BR>
This track shows <em>D. mojavensis</em> genomic alignments to the selected genome using 
blastz and axtChain. See the Human Chain track description for more information.

<P><B>D. moj. mRNAs</B><BR>
<em>D. mojavensis</em> mRNAs. See the description for Human mRNAs.

<P><B>D. mojavensis Net </B><BR>
This track shows the best <em>D. mojavensis</em> chain for every part of the selected 
genome.  See the Human Net track description for more information.

<P><B>D. pseudo. Aligns </B> (<em>D. melanogaster</em>)<BR>
This track shows blastz scores from the 
<em>D. pseudoobscura/D. melanogaster</em> alignment Net track, i.e. the best 
<em>D. pseudoobscura/D. melanogaster</em> chain for every part of the 
<em>D. melanogaster</em> genome. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	In dense display mode, darker shading indicates those best chained and
	netted alignments that have the higher scores. In full display mode, 
	the height of the vertical line corresponds to the score.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show position information
	and -- when zoomed in to the base level -- the base composition of the 
	alignments. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>axtNetX</I> positional table in the Table Browser, where <em>X</em>
	represents the assembly that is being aligned, e.g. Dp1. To view the 
	table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#ScoredRef"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>D. pseudo. Chain </B> (<em>D. melanogaster</em>)<BR>
This track shows <em>D. pseudoobscura</em> genomic alignment to the selected 
genome using blastz and axtChain. See the Human Chain track description for 
more information.

<P><B>D. pseudo. ESTs</B><BR>
<em>D. pseudoobscura</em> ESTs. See the description for Human ESTs.

<P><B>D. pseudo. mRNAs</B><BR>
<em>D. pseudoobscura</em> mRNAs. See the description for Human mRNAs.

<P><B>D. pseudo. Net </B> (<em>D. melanogaster</em>)<BR>
This track shows the best <em>D. pseudoobscura</em> chain for every part of 
the selected genome. See the Human Net track description for more information.

<P><B>D. virilis Chain </B><BR>
This track shows <em>D. virilis</em> genomic alignments to the selected genome using 
blastz and axtChain. See the Human Chain track description for more information.

<P><B>D. vir. mRNAs</B><BR>
<em>D. virilis</em> mRNAs. See the description for Human mRNAs.

<P><B>D. virilis Net </B><BR>
This track shows the best <em>D. virilis</em> chain for every part of the selected 
genome.  See the Human Net track description for more information.

<P><B>D. yakuba Chain </B><BR>
This track shows <em>D. yakuba</em> genomic alignments to the selected genome using 
blastz and axtChain. See the Human Chain track description for more information.

<P><B>D. yakuba ESTs</B><BR>
<em>D. yakuba</em> ESTs. See the description for Human ESTs.

<P><B>D. yakuba mRNAs</B><BR>
<em>D. yakuba</em> mRNAs. See the description for Human mRNAs.

<P><B>D. yakuba Net </B><BR>
This track shows the best <em>D. yakuba</em> chain for every part of the selected 
genome.  See the Human Net track description for more information.


<P><B>Duplications </B><BR>
This track shows genomic 
<a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11381028&dopt=Abstract">
duplications</a> greater than 1,000 bp if not overly repeat-masked. Duplications are 
more common in unordered and unassigned contigs (i.e., difficult to assign to their 
proper position). 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Duplications are show as red if 99%+ identity (likely unmerged assembly 
	overlap artifact), as yellow for 98-99% identity (origin uncertain), and as 
	gray for 90-98% identity (below draft sequencing error, often genuine 
	pericentromeric or subtelomeric duplication). </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	In the left panel of the browser, duplications are named by the chromosome 
	and start of the other member of the duplication pair. Clicking on an 
	individual duplication gives details of both locations and match quality 
	parameters. </TD></TR>
</TABLE></P>

<P><B>D. yakuba ESTs</B><BR>
See the description for Human ESTs.

<P><B>D. yakuba mRNAs</B><BR>
See the description for Human mRNAs.

<A NAME="E_tracks"></A>

<P><B>ECgene Genes</B><BR>
This track shows gene predictions generated by combining genome-based EST 
clustering and transcript assembly methods. The EST clustering is based on 
genomic alignment of mRNA and ESTs similar to that of NCBI's UniGene for the 
human genome. The transcript assembly procedure yields gene models for each 
cluster that include alternative splicing variants. This algorithm was developed 
by Prof. Sanghyuk Lee's Lab of Bioinformatics at Ewha Womans University in 
Seoul, Korea. For more information, see the 
<A HREF="http://genome.ewha.ac.kr/ECgene" TARGET=_blank>ECgene</A> web site. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode brings up basic track
	information, links to the predicted mRNA, genomic sequence, and 
	comparative sequence. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>ECgene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Elegans Blastz</B><BR>
This track displays <I>Blastz</I> alignments of the <em>C. elegans</em> assembly
to the <em>C. briggsae</em> assembly 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for cross-species synteny 
	tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show human and 
	mouse position information, alignment details, and a link to a detailed
	view of parts of the alignment.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter that can be used to turn 
	on the chromosome color track or to filter the display output by 
	chromosome.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>blastzX</I> positional table in the Table Browser, where <em>X</em>
	represents the <em>C. elegans</em> assembly, e.g. Ce2. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Elegans Chain</B><BR>
This track shows <em>C. elegans</em> genomic alignment to the selected 
genome using blastz and axtChain. See the Human Chain track description for 
more information.

<P><B>ENCODE Regions</B><BR>
This track depicts target regions for the NHGRI ENCODE project, which is 
described in <A HREF="http://www.genome.gov/10005107" TARGET=_blank>The 
ENCODE Project: ENCyclopedia of DNA Elements</A>. The long-term goal of this 
project is to identify all functional elements in the human genome sequence to 
facilitate a better understanding of human biology and disease. 
During the pilot phase, 44 regions comprising 30 Mb -- approximately 1% of 
the human genome -- have been selected for intensive study to identify, locate 
and analyze functional elements within the regions. These targets are being 
studied by a diverse public research consortium to test and evaluate the 
efficacy of various methods, technologies, and strategies for locating genomic 
features. The outcome of this initial phase will form the basis for a 
larger-scale effort to analyze the entire human genome. 
See the track description page for more information.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	ENCODE regions are represented by blocks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode shows position information
	 and provides a link to the DNA sequence associated with this feature.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>encodeRegions</I> positional table in the Table Browser, where <I>X</I>
	is the assembly version in which you're interested. To view the table
	definition, click <A
	HREF="../gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Ensembl ESTs</B><BR>
This track shows gene predictions from <A
HREF="http://www.ensembl.org">Project Ensembl</A> based on ESTs. For a description of
the methods used in Ensembl gene prediction, refer to
<A HREF="http://nar.oupjournals.org/cgi/content/full/30/1/38" TARGET=_blank>The Ensembl genome database project</A>, Nucleic Acids Research, 2002, 30(1) 38-41.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode brings up basic track
	information, links to the translated protein, predicted mRNA, and
	genomic sequences, plus a link to detailed supporting evidence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>ensEst</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Ensembl Genes</B><BR>
This track shows gene predictions from <A
HREF="http://www.ensembl.org">Project Ensembl</A>. For a description of
the methods used in Ensembl gene prediction, refer to
<A HREF="http://nar.oupjournals.org/cgi/content/full/30/1/38" TARGET=_blank>The Ensembl genome database project</A>, Nucleic Acids Research, 2002, 30(1) 38-41.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode brings up basic track
	information, links to the translated protein, predicted mRNA, and
	genomic sequences, plus a link to detailed supporting evidence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>ensGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Exofish ecores</B><BR>
The track displays conserved regions, generally protein-coding homologs, 
from a genomics project of long random reads in the 
<A HREF="http://www.genoscope.cns.fr/exofish" TARGET=_blank>pufferfish</A>, Tetraodon 
nigroviridis.  Ecore is an acronym for <B>e</B>volutionary <B>co</B>nserved 
<b>re</b>gion. The data was 
<A HREF="http://www.ncbi.nlm.nih.gov:80/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=10835645&dopt=Abstract" TARGET=_blank>
used</A> to estimate the number of human genes. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Regions of greater homology are displayed in a darker blue. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full mode brings up information on a particular 
	ecore, access the relevant Genoscope page, and the relevant DNA.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>exoFish</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#ExoFish"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Exoniphy</B><BR>
The exoniphy program identifies evolutionarily conserved protein-coding exons in multiple
sequences aligned with multiz using a phylogenetic hidden Markov model (phylo-HMM), a 
statistical model that simultaneously describes exon structure and exon evolution. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Evoluntionarily-conserved protein-coding exons are represented by blocks.
	Arrows indicated the direction of transcription.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full mode displays positional information and related 
	sequence links. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>exoniphy</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredExt"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="F_tracks"></A>

<!--commented out until release
<P><B>FEEP</B> (<em>D. melanogaster)</em><BR>
The full euchromatic expression profile (FEEP) track shows RNA expression of 
protein-coding and noncoding sequences at each major life cycle stage, as 
described in Stolc, V. <em>et al</em>., 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=15499012"
TARGET=_blank>A gene expression map for the euchromatic genome of Drosophila 
melanogaster</A>, <em>Science</em> <B>306</B>(5696), 655-60 (2004). Probe 
sequences were mapped to Release 4 genomic sequence at UCSC using BLAT. For
more information about the methods and probes used to produce the data
displayed in this track, see the track description page.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Mapped probe sequences are represented by blocks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	In dense display mode, the track shows the genomic locations of 
	mapped probe sequences. In the fuller display modes, the track shows
	the medians of all experiments for each life cycle stage. Clicking on 
	an entry displays a full euchromatic expression profile.</TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track decription page provides an option to select which of the
	the three probe sets should be displayed (all probes, probes expressed
	above background, probes differentially expressed by ANOVA).</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>flyFeepComposite</I> positional table in the Table Browser. To 
	view the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#FlyFeepComposite"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>
-->

<P><B>Fgenesh++ Genes</B><BR>
This tracks shows Fgenesh++ gene predictions based on <A
HREF="http://www.softberry.com">Softberry Inc.'s</A> gene-finding
software. Fgenesh++ uses both hidden Markov models (HMMs) and protein 
similarity to find genes in a completely automated manner, with less
emphasis on cDNA/EST data. See the paper 
<A HREF="http://www.genome.org/cgi/content/full/10/4/516" TARGET=_blank>"Ab 
initio gene finding in Drosophila genomic DNA"</A>, Genome Research 10(4) 516-522 
for more information. 
<P>
NOTE: The data in the softberryGene.txt and 
softberryPep.txt tables is free for academic and nonprofit use. Commercial users 
should contact <A HREF="http://www.softberry.com" TARGET=_blank>Softberry Inc.</A>.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode brings up basic track
	information, links to the translated protein, predicted mRNA, and
	genomic sequences, plus a link to the best matching protein at GenBank.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version data set, select the 
	<I>softberryGene</I> positional table in the Table Browser. 
	To view the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>First EF</B><BR>
This track displays predictions from First Exon Finder (FirstEF), a 5' terminal 
exon and promoter prediction program.
Three types of predictions are shown: exon, promoter and CpG window. If two 
consecutive predictions are separated by less than 1000 bp, FirstEF treats them 
as one cluster of alternative first exons that may belong to same gene. 
Each predicted exon is either CpG-related or non-CpG-related, based on a score 
of the frequency of CpG dinucleotides. An exon is classified as CpG-related if 
the CpG score is greater than a threshhold value and non-CpG-related if 
less than the threshold. If an exon is CpG-related, its associated CpG-window is displayed. 


<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for BLAT tracks. Higher CpG 
	scores are shown in darker shades of gray/black. The cluster 
	number is displayed in parentheses in the item label. For example, 
	&quot;exon(405-)&quot; represents the exon prediction in cluster number 
	405 on the minus strand. The exon, promoter and CpG-window are 
	interconnected by this cluster number. Alternative predictions within 
	the same cluster are denoted by &quot;#N&quot; where &quot;N&quot; is the 
	serial number of an alternative prediction in the cluster.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode displays probability, size, 
	and position information, and a link to the DNA sequence. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>firstEF</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Fish Blat </B><BR>
This track displays translated alignments of 728 million bases
of Tetraodon nigroviridis whole genome shotgun reads vs. the
draft human genome. Areas painted by this track are quite likely to be coding
regions. The alignments were done with BLAT in translated protein mode
requiring 2 nearby 4-mer matches to trigger a detailed alignment. The human 
genome was masked with RepeatMasker and Tandem Repeats Finder before running 
BLAT. The Tetraodon sequence was provided by <A
HREF="http://www.genoscope.cns.fr" TARGET=_blank">Genoscope</A>. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for BLAT tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode brings up links to the DNA
	sequence and the alignment. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>blatFish</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>FISH Clones</B><BR>
Shows the location of fluorescent in situ hybridization (FISH) mapped clones along 
the draft assembly sequence. 
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	The green blocks depict the clone placements on the assembly sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B>
	Clicking on a specific marker in full display mode will show information on 
	clone placement, clone registry, its STS markers, and BAC end sequences.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>fishClones</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#FishClones"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>FlyBase Genes</B> (<em>D. melanogaster</em>)<BR>
This track shows protein-coding genes annotated by FlyBase (version 3.2). 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display provides links to the 
	corresponding protein, mRNA, and genomic sequences, microarray expression data, 
	protein structure information, homology information and related links to 
	several tools and databases, including FlyBase. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>bdgpGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredExt"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>FlyBase Non-coding</B> (<em>D. melanogaster)</em><BR>
This track shows non-coding genes annotated by FlyBase (version 3.2). 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display provides links to FlyBase and BDGP,
	basic track information, and links to the corresponding predicted genomic and
	predicted mRNA sequences.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>bdgpGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredExt"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>FlyReg</B> (<em>D. melanogaster)</em><BR>
This track shows DNase I Footprint data from the
<A HREF="http://www.flyreg.org/" TARGET=_blank>FlyReg</A> version 2.0 
database. FlyReg provides access to results of the systematic curation and 
genome annotation of 1,350 DNase I footprints for the fruitfly 
<em>D. melanogaster</em> reported in Bergman, C.M. <em>et al</em>., 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=15572468" 
TARGET=_blank>Drosophila DNase I footprint database: a systematic genome 
annotation of transcription factor binding sites in the fruitfly, D. 
melanogaster</A>, <em>Bioinformatics</em> 2004 Nov. 30 [Epub ahead of 
print].

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Footprint regions are represented by blocks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display provides footprint information,
	links to external databases, and a link to view the related genomic
	DNA. When available, a footprint motif is also displayed, based on a 
	MEME matrix computed by Dan Pollard on the set of footprints for 
	the factor. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>flyreg2</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#FlyReg2"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Fosmid End Pairs</B><BR>
A valid pair of fosmid end sequences must be at least 32Kb but no more than 47Kb 
away from each other. The orientation of the first fosmid end sequence must be 
"+" and the orientation of the second fosmid end sequence must be "-". End 
sequences were trimmed at the NCBI using ssahaCLIP written by Jim Mullikan. 
The trimmed end sequences were placed on the assembled sequence using Jim 
Kent's BLAT program. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for PSL alignment tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B>
	Clicking on an entry in full display mode will show position information 
	and provide links to genomic alignments of the Fosmid ends.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>fosEndPairs</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="../gbdDescriptions.html#FosEndPairs"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Fugu Blat </B><BR>
This track displays translated BLAT alignments of Takifugu rubripes whole genome 
shotgun reads against the currently selected genome assembly. The alignments 
were done with BLAT in translated protein mode
requiring 2 nearby 4-mer matches to trigger a detailed alignment. The human 
genome was masked with RepeatMasker and Tandem Repeats Finder before running 
BLAT. The fugu sequence used in this track is version 3.0, provided by the
<A HREF="http://genome.jgi-psf.org/" TARGET=_blank>DOE Joint Genome Institute</A> (JGI).

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for BLAT tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode brings up links to the DNA
	sequence and the alignment. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>blatFugu</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Fugu Chain </B><BR>
This track shows Fugu genomic alignment to the selected genome using blastz and
axtChain. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for chain tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode shows position information
	in the Fugu and the selected genome, the chain ID and score, and provides 
	links to view details of parts of the chain or to open the Fugu browser 
	to the corresponding position. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>chainFrX</I> positional table in the Table Browser, where <I>X</I>
	is the assembly version in which you're interested. To view the table
	definition, click <A
	HREF="../gbdDescriptions.html#Chain"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Fugu Ecores</B><BR>
This track shows Fugu evolutionary conserved regions (ecores) computed against 
the selected assembly genome by the Exofish program at Genoscope. For 
information on this track, see the Human Ecores track description.

<P><B>Fugu ESTs</B><BR>
See the description for Human ESTs.

<P><B>Fugu mRNAs</B><BR>
See the description for Human mRNAs.

<P><B>Fugu Net </B><BR>
This track shows the best Fugu chain for every part of the selected genome.
See the Human Net track description for more information.

<A NAME="G_tracks"></A>

<P><B>Gap</B><BR>
This track depicts gaps in the assembly. These gaps - with the
exception of intractable heterochromatic gaps - are closed
during the finishing process.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Gaps are represented as black boxes. If the relative order and 
	orientation of the contigs on either side of the gap is known from mRNA, 
	ESTs, or paired BAC end reads, it is a "bridged" gap and a white line is 
	drawn through the black box representing the gap. The display must be 
	sufficiently zoomed in to view these features: at a coarser scale, the 
	gaps can appear to coalesce in the the graphic. In full display mode, 
	the gap label indicates the type
	of gap (see below) and whether the gap is bridged.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top"></TD><TD>
	Strand misorientation errors can occur adjacent to unbridged gaps.
	Unbridged gaps are assigned somewhat arbitrary sequence sizes (runs of 
	N's in genomic DNA); uncertainty is less with bridged gaps. A gap within 
	contig fragments of a clone is typically smaller than a gap between 
	draft clones. </TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode will show the gap type,
	bridging information and position information about the gap. Unfinished 
	clones may have gaps of uncertain size between fragments.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>gap</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GapInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>GC Percent</B><BR>
This shows the percentage of bases that are G or C in a 5-base window. High GC 
content is associated with gene rich areas. 
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track uses the display conventions for wiggle tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains options to configure 
	several of the graph characteristics. To read more information
	about each option, click the &quot;Graph configuration help&quot; link. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on one of the peaks in the track when in full display mode will 
	show position information and provide a link to the underlying DNA
	sequence.  In full mode in a zoomed-out display, detailed information 
	may not be accessible for an individual feature. In this case, zoom in 
	and try again. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>gc5Base</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#WiggleTrack2"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>GC Samples </B>(Zoo)<BR>
This track shows the percentage of bases that are G or C in small regions across 
the genome. Windows with high GC content are plotted at higher vertical 
positions than windows with low GC content. High GC content is associated with 
gene-rich areas.
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track uses the display conventions for wiggle tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter that can be used to turn 
	on the chromosome color track or to filter the display output by 
	chromosome.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on one of the peaks in the track when in full display mode will 
	show position information and provide a link to the underlying DNA
	sequence.  In full mode in 
	a zoomed-out display, detailed information may not be accessible for an 
	individual feature. In this case, zoom in and try again. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>pGC</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#WiggleTrack"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Gene Annotations </B>(Zoo)<BR>
This track displays NISC-generated, curated gene annotations for each of the 
zoo species and human. Exon locations of known genes were determined by aligning 
human and/or mouse reference cDNA sequences with each species' assembled genomic 
sequence. Any splice-site consensus, exon structure, or protein translation 
errors encountered were manually inspected and corrected.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track uses the display conventions for gene prediction tracks. As with other
	tracks in this category, it may be necessary to zoom in to 
	display entries in full display mode due to the 300 line display 
	limit.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full mode provides position, size, and score 
	information, and a link to dsplay the DNA sequence associated with this 
	feature.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>pjt_gene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Gene Bounds</B><BR>
This track shows the boundaries of genes and the direction of transcription as 
deduced from clustering spliced ESTs and mRNAs against the genome. When there 
are many spliced variants of the same gene, this track shows the variant that 
spans the greatest distance in the genome. 
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track uses the display conventions for EST tracks. As with other
	tracks in this category, it may be necessary to zoom in to 
	display entries in full display mode due to the 300 line display 
	limit.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full mode provides position, size, and score 
	information, and a link to dsplay the DNA sequence associated with this 
	feature.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>rnaCluster</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#rnaCluster"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Geneid Gene Predictions</B><BR>
This track shows gene predictions from the geneid program, which is
being developed at the Grup de Recerca en Informatica Biomedica at IMIM
in Barcelona. Geneid uses information from sequence signals involved in
genes specification, as well as coding coding statistics, to define exons
and genes. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode brings up basic track
	information, plus links to the translated protein, predicted mRNA, and
	genomic sequences.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>geneid</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>GenMapDB Clones</B><BR>
This track shows BAC clones from the 
<A HREF="http://genomics.med.upenn.edu/genmapdb/" TARGET=_blank>GenMapDB</A> 
database placed on the draft sequence using BAC end sequence information and 
confirmed using STS markers by Vivian Cheung's lab at the Department of 
Pediatrics, University of Pennsylvania. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	The clone positions are shown in black. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode brings up basic position
	information and additional information about the clones and STS markers. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>genMapDb</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#genMapDb"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Genoscope GAZE</B> (Tetraodon)<BR>
This track shows Genoscope gene model annotations from the GAZE program 
written by Kevin Howe. These annotaions were computed using GAZE with a i
custom-designed gene model. GAZE integrates information from Geneid, 
Genscan, Exofish (Human, Mouse and Fugu), Genewise (Human and Mouse) and 
Tetraodon cDNAs.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track uses the display conventions for gene prediction tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a control that can be used to turn 
	on codon coloring.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on a track feature in full display mode shows position 
	information and provides links to the translated protein, predicted mRNA,
	and genome sequences.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>gaze</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Genscan Genes</B><BR>
This track shows gene predictions from <A HREF="http://genes.mit.edu/GENSCAN.html" TARGET=_blank>
GenScan</A>. These predictions are based on transcriptional, translational, 
and donor and acceptor splicing signals, plus length and compositional 
distributions of exons, introns and intergenic regions. The program does not 
use similarity to known proteins like 
<A HREF="http://www.ncbi.nlm.nih.gov:80/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11337476&dopt=Abstract" TARGET=_blank>
GenomeScan</A>.  For a description of the Genscan program and the model that
underlies it, refer to Burge, C. and Karlin, S. (1997) 
<A HREF="http://www.sciencedirect.com/science?_ob=ArticleURL&_udi=B6WK7-45VGF7T-9&_coverDate=04%2F25%2F1997&_alid=171225362&_rdoc=1&_fmt=&_orig=search&_qd=1&_cdi=6899&_sort=d&view=c&_acct=C000050221&_version=1&_urlVersion=0&_userid=10&md5=0e469be9c055b80c520ef6dde88133ba"
TARGET=_blank>Prediction of complete gene structures in human genomic
DNA</A>. <em>J. Mol. Biol.</em> 268(1):78-94.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show basic track
	information, plus links to the translated protein, predicted mRNA, and
	genomic sequences.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>genscan</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Genscan Genes </B>(Zoo)<BR>
This track shows gene predictions from the 
<A HREF="http://genes.mit.edu/GENSCAN.html" TARGET=_blank>
GenScan</A> program written by Chris Burge. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode shows basic track
	information and provides a link to view the underlying genomic sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>pjt_genscan</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>GNF Atlas 2</B><BR>
This track shows expression data from the GNF (The Genomics Institute of the
Novartis Research Foundation) Gene Expression Atlas 2. The data contains 2 
replicates each of 61 mouse tissues and 79 human tissues run over Affymetrix 
microarrays. For more
information about the experiments, refer to the description page for this track.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses display conventions similar to those for PSL alignment tracks. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	In dense or packed display mode, the track shows averages of related
	tissues. In full display mode, all tissues are displayed. Click on an 
	individual tissue entry to show a gene expression map for all tissues in 
	the experiment. 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>gnfAtlas2</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>GNF Ratio</B><BR>
This track shows expression data from GNF (The Genomics Institute of the
Novartis Research Foundation) using Affymetrix GeneChips. For more
information about the experiments, refer to the description page for this track.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	In dense display mode, the track color denotes the average signal over
	all experiments on a log base 2 scale. Lighter colors correspond
	to lower signals and darker colors correspond to higher signals. 
	In full display mode, the color of each item represents the log base 2 ratio 
	of the signal of that particular experiment to the median signal of all 
	experiments for that probe. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains options to change the display mode,
	group the displayed results, and change the display colors.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>affyRatio</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#LFS"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>GNF U74A</B><BR>
This track shows expression data from GNF using the Affymetrix U74A chip. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows display conventions similar to PSL alignment tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	In dense or packed display mode, the track shows averages of related
	tissues. In full display mode, all tissues are displayed. Click on an 
	individual tissue entry to show a gene expression map for all tissues in 
	the experiment. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>affyGnfU74A</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>GNF U74B</B><BR>
This track shows expression data from GNF using the Affymetrix U74B chip. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows display conventions similar to PSL alignment tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	In dense or packed display mode, the track shows averages of related
	tissues. In full display mode, all tissues are displayed. Click on an 
	individual tissue entry to show a gene expression map for all tissues in 
	the experiment. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>affyGnfU74B</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>GNF U74C</B><BR>
This track shows expression data from GNF using the Affymetrix U74C chip. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows display conventions similar to PSL alignment tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	In dense or packed display mode, the track shows averages of related
	tissues. In full display mode, all tissues are displayed. Click on an 
	individual tissue entry to show a gene expression map for all tissues in 
	the experiment. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>affyGnfU74C</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="H_tracks"></A>

<P><B>Haplotype Blocks </B> (chr21)<BR>
See the description for <em>Perlegen</em>.
</P>

<P><B>Haplotype Blocks</B> (chr22)<BR>
This track shows haplotype blocks on Chromosome 22 from The University of 
Oxford and The Wellcome Trust Sanger Institute, as described in
Dawson E. et. al. (2002), 
"<A HREF="http://bioinformatics.well.ox.ac.uk/~lon/chr22/data/chr22ld.pdf" 
TARGET=_blank>A first-generation linkage disequilibrium map of human 
chromosome 22</A>", Nature 418:544-8.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	The location of each haplotype block is represented by a blue horizontal 
	line with tall vertical blue bars at the first and last SNPs of
	the block. Blocks are displayed as starting at the first SNP and ending 
	at the last SNP of the block. Individual SNPs are denoted
	by smaller black vertical bars. At multi-megabase resolution in dense 
	display mode, clusters of tall blue bars may indicate hotspots
	for recombination.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show basic position
	information, SNP information for the block, and a link to sequence 
	information for the feature. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>haplotype</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>H-Inv </B><BR>
This track shows alignments of full-length cDNAs used as the basis of the 
H-Invitational Gene Database (HInv-DB). Entries in this track include 
gene structures, functions, functional domains, metabolic pathways, and many
other entities.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for PSL alignment tracks. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode provides gene ID, cluster
	ID, and cDNA accession information and a link to sequence 
	information for the feature. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>HInvGeneMrna</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>HOX Genes</B> (Tetraodon chr2, 8, 9, 17, 21, and un_random only)<BR>
This track shows HOX genes annotated using information on 
sequence similarity between Tetraodon vs. human, mouse and zebrafish HOX 
protein and nucleotide sequences. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene prediction tracks. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode provides position, size, and
	CDS information, and links to the related translated protein, predicted
	mRNA, and genomic sequences.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>hoxGenes</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Human Best </B><BR>
This track displays best-in-genome alignments of the specified human draft
assembly to the selected genome baed on the associated alignment net, i.e. the
best chain for every part of the human genome. See the track description page
for the Human Net track for more information about the methods used to generate
this track.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for cross-species synteny 
	tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show position information
	for the human alignment and the selected genome, alignment details, and 
	a link to a detailed view of parts of the alignment.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>axtNetHgX</I> positional table in the Table Browser, where 
	<em>X</em> is the assembly version in which you're interested. 
	To view the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Human Blat </B><BR>
This track displays translated BLAT alignments of the human genome
vs. the mouse genome. 
The alignments were done with BLAT in translated protein mode using the
parameters -q=dnax -t=dnax and otherwise the default settings. Both
genomes were masked with RepeatMasker and Tandem Repeats Finder before
running BLAT. Places where more than 250 alignments occurred
over the same place were filtered out. Beware of alignments of greater
than 97% identity: these may reflect mouse contamination in the human
genome or human contamination in the mouse genome.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for a BLAT alignment track.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show basic track
	information and a link to sequence information for the alignment. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>blatHuman</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Human Chain </B><BR>
This track shows human genomic alignment to the selected genome using blastz and
axtChain. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for chain tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode shows position information
	in the mouse and the selected genome, the chain ID and score, and 
	provides links to view details of parts of the chain or to open the 
	mouse browser to the corresponding position. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>chainX</I> positional table in the Table Browser, where <I>X</I>
	is the assembly version in which you're interested, e.g. Hg16. To view 
	the table definition, click <A
	HREF="../gbdDescriptions.html#Chain"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Human Cons </B><BR>
This track displays the conservation between the mouse and human genomes for 
50bp windows in the mouse genome that have at least 15bp aligned to human. The 
score for a window reflects the probability that the level of observed 
conservation in that 50bp region would occur by chance under neutral evolution. 
It is given on a logarithmic scale, and thus it is called the "L-score". An 
L-score of 1 means there is a 1/10 probability that the observed conservation 
level would occur by chance, an L-score of 2 means a 1/100 probability, an 
L-score of 3 means a 1/1000 probability, etc. The L-scores display as 
"mountain ranges".

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for wiggle tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains options to configure 
	several of the graph characteristics. To read more information
	about each option, click the &quot;Graph configuration help&quot; link. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on one of the peaks in the track when in full display mode will 
	show human genome 
	position information for the window, as well as position information for 
	the corresponding region in the human genome. The details page provides 
	links to the underlying sequence for the alignment and to the sequence 
	underlying the individual alignment windows. In full mode in 
	a zoomed-out display, detailed information may not be accessible for an 
	individual feature. In this case, zoom in and try again. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>musHumL</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#WiggleTrack"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Human Deletions</B><BR>
This track displays regions of the chimp assembly that are deleted in the 
human assembly. Only regions of between 80 and 12000 bases are included. 
The name of each deletion is a unique pointer to that deletion followed by an 
underscore and then its length. A similar track, showing chimp deletions in the 
human assembly, appears in the human Genome Browser.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Deletions are indicated by thin vertical lines. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on a deletion in full display mode provides the item name,
	position, genomic size, and a link to the DNA underlying the feature.
	</TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>humanDels</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Human Ecores</B><BR>
This track shows human evolutionary conserved regions (ecores) computed against 
the selected assembly by the Exofish program at Genoscope. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses display conventions similar to PSL alignment tracks. Each 
	singleton block corresponds to an ecore. Blocks connected by a 
	line correspond to an "ecotig", a set of colinear ecores in a 
	syntenic region.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show positional 
	information, 
	a link to the Exofish browser, and a link to display the DNA sequence.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>ecores<X></I> positional table in the Table Browser, where 
	<em>X</em> is the assembly version in which you're interested (e.g. 
	hg16). 
	To view the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Human ESTs </B><BR>
This track shows alignments between Expressed Sequence Tags (ESTs) -
spliced and unspliced - in GenBank and the genome. ESTs are single read 
(typically approximately 500 base) sequences 
that usually represent fragments of transcribed genes. 
To view only spliced ESTs, see the Spliced EST track.
This track was generated by aligning human ESTs from GenBank against the
genome using the BLAT program. Note that the maximum intron length
allowed by BLAT is 500,000 bases, which may eliminate some ESTs with
very long introns that might otherwise align. When a single EST aligns
in multiple places, the alignment having the highest base identity is
found. Only alignments that have a base identity level within 1% of the
best are kept. Alignments must also have at least 93% base identity to be 
kept.
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for EST tracks.  Hatch marks on the 
	introns indicate the direction of transcription. In the December 2001 
	assembly and later, this direction is taken by looking at the splice sites. 
	In previous assemblies, the direction of transcription was taken from the
	GenBank annotations, which frequently were inaccurate; these should be 
	verified in the details page alignment on the basis of the GT-AG splice 
	donor/acceptor rule.  As with otherEST tracks, it may be necessary to zoom 
	in to display entries in full display mode due to the 300 line display 
	limit.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter to restrict the display to 
	only those entries that match certain keywords and to change the color 
	of the selected entries.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an individual EST entry will show basic track information,
	the accession number, GenBank highlights, and links to EST/Genomic
	Alignments.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>all_est</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Human mRNAs</B><BR>
This track shows alignments between mRNAs from the current assembly in GenBank 
and the genome.  The mRNAs are aligned against the genome using the BLAT 
program. When a single mRNA aligns 
in multiple places, the alignment having the highest base identity is found. 
Only alignments that have a base identity level within 1% of the best
are kept. Alignments must also have at least 95% base identity
to be kept.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track follows the display conventions for PSL alignment tracks. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter to restrict the display to 
	only those entries that match certain keywords and to change the color 
	of the selected entries.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show a summary of the GenBank 
	entry on that mRNA, convenient hyperlinks to major databases, and base-by-base 
	mRNA/genomic alignments. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>all_mrna</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Human Net</B><BR>
This track shows the best human chain for every part of the selected genome.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for net tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	In full display mode, clicking on a box shows details information about 
	the chain as a whole, while clicking on a line shows information
	about the gap. The detailed information is useful in determining the 
	cause of the gap or, for lower level chains, the genomic 
	rearrangement.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>netX</I> positional table in the Table Browser, where <I>X</I>
	is the assembly version in which you're interested, e.g. Hg16. To view 
	the table definition, click <A
	HREF="../gbdDescriptions.html#NetAlign"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Human Proteins</B><BR>
This track shows tBLASTn alignments of the peptides from the predicted and 
known genes indentified in the specified Known Genes track.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track follows the display conventions for PSL alignment tracks. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains options to configure the information shown
	in the track element labels.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show position information, 
	external links to human mRNA and SwissPort information, and links to the alignment
	and peptide sequences.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>blastHg16KG</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Human Synteny</B><BR>
This track shows syntenous (corresponding) regions between mouse and human 
chromosomes. The track was created by passing a 100k non-overlapping window over 
the genome and using the blastz best in human genome alignments to look for
high-scoring regions where at least 40% of the bases aligned with the same 
region in mouse. 100k segments were joined together if they agreed in direction 
and were within 500kb of each other in the mouse genome and within 4mb of each 
other in the human. Gaps were joined between syntenic anchors if the bases 
between two flanking regions agreed with synteny (direction and human location). 
The syntenic block was extended to include these areas.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for synteny tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode shows details of the
	mouse-human correspondence and provides a link to the genomic sequence 
	for this feature. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>syntenyHuman</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="../gbdDescriptions.html#SyntenyHuman"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Human Tight</B><BR>
This track displays <I>Blastz</I> alignments of the human
assembly to the current genome, filtered by <I>axtBest</I> and
<I>subsetAxt</I> with very stringent constraints. See the track's description
page for more details. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for cross-species synteny 
	tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show position information
	for the human and the current genome, alignment details, and a link to 
	a detailed view of parts of the alignment.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter that can be used to turn 
	on the chromosome color track or to filter the display output by 
	chromosome.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>blastzTightHg<X></I> positional table in the Table Browser, where
	<em>X</em> is the number of the assembly. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="J_tracks"></A>

<P><B>JGI Genes</B> (<em>C. intestinalis</em>)<BR>
This track contains alignments of predicted transcripts from the DOE Joint Genome Institute 
(JGI) to the <em>C. intestinalis</em> genome. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode brings up basic track
	information, plus links to the translated protein, predicted mRNA, and
	genomic sequences.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>jgiGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="K_tracks"></A>

<P><B>Known Genes </B><BR>
The Known Genes track shows known protein coding genes based on proteins from 
SWISS-PROT, TrEMBL, and TrEMBL-NEW and their corresponding mRNAs from GenBank. 
Features are colored according to origin and review status:
<UL>
<LI>
<B>Black</B>: Feature has a corresponding entry in the Protein Data Bank (PDB).
<LI>
<B>Dark blue</B>: Feature has a corresponding protein in SWISS-PROT or a corresponding 
NCBI Reference Sequence mRNA with a "Reviewed" status. 
<LI>
<B>Lighter blue</B>: Feature has a corresponding NCBI Reference 
Sequence mRNA with "Provisional" status. 
<LI>
<B>Lightest blue</B>: All other features.
</UL>
All mRNAs of a species are aligned against the genome using the BLAT program. 
When a single mRNA aligns in multiple places, only the best alignments with at
least 98% sequence identity are kept. The set of alignments is further reduced 
by keeping only those mRNAs that are referenced by a protein in SWISS-PROT, 
TrEMBL, or TrEMBL-NEW. When a single protein references multiple mRNAs, the best 
mRNA is chosen based on a quality score that depends on its length, how good its 
translation matches the protein sequence, and its release date. The resulting 
list of mRNA and protein pairs is trimmed by removing short invalid entries and 
consolidating entries with identical CDS regions. Finally, RefSeq entries 
derived from DNA sequences instead of mRNA sequences are added. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for known genes and gene 
	predictions tracks. See above description of color coding conventions. 
	</TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display provides links to the 
	corresponding protein, mRNA, and genomic sequences, microarray expression data, 
	protein structure information, homology information and related links to 
	several tools and databases. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>knownGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#KnownGene"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="M_tracks"></A>

<P><B>Map Contigs</B><BR>
This track shows the locations of contigs of clones on the physical map. 
In assembly versions prior to the August 6 2001 assembly, this track was
based on the Washington University accession map, which in turn was
based on a fingerprint contig (FPC) map described in 
<A
HREF="http://www.nature.com/cgi-taf/DynaPage.taf?file=/nature/journal/v409/n6822/full/409934a0_fs.html"
TARGET=_blank">'A physical map of the human genome'</A> in Nature volume 409 pages 934-941.
Starting with the August 6 2001 assembly, this track is based on tiling path 
fingerprint (TPF) maps curated by the sequencing centers responsible for each 
chromosome.  The TPF maps are merged with the FPC map,
favoring the TPF map where conflicts occur. This step increases the
clone coverage substantially over that in the TPF maps. The clone contigs in
this merged map are shown in this track.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	The clone contigs are represented by blocks that indicate their location
	on the assembly sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode will show a brief description of the 
	extent and number of clones in contig, and the DNA sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>ctgPos</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#ContigInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Mapped Human</B> (<em>X. tropicalis</em>)<BR>
This track contains tBLASTn alignments of the peptides from the predicted 
and known genes identified in the specified version of the Known Genes track.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track follows the display conventions for BLAT tracks.</TD></TR>	
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode shows positional information,
	related mRNA and SwissProt links, the protein length, and links to the
	alignment an peptide sequences.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>blastHg17KG</I> positional table in the Table Browser. To view the 
	table definition, click <A
	HREF="../gbdDescriptions.html#PslInfo" TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>MCS Binomial</B> (Zoo)<BR>
This track shows Multi-Species Conserved Sequences (MCSs) defined using a 
conservation score threshold from the binomial-based method
that selects the top 5% most conserved sequence. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Sequence positions are depicted by blocks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode shows position and size 
	information, and provides a link to the underlying DNA sequence. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>mcs_b</I> positional table in the Table Browser. To view the 
	table definition, click <A
	HREF="../gbdDescriptions.html#BED" TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>MCS Inter </B>(Zoo)<BR>
This track shows Multi-Species Conserved Sequences (MCSs) found by intersecting 
the sequences generated using the binomial-based and parsimony p-value-based 
methods (see the MCS Binomial and MCS ParsPVal track descriptions). 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Sequence positions are depicted by blocks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode shows position and size 
	information, and provides a link to the underlying DNA sequence. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>intersectingMCS</I> positional table in the Table Browser. To view the 
	table definition, click <A
	HREF="../gbdDescriptions.html#BED" TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>MCS InterMerged </B>(Zoo)<BR>
Same as MCS Inter, but regions less than 10 bp apart have been merged into one 
region.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Sequence positions are depicted by blocks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode shows position and size 
	information, and provides a link to the underlying DNA sequence. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>intersectingMergedMCS</I> positional table in the Table Browser. To 
	view the table definition, click <A
	HREF="../gbdDescriptions.html#BED" TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>MCS ParsPVal</B> (Zoo)<BR>
This track shows Multi-Species Conserved Sequences (MCSs) defined using a 
conservation score threshold from the parsimony p-value method
that selects the top 5% most conserved sequence. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Sequence positions are depicted by blocks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode shows position and size 
	information, and provides a link to the underlying DNA sequence. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>mcs_p</I> positional table in the Table Browser. To view the 
	table definition, click <A
	HREF="../gbdDescriptions.html#BED" TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>MGC Genes</B><BR>
This track displays alignments between <A HREF="http://mgc.nci.nih.gov/" TARGET=_blank>Mammalian Gene Collection</A> (MGC) human mRNAs with full-length ORFs and the human genome. 
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show basic track
	information and links to the mRNA, genomic, and comparative sequences.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page offers configuration options that can be used to turn
	on codon coloring for the track. Click the &quot;Codon coloring 
	help&quot; link on the page for more information about this feature.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>mgcGenes</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Microsatellites</B><BR> 
This track contains all perfect microsatellite repeats with between 2 and 10 bp 
repeat units and 10 or more perfect copies. Over 90% of the items will be 
multi-allelic polymorphisms. This track was created by using three programs: 
Tandyman, display_VNTR and Primeleftright. See the track description page for
more information.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	Microsatellite locations are indicated by thin grey vertical lines.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
 	Clicking on an item in full display mode shows position information, 
	details about the repeat, and a link to the DNA sequence associated with
	the feature.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>vntr</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#Vntr"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>


<P><B>miRNA </B><BR>
This track shows microRNAs from the 
<A HREF="http://www.sanger.ac.uk/Software/Rfam/mirna/" TARGET=_blank>miRNA Registry</A> 
at the Wellcome Trust Sanger Institute. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	Mature miRNAs (miRs) are represented by thick lines; the predicted 
	stem-loop portions of the primary transcripts are indicated by thinner 
	lines. miRNAs in the sense orientation are shown in black; those in the 
	reverse orientation are colored grey. When a single precursor produces 
	two mature miRs from its 5' and 3' parts, it is displayed twice with the 
	two different positions of the mature miR. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking in full mode displays positional information and 
	a link to item's entry in the miRNA registry.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>miRNA</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Most Conserved</B><BR>
This track shows predictions of conserved elements produced by the phastCons program. 
Predictions are based on a phylogenetic hidden Markov model (phylo-HMM), a type of 
probabilistic model that describes both the process of DNA substitution at each site in 
a genome and the way this process changes from one site to the next. For more 
information on the methods used to produce this track, see the track's description
page.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Rectangular blocks indicate the positions of conserved elements.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show position information and
	a list of the top-scoring elements in the window.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the <I>phastConsElements</I>
	positional table from the Table Browser.  To view the table definition, click
	<A HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.
	</TD></TR>
</TABLE>

<P><B>Mouse Blat </B><BR>
This track displays alignments of mouse genome to human. The mouse genomic 
DNA is from the Arachne assembly of mouse whole genome shotgun reads available 
at the <A HREF="http://www.ncbi.nlm.nih.gov/Traces/" TARGET=_blank>Trace Repository</A>. 
The alignments were done with BLAT in translated protein mode requiring 2 nearby 
4-mer matches to trigger a detailed alignment. The human genome was masked with 
<A HREF="http://www.repeatmasker.org" TARGET=_blank>RepeatMasker</A> 
and <A HREF="http://tandem.bu.edu/trf/trf.html" TARGET=_blank>Tandem Repeats Finder</A> 
prior to running BLAT. Regions with more than 300 aligned reads were assumed novel 
repeat elements and filtered out. Alignments of greater than 97% identity may represent 
mouse contamination of human or vice versa.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows the display conventions for BLAT tracks.</TD></TR>	
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item will show details on mouse sequence fragment and 
	alignment details. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>blatMouse</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Mouse Chain </B><BR>
This track shows mouse genomic alignment to the selected genome using blastz and
axtChain. See the Human Chain track description for more information.

<P><B>Mouse Cons </B><BR>
This track displays the conservation between the human and mouse genomes for 
50bp windows in the human genome that have at least 15bp aligned to mouse. The 
score for a window reflects the probability that the level of observed 
conservation in that 50bp region would occur by chance under neutral evolution. 
It is given on a logarithmic scale, and thus it is called the "L-score". An 
L-score of 1 means there is a 1/10 probability that the observed conservation 
level would occur by chance, an L-score of 2 means a 1/100 probability, an 
L-score of 3 means a 1/1000 probability, etc. The L-scores display as 
"mountain ranges".

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for wiggle tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains options to configure 
	several of the graph characteristics. To read more information
	about each option, click the &quot;Graph configuration help&quot; link. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on one of the peaks in the track when in full display mode will 
	show human genome 
	position information for the window, as well as position information for 
	the corresponding region in the mouse genome. The details page provides 
	links to the underlying sequence for the alignment and to the sequence 
	underlying the individual alignment windows. In full mode in 
	a zoomed-out display, detailed information may not be accessible for an 
	individual feature. In this case, zoom in and try again. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>humMusL</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#WiggleTrack"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Mouse Ecores</B><BR>
This track shows Mouse evolutionary conserved regions (ecores) computed against 
the selected assembly genome by the Exofish program at Genoscope. For 
information on this track, see the Human Ecores track description.

<P><B>Mouse ESTs </B><BR>
This track shows alignments between mouse Expressed Sequence Tags (ESTs) -
spliced and unspliced - in GenBank and the genome. This track was generated by 
aligning mouse ESTs from GenBank against the
genome using the BLAT program. Note that the maximum intron length
allowed by BLAT is 500,000 bases, which may eliminate some ESTs with
very long introns that might otherwise align. When a single EST aligns
in multiple places, the alignment having the highest base identity is
found. Only alignments that have a base identity level within 1% of the
best are kept. Alignments must also have at least 93% base identity to be 
kept.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for EST tracks. As with other
	EST tracks, it may be necessary to zoom in to display entries in full
	display mode due to the 300 line display limit.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an individual EST entry will show basic track information,
	the accession number, GenBank highlights, and links to EST/Genomic
	Alignments.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>all_est</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Mouse mRNAs</B><BR>
See the description for Human mRNAs.

<P><B>Mouse Net </B><BR>
This track shows the best mouse chain for every part of the selected genome.
See the Human Net track description for more information.

<P><B>Mouse Synteny (hg13+)</B><BR>
This track shows syntenous (corresponding) regions between human and mouse 
chromosomes. The track was created by passing a 100k non-overlapping window over 
the genome and using the blastz best in mouse genome alignments to look for
high-scoring regions where at least 40% of the bases aligned with the same 
region in mouse. 100k segments were joined together if they agreed in direction 
and were within 500kb of each other in the human genome and within 4mb of each 
other in the mouse. Gaps were joined between syntenic anchors if the bases 
between two flanking regions agreed with synteny (direction and mouse location). 
The syntenic block was extended to include these areas.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for synteny tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode shows details of the
	mouse-human correspondence and provides a link to the genomic sequence 
	for this feature. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>syntenyMouse</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="../gbdDescriptions.html#SyntenyMouse"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Mouse Synteny (hg12 only)</B><BR>
This track shows orthologous (syntenic) regions between mouse and human chromosomes based on data from Michael Kamal at the Whitehead Institute.
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for synteny tracks.</TD></TR>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode will show details of the
	mouse-human correspondence, as well links to genomic dotplots, 
	additional information, and an alternative synteny map based on 
	orthologous genes. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>mouseSynWhd</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#mouseSynWhd"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Multiz Yeast</B><BR>
This track displays multiz multiple alignments of seven species of the genus 
<em>Saccharomyces</em>. The graphic display shows the alignment projected onto 
<em>S. cerevisiae</em>. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	In full display mode, the track displays pairwise alignments of several
	species aligned to the <em>S. cerevisiae</em> genome. The pairwise 
	alignments 
	are displayed in standard UCSC browser "dense" mode using a grey-scale 
	density gradient.  When zoomed-in to the base-display level, the track 
	shows the base composition of each alignment. The numbers and symbols on 
	the "hidden gap" line indicate the lengths of gaps in the <em>S. 
	cerevisiae</em> sequence 
	at those alignment positions. If the gap size is greater than 9, the 
	"+" symbol is displayed. The track must be zoomed-in to 30,000
	or fewer bases to access detailed information about the 
	alignments at a specific position.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show detailed alignment 
	information for each of the organisms. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>multizYeast</I> positional table in the Table Browser. To view the 
	table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#multizYeast"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="N_tracks"></A>

<P><B>NIA Genes</B><BR>
This track displays alignments of the National Institute on Aging (NIA) Mouse 
Gene Index (Version 3) against the mouse genome. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for Blat alignment tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an individual track item in full mode displays a page
	with alignment details and a link to the corresponding NIA Mouse Gene 
	Index entry. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>NIAGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>NCI60</B><BR>
This track displays cDNA expression data for a micro-array of 8,000 genes for 
60 cell lines used in an <A HREF="http://genome-www.stanford.edu/nci60/" TARGET=_blank>NCI 
cancer drug screen</A>.  

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	To understand the track, experiment with it set to 
	full mode but other tracks (except known genes) hidden, while reading its
	<A HREF="http://genome.ucsc.edu/cgi-bin/hgTrackUi?hgsid=1017057&c=chr3&g=nci60" TARGET=_blank> 
	filter page</A>. Each column of colored boxes represents variation in 
	transcript levels for a given cDNA across array experiments; each row 
	represents the measured transcript levels for all genes in a single sample. 
	The saturation of a color corresponds to the magnitude of transcript 
	variation. Black  indicates an undetectable change in expression; gray shows 
	missing data. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter to limit the tissue types 
	displayed and change the color scheme of the display.  </TD></TR>
	<B><I>Track details:</I></B> 
	Clicking on an individual track item in full mode opens a rich details 
	page.  The exon probe and experiment selected are highlighted in blue 
	under item name. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>nci60</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Non-A. gambiae mRNAs </B><BR>
See the description for Non-Human mRNAs.

<P><B>Non-A. mel. mRNAs </B><BR>
mRNAs of organisms other than <em>A. mellifera</em>. See the description for Non-Human 
mRNAs.

<P><B>Non-Chicken mRNAs </B><BR>
See the description for Non-Human mRNAs.

<P><B>Non-Chicken RefSeq</B><BR>
This track shows known protein-coding genes from organisms other than chicken, taken from 
the mRNA reference sequences collection (RefSeq) compiled at NCBI. See the Non-Dog RefSeq 
track for more information.

<P><B>Non-Chimp ESTs </B><BR>
See the description for Non-Human ESTs.

<P><B>Non-Chimp mRNAs </B><BR>
See the description for Non-Human mRNAs.

<P><B>Non-<em>C. intestinalis</em> mRNAs </B><BR>
See the description for Non-Human mRNAs.

<P><B>Non-Ciona mRNAs </B><BR>
See the description for Non-Human mRNAs.

<P><B>Non-Ciona RefSeq</B><BR>
This track shows known protein-coding genes from organisms other than 
<em>C. intestinalis</em>, taken from the mRNA reference sequences collection (RefSeq) 
compiled at NCBI. See the Non-Dog RefSeq track for more information.

<P><B>Non-Dog RefSeq</B><BR>
This track shows known protein-coding genes from organisms other than dog, taken from 
the mRNA reference sequences collection (RefSeq) compiled at NCBI. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene 
	predictions tracks. </TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display provides links to the 
	corresponding protein, mRNA, and genomic sequences, and related links to 
	external databases. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains configuration options to
	change the feature label to display the gene name (default), the 
	accession name, both the gene and accession names, or no label at all.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>xenoRefGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Non-D. anan. mRNAs </B><BR>
mRNAs of organisms other than <em>D. ananassae</em>. See the description for Non-Human 
mRNAs.

<P><B>Non-D. anan. RefSeq </B><BR>
This track shows known protein-coding genes from organisms other than <em>D.
ananassae</em>, taken from the mRNA reference sequences collection (RefSeq) compiled 
at NCBI. See the description for Non-Human mRNAs.

<P><B>Non-D. melanogaster mRNAs </B><BR>
See the description for Non-Human mRNAs.

<P><B>Non-D. moj. mRNAs </B><BR>
mRNAs of organisms other than <em>D. mojavensis</em>. See the description for Non-Human 
mRNAs.

<P><B>Non-D. moj. RefSeq </B><BR>
This track shows known protein-coding genes from organisms other than <em>D.
mojavensis</em>, taken from the mRNA reference sequences collection (RefSeq) compiled 
at NCBI. See the description for Non-Human mRNAs.

<P><B>Non-D. pseudo. mRNAs </B><BR>
mRNAs of organisms other than <em>D. pseudoobscura</em>. See the description for 
Non-Human mRNAs.

<P><B>Non-D. pseudo. RefSeq </B><BR>
This track shows known protein-coding genes from organisms other than <em>D.
pseudoobscura</em>, taken from the mRNA reference sequences collection (RefSeq) compiled 
at NCBI. See the description for Non-Human mRNAs.

<P><B>Non-D. vir. mRNAs </B><BR>
mRNAs of organisms other than <em>D. virilis</em>. See the description for 
Non-Human mRNAs.

<P><B>Non-D. vir. RefSeq </B><BR>
This track shows known protein-coding genes from organisms other than <em>D.
virilis</em>, taken from the mRNA reference sequences collection (RefSeq) compiled 
at NCBI. See the description for Non-Human mRNAs.

<P><B>Non-D. yakuba mRNAs </B><BR>
See the description for Non-Human mRNAs.

<P><B>Non-Fugu mRNAs </B><BR>
See the description for Non-Human mRNAs.

<P><B>Non-Human ESTs </B><BR>
This track displays translated BLAT alignments of non-human vertebrate
ESTs from GenBank. The alignments were passed through a piecewise
near-best-in-genome filter. In cases where ESTs align to multiple places in the 
genome, only the near-best are retained. Matches help validate human exon 
identification.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for EST tracks. As with other
	EST tracks, it may be necessary to zoom in to 
	display entries in full display mode due to the 300 line display 
	limit.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display will show a summary of the GenBank 
	entry, a hyperlink to the full record, PubMed links by gene name, 
	product, and author, and mRNA/genomic alignment information. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>xenoEst</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Non-Human mRNAs </B><BR>
This track displays translated BLAT alignments of vertebrate and invertebrate 
mRNA (other than that from the selected genome) from GenBank. The alignments 
were passed through a near-best-in-genome
filter. Some mRNAs align to multiple places but only the near-best are retained. 
For a distant vertebrate, alignment requires a fairly slowly evolving gene. 
Matches are often plausible ortholog candidates.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for PSL alignment tracks. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter to restrict the display to 
	only those entries that match certain keywords and to change the color 
	of the selected entries.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show a summary of the GenBank 
	entry on that mRNA, convenient hyperlinks to major databases, and 
	base-by-base mRNA/genomic alignments. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>xenoMrna</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Non-Mouse ESTs </B><BR>
See the description for Non-Human ESTs.

<P><B>Non-Mouse mRNAs </B><BR>
See the description for Non-Human mRNAs.

<P><B>Non-Opossum mRNAs </B><BR>
See the description for Non-Human mRNAs.

<P><B>Non-Rat ESTs </B><BR>
See the description for Non-Human ESTs.

<P><B>Non-Rat mRNAs </B><BR>
See the description for Non-Human mRNAs.

<P><B>Non-Tetraodon mRNAs </B><BR>
See the description for Non-Human mRNAs.

<P><B>Non-Zebrafish mRNAs </B><BR>
See the description for Non-Human mRNAs.

<A NAME="O_tracks"></A>

<P><B>Opossum Chain </B><BR>
This track shows opossum genomic alignments to the selected genome using 
blastz and axtChain. See the Human Chain track description for more information.

<P><B>Opossum mRNAs</B><BR>
See the description for Human mRNAs.

<P><B>Opossum Net </B><BR>
This track shows the best opossum chain for every part of the selected 
genome.  See the Human Net track description for more information.

<P><B>Overlap SNPs</B> (older archived versions only)<BR>
This track shows single nucleotide polymorphisms found on overlapping contigs. 
Overlap SNPs are useful markers in mapping genes and diseases. This data was provided 
by the <A HREF="http://snp.cshl.org/" TARGET=_blank>SNP Consortium</A>.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	SNP locations are indicated by thin grey vertical lines.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking in full mode brings up a location description and 
	<A HREF="http://www.ncbi.nlm.nih.gov/SNP/" TARGET=_blank>dbSNP</A> link.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>snpNih</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#SNPsites"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="P_tracks"></A>

<P><B>Parsimony P-Values</B> (Zoo)<BR>
This track displays the level of conservation of Zoo sequences at any given 
position. More precisely, it computes the p-value of observing a certain 
parsimony score for a given column of the multiple alignment under the null 
model of neutral evolution.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for wiggle tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains options to configure 
	several of the graph characteristics. To read more information
	about each option, click the &quot;Graph configuration help&quot; link. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on one of the peaks in the track when in full display mode will 
	show genome position information for the window and a link to the
	underlying DNA sequence. In full mode in 
	a zoomed-out display, detailed information may not be accessible for an 
	individual feature. In this case, zoom in and try again. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>parsPValues</I> positional table in the Table Browser. To view the 
	table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#WiggleTrack"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Perlegen </B>(chr21 only)<BR>
This track shows haplotype blocks derived from common single nucleotide 
polymorphisms (SNPs) on Chromosome 21 by Perlegen Sciences, as
described in Patil N et. al. (2001), 
"<A HREF="http://www.sciencemag.org/cgi/content/full/294/5547/1719" TARGET=_blank>Blocks of Limited Haplotype Diversity Revealed by High-Resolution Scanning</A>", 
<em>Science</em> 294:1719-1723. General information on the blocks is available 
from Perlegen's <A HREF="http://www.perlegen.com/haplotype/" TARGET=_blank>Chromosome 21 Haplotype Browser</A>. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	The location of each haplotype block is represented by a blue horizontal 
	line with tall vertical blue bars at the first and last SNPs of
	the block. Blocks are displayed as starting at the first SNP and ending 
	at the last SNP of the block. This is slightly different from the
	representation on the Perlegen web site in which blocks are stretched 
	until they abut each other. The shade of the blue indicates the
	minimum number of SNPs required to discriminate between haplotype 
	patterns that account for at least 80% of genotyped
	chromosomes. Darker colors indicate fewer SNPs are necessary. Individual 
	SNPs are denoted by smaller black vertical bars. At
	multi-megabase resolution in dense display mode, clusters of tall blue 
	bars may indicate hotspots for recombination.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show basic position
	information, SNP information for the block, and a link to sequence 
	information for the feature. To view additional information on a 
	particular block,  click on <em>Outside Link</em> on the item's details 
	page. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>perlegen</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="Q_tracks"></A>

<P><B>QTL</B><BR>
This track shows approximate positions of quantititive trait loci (QTL) based on 
reported peak LOD scores taken from Jackson Lab's 
<A HREF="http://www.informatics.jax.org/" TARGET=_blank> Mouse Genome Informatics (MGI) Database</A>. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	QTL locations are indicated by thin vertical bars.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full mode provides a link to the MGI record for the selected QTL,
	information about the QTL, and position information.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>jaxQTL</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#JaxQTL"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Quality Scores</B><BR>
This track shows the sequencing quality score (between 0 and 100) 
of each base in the assembly. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	The height at each position of the track indicates the quality of the 
	base. When zoomed out to a large range, the heights reflect the averaged 
	scores. Scores of 40 or higher reflect high confidence in the sequence 
	(with an error rate of less than 1/10,000); scores of 20 or higher 
	reflect reasonable confidence (of working draft quality). </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>quality</I> positional table in the Table Browser.
	To view the table definition, click <A
	HREF="../gbdDescriptions.html#WiggleTrack2"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="R_tracks"></A>

<P><B>Radiation Hybrid Map</B> (Zebrafish)<BR>
This track shows alignments between zebrafish Radiation Hybrid (RH) map sequences 
(consisting of ESTs and other genetic markers) and the genome. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for EST tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
 	Clicking in full display mode provides a link to the genomic alignment. 
 	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>rhMap</I> positional table in the Table Browser.
	To view the table definition, click <A
	HREF="../gbdDescriptions.html#PslInfo" TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Random SNPs</B> (older archived versions only)<BR> 
This track displays single nucleotide polymorphisms (SNPs)found by random 
sequencing. Random SNPs are useful markers in mapping genes and complex traits.
Because these SNPs are detected from random reads, they are useful for
evolutionary studies. This data was provided by the <A HREF="http://snp.cshl.org/" TARGET=_blank>SNP 
Consortium</A>.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	SNP locations are indicated by thin grey vertical lines.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
 	Clicking in full mode brings up a location description 
 	and <A HREF="http://www.ncbi.nlm.nih.gov/SNP/" TARGET=_blank>dbSNP</A> link.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>snpTsc</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#SNPsites"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Rat Chain </B><BR>
This track shows rat genomic alignment to the selected genome using blastz and
axtChain. See the Human Chain track description for more information.

<P><B>Rat Ecores</B><BR>
This track shows rat evolutionary conserved regions (ecores) computed against 
the selected assembly genome by the Exofish program at Genoscope. For 
information on this track, see the Human Ecores track description.

<P><B>Rat ESTs </B><BR>
See the description for the Human ESTs track.

<P><B>Rat mRNAs</B><BR>
See the description for the Human mRNAs track.

<P><B>Rat Net </B><BR>
This track shows the best rat chain for every part of the selected genome. See 
the Human Net track description for more information.

<P><B>Rat Synteny </B><BR>
This track shows syntenous (corresponding) regions between human and rat 
chromosomes. The track was created by passing a 100k non-overlapping window over 
the genome and using the blastz best in mouse genome alignments to look for
high-scoring regions where at least 40% of the bases aligned with the same 
region in rat. 100k segments were joined together if they agreed in direction 
and were within 500kb of each other in the human genome and within 4mb of each 
other in the rat. Gaps were joined between syntenic anchors if the bases 
between two flanking regions agreed with synteny (direction and rat location). 
The syntenic block was extended to include these areas.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for synteny tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode shows details of the
	rat-human correspondence and provides a link to the genomic sequence 
	for this feature. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>syntenyRat</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="../gbdDescriptions.html#SyntenyRat"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Recomb Rate</B><BR> 
This track shows calculated sex-averaged rates of recombination based on either 
the deCODE, Marshfield, or Genethon genetic maps. The deCODE map rates are 
displayed by default.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	Each 1Mb window represents the average recombination rate of the bases 
	contained within that window. Darker shades of gray correspond to higher
	recombination rates.
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter to map distances 
	(sex-averaged distance, female distance, or male distance) for the 3 
	different types of maps.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
 	Clicking in full mode shows position information 
 	and a list of all the recombination rate values for that 1Mb window.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>recombRate</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#recombRate"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>RefSeq Genes </B><BR>
The RefSeq Gene track shows known protein coding genes taken from the mRNA
reference sequences collection (RefSeq) compiled at NCBI. Refseq mRNAs are aligned 
against the genome using the BLAT program. When a single mRNA aligns in multiple places, 
only the best alignments are kept. Alignments must also have at least 98% sequence 
identity to be kept. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for known genes and gene 
	predictions tracks. Non-coding RNA genes have their own track in some 
	assemblies. The color shading indicates the level of review that the 
	RefSeq record has undergone: predicted (light), provisional (medium), 
	reviewed (dark). </TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display provides links to the 
	corresponding protein, mRNA, and genomic sequences, and related links to 
	outside databases. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains configuration options to
	change the feature label to display the gene name (default), the 
	accession name, both the gene and accession names, or no label at all.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>refGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Reg. CHIP/CHIP</B><BR>
This track shows the location of the probes spotted on a slide in the chromatin 
immunoprecipitation/microarray hybridization (CHIP/CHIP) experiments described in 
Harbison, C.T. <em>et al.</em>, 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=15343339" 
TARGET=_blank>Transcriptional regulatory code of a eukaryotic genome</A>. 
<em>Nature</em> <B>431</B>(7004), 99-104 (2004). See also the "Regulatory Code" track 
for the position of the individual motifs.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Black boxes indicate the probe locations.  
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode shows position information, as well as
	which transcription factors pulled down DNA that is enriched for the probe 
	sequence, which transcription factor binding site motifs are present in the 
	probe, and whether these motifs are conserved in related yeast species. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>transRegCodeProbe</I> positional table in the Table Browser. To view 
	the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#TranRefCode"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Regulatory Code</B><BR>
This track shows putative regulatory elements in <em>S. cerevisiae</em> that are 
supported by cross-species evidence, as described in  (Harbison, C.T. <em>et al.</em>, 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=15343339" 
TARGET=_blank>Transcriptional regulatory code of a eukaryotic genome</A>. 
<em>Nature</em> <B>431</B>(7004), 99-104 (2004)). The authors performed a genome-wide 
location analysis with 203 known DNA-binding transcriptional regulators (some under 
multiple environmental conditions) and identified 11,000 high-confidence interactions 
between regulators and promoter regions. They then compiled a compendium of motifs for 
102 transcriptional regulators based on a combination of their experimental results, 
cross-species conservation data for four species of yeast, and motifs from the literature,
and finally, they mapped these motifs to the <em>S. cerevisiae</em> genome. This track 
shows positions at which these motifs matched the genome with high confidence and at 
which the matching sequence was well conserved across yeast species. See the 
"Reg. CHIP/CHIP" track for additional related information. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Regulatory elements are represented by rectangular blocks. The darker the shade
	of the block, the stronger the evidence supporting it.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode shows the sequence at that site 
	compared to the position-specific probability matrix for the associated 
	transcriptional regulator (shown as both a table and a graphical logo). It also 
	indicates whether the binding site is supported by experimental (CHIP/CHIP) 
	results and the number of other yeast species in which it is conserved.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The user can set the minimum unnormalized score criteria that data must 
	meet to be displayed in the track.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>tranRegCode</I> positional table in the Table Browser. To view 
	the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#TranRegCode"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Regulatory Module</B><BR>
This track shows predicted transcription factor binding sites based on sequence 
similarities upstream of coordinately expressed genes. 
Clicking on a particular predicted binding site brings up a page which indicates the sequence motif associated with the predicted transcription factor, and the sequence at the predicted binding site. Where known motifs have been identified by this method, they are named, otherwise they are assigned a Motif_ number. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	In dense display mode, gold areas indicate the extent of the area 
	searched for binding sites; black boxes indicate the actual binding 
	sites. In fuller display modes, the gold areas disappear and only the 
	binding sites are displayed. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode provides positional and
	score information about the item, and a link to display the DNA
	underlying the feature. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The user can set the minimum unnormalized score criteria that data must 
	meet to be displayed in the track.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>esRegGeneToMotif</I> positional table in the Table Browser. To view 
	the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#EsRegGeneToMotif"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>RepeatMasker</B><BR>
The track shows dispersed repeats as determined by 
<A HREF="http://www.repeatmasker.org" TARGET=_blank>RepeatMasker</A>
using the
<A HREF="http://www.girinst.org/Repbase_Update.html"TARGET=_blank>Repbase Update</A>
library of repetitive sequences from the Genetic Information Research 
Institute.  
These elements include SINE, LINE, LTR, DNA, simple, low complexity, 
micro-satellite, tRNA, and other repeat families. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Darker gray boxes indicates a more recent age (as dated by alignment to a 
	canonical repeat). In dense mode for large chromosomal regions, the black 
	boxes representing repeats may merge into misleading single blocks in the 
	display. Full display mode shows 9 tracks representing the major repeat 
	classes.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>rmsk</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#RepeatInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Retroposed Genes</B><BR>
This track shows processed mRNAs that have been inserted back into the genome after the 
mouse/human split. Retrogenes can be functional genes that have acquired a promoter 
from a neighboring gene, non-functional pseudogenes, or transcribed pseudogenes. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows the display conventions used by gene prediction tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show detailed information
	about the pseudogene's characteristics, alignment links, and Genome Browser
	annotation links for the gene locus that spawned the retrogene.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>pseudoGeneLink</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PseudoGeneLink"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>RGD ESTs </B><BR>
This track shows alignments expressed sequence tags (ESTs) from the Rat Genome
Database (RGD).

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for EST tracks. As with other
	EST tracks, it may be necessary to zoom in to display entries in full
	display mode due to the 300 line display limit.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an individual EST entry will show basic track information,
	the accession number, GenBank highlights, links to EST/genomic
	alignments, and an external link to the RGD EST report.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>rgdEst</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>RGD Genes</B><BR>
This track shows RefSeq genes curated by the Rat Genome Database (RGD). 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows the display conventions used by gene prediction tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will display position 
	information, links to sequence displays, and links to outside sources
	showing additional information. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter that can be used to turn
	on codon coloring for the track. Click the &quot;Codon coloring help&quot;
	link on the page for more information about this feature.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>rgdGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>RGD QTL</B><BR>
This track shows quantitative trait loci (QTLs) curated by the Rat Genome 
Database (RGD). 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	Loci positions are represented by blocks.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will display genomic 
	information and a link to the corresponding entry in the RGD. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>rgdQtl</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>RGD SSLP</B><BR>
This track shows simple sequence length polymorphisms (SSLPs), also known as
microsatellite DNA, from the Rat Genome Database (RGD). SSLPs consist of 
1 - 6 simple nucleotide repeat sequences, highly polymorphic in repeat length 
among strains. They are often used as genetic markers for genotyping.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	SSLP positions are represented by thin blocks.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will display genomic 
	information and a link to the corresponding entry in the RGD. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>rgdSslp</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Rinn Sex Exp</B><BR>
This track shows gene expression differences between adult male and female tissues as 
described in Rinn, JL et al. (2004) 
<A HREF="http://www.developmentalcell.com/content/article/abstract?uid=PIIS1534580704001649" 
TARGET=_blank>Major Molecular Differences between Mammalian Sexes Are Involved in Drug 
Metabolism and Renal Function</A>. <em>Developmental Cell</em> 6:791-800.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track follows display conventions similar to PSL alignment tracks.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	In dense or packed display mode, the track shows averages of related
	tissues. In full display mode, all tissues are displayed. Click on an 
	individual tissue entry to show a gene expression map for all tissues in 
	the experiment. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>rinnSex</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>RNA Genes</B><BR>
This track shows human non-protein-coding RNA genes and RNA pseudogenes. This 
includes tRNAs, rRNAs, SRPs, methylation guide snoRNAs, U2-U6 RNA-like, and 
similar elements both known and predicted (by tRNAscanSE etc). The density of 
these elements can range from 1-4 per million bp in some chromosomes. Protein 
coding genes are displayed elsewhere in the RefSeq Genes track.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	RNA genes are represented by light-colored blocks. Pseudogenes are
	represented by darker blocks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will display feature 
	classification, method of prediction, location, reliability score, and
	a link to DNA sequence for the feature. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>rnaGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#RNAgenes"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Rosetta</B><BR>
This track displays microarray tissue 
<A HREF="http://www.ncbi.nlm.nih.gov:80/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11237012&dopt=Abstract" TARGET=_blank>
expression data</A> provided by <A HREF="http://www.rii.com" TARGET=_blank>Rosetta Inpharmatics</A>. 
It helps catalogue genes and transcriptome, complements computational gene 
predictions, and seeks to define gene boundaries via co-regulated expression 
of nearby exons. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	The data is stratified by various tissue and disease-specific 
	conditions of gene expression.  Confirmed and predicted exons are placed in 
	separate tracks. Click on the mini-button to the left of the track to
	display a track description page containing more information.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter to fine-tune the track 
	display. The following display options are configurable:
	<BLOCKQUOTE>
		o Reference Sample: Determines which of the 69 different
		experiments are displayed, based on the reference sample
		used.<BR> 
		o Exons Shown: Determines whether data is shown for probes 
		corresponding to confirmed, predicted, or all exons.<BR> 
		o Color Scheme: Data are presented using two color false display. 
		The default colors can be changed for those who are colorblind. Gray 
		values indicate missing data.<BR> 
	</BLOCKQUOTE></TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on a Rosetta exon in full display mode will show an averaged display 
	that can be expanded via checkboxes to a graph of actual intensities for an 
	individual experiment.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>rosetta</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="S_tracks"></A>

<P><B>Sanger 22 and Sanger 22 Pseudo</B><BR>
These tracks contain the Sanger Centre annotations of chromosome 22. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on the track while in full display mode will show details of
	the methods used to construct this track as well as the contributors.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>sanger22</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>S. cerevisiae ESTs</B><BR>
See the Human ESTs track description.

<P><B>S. cerevisiae mRNAs</B><BR>
See the Human mRNAs track description.

<P><B>Scaffolds</B> (Chimp)<BR>
This track shows the chromosome-based version of the November 2003 Arachne chimp assembly 
(panTro1) from the Chimp Genome Sequencing Consortium. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	In dense display mode, this track depicts in alternating gold and brown the 
	individual scaffolds used to create the assembled sequence. 
	Where gaps exist in the path, spaces are shown between the gold and brown blocks. 
	Relative order and orientation of the scaffolds, as determined from the human/chimp 
	alignments, is implied for the non-random chromosomes.  The display must be 
	sufficiently zoomed in to view the gap features.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode will show information about the
	scaffold fragment and bases and position information.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>gold</I> positional table in the Chimp Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GpInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Scaffolds</B> (Fugu, Zebrafish, Tetraodon)<BR>
This track shows the draft and finished clones that comprise the assembly. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	In dense display mode, this track depicts in alternating gold and brown the 
	individual clones used to create the assembled sequence. 
	Where gaps exist in the path, spaces are shown between the gold and brown blocks. 
	If the relative order and orientation of the contigs between the two blocks is 
	known, a line is drawn to bridge the blocks.  The display must be 
	sufficiently zoomed in to view the gap features.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode will show contig and position information.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>gold</I> positional table in the Fugu or Zebrafish Table Browser. To view 
	the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GpInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Segmental Duplications</B><BR>
This track shows regions detected as putative genomic duplications within the 
golden path. 
For a description of the 'fuguization' detection method,
see Bailey JA et. al. (2001). 
"<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11381028&dopt=Abstract" TARGET=_blank>Segmental duplications: organization and impact within the current human genome 
project assembly</A>". <em>Genome Res</em> 11:1005-17. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Orange, yellow, dark-light gray represent
	similarities of greater than 99%, 99-98% and 98-90% respectively. Duplications 
	greater than 98% similarity that lack sufficient SDD evidence (likely 
	missed overlaps) are shown as red. Cut off values were at least 1 kb of 
	total sequence aligned (containing at least 500 bp non-RepeatMasked 
	sequence) and at least 90% sequence identity. </TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode provides position, filter,
	and alignment information as well as links to the corresponding genomic
	sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>genomicSuperDups</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenomicSuperDups"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Self Chain </B><BR>
This track shows alignments of the selected genome with itself using blastz and
axtChain. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for chain tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode shows position information, the 
	chain ID and score, and provides 
	links to view details of parts of the chain or to open the browser 
	to the corresponding aligning position. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>chainSelf</I> positional table in the Table Browser. To view the 
	table definition, click <A
	HREF="../gbdDescriptions.html#Chain"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>SGD Genes </B><BR>
This track shows annotated genes and open reading frames (ORFs) of 
<em>Saccharomyces cerevisiae</em> obtained from the Saccharomyces Genome 
Database (<A HREF="http://www.yeastgenome.org/" TARGET=_blank>SGD</A>).

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for known genes and gene 
	predictions tracks.  </TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode provides a wealth of 
	genomic, mRNA, and protein information about the gene as well as 
	links to several associated external sites. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>sgdGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>SGD Other </B><BR>
This track shows a variety of features in the <em>Saccharomyces cerevisiae</em>
genome, including tRNAs, transposons, centromeres, and open reading frames (ORFs) classified as dubious. 
The data was downloaded from the Saccharomyces Genome Database 
(<A HREF="http://www.yeastgenome.org/" TARGET=_blank>SGD</A>). 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for known genes and gene 
	predictions tracks.  </TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode provides genomic information 
	and a link to view the DNA underlying the track.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>sgdOther</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>SGP Genes </B><BR>
This track shows gene predictions from the SGP program, which is being developed 
at the Grup de Recerca en Informtica Biomdica (GRIB) at Institut Municipal 
d'Investigaci Mdica (IMIM) in Barcelona. To predict genes in a genomic query, 
SGP combines geneid predictions with tblastx comparisons of the genomic query 
against other genomic sequences.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for known genes and gene 
	predictions tracks.  </TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode provides position information
	as well as links to the corresponding protein, mRNA, genomic, and 
	comparative sequences.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>sgpGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Short Match</B><BR>
This track shows all occurrences of a selected short motif within the displayed 
position range of the assembly sequence. It is useful for finding 
oligonucleotides, restriction sites, or other recurring short sequences within 
the assembly. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Perfect matches are indicated by a solid rectangle. In full display mode,
	each occurrence is labeled by the strand on which the match is located,
	followed by the starting coordinate of the match. In cases where the 
	input motif sequence is identical to its reverse complement, only the 
	match on the &quot;+&quot; strand is shown.
	</TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode provides position information
	about the match.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track may be configured to search for any short sequence of 2-30 
	bases in length. To change the sequence, type a new sequence into the 
	text box at the top of the track description page.
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	Unlike most annotations in the Genome Browser, this track is not 
	table-based.</TD></TR>
</TABLE></P>

<P><B>Simple Repeats</B><BR>
This track displays simple tandem repeats (possibly imperfect) located by 
<A HREF="http://tandem.bu.edu/trf/trf.html" TARGET=_blank>Tandem Repeats Finder</A>, 
which is specialized to this purpose. These repeats can occur within coding regions of 
genes and may be quite polymorphic. Repeat expansions are sometimes associated with 
specific diseases.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	Repeats are represented by block boxes that indicate location in the
	assembly.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an item in full display mode will show the simple repeat 
	sequence, copy number, and quality of match statistics.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>simpleRepeats</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#SimpleRepeats"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Slam Human, Slam Mouse, Slam Rat</B><BR>
These tracks predict coding exons and conserved noncoding regions in a pair of 
homologous DNA sequences, incorporating both statistical sequence properties and 
the degree of conservation in making the predictions. The model is symmetric: the 
same gene structure (with possibly different exon lengths) is predicted in both 
sequences. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show basic track
	information, plus links to the predicted mRNA, 
	genomic sequence, and comparative sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the appropriate
	<I>slam</I> positional table in the Table Browser (<I>e.g. slamHuman, 
	slamMouse, slamRat</I>). To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>SlamHumanNonCoding, SlamMouseNonCoding, SlamRatNonCoding</B><BR>
These tables show slam predictions of conserved noncoding regions in homologous 
DNA sequences.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	The data in these tables are displayed as part of the Slam Human, 
	Slam Mouse, and Slam Rat tracks. See the track descriptions for these
	tracks for display information.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the appropriate
	<I>slamNonCoding</I> positional table in the Table Browser (<I>e.g. slamHumanNonCoding, 
	slamMouseNonCoding, slamRatNonCoding</I>). To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>SNAP Genes</B> (<em>C. intestinalis</em>)<BR>
This track shows gene predictions determined by SNAP (Semi-HMM-based Nucleic Acid Parser),
a general purpose gene-finding program written by Ian Korf. SNAP is suitable for both 
eukaryotic and prokaryotic genomes. For this set of gene predictions, it was trained on 
gene annotations from the DOE Joint Genome Institute (JGI).

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode brings up basic track
	information, plus links to the translated protein, predicted mRNA, and
	genomic sequences.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>snapGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredExt"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Sno/miRNA</B><BR>
This track displays positions of four different types of RNA in the human genome: 
microRNAs from the miRNA Registry at the Wellcome Trust Sanger Institute, small nucleolar 
RNAs (C/D box and H/ACA box snoRNAs) and Cajal body-specific RNAs (scaRNAs) from the 
snoRNA-LBME-DB maintained at the Laboratoire de Biologie Molculaire Eucaryote. C/D box 
and H/ACA box snoRNAs are guides for the 2'O-ribose methylation and the pseudouridilation,
respectively, of rRNAs and snRNAs, although many of them have no documented target RNA. 
The scaRNAs guide modifications of the spliceosomal snRNAs transcribed by RNA polymerase 
II, and often contain both C/D and H/ACA domains. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Mature miRNAs (miRs) are represented by thick red blocks; the predicted stem-loop 
	portions of the primary transcripts are indicated by thinner blocks. When a 
	single precursor produces two mature miRs from its 5' and 3' parts, both positions
	of the mature miRNA are individually displayed. C/D box, H/ACA box snoRNAs and 
	scaRNAs are represented by blue, green and magenta blocks, respectively. At a 
	zoomed-in resolution, arrows superimposed on the blocks indicate the sense 
	orientation of the snoRNAs.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode provides a link to the miRNA
	Registry, the RNA type, position information, and a link to the DNA sequence
	associated with the feature.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>wgRna</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#WgRna"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>SNPs</B> (Human assemblies hg17 and later) <BR> 
This track displays all the known Simple Nucleotide Polymorphisms (SNPs) that
can be mapped against the current genome assembly. These data were obtained from
<A HREF="http://www.ncbi.nlm.nih.gov/SNP/" TARGET=_blank>dbSNP</A> and 
commercially-available genotyping arrays and include known point 
mutations (Single Nucleotide Polymorphisms), insertions, deletions, multiple
See the track description page for more information about the SNP 
classifications.
These data were provided by the <A HREF="http://snp.cshl.org/" 
TARGET=_blank>SNP Consortium</A> and 
<A HREF="http://www.affymetrix.com/index.affx"
TARGET=_blank>Affymetric, Inc.</A>. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	Variants are drawn a single tick marks at most zoom levels. 
	When viewing the track at or near base-level resolution, the displayed 
	width of the SNP corresponds to the width of the variant in the 
	reference sequence. Insertions are indicated by a single tick mark 
	displayed between two nucleotides, single nucleotide polymorphisms are 
	displayed as the width of a single base, and multiple nucleotide 
	variants are represented by a block that spans two or more bases. 
</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
 	Clicking in full mode brings up position information about the reference
	sequence, detailed information about the variant, and a link to the
	record in dbSNP.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The colors of variants may be changed to highlight their source, 
	molecule type, variant class, validation status, or functional 
	classification. Variants can be excluded from the display based on 
	these same criteria or if they fall below the user-specified minimum 
	average heterozygosity. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>snp</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#Snp"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>SNPs</B> (Human assemblies prior to hg17) <BR> 
This track displays all the known Simple Nucleotide Polymorphisms (SNPs) that
can be mapped against the current genome assembly. These include known point 
mutations (Single Nucleotide Polymorphisms), insertions, deletions, and 
segmental mutations from the current build of dbSnp (shown in the Genome
Browser <A HREF="/goldenPath/releaseLog.html">release log</A>). See the track
description page for more information about major cases that are not mapped.
These data were provided by the <A HREF="http://snp.cshl.org/" TARGET=_blank>SNP 
Consortium</A>.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> 
	SNP locations are indicated by thin grey vertical lines.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
 	Clicking in full mode brings up a location description 
 	and <A HREF="http://www.ncbi.nlm.nih.gov/SNP/" TARGET=_blank>dbSNP</A> link.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>snpMap</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#SnpMap"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Spliced ESTs</B><BR>
The Spliced EST track displays the subset of Expressed Sequence Tags (ESTs)
from GenBank that show signs of splicing when aligned against
the genome. ESTs are single read (typically approximately 500 base) sequences 
that usually represent fragments of transcribed genes. By requiring splicing, 
the level of contamination in the EST databases is drastically reduced at the 
expense of eliminating many genuine 3' ESTs. This track may also suggest 
alternate splices. The GT-AG splice donor/acceptor rule has not been applied.  
For a display of all ESTs (including unspliced), see the Human EST 
track.
This track was generated by aligning human ESTs from GenBank against the
genome using the BLAT program. Note that the maximum intron length
allowed by BLAT is 500,000 bases, which may eliminate some ESTs with
very long introns that might otherwise align. When a single EST aligns
in multiple places, the alignment having the highest base identity is
found. Only alignments that have a base identity level within 1% of the
best are kept. Alignments must also have at least 93% base identity to be 
kept.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for EST tracks 
	As with other EST tracks, it may be necessary to zoom in to 
	display entries in full display mode due to the 300 line display 
	limit.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an individual EST entry will show basic track information,
	the accession number, GenBank highlights, and links to translated
	protein, predicted mRNA, and genomic sequences.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>intronEst</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>STS Markers (human)</B><BR>
This track shows the locations of Sequence Tagged Site (STS) markers along
the draft assembly. The STS markers have been mapped using either
genetic (Genethon and Marshfield maps), radiation hybridization (the Stanford,
Whitehead RH, and GeneMap99 maps) or YAC mapping (the Whitehead YAC map)
techniques. In releases prior to August 2001, this track also shows the 
approximate position of FISH mapped clones. Starting with the August 2001 assembly, 
the FISH clones have been moved into a separate track. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Genetic map markers are shown in blue, and radiation hybrid
	map markers are shown in black. When a marker maps to multiple
	positions in the genome, it is shown in a lighter color.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on a specific marker in full display mode will show synonyms, 
	position information, primers, YAC map position, and uniSTS and GDB links.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>stsMap</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#STSMap"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>STS Markers (mouse)</B><BR>
This track shows the locations of Sequence Tagged Site (STS) markers
along the draft assembly of the mouse genome. These STS markers appear on the MGI consensus
mouse genetic map. Information about the genetic map and STS marker
primer sequences are provided by the <A HREF="http://www.informatics.jax.org/"
TARGET=_blank> Mouse Genome Informatics</A> database group at The Jackson 
Laboratory.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Genetic map markers are shown as thin black boxes. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on a specific marker in full display mode will show position 
	information, primers, genetic map position, genomic alignments, and MGI
	links.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>stsMapMouse</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#STSMap"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>STS Markers (rat)</B><BR>
This track shows locations of Sequence Tagged Site (STS) markers along the draft 
assembly of the rat genome. These STSs have been mapped using either
genetic (rat FHH x ACI F2 intercross genetic map, rat SHRSP x BN F2 intercross genetic map) or radiation hybridization (RH
map.2.2).

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Genetic map markers are shown as thin black boxes. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on a specific marker in full display mode will show position 
	information, primers, genetic map position, and links to the corresponding
	UniSTS and RGD entries. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>stsMapRat</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#STSmapRat"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Superfamily </B><BR>
The Superfamily track shows proteins having homologs with known structures or 
functions. Each entry in the track shows the coding region of a gene (based on 
Ensembl gene prediction). The label, shown in Full mode at the left
hand side of each entry, consists of the names of all known protein domains 
coded by this gene. This usually contains structural and/or function descriptions
that provide valuable information for getting a quick grasp of the biological 
significance for the gene. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Coding regions are indicated by red blocks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show the Superfamily 
	domains of the corresponding protein and provide a link to the Superfamily	  database entry.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>superfamily</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#BED"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="T_tracks"></A>

<P><B>Tetraodon Chain </B><BR>
This track shows Tetraodon genomic alignments to the selected 
genome using blastz and axtChain. See the Human Chain track description for 
more information.

<P><B>Tetraodon Ecores</B><BR>
This track shows Tetraodon evolutionary conserved regions (ecores) computed 
against the selected assembly genome by the Exofish program at Genoscope. For 
information on this track, see the Human Ecores track description.

<P><B>Tetraodon Net </B><BR>
This track shows the best Tetraodon chain for every part of the selected genome.
See the Human Net track description for more information.

<P><B>TFBS Conserved</B><BR>
This tracks shows the location and score of transcription factor binding sites 
(TFBSs) conserved in a multiple species alignment. A binding site is considered 
to be conserved across the alignment if its score meets the threshold score for 
that binding site in all aligned species. The score and threshold are computed 
with the Transfac Matrix Database created by Biobase. The data are purely 
computational, and as such not all binding sites listed here are biologically 
functional binding sites.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B> TFBSs are represented by thin blocks.
	The darker the shading, the better the match of the binding site. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on a block in full display mode will show the the entry's 
	Transfac ID, positional information, and a link to its Transfac Matrix
	(free registration with Transfac is required).
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>tfbsCons</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#tfbsCons"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Tight Mouse</B><BR>
This track displays <I>Blastz</I> alignments of the specified mouse
draft assembly to the human genome, filtered by <I>axtBest</I> and
<I>subsetAxt</I> with very stringent constraints. See the track's description
page for more details. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for cross-species synteny 
	tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show human and 
	mouse position information, alignment details, and a link to a detailed
	view of parts of the alignment.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter that can be used to turn 
	on the chromosome color track or to filter the display output by 
	chromosome.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>blastzTightMouse</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Tight Rat</B><BR>
This track displays <I>Blastz</I> alignments of the Nov. 2002 rat
draft assembly to the mouse genome, filtered by <I>axtBest</I> and
<I>subsetAxt</I> with very stringent constraints. See the track's description
page for more details. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for cross-species synteny 
	tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show mouse and 
	rat position information, alignment details, and a link to a detailed
	view of parts of the alignment.  </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter that can be used to turn 
	on the chromosome color track or to filter the display output by 
	chromosome.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>blastzTightRat</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Tigr Gene Index</B><BR>
This track displays alignments of the 
<A HREF="http://www.tigr.org/tdb/hgi/" TARGET=_blank>TIGR Gene Index</A> (TGI) 
against the human genome. The TGI is based largely on assemblies of EST
sequences from the public databases and TIGR.  Entries are 
actual assemblies providing a consensus sequence, not simply clusters of 
overlapping sequences, but generally don't predict whole genes. Gene
indices for mouse, rat, cow, and pig also contribute to this track. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show basic track
	information, links to the  predicted mRNA and
	genomic sequences, and a link to supporting information in the TIGR
	Human Gene Index.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>tigrGeneIndex</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Transcriptome</B><BR>
This track shows transcriptome data for chromosomes 21 and 22 from Affymetrix, 
as described in "Large-Scale Transcriptional Activity in Chromosomes 21 and 22", 
Kapranov, P., Cawley, S. E., Drenkow, J., Bekiranov, S, Strausberg, R. L., 
Fodor, S.P.A. and Gingeras, T.R.. In general, the data presented is the perfect 
match - mismatch value. Different experiments were normalized by setting the 
average value to be the same for each chip. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for wiggle tracks. To present a 
	more interpretable display when zoomed out, averages have been 
	precalculated over the chromosome at two different resolutions in 
	addition to the raw data. For example, when zoomed out, there may appear 
	to be a peak at the center of a gene rather than a signal at every exon. 
	Zooming in will reveal the raw data for that region.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	<B><I>Track details:</I></B> 
	In dense mode (or at the top of the track in full mode), the 
	transcriptome wiggle track displays the maximum value over all 
	experiments for that probe, the idea 
	being to paint as many transcribed regions as possible. In full mode, 
	the track displays a separate wiggle track for each cell type, showing 
	the data produced by averaging together replicates for that cell type.
	Clicking on an individual data point will show positional information and
	a link to view the sequence for the feature.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>affyTranscriptome</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#sample9"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Twinscan Gene Predictions</B><BR>
This track displays Twinscan gene predictions. Twinscan predicts genes in a 
manner similar to Genscan, except that Twinscan takes advantage of genome 
comparison to improve gene prediction accuracy. More information and a web 
server can be found at the Washington University St. Louis 
<A HREF="http://genes.cs.wustl.edu/" TARGET=_blank>Twinscan</A> web page.

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene predictions tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show basic track
	information, links to the  predicted protein, predicted mRNA, 
	genomic sequence, and comparative sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>twinscan</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="U_tracks"></A>

<P><B>UCSC Synteny (hg12)</B><BR>
See description for Mouse Synteny (hg13+).
</P>

<P><B>UniGene</B><BR>
This track shows <A HREF="http://www.ncbi.nlm.nih.gov/UniGene/" TARGET=_blank>UniGene</A> 
clusters of ESTs aligned to the genome using BLAT. This data imperfectly 
indicates exon/intron boundaries and to a lesser extent genes. The Serial 
Analysis of Gene Expression <A HREF="http://www.ncbi.nlm.nih.gov/SAGE/" TARGET=_blank>
(SAGE)</A> project at NCBI - a quantitative measurement of gene expression in 
various tissues and disease states - mapped small 10bp bar-coded expression 
markers onto these UniGene clusters in the SageMap project. The Genome Browser 
uses this mapping to provide access to the SAGE data associated with the 
UniGene track in the assembled and annotated genome. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for PSL alignment tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on a UniGene feature will show an elaborate display of tabular and 
	graphical data from the SAGE project relevant to that cluster.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>uniGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="V_tracks"></A>

<P><B>Vega Genes </B>(selected human chromosomes only)<BR> 
This track shows genes from the <A HREF="http://vega.sanger.ac.uk/" TARGET=_blank>Vertebrate Genome Annotation (VEGA) database</A>, a central repository for manual annotation of different vertebrate finished genome sequence. In collaboration with the genome sequencing centers, Vega attempts to present consistent high-quality curation of the published chromosome sequences. Finished genomic sequence is analysed on a clone by clone basis using a combination of similarity searches against DNA and protein databases as well as a series of <I>ab initio</I> gene predictions (GENSCAN, Fgenes). In addition, comparative analysis using vertebrate datasets such as the Riken mouse cDNAs and Genoscope Tetraodon nigroviridis Ecores (Evolutionary Conserved Regions) are used for novel gene discovery.
 
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene prediction tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show basic track
	information, a link to the Vega web page for the item, and links to the 
	predicted protein, predicted mRNA, 
	genomic sequence, and comparative sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>vegaGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>Vega Pseudogenes </B>(selected human chromosomes only)<BR> 
This track shows annotated pseudogenes and immunoglobulin segments from the <A HREF="http://vega.sanger.ac.uk/" TARGET=_blank>Vertebrate Genome Annotation (VEGA) database</A>, a central repository for manual annotation of different vertebrate finished genome sequence. In collaboration with the genome sequencing centers, Vega attempts to present consistent high-quality curation of the published chromosome sequences. Finished genomic sequence is analysed on a clone by clone basis using a combination of similarity searches against DNA and protein databases as well as a series of <I>ab initio</I> gene predictions (GENSCAN, Fgenes). In addition, comparative analysis using vertebrate datasets such as the Riken mouse cDNAs and Genoscope Tetraodon nigroviridis Ecores (Evolutionary Conserved Regions) are used for novel gene discovery.
 
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene prediction tracks.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show basic track
	information, a link to the Vega web page for the item, and links to the 
	predicted mRNA, genomic sequence, and comparative sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>vegaPseudoGene</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="W_tracks"></A>

<P><B>WashU Clones</B><BR> 
This track displays the location of clones (mostly lambda and cosmid clones) 
from Washington University in St. Louis using the names assigned by that group. 
The data was downloaded from the Saccharomyces Genome Database 
(<A HREF="http://www.yeastgenome.org/" TARGET=_blank>SGD</A>). 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	The clone positions are shown in black. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode brings up genomic 
	information and a link to the American Type Culture Collection (ATCC) 
	website.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>sgdClone</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#SgdClone"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>WormBase Genes</B><BR> 
This track shows Sanger Gene predictions from the Wormbase files 
downloaded from the Sanger Institute FTP site. The data shown is the subset
annotated with "curated", "DNA", or "RNA" followed by one of these strings: 
"intron", "exon", "cds", "sequence", or "transcri". 
 
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene prediction tracks. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show a wealth of 
	information about the gene and its associated mRNA and protein, as
	well as links to many related external sites.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter that can be used to turn
	on codon coloring for the track. Click the &quot;Codon coloring 
	help&quot; link on the page for more information about this feature.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>sangerGene</I> positional table in the Table Browser. To view 
	the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>WormBase Genefinder </B><BR> 
This track shows Sanger Gene predictions from the Wormbase files 
downloaded from the Sanger Institute FTP site. The data shown is the subset
annotated with the string "Genefinder" and found by the gene prediction 
program, Genefinder.
 
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene prediction tracks. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show genomic
	information  and links to the 
	translated protein, predicted mRNA, genomic sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains a filter that can be used to turn
	on codon coloring for the track. Click the &quot;Codon coloring 
	help&quot; link on the page for more information about this feature.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>sangerGenefinder</I> positional table in the Table Browser. To view 
	the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>WSSD Coverage</B><BR> 
This track represents coverage of clones that were assayed for segmental duplications 
using high-depth Celera reads. Absent regions were not assessed by this version of the 
Segmental Duplication Database (SDD). For a description of the whole-genome shotgun 
sequence detection (WSSD) &quot;fuguization&quot; method, see Bailey, JA. <em>et al.</em>,
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11381028&dopt=Abstract"
TARGET=_blank>Segmental duplications: organization and impact within the current human 
genome project assembly</A>, <em>Genome Res.</em> <B>11</B>(6), 1005-17 (2001). 
 
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Rectangular blocks indicate the positions of clones.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show position information and a 
	link to the underlying DNA for the feature.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>celeraCoverage</I> positional table in the Table Browser. To view 
	the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#CeleraCoverage"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>WSSD Duplication </B><BR> 
High-depth sequence reads from the Celera project were used to detect paralogy in the 
human genome reference sequence. This track shows confirmed segmental duplications, 
defined as having similarity to sequences in the Segmental Duplication Database (SDD) 
of greater than 90% over more than 250 bp of repeatmasked sequence. For a description 
of the whole-genome shotgun sequence detection (WSSD) 'fuguization' method, see 
Bailey, JA. <em>et al.</em>, 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11381028&dopt=Abstract"
TARGET=_blank>Segmental duplications: organization and impact within the current human 
genome project assembly</A>, <em>Genome Res.</em> <B>11</B>(6), 1005-17 (2001). 
 
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Rectangular blocks indicate the positions of duplications.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show the WSSD ID number, 
	position information, and a link to the underlying DNA for the feature.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>celeraDupPositive</I> positional table in the Table Browser. To view 
	the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#CeleraDupPositive"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>WSSD Overlay</B><BR> 
This track shows regions detected as overlays of Celera whole-genome shotgun sequence 
assembly on the public human assembly. 
 
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	Regions are represented by rectangular blocks. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show position information and a 
	link to the underlying DNA for the feature.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>celeraOverlay</I> positional table in the Table Browser. To view 
	the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#CeleraOverlay"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>


<A NAME="X_tracks"></A>

<P><B><em>X. tropicalis</em> Chain </B><BR>
This track shows <em>X. tropicalis</em> genomic alignments to the selected 
genome using blastz and axtChain. See the Human Chain track description for 
more information.

<P><B><em>X. tropicalis</em> ESTs </B><BR>
See the description for the Human ESTs track.

<P><B><em>X. tropicalis</em> mRNAs</B><BR>
See the description for the Human mRNAs track.

<P><B><em>X. tropicalis</em> Net </B><BR>
This track shows the best <em>X. tropicalis</em> chain for every part of the selected genome. See 
the Human Net track description for more information.

<A NAME="Y_tracks"></A>

<P><B>Yale Pseudo</B><BR> 
This track shows pseudogenes identified in the 
<A HREF="http://www.pseudogene.org" TARGET=_blank>Yale Pseudogene 
Database</A>. 
 
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for gene prediction tracks. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show genomic
	information, a link to the corresponding pseudogene entry in the 
	Yale Pseudogene Database, and links to predicted mRNA and genomic 
	sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page provides an option to turn
	on codon coloring for the track. Click the &quot;Codon coloring 
	help&quot; link on the page for more information about this feature.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>pseudoYale</I> positional table in the Table Browser. To view 
	the table definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<A NAME="Z_tracks"></A>

<P><B>Zebrafish Chain </B><BR>
This track shows zebrafish genomic alignments to the selected 
genome using blastz and axtChain. See the Human Chain track description for 
more information.

<P><B>Zebrafish ESTs</B><BR>
See the Human ESTs track description.

<P><B>Zebrafish mRNAs</B><BR>
See the Human mRNAs track description.

<P><B>Zebrafish Net </B><BR>
This track shows the best Zebrafish chain for every part of the selected genome.
See the Human Net track description for more information.

<P><B>Zfish WZ EST Clusters</B><BR> (Zebrafish)
This track shows alignments between zebrafish WZ expressed sequence tags (ESTs) from 
WashU-Zebrafish Genome Resources and the genome. These WZ ESTs are compiled to produce 
longer reads by clustering together ESTs that originate from the same transcript. 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for EST tracks.  Hatch marks on the 
	introns indicate the direction of transcription. 
	As with otherEST tracks, it may be necessary to zoom 
	in to display entries in full display mode due to the 300 line display 
	limit.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an individual EST entry displays a link to the genomic alignment. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>wz_ests</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#PslInfo"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>ZGC Genes </B> (Zebrafish)<BR>
This track shows alignments to the genome of zebrafish mRNAs from the Mammalian Gene 
Collection (MGC) having full-length open reading frames (ORFs). 

<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for known genes and gene 
	predictions tracks.  </TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display provides links to the 
	corresponding protein, mRNA, and genomic sequences, and related links to 
	outside databases. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page offers configuration options that can be used to turn
	on codon coloring for the track. Click the &quot;Codon coloring 
	help&quot; link on the page for more information about this feature.
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>mgcGenes</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#GenePredictions"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

<P><B>ZooSeq </B>(Zoo)<BR> 
This track displays the level of conservation for each zoo species based on 
blastz pair-wise alignments.
 
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Display conventions:</I></B>
	This track uses the display conventions for wiggle tracks. 
	Gray horizontal bars denote a mapping/sequencing gap. Percent identities 
	of less than 50% are not shown because they fall below the minimum
	y-axis value. </TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track configuration:</I></B> 
	The track description page contains options to configure 
	several of the graph characteristics. To read more information
	about each option, click the &quot;Graph configuration help&quot; link. 
	</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Track details:</I></B> 
	Clicking on an entry in full display mode will show basic track
	information, a link to the Vega web page for the item, and links to the 
	predicted mRNA, genomic sequence, and comparative sequence.</TD></TR>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<B><I>Table details:</I></B> 
	To view a text version of the data set, select the 
	<I>zoo</I> positional table in the Table Browser. To view the table
	definition, click <A
	HREF="http://genome.ucsc.edu/goldenPath/gbdDescriptions.html#WiggleTrack"
	TARGET=_blank>here</A>.</TD></TR>
</TABLE></P>

	</TD><TD WIDTH=15></TD></TR></TABLE>
    <br></TD></TR></TABLE> </TD></TR></TABLE>

</TD></TR></TABLE>

</BODY></HTML>

