<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: User Guide</TITLE>
	<LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
      <script type="text/javascript" src="/js/jquery.js"></script>
	  <script type="text/javascript" src="/js/staticDoc.js"></script>
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../inc/help.topbar.html"--></TD></TR>

<!--Content Tables----------------------------------------------------->  
<TR><TD ROWSPAN=2 VALIGN="top"></TD></TR>
<TR><TD>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp;Genome Browser User Guide </B></FONT> 
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

<B><FONT SIZE=4> Table of Contents: </FONT></B>
<TABLE BORDER=0 cellpadding=0 cellspacing=0><TR><TH HEIGHT=10></TH></TR>
<TR><TD><UL><B>
  		<LI><A CLASS="toc" HREF="#What">
			What does the Genome Browser do?</A>
  		<LI><A CLASS="toc" HREF="#GetStarted">
			Getting started: Genome Browser gateways</A>
  		<LI><A CLASS="toc" HREF="#FineTuning">
			Fine-tuning the Genome Browser display</A>
  		<LI><A CLASS="toc" HREF="#IndivTracks">
			Annotation track descriptions</A>
  		<LI><A CLASS="toc" HREF="#BLATAlign">
			Using BLAT alignments</A>
  		<LI><A CLASS="toc" HREF="#TableBrowser">
			Getting started on the Table Browser</A>
  		<LI><A CLASS="toc" HREF="#Sessions">
			Getting started using Sessions</A>
	</B></UL></TD><TD><UL><B>
		<LI><A CLASS="toc" HREF="#GenomeGraphs">
			Getting started on Genome Graphs</A>
		<LI><A CLASS="toc" HREF="#VisiGeneHelp">
			Using the VisiGene Image Browser</A>
		<LI><A CLASS="toc" HREF="#TrackFormatDNA">
			DNA text formatting</A>
		<LI><A CLASS="toc" HREF="#Convert">
			Converting data between assemblies</A>
		<LI><A CLASS="toc" HREF="#Download">
			Downloading genome data</A>
		<LI><A CLASS="toc" HREF="#CustomTracks">
			Creating and Managing custom annotation tracks</A>
	</B></UL></TD></TR></TABLE>
	<P>
        <FORM name="googleForm1" method="GET" action="http://www.google.com/search" onSubmit="document.googleForm1.q.value=document.googleForm1.qq.value+'   site:genome.ucsc.edu/goldenPath/help';">
        Search the Genome Browser help pages: &nbsp;
        <input type="hidden" name="q" value="">
        <input type="hidden" name="num" value="10">
        <input type="hidden" name="filter" value="0">
        <INPUT TYPE=text name=qq size=30 maxlength=255 value="">
        <input TYPE="submit" VALUE="Submit">
        </FORM>
        </p>
	<P>
 	<FORM name="googleForm2" method="GET" action="http://www.google.com/search" onSubmit="document.googleForm2.q.value=document.googleForm2.qq.value+'   site:lists.soe.ucsc.edu/pipermail/genome';">
 	Search the Genome mailing list archives: &nbsp;
 	<input type="hidden" name="q" value="">
 	<input type="hidden" name="num" value="10">
 	<input type="hidden" name="filter" value="0">
 	<INPUT TYPE=text name=qq size=30 maxlength=255 value="">
 	<input TYPE="submit" VALUE="Submit">
 	</FORM>
 	<P>
        See also the <A HREF="http://www.openhelix.com/downloads/ucsc/ucsc_home.shtml" TARGET=_blank>Open Helix tutorial and training materials</A>.
        </p><p>
        <A HREF="../../contacts.html">Questions and feedback are welcome</A>.
        </p>
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="What"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><a NAME="TOC"></a><b>&nbsp; What does
			the Genome Browser do?  </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
As vertebrate genome sequences near completion and research re-focuses on 
their analysis, the issue of effective sequence display becomes critical: it is not 
helpful to have 3 billion letters of genomic DNA shown as plain text! As an 
alternative, the UCSC Genome Browser provides a rapid and reliable display of any 
requested portion of genomes at any scale, together with dozens of 
aligned annotation tracks (known genes, predicted genes, ESTs, mRNAs, CpG islands, 
assembly gaps and coverage, chromosomal bands, mouse homologies, and more). Half of 
the annotation tracks are computed at UCSC from publicly available
sequence data. The remaining tracks are provided by collaborators worldwide. Users can 
also add their own custom tracks to the browser for educational or research
purposes.
<P>
The Genome Browser stacks annotation tracks beneath genome 
coordinate positions, allowing rapid visual correlation of different types of 
information. The user can look at a whole chromosome to get a feel for gene 
density, open a specific cytogenetic band to see a positionally mapped disease gene 
candidate, or zoom in to a particular gene to view its spliced ESTs and possible 
alternative splicing. The Genome Browser itself does not draw conclusions; rather, it 
collates all relevant information in one location, leaving the exploration and 
interpretation to the user.
<P>
The Genome Browser supports text and sequence based searches that provide quick, precise 
access to any region of specific interest. Secondary links from individual
entries within annotation tracks lead to sequence details and supplementary off-site 
databases.  To control information overload, tracks need not be displayed in full. Tracks
can be hidden, collapsed into a condensed or single-line display, or filtered according 
to the user's criteria.  Zooming and scrolling controls help to narrow or broaden the 
displayed chromosomal range to focus on the exact region of interest. 
Clicking on an individual item within a track opens a details page containing a 
summary of properties and links to off-site repositories such as 
PubMed, GenBank, Entrez, and OMIM. The page provides item-specific information on 
position, cytoband, strand, data source, and encoded protein, mRNA, genomic sequence 
and alignment, as appropriate to the nature of the track.
<P>
A blue navigation bar at the top of the browser provides links to several other tools 
and data sources. For instance, the DNA link
enables the user to view the raw genomic DNA sequence for the coordinate range 
displayed in the browser window. This DNA can encode track features via elaborate 
text formatting options. Other links tie the Genome Browser to the BLAT 
alignment tool, provide access to the underlying relational database via the
Table Browser, convert coordinates across different assembly dates, and open the 
window at the complementary <A HREF="http://www.ensembl.org/" TARGET=_blank>Ensembl</A>
or <A HREF="http://www.ncbi.nlm.nih.gov/mapview/" TARGET=_blank>NCBI Map 
Viewer</A> annotation.
<P>
The browser data represents an immense 
<A HREF="../credits.html">collaborative effort</A> 
involving thousands of people from the international biomedical research community. 
The UCSC Bioinformatics Group itself does no sequencing. Although it creates 
the majority of the annotation tracks in-house, the annotations are based on
publicly available data contributed by many labs and research groups throughout
the world. Several of the Genome Browser annotations are generated in 
collaboration with outside individuals or are contributed wholly by external
research groups. UCSC's
other major roles include building genome assemblies, creating the Genome Browser work 
environment, and serving it online. The majority of the sequence data, annotation tracks, and even 
software are in the public domain and are available for anyone
to <A HREF="#Download">download</A>. 
<P>
In addition to the Genome Browser, the UCSC Genome Bioinformatics group provides 
several other tools for viewing and interpreting genome data: 
<UL>
<LI><A HREF="../../cgi-bin/hgBlat?command=start">BLAT<A> - a 
fast sequence-alignment tool similar to BLAST. <A HREF="#BLATAlign">Read more</A>.
<LI><A HREF="../../cgi-bin/hgTables">Table Browser</A> - 
convenient text-based access to the database underlying the Genome Browser. 
<A HREF="hgTablesHelp.html">Read
more</A>.
<LI><A HREF="../../cgi-bin/hgGenome">Genome Graphs</A> - 
a tool that allows you to upload and display genome-wide data sets such as the 
results of genome-wide SNP association studies, linkage studies and 
homozygosity mapping.
<A HREF="hgGenomeHelp.html">Read more</A>.
<LI><A HREF="../../cgi-bin/hgNear">Gene Sorter</A> - 
expression, homology, and other information on groups of genes that can
be related in many ways. 
<A HREF="hgNearHelp.html">Read more</A>.
<LI>Proteome Browser (accessible from Known Genes details pages) - 
protein property data and links to a wealth of related information. 
<A HREF="pbTracksHelpFiles/pbTracksHelp.shtml">Read more</A>.
</UL>
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="GetStarted"></A><BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp; Getting Started: 
		Genome Browser gateways</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
The UCSC Genome Bioinformatics <A HREF="../../index.html" 
TARGET=_blank>home page</A>
provides access to Genome Browsers on several different genome assemblies. 
To get started, click the Browser link on the blue sidebar. This will take 
you to a Gateway page where you can select which genome to display.
<P>
<B>Opening the Genome Browser at a specific position</B><BR> 
To get oriented in using the Genome Browser, try viewing a gene or region of 
the genome with which you are already familiar, or use the default position. 
To open the Genome Browser window:
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top"><B>1.</B></TD><TD>
	Select the clade, genome and assembly that you wish to display from 
	the corresponding pull-down menus. To access older assembly versions 
	that are no longer available from the menu, look in the Genome 
	Browser <A HREF="http://genome-archive.cse.ucsc.edu/"
	TARGET=_blank>archives</A>.
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top"><B>2.</B></TD><TD>
	Specify the genome location you'd like the Genome Browser to open to.
	To select a location, enter a valid position query in the 
	<I>position or search term</I> text box at the top of the Gateway 
	page or accept the default position already 
	displayed. The search supports several different types of queries: gene 
	symbols, mRNA or EST accession numbers, chromosome 
	bands, descriptive terms likely to occur in GenBank text, or specific chromosomal 
	ranges. To display a region encompassed by two features on the same 
	chromosome, use a semi-colon, e.g. CRYBB3;CRYBB1. The Gateway page shows examples of some of the search requests supported by
	the Genome Browser.
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top"><B>3.</B></TD><TD>
	Click the <I>submit</I> button to open up the Genome Browser window to the
	requested location. In cases where a specific term (accession, gene name,
	etc.) was queried, the item will be highlighted in the display.
	</TD></TR>
</TABLE>
<P>
Occasionally the Gateway page returns a list of several matches in response 
to a search, rather than immediately displaying the Genome Browser 
window. When this occurs, click on the item in which you're interested and 
the Genome Browser will open to that location.
<P>
The search mechanism is not a site-wide search engine. Instead, it 
primarily searches GenBank mRNA records whose text annotations can include 
gene names, gene symbols, journal title words, author names, and RefSeq 
mRNAs. Searches on other selected identifiers, such as NP and NM accession 
numbers, OMIM identifiers, and Entrez Gene IDs are supported.  However, some 
types of queries will return an error, e.g. post-assembly GenBank 
entries, withdrawn gene names, and abandoned synonyms. If your initial
query is unsuccessful, try entering a different related term that
may produce the same location. For example, if a query on a gene symbol
produces no results, try entering an mRNA accession, gene ID number, or
descriptive words associated with the gene.
<P>
<B>Finding a genome location using BLAT</B><BR>
If you have genomic, mRNA, or protein sequence, but don't know the name or
the location to which it maps in the genome, the <A HREF="../../cgi-bin/hgBlat">
BLAT</A> tool will rapidly locate the position by homology alignment, provided that the 
region has been sequenced. This search will find close members of the gene
family, as well as assembly duplication artifacts. An entire set of query sequences can 
be looked up simultaneously when provided in fasta format. 
<P>
A successful BLAT search 
returns a list of one or more genome locations that match the input sequence. To
view one of the alignments in the Genome Browser, click the <I>browser</I> link for the 
match. The <I>details</I> link can be used to preview the alignment to determine if
it is of sufficient match quality to merit viewing in the Genome Browser. If too many BLAT hits 
occur, try narrowing the search by filtering the sequence in slow mode with
<A HREF="http://www.repeatmasker.org" TARGET=_blank>RepeatMasker</A>, then rerunning the BLAT search. 
<P>
For more information
on conducting and fine-tuning BLAT searches, refer to the <A HREF="#BLATAlign">
BLAT section</A> of this document.
<P>
<B>Opening the Genome Browser with a custom annotation track</B><BR>
You can open the Genome Browser window with a custom annotation track displayed
by using the Add Custom Tracks feature available from the gateway and annotation
tracks pages.
For more information on creating and using custom annotation tracks, refer to 
the <A HREF="#CustomTracks">Creating custom annotation tracks</A> section.
<P>
Annotation track data can be entered in one of three ways:
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Enter the file name for an annotation track source file in the
	Annotation File text box.
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Type or paste the annotation track data into the large text box.
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	If the annotation data are accessible through a URL, enter the URL name in the 
	large text box.
	</TD></TR>
</TABLE>
<P>
Once you've entered the annotation information, click the <I>submit</I> button at the top of 
the Gateway page to open up the Genome Browser with the annotation track
displayed.
<P>
The Genome Browser also provides a <A HREF="../customTracks/custTracks.html" 
TARGET=_blank>collection of custom annotation tracks</A> contributed by the UCSC
Genome Bioinformatics group and the research community. 

<P><B>NOTE:</B> If an annotation track does not display correctly when
you attempt to upload it, you may need to reset the Genome Browser to 
its default settings, then reload the track. For information on
troubleshooting display problems with custom annotation tracks, refer to the
troubleshooting section in the <A HREF="#CustomTracks">Creating custom
annotation tracks</A> section.
<P>
<B>Viewing genome data as text</B><BR> 
The 
<A HREF="../../cgi-bin/hgTables" TARGET=_blank>Table Browser</A>,
a portal to the underlying open source <A
HREF="http://www.MySQL.com" TARGET=_blank>MySQL</A> relational database driving the Genome
Browser, displays genomic data as columns of text rather than as
graphical tracks. For more information on using the Table Browser, see the
section <A HREF="#TableBrowser"> Getting Started: on the Table Browser</A>.
<P>
<B>Opening the Genome Browser from external gateways</B><BR> 
Several external gateways provide direct links into the Genome Browser. Examples
include: <A HREF="http://www.ncbi.nlm.nih.gov/entrez/" TARGET=_blank>Entrez Gene</A>, 
<A HREF="http://www.ncbi.nlm.nih.gov/IEB/Research/Acembly/index.html"
TARGET=_blank>AceView</A>,
<A HREF="http://www.ensembl.org/" TARGET=_blank>Ensembl</A>, <A
HREF="http://supfam.mrc-lmb.cam.ac.uk/SUPERFAMILY/" TARGET=_blank>SuperFamily</A>, 
and <A
HREF="http://www.genecards.org/" TARGET=_blank>GeneCards</A>. Journal
articles can also link to the browser and provide custom tracks. Be sure to use
the assembly date appropriate to the provided coordinates when using data from
a journal source.
<P>
<B>Tips for Use</B><BR>
To facilitate your return to regions of interest within the Genome
Browser, save the coordinate range or bookmark the page of displays that you plan to 
revisit or wish to share with others. 
<P>
It is usually best to work with the most recent assembly even though a full 
set of tracks might not yet be ready. Be aware that the coordinates of a given feature 
on an unfinished chromosome may change from one assembly to the next as
gaps are filled, artifactual duplications are reduced, and strand orientations
are corrected.  The Genome Browser offers multiple tools that can correctly 
convert coordinates between different assembly releases. For more information
on conversion tools,
see the section <A HREF="#Convert">Converting data between assemblies</A>.
<P>
To ensure uninterrupted browser services for your research during UCSC server 
maintenance and power outages, bookmark a
<A HREF="../../mirror.html" TARGET=_blank>mirror</A> site that 
replicates the UCSC genome browser. 
<P>
Bear in mind that the Genome Browser cannot outperform the underlying quality of 
the draft genome. Assembly errors and sequence gaps may still occur well into
the sequencing process due to regions that are intrinsically difficult to sequence. 
Artifactual duplications arise as unavoidable compromises during a build, causing 
misleading matches in genome coordinates found by alignment. 
<BR>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<A NAME="FineTuning"></A><BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp; Interpreting and fine-tuning the
		Genome Browser display</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
<p>
The Genome Browser annotation tracks page displays a genome location 
specified through a Gateway search, a BLAT search, or an uploaded 
custom annotation track. There are five main features on this page: a set of 
<A HREF="#NAV">navigation controls</A>, a 
chromosome ideogram, the annotations tracks image, 
<A HREF=#TRACK_CONFIG>display configuration buttons</A>, and a set of track 
<A HREF="#TRACK_CONT">display controls</A>.
<P>
The first time you open the
Genome Browser, it will use the application default values to configure the
annotation tracks display. By manipulating the navigation, configuration 
and display controls, you can customize the annotation tracks display to 
suit your needs. 
For a complete description of the annotation tracks available in all assembly
versions supported by the Genome Browser, see the <A HREF="#IndivTracks">
Annotation Track Descriptions</A> section.
<P>
The Genome Browser retains user preferences from session to session within the
same web browser, although it never monitors or records user activities or 
submitted data. 
To restore the default settings, click the &quot;Click here to reset&quot; 
link on the Genome Browser Gateway page. To return the display to the default 
set of tracks (but retain custom tracks and other configured Genome Browser 
settings), click the <I>default tracks</I> button on the Genome Browser page.
<P>
<B>Display conventions</B><BR>
The annotation tracks displayed in the Genome Browser use a common set of
display conventions:<BR>
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Annotation track
	descriptions:</B> 
	Each annotation track has an associated description page that contains a
	discussion of the track, the methods used to create the annotation, the 
        data 
	sources and credits for the track, and (in some cases) filter and
	configuration options to
	fine-tune the information displayed in the track. To view the
	description page, click on the mini-button to the left of a displayed
	track or on the label for the track in the Track Controls section.</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Annotation track
	details pages:</B> 
	When an annotation track is displayed in full, pack, or squish mode, 
	each line item
	within the track has an associated details page that can be displayed by 
	clicking on the item or its label. The
	information contained in the details page varies by annotation track, but
	may include basic position information about the item, related links 
	to outside sites 
	and databases, links to genomic alignments, or links to corresponding
	mRNA, genomic, and protein sequences.
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><A NAME="GeneDisplay"></A><B>Gene prediction tracks:</B> 
	Coding exons are represented by blocks connected by horizontal lines
	representing introns. The 5' and 3' untranslated regions (UTRs) are displayed 
	as thinner blocks on the leading and trailing ends of the aligning
	regions. In full display mode, arrowheads on
	the connecting intron lines indicate the direction of transcription.
	In situations where no intron is visible (e.g. single-exon genes,
	extremely zoomed-in displays), the arrowheads are displayed on the exon 
	block itself.</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><A NAME="PSLDisplay"></A><B>Pat space layout (PSL) alignment 
	tracks:</B> 
	Aligning regions (usually exons when the query is cDNA) are shown as black blocks. In dense display 
	mode, the degree of darkness corresponds to the number of features aligning to 
	the region or the degree of quality of the match. In pack or full display mode, 
	the aligning regions are connected by lines 
	representing gaps in the alignment (typically spliced-out introns), with arrowheads indicating 
	the orientation of the alignment, pointing right if the query sequence was aligned
	to the forward strand of the genome and left if aligned to the reverse strand.  
	Two parallel lines are drawn over double-sided alignment gaps, which skip over
        unalignable sequence in both target and query.  
	For alignments of ESTs, the arrows may be reversed to show the apparent 
	direction of transcription deduced from splice junction sequences.
	In situations where no gap lines are 
	visible, the arrowheads are displayed on the block itself. 
	To prevent display problems, the Genome Browser imposes an upper 
	limit on the number of alignments that can be viewed simultaneously 
	within the tracks image. When this limit is exceeded, the Browser 
	displays the best several hundred alignments in a condensed display
	mode, then lists the number of undisplayed alignments in the last 
	row of the track. In this situation, try zooming in to display 
	more entries or to return the track to full display mode. 
	For some PSL tracks, <A HREF="hgCodonColoringMrna.html">extra coloring</A> 
	to indicate mismatching bases and query-only gaps may be available.</TD></TR>

<A NAME="ChainDisplay"></A>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>&quot;Chain&quot; tracks 
	(2-species alignment):</B> 
	Chain tracks display boxes joined together by either single or double 
	lines. The boxes represent aligning regions. Single lines indicate gaps 
	that are largely due to a deletion in the genome of the first species or 
	an insertion in the genome of the second species. Double lines represent 
	more complex gaps that involve substantial sequence in both species. 
	This may result from inversions, overlapping deletions, an abundance of 
	local mutation, or an unsequenced gap in one species. In cases where 
	there are multiple chains over a particular portion of the genome, 
	chains with single-lined gaps are often due to processed pseudogenes, 
	while chains with double-lined gaps are more often due to paralogs and 
	unprocessed pseudogenes. In the fuller display modes, the individual 
	feature names indicate the chromosome, strand, and location 
	(in thousands) of the match for each matching alignment. </TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><A NAME="NetDisplay"></A><B>&quot;Net&quot; tracks 
	(2-species alignment):</B> 
	Boxes represent ungapped alignments, while lines represent gaps. Clicking 
	on a box displays detailed information about the chain as a whole, while 
	clicking on a line shows information on the gap. The detailed information 	  
	is useful in determining the cause of the gap or, for lower level chains, 	    
	the genomic rearrangement.  Individual items in the display are 
	categorized as one of four types (other than gap):
	<UL>
	<LI>
	<em>Top</em> - The best, longest match. Displayed on level 1. 
	<LI>
	<em>Syn</em> - Lineups on the same chromosome as the gap in the level 
	above it. 
	<LI>
	<em>Inv</em> - A lineup on the same chromosome as the gap above it, but 
	in the opposite orientation. 
	<LI>
	<em>NonSyn</em> - A match to a chromosome different from the gap in the 
	level above. 
	</UL></TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>&quot;Wiggle&quot; 
	tracks:</B> 
	These tracks plot a continuous function along a chromosome. Data is 
	displayed in windows of a set number of base pairs in width. 
	The score for each window 
	displays as "mountain ranges". The display characteristics vary among
	the tracks in this group. See the individual track descriptions for more
	information on interpreting the display. If the "mountain peak" is
	taller or shorter than what can be shown in the display, it is 
	clipped and colored magenta.</TD></TR>
</TABLE> 
<P>
<A NAME="TRACK_CONT"></A>
<B>Changing the display mode of an individual annotation track</B><BR>
Each annotation track within the window may have up to five display modes: 
<P>
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Hide:</B> the
	track is not displayed at all. To hide all the annotation tracks,
	click the <I>hide all</I> button. This mode is useful for restricting
	the display to only those tracks in which you are interested. For
	example, someone who is not interested in SNPs or mouse synteny may want 
	to hide these tracks to reduce track clutter and improve speed. There
	are a few annotation tracks that pertain only to one specific
	chromosome, e.g. Sanger22, Rosetta. In these cases, the track and its 
	associated controller will be hidden automatically when the track window is 
	not open to the relevant chromosome.</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Dense:</B> the 
	track is displayed with all features collapsed into a single 
	line. This mode is useful for reducing the amount of space used by a
	track when you don't need individual line item details or when you just
	want to get an overall view of an annotation. For example, by opening an
	entire chromosome and setting the RefSeq Genes track to dense, you can get 
	a feel for the known gene density of the chromosome without displaying
	excessive detail. </TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Full:</B> the
	track is displayed with each annotation feature on a separate line. It is
	recommended that you use this option sparingly, due to the large number
	of individual track items that may potentially align at the selected position. For
	example, hundreds of ESTs might align with a specified gene. When
	the number of lines within a requested track location exceeds 250, the
	track automatically defaults to a more tightly-packed display mode. In 
	this situation,
	you can restore the track display to full mode by narrowing the chromosomal 
	range displayed or by using a track filter to reduce the number of items 
	displayed. On tracks that contain only hide, dense, and full modes, you 
	can toggle between full and dense display modes by clicking on the
	track's center label.</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Squish:</B> the 
	track is displayed with each annotation feature shown separately, but
	at 50% the height of full mode. Features are unlabeled, and more than one
	may be drawn on the same line. This mode is useful for reducing the 
	amount of space used by a track when you want to view a large number of
	individual features and get an overall view of an annotation. It is
	particularly good for displaying tracks in which a large number of 
	features align to a particular section of a chromosome, e.g. EST tracks. 
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Pack:</B> the 
	track is displayed with each annotation feature shown separately and
	labeled, but not necessarily displayed on a separate line. 
	This mode is useful for reducing the 
	amount of space used by a track when you want to view the large number of
	individual features allowed by squish mode, but need the labeling and
	display size provided by full mode. When the number of lines within the
	requested track location exceeds 250, the track automatically defaults
	to squish display mode. In this situation, you can restore the track 
	display to pack mode by narrowing the chromosomal range displayed or by 
	using a track filter to reduce the number of items 
	displayed.  To toggle between pack and full display modes, click on 
	the track's center label.
	</TD></TR>
</TABLE> 
<P>
The track display controls are grouped into categories that reflect the type 
of data in the track, e.g. Gene Prediction Tracks, mRNA and EST tracks, etc. 
To change the display mode for a track, find the track's controller in the
Track Controls section at the bottom of the Genome Browser page, select the desired
mode from the control's display menu, and then click the <I>refresh</I> button. 
Alternatively, you can change the display mode by using the Genome
Browser's <A HREF="#RIGHT_CLICK_NAV">right-click navigation</A> feature,
or can toggle between dense and full modes 
for a displayed track (or pack mode when available) by clicking on the optional 
center label for the track. 
<P>
<B>Changing the display mode for a group of tracks</B><BR>
Track display modes may be set individually or as a group on the Genome
Browser Track Configuration page. To access the configuration page, click the
<em>configure</em> button on the annotation tracks page or the 
<em>configure tracks and display</em> button on the Gateway page. Exercise
caution when using the <em>show all</em> buttons on track groups or 
assemblies that contain a large number tracks; this may seriously impact the
display performance of the Genome Browser or cause your Internet browser to
time out.
<P>
<B>Hiding the track display controls</B><BR>
The entire set of track display controls at the bottom of the annotation 
tracks page may be hidden from view by checking the <em>Show track controls 
under main graphic</em> option in the Configure Image section of the Track 
Configuration page.
<P>
<B>Changing the display of a track by using filters and configuration options</B><BR>
Some tracks have additional filter and configuration capabilities, 
e.g. EST tracks, mRNA tracks, NC160, etc. These options
let the user modify the color or restrict the data displayed within an annotation track.
Filters are useful for focusing attention on items relevant to the current task
in tracks that contain large amounts of data. For example, to highlight ESTs
expressed in the liver, set the EST track filter to display items in a
different color when the associated tissue keyword is "liver". Configuration
options let the user adjust the display to best show the data of interest.
For example, the min vertical viewing range value on wiggle tracks
can be used to establish a data threshold. By setting the min value to &quot;50&quot;, only data values greater than 50 percent will display.
<P>
To access filter and configuration options for a specific annotation track, open
the tracks' description page by clicking the label for the track's control menu 
under the Track Controls section, the mini-button to the left of the displayed 
track, or the &quot;Configure...&quot; option from the Genome Browser's 
<A HREF="#RIGHT_CLICK_NAV">right-click</A> popup menu.
The filter and configration section is located at the top of the 
description page. In most instances, more information about the configuration 
options is available within the description text or through a special help link 
located in the configuration section. 
<P>
Filter and configuration settings are persistent from session to session on the 
same web browser. To return the Genome Browser display to the default set of 
tracks (but retain custom tracks and other configured Genome Browser settings),
click the 
<I>default tracks</I> button on the Genome Browser tracks page.
To remove all user configuration settings and custom tracks, and completely 
restore the defaults, click the &quot;Click here to reset&quot; link on the 
Genome Browser Gateway page. 
<P>
<A NAME="NAV"></A>
<B>Zooming and scrolling the tracks display</B><BR>
At times you may want to adjust the amount of flanking region displayed in the
annotation tracks window or adjust the scale of the display. At a scale of 1 pixel per 
base pair, the window accurately displays the width of exons and introns, and 
indicates the direction of transcription (using arrowheads) for multi-exon features. At a 
grosser scale, certain features - such as thin exons - may disappear. Also, some
exons may falsely appear to fall within RepeatMasker features at some scales. 
<P>
Click the <I>zoom in</I> and <I>zoom out</I> buttons at the top of the 
Genome Browser page to zoom in or out on the center of the annotation tracks window by 1.5, 3 or
10-fold. Alternatively, you can zoom in 3-fold on the display by clicking
anywhere on the
Base Position track. In this case, the zoom is centered on the coordinate of the
mouse click.  To view the base composition of the sequence underlying the
current annotation track display, click the <em>base</em> button. 
<P>
Quickly zoom to a specific region of interest by using the browser's
&quot;drag-and-zoom&quot; feature. To define the region you wish to zoom to, 
click-and-hold the mouse button on one edge of the desired zoom area in the Base
Position track, drag the mouse right or left to highlight the selection area, 
then release the mouse button. The annotation tracks image will automatically 
zoom to the new region. To drag-and-zoom in a part of the image other
than the Base Position track, depress the shift key before clicking and dragging
the mouse. 
Note that the <em>Enable advanced javascript 
features</em> option on the Track Configuration page must be toggled on to use 
this feature. 
<P>
To scroll (pan) the view of the entire tracks image horizontally, click on the 
image and drag the cursor to the left or right, then release the mouse button, 
to shift the displayed region in the corresponding direction. The view may be 
scrolled by up to one image width.
To scroll the annotation tracks horizontally by set increments of 10%, 50%, or 
95% of the displayed
size (as given in base pairs), click the corresponding <I>move</I> arrow. It is also 
possible to scroll the left or right side of the tracks by a specified number
of vertical gridlines while keeping the
position of the opposite side fixed. To do this, click the appropriate <I>move 
start</I> or <I>move end</I> arrow, located under the annotation tracks window.
For example, to keep the left-hand display coordinate fixed but increase the
right-hand coordinate, you would click the right-hand <I>move end</I> arrow. To increase
or decrease the gridline scroll interval, edit the value in the <I>move start</I> or
<I>move end</I> text box. 
<P>
<B>Changing the displayed track position</B><BR>
To display a completely different position in the genome, enter the new query in
the <I>position/search</I> text box, then click the <I>jump</I> button. For more information on
valid entries for this text box, refer to the <A HREF="#GetStarted">Getting
Started</A> section.
<P>
If a chromosome image (ideogram) is available above the track display, click anywhere
on the chromosome to  move to that position (the current window size will be
maintained). Select a region of any size by clicking and dragging in the image.
Finally, hold the &quot;control&quot; key while clicking on a chromosome band
to select the entire band.
<P>
<B>Changing the order of the displayed tracks</B><BR>
To vertically reposition a track in the annotation track window, click-and-hold 
the mouse button on the side label, then drag the highlighted track up or down 
within the image. Release the mouse button when the track is in the desired 
position. To move an entire group of associated tracks (such as all the
displayed subtracks in a composite track), click-and-hold the gray mini-button 
to the left of the tracks, then drag.
<P>
<A NAME="TRACK_CONFIG"></A>
<B>Changing the width of the annotation track window</B><BR>
The first time the annotation track window is displayed, or after the
Genome Browser has been reset, the size of the track window is 
set by default to the width that best fits your Internet browser window. If you
horizontally resize the browser window, you can automatically adjust the 
annotation track image size to the new width by clicking the <em>resize</em> 
button under the track image. To manually override the default width, enter a 
new value in the <em>image width</em> text box on the Track Configuration page, 
then click the <em>submit</em> button. The maximum supported width is 5000 pixels.
<P>
<B>Changing the width of the label area to the left of the image</B><BR>
The item labels (or track label, when viewed in dense mode) are displayed to
the left of the annotation image. The width of this area is set to 17 
characters by default. To change the width, edit the value in the <em>label
area width</em> text box on the Track Configuration page, then click 
<em>Submit</em>.
<P>
<B>Changing the text size in the annotation track image</B><BR>
The annotation track image may be adjusted to display text in a range of
fonts from &quot;tiny&quot; to &quot;huge&quot;. To change the size of the
text, select an option from the <em>text size</em> pull-down menu on the 
Track Configuration page, then click <em>Submit</em>. The text size is set to
&quot;small&quot; by default.
<P>
<B>Hiding the annotation track labels</B><BR>
The track and element labels displayed above and to the left of the tracks
in the annotation tracks image may be hidden from view by unchecking the
<em>Display track descriptions above each track</em> and <em>Display labels 
to the left of items in tracks</em> boxes, respectively, on the Track 
Configuration page.
<P>
<B>Hiding the display grid on the annotation tracks image</B><BR>
The light blue vertical guidelines on the annotation tracks image may be
removed by unchecking the <em>Show light blue vertical guidelines</em> box
on the Track Configuration page.
<P>
<B>Hiding the chromosome ideogram</B><BR>
The chromosome ideogram, located just above the annotation tracks image,
provides a graphical overview of the features on the selected chromosome, 
including its bands, the position of the centromere, and an indication of
the region currently displayed in the annotation tracks image. 
To hide the ideogram, uncheck the <em>Display chromosome ideogram above 
main graphic</em> box on the Tracks Configuration page.
<P>
<B>Enabling item and exon navigation</B><BR>
When the <em>Next/previous item navigation</em> configuration option is 
toggled on, on the Track Configuration page,  
gray double-headed arrows display in the Genome Browser tracks image on both 
sides of the track labels of gene, mRNA and EST tracks (or any standard tracks 
based on BED, PSL or genePred format). Clicking on the gray arrows shifts the
image window toward that end of the chromosome so that the next item in the
track is displayed. Similarly, the
<em>Next/previous exon navigation</em> configuration option displays white 
double-headed arrows on 
both the 5' and 3' end of each track item that has exons positioned beyond the 
edges of the current image. Clicking on one of the white arrows shifts the image
window to the next exon located towards that end of the feature.
<P>
<A NAME="RIGHT_CLICK_NAV"></A>
<B>Enabling the right-click navigation feature</B><BR>
Several of the common display and navigation operations offered on the
Genome Browser tracks page may be quickly accessed by right-clicking on
a feature on the tracks image and selecting an option from the displayed
popup menu. Depending on context, the right-click feature allows the
user to:
<UL>
<LI>change the track display mode
<LI>zoom in or out to the exact position coordinates of the feature
<LI>open the &quot;Get DNA&quot; window at the feature's coordinates
<LI>display details about the feature 
<LI>open a popup window to configure the track's display
<LI>display the entire tracks image in a
separate window for inclusion in spreadsheets or other documents. (Note
that the Genome Browser &quot;PDF/PS&quot; described below can also be
used to generate a high-quality annotation tracks image suitable for
printing.)
</UL>
<P>
To use the right-click feature, make sure the <em>Enable advanced
javascript features</em> option on the Track Configuration page is checked,
and configure your internet browser to allow the display of popup
windows from genome.ucsc.edu. When enabled, the right-click navigation
feature replaces the default contextual popup menu typically displayed by the
internet browser when a user right-clicks on the tracks image. A few
combinations of the Mozilla Firefox browser on Mac OS do not support the
right-click menu functionality using secondary click; in these
instances, ctrl+left-click must be used to display the menu. 
<P>
<B>Printing a copy of the annotation track window</B><BR>
The Genome Browser provides a mechanism for saving a copy of 
the currently displayed annotation tracks image to a file that can be printed 
or edited. Images saved in PostScript format can be printed at high resolution
and edited by drawing programs such as Adobe Illustrator. This is useful for
generating figures intended for publication. Images can also be saved in PDF
format for viewing by Adobe Acrobat Reader.
<P>
To print or save the image to a file:
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top"><B>1.</B></TD><TD>
	Click the PDF/PS link in the menu on the annotation tracks page.</TD></TR>
	<TR><TD WIDTH="20" VALIGN="top"><B>2.</B></TD><TD>
	Click the PostScript or PDF link.</BR></BR>
	
</TD></TR></TABLE>
NOTE: If you have configured your browser image to use one of the larger
font sizes, the text in the resulting screen shot may not display correctly.
If you encounter this problem, reduce the Genome Browser font size using the 
<A HREF="#TRACK_CONFIG">Configuration utility</A>, then repeat the save/print 
process.</P>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>

	<A NAME="BLATAlign"></A><BR>	

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Using BLAT alignments </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
BLAT (BLAST-Like Alignment Tool) is a very fast sequence alignment tool similar to 
BLAST. 
For more information on BLAT's internal scoring schemes and its overall n-mer 
alignment seed strategy, refer to W. James Kent (2002) <A
HREF="http://www.genome.org/cgi/content/abstract/12/4/656" TARGET=_blank>BLAT -
The BLAST-Like Alignment Tool</A>, <em>Genome Res</em> 12:4 656-664. 
<P>
On DNA queries, BLAT is designed to quickly find sequences with 95% or greater 
similarity of length 25 bases or more. It may miss genomic alignments that
are more divergent or shorter than these minimums, although it will find perfect 
sequence matches of 33 bases and sometimes as few as 22 bases. The tool is
capable of aligning sequences that contain large introns. 
On protein queries, BLAT rapidly locates genomic sequences with 80% or greater 
similarity of length 20 amino acids or more. In general, gene family members that 
arose within the last 350 million years can generally be detected. More
divergent sequences can be aligned to the human genome by using NCBI's BLAST and
psi-BLAST, then using BLAT to align the resulting match onto the UCSC genome 
assembly. In practice DNA BLAT works well on primates, and protein BLAT works
well on land vertebrates. 
<P>
Some common uses of BLAT include: 
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	finding the genomic coordinates of mRNA or protein within a given assembly
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	determining the exon structure of a gene
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	displaying a coding region within a full-length gene
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	isolating an EST of special interest as its own track
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	searching for gene family members
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD> 
	finding human homologs of a query from another species. 
	</TD></TR> 
</TABLE>	
	
<P><B>Making a BLAT query</B><BR>
To locate a nucleotide or protein within a genome using BLAT: 
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top"><B>1.</B></TD><TD>
	Open the <A HREF="../../cgi-bin/hgBlat?command=start" 
	TARGET=_blank>BLAT Search Genome page</A> by clicking the BLAT link on the top 
	blue menu bar of any of the Genome Browser pages.</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top"><B>2.</B></TD><TD>
	Select the genome, assembly, query type, output sort order, and output
	type. To order the search results based on the closeness of the sequence
	match, choose one of the score options in the <I>Sort output</I> menu.
	The score is determined by the number of matches vs. mismatches in the
	final alignment of the query to the genome.  
	</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top"><B>3.</B></TD><TD>
	If the sequence to be uploaded is in an unformatted plain text file, enter 
	the file name in the <I>Upload sequence</I> text box, then click the
	<I>submit file</I> 
	button.  Otherwise, paste the sequence or fasta-formatted list into the large 
	edit box, and then click the <I>submit</I> button. Input sequence can be
	obtained from the Genome Browser as well as from a custom annotation
	track.</TD></TR> 
</TABLE> 
<P>
Header lines may be included in the input text if they are preceded by > and
contain unique names. 
Multiple sequences may be submitted at the same time if they are of the
same type and are preceded by unique header lines. Numbers, spaces, and extraneous 
characters are ignored: 
<P>
<PRE>
>sequence_1
ATGCAGAGCAAGGTGCTGCTGGCCGTCGCCCTGTGGCTCTGCGTGGAGAC
CCGGGCCGCCTCTGTGGGTTTGCCTAGTGTTTCTCTTGATCTGCCCAGGC
>sequence_2
ATGTTGTTTACCGTAAGCTGTAGTAAAATGAGCTCGATTGTTGACAGAGA
TGACAGTAGTATTTTTGATGGGTTGGTGGAAGAAGATGACAAGGACAAAG
>sequence_3
ATGCTGCGAACAGAGAGCTGCCGCCCCAGGTCGCCCGCCGGACAGGTGGC
CGCGGCGTCCCCGCTCCTGCTGCTGCTGCTGCTGCTCGCCTGGTGCGCGG
</PRE>

<P><B>BLAT limitations</B><BR>
DNA input sequences are limited to a maximum length of 25,000 bases. Protein or
translated input sequences must not exceed 5000 letters. As many as 25 multiple 
sequences may be submitted at the same time. The maximum combined length of DNA 
input for multiple sequence submissions is 50,000 bases (with a 25,000 base 
limit per individual sequence). For 
protein or translated input, the maximum combined input length is 12,500 letters 
(with a 5000 letter limit per individual sequence). 
<P>
NOTE: Program-driven BLAT use is
limited to a maximum of one hit every 15 seconds and no more than 5000 hits per
day.
<P><B>BLAT query search results</B><BR>
If a query returns successfully, BLAT will display a flat database file that 
summarizes the alignments found. A BLAT query often generates multiple 
hits. This can happen when the genome contains 
multiple copies of a sequence, paralogs, pseudogenes, statistical coincidences, 
artifactual assembly duplications, or when the query itself contains repeats or common 
retrotransposons. When too many hits occur, try resubmitting the query sequence
after filtering in slow mode with RepeatMasker. 
<P>
Items in the search results list are ordered by the criteria specified in the 
<I>Sort output</I> menu. Each line item provides links to view the details of the sequence 
alignment or to open the corresponding view in the Genome Browser. The
<I>details</I> link gives the letter-by-letter alignment of the sequence to the 
genome. It is recommended 
that you first examine the details of the alignment for match quality before viewing
the sequence in the Genome Browser. 
<P>
When several nearby BLAT matches occur on a single chromosome, a 
simple trick can be used to quickly adjust the Genome Browser track window to display 
all of them: open the Genome Browser with the match that has the lowest chromosome 
start coordinate, paste in the highest chromosome end coordinate from the list of 
matches, then click the <I>jump</I> button. 
<P><B>Creating a custom annotation track from BLAT output</B><BR>
To make a custom track directly from BLAT, select the 
<A HREF="#PSL">PSL format</A>
output option. The resulting PSL track can be uploaded into the Genome Browser by 
pasting the data into the data text box on the Genome Browser 
<A HREF="../../cgi-bin/hgCustom?">Add Custom 
Tracks</A> page, accessed via the &quot;add custom tracks&quot; button on the
Browser gateway and annotation tracks pages. See the 
<A HREF="#CustomTracks">Creating custom annotation tracks</A> section for more 
information.
<P><B>Using BLAT for large batch jobs or commercial use</B><BR>
For large batch jobs or internal parameter changes, it is best to install
command line BLAT on your own Linux server. Sources and executables are free for
academic, personal, and non-profit purposes. BLAT source may be downloaded from 
<A HREF="http://www.soe.ucsc.edu/~kent/src" 
TARGET=_blank>http://www.soe.ucsc.edu/~kent</A> (look for the <em>blatSrc*.zip</em> file 
with the most recent date). For BLAT executables, go to 
<A HREF="http://genome-test.cse.ucsc.edu/~kent/exe/" 
TARGET=_blank>http://genome-test.cse.ucsc.edu/~kent/exe/</A>; binaries are sorted by 
platform. 
Non-exclusive commercial licenses are available from the 
<A HREF="http://www.kentinformatics.com/" TARGET=_blank>Kent Informatics</A>
website. 
<P><B>BLAT documentation</B><BR>
For more information on the BLAT suite of programs, see the 
<A HREF="blatSpec.html">BLAT
Program Specifications</A> and the 
<A HREF="../../FAQ/FAQblat.html">Blat section</A> of the Genome 
Browser FAQ.
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="IndivTracks"></A><BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp; Annotation track descriptions</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
Detailed information about an individual annotation track, including display
characteristics, configuration information, and associated database tables, 
may be obtained from the track description page accessed by clicking the 
mini-button to the left of the displayed track in the Genome Browser, or 
by selecting the &quot;Open details...&quot; or &quot;Show
details...&quot; option from the Genome Browser's 
<A HREF="#RIGHT_CLICK_NAV">right-click</A> menu.
Click the &quot;View table schema&quot; link on the track description page to
display additional information about the primary database table underlying the
track. Table schema information may also be accessed via the &quot;describe 
table schema&quot; button in the <A HREF="#TableBrowser">Table Browser</A>. For 
more information on configuring and using the tracks 
displayed in the Genome Browser track window, see the section 
<A HREF="#FineTuning">Interpreting and Fine-tuning the Genome Browser display</A>.</P>

<P><B>Tips for viewing annotation track data</B><BR>
<TABLE CELLPADDING=0 CELLSPACING=0>
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	To display a description page with more information about the track,
	click on the mini-button to the left of a track.</TD></TR>   
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	To display a details page with additional information about a specific line 
	item within a track in full display mode, click on the item or its label. </TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	A track does not appear in the browser if its display mode is set to 
	<I>hide</I>. To restrict the browser's display to only those tracks in
	which you're interested, set the display mode of the unwanted tracks to
	<I>hide</I>.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	A track set to <I>full</I> display mode will default to a more tightly
	packed display mode if 
	the total number of lines in the track exceeds 250.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	To quickly toggle between <I>full</I> and <I>dense</I> or <I>pack</I> display modes, click on 
	the track's center label.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Only the most recent assemblies are fully active. Older assemblies may be <A HREF="http://genome-archive.cse.ucsc.edu/" TARGET=_blank>archived</A>.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Not all tracks appear in all assemblies. Only a basic set of tracks 
	appears initially in a new assembly.</TD></TR>  
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	Track data can be viewed as text tables using
	the <A HREF="#TableBrowser">Table Browser</A>. </TD></TR> 
		<TR><TD WIDTH="20" VALIGN="top">--</TD><TD>
	<A HREF="../credits.html" TARGET=_blank>Credit</A>
	goes to many individuals and institutions for generously contributing
	the tracks. For specific information about the contributors of a given 
	track, look at the Credits section on a track's description page.</TD></TR>  
</TABLE> 
	</TD><TD WIDTH=15></TD></TR></TABLE>
    <br></TD></TR></TABLE> </TD></TR></TABLE>

	<A NAME="TableBrowser"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Getting started on the Table Browser  </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
The <A HREF="../../cgi-bin/hgTables" TARGET=_blank>Table
Browser</A> provides text-based access to the genome assemblies and annotation
data stored in the Genome Browser database. As a flexible alternative to the 
graphical-based Genome Browser, this tool offers an enhanced level of 
query support that includes restrictions based on field values, free-form 
SQL queries, and combined queries on multiple tables. Output can be filtered 
to restrict the fields and lines returned, and may be organized into one of 
several formats, including a simple tab-delimited file that can be loaded into 
a spreadsheet or database as well as advanced formats that may be uploaded into 
the Genome Browser as custom annotation tracks. 
The Table Browser provides a convenient
alternative to downloading and manipulating the entire genome and its massive 
data tracks.
(See the <A
HREF="#Download">Downloading Genome Data</A> section.) 
<P>
For information on using the Table Browser features, refer to the <A HREF="hgTablesHelp.html">Table Browser User Guide</A>.
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="Sessions"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Getting started using Sessions  </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>The <A HREF="../../cgi-bin/hgSession" TARGET=_blank>Sessions tool</A> 
allows users to configure their browsers with specific track 
combinations, including custom tracks, and save the configuration options. 
Multiple sessions may be saved for future reference, for comparison of 
scenarios or for sharing with colleagues. Saved sessions persist for
four months after the last access, unless deleted.
<A HREF="/goldenPath/help/hgSessionHelp.html#CTs">User-generated tracks</A>
can be saved within sessions.</P>

<P>This tool may be accessed via the <B>Sessions</B> link in the top blue 
navigation bar in any assembly. To ensure privacy and security, users must 
login to 
the <A HREF="http://genomewiki.cse.ucsc.edu/index.php">genomewiki</A> site 
and create a username and password. Individual sessions 
may be designated by the user as either &quot;shared&quot; or 
&quot;non-shared&quot; to protect the privacy of confidential data. To 
avoid having a new shared session from someone else override existing 
Genome Browser settings, users are encouraged to open a new web-browser 
instance or to save existing settings in a session before loading a new 
shared session.</P>
<P>For more detailed information on using the Session tool, see the 
<A HREF="hgSessionHelp.html">Sessions User Guide</A>.</P>
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>


	<A NAME="GenomeGraphs"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Getting started on Genome Graphs  </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
The <A HREF="../../cgi-bin/hgGenome" TARGET=_blank>Genome Graphs</A> tool can 
be used to display genome-wide data sets such as the results of genome-wide 
SNP association studies, linkage studies, and homozygosity mapping. This tool 
is not pre-loaded with any sample data; instead, you can upload your own 
data for display by the tool.</P>
<P>
Once you have uploaded your data, you can view it in a variety of ways.
You can view multiple sets of genome-wide data simultaneously either as
superimposed graphs or side-by-side graphs.  Once you see an area of interest
in the Genome Graphs view, you can click on it to go directly to the 
Genome Browser at that position. You can also set a significance threshold 
for your data and view only regions or gene sets that meet that threshold.
<P>
For information on using the Genome Graphs features, refer to the <A HREF="hgGenomeHelp.html">Genome Graphs User Guide</A>.
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="VisiGeneHelp"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Using the VisiGene Image Browser</b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>
<A HREF="../../cgi-bin/hgVisiGene">VisiGene</A> is a browser for viewing 
<em>in situ</em> images. It enables the user to examine cell-by-cell as well as 
tissue-by-tissue expression patterns. The browser serves as a virtual 
microscope, allowing users to retrieve images that meet specific search 
criteria, then interactively zoom and scroll across the collection.</P>
<P>
To start the VisiGene browser, click the VisiGene link in the left-hand
sidebar menu on the <A HREF="../../index.html">Genome Browser</A> home page.</P>

<P><B>Images Available</B><BR>
The following image collections are currently available for browsing: 
<UL>
<LI>High-quality high-resolution images of eight-week-old male mouse
sagittal brain slices with reverse-complemented mRNA hybridization probes
from the <A HREF="http://brain-map.org/" TARGET=_blank>Allen Brain
Atlas</A>, courtesy of the <A HREF="http://www.alleninstitute.org/"
TARGET=_blank>Allen Institute for Brain Science</A>
<LI>Mouse <em>in situ</em> images from the 
<A HREF="http://www.informatics.jax.org/expression.shtml"
TARGET=_blank>Jackson Lab Gene Expression Database</A> (GXD) at MGI 
<LI>Transcription factors in mouse embryos from the Mahoney Center for 
Neuro-Oncology
<LI>Mouse head and brain <em>in situ</em> images from NCBI's 
<A HREF="http://www.ncbi.nlm.nih.gov/projects/gensat/" TARGET=_blank>Gene 
Expression Nervous System Atlas</A> (GENSAT) database
<LI><em>Xenopus laevis in situ</em> images from the 
<A HREF="http://www.nibb.ac.jp/en/index.php" TARGET=_blank>National Institute 
for Basic Biology</A> (NIBB) XDB project
</UL></P>

<P><B>Searching the Image Database</B><BR>
The image database may be searched by gene symbols, authors, years of 
publication, body parts, GenBank or UniProtKB accessions, organisms, 
<A HREF="http://genex.hgu.mrc.ac.uk/Atlas/intro.html" 
TARGET=_blank>Theiler stages</A> (mice), and 
<A HREF="http://www.xenbase.org/atlas/NF/NF-all.html"
TARGET=_blank>Nieuwkoop/Faber stages</A> (frogs). The search returns only those 
images that match all the specified criteria. For a list of sample search
strings, see the VisiGene Gateway page.</P>
<P>
The wildcard characters * and ? are supported for gene name searches. For 
example, to view the images of all genes in the Hox A cluster, search for 
<em>hoxa*</em>. When searching on author names that include initials, use the 
format <em>Smith AJ</em>. </P>

<P><B>Image Navigation</B><BR>
Following a successful search, VisiGene displays a list of thumbnails of images 
matching the search criteria in the lefthand pane of the browser. By default, 
the image corresponding to the first thumbnail in the list is displayed in the 
main image pane. If more than 25 images meet the search criteria, links at the 
bottom of the thumbnail pane allow the user to toggle among pages of search 
results. To display a different image in the main browser pane, click the 
thumbnail of the image you wish to view. </P>
<P>
By default, an image is displayed at a resolution that provides optimal viewing 
of the overall image. This size varies among images. The image may be zoomed in 
or out, sized to match the resolution of the original image or best fit the 
image display window, and moved or scrolled in any direction to focus on areas 
of interest. The original full-sized image may also be downloaded.</P>
<P>
<B>Zooming in:</B> To enlarge the image by 2X, click the Zoom <em>in</em> button
above the image or click on the image using the left mouse button. 
Alternatively, the + key may be used to zoom in when the main image pane is the 
active window.</P>
<P>
<B>Zooming out:</B> To reduce the image by 2X, click the Zoom <em>out</em> 
button above the image or click on the image using the right mouse button. 
Alternatively, the - key may be used to zoom out when the main image pane is
the active window. </P> 
<P>
<B>Sizing to full resolution:</B> Click the Zoom <em>full</em> button above the 
image to resize the image such that each pixel on the screen corresponds to a 
pixel in the digitized image.</P> 
<P>
<B>Sizing to best fit:</B> Click the Zoom <em>fit</em> button above the image 
to zoom the image to the size that best fits the main image pane.</P> 
<P>
<B>Moving the image:</B> To move the image viewing area in any direction, click 
and drag the image using the mouse. Alternatively, the following keyboard 
shortcuts may be used after clicking on the image: 
<UL>
<LI><B>Scroll left in the image:</B> Left-arrow key or <em>Home</em> key 
<LI><B>Scroll right in the image:</B> Right-arrow key or <em>End</em> key 
<LI><B>Scroll up in the image:</B> Up-arrow key or <em>PgUp</em> key 
<LI><B>Scroll down in the image:</B> Down-arrow key or <em>PgDn</em> key 
</UL></P>
<P>
<B>Downloading the original full-sized image:</B> Most images may be viewed
in their original full-sized format by clicking the &quot;download&quot; link
at the bottom of the image caption. NOTE: due to the large size of some images,
this action may take a long time and could potentially exceed the capabilities 
of some Internet browsers. </P>
<P>
If you have an image set you would like to contribute for display in the VisiGene
Browser, contact <A HREF="mailto:kent@soe.ucsc.edu">Jim Kent</A>.  </P>

		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>



	<A NAME="TrackFormatDNA"></A><BR>	

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; DNA text formatting</b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
The Genome Browser provides a feature to configure the retrieval, formatting,
and coloring of the text used to depict the DNA sequence underlying the features in the displayed 
annotation tracks window. Retrieval options allow the user to add a padding of 
extra bases to the upstream or downstream end of the sequence. Formatting options range from simply displaying exons in upper case to 
elaborately marking up a sequence according to multiple track data. The DNA sequence covered by 
various tracks can be highlighted by case, underlining, bold or italic fonts, and color.
<P>
The DNA display configuration feature can be useful to highlight features within a
genomic sequence, point out overlaps between two types of features (for example, known
genes vs. gene predictions), or mask out unwanted features. 
<P><B>Using the DNA text formatting feature</B><BR>
To access the feature, click on the DNA link on the top blue menu
bar on the Genome Browser page, or select the &quot;Get DNA...&quot; option 
from the Genome Browser's <A HREF="#RIGHT_CLICK_NAV">right-click</A> menu.
The Get DNA in Window page that appears contains 
sections for configuring the retrieval and output format. 
<P>
To display extra bases upstream of the 5' end of your sequence or downstream
of the 3' end of the sequence, enter the number of bases in the corresponding
text box. This option is useful in looking for regulatory regions.
<P>
The Sequence Formatting section lists several options for
adjusting the case of all or part of the DNA sequence. To choose one of these formats,
click the corresponding option button, then click the <I>get DNA</I> button. To access a table 
of extended formatting options, click the <I>Extended case/color options</I> button. 
<P>
The Extended DNA Case/Color page presents a table with many more format options. The page
provides instructions for using the formatting table, as well as examples of its use. The
list of tracks in the Track Name column is automatically generated from the list of tracks
available on the current genome.
<P><B>Tips for Use</B><BR>
A few caveats mentioned on the Extended DNA Case/Color page bear repeating. Keep the formatting simple at first: it is easy to 
make a display that is pretty to look at but is also completely cryptic. Also, be careful when 
requesting complex formatting for a large chromosomal region: when all the HTML tags have been 
added to the output page, the file size may exceed the size limits that your internet browser, 
clipboard, and other software can safely display. The maximum size of genome that can be 
formatted by the tool is approximately 10 Mbp.
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="Convert"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Converting data between assemblies</b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>Coordinates of features frequently
change from one assembly to the next as gaps are closed, strand orientations are
corrected, and duplications are reduced. Occasionally, a chunk of sequence may
be moved to an entirely different chromosome as the map is
refined. There are three different methods available for migrating data from one
assembly to another: BLAT alignment, coordinate conversion, and coordinate 
lifting. The BLAT alignment tool is described in the section 
<A HREF="#BLATAlign">Using BLAT alignments</A>.
<P><B>Coordinate conversion</B><BR>
The Genome Browser Convert utility is useful for 
locating the position of a feature of interest in a different release of the
same genome or (in some cases) in a genome assembly of another species.
During the conversion process, portions of the genome in the coordinate range 
of the original assembly are aligned to the new assembly while preserving their 
order and orientation. In general, it is easier to achieve successful 
conversions with shorter sequences. 
<P>
When coordinate conversion is available for an assembly, a Convert link
is displayed in the top menu bar on the Genome Browser tracks page. Click this
link to convert the currently-displayed coordinate range. You will be presented
with a list of the genome/assembly conversion options 
available for the
current assembly. Select the genome and assembly to which you'd like to convert
the coordinates, then click the Submit button. If the conversion is successful,
the browser will return a list of regions in the new assembly, along with
the percent of bases and span covered by that region. Click on a region to 
display it in the browser. If the conversion is unsuccessful, the utility 
returns a failure message.

<A NAME="Liftover"></A><BR>

<P><B>Lifting coordinates</B><BR>
The liftOver tool is useful if you wish to convert a large number of 
coordinate ranges between assemblies. This tool is available in both 
web-based and command line forms, and supports forward/reverse conversions
as well as conversions between species.

<P><B>Web-based coordinate lifting</B><BR>
To access the graphical version of the liftOver tool, click the Utilities link
in the left-hand sidebar on the Genome Browser home page, then select the
<a class="change" href="http://genome.ucsc.edu/cgi-bin/hgLiftOver">Batch Coordinate Conversion 
</a> (liftOver) link.
<P>
To convert one or more coordinate ranges using the default conversion settings: 
<OL>
<LI>
Select the genome and assembly from which the ranges were taken 
(&quot;Original&quot;), as well as the genome and assembly to which the 
coordinates should be converted (&quot;New&quot;). 
<LI>Select the Data Format option: 
Browser Extensible Data format (<A HREF="../../FAQ/FAQformat.html#format1">BED</A>) 
or position (coordinates of the form <em>chrN:start-end</em>). 
<LI>Enter coordinate ranges in the selected data format into the large text 
box, one per line.
<LI>Click Submit. 
</OL></P>
<P>
Alternatively, you may load the coordinate ranges from an existing data
file by entering the file name in the upload box at the bottom of the screen, 
then clicking the Submit File button. 
<P>
The default parameter settings are recommended for general purpose use of the
liftOver tool. However, you may want to customize settings if you have several
very large regions to convert. 
<!--
<UL>
<LI><B>Minimum ratio of bases that must remap</B> (default 0.95)<B>:</B> It may 
be beneficial to reduce this value when working with poor-quality assemblies or
cross-species conversions. For large-area conversions under these conditions,
a value as low as 0.01 may be used.
<LI><B>Minimum chain size in target</B> (default 0)<B>:</B> This parameter is 
applied only when the &quot;Allow multiple output regions&quot; option is 
checked. This setting may be increased to reduce fragmentation. For example, 
when doing large-scale conversions, we have found that a setting of 4000
worked well for this option.
<LI><B>Minimum hit size in query</B> (default 0)<B>:</B> This parameter is 
applied only when the &quot;Allow multiple output regions&quot; option is 
checked. 
<LI><B>Allow multiple output regions</B> (default off)<B>:</B> By default, 
liftOver does not return a match if the region is split in the new assembly. 
Consider checking this option for conversions involving high-quality data 
within the same species. This option should not be used when mapping large 
regions, doing cross-species conversions or using fragmented poor-quality 
assemblies.
<LI><B>Min ratio of alignment blocks/exons that must map</B> (default 1)<B>:</B>
<LI><B>If thickStart/thickEnd is not mapped, use the closest mapped bas</B>
(default off)<B>:</B>
</UL></P>
-->

<P><B>Command-line coordinate lifting</B><BR>
The command-line version of liftOver offers the increased flexibility and 
performance gained by running the tool on your local server. This utility 
requires access to a Linux platform. The executable file 
may be downloaded 
<A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/">here</A>.
Command-line liftOver requires a UCSC-generated <em>over.chain</em> file as 
input. Pre-generated files for a given assembly can be accessed from the 
assembly's &quot;LiftOver files&quot; link on the 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">Downloads</A> page. If the desired conversion file is not listed, 
send a request to the 
<A HREF="../../contacts.html">genome mailing list</A> and we may be able
to generate one for you. 

		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="Download"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Downloading genome data</b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>Most of the underlying tables containing the genomic sequence and annotation data 
displayed in the Genome Browser can be downloaded. All of the tables are
freely usable for any purpose except as indicated in the README.txt file
in the download directories. This data was contributed by many researchers, as 
listed on the Genome Browser 
<A HREF="../credits.html" TARGET=_blank>Credits</A> page. Please
acknowledge the contributor(s) of the data you use. 
<P><B>Downloading the data</B><BR>
Genome data can be downloaded in two different ways:
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Via
	ftp:</B>The UCSC Genome Bioinformatics ftp site contains download
	directories for all genome versions currently accessible in the
	Genome Browser. The ftp command
	<tt>ftp://hgdownload.cse.ucsc.edu/goldenPath/</tt> will take you to a
	directory that contains the genome download directories. This 
	download method is recommended if you plan to download a large file
	or multiple files from a single directory. Use the <I>mget</I> command
	to download multiple files: mget <I>filename1 filename2</I>, or
	mget -a (to download all the files in the directory).</TD></TR>

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>Via the Downloads link:</B> Click the
	<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html"
	TARGET=_blank>Downloads</A> link on the left side bar on the UCSC 
	Genome Bioinformatics home page to display a list of all database directories 
	available for 
	download. If the data you wish to download pre-dates the assembly 
	versions listed, look in the archives accessible from the
	<A HREF="http://genome-archive.cse.ucsc.edu/"
	TARGET=_blank>Archive</A> link on the home page.
	</TD></TR>
</TABLE> 
<P><B>Types of data available</B><BR>
There may be several download directories associated with each version of a genome
assembly: the full data set <I>(bigZips)</I>, the full data set by 
chromosome <I>(chromosome)</I>, the annotation database tables
<I>(database)</I>, and one or more sets of comparative cross-species alignments. 
<P>
<I>BigZips</I> contains the entire draft of the 
genome in chromosome and/or contig form. Depending on the genome, this 
directory may contain some or all of the following files:<BR> 
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>chromAgp.zip:</B>
    	Description of how the assembly was generated, unpacking to one file
	per chromosome. 
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>chromFa.zip:</B>
	The assembly sequence chromosomes, in one file per chromosome. 
	Repeats from RepeatMasker and Tandem Repeats Finder are shown in 
	lower case; non-repeating sequence is in upper case.
	The main assembly is contained in the <em>chrN.fa</em> files, where 
	<em>chrN</em> is the name of the chromosome.  The <em>chrN_random.fa</em> 
	files contain clones that are not yet finished or cannot be placed
	with certainty at a specific place on the chromosome. In some
	cases, including the human HLA region on chromosome 6, the 
	<em>chrN_random.fa</em> files also contain haplotypes that differ from
	the main assembly.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>chromFaMasked.zip:</B>
	The assembly sequence chromosomes, in one file per chromosome. 
	Repeats are masked by capital Ns; non-repeating sequence is shown in upper
	case.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>chromOut.zip:</B>
	RepeatMasker <em>.out</em> file for chromosomes, generated by RepeatMasker
	at the <I>-s</I> sensitive setting.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>chromTrf.zip:</B>
	Tandem Repeats Finder locations, filtered to keep repeats with period 
	less than or equal to 12, translated into one <em>.bed</em> file per 
	chromosome.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>contigAgp.zip:</B>
	Description of how the assembly was generated from fragments at a contig 
	layout level. 
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>contigFa.zip:</B>
	The assembly sequence contigs, in one file per contig. All 
	contigs are in forward orientation relative to the chromosome. 
        In some cases, this means that contigs will be reversed 
	relative to their orientation in the NCBI assembly. Repeats are
	shown in lower case; non-repeating sequence is shown in upper case.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>contigFaMasked.zip:</B>
	The assembly sequence contigs, in one file per contig. 
	Repeats are masked by capital <em>N</em>s; non-repeating sequence is 
	shown in upper case.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>contigOut.zip:</B>
	RepeatMasker <em>.out</em> file for contigs, generated by RepeatMasker 
	at the <I>-s</I> sensitive setting.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>contigTrf.zip:</B>
	Tandem Repeats Finder locations, filtered to keep repeats with period 
	less than or equal to 12, and translated into one <em>.bed</em> file per 
	contig.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>database.zip:</B>
	The Genome Browser database as tab-delimited files and associated 
	MySQL table-creation tiles (eliminated in later assemblies due to
	size restrictions).
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>est.fa.zip:</B>
	Sequences of all GenBank ESTs for the selected species. 
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>liftAll.zip:</B>
	The offsets of contigs within chromosomes.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>mrna.zip:</B>
	mRNAs in GenBank from the selected species. 
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>refmrna.zip:</B>
	RefSeq mRNAs from the selected species.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>upstream1000.zip:</B>
	Sequences 1000 bases upstream of annotated transcription start of RefSeq 
	genes. This includes only cases where the transcription start is 
	annotated separately from the coding region start.
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>upstream2000.zip:</B>
	Same as upstream1000, but with 2000 bases. 
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>upstream5000.zip:</B>
	Same as upstream1000, but with 5000 bases. 
	</TD></TR> 

	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>xenoMrna.zip:</B>
	All GenBank mRNAs from species other than that of the selected one.
	</TD></TR> 
</TABLE> 
<P>
<I>Chromosomes</I> contains the assembled sequence for the 
genome in separate files for each chromosome in a zipped fasta format.
The main assembly can be found in the <em>chrN.fa</em> files, where <em>N</em> 
is the name of the chromosome.  The <em>chrN_random.fa</em> files contain clones
that are not yet finished or cannot be placed with certainty
at a specific place on the chromosome. In some cases, the <em>chrN_random.fa</em>
files also contain haplotypes that differ from the main assembly.
<P>
<I>Database</I> contains all of the 
positional and non-positional tables in the genome annotation database. Each 
table is represented by 2 files:
<TABLE>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>.sql file:</B>
	the MySQL commands used to create the table.
	</TD></TR>
	<TR><TD WIDTH="20" VALIGN="top">--</TD><TD><B>.txt.gz file:</B>
	the MySQL database table data in tab-delimited format and 
	compressed with gzip.
	</TD></TR>
</TABLE> 
<P>
Schema descriptions for all tables in the genome annotation database may be
viewed by using the &quot;describe table schema&quot; button in the 
<A HREF="#TableBrowser">Table Browser</A>. 
<P>
<I>Cross-species alignments</I> directories, such as the <I>vsMm4</I> 
and <em>humorMm3Rn3</em> directories in the hg16 assembly, contain pairwise
and multiple species alignments and filtered alignment files 
used to produce cross-species annotations. For more information, refer to the 
READMEs in these directories and the description of the
<A HREF="../../FAQ/FAQformat.html#format5">Multiple Alignment Format</A> (MAF).
		</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
</TD></TR></TABLE>

	<A NAME="CustomTracks"></A><BR>

<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
			<FONT SIZE="4"><b>&nbsp; Creating custom annotation tracks </b></FONT>
		</TD></TR></TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
			<TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
<!--#include virtual="customTrackText.html"-->
	</TD><TD WIDTH=15></TD></TR></TABLE>
	<br></TD></TR></TABLE>
	</TD></TR></TABLE>
	
</TD></TR></TABLE>

</BODY></HTML>

