<!doctype html public "-//W3C//DTD HTML 4.01 Transitional//EN"
          "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<title>Quick Start Guide to Assembly Hubs</title>
<meta http-equiv="content-type" content="text/html; charset=utf-8">
<link rel="stylesheet" href="/style/HGStyle.css">
</head>
<body class="oldStaticPage">
<div class="header">
<img src="../../images/title.jpg" alt="UCSC Genome Bioinformatics">
</div><!-- .header -->
<div class="nav">
<!--#include virtual="../../cgi-bin/hgMenubar"-->
</div><!-- .nav -->

<div class="content"> <!-- start of page content -->
<div class="section">
<h1>Quick Start Guide to Assembly Hubs</h1> 
<div class="sectionContent">
<p>
Assembly Hubs allow researchers to create Track Data Hubs on assemblies that are not in
the UCSC Browser. By including the underlying reference sequence in UCSC
<a href="twoBit.html" target="_blank">twoBit</a> format,
as well as data tracks, researchers can browse and annotate any genome. For more information
please refer to the <a href="http://genomewiki.ucsc.edu/index.php/Assembly_Hubs"
target="_blank">Assembly Hub Wiki</a>. Below is also a section about starting <a href="#blatGbib">GBiB Assembly Hubs</a>.
</p>
<p>
<ul>
<b>STEP 1:</b> In a publicly-accessible directory, copy this <em>Arabidopsis thaliana</em>
plant assembly hub, which includes an araTha1.2bit file, using the following wget command:
<p>
<tt>wget -r --no-parent --reject "index.html*" -nH --cut-dirs=3 http://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubAssembly/plantAraTha1/</tt>
</p>
<p>
<ul>
Alternatively, <b>if you do not have wget installed,</b> you can curl these files individually.
Perform the curl -O option in the location you wish to copy the files:
<tt>curl -O http://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubAssembly/plantAraTha1/hub.txt</tt>
</p>
<p>
<em>If you use curl</eM>, be sure to recreate the structure with matching araTha1 and
araTha1/bbi directories. Double check you have all the files by looking here:
<tt>
<a href="http://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubAssembly/plantAraTha1/" 
target="_blank">http://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubAssembly/plantAraTha1/</a>
</tt>
</p>
</ul>
<p>
<b>STEP 2:</b> Paste your hub.txt link (<tt>http://yourURL/hub.txt</tt>) into the
<a href="../../cgi-bin/hgHubConnect" target="_blank">My Hubs</a>
tab of the Track Data Hubs page, click the "Add Hub" button, and then click the
"Genome Browser" link from the top bar.
<ul>
Alternatively build a URL that will directly load your assembly hub and display it
on hgGateway. Then click the "Genome Browser" link from the top bar to view your assembly hub: 
<p>
<tt>
http://genome.ucsc.edu/cgi-bin/hgHubConnect?hgHub_do_redirect=on&hgHubConnect.remakeTrackHub=on&hgHub_do_firstDb=1&<b>hubUrl=http://yourURL/hub.txt</b>
</tt>
<p>
This URL should work the same as using the original data just copied:<br>
<tt>
<a href="http://genome.ucsc.edu/cgi-bin/hgHubConnect?hgHub_do_redirect=on&hgHubConnect.remakeTrackHub=on&hgHub_do_firstDb=1&hubUrl=http://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubAssembly/plantAraTha1/hub.txt"
target="_blank">http://genome.ucsc.edu/cgi-bin/hgHubConnect?hgHub_do_redirect=on&hgHubConnect.remakeTrackHub=on&hgHub_do_firstDb=1&<b>hubUrl=http://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubAssembly/plantAraTha1/hub.txt</b></a>
</tt>
</p>
</ul>
<p>
<b>STEP 3:</b> Congratulations! Your assembly hub should display! 
</p>
<ul>
<p>If you are having problems, be sure all your files and the directories are
publicly-accessible. You may also wish to <a href="../../cgi-bin/cartReset"
target="_blank">reset</a> the browser occastionally to
clear all existing data. For hubs to work, your server must also accept byte-ranges.
You can check using the following command to verify "<tt>Accept-Ranges: bytes</tt>"
displays: <tt>curl -I http://yourURL/hub.txt</tt>
</p>
</ul>
<p>
Now that you have the assembly hub copied from above, you can copy the directory
and start to edit some of the documents like genomes.txt, groups.txt, and trackDb.txt to understand
how they work. Refer to the <a href="http://genomewiki.ucsc.edu/index.php/Assembly_Hubs"
target="_blank">Assembly Hub Wiki</a> to understand how to build a 
<a href="twoBit.html" target="_blank">twoBit</a> file for your
own original fasta files. Read more about <a href="trackDb/trackDbHub.html"
target="_blank">trackDb settings</a> in the definition document.
</p>
This assembly hub is a an abbreviated version of a larger plant assembly Public Hub.
You can explore the larger hub structure
<a href="http://genome-test.cse.ucsc.edu/~hiram/hubs/Plants/" target="_blank">here</a>.
<p>
Please note that the Browser waits 300 seconds before checking for 
any changes to these files. <b>When editing hub.txt, genomes.txt, trackDb.txt, and
related hub files shorten this delay by adding <tt>udcTimeout=1</tt> to your URL.</b>
For more information, please see the <a href="hgTrackHubHelp.html#Debug"
target="_blank">Debugging and Updating Track Hubs</a> section of the
<a href="hgTrackHubHelp.html" target="_blank">Track Hub User Guide</a>.
Also, for more detailed instructions on setting up a regular hub, please see the
<a href="hgTrackHubHelp.html#Setup" target="_blank">Setting Up Your Own Track Hub</a>
section of the Track Hub User Guide. 
</ul>
<b><font color="#006666" size=4>Setting up Blat for an Assembly Hub</font></b>
<ul>
<li><b><a href="#blat">Starting Blat for an Assembly Hub</a></b><br>
By running gfServers from your institution, you can enable blat on your assembly hubs.
</ul>
<p>
<b><font color="#006666" size=4>Setting up an Assembly Hub on GBiB with Blat included</font></b>
<ul>
<li><b><a href="#blatGbib">Starting a Blat enabled Assembly Hub on GBiB</a></b><br>
With an operational Genome Browers in a Box (GBiB), you can quickly and easily acquire
an example assembly hub and run gfServers locally on the GBiB to enable blat.
<p>
</ul>
<b><font color="#006666" size=4> Resources: </font></b>
<ul>
<li><b><a href="http://genomewiki.ucsc.edu/index.php/Assembly_Hubs"
target="_blank">Assembly Hubs Wiki</a></b>
<li><b><a href="hgTrackHubHelp.html" target="_blank">Track Hub User Guide</a></b> 
<li><b><a href="trackDb/trackDbHub.html"
target="_blank">Track Database (trackDb) Definition Document</a></b> 
<li><b><a href="http://genomewiki.ucsc.edu/index.php/Public_Hub_Guidelines"
target="_blank">Public Hub Guidelines Wiki</a></b>
<li><b><a href="hubQuickStart.html"
target="_blank">Quick Start Guide to a Basic Hub</a></b>
<li><b><a href="hubQuickStartGroups.html"
target="_blank">Quick Start Guide to Organizing Hubs</a></b>
</ul>
</p>
</ul>
</div> <!-- end sectionContent-->
</div> <!-- end section-->


<!-- ========= assembly blat ============================== -->

<a name="blat"></a><br>
<div class="section">
<h1>Starting Blat for an Assembly Hub</h1>
<div class="sectionContent">
<p>
From the location of yourAssembly.2bit file,
<tt>http://yourURL/yourAssembly/yourAssembly.2bit</tt>,
you can start two gfServers, specifing a port for the assembly hub to access amino
acid sequence, <tt>17777 -trans</tt>, or DNA sequence, <tt>17779</tt>, in
this example:<br>
<ul>
<tt>
gfServer start localhost 17777 -trans -mask yourAssembly.2bit &<br>
gfServer start localhost 17779 -stepSize=5 yourAssembly.2bit &<br>
</tt>
</ul>
Then you can edit the genomes.txt file of your assembly hub to include two lines
in the stanza referring to yourAssembly, that would have matching port numbers:<br>
<ul>
<tt>
transBlat yourLab.yourInstitution.edu 17777<br>
blat yourLab.yourInstitution.edu 17779<br>
</tt>
</ul>
See an example genomes.txt with commented out lines 
<a href="http://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubAssembly/plantAraTha1/genomes.txt"
target="_blank">here</a>, and please note capital B in transBlat. For more information, see the &quot;Adding BLAT servers&quot;
section of the <a href="http://genomewiki.ucsc.edu/index.php/Assembly_Hubs#Adding_BLAT_servers" 
target="_blank">Assembly Hub Wiki</a>.
The <a href="http://hgdownload.cse.ucsc.edu/downloads.html#source_downloads" 
target="_blank">Source Downloads</a> page offers access to utilities with pre-compiled 
binaries such as gfServer found in a blat/ directory for your machine type
<a href="http://hgdownload.cse.ucsc.edu/admin/exe/" target="_blank">here</a> and further
<a href="blatSpec.html"
target="_blank">blat documentation here</a>. Please note that because the <tt>-mask</tt> option 
in the above <tt>17777 -trans</tt> gfServer option will mask all lower-case sequence from
being matched, you may not wish to include it. See the above blat links and
gfServer usage statement for more information.
</p>
<p>
If you are having trouble with your blat servers connecting to the browser  it may help to know
that some institutions have firewalls that will prevent the browser from sending multiple inquiries,
in which case you may need to request your admins add this IP range as exceptions that are not
limited: <tt>128.114.119.*</tt> That will cover the U.S. genome.ucsc.edu site. In case you may
wish the requests to work from our European Mirror genome-euro.ucsc.edu site, you would want to
include <tt>129.70.40.120</tt> also to the exception list.
</p>
</div> <!--- sectionContent --->
</div> <!--- section -->

<!-- ========== blat assembly hub GBiB ============================== -->

<a name="blatGbib"></a><br>
<div class="section">
<h1>Starting a Blat enabled Assembly Hub on GBiB</h1>
<div class="sectionContent">
<p>
<b>Acquiring GBiB and Assembly Hub</b>
</p>
<p>
<em>Install GBiB</em>
<ul>
1. Acquire and install an operational Genome Browser in the Box:
<a href="gbib.html" 
target="_blank">http://genome.ucsc.edu/goldenPath/help/gbib.html</a>
</p>
<p>
You may wish to read this <a href="../../blog/genome-browser-in-a-box-gbib-origins/"
target="_blank">blog post</a> and the related OpenHelix
<a href="http://blog.openhelix.eu/?p=20193" target="_blank">blog post</a>.
</p>
</ul>
<em>SSH into your GBiB</em>
<ul>
<p>
2. With your GBiB operational, use your computer's terminal program to ssh
into your GBiB: <tt>ssh browser@localhost -p 1235</tt>, using <tt>browser</tt> for the password.
</p>
<p>
</ul>
<em>Wget example hub</em>
<ul>
<p>
3. Navigate to the GBiB's folders directory and then use sudo to wget this
assembly hub:
</p>
<p>
<ul>
<p>
<b><tt>cd /folders</tt></b>
</p>
<p>
<tt>sudo wget -r --no-parent --reject "index.html*" -nH --cut-dirs=3 http://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubAssembly/plantAraTha1/</tt>
</p>
</ul>
</p>
</ul>
<em>Load hub on your GBiB</em>
<p>
<ul>
4. You now have all the required files on your local machine and can load this plant assembly hub
by using this URL and selecting it under the &quot;group&quot; category where
&quot;Plant araTha1&quot; displays:<br>
</p>
<p>
</ul>
<ul>
<tt>
<a href="http://127.0.0.1:1234/cgi-bin/hgGateway?genome=araTha1&hubUrl=http://127.0.0.1:1234/folders/hubExamples/hubAssembly/plantAraTha1/hub.txt"
target="_blank">http://127.0.0.1:1234/cgi-bin/hgGateway?genome=araTha1&amp;hubUrl=http://127.0.0.1:1234/folders/hubExamples/hubAssembly/plantAraTha1/hub.txt</a>
</tt>
</ul>
</p>
<p>
<b>Enabling Blat</b>
</p>
<em>Acquire gfServer utility to create local blat servers</em>
<ul>
<p>
5. To enable blat you must acquire the gfServer utility. The UCSC Genome Browser and Blat
software are free for academic, nonprofit, and personal use. Commercial download and
installation of the Blat and In-Silico PCR software may be licensed through Kent Informatics
(<a href="http://www.kentinformatics.com">http://www.kentinformatics.com</a>).
</p>
<p>
You can obtain just the gfServer utility on your GBiB with the following command that will create
a bin directory and install the tool.
<p>
<ul>
<tt>mkdir ~/bin -p; rsync -avP hgdownload.cse.ucsc.edu::genome/admin/exe/linux.x86_64/blat/gfServer ~/bin/</tt>
</ul>
</p>
<p>
The GBiB also includes a tool you can run on the commandline to download an entire suite
of tools including gfServer: <tt>gbibAddTools</tt>
</p>
</ul>
<em>Enable blat lines in genomes.txt</em>
<ul>
<p>
6. Navigate to the genomes.txt file of this assembly hub: <tt>cd /folders/hubExamples/hubAssembly/plantAraTha1/</tt>
</p>
<p>
Edit the currently commented-out blat lines with <tt>sudo vi genomes.txt</tt> and
use &quot;x&quot; when the cursor is over the <tt>#</tt> at the start of the line to remove it and
<tt>:w!</tt> to save the changes and <tt>:q</tt> to quit.
</p>
<p>
<ul>
<tt>
blat localhost 17779<br>
transBlat localhost 17777<br>
</tt>
</ul>
<p>
Please note that if you loaded your hub earlier, it will take five minutes (300 seconds)
for the browser to check for any changes to genomes.txt, and that this delay can be
shortened temporarily by adding <tt>&udcTimeout=10</tt> to the URL. See more information in the
<a href="hgTrackHubHelp.html#Debug" target="_blank">Debugging and Updating</a> section of the
Track Hub User Guide.</p>
</ul>
<em>Start gfServers on your assembly hub</em>
<ul>
<p>
7. Change directories to the 2bit file: <tt>cd /folders/hubExamples/hubAssembly/plantAraTha1/araTha1</tt> 
</p>
<p>
Run the two gfServer commands to start the blat servers:
</p>
<p>
<ul>
<tt>
gfServer start localhost 17777 -trans -mask araTha1.2bit &<br>
gfServer start localhost 17779 -stepSize=5 araTha1.2bit &<br>
</tt>
</ul>
</p>
</ul>
<em>Load hub and use blat</em>
<ul>
<p>
8. Load this plant assembly hub by using this URL and selecting it under the &quot;group&quot; category where
&quot;Plant araTha1&quot; displays:<br>
<ul>
<tt>
<a href="http://127.0.0.1:1234/cgi-bin/hgGateway?genome=araTha1&hubUrl=http://127.0.0.1:1234/folders/hubExamples/hubAssembly/plantAraTha1/hub.txt"
target="_blank">http://127.0.0.1:1234/cgi-bin/hgGateway?genome=araTha1&amp;hubUrl=http://127.0.0.1:1234/folders/hubExamples/hubAssembly/plantAraTha1/hub.txt</a>
</tt>
</ul>
</p>
<p>
On the blat page, <tt><a href="http://127.0.0.1:1234/cgi-bin/hgBlat" target="_blank">
http://127.0.0.1:1234/cgi-bin/hgBlat</a></tt>, you can now select the <em>Arabidopsis thaliana</em>
assembly and blat plant amino acid sequences, like <tt>IYQTRENKYIIGEIQITESERDRRRSSLPGNH</tt>
or DNA sequences, like <tt>TAAGTAAAAAATAATATGATTAAGACTAATAAATCTTAATAGTTAATACT</tt>.
</div> <!--- section-->
</div> <!--  end content-->
</body>
</html>
