<!DOCTYPE html>
<!--#set var="TITLE" value="Genome Browser Net Format" -->
<!--#set var="ROOT" value="../.." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<h1>Net Format</h1>
<p>
The <em>net</em> file format is used to describe the axtNet data that underlie the net alignment 
annotations in the Genome Browser. For a detailed description of the methods used to generate these 
data, refer to the Genome Browser description pages that accompany the 
<a href="http://hgdownload.gi.ucsc.edu/downloads.html">downloadable net alignment tracks</a>.
You can find a detailed
<a href="https://genomewiki.ucsc.edu/index.php?title=Chains_Nets#Net_construction_example"
target="_blank">example</a> how nets are constructed on our wiki.</p>
<p>
At the beginning of each target species chromosome, a &quot;net&quot; line appears with the 
format:</p> 
<pre><code>net chromName chromSize</code></pre>
<p>
<em>chromName</em> is the target species chromosome name, and <em>chromSize</em> is the
chromosome size. For example:</p> 
<pre><code>net chr2L 23011544</code></pre>
<p>
Each target chromosome section in the file  starts with a net line. The net line is followed by a 
set of &quot;fill&quot; and &quot;gap&quot; lines.</p> 
<p>
<strong>File indentation:</strong> Line indentation level represents the parent/child relationship 
between records and is a necessary part of the net file format. Child records are indented one 
space from the parent, as shown in the example net file below.</p>
<pre><code>net chr2L 23011544
 fill 6004 3278 chrXR_group3a - 1396397 2164 id 25606 score 23114 ali 782 qDup 576 type top tN 0 qN 0 tR 36 qR 0 tTrf 0 qTrf 0
  gap 6065 2 chrXR_group3a - 1398498 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
  gap 6096 1485 chrXR_group3a - 1397572 897 tN 0 qN 0 tR 36 qR 0 tTrf 0 qTrf 0
   fill 6096 513 chrU - 5570675 533 id 48675 score 4435 ali 465 qDup 533 type nonSyn tN 0 qN 0 tR 0 qR 13 tTrf 0 qTrf 0
    gap 6116 8 chrU - 5571188 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
    gap 6156 5 chrU - 5571156 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
    gap 6184 3 chrU - 5571133 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
    gap 6212 18 chrU - 5571106 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
    gap 6244 9 chrU - 5571092 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
    gap 6340 2 chrU - 5570996 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
    gap 6515 3 chrU - 5570771 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
  gap 7623 1 chrXR_group3a - 1397530 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
  gap 7664 1007 chrXR_group3a - 1397008 482 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
   fill 7664 382 chrXL_group1e - 8262003 506 id 25608 score 10609 ali 364 qDup 506 type nonSyn tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
    gap 7784 4 chrXL_group1e - 8262361 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
    gap 7792 3 chrXL_group1e - 8262357 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
    gap 7921 2 chrXL_group1e - 8262126 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
    gap 7949 9 chrXL_group1e - 8262092 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
  gap 8693 1 chrXR_group3a - 1396985 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
 fill 9833 1251 chrU - 5562980 1239 id 48675 score 10720 ali 1124 qDup 1094 type top tN 0 qN 0 tR 22 qR 88 tTrf 0 qTrf 0
  gap 9966 7 chrU - 5564075 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
  gap 10015 3 chrU - 5564030 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
  gap 10088 2 chrU - 5563957 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0
  gap 10101 8 chrU - 5563946 0 tN 0 qN 0 tR 0 qR 0 tTrf 0 qTrf 0 </code></pre>

<h2>Field definitions</h2>
<p>
The net file consists of 7 fixed fields and a set of optional name/value pair fields. In the 
descriptions below, <em>target</em> refers to the reference species and <em>query</em> refers to the
aligning species.</p>

<h3>Fixed fields</h3>
<ul> 
  <li> 
  <strong><em>Class</em></strong> -- Either <em>fill</em> or <em>gap</em>. Fill refers to a portion 
  of a chain</li> 
  <li> 
  <em><strong>Start in chromosome</em></strong> --  (target species)</li> 
  <li> 
  <em><strong>Size</em></strong>  -- target species)</li> 
  <li> 
  <em><strong>Chromosome name</em></strong> -- (query species)</li> 
  <li> 
  <em><strong>Relative orientation</em></strong> -- between target and query species</li>
  <li> 
  <em><strong>Start in chromosome</em></strong> --  (query species)</li> 
  <li> 
  <em><strong>Size</em></strong> -- (query species)</li> 
</ul>

<h3>Optional fields (Name/value pairs)</h3>
<ul> 
  <li> 
  <em><strong>id</strong></em> -- ID of associated chain (gapped alignment), if any.</li>  
  <li> 
  <em><strong>score</strong></em> -- Score of associated chain.</li>  
  <li> 
  <em><strong>ali</strong></em>   -- Number of bases in alignments in chain.</li>  
  <li> 
  <em><strong>qFar</strong></em> -- For fill that is on the same chromosome as parent, how far fill 
  is from position predicted by parent. This helps determine if a rearrangement is local or if a 
  duplication is tandem.</li>  
  <li> 
  <em><strong>qOver</strong></em> -- Number of bases overlapping with parent gap on query side. 
  Generally, this will be near zero, except for inverts.</li>  
  <li> 
  <em><strong>qDup</strong></em> -- Number of bases in query region that are used twice or more in 
  net. This helps distinguish between a rearrangement and a duplication.</li>  
  <li> 
  <em><strong>type</em></strong> -- One of the following values: 
  <ul> 
    <li> 
    <em>top</em> -- Chain is top-level, not a gap filler.</li>  
    <li> 
    <em>syn</em> -- Chain is on same chromosome and in same direction as parent.</li>  
    <li> 
    <em>inv</em> -- Chain is on same chromosome on opposite direction from parent.</li>  
    <li> 
    <em>nonSyn</em> -- Chain is on a different chromosome from parent.</li>  
  </ul> 
  <li> 
  <em><strong>tN</strong></em> -- Number of unsequenced bases (<em>N</em>s) on target side.</li>  
  <li> 
  <em><strong>qN</strong></em> -- Number of unsequenced bases on query side.</li>  
  <li> 
  <em><strong>tR</strong></em> -- Number of bases in RepeatMasker masked repeats on target.</li>  
  <li> 
  <em><strong>qR</strong></em> -- Number of bases in RepeatMasker masked repeats on query.</li>  
  <li> 
  <em><strong>tNewR</strong></em> -- Bases in lineage-specific repeats on target.</li>  
  <li> 
  <em><strong>qNewR</strong></em> -- Bases in lineage-specific repeats on query.</li>  
  <li> 
  <em><strong>tOldR</strong></em> -- Bases in repeats predating split on target.</li>  
  <li> 
  <em><strong>qOldR</strong></em> -- Bases in repeats predating split on query.</li>  
  <li> 
  <em><strong>tTrf</strong></em> -- Bases in trf (Tandem Repeat Finder) repeats on target.</li>  
  <li> 
  <em><strong>qTrf</strong></em> -- Bases in trf repeats on query.</li>  
</ul>

<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->
