<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: Net Format</TITLE>
	<LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../inc/topbar.html"--></TD></TR>
<!--Content Tables------------------------------------------------------->
    <TR><TD COLSPAN=2 CELLPADDING=10>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    	    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		<TR><TD>	
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
		    <TR><TD>
			<FONT SIZE="4"><b>&nbsp;  Net Format</b></FONT>
		    </TD></TR>
		</TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
		    <TR><TH HEIGHT=10></TH></TR>
		    <TR><TD WIDTH=10>&nbsp;</TD><TD>
			<P>
			The <em>net</em> file format is used to describe
			the axtNet data that underlie the net alignment
			annotations in the Genome Browser. For a detailed
			description of the methods used to generate these
			data, refer to the Genome Browser description pages
			that accompany the Net alignment tracks.
			<P>
			<B><em>Example:</em></B>
			<FONT color="#0000FF"><PRE><TT>  fill 58159 3681 chr7 + 17527587 9297 id 43830 score 143688 qDup 8727 
	type top tN 0 qN 0 tR 309 qR 345 tNewR 309 qNewR 246 tOldR 0 qOldR 0 tTrf 29 qTrf 0
  gap 59899 104 chr7 + 17531210 4178 tN 0 qN 0 tR 0 qR 246 tNewR 0 qNewR 246 tOldR 0 qOldR 0 
  	tTrf 0 qTrf 0
  gap 60324 117 chr7 + 17535966 0 tN 0 qN 0 tR 0 qR 0 tNewR 0 qNewR 0 tOldR 0 qOldR 0 
	tTrf 0 qTrf 0
  fill 60325 101 chr7 + 17536081 101 id 5137619 score 6111 qOver 0 qFar 115 qDup 101 
	type syn tN 0 qN 0 tR 0 qR 0 tNewR 0 qNewR 0 tOldR 0 qOldR 0 tTrf 0 qTrf 0
  gap 60558 106 chr7 + 17536066 0 tN 0 qN 0 tR 0 qR 0 tNewR 0 qNewR 0 tOldR 0 qOldR 0 
	tTrf 0 qTrf 0
  fill 60558 106 chr7 - 17531782 4202 id 1670092 score 15263 qOver 0 qFar 82 qDup 4202 
	type inv tN 0 qN 0 tR 0 qR 246 tNewR 0 qNewR 246 tOldR 0 qOldR 0 tTrf 0 qTrf 0
  gap 60671 63 chr7 + 17536073 0 tN 0 qN 0 tR 0 qR 0 tNewR 0 qNewR 0 tOldR 0 qOldR 0 
	tTrf 0 qTrf 0
  fill 60671 26 chr7 - 17531815 26 id 1670092 score 15263 qOver 0 qFar 4232 qDup 26 
	type inv tN 0 qN 0 tR 0 qR 0 tNewR 0 qNewR 0 tOldR 0 qOldR 0 tTrf 0 qTrf 0
  gap 61069 320 chr7 + 17536433 0 tN 0 qN 0 tR 309 qR 0 tNewR 309 qNewR 0 tOldR 0 qOldR 0 
	tTrf 29 qTrf 0</TT></PRE></FONT>
			<P>
			<B>File format</B>
			<P>
			The net file consists of 7 fixed fields and a set
			of optional name/value pairs. In the descriptions
			below, <em>target</em> refers to the reference
			species and <em>query</em> refers to the aligning
			species.
			<P>
			<UL>
			<LI>
			<B>Fixed fields:</B>
			<UL>
			<LI>
			<B><em>Class</em></B>. Either <em>fill</em> or 
			<em>gap</em>.  Fill refers to a portion of a chain.
			<LI>
			<em><B>Start in chromosome</em></B> (target species) 
			<LI>
			<em><B>Size</em></B> (target species)
			<LI>
			<em><B>Chromsome name</em></B> (query species)
			<LI>
			<em><B>Relative orientation</em></B> between target 
			and query species.
			<LI>
			<em><B>Start in chromsome</em></B> (query species)
			<LI>
			<em><B>Size</em></B> (query species)
			</UL>
			<P>
			<LI>
			<B>Name/value pairs (optional):</B>
			<UL>
			<LI>
			<em><B>id</B></em> -- ID of associated chain (gapped 
			alignment), if any.
			<LI>
			<em><B>score</B></em> -- Score of associated chain.
			<LI>
			<em><B>ali</B></em>   -- Number of bases in 
			alignments in chain.
			<LI>
			<em><B>qFar</B></em> -- For fill that is on the same 
			chromosome as parent, how far fill is from position 
			predicted by parent.   This helps determine if a
       			rearrangement is local or if a duplication is tandem.
			<LI>
			<em><B>qOver</B></em> -- Number of bases overlapping 
			with parent gap on query side.  Generally, this will 
			be near zero, except for inverts. 
			<LI>
			<em><B>qDup</B></em> -- Number of bases in query 
			region that are used twice or more in net. This helps
			distinguish between a rearrangement and a duplication.
			<LI>
			<em><B>type</em></B> -- One of the following values:
   			<UL>
			<LI>
			<em>top</em> -- Chain is top-level, not a gap filler
   			<LI>
			<em>syn</em> -- Chain is on same chromosome and in 
			same direction as parent
   			<LI>
			<em>inv</em> -- Chain is on same chromosome on 
			opposite direction from parent 
			<LI>
			<em>nonSyn</em> -- Chain is on a different chromosome 
			from parent
			</UL>
			<LI>
			<em><B>tN</B></em> -- Number of unsequenced bases 
			(<em>N</em>s) on target side
			<LI>
			<em><B>qN</B></em> -- Number of unsequenced 
			bases on query side
			<LI>
			<em><B>tR</B></em> -- Number of bases in RepeatMasker 
			masked repeats on target.
			<LI>
			<em><B>qR</B></em> -- Number of bases in RepeatMasker 
		 	masked repeats on query.
			<LI>
			<em><B>tNewR</B></em> -- Bases in lineage-specific 
			repeats on target.
			<LI>
			<em><B>qNewR</B></em> -- Bases in lineage-specific 
			repeats on query.
			<LI>
			<em><B>tOldR</B></em> -- Bases in repeats predating 
			split on target.
			<LI>
			<em><B>qOldR</B></em> -- Bases in repeats predating 
			split on query.
			<LI>
			<em><B>tTrf</B></em> -- Bases in trf (Tandem Repeat 
			Finder) repeats on target.
			<LI>
			<em><B>qTrf</B></em> -- Bases in trf repeats on query.
			</UL>
			</UL>
		    </TD><TD WIDTH=15></TD>
		</TR>
		</TABLE>
	    <BR></TD></TR>
	    </TABLE>
	</TD></TR>
	</TABLE>
    </TD></TR>
    </TABLE>
</BODY></HTML>
