<H2>Description</H2>
<P>
This track shows all occurrences of a selected short motif within the 
displayed position range of the assembly sequence. It is useful for finding 
oligonucleotides, restriction sites, or other recurring short sequences within
the assembly.  In full display mode, each motif occurrence is labeled by the 
strand on which the match is located, followed by the starting coordinate of 
the match. In cases where the input motif sequence is identical to its reverse
complement, only the match on the &quot;+&quot; strand is shown.</P>
<P>
The track may be configured to search for any short sequence of 2 - 30 bases 
in length. Sequences may include <A HREF="iupac.html">IUPAC ambiguity
codes</A>. To change the motif, open the track's description page (by clicking
the track control label or the mini-button to the left of the track), then 
type a new sequence into the text box.</P>

<H2>Credits</H2>
<P>
This track was generated by 
<A HREF="mailto:&#107;&#101;&#110;t&#64;&#115;&#111;e.
&#117;&#99;&#115;&#99;.
&#101;&#100;&#117;">Jim Kent</A> of the UCSC Genome Bioinformatics Group.</P>

