<!DOCTYPE html>
<!--#set var="TITLE" value="Genome Browser Queries" -->
<!--#set var="ROOT" value="../.." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<h1>Querying the Genome Browser</h1> 
<p> 
From the <a href="../../cgi-bin/hgGateway">Genomes page</a>, you can jump to the default 
position of an assembly by clicking the &quot;Go&quot; 
button or you can specify a particular genome position in a variety of formats. These same
formats are valid in the search bar above the main 
<a href="../../cgi-bin/hgTracks">Genome Browser track display</a>.</p>
<p>
In addition to the positional queries described below, any search term can be used to 
find matches in track data, track names and/or descriptions, help docs, and public hub 
track names and/or descriptions. See our <a href="/cgi-bin/hgSearch">search
page</a> for more details on the search functionality.</p>
 
<p> 
Valid position queries can include:</p> 
<ul> 
  <li>
  Chromosome numbers</li> 
  <li>
  Chromosomal coordinate ranges</li> 
  <li>
  Gene names</li> 
  <li>
  Accession numbers</li> 
  <li>
  An mRNA, EST or STS marker</li> 
  <li>
  Keywords from the GenBank description of an mRNA</li> 
  <li>
  <a href="http://varnomen.hgvs.org/" target="_blank">HGVS</a> terms</li>
  <li>
  gnomAD variant IDs</li>
  <li>
  HGVS and accession searches on outdated RefSeq accession versions is available on hg38</li>
</ul> 
<p> 
To specify a genome position:</p> 
<ol> 
  <li>
  Select the desired clade, genome and assembly</li> 
  <li>
  Enter the desired query in the &quot;Position/Search Term&quot; box (see sample queries 
  below)</li> 
  <li>
  Click the &quot;Go&quot; button</li> 
</ol> 

<p> 
A query may have multiple results. If this is the case, a results page will appear listing 
each result along with the track it is associated with. Once selected, the result will be displayed 
in the Browser with a highlighted label, making it easier to identify. If you have further
questions, you can search the <a href="/FAQ/index.html" target="_blank">Genome Browser FAQ</a>
page and find links to further resources. Also, developers of track hubs can create
<a href="hubQuickStartSearch.html" target="_blank">searchable track hubs</a> using the
<a href="trix.html" target="_blank"><code>searchTrix</code></a> setting.</p>

<p>
To quickly jump to a codon or exon of a gene transcript:</p> 
<ol> 
  <li>
  Use one of the searches below to jump to a gene, to show all transcripts of a gene or range of interest
  <li>
  Right-click any transcript, select "Choose exon" or "Zoom to codon" and enter the exon
  or codon position of interest
</ol>
</p>

<h2>Sample queries</h2> 
<p> 
Below is a list of examples that might be used to query the Genome Browser. Note that not every 
query listed here will produce a result in every assembly. The list serves only to illustrate the 
different types of queries that can be performed.  
<table border="1"> 
  <tr><th width="200">Query</th><th width="250">Genome Browser Response</th></tr> 
  <tr>
    <td>chr7</td>
    <td>Displays all of chromosome 7</td></tr> 
  <tr>
    <td>chr3:1-1000000</td>
    <td>Displays the first million bases of chromosome 3, counting from the p-arm telomere</td></tr>
  <tr>
    <td>3:1-1000000</td>
    <td>Displays the first million bases of chromosome 3, Ensembl format chromosome names</td></tr>
  <tr>
    <td>chr3 0 1000000</td>
    <td>Displays the first million bases of chromosome 3; BED format</td></tr>
  <tr>
    <td>NC_000007.14:1-1000000</td>
    <td>Displays the first million bases of chromosome 3, RefSeq format</td></tr>
  <tr>
    <td>CM000665.2:1-1000000</td>
    <td>Displays the first million bases of chromosome 3, GenBank/INSDC format</td></tr>
  <tr>
    <td>chr3:1000000+2000</td>
    <td>Displays a region of chromosome 3 that spans 2000 bases, starting with position 1000000</td>
    </tr> 
  <tr>
    <td>chrUn_GL000213v1</td>
    <td>Displays all of the unplaced contig GL000213v1</td></tr>
  <tr>
    <td>chr3_GL000221v1_random</td>
    <td>Displays the unlocalized contig GL000221v1</td></tr>
  <tr>
    <td>chr1_KN196472v1_fix</td>
    <td>Displays all of patch fix KN196472v1</td></tr>
  <tr>
    <td>20p13</td>
    <td>Displays the region for band p13 on chromosome 20</td></tr> 
  <tr>
  <tr>
    <td>GTATGTAGCCACGGAGCACCATTACCTGTCACCATTACCTGAATGGCTA</td>
    <td>Displays the first best match to this DNA sequence, e.g. chr21:33034835-33034883 for hg19</td></tr> 
  <tr>
    <td>AA205474</td>
    <td>Displays the region containing the EST with GenBank accession AA205474 in the BRCA1 cancer 
    gene on chromosome 17</td></tr> 
  <tr>
    <td>AC008101</td>
    <td>Displays the region containing the clone with GenBank accession AC008101</td></tr> 
  <tr>
    <td>AF083811</td>
    <td>Displays the region containing the mRNA with GenBank accession number AF083811</td></tr> 
  <tr>
    <td>NM_017414</td>
    <td>Displays the region containing RefSeq identifier NM_017414</td></tr> 
  <tr>
    <td>NP_059110</td>
    <td>Displays the region containing protein accession number NP_059110</td></tr> 
  <tr>
    <td>PRNP</td>
    <td>Displays the region containing HUGO Gene Nomenclature Committee identifier PRNP</td></tr> 
  <tr>
  <tr>
    <td>Q99697</td>
    <td>Displays the region containing the alignment of the UniProt/SwissProt protein sequence with accession Q99697 (PITX2)</td></tr> 
  <tr>
    <td nowrap>RH18061;RH80175<br>15q11;15q13<br>NM_012090.5;NM_012421.4</td>
    <td nowrap>Displays the region between genome landmarks, such as the STS markers RH18061 and 
    RH80175, or chromosome<br> bands 15q11 to 15q13, or SNPs NM_000310.4 and NM_012090.5. This syntax
    may also be used for other range queries,<br>such as between uniquely determined ESTs, mRNAs, 
    refSeqs, SNPS, etc.</td></tr>
  <tr>
    <td>NR_026861.1:1-1000</td>
    <td>Works with any other type of accession from this page: Displays the first 1000bp of NR_026861.1</td></tr> 
  <tr id="HGVS">
    <td nowrap>NM_000310.4(PPT1):c.271_287del17insTT<br> NM_007262.5(PARK7):c.-24+75_-24+92dup<br>
    NM_006172.4(NPPA):c.456_*1delAA<br> MYH11:c.503-14_503-12del<br> 
    NM_198576.4(AGRN):c.1057C&gt;T<br> NM_198056.3:c.1654G&gt;T<br> NP_002993.1:p.Asp92Glu<br>
    NP_002993.1:p.D92E<br> BRCA1 Ala744Cys<br> BRCA1 A744C<br> LRG_100t1:c.4G>A<br> LRG_100t1:n.1<br>
    LRG_456p1:p.Ser190Leu<br>LRG_321:g.16409_16461del<br>ENST00000002596.6:c.-108-6848A&gt;G<br>
    ENSP00000005178.5:p.Val20Gly<br>
    chrX:g.31500000_31600000del<br> NR_111987:n.-1 <br> NM_015102.5:n.3038-2<br>
    NM_001372044:c.1528_1530del</td>
    <td>Displays the region that matches the <a href="http://varnomen.hgvs.org/" target="_blank">HGVS</a> 
        expression, usually in the format <tt>&lt;transcript or protein&gt;:&lt;position&gt; &lt;amino acid or nucleotide change&gt;</tt><br>If a gene symbol is used, HGVS search will try all RefSeq transcripts to find the nucleotide or amino acid at the position indicated in the expression. If there are multiple matches, a disambiguation page will be shown. If the RefSeq sequence differs from the genome sequence, then currently the search will use the genome, not the transcript, for codon counting and amino acid / nucleotide  comparison. Please contact us if this is inconvenient.</td></tr> 
  <tr>
    <td>NM_198056.2:c.1A&gt;C</td>
    <td>An example of an HGVS search on a previous NM version that is now outdated. 
        Support for previous NM accessions is only available on hg38.</td></tr>
  <tr>
    <td>1-55051215-G-GA</td>
    <td>Displays the region that matches the gnomAD variant ID, 1-55051215-G-GA</td>
<!-- commented out -- not working 
  <tr> 
    <td>15q11;15q13</td>
    <td>bands 15q11 to 15q13, or SNPs rs1042522 and rs1800370. This syntax may also be used for 
    other range queries</td></tr> 
  <tr>
    <td>rs1042522;rs1800370</td> 
    <td>such as between uniquely determined ESTs, mRNAs, refSeqs, etc.</td></tr> 
--> 
  <tr>
    <td>essv8694097</td>
    <td>Displays the region covering the copy number variant with the accession essv8694097 in the
        Database of Genomic Variants (DGV)</td></tr>
  <tr>
    <td>nssv3446126</td>
    <td>Displays the region covering the copy number variant with the accession nssv3446126 in the cases of developmental delay</td></tr>
  <tr>
    <td>CTD-3071L10</td>
    <td>Displays the region covering the CTD-3071L10 NCBI clone end mapping in the NCBI Clone DB database</td></tr>
  <tr>
    <td>nssv16167444</td>
    <td>Displays the region covering the common copy number genomic variant with the accession nssv16167444 in the 
        nstd186 (NCBI Curated Common Structural Variants) dataset</td></tr>
  <tr>
    <td>rs1333049</td>
    <td>Displays results for annotations matching this rsID, including dbSNP database</td></tr>
  <tr>
    <td>COSM6161404</td>
    <td>Displays the region covering COSM6161404 in the Catalogue Of Somatic Mutations In Cancer (COSMIC) database</td></tr>
  <tr>
    <td>nssv3395351</td>
    <td>Displays the region covering ClinVar Copy Number Variant with the accession nssv3395351 in the ClinVar database</td></tr>
  <tr>
    <td>BRCT_assoc</td>
    <td>Displays the region covering the manually-curated Pfam-A domain BRCT_assoc found in GENCODE Genes</td></tr>
  <tr>
    <td>U133A:219211_at</td>
    <td>Displays the region containing the consensus and exemplar sequences used for the selection of probes on the Affymetrix HG-U133A chips</td></tr>
  <tr>
    <td>chr1 0 1000</td>
    <td>When entered without ":" and "-", uses 0-based, half-open coordinates (like custom tracks and internal table coordinates), so displays chr1:1-1000</td>
  </tr> 
  <tr>
    <td>pseudogene mRNA</td> 
    <td>Lists transcribed pseudogenes, but not cDNAs</td></tr> 
  <tr> 
    <td>p53</td>
    <td>Lists mRNAs related to the p53 tumor suppressor</td></tr> 
  <tr> 
    <td>T-cell receptor</td> 
    <td>Lists mRNAs for T-cell receptor genes in GenBank</td></tr> 
  <tr>
    <td>breast cancer</td> 
    <td>Lists mRNAs associated with breast cancer</td></tr> 
  <tr> 
    <td>homeobox caudal</td> 
    <td>Lists mRNAs for caudal homeobox genes</td></tr> 
  <tr> 
    <td>zinc finger</td> 
    <td>Lists zinc finger mRNAs</td></tr> 
  <tr> 
    <td>kruppel zinc finger</td> 
    <td>Lists only kruppel-like zinc fingers</td></tr> 
  <tr> 
    <td>huntington</td> 
    <td>Lists candidate genes associated with Huntington's disease</td></tr> 
  <tr> 
    <td>zahler</td> 
    <td>Lists mRNAs deposited by a scientist named Zahler</td></tr> 
  <tr>
    <td>Evans,J.E.</td> 
    <td>Lists mRNAs deposited by co-author J.E. Evans</td></tr> 
</table>
<p> 
Use this last format for author queries. Although GenBank requires the search format 
<em>Evans JE</em>, internally it uses the format <em>Evans,J.E.</em>.</p>

<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->
