<!DOCTYPE html>
<!--#set var="TITLE" value="Genome Browser Queries" -->
<!--#set var="ROOT" value="../.." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<h1>Querying the Genome Browser</h1> 
<p> 
You may browse to the default genome position of an assembly when clicking the &quot;submit&quot; 
button or you may specify a genome position in a variety of formats.</p> 
<p> 
Valid position queries can include:</p> 
<ul> 
  <li>
  Chromosome numbers</li> 
  <li>
  Chromosomal coordinate ranges</li> 
  <li>
  Gene names</li> 
  <li>
  Accession numbers</li> 
  <li>
  An mRNA, EST or STS marker</li> 
  <li>
  Keywords from the GenBank description of an mRNA</li> 
  <li>
  <a href="http://varnomen.hgvs.org/" target="_blank">HGVS</a> terms</li>
</ul> 
<p> 
To specify a genome position:</p> 
<ol> 
  <li>
  Select the desired clade, genome and assembly</li> 
  <li>
  Enter the desired query in the &quot;search term&quot; box (see sample queries below)</li> 
  <li>
  Click the &quot;submit&quot; button</li> 
</ol> 
<p> 
Note: A query may have multiple results. If this is the case, a results page will appear listing 
each result along with the track it is associated with. Once selected, the result will be displayed 
in the Browser with a highlighted tag, making it easier to identify.</p> 

<h2>Sample queries</h2> 
<p> 
Below is a list of examples that might be used to query the Genome Browser. Note that not every 
query listed here will produce a result in every assembly. The list serves only to illustrate the 
different types of queries that can be performed.  
<table border="1"> 
  <tr><th width="200">Query</th><th>Genome Browser Response</th></tr> 
  <tr>
    <td>chr7</td>
    <td>Displays all of chromosome 7</td></tr> 
  <tr>
    <td>chr3:1-1000000</td>
    <td>Displays the first million bases of chromosome 3, counting from the p-arm telomere</td></tr>
  <tr>
    <td>chr3:1000000+2000</td>
    <td>Displays a region of chromosome 3 that spans 2000 bases, starting with position 1000000</td>
    </tr> 
  <tr>
    <td>chrUn_gl000212</td>
    <td>Displays all of the unplaced contig gl000212</td></tr> 
  <tr>
    <td>chr6_GL285171_random</td>
    <td>Displays the unlocalized contig GL285171</td></tr> 
  <tr>
    <td>GL192360</td>
    <td>Displays all of scaffold GL192360</td></tr> 
  <tr>
    <td>20p13</td>
    <td>Displays the region for band p13 on chromosome 20</td></tr> 
  <tr>
  <tr>
    <td>GTATGTAGCCACGGAGCACCATTACCTGTCACCATTACCTGAATGGCTA</td>
    <td>Displays the first best match to this DNA sequence, e.g. chr21:33034835-33034883 for hg19</td>
  </tr> 
  <tr>
    <td>374180</td>
    <td>Displays the region containing Entrez Gene identifier 374180</td></tr> 
  <tr>
    <td>AA205474</td>
    <td>Displays the region containing the EST with GenBank accession AA205474 in the BRCA1 cancer 
    gene on chromosome 17</td></tr> 
  <tr>
    <td>AC008101</td>
    <td>Displays the region containing the clone with GenBank accession AC008101</td></tr> 
  <tr>
    <td>AF083811</td>
    <td>Displays the region containing the mRNA with GenBank accession number AF083811</td></tr> 
  <tr>
    <td>NM_017414</td>
    <td>Displays the region containing RefSeq identifier NM_017414</td></tr> 
  <tr>
    <td>NP_059110</td>
    <td>Displays the region containing protein accession number NP_059110</td></tr> 
  <tr>
    <td>PRNP</td>
    <td>Displays the region containing HUGO Gene Nomenclature Committee identifier PRNP</td></tr> 
  <tr>
    <td nowrap>RH18061;RH80175<br>15q11;15q13<br>rs1042522;rs1800370</td> 
    <td nowrap>Displays the region between genome landmarks, such as the STS markers RH18061 and 
    RH80175, or chromosome<br> bands 15q11 to 15q13, or SNPs rs1042522 and rs1800370. This syntax 
    may also be used for other range queries,<br>such as between uniquely determined ESTs, mRNAs, 
    refSeqs, etc.</td></tr> 
  <tr id="HGVS">
    <td nowrap>NM_000310.3(PPT1):c.271_287del17insTT<br> NM_007262.4(PARK7):c.-24+75_-24+92dup<br> 
    NM_006172.3(NPPA):c.456_*1delAA<br> MYH11:c.503-14_503-12del<br> 
    NM_198576.3(AGRN):c.1057C&gt;T<br> NM_198056.2:c.1654G&gt;T<br> NP_002993.1:p.Asp92Glu<br> 
    NP_002993.1:p.D92E<br> BRCA1 Ala744Cys<br> BRCA1 A744C<br> LRG_100t1:c.4G>A<br> LRG_100t1:n.1<br>
    LRG_456p1:p.Ser190Leu<br>LRG_321:g.16409_16461del<br>ENST00000002596.5:c.-108-6848A&gt;G<br>
    ENSP00000005178.5:p.Val20Gly<br>
    chrX:g.31500000_31600000del<br> NR_111987:n.-1 <br> NM_015102.4:n.3086-2<br>
    NM_033517:c.1307_1309del</td> 
    <td>Displays the region containing <a href="http://varnomen.hgvs.org/" target="_blank">HGVS</a> 
    term</td></tr> 
<!-- commented out -- not working 
  <tr> 
    <td>15q11;15q13</td>
    <td>bands 15q11 to 15q13, or SNPs rs1042522 and rs1800370. This syntax may also be used for 
    other range queries</td></tr> 
  <tr>
    <td>rs1042522;rs1800370</td> 
    <td>such as between uniquely determined ESTs, mRNAs, refSeqs, etc.</td></tr> 
--> 
  <tr>
    <td>chr1 0 1000</td>
    <td>When entered without ":" and "-", uses 0-based, half-open coordinates (like custom tracks and internal table coordinates), so displays chr1:1-1000</td>
  </tr> 
  <tr>
    <td>pseudogene mRNA</td> 
    <td>Lists transcribed pseudogenes, but not cDNAs</td></tr> 
  <tr> 
    <td>p53</td>
    <td>Lists mRNAs related to the p53 tumor suppressor</td></tr> 
  <tr> 
    <td>T-cell receptor</td> 
    <td>Lists mRNAs for T-cell receptor genes in GenBank</td></tr> 
  <tr>
    <td>breast cancer</td> 
    <td>Lists mRNAs associated with breast cancer</td></tr> 
  <tr> 
    <td>homeobox caudal</td> 
    <td>Lists mRNAs for caudal homeobox genes</td></tr> 
  <tr> 
    <td>zinc finger</td> 
    <td>Lists zinc finger mRNAs</td></tr> 
  <tr> 
    <td>kruppel zinc finger</td> 
    <td>Lists only kruppel-like zinc fingers</td></tr> 
  <tr> 
    <td>huntington</td> 
    <td>Lists candidate genes associated with Huntington's disease</td></tr> 
  <tr> 
    <td>zahler</td> 
    <td>Lists mRNAs deposited by a scientist named Zahler</td></tr> 
  <tr>
    <td>Evans,J.E.</td> 
    <td>Lists mRNAs deposited by co-author J.E. Evans</td></tr> 
</table>
<p> 
Use this last format for author queries. Although GenBank requires the search format 
<em>Evans JE</em>, internally it uses the format <em>Evans,J.E.</em>.</p>

<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->
