<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=286 HEIGHT=215>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
    TARGET=_blank>
    <IMG SRC="/images/Oryctolagus_cuniculus.jpg" WIDTH=261 HEIGHT=200 ALT="Rabbit"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Oryctolagus cuniculus</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of Wikimedia Commons
      (<A HREF="http://cs.wikipedia.org/wiki/Wikipedista:Dezidor"
      TARGET=_blank>Králík divoký</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Apr. 2009 <em>Oryctolagus cuniculus</em> draft assembly 
(Broad Institute oryCun2 (NCBI project 12819, AAGW00000000)) was
produced by the Broad Institute at MIT and Harvard. For more information
about this assembly, see
<A HREF="http://www.ncbi.nlm.nih.gov/assembly/5758/" TARGET="_blank">
OryCun2.0</A> in the NCBI Assembly database.

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
Rabbit genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>chrM</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of chromosome M</TD>
    </TR>
    <TR><TD VALIGN=Top>chr3:1-1000000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of chr 3</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr3:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr 3 that spans 2000 bases, starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>EF606761</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number EF606761</TD>
    </TR>
    <TR><TD VALIGN=Top>DN888572</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays selection of regions of EST alignments
		with GenBank accession DN888572</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>edery</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Edery</TD>
    </TR>
    <TR><TD VALIGN=Top>Wistow,G.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author G. Wistow</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Wistow G</em>, internally it
	uses the format <em>Wistow,G.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
The genome has been sequenced to 7.48X coverage.  Included in this assembly
is sequence of length 2,247,769,349 bases for chromosomes 1 thru 21,
the X chromosome as chrX and the mitochondrian sequence
(genbank accession NC_001913) as chrM.
Also included are 3,219 unplaced sequences of length
489,721,152 bases with names of chrUn0001 thru chrUn3219.
<BR>
See also:
<A HREF="http://www.broadinstitute.org/science/projects/mammals-models/rabbit/rabbit-genome-sequencing-project"
TARGET=_blank>Broad Institute</A>
<A HREF="http://www.broadinstitute.org/ftp/pub/assemblies/mammals/rabbit/oryCun2/"
TARGET=_blank>Broad oryCun2 data</A>
</P>

<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/oryCun2/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#rabbit">Downloads</A> page. 

Many thanks to the Broad Institute of MIT/Harvard, NHGRI, and the many
institutions who contributed to the sequencing and mapping effort for this
release.  The Rabbit browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#rabbit_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
<HR>
<H3>Genbank Pipeline Details</H3>
For the purposes of the Genbank alignment pipeline, this assembly is considered to be: <B>well-ordered</B>.
