<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 3.2//EN">

<HTML>
  <HEAD>
    <META name="generator" content=
    "HTML Tidy for Linux/x86 (vers 1st June 2002), see www.w3.org">

    <TITLE>GenBank/RefSeq Genome Setup</TITLE>
  </HEAD>

  <BODY bgcolor="white">
    <H1>GenBank/RefSeq Genome Setup</H1>
    The page describes that process of setting up the GenBank/RefSeq update
    process for a new genome or new assembly. It's assumed that the
    automated download and build already in place and in <CODE>/cluster/data/genbank/</CODE>.

    <H3>Recent changes</H3>
    <OL>
    <LI> The working directory for initial alignments is now removed
    unless <CODE>-keep</CODE> is specified.
    <LI> <CODE>gbBlat</CODE> no longer needs to be modified to specify
    the ooc file.  It is now specified in the <CODE>genbank.conf</CODE>
    file.
    <LI> Many changed to the <a href="config.html"><CODE>genbank.conf</CODE></a>
         file.
    </OL>


    <H3>Initial Alignment</H3>
    Building the initial alignments is a long process, involving gigabytes
    of disk space and over a day of cluster time.  Do not be surprised if
    it does not go smoothly.  Manual intervention maybe required to
    correct problems.

    <OL>
      <LI> Make sure <CODE>ssh</CODE> is configure to
           <A HREF="ssh.html">not require a passphase</A>.
      <LI> In this document, <CODE>$db</CODE> refers to the database being aligned.
           Substitute the actual database name (e.g. <CODE>hg15</CODE>).
      <LI> If this is the first time this organism has been aligned, some source files
           need to be edited.  The genbank update code is under 
           <CODE>kent/src/hg/makeDb/genbank/</CODE>
      <UL>
      <LI> A mapping between the prefix used in the UCSC databases names
           (e.g. <CODE>hg</CODE> of <CODE>hg13</CODE>) and the organism names
           used in GenBank needs to be defined.  This is done by editing
           <CODE>genbank/src/lib/gbGenome.c</CODE> and rebuilding the programs.  It maybe
           necessary to define multiple organism name mappings.  A list of 
           organism in GenBank/RefSeq, along with the count of cDNAs is in:<BR>
           <PRE>
           /cluster/data/genbank/data/organism.lst
           </PRE>
      <LI> <CODE>cd</CODE> to the top of the genbank source (<CODE>kent/src/hg/makeDb/genbank/</CODE>)
      <LI> <CODE>make</CODE> to test if the source builds
      <LI> <CODE>make install-server</CODE> to update <CODE>/cluster/data/genbank/</CODE>.
      <LI> Once these changes are debugged and committed, ask <A HREF="mailto:markd@cse.ucsc.edu"><CODE>markd</CODE></A>
           to update the round-robin code.

      </UL>
      <LI> Edit <A HREF="config.html"><CODE>kent/src/hg/makeDb/genbank/etc/genbank.conf</CODE></A> to 
           configure this databases.   Must set:
           <UL>
           <LI> <CODE>$db.serverGenome</CODE>
           <LI> <CODE>$db.clusterGenome</CODE>
           <LI> <CODE>$db.lift</CODE>
           </UL>
           You may want to set some database load options to override the defaults.
           If you have a pseudo chromosome with unplaced sequence, be sure to
           specify <code><em>$db</em>.align.maxGapChrs</code> and a lift file.
           <p>
           Commit your changes and then go to the top of the genbank source
           tree and update the installed genbank etc files with:
           <PRE>
           make etc-update
           </PRE>
           
      <LI><CODE>ssh <em>fileServer</em></CODE><BR>
          Where <CODE><em>fileServer</em></CODE> is the NFS server with
          <CODE>/cluster/data/genbank/</CODE>.

      <LI><CODE>cd /cluster/data/genbank</CODE><BR>
          This directory is <CODE>$gbRoot</CODE>.

      <LI> <CODE>nice bin/gbAlignStep -initial <EM>$db</EM>&amp;</CODE><BR>
           <P>This will run the entire alignment process.
           The <CODE>-initial</CODE> option defaults several parameters for
           and initial alignment and prevents this alignment from blocking
           the automatic daily alignments.
           <BLOCKQUOTE>
           Warning: <CODE>gbAlignStep</CODE> and other GenBank do not currently accept options after
           the positional arguments (i.e. the databases).
           </BLOCKQUOTE>

           <P>
           All output is saved in the log file.
           <P> If your organism has xeno ESTs enabled,
               it's a good idea to start out by aligning and loading just the
               the mRNAs, as this will go much faster.  Two options control
               what is aligned:
               <UL>
               <LI> <CODE>-srcDb=<EM>name</EM></CODE> - Restrict the source
                  database to either <CODE>genbank</CODE> or <CODE>refseq</CODE>.
               <LI> <CODE>-type=<EM>name</EM></CODE> - Restrict the type of sequence
                  processeed to either <CODE>mrna</CODE> or <CODE>est</CODE>.
               </UL>
               Note that since the alignment processs only aligns what needs
               to be aligned, no option is required when doing the ESTs after
               an initial mRNA alignment.

          <P> If anything fails, a subset of the tasks done by
              <CODE>gbAlignStep</CODE> script can be rerun after correcting
              the problem.  This is done using the
              <CODE>-continue=<EM>subtask</EM></CODE> option with
              <CODE><EM>subtask</EM></CODE> is either
              <UL>
              <LI><CODE>copy</CODE> - continue with coping to the iserver,
              this skips extracting the sequences to align.
              <LI><CODE>run</CODE> - Continue with parasol blat run.
              <LI><CODE>finish</CODE> - finish, alignments, doing
              lifting and filtering.
              </UL>

              If the parasol alignment run fails,  then can be
              continued using parasol directly,
              followed by an <CODE>gbAlignStep</CODE> with
              <CODE>-continue=finish</CODE>.   If parasol loses track of
              the jobs, one can use the <CODE>parasol recover</CODE>
              command to generate a new jobs file with the jobs
              that have not completed.
    </OL>

    <H3>Initial Database Load</H3>

    <OL>
      <LI><CODE>nice bin/gbDbLoadStep -drop -initialLoad $db</CODE></LI>
      <LI>
          <UL>
          <LI> The <CODE>-drop</CODE> option drops any existing GenBank or RefSeq
               tables before loading.
          <LI> If an initial load was done using only the mRNAs, it will be
               most likely be <B>much</B> faster to drop all of the GenBank
               tables and load with the <CODE>-initialLoad</CODE> option
               when loading the ESTs.
          </UL>
      <LI> After an initial review of the loaded alignments:
          <ul>
          <li> Enable daily alignment of the databases by adding the databases to <code>etc/align.dbs</code>.
          <li> Enable databases update of hgwdev by adding the databases to <code>etc/hgwdev.dbs</code>.
          <li> <code> make update-etc</code>

          </ul>
    </OL>

    <H3>Realigning Tracks</H3>
    It maybe necessary to realign and reload tracks to change alignment
    parameters or other attributes.  This is fairly straight forward
    when a genome databases is initially being built.  It's more complex
    if one has to sync up multiple systems.

    <OL>
      <LI> If automated alignment or update has been enabled for the database,
           disable it by editing <CODE>$gbRoot/etc/align.dbs</CODE>.
      <LI> Make sure an automated alignment isn't current running.
      <LI> To triger a realignment, on needs to remove the related files
           for some partation of the data for all updates.  These live
           under either the genbank or refseq alignment directories,
           for example:
           <UL>
           <LI><CODE>data/aligned/genbank.139.0/hg16/</CODE>
           <LI><CODE>data/aligned/refseq.139.0/hg16/</CODE>
           </UL>
           To realign native RefSeq mRNAs for hg16, one would remove:
           <UL>
           <LI><CODE>data/aligned/refseq.139.0/hg16/*/mrna.native.*</CODE>
           </UL>
           To realign xeno GeneBank ESTs for hg16, one would remove:
           <UL>
           <LI><CODE>data/aligned/refseq.139.0/hg16/*/est.*.xeno.*</CODE>
           </UL>

      <LI> Do an initial alignment as described above, restricting 
           with <CODE>-srcDb</CODE> and <CODE>-type</CODE>.
      <LI> Reload the database with the partation of data that was realigned.
           The <CODE>-srcDb</CODE> and <CODE>-type</CODE> options restrict
           the subset.  The organism category (native or xeno) isn't 
           specified.  Reloading of ESTs isn't supported, use <CODE>-drop</CODE>
           and <CODE>-initialLoad</CODE> instead.
          <UL>
          <LI><CODE>nice bin/gbDbLoadStep -reload -srcDb=genbank -type=mrna $db</CODE></LI>
          </UL>
      </OL>
<!--
FIXME: 
  -  mention cd-ing to /cluster/data/genbank and then doing a du to find machine.
  - mention that build takes place on kk.
  - alignment directory, monitor with list batches
-->

  </BODY>
</HTML>

