<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=185>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Ornithorhynchus%20anatinus"
    TARGET=_blank>
    <IMG SRC="/images/platypus.jpg" WIDTH=200 HEIGHT=135 ALT="Platypus"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1>Platypus (<em>Ornithorhynchus anatinus</em>)<BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of 
      (<A HREF="http://commons.wikimedia.org/wiki/Image:Platypus.jpg" 
      TARGET=_blank>Wikimedia Commons</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Dec. 2006 platypus (<em>Ornithorhynchus anatinus</em>) draft assembly 
(<A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Ornithorhynchus%20anatinus"
TARGET=_BLANK>WUSTL version 6.0?</A>) was produced by the Genome
Sequencing Center at Washington University, St. Louis.

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
Platypus genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>chrX5</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of chromosome X5</TD>
    </TR>
    <TR><TD VALIGN=Top>chr3:1-1000000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of chr 3</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr3:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr 3 that spans 2000 bases, starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>BX950328</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number BX950328</TD>
    </TR>
    <TR><TD VALIGN=Top>BU335953</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		BU335953</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>pseudogene mRNA</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>zhang</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zhang</TD>
    </TR>
    <TR><TD VALIGN=Top>Taguchi,T.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author T. Taguchi</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Taguchi T</em>, internally it
	uses the format <em>Taguchi,T.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
From the assembly release notes:</P>
<P>
The platypus (<em>Ornithorhynchus anatinus</em>) genome of a female nicknamed
&quot;Glennie&quot; (collected at the Upper Barnard River on Glen Rock Station,
New South Wales) was sequenced to a total of 6x whole genome coverage.
The sequencing strategy utilized combined whole genome shotgun
plasmid, fosmid and BAC end sequences.  The combined sequence reads
were assembled using the PCAP software (<A 
HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=pubmed&cmd=Retrieve&dopt=AbstractPlus&list_uids=12952883&query_hl=1&itool=pubmed_docsum" 
TARGET=_BLANK>Genome Res. 13(9):2164-70
2003</A>) using stringent parameters. After initial assembly with the PCAP
software, supercontigs (ordered/oriented contigs; contigs are
contiguous sequences not interrupted by gaps) were linked to the
physical map (Washington University Genome Sequencing Center) using
BAC end sequences and in silico digests of the sequence itself. The
physical map, then, was used to organize the supercontigs into
"ultracontigs" (ordered/oriented supercontigs).  With the exception of
those supercontigs with alignments at >95% identity to a platypus
EST (Washington University Genome Sequencing Center), supercontigs smaller
than 2kb were removed from the data set prior to submission if they
were >97% identical over >97% of their length to other ultracontigs
larger than 2kb or if they were deemed to be >95% repetitive (based on
analysis using RECON (<A 
HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=pubmed&cmd=Retrieve&dopt=AbstractPlus&list_uids=12176934&query_hl=3&itool=pubmed_docsum" 
TARGET=_BLANK>Bao and Eddy, 2002</A>) for repeat
identification). Further, singleton contigs (those not part of a
supercontig or ultracontig) smaller than 500bp that did not have
an alignment of >95% identity to a platypus EST were not submitted.
</P>
<P>
The assembly is composed of 205,534 supercontigs (of those, using the
physical map 4,197 supercontigs were organized into 689 ultracontigs)
covering 1.84 Gb of actual sequence (without including estimated gap
sizes) or almost 2.0Gb including gap sizes. Of the 1.84Gb, 437Mb (1507
supercontigs organized into 145 ultracontigs) have been anchored and
ordered along platypus chromosomes using the physical map in
combination with FISH data. The N50 statistic is defined as the length
L such that 50% of all nucleotides are contained in contigs of size at
least L.  The N50 number is 298 and the N50 size is 967kb.
</P>
<P>
Future improvements to the platypus sequence assembly will be
dependent on the availability of funding and improvements to existing
assembler software. Funding for the sequencing of the platypus genome
was provided by the National Human Genome Research Institute (NHGRI),
National Institutes of Health (NIH).
</P>

<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/ornAna1/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#platypus">Downloads</A> page. 

These data have
<A HREF="../goldenPath/credits.html#platypus_use">specific 
conditions for use</A>. 

The Platypus browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#platypus_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
