<H2>Description</H2>
<P>
The miRNA track shows microRNAs from the
<A HREF="http://microrna.sanger.ac.uk/" TARGET=_blank>
miRBase</A> at <A HREF="http://www.sanger.ac.uk/" TARGET=_blank>The 
Wellcome Trust Sanger Institute</A>. </P>

<H2>Display Conventions and Configuration</H2>
<P>
The precursor forms of
microRNAs (mirs) are represented by thin blocks. miRNAs in the sense
orientation are shown in black; those in the reverse orientation are colored
grey.</P>

<P>To display only those items
that exceed a specific unnormalized score, enter a minimum score between 0 and
1000 in the text box at the top of the track description page. </P>

<H2>Methods</H2>
<P>
Precursor miRNAs from 
<a href="http://microrna.sanger.ac.uk/" target="_blank">
miRBase</a>
were aligned against the genome
using blat. The extents of the precursor sequences were not generally known,
and were predicted based on base-paired hairpin structure. miRBase is described
in Griffiths-Jones, S. et al. (2006). The miRNA Registry is described in
Griffiths-Jones, S. (2004) and Weber, M.J. (2005) in the References section
below.</p>

<H2>Credits</H2>
<P>

This track was created by Michel Weber of <a
href="http://www-lbme.biotoul.fr/uk/index.html" target="_blank">
Laboratoire de Biologie Moléculaire Eucaryote,
CNRS Université Paul Sabatier</a>
(Toulouse, France), Yves Quentin of <a
href="http://www-lmgm.biotoul.fr/uk/index.html" target="_blank">Laboratoire de Microbiologie et Génétique
Moléculaires</a> (Toulouse, France) and Sam Griffiths-Jones of
<a href="http://www.sanger.ac.uk" target="_blank">
The Wellcome Trust Sanger Institute</a>
(Cambridge, UK).
</P>
<H2>References</H2>
<P>
When making use of these data, please cite:</P>
<P>
Griffiths-Jones S, Grocock RJ, van Dongen S, Bateman A, Enright AJ.
<A HREF="http://nar.oxfordjournals.org/cgi/content/full/34/suppl_1/D140"
TARGET=_BLANK>miRBase: microRNA sequences, targets and gene nomenclature</A>.
<em>Nucl. Acids Res.</em> 2006 Jan 1;34(Database Issue):D140-4.</P>
<P>
Griffiths-Jones S. 
<A HREF="http://nar.oupjournals.org/cgi/content/full/32/suppl_1/D109"
TARGET=_blank>The microRNA Registry</A>.
<em>Nucl. Acids Res.</em> 2004 Jan 1;32(Database issue):D109-11.</P>
<P>
Weber MJ. 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Citation&list_uids=15634332"
TARGET=_blank>New human and mouse microRNA genes found by homology search</A>.
<em>Febs J.</em> 2005 Jan;272(1):59-73.</P>
<P>
You may also want to cite The Wellcome Trust Sanger Institute 
<A HREF="http://www.sanger.ac.uk/Software/Rfam/mirna/"
TARGET=_blank>miRNA Registry</A>.</P>
<P>
The following publication provides guidelines on miRNA annotation:
Ambros V, Bartel B, Bartel DP, Burge CB, Carrington JC, Chen X, 
Dreyfuss G, Eddy SR, Griffiths-Jones S, Marchall M, <em>et al</em>. 
<A HREF="http://www.rnajournal.org/cgi/content/full/9/3/277" 
TARGET=_blank>A uniform system for microRNA annotation</A>.  
<em>RNA.</em> 2003 Mar;9(3):277-9.</P>
<P>
For more information on blat, see 
Kent WJ.
<A HREF="http://www.genome.org/cgi/content/abstract/12/4/656" 
TARGET=_blank>BLAT - the BLAST-like alignment tool</A>.
<em>Genome Res.</em> 2002 Apr;12(4):656-64.</P>

