<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=230 HEIGHT=285>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://genome.jgi.doe.gov/Xentr4/Xentr4.home.html"
    TARGET=_blank>
    <IMG SRC="/images/Xenopus_tropicalis.jpg" WIDTH=207 HEIGHT=260 ALT="Frog"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Xenopus (Silurana) tropicalis</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of UNC Chapel Hill,
      (<A HREF="http://www.unc.edu/"
      TARGET=_blank>Chris Showell,</A> all rights reserved.)
      <P>
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Nov. 2009 <em>Xenopus (Silurana) tropicalis</em> (western clawed frog)
draft assembly (GCA_000004195.1) produced by the US DOE Joint Genome
Institute (JGI-PGF).  See also:
<A HREF="http://www.ncbi.nlm.nih.gov/Traces/wgs/?val=AAMC01"
TARGET=_blank>WGS record at NCBI.</A>

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a sequenced 
genomic clone, an mRNA or EST,
a chromosomal coordinate range, or keywords from 
the GenBank description of an mRNA. The following list provides examples 
of various types of position queries for the frog genome. See the 
<A HREF="../goldenPath/help/hgTracksHelp.html" 
TARGET=_blank>User's Guide</A> for more information. 
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><BR></TD>
    </TR>
	
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>scaffold_3</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays all of scaffold 3</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>scaffold_3:1-2000000</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays first 2 million bases of scaffold_3</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>scaffold_3:100000+2000</TD>
	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of scaffold_3 that spans 2000 bases, starting with position 100000 </TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>CR848332</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays location of mRNA with GenBank accession CR848332</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>AL680330</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays location of EST with GenBank accession AL680330</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>scaffold_1001.7</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays location of Genscan gene prediction with identifier scaffold_1001.7</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>

    <TR><TD VALIGN=Top NOWRAP>dazl</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists dazl mRNAs</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists zinc finger mRNAs</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>asashima</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientists named Asashima</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Papalopulu,N.</TD>
	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by co-author N. Papalopulu</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3">Use this last format for author queries. 
			Although GenBank requires the search format 
			<em>Papalopulu N</em>, internally it uses 
			the format <em>Papalopulu,N.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly Details</H3>
<P>
Assembly version 4.2--which was constructed from WGS reads using the JGI 
assembler, Jazz--contains 19,550 scaffolds with an average coverage of 7.65X. 
Roughly half the genome is contained in 270 scaffolds, each of at least 1.56  
Mb in length.  For more information about this assembly, see the JGI
<A HREF="http://www.jgi.doe.gov/xenopus" 
TARGET=_blank><em>X. tropicalis</em> website</A>.</P>

<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/xenTro3/">FTP server</A>
or the
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#xenTro">Downloads</A>
page.</P>

<P>
Many thanks to the JGI and the other institutions who contributed to
the sequencing and mapping effort for this release.
The X. tropicalis (western clawed frog) browser annotation tracks were
generated by UCSC and collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#xenTro_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release.</P>

<P>
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#xenTro">downloads</A> 
page. These data have 
<A HREF="http://genome.jgi.doe.gov/Xentr4/Xentr4.download.html" 
TARGET=_blank>specific conditions for use</A>.</P>
