<H3>Description</H3>
<P>
This column shows the gene expression data from the Mouse Gene Expression
Atlas 3 from the Genomics Institute of the Novartis Research Foundation (GNF).
<P>
By default, the column shows the ratio of expression of a gene in a specific 
set of tissues to the expression of the gene overall. A gene that is more highly 
expressed in a tissue is colored red, and a less expressed gene is shown in 
green. The values are colored on a logarithmic scale. This coloring is standard, 
but is the opposite of what an inexperienced user 
might expect: in this case, red means go and green means stop!  
Black indicates that a gene is neither over- nor under-expressed in the tissue. 
Uncolored boxes (white on most browsers) represent missing data.  

<H3>Column Configuration</H3>
Several column configuration options are available on the Configuration page, 
which can be accessed via the <em>configure</em> button in the top menu.
<UL>
<LI>
<B>Absolute expression values:</B> The display can be configured to show 
absolute expression values rather than ratio values. To toggle the display, 
select the <em>absolute</em> option from the <em>values</em> pulldown menu. 
Absolute expression values
are shown in shades of gray, with higher expression levels shown in lighter 
shades. As with ratio values, absolute values are colored on a logarithmic 
scale.
<LI>
<B>Color scheme:</B> Color-blind users can switch the coloring scheme from red/green to 
yellow/blue via the <em>Expression ratio colors</em> pulldown menu at the top
of the Configuration page.
<LI>
<B>Level of tissue detail:</B> The <em>tissue</em> pulldown menu
controls the amount and type of tissue data displayed. By default, only the 
median values for selected tissues are shown.  To display the value of all 
individual experimental replicates of all tissues, 
select the <em>all replicates</em> option. To view a single median value for each 
tissue, choose the <em>median of replicates</em> option.
<LI>
<B>Brightness:</B> To increase or decrease the brightness of the expression colors,
edit the number in the column's <em>brightness</em> text box. Values greater 
than 1.0 increase the brightness, while those less than 1.0 dim the color. 
</UL>

<H3>Methods</H3>
<P>
All cell lines and tissues were sourced from 8-10 week old male C57Bl/6 mice,
with the exception of female-specific organs, which were sourced from female
mice.<BR>
Lattin JE, Schroder K, Su AI, Walker JR, Zhang J, Wiltshire T, Saijo K,
Glass CK, Hume DA, Kellie S, Sweet MJ.
<A HREF="http://www.ncbi.nlm.nih.gov/sites/entrez?Db=Pubmed&term=18442421[UID]"
TARGET=_blank>Expression analysis of G Protein-Coupled Receptors in mouse
macrophages</A>. <em>Immunome Research</em>. 2008 Apr 29;4(1):5

<P>When calculating expression ratios, the overall expression level in the
denominator was calculated by first taking the median of replicates for
each tissue,  and then taking the median of these medians.
</P>
<P>To associate particular genes with expression data we use a fairly simple-minded
scheme.  First we align the cDNA sequences around which the probes were designed to
the genome.  Then for each gene we pick the cDNA sequence with the alignment that most
overlaps the gene as the source for expression data for that gene.  Since many of the cDNA 
sequences and probes designed for the cDNA sequences just cover part of the gene, in most
cases different splice varients of a gene will end up mapped to the same cDNA sequence. 
</P>

<H3>Credits</H3>
The data for this column was kindly provided by 
<A HREF="http://web.gnf.org/" TARGET=_blank>GNF</A>
as part of the <A HREF="http://expression.gnf.org/" TARGET=_blank>GNF Gene Expression Atlas</A>.

