<H2>Description</H2> 
<P>
This track shows a measure of evolutionary conservation in seven species of
the genus <em>Saccharomyces</em> based on a phylogenetic hidden Markov model
(phastCons).  The graphic display shows the alignment projected onto
<em>S. cerevisiae</em>. 
<P>
The genomes were downloaded from:<BR>
<UL>
<LI><em>S. cerevisiae</em> - <A HREF="ftp://genome-ftp.stanford.edu/pub/yeast/data_download/sequence/genomic_sequence" TARGET=_blank>ftp://genome-ftp.stanford.edu/pub/yeast/data_download/sequence/genomic_sequence</A>
<LI><em>S. paradoxus</em> - <A HREF="http://www.broad.mit.edu/ftp/pub/annotation/fungi/comp_yeasts/S1a.Assembly/Spar_contigs.fasta" TARGET=_blank>http://www.broad.mit.edu/ftp/pub/annotation/fungi/comp_yeasts/S1a.Assembly/Spar_contigs.fasta</A>
<LI><em>S. mikatae</em> - <A HREF="http://www-genome.wi.mit.edu/ftp/pub/annotation/fungi/comp_yeasts/S1a.Assembly/Smik_contigs.fasta" TARGET=_blank>http://www-genome.wi.mit.edu/ftp/pub/annotation/fungi/comp_yeasts/S1a.Assembly/Smik_contigs.fasta</A>
<LI><em>S. kudriavzevii</em> - <A HREF="http://www.genetics.wustl.edu/saccharomycesgenomes/Contigs/YM6553.fsa.gz" TARGET=_blank>http://www.genetics.wustl.edu/saccharomycesgenomes/Contigs/YM6553.fsa.gz</A>
<LI><em>S. bayanus</em> - <A HREF="http://www-genome.wi.mit.edu/ftp/pub/annotation/fungi/comp_yeasts/S1a.Assembly/Sbay_contigs.fasta" TARGET=_blank>http://www-genome.wi.mit.edu/ftp/pub/annotation/fungi/comp_yeasts/S1a.Assembly/Sbay_contigs.fasta</A>
<LI><em>S. castelli</em> - <A HREF="http://www.genetics.wustl.edu/saccharomycesgenomes/Contigs/YM476.fsa.gz" TARGET=_blank>http://www.genetics.wustl.edu/saccharomycesgenomes/Contigs/YM476.fsa.gz</A>
<LI><em>S. kluyveri</em> - <A HREF="http://www.genetics.wustl.edu/saccharomycesgenomes/Contigs/YM479.fsa.gz" TARGET=_blank>http://www.genetics.wustl.edu/saccharomycesgenomes/Contigs/YM479.fsa.gz</A>
</UL>
</P>
<P>
In full display mode, this track shows the overall conservation score across all
species as well as pairwise alignments 
of each species with <em>$Organism</em>.  The pairwise alignments are
shown in dense display mode using a grayscale 
density gradient.  The checkboxes in the track configuration section allow
the exclusion of species from the pairwise display; however, this does not
remove them from the conservation score display.  </P>
<P>
When zoomed-in to the base-display level, the track shows the base 
composition of each alignment. The numbers and symbols on the Gaps
line indicate the lengths of gaps in the <em>$Organism</em> sequence at those 
alignment positions relative to the longest non-<em>$Organism</em> sequence. 
If there is sufficient space in the display, the size of the gap is shown; 
if not, and if the gap size is a multiple of 3, a &quot;*&quot; is displayed, 
otherwise &quot;+&quot; is shown.  
To view detailed information about the alignments at a specific position,
zoom in the display to 30,000 or fewer bases, then click on the alignment.</P>
<P>
This track may be configured in a variety of ways to highlight different aspects 
of the displayed information. Click the 
<A HREF="../goldenPath/help/hgWiggleTrackHelp.html" TARGET=_blank>Graph 
configuration help</A> link for an explanation of the configuration options.</P>

<H2>Methods</H2>
<P> 
Best-in-genome pairwise alignments were generated for each species 
using blastz, followed by chaining and netting.  The pairwise alignments
were then multiply aligned using multiz, and
the resulting multiple alignments were assigned 
conservation scores by phastCons.</P>
<P>
The phastCons program computes conservation scores based on a phylo-HMM, a
type of probabilistic model that describes both the process of DNA
substitution at each site in a genome and the way this process changes from
one site to the next (Felsenstein and Churchill 1996, Yang 1995, Siepel and
Haussler 2005).  PhastCons uses a two-state phylo-HMM, with a state for
conserved regions and a state for non-conserved regions.  The value plotted
at each site is the posterior probability that the corresponding alignment
column was "generated" by the conserved state of the phylo-HMM.  These
scores reflect the phylogeny (including branch lengths) of the species in
question, a continuous-time Markov model of the nucleotide substitution
process, and a tendency for conservation levels to be autocorrelated along
the genome (i.e., to be similar at adjacent sites).  The general reversible
(REV) substitution model was used.  Note that, unlike many
conservation-scoring programs, phastCons does not rely on a sliding window
of fixed size, so short highly-conserved regions and long moderately
conserved regions can both obtain high scores.  More information about
phastCons can be found in Siepel et al. (2005).</P> 
<P> 
PhastCons currently treats alignment gaps as missing data, which
sometimes has the effect of producing undesirably high conservation scores
in gappy regions of the alignment.  We are looking at several possible ways
of improving the handling of alignment gaps.</P>

<H2>Credits</H2>
<P>
This track was created at UCSC using the following programs:
<UL>
<LI>
Blastz and multiz by Minmei Hou, Scott Schwartz and Webb Miller of the 
<A HREF="http://bio.cse.psu.edu/" TARGET=_blank>Penn State Bioinformatics 
Group</A>. 
<LI>
AxtBest, axtChain, chainNet, netSyntenic, and netClass 
by Jim Kent at UCSC. 
<LI> PhastCons by Adam Siepel at Cornell University. 
<LI>"Wiggle track" plotting software by Hiram Clawson at UCSC.
</UL>
</P>

<H2>References</H2>

<H3>Phylo-HMMs and phastCons:</H3>
<P> 
Felsenstein J, Churchill GA.
<A HREF="http://mbe.oxfordjournals.org/cgi/content/abstract/13/1/93"
TARGET=_blank>A Hidden Markov Model approach to
variation among sites in rate of evolution</A>.
<EM>Mol Biol Evol</EM>. 1996 Jan;13(1):93-104.  </P>
<P> 
Siepel A, Haussler D.
<A HREF="http://compgen.bscb.cornell.edu/~acs/phylohmm.pdf"
TARGET=_blank>Phylogenetic Hidden Markov Models</A>.
In R. Nielsen, ed., <em>Statistical Methods in Molecular Evolution</em>,
pp. 325-351, Springer, New York (2005). </P>
<P>
Siepel A, Bejerano G, Pedersen JS, Hinrichs AS, Hou M, Rosenbloom K,
Clawson H, Spieth J, Hillier LW, Richards S, <EM>et al.</EM>
<A HREF="http://www.genome.org/cgi/doi/10.1101/gr.3715005"
TARGET=_blank>Evolutionarily conserved elements in vertebrate, insect, worm,
and yeast genomes</A>.
<EM>Genome Res</EM>. 2005 Aug;15(8):1034-50.  </P>
<P>
Yang Z.
<A HREF="http://www.genetics.org/cgi/content/abstract/139/2/993"
TARGET=_blank>A space-time process model for the evolution of DNA
sequences</A>.
<EM>Genetics</EM>. 1995 Feb;139(2):993-1005.  </P>

<H3>Chain/Net:</H3>
<P>
Kent WJ, Baertsch R, Hinrichs A, Miller W, Haussler D.
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1"
TARGET=_blank>Evolution's cauldron:
duplication, deletion, and rearrangement in the mouse and human genomes</A>.
<EM>Proc Natl Acad Sci U S A</EM>. 2003 Sep 30;100(20):11484-9.  </P>

<H3>Multiz:</H3>
<P>
Blanchette M, Kent WJ, Riemer C, Elnitski L, Smit AF, Roskin KM,
Baertsch R, Rosenbloom K, Clawson H, Green ED, <EM>et al.</EM>
<A HREF="http://www.genome.org/cgi/content/abstract/14/4/708"
TARGET=_blank>Aligning multiple genomic sequences with the threaded blockset aligner</A>.
<EM>Genome Res</EM>. 2004 Apr;14(4):708-15.  </P>

<H3>Blastz:</H3>
<P>
Chiaromonte F, Yap VB, Miller W.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract"
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>.
<EM>Pac Symp Biocomput</EM>. 2002;:115-26.  </P>
