<H2>Description and Methods </H2>
<P>
Excerpted from the <A HREF="http://vega.sanger.ac.uk/" TARGET=_BLANK>Vega</A> 
main page:</P>
<P>
"The 
<A HREF="http://vega.sanger.ac.uk/" TARGET=_BLANK>
Vertebrate Genome Annotation (VEGA)</A> 
 is designed to be a central repository for high quality, frequently updated  
manual annotation of different vertebrate finished genome sequence.
In collaboration with the genome sequencing centres Vega attempts to
present consistent high-quality curation of the published chromosome
sequences."</P>
<P>
"Finished genomic sequence is analysed on a clone by clone basis using
a combination of similarity searches against DNA and protein databases
as well as a series of <em>ab initio</em> gene predictions (GENSCAN, Fgenes)."</P>
<P>
"In addition, comparative analysis using vertebrate datasets such as
the Riken mouse cDNAs and Genoscope <em>Tetraodon nigroviridis</em> Ecores
(Evolutionary Conserved Regions) are used for novel gene discovery. The 
annotation is based on supporting evidence only"</P>
<P>
NOTE: VEGA annotations are shown only on chromosomes 1-25 in this assembly.

For zebrafish, the <AHREF "http://vega.sanger.ac.uk/Danio_rerio/index.html" 
_TARGET=blank>VEGA</A> annotations consist of BAC and PAC clones that were 
sequenced for the zebrafish clone mapping and sequencing project. It is based
on the tiling path information from the fpc database.

<H2> Credits </H2>
<P>
Thanks to Kerstin Howe at the
<A HREF="http://www.sanger.ac.uk/" TARGET=_BLANK>
Wellcome Trust Sanger Institute</A> 
for providing the GFF files for the Vega annotations. Vega gene annotations are 
generated by manual annotation from the Zebrafish sequence analysis group and 
the <A HREF="http://www.sanger.ac.uk/HGP/havana/" TARGET=_blank>HAVANA</A> 
group at the Sanger Institute and <A HREF="http://www.zfin.org/" 
TARGET=_BLANK>ZFIN</A>.
<P>
