<p class="p1" style="text-align:center"><b>PhyloCSF Track Hub</b><br><br></p>
<p class="p2"><b>Description</b></p>
<p class="p3"><br></p>
<p class="p4">These tracks show evolutionary protein-coding potential as determined by PhyloCSF [1] to help identify conserved, functional, protein-coding regions of genomes. PhyloCSF examines evolutionary signatures characteristic of alignments of conserved coding regions, such as the high frequencies of synonymous codon substitutions and conservative amino acid substitutions, and the low frequencies of other missense and nonsense substitutions (CSF = Codon Substitution Frequencies). PhyloCSF provides more information than conservation of the amino acid sequence, because it distinguishes the different codons that code for the same amino acid. One of PhyloCSF's main current applications is to help distinguish protein-coding and non-coding RNAs represented among novel transcript models obtained from high-throughput transcriptome sequencing. More information on PhyloCSF can be found on the <a href="https://github.com/mlin/PhyloCSF/wiki"><span class="s1">PhyloCSF wiki</span></a>.</p>
<p class="p3"><br></p>
<p class="p4">The <b>Raw PhyloCSF</b> tracks show the PhyloCSF score for each codon in each of 6 frames. Regions in which most codons have score greater than 0 are likely to be protein-coding in that frame. No score is shown when the relative branch length is less than 0.1 (see <b>PhyloCSF Power</b>).</p>
<p class="p3"><br></p>
<p class="p4">The <b>Smoothed PhyloCSF</b> tracks show the scores smoothed using an HMM. No score is shown when the relative branch length is less than 0.1 (see <b>PhyloCSF Power</b>).</p>
<p class="p3"><br></p>
<p class="p4">The <b>PhyloCSF Regions</b> tracks show the regions that are protein-coding in the most-likely-path through the HMM. They are something like predicted exons, except that splice sites, start codons, and stop codons are not considered so the boundaries are approximate. The gray scale is an indication of the maximum log-odds that any codon in the region is coding according to the HMM.</p>
<p class="p3"><br></p>
<p class="p4">The <b>PhyloCSF Power</b> track shows the branch length score at each codon, i.e., the ratio of the branch length of the species present in the local alignment to the total branch length of all species in the full genome alignment. It is an indication of the statistical power available to PhyloCSF. Codons with branch length score less than 0.1 have been excluded altogether (from all tracks) because PhyloCSF does not have sufficient power to get a meaningful score at these codons. Codons with branch length score greater than 0.1 but much less than 1 should be considered less certain.</p>
<p class="p3"><br></p>
<p class="p4">The <b>Splice Prediction</b> tracks show canonical splice predictions for each strand. This can be useful for predicting where novel exons start and end. The bars are on the first or last base of the hypothetical intron, i.e., the G of GT or AG. Green bars indicate splice donors and red bars indicate acceptors, so an intron would extend from a green bar to a red bar. Taller bars are more likely to be true splice sites.</p>
<p class="p3"><br></p>
<p class="p4">The <b>PhyloCSF Novel</b> tracks (a.k.a., <b>PhyloCSF Candidate Coding Region</b> or <b>PCCR</b> tracks) show regions that could be protein-coding but are not currently annotated as protein-coding exons or pseudogenes in a specified gene set. Green regions are on the plus strand and red regions are on the minus strand. These regions have been ranked using an SVM with ones most likely to be real novel coding or pseudogene regions having low rank (darker shading), and ones that are more likely to be non-coding false positives having high rank (lighter shading). The rank is listed next to the region.</p>
<p class="p3"><br></p><p class="p4"> Select on a PCCR to bring up its details page. This includes useful information for deciding if it is protein-coding including a link to <a href="https://data.broadinstitute.org/compbio1/cav.php"><span class="s1">CodAlignView</span></a> [2], which shows the alignment of that region, color-coded to indicate protein-coding signatures.<p>
<p class="p3"><br></p>
<p class="p2"><b>Caveats</b></p>
<ul class="ul1">
  <li class="li4">Around 10% of annotated protein-coding regions get score less than 0. This can happen for various reasons. For example, the region could be coding in the reference species but not in other species, or the alignment does not represent a true orthology relationship between the species.</li>
  <li class="li4">Protein-coding regions will often have positive score on the reverse strand in the frame in which the third codon positions match up (the "antisense" frame), though the score will usually be higher on the correct strand.</li>
  <li class="li4">Pseudogenes will often get positive scores even though they are not protein-coding. One reason for this is that since PhyloCSF is measuring coding potential on the whole species tree, if a unitary pseudogene was coding in the common ancestor but is no longer protein-coding in the reference species it will often still have positive score. Another reason is that the aligner might align a duplicated pseudogene to the orthologs of its parent. The CACOFONy score in the details page can help identify this situation; CACOFONy, short for Coding Alignment COllision FractiON, is the fraction of the alignment of the region that is identical to the alignment of some annotated coding region.
</li>
</ul>
<p class="p3"><br></p>
<p class="p2"><b>Methods</b></p>
<p class="p3"><br></p>
<p class="p4">PhyloCSF was run with the "fixed" strategy on every codon in every frame on each strand in the assembly.</p>
<p class="p3"><br></p>
<p class="p4">The alignments and PhyloCSF parameters for the various assemblies are as follows:</p>
<table cellspacing="0" cellpadding="0" class="t1">
  <tbody>
    <tr>
      <td valign="middle" class="td1">
        <p class="p4"><span class="s2">Assembly</span></p>
      </td>
      <td valign="middle" class="td2">
        <p class="p4"><span class="s2">PhyloCSF Parameters</span></p>
      </td>
      <td valign="middle" class="td3">
        <p class="p4"><span class="s2">Alignments</span></p>
      </td>
    </tr>
    <tr>
      <td valign="middle" class="td1">
        <p class="p4">hg19</p>
      </td>
      <td valign="middle" class="td2">
        <p class="p4">29mammals</p>
      </td>
      <td valign="middle" class="td3">
        <p class="p4">29-mammals subset of the 46-vertebrates hg19 alignment</p>
      </td>
    </tr>
    <tr>
      <td valign="middle" class="td1">
        <p class="p4">hg38</p>
      </td>
      <td valign="middle" class="td2">
        <p class="p4">58mammals</p>
      </td>
      <td valign="middle" class="td3">
        <p class="p4">58-mammals subset of the 100-vertebrates hg38 alignment</p>
      </td>
    </tr>
    <tr>
      <td valign="middle" class="td1">
        <p class="p4">mm10</p>
      </td>
      <td valign="middle" class="td2">
        <p class="p4">29mammals</p>
      </td>
      <td valign="middle" class="td3">
        <p class="p4">29-mammals subset of the 60-vertebrates mm10 alignment</p>
      </td>
    </tr>
    <tr>
      <td valign="middle" class="td1">
        <p class="p4">dm6</p>
      </td>
      <td valign="middle" class="td2">
        <p class="p4">23flies</p>
      </td>
      <td valign="middle" class="td3">
        <p class="p4">23-<I>Drosophila</I> subset of the 27-insect dm6 alignment</p>
      </td>
    </tr>
    <tr>
      <td valign="middle" class="td1">
        <p class="p4">ce11</p>
      </td>
      <td valign="middle" class="td2">
        <p class="p4">8worms</p>
      </td>
      <td valign="middle" class="td3">
        <p class="p4">8-<I>Caenorhabditis</I> subset of the 26-worm ce11 alignment</p>
      </td>
    </tr>
    <tr>
      <td valign="middle" class="td1">
        <p class="p4">galGal4</p>
      </td>
      <td valign="middle" class="td2">
        <p class="p4">49birds</p>
      </td>
      <td valign="middle" class="td3">
        <p class="p4">49 sauropsids alignment (unpublished)</p>
      </td>
    </tr>
    <tr>
      <td valign="middle" class="td1">
        <p class="p4">galGal6</p>
      </td>
      <td valign="middle" class="td2">
        <p class="p4">53birds</p>
      </td>
      <td valign="middle" class="td3">
        <p class="p4">53-bird subset of the 77-vertebrates galGal6 alignment</p>
      </td>
    </tr>
  </tbody>
</table>
<p class="p3"><br></p>
<p class="p4">The scores were smoothed using a Hidden Markov Model (HMM) with 4 states, one representing coding regions and three representing non-coding regions. The emission of each codon is its PhyloCSF score. The ratio of the emissions probabilities for the coding and non-coding models are computed from the PhyloCSF score, since it represents the log-likelihood ratio of the alignment under the coding and non-coding models. The three non-coding states have the same emissions probabilities but different transition probabilities (they can only transition to coding) to better capture the multimodal distribution of gaps between same-frame coding exons. These transition probabilities represent the best approximation of this gap distribution as a mixture model of three exponential distributions, computed using Expectation Maximization.</p>
<p class="p3"><br></p>
<p class="p4">The HMM defines a probability that each codon is coding, based on the PhyloCSF scores of that codon and nearby codons on the same strand in the same frame, without taking into account start codons, stop codons, or potential splice sites. PhyloCSF+0 shows the log-odds that codons in frame 0 on the '+' strand are in the coding state according to the HMM, and similarly for strand '-' and frames 1 and 2. The PhyloCSF Regions tracks show the genomic intervals that are in the protein-coding state in the most likely path through the HMM.</p>
<p class="p3"><br></p>
<p class="p4">Splice sites were predicted using the maximum entropy method [3], trained on <I>D.</I> <I>melanogaster</I> splice sites for dm6, on <I>C.</I> <I>elegans</I> splice sites for ce11, and on human splice sites for the other assemblies.</p>
<p class="p3"><br></p>
<p class="p4">The PhyloCSF Novel (PCCR) tracks were created as follows. All regions from the PhyloCSF Regions tracks were compared to protein-coding and pseudogene annotations from the specified gene set. Regions contained in annotated pseudogene regions, or in protein-coding regions in the same frame or the antisense frame, were eliminated. If part of a region was contained in the annotated region, the region was trimmed to the unannotated portion. Regions less than nine codons long were eliminated. PhyloCSF scores for each region were recomputed using the "mle" strategy except for hg19, galGal4, mm10 version 5, and GENCODE hg38 versions 25 and earlier. Regions that were more likely to be antisense to a novel region than to be novel themselves were distinguished using a three-feature SVM and excluded, the features being the PhyloCSF score, the difference between the PhyloCSF scores on the two strands, and the length of the region. Ranks were assigned using a four-feature SVM, trained to distinguish regions that are contained in coding annotations from ones that are not, the features being the three features mentioned previously and the relative branch length of the species in the local alignment of the region.</p>
<p class="p3"><br></p>
<p class="p2"><b>Credits</b></p>
<p class="p3"><br></p>
<p class="p4">Questions should be directed to <a href="mailto:iljungr@csail.mit.edu"><span class="s1">Irwin Jungreis</span></a>.</p>
<p class="p3"><br></p>
<p class="p2"><b>Citing the PhyloCSF Tracks</b></p>
<p class="p3"><br></p>
<p class="p4">If you use the PhyloCSF browser tracks, please cite Mudge et al. 2019 [4].</p>
<p class="p3"><br></p>
<p class="p2"><b>References</b></p>
<p class="p3"><br></p>
<p class="p5"><span class="s3">[1] </span>Lin MF, Jungreis I, and Kellis M (2011). PhyloCSF: a comparative genomics method to distinguish protein-coding and non-coding regions. <a href="http://bioinformatics.oxfordjournals.org/content/27/13/i275.abstract"><span class="s4"><i>Bioinformatics</i></span><span class="s1"> 27:i275-i282</span></a> (ISMB/ECCB 2011).</p>
<p class="p5"><span class="s3">[2] </span>Jungreis I, Lin MF, Chan CS, Kellis M (2016). CodAlignView: The Codon Alignment Viewer [Internet]. Available from: <a href="http://data.broadinstitute.org/compbio1/cav.php"><span class="s4">http://data.broadinstitute.org/compbio1/cav.php</span></a>.</p>
<p class="p5"><span class="s3">[3] </span>Yeo G, Burge CB (2004). Maximum entropy modeling of short sequence motifs with applications to RNA splicing signals. <a href="http://online.liebertpub.com/doi/abs/10.1089/1066527041410418"><span class="s4"><i>Journal of Computational Biology : a Journal of Computational Molecular Cell Biology</i></span><span class="s1"> 11(2-3), 377â€“394</span></a>.</p>
<p class="p5"><span class="s3">[4] </span>Mudge JM, Jungreis I, Hunt T, Gonzalez JM, Wright J, Kay M, Davidson C, Fitzgerald S, Seal R, Tweedie S, He L, Waterhouse RM, Li Y, Bruford E, Choudhary J, Frankish A, Kellis M (2019). Discovery of high-confidence human protein-coding genes and exons by whole-genome PhyloCSF helps elucidate 118 GWAS loci. <a href="https://genome.cshlp.org/content/early/2019/09/19/gr.246462.118.abstract"><em>Genome Research</em> gr-246462. doi: 10.1101/gr.246462.118</a>.</p>
