<H2>Description</H2>
<p>
This track shows predicted and experimental representations of the
SARS-CoV-2 transcriptome based on long-read Nanopore sequencing.<p>

<h2>Display Conventions and Configuration</h2>
<p>
SARS-CoV-2 generates sub-genomic mRNAs (sgmRNAs) for all ORFs. The virus
achieves this by recombination mechanisms in which replication machinery
jumps from one of many TRS-B site (transcription regulatory sequence, body) to
the TRS-L (leader sequence) during negative strand synthesis.
These negative strands are then used as templates for mRNA synthesis.</p>

<p>
On these tracks we depict the predicted mRNAs with the excised sequence
drawn like introns. The ORFs predicted to be translated by these mRNAs are
shown in thick boxes.  The thin bars function as UTRs for that particular mRNA
species.<p>

<p>Multiple subtracks are available:</p>
<ul>
<li><b>TRS sites</b>: Annotated core TRS sequence (ACGAAC) in the viral genome
that allows recombination. One site TRS* differs from the canonical TRS site by
1 nt, but has experimental support and is required to generate a 7b mRNA.</li>
<li><b>SARS-CoV-2 Transcripts</b>: Canonical SARS-CoV-2 Transcripts (gRNA and
mRNA). Generated by recombining all TRS-B sites in the above track with the
leader.  Note the actual recombination breakpoints can often be drawn in
multiple ways (since the TRS core motif is identical), and direct RNA
sequencing suggests slightly different breakpoints, depending on the mRNA.  
See the experimental tracks if your analysis requires a detailed understanding of 
the breaks. The reported breaks in this track are only meant to be approximate.</li>
<li><b>Kim Recombined transcripts</b>: All RNA recombination products reported by 
Nanopore direct RNA sequencing in Kim et al, 2020, Cell.</li>
<li><b>Kim Recomb. TRS transcripts</b>: RNA recombination products that involve the 
leader on the 5' end; validated by nanopore sequencing. The score is the number of 
reads divided by 100. RNAs with read support &gt;100,000 are rounded to a score of 1000.</li>
<li><b>Kim Recomb. Novel transcripts</b>: RNA recombination products that do <b>not</b> 
involve the leader, discovered by nanopore sequencing. RNAs with less than 100 reads 
were discarded. The score is the number of reads. RNAs with read 
support &gt;1000 are listed as 1000.</li>
</ul>

<h2>Methods</h2>
<ul>
<li><b>TRS sites</b>: The core TRS sequence (ACGAAC) was identified using the findMotif tool. 
The TRS* (AaGAAC) site identified in Kim et al was manually added to create overall 
agreement with their Figure 1.</li>
<li><b>SARS CoV-2 Transcripts</b>: Using the TRS track, we generated mRNAs which span from 
nt 1-75 (the end of the TRS-L core sequence) and resume at the 3' end of all TRS-B 
sequences. Neither the 5' and 3' terminal ends of these RNAs, nor their internal 
breakpoints, should be considered exact.  CDS sequences match Figure 1 from Kim et al.</li>
<li><b>Kim Recombined transcripts</b>: was generated from Table S2 of Kim et al, 2020. This 
represents all RNA species identified by direct RNA sequencing. Scores are the number 
of reads divided by 100 (RNAs with &gt;1000 reads are rounded to 1000).</li>
<li><b>Kim Recomb. TRS transcripts</b>: generated from Table S3 of Kim et al, 2020. These 
viral transcripts all contain the TRS-L recombined with 3' viral sequence.</li>
<li><b>Kim Recomb. Novel transcripts</b>: this track was generated from Table S4. These 
viral transcripts contain novel recombination events that do <b>not</b> involve the leader.</li>
</ul>

<h2>Data Access</h2>
<p>
The raw data can be explored interactively with the
<a href="hgTables">Table Browser</a>, or combined with other datasets in the
<a href="hgIntegrator">Data Integrator</a> tool.
For automated analysis, the genome annotation is stored in
a bigBed file that can be downloaded from
<a href="http://hgdownload.soe.ucsc.edu/gbdb/wuhCor1/bbi/kim2020">the download server</a>.</p>

<p>
Annotations can
be converted from binary to ASCII text by our command-line tool bigBedToBed.
Instructions for downloading this command can be found on our
<a href="http://hgdownload.soe.ucsc.edu/downloads.html#utilities_downloads">utilities page</a>.
The tool can also be used to obtain features within a given range without downloading the file,
for example:</p>
<code>bigBedToBed http://hgdownload.soe.ucsc.edu/gbdb/wuhCor1/bbi/kim2020/TRS.bb -chrom=NC_045512v2 -start=0 -end=29902 stdout</code>

<p>
Please refer to our
<a href="https://groups.google.com/a/soe.ucsc.edu/forum/#!forum/genome">mailing list archives</a>
for questions, or our
<a href="../../FAQ/FAQdownloads.html#download36">Data Access FAQ</a>
for more information.</p>

<h2>Credits</h2>
<p>
Thanks to Jason Fernandes (Haussler-lab, UCSC) for preparing this track.</p>

<h2>References</h2>
<p>
Kim D, Lee JY, Yang JS, Kim JW, Kim VN, Chang H.
<a href="https://linkinghub.elsevier.com/retrieve/pii/S0092-8674(20)30406-2" target="_blank">
    The Architecture of SARS-CoV-2 Transcriptome</a>.
<em>Cell</em>. 2020 Apr 18;.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32330414" target="_blank">32330414</a>; PMC: <a
    href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7179501/" target="_blank">PMC7179501</a>
</p>

