<h2>Description</h2>

<p>
This track shows protein sequence annotations from the <a
href="https://www.uniprot.org/" target="_blank">UniProt/SwissProt</A> database,
mapped to genomic coordinates. It shows how the protein sequences in this database 
map to the genome.
The data has been curated from scientific publications by the UniProt/SwissProt staff.
</p> 

<p>Note that the data is not online at UniProt itself yet, so the Genome
Browser is the only place where you can see it, at the time of release of this Genome Browser. It will be
released on April 22nd.</p>

<h2>Display Conventions and Configuration</h2>

<p>
Genomic locations of UniProt/SwissProt annotations are labeled with a short name for
the type of annotation (e.g. &quot;glyco&quot;, &quot;disulf bond&quot;, &quot;Signal peptide&quot;
etc.). A click on them shows the full annotation and provides a link to the UniProt/SwissProt
record for more details. 
</p>

<p>
Mouse over a feature to see the full UniProt annotation comment. For variants, the mouse over will
show the full name of the UniProt disease acronym.
</p>

<h2>Methods</h2>

<p>
UniProt sequences were aligned to UCSC/Gencode transcript sequences first with
BLAT, filtered with pslReps (93% query coverage, within top 1% score), lifted
to genome positions with pslMap and filtered again.  UniProt annotations were
obtained from the UniProt XML file.  The annotations were then mapped to the
genome through the alignment using the pslMap program.  This mapping approach
draws heavily on the <A HREF="http://modbase.compbio.ucsf.edu/LS-SNP/"
TARGET="_BLANK">LS-SNP</A> pipeline by Mark Diekhans. For human and mouse, the
alignments were filtered by retaining only proteins annotated with
a given transcript in the Genome Browser table kgXref. Like all Genome Browser
source code, the main script used to build this track can be found on 
<a href="https://github.com/ucscGenomeBrowser/kent/tree/master/src/hg/utils/otto/uniprot/doUniprot">github</a>.
</p>

<h2>Data Access</h2>

<p>
The raw data can be explored interactively with the
<a href="../cgi-bin/hgTables">Table Browser</a>, or the
<a href="../cgi-bin/hgIntegrator">Data Integrator</a>.
For automated analysis, the genome annotation is stored in a bigBed file that 
can be downloaded from the
<a href="http://hgdownload.soe.ucsc.edu/gbdb/$db/uniprot/" target="_blank">download server</a>.
The exact filenames can be found in the 
<a href="https://github.com/ucscGenomeBrowser/kent/blob/master/src/hg/makeDb/trackDb/trackDb.uniprot.ra">track configuration file</a>. 
Annotations can be converted to ASCII text by our tool <tt>bigBedToBed</tt>
which can be compiled from the source code or downloaded as a precompiled
binary for your system. Instructions for downloading source code and binaries can be found
<a href="http://hgdownload.soe.ucsc.edu/downloads.html#utilities_downloads">here</a>.
The tool can also be used to obtain only features within a given range, for example:
<br><br> 
<tt>bigBedToBed http://hgdownload.soe.ucsc.edu/gbdb/wuhCor1/uniprot/unipAliSwissprotCov2.bb -chrom=NC_045512v2 -start=0 -end=29903 stdout</tt> 
<br><br>
This track is updated every month. The MySQL table hgFixed.trackVersion
contains the name of the currently available data on the website. Older
versions of the data files can be downloaded from the <a
href="http://hgdownload.soe.ucsc.edu/goldenPath/$db/archive/">archive
folder</a> of our downloads server.  <br>
Please refer to our
<a href="https://groups.google.com/a/soe.ucsc.edu/forum/#!forum/genome">mailing list archives</a>
for questions, or our
<a href="../FAQ/FAQdownloads.html#download36">Data Access FAQ</a>
for more information. 
</p>

<h2>Credits</h2>
The UCSC SARS-CoV-2 genome browser and associated data annotation tracks were
funded by generous private donors to the UC Santa Cruz Genomics Institute.

This track was created by Maximilian Haeussler at UCSC, with help from Chris
Lee, Mark Diekhans and Brian Raney, feedback from the UniProt staff and Alejo
Mujica, Regeneron Pharmaceuticals.  Thanks to UniProt for making all data
available for download.
</p>

<h2>References</h2>

<p>
UniProt Consortium.
<a href="https://academic.oup.com/nar/article/40/D1/D71/2903687/Reorganizing-the-protein-space-at-
the-Universal" target="_blank">
Reorganizing the protein space at the Universal Protein Resource (UniProt)</a>.
<em>Nucleic Acids Res</em>. 2012 Jan;40(Database issue):D71-5.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22102590" target="_blank">22102590</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3245120/" target="_blank">PMC3245120</a>
</p>

<p>
Yip YL, Scheib H, Diemand AV, Gattiker A, Famiglietti LM, Gasteiger E, Bairoch A.
<a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/humu.20021" target="_blank">
The Swiss-Prot variant page and the ModSNP database: a resource for sequence and structure
information on human protein variants</a>.
<em>Hum Mutat</em>. 2004 May;23(5):464-70.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/15108278" target="_blank">15108278</a>
</p>
