<H2>Description</H2>
<P>
This track shows syntenous (corresponding) regions between $o_organism and 
$organism chromosomes. The $o_date ($o_db) $o_organism assembly was used for 
this annotation.

<H2>Methods</H2>
<P>
We passed a 100k non-overlapping window over the genome and - using the 
<A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>blastz</A>
best-in-$o_organism genome alignments - looked for high-scoring regions with at 
least 40% of the bases aligning with the same region in $organism. 100k segments 
were joined together if they agreed in direction and were within 500 kb of each 
other in the $organism genome and within 4 Mb of each other in the $o_organism. 
Gaps were joined between syntenic anchors if the bases between two flanking 
regions agreed with synteny (direction and $o_organism location). Finally, we 
extended the syntenic block to include those areas.<p>

<H2>Credits</H2>
<P>
Contact <A HREF="mailto:&#98;&#97;&#101;&#114;&#116;s&#99;h&#64;&#99;&#115;&#101;.&#117;&#99;s&#99;.e&#100;&#117;">Robert 
Baertsch</A> at UCSC for more information about this track.
Thanks to the Mouse Genome Sequencing Consortium for providing the $organism 
sequence data. 
