<H2>Description</H2>
This track shows blastz scores from the 
D.pseudoobscura/$organism Alignment Net track, i.e. 
the best D.pseudoobscura/$organism chain for 
every part of the $organism genome.  
The D.pseudoobscura sequence used in this annotation is from the 
<A HREF="http://www.hgsc.bcm.tmc.edu/projects/drosophila/" TARGET=_BLANK>
Baylor HGSC August 2003 Freeze 1 assembly</A>.
<P>
In the graphical display, the shade (in dense display mode) 
or height (in full mode) of a vertical line indicates the score 
of the best chained and netted alignment at that location.  
Clicking on the track brings up detailed information about the 
blastz alignments (taken from the axt-formatted output) in the 
currently displayed range of positions.  
<BR>

<H2>Methods</H2>
First, chains are derived from 
<A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>blastz</A> 
alignments as
described in the details pages of the chain tracks
and sorted so that the highest scoring chain in the
genome is first. The program <em>chainNet</em> then places
chains one at a time, trimming them as necessary to
fit into section that is not already covered by 
a higher scoring chain.  During this process, a
natural hierarchy emerges where chains that fill gaps
in a previous chain are considered underneath the
previous chain.   The program <em>netSyntenic</em> fills in
information about the relationship between upper
and lower level chains, including whether a lower level
chain is syntenic with the higher level chain,  whether
it is inverted with respect to the higher level chain,
and so forth.  The program <em>netClass</em> then fills in 
how much of the gaps and chains are filled with N's
(sequencing gaps) in one or both species,  how much
is filled with transposons inserted before and after
D.pseudoobscura and $organism diverged, and so forth.
The program <em>netToAxt</em> translates the net back 
to axt.  <em>netToAxt</em> is run with -maxGap=300.  

<H2>Credits</H2>
<P>The chainNet, netSyntenic, netClass and netToAxt programs were
developed at the University of California
at Santa Cruz by Jim Kent.</P>

<P>Lineage-specific repeats were identified by Arian Smit and his
program <A HREF="http://repeatmasker.genome.washington.edu/cgi-bin/RepeatMasker"TARGET=_blank>RepeatMasker</A>.</P>

