<H2>Description</H2>
<P>
The track shows the May 2011 <em>Gorilla gorilla gorilla</em> draft assembly
produced by the
<A HREF="http://www.sanger.ac.uk/resources/downloads/gorilla/"
TARGET=_blank>Wellcome Trust Sanger Institute</A>
(<A HREF="http://www.ncbi.nlm.nih.gov/bioproject/31265"
TARGET=_blank>NCBI project 31265</A>, GCA_000151905.1).

<P>
See also: <A HREF="http://www.ncbi.nlm.nih.gov/Traces/wgs/?val=CABD02"
TARGET=_blank>WGS CABD02</A><BR>
Assembly involved several phases, starting with a de novo assembly of the
initial data using the ABySS and Phusion assemblers. Contigs from this
'seed' assembly were then grown by assembling and attaching Solexa read
pairs from the initial data. To improve long-range structure, supercontig
construction was guided by placing (where possible) seed contigs in
accordance with their homologous locations on the human genome, breaking
supercontigs wherever a potential break between human and gorilla was
inferred in an alignment of all the Solexa data to human.
</P><P>
This draft assembly contains all the assembled supercontigs plus a
mitochondrial sequence (chrM - NC_011120).
</P><P>
Two types of sequencing were used in this assembly:
<UL>
<LI> 2.1x WGS capillary sequenced read pairs.
<LI> 35x Solexa sequenced read pairs with insert sizes 150 bp and 450 bp.
</UL>
All sequence was derived from DNA sampled from a single female Western
Lowland gorilla (<em>Gorilla gorilla gorilla</em>), Kamilah. 
</P>

<P>
In dense mode, this track depicts the contigs that make up the currently
viewed chromosome.  Contig boundaries are distinguished by the use of
alternating gold and brown coloration. Where gaps exist between contigs,
spaces are shown between the gold and brown blocks.  Where the relative order
and orientation of the contigs are known, a line is drawn in the graphical
display to bridge the blocks.</P>
<P>
The mitochondrial sequence, chrM, is fragment type &quot;F&quot; (finished).
There are Whole Genome Shotgun contigs indicated as fragment type &quot;W&quot;.
Other components are indicated as fragment type &quot;O&quot;.
</P>
