<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=171>
  <TR><TD ALIGN=RIGHT><A HREF="https://www.hgsc.bcm.edu/content/honey-bee-genome-project"
    TARGET="_blank">
    <IMG SRC="/images/Apis_mellifera.jpg" WIDTH=156 HEIGHT=170 ALT="Bee"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Apis mellifera</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      (NHGRI <A HREF="http://www.genome.gov/dmd/img.cfm?node=Photos/Animals/Insects/Bee%20Apis%20mellifera&id=79117" 
      TARGET=_blank>Press Photos</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The July 2004 <em>Apis mellifera</em> genome assembly 
is based on the Amel_1.2 assembly produced by the 
Human Genome Sequencing Center (HGSC) at 
<A HREF="https://www.hgsc.bcm.edu/content/honey-bee-genome-project" 
TARGET="_blank">Baylor School of Medicine</A>.  To see a list
of the assembly's largest scaffolds ranked by size, click 
<A HREF="/cgi-bin/hgTracks?db=apiMel1&chromInfoPage=">here</A>.

<H3>Sample position queries</H3>
<P>
A genome position can be specified by a scaffold coordinate range,
the accession number of an mRNA or RefSeq entry, a gene name, or
keywords from the GenBank description of an mRNA. The following
list shows examples of valid position queries for the
<em>A. mellifera</em> genome.  Note that some position queries
(e.g. "huntington") may return matches to the mRNA records of
other species. In these cases, the mRNAs are mapped to their 
homologs in <em>A. mellifera</em>. 
See the <A HREF="../goldenPath/help/hgTracksHelp.html" 
TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top><B>Genome Browser Response:</B><BR></TD>
    </TR>
    <TR><TD COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top NOWRAP>Group6.15</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of scaffold Group6.15</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Group10.1:1-50,000</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first 50,000 bases of scaffold Group10.1</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>BI510436</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of genome aligning to EST with GenBank accession BI510436</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>AB193550</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of genome aligning to mRNA with GenBank accession AB193550</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>dock</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of genome aligning to gene with identifier dock (a <em>D. melanogaster</em> gene)</TD>
    </TR>
<!--
    <TR><TD VALIGN=Top NOWRAP>NM_057677</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of genome aligning to RefSeq accession with identifier NM_057677 (a <em>D. melanogaster</em> RefSeq entry)</TD>
    </TR>
-->
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>preprosecapin</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists preprosecapin mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>odorant binding</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists odorant-binding protein mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>huntington</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists candidate genes associated with Huntington's disease</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>robinson</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists accessions deposited by authors named Robinson</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Robinson,G.E.</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author G.E. Robinson</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Robinson GE</em>, internally it uses 
			the format <em>Robinson,G.E.</em></TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
Amel_1.2 is a whole genome shotgun assembly produced using Baylor 
HGSC's assembly engine, Atlas. The assembly, which provides approximately 6x 
coverage of the clonable portion of the genome, contains 9751 scaffolds.  
Scaffolds are named according to 17 linkage groups (Group1.* - Group16.* 
and GroupUn.*).
The total scaffold size for this assembly is 213 Mb including gaps between 
contigs.  Total contig size is 208 Mb.  
The median scaffold size is 2,189 bp, average size is 21,884 bp, 
and the largest scaffold is 2,167,338 bp. The N50 scaffold size, 
defined as the length such that 50% of the assembled
genome lies in blocks of the N50 size or longer, is 260 kb.  
More details about Amel_1.2 can be found 
<A HREF="ftp://ftp.hgsc.bcm.tmc.edu/pub/data/Amellifera/fasta/Amel20040120-freeze/README.TXT" TARGET=_BLANK>here</A>.
<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/apiMel1/">FTP server</A> or the 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#apiMel">Downloads</A> page. 
These data are made available with 
<A HREF="https://www.hgsc.bcm.edu/content/conditions-use" 
TARGET=_BLANK>specific conditions for use</A>. 
The apiMel1 annotation tracks were generated by UCSC and collaborators worldwide. 
See the <A HREF="../goldenPath/credits.html#apiMel_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
<HR>
<H3>Genbank Pipeline Details</H3>
For the purposes of the Genbank alignment pipeline, this assembly is considered to be: <B>low-coverage</B>.
