<H2>Description</H2>
<P>
This track shows the draft assembly of the $organism genome.  
</P>

<P>
The <EM>Pongo pygmaeus abelii</EM> whole genome shotgun data from primary
donor-derived reads (Susie, a female sumatran orangutan housed at the
Gladys Porter Zoo (Brownsville, TX)) were assembled using PCAP (Huang,
2006) using stringent parameters derived by eliminating detectable
global mis-assemblies (interchromosomal cross-overs determined by
alignment of the orangutan genome against the human genome) larger
than 50 kb. Sequences were obtained from plasmids, fosmids, and BAC-end
sequences (from CHORI-276 also obtained from "Susie"). A fingerprint
map with a target of 12X clone coverage is currently in progress.
</P>

<P>
Of the 3.09Gb of total sequence, 3.08Gb are ordered and oriented along
the chromosomes. Gap sizes between supercontigs were estimated based on
their size in human with a maximum gap size allowed of 30kb.
</P>

<P>
The assembly data were aligned against the human genome at UCSC
(B. Raney) utilizing BLASTZ (Schwartz 2003) to align and score
non-repetitive orangutan regions against repeat-masked human
sequence. Alignment chains differentiated between orthologous and
paralogous alignments (Kent 2003) and only "reciprocal best"
alignments were retained in the alignment set. The orangutan AGP files
were generated from these alignments in a manner similar to that
already described (The Chimpanzee Genome Sequencing and Analysis
Consortium 2005).  Documented inversions based on primarily on FISH data (Rocchi,
personal communication) as well as inversions suggested by the
assembly and supported by additional mapping data (e.g. fosmid end
sequences against the human assembly, (Chen and Eichler, personal
communication)) were introduced as was the separation of alignments to
human chromosome 2 into chimpanzee chromosomes 2A and 2B.  Finally, 78
finished BAC (CHORI-273) clones were integrated into the final
chromosomal sequences.
</P>

<P>
In dense mode, this track depicts the contigs that make up the 
currently viewed chromosome. 
Contig boundaries are distinguished by the use of alternating gold and brown 
coloration. Where gaps
exist between contigs, spaces are shown between the gold and brown
blocks.  The relative order and orientation of the contigs
within a chromosome is always known; therefore, a line is drawn in the
graphical display to bridge the blocks.</P>
