<H2> Description </H2>
<P>
This track shows RNA secondary structure predictions made with the
EvoFold program, a comparative method that exploits the evolutionary signal
of genomic multiple-sequence alignments for identifying conserved
functional RNA structures. </P>

<H2> Display Conventions and Configuration </H2>
<P>
Track elements are labeled using the convention <em>ID_strand_score</em>.
When zoomed out beyond the base level, secondary structure prediction regions
are indicated by blocks, with the stem-pairing regions shown in a darker shade 
than unpaired regions. Arrows indicate the predicted strand.
When zoomed in to the base level, the specific secondary structure predictions 
are shown in parenthesis format. The confidence score for each position is
indicated in grayscale, with darker shades corresponding to higher scores.
<P>
The details page for each track element shows the predicted secondary
structure (labeled <em>SS anno</em>), together with details of the
multiple species alignments at that location. Substitutions relative
to the <em>Drosophila melanogaster</em> sequence are color-coded
according to their compatibility with the predicted secondary
structure (see the color legend on the details page). Each prediction
is assigned an overall score and a sequence of position-specific
scores. The overall score measures evidence for any functional RNA
structures in the given region, while the position-specific scores (0
- 9) measure the confidence of the base-specific
annotations. Base-pairing positions are annotated with the same pair
symbol. The <em>offsets</em> are provided to ease visual navigation of
the alignment in terms of the <em>D. melanogaster</em> sequence. The
offset is calculated (in units of ten) from the start position of the
element on the positive strand or from the end position when on the
negative strand.</P>
<P>
The graphical display may be filtered to show only those track elements 
with scores that meet or exceed a certain threshhold. To set a 
threshhold, type the minimum score into the text box at the top of the 
description page.</P>

<H2>Methods</H2>
<P>
Evofold makes use of phylogenetic stochastic context-free grammars
(phylo-SCFGs), which are combined probabilistic models of RNA
secondary structure and primary sequence evolution. The predictions
consist of both a specific RNA secondary structure and an overall
score. The overall score is essentially a log-odd score between a
phylo-SCFG modeling the constrained evolution of stem-pairing regions
and one which only models unpaired regions.</P>
<P>
The predictions for this track were based on the conserved elements of
a 12-way <em>Drosophila</em> alignment of the <em>D. melanogaster</em>
(dm2), <em>D. simulans</em> (droSim1), <em>D. sechellia</em>
(droSec1), <em>D. yakuba</em> (droYak2), <em>D. erecta</em> (droEre2),
<em>D. ananassae</em> (droAna3), <em>D. pseudoobscura</em> (dp4),
<em>D. persimilis</em> (droPer1), <em>D. willistoni</em> (droWil1),
<em>D. virilis</em> (droVir3), <em>D. mojavensis</em> (droMoj3), and
<em>D. grimshawi</em> (droGri2) assemblies. The 12-way
<em>Drosophila</em> alignment was extracted from a 15-way insect
alignment, which is the one displayed in the Conservation track.</P>

<H2>Credits</H2> 
<P>
The EvoFold program and browser track were developed by 
<A HREF="mailto:jsp@binf.ku.dk">Jakob Skou Pedersen</A>.</P>

<H2>References</H2>

<H3>Prediction analysis</H3>
<p>
Stark A, Lin MF, Kheradpour P, Pedersen JS, Parts L, Carlson JW, Crosby MA, Rasmussen MD, Roy S,
Deoras AN <em>et al</em>.
<a href="http://www.nature.com/nature/journal/v450/n7167/abs/nature06340.html" target="_blank">
Discovery of functional elements in 12 Drosophila genomes using evolutionary signatures</a>.
<em>Nature</em>. 2007 Nov 8;450(7167):219-32.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/17994088" target="_blank">17994088</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2474711/" target="_blank">PMC2474711</a>
</p>

<H3>EvoFold</H3>
<p>
Pedersen JS, Bejerano G, Siepel A, Rosenbloom K, Lindblad-Toh K, Lander ES, Kent J, Miller W,
Haussler D.
<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.0020033" target="_blank">
Identification and classification of conserved RNA secondary structures in the human genome</a>.
<em>PLoS Comput Biol</em>. 2006 Apr;2(4):e33.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/16628248" target="_blank">16628248</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1440920/" target="_blank">PMC1440920</a>
</p>

<H3>Phylo-SCFGs</H3>
<p>
Knudsen B, Hein J.
<a href="http://bioinformatics.oxfordjournals.org/content/15/6/446.abstract" target="_blank">
RNA secondary structure prediction using stochastic context-free grammars and evolutionary
history</a>.
<em>Bioinformatics</em>. 1999 Jun;15(6):446-54.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/10383470" target="_blank">10383470</a>
</p>

<p>
Pedersen JS, Meyer IM, Forsberg R, Simmonds P, Hein J.
<a href="http://nar.oxfordjournals.org/content/32/16/4925.full" target="_blank">
A comparative method for finding and folding RNA secondary structures within protein-coding
regions</a>.
<em>Nucleic Acids Res</em>. 2004;32(16):4925-36.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/15448187" target="_blank">15448187</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC519121/" target="_blank">PMC519121</a>
</p>

<H3>PhastCons</H3>
<p>
Siepel A, Bejerano G, Pedersen JS, Hinrichs AS, Hou M, Rosenbloom K, Clawson H, Spieth J, Hillier
LW, Richards S <em>et al</em>.
<a href="http://genome.cshlp.org/content/15/8/1034" target="_blank">
Evolutionarily conserved elements in vertebrate, insect, worm, and yeast genomes</a>.
<em>Genome Res</em>. 2005 Aug;15(8):1034-50.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/16024819" target="_blank">16024819</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1182216/" target="_blank">PMC1182216</a>
</p>
