<H2>Description</H2>
<P>This track displays Positive Selection analysis results of VAX003 subtype 
AE or subtype B HIV-1 gp120 nucleotide sequences.  Each positive selection site is labeled by the corresponding amino acid of the consensus sequence.
</P>

<H2>Methods</H2>
<P>The standard method for detecting adaptive molecular evolution in 
protein-coding DNA sequences is through comparison of nonsynonymous 
(amino acid changing; <i>d</i><sub>N</sub>) and synonymous 
(silent; <i>d</i><sub>S</sub>) substitution rates through the 
<i>d</i><sub>N</sub>/<i>d</i><sub>S</sub> ratio [&omega; or acceptance rate]. 
&omega; measures the difference between both rates based on a codon
substitution model. If an amino acid substitution is neutral, it will be fixed
at the same rate as a synonymous mutation, with &omega; = 1.  If the amino 
acid change is deleterious, purifying or negative selection (i.e., natural 
selection against deleterious mutations with negative selection coefficients) 
will reduce its fixation rate, thus &omega; &lt; 1.  Only when the amino acid 
change offers a selective advantage is it fixed at higher rate than a 
synonymous mutation, with &omega; &gt; 1.  Therefore, an &omega; ratio 
significantly higher than one is convincing evidence for adaptive or 
diversifying selection. In the HIV context, this is the
principle means of identifying changes and therefore sites under antigenic
selection and therefore most important in vaccine targets. Here 
<i>d</i><sub>N</sub>/<i>d</i><sub>S</sub> ratios in each subtype and 
placebo and vaccine samples were compared using methods implemented in 
PAML v3.14.  &omega; and the proportion of sites (&rho;) with &omega;
&gt; 1 were estimated under the site-specific models of
Goldman <em>et al</em>. (1994) and Yang (2000).
Tests of positive selection were performed by comparing likelihood scores 
(likelihood ratio test) between the M1 (neutral) and M2 (selection) and 
between the M7 (beta) and M8 (beta&&omega;) per-site nested models. 
More stringent models in PAML (M0) of per-gene selection were also 
estimated for comparison. If adaptive selection was identified, we then 
applied the Bayesian test developed by Yang <em>et al</em>. and implemented 
in PAML to identify the potential sites under diversifying selection as 
indicated by a posterior probability &gt; 0.95. Maximum likelihood trees 
were estimated for each subtype using PhyML under the best-fit substitution 
models. Each analysis was run twice under &omega; = 1.5 and 0.5. 
</P>

<P>
Simulations published by Anisimova <em>et al</em>. (2003) and Shriner 
<em>et al</em>. (2003) assessed the accuracy and power of the LRT and Bayes 
test implemented in PAML in the presence of recombination. General conclusions 
from these analyses indicate that excessive recombination (&rho; = 0.01), 
like usually observed in HIV sequences, can cause false positives in the Bayes 
test and makes the LRT unrealistic as it often mistakes recombination as 
evidence for positive selection. The LRT test that compares models M7 and M8 
seems to be more robust to recombination and the detection of sites under 
positive selection seems to be less affected by recombination. 
Nevertheless, a new coalescent model has been recently described that estimates 
the <i>d</i><sub>N</sub>/<i>d</i><sub>S</sub> ratio in the presence of 
recombination and hence generates simultaneous estimates of &omega; and &rho; 
using Bayesian inference (Wilson <em>et al</em>. 2006). 
Such a model is implemented in omegaMap and has been 
applied to our subtype and vaccine and placebo samples. We ran omegaMap under 
a constant model for variation (i.e., all sites are assumed to share common 
&omega; and &rho;) and the following parameter settings: 
<LI>N&#176; orders = 10</LI>
<LI>N&#176; iterations = 10<super>6</super></LI>
<LI>thinning = 100</LI>
<LI>priors = improper inverse</LI>
</P>

<H3>Results</H3>
<P>
Selection pressure, as indicated by the <i>d</i><sub>N</sub>/<i>d</i>
<sub>S</sub> ratio per gene and per site, and the proportion of sites 
under selection (p) was high for both subtypes, although subtype B 
showed higher values than subtype AE for both parameters. 
The Bayesian approach detected numerous sites under selection (n) in 
both datasets, although up two times more positively selected sites were 
observed in subtype AE than in subtype B. These differences are probably a 
consequence of the uneven sample sizes of these two datasets (181 
and 29 sequences, respectively). Simultaneous estimates of selection 
and recombination also showed higher <i>d</i><sub>N</sub>/<i>d</i><sub>S</sub>  
estimates for subtype B than for subtype AE; the recombination rate (&rho;), 
however, was almost four times higher for subtype AE than for subtype B 
(<B>Table 1</B>). Subtype AE is of recombinant origin, so one should expect 
higher recombination rates for this subtype than for subtype B. Despite this, 
recombination could inflate <i>d</i><sub>N</sub>/<i>d</i><sub>S</sub> 
rates, but this does not seem to be the case, since 
<i>d</i><sub>N</sub>/<i>d</i><sub>S</sub> rates (as estimated in PAML) 
are higher for subtype B, hence suggesting that the high frequency of 
subtype AE in Thailand is rather a founder event that could have taken 
place approximately 25 years ago.
</P>

<TABLE BORDER=1>
 <TR>
  <TD>&nbsp;</TD>
  <TD>Ns</TD>
  <TD>&omega;<sub>MO</sub></TD>
  <TD>-lnL<sub>M1</sub></TD>
  <TD>-lnL<sub>M2</sub></TD>
  <TD>&omega;<sub>M2</sub></TD>
  <TD>&rho;<sub>M2</sub></TD>
  <TD>n<sub>M2</sub></TD>
  <TD>-lnL<sub>M7</sub></TD>
  <TD>-lnL<sub>M8</sub></TD>
  <TD>&omega;<sub>M8</sub></TD>
  <TD>&rho;<sub>M8</sub></TD>
  <TD>n<sub>M8</sub></TD>
  <TD>&omega;</TD>
  <TD>&rho;</TD>
 </TR>

 <TR>
  <TD><B>Subtype AE</B></TD>
  <TD>181</TD>
  <TD>0.561</TD>
  <TD>33555.5</TD>
  <TD>32999.5</TD>
  <TD>3.29</TD>
  <TD>0.107</TD>
  <TD>42</TD>
  <TD>33450.1</TD>
  <TD>32949</TD>
  <TD>2.95</TD>
  <TD>0.108</TD>
  <TD>44</TD>
  <TD>0.404 (0.366-0.443)</TD>
  <TD>15.56 (14.65-16.65)</TD>
 </TR>

 <TR>
  <TD><B>&nbsp;&nbsp;&nbsp; Placebo</B></TD>
  <TD>93</TD>
  <TD>0.561</TD>
  <TD>19183.7</TD>
  <TD>18915.3</TD>
  <TD>3.23</TD>
  <TD>0.105</TD>
  <TD>33</TD>
  <TD>19186.9</TD>
  <TD>18918.2</TD>
  <TD>2.87</TD>
  <TD>0.119</TD>
  <TD>42</TD>
  <TD>0.424 (0.375-0.480)</TD>
  <TD>15.81 (14.02-17.84)</TD>
 </TR>

 <TR>
  <TD><B>&nbsp;&nbsp;&nbsp; Vaccine</B></TD>
  <TD>88</TD>
  <TD>0.536</TD>
  <TD>18145.8</TD>
  <TD>17870.7</TD>
  <TD>3.57</TD>
  <TD>0.096</TD>
  <TD>34</TD>
  <TD>18106.8</TD>
  <TD>17851.6</TD>
  <TD>3.07</TD>
  <TD>0.100</TD>
  <TD>30</TD>
  <TD>0.447 (0.396-0.504)</TD>
  <TD>11.17 (10.07-12.39)</TD>
 </TR>

 <TR>
  <TD><B>Subtype B</B></TD>
  <TD>29</TD>
  <TD>0.756</TD>
  <TD>9303</TD>
  <TD>9179.9</TD>
  <TD>3.68</TD>
  <TD>0.127</TD>
  <TD>34</TD>
  <TD>9321.7</TD>
  <TD>9188.2</TD>
  <TD>3.22</TD>
  <TD>0.17</TD>
  <TD>39</TD>
  <TD>0.778 (0.673-0.901)</TD>
  <TD>3.95 (3.45-4.53)</TD>
 </TR>

 <TR>
  <TD><B>&nbsp;&nbsp;&nbsp; Placebo</B></TD>
  <TD>16</TD>
  <TD>0.751</TD>
  <TD>6254.2</TD>
  <TD>6185.1</TD>
  <TD>3.94</TD>
  <TD>0.127</TD>
  <TD>23</TD>
  <TD>6262.9</TD>
  <TD>6187.6</TD>
  <TD>3.79</TD>
  <TD>0.140</TD>
  <TD>27</TD>
  <TD>0.732 (0.611-0.869)</TD>
  <TD>8.79 (6.56-11.76)</TD>
 </TR>

 <TR>
  <TD><B>&nbsp;&nbsp;&nbsp; Vaccine</B></TD>
  <TD>13</TD>
  <TD>0.774</TD>
  <TD>5312.3</TD>
  <TD>5265.4</TD>
  <TD>3.99</TD>
  <TD>0.136</TD>
  <TD>15</TD>
  <TD>5318.1</TD>
  <TD>5267.0</TD>
  <TD>3.89</TD>
  <TD>0.147</TD>
  <TD>21</TD>
  <TD>0.873 (0.722-1.059)</TD>
  <TD>1.9 (1.55-2.34)</TD>
 </TR>
</TABLE>

<P><B>Table 1.</B> Test of adaptive selection for the Thailand HIV-1 
subtypes B and AE from placebo and vaccinated individuals in PAML 
and omegaMap. All model comparisons in PAML were significant 
(<i>P</i> &lt; 0.001). The recombination rate (&rho;) under 
selection (omegaMap) is also provided. 95% HPD intervals are 
indicated between parentheses.
</P>

<H2>Credits</H2>
<P>
The data for this track were provided by Keith A. Crandall at 
<A HREF="http://www.genoma-llc.com" TARGET=_BLANK>Genoma LLC</A>.
</P>

<H2>References</H2>
<P>Anisimova M, Nielsen R, Yang Z.
<A HREF="http://www.genetics.org/cgi/content/abstract/164/3/1229" TARGET=_blank>Effect of recombination on the accuracy of the likelihood method for detecting positive selection at amino acid sites</A>. 
<EM>Genetics</EM>. 2003 Jul;164(3):1229-36.
</P>

<P>
Crandall KA, Kelsey CR, Imamichi H, Lane HC, Salzman NP. 
<A HREF="http://mbe.oxfordjournals.org/cgi/reprint/16/3/372.pdf" TARGET=_blank>Parallel evolution of drug resistance in HIV: failure of nonsynonymous/synonymous substitution rate ratio to detect selection</A>.
<EM>Mol Biol Evol</EM>. 1999 Mar;16(3):372-82.
</P>

<P>
Goldman N, Yang Z. 
<A HREF="http://mbe.oxfordjournals.org/cgi/content/abstract/11/5/725" TARGET=_blank>A codon-based model of nucleotide substitution for protein-coding DNA sequences</A>. 
<EM>Mol Biol Evol</EM>. 1994 Sep;11(5):725-36.
</P>

<P>
Miyata T, Yasunaga T. 
<A HREF="http://www.springerlink.com/content/v2645n103qu42633/" TARGET=_blank>Molecular evolution of mRNA: a method for estimating evolutionary rates of synonymous and amino acid substitutions from homologous nucleotide sequences and its application</A>. 
<EM>J Mol Evol</EM>. 1980 Mar;16(1):23-36.
</P>

<P>
Shriner, D., D. C. Nickle, M. A. Jensen, and J. I. Mullins. 
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/12872913" TARGET=_blank>Potential impact of recombination on sitewise approaches for detecting positive natural selection</A>.
<EM>Genet Res</EM>. 2003 Apr;81(2):115-21.
</P>

<P>
Wilson DJ, McVean G. 
<A HREF="http://www.genetics.org/cgi/content/abstract/172/3/1411" TARGET=_blank>Estimating diversifying selection and functional constraint in the presence of recombination</A>. 
<EM>Genetics</EM>. 2006 Mar;172(3):1411-25.
</P>

<P>
Yang Z. 
<A HREF="http://mbe.oxfordjournals.org/cgi/content/abstract/24/8/1586" TARGET=_blank>PAML 4: Phylogenetic Analysis by Maximum Likelihood</A>.
<EM>Mol Biol Evol</EM>. 2007 Aug;24(8):1586-91.
</P>

<P>
Yang Z, Nielsen R, Goldman N, Pedersen A-MK. 
<A HREF="http://www.genetics.org/cgi/content/abstract/155/1/431" TARGET=_blank>Codon-substitution models for heterogeneous selection pressure at amino acid sites</A>. 
<EM>Genetics</EM>. 2000 May;155(1):431-449.
</P>

<P>
Yang Z, Wong WSW, Nielsen R. 
<A HREF="http://mbe.oxfordjournals.org/cgi/content/abstract/msi097v1?ct" TARGET=_blank>Bayes empirical bayes inference of amino acid sites under positive selectioni</A>. 
<EM>Mol Biol Evol</EM>. 2005 Apr;22(4):1107-18.
</P>
