<H2>Description</H2>
<P>
This track shows the location of sites showing transcription 
over the first  24 hours of <em>D. melanogaster</em> development in 
two hour increments, measured by a tiling array as described in 
Manak <em>et al</em>. (2006) (see References).
Clustered sites are shown in
separate subtracks for each of the twelve timepoints.</P>

<H2>Display Conventions and Configuration</H2>
<P>
To show only selected subtracks, uncheck the boxes next to 
the tracks that you wish to hide.  
<P>
Color differences among the subtracks are arbitrary. They provide a
visual cue for distinguishing between the different timepoints.</P>

<H2>Methods</H2>
<P>
The data were processed into signal and transfrags as described in 
Cheng <em>et al</em>. (2005) and Kampa <em>et al</em>. (2004).  
The data from replicate arrays were quantile-normalized and all arrays
were scaled to a median array intensity of 25.  Within a sliding 101
bp window centered on each probe, an estimate of RNA abundance
(signal) was found by calculating the median of all pairwise average
PM-MM values, where PM is a perfect match and MM is a mismatch.</P>

<H2>Verification</H2>
<P>
Samples were hybridized
to duplicate arrays (three technical replicates). Transcribed regions
(see the Affy Signal track) 
were generated from the composite signal track by merging genomic positions
to which probes are mapped. This merging was based on a 5% false
positive rate cutoff in negative bacterial controls, a maximum
gap (MaxGap) of 50 base-pairs and minimum run (MinRun) of 90 base-pairs.

<H2>Credits</H2>
<P>
These data were generated and analyzed by the Tom Gingeras group at 
<A HREF="http://www.affymetrix.com/estore/" TARGET=_blank>Affymetrix</A>.</P>

<H2>References</H2>
<P>
Please see the 
<A HREF="http://www.gingeras.org/affy_archive_data/local_index.html"
TARGET=_blank>Affymetrix Transcriptome</A> site for a project overview and
additional references to Affymetrix tiling array publications.</P>

<p>
Cheng J, Kapranov P, Drenkow J, Dike S, Brubaker S, Patel S, Long J, Stern D, Tammana H, Helt G
<em>et al</em>.
<a href="http://www.sciencemag.org/content/308/5725/1149.abs" target="_blank">
Transcriptional maps of 10 human chromosomes at 5-nucleotide resolution</a>.
<em>Science</em>. 2005 May 20;308(5725):1149-54.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/15790807" target="_blank">15790807</a>
</p>

<p>
Kampa D, Cheng J, Kapranov P, Yamanaka M, Brubaker S, Cawley S, Drenkow J, Piccolboni A, Bekiranov
S, Helt G <em>et al</em>.
<a href="http://genome.cshlp.org/content/14/3/331.long" target="_blank">
Novel RNAs identified from an in-depth analysis of the transcriptome of human chromosomes 21 and
22</a>.
<em>Genome Res</em>. 2004 Mar;14(3):331-42.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/14993201" target="_blank">14993201</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC353210/" target="_blank">PMC353210</a>
</p>

<p>
Manak JR, Dike S, Sementchenko V, Kapranov P, Biemar F, Long J, Cheng J, Bell I, Ghosh S, Piccolboni
A <em>et al</em>.
<a href="http://www.nature.com/ng/journal/v38/n10/abs/ng1875.html" target="_blank">
Biological function of unannotated transcription during the early development of Drosophila
melanogaster</a>.
<em>Nat Genet</em>. 2006 Oct;38(10):1151-8.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/16951679" target="_blank">16951679</a>
</p>
