<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=245>
  <TR><TD ALIGN=RIGHT><A HREF="https://www.hgsc.bcm.edu/content/rat-genome-project" 
    TARGET=_blank>
    <IMG SRC="/images/Rattus_norvegicus.jpg" WIDTH=230 HEIGHT=153 ALT="Rat"></A> 
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1> <em>Rattus norvegicus</em><BR>
    </FONT>
    <FONT SIZE=-2> 
      (NHGRI <A HREF="http://www.genome.gov/dmd/img.cfm?node=Photos/Animals/Rat%20(brown%20norway)&id=79111"
      TARGET=_blank>Press Photos</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The June 2003 rat (<em>Rattus norvegicus</em>) genome assembly is
based on version 3.1 produced by the Atlas group at 
<A HREF="https://www.hgsc.bcm.edu/" TARGET=_blank>
Baylor Human Genome Sequencing Center</A> 
(HGSC) as part of the Rat Genome Sequencing Consortium. 

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a 
sequenced genomic clone, an mRNA or EST or STS marker, 
a chromosomal coordinate range, or keywords from 
the GenBank description of an mRNA. The following list shows 
examples of valid position queries for the rat genome. 
See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
TARGET=_blank> User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><BR></TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr16</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of chromosome 16</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr16:1-5000000</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first 5 million bases of chr 16</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr16:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr 16 that spans 2000 bases, starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>RNOR03233282;RNOR03233294</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region between Assembly IDs RNOR03233282 and RNOR03233294</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>AI501130</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of EST with GenBank accession AI501130</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>AF199335</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession AF199335</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>apoe</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of genome with gene identifier apoE</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>NM_145881</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of genome with RefSeq identifier NM_145881</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>25728</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of genome with Entrez Gene identifier 25728</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>kruppel zinc finger</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>huntington</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists candidate genes associated with Huntington's disease</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Jones,R.</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author R. Jones</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Jones R</em>, internally it uses 
			the format <em>Jones,R.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
This sequence was assembled using a hybrid approach that combines the 
clone-by-clone and whole genome shotgun methods.
This assembly is a minor update to the 3.0 release. Sequence changes 
affect only chromosomes 7 and X. No additional assembly releases are planned 
prior to the publication of the rat genome analysis papers. 
<P>
The 3.x assemblies reflect several sequence 
additions and software improvements over the previous 2.x assemblies, 
including the sequencing of over 1100 new BACs to cover gaps, an improved 
marker set from the Medical College of Wisconsin, a new FPC map from the 
BC Cancer Agency Genome Sciences Centre, and improved linking of bactigs. 
For detailed information and statistics about the 3.x assemblies, see the 
Baylor HGSC <A HREF="https://www.hgsc.bcm.edu/content/rat-genome-project"
TARGET=_blank>Rat Genome Project</A> web page. 
<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/rn3/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#rat">Downloads</A> page. 
These data are made available with 
<A HREF="../goldenPath/credits.html#rat_use">specific conditions for use</A>.
The rn3 annotation tracks were generated by UCSC and 
collaborators worldwide. See the <A HREF="../goldenPath/credits.html#rat_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 

<HR>
<H3>Genbank Pipeline Details</H3>
For the purposes of the Genbank alignment pipeline, this assembly is considered to be: <B>well-ordered</B>.
