<H2>Description</H2>
<P>
This track shows the draft assembly of the $organism genome.  

The assembly is 9.5X WGS coverage. The 3,510 supercontigs are displayed
in the UCSC Genome browser as a single artifical chromosome chrUn with an
artifical gap of 1,000 N's between supercontigs. The largest supercontig
is 4,147,112 base pairs with 121 supercontigs larger than 377,562 base pairs (N50) containing
over 50% of the sequence.

<P>
In dense mode, this track depicts the supercontigs.  Supercontig boundaries
are distinguished by the use of alternating gold and brown 
coloration. Arbitrary 1000 base pair gaps exist between the supercontigs, shown as
spaces between the gold and brown blocks.
</P>
<P>
All components within this track are of fragment type &quot;W&quot;: 
Whole Genome Shotgun contig. </P>

<H2>Credits</H2>
<P>
The Feb. 2008 <em>Caenorhabditis brenneri</em>
(aka: <em>Caenorhabditis</em> n. sp. 4. PB2801) assembly
produced by
<A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Caenorhabditis%20brenneri"
TARGET=_blank>
Washington University School of Medicine GSC</A> with the following
<A HREF="http://genome.wustl.edu/data.cgi" TARGET=_blank>data usage policy.</A>
</P>
<P>
As described at WormBase: "This is a 20X inbred (one gravid female per
generation) derivative of LKC28, which is conspecific with CB5161."
Recently renamed to <em>C. brenneri</em>.  Other conspecific
strains: SB129, SB280.  See also:
<A HREF="http://www.mapress.com/zootaxa/2007f/zt01456p062.pdf"
	TARGET=_blank>Zootaxa 1456: 45-62 (25 Apr. 2007)</A>,
	 Walter Sudhaus and Karin Kiontke.
</P>
