<H3>Description</H3>
<P>
This column shows microarray gene expression data from the Affymetrix All Exon
dataset. An associated Genome Browser track is available on the Human Genome
Browser.
</P>
<P>
By default, the column shows the ratio of expression of a gene in a specific 
set of tissues to the expression of the gene overall. A gene that is more 
highly expressed in a tissue is colored red, and a less expressed gene is 
shown in green. The values are colored on a logarithmic scale. This coloring 
is standard, but is the opposite of what an inexperienced user 
might expect: in this case, red means go and green means stop!  
Black indicates that a gene is neither over- nor under-expressed in the tissue. 
Uncolored boxes (white on most browsers) represent missing data.  </P>

<H3>Column Configuration</H3>
<P>
Several column configuration options are available on the Configuration page, 
which can be accessed via the <em>configure</em> button in the top menu.
<UL>
<LI>
<B>Absolute expression values:</B> The display can be configured to show 
absolute expression values rather than ratio values. To toggle the display, 
select the <em>absolute</em> option from the <em>values</em> pulldown menu. 
Absolute expression values
are shown in shades of gray, with higher expression levels shown in lighter 
shades. As with ratio values, absolute values are colored on a logarithmic 
scale.
<LI>
<B>Color scheme:</B> Color-blind users can switch the coloring scheme from 
red/green to yellow/blue via the <em>Expression ratio colors</em> pulldown 
menu at the top of the Configuration page.
<LI>
<B>Level of tissue detail:</B> The <em>tissue</em> pulldown menu
controls the amount and type of tissue data displayed. By default, the median
of replicate tissues are shown.  To display the value of all 
individual experimental replicates of all tissues, 
select the <em>all replicates</em> option.
<LI>
<B>Brightness:</B> To increase or decrease the brightness of the expression 
colors, edit the number in the column's <em>brightness</em> text box. Values 
greater than 1.0 increase the brightness, while those less than 1.0 dim the 
color. 
</UL></P>

<H3>Methods</H3>
<P>
RNA (from a commercial source) from 11 tissues were hybridized to Affymetrix 
Human Exon 1.0 ST arrays. For each tissue, 3 replicate experiments were done 
for a total of 33 arrays.  The arrays' raw signal intensity was normalized
with a quantile normalization method, then run through the PLIER algorithm.
The normalized data were then converted to log-ratios, which are displayed
as green for negative log-ratios (under-expression), and red for positive 
(over-expression).</P>
<P>When calculating expression ratios, the overall expression level in the
denominator was calculated by first taking the median of replicates for
each tissue and then taking the median of these medians. </P>
<P>
To associate particular genes with expression data, the following algorithm
was used: for each gene, the overlapping exons from the microarray dataset
were collected and the median intensity of these was assigned to the gene.
If the gene contained exons not overlapping an exon from the microarray,
those exons were ignored. If the microarray dataset contained exons not
overlapping a gene, those were ignored as well.  Once the overall expression
level of each gene was determined, this was then log-ratio'd.  In addition,
the median was taken for replicate arrays per tissue for both the absolute 
and log-ratio varieties.

<H2>Credits</H2>
<P>
The data for this track was provided and analyzed by 
<A HREF="http://www.affymetrix.com/" TARGET=_blank>Affymetrix</A>.
</P>

<H2>Links</H2>
<P><UL><LI>
<A HREF="http://www.affymetrix.com/products/arrays/specific/exon.affx" TARGET=_blank>AffyMetrix Human Exon 1.0 ST array web page</A></LI>
<LI><A HREF="http://www.affymetrix.com/support/technical/sample_data/exon_array_data.affx" TARGET=_blank>AffyMetrix Human Exon data</A></LI>
<LI><A HREF="http://www.affymetrix.com/support/technical/technotes/plier_technote.pdf" TARGET=_blank>PLIER algorithm documentation (PDF).</A></LI></UL></P>
