<H2>Description</H2>
<P>
This track shows the finished assembly of the <em>$organism</em> genome.  

This assembly merges contigs from overlapping drafts and
finished clones into longer sequence contigs.  The sequence
contigs are ordered and oriented when possible by mRNA, EST,
paired plasmid reads (from the SNP Consortium) and BAC end
sequence pairs. In the WS220 assembly, all clones are finished
and are of clone type &quot;F&quot;.</P>

<P>In dense mode, this track depicts the path through the 
finished clones (aka the golden path) used to create the assembled sequence. 
Clone boundaries are distinguished by the use of alternating gold and brown 
coloration. There are no gaps in this assembly.
</P>

<H2>Credits</H2>
<P>
The $date <em>$Organism</em> assembly 
is based on sequence version WS220 deposited into 
<A HREF="http://www.wormbase.org/" TARGET="_blank">WormBase</A>
as of Jan. 2010.  The sequence was produced jointly by the Sanger
Institute in Hinxton, England and
<A HREF="http://genome.wustl.edu/genomes/detail/caenorhabditis-elegans/"
TARGET="_blank">The Genome Institute</A> at Washington University in St. Louis.</P>
