<H2>Description</H2>

<P>
This track displays <EM>Blastz</EM> alignments of the Nov. 2002 human
assembly to the mouse genome, filtered by <em>axtBest</em> and 
<em>subsetAxt</em> with very stringent constraints as described below.    

<P>Each item in the display is identified by the chromosome, strand, and 
location of the match (in thousands).

<H2>Methods</H2>
<em>Blastz</em> uses 12 of 19 seeds and this matrix: 
<br>
<table>
<tr><td>&nbsp;</td><td>  A</td><td>  C</td><td>  G</td><td>  T</td></tr>
<tr><td>A</td><td>&nbsp;&nbsp;91</td><td>-114</td><td>&nbsp;-31</td><td>-123</td></tr>
<tr><td>C</td><td>-114</td><td>&nbsp;100</td><td>-125</td><td>&nbsp;-31</td></tr>
<tr><td>G</td><td>&nbsp;-31</td><td>-125</td><td>&nbsp;100</td><td>-114</td></tr>
<tr><td>T</td><td>-123</td><td>&nbsp;-31</td><td>-114</td><td>&nbsp;91</td></tr>
</table>
<br>
  O = 400, E = 30, K = 3000, L = 3000, M = 50
<br><br>
A second pass is done at reduced stringency (7mer seeds and
MSP threshold of K=2200) to attempt to fill in gaps (up to about 10K bp).
Lineage specific repeats are abridged during this alignment.
<P>
<em>axtBest</em> selects only the best alignment for any given region
of the genome.  
<P>
<em>subsetAxt</em> was run on <em>axtBest</em>-filtered alignments 
with this matrix:
<br>
<table>
<tr><td>&nbsp;</td><td>  A</td><td>  C</td><td>  G</td><td>  T</td></tr>
<tr><td>A</td><td>100</td><td>-200</td><td>-100</td><td>-200</td></tr>
<tr><td>C</td><td>-200</td><td>100</td><td>-200</td><td>-100</td></tr>
<tr><td>G</td><td>-100</td><td>-200</td><td>100</td><td>-200</td></tr>
<tr><td>T</td><td>-200</td><td>-100</td><td>-200</td><td>100</td></tr>
</table>
<br>

with a gap open penalty of 2000 and a gap extension penalty of 50. 
The minimum score threshold was 3400.

<H2>Using the Filter</H2>
<P>The track filter can be used to turn on the chromosome color track or to 
filter the display output by chromosome.
<UL>
<LI><B>Color track</B>: To display the chromosome color track, click the <I>on</I>
button next to "Color track based on chromosome". When the color track is 
activated, each of the items within the annotation track will be colored to 
show the chromosome in the genome assembly from which the alignment 
originated.
<LI><B>Chromosome filter</B>: To display only alignments from a specific 
chromosome in the aligning assembly, enter the chromosome number (in the form
chrN) in the text box to the right of "Filter by chromosome". For example, to
display alignments from chromosome 6, type "chr6".
</UL></P>
When you have finished configuring the filter, click the <I>Submit</I> button.

<H2>Credits</H2>
The alignments are contributed by Scott Schwartz, 
<A HREF="http://bio.cse.psu.edu/" TARGET=_blank>
Penn State Bioinformatics Group</A>. 
The best in genome filtering is done by UCSC's axtBest and subsetAxt
programs.
Mouse sequence data are provided by the 
Mouse Genome Sequencing Consortium. 

