<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=315>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Petromyzon%20marinus"
    TARGET=_blank>
    <IMG SRC="/images/Petromyzon_marinus.jpg" WIDTH=300 HEIGHT=198 ALT="Lamprey"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Petromyzon marinus</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of
      (<A HREF="http://epa.gov/greatlakes/image/index.htm"
      TARGET=_blank>U.S. Environmental Protection Agency</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Mar. 2007 <em>Petromyzon marinus</em> draft assembly 
(WUSTL v.3.0 (March 2007)) was produced by the
<A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Petromyzon%20marinus"
TARGET=_blank>Genome Sequencing Center at Washington University School of
Medicine in St. Louis</A> (WUSTL).
</P>

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
Lamprey genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>chrZ</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of chromosome Z</TD>
    </TR>
    <TR><TD VALIGN=Top>chr3:1-1000000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of chr 3</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr3:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr 3 that spans 2000 bases, starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>BX950328</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number BX950328</TD>
    </TR>
    <TR><TD VALIGN=Top>BU335953</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		BU335953 on chr Z</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>pseudogene mRNA</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>zhang</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zhang</TD>
    </TR>
    <TR><TD VALIGN=Top>Taguchi,T.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author T. Taguchi</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Taguchi T</em>, internally it
	uses the format <em>Taguchi,T.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
The genome has been sequenced to 5.9X coverage.  Please note the
description of the assembly in the WUSTL
<href="ftp://genome.wustl.edu/pub/organism/Other_Vertebrates/Petromyzon_marinus/assembly/Petromyzon_marinus-3.0/ASSEMBLY"
TARGET=_blank>README</A> file.
</P>
<P>
The UCSC browser displays the 108,241 contigs, containing 831,696,438
bases of sequence, and 195,562,529 bases in gaps (N's in the sequence)
between fragments of the contigs.  The largest contig, Contig0 is 273,059 bases,
and one half of the sequence is contained in 21,088 contigs of
size 14,239 (N50) bases are larger.
</P>
<P>
The mitochondrial genome sequence has been included as chrM.  This
is sequence from the genbank record NC_001626 accession 5835065.
</P>

<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/petMar1/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#lamprey">Downloads</A> page. 

These data have
<A HREF="../goldenPath/credits.html#lamprey_use">specific 
conditions for use</A>. 

The Lamprey browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#lamprey_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
