<H2>Description</H2>
<P>
This track shows the Zebrafish Zv9 (Jul. 2010) assembly provided by 
<A HREF="http://www.sanger.ac.uk/resources/zebrafish/genomeproject.html"
TARGET=_blank>The Wellcome Trust Sanger Institute</A>. 
See also: <A HREF="http://www.sanger.ac.uk/resources/zebrafish/faq.html"
TARGET=_blank><em>Danio rerio</em> Sequencing Project - FAQ</A>, and:
<A HREF="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/zebrafish/index.shtml"
TARGET=_blank>GRC <em>Danio rerio</em> genome overview</A>.
It contains scaffolds (supercontigs) totaling 1,412,464,843 bp. 
The assembly has been tied to the fingerprint contig (FPC) map 
(data freeze April 1, 2010), which provides a tiling path of sequenced clones.
1.41 Gb of sequence from 9,816 sequenced clones (8,276 finished and 1,090 
unfinished) were used as a scaffold for the assembly. Gaps were filled with 
contigs from a whole genome shotgun (WGS) assembly of 6.5-7x coverage comprised
of reads from a library created from a single Tuebingen doubled haploid 
zebrafish. Approximately 1.36 Gbp (96%) of the resulting integrated assembly 
were placed on chromosomes 1-25, including estimated gap sizes and 100 bp gaps 
inserted betweeen scaffolds. The complete sequence of the mitochondrion genome,
which is shown as chrM in the Genome Browser, was obtained from GenBank,
accession NC_002333.2.  </P>
<P>
The entire assembly consists of chrM, 25 chromosomes, and 
1,107 unplaced scaffolds that fall into two groups:
<UL>
<LI><B>Zv9_scaffold3453 - Zv9_scaffold3564</B> - sequences based on FPC
contigs or linked to chromosomes via a marker (in 112 unmapped scaffolds)
where scaffold3453 is the scaffold identifier.
<LI><B>Zv9_NA1 - Zv9_NA999</B> - WGS contigs that could not be related
to any FPC contig and could not be be placed on a chromosome
(in 995 unmapped scaffolds) where NA1 is the scaffold identifier.
</UL>
The unplaced scaffolds contain 100 bp gaps that are shown in the 
Gap annotation. 
<P>
Components within this track are marked type &quot;F&quot; for finished
sequences, type &quot;W&quot;  for WGS contigs, and type &quot;A&quot; for  
active finishing.
<P>
In dense mode, this track depicts the path through the draft and
finished clones (aka the golden path) used to create the assembled sequence.
Clone boundaries are distinguished by the use of alternating gold and brown
coloration. Where gaps exist in the path, spaces are shown between the gold 
and brown blocks.  If the relative order and orientation of the scaffolds
between the two blocks is known, a line is drawn to bridge the
blocks.</P>
<P>For detailed information on the methods used to produce this assembly, see 
the Wellcome Trust Sanger Institute 
<A HREF="http://www.sanger.ac.uk/resources/zebrafish/genomeproject.html"
TARGET=_blank><em>Danio rerio</em> Sequencing Project</A> website.

<H2>Credits</H2>
<P>
The Zv9 Zebrafish assembly was produced by The Wellcome Trust Sanger Institute,
in collaboration with the Max Planck Institute for Developmental Biology, 
the Netherlands Institute for Developmental Biology (Hubrecht Laboratory),
and Yi Zhou, Anthony DiBiase and Leonard Zon from the Boston Children's 
Hospital.</P> 
