<H2>Description</H2>
<P>
This track shows genome-wide DNaseI hypersensitive (HS) sites as
determined in human CD4+ T-cells.  DNaseI HS sites identify regions of
open chromatin believed to be largely free of nucleosomes and have
been shown to accurately identify the locations of genetic regulatory
elements, including promoters, enhancers, silencers, insulators, and
locus control regions.  DNaseI HS sites are believed to differ in
degree of openness.  Scores shown in this annotation reflect an
estimate of this openness.
</P>

<H2>Display Conventions and Configuration</H2>
<P>
In full and pack display modes, DNaseI HS scores are displayed as a
&quot;wiggle&quot; (histogram), where the height reflects the size of
the score.  In dense display mode, DNaseI HS is shown in grayscale
using darker values to indicate higher levels of overall
hypersensitivity.
</p>
<p>
The DNaseI HS wiggle can be configured in a variety of ways to
highlight different aspects of the displayed information. Click the <A
HREF="../goldenPath/help/hgWiggleTrackHelp.html" TARGET=_blank>Graph
configuration help</A> link for an explanation of the configuration
options.
</p>

<H2>Methods</H2>
<P>
DNaseI HS were determined for human CD4+ T-cells using
DNase-sequencing and DNase-chip.  Over 12.5 million uniquely aligned
DNase sequence tags were generated using Illumina (Solexa) and 454
Life Sciences (Roche) sequencers from the same biological sample.
Each base was assigned a score using Parzen windows kernal density
estimation.  The Nimblegen 38-array whole genome platform was used to
generate data from 2 biological replicates.  Ratio scores for the two
were averaged.  These data were combined by re-scaling each using
Z-scores and then summing.  Resulting scores above 0 are considered
hypersensitive.
</P>

<H2>Credits</H2>
<P>
This annotation was created by Alan Boyle, Terry Furey, and Greg
Crawford at Duke University's <A HREF="http://www.genome.duke.edu"
TARGET=_blank>Institute for Genome Sciences & Policy (IGSP)</A>. </P>

<H2>References</H2>
<P>
Boyle AP, Davis S, Shulha HP, Meltzer P, Margulies EH, Weng Z, Furey TS,
Crawford GE.
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/18243105?ordinalpos=1&itool=EntrezSystem2.PEntrez.Pubmed.Pubmed_ResultsPanel.Pubmed_RVDocSum"
TARGET=_blank>High-resolution mapping and characterization of open chromatin
across the genome</A>. <em>Cell</em>. 2008 Jan 25;132(2):311-22.</P>

<p>
Crawford GE, Davis S, Scacheri PC, Renaud G, Halawi MJ, Erdos MR,
Green R, Meltzer PS, Wolfsberg TG, Collins FS. <A
HREF="http://www.nature.com/nmeth/journal/v3/n7/abs/nmeth888.html"
TARGET=_blank>DNase-chip: a high-resolution method to identify DNase I
hypersensitive sites using tiled microarrays</a>. <em>Nat
Methods</em>. 2006 Jul;3(7):503-9.
</p>

<P>
Crawford GE, Holt IE, Whittle J, Webb BD, Tai D, Davis S, Margulies
EH, Chen Y, Bernat JA, Ginsburg D, Zhou D, Luo S, Vasicek TJ, Daly MJ,
Wolfsberg TG, Collins FS.  <A
HREF="http://www.genome.org/cgi/content/full/16/1/123"
TARGET=_blank>Genome-wide mapping of DNase hypersensitive sites using
massively parallel signature sequencing (MPSS)</a>.  <em>Genome
Res.</em> 2006 Jan;16(1):123-31. 
</p>
