<H2>Description</H2>
<P>
This track displays human-centric multiple sequence alignments in the 
ENCODE regions for the 28 vertebrates included in the 
<A TARGET=_BLANK HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg17/encode/alignments/SEP-2005/">September 2005 ENCODE MSA freeze</A>,
based on comparative sequence data generated for the ENCODE project
as well as whole-genome assemblies residing at UCSC, as listed:
<UL>
<LI>human (May 2004, hg17)
    <LI>armadillo (NISC, and May 2005 Broad Assisted Assembly v 1.0)
    <LI>baboon (NISC)
    <LI>chicken (Feb 2004, galGal2)
    <LI>chimp (Nov 2003, panTro1)
    <LI>colobus_monkey (NISC)
    <LI>cow (BCM)
    <LI>dog (July 2004, canFam1)
    <LI>dusky_titi (NISC)
    <LI>elephant (NISC, and May 2005 Broad Assisted Assembly v 1.0)
    <LI>fugu (Aug 2002, fr1)
    <LI>galago (NISC)
    <LI>hedgehog (NISC)
    <LI>macaque (Jan 2005, rheMac1)
    <LI>marmoset (NISC)
    <LI>monodelphis (Oct 2004, monDom1)
    <LI>mouse (Mar 2005, mm6)
    <LI>mouse_lemur (NISC)
    <LI>owl_monkey (NISC)
    <LI>platypus (NISC, and Aug 2005 Mullikin Phusion Assembly of WUGSC Traces)
    <LI>rabbit (NISC, and May 2005 Broad Assisted Assembly v 1.0)
    <LI>rat (June 2003, rn3)
    <LI>rfbat (NISC)
    <LI>shrew (NISC, and Sep 2005 Mullikin Phusion Assembly of Broad Traces)
    <LI>tenrec (Apr 2005 Mullikin Phusion Assembly of Broad Traces)
    <LI>tetraodon (Feb 2004, tetNig1)
    <LI>xenopus (Oct 2004, xenTro1)
    <LI>zebrafish (June 2004, danRer2)
</UL>
</P>
The alignments in this track were generated using the 
<A HREF="http://lagan.stanford.edu/" TARGET=_blank>LAGAN</A> Alignment Toolkit.
The Genome Browser companion tracks, MLAGAN Cons and MLAGAN Elements, 
display conservation scoring and conserved elements for these alignments based 
on various conservation methods.</P>

<H2>Display Conventions and Configuration</H2>
<P>
In full display mode, this track shows pairwise alignments
of each species aligned to the human genome.
The alignments are shown in dense display mode using a gray-scale
density gradient.  The checkboxes in the track configuration section allow
the exclusion of species from the pairwise display.
<P>
When zoomed-in to the base-display level, the track shows the base
composition of each alignment. The numbers and symbols on the &quot;human
gap&quot; line indicate the lengths of gaps in the human sequence at those
alignment positions relative to the longest non-human sequence. If the gap
size is greater than 9, a non-numeric symbol is shown; if the
gap size is a multiple of 3, a &quot;*&quot; is displayed,
otherwise &quot;+&quot; is shown.
To view detailed information about the
alignments at a specific position, zoom in the display to 30,000 or fewer
bases, then click on the alignment.  </P>

<H2>Methods</H2>
<P>MLAGAN alignments were produced by a pipeline specifically designed for
ENCODE. First, WU-BLAST [1] was used to find local similarities (anchors) 
between the human sequence and the sequence of every other species. Then, 
Shuffle-LAGAN [2] was used to calculate the highest-scoring
human-monotonic 
chain of these local similarities (according to a scoring scheme that 
penalized evolutionary rearrangements), and (with the help of a utility 
called SuperMap) produce a map of orthologous segments, in increasing human 
coordinates. This map was used to undo the genomic rearrangements of the 
other sequence and convert it to a form that was directly alignable to the
human sequence. The new humanized sequences, together with the human 
sequence, were then multiply aligned using <A 
HREF="http://lagan.stanford.edu/" TARGET=_blank>MLAGAN</A> [3]. The 
resulting alignments were subsequently refined using <A 
HREF="http://www.drive5.com/muscle/" TARGET=_blank>MUSCLE</A> [4], which 
processed small non-overlapping alignment windows and realigned them in an
iterative fashion, keeping the refined alignment if it had a better 
sum-of-pairs score than the original.  Finally, a pairwise refinement
round was performed, during which the pieces that had very low identity (in the 
induced pairwise alignments between human and each species) were removed 
from the alignment.</P>

<H2>Credits</H2>
<P>
The MLAGAN alignments were generated by George Asimenos from Stanford's 
<A HREF="http://www-shgc.stanford.edu/genetics/encode.html"
TARGET=_blank>ENCODE group</A>.</P>
<P>
Shuffle-LAGAN, SuperMap and MLAGAN were written by Mike Brudno.</P>
<P>
MUSCLE was authored by Bob Edgar.</P>

<H2>References</H2>
<P>
Gish, W. (1996-2004) 
<A TARGET=_BLANK HREF="http://blast.wustl.edu">WU-BLAST</A>
<P>
Brudno, M., Do, C., Cooper, G., Kim, M.F., Davydov, E., Green, E.D., Sidow, A.
and Batzoglou, S. 
<A HREF="http://www.genome.org/cgi/content/abstract/13/4/721"
TARGET=_blank>LAGAN and Multi-LAGAN: efficient tools for large-scale multiple 
alignment of genomic DNA</A>.
<em>Genome Res.</em> <B>13</B>(4), 721-31 (2003).</P>
<P>
Brudno, M., Malde, S., Poliakov, A., Do, C., Courone, O., Dubchak, I. and 
Batzoglou, S.
<A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/19/suppl_1/i54"
TARGET=_blank>Glocal alignment: finding rearrangements during alignment</A>.
<em>Bioinformatics</em> <B>19</B>(Suppl. 1), i54-i62 (2003).</P>
<P>
Edgar, R.C. 
<A HREF="http://nar.oupjournals.org/cgi/content/full/32/5/1792?ijkey=48Nmt1tta0fMg&keytype=ref"
TARGET=_blank>MUSCLE: multiple sequence alignment with high
accuracy and high throughput</A>.
<em>Nucl. Acids Res.</em> <B>32</B>(5), 1792-97 (2004).</P>
