<H2>Description</H2>
<P>
This track shows binding sites for p53, STAT1, and c-Myc,
as determined by chromatin
immunoprecipitation (ChIP) and paired-end di-tag (PET) sequencing.  
The p53 and c-Myc site data is genome-wide; data for STAT1
is restricted to the ENCODE regions.</P>
<P>
The p53 protein is a transcription factor involved in the control 
of cell growth that is often expressed at high levels in cancer cells.  
STAT1 is a signal transducer and transcription
factor that binds to gamma interferon activating sequence.
The c-Myc (cellular myelocytomatosis) protein is a transcription
factor associated with cell proliferation, differentiation, and
neoplastic disease. 
<P>
The PET sequences in this track are derived from individual ChIP fragments
as follows:
<P>
<TABLE BORDER='1'>
<TR><TH>Factor</TH><TH>Fragments</TH><TH>Cell line</TH><TH>Treatment</TH></TR>
<TR>
<TD>p53</TD>
<TD>65,572</TD>
<TD>HCT116</TD>
<TD> 6hrs 5-fluorouracil (5FU)</TD>
</TR>

<TR>
<TD>STAT1</TD>
<TD>263,901</TD>
<TD>HeLa</TD>
<TD>none</TD>
</TR>

<TR>
<TD>STAT1</TD>
<TD>327,838</TD>
<TD>HeLa</TD>
<TD>gamma interferon (gIFN)</TD>
</TR>

<TR>
<TD>c-Myc</TD>
<TD>273,566</TD>
<TD>P493 B cell with tetracycline-repressible c-Myc transgene</TD>
<TD> none </TD>
</TR>
</TABLE>

<P>
For the STAT1 experiments,
a total of 4,007 of the PETs from the stimulated cells and
3,180 PETs from unstimulated cells were mapped to the ENCODE regions.
The data from the unstimulated cells were used as the negative control.
Only STAT1 PETs mapped to the ENCODE regions are shown in this track.</P>
</P>

<H2>Display Conventions and Configuration</H2>
<P>
In the graphical display, PET sequences are shown as two blocks,
representing the ends of the pair, connected by a thin arrowed
line. Overlapping PET clusters (PET fragments that overlap one
another) originating from the ChIP enrichment process define the
genomic loci that are potential transcription factor binding sites (TFBSs). 
PET singletons, from non-specific ChIP fragments that did not cluster, are 
not shown.</P>
<P>
In full and packed display modes, the arrowheads on the horizontal line 
represent the orientation of the PET sequence, and an ID of the format 
<em>XXXXX-M</em> is shown to the left of each PET,
where <em>X</em> is the unique ID for each PET 
and <em>M</em> is the number of PET sequences at this location.
The track coloring reflects the value of <em>M</em>: 
light gray indicates one or two sequences (score = 333), dark gray is used for 
three sequences (score = 800) and black indicates four or more PET sequences 
(score = 1000) at the location.</P>

<H2>Methods</H2>
<P>
The cross-linked chromatin was sheared and precipitated with a high affinity
antibody. 
The DNA fragments were end-polished and cloned into the plasmid
vector, pGIS3. pGIS3 contains two <I>MmeI</I> recognition sites that
flank the cloning site, which were used to produce a 36 bp PET from the 
original ChIP DNA fragments (18 bp from each of the 5' and 3' ends). 
Multiple 36 bp PETs were concatenated and cloned into pZero-1 for 
sequencing, where each sequence read can generate 10-15 PETs. The PET 
sequences were extracted from raw sequence reads and mapped to the genome, 
defining the boundaries of each ChIP DNA fragment.  The following specific
mapping criteria were used:
<UL>
<LI>both 5' and 3' signatures must be present on the same chromosome
<LI>their 5' to 3' orientation must be correct
<LI>a minimal 17 bp match must exist for each 18 bp 5' and 3' signature
<LI>the tags must have genomic alignments within 7 Kb of each other
</UL></P>
<P>
Due to the known possibility of <I>MmeI</I> slippage (+/- 1 bp) that
leads to ambiguities at the PET signature boundaries, a minimal 17 bp
match was set for each 18 bp signature.  The total count of PET sequences
mapped to the same locus but with slight nucleotide differences may reflect
the expression level of the transcripts. Only PETs with specific mapping 
(one location) to the genome were considered.  PETs that mapped to multiple 
locations may represent low complexity or repetitive sequences, and therefore 
were not included for further analysis.</P>

<H2>Verification</H2>
<P>
Statistical and experimental verification exercises have shown that the 
overlapping PET clusters result from ChIP enrichment events. </P>

<H3>P53 HCT116</H3>
<P>
Monte Carlo simulation using the p53 ChIP-PET data estimated that
about 27% of PET-2 clusters (PET clusters with two overlapping
members), 3% of the PET clusters with 3 overlapping members (PET-3
clusters), and less than 0.0001% of PET clusters with more than 3
overlapping members were due to random chance. This suggests that the
PET clusters most likely represent the real enrichment events by ChIP
and that a higher number of overlapping fragments correlates to a
higher probability of a real ChIP enrichment event. Furthermore, based
on goodness-of-fit analysis for assessing the reliability of PET
clusters, it was estimated that less than 36% of the PET-2 clusters
and over 99% of the PET-3+ clusters (clusters with three or more
overlapping members) are true enrichment ChIP sites. Thus, the
verification rate is nearly 100% for PET-3+ ChIP clusters, and the
PET-2 clusters contain significant noise.</P>
<P>
In addition to these statistical analyses, 40 genomic locations
identified by PET-3+ clusters were randomly selected and analyzed by
quantitative real-time PCR. The relative enrichment of candidate
regions compared to control GST ChIP DNA was determined and all 40
regions (100%) were confirmed to have significant enrichment of p53
ChIP clusters. </P>
<H3>STAT1 HeLa</H3>
<P>
Monte Carlo simulation using the STAT1 ChIP-PET data from interferon 
gamma-stimulated dataset estimated that random chance accounted for about 
58% of PET-3 clusters (maximal numbers of PETs within the overlap region 
of any cluster), 21% of the PET clusters with 4 overlapping members (PET-4 
clusters), and less than 0.5% of PET clusters with more than 5 overlapping 
members. This suggests that the PET-5+ clusters represent the real enrichment 
events by ChIP and that a higher number of overlapping fragments correlates 
to a higher probability of a real ChIP enrichment event. Furthermore, based 
on goodness-of-fit analysis for assessing the reliability of PET clusters, it 
was estimated that less than 30% of the PET-4 clusters and over 90% of the 
PET-5+ clusters (clusters with five or more overlapping members) are true 
enrichment ChIP sites. </P>
<P>
In addition to these statistical analyses, 9 out of 14 genomic locations (64%) 
identified by PET-5+ clusters in the ENCODE regions were supported by ChIP-chip 
data from Yale using the same ChIP DNA as hybridization material. </P>

<H3>c-Myc P493</H3>
<P>
Monte Carlo simulation using the c-Myc ChIP-PET data estimated that 
about 32% of PET-3 clusters (maximal numbers of PETs within the overlap 
region of any cluster) and 4% of the PET clusters with 4 or more overlapping
 members (PET-4+ clusters) were due to random chance. This suggests 
that ~ 70% of PET-3+ clusters represent the real enrichment events by 
ChIP and that a higher number of overlapping fragments correlates to
 a higher probability of a real ChIP enrichment event. In addition to these
 statistical analyses, 29 genomic locations identified by PET-3+ clusters
 and 19 genomic locations defined by PET-2 clusters were randomly selected
 and subjected for quantitative real-time PCR analyses. The relative 
enrichment of candidate regions compared to control GST ChIP DNA was
 determined and all 29 PET-3+ regions (100%) and 19 PET-2 regions (47%) were
 confirmed significant enrichment of c-Myc ChIP, indicating that all of the
 PET-3+ and 47% of the PET-2 clusters defined regions are true c-Myc bound
 targets.</P

<H2>Credits</H2>
<P>
The ChIP-PET library and sequence data were produced at the 
<A HREF="http://www.gis.a-star.edu.sg/internet/site/"
TARGET=_blank>Genome Institute of Singapore</A>. 
The data were mapped
and analyzed by scientists from the Genome Institute of Singapore, 
the <A HREF="http://www.bii.a-star.edu.sg/" TARGET=_blank>Bioinformatics 
Institute, Singapore</A>, and Boston University.</P>
<P>
The STAT1 ChIP fragment prep was provided by Ghia Euskirchen from the 
<A HREF="http://www.yale.edu/snyder/" TARGET=_BLANK>Snyder lab</A> at Yale.
The c-Myc ChIP fragment prep was provided by 
Karen Zeller from the
<A HREF="http://www.hopkinsmedicine.org/cellbio/profiles/profdisplay.cfm?senduserID=18&sendpage=directory" TARGET=_BLANK>
Dang lab</A> at Johns Hopkins University.</P>

<H2>References</H2>
<P>
Ng P, Wei CL, Sung WK, Chiu KP, Lipovich L, Ang CC, Gupta S, Shahab A, 
Ridwan A, Wong CH, <EM>et al</EM>.  
<A HREF="http://www.nature.com/nmeth/journal/v2/n2/abs/nmeth733.html"
TARGET=_blank>Gene identification signature (GIS) analysis for
transcriptome characterization and genome annotation</A>.
<EM>Nat Methods</EM>. 2005 Feb;2(2):105-11. </P>
<P>
Wei CL, Wu Q, Vega VB, Chiu KP, Ng P, Zhang T, Shahab A, Yong HC, Fu Y, Weng Z
<em>et al</em>.
<A HREF="http://www.cell.com/content/article/abstract?uid=PIIS0092867405013991"
TARGET=_BLANK>A Global Map of p53 Transcription-Factor Binding Sites in the 
Human Genome</A>.
<em>Cell</em> 2006 Jan 13;124(1):207-19. </P>
<P>
Chiu KP, Wong CH, Chen Q, Ariyaratne P, Ooi HS, Wei CL, Sung WK, Ruan Y.
<A HREF="http://www.biomedcentral.com/1471-2105/7/390" TARGET=_BLANK>
PET-Tool: a software suite for comprehensive processing and managing of 
Paired-End diTag (PET) sequence data</A>.
<em>BMC Bioinformatics</em>. 2006 Aug 25;7:390. </P>
<P>
Zeller KI, Zhao X, Lee CW, Chiu KP, Yao F, Yustein JT, Ooi HS, Orlov YL, 
Shahab A, Yong HC <em>et al</em>.
<A HREF="http://www.pnas.org/cgi/content/full/103/47/17834"
TARGET=_blank>Global mapping of c-Myc binding sites and target gene networks in 
human B cells</A>.
<em>Proc Natl Acad Sci U S A</em>. 2006 Nov 21;103(47):17834-9. </P>
