<H2>Description</H2>
<P>
This track shows the map of signal intensity (estimating the fold
enrichment [log2 scale] of chromatin immunoprecipitated DNA vs. input DNA) 
for ChIP-chip using DNA from immunoprecipitated chromatin from either human 
<A HREF="http://www.atcc.org/common/catalog/numSearch/numResults.cfm?atccNum=CCL-2.2" TARGET=_blank>HelaS3</A> (cervix epithelial adenocarcinoma), <A HREF=
"http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM06990&PgId=166"
TARGET=_blank>GM06990</A>
(lymphoblastoid) or 
<A HREF="http://www.atcc.org/common/catalog/numSearch/numResults.cfm?atccNum=CCL-243" TARGET=_blank>K562</A> (myeloid leukemia-derived) cells hybridized to 
maskless photolithographic arrays. The arrays consist of 50-mer 
oligonucleotides tiled with 12-nt overlaps covering most of the 
non-repetitive DNA sequence of the ENCODE regions. Chromatin
immunoprecipitation was carried out for each experiment using antibodies 
against the following targets: BAF155, BAF170, INI1/BAF47, c-Fos, c-Jun, 
TAF1/TAFII250, RNA polymerase II, histone H4 tetra-acetylated lysine 
(H4Kac4), histone H3 tri-methylated lysine (H3K27me3), STAT1, nuclear factor 
kappa B (NFKB) p65, SMARCA4/BRG1, SMARCA6 and NRSF. Additionally, HeLa S3 cells 
immunoprecipitated with STAT1 were pre-treated with interferon-alpha and 
HeLa S3 cells immunoprecipitated with NFKB antibody were pre-treated with 
tumor necrosis factor-alpha (TNF-alpha) (see table below).
<P>
This track shows the combined results of three or four multiple biological
replicates. For all arrays, the ChIP DNA was labeled with Cy5 and the
control DNA was labeled with Cy3.
These data are available at <A HREF="http://www.ncbi.nlm.nih.gov/geo/" 
TARGET=_blank>NCBI GEO</A>  (see table below for links), which also provides 
additional information about the experimental protocols.
<P>

<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4>
<TR>
  <TH align=left>Target</TH>
  <TH align=left>GEO Accession(s)</TH>
  <TH align=left>Description</TH>
</TR>
<TR>
  <TD><A HREF="http://www.scbt.com/product.php?datasheet=10756" TARGET=_blank>
BAF155 (H-76)</A></TD>
  <TD><A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE3549" 
TARGET=_blank>GSE3549</A> (HeLa S3 cells) and 
  <A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE6898" 
TARGET=_blank>GSE6898</A> (K562 cells)</TD>
  <TD>BAF155 (Brg1-Associated Factor, 155 kD) is a human homolog of yeast 
SWI3. The Swi-Snf chromatin-remodeling complex was first described in yeast, 
and similar proteins have been found in mammalian cells. The human Swi-Snf
complex is comprised of at least nine polypeptides, including two ATPase
subunits, Brm and Brg-1. Other members of the human Swi-Snf complex are
termed BAFs for Brg1-associated factors. BAF155 is a conserved
(core) component that stimulates the chromatin remodeling activity of
Brg1.
</TD>
</TR>
<TR>
  <TD><A HREF="http://www.scbt.com/product.php?datasheet=10757" TARGET=_blank>
BAF170 (H-116)</A></TD>
  <TD><A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE3550" 
TARGET=_blank>GSE3550</A> (HeLa S3 cells) and 
  <A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE6896" 
TARGET=_blank>GSE6896</A> (K562 cells)</TD>
  <TD>BAF170 (Brg1-Associated Factor, 170 kD) is a human homolog of yeast
SWI3, a protein important in chromatin remodeling. It is a conserved (core) 
component of the Swi-Snf complex that stimulates the chromatin remodeling 
activity of Brg1 (see the description for BAF155).</TD>
</TR>
<TR>
  <TD><A HREF="http://www.scbt.com/product.php?datasheet=13055" TARGET=_blank>
INI1/BAF47 (H-300)</A></TD>
  <TD><A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE6897" 
TARGET=_blank>GSE6897</A> (K562 cells)</TD>
  <TD>INI1 (Integrase Interactor 1) or BAF47 is a human homolog of yeast 
SNF5, a protein important in chromatin remodeling.</TD>
</TR>
<TR>
  <TD>c-Fos</TD>
  <TD><A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE3449" 
TARGET=_blank>GSE3449</A> (HeLa S3 cells)</TD>
  <TD>c-Fos (transcription factor) is the cellular homolog of the v-fos 
viral oncogene. It is a member of the leucine zipper protein family and its 
transcriptional activity has been implicated in cell growth, differentiation, 
and development. Fos is induced by many stimuli, ranging from mitogens to 
pharmacological agents. c-Fos has been shown to be associated with another 
proto-oncogene, c-Jun, and together they bind to the AP-1 binding site to 
regulate gene transcription. Like CREB, c-Fos is regulated by p90Rsk.</TD>
</TR>
<TR>
  <TD>c-Jun</TD>
  <TD><A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE3448" 
TARGET=_blank>GSE3448</A> (HeLa S3 cells)</TD>
  <TD>c-Jun (transcription factor), also known as AP-1 (activator protein 1), 
is the cellular homolog of the avian sarcoma virus oncogene v-jun, and as 
such can be referred to as a proto-oncogene.</TD>
</TR>
<TR>
  <TD>TAF1/TAFII250</TD>
  <TD><A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE3450" 
TARGET=_blank>GSE3450</A> (HeLa S3 cells)</TD>
  <TD>TAF1 (TATA box binding protein (TBP)-associated factor, with molecular
weight 250 kD, also known as TAFII250) is involved in the 
initiation of transcription by RNA polymerase II. It has histone 
acetyltransferase activity, which can relieve the binding between DNA and 
histones in the nucleosome. It is the largest subunit of the basal 
transcription factor, TFIID.</TD>
</TR>
<TR>
  <TD><A HREF="http://www.scbt.com/product.php?datasheet=899" 
TARGET=_blank>RNA polymerase II (N-20), N-terminus</A></TD>
  <TD><A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE6390" 
TARGET=_blank>GSE6390</A> (HeLa S3 cells) and 
  <A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE6392" 
TARGET=_blank>GSE6392</A> (GM06990 cells)</TD>
  <TD>RNA polymerase II (pol II) catalyzes transcription of DNA for the 
production of mRNAs and most snoRNAs. </TD>
</TR>
<TR>
  <TD>
<A HREF="http://store.crpinc.com/datasheet.aspx?UpProd=&catalogno=MMS-126R"
TARGET=_blank>RNA polymerase II (8WG16), C-terminus</A></TD>
  <TD><A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE6391" 
TARGET=_blank>GSE6391</A> (HeLa S3 cells) and 
  <A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE6394" 
TARGET=_blank>GSE6394</A> (GM06990 cells)</TD>
  <TD>RNA polymerase II (pol II) catalyzes transcription of DNA for the 
production of mRNAs and most snoRNAs. This antibody targets the pre-initiation 
complex form recognizing the C-terminal hexapeptide repeat of the large 
subunit of pol II. The initiation-complex form of RNA polymerase II is 
associated with the transcription start site.</TD>
</TR>
<TR>
  <TD><A HREF="http://www.upstate.com/browse/productdetail.asp?ProductId=06-866"
TARGET=_blank>H4Kac4</A></TD>
  <TD><A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE6389" 
TARGET=_blank>GSE6389</A> (HeLa S3 cells) and 
  <A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE6393" 
TARGET=_blank>GSE6393</A> (GM06990 cells)</TD>
  <TD>H4Kac4 (Histone H4 tetra-acetylated lysine) is a post-translational
modification of the histone which affects chromatin remodeling. Histone H4
is found in transcriptionally active euchromatin.</TD>
</TR>
<TR>
  <TD><A HREF="http://www.upstate.com/browse/productdetail.asp?ProductId=07-449"
TARGET=_blank>H3K27me3</A></TD>
  <TD><A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE8073"
TARGET=_blank>GSE8073</A> (HeLa S3 cells)</TD>
  <TD>H3K27me3 (Histone H3 tri-methylated lysine) is a post-translational
modification of the histone which affects chromatin remodeling. It is known
to be associated with heterochromatin.</TD>
</TR>
<TR>
  <TD><A HREF="http://www.scbt.com/product.php?datasheet=345"
TARGET=_blank>STAT1 p91 (C-24)</A></TD>
  <TD><A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE6892" 
TARGET=_blank>GSE6892</A> (HeLa S3 cells, interferon-alpha stimulated)</TD>
  <TD>STAT1 (Signal Transducer and Activator of Transcription 1) responds to
many cytokines and growth factors and regulates genes important for apoptosis,
inflammation, and the immune system.</TD>
</TR>
<TR>
  <TD><A HREF="http://www.scbt.com/product.php?datasheet=109"
TARGET=_blank>NFKB p65, N-terminus</A></TD>
  <TD><A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE6900" 
TARGET=_blank>GSE6900</A> (HeLa S3 cells, TNF-alpha stimulated)</TD>
  <TD>NFKB p65 (RelA) is the strongest transcriptional-activator among the five
members of the mammalian NF-kB/Rel family and plays an essential role in
regulating the induction of genes involved in several physiological processes,
including immune and inflammatory responses.</TD>
</TR>
<TR>
  <TD><A HREF="http://www.scbt.com/product.php?datasheet=372"
TARGET=_blank>NFKB p65 (C-20), C-terminus</A></TD>
  <TD><A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE6899" 
TARGET=_blank>GSE6899</A> (HeLa S3 cells, TNF-alpha stimulated)</TD>
  <TD>NFKB p65 (RelA) is the strongest transcriptional-activator among the five
members of the mammalian NF-kB/Rel family and plays an essential role in
regulating the induction of genes involved in several physiological processes,
including immune and inflammatory responses.</TD>
</TR>
<TR>
  <TD><A HREF="http://www.scbt.com/product.php?datasheet=17796"
TARGET=_blank>SMARCA4/BRG1</A></TD>
  <TD><A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE7370" 
TARGET=_blank>GSE7370</A> (HeLa S3 cells)</TD>
  <TD>SMARCA4 (BRG1) is a catalytic subunit of the SWI/SNF chromatin remodeling 
complex. It is a member of the SNF2 family of chromatin remodeling ATPases.</TD>
</TR>
</TR>
<TR>
  <TD>
<A HREF="http://www.abcam.com/index.html?pageconfig=datasheet&intAbID=3851"
TARGET=_blank>SMARCA6</A></TD>
  <TD><A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE7371" 
TARGET=_blank>GSE7371</A> (HeLa S3 cells)</TD>
  <TD>SMARCA6 is a SNF2-like helicase linked to cell proliferation and 
DNA methylation. It is a member of the SNF2 family of chromatin remodeling
ATPases.</TD>
<TR>
  <TD><A HREF="http://www.upstate.com/browse/productdetail.asp?ProductId=07-579"
TARGET=_blank>NRSF</A></TD>
  <TD><A HREF="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE7372"
TARGET=_blank>GSE7372</A> (HeLa S3 cells)</TD>
  <TD>NRSF (neuron-restrictive silencer factor) represses neuron-specific 
genes in non-neuronal cells.</TD>
</TABLE></P>

<H2>Display Conventions and Configuration</H2>
<P>
The subtracks within this annotation 
may be configured in a variety of ways to highlight different aspects of the 
displayed data. The graphical configuration options are shown at the top of 
the track description page, followed by a list of subtracks.  To display only 
selected subtracks, uncheck the boxes next to the tracks you wish to hide. 
For more information about the graphical configuration options, click the 
<A HREF="../goldenPath/help/hgWiggleTrackHelp.html" TARGET=_blank>Graph
configuration help</A> link.</P>

<H2>Methods</H2>
<P>
The data from replicates were quantile-normalized and median-scaled to each
other (both Cy3 and Cy5 channels). Using a 1000 bp sliding window centered
on each oligonucleotide probe, a signal map (estimating the fold enrichment
[log2 scale] of ChIP DNA) was generated by computing the pseudomedian
signal of all log2(Cy5/Cy3) ratios (median of pairwise averages) within the
window, including replicates. Using the same procedure, a -log10(P-value)
map (measuring significance of enrichment of oligonucleotide probes in the
window) for all sliding windows was made by computing P-values using the
Wilcoxon paired signed rank test comparing fluorensent intensity between
Cy5 and Cy3 for each oligonucleotide probe (Cy5 and Cy3 signals from the
same array). A binding site was determined by thresholding oligonucleotide
positions with -log10(P-value) (>= 4), extending qualified positions
upstream and downstream 250 bp, and requiring 1000 bp space between two
sites. Top 400 sites are retained.

<H2>Verification</H2>
<P>
ChIP-chip binding sites were verified by comparing "hit lists" generated
from combinations of different biological replicates. Only experiments that
yielded a significant overlap (greater than 50 percent) were accepted. As
an independent check (for maskless arrays), data on the microarray were
randomized with respect to position and re-scored; significantly fewer hits
(consistent with random noise) were generated this way.

<H2>Credits</H2>
<P>
These data were generated and analyzed by the labs of Michael Snyder, Mark
Gerstein and Sherman Weissman at Yale University.

<H2>References</H2>
<P>
Cawley S, Bekiranov S, Ng HH, Kapranov P, Sekinger EA, Kampa
D, Piccolboni A, Sementchenko V, Cheng J, Williams AJ <em>et al</em>.
<a href = "http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=14980218&dopt=Abstract"
TARGET=_blank>
Unbiased mapping of transcription factor binding sites along human
chromosomes 21 and 22 points to widespread regulation of noncoding RNAs</a>.
<em>Cell</em> 2004 Feb 20;116(4):499-509.
<P>
Euskirchen G, Royce TE, Bertone P, Martone R, Rinn JL, Nelson
FK, Sayward F, Luscombe NM, Miller P, Gerstein M <em>et al</em>.
<a href = "http://mcb.asm.org/cgi/content/full/24/9/3804?view=long&pmid=15082775"
TARGET=_blank>
CREB binds to multiple loci on human chromosome 22</a>.
<em>Mol Cell Biol.</em> 2004 May;24(9):3804-14.
<P>
Martone R, Euskirchen G, Bertone P, Hartman S, Royce TE,
Luscombe NM, Rinn JL, Nelson FK, Miller P, Gerstein M <em>et al</em>.
<a href = "http://www.pnas.org/cgi/content/full/100/21/12247"
TARGET=_blank>
Distribution of NF-kappaB-binding sites across human chromosome 22</a>.
<em>Proc Natl Acad Sci U S A.</em> 2003 Oct 14;100(21):12247-52.
<P>
Quackenbush J.
<a href = "http://www.nature.com/ng/journal/v32/n4s/full/ng1032.html;jsessionid=1FB90778A00BAB3560FAB110F7347905"
TARGET=_blank>
Microarray data normalization and transformation</a>
<em> Nat Genet.</em> 2002 Dec;32(Suppl):496-501.


