<H2>Description</H2>
<P>
This track shows STAT1 binding sites as determined by chromatin 
immunoprecipitation (ChIP) and paired-end di-tag (PET) sequencing.  
<P>
The PET sequences in this track are derived from
327,838 STAT1 ChIP fragments of interferon gamma-stimulated HeLa cells and
263,901 STAT1 ChIP fragments of non-stimulated HeLa cells.  
Of these individual ChIP fragments, 3,180 of the PETs from the stimulated cells and
4,007 PETs from unstimulated cells were mapped to the ENCODE regions.  
The data from the unstimulated cells were used as the negative control.</P>
<P>
Only PETs mapped to the ENCODE regions are shown in this track.  

<H2>Display Conventions and Configuration</H2>
<P>
In the graphical display, PET sequences are shown as two blocks,
representing the ends of the pair, connected by a thin, arrowed
line. Overlapping PET clusters (PET fragments that overlap one
another) originating from the ChIP enrichment process define the
genomic loci that are potential transcription factor binding sites (TFBSs).
PET singletons, from non-specific ChIP fragments that did not cluster, are
not shown.</P>
<P>
In full and packed display modes, the arrowheads on the horizontal line
represent the orientation of the PET sequence, and an ID of the format
<em>XXXXX-M</em> is shown to the left of each PET,
where <em>X</em> is the unique ID for each PET
and <em>M</em> is the number of PET sequences at this location.
The track coloring reflects the value of <em>M</em>:
light gray indicates one or two sequences (score = 333), dark gray is used for
three sequences (score = 800) and black indicates four or more PET sequences
(score = 1000) at the location.</P>

<H2>Methods</H2>
<P>
The STAT1 chromatin immuno-precipitated DNA fragments from stimulated and 
non-stimulated control cells were end-polished and cloned into the plasmid 
vector, pGIS3.  pGIS3 contains two <I>MmeI</I> recognition sites that flank 
the cloning site, which were used to produce a 36 bp PET from the original 
ChIP DNA fragments (18 bp from each of the 5' and 3' ends). Multiple 36 bp 
PETs were concatenated and cloned into pZero-1 for sequencing, where each 
sequence read can generate 10-15 PETs. The PET sequences were extracted from 
raw sequence reads and mapped to the genome, defining the boundaries of each 
ChIP DNA fragment. The following specific mapping criteria were used: 
<UL>
<LI>both 5' and 3' signatures must be present on the same chromosome 
<LI>their 5' to 3' orientation must be correct 
<LI>a minimal 17 bp match must exist for each 18 bp 5' and 3' signature 
<LI>the tags must have genomic alignments within 6 Kb of each other 
</UL></P>
<P>
Due to the known possibility of <I>MmeI</I> slippage (+/- 1 bp) that leads to 
ambiguities at the PET signature boundaries, a minimal 17 bp match was set 
for each 18 bp signature. Only PETs with specific mapping (one location) to 
the genome were considered. PETs that mapped to multiple locations may 
represent low complexity or repetitive sequences, and therefore were not 
included for further analysis.</P>
  
<H2>Verification</H2>
<P>
Statistical and experimental verification exercises have shown that the 
overlapping PET clusters result from ChIP enrichment events. </P>
<P>
Monte Carlo simulation using the STAT1 ChIP-PET data from interferon 
gamma-stimulated dataset estimated that random chance accounted for about 
58% of PET-3 clusters (maximal numbers of PETs within the overlap region 
of any cluster), 21% of the PET clusters with 4 overlapping members (PET-4 
clusters), and less than 0.5% of PET clusters with more than 5 overlapping 
members. This suggests that the PET-5+ clusters represent the real enrichment 
events by ChIP and that a higher number of overlapping fragments correlates 
to a higher probability of a real ChIP enrichment event. Furthermore, based 
on goodness-of-fit analysis for assessing the reliability of PET clusters, it 
was estimated that less than 30% of the PET-4 clusters and over 90% of the 
PET-5+ clusters (clusters with five or more overlapping members) are true 
enrichment ChIP sites. </P>
<P>
In addition to these statistical analyses, 9 out of 14 genomic locations (64%) 
identified by PET-5+ clusters in the ENCODE regions were supported by ChIP-chip 
data from Yale using the same ChIP DNA as hybridization material. </P>

<H2>Credits</H2>
<P>
The ChIP fragment prep was provided by Ghia Euskirchen from Michael Snyder's 
lab at Yale.  The ChIP-PET library and sequence data were produced at the
<A HREF="http://www.gis.a-star.edu.sg/internet/site/"
TARGET=_blank>Genome Institute of Singapore</A>.  The data were mapped
and analyzed by scientists from the Genome Institute of Singapore and
the <A HREF="http://www.bii.a-star.edu.sg/" TARGET=_blank>Bioinformatics 
Institute, Singapore</A>.</P>

<H2>References</H2>
<P>
Ng, P. <em>et al</em>.  
<A HREF="http://www.nature.com/nmeth/journal/v2/n2/abs/nmeth733.html"
TARGET=_blank>Gene identification signature (GIS) analysis for
transcriptome characterization and genome annotation</A>.  <em>Nature
Methods</em> <B>2</B>, 105-111 (2005). </P>
