<H2>Description</H2>
<P>
Chromatin IP was performed as described in Mortazavi et al. (2006).  Libraries were prepared from the ChIP DNA using the 
ligation mediated PCR Solexa protocol.  The ChIP libraries were then used as a substrate for the Solexa sequencing platform.

<H2>Methods</H2>
<P>
This track represents raw 25mer ChIPSeq reads mapped back to the genome with one or fewer mismatched nucleotides.

<H2>Verification</H2>
<P>
These data represent a pool of 70 replicate immunoprecipitations.  We have also included a control track 
(a non-immunoenriched, sequenced Solexa library).  Using a set of 74 known positive loci and 139 known negative 
loci (Mortazavi et al., 2006), we estimate a sensitivity of 87% and a specificity of 97% for clusters of 13 tags 
separated by no more than 100bp.

<H2>Credits</H2>
<P>
Myers Group: David Johnson, Betsy Anton, Loan Nguyen, Cat Medina, Richard Myers.
<P>
Wold Group: Barbara Wold, Ali Mortazavi, Kenneth McCue.
<P>
Solexa/Illumina Sequencing: Gary Schroth.

<H2>References</H2>
<P>
Mortazavi A, Thompson EC, Garcia ST, Myers RM, Wold B. Comparative genomics modeling of the NRSF/REST repressor network: 
from single conserved sites to genome-wide repertoire.
Genome Res. 2006 Oct;16(10):1208-21. 
