<H2>Description</H2>
<P>
The exoniphy program identifies evolutionarily conserved protein-coding exons in
multiple, aligned sequences using a phylogenetic hidden Markov model
(phylo-HMM), a kind of statistical model that simultaneously describes exon
structure and exon evolution.  This track shows exoniphy predictions
for the human Jul. 2003 (hg16), mouse Feb. 2003 (mm3), and rat Jun. 2003
(rn3) genomes, as aligned by the multiz program.  These predictions have
been mapped to the human May 2004 (hg17) assembly using the liftOver program.
</P>

<H2>Methods</H2>
<P>
For a description of exoniphy, see Siepel <em>et al.</em> (2004).
Multiz is described in Blanchette <em>et al.</em> (2004).

<H2>References</H2>
<P>
Blanchette, M. <em>et al.</em> 
<A HREF="http://www.genome.org/cgi/content/abstract/14/4/708"
TARGET=_blank>Aligning multiple genomic sequences with the threaded
blockset aligner</A>. 
<em>Genome Res.</em> <B>14</B>, 708-175 (2004).</P>
<P>
Siepel, A. and Haussler, D. 
<A HREF="http://compgen.bscb.cornell.edu/~acs/recomb2004.pdf"
TARGET=_blank>Computational identification of evolutionarily conserved
exons</A>. <em>RECOMB '04</em> (2004).</P>
