<H2>Description</H2>
<P>
This annotation consolidates mutations from locus-specific databases and a 
genome-wide mutation database into a single track.
The prototype version includes entries from the few locus-specific databases
listed in the Methods section below. 
</P>

<H2>Display Conventions and Configuration</H2>
<P>
This track is color-coded by the disease-association or mutation type.<BR />
The default colors for disease-association are:
<UL>
<LI>
known to be disease-associated = <font color="#DD0000">red</font>
<LI>
likely to be phenotype-associated = <font color="#FF9900">orange</font>
<LI>
not disease-associated = <font color="#009900">green</font>
<LI>
remaining mutations = <font color="GRAY">gray</font>
</UL>
The default colors for mutation type are:
<UL>
<LI>
substitution = <font color="#CC00FF">purple</font>
<LI>
insertion = <font color="#009900">green</font>
<LI>
duplication = <font color="#FF9900">orange</font>
<LI>
deletion = <font color="#0000FF">blue</font>
<LI>
complex = <font color="#643200">brown</font>
<LI>
unknown = black
</UL>
The color choices can be changed using the popup menus on the description page.
</P>
<P>
By default items are labeled with the Human Genome Variation Society (HGVS) 
name.  Alternatively, you may configure the browser to display the common name, 
the feature ID, or no label by changing which label option is selected at the 
top of the description page.</P>
<P>
All mutation types, locations and data sources are included in the default
display. To filter the display based on one or more of these characteristics,
check the appropriate boxes in the exclusion lists at the top of the 
description page.
</P>

<a name="source">
<H2>Methods</H2>
<P>
The data shown in this track were obtained from the following variant sources:
<UL>
<LI>
<A HREF="http://globin.bx.psu.edu/hbvar/menu.html" TARGET=_blank>HbVar 
database</A> - variants and thalassemia mutations in the globin loci.
<LI>
<A HREF="http://www.pahdb.mcgill.ca/" TARGET=_blank>PAHdb</A> - mutations 
in the phenylalanine hydroxylase (PAH) gene.
<LI>
<A HREF="http://www.ncbi.nlm.nih.gov/projects/mhc/xslcgi.fcgi?cmd=bgmut/home" 
TARGET=_blank>Blood Group Antigen Gene Mutation Database</A> (BGMUT) - 
mutations in many genes.
<LI>
<A HREF="http://www.androgendb.mcgill.ca/" TARGET=_blank>Androgen Receptor Gene 
Mutations Database</A> (ARdb) - mutations in the AR gene.
<LI>
<A HREF="http://www.genet.sickkids.on.ca/cftr/" TARGET=_blank>Cystic Fibrosis 
Mutation Database</A> (CFMDB) - mutations in the CFTR gene.
<LI>
<A HREF="http://www.expasy.org/sprot/" TARGET=_blank>UniProt 
(Swiss-Prot/TrEMBL)</A> - 
genome-wide data on variants within coding regions.  Only those that are from
UCSC Known Genes.
<!--
<LI>
A possible connection with <A HREF="http://www.centralmutations.org" 
TARGET=_blank>WayStation</A>, the HGVS community-wide database.
-->
</UL>
<p>The disease association is determined by a database curator listing the
mutations that are "known to be disease-associated" or "not disease-associated".
Of the remaining mutations those that are insertions or deletions or 
non-synonymous substitutions in coding regions are considered to be
"likely to be phenotype-associated", as well as known splicing mutations.
Also mutations that intersect with 
regions in the most conserved track are considered to be 
"likely to be phenotype-associated".  The remaining entries are not known and
left blank (gray by default).

<H2>Credits</H2>

<H4>PhenCode developers</H4>
<DL>
<DT>Belinda Giardine, Ross Hardison, Webb Miller, Cathy Riemer
<DD>Center for Comparative Genomics and Bioinformatics, Penn State University, University Park, PA
<DT>Fan Hsu, Jim Kent, Andrew Kern, Robert Kuhn, Heather Trumbower
<DD>Center for Biomolecular Science and Engineering, University of California, Santa Cruz, CA
<DT>Richard Gibbons, Doug Higgs, Jim Hughes
<DD>Weatherall Institute of Molecular Medicine, Oxford UK
<DT>Garry Cutting, Andrew P. Feinberg
<DD>Johns Hopkins University School of Medicine, Baltimore, MD
</DL> 

<H4>Cooperating databases</H4>
<DL>
<DT>ARdb:
<DD>Dr. Bruce Gottlieb,
Lady Davis Institute for Medical Research,
Sir Mortimer B. Davis-Jewish General Hospital
Montreal, Quebec, Canada
<DT>BGMUT:
<DD>Olga O. Blumenfeld, Ph.D., Professor <br>
Santosh K. Patnaik<br>
Dept. of Biochemistry,
Albert Einstein College of Medicine,
Bronx, NY
<DT>Genes BRCA1 and BRCA2:
<DD>Mary-Claire King, Ph.D., University of Washington School of Medicine, WA
<DT>CFMDB:
<DD>Dr. Julian Zielenski, Dr. Richard Sang,
The Hospital for Sick Children,
Genetics and Genomic Biology,
Ontario, Canada
<DT>HbVar:
<DD>George Patrinos, Erasmus University, Rotterdam, Netherlands<br>
Henri Wajcman, Hospital Henri Mondor, Creteil, France<br>
David H.K. Chui, Boston University, Boston, MA<br>
Nicholas Anagnou, U. Athens and IIBEAA, Athens, Greece<br>
Georgi D. Efremov, Macedonian Academy of Sciences and Arts, RCGEB, Skopje, Macedonia
<DT>PAH:
<DD>Charles R. Scriver MDCM FRS,
Alva Professor Emeritus of Human Genetics,
McGill University,
Montreal Children's Hospital Research Institute, Canada
<DT>Gene srd5a2:
<DD>Prof. Juergen Reichardt, University of Sydney, Australia<br>
Nick Makridakis, Ph.D., University of Southern California, California
<DT>UniProt (Swiss-Prot/TrEMBL):
<DD>Swiss Institute of Bioinformatics, Geneva, Switzerland
</DL>
