<H2>Description</H2>
<P>
This track shows Human Accelerated Regions (HARs), genomic elements that 
are conserved in other vertebrates but significantly changed in humans. 
The HARs were identified in a genome-wide scan in two stages. First, all 
regions of at least 100bp that are 96% or more identical between the 
chimp, mouse and rat genomes were identified. Then, each of these regions
was examined for evidence of substitution rate acceleration on the human 
lineage compared to 11 other vertebrates (including mammalian, amphibian, 
bird, and fish species) plus a parsimony inferred chimp-human ancestor. 
The 202 HARs displayed in this track all showed statistically significant 
(FDR adjusted p<0.1) acceleration in the human lineage. Details of the
HARs are available <A href="http://docpollard.com/HARs.html" TARGET=_blank>
here</A>. 
</P>

<H2>Methods</H2>
<P>
Multiz alignments of the following assemblies were used to generate this 
track: 
<UL>
<LI>human (May 2004, hg17)
<LI>chimp (Nov 2003, panTro1)
<LI>macaque (Jan 2006, rheMac2)
<LI>mouse (Aug 2005, mm7)
<LI>rat (Jun 2003, rn3)
<LI>dog (May 2005, canFam2)
<LI>cow (Mar 2005, bosTau2)
<LI>armadillo (May 2005, dasNov1)
<LI>elephant (May 2005, loxAfr1)
<LI>tenrec (Jul 2005, echTel1)
<LI>opossum (Jun 2005, monDom2)
<LI>chicken (Feb 2004, galGal2)
<LI>frog (Oct 2004, xenTro1)
<LI>zebrafish (May 2005, danRer3)
<LI>tetraodon (Feb 2004, tetNig1)
<LI>fugu (Aug 2002, fr1)
</UL>

Alignment gaps and CpG dinucleotides were excluded from the analysis. 
The phylogeny used was the same as that employed in the 17 species
vertebrate conservation track, described 
<A href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?hgsid=77729200&c=chrX&g=multiz17way" TARGET=_blank> 
here</A>. Filters were used to remove cases of assembly errors, alignment
errors, human pseudogenes, and misaligned paralogous sequences. 
</P>

<P>
Each chimp-rodent conserved region was assessed for evidence of accelerated 
substitution rate in the human lineage using a Likelihood Ratio Test (LRT). 
The LRT statistic compares the likelihood of the alignment data under a 
molecular evolutionary model with a human substitution rate parameter 
(constrained to be accelerated only) to a model without this parameter. Both 
models are fit to each alignment by scaling a genome-wide model for conserved 
sequences. The general time-reversible (REV) single-nucleotide model for 
molecular evolution was used. Large values of the LRT statistic indicate more 
evidence for acceleration in the human lineage. Significance was assessed by 
simulation from the genome-wide (no acceleration) model. P-values were 
adjusted for multiple comparisons using the false discovery rate controlling 
procedure of Benjamini &amp; Hochberg (1995).
</P>

<P>
More information about the methods can be found in the two references by 
Pollard et al. (2006). 
</P>

<H2>Credits</H2>
<P>
The genome-wide HAR screen was developed and executed at UCSC by Katherine 
Pollard, using programs from the PHAST library written by Adam Siepel and 
scripts contributed by Gill Bejerano. 
</P>

<H2>References</H2>
<P>
K.S. Pollard, S.R. Salama, B. King, A.D. Kern, T. Dreszer, S. Katzman, A. 
Siepel, J.S. Pedersen, G. Bejerano, R. Baertsch, K.R. Rosenbloom, J. Kent, 
D. Haussler <A href="http://genetics.plosjournals.org/perlserv/?request=get-document&doi=10.1371/journal.pgen.0020168.eor" TARGET=_blank> Forces shaping the 
fastest evolving regions in the human genome.</A> PLoS Genetics, early 
online publication (August 23, 2006).
</P>
<P>
K.S. Pollard, S.R. Salama, N. Lambert, M.A. Lambot, S. Coppens, J.S. 
Pedersen, S. Katzman, B. King, C. Onodera, A. Siepel1, A.D. Kern, C. Dehay, 
H. Igel, M. Ares Jr, P. Vanderhaeghen, D. Haussler <A href="http://www.nature.com/nature/journal/vaop/ncurrent/full/nature05113.html" TARGET=_blank> An RNA 
gene expressed during cortical development evolved rapidly in humans.</A>
 Nature, early online publication (August 16, 2006). 
</P>
<P>
Y. Benjamini, Y. Hochberg. Controlling the False Discovery Rate - a 
practical and powerful approach to multiple testing. Journal of the Royal 
Statistical Society Series B-Methodological <B>57</B>, 289-300 (1995).
</P>