<H2>Description</H2>
This track displays positions of four different types of RNA in the human 
genome: 
<UL>
<LI>microRNAs from the
<A HREF="http://www.sanger.ac.uk/Software/Rfam/mirna/" TARGET=_blank>
miRNA Registry</A> at the <A HREF="http://www.sanger.ac.uk/" TARGET=_blank>
Wellcome Trust Sanger Institute</A>
<LI>small nucleolar RNAs (C/D box and H/ACA box snoRNAs) and Cajal body-specific RNAs (scaRNAs) from the snoRNA-LBME-DB maintained at the 
<A HREF="http://www-lbme.biotoul.fr/" TARGET=_blank>Laboratoire de Biologie 
Moléculaire Eucaryote</A>. 
</UL>
C/D box and H/ACA box snoRNAs are guides for the 2'O-ribose methylation and 
the pseudouridilation, respectively, of rRNAs and snRNAs, although many of 
them have no documented target RNA.  The scaRNAs guide modifications of the
spliceosomal snRNAs transcribed by RNA polymerase II, and often contain both C/D and H/ACA domains.
<P>
Mature miRNAs (miRs) are represented by thick red blocks; the predicted 
stem-loop portions of the primary transcripts are indicated by thinner 
blocks.  When a single precursor produces two mature miRs from its 5' and 3' 
parts, both positions of the mature miRNA are individually displayed.
<P>
C/D box, H/ACA box snoRNAs and scaRNAs are represented by blue, green and magenta blocks, 
respectively. At a zoomed-in resolution, arrows superimposed on the blocks indicate 
the sense orientation of the snoRNAs. 
<H2>Methods</H2>
<P>
Mature and precursor miRNAs from the miRNA Registry were aligned against the 
genome using
<A HREF="http://genome.ucsc.edu/cgi-bin/hgBlat" TARGET=_blank>blat</A>.
The extents of the precursor sequences were not generally known and were
predicted based on base-paired hairpin structure.  The miRNA Registry is
described in Griffiths-Jones, S.,
<A HREF="http://nar.oupjournals.org/cgi/content/full/32/suppl_1/D109"
TARGET=_blank>The microRNA Registry</A>, 
<em>Nucl. Acids Res.</em> <B>32</B>, D109-D111 (2004).
<P>
The snoRNAs and scaRNAs from the snoRNA-LBME-DB were aligned against the human genome using blat. 
</P>

<H2>Credits</H2>
<P>
The miRNA annotation was contributed by Michel Weber of 
<A HREF="http://www-lbme.biotoul.fr/" TARGET=_blank>Laboratoire de Biologie 
Moléculaire Eucaryote, CNRS Université Paul Sabatier </A> (UMR5099, Toulouse, France) 
and Sam Griffiths-Jones of <A HREF="http://www.sanger.ac.uk" TARGET=_blank>The
Wellcome Trust Sanger Institute</A> (Cambridge, UK). 
<P>
The snoRNA annotations were contributed by Michel Weber and 
Laurent Lestrade of the 
<A HREF="http://www.iefg.biotoul.fr/" TARGET=_blank>Institut d'Exploration 
Fonctionnelle des Génomes</A> (IFR109, Toulouse, France). 
<P> 
<A HREF="mailto:&#102;a&#110;&#104;s&#117;&#64;&#115;&#111;&#101;.&#117;c&#115;&#99;.&#101;&#100;&#117;">Fan Hsu</A> from the UCSC Genome Bioinformatics 
Group created the combined annotation track.

<H2>References</H2>
<P>
If you make use of the miRNA data, please cite
the article listed in the Methods section in addition to the primary data 
sources.
You may also want to cite The Wellcome Trust Sanger 
Institute <A HREF="http://www.sanger.ac.uk/Software/Rfam/mirna/"
TARGET=_blank>miRNA Registry</A>.
The following publication provides guidelines on miRNA annotation:
Ambros V. et al., 
<A HREF="http://www.rnajournal.org/cgi/content/full/9/3/277" TARGET=_blank>
A uniform system for microRNA annotation</A>.  <em>RNA</em>, <B>9</B>(3), 277-279 (2003).
