<H2>Description</H2>
<P>
The Modern Human Variants track shows variant calls made from sequence
reads of eleven individuals mapped to the $organism genome. The purpose of
this track is to put the divergence of the Denisova genome into
perspective with regard to present-day humans.
</P>

<H2>Methods</H2>
<P>
DNA was obtained for each of ten individuals from the 
<a href="http://www.cephb.fr/en/hgdp_panel.php"
target="_blank">CEPH-Human Genome Diversity Panel (HGDP)</a>:
<UL>
<LI>HGDP00456 (Mbuti)</LI>
<LI>HGDP00521 (French)</LI>
<LI>HGDP00542 (Papuan)</LI>
<LI>HGDP00665 (Sardinian)</LI>
<LI>HGDP00778 (Han)</LI>
<LI>HGDP00927 (Yoruba)</LI>
<LI>HGDP00998 (Karitiana)</LI>
<LI>HGDP01029 (San)</LI>
<LI>HGDP01284 (Mandenka)</LI>
<LI>HGDP01307 (Dai)</LI>
</UL>
DNA was also extracted from a Dinka individual from Sudan (DNK02).
To minimize biases due to instrument variability, the samples were pooled
for sequencing, using four barcoded libraries per sample.
The paired-end reads were aligned to the $organism genome
using the <A HREF="http://bio-bwa.sourceforge.net/"
TARGET=_blank>Burrows-Wheeler Aligner</A> and potential PCR duplicates
were filtered using
<A HREF="http://picard.sourceforge.net/" TARGET=_BLANK>Picard</A>.
</P>
Genotype calls for single nucleotide variants and small insertions and
deletions were made using the Unified Genotyper from the
<A HREF="http://www.broadinstitute.org/gatk/"
TARGET=_BLANK>Genome Analysis Toolkit (GATK)</A>, with an additional iteration
using a modified reference genome in order to reduce reference bias
(Note 6,
<A HREF="http://www.sciencemag.org/content/suppl/2012/08/29/science.1224344.DC1/Meyer.SM.pdf"
TARGET=_BLANK>supplementary online materials</A> of Meyer, 2012).
</P>
<P>
Variant Call Format (VCF) files were enhanced by adding information from
<A HREF="ftp://ftp.ensembl.org/pub/release-64/emf/ensembl-compara/"
TARGET=_BLANK>
Ensembl Compara EPO alignments</A> of 6 primates and of 35 Eutherian mammals,
phastCons conservation scores generated using EPO alignments,
1000 Genomes Project
<A HREF="ftp://ftp.ncbi.nlm.nih.gov/1000genomes/ftp/release/20101123/interim_phase1_release/"
TARGET=_BLANK>integrated variant call files</A>,
University of Washington
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/19424416"
TARGET=_BLANK>background selection scores</A>,
ENCODE/Duke Uniqueness of 20mers (see the Mappability track),
segmental duplications from the Eichler lab (see the Segmental Dups track),
and
<A HREF="http://samtools.sourceforge.net/mpileup.shtml"
TARGET=_BLANK>samtools mpileup</A> summaries of mapped reads.
</P>

<H2>Credits</H2>
<P>
Thanks to the
<A HREF="http://www.eva.mpg.de/"
TARGET=_BLANK>Max Planck Institute for Evolutionary Anthropology</A>
for providing the variant-only VCF files used for this track.
</P>

<H2>References</H2>
<p>
Meyer M, Kircher M, Gansauge MT, Li H, Racimo F, Mallick S, Schraiber JG, Jay F, Pr&#252;fer K, de
Filippo C <em>et al</em>.
<a href="http://www.sciencemag.org/content/338/6104/222.abs" target="_blank">
A high-coverage genome sequence from an archaic Denisovan individual</a>.
<em>Science</em>. 2012 Oct 12;338(6104):222-6.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/22936568" target="_blank">22936568</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3617501/" target="_blank">PMC3617501</a>;
<a href="http://www.sciencemag.org/content/338/6104/222/suppl/DC1" 
target="_blank">supplementary online materials</A>, Note 6.
</p>
