<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=441 HEIGHT=655>
  <TR><TD ALIGN=RIGHT>
    <IMG SRC="/images/denisovaCave.jpg" WIDTH=426 HEIGHT=640
	ALT="Denisova cave">
  </TD></TR>
  <TR><TD ALIGN=RIGHT>
    <FONT SIZE=-1>Denisova cave entrance in the Altai Mountains
	of Siberia, Russia where the bones were found from which
	DNA was sequenced
    </FONT>
    <FONT SIZE=-2> (Copyright (C) 2010, Johannes Krause)
    </FONT>
  </TD></TR>
</TABLE>

<H2>Description</H2>
<P>
The Denisova Variants track shows variant calls made using
high-coverage sequence reads from an archaic Denisovan individual
mapped to the $organism genome reference assembly.
The Denisova DNA was extracted from a phalanx bone excavated from
Denisova Cave in the Altai Mountains in southern Siberia.
</P>

<H2>Methods</H2>
<P>
A novel single-stranded DNA library preparation method (Meyer, 2012)
was applied to DNA previously extracted from 40mg of bone (Reich, 2010).
Using single-stranded DNA greatly increased the genomic coverage to
30X compared to an earlier 1.9X sequence (Reich, 2010).
Sequence reads were aligned to $organism sequence $date
(downloaded from the
<A HREF="http://1000genomes.org"
TARGET=_BLANK>1000 Genomes Project</A>) using the
<A HREF="http://bio-bwa.sourceforge.net/"
TARGET=_blank>Burrows-Wheeler Aligner.</A>
Genotype calls for single nucleotide variants and small insertions and
deletions were made using the Unified Genotyper from the
<A HREF="http://www.broadinstitute.org/gatk/"
TARGET=_BLANK>Genome Analysis Toolkit (GATK)</A>, with an additional iteration
using a modified reference genome in order to reduce reference bias
(Note 6,
<A HREF="http://www.sciencemag.org/content/suppl/2012/08/29/science.1224344.DC1/Meyer.SM.pdf"
TARGET=_BLANK>supplementary online materials</A> of Meyer, 2012).
</P>
<P>
Variant Call Format (VCF) files were enhanced by adding information from
<A HREF="ftp://ftp.ensembl.org/pub/release-64/emf/ensembl-compara/"
TARGET=_BLANK>
Ensembl Compara EPO alignments</A> of 6 primates and of 35 Eutherian mammals,
phastCons conservation scores generated using EPO alignments,
1000 Genomes Project
<A HREF="ftp://ftp.ncbi.nlm.nih.gov/1000genomes/ftp/release/20101123/interim_phase1_release/"
TARGET=_BLANK>integrated variant call files</A>,
University of Washington
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/19424416"
TARGET=_BLANK>background selection scores</A>,
ENCODE/Duke Uniqueness of 20mers (see the Mappability track),
segmental duplications from the Eichler lab (see the Segmental Dups track),
and
<A HREF="http://samtools.sourceforge.net/mpileup.shtml"
TARGET=_BLANK>samtools mpileup</A> summaries of mapped reads.
</P>
<P>
<A HREF="http://cdna.eva.mpg.de/denisova/VCF/hg19_1000g/"
TARGET=_BLANK>Comprehensive VCF files</A> that include information for homozygous-reference
bases and uncovered bases are available.  This track uses VCF files that were filtered
to retain only those locations that clearly differ from the $organism reference genome.
Reference genome bases without variant calls may indicate either matching Denisova sequence
or insufficient data.
</P>

<H2>Credits</H2>
<P>
Thanks to the
<A HREF="http://www.eva.mpg.de/"
TARGET=_BLANK>Max Planck Institute for Evolutionary Anthropology</A>
for providing the variant-only VCF files used for this track.
</P>

<H2>References</H2>
<P>
Meyer M, Kircher M, Gansauge MT, Li H, Racimo F, Mallick S, Schraiber JG, Jay F, Prüfer K, de Filippo C <EM>et al.</EM>
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/22936568"
TARGET=_BLANK>A High-Coverage Genome Sequence from an Archaic Denisovan Individual.</A>
<EM>Science.</EM> 2012 Aug 31. [Epub ahead of print]
<A HREF="http://www.sciencemag.org/content/suppl/2012/08/29/science.1224344.DC1/Meyer.SM.pdf"
TARGET=_BLANK>supplementary online materials</A>, Note 6</P>
<P>
Reich D, Green RE, Kircher M, Krause J, Patterson N, Durand EY, Viola B,
Briggs AW, Stenzel U, Johnson PLF <EM>et al.</EM>
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/21179161"
TARGET=_BLANK>Genetic history of an archaic hominin group from Denisova Cave in Siberia</A>.
<EM>Nature.</EM> 2010 Dec 23;468:1053-1060.</P>
