<h2>Description</h2>
<p>
The tracks that are listed here contain data from unrelated individuals sequenced as part of
various population-genetic and disease-specific <a target="_blank"
href="https://gnomad.broadinstitute.org/about">studies</a> collected by the <a target="_blank"
href="https://gnomad.broadinstitute.org/">Genome Aggregation Database (gnomAD)</a>.
Individuals affected by severe pediatric diseases and first-degree relatives were excluded from the
studies. However, some individuals with severe disease may still have remained in the datasets,
although probably at an equivalent or lower frequency than observed in the general population. Raw
data from all studies have been reprocessed using a standardized pipeline and jointly variant-called
process, which aims to increase consistency between projects. For more information on the processing
pipeline and population annotations, see the following blog post <a target="_blank"
href="https://macarthurlab.org/2017/02/27/the-genome-aggregation-database-gnomad">gnomAD</a>,
 <a target="_blank"href="https://macarthurlab.org/2018/10/17/gnomad-v2-1">gnomAD v2.1</a> 
and the 2.0.2 <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/README.txt"
target="_blank">README</a>.</p>

<p>
The available data tracks are:
<ul>
<li><a href="/cgi-bin/hgTrackUi?db=hg19&g=gnomadGenomes"><strong>Genome Variants 
(gnomAD Genomes)</strong></a> - Shows single nucleotide
variants (SNVs) and small insertion/deletion variants of &lt;50 nucleotides (indels) of 15,708 
unrelated individuals' genome sequences from the v2.1.1 release.</li> 
<li><a href="/cgi-bin/hgTrackUi?db=hg19&g=gnomadExomes"><strong>Exome 
Variants (gnomAD Exomes)</strong></a> - Shows single nucleotide
variants (SNVs) and small insertion/deletion variants of &lt;50 nucleotides (indels) of 125,748 
unrelated individuals' exome sequences from the v2.1.1 release.</li>
<li><a href="/cgi-bin/hgTrackUi?db=hg19&g=gnomadCoverage"><strong>Genome and Exome 
Sample Coverage (gnomAD Coverage)</strong></a> - Shows various read depth variant 
metrics calculated separately for exomes and genomes on a ~10% subset of the v2.0.2 samples.</li>
<li><a href="/cgi-bin/hgTrackUi?db=hg19&g=gnomadPext"><strong>Proportion Expression 
Across Transcript Scores (pext)</a></strong> - Shows exon-level expression for 
53 GTEx tissues.</li>
<li><a href="/cgi-bin/hgTrackUi?db=hg19&g=gnomadStructuralVariants"><strong>Structural 
Variants (SV) (gnomAD Structural Variants)</strong></a> - Shows structural 
variants calls (variants &gt;=50 nucleotides) from the gnomAD v2.1 release 
on 10,847 unrelated genomes.</li>
<li><a href="/cgi-bin/hgTrackUi?db=hg19&g=gnomadPLI"><strong>Predicted Constraint 
Metrics (gnomAD Constraint Metrics)</strong></a> - Contains metrics of 
pathogenicity per-gene as predicted for gnomAD v2.1.1.</li></ul>
<li><a href="/cgi-bin/hgTrackUi?db=hg19&g=gnomadMPc"><strong>Missense Deleteriousness Prediction by Constraint (MPC)</strong></a> - Regions of transcripts (not entire transcripts) that show enrichment of missense mutations</li></ul>
</p>

<h2>Display Conventions</h2>
<p>
These tracks are multi-view composite tracks that contain multiple data types (views). Each view
within a track has separate display controls, as described <a target="_blank"
href="https://genome.ucsc.edu/goldenPath/help/multiView.html">here</a>. Most gnomAD tracks contain
multiple subtracks, corresponding to subsets of data. If a track contains many subtracks, only some
subracks will be displayed by default. The user can select which subtracks are displayed via the
display controls on the track details page.
</p> 

<h2>Data Access</h2>
<p>
The raw data can be explored interactively with the <a target="_blank" href="../cgi-bin/hgTables">
Table Browser</a>, or the <a target="_blank" href="../cgi-bin/hgIntegrator">Data Integrator</a>. For
automated analysis, the data may be queried from our <a target="_blank" 
href="/goldenPath/help/api.html">REST API</a>, and the genome annotations are stored in files that 
can be downloaded from our <a
href="http://hgdownload.soe.ucsc.edu/gbdb/$db/gnomAD/" target="_blank">download server</a>, subject
to the conditions set forth by the gnomAD consortium (see below). Coverage values
for the genome are in <a target="_blank" href="../goldenPath/help/bigWig.html">bigWig</a> files in
the coverage/ subdirectory. Variant VCFs can be found in the vcf/ subdirectory.</p>
<p>
The data can also be found directly from the gnomAD <a target="_blank" 
href="https://gnomad.broadinstitute.org/downloads">downloads page</a>. Please refer to 
our <a href="https://groups.google.com/a/soe.ucsc.edu/forum/#!forum/genome"
target="_blank">mailing list archives</a> for questions, or our <a target="_blank"
href="../FAQ/FAQdownloads.html#download36">Data Access FAQ</a> for more information.</p>

<p>
More information about using and understanding the gnomAD data can be found in the
<a target="_blank" href="https://gnomad.broadinstitute.org/faq">gnomAD FAQ</a> site.
</p>

<h2>Credits</h2>
<p>
Thanks to the <a href="https://gnomad.broadinstitute.org/about" target="_blank">Genome Aggregation
Database Consortium</a> for making these data available. The data are released under the <a
href="https://opendatacommons.org/licenses/odbl/1.0/" target="_blank">ODC Open Database License
(ODbL)</a> as described <a href="https://gnomad.broadinstitute.org/terms" target="_blank">here</a>.
</p>

<h2>References</h2>

<p>
Lek M, Karczewski KJ, Minikel EV, Samocha KE, Banks E, Fennell T, O'Donnell-Luria AH, Ware JS, Hill
AJ, Cummings BB <em>et al</em>.
<a href="https://www.ncbi.nlm.nih.gov/pubmed/27535533" target="_blank">
Analysis of protein-coding genetic variation in 60,706 humans</a>.
<em>Nature</em>. 2016 Aug 18;536(7616):285-91.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27535533" target="_blank">27535533</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5018207/" target="_blank">PMC5018207</a>
</p>

<p>
Karczewski KJ, Francioli LC, Tiao G, Cummings BB, Alf&#246;ldi J, Wang Q, Collins RL, Laricchia KM,
Ganna A, Birnbaum DP <em>et al</em>.
<a href="https://www.ncbi.nlm.nih.gov/pubmed/32461654" target="_blank">
The mutational constraint spectrum quantified from variation in 141,456 humans</a>.
<em>Nature</em>. 2020 May;581(7809):434-443.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32461654" target="_blank">32461654</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7334197/" target="_blank">PMC7334197</a>
</p>

<p>
Collins RL, Brand H, Karczewski KJ, Zhao X, Alf&#246;ldi J, Francioli LC, Khera AV, Lowther C,
Gauthier LD, Wang H <em>et al</em>.
<a href="https://www.ncbi.nlm.nih.gov/pubmed/32461652" target="_blank">
A structural variation reference for medical and population genetics</a>.
<em>Nature</em>. 2020 May;581(7809):444-451.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32461652" target="_blank">32461652</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7334194/" target="_blank">PMC7334194</a>
</p>
<p>
Cummings BB, Karczewski KJ, Kosmicki JA, Seaby EG, Watts NA, Singer-Berk M, Mudge JM, Karjalainen J,
Satterstrom FK, O'Donnell-Luria AH <em>et al</em>.
<a href="https://www.ncbi.nlm.nih.gov/pubmed/32461655" target="_blank">
Transcript expression-aware annotation improves rare variant interpretation</a>.
<em>Nature</em>. 2020 May;581(7809):452-458.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32461655" target="_blank">32461655</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7334198/" target="_blank">PMC7334198</a>
</p>

