<H2>Description</H2>
<P>
This track shows 5' cap analysis gene expression (CAGE) tags and clusters in 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=rnaExtract">RNA extracts</A> 
from different 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=localization">
sub-cellular localizations</A> 
in multiple
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=cellType">cell lines</A>.
A CAGE cluster is a region of overlapping tags with
an assigned value that represents the expression level.
The data in this track were produced as part of the ENCODE 
Transcriptome Project. 

<H2>Display Conventions and Configuration</H2>
<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are 
<A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.</P>

<P>To show only selected subtracks, uncheck the boxes next to the tracks that 
you wish to hide.</P>

<P>This track contains the following views:
<DL>
<DT><I>Plus</I> and <I>Minus</I> <I>Signals</I>
<DD>These views display clusters of overlapping read mappings on the
forward and reverse genomic strands.
<DT><i>Alignments</i></DT>
<DD>The Alignments view shows reads mapped to the genome and indicates where
bases may mismatch. Every mapped read is displayed, i.e. uncollapsed. 
The alignment file follows the standard SAM format of Bowtie output. The custom tag XP can be ignored. See the <a href="http://bowtie-bio.sourceforge.net/manual.shtml#sam-bowtie-output" target=_blank>Bowtie Manual</a> for more information about the SAM Bowtie output (including other tags) and the <a href="http://samtools.sourceforge.net/SAM1.pdf#page=4" target=_blank>SAM Format Specification</a> for more information on the SAM/BAM file format.
 Where mapping quality is not available for this track a score of 255 is used in accordance with the SAM Format Specification. Also, where the sequence quality scores are not available, all scores are displayed as 40.
</dd></dl>

<P>Color differences in subtracks may be set as a visual cue to
distinguish between the different cell types or between annotations
on the plus and minus strand.

<H2>Methods</H2>
<P>
Cells were grown according to the approved 
<A HREF="../../ENCODE/protocols/cell" TARGET=_BLANK>
ENCODE cell culture protocols</A>.
RNA molecules longer than 200 nt 
and present in the RNA population isolated from each subcellular compartment
were fractionated into polyA+ and polyA- fractions as described in 
<A HREF="../../ENCODE/protocols/general/rnaExtracts.txt" TARGET=_BLANK>these 
protocols</A>.
The CAGE tags were sequenced from the 5' ends of cap-trapped cDNAs produced
using RIKEN CAGE technology 
(Kodzius <EM>et al</EM>. 2006; Valen <EM>et al</EM>. 2009). 
To create the tag, a linker was attached to the 5'
end of polyA+ or polyA- reverse-transcribed cDNAs which were selected by cap
trapping (Carninci <EM>et al</EM>. 1996). The first 27 bp of
the cDNA were cleaved using class II restriction enzymes. A linker was then
attached to the 3' end of the cDNA. 
</P>
<P>
After PCR amplification, the tags were sequenced (36 bp single reads) using Illumina's Genome analyzer.
Tags were mapped to the human genome (hg19) using the program delve (T. Lassmann manuscript in preparation).
Delve is a new probabilistic aligner focused on giving the best possible alignment of reads to genome rather
than focusing on speed. It calculates the mapping accuracy (probability of each alignment being true or not)
for each alignment. There is no set limit on the number of errors allowed and therefore the mapping rate is
commonly 100%. However, for analysis it is recommended to discard alignments with low mapping qualities.
An exception to the above are K562 polyA- RNA from the cytosol and nucleus and the prostate cells samples
which were sequenced using ABI <A
HREF="http://www.appliedbiosystems.com/absite/us/en/home/applications-technologies/solid-next-generation-sequencing.html"
TARGET=_blank>SOLiD</A>  technology. These reads were mapped using Bowtie using default parameters.
 Clusters were defined as regions of overlapping CAGE reads. The expression level was computed as the number
 of reads making up the cluster, divided by the total number of reads sequenced, times 1 million.
</P>

<H2>Release Notes</H2>
<P>
Release 1 on hg19 contains 
the original <A TARGET=_BLANK HREF="http://genome.ucsc.edu//cgi-bin/hgTrackUi?db=hg18&g=wgEncodeRikenCage"> data on hg18</A> 
that was remapped and indicated in this release as Generation 0 since that data had no replicates. The new data for this track was done with a different 
process and in some cases replaced the data from hg18. The new data has standard replicate numbers. The replicate labeling in the genome browser view is a
counter indicating the total number of replicates submitted.  The producing lab has replicate numbers that correspond to their internal bio-replicate numbering. 
Where these two numbering systems conflict, both are listed in the long label of the specific track.
</P>

<H2>Credits</H2>
<P>
These data were generated and analyzed by Timo Lassmann, Phil Kapranov, 
Hazuki Takahashi, Yoshihide Hayashizaki, Carrie Davis, Tom Gingeras, and Piero Carninci.
<P>
<P>Contact: <A HREF="mailto:&#99;&#97;&#114;&#110;i&#110;&#99;i&#64;&#114;&#105;k&#101;&#110;.&#106;&#112;">
Piero Carninci</A> at <A TARGET=_BLANK HREF="http://www.osc.riken.jp/english/">
RIKEN Omics Science Center</A>
<!-- above address is carninci at riken.jp -->
</P>

<H2>References</H2>
<P>
Kodzius R, Kojima M, Nishiyori H, Nakamura M, Fukuda S, Tagami M, Sasaki D,
Imamura K, Kai C, Harbers M, <EM>et al</EM>. 
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/pubmed/16489339">
CAGE: cap analysis of gene expression.</A>
<EM>Nat Methods</EM>. 2006 March 1; 3(3):211-222.
</P>

<P>
Valen E, Pascarella G, Chalk A, Maeda N, Kojima M, Kawazu C, Murata M, Nishiyori
H, Lazarevic D, Motti D, <EM>et al</EM>.
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/pubmed/19074369">
Genome-wide detection and analysis of hippocampus core promoters using
DeepCAGE.</A>
<EM>Genome Res</EM>. 2009 February; 19(2):255-265.
</P>
<P>
Carninci P, Kvam C, Kitamura A, Ohsumi T, Okazaki Y, Itoh M, Kamiya M, 
Shibata K, Sasaki N, Izawa M, <EM>et al</EM>.
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/pubmed/8938445">
High-efficiency full-length cDNA cloning by biotinylated CAP trapper.</A>
<EM>Genomics</EM>. 1996 November 1; 37(3):327-336.
</P>

<H2> Data Release Policy </H2>

<P> Data users may freely use ENCODE data, but may not, without prior 
consent, submit publications that use an unpublished ENCODE dataset until 
nine months following the release of the dataset.  This date is listed in 
the <EM>Restricted Until</EM> column, above.  The full data release policy 
for ENCODE is available 
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P> 

