<h2>Description</h2>
<p>
This track shows the percentage of molecules methylated at specific CpG dinucleotides.  In general, methylation of CpG sites within a promoter causes silencing of the gene associated with that promoter.
Reduced Representation Bisulfite Sequencing (RRBS) is used to measure methylation status at more than 500,000 CpG dinucleotides in the genome.  Genomic DNA is digested with the methyl-insensitive restriction enzyme Msp1, small fragments are purified by gel electrophoresis, and they are then used to construct an Illumina sequencing library that is treated with bisulfite and amplified by PCR to convert every unmethylated cytosine to a thymidine.  The sequenced fragments are aligned to a customized reference genome sequence and used to calculate the number of reads at each CpG assayed and the percentage of those reads that detect methylation.
<H2> Release Notes</H2>
This is release 1 of this track.

<h2>Track Conventions</h2>
Methylation status is represented with an 11-color gradient using the following convention:
<ul>
<li>red = 100% of molecules sequenced are methylated</li>
<li>yellow = 50% of molecules sequenced are methylated</li>
<li>green = 0% of molecules sequenced are methylated</li>
</ul>
<a name="Methods"></a><h2>Methods</h2>
CpG sites were assayed with Reduced Representation Bisulfite Sequencing (RRBS; Meissner et al., 2008).  RRBS uses the Msp1 restriction enzyme followed by size selection of small fragments to produce a "reduced" genome.  After Msp1 treatment and size selection, the fragments are treated with sodium bisulfite, which converts unmethylated cytosines to uracils and leaves methylated cytosines unchanged.  Bisulfite-treated fragments are then sequenced on a next-generation DNA sequencing platform (HudsonAlpha uses the Illumina Genome Analyzer IIX).  Sequence reads are aligned to a reference human genome sequence that takes into account bisulfite treatment (see data analysis below).  The fraction of molecules methylated at each CpG site is then directly calculated from the alignment and represents the percentage of methylated molecules in the mixture of cells from which the DNA was obtained.
In an RRBS experiment, a single sequencing lane of approximately 20 million total reads is used to generate data at 500,000 CpGs with greater than 10-fold coverage.  More than 40% of these are located near the transcriptional start sites of genes, and about 70% are located in CpG islands.  Biological replicates of cell lines show 99% concordance on average in methylation status measured by RRBS.

<h3>Isolation of genomic DNA</h3>
Genomic DNA is isolated from biological replicates of each cell line by using the QIAGEN DNeasy Blood & Tissue Kit according to the instructions provided by the manufacturer.  DNA concentrations and a level of quality of each preparation are determined by UV absorbance.  One microgram of DNA is used to make an RRBS library; however, we have also had success in making libraries with 500 ng genomic DNA.

<h3>RRBS library construction and sequencing</h3>
RRBS library construction starts with Msp1 digestion of genomic DNA, which cuts at every CCGG regardless of methylation status.  DNA fragments are then end repaired and an adenosine is added to 3 prime ends.  Next, a methylated version of the illumina paired-end adapters is ligated onto the DNA.  Adapter ligated DNA between 105 and 185 basepairs is purified using gel electrophoresis and subsequently treated with sodium bisulfite, which converts unmethylated cytosines to uracils and leaves methylated cytosines unchanged.  Bisulfite treated DNA is amplified in a final PCR reaction (changing uracils to thymidines).  The sample is now ready for sequencing on the Illumina sequencing platform.  These libraries were sequenced with an Illumina Genome Analyzer IIx according to the manufacturer's recommendations.  The full RRBS protocol can be found 
<A HREF="http://hudsonalpha.org/myers-lab/protocols" TARGET=_BLANK>here</A>.</P> 
<h3>Data analysis</h3>
To analyze the sequence data a reference genome is created that contains only the 36 base pairs downstream of every MspI site and every C in those sequences is changed to T.  A converted sequence read file is then created by changing each C in the original sequence reads to a T.  The converted sequence reads are aligned to the converted reference genome, and only reads that map uniquely to the reference genome are kept.   Once reads are aligned the percent methylation is calculated for each CpG using the original sequence reads.  The percent methylation and number of reads is reported for each CpG.

<a name="Credits"></a><h2>Credits</h2>
<p>
<li>Dr. K-T Varley; 
<A HREF="mailto:&#107;t&#118;&#97;&#114;&#108;&#101;y&#64;&#104;&#117;&#100;&#115;on&#97;&#108;&#112;&#104;&#97;.
&#111;&#114;&#103;">
&#107;t&#118;&#97;&#114;&#108;&#101;y&#64;&#104;&#117;&#100;&#115;on&#97;&#108;&#112;&#104;&#97;.
&#111;&#114;&#103;</A></li>
<!-- above address is ktvarley at hudsonalpha.org -->
<li>Dr. Jason Gertz; 
<A HREF="mailto:&#106;g&#101;&#114;&#116;z&#64;&#104;&#117;d&#115;&#111;&#110;&#97;&#108;p&#104;a.
&#111;&#114;&#103;">
&#106;g&#101;&#114;&#116;z&#64;&#104;&#117;d&#115;&#111;&#110;&#97;&#108;p&#104;a.
&#111;&#114;&#103;</A></li>
<!-- above address is jgertz at hudsonalpha.org -->
<li>Dr. Richard M. Myers; 
<A HREF="mailto:&#114;m&#121;&#101;&#114;s&#64;&#104;&#117;d&#115;&#111;&#110;&#97;&#108;p&#104;a.
&#111;&#114;&#103;">
&#114;m&#121;&#101;&#114;s&#64;&#104;&#117;d&#115;&#111;&#110;&#97;&#108;p&#104;a.
&#111;&#114;&#103;</A></ul>
<!-- above address is rmyers at hudsonalpha.org -->

<p>
Contact:
<A HREF="mailto:&#114;r&#97;&#117;&#99;h&#64;&#115;&#116;a&#110;&#102;o&#114;&#100;.
e&#100;&#117;">Rami Rauch</A>
<!-- above address is rrauch at stanford.edu -->
<br />
</p>
<a name="References"></a><h2>References</h2>
<p>
1. Meissner A, Gnirke A, Bell GW, Ramsahoye B, Lander ES, Jaenisch R. Reduced representation bisulfite sequencing for comparative high-resolution DNA methylation analysis. Nucleic Acids Res. 2005 Oct 13;33(18):5868-77.

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior 
consent, submit publications that use an unpublished ENCODE dataset until 
nine months following the release of the dataset.  This date is listed in 
the <EM>Restricted Until</EM> column, above.  The full data release policy 
for ENCODE is available 
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P> 


