<H2>Description</H2>

<P>This track is produced as part of the ENCODE Project.  
Transcription factors (TF) are hypothesized to be knocked down through the interference of short hairpin RNA (shRNA).  The derivative sublines may show 
differentially expressed genes where TF binding sites are proximal.  
This increases evidence of potential genomic targets for a TF.
</P>

<P>K562, a non-adherent human erythromyeloblastoid leukemia cell line, 
was transduced with lentiviral vectors carrying an inducible shRNA 
that targets a specific TF to reduce its expression.  
Cells with stable integration of shRNA constructs were selected with puromycin.
Doxycycline was used to induce shRNA, 
which reduced the expression (measured by QPCR) of the targeted gene 
by at least 70% relative to cells not treated with doxycycline. </P>

<P>The cell lines are named shX_K562, where X is the TF targeted 
by shRNA and K562 denotes the parent cell line.  For example, 
shATF3_K562 cells are K562 derived cells selected for stable 
integration of shRNA targeting the TF ATF3 gene and showed 
at least a 70% reduction in the expression of ATF3 gene.</P> 

<P>RNA-seq (Mortazavi <EM>et al.</EM>, 2008) was used to quantify 
the transcriptome of these cell lines both for the doxycycline treatment 
with induced shRNA and for no doxycycline.  The cDNA was sequenced on an 
Illumina Genome Analyzer (GAI or GAIIx).  Biological replicates 
were performed for each experiment.  </P>  

<H2>Display Conventions and Configuration</H2>

<P>This is a composite track that contains multiple data types 
(<EM>views</EM>). For each view, there are multiple subtracks 
(cell lines, replicates and growth conditions) that display 
individually on the browser. Instructions for configuring multi-view 
tracks are <A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.  The following views are in this track: </DL>

<DT><I>FPKM</I>
<DD>Gencode.v3c (Harrow <EM>et al.</EM>, 2006) gene models are shaded with a <A HREF="http://genome.ucsc.edu/FAQ/FAQformat.html#format1" TARGET=_BLANK>gray scale</A> according to a score calculated from the FPKM value (Fragments Per Kilobase of exon per Million reads, Cufflinks v0.9.3, Roberts <EM>et al.</EM>, 2011).  Score is equal to 100 x log<sub>2</sub>(FPKM +1) and is capped at 1000.  The gray scale becomes darker as score and FPKM increase, thereby assisting in visualizing the relative amount of a given transcript across multiple samples. 
<P>
<DT><I>Alignments</I>
<DD>The Alignments view shows reads mapped to the genome. 
</DD>
<P>
<DT><I>Raw Signal</I>
<DD>Density graph of signal enrichment based on a normalized aligned read density (Read Per Million, RPM).  RPM is reported in the score field and is equal to the number of reads at that position divided by the total number of reads divided by one million.  The RawSignal view displays dense, continuous data as a graph and the RPM measure assists in visualizing the relative amount of a given transcript across multiple samples.  Raw Signals are colored by cell type.</DD>

<H2>Methods</H2>
<P>
<H3>Experimental Procedures</H3>

<P>Cells were grown according to the approved <A HREF="../../ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture protocols</A>.  Messenger RNA was isolated, reverse transcribed to cDNA and sequenced according to the protocol in Mortazavi <EM>et al.</EM> (2008).   The Genome Analyzer flowcell was used according to the protocol for the ChIPSeq DNA genomic DNA kit (Illumina).  The sequencing libraries were size-selected around 225 bp and amplified with 15 rounds of PCR.  Libraries were sequenced with an Illumina Genome Analyzer IIx according to the manufacturer's recommendations. Single end reads of 36 nt in length were obtained. 
</P>

<H3>Data Processing and Analysis</H3>

<P>Fastq files were made from qseq files generated by the Illumina pipeline 
(Casava 1.7).  The Raw Signal files (bigWig) were generated from 
bedgraph files and the score was calculated as the number of reads 
at that position divided by the total number of reads divided by one million.  
Casava export files were aligned to the NCBI Build37 (hg19) version of the 
human genome with ELAND (Illumina), generating SAM files.  SAM files 
were converted to BAM with SAMtools (Li <EM>et al.</EM>, 2009).
The first 10 residues of sequencing have a weak characteristic 
nucleotide bias of unknown origin. This RNA-seq protocol does 
not specify the coding strand.  As a result, there will be 
ambiguity at loci where both strands are transcribed.</P>

<P>RNA-seq reads were aligned to Gencode.v3c (Harrow <EM>et al.</EM>, 2006) 
gene models and gene expression was measured in Fragments Per Kilobase exon 
per Million reads (FPKM) using Cufflinks v0.9.3 (Roberts <EM>et al.</EM>, 
2011).  FPKM is calculated by dividing the total number of fragments 
that align to the gene model by the size of the spliced transcript (exons) 
in kilobases.  This number is then divided by the total number of reads 
in millions for the experiment.  FPKM is reported in the last column 
of the gtf (TranscriptGencV3c) files.</P>

<P>RawData (fastq), RawSignal (bigWig), Alignments (bam) and TranscriptGencV3c (gtf) files are available from the <A HREF="http://hgwdev.cse.ucsc.edu/cgi-bin/hgFileUi?g=wgEncodeHaibRnaSeq" TARGET=_BLANK>Downloads</A> page.</P>   


<H2>Verification</H2>
<UL>
<LI>Known exon maps as displayed on the genome browser are confirmed by the alignment of sequence reads.</LI>
<LI>Biological replicates confirm expression measurements with r > 0.90
</UL>

<H2>Credits</H2>

<P>These data were produced by the <A TARGET=_BLANK HREF="http://hudsonalpha.org/myers-lab"> Dr. Richard Myers Lab</A>  at the <A TARGET=_BLANK HREF="http://hudsonalpha.org/"> Hudson Alpha Institute for Biotechnology</A>. 
</P>

<P>Contact: <A HREF="mailto:&#102;p&#97;&#117;&#108;&#105;&#64;&#104;&#117;&#100;&#115;&#111;&#110;&#97;&#108;&#112;&#104;a.
&#111;r&#103;">Dr. Florencia Pauli</A>.
<!-- above address is fpauli at hudsonalpha.org (encodeEmail.pl) -->
</P>

<H2>References</H2>

<P>Harrow J, Denoeud F, Frankish A, Reymond A, Chen CK, Chrast J, Lagarde J, Gilbert JG, Storey R, Swarbreck D, Rossier C, Ucla C, Hubbard T, Antonarakis SE and Guigo R.
<A TARGET=_BLANK HREF="http://www.gencodegenes.org/">GENCODE: producing a reference annotation for ENCODE</A>
<EM>Genome Biology</EM>. 2006; 7 Suppl 1;S4.1-9</P>

<P>Li H, Handsaker B, Wysoker A, Fennell T, Ruan J, Homer N, Marth G, Abecasis G, Durbin R and 1000 Genome Project Data Processing Subgroup. <A TARGET=_BLANK HREF="http://bioinformatics.oxfordjournals.org/content/25/16/2078.long"><U>The Sequence alignment/map (SAM) format and SAMtools</U></A> <EM>Bioinformatics</EM>. 2009; 25:2078-9.</P>

<P>Mortazavi A, Williams BA, McCue K, Schaeffer L, and Wold BJ. 
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/pubmed/18516045">Mapping and 
quantifying mammalian transcriptomes by RNA-Seq</A>
<EM>Nature Methods</EM>. 2008 Jul; 5(7):621-628.</P> 

<P> Roberts A, Trapnell C, Donaghey J, Rinn JL, Patcher L.
<A TARGET=_BLANK HREF="http://genomebiology.com/2011/12/3/R22/abstract"><U>Improving RNA-Seq expression estimates by correcting for fragment bias </U></A><EM>Genome Biology</EM>. 2011 Mar; 12:R22.</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column, above.  The full data release policy
for ENCODE is available
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>

