<H2> Description </H2>
<P>This track is produced as part of the ENCODE project.  The tracks displays copy number variation (CNV) as determined by the Illumina Human 1M-Duo Infinium HD BeadChip assay and circular binary segmentation (CBS).  The Human 1M-Duo contains more than 1,100,000 tagSNP markers and a set of &#126;60,000 additional CNV-targeted markers.  The median spacing between markers is 1.5 kb and the mean spacing is 2.4 kb.  The B-allele frequency and genotyping single nucleotide polymorphism (SNP) data generated by the experiment are not displayed, but are available for download from the Downloads page.  
</P>

<P>Where applicable, biological replicates of each cell line are reported separately.  Possible uses of the data include correction of copy number in peak-calling for ChIP-seq, transcriptome, DNase hypersensitivity, and methylation determinations.
</P>

<H2>Display Conventions and Configuration</H2>
<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>). For each view, there are multiple subtracks that display
individually on the browser. Instructions for configuring multi-view tracks
are <A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
</P>
<P> The track displays regions of the genome where copy number variation has been assessed.  CNV regions are colored by type: </P>

<UL>
<LI> <span style="color:#0000FF">blue</span>  = amplified 
<LI> <span style="color:#000000">black</span> = normal
<LI> <span style="color:#FF9933">orange</span>  = heterozygous deletion 
<LI> <span style="color:#FF0000">red</span>  = homozygous deletion
</UL>

<P> The mean log R ratio for each region can be seen by clicking on each individual region. See Methods below for significance of log R ratio values.   The mean log R ratio for each region is reported in the .bed file available for download.
</P>
<P>The Illumina 1M-Duo B-allele frequency data is available from the <A TARGET=_BLANK HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeHaibGenotype/supplemental/">Supplemental Materials</A> directory on the <A TARGET=_BLANK HREF="/cgi-bin/hgFileUi?g=wgEncodeHaibGenotype ">Downloads</A> page.  The file (<A TARGET=_BLANK HREF=" http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeHaibGenotype/supplemental/wgEncodeHaibGenotypeBalleleSnp.txt">wgEncodeHaibGenotypeBalleleSnp.txt</A>) was generated using the standard Illumina protocol and contains the B-allele frequency for all cell types tested.  The genotype calls for all cell types tested are also available for download (<A TARGET=_BLANK HREF=" http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeHaibGenotype/supplemental/wgEncodeHaibGenotypeGtypeSnp.txt">wgEncodeHaibGenotypeGtypeSnp.txt</A>).  Genotyping calls with a Gencall value greater than 0.6 are considered significant.  
</P>

<H3>Replicate Numbering</H3>
<P>
The replicate labeling in the genome browser view is a
counter indicating the total number of replicates submitted (UCSC Rep). 
 The producing lab has replicate numbers (Lab Rep) that correspond to their internal bio-replicate numbering.
Where these two numbering systems conflict, both are listed in the long label of the specific track.
When comparing data across tracks, the lab replicate number should be considered. In the downloads directory both replicate numbers are listed.  The files are labeled with the lab replicate number.
</P>


<H2>Methods</H2>

<H3> Isolation of genomic DNA and hybridization</H3>
<P> Cells were grown according to the approved <A TARGET=_BLANK HREF="http://genome.ucsc.edu/ENCODE/cellTypes.html"> ENCODE cell culture protocols</A> by the Myers lab and by other ENCODE production groups.  The production group is reported in the metadata.  Genomic DNA was isolated using the DNeasy Blood and Tissue Kit (Qiagen).  DNA concentration and quality were determined by fluorescence (Invitrogen Quant-iT dsDNA High Sensitivity Kit and Qubit Fluorometer), and 400 nanograms of each sample were hybridized to Illumina 1M-Duo DNA Analysis BeadChips.
</P>

<H3> Processing and Analysis </H3>
<P>
The genotypes from the 1M-Duo Arrays were ascertained with BeadStudio by using default settings and formatting with the A/B genotype designation for each SNP. Primary QC for each sample was a cut-off at a call rate of 0.95.
</P>
<P>
Copy Number Variation (CNV) analysis was performed with circular binary segmentation (DNAcopy) of the log R ratio values at each probe (Olshen et al., 2004). The parameters used were alpha=0.001, nperm=5000, sd.undo=1. The copy number segments are reported with the mean log R ratio for each chromosomal segment called by CBS. Log ratios of ~-0.2 to -1.5 can be considered heterozygous deletions, &lt; -1.5 homozygous deletions, and &gt; 0.2 amplifications. Primary QC for each sample was SD of &lt; 0.6.
</P>

<H2> Release Notes </H2>
<P>This is the NCBI Build37 (hg19) release of this track.  This release includes the 3 cell types previously released on NCBI Build36 (hg18) which were lifted to NCBI Build37 (hg19) and adds data for many more cell types.  The track includes a single display for each cell type and reports the Log ratio in the .bed files.  The B-allele frequency and SNP genotyping files are not displayed, but are available for download for the entire dataset from the downloads page.
</P>


<H2>Credits</H2>
<P>These data were produced by the <A TARGET=_BLANK HREF="http://hudsonalpha.org/myers-lab"> Dr. Richard Myers Lab</A> and the Dr. Devin Absher lab at the <A TARGET=_BLANK HREF="http://hudsonalpha.org/"> HudsonAlpha Institute for Biotechnology</A>.  
<p>Cells were grown by the Myers Lab and other ENCODE production groups. 
</P>
Contact:
<A HREF="mailto:&#102;p&#97;&#117;&#108;&#105;&#64;&#104;&#117;&#100;&#115;&#111;&#110;&#97;&#108;&#112;&#104;a.
&#111;r&#103;">Dr. Florencia Pauli</A>.
<!-- above address is fpauli at hudsonalpha.org (encodeEmail.pl) -->


<H2> References </H2>
<P>Olshen, A. et al. (2004).  
<A HREF="http://biostatistics.oxfordjournals.org/content/5/4/557.long"
	class="external test"
	title="http://biostatistics.oxfordjournals.org/content/5/4/557.long"
	rel="nofollow" TARGET = _BLACK>
Circular binary segmentation for the analysis of array-based DNA copy number data</A>. 
<em>Biostatistics</em> (Oxford, England) vol. 5 (4) pp. 557-572.

</P>

<H2> Data Release Policy </H2>
<P>Data users may freely use ENCODE data, but may not, without prior consent, submit publications that use an unpublished ENCODE dataset until nine months following the release of the dataset. This date is listed in the <EM> Restricted Until</EM> column on the track configuration page and the download page. The full data release policy for ENCODE is available <A TARGET=_BLANK HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.
</P>
