<H2>Description</H2>
<P>
These tracks display the level of sequence uniqueness of the reference <B>GRCh37/hg19</B>
genome assembly.  They were generated using different window sizes, and high signal
will be found in areas where the sequence is unique.

<H2> Display Conventions and Configuration </H2>
<P>This track is a multi-view composite track that contains multiple data types
separated as separate (views). For each view, there are
multiple subtracks representing different sequence lengths or methods of preparation.
Instructions for configuring multi-view tracks are
<A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
<B>Mapability</B> tracks consist of the following views:
<DL>
<DT><I>Alignability</I></DT>
<DD>These tracks provide a measure of how often the sequence found at the particular
location will align with the whole genome.  Unlike measures of uniqueness, alignability
will tolerate up to 2 mismatches. These tracks are in the form of signals ranging from
0 to 1 and have several configuration options.</DD>
</P>
<DT><I>Uniqueness</I></DT>
<DD>These tracks are a direct measure of sequence uniqueness throughout the reference
genome. These tracks are in the form of signals ranging from 0 to 1 and have several
configuration options.</DD>
</P>
<P><DT><I>Blacklisted Regions</I></DT>
<DD>Both tracks of blacklisted regions attempt to identify regions of the reference
genome which are troublesome for high throughput sequencing aligners.  Troubled
regions may be due to repetitive elements or other anomalies.  Each track contains a
set of regions of varying length with no special configuration options.</DD>
</P>

<H2>Methods</H2>
<H4>Alignability</H4>
The <em>CRG Alignability</em> tracks display how uniquely k-mer sequences align
to a region of the genome. To generate the data, the GEM-mappability
<A HREF="http://gemlibrary.sourceforge.net" TARGET=_BLANK>program</A> has
been employed. The method is equivalent to mapping sliding windows of k-mers
(where k has been set to 36, 40, 50, 75 or 100 nts to produce these tracks)
back to the genome using the GEM mapper aligner (up to 2 mismatches were
allowed in this case). For each window, a mapability score was computed
(S = 1/(number of matches found in the genome): S=1 means one match in the
genome, S=0.5 is two matches in the genome, and so on). The
<em>CRG Alignability</em> tracks were
generated independently of the ENCODE project, in the framework of the GEM
(GEnome Multitool) project.

<H4>Uniqueness</H4>
The <em>Duke uniqueness</em> tracks display how unique is each sequence on the
positive strand starting at a particular base and of a particular length.
Thus, the 20 bp track reflects the uniqueness of all 20 base sequences with
the score being assigned to the first base of the sequence. Scores are
normalized to between 0 and 1 with 1 representing a completely unique
sequence and 0 representing the sequence occurs &#62; 4 times in the genome
(excluding chrN_random and alternative haplotypes).  A score of 0.5
indicates the sequence occurs exactly twice, likewise 0.33 for three times
and 0.25 for four times. The <em>Duke uniqueness</em> tracks were generated
for the ENCODE project as tools in the development of the
<A HREF="hgTrackUi?g=wgEncodeDNAseSuper">Open Chromatin</A>:
<A HREF="hgTrackUi?g=wgEncodeOpenChromDnase">DNaseI HS</A>,
<A HREF="hgTrackUi?g=wgEncodeOpenChromFaire">FAIRE</A>,
<A HREF="hgTrackUi?g=wgEncodeOpenChromDnase">TFBS</A> and
<A HREF="hgTrackUi?g=wgEncodeOpenChromDnase">Synthesis</A></B> tracks.

<H4>Blacklisted Regions</H4>
The <em>DAC Blacklist regions</em> aim to identify a comprehensive set of
regions in the human genome that have anomalous, unstructured, high
signal/read counts in next gen sequencing experiments independent of
cell line and type of experiment. We use 80 open chromatin tracks
(DNase and FAIRE datasets) and 20 ChIP-seq input/control tracks spanning
&#126;60 human tissue types/cell lines in total to identify these regions
with signal artifacts. These regions tend to have a very high ratio of
multi-mapping to unique mapping reads and high variance in mapability.
Some of these regions overlap pathological repeat elements such as satellite,
centromeric and telomeric repeats. However, simple mapability based filters
do not account for most of these regions. Hence, it is recommended to use this
blacklist alongside mapability filters.  The
<em>DAC Blacklist regions</em> track was generated for the ENCODE project.
<!-- Complete methods can be found
<A HREF="ftp://encodeftp.cse.ucsc.edu/users/akundaje/rawdata/blacklists/hg19/hg19-blacklist-README.pdf">here</a>.
-->

<P>
The <em>Duke excluded regions</em> track displays genomic regions for
which mapped sequence tags were filtered out before signal generation
and peak calling for
<A HREF="hgTrackUi?g=wgEncodeDNAseSuper">Open Chromatin</A>:
<A HREF="hgTrackUi?g=wgEncodeOpenChromDnase">DNaseI HS</A> and
<A HREF="hgTrackUi?g=wgEncodeOpenChromFaire">FAIRE</A> tracks.
This track contains problematic regions for short sequence tag signal
detection (such as satellites and rRNA genes). The
<em>Duke excluded regions</em> track was generated for the ENCODE project.
</P>

<h2>Release Notes</h2>
<P>
This is Release 3 (July 2011) of this track, which now includes the DAC
Blacklisted regions, Duke Uniqueness and Duke Excluded regions.

<H2>Credits</H2>
<P>
The <em>CRG Alignability</em> track was created by Thomas Derrien and <A HREF="mailto:&#112;&#97;&#111;l&#111;.
&#114;&#105;&#98;&#101;&#99;&#97;&#64;&#103;&#109;&#97;&#105;l.
&#99;o&#109;">Paolo
Ribeca</A>
<!-- above address is paolo.ribeca at gmail.com -->
 in <A HREF="http://pasteur.crg.es/portal/page/portal/Internet/02_Research/01_Programmes/01_Bioinformatics_Genomics"
TARGET=_BLANK>Roderic Guigo's lab</A> at the Centre for Genomic
Regulation (CRG), Barcelona, Spain. Thomas Derrien was supported by funds from NHGRI
for the ENCODE project, while Paolo Ribeca was funded by a Consolider grant
CDS2007-00050 from the Spanish Ministerio de Educaci&#243;n y Ciencia.
</P>

<P>
The <em>Duke uniqueness</em> and <em>Duke excluded regions</em> tracks were created
by
<A HREF="mailto:&#116;&#115;&#102;ur&#101;&#121;&#64;&#100;&#117;&#107;&#101;.e&#100;u"
title="&#116;&#115;&#102;ur&#101;&#121;&#64;&#100;&#117;&#107;&#101;.e&#100;u" rel="nofollow" TARGET=_BLANK>Terry Furey</A>
<!-- above address is tsfurey at duke.edu -->
and Debbie Winter at Duke Univerisity's
<A HREF="http://www.genome.duke.edu/index.php" TARGET=_blank>Institute for Genome Sciences & Policy (IGSP)</A>;
and Stefan Graf at the
<A HREF="http://www.ebi.ac.uk/" TARGET=_blank>European Bioinformatics Insitute (EBI)</A>.
We thank NHGRI for ENCODE funding support.
</P>

<P>
The <em>DAC Blacklisted regions</em> were created by
<A HREF="http://sites.google.com/site/anshulkundaje/" TARGET=_BLANK>Anshul Kundaje</A>
at Stanford University in the labs of
<A HREF="http://nature.stanford.edu/" TARGET=_BLANK>Batzoglou</A> and
<A HREF="http://mendel.stanford.edu/SidowLab/" TARGET=_BLANK>Sidow</A>
and in cooperation with
<A HREF="http://www.ebi.ac.uk/~birney/" TARGET=_BLANK>Ewan Birney</A>
at the
<A HREF="http://www.ebi.ac.uk/" TARGET=_blank>European Bioinformatics Insitute (EBI)</A>.
We thank NHGRI for ENCODE funding support.  (Contact:
<A HREF="mailto:&#97;k&#117;&#110;d&#97;&#106;&#101;&#64;&#115;&#116;&#97;n&#102;or&#100;.&#101;&#100;&#117;"
title="&#97;k&#117;&#110;d&#97;&#106;&#101;&#64;&#115;&#116;&#97;n&#102;or&#100;.&#101;&#100;&#117;" rel="nofollow" TARGET=_BLANK>Anshul Kundaje</A>).
</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use all data in this track.  ENCODE labs that
contributed annotations have exempted the data displayed here from the
ENCODE data release policy restrictions.
