<H2>Description</H2>
<P>These tracks display DNaseI hypersensitivity (HS) evidence as part of the
four <a href="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeDNAseSuper&hgTracksConfigPage=configure">Open Chromatin</a> track sets.
DNaseI is an enzyme that has long been used to map general
chromatin accessibility and DNaseI "hypersensitivity" is a feature of active
<i>cis</i>-regulatory sequences. The use of this method has led to the discovery of
functional regulatory elements that include promoters, enhancers, silencers,
insulators, locus control regions, and novel elements. DNaseI hypersensitivity
signifies chromatin accessibility following binding of <i>trans</i>-acting factors in
place of a canonical nucleosome.
</P>
<P>Together with FAIRE and
ChIP-seq experiments, these tracks display the locations of active regulatory
elements identified as open chromatin in
<A TARGET=ucscVocab HREF="/cgi-bin/hgEncodeVocab?type=cellType">multiple cell types</A>
from the Duke, UNC-Chapel Hill, UT-Austin, and EBI ENCODE group.
Within this project, open chromatin was identified using two
independent and complementary methods: these DNaseI HS assays
and Formaldehyde-Assisted Isolation of Regulatory Elements (FAIRE),
combined with chromatin immunoprecipitation (ChIP) for select
regulatory factors. DNaseI HS and FAIRE provide assay
cross-validation with commonly identified regions delineating the
highest confidence areas of open chromatin.  ChIP assays provide
functional validation and preliminary annotation of a subset of
open chromatin sites. Each method employed Illumina (formerly Solexa)
sequencing by synthesis as the detection platform.
The Tier 1 and Tier 2 cell types were additionally verified using
high-resolution 1% ENCODE tiled microarrays supplied by NimbleGen.
</P>
<P>Other <B><em>Open Chromatin</em></B> track sets:
<UL>
<LI>Data for the FAIRE experiments can be found in
<A HREF="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromFaire">UNC FAIRE</A>.
<LI>Data for the ChIP experiments can be found in
<A HREF="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromChip">UTA TFBS</A>.
<LI>A synthesis of all the open chromatin assays for select cell lines can
be previewed in
<A HREF="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromSynth">Open Chrom Synth</a>.
</UL>

<H2> Display Conventions and Configuration </H2>
<P>This track is a multi-view composite track that contains a single data type
with multiple levels of annotation (views). For each view, there are
multiple subtracks representing different cell types that display individually
on the browser. Instructions for configuring multi-view tracks are
<A HREF="/goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
Chromatin data displayed here represents a continuum of signal intensities.
The Crawford lab recommends setting the &quot;Data view scaling: <EM>auto-scale</EM>&quot;
option when viewing signal data in full mode to see the <EM>full</EM> dynamic
range of the data. Note that in regions that do not have open chromatin sites,
autoscale will rescale the data and inflate the background signal, making the
regions appear noisy.  Changing back to fixed scale will alleviate this issue.
In general, for each experiment in each of the cell types, the
<B>Duke DNaseI HS</B> tracks contain the following views:
<DL>
<DT><I>Peaks</I></DT>
<DD>Regions of enriched signal in DNaseI HS experiments.
Peaks were called based on signals created using F-Seq, a software program
developed at Duke (Boyle <em>et al.</em>, 2008b). Significant regions were
determined by fitting the data to a gamma distribution to calculate p-values.
Contiguous regions where p-values were below a 0.05/0.01 threshold were
considered significant.  The solid vertical line in the peak represents the
point with the highest signal.</DD>
</P>
<P><DT><I>F-Seq Density Signal</I></DT>
<DD>Density graph (wiggle) of signal
enrichment calculated using F-Seq for the combined set of sequences from all
replicates. F-Seq employs Parzen kernel density estimation to create base pair
scores (Boyle <em>et al.</em>, 2008b). This method does not look at fixed-length
windows, but rather weights contributions of nearby sequences in proportion to
their distance from that base.  It only considers sequences aligned 4 or less
times in the genome and uses an alignability background model to try to correct
for regions where sequences cannot be aligned. For each cell type (especially
important for those with an abnormal karyotype), a model to try to correct for
amplifications and deletions that is based on control input data was also used.</DD>
</P>
<P><DT><I>Base Overlap Signal</I></DT>
<DD>An alternative version of the
<I>F-Seq Density Signal</I> track annotation that provides a higher resolution
view of the raw sequence data. This track also includes the combined set of
sequences from all replicates.  For each sequence, the aligned read is
extended 5 bp in both directions from its 5' aligned end where DNase cut
the DNA.  The score at each base pair represents the number of
extended fragments that overlap the base pair.</DD>
</DL>
</P>
<P>
Peaks and signals displayed in this track are the results of pooled replicates. The raw
sequence and alignment files for each replicate are available for
<A HREF="/cgi-bin/hgFileUi?g=wgEncodeOpenChromDnase">download</A>.
</P>
<p>Metadata for a particular subtrack can be found by clicking the down arrow in the list of subtracks.</p>


<H2>Methods</H2>
<P>Cells were grown according to the approved
<A HREF="/ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture protocols</A>.
</P>
<P><em>DNaseI hypersensitive sites</em> were isolated using methods called DNase-seq or
DNase-chip (Song and Crawford, 2010; Boyle <em>et al.</em>, 2008a; Crawford <em>et al.</em>, 2006).
Briefly, cells were lysed with NP40, and intact nuclei were digested with optimal
levels of DNaseI enzyme. DNaseI-digested ends were captured from three different
DNase concentrations, and material was sequenced using Illumina (Solexa)
sequencing. DNase-seq data for Tier 1 and Tier 2 cell lines were verified by comparing
multiple independent growths (replicates) and determining the reproducibility of the
data. In general, cell lines were verified if 80% of the top 50,000 peaks in
one replicate were detected in the top 100,000 peaks of a second replicate. For
some cell types, additional verification was performed using similar material
hybridized to NimbleGen Human ENCODE tiling arrays (1% of the genome) along with
the input DNA as reference (DNase-chip). A more detailed protocol is available
<A HREF="/ENCODE/protocols/general/Duke_DNase_protocol.pdf" target="_blank">here</A>.
</P>
<P>The read length for sequences from DNase-seq was 20 bases long due to a MmeI
cutting step of the approximately 50 kb DNA fragments extracted after DNaseI
digestion. Sequences from each experiment were aligned to the genome using
BWA (Li <em>et al.</em>, 2008) for the GRCh37 (hg19) assembly.
<DIV style='position:relative; left:25px;'><DL><DT>The command used for these alignments was:</DT>
<DD><CODE>&gt; bwa aln -t 8 genome.fa s_1.sequence.txt.bfq &gt; s_1.sequence.txt.sai</CODE></DD></DL>
where genome.fa is the whole genome sequence and s_1.sequence.txt.bfq is one lane
of sequences converted into the required bfq format.</DIV>
</P>
<P>Sequences from multiple lanes
were combined for a single replicate using the bwa samse command and converted
to the sam/bam format using SAMtools.
</P>
<P>Only those sequences that aligned to 4 or fewer locations were retained. Other sequences
were also filtered based on their alignment to problematic regions
(such as satellites and rRNA genes - see
<A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeOpenChromDnase/supplemental/"
 TARGET=ucscDownloads>supplemental materials</a>).
The mappings of these short reads to the genome are available for
<A HREF="/cgi-bin/hgFileUi?g=wgEncodeOpenChromDnase">download</A>.</P>
<P> Using F-seq, the resulting digital signal was converted to a continuous wiggle track that employs a Parzen kernel density estimation to create base pair scores
(Boyle <em>et al.</em>, 2008b).  Input data was generated for several
cell lines. These were used directly to create a control/background model used
for F-Seq when generating signal annotations for these cell lines.
These models were meant to correct for sequencing biases, alignment artifacts,
and copy number changes in these cell lines. Input data was not generated
directly for other cell lines. For cell lines for which there is 
no input experiment available, the peaks were generated using the control
of generic_male or generic_female, as an attempt to create a general
background based on input data from several cell types. These files
are in &quot;iff&quot; format, which is used when calling peaks with
F-seq software, and can be downloaded from the
<a href="http://fureylab.web.unc.edu/software/fseq/" target="_blank">production lab directly</a>
from under the section titled &quot;Copy number / karyotype correction.&quot;
Using a general background model derived from the available Input data sets provided corrections for
sequencing biases and alignment artifacts, but did not correct for cell type-specific copy number changes.
<DIV style='position:relative; left:25px;'><DL><DT>The exact command used for this step was:</DT>
<DD><CODE>&gt; fseq -l 600 -v -f 0 -b &lt;bff files&gt; -p &lt;iff files&gt; aligments.bed</CODE></DD></DL>
where the bff files are the background files based on alignability, the
iff files are the background files based on the Input experiments,
and alignments.bed is a bed file of filtered sequence alignments.</DIV>
</P>
<P>Discrete DNaseI HS sites (peaks) were identified from DNase-seq F-seq density signal.
Significant regions were determined by fitting the data to a gamma distribution to
calculate p-values. Contiguous regions where p-values were below a 0.05/0.01
threshold were considered significant.
</P>
<P>Data from the high-resolution 1% ENCODE tiled microarrays supplied by
NimbleGen were normalized using the Tukey biweight normalization and peaks
were called using ChIPOTle (Buck <em>et al.</em>, 2005) at multiple levels
of significance. Regions matched on size to these peaks that were devoid of
any significant signal were also created as a null model.  These data were used
for additional verification of Tier 1 and Tier 2 cell lines by ROC analysis.
Files containing this data can be found in the 
<a href="/cgi-bin/hgFileUi?db=hg19&g=wgEncodeOpenChromDnase" >Downloads</a> 
directory, labeled 'Validation' in the View column.
</P>



<h2>Release Notes</h2>
<P>
This is Release 3 (August 2012) of the track. It includes 27 new experiments including 18 new cell lines. 

</P>
<P>
<UL>
<li>A synthesis of open chromatin evidence from the three assay types was
compiled for Tier 1 and 2 cell lines and can be viewed in
<a href="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromSynth">Open Chromatin Synthesis</a>.
</li>
<li><B>Enhancer and Insulator Functional assays:</B>  A subset of DNase and FAIRE
regions were cloned into functional tissue culture reporter assays to test for
enhancer and insulator activity.  Coordinates and results from these
experiments can be found in the
<A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeOpenChromDnase/supplemental/"
 TARGET=ucscDownloads>supplemental materials</a>.  <!-- WARNING: Change to hgdownload.cse.ucsc.edu on release -->
</li>
</UL>
</P>

<H2>Credits</H2>
<P>
These data and annotations were created by a collaboration of multiple
institutions (contact:
<A HREF="mailto:&#116;s&#102;&#117;&#114;e&#121;&#64;&#101;m&#97;&#105;&#108;.
&#117;&#110;c.
&#101;&#100;u">
Terry Furey</A>)
<!-- above address is tsfurey at email.unc.edu -->

</P><P>
<UL>
<LI>Duke University's <A HREF="http://www.genome.duke.edu/index.php"
TARGET=_blank>Institute for Genome Sciences & Policy (IGSP)</A>: Alan Boyle,
Lingyun Song, and
<A HREF="http://www.genome.duke.edu/directory/faculty/crawford/"
TARGET=_blank>Greg Crawford</a>
</P><P>
<LI>University of North Carolina at Chapel Hill: Paul Giresi,
<A HREF="http://lieblab.bio.unc.edu/"
TARGET=_blank>Jason Lieb</a>, and <A HREF="http://genetics.unc.edu/faculty/terry-furey"
TARGET=_blank>Terry Furey</A>
</P><P>
<LI>Universty of Texas at Austin: Zheng Liu, Ryan McDaniell, Bum-Kyu Lee, and
<A HREF="http://microarray.icmb.utexas.edu/research.html"
TARGET=_blank>Vishy Iyer</a>
</P><P>
<LI><A HREF="http://www.ebi.ac.uk/" TARGET=_blank>European Bioinformatics
Insitute</A>: Paul Flicek, Damian Keefe, and
<A HREF="http://www.ebi.ac.uk/~birney/" TARGET=_blank>Ewan Birney</a>
</P><P>
<LI><A HREF="http://www.oncology.cam.ac.uk/" TARGET=_blank>University of
Cambridge, Department of Oncology and CR-UK Cambridge Research Institute (CRI)</A>: Stefan Graf
</UL>
</P><P>
We thank NHGRI for ENCODE funding support.
</P>

<H2>References</H2>

<p>
Bhinge AA, Kim J, Euskirchen GM, Snyder M, Iyer VR.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/17568006" target="_blank">
Mapping the chromosomal targets of STAT1 by Sequence Tag Analysis of Genomic Enrichment (STAGE)</a>.
<em>Genome Res</em>. 2007 Jun;17(6):910-6.
</p>

<p>
Boyle AP, Davis S, Shulha HP, Meltzer P, Margulies EH, Weng Z, Furey TS, Crawford GE.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/18243105" target="_blank">
High-resolution mapping and characterization of open chromatin across the genome</a>.
<em>Cell</em>. 2008 Jan 25;132(2):311-22.
</p>

<p>
Boyle AP, Guinney J, Crawford GE, Furey TS.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/18784119" target="_blank">
F-Seq: a feature density estimator for high-throughput sequence tags</a>.
<em>Bioinformatics</em>. 2008 Nov 1;24(21):2537-8.
</p>

<p>
Buck MJ, Nobel AB, Lieb JD.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/16277752" target="_blank">
ChIPOTle: a user-friendly tool for the analysis of ChIP-chip data</a>.
<em>Genome Biol</em>. 2005;6(11):R97.
</p>

<p>
Crawford GE, Davis S, Scacheri PC, Renaud G, Halawi MJ, Erdos MR, Green R, Meltzer PS, Wolfsberg TG, Collins FS.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/16791207" target="_blank">
DNase-chip: a high-resolution method to identify DNase I hypersensitive sites using tiled microarrays</a>.
<em>Nat Methods</em>. 2006 Jul;3(7):503-9.
</p>

<p>
Crawford GE, Holt IE, Whittle J, Webb BD, Tai D, Davis S, Margulies EH, Chen Y, Bernat JA, Ginsburg D <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/16344561" target="_blank">
Genome-wide mapping of DNase hypersensitive sites using massively parallel signature sequencing (MPSS)</a>.
<em>Genome Res</em>. 2006 Jan;16(1):123-31.
</p>

<p>
ENCODE Project Consortium, Birney E, Stamatoyannopoulos JA, Dutta A, Guig&#243; R, Gingeras TR, Margulies EH, Weng Z, Snyder M, Dermitzakis ET <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/17571346" target="_blank">
Identification and analysis of functional elements in 1% of the human genome by the ENCODE pilot project</a>.
<em>Nature</em>. 2007 Jun 14;447(7146):799-816.
</p>

<p>
Giresi PG, Kim J, McDaniell RM, Iyer VR, Lieb JD.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/17179217" target="_blank">
FAIRE (Formaldehyde-Assisted Isolation of Regulatory Elements) isolates active regulatory elements from human chromatin</a>.
<em>Genome Res</em>. 2007 Jun;17(6):877-85.
</p>

<p>
Giresi PG, Lieb JD.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/19303047" target="_blank">
Isolation of active regulatory elements from eukaryotic chromatin using FAIRE (Formaldehyde Assisted Isolation of Regulatory Elements)</a>.
<em>Methods</em>. 2009 Jul;48(3):233-9.
</p>

<p>
Li H, Ruan J, Durbin R.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/18714091" target="_blank">
Mapping short DNA sequencing reads and calling variants using mapping quality scores</a>.
<em>Genome Res</em>. 2008 Nov;18(11):1851-8.
</p>

<p>
Song L, Crawford GE.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/20150147" target="_blank">
DNase-seq: a high-resolution technique for mapping active gene regulatory elements across the genome from mammalian cells</a>.
<em>Cold Spring Harb Protoc</em>. 2010 Feb;2010(2):pdb.prot5384.
</p>

<H2>Data Release Policy</H2>
Data users may freely use ENCODE data, but may not, without prior consent, submit publications 
that use an unpublished ENCODE dataset until nine months following the release of the dataset. 
This date is listed in the <em>Restricted Until</em> column, above. The full data release policy for ENCODE is available
<A HREF="../ENCODE/terms.html"
TARGET=_BLANK>here</A>.
