<h1>Overview</h1>
<p>
This set of tracks represents multivariate genome-segmentation results based on ENCODE data 
(ENCODE Project Consortium, 2012). 
Using two different unsupervised machine learning techniques (ChromHMM and Segway), the genome
was automatically segmented into disjoint segments. 
Each segment belongs to one of a few specific genomic "states" which is assigned an intuitive label.
Each genomic state represents a particular combination and distribution of different ENCODE
functional data tracks such as histone modifications, open chromatin data and specific TF
binding data. 
A consensus unified segmentation was also generated by reconciling results from the 
individual segmentations. 
The specific descriptions for each segmentation are listed below.</p>
<p>
These segmentations were performed on
<a target=ucscVocab href="http://genome.ucsc.edu/cgi-bin/hgEncodeVocab?term=GM12878,H1-hESC,K562,HeLa-S3,HepG2,HUVEC">six human cell types</a>
(GM12878, K562, H1-hESC, HeLa-S3, HepG2, and HUVEC), integrating ChIP-seq data for 
<a target=ucscVocab 
href="http://genome.ucsc.edu/cgi-bin/hgEncodeVocab?term=CTCF,H3K4me1,H3K4me2,H3K4me3,H3K27ac,H3K9ac,H3K36me3,H4K20me1,H3K27me3,Pol2,Input">
8 chromatin marks, RNA Polymerase II, the CTCF transcription factor, and input data</a>.
In total, twenty-five states were used to segment the genome, and these states were then 
grouped and colored to highlight predicted functional elements.
</p>

<h2> Display Conventions and Configuration </h2>
<p>
The number and type of Segmentation states from the individual segmentations differ, 
but are unified via grouping by color (10 groups for ChromHMM and Segway, 7 for the Combined).
The display can be filtered to selected groups using the 'Filter by Segment Type' control on the
track configuration page.  Groupings that are not represented in the Combined tracks are marked
in the menu with an asterisk.</p>

<hr>
<h1>Combined Segmentations</h1>

<h2>Description</h2>

<p>These tracks display chromatin state segmentations from 6 cell lines, using a consensus 
merge of the segmentations produced by the ChromHMM and Segway software.  
In both segmentations, twenty-five states were used to segment the genome, however
for ease of comprehension and display, the merged segmentation uses only seven states.</>

<h2> Display Conventions and Configuration </h2>
<p>
The seven states of the combined segmentation, the candidate annotations and associated segment colors are as follows:</p>
<table>
<tr>
<td>TSS </td><td bgcolor='#FF0000'>Bright Red</td><td> Predicted promoter region including TSS</td>
</tr>
<tr>
<td>PF </td><td bgcolor='#FF6969'>Light Red</td><td> Predicted promoter flanking region</td>
</tr>
<tr>
<td>E  </td><td bgcolor='#FACA00'>Orange</td><td> Predicted enhancer</td>
</tr>
<tr>
<td>WE </td><td bgcolor='#FFFC04'>Yellow</td><td> Predicted weak enhancer or open chromatin cis regulatory element</td>
</tr>
<tr>
<td>CTCF</td><td bgcolor='#0ABEFE'>Blue</td><td>  CTCF enriched element</td>
</tr>
<tr>
<td>T  </td><td bgcolor='#00B050'>Dark Green</td><td> Predicted transcribed region</td>
</tr>
<tr>
<td>R  </td><td bgcolor='#7F7F7F'>Gray</td><td> Predicted Repressed or Low Activity region</td>
</tr></table>

<h2>Methods</h2>
<p>ChIP-seq data from the ENCODE Consortium was used to generate this track, and the ChromHMM 
and Segway programs were used to perform the segmentation.  
Methods for the ChromHMM and Segway segmentations are described below.</p>
<p>
To form the combined segmentation, for each original segmentation, states that could be grouped
together based on similar signal patterns were identified.
For the ChromHMM segmentation, the states were grouped manually based on the mean signal values 
across multiple cell lines. 
For the Segway segmentations run independently over multiple cell lines, multiple hierarchical 
clustering techniques were applied across all states in the segmentations to identify the most 
consistent clustering of states, both across cell lines and with respect to existing 
biological knowledge.  
Using these criteria, the Ward clustering on euclidean distances between mean signal scores 
transformed to the unit interval was chosen to cluster the Segway state labels. 
Subsequently, pairwise relationships between the ChromHMM and Segway merged states were 
identified using both overlap calculations and manual annotation (Hoffman, Ernst et al. 2013). 
Pairs of states that were viewed as concordant were assigned to one of the seven state classes. 
Regions of the genome occupied by concordant states between the two initial segmentations were 
reassigned to the new summary labels. 
In some cases there were combinations of states between the two segmentations that could not 
be reconciled and these combinations were viewed as discordant. 
Regions with discordant states were not assigned a state label, and were dropped from
the summary combined segmentation.</p>
<hr>

<h1>ChromHMM Segmentations</h1>

<h2>Description</h2>
<p>
A common set of states across 6 human cell types were learned by computationally integrating ENCODE
ChIP-seq, DNase-seq, and FAIRE-seq data using a Hidden Markov Model (HMM).
Twenty-five states were used to segment the genome, and these states were then 
grouped and colored to highlight predicted functional elements.
There are 6 ChromHMM tracks. Each track represents the segmentation results for each of the 
six cell lines.</p>
<p>
A related ChromHMM browser track, 
<a href="../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeBroadHmm" target="_blank">
Chromatin State Segmentation by HMM from ENCODE/Broad (Broad ChromHMM)</a>
(Ernst et. al. 2011) reports segmentations for 9 cell types and is based solely on histone data.

<h2> Display Conventions and Configuration </h2>
<p>
The candidate annotations and associated segment colors are as follows:</p>
<table>
<tr><td>Tss, TssF</td><td bgcolor='#FF0000'>Bright Red</td><td>Active Promoter</td></tr>
<tr><td>PromF</td><td bgcolor='#FF6969'>Light Red</td><td>Promoter Flanking</td></tr>
<tr><td>PromP</td><td bgcolor='#CF0BC6'>Purple</td><td>Inactive Promoter</td></tr>
<tr><td>Enh, EnhF</td><td bgcolor='#FACA00'>Orange</td><td>Candidate Strong enhancer</td></tr>
<tr><td>EnhWF, EnhW, DNaseU, DNaseD, FaireW</td><td bgcolor='#FFFC04'>Yellow</td><td>Candidate Weak enhancer/DNase</td></tr>
<tr><td>CtrcfO, Ctcf</td><td bgcolor='#0ABEFE'>Blue</td><td>Distal CTCF/Candidate Insulator</td></tr>
<tr><td>Gen5', Elon, ElonW, Gen3', Pol2, H4K20</td><td bgcolor='#00B050'>Dark Green</td><td>Transcription associated</td></tr>
<tr><td>Low</td><td bgcolor='#99FF66'>Light Green</td><td> Low activity proximal to active states</td></tr>
<tr><td>ReprD, Repr, ReprW</td><td bgcolor='#7F7F7F'>Gray</td><td>Polycob repressed</td></tr>
<tr><td>Quies, Art</td><td bgcolor='#F5F5F5'>Light Gray</td><td>Heterochromatin/Repetitive/Copy Number Variation</td></tr>
</table>

<h2>Methods</h2>
<p>
Data from the ENCODE Consortium was used to generate this track, and the ChromHMM 
program was used to perform the segmentation. 
Datasets for 10 factors plus input in 6 cell types were binarized separately at a 200 base pair 
resolution using a Poisson background model and fold enrichment cut-offs. 
The chromatin states were learned from this binarized data using a multivariate Hidden Markov 
Model (HMM) that explicitly models the combinatorial patterns of observed modifications 
(Ernst and Kellis, 2010).
To learn a common set of states across the six cell types, first the genomes were concatenated
across the cell types. For each of the six cell types, each 200 base pair interval was then 
assigned to its most likely state under the model. </p>
<hr>

<h1>Segway Segmentations</h1>

<h2>Description</h2>
<p>
Sets of states across 6 human cell types were learned by computationally integrating
ENCODE ChIP-seq, DNAse-seq and FAIRE-seq data using a Dynamic Bayesian Network (DBN). 
Twenty-five states were used to segment the genome (listed below in the Display Conventions
and Configuration section by their prefixes - such as PromP for PromP1, PromP2, etc.),
and these states were then grouped and colored to highlight predicted functional elements
(such as the color purple for an inactive promoter region). There are 6 Segway tracks, each
representing the segmentation results for a separate cell line.  Not every
segmentation state is found in each cell line. If you have further questions about the
tracks, please contact the authors listed under the Credits section. 
</p>

<h2> Display Conventions and Configuration </h2>
<p>
The segment state prefixes, associated colors, and candidate annotations are:</p>
<table>
<tr><td>Tss, DnaseD</td><td bgcolor='#FF0000'>Bright Red</td><td>Active Promoter</td></tr>
<tr><td>TssF, PromF</td><td bgcolor='#FF6969'>Light Red</td><td>Promoter Flanking</td></tr>
<tr><td>PromP</td><td bgcolor='#CF0BC6'>Purple</td><td>Inactive Promoter</td></tr>
<tr><td>Enh, EnhF, EnhPr, EnhP</td><td bgcolor='#FACA00'>Orange</td><td>Candidate Strong enhancer</td></tr>
<tr><td>EnhW, EnhWf</td><td bgcolor='#FFFC04'>Yellow</td><td>Candidate Weak enhancer</td></tr>
<tr><td>Ctcf, CtcfO</td><td bgcolor='#0ABEFE'>Blue</td><td>Distal CTCF/Candidate Insulator</td></tr>
<tr><td>Gen3', Gen5', Elon, ElonW</td><td bgcolor='#00B050'>Dark Green</td><td>Transcription associated</td></tr>
<tr><td>Low</td><td bgcolor='#99FF66'>Light Green</td><td>Low activity proximal to active states</td></tr>
<tr><td>Repr</td><td bgcolor='#7F7F7F'>Gray</td><td>Polycomb repressed</td></tr>
<tr><td>Quiesc</td><td bgcolor='#F5F5F5'>Light Gray</td><td>Heterochromatin/Repetitive/Copy Number Variation</td></tr>
</table>

<h2>Methods</h2>
<p>
Data from the ENCODE Consortium was used to generate this track, and the Segway program 
was used to perform the segmentation. 
Data for 10 factors plus input in 6 cell types
was converted to real valued signal data using the Wiggler program. 
Using the ENCODE regions (spanning 1% of the human genome) the chromatin states were 
learned from this data using a Dynamic Bayesian Network (DBN) (Hoffman, et al. 2012). 
Models were learned separately for each of the six cell types. For each cell type, 
the Viterbi algorithm was used to assign genomic regions to individual state labels at 
single base pair resolution over the entire genome.</p>

<hr>
<A NAME="TRACK_CREDITS"></A>
<h1> Credits </h1>
<p>
The <b>ChromHMM</b> segmentation was produced at the 
<a href="http://compbio.mit.edu" target=_blank>
MIT Computational Biology Group (Kellis lab)</a> by
<A HREF="mailto:&#106;&#97;&#115;&#111;&#110;.
e&#114;&#110;s&#116;&#64;&#117;&#99;l&#97;.
&#101;&#100;u">
<!-- above address is jason.ernst at ucla.edu -->
Jason Ernst</a> now at 
<a target=_blank href="http://biolchem.ucla.edu/labs/ernst/">UCLA</a>.</p>
<p>
The <b>Segway</b> segmentation was produced at the 
<a href="http://noble.gs.washington.edu/" target=_blank>
Noble Research Lab</a> by
<A HREF="mailto:&#109;&#104;&#111;&#102;f&#109;a&#110;&#64;&#117;&#104;&#110;&#114;&#101;&#115;&#101;a&#114;&#99;&#104;.
c&#97;">
<!-- above address is mhoffman at uhnresearch.ca -->
Michael Hoffman</a>, now at the 
<a target=_blank href="http://pmgenomics.ca/hoffmanlab/">
Princess Margaret Cancer Center</a>, Toronto.
<p>
The <b>Combined</b> segmentation was produced at the European Bioinformatics Institute (EMBL-EBI,
Flicek team), by
<A HREF="mailto:&#119;i&#108;&#100;e&#114;&#64;&#101;&#98;&#105;.
&#97;&#99;.
&#117;&#107;">
<!-- above address is wilder at ebi.ac.uk -->
Steven Wilder</a> and Ian Dunham,
as part of the work of the ENCODE Data Analysis Center (Ewan Birney).</p>
<h1>References</h1>
<p>
ENCODE Project Consortium.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/22955616" target="_blank">
An integrated encyclopedia of DNA elements in the human genome</a>.
<em>Nature</em>. 2012 Sep 6;489(7414):57-74.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/22955616" target="_blank">22955616</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3439153" target="_blank">PMC3439153</a>
</p>

<p>
Ernst J, Kellis M.
<a href="http://dx.doi.org/10.1038/nmeth.1906" target="_blank">
ChromHMM: automating chromatin-state discovery and characterization</a>.
<em>Nat Methods</em>. 2012 Feb 28;9(3):215-6.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/22373907" target="_blank">22373907</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3577932" target="_blank">PMC3577932</a>
</p>
</p>

<p>
Ernst J, Kellis M.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/20657582" target="_blank">
Discovery and characterization of chromatin states for systematic annotation of the human
genome</a>.
<em>Nat Biotechnol</em>. 2010 Aug;28(8):817-25.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/20657582" target="_blank">20657582</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2919626" target="_blank">PMC2919626</a>
</p>

<p>
Hoffman MM, Buske OJ, Wang J, Weng Z, Bilmes JA, Noble WS.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/22426492" target="_blank">
Unsupervised pattern discovery in human chromatin structure through genomic segmentation</a>.
<em>Nat Methods</em>. 2012 Mar 18;9(5):473-6.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/22426492" target="_blank">22426492</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3340533" target="_blank">PMC3340533</a>
</p>

<p>
Hoffman MM, Ernst J, Wilder SP, Kundaje A, Harris RS, Libbrecht M, Giardine B, Ellenbogen PM, Bilmes
JA, Birney E <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/23221638" target="_blank">
Integrative annotation of chromatin elements from ENCODE data</a>.
<em>Nucleic Acids Res</em>. 2013 Jan;41(2):827-41.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/23221638" target="_blank">23221638</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3553955" target="_blank">PMC3553955</a>
</p>

<h1> Data Release Policy </h1>
<p>
The data used to generate these segmentations are covered by the ENCODE data release policy
<a href="http://genome.ucsc.edu/ENCODE/terms.html" target=_blank>here</a>, and
so were subject to some usage restrictions for a 9 month period.
There are no restrictions on the use of the ENCODE segmentation data.
</p>
