<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"><html><head><META http-equiv="Content-Type" content="text/html; charset=utf-8"></head><body><h2>Description</h2>
<p>This track is produced as part of the <a href="http://encodeproject.org/ENCODE/" target="_blank">
Encyclopedia of DNA Elements (ENCODE) Project</a>
and displays 3&#39; UTR regions associated with RNA binding proteins in the 
<a href="http://genome.ucsc.edu/cgi-bin/hgEncodeVocab?ra=encode%2Fcv.ra&term=%22HT%2D1080%22" target="_blank">HT-1080</a>
cell line using <a href="http://switchgeargenomics.com" target="_blank"> reporters</a>.
</p>


<p>In eukaryotic organisms gene regulatory networks require an additional
level of coordination that links transcriptional and post-transcriptional
processes. Messenger RNAs have traditionally been viewed as passive
molecules in the pathway from transcription to translation. However,
it is now clear that RNA-binding proteins play a major role in
regulating multiple mRNAs in order to facilitate gene expression patterns.
These tracks show RNA binding protein associated (predicted via RIP-chip) mRNA 3&#39; UTRs
whose attachment to a reporter gene led to a change of that reporter&#39;s
expression when the associated RBP&#39;s level was altered. This experiment was proposed as
a validation for the data found in <a href="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeSunyalbanyRnaGeneChip">SUNY RBP</a>
which is the precursor to the hg19 track <a href="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSunyAlbanyGeneSt">SUNY GeneST</a>.
It was done on a small set of targets chosen from the earliest RIP arrays. 
</p>


<h2>Display Conventions and Configuration</h2>
<p>This is a composite track containing multiple subtracks that display individually 
in the browser. The subtracks within this track correspond to different
 mRNA 3&#39; UTRs attached to a reporter and tested for a response to a change in a particular
RNA binding protein&#39;s level.
To display only selected subtracks, uncheck the boxes next to the tracks you wish to hide.
<!--</p>
Instructions for configuring multi-view tracks are
<a href="../../goldenPath/help/multiView.html" target="_blank">here</a>.
-->

<h2>Methods</h2>
<p>
IGF2BP1 targets were identified by RIP-chip and cross referenced with Switchgear&#39;s 3&#39; UTR luciferase reporter clone inventory.
50 ng of putative IGF2BP1 target 3&#39;UTR reporter DNA was individually co-transfected with either 20 ng IGF2BP1(IMP1) cDNA (Origene, SC116030)
or alone (untreated). Each transfection was performed in triplicate. The DNA was combined with FuGene transfection reagent (Roche)
and allowed to complex for 30 minutes before being added to 7500 HT-1080 (ATCC) cells resuspended in warmed, complete media before
being aliquotted into 96 well plates. Cells were returned to the incubator for 24 hours before 100 uL SteadyGlo Luciferase assay reagent
was added to each well. The plate was incubated for 30 minutes before being read on a LmaxII-384 luminometer. Signals were normalized using
controls and significance of treatment versus control was calculated via a two-tailed t-test.
</p>
<p>
ELAVL1 targets were identified by RIP-chip and cross referenced with SwitchGear&#39;s 3&#39; UTR luciferase reporter clone inventory.
We seeded 5,000 HT-1080 cells into 96-well plates the day before transfection and incubated overnight. 50 ng of putative ELAVL1 target 3&#39;UTR
reporter DNA was individually co-transfected with either 10ng ELAVL1(HuR) cDNA ( Origene, SC119271) or alone (untreated). Each transfection
was performed in triplicate. The DNA was combined with FuGene transfection reagent (Roche) and allowed to complex for 30 minutes before being
added to 7500 HT-1080 (ATCC) cells resuspended in warmed, complete media before being aliquotted into 96 well plates. Cells were returned to
the incubator for 24 hours before 100 uL SteadyGlo Luciferase assay reagent was added to each well. The plate was incubated for 30 minutes before
being read on a LmaxII-384 luminometer. Signals were normalized using controls and significance of treatment versus control was calculated via a
two-tailed t-test.
</p>



<p>For additional Switchgear 3&#39; UTR reporter assay information, see
<a href="http://switchgeargenomics.com/products/utr-reporter-collection/" target="_blank"> SwitchGear Genomics</a>.


<h2>Verification</h2>

<p>
All experiments (including controls) performed in and analyzed as triplicates.
</p>

<h2>Credits</h2>

<p>These data were produced and analyzed by a collaboration between the
<A HREF="http://www.albany.edu/cancergenomics/faculty/stenenbaum/index.shtml"
TARGET="_BLANK">
Tenenbaum lab</A>
at the University at Albany-SUNY, College of Nanoscale
Science and Engineering, and
<a href="http://switchgeargenomics.com" target="_blank">
SwitchGear Genomics</a>.

<P>
Contact:
<A HREF="mailto:&#83;&#84;&#101;&#110;&#101;&#110;&#98;&#97;um&#64;&#117;&#97;m&#97;&#105;&#108;.
&#97;&#108;&#98;&#97;ny.
&#101;&#100;&#117;">
Scott Tenenbaum</A>
<!-- above address is STenenbaum at uamail.albany.edu -->
<br />

<h2>References</h2>
<h3>SwitchGear Reporter Related:</h3>
<p><a href="http://switchgeargenomics.com/resources/publications/" target="_blank">
SwitchGear Publications
</a>
</p>


<h3>RIP Related:</h3>


<p>Baroni TE, Chittur SV, George AD, Tenenbaum SA.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/18369977" target="_blank">
Advances in RIP-chip analysis : RNA-binding protein
immunoprecipitation-microarray profiling</a>.
<em>Methods Mol Biol</em>. 2008;419:93-108.
</p>

<p>George AD, Tenenbaum SA.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/17114943" target="_blank">
MicroRNA modulation of RNA-binding protein regulatory elements</a>.
<em>RNA Biol</em>. 2006;3(2):57-9. Epub 2006 Apr 1.
</p>

<p>Keene JD, Tenenbaum SA.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/12086614" target="_blank">
Eukaryotic mRNPs may represent posttranscriptional operons</a>.
<em>Mol Cell</em>. 2002;9(6):1161-7.
</p>

<p>Penalva LO, Tenenbaum SA, Keene JD.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/14770002" target="_blank">
Gene expression analysis of messenger RNP complexes</a>.
<em>Methods Mol Biol</em>. 2004;257:125-34.
</p>

<p>Tenenbaum SA, Lager PJ, Carson CC, Keene JD.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/12054896" target="_blank">
Ribonomics: identifying mRNA subsets in mRNP complexes using
antibodies to RNA-binding proteins and genomic arrays</a>.
<em>Methods</em>. 2002 Feb;26(2):191-8.
</p>





<h2> Data Release Policy </h2>

<p>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <em>Restricted Until</em> column, above.  The full data release policy
for ENCODE is available
<a href="http://genome.ucsc.edu/ENCODE/terms.html" target="_blank">here</a>.</p>

</p></body></html>
