<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <title>wgEncodeGencodeSuper</title>
</head>
<body>
</P>
<H2>Description</H2>
<P>
The aim of the <a href="http://www.gencodegenes.org/" target="_blank">GENCODE 
Genes project</a> (Harrow <em>et al.</em>, 2006) is to produce a set of 
highly accurate annotations of evidence-based gene features on the human reference genome.
This includes the identification of all protein-coding loci with associated
alternative splice variants, non-coding with transcript evidence in the public 
databases (NCBI/EMBL/DDBJ) and pseudogenes. A high quality set of gene
structures is necessary for many research studies such as comparative or 
evolutionary analyses, or for experimental design and interpretation of the 
results.</p>
<p> 
The GENCODE Genes tracks display the high-quality manual annotations merged 
with evidence-based automated annotations across the entire
human genome. The GENCODE gene set presents a full merge
between HAVANA manual annotation and Ensembl automatic annotation.
Priority is given to the manually curated HAVANA annotation using predicted
Ensembl annotations when there are no corresponding manual annotations. With 
each release, there is an increase in the number of annotations that have undergone
manual curation.  
This annotation was carried out on the GRCh37 (hg19) genome assembly.
</p>
<p>Experimental verification details are given in each descriptions for each
track. Transcript Support Levels were determined for version 10 onwards based 
on evidence provided by GenBank mRNA and EST sequences. Versions 7 and 10 are
being used in data analysis by the ENCODE consortium.</p>

NOTE</B>: Due to the UCSC Genome Browser using the NC_001807 mitochondrial 
genome sequence
(chrM) and GENCODE annotating the NC_012920 mitochondrial sequence, the
GENCODE mitochondrial sequences are not available in the UCSC Genome Browser.
These annotations are available for download in the 
<a HREF="http://www.gencodegenes.org/releases/"TARGET="_BLANK"> GENCODE GTF files</a>.
</p>

<h2>Display Conventions</h2>
<P>
These are multi-view composite tracks that contain differing data sets
(<EM>views</EM>).  Instructions for configuring multi-view tracks are
<A HREF="/goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
Only some subtracks are shown by default. The user can select which subtracks
are displayed via the display controls on the track details pages.
Further details on display conventions and data interpretation are available in the track descriptions.</p>

<h2>Release Notes</h2>
<p>
<span style="font-weight: bold;">GENCODE version 19 corresponds to Ensembl 74 and Vega 54.</p> <span style="font-weight: bold;">GENCODE version 17</span>
corresponds to Ensembl 72 and Vega 52 and is the most current release.
<br>
<br>
<span style="font-weight: bold;">GENCODE version 14</span>
corresponds to Ensembl 69 and Vega 49
<br>
<br>
<span style="font-weight: bold;">GENCODE version 7</span>
corresponds to Ensembl 62 and Vega 42 
and is used in ENCODE analysis.
</p>
<p>
See also: <a href="http://www.gencodegenes.org/releases" target="_blank">The GENCODE Project Release History</a>.
</p>

<h2>Credits</h2>
<p> This GENCODE release is the result of a collaborative effort among
the following laboratories: (contact: <A HREF="mailto:&#103;&#101;n&#99;o&#100;&#101;&#64;&#115;an&#103;&#101;&#114;.&#97;&#99;.u&#107;">
GENCODE at the Sanger Institute</A>)
<!-- above address is gencode at sanger.ac.uk -->
</p>

<table style="text-align: left; width: 100%;" border="1" cellpadding="2" cellspacing="2">
  <tbody>
    <tr>
      <td style="vertical-align: top; text-align: left;"><b>Lab/Institution</b></td>
      <td style="vertical-align: top; text-align: left;"><b>Contributors</b></td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;"><a href="http://www.sanger.ac.uk/research/faculty/thubbard/" target="_blank">GENCODE Principal Investigator</a></td>
      <td style="vertical-align: top;">Tim Hubbard</td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;"><a href="http://www.sanger.ac.uk/research/projects/vertebrategenome/havana/" target="_blank">HAVANA manual annotation group</a>, Wellcome Trust Sanger Insitute (WTSI), Hinxton, UK</td>
      <td style="vertical-align: top;">Jennifer Harrow, Stephen Searle, Timothy Cutts, Bronwen Aken, James Gilbert, Jyoti Choudhary, Ed Griffiths, Ewan Birney, Jose Manuel Gonzalez, Electra Tapanari, Daniel Barrell, Adam Frankish, Andrew Berry, Alexandra Bignell, Veronika Boychenko, Claire Davidson, Gloria Despacio-Reyes, Mike Kay, Deepa Manthravadi, Gaurab Mukherjee, Catherine Snow, Gemma Barson, Matt Hardy, Joanne Howes</td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;"><a href="http://big.crg.cat/computational_biology_of_rna_processing" target="_blank">Centre de Regulaci&oacute; Gen&ograve;mica (CRG)</a>, Barcelona, Spain</td>
      <td style="vertical-align: top;">Roderic Guig&oacute;, Julien Lagarde, Barbara Uszczy&#x1102A;ska</td>
    </tr>
    <tr>
      <td><a href="http://genome.ucsc.edu/" target="_blank">Genome Bioinformatics, University of California Santa Cruz</a> (UCSC), USA</td>
      <td style="vertical-align: top;">David Haussler, Rachel Harte, Mark Diekhans, Benedict Paten, Joel Armstrong</td>
    </tr>
    <tr>
      <td><a href="http://www.csail.mit.edu/" target="_blank">Computer Science and Artificial Intelligence Lab</a>,<a href="http://www.broadinstitute.org/" target="_blank">Broad Institute of MIT and Harvard</a>, USA</td>
      <td style="vertical-align: top;">Manolis Kellis, Irwin Jungreis</td>
    </tr>
    <tr>
      <td><a href="http://medicine.yale.edu/bbs/computational/index.aspx" target="_blank">Computational Biology and Bioinformatics, Yale University</a> (Yale), USA</td>
      <td>Mark Gerstein, Suganthi Balasubramanian, Ekta Khurana, Cristina Sisu, Baikang Pei, Yan Zhang, Mihali Felipe</td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;"><a href="http://www.unil.ch/cig" target="_blank">Center for Integrative Genomics</a>,University of Lausanne, Switzerland</td>
      <td style="vertical-align: top;">Alexandre Reymond, Cedric Howald, Anne-Maud Ferreira, Jacqueline Chrast</td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;"><a href="http://www.cnio.es/ing/grupos/plantillas/presentacion.asp?pag=726" target="_blank">Structural Computational Biology Group</a>, Centro Nacional de Investigaciones Oncologicas (CNIO), Madrid, Spain</td>
      <td style="vertical-align: top;">Alfonso Valencia, Michael Tress, Jos&eacute; Manuel Rodr&iacute;guez, Victor de la Torre</td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;">Former members of the GENCODE project</td>
      <td style="vertical-align: top;">Felix Kokocinski, Toby Hunt, Gary Saunders, Sarah Grubb, Thomas Derrien, Andrea Tanzer, Gang Fang, Mihali Felipe, Michael Brent, Randall Brown, Jeltje van Baren</td>
    </tr>
  </tbody>
</table>

<P>
<b>Contact:</b> <A HREF="mailto:&#103;&#101;n&#99;o&#100;&#101;&#64;&#115;an&#103;&#101;&#114;.&#97;&#99;.u&#107;">
GENCODE at the Sanger Institute</A>
<!-- above address is gencode at sanger.ac.uk -->
</p>


<h2>References</h2>

<p>
Coffey AJ, Kokocinski F, Calafato MS, Scott CE, Palta P, Drury E, Joyce CJ, Leproust EM, Harrow J, Hunt S <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/21364695" target="_blank">
The GENCODE exome: sequencing the complete human exome</a>.
<em>Eur J Hum Genet</em>. 2011 Jul;19(7):827-31.
</p>

<p>
Flicek P, Amode MR, Barrell D, Beal K, Brent S, Carvalho-Silva D, Clapham P, Coates G, Fairley S, Fitzgerald S <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/22086963" target="_blank">
Ensembl 2012</a>.
<em>Nucleic Acids Res</em>. 2012 Jan;40(Database issue):D84-90.
</p>

<p>
Harrow J, Denoeud F, Frankish A, Reymond A, Chen CK, Chrast J, Lagarde J, Gilbert JG, Storey R, Swarbreck D <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/16925838" target="_blank">
GENCODE: producing a reference annotation for ENCODE</a>.
<em>Genome Biol</em>. 2006;7 Suppl 1:S4.1-9.
</p>


<h2>Publications</h2>

<p>
Djebali S, Lagarde J, Kapranov P, Lacroix V, Borel C, Mudge JM, Howald C, Foissac S, Ucla C, Chrast J <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/22238572" target="_blank">
Evidence for transcript networks composed of chimeric RNAs in human cells</a>.
<em>PLoS One</em>. 2012;7(1):e28213.
</p>

<p>
Ezkurdia I, Del Pozo A, Frankish A, Rodriguez JM, Harrow J, Ashman K, Valencia A, Tress ML.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/22446687" target="_blank">
Comparative Proteomics Reveals a Significant Bias Toward Alternative Protein Isoforms with Conserved Structure and Function</a>.
<em>Mol Biol Evol</em>. 2012 Apr 17;.
</p>

<a href="http://www.gencodegenes.org/publications.html"target="_blank">Publications link for the GENCODE project</a>

<h2> Data Release Policy </h2>
<p>GENCODE data are available for use without restrictions.</p>
