<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <title>wgEncodeRbpSuper.html</title>
  <meta charset="utf-8" id="webkit-interchange-charset">
</head>
<body>
<h2>Description</h2>
<p>In eukaryotic organisms, gene regulatory networks require an
additional level of coordination that links transcriptional and
post-transcriptional processes. Messenger RNAs (mRNAs) have
traditionally been viewed as passive molecules in the pathway from
transcription to translation. However, it is now clear that RNA-binding
proteins (RBPs) play a major role in regulating multiple mRNAs in order
to facilitate gene expression patterns. <br>
</p>
<p>These tracks can elucidate RNA processing by identifying RNA
molecules that interact with specific RBPs. They were developed using
assays that first purify mRNA-RBP complexes and then separate the
complexes to identify the target mRNAs bound to specific RBPs. The
mRNAs can be identified by methods including sequencing, microarrays,
and SAGE.  
<p>
The tracks in this supertrack contain two forms of information: genes
whose transcripts were bound by the given RBP (such as SUNY RIP GeneSt)
and approximate location of the RBP binding site in the mRNA sequence
(such SUNY RIP Tiling and SUNY RIP-seq).<br>
</p>

<h3>Please note:</h3>
<p>
RIP input tracks (both array and sequencing based) were created for use
in downstream informatic analysis to produce RBP specific RIP tracks.
Low abundance RNA that is undetectable in the input samples may be
proportionally enriched to the point of detection in the RIPs. This may
be confusing to some users expecting
to see RIP as a subset of input. Users seeking information on total
RNA should examine the "expression" RNA-seq tracks produced by other
ENCODE groups.
</p>
<h2>Display Conventions</h2>
<p>
These tracks are multi-view composite tracks that contains multiple
data types (views). Each view within each track
has separate display controls, as described <a
 href="http://hgwdev.cse.ucsc.edu/goldenPath/help/multiView.html">here</a>.
Most
ENCODE
tracks contain multiple subtracks, corresponding to
multiple experimental conditions. If a track contains a large
number of subtracks, only some subtracks will be displayed by default.
The user can select which subtracks are displayed via the display
controls
on the track details pages.</p>
<h2>Credits</h2>
<p>
These data were generated and analyzed as part of the <a
 href="http://www.genome.gov/10005107">ENCODE</a> project, a
genome-wide consortium project with the aim of cataloging all
functional elements in the human genome. This effort includes
collecting a variety of data across related experimental conditions, to
facilitate integrative analysis. Consequently, additional ENCODE tracks
may contain data that is relevant to the data in these tracks.</p>
<h2>References</h2>
<p>Baroni TE, Chittur SV, George AD, Tenenbaum SA. <a
 href="http://www.ncbi.nlm.nih.gov/pubmed/18369977">
Advances in RIP-chip analysis : RNA-binding protein
immunoprecipitation-microarray
profiling.</a>
<em>Methods Mol Biol. </em> 2008;419:93-108.
</p>
<p>George AD, Tenenbaum SA. <a
 href="http://www.ncbi.nlm.nih.gov/pubmed/17114943">
MicroRNA modulation of RNA-binding protein regulatory elements. </a>
<em>RNA Biol</em>. 2006;3(2):57-9. Epub 2006 Apr 1.
</p>
<p>Keene JD, Tenenbaum SA. <a
 href="http://www.ncbi.nlm.nih.gov/pubmed/12086614">
Eukaryotic mRNPs may represent posttranscriptional operons. </a>
<em>Mol Cell</em>. 2002;9(6):1161-7.
</p>
<p>Penalva LO, Tenenbaum SA, Keene JD. <a
 href="http://www.ncbi.nlm.nih.gov/pubmed/14770002">
Gene expression analysis of messenger RNP complexes. </a>
<em>Methods Mol Biol</em>. 2004;257:125-34.
</p>
<p>
Tenenbaum SA, Lager PJ, Carson CC, Keene JD. <a
 href="http://www.ncbi.nlm.nih.gov/pubmed/12054896">
Ribonomics: identifying mRNA subsets in mRNP complexes using antibodies
to RNA-binding proteins and genomic arrays.
</a><em>Methods</em>. 2002 Feb;26(2):191-8.
</p>

    <h2>Data Release Policy</h2>
<p>
Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset
until nine months following the release of the dataset. This date is
listed in the Restricted Until column on the track configuration page
and the download page. The full data release policy for ENCODE is
available <a href="http://genome.ucsc.edu/ENCODE/terms.html">here</a>.
</p>
</body>
</html>
