<H2>Release Note</H2>
<P>This is the NCBI Build37 (hg19) release of this track.  This release includes the 3 datasets previously released on NCBI Build36 (hg18) and adds data for several more cell types and growth conditions in replicate.  Four types of download files are available for each replicate including the raw data (fastq) and 3 display files: FPKM's for GencodeV3c gene models (gtf), raw signal (bigwig), and alignments (bam).</P>

<H2>Description</H2>

<P>This track is produced as part of the ENCODE Project. RNA-seq is a method for mapping and quantifying the transcriptome of any organism that has a genomic DNA sequence assembly (Mortazavi <EM>et al.</EM>, 2008). Biological replicates of ENCODE cell lines were grown on separate culture plates, total RNA was purified and 
polyA selected two times.  mRNA was then fragmented by magnesium-catalyzed hydrolysis, reverse transcribed to cDNA by random priming and amplified.  The cDNA was sequenced on an Illumina Genome Analyzer (GAI or GAIIx).</P>  

</P>The DNA sequences were aligned to the NCBI Build37 (hg19) version of the human genome using the sequence alignment programs ELAND (Illumina) or Bowtie (Langmead <EM>et al.</EM>, 2009).  The first 10 residues of sequencing have a weak characteristic nucleotide bias of unknown origin. This RNA-seq protocol does not specify the coding strand.  As a result, there will be ambiguity at loci where both strands are transcribed.</P>
 

<H2>Display Conventions and Configuration</H2>

<P>This track is a multi-view composite track that contains multiple data types (<EM>views</EM>). For each view, there are multiple subtracks (cell lines, replicates and growth conditions) that display individually on the browser. Instructions for configuring multi-view tracks are <A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.  The following views are in this track: </DL>

<DT><I>FPKM</I>
<DD>Gencode.v3c (Harrow <EM>et al.</EM>, 2006) gene models are shaded with a <A HREF="http://genome.ucsc.edu/FAQ/FAQformat.html#format1" TARGET=_BLANK>gray scale</A> according to a score calculated from the FPKM value (Fragments Per Kilobase of exon per Million reads, Cufflinks v0.9.3, Roberts <EM>et al.</EM>, 2011).  Score is equal to 100 x log<sub>2</sub>(FPKM +1) and is capped at 1000.  The gray scale becomes darker as score and FPKM increase, thereby assisting in visualizing the relative amount of a given transcript across multiple samples. 
<P>
<DT><I>Alignments</I>
<DD>The Alignments view shows reads mapped to the genome. 
</DD>
<P>
<DT><I>Raw Signal</I>
<DD>Density graph of signal enrichment based on a normalized aligned read density (Read Per Million, RPM).  RPM is reported in the score field and is equal to the number of reads at that position divided by the total number of reads divided by one million.  The Raw Signal view displays dense, continuous data as a graph and the RPM measure assists in visualizing the relative amount of a given transcript across multiple samples.</DD>

<H2>Methods</H2>

<H3>Experimental Procedures</H3>

<P>Cells were grown according to the approved <A HREF="../../ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture protocols</A> except for H1-hESC for which frozen cell pellets were purchased from Cellular Dynamics.  Cells were lysed in RLT buffer (Qiagen RNEasy kit) and processed on RNEasy midi columns according to the manufacturer's protocol, with the inclusion of the "on-column" DNase digestion step to remove residual genomic DNA.  mRNA was isolated from at least 10 ug of total RNA with oligo(dT) two times (Dynabeads mRNA PurificationgKit, Invitrogen).  Alternatively, cells were lysed and mRNA was purified directly two times with oligo(dT) (Dynabeads mRNA DIRECT Kit, Invitrogen).  100 ng of mRNA was fragmented by magnesium-catalyzed hydrolysis and reverse transcribed to cDNA by random priming according to the protocol in Mortazavi <EM>et al.</EM> (2008).  cDNA was prepared for sequencing on the Genome Analyzer flowcell according to the protocol for the ChIPSeq DNA genomic DNA kit (Illumina).  The sequencing libraries were size-selected around 225 bp and amplified with 15 rounds of PCR.</P>

<P>Libraries were sequenced with an Illumina Genome Analyzer I or an Illumina Genome Analyzer IIx according to the manufacturer's recommendations. Single end reads of 36 nt in length were obtained. 
</P>

<H3>Data Processing and Analysis</H3>

<P>Fastq files were made from qseq files generated by the Illumina pipeline (Casava 1.7).  The Raw Signal files (bigWig) were generated from bedgraph files and the score was calculated as the number of reads at that position divided by the total number of reads divided by one million.</P>

<P>Casava export files were aligned to the NCBI Build37 (hg19) version of the human genome with ELAND (Illumina), generating SAM files.  Fastq files of experiments that were previously aligned to NCBI Build36 (hg18) were aligned to NCBI Build37 (hg19) using Bowtie (Langmead <EM>et al.</EM>, 2009; parameters: -S -n 2 -k 11 -m 10 --best), also generating SAM files.  SAM files were converted to BAM with SAMtools (Li <EM>et al.</EM>, 2009).</P>   
 
<P>RNA-seq reads were aligned to Gencode.v3c (Harrow <EM>et al.</EM>, 2006) gene models and gene expression was measured in Fragments Per Kilobase exon per Million reads (FPKM) using Cufflinks v0.9.3 (Roberts <EM>et al.</EM>, 2011).  FPKM is calculated by dividing the total number of fragments that align to the gene model by the size of the spliced transcript (exons) in kilobases.  This number is then divided by the total number of reads in millions for the experiment.  FPKM is reported in the last column of the gtf (TranscriptGencV3c) files.</P>

<P>RawData (fastq), RawSignal (bigWig), Alignments (bam) and TranscriptGencV3c (gtf) files are available from the <A HREF="http://hgwdev.cse.ucsc.edu/cgi-bin/hgFileUi?g=wgEncodeHaibRnaSeq" TARGET=_BLANK>Downloads</A> page.</P>   


<H2>Verification</H2>
<UL>
<LI>Known exon maps as displayed on the genome browser are confirmed by the alignment of sequence reads.</LI>
<LI>Biological replicates confirm expression measurements with r > 0.90
</UL>

<H2>Credits</H2>

<P>These data were produced by the <A TARGET=_BLANK HREF="http://hudsonalpha.org/myers-lab"> Dr. Richard Myers Lab</A>  at the <A TARGET=_BLANK HREF="http://hudsonalpha.org/"> HudsonAlpha Institute for Biotechnology</A>. 

<P>Contact: <A HREF="mailto:&#102;p&#97;&#117;&#108;&#105;&#64;&#104;&#117;&#100;&#115;&#111;&#110;&#97;&#108;&#112;&#104;a.
&#111;r&#103;">Dr. Florencia Pauli</A>.
<!-- above address is fpauli at hudsonalpha.org (encodeEmail.pl) -->


</P>

<H2>References</H2>

<P>Harrow J, Denoeud F, Frankish A, Reymond A, Chen CK, Chrast J, Lagarde J, Gilbert JG, Storey R, Swarbreck D, Rossier C, Ucla C, Hubbard T, Antonarakis SE and Guigo R.
<A TARGET=_BLANK HREF="http://www.gencodegenes.org/">GENCODE: producing a reference annotation for ENCODE</A>
<EM>Genome Biology</EM>. 2006; 7 Suppl 1;S4.1-9

<P>Langmead B, Trapnell C, Pop M, Salzberg SL.
<A TARGET=_BLANK HREF="http://genomebiology.com/2009/10/3/R25"><U>Ultrafast and 
memory-efficient alignment of short DNA sequences to the human genome</U></A>
<EM>Genome Biology</EM>. 2009 Mar; 10:R25.</P>

<P>Li H, Handsaker B, Wysoker A, Fennell T, Ruan J, Homer N, Marth G, Abecasis G, Durbin R and 1000 Genome Project Data Processing Subgroup. <A TARGET=_BLANK HREF="http://bioinformatics.oxfordjournals.org/content/25/16/2078.long"><U>The Sequence alignment/map (SAM) format and SAMtools</U></A> <EM>Bioinformatics</EM>. 2009; 25:2078-9.

<P>Mortazavi A, Williams BA, McCue K, Schaeffer L, and Wold BJ. 
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/pubmed/18516045">Mapping and 
quantifying mammalian transcriptomes by RNA-Seq</A>
<EM>Nature Methods</EM>. 2008 Jul; 5(7):621-628.</P> 

<P> Roberts A, Trapnell C, Donaghey J, Rinn JL, Patcher L.
<A TARGET=_BLANK HREF="http://genomebiology.com/2011/12/3/R22/abstract"><U>Improving RNA-Seq expression estimates by correcting for fragment bias </U></A><EM>Genome Biology</EM>. 2011 Mar; 12:R22.</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column, above.  The full data release policy
for ENCODE is available
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>

