<h2> Release Note </h2>
<p> This is the NCBI Build37 (hg19) release of this track.  This release includes the 3 cell types previously released on NCBI Build36 (hg18) and adds data for many more cell types. 
</p> 
<h2> Description </h2>
<p>This track is produced as part of the ENCODE project.  The track displays the methylation status of specific CpG dinucleotides in the given cell types as identified by the <a target='_blank' href="http://www.illumina.com/pages.ilmn?ID=243">Illumina Infinium HumanMethylation27 BeadArray platform</a>.  In general, methylation of CpG sites within a promoter causes silencing of the gene associated with that promoter. 
</p>
<p>
Detailed information for the CpG targets is in an XLS formatted spreadsheet on the Myers' lab  
<a target='_blank' href="http://hudsonalpha.org/myers-lab/protocols">protocols</a> website.
</a>
</p>
<h2>Display Conventions</h2>
<p>
Scores associated with each site are beta values (see Methods) multiplied by 1000.  Methylation
status is color-coded as:
<ul>
<li><font color="#FF7f00">orange</font> = methylated (score >= 600) </li>
<li><font color="#800080">purple</font> = partially methylated (200 < score < 600)</li>
<li><font color="#0000CD">bright blue</font> = unmethylated (0 < score <= 200) </li>
<li>black = NA (score = 0)</li>
</ul>
</p>
<p>
</p>

<a name="Methods"></a><h2> Methods </h2>

<P>
Cells were grown according to the approved
<A HREF="http://hgwdev.cse.ucsc.edu/cgi-bin/hgEncodeVocab?type=cellType" TARGET=_BLANK>ENCODE cell culture protocols</A>.</P>

<p>Genomic
DNA was isolated from each cell line with the QIAGEN DNeasy Blood &amp; Tissue Kit according to the instructions provided by the manufacturer.  DNA concentrations and a level of quality of each preparation was determined by fluorescence with the Qubit Fluorometer (Invitrogen).  
</p>
<p>
The Methyl27K platform uses bisulfite treated genomic DNA to assay the methylation status of 27,578 CpG sites within more than 14,000 genes.  When genomic DNA is treated with sodium bisulfite, unmethylated cytosine of CpG dinucleotides are converted into uracils; methylated cytosines do not get converted.  After bisulfite treatment, the methylation status of a site is assayed by single base-pair extension with a Cy3 or Cy5 labeled nucleotide on oligo-beads specific for the methylated or unmethylated state. A beta value is calculated by Illumina's Bead Studio software for each CpG target.  This value represents the intensity value from the methylated bead type divided by the sum of the intensity values from the methylated and unmethylated bead types for any given CpG target.
</p>
<p>Bisulfite conversion reaction was done using the
<a target='_blank' href="http://www.zymoresearch.com/product/ez-96-dna-methylation-kit-d5003">Zymo Research EZ-96 DNA Methylation<sup>TM</sup> Kit</a>.  One step of the protocol was modified.  During the incubation, a 30 sec 95<sup>o</sup>C denaturing step every hour was included to increase reaction efficiency as recommended by the Illumina Infinium Human Methylation27 protocol. 
</p>
<p>The bead arrays were run according to the 
<a target='_blank' href="http://www.illumina.com/pagesnrn.ilmn?ID=275">protocol</a> provided by Illumina.
</p>
<p>The intensity data from the BeadArray was processed using Illumina's BeadStudio software with the Methylation Module v3.2.  The data was then quality-filtered using p-values.  
</p>
<p>
Any beta value equal to or greater than 0.6 is considered fully methylated.  Any beta value equal to or less than 0.2 is considered to be fully unmethylated.  Beta values between 0.2 and 0.6 are considered to be partially methylated.  Beta-values are quality filtered and spots that fall below the minimum intensity threshold are displayed as "NA".
</p>
<a name="Credits"></a><h2> Credits </h2>
<p>
<P>These data were produced by the <A TARGET=_BLANK HREF="http://hudsonalpha.org/myers-lab"> Dr. Richard Myers Lab</A> and the Dr. Devin Absher lab at the <A TARGET=_BLANK HREF="http://hudsonalpha.org/"> HudsonAlpha Institute for Biotechnology</A>.  
<p>Cells were grown by the Myers Lab and other ENCODE production groups. 
</P>
Contact:
<A HREF="mailto:&#102;p&#97;&#117;&#108;&#105;&#64;&#104;&#117;&#100;&#115;&#111;&#110;&#97;&#108;&#112;&#104;a.
&#111;r&#103;">Dr. Florencia Pauli</A>.
<!-- above address is fpauli at hudsonalpha.org (encodeEmail.pl) -->


<h2> Data Release Policy </h2>

<p>Data users may freely use ENCODE data, but may not, without prior consent, submit publications that use an unpublished ENCODE dataset until nine months following the release of the dataset.  This date is listed in the <EM>Restricted Until</EM> column on the track configuration page and the download page.  The full data release policy for ENCODE is available
<a href="../ENCODE/terms.html" TARGET=_BLANK>here</a>.</p>

