<h2>Description</h2>
<p>This track displays maps of chromatin state generated by the Broad/MGH
ENCODE group using ChIP-seq. Chemical modifications (methylation, acetylation)
to the histone proteins present in chromatin influence gene expression
by changing how accessible the chromatin is to transcription.
</p>

<p>The ChIP-seq method involves first using formaldehyde to cross-link histones
and other DNA-associated proteins to genomic DNA within cells. The cross-linked
chromatin is subsequently extracted, mechanically sheared, and
immunoprecipitated using specific antibodies. After reversal of cross-links,
the immunoprecipitated DNA is sequenced and mapped to the human reference
genome. The relative enrichment of each antibody-target (epitope) across the
genome is inferred from the density of mapped fragments.


<H2> Display Conventions and Configuration </H2>
<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are <A HREF="/goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
ENCODE tracks typically contain one or more of the following views:
<DL>
<DT><EM>Peaks</EM></DT><DD> Regions of statistically significant signal
enrichment. The score associated with each enriched interval is the mean
signal value across the interval. (Note that a broad region with moderate
enrichment may deviate from the background more significantly than a short
region with high signal.)
</DD>
<DT><EM>Signal</EM></DT><DD> Density graph (wiggle) of signal enrichment.
At each base-pair position the density is calculated as the number of sequenced
tags overlapping a 25 bp window centered at that position.</DD>
</DL>
</P>
<P>Peaks and signals displayed in this track are the results of pooled replicates. The raw
sequence and alignment files for each replicate are available for
<A HREF="/cgi-bin/hgFileUi?g=wgEncodeBroadHistone">download</A>.
</P>
<p>Metadata for a particular subtrack can be found by clicking the down arrow in the list of subtracks.</p>

<h2>Methods</h2>
<P><B><em>ChIP-seq</em></B>: Cells were grown according to the approved
<A HREF="/ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture protocols</A>.
Cells were fixed in 1% formaldehyde and resuspended in lysis buffer. Chromatin
was sheared to 200-700 bp using a Diagenode Bioruptor. Solubilized chromatin was
immunoprecipitated with antibodies against each of the histone antibodies
listed above. Antibody-chromatin complexes were pulled-down using protein
A-sepharose (or anti-IgM-conjugated agarose for RNA polymerase II), washed and
then eluted. After cross-link reversal and proteinase K treatment,
immunoprecipitated DNA was extracted with phenol-chloroform, ethanol
precipitated, treated with RNAse and purified. A quantity of 1-10 ng of DNA was
end-repaired, adapter-ligated and sequenced by Illumina Genome Analyzers as
recommended by the manufacturer.</p>

<P><B><em>Alignment</em></B>: Sequence reads from each IP experiment were aligned to the
human reference genome (<B>GRCh37/hg19</B>) using
<a href="http://maq.sourceforge.net/" title='http://maq.sourceforge.net/' target="_blank">MAQ</a>
with default parameters, except '<code>-C 11</code>' and
'<code>-H output_file</code>' were added, which added outputs up to 11 additional best matches
for each read (if any are found) to a file. This information was used to filter
out any read that had more than 10 best matches on the genome.  Note: It is
likely that instances where multiple reads align to the same position and with
the same orientation are due to enhanced PCR amplification of a single DNA
fragment. No attempt has been made, however, to remove such artifacts from the
data, following ENCODE practices.</p>

<P><B><em>Signal</em></B>: Fragment densities were computed by counting the number of
reads overlapping each 25 bp bin along the genome. Densities were computed using
<a href="http://www.broadinstitute.org/igv/node/173" title='http://www.broadinstitute.org/igv/node/173' target="_blank">igvtools count</a>
with default parameters (in particular, '<code>-w 25</code>' to set window size
of 25 bp and '<code>-f mean</code>' to report the mean value across the window),
except for '<code>-e</code>' which was set to extend the reads to 200 bp, and the
<em>.wig</em> output was converted to <em>bigWig</em> using <code>wigToBigWig</code>
from the UCSC Kent software package.</p>

<P><B><em>Peaks</em></B>: Discrete intervals of ChIP-seq fragment enrichment were
identified using <a href="http://www.broadinstitute.org/software/scripture/" title='http://www.broadinstitute.org/software/scripture/' target="_blank">Scripture</a>,
a scan statistics approach, under the assumption of uniform background signal.
All data sets where processed with '<code>-task chip</code>', and with
'<code>-windows 100,200,500,1000,5000,10000,100000</code>'. (No mask file nor
the '<code>-trim</code>' option have been used.) The resulting called segments
were then further filtered to remove intervals that are significantly enriched
only because they contain smaller enriched intervals within them. This
post-processing step has been implemented using <code>Matlab</code>. The use of the
post-processing step allowed very large enriched intervals (of the order of
Mbps for H3K27me3, for instance) to be detected, as well as much smaller
intervals, without the need to tailor the parameters of <code>Scripture</code>
based on prior expectations.

<h2>Release Notes</h2>

<p>
<ul>
<li>This is Release 2 (Mar 2012). It contains 64 new experiments including 1 new cell line (Dnd41) and 10 new antibodies.</li>
<li>Three data sets have added additional data: GM12878/H3K27me3 (UCSC Accession: wgEncodeEH000031), GM12878/H3K4me1 (UCSC Accession: wgEncodeEH000033) and GM12878/H3K4me3 (UCSC Accession: wgEncodeEH000028). Files that have been reanalyzed have a version number appended to the name, <i>e.g.</i>V2. </li>
</ul>
</p>


<h2>Credits</h2>
<p>The ChIP-seq data were generated at the
 <a href="http://www.broadinstitute.org/science/projects/epigenomics/epigenomics-initiative" title="http://www.broad.mit.edu" target="_blank">
 Broad Institute</a> and in the
 <a href="https://csb.mgh.harvard.edu/bradley_bernstein" title="https://csb.mgh.harvard.edu/bradley_bernstein" target="_blank">
 Bradley E. Bernstein lab</a> at the Massachusetts General Hospital/Harvard Medical School.
  &nbsp;&nbsp;

<p>Data generation and analysis was supported by funds from the NHGRI, the
Burroughs Wellcome Fund, Massachusetts General Hospital and the Broad Institute.
</p>

</p>
Contact: <A HREF="mailto:&#110;&#115;h&#111;&#114;&#101;&#115;h&#64;&#98;r&#111;&#97;&#100;.&#109;it.&#101;&#100;&#117;"
  title="&#110;&#115;h&#111;&#114;&#101;&#115;h&#64;&#98;r&#111;&#97;&#100;.&#109;it.&#101;&#100;&#117;" rel="nofollow"
  TARGET = _BLANK>Noam Shoresh</a>
</p>

<h2>References</h2>

<p>
Bernstein BE, Kamal M, Lindblad-Toh K, Bekiranov S, Bailey DK, Huebert DJ, McMahon S, Karlsson EK, Kulbokas EJ 3rd, Gingeras TR <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/15680324" target="_blank">
Genomic maps and comparative analysis of histone modifications in human and mouse</a>.
<em>Cell</em>. 2005 Jan 28;120(2):169-81.
</p>

<p>
Bernstein BE, Mikkelsen TS, Xie X, Kamal M, Huebert DJ, Cuff J, Fry B, Meissner A, Wernig M, Plath K <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/16630819" target="_blank">
A bivalent chromatin structure marks key developmental genes in embryonic stem cells</a>.
<em>Cell</em>. 2006 Apr 21;125(2):315-26.
</p>

<p>
Ernst J, Kheradpour P, Mikkelsen TS, Shoresh N, Ward LD, Epstein CB, Zhang X, Wang L, Issner R, Coyne M <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/21441907" target="_blank">
Mapping and analysis of chromatin state dynamics in nine human cell types</a>.
<em>Nature</em>. 2011 May 5;473(7345):43-9.
</p>

<p>
Guttman M, Garber M, Levin JZ, Donaghey J, Robinson J, Adiconis X, Fan L, Koziol MJ, Gnirke A, Nusbaum C <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/20436462" target="_blank">
Ab initio reconstruction of cell type-specific transcriptomes in mouse reveals the conserved multi-exonic structure of lincRNAs</a>.
<em>Nat Biotechnol</em>. 2010 May;28(5):503-10.
</p>

<p>
Mikkelsen TS, Ku M, Jaffe DB, Issac B, Lieberman E, Giannoukos G, Alvarez P, Brockman W, Kim TK, Koche RP <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/17603471" target="_blank">
Genome-wide maps of chromatin state in pluripotent and lineage-committed cells</a>.
<em>Nature</em>. 2007 Aug 2;448(7153):553-60.
</p>

<h2>Publications</h2>

<p>
Ram O, Goren A, Amit I, Shoresh N, Yosef N, Ernst J, Kellis M, Gymrek M, Issner R, Coyne M <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/22196736" target="_blank">
Combinatorial patterning of chromatin regulators uncovered by genome-wide location analysis in human cells</a>.
<em>Cell</em>. 2011 Dec 23;147(7):1628-39.
</p>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM>, above.  The full data release policy for ENCODE is available
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>
