<H2>Description</H2>

<P>This track is produced as part of the ENCODE Project.
This track displays genome-wide maps of histone modifications in different
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=cellType">cell lines</A>,
using ChIP-seq high-throughput sequencing.

<H2>Display Conventions and Configuration</H2>
<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>). For each view, there are multiple subtracks that display
individually on the browser. Instructions for configuring multi-view tracks
are <A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
<P>
For each cell type, this track contains the following views:
<DL>
<DT><I>HotSpots</I>
<DD> ChIP-seq affinity zones identified using the HotSpot algorithm.
<DT><I>Peaks</I>
<DD> ChIP-seq affinity sites identified as signal peaks within
FDR 1.0% affinity zones.
<DT><I>Raw Signal</I>
<DD> The density of tags mapping within a 150 bp sliding window
(at a 20 bp step across the genome).
</DL>
<H2>Methods</H2>

<P>
Cells were grown according to the approved
<A HREF="../../ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture protocols</A>.
Cells were crosslinked with 1% formaldehyde, and the reaction was quenched
by the addition of glycine. Fixed cells were rinsed with PBS, lysed in nuclei
lysis buffer, and the chromatin was sheared to 200-500 bp fragments using
Fisher Dismembrator (model 500). Sheared chromatin fragments were
immunoprecipitated with specific polyclonal antibody at 4 degrees C
with gentle rotation. Antibody-chromatin complexes were washed and eluted.
The cross linking in immunoprecipitated DNA was reversed and treated with
RNase-A.  Following proteinase K treatment, the DNA fragments were purified
by phenol-chloroform-isoamyl alcohol extraction and ethanol precipitation.
20-50 ng of ChIP DNA was end-repaired, followed by the addition of adenine,
ligated to Illumina adapters and made in to Solexa library for sequencing.
<P>
ChIP-seq affinity is directly reflected in
raw tag density (<I>Raw Signal</I>), which is shown in the track as density of tags
mapping within a 150 bp sliding window (at a 20 bp step across the genome).
ChIP-seq affinity zones (<I>HotSpots</I>) were identified using the
HotSpot algorithm described in Sabo <EM>et al.</EM> (2004).
One percent false discovery rate thresholds (FDR 0.01) were computed for each cell type by
applying the HotSpot algorithm to an equivalent number of random
uniquely mapping 36mers. ChIP-Seq affinity  (<I>Peaks</I>)
were identified as signal peaks within FDR 1.0% hypersensitive
zones using a peak-finding algorithm.
</P>

<H2>Verification</H2>

<P>
Data were verified by sequencing biological replicates displaying correlation
coefficient >0.9.
</P>

<h2>Release Notes</h2>
<p>
Release 1 (Feb 2011) of this track has 4 bam files found in the
<A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeUwHistone/"
 TARGET=_BLANK>downloads page</A> that contain mis-mappings to chrY, with long
 stretches of n's in the alignments. The associated processed data files will contain small
 number of mis-mapping to chrY. The files directly affected are:
 <ul>
 <li>wgEncodeUwHistoneCaco2H3k4me3StdAlnRep1.bam</li>
 <li>wgEncodeUwHistoneHepg2H3k4me3StdAlnRep1.bam</li>
 <li>wgEncodeUwHistoneHepg2H3k4me3StdAlnRep2.bam</li>
 <li>wgEncodeUwHistoneHuvecH3k27me3StdAlnRep2.bam</li>
 </ul>
 </p>



<H2>Credits</H2>

<P>These data were generated by the UW ENCODE group. </P>
<P>Contact: <A HREF="mailto:&#115;&#117;l&#108;&#64;&#117;.w&#97;&#115;h&#105;&#110;&#103;&#116;&#111;n.&#101;&#100;&#117;">Richard Sandstrom</A>
<!-- above address is sull at u.washington.edu -->
</P>

<H2>References</H2>

<P>Sabo PJ, Hawrylycz M, Wallace JC, Humbert R, Yu M, Shafer A, Kawamoto J,
Hall R, Mack J, Dorschner MO, <span style="font-style: italic;">et al</span>.
<A TARGET=_BLANK HREF="http://www.pnas.org/content/101/48/16837">Discovery of functional noncoding elements by digital analysis of chromatin structure.</A>
<EM>PNAS</EM>. 2004;101:16837-16842.</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column, above.  The full data release policy
for ENCODE is available
<A TARGET=_BLANK HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>

