<H2>Description</H2>

<P>These tracks were generated by the 
<A TARGET=_BLANK HREF="http://www.genome.gov/10005107">ENCODE
Consortium</A>.  They contain information about human RNAs &gt; 200
nucleotides in length that were obtained as short reads from the Illumina GAIIx
platform.  Data are available from biological replicates of several
cell lines.  In addition to profiling Poly-A+ and Poly-A- RNA from
whole cells, there are also data from various subcellular
compartments.  In many cases, there are Cap Analysis of Gene
Expression (CAGE, see the
<A TARGET=_BLANK HREF="../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRikenCage">RIKEN CAGE Loc</A> track),
Small RNA-seq (&lt;200 nucleotides, see the
<A TARGET=_BLANK HREF="../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlShortRnaSeq">CSHL Sm RNA-seq</A> track) and Pair-End
di-TAG-RNA (PET-RNA, see the
<A TARGET=_BLANK
HREF="../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisRnaPet">GIS RNA PET</A> track) datasets available from the same biological replicates.   

<h2>Display Conventions and Configuration</h2>

<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are
<A HREF="/goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.</P>

<p>To show only selected subtracks, uncheck the boxes next to the tracks that
you wish to hide.</p>

<p>Color differences among the views are arbitrary. They provide a
visual cue for
distinguishing between the different cell types and compartments.</p>

<dl>
<dt><i>Contigs</i>
</dt><dd>The Contigs represent blocks of overlapping mapped reads from the pooled biological replicates. Specific column specifications can be found in the <A TARGET=_BLANK HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeCshlLongRnaSeq/supplemental/">supplemental directory</A>.
</dd><dt><i>Signals</i>
</dt><dd>The Plus Signal and Minus Signal views show the density of mapped reads on the plus and minus strands (wiggle format), respectively.
</dd>
<dt><i>Alignments</i>
</dt><dd>The Alignments view shows individual reads mapped from biological replicates to the genome and indicates where
bases may mismatch. Every mapped read is displayed, i.e. uncollapsed.  The alignment file follows the standard SAM format. See the <a href="http://samtools.sourceforge.net/SAM1.pdf#page=4" target=_blank>SAM Format Specification</a> for more information on the SAM/BAM file format. 
</dd>
<dt><i>Splice Junctions</i>
</dt><dd>Subset of aligned reads that cross splice junctions. Specific column specifications can be found in the <A TARGET=_BLANK HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeCshlLongRnaSeq/supplemental/">supplemental directory</A>.
</dd>
</dl>
<p>Metadata for a particular subtrack can be found by clicking the down arrow in the list of subtracks.</p>

<P>
<!--echo var="BROWSER_HOST" -->
Additional <A TARGET=_BLANK HREF="../cgi-bin/hgEncodeVocab?tag=EXONGENCV7,FASTQRD1,FASTQRD2,GENEDENO,GENEGENCV7,GENEGENCV10,EXONGENCV10,PRTCL,SPKNS,TRANSGENCV7,TRANSDENOVO">views</A> are available on the Downloads page. 
</P>

<H2>Methods</H2>

<H3>Cell Culture</H3>
<P>Cells were grown according to the approved 
<A TARGET=_BLANK
HREF="/ENCODE/protocols/cell/"> ENCODE
cell culture protocols</A>.  
</P>

<H3>Library Preparation</H3>
<P>The published 
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/pubmed/19620212">
cDNA sequencing protocol</A> was used.  This protocol generates directional libraries
and reports the transcripts' strand of origin.  Exogenous RNA spike-ins
were added to each endogenous RNA isolate and carried
through library construction and sequencing.  The Illumina PhiX
control library was also spiked-in at 1&#37; to each completed human
library just prior to cluster formation.
 Accompanying each RNA-seq dataset is a
&ldquo;Protocol Document,&rdquo; available for download.  This document contains details
about the RNA isolations and treatments, library construction,
spike-ins as well as quality control figures for individual libraries.   
The spike-in sequence and the concentrations are available for download
in the <A TARGET=_BLANK HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeCshlLongRnaSeq/supplemental/">supplemental directory</A>.
</P>

<H3>Sequencing and Mapping</H3>
<P>The libraries were sequenced on the Illumina GAIIx platform
as paired-ends for 76 or 101 cycles for each read.
The average depth of sequencing was &#126;200 million reads 
(100 million paired-ends). 
The data were mapped against hg19 using Spliced Transcript Alignment
and Reconstruction (STAR) written by Alex Dobin (CSHL).  More
information about STAR, including the parameters used for these data,
is available from the
<A TARGET=_BLANK
HREF="http://gingeraslab.cshl.edu/STAR/">Gingeras lab</A>.
</P>

<P>
For each experiment there are additional 
<a target="_BLANK" href="../cgi-bin/hgEncodeVocab?tag=EXONSDENOVO,EXONGENCV7,EXONGENCV10,GENEDENO,GENEGENCV7,GENEGENCV10,TRANSDENOVO,TRANSGENCV7,TRANSGENCV10">element data views</a>
data files available for <a target="_BLANK" href="../cgi-bin/hgFileUi?db=hg19&g=wgEncodeCshlLongRnaSeq">download</a>.
These elements were assessed for reproducibility using a nonparametric 
irreproducible detection (IDR) rate script.  The IDR values for each element
 are included in the files for end-users to use as a threshold. An IDR value of 0.1 means
 that the probability of detecting that element in a third experiment equivalent 
in depth to the sum of the bioreplicates is 90&#37;.  In addition,
 expression values for annotated genes, transcripts and exons were computed. Further explanation of these 
files is available for download in the
 <A TARGET=_BLANK HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeCshlLongRnaSeq/supplemental/">supplemental directory</A>.
</P>

<H2>Verification</H2>

<P>
FPKM (fragments per kilobase of exon per million fragments mapped) values were calculated 
for annotated Gencode exons and Spearman values were compared.  
In general, Rho values are &#62; .90 between biological replicates.   
Additional quality control metrics for the samples can be found 
<A TARGET=_BLANK
HREF="http://genome.crg.es/~jlagarde/encode_RNA_dashboard/">Transcriptome Data Portal</A>.
</P>

<H2>Release Notes</H2>

<P> 
This is release 2 (May 2012) of this track for hg19. It includes 25 new cell lines:
CD20+, CD34+_Mobilized, HAoAF, HAoEC, HCH, HFDPC, HMEpC, hMNC-CB, hMNC-PB, hMSC-AT, hMSC-BM, hMSC-UC, HOB, HPC-PL, HPIEpC, HSaVEC, HVMF, HWP, IMR90, Monocytes-CD14+, NHDF, NHEM.f_M2, NHEM_M2, SkMC, and SK-N-SH.  The downloadable elements files (Transcripts, Genes and Exons) in this release were generated using GENCODE V10, while the older datasets were generated using GENCODE V7.  The metadata will specify V7 or V10 for these files.
</P>


<H2>Credits</H2>

<P>These data were generated and analyzed by the transcriptome group at
Cold Spring Harbor Laboratories and the Center for Genomic
Regulation (CRG in Barcelona), who are participants in the ENCODE Transcriptome Group.
</P>
<h3>Contacts</h3>
<dl>
<dd>
<A HREF="mailto:davisc@cshl.edu">Carrie Davis</A> (experimental),
<!-- above address is davisc at cshl.edu -->
</dd>
<dd>
<A HREF="mailto:dobin@cshl.edu">Alex Dobin</A> (computational),
<!-- above address is dobin at cshl.edu -->
</dd>
<dd>
<A HREF="mailto:schlesin@cshl.edu">Felix
Schlesinger</A> (computational),
<!--above address is schlesin at cshl.edu -->
</dd>
<dd>
<A HREF="mailto:gingeras@cshl.edu">Tom Gingeras</A> (primary investigator), and
<!--above address is gingeras at cshl.edu -->
</dd>
<dd>
<A HREF="mailto:rguigo@imim.es">Roderic Guigo</A>'s group at the
<A TARGET=_BLANK
HREF="http://pasteur.crg.es/portal/page/portal/Internet/">CRG</A>.
<!-- above address is rguigo at imim.es -->
</dd>
</dl>
</P>

<H2>References</H2>

<p>
Parkhomchuk D, Borodina T, Amstislavskiy V, Banaru M, Hallen L, Krobitsch S, Lehrach H, Soldatov A.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/19620212" target="_blank">
Transcriptome analysis by strand-specific sequencing of complementary DNA</a>.
<em>Nucleic Acids Res</em>. 2009 Oct;37(18):e123.
</p>

<h2>Publications</h2>

<p>
Deng X, Hiatt JB, Nguyen DK, Ercan S, Sturgill D, Hillier LW, Schlesinger F, Davis CA, Reinke VJ, Gingeras TR <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/22019781" target="_blank">
Evidence for compensatory upregulation of expressed X-linked genes in mammals, Caenorhabditis elegans and Drosophila melanogaster</a>.
<em>Nat Genet</em>. 2011 Oct 23;43(12):1179-85.
</p>

<p>
Jiang L, Schlesinger F, Davis CA, Zhang Y, Li R, Salit M, Gingeras TR, Oliver B.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/21816910" target="_blank">
Synthetic spike-in standards for RNA-seq experiments</a>.
<em>Genome Res</em>. 2011 Sep;21(9):1543-51.
</p>

<H2>Data Release Policy</H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column, above.  The full data release policy
for ENCODE is available
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>
