<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <title>wgEncodeGencodeSuper</title>
</head>
<body>
</P>
<H2>Description</H2>
<P>
The aim of the <a href="http://www.gencodegenes.org/" target="_blank">GENCODE 
Genes project</a> (Harrow <em>et al.</em>, 2006) is to produce a set of 
highly accurate annotations of evidence-based gene features on the human reference genome.
This includes the identification of all protein-coding loci with associated
alternative splice variants, non-coding with transcript evidence in the public 
databases (NCBI/EMBL/DDBJ) and pseudogenes. A high quality set of gene
structures is necessary for many research studies such as comparative or 
evolutionary analyses, or for experimental design and interpretation of the 
results.</p>
<p> 
The GENCODE Genes tracks display the high-quality manual annotations merged 
with evidence-based automated annotations across the entire
human genome. The GENCODE gene set presents a full merge
between HAVANA manual annotation and Ensembl automatic annotation.
Priority is given to the manually curated HAVANA annotation using predicted
Ensembl annotations when there are no corresponding manual annotations. With 
each release, there is an increase in the number of annotations that have undergone
manual curation.  
This annotation was carried out on the GRCh37 (hg19) genome assembly.
</p>
<p>Experimental verification details are given in each descriptions for each
track. Transcript Support Levels were determined for version 10 onwards based 
on evidence provided by GenBank mRNA and EST sequences. Versions 7 and 10 are
being used in data analysis by the ENCODE consortium.</p>

NOTE</B>: Due to the UCSC Genome Browser using the NC_001807 mitochondrial 
genome sequence
(chrM) and GENCODE annotating the NC_012920 mitochondrial sequence, the
GENCODE mitochondrial sequences are not available in the UCSC Genome Browser.
These annotations are available for download in the 
<a HREF="http://www.gencodegenes.org/releases/"TARGET="_BLANK"> GENCODE GTF files</a>.
</p>

<h2>Display Conventions</h2>
<P>
These are multi-view composite tracks that contain differing data sets
(<EM>views</EM>).  Instructions for configuring multi-view tracks are
<A HREF="/goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
Only some subtracks are shown by default. The user can select which subtracks
are displayed via the display controls on the track details pages.
Further details on display conventions and data interpretation are available in the track descriptions.</p>

<h2>Release Notes</h2>
<p>
<span style="font-weight: bold;">GENCODE version 27lift37 (mapped from GRCh38 to GRCh37)</span>
corresponds to Ensembl 90.
<br>
<br>
<p>
<span style="font-weight: bold;">GENCODE version 26lift37 (mapped from GRCh38 to GRCh37)</span>
corresponds to Ensembl 88.
<br>
<br>
<p>
<span style="font-weight: bold;">GENCODE version 24lift37 (mapped from GRCh38 to GRCh37)</span>
corresponds to Ensembl 83.
<br>
<br>
<span style="font-weight: bold;">GENCODE version 19</span> corresponds to Ensembl 74 and Vega 54.
<br>
<br>
<span style="font-weight: bold;">GENCODE version 17</span>
corresponds to Ensembl 72 and Vega 52.
<br>
<br>
<span style="font-weight: bold;">GENCODE version 14</span>
corresponds to Ensembl 69 and Vega 49
<br>
<br>
<span style="font-weight: bold;">GENCODE version 7</span>
corresponds to Ensembl 62 and Vega 42 
and is used in ENCODE analysis.
</p>
<p>
See also: <a href="http://www.gencodegenes.org/releases" target="_blank">The GENCODE Project Release History</a>.
</p>

<!--#insert file="../../wgEncodeGencodeCredits1.shared.html"-->
