<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <title>wgEncodeProtGenoSuper.html</title>
</head>
<body>

<h2>Description</h2>
<p>These tracks display proteogenomomic mappings of peptides from <a
 href="http://www.genome.gov/10005107">ENCODE</a> cell
types, mapped to the human genome.  These peptides represent
fragments of proteins translated in the respective cell lines. 
Proteogenomic methods differ from conventional mass-spec proteomic
methods, which identify peptides by comparing them to peptides produced
from known proteins, and through this comparison identify the
translated proteins.  In contrast, proteogenomic methods compare
the peptides to all peptides that might theoretically be produced by
the six possible translation frames of the genome (three on the forward
strand plus three on the reverse strand) to identify the genomic region
from which the peptides were produced.  Study of proteogenomic
data offers insights on numerous regulatory mechanisms, including:
translation, pre-mRNA splicing and transcript diversity,
nonsense-mediated decay, and transcription of novel protein-coding
genes.<br>
</p>

<h2>Display Conventions and Configuration</h2>
<p>
Most ENCODE tracks contain multiple subtracks, corresponding to
multiple experimental conditions. If a track contains a large
number of subtracks, only some subtracks will be displayed by default.
The user can select which subtracks are displayed via the display
controls
on the track details pages.
</p>

<h2>Credits</h2>
<p>
These data were generated and analyzed as part of the <a
 href="http://www.genome.gov/10005107">ENCODE</a> project, a
genome-wide consortium project with the aim of cataloging all
functional elements in the human genome. This effort includes
collecting a variety of data across related experimental conditions, to
facilitate integrative analysis. Consequently, additional ENCODE tracks
may contain data that is relevant to the data in these tracks.
</p>

<h2>References</h2>
<p>
Castellana N, Bafna V.  <a
 href="http://www.ncbi.nlm.nih.gov/pubmed/20620248">Proteogenomics to
discover the full coding content of genomes: a computational
perspective</a>  
<span style="font-style: italic;">J Proteomics</span> 
2010 Oct 10;73(11):2124-35.</p>
<p>Krug K, Nahnsen S, Macek B. <a
 href="http://www.ncbi.nlm.nih.gov/pubmed/20967315">Mass spectrometry
at the interface of proteomics and genomics</a> 
<span style="font-style: italic;">Mol Biosyst.</span> 
2011 Feb;7(2):284-91.
</p>

<h2>Data Release Policy</h2>
<p>
Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset
until nine months following the release of the dataset. This date is
listed in the <span style="font-style: italic;">Restricted Until</span> 
column on the track configuration page and the download page. The full 
data release policy for ENCODE is available 
<a href="http://genome.ucsc.edu/ENCODE/terms.html">here</a>.
</p>
</body>
</html>
