<HR ALIGN='bottom' style='position:relative; top:1em;'><table class='windowSize'><tr valign='top'><td rowspan=2><A NAME='TRACK_HTML'></A>
<h2>Description</h2>
<p>
The CSHL small RNA track depicts <a target="_BLANK" href="../cgi-bin/hgEncodeVocab?term=%22shortTotal%22">short total RNA</a> sequencing data from <a target="_BLANK" href="../cgi-bin/hgEncodeVocab?type=cellType">ENCODE tissues</a> or <a target="_BLANK" href="../cgi-bin/hgEncodeVocab?type=localization">sub-cellular compartments</a> of <a target="_BLANK" href="../cgi-bin/hgEncodeVocab?type=cellType">ENCODE cell lines</a>. The protocol used to generate these data produced directional reads from the 5' end of short RNAs (< 200 nt). Libraries were sequenced using an Illumina GAIIx<!-- or HiSeq-->. These data were generated by Cold Spring Harbor Laboratories as a part of the ENCODE Consortium. The ENCODE project seeks to identify and characterize all functional elements in the human genome.  In many cases, there are Cap Analysis of Gene Expression (CAGE, see the <a target="_BLANK" href="../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRikenCage">RIKEN CAGE Loc</a> track), Long RNA-seq (>200 nucleotides, see the  <a target="_BLANK" href="../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlLongRnaSeq">CSHL Long RNA-seq</a> track) and Pair-End di-TAG-RNA (PET-RNA, see the <a target="_BLANK" href="../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisRnaPet">GIS RNA PET</a> track) datasets available from the same biological replicates.</p>

<p>
Many of the datasets produced by the Hannon lab (Generation 0 datasets) in Release 1 have been replaced by newly generated data from the Gingeras lab in Release 2. Of all Generation 0 datasets, only data from K562 and Prostate tissue are still displayed. All Generation 0 datasets are still available for <a target="_BLANK" href="../cgi-bin/hgFileUi?db=hg19&g=wgEncodeCshlShortRnaSeq">download</a>.</p>


<h2>Display Conventions and Configuration</h2>

<p>
This track is a multi-view composite track that contains multiple data types
(<em>views</em>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are <a href="/goldenPath/help/multiView.html" TARGET=_BLANK>here</a>.
To show only selected subtracks, uncheck the boxes next to the tracks that
you wish to hide.  Color differences among the views provide a visual cue for
distinguishing between the different cell types and compartments.</p>

<P>This track contains the following views:

<dl>
<dt><i>Contigs</i></dt>
<dd>The Contigs are BED format files representing blocks of overlapping mapped reads from pooled biological replicates. The corresponding number of mapped reads, the RPKM value, and the non-parametric IDR (npIDR) are reported for each contig.</dd>
<dt><i>Raw Signal</i></dt>
<dd>The Raw Signal view shows the density of mapped reads on the plus and minus strands (wiggle format).</dd>
</dl>

<P>Metadata for a particular subtrack can be found by clicking the down arrow in the list of subtracks.</P>

<h3>Downloadable Files</h3>
The following files can be found on the <a target="_BLANK" href="../cgi-bin/hgFileUi?db=hg19&g=wgEncodeCshlShortRnaSeq">Downloads Page</a>.

<dl>
<dt><i>Alignments</i></dt>
<dd>The Alignments contain individual reads mapped to the genome and indicates where bases may mismatch. Every mapped read is displayed. The alignments file follows the standard SAM format. See the <a target="_BLANK" href="http://samtools.sourceforge.net/SAM1.pdf#page=4">SAM Format Specification</a> for more information on the SAM/BAM file format.
</dd>
<dt><i>GENCODE V7</i></dt>
<dd>For each GENCODE V7 exon, the raw read count on the plus or minus strand, reads per million mapped reads (RPM) and the non-parametric IDR (npIDR) are reported in GFF format.</dd>
<dt><i>Transfrags</i></dt>
<dd>These data are now available for download only and have been replaced by Contigs. Small RNA reads were assembled into transcribed fragments, "Transfrags," by merging reads with one or more overlapping nucleotides. Only uniquely mapping reads were used to generate Transfrags. The BED6+ format of the transfrag files were created from the "intervals-to-contigs" Galaxy tool written by Assaf Gordon in the Hannon lab at CSHL. The transfrags data were not filtered.</dd>
<dt><i>Fastq</i></dt>
<dd>Raw sequence information is available in fastq format.</dd>
</dl>


<h2>Methods</h2>
<h3>Experimental Procedures</h3>
<p>Cells were grown according to the approved ENCODE <A href="/ENCODE/protocols/cell" target="_BLANK">cell culture protocols</a>. Short RNAs between 20-200 nt were isolated from total RNA using a Qiagen RNeasy kit (Qiagen #74204) according to manufacturer protocol. Purified small RNAs were treated with Ribominus (Invitrogen # A10837-08) according to manufacturer protocols to remove ribosomal RNA. The 5' structures were removed by treatment with Tobacco Alkaline Pyrophosphatase to allow for ligation of a 5' linker. RNA fragments were poly-adenylated using poly-A polymerase (or poly-cytidylated, in the case of Generation 0 data) and a 5' linker was ligated using T4 RNA ligase. An anchored oligo-dT was used to prime the reverse transcriptase reaction. cDNA was amplified using universal PCR primers introduced during the reverse transcriptase reaction. Resulting libraries were gel purified and used in cluster generation on an Illumina GAIIx<!-- or HiSeq-->. The libraries were sequenced as a single read from the 5' end of the inserts for a total of 36 <!--or 101--> cycles.
</p>
<p>
Complete protocols are available in the <a target="_BLANK" href="../cgi-bin/hgFileUi?db=hg19&g=wgEncodeCshlShortRnaSeq">Downloads Page</a>.
</p>



<h3>Data Processing and Analysis</h3>
<p>Data from the Gingeras and Guigo labs were preprocessed to remove experimentally derived poly-A tails and Illumina 3' linkers from raw reads. The best alignment to the Illumina 3' linker for each read was determined. If the number of mismatches in the alignment was less than 20% of the aligned length, the read was clipped from the first aligned base. Pre-processed reads were mapped using the STAR algorithm. For a description of STAR, the source code and mapping parameters used, see <a target="_BLANK" href="http://gingeraslab.cshl.edu/STAR/">the STAR project website</a>. Reads mapping 10 times or less  are reported in the Raw Signal and Alignment files.</p>

<p>Mapped reads were discarded if they fell into one of the following categories: 1) it contained five or more consecutive A's, 2) it was less than 16 nt in length, 3) it mapped to more than one genomic position (multiply-mapped reads), 4) it mapped upstream of genomically encoded poly-A sequences. The remaining reads were used both to call contigs and to produce expression values over GENCODE V7 exons. Contigs were generated from overlapping reads in pooled biological replicates.</p>


<h4>Generation 0 data</h4>
Reads were trimmed to discard any bases following a quality score less than or equal to 20 and converted into FASTA format, thereby discarding quality information for the rest of the pipeline. As a result, the sequence quality scores in the BAM output are all displayed as "40" to indicate no quality information. The read lengths may exceed the insert sizes and consequently introduce 3' adapter sequence into the 3' end of the reads. The 3' sequencing adapter was removed from the reads using a custom clipper program (available at <a href="http://hannonlab.cshl.edu/fastx_toolkit" target="_BLANK">http://hannonlab.cshl.edu/fastx_toolkit/</a>), which aligned the adapter sequence to the short-reads using up to 2 mismatches and no indels. Regions that aligned were clipped off from the read. Terminal C nucleotides introduced at the 3' end of the RNA via the cloning procedure were also trimmed. Reads were aligned to the human genome (version hg19, using the gender build appropriate to the sample in question -  <a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/referenceSequences/female.hg19.fa.gz" target="_BLANK">female</a>/<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/referenceSequences/male.hg19.fa.gz" target="_BLANK">male</a>) using Bowtie (Langmead B <em>et al</em>., 2009). Reads that mapped 20 or fewer times with 2 or less mismatches were reported. See Release Notes for more information on Generation 0 datasets.


<h2>Verification</h2>

<p> The mapped data were visually inspected to verify the majority of the reads were mapping the 5' ends of annotated small RNA classes.
</p>

<h2>Release Notes </h2>
<p>
Update to Release 2 (July 2012): Alignments are no longer being displayed. They are now available as downloads only.
</p>
<p>
This is Release 2 (February 2012) of CSHL Small RNA-seq with new data from the Gingeras lab. It includes ten new cell lines (A549, AG04450, BJ, H1-hESC, HeLa-S3, HepG2, HUVEC, MCF-7, NHEK, and SK-N-SH_RA), a new displayed view (Contigs), and two new download files (GENCODE Predicted Exons and a Protocol Document).
</p>

<P>
Release 1 contained data produced by the Hannon lab that was remapped from hg18 and indicated in this release as Generation 0 since the older data had no replicates. When there is both Generation 0 and new data available, only the new data is displayed. The older data is available for download only. From the original eleven experiments displayed with Release 1, only two (prostate and K562 polysome) are still displayed. The new data for this track were created with a different process and have standard replicate numbers. The replicate labeling in the genome browser view is a counter indicating the total number of replicates submitted. The producing lab has replicate numbers that correspond to their internal bio-replicate numbering. Where these two numbering systems conflict, both are listed in the long label of the specific track. 
</P>

<p>
<em><strong>Discrepancies between hg18 and hg19 versions of Generation 0 CSHL small RNA data:</em></strong> The alignment pipeline for the CSHL small RNA data was updated upon the release of the human genome version hg19, resulting in a few noteworthy discrepancies with the hg18 dataset.  First, mapping was conducted with the open-source Bowtie algorithm (<a href="http://bowtie-bio.sourceforge.net/index.shtml" target="_BLANK">http://bowtie-bio.sourceforge.net/index.shtml</a>) rather than the custom NexAlign software.  As each algorithm uses different strategies to perform alignments, the mapping results may vary even in genomic regions that do not differ between builds.  The read processing pipeline also varies slightly in that we no longer retain information regarding whether a read was clipped off an adapter sequence. 
</p>



<h2>Credits</h2>
Hannon lab members: Katalin Fejes-Toth, Vihra Sotirova, Gordon  Assaf, Jon Preall</p>
<p>Gingeras and Guigo laboratories: Carrie A. Davis, Lei-Hoon See, Wei Lin </p>

<h2>Contacts:</h2> 
<dl>
<dd><a href="mailto:&#106;&#112;r&#101;&#97;&#108;&#108;&#64;&#99;&#115;&#104;l.
&#101;&#100;&#117;">Jonathan Preall</a> (Generation 0 Data from Hannon Lab) 
<!-- above address is jpreall at cshl.edu -->
</dd>
<dd>
<a href="mailto:&#100;a&#118;&#105;s&#99;&#64;&#99;sh&#108;.&#101;&#100;&#117;">Carrie Davis</a> (experimental)
<!-- above address is davisc at cshl.edu -->
</dd>
<dd>
<a href="mailto:&#100;&#111;&#98;i&#110;&#64;&#99;&#115;&#104;l.&#101;d&#117;">Alex Dobin</a> (computational)
<!-- above address is dobin at cshl.edu -->
</dd>
<dd>
<a href="mailto:&#119;&#108;&#105;&#110;&#64;&#99;s&#104;&#108;.&#101;&#100;&#117;">Wei Lin</a> (computational)
<!--above address is wlin at cshl.edu -->
</dd>
<dd>
<a href="mailto:&#103;&#105;&#110;&#103;e&#114;a&#115;&#64;&#99;&#115;&#104;&#108;.&#101;d&#117;">Tom Gingeras</a> (primary investigator)
<!--above address is gingeras at cshl.edu -->
</p>
</dd>
</dl>

<h2>References</h2>

<p>
Affymetrix ENCODE Transcriptome Project, Cold Spring Harbor Laboratory ENCODE Transcriptome Project.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/19169241" target="_blank">
Post-transcriptional processing generates a diversity of 5'-modified long and short RNAs</a>.
<em>Nature</em>. 2009 Feb 19;457(7232):1028-32.
</p>

<p>
Langmead B, Trapnell C, Pop M, Salzberg SL.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/19261174" target="_blank">
Ultrafast and memory-efficient alignment of short DNA sequences to the human genome</a>.
<em>Genome Biol</em>. 2009;10(3):R25.
</p>

<!--<p>Mortazavi A, Williams BA, McCue K, Schaeffer L, and Wold BJ. 
<a target="_BLANK" href="http://www.ncbi.nlm.nih.gov/pubmed/18516045">Mapping and quantifying mammalian transcriptomes by RNA-seq</a>.
<em>Nature Methods</em>. 2008 Jul; 5(7):621-628.</p> 
-->

<h2> Data Release Policy </h2>
<p>Data users may freely use ENCODE data, but may not, without prior 
consent, submit publications that use an unpublished ENCODE dataset until 
nine months following the release of the dataset.  This date is listed in 
the <em>Restricted Until</em> column in the track configuration page and the 
download page.  The full data release policy 
for ENCODE is available 
<a href="../ENCODE/terms.html" TARGET=_BLANK>here</a>.</p> 

