<h2>Description</h2>

<p>
This track maps genome-wide human transcription factor binding sites using
second-generation massively parallel sequencing. This mapping uses expressed
transcription factors as GFP-tagged fusion proteins after bacterial artificial
chromosome (BAC) recombineering (recombination-mediated genetic engineering).
</p>

<p>
The University of Chicago and Max Planck Institute (Dresden) pipeline generates
recombineered BACs for the production of cell lines or animals that express fusion
proteins from epitope-tagged transgenes.
</p>

<h2>Display Conventions and Configuration</h2>

<p>
This track is a multi-view composite track that contains multiple data types
(<em>views</em>). For each view, there are multiple subtracks that display
individually on the browser. Instructions for configuring multi-view tracks
are <a href="/goldenPath/help/multiView.html" target="_blank">here</a>.
</p>

<p>
For each cell type, this track contains the following views:
<dl>
<dt><i>Peaks</i>
<dd>Regions of signal enrichment based on processed data (usually normalized data from pooled replicates).
<dt><i>Signal</i>
<dd>Density graph (wiggle) of signal enrichment based on aligned read density.
</dl>
</p>

<p>
Peaks and signals displayed in this track are the results of pooled replicate
sequence. Alignment files for each replicate are available for
<a href="/cgi-bin/hgFileUi?g=wgEncodeUchicagoTfbs" target="_blank">download</a>.
</p>

<h2>Methods</h2>

<p>
Cells were grown according to the approved
<a href="/ENCODE/protocols/cell" target="_blank">ENCODE cell culture protocols</a>.
</p>

<h3>Recombineering Strategy</h3>

<p>
To facilitate high-throughput production of the transgenic constructs, the
program BACFinder (Crowe <em>et al</em>., 2002) automatically selects the
most suitable BAC clone for any given human gene and generates the sets of
PCR primers required for tagging and verification (Poser <em>et al</em>.,
2008). Recombineering is used for tagging cassettes at either the N or C
terminus of the protein. The N-terminal cassette has a dual eukaryotic-prokaryotic
promoter (PGK-gb2) driving a neomycin-kanamycin resistance gene within an
artificial intron inside the tag coding sequence. The selection cassette is
flanked by two loxP sites and can be permanently removed by Cre
Recombinase-mediated excision. The C-terminal cassette contains the sequence
encoding the tag followed by an internal ribosome entry site (IRES) in front
of the neomycin resistance gene. In addition, a short bacterial promoter (Gb3)
drives the expression of the neomycin-kanamycin resistance gene in <em>E. coli</em>.
</p>

<p>
The tagging cassettes, containing 50 nucleotides of PCR-introduced homology arms,
were inserted into the BAC by recombineering, either behind the start codon (for
the N-terminal tag) or in front of the stop codon (for the C-terminal tag) of the
gene. <em>E. coli</em> cells that had successfully recombined the cassette were
selected for kanamycin resistance in liquid culture.  Each saturated culture from
a specific recombineering reaction derived 10-200 independent recombination events.
</p>

<p>
Two independent clones were checked for each PCR through the tag insertion point and
97% (85/88) yielded a PCR product of the expected size. Most of the clones that failed
to grow were missing the targeted genomic region. An estimated 10% of the BACs used
were chimeric, rearranged or wrongly mapped. Thus, initial results indicated that the
necessary recombineering steps could be carried out with high fidelity.
</p>

<p>
The White lab produced all epitope tagged transcription and chromatin factor BACs,
as well as the genome-wide ChIP data and analysis.  An application of this approach
to the analysis of closely related paralogs (RARa and RARg) yielded transcription
factors, chromatin factors, cell lines, ChIP-chip data and ChIP-seq data (Hua
<em>et al</em>., 2009). Such paralogous transcription factors often cannot otherwise
be distinguished by antibodies.
</p>
 
<h3>Sample Preparation</h3>

<p>
ChIP DNA from samples were sheared to approximately 800 bp using a nebulizer. The ends of the DNA
were polished and two unique adapters were ligated to the fragments. Ligated fragments of
150-200 bp were isolated by gel extraction and amplified using limited cycles of PCR.
</p>

<h3>Sequencing System</h3>

<p>
Illumina GAIIx and HySeq next-generation sequencing were used to produce all ChIP-seq data.
</p>

<h3>Processing and Analysis Software</h3>

<p>
Raw sequencing reads were aligned using
<a href="http://bowtie-bio.sourceforge.net/index.shtml" target="_blank">Bowtie</a> 0.12.5
(Langmead <em>et al</em>., 2009). The &quot;-m 1&quot; parameter was applied to suppress
alignments mapping more than once in the genome. Reads were aligned to the UCSC hg19 assembly.
Wiggle format signal files were generated with
<a href="http://compbio.med.harvard.edu/Supplements/ChIP-seq/" target="_blank">SPP</a> 2.7.1
(Kharchenko <em>et al</em>., 2008) for R 2.7.1.
<a href="http://liulab.dfci.harvard.edu/MACS/index.html" target="_blank">MACS</a> 1.3.7
was used to call peaks. The MACS parameters used varied by experiment.
</p>

<p>
The White lab used goat anti-GFP antibody to perform ChIP in untagged K562 cells as a background control.
The test IP was performed in the same manner as the background control. Results were expressed as values
of the test normalized to the background.
</p>

<h2>Credits</h2>

<p>
These data and annotations were created by a collaboration of University of Chicago and Argonne National Laboratory:
</p>

<ul>
<li><a href="mailto:&#107;p&#119;h&#105;&#116;&#101;&#64;&#117;c&#104;&#105;&#99;a&#103;&#111;.
e&#100;&#117;">
Kevin White</a> (Principal Investigator at University of Chicago)</li>
<!-- above address is kpwhite at uchicago.edu -->

<li><a href="mailto:&#115;&#117;&#98;&#104;ra&#100;&#105;&#112;&#64;&#117;&#99;h&#105;&#99;&#97;g&#111;.
&#101;&#100;&#117;">
Subhradip Karmakar</a> (Project Lead at University of Chicago)</li>
<!-- above address is subhradip at uchicago.edu -->

<li><a href="mailto:&#110;&#98;&#105;l&#100;&#64;&#98;&#115;&#100;.
u&#99;h&#105;&#99;&#97;&#103;&#111;.
&#101;&#100;&#117;">
Nick Bild</a> (Data Analyst at University of Chicago)</li>
<!-- above address is nbild at bsd.uchicago.edu -->

<li><a href="mailto:&#97;&#99;ho&#117;&#100;&#104;&#117;&#114;&#121;&#64;&#117;&#99;&#104;&#105;&#99;&#97;go.
&#101;&#100;u">
Alina Choudhury</a> (Laboratory Technician at University of Chicago)</li>
<!-- above address is achoudhury at uchicago.edu -->
 
<li><a href="mailto:&#109;&#100;&#111;&#109;&#97;n&#117;s&#64;&#97;&#110;&#108;.
&#103;&#111;v">
Marc Domanus</a> (Sequencing Technician at Argonne National Lab)</li>
<!-- above address is mdomanus at anl.gov -->
</ul>

<h2>References</h2>

<p>
Crowe ML, Rana D, Fraser F, Bancroft I, Trick M.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/12409477" target="_blank">
BACFinder: genomic localisation of large insert genomic clones based on restriction fingerprinting</a>.
<em>Nucleic Acids Res</em>. 2002 Nov 1;30(21):e118.
</p>

<p>
Hua S, Kittler R, White KP.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/19563758" target="_blank">
Genomic antagonism between retinoic acid and estrogen signaling in breast cancer</a>.
<em>Cell</em>. 2009 Jun 26;137(7):1259-71.
</p>

<p>
Kharchenko PV, Tolstorukov MY, Park PJ.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/19029915" target="_blank">
Design and analysis of ChIP-seq experiments for DNA-binding proteins</a>.
<em>Nat Biotechnol</em>. 2008 Dec;26(12):1351-9.
</p>

<p>
Langmead B, Trapnell C, Pop M, Salzberg SL.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/19261174" target="_blank">
Ultrafast and memory-efficient alignment of short DNA sequences to the human genome</a>.
<em>Genome Biol</em>. 2009;10(3):R25.
</p>

<p>
Poser I, Sarov M, Hutchins JR, H&#233;rich&#233; JK, Toyoda Y, Pozniakovsky A, Weigl D, Nitzsche A, Hegemann B, Bird AW <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/18391959" target="_blank">
BAC TransgeneOmics: a high-throughput method for exploration of protein function in mammals</a>.
<em>Nat Methods</em>. 2008 May;5(5):409-15.
</p>

<h2>Data Release Policy</h2>

<p>
Data users may freely use ENCODE data, but may not, without prior consent,
submit publications that use an unpublished ENCODE dataset until nine months
following the release of the dataset.  This date is listed in the <em>Restricted Until</em>
column, above. The full data release policy for ENCODE is available
<a href="../ENCODE/terms.html" target="_blank">here</a>.
</p>
