<h2>Description</h2>
<p>
The GENCODE Genes track (version 7, May 2011) shows high-quality manual
annotations merged with evidence-based automated annotations across the entire
human genome generated by the
<a href="http://www.gencodegenes.org/" target="_blank">GENCODE project</a>.
The GENCODE gene set presents a full merge
between HAVANA manual annotation and ENSEMBL automatic annotation. 
Priority is given to the manually curated HAVANA annotation, using predicted
ENSEMBL annotations when there are no corresponding manual annotations.
The annotation was carried out on genome assembly GRCh37 (hg19).
</p>
<p>
<B>NOTE</B>: Due to UCSC Genome Browser using the NC_001807 mitochondrial genome sequence
(chrM) and GENCODE annotating the NC_012920 mitochondrial sequence, the
GENCODE mitochondrial sequences are not available in the UCSC Genome Browser.
These annotations are available for download in the GENCODE GTF files.
</p>

<p>
<B>NOTE</B>: We try and synchronize the release cycles for GENCODE, Havana and
Ensembl. This GENCODE version 7 corresponds to Ensembl 62 from 13 April 2011
and Vega 23-03-2011.  Also see:
<a href="http://www.gencodegenes.org/" target="_blank">GENCODE project</a>.
</p>


<h2> Display Conventions and Configuration </h2>
<p>
The annotations are divided into separate tracks based on type of annotation.
The <em>basic set</em> of coding and non-coding transcripts is a subset of
the <em>comprehensive set</em> selected to provide a simplified view of the transcript
set designed to suit the needs of a majority of users.  The selection algorithm is described in
the <a href="#basicSetSelection">next section</a>.
The available tracks are:

</p>
<ul>
  <li><em>GENCODE Basic set</em> - subset of the GENCODE coding and non-coding
    transcript annotations, including polymorphic pseudogenes.  This includes
    both manual and automatic annotations.  The selection criteria is
    described in the <a href="#basicSetSelection">next section</a>.
    This is subset of the <em>comprehensive set</em>.

  <li><em>GENCODE Comprehensive set</em> - all GENCODE coding and non-code transcript annotations,
    including polymorphic pseudogenes.  This includes both manual and
    automatic annotations, except pseudogenes.  This is super-set of
    the <em>basic set</em>.
  <li><em>GENCODE Pseudogenes</em> - all pseudogene annotations except polymorphic pseudogenes

  <li><em>GENCODE 2-way Pseudogenes</em> - Pseudogenes predicted by the Yale
    Pseudopipe and UCSC Retrofinder pipelines, but not annotated by the HAVANA
    manual annotation process.  The set was derived by looking for a 50 base pairs
    of overlap between pseudogenes derived from both sets based on their genomic
    locations i.e. chromosomal coordinates.  When multiple Pseudopipe
    predictions map to a single Retrofinder prediction, only one match is kept
    for the 2-way consensus set.
  <li><em>GENCODE PolyA</em> - This track contains polyA signals and sites manually annotated on
    the genome based on transcribed evidence (ESTs and cDNAs) of 3' end of
    transcripts containing at least 3 A's not matching to the genome.
</ul>


<h4><a name="basicSetSelection">GENCODE <em>basic set</em> selection</a></h4>
<p>
The GENCODE <em>basic set</em> is intended to provide a simplified subset of
the GENCODE transcript annotations that will be useful to the majority of
users.  Selection for the GENCODE annotations to include in the <em>basic set</em>
is done on a per-locus basis and then for coding and non-coding transcripts
within that locus.  The goal is to use the better quality
transcript annotations while still having some annotation present for
each locus.
</p>

<p>The selection criteria for a given locus is:</p>
<ul>
  <li> Coding transcripts (including polymorphic pseudogenes):
    <ul>
      <li> If there are any full length coding transcripts that are not
        nonsense-mediated decay or problem transcripts, then only they are
        included in the basic set.
      <li> Otherwise, use the coding transcript with the largest CDS.
    </ul>
  <li> Non-coding transcripts:
    <ul>
      <li> If there are any full length non-coding transcripts and they have a
        well characterized BioType (see below), then only they are included in the
        basic set.
      <li> Otherwise, use the largest non-coding transcript.
    </ul>
</ul>

<h4>Non-coding transcript categories</h4>
<p>Non-coding transcripts are categorized using
their <a href="http://www.gencodegenes.org/gencode_biotypes.html">BioType</a>
and the following criteria:
</p>

<ul>
  <li> well characterized: <em>antisense, lincRNA, miRNA, Mt_rRNA, Mt_tRNA,
      rRNA, snoRNA, snRNA</em>
  <li> poorly characterized: <em>non_coding, processed_transcript,
      retrotransposed, misc_RNA</em>
</ul>
 
<h4>Filtering</h4>
<p>
Items in the GENCODE Basic, Comprehensive and Pseudogene tracks
can be filter using the following criteria:
</p>
<ul>
  <li> Transcript class: Filter by the basic biological function of a transcript
    annotation.
    <ul>
     <li> All - Don't filter by transcript class.
     <li> coding - Display protein coding transcripts, including polymorphic pseudogenes.
     <li> nonCoding - Display non-protein coding transcripts.
     <li> pseudo - Display pseudogene transcript annotations.
     <li> problem - Display problem transcripts.
   </ul>

  <li> Annotation Method: Filter by the method used to create the annotation.
   <ul>
     <li> All - Don't filter by transcript class.
     <li> manual - display manually created annotations, including those that are 
       also created automatically.
     <li> automatic - display automatically created annotations, including those that are 
       also created manually.
     <li> manual_only - display manually created annotations there were
       not annotated by the automatic method.
     <li> automatic_only - display automatically created annotations there were
       not annotated by the manual method.
   </ul>

  <li> Transcript Type: filter transcripts by <a href="http://www.gencodegenes.org/gencode_biotypes.html">BioType</a>.
</ul>

<h4>Coloring</h4>
<p>
The gene annotations are colored based on the annotation type:
</p>
<table border="1">
  <tr><th>Manual and automatic
      <td><font color="#0c0c78">coding</font>
      <td><font color="#006400">non-coding</font>
      <td><font color="#ff33ff">pseudogene</font>
      <td><font color="#fe0000">problem</font>
  </tr>
  <tr><th>2-way pseudogene
      <td colspan="4"><font color="#ff33ff">all</font>
  </tr>
  <tr><th>PolyA annotations
      <td colspan="4"><font color="#000000">all</font>
  </tr>
</table>

<h2>Methods</h2>

<p>
We aim to annotate all evidence-based gene features at high accuracy on
the human reference sequence. This includes identifying all
protein-coding loci with associated alternative variants, non-coding
loci which have transcript evidence, and pseudogenes. We integrate
computational approaches (including comparative methods), manual
annotation and targeted experimental verification.
</p>
<p>For a detailed description of the methods and references used, see
Harrow <em>et al</em> (2006).
</p>

<h2>Verification</h2>

<p> See Harrow <em>et al.</em> (2006) for information on verification
techniques.
</p>

<h2>Credits</h2>

<p> This GENCODE release is the result of a collaborative effort among
the following laboratories: (contact: <A HREF="mailto:&#103;&#101;n&#99;o&#100;&#101;&#64;&#115;an&#103;&#101;&#114;.
&#97;&#99;.
u&#107;">GENCODE at the Sanger Institute</A>.
<!-- above address is gencode at sanger.ac.uk -->)
</p>

<table style="text-align: left; width: 100%;" border="1" cellpadding="2" cellspacing="2">
  <tbody>
    <tr>
      <td style="vertical-align: top; text-align: left;"><b>Lab/Institution</b></td>
      <td style="vertical-align: top; text-align: left;"><b>Contributors</b></td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;"><a href="http://www.sanger.ac.uk/research/faculty/thubbard/" target="_blank">GENCODE Principal Investigator</a></td>
      <td style="vertical-align: top;">Tim Hubbard</td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;"><a href="http://www.sanger.ac.uk/HGP/havana/havana.shtml" target="_blank">HAVANA manual annotation group</a>,
          Wellcome Trust Sanger Insitute (WTSI), Hinxton, UK</td>
      <td style="vertical-align: top;">Adam Frankish, Jose Manuel Gonzales, Mike Kay, Alexandra Bignell, 
        Gloria Despacio-Reyes, Garaub Mukherjee, Gary Sanders, Veronika Boychenko, Jennifer Harrow</td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;"><a href="http://genome.imim.es/" target="_blank">Genome Bioinformatics Lab (CRG)</a>,
          Barcelona, Spain</td>
      <td style="vertical-align: top;">Thomas Derrien, Tyler Alioto, Roderic Guig&oacute;</td>
    </tr>
    <tr>
      <td><a href="http://genome.ucsc.edu/" target="_blank">Genome Bioinformatics, University of California Santa Cruz</a> (UCSC), USA</td>
      <td>Rachel Harte, Mark Diekhans, Robert Baertsch, David Haussler</td>
    </tr>
    <tr>
      <td><a href="http://mblab.wustl.edu/" target="_blank">Comp. Genomics Lab, Washington University</a> St. Louis (WUSTL), USA</td>
      <td>Jeltje van Baren, Charlie Comstock, David Lu, Michael Brent</td>
    </tr>
    <tr>
      <td><a href="http://www.csail.mit.edu/" target="_blank">Computer Science and Artificial Intelligence Lab</a>,
          <a href="http://www.broad.mit.edu/" target="_blank">Broad Institute of MIT and Harvard</a>, USA</td>
      <td>Mike Lin, Manolis Kellis</td>
    </tr>
    <tr>
      <td><a href="http://medicine.yale.edu/bbs/computational/index.aspx" target="_blank">Computational Biology and Bioinformatics, Yale University</a> (Yale), USA</td>
      <td>Philip Cayting, Mark Gerstein</td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;"><a href="http://www.unil.ch/cig" target="_blank">Center for Integrative Genomics</a>,
          University of Lausanne, Switzerland</td>
      <td style="vertical-align: top;"> Cedric Howald, Alexandre Reymond</td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;"><a href="http://www.ensembl.org/info/docs/genebuild/genome_annotation.html" target="_blank">ENSEMBL genebuild group</a>,
          Wellcome Trust Sanger Insitute (WTSI), Hinxton, UK</td>
      <td style="vertical-align: top;">Steve Searle, Bronwen Aken, Amonida Zadissa, Daniel Barrell
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;"><a href="http://www.cnio.es/ing/grupos/plantillas/presentacion.asp?pag=726" target="_blank">
Structural Computational Biology Group</a>, Centro Natcional de Investigaciones Oncologicas (CNIO), Madrid, Spain</td>
      <td>Jos&eacute; Manuel Rodr&iacute;guez, Michael Tress, Alfonso Valencia</td>
    </tr>
  </tbody>
</table>

<h2>References</h2>

<p> Flicek <em>et al</em>.
<a href="http://nar.oxfordjournals.org/content/39/suppl_1/D800.full" target="_blank">Ensembl 2011</a>.
<em>Nucleic Acids Research</em>. 2011;39 Database issue:D800-D806
</p>

<p> Harrow J, Denoeud F, Frankish A, Reymond A, Chen CK, Chrast J, Lagarde J, Gilbert JG, Storey R, Swarbreck D <em>et al</em>.
<a href="http://genomebiology.com/2006/7/S1/S4" target="_blank">GENCODE: producing a reference annotation
for ENCODE</a>. <em>Genome Biol</em>. 2006;7 Suppl
1:S4.1-9.</p>


<h2> Data Release Policy </h2>

<p>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column on the track configuration page and
the download page.  The full data release policy for ENCODE is available
<a HREF="../ENCODE/terms.html" TARGET=_BLANK>here</a>.</p>

<!--  LocalWords:  pseudogenes Pseudopipe Retrofinder GENCODE
 -->
