<H2>Description</H2>
<P>

This track displays human tissue microarray data using <A HREF="http://www.affymetrix.com/estore/browse/products.jsp?navMode=34000&productId=131452&navAction=jump&aId=productsNav#1_1" 
TARGET=_blank>Affymetrix Human Exon 1.0</A> 
ST expression arrays. This RNA expression track was produced as part of the
ENCODE Project. The RNA was extracted from cells that were also analyzed by DNaseI hypersensitivity, (<a href="../../cgi-bin/hgTrackUi?g=wgEncodeOpenChromDnase" TARGET=_BLANK>Duke DNaseI HS</A>), FAIRE (<a href="../../cgi-bin/hgTrackUi?g=wgEncodeOpenChromFaire" TARGET=_BLANK>UNC FAIRE</A>), and ChIP (<a href="../../cgi-bin/hgTrackUi?g=wgEncodeOpenChromChip" TARGET=_BLANK>UTA TFBS</A>).
</P>

<H2>Display Conventions and Configuration</H2>
<P>
In contrast to the hg18 annotation, this track now displays exon array data
that has been aggregated to the gene level for those probes that have been
linked to genes. Probes not linked to genes are displayed at the probe level.
The display for this track shows gene probe location and signal value as
grayscale-colored items where higher signal values correspond to darker-colored
blocks.
</P>

<P>
Items with scores between 900-1000 are in the highest 10% quantile for signal
value of that particular cell type. Similarly, items scoring 800-900 are the
next 10% quantile and at the bottom of scale, items scoring 100-200 are in the
lowest 20% quantile for signal value.
</P>
<P>
The subtracks within this composite annotation track correspond to data from different
cell types and tissues. The configuration options are shown at the top of the track
description page, followed by a list of subtracks. To display only selected subtracks,
uncheck the boxes next to the tracks you wish to hide.
</P>
<P>
For information regarding specific microarray probes, turn on the <a
href="../../cgi-bin/hgTrackUi?g=affyExonArray" TARGET=_BLANK>Affy Exon Probes</A> track, which 
can be found in the <B>Expression</B> track group.  See Methods for a description
as to how probe level data was processed to produce gene level annotations.
</P>

<H2>Methods</H2>
<P>

Cells were grown according to the approved
 <A HREF="../../ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture protocols</A>.
Total RNA was isolated from these cells using trizol extraction followed by
cleanup on RNEasy column (Qiagen) that included a DNaseI step. The RNA was
checked for quality using a nanodrop and an Agilent Bioanalyzer. RNA
(1 &micro;g) deemed to be of good quality was then processed according to the
standard Affymetrix Whole transcript Sense Target labeling protocol that
included a riboreduction step. The fragmented biotin-labeled cDNA was
hybridized over 16 h to Affymetrix Exon 1.0 ST arrays and scanned on an
Affymetrix Scanner 3000 7G using AGCC software.
</P>
<P>
Exon expression analyses were carried out using Affymetrix Expression Console
1.1 software tools. All samples were processed together using the sketch-quantile
normalization method. Gene level analysis was done at the "extended" annotation
confidence level. Probes were summarized with median polish as used in RMA, and
the background was corrected with PM and RMA background correction options. And
all library files used were default library files. Gene names were assigned based on
the output annotation file from this software. The accession number for the annotated
transcript was used if a gene name was not assigned. If neither was present, the probe
number was used.
</P>

<H2>Verification</H2>
<P>
Data were verified by analyzing biological replicates displaying Pearson correlation
coefficient &gt; 0.9, when replicates were available.
</P>

<H2>Release Notes</H2>
<P>
This is release 1 of this track for the hg19 assembly. This initial release
includes 45 distinct cell types and conditions with most having multiple
replicates. Data from 26 of these cell types were previously released on the
hg18 assembly.
</P>

<H2>Credits</H2>

<P>
RNA was extracted from each cell type by Greg Crawford's group at Duke University.
RNA was purified and hybridized to Affymetrix Exon arrays by Sridar Chittur and
Scott Tenenbaum at the University of Albany-SUNY. Data analyses were performed by
Zhancheng Zhang (UNC Chapel Hill) and Darin London (Duke University).
</P>

<p>Contact: <A HREF="mailto:&#116;&#101;&#114;&#114;&#121;.
&#102;&#117;rey&#64;&#100;&#117;&#107;e.
&#101;&#100;&#117;">
Terry Furey
</A></p>

<H2>Data Release Policy</H2> 
<P>
Data users may freely use ENCODE data, but
may not, without prior consent, submit publications that use an
unpublished ENCODE dataset until nine months following the release of
the dataset. This date is listed in the Restricted Until column,
above. The full data release policy for ENCODE is available
<A href="http://genome.ucsc.edu/ENCODE/terms.html">here</A>.
</P>
