<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <title>wgEncodeHistoneSuper.html</title>
</head>
<body>
<h2>Description</h2>
<p>These tracks display maps of chromatin state
of <a href="http://www.genome.gov/10005107">ENCODE</a> cell
types. Histones are a group of closely-related proteins that together
form the nucleosome. There are six histone families: H1, H2A, H2B,
H3, H4, and H5.  Each nucleosome core is formed by two H2A-H2B dimers and
a H3-H4 tetramer, while H1 and H5 bind the nucleosome and DNA.  When
histone proteins undergo post-transcriptional modifications at specific
amino acids (methylation, acyetylation), these modifications (referred
to as <em>marks</em>) perform a role in regulating the winding of the
DNA around the nucleosome. This in turn regulates gene expression by
controlling the accessibility of the chromatin. Histone marks vary in
their effect. Promoters, enhancers, transcribed regions, and silenced
regions are each associated with specific histone marks.</p>
<p>
The ChIP-seq method enables identifying regions of DNA that interact
with specific histone marks. This method involves cross-linking
histones and other DNA associated proteins to genomic DNA within cells
using formaldehyde. The cross-linked chromatin is subsequently
extracted, mechanically sheared, and immunoprecipitated using specific
antibodies. This has the effect of capturing segments of DNA that are
bound to the protein selected by the antibody. After reversal of
cross-links, the immunoprecipitated DNA is sequenced and mapped to the
human reference genome. The relative enrichment of each antibody-target
(epitope) across the genome is inferred from the density of mapped
fragments. If the antibody is specific to a given histone mark, then
this procedure identifies segments of the genome associated with that
mark.</p>

<h2>Display Conventions</h2>
<p>
These tracks are multi-view composite tracks that contains multiple
data types (views). Each view within each track
has separate display controls, as described <a
 href="http://hgwdev.cse.ucsc.edu/goldenPath/help/multiView.html">here</a>.
Most ENCODE tracks contain multiple subtracks, corresponding to
multiple experimental conditions. If a track contains a large
number of subtracks, only some subtracks will be displayed by default.
The user can select which subtracks are displayed via the display controls
on the track details pages.</p>
<h2>Credits</h2>
<p>
These data were generated and analyzed as part of the <a
 href="http://www.genome.gov/10005107">ENCODE</a> project, a
genome-wide consortium project with the aim of cataloging all
functional elements in the human genome. This effort includes
collecting a variety of data across related experimental conditions, to
facilitate integrative analysis.  Consequently, additional ENCODE tracks may 
contain data that is relevant to the data in these tracks.</p>
<h2>References</h2>
<p>
Geiman TM, Robertson KD. <a href="http://www.ncbi.nlm.nih.gov/pubmed/12244565">Chromatin remodeling, histone
modifications, and DNA methylation-how does it all fit together?</a>
<span style="font-style: italic;">J Cell Biochem</span>. 2002;87(2):117-25.</p>
<p>
Segal E, Widom J. <a href="http://www.ncbi.nlm.nih.gov/pubmed/19596482">
What controls nucleosome positions?</a><span style="font-style: italic;">Trends Genet</span>. 2009
Aug;25(8):335-43.</p>

<h2>Data Release Policy</h2>
<p>
Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset
until nine months following the release of the dataset. This date is
listed in the Restricted Until column on the track configuration page
and the download page. The full data release policy for ENCODE is
available <a href="http://genome.ucsc.edu/ENCODE/terms.html">here</a>.
</p>
</body>
</html>
