<H2>Description</H2>
<P>
These tracks display the level of sequence uniqueness of the reference <B>GRCh37/hg19</B>
genome assembly.  They were generated using different window sizes, and high signal
will be found in areas where the sequence is unique.

<H2>Methods</H2>
<P>
The <em>CRG Alignability</em> tracks display how uniquely k-mer sequences align
to a region of the genome. To generate the data, the GEM-mappability
<A HREF="http://gemlibrary.sourceforge.net" TARGET=_BLANK>program</A> has
been employed. The method is equivalent to mapping sliding windows of k-mers
(where k has been set to 36, 40, 50, 75 or 100 nts to produce these tracks)
back to the genome using the GEM mapper aligner (up to 2 mismatches were
allowed in this case). For each window, a mapability score was computed
(S = 1/(number of matches found in the genome): S=1 means one match in the
genome, S=0.5 is two matches in the genome, and so on). The
<em>CRG Alignability</em> tracks were
generated independently of the ENCODE project, in the framework of the GEM
(GEnome Multitool) project.

<H2>Credits</H2>
<P>
The <em>CRG Alignability</em> track was created by Thomas Derrien and <A HREF="mailto:&#112;&#97;&#111;l&#111;.
&#114;&#105;&#98;&#101;&#99;&#97;&#64;&#103;&#109;&#97;&#105;l.
&#99;o&#109;">Paolo
Ribeca</A>
<!-- above address is paolo.ribeca at gmail.com -->
 in <A HREF="http://pasteur.crg.es/portal/page/portal/Internet/02_Research/01_Programmes/01_Bioinformatics_Genomics"
TARGET=_BLANK>Roderic Guigo's lab</A> at the Centre for Genomic
Regulation (CRG), Barcelona, Spain. Thomas Derrien was supported by funds from NHGRI
for the ENCODE project, while Paolo Ribeca was funded by a Consolider grant
CDS2007-00050 from the Spanish Ministerio de Educaci&#243;n y Ciencia.
</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use all data in this track.  ENCODE labs that
contributed annotations have exempted the data displayed here from the
ENCODE data release policy restrictions.
