<H2>Description</H2>
<P><P>These tracks display a synthesis of evidence from different assays
as part of the four <B><em>Open Chromatin</em></B> track sets.
This track displays open chromatin regions and/or transcription factor binding
sites identified in
<A TARGET=ucscVocab HREF="/cgi-bin/hgEncodeVocab?type=cellType">multiple cell types</A>
by one or more complementary methodologies, DNaseI hypersensitivity (HS)
(<A HREF="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromDnase">Duke DNaseI HS</A>),
Formaldehyde-Assisted Isolation of Regulatory Elements (FAIRE)
(<A HREF="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromFaire">UNC FAIRE</A>),
and chromatin immunoprecipitation (ChIP) for select regulatory factors
(<A HREF="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromChip">UTA TFBS</A>).
Each methodology was performed on the same cell type using identical growth
conditions. (Note: Data for some or all ChIP experiments may not be available
for all cell types). Regions that overlap between methodologies identify
regulatory elements that are cross-validated indicating high confidence
regions.  In addition, multiple lines of evidence suggest that regions detected
by a single assay (e.g., DNase-only or FAIRE-only) are also biologically
relevant (Song et al., submitted).
<P><A HREF="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromDnase">DNaseI HS data</A>:
DNaseI is an enzyme that has long been used to map general
chromatin accessibility, and DNaseI "hypersensitivity" is a feature of active
cis-regulatory sequences.The use of this method has led to the discovery of
functional regulatory elements that include promoters, enhancers, silencers,
insulators, locus control regions, and novel elements. DNaseI hypersensitivity
signifies chromatin accessibility following binding of trans-acting factors in
place of a canonical nucleosome.
</P>
<P><A HREF="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromFaire">FAIRE data</A>:
FAIRE (Formaldehyde Assisted Isolation of Regulatory
Elements) is a method to isolate and identify nucleosome-depleted regions of
the genome.  FAIRE was initially discovered in yeast and subsequently shown to
identify active regulatory elements in human cells (Giresi <em>et al.</em>,
2007). Similar to DNaseI HS, FAIRE appears to identify functional regulatory
elements that include promoters, enhancers, silencers, insulators, locus
control regions and novel elements.
</P>
<P><A HREF="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromChip">ChIP data</A>:
ChIP (Chromatin Immunoprecipitation) is a method to
identify the specific location of proteins that are directly or indirectly
bound to genomic DNA. By identifying the binding location of sequence-specific
transcription factors, general transcription machinery components, and
chromatin factors, ChIP can help in the functional annotation of the open
chromatin regions identified by DNaseI HS mapping and FAIRE.
</P>
<P><A HREF="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromChip">Input data</A>:
As a background control experiment, we sequenced
the input genomic DNA sample that was used for ChIP. Crosslinked chromatin
is sheared and the crosslinks are reversed without carrying out the
immunoprecipitation step. This sample is otherwise processed in a manner
identical to the ChIP sample as described below. The input track is
useful in revealing potential artifacts arising from the sequence
alignment process such as copy number differences between the
reference genome and the sequenced samples, as well as regions of
poor sequence alignability.</P>

<H2> Display Conventions and Configuration </H2>
<P>This track contains multiple subtracks representing different cell types
that display individually on the browser. Instructions for configuring tracks
with multiple subtracks are
<A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
<P>To facilitate analyses, each region has been assigned an <em>Open Chromatin Code (OC Code)</em>,
based on the assay(s) by which it was detected, and a color, based on its level of validation
(which was determined by the combination of its OC Code and its statistical significance):
<DL>
<DT><B>Validated</B>, OC Code = 1:</DT>
<DD><ul style="list-style-type:none;">
<li><b>Black</b>:
Regions identified as peaks by both the DNaseI HS assay and FAIRE assay.  Peaks
for DNaseI HS have DNase peak calling p-values <code>&lt; 0.05</code>
(<code>-log<sub>10</sub>(p-value) &gt; 1.3</code>) and peaks for FAIRE have FAIRE
peak calling p-values <code>&lt;  0.1</code> (<code>-log<sub>10</sub>(p-value) &gt; 1.0</code>).
</li></DD>
</ul>
<br />
<DT><B>Open Chromatin</B>, OC Code = 2 or 3:</DT>
<DD><ul style="list-style-type:none;">
<li><span style='color: #0000FF;'><b>Blue</b></span> (high significance):
Regions not identified as peaks in both DNaseI HS and FAIRE, but for which the
combination of peak calling p-values from these assays using Fisher's combined
probability test results in a p-value <code>&lt; 0.01</code>
(<code>-log<sub>10</sub>(p-value) &gt; 2</code>).</li></DD>
</ul>
<br />
<DT><B>DNase</B>, OC Code = 2:</DT>
<DD><ul style="list-style-type:none;">
<li><b><span style='color: #009900;'>Green</span></b> (low significance):
Regions identified by DNaseI HS as peaks (DNase peak calling p-value <code>&lt; 0.05</code>
(<code>-log<sub>10</sub>(p-value) &gt; 1.3</code>))
and not identified by FAIRE as peaks (FAIRE peak calling p-value <code>&lt;  0.1</code>
(<code>-log<sub>10</sub>(p-value) &gt; 1.0</code>)),
and with a Fisher's combined DNaseI HS and FAIRE p-value <code>&gt;= 0.01</code>
(<code>-log<sub>10</sub>(p-value) &lt;= 2</code>).</li>
<li><b><span style='color: #0000FF;'>Blue</b></span> (high significance):
see Open Chromatin above.</li></DD>
</ul>
<br />
<DT><B>FAIRE</B>, OC Code = 3:</DT>
<DD><ul style="list-style-type:none;">
<li><b><span style='color: #990000;'>Dark Red</span></b> (low significance):
Regions identified by FAIRE as peaks (FAIRE peak calling p-value <code>&lt;  0.1</code>
(<code>-log<sub>10</sub>(p-value) &gt; 1.0</code>))
identified by DNaseI HS as peaks (DNase peak calling p-value <code>&lt; 0.05</code>
(<code>-log<sub>10</sub>(p-value) &gt; 1.3</code>)), and with a
Fisher's combined DNaseI HS and FAIRE p-value <code>&gt;= 0.01</code>
(<code>-log<sub>10</sub>(p-value) &lt;= 2</code>).</li>
<li><b><span style='color: #0000FF;'>Blue</b></span> (high significance):
see Open Chromatin above.</li></DD>
</ul>
<br />
<DT><B>ChIP-seq</B>, OC Code = 4:</DT>
<DD><ul style="list-style-type:none;">
<li><b><span style='color: #FF00FF;'>Pink</span></b>:
Regions identified by ChIP-seq as peaks (at least one of the peak calling p-values for
the three ChIP experiment is <code>&lt; 0.05 </code>(<code>-log<sub>10</sub>(p-value) &gt; 1.3</code>))
indicating binding sites for one
or more of RNA Pol II, CTCF, and c-Myc described here and not identified by DNaseI HS
or FAIRE as peaks. Peaks for ChIP-seq have p-values <code>&lt; 0.05</code>
(<code>-log<sub>10</sub>(p-value) &gt; 1.3</code>). For RNA Pol II, only sites that
overlap annotated transcription start sites by the UCSC Genes track are considered.
</li></DD>
</ul>
</DL>
</P>
<P>All signal values, <code>-log<sub>10</sub></code>(p-values), and the OC Code is
displayed on the detail page for each element and is available in the
corresponding bed file.
</P>

<H2>Methods</H2>
<P>For each site, the maximum F-Seq Density Signal value has been calculated
for each assay that was performed in that cell type. F-Seq employs Parzen
kernel density estimation to create base pair scores (Boyle et al., 2008b).
Significant regions, or peaks, were determined by fitting the data to a gamma
distribution to calculate p-values. Contiguous regions where p-values were
below a 0.05 (DNaseI HS, ChIP) or 0.1 (FAIRE) threshold were considered
significant. See assay specific description pages (
<A HREF="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromDnase">Duke DNaseI HS</A>,
<A HREF="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromFaire">UNC FAIRE</A> and
<A HREF="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromChip">UTA TFBS</A>)
for more details.

<P>A Fisher's Combined P-value for DNaseI HS and FAIRE was calculated using
Fisher's combined probability test.  First, a test statistic is calculated
using the formula:</P>
<P>
<PRE style='font-size:1.2em;'>    <B>X<sup>2</sup> = -2</B><span style='font-size:2.6em;'>&#8721;</span><B>log<sub>e</sub>(p<sub>i</sub>)</B></PRE>
</P><P>
where <B><code style='font-size:1.2em;'>p<sub>i</sub></code></B> are the p-values calculated for DNaseI HS
and FAIRE.  <B><code style='font-size:1.2em;'>X<sup>2</sup></code></B> follows a chi-squared distribution,
thus a combined p-value can be assigned to this test statistic.
</P>

<h2>Release Notes</h2>
<P>
This is release 1 (May 2011) of this track and is based upon the first release on <B>GRCh37/hg19</B>
of the three Open Chromatin tracks:
<A HREF="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromDnase">Duke DNaseI HS</A>,
<A HREF="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromFaire">UNC FAIRE</A>,
and
<A HREF="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromChip">UTA TFBS</A>.
</P><P>
<B>Enhancer and Insulator Functional assays:</B>  A subset of DNase and FAIRE
regions were cloned into functional tissue culture reporter assays to test for
enhancer and insulator activity.  Coordinates and results from these
experiments can be found
<A HREF="http://hgdownload-test.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeOpenChromDnase/supplemental/"
 TARGET=ucscDownloads>here</a>.  <!-- WARNING: Change to hgdownload.cse.ucsc.edu on release -->
</P>

<H2>Credits</H2>
<P>
These data and annotations were created by a collaboration of multiple
institutions (contact:
<A HREF="mailto:&#116;&#115;&#102;ur&#101;&#121;&#64;&#100;&#117;&#107;&#101;.e&#100;u">
Terry Furey</A>):
<!-- above address is tsfurey at duke.edu -->
</P><P>
<UL>
<LI>Duke University's <A HREF="http://www.genome.duke.edu/index.php"
TARGET=_blank>Institute for Genome Sciences & Policy (IGSP)</A>: Zhancheng Zhang,
Lingyun Song, <A HREF="http://www.genome.duke.edu/people/faculty/furey/"
TARGET=_blank>Terry Furey</A>, and
<A HREF="http://www.genome.duke.edu/people/faculty/crawford/"
TARGET=_blank>Greg Crawford</a>
</P><P>
<LI>University of North Carolina at Chapel Hill: Paul Giresi and
<A HREF="http://www.bio.unc.edu/faculty/lieb/labpages/default.shtml"
TARGET=_blank>Jason Lieb</a>
</P><P>
<LI>Universty of Texas at Austin: Zheng Liu, Ryan McDaniell, Bum-Kyu Lee, and
<A HREF="http://microarray.icmb.utexas.edu/research.html"
TARGET=_blank>Vishy Iyer</a>
</P><P>
<LI><A HREF="http://www.ebi.ac.uk/" TARGET=_blank>European Bioinformatics
Insitute</A>: Paul Flicek, Damian Keefe, and
<A HREF="http://www.ebi.ac.uk/~birney/" TARGET=_blank>Ewan Birney</a>
</P><P>
<LI><A HREF="http://www.oncology.cam.ac.uk/" TARGET=_blank>University of
Cambridge, Department of Oncology and CR-UK Cambridge Research Institute (CRI)
</A>: Stefan Graf
</UL>
</P><P>
We thank NHGRI for ENCODE funding support.
</P>

<H2>References</H2>

<P>
Bhinge AA, Kim J, Euskirchen GM, Snyder M, Iyer, VR.
<A HREF="http://genome.cshlp.org/cgi/content/abstract/17/6/910" TARGET=_blank>
Mapping the chromosomal targets of STAT1 by Sequence Tag Analysis of Genomic
Enrichment (STAGE)</A>. <EM>Genome Res.</EM> 2007 Jun;17(6):910-6.
</P><P>

Boyle AP, Davis S, Shulha HP, Meltzer P, Margulies EH, Weng Z, Furey TS,
Crawford GE. <A HREF="http://www.cell.com/abstract/S0092-8674(07)01613-3"
TARGET=_blank>High-resolution mapping and characterization of open chromatin
across the genome</A>. <EM>Cell.</EM> 2008 Jan 25;132(2):311-22.
</P><P>

Boyle AP, Guinney J, Crawford GE, and Furey TS.
<A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btn480"
TARGET=_blank>F-Seq: a feature density estimator for high-throughput sequence
tags</A>. <EM>Bioinformatics.</EM> 2008 Nov 1;24(21):2537-8.
</P><P>

Buck MJ, Nobel AB, Lieb JD.
<A HREF="http://genomebiology.com/2005/6/11/R97" TARGET=_blank>ChIPOTle: a
user-friendly tool for the analysis of ChIP-chip data</A>.
<EM>Genome Biol.</EM> 2005;6(11):R97.
</P><P>

Crawford GE, Davis S, Scacheri PC, Renaud G, Halawi MJ, Erdos MR, Green R,
Meltzer PS, Wolfsberg TG, Collins FS.
<A HREF="http://www.nature.com/nmeth/journal/v3/n7/abs/nmeth888.html"
TARGET=_blank>DNase-chip: a high-resolution method to identify DNase I
hypersensitive sites using tiled microarrays</A>.
<EM>Nat Methods.</EM> 2006 Jul;3(7):503-9.
</P><P>

Crawford GE, Holt IE, Whittle J, Webb BD, Tai D, Davis S, Margulies EH, Chen Y,
Bernat JA, Ginsburg D <EM>et al.</EM>
<A HREF="http://www.genome.org/cgi/content/abstract/16/1/123"
TARGET=_blank>Genome-wide mapping of DNase hypersensitive sites using massively
parallel signature sequencing (MPSS)</A>.
<EM>Genome Res.</EM> 2006 Jan;16(1):123-31.
</P><P>

The ENCODE Project Consortium.
<A HREF="http://www.nature.com/nature/journal/v447/n7146/abs/nature05874.html"
TARGET=_blank>Identification and analysis of functional elements in 1% of the
human genome by the ENCODE pilot project</A>. <EM>Nature.</EM>
2007 Jun 14;447(7146):799-816.
</P><P>

Giresi PG, Kim J, McDaniell RM, Iyer VR, Lieb JD.
<A HREF="http://www.genome.org/cgi/content/abstract/17/6/877" TARGET=_blank>
FAIRE (Formaldehyde-Assisted Isolation of Regulatory Elements) isolated active
regulatory elements in human chromatin</A>.
<EM>Genome Res.</EM> 2007 Jun;17(6):877-85.
</P><P>

Giresi PG, Lieb JD. <A HREF="http://www.ncbi.nlm.nih.gov/pubmed/19303047"
TARGET=blank>Isolation of active regulatory elements from eukaryotic chromatin
using FAIRE (Formaldehyde Assisted Isolation of Regulatory Elements)</A>.
<EM>Methods.</EM> 2009 Jul;48(3):233-9.
</P><P>

Li H, Ruan J, and Durbin R.
<A HREF="http://genome.cshlp.org/content/early/2008/09/25/gr.078212.108.short"
TARGET=_blank>Mapping short DNA sequencing reads and calling variants using
mapping quality scores</A>. <EM>Genome Res.</EM> 2008 Nov;18(11):1851-8.
</P><P>
Song L and Crawfor GE.
<A HREF="http://cshprotocols.cshlp.org/cgi/content/full/2010/2/pdb.prot5384"
TARGET=_BLANK>DNase-seq: a high-resolution technique for mapping active
gene regulatory elements across the genome from mammalian cells.</A>
<Em>Cold Spring Harb. Protoc.</EM>; 2010;Issue 2.

</P>

<H2> Data Release Policy </H2>
Data users may freely use ENCODE data, but may not, without prior consent,
submit publications that use an unpublished ENCODE dataset until nine months
following the release of the dataset.  This date is listed in the <EM>Restricted
Until</EM> column on the track configuration page and the download page.  The
full data release policy for ENCODE is available <A HREF="../ENCODE/terms.html"
TARGET=_BLANK>here</A>.
