<P>
The August 2005 <em>Drosophila virilis</em> genome assembly 
was produced by 
Agencourt Bioscience (now a part of <a href="http://www.beckmangenomics.com" target="_blank">Beckman Coulter Genomics</a>) and can be downloaded 
<A HREF="http://rana.lbl.gov/drosophila/assemblies/dvir_agencourt_arachne_01aug05.tar.gz" TARGET=_BLANK>here</A>.  
To see a list of the assembly's largest 1000 scaffolds ranked by size, click 
<A HREF="/cgi-bin/hgTracks?db=droVir2&chromInfoPage=">here</A>.
</P>

<H3>Sample position queries</H3>
<P>
A genome position can be specified by a scaffold coordinate range, the accession 
number of an mRNA or RefSeq entry, a gene name, or keywords from 
the GenBank description of an mRNA. The following list shows 
examples of valid position queries for the <em>D. virilis</em> genome. 
Note that some position queries (e.g. "huntington") may return matches to the 
mRNA records of other species. In these cases, the mRNAs are mapped to their 
homologs in <em>D. virilis</em>. 
See the
<A HREF="../goldenPath/help/hgTracksHelp.html" TARGET=_blank>User's Guide</A>
for more information.
</P>
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><BR></TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>scaffold_12472</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays all of scaffold_12472</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>scaffold_12472:1-100,000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays first 100,000 bases of scaffold_12472</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
<!--
    <TR><TD VALIGN=Top NOWRAP>AF419620</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of genome aligning to mRNA with GenBank accession AF419620</TD>
    </TR>
-->
    <TR><TD VALIGN=Top NOWRAP>CX159558</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of genome aligning to EST with GenBank accession CX159558</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>CG3727-RA</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of genome aligning to FlyBase accession CG3727-RA</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>dock</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of genome aligning to gene with identifier dock (a <em>D. melanogaster</em> gene)</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>NM_057677</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of genome aligning to RefSeq accession with identifier NM_057677 (a <em>D. melanogaster</em> RefSeq entry)</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>zinc finger 312</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists only zinc finger protein 312</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>huntington</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists candidate genes associated with Huntington's disease</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>cline</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Cline</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Strunkelnberg,M.</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author M. Strunkelnberg</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Strunkelnberg M</em>, internally it uses 
			the format <em>Strunkelnberg,M.</em></TD>
    </TR>
</TABLE>
</P>
<P>
<HR>
<H3>Assembly details</H3>
<P>
Agencourt Bioscience Corporation produced the 1 August 2005 assembly using the 
<A HREF="http://www.broad.mit.edu/wga/" TARGET=_BLANK>Arachne</A> assembler.  
The assembly contains 13,562 scaffolds ranging in size from 57 bases to 
25,269,527 bases, with a mean size of 15263.1 and median of 1249.  
</P>
<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/droVir2/">FTP server</A> or the 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#droVir">Downloads</A> page. 
The droVir2 annotation tracks were generated by UCSC and collaborators worldwide. 
See the <A HREF="../goldenPath/credits.html#droVir_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to this release. 
</P>
<HR>
<H3>Genbank Pipeline Details</H3>
For the purposes of the Genbank alignment pipeline, this assembly is considered to be: <B>low-coverage</B>.
