<H2>Display</H2>
<P>This track displays alignments of the Feb. 2002 mouse draft assembly
against regions of the human genome contained in an ancient copies of
transposable elements. In this case <EM>ancient</EM> means that RepeatMasker's
annotation indicates that the copy was fixed as an interspersed repeat in
a common ancestor of human and mouse. These regions are of interest
because they, more likely then any other region, have not been under
functional constraint.
Each block in the alignment is displayed as a colored block on the track
with a line connecting all the blocks. The color of each alignment
indicates the percent identity of aligned residues over all blocks of the
alignment. 50% identity and below is lightly colored and the color gets
linearly darker as the percent identity approaches 100%.
In the alignments, lower case letters indicate that RepeatMasker annotated
them as an interspersed repeat. Because of the high substitution rate in
the mouse lineage, the element often only was recognized in the human
genome. The original alignments often are much longer, but only the region
witin the repeat is displayed.

<H2>Methods</H2>
<P>The sequences were aligned with blastz (discontiguous exact seeds,
   ungapped extension, local alignments via dynamic programming) and
   postprocessed for single coverage.
   
<br>
<H2>Credits</H2>
Alignments contributed by Scott Schwartz, Penn State Bioinformatics Group.
