<H2>Description</H2>
<P>
The <A HREF="http://www.affymetrix.com/estore/products_services/arrays/index.affx"
TARGET=_blank>Exon GeneChip</A> contains over one million probe 
sets
designed to interrogate individual exons rather than the 3' ends of transcripts
as in traditional GeneChips. Exons were derived from a variety of
annotations that have been divided into the classes Core, Extended
and Full. 
<UL> 
  <LI><B>Core:
    </B>RefSeq transcripts, full-length GenBank mRNAs</LI>
  <LI><B>Extended:</B> 
    dbEst alignments, Ensembl annotations, syntenic mRNA from human, rat and mouse, 
    microRNA annotations, MITOMAP annotations, Vega genes, Vega pseudogenes
  <LI><B>Full:</B>
    Geneid genes, Genscan genes, Genscan Subopt, Exoniphy, RNA genes, SGP genes,
    Twinscan genes
  </LI>
</UL></P> 

<P>
Probe sets are colored by class with the Core probe sets being
the darkest and the Full being the lightest color. Additionally, probe
sets that do not overlap the exons of a transcript cluster, but fall
inside of its introns, are considered bounded by that transcript
cluster and are colored slightly lighter. Probe sets that overlap the
coding portion of the Core class are colored slightly darker.</P>
<P>
The microarray track using this probe set can be displayed by turning
on the <B>Affy Exon Tissue</B> track.</P>

<H2>Credits and References</H2>
<P>
The exons interrogated by the probe sets displayed in this track are
from the Affymetrix Exon 1.0 GeneChip and were derived from a
number of sources. In addition to the millions of cDNA sequences
contributed to the 
<A HREF="http://www.ncbi.nlm.nih.gov/genbank/" 
TARGET=_blank>GenBank</A>, 
<A HREF="http://www.ncbi.nlm.nih.gov/dbEST/" TARGET=_blank>dbEst</A> and 
<A HREF="http://www.ncbi.nlm.nih.gov/refseq/" TARGET=_blank>RefSeq</A> 
databases by
individual labs and scientists, the following annotations were used:
<P>
<A HREF="http://www.ensembl.org/index.html" 
TARGET=_blank><B>Ensembl</B></A><B>: </B>
Hubbard T, Barker D, Birney E, Cameron G, Chen Y, Clark L, Cox T, Cuff J, Curwen V, Down T <em>et
al</em>.
<a href="http://nar.oxfordjournals.org/content/30/1/38.abstract" target="_blank">
The Ensembl genome database project</a>.
<em>Nucleic Acids Res</em>. 2002 Jan 1;30(1):38-41.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/11752248" target="_blank">11752248</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC99161/" target="_blank">PMC99161</a>
</p>

<P>
<a href="http://compgen.bscb.cornell.edu/phast/"><B>Exoniphy</a>:</B> Siepel, A., Haussler, D. 
<A HREF="http://compgen.bscb.cornell.edu/~acs/recomb2004.pdf"
TARGET=_blank>Computational identification of evolutionarily conserved 
exons</A>.
<em>Proc. 8th Int'l Conf. on Research in Computational Molecular Biology</em>, 
177-186 (2004).
</P>

<P>
<B><A HREF="http://genome.crg.es/geneid.html" TARGET=_blank>Geneid Genes</A>:</B>
Parra G, Blanco E, Guig&#243; R.
<a href="http://genome.cshlp.org/content/10/4/511.abstract" target="_blank">
GeneID in Drosophila</a>.
<em>Genome Res</em>. 2000 Apr;10(4):511-5.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/10779490" target="_blank">10779490</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC310871/" target="_blank">PMC310871</a>
</p>

<P>
<B><A HREF="http://genes.mit.edu/GENSCAN.html" TARGET=_blank>Genscan Genes</A>:</B> 
Burge C, Karlin S.
<a href="http://www.sciencedirect.com/science/article/pii/S0022283697909517" target="_blank">
Prediction of complete gene structures in human genomic DNA</a>.
<em>J Mol Biol</em>. 1997 Apr 25;268(1):78-94.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/9149143" target="_blank">9149143</a>
</p>

<P>
<B><A HREF="http://www.mirbase.org/" TARGET=_blank>microRNA</A>:</B>
Griffiths-Jones S.
<a href="http://nar.oxfordjournals.org/content/32/suppl_1/D109.full" target="_blank">
The microRNA Registry</a>.
<em>Nucleic Acids Res</em>. 2004 Jan 1;32(Database issue):D109-11.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/14681370" target="_blank">14681370</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC308757/" target="_blank">PMC308757</a>
</p>

<P>
<B><A HREF="http://www.mitomap.org/MITOMAP" TARGET=_blank>MITOMAP</A>:</B>
Kogelnik AM, Lott MT, Brown MD, Navathe SB, Wallace DC.
<a href="http://nar.oxfordjournals.org/content/33/suppl_1/D611.full" target="_blank">
MITOMAP: a human mitochondrial genome database</a>.
<em>Nucleic Acids Res</em>. 1996 Jan 1;24(1):177-9.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/8594574" target="_blank">8594574</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC145607/" target="_blank">PMC145607</a>
</p>

<P>
<B><A HREF="http://selab.janelia.org/tRNAscan-SE/" TARGET=_blank>RNA Genes</A>:</B>
Lowe TM, Eddy SR.
<a href="http://nar.oxfordjournals.org/content/25/5/0955.full" target="_blank">
tRNAscan-SE: a program for improved detection of transfer RNA genes in genomic sequence</a>.
<em>Nucleic Acids Res</em>. 1997 Mar 1;25(5):955-64.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/9023104" target="_blank">9023104</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC146525/" target="_blank">PMC146525</a>
</p>

<P>
<B><A HREF="http://grib.imim.es" TARGET=_blank>SGP Genes</A>: </B> 
Wiehe T, Gebauer-Jung S, Mitchell-Olds T, Guig&#243; R.
<a href="http://genome.cshlp.org/content/11/9/1574.full" target="_blank">
SGP-1: prediction and validation of homologous genes based on sequence alignments</a>.
<em>Genome Res</em>. 2001 Sep;11(9):1574-83.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/11544202" target="_blank">11544202</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC311140/" target="_blank">PMC311140</a>
</p>

<P>
<B><A HREF="http://mblab.wustl.edu/" TARGET=_blank>Twinscan Genes</A>:</B>
Korf I, Flicek P, Duan D, Brent MR.
<a href="http://bioinformatics.oxfordjournals.org/content/17/suppl_1/S140.abstract" target="_blank">
Integrating genomic homology into gene structure prediction</a>.
<em>Bioinformatics</em>. 2001;17 Suppl 1:S140-8.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/11473003" target="_blank">11473003</a>
</p>

<P>
<B><A HREF="http://vega.sanger.ac.uk/index.html" TARGET=_blank>Vega Genes 
and Pseudogenes</A>: </B>The HAVANA group, 
<A HREF="http://www.sanger.ac.uk/" TARGET=_blank>Wellcome Trust Sanger 
Institute</A>.</P>
