<H2>Description</H2>
<P>
This track contains tBLASTn alignments of the peptides from the predicted 
and known genes identified in the hg16 Known Genes track.
</P>

<H2>Methods</H2>
<P>
First, the predicted proteins from the human Known Genes track were aligned 
with the human genome using the Blat program to discover exon boundaries.  
Next, the amino acid sequences that make up each exon were aligned with the 
$organism sequence using the tBLASTn program.
Finally, the putative <em>$organism</em> exons were chained together using an 
organism-specific maximum gap size but no gap penalty. The single best exon 
chains extending over more than 60% of the query protein were included. Exon 
chains that extended over 60% of the query and matched at least 60% of the 
protein's amino acids were also included.</P>

<H2>Credits</H2> 
<P>
tBLASTn is part of the NCBI BLAST tool set. For more information on BLAST, see
Altschul SF, Gish W, Miller W, Myers EW, Lipman DJ.
<A HREF="http://www.sciencedirect.com/science/article/pii/S0022283605803602"
TARGET=_blank>Basic local alignment search tool</A>.
<em>J Mol Biol.</em> 1990 Oct 5;215(3):403-410.</P>
<P>
<A HREF="http://www.genome.org/cgi/content/abstract/12/4/656" 
TARGET=_blank>Blat</A> was written by Jim Kent. The remaining utilities 
used to produce this track were written by Jim Kent or Brian Raney.</P>
