<H2>Description</H2>
<P>
This track contains tBLASTn alignments of the peptides
from the predicted and known genes identified in the <em>D. melanogaster</em>
FlyBase as of 24 July 2004 to the <EM>$organism</EM> sequence.

<H2>Methods</H2>
<P>
First, predicted proteins from the <em>D. melanogaster</em> FlyBase track were
aligned with the <em>D. melanogaster</em> genome using the blat program to 
discover exon boundaries.  
Next, the amino acid sequences that make up each exon were aligned with the 
<em>$organism</em> sequence using the tBLASTn program.
Finally, the putative <em>$organism</em> exons were chained together using an 
organism-specific maximum gap size but no gap penalty. The single best exon 
chains extending over more than 60% of the query protein were included. Exon 
chains that extended over 60% of the query and matched at least 60% of the 
protein's amino acids were also included.

<H2>Credits</H2> 
<P>
tBLASTn is part of the NCBI Blast tool set. For more information on blast, see
Altschul, S.F., Gish, W., Miller, W., Meyers, E.W. & Lipman, D.J. 
<A HREF="http://www.sciencedirect.com/science?_ob=ArticleURL&_udi=B6WK7-45PV640-2J&_user=10&_handle=B-WA-A-A-A-MsSAYWA-UUW-AUEWDWWYUW-AUEUBUBZUW-CWCBCZDUC-A-U&_fmt=summary&_coverDate=10%2F05%2F1990&_rdoc=1&_orig=browse&_srch=%23toc%236899%231990%23997849996%23312255!&_cdi=6899&view=c&_acct=C000050221&_version=1&_urlVersion=0&_userid=10&md5=347dee8971445b77e0b0f342f73fafde" 
TARGET=_blank>Basic local alignment search tool</A>. 
<em>J. Mol. Biol.</em> <B>215</B>(3), 403-410 (1990).
<P>
<A HREF="http://www.genome.org/cgi/content/abstract/12/4/656" 
TARGET=_blank>Blat</A> was written by Jim Kent. The remaining utilities 
required to produce this track were written by Jim Kent or Brian Raney.
