<H2>Description</H2>
<P>
The Fugu v.3.0 whole genome shotgun assembly was provided by the
<A HREF="http://jgi.doe.gov/" TARGET=_blank>US DOE Joint 
Genome Institute</A> (JGI). The assembly was constructed with the JGI
assembler, JAZZ, from paired end sequencing reads produced at JGI, Myriad 
Genetics, and Celera Genomics, resulting in a sequence coverage of 5.7X. All 
reads are plasmid, cosmid, or BAC end-sequences, with the predominant coverage
derived from 2 Kb insert plasmids. This assembly contains 20,379
scaffolds totaling 319 million base pairs. The largest 679 scaffolds
total 160 million base pairs.</P>
<P>
The strand information (+/-) for this track is in two parts.  The
first + or - indicates the orientation of the query sequence whose
translated protein produced the match.  The second + or - indicates the
orientation of the matching translated genomic sequence.  Because the two
orientations of a DNA sequence give different predicted protein sequences,
there are four combinations.  ++ is not the same as --; nor is +- the same
as -+.</P>

<H2>Methods</H2>
<P>
The alignments were made with blat in translated protein mode requiring two
nearby 4-mer matches to trigger a detailed alignment.  The $organism
genome was masked with RepeatMasker and Tandem Repeat Finder before 
running blat. </P>

<H2>Credits</H2> 
<P>
The 3.0 draft from the
<A HREF="http://genome.jgi-psf.org/fugu6/fugu6.info.html" TARGET=_blank>
JGI <em>Fugu rubripes</em> website</A> was used in the
UCSC Genome Browser Fugu blat alignments. These data were freely provided 
by the JGI for use in this publication only.</P>

<H2>References</H2> 
<p>
Kent WJ.
<a href="http://genome.cshlp.org/content/12/4/656.abstract" target="_blank">
BLAT--the BLAST-like alignment tool</a>.
<em>Genome Res</em>. 2002 Apr;12(4):656-64.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/11932250" target="_blank">11932250</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC187518/" target="_blank">PMC187518</a>
</p>
