<H2>Description</H2>
<P>
This track shows alignments of $o_organism ($o_db, $o_date) to the
$organism genome using a gap scoring system that allows longer gaps 
than traditional affine gap scoring systems. It can also tolerate gaps in both
$o_organism and $organism simultaneously. These 
&quot;double-sided&quot; gaps can be caused by local inversions and 
overlapping deletions in both species. 
<P>
The chain track displays boxes joined together by either single or
double lines. The boxes represent aligning regions.
Single lines indicate gaps that are largely due to a deletion in the
$o_organism assembly or an insertion in the $organism 
assembly.  Double lines represent more complex gaps that involve substantial
sequence in both species. This may result from inversions, overlapping
deletions, an abundance of local mutation, or an unsequenced gap in one
species.  In cases where multiple chains align over a particular region of
the $organism genome, the chains with single-lined gaps are often 
due to processed pseudogenes, while chains with double-lined gaps are more 
often due to paralogs and unprocessed pseudogenes.</P> 
<P>
In the "pack" and "full" display
modes, the individual feature names indicate the chromosome, strand, and
location (in thousands) of the match for each matching alignment.</P>

<H2>Methods</H2>
<P>
Transposons that have been inserted since the $o_organism/$organism
split were removed from the assemblies. The abbreviated genomes were
aligned with blastz, and the transposons were added back in.
The resulting alignments were converted into axt format using the lavToAxt
program. The axt alignments were fed into axtChain, which organizes all
alignments between a single $o_organism chromosome and a single
$organism chromosome into a group and creates a kd-tree out
of the gapless subsections (blocks) of the alignments. A dynamic program
was then run over the kd-trees to find the maximally scoring chains of these
blocks.

$matrix

Chains scoring below a threshold were discarded; the remaining
chains are displayed in this track.</P>

<H2>Credits</H2>
<P>
Blastz was developed at <A HREF="http://bio.cse.psu.edu" 
TARGET=_blank>Pennsylvania State University</A> by 
Minmei Hou, Scott Schwartz, Zheng Zhang, and Webb Miller with advice from
Ross Hardison.</P>
<P>
Lineage-specific repeats were identified by Arian Smit and his 
<A HREF="http://www.repeatmasker.org" TARGET=_blank>RepeatMasker</A>
program.</P>
<P>
The axtChain program was developed at the University of California at 
Santa Cruz by Jim Kent with advice from Webb Miller and David Haussler.</P>
<P>
The browser display and database storage of the chains were generated
by Robert Baertsch and Jim Kent.</P>

<H2>References</H2>
<P>
Chiaromonte F, Yap VB, Miller W. 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" 
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. 
<em>Pac Symp Biocomput.</em> 2002;:115-26.</P>
<P>
Kent WJ, Baertsch R, Hinrichs A, Miller W, Haussler D.
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" 
TARGET=_blank>Evolution's cauldron: Duplication, deletion, and rearrangement
in the mouse and human genomes</A>.
<em>Proc Natl Acad Sci U S A</em>. 2003 Sep 30;100(20):11484-9.</P>
<P>
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison RC,
Haussler D, Miller W.
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103" 
TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>. 
<em>Genome Res.</em> 2003 Jan;13(1):103-7.</P>

