<H2>Description</H2>
<P>
The Eponine program provides a probabilistic method for detecting 
transcription start sites (TSS) in mammalian genomic sequence, with 
good specificity and excellent positional accuracy.</P>

<H2>Methods</H2>
<P>
Eponine models consist of a set of DNA weight matrices recognizing
specific sequence motifs. Each of these is associated with a position
distribution relative to the TSS.</P>

<P>
Eponine has been tested by comparing the output with annotated mRNAs
from human chromosome 22. From this work, we estimate that using the
default threshold (0.999) it detects &gt;50% of transcription start
sites with approximately 70% specificity. However, it does not always
predict the direction of transcription correctly&mdash;an effect that
seems to be common among computational TSS finders.</P>

<H2>Credits</H2>
<P>
Thanks to Thomas Down at the 
<A HREF="http://www.sanger.ac.uk/" TARGET=_BLANK>Sanger Institute</A> 
for providing the 
<A HREF="http://www.sanger.ac.uk/Software/analysis/eponine/" 
TARGET=_BLANK>Eponine</A> program (version 2, March 6, 2002) which was run 
at UCSC to produce this track.</P>

<H2>References</H2>
<P>
Down TA, Hubbard TJP. 
<A HREF="http://www.genome.org/cgi/content/abstract/12/3/458" TARGET=_BLANK>
Computational detection and location of transcription start sites 
in mammalian genomic DNA</A>. 
<em>Genome Res.</em> 2002 Mar;12(3):458-61.</P>
