<h2>Description</h2>

<p>
This track shows predictions from the
<a href="http://genes.mit.edu/GENSCAN.html" target="_blank">Genscan</a> program
written by <a href="http://genes.mit.edu/burgelab/" target="_blank">Chris Burge</a>.
The predictions are based on transcriptional, translational and donor/acceptor
splicing signals as well as the length and compositional distributions of exons,
introns and intergenic regions.
</p>

<h2>Display Conventions and Configuration</h2>

<p>
This track follows the display conventions for
<a href="/goldenPath/help/hgTracksHelp.html#GeneDisplay">gene prediction
tracks</a>.
</p>

<p>
The track description page offers the following filter and configuration
options:
<ul>
<li><b>Color track by codons:</b> Select the <em>genomic codons</em> option
to color and label each codon in a zoomed-in display to facilitate validation
and comparison of gene predictions. Go to the
<a HREF="/goldenPath/help/hgCodonColoring.html">
Coloring Gene Predictions and Annotations by Codon</a> page for more
information about this feature.</li>
</ul>
</p>

<h2>Methods</h2>

<p>
For a description of the Genscan program and the model that underlies it,
refer to Burge and Karlin (1997) in the References section below.
The splice site models used are described in more detail in Burge (1998)
below.
</p>

<h2>Credits</h2>

Thanks to Chris Burge for providing the Genscan program.

<h2>References</h2>

<p>
Burge C.
Modeling Dependencies in Pre-mRNA Splicing Signals.
In: Salzberg S, Searls D, Kasif S, editors.
<a href="http://www.cbcb.umd.edu/~salzberg/compbio-book.html" target="_blank">
Computational Methods in Molecular Biology</a>.
Amsterdam: Elsevier Science; 1998. p. 127-163.
</p>

<p>
Burge C, Karlin S.
<a href="http://www.sciencedirect.com/science/article/pii/S0022283697909517" target="_blank">
Prediction of complete gene structures in human genomic DNA</a>.
<em>J. Mol. Biol.</em> 1997 Apr 25;268(1):78-94.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/9149143" target="_blank">9149143</a>
</p>
