<h2>Description</h2>
<p>
This track represents the genome-wide predicted binding 
sites for TF (transcription factor) binding profiles in the 
<a href="https://jaspar.elixir.no/genome-tracks/" target="_blank">JASPAR</a> 
database CORE collection.
</p>

<h2>Display Conventions and Configuration</h2>
<p>
Shaded boxes represent predicted binding sites for each of the TF profiles
in the JASPAR CORE collection. The shading of the boxes indicates 
the p-value of the profile's match to that position (scaled between 
0-1000 scores, where 0 corresponds to a p-value of 1 and 1000 to a 
p-value &le; 10<sup>-10</sup>). Thus, the darker the shade, the 
lower (better) the p-value.</p>

<p>
The default view shows only predicted binding sites with scores of 400 or greater but
can be adjusted in the track settings. Multi-select filters allow viewing of
particular transcription factors. At window sizes of greater than
10,000 base pairs, this track turns to density graph mode. 
Zoom to a smaller region and click into an item to see more detail.</p>

<p>
<em>From <a href="../../FAQ/FAQformat.html#format1">BED format documentation</a>:
  </em>
<table style="box-sizing: border-box; border-collapse: collapse; border-spacing: 0px; border: 2px solid gray; margin-top: 10px; margin-left: 15px; font-size: 13px; color: rgb(0, 0, 0); font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: normal; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-style: initial; text-decoration-color: initial;">
  <tbody style="box-sizing: border-box;">
    <tr style="box-sizing: border-box;">
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221);">shade</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221); background-color: rgb(226, 226, 226);">&nbsp;</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221); background-color: rgb(198, 198, 198);">&nbsp;</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221); background-color: rgb(170, 170, 170);">&nbsp;</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221); background-color: rgb(141, 141, 141);">&nbsp;</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221); background-color: rgb(113, 113, 113);">&nbsp;</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221); background-color: rgb(85, 85, 85);">&nbsp;</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221); background-color: rgb(56, 56, 56);">&nbsp;</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221); background-color: rgb(28, 28, 28);">&nbsp;</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221); background-color: rgb(0, 0, 0);">&nbsp;</td>
    </tr>
    <tr style="box-sizing: border-box;">
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221);">score in range</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221);">&le; 166</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221);">167-277</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221);">278-388</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221);">389-499</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221);">500-611</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221);">612-722</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221);">723-833</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221);">834-944</td>
      <td style="box-sizing: border-box; padding: 1px 15px; text-align: left; border-bottom: 1px solid rgb(221, 221, 221);">&ge; 945</td>
    </tr>
  </tbody>
</table>

<p><em>Conversion table:</em></p>
<table border="2" style="padding: 10px; border: 1px solid black; border-collapse: collapse;">
  <tr>
    <td style="padding:10px"><strong>Item score</strong></td>
    <td style="padding:10px">0</td>
    <td style="padding:10px">100</td>
    <td style="padding:10px">131</td>
    <td style="padding:10px">200</td>
    <td style="padding:10px">300</td>
    <td style="padding:10px">400</td>
    <td style="padding:10px">500</td>
    <td style="padding:10px">600</td>
    <td style="padding:10px">700</td>
    <td style="padding:10px">800</td>
    <td style="padding:10px">900</td>
    <td style="padding:10px">1000</td>
  </tr> 
  <tr>
    <td style="padding:10px"><strong>p-value</strong></td>
    <td style="padding:10px">1</td>
    <td style="padding:10px">0.1</td>
    <td style="padding:10px">0.049</td>
    <td style="padding:10px">10<sup>-2</sup></td>
    <td style="padding:10px">10<sup>-3</sup></td>
    <td style="padding:10px">10<sup>-4</sup></td>
    <td style="padding:10px">10<sup>-5</sup></td>
    <td style="padding:10px">10<sup>-6</sup></td>
    <td style="padding:10px">10<sup>-7</sup></td>
    <td style="padding:10px">10<sup>-8</sup></td>
    <td style="padding:10px">10<sup>-9</sup></td>
    <td style="padding:10px">&le; 10<sup>-10</sup></td>
  </tr>
</table>

<h2>Methods</h2>
<p>
For each TF binding profile in the JASPAR database CORE collection, genomes were scanned for matches.
</p>
<p>
For the computation of relative scores and p-values, we used PWMScan (Ambrosini et al. 2018). 
We selected TFBS predictions with a PWM relative score ≥ 0.8 and a p-value &lt; 0.05. 
P-values were scaled between 0 (corresponding to a p-value of 1) and 1000 (p-value ≤ 10-10) 
for colouring of the genome tracks and to allow for comparison of prediction confidence between 
different profiles.
</p>
<p>
Please refer to the supplementary information of the <a href="https://doi.org/10.1093/nar/gkz1001">JASPAR 2020 manuscript</a> for more details.
</p>
<h3>Brief overview of each release</h3>
<p>
The JASPAR 2026 update expanded the JASPAR CORE collection by 12% (306
added or upgraded profiles), culminating to a set of 2633 non-redundant
TF binding profiles.  Genome sequences were scanned with JASPAR 2026
CORE TF binding profiles for each taxon independently using PWMScan.
TFBS predictions were selected with a PWM relative score ≥ 0.8 and a
p-value &lt; 0.05. P-values were scaled between 0 (corresponding to a
p-value of 1) and 1000 (p-value ≤ 10⁻¹⁰) for coloring of the genome
tracks and to allow for comparison of prediction confidence between
different profiles. More information on the methods can be found in the
<a href="https://doi.org/10.1093/nar/gkaf1209" target="_blank">JASPAR 2026 
publication</a> or on the
<a href="https://jaspar.elixir.no/" target="_blank">JASPAR website</a>.</p>

<p>
The JASPAR 2024 update expanded the JASPAR CORE collection by 20% (329 added and 72 upgraded
profiles). The new profiles were introduced after manual curation, in which 26 629 TF binding
motifs were curated and obtained as PFMs or discovered from ChIP-seq/-exo or DAP-seq data. 2500
profiles from JASPAR 2022 were revised to either promote them to the CORE collection, update the
associated metadata, or remove them because of validation inconsistencies or poor quality. The
JASPAR database stores and focuses mostly on PFMs as the model of choice for TF-DNA interactions.
More information on the methods can be found in the
<a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkad1059" target="_blank">
JASPAR 2024 publication</a> or on the
<a href="https://jaspar.genereg.net/" target="_blank">JASPAR website</a>.</p>

<p>
JASPAR 2022 contains updated transcription factor binding sites
with additional transcription factor profiles. More information on the methods can be found in the
<a href="https://www.ncbi.nlm.nih.gov/pubmed/34850907" target="_blank">
JASPAR 2022 publication</a>
JASPAR 2022 publication or on the
<a href="https://jaspar.genereg.net/" target="_blank">JASPAR website</a>.</p>

<p>
JASPAR 2020 scanned DNA sequences with JASPAR CORE TF-binding profiles 
for each taxa independently using PWMScan. TFBS predictions were selected with 
a PWM relative score &ge; 0.8 and a p-value &lt; 0.05. P-values were scaled 
between 0 (corresponding to a p-value of 1) and 1000 (p-value &le; 10<sup>-10</sup>) for 
coloring of the genome tracks and to allow for comparison of prediction 
confidence between different profiles.</p>

<p> 
JASPAR 2018 used the TFBS Perl module (Lenhard and Wasserman 2002) 
and FIMO (Grant, Bailey, and Noble 2011), as distributed within the MEME suite 
(version 4.11.2) (Bailey <em>et al.</em> 2009). For scanning genomes with the 
BioPerl TFBS module, profiles were converted to PWMs and matches were kept with a 
relative score &ge; 0.8. For the FIMO scan, profiles were reformatted to MEME motifs 
and matches with a p-value &lt; 0.05 were kept. TFBS predictions that were not 
consistent between the two methods (TFBS Perl module and FIMO) were removed. The 
remaining TFBS predictions were colored according 
to their FIMO p-value to allow for comparison of prediction confidence between 
different profiles.</p>

<h2>Data Access</h2>
<p>
JASPAR Transcription Factor Binding data includes billions of items. Limited regions can 
be explored interactively with the 
<a href="../cgi-bin/hgTables">Table Browser</a> and cross-referenced with 
<a href="../cgi-bin/hgIntegrator">Data Integrator</a>, although positional
queries that are too big can lead to timing out. This results in a black page
or truncated output. In this case, you may try reducing the chromosomal query to
a smaller window.</p>
<p>
For programmatic access, 
the track can be accessed using the Genome Browser's 
<a href="../../goldenPath/help/api.html">REST API</a>. 
JASPAR annotations can be downloaded from the
<a href="http://hgdownload.soe.ucsc.edu/gbdb/$db/jaspar">Genome Browser's download server</a>
as a bigBed file. This compressed binary format can be remotely queried through
command line utilities. Please note that some of the download files can be quite large.</p>
<p>
The utilities for working with bigBed-formatted binary files can be downloaded
<a href="http://hgdownload.soe.ucsc.edu/downloads.html#utilities_downloads"
   target=_blank>here</a>.
Run a utility with no arguments to see a brief description of the utility and its options.
<ul>
  <li><b>bigBedInfo</b> provides summary statistics about a bigBed file including the number of
    items in the file.  With the <b>-as</b> option, the output includes an
    autoSql
    definition of data columns, useful for interpreting the column values.</li>
  <li><b>bigBedToBed</b> converts the binary bigBed data to tab-separated text.
    Output can be restricted to a particular region by using the -chrom, -start
    and -end options.</li>
</ul>
</p>

<h4>Example: retrieve all JASPAR items in chr1:200001-200400</h4>

<pre><tt>bigBedToBed http://hgdownload.soe.ucsc.edu/gbdb/$db/jaspar/JASPAR2024.bb -chrom=chr1 -start=200000 -end=200400 stdout</tt></pre>

<p> 
All data are freely available.
Additional resources are available directly from the JASPAR group:</p>
<ul>
  <li>Binding site predictions for all and individual TF profiles are available for download 
    at <a href="https://mencius.uio.no/JASPAR/JASPAR_genome_browser_tracks/" target="_blank">https://mencius.uio.no/JASPAR/JASPAR_genome_browser_tracks/</a>.</li>
  <li>Code used to create the UCSC tracks are available at <a href="https://github.com/ievarau/JASPAR-UCSC-tracks/tree/master">https://github.com/ievarau/JASPAR-UCSC-tracks/tree/master</a>.</li>
  <li>The underlying JASPAR motif data is available through the JASPAR website at <a href="https://jaspar.elixir.no/" target="_blank">https://jaspar.elixir.no</a>.</li>
</ul>


<h2>Other Genomes</h2>
<p>The JASPAR group provides TFBS predictions for many additional species and 
genomes. The 2026 release is available as a native track on the following genomes, and additionally 
on mm10 and araTha1 by connection to their 
<a href="../cgi-bin/hgHubConnect#publicHubs">
Public Hub</a> or by clicking the assembly links below:</p>
<table width="458" border="1">
  <tbody>
    <tr>
      <td><strong>Species</strong></td>
      <td><strong>Genome assembly versions</strong></td>
    </tr>
    <tr>
      <td width="300">Human - <em>Homo sapiens</em></td>
      <td><a target="_blank"
href="../cgi-bin/hgTrackUi?db=hg38&g=jaspar">hg38</a></td>
    </tr>
    <tr>
      <td>Mouse - <em>Mus musculus</em></td>
      <td><a target="_blank"
href="../cgi-bin/hgTrackUi?db=mm39&g=jaspar">mm39</a></td>
    </tr>
    <tr>
      <td>Zebrafish - <em>Danio rerio</em></td>
      <td><a target="_blank" 
href="../cgi-bin/hgTrackUi?db=danRer11&g=jaspar">danRer11</a></td>
    </tr>
    <tr>
      <td>Fruitfly - <em>Drosophila melanogaster</em></td>
      <td><a target="_blank" 
href="../cgi-bin/hgTrackUi?db=dm6&g=jaspar">dm6</a></td>
    </tr>
    <tr>
      <td>Nematode - <em>Caenorhabditis elegans</em></td>
      <td><a target="_blank"
href="../cgi-bin/hgTrackUi?db=ce11&g=jaspar">ce11</a></td>
    </tr>
    <tr>
      <td>Vase tunicate - <em>Ciona intestinalis</em></td>
      <td><a target="_blank"
href="../cgi-bin/hgTrackUi?db=ci3&g=jaspar">ci3</a></td>
    </tr>
    <tr>
      <td>Thale cress - <em>Arabidopsis thaliana</em></td>
      <td><a target="_blank" 
href="../cgi-bin/hgTracks?genome=araTha1&hubUrl=https://mencius.uio.no/JASPAR/JASPAR_genome_browser_tracks/current/hub.txt">araTha1</a></td>
    </tr>
    <tr>
      <td>Yeast - <em>Saccharomyces cerevisiae</em></td>
      <td><a target="_blank" 
href="../cgi-bin/hgTrackUi?db=sacCer3&g=jaspar">sacCer3</a></td>
    </tr>
    <tr>
      <td>Chicken - <em>Gallus gallus</em></td>
      <td><a target="_blank"
href="../cgi-bin/hgTrackUi?db=galGal6&g=jaspar">galGal6</a></td>
    </tr>
  </tbody>
</table>

<h2>Credits</h2>
<p>
The JASPAR database is a joint effort between several labs (please see the latest JASPAR 
paper, below). Binding site predictions and UCSC tracks were computed by the CBGR team 
at NCMBM using code developed at the Wasserman Lab. For enquiries about the data, 
please contact Anthony Mathelier (
<A HREF="mailto:&#97;&#110;&#116;&#104;&#111;&#110;y.
&#109;&#97;&#116;&#104;e&#108;&#105;&#101;&#114;&#64;&#110;&#99;&#109;b&#109;.
&#117;i&#111;.
&#110;o">
&#97;&#110;&#116;&#104;&#111;&#110;y.
&#109;&#97;&#116;&#104;e&#108;&#105;&#101;&#114;&#64;&#110;&#99;&#109;b&#109;.
&#117;i&#111;.
&#110;o</A>
<!-- above address is anthony.mathelier at ncmbm.uio.no -->
) or Ieva Rauluseviciute (
<A HREF="mailto:&#105;&#101;&#118;&#97;.
&#114;&#97;&#117;&#108;us&#101;&#118;&#105;&#99;&#105;u&#116;&#101;&#64;&#110;&#99;m&#98;m.
&#117;&#105;&#111;.
&#110;o">
&#105;&#101;&#118;&#97;.
&#114;&#97;&#117;&#108;us&#101;&#118;&#105;&#99;&#105;u&#116;&#101;&#64;&#110;&#99;m&#98;m.
&#117;&#105;&#111;.
&#110;o</A>
<!-- above address is ieva.rauluseviciute at ncmbm.uio.no -->
).
</p>

<blockquote>
<p><em><a href="https://mathelierlab.com/">CBGR</a></em><br>
Computational Biology &amp; Gene Regulation<br>
Norwegian Centre for Molecular Biosciences and Medicine (NCMBM)<br>
University of Oslo<br>
Oslo, Norway
</p>
</blockquote>
<blockquote>
<p><em><a href="https://cisreg.info/">Wasserman Lab</a></em><br>
Centre for Molecular Medicine and Therapeutics<br>
BC Children's Hospital Research Institute<br>
Department of Medical Genetics<br>   
University of British Columbia<br>
Vancouver, Canada
</p>
</blockquote>


<h2>References</h2>
<p>
Ovek Baydar D, Rauluseviciute I, Aronsen DR, Blanc-Mathieu R, Bonthuis I, de Beukelaer H, Ferenc K,
Jegou A, Kumar V, Lemma RB <em>et al</em>.
<a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkaf1209" target="_blank">
JASPAR 2026: expansion of transcription factor binding profiles and integration of deep learning models</a>.
<em>Nucleic Acids Res</em>. 2026;
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/41325984" target="_blank">41325984</a>; PMC: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC12807658/" target="_blank">PMC12807658</a>
</p>

<p>
Sandelin A, Alkema W, Engstrom P, Wasserman WW, Lenhard B.
<a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkh012" target="_blank">
JASPAR: an open-access database for eukaryotic transcription factor binding profiles</a>.
<em>Nucleic Acids Res</em>. 2004;.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/14681366" target="_blank">14681366</a>
</p>

