<H2>Description</H2>
<P>
The UCSC Known Genes track shows known protein-coding genes based on 
protein data from SWISS-PROT, TrEMBL, and TrEMBL-NEW and their
corresponding mRNAs from 
<A HREF="http://www.ncbi.nlm.nih.gov/Genbank/index.html" 
TARGET=_blank>GenBank</A>.</P>

<H2>Display Conventions and Configuration</H2>
<P>
This track follows the display conventions for
<A HREF="../goldenPath/help/hgTracksHelp.html#GeneDisplay">gene prediction
tracks</A>. Black coloring indicates features that have corresponding entries
in the Protein Databank (PDB). Blue indicates features associated with
mRNAs from NCBI RefSeq or (dark blue) items having associated proteins in
the SWISS-PROT database. The variation in blue shading of RefSeq items
corresponds to the level of review the RefSeq record has undergone:
predicted (light), provisional (medium), or reviewed (dark). </P>
<P>
This track contains an optional codon coloring
feature that allows users to quickly validate and compare gene predictions.
To display codon colors, select the <em>genomic codons</em> option from the
<em>Color track by codons</em> pull-down menu. Click
<A HREF="../goldenPath/help/hgCodonColoring.html">here</A> for more
information about this feature. </P>

<H2>Methods</H2>
<P>
mRNA sequences were aligned against the $organism genome using blat. When a 
single mRNA aligned in multiple places, only alignments having at least 98% 
base identity with the genomic sequence were kept. This set of mRNA 
alignments was further reduced by keeping only those mRNAs referenced by a 
protein in SWISS-PROT, TrEMBL, or TrEMBL-NEW.</P>
<P>
Among multiple mRNAs referenced by a single protein, the best mRNA was 
selected, based on a quality score derived from its length, the level of the
match between its translation and the protein sequence, and its release date.
The resulting mRNA and protein pairs were further filtered by removing 
short invalid entries and consolidating entries with identical CDS regions.
</P>
<P>
Finally, RefSeq entries derived from DNA sequences instead of 
mRNA sequences were added to produce the final data set shown in this track. 
Disease annotations were obtained from SWISS-PROT.</P>

<H2>Credits</H2>
<P>
The Known Genes track was produced at UCSC based primarily on cross-references
between proteins from 
<A HREF="http://www.expasy.org/sprot/" TARGET=_blank>SWISS-PROT</A> 
(including TrEMBL and TrEMBL-NEW) and mRNAs from 
<A HREF="http://www.ncbi.nlm.nih.gov/Genbank/index.html"
TARGET=_blank>GenBank</A>
contributed by scientists worldwide.  
<A HREF="http://www.ncbi.nlm.nih.gov/RefSeq/" TARGET=_blank>NCBI RefSeq</A> 
data were also included in this track.</P>

<H2>Data Use Restrictions</H2>
<P>
The SWISS-PROT entries in this annotation track are copyrighted. They are 
produced through a collaboration 
between the Swiss Institute of Bioinformatics and the EMBL Outstation - the 
European Bioinformatics Institute. There are no restrictions on their use by  
non-profit institutions as long as their content is in no way modified and 
this statement is not removed. Usage by and for commercial entities requires 
a license agreement (see <A HREF="http://www.isb-sib.ch/announce/" 
TARGET=_blank>http://www.isb-sib.ch/announce/</A> or send an email to 
<A HREF="mailto:license@isb-sib.ch">license@isb-sib.ch</A>).</P>

<H2>References</H2>
<P>
Hsu, F., Kent, W.J., Clawson, H., Kuhn, R.M., Diekhans, M., and Haussler, D. 
<A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btl048?ijkey=qbTw0578H4cXuFw&keytype=ref" 
TARGET=_blank>The UCSC Known Genes</A>. 
<em>Bioinformatics</em> <B>22</B>(9), 1036-46 (2006).
<P>
Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Wheeler DL. (2004)
<A HREF="http://nar.oupjournals.org/cgi/content/full/32/suppl_1/D23" TARGET=_blank>GenBank: update</A>. 
<em>Nucleic Acids Res.</em> 32 Database issue:D23-6.
<P>
Kent, W.J.
<A HREF="http://www.genome.org/cgi/content/abstract/12/4/656"
TARGET=_blank>BLAT - the BLAST-like alignment tool</A>.
<em>Genome Res.</em> <B>12</B>(4), 656-664 (2002).</P>

