<h2>Description</h2>

<p>
The mRNA track shows alignments between $organism mRNAs
in <a href="http://www.ncbi.nlm.nih.gov/genbank/" target="_blank">
GenBank</a> and the genome.</p>

<h2>Display Conventions and Configuration</h2>

<p>
This track follows the display conventions for
<a href="/goldenPath/help/hgTracksHelp.html#PSLDisplay" target="_blank">
PSL alignment tracks</a>. In dense display mode, the items that
are more darkly shaded indicate matches of better quality.
</p>

<p>
The description page for this track has a filter that can be used to change
the display mode, alter the color, and include/exclude a subset of items
within the track. This may be helpful when many items are shown in the track
display, especially when only some are relevant to the current task.
</p>

<p>
To use the filter:
<ol>
<li>Type a term in one or more of the text boxes to filter the mRNA
display. For example, to apply the filter to all mRNAs expressed in a specific
organ, type the name of the organ in the tissue box. To view the list of
valid terms for each text box, consult the table in the Table Browser that
corresponds to the factor on which you wish to filter. For example, the
&quot;tissue&quot; table contains all the types of tissues that can be
entered into the tissue text box. Multiple terms may be entered at once,
separated by a space. Wildcards may also be used in the filter.</li>
<li>If filtering on more than one value, choose the desired combination
logic. If &quot;and&quot; is selected, only mRNAs that match all filter
criteria will be highlighted. If &quot;or&quot; is selected, mRNAs that
match any one of the filter criteria will be highlighted.</li>
<li>Choose the color or display characteristic that should be used to
highlight or include/exclude the filtered items. If &quot;exclude&quot; is
chosen, the browser will not display mRNAs that match the filter criteria.
If &quot;include&quot; is selected, the browser will display only those
mRNAs that match the filter criteria.</li>
</ol>
</p>

<p>
This track may also be configured to display codon coloring, a feature that
allows the user to quickly compare mRNAs against the genomic sequence. For more
information about this option, go to the
<a href="/goldenPath/help/hgCodonColoringMrna.html" target="_blank">
Codon and Base Coloring for Alignment Tracks</a> page.
Several types of alignment gap may also be colored;
for more information, go to the
<a href="/goldenPath/help/hgIndelDisplay.html" target="_blank">
Alignment Insertion/Deletion Display Options</a> page.
</p>

<h2>Methods</h2>

<p>
GenBank $organism mRNAs were aligned against the genome using the
blat program. When a single mRNA aligned in multiple places,
the alignment having the highest base identity was found.
Only alignments having a base identity level within 0.5% of
the best and at least 96% base identity with the genomic sequence were kept.
</p>

<h2>Credits</h2>

<p>
The mRNA track was produced at UCSC from mRNA sequence data
submitted to the international public sequence databases by
scientists worldwide.
</p>

<h2>References</h2>
<P>
Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Wheeler DL.
<A HREF="http://nar.oxfordjournals.org/content/32/suppl_1/D23.abstract"
TARGET=_blank>GenBank: update</A>.
<em>Nucleic Acids Res</em>. 2004 Jan 1;32(Database issue):D23-6.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/14681350" target="_blank">14681350</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC308779/" target="_blank">PMC308779</a>
</p>

<P>
Kent WJ.
<A HREF="http://genome.cshlp.org/content/12/4/656.abstract" 
TARGET=_blank>BLAT - the BLAST-like alignment tool</A>.
<em>Genome Res</em>. 2002 Apr;12(4):656-64.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/11932250" target="_blank">11932250</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC187518/" target="_blank">PMC187518</a>
</p>
