<H2>Description</H2>
<P>
The Neandertal Seq track shows Neandertal sequence reads mapped to the $organism
genome. The Neandertal sequence was generated from six Neandertal fossils found
in Croatia, Germany, Spain and Russia.
</P>

<H2>Display Conventions and Configuration</H2>
<P>
The sequence reads (query sequences) from each of the six samples are contained
in separate subtracks. Use the checkboxes to select which samples 
will be displayed in the browser. Click and drag the sample name to
reorder the subtracks. The order in which the subtracks appear in the subtrack
list will be the order in which they display in the browser.
<P>
The query sequences in the SAM/BAM alignment representation
are normalized to the + strand of the reference genome
(see the <A HREF="https://samtools.github.io/hts-specs/SAMv1.pdf"
TARGET=_blank>SAM Format Specification</A>
for more information on the SAM/BAM file format). If a query sequence was
originally the reverse of what has been stored and aligned, it will have the
following
<A HREF="https://samtools.github.io/hts-specs/SAMv1.pdf#page=4"
TARGET=_blank>flag</A>:
<PRE>
(0x10) Read is on '-' strand.
</PRE></P>
<P>
<P>
BAM/SAM alignment representations also have <A HREF="https://samtools.github.io/hts-specs/SAMv1.pdf#page=6"
TARGET=_blank>tags</A>. Some tags are predefined and others (those beginning
with X, Y or Z) are defined by the aligner or data submitter. 
The following tag is associated with this track: 
<UL>
<LI><A HREF="https://samtools.github.io/hts-specs/SAMv1.pdf#page=6"
TARGET=_blank><B>AS</B></A>: Alignment score generated by aligner</LI>
</UL>
</P>
<P>
The item labels and display colors of features within this track can be
configured through the controls at the top of the track description page.
</P>
<UL>
<LI><B>Display Read Names</B>: By default, read names are not displayed. To 
display the read names, selected the check box next to "Display read names".
</LI>

<LI><B>Minimum alignment quality</B>: Excludes alignments with quality less than
the given number. The default is 0.</LI>

<LI><B>Color track by bases</B>: By default, mismatching bases are highlighted
in the display. Change the selection to &quot;item bases&quot; to see all base
values from the query sequence, or &quot;OFF&quot; to ignore query sequence.
Click <A HREF="/goldenPath/help/hgBaseLabel.html">here</A> for additional 
information.</LI>

<LI><B>Alignment Gap/Insertion Display Options</B>: Click 
<A HREF="/goldenPath/help/hgIndelDisplay.html">here</A> for help with 
these options.

<LI><B>Additional coloring modes</B>: Other aspects of the alignments can be
displayed in color or grayscale.</LI>
    <UL>
    <LI><B>Color by strand</B>: Alignments on the reverse strand are colored 
    dark red, alignments on the forward strand are colored dark blue.</LI>

    <LI><B>Grayscale</B>: Items are shaded according to the chosen method:
    alignment quality or base qualities. The alignment qualities of individual 
    items are shaded on a scale of 0 (lightest) to 99 (darkest).
    Base qualities are shaded on a scale of 0 (lightest) to 40 (darkest).
    Alignment quality is the default.</LI>
    </UL>
    </LI>
</UL>
</P>

<H2>Methods</H2>
<P>
The Neandertal sequence was genereated from six Neandertal fossils. Vi33.16
(54.1% genome coverage), Vi33.25 (46.6%) and Vi33.26 (45.2%) were discovered in
the Vindija cave in Croatia. Feld1 (0.1%) is from the Neandertal type specimen
from the Neander Valley in Germany, Sid1253 (0.1%) is from El Sidron cave in
Asturias, Spain, and Mez1 (2%) is from Mezmaiskaya in the Altai Mountains,
Russia.</P> 
<P>
To increase the fraction of endogenous Neandertal DNA in the sequencing
libraries, restriction enzymes were used to deplete libraries of microbial DNA.
This was done by identifying Neandertal sequencing reads whose best alignment
was to a primate sequence, and selecting enzymes that would differentially cut
non-primate fragments.  These enzymes all contained CpG dinucleotides in their
recognition sequences, reflecting the particularly low abundance of this
dinucleotide in mammalian DNA. Sequencing was carried out on the 454 FLX and
Titanium platforms and the Illumina GA. Neandertal reads were mapped to the
$organism genome ($db) using a custom mapper called
<A HREF="https://bioinf.eva.mpg.de/anfo"
title="https://bioinf.eva.mpg.de/anfo" TARGET=_blank>ANFO</A>. This custom
alignment program was developed to take into account the characteristics of 
ancient DNA. Following the observation and implementation by Briggs 
<EM>et al</EM>., ANFO
uses different substitution matrices for DNA thought to be double-stranded
versus single-stranded and changes between them if doing so affords a better
score.
</P>

<H2>Credits</H2>
<P>
This track was produced at UCSC using data generated by
<A HREF="mailto:&#101;&#100;&#64;&#115;&#111;e.&#117;&#99;s&#99;.&#101;&#100;&#117;"
title="&#101;&#100;&#64;&#115;&#111;e.&#117;&#99;s&#99;.&#101;&#100;&#117;" rel="nofollow"
TARGET=_BLANK>Ed Green</A>.
</P>

<H2>References</H2>
<p>
Briggs AW, Stenzel U, Johnson PL, Green RE, Kelso J, Pr&#252;fer K, Meyer M, Krause J, Ronan MT,
Lachmann M <em>et al</em>.
<a href="http://www.pnas.org/content/104/37/14616.abs" target="_blank">
Patterns of damage in genomic DNA sequences from a Neandertal</a>.
<em>Proc Natl Acad Sci U S A</em>. 2007 Sep 11;104(37):14616-21.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/17715061" target="_blank">17715061</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1976210/" target="_blank">PMC1976210</a>
</p>

<p>
Green RE, Krause J, Briggs AW, Maricic T, Stenzel U, Kircher M, Patterson N, Li H, Zhai W, Fritz MH
<em>et al</em>.
<a href="http://www.sciencemag.org/content/328/5979/710.abs" target="_blank">
A draft sequence of the Neandertal genome</a>.
<em>Science</em>. 2010 May 7;328(5979):710-22.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/20448178" target="_blank">20448178</a>
</p>
