<H2>Description</H2>
<P>
The Neandertal Sequence Contigs track shows consensus contigs 
called (after duplicate reads from each library were merged) from 
overlapping, non-redundant reads that passed mapping and base
quality criteria. 
</P>

<H2>Display Conventions and Configuration</H2>
<P>
The contigs (query sequences) from each of the six samples are contained in separate
subtracks. Use the checkboxes to select which samples will be
displayed in the browser. Click and drag the sample name to reorder the
subtracks. The order in which the subtracks appear in the subtrack list will
be the order in which they display in the browser.
<P>
The query sequences in the SAM/BAM alignment representation
are normalized to the + strand of the reference genome
(see the <A HREF="https://samtools.github.io/hts-specs/SAMv1.pdf"
TARGET=_blank>SAM Format Specification</A>
for more information on the SAM/BAM file format). If a query sequence was
originally the reverse of what has been stored and aligned, it will have the
following
<A HREF="https://samtools.github.io/hts-specs/SAMv1.pdf#page=4"
TARGET=_blank>flag</A>:
<PRE>
(0x10) Read is on '-' strand.
</PRE></P>
<P>
BAM/SAM alignment representations also have <A HREF="https://samtools.github.io/hts-specs/SAMv1.pdf#page=6"
TARGET=_blank>tags</A>. Some tags are predefined and others (those beginning
with X, Y or Z) are defined by the aligner or data submitter. 
The following tag is associated with this track:
<UL>
<LI><A HREF="https://samtools.github.io/hts-specs/SAMv1.pdf#page=6"
TARGET=_blank><B>AS</B></A>: Alignment score generated by aligner</LI>
</UL>
</P>
<P>
The item labels and display
colors of features within this track can be configured through the controls at
the top of the track description page.
</P>
<UL>
<LI><B>Display Read Names</B>: By default, read names are not displayed. To 
display the read names, select the check box next to "Display read names".</LI>

<LI><B>Minimum alignment quality</B>: Excludes alignments with quality less than
the given number. The default is 0.</LI>

<LI><B>Color track by bases</B>: By default, mismatching bases are highlighted
in the display. Change the selection to &quot;item bases&quot; to see all base
values from the query sequence, or &quot;OFF&quot; to ignore query sequence.
Click <A HREF="/goldenPath/help/hgBaseLabel.html">here</A> 
for additional information.</LI>

<LI><B>Alignment Gap/Insertion Display Options</B>: Click
<A HREF="/goldenPath/help/hgIndelDisplay.html">here</A> for help with 
these options.

<LI><B>Additional coloring modes</B>: Other aspects of the alignments can be
displayed in color or grayscale.</LI>
    <UL>
    <LI><B>Color by strand</B>: Alignments on the reverse strand are colored 
    dark red, alignments on the forward strand are colored dark blue.</LI>

    <LI><B>Grayscale</B>: Items are shaded according to the chosen method:
    alignment quality or base qualities.  The alignment qualities of individual 
    items are shaded on a scale of 0 (lightest) to 99 (darkest).
    Base qualities are shaded on a scale of 0 (lightest) to 40 (darkest).
    Alignment quality is the default.</LI>
   </UL>
   </LI>
</UL>
</P>

<H2>Methods</H2>
<P>
All Neandertal sequence reads from each of the six samples were aligned
to the $organism ($db) genome using the short read aligner/mapper 
<A HREF="https://bioinf.eva.mpg.de/anfo/" TARGET=_blank>ANFO</A>.
</P>
<P>
To reduce the effects of sequencing error, the alignments of Neandertal reads to
the human and chimpanzee reference genomes were used to construct human-based 
and chimpanzee-based consensus "minicontigs". To generate the consensus, 
uniquely placed, overlapping alignments were selected (ANFO MAPQ &ge; 90) and 
these were merged into a single multi-sequence alignment using the common 
reference genome sequence.
</P>
<P>
At each position in the resulting alignment, for each observed base, and for 
each possible original base: i) The likelihood of the observation was 
calculated, ii) the likely length of single-stranded overhangs was estimated, 
and iii) the potential for ancient DNA damage using the Briggs-Johnson model was
considered (Briggs <EM>et al</EM>. 2007). 
If most observations in a given position showed a gap, the consensus became a 
gap; otherwise the base with the highest quality score (calculated by dividing 
each likelihood by the total likelihood) was used as the consensus.
</P>
<P>
At the current coverage, heterozygous sites will appear as low quality bases 
with the second base (not shown) having a similar likelihood to the consensus 
base. 
Likewise, heterozygous indels are included only by chance or may show up as 
stretches of low quality bases. 
</P>

<H2>Credits</H2>
<P>
This track was produced at UCSC using data generated by
<A HREF="mailto:&#101;&#100;&#64;&#115;&#111;e.&#117;&#99;s&#99;.&#101;&#100;&#117;"
title="&#101;&#100;&#64;&#115;&#111;e.&#117;&#99;s&#99;.&#101;&#100;&#117;" rel="nofollow"
TARGET = _BLANK>Ed Green</a>.
</P>

<H2>Reference</H2>
<p>
Briggs AW, Stenzel U, Johnson PL, Green RE, Kelso J, Pr&#252;fer K, Meyer M, Krause J, Ronan MT,
Lachmann M <em>et al</em>.
<a href="http://www.pnas.org/content/104/37/14616.abs" target="_blank">
Patterns of damage in genomic DNA sequences from a Neandertal</a>.
<em>Proc Natl Acad Sci U S A</em>. 2007 Sep 11;104(37):14616-21.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/17715061" target="_blank">17715061</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1976210/" target="_blank">PMC1976210</a>
</p>

<p>
Green RE, Krause J, Briggs AW, Maricic T, Stenzel U, Kircher M, Patterson N, Li H, Zhai W, Fritz MH
<em>et al</em>.
<a href="http://www.sciencemag.org/content/328/5979/710.abs" target="_blank">
A draft sequence of the Neandertal genome</a>.
<em>Science</em>. 2010 May 7;328(5979):710-22.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/20448178" target="_blank">20448178</a>
</p>
