<H2>Description</H2>
<P>This track is based on text-mining of full-text biomedical articles and includes two types of subtracks:</P>

<ul>
<li>Sequences found in publications, grouped by article and searched in genomes with BLAT</li>
<li>Identifiers in publications that directly relate to chromosome locations (e.g., gene symbols, SNP identifiers, etc)</li>
</ul>
<P>Both sources of information are linked to the respective articles.
Background information on how permission to full-text data was obtained can be found on the <A
href="http://text.soe.ucsc.edu" target="_blank">project website</A>. 
<H2>Display Convention and Configuration</H2>
<P>The sequence subtrack indicates the location of sequences in publications
mapped back to the genome, annotated with the first author and the year of the
publication.  All matches of one article are grouped ("chained") together.
Article titles are shown when you move the mouse cursor over the features.
Thicker parts of the features (exons) represent matching sequences,
connected by thin lines to matches from the same article within 30 kbp.</P>

<P>The subtrack "individual sequence matches" activates automatically when
the user clicks a sequence match and follows the link "Show sequence matches individually" 
from the details page.  Mouse-overs show flanking text around the sequence, and clicking
features links to BLAT alignments.
</P>

<P>All other subtracks (i.e. bands, genes, SNPs) show the number of matching articles as
 the feature description.  Clicking on them shows the sentences and sections in articles 
where the identifiers were found.</P>

<P>The track configuration includes a keyword and year filter. Keywords are space-separated
and are searched in the article's title, author list, and abstract.</P>

<H2>Data</H2>
<P>The track is based on text from biomedical research articles, obtained as
part of the <A HREF="http://text.soe.ucsc.edu" target="_blank">UCSC Genocoding Project</A>.</P>

<P>The current dataset consists of about 600,000 files (main text and
supplementary files) from <A HREF="http://www.ncbi.nlm.nih.gov/pmc/"
TARGET=_blank>PubMed Central</A> (Open-Access set) and around 6 million text
files (main text) from <A HREF="http://www.elsevier.com"
TARGET=_blank>Elsevier</A> (as part of the <A
HREF="http://www.applications.sciverse.com/action/userhome"
TARGET=_blank>Sciverse Apps</A> program). </P>

<H2>Methods</H2>
<P>
All file types (including XML, raw ASCII, PDFs and various Microsoft
Office formats (Excel, Word, PowerPoint)) were converted to text. The results were processed 
to find groups of words that look like DNA/RNA sequences or
words that look like protein sequences. These were then mapped with BLAT to the
human genome and these model organisms: mouse (mm9), rat (rn4), zebrafish
(danRer6), <em>Drosophila melanogaster</em> (dm3), <em>X. tropicalis</em> (xenTro2), Medaka
(oryLat2), <em>C. intestinalis</em> (ci2), <em>C. elegans</em> (ce6) and yeast (sacCer2).

The pipeline roughly proceeds through these steps:
<UL>
<LI>For sequences, the best match across all genomes is used, if it is longer than 17 bp and matches at 90% identity. 
Two sets of BLAT parameters are tried, the default ones for sequences longer than 25 bp, very sensitive ones (stepSize=5) for shorter sequences.</LI>
<LI>Sequences are mapped to genomic DNA. Those that do not match are mapped to RefSeq cDNAs.</LI>
<LI>Hits from the same article that are closer than 30 kbp are
joined into one feature (shown as exon-blocks on the browser).</LI>
<LI>All parts of a joined feature have to match at least 25 bp.</LI>
<LI>Non-unique hits are kept in the joined feature with the most members.</LI>
<LI>Joined features with identical members in two different genomes are kept in both genomes.</LI>
</UL>

<P>Note that due to the 90% identity filter, some sequences do not match
anywhere in the genome. Examples include primers with added restriction sites,
mutation primers, or any other sequence that joins or mixes two pieces of genomic
DNA not part of RefSeq.  Also note that some gene symbols correspond to 
English words which can sometimes lead to many false positives.</P>

<H2>Credits</H2>
<P>Software and processing by Maximilian Haeussler. UCSC Track visualisation by
Larry Meyer and Hiram Clawson. Elsevier support by Max Berenstein, Raphael
Sidi, Judd Dunham, Scott Robbins and colleagues. Original version written at the <A
HREF="http://bergmanlab.smith.man.ac.uk/" target="_blank">Bergman Lab</A>, University of
Manchester, UK. Testing by Mary Mangan, OpenHelix Inc, and Greg Roe, UCSC.</P>

<H2>Feedback</H2>
Please send ideas, comments or feedback on this track to
<A HREF="mailto:&#109;a&#120;&#64;&#115;&#111;&#101;.uc&#115;&#99;.e&#100;&#117;">
&#109;a&#120;&#64;&#115;&#111;&#101;.uc&#115;&#99;.e&#100;&#117;</A>.
<!-- above address is max at soe.ucsc.edu -->
We are very interested in getting access to more articles from publishers for this
dataset; see the <A HREF="http://text.soe.ucsc.edu" target="_blank">project website</a>.
</P>

<H2>References</H2>

<p>
Aerts S, Haeussler M, van Vooren S, Griffith OL, Hulpiau P, Jones SJ, Montgomery SB, Bergman CM, Open Regulatory Annotation Consortium.
<a href="http://genomebiology.com/content/9/2/R31" target="_blank">
Text-mining assisted regulatory annotation</a>.
<em>Genome Biol</em>. 2008;9(2):R31.
</p>

<p>
Haeussler M, Gerner M, Bergman CM.
<a href="http://bioinformatics.oxfordjournals.org/content/27/7/980.long" target="_blank">
Annotating genes and genomes with DNA sequences extracted from biomedical articles</a>.
<em>Bioinformatics</em>. 2011 Apr 1;27(7):980-6.
</p>

<p>
Van Noorden R.
<a href="http://www.nature.com/news/trouble-at-the-text-mine-1.10184" target="_blank">
Trouble at the text mine</a>.
<em>Nature</em>. 2012 Mar 7;483(7388):134-5.
</p>
