<H2>Description</H2>
<P>
This track provides automatically-designed RT-qPCR primers for measuring the abundance of
human and mouse transcripts using SYBR-based qPCR (qPCR with double-stranded DNA-binding
reporter dye). The primers were generated by a
procedure that targets all transcripts and all &quot;possible&quot; exon-exon and intron-exon
junctions in the human and mouse transcriptomes.
</P>
<P>
Not all consecutive exon-intron-exon triplets generate &quot;possible&quot; primer pairs.
&quot;Possible&quot; primer pairs are defined as satisfying a set of imposed design rules:
<OL>
<LI> The first exon-exon junction is not addressed, to avoid problems related to abortive
transcription.</LI>
<LI>Intron length should be more than 800 bp to avoid problems of double products in
amplification.</LI>
<LI>Only junction primers are designed: forward and reverse primers must flank the junction.</LI>
<LI>Melting temperature of the primers should be between 60&deg;C and 63&deg;C (optimally
60.5&deg;C, according to Breslauer <em>et al.</em>, 1986).</LI>
<LI>Primer length should be 18-25 bp.</LI>
<LI>Product size should be 60-125 bp.</LI>
<LI>Primers are designed first for the intron-exon (pre-mRNA) junctions, and the two best primer
pairs for each junction are chosen.
<LI>Then, for the reverse primer of each pair, two options are
designed for the corresponding forward primer of the exon-exon (mRNA) junction.</LI>
<LI>The &quot;Primer Mispriming Library&quot; of the <A HREF="http://frodo.wi.mit.edu/"
TARGET=_blank>primer3</A> software is used: &quot;human&quot; for the human transcriptome and
&quot;rodent&quot; for the mouse transcriptome.</LI>
</OL>
</P>
<P>
The track provides easy access to primers for almost all transcripts in the transcriptome,
eliminating the need for a tedious, error-prone design process.
</P>

<H2>Methods</H2>
<p>
The <a href="../cgi-bin/hgGene?hgg_do_kgMethod=1&org=$organism">UCSC Genes model</a>
was used as a reference of the gene structure and the
<A HREF="http://frodo.wi.mit.edu/" TARGET=_blank>primer3</A> software as the design engine. The
software goes over all possible exon-exon junctions in the transcriptome and applies our design
rules/parameters to provide two primer pairs for every &quot;possible&quot; intron-exon junction
and four pairs for every &quot;possible&quot; exon-exon junction.
</P>
<H2>Display Conventions</H2>
<P>
The primers to amplify pre-mRNA (intron-exon junctions) are shown in <font color="red"><b>red</b>
</font> and the primers to amplify
mRNA (exon-exon junctions) in <font color="blue"><b>blue</b></font>. For each pre-mRNA primer pair,
there are two corresponding mRNA primers (that use the same reverse primer, if possible). Each pair
has a unique code which stands for the gene name and the junction name. For example, the human pair
&quot;JAG1_uc002wnw.2_11_1&quot; amplifies pre-mRNA, and the corresponding mRNA primers are
&quot;JAG1_uc002wnw.2_11_1_1&quot; and &quot;JAG1_uc002wnw.2_11_1_2.&quot;</p>
<p>
Using <b>JAG1_uc002wnw.2_11_1_2</b> to illustrate the naming scheme:
<ul>
<li><b>JAG1</b> is the gene symbol.
<li><b>uc002wnw.2</b> is the UCSC Genes identifier of the isoform.
<li><b>11</b> identifies the exon-intron-exon triplet.
<li><b>1</b> is the number (1 or 2) of the intron-exon junction (step 7 in the <em>Description</em>
section above). The names of primer pairs that cover intron-exon junctions
end here. For the exon-exon junctions that use the same reverse primer, there is one additional
number.
<li><b>2</b> is the number (1 or 2) of the exon-exon junction pair (step 8 in the
<em>Description</em> section above).
</ul>
<p>
Clicking on a primer pair
will take you to a new page with details for that pair. Additional properties for the primer pair,
including forward and reverse sequence, melting temperature, GC%, and product size, are available
by clicking on the number next to the instruction &quot;Click here for primer details.&quot;
There is also a
<A HREF="http://www.weizmann.ac.il/complex/compphys/software/Amit/primers/batch_query_qpcr_primers.htm"
TARGET=_blank>batch query website</A> available to download details for a large number of primers.
</p>

<H2>References</H2>

<P>
Breslauer KJ, Frank R, Bl&ouml;cker H, Marky LA. <A HREF="http://www.pnas.org/content/83/11/3746"
TARGET=_blank>Predicting DNA duplex stability from the base sequence</A>. <EM>Proc Natl Acad Sci
U S A</EM>. 1986 Jun;83(11):3746-50.
</P>

<P>
Rozen S, Skaletsky HJ. <A HREF="http://www.ncbi.nlm.nih.gov/pubmed/10547847" TARGET=_blank>Primer3
on the WWW for general users and for biologist programmers</A>. <EM>Methods Mol Biol</EM>.
2000;132:365-86.
</P>

<P>
Zeisel A, Yitzhaky A, Bossel Ben-Moshe N, Domany E.
<A HREF="http://bioinformatics.oxfordjournals.org/content/early/2013/03/28/bioinformatics.btt145"
TARGET=_blank>An accessible database for mouse and human whole transcriptome qPCR primers</A>.
<i>Bioinformatics</i>. 2013 Mar 28. [Epub ahead of print] PMID: 23539303.
</P>
