<H2>Description</H2>
<P>
This track shows the graphs constructed by analyzing experimental RNA
transcripts and serves as basis for the predicted alternative splicing
transcripts shown in the SIB Genes track.  The blocks represent exons; lines
indicate introns. The graphical display is drawn such that no exons
overlap, making alternative events easier to view when the track is in full
display mode and the resolution is set to approximately gene-level.  </P>
<P>Further information on the graphs can be found on the
<A HREF="http://ccg.vital-it.ch/tromer/" TARGET="_BLANK">Transcriptome 
Web interface</A>.</P>

<H2>Methods</H2>
<P>
The splicing graphs were generated using a multi-step pipeline:  
<OL>
<LI>RefSeq and GenBank RNAs and ESTs are aligned to the genome with
<A HREF="http://sibsim4.sourceforge.net" TARGET="_BLANK">SIBsim4</A>, keeping 
only the best alignments for each RNA.
<LI>Alignments are broken up at non-intronic gaps, with small isolated 
fragments thrown out.
<LI>A splicing graph is created for each set of overlapping alignments.  This
graph has an edge for each exon or intron, and a vertex for each splice site,
start, and end.  Each RNA that contributes to an edge is kept as evidence for
that edge.
<LI>Graphs consisting solely of unspliced ESTs are discarded.
</OL></P>

<H2>Credits</H2>
<P>
The SIB Alternative Splicing Graphs track was produced on the <A
HREF="http://www.vital-it.ch" TARGET="_BLANK">Vital-IT</A> high-performance 
computing platform
using a computational pipeline developed by Christian Iseli with help from
colleagues at the <A HREF="http://www.ludwigcancerresearch.org/" TARGET="_BLANK">Ludwig 
Institute for Cancer
Research</A> and the <A HREF="http://www.isb-sib.ch" TARGET="_BLANK">Swiss 
Institute of Bioinformatics</A>.  It is based on data from NCBI <A
HREF="http://www.ncbi.nlm.nih.gov/refseq/" TARGET="_blank">RefSeq</A> and <A
HREF="http://www.ncbi.nlm.nih.gov/genbank/"
TARGET="_blank">GenBank</A>/<A HREF="http://www.embl.de" 
TARGET="_BLANK">EMBL</A>.  Our
thanks to the people running these databases and to the scientists worldwide
who have made contributions to them.</P>
