<H2>Description</H2>
<P>
This track consolidates all the Simple Nucleotide Polymorphisms 
into a single track.  </P>

<H2>Filtering</H2>
<P>
The SNPs in this track include all known polymorphisms that
can be mapped against the current assembly.  These include known point
mutations (Single Nucleotide Polymorphisms), insertions, deletions,
and segmental mutations from the current build of 
<A HREF="http://www.ncbi.nlm.nih.gov/SNP/" TARGET=_blank>dbSnp</A>, 
which is shown in the Genome Browser 
<A HREF="../goldenPath/releaseLog.html">release log</A>.
</P>
<P>
There are three major cases that are not mapped and/or annotated:
<UL>
<LI> 
Submissions that are completely masked as repetitive elements. 
These are dropped from any further computations. This set of
reference SNPs is found in chromosome &quot;rs_chMasked&quot;
on the <A HREF="ftp://ftp.ncbi.nih.gov/snp/" TARGET=_blank>dbSNP ftp
site</A>.
<LI>
Submissions that are defined in a cDNA context with extensive
splicing. These SNPs are typically annotated on refSeq mRNAs through a
separate annotation process. Effort is being made to reverse map these
variations back to contig coordinates, but that has not been
implemented. For now, you can find this set of variations in
&quot;rs_chNotOn&quot; on the <A HREF="ftp://ftp.ncbi.nih.gov/snp/"
TARGET=_blank>dbSNP ftp site</A>.
<LI>
Submissions with excessive hits to the genome. Variations with 3+ hits
to the genome are not included in the tracks, but are available in
&quot;rs_chMulti&quot; on the <A HREF="ftp://ftp.ncbi.nih.gov/snp/"
TARGET=_blank>dbSNP ftp site</A>.
</UL>
</P>
<P>The heuristics for the non-SNP variations (i.e. named elements and
STRs) are quite conservative; therefore, some of these are probably lost. This
approach was chosen to avoid false annotation of variation in
inappropriate locations.</P>

<H2>Supporting Details</H2>
<P>
Positional information can be found in the annotations section
of the Genome Browser 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">downloads</A> page, 
which is organized by species and assembly. Non-positional information
displayed on this page can be found in the 
<A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hgFixed/database/">shared
data</A> section of the same page, where it is split into tables by
organism: 
<TT><B><FONT size=3>dbSnpRsHg</FONT></B></TT> for Human, 
<TT><B><FONT size=3>dbSnpRsMm</FONT></B></TT> for Mouse, and 
<TT><B><FONT size=3>dbSnpRsRn</FONT></B></TT> for Rat.

<H2>Credits</H2>
<P>
Thanks to NIH's <A HREF="http://www.ncbi.nlm.nih.gov/SNP/"
TARGET=_blank>dbSNP</A> for providing the public data, which 
are available from <A HREF="http://www.ncbi.nlm.nih.gov/SNP/"
TARGET=_blank>dbSnp</A> at the NCBI.</P>

