<h2>Description</h2>

<p>
This track shows protein sequence annotations from the <a
href="http://www.uniprot.org/" target="_blank">UniProt/SwissProt</A> database,
mapped to genomic coordinates. It also shows how the protein sequences in this database 
map to the genome.
The data has been curated from scientific publications by the UniProt/SwissProt staff.
The annotations are divided into multiple subtracks, based on their &quot;feature type&quot; in UniProt.
</p> 

<p>
For consistency, the subtrack &quot;UniProt/SwissProt Variants&quot; is a copy of the track
&quot;UniProt Variants&quot; in the track group &quot;Phenotype and Literature&quot;.
</p>

<h2>Display Conventions and Configuration</h2>

<p>
Genomic locations of UniProt/SwissProt annotations are labeled with a short name for
the type of annotation (e.g. &quot;glyco&quot;, &quot;disul&quot;, &quot;signal pep&quot; etc.).
A click on them shows the full annotation and provides a link to the UniProt/SwissProt
record for more details.  </p>

<p>
Mouse over a feature to see the full UniProt annotation comment.
</p>

<p>
Modified residues are highlighted in light blue, transmembrane regions in blue,
glycosylation sites in yellow, disulfide bonds in grey, topological domains in
red.
</p>

<p>
The subtracks for subcellular location are sorted from outside to inside of the cell: signal, extracellular, cytoplasmic, nucleus.
</p>

<p>Note that for the human hg38 assembly, there also is a 
<a href="hgHubConnect">public track hub</a> prepared by UniProt itself, with 
genome annotations produced and maintained by UniProt using their mapping
method.</p>

<h2>Methods</h2>

<p>
UniProt sequences were aligned to UCSC/Gencode transcript sequences first with
BLAT, then lifted to genome positions with pslMap.  UniProt annotations were
obtained from the UniProt XML file.  The annotations were then mapped to the genome
through the alignment using the pslMap program.  This mapping approach
draws heavily on the <A HREF="http://modbase.compbio.ucsf.edu/LS-SNP/"
TARGET="_BLANK">LS-SNP</A> pipeline by Mark Diekhans. The complete script is
part of the kent source tree and is located in src/hg/utils/otto/uniprot. The exact command
that were used to build this track can be found on <a
href="https://github.com/ucscGenomeBrowser/kent/tree/master/src/hg/makeDb/doc/hg38/uniprot.txt">github</a>.
</p>

<h2>Data Access</h2>

<p>
The raw data can be explored interactively with the
<a href="../cgi-bin/hgTables">Table Browser</a>, or the
<a href="../cgi-bin/hgIntegrator">Data Integrator</a>.
For automated analysis, the genome annotation is stored in a bigBed file that 
can be downloaded from the
<a href="http://hgdownload.soe.ucsc.edu/gbdb/$db/uniprot/" target="_blank">download server</a>.
The exact filenames can be found in the <a href="https://github.com/ucscGenomeBrowser/kent/blob/master/src/hg/makeDb/trackDb/trackDb.uniprot.ra">track configuration file</a>. 
Annotations can be converted to ASCII text by our tool <tt>bigBedToBed</tt>
which can be compiled from the source code or downloaded as a precompiled
binary for your system. Instructions for downloading source code and binaries can be found
<a href="http://hgdownload.soe.ucsc.edu/downloads.html#utilities_downloads">here</a>.
The tool can also be used to obtain only features within a given range, for example:
<br> 
<tt>bigBedToBed http://hgdownload.soe.ucsc.edu/gbdb/$db/uniprot/unipStruct.bb -chrom=chr6 -start=0 -end=1000000 stdout</tt> 
<br>
Please refer to our
<a href="https://groups.google.com/a/soe.ucsc.edu/forum/#!forum/genome">mailing list archives</a>
for questions, or our
<a href="../FAQ/FAQdownloads.html#download36">Data Access FAQ</a>
for more information. 
</p>

<h2>Credits</h2>

<p>
This track was created by Maximilian Haeussler, with advice from Mark Diekhans and Brian Raney, feedback from the UniProt staff and Alejo Mujica, Regeneron.
Thanks to UniProt for making all data available for download.
</p>

<h2>References</h2>

<p>
UniProt Consortium.
<a href="http://nar.oxfordjournals.org/content/40/D1/D71.long" target="_blank">
Reorganizing the protein space at the Universal Protein Resource (UniProt)</a>.
<em>Nucleic Acids Res</em>. 2012 Jan;40(Database issue):D71-5.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22102590" target="_blank">22102590</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3245120/" target="_blank">PMC3245120</a>
</p>

<p>
Yip YL, Scheib H, Diemand AV, Gattiker A, Famiglietti LM, Gasteiger E, Bairoch A.
<a href="http://onlinelibrary.wiley.com/doi/10.1002/humu.20021/abstract" target="_blank">
The Swiss-Prot variant page and the ModSNP database: a resource for sequence and structure
information on human protein variants</a>.
<em>Hum Mutat</em>. 2004 May;23(5):464-70.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/15108278" target="_blank">15108278</a>
</p>
