<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=223 HEIGHT=185>
  <TR><TD ALIGN=RIGHT><A HREF="http://www.sanger.ac.uk/Projects/M_musculus/" 
    TARGET=_blank>
    <IMG SRC="/images/Mus_musculus.jpg" WIDTH=208 HEIGHT=160 ALT="Mouse"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1> <em>Mus musculus</em><BR>
    </FONT>
    <FONT SIZE=-2> (Photo courtesy of
      <A HREF="http://www.jax.org/" TARGET=_blank>The Jackson Laboratory</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The March 2005 mouse (<em>Mus musculus</em>) draft genome data was 
obtained from the Build 34 assembly by NCBI. 

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a 
sequenced genomic clone, an mRNA or EST or STS marker, or 
a cytological band, a chromosomal coordinate range, or keywords 
from the GenBank description of an mRNA. The following list 
provides examples of various types of position queries for the 
mouse genome. See the 
<A HREF="../goldenPath/help/hgTracksHelp.html" 
TARGET=_blank>User's Guide</A> for more information. 
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top><B>Genome Browser Response:</B><BR></TD>
    </TR>
	
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chr16</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays all of chromosome 16</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>chr16:1-5000000</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays first 5 million bases of chr 16</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>AC072048.4</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays region of contig AC072048.4</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>D16Mit120</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays region around STS marker DMit16120 from the MGI consensus genetic map, including 100,000 bases on each side as well</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>D16Mit120;D16Mit149</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays region between STS markers D16Mit120 and D16Mit149</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>

    <TR><TD VALIGN=Top NOWRAP>BC012683</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays alignment location of mRNA with GenBank accession BC012683</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>AW045217</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays alignment location of EST with GenBank accession AW045217</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>Ncam2</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays region of genome with official MGI mouse genetic nomenclature Ncam2</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>

    <TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists transcribed pseudogenes but not cDNAs</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>kruppel zinc finger</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>huntington</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists candidate genes associated with Huntington's disease</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>Smith,D.</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by co-author D. Smith</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Smith D</em>, internally it uses 
			the format <em>Smith,D.</em>.</TD>
    </TR>

</TABLE>
<BR>
<HR>
<H3>Assembly Details</H3>
<P>
<!--
This assembly is a composite version in which phase 3 High Throughput Genome 
Sequence (HTGS) was merged with the Mouse Genome Sequencing Consortium v3 Whole 
Genome Shotgun Assembly (MGSCv3). The assembly was performed by NCBI using a 
&quot;combined&quot; tiling path that was created automatically for the most 
part, but was manually curated in places. This facilitated the placement of 
finished sequence in the context of the MGSCv3 assembly. Draft sequence was not 
included in this build: the slight increase in coverage gained by using this 
would have been offset by the increase in build errors. 
<P>
-->
Build 34 includes approximately 2.6 Gb of sequence, of which about
1.9 Gb is finished sequence, contained in 
chromosomes 1-19, X, Y, M (mitochondrial DNA) and Un (unmapped clone contigs).
Chromosomes 2, 4, 11 and X are finished in Build 34. 
More information about this version is available on the NCBI 
<A HREF="http://www.ncbi.nlm.nih.gov/mapview/stats/BuildStats.cgi?taxid=10090&build=34&ver=1"
TARGET=_blank>Build 34 Data</A> web page. </P>
<P>
NOTE: In the Genome Browser mm6 assembly, chrY_random erroneously contains a 
region duplicated from chrY. For more information about this issue, see the 
Genome Browser <A HREF="../FAQ/FAQdownloads#download30">FAQ</A>. Please also 
note that the UCSC mm6 database contains only the reference strain C57BL/6J.
</P>
<!--The whole genome N50 for Build 34 is 22.3 Mb, in 
comparison to 17.7 Mb for the previous build.
-->
<P>
The mm6 sequence and annotation data may be downloaded from the Genome 
Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/mm6/">FTP server</A> or
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#mouse">Downloads</A> 
web page.
The mm6 annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="../goldenPath/credits.html#mouse_credits">credits</A> page
for a detailed list of the organizations and individuals who contributed 
to this release. </P>
