<!-- Display image in righthand corner -->
<!--
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=185>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Caenorhabditis+japonica"
    TARGET=_blank>
    <IMG SRC="/images/Caenorhabditis_japonica.jpg" WIDTH=*** HEIGHT=*** ALT="C. japonica">
    </A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT>&nbsp;
    <FONT SIZE=-1><em>Caenorhabditis japonica</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of ***
      (<A HREF="*** link to source" 
      TARGET=_blank>***source</A>)
    </FONT>
  </TD></TR>
</TABLE>
-->

<P>
The Mar. 2008 <em>Caenorhabditis japonica</em> draft assembly was produced by 
the Genome Sequencing Center at the 
<A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Caenorhabditis+japonica" 
TARGET=_blank>Washington University in St. Louis (WUSTL) School of Medicine</A>.
The UCSC caeJap1 version corresponds to WUSTL version 3.0.2.</P>

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
<em>C. japonica</em> genome.  Note that the search results may include 
other organisms in addition to the current genome. See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>chrUn:1-1000000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of chrUn</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chrUn:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chrUn that spans 2000 bases, 
starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>BC058916</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of human mRNA with GenBank accession
		number BC058916</TD>
    </TR>
    <TR><TD VALIGN=Top>FF131130</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of <em>C. japonica</em> EST with GenBank accession
		FF131130</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>succinate-CoA</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists succinate-CoA RefSeq Genes and mRNAs from organisms other than <em>C. japonica</em></TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>rand</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Rand</TD>
    </TR>
    <TR><TD VALIGN=Top>Jones,D.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author D. Jones</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Jones D</em>, internally it
	uses the format <em>Jones,D.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
This genome has been sequenced to a depth of 6.3X.  
The assembly is comprised of 35,273 contigs containing 129,347,181 nucleotides.
These smaller contigs have been assembled into 4,657 supercontigs
named <em>Contig0</em> through <em>Contig10115</em> that contain 151,722,573 
nucleotides.  The UCSC Genome Browser combines the supercontigs together into an
artificial chromosome <em>chrUn</em> with 1,000 base gaps inserted between 
the supercontigs. This chrUn sequence contains 156,378,573 bases, with the 
artificial 1,000-base gaps covering 4,656,000 bases.
For more information on this assembly, see the WUSTL
<A HREF="http://genome.wustl.edu/pub/organism/Invertebrates/Caenorhabditis_japonica/assembly/Caenorhabditis_japonica-3.0.2/ASSEMBLY"
TARGET=_blank>assembly notes</A>.
</P>
<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/caeJap1/">FTP
server</A> or the 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#c_japonica">Downloads</A>
page. Refer to the WUSTL 
<A HREF="http://genome.wustl.edu/data.cgi" TARGET=_blank>data user policy</A>
for data use restrictions and citation information. </P>
<P>
The <em>C. japonica</em> browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#worm_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to this release.  </P>
