<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=245>
  <TR><TD ALIGN=RIGHT><A HREF="https://www.hgsc.bcm.edu/content/rat-genome-project" 
    TARGET=_blank>
    <IMG SRC="/images/Rattus_norvegicus.jpg" WIDTH=230 HEIGHT=153 ALT="Rat"></A> 
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1> <em>Rattus norvegicus</em><BR>
    </FONT>
    <FONT SIZE=-2> 
      (NHGRI <A HREF="http://www.genome.gov/dmd/index.cfm?node=Photos/Animals/Rat%20(brown%20norway)" 
      TARGET=_blank>Press Photos</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The November 2002 rat (<em>Rattus norvegicus</em>) genome assembly 
was produced by the 
Atlas group at <A HREF="http://hgsc.bcm.tmc.edu/" 
TARGET=_blank>Baylor Human Genome Sequencing Center</A> (HGSC) as 
part of the Rat Genome Sequencing Consortium. 

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a 
sequenced genomic clone, an mRNA or EST or STS marker, 
a chromosomal coordinate range, or keywords from 
the GenBank description of an mRNA. The following list shows 
examples of valid position queries for the rat genome. 
See the <A HREF="../goldenPath/help/hgTracksHelp.html"
TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><BR></TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr16</TD>
	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of chromosome 16</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr16:1-5000000</TD>
	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first 5 million bases of chr 16</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>AI501130</TD>
	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of EST with GenBank accession AI501130</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>AF199335</TD>
	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession AF199335</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>apoe</TD>
	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of genome with gene identifier apoE</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>NM_145881</TD>
	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of genome with RefSeq identifier NM_145881</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>25728</TD>
	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of genome with Entrez Gene identifier 25728</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>kruppel zinc finger</TD>
	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>huntington</TD>
	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists candidate genes associated with Huntington's disease</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Jones,R.</TD>
	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author R. Jones</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Jones R</em>, internally it uses 
			the format <em>Jones,R.</em>.</TD>
    </TR>
</TABLE>
<P>
<HR>
<H3>Assembly details</H3>
<P>
This sequence was assembled using a hybrid approach that combines the 
clone-by-clone and whole genome shotgun methods. A new software program - 
ATLAS - was developed for this effort. The assembly process resulted in a 
6.5-fold coverage of the rat genome, which is estimated to be approximately 
2.8 Gigabases in size. For more information on the rat genome, the assembly 
process, and the Rat Genome Sequencing Consortium, refer to the Human Genome 
Sequencing Center at Baylor College of Medicine 
<A HREF="https://www.hgsc.bcm.edu/content/rat-genome-project" TARGET=_blank>rat
website</A>.</P>
<P>
Sequence and annotation downloads are available from the Genome Browser
<A HREF="ftp://genome-archive.cse.ucsc.edu/goldenPath/rnNov2002/">FTP server</A> or 
<A HREF="http://genome-archive.cse.ucsc.edu/downloads.html#rat">Downloads</A> page. 
These data are made available with 
<A HREF="http://genome.ucsc.edu/goldenPath/credits.html#rat_use">specific conditions for use</A>.
</P>
<P>
Thanks to the Atlas group at Baylor HGSC and the Rat Genome Sequencing 
Consortium for providing the sequence data and assembly for this release.
The annotation tracks displayed in the Genome Browser were generated by UCSC and
collaborators worldwide. See the 
<A HREF="http://genome.ucsc.edu/goldenPath/credits.html#rat_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to this release. 
