<H2>Description</H2>
<P>
Bacterial artificial chromosomes (BACs) are a key part of many large-scale 
sequencing projects.  A BAC typically consists of 50 - 300 kb of DNA.  
During the early phase of a sequencing project, it is common to sequence a 
single read (approximately 500 bases) off each end of a large number of BACs. 
Later in the project, these BAC end reads are mapped to the genome sequence. 
A valid pair of BAC end sequences must be at least 2 kb but no more than 
800 kb away from each other. The orientation of the first BAC end sequence 
must be &quot+&quot;; that of the second BAC end sequence must be 
&quot;-&quot;.  
<P>
These BAC end pairs can be useful for validating the assembly over relatively 
long ranges.  In some cases, the BACs are useful biological reagents.  This 
annotation can also be used to determine which BAC contains a given gene, 
useful information for certain wet lab experiments.</P>
<P>
This track shows mappings in cases where only one BAC end was mapped, either
because only one end sequence exists or only one end sequence
could be mapped to the genome with an alignment that met the criteria 
described in the Methods section.
For the $organism assembly, the BACs are approximately 150-200 kb in size and 
are used for both the fingerprint contig (FPC) and radiation hybrid (RH) maps.
Some BAC end sequences have replicate reads; in these cases, a single read 
with the longest alignment was chosen to represent the end for a BAC clone in 
the browser and this same read is used to represent the BAC clone for every 
alignment of this BAC end.
</P>
<P>
The scoring scheme used for this annotation assigns 1000 to an alignment 
when a single BAC end aligns to only one location in the genome (after 
filtering).  When a single BAC end aligns to multiple locations, the score 
is calculated as 1500/(number of alignments).</P>

<H2>Display Conventions and Configuration</H2>
<P>
This track follows the display conventions for 
<A HREF="../goldenPath/help/hgTracksHelp.html#PSLDisplay" 
TARGET=_blank>PSL alignment tracks</A>. On the track description page, the 
display may be configured to show only those items with an unnormalized score
that equals or exceeds a user-specified minimum.</P>
<P>
To view the registry entry for a specific clone, open the details page for the
clone and click on its name at the top of the page. Not all $organism BAC 
clones have been submitted to NCBI Clone Registry as of July 2004; therefore, 
some of the clone links to NCBI may not yet be active. 
Information about the libraries may be found on the Sanger Institute 
<A HREF="http://www.sanger.ac.uk/Projects/D_rerio/library_details.shtml" 
TARGET=_blank>Zebrafish Library Details</A> page. Additional information about
some of the clones, including how they can be obtained, may be found at the
<A HREF="http://www.ncbi.nlm.nih.gov/genome/clone/index.html" 
TARGET=_blank>NCBI Clone Registry</A>.
<P>
Information about STS markers associated with a BAC clone is displayed when
available. Aliases include those for the BAC clone and those for associated
STS markers. The BAC ends tracks may be searched using the clone name, the 
Sanger internal BAC clone name, the Sanger STS name or any of these aliases. 
The UniSTS ID(s) shown are those associated with the STS aliases for each 
Sanger STS name. </P>
<P>
NOTE: The primer sequences shown may differ from those associated with the
uniSTS IDs in UniSTS. In cases where more than one UniSTS ID exists, primer 
sequences may be the same as those in UniSTS for one of the UniSTS IDs.
The value in the relationship field indicates the method used to find the 
STS markers associated with a particular BAC clone:
<UL>
<LI><B>n/a</B> - no information is available for this STS or there is no STS 
marker for this BAC clone
<LI><B>BAC end</B> - STS is designed from the end sequence of this BAC clone
<LI><B>e-PCR genomic</B> - STS is found by e-PCR in the genomic sequence of 
this BAC clone
<LI><B>e-PCR WGS</B> - STS is found by e-PCR in the whole genome shotgun 
(WGS) contig that has been aligned to this region (approximately +/- 150 kb). 
There is no direct relationship with the clone. These data are covered by the 
same caveats that apply to the WGS contigs.
</UL></P>

<H2>Methods</H2>
<P>BAC end sequences were placed on the assembled sequence using blat, 
followed by pslReps using the parameters <TT>-nearTop=0.02 -minCover=0.40 
-minAli=0.85 -noIntrons</TT>. This ensured that only alignments with 
least 85% identity, a minimum sequence coverage of 40% and a base identity 
level within 2% of the best were kept. No penalty was imposed for sequences 
that lacked introns. Furthermore, a base identity of at least 91% was required 
of at least one BAC end of the pair. </P>

<H2>Credits</H2>
<P>
The $Organism BAC End Pairs track was produced at UCSC from 
data obtained from the following sources:
<UL>
<LI>BAC end pairs sequence data &mdash; 
<A HREF="http://www.eb.tuebingen.mpg.de/dept3/geisler/home.html" 
TARGET=_blank>Robert Geisler</A> at the Max Planck Institute for 
Developmental Biology, Germany and the 
<A HREF="http://www.niob.knaw.nl" TARGET=_blank>Netherlands Institute for 
Developmental Biology</A> (Hubrecht Laboratory) 
<LI>GenBank accessions for BAC clones, STS marker data &mdash; 
<A HREF="http://www.sanger.ac.uk" TARGET=_blank>The Wellcome Trust Sanger 
Institute</A>, Cambridge, UK. 
<LI>UniSTS IDs &mdash; NCBI <A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=unists"
TARGET=_blank>UniSTS</A> database
</UL></P>
<P>
This track was produced in collaboration with the 
<A HREF="http://zon.tchlab.org"
TARGET=_blank>Zebrafish Genome Initiative</A> at Childrens Hospital, Boston.
</P>

<H2>References</H2>
<P>
Kent, W.J.
<A HREF="http://www.genome.org/cgi/content/abstract/12/4/656" 
TARGET=_blank>BLAT - the BLAST-like alignment tool</A>.
<em>Genome Res.</em> <B>12</B>(4), 656-664 (2002).</P>
