<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=210 HEIGHT=250>
  <TR><TD ALIGN=RIGHT><A HREF="http://www.hgsc.bcm.tmc.edu/projects/seaurchin/" 
    TARGET=_blank>
    <IMG SRC="/images/S_purpuratus.jpg" WIDTH=200 HEIGHT=188 ALT="Purple Sea Urchin"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1> <em>Strongylocentrotus purpuratus</em><BR>
    </FONT>
    <FONT SIZE=-2> (Photo courtesy of
      <A HREF="http://biology.fullerton.edu/people/faculty/steve-murray/" TARGET=_blank>Steven Murray, CSU Fullerton</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Sep. 2006 <em>Strongylocentrotus purpuratus</em> draft assembly 
(Baylor release Spur_2.1) was produced by the Baylor College of Medicine's 
Human Genome Sequencing Center (BCM HGSC). 

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA, a chromosome (or scaffold) coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
<em>S. purpuratus</em> genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top><B>Genome Browser Response:</B><BR></TD>
    </TR>
    <TR><TD COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top NOWRAP>Scaffold12150</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all 1,373,087 bases of Scaffold12150</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Scaffold12150:1-10,000</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first 10,000 bases of Scaffold12150</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>AF321312</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Searches for regions of genome aligning to mRNA with GenBank accession AF321312</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>tbp</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Searches for regions of genome aligning to tbp gene</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>cyclin dependent</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists cyclin-dependent protein mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>davidson</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists accessions deposited by authors named Davidson</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Wessel,G.M.</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author G.M. Wessel</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although 
	GenBank requires the search format <em>Wessel GM</em>, internally it 
	uses the format <em>Wessel,G.M.</em></TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
The Sep. 2006 release of the purple sea urchin genome (<em>Strongylocentrotus 
purpuratus</em>) is now available in the UCSC Genome Browser. This assembly, 
UCSC version strPur2, was produced by the Baylor College of Medicine Human 
Genome Sequencing Center (BCM HGSC) and corresponds to their Spur_2.1 assembly. 
</P>
<P>
The Spur_2.1 release was assembled from BAC sequence reads (approximately 2x 
coverage) and whole genome shotgun reads (6x coverage), and utilizes BAC tiling 
path information. The BCM HGSC Atlas-2.0 genome assembly system was used to 
generate this assembly. 
The total length of all contigs greater than 1kb is 804 Mbps.  When the 
gaps between contigs in scaffolds are included, the total span of the 
assembly is 907 Mbps.  The estimated size of the genome based on the 
assembly is 814 Mbps.</P>
<P>
Compared to previous sea urchin releases, the Spur_2.1 
assembly is more continuous and has fewer false duplications; contaminations 
identified in the previous Spur_2.0 assembly have been removed. This draft 
assembly may contain errors; therefore, users should exercise caution. Typical 
errors may include misassemblies of repeat sequences, collapses of repeat 
regions, and artificial duplications in polymorphic regions. However, base 
accuracy in contigs is usually very high with most errors near the ends of 
contigs. </P>
<P>
More assembly details can be found in the 
<A HREF="ftp://ftp.hgsc.bcm.tmc.edu/pub/data/Spurpuratus/fasta/Spur_v2.1/README_Spur_v2.1.txt"
TARGET=_blank>Spur_2.1 README file</A> and on the BCM HGSC 
<A HREF="http://www.hgsc.bcm.tmc.edu/projects/seaurchin/"
TARGET=_blank>Sea Urchin Genome Project</A> web page.</P>
<P>
Bulk downloads of the sequence and annotation data are available from the 
Genome Browser
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/strPur2">FTP server</A>
or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#urchin">Downloads</A> 
page. These data have 
<A HREF="http://www.hgsc.bcm.tmc.edu/projects/conditions_for_use.html" 
TARGET=_blank>specific conditions for use</A>. The initial
set of strPur2 annotation tracks was generated by UCSC. 
See the <A HREF="../goldenPath/credits.html#urchin_credits">Credits</A>
page for a detailed list of the organizations and 
individuals who contributed to this release. </P>
