<H2>Description</H2>
<P>
This track shows the gaps in the GRCh38 (hg38) genome assembly defined in the
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/hg38.agp.gz"
TARGET=_blank>AGP file</A> delivered with the sequence. These gaps are being closed during the 
finishing process on the human genome. For information on the AGP file format, see the NCBI 
<A HREF="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/agp/AGP_Specification.shtml"
TARGET=_blank>AGP Specification</A>.  The NCBI website also provides an 
<A HREF="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/assembly.shtml"
TARGET=_blank>overview of genome assembly procedures</A>, as well as  
<A HREF="http://www.ncbi.nlm.nih.gov/assembly/883148/"
TARGET="_blank">specific information</A> about the hg38 assembly.
</P>
<P>
Gaps are represented as black boxes in this track.
If the relative order and orientation of the contigs on either side
of the gap is supported by read pair data, 
it is a <em>bridged</em> gap and a white line is drawn 
through the black box representing the gap. 
</P>
<P>This assembly contains the following principal types of gaps:
<UL>
<LI><B>short_arm</B> - short arm gaps (count: 5; size range: 5,000,000 - 16,990,000 bases)</LI>
<LI><B>heterochromatin</B> - heterochromatin gaps (count: 11; size range: 20,000 - 30,000,000 bases)</LI>
<LI><B>telomere</B> - telomere gaps (count: 48; all of size 10,000 bases)</LI>
<LI><B>contig</B> - gaps between contigs in scaffolds (count: 285; size range: 100 - 400,000 bases)</LI>
<LI><B>scaffold</B> - gaps between scaffolds in chromosome assemblies (count: 470; size range: 10 - 624,000 bases)</LI>
</UL></P>
