<h2>Description</h2>

<p>
This track shows the genomic positions of somatic variants found through whole genome sequencing of tumors
as part of The Cancer Genome Atlas (TCGA) by the National Cancer Institute, made available through
the <a href="https://portal.gdc.cancer.gov" target="_blank">Genomic Data Commons Portal</a>. The
data shown here is sometimes called the &quot;Pan-Cancer dataset&quot;, a collection of thirty-three
TCGA projects processed in a uniform way.</p>

<h2>Display Conventions and Configuration</h2>
<p>
Variants can be filtered by project ID and gender from the track details page. Pressing the
&quot;All&quot; button allows the user to specify whether the checked values all have to be
true of a particular variant, or if only one of them need be present to satisfy the filter.</p>

<p>
The vertical viewing range in full mode can also be used to filter what variants are shown. Variants
that have a sampleCount more or less than the min and max values specificed in the viewing range are
not displayed.</p>

<H2>Data access</H2>
<p>
The raw data can be explored interactively with the <a href="../cgi-bin/hgTables"
target="_blank">Table Browser</a> or the <a href="../cgi-bin/hgIntegrator" target="_blank">Data
Integrator</a>.

<p>
For automated download and analysis, the genome annotation for all the thirty-three projects is
stored in a bigBed file that can be downloaded from
<a href="http://hgdownload.soe.ucsc.edu/gbdb/hg38/gdcCancer/gdcCancer.bb" target="_blank">our
download server</a>. There are also bigBed files for each of the thirty-three projects in that
directory. Individual regions or the whole genome annotation can be obtained using our tool
<tt>bigBedToBed</tt> which can be compiled from the source code or downloaded as a precompiled
binary for your system. Instructions for downloading source code and binaries can be found
<a href="http://hgdownload.soe.ucsc.edu/downloads.html#utilities_downloads"
target="_blank">here</a>. The tool can also be used to obtain only features within a given range,
e.g.,</p>
<pre>
bigBedToBed http://hgdownload.soe.ucsc.edu/gbdb/hg38/gdcCancer/gdcCancer.bb -chrom=chr21 -start=0 -end=100000000 stdout
</pre>


<h2>Methods</h2>

<p>
All MuTect Variant calls were downloaded from the GDC portal in January 2019 and reformatted at UCSC
to the <a href="../goldenPath/help/bigBed.html" target="_blank">bigBed</a> format with a short
script, <a href="https://github.com/ucscGenomeBrowser/kent/blob/master/src/utils/cancerMafToBigBed"
target="_blank">cancerMafToBigBed</a>.
</p>

<h2>Credits</h2>
<p>
Thanks to GDC for making the TCGA data available on their web site.
</p>
