<h2>Description</h2>
<h3>Agilent Arrays</h3>
<p>
The arrays listed in this track are probes from the
<a href="https://www.genomics.agilent.com/article.jsp?pageId=75"
target="_blank">Agilent Catalog Oligonucleotide Microarrays</a>.
</P>
<p>Please note that more microarray tracks are available on the hg19 genome assembly. 
To view those tracks, please 
<a href="/cgi-bin/hgTrackUi?db=hg19&g=genotypeArrays">click this link for hg19 microarrays</a>.
Microarrays that are not listed can be added as Custom Tracks with data from the companies.
</p>
<p><a target="_blank" 
href="https://www.agilent.com/en/product/cgh-cgh-snp-microarray-platform/cgh-cgh-snp-microarrays/human-microarrays">
Agilent GenetiSure Cyto</a>
</p>
<P>
Agilent's oligonucleotide CGH (Comparative Genomic Hybridization) platform enables the
study of genome-wide DNA copy number changes at a high resolution. The CGH probes on Agilent
CGH microarrays are 60-mer oligonucleotides synthesized in situ using Agilent's inkjet
SurePrint technology. The probes represented on the Agilent CGH microarrays have been
selected using algorithms developed specifically for the CGH application, assuring optimal
performance of these probes in detecting DNA copy number changes.
</P>

<h3>Illumina 450k and 850k Methylation Arrays</h3>
<p>
With the <a href="https://www.illumina.com/products/by-type/microarray-kits/infinium-methylation-epic.html"
target="_blank">Infinium MethylationEPIC BeadChip Kit</a>, researchers can interrogate over 850,000
methylation sites quantitatively across the genome at single-nucleotide resolution. Multiple
samples, including FFPE, can be analyzed in parallel to deliver high-throughput power while
minimizing the cost per sample. These tracks show positions being measured on the Illumina 450k and
850k (EPIC) microarray tracks. More information about the arrays can be found on the
<a href="https://www.illumina.com/products/by-type/microarray-kits/infinium-methylation-epic.html"
target="_blank">Infinium MethylationEPIC Kit</a> website.

<h3>Illumina CytoSNP 850K Probe Array</h3>
<p>
The <a href="https://www.illumina.com/products/by-type/clinical-research-products/infinium-cytosnp-850k.html"
target="_blank">Infinium CytoSNP-850K v1.2 BeadChip</a> provides comprehensive coverage of
cytogenetically relevant genes on a proven platform, helping researchers find valuable information
that may be missed by other technologies. It contains approximately 850,000 empirically selected
single nucleotide polymorphisms (SNPs) spanning the entire genome with enriched coverage for 3,262
genes of known cytogenetics relevance in both constitutional and cancer applications. 
</p>

<h3>Affymetrix Cytoscan HD GeneChip Array</h3>
<p>
The CytoScan HD Array, which is included in the
<a href="https://www.thermofisher.com/order/catalog/product/901835"
target="_blank">CytoScan HD Suite</a>, provides the broadest coverage and highest performance for
detecting chromosomal aberrations. CytoScan HD Suite has greater than 99% sensitivity and can
reliably detect 25-50kb copy number changes across the genome at high specificity with
single-nucleotide polymorphism (SNP) allelic corroboration. With more than 2.6 million copy number
markers, CytoScan HD Suite covers all OMIM and RefSeq genes.
</p>

<h3>Bionano DLE-1 CTTAAG sites</h3>

<p>
Bionano Laboratories provides access to Optical Genome Mapping (OGM) data for projects across a variety of
applications for researchers, clinicians, and pharmaceutical companies.</p>
<p>This track shows the CTTAAG sites used by the 
<a target="blank" href="https://www.bionano.com">Bionano Optical Genome Mapping system</a>,
an assay to detect structure variants.
</p>

<h2>Display Conventions and Configuration</h2>

<p>
Items in this track are colored according to their strand orientation. <b><font color=blue>Blue
indicates alignment to the negative strand</font></b>, and <b><font color=red>red indicates
alignment to the positive strand</font></b>.
</p>


<h2>Methods</h2>
<p>
The Agilent arrays were downloaded from their 
<a href="https://earray.chem.agilent.com/suredesign/search.htm" 
target="_blank">Agilent SureDesign</a> website tool on March 2022.</p>
<p>
The Illumina 450k and 850k (EPIC) tracks were created using a few columns from the
<a href="https://support.illumina.com/array/array_kits/infinium-methylationepic-beadchip-kit/downloads.html"
target="_blank">Infinium MethylationEPIC v1.0 B5 Manifest File (CSV Format)</a>
and was then converted into a bigBed.</p>
<p>
The Illumina CytoSNP-850K track was created by downloading the
<a href="https://webdata.illumina.com/downloads/productfiles/cytosnp-850k/v1-2/cytosnp-850k-v1-2-ns550-d2-manifest-file-csv.zip"
target="_blank">CytoSNP-850K v1.2 Manifest File (CSV Format) (GRCh38)</a> file and then converted
into a bigBed file.
</p>
<p>
The Affymetrix Cytoscan HD GeneChip Array track was created by converting the 
<a href="https://hgdownload.soe.ucsc.edu/gbdb/hg38/genotypeArrays/CytoScanHD_Accel_Array.na36.bed.zip"
target="_blank">CytoScanHD_Accel_Array.na36.bed.zip</a>
into a bigBed file.
</p>

<h2>Data Access</h2>
<P>
The raw data can be explored interactively with the
<a target="_blank" href="/cgi-bin/hgTables">Table Browser</a> or the
<a target="_blank" href="/cgi-bin/hgIntegrator">Data Integrator</a>.
For automated analysis, the data may be queried from our
<a target="_blank" href="/goldenPath/help/api.html">REST API</a> 
or downloaded from our <a target="_blank" href="https://hgdownload.soe.ucsc.edu/gbdb/hg38/">
Downloads site</a>. Please refer to our
<a target="_blank" href="https://groups.google.com/a/soe.ucsc.edu/forum/#!forum/genome">
mailing list archives</a> for questions, or our
<a target="_blank" href="/FAQ/FAQdownloads.html#downloads36">
Data Access FAQ</a> for more information.
</P>

<H2>Credits</H2>
<P>
Thanks to the Aliglent and Illumina support teams for sharing the data and the UCSC Genome Browser
engineers for configuring the data.</p>
