<h2>Description</h2>
<p>
The gnomAD v3 track shows variants and derived information from 71,702 whole genomes (and no exomes), all mapped to the 
GRCh38/hg38 reference sequence. Most of the genomes from v2 are included in v3. For more detailed 
information on gnomAD v3, see the related <a target="_blank" 
href="https://macarthurlab.org/2019/10/16/gnomad-v3-0/">blog post</a>.</p>

<p>
The gnomAD v2 tracks show variants from 125,748 exomes and 15,708 whole genomes, all mapped to 
the GRCh37/hg19 reference sequence and lifted to the GRCh38/hg38 assembly. The data originate 
from 141,456 unrelated individuals sequenced as part of various population-genetic and 
disease-specific <a target="_blank" href="https://gnomad.broadinstitute.org/about">studies</a> 
collected by the <a target="_blank" 
href="https://gnomad.broadinstitute.org/">Genome Aggregation Database (gnomAD)</a>, release 2.1.1.
Raw data from all studies have been reprocessed through a unified pipeline and jointly
variant-called to increase consistency across projects. For more information on the processing
pipeline and population annotations, see the following <a target="_blank"
href="https://macarthurlab.org/2017/02/27/the-genome-aggregation-database-gnomad">blog post</a>
and the 2.1.1 <a href="https://storage.googleapis.com/gnomad-public/release/2.1.1/README.txt"
target="_blank">README</a>.</p>
<p>
gnomAD v2 data are based on the GRCh37/hg19 assembly. These tracks display the 
GRCh38/hg38 lift-over provided by gnomAD on their downloads site.
</p>

<p>On hg38 only, a subtrack "Gnomad mutational constraint" aka "Genome
non-coding constraint of haploinsufficient variation (Gnocchi)" captures the
depletion of variation caused by purifying natural selection.
This is similar to negative selection on loss-of-function (LoF) for genes, but
can be calculated for non-coding regions, too. Briefly, for any 1kbp window in
the genome, a model based on trinucleotide sequence context, base-level
methylation, and regional genomic features predicts expected number of mutations,
and compares this number to the observed number of mutations using a Z-score (see Chen et al 2024 
in the Reference section for details). The chrX scores were added as received from the authors, 
as there are no mutations available for chrX, they are more speculative than the ones on the autosomes.</p>

<p>The gnomAD MPC score (Missense Deleteriousness Prediction by Constraint) is available only on hg19.</p>

<p>
For questions on the gnomAD data, also see the <a target="_blank" 
href="https://gnomad.broadinstitute.org/faq">gnomAD FAQ</a>.</p>

<h2>Display Conventions</h2>
<dl>
  <dt></dt>
  <dd><li>In <button type="button">dense</button> mode, a vertical line is drawn at the position of
    each variant.</li>
  <li>In <button type="button">pack</button> mode, &quot;ref&quot; and &quot;alt&quot; alleles are
    displayed to the left of a vertical line with colored portions corresponding to allele counts.
    Hovering the mouse pointer over a variant pops up a display of alleles and counts.</li></dd>
</dl>

<h2>Data Access</h2>

<p>
The raw data can be explored interactively with the <a target="_blank" href="../cgi-bin/hgTables">
Table Browser</a>, or the <a target="_blank" href="../cgi-bin/hgIntegrator">Data Integrator</a>. For
automated analysis, the data may be queried from our <a target="_blank" 
href="/goldenPath/help/api.html">REST API</a>, and the genome annotations are stored in files that
can be downloaded from our <a
href="http://hgdownload.soe.ucsc.edu/gbdb/$db/gnomAD/" target="_blank">download server</a>, subject
to the conditions set forth by the gnomAD consortium (see below). Coverage values
and constraint scores for the genome are in <a target="_blank" href="../goldenPath/help/bigWig.html">bigWig</a> files in
the coverage/ subdirectory. Variant VCFs can be found in the vcf/ subdirectory.</p>
<p>
The data can also be found directly from the gnomAD <a target="_blank" 
href="https://gnomad.broadinstitute.org/downloads">downloads page</a>. Please refer to
our <a href="https://groups.google.com/a/soe.ucsc.edu/forum/#!forum/genome"
target="_blank">mailing list archives</a> for questions, or our <a target="_blank"
href="../FAQ/FAQdownloads.html#download36">Data Access FAQ</a> for more information.</p>

<p>The mutational constraints score ("Gnocchi") was updated in October 2022 from a previous,
now deprecated, pre-publication version. The old version can be found in our 
<a href="https://hgdownload.soe.ucsc.edu/goldenPath/archive/hg38/gnomad/">archive
directory</a> on the download server. It can be loaded by copying the URL into
our "Custom tracks" input box.  </p>

<h2>Credits</h2>
<p>
Thanks to the <a href="https://gnomad.broadinstitute.org/about" target="_blank">Genome Aggregation
Database Consortium</a> for making these data available. The data are released under the <a
href="https://opendatacommons.org/licenses/odbl/1.0/" target="_blank">ODC Open Database License
(ODbL)</a> as described <a href="https://gnomad.broadinstitute.org/terms" target="_blank">here</a>.
</p>

<h2>References</h2>

<p>
Karczewski KJ, Francioli LC, Tiao G, Cummings BB, Alfoldi J, Wang Q, Collins RL, Laricchia KM,
Ganna A, Birnbaum DP <em>et al</em>.
<a target="_blank" href="https://www.biorxiv.org/content/10.1101/531210v3.full">Variation across 
141,456 human exomes and genomes reveals the spectrum of loss-of-function intolerance across human 
protein-coding genes</a>. doi: https://doi.org/10.1101/531210.
</p>
<p>
Lek M, Karczewski KJ, Minikel EV, Samocha KE, Banks E, Fennell T, O'Donnell-Luria AH, Ware JS, Hill
AJ, Cummings BB <em>et al</em>.
<a href="https://www.nature.com/articles/nature19057" target="_blank">Analysis of protein-coding
genetic variation in 60,706 humans</a>. <em>Nature</em>. 2016 Aug 17;536(7616):285-91.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27535533" target="_blank">27535533</a>;
PMC: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5018207/" target="_blank">PMC5018207</a>
</p>
<p>
Chen S, Francioli LC, Goodrich JK, Collins RL, Kanai M, Wang Q, Alf&#246;ldi J, Watts NA, Vittal C,
Gauthier LD <em>et al</em>.
<a href="https://doi.org/10.1038/s41586-023-06045-0" target="_blank">
    A genomic mutational constraint map using variation in 76,156 human genomes</a>.
<em>Nature</em>. 2024 Jan;625(7993):92-100.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/38057664" target="_blank">38057664</a><br>
(We added the data in 2021, then later referenced the 2022 Biorxiv preprint, in which the track was not called "Gnocchi" yet)
</p>

