<h2>Description</h2>
<p>
This track shows genetic variants likely affecting proximal gene expression in 49 human tissues
from the
<a target="_blank" href="https://gtexportal.org/home/">Genotype-Tissue Expression (GTEx)</a> V8
data release.
The data items displayed are gene expression quantitative trait loci within 1MB
of gene transcription start sites (cis-eQTLs), significantly associated with
gene expression and in the credible set of variants for the gene at a high
confidence level (95%). Each eQTL annotation includes the tissue and causal
posterior probability (CPP, described in
<a target="_blank" href="https://www.genetics.org/content/198/2/497.long">Hormozdiari et al.</a>)
</p>

<h2>Display Conventions</h2>
<p>
The track shows gene/variant pairs for 49 GTEx tissues, following the display
conventions for
<a href="../goldenPath/help/interact.html" target="_blank">interact tracks</a>
where an arc is drawn between the variant and
gene. Items in the track are colored according to their tissue, with the color
matching those in the
<a target="_blank" href="/cgi-bin/hgGtexTrackSettings?g=gtexGeneV8">GTEx Gene V8 Track</a>
Hovering over the
center-point on that arc will display the dbSNP rsID, target gene, tissue, and
CPP. Clicking an item will show the details of that interaction as well as link
to view more details on the GTEx website.
</p>

<p>
Track configuration supports filtering by tissue, gene, or CPP.
</p>

<h2>Methods</h2>
<p>
Laboratory and RNA-seq analysis methods for GTEx V8 are summarized in the 
<a target="_blank" href="/cgi-bin/hgGtexTrackSettings?g=gtexGeneV8">GTEx Gene Track</a> description page.
</p>

<p>
Cis-eQTL's were identified from GTEx RNA-seq and genotype data (variants with minor allele 
frequency &gt;= 1%) in 49 tissues (those with sample size &gt;=70) using 
the <a target="_blank" href="http://fastqtl.sourceforge.net">FastQTL</a> mapper at 5% FDR threshold, 
as part of the GTEx project v8 analysis.  
These cis-eQTL's were then analyzed together with genome variation information (LD) using the 
<a target="_blank" href="http://genetics.cs.ucla.edu/caviar/">CAVIAR</a> statistical framework
to quantify the probability a variant is causal as part of GTEx downstream analysis.
The UCSC track was created using the CAVIAR high-confidence set, which
represents the high causal variants that have a (CPP) of &gt; 0.1).
</p>
<p>
Raw data for these analyses are available from the
<a href="https://gtexportal.org/home/protectedDataAccess" target="_blank">GTEx Portal</a>.
</p>

<h2>Data Access</h2>
<p>
The raw data for this track can be accessed in multiple ways. It can be explored interactively 
using the <a href="../cgi-bin/hgTables" target="_blank">Table Browser</a> or 
<a href="../cgi-bin/hgIntegrator" target="_blank">Data Integrator</a>. You can also access the data
entries in JSON format through our <a href="http://genome.ucsc.edu/goldenPath/help/api.html">
JSON API</a>.</p>

<p>
The data in this track are organized in <a href="../goldenPath/help/interact.html"
target="_blank">bigInteract</a> file format. The underlying file
<a href="http://hgdownload.soe.ucsc.edu/gbdb/$db/gtex/eQtl/gtexEqtlsCaviarHighConf.interact.bb"
target="_blank">gtexEqtlsCaviarHighConf.interact.bb</a>
can be obtained from our downloads server.
Individual regions or the whole set of genome-wide annotations can be obtained using our tool
<tt>bigBedToBed</tt> which can be compiled from the source code or downloaded as a precompiled
binary for your system from the utilities directory linked below. For example, to extract only
annotations in a given region, you could use the following command:</p>
<p>
<tt>bigBedToBed http://hgdownload.soe.ucsc.edu/gbdb/$db/gtex/eQtl/gtexEqtlsCaviarHighConf.interact.bb
-chrom=chr16 -start=34990190 -end=36727467 stdout</tt></p>

<h2>Credits</h2>
Thanks to GTEx investigators, analysts, and portal team for providing this data.

<h2>References</h2>
<p>
GTEx Consortium.
<a href="https://www.science.org/doi/full/10.1126/science.aaz1776" target="_blank">
The GTEx Consortium atlas of genetic regulatory effects across human tissues</a>.
<em>Science</em>. 2020 Sep 11;369(6509):1318-1330.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32913098" target="_blank">32913098</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7737656/" target="_blank">PMC7737656</a>
</p>
<p>
Ongen H, Buil A, Brown AA, Dermitzakis ET, Delaneau O.
<a href="https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btv722"
target="_blank">
Fast and efficient QTL mapper for thousands of molecular phenotypes</a>.
<em>Bioinformatics</em>. 2016 May 15;32(10):1479-85.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26708335" target="_blank">26708335</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4866519/" target="_blank">PMC4866519</a>
</p>
<p>
Hormozdiari F, Kostem E, Kang EY, Pasaniuc B, Eskin E.
<a href="https://www.genetics.org/content/198/2/497.long" target="_blank">
Identifying causal variants at loci with multiple signals of association</a>.
<em>Genetics</em>. 2014 Oct;198(2):497-508.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25104515" target="_blank">25104515</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4196608/" target="_blank">PMC4196608</a>
</p>
<p>
GTEx Consortium.
<a href="https://www.nature.com/articles/ng.2653" target="_blank">
The Genotype-Tissue Expression (GTEx) project</a>.
<em>Nat Genet</em>. 2013 Jun;45(6):580-5.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23715323" target="_blank">23715323</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4010069/" target="_blank">PMC4010069</a>
</p>
<p>
<a href="https://www.gtexportal.org/home/documentationPage">
GTEx Portal Documentation</a>
</p>
