<h2>Description</h2>
<p>
The GENCODE track is composed of all the gene models in the <a target="_blank"
href="https://www.gencodegenes.org/">GENCODE v32 release.</a> By default, only the basic gene set is
displayed, which is a subset of the comprehensive gene set. The basic set represents transcripts
that GENCODE believes will be useful to the majority of users.</p>

<p>
The track includes protein-coding genes, non-coding RNA genes, and pseudo-genes, though pseudo-genes
are not displayed by default. It contains annotations on the reference chromosomes as well as
assembly patches and alternative loci (haplotypes).</p>

<p>
The following table provides statistics for the v32 release derived from the GTF file that contains
annotations only on the main chromosomes. More information on how they were generated can be found
in the <a target="_blank" href="https://www.gencodegenes.org/human/stats.html">GENCODE site</a>.</p>

<p>
<blockquote><table class="stdTbl">
<tr><th COLSPAN=4>GENCODE v32 Release Stats</th></tr>
<tr align=left><th>Genes</th><th>Observed</th><th>Transcripts</th><th>Observed</th></tr>
<tr align=left><td>Protein-coding genes</td><td>19,965</td><td>Protein-coding transcripts</td><td>83,986</td></tr>
<tr align=left><td>Long non-coding RNA genes</td><td>17,910</td><td><font size="-1">- full length protein-coding</font></td><td>57,935</td></tr>
<tr align=left><td>Small non-coding RNA genes</td><td>7,576</td><td><font size="-1">- partial length protein-coding</font></td><td>26,051</td></tr>
<tr align=left><td>Pseudogenes</td><td>14,749</td><td>Nonsense mediated decay transcripts</td><td>15,811</td></tr>
<tr align=left><td>Immunoglobulin/T-cell receptor gene segments</td><td>645</td><td>Long non-coding RNA loci transcripts</td><td>48,351</td></tr>
</table><BR>
</blockquote></p>

<p>
For more information on the different gene tracks, see our <a target="_blank"
href="/FAQ/FAQgenes.html">Genes FAQ</a>.</p>

<h2>Display Conventions and Configuration</h2>
<p>
By default, this track displays only the basic GENCODE set, splice variants, and non-coding genes.
It includes options to display the comprehensive GENCODE set and pseudogenes. To customize these
options, the respective boxes can be checked or unchecked at the top of this description page. Our
FAQ includes examples of <a target="_blank" href="/FAQ/FAQgenes.html#justsingle">how to display a
single transcript per gene</a> and <a target="_blank" href="/FAQ/FAQgenes.html#gencode">switching
between the basic and comprehensive gene sets</a>.</p>

<p>
This track also includes a variety of labels which identify the transcripts when visibility is set
to &quot;full&quot; or &quot;pack&quot;. Gene symbols (e.g. NIPA1) are displayed by default, but
additional options include GENCODE Transcript ID (ENST00000561183.5), UCSC Known Gene ID
(uc001yve.4), UniProt Display ID (Q7RTP0) and OMIM ID (608145). Additional information about gene
and transcript names can be found in our
<a target="_blank" href="/FAQ/FAQgenes.html#genename">FAQ</a>.</p>

<p>
This track, in general, follows the display conventions for <a target="_blank"
href="../goldenPath/help/hgTracksHelp.html#GeneDisplay">gene prediction tracks</a>. The exons for
putative non-coding genes and untranslated regions are represented by relatively thin blocks, while
those for coding open reading frames are thicker. The following color key is used:</p>

<ul>
  <li><span style="font-weight: bold;">Black</span> -- feature has a corresponding entry in the
  Protein Data Bank (PDB)</li>
  <li><span style="color: #0C0C78; font-weight: bold;">Dark blue</span> -- transcript has been
  reviewed or validated by either the RefSeq or SwissProt staff</li>
  <li><span style="color: #5050A0; font-weight: bold;">Medium blue</span> -- other RefSeq
  transcripts</li>
  <li><span style="color: #8282D2; font-weight: bold;">Light blue</span> -- non-RefSeq
  transcripts</li>
</ul>

<p>
This track contains an optional <a target="_blank"
href="../goldenPath/help/hgCodonColoring.html">codon coloring feature</a> that allows users to
quickly validate and compare gene predictions. There is also an option to display the data as
a <a target="_blank" href="../goldenPath/help/hgWiggleTrackHelp.html">density graph</a>, which
can be helpful for visualizing the distribution of items over a region.</p>

<h2>Methods</h2>
<p>
The GENCODE v32 track was built from the <a href="https://www.gencodegenes.org/human/" 
target="_blank">GENCODE downloads</a> file 
<code>gencode.v32.chr_patch_hapl_scaff.annotation.gff3.gz</code>. Data from other sources 
were correlated with the GENCODE data to build the knownTo tables.</p>

<h2>Related Data</h2>
<p>
The GENCODE Genes transcripts are annotated in numerous tables, each of which is also available as a
<a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/" target="_blank">downloadable
file</a>. These include tables that link GENCODE Genes transcripts to external datasets (such as
<i>knownToLocusLink</i>, which maps GENCODE Genes transcripts to Entrez identifiers, previously
known as Locus Link identifiers), and tables that detail some property of GENCODE Genes transcript
sequences (such as <i>knownToPfam</i>, which identifies any Pfam domains found in the GENCODE Genes
protein-coding transcripts).</p>

<p>
One can see a full list of the associated tables in the <a href="/cgi-bin/hgTables"
target="_blank">Table Browser</a> by selecting GENCODE Genes from the <b>track</b> menu; this list
is then available on the <b>table</b> menu. Note that some of these tables refer to GENCODE Genes
by its former name of <i>Known Genes</i>, sometimes abbreviated as <i>known</i> or <i>kg</i>. While
the complete set of annotation tables is too long to describe, some of the more important tables are
described below.</p>
<ul> 
  <li> <b>kgXref</b> identifies the RefSeq, SwissProt, Rfam, or tRNA sequences (if any) which are
  associated with each transcript.</li>
  <li> <b>knownToRefSeq</b> identifies the RefSeq transcript that each GENCODE Genes transcript is
  most closely associated with. That RefSeq transcript is the RefSeq transcript that the GENCODE
  Genes transcript overlaps at the most bases.</li>
  <li> <b>knownGeneMrna</b> contains the genomic sequence for each of the GENCODE Genes models.
  This may not be the same as the actual mRNA used to validate the gene model.</li>
  <li> <b>knownGenePep</b> contains the protein sequences derived from the knownGeneMrna transcript
  sequences. Any protein-level annotations, such as the contents of the knownToPfam table, are based
  on these sequences.</li>
  <li> <b>knownIsoforms</b> maps each transcript to a <em>cluster ID</em>, a cluster of isoforms of
  the same gene.</li>
  <li> <b>knownCanonical</b> identifies the canonical isoform of each cluster ID or gene using the
  ENSEMBL gene IDs to define each cluster. The canonical transcript is chosen using the APPRIS
  principal transcript when available. If no APPRIS tag exists for any transcript associated with
  the cluster, then a transcript in the BASIC set is chosen. If no BASIC transcript exists, then the
  longest isoform is used.</li>
</ul>

<h2>Data access</h2>
<p>
GENCODE Genes and its associated tables can be explored interactively using the
<a href="../goldenPath/help/api.html" target="_blank">REST API</a>, the
<a href="/cgi-bin/hgTables" target="_blank">Table Browser</a> or the
<a href="/cgi-bin/hgIntegrator" target="_blank">Data Integrator</a>. 
The genePred format files for hg38 are available from our 
<a target="_blank" href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/">
downloads directory</a> as knownGene.txt.gz or in our
<a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/genes/" target="_blank">
GTF download directory</a> as hg19.knownGene.gtf.gz. 
All the tables can also be queried directly from our public MySQL
servers, with more information available on our
<a target="_blank" href="/goldenPath/help/mysql.html">help page</a> as well as on
<a target="_blank" href="http://genome.ucsc.edu/blog/tag/mysql/">our blog</a>.</p>

<h2>Credits</h2>
<p>
The GENCODE Genes track was produced at UCSC from the GENCODE comprehensive gene set using a
computational pipeline developed by Jim Kent and Brian Raney.</p>

<h2>References</h2>
<p>
Harrow J, Frankish A, Gonzalez JM, Tapanari E, Diekhans M, Kokocinski F, Aken BL, Barrell D, Zadissa
A, Searle S <em>et al</em>.
<a href="https://genome.cshlp.org/content/22/9/1760.long" target="_blank">
GENCODE: the reference human genome annotation for The ENCODE Project</a>.
<em>Genome Res</em>. 2012 Sep;22(9):1760-74.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22955987" target="_blank">22955987</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3431492/" target="_blank">PMC3431492</a>
</p>

<p>
Harrow J, Denoeud F, Frankish A, Reymond A, Chen CK, Chrast J, Lagarde J, Gilbert JG, Storey R,
Swarbreck D <em>et al</em>.
<a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2006-7-s1-s4" target="_blank">
GENCODE: producing a reference annotation for ENCODE</a>.
<em>Genome Biol</em>. 2006;7 Suppl 1:S4.1-9.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/16925838" target="_blank">16925838</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1810553/" target="_blank">PMC1810553</a>
</p>

<p>A full list of GENCODE publications is available
at <a href="https://www.gencodegenes.org/pages/publications.html" target="_blank">The GENCODE
Project web site</a>.
</p>

<h2>Data Release Policy</h2>
<p>GENCODE data are available for use without restrictions.</p>
