<H2>Description</H2>
<P>
This is a super track for non-coding RNA data, subtracks represent some form of non-coding RNA data. 
</P>
<H2>Credits</h2> 

<p>The body map RNA-Seq data was kindly provided by the Gene Expression
Applications research group at <a href="https://www.illumina.com" target=_blank>Illumina</a>.</p>

Genome coordinates for the sno/miRNA track were obtained from the miRBase sequences
<A HREF="ftp://mirbase.org/pub/mirbase/CURRENT/" 
TARGET=_BLANK>FTP site</A> and from 
<A HREF="https://www-snorna.biotoul.fr//coordinates.php" TARGET=_BLANK>
snoRNABase coordinates download page.</A>
<P>

<h2>References</h2> 

<P>
When making use of these data, please cite the folowing articles in addition to
the primary sources of the miRNA sequences:</P>
<P>
Griffiths-Jones S, Saini HK, van Dongen S, Enright AJ.
<A HREF="https://academic.oup.com/nar/article/36/suppl_1/D154/2507930/miRBase-tools-for-microRNA-
genomics" TARGET=_BLANK>miRBase: tools for microRNA genomics</A>.
<EM>Nucleic Acids Res.</EM> 2008 Jan 1;36(Database issue):D154-8.</P>
<P>
Griffiths-Jones S, Grocock RJ, van Dongen S, Bateman A, Enright AJ.
<A HREF="https://academic.oup.com/nar/article/34/suppl_1/D140/1133563/miRBase-microRNA-sequences-
targets-and-gene" TARGET=_BLANK>miRBase: microRNA sequences, targets and gene nomenclature</A>.
<EM>Nucleic Acids Res.</EM> 2006 Jan 1;34(Database issue):D140-4.</P>
<P>
Griffiths-Jones S.
<A HREF="https://academic.oup.com/nar/article/32/suppl_1/D109/2505171/The-microRNA-Registry"
TARGET=_blank>The microRNA Registry</A>.
<EM>Nucleic Acids Res.</EM> 2004 Jan 1;32(Database issue):D109-11.</P>
<P>
Weber MJ.
<A HREF="http://onlinelibrary.wiley.com/doi/10.1111/j.1432-1033.2004.04389.x/abstract"
TARGET=_blank>New human and mouse microRNA genes found by homology search</A>.
<P>
You may also want to cite The Wellcome Trust Sanger Institute 
<A HREF="http://www.mirbase.org/"
TARGET=_blank>miRBase</A> and The Laboratoire de Biologie Moleculaire 
Eucaryote <A HREF="https://www-snorna.biotoul.fr//" TARGET=_blank>snoRNABase</A>.</P>
<P>
The following publication provides guidelines on miRNA annotation:
Ambros V. <em>et al</em>., 
<A HREF="http://rnajournal.cshlp.org/content/9/3/277.abstract" 
TARGET=_blank>A uniform system for microRNA annotation</A>.  
<em>RNA.</em> 2003;9(3):277-9.</P>
<P>


<p>
Cabili MN, Trapnell C, Goff L, Koziol M, Tazon-Vega B, Regev A, Rinn JL.
<a href="http://genesdev.cshlp.org/content/25/18/1915" target="_blank">
Integrative annotation of human large intergenic noncoding RNAs reveals global properties and
specific subclasses</a>.
<em>Genes Dev</em>. 2011 Sep 15;25(18):1915-27.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21890647" target="_blank">21890647</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3185964/" target="_blank">PMC3185964</a>
</p>

<p>
Trapnell C, Williams BA, Pertea G, Mortazavi A, Kwan G, van Baren MJ, Salzberg SL, Wold BJ, Pachter
L.
<a href="https://www.nature.com/articles/nbt.1621" target="_blank">
Transcript assembly and quantification by RNA-Seq reveals unannotated transcripts and isoform
switching during cell differentiation</a>.
<em>Nat Biotechnol</em>. 2010 May;28(5):511-5.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/20436464" target="_blank">20436464</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3146043/" target="_blank">PMC3146043</a>
</p>


