<H2>Description</H2> 
<P>
The recombination rate track represents calculated rates of recombination based
on the genetic maps from deCODE (Halldorsson <EM>et al.</EM>, 2019) and 1000 Genomes
(2013 Phase 3 release, lifted from hg19). The deCODE map is more recent, has a higher 
resolution and was natively created on hg38 and therefore recommended. 
For the Recomb. deCODE average track, the recombination rates for chrX represent the female rate.
</P>

<p>This track also includes a subtrack with all the
individual deCODE recombination events and another subtrack with several thousand
de-novo mutations found in the deCODE sequencing data. These two tracks are hidden by
default and have to be switched on explicitly on the configuration page.
</p>

<h2>Display Conventions and Configuration</h2>
<p>
This is a super track that contains different subtracks, three with the deCODE
recombination rates (paternal, maternal and average) and one with the 1000
Genomes recombination rate (average). These tracks are in 
<a href="https://genome.ucsc.edu/goldenPath/help/hgWiggleTrackHelp.html" target=_blank>signal graph</a>
(wiggle) format. By default, to show most recombination hotspots, their maximum
value is set to 100 cM, even though many regions have values higher than 100.
The maximum value can be changed on the configuration pages of the tracks.
</p>

<p>
There are two more tracks that show additional details provided by deCODE: one
subtrack with the raw data of all cross-overs tagged with their proband ID and
another one with around 8000 human de-novo mutation variants that are linked to
cross-over changes.
</p>

<H2>Methods</H2>
<P>
The deCODE genetic map was created at 
<A HREF="https://www.decode.com/" TARGET=_blank>deCODE Genetics</A>. It is based 
on microarrays assaying 626,828 SNP markers that allowed to identify 1,476,140 crossovers in
56,321 paternal meioses and 3,055,395 crossovers in 70,086 maternal meioses.
In total, the data is based on 4,531,535 crossovers in 126,427 meioses. By
using WGS data with 9,305,070 SNPs, the boundaries for 761,981 crossovers were
refined: 247,942 crossovers in 9423 paternal meioses and 514,039 crossovers in
11,750 maternal meioses. The average resolution of the genetic map is 682 base
pairs (bp): 655 and 708 bp for the paternal and maternal maps, respectively.
</p>

<p>The 1000 Genomes genetic map is based on the <a
    href="https://mathgen.stats.ox.ac.uk/impute/1000GP_Phase3.html"
target=_blank>IMPUTE genetic map based on 1000 Genomes Phase 3</a>, on hg19 coordinates. It
was converted to hg38 by Po-Ru Loh at the Broad Institute.  After a run of 
liftOver, he post-processed the data to deal with situations in which
consecutive map locations became much closer/farther after lifting. The
heuristic used is sufficient for statistical phasing but may not be optimal for
other analyses. For this reason, and because of its higher resolution, the DeCODE
map is therefore recommended for hg38.
</p>

<p>As with all other tracks, the data conversion commands and pointers to the
original data files are documented in the 
<a href="https://github.com/ucscGenomeBrowser/kent/blob/master/src/hg/makeDb/doc/hg38/recombRate.txt"
target=_blank>makeDoc file</a> of this track.</p>

<h2>Data Access</h2>
<p>
The raw data can be explored interactively with the <a href="../hgTables">Table Browser</a>, or
the <a href="../hgIntegrator">Data Integrator</a>. For automated access, this track, like all
others, is available via our <a href="../goldenPath/help/api.html">API</a>.  However, for bulk
processing, it is recommended to download the dataset.
</p>

<p>
For automated download and analysis, the genome annotation is stored at UCSC in bigWig and bigBed
files that can be downloaded from
<a href="http://hgdownload.soe.ucsc.edu/gbdb/hg38/" target="_blank">our download server</a>.
Individual regions or the whole genome annotation can be obtained using our tools <tt>bigWigToWig</tt>
or <tt>bigBedToBed</tt> which can be compiled from the source code or downloaded as a precompiled
binary for your system. Instructions for downloading source code and binaries can be found
<a href="http://hgdownload.soe.ucsc.edu/downloads.html#utilities_downloads">here</a>.
The tools can also be used to obtain features confined to a given range, e.g.,
<br><br>
<tt>bigWigToBedGraph -chrom=chr17 -start=45941345 -end=45942345 http://hgdownload.soe.ucsc.edu/gbdb/hg38/recombRate/recombAvg.bw stdout</tt>
<br>
</p>

<p>
Please refer to our
<a HREF="../FAQ/FAQdownloads.html#download36" target=_blank>Data Access FAQ</a>
for more information.
</p>

<H2>Credits</H2>
<P>
This track was produced at UCSC using data that are freely available for
the <A HREF="https://www.decode.com/" TARGET=_blank>deCODE</A>
and <a href="https://mathgen.stats.ox.ac.uk/impute/1000GP_Phase3.html"
target=_blank>1000 Genomes genetic maps</a>. Thanks to Po-Ru Loh at the
Broad Institute for providing the code to lift the hg19 1000 Genomes map data to hg38.
</P>  

<H2>References</H2>
<p>
1000 Genomes Project Consortium., Abecasis GR, Altshuler D, Auton A, Brooks LD, Durbin RM, Gibbs RA,
Hurles ME, McVean GA.
<a href="https://www.ncbi.nlm.nih.gov/pubmed/20981092" target="_blank">
    A map of human genome variation from population-scale sequencing</a>.
<em>Nature</em>. 2010 Oct 28;467(7319):1061-73.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/20981092" target="_blank">20981092</a>; PMC: <a
    href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3042601/" target="_blank">PMC3042601</a>
</p>

<p>
Halldorsson BV, Palsson G, Stefansson OA, Jonsson H, Hardarson MT, Eggertsson HP, Gunnarsson B,
Oddsson A, Halldorsson GH, Zink F <em>et al</em>.
<a href="https://www.ncbi.nlm.nih.gov/pubmed/30679340" target="_blank">
    Characterizing mutagenic effects of recombination through a sequence-level genetic map</a>.
<em>Science</em>. 2019 Jan 25;363(6425).
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30679340" target="_blank">30679340</a>
</p>
