<H2>Description</H2>
<P>
<a href="https://cancergenome.nih.gov/" target="_blank">
The Cancer Genome Atlas (TCGA)</a>, a collaboration between the
<a href="https://www.cancer.gov/" target="_blank">National Cancer Institute</a> (NCI)
and <a href="https://www.genome.gov/" target="_blank">
National Human Genome Research Institute</a> (NHGRI), has generated comprehensive,
multi-dimensional maps of the key genomic changes in 33 types of cancer. The TCGA
dataset, 2.5 petabytes of data describing tumor tissue and matched normal tissues from
more than 11,000 patients, is publically available and has been used widely by the
research community.

The Cancer Genome Atlas is a NIH-funded project to catalog genetic mutations
responsible for cancer. The data shown here is RNA-seq expression data produced by the
consortium.</P>

<P>For questions or feedback on the data, please contact 
<a href="https://cancergenome.nih.gov/abouttcga/peoplecontacts/tcgacontacts"
target="_blank">TCGA</a>.
</p>

<H3>TCGA Gene Expression</H3>
<p>
The gene track shows RNA expression level for each TCGA tissue in GENCODE canonical
genes. The gene scores are a total of all transcripts in that gene. </p>

<H3>TCGA Transcript Expression</H3>
<p>
The transcript track shows RNA expression levels for each TCGA tissue using GENCODE v23
transcripts.</p>


<H2>Display Conventions</H2>
<P>
In Full and Pack display modes, expression for each genomic item (gene/transcript) is
represented by a colored bar chart, where the height of each bar represents the median
expression level across all samples for a tissue, and the bar color indicates the
tissue.</p>
<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<img border="1" src="../images/gtex/gtexGeneTcap.png"><br>
The bar chart display has the same width and tissue order for all genomic items.
Mouse hover over a bar will show the tissue and median expression levels
The Squish display mode draws a rectangle for each gene, colored to indicate the tissue
with highest expression level if it contributes more than 10% to the overall expression
(and colored black if no tissue predominates).
In Dense mode, the darkness of the grayscale rectangle displayed for the gene reflects the total
median expression level across all tissues.</p>

This track was designed to be used in conjunction with the GTEx expression tracks that can act as a control.

The color of each cancer was derived by mapping the tissue of origin to the closest GTEx tissue, then
taking the GTEx tissues color.  Five cancers did not have a matching GTEx tissue and were assigned 
a rainbow color scheme, these cancers are Cholangiocarcinoma, Esophageal carcinoma, Head and Neck 
squamous cell carcinoma, Sarcoma and Uveal Melanoma. 

The ordering of the cancers is based on the alphabetical ordering of their GTEX tissues. The five cancers
that did not match were ordered alphabetically. 
<P>

<H2>Methods</H2>

<p>TCGA chose cancers for study based on two broad criteria; poor prognosis/overall 
public health impact and availability of human tumor and matched normal tissue samples that meet 
<a HREF ="https://cancergenome.nih.gov/cancersselected/biospeccriteria">TCGA standards</a>. 
</p>
RNA sequencing was performed using a polyA library and the Illumina HiSeq 2000 platform. It was all
performed by UNC.
<p>
Sequence reads for this track were quantified to the hg38/GRCh38 human genome using kallisto assisted
by the GENCODE v23 transcriptome definition. Read quantification was performed at UCSC by the 
Computational Genomics lab, using the 
<a href="https://xenabrowser.net/datapages/?host=https://toil.xenahubs.net">Toil pipeline</a>
. The resulting kallisto files were combined to 
generate a transcript per million (tpm) expression matrix using the UCSC tool kallistoToMatrix.
By totaling the TPM values for all transcripts
associated to the canonical transcript/gene, a condensed gene per million (gpm) matrix was made.
</p>
<p>
For both matrices average expression values for each tissue were calculated and used to generate a bed6+5 file 
that is the base of each track. This was done using the UCSC tool expMatrixToBarchartBed. The bed 
track was then converted to a bigBed file using the UCSC tool bedToBigBed.
</p>

<H2>Credits</H2>
Data shown here are in whole based upon data generated by the 
<a href="http://cancergenome.nih.gov/" target="_blank">TCGA Research Network</a>.
John Vivian, Melissa Cline, and Benedict Paten of the UCSC Computational Genomics lab were
responsible for the sequence read quantification used to produce this track. Chris Eisenhart 
and Kate Rosenbloom of the UCSC Genome Browser group were responsible for data file
post-processing, track configuration and display type.
<P>

<H2>References</H2>
<p>
<p>
J. Vivian et al., 
<a href="http://biorxiv.org/content/early/2016/07/07/062497" target="_blank">
	Rapid and efficient analysis of 20,000 RNA-seq samples with Toil</a>
<em> bioRxiv</em>  bioRxiv, vol. 2, p. 62497, 2016.
</p>
