<h2>Description</h2>
<p>
These tracks contain information relevant to the regulation of transcription from the
<a href="/ENCODE/" target="_blank">ENCODE Project</a>.

<ul>
  <li>The <i>TF rPeak Clusters</i> track shows genomic regions bound by DNA-associated proteins 
involved in transcriptional regulation from ENCODE 4.</li>
  <li>The <i>Transcription</i> track shows transcription
levels assayed by sequencing of polyadenylated RNA from a variety of cell types.</li>
  <li>The <i>Layered H3K4Me1</i> and <i>Layered H3K27Ac</i> tracks show where modification of histone proteins
is suggestive of enhancer and, to a lesser extent, other regulatory activity.  These histone 
modifications, particularly H3K4Me1, are quite broad.  The actual enhancers are typically just a 
small portion of the area marked by these histone modifications.</li>
  <li>The <i>Layered H3K4Me3</i> 
track shows a histone mark associated with promoters.</li>
  <li>The <i>DNase I Hypersensitivity</i> tracks indicate
where chromatin is hypersensitive to cutting by the DNase enzyme, which has 
been assayed in a large number of cell types. Regulatory regions, in general, tend to be 
DNase-sensitive, and promoters are particularly DNase-sensitive.</li>
  <li>The <i>Txn Factor ChIP</i>
tracks show DNA regions where transcription factors, proteins responsible for 
modulating gene transcription, bind as assayed by chromatin immunoprecipitation with antibodies 
specific to the transcription factor followed by sequencing of the precipitated DNA (ChIP-seq).</li>
</ul>
</p>

<p>
These tracks complement each other and together can shed much light on regulatory DNA. The histone
marks are informative at a high level, but they have a resolution of just ~200 bases and do not
provide much in the way of functional detail.  The DNase hypersensitivity assay is higher in
resolution at the DNA level and can be done on a large number of cell types since it's just 
a single assay. At the functional level, DNase hypersensitivity suggests that a 
region is very likely to be regulatory in nature, but provides little information beyond that.
The transcription factor ChIP assay has a high resolution at the DNA level and, due to the very
specific nature of the transcription factors, is often informative with respect to functional
detail.  However, since each transcription factor must be assayed separately, the information is
only available for a limited number of transcription factors on a limited number of cell lines. 
Though each assay has its strengths and weaknesses, the fact that all of these assays are 
relatively independent of each other gives increased confidence when multiple tracks are 
suggesting a regulatory function for a region.
</p>

<p>
For additional information, please click on the hyperlinks for the individual tracks above.
Also note that additional histone marks and transcription information is available in other
ENCODE tracks.  This integrative supertrack just shows a selection of the most informative data of
most general interest.
</p>

<h2>Display Conventions</h2>
<p>
By default, the transcription and histone mark displays use a transparent overlay method of 
displaying data from a number of cell lines in a single track.  Each of the cell lines in this track
is associated with a particular color, and these colors are relatively light and saturated so
as to work best with the transparent overlay. The color of the transcription and histone mark tracks
match their versions from their lifted source on the hg19 assembly.</p>
<p>
The DNase tracks, which were not lifted from hg19, are colored differently 
to reflect similarity of cell types. There are three DNase tracks starting with a transparent
overlay DNase Signal Track to allow viewing signals from all 95 cell types in one track.
The individual signals and the same coloring scheme can also be found in the DNase HS Track
where processed peaks and hotspots are also called out as gray boxes with the darkness of
each box reflecting the underlying signal value. Lastly, in the DNase Clusters track all observed
hypersensitive regions in the different cell lines at the same location were clustered into a single box
where a number to the left of the box indicates how many cell types showed a hypersensitivity 
region and the darkness of the grey box is proportional to the the maximum value seen from one of
the underlying cell lines. Clicking on these item takes you to a details page where
additional information displays, such as the list of cell types that combined to form
the cluster in the DNase Clusters track.
</p>

<h2>Data Access</h2>
<p>
The raw data for ENCODE 3 Regulation tracks can be accessed from 
<a href="hgTables?db=hg38">
Table Browser</a> or combined with other data-sets through <a href="hgIntegrator">
Data Integrator</a>. For automated analysis and downloads, the track data files can be downloaded 
from our <a href="https://hgdownload.soe.ucsc.edu/gbdb/hg38/bbi/">downloads server</a> or queried
using the <a href="../../goldenPath/help/api.html">JSON API</a> or the 
<a href="../../goldenPath/help/mysql.html">Public SQL</a> Individual regions or the whole genome 
annotation can be accessed as text using our utility bigBedToBed. Instructions for downloading 
the utility can be found 
<a href="http://hgdownload.soe.ucsc.edu/downloads.html#utilities_downloads">here</a>. That 
utility can also be used to obtain features within a given range, e.g. 
<tt>bigBedToBed http://hgdownload.soe.ucsc.edu/gbdb/hg38/wgEncodeRegDnase/wgEncodeRegDnaseUwA549Hotspot.broadPeak.bb -chrom=chr21 -start=0 -end=100000000 stdout</tt></p>
<p>
For sorting transcription factor binding sites by cell type, we recommend you use the following
download 
<a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/encRegTfbsClustered/">file for hg38</a>.
</p>


<H2>Credits</H2>
<p>
Specific labs and contributors for these datasets are listed in the Credits section 
of the individual tracks in this super-track. The integrative view presented here was developed by Jim Kent at UCSC.</P>

<H2> Data Use Policy </H2>
<P> <B>Users may freely download, analyze and publish results based on any ENCODE data without 
restrictions.</B>
Researchers using unpublished ENCODE data are encouraged to contact the data producers to discuss possible coordinated publications; however, this is optional. </p>
<B><I>Users of ENCODE datasets are requested to cite the ENCODE Consortium and ENCODE
production laboratory(s) that generated the datasets used, as described in
<A target="_blank" href="https://www.encodeproject.org/help/citing-encode/">Citing ENCODE</A>.</B></I></p>
