</P>
<H2>Description</H2>
<P>
The aim of the <a href="https://www.gencodegenes.org/" target="_blank">GENCODE 
Genes project</a> (Harrow <em>et al.</em>, 2006) is to produce a set of 
highly accurate annotations of evidence-based gene features on the human reference genome.
This includes the identification of all protein-coding loci with associated
alternative splice variants, non-coding with transcript evidence in the public 
databases (NCBI/EMBL/DDBJ) and pseudogenes. A high quality set of gene
structures is necessary for many research studies such as comparative or 
evolutionary analyses, or for experimental design and interpretation of the 
results.</p>
<p> 
The GENCODE Genes tracks display the high-quality manual annotations merged 
with evidence-based automated annotations across the entire
human genome. The GENCODE gene set presents a full merge
between HAVANA manual annotation and Ensembl automatic annotation.
Priority is given to the manually curated HAVANA annotation using predicted
Ensembl annotations when there are no corresponding manual annotations. With 
each release, there is an increase in the number of annotations that have undergone
manual curation.  
This annotation was carried out on the GRCh38 (hg38) genome assembly.
</p>

<p>
For more information on the different gene tracks, see our <a target=_blank 
href="/FAQ/FAQgenes.html">Genes FAQ</a>.</p>

<h2>Display Conventions</h2>
<P>
These are multi-view composite tracks that contain differing data sets
(<EM>views</EM>).  Instructions for configuring multi-view tracks are
<A HREF="/goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
Only some subtracks are shown by default. The user can select which subtracks
are displayed via the display controls on the track details pages.
Further details on display conventions and data interpretation are available in the track descriptions.</p>

<h2>Data access</h2>
<p>
GENCODE Genes and its associated tables can be explored interactively using the
<a href="../goldenPath/help/api.html" target="_blank">REST API</a>, the
<a href="/cgi-bin/hgTables" target="_blank">Table Browser</a> or the
<a href="/cgi-bin/hgIntegrator" target="_blank">Data Integrator</a>.
The GENCODE data files for hg38 are available in our
<a target="_blank" href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/">
downloads directory</a> as wgEncodeGencode* files in genePred format.
All the tables can also be queried directly from our public MySQL
servers, with instructions on this method available on our
<a target="_blank" href="/goldenPath/help/mysql.html">MySQL help page</a> as well as on
<a target="_blank" href="http://genome.ucsc.edu/blog/tag/mysql/">our blog</a>.</p>

<h2>Release Notes</h2>
<p>
<span style="font-weight: bold;">GENCODE version 48</span>
corresponds to Ensembl 114.
</p>
<span style="font-weight: bold;">GENCODE version 47</span>
corresponds to Ensembl 113.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 46</span>
corresponds to Ensembl 112.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 45</span>
corresponds to Ensembl 111.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 44</span>
corresponds to Ensembl 110.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 43</span>
corresponds to Ensembl 109.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 42</span>
corresponds to Ensembl 108.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 41</span>
corresponds to Ensembl 107.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 40</span>
corresponds to Ensembl 106.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 39</span>
corresponds to Ensembl 105.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 38</span>
corresponds to Ensembl 104.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 37</span>
corresponds to Ensembl 103.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 36</span>
corresponds to Ensembl 102.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 35</span>
corresponds to Ensembl 101.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 34</span>
corresponds to Ensembl 100.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 33</span>
corresponds to Ensembl 99.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 30</span>
corresponds to Ensembl 96.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 29</span>
corresponds to Ensembl 94.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 28</span>
corresponds to Ensembl 92.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 27</span>
corresponds to Ensembl 90.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 26</span>
corresponds to Ensembl 88.
</p>
<p>
<span style="font-weight: bold;">GENCODE version 24</span>
corresponds to Ensembl 84.
</p>
<span style="font-weight: bold;">GENCODE version 23</span>
corresponds to Ensembl 81.
</p>
<span style="font-weight: bold;">GENCODE version 22</span>
corresponds to Ensembl 79.
</p>
<span style="font-weight: bold;">GENCODE version 20</span>
corresponds to Ensembl 76.
</p>
<p>
See also: <a href="https://www.gencodegenes.org/human/releases.html" 
target="_blank">The GENCODE Project Release History</a>.
</p>

<!--#insert file="../../wgEncodeGencodeCredits1.shared.html"-->
