<H2>Description</H2>
<P>
This track shows expression data from the <A HREF="http://zon.tchlab.org/" 
TARGET=_blank">Zon Lab</A> at Children's Hospital Boston using the
<A HREF="http://www.affymetrix.com/estore/browse/products.jsp?navMode=
34000&productId=131530&navAction=jump&aId=productsNav#1_1"
TARGET=_blank>Affymetrix Zebrafish GeneChip Genome Array</A>. Data is 
displayed in two subtracks showing the alignments of either the probeset
consensus sequences or target sequences for the Affymetrix Zebrafish GeneChip
(see the Affymetrix Zebrafish track for more details).
</P>

<H2>Methods</H2>
<P>
For detailed information about the experimental methods, see Weber 
<em>et al.</em> (2005) below. This expression data is for five strains of wild 
type zebrafish whole embryos at different developmental stages.</P>
<P>
Microarray data were preprocessed using MAS background correction, MAS
probe specific correction, and Loess normalization. 
<A HREF="http://bioconductor.org/download" TARGET=_blank">
BioConductor's</A> affy 
package was used to perform this processing. CEL files were normalized as
a single group.</P>
<P>  
The track color denotes the signal ratio for a specific 
developmental stage and strain on a log base 2 scale. The ratio is the 
expression value for a probeset divided by the median of expression values 
for that probeset across all experiments. Over-expressed genes are colored red; 
under-expressed genes are show in green. Black indicates that a gene
is neither over- nor under-expressed in the tissue. Uncolored boxes
(white on most browsers) represent missing data. </P>

<P>
Each row on the track or details page shows gene expression data for a 
single developmental stage and a single strain (see Strains and Stages table 
below) from wild type, whole embryo zebrafish. Data was collected by the 
<A HREF="http://zfrhmaps.tch.harvard.edu/ZonRHmapper/" 
TARGET=_blank>Children&#8217;s Hospital Boston Zebrafish Genome Initiative</A>.
</P>

<H2>Display Conventions and Configuration</H2>
<P>
This track follows the display conventions for composite tracks.
To display only selected subtracks, uncheck the boxes next to the tracks you 
wish to hide.</P>
<P>
This track may be configured to vary the type of experiment information 
shown. It is possible to display all arrays, arrays grouped by 
replicate means, or arrays grouped by replicate medians.  Choose the 
desired set of arrays from the <B>Combine arrays</B> list above.  
To display all arrays for each strain and developmental stage, select 
<EM>All Arrays</EM> from the list. The default display shows the Replicate 
Medians.
</P>

<H2>Strains and Stages Table</H2>
<P><B>Key: </B>TL = Tuebingen longfin; TU = Tuebingen; hpf. = hours 
post-fertilization.</P>
<P>
<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4>
<TR>
  <TH align=left>Strain</TH>
  <TH align=left>Stage</TH>
  <TH align=left>Replicates</TH>
</TR>
<TR>
  <TD>AB</TD>
  <TD>14 Somites</TD>
  <TD>6</TD>
</TR>
<TR>
  <TD>AB</TD>
  <TD>36 hpf.</TD>
  <TD>9</TD>
</TR>
<TR>
  <TD>golden</TD>
  <TD>14 somites</TD>
  <TD>1</TD>
</TR>
<TR>
  <TD>TL</TD>
  <TD>14 somites</TD>
  <TD>2</TD>
</TR>
<TR>
  <TD>TL</TD>
  <TD>36 hpf.</TD>
  <TD>7</TD>
</TR>
<TR>
  <TD>TU</TD>
  <TD>14 somites</TD>
  <TD>2</TD>
</TR>
<TR>
  <TD>TU</TD>
  <TD>15 somites</TD>
  <TD>3</TD>
</TR>
<TR>
  <TD>TU</TD>
  <TD>36 hpf.</TD>
  <TD>3</TD>
</TR>
<TR>
  <TD>WIK</TD>
  <TD>14 somites</TD>
  <TD>1</TD>
</TR>
</TABLE></P>
<P></P>

<H2>Credits</H2>
<P>
Thanks to the <A href="http://zon.tchlab.org/" TARGET=_blank>Zon Lab</A> for
kindly providing these data as part of the following NIH grants: PO1
DK 055381 - Characterization of the Zebrafish Genome, 
UO1 DK 63328 - SCGAP, and P30 DK49216 - Center Grant Core. Data 
collection and curation was performed by Gerhard Weber, Yi Zhou, 
Anhua Song and Tony DiBiase.</P>

<H2>References</H2>
<P>
Weber GJ, Choe SE, Dooley KA, Paffett-Lugassy NN, Zhou Y, Zon LI. 
<A HREF="http://www.bloodjournal.org/cgi/content/abstract/106/2/521" 
TARGET=_blank>Mutant-specific gene programs in the zebrafish</A>. 
<EM>Blood</EM>. 2005 Jul 15;106(2):521-30.</P>
