<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=315>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/"
    TARGET=_blank>
    <IMG SRC="/images/Danio_rerio.jpg" WIDTH=300 HEIGHT=98 ALT="Zebrafish"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Danio rerio</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of NHGRI
      (<A HREF="http://www.genome.gov/10005141" 
      TARGET=_blank>Press Photos</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The July 2007 zebrafish (<em>Danio rerio</em>) Zv7 assembly was produced by 
The Wellcome Trust Sanger Institute in collaboration with the Max Planck 
Institute for Developmental Biology in Tuebingen, Germany, and the Netherlands
Institute for Developmental Biology (Hubrecht Laboratory), Utrecht, The 
Netherlands. 

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
zebrafish genome.  Note that some position queries (e.g. "huntington") may 
return matches to the mRNA records of other species. In these cases, the 
mRNAs are mapped to their homologs in zebrafish. See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
<TR><TD VALIGN=Top NOWRAP><B>Request:</B><br></TD>
	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><br></TD></TR>
	
<TR><TD VALIGN=Top><br></TD></TR>
	
<TR><TD VALIGN=Top NOWRAP>chr1</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays all of chromosome 1</TD></TR>
<TR><TD VALIGN=Top NOWRAP>chr1:1-200000</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays first two hundred thousand bases of chromosome 1</TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chr1:100000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr 1 that spans 2000 bases, starting with position 100000 </TD>
    </TR>
<TR><TD VALIGN=Top><br></TD></TR>

<TR><TD VALIGN=Top NOWRAP>U30710</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays region containing zebrafish mRNA with GenBank accession number U30710</TD></TR>

<TR><TD VALIGN=Top NOWRAP>AA658622</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays region containing zebrafish EST with GenBank accession AA658622
</TD></TR>
<TR><TD VALIGN=Top NOWRAP>ENSDART00000025573</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays region containing Ensembl gene prediction transcript ENSDART00000025573
</TD></TR>
<TR><TD VALIGN=Top><br></TD></TR>

<TR><TD VALIGN=Top NOWRAP>p53</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs related to the p53 tumor suppressor</TD></TR> 
</TD></TR>
<TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists transcribed pseudogenes but not cDNAs, in GenBank</TD></TR>
<TR><TD VALIGN=Top NOWRAP>homeobox caudal</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes in GenBank</TD></TR>
<TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD></TR>
<TR><TD VALIGN=Top NOWRAP>kruppel zinc finger</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD></TR>
<TR><TD VALIGN=Top NOWRAP>huntington</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs associated with Huntington's disease</TD></TR>

<TR><TD VALIGN=Top><br></TD></TR>

<TR><TD VALIGN=Top NOWRAP>porter</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientists named Porter</TD></TR>
<TR><TD VALIGN=Top NOWRAP>Amsterdam,A.</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by co-author A. Amsterdam</TD></TR>

<TR><TD VALIGN=Top><br></TD></TR>
	
<TR><TD COLSPAN="3" > Use this last format for author queries. Although 
GenBank requires the search format <em>Amsterdam A</em>, internally it uses 
the format <EM>Amsterdam,A.</EM>
<HR>
<H3>Assembly details</H3>
<P>
The Zv7 assembly consists of 1,440,582,308 bp in 5,036 fragments 
(N50 = 1,153,933, n = 277). 
This assembly was constructed using two different strategies. 
One approach used traditional clone mapping and sequencing techniques to 
produce a higher-quality genome sequence. BAC libraries were selected and 
fingerprinted to generate a fingerprint contig (FPC) map (data freeze 11 April
2007). From this map, 
a tiling path was calculated that covered the genome sequence clone-by-clone. 
Clones from this path were selected for high-quality sequencing and then 
pieced together to form the genome sequence. The other strategy involved a whole
genome shotgun (WGS) approach.
The WGS assembly, of lower quality sequence, was used to fill gaps in 
the tiling path. It was generated using 13,756,367 reads from a single
Tuebingen doubled haploid zebrafish. The reads comprised 10,891,216,277 bp 
and were clustered using Phusion, resulting in a coverage of 5.5x.</P>
<P>
Clone sequences and WGS contigs were integrated by considering sequence 
alignments, BAC end placements and zebrafish cDNAs and markers. Improvements 
to the integration algorithm allowed the placement of the WGS contigs that 
contained markers, but could not be linked to the FPC contigs. In cases where 
markers from different chromosomes appeared on the same contig, priority was 
given to the Heat Shock Diploid Cross Panel (HS) and the Boston MGH Cross 
Map (MGH). Some of these discrepancies may be attributed to misassemblies, but 
there may also be inconsistencies between the zebrafish marker panels. </P>
<P>
The integrated assembly contains 1.02 Gb from 7,823 sequenced clones, 
(7,139 finished and 684 unfinished).  
89% of the sequence -- 1,277,075,233 bp including estimated gap sizes and
100 bp gaps between scaffolds -- lies in scaffolds that are placed on 
chromosomes 1-25 (linkage groups 1-25) after integration of the WGS 
assembly with the clone sequences. Integration was 
achieved by considering a combination of sequence alignment and BAC end 
positions, as well as features such as zebrafish cDNAs and markers. 
45,800,611 bp of the sequence are in 166 scaffolds tied to unplaced 
FPC contigs; 117,689,868 bp are in 4,844 NA scaffolds, which are unplaced WGS 
contigs. 
The zebrafish mitochondrial sequence is also
available as the virtual chromosome &quot;chrM&quot;.</P>
<P>
The Sanger Institute notes this assembly release is still preliminary. 
Highly variable regions within the genome posed assembly difficulties, 
most likely because the sequences originated from different haplotypes. 
This also results in assembly dropouts and false duplications. 
While generating this assembly, special attention was paid to such issues and 
more than 200 Mb of duplicated sequence was removed in comparison to the Zv6
assembly (UCSC danRer4). 
For more information about this Zv7 assembly, see the Sanger Institute 
page for the <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/Zv7_assembly_information.shtml"
TARGET=_blank><em>Danio rerio</em> Sequencing Project</A>. </P>
<P>
Downloads of the Zebrafish data and annotations can be obtained from the UCSC
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/danRer5/" TARGET=_blank>FTP 
site</A> or 
<A HREF=http://hgdownload.cse.ucsc.edu/downloads.html#zebrafish>Downloads 
page</A>. This data set has 
<A HREF="http://www.sanger.ac.uk/notices/use-policy.shtml" 
TARGET=_blank>specific conditions for use</A>.
The danRer5 annotation tracks were generated by UCSC and collaborators 
worldwide. 
Special thanks to the Zebrafish Genome Initiative at Children's Hospital in 
Boston, MA, USA for their collaboration on this release.
See the <A HREF="../goldenPath/credits.html#zebrafish_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to this release. </P>
