<H2>Description and Methods </H2>
<P>
This track shows gene annotations from the Vertebrate Genome Annotation (Vega) 
database.</P>
<P>
The following information is excerpted from the 
<A HREF="http://vega.sanger.ac.uk/" TARGET=_BLANK>Vertebrate Genome Annotation</A> 
home page:</P>
<P>
&quot;The Vega database
is designed to be a central repository for high-quality, frequently updated  
manual annotation of different vertebrate finished genome sequence.
Vega attempts to present consistent high-quality curation of the published 
chromosome sequences. Finished genomic sequence is analysed on a 
clone-by-clone basis using
a combination of similarity searches against DNA and protein databases
as well as a series of <em>ab initio</em> gene predictions (GENSCAN, Fgenes).
The annotation is based on supporting evidence only.&quot;</P>
<P>
In this assembly 
<A HREF="http://vega.sanger.ac.uk/Danio_rerio/Info/Index"
TARGET=blank>Vega</A> annotations are shown only on chromosomes 1-25,
and on some unplaced clones &mdash; not on the WGS sequence. 
The Vega annotation consists of BAC and PAC clones that 
were sequenced for the zebrafish clone mapping and 
sequencing <A HREF="https://www.sanger.ac.uk/resources/zebrafish/librarydetails.html"
TARGET=_BLANK>project</A>. This is based on the tiling path information from 
the FPC database.</P>

<H2>Display Conventions and Configuration</H2>
<P>
This track follows the display conventions for
<A HREF="../goldenPath/help/hgTracksHelp.html#GeneDisplay">gene prediction
tracks</A> using the following color scheme to indicate the status of the gene 
annotation:
<UL>
<LI><B>Known</B> (Dark blue): genes that are identical to known zebrafish 
complementary DNA or protein sequences and have an entry in the 
species-specific model organism database, 
<A HREF="http://zfin.org/" TARGET=_blank>ZFIN</A>.
<LI><B>Novel</B> (Dark blue): genes that have an open reading 
frame (ORF) and are identical or homologous to zebrafish cDNAs, zebrafish ESTs,
or proteins in all species.  Novel transcripts are genes that fit the 
criteria of novel genes with the exception that an unambiguous ORF cannot be 
assigned.
<LI><B>Putative</B> (Medium blue): genes whose sequences are identical or
homologous to zebrafish ESTs but do not contain an ORF.
<LI><B>Predicted</B> (Light blue): genes based on <em>ab initio</em> prediction and 
for which at least one exon is supported by biological data (unspliced ESTs, 
protein sequence similarity with mouse or tetraodon genomes, or 
expression data from Rosetta).
<LI><B>Unclassified</B> (Gray).
</UL></P>
<P>
The details pages show the only the Vega gene type and not the transcript type.
A single gene can have more than one transcript which can belong to 
different classes, so the gene as a whole is classified according to the 
transcript with the &quot;highest&quot; level of classification. Transcript 
type (and other details) may be found by clicking on the transcript 
identifier which forms the outside link to the Vega transcript details page.
Further information on the gene and transcript classification may be found
<A HREF="http://vega.sanger.ac.uk/info/about/gene_and_transcript_types.html"
TARGET=_blank>here</A>.
</P>

<H2> Credits </H2>
<P>
Thanks to Tina Eyre, Ian Sealy and Kerstin Howe at the
<A HREF="http://www.sanger.ac.uk/" TARGET=_BLANK>
Wellcome Trust Sanger Institute</A> 
for providing the GTF files and additional annotations for the Vega genes. 
Vega gene annotations are 
generated by manual annotation by the 
<A HREF="https://www.sanger.ac.uk/resources/zebrafish/genomeproject.html" TARGET=_BLANK>
Zebrafish sequence analysis group</A> and 
the <A HREF="https://www.sanger.ac.uk/research/projects/vertebrategenome/havana/" TARGET=_blank>HAVANA</A>
group at the Sanger Institute and <A HREF="http://zfin.org/"
TARGET=_BLANK>ZFIN</A>.
<P>
