<H2>Description</H2>

<P>
This track shows gene predictions using the N-SCAN gene structure prediction
software provided by the Computational Genomics Lab at Washington University 
in St. Louis, MO, USA.
</P>

<H2>Methods</H2>

<H3>N-SCAN PASA-EST</H3>
<P>
N-SCAN combines biological-signal modeling in the target genome sequence along 
with information from a multiple-genome alignment to generate <EM>de novo</EM> 
gene predictions. It extends the TWINSCAN target-informant genome pair to allow for 
an arbitrary number of informant sequences as well as richer models of 
sequence evolution. N-SCAN models the phylogenetic relationships between the 
aligned genome sequences, context-dependent substitution rates, insertions, 
and deletions.
</P>
<P>
For creating predictions on cow, N-SCAN uses mouse (mm9) as the informant.
</P>
<P>
N-SCAN PASA-EST combines EST alignments into N-SCAN. Similar to the conservation
sequence models in TWINSCAN, separate probability models are developed for EST 
alignments to genomic sequence in exons, introns, splice sites and UTRs, 
reflecting the EST alignment patterns in these regions. N-SCAN PASA-EST is more 
accurate than N-SCAN while retaining the ability to discover novel genes to 
which no ESTs align.
</P>
<P>
No manual annotation was performed to generate any of the gene models.
</P>
<H2>Credits</H2>

<P>
Thanks to Michael Brent's <A HREF="http://mblab.wustl.edu/" 
TARGET=_blank>Computational Genomics Group</A> at Washington 
University St. Louis for providing these data.
</P>
<p>
Special thanks for this implementation of N-SCAN to Aaron Tenney in
the Brent lab, and Robert Zimmermann, currently at Max F. Perutz
Laboratories in Vienna, Austria.
</p>


<H2>References</H2>
<p>
Gross SS, Brent MR.
<a href="http://online.liebertpub.com/doi/abs/10.1089/cmb.2006.13.379" target="_blank">
Using multiple alignments to improve gene prediction</a>.
<em>J Comput Biol</em>. 2006 Mar;13(2):379-93.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/16597247" target="_blank">16597247</a>
</p>

<P>
Haas BJ, Delcher AL, Mount SM, Wortman JR, Smith RK Jr, Hannick LI, Maiti R, Ronning CM,
Rusch DB, Town CD <em>et al</em>.
<a href="http://nar.oxfordjournals.org/content/31/19/5654.abstract" target="_blank">
Improving the <em>Arabidopsis</em> genome annotation using maximal transcript
alignment assemblies</a>.
<em>Nucleic Acids Res</em>. 2003 Oct 1;31(19):5654-66.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/14500829" target="_blank">14500829</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC206470/" target="_blank">PMC206470</a>
</p>

<p>
Korf I, Flicek P, Duan D, Brent MR.
<a href="http://bioinformatics.oxfordjournals.org/content/17/suppl_1/S140.abstract"
target="_blank">
Integrating genomic homology into gene structure prediction</a>.
<em>Bioinformatics</em>. 2001;17 Suppl 1:S140-8.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/11473003" target="_blank">11473003</a>
</p>

<p>
van Baren MJ, Brent MR.
<a href="http://genome.cshlp.org/content/16/5/678.abstract" target="_blank">
Iterative gene prediction and pseudogene removal improves genome annotation</a>.
<em>Genome Res</em>. 2006 May;16(5):678-85.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/16651666" target="_blank">16651666</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1457044/" target="_blank">PMC1457044</a>
</p>
