<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
  <meta http-equiv="Content-Type"
 content="text/html; charset=ISO-8859-1">
  <meta name="description" content="ENCODE Project at UCSC">
  <meta name="keywords"
 content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
  <title>ENCODE Project at UCSC</title>
  <link rel="STYLESHEET" href="../style/HGStyle.css">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
</head>
<body style="font-family: arial, Geneva, Helvetica, sans-serif;" alink="#6600ff"
 link="#0000cc" vlink="#330066">
<a name="TOP"></a><!-- Header Images and links -->
<table border="0" cellpadding="0" cellspacing="0" width="100%">
  <tbody>
    <tr>
      <th colspan="2" align="left">
<img src="../images/ENCODE_scaleup_logo.png" alt="ENCODE Project at NHGRI" height="75"> &nbsp;&nbsp;
 <font size="6" face="arial, Geneva, Helvetica, sans-serif"><font color="004D99">
 Encyclopedia of DNA Elements</font>
       </th>
    </tr>
    <tr height="100%" valign="top">
      <td rowspan="1" height="100%" valign="top"><!-- outer table is to align leftbar & content boxes at top of page -->
      <table border="0" cellpadding="0" cellspacing="0" width="100%">
        <tbody>
          <tr>
            <td><img src="../images/DOT.gif" alt="" height="7"></td>
          </tr>
          <tr height="100%">
            <td height="100%"><!--outer table is for black border-->
            <table bgcolor="#004D99" border="0" cellpadding="1" cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table border="0" cellpadding="10" cellspacing="0" width="100%">
                    <tbody>
                      <tr height="100%">
                        <td bgcolor="#0004D99" height="100%" valign="top">
			<font color="#FFCC00"><b><em>Human</em></b></font><br>
                        <img src="../images/DOT.gif" alt="" height="8" width="0"><br>
			     <a class="leftbar" href="/ENCODE/dataSummary.html">Data Summary</a>
                        <hr> <a class="leftbar" href="search.html">Search</a>
                        <hr> <a class="leftbar" href="downloads.html">Downloads</a>
			<hr> <a class="leftbar" href="/cgi-bin/hgTracks?db=hg19">Genome Browser (hg19)</a>
			<hr> <a class="leftbar" href="http://genome-preview.ucsc.edu/cgi-bin/hgTracks?db=hg19">Preview Browser (hg19)</a>
			<hr> <a class="leftbar" href="http://genomewiki.cse.ucsc.edu/index.php/Genome_Browser_Session_Gallery"
			TARGET=_blank>Session Gallery</a>
                        <hr> <a class="leftbar" href="cellTypes.html">Cell Types</a>
                        <hr>
			<img src="../images/DOT.gif" alt="" height="5" width="0"><br>
			<font color="#FFCC00"><b><em>Mouse</em></b></font><br>
			<img src="../images/DOT.gif" alt="" height="8" width="0"><br>
			     <a class="leftbar" href="/ENCODE/dataSummaryMouse.html">Data Summary</a>
                        <hr> <a class="leftbar" href="search.html">Search</a>
                        <hr><a class="leftbar" href="downloadsMouse.html">Downloads</a>
                        <hr><a class="leftbar" href="/cgi-bin/hgTracks?db=mm9">Genome Browser (mm9)</a>
			<hr><a class="leftbar" href="http://genome-preview.ucsc.edu/cgi-bin/hgTracks?db=mm9">Preview Browser (mm9)</a>
			<hr><a class="leftbar" href="cellTypesMouse.html">Cell Types</a>
			<hr>
                        <img src="../images/DOT.gif" alt="" height="5" width="0"><br>
			<font color="FFCC00"><b><em>General</em></b></font><br>
                        <img src="../images/DOT.gif" alt="" height="8"><br>
			     <a class="leftbar" href="otherTerms.html">Registered Variables</a>
                        <hr> <a class="leftbar" href="antibodies.html">Antibodies</a>
                        <hr> <a class="leftbar" href="usageResources.html">Education and Outreach</a>
			<hr> <a class="leftbar" href="releaseLog.html">Release Log</a>
                        <hr> <a class="leftbar" href="terms.html">Data Policy</a>
			<hr> <a class="leftbar" href="dataStandards.html">Data Standards</a>
		    	<hr> <a class="leftbar" href="pubs.html">Publications</a>
                        <hr> <a class="leftbar" href="contributors.html">Contributors</a>
                        <hr> <a class="leftbar" href="pilot.html">Pilot Project</a>
                        <hr> <a class="leftbar" href="http://www.cbse.ucsc.edu/jobs/" target="_blank">Jobs</a>
                        <hr> <a class="leftbar" href="contacts.html">Contact Us</a>
                        <hr> <br>
                        <img src="../images/DOT.gif" alt="" height="970" width="1"> </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            </td>
          </tr>
        </tbody>
      </table>
      </td>
<!-- Content Tables --> <td valign="top">
      <table border="0" cellpadding="5">
        <tbody>
          <tr>
            <td><!--outer table is for border purposes-->
            <table bgcolor="#888888" border="0" cellpadding="1"
 cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table bgcolor="#FFFFFF" border="0" cellpadding="0"
 cellspacing="0" width="100%">
                    <tbody>
                      <tr>
                        <td>
                        <table background="../images/hr.gif"
 bgcolor="#FFCC00" border="0" width="100%">
                          <tbody>
                            <tr>
                              <td><a name="About"></a><font size="4"><b>&nbsp;
About ENCODE Data</b></font> </td>
                            </tr>
                          </tbody>
                        </table>
                        <table bgcolor="#FFFFFF" border="0"
 cellpadding="0" width="100%">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="10"><br>
                              </td>
                              <td>
<p>The <a href="http://www.genome.gov/10005107" target="_blank">Encyclopedia of DNA Elements</a>
(ENCODE) Consortium is an international collaboration
of research groups funded by the National Human Genome Research
Institute (<a href="http://www.genome.gov/" target="_blank">NHGRI</a>).
The goal of ENCODE is to build a comprehensive parts list of
functional elements in the human genome, including elements that
act at the protein and RNA levels,
and regulatory elements that control cells and
circumstances in which a gene is active.
</p>
<!--#if expr='"$SERVER_NAME" = /hgwdev.*.cse.ucsc.edu/ || "$SERVER_NAME" = "genome-test.cse.ucsc.edu"'-->
<p>
<b>WARNING: This is a test site.</b>
Data and tools here are under 
construction, have not been quality reviewed, and are subject to change 
at any time.  
For 
high-quality reviewed annotations on our production server, visit
<a HREF="http://encodeproject.org">http://encodeproject.org</a>.</p>
<!--#endif -->

<!--#if expr='"$SERVER_NAME" = /genome-preview.*.ucsc.edu/' -->
<p>
<b>WARNING: This is our preview site.</b>
This website is a weekly mirror of our internal 
development server for public access.  
Data and tools here are under 
construction, have not been quality reviewed, and are subject to change 
at any time.  We provide this site for early access, with the warning that it is
less available and stable than our public site. For 
high-quality reviewed annotations on our production server, visit
<a HREF="http://encodeproject.org">http://encodeproject.org</a>.</p>
<!--#endif -->

<table><tr>
<td><a href="/ENCODE/aboutScaleup.html" target="_blank"><img src="../images/EncodeNatureGraphic.png" alt="" height="200">
<em><font size=-2>Click to enlarge</em></font></td>

<td>
ENCODE data are now available for the entire human genome.
<b><em>All ENCODE data are free and available for immediate use via </em></b>:
<ul class="simple">
<li><a href="/ENCODE/search.html" target="_blank">Search</a> for displayable tracks and downloadable files
</li>
<li><a href="/ENCODE/downloads.html" target="_blank">Download</a> of data files
</li>
<li><a href="http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&hgS_otherUserName=Kate&hgS_otherUserSessionName=encodePortalSession"
target="_blank">Visualization</a> in the UCSC Genome Browser (ENCODE data marked with
the <img src="../images/nhgri.gif" alt="" height="16" width="16"> NHGRI logo)
</li>
<li><a href="/cgi-bin/hgTables?db=hg18&hgta_group=regulation&hgta_track=wgEncodeHudsonalphaChipSeq"
target="_blank">Data mining</a> with the UCSC Table Browser and other
<a href="http://genome.ucsc.edu"> UCSC Genome Bioinformatics tools</a>
</li>
</ul>
</td>

</tr></table>
</p>
<p>
To search for ENCODE data related to your area of interest and set up a browser view, use the UCSC
<a href="/cgi-bin/hgTracks?hgt_tSearch=1" target="_blank">Track Search tool</a> (<em>Advanced</em> features).
The <a href="dataSummary.html" target="_blank">Data Summary</a> shows a comprehensive listing of
ENCODE data that is released or in preparation.
Early access to pre-release ENCODE data is provided at
<a href="http://genome-preview.ucsc.edu" target="_blank">http://genome-preview.ucsc.edu</a>.
If you would like to receive notifications of ENCODE data releases and related news by email,
subscribe to the
<a href="https://lists.soe.ucsc.edu/mailman/listinfo/encode-announce" target="_blank">encode-announce mailing list</a>.
For more information about how to access this data, see the free online
<a target=="_blank" href="http://www.openhelix.com/ENCODE/" target="_blank">OpenHelix ENCODE tutorial</a>.
<p>
To complement the human ENCODE data, Mouse ENCODE experiments are currently underway. Early
access to this data is available on the Mouse mm9/NCBI37 browser at the UCSC preview site.
The <a href="/encode/dataSummaryMouse.html" target="_blank">Mouse ENCODE
Data Summary</a> lists experiments that are planned or in progress.
</p>
<p>
All ENCODE data is freely available for download and analysis.
However, before publishing research that uses ENCODE data,
please read the <a href="terms.html" target="_blank">ENCODE Data Release Policy</a>,
which places some restrictions on publication use of data for nine
months following data release. &nbsp;&nbsp;
<a href="/ENCODE/aboutScaleup.html">Read more</a> about ENCODE data at UCSC.</p>

                              </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        <br>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
<!--outer table is for border purposes-->
            <table bgcolor="#888888" border="0" cellpadding="1"
 cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table bgcolor="#FFFFFF" border="0" cellpadding="0"
 cellspacing="0" width="100%">
                    <tbody>
                      <tr>
                        <td>
                        <table background="../images/hr.gif"
 bgcolor="#FFCC00" border="0" width="100%">
                          <tbody>
                            <tr>
                              <td><!--cheap trick to get background on row to be continuous-->
                              <table border="0" cellpadding="0" cellspacing="0" width="100%">
                                <tbody>
                                  <tr>
                                    <td align="left"><font size="4"><a name="TOC"></a><b>&nbsp; News</b></font> </td>
                                    <td align="right"><a href="newsarch.html"><img src="../images/news.jpg" alt="" align="right" border="0"></a> </td>
                                    <td align="right">&nbsp;</td>
                                  </tr>
                                </tbody>
                              </table>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        <table bgcolor="#FFFFFF" border="0" cellpadding="0" width="100%">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="10"><br>
                              </td>
                              <td style="font-family: arial, Geneva, Helvetica, sans-serif;">


<!-- start news item -->
<p>
<font size="3"><b> 8 July 2011 - Mouse ENCODE data releases: DNaseI hypersensitivity (UW DNaseI HS) and histone modifications (LICR Histone)
</b></font>
<p>
Two tracks of ENCODE data were released on the mm9 genome browser, from the UCSD/Ludwig Institute for Cancer Research and the University of Washington Mouse ENCODE groups.
</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrHistone">Histone Modifications by ChIP-seq from ENCODE/LICR</b></a>:
This track shows a comprehensive survey of cis-regulatory elements in the mouse genome by using ChIP-seq to identify transcription factor binding sites and chromatin modification profiles in many mouse tissues and primary cells, including bone marrow, cerebellum, cortex, heart, kidney, liver, lung, spleen, mouse embryonic fibroblast cells (MEFs) and embryonic stem (ES) cells.
</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=mm9&g=wgEncodeUwDnase">DNaseI Hypersensitivity by Digital DNaseI from ENCODE/University of Washington</b></a>:
This track shows DNaseI sensitivity measured genome-wide in mouse tissues and cell lines using the Digital DNaseI methodology and DNaseI hypersensitive sites.
</p>
<hr>
<!-- end news item -->
<!-- start news item -->
<p>
<font size="3"><b> 1 July 2011 - ENCODE data releases: Broad ChromHMM, Open Chrom Synth, UChicago TFBS, Duke Affy Exon
</b></font>
<p>
Four tracks of ENCODE production data and analysis were released in June, from the 
Broad Institute (Kellis lab), OpenChromatin (Duke, UNC, UT-A) and University of Chicago
(White Lab) ENCODE groups.  This is the first data release from the University of Chicago ENCODE group, 
which joined the Consortium as part of the NIH ARRA stimulus grants.
</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeBroadHmm">Chromatin State Segmentation by HMM from ENCODE/Broad</b></a>:
This track, and the companion hg18 track, display chromatin state segmentation of the human genome into
fifteen states grouped to predict functional elements.
</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromSynth">DNaseI/FAIRE/ChIP Synthesis from ENCODE/OpenChrom(Duke/UNC/UTA)</b></a>:
This track displays a synthesis of open chromatin regions and binding of selected regulatory factors, 
based on three complementary methodologies.
</p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUchicagoTfbs">Transcription Factor Binding Sites by Epitope-Tag ChIP-seq from ENCODE/University of Chicago</b></a>:
This track maps human transcription factor binding sites genome-wide using expressed transcription factors as GFP tagged fusion proteins after BAC recombineering.
</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeDukeAffyExonArray">Affymetrix Exon Array from ENCODE/Duke</b></a>:
This track displays human tissue microarray data using Affymetrix Human Exon 1.0 ST expression arrays, 
including annotations at the gene level.
</p>
<hr>
<!-- end news item -->

<!-- start news item -->
<p>
<font size="3"><b> 24 June 2011 - ENCODE RNA-seq data standards now available
</b></font>
<p>
The ENCODE Consortium has finalized <em><b>'Standards, Guidelines and Best Practices for RNA-Seq V1.0'</b></em>, as part of the Consortium's continuing
effort to generate data standards.   The document is available at the ENCODE portal via  the
<a target=_blank href="http://encodeproject.org/ENCODE/dataStandards.html"><b>Data Standards</b></a> link.
</p>
<p>
RNA-Seq is a directed experimental approach aimed at characterizing transcription in biological samples. This document presents a set of guidelines and standards focused on best practices for creating 'reference quality' transcriptome measurements. 
sets.</p>
<hr>
<!-- end news item -->
<!-- start news item -->
<p>
<font size="3"><b> 1 June 2011 - ENCODE data releases in April and May
</b></font>
<p>
Five tracks of ENCODE production and analysis data were released in April and May on the GRCh37/hg19 human assembly from
the Caltech, Broad Institute, HudsonAlpha Institute for Biotechnology, Duke University (Open Chromatin), SUNY Albany,
University of Washington, Boston University, Stanford/Yale/Davis/Harvard and UCSC ENCODE groups
</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeReg">Integrated Regulation from ENCODE</b></a>:
This collection of tracks displays integrated signal and clustering annotations from multiple cell lines, using ENCODE
primary data from RNA-seq, ChIP-seq, and DNase-seq assays.  This track is a companion to the hg18 ENCODE Regulation track.
</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromDnase">
 Open Chromatin by DNaseI HS from ENCODE/OpenChrom(Duke University)</b></a>:
This track displays DNaseI hypersensitivity evidence as part of the four Open Chromatin track set.
</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSunyAlbanyGeneSt">RNA Binding Protein Associated RNA by RIP-chip GeneST from ENCODE/SUNY Albany</b></a>:
This track displays transcriptional fragments associated with RNA 
binding proteins in different cell lines using RIP-Chip (Ribonomic) 
profiling on Affymetrix GeneChip® Human Gene 1.0 ST Arrays.
</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase">DNaseI Hypersensitivity by Digital DNaseI form ENCODE/University of Washington</b></a>:
This track shows DNaseI sensitivity measured genome-wide in different 
cell lines using the Digital DNaseI methodology.
</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeBuOrchid">OrChID Predicted DNA Cleavage Sites from ENCODE/Boston Univ (Tullius Lab)</b></a>:
This track display predicted hydroxyl radical cleavage intensity on naked DNA for each nucleotide in the genome. 
</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhHistone">Histone Modifications by ChIP-seq from ENCODE/Stanford/Yale/Davis/Harvard</b></a>:
This track displays maps of histone modifications genome-wide using ChIP-seq in different cell lines. 
</p>
<hr>
<!-- end news item -->

<!-- start archived item section -->
<p>
<font size="3"><b> 22 April 2011 - ENCODE data releases: UTA TFBS, UW CTCF, UNC FAIRE, RIKEN CAGE Loc, UW Affy Exon, UW DNaseI DGF & Duke DNaseI HS</b></font>
<p>
Seven tracks of ENCODE data on the GRCh37/hg19 human assembly were released in March and April from
from the University of Texas at Austin (Open Chromatin), University of Washington, University of North Carolina (Open Chromatin), RIKEN, and Duke University (Open Chromatin) ENCODE groups.
<a href="newsarch.html#042211">Read more</a>.</p>
<!-- end archived news item -->
<br>
<!-- end archived news item -->
<!-- end archived item section--> </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
<!--outer table is for border purposes-->
            <table bgcolor="#888888" border="0" cellpadding="1"
 cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table bgcolor="#FFFFFF" border="0" cellpadding="0"
 cellspacing="0" width="100%">
                    <tbody>
                      <tr>
                        <td>
                        <table background="../images/hr.gif"
 bgcolor="#FFCC00" border="0" width="100%">
                          <tbody>
                            <tr>
                              <td><a name="Conditions"></a><font
 size="4"><b>&nbsp; Conditions of Use</b></font> </td>
                            </tr>
                          </tbody>
                        </table>
                        <table bgcolor="#FFFFFF" border="0"
 cellpadding="0" width="100%">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="10"><br>
                              </td>
                              <td>
                              <p>The sequence and annotation data
displayed in the Genome Browser are freely available for academic,
nonprofit, and personal use with the following conditions: </p>
                              <ul>
                                <li>ENCODE data are covered by the <a
 href="terms.html">ENCODE Consortium Data Release Policy</a>. </li>
                                <li>The general <a
 href="http://genome.ucsc.edu/#Conditions">Conditions of Use</a> for
the UCSC Genome Browser apply. </li>
                              </ul>
                              </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
            </td>
          </tr>
        </tbody>
      </table>
      </td>
    </tr>
  </tbody>
</table>
</body>
</html>
