<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="ENCODE Project at UCSC">
    <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
    <TITLE>ENCODE Project at UCSC</TITLE>
    <LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
  <A NAME="TOP"></A>

    <!-- Header Images and links -->
    <TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
      <TR>
        <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
          <A HREF="http://www.genome.gov/10005107" TARGET=_blank><IMG SRC="/images/encodelogo.gif" height=50 ALT="ENCODE Project at NHGRI"></A>
          &nbsp;&nbsp;
          <IMG SRC="/images/encode.jpg" ALT="ENCODE Project at UCSC">
        </TH>
      </TR>

      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.home.topbar.html"--></TD></TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
        <TR><TD><IMG SRC="/images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="encode.hg17.html">
		   Regions (hg17)</A><HR>
	        <A CLASS="leftbar" HREF="encode.hg16.html">
		   Regions (hg16)</A><HR>
                <A CLASS="leftbar" HREF="trackStatus.html">
                    Data Status</a> <HR>
	        <A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/goldenPath/encode/index.html">
		   Downloads</A><HR>
	        <A CLASS="leftbar" HREF="submission.html">
		   Submission</A><HR>
	        <A CLASS="leftbar" HREF="/cgi-bin/hgLiftOver">
		   Tools</A><HR>
                <A CLASS="leftbar" HREF="releaseLog.html">
                    Release Log</a> <HR>
<!-- Commented out until functional
                <A CLASS="leftbar" HREF="customTracks/index.html">
                    Custom Tracks</a> <HR>
-->
		<A class="leftbar" HREF="contributors.html">
		   Contributors</a><HR>
	        <A CLASS="leftbar" HREF="terms.html">
		   Terms of Use</a><HR>
<!--                <A CLASS="leftbar" HREF="releaseLog.html">  Release Log</A><HR>-->
		<BR><BR>&nbsp<HR>
                <A CLASS="leftbar" HREF="/cgi-bin/hgGateway">
		   Genome Browser</A><HR>
 	        <BR>
	        <IMG SRC="/images/DOT.gif" ALT="" HEIGHT=970 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-------------------->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	

	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; 
		    About the ENCODE Project</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                   <TR><TH HEIGHT=3></TH></TR>
                   <TR><TD WIDTH=10></TD><TD>
	             <P> This site contains information related to the 
		     <A HREF="http://www.genome.gov/10005107" 
		     TARGET=_blank>ENCODE project</A> at 
		     <A HREF="http://www.genome.gov/" 
		     TARGET=_blank>NHGRI</A>. The UCSC Genome Bioinformatics 
		     Group manages the official repository of 
		     sequence-related data for the ENCODE consortium and 
		     supports the coordination of data submission, storage, 
		     retrieval, and visualization.  
                     A summary of the status of datasets submitted to UCSC 
                     by ENCODE contributors is available on the 
                     <A TARGET=_BLANK HREF="trackStatus.html">
                     ENCODE data status page</A>.
                     <P>
                     UCSC also has a special 
		     interest in comparative genomics, and we provide 
		     resources for the ENCODE multiple sequence alignment 
		     interest group. <A HREF="http://www.ensembl.org" 
		     TARGET=_blank> Ensembl</A> also provides an 
		     <A HREF="http://www.ensembl.org/Homo_sapiens/encode.html"
		     TARGET=_blank> ENCODE resource page</A>.</P>
		     <P>
		     We'd like to thank NHGRI and the contributors of 
		     annotations and analyses to this project. The team at 
		     UCSC that develops and maintains this ENCODE site is 
		     made up of <A HREF="http://www.soe.ucsc.edu/~daryl" 
		     TARGET=_blank>Daryl Thomas</A>, 
		     <A HREF="mailto:&#107;&#97;t&#101;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u">Kate Rosenbloom</A>, 
		     <A HREF="http://www.soe.ucsc.edu/~kent/" 
		     TARGET=_blank>Jim Kent</A>, and the 
		     <A HREF="../staff.html">UCSC Genome Bioinformatics 
		     staff</A>. <A HREF="about.html">Read more.</A></P>
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%" BORDER=0>
                  <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                      <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT>
 		    	<TD ALIGN="right"><A HREF="newsarch.html"><IMG SRC="/images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>

		      </TD><TD ALIGN="right">&nbsp;</TD></TR>
		    </TABLE>
	          </TD></TR>
		</TABLE>

		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
    		  <TR><TH HEIGHT=3></TH></TR>
    		  <TR><TD WIDTH=10></TD><TD>

		    <P>
<!-- start news item -->
		    <FONT FACE="courier" SIZE=3><B>9 Jan. 2007 - Winter 2006 ENCODE News</B></FONT><BR>
		    <P>
		    To improve communication, we have posted instructions for 
		    our ENCODE ftp site on the ENCODE Wiki and have set up an 
		    email alias for notifying UCSC about your ENCODE data 
		    submissions: <A HREF="mailto:encode@soe.ucsc.edu">encode@soe.ucsc.edu</A>.
		    The current UCSC recipients are Kate Rosenbloom, Daryl 
		    Thomas, Ting Wang, and Rachel Harte.</P>
		    <P>
		    During November and December, four new/improved data tracks 
		    were released:
		    <UL>
		    <LI>
		    <B>Chip-PET from the Genome Institute of Singapore</B> -  
		    Genome-wide data for c-Myc in P493 B cells was added as a 
		    new subtrack (cMyc P493, encodeGisChipPetMycP493) to the 
		    existing GIS ChIP-PET track.
		    <LI>
		    <B>DNaseI Hs from Duke University</B> - 
		    Existing NHGRI data and new data from the Crawford lab at 
		    Duke University (raw and p-value data for the HepG2 cell 
		    line) were merged into the Duke/NHGRI DNase track. The 
		    newer data is based on DNase-chip technology.
		    <LI>
		    <B>STAGE tags from University of Texas</B> - 
		    Raw tags data for STAT1 in HeLa cells were added
		    as a new subtrack (UT STAT1 HeLa Tags, 
		    encodeUtexStageStat1HelaTags)
		    to the existing UT-Austin Stage track.
		    <LI>
		    <B>DNaseI Hs from  University of Washington</B> - 
		    The existing three UW/Regulome DNaseI Sens tracks were 
		    replaced with a single new track (UW/Reg QCP DNaseI Sens) 
		    based on quantitative chromatin profiling (QCP) methods in 
		    16 cell types.
		    </UL>
		    <P>
		    Thanks to the UCSC staff who worked on these tracks:
		    Rachel Harte, Kate Rosenbloom, and Hiram Clawson 
		    (development), Ann Zweig, Brooke Rhead, and Kayla Smith 
		    (quality review).
		    <P>
<!-- end news item --> 
<!-- start archived item section-->
		    <HR>
		    <FONT FACE="courier" SIZE=3><B>7 Oct. 2006 - Comparative Genomics Data Release: </B></FONT>
		    Twelve tracks of data produced by the ENCODE Multi-Species
		    Sequence Analysis group have been released to the UCSC 
		    public server. 
		    <A HREF="newsarch.html#100706">Read more</A>.</P>
		    <P>
		    <FONT FACE="courier" SIZE=3><B>31 Aug. 2006 - Summer 2006
		    Data Releases: </B></FONT>
		    Since mid-June, we have released new ENCODE data from three
		    labs (Sanger Institute, Uppsala University, and University
		    of North Carolina) and have a track in
		    progress for newly submitted data from a fourth lab 
		    (NHGRI/Duke University).
		    <A HREF="newsarch.html#083106">Read more</A>.</P>
<!-- end archived item section--> 

    	          </TD><TD WIDTH=15></TD></TR>
	        </TABLE>
	      <BR></TD></TR>
	    </TABLE>
          </TD></TR>
        </TABLE>
        <BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH="100%">
		  <TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	          </TD></TR>
		</TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome 
		    Browser are freely available for academic, nonprofit, and 
		    personal use with the following conditions:
		    <UL>
		      <LI>
		      The general 
		      <A HREF="http://genome.ucsc.edu/#Conditions">Conditions of Use</A> 
		      for the UCSC Genome Browser apply.
		      <LI>
		      The ENCODE-specific conditions of use are
		      still being developed and will be displayed here
		      in the future.  Until then, <A
		      HREF="mailto:genome@soe.ucsc.edu">please contact
		      us</A> for the conditions on the use of this data.
		    </UL>
	          </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      <BR></TD></TR>
	    </TABLE>
          </TD></TR>
	</TABLE>
      <BR></TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>

