<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
 	<META name="description" content="ENCODE Project at UCSC - News Archives">
        <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
        <TITLE>ENCODE Project at UCSC: News Archives</TITLE>
        <LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<!-- Header Images and links -->
<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
  <TR>
    <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
      <A HREF="http://www.genome.gov/10005107" TARGET=_BLANK><IMG SRC="../images/ENCODE_scaleup_logo.png" height=50 ALT="ENCODE Project at NHGRI"></A>
        &nbsp;&nbsp;
          <A HREF="/encode/"><IMG SRC="../images/encodeDcc.jpg" ALT="ENCODE Data Coordination Center at UCSC"></A>
    </TH>
  </TR>

  <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.topbar.html"--></TD></TR>

<!--Content Tables------------------------------------------------------->
  <TR><TD COLSPAN=2 CELLPADDING=10>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1>
      <TR><TD>
      	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="../images/hr.gif" WIDTH="100%">
	  <TR><TD>
	    <FONT SIZE="4"><B> &nbsp;News Archives </B></FONT>
	  </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0>
	  <TH HEIGHT=10></TH>
	  <TR><TD WIDTH=10></TD><TD>
	    <UL>
	      <LI><A CLASS="toc" HREF="#2011">2011 News</A></LI>
	      <LI><A CLASS="toc" HREF="#2010">2010 News</A></LI>
	      <LI><A CLASS="toc" HREF="#2009">2009 News</A></LI>
	      <LI><A CLASS="toc" HREF="#2008">2008 News</A></LI>
	      <LI><A CLASS="toc" HREF="#2007">2007 News</A></LI>
	      <LI><A CLASS="toc" HREF="#2006">2006 News</A></LI>
	      <LI><A CLASS="toc" HREF="#2005">2005 News</A></LI>
	      <LI><A CLASS="toc" HREF="#2004">2004 News</A></LI>
	    </UL>
	    <HR>

<!-- Begin 2010 new items -->
           <a name="2011"></a>
<!-- start news item -->
<a name="042211"></a>
<p>
<font size="3"><b> 22 April 2011 - ENCODE data releases: UTA TFBS, UW CTCF, UNC FAIRE, RIKEN CAGE
Loc, UW Affy Exon, UW DNaseI DGF & Duke DNaseI HS</b></font>
<p>
Seven tracks of ENCODE data on the GRCh37/hg19 human assembly were released in March and April from
the University of Texas at Austin (Open Chromatin), University of Washington, University of North Carolina (Open 
Chromatin), RIKEN, and Duke University (Open Chromatin) ENCODE groups (6 of the tracks are new to hg19, and one is a Release 2 on hg19):
</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromChip"><b>UTA TFBS</b></a>:
This track displays chromatin immunoprecipitation (ChIP-seq) evidence as part of the four Open
Chromatin track sets.</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwTfbs"><b>UW CTCF</b></a>:
This track displays maps of genome-wide binding of the CTCF transcription factor in different cell
lines using ChIP-seq high-throughput sequencing.
</p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromFaire"><b>UNC FAIRE</b></a>:
This track displays Formaldehyde-Assisted Isolation of Regulatory Elements (FAIRE) evidence as part
of the four Open Chromatin track sets.</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRikenCage"><b>RIKEN CAGE Loc</b></a>:
This track shows 5' cap analysis gene expression (CAGE) tags and clusters in RNA extracts from
different sub-cellular localizations in multiple cell lines.</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwAffyExonArray"><b>UW Affy Exon</b></a>:
This track displays human tissue microarray data using Affymetrix Human Exon 1.0 GeneChip.</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDgf"<b>UW DNaseI DGF</b></a> (Release 2):
This track contains deep sequencing DNase data that will be used to identify sites where regulatory
factors bind to the genome (footprints).</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromDnase"><b>Duke DNaseI HS</b></a>:
This track displays DNaseI hypersensitivity (HS) evidence as part of the four Open Chromatin track
sets.</p>
<hr>
<!-- end news item -->
<!-- start news item -->
<a name="021411"></a>
<p>
<font size=4 color="#006666">14 February 2011 - ENCODE data releases: Caltech RNA-seq, Broad Histone, UW Histone & SUNY RIP Tiling</b></font>
<p>
Four tracks of ENCODE data on the GRCh37/hg19 human assembly
from the Caltech, Broad/MGH, SUNY Albany and University of Washington ENCODE groups were released:
</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCaltechRnaSeq"><b>Caltech RNA-seq</b></a>:
This track shows transcriptome measurements performed on polyA+ RNA using both stranded and unstranded protocols.  </p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeBroadHistone"><b>Broad Histone</b></a>,
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwHistone"><b>UW Histone</b></a>:
These tracks display maps of histone modifications reflective of chromatin state changes identified by ChIP-seq.
</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSunyAlbanyTiling"><b>SUNY RIP Tiling</b></a>:
This track displays transcriptional fragments associated with RNA binding proteins in different cell lines,
using RIP-Chip (Ribonomic) profiling on Affymetrix GeneChip ENCODE 2.0R Tiling Arrays.</p>

<hr>
<!-- start news item -->
<a name="020911"></a>
<p>
<font size=4 color="#006666">9 February 2011 - First Mouse ENCODE data release: Transcription Factor Binding Sites by ChIP-seq from Stanford/Yale </b></font>
<p>
The first Mouse ENCODE data is now available on the mm9 (NCBI37) genome assembly.  The
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhTfbs">Stan/Yale TFBS</b></a> track
in the 'Regulation' track group shows probable binding sites of the following transcription
factors: c-MYB (H-141), CTCF (C-20), Max, NELFE, p300 (N-15), Rad21, and
USF2, in the MEL leukemia (K562 analog) cell line as determined by ChIP-seq.
Thanks to all who had a hand in generating this data and to the UCSC
wrangler, Venkat Malladi, and Q/A staff who made this data release
possible.
<hr>
<!-- end news item -->

<!-- Begin 2010 new items -->
           <a name="2010"></a>
<!-- start news item -->
<a name="121610"></a>
<p>
<font size=4 color="#006666">16 December 2010 - Release of DNA Methylation and DNaseI Sensitivity data</b></font>
<p>
Three tracks of ENCODE DNA Methylation and DNaseI Sensitivity data have been released on the GRCh37/hg19 human
assembly. All three tracks are in the browser 'Regulation' track group; one of the tracks is in the
new <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeDnaMethylSuper">
ENC DNA Methyl</a> super-track and the other two are in the new
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeDNAseSuper">ENC DNase/FAIRE</a>
super-track (super-tracks provide additional documentation and organization by data type). The three
newly released tracks are: 
</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibMethylRrbs"><b>HAIB Methyl
RRBS</b></a>: This track reports the percentage of DNA molecules that exhibit cytosine methylation
at specific CpG dinucleotides. In general, DNA methylation within a gene's promoter is associated
with gene silencing, and DNA methylation within the exons and introns of a gene is associated with
gene expression.</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDgf"><b>UW DNaseI DGF</b></a>:
This track contains deep sequencing DNase data that will be used to identify sites where regulatory
factors bind to the genome (footprints).</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase"><b>UW DNaseI HS</b></a>:
This track shows DNaseI sensitivity measured genome-wide in different cell lines using the Digital
DNaseI methodology and DNaseI hypersensitive sites.</p>
<p>
Some of the data that comprise these tracks were originally released on hg18 and have been remapped
to hg19; in such cases, subtracks have 'origAssembly hg18' as part of their metadata.</p>
<hr>
<!-- end news item -->

<!-- start news item -->
<a name="111610"></a>
<p>
<font size=4 color="#006666">16 November 2010 - Release of the first ENCODE RNA-seq data on hg19</b></font>
<p>
We are pleased to announce the release of the first ENCODE RNA-seq data on the GRCh37/hg19 human
browser. Two tracks have just been released; these are organized in the new
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?g=wgEncodeRnaSeqSuper">ENC RNA-seq</a> super-track
within the browser 'Expression' track group. The super-track provides additional documentation and
organization by data type.  The two tracks released are:
</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlShortRnaSeq"><b>CSHL Sm
RNA-seq</b></a>: This track depicts NextGen sequencing information for RNAs between the sizes of
20-200 nt isolated from RNA samples from tissues or subcellular compartments from ENCODE cell lines.
</p>
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisRnaSeq"><b>GIS RNA-seq</b></a>:
This track shows high throughput sequencing of RNA samples from tissues or subcellular compartments
from cell lines included in the ENCODE Transcriptome subproject.</p>
<p>
All of the data that comprise these tracks were originally released on hg18 and have been remapped
to hg19.</p>
<hr>
<!-- end news item -->

<!-- start news item -->
<a name="111510"></a>
<font size=4 color="#006666">15 Nov 2010 - New ENCODE Tutorial at OpenHelix</font><br>
<p>
		OpenHelix, together with the UCSC Genome Bioinformatics group, anounce a new online
		tutorial suite to teach users how to access the ENCODE data in the UCSC Genome
		Browser. This tutorial introduces the types of data available under ENCODE, and
		presents methods to access the data via the Genome Browser, Table Browser, and
		downloads. This tutorial suite is freely available at
		<a href="http://www.openhelix.com/encode">OpenHelix</a>
</p><hr>
<!-- end news item -->

<!-- start news item -->
<a name="091610"></a>
<font size=4 color="#006666">16 September 2010 - First Production ENCODE Data on hg19 has been Released</font><br>
<p>
		We are pleased to announce the release of the first sets of production ENCODE data
		on hg19:
		</p>
		<p>
		<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisDnaPet"
		><b>GIS DNA PET</b></a>: This track shows the starts and ends of DNA fragments from
		different cell lines determined by paired-end ditag (PET) sequencing using different
		DNA fragment sizes for analysis of genome structural variation. The data in this
		track uses the new BAM data format. For more information about SAM/BAM, click
		<a href="http://samtools.sourceforge.net/SAM1.pdf">here</a>. All of the subtracks
		that comprise this track were originally released on hg18 and have been remapped to
		hg19.
		</p>
		<p>
		<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencode"
		><b>Gencode Genes</b></a>:
		This track (version 4, May 2010) shows high-quality manual annotations merged with
		evidence-based automated annotations across the entire human genome generated by the
		GENCODE project. Previous versions of this data were released on hg18, but this
		newest version is available solely on hg19.
</p><hr>
<!-- end news item -->
<!-- start news item -->
<a name="082010"></a>
<font size=4 color="#006666">20 August 2010 - New ENCODE Integrated Regulation Super-track Released</font><br>
 <P>
                    We are pleased to announce the release of the ENCODE Integrated
                    Regulation super-track, a collection of regulatory tracks containing
                    state-of-the-art information about the mechanisms that turn genes on and
                    off at the transcription level. Individual tracks within the set show
                    enrichment of histone modifications suggestive of enhancer and promoter
                    activity, DNAse clusters indicating open chromatin, regions of
                    transcription factor binding, and transcription levels. When viewed in
                    combination, the complementary nature of the data within these tracks
                    has the potential to greatly facilitate our understanding of regulatory
                    DNA.
                    </P>
                    <P>
                    The data comprising these tracks were generated from hundreds of
                    experiments on multiple cell lines conducted by labs participating in
                    the ENCODE project, and were submitted to
                    the UCSC ENCODE Data Coordination Center for display on the Genome
                    Browser.
                    </P>
                    <P>
                    Faced with the problem of how to display such a large amount of data
                    in a manner facilitating analysis, UCSC has developed new visualization
                    methods that cluster and overlay the data, and then display the
                    resulting tracks on a single screen. Each of the cell lines in a track
                    is associated with a particular color. Light, saturated colors are used
                    to produce the best transparent overlay.
                    </P>
                    <P>
                    <A HREF="http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&hgS_otherUserName=Example1&hgS_otherUserSessionName=hg18EncReg"><img
		    style="border: 2px solid black; margin-left:auto; margin-right:auto; display: block;" src="/images/encodeRegShot.gif"
                    alt="ENCODE Regulatory track screenshot" /> </A>
                    </P>
                    <P>
                    Currently, the ENCODE Regulation data are available only on the
                    March 2006 (NCBI Build 36, UCSC version hg18) assembly of the human
                    genome.
                    </P>
                    <P>
                    For a detailed description of the datasets contained in this super-track
                    and a discussion of how the tracks can be used synergistically to
                    examine regions of regulatory functionality within the genome, see the
                    <A HREF="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&c=chr1&g=wgEncodeReg">track description</A> page.
                    </P><HR>
<!-- end news item -->
<!-- start news item -->
<a name="080610"></a>
<font size=4 color="#006666">6 August 2010 - June and July ENCODE news</font><br>
                              <br>
                                <b>Initial Release of the
                                <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaRnaSeq">HudsonAlpha RNA-seq</a>
				track:</b> This track shows short
                                tag sequencing of cDNA obtained from biological replicate samples
                                (different culture plates) of the ENCODE cell lines. The sequences
                                were aligned to the human genome (hg18) and UCSC known-gene splice
                                junctions.<br>
                                <br>
                                <b>Release 2 of the
                                <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeCaltechRnaSeq">Caltech RNA-seq</a>
                                track: </b></font> This track shows alignments, signal density, and
                                splice sites based on 75 bp paired reads and 32 bp strand-specific
                                single reads of polyA+ RNA aligned to the human genome (hg18) and
                                UCSC known-gene splice junctions. Also included with the track as
                                downloadable files are RPKM expression level measurements at the
                                gene-level and exon-level, and candidate novel exons.  Release 2
                                of this track adds five new cell types: H1-hESC, HeLa-S3, HepG2,
                                HUVEC, and NHEK.<br>
                                <br>
                                <b>Initial Release of the
                                <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeGisPet">GIS PET Loc</a>
                                track: </b></font> This track shows starts and ends of full length
                                mRNA transcripts determined by PET sequencing of polyA+ and total
                                RNA from 6 subcellular compartments and whole cell, in 8 cell lines.<br>
                                <br>
                                <b>Release 2 of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaChipSeq">HAIB TFBS</a>
                                track: </b></font> This track shows signal density and binding sites
                                of selected transcription factors in a variety of cell types.
                                Release 2 of this track adds 73 new experiments covering 13 new cell
                                lines and 27 antibodies. Additionally, DEX and EtOH treatments have
                                been included in the A549 cell line.<br>
                                <br>
                                <b>Initial Release of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBuOrchid">BU ORChID</a>
                                track: </b></font> This track displays the predicted hydroxyl
                                radical cleavage intensity on naked DNA for each nucleotide in the
                                genome.<br>
                                <br>
                                <b>Update of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeMapability">Mapability</a>
                                track: </b></font> This track displays the level of sequence
                                uniqueness of the reference hg18 genome. The update adds CRG
                                Alignability data, which displays how uniquely k-mer sequences align
                                to a region of the genome.<br>
				<br>
			      <hr>
<!-- end news item -->
<!-- start news item -->
<a name="071910"></a>
<font size=4 color="#006666">19 July 2010 - April and May ENCODE news</b></font><br>
				<br>
				<b>Initial Release of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeGisRnaSeq">GIS RNA-seq</a>
				track: </b></font> This track shows RNA-seq of high-quality PolyA+
				RNA in ENCODE Tier1 cell lines and H1 ESC sequenced on the ABI Solid
				platform.<br>
                              	<br>
				<b>Release 3 of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeYaleChIPseq">Yale TFBS</a>

				track: </b></font> This track shows probable binding sites of the
				specified transcription factors (TFs) in the given cell types as
				determined by chromatin immunoprecipitation followed by high
				throughput sequencing (ChIP-Seq). Release 3 adds 64 experiments
				and 26 input/control datasets for a total of 54 factors in 21 cell lines.<br>
                              	<br>
				<b>Release 2 of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeRikenCage">RIKEN CAGE Loc</a>
				track: </b></font> This track shows 5' cap analysis gene expression
				(CAGE) tags and clusters in RNA extracts from different sub-cellular
				localizations. Release 2 of this track adds data for eight new
				cell-type/compartment combinations (GM12878 Nucleus, H1-hESC whole
				cell, HepG2 cytosol/nucleus/nucleolus, HUVEC cytosol, and NHEK
				cytosol/nucleus).<br>
                              	<br>

				<b>Initial Release of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeDukeAffyExonArray">Duke Affy Exon</a>
				track: </b></font> This track shows gene expression by microarray of
				RNA extracted from 28 cell lines that were also analyzed by DNaseI
				hypersensitivity, FAIRE, and ChIP assays (Open Chromatin track).<br>
                              	<br>
				<b>Initial Release of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeUwDGF">UW DNase DGF</a>
				track: </b></font> This track shows high-resolution DNase
				annotations from samples sequenced to depths of 300-fold or greater,
				in 5 cell lines.<br>

                              	<br>
				<b>Initial Release of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeGisDnaPet">GIS DNA PET</a>
				track: </b></font> This track shows the starts and ends of DNA
				fragments from different cell lines determined by paired-end ditag
				(PET) sequencing using different DNA fragment sizes for analysis of
				genome structural variation.<br>
			      	<br>
                              <hr>
<!-- end news item -->
<!-- start news item -->
<a name="031810"></a>
<font size=4 color="#006666">18 March 2010 - February and March 2010 ENCODE news</font><br>

                              <br>
				<b>Release 3 of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeChromatinMap">Open Chromatin</a>
				track: </b></font> This track displays evidence of open chromatin in
				multiple cell types from the Duke/UNC/UT-Austin/EBI ENCODE group.
				Release 3 of this track includes 18 new cell line or cell/treatment
				experiments. In addition, a number of new experiments were added to
				existing cell lines. Almost all Peaks have been called anew using
				improved cut-offs and p-Values. Finally, a second type of peak
				called using a ZINBA algorithm has been provided for several of the
				FAIRE-seq experiments.<br>
                              	<br>
				<b>Release 3 of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBroadChipSeq">Broad Histone</a>

				track: </b></font> This track shows maps of chromatin state
				generated using CHIP-seq. Release 3 of this track adds the HSMM cell
				line and includes new experiments for H1-hESC and NHLF.<br>
			      <br>
                              <font size="3"><b>Release 2
of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeUwAffyExonArray">UW Affy Exon</a> track: </b></font>
                              <meta charset="utf-8"
 id="webkit-interchange-charset">
                              This track displays
human tissue microarray data using the 

<a href="http://www.affymetrix.com/estore/browse/products.jsp?productId=131452">Affymetrix Human Exon 1.0</a> GeneChip.  This release includes 28 new cell types, and replaces the data for four existing tables (replicate 1 for K562, NB4, and SKMC; replicate 2 for HeLa-S3).
<br>
                              <br>
                              <font size="3"><b>Initial release
of the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeUwChIPSeq">UW Histone</a> track: </b></font>
                              This track displays
maps of histone modifications genome-wide in different cell lines, using ChIP-seq high-throughput sequencing.<br>
                              <br>

                              <font size="3"><b>Release 2 of the <a
 href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeHudsonalphaMethylSeq">HudsonAlpha
Methyl-seq</a> track: </b>
                         Release 2 adds data for five new cell
types.<br>
                              <font size="3"><b><br>
Release 3 of the <a
 href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeSangerGencode">Gencode
Genes</a> track:</b> </font>shows high-quality
manual annotations in the ENCODE regions generated by the 

<a href="http://www.sanger.ac.uk/PostGenomics/encode/" target="_blank">GENCODE
project</a>.<font size="3"><b> </b></font>
                              Version 3 of the
Gencode gene set presents a full merge between HAVANA and ENSEMBL,
giving priority to the manually curated Havana objects and using
ENSEMBL objects where they are different or fall into un-annotated
regions.<br>
                              <font size="3"><br>
                              <b>Initial release of the <a
 href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeCshlShortRnaSeq">CSHL
                              </a></b></font>
                              <a
 href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeCshlShortRnaSeq">Small
RNA-seq</a> track: This track depicts
NextGen sequencing information for RNAs between the sizes of 20-200 nt
isolated from RNA samples from tissues of sub cellular compartments
from ENCODE cell lines. <br>

                              <br>
<b>Release 3 of the <a
 href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeUwDnaseSeq">UW
DNaseI HS</a> track: </b></font>
                              This track shows
DNaseI sensitivity measured genome-wide in different using the Digital DNaseI
methodology, and DNaseI hypersensitive sites. This
release includes 19 new cell lines as
well as new version of NB4 replicate 1.<br><hr>
<!-- end news item -->
<!-- start news item -->
	    <a name="010610"></a>
                <font size=4 color="#006666">6 January 2010 - December 2009 ENCODE news</font></br>
		<p>
		</font><a href="http://nar.oxfordjournals.org/cgi/content/full/38/suppl_1/D620"><span
 style="font-weight: bold;">&quot;ENCODE whole-genome data in the UCSC Genome Browser&quot;</span></a>:
		This paper addresses the history of the ENCODE project, summarizes the datasets
		available as of September 2009, and outlines methods to access the data. See
		Nucleic Acids Res. 2010 Jan;38(Database issue):D620-5. <br>
		<br>
		<span style="font-weight: bold;">Initial release of the
		<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeCaltechRnaSeq">Caltech RNA-seq</a>
		track: </span>This track contains sequence reads and RPKM transcript abundance
		measures for sequences that map to either the genome or to known RNA splice sites.
		The results of four different mapping algorithms are provided, enabling
		comparison between different mapping algorithms.&nbsp; Results are available for
		polyA+ and total RNA for the two ENCODE Tier 1 cell lines.<br>
                <br>
                <span style="font-weight: bold;">Release 2 of the
		<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeBroadChipSeq">Broad Histone</a>
		track: </span>This track displays maps of chromatin state generated using CHIP-seq.
		&nbsp; Release 2 adds data for the ENCODE Tier 2 cell lines H1-hESC and HepG2, plus
		NHLF (normal human lung fibroblasts) and HMEC (human mammary epithelial) cells.	This
		expands the track data to 9 cell lines, and 11 antibodies plus an input control. <br>
		<br>
		<span style="font-weight: bold;">Release 2 of the
		<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeCshlLongRnaSeq">CSHL Long RNA-seq</a>
		track: </span>This track depicts sequencing of long RNAs of more than 200
		nucleotides in length. Release 2 adds data from strand-specific assays of
		total RNA for the two ENCODE Tier 1 cell lines.<br>
                <br>
                <span style="font-weight: bold;">Release 2 of the
		<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeChromatinMap">ENCODE Open Chromatin</a>
		track: </span>This track displays evidence of open chromatin as identified by two
		complementary methods, DNaseI hypersensitivity and FAIRE, combined with ChIP
		identification methods. Release 2 adds data from eight additional cell types,
		expanding the track to 41 experiments in 13 cell lines.<br>
		<hr>
<!-- end news item -->
<!-- Begin 2009 news items -->
	    <A NAME="2009"></A>
<!-- start news item -->
	    <a name="110709"></a>
		<font size=4 color="#006666">7 November 2009 - October ENCODE News</font></br>
		<p>
		<b>Sep 2009 data freeze complete:</b>&nbsp;&nbsp;The ENCODE Consortium has just
		completed data submissions for the fourth production data freeze (Sep 09). The first
		set of data from this freeze to complete quality review is now available on the UCSC
		public server, in Release 2 of the
		<a href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeYaleChIPseq">ENCODE Transcription Factor Binding Sites from Yale/UC-Davis/Harvard</a>
		track. Release 2 adds 59 ChIP-seq experiments to this track.</p>
                <p>
		<b>Other October track releases:</b>&nbsp;&nbsp;The
		<a href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeAffyRnaChip">Affymetrix/CSHL Subcellular RNA Localization by Tiling Array</a>
		track was expanded to include 4 additional experiments. </p>
                <p>
		<b>encodeproject.org:</b>&nbsp;&nbsp;By request of the ENCODE Consortium, the domain
		<i>encodeproject.org</i> has been registered by the ENCODE Data Coordination Center,
		and is redirected to the ENCODE portal at UCSC. </p>
                <p>
		<b>New grants funded:</b>&nbsp;&nbsp;NHGRI has funded 5 new ENCODE grants, as part
		of the American Investment and Recovery Act. The new grants include expansion of
		ENCODE to the mouse genome and proteogenomics. </p>
		<p>
		<b>Job openings at UCSC:</b>&nbsp;&nbsp;The UCSC Genome Browser and ENCODE projects
		are currently accepting applications for
		<a href="http://www.cbse.ucsc.edu/jobs/staff#sftwr">Software Developer</a> and
		<a href="http://www.cbse.ucsc.edu/jobs/staff#bdtust">Biological Database Testing/User Support Technician</a>
		positions. We are looking for talented individuals who would like to use their
		skills in computer science, biology, and bioinformatics on fast-paced projects
		featuring the work of top genomics scientists worldwide. </p>
		<hr>
<!-- end news item -->
<!-- start news item -->
	    <a name="092409"></a>
		<font size=4 color="#006666">24 September 2009 - ENCODE data releases since July 1</font></br>
		<p>
		During this period a total of 10 new ENCODE tracks were released to the UCSC public
		server. Functional elements and region characterization in these tracks include: </p>
		<ul>
		  <li>
		  <a href="../../cgi-bin/hgTrackUi?g=wgEncodeUwDnaseSeq">DNaseI</a>
		  hypersensitive sites and hotspots (University of Washington) </li>
		  <li>
		  Regions of DNA methylation, by
		  <a href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeHudsonalphaMethylSeq">Methyl-seq</a>
		  and <a href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeHudsonalphaMethyl27">Illumina 27</a>
		  array (HudsonAlpha Institute)</li>
		  <li>
		  <a href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeNhgriBip">Bi-directional promoters</a>
		  and <a href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeNhgriNre">negative regulatory elements</a>
		  (NHGRI Elnitski lab) </li>
		  <li>
		  Transcriptome profiling by RNA-seq, including single-molecule sequencing
		  (<a href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeCshlLoRnaSeq">CSHL</a>,
		  <a href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeHelicosRnaSeq">Helicos</a>)</li>
		  <li>
		  Expression levels by <a href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeUwAffyExonArray">exon array</a>
		  (University of Washington) </li>
		  <li>
		  Regions of <a href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeHudsonalphaCnv">copy number variation</a>
		  (HudsonAlpha Institute) </li>
		  <li>
		  <a href="../../cgi-bin/hgTrackUi?db=hg18&amp;g=wgEncodeMapability">Mappability/uniqueness</a>
		  of nmers (University of Massachusetts, Duke, Broad, Rosetta) </li>
		</ul>
		<p>
		For track names and file access, see the <em>Release Log</em> and <em>Downloads</em>
		links listed in the left menu bar. </p>
		<p>
		We would like to thank the contributing ENCODE labs and the the DCC team at UCSC for their efforts
		completing these tracks. </p>
		<hr>
<!-- end news item -->

<!-- start news item -->
	    <A NAME="070109"></A>
	      <FONT SIZE=4 COLOR="#006666">1 July 2009 - ENCODE data releases for the period April - June 2009</FONT></BR>
		<P>
		The following ENCODE tracks were released to the ENCODE DCC/UCSC public server during this period:
		<UL>
		  <LI><A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeAffyRnaChip"> Affymetrix/CSHL Subcellular RNA Localization by Tiling Array</A>
		  <LI><A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeChromatinMap">Open Chromatin, Duke/UNC/UT</A>
		  <LI><A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeRikenCage">RIKEN RNA Subcellular Localization by CAGE Tags</A>
		  <LI><A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaChipSeq">HudsonAlpha Transcription Factor Binding Sites by ChIP-seq</A>
		  <LI><A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeSangerGencode">Gencode Gene Annotations</A>
		  <LI><A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeSunyalbanyRnaGeneChip">SUNY Albany RNA Binding Proteins by RIP-chip</A>
		  <LI><A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBroadChipSeq">Histone Modifications by Broad Institute ChIP-seq</A>
		</UL>
		<P>
		The <EM>Release Log</EM> listed in the left menu bar shows all released ENCODE tracks.
		<P>
		We would like to thank the HudsonAlpha, Transcriptome, Open Chromatin, Broad, and SUNY ENCODE
		groups and the DCC team at UCSC for their efforts completing these tracks.</P>
		<HR>
<!-- end news item --> 

<!-- start news item -->
	    <A NAME="031309"></A>
<FONT SIZE=4 COLOR="#006666">13 March 2009 - ENCODE Data Release: Transcription Factor Binding Sites from Yale/UC-Davis/Harvard</FONT></BR>
<P>
We are pleased to announce the release to the ENCODE DCC/UCSC public server
of the 
<A TARGET=_BLANK HREF="/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeYaleChIPseq">
ENCODE Transcription Factor Binding Sites by ChIP-seq from Yale/UC-Davis/Harvard
</A>
(Yale TFBS, in the <I>Regulation</I> group).
This track shows probable binding sites of 12 transcription factors
and RNA polymerase II in 7 cell types, as determined by chromatin
immunoprecipitation followed by high-throughput sequencing.
The Genome Browser displays discrete <I>Peaks</I>
of enrichment and <I>Signal</I> graphs of enrichment density for these
experiments.
The sequence reads, quality scores, and sequence alignment coordinates from
these experiments are available for 
<A TARGET=_BLANK HREF="/ENCODE/downloads.html">download</A>.
<P>
<P>
We would like to acknowledge the efforts of the Yale/UC-Davis/Harvard
ENCODE group and the work of the UCSC data wrangler for this group,
Tim Dreszer, for completing this track.  We also thank
the entire UCSC ENCODE team and the UCSC Quality Assurance
group for contributing to this first ENCODE track release.
</P>
<!-- end news item --> 

<HR>
<!-- start news item -->
	    <A NAME="022709"></A>
		    <FONT SIZE=4 COLOR="#006666">27 Feb 2009 - ENCODE February 2009 data freeze</FONT><BR>
		    <P>
                    The February 2009 ENCODE data freeze supplements the data contributed for the
                    November 2008 freeze, and will be used together with the earlier freeze
                    for the initial analysis effort of the ENCODE Consortium. Data from
                    this freeze is being incorporated into tracks created from the first
                    data freeze, and is being reviewed by the UCSC Quality Assurance team.
		    </P>
<!-- end news item -->
<HR>
<!-- Begin 2008 news items -->
	    <A NAME="2008"></A>
<!-- start news item -->
	    <A NAME="091208"></A>
		    <FONT SIZE=4 COLOR="#006666">9 Dec. 2008 - First ENCODE whole-genome data freeze completed</FONT><BR>
		    <P>
		    The ENCODE Consortium has just completed the first freeze
		    (November 2008) of whole-genome experimental data 
		    produced for the ENCODE production phase.  Data submitted
		    to the DCC for this freeze include:
		    <UL>
		      <LI>transcription factor binding sites
		      <LI>histone modifications
		      <LI>DNaseI hypersensitive sites
		      <LI>DNA methylation
		      <LI>transcription maps and tags, localized to subcellular 
		          compartments
		      <LI>GENCODE gene annotations
		    </UL>
		    <P>
		    Experiments during this freeze focused on the ENCODE 
		    Tier1 cell lines -- K562 leukemia, and GM12878 lymphoblastoid
                    (which is also a 
                    <A HREF="http://www.1000genomes.org/page.php" TARGET=_BLANK>1000 Genomes project</A>
                    sample designated for in-depth analysis of genetic variation).
		    The freeze also includes data 
                    from some ENCODE Tier2 and Tier3 cell lines (see
                    <A HREF="cellTypes.html" TARGET=_BLANK>Cell Types</A>). The majority 
		    of these experiments were assayed by high-throughput 
		    sequencing (ChIP-seq, DNase-seq, and RNA-seq).  
		    </P>
		    <P>
		    The UCSC quality team is currently reviewing these data. 
		    When the review is complete, the browser tracks and 
		    associated downloads will be released to the UCSC public 
		    Genome Browser.
		    </P>
		    <P>
		    Thanks to the many labs who contributed data for the
		    initial phase of this project. We'd also like to acknowledge
		    the UCSC ENCODE team for data wrangling during the freeze, 
		    and for the development and maintenance of 
		    the ENCODE automated data submission pipeline and 
		    associated tools: Kate Rosenbloom, Tim Dreszer, Larry Meyer,
		    Michael Pheasant, Ting Wang, Galt Barber, and Andy Pohl. 
		    </P>
                    <HR>
<!-- end news item --> 
<!-- Begin 2007 news items -->
	    <A NAME="2007"></A>

	    <A NAME="100407"></A>
  	      <FONT SIZE=4 COLOR="#006666">4 Oct. 2007 - ENCODE Genome Browser Released for hg18 Assembly</B></FONT><BR>
		    <P>
		    The ENCODE browser for UCSC human genome assembly hg18 
		    (NCBI Build 36) is now available. You can access the browser
		    directly at 
		    <A HREF="encode.hg18.html">http://genome.ucsc.edu/ENCODE/encode.hg18.html</A> 
		    or by clicking the ENCODE link on the Genome Browser 
		    <A HREF="../index.html">home page</A>, then
		    selecting the Regions (hg18) item in the sidebar menu on 
		    the ENCODE portal page.</P>
		    <P>
		    The hg18 ENCODE browser includes 540 data tables in 59 
		    browser tracks that were migrated from the hg17 browser. 
		    The hg17 data coordinates were converted to hg18 coordinates
		    using the UCSC liftOver process.</P>
		    <P>
		    To improve the accessibility of the data, related ENCODE 
		    tracks have been gathered into new configuration groupings 
		    (&quot;super-tracks&quot;) that can be displayed or hidden
		    using a single visiblity control.  We have also reduced the
 		    number of track groups and have modified some of the group 
		    names for clarity. </P>
		    <P>
		    The following table
		    summarizes the data currently present in the hg18 ENCODE 
		    browser:
		    <BR><BR>
            	    <TABLE BORDER=1 BORDERCOLOR="#aaaaaa" WIDTH="100%">        
		      <TR>
		 	<TD nowrap><FONT color="#006666"><B>Group</B></FONT></TD>
                	<TD nowrap><FONT color="#006666"><B>Super-tracks</B></FONT></TD>
                	<TD nowrap><FONT color="#006666"><B>Tracks</B></FONT></TD>
                	<TD nowrap><FONT color="#006666"><B>Tables</B></FONT></TD>
            	      </TR>
  			<TR><TD>Regions and Genes</TD><TD>2</TD><TD>12</TD><TD>73</TD></TR>
  			<TR><TD>Transcription</TD><TD>2</TD><TD>11</TD><TD>67</TD></TR>
  			<TR><TD>Chromatin Immunoprecipitation</TD><TD>8</TD><TD>28</TD><TD>349</TD></TR>
  			<TR><TD>Chromatin Structure</TD><TD>2</TD><TD>8</TD><TD>51</TD></TR>
		    </TABLE></P>
		    <P>
		    Note that the Variation and Comparative Genomics data were 
		    not lifted during this migration; instead, they will be 
		    replaced by new data. The first ENCODE MSA alignment for 
		    hg18 (TBA) is currently in progress on the UCSC 
		    <A HREF="http://hgwdev.cse.ucsc.edu/ENCODE/">development 
		    server</A>.
		    <P>
		    During the migration, ENCODE tracks with whole-genome 
		    data were moved into the standard browser track 
		    groups. These include the GIS PET and UCSD/LI
		    TAF1 tracks. Future submissions of whole-genome ENCODE data
 		    will be loaded directly into the standard track groups.
		    </P>
		    <P>
		    We have expanded the ENCODE downloads site to include 
		    original data for all &quot;wiggle&quot; datasets. These 
		    data files now have filename extensions indicating the 
		    wiggle input format (fixed step, variable step, or 
		    bedGraph).</P>
		    <P>
		    You can find a description of the migration project and 
		    full details of the tables, tracks, and super-tracks 
		    available at the UCSC ENCODE portal on the UCSC
                    <A HREF="http://genomewiki.ucsc.edu/index.php/ENCODE_Hg18_Migration" TARGET=_BLANK>
                    genomeWiki</A>.</P>
		    <P>
		    The UCSC team members who contributed to this effort were:
		    Andy Pohl (data conversion and database loading), Ting Wang
		    and Donna Karolchik (super-track documentation), Bob Kuhn 
		    (portal updates), Brooke Rhead, Kayla Smith, and Ann Zweig (quality 
		    assurance), and Kate Rosenbloom (super-track development, 
		    project management).
		    <P>
		    <HR>

	    <A NAME="061307"></A>
  	      <FONT SIZE=4 COLOR="#006666">13 Jun. 2007 - ENCODE Findings Published in <em>Nature</em> and <em>Genome Research</em></B></FONT><BR>
		   <P>
		    The findings of the ENCODE project have been
		    released to the public today, the culmination of a 
		    four-year effort to catalog the biologically functional 
		    elements in 1 percent of the human genome. The publications,
		    which include a group paper in the 14 June 2007 issue of 
		    <A HREF="http://www.nature.com/nature/journal/v447/n7146/full/nature05874.html"
		    TARGET=_blank><em>Nature</em></A> and 28 companion papers 
		    in the June 2007 issue of 
		    <A HREF="http://www.genome.org/content/vol17/issue6/"
		    TARGET=_blank><em>Genome Research</em></A>, were authored by
		    researchers from academic, governmental, and industry
		    organizations located in 11 countries. The <em>Nature</em> 
		    issue includes a pull-out poster featuring a screenshot of 
		    the UCSC Genome Browser displaying a broad range of the 
		    ENCODE data.</P>
		    <P>
		    In the <A HREF="http://genome.gov/25521554"
		    TARGET=_blank>press release</A> accompanying the publication
		    rollout,
		    NHGRI Director Francis S. Collins is quoted as saying
		    &quot;This impressive effort has uncovered many exciting 
		    surprises and blazed the way for future efforts to explore 
		    the functional landscape of the entire human genome. Because
		    of the hard work and keen insights of the ENCODE consortium,
		    the scientific community will need to rethink some long-held
		    views about what genes are and what they do, as well as how 
	   	    the genome's functional elements have evolved. This could 
		    have significant implications for efforts to identify the 
		    DNA sequences involved in many human diseases.&quot;</P>
		    <P>
		    For more information on the ENCODE project, 
		    including the consortium's data release and accessibility 
		    policies and a list of NHGRI-funded participants, see the
		    <A HREF="http://www.genome.gov/ENCODE"
		    TARGET=_blank>NHGRI ENCODE website</A>. 
		    <P>
		    <HR>
	    <A NAME="061207"></A>
  	      <FONT SIZE=4 COLOR="#006666">12 Jun. 2007 - Spring 2007 ENCODE News</B></FONT><BR>
		    <P>
		    Between January and May of 2007, several new or upgraded 
		    data tracks were released by UCSC:
		    <UL>
		    <LI><B>Gencode March 2007 Genes</B> and <B>Gencode 
		    RACEfrags</B> -- 
		    Reannotation of 69 loci consisting of 132 transcripts
		    based on RACE, array, and sequencing analyses.  New features
		    include the addition of PolyA features, polymorphic
		    gene type, and integration of experimental intron 
		    validation.
		    <LI><B>Yale ChIP-chip RFBR</B> -- 
		    Analysis of ChIP-chip data to identify regulatory factor 
		    binding regions, resulting from over 105 experiments 
		    representing 29 transcription factors in 9 cells lines, 
		    performed in 7 labs using 3 microarray platforms. Data 
		    provided by the ENCODE Transcriptional Regulation analysis 
		    group.
		    <LI><B>University of Virginia DNA Replication Origins</B> --
		    Three new datasets representing two new experimental
		    methods for origin mapping (bubble trapping and nascent 
		    strand) in HeLa and GM06990 cells.
		    <LI><B>University of Vienna RNAz</B> --
		    RNA secondary structure prediction as predicted by RNAz,
		    based on evolutionary conservation and thermodynamic 
		    stability.
		    <LI><B>Duke DNaseI Hypersensitivity</B> -- 
		    DNase-chip results in IMR90 (fibroblast), K562 (leukemia) 
		    and H9 (stem) cells.
		    <LI><B>UCSC EvoFold</B> -- Replaces TBA23 EvoFold.
		    <LI><B>UC Davis ChIP Hits</B> -- Updated cMyc and E2F1.
		    </UL></P>
		    <P>
		    Thanks to the UCSC staff who worked on these tracks:
		    Rachel Harte and Kate Rosenbloom (development), and Ann 
		    Zweig, Archana Thakkapallayil, and Kayla Smith (quality 
		    review).</P>
		    <P>
		    <HR>

	    <A NAME="010907"></A>
  	      <FONT SIZE=4 COLOR="#006666">9 Jan. 2007 - Winter 2006 ENCODE News</B></FONT><BR>
		    <P>
		    To improve communication, we have posted instructions for 
		    our ENCODE ftp site on the ENCODE Wiki and have set up an 
		    email alias for notifying UCSC about your ENCODE data 
		    submissions: <A HREF="mailto:encode@soe.ucsc.edu">encode@soe.ucsc.edu</A>.
		    The current UCSC recipients are Kate Rosenbloom, Daryl 
		    Thomas, Ting Wang, and Rachel Harte.</P>
		    <P>
		    During November and December, four new/improved data tracks 
		    were released:
		    <UL>
		    <LI>
		    <B>Chip-PET from the Genome Institute of Singapore</B> -  
		    Genome-wide data for c-Myc in P493 B cells was added as a 
		    new subtrack (cMyc P493, encodeGisChipPetMycP493) to the 
		    existing GIS ChIP-PET track.
		    <LI>
		    <B>DNaseI Hs from Duke University</B> - 
		    Existing NHGRI data and new data from the Crawford lab at 
		    Duke University (raw and p-value data for the HepG2 cell 
		    line) were merged into the Duke/NHGRI DNase track. The 
		    newer data is based on DNase-chip technology.
		    <LI>
		    <B>STAGE tags from University of Texas</B> - 
		    Raw tags data for STAT1 in HeLa cells were added
		    as a new subtrack (UT STAT1 HeLa Tags, 
		    encodeUtexStageStat1HelaTags)
		    to the existing UT-Austin Stage track.
		    <LI>
		    <B>DNaseI Hs from  University of Washington</B> - 
		    The existing three UW/Regulome DNaseI Sens tracks were 
		    replaced with a single new track (UW/Reg QCP DNaseI Sens) 
		    based on quantitative chromatin profiling (QCP) methods in 
		    16 cell types.
		    </UL>
		    <P>
		    Thanks to the UCSC staff who worked on these tracks:
		    Rachel Harte, Kate Rosenbloom, and Hiram Clawson 
		    (development), Ann Zweig, Brooke Rhead, and Kayla Smith 
		    (quality review).
		    <P>
		    <HR>

	    <A NAME="2006"></A>

	    <A NAME="100706"></A>
  	      <FONT SIZE=4 COLOR="#006666">7 Oct. 2006 - Comparative Genomics Data Release</B></FONT><BR>
		    <P>
		    Twelve tracks of data produced by the ENCODE Multi-Species
		    Sequence Analysis group have been released to the UCSC 
		    public server. These tracks contain multiple sequence 
		    alignments, conservation, and conserved (constrained) 
		    elements produced by four conservation methods
		    (phastCons, binCons, GERP, SCONE) applied to three sequence 
		    alignments (TBA, MLAGAN, MAVID), and also an assessment of 
		    the agreement among the alignment methods. The alignments 
		    were based on genomic sequence in the ENCODE regions of 28 
		    vertebrate species, as defined in the 
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg17/encode/alignments/SEP-2005"
		    TARGET=_blank>MSA September 2005 sequence freeze</A>.
		    <P>
		    The following tracks can now be found in the ENCODE 
	   	    Comparative Genomics track group on the public ENCODE 
		    browser:
		    <UL>
		    <LI>MSA Consensus Constrained Elements
		    <LI>TBA Alignments, TBA Conservation, TBA Conserved Elements
		    <LI>MLAGAN Alignments, MLAGAN Conservation, MLAGAN Conserved
		    Elements
		    <LI>MAVID Alignments, Conservation, Conserved Elements
		    <LI>MSA Alignment Agreement,  MSA Alignment Gaps
		    </UL></P>
		    <P>
		    Thanks to the following providers of this data:
		    <UL>
     		    <LI>Elliott Margulies (NHGRI) - TBA alignments, binCons and 
		    phastCons conservation, Consensus elements, sequence freeze
		    <LI>Daryl Thomas (Haussler lab, UCSC) - sequence freeze
     		    <LI>George Asimenos (Batzoglou lab, Stanford University) - 
		    MLAGAN alignments
		    <LI>Colin Dewey (while at the Pachter lab, UC Berkeley) - 
		    MAVID alignments
		    <LI>Greg Cooper (while at the Sidow lab, Stanford 
		    University) - GERP conservation
     		    <LI>Saurabh Asthana (Sunyaev lab, Harvard Genetics) - 
		    SCONE conservation
     		    <LI>Ariel Schwartz (Pachter lab, UC Berkeley) - Alignment
		    agreement
		    </UL></P>
		    <P>
		    Also, thanks to the UCSC team that produced these tracks in 
		    the browser: Kate Rosenbloom (track development), Ann Zweig,
		    Kayla Smith, and Archana Thakkapallayil (quality review).
		    </P>
		    <P>
		    <HR>

	    <A NAME="083106"></A>
  	      <FONT SIZE=4 COLOR="#006666">31 Aug. 2006 - Summer ENCODE data activity</B></FONT><BR>
		    <P>
		    Since mid-June, UCSC has released new ENCODE data from three
		    labs (Sanger Institute, Uppsala University, and University
		    of North Carolina) and has a track in
		    progress for newly submitted data from a fourth lab 
		    (NHGRI/Duke University): </P>
		    <P>
		    <UL>
		    <LI>
		    <B>Sanger ChIP-chip (MOLT4 and PTR8 cells)</B>  - Eight new 
		    datasets were added to the Sanger Chip/chip track.
		    The new data show sites of H3 histone methylation and
		    acetylation in MOLT4 (lymphoblastic leukemia) and also
		    in the chimpanzee PTR8 cell line, used for comparative
		    analysis.  Thanks to Rob Andrews at the Ian Dunham lab
		    for providing these data.
		    <LI>
		    <B>Uppsala University Chip/chip Butyrate</B> - This 
		    track shows the effects of Na-butyrate treatment of
		    HepG2 (liver carcinoma) cells on histone H3 and H4 
		    acetylation, assayed on Sanger microarrays.  Thanks to Adam 
		    Ameur at the Claes Wadelius lab for providing these data.
		    <LI>
		    <B>University of North Carolina FAIRE (Peaks data 
		    update)</B> - 
		    This track was updated to include a subtrack of peaks
		    generated by an alternate peak-finding algorithm,
		    ChIPOTle.  The FAIRE data were generated from 2091
		    fibroblast cells hybridized to NimbleGen ENCODE arrays.
		    Thanks to Paul Giresi at the Jason Leib lab for providing
		    these data.
		    <LI>
		    <B>NHGRI DNaseI-HS Raw (in progress)</B> - DNase-chip raw 
		    and p-value data in GM06990, CD4+, and Hela S3 cells.
		    </UL></P>
		    <P>
		    Thanks to our UCSC staff who worked on these tracks:
		    Rachel Harte and Hiram Clawson (development), Ann Zweig
		    and Archana Thakkapallayil (quality review), and Donna
		    Karolchik (documentation).</P>
		    <P>
		    The ENCODE data status page has been updated to reflect the
	   	    recent activity.</P>
		    <P>
		    <HR>

	    <A NAME="061406"></A>
  	      <FONT SIZE=4 COLOR="#006666">14 June 2006 - New ENCODE data at UCSC</B></FONT><BR>
		    <P>
		    During the build-up to the analysis paper submissions,
		    UCSC received a flurry of ENCODE data submissions
		    (6 during the month of May).  We have recently
		    released three data sets to our public server; 
		    the remaining tracks are in progress, as indicated below.
		    <P>
		    <B>Released data:</B>
		    <P>
		    <UL>
		    <LI>
		    Sanger ChIP-chip (HFL-1 cells) - 
		    New data added to the Sanger ChIP track show the
		    location of modified histones in HFL-1 (embryonic lung
		    fibroblast) cells.   Thanks to Rob Andrews and
		    Christopher Koch for providing this data.
		    <LI>
		    DLESS (Detection of LinEage Specific Selection) - 
		    This track shows elements predicted by the DLESS program
		    to be under lineage-specific selection, based on
		    alignments of 17 mammalian species from NHGRI/PSU 
		    TBA ENCODE alignments. DLESS is based on a phylo-HMM 
		    with states for neutrally evolving and conserved regions, 
		    and for gains and losses on each branch of the tree.  
		    Thanks to Adam Siepel of Cornell University, who 
		    developed the DLESS program, generated the data, and 
		    loaded the annotation track.
		    <LI>
		    UW/Regulome Dnase/Array - 
		    This track displays DNaseI sensitivity in GM06990 cells,
		    using the DNase/Array methodology.  Dnase/Array is a novel
 		    method for isolating DNA segments corresponding to 
		    specific DNaseI cleavage events on individual nuclear 
		    chromatin templates. Thanks to Scott Kuehn at the
 		    University of Washington for providing these data.
		    </UL></P>
		    <P>
		    Thanks to our UCSC staff who worked on these tracks: 
		    Rachel Harte (development), Ann Zweig, Kayla Smith, 
		    and Archana Thakkapallayil (quality review).
		    <P>
		    <B>In quality review:</B>
		    <UL>
		    <LI>UNC FAIRE Peaks (Update, 2091 fibroblasts)
		    </UL></P>
		    <P>
		    <B>In progress:</B>
		    <UL>
		    <LI>University of Uppsala Chip/Chip (butyrate treatment in HepG2)
		    <LI>UW/Regulome QCP Dnase HS (update and new cell lines -- 16 tissues)
		    <LI>PECAN Alignments from EBI
		    </UL></P>
		    <P>
		    The ENCODE data status page has been updated to reflect the 
		    recent activity.</P>
		    <P>
		    <HR>

	    <A NAME="011306"></A>
  	      <FONT SIZE=4 COLOR="#006666">13 January 2006 - Release of October freeze data complete</B></FONT><BR>
		    <P>
		    All datasets submitted for the October 2005 ENCODE freeze
		    have now been released.  The tracks with their release
		    dates are:
		    <UL>
		    <LI>13 Jan. - Yale ChIP-chip (5 new transcription factors)
		    <LI>3 Jan. - University of Uppsala Chip/chip (data correction)
		    <LI>2 Jan. - Yale RNA (update)
		    <LI>29 Dec. - U Virginia DNA Replication Timing and Predicted Origins
		    <LI>28 Dec. - UT Austin STAGE
		    <LI>21 Dec. - Affy RNA and Transfrags (HeLa update)
		    <LI>22 Nov. - Sanger ChIP-chip (5 histone modifications)
		    <LI>19 Nov. - UC Davis ChIP-chip (Pol2, Myc) in HeLa
		    <LI>18 Nov. - U North Carolina FAIRE (Formaldehyde Assisted Isolation of Regulatory Elements) 
		    <LI>1 Nov. - Affy RNA and Transfrags (update)
		    <LI>1 Nov. - UT-Austin Ng ChIP-chip (E2F4, c-Myc factors)
		    <LI>1 Nov. - GIS PET RNA (update)
		    <LI>1 Nov. - Gencode Genes and Introns (update)
		    <LI>1 Nov. - NHGRI DNaseI HS (update and new GM06990)
		    <LI>27 Oct. - UCSD/LI Ng ChIP-chip (update and new TFs)
		    <LI>14 Oct. - GIS ChIP/PET (STAT1)
		    <LI>21 Sep. - Boston University ORChID (update)
		    </UL></P>
		    <P>
		    <HR>

<!- Begin 2005 news items -->
	    <A NAME="2005"></A>

	    <A NAME="101705"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 17 Oct. 2005 - ENCODE hg17 Genome Browser Released</B></FONT><BR>
		    <P>
		    The ENCODE browser for UCSC human genome assembly hg17 
		    (NCBI Build 35) is now available at
    		    <A TARGET=_BLANK HREF="http://genome.ucsc.edu/ENCODE/encode.hg17.html">
                    http://genome.ucsc.edu/ENCODE/encode.hg17.html</A>
		    and from the
    		    <A TARGET=_BLANK HREF="http://genome.ucsc.edu/ENCODE/encode.hg17.html">Regions (hg17)</A>
                    item in the left (blue bar) menu of the ENCODE portal.
		    <P>
		    All datasets originally submitted in hg17 coordinates for 
		    the June data freeze were directly loaded; the remaining 
		    data  were coordinate-converted using the UCSC liftOver 
		    process. A total of 351 data tables were loaded into our 
		    database.
		    <B>NOTE:</B> Many of these tracks will be updated with new 
		    data from the October ENCODE data freeze.</P>
		    <P>
		    Many thanks to all the ENCODE consortium members who
		    contributed data for this release. We'd also like to 
		    thank UCSC team members Kate Rosenbloom for portal and 
		    track development and Jennifer Jackson, Kayla Smith, Ann 
		    Zweig, and Bob Kuhn for quality assurance.</P>
		    <P>
		    <HR>

	    <A NAME="071505"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 15 July 2005 - Regulome DnaseI HS, Oxford Recombination, and MSA Data Released</FONT><BR>
		    <P>
		    The remaining datasets submitted for the June 2005 ENCODE 
		    data freeze have now been released as annotation tracks in 
		    the hg16 browser.  These tracks and their release dates are:
		    <UL>
		    <LI>15 Jul. - TBA23 EvoFold RNA Structure Predictions
		    <LI>14 Jul. - Regulome DNaseI HS and Sens
		    <LI>14 Jul. - Regulome Plate Q/A and Amplicons
		    <LI>14 Jul. - MSA Consensus Elements
		    <LI>14 Jul. - TBA alignments with Conservation
		    <LI>14 Jul. - Stanford MLAGAN Alignments and Conservation
		    <LI>13 Jul. - Oxford Recombination
		    </UL></P>
		    <P>
		    <HR>

	    <A NAME="071205"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 12 July 2005 - EGASP genes, RIKEN CAGE tags, and Stanford RTPCR, ChIP-chip and DNA Methylation data released</FONT><BR>
		    <P>
		    Several new ENCODE data sets have been released in the UCSC
		    Genome Browser. </P>
		    <P>
		    <B>EGASP Full, Partial, and Update: </B> These three 
		    gene prediction tracks are from the ENCODE Gene Annotation 
		    Assessment  Project (EGASP) Prediction Workshop 2005. 
		    The EGASP Full track shows 20 sets of gene predictions 
		    originally submitted for the workshop, covering all 44 
		    ENCODE regions.  THE EGASP Partial track shows eight sets of
		    gene precdictions that were submitted for the workshop, but 		    do not cover all ENCODE regions. The EGASP Update track 
		    shows updated versions of some of the submitted predictions.
		    Thanks to Julien Lagarde at IMIM for providing the EGASP 
		    Full and EGASP Partial data sets.  Thanks to Tyler Alioto of
		    IMIM (GeneID-U12 and SGP2-U12), Deyou Zheng of Yale (Yale 
		    Pseudogenes), Sarah Djebali of Ecole Normale Suprieure 
		    (Exogean), Jonathan Allen of TIGR/Univ. Maryland (Jigsaw) 
		    and Mario Stanke of the University of Gottingen (Augustus) 
		    for providing their EGASP Update gene sets.</P>
		    <P>
		    <B>RIKEN CAGE Predicted Gene Start Sites: </B> This track 
		    shows the numbers of 5' cap analysis gene expression (CAGE) 
		    tags that map to the genome at specific locations. Areas in 
		    which many tags map to the same region may indicate a 
		    significant transcription start site. Thanks to Albin 
		    Sandelin at RIKEN and the FANTOM (Functional Annotation
		    of Mouse) Consortium for providing these data.</P>
		    <P>
		    <B>Stanford RTPCR: </B> This track displays absolute 
		    transcript copy numbers for 136 genes and 12 negative 
		    control intergenic regions, determined by real-time PCR in
		    HCT116 cells.</P>
		    <P>
		    <B>Stanford ChIP-chip, Stanford ChIP-chip Smoothed 
		    Score: </B> These tracks display raw scores and 
		    sliding-window mean scores for regions bound by the Sp1 
		    and Sp3 transcription factors in three cell lines (HCT116, 
		    Jurkat, K562), assayed by chromatin immunoprecipitation and 
		    hybridization to Nimblegen oligo tiling arrays. The 
		    Smoothed Score track is provided solely for visualization 
		    purposes -- the raw scores should be used for data analysis.
		    </P>
		    <P>
		    <B>Stanford Methylation Digest: </B> This track displays 
		    experimentally determined regions of unmethylated 
		    CpGs in eight cell lines (BE2C, CRL1690, HCT116, HT1080, 
		    HepG2, JEG3, SNU182, and U87MG). Genomic DNA was digested 
		    with a cocktail of six methyl-sensitive restriction enzymes,
		    size-selected, amplified, labeled, and hybridized to 
		    Nimblegen oligo tiling arrays. 
		    <P>
		    Thanks to Nathan Trinklein for providing the Stanford 
		    datasets.</P>
		    <P>
		    We'd like to acknowledge the work of the UCSC Genome
		    Bioinformatics team members who produced these tracks:
		    Kate Rosenbloom, Angie Hinrichs, and Hiram Clawson 
		    (development), Ann Zweig, Bob Kuhn, Galt Barber, Rachel
		    Hart, and Ali Sultan-Quarrie (quality review), and Donna 
		    Karolchik (documentation).</P>
		    <P>
		    <HR>

	    <A NAME="070605"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 6 July 2005 - NHGRI DIPs, HapMap SNPs, Sanger Assoc annotation tracks released</FONT><BR>
		    <P>
		    The first datasets in the ENCODE Variation group are now
		    available in the UCSC browser.</P>
		    <P>
		    <B>NHGRI Deletion/Insertion Polymorphisms: </B> All human 
		    trace data from NCBI's trace archive were aligned to the 
		    genome and processed using the programs ssahaSNP and
		    ssahaDIP to detect deletion and insertion polymorphisms.
		    Thanks to Jim Mullikin at NHGRI for performing the analyses 
		    and providing these data.</P>
		    <P>
		    <B>HapMap Allele Frequencies: </B> This track shows allele 
		    frequencies for the four HapMap populations in the ten 
		    ENCODE regions that have been resequenced for variation 
		    (manually selected regions m010, m013, and m014 and 
		    randomly selected regions r112, r113, r123, r131, r213, 
		    r232, and r321).  These data were obtained from HapMap 
		    public release #16c.1. Thanks to the International HapMap 
		    Project for making this information available.</P>
		    <P>
		    <B>Sanger Genotype-Expression Association: </B> This track 
		    displays associations among gene expression data from the
		    60 unrelated Centre d'Etude du Polymorphisme Humain (CEPH) 
		    individuals of the International HapMap Project with SNPs 
		    genotyped by HapMap, 
		    in eight ENCODE regions (m010, m013, m014 and r123, r131,
		    r213, r232, and r321). The CEPH population is composed 
		    of Utah residents with ancestry from northern and western 
		    Europe.  The expression data were generated with the 
		    Illumina platform at the Wellcome Trust Sanger Institute.
		    Thanks to Manolis Dermitzakis at the Sanger Institute for 
		    providing these data.</P>
		    <P>
		    We'd also like to acknowledge the UCSC ENCODE team members 
		    who worked on these tracks: Heather Trumbower, Daryl Thomas,
		    and Angie Hinrichs (development), Galt Barber and Ali 
		    Sultan-Qurraie (quality assurance), and Donna Karolchik
		    (documentation). </P>
		    <P>
		    <HR>

	    <A NAME="062305"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 23 June 2005 - Yale and Affymetrix ChIP-chip and transcription data release; New ENCODE track groupings</FONT><BR>
		    <P>
		    To aid ENCODE analysis and reduce visual clutter, we have 
		    split the Genome Browser ENCODE track group into six new 
		    groups:
		    <UL>
		    <LI>ENCODE Regions and Genes
     		    <LI>ENCODE Transcript Levels
     		    <LI>ENCODE Chromatin Immunoprecipitation
     		    <LI>ENCODE Chromosome, Chromatin and DNA Structure
     		    <LI>ENCODE Variation
     		    <LI>ENCODE Comparative Genomics
		    </UL>
		    All of these track groups are visible on the UCSC test 
		    browser. The last two groups, Variation and Comparative 
		    Genomics, do not yet have published tracks on the public 
		    server and therefore are not visible on that server.
		    <P>
		    We have also released a set of new Yale 
		    data and an extensive update of Affymetrix data.  The track 
		    controls for these datasets can be found in the
		    track groups ENCODE Transcript Levels and ENCODE Chromatin 
		    Immunoprecipitation.</P>
		    <P>
		    <B>Yale ChIP-chip and RNA:</B> Three tracks of ChIP-chip 
		    data from Yale, evaluating microarray platforms, have been 
		    released: Yale ChIP pVal, Yale ChIP Sig, and Yale ChIP 
		    Sites.  These tracks show results of ChIP experiments using
		    STAT1 antibody in HeLa cells on four different microarrays 
		    -- three custom maskless photolithographic oligo arrays, 
		    designed at different resolutions, and the PCR amplicon 
		    array developed by the Ren lab at the Ludwig Institute/UCSD.
		    </P>
		    <P>
		    Two tracks of RNA transcript data from Yale have been 
		    released: Yale RNA and Yale TAR. These tracks show 
		    transcriptionally active regions and transcribed
		    fragments for three cell types (neutrophil, placenta, and 
		    NB4 variously treated for differentiation).</P>
		    <P>
		    Thanks to Joel Rozowsky at Yale for providing this data.
		    Additional Yale ChIP-chip data is currently under review by 
		    our quality assurance group.</P>
		    <P>
		    <B>Affymetrix ChIP-chip and RNA: </B>
		    The Affymetrix ChIP-chip dataset now contains experimental 
		    results for ten factors (Brg1, CEBPe, CTCF, H3K27me3, 
		    H4Kac4, P300, PU1, Pol2, RARA, and SIRT1) in HeLa cells at 
		    four timepoints after retinoic acid treatment, plus TFIIB 
		    for the final timepoint only. The data is displayed in 
		    eight tracks: Affy PVal 0h, 2h, 8h, 32h and Affy Sites 0h, 
		    2h, 8h, 32h. We acknowledge that this track grouping 
		    is a bit awkward and are working composite track 
		    enhancements to provide more flexibility.</P>
		    <P>
		    The Affy RNA tracks show RNA abundance and transfrags in 
		    retinoic acid-stimulated HL-60 cells at four timepoints,
		    and in GM06990 and HeLa cells: Affy RNA Signal and Affy 
		    Transfrags. </P>
		    <P> 
		    Thanks to Stefan Bekiranov and Srinka Ghosh at Affymetrix 
		    for providing these data. </P>
		    <P>
		    Thanks also to the UCSC ENCODE 
		    team members who developed and reviewed these tracks and 
		    to Rachel Harte in the UCSC browser group
		    for her assistance with track review.</P>
		    <P>
		    <HR>

	    <A NAME="061505"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 15 June 2005 - Six data sets released in Genome Browser</FONT><BR>
		    <P>
		    Six more ENCODE annotation tracks have been added to the
		    Genome Browser this week:</P>
		    <P>
		    <B>NHGRI DNaseI-Hypersensitive Sites (update): </B>The 
		    NHGRI DNaseI-HS track has been updated with new data. The
		    track now includes DNaseI-hypersensitive sites in CD4+ 
		    T-cells before and after activation by anti-CD3 and 
		    anti-CD28 antibodies. Thanks to Greg Crawford at the 
		    Collins lab (NHGRI) for providing these data.</P>
		    <P>
		    <B>Genome Institute of Singapore PET of PolyA+ RNA: </B>
		    The GIS PET RNA track displays starts and ends of mRNA 
		    transcripts determined by paired-end ditag sequencing in 
		    two cell lines, MCF7 and HCT116 treated with 5 
		    fluoro-uracil. A total of 584,624 PETs were generated for 
		    MCF7 and 280,340 were generated for HCT116. More than
		    80% of the PETs in each group were mapped to the genome.
		    Thanks to Atif Shahab, Yijun Ruan, the GIS, and the 
		    Bioinformatics Institute of Singapore for providing these 
		    data.</P>
		    <P>
		    <B>Gencode Gene Annotations and Intron Validation: </B>
		    The Gencode Genes track displays high-quality manual 
		    annotations in the ENCODE regions generated by the 
		    GENCODE project.  A companion track, Gencode Introns, 
		    shows experimental gene structure validations for these
		    annotations.  Thanks to the HAVANA team at the Wellcome
		    Trust Sanger Institute; France Denoeud, Julien Lagarde,
		    and Roderic Guigo at the IMIM; and Alexandre Reymond
		    at the University of Geneva for providing the
		    annotations and experimental confirmation, as well as  
		    working with UCSC to develop the track display.</P>
		    <P>
		    <B>Boston University Hydroxyl Radical Cleavage: </B>
		    The BU ORChID track displays predicted hydroxyl radical 
		    cleavage intensity on naked DNA for each nucleotide in 
		    the ENCODE regions.  The prediction algorithm draws data 
		    from a database of 150 experimentally-determined cleavage 
		    patterns. Thanks to Jay Greenbaum at the Tullius lab for 
		    providing these data.</P>
		    <P>
		    <B>UC Davis ChIP-chip: </B> The UCD Ng E2F1 track shows 
		    ChIP analysis of HeLa cells using E2F1 antibody as 
		    assayed on Nimblegen arrays. The E2F1 factor is associated
		    with cell cycle control, transcriptional regulation, and 
		    apoptosis. Thanks to the Farnham lab and Kyle Munn, Todd 
		    Richmond and Roland Green of Nimblegen for providing 
		    these data.</P>
		    <P>
		    <B>Ludwig Institute/UCSD ChIP-chip: </B>
		    Three tracks of ChIP-chip data using Nimblegen arrays 
		    have been released: LI Ng TAF1 IMR90, LI Ng Validation,
     		    and LI Ng gIF ChIP. The TAF1 track shows genome-wide 
		    TAF1 binding sites as determined by ChIP in IMR90 
		    (fibroblast) cells assayed on Nimblegen high-density 
		    oligo arrays. The peaks from the genome scan experiments 
		    were verified using four antibodies associated with 
		    transcription start (Pol2, H3ac, H3K4me2, and TAF1) on 
		    condensed arrays covering the putative TAF1 binding sites.
		    The gIF track shows the results of ChIP-chip experiments 
		    using gamma interferon-treated HeLa cells, with Pol2 and 
		    H3K4me3 antibodies.</P>
		    <P>
		    Two additional tracks, LI ChIP Various and LI gIF ChIP, 
 		    display ChIP-chip data assayed on Ren lab ENCODE PCR
		    tiling arrays for five antibodies (Pol2, TAF1, H3K4me2, 
		    SUZ12, H3K27me3) and four cell lines (HeLa, THP1, IMR90, 
		    HCT116), as well as gamma interferon experiments using 
		    seven antibodies (Pol2, TAF1, H3K4me2, H3K4me3, H3ac, 
		    H4ac, and STAT1) in HeLa cells.
		    Thanks to Chunxu Qu and Bing Ren, at the Ren lab for 
		    providing these data and assisting with track display 
		    issues.</P>
		    <P>
		    We'd also like to acknowledge the UCSC team members who 
		    developed these tracks: Kate Rosenbloom, Hiram Clawson, 
		    and Angie Hinrichs for track development; Bob Kuhn, Ali
		    Sultan-Qurraie and Galt Barber for quality assurance; 
		    and Donna Karolchik and Jim Kent for documentation.</P>
		    <P>
		    <HR>

	    <A NAME="060905"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 9 June 2005 - Sanger Histone Modification ChIP-chip track updated</FONT><BR>
		    <P>
		    The Sanger Institute has submitted ChIP-chip data for 
		    additional antibodies and cell lines, which we have 
		    incorporated into the existing Sanger ChIP browser track. 
		    The expanded track now contains data for five antibodies
		    (H3K4me1, H3K4me2, H3K4me3, H3ac, H4ac) and two cell 
		    lines (GM06990, K562 (leukemia)).</P>
		    <P>
		    Thanks to Rob Andrews and Chris Koch at the Dunham lab 
		    for providing these data.  UCSC team members who developed
		    this track include Hiram Clawson (track development), Ali 
		    Sultan-Qurraie (quality assurance), and Donna Karolchik 
		    and Jim Kent (documentation).</P>
		    <P>
		    <HR>

	    <A NAME="060605b"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 6 June 2005 - Changes to ENCODE track labels in Genome Browser</FONT><BR>
		    <P>
		    We have modified the Genome Browser labels for the 
		    existing ENCODE data tracks to trim overly-long labels
		    that were truncated in the display and to facilitate 
		    cross-track analysis. The new label format shows the 
		    submitter and the experiment, followed by the cell line 
		    (in tracks where the data includes only one cell line).
		    </P>
		    <P>
		    <HR>

	    <A NAME="060605a"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 6 June 2005 - Stanford Promoters and UVa DNA Replication tracks released</FONT><BR>
		    <P>
		    We are pleased to announce the first UCSC Genome Browser 
		    tracks released for the June 2005 ENCODE data freeze:
		    Stanford Promoters and UVa DNA Replication Temporal 
		    Profiling.
		    <P>
		    Stanford has provided an update of their promoter activity
		    data based on transient transfection luciferase reporter 
		    assays of 643 putative promoter fragments in the ENCODE 
		    regions.  The update includes two additional cell lines 
		    and activity averaged across all cell lines.  The data 
		    tables now contain additional experimental detail to 
		    facilitate analysis.  This track, containing 17 subtracks 
		    (16 cell lines and the average), is labeled 				  &quot;Stanf. Promoter&quot;. Thanks to Sara Hartman at 
		    the Myers lab for providing these data.</P>
		    <P>
		    The Dutta lab at Univerity of Virginia (UVa) has completed
		    the second biological replicate of their temporal 
		    profiling of HeLa cell replication products and has
		    provided a dataset containing merged data from the two 
		    replicates. The track, containing five subtracks 
		    representing two-hour intervals, is labeled
		    &quot;UVa DNA Rep&quot;. Thanks to Christopher Taylor at 
		    the Dutta lab for providing these data.</P>
		    <P>
		    We'd also like to acknowledge the UCSC team members who
		    worked on these annotation tracks: Angie Hinrichs (track
		    development), Galt Barber and Ali Sultan-Qurraie (QA), and
		    Jim Kent and Donna Karolchik (track documentation).</P>
		    <P>
		    <HR>

	    <A NAME="052405b"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 24 May 2005 - New MSA sequence data freeze available</FONT><BR>
		    <P>
		    A new ENCODE MSA sequence data freeze is available on the 
		    UCSC downloads server. The latest freeze contains 
		    sequences from 23 vertebrates provided by NISC, Baylor, 
		    the Broad Institute (2X) and the whole genome shotgun 
		    (WGS) assemblies.  The data may be downloaded as 
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/alignments/MAY-2005/">individual data files</A> or a
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/alignments/MAY-2005.tgz">directory tarball</A>. 
		    Aligners are encouraged to upload alignments and related 
		    data (such as conservation scores and elements) to the 
		    UCSC ENCODE ftp site as soon as possible and then notify 
		    <A HREF="mailto:kate@soe.ucsc.edu">Kate Rosenbloom</A>. 
		    Other data, (conservation, trees, etc.) 
		    will be generated based on this dataset.</P>
		    <P>
		    The following is a summary of data updates from the
		    previous release:
		    <UL>
		    <LI>
		    The human assembly version remains at hg16 (Jul. 2003).
		    <LI>
		    The mouse assembly has been updated from mm5 (May 2004) to
		    mm6 (Mar. 2005).
		    <LI>
		    The multiple rat sequences have been replaced by a single
		    sequence: rn3 (Jun. 2003).
		    <LI>
		    The cow sequences have been updated using an assembly of 
		    BAC-based sequences provided by Baylor College of 
		    Medicine.
		    <LI>
		    Fugu (fr1), macaque (rheMac1), opossum (monDom1), 
		    Tetraodon (tetNig1), Xenopus (xenTro1), and zebrafish 
		    (danRer2) have been added.  The macaque sequence was 
		    obtained from a Baylor 
		    College of Medicine assembly that has not yet been 
		    officially released. 
		    <LI>
		    A new NISC species, rfbat 
		    (<em>Rhinolophus ferrumequinum</em>), is now available.
		    <LI>
		    Platypus data from regions 
		    where NISC has not yet generated data were provided by 
		    Jim Mullikin from a preliminary assembly of Washington 
		    University WGS reads. 
		    <LI>
		    This freeze includes low-redundancy sequence data from 
		    tenrec, elephant, armadillo, and rabbit. Only one set of 
		    sequences are provided per species/target combination; 
		    where available, NISC data is provided instead of 
		    the 2X assemblies. These data are not yet accessioned at 
		    NCBI, but were made available by the Broad Institute 
		    (rabbit, elephant, armadillo) and Jim Mullikin (tenrec).
		    <LI>
		    Orthology predictions for Fugu were made only by 
		    MAVID/Mercator; predictions for all other assemblies 
		    supported by the UCSC Genome Browser represent a union 
		    with UCSC predictions as well. Because no additional 
		    post-processing was done on the Fugu predictions, they 
		    contain a few very small contigs.
		    </UL>
		    <P>
		    Thanks to the many people, particularly Elliott Margulies
		    and Daryl Thomas, who made this release possible.</P>
		    <P>
		    <HR>

	    <A NAME="052405a"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 24 May 2005 - ChIP-PET/GIS annotation track (Genome Institute of Singapore) released</FONT><BR>
		    <P>
		    The ENCODE Genome Browser now features the ChIP-PET/GIS 
		    annotation track, which shows paired-end 
		    ditag (PET) sequences derived from 65,572 individual p53 
		    ChIP fragments of 5-fluorouracil (5FU) stimulated HCT116 
		    (colon) cells. Only PETs with a single specific mapping 
		    to the genome are included in this track.
		    <P>
		    Thanks to Atif Shahab, Chia-Lin Wei, and Yijun Ruan at the
		    <A HREF="http://www.gis.a-star.edu.sg/internet/site/"
		    TARGET=_blank>Genome Institute of Singapore</A> for 
		    providing the p53 ChIP-PET library and sequence data. The 
		    data were mapped and analyzed by scientists from the 
		    Genome Institute of Singapore, the Bioinformatics 
		    Institute, Singapore, and Boston University. For more
		    information about this annotation, see the ChIP-PET/GIS
		    track description page.</P>
		    <P>
		    <HR>

	    <A NAME="052305"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 23 May 2005 - Boston University First Exon annotation track released</FONT><BR>
		    <P>
		    The First Exon/BU annotation track, contributed by the
		    <A HREF="http://zlab.bu.edu/zlab/index.shtml"
		    TARGET=_blank>ZLAB</A> at Boston University, is now 
		    available in the UCSC Genome Browser. This track displays 
		    expression levels of computationally identified first 
		    exons and a constitutive exon of 20 genes in the ENCODE 
		    regions.</P>
		    <P>
		    For each gene, all alternative first exons were identified
		    based on manual selection of predictions from the 
		    <A HREF="http://biowulf.bu.edu/zlab/Promoser/"
		    TARGET=_blank>PromoSer</A>
		    program. The expression levels of exons were then 
		    quantified using rcPCR in ten normal human tissues.</P>
		    <P>
		    Thanks to Ulas Karaoz and the Zhiping Weng lab at 
		    Boston University for providing these data. For more
		    information about this annotation, as well as a complete
		    list of the individuals who contributed to this track,
		    see the First Exon/BU track description page.</P>
		    <P>
		    <HR>

	    <A NAME="050705"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 7 May 2005 - ENCODE status page now available</FONT><BR>
		    <P>
		    A simple summary page has been added to the UCSC ENCODE
		    portal to show the status of datasets submitted to UCSC by
		    ENCODE contributors. The page may be found at
		    <A HREF="http://genome.ucsc.edu/ENCODE/trackStatus.html">http://genome.ucsc.edu/ENCODE/trackStatus.html</A> and can be accessed via
		    links on the ENCODE home page and the ENCODE data
		    submission page. The status page will be updated 
		    approximately once a week.
		    <P>
		    To review the latest ENCODE data releases, see the
		    <A HREF="releaseLog.html">release log</A>.
		    <P>
		    <HR>

	    <A NAME="050605"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 6 May 2005 - UCSD/Ludwig Institute histone modification ChIp/Chip data released</FONT><BR>
                    <P>  
                    Two additional sets of ChIp/Chip data from UCSD/Ludwig
                    Institute are now available:
<PRE>
    ChIp/LI AcH3     
    ChIp/LI MeH3K4   
</PRE>               
                    These tracks display locations of acetylated H3 and
                    dimethylated K4H3 binding in IMR90 (lung fibroblast) 
		    cells.
                    <P>
                    To consolidate viewing in the browser, the previously 
		    released eight datasets from UCSD/LI have been reformatted
                    as two composite tracks (one track per antibody) with 
		    each track containing four subtracks (one per cell line). 
		    These tracks are:
<PRE>
    ChIp/LI Pol2
    ChIp/LI TAF1
</PRE>
                    To facilitate data analysis, the data were also reloaded
                    in a format that allows extraction of the original data 
		    values via the UCSC table browser.
                    <P>
                    Thanks to Chunxu Qu and the
                    <A HREF="http://licr-renlab.ucsd.edu/" TARGET=_blank>Ren 
		    lab</A> for providing these data.
                    <P>
                    <HR>

	    <A NAME="021705"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 17 Feb. 2005 - ENCODE downloads released</FONT><BR>
   		<P>
		    The ENCODE download area has been reorganized and updated 
		    on our public download server.  The downloads access page 
		    is now:
		    <P>
		    &nbsp; &nbsp; &nbsp; &nbsp;
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/encode/"
		    TARGET=_blank>http://hgdownload.cse.ucsc.edu/goldenPath/encode/</A>
		    <P>
		    and the annotations are now located in the 
		    assembly-specific download directory, currently:
		    <P>
		    &nbsp; &nbsp; &nbsp; &nbsp;
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/"
		    TARGET=_blank>http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/</A>
		    <P>
		    Any web pages referencing the previous UCSC ENCODE 
		    downloads will need to be updated.  Please contact us if 
		    you have any difficulties.
		    <P>
		    <HR>

	    <A NAME="020205"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 2 Feb. 2005 - Affymetrix ChIp/Chip and PolyA RNA data released</FONT><BR>
		    <P>
		    A second set of ENCODE ChIp/Chip data is now available on 
		    the July 2003 human genome assembly:
<PRE>
    ChIp/Affy Pol2 Pval
    ChIp/Affy Pol2 Sites
    RNA/Affy Signal
    RNA/Affy Sites
</PRE>
		    These tracks show RNA Polymerase II precipitation and 
		    RNA abundance in retinoic acid-stimulated HL-60 cells at 
		    0, 2, 8, and 32 hours, as measured by Affymetrix tiling 
		    arrays in the non-repetitive ENCODE regions.  The 
		    <em>Pval</em> and <em>Signal</em> tracks show values 
		    for each tiled probe; the <em>Sites</em> tracks show 
		    contiguous regions of enrichment.
		    <P>
		    A new <em>composite</em> track display was developed to 
		    concisely display multiple data sets of similar types, a 
		    common feature of ENCODE data.  Each of these new tracks 
		    contains 4 subtracks, one for each time interval.  The 
		    subtracks share a single description page and set of 
		    visibility controls.  Checkboxes on the track 
		    configuration page allow selected subtracks to be hidden 
		    in the display.
		    <P>
		    These data were generated and analyzed by Tom Gingeras' 
		    group at 
		    <A HREF="http://www.affymetrix.com"
		    TARGET=_blank>Affymetrix</A> and  
		    <A HREF="http://www.hms.harvard.edu/dms/bbs/fac/struhl.html" 
		    TARGET=_blank>Kevin Struhl's group at Harvard Medical 
		    School</A>. We would like to thank Stefan Bekiranov at 
		    Affymetrix for submitting the data and working closely 
		    with us to clarify the experimental methods and 
		    verification descriptions.
		    <P>
		    <HR>

<!- Begin 2004 news items -->
	    <A NAME="2004"></A>
	    <A NAME="110504"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 5 Nov. 2004 - First ENCODE data tracks in the UCSC Browser</FONT><BR>
		    <P>
		    The first datasets submitted for the ENCODE project are 
		    now publicly available:
		    <UL>
		      <LI>
 		      ChIP-chip and transcription (Ludwig Institute/UCSD)
		      <LI>
		      Temporal profiling of DNA replication (University of 
		      Virginia)
		      <LI>
		      Promoter activity (Stanford)
		      <LI>
		      DNaseI hypersensitive sites (NHGRI)
		    </UL>
		    These tracks are visible in the ENCODE track group of the 
		    July 2003 (hg16) human genome assembly.  We would like to 
		    thank the labs of Bing Ren (LI/UCSD), Anindya Dutta (UVA),
		    Rick Myers (Stanford), and Francis Collins (NHGRI) for 
		    contributing the initial ENCODE data sets.  
		    <P>
                    <HR>

	    <A NAME="102604"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 26 Oct. 2004 - Sequence Freeze For Multiple Alignments</FONT><BR>
		    <P>
		    We are pleased to release the first "official" sequence 
		    data freeze for the ENCODE multiple sequence alignment 
		    projects. The data formats are described in the 
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/alignments/OCT-2004/README.txt" 
		    TARGET=_blank>README</A> file, and the sequences and 
		    supporting information is collected in the 
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/alignments/OCT-2004" 
		    TARGET=_blank> data directory</A>.
		    <P>
		    The species included in this freeze are as follows:
<PRE>
       _SPECIES_      _SOURCE_
        Human           hg16
        Chimpanzee      panTro1
        Dog             canFam1
        Rat             rn3
        RatB	        BCM
        Mouse           mm5
        Chicken		galGal2
        Galago		NISC
        Baboon		NISC
        Marmoset        NISC
        Armadillo       NISC
        Platypus        NISC
</PRE>
		    You will notice that we have worked hard to include a 
		    number of species for which genome-wide sequence data was 
		    already available. The process by which these orthologous 
		    regions were identified is still an area of active 
		    research development, the details of which will be 
		    presented at the upcoming ENCODE meeting at CSHL.
		    <P>
		    Please note that we have also included a *second* rat 
		    sequence (ratB) in this freeze.  RatB represents an 
		    initiative to standardize the quality level of sequences 
		    in the ENCODE regions for species with genome-wide 
		    sequence data. The data were made available just before 
		    our freeze date, so we decided to include both versions of
		    the rat sequence for now. Eventually these sequences will 
		    likely be rolled into future genome assemblies.
		    <P>
		    Remember, this is a work in progress, so not all targets 
		    have sequence from all species. And some species/target 
		    combinations may not be complete yet. Progress on the 
		    NISC-generated sequences can be found at the 
		    <A HREF="http://www.nisc.nih.gov/open_page.cgi?path=/projects/encode/index.cgi"
		    TARGET=_blank> NISC ENCODE Project: Comparative 
		    Sequencing</A>.
		    <P>
		    Many people have worked very hard to make these data 
		    available. Special thanks to Daryl Thomas, Kate 
		    Rosenbloom and the entire UCSC Team; Greg Schuler and the 
		    NCBI team; Colin Dewey and Lior Pachter; Pam Thomas and 
		    the NISC team; and David Wheeler and the BCM team.
		    <P>
		    Please send any comments to the 
		    <A HREF="mailto: <ENCODE_MSA-L@LIST.NIH.GOV>" 
		    TARGET=_blank>ENCODE MSA Mailing List</A>.
		    <P>
		    &nbsp; &nbsp; Elliott
		    <BR>
		    -- 
		    <BR>
		    Elliott H. Margulies, Ph.D.
		    Genome Technology Branch, NHGRI
		    <P>
                    <HR>

	    <A NAME="062404"></A>
  	      <FONT SIZE=4 COLOR="#006666">Posted on 24 Jun. 2004 - ENCODE Project Portal Released</FONT><BR>
		    <P>
		    We are proud to announce the release of features
		    in the UCSC Genome Browser that are tailored to
		    the ENCODE project community, including this home
		    page to consolidate these resources.
		    <P>
		    The initial resources include sequences for the current 
		    human assemblies (hg16, hg15, hg13, and hg12), sequence of
		    the comparative species from 
		    <A HREF="http://www.nisc.nih.gov/" TARGET=_blank>NISC</A>,
		    tools for coordinate conversion between human
		    assemblies, format descriptions for data
		    submission, and contact information for help with
		    submitting annotation data and analyses.
		    <P>
		    Bulk downloads of the sequence and annotations may
		    be obtained from the ENCODE Project 
		    <!--<A HREF="ftp://genome-test.cse.ucsc.edu/encode/downloads/">FTP</A> or -->
		    <A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/encode/">Downloads</A> 
		    page.  The sequences available here are repeat-masked 
		    versions of the GenBank records.
	    <BR><BR>
	  </TD><TD WIDTH=15></TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>
