<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="ENCODE Project at UCSC">
    <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
    <TITLE>ENCODE Pilot Project at UCSC</TITLE>
    <LINK REL="STYLESHEET" HREF="../style/HGStyle.css" TYPE="text/css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
  <A NAME="TOP"></A>

    <!-- Header Images and links -->
    <TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
      <TR><TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
          <A HREF="http://www.genome.gov/10005107" TARGET=_blank><IMG SRC="/images/encodelogo.gif" height=50 ALT="ENCODE Project at NHGRI"></A>&nbsp;&nbsp;
      <!-- TODO: Replace with image -->
        <A HREF="/encode/"><FONT SIZE=5 COLOR="gold">ENCODE Pilot Project</FONT>&nbsp;<FONT SIZE=5 COLOR="blue">at UCSC</A>
      </TH></TR>

      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.topbar_pilot.html"--></TD></TR>

      <TR HEIGHT="100%" VALIGN="top">
        <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
        <!-- outer table is to align leftbar & content boxes at top of page -->
          <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
            <TR><TD><IMG SRC="../images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
            <TR HEIGHT="100%"><TD HEIGHT="100%">
            <!--outer table is for black border-->
              <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
                <TR><TD>
                  <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                    <TR HEIGHT="100%">
	              <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%"> 
                	<A CLASS="leftbar" HREF="../cgi-bin/hgGateway">
		   	Genome Browser</A><HR>
	                <A CLASS="leftbar" HREF="encode.hg18.html">
		    	Regions (hg18)</A><HR>
	        	<A CLASS="leftbar" HREF="encode.hg17.html">
		   	Regions (hg17)</A><HR>
                	<A CLASS="leftbar" HREF="trackStatus.html">
                    	Released Data</a> <HR>
	        	<A CLASS="leftbar" HREF="http://hgdownload.cse.ucsc.edu/goldenPath/encode/index.html">
		   	Downloads</A><HR>
			<A class="leftbar" HREF="contributors_pilot.html">
		   	Contributors</a>
			<HR>
	        	<A CLASS="leftbar" HREF="terms_pilot.html">
		   	Terms of Use</a><HR>
<!--
			<BR>&nbsp<HR>
                	<A CLASS="leftbar" HREF="../cgi-bin/hgGateway">
		   	Genome Browser</A><HR>
-->
 	        	<BR>
	      		<IMG SRC="../images/DOT.gif" ALT="" HEIGHT=970 WIDTH=1>
   	    	      </TD></TR>
		    </TABLE>
              </TD></TR>
	    </TABLE>
      </TD></TR>
    </TABLE>
  </TD>

<!--Content Tables-------------------->  
  <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0>
      <TR><TD>
        <!--outer table is for border purposes-->
          <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
                <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                  <TR><TD>	
	              <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
			<TR><TD>
	                    <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; 
		            About the UCSC ENCODE pilot phase repository</B></FONT>	        
		        </TD></TR>
		      </TABLE>
	              <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                        <TR><TH HEIGHT=3></TH></TR>
                        <TR><TD WIDTH=10></TD><TD>
			  <P>
		  	  The University of California Santa Cruz 
		          (<A HREF="http://www.ucsc.edu/public/" 
		    	  TARGET=_blank>UCSC</A>) manages the official 
			  repository of sequence-related data submitted during 
			  the pilot phase (2003-2007) of the 
		 	  <A HREF="http://www.genome.gov/10005107" 
		    	  TARGET=_blank>ENCODE</A> project at 
		  	  <A HREF="http://www.genome.gov/" 
			  TARGET=_blank>NHGRI</A>. Data analysis during the
			  pilot phase was coordinated by the
		      	  <A HREF="http://www.ensembl.org/Homo_sapiens/encode.html" 
		    	  TARGET=_blank>Ensembl</A> group, a joint project of
		          EBI and the Wellcome Trust Sanger Institute. </P>
		          <!--
		    	  <A HREF="about.html">Read more</A>. </P> 
		    	  -->
                          <P>
			  The data collected during the pilot phase can be 
			  accessed from the links on this page. To view the 
			  ENCODE data on 
			  the NCBI Build 36 human assembly (March 2006), click 
			  the <A HREF="encode.hg18.html">Regions (hg18)</A> 
			  link. 
			  To view data on the previous ENCODE reference assembly
		 	  (May 2004, hg17, Build 35) that was used as the basis 
			  for the June 2007 publications in <em>Nature</em> and 
			  <em>Genome Research</em>, click the 
			  <A HREF="encode.hg17.html">Regions (hg17)</A> link.
			  </P>
		          <P>
                          ENCODE data submissions released by UCSC during the
			  pilot phase are listed on the 
                          <A HREF="trackStatus.html">released data 
		          page</A>. Primary data for ENCODE is available from 
			  the 
                          <A HREF="http://www.ncbi.nlm.nih.gov/projects/geo/info/ENCODE.html"
		          TARGET=_blank>NCBI GEO</A> and 
		          <A HREF="http://www.ebi.ac.uk/arrayexpress/"
		          TARGET=_blank>EBI ArrayExpress</A> public array data 
		          repositories. Ensembl provides an 
		          <A HREF="http://www.ensembl.org/Homo_sapiens/encode.html"
                          TARGET=_blank>ENCODE resource page</A>, and NHGRI 
			  provides the 
			  <A HREF="http://research.nhgri.nih.gov/ENCODEdb"
                          TARGET=_blank>ENCODEdb</A> portal.</P>
		          <P>
		          We thank NHGRI and those who have contributed 
			  annotations and analyses to this project. This portal 
			  is maintained by the UCSC Genome Bioinformatics Group,
			  a cross-departmental team within the Center for 
			  Biomolecular Science and Engineering 
			  (<A HREF="http://www.cbse.ucsc.edu/" 
		          TARGET=_blank>CBSE</A>) at UCSC. Click the 
			  <A HREF="contributors_pilot.html">Contributors</A> link for
			  a complete list of ENCODE pilot phase data providers and the
			  UCSC staff who develop and maintain this website.</P>
<!--
			  The individuals who 
		          developed and maintained this website include
                          <A HREF="mailto:&#107;&#97;t&#101;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u"
		          TARGET=_blank>Kate Rosenbloom</A>, 
                          <A HREF="mailto:&#104;&#97;rt&#101;r&#97;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u"
		          TARGET=_blank>Rachel Harte</A>, 
                          <A HREF="mailto:t&#105;n&#103;w&#97;n&#103;&#64;&#115;&#111;e.&#117;&#99;&#115;&#99;.&#101;&#100;u"
                          TARGET=_blank>Ting Wang</A>, 
                          <A HREF="http://www.soe.ucsc.edu/~kent/"
		          TARGET=_blank>Jim Kent</A>, and the 
		          <A HREF="../staff.html">UCSC Genome Bioinformatics 
		          staff</A>, with input from
                          <A HREF="http://www.soe.ucsc.edu/~daryl" 
		          TARGET=_blank>Daryl Thomas</A>.</P>
-->
		          <P>
 		          The ENCODE project has now expanded to cover the full 
			  human genome. UCSC is the official 
                          <A HREF="index.html">Data Coordination Center</A>
                          for this endeavor. The new data 
			  are accessible on the most recent human assemblies in 
			  the <A HREF="../cgi-bin/hgGateway?org=human">Genome 
			  Browser</A>, along with the sequence and 
			  annotation data for a large collection of genome 
			  assemblies. </P>
	              </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>

        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
              <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	            <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%" BORDER=0>
                      <TR><TD>
		      <!--cheap trick to get background on row to be continuous-->
		          <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                            <TR><TD>
  		            <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; About the ENCODE project pilot phase</B></FONT>
		      	    </TD></TR>
		          </TABLE>
	              </TD></TR>
		    </TABLE>

		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
    		  <TR><TH HEIGHT=3></TH></TR>
    		  <TR><TD WIDTH=10></TD><TD>
                          <P> Following the release of the completed  human 
		  	  genome sequence in April 2003, the scientific 
			  community intensified its  efforts to mine the data 
			  for clues about how the body works in health and in 
			  disease. A basic requirement for this understanding of
			  human biology is the ability to identify and 
			  characterize sequence-based functional elements 
			  through experimentation and computational analysis. 
			  In September 2003, the  NHGRI introduced the ENCODE 
			  project to facilitate the identification and analysis 
			  of the complete set of functional elements in the 
			  human genome sequence. During the initial pilot and 
			  technology development phases of the project, 44 
			  regions&mdash;approximately 1% of the human 
			  genome&mdash;were targeted for analysis using a 
			  variety of experimental and computational methods with
		 	  the aim of assembling a comprehensive encyclopedia of 
			  the functional elements in these regions, showing 
			  their identity and precise location. The pilot project
			  established protocols for scaling up to full-genome 
			  coverage and  produced a wealth of data, elucidating 
			  elements such as protein-coding genes, transcription 
			  units, protein binding sites, conserved DNA elements, 
			  features of chromatin assembly and modification, and 
			  single nucleotide polymorphisms. </P>                
                          <P>
			  During the pilot phase, UCSC collected, processed, 
			  and released more than 500 ENCODE data sets 
			  representing a broad range of experimental methods and
			  diverse tissues and cell lines. In addition to the two
			  designated ENCODE cell lines, HeLa cervical carcinoma 
			  and GM06990 lymphoblastoid, more than 40 cell types 
			  are represented. A substantial proportion of the data 
			  is the product of chromatin immunoprecipitation 
			  (ChIP-CHIP) experiments used to determine binding 
			  sites for transcription factors&mdash;eight groups 
			  have produced ChIP/CHIP data from four microarray 
			  platforms, investigating more than two dozen 
			  transcription  factors and histone modifications. 
			  Several experimental groups have provided time course 
			  data and varied cell treatments. Other notable 
			  experimental data include localization of RNA 
			  transcription starts, identification of regions of 
			  DNaseI hypersensitivity, and temporal profiling of DNA
			  replication.</P>
                          <P>
			  Accompanying the ENCODE experimental data, UCSC also 
			  hosts the ENCODE high-quality gene set, provided by 
			  the Gencode project, and a  variety of computationally
			  derived annotations, including gene predictions from  
			  the ENCODE Gene Annotation Assessment Project (EGASP),
			  pseudogene annotations from four projects, and RNA 
			  secondary structure predictions from two contributors.
			  The comparative 
			  genomics tracks include multiple alignments of 28 
			  vertebrate species in the ENCODE regions, produced 
			  with three sequence  alignment methods and four 
			  different conservation algorithms. The 
			  Genome  Browser provides a full set of genome-wide 
			  comparative genomics tracks that complement the 
			  ENCODE tracks, including a genome-wide multiple 
			  alignment covering nearly 30 vertebrate species. </P>
			  <P>
			  You can find more information about the ENCODE pilot 
			  phase at UCSC in the <A HREF="newsarch.html">news 
			  archives</A>.</P>
		    <P></P>
    	          </TD><TD WIDTH=15></TD></TR>
	        </TABLE>
	      </TD></TR>
	    </TABLE>
          </TD></TR>
        </TABLE>
	<BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
		  <TR><TD>
	            <A NAME="Conditions"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	          </TD></TR>
		</TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The sequence and annotation data displayed in the Genome 
		    Browser are freely available for academic, nonprofit, and 
		    personal use with the following conditions:
		    <UL>
		      <LI>
		      The general 
		      <A HREF="http://genome.ucsc.edu/#Conditions">Conditions of Use</A> 
		      for the UCSC Genome Browser apply.
		      <LI>
                      The 
		      <A HREF="terms_pilot.html">
                      ENCODE Consortium Data Release Policy (2003-2007)</A>
                      applies to all pilot phase ENCODE data.
		    </UL>
	          </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      </TD></TR>
	    </TABLE>
          </TD></TR>
	</TABLE>
      <BR></TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>

