<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html>
    <head>
        <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
        <title>Quality Metrics</title>
        <link rel="stylesheet" href="/style/HGStyle.css" type="text/css">
        <link rel="stylesheet" href="/style/encodeProject.css" type="text/css">
    </head>

    <body>
        <!--#include virtual="/inc/encodeProject.topbar.html"-->

        <div class="encodeHeader">
            <a href="http://www.genome.gov/10005107"><img src="/images/ENCODE_scaleup_logo.png" alt="ENCODE Project at NHGRI" title="ENCODE Project at NHGRI"></a>
            <span class="txt">Quality Metrics</span>
        </div>
       <div class="wrapper">
            <div class="bar"><h4 class="title">Contents</h4></div>
            <div class="content">
                <!--CONTENT GOES HERE-->

                <p>The ENCODE consortium analyzes the quality of the data produced using a 
                variety of metrics.  This page includes spreadsheets showing quality metrics for
                many ENCODE datasets, along with descriptions of what these 
                metrics are, and what they appear to measure.  
                These quality metrics will be updated on occasion to include analysis 
                of more recent data.
               <ul>
                   <li><a href='#spreadsheets'>Spreadsheets of quality metrics for datasets</a></li>
                   <li><a href='#definitions'>Definitions of quality metrics</a></li>
                   <li><a href='#chipSeq'>Definitions of ChIP-seq specific metrics</a></li>
               </ul>
                </p>
                <p>It is important to note that quality metrics for evaluating epigenomic 
                assays is an area of research, so standards are emerging as more metrics 
                are used with more datasets and types of experiments.  The typical values for 
                a quality metric can be quite different with different experiment types, or 
                even comparing different features in the same experiment type, such as different 
                antibodies used in ChIP-seq experiments.  Currently there is no single 
                measurement that identifies all high-quality or low-quality samples.  As with 
                quality control for other types of experiments, multiple assessments (including 
                manually inspection of tracks) are useful because they may capture different 
                concerns.  Comparison within an experimental method can be very helpful in 
                identifying stochastic error in experiments (i.e. comparing replicates to 
                each other, or comparing values for one antibody in several cell types, or the same 
                antibody and cell type in different labs).
                </p>
                <p>Many of the software tools used for quality metrics, along with their citations, 
                can be found on the <a target=_blank href="softwareTools.html">Software Tools</a> page.
                Other documents related to data quality are the <a target=_blank href="dataStandards">Data Standards</a> page
                and the antibody validation documents accessed from the <a target=blank href="antibodies.html">Antibodies</a> page.
                For questions about the ENCODE Quality Metrics, contact
                <a target=_blank href="http://www.genome.gov/27545735">Mike Pazin, NHGRI</a>.</p>
                </p>
                <p class="date">Updated 18 October 2012</p>
            </div><!--end content-->
        </div><!--end wrapper-->

        <div class="wrapper">
            <div class="bar">
                <a name="spreadsheets"></a> 
                <h4 class="title">Quality metric spreadsheets</h4>
            </div>
            <div class="content">

                    <p>Datasets are divided into DNase-seq, FAIRE-seq, TF ChIP-seq, Histone ChIP-seq, 
                    and ChIP Controls.  The ReadMe worksheet provides a summary description of the metrics (described in more detail below).
                    The <em>Assays</em>, <em>Cells</em>, and <em>Treatments</em> are defined in the 
                    <a target=_blank href="otherTerms.html">ENCODE Controlled Vocabulary</a> 
                    of registered terms. The <em>Identifiers</em> reference the dataset at UCSC 
                    (via the Table Browser, metaDb table) and the ENCODE analysis site (filenames).
                    </p>
                    <p>
                    <b><em>Download spreadsheets in:</em></b>&nbsp;&nbsp;
                  <a href='https://docs.google.com/spreadsheet/pub?key=0AvQL5qBL6AfEdDVaa1FNZWZSdE8zX1hwT0RiTWR0M0E&output=xls'>
                      Excel format (.xls)</a>&nbsp;&nbsp;
                  <a href='https://docs.google.com/spreadsheet/pub?key=0AvQL5qBL6AfEdDVaa1FNZWZSdE8zX1hwT0RiTWR0M0E&output=ods'>
                      OpenOffice spreadsheet format (.ods)</a>&nbsp;&nbsp;
                    </p>
                    <p><iframe width='1450' height='800' frameborder='0' 
                        src='https://docs.google.com/spreadsheet/pub?key=0AvQL5qBL6AfEdDVaa1FNZWZSdE8zX1hwT0RiTWR0M0E&output=html&widget=true'></iframe>
                    </p>                	
            </div><!--end content-->
        </div><!--end wrapper-->

        <div class="wrapper">
            <div class="bar">
                <a name="definitions"></a> 
                <h4 class="title">Definitions of quality metrics</h4>
            </div>
            <div class="content">

                	<p><b>Uniquely mappable reads (N_uniq map reads):</b><br>
                	The count of the number of sequence reads for this sample that can be 
                	aligned to a single genomic location; this does not distinguish between reads 
                	that were obtained multiple times (redundant reads) and reads obtained only once 
                	(non-redundant reads).  A larger number of reads from a sufficiently complex 
                	library increases the chances of finding all true binding sites; however, the 
                	number of reads required is not known with certainty, and likely depends on 
                	enrichment, antibody quality in ChIP experiments, and the fraction of the genome 
                	containing the feature being measured.
                	</p>

                	<p><b>Self-consistent peaks, IDR n (Self Cons IDR):</b><br>
                	An estimate of the number of enriched regions in a single sample.  A dataset 
                	is divided into 2 pseudo-replicates that are analyzed by peak-calling at relaxed 
                	stringency followed by IDR filtering at the indicated IDR threshold.
                	</p>

                        <p><b>Replicate-consistent peaks, IDR n (Rep Cons IDR):</b><br>
                        The number of enriched regions, determined using IDR (Irreproducible Discovery Rate) 
                        using this sample and a replicate.  Potential enriched regions are identified using a 
                        peak caller at very low stringency, then the IDR method is used to determine which peaks 
                        are signal and which are noise, at the indicated IDR threshold.  As this analysis is 
                        performed using pairs of datasets, the output number of peaks is identical for these two 
                        datasets using this method.
                        </p>

                        <p><b>Signal Portion of Tags (SPOT):</b><br>
                        A measure of enrichment, analogous to the commonly used 
                        fraction of reads in peaks metric.  SPOT calculates the fraction of reads that fall in 
                        tag-enriched regions identified using the Hotspot program, (Hotspot and SPOT are described 
                        on the ENCODE Software Tools page) from a sample of 10 million reads.  Note that because methods of 
                        measuring enrichment based on determining the fraction of reads that fall in peaks are 
                        sensitive to the determination of enriched regions, comparison is possible only when using 
                        the identical peak caller and parameters.  Larger SPOT values indicate higher signal to 
                        noise; 1.0 is the maximum possible value (all reads are signal) and 0 is the minimum possible 
                        value (all reads are noise).  For FAIRE, more than 10 million reads are typically required to 
                        reliably detect peaks.
                        </p>

                        <p><b>PCR Bottleneck Coefficient (PBC):</b><br>
                        A measure of library complexity, i.e. how skewed the distrubition of read counts per location
                        is towards 1 read per location.
                        </p>
                        <p>
                        PBC = N1/Nd
                        </p>
                        <p>
                        (where N1= number of genomic locations to which EXACTLY one unique mapping read maps,
                        and Nd = the number of genomic locations to which AT LEAST one unique mapping read maps,
                        i.e. the number of non-redunant, unique mapping reads).</p>
                        <p>
                        PBC is further described on the ENCODE Software Tools page.  Provisionally, 0-0.5 is severe bottlenecking, 0.5-0.8 
                        is moderate bottlenecking, 0.8-0.9 is mild bottlenecking, while 0.9-1.0 is no bottlenecking.  
                        Very low values can indicate a technical problem, such as PCR bias, or a biological finding, 
                        such as a very rare genomic feature.  Nuclease-based assays (DNase, MNase) detecting features 
                        with base-pair resolution (transcription factor footprints, positioned nucleosomes) are 
                        expected to recover the same read multiple times, resulting in a lower PBC score for these 
                        assays. Note that the most complex library, random DNA, would approach 1.0, thus the very 
                        highest values can indicate technical problems with libraries.  It is the practice for some 
                        labs outside of ENCODE to remove redundant reads; after this has been done, the value for this 
                        metric is 1.0, and this metric is not meaningful.  82% of TF ChIP, 89% of His ChIP, 77% of 
                        DNase, 98% of FAIRE, and 97% of control ENCODE datasets have no or mild bottlenecking.
                        </p>

                        <p><b>Normalized Strand Cross-correlation coefficient (NSC):</b><br>
                        A measure of enrichment derived without dependence on prior determination of enriched regions.
                        Forward and reverse strand read coverage signal tracks are computed (number of unique mapping
                        read starts at each base in the genome on the + and - strand counted separately). The forward
                        and reverse tracks are shifted towards and away from each other by incremental distances and
                        for each shift, the Pearson correlation coefficient is computed. In this way, a cross-correlation
                        profile is computed, representing the correlation between forward and reverse strand coverage at
                        different shifts. The highest cross-correlation value is obtained at a strand shift equal to the
                        predominant fragment length in the dataset as a result of clustering/enrichment of relative
                        fixed-size fragments around the binding sites of the target factor or feature. 
                        </p>

                        <p>
                        The NSC is the ratio of the maximal cross-correlation value (which occurs at strand shift equal
                        to fragment length) divided by the background cross-correlation (minimum cross-correlation value
                        over all possible strand shifts). Higher values indicate more enrichment, values less than 1.1 are
                        relatively low NSC scores, and the minimum possible value is 1 (no enrichment). This score is
                        sensitive to technical effects; for example, high-quality antibodies such as H3K4me3 and CTCF score
                        well for all cell types and ENCODE production groups, and variation in enrichment in particular IPs
                        is detected as stochastic variation. This score is also sensitive to biological effects; narrow marks
                        score higher than broad marks (H3K4me3 vs H3K36me3, H3K27me3) for all cell types and ENCODE production
                        groups, and features present in some individual cells, but not others, in a population are expected
                        to have lower scores.
                        </p>

                        <p><b>Relative Strand Cross-correlation coefficient (RSC):</b><br>
                        A measure of enrichment derived without dependence on prior determination of enriched regions.
                        Forward and reverse strand read coverage signal tracks are computed (number of unique mapping read
                        starts at each base in the genome on the + and - strand counted separately). The forward and reverse
                        tracks are shifted towards and away from each other by incremental distances and for each shift, the
                        Pearson correlation coefficient is computed. In this way, a cross-correlation profile is computed
                        representing the correlation values between forward and reverse strand coverage at different shifts.
                        The highest cross-correlation value is obtained at a strand shift equal to the predominant fragment
                        length in the dataset as a result of clustering/enrichment of relative fixed-size fragments around
                        the binding sites of the target factor. For short-read datasets (&lt; 100 bp reads) and large genomes
                        with a significant number of non-uniquely mappable positions (e.g., human and mouse), a
                        cross-correlation phantom-peak is also observed at a strand-shift equal to the read length. This
                        read-length  peak is an effect of the variable and dispersed mappability of positions across the
                        genome. For a significantly enriched dataset, the fragment length cross-correlation peak (representing
                        clustering of fragments around target sites) should be larger than the mappability-based read-length peak. 
                        </p>

                        <p>
                        The RSC is the ratio of the fragment-length cross-correlation value minus the background
                        cross-correlation value, divided by the phantom-peak cross-correlation value minus the background
                        cross-correlation value. The minimum possible value is 0 (no signal), highly enriched experiments have
                        values greater than 1, and values much less than 1 may indicate low quality.
                        </p>

            </div><!--end content-->
        </div><!--end wrapper-->

        <div class="wrapper">
            <div class="bar">
                <a name="chipSeq"></a> 
                <h4 class="title">Definitions of ChIP-seq specific quality metrics</h4>
            </div>
            <div class="content">
                        <p><b>MACS FDR 0.01:</b><br>
                        This is the number of enriched regions identified by MACS using an FDR threshold of 0.01 (1%).</p>

                        <p><b>Under seq:</b><br>
                        If set to 1, this means it was manually annotated that this dataset is likely to be 
                        undersequenced.
                        </p>
                        
                        <p><b>Diff rep:</b><br>
                        If set to 1, it means this row is a replicate that is different from the other replicates (based 
                        on self-consistency, NSC, or RSC).  Therefore, this sample should not be used for replicate-based 
                        comparisons such as IDR.
                        </p>
                        
                        <p><b>Manual low S/N:</b><br>
                        If set to 1 this means it was manually annotated that the data has low signal to noise.  This 
                        could be the result of under-sequencing, poor enrichment during ChIP, poor antibody quality, or 
                        the biological nature of the feature being examined.
                        </p>
                        
                        <p><b>Auto low S/N:</b><br>
                        If set to 1 this means the data has low signal to noise, scored by NSC &lt; 1.09 and RSC &lt; 0.9.  
                        This could be the result of under-sequencing, poor enrichment during ChIP, poor antibody quality, 
                        or the biological nature of the feature being examined.
                        </p>
                        
                        <p><b>Revoke flag:</b><br>
                        These datasets have been revoked by the production lab.  R = revoked dataset, D = duplicate good 
                        dataset.
                        </p>
                        


            </div><!--end content-->
       </div><!--end wrapper-->
    </body>
</html>
