<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>

	
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META http-equiv="Content-Script-Type" content="text/javascript">
	<META HTTP-EQUIV="Pragma" CONTENT="no-cache">
	<META HTTP-EQUIV="Expires" CONTENT="-1">
	<TITLE>
Track and Table Releases	</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css" TYPE="text/css">

</HEAD>
<BODY BGCOLOR="#FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">

<TR><TH COLSPAN=1 ALIGN="left">
<A HREF="http://www.genome.gov/10005107" TARGET="_BLANK"><IMG SRC="../images/encodelogo.gif" height=50 ALT="ENCODE Project at NHGRI"></A>
<IMG SRC="../images/encode.jpg" ALT="ENCODE Project at UCSC">
</TH></TR>


<!-- +++++++++++++++++++++ HOTLINKS BAR +++++++++++++++++++ -->
<TR><TD COLSPAN=3 HEIGHT=40 >
<table bgcolor="#000000" cellpadding="1" cellspacing="1" width="100%%" height="27">
<tr bgcolor="#2636D1"><td valign="middle">
	<table BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="#2636D1" height="24"><TR>
	 	<TD VALIGN="middle"><font color="#89A1DE">&nbsp;

&nbsp;<A HREF="../encode/" class="topbar">
           Home</A>
       <A HREF="../cgi-bin/hgTables?hgta_doMainPage=1" class="topbar">
           Tables</A> &nbsp;&nbsp;&nbsp;
       <A HREF="../cgi-bin/hgPcr" class="topbar">
           PCR</A> &nbsp;&nbsp;&nbsp;
<A HREF="../cgi-bin/hgSession?hgS_doMainPage=1" class="topbar">Session</A>&nbsp;&nbsp;&nbsp;
       <A HREF="../FAQ/" class="topbar">
           FAQ</A> &nbsp;&nbsp;&nbsp;

       <A HREF="../goldenPath/help/hgTracksHelp.html"
       class="topbar">
           Help</A> 
&nbsp;</font></TD>
       </TR></TABLE>
</TD></TR></TABLE>
</TD></TR>	


<!-- +++++++++++++++++++++ CONTENT TABLES +++++++++++++++++++ -->
<TR><TD COLSPAN=3>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp;
Track and Table Releases</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
	

 This page contains track and table release information for the following genome assemblies:<br>
<ul>
<li><a CLASS="toc" HREF="#recent" ><b>10 Latest Changes (all assemblies)</b></a></li><li><a CLASS="toc" HREF="#hg18">Human Mar. 2006 (hg18)</a></li>
<li><a CLASS="toc" HREF="#hg17">Human May 2004 (hg17)</a></li>
<li><a CLASS="toc" HREF="#hg16">Human July 2003 (hg16)</a></li>
</ul>
<p>
 For more information about the tracks and tables listed on this page, refer to the <a href=/goldenPath/gbdDescriptions.html>Description of the annotation database</a> and the <a href=/goldenPath/help/hgTracksHelp.html#IndivTracks>User's Guide</a>.<p>
<em>Last updated 07 Aug 2008. <a HREF="mailto:genome@soe.ucsc.edu">Inquiries and feedback welcome</a>.</em>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=recent></A> 10 Latest Changes (all assemblies)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR>
<TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
<TD nowrap><FONT color="#006666"><B>Assembly</B></FONT></TD>
<TD nowrap><FONT color="#006666"><B>Release Date</B></FONT></TD>
</TR>
<TR valign=top><TD align=left>
Affymetrix ENCODE Extension Transcription Sites & Signal</td>
<td>hg17,hg18</td>
<td>16 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI DIPs (lift from hg17)</td>
<td>hg18</td>
<td>23 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE TBA SCONE Conservation (Update)</td>
<td>hg17</td>
<td>19 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Whole-Genome ENCODE LI/UCSD TAF1 Super Track</td>
<td>hg18</td>
<td>17 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE data migration from hg17</td>
<td>hg18</td>
<td>08 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UC Davis ChIP-chip NimbleGen (E2F1, c-Myc, TAF, POLII)</td>
<td>hg17</td>
<td>24 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sites - Jun07 batch</td>
<td>hg17</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sig - Jun07 batch</td>
<td>hg17</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Pval - Jun07 batch</td>
<td>hg17</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sites - Apr07 batch</td>
<td>hg17</td>
<td>13 Jul 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=hg18></A>Human Mar. 2006 (hg18, NCBI Build 36.1)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Affymetrix ENCODE Extension Transcription Sites & Signal</td>
<td>16 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI DIPs (lift from hg17)</td>
<td>23 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Whole-Genome ENCODE LI/UCSD TAF1 Super Track</td>
<td>17 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE data migration from hg17</td>
<td>08 Oct 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=hg17></A>Human May 2004 (hg17, NCBI Build 35)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
Affymetrix ENCODE Extension Transcription Sites & Signal</td>
<td>16 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
ENCODE TBA SCONE Conservation (Update)</td>
<td>19 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UC Davis ChIP-chip NimbleGen (E2F1, c-Myc, TAF, POLII)</td>
<td>24 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sites - Jun07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sig - Jun07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Pval - Jun07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sites - Apr07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sig - Apr07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Pval - Apr07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sites - Update to Jan07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sig - Jan07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Pval - Jan07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sites Update to Nov06 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sig - Nov06 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE University of Virginia DNA Replication Origins (data update)</td>
<td>30 May 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Duke Dnase/chip - Feb07 batch (data addition)</td>
<td>21 May 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gencode RACEfrags</td>
<td>18 May 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gencode Gene Annotations (March 2007)</td>
<td>04 May 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UCD Ng ChIP Hits (update of cMyc and E2F1 data)</td>
<td>17 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale RFBR (Regulatory Factor Binding Regions)</td>
<td>21 Feb 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: UCSC EvoFold (Encode predictions analyzed in Washietl, Pedersen, et al., 2007).  Replaces TBA23 EvoFold data. </td>
<td>13 Feb 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE - University of Vienna, RNA secondary structure predicted by RNAz</td>
<td>16 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale RNA (update with October 2005 freeze data)</td>
<td>10 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UW/Reg QCP DNaseI Sens (replaces the following three tracks: UW/Reg DNaseI HSs,  UW/Reg Q/A,  UW DNase GM)</td>
<td>07 Dec 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Duke/NHGRI DNase (New Chip method in HepG2, Hela, GM, and CD4+)</td>
<td>05 Dec 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UT-Austin STAGE (update to add STAT1 tags)</td>
<td>29 Nov 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE GIS Chip-PET (P53 in HCT116, STAT1 in HeLa, c-Myc in P493)</td>
<td>07 Nov 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI/PSU/UCSC TBA Conserved Elements</td>
<td>05 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI/PSU/UCSC TBA Conservation</td>
<td>05 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI/PSU TBA Alignments</td>
<td>05 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE MSA Consensus Constrained Elements</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UC Berkeley  MAVID Alignments</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UC Berkeley MAVID Conservation</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UC Berkeley MAVID Conserved Elements</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford MLAGAN Conserved Elements</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford MLAGAN Conservation</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford MLAGAN Alignments</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE MSA Alignment Gaps</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE MSA Alignment Agreement</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UNC FAIRE track (data update)</td>
<td>21 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
8 new subtracks for ENCODE Sanger ChIP-chip track</td>
<td>17 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE University of Uppsala, Sweden ChIP-chip Na-butyrate time series</td>
<td>26 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UNC FAIRE Peaks (ChIPOTle) Apr. 2006 Update, new subtrack added to UNC FAIRE</td>
<td>15 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UW DNase/Array GM06990</td>
<td>13 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Sanger ChIP (additional data in HFL-1 cell line)</td>
<td>24 May 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Detection of LinEage Specific Selection (DLESS)</td>
<td>19 May 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP Sites (data replaced with 5 new transcription factors). Previous track Yale ChIP Sites renamed Yale STAT1 Sites.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP Signal (data replaced with 5 new transcription factors). Previous track Yale ChIP Signal renamed Yale STAT1 Signal.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP pVal (data replaced with 5 new transcription factors). Previous track Yale ChIP pVal renamed Yale STAT1 pVal.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE EGASP Full (Correction)</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE EGASP Partial (data correction)</td>
<td>11 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Recomb Hotspots: ENCODE recombination hotspots and rates</td>
<td>06 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Uppsala Chip (University of Uppsala, Sweden ChIP-chip coordinate corrections)</td>
<td>03 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale TAR (Replace June data with new methods)</td>
<td>02 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UVa DNA Rep Ori (University of Virginia DNA Replication Origins, Oct 2005 Freeze)</td>
<td>29 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UT-Austin STAGE (Sequence Tag Analysis of Genomic Enrichment), c-Myc in HeLa cells</td>
<td>28 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy Transfrags (updated HeLa data)</td>
<td>21 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy RNA Signal (updated HeLa data)</td>
<td>21 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UVa DNA Rep TR50 (University of Virginia DNA Smoothed Timing at 50% Replication, Oct 2005 Freeze)</td>
<td>16 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UVa DNA Rep Seg (University of Virginia DNA Replication Temporal Segmentation, Oct 2005 Freeze)</td>
<td>02 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Sanger ChIP Hits (Peaks and Centers, Oct 2005 Freeze)</td>
<td>22 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Sanger ChIP (H3, H4 histones in GM, K562 and HeLa cells, Oct 2005 Freeze)</td>
<td>22 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
 ENCODE UNC FAIRE (Formaldehyde Assisted Isolation of Regulatory Elements, Oct 2005 Freeze)
</td>
<td>22 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UCD Ng ChIP (UC Davis ChIP-chip of E2F1, Myc in HeLa, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Uppsala Chip (University of Uppsala, Sweden ChIP-chip of H3ac,HNF3b,HNF4a,USF1 in HepG2 cells, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy Strict Sites (Affymetrix ChIP-chip of 5 factors in HL60 cells, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy Strict (Affymetrix ChIP-chip of 5 factors in HL60 cells, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy Strict Pval (Affymetrix ChIP-chip on 5 factors, in HL60 cells, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy RNA (RNA abundance and transfrags, Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UT-Austin ChIP (c-Myc and EF24 in HeLa and 20901 fibroblasts, Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI DnaseI-HS (MPSS and Chip methods, in CD4 and GM09660, Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE GIS PET RNA (update Oct 2005 Freeze and new estrogen-treated MCF7)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gencode Genes (Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE LI Ng gIF ChIP (Addition of 5 antibodies, in HeLa cells)</td>
<td>27 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE EGASP Update (Add Fgenesh++ corrected predictions)</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE GIS STAT1 +/- gIF HeLa</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UVa DNA Rep Seg, UVa DNA Rep Ori (DNA replication timing segmentation and origins)</td>
<td>04 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE YALE MAS TAR (additional data)</td>
<td>27 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE YALE MAS RNA (additional data from June freeze)</td>
<td>27 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE BU Orchid</td>
<td>21 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI DIPs: Whole Genome Shotgun insertion/deletion polymorphisms from Jim Mullikin/SSAHA</td>
<td>21 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: LI Ng gIF ChIP (HeLa)</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: LI Ng Validation (Pol2, Taf1, H3K4me, H3ac)</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Sanger ChIP-chip</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Ludwig Institute/UCSD ChIP-chip - Pol II, TAF1, AcH3, H3K4me, SUZ12, and H3K27me antibodies - repackage three existing tracks into one</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, P-Values)</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: LI gIF ChIP (HeLa)
</td>
<td>19 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE HapMap Allele Frequencies</td>
<td>19 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: GIS P53 5FU HCT116</td>
<td>19 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENENCODE: Affy Transfrags, Affy pVal {0,2,8,32}h, Affy Sites {0,2,8,32}h</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Affy RNA Signal </td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale MAS array, RNA Transcript Map, Placenta, Nimblegen Protocol</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale MAS array, RNA Transcriptionally Active Regions, Placenta, Nimblegen Protocol</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, Signal)</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, Binding Sites)</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale TAR </td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Stanford Endogenous Transcript Levels in HCT116 Cells; 
Stanford ChIP-chip (HCT116, Jurkat, K562 cells; Sp1, Sp3 ChIP); 
Stanford ChIP-chip Smoothed Score; 
Stanford Methylation Digest: Be2C, CRL1690, HCT116, HT1080, HepG2, JEG3, Snu182, U87; 
Stanford Methylation Digest Smoothed Score</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Stanf Promoter</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: BU First Exon</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale RNA </td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Sanger Assoc</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: UC Davis ChIP-chip Nimblegen - E2F1 ab, HeLa Cells</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: GIS RNA PET</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: TBA23 EvoFold (Predictions of RNA Secondary Structure Using TBA23, ENCODE Regions only)</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Known+Pred RNA (Known and Predicted RNA Transcription)</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: UW/Regulome DNase I Hypersensitivity Probability</td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Regulome DNase I Hypersensitivity</td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: NHGRI DNaseI-HS </td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Riken CAGE - predicted gene start sites</td>
<td>13 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gene Prediction Workshop (EGASP) Updates</td>
<td>12 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Regions</td>
<td>15 Sep 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="#FFFEE8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="#D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%"><TR><TD>
		<FONT SIZE="4"><b>&nbsp; 
<A NAME=hg16></A>Human July 2003 (hg16, NCBI Build 34)	</b></FONT></TD></TR></TABLE>
	<TABLE BGCOLOR="#FFFEE8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">
<TR><TD nowrap><FONT color="#006666"><B>Track/Table Name</B></FONT></TD>
    <TD nowrap><FONT color="#006666"><B>Release Date</B></FONT>
</TD></TR>
<TR valign=top><TD align=left>
ENCODE EGASP Full (Correction)</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Recomb Hotspots: ENCODE recombination hotspots and rates</td>
<td>06 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE GIS PET RNA (update)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE YALE MAS TAR (additional data)</td>
<td>27 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE YALE MAS RNA (additional data from June freeze)</td>
<td>27 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Consensus Unannotated Transfrags</td>
<td>30 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE All Introns Proximal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE All Introns Distal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE All Intergenic Proximal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE All Intergenic Distal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Introns Proximal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Introns Distal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Intergenic Proximal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Intergenic Distal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
TBA23 EvoFold (Predictions of RNA Secondary Structure Using TBA23, ENCODE Regions only)</td>
<td>15 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford LAGAN Alignments</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: NHGRI/PSU TBA Alignments</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Recombination in 10 ENCODE regions (cM/Mb)</td>
<td>13 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford LAGAN Conserved Elements </td>
<td>11 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford LAGAN Conservation</td>
<td>11 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Prediction Workshop (EGASP) for Partial ENCODE Regions</td>
<td>06 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Prediction Workshop (EGASP) All ENCODE Regions</td>
<td>06 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Riken CAGE - predicted gene start sites</td>
<td>06 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gene Prediction Workshop (EGASP) Updates</td>
<td>05 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE HapMap Allele Frequencies</td>
<td>05 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Sanger Assoc</td>
<td>01 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale MAS array, RNA Transcriptionally Active Regions, Placenta, Nimblegen Protocol</td>
<td>27 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale MAS array, RNA Transcript Map, Placenta, Nimblegen Protocol</td>
<td>27 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Affy RNA Signal, Affy Transfrags, Affy pVal {0,2,8,32}h, Affy Sites {0,2,8,32}h</td>
<td>17 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale RNA Transcriptionally Active Regions (TARs)</td>
<td>17 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale RNA Transcript Map (Neutrophil, Placenta and NB4 cells)</td>
<td>17 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, P-Values)</td>
<td>15 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, Binding Sites)</td>
<td>15 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Boston University ORChID</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: UC Davis ChIP-chip Nimblegen - E2F1 ab, HeLa Cells</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, Signal)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Gencode Introns (Intron validation for selected Gencode Genes)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Gencode Genes (Manual annotations in the ENCODE regions from the Gencode project)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: ChIP-PET/GIS (data update, has cluster counts)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Ludwig Institute/UCSD ChIP-chip - Pol II, TAF1, AcH3, H3K4me, SUZ12, and H3K27me antibodies - repackage three existing tracks into one</td>
<td>13 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Ludwig Institute/UCSD ChIP-chip - gamma interferon experiments</td>
<td>13 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: GIS RNA PET</td>
<td>10 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: NHGRI DNaseI-HS (data update; now includes sites from activated as well as unactivated cells)</td>
<td>10 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Sanger ChIP-chip (histones H3,H4 GM06990)</td>
<td>10 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix ChIP-chip (Pol2 RA-stim HL60) Signal in ENCODE Regions (changed from wig to bedGraph)</td>
<td>08 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix ChIP-chip (Pol2 RA-stim HL60) PValue in ENCODE Regions (changed from wig to bedGraph)</td>
<td>08 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Stanf. Promoter in ENCODE regions (data update; changed to composite track -  old name: Promoter/Stanford) </td>
<td>06 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
DNA Rep/UVA in ENCODE regions (reloaded with new data)</td>
<td>03 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP-PET/GIS (P53 TFBS in ENCODE regions)</td>
<td>24 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Promoters/BU (Boston Univ. Promoters in ENCODE regions)</td>
<td>07 May 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UCSD/Ludwig Chip (acetylated and methylated histone binding)</td>
<td>06 May 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UCSD Ludwig RNAP, ENCODE UCSD Ludwig TAF1 (tracks combined into composite format)</td>
<td>29 Apr 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIp/Sanger (Lysine 4 methylation of histone in ENCODE regions)</td>
<td>07 Apr 2005</td>
</tr>
<TR valign=top><TD align=left>
RNA/Affy Sites (Affymetrix PolyA+ RNA from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
RNA/Affy Signal (Affymetrix PolyA+ RNA abundance from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP Affy Pol2 Sites (Affymetrix ChIP-chip of Pol2 from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP Affy Pol2 Pval (Affymetrix ChIP-chip enrichment of Pol2 from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP-chip data in ENCODE regions</td>
<td>05 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
DNA Rep/UVA (ENCODE DNA Replication data from Dutta Lab, University of VIrginia)</td>
<td>03 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
Promoter/Stanford (Promoters in ENCODE regions, from Myers lab at Stanford)</td>
<td>02 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
DNaseI-HS/NHGRI (ENCODE DNaseI Hypersensitive sites, from Collins lab, NHGRI)</td>
<td>01 Nov 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</TD></TR></TABLE>

</BODY></HTML>
