<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<HTML>
<HEAD>

	
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META http-equiv="Content-Script-Type" content="text/javascript">
	<META HTTP-EQUIV="Pragma" CONTENT="no-cache">
	<META HTTP-EQUIV="Expires" CONTENT="-1">
	<TITLE>Track and Table Releases	</TITLE>
    <LINK rel='STYLESHEET' href='../style/HGStyle.css' TYPE='text/css' />
</HEAD>
<BODY CLASS="qaPushQ cgi">
<script type='text/javascript'>
document.write("<center><div id='warnBox' style='display:none;'><CENTER><B id='warnHead'></B></CENTER><UL id='warnList'></UL><CENTER><button id='warnOK' onclick='hideWarnBox();return false;'></button></CENTER></div></center>");
function showWarnBox() {document.getElementById('warnOK').innerHTML='&nbsp;OK&nbsp;';var warnBox=document.getElementById('warnBox');warnBox.style.display=''; warnBox.style.width='65%';document.getElementById('warnHead').innerHTML='Warning/Error(s):';window.scrollTo(0, 0);}
function hideWarnBox() {var warnBox=document.getElementById('warnBox');warnBox.style.display='none';warnBox.innerHTML='';var endOfPage = document.body.innerHTML.substr(document.body.innerHTML.length-20);if(endOfPage.lastIndexOf('-- ERROR --') > 0) { history.back(); }}
window.onunload = function(){}; // Trick to avoid FF back button issue.
</script>

<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">

<TR><TH COLSPAN=1 ALIGN="left">
<A HREF="http://www.genome.gov/10005107" TARGET="_BLANK"><IMG SRC="../images/ENCODE_scaleup_logo.png" height=50 ALT="ENCODE Project at NHGRI"></A>
<IMG SRC="../images/encodeDcc.jpg" ALT="ENCODE Project at UCSC">
</TH></TR>


<noscript><div class='noscript'><div class='noscript-inner'><p><b>JavaScript is disabled in your web browser</b></p><p>You must have JavaScript enabled in your web browser to use the Genome Browser</p></div></div></noscript>
<script type='text/javascript' SRC='../js/jquery.js'></script>
<script type='text/javascript' SRC='../js/jquery.plugins.js'></script>
<LINK rel='STYLESHEET' href='../style/nice_menu.css' TYPE='text/css' />
<div id="main-menu-whole">

		<div id="main-menu"><div id="block-nice_menus-1" class="clear-block block block-nice_menus">

	<div class="upcorner"></div>
	<div class="blockbg">
		<div class="blockcontent">
						<div class="content">

<ul class="nice-menu nice-menu-down sf-js-enabled" id="nice-menu-1">

<li class="first even" id="home"><a href="../index.html" ><span id="home-link">Home</span></a></li> 

<li class="menuparent" id="tools1"><a href="../cgi-bin/hgGateway">Genomes</a>
	<ul style="display: none; visibility: hidden;">
	<li class="first"><a href="../cgi-bin/hgTracks?db=hg38&amp;position=lastDbPos">Human GRCh38/hg38</a></li>
	<li><a href="../cgi-bin/hgTracks?db=hg19&amp;position=lastDbPos">Human GRCh37/hg19</a></li>
	<li><a href="../cgi-bin/hgTracks?db=mm10&amp;position=lastDbPos">Mouse GRCm38/mm10</a></li>
	<li><a href="../cgi-bin/hgTracks?db=mm9&amp;position=lastDbPos">Mouse NCBI37/mm9</a></li>
	<li class="last"><a href="../cgi-bin/hgGateway">Other</a></li>
        </ul>
</li>

<li class="menuparent" id="tools2"><a href="../cgi-bin/hgTracks">Genome Browser</a>
	<ul style="display: none; visibility: hidden;">
	<li class="first"><a id="configureMenuLink" href="../cgi-bin/hgTracks?hgTracksConfigPage=configure">Configure</a></li>
	<li><a id="trackSearchMenuLink" href="../cgi-bin/hgTracks?hgt_tSearch=track+search">Track Search</a></li>
	<li><a id="cartResetMenuLink" href="../cgi-bin/cartReset">Reset All User Settings</a></li>
        </ul>
</li>


<li class="menuparent" id="tools3"><span>Tools</span>
	<ul style="display: none; visibility: hidden;">
	<li class="first"><a id="blatMenuLink" href="../cgi-bin/hgBlat?command=start">Blat</a></li>
	<li><a id="tableBrowserMenuLink" href="../cgi-bin/hgTables">Table Browser</a></li>
	<li><a id="vaiMenuLink" href="../cgi-bin/hgVai">Variant Annotation Integrator</a></li>
	<li><a id="diMenuLink" href="../cgi-bin/hgIntegrator">Data Integrator</a></li>
	<li><a id="gsMenuLink" href="../cgi-bin/hgNear">Gene Sorter</a></li>
	<li><a id="ggMenuLink" href="../cgi-bin/hgGenome">Genome Graphs</a></li>
	<li><a id="ispMenuLink" href="../cgi-bin/hgPcr">In-Silico PCR</a></li>
	<li><a id="liftOverMenuLink" href="../cgi-bin/hgLiftOver">LiftOver</a></li>
	<li><a id="visiGeneMenuLink" href="../cgi-bin/hgVisiGene">VisiGene</a></li>
	<li class="last"><a href="../util.html">Other Utilities</a></li>
	</ul>
</li>
<li class="menuparent" id="mirrors"><span>Mirrors</span>
	<ul style="display: none; visibility: hidden;">
	<li class="first"><a href="../mirror.html">Third Party Mirrors</a></li>
	<li><a href="../goldenPath/help/mirror.html">Mirroring Instructions</a></li>
	
	</ul>

</li>
<li class="menuparent" id="downloads"><span>Downloads</span>
	<ul style="display: none; visibility: hidden;">
	<li class="first"><a href="http://hgdownload.gi.ucsc.edu/downloads.html">Genome Data</a></li>
	<li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#source_downloads">Source Code</a></li>
  <li><a href="https://genome-store.ucsc.edu">Genome Browser Store</a></li>
	<li><a href="http://hgdownload.gi.ucsc.edu/downloads.html#utilities_downloads">Utilities</a></li>
	<li class="last"><a href="../goldenPath/help/ftp.html">FTP</a></li>
	<li class="last"><a href="../goldenPath/help/mysql.html">MySQL Access</a></li>
	</ul>
</li>
<li class="menuparent" id="myData"><span>My Data</span>
	<ul style="display: none; visibility: hidden;">
	<li><a id="sessionsMenuLink" href="../cgi-bin/hgSession?hgS_doMainPage=1">My Sessions</a></li>
	<li><a id="publicSessionsMenuLink" href="../cgi-bin/hgPublicSessions">Public Sessions</a></li>
	<li><a id="trackHubsMenuLink" href="../cgi-bin/hgHubConnect">Track Hubs</a></li>
	<li class="last"><a id="customTracksMenuLink" href="../cgi-bin/hgCustom">Custom Tracks</a></li>
	</ul>
</li>

<!-- OPTIONAL_VIEW_MENU -->

<li class="menuparent" id="help"><span>Help</span><ul style="display: none; visibility: hidden;">
<!-- CONTEXT_SPECIFIC_HELP -->
<li><a href="../FAQ/">FAQs & Search</a></li>
<li><a href="../goldenPath/help/hgTracksHelp.html">Browser Documentation</a></li>
<li><a href="../training/index.html">Training</a></li>
<li><a href="../contacts.html">Mailing Lists</a></li>
<li class="last"><a href="http://genomewiki.ucsc.edu/" target="_blank">GenomeWiki</a></li>
</ul>
</li>

<li class="menuparent last" id="aboutUs"><span>About Us</span>
	<ul style="display: none; visibility: hidden;">
	<li class="first odd"><a href="../goldenPath/newsarch.html">News</a></li>
	<li class="even"><a href="../goldenPath/pubs.html">Publications</a></li>
	<li class="even"><a href="http://genome.ucsc.edu/blog/">Blog</a></li>
	<li class="odd"><a href="../cite.html">Cite Us</a></li>
	<li class="even"><a href="../goldenPath/credits.html">Credits</a></li>
	<li class="odd"><a href="../goldenPath/releaseLog.html">Release Log</a></li>
	<li class="even"><a href="../staff.html">Staff</a></li>
	<li class="odd"><a href="../contacts.html">Contact Us</a></li>
	<li class="even"><a href="../conditions.html">Conditions of Use</a></li>
	<li class="odd"><a href="https://ucscgenomics.soe.ucsc.edu/careers/" TARGET="_blank">Jobs</a></li>
	<li class="even last"><a href="../license/">Licenses</a></li>
</ul>
</li>

</ul>
</div>
		</div>
	</div>
	<div class="downcorner"></div>

</div>
</div><!-- end main-menu -->
</div><!-- end main-menu-whole -->

<!-- +++++++++++++++++++++ CONTENT TABLES +++++++++++++++++++ -->
<TR><TD COLSPAN=3>
<div id=firstSection>      <!--outer table is for border purposes-->
      <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>
     <div class='subheadingBar'><div class='windowSize' id='sectTtl'>
Track and Table Releases     </div></div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


 This page contains track and table release information for the following genome assemblies:<br>
<ul>
<li><a CLASS="toc" HREF="#hg38">Human Dec. 2013 (GRCh38/hg38) (hg38)</a></li>
<li><a CLASS="toc" HREF="#hg19">Human Feb. 2009 (GRCh37/hg19) (hg19)</a></li>
<li><a CLASS="toc" HREF="#hg18">Human Mar. 2006 (NCBI36/hg18) (hg18)</a></li>
<li><a CLASS="toc" HREF="#hg17">Human May 2004 (NCBI35/hg17) (hg17)</a></li>
<li><a CLASS="toc" HREF="#hg16">Human July 2003 (NCBI34/hg16) (hg16)</a></li>
<li><a CLASS="toc" HREF="#mm10">Mouse Dec. 2011 (GRCm38/mm10) (mm10)</a></li>
<li><a CLASS="toc" HREF="#mm9">Mouse July 2007 (NCBI37/mm9) (mm9)</a></li>
<li><a CLASS="toc" HREF="#bosTau4">Cow Oct. 2007 (Baylor 4.0/bosTau4) (bosTau4)</a></li>
</ul>
<p>
	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=hg38></A>Human Dec. 2013 (GRCh38/hg38) (hg38, GRCh38 Genome Reference Consortium Human Reference 38 (GCA_000001405.2))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeGencodeV24" target=_blank>GENCODE V24 release</a></td>
<td>14 Jul 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeGencodeV23" target=_blank>GENCODE V23 release</a></td>
<td>17 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeRegDnaseClustered" target=_blank>ENCODE Regulation: DNase I Hypersensitivity Peak Clusters</a></td>
<td>19 Jun 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeRegDnaseWig" target=_blank>ENCODE Regulation:  DNase I Hypersensitivity Signal Colored by Similarity</a></td>
<td>19 Jun 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeRegDnase" target=_blank>ENCODE Regulation:  DNase I Hypersensitivity in 95 cell types</a></td>
<td>19 Jun 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeReg" target=_blank>ENCODE Regulation (Lifted from hg19)</a></td>
<td>19 Jun 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeGencodeV22" target=_blank>GENCODE Genes V22</a></td>
<td>30 Apr 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg38&g=wgEncodeGencodeV20" target=_blank>GENCODE V20 / Ensembl 76 release for GRCh38</a></td>
<td>13 Oct 2014</td>
</tr>
<TR valign=top><TD align=left>
Initial hg38 release (using GRCh38 Genome Reference Consortium Human Reference 38 (GCA_000001405.2)): Assembly, Hg19 Diff, Alt Map, Gap, GC Percent, GRC Contigs, INSDC, Scaffolds, RefSeq Genes, GENCODE Genes V19, Genscan Genes, MGC Genes, ORFeome Clones, Other RefSeq, Human mRNAs, Spliced ESTs, Human ESTs, Other ESTs, Other mRNAs, CpG Islands, Dog Chain/Net, Mouse Chain/Net, Primate Chain/Net, Self Chain, RepeatMasker, Interrupted Repeats, Simple Repeats, WM + SDust</td>
<td>06 Mar 2014</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=hg19></A>Human Feb. 2009 (GRCh37/hg19) (hg19, GRCh37 Genome Reference Consortium Human Reference 37 (GCA_000001405.1))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV24lift37" target=_blank>GENCODE V24 mapped from GRCh38/hg38 to GRCh37/hg19</a></td>
<td>02 May 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeAwgDnaseMasterSites" target=_blank>DNaseI Hypersensitive Site Master List (125 cell types) from ENCODE/Analysis </a></td>
<td>24 Nov 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRegDnaseClustered" target=_blank>ENCODE Regulation: DNase Clusters in 125 cell types from ENCODE (update to include clusters from 1 cell type, and to extend schema to include sources)</a></td>
<td>03 Nov 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeAwgSegmentation" target=_blank>Genome State Segmentations from ENCODE</a></td>
<td>17 Apr 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRegTfbsClusteredV3" target=_blank>ENCODE Regulation: Txn Factor ChIP Clusters (update to 161 TF targets, from uniform processing of June 2012 freeze)</a></td>
<td>24 Jan 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV19" target=_blank>GENCODE V19 release</a></td>
<td>15 Jan 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV17" target=_blank>GENCODE Version 17 release</a></td>
<td>05 Aug 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeAwgTfbsUniform" target=_blank>Transcription Factor ChIP-seq Uniform Peaks from ENCODE/Analysis</a></td>
<td>10 May 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRegDnaseClusteredV2" target=_blank>ENCODE Regulation: DNase Clusters (update to 125 cell types using ENCODE uniform peaks from Jan 2011 freeze)</a></td>
<td>25 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeAwgDnaseUniform" target=_blank>DNaseI Hypersensitivity Uniform Peaks from ENCODE/Analysis</a></td>
<td>22 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV14" target=_blank>GENCODE Version 14 release</a></td>
<td>17 Jan 2013</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlLongRnaSeq" target=_blank>Long RNA-seq from ENCODE/Cold Spring Harbor Lab (Release 3)</a></td>
<td>26 Sep 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromFaire" target=_blank>Open Chromatin by FAIRE from ENCODE/OpenChrom(UNC Chapel Hill) (Release 2)</a></td>
<td>19 Sep 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUncBsuProtGenc" target=_blank>Proteogenomics Hg19 and GENCODE Mapping from ENCODE/Univ. North Carolina/Boise State Univ.</a></td>
<td>11 Sep 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/HAIB (Release 3)</a></td>
<td>10 Sep 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromDnase" target=_blank>Open Chromatin by DNaseI HS from ENCODE/OpenChrom(Duke University) (Release 3)</a></td>
<td>28 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromChip" target=_blank>Open Chromatin TFBS by ChIP-seq from ENCODE/Open Chrom(UT Austin) (Release 2)</a></td>
<td>27 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/Stanford/Yale/USC/Harvard (Release 3)</a></td>
<td>20 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisRnaPet" target=_blank>RNA Sub-cellular Localization by Paired-end diTag Sequencing from ENCODE/GIS (Release 2)</a></td>
<td>14 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCaltechRnaSeq" target=_blank>RNA-seq from ENCODE/Caltech (Release 4)</a></td>
<td>08 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeBroadHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/Broad Institute (Release 3)</a></td>
<td>06 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDgf" target=_blank>DNaseI Digital Genomic Footprinting from ENCODE/University of Washington (Release 4)</a></td>
<td>06 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwAffyExonArray" target=_blank>Affymetrix Exon Array from ENCODE/University of Washington (Release 4)</a></td>
<td>31 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeFsuRepliChip" target=_blank>Replication Timing by Repli-chip from ENCODE/FSU (Release 2)</a></td>
<td>31 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibMethylRrbs" target=_blank>DNA Methylation by Reduced Representation Bisulfite Seq from ENCODE/HudsonAlpha (Release 3)</a></td>
<td>30 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRikenCage" target=_blank>RNA Subcellular CAGE Localization from ENCODE/RIKEN (Release 4)</a></td>
<td>30 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/University of Washington (Release 5)</a></td>
<td>24 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase" target=_blank>DNaseI Hypersensitivity by Digital DNaseI from ENCODE/University of Washington (Release 6)</a></td>
<td>23 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUncBsuProt" target=_blank>Proteogenomics Hg19 Mapping from ENCODE/Univ. North Carolina/Boise State Univ. (Release 2)</a></td>
<td>18 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwAffyExonArray" target=_blank>Affymetrix Exon Array from ENCODE/University of Washington (Release 3)</a></td>
<td>12 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRegTxn" target=_blank>Transcription Levels Assayed by RNA-seq on 9 Cell Lines from ENCODE</a></td>
<td>10 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlShortRnaSeq" target=_blank>Small RNA-seq from ENCODE/Cold Spring Harbor Lab (Release 3)</a></td>
<td>09 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUmassDekker5C" target=_blank>Chromatin Interactions by 5C from ENCODE/Dekker Univ. Mass. (Release 2)</a></td>
<td>09 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlLongRnaSeq" target=_blank>Long RNA-seq from ENCODE/Cold Spring Harbor Lab (Release 2)</a></td>
<td>21 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV12" target=_blank>Gene Annotations from ENCODE/GENCODE Version 12</a></td>
<td>21 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/Stanford/Yale/USC/Harvard (Release 3)</a></td>
<td>15 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase" target=_blank>DNaseI Hypersensitivity by Digital DNaseI from ENCODE/University of Washington (Release 5)</a></td>
<td>12 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisChiaPet" target=_blank>Chromatin Interaction Analysis Paired-End Tags (ChIA-PET) from ENCODE/GIS-Ruan</a></td>
<td>24 May 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/Stanford/Yale/USC/Harvard (Release 2)</a></td>
<td>14 May 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRikenCage" target=_blank>RNA Subcellular CAGE Localization from ENCODE/RIKEN (Release 3)</a></td>
<td>02 May 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCaltechRnaSeq" target=_blank>RNA-seq from ENCODE/Caltech (Release 3)</a></td>
<td>19 Apr 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeFsuRepliChip" target=_blank>Replication Timing by Repli-chip from ENCODE/FSU</a></td>
<td>18 Apr 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/University of Washington (Release 4)</a></td>
<td>11 Apr 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeDukeAffyExon" target=_blank>Affymetrix Exon Array from ENCODE/Duke (Release 3)</a></td>
<td>04 Apr 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromDnase" target=_blank>Open Chromatin by DNaseI HS from ENCODE/OpenChrom(Duke University) (Release 2)</a></td>
<td>28 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=encodeRegions" target=_blank>ENCODE Pilot (regions lifted from hg18)</a></td>
<td>26 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUmassDekker5C" target=_blank>Chromatin Interactions by 5C from ENCODE/Dekker Univ. Mass.</a></td>
<td>22 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV11" target=_blank>Gene Annotations from ENCODE/GENCODE Version 11</a></td>
<td>08 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeBroadHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/Broad Institute (Release 2)</a></td>
<td>06 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase" target=_blank>DNaseI Hypersensitivity by Digital DNaseI from ENCODE/University of Washington (Release 4)</a></td>
<td>06 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/HAIB (Release 2)</a></td>
<td>29 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwTfbs" target=_blank>CTCF Binding Sites by ChIP-seq from ENCODE/University of Washington (Release 3)</a></td>
<td>29 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibRnaSeq" target=_blank>RNA-seq from ENCODE/HAIB</a></td>
<td>23 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV10" target=_blank>Gene Annotations from ENCODE/GENCODE Version 10</a></td>
<td>22 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlShortRnaSeq" target=_blank>Small RNA-seq from ENCODE/Cold Spring Harbor Lab (Release 2)</a></td>
<td>22 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/University of Washington (Release 3)</a></td>
<td>16 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwRepliSeq" target=_blank>Replication Timing by Repli-seq from ENCODE/University of Washington</a></td>
<td>14 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDgf" target=_blank>DNaseI Digital Genomic Footprinting from ENCODE/University of Washington (Release 3)</a></td>
<td>13 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromSynth" target=_blank>DNaseI/FAIRE/ChIP Synthesis from ENCODE/OpenChrom(Duke/UNC/UTA) (Release 2)</a></td>
<td>10 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibGenotype" target=_blank>Genotype (CNV and SNP) by Illumina 1MDuo and CBS from ENCODE/HudsonAlpha (Release 2)</a></td>
<td>20 Jan 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCaltechRnaSeq" target=_blank>RNA-seq from ENCODE/Caltech (Release 2)</a></td>
<td>17 Jan 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwTfbs" target=_blank>CTCF Binding Sites by ChIP-seq from ENCODE/University of Washington (Release 2)</a></td>
<td>17 Jan 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwAffyExonArray" target=_blank>Affymetrix Exon Array from ENCODE/University of Washington (Release 2)</a></td>
<td>04 Jan 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRikenCage" target=_blank>RNA Subcellular CAGE Localization from ENCODE/RIKEN</a></td>
<td>22 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibMethyl450" target=_blank>CpG Methylation by Methyl 450K  Bead Arrays from ENCODE/HAIB</a></td>
<td>01 Dec 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/University of Washington</a></td>
<td>17 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/Stanford/Yale/USC/Harvard</a></td>
<td>29 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeMapability" target=_blank>ENCODE Mapability (hg19 release 3)</a></td>
<td>28 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase" target=_blank>DNaseI Hypersensitivity by Digital DNaseI from ENCODE/University of Washington (Release 3)</a></td>
<td>19 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisRnaPet" target=_blank>RNA Subcellular Localization by Paired End diTag Sequencing from GIS/ENCODE</a></td>
<td>17 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlLongRnaSeq" target=_blank>Long RNA-seq from ENCODE/Cold Spring Harbor Lab</a></td>
<td>17 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeDukeAffyExon" target=_blank>Affymetrix Exon Array from ENCODE/Duke (Release 2)</a></td>
<td>13 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUw5C" target=_blank>Chromatin Interactions by 5C from ENCODE/University of Washington</a></td>
<td>10 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/SYDH (Release 2)</a></td>
<td>07 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSunyAlbanyGeneSt" target=_blank>RNA Binding Protein Associated RNA by RIP-chip GeneST from ENCODE/SUNY Albany (Release 2 - fixed p & q values and scores)</a></td>
<td>04 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibMethylRrbs" target=_blank>DNA Methylation by Reduced Representation Bisulfite Seq from ENCODE/HudsonAlpha (Release 2)</a></td>
<td>30 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUncBsuProt" target=_blank>Proteogenomics from ENCODE/Univ. North Carolina/Boise State Univ.</a></td>
<td>29 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhRnaSeq" target=_blank>RNA-seq from ENCODE/Stanford/Yale/USC/Harvard</a></td>
<td>08 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?g=wgEncodeSunyRipSeq" target=_blank>RIP-seq from ENCODE/SUNY Albany</a></td>
<td>29 Aug 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibGenotype" target=_blank>Genotype (CNV and SNP) by Illumina 1MDuo and CBS from ENCODE/HudsonAlpha </a></td>
<td>16 Aug 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSunySwitchgear" target=_blank>RNA Binding Protein Associated RNA by SwitchGear from ENCODE/SUNY Albany</a></td>
<td>15 Aug 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhNsome" target=_blank>Nucleosome Position by MNase-seq from ENCODE/Stanford/BYU</a></td>
<td>29 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/HAIB</a></td>
<td>18 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV7" target=_blank>Gene Annotations from ENCODE/GENCODE Version 7</a></td>
<td>11 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeBroadHmm" target=_blank>Chromatin State Segmentation by HMM from ENCODE/Broad</a></td>
<td>24 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeDukeAffyExon" target=_blank>Affymetrix Exon Array from ENCODE/Duke</a></td>
<td>14 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromSynth" target=_blank>DNaseI/FAIRE/ChIP Synthesis from ENCODE/OpenChrom(Duke/UNC/UTA)</a></td>
<td>10 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUchicagoTfbs" target=_blank>Transcription Factor Binding Sites by Epitope-Tag ChIP-seq from ENCODE/University of Chicago </a></td>
<td>03 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeReg" target=_blank>Integrated Regulation from ENCODE Tracks (Transcription, Layered H3K4Me1, Layered H3K4Me3, Layered H3K27Ac, DNase Clusters, Txn Factor ChIP)</a></td>
<td>19 May 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSydhHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/Stanford/Yale/Davis/Harvard</a></td>
<td>19 May 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeBuOrchid" target=_blank>ORChID Predicted DNA Cleavage Sites from ENCODE/Boston Univ (Tullius Lab)</a></td>
<td>18 May 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase" target=_blank>DNaseI Hypersensitivity by Digital DNaseI from ENCODE/University of Washington</a></td>
<td>04 May 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSunyAlbanyGeneSt" target=_blank>RNA Binding Protein Associated RNA by RIP-chip GeneST from ENCODE/SUNY Albany</a></td>
<td>28 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromDnase" target=_blank>Open Chromatin by DNaseI HS from ENCODE/Duke University</a></td>
<td>19 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDgf" target=_blank>DNaseI Digital Genomic Footprinting from ENCODE/University of Washington (Release 2)</a></td>
<td>07 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwAffyExonArray" target=_blank>Affymetrix Exon Array from ENCODE/University of Washington</a></td>
<td>01 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRikenCage" target=_blank>RNA CAGE Subcellular Localization from ENCODE/RIKEN</a></td>
<td>23 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromFaire" target=_blank>Open Chromatin by FAIRE-seq from ENCODE/UNC</a></td>
<td>21 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwTfbs" target=_blank>CTCF Binding Sites by ChIP-seq from ENCODE/University of Washington</a></td>
<td>17 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromChip" target=_blank>Open Chromatin TFBS by ChIP-seq from ENCODE/Open Chrom(UT Austin)</a></td>
<td>17 Mar 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="./../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeSunyAlbanyTiling" target=_blank>RNA Binding Protein Associated RNA by Tiling Array from ENCODE/SUNY Albany</a></td>
<td>14 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/University of Washington</a></td>
<td>11 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeBroadHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/Broad Institute</a></td>
<td>10 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?c=chr21&g=wgEncodeCaltechRnaSeq" target=_blank>ENCODE Caltech (Wold Lab) RNA-seq</a></td>
<td>08 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisDnaPet" target=_blank>ENCODE GIS DNA PET (update: added fastq files)</a></td>
<td>12 Jan 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase" target=_blank>ENCODE UW DNaseI HS</a></td>
<td>16 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibMethylRrbs" target=_blank>ENCODE HAIB Methyl RRBS</a></td>
<td>15 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDgf" target=_blank>ENCODE UW DNaseI DGF</a></td>
<td>14 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hgs19&g=wgEncodeAffyRnaChip" target=_blank>ENCODE Affymetrix/CSHL Subcellular RNA Localization by Tiling Array</a></td>
<td>03 Dec 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlShortRnaSeq" target=_blank>ENCODE CSHL Sm RNA-seq</a></td>
<td>17 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisRnaSeq" target=_blank>ENCODE Genome Institute of Singapore RNA-seq
</a></td>
<td>17 Nov 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencode" target=_blank>ENCODE Gencode Gene Annotations (V4)</a></td>
<td>15 Sep 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisDnaPet" target=_blank>ENCODE Genome Institute of Singapore DNA Paired-End Ditags</a></td>
<td>08 Sep 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeMapability" target=_blank>ENCODE Mapability or Uniqueness of Reference Genome</a></td>
<td>03 Jun 2010</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=hg18></A>Human Mar. 2006 (NCBI36/hg18) (hg18, NCBI Build 36.1)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeUwDnaseSeq" target=_blank>ENCODE Univ. Washington DNaseI Hypersensitivity by Digital DNaseI - Validation data (Release 5)</a></td>
<td>11 Oct 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBroadHmm" target=_blank>ENCODE Chromatin State Segmentation by HMM from Broad Institute, MIT and MGH</a></td>
<td>24 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeYaleChIPseq" target=_blank>ENCODE Transcription Factor Binding Sites by ChIP-seq from Yale/UC-Davis/Harvard</a></td>
<td>03 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaCnv" target=_blank>ENCODE Common Cell CNV (Release 2): colors in Regions view subtracks updated to new standards</a></td>
<td>08 Apr 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeUwDnaseSeq" target=_blank>ENCODE UW DNaseI HS (Release 4): Jurkat (G1 Phase) and HIPEpic replicates have been revoked</a></td>
<td>24 Feb 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeReg" target=_blank>ENCODE Regulation</a></td>
<td>12 Aug 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaRnaSeq" target=_blank>ENCODE HudsonAlpha RNA-seq</a></td>
<td>30 Jul 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeYaleRNAseq" target=_blank>ENCODE RNA-seq from Yale/UC-Davis/Harvard</a></td>
<td>09 Jul 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBuOrchid" target=_blank>ENCODE Boston Univ (Tullius Lab) ORChID Predicted DNA Cleavage Sites</a></td>
<td>29 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBroadChipSeq" target=_blank>ENCODE Broad Histone (update to Release 3) - improved display of Signal subtracks when zoomed in to a range spanning less than 16,500 base pairs.</a></td>
<td>16 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaChipSeq" target=_blank>ENCODE HAIB TFBS (release 2)</a></td>
<td>11 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeCaltechRnaSeq" target=_blank>ENCODE Caltech RNA-seq (Release 2)</a></td>
<td>10 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeMapability" target=_blank>ENCODE Mapability or Uniqueness of Reference Genome (update: adding CRG Alignability data)</a></td>
<td>03 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeGisPet" target=_blank>ENCODE GIS Subcellular RNA Localization by Paired End diTag Sequencing</a></td>
<td>01 Jun 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeGisRnaSeq" target=_blank>ENCODE Genome Institute of Singapore RNA-seq</a></td>
<td>26 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeYaleChIPseq" target=_blank>ENCODE Yale TFBS (release 3)</a></td>
<td>26 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeRikenCage" target=_blank>ENCODE RIKEN RNA Subcellular Localization by CAGE Tags (Release 2)</a></td>
<td>24 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeDukeAffyExonArray" target=_blank>ENCODE Duke Affy All-Exon Arrays</a></td>
<td>24 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeUwDGF" target=_blank>ENCODE Univ. Washington Digital DNase Genomic Footprinting</a></td>
<td>17 May 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeGisDnaPet" target=_blank>ENCODE Genome Institute of Singapore DNA Paired-End Ditags</a></td>
<td>19 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
Fixed corrupt table from ENCODE pilot project: Gencode Gene Annotations (March 2007)</td>
<td>09 Apr 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeChromatinMap" target=_blank>ENCODE Open Chromatin (release 3)</a></td>
<td>31 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBroadChipSeq" target=_blank>ENCODE Histone Modifications by Broad Institute ChIP-seq (release 3)</a></td>
<td>23 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeUwAffyExonArray" target=_blank>ENCODE UW Affy All-Exon Arrays (release 2)</a></td>
<td>18 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeUwChIPSeq" target=_blank>ENCODE Histone Modifications by Univ. Washington ChIP-seq</a></td>
<td>11 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaMethylSeq" target=_blank>ENCODE HudsonAlpha Methyl-seq</a></td>
<td>05 Mar 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeSangerGencode" target=_blank>ENCODE Gencode Gene Annotations (Version 3c)</a></td>
<td>26 Feb 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeCshlShortRnaSeq" target=_blank>ENCODE Cold Spring Harbor Labs Small RNA-seq</a></td>
<td>26 Feb 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeUwDnaseSeq" target=_blank>ENCODE Univ. Washington DNaseI Hypersensitivity by Digital DNaseI (Release 2)</a></td>
<td>03 Feb 2010</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBroadChipSeq" target=_blank>ENCODE Broad Histone Release 2 (release 2)</a></td>
<td>18 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeCaltechRnaSeq" target=_blank>ENCODE Caltech RNA-seq Map Algorithm</a></td>
<td>18 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeCshlLongRnaSeq" target=_blank>ENCODE Cold Spring Harbor Labs Long RNA-seq (Release 2: GM12878, K562 whole cell)</a></td>
<td>17 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeChromatinMap" target=_blank>ENCODE Open Chromatin, Duke/UNC/UT (Release 2)</a></td>
<td>04 Dec 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeRikenCage" target=_blank>ENCODE RIKEN CAGE Loc: updated download files</a></td>
<td>30 Nov 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeYaleChIPseq" target=_blank>ENCODE Yale TFBS (Update several subtracks; add new subtracks)</a></td>
<td>14 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeAffyRnaChip" target=_blank>ENCODE Affymetrix/CSHL Subcellular RNA Localization by Tiling Array (Update:  GM12878 Nucleolus Total, HepG2 Nucleus PolyA+ and PolyA-, HepG2 Nucleolus Total)</a></td>
<td>02 Oct 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeNhgriNre" target=_blank>ENCODE NHGRI Elnitski Negative Regulatory Elements</a></td>
<td>24 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeUwAffyExonArray" target=_blank>ENCODE UW Affy All Exon Arrays</a></td>
<td>15 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaCnv" target=_blank>ENCODE Common Cell-Type Copy Number Variation</a></td>
<td>11 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaMethylSeq" target=_blank>ENCODE HudsonAlpha Methyl-seq</a></td>
<td>11 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaMethyl27" target=_blank>ENCODE HudsonAlpha CpG Methylation by Illumina Methyl27</a></td>
<td>11 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?g=wgEncodeUwDnaseSeq" target=_blank>ENCODE Univ. Washington DNaseI Hypersensitivity by Digital DNaseI</a></td>
<td>11 Sep 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaChipseq" target=_blank>ENCODE HudsonAlpha CHIP-seq (Correction to labels and metadata - TAF1)</a></td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeRikenCage" target=_blank>ENCODE RIKEN RNA Subcellular Localization by CAGE Tags: PolyA- in K562 nucleus and cytosol, PolyA- in prostate whole cell</a></td>
<td>13 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeCshlLoRnaSeq" target=_blank>ENCODE Cold Spring Harbor Labs Long RNA-seq (K562)</a></td>
<td>11 Aug 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHelicosRnaSeq" target=_blank>ENCODE Helicos RNA-seq (K562)</a></td>
<td>28 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeMapability" target=_blank>ENCODE Mapability or Uniqueness of Reference Genome</a></td>
<td>24 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeAffyRnaChip" target=_blank>ENCODE Affymetrix/CSHL Subcellular RNA Localization by Tiling Array</a></td>
<td>06 Jul 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeChromatinMap" target=_blank>ENCODE Open Chromatin, Duke/UNC/UT</a></td>
<td>26 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg18&g=wgEncodeRikenCage" target=_blank>ENCODE RIKEN RNA Subcellular Localization by CAGE Tags</a></td>
<td>09 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeHudsonalphaChipSeq" target=_blank>ENCODE HudsonAlpha Transcription Factor Binding Sites by ChIP-seq</a></td>
<td>05 Jun 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeSangerGencode" target=_blank>ENCODE Gencode Gene Annotations</a></td>
<td>08 May 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeSunyalbanyRnaGeneChip" target=_blank>ENCODE SUNY Albany RNA Binding Proteins by RIP-chip</a></td>
<td>08 Apr 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBroadChipSeq" target=_blank>ENCODE Histone Modifications by Broad Institute ChIP-seq (Raw Signal tracks)</a></td>
<td>03 Apr 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeBroadChipSeq" target=_blank>ENCODE Histone Modifications by Broad Institute ChIP-seq</a></td>
<td>03 Apr 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=wgEncodeYaleChIPseq" target=_blank>ENCODE Transcription Factor Binding Sites by ChIP-seq from Yale/UC-Davis/Harvard</a></td>
<td>13 Mar 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../ENCODE/protocols/cell" target=_blank>ENCODE cell culture protocols</a></td>
<td>09 Mar 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../ENCODE" target=_blank>ENCODE portal update for production phase</a></td>
<td>23 Jan 2009</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=hg18&g=encodeMsaTbaDec07&" target=_blank>ENCODE 36-Way TBA Alignments and Conservation</a></td>
<td>19 Dec 2008</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix ENCODE Extension Transcription Sites & Signal</td>
<td>16 Apr 2008</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI DIPs (lift from hg17)</td>
<td>23 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
Whole-Genome ENCODE LI/UCSD TAF1 Super Track</td>
<td>17 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE data migration from hg17</td>
<td>08 Oct 2007</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=hg17></A>Human May 2004 (NCBI35/hg17) (hg17, NCBI Build 35)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
ENCODE TBA SCONE Conservation (Update)</td>
<td>19 Oct 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UC Davis ChIP-chip NimbleGen (E2F1, c-Myc, TAF, POLII)</td>
<td>24 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sites - Jun07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sig - Jun07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Pval - Jun07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sites - Apr07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sig - Apr07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Pval - Apr07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sites - Update to Jan07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sig - Jan07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Pval - Jan07 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sites Update to Nov06 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP-chip Sig - Nov06 batch</td>
<td>13 Jul 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE University of Virginia DNA Replication Origins (data update)</td>
<td>30 May 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Duke Dnase/chip - Feb07 batch (data addition)</td>
<td>21 May 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gencode RACEfrags</td>
<td>18 May 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gencode Gene Annotations (March 2007)</td>
<td>04 May 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UCD Ng ChIP Hits (update of cMyc and E2F1 data)</td>
<td>17 Apr 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale RFBR (Regulatory Factor Binding Regions)</td>
<td>21 Feb 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: UCSC EvoFold (Encode predictions analyzed in Washietl, Pedersen, et al., 2007).  Replaces TBA23 EvoFold data. </td>
<td>13 Feb 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE - University of Vienna, RNA secondary structure predicted by RNAz</td>
<td>16 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale RNA (update with October 2005 freeze data)</td>
<td>10 Jan 2007</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UW/Reg QCP DNaseI Sens (replaces the following three tracks: UW/Reg DNaseI HSs,  UW/Reg Q/A,  UW DNase GM)</td>
<td>07 Dec 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Duke/NHGRI DNase (New Chip method in HepG2, Hela, GM, and CD4+)</td>
<td>05 Dec 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UT-Austin STAGE (update to add STAT1 tags)</td>
<td>29 Nov 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE GIS Chip-PET (P53 in HCT116, STAT1 in HeLa, c-Myc in P493)</td>
<td>07 Nov 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI/PSU/UCSC TBA Conserved Elements</td>
<td>05 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI/PSU/UCSC TBA Conservation</td>
<td>05 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI/PSU TBA Alignments</td>
<td>05 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE MSA Consensus Constrained Elements</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UC Berkeley  MAVID Alignments</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UC Berkeley MAVID Conservation</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UC Berkeley MAVID Conserved Elements</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford MLAGAN Conserved Elements</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford MLAGAN Conservation</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford MLAGAN Alignments</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE MSA Alignment Gaps</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE MSA Alignment Agreement</td>
<td>04 Oct 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UNC FAIRE track (data update)</td>
<td>21 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
8 new subtracks for ENCODE Sanger ChIP-chip track</td>
<td>17 Aug 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE University of Uppsala, Sweden ChIP-chip Na-butyrate time series</td>
<td>26 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UNC FAIRE Peaks (ChIPOTle) Apr. 2006 Update, new subtrack added to UNC FAIRE</td>
<td>15 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UW DNase/Array GM06990</td>
<td>13 Jun 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Sanger ChIP (additional data in HFL-1 cell line)</td>
<td>24 May 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Detection of LinEage Specific Selection (DLESS)</td>
<td>19 May 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP Sites (data replaced with 5 new transcription factors). Previous track Yale ChIP Sites renamed Yale STAT1 Sites.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP Signal (data replaced with 5 new transcription factors). Previous track Yale ChIP Signal renamed Yale STAT1 Signal.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale ChIP pVal (data replaced with 5 new transcription factors). Previous track Yale ChIP pVal renamed Yale STAT1 pVal.</td>
<td>13 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE EGASP Partial (data correction)</td>
<td>11 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
Recomb Hotspots: ENCODE recombination hotspots and rates</td>
<td>06 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Uppsala Chip (University of Uppsala, Sweden ChIP-chip coordinate corrections)</td>
<td>03 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Yale TAR (Replace June data with new methods)</td>
<td>02 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UVa DNA Rep Ori (University of Virginia DNA Replication Origins, Oct 2005 Freeze)</td>
<td>29 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UT-Austin STAGE (Sequence Tag Analysis of Genomic Enrichment), c-Myc in HeLa cells</td>
<td>28 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy Transfrags (updated HeLa data)</td>
<td>21 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy RNA Signal (updated HeLa data)</td>
<td>21 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UVa DNA Rep TR50 (University of Virginia DNA Smoothed Timing at 50% Replication, Oct 2005 Freeze)</td>
<td>16 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UVa DNA Rep Seg (University of Virginia DNA Replication Temporal Segmentation, Oct 2005 Freeze)</td>
<td>02 Dec 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Sanger ChIP Hits (Peaks and Centers, Oct 2005 Freeze)</td>
<td>22 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Sanger ChIP (H3, H4 histones in GM, K562 and HeLa cells, Oct 2005 Freeze)</td>
<td>22 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
 ENCODE UNC FAIRE (Formaldehyde Assisted Isolation of Regulatory Elements, Oct 2005 Freeze)
</td>
<td>22 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UCD Ng ChIP (UC Davis ChIP-chip of E2F1, Myc in HeLa, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Uppsala Chip (University of Uppsala, Sweden ChIP-chip of H3ac,HNF3b,HNF4a,USF1 in HepG2 cells, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy Strict Sites (Affymetrix ChIP-chip of 5 factors in HL60 cells, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy Strict (Affymetrix ChIP-chip of 5 factors in HL60 cells, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy Strict Pval (Affymetrix ChIP-chip on 5 factors, in HL60 cells, Oct 2005 Freeze)</td>
<td>18 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Affy RNA (RNA abundance and transfrags, Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UT-Austin ChIP (c-Myc and EF24 in HeLa and 20901 fibroblasts, Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI DnaseI-HS (MPSS and Chip methods, in CD4 and GM09660, Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE GIS PET RNA (update Oct 2005 Freeze and new estrogen-treated MCF7)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gencode Genes (Oct 2005 Freeze)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE LI Ng gIF ChIP (Addition of 5 antibodies, in HeLa cells)</td>
<td>27 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE EGASP Update (Add Fgenesh++ corrected predictions)</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE GIS STAT1 +/- gIF HeLa</td>
<td>14 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UVa DNA Rep Seg, UVa DNA Rep Ori (DNA replication timing segmentation and origins)</td>
<td>04 Oct 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE YALE MAS TAR (additional data)</td>
<td>27 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE YALE MAS RNA (additional data from June freeze)</td>
<td>27 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE BU Orchid</td>
<td>21 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE NHGRI DIPs: Whole Genome Shotgun insertion/deletion polymorphisms from Jim Mullikin/SSAHA</td>
<td>21 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: LI Ng gIF ChIP (HeLa)</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: LI Ng Validation (Pol2, Taf1, H3K4me, H3ac)</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Sanger ChIP-chip</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Ludwig Institute/UCSD ChIP-chip - Pol II, TAF1, AcH3, H3K4me, SUZ12, and H3K27me antibodies - repackage three existing tracks into one</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, P-Values)</td>
<td>20 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: LI gIF ChIP (HeLa)
</td>
<td>19 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE HapMap Allele Frequencies</td>
<td>19 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: GIS P53 5FU HCT116</td>
<td>19 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENENCODE: Affy Transfrags, Affy pVal {0,2,8,32}h, Affy Sites {0,2,8,32}h</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Affy RNA Signal </td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale MAS array, RNA Transcript Map, Placenta, Nimblegen Protocol</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale MAS array, RNA Transcriptionally Active Regions, Placenta, Nimblegen Protocol</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, Signal)</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, Binding Sites)</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale TAR </td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Stanford Endogenous Transcript Levels in HCT116 Cells; 
Stanford ChIP-chip (HCT116, Jurkat, K562 cells; Sp1, Sp3 ChIP); 
Stanford ChIP-chip Smoothed Score; 
Stanford Methylation Digest: Be2C, CRL1690, HCT116, HT1080, HepG2, JEG3, Snu182, U87; 
Stanford Methylation Digest Smoothed Score</td>
<td>16 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Stanf Promoter</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: BU First Exon</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale RNA </td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Sanger Assoc</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: UC Davis ChIP-chip Nimblegen - E2F1 ab, HeLa Cells</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: GIS RNA PET</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: TBA23 EvoFold (Predictions of RNA Secondary Structure Using TBA23, ENCODE Regions only)</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Known+Pred RNA (Known and Predicted RNA Transcription)</td>
<td>15 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: UW/Regulome DNase I Hypersensitivity Probability</td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Regulome DNase I Hypersensitivity</td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: NHGRI DNaseI-HS </td>
<td>14 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Riken CAGE - predicted gene start sites</td>
<td>13 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gene Prediction Workshop (EGASP) Updates</td>
<td>12 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Regions</td>
<td>15 Sep 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=hg16></A>Human July 2003 (NCBI34/hg16) (hg16, NCBI Build 34)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
ENCODE EGASP Full (Correction)</td>
<td>12 Jan 2006</td>
</tr>
<TR valign=top><TD align=left>
ENCODE GIS PET RNA (update)</td>
<td>07 Nov 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE YALE MAS TAR (additional data)</td>
<td>27 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE YALE MAS RNA (additional data from June freeze)</td>
<td>27 Sep 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Consensus Unannotated Transfrags</td>
<td>30 Aug 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE All Introns Proximal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE All Introns Distal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE All Intergenic Proximal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE All Intergenic Distal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Introns Proximal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Introns Distal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Intergenic Proximal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Intergenic Distal</td>
<td>17 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
TBA23 EvoFold (Predictions of RNA Secondary Structure Using TBA23, ENCODE Regions only)</td>
<td>15 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford LAGAN Alignments</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: NHGRI/PSU TBA Alignments</td>
<td>14 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Recombination in 10 ENCODE regions (cM/Mb)</td>
<td>13 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford LAGAN Conserved Elements </td>
<td>11 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Stanford LAGAN Conservation</td>
<td>11 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Prediction Workshop (EGASP) for Partial ENCODE Regions</td>
<td>06 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
Gene Prediction Workshop (EGASP) All ENCODE Regions</td>
<td>06 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Riken CAGE - predicted gene start sites</td>
<td>06 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE Gene Prediction Workshop (EGASP) Updates</td>
<td>05 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE HapMap Allele Frequencies</td>
<td>05 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Sanger Assoc</td>
<td>01 Jul 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale MAS array, RNA Transcriptionally Active Regions, Placenta, Nimblegen Protocol</td>
<td>27 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale MAS array, RNA Transcript Map, Placenta, Nimblegen Protocol</td>
<td>27 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Affy RNA Signal, Affy Transfrags, Affy pVal {0,2,8,32}h, Affy Sites {0,2,8,32}h</td>
<td>17 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale RNA Transcriptionally Active Regions (TARs)</td>
<td>17 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale RNA Transcript Map (Neutrophil, Placenta and NB4 cells)</td>
<td>17 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, P-Values)</td>
<td>15 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, Binding Sites)</td>
<td>15 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Boston University ORChID</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: UC Davis ChIP-chip Nimblegen - E2F1 ab, HeLa Cells</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Yale ChIP-chip (STAT1 ab, Hela cells, Signal)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Gencode Introns (Intron validation for selected Gencode Genes)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Gencode Genes (Manual annotations in the ENCODE regions from the Gencode project)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: ChIP-PET/GIS (data update, has cluster counts)</td>
<td>14 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Ludwig Institute/UCSD ChIP-chip - Pol II, TAF1, AcH3, H3K4me, SUZ12, and H3K27me antibodies - repackage three existing tracks into one</td>
<td>13 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Ludwig Institute/UCSD ChIP-chip - gamma interferon experiments</td>
<td>13 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: GIS RNA PET</td>
<td>10 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: NHGRI DNaseI-HS (data update; now includes sites from activated as well as unactivated cells)</td>
<td>10 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE: Sanger ChIP-chip (histones H3,H4 GM06990)</td>
<td>10 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix ChIP-chip (Pol2 RA-stim HL60) Signal in ENCODE Regions (changed from wig to bedGraph)</td>
<td>08 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Affymetrix ChIP-chip (Pol2 RA-stim HL60) PValue in ENCODE Regions (changed from wig to bedGraph)</td>
<td>08 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
Stanf. Promoter in ENCODE regions (data update; changed to composite track -  old name: Promoter/Stanford) </td>
<td>06 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
DNA Rep/UVA in ENCODE regions (reloaded with new data)</td>
<td>03 Jun 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP-PET/GIS (P53 TFBS in ENCODE regions)</td>
<td>24 May 2005</td>
</tr>
<TR valign=top><TD align=left>
Promoters/BU (Boston Univ. Promoters in ENCODE regions)</td>
<td>07 May 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UCSD/Ludwig Chip (acetylated and methylated histone binding)</td>
<td>06 May 2005</td>
</tr>
<TR valign=top><TD align=left>
ENCODE UCSD Ludwig RNAP, ENCODE UCSD Ludwig TAF1 (tracks combined into composite format)</td>
<td>29 Apr 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIp/Sanger (Lysine 4 methylation of histone in ENCODE regions)</td>
<td>07 Apr 2005</td>
</tr>
<TR valign=top><TD align=left>
RNA/Affy Sites (Affymetrix PolyA+ RNA from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
RNA/Affy Signal (Affymetrix PolyA+ RNA abundance from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP Affy Pol2 Sites (Affymetrix ChIP-chip of Pol2 from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP Affy Pol2 Pval (Affymetrix ChIP-chip enrichment of Pol2 from retinoic acid-stimulated HL60 cells at 4 timepoints, in ENCODE regions)
</td>
<td>31 Jan 2005</td>
</tr>
<TR valign=top><TD align=left>
ChIP-chip data in ENCODE regions</td>
<td>05 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
DNA Rep/UVA (ENCODE DNA Replication data from Dutta Lab, University of VIrginia)</td>
<td>03 Nov 2004</td>
</tr>
<TR valign=top><TD align=left>
DNaseI-HS/NHGRI (ENCODE DNaseI Hypersensitive sites, from Collins lab, NHGRI)</td>
<td>01 Nov 2004</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=mm10></A>Mouse Dec. 2011 (GRCm38/mm10) (mm10, Genome Reference Consortium Mouse Build 38 (GCA_000001635.2))     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=wgEncodeGencodeVM11" target=_blank>GENCODE VM11 release</a></td>
<td>09 Nov 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=wgEncodeGencodeVM7" target=_blank>GENCODE Version M9</a></td>
<td>22 Apr 2016</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm10&g=wgEncodeGencodeVM7" target=_blank>GENCODE Version M7</a></td>
<td>12 Nov 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/db=mm10&g=wgEncodeGencodeVM4" target=_blank>GENCODE Version M4</a></td>
<td>22 Jan 2015</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm10&g=wgEncodeGencodeVM3" target=_blank>GENCODE VM3 import from http://www.gencodegenes.org/</a></td>
<td>28 Oct 2014</td>
</tr>
<TR valign=top><TD align=left>
<a href=" ../../cgi-bin/hgTrackUi?db=mm10&g=wgEncodeGencodeVM2" target=_blank>GENCODE Gene annotations for mouse, Version M2</a></td>
<td>11 Mar 2014</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=mm9></A>Mouse July 2007 (NCBI37/mm9) (mm9, NCBI Build 37)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeUwDnase" target=_blank>DNaseI Hypersensitivity by Digital DNaseI from ENCODE/University of Washington (Release 2)</a></td>
<td>25 Sep 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeUwDgf" target=_blank>DNaseI Digital Genomic Footprinting from ENCODE/University of Washington</a></td>
<td>07 Sep 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/LICR (Release 3)</a></td>
<td>05 Sep 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeCshlLongRnaSeq" target=_blank>Long RNA-seq from ENCODE/Cold Spring Harbor Lab (Release 3)</a></td>
<td>05 Sep 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhHist" target=_blank>Histone Modifications by ChIP-seq from ENCODE/SYDH (Release 2)</a></td>
<td>20 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/Stanford/Yale (Release 4)</a></td>
<td>16 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodePsuDnase" target=_blank>DNaseI Hypersensitivity by Digital DNaseI from ENCODE/PSU</a></td>
<td>16 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/LICR (Release 3)</a></td>
<td>13 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodePsuTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/PSU (Release 2)</a></td>
<td>07 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodePsuHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/PSU (Release 2)</a></td>
<td>06 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhRnaSeq" target=_blank>RNA-seq from ENCODE/Stanford/Yale (Release 2)</a></td>
<td>01 Aug 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeUwRnaSeq" target=_blank>RNA-seq from ENCODE/UW</a></td>
<td>10 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeFsuRepliChip" target=_blank>Replication Timing by Repli-chip from ENCODE/FSU</a></td>
<td>03 Jul 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeCshlLongRnaSeq" target=_blank>Long RNA-seq from ENCODE/Cold Spring Harbor Lab (Release 2)</a></td>
<td>29 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhRnaSeq" target=_blank>RNA-seq from ENCODE/Stanford</a></td>
<td>25 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeMapability" target=_blank>Mapability or Uniqueness of Reference Genome from ENCODE</a></td>
<td>19 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/Stanford/Yale (Release 3)</a></td>
<td>15 Jun 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeCaltechRnaSeq" target=_blank>RNA-seq from ENCODE/Caltech</a></td>
<td>07 May 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrRnaSeq" target=_blank>RNA-seq from ENCODE/LICR (Release 2)</a></td>
<td>04 May 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodePsuHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/PSU</a></td>
<td>30 Apr 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/LICR (Release 2)</a></td>
<td>13 Apr 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/LICR (Release 2)</a></td>
<td>28 Mar 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeCshlLongRnaSeq" target=_blank>Long RNA-seq from ENCODE/Cold Spring Harbor Lab</a></td>
<td>29 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodePsuTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/PSU</a></td>
<td>13 Feb 2012</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeCaltechHist" target=_blank>Histone Modifications by ChIP-seq by ChIP-seq from ENCODE/Caltech</a></td>
<td>28 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeCaltechTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/Caltech</a></td>
<td>23 Nov 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/Stanford/Yale (Release 2)</a></td>
<td>19 Sep 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrRnaSeq" target=_blank> 	
RNA-seq from ENCODE/LICR</a></td>
<td>04 Aug 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/LICR</a></td>
<td>22 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhHist" target=_blank>Histone Modifications by ChIP-seq from ENCODE/Stanford/Yale</a></td>
<td>22 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeUwDnase" target=_blank>DNaseI Hypersensitivity by Digital DNaseI from ENCODE/University of Washington</a></td>
<td>01 Jul 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrHistone" target=_blank>Histone Modifications by ChIP-seq from ENCODE/LICR</a></td>
<td>24 Jun 2011</td>
</tr>
<TR valign=top><TD align=left>
<a href="../../cgi-bin/hgTrackUi?db=mm9&g=wgEncodeSydhTfbs" target=_blank>Transcription Factor Binding Sites by ChIP-seq from ENCODE/Stanford/Yale </a></td>
<td>07 Feb 2011</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
<div>
<!-- +++++++++++++++++++++ START NEW SECTION +++++++++++++++++++ -->
<BR>

   <!--outer table is for border purposes-->
   <TABLE WIDTH='100%' BGCOLOR='#888888' BORDER='0' CELLSPACING='0' CELLPADDING='1'><TR><TD>
    <TABLE BGCOLOR='#FFFEE8' WIDTH='100%'  BORDER='0' CELLSPACING='0' CELLPADDING='0'><TR><TD>

<div class='subheadingBar' class='windowSize'>
<A NAME=bosTau4></A>Cow Oct. 2007 (Baylor 4.0/bosTau4) (bosTau4, Baylor Release 4.0)     </div>
     <TABLE BGCOLOR='#FFFEE8' WIDTH='100%' CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
     <TR><TD WIDTH=10>&nbsp;</TD><TD>


<TABLE CELLPADDING=4 style='border:1px solid #aaaaaa; width:100%;'>
<TR><TD nowrap><B style='color:#006666;'>Track/Table Name</B></TD>
    <TD nowrap><B style='color:#006666;'>Release Date</B>
</TD></TR>
<TR valign=top><TD align=left>
ENCODE NHGRI Elnitski Bi-directional Promoters</td>
<td>30 Jul 2009</td>
</tr>
</table>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>
	</TD></TR></TABLE>
	
</div>
</TD></TR></TABLE>


<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('require', 'displayfeatures');
  ga('create', 'UA-4289047-6', 'auto');
  ga('send', 'pageview');

</script></BODY></HTML>
