<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="ENCODE Project at UCSC: Data Submission">
    <META name="keywords" content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
    <TITLE>ENCODE Project at UCSC: Data Submission</TITLE>
    <LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
  <A NAME="TOP"></A>

<!-- Header Images and links -->
    <TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
      <TR>
        <TH COLSPAN=2 ALIGN="left">&nbsp;&nbsp;
          <A HREF="http://www.genome.gov/10005107" TARGET=_BLANK><IMG SRC="../images/encodelogo.gif" height=50 ALT="ENCODE Project at NHGRI"></A>
          &nbsp;&nbsp;
          <A HREF="/encode/"><IMG SRC="../images/encode.jpg" ALT="ENCODE Project at UCSC"></A>
        </TH>
      </TR>

<!-- Topbar menu -------------->
      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/encode.topbar.html"--></TD></TR>

<!-- Overview ------------------------------------------------------->
      <TR HEIGHT="100%" VALIGN="top"><TD VALIGN="top">
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	      <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
		  <TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Submission of ENCODE data </B></FONT>
		  </TD></TR> 
		</TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
		  <TR><TH HEIGHT=3></TH></TR> 
		  <TR><TD WIDTH=10></TD><TD> 
		    <P>
		    We are looking forward to the submission of
                    ENCODE-related data, and are ready to help
                    throughout the process.  We recommend the
                    following steps: 
		    <UL>
			<LI>Choose an appropriate format 
			(<A HREF="#BED">BED</A>, <A HREF="#GFF">GFF</A>,
		        <A HREF="#GTF">GTF</A>, <A HREF="#MAF">MAF</A>, or 
			<A HREF="#WIG">WIG</A>) for your data from the
		        descriptions below and create a 
		        file in that format. The submitted data file should be
			in plain-text (or compressed plain-text) format.</LI>

		        <LI>Data coordinates should be based on the NCBI
			Build 35 assembly (May 2004, hg17). If necessary, 
			data in BED format from previous assemblies may be
			transformed to the Build 35 assembly with the UCSC
		        <a href="/cgi-bin/hgLiftOver" >liftOver</a>
		        tools.</LI>

		        <LI>Test your file by loading it into the
                        browser as a <a
                        href="http://genome.ucsc.edu/goldenPath/help/customTrack.html">
			custom track</a>.</LI>

			<LI>Generate a description page to accompany
			your data. The page should contain a brief
			description of the data, methods used
			to generate the data, the techniques used to
			verify the data, acknowledgements of the
			individuals or organizations involved in data
			collection and analysis, and any related literature 
			references (optional). Use the 
			<A HREF="description.txt">HTML template</A> 
			to organize and format your text.

		        <LI>When your data is displayed as you desire,
		        please <A HREF="mailto:encode@soe.ucsc.edu">
		        contact us</A> by email to encode@soe.ucsc.edu,
                        to initiate loading the tracks into the
		        database.  This makes the data available to
		        the rest of the ENCODE community and to the
		        public.</LI> 

                        <LI>
                        The status of your track at UCSC can be checked on the
                        <A TARGET=_BLANK HREF="trackStatus.html">
                        ENCODE data status page</A>.
                        </LI>
		    </UL>
                  </TD><TD WIDTH=15></TD></TR>
	        </TABLE>
	      <BR></TD></TR>
	    </TABLE>
          </TD></TR>
      </TABLE>
      <BR>

<!-- BED ------------------------------------------------------->
      <!--outer table is for border purposes-->
      <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>
          <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
            <TR><TD>
	      <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
		<TR><TD>
		   <FONT SIZE="4"><A name="BED"></A><B>&nbsp;Browser Extensible Data Format (BED)</B></FONT>
		  </TD></TR>
	 	</TABLE>
	  	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	    	  <TR><TH HEIGHT=10></TH></TR>
		  <TR><TD WIDTH=10>&nbsp;</TD><TD>
		    <P>
		        The BED format was defined for the efficient
		        storage and retrieval of genomic annotations.
		        It provides a flexible way to define the data
		        lines that are displayed in an annotation
		        track.  Please see the
			<A HREF="http://genome.ucsc.edu/goldenPath/help/customTrack.html#BED">
		        BED format description</A> on the 
		        <A HREF="http://genome.ucsc.edu/goldenPath/help/customTrack.html">
		        custom tracks</A> page for more details.
                  </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      <BR></TD></TR>
	    </TABLE>
	  </TD></TR>
        </TABLE>
	<BR>

<!-- GFF ------------------------------------------------------->
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
		  <TR><TD>
		    <FONT SIZE="4"><A name="GFF"></A><B>&nbsp; General Feature Format (GFF) </B> </FONT>
		  </TD></TR>
		</TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
		  <TR><TH HEIGHT=10></TH></TR>
		  <TR><TD WIDTH=10>&nbsp;</TD><TD>
		    <P>
			GFF is used for data where there are a set of
			linked features, such as gene models that have
			introns, exons, promoters, and transcription
			start/end sites.  GFF lines have nine required
			fields that must be tab-separated. If the
			fields are separated by spaces instead of
			tabs, the track will not display correctly.

			Please see the
			<A HREF="http://genome.ucsc.edu/goldenPath/help/customTrack.html#GFF">
		        GFF description</A> on the 
		        <A HREF="http://genome.ucsc.edu/goldenPath/help/customTrack.html">
		        custom tracks</A> page for more details.
                  </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      <BR></TD></TR>
	    </TABLE>
	  </TD></TR>
        </TABLE>
	<BR>

<!-- GTF ------------------------------------------------------->
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>
		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
		  <TR><TD>
		    <FONT SIZE="4"><A name="GTF"></A><B>&nbsp; Gene Transfer Format (GTF)</B></FONT>
		  </TD></TR>
		</TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
		  <TR><TH HEIGHT=10></TH></TR>
		  <TR><TD WIDTH=10>&nbsp;</TD><TD>
		    <P>
			GTF is a refinement of GFF that tightens the
			specification and allows arbitrary
			supplemental information for each gene.
			Please see the
			<A HREF="http://genome.ucsc.edu/goldenPath/help/customTrack.html#GTF">
		        GTF description</A> on the 
		        <A HREF="http://genome.ucsc.edu/goldenPath/help/customTrack.html">
		        custom tracks</A> page for more details.
                  </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      <BR></TD></TR>
	    </TABLE>
	  </TD></TR>
        </TABLE>
	<BR>

<!-- MAF ------------------------------------------------------->
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
	   	  <TR><TD>
	            <FONT SIZE="4"><A NAME="MAF"></A><B>&nbsp; Multiple Alignment Format (MAF)</B></FONT>
		  </TD></TR>
		</TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=10></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
		    <P>
		        The multiple alignment format stores a series
		        of multiple alignments in a format that is
		        easy to parse and relatively easy to read.
		        This format stores multiple alignments at the
		        DNA level between entire genomes. 

			Please see the
			<A HREF="http://genome.ucsc.edu/FAQ/FAQformat.html#format5">
		        MAF description help page</A> 
<!--commented out until MAF-as-custom-track is implemented
			and the 
		        <A HREF="http://genome.ucsc.edu/goldenPath/help/customTrack.html">
		        custom tracks</A> page 
-->
			for more details.
                  </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      <BR></TD></TR>
	    </TABLE>
	  </TD></TR>
        </TABLE>
	<BR>

<!-- WIG ------------------------------------------------------->
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
          <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
              <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
		  <TR><TD>
		    <FONT SIZE="4"><A NAME="WIG"></A><B>&nbsp; Wiggle Format (WIG)</B></FONT>
		  </TD></TR>
		</TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=10></TH></TR>
                  <TR><TD WIDTH=10></TD><TD>
		    <P>
		        The wiggle (WIG) format allows display of
		        continuous-valued data in track format.  This
		        is useful for GC percent, probability scores,
		        and transcriptome data.

			Please see the
			<A HREF="http://genome.ucsc.edu/goldenPath/help/wiggle.html">
		        WIG description help page</A> and the 
		        <A HREF="http://genome.ucsc.edu/goldenPath/help/customTrack.html">
		        custom tracks</A> page for more details.
                  </TD><TD WIDTH=15></TD></TR>
		</TABLE>
	      <BR></TD></TR>
	    </TABLE>
	  </TD></TR>
        </TABLE>
      <BR></TD></TR>
    </TABLE>
  </BODY>
</HTML>
