<H2>Description</H2>
<P>
The <A HREF="http://www.affymetrix.com/estore/browse/products.jsp?navMode=
34000&productId=131452&navAction=jump&aId=productsNav#1_1"
TARGET=_blank>Human Exon 1.0 ST</A> GeneChip contains over 1.4 million probe 
sets
designed to interrogate individual exons rather than the 3' ends of transcripts
as in traditional GeneChips. Exons were derived from a variety of
annotations that have been divided into the classes Core, Extended
and Full. 
<UL> 
  <LI><B>Core:
    </B>RefSeq transcripts, full-length GenBank mRNAs</LI>
  <LI><B>Extended:</B> 
    dbEst alignments, Ensembl annotations, syntenic mRNA from rat and mouse, 
    microRNA annotations, MITOMAP annotations, Vega genes, Vega pseudogenes
  <LI><B>Full:</B>
    Geneid genes, Genscan genes, Genscan Subopt, Exoniphy, RNA genes, SGP genes,
    Twinscan genes
  </LI>
</UL></P> 

<P>
Probe sets are colored by class with the Core probe sets being
the darkest and the Full being the lightest color. Additionally, probe
sets that do not overlap the exons of a transcript cluster, but fall
inside of its introns, are considered bounded by that transcript
cluster and are colored slightly lighter. Probe sets that overlap the
coding portion of the Core class are colored slightly darker.</P>
<P>
The microarray track using this probe set can be displayed by turning
on the <B>Affy All Exon</B> track.</P>

<H2>Credits and References</H2>
<P>
The exons interrogated by the probe sets displayed in this track are
from the Affymetrix Human Exon 1.0 GeneChip and were derived from a
number of sources. In addition to the millions of cDNA sequences
contributed to the 
<A HREF="https://www.ncbi.nlm.nih.gov/Genbank/index.html" 
TARGET=_blank>GenBank</A>, 
<A HREF="https://www.ncbi.nlm.nih.gov/dbEST/" TARGET=_blank>dbEst</A> and 
<A HREF="https://www.ncbi.nlm.nih.gov/RefSeq/" TARGET=_blank>RefSeq</A> 
databases by
individual labs and scientists, the following annotations were used:
<P>
<A HREF="http://www.ensembl.org/index.html" 
TARGET=_blank><B>Ensembl</B></A><B>: </B>
Hubbard T, Barker D, Birney E, Cameron G, Chen Y, Clark L, Cox T, Cuff J,
Curwen V, Down T <em>et al.</em>.
<A HREF="https://academic.oup.com/nar/article/30/1/38/1332872"
TARGET=_blank>The Ensembl genome database project</A>.
<em>Nucleic Acids Research</em>. 2002 Jan 1;30(1):38-41.</P>
<P>
<B>Exoniphy:</B> Siepel, A., Haussler, D. 
<A HREF="http://compgen.cshl.edu/~acs/recomb2004.pdf"
TARGET=_blank>Computational identification of evolutionarily conserved 
exons</A>.
<em>Proc. 8th Int'l Conf. on Research in Computational Molecular Biology</em>, 
177-186 (2004).</P>
<P>
<B><A HREF="http://genome.crg.es/geneid.html" TARGET=_blank>Geneid Genes</A>:</B>
Parra, G., Blanco, E., Guigo, R. 
<A HREF="https://genome.cshlp.org/content/10/4/511.abstract"
TARGET=_blank>Geneid in Drosophila</A>.
<em>Genome Res.</em> <B>10</B>(4), 511-515 (2000).</P>
<P>
<B><A HREF="http://genes.mit.edu/GENSCAN.html" TARGET=_blank>Genscan Genes</A>:
</B> Burge, C., Karlin, S. 
<A HREF="https://www.sciencedirect.com/science/article/pii/S0022283697909517"
TARGET=_blank>Prediction of Complete Gene Structures in Human Genomic DNA</A>.
<em>J. Mol. Biol.</em> <B>268</B>(1), 78-94 (1997).</P>
<P>
<B><A HREF="http://www.mirbase.org/ftp.shtml" 
TARGET=_blank>microRNA</A>:
</B>  Griffiths-Jones, S. 
<A HREF="https://academic.oup.com/nar/article/32/suppl_1/D109/2505171" 
TARGET=_blank>The microRNA Registry</A>. 
<em>Nucl. Acids Res.</em> <B>32</B>, D109-D111 (2004).</P>
<P>
<B><A HREF="https://www.mitomap.org/MITOMAP" TARGET=_blank>MITOMAP</A>:
</B> Brandon, M. C., Lott, M. T., Nguyen, K. C., Spolim, S., Navathe, S. B., 
Baldi, P. &amp; Wallace, D. C.
<A HREF="https://academic.oup.com/nar/article/33/suppl_1/D611/2505354"
TARGET=_blank>MITOMAP: a human mitochondrial genome database--2004 update</A>
<em>Nucl. Acids Res.</em> <B>33</B>(Database Issue):D611-613 (2005). </P>
<P>
<B><A HREF="http://eddylab.org" TARGET=_blank>RNA Genes</A>:
</B> Lowe, T. M., Eddy, S. R. 
<A HREF="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC146525/"
TARGET=_blank>tRNAscan-SE: A Program for Improved Detection of Transfer RNA 
Genes in Genomic Sequence</A>.
<em>Nucleic Acids Res.</em>, <B>25</B>(5), 955-964 (1997).</P>
<P>
<B><A HREF="http://grib.imim.es" TARGET=_blank>SGP Genes</A>: </B> 
Wiehe, T., Gebauer-Jung, S.,  Mitchell-Olds, T., Guigo, R. 
<A HREF="https://genome.cshlp.org/content/11/9/1574.abstract"
TARGET=_blank>SGP-1: prediction and validation of homologous genes based on 
sequence alignments</A>.
<em>Genome Res.</em>, <B>11</B>(9), 1574-83 (2001).</P>
<P>
<B><A HREF="http://mblab.wustl.edu/" TARGET=_blank>Twinscan Genes</A>:
</B> Korf, I., Flicek, P., Duan, D., Brent, M.R. 
<A HREF="https://academic.oup.com/bioinformatics/article/17/suppl_1/S140/261903"
TARGET=_blank>Integrating genomic homology into gene structure prediction</A>.
<em>Bioinformatics</em> 17, S140-148 (2001).
<P>
<B><A HREF="http://vega.archive.ensembl.org/index.html" TARGET=_blank>Vega Genes 
and Pseudogenes</A>: </B>The HAVANA group, 
<A HREF="https://www.sanger.ac.uk/" TARGET=_blank>Wellcome Trust Sanger 
Institute</A>.</P>
