<H2>Description</H2>
<P>
This track displays transcriptome data from tiling GeneChips produced
by <A HREF="http://www.affymetrix.com/estore/"
TARGET=_blank>Affymetrix</A>. For the ten chromosomes 6, 7, 13, 14,
19, 20, 21, 22, X, and Y, more than 74 million probes were tiled every
5 bp in non-repeat-masked areas and hybridized to mRNA from 11
different cell lines (some cell lines were female and contain no data
for chrY). For HepG2, some samples were depleted
of polyA transcripts rather than enriched. For experimental details
and results, see Cheng <em>et al</em>. in the References section
below.</P>
<BR>
<BR>
<CENTER><img src='../images/affyTxn2Example.jpg'></CENTER>

<H2> Display Conventions and Configuration</H2>
<P>
This annotation follows the display conventions for composite 
tracks. The subtracks within this annotation may be configured in a variety of 
ways to highlight different aspects of the displayed data. The graphical 
configuration options are shown at the top of the track description page, 
followed by a list of subtracks. For more information about the 
graphical configuration options, click the 
<A HREF="../goldenPath/help/hgWiggleTrackHelp.html">Graph configuration 
help</A> link. To display only selected subtracks, uncheck the boxes next to 
the tracks you wish to hide. </P>
<P>
Each subtrack is colored blue in areas that are thought to be transcribed
at a statistically significant level as described in the accompanying
transfrags (transcribed fragments) track. Transfrags that have a
significant blat hit elsewhere in the genome are displayed in a
lighter shade of blue, and transfrags that overlap putative
pseudogenes are colored an even lighter shade of blue. All other
regions of the track are colored brown. While the raw data are based
on perfect match minus mismatch (PM - MM) probe values and may contain
negative values, the track has a minimum value of zero for visualization
purposes.</P>

<H2>Methods</H2>
<P>
For each data point, probes within 30 bp on either side were used to
improve the estimate of expression level for a particular probe. This
helped to smooth the data and produce a more robust estimate of the
transcription level at a particular genomic location. The following
analysis method was used:
<OL>
<LI>Replicate arrays were quantile-normalized and the median
intensity (using both PM and MM intensities) of each array was
scaled to a target value of 44.
<LI>The expression level was estimated for each mapped probe position by 
  <UL>
  <LI>collecting all the probe pairs that fell within a window of &#177;
  30 bp
  <LI>calculating all non-redundant pairwise averages of PM - MM
  values of all probe pairs in the window
  <LI>taking the median of all resulting pairwise averages
  </UL>
<LI>The resulting signal value is the Hodges-Lehmann estimator
associated with the Wilcoxon signed-rank statistic of the PM - MM
values that lie within &#177; 30 bp of the sliding window centered at
every genomic coordinate.
</OL></P>

<H2>Credits</H2>
<P>
Data generation and analysis was performed by the transcriptome group at 
Affymetrix: Bekiranov S, Brubaker S, Cheng J, Dike S, Drenkow J, Ghosh S, Gingeras T, Helt G, Kampa D,
Kapranov P, Long J, Madhavan G, Manak J, Patel S, Piccolboni A, Sementchenko V, and Tammana H.</P>

<P>UCSC annotation performed by Chuck Sugnet.

<H2>References</H2>

<p>
Cheng J, Kapranov P, Drenkow J, Dike S, Brubaker S, Patel S, Long J, Stern D, Tammana H, Helt G
<em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/15790807" target="_blank">
Transcriptional maps of 10 human chromosomes at 5-nucleotide resolution</a>.
<em>Science</em>. 2005 May 20;308(5725):1149-54.
</p>
