<H2>Description</H2>
<P>
This track shows alternate sequence indicated by  
<a href="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/" 
target=_blank>GRC</a> (Genome Reference Consortium) to the GRCh37 (hg19) 
reference sequence in an effort to provide the best possible reference 
assembly for the human genome. GRC does this by both generating multiple 
representations (Haplotypes/alternate loci) for regions that are too complex 
to be represented by a single path and by releasing regional fixes to short 
sections of DNA known as Patches. This allows users who are interested in a 
specific locus to get an improved representation without affecting users who 
need chromosome coordinate stability.
</p>

<a href='ftp://hgdownload.cse.ucsc.edu/gbdb/hg19/hg19Patch5/' target=_blank>Download</a>
the GRC Patch Release track data sets from the Genome Browser downloads server.

<h3>Haplotypes</h3>
<p>
Items in the <em>Haplotype</em> track show sequence variation present in the
population for certain sections of the genome. Items in this track were either 
designated as alternate loci or novel patches by GRC. To view the alternate 
sequence, click on the &quot;Fetch alternate sequence&quot; link that appears on the sequence 
details pages.
</p>

<h3>Patches</h3>
<p>
Items in the <em>Patch</em> track indicate areas of the sequence that have 
been corrected by GRC and will be updated to new sequence in the next full 
human assembly release. To view the new, corrected sequence, click on the 
&quot;Fetch alternate sequence&quot; link that appears on the sequence 
details pages.
</p>

<H3>Alignment</H3>
<P>
The <em>Alignment</em> track is the PSL representation of the <em>Chain</em> 
track.  This PSL representation allows the side-by-side alignment to be
viewed.  It was created from the Chain track data using
the <em>chainToPsl</em> kent source utility. This track follows the display 
conventions for 
<a href="../goldenPath/help/hgTracksHelp.html#PSLDisplay" target=_blank>
PSL alignment tracks</a>.
</p>

<H3>Chain Track</H3>
<P>
The chain track shows the alignment of human genome alternate sequence to the 
human $date genome sequence using a gap scoring system that allows
longer gaps than traditional affine gap scoring systems. It can also
tolerate gaps in both human and the alternate sequence simultaneously.
</p><P>
The chain track displays boxes joined together by either single or
double lines. The boxes represent aligning regions.
Single lines indicate gaps that are largely due to a deletion in the
human assembly or an insertion in the alternate sequence.
Double lines represent more complex gaps that involve substantial
sequence in both sequences.
</p><P>
In the "pack" and "full" display
modes, the individual feature names indicate the chromosome, strand, and
location (in thousands) of the match for each matching alignment. 
By default, the chains to chromosome-based assemblies are colored
based on which chromosome they map to in the aligning organism. To turn
off the coloring, check the &quot;off&quot; button next to: Color
track based on chromosome (click "Chains" text in controls to view options).
</P><P>
To display only the chains of one chromosome in the aligning
sequence, enter the name of that chromosome (e.g. chr4) in box next to:
Filter by chromosome (click "Chains" text in controls to view options).
</P>

<H3>Net Track</H3>
<P>
The net track shows the best human/alternate chain for 
every part of the alternate sequence. It is useful for
finding orthologous regions and for studying genome
rearrangement.  The human sequence used in this annotation is from
the $date assembly.
</P><P>
In full display mode, the top-level (level 1)
chains are the largest, highest-scoring chains that
span this region.  In many cases gaps exist in the
top-level chain.  When possible, these are filled in by
other chains that are displayed at level 2.  The gaps in
level 2 chains may be filled by level 3 chains and so
forth.
</P><P>
In the graphical display, the boxes represent ungapped
alignments; the lines represent gaps.  Click
on a box to view detailed information about the chain
as a whole; click on a line to display information
about the gap.  The detailed information is useful in determining
the cause of the gap or, for lower level chains, the genomic
rearrangement.
</P><P>
Individual items in the display are categorized as one of four types
(other than gap):
</P><P><UL>
<LI><B>Top</B> - the best, longest match. Displayed on level 1.
<LI><B>Syn</B> - line-ups on the same chromosome as the gap in the level above
it.
<LI><B>Inv</B> - a line-up on the same chromosome as the gap above it, but in
the opposite orientation.							
<LI><B>NonSyn</B> - a match to a chromosome different from the gap in the
level above.
</UL></P>

<H2>Methods</H2>
<h3>Haplotype/Patch tracks</h3>
<P>
The locations of the Patches and Haplotypes were established by NCBI and 
specified in the files at:<br> 
<A HREF="ftp://ftp.ncbi.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Homo_sapiens/GRCh37/"
TARGET=_blank>ftp://ftp.ncbi.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Homo_sapiens/
GRCh37/</A>ALT_REF_LOCI_*/placed_scaffolds/alt_locus_scaf*primary.pos
</p>

<p>And the patch release site:<br>
<A HREF="ftp://ftp.ncbi.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Homo_sapiens/"
TARGET=_blank>ftp://ftp.ncbi.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Homo_sapiens/</A>
GRCh37.p*/PATCHES/alt_scaffolds/alt_scaffold_placement.txt
</p>

<H3>Chain track</H3>
<P>
The lastz alignments between the human sequence and the alternate sequence
were restricted to only the region that the haplotype overlaps the human
sequence or the region of the human sequence that is replaced by the
updated patch sequence.
The resulting alignments were converted into axt format using the lavToAxt
program. The axt alignments were fed into axtChain, which organizes all
alignments between the single human region of the alignment and a single
region from the alternate sequence into a group and creates a kd-tree out
of the gapless subsections (blocks) of the alignments. A dynamic program
was then run over the kd-trees to find the maximally scoring chains of these
blocks.

$matrix

<P>
Chains scoring below a minimum score of '$chainMinScore' were discarded;
the remaining chains are displayed in this track.  The linear gap
matrix used with axtChain:<BR>
$chainLinearGap
</P>

<H3>Net track</H3>
<P>
Chains were derived from lastz alignments, using the methods
described in the chain track description above, and sorted with the 
highest-scoring chains in the genome ranked first. The program
chainNet was then used to place the chains one at a time, trimming them as 
necessary to fit into sections not already covered by a higher-scoring chain. 
During this process, a natural hierarchy emerged in which a chain that filled 
a gap in a higher-scoring chain was placed underneath that chain. The program 
netSyntenic was used to fill in information about the relationship between 
higher- and lower-level chains, such as whether a lower-level
chain was syntenic or inverted relative to the higher-level chain. 
The program netClass was then used to fill in how much of the gaps and chains 
contained <em>N</em>s (sequencing gaps) in one or both species and how much
was filled with transposons inserted before and after the two sequences 
diverged.
</P>

<H2>Credits</H2>
<P>
Lastz (previously known as blastz) was developed at
<A HREF="http://bio.cse.psu.edu/" 
TARGET=_blank>Pennsylvania State University</A> by 
Minmei Hou, Scott Schwartz, Zheng Zhang, and Webb Miller with advice from
Ross Hardison.</P>
<P>
Lineage-specific repeats were identified by Arian Smit and his 
<A HREF="http://www.repeatmasker.org" TARGET=_blank>RepeatMasker</A>
program.</P>
<P>
The axtChain program was developed at the University of California at 
Santa Cruz by Jim Kent with advice from Webb Miller and David Haussler.</P>
<P>
The browser display and database storage of the chains and nets were created
by Robert Baertsch and Jim Kent.</P>
<P>
The chainNet, netSyntenic, and netClass programs were
developed at the University of California
Santa Cruz by Jim Kent.</P>
<P>

<H2>References</H2>
<P>
Chiaromonte F, Yap VB, Miller W. 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" 
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. 
<em>Pac Symp Biocomput.</em> 2002;:115-26.</P>
<P>
Kent WJ, Baertsch R, Hinrichs A, Miller W, Haussler D.
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" 
TARGET=_blank>Evolution's cauldron: Duplication, deletion, and rearrangement
in the mouse and human genomes</A>.
<em>Proc Natl Acad Sci U S A</em>. 2003 Sep 30;100(20):11484-9.</P>
<P>
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison RC,
Haussler D, Miller W.
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103" 
TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>. 
<em>Genome Res.</em> 2003 Jan;13(1):103-7.</P>
</P>
