<h2>Description</h2>

<p>
Enrichment of large copy number variants (CNVs) has been linked to severe pediatric disease
including developmental delay, intellectual disability and autism spectrum disorder. The
association of individual loci with specific disorders, however, has still been problematic.
</p>

<p>
This track shows CNVs from cases of developmental delay along with healthy control sets from two
separate studies. The study by Cooper <em>et al</em>. (2011) analyzed samples from 15,767 children
with various developmental disabilities and compared them with samples from 8,329 adult controls to
produce a detailed genome-wide morbidity map of developmental delay and congenital birth defects.
The study by Coe <em>et al</em>. (2014) further expanded the morbidity map by analyzing 13,318 new
case samples along with 11,255 new controls.
</p>

<h2>Display Conventions and Configuration</h2>

<p>
This is a composite track consisting of a Case subtrack and a Control subtrack. To turn a subtrack
on or off, toggle the checkbox to the left of the subtrack name in the track controls at the top of
the track description page.
</p>

<p>
Items in this track are colored <b><font color="red">red for copy number loss</font></b> and
<b><font color="blue">blue for copy number gain</font></b>.
</p>

<h2>Methods</h2>

<p>
The samples were analyzed using nine different CGH platforms with initial CNV calls filtered as
described in Coe <em>et al</em>. (2014).
</p>

<p>
Final CNV calls were decoupled from identifying information and submitted to dbVar as
<a href="https://www.ncbi.nlm.nih.gov/dbvar/studies/nstd54/" target="_blank">nstd54</a> and
<a href="https://www.ncbi.nlm.nih.gov/dbvar/studies/nstd100/" target="_blank">nstd100</a>
for unrestricted release.
</p>

<p>
The 15,767 case individuals from the Cooper study comprise nstd54 sampleset 1, while the 8,329
control individuals from the Cooper study comprise nstd54 samplesets 2-12. The 13,318 case
individuals from the Coe study were combined with the Cooper case individuals to comprise nstd100
sampleset 1. The 11,255 control individuals from the Coe study comprise nsdt100 samplesets 2 and 3.
</p>

<p>
The Case subtrack was constructed using nstd100 sampleset 1. The Control subtrack was constructed by
combining nstd100 samplesets 2 and 3 with nstd54 samplesets 2-12.
</p>

<h2>Credits</h2>

<p>
We would like to thank Gregory Cooper, Brad Coe and the
<a href="https://eichlerlab.gs.washington.edu/" target="_blank">Eichler Lab</a> at the University of
Washington for providing the data for this track.
</p>

<h2>References</h2>

<p>
Coe BP, Witherspoon K, Rosenfeld JA, van Bon BW, Vulto-van Silfhout AT, Bosco P, Friend KL, Baker C,
Buono S, Vissers LE <em>et al</em>.
<a href="http://www.nature.com/ng/journal/v46/n10/abs/ng.3092.html" target="_blank">
Refining analyses of copy number variation identifies specific genes associated with developmental
delay</a>.
<em>Nat Genet</em>. 2014 Oct;46(10):1063-71.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25217958" target="_blank">25217958</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4177294/" target="_blank">PMC4177294</a>
</p>

<p>
Cooper GM, Coe BP, Girirajan S, Rosenfeld JA, Vu TH, Baker C, Williams C, Stalker H, Hamid R, Hannig
V <em>et al</em>.
<a href="http://www.nature.com/ng/journal/v43/n9/abs/ng.909.html" target="_blank">
A copy number variation morbidity map of developmental delay</a>.
<em>Nat Genet</em>. 2011 Aug 14;43(9):838-46.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21841781" target="_blank">21841781</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3171215/" target="_blank">PMC3171215</a>
</p>
