<h2>Description</h2>
The track <b>NCBI dbVar Curated Common SVs: Conflicts with Pathogenic</b> highlights loci where
common copy number variants from
<a target=_blank href="https://www.ncbi.nlm.nih.gov/dbvar/studies/nstd186/">nstd186 (NCBI Curated 
Common Structural Variants)</a> overlap with structural Variants with clinical assertions,
submitted to ClinVar by external labs <a target=_blank 
href="https://www.ncbi.nlm.nih.gov/dbvar/content/var_summary/#nstd102">(Clinical Structural 
Variants - nstd102)</a>.
</p>

<p>
<em>Overlap</em> in the track refers to reciprocal overlap between variants in the <b><em>common</em>
(NCBI Curated Common Structural Variants)</b> versus <b><em>clinical</em> (ClinVar Long Variants)</b>
tracks. Reciprocal overlap values can be anywhere from 10% to 100%.
</p>

<p>
For more information on the number of variant calls and latest statistics for nstd186 see
<a target=_blank href="https://www.ncbi.nlm.nih.gov/dbvar/content/common_summary/">Summary of nstd186</a>
(NCBI Curated Common Structural Variants).
</p>

<h2>Display Conventions and Configuration</h2>

<p>
Items in all subtracks follow the same conventions: items are colored by variant type, and are
based on the dbVar colors described in the
<a target="_blank" href="https://www.ncbi.nlm.nih.gov/dbvar/content/overview/">dbVar Overview page</a>.
<b><font color="red">Red</font></b> for copy number loss or deletion,
<b><font color="blue">blue</font></b> for copy number gain or duplication, and
<b><font color="#662180">violet</font></b> for copy number variation. 
</p>

<p>
<b>Mouseover</b> on items indicates genes affected, size, variant type, and allele frequencies (AF). 
All tracks can be filtered according to the <b>variant length</b>, <b>variant type</b> and 
<b>variant overlap</b>. This last filter defines four bins within that range from which the 
user can choose.
</p>
 

<h2>Data Access</h2>
The raw data can be explored interactively with the
<a href="https://genome.ucsc.edu/cgi-bin/hgTables">Table Browser</a>, or the
<a href="https://genome.ucsc.edu/cgi-bin/hgIntegrator">Data Integrator</a>. For automated analysis,
the data may be queried from our
<a href="https://genome.ucsc.edu/goldenPath/help/api.html">REST API</a>. The data can also be found
directly from the dbVar nstd186 <a target=_blank
href="https://www.ncbi.nlm.nih.gov/dbvar/content/common_summary/#data_access">data access</a>, as
well as in the <a href="https://genome.ucsc.edu/cgi-bin/hgHubConnect?hgsid=910263067_X3tU6DiJPhKMr71o1h9Zj2K5XQ4u"
>dbVar Track Hub</a>, where additional subtracks are included. For questions about dbVar track
data, please contact <em>dbvar@ncbi.nlm.nih.gov</em>.
</p>

<h2>Credits</h2>
Thanks to the dbVAR team at NCBI, especially John Lopez and Timothy Hefferon for technical 
coordination and consultation, and to Christopher Lee, Anna Benet-Pages, and Daniel Schmelter of 
the Genome Browser team for engineering the track display.
</p>
 
<h2>References</h2>
<p>
Lappalainen I, Lopez J, Skipper L, Hefferon T, Spalding JD, Garner J, Chen C, Maguire M, Corbett M,
Zhou G <em>et al</em>.
<a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gks1213" target="_blank">
DbVar and DGVa: public archives for genomic structural variation</a>.
<em>Nucleic Acids Res</em>. 2013 Jan;41(Database issue):D936-41.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23193291" target="_blank">23193291</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3531204/" target="_blank">PMC3531204</a>
</p>
 
