<H2>Description</H2>

<div class="warn-note" style="border: 2px solid #9e5900; padding: 5px 20px; background-color: #ffe9cc;">
 <P><span style="font-weight: bold; color: #c70000;">NOTE:</span><br> 
 While the DECIPHER database is 
 open to the public, users seeking information about a personal medical or
 genetic condition are urged to consult with a qualified physician for
 diagnosis and for answers to personal questions.
 </P>
 <P>Because the UCSC Genes mappings for CNVs are based on associations from
 RefSeq and UniProt, they are dependent on any interpretations from those
 sources.  Furthermore, because many DECIPHER records refer to multiple gene
 names, or syndromes not tightly mapped to individual genes, the associations
 in this track should be treated with skepticism and any conclusions
 based on them should be carefully scrutinized using independent
 resources.
 </P>
 <p><b>Data Display Agreement Notice</b><br>
 The CNV/SNV data are only available for display in the Browser, and not for bulk
 download.  Access to bulk data may be obtained directly from DECIPHER
 (<a href='https://www.deciphergenomics.org/about/data-sharing' target='_blank'
 >https://www.deciphergenomics.org/about/data-sharing</a>) and is subject to a
 Data Access Agreement, in which the user certifies that no attempt to
 identify individual patients will be undertaken.  The same restrictions
 apply to the public data displayed at UCSC in the UCSC Genome Browser;
 no one is authorized to attempt to identify patients by any means.
 </p>
 <p>These data are made available as soon as possible and may be a
 pre-publication release.  For information on the proper use of DECIPHER
 data, please see <a href='https://www.deciphergenomics.org/about/data-sharing'
 target='_blank'>https://www.deciphergenomics.org/about/data-sharing</a>.
 </p>
 <p>The DECIPHER consortium provides these data in good faith as a research
 tool, but without verifying the accuracy, clinical validity, or utility of
 the data.  The DECIPHER consortium makes no warranty, express or implied,
 nor assumes any legal liability or responsibility for any purpose for
 which the data are used.
 </p>
</div>

<P>
The 
<A HREF="https://decipher.sanger.ac.uk" TARGET=_BLANK>DECIPHER</A>
database of submicroscopic chromosomal imbalance 
collects clinical information about chromosomal 
microdeletions/duplications/insertions, translocations and inversions, 
and displays this information on the human genome map.
<p>
The <b>CNVs and SNVs</b> tracks show genomic regions of reported cases and their 
associated phenotype information.  All data have passed the strict
consent requirements of the DECIPHER project and are approved for
unrestricted public release.  Clicking the Patient View ID link
brings up a more detailed informational page on the patient at the 
DECIPHER web site.</p>

<p>
The <b>Population CNVs</b> track shows common copy-number variants (CNVs) and their
population frequencies, lifted over from the hg19 assembly.</p>

<p>
The <strong>Developmental Disorders Genotype-to-Phenotype (DDG2P)</strong> track displays genes
associated with severe developmental disorders. The track can be used to filter genomic sequencing
data from individuals with genetic disorders to identify likely causative variants and accelerate
diagnosis.
</p>

<H2>Display Conventions and Configuration</H2>
<P>
The genomic locations of DECIPHER variants are labeled with the DECIPHER variant descriptions. 
<b>Mouseover</b> on items shows variant details, clinical interpretation, and associated conditions. 
Further information on each variant is displayed on the details page by a click onto any variant. 
</p>

<P>
For the <b>CNVs track</b>, the entries are colored by the <b>type of variant</b>:
<ul>
 <li><b><font color="red">red</font></b> for loss</li>
 <li><b><font color="blue">blue</font></b> for gain</li>
 <li><b><font color="grey">grey</font></b> for amplification</li>
</ul>
</P>

<P>
A light-to-dark color gradient indicates the <b>clinical significance</b> of each variant, with 
the lightest shade being benign, to the darkest shade being pathogenic. Detailed information on the 
CNV color code is described <a href="../../goldenPath/help/hgCnvColoring.html">here</a>.
Items can be filtered according to the size of the variant, variant type, and clinical significance 
using the track <b>Configure</b> options.
</P>
 
<P>
For the <b>SNVs track</b>, the entries are colored according to the estimated <b>clinical significance</b> 
of the variant:
<ul>
 <li><b><font color="black">black</font></b> for likely or definitely pathogenic</li>
 <li><b><font color="#888">dark grey</font></b> for uncertain or unknown</li>
 <li><b><font color="#c8c8c8">light grey</font></b> for likely or definitely benign</li>
 </ul>
</P>

<p>
For the <b>Population CNVs track</b>, genomic variants are visually differentiated to facilitate quick and
clear identification. Variants are colored according to their clinical significance and type:
</P>
<ul>
  <li><font color="#FF0000"><strong>Red</strong></font> - exclusively deletion site. (deletions)</li>
  <li><font color="#0000FF"><strong>Blue</strong></font> - exclusively duplication site. (duplication)</li>
  <li><font color="#808080"><strong>Grey</strong></font> - deletions and duplications site. (del/dup)</li>
</ul>

<P>
The Population CNVs track's <b>mouseover</b> tooltip provides the following information
about the data:
</P>
<ul>
  <li><strong>Position:</strong> Specifies the chromosomal range of the CNV.</li>
  <li><strong>Type of CNV:</strong> Indicates if the variation is a loss, gain, or
      deletions/duplications(del/dup).</li>
  <li><strong>Frequency of CNV:</strong> Reflects how often the CNV occurs in the sampled
      population.</li>
  <li><strong>Number of Observations:</strong> The count of times this CNV was observed in the
      dataset.</li>
  <li><strong>Sample Size of Study:</strong> The total number of samples examined.</li>
</ul>


<p>
For the <b>DDG2P track</b>, items are colored according to the likelihood that the gene-disease
association is true:</p>
<ul>
    <li> <font style="color: green;"><b>Green</b></font> - Definitive</li>
    <li> <font style="color: blue;"><b>Blue</b></font> - Strong</li>
    <li> <font style="color: orange;"><b>Orange</b></font> - Moderate</li>
    <li> <font style="color: red;"><b>Red</b></font> - Limited</li>
    <li> <font style="color: gray;"><b>Gray</b></font> - Refuted</li>
</ul>

<p>Each <b>mouseover</b> tooltip provides the following information:</p>
<ul>
  <li><strong>G2P ID</strong>: Unique identifier assigned by the Gene2Phenotype (G2P) database.</li>
  <li><strong>Variant Consequence</strong>: Predicted effect each allele of a variant has on a
      transcript.</li>
  <li><strong>Disease Name</strong>: Name of the disease associated with the variant.</li>
  <li><strong>PubMed IDs</strong>: Publications associated with the variant.</li>
  <li><strong>Molecular Mechanism</strong>: Description of the molecular processes and interactions
      causing pathogenic effects.</li>
  <li><strong>Allelic Requirements</strong>: Number of alleles required at a locus to produce a
      pathogenic phenotype (e.g., monoallelic, biallelic).</li>
  <li><strong>Date of Last Review</strong>: Most recent date the entry was manually reviewed.</li>
</ul>


<H2>Method</H2>
<P>
Data provided by the DECIPHER project group are imported and processed
to create a simple BED track to annotate the genomic regions associated
with individual patients.
</p>
<p>
Data provided by the Gene2Phenotype project group were imported and processed to create a BED-based track
annotating genomic regions associated with individual patients. Standard genome assembly coordinates
and gene annotations were used to map DDG2P entries to the browser.
</P>


<H2>Contact</H2>
<P>
For more information on DECIPHER, please contact
<A HREF="mailto:&#99;&#111;n&#116;&#97;c&#116;&#64;&#100;&#101;&#99;&#105;p&#104;&#101;&#114;&#103;&#101;&#110;&#111;&#109;&#105;c&#115;.
&#111;&#114;g">
&#99;&#111;n&#116;&#97;c&#116;&#64;&#100;&#101;&#99;&#105;p&#104;&#101;&#114;&#103;&#101;&#110;&#111;&#109;&#105;c&#115;.
&#111;&#114;g</A>
</P>

<h2>Data Access</h2>
<p>
The DECIPHER data access and documentation can be found at
<a href="https://www.deciphergenomics.org/about/downloads" target="_blank">DECIPHER Downloads</a>.
</p>
<p>
Source data for the DDG2P track are available directly from
<a href="https://www.ebi.ac.uk/gene2phenotype/" target="_blank">Gene2Phenotype</a>. 
</p>

<H2>References</H2>
<p>
Firth HV, Richards SM, Bevan AP, Clayton S, Corpas M, Rajan D, Van Vooren S, Moreau Y, Pettett RM,
Carter NP.
<a href="https://www.cell.com/ajhg/abstract/S0002-9297(09)00107-4" target="_blank">
DECIPHER: Database of Chromosomal Imbalance and Phenotype in Humans Using Ensembl Resources</a>.
<em>Am J Hum Genet</em>. 2009 Apr;84(4):524-33.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19344873" target="_blank">19344873</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2667985/" target="_blank">PMC2667985</a>
</p>
<p>
Thormann A, Halachev M, McLaren W, Moore DJ, Svinti V, Campbell A, Kerr SM, Tischkowitz M, Hunt SE,
Dunlop MG <em>et al</em>.
<a href="https://doi.org/10.1038/s41467-019-10016-3" target="_blank">
Flexible and scalable diagnostic filtering of genomic variants using G2P with Ensembl VEP</a>.
<em>Nat Commun</em>. 2019 May 30;10(1):2373.
DOI: <a href="https://doi.org/10.1038/s41467-019-10016-3"
target="_blank">10.1038/s41467-019-10016-3</a>; PMID: <a
href="https://www.ncbi.nlm.nih.gov/pubmed/31147538" target="_blank">31147538</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6542828/" target="_blank">PMC6542828</a>
</p>
