<H2>Description</H2>
<P>
This track shows the forward (+) and reverse (-) strand transcript map of 
intensity scores (estimating RNA abundance) for human NB4 cell total RNA,
and human placental Poly(A)+ RNA, hybridized 
to the Yale MAS (Maskless Array Synthesizer) ENCODE oligonucleotide 
microarray, transcription mapping design #1. This array has 36-mer 
oligonucleotide probes approximately every 36 bp (<i>i.e.</i> 
end-to-end) covering all the non-repetitive DNA sequence of the ENCODE 
regions ENm001-ENm012. See NCBI
<A HREF="http://www.ncbi.nlm.nih.gov/projects/geo/query/acc.cgi?acc=GPL2105"
    TARGET=_blank>GEO GPL2105</A> for details of this array design. </P>
<P>
This transcript map is a combined signal from three biological replicates, 
each with at least two technical replicates. Arrays were hybridized using 
either the standard Nimblegen protocol or the protocol described in Bertone 
<em>et al</em>. (2004). The label of each subtrack in this annotation 
indicates the specific protocol used for that particular data set.</P>

<H2>Display Conventions and Configuration</H2>
<P>
This annotation follows the display conventions for composite 
tracks. The subtracks within this annotation 
may be configured in a variety of ways to highlight different aspects of the 
displayed data. The graphical configuration options are shown at the top of 
the track description page, followed by a list of subtracks.  To display only 
selected subtracks, uncheck the boxes next to the tracks you wish to hide. 
For more information about the graphical configuration options, click the 
<A HREF="../goldenPath/help/hgWiggleTrackHelp.html" TARGET=_blank>Graph
configuration help</A> link.</P>

<H2>Methods</H2>
<P>
A score was assigned to each oligonucleotide probe position by combining 
two or more technical replicates and by using a sliding window 
approach. Within a sliding window of 160 bp (corresponding to 5 
oligos), the hybridization intensities for all replicates of each 
oligonucleotide probe were compared to their respective array median 
score. Within the window and across all the replicates, the number of 
probes above and below their respective median were counted. Using the 
sign test, a one-sided P-value was then calculated and a score defined 
as score=-log(<i>P-value</i>) was assigned to the oligo in the center of 
the window.</P>
<P>
Three independent biological replicates were generated and each was 
hybridized to at least 2 different arrays (technical replicates).</P>

<H2>Verification</H2>
<P>
Reasonable correlation coefficients between replicates were ensured. 
Additionally, transcribed regions (TARs/transfrags) were called and 
compared between technical and biological replicates to ensure 
significant overlap.</P>

<H2>Credits</H2>
<P>
These data were generated and analyzed by the labs of Michael Snyder, 
Mark Gerstein and Sherman Weissman at Yale University.</P>

<H2>References</H2>
<P>
Bertone, P., Stolc, V., Royce, T.E., Rozowsky, J.S., Urban, A.E., Zhu, X., 
Rinn, J.L., Tongprasit, W., Samanta, M. <em>et al</em>.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=15539566&query_hl=7"
TARGET=_blank>Global identification of human transcribed sequences with 
genome tiling arrays</A>. 
<i>Science</i> <B>306</B>(5705), 2242-6 (2004).</P>
<P>
Cheng, J., Kapranov, P., Drenkow, J., Dike, S., Brubaker, S., Patel, S., 
Long, J., Stern, D., Tammana, H. <em>et al</em>.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=15790807&query_hl=5"
TARGET=_blank>Transcriptional maps of 10 human chromosomes at 5-nucleotide 
resolution</A>. 
<i>Science</i> <B>308</B>(5725), 1149-54 (2005).</P>
<P>
Kapranov, P., Cawley, S.E., Drenkow, J., Bekiranov, S., Strausberg, R.L., 
Fodor, S.P. and Gingeras, T.R.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=11988577&query_hl=5"
TARGET=_blank>Large-scale transcriptional activity in chromosomes 21 and 
22</A>. 
<i>Science</i> <B>296</B>(5569), 916-9 (2002).</P>
<P>
Kluger, Y., Tuck, D.P., Chang, J.T., Nakayama, Y., Poddar, R., Kohya, N., 
Lian, Z., Ben Nasr, A., Halaban, H.R. <em>et al</em>.
<A HREF="http://www.pnas.org/cgi/content/full/101/17/6508"
TARGET=_blank>Lineage specificity of gene expression patterns</A>. 
<i>Proc Natl Acad Sci U S A</i> <B>101</B>(17), 6508-13 (2004).</P>
<P>
Rinn, J.L., Euskirchen, G., Bertone, P., Martone, R., Luscombe, N.M., 
Hartman, S., Harrison, P.M., Nelson, F.K., Miller, P. <em>et al.</em>
<A HREF="http://www.genesdev.org/cgi/content/full/17/4/529"
TARGET=_blank>The transcriptional activity of human Chromosome 22</A>. 
<i>Genes Dev</i> <B>17</B>(4), 529-40 (2003).</P>
