<H2>Description</H2>
<P>
This track shows allele frequencies for the four HapMap populations in the 
ten ENCODE regions that have been resequenced for variation. The data for
each population is shown in a separate subtrack:
<UL>
<LI>HapMap Allele Frequencies (CEU): Utah residents with ancestry from northern and western Europe
<LI>HapMap Allele Frequencies (CHB): Han Chinese in Beijing, China
<LI>HapMap Allele Frequencies (JPT): Japanese in Tokyo, Japan
<LI>HapMap Allele Frequencies (YRI): Yoruba in Ibadan, Nigeria
</UL></P>
<P>
The ENCODE regions targeted in this annotation include: 
<UL>
<LI>ENr112 (chr2)
<LI>ENr131 (chr2)
<LI>ENr113 (chr4)
<LI>ENm010 (chr7)
<LI>ENm013 (chr7)
<LI>ENm014 (chr7)
<LI>ENr321 (chr8)
<LI>ENr232 (chr9)
<LI>ENr123 (chr12)
<LI>ENr213 (chr18)
</UL></P>
<P>
See the Methods section for a discussion of the scoring method used in this 
annotation.</P>

<P>The data set combines SNPs from the HapMap resequencing project, in addition to
SNPs discovered previously.

<H2>Display Conventions and Configuration</H2>
<P>
The complete list of subtracks available in this annotation is shown at
the top of the track description page. To display
only selected subtracks, uncheck the boxes next to the tracks you wish to
hide.</P>
<P>
Allele locations are indicated by tickmarks using a grayscale coloring 
scheme based on score, where darker shading indicates a higher score. A lower
score indicates little or no variation; a higher score indicates a split 
between the reference and variant observations in the population.
The track details page for an individual allele displays the variant and
reference sequences, the allele frequencies, the origination of the data,
and the total sample count.</P>

<H2>Methods</H2>
<P>
See the <A HREF="http://www.hapmap.org" TARGET=_blank>International HapMap 
Project</A> website for information about how these data were collected and
analyzed. </P>
<P>
The score calculation in this annotation is a function of the minor allele 
frequency (maf), which varies from 0.0 to 0.5. The score has been normalized 
to a range of 500 to 1000 using the formula <TT>score = 500 + (maf * 1000)</TT>.
Thus, a score of 500 indicates no variation;  a score of 1000 indicates an even 
split between reference and variant observations in the population.</P>

<H2>Credits</H2>
<P>
These data were obtained from HapMap public release 16c.1. Thanks to the
International HapMap Project for making this information available.</P>
