<H2>Description</H2>
<P>
This track shows the locations of the 25 bp probes on the Affymetrix ENCODE 
tiling array, which are spaced every 22 bp on average.  The 
chip, which was produced using Affymetrix GeneChip technology, has 737,680 
probes representing all the non-repetitive DNA sequence of the ENCODE 
regions. This chip was designed for high throughput experiments to explore the 
human transcriptome at high resolution. The probes represent all the 
transcribed regions, including mRNAs as well as non-coding RNAs that are used 
both structurally and in the regulation of gene expression. Disruption of these
structures or changes in the levels of transcription or translation may play a 
role in disease pathogenesis; therefore, this array is a valuable tool for the 
discovery and elucidation of disease processes.</P>

<H2>Display Conventions and Configuration</H2>
<P>
Probe locations are indicated by solid blocks in the graphical display.</P>

<H2>Methods</H2>
<P>
Probe positions were provided by Affymetrix, and the sequence was verified upon
mapping to the genome. The array can be utilized to study transcribed
regions (see Affy RNA Signal and Affy Transfrags tracks), transcription factor 
binding sites (Affy pVal and Affy Sites tracks), sites of chromatin 
modification, sites for DNA methylation and chromosomal origins of replication.
</P>

<H2>Credits</H2>
<P>
This chip was generated and analyzed by the Gingeras/Struhl
collaboration with the Tom Gingeras group at 
<A HREF="http://www.affymetrix.com/" TARGET=_blank>Affymetrix</A> and the 
<A HREF="http://www.hms.harvard.edu/dms/bbs/fac/struhl.html"
TARGET=_blank>Kevin Struhl</A> group at Harvard Medical School.</P>

<H2>References</H2>
<P>
Please see the 
<A HREF="http://transcriptome.affymetrix.com/local_index.html"
TARGET=_blank>Affymetrix Transcriptome</A> site for a project overview and
additional references to Affymetrix tiling array publications.</P>
<P>
Bolstad, B. M., Irizarry, R. A., Astrand, M., and Speed, T. P. 
<A HREF="http://bioinformatics.oupjournals.org/cgi/content/abstract/19/2/185" 
TARGET=_blank>A comparison of normalization methods for high density 
oligonucleotide array data based on variance and bias</A>. 
<em>Bioinformatics</em> <B>19</B>(2), 185-193 (2003).</P>
<P>
Cawley, S., Bekiranov, S., Ng, H. H., Kapranov, P., Sekinger,
E. A., Kampa, D., Piccolboni, A., Sementchenko, V., Cheng, J.,
Williams, A. J., <em>et al</em>. 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=14980218" 
TARGET=_blank>Unbiased mapping of transcription factor binding sites along 
human chromosomes 21 and 22 points to widespread regulation of noncoding 
RNAs</A>. 
<em>Cell</em> <B>116</B>(4), 499-509 (2004).</P>
<P>
Kapranov, P., Cawley, S. E., Drenkow, J., Bekiranov, S., Strausberg,
R.  L., Fodor, S. P., and Gingeras, T. R. 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=11988577" 
TARGET=_blank>Large-scale transcriptional activity in chromosomes 21 and 
22</A>. 
<em>Science</em> <B>296</B>(5569), 916-919 (2002).</P>
