<H2> Description </H2>
<P>
This track shows the locations of known and predicted non-protein-coding RNA 
genes and pseudogenes that fall within the ENCODE regions. It contains all 
information in Sean Eddy's RNA Genes track for these regions, combined with 
computational predictions generated by Jakob Skou Pedersen's EvoFold algorithm. 
In addition to the fields contained in the RNA Genes track, this track also 
includes ENCODE-related fields describing overlap with transcribed regions and 
repeats.</P>
<P>
Feature types in this annotation include:
<UL>
<LI><B>tRNA:</B> transfer RNA (or pseudogene) 
<LI><B>rRNA:</B>  ribosomal RNA (or pseudogene)
<LI><B>scRNA:</B> small cytoplasmic RNA (or pseudogene)
<LI><B>snRNA:</B> small nuclear RNA (or pseudogene)
<LI><B>snoRNA:</B> small nucleolar RNA (or pseudogene)
<LI><B>miRNA:</B> microRNA (or pseudogene)
<LI><B>misc_RNA:</B> miscellaneous other RNA, such as Xist (or pseudogene)
<LI><B>"-":</B> unknown RNA
</UL></P>

<H2>Display Conventions and Configuration</H2>
<P>
The locations of the RNA genes and pseudogenes are represented by blocks in the 
graphical display, color-coded as follows: 
<UL>
<LI><B><FONT COLOR="000000">Black:</B></FONT> region is Repeatmasked.
<LI><B><FONT COLOR="79AA3D">Green:</B></FONT> region is transcribed.
<LI><B><FONT COLOR="FF0000">Red:</B></FONT> region is from the RNA Genes track 
and is not transcribed.
<LI><B><FONT COLOR="0000FF">Blue:</B></FONT> region is an EvoFold prediction 
and is not transcribed.
</UL></P>
<P>
The display may be filtered to show only those items 
with unnormalized scores that meet or exceed a certain threshhold. To set a 
threshhold, type the minimum score into the text box at the top of the 
description page.</P>

<H2>Methods</H2>
<P>
The RNA Genes track was supplemented with EvoFold predictions and filtered to 
include only those items that lie within the ENCODE regions.  
Regions that are at least 10 percent Repeatmasked are flagged because no 
transcriptional data is available for them.  A region is considered transcribed 
if at least 10 percent overlaps with any Affymetrix transcribed fragment 
(transfrag), derived from six microarray experiments, or Yale 
transcriptionally-active region (TAR), derived from 15 microarray experiments.  
In these cases, each array from which the overlapped transfrags and TARs were 
derived is listed.</P>
<P>
EvoFold is a comparative method that exploits the evolutionary signal
of genomic multiple-sequence alignments for identifying conserved
functional RNA structures. The method makes use of phylogenetic
stochastic context-free grammars (phylo-SCFGs), which are combined
probabilistic models of RNA secondary structure and primary sequence
evolution. The predictions consist both of a specific RNA secondary
structure and an overall score. The overall score is essentially a
log-odd score phylo-SCFG modeling the constrained evolution of
stem-pairing regions and one which only models unpaired regions.</P>
<P>
Two sets of EvoFold predictions are included in this track.  The first,
labeled EvoFold, contains predictions based on the conserved elements of an 
8-way vertebrate alignment of the human, chimpanzee, mouse, rat, dog, chicken, 
zebrafish, and Fugu assemblies. The second set of predictions, TBA23_EvoFold, 
was based on the conserved elements of the 23-way TBA alignments present in the 
ENCODE regions. When a pair of these predictions overlap, only the EvoFold 
prediction is shown.</P>

<H2>Credits</H2>
<P>
These data were kindly provided by Sean Eddy at Washington University,
Jakob Skou Pedersen at UC Santa Cruz, and The Encode Consortium.</P>
<P>
This annotation track was generated by Matt Weirauch.</P>

<H2>References</H2>
<P>
Knudsen, B. and J.J. Hein.  
<A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/15/6/446"
TARGET=_blank>RNA secondary structure prediction using stochastic context-free 
grammars and evolutionary history</A>.
<em>Bioinformatics</em> <B>15</B>(6), 446-54 (1999).</P>
<P>
Pedersen, J.S., Bejerano, G. and Haussler, D. Identification and
classification of conserved RNA secondary structures in the human
genome. <em>(In preparation)</em>.
