<H2>Description</H2>
<P>
This track displays activity levels of 643 putative promoter fragments
in the ENCODE regions, based on high-throughput transient transfection
luciferase reporter assays.  The activity of each putative promoter is
indicated by color, ranging from black (no activity) to red (strong
activity).  Each of the fragments was tested in a panel of 16 cell
lines:
</P><P>
<BLOCKQUOTE>
  <TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4>
  <TR><TH>Cell Line</TH><TH>Classification</TH><TH>Isolated From</TH></TR>
  <TR><TD><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=CRL-1739&Template=cellBiology" TARGET=_blank>AGS</A></TD><TD>gastric adenocarcinoma</TD><TD>stomach</TD></TR>
  <TR><TD><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=CRL-2268&Template=cellBiology" TARGET=_blank>BE(2)-C</A></TD><TD>neuroblastoma</TD><TD>brain (metastatic, from bone marrow)</TD></TR>
  <TR><TD><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=CRL-1690&Template=cellBiology" TARGET=_blank>T98G</A> (CRL-1690)</TD><TD>glioblastoma</TD><TD>brain</TD></TR>
  <TR><TD><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=CRL-1440&Template=cellBiology" TARGET=_blank>G-402</A></TD><TD>renal leiomyoblastoma</TD><TD>kidney</TD></TR>
  <TR><TD><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=CCL-247&Template=cellBiology" TARGET=_blank>HCT 116</A></TD><TD>colorectal carcinoma</TD><TD>colon</TD></TR>
  <TR><TD><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=CRL-9607&Template=cellBiology" TARGET=_blank>HMCB</A></TD><TD>melanoma</TD><TD>skin</TD></TR>
  <TR><TD><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=CCL-121&Template=cellBiology" TARGET=_blank>HT-1080</A></TD><TD>fibrosarcoma</TD><TD>connective tissue</TD></TR>
  <TR><TD><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=HTB-11&Template=cellBiology" TARGET=_blank>SK-N-SH</A> (HTB-11)</TD><TD>neuroblastoma</TD><TD>brain (metastatic, from bone marrow)</TD></TR>
  <TR><TD><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=CCL-2&Template=cellBiology" TARGET=_blank>HeLa</A></TD><TD>adenocarcinoma</TD><TD>cervix</TD></TR>
  <TR><TD><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=HB-8065&Template=cellBiology" TARGET=_blank>HepG2</A></TD><TD>hepatocellular carcinoma</TD><TD>liver</TD></TR>
  <TR><TD><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=HTB-36&Template=cellBiology" TARGET=_blank>JEG-3</A></TD><TD>choriocarcinoma</TD><TD>placenta</TD></TR>
  <TR><TD><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=CRL-1427&Template=cellBiology" TARGET=_blank>MG-63</A></TD><TD>osteosarcoma</TD><TD>bone</TD></TR>
  <TR><TD><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=CCL-171&Template=cellBiology" TARGET=_blank>MRC-5</A></TD><TD>fibroblast</TD><TD>lung</TD></TR>
  <TR><TD><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=CRL-1469&Template=cellBiology" TARGET=_blank>PANC-1</A></TD><TD>epithelioid carcinoma</TD><TD>pancreas (duct)</TD></TR>
  <TR><TD><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=CRL-2235&Template=cellBiology" TARGET=_blank>SNU-182</A></TD><TD>hepatocellular carcinoma</TD><TD>liver</TD></TR>
  <TR><TD><A HREF="https://www.atcc.org/ATCCAdvancedCatalogSearch/ProductDetails/tabid/452/Default.aspx?ATCCNum=HTB-14&Template=cellBiology" TARGET=_blank>U-87 MG</A></TD><TD>glioblastoma-astrocytoma</TD><TD>brain</TD></TR>
  </TABLE>
</BLOCKQUOTE>
</P>

<H2>Methods</H2>
<P>
Promoters in the ENCODE region were predicted using a variation on methods
previously described (Trinklein <em>et al</em>., 2003, Trinklein <em>et 
al</em>., 2004). Using BLAT alignments of human cDNAs in Genbank to the 
genome, those with at least one bp of exon overlap were merged,
generating gene models. The transcription start sites were predicted
by assigning the 5' end of each gene model as one transcription start
site and alternative 5' ends that were at least 500 bp downstream and
supported by full-length cDNAs as other start sites. Promoters were 
defined as the regions approximately 600 bp upstream and 100 bp
downstream of each transcription start site. 
</P>
<P>
Primer3 was used to pick primers yielding approximately 500 bp
amplicons containing the predicted transcription start site. Each
fragment of DNA represented in this track was cloned into a
luciferase reporter vector (pGL3-Basic, Promega) using the BD
Clontech Infusion Cloning System. The Dual Luciferase system
(Promega) was used to co-transfect the experimental DNA along with a
control plasmid expressing Renilla - to control for variation in 
transcription efficiency - in 96-well format into one of the sixteen 
cell types using FuGENE Transfection Reagent (Roche). Each
transfection was done in duplicate. 
</P>
<P>
Data are reported as normalized and log<sub>2</sub> transformed averages of the
Luciferase/Renilla ratio. This normalization was based on the
activity of 102 random genomic fragments (negative controls) derived 
from exons and intergenic regions.  Such a normalization allows
for a meaningful comparison between cell types. The average log transformed 
Luciferase/Renilla ratio was scaled linearly to create a score where the
maximum value is 1000 and the minimum value is 0. This score is arbitrary
and for visualization purposes only; the raw ratio values should be used
for all analyses. 
</P>

<H2>Verification</H2>
<P>
Data were verified by repeating the preparation and measurement of
48 random fragments. No significant variation between the two
preparations was detected.
</P>
<P>
A spreadsheet containing the negative control data can be downloaded 
<A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hg16/encode/datafiles/stanfordPromoters/" TARGET=_BLANK>here</A>.
</P>

<H2>Credits</H2>
<P>This
work was done in collaboration at the 
Myers Lab at Stanford University (now at <a href="http://hudsonalpha.org/myers-lab" 
target="_blank">HudsonAlpha Institute for Biotechnology</a>). The following people contributed: Sara J. Cooper, Nathan D. Trinklein, Elizabeth D. Anton, Loan Nguyen, and Richard M. Myers. 
</P>

<H2>References</H2>
Cooper SJ, Trinklein ND, Anton ED, Nguyen L, Myers RM.
<A HREF = "http://www.genome.org/cgi/content/full/16/1/1" 
TARGET = _BLANK>Comprehensive analysis of transcriptional promoter structure 
and function in 1% of the human genome</A>.
<EM>Genome Res.</EM> 2006 Jan;16(1):1-10. Epub 2005 Dec 12.
</P>

<P>Trinklein ND, Aldred SJ, Saldanha AJ, Myers RM.
<A HREF="http://www.genome.org/cgi/content/short/13/2/308" 
TARGET=_BLANK>Identification and functional analysis of human transcriptional 
promoters</A>.
<EM>Genome Res.</EM> 2003 Feb;13(2):308-12.
</P>

<P>
Trinklein ND, Aldred SF, Hartman SJ, Schroeder DI, Otillar RP,
Myers RM.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;list_uids=14707170&amp;dopt=Abstract" 
TARGET=_BLANK>An abundance of bidirectional promoters in the human genome</A>.
<EM>Genome Res.</EM> 2004 Jan;14(1):62-6. 
</P>
