<H2>Overview</H2>
This super-track combines related tracks of the ENCODE Extension data generated 
by Affymetrix.  There are two member tracks:
<UL>
<LI><B>Affymetrix ENCODE Extension Transcription Sites</B> the transcribed 
fragments (transfrags) based on the signal.</LI>
<LI><B>Affymetrix ENCODE Extension Transcription Signal</B> - RNA abundance 
signal.</LI>
</UL>

<H2>Methods</H2>
The data from replicate arrays were quantile-normalized (Bolstad <EM>et al</EM>., 2003) 
and all arrays were scaled to a median array intensity of 330. Using two 
different approaches: i) no sliding window ii) sliding 51-bp window centered 
on each probe, an estimate of RNA abundance (signal) was computed by 
calculating the median of all pairwise average PM-MM values, where PM is a 
perfect match and MM is a mismatch. Both Kapranov <EM>et al</EM>. (2002) and Cawley 
<EM>et al</EM>. (2004) are good references for the experimental methods.  The latter
also describe the analytical methods.

<H3>Verification</H3>
Single biological replicates were generated and hybridized to duplicate arrays 
(two technical replicates). Transcribed regions were generated from the 
composite signal track by merging genomic positions to which probes are mapped.
This merging was based on a 5% false positive rate cutoff in negative bacterial
controls, a maximum gap (MaxGap) of 25 basepairs and minimum run (MinRun) of 
25 basepairs (see the Affy TransFrags track for the merged regions). 
 
<H2>Credits</H2>
These data were generated and analyzed by the collaboration of the following 
groups: the Tom Gingeras group at Affymetrix, Roderic Guigo group at Centre 
de Regulacio Genomica, Alexandre Reymond group at the University of Lausanne 
and Stylianos Antonarakis group at the University of Geneva. 
 
<H2>References</H2>
Please see the <a href="http://www.affymetrix.com/transcriptome" TARGET = _BLANK>
Affymetrix Transcriptome</a> site for a project overview and additional 
references to Affymetrix tiling array publications.
<P>
Bolstad BM, Irizarry RA, Astrand M, Speed TP. 
<A HREF="http://bioinformatics.oupjournals.org/cgi/content/abstract/19/2/185"
TARGET = _BLANK>A comparison of normalization methods for high density oligonucleotide
array data based on variance and bias</a>. 
<em>Bioinformatics.</em> 2003 Jan 22;19(2):185-93.
<P>
Cawley S, Bekiranov S, Ng HH, Kapranov P, Sekinger EA,
Kampa D, Piccolboni A, Sementchenko V, Cheng J, Williams  AJ <em>et al</em>.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=14980218"
TARGET = _BLANK>Unbiased mapping of transcription factor 
binding sites along human chromosomes 21 and 22 points to widespread 
regulation of noncoding RNAs</A>. <EM>Cell.</EM> 2004 Feb 20;116(4):499-509.
<P>
Kapranov P, Cawley SE, Drenkow J, Bekiranov S, Strausberg RL, Fodor SP, 
Gingeras TR. <A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=11988577"
TARGET = _BLANK>
Large-scale transcriptional activity in chromosomes 21 and 22</a>. 
<em>Science.</em> 2002 May 3;296(5569):916-9.
