<H2>Overview</H2>
<P>
This super-track combines related tracks of ChIP-chip data
generated by the Affymetrix/Harvard ENCODE collaboration.
ChIP-chip, also known as genome-wide location analysis, is a technique for
isolation and identification of DNA sequences bound by specific proteins in
cells.
</P>
<P>
These tracks contain ChIP-chip data of multiple transcription
factors, RNA polymerase II and histones, in multiple cell lines,
including HL-60 (leukemia) and ME-180 (cervical carcinoma),
and at different time points after drug cell treatment.
Binding was assayed on Affymetrix ENCODE tiling arrays.
Data are displayed as signals, 
median p-values, &quot;strict&quot; p-values and sites.
</P>

<H2>Credits</H2>
<P>
These data were generated and analyzed by collaboration of
the Tom Gingeras group at
<A HREF="http://www.affymetrix.com/" TARGET=_blank>Affymetrix</A> and the
<A HREF="http://struhl.med.harvard.edu/" TARGET=_BLANK>Kevin Struhl lab</A>
at Harvard Medical School.</P>

<H2>References</H2>
<P>
Please see the <A HREF="http://transcriptome.affymetrix.com/local_index.html"
TARGET=_blank>Affymetrix Transcriptome</A> site for a project overview and
additional references to Affymetrix tiling array publications.</P>
<P>
Bolstad BM, Irizarry RA, Astrand M, and Speed TP.
<A HREF="http://bioinformatics.oupjournals.org/cgi/content/abstract/19/2/185"
TARGET=_blank>A comparison
of normalization methods for high density oligonucleotide array data based
on variance and bias</A>.
<em>Bioinformatics</em>. 2003 Jan 22;19(2):185-93.</P>
<P>
Cawley S, Bekiranov S, Ng HH, Kapranov P, Sekinger EA,
Kampa D, Piccolboni A, Sementchenko V, Cheng J,
Williams AJ <em>et al</em>.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=pubmed&dopt=Abstract&list_uids=14980218" TARGET=_blank>Unbiased mapping of
transcription factor binding sites along human chromosomes 21 and 22 points
to widespread regulation of noncoding RNAs</A>.
<em>Cell</em>. 2004 Feb 20;116(4):499-509.

