<H2>Overview</H2>
<P>
This super-track combines related tracks of ChIP data generated by
the <A HREF="https://molecularbiosci.utexas.edu/component/cobalt/item/16-molecular-biosciences/164-iyer-vishwanath-r?Itemid=1224" TARGET=_blank>Iyer laboratory</A> at 
The University of Texas at Austin.
Two technologies are presented in this super-track: ChIP-chip and ChIP-STAGE.
ChIP-chip, also known as genome-wide location analysis, is a technique for
isolation and identification of DNA sequences bound by specific proteins in
cells. Instead of detecting bound fragments by microarray, ChIP-STAGE uses
Sequence Tag Analysis of Genomic Enrichment, or STAGE, technology by cloning 
STAGE tags, sequencing and mapping to the human genome.
</P>
<P>
These tracks contain ChIP data for several transcription
factors, including c-Myc, E2F4 and STAT1, in cell lines
including 2091 (foreskin fibroblast) and HeLa (cervical carcinoma).
</P>

<H2>Credits</H2>
<P>
ChIP-chip data were contributed by Jonghwan Kim, Akshay Bhinge, and Vishy Iyer 
from the Iyer lab
at The University of Texas at Austin, in collaboration with Mike Singer, 
Nan Jiang, and Roland Green of <A HREF="http://sequencing.roche.com/en/products-solutions/by-category/target-enrichment/hybridization.html"
TARGET=_blank>NimbleGen Systems, Inc</A>.</P>
<P>
ChIP-STAGE data were contributed by Jonghwan Kim, Akshay Bhinge, and Vishy Iyer 
from the Iyer lab, and by Ghia Euskirchen and Michael Snyder of the 
<A HREF="https://www.yale.edu/snyder" TARGET=_BLANK>Snyder lab</A> at
Yale University. </P>

<H2>References</H2>
<P>
Bhinge AA, Kim J, Euskirchen G, Snyder M, Iyer VR.
<A HREF="https://genome.cshlp.org/content/17/6/910.abstract" TARGET=_blank>
Mapping the chromosomal targets of STAT1 by Sequence Tag Analysis of Genomic 
Enrichment (STAGE)</A>. 
<em>Genome Res</em>. 2007 Jun;17(6):910-6.</P>
<P>
Kim J, Bhinge A, Morgan XC, Iyer VR. 
<A HREF="https://www.nature.com/articles/nmeth726"
TARGET=_blank>Mapping DNA-protein interactions in large genomes by sequence tag 
analysis of genomic enrichment</A>. <em>Nat Methods</em>. 2005 Jan;2(1):47-53.
</P>

