<h2>Description</h2>

<p>
These tracks display genes associated with severe developmental disorders. The track can be used to
filter genomic sequencing data from individuals with genetic disorders to identify likely causative
variants and accelerate diagnosis.
</p>

<h2>Display Conventions and Configuration</h2>

<p>
For each track, items are colored according to the likelihood that the gene-disease
association is true:</p>
<ul>
    <li> <font style="color: rgb(39,103,73);"><b> Dark-green</b></font> - Definitive</li>
    <li> <font style="color: rgb(56,161,105);"><b>Green</b></font> - Strong</li>
    <li> <font style="color: rgb(104,211,145);"><b>Light-green</b></font> - Moderate</li>
    <li> <font style="color: rgb(252,129,129);"><b>Pink</b></font> - Limited</li>
    <li> <font style="color: rgb(229,62,62);"><b>Red</b></font> - Disputed</li>
    <li> <font style="color: rgb(155,44,44);"><b>Dark-red</b></font> - Refuted</li>
</ul>

<p>Each <b>mouseover</b> tooltip provides the following information:</p>
<ul>
  <li><strong>G2P ID</strong>: Unique identifier assigned by the Gene2Phenotype (G2P) database.</li>
  <li><strong>Variant Consequence</strong>: Predicted effect each allele of a variant has on a
      transcript.</li>
  <li><strong>Disease Name</strong>: Name of the disease associated with the variant.</li>
  <li><strong>PubMed IDs</strong>: Publications associated with the variant.</li>
  <li><strong>Molecular Mechanism</strong>: Description of the molecular processes and interactions
      causing pathogenic effects.</li>
  <li><strong>Allelic Requirements</strong>: Number of alleles required at a locus to produce a
      pathogenic phenotype (e.g., monoallelic, biallelic).</li>
  <li><strong>Date of Last Review</strong>: Most recent date the entry was manually reviewed.</li>
</ul>

<h2>Method</h2>
<p>
Expert-curated gene disease models released by the Gene2Phenotype project were imported and
processed to create a BED-based track annotating genomic regions reported to be associated with
disease in the literature. Standard genome assembly coordinates and gene annotations were used to
map entries to the browser.
</p>

<h2>Contact</h2>
<p>
For more information on the Gene2Phenotype project, please contact 
<A HREF="mailto:&#103;&#50;p&#45;&#104;&#101;&#108;p&#64;&#101;&#98;&#105;.
a&#99;.
&#117;&#107;">
&#103;&#50;p&#45;&#104;&#101;&#108;p&#64;&#101;&#98;&#105;.
a&#99;.
&#117;&#107;</A>
<!-- above address is g2p-help at ebi.ac.uk -->
</p>

<h2>Data Access</h2>
<p>
Source data for these tracks are available directly from
<a href="https://www.ebi.ac.uk/gene2phenotype/" target="_blank">Gene2Phenotype</a>. 
</p>

<h2>References</h2>
<p>
Thormann A, Halachev M, McLaren W, Moore DJ, Svinti V, Campbell A, Kerr SM, Tischkowitz M, Hunt SE,
Dunlop MG <em>et al</em>.
<a href="https://doi.org/10.1038/s41467-019-10016-3" target="_blank">
Flexible and scalable diagnostic filtering of genomic variants using G2P with Ensembl VEP</a>.
<em>Nat Commun</em>. 2019 May 30;10(1):2373.
DOI: <a href="https://doi.org/10.1038/s41467-019-10016-3"
target="_blank">10.1038/s41467-019-10016-3</a>; PMID: <a
href="https://www.ncbi.nlm.nih.gov/pubmed/31147538" target="_blank">31147538</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6542828/" target="_blank">PMC6542828</a>
</p>
<p>
Yates TM, Ansari M, Thompson L, Hunt SE, Uhalte EC, Hobson RJ, Marsh JA, Wright CF, Firth HV.
<a href="https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-024-01398-1"
target="_blank">
Curating genomic disease-gene relationships with Gene2PhenotypeÂ (G2P)</a>.
<em>Genome Med</em>. 2024 Nov 6;16(1):127.
DOI: <a href="https://doi.org/10.1186/s13073-024-01398-1"
target="_blank">10.1186/s13073-024-01398-1</a>; PMID: <a
href="https://www.ncbi.nlm.nih.gov/pubmed/39506859" target="_blank">39506859</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11539801/" target="_blank">PMC11539801</a>
</p>
