<h2>Description</h2>
<p>
The
<a target="_blank" href="https://commonfund.nih.gov/GTEx">NIH Genotype-Tissue Expression (GTEx)
project</a> was created to establish a sample and data resource for studies on the relationship
between genetic variation and gene expression in multiple human tissues. 
This track displays median transcript expression levels in 53 tissues, based on
RNA-seq data from the GTEx midpoint milestone data release (V6, October 2015). 
Data for this track were computed at UCSC from GTEx RNA-seq sequence data using the
<a href="http://toil.ucsc-cgl.org/" target="_blank">Toil</a>
pipeline running the kallisto transcript-level quantification tool.</p>

<h2>Display Conventions</h2>
<p>
In Full and Pack display modes, expression for each transcript is represented by a colored 
bar chart, where the height of each bar represents the median expression level across all 
samples for a tissue, and the bar color indicates the tissue.
<div><img border="1" src="../images/gtex/gtexGeneTcap.png"></div></p>
<p>
The bar chart display has the same width and tissue order for all transcripts.
Mouse hover over a bar will show the tissue and median expression level.
The Squish display mode draws a rectangle for each gene, colored to indicate the tissue
with highest expression level if it contributes more than 10% to the overall expression
(and colored black if no tissue predominates).
In Dense mode, the darkness of the grayscale rectangle displayed for the transcript reflects 
the total median expression level across all tissues.</p>
<p>
Click-through on a graph displays a boxplot of expression level quartiles with outliers, 
per tissue.</p>

<h2>Methods</h2>
<p>
Tissue samples were obtained using the GTEx standard operating procedures for informed consent
and tissue collection, in conjunction with the 
<a target="_blank" href="http://biospecimens.cancer.gov/resources/sops/gtex.asp">
National Cancer Institute Biorepositories and Biospecimen</a>.
All tissue specimens were reviewed by pathologists to characterize and
verify organ source.
Images from stained tissue samples can be viewed via the 
<a target="_blank" href="https://brd.nci.nih.gov/brd/image-search/searchhome">
NCI histopathology viewer</a>.
The Qiagen PAXgene non-formalin tissue preservation product was used to stabilize 
tissue specimens without cross-linking biomolecules.</p>
<p>
RNA-seq was performed by the GTEx Laboratory, Data Analysis and Coordinating Center 
(LDACC) at the Broad Institute.
The Illumina TruSeq protocol was used to create an unstranded polyA+ library sequenced
on the Illumina HiSeq 2000 platform to produce 76-bp paired end reads at a depth 
averaging 50M aligned reads per sample.</p>
<p>
Sequence reads for this track were quantified to the hg38/GRCh38 human genome using kallisto
assisted by the GENCODE v23 transcriptome definition. Read quantification was performed at UCSC
by the Computational Genomics lab, using the <a href="http://toil.ucsc-cgl.org/"
target="_blank">Toil</a> pipeline. The resulting kallisto files were
combined to generate a transcript per million (TPM) expression matrix using the UCSC tool,
<i>kallistoToMatrix</i>. Average TPM expression values for each tissue were calculated and 
used to generate a bed6+5 file that is the base of the track. This was done using the UCSC
tool, <i>expMatrixToBarchartBed</i>. The bed track was then converted to a bigBed file using the 
UCSC tool, <i>bedToBigBed</i>.</p>
<p>
The data in the hg19/GRCh37 version of this track was generated by coordinate converting 
the hg38/GRCh38 track data. 
Of the 189,615 BED entries from the original hg38 track, 176,220 were mapped over by transcript
name to hg19 using wgEncodeGencodeCompV24lift37 (~93% coverage).</p>

<h2>Subject and Sample Characteristics</h2>
<p>
The scientific goal of the GTEx project required that the donors and their biospecimen 
present with no evidence of disease. The tissue types collected were chosen based on their 
clinical significance, logistical feasibility and their relevance to the scientific goal 
of the project and the research community. Postmortem samples were collected from 
non-diseased donors with ages ranging from 20 to 79. 34.4% of donors were female and
65.6% male. 
<div> <img border="1" src="/images/gtex/gtexSampleRin.V6.png"></div></p>
<p>
<div><img border="1" src="/images/gtex/gtexSampleAge.V6.png"></div></p>
<p>
Additional summary plots of GTEx sample characteristics are available at the 
<a target="_blank" href="https://gtexportal.org/home/tissueSummaryPage">
GTEx Portal Tissue Summary</a> page.</p>

<h2>Credits</h2>
<p>
Samples were collected by the GTEx Consortium.
RNA-seq was performed by the GTEx Laboratory, Data Analysis and Coordinating Center 
(LDACC) at the Broad Institute.
John Vivian, Melissa Cline, and Benedict Paten of the UCSC Computational Genomics lab were
responsible for the sequence read quantification used to produce this track. Kate Rosenbloom 
and Chris Eisenhart of the UCSC Genome Browser group were responsible for data file
post-processing and track configuration.</p>

<h2>References</h2>
<p>
J. Vivian et al., 
<a href="http://www.biorxiv.org/content/early/2016/07/07/062497" target="_blank">
Rapid and efficient analysis of 20,000 RNA-seq samples with Toil</a>
<em> bioRxiv</em>  bioRxiv, vol. 2, p. 62497, 2016. </p>
<p>
GTEx Consortium.
<a href="https://www.nature.com/ng/journal/v45/n6/full/ng.2653.html" target="_blank">
The Genotype-Tissue Expression (GTEx) project</a>.
<em>Nat Genet</em>. 2013 Jun;45(6):580-5.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23715323" target="_blank">23715323</a>; 
PMC: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4010069/" target="_blank">PMC4010069</a> </p>

<p>
Carithers LJ, Ardlie K, Barcus M, Branton PA, Britton A, Buia SA, Compton CC, DeLuca DS, Peter-Demchok J, Gelfand ET <em>et al</em>.
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/26484571/" target="_blank">
A Novel Approach to High-Quality Postmortem Tissue Procurement: The GTEx Project</a>.
<em>Biopreserv Biobank</em>. 2015 Oct;13(5):311-9.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26484571" target="_blank">26484571</a>; 
PMC: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4675181/" target="_blank">PMC4675181</a></p>

<p>
Mel&#233; M, Ferreira PG, Reverter F, DeLuca DS, Monlong J, Sammeth M, Young TR, Goldmann JM,
Pervouchine DD, Sullivan TJ <em>et al</em>.
<a href="http://science.sciencemag.org/content/348/6235/660.long" target="_blank">
Human genomics. The human transcriptome across tissues and individuals</a>.
<em>Science</em>. 2015 May 8;348(6235):660-5.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25954002" target="_blank">25954002</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4547472/" target="_blank">PMC4547472</a></p>

<p>
DeLuca DS, Levin JZ, Sivachenko A, Fennell T, Nazaire MD, Williams C, Reich M, Winckler W, Getz G.
<a href="https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/bts196"
target="_blank">
RNA-SeQC: RNA-seq metrics for quality control and process optimization</a>.
<em>Bioinformatics</em>. 2012 Jun 1;28(11):1530-2.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22539670" target="_blank">22539670</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3356847/" target="_blank">PMC3356847</a></p>

