<h2>Description</h2>

<p>
This track displays single nucleotide polymorphisms (SNPs) identified by published 
Genome-Wide Association Studies (GWAS), collected in the 
<a href="http://www.ebi.ac.uk/gwas/" target="_blank">NHGRI-EBI GWAS Catalog</a>
published jointly by the <a href="https://www.genome.gov/" target="_blank">National
Human Genome Research Institute</A> (NHGRI) and the <a href="https://www.ebi.ac.uk/"
target="_blank">European Bioinformatics Institute (EMBL-EBI)</a>.
Some <a href="http://www.ebi.ac.uk/gwas/docs/abbreviations" target="_blank">abbreviations</a>
are used above.
</p>
<p>
From <a href="http://www.ebi.ac.uk/gwas/docs/about" target="_blank">http://www.ebi.ac.uk/gwas/docs/about</a>:
<blockquote>
The Catalog is a quality controlled, manually curated, literature-derived
collection of all published <a href="https://en.wikipedia.org/wiki/Genome-wide_association_study"
target="_blank">genome-wide association studies</a> assaying at least
100,000 SNPs and all SNP-trait associations with p-values &lt; 1.0 x
10<sup>-5</sup> (Hindorff et al., 2009). For more details about the Catalog
curation process and data extraction procedures, please refer to the
<a href="http://www.ebi.ac.uk/gwas/docs/methods" target="_blank">Methods page</a>.
</blockquote>
</p>

<h2>Methods</h2>

<p>
From <a href="http://www.ebi.ac.uk/gwas/docs/methods" target="_blank">http://www.ebi.ac.uk/gwas/docs/methods</a>:
<blockquote>
The GWAS Catalog data is extracted from the literature. Extracted information
includes publication information, study cohort information such as cohort size,
country of recruitment and subject ethnicity, and SNP-disease association
information including SNP identifier (i.e. RSID), p-value, gene and risk
allele. Each study is also assigned a trait that best represents the phenotype
under investigation. When multiple traits are analysed in the same study either
multiple entries are created, or individual SNPs are annotated with their
specific traits. Traits are used both to query and visualise the data in the
Catalog's web form and diagram-based query interfaces.
<br><br>
Data extraction and curation for the GWAS Catalog is an expert activity; each
step is performed by scientists supported by a web-based tracking and data
entry system which allows multiple curators to search, annotate, verify and
publish the Catalog data. Papers that qualify for inclusion in the Catalog are
identified through weekly PubMed searches. They then undergo two levels of
curation. First all data, including association information for SNPs, traits
and general information about the study, are extracted by one curator. A second
curator then performs an additional round of curation to double-check the
accuracy and consistency of all the information. Finally, an automated pipeline
performs validation of the extracted data, see the
<a href="https://www.ebi.ac.uk/gwas/docs/methods#mapping"
target="_blank">Quality control and SNP mapping section</a> below for more
details. This information is then used for queries and in the production of the
diagram.
</blockquote>
</p>

<h2>References</h2>
<p>
Hindorff LA, Sethupathy P, Junkins HA, Ramos EM, Mehta JP, Collins FS, Manolio TA.
<a href="http://www.pnas.org/content/106/23/9362.long" target="_blank">
Potential etiologic and functional implications of genome-wide association loci for human diseases
and traits</a>.
<em>Proc Natl Acad Sci U S A</em>. 2009 Jun 9;106(23):9362-7.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19474294" target="_blank">19474294</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2687147/" target="_blank">PMC2687147</a>
</p>
