<H2>Description</H2>
<P>
Consolidates mutations from locus-specific databases and a genome wide
mutation database into a composite track,
with an emphasis on those associated with phenotypic effects.
This prototype version includes entries mostly from the globin loci, plus
a few sample entries from the CFTR region for the locus-specific subtrack.
</P>

<a name="source">
<H2>Variant Sources</H2>
<UL>
<LI>
The HbVar database describes variants and thalassemia mutations in the
globin loci.
<A HREF="http://www.bx.psu.edu/" TARGET="_blank">http://www.bx.psu.edu/</A>
<LI>
Coming soon: CFTR data.
(A few sample variants from this region have already been added.)
<LI>
Swiss-Prot data on variants within coding regions is available genome wide.
<LI>
A possible connection with WayStation, the HGVS community-wide database.
<A HREF="http://www.genomic.unimelb.edu.au/mdi/centraldb.html" TARGET="_blank">http://www.genomic.unimelb.edu.au/mdi/centraldb.html</A>
</UL>

<H2>Filtering</H2>
<UL>
<LI>
Filtering can be performed on the location of the mutations with respect to
genes, i.e. to exclude those in exons, introns, UTR regions, or those
not within any known transcription unit.
<LI>
One can also exclude mutations by type: substitution, insertion, deletion,
duplication, or complex.
</UL>

<H2>Color key</H2>
The track is color coded by the mutation type.  Also the mutations with
darker colors indicate that there is a link to clinical data on the
details page.
<UL>
<LI>
substitution = purple
<LI>
insertion = green
<LI>
deletion = blue
<LI>
duplication = orange
<LI>
complex = red
<LI>
unknown = black
</UL>

<H2>Supporting Details</H2>
<P> 
Positional information can be found in the annotations section
of the Genome Browser 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">downloads</A> page, 
which is organized by species and assembly. Non-positional information
can be found in the 
<A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hgFixed/database/">shared
data</A> section of the same page.

<H2>Credits</H2>
<DL>
<DT>Center for Comparative Genomics and Bioinformatics, Penn State University, University Park, PA
<DD>Belinda Giardine, Ross Hardison, Webb Miller, Cathy Riemer
<DT>Erasmus University, Rotterdam, Netherlands
<DD>George Patrinos
<DT>Hospital Henri Mondor, Creteil, France
<DD>Henri Wajcman
<DT>Boston University
<DD>David H.K. Chui
<DT>U. Athens and IIBEAA, Athens, Greece
<DD>Nicholas Anagnou
<DT>Macedonian Academy of Sciences and Arts, RCGEB, Skopje, Macedonia
<DD>Georgi D. Efremov
<DT>Weatherall Institute of Molecular Medicine, Oxford UK
<DD>Richard Gibbons, Doug Higgs, Jim Hughes
<DT>Johns Hopkins University School of Medicine, Baltimore, MD
<DD>Garry Cutting, Andrew P. Feinberg
<DT>Center for Biomolecular Science and Engineering, University of California, Santa Cruz, CA
<DD>Fan Hsu, Jim Kent, Andrew Kern, Robert Kuhn, Heather Trumbower
</DL> 

