<H2>Description</H2>
<P>
This track shows validated regions of structural variation in nine 
individuals from <em>Kidd et al.</em>.  
<B><FONT color="C00000">Deletions</FONT></B>, 
<B><FONT color="0000C0">insertions</FONT></B> and
<B><FONT color="00C000">inversions</FONT></B> are included.  
For inversions, sites corresponding to both breakpoints may be
depicted.  Clones corresponding to only a single breakpoint were
selected to validate the site.  Coordinates correspond to the variant
region predicted by end-sequence pairs (ESPs), not to sequence-derived
breakpoints.
</P>
<P>
Each site was validated by at least one of these methods:
<UL>
<LI><B>Agi:</B> Agilent CGH</LI>
<LI><B>FISH:</B> Inversion FISH assay</LI>
<LI><B>MCD:</B> Clone fingerprint</LI>
<LI><B>NIL:</B> Overlap with &quot;novel&quot; insertion locus</LI>
<LI><B>Nim:</B> NimbleGen CGH</LI>
<LI><B>Seq:</B> Clone sequencing</LI>
</UL
</P>
<P>
Each individual's validated sites are in a different 
subtrack.  The nine individuals' labels used in <em>Kidd et al.</em>, 
populations of origin, and 
<A HREF="http://ccr.coriell.org/" 
TARGET=_BLANK>Coriell Cell Repository</A> catalog IDs are shown here:
</P>
<TABLE>
<TR><TD><B>Individual&nbsp;&nbsp;</B></TD><TD><B>Population&nbsp;&nbsp;</B></TD><TD><B>Coriell ID</B></TD></TR>
<TR><TD>ABC14</TD><TD>CEPH</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM12156" TARGET=_BLANK>NA12156</TD></TR>
<TR><TD>ABC13</TD><TD>Yoruba</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM19129" TARGET=_BLANK>NA19129</TD></TR>
<TR><TD>ABC12</TD><TD>CEPH</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM12878" TARGET=_BLANK>NA12878</TD></TR>
<TR><TD>ABC11</TD><TD>China</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM18555" TARGET=_BLANK>NA18555</TD></TR>
<TR><TD>ABC10</TD><TD>Yoruba</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM19240" TARGET=_BLANK>NA19240</TD></TR>
<TR><TD>ABC9</TD><TD>Japan</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM18956" TARGET=_BLANK>NA18956</TD></TR>
<TR><TD>ABC8</TD><TD>Yoruba</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM18507" TARGET=_BLANK>NA18507</TD></TR>
<TR><TD>ABC7</TD><TD>Yoruba</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM18517" TARGET=_BLANK>NA18517</TD></TR>
<TR><TD>G248</TD><TD>Unknown</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM15510" TARGET=_BLANK>NA15510</A></TD></TR>
</TABLE>

<H2>Methods</H2>
<P>
Excerpted from <em>Kidd et al.</em>:</P>
<TABLE>
<TR><TD>&nbsp;&nbsp;&nbsp;</TD><TD>
We selected eight individuals as part of the first phase of the Human
Genome Structural Variation Project. This included four individuals of
Yoruba Nigerian ethnicity and four individuals of non-African
ethnicity. For each individual we constructed a whole genomic library
of about 1 million clones by using a fosmid subcloning strategy
For each individual, a fosmid subcloning strategy was used to construct 
a whole genomic library of about 1 million clones.  
Each library was arrayed and both ends of each clone insert were
sequenced to generate a pair of high-quality end sequences (termed an
end-sequence pair (ESP)).  
The overall approach generated a physical clone map for each
individual human genome, flagging regions discrepant by size or
orientation on the basis of the placement of end sequences against the
reference assembly.
Across all eight libraries, we mapped 6.1 million clones to distinct
locations against the reference sequence
(<A HREF="http://hgsv.washington.edu" 
TARGET=_BLANK>http://hgsv.washington.edu</A>).  
Of these, 76,767 were discordant by length and/or orientation,
indicating potential sites of structural variation. About 0.4%
(23,742) of the ESPs mapped with only one end to the reference
assembly despite the presence of high-quality sequence at the other
end (termed one-end anchored (OEA) clones).
<BR><BR>

Fosmid clones discordant by size (n = 3,371 fosmid clones) were
subjected to fingerprint analysis using four multiple complete
restriction enzyme digests (MCD analysis) to confirm insert size and
eliminate rearranged clones. Two high-density customized
oligonucleotide microarrays (Agilent and NimbleGen) were designed to
confirm sites of deletion and insertion (GEO accessions GSE10008 and
GSE10037). We developed a new, expectation maximization-based
clustering approach to genotype deletions with the use of data from
the Illumina Human1M BeadChip collected for 125 HapMap DNA samples.
We found that more than 98% of the children's genotypes were
consistent with mendelian transmission on the basis of an analysis of
28 parent-child trios.
</TD></TR>
</TABLE>

<H2>References</H2>
<P>
Kidd JM, Cooper GM, Donahue WF, Hayden HS, Sampas N, Graves T, Hansen
N, Teague B, Alkan C, Antonacci F, Haugen E, Zerr T, Yamada NA, Tsang
P, Newman TL, Tüzün E, Cheng Z, Ebling HM, Tusneem N, David R, Gillett
W, Phelps KA, Weaver M, Saranga D, Brand A, Tao W, Gustafson E,
McKernan K, Chen L, Malig M, Smith JD, Korn JM, McCarroll SA,
Altshuler DA, Peiffer DA, Dorschner M, Stamatoyannopoulos J, Schwartz
D, Nickerson DA, Mullikin JC, Wilson RK, Bruhn L, Olson MV, Kaul R,
Smith DR, Eichler EE.
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/18451855?ordinalpos=1&itool=EntrezSystem2.PEntrez.Pubmed.Pubmed_ResultsPanel.Pubmed_RVDocSum" 
TARGET=_BLANK>Mapping and sequencing of structural variation from eight 
human genomes.</A>
Nature. 2008 May 1;453(7191):56-64.</P>

