<H2>Description</H2>
<P>
This track shows the mapping of clone ends from nine individuals 
to the reference $Organism genome, specifically clones from Kidd,
<EM>et al.</EM> whose end mappings were discordant with the reference
genome in one of the following ways:
<UL>
<LI><B><FONT color="C00000">deletion:</FONT></B> Clone mapping too large relative to reference</LI>
<LI><B><FONT color="0000C0">insertion:</FONT></B> Clone mapping too small relative to reference</LI>
<LI><B><FONT color="00C000">inversion:</FONT></B> In appropriate orientation, clone mapping spans 
                      potential inversion breakpoint</LI>
<LI><B><FONT color="F0A040">OEA:</FONT></B> One End Anchored clones (only one end could be mapped
                to reference)</LI>
<LI><B><FONT color="000000">transchrm</FONT>:</B> Clone ends map to different chromosomes
(name indicates identity of other chromosome after the underscore).
</LI>

</UL>
</P>
<P>
Each individual's discordant clone end mappings are in a different 
subtrack.  The nine individuals' labels used in Kidd, <EM>et al.</EM>, 
populations of origin, and 
<A HREF="http://ccr.coriell.org/" 
TARGET=_BLANK>Coriell Cell Repository</A> catalog IDs are shown here:
</P>
<TABLE>
<TR><TD><B>Individual&nbsp;&nbsp;</B></TD><TD><B>Population&nbsp;&nbsp;</B></TD><TD><B>Coriell ID</B></TD></TR>
<TR><TD>ABC14</TD><TD>CEPH</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM12156" TARGET=_BLANK>NA12156</TD></TR>
<TR><TD>ABC13</TD><TD>Yoruba</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM19129" TARGET=_BLANK>NA19129</TD></TR>
<TR><TD>ABC12</TD><TD>CEPH</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM12878" TARGET=_BLANK>NA12878</TD></TR>
<TR><TD>ABC11</TD><TD>China</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM18555" TARGET=_BLANK>NA18555</TD></TR>
<TR><TD>ABC10</TD><TD>Yoruba</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM19240" TARGET=_BLANK>NA19240</TD></TR>
<TR><TD>ABC9</TD><TD>Japan</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM18956" TARGET=_BLANK>NA18956</TD></TR>
<TR><TD>ABC8</TD><TD>Yoruba</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM18507" TARGET=_BLANK>NA18507</TD></TR>
<TR><TD>ABC7</TD><TD>Yoruba</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM18517" TARGET=_BLANK>NA18517</TD></TR>
<TR><TD>G248</TD><TD>Unknown</TD><TD><A HREF="http://ccr.coriell.org/Sections/Search/Sample_Detail.aspx?Ref=GM15510" TARGET=_BLANK>NA15510</A></TD></TR>
</TABLE>

<H2>Methods</H2>
<P>
Excerpted from Kidd, <EM>et al.</EM>:</P>
<BLOCKQUOTE>
We selected eight individuals as part of the first phase of the Human
Genome Structural Variation Project. This included four individuals of
Yoruba Nigerian ethnicity and four individuals of non-African
ethnicity. For each individual we constructed a whole genomic library
of about 1 million clones, using a fosmid subcloning strategy.
Each library was arrayed and both ends of each clone insert were
sequenced to generate a pair of high-quality end sequences (termed an
end-sequence pair (ESP)).  
The overall approach generated a physical clone map for each
individual human genome, flagging regions discrepant by size or
orientation on the basis of the placement of end sequences against the
reference assembly.

Across all eight libraries, we mapped 6.1 million clones to distinct
locations against the reference sequence
(<A HREF="http://hgsv.washington.edu" 
TARGET=_BLANK>http://hgsv.washington.edu</A>).  
Of these, 76,767 were discordant by length and/or orientation,
indicating potential sites of structural variation. About 0.4%
(23,742) of the ESPs mapped with only one end to the reference
assembly despite the presence of high-quality sequence at the other
end (termed one-end anchored (OEA) clones).
</BLOCKQUOTE>
<P>
Note: This track contains many more than the 76,767 + 23,742 items
mentioned above because it also includes clones whose ends map to
different chromosomes (transchrm).
</P>

<H2>References</H2>
<P>
Kidd JM, Cooper GM, Donahue WF, Hayden HS, Sampas N, Graves T, Hansen N, 
Teague B, Alkan C, Antonacci F, <EM>et al.</EM>
<!--
Haugen E, Zerr T, Yamada NA, Tsang
P, Newman TL, Tüzün E, Cheng Z, Ebling HM, Tusneem N, David R, Gillett
W, Phelps KA, Weaver M, Saranga D, Brand A, Tao W, Gustafson E,
McKernan K, Chen L, Malig M, Smith JD, Korn JM, McCarroll SA,
Altshuler DA, Peiffer DA, Dorschner M, Stamatoyannopoulos J, Schwartz
D, Nickerson DA, Mullikin JC, Wilson RK, Bruhn L, Olson MV, Kaul R,
Smith DR, Eichler EE.
-->
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/18451855?ordinalpos=1&itool=EntrezSystem2.PEntrez.Pubmed.Pubmed_ResultsPanel.Pubmed_RVDocSum" 
TARGET=_BLANK>Mapping and sequencing of structural variation from eight 
human genomes.</A>
<EM>Nature</EM>. 2008 May 1;453(7191):56-64.</P>

