<H2>Description</H2>
<P>
This track shows the S score (Z-score +- variance) for positive selection in humans within a 100 kb
window surrounding each polymorphic position in the five modern human
sequences and the human reference genome as described in Green <em>et al.</em>,
Supplemental Online Material Text 13, Burbano <em>et al.</em>. 
A positive score indicates more derived alleles in Neandertal than
expected, given the frequency of derived alleles in human. A negative
score indicates fewer derived alleles in Neandertal, and may indicate an 
episode of positive selection in early humans.
</P>
<P>
To view the polymorphic sites on which the S score was computed, open
the <A HREF="/cgi-bin/hgTracks?db=$db&ntSssSnps=pack">S
SNPs</A> track.</P>

<H2>Methods</H2>
<P>
Green <em>et al.</em> identified single-base sites that are
polymorphic among five modern human genomes of diverse ancestry
(in the 
<A HREF="/cgi-bin/hgTrackUi?&c=chr1&g=ntModernHumanSeq">Modern Human Seq</A> 
track) plus the human reference
genome.  CpG sites were excluded because of the higher mutation rate
at CpG sites.
The ancestral or derived state of each single nucleotide polymorphism
(SNP) was determined by comparison with the chimpanzee genome.
The SNPs are displayed in the 
<A HREF="/cgi-bin/hgTrackUi?&c=chr1&g=ntSssSnps">S SNPs</A> track.
The fact that SNPs with higher frequencies of the derived
allele in modern humans were more likely to show the derived allele in
Neandertals was used to calculate the expected number of derived alleles in
Neandertal within a given region of the human genome.
The observed numbers of derived alleles were compared to the expected
numbers to identify regions where the Neandertals carry fewer
derived alleles than expected given the human allelic states.  The
score assigned to each SNP is
the <A HREF="http://en.wikipedia.org/wiki/Z-score"
TARGET=_BLANK>z-score</A> of the observed and expected counts relative
to the variance in the number of the expected counts of derived
alleles within the 100,000-base window around the SNP.
</P>
<P>
<B>Note:</B> In order to display both the score and the variance within
the same track in the UCSC Genome Browser, the scores were modified as
follows: at the SNP position, the value displayed is the score plus
the variance.  At the position following the SNP position, the score
minus the variance is displayed.  When viewing large regions (at least
100,000 bases), the default mean+whiskers condensation of the scores
provides an indication of the range covered by the variance.
</P>

<H2>Reference</H2>
<p>
Green RE, Krause J, Briggs AW, Maricic T, Stenzel U, Kircher M, Patterson N, Li H, Zhai W, Fritz MH
<em>et al</em>.
<a href="http://www.sciencemag.org/content/328/5979/710.abs" target="_blank">
A draft sequence of the Neandertal genome</a>.
<em>Science</em>. 2010 May 7;328(5979):710-22.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/20448178" target="_blank">20448178</a>
</p>
