<H2>Description</H2>
<P>
This track shows single nucleotide polymorphisms (SNPs) used in a
genome-wide scan for signals of positive selection in the human
lineage since divergence from the Neandertal lineage.</P>
<P>
SNP labels represent the ancestral (A) or derived (D) status,
determined by comparison with the chimpanzee reference genome,
of alleles in the human reference assembly, five modern human
genomes of diverse ancestry (see the 
<A HREF="/cgi-bin/hgTrackUi?&c=chr1&g=ntModernHumans">Modern Human Seq</A> 
track), and Neandertals.  
The first six characters of an item name show the status of the allele
(A, D or _ if not known)
in six genomes: human reference, San, Yoruba, Han, Papuan, and French, in that
order. These characters are followed by a colon, the number of derived alleles
found in Neandertals, a comma and the number of ancestral alleles found in
Neandertals.  
For example, a SNP labeled AAADAA:0D,2A has the ancestral allele in
the reference human genome and in all of the modern human genomes
except Han.  Among Neandertals, two instances of the ancestral allele 
were found, but no instances of the derived allele.  
</P>
<P>
SNPs are colored <B><FONT COLOR = "red">red</FONT></B> when at least four of the six modern human
genomes are derived while all observed Neandertal alleles are
ancestral.  An overrepresentation of such SNPs in a region would 
imply that the region had undergone positive selection in the 
modern human lineage since divergence from Neandertals; the 
<A HREF="/cgi-bin/hgTrackUi?&c=chr1&g=ntSssZScorePMVar">Sel Swp Scan 
(S)</A> track displays a signal calculated from these SNPs, and the 
<A HREF="/cgi-bin/hgTrackUi?&c=chr1&g=ntSssTop5p">5% Lowest S</A>
track contains the regions in which the signal most strongly indicates selective
pressure on the modern human lineage.
</P>

<H2>Display Conventions and Configuration</H2>
<P>
<B><FONT COLOR = "red">Red</FONT></B>
SNPs are those where at least four of the six modern human
genomes are derived while all observed Neandertal alleles are
ancestral. All other SNPs are <B>black</B>.
</P>

<H2>Methods</H2>
<P>
For the purposes of this analysis, SNPs were defined as single-base
sites that are polymorphic among 5 modern human genomes of diverse
ancestry (see the 
<A HREF="/cgi-bin/hgTrackUi?&c=chr1&g=ntModernHumans">Modern Human Seq</A> 
track) plus the human reference
genome.  SNPs at CpG sites were excluded because of the higher
mutation rate at CpG sites.  
Ancestral or derived state was determined by comparison with the 
chimpanzee genome.  
</P>

<H2>Credits</H2>
<P>
This track was produced at UCSC using data generated by
<A HREF="mailto:&#101;&#100;&#64;&#115;&#111;e.&#117;&#99;s&#99;.&#101;&#100;&#117;"
title="&#101;&#100;&#64;&#115;&#111;e.&#117;&#99;s&#99;.&#101;&#100;&#117;" rel="nofollow"
TARGET = _BLANK>Ed Green</A>.
</P>

<H2>Reference</H2>
<p>
Green RE, Krause J, Briggs AW, Maricic T, Stenzel U, Kircher M, Patterson N, Li H, Zhai W, Fritz MH
<em>et al</em>.
<a href="http://www.sciencemag.org/content/328/5979/710.abs" target="_blank">
A draft sequence of the Neandertal genome</a>.
<em>Science</em>. 2010 May 7;328(5979):710-22.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/20448178" target="_blank">20448178</a>
</p>
