<H2>Description</H2>
<P>
This track shows regions of the human genome with a strong signal for depletion
of Neandertal-derived alleles (regions from the 
<A HREF="/cgi-bin/hgTrackUi?&c=chr1&g=ntSssZScorePMVar">Sel Swp Scan 
(S)</A> track with S scores in the lowest 5%),
which may indicate an episode of positive selection in early
humans.</P>

<H2>Display Conventions and Configuration</H2>
<P>
Grayscale shading is used as a rough indicator of the strength of the
score; the darker the item, the stronger its negative score.  The
strongest negative score (-8.7011) is shaded black, and the shading
lightens from dark to light gray as the negative score weakens (weakest
score is -4.3202).
</P>

<H2>Methods</H2>
<P>
Green <em>et al.</em> identified single-base sites that are
polymorphic among five modern human genomes of diverse ancestry
(in the 
<A HREF="/cgi-bin/hgTrackUi?&c=chr1&g=ntModernHumans">Modern Human 
Seq</A> track) plus the human reference
genome, and determined ancestral or derived state of each
single nucleotide polymorphism (SNP) by comparison with the chimpanzee
genome.  The SNPs are displayed in the 
<A HREF="/cgi-bin/hgTrackUi?&c=chr1&g=ntSssSnps">S SNPs</A> track.
The human allele states were used to estimate an expected number
of derived alleles in Neandertal in the 100,000-base window around 
each SNP, and a measure called the S score was developed, displayed in the 
<A HREF="/cgi-bin/hgTrackUi?&c=chr1&g=ntSssZScorePMVar">Sel Swp Scan 
(S)</A> track, to compare the observed
number of Neandertal alleles in each window to the expected number.  
An S score significantly less than zero indicates a reduction of 
Neandertal-derived alleles (or an increase of human-derived alleles not found in
Neandertal), consistent with the scenario of positive selection in
the human lineage since divergence from Neandertals.
</P>
<P>
Genomic regions of 25,000 or more bases in which all polymorphic sites
were at least 2 standard deviations below the expected value were
identified, and S was recomputed on each such region.  Regions with S
scores in the lowest 5% (strongest negative scores) were prioritized
for further analysis as described in Green <em>et al.</em>.
</P>

<H2>Credits</H2>
<P>
This track was produced at UCSC using data generated by
<A HREF="mailto:&#101;&#100;&#64;&#115;&#111;e.&#117;&#99;s&#99;.&#101;&#100;&#117;"
title="&#101;&#100;&#64;&#115;&#111;e.&#117;&#99;s&#99;.&#101;&#100;&#117;" rel="nofollow"
TARGET=_BLANK>Ed Green</a>.
</P>

<H2>References</H2>
<p>
Green RE, Krause J, Briggs AW, Maricic T, Stenzel U, Kircher M, Patterson N, Li H, Zhai W, Fritz MH
<em>et al</em>.
<a href="http://www.sciencemag.org/content/328/5979/710.long" target="_blank">
A draft sequence of the Neandertal genome</a>.
<em>Science</em>. 2010 May 7;328(5979):710-22.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/20448178" target="_blank">20448178</a>
</p
