<h2>Description</h2>
<p>
The PanelApp tracks show regions that are related to human disorders. These can be either
genes, short tandem repeats, or copy number variants. The regions were curated by groups of
specialists collaborating using the PanelApp web tool. The primary website is <a target="_blank"
href="https://panelapp.genomicsengland.co.uk/#!">Genomics England PanelApp</a>.
Another deployment of the website, with different data, is 
<a target=_blank href="https://panelapp-aus.org/">PanelApp Australia</a>.
</p>

<p>
Originally, PanelApp was developed to aid interpretation of participant genomes in the
<a href="https://www.genomicsengland.co.uk/initiatives/100000-genomes-project" target="_blank">
100,000 Genomes Project</a>.
<a target="_blank" href="https://panelapp.genomicsengland.co.uk/#!">Genomics England PanelApp</a>
is now being used as the platform for achieving consensus on gene panels in the <a target="_blank"
    href="https://www.england.nhs.uk/genomics/nhs-genomic-med-service/"> NHS
    Genomic Medicine Service (GMS)</a>. Later, the same platform was also deployed by
<a target=_blank
   href="https://www.australiangenomics.org.au/tools-and-resources/panelapp-australia/">Australian
   Genomics</a>.
</p>

<p>
<a href="https://panelapp.genomicsengland.co.uk/panels/entities/" target="_blank">Genes and genomic
entities</a>, so short tandem repeats/STRs and copy number variants/CNVs,
have been reviewed by experts to enable a community consensus to be reached on which
genes and genomic entities should appear on a <b>diagnostics grade panel</b> for each disorder.
A <b>rating system</b> (confidence level 0 - 3) is used to classify the level of evidence
supporting association with phenotypes covered by the gene panel in question.
</p>

<p>
There are six subtracks in total: Three different types (genes, STRs, and CNVs), and these 
three exist for both countries, England and Australia. The three types of tracks are:</p>

<ul>
  <li>
    <b>PanelApp Genes (PanelApp Genes):</b>
    <br>
    shows genes with evidence supporting a gene-disease relationship.
</li>
  <br>
  <li>
    <b>PanelApp STRs (PanelApp STRs):</b>
    <br>
    shows short tandem repeats that can be disease-causing when a particular number of repeats is
    present.</li>
  <br>
  <li>
    <b>Only on hg38: PanelApp Regions (PanelApp CNV Regions):</b>
    <br>
    shows copy-number variants (region-loss and region-gain) with evidence supporting a gene-disease
    relationship.</li>
</ul>


<h2>Display Conventions</h2>

<p>
There are a few differences between the Genomics England and the Australian Genomics tracks:<br><br>
<b>Genomics England</b>
<ul>
<li>By default, only items with a version greater than or equal to 1 are displayed. This
can be changed in the track configuration menu.</li>
</ul>
<b>Australian Genomics</b>
<ul>
<li>For the <b>PanelApp Genes</b> track, only items from the <b>Mendeliome</b> and
<b>Incidentalome</b> panels are displayed by default. The sum total of these two panels represents all the
gene-disease relationships available on the platform and provides an overarching assessment of the
association between each gene and disease(s). Pulling information from other panels can be
confusing, as the same gene may have different ratings across different panels.</li>

<li>By default, all versions are displayed (versions greater than 0). This
can be changed in the track configuration menu.</li>
</ul></p>

<p>
The individual tracks are colored by <b>confidence level:</b>

<ul>
<li><b><font color="#32CD32">Score 3 (lime green)</font></b> - <b>High level of evidence</b> 
for this gene-disease association. Demonstrates confidence that this gene should be 
used for genome interpretation.</li>
<li><b><font color="#FFBF00">Score 2 (amber)</font></b> - <b>Moderate evidence</b> 
for this gene-disease association. This gene should not be used for genomic 
interpretation.</li>
<li><b><font color="red">Score 0 or 1 (red)</font></b> - <b>Not enough evidence</b> 
for this gene-disease association. This gene should not be used for 
genomic interpretation. </li>
</ul>
<p>
Mouseover on items shows the gene name, panel associated, mode of inheritance 
(if known), phenotypes related to the gene, and confidence level. Tracks can 
be filtered according to the confidence 
level of disease association evidence. For more information on 
the use of this data, see the PanelApp
<a target="_blank" href="https://panelapp.genomicsengland.co.uk/#!FAQs">FAQs</a>.
</p>

<h2>Data Access</h2>
<p>
The raw data can be explored interactively with the
<a target="_blank" href="/cgi-bin/hgTables">Table Browser</a> or the
<a target="_blank" href="/cgi-bin/hgIntegrator">Data Integrator</a>.
For automated analysis, the data may be queried from our
<a target="_blank" href="/goldenPath/help/api.html">REST API</a>.
</p>
<p>
For automated download and analysis, the genome annotation is stored in a bigBed file that
can be downloaded from
<a href="http://hgdownload.soe.ucsc.edu/gbdb/$db/panelApp/" target="_blank">our download server</a>.
The files for this track are called <tt>genes.bb</tt>, <tt>tandRep.bb</tt>, and <tt>cnv.bb</tt>. Individual
regions or the whole genome annotation can be obtained using our tool <tt>bigBedToBed</tt>,
which can be compiled from the source code or downloaded as a precompiled
binary for your system. Instructions for downloading source code and binaries can be found
<a href="http://hgdownload.soe.ucsc.edu/downloads.html#utilities_downloads">here</a>.
The tool
can also be used to obtain only features within a given range, e.g. 
<tt>bigBedToBed http://hgdownload.soe.ucsc.edu/gbdb/$db/panelApp/genes.bb -chrom=chr21 -start=0 -end=100000000 stdout</tt></p>

<p>
Please refer to our
<a target="_blank" href="https://groups.google.com/a/soe.ucsc.edu/forum/#!forum/genome">
mailing list archives</a> for questions, or our
<a target="_blank" href="/FAQ/FAQdownloads.html#downloads36">
Data Access FAQ</a> for more information.
</p>
<p>
Data is also freely available on the
<a target="_blank" href="https://panelapp.genomicsengland.co.uk/#!API">Genomics England PanelApp API</a>
and the <a target=_blank href="https://panelapp-aus.org/api/docs/">Australia PanelApp API</a>.
</p>

<h2>Updates and archiving of old releases</h2>
<p>
This track is updated automatically every week. If you need to access older releases of the data,
you can download them from our <a href="https://hgdownload.soe.ucsc.edu/goldenPath/archive/$db/panelApp/">archive directory</a> on the download server. To load them into the browser, select a week on the archive directory, copy the link to a file, go to <a href="http://genome.ucsc.edu/cgi-bin/hgCustom">My Data &gt; Custom Tracks</a>, click "Add custom track", paste the link into the box, and click "Submit".
</p>

<h2>Methods</h2>
<p>
PanelApp files were reformatted at UCSC to the <a target="_blank"
href="/goldenPath/help/bigBed.html">bigBed</a> format. The script that updates the track is called 
<tt>doPanelApp.py</tt> and can be found in our <a target="_blank"
href="https://github.com/ucscGenomeBrowser/kent/tree/master/src/hg/utils/otto/panelApp/doPanelApp.py">GitHub repository</a>.
</p>

<h2>Credits</h2>
<p>
Thank you to Genomics England PanelApp, especially Catherine Snow for technical
coordination and consultation, and Zornitza Stark from Australia PanelApp.
Thanks to Beagan Nguy, Lou Nassar, Christopher Lee, Daniel Schmelter, Ana
Benet-Pag&egrave;s and Maximilian Haeussler of the Genome Browser team for the
creation of the tracks.
</p>

<h2>Reference</h2>
<p>
    Martin AR, Williams E, Foulger RE, Leigh S, Daugherty LC, Niblock O, Leong IUS, Smith KR,
    Gerasimenko O, Haraldsdottir E <em>et al</em>.
    <a href="https://www.ncbi.nlm.nih.gov/pubmed/31676867" target="_blank">
    PanelApp crowdsources expert knowledge to establish consensus diagnostic gene panels</a>.
    <em>Nat Genet</em>. 2019 Nov;51(11):1560-1565.
    PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31676867" target="_blank">31676867</a>
</p>
