<h2>Description</h2>

<p>
The "Prediction Scores" container track includes subtracks showing the results of prediction
scores.</p>

<h3>BayesDel</h3>
<p>BayesDel is a deleteriousness meta-score for coding and non-coding variants, single nucleotide
variants, and small insertion/deletions. The range of the score is from -1.29334 to 0.75731.
The higher the score, the more likely the variant is pathogenic.</p>

<p>
For gene discovery research, a universal cutoff value (0.0692655 with MaxAF, -0.0570105 without
MaxAF) was obtained by maximizing sensitivity and specificity in classifying ClinVar variants;
Version 1 (build date 2017-08-24).</p>
<p>
For clinical variant classification, Bayesdel thresholds have been calculated for a variant to
reach various levels of evidence; please refer to Pejaver et al. 2022 for general application
of these scores in clinical applications.
</p>

<h2>Display Conventions and Configuration</h2>

<h3>BayesDel</h3>
<p>There are eight subtracks for the BayesDel track: four include pre-computed MaxAF-integrated BayesDel
scores for missense variants, one for each base. The other four are of the same format, but scores
are not MaxAF-integrated.</p>

<p>For SNVs, at each genome position, there are three values per position, one for every possible
nucleotide mutation. The fourth value, &quot;no mutation&quot;, representing the reference allele,
(e.g. A to A) is always set to zero.</p>

<p><b>Note:</b> There are cases in which a genomic position will have one value missing.
</p>

<p>When using this track, zoom in until you can see every base pair at the top of the display.
Otherwise, there are several nucleotides per pixel under your mouse cursor and instead of an actual
 score, the tooltip text will show the average score of all nucleotides under the cursor. This is
indicated by the prefix &quot;~&quot; in the mouseover.
</p>

<h2>Data Access</h2>
<p>BayesDel scores are available at the
<a target="_blank" href="https://fenglab.chpc.utah.edu/BayesDel/BayesDel.html">BayesDel website</a>.

<h2>Methods</h2>
<p>Data was converted from the files provided on the
<a target="_blank" href="https://drive.google.com/drive/folders/1K4LI6ZSsUGBhHoChUtegC8bgCt7hbQlA?usp=sharing">BayesDel_170824 Database</a>.
The number 170824 is the date (2017-08-24) the scores were created. Both sets of BayesDel scores are
available in this database, one integrated MaxAF (named BayesDel_170824_addAF) and one without
(named BayesDel_170824_noAF). Data conversion was performed using
<a a target="_blank"  href="https://github.com/ucscGenomeBrowser/kent/tree/master/src/hg/makeDb/outside/bayesDel/hg19">
custom Python scripts</a>.
</p>

<h2>Credits</h2>
<p>Thanks to the BayesDel team for providing precomputed data, and to Tiana Pereira, Christopher
Lee, Gerardo Perez, and Anna Benet-Pages of the Genome Browser team.</p>

<h2>References</h2>
<p>
Feng BJ.
<a href="https://doi.org/10.1002/humu.23158" target="_blank">
PERCH: A Unified Framework for Disease Gene Prioritization</a>.
<em>Hum Mutat</em>. 2017 Mar;38(3):243-251.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27995669" target="_blank">27995669</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5299048/" target="_blank">PMC5299048</a>
</p>

<p>
Pejaver V, Byrne AB, Feng BJ, Pagel KA, Mooney SD, Karchin R, O'Donnell-Luria A, Harrison SM,
Tavtigian SV, Greenblatt MS <em>et al</em>.
<a href="https://linkinghub.elsevier.com/retrieve/pii/S0002-9297(22)00461-X" target="_blank">
Calibration of computational tools for missense variant pathogenicity classification and ClinGen
recommendations for PP3/BP4 criteria</a>.
<em>Am J Hum Genet</em>. 2022 Dec 1;109(12):2163-2177.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36413997" target="_blank">36413997</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9748256/" target="_blank">PMC9748256</a>
</p>

<p>
Tian Y, Pesaran T, Chamberlin A, Fenwick RB, Li S, Gau CL, Chao EC, Lu HM, Black MH, Qian D.
<a href="https://doi.org/10.1038/s41598-019-49224-8" target="_blank">
REVEL and BayesDel outperform other in silico meta-predictors for clinical variant
classification</a>.
<em>Sci Rep</em>. 2019 Sep 4;9(1):12752.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31484976" target="_blank">31484976</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6726608/" target="_blank">PMC6726608</a>
</p>
