<H2>Description</H2>
<P>
This track displays regulatory potential (RP) scores computed from
alignments of human, chimpanzee (panTro2), macaque (rheMac2), mouse (mm8), rat
(rn4), dog (canFam2), and cow (bosTau2).</P>

<P>RP scores compare frequencies of short alignment patterns between known
regulatory elements and neutral DNA. 
The sensitivity and specificity of RP scores were calibrated on the
hemoglobin beta gene cluster.  These results suggest a threshold of ~0.00 
for the identification of new putative regulatory elements. </P>

<P>The default viewing range for this track is from 0.0 to 0.1.
Score values below the 0.0 default lower limit indicate resemblance to 
alignment patterns
typical of neutral DNA, while score values above the 0.1 default upper limit 
indicate very
marked resemblance to alignment patterns typical of regulatory elements in the
training set. The range of RP scores from 0.0 to 0.1 contains the prediction
threshold suggested by calibration studies, and provides an effective
visualization of the score for most genomic loci. However, the user can specify
different viewing ranges if desired. Note: Absence of a score value at a given
location indicates lack of sufficient alignment -- scores are computed
for all regions of the reference genome in which no region of more than
100 bases lacks alignment in at least three non-human species.</P>

<P>This track may be configured in a variety of ways to highlight different
aspects of the displayed information. Click the 
<A HREF="../goldenPath/help/hgWiggleTrackHelp.html">Graph 
configuration help</A> link for an explanation of the configuration options.</P>

<H2>Methods</H2>

<P>The comparison employs log-ratios of transitions probabilities from two
variable order Markov models. Training the score entails selecting
appropriate alphabet (alignment column symbols) and maximal order (length of
the longest patterns = order + 1) for the Markov models, and estimating their
transition probabilities, based on alignment data from known
regulatory elements and ancestral repeats. The scores in this track 
are computed using a maximal order of 2. </P>

<P>In the track, score values are displayed using a system of overlapping
windows of size 100 bp along sufficiently alignable portions of the
human sequence. Log-ratios are added over positions in a window, and
the sum is normalized for length. </P>

<H2>Credits</H2>

<P>Work on RP scores is performed by members of the 
<A HREF="http://www.bx.psu.edu/" TARGET="_blank">Comparative Genomics and
Bioinformatics Center at Penn State University</A>. More information on this
research and the collection of known regulatory elements used in training the
score can be found at this site. </P>

<H2>References</H2>

<P>King DC, Taylor J, Elnitski L, Chiaromonte F, Miller W, Hardison
RC. 
<A HREF="http://www.genome.org/cgi/content/abstract/15/8/1051" 
TARGET = _BLANK>Evaluation of regulatory potential and conservation scores for 
detecting cis-regulatory modules in aligned mammalian genome sequences.</A>. 
<em>Genome Res</em>. 2005 Aug;<B>15</B>(8):1051-60.</P>

<P>Kolbe D, Taylor J, Elnitski L, Eswara P, Li J, Miller W, Hardison RC,
Chiaromonte F. 
<A HREF="http://www.genome.org/cgi/content/full/14/4/700"
TARGET="_BLANK">Regulatory potential scores from genome-wide three-way
alignments of human, mouse, and rat</A>. 
<em>Genome Res.</em> 2004 Apr;<B>14</B>(4):700-7. </P>
