<H2>Description</H2>
<P>
This track consolidates all the Simple Nucleotide Polymorphisms
into a single track.  It is the union of the Overlap SNPs,
Random SNPs, Affymetrix 120K SNP, and Affymetrix 10K SNP tracks that
previously existed in the Genome Browser.  </P>

<a name="source">
<B>Variant Sources</B>
<UL>
<LI>
The <B>Random</B> variants were detected by aligning reads from random
genomic clones from a diverse pool of human DNA against the
genome. <BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(These were previously located in 
the Random SNPs track.)
<LI>
The <B>Mixed</B> variants were detected by both clone overlaps and the
random genomic clones. <BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(These were 
previously located in the Random SNPs track.)
<LI>
The <B>BAC Overlaps</B> variants were detected primarily by
looking at overlaps between clones that cover the same region of the
genome. <BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(These were previously located in 
the Overlap SNPs track.)
<LI>
The <B>Other</B> variants were found using alternative
methods. <BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(These were previously located
in the Overlap SNPs track.)
<LI>
The <B>Affymetrix Genotyping Array 10K</B> variants are those Single
Nucleotide Polymorphisms that have been included in the Affymetrix
10K SNP Genotyping Array.  <BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(These were 
previously located in the Affymetrix 10K SNP track.)
<LI>
The <B>Affymetrix Genotyping Array 120K</B> variants are those Single
Nucleotide Polymorphisms that have been included in the Affymetrix
120K SNP Genotyping Array. <BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(These were 
previously located in the Affymetrix 120K SNP track.)
</UL>

<a name="type">
<B>Variant Types</B>
<UL>
<LI> <B>Single Nucleotide Polymorphisms:</B> variants where the two alleles are each a single base long.
<LI> <B>Insertions and Deletions:</B> variants where one allele contains no bases.
<LI> <B>Segmental Mutations:</B> variants with multiple nucleotide differences.
</UL>

<H2>Filtering</H2>
<P>
The SNPs in this track include all known polymorphisms that
can be mapped against the current assembly.  These include known point
mutations (Single Nucleotide Polymorphisms), insertions, deletions,
and segmental mutations from the current build of 
<A HREF="http://www.ncbi.nlm.nih.gov/SNP/" TARGET=_blank>dbSnp</A>, which is 
shown in the Genome Browser <A HREF="/goldenPath/releaseLog.html">release log</A>.
</P>
<P>
There are three major cases that are not mapped and/or annotated:
<UL>
<LI> 
Submissions that are completely masked as repetitive elements. 
These are dropped from any further computations. This set of
reference SNPs is can be found in chromosome &quot;rs_chMasked&quot;
on the <A HREF="ftp://ftp.ncbi.nih.gov/snp/" TARGET=_blank>dbSNP ftp
site</A>.
<LI>
Submissions that are defined in a cDNA context with extensive
splicing. These SNPs are typically annotated on refSeq mRNAs through a
separate annotation process. Effort is being made to reverse map these
variations back to contig coordinates, but that has not been
implemented. For now, you can find this set of variations in
&quot;rs_chNotOn&quot; on the <A HREF="ftp://ftp.ncbi.nih.gov/snp/"
TARGET=_blank>dbSNP ftp site</A>.
<LI>
Submissions with excessive hits to the genome. Variations with 3+ hits
to the genome are not included in the tracks, but are available in
&quot;rs_chMulti&quot; on the <A HREF="ftp://ftp.ncbi.nih.gov/snp/"
TARGET=_blank>dbSNP ftp site</A>.
</UL>
</P>
<P>The heuristics for the non-SNP variations (i.e. named elements and
STRs) are quite conservative; therefore, some of these are probably lost. This
approach was chosen to avoid false annotation of variation in
inappropriate locations.</P>

<H2>Supporting Details</H2>
<P> 
Positional information can be found in the annotations section
of the Genome Browser 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">downloads</A> page, 
which is organized by species and assembly. Non-positional information
can be found in the 
<A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/hgFixed/database/">shared
data</A> section of the same page, where it is split into tables by
organism: 
<TT><B><FONT size=3>dbSnpRsHg</FONT></B></TT> for Human, 
<TT><B><FONT size=3>dbSnpRsMm</FONT></B></TT> for Mouse, and 
<TT><B><FONT size=3>dbSnpRsRn</FONT></B></TT> for Rat.

<H2>Credits</H2>
<P>
Thanks to the <A HREF="http://snp.cshl.org/" TARGET=_blank>SNP
Consortium</A> and <A HREF="http://www.ncbi.nlm.nih.gov"
TARGET=_blank>NIH</A> for providing the public data, which are available from 
<A HREF="http://www.ncbi.nlm.nih.gov/SNP" TARGET=_blank>dbSnp</A> at 
NCBI.</P>
<P>
Thanks to <A HREF="http://www.perlegen.com" TARGET=_blank>Perlegen Sciences, 
Inc.</A> for providing additional SNPs from their database.
Additional information about the Perlegen SNP discovery process can be
found in Patil, N. (2001) 
<A HREF="http://www.sciencemag.org/cgi/content/abstract/294/5547/1719"
TARGET=_blank> 
Blocks of Limited Haplotype Diversity Revealed by High-Resolution
Scanning of Human Chromosome 21</A>. <i>Science</i> 294:1719-1723.
</p>
<P>
Thanks to <A HREF="http://www.affymetrix.com"
TARGET=_blank>Affymetrix, Inc.</A> for developing the genotyping
arrays. For more details on this genotyping assay, please see the
supplemental information on the 
<A HREF="http://www.affymetrix.com/support/technical/byproduct.affx?product=10k" 
TARGET=_blank>Affymetrix 10K SNP</A> and 
<A HREF="http://www.affymetrix.com/support/technical/sample_data/genotyping_data.affx" 
TARGET=_blank>Affymetrix 120K SNP</A> products. Additional information, 
including genotyping data, is available from the details pages for the 
Affymetrix 120K SNP and Affymetrix 10K SNP tracks.</P>

<H2>Terms of Use for the Affymetrix data</H2>
<P> Please see the <A HREF="http://www.affymetrix.com/site/terms.affx"
TARGET=_blank>Terms and Conditions</A> page on the Affymetrix website for 
restrictions on the use of their data. 

<!--
<P>During sample preparation, an <B>enzyme</B> is used to fragment the
genomic DNA into smaller pieces.  The restriction endonucleases
<I><B>HindIII</B></I> and <I><B>XbaI</B></I> have been used to generate this
data.</P>

<P>Genotypes are reported as pairs of basecalls representing the two
observed alleles.  Individual genotypes for which no call was made are
labeled 'NC'.  A minimum of 46 genotype calls in the set of 54
individuals are included for each SNP.</P>

<P><B>Allele Frequency</B> for the 'A' allele is two times the number
of 'AA' calls plus the number of 'AB' calls, divided by two times the
total number of calls.  Allele Frequency for the 'B' allele is two
times the number of 'BB' calls plus the number of 'AB' calls, divided
by two times the total number of calls.  The <B>Minimum Allele
Frequency</B> is the lesser of these two values.
<B>Heterozygosity</B> is two times the product of the two allele
frequencies.</P> 
-->

