<!DOCTYPE html>
<html>
<head>
</head>

<body>
<h2>Description</h2>
<p>
SpliceAI is an <a href="https://github.com/Illumina/SpliceAI" target="_blank">open-source</a> deep
learning splicing prediction algorithm that can predict splicing alterations caused by DNA variations. 
Such variants may activate nearby cryptic splice sites, leading to abnormal transcript isoforms.
SpliceAI was developed at Illumina; a 
<a href="https://spliceailookup.broadinstitute.org" target="_blank">lookup tool</a> 
is provided by the Broad institute.<br><br>
<b>Important</b>: The SpliceAI data on the UCSC Genome Browser is directly from 
Illumina (See Data Access below). However, since SpliceAI refers to the algorithm, and not the computed dataset, 
the data on the Broad server or other sources may have some differences between them.
</p>

<h2>Display Conventions and Interpretation</h2>
<p>
Variants are colored by their predicted effects:
<ul>
<li><b><font color="#FF0000">Acceptor gain (red)</font></b> </li>
<li><b><font color="#FF8000">Acceptor loss (orange)</font></b> </li>
<li><b><font color="#0000FF">Donor gain (blue)</font></b> </li>
<li><b><font color="#D400FF">Donor loss (violet)</font></b> </li>
</ul>
Mouseover on items shows the variant, gene name, type of change (donor gain/loss, acceptor gain/loss), 
location of affected cryptic splice, and spliceAI score. Clicking on any item brings up a table with this
information.
</p>
<p>
The scores range from 0 to 1 and can be interpreted as the 
probability of the variant being splice-altering. In the paper, a detailed characterization is 
provided for 0.2 (high recall), 0.5 (recommended), and 0.8 (high precision) cutoffs.</p>

<h2>Data Access</h2>
These data are not available for download from the Genome Browser. 
The raw data can be found directly on
<a target="_blank" href="https://basespace.illumina.com/s/otSPW8hnhaZR">Illumina</a>. 
See below for a copy of the license restrictions pertaining to these data.
</p>

<h2>License</h2>
<p>
FOR ACADEMIC AND NOT-FOR-PROFIT RESEARCH USE ONLY. The SpliceAI scores are 
made available by Illumina only for academic or not-for-profit research only. 
By accessing the SpliceAI data, you acknowledge and agree that you may only 
use this data for your own personal academic or not-for-profit research only, 
and not for any other purposes. You may not use this data for any for-profit, 
clinical, or other commercial purpose without obtaining a commercial license 
from Illumina, Inc.
</p>

<h2>Methods</h2>
<p>
The data were downloaded from <a 
target="_blank" href="https://basespace.illumina.com/s/otSPW8hnhaZR">Illumina</a>. 
The spliceAI scores are represented in the VCF INFO field as 
<code style="background-color: lightgray;">SpliceAI=G|OR4F5|0.01|0.00|0.00|0.00|-32|49|-40|-31</code> <br><br>
Here, the pipe-separated fields contain 
<ul>
<li>ALT allele</li>
<li>Gene name</li>
<li>Acceptor gain score</li>
<li>Acceptor loss score</li>
<li>Donor gain score</li>
<li>Donor loss score</li>
<li>Relative location of affected cryptic acceptor</li>
<li>Relative location of affected acceptor</li>
<li>Relative location of affected cryptic donor</li>
<li>Relative location of affected donor</li>
</ul>
Since most of the values are 0 or almost 0, we selected only those variants 
with a score equal to or greater than 0.02.<br><br>
The complete processing of this track can be found in the <a target="_blank" 
href="https://github.com/ucscGenomeBrowser/kent/blob/master/src/hg/makeDb/scripts/spliceAI/spliceAI.py">
makedoc</a>.
</p>

<h2>References</h2>
<p>
Jaganathan K, Kyriazopoulou Panagiotopoulou S, McRae JF, Darbandi SF, Knowles D, Li YI, Kosmicki JA,
Arbelaez J, Cui W, Schwartz GB <em>et al</em>.
<a href="https://linkinghub.elsevier.com/retrieve/pii/S0092-8674(18)31629-5" target="_blank">
Predicting Splicing from Primary Sequence with Deep Learning</a>.
<em>Cell</em>. 2019 Jan 24;176(3):535-548.e24.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30661751" target="_blank">30661751</a>
</p>
</body>
</html>
