<br><br>
<b>Important</b>: The SpliceAI variant impact data on the UCSC Genome Browser is directly from 
Illumina (See <a href="#dataAccess">Data Access</a> below). However, since SpliceAI refers to the
algorithm, and not the computed dataset, the data on the Broad server or other sources may have
some differences between them.
</p>
<h2>Description</h2>
<p>
<a href="https://github.com/Illumina/SpliceAI" target="_blank">SpliceAI</a> is an open-source deep
learning algorithm that predicts splicing probability for nucleotides and 
as a result can score DNA variants for splicing impact.
Such variants may activate nearby cryptic splice sites, leading to abnormal transcript isoforms.
SpliceAI was developed at Illumina; a 
<a href="https://spliceailookup.broadinstitute.org" target="_blank">lookup tool</a> 
is provided by the Broad institute.
</p>

<p>The spliceAI algorithm is run on the genome sequence itself and scores each
nucleotide for the probability that it is a donor or acceptor site, on both the
forward and the reverse strand. Then variants are added and the new sequence is
scored again. The "wildtype" container track shows the scores for the genome
sequence itself and the "variants" container track shows the impact of all
possible variants close to known splice sites. The "wildtype" subtracks are
useful when looking at new transcript models, to evaluate how likely exon
boundaries are. The "variants" subtracks are used to evaluate the impact of
variants onto splicing, typically in medical diagnostics.
</p>

<h3>Why are some variants not scored by SpliceAI?</h3>
<p>
SpliceAI only annotates variants close to splice sites of genes defined by the 
Gencode gene annotation track. Additionally, SpliceAI does not annotate variants if they are
close to chromosome ends (5kb on either side), deletions of length greater than
twice the input parameter -D, or inconsistent with the reference fasta file.
</p>

<h3>What are the differeneces between masked and unmasked tracks?</h3>
<p>
The unmasked tracks include splicing changes corresponding to strengthening annotated splice sites
and weakening unannotated splice sites, which are typically much less pathogenic than weakening
annotated splice sites and strengthening unannotated splice sites. The delta scores of such splicing
changes are set to 0 in the masked files. We recommend using the unmasked tracks for alternative
splicing analysis and masked tracks for variant interpretation.
</p>

<h2>Display Conventions and Interpretation</h2>
<p>
Variants are colored according to Walker et al. 2023 splicing imact:
</p>
<ul>
<li><b><font color="#FF8000">Predicted impact on splicing: Score &gt;&#61; 0.2 </font></b> </li>
<li><b><font color="#808080">Not informative: Score &lt; 0.2 and &gt; 0.1 </font></b> </li>
<li><b><font color="#0000FF">No impact on splicing: Score &lt;&#61; 0.1 </font></b> </li>
</ul>
</p>
Mouseover on items shows the variant, gene name, type of change (donor gain/loss, acceptor
gain/loss), location of affected cryptic splice, and spliceAI score. Clicking on any item brings up
a table with this information.
</p>
<p>
The scores range from 0 to 1 and can be interpreted as the 
probability of the variant being splice-altering. In the paper, a detailed characterization is 
provided for 0.2 (high recall), 0.5 (recommended), and 0.8 (high precision) cutoffs.</p>

<h2>Methods</h2>
<p>
The data were downloaded from <a 
target="_blank" href="https://basespace.illumina.com/s/otSPW8hnhaZR">Illumina</a>. 
The spliceAI scores are represented in the VCF INFO field as 
<code style="background-color: lightgray;">SpliceAI=G|OR4F5|0.01|0.00|0.00|0.00|-32|49|-40|-31</code> <br><br>
Here, the pipe-separated fields contain 
<ul>
  <li>ALT allele</li>
  <li>Gene name</li>
  <li>Acceptor gain score</li>
  <li>Acceptor loss score</li>
  <li>Donor gain score</li>
  <li>Donor loss score</li>
  <li>Relative location of affected cryptic acceptor</li>
  <li>Relative location of affected acceptor</li>
  <li>Relative location of affected cryptic donor</li>
  <li>Relative location of affected donor</li>
</ul>
<p>
Since most of the values are 0 or almost 0, we selected only those variants 
with a score equal to or greater than 0.02.
</p>
<p>
The complete processing of this track can be found in the <a target="_blank" 
href="https://github.com/ucscGenomeBrowser/kent/blob/master/src/hg/makeDb/scripts/spliceAI/spliceAI.py">
makedoc</a>.
</p>

<a name="dataAccess"></a>
<h2>Data Access</h2>
These data are not available for download from the Genome Browser. 
The raw data can be found directly on
<a target="_blank" href="https://basespace.illumina.com/s/otSPW8hnhaZR">Illumina</a>. 
See below for a copy of the license restrictions pertaining to these data.
</p>

<h2>License</h2>
<p>
FOR ACADEMIC AND NOT-FOR-PROFIT RESEARCH USE ONLY. The SpliceAI scores are 
made available by Illumina only for academic or not-for-profit research only. 
By accessing the SpliceAI data, you acknowledge and agree that you may only 
use this data for your own personal academic or not-for-profit research only, 
and not for any other purposes. You may not use this data for any for-profit, 
clinical, or other commercial purpose without obtaining a commercial license 
from Illumina, Inc.
</p>

<h2>Credits</h2>
<p>
Thanks to Illumina for making the data available. Thanks to Michael Hiller, Francois Lecoquierre and
Jean-Madeleine de Sainte Agathe for making available and suggesting the SpliceAI wildtype tracks.
</p>

<h2>References</h2>
<p>
Jaganathan K, Kyriazopoulou Panagiotopoulou S, McRae JF, Darbandi SF, Knowles D, Li YI, Kosmicki JA,
Arbelaez J, Cui W, Schwartz GB <em>et al</em>.
<a href="https://linkinghub.elsevier.com/retrieve/pii/S0092-8674(18)31629-5" target="_blank">
Predicting Splicing from Primary Sequence with Deep Learning</a>.
<em>Cell</em>. 2019 Jan 24;176(3):535-548.e24.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30661751" target="_blank">30661751</a>
</p>

<p>
Walker LC, Hoya M, Wiggins GAR, Lindy A, Vincent LM, Parsons MT, Canson DM, Bis-Brewer D, Cass A,
Tchourbanov A <em>et al</em>.
<a href="https://linkinghub.elsevier.com/retrieve/pii/S0002-9297(23)00203-3" target="_blank">
Using the ACMG/AMP framework to capture evidence related to predicted and observed impact on
splicing: Recommendations from the ClinGen SVI Splicing Subgroup</a>.
<em>Am J Hum Genet</em>. 2023 Jul 6;110(7):1046-1067.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37352859" target="_blank">37352859</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10357475/" target="_blank">PMC10357475</a>
</p>
