<H2>Description</H2>
<P>
Genome wide methylation (MeDIP-seq and MRE-seq), histone H3 lysine 4 trimethylation 
(H3K4me3) and gene expression (RNA-seq and RNA-seq (SMART)) data were generated from 
postmortem human frontal cortex gray matter of a 57 year-old male. This was done
to investigate the role that intragenic, tissue-specific CpG island methylation 
plays in controlling gene expression 
(<a href="http://www.nature.com/nature/journal/v466/n7303/full/nature09165.html" TARGET=_BLANK>Maunakea, et al. 2010</a>).
</P>

<H2>Display Conventions and Configuration</H2>
<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are <A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
The following views are in this track:
</P>
<DL>
<DT><I>Raw Singal</I>
<DD>Density graph (wiggle) of signal enrichment.
<DT><I>CpG score</I>
<DD>DNA methylation score on CpG sites.
</DL>

<h2>Methods</h2>
<p>
DNA, RNA and native chromatin were extracted using standard methods; assay specific 
methods are described below. MRE-seq, MeDIP-seq, H3K4me3 ChIP-seq, RNA-seq and RNA-seq 
(SMART) libraries were sequenced using an Illumina Genome Analyzer II. Sequencing reads 
are available through the NCBI SRA 
(<a href="http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?study=SRP002318" TARGET=_BLANK>study accession number SRP002318</a>).
</p>

<h3>MeDIP-seq (Methylated DNA immunoprecipitation and sequencing)</h3>
<p>
MeDIP-seq uses immunoprecipitation to extract the methylated fraction of the genome. 
Purified DNA was first sheared and processed following the Illumina Genomic DNA Library 
Kit protocol. These DNA fragments were then immunoprecipitated using an antibody raised 
against 5-methylcytosine, the methylated form of cytosine, before constructing a 
library, which was sequenced and mapped to the genome.
</p>

<h3>MRE-seq (Methyl-sensitive restriction enzyme digest and sequencing)</h3>
<p>
MRE-seq identifies unmethylated CpG sites by sequencing size-selected fragments from 
parallel DNA digestions with the MREs HpaII, Hin6I, and AciI. Since these enzymes 
require unmethylated CpG sites within their recognition sequences to cut DNA, 
identifying the end of each fragment generated allows inference of a single 
unmethylated cytosine. The 3 digests were combined and size-selected by gel 
electrophoresis to enrich for unmethylated CpG sites in close proximity. A library was 
constructed and sequenced; the sequence reads were then mapped to the genome with the 
additional requirement that they map to a known MRE recognition site.
</p>

<h3>H3K4me3 ChIP-seq (Histone H3 lysine 4 trimethylation chromatin immunoprecipitation 
and sequencing)</h3>
<p>
Chromatin immunoprecipitation was performed to enrich for histone H3 modified at lysine 
position 4 with trimethylation (H3K4me3), as this histone modification is associated 
with promoters. A ChIP-seq library was constructed as described in Robertson, et al. 
2007, sequenced and mapped to the genome.
</p>

<h3>RNA-seq and RNA-seq SMART (RNA sequencing and SMART-tagged RNA sequencing)</h3>
<p>
Sheared RNA was used to synthesize full-length single-stranded cDNAs as described by 
Morin, et al. 2008. A library was constructed and sequenced, and sequence reads are 
mapped to the genome. The 5&#39; end of transcripts were tagged with a sequence tag, 
called a &quot;SMART tag&quot;, while making cDNA library for sequencing. SMART tagged 
reads were used to infer transcription initiation, while all reads together are used to 
infer gene expression level.
</p>

<h2>Verification</h2>

<h3>MeDIP-seq</h3>
<p>
Each post-amplification library was QC&#39;d for quantity, quality and size 
distribution by spectrophotometry and Agilent DNA Bioanalyzer analysis. Four 
independent PCR reactions were performed to confirm enrichment for methylated and 
de-enrichment for unmethylated sequences, compared to input sonicated DNA. Visual 
inspection of extended coverage browser tracks confirmed expectations: lack of MeDIP 
signal in most 5&#39; CpG island promoters and in regions devoid of CpG sites, as well 
as high MeDIP signal at known methylated sites (i.e. some imprinted regions).
</p>

<h3>MRE-seq</h3>
<p>
Each post-amplification library was QC&#39;d for quantity, quality and size 
distribution by Nanodrop spectrophotometry and Agilent DNA Bioanalyzer analysis. Prior 
to high-throughput sequencing, a portion of each library was cloned into a sequencing 
vector and &#126;24 individual clones were Sanger sequenced to confirm the presence of 
MRE sites at the ends of each insert. Illumina sequencing reads were filtered to only 
include those that map to MRE sites in the reference. MRE reads occured frequently in 
5&#39; CpG islands, which are often unmethylated and are enriched for the MRE 
recognition sequences relative to rest of the genome.
</p>

<h3>H3K4me3 ChIP-seq</h3>
<p>
Each post-amplification library was examined for quantity, quality and size 
distribution by Nanodrop spectrophotometry, Qubit fluoremetry and Agilent DNA 
Bioanalyzer. Fold H3K4me3 enrichment was confirmed by comparison to non-specific 
rabbit IgG enrichment. Visual inspection of the browser track of called peaks confirmed 
enrichment at a subset of annotated promoters.
</p>

<h3>RNA-seq and RNA-seq (SMART)</h3>
<p>
Each post-amplification library was examined for quantity, quality and size 
distribution by Nanodrop spectrophotometry, Qubit fluoremetry and Agilent DNA 
Bioanalyzer. Visual inspection of the browser track of extended reads confirmed 
enrichment at annotated exons and UTRs. SMART-tagged reads were enriched at known 
promoters, as expected. 
</p>

<H2>Credits</H2>

<P>
UCSF: Joseph Costello, Raman Nagarajan, Shaun Fouse, Brett Johnson, Chibo Hong, Ksenya 
Shchors, Vivi M. Heine, David H. Rowitch
</P>
<P>
Genome Sciences Centre, BC Cancer Agency: Mikhail Bilenky, Cletus D'Souza, Cydney 
Nielsen, Yongjun Zhao, Allen Delaney, Richard Varhol, Nina Thiessen, Steven S.J. Jones, 
Marco A. Marra, Martin Hirst
</P>
<P>
Washington University, St. Louis, MO: Ting Wang, Xiaoyun Xing, Chris Fiore, Maximiliaan 
Schillebeeckx
</P>
<P>
UCSC: Tracy J. Ballinger, David Haussler
</P>
<P>
McGill: Gustavo Turecki
</P>

<H2>References</H2>
<p>
Maunakea AK, Nagarajan RP, Bilenky M, Ballinger TJ, D'Souza C, Fouse SD, Johnson BE, Hong C, Nielsen
C, Zhao Y <em>et al</em>.
<a href="http://www.nature.com/nature/journal/v466/n7303/full/nature09165.html"
target="_blank">
Conserved role of intragenic DNA methylation in regulating alternative promoters</a>.
<em>Nature</em>. 2010 Jul 8;466(7303):253-7.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/20613842" target="_blank">20613842</a>
</p>

<p>
Morin R, Bainbridge M, Fejes A, Hirst M, Krzywinski M, Pugh T, McDonald H, Varhol R, Jones S, Marra
M.
<a href="http://www.biotechniques.com/BiotechniquesJournal/2008/July/Profiling-the-HeLa-S3-transcriptome-using-randomly-primed-cDNA-and-massively-parallel-short-read-sequencing/biotechniques-45303.html"
target="_blank">
Profiling the HeLa S3 transcriptome using randomly primed cDNA and massively parallel short-read
sequencing</a>.
<em>Biotechniques</em>. 2008 Jul;45(1):81-94.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/18611170" target="_blank">18611170</a>
</p>

<p>
Robertson G, Hirst M, Bainbridge M, Bilenky M, Zhao Y, Zeng T, Euskirchen G, Bernier B, Varhol R,
Delaney A <em>et al</em>.
<a href="http://www.nature.com/nmeth/journal/v4/n8/abs/nmeth1068.html"
target="_blank">
Genome-wide profiles of STAT1 DNA association using chromatin immunoprecipitation and massively
parallel sequencing</a>.
<em>Nat Methods</em>. 2007 Aug;4(8):651-7.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/17558387" target="_blank">17558387</a>
</p>

