<H2>Description</H2>
<p>The tracks that are listed here contain genetic variants and links to scientific publications that 
mention them.</p>
<ul>
<li>The <b>Mastermind</b> track was created by Genomenon, a company that analyzes fulltext 
of publications with their own proprietary software with an unknown false
positive rate.</li>
<li>The <b><a target="_blank" href="https://varchat.engenome.com/">VarChat</a></b> 
track was created by enGenome and links to its proprietary 
software, VarChat, with an unknown false positive rate.</li>
<li>The <b>AVADA</b> track was created in the Bejerano lab at
Stanford by J. Birgmeier also on fulltext papers, using sophisticated machine learning
methods and was evaluated to have a false positive rate of around 50% in their study.</li>
<!--
<li>The <b>PubTator rsIDs</b> track was created using 
<a href="https://ftp.ncbi.nlm.nih.gov/pub/lu/PubTator3/">PubTator 3 data</a>.</li>
-->
<li><b>Varaico</b> was created using literature mining in a fashion similar to AVADA. Coloring
is a gradient between blue and red, and represent the number of publications per variant. See
the <a href="https://varaico.com/">Varaico website</a> for more details.</li>
</ul>

</p><p>
For additional information please click on the hyperlink of the respective track above.
<H2>Display conventions</H2>
</p><p>
By default, each variant is labeled with the nucleotide change. Hover over the
feature to see more information, explained on the track details page of the particular track
or when clicking onto the feature.  </p>
<H2>Credits</H2>
<p>
For data provenance, access and descriptions, please click the documentation via the link above.
</p>
