<P>
The January 2006 chimpanzee (<em>Pan troglodytes</em>) browser display
data from the October 2005 6X draft assembly &mdash; Build 
2 Version 1 &mdash; was produced by the Chimpanzee Genome Sequencing Consortium.

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the HUGO Gene Nomenclature Committee gene 
name of a human RefSeq, the accession number of a
mRNA or EST, a chromosomal coordinate range, 
or keywords from 
the GenBank description of an mRNA. The following list shows 
examples of valid position queries for the chimpanzee genome.
See the
<A HREF="../goldenPath/help/hgTracksHelp.html" TARGET=_blank>
User's Guide</A> for more information.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><br></TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chr22</TD>
        <TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays all of chromosome 22</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr6:118,000,000-119,000,000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a million bases of chromosome 6, beginning at 118 million bases offset from p arm telomere</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr6:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr 6 that spans 2000 bases, starting at position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>BRCA1</TD>
       	<TD WIDTH=14></TD>
       	<TD>Displays region of genome where human RefSeq gene BRCA1 aligns</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>AF115459</TD>
       	<TD WIDTH=14></TD>
   	<TD VALIGN=Top>Displays region of genome with mRNA with GenBank accession number AF115459</TD>
    </TR>
    <TR>
	<TD VALIGN=Top NOWRAP>348</TD>
	<TD><BR></TD>
	<TD VALIGN=Top NOWRAP>Displays the region of genome with LocusLink identifier 348</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>sialic acid</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs and RefSeqs with sialic acid in the GenBank keywords</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>huntington</TD>
       	<TD WIDTH=14></TD>
 	<TD VALIGN=Top>Lists mRNAs associated with Huntington's disease</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Paabo,S.</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author S. Paabo</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Paabo S</em>, internally it uses 
			the format <em>Paabo,S.</em>.</TD>
    </TR>
</TABLE>
<P>
<HR>
<H3>Assembly details</H3>
This assembly covers about 97 percent of the genome and is based on 6X
sequence coverage. It is composed of 265,882 contigs with an N50
length of 29 kb, and 44,460 supercontigs with an N50 length of 9.7
Mb. The total contig length, not including estimated gap sizes, is
2.97 Gb.  Of that total, 2.82 Gb has been ordered and oriented along
specific chimpanzee chromosomes, 107 Mb has been placed in chr*_random,
and 50 Mb remains in chrUn.
<P>
The whole genome shotgun data from primary donor-derived reads (Clint,
a captive-born male chimpanzee from the Yerkes Primate Research Center
(Atlanta, USA)) and were assembled using PCAP (Huang 2006) using
stringent parameters derived by eliminating detectable global
mis-assemblies (interchromosomal cross-overs determined by alignment
of the chimpanzee genome against the human genome) larger than 50kb.
<P>
The assembly data were aligned against the human genome at UCSC
(B. Raney) utilizing BLASTZ (Schwartz 2003) to align and score
non-repetitive chimpanzee regions against repeat-masked human
sequence. Alignment chains differentiated between orthologous and
paralogous alignments (Kent 2003) and only "reciprocal best"
alignments were retained in the alignment set. The chimpanzee AGP
files were generated from these alignments in a manner similar to that
already described (The Chimpanzee Genome Sequencing Consortium 2005).
Centromeres were introduced into the chimp sequence at the positions
of the centromeres in the human chromosomes.  Ten documented/known
human inversions (Yunis 1982) supported by the assembly were
introduced into the ordering as was the separation of alignments to
human chromosome 2 into chimpanzee chromosomes 2A and 2B.  We replaced
the regions in the WGS assembly corresponding to the finished
chromosome 21 sequence (see Credits), chr Y sequence (see Credits) and
a 5 Mb finished region from chimpanzee chromosome 7 (see Credits) with
the corresponding finished AGPs/sequences.
<P>


<P>
<A HREF="http://www.genome.gov/Pages/Research/Sequencing/SeqProposals/ChimpGenome2.pdf" TARGET=_blank>Background information</A> on the chimp genome 
sequencing project and the <A HREF="http://www.genome.gov/11509418" 
TARGET=_blank>initial news release</A> about the chimp assembly can be 
found on the NHGRI website.
<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/panTro2/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#chimp">Downloads</A> page. 
The complete set of sequence reads is available at the 
<A HREF="http://www.ncbi.nlm.nih.gov/Traces" TARGET=_blank>NCBI trace archive</A>.
The chimpanzee browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the <A HREF="../goldenPath/credits.html#chimp_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release and conditions for use of the data.
<P>
