<!-- Display image at top right of page -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=315>
  <TR><TD ALIGN=RIGHT><A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Pan%20troglodytes" TARGET=_blank>
      <IMG SRC="/images/Pan_troglodytes.jpg" WIDTH=300 HEIGHT=219 ALT="Chimpanzee"></A>
    </TD></TR>
  <TR><TD ALIGN=RIGHT>
  <FONT SIZE=-1> <em>Pan troglodytes</em><BR>
  </FONT>
  <FONT SIZE=-2>
    Photo credit: <A HREF="http://www.smm.org/" TARGET=_blank>Science Museum of 
	Minnesota</A>. All rights reserved.<BR>
    Courtesy of
    <A HREF = "http://www.nsf.gov/news/mmg/mmg_disp.cfm?med_id=57020&from=search_list"
    TARGET = _BLANK>National Science Foundation</A><BR>
</FONT>
  </TD></TR>
</TABLE>

<P>
The March 2006 chimpanzee (<em>Pan troglodytes</em>) browser displays
data from the 6X whole genome shotgun draft assembly (Build 
2 Version 1, Oct. 2005) produced by the 
<A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Pan%20troglodytes&SECTION=collaborators" 
TARGET=_BLANK>Chimpanzee Sequencing and Analysis Consortium</A>.
This assembly contains sequence from the initial 4X chimpanzee assembly
described and analyzed in <em>Nature</em> (The Chimpanzee Sequencing 
and Analysis Consortium, 2005),
with additional 2X sequence generated, assembled, and assigned to 
chromosomes by the 
<A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Pan%20troglodytes" 
TARGET=_BLANK>Genome Sequencing Center of Washington University School of 
Medicine</A>, St. Louis, MO, USA.</P>
<P>
This assembly uses a new chromosomal numbering scheme that reflects
orthology between the human and chimpanzee chromosomes. For details, see 
the Assembly details section below and the Genome Browser 
<A HREF="../FAQ/FAQdownloads#download25">FAQ</A>. To read more about the
chimpanzee assembly, see the Washington University in St. Louis School of 
Medicine 
<A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Pan%20troglodytes" 
TARGET=_blank><em>Pan troglodytes</em></A> web page and the National Institutes 
of Health NIH News <A HREF="http://www.genome.gov/15515096" 
TARGET=_blank>summary of the chimpanzee analysis paper</A>.</P>
<P>
The chimpanzee is the species most closely related to humans, but is
<A HREF="http://ecos.fws.gov/species_profile/servlet/ 
gov.doi.species_profile.serv
lets.SpeciesProfile?spcode=A06C" TARGET=_blank> endangered</A>.
Consequently, it is the focus of multiple 
<A HREF="http://www.janegoodall.org/chimp_central/conservation/default.asp"  
TARGET=_blank>conservation efforts</A>.</P>

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the HUGO Gene Nomenclature Committee gene 
name of a human RefSeq, the accession number of an EST or mRNA,
a chromosomal coordinate range, or keywords from 
the GenBank description of an mRNA. The following list shows 
examples of valid position queries for the chimpanzee genome.
See the
<A HREF="../goldenPath/help/hgTracksHelp.html" TARGET=_blank>
User's Guide</A> for more information.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><br></TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chr22</TD>
        <TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays all of chromosome 22</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr2a:11,250,001-12,250,000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a million bases of chromosome 2a, beginning at 
base 11,250,001. Note that <BR>
	chromosome 2 in this assembly has been split into two parts: 2a
and 2b.</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr2a:11,250,001+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr 2a that spans 2000 bases, starting at position 11,250,001</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>BRCA1</TD>
       	<TD WIDTH=14></TD>
       	<TD>Displays a list of genomic regions where human RefSeq gene BRCA1 (or features associated with BRCA1) aligns</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>AF115459</TD>
       	<TD WIDTH=14></TD>
   	<TD VALIGN=Top>Displays region of genome with mRNA with GenBank accession number AF115459</TD>
    </TR>
    <TR>
	<TD VALIGN=Top NOWRAP>348</TD>
	<TD><BR></TD>
	<TD VALIGN=Top NOWRAP>Displays the region of genome with Entrez Gene identifier 348</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>sialic acid</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs and RefSeqs with GenBank keywords sialic acid </TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>huntington</TD>
       	<TD WIDTH=14></TD>
 	<TD VALIGN=Top>Lists mRNAs associated with Huntington's disease</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Paabo,S.</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author S. Paabo</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Paabo S</em>, internally it uses 
			the format <em>Paabo,S.</em>.</TD>
    </TR>
</TABLE>
<P>
<HR>
<H3>Assembly details</H3>
<P>
This assembly covers about 97 percent of the genome and is based on 6X
sequence coverage. It is composed of 265,882 contigs with an N50
length of 29 kb and 44,460 supercontigs with an N50 length of 9.7
Mb. The total contig length, not including estimated gap sizes, is
2.97 Gb.  Of that total, 2.82 Gb of sequence have been ordered and oriented 
along specific chimpanzee chromosomes, 107 Mb have been placed in chr*_random,
and 50 Mb remain in chrUn.</P>
<P>
The whole genome shotgun data were derived primarily from the donor 
Clint, a captive-born male chimpanzee from the Yerkes Primate Research Center
in Atlanta, GA, USA. The reads were assembled with the whole-genome assembly
program PCAP (Huang, 2006), using
stringent parameters derived by eliminating detectable global
misassemblies -- interchromosomal cross-overs determined by alignment
of the chimpanzee genome against the human genome -- larger than 50 Kb.</P>
<P>
The assembly data were aligned against the human genome at UCSC
utilizing BLASTZ (Schwartz, 2003) to align and score
non-repetitive chimpanzee regions against repeat-masked human
sequence. The alignment chains differentiated between orthologous and
paralogous alignments (Kent, 2003); only &quot;reciprocal best&quot;"
alignments were retained in the alignment set. The chimpanzee AGP
files were generated from these alignments in a manner similar to that
described in The Chimpanzee Sequencing and Analysis Consortium (2005).
Centromeres were introduced into the chimp sequence at the positions
of the centromeres in the human chromosomes.  Ten documented/known
human inversions supported by the assembly were
introduced into the ordering, as was the separation of alignments to
human chromosome 2 into chimpanzee chromosomes 2a and 2b.  
The regions in the WGS assembly corresponding to the finished sequences for
chromosomes 21 and Y and
a 5-Mb finished region from chimpanzee chromosome 7 were replaced
with the corresponding finished AGPs/sequences. See the 
<A HREF="../goldenPath/credits.html#chimp_credits">Credits</A> page for 
acknowledgements for these chromosomal regions.</P>
<P>
A major difference between this assembly and the previous Nov. 2003 version is 
the chromosomal numbering scheme, which has been changed to reflect a
new standard that preserves orthology with human chromosomes.
Proposed by E.H. McConkey in 2004, 
the new numbering convention was subsequently endorsed by the 
International Chimpanzee Sequencing and Analysis Consortium.
This standard assigns the identifiers &quot;2a&quot; and &quot;2b&quot; to the 
two chimp chromosomes that fused in the human genome to form chromosome 2. Note 
that
the genome assembly shown in the Nov. 2003 panTro1 Genome Browser retains the
older numbering scheme. To view a table showing the correspondence between
human and chimp chromosomes, see the 
<A HREF="../FAQ/FAQdownloads#download25">FAQ</A>.</P> 
<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/panTro2/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#chimp">Downloads</A> page. 
The complete set of sequence reads is available at the 
<A HREF="http://www.ncbi.nlm.nih.gov/Traces" TARGET=_blank>NCBI trace archive</A>. These data have <A HREF="../goldenPath/credits.html#chimp_use">specific 
conditions for use</A>.</P>
<P>
The chimpanzee browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="../goldenPath/credits.html#chimp_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to this release. </P> 
<P>
<HR>
<H3>References</H3>
<P>
The Chimpanzee Sequencing and Analysis Consortium. 
<A HREF="http://www.nature.com/nature/journal/v437/n7055/full/nature04072.html" TARGET=_blank>Initial sequence of the chimpanzee genome and comparison with the 
human genome</A>. <em>Nature</em> <B>437</B>(7005), 69-87 (2005). </P>
<P>
Huang, X., Yang, S., Chinwalla, A.T., Hillier, L.W., Minx, P., Mardis, E.R. 
and Wilson, R.K.
<A HREF="http://nar.oxfordjournals.org/cgi/content/full/34/1/201"
TARGET=_blank>Application of a superword array in genome assembly</A>.
<em>Nucleic Acids Res.</em> <B>34</B>(1), 201-5 (2006).</P>
<P>
Kent, W.J., Baertsch, R., Hinrichs, A., Miller, W. and Haussler, D. 
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1"
TARGET=_blank>Evolution's cauldron: duplication, deletion, and rearrangement in 
the mouse and human genomes</A>. 
<em>P. Natl. Acad. Sci. USA</em> <B>100</B>(20), 11484-11489 (2003).</P>
<P>
McConkey, E.H.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=pubmed&cmd=Retrieve&dopt=Abstract&list_uids=15218271&query_hl=1&itool=pubmed_docsum"
TARGET=_blank>Orthologous numbering of great ape and human chromosomes is 
essential for comparative genomics</A>.
Cytogenet Genome Res. <B>105</B>(1), 157-8 (2004).
<P>
Schwartz, S., Kent, W.J., Smit, A., Zhang, Z., Baertsch, R., Hardison, R., 
Haussler, D., and Miller, W. 
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103"
TARGET=_blank>Human-mouse alignments with BLASTZ</A>. 
<em>Genome Res.</em> <B>13</B>(1), 103-107 (2003). </P>
