<P>
The November 2002 human genome assembly is based on NCBI Build 31.  

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic clone, an mRNA or EST or STS marker, or
a cytological band, a chromosomal coordinate range, or keywords from 
the GenBank description of an mRNA. The following list shows 
examples of valid position queries for the human genome.
See the
<A HREF="../goldenPath/help/hgTracksHelp.html" TARGET=_blank>
User's Guide</A> for more information.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><br></TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chr7</TD>
        <TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays all of chromosome 7</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>20p13</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region for band p13 on chr 20</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr3:1-1000000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays first million bases of chr 3, counting from p arm telomere</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>scf1:1-1000000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays first million bases of scaffold 1 of an unmapped genome assembly</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>D16S3046</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region around STS marker D16S3046 from the Genethon/Marshfield maps.
                      Includes 100,000 bases on each side as well.</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>RH18061;RH80175</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region between STS markers RH18061;RH80175.
                      Includes 100,000 bases on each side as well.</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>AA205474</TD>
       	<TD WIDTH=14></TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession AA205474 in BRCA1 cancer gene on chr 17</TD>
    </TR>
<!-- 
    <TR><TD VALIGN=Top NOWRAP>ctgchr7_ctg</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of the clone contig ctgchr7_ctg 
                       (set "Map contigs" track to "dense" and refresh to see contigs)</TD>
    </TR> 
-->
    <TR><TD VALIGN=Top NOWRAP>AC008101</TD>
    	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of clone with GenBank accession AC008101</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>AF083811</TD>
       	<TD WIDTH=14></TD>
   	<TD VALIGN=Top>Displays region of mRNA with GenBank accession number AF083811</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>PRNP</TD>
       	<TD WIDTH=14></TD>
       	<TD>Displays region of genome with HUGO Gene Nomenclature Committee identifier PRNP</TD>
    </TR>
    <TR>
	<TD VALIGN=Top NOWRAP>NM_017414</TD>
	<TD><BR></TD>
	<TD VALIGN=Top>Displays the region of genome with RefSeq identifier NM_017414</TD>
    </TR>
    <TR>
	<TD VALIGN=Top NOWRAP>NP_059110</TD> 
	<TD><BR></TD>
 	<TD VALIGN=Top NOWRAP> Displays the region of genome with protein accession number NP_059110</TD>
    </TR>
    <TR>
	<TD VALIGN=Top NOWRAP>11274</TD>
	<TD><BR></TD>
	<TD VALIGN=Top NOWRAP>Displays the region of genome with Entrez Gene identifier 11274</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>homeobox caudal</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>kruppel zinc finger</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>huntington</TD>
       	<TD WIDTH=14></TD>
 	<TD VALIGN=Top>Lists candidate genes associated with Huntington's disease</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>zahler</TD>
       	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zahler</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>Evans,J.E.</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author J.E.  Evans</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Evans JE</em>, internally it uses 
			the format <em>Evans,J.E.</em>.</TD>
    </TR>
</TABLE>
<BR>
<HR>
<H3>Assembly details</H3>
Build 31 is an excellent high-quality assembly that 
shows a remarkable amount of progress toward the milestone of 
finishing the human genome. 
More than 95% of the euchromatic 
region of the genome is complete, with greater than 90% of 
the sequence in a finished state. The number of clone contig gaps 
has decreased by one third from the previous assembly, and the 
overall number of sequenced contigs has been reduced by one half. 
Seven chromosomes are considered to be in a finished state: 
6, 7, 13, 20, 21, 22, and Y. 
<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/14nov2002/">FTP server</A>
or <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#human">Downloads</A> page. 
The annotation tracks displayed in the Genome Browser were generated by UCSC and 
collaborators worldwide. See the <A HREF="../goldenPath/credits.html#human_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
<P>
<HR>
<H3>Statistical information</H3>
<UL>
<LI><A HREF="../goldenPath/stats.html">Summary Statistics</A>
<LI><A HREF="http://genome-archive.cse.ucsc.edu/goldenPath/chromReports">Chromosome Reports</A>
<LI><A HREF="http://genome-archive.cse.ucsc.edu/goldenPath/mapPlots/">Genome Map Plots</A>
</UL>
