<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=310>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Caenorhabditis+remanei"
    TARGET=_blank>
    <IMG SRC="/images/Caenorhabditis_remanei.jpg" WIDTH=295 HEIGHT=400 ALT="C. remanei"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Caenorhabditis remanei</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of
      <A HREF="http://www.life.umd.edu/biology/faculty/haag/Haag_Lab_Members_Page.html" 
      TARGET=_blank>Eric Haag,</A> All Rights Reserved.
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Jul. 2007 <em>Caenorhabditis remanei</em> draft assembly was
produced by the
<A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Caenorhabditis+remanei"
TARGET=_blank>Genome Sequencing Center</A> at Washington University in St.
Louis (WUSTL) School of Medicine. 

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
<em>C. remanei</em> genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>chrUn:1-1000000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of chrUn</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chrUn:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chrUn that spans 2000 bases, starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>BX950328</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number BX950328</TD>
    </TR>
    <TR><TD VALIGN=Top>DR777235</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		DR777235</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>succinate-CoA</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists non-C. remanei RefSeq Genes</TD>
    </TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>zhang</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zhang</TD>
    </TR>
    <TR><TD VALIGN=Top>Taguchi,T.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author T. Taguchi</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Taguchi T</em>, internally it
	uses the format <em>Taguchi,T.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
Quote from <A HREF="http://genome.wustl.edu/pub/organism/Invertebrates/Caenorhabditis_remanei/assembly/Caenorhabditis_remanei-15.0.1/ASSEMBLY"
TARGET=_blank>genome.wustl.edu</A>
"The Caenorhabditis remanei genome was sequenced to a depth of 9.2X using
using plasmid and fosmid libraries made from DNA extracted from a
mixed population of females and males from all developmental stages.
Coverage was based on an expected genome size of 130 Mb. The whole-genome
shotgun plasmid and fosmid paired end reads were assembled using PCAP
(Genome Res. 13(9):2164-70 2003).  The sequence cat be downloaded
from the <A HREF="ftp://genome.wustl.edu/pub/organism/Invertebrates/Caenorhabditis_remanei/assembly/Caenorhabditis_remanei-15.0.1/output"
TARGET=_blank>GSC ftp site.</A>"
</P><P>
The UCSC Genome browser displays the 3,670 scaffolds as a single
artifical chromosome chrUn. There is 138,406,203 bases of sequence
and 10,705,348 N's in gaps.
</P><P>
For more information on this assembly, see the WUSTL 
<A HREF="http://genome.wustl.edu/pub/organism/Invertebrates/Caenorhabditis_remanei/assembly/Caenorhabditis_remanei-15.0.1/ASSEMBLY"
TARGET=_blank>assembly notes</A>.</P>
<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/caeRem3/">FTP
server</A> or the 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#caeRem">Downloads</A> 
page. These data have <A HREF="http://genome.wustl.edu/data.cgi" TARGET=_blank>
specific conditions for use</A>. </P>
<P>
The <em>C. remanei</em> browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="../goldenPath/credits.html#worm_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to this release. 
</P>
