<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=282 HEIGHT=215>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://www.broadinstitute.org/science/projects/mammals-models/elephant/elephant"
    TARGET=_blank>
    <IMG SRC="/images/Loxodonta_africana.jpg" WIDTH=267 HEIGHT=200 ALT="African Savannah Elephant"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Loxodonta africana</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of
      (<A HREF="http://commons.wikimedia.org/wiki/File:African_Bush_Elephants.jpg"
      TARGET=_blank>Wikimedia Commons</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Jul. 2009 <em>Loxodonta africana</em> draft assembly 
(Broad Institute loxAfr3 (NCBI project 12569, AAGU03000000)) was produced
by the Broad Institute sequencing center. For more information about this assembly, see
<A HREF="http://www.ncbi.nlm.nih.gov/assembly/299248/" TARGET="_blank">
Loxafr3.0</A> in the NCBI Assembly database.

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
Elephant genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>scaffold_17</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of scaffold 17</TD>
    </TR>
    <TR><TD VALIGN=Top>scaffold_17:1-1000000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of scaffold 17</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>scaffold_17:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of scaffold 17 that spans 2000 bases, starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>BX950328</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number BX950328</TD>
    </TR>
    <TR><TD VALIGN=Top>AK131176</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of Other mRNA with GenBank accession
		AK131176</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>janke</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Janke</TD>
    </TR>
    <TR><TD VALIGN=Top>Taguchi,T.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author T. Taguchi</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Taguchi T</em>, internally it
	uses the format <em>Taguchi,T.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
The genome has been sequenced to 7X coverage.  There are 2,352 scaffolds
with a total size of 3,196,743,967 bases, and the mitochondrial genome
chrM (genbank accession NC_000934, size 16,866 bases).<BR>
<BR>
The N50 size of the scaffolds is 46,401,353 bases with 21 scaffolds larger,
and 2,331 scaffolds smaller.  The mean scaffold size is 1,359,159 bases,
the median scaffold size is 10,119.<BR>
<BR>
See also: <A HREF="http://www.broadinstitute.org/ftp/pub/assemblies/mammals/elephant/loxAfr3/"
TARGET=_blank>Broad Institute Elephant sequence statistics</A>
</P>

<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/loxAfr3/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#elephant">Downloads</A> page. 

These data have
<A HREF="../goldenPath/credits.html#elephant_use">specific 
conditions for use</A>. 

The Elephant browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#elephant_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
<HR>
<H3>Genbank Pipeline Details</H3>
For the purposes of the Genbank alignment pipeline, this assembly is considered to be: <B>well-ordered</B>.
