<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="CONTENT-TYPE" CONTENT="text/html; charset=utf-8">
	<TITLE></TITLE>
	<META NAME="GENERATOR" CONTENT="OpenOffice.org 2.3  (Linux)">
	<META NAME="CREATED" CONTENT="0;0">
	<META NAME="CHANGED" CONTENT="20080505;18391700">
</HEAD>
<BODY LANG="en-US" DIR="LTR">
<H2>Description</H2>
<P>The transcriptome track shows gene predictions based on data from
RefSeq and EMBL/GenBank. This is a moderately conservative set of
predictions, requiring the support of either one GenBank full length
RNA sequence, one RefSeq RNA, or one spliced EST. The track includes
both protein-coding and non-coding transcripts. The CDS are predicted
using <A HREF="http://estscan.sourceforge.net/" target="_blank">ESTScan</A>.</P>
<H2>Display Conventions and Configuration</H2>
<P>This track in general follows the display conventions for <A HREF="../goldenPath/help/hgTracksHelp.html#GeneDisplay">gene
prediction tracks</A>. The exons for putative noncoding genes and
untranslated regions are represented by relatively thin blocks, while
those for coding open reading frames are thicker.</P>
<P>This track contains an optional codon coloring feature that allows
users to quickly validate and compare gene predictions. To display
codon colors, select the <EM>genomic codons</EM> option from the
<EM>Color track by codons</EM> pull-down menu. Click <A HREF="../goldenPath/help/hgCodonColoring.html">here</A>
for more information about this feature. 
</P>
<P>Further information on the predicted transcripts can be found on
the <A HREF="http://ccg.vital-it.ch/tromer/" target="_blank">Transcriptome Web
interface</A>.</P>
<H2>Methods</H2>
<P>The transcriptome is built using a multi-step pipeline: 
</P>
<OL>
	<LI><P STYLE="margin-bottom: 0in">RefSeq and GenBank RNAs and ESTs
	are aligned to the genome with <A HREF="http://sibsim4.sourceforge.net/" target="_blank">SIBsim4</A>,
	keeping only the best alignments for each RNA. 
	</P>
	<LI><P STYLE="margin-bottom: 0in">Alignments are broken up at
	non-intronic gaps, with small isolated fragments thrown out. 
	</P>
	<LI><P STYLE="margin-bottom: 0in">A splicing graph is created for
	each set of overlapping alignments. This graph has an edge for each
	exon or intron, and a vertex for each splice site, start, and end.
	Each RNA that contributes to an edge is kept as evidence for that
	edge. 
	</P>
	<LI><P STYLE="margin-bottom: 0in">The graph is traversed to generate
	all unique transcripts. The traversal is guided by the initial RNAs
	to avoid a combinatorial explosion in alternative splicing. 
	</P>
	<LI><P>Protein predictions are generated. 
	</P>
</OL>
<H2>Credits</H2>
<P>The transcriptome track was produced on the <A HREF="http://www.vital-it.ch/" target="_blank">Vital-IT</A>
high-performance computing platform using a computational pipeline
developed by Christian Iseli with help from colleagues at the <A HREF="http://www.ludwigcancerresearch.org/" target="_blank">Ludwig
institute for Cancer Research</A> and the <A HREF="http://www.isb-sib.ch/" target="_blank">Swiss
Institute of Bioinformatics</A>. It is based on data from NCBI <A HREF="http://www.ncbi.nlm.nih.gov/refseq/" TARGET="_blank">RefSeq</A>
and <A HREF="http://www.ncbi.nlm.nih.gov/genbank/" TARGET="_blank">GenBank</A>/<A HREF="http://www.embl.org/" target="_blank">EMBL</A>.
Our thanks to the people running these databases and to the
scientists worldwide who have made contributions to them. 
</P>
<H2>References</H2>
<p>
Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Wheeler DL.
<a href="http://nar.oxfordjournals.org/content/32/suppl_1/D23.abstract" target="_blank">
GenBank: update</a>.
<em>Nucleic Acids Res</em>. 2004 Jan 1;32(Database issue):D23-6.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/14681350" target="_blank">14681350</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC308779/" target="_blank">PMC308779</a>
</p>
</BODY>
</HTML>
