<H2>Description</H2>

<p>
These tracks were generated by the ENCODE Consortium. They contain information 
about mouse RNAs greater than 200 nucleotides in length obtained as short 
reads off the Illumina platform. Data are available from biological replicates. 
</p>

<h2>Display Conventions and Configuration</h2>
<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are
<a href="/goldenPath/help/multiView.html" target="_blank">here</a>.</P>

<p>To show only selected subtracks, uncheck the boxes next to the tracks that
you wish to hide.</p>

<p>Color differences among the views are arbitrary. They provide a
visual cue for
distinguishing between the different cell types and compartments.</p>

<dl>
<dt><i>Contigs</i>
</dt><dd>The Contigs represent blocks of overlapping mapped reads from the pooled biological replicates.
</dd><dt><i>Raw Signals</i>
</dt><dd>The Plus Raw Signal and Minus Raw Signal views show the density of mapped reads on the plus and minus strands (wiggle format), respectively.
</dd>
<dt><i>Alignments</i>
</dt><dd>The Alignments view shows individual reads mapped from biological replicates to the genome and indicates where
bases may mismatch. Every mapped read is displayed, i.e. uncollapsed.  The alignment file follows the standard SAM format of Bowtie output. See the <a href="http://bowtie-bio.sourceforge.net/manual.shtml#sam-bowtie-output" target=_blank>Bowtie Manual</a> for more information about the SAM Bowtie output (including other tags) and the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf#page=4" target=_blank>SAM Format Specification</a> for more information on the SAM/BAM file format. 
</dd>
<dt><i>Splice Junctions</i>
</dt><dd>Subset of aligned reads that cross splice junctions. Specific column specifications can be found in the <A TARGET=_BLANK HREF="http://hgdownload.cse.ucsc.edu/goldenPath/mm9/encodeDCC/wgEncodeCshlLongRnaSeq/supplemental/">supplemental directory</A>.
</dd>
</dl>
<p>Metadata for a particular subtrack can be found by clicking the down arrow in the list of subtracks.</p>

<P>
<!--echo var="BROWSER_HOST" -->
Additional <A TARGET=_BLANK HREF="../cgi-bin/hgEncodeVocab?tag=FASTQRD1,FASTQRD2,SPKNS,TRANSDENOVO">views</A> are available on the Downloads page. 
</P>

<H2>Methods</H2>

<H3>Tissue Samples</H3>
<P> Individual tissues were harvested from mouse strain C57BL/6J at different timepoints
according to <a href="/ENCODE/protocols/cell/mouse" target="_blank">ENCODE
cell culture protocols</A>. 
Whenever possible, biological replicates were obtained from littermates.
</P>

<H3>Library Preparation</H3>
<P>The published 
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/pubmed/19620212">
cDNA sequencing protocol</A> was used.  This protocol generates directional libraries
and reports the transcripts' strand of origin.  Exogenous RNA spike-ins
were added to each endogenous RNA isolate and carried
through library construction and sequencing.  
The spike-in sequence and the concentrations are available for download
in the <A TARGET=_BLANK HREF="http://hgdownload.cse.ucsc.edu/goldenPath/mm9/encodeDCC/wgEncodeCshlLongRnaSeq/supplemental/">supplemental directory</A>.
</P>

<H3>Sequencing and Mapping</H3>
<P>The libraries were sequenced on the Illumina platform (either GAIIx or
Hi-Seq) in mate-pair fashion (either pair-end 76 or pair-end 101) to an average depth of 100 million
mate-pairs. 
The data were mapped against mm9 using Spliced Transcript Alignment
and Reconstruction (STAR) written by Alex Dobin (CSHL).  More
information about STAR, including the parameters used for these data,
is available from the
<A TARGET=_BLANK
HREF="http://gingeraslab.cshl.edu/STAR/">Gingeras lab</A>.
</P>

<P>
For each experiment, there are additional
<a target="_BLANK" href="../cgi-bin/hgEncodeVocab?tag=EXONSDENOVO,EXONENSV65,GENEDENO,GENENSV65,TRANSDENOVO,TRANSENSV65">element data views</a>
data files available for <a target="_BLANK" href="../cgi-bin/hgFileUi?db=mm9&g=wgEncodeCshlLongRnaSeq">download</a>.
These elements were assessed for reproducibility using a nonparametric
irreproducible detection (IDR) rate script.  The IDR values for each element
are included in the files for end-users to use as a threshold. An IDR value of 0.1 means
that the probability of detecting that element in a third experiment equivalent
in depth to the sum of the bioreplicates is 90&#37;.  In addition,
expression values for annotated genes, transcripts and exons were computed. Further explanation of these
files is available for download in the
<A TARGET=_BLANK HREF="http://hgdownload.cse.ucsc.edu/goldenPath/mm9/encodeDCC/wgEncodeCshlLongRnaSeq/supplemental/">supplemental directory</A>.
</P>


<H2>Verification</H2>
<P>
FPKM (fragments per kilobase of exon per million fragments mapped) values were calculated 
for annotated exons and Spearman correlation coefficients were computed.   
In general, Rho values are greater than 0.90 between biological replicates.   
</P>

<H2>Release Notes</H2>

<P>
This is release 3 (Sept 2012) of this track. It adds data for bladder, cerebellum, CNS, cortex, frontal lobe, limb, liver, placenta, and whole brain. The samples for CNS, liver, limb and whole brain vary over age (developmental stage). 
This release also contains replacement BAM files for the previous ones had the second read reverse complemented.
</P>

<H2>Credits</H2>
<P>These data were generated and analyzed by the transcriptome group at
Cold Spring Harbor Laboratories and the Center for Genomic
Regulation (CRG in Barcelona), who are participants in the ENCODE Transcriptome Group.
</P>
<b>Contacts:</b>
<A HREF="mailto:&#100;a&#118;&#105;s&#99;&#64;&#99;sh&#108;.&#101;&#100;&#117;">Carrie Davis</A> (experimental),
<!-- above address is davisc at cshl.edu -->
<A HREF="mailto:&#114;&#103;&#117;i&#103;o&#64;&#105;&#109;&#105;m.e&#115;">Roderic Guigo</A> and lab (data processing),
<!-- above address is rguigo at imim.es -->
<A HREF="mailto:&#103;&#105;&#110;&#103;e&#114;a&#115;&#64;&#99;&#115;&#104;&#108;.&#101;d&#117;">Tom Gingeras</A> (primary investigator)
<!--above address is gingeras at cshl.edu -->

</P>

<H2>References</H2>
<p>
Jiang L, Schlesinger F, Davis CA, Zhang Y, Li R, Salit M, Gingeras TR, Oliver B.
<a href="http://genome.cshlp.org/content/21/9/1543.long" target="_blank">
Synthetic spike-in standards for RNA-seq experiments</a>.
<em>Genome Res</em>. 2011 Sep;21(9):1543-51.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/21816910" target="_blank">21816910</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3166838/" target="_blank">PMC3166838</a>
</p>

<p>
Parkhomchuk D, Borodina T, Amstislavskiy V, Banaru M, Hallen L, Krobitsch S, Lehrach H, Soldatov A.
<a href="http://nar.oxfordjournals.org/content/37/18/e123.long" target="_blank">
Transcriptome analysis by strand-specific sequencing of complementary DNA</a>.
<em>Nucleic Acids Res</em>. 2009 Oct;37(18):e123.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/19620212" target="_blank">19620212</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2764448/" target="_blank">PMC2764448</a>
</p>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column, above.  The full data release policy
for ENCODE is available
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>
