<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- saved from url=(0074)http://genome-test.cse.ucsc.edu/cgi-bin/hgTrackUi?db=mm9&g=wgEncodePsuTfbs -->
<html><head><meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1">

<h2> Description </h2>

<h2>Rationale for the Mouse ENCODE project</h2>
<p>Our knowledge of the function of genomic DNA sequences comes from three 
basic approaches. Genetics uses changes in behavior or structure of a cell or organism
in response to changes in DNA sequence to infer function of the altered sequence. 
Biochemical approaches monitor states of histone modification, binding of specific 
transcription factors, accessibility to DNases and other epigenetic features along 
genomic DNA. In general, these are associated with gene activity, but the precise 
relationships remain to be established. The third approach is evolutionary, using 
comparisons among homologous DNA sequences to find segments that are evolving 
more slowly or more rapidly than expected given the local rate of neutral change. These 
are inferred to be under negative or positive selection, respectively, and we interpret 
these as DNA sequences needed for a preserved (negative selection) or adaptive 
(positive selection) function. </p>

<p>The ENCODE project aims to discover all the DNA sequences associated with 
various epigenetic features, with the reasonable expectation that these will also be 
functional (best tested by genetic methods). However, it is not clear how to relate these 
results with those from evolutionary analyses. The mouse ENCODE project aims to 
make this connection explicitly and with a moderate breadth. Assays identical to those 
being used in the ENCODE project are performed in cell types in mouse that are similar 
or homologous to those studied in the human project. Thus we will be able to discover 
which epigenetic features are conserved between mouse and human, and we can 
examine the extent to which these overlap with the DNA sequences under negative 
selection. The contribution of DNA that with a function preserved in mammals versus 
that with a function in only one species will be discovered. The results will have a 
significant impact on our understanding of the evolution of gene regulation.
</p>

<h2>Maps of Occupancy by Transcription Factors</h2>
Genome-wide occupancy maps of transcription factors (TFs) are generated by 
ChIP-seq. A ChIP-Seq experiment combines a chromatin immunoprecipitation (ChIP) experiment that 
enriches genomic DNA for the segments bound by specific proteins (the 
antigens recognized by the antibody) with high-throughput short read 
sequencing of the enriched DNA fragments (Wold & Myers, 2008). Proteins are crosslinked to DNA (usually with formaldehyde), 
chromatin is sheared and immunoprecipitated with the antibody of interest. The immunoprecipitated material is turned into a sequencing library and sequenced. 
The sequencing reads are then aligned to the genome. An control sample consiting of sonicated chromatin that has not been immunoprecipitated or 
immunoprecipitated with a non-specific immunoglobulin is also sequenced. The ChIP and the control datasets are analyzed with a variety of software packages 
to identify regions occupied by the target protein. The sequencing data, alignments and analysis files for these experiments are available for download. <p></p>

<h2> Display Conventions and Configuration </h2>
<p>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are <A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
This track contains the following views:
</p>
<dl>
<dt><i>Peaks</i></dt><dd> Regions of signal enrichment based on processed data. Intensity is represented in 
grayscale, the darker shading shows higher intensity (a solid vertical line
in the peak region represents the the point with the highest signal).</dd>

<dt><i>Signal</i></dt><dd> Density graph (wiggle) of signal enrichment based on
processed data of all mapped read Intensity of the signal is represented as RPM (Read Per Million)</dd>

</dl>

<h2> Methods </h2>
<p>
Cells were grown according to the approved
<a href="http://genome-test.cse.ucsc.edu/ENCODE/protocols/cell/mouse" target="_BLANK">ENCODE cell culture protocols</a>.
</p>
<p>
Chromatin immunoprecipitation followed published methods (Johnson & Mortazavi <em> et al.</em>, 2007) with the exception of certain experiments for which glutaraldehyde was added to the crosslink reaction.
Information on antibodies used is available via the hyperlinks in the "Select
subtracks" menu. Libraries were contructed using the Illumina ChIP-seq 
Sample Preparation Kit or using a modified protocol that includes the addition of multiplexing tags to the fragments. 
DNA fragments were repaired to generate blunt ends and a single A nucleotide 
was added to each end. Double-stranded Illumina adaptors or Double-stranded Illumina adaptors with multiplexing tags were ligated to the 
fragments. Ligation products were amplified by 18 cycles of PCR, and the DNA between 
150-250 bp was gel purified. Completed libraries were quantified with Quant-iT dsDNA 
HS Assay Kit. The DNA library was sequenced on the Illumina GAII and GAIIx
sequencing systems, and more recently, for multiplexed libraries, several of them were pooled and sequenced on the HiSeq platform. Cluster generation, linearization, 
blocking and sequencing primer reagents were provided in the Illumina Cluster 
Amplification kits. Older libraries were generated using 2 rounds of PCR. Matched input samples were sequenced for each variation of 
fixation conditions and the number of PCR rounds. Reads of 32bp, 36bp or 50bp length were generated
</p>

<p>
Sequencing reads (fastq files) were assigned to the corresponding libraries based on the multiplexing tag for pooled libraries 
(all tags have been removed from reads in the fastq files avaialble for download) or directly processed. Bowtie (Langmead <em>et al.</em>, 2009) was used to map reads 
to the male of female version of the mouse genome (excluding the _random chromomosomes in the assembly) depending on the cell line sex. The following paramteres were used:
"-v 2 -k 11 -m 10 -t --best --strata". Aligned reads were converted into rds files using the ERANGE package (Johnson & Mortazavi <em>et al.</em>, 2007) and the findall.py program in 
ERANGE was used to identify enriched regions against the matching input sample. The following settings were used for point-source transcription factors: 
 "--shift learn --ratio 3 --minimum 2 --listPeak --revbackground". For histone modifications, the settings were changed to
"--notrim --nodirectionality --spacing 100 --ratio 3 --minimum 2 --listPeak --revbackground".
</p>

<h2> Credits </h2>

<p>
Cell growth, ChIP, and Illumina library construction were done in the laboratory 
of Barbara Wold, (California Institute of Techniology). Sequencing was done at the Millard and Muriel Jacobs Genetics and Genomics Laboratory at the California Institute of Technology, initial HiSeq data was generated at Illumina Inc., Hawyard, CA.
<br><br>
<b>Cell growth and ChIP:</b>

Georgi Marinov, Katherine Fisher, Gordon Kwan, Antony Kirilusha, Ali Mortazavi, Gilberto DeSalvo, Brian Williams<br>

<b>Library Construction, Sequencing and Primary Data Handling:</b>

Lorianne Schaeffer, Diane Trout , Igor Antoschechkin (California Institute of Technology), 
Lu Zhang, Gary Schroth (Illumina Inc.)<br>

<b>Data Processing and Submission:</b>

Georgi Marinov, Diane Trout<br>
<br>
<b>Contact:</b>
<a href="mailto:&#119;o&#108;&#100;&#98;&#64;&#99;&#97;&#108;&#116;e&#99;&#104;.
&#101;d&#117;"target="_BLANK">Barbara Wold</a>, 
<a href="mailto:&#103;&#101;&#111;r&#103;i&#64;&#99;&#97;lt&#101;c&#104;.
&#101;&#100;&#117;" target="_BLANK">Georgi K. Marinov</a>, 
<a href="mailto:&#100;&#105;a&#110;&#101;&#64;&#99;&#97;&#108;&#116;&#101;&#99;h.
e&#100;u" target="_BLANK">Diane Trout</a><br>

</p>
<h2> References </h2>

<p>
Langmead B, Trapnell C, Pop M, Salzberg SL.
 <a href="http://genomebiology.com/2009/10/3/R25" class="external text" title="http://genomebiology.com/2009/10/3/R25" rel="noFollow" target="_BLANK">Ultrafast and memory-efficient alignment of 
short DNA sequences to the human genome</a>.
<em>Genome Biol</em> 2009 March; 10(3): R25.
</p>

Johnson DS, Mortazavi A, Myers RM, Wold B.
 <a href="http://www.sciencemag.org/content/316/5830/1497.short" class="external text" title="http://www.sciencemag.org/content/316/5830/1497.short" rel="noFollow" target="_BLANK">Genome-wide mapping of in vivo protein-DNA interactions</a>.
<em>Science. </em> 2007 Jun 8; 316(5830):1497-502. 
</p>

<p>
Wold B, Myers RM.
 <a href="http://www.nature.com/nmeth/journal/v5/n1/full/nmeth1157.html" class="external text" title="http://www.nature.com/nmeth/journal/v5/n1/full/nmeth1157.html" rel="noFollow" target="_BLANK">Sequence census methods for functional genomics</a>.
<em> Nat Methods </em> 2008 Jan; 5(1): 19-21.
</p>

<h2> Data Release Policy </h2>

<p>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <em>Restricted Until</em> column on the track configuration page and
the download page.  The full data release policy for ENCODE is available
<a href="../ENCODE/terms.html" target="_BLANK">here</a>.</p>


