<H2>Description</H2>

<P>
This track shows enrichment of RNA sequence tags mapped to the mouse genome generated by high 
throughput sequencing (RNA-Seq).  Double stranded cDNA was synthesized from enriched RNA
that was obtained after depletion of ribosomal RNA.  Pieces of cDNA, 300-350 nucleotides 
in length, were PCR amplified, adapter ligated, and sequenced on an Illumina HiSeq sequencer.
</p>

<H2>Display Conventions and Configuration</H2>

<P>
The Alignments view shows reads mapped to the genome and indicates where
bases may mismatch.  The alignment file follows the standard SAM format of Bowtie output.  See the <a href="http://bowtie-bio.sourceforge.net/manual.shtml#sam-bowtie-output" target=_blank>Bowtie Manual</a> for more information about the SAM Bowtie output and the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf#page=4" target=_blank>SAM Format Specification</a> for more information on the SAM/BAM file format. 
</P>

<H2>Methods</H2>

<P>
Cells were grown according to the approved
<a href="/ENCODE/protocols/cell" target="_blank">ENCODE cell culture protocols</a>.

Total RNA was extracted using RNeasy Mini Kit (74104, Aiagen), following the manufacturer's protocol.  Ribosomal RNA was removed from total RNA using the Ribo-Zero Gold Kits (MRZG126, Epicentre).  Double-stranded cDNA synthesis was performed on the rRNA depleted RNA using random primers and the SuperScript double-stranded cDNA synthesis kit (11917-010, Life Tech).  After first strand cDNA synthesis, NucAway Spin Column (Ambion cat. 100070-30) was used to remove dNTPs.  In the second strand cDNA synthesis reaction, dTTP in the dNTP mix was substituted with dUTP.  After end repair and addition of 'A' base to 3' end, illumina paired-end adapter was ligated to Double-stranded cDNA library.  After gel size selection of adapter ligated cDNA (300-350), Uracil-N-Glycosylase (UNG: Applied Biosystems) was used to digest the second strand cDNA (Parkhomchuk <EM>et al. </EM>, 2009).  PCR amplified adapter ligated cDNA was sequenced using Illumina HiSeq.  Sequence reads of 2x101 nt long with 0-2 mismatches were mapped to the mouse genome (version mm9) using the BWA aligner, version 0.5.7.  The signal height corresponds to the number of overlapping fragments at each nucleotide position in the genome.
</P>

<H2>Release Notes</H2>

<P>
This is Release 2 (August 12) of this track. The bigwig file for the MEL cell, replicate 2 with no treatment, was corrupt and has been replaced.  
</P>

<H2>Credits</H2>

<P>These data were generated and analyzed by the labs of
<A HREF="http://snyderlab.stanford.edu/"
  class="external text" title="http://snyderlab.stanford.edu/"
  target="_blank">Michael Snyder</a>.
<P>
&nbsp;Contact: <A 
HREF="mailto:&#112;c&#97;&#121;&#116;&#105;n&#103;&#64;&#115;&#116;&#97;n&#102;
o&#114;&#100;.&#101;d&#117;"
 title="mailto:&#112;c&#97;&#121;&#116;&#105;n&#103;&#64;&#115;&#116;&#97;n&#102;
 o&#114;&#100;.&#101;d&#117;" rel="nofollow" TARGET=_BLANK>Philip Cayting</A>.
 <!-- above address is (Philip Cayting) pcayting at stanford.edu -->
</P>

<H2>References</H2>

<p>
Parkhomchuk D, Borodina T, Amstislavskiy V, Banaru M, Hallen L, Krobitsch S, Lehrach H, Soldatov A.
<a href="http://nar.oxfordjournals.org/content/37/18/e123.long" target="_blank">
Transcriptome analysis by strand-specific sequencing of complementary DNA</a>.
<em>Nucleic Acids Res</em>. 2009 Oct;37(18):e123.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/19620212" target="_blank">19620212</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2764448/" target="_blank">PMC2764448</a>
</p>

<H2>Data Release Policy</H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column on the track configuration page and
the download page.  The full data release policy for ENCODE is available
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>
