
<H2> Description </H2>

<P>
Using RNA-Seq (Mortazavi <EM>et al.</EM>, 2008), high-resolution
genome-wide maps of the mouse transcriptome across multiple mouse (C57Bl/6) 
tissues and primary cells were generated.
</P>

<H2> Display Conventions and Configuration </H2>
<P>
This is a composite track that contains multiple data types
(<EM>views</EM>). For each view, there are multiple subtracks that
display individually on the browser. Instructions for configuring composite
tracks are <A HREF="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
This track contains the following views:
<DL>
<DT><I>Signal</I></DT><DD>Density graph (wiggle) of signal enrichment based
on processed data.</DD>
<DT><I>Alignments</I></DT><DD>Mappings of short reads to the genome. <DD>
</DL>
</P>

<H2> Methods </H2>
<P>
Cells were grown according to the approved
<A HREF="../../ENCODE/protocols/cell/mouse" TARGET=_BLANK>ENCODE cell culture protocols</A>.
</P>

<P><DT><B><I>RNA-Seq</I></B></DT>
RNA samples from tissues and primary cells were extracted
from Trizol&#174; according to protocol (Invitrogen). PolyA+ RNA
was purified with the Dynabeads mRNA purification kit (Invitrogen).
The mRNA libraries were prepared for strand-specific sequencing as
described previously (Parkhomchuk <EM>et al.</EM>, 2009).
</P>


<P><DT><B><I>Sequencing and Analysis</I></B></DT>
Samples were sequenced on Illumina Genome Analyzer II,
Genome Analyzer IIx and HiSeq 2000 platforms
for 36 cycles.  Image analysis, base calling and 
alignment to the mouse genome version mm9 were performed 
using Illumina&#39;s RTA.
Alignment to the mouse genome was performed using
TopHat (Trapnell <EM>et al</EM>., 2009).  Wig files were 
generated by TopHat and expression levels were 
calculated with Cufflinks (Trapnell <EM>et al.</EM>, 2010).
</P>

<H2> Credits </H2>

<P>
These data were generated and analyzed in
<A HREF="http://licr-renlab.ucsd.edu/" TARGET=_BLANK>Bing Ren&#39;s laboratory</A>
at the Ludwig Institute for Cancer Research.
</P>
<P>
Contact: <A HREF="mailto:&#121;&#55;&#115;h&#101;n&#64;&#117;&#99;&#115;&#100;.
&#101;&#100;u">Yin Shen</A>
<!-- above address is y7shen at ucsd.edu -->
</P>


<H2> References </H2>
<P>
Mortazavi, A., Williams, B.A., McCue, K., Schaeffer, L., and Wold, B.
<A HREF="http://www.nature.com/nmeth/journal/v5/n7/full/nmeth.1226.html"
    class="external text"
	title="http://www.nature.com/nmeth/journal/v5/n7/full/nmeth.1226.html"
	rel="noFollow" TARGET = _BLANK>Mapping and quantifying mammalian transcriptomes by RNA-Seq</A>.
<EM>Nat Methods</EM> 2008 May; 5: 621&#45;628.
</P>
<P>
Parkhomchuk, D., Borodina, T., Amstislavskiy, V., Banaru, M., Hallen, L., Krobitsch, S., Lehrach, H., 
and Soldatov, A. 
<A HREF="http://nar.oxfordjournals.org/content/37/18/e123.full"
    class="external text"
	title="http://nar.oxfordjournals.org/content/37/18/e123.full"
	rel="noFollow" TARGET = _BLANK>Transcriptome analysis by strand-specific sequencing of
complementary DNA</A>.
<EM>Nucleic Acids Res</EM> 2009 Jun; 37&#40;18&#41;: e123.
</P>
<P>
Trapnell C, Pachter L, and Salzberg SL
<A HREF="http://bioinformatics.oxfordjournals.org/content/25/9/1105.full"
    class="external text"
	title="http://bioinformatics.oxfordjournals.org/content/25/9/1105.full"
	rel="noFollow" TARGET = _BLANK>TopHat: discovering splice junctions with RNA-Seq</A>.
<EM>Bioinformatics</EM> 2009 Feb; 25&#40;9&#41;: 1105&#45;1111.
</P>
<P>
Trapnell C, Williams BA, Pertea G, Mortazavi A, Kwan G, van Baren MJ, Salzberg SL, 
Wold BJ, and Pachter L. 
<A HREF="http://www.nature.com/nbt/journal/v28/n5/abs/nbt.1621.html"
    class="external text"
	title="http://www.nature.com/nbt/journal/v28/n5/abs/nbt.1621.html"
	rel="noFollow" TARGET = _BLANK>Transcript assembly and quantification by RNA-Seq 
reveals unannotated transcripts and isoform switching during cell differentiation</A>.
<EM>Nature</EM> 2010 Feb; 28: 511&#45;515.
</P>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column on the track configuration page and
the download page.  The full data release policy for ENCODE is available
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>





