<H2>Description</H2>

<P>
This track shows a comprehensive survey of <i>cis</i>-regulatory elements
in the mouse genome by using ChIP-seq &#40;Robertson <EM> et al.</EM>, 2007&#41; to identify
transcription factor binding sites and chromatin modification
profiles in various mouse &#40;C57Bl/6&#41; tissues, primary cells, and cell lines. 
</P> 
<P>
The Ren lab examined RNA polymerase II &#40;PolII&#41;, 
co-activator protein p300, the insulator protein CTCF, and 
the following chromatin modification marks: H3K4me3 and H3K4me1, 
H3K27ac, H3K36me3, H3K9me3, and H3K27me3 due to their 
demonstrated utilities in identifying promoters, enhancers, 
insulator elements, actively transcribed gene bodies, and 
silent chromatin regions &#40;Barski <EM>et al.</EM>, 2007; Bernstein <EM>et al.</EM>, 2006; Blow <EM>et al.</EM>, 2010; 
Creyghton <EM>et al.</EM>, 2010; Francis <EM>et al.</EM>, 2004; Hawkins <EM>et al.</EM>, 2011; Heintzman <EM>et al.</EM>, 2009; Kim <EM>et al.</EM>, 2007; Kim <EM> et al.</EM>, 2005; Krogan <EM>et al.</EM>, 2003; Li <EM>et al.</EM>, 2002; Peters <EM> et al.</EM>, 2001; Rada-Iglesias <EM>et al.</EM>, 2011; Schotta <EM>et al.</EM>, 2002; Visel <EM>et al.</EM>, 2009&#41;.
Enrichment of PolII signals is a strong indicator of 
an active promoter and the presence of p300 outside of
promoter regions has been used as a mark for enhancers.  
CTCF binding sites are considered as a mark for 
potential insulator elements.  H3K4me3 is an active mark 
for promoters and H3K27ac is an active mark for both promoters 
and enhancers.  In the absence of H3K4me3, H3K4me1 serves as an
active mark for enhancers.  H3K36me3 is normally found in actively 
transcribed gene bodies whereas both H3K9me3 and H3K27me3 are common
repressive marks for transcriptionally silent chromatin regions.
For each transcription factor or chromatin mark in each tissue, 
ChIP-seq was carried out with at least two biological 
replicates. Each experiment produced 20-30 million uniquely-mapped monoclonal tags.
</P>

<H2>Display Conventions and Configuration</H2>

<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are <a href="/goldenPath/help/multiView.html" target="_blank">here</a>.
This track contains the following views:
<DL>
<DT><I>Peaks</I></DT><DD> Regions of signal enrichment based on processed data
(normalized data from pooled replicates). Intensity is represented in
grayscale; darker shading shows higher intensity (a solid vertical line
	in the peak region represents the point with the highest signal).
</DD>
<DT><I>Signal</I></DT><DD>Density graph (wiggle) of signal enrichment based on
processed data.</DD>
</DL>
</P>
<p>Metadata for a particular subtrack can be found by clicking the down arrow in the list of subtracks.</p>
<P>
<!--echo var="BROWSER_HOST" -->
Additional views are available on the Downloads page. 
</P>

<H2>Methods</H2>
<P>
Cells were grown according to the approved
<a href="/ENCODE/protocols/cell/mouse" target="_blank">ENCODE cell culture protocols</a>.
</P>

<P><DT><B><I>Enrichment and Library Preparation</I></B></DT>

Chromatin immunoprecipitation was performed according to the <A HREF="http://bioinformatics-renlab.ucsd.edu/RenLabChipProtocolV1.pdf" 
TARGET=_BLANK>Ren Lab ChIP Protocol</A>.
</P>
<P>
Library construction was performed according to the <A HREF="http://bioinformatics-renlab.ucsd.edu/RenLabLibraryProtocolV1.pdf"
TARGET=_BLANK>Ren Lab Library Protocol</A>.
</P>

<P><DT><B><I>Sequencing and Analysis</I></B></DT>
Samples were sequenced on Illumina Genome Analyzer II
Genome Analyzer IIx and HiSeq 2000 platforms
for 36 cycles.  Image analysis, base calling and 
alignment to the mouse genome version NCBI37/mm9 were performed 
using Illumina&#39;s RTA and Genome Analyzer Pipeline software. 
Alignment to the mouse genome was performed using ELAND
or Bowtie (Langmead <EM>et al.</EM>, 2009) with 
a seed length of 25 and allowing up to two mismatches. 
Only the sequences that mapped to one location were used 
for further analysis.  Of those sequences, clonal reads,
defined as having the same start position on the same 
strand, were discarded.  BED and wig files were
created using custom perl scripts.
</P>

<h2>Release Notes</h2>

<p>
This is Release 2 (Mar 2012). It contains a total of 99 ChIP-seq experiments on histone modifications with the addition of 69 new experiments.
</p>

<H2>Credits</H2>

<P>
These data were generated and analyzed in
<A HREF="http://bioinformatics-renlab.ucsd.edu/rentrac" TARGET=_BLANK>Bing Ren&#39;s laboratory</A>
at the Ludwig Institute for Cancer Research.
</P>
<P>
Contact: <A HREF="mailto:&#121;&#55;&#115;h&#101;n&#64;&#117;&#99;&#115;&#100;.
&#101;&#100;u">Yin Shen</A>
<!-- above address is y7shen at ucsd.edu -->
</P>

<H2>References</H2>

<p>
Barski A, Cuddapah S, Cui K, Roh TY, Schones DE, Wang Z, Wei G, Chepelev I, Zhao K.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/17512414" target="_blank">
High-resolution profiling of histone methylations in the human genome</a>.
<em>Cell</em>. 2007 May 18;129(4):823-37.
</p>

<p>
Bernstein BE, Mikkelsen TS, Xie X, Kamal M, Huebert DJ, Cuff J, Fry B, Meissner A, Wernig M, Plath K <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/16630819" target="_blank">
A bivalent chromatin structure marks key developmental genes in embryonic stem cells</a>.
<em>Cell</em>. 2006 Apr 21;125(2):315-26.
</p>

<p>
Blow MJ, McCulley DJ, Li Z, Zhang T, Akiyama JA, Holt A, Plajzer-Frick I, Shoukry M, Wright C, Chen F <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/20729851" target="_blank">
ChIP-Seq identification of weakly conserved heart enhancers</a>.
<em>Nat Genet</em>. 2010 Sep;42(9):806-10.
</p>

<p>
Creyghton MP, Cheng AW, Welstead GG, Kooistra T, Carey BW, Steine EJ, Hanna J, Lodato MA, Frampton GM, Sharp PA <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/21106759" target="_blank">
Histone H3K27ac separates active from poised enhancers and predicts developmental state</a>.
<em>Proc Natl Acad Sci U S A</em>. 2010 Dec 14;107(50):21931-6.
</p>

<p>
Francis NJ, Kingston RE, Woodcock CL.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/15567868" target="_blank">
Chromatin compaction by a polycomb group protein complex</a>.
<em>Science</em>. 2004 Nov 26;306(5701):1574-7.
</p>

<p>
Hawkins RD, Hon GC, Yang C, Antosiewicz-Bourget JE, Lee LK, Ngo QM, Klugman S, Ching KA, Edsall LE, Ye Z <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/21876557" target="_blank">
Dynamic chromatin states in human ES cells reveal potential regulatory sequences and genes involved in pluripotency</a>.
<em>Cell Res</em>. 2011 Oct;21(10):1393-409.
</p>

<p>
Heintzman ND, Hon GC, Hawkins RD, Kheradpour P, Stark A, Harp LF, Ye Z, Lee LK, Stuart RK, Ching CW <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/19295514" target="_blank">
Histone modifications at human enhancers reflect global cell-type-specific gene expression</a>.
<em>Nature</em>. 2009 May 7;459(7243):108-12.
</p>

<p>
Kim TH, Abdullaev ZK, Smith AD, Ching KA, Loukinov DI, Green RD, Zhang MQ, Lobanenkov VV, Ren B.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/17382889" target="_blank">
Analysis of the vertebrate insulator protein CTCF-binding sites in the human genome</a>.
<em>Cell</em>. 2007 Mar 23;128(6):1231-45.
</p>

<p>
Kim TH, Barrera LO, Qu C, Van Calcar S, Trinklein ND, Cooper SJ, Luna RM, Glass CK, Rosenfeld MG, Myers RM <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/15899964" target="_blank">
Direct isolation and identification of promoters in the human genome</a>.
<em>Genome Res</em>. 2005 Jun;15(6):830-9.
</p>

<p>
Krogan NJ, Kim M, Tong A, Golshani A, Cagney G, Canadien V, Richards DP, Beattie BK, Emili A, Boone C <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/12773564" target="_blank">
Methylation of histone H3 by Set2 in Saccharomyces cerevisiae is linked to transcriptional elongation by RNA polymerase II</a>.
<em>Mol Cell Biol</em>. 2003 Jun;23(12):4207-18.
</p>

<p>
Langmead B, Trapnell C, Pop M, Salzberg SL.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/19261174" target="_blank">
Ultrafast and memory-efficient alignment of short DNA sequences to the human genome</a>.
<em>Genome Biol</em>. 2009;10(3):R25.
</p>

<p>
Li J, Moazed D, Gygi SP.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/12381723" target="_blank">
Association of the histone methyltransferase Set2 with RNA polymerase II plays a role in transcription elongation</a>.
<em>J Biol Chem</em>. 2002 Dec 20;277(51):49383-8.
</p>

<p>
Peters AH, O'Carroll D, Scherthan H, Mechtler K, Sauer S, Sch&#246;fer C, Weipoltshammer K, Pagani M, Lachner M, Kohlmaier A <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/11701123" target="_blank">
Loss of the Suv39h histone methyltransferases impairs mammalian heterochromatin and genome stability</a>.
<em>Cell</em>. 2001 Nov 2;107(3):323-37.
</p>

<p>
Rada-Iglesias A, Bajpai R, Swigut T, Brugmann SA, Flynn RA, Wysocka J.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/21160473" target="_blank">
A unique chromatin signature uncovers early developmental enhancers in humans</a>.
<em>Nature</em>. 2011 Feb 10;470(7333):279-83.
</p>

<p>
Robertson G, Hirst M, Bainbridge M, Bilenky M, Zhao Y, Zeng T, Euskirchen G, Bernier B, Varhol R, Delaney A <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/17558387" target="_blank">
Genome-wide profiles of STAT1 DNA association using chromatin immunoprecipitation and massively parallel sequencing</a>.
<em>Nat Methods</em>. 2007 Aug;4(8):651-7.
</p>

<p>
Schotta G, Ebert A, Krauss V, Fischer A, Hoffmann J, Rea S, Jenuwein T, Dorn R, Reuter G.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/11867540" target="_blank">
Central role of Drosophila SU(VAR)3-9 in histone H3-K9 methylation and heterochromatic gene silencing</a>.
<em>EMBO J</em>. 2002 Mar 1;21(5):1121-31.
</p>

<p>
Visel A, Blow MJ, Li Z, Zhang T, Akiyama JA, Holt A, Plajzer-Frick I, Shoukry M, Wright C, Chen F <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/19212405" target="_blank">
ChIP-seq accurately predicts tissue-specific activity of enhancers</a>.
<em>Nature</em>. 2009 Feb 12;457(7231):854-8.
</p>

<H2>Data Release Policy</H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column, above.  The full data release policy for ENCODE is available
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>
