<H2>Description</H2>
<P>
This track shows alignments of the <em>$organism</em> genome with itself, using
a gap scoring system that allows longer gaps than traditional
affine gap scoring systems. The system can also tolerate gaps
in both sets of sequence simultaneously. After filtering out the 
&quot;trivial&quot; alignments produced when identical locations of the genome 
map to one another (e.g. chrN maps to chrN), the remaining 
alignments point out areas of duplication within the <em>organism</em> genome.
A second filter was applied to remove chains scoring less than 20,000. 
<P>
The chain track displays boxes joined by either single or
double lines. The boxes represent aligning regions. 
Single lines indicate gaps that are largely due to a deletion in the query 
assembly or an insertion in the target assembly.
Double lines represent more complex gaps that involve substantial
sequence in both assemblies. This may result from inversions, overlapping
deletions, an abundance of local mutation, or an unsequenced gap in one of the
assemblies.
In cases where there are multiple
chains over a particular portion of the <em>$o_organism</em> genome, chains with
single-lined gaps are often due to processed pseudogenes, while chains
with double-lined gaps are more often due to paralogs and unprocessed
pseudogenes.
<P>
The display indicates the chromosome, strand, and location of the match
for each matching alignment (in thousands).

<H2>Methods</H2>
<P>
The genome was aligned to itself using
<A HREF="http://bio.cse.psu.edu/dist/blastz.pdf" TARGET=_blank>blastz</A>.  
Trivial alignments were filtered out. The remaining alignments were converted 
into axt format
and the resulting axts are fed into axtChain.  This program organizes all 
the alignments between a target and a single query chromosome
into a group and makes a kd-tree out of all the gapless subsections
(blocks) of the alignments.  Next, maximally scoring chains of these
blocks were found by running a dynamic program over the kd-tree.  Chains
scoring below a threshold were discarded. The remaining chains are
displayed in this track.

<H2>Credits</H2>
<P>
Blastz was developed at <A HREF="http://bio.cse.psu.edu" 
TARGET=_blank>Pennsylvania State University</A> by 
Scott Schwartz, Zheng Zhang, and Webb Miller with advice from
Ross Hardison.</P> 
<P>
Lineage-specific repeats were identified by Arian Smit and his program
<A HREF="http://www.repeatmasker.org" TARGET=_blank>RepeatMasker</A>. For this 
track, the RepeatMasker output for the <em>$o_organism</em> assembly was used 
for lineage-specific repeats of both query and target. </P>
<P>
The axtChain program was developed at the University of California
at Santa Cruz by Jim Kent with advice from Webb Miller and David Haussler.</P>
<P>
The browser display and database storage of the chains were made
by <A
HREF="mailto:mailto:&#104;a&#114;&#116;&#101;&#114;a&#64;&#115;&#111;&#101;.
&#117;&#99;&#115;c.
&#101;&#100;&#117;">Rachel Harte</A>
<!-- above address is hartera at soe.ucsc.edu -->
and <A HREF="http://www.soe.ucsc.edu/~kent/">Jim Kent</A>.</P>

<H2>References</H2>
<P>
Chiaromonte, F., Yap, V.B., Miller, W. 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. 
<em>Pac Symp Biocomput 2002</em>, 115-26 (2002).</P>
<P>
Schwartz, S., Kent, W.J., Smit, A., Zhang, Z., Baertsch, R., Hardison, R., 
Haussler, D., and Miller, W.
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103" TARGET=_blank>Human-Mouse 
Alignments with BLASTZ</A>. <em>Genome Res.</em> <B>13</B>(1), 103-7 (2003).
</P>
