<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=315>
  <TR><TD ALIGN=RIGHT>
	<A HREF="http://www.wormbase.org/about/about_Celegans.html"
		TARGET=_blank>
    <IMG SRC="/images/Caenorhabditis_briggsae.jpg" WIDTH=300 HEIGHT=238 ALT="C. briggsae"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Caenorhabditis briggsae</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of
      <A HREF="http://www.life.umd.edu/biology/faculty/haag/Haag_Lab_Members_Page.html" 
      TARGET=_blank>Eric Haag,</A> All Rights Reserved.
    </FONT>
  </TD></TR>
</TABLE>

<P>
The <em>Caenorhabditis briggsae</em> genome assembly 
is based on sequence version cb25.agp8 deposited into 
<A HREF="http://www.wormbase.org/" TARGET=_blank>WormBase</A> 
as of August 2005. 

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a 
sequenced genomic clone, an mRNA or EST, or a gene identifier, 
chromosomal coordinate range, or keywords from the GenBank description of 
an mRNA. The following list shows examples of valid position queries for 
the <em>C. briggsae</em> genome. 
See the
<A HREF="../goldenPath/help/hgTracksHelp.html" TARGET=_blank>
User's Guide</A> for more information.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
       	<TD VALIGN=Top COLSPAN=2><B>&nbsp;&nbsp; Genome Browser Response:</B><BR></TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chrI</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays all of chromosome I</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>chrI:1-1000000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays the first million bases of chrI</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>R02870</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays region of EST associated with GenBank accession R02870</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>U48289</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays the region associated with the mRNA accession U48289</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>protein phosphatase PP2C </TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists only records pertaining to protein phophatase PP2C</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>Kimble,J.</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author J. Kimble</TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries.
	Although GenBank requires the search format <em>Kimble J</em>,
	internally it uses the format <em>Kimble,J.</em>.</TD>
    </TR>
</TABLE>
<P>
<HR>
<H3>Assembly details</H3>
<P>
This version was assembled from 2.05 million whole genome shotgun
(WGS) reads, of which 88.2% were in read pairs. The sequence reads
were generated at Washington University School of Medicine's Genome
Sequencing Center and the Sanger Centre (Hinxton, UK).  The Phusion
(Mullikin, J) assembler was used to assemble the WGS reads into
contigs based on overlap information, and then into supercontigs using
read pair information to cross gaps. These supercontigs were assembled
into mapped ultracontigs based on FPC fingerprint mapping (generated
at Washington University School of Medicine's Genome Sequencing
Center), with material from previously finished clones used to bridge
gaps. Using the sequence data to choose primers along major
ultracontigs, a genetic map (Ray Miller, Washington University School
of Medicine) was constructed which yielded 6 chromosomes.  The genetic
mapping data along with data from 1:1 elegans:briggsae orthologs were
used to refine ultracontigs and to order and orient the sequences
along the C. briggsae chromosomes.
</P>
<P>
The six chromosomes are named: chrI, chrII, chrIII, chrIV, chrV and chrX,
containing 89,604,507 bp of ordered and oriented sequence.  
An additional 11,714,720 bp of sequence is localized to a
specific chromosome but not ordered and displayed on the artifical
chr*_random chromosomes.  7,825,149 bp of sequence is unordered and
not localized, displayed on the chrUn artifical chromosome.
</P>
<P>
The sequencing centers
estimate that this whole genome shotgun assembly achieved 98% coverage
of the genome.
</P>
<P>
The cb2 sequence and annotation data may be downloaded from the UCSC Genome 
Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/cb2/">FTP server</A>
or
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#c_briggsae">Downloads</A>
web page.
The sequence data have
<A HREF="http://genome.wustl.edu/data/data_use_policy" TARGET=_blank>specific conditions for use</A>. 
The cb2 annotation tracks were generated by UCSC and 
collaborators worldwide. See the <A HREF="../goldenPath/credits.html#worm_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to this release. 
</P>
