<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=215>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://www.genoscope.cns.fr/spip/Tetraodon-nigroviridis-a-fish-with.html"
    TARGET=_blank>
    <IMG SRC="/images/Tetraodon_nigroviridis.jpg" WIDTH=200 HEIGHT=96
      ALT="Tetraodon"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Tetraodon nigroviridis</em><BR> 
    </FONT>
    <FONT SIZE=-2> (Photo courtesy of
      <A HREF="http://www.genoscope.cns.fr/spip/" TARGET=_blank>Genoscope</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The <em>Tetraodon nigroviridis</em> V8 assembly (March 2007)
was provided by 
<A HREF="http://www.genoscope.cns.fr/spip/" TARGET=_blank>Genoscope</A>, Evry, 
France, in collaboration with the
<A HREF="http://www.broadinstitute.org/annotation/tetraodon/background.html"
TARGET=_blank>Broad Institute</A>, Cambridge, MA, USA. The project was 
supported by the Consortium National de Recherche 
en Genomique and the National Human Genome Research Institute (NHGRI).
For more information about this assembly, see
<A HREF="http://www.ncbi.nlm.nih.gov/assembly/228438/" TARGET="_blank">
ASM18073v1</A> in the NCBI Assembly database.
</P>

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of
an mRNA or EST, a chromosome range, or keywords from 
the GenBank description of an mRNA. The following list provides 
examples of various types of position queries for the Tetraodon
genome. Note that some position queries (e.g. "huntington") may
return matches to the mRNA records of other species. In these
cases, the mRNAs are mapped to their homologs in Tetraodon.
See the <A HREF="../goldenPath/help/hgTracksHelp.html"
TARGET=_blank>User Guide</A> for more information. 
</P><P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
<TR><TD VALIGN=Top NOWRAP><B>Request:</B><br></TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top><B>Genome Browser Response:</B><br></TD></TR>
	
<TR><TD VALIGN=Top><br></TD></TR>
	
<TR><TD VALIGN=Top NOWRAP>chr1</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays all of chromosome 1</TD></TR>
<TR><TD VALIGN=Top NOWRAP>chr1:1-200000</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays first two hundred thousand bases of chromosome 1</TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chr1:100000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chromosome 1 that spans 2000 bases, starting with position 100000 </TD>
    </TR>
<TR><TD VALIGN=Top><br></TD></TR>

<TR><TD VALIGN=Top NOWRAP>CR731623</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays region containing Tetraodon mRNA with GenBank accession number CR731623</TD></TR>

<TR><TD VALIGN=Top NOWRAP>BC083244</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays region containing Other mRNA with GenBank accession BC083244</TD></TR>

<TR><TD VALIGN=Top><br></TD></TR>

<TR><TD VALIGN=Top NOWRAP>IL11</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs related to IL11 (interleukin-11)</TD>
</TR>


<TR><TD VALIGN=Top NOWRAP>T-cell receptor</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs for T-cell receptor genes in GenBank</TD></TR>
<TR><TD VALIGN=Top NOWRAP>kinase</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists many kinase mRNAs</TD></TR>
<TR><TD VALIGN=Top NOWRAP>tyrosine kinase</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists only tyrosine kinases</TD></TR>
<TR><TD VALIGN=Top NOWRAP>breast cancer</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs associated with breast cancer</TD></TR>

<TR><TD VALIGN=Top><br></TD></TR>

<TR><TD VALIGN=Top NOWRAP>abkevich</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Abkevich</TD></TR>
<TR><TD VALIGN=Top NOWRAP>Hashimoto,H.</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by co-author H. Hashimoto</TD></TR>

<TR><TD VALIGN=Top><br></TD></TR>
	
<TR><TD COLSPAN="3" > Use this last format for author queries. Although 
GenBank requires the search format <em>Hashimoto H</em>, internally it uses
the format <em>Hashimoto,H.</em>
</TD></TR>
</TABLE>
</P>

<HR>
<H3>Assembly details</H3>
<P>
The Tetraodon was sequenced through a collaboration between Genoscope and the 
Broad Institute using a whole genome sequencing strategy. 
The sequence data were assembled using Arachne, resulting in a sequence coverage
of about 7.9X. This assembly, which has been improved from the v7 assembly by 
using a new fosmid library of around 6x of clone coverage, contains 1,849 
placed scaffolds (N50 size 1349163) and 15,296 unplaced scaffolds (N50 size 
54038) covering more than 90% of the genome. 
For detailed sequencing and assembly information, see the Genoscope
<em>Tetraodon nigroviridis</em> 
<A HREF="http://www.genoscope.cns.fr/spip/Tetraodon-nigroviridis-whole.html"
TARGET=_blank>assembly</A> page. 
</P>
<P>
Downloads of the Tetraodon data and annotations may be obtained from the UCSC
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/tetNig2/" TARGET=_blank>FTP 
site</A> or <A HREF=http://hgdownload.cse.ucsc.edu/downloads.html#tetraodon>
Downloads</A> page. 
These data have been freely provided by Genoscope for use in the UCSC Genome 
Browser with 
<A HREF="../goldenPath/credits#tetraodon_use"
TARGET=_blank>specific conditions for use</A>. For more information about
the <em>Tetraodon nigroviridis</em> project, see the Genoscope
<A HREF="http://www.genoscope.cns.fr/spip/-Tetraodon-nigroviridis,387-.html"
TARGET=_blank>website</A>.
</P>
<P>
The tetNig2 annotation tracks were generated by UCSC and 
its collaborators. See the 
<A HREF="../goldenPath/credits.html#tetraodon_credits">credits</A> page
for a detailed list of the organizations and individuals who contributed 
to this release. 
</P>
<HR>
<H3>Genbank Pipeline Details</H3>
For the purposes of the Genbank alignment pipeline, this assembly is considered to be: <B>well-ordered</B>.
