<H2>Description</H2>
<P>
The RefSeq Genes track shows known protein-coding genes taken from 
the NCBI mRNA reference sequences collection (RefSeq). On assemblies in 
which incremental GenBank downloads are supported, the data underlying this 
track are updated nightly.</P>

<H2>Display Conventions and Configuration</H2>
<P>
This track follows the display conventions for 
<A HREF="../goldenPath/help/hgTracksHelp.html#GeneDisplay">gene prediction 
tracks</A>.
The color shading indicates the level of review the RefSeq record has 
undergone: predicted (light), provisional (medium), reviewed (dark). 
In some assemblies, non-coding RNA genes are shown in a separate track.</P>
<P>
The item labels and display colors of features within this track can be
configured through the controls at the top of the track description page. 
This page is accessed via the small button to the left of the track's 
graphical display or through the link on the track's control menu. 
<UL>
<LI><B>Label:</B> By default, items are labeled by gene name. Click the 
appropriate Label option to display the accession name instead of the gene
name, show both the gene and accession names, or turn off the label 
completely.
<LI><B>Codon coloring:</B> This track contains an optional codon coloring 
feature that allows users to quickly validate and compare gene predictions.
To display codon colors, select the <em>genomic codons</em> option from the
<em>Color track by codons</em> pull-down menu. Click 
<A HREF="../goldenPath/help/hgCodonColoring.html">here</A> for more 
information about this feature.
</UL></P>
<P>
After you have made your selections, click the <em>Submit</em> button to 
return to the tracks display page.</P>

<H2>Methods</H2>
<P>
RefSeq mRNAs were aligned against the <em>$Organism</em> genome using blat; 
those with an alignment of less than 15% were discarded. When a single mRNA 
aligned in multiple places, the alignment having the highest base identity 
was identified.  Only alignments having a base identity level within 0.1% of 
the best and at least 96% base identity with the genomic sequence were kept.
</P>


<H2>Credits</H2>
<P>
This track was produced at UCSC from mRNA sequence data
generated by scientists worldwide and curated by the 
NCBI <A HREF="http://www.ncbi.nlm.nih.gov/RefSeq/" 
TARGET=_blank>RefSeq project</A>.  </P>

<H2>References</H2>
<P>
Kent, W.J.
<A HREF="http://www.genome.org/cgi/content/abstract/12/4/656"
TARGET=_blank>BLAT - the BLAST-like alignment tool</A>.
<em>Genome Res.</em> <B>12</B>(4), 656-664 (2002).</P>
  
<P>Pruitt K.D., Tatusova, T., Maglott D.R. 
<A HREF="http://nar.oxfordjournals.org/cgi/content/full/33/suppl_1/D501?ijkey=06NkMzN3kUcez&keytype=ref" 
TARGET=_BLANK>NCBI Reference Sequence (RefSeq): a curated non-redundant 
sequence database of genomes, transcripts and proteins</A> <EM>Nucleic Acids 
Res.</EM> <B>33</B>(1), D501-D504 (2005).
</P>


