<H2>Description</H2>
<P>
This track shows multiple alignments of 13 vertebrate
species and measurements of evolutionary conservation using
two methods (<em>phastCons</em> and <em>phyloP</em>) from the
<A HREF="http://compgen.bscb.cornell.edu/phast/" target=_BLANK>
PHAST package</A>, for all 13 species. 
The multiple alignments were generated using multiz and
other tools in the UCSC/<A HREF="http://bio.cse.psu.edu/" 
TARGET=_blank>Penn State Bioinformatics</A>
comparative genomics alignment pipeline.
Conserved elements identified by phastCons are also displayed in
this track. 
</P>
<P>
PhastCons (which has been used in previous Conservation tracks) is a hidden
Markov model-based method that estimates the probability that each
nucleotide belongs to a conserved element, based on the multiple alignment.
It considers not just each individual alignment column, but also its
flanking columns.  By contrast, phyloP separately measures conservation at
individual columns, ignoring the effects of their neighbors.  As a
consequence, the phyloP plots have a less smooth appearance than the
phastCons plots, with more "texture" at individual sites.  The two methods
have different strengths and weaknesses.  PhastCons is sensitive to "runs"
of conserved sites, and is therefore effective for picking out conserved
elements.  PhyloP, on the other hand, is more appropriate for evaluating
signatures of selection at particular nucleotides or classes of nucleotides
(e.g., third codon positions, or first positions of miRNA target sites).
</P>
<P>
Another important difference is that phyloP can measure acceleration
(faster evolution than expected under neutral drift) as well as
conservation (slower than expected evolution).  In the phyloP plots, sites
predicted to be conserved are assigned positive scores (and shown in blue),
while sites predicted to be fast-evolving are assigned negative scores (and
shown in red).  The absolute values of the scores represent -log p-values
under a null hypothesis of neutral evolution.  The phastCons scores, by
contrast, represent probabilities of negative selection and range between 0
and 1.
</P>

The multiple alignments were generated using multiz and 
other tools in the UCSC/<A HREF="http://bio.cse.psu.edu/" 
TARGET=_blank>Penn State Bioinformatics</A>
comparative genomics alignment pipeline.
The conservation measurements were created using the phastCons package from
<A HREF="http://compgen.bscb.cornell.edu/~acs/software.html" TARGET=_blank>
Adam Siepel</A> at Cornell University. </P>
<P>
UCSC has repeatmasked and aligned the low-coverage genome assemblies, and
provides the sequence for download; however, we do not construct
genome browsers for them. Missing sequence in the low-coverage assemblies is
highlighted in the track display by regions of yellow when zoomed out
and Ns displayed at base level (see <EM>Gap Annotation</EM>, below). </P>
<P>
<BLOCKQUOTE><TABLE BORDER=1 CELLPADDING=4 BORDERCOLOR="#aaaaaa">
<TR ALIGN=left><TH>Organism</TH><TH>Species</TH><TH>Release date</TH><TH>UCSC version</TH><TH>alignment type</TH></TR>
<TR ALIGN=left><TD>Marmoset</TD><TD>Callithrix jacchus</TD><TD>
    Mar. 2009</TD><TD><A HREF="../cgi-bin/hgGateway?db=calJac3"
    TARGET=_blank>calJac3</A></TD><TD>reference species</TD></TR>
<TR ALIGN=left><TD>Human</TD><TD>Homo sapiens</TD><TD>
    Feb. 2009</TD><TD> <A HREF="../cgi-bin/hgGateway?db=hg19"
    TARGET=_blank>hg19/GRCh37</A></TD><TD>Syntenic Net</TD></TR>
<TR ALIGN=left><TD>Baboon</TD><TD>Papio hamadryas</TD><TD>Nov. 2008</TD>
    <TD><A HREF="ftp://hgdownload.cse.ucsc.edu/gbdb/papHam1/"TARGET=_blank>papHam1</A>*</TD>
    <TD>Reciprocal Best</TD></TR>
<TR ALIGN=left><TD>Bushbaby</TD><TD>Otolemur garnettii</TD><TD>Dec. 2006</TD>
    <TD><A HREF="ftp://hgdownload.cse.ucsc.edu/gbdb/otoGar1/"TARGET=_blank>otoGar1</A>*</TD>
    <TD>Reciprocal Best</TD></TR>
<TR ALIGN=left><TD>Chimp</TD><TD>Pan troglodytes</TD><TD>
    Mar. 2006</TD><TD><A HREF="../cgi-bin/hgGateway?db=panTro2"
    TARGET=_blank>panTro2</A></TD><TD>Syntenic Net</TD></TR>
<TR ALIGN=left><TD>Dog</TD><TD>Canis lupus familiaris</TD><TD>
    May 2005</TD><TD><A HREF="../cgi-bin/hgGateway?db=canFam2"
    TARGET=_blank>canFam2</A></TD><TD>Syntenic Net</TD></TR>
<TR ALIGN=left><TD>Gorilla</TD><TD>Gorilla gorilla gorilla</TD><TD>Oct. 2009</TD>
    <TD><A HREF="ftp://hgdownload.cse.ucsc.edu/gbdb/gorGor2/"TARGET=_blank>gorGor2</A>*</TD>
    <TD>Syntenic Net</TD></TR>
<TR ALIGN=left><TD>Mouse</TD><TD>Mus musculus</TD><TD>
    July 2007</TD><TD><A HREF="../cgi-bin/hgGateway?db=mm9"
    TARGET=_blank>mm9</A></TD><TD>Syntenic Net</TD></TR>
<TR ALIGN=left><TD>Opossum</TD><TD>Monodelphis domestica</TD><TD>
    Oct. 2006</TD><TD><A HREF="../cgi-bin/hgGateway?db=monDom5"
    TARGET=_blank>monDom5</A></TD><TD>Syntenic Net</TD></TR>
<TR ALIGN=left><TD>Orangutan</TD><TD>Pongo pygmaeus abelii</TD><TD>
    July 2007</TD><TD><A HREF="../cgi-bin/hgGateway?db=ponAbe2"
    TARGET=_blank>ponAbe2</A></TD><TD>Syntenic Net</TD></TR>
<TR ALIGN=left><TD>Rhesus</TD><TD>Macaca mulatta</TD><TD>
    Jan. 2006</TD><TD><A HREF="../cgi-bin/hgGateway?db=rheMac2"
    TARGET=_blank>rheMac2</A></TD><TD>Syntenic Net</TD></TR>
<TR ALIGN=left><TD>Mouse lemur</TD><TD>Microcebus murinus</TD><TD>Jun. 2003</TD>
    <TD><A HREF="ftp://hgdownload.cse.ucsc.edu/gbdb/micMur1"TARGET=_blank>micMur1</A>*</TD>
    <TD>Reciprocal Best</TD></TR>
<TR ALIGN=left><TD>Tarsier</TD><TD>Tarsier syrichta</TD><TD>Aug. 2008</TD>
    <TD><A HREF="ftp://hgdownload.cse.ucsc.edu/gbdb/tarSyr1/"TARGET=_blank>tarSyr1</A>*</TD>
    <TD>Reciprocal Best</TD></TR>
</TABLE><BR>

<B>Table 1.</B> <EM>Genome assemblies included in the 13-way Conservation 
track.</EM>
<BR>* Data download only, browser not available.
</BLOCKQUOTE></P>

Downloads for data in this track are available:
<UL>
<LI>
<A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/calJac3/multiz13way/">Multiz alignments</A> (MAF format), and phylogenetic trees <LI>
<A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/calJac3/phyloP13way/">PhyloP conservation</A> (WIG format)
<LI>
<A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/calJac3/phastCons13way/">PhastCons conservation</A> (WIG format)
</UL>


<H2>Display Conventions and Configuration</H2>
<P>
The track configuration options allow the user to display either
the vertebrate or placental mammal conservation scores, or both
simultaneously.
In full and pack display modes, conservation scores are displayed as a
<EM>wiggle track</EM> (histogram) in which the height reflects the 
size of the score. 
The conservation wiggles can be configured in a variety of ways to 
highlight different aspects of the displayed information. 
Click the <A HREF="../goldenPath/help/hgWiggleTrackHelp.html" 
TARGET=_blank>Graph configuration help</A> link for an explanation 
of the configuration options.</P>
<P>
Pairwise alignments of each species to the $organism genome are 
displayed below the conservation histogram as a grayscale density plot (in 
pack mode) or as a wiggle (in full mode) that indicates alignment quality.
In dense display mode, conservation is shown in grayscale using
darker values to indicate higher levels of overall conservation 
as scored by phastCons. </P>
<P>
Checkboxes on the track configuration page allow selection of the
species to include in the pairwise display.  
Note that excluding species from the pairwise display does not alter the
the conservation score display.</P>
<P>
To view detailed information about the alignments at a specific
position, zoom the display in to 30,000 or fewer bases, then click on
the alignment.</P>

<H3>Gap Annotation</H3>
<P>
The <EM>Display chains between alignments</EM> configuration option 
enables display of gaps between alignment blocks in the pairwise alignments in 
a manner similar to the Chain track display.  The following
conventions are used:
<UL>
<LI><B>Single line:</B> no bases in the aligned species. Possibly due to a
lineage-specific insertion between the aligned blocks in the $organism genome
or a lineage-specific deletion between the aligned blocks in the aligning
species.
<LI><B>Double line:</B> aligning species has one or more unalignable bases in
the gap region. Possibly due to excessive evolutionary distance between 
species or independent indels in the region between the aligned blocks in both
species. 
<LI><B>Pale yellow coloring:</B> aligning species has Ns in the gap region.
Reflects uncertainty in the relationship between the DNA of both species, due
to lack of sequence in relevant portions of the aligning species. 
</UL></P>

<H3>Genomic Breaks</H3>
<P>
Discontinuities in the genomic context (chromosome, scaffold or region) of the
aligned DNA in the aligning species are shown as follows: 
<UL>
<LI>
<B>Vertical blue bar:</B> represents a discontinuity that persists indefinitely
on either side, <em>e.g.</em> a large region of DNA on either side of the bar
comes from a different chromosome in the aligned species due to a large scale
rearrangement. 
<LI>
<B>Green square brackets:</B> enclose shorter alignments consisting of DNA from
one genomic context in the aligned species nested inside a larger chain of
alignments from a different genomic context. The alignment within the
brackets may represent a short misalignment, a lineage-specific insertion of a
transposon in the $organism genome that aligns to a paralogous copy somewhere
else in the aligned species, or other similar occurrence.
</UL></P>

<H3>Base Level</H3>
<P>
When zoomed-in to the base-level display, the track shows the base 
composition of each alignment, with bases in annotated repetitive 
elements displayed in lower case. The numbers and symbols on the Gaps
line indicate the lengths of gaps in the $organism sequence at those 
alignment positions relative to the longest non-$organism sequence. 
If there is sufficient space in the display, the size of the gap is shown. 
If the space is insufficient and the gap size is a multiple of 3, a 
&quot;*&quot; is displayed; other gap sizes are indicated by &quot;+&quot;.</P>
<P>
Codon translation is available in base-level display mode if the
displayed region is identified as a coding segment. To display this annotation,
select the species for translation from the pull-down menu in the Codon
Translation configuration section at the top of the page. Then, select one of
the following modes:
<UL>
<LI> 
<B>No codon translation:</B> the gene annotation is not used; the bases are
displayed without translation. 
<LI>
<B>Use default species reading frames for translation:</B> the annotations from the genome
displayed 
in the <EM>Default species for translation</EM>; pull-down menu are used to
translate all the aligned species present in the alignment. 
<LI>
<B>Use reading frames for species if available, otherwise no translation:</B> codon
translation is performed only for those species where the region is
annotated as protein coding.
<LI><B>Use reading frames for species if available, otherwise use default species:</B>
codon translation is done on those species that are annotated as being protein
coding over the aligned region using species-specific annotation; the remaining
species are translated using the default species annotation. 
</UL></P>
<P>
Codon translation uses the following gene tracks as the basis for
translation, depending on the species chosen (<B>Table 2</B>).  
Species listed in the row labeled &quot;None&quot; do not have 
species-specific reading frames for gene translation.

<BLOCKQUOTE><TABLE BORDER=1 CELLPADDING=4 BORDERCOLOR="#aaaaaa">
<TR ALIGN=left><TD><B>Gene Track</B></TD><TD><B>Species</B></TD></TR>
<TR ALIGN=left><TD>Known Genes</TD><TD>hg19, mm9</TD></TR>
<TR ALIGN=left><TD>Ensembl Genes</TD><TD> catJac3, ponAbe2, panTro2, rheMac2, tarSyr1, micMur1, otoGar1, canFam2, monDom5
	</TD></TR>
<TR ALIGN=left><TD>Xeno mRNAs</TD><TD>papHam1, gorGor2</TD></TR>
</TABLE>
<B>Table 2.</B> <EM>Gene tracks used for codon translation.</EM>
</BLOCKQUOTE></P>

<H2>Methods</H2>
<P> 
Pairwise alignments with the marmoset genome were generated for 
each species using blastz from repeat-masked genomic sequence. 
Pairwise alignments were then linked into chains using a dynamic programming
algorithm that finds maximally scoring chains of gapless subsections
of the alignments organized in a kd-tree.
The scoring matrix and parameters for pairwise alignment and chaining
were tuned for each species based on phylogenetic distance from the reference.
High-scoring chains were then placed along the genome, with
gaps filled by lower-scoring chains, to produce an alignment net.
For more information about the chaining and netting process and 
parameters for each species, see the description pages for the Chain and Net 
tracks.</P>
<P>
An additional filtering step was introduced in the generation of the 9-way
conservation track to reduce the number of paralogs and pseudogenes from the 
high-quality assemblies and the suspect alignments from the low-quality 
assemblies:
the pairwise alignments of high-quality mammalian 
sequences (placental and marsupial) were filtered based on synteny; 
those for 2X mammalian genomes were filtered to retain only 
alignments of best quality in both the target and query (&quot;reciprocal 
best&quot;).</P>
<P>
The resulting best-in-genome pairwise alignments
were progressively aligned using multiz/autoMZ, 
following the tree topology diagrammed above, to produce multiple alignments.
The multiple alignments were post-processed to
add annotations indicating alignment gaps, genomic breaks,
and base quality of the component sequences.
The annotated multiple alignments, in MAF format, are available for
<A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/calJac3/multiz13way"
TARGET=_blank>bulk download</A>.
An alignment summary table containing an entry for each
alignment block in each species was generated to improve
track display performance at large scales.
Framing tables were constructed to enable
visualization of codons in the multiple alignment display.</P>
<P>
Conservation scoring was performed using the PhastCons package (A. Siepel),
which computes conservation based on a two-state phylogenetic hidden Markov
model (HMM).
PhastCons measurements rely on a tree model containing the tree topology,
branch lengths representing evolutionary distance at neutrally
evolving sites, the background distribution of nucleotides, and a substitution
rate matrix.  The 
<A HREF="http://hgdownload.cse.ucsc.edu/goldenPath/calJac3/phastCons13way/all.mod"
TARGET=_blank>vertebrate tree model</A> for this track was
generated using the phyloFit program from the phastCons package  
(REV model, EM algorithm, medium precision) using multiple alignments of 
4-fold degenerate sites extracted from the 28-way human(hg18) alignment
(msa_view).  The 4d sites were derived from the 
<A HREF="http://www.sanger.ac.uk/resources/databases/encode/" TARGET=_blank>Oct 2005 Gencode 
Reference Gene set</A>,
which was filtered to select single-coverage long transcripts.
The phastCons parameters used for the conservation measurement
were: expected-length=45, target-coverage=.3 and rho=.31</P>
<P>
The phastCons program computes conservation scores based on a phylo-HMM, a
type of probabilistic model that describes both the process of DNA
substitution at each site in a genome and the way this process changes from
one site to the next (Felsenstein and Churchill 1996, Yang 1995, Siepel and
Haussler 2005).  PhastCons uses a two-state phylo-HMM, with a state for
conserved regions and a state for non-conserved regions.  The value plotted
at each site is the posterior probability that the corresponding alignment
column was "generated" by the conserved state of the phylo-HMM.  These
scores reflect the phylogeny (including branch lengths) of the species in
question, a continuous-time Markov model of the nucleotide substitution
process, and a tendency for conservation levels to be autocorrelated along
the genome (i.e., to be similar at adjacent sites).  The general reversible
(REV) substitution model was used.  Unlike many conservation-scoring programs, 
note that phastCons does not rely on a sliding window
of fixed size; therefore, short highly-conserved regions and long moderately
conserved regions can both obtain high scores.  More information about
phastCons can be found in Siepel <EM>et al</EM>. 2005.</P> 
<P> 
PhastCons currently treats alignment gaps as missing data, which
sometimes has the effect of producing undesirably high conservation scores
in gappy regions of the alignment.  We are looking at several possible ways
of improving the handling of alignment gaps.</P>

<H2>Credits</H2>
<P> This track was created using the following programs:
<UL>
<LI> Alignment tools: blastz and multiz by Minmei Hou, Scott Schwartz and Webb 
Miller of the <A HREF="http://bio.cse.psu.edu/" 
TARGET=_blank>Penn State Bioinformatics Group</A>
<LI> Chaining and Netting:  axtChain, chainNet by Jim Kent at UCSC
<LI> Conservation scoring: PhastCons, phyloFit, tree_doctor, msa_view by 
<A HREF="http://compgen.bscb.cornell.edu/~acs/software.html" 
TARGET=_blank>Adam Siepel</A> while at UCSC, now at Cornell University
<LI> MAF Annotation tools: mafAddIRows by Brian Raney, UCSC; mafAddQRows
by Richard Burhans, Penn State; genePredToMafFrames by Mark Diekhans, UCSC
<LI> Tree image generator: phyloGif by Galt Barber, UCSC
<LI> Conservation track display: Kate Rosenbloom, Hiram Clawson (wiggle 
display), and Brian Raney (gap annotation and codon framing) at UCSC
</UL>
</P>
<P>The phylogenetic tree is based on Murphy <EM>et al</EM>. (2001) and general 
consensus in the vertebrate phylogeny community as of March 2007.
</P>

<H2>References</H2>

<H3>Phylo-HMMs and phastCons:</H3>
<P>
Felsenstein J, Churchill GA.
<A HREF="http://mbe.oxfordjournals.org/cgi/content/abstract/13/1/93"
TARGET=_blank>A Hidden Markov Model approach to
variation among sites in rate of evolution</A>.
<EM>Mol Biol Evol</EM>. 1996 Jan;13(1):93-104.
</P>

<P>
Siepel A, Haussler D.
<A HREF="http://compgen.bscb.cornell.edu/~acs/phylohmm.pdf"
TARGET=_blank>Phylogenetic Hidden Markov Models</A>.
In R. Nielsen, ed., <em>Statistical Methods in Molecular Evolution</em>,
pp. 325-351, Springer, New York (2005). </P>

<P>
Siepel A, Bejerano G, Pedersen JS, Hinrichs AS, Hou M, Rosenbloom K,
Clawson H, Spieth J, Hillier LW, Richards S, <EM>et al.</EM>
<A HREF="http://www.genome.org/cgi/doi/10.1101/gr.3715005"
TARGET=_blank>Evolutionarily conserved elements in vertebrate, insect, worm,
and yeast genomes</A>.
<EM>Genome Res</EM>. 2005 Aug;15(8):1034-50.
</P>

<P>
Yang Z.
<A HREF="http://www.genetics.org/cgi/content/abstract/139/2/993"
TARGET=_blank>A space-time process model for the evolution of DNA
sequences</A>.
<EM>Genetics</EM>. 1995 Feb;139(2):993-1005.
</P>

<H3>Chain/Net:</H3>
<P>
Kent WJ, Baertsch R, Hinrichs A, Miller W, Haussler D.
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1"
TARGET=_blank>Evolution's cauldron:
duplication, deletion, and rearrangement in the mouse and human genomes</A>.
<EM>Proc Natl Acad Sci U S A</EM>. 2003 Sep 30;100(20):11484-9.
</P>

<H3>Multiz:</H3>
<P>
Blanchette M, Kent WJ, Riemer C, Elnitski L, Smit AF, Roskin KM,
Baertsch R, Rosenbloom K, Clawson H, Green ED, <EM>et al.</EM>
<A HREF="http://www.genome.org/cgi/content/abstract/14/4/708"
TARGET=_blank>Aligning multiple genomic sequences with the threaded blockset aligner</A>.
<EM>Genome Res</EM>. 2004 Apr;14(4):708-15.
</P>

<H3>Blastz:</H3>
<P>
Chiaromonte F, Yap VB, Miller W.
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract"
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>.
<EM>Pac Symp Biocomput</EM>. 2002;:115-26.
</P>

<P>
Schwartz S, Kent WJ, Smit A, Zhang Z, Baertsch R, Hardison RC,
Haussler D, Miller W.
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103"
TARGET=_blank>Human-mouse alignments with BLASTZ</A>.
<EM>Genome Res</EM>. 2003 Jan;13(1):103-7.
</P>

<H3>Phylogenetic Tree:</H3>
<P>
Murphy WJ, Eizirik E, O'Brien SJ, Madsen O, Scally M, Douady CJ, Teeling E,
Ryder OA, Stanhope MJ, de Jong WW, Springer MS.
<A HREF="http://www.sciencemag.org/cgi/content/abstract/294/5550/2348"
TARGET=_blank>Resolution of the early placental mammal radiation using Bayesian phylogenetics</A>.
<EM>Science</EM>. 2001 Dec 14;294(5550):2348-51.
</P>
