<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=273>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://genome.wustl.edu/genomes/view/callithrix_jacchus/"
    TARGET=_blank>
    <IMG SRC="/images/Callithrix_jacchus.jpg" WIDTH=258 HEIGHT=200 ALT="Marmoset"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Callithrix jacchus</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of NHGRI
      (<A HREF="http://www.genome.gov/pressDisplay.cfm?photoID=82" 
      TARGET=_blank>press photos</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Mar. 2009 <em>Callithrix jacchus</em> draft assembly 
(WUGSC 3.2 (GCA_000004665.1)) was produced by the Washington 
University St. Louis (WUSTL) School of Medicine 
<A HREF="http://genome.wustl.edu/genomes/view/callithrix_jacchus/"
TARGET=_blank>Genome Sequencing Center</A> in St. Louis, MO, USA
and the Baylor College of Medicine (BCM)
<a href="http://www.hgsc.bcm.tmc.edu/"
target=_blank>Human Genome Sequencing Center</a>
in Houston, TX, USA.</P>


<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
Marmoset genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>chr6_GL285171_random</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays the non-localized contig GL285171</TD>
    </TR>
    <TR><TD VALIGN=Top>chr3:1-1000000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of chr 3</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr3:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr 3 that spans 2000 bases, starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>EF215130</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number EF215130</TD>
    </TR>
    <TR><TD VALIGN=Top>EH381683</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		EH381683 on chr 15</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>pseudogene</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists RefSeq or mRNA pseudogenes</TD>
    </TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>Datson</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Datson</TD>
    </TR>
    <TR><TD VALIGN=Top>Ouwendijk,W.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author W. Ouwendijk</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Ouwendijk W</em>, internally it
	uses the format <em>Ouwendijk,W.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
The <em>C. jacchus</em> genome was sequenced to 6X coverage 
using DNA from a female marmoset provided by the Southwestern
National Primate Research Center in San Antonio, TX, USA.
DNA from a full brother of the female was used as the source
for the CHORI-259 BAC library. The combined sequence reads were 
assembled using <A HREF="http://www.genome.org/cgi/content/full/13/9/2164"
TARGET=_blank>PCAP</A> and filtered for all known non-marmoset sequence 
contaminants and unscaffolded singleton contigs of less than 2 kb in length. 
</P> 
<P>
For more statistics and details on the assembly process, refer to the
<A HREF="http://genome.wustl.edu/pub/organism/Primates/Callithrix_jacchus/assembly/Callithrix_jacchus-3.2/ASSEMBLY"
TARGET=_blank>WUSTL Callithrix_jacchus-3.2 assembly information</A>.
There is no mitochondrial sequence available for the marmoset at this time.
</P>

<P>
Bulk downloads of the calJac3 sequence and annotations may 
be obtained from the Genome Browser 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/calJac3/">FTP server</A> or 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#marmoset">Downloads</A> 
page. The sequence data can also be obtained directly from 
<A HREF="ftp://ftp.ncbi.nlm.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Callithrix_jacchus/Callithrix_jacchus-3.2/"
TARGET=_blank>NCBI Genbank</A>. See the WUSTL 
<A HREF="http://genome.wustl.edu/data/data_use_policy"
TARGET=_blank>data use policy</A> for conditions of use.
Please acknowledge the WUSTL School of Medicine Genome 
Sequencing Center and the BCM Human Genome Sequencing Center
in any publications that result from the use of this sequence
assembly.</P>
<P>
We'd like to thank WUSTL and BCM for providing this assembly. The 
initial set of marmoset browser annotation tracks were 
generated by UCSC.  See the 
<A HREF="../goldenPath/credits.html#marmoset_credits">Credits</A>
page for a detailed list of the organizations and
individuals who contributed to this release.</P>
