<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
  <meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1">
  <meta name="description" content="Neandertal Genome Analysis Consortium Tracks at UCSC">
  <meta name="keywords" content="Neandertal, Neanderthal, genome, Neandertal genome, Neanderthal genome, genome browser, human genome, UCSC, bioinformatics, Max Planck Institute, Ed Green, Richard Green, Green RE, Svante Paabo">
  <title>Neandertal Genome Analysis Consortium Tracks at UCSC</title>
  <link rel="STYLESHEET" href="../style/HGStyle.css">
</HEAD>
<BODY style="background-color: rgb(255, 249, 210);" link="#0000cc"
 vlink="#330066" alink="#6600ff">
<A NAME="TOP"></a><!-- Header Images and links -->

<TABLE width="100%" border="0" cellpadding="0" cellspacing="0">
  <TBODY>
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../images/title.jpg" alt="Neandertal Genome Analysis Consortium Tracks at UCSC"></th>
    </TR>
    <TR><TD colspan="2" height="40"><!--#include virtual="../cgi-bin/hgMenubar"-->
    </TD></TR>

<!-- Left Sidebar Menu ======================================= -->  
    <TR HEIGHT="100%" VALIGN="top">
      <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
      <!-- outer table is to align leftbar & content boxes at top of page -->
        <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0>
          <TR><TD><IMG SRC="../images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
          <TR HEIGHT="100%"><TD HEIGHT="100%">
          <!--outer table is for black border-->
            <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
              <TR><TD>
                <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                  <TR HEIGHT="100%">
                    <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
                    <A CLASS="leftbar" HREF="../cgi-bin/hgGateway">
                        Genome Browser</A><HR>
                    <A CLASS="leftbar" HREF="/ebolaPortal/">
                        Ebola</A><HR>
                    <A CLASS="leftbar" HREF="../cgi-bin/hgBlat">
                        Blat</A><HR>
                    <A CLASS="leftbar" HREF="../cgi-bin/hgTables">
                        Table Browser</A><HR>
                    <A CLASS="leftbar" HREF="../cgi-bin/hgNear">
                        Gene Sorter</A><HR>
                    <A CLASS="leftbar" HREF="../cgi-bin/hgPcr">
                        In Silico PCR</A><HR>
                    <A CLASS="leftbar" HREF="../cgi-bin/hgGenome">
                        Genome Graphs</A><HR>
                    <A CLASS="leftbar" HREF="http://galaxy.psu.edu"
			TARGET=_blank>Galaxy</A><HR>
                    <A CLASS="leftbar" HREF="../cgi-bin/hgVisiGene">
                        VisiGene</A><HR>
                    <A CLASS="leftbar" HREF="../util.html">
                        Utilities</A><HR>
                    <A CLASS="leftbar" HREF="http://hgdownload.gi.ucsc.edu/downloads.html">
                        Downloads</A><HR>
                    <A CLASS="leftbar" HREF="../goldenPath/releaseLog.html">
                        Release Log</A><HR>
                    <A CLASS="leftbar" HREF="../goldenPath/customTracks/custTracks.html">
                        Custom Tracks</A><HR>
                    <A CLASS="leftbar" HREF="http://microbes.ucsc.edu/">
                        Microbial Genomes</A><HR>
                    <A CLASS="leftbar" HREF="../mirror.html">
                        Mirrors</A><HR>
                    <A CLASS="leftbar" HREF="../training/index.html" TARGET=_blank>
                        Training</A><HR>
                    <A CLASS="leftbar" HREF="../goldenPath/credits.html">
                        Credits</A><HR>
                    <A CLASS="leftbar" HREF="../goldenPath/pubs.html">
                        Publications</A><HR>
                    <A CLASS="leftbar" HREF="../cite.html">
                        Cite Us</A><HR>
                    <A CLASS="leftbar" HREF="../license/">
                        Licenses</A><HR>
                    <A CLASS="leftbar" HREF="https://ucscgenomics.soe.ucsc.edu/careers/" TARGET=_blank>
                        Jobs</A><HR>
                    <A CLASS="leftbar" HREF="../staff.html">
                        Staff</A><HR>
                    <A CLASS="leftbar" HREF="../contacts.html">
                        Contact Us</A><HR>
                    <BR>
                    <IMG SRC="../images/DOT.gif" ALT="" HEIGHT=700 WIDTH=1>
                  </TD></TR>
		</TABLE>
            </TD></TR>
	  </TABLE>
      </TD></TR>
    </TABLE>
  </TD>
<!--Content Tables--> 
  <TD valign="top">
    <TABLE border="0" cellpadding="5">
      <TBODY><TR><TD><!--outer table is for border purposes-->
            <TABLE width="100%" bgcolor="#888888" border="0" cellpadding="1" cellspacing="0">
              <TBODY><TR><TD>
                  <TABLE width="100%" bgcolor="#fffee8" border="0"
 cellpadding="0" cellspacing="0">
                    <TBODY><TR><TD>
                        <TABLE width="100%" background="../images/hr.gif" bgcolor="#d9e4f8" border="0">
                          <TBODY><TR>
			      <TD><A NAME="About"></A><FONT size="4"><B>&nbsp; Neandertal Genome Analysis Consortium Tracks at UCSC</B></FONT> 
			  </TD></TR></TBODY>
                        </TABLE>
                        <TABLE width="100%" bgcolor="#fffee8" border="0" cellpadding="0">
                          <TBODY><TR><TH height="3"><BR></TH>
                              <TD><BR></TD>
                            </TR>
                            <TR>
                              <TD width="10"><BR>
                              </TD>
                              <TD>
<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=250>
  <TR><TD ALIGN=RIGHT>
      <IMG SRC="Neandertal_reconstruction.jpg" HEIGHT=251 WIDTH=167 ALT="Artist's rendering of Neandertal">
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
      <FONT SIZE=-1>Artist's rendering of Neandertal man, from <BR> Neandertal museum in Mettmann, Germany<BR> 
      </FONT>
      <FONT SIZE=-2>Copyright: Johannes Krause, 
	<A HREF="http://www.eva.mpg.de/neandertal/" 
	TARGET=_BLANK>Max Planck Institute <BR>for Evolutionary 
	Anthropology</A>.  All rights reserved.
      <P>&nbsp;</P>
      </FONT>
  </TD></TR>
</TABLE>

<P>
Neandertals are the closest extinct relatives of humans. They lived from
several hundred thousand years ago until their disappearance approximately
30,000 years ago. The Neandertal genome sequence 
(published by Green <em>et al.</em> in <A HREF="http://www.sciencemag.org/special/neandertal/" 
TARGET=_BLANK><em>Science</em></A> May 2010) consists of short
sequence fragments, usually about 50 base pairs long, mapped to the
human reference genome. The DNA was extracted largely from three
Neandertal bones, each about 40,000 years old, from the Vindija Cave
in Croatia: Vi33.16, Vi33.25, and Vi33.26. The bulk sequencing was
carried out on the Illumina GAII platform. Neandertal DNA was
identified from among the background of microbial sequences in the
bone by similarity to the human or chimpanzee genomes.</P>
<P>
This portal provides access to the sequence data and alignments to the
reference human genome (NCBI 
Build 36/<A HREF="../cgi-bin/hgGateway?db=hg18">hg18</A>,
GRCh37/<A HREF="../cgi-bin/hgGateway?db=hg19">hg19</A>) as well as the reference
chimpanzee genome (CGSC 2.1/<A HREF="../cgi-bin/hgGateway?db=panTro2">panTro2</A>) and several associated analyses
(see <A HREF="#Downloads">Downloads</A> and <A HREF="#References">References</A>). 
</P>
<P>
The following annotations are available on the human hg18 and hg19 genome assemblies,
except as noted:
<UL>
<LI>
<A HREF="../cgi-bin/hgTrackUi?db=hg18&c=chr10&position=chr10:69713986-69714099&g=ntHumChimpCodingDiff">Neandertal 
Alleles in Human/Chimp Coding Non-synonymous Differences in Human Lineage</A>: 
Displays Neandertal alleles for human-chimp protein-coding differences 
on the human lineage using orangutan as the outgroup to determine which allele 
is more likely to be ancestral.
<LI>
<A HREF="../cgi-bin/hgTrackUi?db=hg18&c=chr2&position=chr2:43309001-43314000&g=ntSssZScorePMVar">Selective 
Sweep Scan (S) on Neandertal vs. Human Polymorphisms</A>: 
Shows the S score, an estimate of positive selection in 
humans within a 100 kb window surrounding each polymorphic position in the 
five modern human sequences and the human reference genome.
<LI>
<A HREF="../cgi-bin/hgTrackUi?db=hg18&c=chr2&position=chr2:43309001-43314000&g=ntSssTop5p">Selective 
Sweep Scan (S): 5% Smallest S scores</A>: Shows regions of the human 
genome with a strong signal for depletion of Neandertal-derived alleles (regions
from the Sel Swp Scan (S) track with S scores in the lowest 5%), which may 
indicate an episode of positive selection in early humans. 
<LI>
<A HREF="../cgi-bin/hgTrackUi?db=hg18&c=chr2&position=chr2:43309001-43314000&g=ntSssSnps">SNPS 
Used for Selective Sweep Scan (S)</A>: Shows single nucleotide polymorphisms 
(SNPs) on which the S score (above) was calculated.
<LI>
<A HREF="../cgi-bin/hgTrackUi?db=hg18&c=chr10&position=chr10:69713986-69714099&g=ntOoaHaplo">Candidate 
Regions for Gene Flow from Neandertal to Non-African Modern Humans</A>: 
Shows 13 regions of the human genome in which there is considerably more 
haplotype diversity among non-African genomes than within African genomes.
10 of these regions provide evidence for gene flow from Neandertals to 
modern humans outside of Africa.
<LI>
<A HREF="../cgi-bin/hgTrackUi?db=hg18&c=chr10&position=chr10:69713986-69714099&g=ntSeqContigs">Neandertal 
Sequence Contigs Generated by Genotype Caller</A>: Shows consensus 
contigs called from overlapping, non-redundant reads that passed mapping and 
base quality criteria.  Also available for panTro2 (chimpanzee).  Not available for hg19.
<LI>
<A HREF="../cgi-bin/hgTrackUi?db=hg18&c=chr10&position=chr10:69713986-69714099&g=ntSeqReads">Neandertal 
Sequence Reads</A>: Shows Neandertal sequence reads mapped to the human genome.
Also available for panTro2 (chimpanzee).
<LI>
<A HREF="../cgi-bin/hgTrackUi?db=hg18&c=chr10&position=chr10:69713986-69714099&g=ntModernHumans">Alignments 
of Sequence Reads from 5 Modern Humans</A>: Shows human sequence reads of
five individuals mapped to the human genome.  These data were used in the 
S score and Candidate Gene Flow analyses (above).  Not available for hg19.
<LI>
<A HREF="../cgi-bin/hgTrackUi?db=hg18&c=chrM&position=chrM&g=ntMito">Neandertal 
Mitochondrial Sequence</A>: Shows the alignment of a complete Neandertal 
mitochondrial sequence from a prior publication (Green <em>et al.</em>, 2008) 
to a modern human mitochondrial sequence.
</UL>

<!-- Display image in lefthand corner -->
<TABLE ALIGN=LEFT BORDER=0 WIDTH=315>
  <TR><TD ALIGN=LEFT>
      <IMG SRC="Krause_DSC_0675_small.JPG" HEIGHT=356 WIDTH=536 ALT="Vindija Cave, Croatia">
  </TD></TR>
  <TR><TD ALIGN=LEFT> 
      <FONT SIZE=-1>Entrance of the Vindija Cave, Croatia<BR> 
      </FONT>
      <FONT SIZE=-2> 
        Copyright: Johannes Krause, <A HREF="http://www.eva.mpg.de/neandertal/"
    	TARGET=_BLANK>Max Planck Institute for Evolutionary Anthropology</A>.
      	All rights reserved.<BR>
      </FONT>
  </TD></TR>
</TABLE>

<!-- Display righthand image -->
<TABLE BORDER=0 WIDTH=315>
  <TR><TD>
      <IMG SRC="Vindija_bones.jpg" HEIGHT=413 WIDTH=300 ALT="Bones from Vindija Cave">
  </TD></TR>
  <TR><TD> 
      <FONT SIZE=-1>The researchers obtained the majority of the DNA used for 
	their study
	from the bone fragments of three female Neandertals that were excavated
	in the Vindija Cave in Croatia.<BR> 
      </FONT>
      <FONT SIZE=-2> 
        Copyright: <A HREF="http://www.eva.mpg.de/neandertal/"
    	TARGET=_BLANK>Max Planck Institute for Evolutionary Anthropology</A>.
      	All rights reserved.<BR>
        <P></P>
      </FONT>
  </TD></TR>
</TABLE>
                              </TD>
                              <TD width="15"><BR></TD>
                          </TR></TBODY>
                        </TABLE>
                        <BR>
                  </TD></TR></TBODY>
                </TABLE>
          </TD></TR></TBODY>
        </TABLE>
        <BR>

<!--outer table is for border purposes-->
        <TABLE width="100%" bgcolor="#888888" border="0" cellpadding="1" cellspacing="0">
          <TBODY>
            <TR><TD>
                <TABLE width="100%" bgcolor="#fffee8" border="0" cellpadding="0" cellspacing="0">
                  <TBODY>
                    <TR><TD>
                        <TABLE width="100%" background="../images/hr.gif" bgcolor="#d9e4f8" border="0">
                          <TBODY>
                            <TR><TD><a name="Downloads"></a><font size="4"><b>&nbsp;
				Downloads</b></font> 
			    </TD></TR>
                          </TBODY>
                        </TABLE>
                        <TABLE width="100%" bgcolor="#fffee8" border="0" cellpadding="0">
                          <TBODY>
                            <TR><TH height="3"><BR></TH>
                              <TD><BR></TD>
                            </TR>
                            <TR><TD width="10"><BR></TD>
                              <TD>
                                <P>
Data files for tracks at UCSC can be downloaded at the locations below.
Files that end with .sql and .txt.gz are dumped from our MariaDB
database; .sql files name the data fields of each row, and .txt.gz
files contain all rows with tab-separated fields.  
All database dump files have a first column &quot;bin&quot; that can
be ignored (it is for database query speedup).  

Files that end in .bam and .bai contain compressed sequence alignments
and index files in the <A
HREF="http://samtools.sourceforge.net/SAM1.pdf" TARGET=_BLANK>BAM</A>
format.  
The free software packages <A HREF="http://samtools.sourceforge.net/"
TARGET=_BLANK>samtools</A> and 
<A HREF="http://picard.sourceforge.net/"
TARGET=_BLANK>PICARD</A> are useful for viewing and manipulating BAM 
files.

The .bw suffix denotes the 
<A HREF="../goldenPath/help/bigWig.html">bigWig</A> format, which stores 
a numeric signal for genomic positions.</P>
<P>
All links below can be fetched using 
<A HREF="http://rsync.samba.org/" TARGET=_BLANK>rsync</A>; replace 
&quot;ftp://&quot; with &quot;rsync://&quot; in the links below.
<UL>
<LI><B>Neandertal Alleles in Human/Chimp Coding Non-synonymous Differences in Human Lineage:</B><BR>
Human (hg18):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntHumChimpCodingDiff.sql"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntHumChimpCodingDiff.sql</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntHumChimpCodingDiff.txt.gz"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntHumChimpCodingDiff.txt.gz</A><BR>
Human (hg19):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntHumChimpCodingDiff.sql"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntHumChimpCodingDiff.sql</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntHumChimpCodingDiff.txt.gz"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntHumChimpCodingDiff.txt.gz</A><BR>
</LI>
<LI><B>Selective Sweep Scan (S) on Neandertal vs. Human Polymorphisms:</B><BR>
Human (hg18):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/bbi/ntSssZScorePMVar.bw"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/bbi/ntSssZScorePMVar.bw</A><BR>
Human (hg19):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/bbi/ntSssZScorePMVar.bw"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/bbi/ntSssZScorePMVar.bw</A><BR>
</LI>
<LI><B>Selective Sweep Scan (S): 5% Smallest S scores:</B><BR>
Human (hg18):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntSssTop5p.sql"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntSssTop5p.sql</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntSssTop5p.txt.gz"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntSssTop5p.txt.gz</A><BR>
Human (hg19):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntSssTop5p.sql"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntSssTop5p.sql</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntSssTop5p.txt.gz"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntSssTop5p.txt.gz</A><BR>
</LI>
<LI><B>SNPS Used for Selective Sweep Scan (S):</B><BR>
Human (hg18):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntSssSnps.sql"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntSssSnps.sql</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntSssSnps.txt.gz"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntSssSnps.txt.gz</A><BR>
Human (hg19):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntSssSnps.sql"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntSssSnps.sql</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntSssSnps.txt.gz"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntSssSnps.txt.gz</A><BR>
</LI>
<LI><B>Candidate Regions for Gene Flow from Neandertal to Non-African Modern Humans:</B><BR>
Human (hg18):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntOoaHaplo.sql"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntOoaHaplo.sql</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntOoaHaplo.txt.gz"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntOoaHaplo.txt.gz</A><BR>
Human (hg19):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntOoaHaplo.sql"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntOoaHaplo.sql</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntOoaHaplo.txt.gz"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntOoaHaplo.txt.gz</A><BR>
</LI>
<LI><B>Neandertal Sequence Contigs Generated by Genotype Caller:</B><BR>
Human (hg18):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/all-hg18.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/all-hg18.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/all-hg18.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/all-hg18.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Feld1-hg18.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Feld1-hg18.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Feld1-hg18.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Feld1-hg18.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Mez1-hg18.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Mez1-hg18.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Mez1-hg18.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Mez1-hg18.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Sid1253-hg18.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Sid1253-hg18.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Sid1253-hg18.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Sid1253-hg18.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Vi33.16-hg18.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Vi33.16-hg18.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Vi33.16-hg18.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Vi33.16-hg18.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Vi33.25-hg18.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Vi33.25-hg18.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Vi33.25-hg18.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Vi33.25-hg18.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Vi33.26-hg18.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Vi33.26-hg18.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Vi33.26-hg18.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/Vi33.26-hg18.bam.bai</A><BR>
Chimpanzee (panTro2):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/all-pt2.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/all-pt2.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/all-pt2.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/all-pt2.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Feld1-pt2.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Feld1-pt2.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Feld1-pt2.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Feld1-pt2.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Mez1-pt2.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Mez1-pt2.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Mez1-pt2.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Mez1-pt2.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Sid1253-pt2.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Sid1253-pt2.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Sid1253-pt2.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Sid1253-pt2.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Vi33.16-pt2.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Vi33.16-pt2.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Vi33.16-pt2.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Vi33.16-pt2.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Vi33.25-pt2.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Vi33.25-pt2.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Vi33.25-pt2.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Vi33.25-pt2.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Vi33.26-pt2.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Vi33.26-pt2.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Vi33.26-pt2.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/Vi33.26-pt2.bam.bai</A><BR>
</LI>
<LI><B>Neandertal Sequence Reads:</B><BR>
Human (hg18):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLFeld1.hg18.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLFeld1.hg18.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLFeld1.hg18.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLFeld1.hg18.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLMez1.hg18.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLMez1.hg18.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLMez1.hg18.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLMez1.hg18.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLSid1253.hg18.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLSid1253.hg18.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLSid1253.hg18.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLSid1253.hg18.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLVi33.16.hg18.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLVi33.16.hg18.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLVi33.16.hg18.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLVi33.16.hg18.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLVi33.25.hg18.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLVi33.25.hg18.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLVi33.25.hg18.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLVi33.25.hg18.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLVi33.26.hg18.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLVi33.26.hg18.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLVi33.26.hg18.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/seqAlis/SLVi33.26.hg18.bam.bai</A><BR>
Human (hg19):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLFeld1.hg19.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLFeld1.hg19.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLFeld1.hg19.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLFeld1.hg19.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLMez1.hg19.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLMez1.hg19.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLMez1.hg19.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLMez1.hg19.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLSid1253.hg19.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLSid1253.hg19.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLSid1253.hg19.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLSid1253.hg19.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLVi33.16.hg19.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLVi33.16.hg19.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLVi33.16.hg19.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLVi33.16.hg19.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLVi33.25.hg19.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLVi33.25.hg19.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLVi33.25.hg19.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLVi33.25.hg19.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLVi33.26.hg19.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLVi33.26.hg19.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLVi33.26.hg19.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/seqAlis/SLVi33.26.hg19.bam.bai</A><BR>
Chimpanzee (panTro2):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLFeld1.pt2.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLFeld1.pt2.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLFeld1.pt2.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLFeld1.pt2.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLMez1.pt2.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLMez1.pt2.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLMez1.pt2.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLMez1.pt2.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLSid1253.pt2.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLSid1253.pt2.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLSid1253.pt2.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLSid1253.pt2.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLVi33.16.pt2.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLVi33.16.pt2.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLVi33.16.pt2.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLVi33.16.pt2.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLVi33.25.pt2.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLVi33.25.pt2.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLVi33.25.pt2.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLVi33.25.pt2.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLVi33.26.pt2.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLVi33.26.pt2.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLVi33.26.pt2.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/panTro2/neandertal/seqAlis/SLVi33.26.pt2.bam.bai</A><BR>
</LI>
<LI><B>Alignments of Sequence Reads from 5 Modern Humans:</B><BR>
Human (hg18):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS4_HGDP00778_Han.paired.qualfilt.rmdup.entropy1.0.sort.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS4_HGDP00778_Han.paired.qualfilt.rmdup.entropy1.0.sort.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS4_HGDP00778_Han.paired.qualfilt.rmdup.entropy1.0.sort.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS4_HGDP00778_Han.paired.qualfilt.rmdup.entropy1.0.sort.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS5_HGDP00542_Papuan.paired.qualfilt.rmdup.entropy1.0.sort.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS5_HGDP00542_Papuan.paired.qualfilt.rmdup.entropy1.0.sort.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS5_HGDP00542_Papuan.paired.qualfilt.rmdup.entropy1.0.sort.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS5_HGDP00542_Papuan.paired.qualfilt.rmdup.entropy1.0.sort.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS6_HGDP00927_Yoruba.paired.qualfilt.rmdup.entropy1.0.sort.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS6_HGDP00927_Yoruba.paired.qualfilt.rmdup.entropy1.0.sort.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS6_HGDP00927_Yoruba.paired.qualfilt.rmdup.entropy1.0.sort.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS6_HGDP00927_Yoruba.paired.qualfilt.rmdup.entropy1.0.sort.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS7_HGDP01029_San.paired.qualfilt.rmdup.entropy1.0.sort.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS7_HGDP01029_San.paired.qualfilt.rmdup.entropy1.0.sort.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS7_HGDP01029_San.paired.qualfilt.rmdup.entropy1.0.sort.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS7_HGDP01029_San.paired.qualfilt.rmdup.entropy1.0.sort.bam.bai</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS8_HGDP00521_French.paired.qualfilt.rmdup.entropy1.0.sort.bam"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS8_HGDP00521_French.paired.qualfilt.rmdup.entropy1.0.sort.bam</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS8_HGDP00521_French.paired.qualfilt.rmdup.entropy1.0.sort.bam.bai"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/humans/MMS8_HGDP00521_French.paired.qualfilt.rmdup.entropy1.0.sort.bam.bai</A><BR>
</LI>
<LI><B>Neandertal Mitochondrial Sequence (2008):</B><BR>
Human (hg18):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntMito.sql"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntMito.sql</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntMito.txt.gz"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg18/database/ntMito.txt.gz</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/ntMito/ntM.fa"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg18/neandertal/ntMito/ntM.fa</A><BR>
Human (hg19):<BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntMito.sql"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntMito.sql</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntMito.txt.gz"
>ftp://hgdownload.gi.ucsc.edu/goldenPath/hg19/database/ntMito.txt.gz</A><BR>
<A HREF="ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/ntMito/ntM.fa"
>ftp://hgdownload.gi.ucsc.edu/gbdb/hg19/neandertal/ntMito/ntM.fa</A><BR>
</LI>
</UL>
</P>
                              </TD>
                              <TD width="15"><BR></TD>
                          </TR></TBODY>
                        </TABLE>
                        <BR>
                  </TD></TR></TBODY>
                </TABLE>
          </TD></TR></TBODY>
        </TABLE>
      <BR>

<!--outer table is for border purposes-->
        <TABLE width="100%" bgcolor="#888888" border="0" cellpadding="1" cellspacing="0">
          <TBODY>
            <TR><TD>
                <TABLE width="100%" bgcolor="#fffee8" border="0" cellpadding="0" cellspacing="0">
                  <TBODY>
                    <TR><TD>
                        <TABLE width="100%" background="../images/hr.gif" bgcolor="#d9e4f8" border="0">
                          <TBODY>
                            <TR><TD><a name="Credits"></a><font size="4"><b>&nbsp;
				Credits</b></font> 
			    </TD></TR>
                          </TBODY>
                        </TABLE>
                        <TABLE width="100%" bgcolor="#fffee8" border="0" cellpadding="0">
                          <TBODY>
                            <TR><TH height="3"><BR></TH>
                              <TD><BR></TD>
                            </TR>
                            <TR><TD width="10"><BR></TD>
                              <TD>
                                <P>
				The Neandertal Genome Project was carried out 
				at the 
				<A HREF="http://www.eva.mpg.de/neandertal/" 
				TARGET=_BLANK>Max Planck Institute for 
				Evolutionary Anthropology</A> under the 
				direction of Svante P&auml;&auml;bo and made 
				possible by a grant from the Presidential 
				Innovation Fund of the Max Planck Society.</P>
                              </TD>
                              <TD width="15"><BR></TD>
                          </TR></TBODY>
                        </TABLE>
                        <BR>
                  </TD></TR></TBODY>
                </TABLE>
          </TD></TR></TBODY>
        </TABLE>
      <BR>

<!--outer table is for border purposes-->
        <TABLE width="100%" bgcolor="#888888" border="0" cellpadding="1" cellspacing="0">
          <TBODY>
            <TR><TD>
                <TABLE width="100%" bgcolor="#fffee8" border="0" cellpadding="0" cellspacing="0">
                  <TBODY>
                    <TR><TD>
                        <TABLE width="100%" background="../images/hr.gif" bgcolor="#d9e4f8" border="0">
                          <TBODY>
                            <TR><TD><a name="References"></a><font size="4"><b>&nbsp;
				References</b></font> 
			    </TD></TR>
                          </TBODY>
                        </TABLE>
                        <TABLE width="100%" bgcolor="#fffee8" border="0" cellpadding="0">
                          <TBODY>
                            <TR><TH height="3"><BR></TH>
                              <TD><BR></TD>
                            </TR>
                            <TR><TD width="10"><BR></TD>
                              <TD>
<H3>Neandertal Genome Analysis Consortium (2010):</H3>
<P>
Green RE, Krause J, Briggs AW, Maricic T, Stenzel U, Kircher M, Patterson N,
Li H, Zhai W, Fritz MH  <em>et al.</em>
<A HREF="http://www.sciencemag.org/cgi/content/full/328/5979/710"
TARGET=_blank>A draft sequence of the Neandertal genome</A>.
<em>Science</em>. 2010 7 May;328(5979):710-22.
</P>
<P>
Burbano HA, Hodges E, Green RE, Briggs AW, Krause J, Meyer M, Good JM, 
Maricic T, Johnson PLF, Xuan Z <em>et al.</em>
<A HREF="http://www.sciencemag.org/cgi/content/full/328/5979/723"
TARGET=_BLANK>Targeted investigation of the Neandertal genome by array-based 
sequence capture</A>.
<em>Science</em>. 2010 7 May;328(5979):723-5.
</P>
                              
<H3>Previously published mitochondrial genome of Vi33.16 (2008):</H3>
<P>
Green RE, Malaspinas AS, Krause J, Briggs AW, Johnson PL, Uhler C, Meyer M, 
Good JM, Maricic T, Stenzel U, <em>et al.</em>
<A HREF="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2602844/?tool=pubmed" TARGET=BLANK>A complete 
Neandertal mitochondrial genome sequence determined by high-throughput 
sequencing.</A>
<em>Cell.</em> 2008 Aug 8;134(3):416-26.
</P>
                              </TD>
                              <TD width="15"><BR></TD>
                          </TR></TBODY>
                        </TABLE>
                        <BR>
                  </TD></TR></TBODY>
                </TABLE>
          </TD></TR></TBODY>
        </TABLE>
      <BR>
<!--outer table is for border purposes-->
        <TABLE width="100%" bgcolor="#888888" border="0" cellpadding="1" cellspacing="0">
          <TBODY>
            <TR><TD>
                <TABLE width="100%" bgcolor="#fffee8" border="0" cellpadding="0" cellspacing="0">
                  <TBODY>
                    <TR><TD>
                        <TABLE width="100%" background="../images/hr.gif" bgcolor="#d9e4f8" border="0">
                          <TBODY>
                            <TR><TD><a name="Conditions"></a><font size="4"><b>&nbsp;
				Conditions of Use</b></font> 
			    </TD></TR>
                          </TBODY>
                        </TABLE>
                        <TABLE width="100%" bgcolor="#fffee8" border="0" cellpadding="0">
                          <TBODY>
                            <TR><TH height="3"><BR></TH>
                              <TD><BR></TD>
                            </TR>
                            <TR><TD width="10"><BR></TD>
                              <TD>
                                <P>
				The sequence and annotation data displayed in 
			  	the Genome Browser are freely available for 
				academic, nonprofit, and personal use (see 
				<A HREF="../#Conditions">Conditions of Use</A>).
				</P>
				<P></P>
                              </TD>
                              <TD width="15"><BR></TD>
                          </TR></TBODY>
                        </TABLE>
                        <BR>
                  </TD></TR></TBODY>
                </TABLE>
          </TD></TR></TBODY>
        </TABLE>
      <BR>
 </TD></TR></TBODY>
</TABLE>
</TD></TR></TBODY>
</TABLE>
</BODY>
</HTML>
