<H2>Description</H2>
<P>
This track shows predictions of
<A HREF="http://augustus.gobics.de/" TARGET=_blank>AUGUSTUS</A>.
TransMap refseqs from human, mouse, rat, dog and chimp as well as 
rhesus ESTs were used as sources of hints.

<H2>Display Conventions and Configuration</H2>
<P>
This track follows the display conventions for 
<A HREF="../goldenPath/help/hgTracksHelp.html#GeneDisplay">gene prediction 
tracks</A>.</P>
<P>
This track contains an optional codon coloring 
feature that allows users to quickly validate and compare gene predictions.
To display codon colors, select the <em>genomic codons</em> option from the
<em>Color track by codons</em> pull-down menu. Click the
<A HREF="../goldenPath/help/hgCodonColoring.html">Help on codon coloring</A> 
link for more information about this feature.</P>

<H2>Methods</H2>
<P>
Augustus uses a generalized hidden Markov model (GHMM) that models coding and 
non-coding sequence, splice sites, the branch point region, translation start 
and end, and lengths of exons and introns. This version has been trained on a 
set of 1284 human genes. </P>

<H2>Credits</H2>
<P>
This track was created by Mario Stanke.</P>

<H2>References</H2>
<P>
Stanke, M.
<A HREF="http://webdoc.sub.gwdg.de/diss/2004/stanke/index.html" 
TARGET=_blank>Gene prediction with a hidden Markov model</A>.
<em>Ph.D. thesis</em>, Universit&auml;t G&ouml;ttingen, Germany (2004).</P>
<P>
Stanke, M., Steinkamp, R., Waack, S. and Morgenstern, B.
<A HREF="http://nar.oupjournals.org/cgi/content/full/32/suppl_2/W309" 
TARGET=_blank>AUGUSTUS: a web server for gene finding in eukaryotes</A>.
<em>Nucl. Acids Res.</em> <B>32</B>, W309-12 (2004).</P>
<P>
Stanke, M. and Waack, S. 
<A HREF="http://bioinformatics.oupjournals.org/cgi/content/abstract/19/suppl_2/ii215" 
TARGET=_blank>Gene prediction with a hidden Markov model and a new intron 
submodel</A>.
<em>Bioinformatics</em> <B>19</B>(Suppl. 2), ii215-25 (2003).</P>
