<!-- Display image at top right of page -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=315>
  <TR><TD ALIGN=RIGHT><A HREF="http://www.hgsc.bcm.tmc.edu/projects/rmacaque/" TARGET=_blank>
      <IMG SRC="/images/Rhesus_macaque.jpg" WIDTH=300 HEIGHT=225 ALT="Rhesus Macaque"></A>
    </TD></TR>
  <TR><TD ALIGN=RIGHT> 
  <FONT SIZE=-1> <em>Macaca mulatta</em><BR> 
  </FONT>
  <FONT SIZE=-2>
    Photo courtesy of 
    <A HREF = "http://www1.wfubmc.edu/pathresearch/faculty/kaplan.htm" 
    TARGET = _BLANK>J.R. Kaplan</A><BR>
</FONT>
  </TD></TR>
</TABLE>

<P>
The Jan. 2006 rhesus macaque (<em>Macaca mulatta</em>) 
draft assembly &mdash; v.1.0, Mmul_051212 &mdash; was obtained from the Baylor 
College of Medicine Human Genome Sequencing Center (BCM HGSC). The sequencing 
and assembly of the rhesus macaque genome is a project of the Macaque Genome 
Sequencing Consortium led by the BCM HGSC, in collaboration with the Genome 
Sequencing Center at Washington University School of Medicine in St. Louis and 
the J. Craig Venter Institute Joint Technology Center.</P>

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced mRNA or EST, a scaffold coordinate range, a gene identifier,
or keywords from the GenBank description of an mRNA.
The following list shows examples of valid position queries for
the rhesus genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
TARGET=_blank>User's Guide</A> for more information. 
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top><B>Genome Browser Response:</B><BR></TD>
    </TR>
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chr19</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of chromosome 19</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr19:1-100000</TD>
       	<TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays the first 100,000 bases of chromosome 19</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr19:100000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr 19 that spans 2000 bases, starting with position 100000 </TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>

    <TR><TD VALIGN=Top NOWRAP>CO582753</TD>
       	<TD WIDTH=14></TD>
      	<TD VALIGN=Top>Displays EST with GenBank accession CO582753</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>BT019700</TD>
       	<TD WIDTH=14></TD>
      	<TD VALIGN=Top>Displays human mRNA with GenBank accession BT019700 aligning to rheMac2 assembly</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>PRNP</TD>
    	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists PRNP genes in other species aligned to rheMac2 chromosome 10</TD>
    </TR>

    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>leptin</TD>
    	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs associated with leptin receptors </TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
    	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>kruppel zinc finger</TD>
    	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>Scinicariello,F.</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author F. Scinicariello</TD>
    </TR>
    <TR><TD VALIGN=Top><br></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Scinicariello F</em>, internally it uses 
			the format <em>Scinicariello,F.</em>.</TD>
    </TR>
</TABLE>
<P>
<HR>
<H3>Assembly details</H3>
<P>
The groups in the Macaque Genome Sequencing Consortium carried out 
preliminary assemblies of the genome data using different and complementary 
approaches. The resulting data were combined into a single, high-density 
&quot;melded&quot; assembly by a team at J. Craig Venter Institute. 
This collaboration made use of published macaque maps (Rogers, 2006; Murphy, 
2005), the BAC fingerprint map from the Michael Smith Genome Sciences Centre, 
and the human genome reference sequence.</P>
<P>
Several whole genome shotgun (WGS) libraries, with inserts of 2-4 kb and 10 kb, 
fosmids with ~35 kb inserts, and BACs with 180 kb inserts were used to produce 
the clone-end sequence data. Approximately 20.1 million reads were used in the 
assembly, representing about 14.9 Gb of sequence and about 5.1X coverage of the 
(clonable) macaque genome. 
The assemblers produced a set of contigs and scaffolds. Scaffolds include 
sequence contigs that could be ordered and oriented with respect to each other 
as well as isolated contigs that could not be linked (single contig scaffolds 
or singletons). The N50 of the contigs is 25.7 kb and the N50 of the scaffolds 
is 5.87 Mb. (The N50 size is the length such that 50% of the assembled genome 
lies in blocks of the N50 size or longer.)
The total length of all contigs is 2.87 Gb. When the gaps between
contigs in scaffolds are included, the total span of the assembly is
3.01 Gb.</P> <!--
<P>
This is a draft sequence and may contain errors; therefore, users should 
exercise caution. Typical errors in draft genome sequences include
misassemblies of repeated sequences, collapses of repeated regions,
and unmerged overlaps (e.g. due to polymorphisms) creating artificial
duplications. However, base accuracy in contigs is usually very high with 
most errors near the ends of contigs. </P>
-->
<P>
For general information about the rhesus monkey sequencing project, see the 
Baylor
<A HREF="http://www.hgsc.bcm.tmc.edu/projects/rmacaque/"
TARGET=_blank>Rhesus Monkey Project</A> web page.</P>
<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/rheMac2/">FTP
server</A> or the 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#rhesus">Downloads</A> 
page. These data have specific 
<A HREF="../goldenPath/credits/#rhesus_use" TARGET=_blank>conditions for 
use</A>. The rheMac2 annotation tracks were 
generated by UCSC and collaborators worldwide. See the 
<A HREF="../goldenPath/credits.html#rhesus_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to this release. </P>
