<H2>Description</H2>
<P>
This track shows the single nucleotide polymorphisms (SNPs) and/or
insertions/deletions (indels) determined by the 
<A HREF="http://www.genomics.org.cn/bgi_new/english/index.htm" TARGET=_BLANK>
Beijing Genomics Institute</A> (BGI).  
Genomic sequence reads from 3 alternate strains of chicken (Broiler,
Layer and Silkie) were used to determine the SNPs relative to the
reference assembly for Red Jungle Fowl (RJF).
</P>
<P>
SNP names are of the format 
snp.[<em>supercontig</em>].[<em>contig</em>].[<em>position</em>].[<em>type</em>].[<em>strain</em>]:
</P>
<UL>
<LI><B><em>supercontig</em></B>: Supercontig number from assembly</LI>
<LI><B><em>contig</em></B>: Contig number (within supercontig) from assembly</LI>
<LI><B><em>position</em></B>: Starting offset within Contig.[<em>supercontig</em>].[<em>contig</em>]</LI>
<LI><B><em>type</em></B>: S=Substitution, I=Insertion or D=Deletion</LI>
<LI><B><em>strain</em></B>: 1=Broiler, 2=Layer, 3=Silkie</LI>
</UL>

<H2>Methods</H2>
<P>
Sequence reads from the Broiler, Layer and Silkie strains were mapped
to the reference chicken assembly with blast. Then, cross_match was
used for the detailed base-by-base comparison.  The Phred quality
scores (-10 * log <SUB> 10</SUB> estimated error probability) are given for the 
SNP base(s) in the assembly and the (single read) SNP read.  
</P>

<H2>Credits</H2>
<P>Thanks to the 
<A HREF="http://www.genomics.org.cn/bgi_new/english/index.htm" TARGET=_BLANK>
Beijing Genomics Institute</A> for providing these data.
</P>

<H2>Terms of Use</H2>
<OL>
<LI>
Users are free to use these data in scientific papers analyzing
particular genes, if the Beijing Genomics Institute (BGI) is
acknowledged.
</LI>
<LI>
BGI and its collaborators reserve the right to publish the initial
analyses of these data, including but not restricted to the large-scale
identification of functional polymorphisms, evolutionary patterns and
signs of selection, correlations with known QTLs, utility of data for
genetic mapping purposes, etc.
</LI>
<LI>
Any redistribution of these data should carry this notice.
</LI>
</OL>

