<!DOCTYPE html>
<!--#set var="TITLE" value="Genome Browser FAQ" -->
<!--#set var="ROOT" value=".." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<h1>Frequently Asked Questions: Custom Annotation Tracks</h1>

<h2>Topics</h2>

<ul>
<li><a href="#custom1">Displaying personal annotation data in the Genome Browser</a></li>
<li><a href="#custom2">Adding a personal annotation track to the Genome Browser website</a></li>
<li><a href="#custom3">Defining filter parameters for custom tracks</a></li>
<li><a href="#custom4">Coloring a custom track using the useScore parameter</a></li>
<li><a href="#custom5">Constructing a Genome Browser URL</a></li>
<li><a href="#custom6">Using the hgsid parameter in a URL</a></li>
<li><a href="#custom7">Creating a details page for a custom annotation track</a></li>
<li><a href="#custom8">Custom annotation track troubleshooting</a></li>
<li><a href="#custom9">Error 500: Internal server error</a></li>
<li><a href="#custom10">Byte-range request error</a></li>
<li><a href="#custom11">BedToBigBed and other utilities fail because of custom track header 
<li><a href="#custom12">Non-UCSC chromosome names</a></li>
<li><a href="#custom13">Apostrophes and special characters in track labels</a></li>
</ul>
<hr>
<p>
<a href="index.html">Return to FAQ Table of Contents</a> </p>

<a name="custom1"></a>
<h2>Displaying personal annotation data in the Genome Browser</h2>
<p>
<h6>How do I display my own personal annotation data in the Genome Browser?</h6>
<p> 
To create an annotation track that will display on the Genome Browser, you must first organize your 
data into a format supported by the browser custom track feature: 
<a href="FAQformat.html#format4">GTF</a>, 
<a href="FAQformat.html#format3">GFF</a>, 
<a href="FAQformat.html#format1">BED</a>, 
<a href="FAQformat.html#format6">WIG</a>, 
or <a href="FAQformat.html#format2">PSL</a>. 
Then, upload your data into the Genome Browser on the 
<a href="../cgi-bin/hgCustom">Add Custom Tracks</a> page. Once you've created your annotation track,
you can share it with others over the internet by putting your annotation file on your website
and then creating a custom URL that allows others to directly start the browser with your track 
displayed.</p>
<p>
Read the <a href="../goldenPath/help/hgTracksHelp.html#CustomTracks">Creating custom annotation 
tracks</a> section in the Genome Browser User's Guide for a step-by-step description of how to 
format and display a custom annotation track and create a custom URL.</p>

<a name="custom2"></a>
<h2>Adding a personal annotation track to the Genome Browser website</h2>
<h6>I have an annotation track that I think might be of general interest to the research 
community. Will UCSC consider including my track in their browser?</h6>
<p> 
We are always interested in receiving new annotation tracks for the Genome Browser, and encourage 
our users to share their tracks with us and others in the research community. Please send the URL 
for your track -- along with a description of the methods, data, and format used -- to 
<a href="mailto:&#103;&#101;&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;sc.e&#100;&#117;">&#103;&#101;&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;sc.e&#100;&#117;</a>. 
<!-- above address is genome at soe.ucsc.edu -->
If we determine that your track is of sufficient general interest to distribute as part of our 
browser, we'll work with you to include it.</p> 
<p>	
In addition to the standard set of tracks displayed on the Genome Browser page, we also have a 
<a href="../goldenPath/customTracks/custTracks.html">Custom Annotation Tracks</a> page that 
contains contributed tracks we are unable to display in the main browser (for example, tracks that 
are of too specific an interest or are too sparse). We welcome contributions to this page.</p>

<a name="custom3"></a>
<h2>Defining filter parameters for custom tracks</h2>
<h6>Is it possible to define a filter parameter for a custom track to highlight certain
features?</h6>
<p>
This feature is not currently implemented.</p>

<a name="custom4"></a>
<h2>Coloring a custom track using the useScore parameter</h2>
<!--FIXME - add itemRGB stuff -->

<h6>When designing a custom track, is there a way to assign specific colors to each segment, as 
is done in the mouse/rat synteny tracks? Is there a way to assign a value range for the 
<em>useScore</em> variable such that I can have four shades and specify the value range for
each?</h6>
<p>
Currently <em>useScore</em> works only with tracks that are black or specific shades of brown or 
blue. The score range is 0-1000. To display four shades, use the scores 0, 333, 666, and 1000.</p>

<a name="custom5"></a>
<h2>Constructing a Genome Browser URL</h2>
<h6>How can I construct a URL to retrieve data from the Genome Browser? What do the various 
parameters in the Genome Browser URLs mean?</h6>
<p> 
One way to determine how to construct a correct URL is to open a Genome Browser link in which you 
are interested and examine how the Genome Browser constructs the URL. See the 
<a href="../goldenPath/help/hgTracksHelp.html#CustomTracks">User's Guide</a> for a discussion of the
basic components of a Genome Browser URL. Note that the <em>c</em> parameter that appears in some 
URLs specifies the chromosome name or the chromosome name and position.</p>

<a name="custom6"></a>
<h2>Using the hgsid parameter in a URL</h2>
<h6>Should I use the hgsid parameter in my URL?</h6>
<p>
Avoid using hgsid -- it is a temporary identifier, and will typically stop working after a day.</p>

<a name="custom7"></a>
<h2>Creating a details page for a custom annotation track</h2>
<h6>While working on a custom track, we noticed that the feature details page for custom track 
looks different from regular tracks on the site. Is the details page for a custom track
customizable?</h6>
<p> 
You can add a link from a details page to an external web page containing additional information 
about the feature by using the track line url attribute. In the annotation file, set the url 
attribute in the track line to point to a publicly available page on a web server. The url 
attribute substitutes each occurrence of '$$' in the URL string with the name defined by the name 
attribute. You can take advantage of this feature to provide individualized information for each 
feature in your track by creating HTML anchors that correspond to the feature names in your web 
page.</p>

<a name="custom8"></a>
<h2>Custom annotation track troubleshooting</h2>
<h6>When I click the Submit button, I get the error message &quot;line # of custom input: BED 
chromStarts[i] must be in ascending order&quot;.</h6>
<p> 
This error is caused by a logical conflict in the Genome Browser software. It accepts custom GFF 
tracks that have multiple &quot;exons&quot; at the same position, but not BED tracks that have this 
feature. Because the browser translates GFF tracks to BED format before storing the custom track 
data, GFF tracks with multiple exons will cause an error when the BED is read back in. To work 
around this problem, remove duplicate lines in the GFF track.</p>

<a name="custom9"></a>
<h2>Error 500: Internal server error</h2>
<h6>I'm getting an Error 500 when trying to upload my custom track(s). What am I doing
wrong?</h6>
<p> 
This error may occur when trying to upload a file that is too large. For larger data sets, we 
suggest using bigBed and bigWig file formats. More information about selecting a format can be found
<a href="http://genomewiki.ucsc.edu/index.php/Selecting_a_graphing_track_data_format">here</a>.</p>

<a name="custom10"></a>
<h2>Byte-range request error</h2>
<h6>When I try to visualize my custom track, I receive the error 'Byte-range request was ignored
by server.</h6>
<p> 
This error occurs when a web server is not configured properly or does not support byte-ranges. 
Click <a href="../goldenPath/help/hgTracksHelp.html#BYTERANGE">here</a> for more information.</p>

<a name="custom11"></a>
<h2>BedToBigBed and other utilities fail because of custom track header lines</h2>
<h6>Why does my custom track fail when I use <code>bedToBigBed</code> or utilities like 
<code>validateFiles</code>?</h6>
<p>
These utilities can fail when the input includes custom-track-specific lines at the top of the file 
that are not considered part of the data. Many of the custom track examples on the 
<a href="FAQformat.html" target="_blank">File Format</a> page include a &quot;<code>track 
type=...</code>&quot; line that is specific for loading the data into the Browser. This line will 
cause raw data files to fail validation by other tools, such as <code>bedToBigBed</code> or 
<code>validateFiles</code>, outside of the Browser. To see an example of using 
<code>bedToBigBed</code> with correct input data types, follow this 
<a href="../goldenPath/help/bigBed.html#Ex3" target="_blank">link</a>. More information about track 
lines can be found <a href="../goldenPath/help/customTrack.html#TRACK" target="_blank">here</a>.</p>

<a name="custom12"></a>
<h2>Non-UCSC style chromosome names</h2>
<h6>Can I upload a custom track with chromosome names (e.g. '1' or 'NC_000001.11') that don't
match the UCSC-style chromosome names (e.g. 'chr1', 'chrX')?</h6>
<p>
As of 2018, you can upload a custom track that contains either
UCSC-style chromosome names (e.g. 'chr1', 'chrX') or one of several supported
aliases, typically from Ensembl (e.g. '1' or 'X'), GenBank (e.g. 'CM000663.2'
or 'CM000685.2'), or RefSeq (e.g. 'NC_000001.11' or 'NC_000023.11'). Many
different custom track types support this, including BED, BAM, and any of
our big* formats like bigBed or bigWig. You can view the list of supported
aliases by clicking on the &quot;View sequences&quot; link for your assembly of
interest from the <a href="../cgi-bin/hgGateway">Assembly Gateway page</a> or
as a &quot;chromAlias.txt&quot; file from <a
href="https://hgdownload.soe.ucsc.edu/downloads.html" >our downloads server</a>
under the link &quot;Genome sequence files and select annotations&quot; for that assembly.

<a name="custom13"></a>
<h2>Apostrophes and special characters in track labels</h2>
<h6>How can I put an apostrophe in my track label?</h6>
<p>
The Genome Browser supports a limited set of characters in the track image. In
addition to the characters <code>a-z</code> and <code>A-Z</code>, it supports
some accented characters such as <code>&eacute;</code> or <code>&aacute;</code> and apostrophes,
specifically "<code>&#697;</code>" (which corresponds to Unicode character U+2019).
If there are characters in your track's short or long label that we don't
currently support, <a href="../contacts.html">contact us</a>.

<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->
