<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
        <META name="description" content="UCSC Genome Browser FAQ">
        <META name="keywords" content="genome, genome browser, genome assembly, UCSC, bioinformatics, sequence, gene prediction, SNP, EST, mRNA">
	<TITLE>UCSC Genome Bioinformatics: FAQ</TITLE>
	<LINK REL="STYLESHEET" HREF="/style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<!--Hotlinks bar---------------------------------------------------->
<TR><TD COLSPAN=2 HEIGHT=40>
    <TABLE bgcolor="000000" cellpadding="1" cellspacing="1" width="100%" height="27"><TR bgcolor="2636D1"><TD valign="middle">
        <TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 bgcolor="2636D1" height="24">
            <TR><TD VALIGN="middle"><font color="#89A1DE">&nbsp;
                &nbsp;<A HREF="/index.html" class="topbar">
                        Home</a> &nbsp; - &nbsp;
                <A HREF="/cgi-bin/hgGateway" class="topbar">
                        Genome Browser</a> &nbsp; - &nbsp;
                <A HREF="/cgi-bin/hgBlat?command=start" class="topbar">
                        Blat Search</a> &nbsp; - &nbsp;
                <A HREF="/cgi-bin/hgText?command=start" class="topbar">
                        Table Browser</a> &nbsp; - &nbsp;
                <A HREF="goldenPath/help/hgTracksHelp.html" class="topbar">
                        User Guide</a> &nbsp;</font></TD>
        </TR></TABLE>
    </TD></TR></TABLE>
</TD></TR>


<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>
            <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
                <TR><TD> <FONT SIZE="4"><b>&nbsp;   Frequently Asked Questions: Data Downloads</b></FONT>
                </TD></TR></TABLE>
            <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
                <TR><TD>&nbsp;</TD><TD>
		<UL>
                <LI><A HREF="#download1">Downloading sequence and annotation data</A>
                <LI><A HREF="#download2">Download restrictions</A>
                <LI><A HREF="#download3">Opening .fa files</A>
                <LI><A HREF="#download4">Data differences between downloaded data and browser display</A>
                <LI><A HREF="#download5">Strange characters in FASTA file</A>
                <LI><A HREF="#download6">Selection of GenBank ESTs</A>
                <LI><A HREF="#download7">EST strand direction</A>
                <LI><A HREF="#download8">Missing RefSeq ID</A>
                <LI><A HREF="#download9">Finish vs. draft segments</A>
                <LI><A HREF="#download10">chrN_random tables</A>
                <LI><A HREF="#download11">Chromosome Un</A>
                <LI><A HREF="#download12">Chromosome M</A>
                <LI><A HREF="#download13">N characters at beginning of human chr22</A>
                <LI><A HREF="#download15">100K bp discrepancy in human Build 31 chr22</A>
                <LI><A HREF="#download16">Repeat-masking data</A>
                <LI><A HREF="#download17">Availability of repeat-masked data</A>
                <LI><A HREF="#download18">Obtaining promoter sequence</A>
                <LI><A HREF="#download19">Data from Evolutionary Conservation Score tracks</A>
                <LI><A HREF="#download20">Minus strand coordinates - axtBest files</A>
                <LI><A HREF="#download21">Mapping UCSC STS marker IDS to those of other groups</A>
                <LI><A HREF="#download22">deCODE map data</A>
		</UL>
		<A HREF="index.html">Return to FAQ Table of Contents</A>
		<BR>
            </TD><TD WIDTH=15></TD></TR></TABLE><BR>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download1"></A>
<TABLE BORDER=0 WIDTH="100%">
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Downloading sequence and annotation data</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"How do I obtain the sequence and/or annotation data for a release?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Sequence and annotation data downloads are usually made available within 
			the first week of the release of a new assembly. The download directories
			are automatically updated nightly to incorporate additions and modifications
			to the data.
			<P>
			We recommend that you download data via our FTP site at 
			<A HREF="ftp://genome.ucsc.edu/goldenPath/">http://genome.ucsc.edu/goldenPath/</A>, 
			particularly if you plan to download multiple files or files of large size. 
			You can also download data from our 
			<A HREF="http://genome.ucsc.edu/downloads.html">Downloads</A>
			page. To download a specific subset of the data or to configure the output
			format of the data, use the Table Browser.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download2"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp;Download restrictions</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Do you have restrictions on the amount of downloads one can do?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Generally, we'd prefer that you not hit our interactive site with programs, 
			unless they are themselves front ends for interactive sites. We can handle 
			the traffic from all the clicks that biologists are likely to generate, 
			but not from programs. Program-driven use is limited to a maximum of one 
			hit every 10 seconds and no more than 5,000 hits per day. 
			<P>
			If you need to run batch Blat jobs, please 
			<A HREF="mailto:www-genome@cse.ucsc.edu">contact us</A> for a copy of 
			Blat you can run locally.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download3"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Opening .fa files</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I am trying to look at the final decoding of the human genome. How can I
			open the *.fa files?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Microsoft Word or any program that can handle large text files will do. 
			Some of the chromosomes begin with long blocks of <em>N</em>'s. You may want 
			to search for an <em>A</em> to get past them. 
			<P>
			Unless you have a particular need 
			to view or use the raw data files, you might find it more interesting to 
			look at the data using the Genome Browser. Type the name of a gene in which
			you're interested into the position box (or use the default position), 
			then click the Submit button. In the resulting Genome Browser  
			display, click the DNA link on the menu bar at the top of the page. 
			Select the Extended case/color options button at the bottom of the 
			next page. Now you can color the DNA sequence to display which portions are 
			repeats, known genes, genetic markers, etc.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download4"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Data differences between downloaded data and browser display</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I downloaded the genome annotations from your MySQL database tables, but the 
			mRNA locations didn't match what was showing in the Genome Browser. Shouldn't they 
			be in synch?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Yes. The Genome Browser and Table Browser are both driven by the same 
			underlying MySQL database. Check that your downloaded tables are from
			the same assembly version as the one you are viewing in the Genome Browser. If the 
			assembly
			dates don't match, the coordinates of the data within the tables may differ. 
			In a very rare instance, you could also be affected 
			by the brief lag time between the update of the live databases underlying the Genome
			Browser and the time it takes for text dumps of these databases to become available 
			in the downloads directory.  
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download5"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Strange characters in FASTA file</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I noticed several characters other than <em>A</em>, <em>C</em>, <em>G</em>, 
			<em>T</em>, and <em>N</em> in my fasta file, for example <em>y</em>, <em>k</em>, 
			<em>s</em>, etc. Is the file corrupted or are these characters valid?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			The characters most commonly seen in sequence are <em>A</em>, <em>C</em>, <em>G</em>, 
			<em>T</em>, and <em>N</em>, but there are 
			several other valid characters that are used in clones to indicate
			ambiguity about the identity of certain bases in the sequence. It's not uncommon to 
			see these &quot;wobble&quot; codes at polymorphic positions in DNA sequences. The
			following chart (IUPAC-IUB Symbols for Nucleotide Nomenclature: Cornish-Bowden
			(1985). <em>Nucl. Acids Res.</em> 13:3021-3030) lists nucleotide symbols, including those
			used for ambiguity:
			<PRE>
          		--------------------------------------
          		Symbol    Meaning      Nucleic Acid
          		--------------------------------------
           		A            A           Adenine
           		C            C           Cytosine
           		G            G           Guanine
           		T            T           Thymine
           		U            U           Uracil
           		M          A or C
           		R          A or G        Purine
           		W          A or T
           		S          C or G
           		Y          C or T        Pyrimidine
           		K          G or T
           		V        A or C or G
           		H        A or C or T
           		D        A or G or T
           		B        C or G or T
           		X      G or A or T or C
           		N      G or A or T or C
			</PRE>
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download6"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Selection of GenBank ESTs</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I am interested in ESTs. How do you select which ones from GenBank to display in the 
			Genome Browser?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			All ESTs in GenBank on the date of the track data freeze for the given organism are 
			used - none are discarded. When two ESTs have identical sequences, both are 
			retained because this can be significant corroboration of a splice site. </P>
			<P>
			ESTs are aligned against the genome using the Blat program. When a single EST aligns 
			in multiple places, the alignment having the highest base identity is found. Only 
			alignments that have a base identity level within a selected percentage of 
			the best are kept. Alignments 
			must also have a minimum base identity to be kept. For more information on the 
			selection criteria specific to each organism, consult the description page accompanying
			the EST track for that organism.</P>
			<P>
			The maximum intron length allowed by Blat is 500,000 bases, which may 
			eliminate some ESTs with very long introns that might otherwise align. If an EST 
			aligns non-contiguously (i.e. an intron has been spliced out), it is also a candidate 
			for the Spliced EST track, provided it meets various quality controls for intron and 
			exon length and match quality.  Start and stop coordinates of each alignment block are 
			available from the appropriate positional table within the 
			<A HREF="http://genome.ucsc.edu/goldenPath/hgText.html">Table Browser</A>.</P>
			<P>
			Note that only 250 EST tracks can be viewed at a time within the browser. If more 
			than 250 tracks exist for the selected region, the display defaults to a denser display
			mode to 
			prevent the user's web browser from being overloaded. You can restore the EST track 
			display to a fuller display mode by zooming in on the chromosomal range or by using 
			the EST track filter to restrict the number of tracks displayed.</P>
			<P>
			For tracks such as Non[Organism] ESTs and Non[Organism] mRNAs, some selection is done
			on the full set at GenBank. If a sequence is too divergent from the organism's genome 
			to generate a significant Blat hit, it is not included in the track.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download7"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; EST strand direction</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Could you help me with my interpretation of EST data? If the EST is taken
			from the minus (-) strand, does this always mean that the transcript is generated 
			on the minus strand? Are two corresponding ESTs that are assigned 
			- and + always complementary? 
			<P>
			I want to confirm the strand assignment for two 
			human ESTs: 
			<UL>
			<LI>BQ016549 (chr22:20676822-20698290): + strand in text and - strand in graphical 
			display 
			<LI>AA928010 (chr22:18713957-18723221): - strand in text and + strand in graphical 
			display. 
			</UL>
			The graphical display goes with the orientation of the gene in that location."
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			When UCSC downloads mRNAs and ESTs from GenBank and aligns them to a genome 
			assembly, each alignment has a direction (forward or reverse), which we 
			record as + or - in the strand field of the database table. 
			<P>
			Some GenBank EST 
			records specify a 3' or 5' read direction, and some don't. If no direction is 
			specified, or if the direction is 5', the Genome Browser's graphical 
			display uses the strand field of the database table (the alignment direction) 
			to determine whether to draw the arrows pointing forward (+) or 
			reverse (-).  However, when the direction is 3' in the GenBank record, 
			the arrows are drawn opposite of the strand (alignment direction).
			<P>
			The alignment details pages and the Table Browser do not take the GenBank 
			record's read direction into account. They show only the alignment of the 
			GenBank sequence (as given) to the genome. If the alignment is used to 
			retrieve DNA sequence from the genome, the DNA sequence will look 
			similar to the GenBank sequence (not its complement).
			The GenBank records for BQ016549 and AA928010 specify 3' as the read 
			direction. Therfore, the graphical display reverses the direction of alignment, but 
			the details page and Table Browser don't. It is admittedly a bit quirky.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download8"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Missing RefSeq ID</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Why isn't my refseq ID in your database?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			It may have been added after we last downloaded data from Genbank, or it may have 
			been replaced or removed. You can check the submission date and status of an accession
			on the 
			<A HREF="http://www.ncbi.nlm.nih.gov:80/entrez/" TARGET=_blank>NCBI 
			Entrez Nucleotide site</A>.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download9"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Finished vs. draft segments</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Do chr<em>N</em>.fa tables contain both finished and draft segments? If so, 
			how do you determine which segments are finished?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Yes, these tables contain both finished and draft segments. Use the 
			corresponding chr<em>N</em>_gold table to look them up. The quality of the draft 
			varies. In 
			general, the larger the contig it is in, the better the quality. The 
			quality of the last 500 bases on either end of a contig tends to be 
			lower than the rest of the contig. 
			<P>
			How do you determine the accuracy? The 
			base-calling program <A HREF="http://www.phrap.org/" TARGET=_blank>Phred</A> analyzes 
			the traces from the sequencing machines 
			and assigns a quality score to these. These quality scores are used by the 
			<A HREF="http://www.phrap.org/" TARGET=_blank>Phrap</A> assembly program, which gives 
			quality scores for the bases on the assembly as well.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download10"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; chrN_random tables</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"What are the chr<em>N</em>_random_[table] files in the human assembly? Why are they 
			called random? Is there something biologically random about the sequence in 
			these tables or are they just not placed within their given chromosomes?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			In the past, these tables contained data related to sequence that is 
			known to be in a particular chromosome, but could not be reliably ordered 
			within the current sequence. 
			<P>
			Starting with the the April 2003 human assembly, these tables also include data for 
			sequence that is not in a finished state, but whose location in the chromosome is 
			known, in addition to the unordered sequence.  
			Because this sequence is not quite finished, it could not be included in the 
			main &quot;finished&quot; ordered and oriented section of the chromosome.  
			<P>
			Also, in 
			a very few cases, the random files contain data related to sequence for alternative 
			haplotypes.  
			This is present primarily in chr6, where we have included two alternative
			versions of the MHC region in chr6_random. There are a few clones in 
			other chromosomes that also correspond to a different haplotype.  Because the 
			primary reference sequence can only display a single haplotype, these 
			alternatives have been been included in random files for now.  In the future, 
			we hope to handle this situation in a better fashion.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download11"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Chromosome Un</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"What is ChrUn?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			ChrUn contains clone contigs that can't be confidently placed on a 
			specific chromosome. For the chr<em>N</em>_random and chrUn_random files, we 
			essentially just concatenate together all the contigs into short 
			pseudo-chromosomes. The coordinates of these are fairly arbitrary, 
			although the relative positions of the coordinates 
			are good within a contig. You can find more information about the data organization 
			and format on the <A HREF="http://genome.ucsc.edu/goldenPath/datorg.html">Data
			Organization and Format</A> page.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download12"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Chromosome M </b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"What is chromosome M (chrM)?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Mitochondrial DNA.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download13"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; N characters at beginning of human chr22</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"When I download human chr22 from your web site, the unzipped file contains only 
			<em>N</em>'s."
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			There is a large block of <em>N</em>'s at the beginning and end of chr22. Search 
			for an <em>A</em> to bypass the initial group of <em>N</em>'s.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download15"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; 100K bp discrepancy in human Build 31 chr22</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"The chr22 download file for human Build 31 is 47748585 bp long, but 
			the browser shows the length as 47848585 bp. Which is correct?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			This is a ~100K bp gap inserted into chr22 just after the centromere.
			<P>
			There are 5 chromosomes (13, 14, 15, 21, 22) which are termed 
			<em>acrocentric</em>, meaning that one of their arms is very 
			short, and thus the centromere is very off-center. The short 
			arms for these chromosomes 
			consist primarily of heterochromatin combined with a 
			&quot;stalk&quot; that mainly 
			consists of RNA genes. These regions are very hard to sequence,
			similar to the centromeres.
			<P>
			We try to represent the true lengths of each chromosome. 
			Therefore, <em>N</em>'s are inserted to represent bases in 
			regions that cannot be sequenced.  For these 5 chromosomes, 
			the short arms consist entirely of <em>N</em>'s.  This 
			is the ~13Mb of <em>N</em>'s that you see at the beginning of 
			chr22.  The remaining sequence is the q-arm for chr22, the 
			only part that has been sequenced.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download16"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Repeat-masking data</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"What RepeatMasker flags do you use on your data?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			We use RepeatMasker <em>-s</em>. For mouse repeats, we also use <em>-m</em>. 
			In addition to RepeatMasker, we also use the Tandem Repeat Finder (trf) program, 
			masking out repeats of period 12 or less. The repeats are 
			just &quot;soft&quot; masked. Alignments are allowed to extend through repeats, but 
			not initiate in them.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download17"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Availability of repeat-masked data</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Are the repeat annotation files available for every chromosome?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Yes, you can obtain the repeat-masked files via the Table Browser or from the 
			organism's annotation database downloads directory. The RepeatMasker annotation 
			tables are named 
			chr<em>N</em>_rmsk (where <em>N</em> represents the chromosome number) and the 
			Tandem Repeat Finder (TRF) tables are named simpleRepeat.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download18"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Obtaining promoter sequence</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"How can I fetch promoter sequence upstream of a gene?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			The UCSC Genome Browser offers several ways to obtain this information, 
			depending on your requirements.
			<P>
			The Genome Browser downloads site provides prepackaged downloads of 1000 bp, 2000 bp, 
			and 5000 bp upstream sequence for RefSeq genes that have annotated
			5' UTRs. You can obtain these from the bigZips downloads
			directory for the assembly of interest. 
			<P>
			To fetch the upstream sequence for a specific gene, use the 
			<A HREF="http://genome.ucsc.edu/cgi-bin/hgText">Table Browser</A>. 
			Enter the genome, assembly, and gene in which you're interested, 
			select the refGene positional table, then click the Advanced query button. 
			On the next page, select the output type &quot;FASTA&quot; and click the Get Results 
			button. You will then have the opportunity to configure the amount of upstream promoter
			sequence to fetch, along with several other display options. Click Get Sequence
			when you've finished configuring the output.
			<P>
			You can also use the Genome Browser to obtain sequence for a specific gene. 
			Open the Genome Browser window to display the gene in which you're
			interested. Click the entry for the gene in the RefSeq or Known Genes track, then
			click the Genomic Sequence link. Alternatively, you can click the DNA link in 
			the top menu bar of the Genome Browser tracks window to access options for displaying 
			the sequence. 
			<P>
			The Stanford Human Promoters track on the 
			<A HREF="http://genome.ucsc.edu/goldenPath/customTracks/custTracks.html">UCSC 
			Custom Annotation Tracks page</A> shows promoters for the hg13 human assembly.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download19"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Data from Evolutionary Conservation Score tracks</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Where can I download the conservation score data from the Human/Mouse 
			Evolutionary Conservation Score track?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			The conservation score data are stored in a group of tables in the annotation 
			database downloads directory. 
			The naming conventions of the tables vary among releases. In earlier
			assemblies, table names are of the form chr<em>N</em>_humMusL, chr<em>N</em>_zoom1_humMusL, and
			or chr<em>N</em>_zoom2500_humMusL. In later releases, the tables are named using 
			specific release numbers, such as chr<em>N</em>_hg16Mm3. The tables within a given set
			differ by the number of bases/score interval and are used to generate the browser
			displays at different zooming levels. 
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download20"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Minus strand coordinates - axtBest files</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I downloaded the axtBest alignments between the latest human and mouse assemblies. 
			I found that some of the alignments listed in the axtBest 
			files do not agree with what is shown in the browser." 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR>
			Is this alignment on the minus strand? Minus strand coordinates in axt files 
			are handled differently from how they are handled in the Genome Browser. To convert 
			axt minus strand coordinates to Genome Browser coordinates, use:
			<PRE>
      	start = chromSize - axtEnd
      	end = chromSize - axtStart
			</PRE>
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    </TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download21"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Mapping UCSC STS marker IDs to those of other groups</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"How do I map the STS genetic marker IDs in the genome browser to the 
			IDs assigned by other groups? "
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			We assign our own IDs to each of the STS markers, but we also track 
			the UniSTS IDs for each marker in the downloadable stsInfo2 table.  
			To determine the location of a specific marker, look up the marker's name
			in the stsAlias table to determine the UCSC ID assigned to the 
			marker, and then use this ID to look it up in the stsMap table where the marker 
			is located. For example, D10S249 has UCSC ID 2880 and is located at chr10:240791-241019.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="download22"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; deCODE map data</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Where can I get more information about the deCODE map?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			You can obtain this information from the combination of a couple of tables. 
			The stsMap table contains the physical position of all STS markers, 
			including those on the deCODE map. This file also contains information about 
			the position on the genome-wide maps, including the deCODE map. A second file, 
			stsInfo2, contains additional information about each marker, including aliases, 
			primer sequence information, etc. This table is related to the first table by an 
			ID (the identNo field in both files). 
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
    </TD><TD WIDTH=15></TD></TR>
</TABLE>
</BODY></HTML>
