<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
        <META name="description" content="UCSC Genome Browser FAQ">
        <META name="keywords" content="genome, genome browser, genome assembly, UCSC, bioinformatics, sequence, gene prediction, SNP, EST, mRNA">
	<TITLE>UCSC Genome Bioinformatics: FAQ</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/FAQ.topbar.html"--></TD></TR>
<!--Content Tables-->
    <TR><TD COLSPAN=2 CELLPADDING=10>
    <!--outer table is for border purposes-->
   	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>
           	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>
            		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
                	    <TR><TD> <FONT SIZE="4"><b>&nbsp;   Frequently Asked Questions: Data File Formats</b></FONT> 
			    </TD></TR>
			</TABLE>
            		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
                	    <TR><TD>&nbsp;</TD><TD>
				<UL>
                		<LI><A HREF="#format1">BED format</A>
                		<LI><A HREF="#format2">PSL format</A>
                		<LI><A HREF="#format3">GFF format</A>
                		<LI><A HREF="#format4">GTF format</A>
                		<LI><A HREF="#format5">MAF format</A>
                		<LI><A HREF="#format6">WIG format</A>
                		<LI><A HREF="#format6.5">Microarray format</A>
                		<LI><A HREF="../goldenPath/help/chain.html">Chain format</A>
                		<LI><A HREF="../goldenPath/help/net.html">Net format</A>
				<LI><A HREF="../goldenPath/help/axt.html">Axt format</A>
                		<LI><A HREF="#format7">.2bit format</A>
                		<LI><A HREF="#format8">.nib format</A>
                		<LI><A HREF="#format9">GenePred table format</A>
				</UL>
				<P>
				<A HREF="index.html">Return to FAQ Table of Contents</A>
				<BR>
            	     	     </TD><TD WIDTH=15></TD></TR>
			</TABLE><BR>
        	     </TD></TR>
		</TABLE>
    	     </TD></TR>
	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format1"></A>
    <!--outer table is for border purposes-->        
    	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
             <TR><TD>            
        	<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    	     <TR><TD>                
  	    		<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
	        	     <TR><TD>                
	        	     <!--cheap trick to get background on row to be continuous-->
	   			<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    		     <TR><TD ALIGN="left"><FONT SIZE="4"><b>&nbsp; BED format</b></FONT></TD> 
		    		     <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            		     </TD></TR>
  	        		</TABLE>                
	        	     </TD></TR>
  	    		</TABLE>                
   	    		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        	    <TR><TH HEIGHT=10></TH></TR>
                	    <TR><TD WIDTH=10>&nbsp;</TD>
		    	    <TD>
	            	    <P>
<P>BED format provides a flexible way to define the data lines that
are displayed in an annotation track. BED lines have three required fields and
nine additional optional fields. The number of fields per line must be consistent
throughout any single set of data in an annotation track. The order of the
optional fields is binding: lower-numbered fields must always be populated if
higher-numbered fields are used.


<P>The first three required BED fields are:</P>
<OL>
        <LI><B>chrom</B> - The name of the chromosome (e.g. chr3, chrY,
        chr2_random) or scaffold (e.g. scaffold10671).
        <LI><B>chromStart</B> - The starting position of the feature in the
        chromosome or scaffold. The first base in a chromosome is numbered 0.
        <LI><B>chromEnd</B> - The ending position of the feature in the
        chromosome or scaffold. The <I>chromEnd</I> base is not included in the
        display of the feature. For example, the first 100 bases of a
        chromosome are defined as <I>chromStart=0, chromEnd=100</I>, and span
        the bases numbered 0-99.
</OL>

<P>The 9 additional optional BED fields are:</P>
<OL START=4>
        <LI><B>name</B> - Defines the name of the BED line.  This label is
        displayed to the left of the BED line in the Genome Browser
        window when the track is open to full display mode or directly to the
        left of the item in pack mode.
        <LI><B>score</B> - A score between 0 and 1000. If the track line
        <I>useScore</I> attribute is set to 1 for this annotation data set, the
        <I>score</I> value will determine the level of gray in which
        this feature is displayed (higher numbers = darker gray).
        <LI><B>strand</B> - Defines the strand - either '+' or '-'.
        <LI><B>thickStart</B> - The starting position at which the feature
        is drawn thickly (for example, the start codon in gene
        displays).
        <LI><B>thickEnd</B> - The ending position at which the feature is
        drawn thickly (for example, the stop codon in gene displays).
        <LI><B>itemRgb</B> - An RGB value of the form R,G,B (e.g. 255,0,0).
        If the track line <em>itemRgb</em> attribute is set to &quot;On&quot;,
        this RBG value will determine the display color of the data contained
        in this BED line. NOTE: It is recommended that a simple color scheme
        (eight colors or less) be used with this attribute to avoid 
	overwhelming the color resources of the Genome Browser and your 
	Internet browser.
        <LI><B>blockCount</B> - The number of blocks (exons) in the
        BED line.
        <LI><B>blockSizes</B> - A comma-separated list of the block
        sizes. The number of items in this list should correspond to
        <I>blockCount</I>.
        <LI><B>blockStarts</B> - A comma-separated list of block starts.
        All of the <I>blockStart</I> positions should be calculated relative to
        <I>chromStart</I>. The number of items in
        this list should correspond to <I>blockCount</I>.
</OL>
<P><B><I>Example:</I></B><BR>
Here's an example of an annotation track that uses a complete BED definition:</P>
<PRE><TT>track name=pairedReads description=&quot;Clone Paired Reads&quot; useScore=1
chr22 1000 5000 cloneA 960 + 1000 5000 0 2 567,488, 0,3512
chr22 2000 6000 cloneB 900 - 2000 6000 0 2 433,399, 0,3601</TT></PRE>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format2"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; PSL format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
<P>PSL lines represent alignments, and are typically taken from files generated by BLAT
or psLayout. See the
<a
href="http://genome.ucsc.edu/goldenPath/help/hgTracksHelp.html#BLATAlign"
TARGET=_blank>BLAT
documentation</a> for more details. All of the following fields are
required on each data line within a PSL file:
<OL>
<LI><B>matches</B> - Number of bases that match that aren't repeats
<LI><B>misMatches</B> - Number of bases that don't match
<LI><B>repMatches</B> - Number of bases that match but are part of repeats
<LI><B>nCount</B> - Number of 'N' bases
<LI><B>qNumInsert</B> - Number of inserts in query
<LI><B>qBaseInsert</B> - Number of bases inserted in query
<LI><B>tNumInsert</B> - Number of inserts in target
<LI><B>tBaseInsert</B> - Number of bases inserted in target
<LI><B>strand</B> - '+' or '-' for query strand. For translated alignments, second '+'or '-' is for genomic strand
<LI><B>qName</B> - Query sequence name
<LI><B>qSize</B> - Query sequence size
<LI><B>qStart</B> - Alignment start position in query
<LI><B>qEnd</B> - Alignment end position in query
<LI><B>tName</B> - Target sequence name
<LI><B>tSize</B> - Target sequence size
<LI><B>tStart</B> - Alignment start position in target
<LI><B>tEnd</B> - Alignment end position in target
<LI><B>blockCount</B> - Number of blocks in the alignment (a block contains no gaps)
<LI><B>blockSizes</B> - Comma-separated list of sizes of each block
<LI><B>qStarts</B> - Comma-separated list of starting positions of each block in query
<LI><B>tStarts</B> - Comma-separated list of starting positions of each block in target
</OL>
<P><B><I>Example:</I></B><BR>
Here is an example of an annotation track in PSL format. Note that line
breaks have been inserted into the PSL lines in this example for
documentation display purposes.
 Click <A HREF="../goldenPath/help/fishBlats.txt">
here</A> for a copy of this example that can be pasted into the browser without editing.
<PRE><TT>track name=fishBlats description=&quot;Fish BLAT&quot; useScore=1
59 9 0 0 1 823 1 96 +- FS_CONTIG_48080_1 1955 171 1062 chr22
    47748585 13073589 13073753 2 48,20,  171,1042,  34674832,34674976,
59 7 0 0 1 55 1 55 +- FS_CONTIG_26780_1 2825 2456 2577 chr22
    47748585 13073626 13073747 2 21,45,  2456,2532,  34674838,34674914,
59 7 0 0 1 55 1 55 -+ FS_CONTIG_26780_1 2825 2455 2676 chr22
    47748585 13073727 13073848 2 45,21,  249,349,  13073727,13073827,
</TT></PRE>

<P>Be aware that the coordinates for a negative strand in a PSL line are handled in a special way. In the
<I>qStart</I> and <I>qEnd</I> fields, the coordinates indicate the position where the query matches from
the point of view of the forward strand, even when the match is on the reverse strand.
However, in the <I>qStarts</I> list, the coordinates are reversed.</P>
<P>
<B><I>Example:</I></B><BR>
Here is a 30-mer containing 2 blocks that align on the minus strand and
2 blocks that align on the plus strand (this sometimes can happen in response to
assembly errors): 
<BR><BR>
<PRE><TT>0         1         2         3 tens position in query  
0123456789012345678901234567890 ones position in query   
            ++++          +++++ plus strand alignment on query   
    --------    ----------      minus strand alignment on query   

Plus strand:   
     qStart=12 
     qEnd=31 
     blockSizes=4,5 
     qStarts=12,26 
                  
Minus strand:   
     qStart=4 
     qEnd=26 
     blockSizes=10,8 
     qStarts=5,19   
</TT></PRE>
Essentially, the minus strand <I>blockSizes</I> and <I>qStarts</I> are
what you would get if you reverse-complemented the query.
However, the <I>qStart</I> and <I>qEnd</I> are not reversed. To convert one to the other:
<PRE>
     qStart = qSize - revQEnd
     qEnd = qSize - revQStart
</PRE>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format3"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; GFF format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
<P>GFF (General Feature Format) lines are based on the GFF standard file format.  GFF
lines have nine required fields that <I>must</I> be tab-separated. If the fields are
separated by spaces instead of tabs, the track will not display correctly. For more
information on GFF format, refer to <A
HREF="http://www.sanger.ac.uk/Software/formats/GFF"
TARGET=_blank>http://www.sanger.ac.uk/Software/formats/GFF</A>. </P>
<P>
Here is a brief description of the GFF fields:</P>

<OL>
        <LI><B>seqname</B> - The name of the sequence.  Must be a chromosome or scaffold.
        <LI><B>source</B> - The program that generated this feature.
        <LI><B>feature</B> - The name of this type of feature. Some examples of
        standard feature types are "CDS", "start_codon", "stop_codon", and
        "exon".
        <LI><B>start</B> - The starting position of the feature in the sequence.  The first base is numbered 1.
        <LI><B>end</B> - The ending position of the feature (inclusive).
        <LI><B>score</B> - A score between 0 and 1000. If the track line
        <I>useScore</I> attribute is set to 1 for this annotation data set, the
        <I>score</I> value will determine the level of gray in which
        this feature is displayed (higher numbers = darker gray). If there is no
        score value, enter ".".        <LI><B>strand</B> - Valid entries include '+', '-', or '.' (for don't know/don't care).        <LI><B>frame</B> - If the feature is a coding exon, <I>frame</I> should be a number between 0-2 that represents the reading frame of the
        first base.  If the feature is not a coding exon, the value should be '.'.        <LI><B>group</B> - All lines with the same group are linked together into a single item.
</OL>
<P><B><I>Example:</I></B><BR>
Here's an example of a GFF-based track.
Click <A HREF ="../goldenPath/help/regulatory.txt">here</A> for a copy of this example that can be 
pasted into the browser without editing. NOTE: Paste operations on some 
operating systems will replace tabs with spaces, which will result in an error 
when the GFF track is uploaded. You can circumvent this problem by pasting the 
URL of the above example
(http://genome.ucsc.edu/goldenPath/help/regulatory.txt) instead of the text 
itselfinto the custom annotation track text box.

<PRE><TT>track name=regulatory description=&quot;TeleGene(tm) Regulatory Regions&quot;
chr22  TeleGene enhancer  1000000  1001000  500 +  .  touch1
chr22  TeleGene promoter  1010000  1010100  900 +  .  touch1
chr22  TeleGene promoter  1020000  1020000  800 -  .  touch2
</TT></PRE>

			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format4"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; GTF format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
<!--#include virtual="../goldenPath/help/GTF.html" -->
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format5"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; MAF format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
                    The multiple alignment format stores a series of
                    multiple alignments in a format that is easy to
                    parse and relatively easy to read.  This format
                    stores multiple alignments at the DNA level
                    between entire genomes. Previously used formats 
		    are suitable for multiple alignments of single proteins or 
		    regions of DNA without rearrangements, but would require 
		    considerable extension to cope with genomic issues such as
                    forward and reverse strand directions, multiple pieces 
		    to the alignment, and so forth.
                    <P>
		    <B>General Structure</B>
		    <P> The <em>.maf</em> format is
                    line-oriented. Each multiple alignment ends with a
                    blank line.  Each sequence in an alignment is on a single 
		    line, which can get quite long, but there is no length limit.
		    Words in a line are delimited by any white space. Lines
                    starting with # are considered to be comments.
                    Lines starting with ## can be ignored by most
                    programs, but contain meta-data of one form or
                    another.
                    <P>
                    The file is divided into paragraphs that terminate
                    in a blank line.  Within a paragraph, the first
                    word of a line indicates its type.  Each multiple
                    alignment is in a separate paragraph that begins
                    with an &quot;a&quot; line and contains an &quot;s&quot; line 
		    for each sequence in the multiple alignment.  
		    Some MAF files may contain other  optional line types: 
		    <UL>
		    <LI>
		    an "i" line containing information about what is in the 
		    aligned species DNA before and after the immediately 
		    preceding "s" line
		    <LI>
		    an "e" line containing information about the size of the 
		    gap between the alignments that span the current 
		    block 
		    <LI>
		    a "q" line indicating the quality of each aligned base
		    for the species
		    </UL>
		    <P>
		    Parsers may ignore any other types of paragraphs and 
		    other types of lines within an alignment paragraph. 
                    <P>
		    <B>Custom Tracks</B>
		    <P>
                    The first line of a custom MAF track must be a "track" line
		    that contains a name=value pair specifying the track name.
		    Here is an example of a minimal track line:
		    
		    <FONT color="#0000FF"><PRE><TT> track name=sample</TT></PRE></FONT>
                    The following variables can be specified in the track line
		    of a custom MAF:
                    <UL>
                    <LI><B>name=sample</B> - Required. Name the track </LI>
                    <LI><B>description="Sample Track"</B> - Optional. 
		    Gives a long name for the track</LI>
                    <LI><B>frames=multiz28wayFrames</B> - Optional. 
		    Tells the browser which table to 
		    grab the gene frames from.  This is usually associated with
		    an N-way alignment where the name ends in the string 
		    "Frames" </LI>
                    <LI><B>mafDot=on</B> - Optional. 
		    Use dots instead of bases when bases are identical</LI>
                    <LI><B>visibility=dense|pack|full</B> - Optional. 
		    Sets the default visibility mode for this track.</LI>
                    <LI><B>speciesOrder="hg18 panTro2"</B> - Optional. 
		    White-space separated list specifying the order in which
		    the sequences in the maf should be displayed.
		    </LI>
		    </UL>
		    The second line of a custom MAF track must be a header
	            line as described below.
		    <P>
		    <B>Header Line</B>
		    <P>
                    The first line of a <em>.maf</em> file begins with ##maf. 
		    This word is followed by white-space-separated
                    variable=value pairs.  There should be <em>no</em> white
                    space surrounding the &quot;=&quot;.  
		    <FONT color="#0000FF"><PRE><TT> ##maf version=1 scoring=tba.v8 </TT></PRE></FONT>
		    The currently defined variables are:
                    <UL>
                    <LI><B>version</B> - Required.  Currently set to one.  </LI>
                    <LI><B>scoring</B> - Optional. A name for the scoring
                                   scheme used for the alignments.
                                   The current scoring schemes are:
                        <UL>
			<LI><em>bit</em> - roughly corresponds to blast bit
                                   values (roughly 2 points per
                                   aligning base minus penalties for
                                   mismatches and inserts).  </LI>
                        <LI><em>blastz</em> - blastz scoring scheme -- roughly
                                   100 points per aligning base.</LI>
                        <LI><em>probability</em> - some score normalized
                                   between 0 and 1.  </LI>
                        </UL>
                    <LI><B>program</B> - Optional. Name of the program
                                   generating the alignment.</LI>
                    </UL>
                    Undefined variables are ignored by the parser.
                    <P>
		    <B>Alignments Parameter Line</B>
		    <P>
                    The second line displays the parameters that were
                    used to run the alignment program.
		    <FONT color="#0000FF"><PRE><TT> # tba.v8 (((human chimp) baboon) (mouse rat))</TT></PRE></FONT>
                    <P>
		    <B>Alignment Block Lines</B> (lines starting with 'a' -- parameters for a new alignment block)
		    <P>                    
		    <FONT color="#0000FF"><PRE><TT> a score=23262.0</TT></PRE></FONT>
                    Each alignment begins with an 'a' line that set 
		    variables for the entire alignment block.  The 'a'
                    is followed by name=value pairs.  There are no
                    required name=value pairs.  The currently defined
                    variables are: 
		    <UL>
                    <LI> score -- Optional. Floating point score. If
                                 this is present, it is good practice
                                 to also define scoring in the first line.  </LI>

                    <LI> pass --  Optional.  Positive integer value.
                                 For programs that do multiple pass
                                 alignments such as blastz, this shows
                                 which pass this alignment came from.
                                 Typically, pass 1 will find the
                                 strongest alignments genome-wide, and
                                 pass 2 will find weaker alignments
                                 between two first-pass alignments.</LI>
                    </UL>
                    <P>
		    <B>Lines starting with 's' -- a sequence within an alignment block</B>
		    <P>
		    <FONT color="#0000FF">
<PRE><TT> s hg16.chr7    27707221 13 + 158545518 gcagctgaaaaca
 s panTro1.chr6 28869787 13 + 161576975 gcagctgaaaaca
 s baboon         249182 13 +   4622798 gcagctgaaaaca
 s mm4.chr6     53310102 13 + 151104725 ACAGCTGAAAATA
</TT></PRE></FONT>
                    The 's' lines together with the 'a' lines define a
                    multiple alignment.  The 's' lines have the
                    following fields which are defined by position
                    rather than name=value pairs.
		    <UL>
                    <LI> src -- The name of one of the source sequences
                               	for the alignment.  For sequences that
                               	are resident in a browser assembly, the
			 	form 'database.chromosome' allows
                               	automatic creation of links to other 
				assemblies.  Non-browser sequences are
                               	typically reference by the species name
                               	alone.</LI>
                    <LI> start -- The start of the aligning region in the 
				source sequence. This is a
                               	zero-based number. If the strand field 
				is '-' then this is the start
                               	relative to the reverse-complemented
                               	source sequence.  </LI>
                    <LI> size -- The size of the aligning region in the
                               	source sequence.  This number is equal
                                to the number of non-dash characters in
                               	the alignment text field below.  </LI>          
		    <LI> strand -- Either '+' or '-'.  If '-', then the           
                               	alignment is to the
                               	reverse-complemented source.  </LI>
                    <LI> srcSize -- The size of the entire source sequence, 
				not just the parts involved in the alignment.</LI>
		    <LI> text -- The nucleotides (or amino acids) in
				the alignment and any insertions
                               (dashes) as well.  </LI> 
                    </UL>          
                                   
  		    <P>
		    <B>Lines starting with 'i' -- information about what's 
		    happening before and after this block in the aligning 
		    species</B> 
                    <P><FONT color="#0000ff">
<PRE><TT> s hg16.chr7    27707221 13 + 158545518 gcagctgaaaaca s panTro1.chr6 28869787 13 + 161576975 gcagctgaaaaca
 i panTro1.chr6 N 0 C 0
 s baboon         249182 13 +   4622798 gcagctgaaaaca
 i baboon       I 234 n 19</TT></PRE></FONT>
		    <P>
		    The 'i' lines contain information about the context of the 
		    sequence lines immediately preceeding them. The following 
		    fields are defined by position rather than name=value pairs:
                    <UL>
                    <LI>src -- The name of the source sequence for the 
		    alignment. Should be the same as the 's' line immediately 
		    above this line.             
		    <LI>leftStatus -- A character that specifies the 
		    relationship between the sequence in this block and the 
		    sequence that appears in the previous block.
		    <LI>leftCount -- Usually the number of bases in the aligning
                    species between the start of this alignment and the end of 
		    the previous one.                          
		    <LI>rightStatus -- A character that specifies the 
		    relationship between the sequence in this block and the 
		    sequence that appears in the subsequent block.
	            <LI>rightCount -- Usually the number of bases in the 
		    aligning species between the end of this alignment and the 
		    start of the next one. 
		    </UL>
		    <P>
		    The status characters can be one of the following values:
                    <UL>                          
		    <LI>C -- the sequence before or after is contiguous with 
		    this block.
		    <LI>I -- there are bases between the bases in this block 
		    and the one before or after it.                          
		    <LI>N -- this is the first sequence from this src chrom or 
		    scaffold.
		    <LI>n -- this is the first sequence from this src chrom or 
		    scaffold but it is bridged by another alignment from a 
		    different chrom or scaffold.                          
		    <LI>M -- there is missing data before or after this block 
		    (Ns in the sequence).
		    <LI>T -- the sequence in this block has been used before
		    in a previous block (likely a tandem duplication)
		    </UL>                        
		    <P>
		    <B>Lines starting with 'e' -- information about empty parts 
		    of the alignment block</B>                        
		    <P><FONT color=#0000ff>
<PRE><TT> s hg16.chr7    27707221 13 + 158545518 gcagctgaaaaca
 e mm4.chr6     53310102 13 + 151104725 I</TT></PRE></FONT>
		    <P>
		    The 'e' lines indicate that there isn't aligning DNA for a 
		    species but that the current block is bridged by a chain 
		    that connects blocks before and after this block. The 
		    following fields are defined by position rather than 
		    name=value pairs.
                    <UL>                          
		    <LI>src -- The name of one of the source sequences for the 
		    alignment.                          
		    <LI>start -- The start of the non-aligning region in the
                    source sequence. This is a zero-based number. If the
		    strand field is '-' then this is the start relative to 
                    the reverse-complemented source sequence.
		    <LI>size -- The size in base pairs of the non-aligning 
		    region in the source sequence.                          
		    <LI>strand -- Either '+' or '-'. If '-', then the
                    alignment is to the reverse-complemented source.
                    <LI>srcSize -- The size of the entire source sequence, 
                    not just the parts involved in the alignment.
                    alignment and any insertions (dashes) as well.
		    <LI>status -- A character that specifies the 
		    relationship between the non-aligning sequence in this 
		    block and the sequence that appears in the previous and 
		    subsequent blocks.
		    </UL>
		    <P>
		    The status character can be one of the following values:
                    <UL>                          
		    <LI>C -- the sequence before and after is contiguous 
		    implying that this region was either deleted in the source 
		    or inserted in the reference sequence. The browser draws a 
		    single line or a '-' in base mode in these blocks.
		    <LI>I -- there are non-aligning bases in the source species 
		    between chained alignment blocks before and after this 
		    block. The browser shows a double line or '=' in base mode.
		    <LI>M -- there are non-aligning bases in the source and 
		    more than 90% of them are Ns in the source. The browser 
		    shows a pale yellow bar.                         
		    <LI>n -- there are non-aligning bases in the source and the 
		    next aligning block starts in a new chromosome or scaffold 
		    that is bridged by a chain between still other blocks. The 
		    browser shows either a single line or a double line based 
		    on how many bases are in the gap between the bridging 
		    alignments.
		    </UL>
		    <B>Lines starting with 'q' -- information about the quality
		    of each aligned base for the species</B> 
                    <P><FONT color=#0000ff>
<PRE><TT> s hg18.chr1                  32741 26 + 247249719 TTTTTGAAAAACAAACAACAAGTTGG
 s panTro2.chrUn            9697231 26 +  58616431 TTTTTGAAAAACAAACAACAAGTTGG
 q panTro2.chrUn                                   99999999999999999999999999
 s dasNov1.scaffold_179265     1474  7 +      4584 TT----------AAGCA---------
 q dasNov1.scaffold_179265                         99----------32239--------- </TT></PRE></FONT>
		    <P>
		    The 'q' lines contain a compressed version of the actual
		    raw quality data, representing the quality of each 
		    aligned base for the species with a single character of 0-9 
		    or F. The following fields are defined by position rather 
		    than name=value pairs:
		    <UL>
                    <LI>src -- The name of the source sequence for the 
		    alignment. Should be the same as the 's' line immediately 
		    preceding this line.             
		    <LI>value -- A MAF quality value corresponding to 
		    the aligning nucleotide acid in the preceding 
		    's' line.  Insertions (dashes) in the preceding 's' line
	       	    are represented by dashes in the 'q' line as well. 
		    The quality value can be 'F' (finished
		    sequence) or a number derived from the actual quality
		    scores (which range from 0-97) or the manually assigned
		    score of 98. These numeric values are calculated as: 
<BLOCKQUOTE><TT>MAF quality value = min( floor(actual quality value/5), 9 )</TT></BLOCKQUOTE>
		    This results in the following mapping: 
		    <P>
            	    <TABLE BORDER=1 BORDERCOLOR="#aaaaaa"> 
		    <TR>
                    <TH nowrap><FONT color="#006666"><B>MAF quality value</B></FONT></TH>
		    <TH nowrap><FONT color="#006666"><B>Raw quality score range</B></FONT></TH>
                    <TH nowrap><FONT color="#006666"><B>Quality level</B></FONT></TH>
                    </TR>
		    <TR>
                    <TD ALIGN="center">0-8</TD>
		    <TD ALIGN="center">0-44</TD>
                    <TD ALIGN="center">Low</TD>
                    </TR>
		    <TR>
                    <TD ALIGN="center">9</TD>
		    <TD ALIGN="center">45-97</TD>
                    <TD ALIGN="center">High</TD>
                    </TR>
		    <TR>
                    <TD ALIGN="center">0</TD>
		    <TD ALIGN="center">98</TD>
                    <TD ALIGN="center">Manually assigned</TD>
                    </TR>
		    <TR>
                    <TD ALIGN="center">F</TD>
		    <TD ALIGN="center">99</TD>
                    <TD ALIGN="center">Finished</TD>
                    </TR>
		    </TABLE>
		    </UL><BR>
                    <P>
		    <B>A Simple Example</B>
		    <P>                     
		    Here is a simple example of a three alignment
                    blocks derived from five starting sequences.
		    The first <B>track</B> line is necessary for custom
		    tracks, but should be removed otherwise.
		    Repeats are shown as lowercase, and each block may
                    have a subset of the input sequences.  All
		    sequence columns and rows must contain at least one nucleotide
                    (no columns or rows that contain only insertions).

		    <FONT color="#0000FF">
<PRE><TT> 
track name=euArc visibility=pack
##maf version=1 scoring=tba.v8 
# tba.v8 (((human chimp) baboon) (mouse rat)) 
                   
a score=23262.0     
s hg18.chr7    27578828 38 + 158545518 AAA-GGGAATGTTAACCAAATGA---ATTGTCTCTTACGGTG
s panTro1.chr6 28741140 38 + 161576975 AAA-GGGAATGTTAACCAAATGA---ATTGTCTCTTACGGTG
s baboon         116834 38 +   4622798 AAA-GGGAATGTTAACCAAATGA---GTTGTCTCTTATGGTG
s mm4.chr6     53215344 38 + 151104725 -AATGGGAATGTTAAGCAAACGA---ATTGTCTCTCAGTGTG
s rn3.chr4     81344243 40 + 187371129 -AA-GGGGATGCTAAGCCAATGAGTTGTTGTCTCTCAATGTG
                   
a score=5062.0                    
s hg18.chr7    27699739 6 + 158545518 TAAAGA
s panTro1.chr6 28862317 6 + 161576975 TAAAGA
s baboon         241163 6 +   4622798 TAAAGA 
s mm4.chr6     53303881 6 + 151104725 TAAAGA
s rn3.chr4     81444246 6 + 187371129 taagga

a score=6636.0
s hg18.chr7    27707221 13 + 158545518 gcagctgaaaaca
s panTro1.chr6 28869787 13 + 161576975 gcagctgaaaaca
s baboon         249182 13 +   4622798 gcagctgaaaaca
s mm4.chr6     53310102 13 + 151104725 ACAGCTGAAAATA
&nbsp
</TT></PRE></FONT>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format6"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; WIG format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
<P>
Wiggle format (WIG) allows the display of continuous-valued data in a track
format. Click <A HREF="../goldenPath/help/wiggle.html">here</A> for more 
information.</P>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format6.5"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Microarray format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
<P>
The datasets for the built-in microarray tracks in the Genome Browser are stored
in BED15 format, an extension of <A HREF="#format1">BED</A> format that includes
three additional fields: expCount, expIds, and expScores. 
To display correctly in the Genome Browser, microarray tracks 
require the setting of several attributes in the trackDb file associated with 
the track's genome assembly. Each microarray track set must also have an 
associated microarrayGroups.ra configuration file that contains additional 
information about the data in each of the arrays.</P>
<P>
User-created microarray custom tracks are similar in format to BED 
custom tracks with the addition of three required track line parameters in the 
header--expNames, expScale, and expStep--that mimic the trackDb and 
microarrayGroups.ra settings of built-in microarray tracks. </P>
<P>
For a complete description of the microarray track format and an explanation
of how to construct a microarray custom track, see the 
<A HREF="http://genomewiki.ucsc.edu/index.php/Microarray_track"
TARGET=_blank>Genome Browser Wiki</A>.
</P>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format7"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; .2bit format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
<P>
A .2bit file stores multiple DNA sequences (up to 4 Gb total) in a compact 
randomly-accessible format. The file contains masking information as well 
as the DNA itself.</P>
<P>
The file begins with a 16-byte header containing the 
following fields:
<UL>
<LI>signature - the number 0x1A412743 in the architecture of the machine that 
created the file
<LI>version - zero for now. Readers should abort if they see a version number 
higher than 0.
<LI>sequenceCount - the number of sequences in the file.
<LI>reserved - always zero for now
</UL>
<P>
All fields are 32 bits unless noted. If the signature value is not as given, 
the reader program should byte-swap the signature and check if the swapped 
version matches. If so, all multiple-byte entities in the file will have to be 
byte-swapped. This enables these binary files to be used unchanged on 
different architectures.</P>
<P>
The header is followed by a file index, which contains one entry for 
each sequence. Each index entry contains three fields:
<UL>
<LI>nameSize - a byte containing the length of the name field
<LI>name - the sequence name itself, of variable length depending on nameSize
<LI>offset - the 32-bit offset of the sequence data relative to the start of the
file
</UL>
<P>
The index is followed by the sequence records, which contain nine fields:
<UL>
<LI>dnaSize - number of bases of DNA in the sequence
<LI>nBlockCount - the number of blocks of Ns in the file (representing unknown 
sequence)
<LI>nBlockStarts - the starting position for each block of Ns
<LI>nBlockSizes - the size of each block of Ns
<LI>maskBlockCount - the number of masked (lower-case) blocks
<LI>maskBlockStarts - the starting position for each masked block
<LI>maskBlockSizes - the size of each masked block
<LI>reserved - always zero for now
<LI>packedDna - the DNA packed to two bits per base, represented as so: 
T - 00, C - 01, A - 10, G - 11.  The first base is in the most significant 
2-bit byte; the last base is in the least significant 2 bits. For example, the
sequence TCAG is represented as 00011011. The packedDna field is padded with 0 
bits as necessary to take an even multiple of 32 bits in the file, which 
improves I/O performance on some machines.
</UL>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format8"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; .nib format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
<P>
The .nib format pre-dates the .2bit format and is less compact. It 
describes a DNA sequence by packing two bases into each byte. 
Each .nib file contains only a single sequence.  The file begins with a 32-bit 
signature that is 0x6BE93D3A in the architecture of the machine that created 
the file (or possibly a byte-swapped version of the same number on another 
machine). This is followed by a 32-bit number in the same format that describes 
the number of bases in the file. Next, the bases themselves are listed, packed 
two bases to the byte. The first base is packed in the high-order 4 bits 
(nibble); the second base is packed in the low-order four bits:
<BLOCKQUOTE><TT>
byte = (base1<<4) + base2
</TT></BLOCKQUOTE></P>
<P>
The numerical representations for the bases are:
<UL>
<LI>0 - T
<LI>1 - C
<LI>2 - A
<LI>3 - G
<LI>4 - N (unknown)
</UL></P>
<P>
The most significant bit in a nibble is set if the base is masked. </P>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format9"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; GenePred table format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
<P>
genePred is a table format commonly used for gene prediction tracks in the
Genome Browser. Variations of the genePred format are listed below.

<H4><A name=GenePredictions></A>Gene Predictions</H4>
The following definition is used for gene prediction tables. In 
alternative-splicing situations, each transcript has a row in this table.
<PRE>table genePred
"A gene prediction."
    (
    string  name;               "Name of gene"
    string  chrom;              "Chromosome name"
    char[1] strand;             "+ or - for strand"
    uint    txStart;            "Transcription start position"
    uint    txEnd;              "Transcription end position"
    uint    cdsStart;           "Coding region start"
    uint    cdsEnd;             "Coding region end"
    uint    exonCount;          "Number of exons"
    uint[exonCount] exonStarts; "Exon start positions"
    uint[exonCount] exonEnds;   "Exon end positions"
    )
</PRE>

<H4><A name=GenePredExt></A>Gene Predictions (Extended)</H4>
The following definition is used for extended gene prediction tables. In 
alternative-splicing situations, each transcript has a row in this table.
The refGene table is an example of the genePredExt format.
<PRE>table genePredExt
"A gene prediction with some additional info."
    (
    string name;        	"Name of gene (usually transcript_id from GTF)"
    string chrom;       	"Chromosome name"
    char[1] strand;     	"+ or - for strand"
    uint txStart;       	"Transcription start position"
    uint txEnd;         	"Transcription end position"
    uint cdsStart;      	"Coding region start"
    uint cdsEnd;        	"Coding region end"
    uint exonCount;     	"Number of exons"
    uint[exonCount] exonStarts; "Exon start positions"
    uint[exonCount] exonEnds;   "Exon end positions"
    uint id;            	"Unique identifier"
    string name2;       	"Alternate name (e.g. gene_id from GTF)"
    string cdsStartStat; 	"enum('none','unk','incmpl','cmpl')"
    string cdsEndStat;   	"enum('none','unk','incmpl','cmpl')"
    lstring exonFrames; 	"Exon frame offsets {0,1,2}"
    )
</PRE>

<H4><A name=RefFlat></A>Gene Predictions and RefSeq Genes with Gene Names</H4>
A version of genePred that associates the gene name with the gene prediction 
information. In alternative-splicing situations, each transcript has a row in
this table.
<PRE>table refFlat
"A gene prediction with additional geneName field."
    (
    string  geneName;           "Name of gene as it appears in Genome Browser."
    string  name;               "Name of gene"
    string  chrom;              "Chromosome name"
    char[1] strand;             "+ or - for strand"
    uint    txStart;            "Transcription start position"
    uint    txEnd;              "Transcription end position"
    uint    cdsStart;           "Coding region start"
    uint    cdsEnd;             "Coding region end"
    uint    exonCount;          "Number of exons"
    uint[exonCount] exonStarts; "Exon start positions"
    uint[exonCount] exonEnds;   "Exon end positions"
    )
</PRE>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>
    </TD><TD WIDTH=15></TD></TR>
</TABLE>
</BODY></HTML>
