<!DOCTYPE html>
<!--#set var="TITLE" value="Genome Browser FAQ" -->
<!--#set var="ROOT" value=".." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<h1>Frequently Asked Questions: Linking to the Genome Browser</h1>

<h2>Topics</h2>

<div class="row">
<div class="col-md-5">
<h6> <a href="#link1">Creating a sharable URL to view specific tracks</a> </h6>
<h6> <a href="#link2">Linking to the Browser at a specific position, or default position</a> </h6>
<h6> <a href="#zoomVariant">Zooming in or out using a link to an HGVS identifier</a> </h6>
<h6> <a href="#trackViz">Setting track visibility via URL</a> </h6>
<h6> <a href="#custUrl">Loading Custom Tracks with the URL</a> </h6>
<h6> <a href="#hubUrl">Loading Track Hubs and Assembly Hubs with the URL</a> </h6>
<h6> <a href="#genes">Linking to gene-specific information</a> </h6>
<h6> <a href="#hgsid">The hgsid parameter</a> </h6>
<h6> <a href="#moreInfo">Additional URL parameters</a> </h6>

<h6><a href="#Videos">Video demonstrations of making links</a></h6>
<ul>
  <li><a href="#LinkVid1"><strong> Understanding the URL</strong></a></li>
  <li><a href="#LinkVid2"><strong> Jump into genes</strong></a></li>
  <li><a href="#LinkVid3"><strong> Composites, custom tracks, spreadsheets</strong></a></li>

</ul>

</div>
<div class="col-md-7">

<form name="googleForm1" method="GET" action="https://www.google.com/search"
onSubmit="document.googleForm1.q.value=document.googleForm1.qq.value+
'   site:genome.ucsc.edu/goldenPath/help';">
<p>
  Search the Genome Browser help pages: &nbsp;
  <input type="hidden" name="q" value="">
  <input type="hidden" name="num" value="10">
  <input type="hidden" name="filter" value="0">
  <input type=text name=qq size=30 maxlength=255 value="">
  <input type="submit" value="Submit">
</form>
</p>
<p><a href="../../contacts.html">Questions and feedback are welcome</a>.</p>

 </div>
</div>


<a name="link1"></a>
<h2>Creating a sharable URL to view specific tracks</h2>
<h6>How do I create a link to the Genome Browser to share my data?</h6>
<p> 
The easiest way to save and share tracks from the URL is by
<a href="../cgi-bin/hgLogin" TARGET="_blank">logging in</a> to your Genome Browser account and
creating a <a href="../goldenPath/help/hgSessionHelp.html">saved session</a>.
Saved sessions are a versatile way to share data that may include native annotations, Custom
Tracks, Track Hubs, and Assembly Hubs. In these examples, text in brackets <b>"&lt;"</b> and <b>"&gt;"</b> 
indicate places where the user supplies information. Note that the brackets are not needed for the 
URL, including the brackets will result in a 'Could not find session' error.</p>
<p>
You will be able to share Genome Browser sessions with the following link format:</p>
<p><code>http://genome.ucsc.edu/s/&lt;userName&gt;/&lt;sessionName&gt;</code></p>
<p>
For instructions on creating a saved session, go to the
<a href="../goldenPath/help/hgSessionHelp.html#Create" target ="_blank">session user guide</a>.
If you want to specify track settings in a URL directly, please read the section on 
<a href="#trackViz">setting track visibility via URL</a> for a complete description. </p>
<h6> Loading sessions on other tools</h6>
<p>
Or if you prefer the older style, which allows you to link to different tools, you may use the 
following parameters:</p>
<ul>
  <li id="hgS_otherUserName"><code>hgS_otherUserName=&lt;userName&gt</code></li>
  <li id="hgS_otherUserSessionName"><code>hgS_otherUserSessionName=&lt;sessionName&gt;</code></li>
  <li id="hgS_doOtherUser"><code>hgS_doOtherUser=submit</code></li>
</ul>
<p>
Here is an example:</p>
<p><code>http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&amp;hgS_otherUserName=&lt;userName&gt;&amp;hgS_otherUserSessionName=&lt;sessionName&gt;</code></p>
<p>
This longer format has the flexibility of replacing &quot;hgTracks&quot; with different tool names to share 
saved settings on the Table Browser (hgTables), Variant Annotation Integrator (hgVai), or Data 
Integrator (hgIntegrator). This will preserve your option selections and can be useful to share.
The following format will bring the recipient to a user's custom Table Browser selections:</p>
<p><code>http://genome.ucsc.edu/cgi-bin/hgTables?hgS_doOtherUser=submit&amp;hgS_otherUserName=&lt;userName&gt;&amp;hgS_otherUserSessionName=&lt;sessionName&gt;</code></p>

<p>Both session link formats have the advantage of being able to add URL parameters to the end.
The shorter link format requires a question mark before any URL parameters, with ampersand 
characters separating different parameters like so: </p>
<p><code><a href=http://genome.ucsc.edu/s/view/clinicalzoom?textSize=18>http://genome.ucsc.edu/s/view/clinicalzoom?textSize=18</a></code></p>
<p>Both formats require an ampersand between each additional parameter, seen in the longer format like so:</p>
<p><code><a href=http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&amp;hgS_otherUserName=view&amp;hgS_otherUserSessionName=clinicalzoom&amp;textSize=18>http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&amp;hgS_otherUserName=view&amp;hgS_otherUserSessionName=clinicalzoom&amp;textSize=18</a></code></p>
<h6>Building URLs to specify settings</h6>
<p>
Another option is to host, modify, or create your own session/settings file and then host it in an 
external location. Nearly every track parameter on the genome browser can be specified in a session
file, as described on our <a href="/goldenPath/help/customTrack.html#optParams" target="_blank">
custom tracks page</a>. This also allows you to circumvent URL character limits.</p>
<p>
Once the file is prepared and hosted, you can load it with the
<code id="hgS_doLoadUrl">hgS_doLoadUrl=submit</code> and
<code id="hgS_loadUrlName">hgS_loadUrlName=&lt;URL&gt;</code> variables, this also has the added
benefit that you can
externally maintain your session,
and update it as you wish. Here is an example:</p>
<p><code><a href="https://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doLoadUrl=submit&hgS_loadUrlName=http://genome.ucsc.edu/goldenPath/help/examples/sessions/session.hg19.stopCodons">
https://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doLoadUrl=submit&hgS_loadUrlName=http://genome.ucsc.edu/goldenPath/help/examples/sessions/session.hg19.stopCodons</a></code></p>

<a name="link2"></a>
<h2>Linking to the Browser at a specific position, or default position</h2>
<h6>How do I make a link to a specific genome, position, or HGVS variant?</h6>
<p>
You can link to a specific genome assembly and position in the Genome Browser using a URL with
the <code>db=</code> and <code>position=</code> parameters.</p>
<p><code>http://genome.ucsc.edu/cgi-bin/hgTracks?db=&lt;assembly&gt;&position=&lt;position&gt;</code></p>
<p>
Where:</p>
<ul>
  <li id="db">
  <code>db</code> - designates a specific genome assembly. For example, <code>db=hg19</code> 
  refers to the Feb. 2009 human genome release. For a list of db parameter values that 
  correspond to UCSC assemblies, see the <a href="FAQreleases.html#release1">list of UCSC 
  releases</a>.</li>
  <li id="position"><code>position</code> - can be any search term for the genome specified, including
  a position range or a gene identifier. This often takes the form of
  <code>position=chr1:35000-40000</code>.</li>
</ul>
<p>The following link is an example of a URL that declares assembly and position:</p>
<p><code><a href="../cgi-bin/hgTracks?db=hg19&position=chr1:35000-40000">http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&position=chr1:35000-40000</a></code></p>

<h6>How do I make a link to the default position of a genome?</h6>
<p>
You can link directly to the default position of an assembly by passing the 
<code>position=default</code> parameter. This can be helpful in cases where
a track exists on multiple assemblies, and you want to build links to each of them. If no
position variable is passed, the Genome Browser assumes the default position for the 
default assembly (hg38).</p>
<p>
Here is an example which opens the uniprot track at the default position for sacCer3:</p>
<p><code><a href="/cgi-bin/hgTrackUi?db=sacCer3&g=uniprot&position=default">http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=sacCer3&g=uniprot&position=default</a></code></p>

<a name="zoomVariant"></a>
<h2>Zooming in or out with a link &mdash; HGVS </h2>
<h6>How do I zoom out using a link to a HGVS identifier?</h6>
<p>
To link to a specific HGVS identifier, you can construct a link with the HGVS identifier
in the position field instead of coordinates. The default padding is 5bp on either side,
but you can always zoom in or out with <code>hgt.out1=submit</code> or <code>hgt.in1=submit</code>.
The numbers 1 through 4 zoom in or out corresponding to the buttons above the track window.
The following lists the zoom levels of each number, applicable to zooming in or out:</p>
<ul>
<li id="hgt.out1"><code>hgt.out1=submit</code> zooms out 1.5x</li>
<li id="hgt.out2"><code>hgt.out2=submit</code> zooms out 3x</li>
<li id="hgt.out3"><code>hgt.out3=submit</code> zooms out 10x</li>
<li id="hgt.out4"><code>hgt.out4=submit</code> zooms out 100x</li>
</ul>
<p>
The following link is an example which leads to the variant NM_00257:c.1208G&gt;T and zooms 
out 3x:</p>
<p>
<code><a href="../cgi-bin/hgTracks?db=hg19&position=NM_000257:c.1208G%3ET&hgt.out2=submit">http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&position=NM_000257:c.1208G>T&hgt.out2=submit</a></code></p>


<a name="trackViz"></a>
<h2>Setting Track Visibility via URL</h2>
<h6>How do I create a custom URL to control the visibility of specific tracks?</h6>
<p>
You can control the visibility of tracks from the URL with the following parameters, 
each linked by the "&amp;" sign:
<ul>
  <li>
  <code>guidelines=on/off</code> - activate or deactivate the blue guidelines -
  <a href="../../cgi-bin/hgTracks?db=hg38&position=chr9%3A133255176-133275214&guidelines=off"
  target="_blank">example link</a> to switch off blue guidelines</li>
  <li>
  <code>hgFind.matches=&lt;listOfNames&gt;</code> - highlight features given their names -
  <a href="../../cgi-bin/hgTracks?db=hg38&position=chr9%3A133255176-133275214&hgFind.matches=ENST00000611156.4,ENST00000538324.2"
  target="_blank">example link</a> to highlight two transcripts of the ABO gene</li>
  <li>
  <code>hgt.reset=1</code> - show only the default tracks -
  <a href="../../cgi-bin/hgTracks?db=hg38&position=chr9%3A133255176-133275214&hgt.reset=1"
  target="_blank" >example link</a></li>
  <li>
  <code>hgt.toggleRevCmplDisp=1</code> - show the reverse-complement -
  <a href="../../cgi-bin/hgTracks?db=hg38&position=chr9%3A133255176-133275214&hgt.toggleRevCmplDisp=1"
  target="_blank">example link</a> to show the reverse-complement of the ABO gene</li>
  <li>
  <code>hgt.labelWidth=&lt;number&gt;</code> - set the size of the left-side label area -
  <a href="../../cgi-bin/hgTracks?db=hg38&position=chr9%3A133255176-133275214&hgt.labelWidth=50"
  target="_blank">example link</a> to increase the label area to 50 characters</li>
  <li>
  <code>hideTracks=1</code> - hide all tracks -
  <a href="../../cgi-bin/hgTracks?db=hg38&position=chr9%3A133255176-133275214&hideTracks=1"
  target="_blank">example link</a> to show no tracks at all</li>
  <li>
  <code>hideTracks=1&amp;&lt;trackName&gt;=full|dense|pack|hide</code> - hide all tracks and show
  other tracks -
  <a href="../../cgi-bin/hgTracks?db=hg38&position=chr9%3A133255176-133275214&hideTracks=1&cytoBand=pack"
  target="_blank">example link</a> to show only the Chromosome Bands track and nothing else</li>
  <li>
  <code>highlight=&lt;db&gt;.&lt;chrom&gt;:&lt;chromStart&gt;-&lt;chromEnd&gt;#&lt;color&gt;|...</code> -
  highlight one or more regions in a given color on the image. Note that the arguments have to be
  URL-encoded for Internet browsers, so &quot;:&quot; becomes &quot;%3A&quot;, &quot;#&quot;
  becomes &quot;%23&quot; and &quot;|&quot; becomes &quot;%7&quot;C. -
  <a href="../../cgi-bin/hgTracks?db=hg38&position=chr9%3A133255176-133275214&highlight=hg38.chr9%3A133255612-133262185%23AA0000%7Chg38.chr9%3A133269124-133274224%230000FF"
  target="_blank">example link</a> to highlight two parts of the ABO locus in red and blue.</li>
  <li>
  <code>ignoreCookie=1</code> - do not load the user's existing settings saved
  in the internet browser's UCSC Genome Browser cookie.  This means that the link will show the
  Genome Browser default
  settings such as track selections, custom tracks, and track hubs. Any changes
  you make in this new session will, however, affect the user's settings.  E.g.,
  if you add a track in this new window, and come back to the genome browser
  later, the track will still be there. This setting is useful if a website
  wants to link to the Genome Browser, starting with a "clean slate" but
  believes the user will come back to the Genome Browser expecting the
  changes to still be there. </li>
  <li>
  <code>ruler=hide</code> - hide the ruler at the top of the browser image -
  <a href="../../cgi-bin/hgTracks?db=hg38&hideTracks=1&knownGene_imgOrd=2&knownGene=pack&gtexGeneV8=full&gtexGeneV8_imgOrd=1&ruler=hide"
  target="_blank">example link</a> to hide the ruler</li>
  <li>
  <code>oligoMatch=pack&amp;hgt.oligoMatch=&lt;dnaSeq&gt;</code> - switch on the Short Match track and
  highlight a matching sequence -
  <a href="../../cgi-bin/hgTracks?db=hg38&position=chr7%3A155796193-155816207&oligoMatch=pack&hgt.oligoMatch=TATAWAR"
  target="_blank">example link</a> to highlight the TATAWAR motif in the ABO locus</li>
  <li>
  <code>pix=&lt;number&gt;</code> - set the width of the image in pixels -
  <a href="../../cgi-bin/hgTracks?db=hg38&position=chr9%3A133255176-133275214&pix=300"
  target="_blank">example link</a> to create a 300-pixel wide image</li>
  <li>
  <code>textSize=&lt;number&gt;</code> - sets browser text size to either 6, 8, 10, 12, 14, 18, 24,
   or 34. Default is a textSize of 12. -
  <a href="../../cgi-bin/hgTracks?db=hg38&position=chr9%3A133255176-133275214&textSize=18"
  target="_blank">example link</a> to increase the text font size to 18 pixels</li>
  <li>
  <code>&lt;trackName&gt;=full|pack|dense|hide</code> - sets specified track or subtrack to a
  chosen visibility: full, pack, dense, or hide track -
  <a href="../../cgi-bin/hgTracks?db=hg38&position=chr9%3A133255176-133275214&cytoBand=pack"
  target="_blank">example link</a> to show the
  Chromosome Bands track set to &quot;pack&quot; and added to your view as saved in your cart.
  Please note that for this feature to work with
  custom tracks you must use the unique name and identifier number
  <code>ct_name_####</code> assigned by our system. You can determine the name for a custom track
  using the url, <a href = "https://genome.ucsc.edu/cgi-bin/cartDump">
  https://genome.ucsc.edu/cgi-bin/cartDump</a>.
  <li>
  <code>&lt;trackName&gt;_imgOrd=&lt;number&gt;</code> - vertically orders the tracks on the image
  based on the numbers provided. You need to specify an order for every visible track when using this
  parameter -
  <a href="../../cgi-bin/hgTracks?db=hg38&hideTracks=1&knownGene_imgOrd=2&knownGene=pack&gtexGeneV8=full&gtexGeneV8_imgOrd=1"
  target="_blank">example link</a> to show two show knownGene track being listed second with gtex first</li>
  <li>
  <code>&lt;trackName&gt;.heightPer=&lt;number&gt;</code> - sets a bigWig track's height to a
  particular number of pixels (between 20-100) -
  <a href="../../cgi-bin/hgTracks?db=hg38&ignoreCookie=1&hideTracks=1&mappability=full&umap24Quantitative.heightPer=100"
  target="_blank">example link</a> to set umap bigWig track height to 100 pixels
  </li>
  <li>
  <code>&lt;trackName&gt;_hideKids=1</code> - hides a specific super track's individual tracks - <a
  href="../../cgi-bin/hgTracks?db=hg38&wgEncodeReg_hideKids=1&ignoreCookie=1" target="_blank">
  example link</a> to hide the Encode Regulation super track</li>
  <li>
  <code>&lt;trackName&gt;_sel=1</code> - selects specific subtrack to be 'checked', allowing
  display - <a href="../../cgi-bin/hgTracks?db=hg38&refSeqComposite=full&refGene_sel=1" target="_blank">example
  link</a> to select the checkbox for UCSC RefSeq subtrack in the refSeq composite track, allowing
  display alongside default tracks
  </li>
</ul>
<p>
For example, you can use the following command to hide every track (hideTracks=1), set the
genome database to hg38 (db=hg38), set the mappability track to full visibility (mappability=full),
and set the umap track height to 100 pixels (umap24Quantitative.heightPer=100). Each of these 
parameters can be used individually or in combination.</p>
<p><code><a 
href="../cgi-bin/hgTracks?db=hg38&ignoreCookie=1&hideTracks=1&mappability=full&umap24Quantitative.heightPer=100">
http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38&hideTracks=1&mappability=full&umap24Quantitative.heightPer=100</a></code></p>

<p>Composite tracks have additional URL parameters that encode options to hide, select, and 
display subtracks.</p>
<ul>
<li><code>&lt;trackName&gt;_hideKids=1</code> - hides a specific composite track's subtracks</li>
<li><code>&lt;trackName&gt;_sel=1</code> - selects specific subtrack to be 'checked', allowing
display</li>
</ul>
<p>For example, the following URL hides all tracks (hideTracks=1), hides a specific composite
track's default subtracks (refSeqComposite_hideKids=1), turns on one specific subtrack 
(ncbiRefSeqCurated=full), and checks a box to display that subtrack 
(ncbiRefSeqCurated_sel=1).</p>

<pre><a href="../cgi-bin/hgTracks?db=hg38&hideTracks=1&refSeqComposite_hideKids=1&ncbiRefSeqCurated=full&ncbiRefSeqCurated_sel=1" target="_blank">https://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38&hideTracks=1&refSeqComposite_hideKids=1&ncbiRefSeqCurated=full&ncbiRefSeqCurated_sel=1</a></pre>


<h2>Loading data with the URL</h2>
<a name="custUrl"></a>
<h3>Loading Custom Track data with the URL</h3>
<h6>How do I create a link to my custom track data?</h6>
<p>
If you have a custom track on a web-accessible server, you can use the location of the file
to load it directly as part of a URL.
You can combine the URL visibility settings with the <code>hgct_customText=</code> parameter using
a track line you would otherwise put in the <a href="../cgi-bin/hgCustom">custom track 
input box</a>. The following example shows the <code>hgct_customText</code> parameter accepting
a bigBed file URL as a custom track:
</p>
<pre><a href="../cgi-bin/hgTracks?db=hg38&position=chr21:34821279-34888690&hgct_customText=https://genome.ucsc.edu/goldenPath/help/examples/bigBedExample.bb" target="_blank">http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38&position=chr21:34821279-34888690&hgct_customText=https://genome.ucsc.edu/goldenPath/help/examples/bigBedExample.bb</a></pre>

<p>
If you want to add more information to the Custom Track, you can do so using the 
<code>hgct_customText</code> parameter. Since this is a URL, you must use "%20" to encode
for spaces and "%0A" for a new line character. For example, the following example shows Custom
Track input pasted in the <a href="../cgi-bin/hgCustom">custom track input box</a> and the 
equivalent input in the URL:</p>

<pre>browser position chr21:33038946-33039092
track type=bam bigDataUrl=https://genome.ucsc.edu/goldenPath/help/examples/bamExample.bam name=Example description=ExampleBAM</pre>
<pre><a href="../cgi-bin/hgTracks?db=hg38&hgct_customText=browser%20position%20chr21:33038946-33039092%0Atrack%20type=bam%20bigDataUrl=https://genome.ucsc.edu/goldenPath/help/examples/bamExample.bam%20name=Example%20description=ExampleBAM">http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38&hgct_customText=browser%20position%20chr21:33038946-33039092%0Atrack%20type=bam%20bigDataUrl=https://genome.ucsc.edu/goldenPath/help/examples/bamExample.bam%20name=Example%20description=ExampleBAM</a></pre>
<p>
More information on custom track parameters can be found in the 
<a href="../goldenPath/help/customTrack.html">Custom Track user guide</a>.</p>


<a name="hubUrl"></a>
<h3>Loading Track Hubs and Assembly Hubs with the URL</h3>
<h6>How do I create a link to my track hub or assembly hub?</h6>
<p>Similar to custom tracks, track hubs can be loaded into the URL using the 
<code>hubUrl=</code> parameter. This parameter takes input similar to the
<a href="../cgi-bin/hgHubConnect#unlistedHubs">track hub input box</a>. The following example
links to the hg19 genome database and an example track Hub:</p>  
<pre><a href=../cgi-bin/hgTracks?db=hg19&hubUrl=https://genome.ucsc.edu/goldenPath/help/examples/hubDirectory/hub.txt>http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&hubUrl=https://genome.ucsc.edu/goldenPath/help/examples/hubDirectory/hub.txt</a></pre>

<p>Track hubs' track visibility can also be changed from the URL parameters. The following link 
specifies the genome database (db=hg19), loads a track hub (hubUrl=http.../hub.txt), hides all tracks
(hideTracks=1), hides the subtrack kids of a particular track (gtexRnaSignalMaleYoung_hideKids=1),
sets a specific subtrack to be displayed (gtexRnaSignalSRR1311243=full), and ignores user settings
(ignoreCookie=1).</p>
<pre><a href="../cgi-bin/hgTracks?db=hg19&hubUrl=http://hgdownload.soe.ucsc.edu/hubs/gtex/hub.txt&hideTracks=1&gtexRnaSignalMaleYoung_hideKids=1&gtexRnaSignalMaleYoung=full&gtexRnaSignalSRR1311243=full&ignoreCookie=1">https://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&hubUrl=http://hgdownload.soe.ucsc.edu/hubs/gtex/hub.txt&hideTracks=1&amp;gtexRnaSignalMaleYoung_hideKids=1&amp;gtexRnaSignalMaleYoung=full&amp;gtexRnaSignalSRR1311243=full&ignoreCookie=1</a></pre>

<p>
To link to an assembly hub and display data on a non-natively supported genome, the same 
parameters apply. To specify the intended genome assembly, instead of using <code>db=</code>,
you must use <code>genome=araTha1</code>, where araTha1 is the assembly name set by
your genomes.txt file in the line <code>genome araTha1</code>.</p>
<pre><a href=../cgi-bin/hgTracks?genome=araTha1&hubUrl=http://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubAssembly/plantAraTha1/hub.txt>https://genome.ucsc.edu/cgi-bin/hgTracks?genome=araTha1&hubUrl=http://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubAssembly/plantAraTha1/hub.txt</a></pre>
<p>
To see the files behind that assembly hub, please visit the
<a href="../goldenPath/help/examples/hubExamples/hubAssembly/plantAraTha1/">
hub's directory</a>. For more information on assembly hubs in general, please see the
<a href="../goldenPath/help/assemblyHubHelp.html" target ="_blank">assembly hub user guide</a>, the <a
href="../goldenPath/help/hgTrackHubHelp.html">track hub user guide</a>,
or the <a href=../goldenPath/help/hubQuickStartAssembly.html>quick start guide
to assembly hubs</a>.
</p>
<h6 id="hubClear">Use <code>hubClear=</code> to remove hubs at the same location</h6>
<p>
Another feature one can use in place of <code>hubUrl=</code> is
<code>hubClear=</code>, which will load a hub while simultaneously disconnecting
or clearing, hubs located at the same location. For example, adding
<code>hubClear=http://university.edu/lab/folder/hub10.txt</code> would connect the
referenced hub10.txt while simultaneously disconnecting any hubs that might be
displayed from  the same <code>http://university.edu/lab/folder/</code> directory
(for example, hub1.txt, hub2.txt, etc.). This feature can be useful for dynamically
generated hubs that might collect in the browser otherwise.
</p>
<h6>Additional hub connection parameters</h6>
<p>
Alongside <code>hubUrl</code> and <code>hubClear</code> there are other parameters
you can add to either if you wished, for instance, to attach an assembly hub and
display the Gateway page, while also sharing your contact email.</p>
<ul>
<li id="hgHubConnect.remakeTrackHub"><code>hgHubConnect.remakeTrackHub=on</code>
acts to connect when pointed to /hgHubConnect</li>
<li id="hgHub_do_firstDb"><code>hgHub_do_firstDb=1</code>
uses the first database in genomes.txt</li>
<li id="hgHub_do_redirect"><code>hgHub_do_redirect=on</code>
redirects the attached hub to the Gateway page</li>
</ul>
<p>
By pointing the URL to <code>/cgi-bin/hgHubConnect</code> the hub would be connected to
and then redirect to the Gateway page to display the page in <code>genomes.txt</code> defined
by the <code>htmlPath ../newOrg.html</code> line. Here is an example:</p>
<pre><code><a href="http://genome.ucsc.edu/cgi-bin/hgHubConnect?hgHub_do_redirect=on&hgHubConnect.remakeTrackHub=on&hgHub_do_firstDb=1&hubUrl=http://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubAssembly/plantAraTha1/hub.txt"
target="_blank">http://genome.ucsc.edu/cgi-bin/hgHubConnect?hgHub_do_redirect=on&hgHubConnect.remakeTrackHub=on&hgHub_do_firstDb=1&<strong>hubUrl=http://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubAssembly/plantAraTha1/hub.txt</strong></a></code>
</pre>
<h6 id="genome">Use <code>genome=</code> for assembly hubs</h6>
<p>
There is also another way to connect an assembly hub using the <code>genome=</code> to arrive at
the Gateway page if you know the genome in <code>genomes.txt</code> that you wish to display.
For instance, the below link skips the redirection step, where your
<code>hub.txt</code> contact email gets displayed, by pointing to the
<code>/cgi-bin/hgGateway</code> page:</p>
<pre><code><a href="http://genome.ucsc.edu/cgi-bin/hgGateway?genome=araTha1&hubUrl=http://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubAssembly/plantAraTha1/hub.txt"
target="_blank"><strong>http://genome.ucsc.edu/cgi-bin/hgGateway?genome=araTha1&hubUrl=http://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubAssembly/plantAraTha1/hub.txt</strong></a> </code></pre>
 
<a name="trackHubLoading"></a>
<h3>Attaching a track hub to an assembly hub</h3>
<h6>How do I attach a track hub to an assembly hub?</h6>
<p>
It is possible to attach a track hub to an assembly hub via the URL using a combination of the
<code>hubUrl=</code> and <code>genome=</code> URL parameters. For example, using the following
assembly hub and track hub:
</p>
<pre>
<b>Arabidopsis thaliana assembly hub</b>
https://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubPlants/cshl2013/hub.txt
</pre>
<pre>
<b>ReMap2020 Reg. Atlas track hub</b>
https://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubRemap/remap.txt
</pre>
<p>
The combination of the two hubs, along with <code>genome=araTha1</code>, allows for the creation of
the following URL that can load both hubs on the Genome Browser.
</p>
<pre>
<a href="/cgi-bin/hgTracks?genome=araTha1&hubUrl=https://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubPlants/cshl2013/hub.txt&hubUrl=https://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubRemap/remap.txt" target="_blank">https://genome.ucsc.edu/cgi-bin/hgTracks?genome=araTha1&hubUrl=https://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubPlants/cshl2013/hub.txt&hubUrl=https://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubRemap/remap.txt</a>
</pre>

<a name="genArkTrackHub"></a>
<h3>Creating a track hub for a GenArk assembly</h3>
<h6>Can I add my own tracks to a GenArk assembly hub?</h6>
<p>
When creating a track hub for a GenArk assembly, there is no need to do any attaching of the
assembly hub itself via the <code>hubUrl=</code> URL parameter. GenArk hubs will automatically
attach themselves if the track hub mentions the GCA_ or GCF_ name identifier of the assembly hub.
Simply load the track hub on the Genome Browser, and the assembly hub will automatically appear. For
example, the following <a href="https://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubGenArkExample/genArkTrackHub/hub.txt"
target="_blank">example track hub</a> will load an additional track for the pig (GCA_002844635.1)
GenArk assembly.
</p>
<p>
To create a link to the track hub that references a GenArk assembly, the
<code>genome=GCA_002844635.1</code> and <code>hubUrl=</code> URL parameters can be used like in the
following example:
</p>
<pre>
<a href="/cgi-bin/hgTracks?genome=GCA_002844635.1&hubUrl=https://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubGenArkExample/genArkTrackHub/hub.txt" target="_blank">https://genome.ucsc.edu/cgi-bin/hgTracks?genome=GCA_002844635.1&hubUrl=https://genome.ucsc.edu/goldenPath/help/examples/hubExamples/hubGenArkExample/genArkTrackHub/hub.txt</a>
</pre>
<p>
You can also read a blog post,
<a href="https://genome-blog.soe.ucsc.edu/blog/2021/08/13/sharing-data-with-sessions-and-urls"
target="_blank">Sharing Data with Sessions and URLs</a>, about how to
<a href="https://genome-blog.soe.ucsc.edu/blog/2021/08/13/sharing-data-with-sessions-and-urls/#assemblyHubs"
target="_blank">build URLs to track hubs on assembly hubs</a> or see further examples of
<a href="https://genome-blog.soe.ucsc.edu/blog/2021/08/13/sharing-data-with-sessions-and-urls/#GenArk"
target="_blank">GenArk hubs</a> loaded with custom data.</p>

<a name="genes"></a>
<a name="link3"></a>
<h2>Linking to gene-specific information</h2>
<h6>How do I link to a specific gene or specific gene description page?</h6>
<p>
To jump directly to a gene's position on the Genome Browser, set the position parameter in the
URL to a gene symbol (e.g., TP53, MTOR, KRAS) and add the parameter
<code>singleSearch=knownCanonical</code>. For example, the following link will open the
Genome Browser for the hg19 human assembly at the position of TP53 on the knownCanonical dataset
</p>
<p><code><a href="../cgi-bin/hgTracks?db=hg19&singleSearch=knownCanonical&position=TP53" target="_blank">http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&singleSearch=knownCanonical&position=TP53</a></code></p>
<p>
<a name="gene"></a>
You can also link directly to gene description pages from the URL. Instead of a position search,
gene descriptions use the <code>hgg_gene=</code> URL parameter. The following URL connecting
to 'hgGene' will open up the Genome Browser description page containing protein function,
expression profile, and links to additional information for the gene TP53.
</p>
<p><code><a href="../cgi-bin/hgGene?db=hg19&hgg_gene=TP53" target="_blank">http://genome.ucsc.edu/cgi-bin/hgGene?db=hg19&hgg_gene=TP53</a> </code></p>



<a name="hgsid"></a>
<h2>The <em>hgsid</em> parameter</h2>
<h6>What is the hgsid parameter and should I include it in Genome Browser links?</h6>
<p>The hgsid is a temporary user ID that stores setting and custom track information in the URL.
Including it in any shared URLs is a privacy concern, and it should be removed when
constructing any links to the Genome Browser. Most significantly, it will change after
you share it. Anyone using it will see that last thing you did, not what you thought
you were sharing.  Creating
<a href="../goldenPath/help/hgSessionHelp.html#Create">Saved Sessions</a> is the recommended way to
share Genome Browser information.
</p>

<a name="moreInfo"></a>
<h2>Additional URL parameters</h2>
<h6>Are there any more resources for URL and link parameters?</h6>
<p>
For more information, please see our <a href=../goldenPath/help/customTrack.html#optParams>section
on URL parameters for custom tracks</a>. If you cannot find what 
you are looking for, please contact our active mailing list by emailing 
<a href="mailto:&#103;e&#110;o&#109;&#101;&#64;&#115;&#111;&#101;.&#117;&#99;&#115;c.
e&#100;u">&#103;e&#110;o&#109;&#101;&#64;&#115;&#111;&#101;.&#117;&#99;&#115;c.e&#100;u</a>. All 
messages sent to that address are publicly archived. If your question includes sensitive data, 
you may send it instead to <a href="mailto:&#103;e&#110;&#111;&#109;&#101;&#45;&#119;w&#119;
&#64;&#115;&#111;&#101;.u&#99;&#115;c.&#101;&#100;&#117;">
&#103;e&#110;&#111;&#109;&#101;&#45;&#119;w&#119;&#64;&#115;&#111;&#101;.
u&#99;&#115;c.&#101;&#100;&#117;</a></p>

<a name="Videos"></a>
<p>
</p>
<h2>Videos</h2>

<a name="LinkVid1"></a>
<p>
</p>
<h3>Video demonstration: Links: Understanding the URL </h3>
<p>

</p>
<!--#if expr="${SERVER_NAME} = /-china/" -->
    <a href='../videos/p6TPvHDmpJk.mp4' title="UCSC Genome Browser Links">
    <img src=../images/videoIcon.png alt='UCSC Genome Browser Basics Part 1 tutorial video'></a>
    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
    Visit our <a href="/videos/" target="_blank">Video Page</a>.
<!--#else -->
    <iframe width="560" height="315" src="https://www.youtube.com/embed/p6TPvHDmpJk?rel=0"
    frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
    allowfullscreen></iframe>
    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
    Visit our
    <a href="https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos"
    target="_blank">YouTube channel</a>.
</p>
<!--#endif -->
</p>


<a name="LinkVid2"></a>
<p>
</p>
<h3>Video demonstration: Links: Jump into genes </h3>
<p>
<!--#if expr="${SERVER_NAME} = /-china/" -->
    <a href='../videos/xmcyz2GyRKA.mp4' title="UCSC Genome Browser Links">
    <img src=../images/videoIcon.png alt='UCSC Genome Browser Jump into genes'></a>
    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
    Visit our <a href="../videos/" target="_blank">Video Page</a>.
<!--#else -->
    <iframe width="560" height="315" src="https://www.youtube.com/embed/xmcyz2GyRKA?rel=0"
    frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
    allowfullscreen></iframe>

    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
    Visit our
    <a href="https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos"
    target="_blank">YouTube channel</a>.
<!--#endif -->
</p>
 

<a name="LinkVid3"></a>
<br>
<h3>Video demonstration: Links: Composites, custom tracks, spreadsheets</h3>
<br>
<p>
<!--#if expr="$SERVER_NAME = /-china/" -->
    <a href='../videos/Yy4UmepYs2Q.mp4' title="UCSC Genome Composites">
    <img src=../images/videoIcon.png alt='UCSC Genome Browser Jump into genes'></a>
    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
    Visit our <a href="../videos/" target="_blank">Video Page</a>.
<!--#else -->
    <iframe width="560" height="315" src="https://www.youtube.com/embed/Yy4UmepYs2Q?rel=0"
    frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
    allowfullscreen></iframe>

    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
    Visit our
    <a href="https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos"
    target="_blank">YouTube channel</a>.
<!--#endif -->
</p>

<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->
