<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=315 HEIGHT=185>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://dolphin.lab.nig.ac.jp/"
    TARGET=_blank>
    <IMG SRC="/images/Oryzias_latipes.jpg" WIDTH=300 HEIGHT=123 ALT="Medaka"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Oryzias latipes</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of
      (<A HREF="http://mlab.cb.k.u-tokyo.ac.jp/index_eng.html" 
      TARGET=_blank>Shinichi Morishita</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Oct. 2005 <em>Oryzias latipes</em> draft assembly 
(MEDAKA1) was produced by the
<A HREF="http://dolphin.lab.nig.ac.jp/"
TARGET=_blank>Institute of Genetics (NIG)</A>
and the <A HREF="http://medaka.utgenome.org/"
TARGET=_blank>University of Tokyo.</A>

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
Medaka genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>chr23</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of chromosome 23</TD>
    </TR>
    <TR><TD VALIGN=Top>chr3:1-1,000,000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of chr 3</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr3:19,840,000+100,000</TD>
       	<TD WIDTH=14></TD>
	<TD VALIGN=Top>Displays a region of chr 3 that spans 100,000 bases, starting with position 19,840,000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>AY158893</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number AY158893</TD>
    </TR>
    <TR><TD VALIGN=Top>BJ733542</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		BJ733542 on chr 6</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>pseudogene mRNA</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>zhang</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zhang</TD>
    </TR>
    <TR><TD VALIGN=Top>Taguchi,T.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author T. Taguchi</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Taguchi T</em>, internally it
	uses the format <em>Taguchi,T.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
The genome has been sequenced to 6.7X coverage.
There is 724,218,853 bp on 24 chromosomes, chr1-chr24.
There are 7,299 scaffolds comprised of 36,494 contigs of unplaced sequence
of 118,243,491 bp (plus 1,021,750 N gap) displayed on the virtual
chromosome chrUn.  The contigs are spaced with a 10 bp gap, and the
scaffold gaps are 100 bp.
</P>
<P>
The medaka mitochondrial sequence is also available as the virtual chromosome 
&quot;chrM&quot;.  (genbank accession: NC_004387)
</P>

<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/oryLat1/"
TARGET=_blank>FTP server</A> or the
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#medaka"
TARGET=_blank>Downloads</A> page. 

These data have
<A HREF="http://medaka.utgenome.org/#policy"
TARGET=_blank>specific conditions for use</A>. 

The Medaka browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#medaka_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
