<H2>Description</H2>
<P>
The UCSC Genes track is a set of gene predictions based on data from RefSeq, GenBank,
and the <a href="hgTrackUi?db=mm10&g=tRNAs">tRNA Genes</a> track. The track
includes both protein-coding genes and
non-coding RNA genes. Both types of genes can produce non-coding transcripts, but non-coding
RNA genes do not produce protein-coding transcripts. This is a moderately conservative set of
predictions.  Transcripts of protein-coding genes require the support of one RefSeq RNA, or one
GenBank RNA sequence plus at least one additional line of evidence. Transcripts of non-coding RNA
genes require the support of one Rfam or tRNA prediction.  Compared to RefSeq, this gene set has
generally about 10% more protein-coding genes, approximately four times as many putative non-coding
genes, and about twice as many splice variants.</P>

<H2>Display Conventions and Configuration</H2>
<P>
This track in general follows the display conventions for
<A HREF="../goldenPath/help/hgTracksHelp.html#GeneDisplay">gene prediction tracks</A>. The exons
for putative non-coding genes and untranslated regions are represented by relatively thin blocks,
while those for coding open reading frames are thicker. The following color key is used:</P>

<ul>
<li><span style="font-weight: bold;">Black</span> -- feature has a corresponding entry in the Protein
 Data Bank (PDB)</li>
<li><span style="color: #0C0C78; font-weight: bold;">Dark blue</span> -- transcript has been reviewed
 or validated by either the RefSeq, SwissProt or CCDS staff</li>
<li><span style="color: #5050A0; font-weight: bold;">Medium blue</span> -- other RefSeq transcripts</li>
<li><span style="color: #8282D2; font-weight: bold;">Light blue</span> -- non-RefSeq transcripts</li>
</ul>

<P>
This track contains an optional <A HREF="../goldenPath/help/hgCodonColoring.html">codon coloring
feature</A> that allows users to quickly validate and compare gene predictions.</P>

<H2>Methods</H2>
<P>
The UCSC Genes are built using a multi-step pipeline:</P>
<OL>
<LI>RefSeq and GenBank RNAs are aligned to the genome with BLAT, keeping only the best alignments
for each RNA.  Alignments are discarded if they do not meet certain sequence identity and coverage
filters.  All sequences must align with high (98%) identity. The sequence coverage must be at least
90% for shorter sequences (those with 2500 or fewer bases), with the coverage threshold
progressively relaxed for longer sequences.</LI>
<LI>Alignments are broken up at non-intronic gaps, with small isolated fragments thrown out.</LI>
<LI>A splicing graph is created for each set of overlapping alignments. This graph has an edge
for each exon or intron, and a vertex for each splice site, start, and end. Each RNA that
contributes to an edge is kept as evidence for that edge. Gene models from the Consensus CDS project
(CCDS) are also added to the graph.</LI>
<LI>A similar splicing graph is created in the human, based on human RNA and ESTs. If the human
graph has an edge that is orthologous to an edge in the mouse graph, that is added to the evidence
for the mouse edge.</LI>
<LI>If an edge in the splicing graph is supported by two or more mouse ESTs, it is added as
evidence for the edge.</LI>
<LI>If there is an Exoniphy prediction for an exon, that is added as evidence.</LI>
<LI>The graph is traversed to generate all unique transcripts. The traversal is guided by the
initial RNAs to avoid a combinatorial explosion in alternative splicing. All RefSeq transcripts are
output. For other multi-exon transcripts to be output, an edge supported by at least one additional
line of evidence beyond the RNA is required.  Single-exon genes require either two RNAs or two
additional lines of evidence beyond the single RNA.</LI>
<LI>Alignments are merged in from the mm10
<A HREF="hgTrackUi?db=mm10&g=tRNAs">tRNA Genes</A> track and from Rfam
in regions that are syntenic with the hg19 human genome.</LI>
<LI>Protein predictions are generated. For non-RefSeq transcripts, we use the txCdsPredict program to
determine if the transcript is protein-coding, and if so, the locations of the start and stop codons.
The program weighs as positive evidence the length of the protein, the presence of a Kozak consensus
sequence at the start codon, and the length of the orthologous predicted protein in other species.
As negative evidence it considers nonsense-mediated decay and start codons in any frame upstream of
the predicted start codon. For RefSeq transcripts, the RefSeq protein prediction is used directly
instead of this procedure.  For CCDS proteins, the CCDS protein is used directly.</LI>
<LI>The corresponding UniProt protein is found, if any.</LI>
<LI>The transcript is assigned a permanent &quot;uc&quot; accession. If the transcript was not in
the previous release of UCSC Genes, the accession ends with the suffix &quot;.1&quot; indicating
that this is
the first version of this transcript.  If the transcript is identical to some transcript in the
previous release of UCSC Genes, the accession is re-used with the same version number. If the
transcript is not identical to any transcript in the previous release but it overlaps a similar
transcript with a compatible structure, the previous accession is re-used with the version number
incremented.</LI>
</OL>

<H2>Related Data</H2>
<P>
The UCSC Genes transcripts are annotated in numerous tables, each of which is also available as a
<a href="http://hgdownload.cse.ucsc.edu/goldenPath/mm10/database/">downloadable file</a>.  These
include tables that link UCSC Genes transcripts to external datasets (such as
<I>knownToLocusLink</I>, which maps UCSC Genes transcripts to Entrez identifiers, previously known
as Locus Link identifiers), and tables that detail some property of UCSC Genes transcript sequences
(such as <I>knownToPfam</I>, which identifies any Pfam domains found in the UCSC Genes
protein-coding transcripts). One can see a full list of the associated tables in the
<a href="/cgi-bin/hgTables">Table Browser</a> by selecting UCSC Genes at the <B>track</B> menu;
this list is then available at the <B>table</B> menu. Note that some of these tables refer to UCSC
Genes by its former name of <I>Known Genes</I>, sometimes abbreviated as <I>known</I> or <I>kg</I>.
While the complete set of annotation tables is too long to describe, some of the more important
tables are described below.</P>
<UL>
<LI> <B>kgXref</B> identifies the RefSeq, SwissProt, Rfam, or tRNA sequences (if any) on which each
transcript was based.</LI>
<LI> <B>knownToRefSeq</B> identifies the RefSeq transcript that each UCSC Genes transcript is most
closely associated with. That RefSeq transcript is either the RefSeq on which the UCSC Genes
transcript was based, if there is one, or it's the RefSeq transcript that the UCSC Genes transcript
overlaps at the most bases.</LI>
<LI> <B>knownGeneMrna</B> contains the mRNA sequence that represents each UCSC Genes transcript. If
the transcript is based on a RefSeq transcript, then this table contains the RefSeq transcript,
including any portions that do not align to the genome.</LI>
<LI> <B>knownGeneTxMrna</B> contains mRNA sequences for each UCSC Genes transcript.  In contrast to
the sequencess in knownGeneMrna, these sequences are derived by obtaining the sequences for each exon
from the reference genome and concatenating these exonic sequences.</LI>
<LI> <B>knownGenePep</B> contains the protein sequences derived from the knownGeneMrna transcript
sequences. Any protein-level annotations, such as the contents of the knownToPfam table, are based
on these sequences.</LI>
<LI> <B>knownGeneTxPep</B> contains the protein translation (if any) of each mRNA sequence in
knownGeneTxMrna.</LI>
<LI> <B>knownIsoforms</B> maps each transcript to a <EM>cluster ID</EM>, a cluster of isoforms of
the same gene.</LI>
<LI> <B>knownCanonical</B> identifies the canonical isoform of each cluster ID, or gene. Generally,
this is the longest isoform.</LI>
</UL>

<H2>Credits</H2>
<P>
The UCSC Genes track was produced at UCSC using a computational pipeline developed by Jim Kent,
Chuck Sugnet, Melissa Cline and Mark Diekhans. It is based on data from NCBI
<A HREF="http://www.ncbi.nlm.nih.gov/refseq/" TARGET=_blank>RefSeq</A>,
<A HREF="http://www.uniprot.org/" TARGET=_blank>UniProt</A>
(including TrEMBL and TrEMBL-NEW),
<A HREF="http://www.ncbi.nlm.nih.gov/CCDS/CcdsBrowse.cgi" TARGET=_blank>CCDS</A>, and
<A HREF="http://www.ncbi.nlm.nih.gov/genbank/"
TARGET=_blank>GenBank</A> as well as data from
<A HREF="http://rfam.xfam.org/" target="_blank">Rfam</A> and the
<A HREF="http://lowelab.ucsc.edu/" target="_blank">Todd Lowe lab</A>.
Our thanks to the people running these databases and to the scientists worldwide who have made
contributions to them.</P>

<H2>References</H2>

<p>
Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Wheeler DL.
<a href="http://nar.oxfordjournals.org/content/32/suppl_1/D23.full" target="_blank">
GenBank: update</a>.
<em>Nucleic Acids Res</em>. 2004 Jan 1;32(Database issue):D23-6.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/14681350" target="_blank">14681350</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC308779" target="_blank">PMC308779</a>
</p>

<p>
Chan PP, Lowe TM.
<a href="http://nar.oxfordjournals.org/content/37/suppl_1/D93.full" target="_blank">
GtRNAdb: a database of transfer RNA genes detected in genomic sequence</a>.
<em>Nucleic Acids Res</em>. 2009 Jan;37(Database issue):D93-7.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/18984615" target="_blank">18984615</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2686519" target="_blank">PMC2686519</a>
</p>

<p>
Gardner PP, Daub J, Tate J, Moore BL, Osuch IH, Griffiths-Jones S, Finn RD, Nawrocki EP, Kolbe DL,
Eddy SR <em>et al</em>.
<a href="http://nar.oxfordjournals.org/content/39/suppl_1/D141.full" target="_blank">
Rfam: Wikipedia, clans and the "decimal" release</a>.
<em>Nucleic Acids Res</em>. 2011 Jan;39(Database issue):D141-5.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/21062808" target="_blank">21062808</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3013711" target="_blank">PMC3013711</a>
</p>

<p>
Hsu F, Kent WJ, Clawson H, Kuhn RM, Diekhans M, Haussler D.
<a href="http://bioinformatics.oxfordjournals.org/content/22/9/1036.full"
target="_blank">
The UCSC Known Genes</a>.
<em>Bioinformatics</em>. 2006 May 1;22(9):1036-46.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/16500937" target="_blank">16500937</a>
</p>


<p>
Kent WJ.
<a href="http://genome.cshlp.org/content/12/4/656.full" target="_blank">
BLAT--the BLAST-like alignment tool</a>.
<em>Genome Res</em>. 2002 Apr;12(4):656-64.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/11932250" target="_blank">11932250</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC187518" target="_blank">PMC187518</a>
</p>

<p>
Lowe TM, Eddy SR.
<a href="http://nar.oxfordjournals.org/content/25/5/0955.full" target="_blank">
tRNAscan-SE: a program for improved detection of transfer RNA genes in genomic sequence</a>.
<em>Nucleic Acids Res</em>. 1997 Mar 1;25(5):955-64.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/9023104" target="_blank">9023104</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC146525" target="_blank">PMC146525</a>
</p>

<p>
UniProt Consortium.
<a href="http://nar.oxfordjournals.org/content/40/D1/D71.full" target="_blank">
Reorganizing the protein space at the Universal Protein Resource (UniProt)</a>.
<em>Nucleic Acids Res</em>. 2012 Jan;40(Database issue):D71-5.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/22102590" target="_blank">22102590</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3245120" target="_blank">PMC3245120</a>
</p>
