<!--News Section----------------------------------------------->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list.</P>

    <!-- start news -->
        <p>
        <font face="courier" size="3"><b>2 July 2013
	- Assembly Data Hubs</b></font>
        </p>
	<p>
	We are pleased to announce a new feature in the UCSC Genome Browser:
	Assembly Data Hubs.
	</p>
	<p>
	Assembly Data Hubs extend the functionality of Track Data Hubs to
	assemblies that are not hosted natively on the Browser. Assembly Data
	Hubs were developed to address the increasing need for researchers to
	annotate sequence for which UCSC does not provide an annotation
	database. They allow researchers to include the underlying reference
	sequence, as well as data tracks that annotate that sequence. Sequence
	is stored in the UCSC twoBit format, and the annotation tracks are
	stored in the same manner as Track Data Hubs.
	</p>
        <p>
	To read more about using Assembly Data Hubs, or about creating your
	own, please see these guides:
	</p>
	<p>
	<a href="http://genome.ucsc.edu/goldenPath/help/hgTrackHubHelp.html">
	http://genome.ucsc.edu/goldenPath/help/hgTrackHubHelp.html</a>
	</p>
	<p>
	<a href="http://genomewiki.ucsc.edu/index.php/Assembly_Hubs">
	http://genomewiki.ucsc.edu/index.php/Assembly_Hubs</a>
	</p>
	<p>
	Many thanks to Hiram Clawson, Brian Raney, and Pauline Fujita for their work on this feature.
	</p>
	
	<p>
        <font face="courier" size="3"><b>2 July 2013
	- New Gene Alleles feature on the details pages of UCSC genes</b></font>
        </p>
	<p>
	We are excited to announce the release the new Gene Alleles feature on
	the details pages of UCSC genes. For each protein-coding gene in the
	UCSC Genes track, variant data from the 2,184 (per autosome) phased
	chromosomes from Phase 1 of the 1000 Genomes Project have been
	distilled into "gene haplotype alleles."  Each haplotype allele is a
	distinct set of variants found on at least one of the 1000 Genomes
	subject chromosomes. For each haplotype, we display its frequency in
	the 1000 Genomes populations and indicate the frequency with which it
	occurs homozygously in each population. Unexpected frequencies of
	occurrence may thus be used to identify alleles that merit further
	study. Predicted protein sequence for common haplotypes can also be
	displayed, allowing differences among alleles to be used to identify
	differences at the amino acid level.
	</p>
	<p>
	To see the Gene Alleles feature, go to the details page for any
	protein-coding gene in the UCSC Genes track and click the "Gene
	Alleles" link in the top right cell of the "Page Index" matrix. This
	will take you to the "Common Gene Haplotype Alleles" section. For
	more information on this feature please see this help doc:
	</p>
	<p>
	<a href="http://genome.ucsc.edu/goldenPath/help/haplotypes.html">
	http://genome.ucsc.edu/goldenPath/help/haplotypes.html</a>
	</p>
	<p>
	Many thanks to Tim Dreszer, Brooke Rhead and Pauline Fujita for their work on this feature.
	</p>

 	<p>
 	<font face="courier" size="3"><b>28 June 2013
	- New UCSC Genes Track Released for GRCh37/hg19</b></font>
	</p>

	<p>
	We're happy to announce the release of an updated UCSC Genes
	track for the GRCh37/hg19 human Genome Browser. 
	The new release has 82,960 total transcripts, compared with 
	80,922 in the previous version. The total number of canonical 
	genes has increased from 31,227 to 31,848. Comparing the new 
	gene set with the previous version:
	<ul>
	<li>74,657 transcripts did not change between versions.
	<li>88 transcripts were not carried forward to the new version.
	<li>5,556 transcripts are "compatible" with those in the previous set, 
	meaning that the two transcripts show consistent splicing. 
	In most cases, the old and new transcripts differ in the 
	lengths of their UTRs.
	<li>621 transcripts overlap with those in the previous set but 
	do not show consistent splicing, i.e., they contain overlapping 
	introns with differing splice sites.
	</ul>
	</p>
	<p>
	More details about the new UCSC Genes track can be found on
	the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=knownGene"
	>UCSC Genes track description page</a>.
	</p>
	<p>
	Many thanks to Brian Raney, Jim Kent, and Luvina Guruvadoo 
	for their work on this track!
	</p> 
 
        <p>
	<hr>

        <p>
        <font face="courier" size="3"><b>27 June 2013
        - New Official European Mirror Released</b></font> Announcing
        an official mirror for European users, with automatic
        redirection.
        <a href="goldenPath/newsarch.html#062713">Read more</a>.
        </p>

	<p>
	<font face="courier" size="3"><b>26 June 2013
	- Suggestion Box Released:</b></font>
	Due to popular demand, we have created a suggestion box for track, assembly,
	tool, and other feature requests.
	<a href="goldenPath/newsarch.html#062613">Read more</a>.</p>

        <!-- start archives -->
	<hr>


    <!-- staged for archive use
       <p>
        <font face="courier" size="3"><b>2 July 2013
        - Assembly Data Hubs</b></font>
        </p>
	<p>
        <a href="goldenPath/newsarch.html#070213">Read more</a>.
        </p>

        <p>
        <font face="courier" size="3"><b>2 July 2013
        - New Gene Alleles feature on the details pages of UCSC genes</b></font>
        </p>
        <a href="goldenPath/newsarch.html#070213b">Read more</a>.
        </p>


    -->

            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
