<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                         <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD>
                  <TD>
                  <P>
                  To receive announcements of new genome 
                  assembly releases, new software features, updates and 
                  training seminars by email, subscribe to the
                  <A HREF="https://lists.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                  TARGET=_blank>genome-announce</A> mailing list.</P>

            
            <!-- start news -->

                <FONT FACE="courier" SIZE="3"><B>15 May 2012 - Naked Mole-Rat
		Assembly Now Available in Genome Browser </B></FONT>
                <P>We are pleased to announce the release of a Genome Browser 
		for the July 2011 naked mole-rat, <em>Heterocephalus glaber</em>
		(BGI version HetGla1.0, UCSC version hetGla1). This draft assembly,  
		produced by the 
		<A HREF="http://en.genomics.cn/" 
		TARGET=_blank>Beijing Genomics Institute</A>, has been sequenced using
		Illumina reads with a sequencing depth of >20x. There are 
		39,266 scaffolds with a total size of 2,643,961,837 bases. 
		The N50 size of the scaffolds is 1,321,980,918 bases with 502 
		scaffolds larger, and 38,764 scaffolds smaller. The mitochondrial 
		sequence is also available as the virtual chromosome "chrM".
		<P>
		Bulk downloads of the sequence and annotation data are available 
		via the Genome Browser 
		<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/hetGla1/">FTP server</A> or the 
		<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#naked_mole-rat" 
		TARGET=_blank>Downloads</A> page. These data have 
		<A HREF="../goldenPath/credits.html#naked_mole-rat_use" 
		>specific conditions for use</A>. The naked mole-rat browser 
		annotation tracks were generated by UCSC and collaborators worldwide. 
		See the 
		<A HREF="../goldenPath/credits.html#naked_mole-rat_credits" 
		>Credits</A> page for a detailed list of the organizations and 
		individuals who contributed to the success of this release. 
                <HR>

		<FONT FACE="courier" SIZE="3"><B>8 May 2012 - OpenHelix
                to Conduct Free Genome Browser Webinars in May</B></FONT>
                <P>
                <A HREF="http://www.openhelix.com" TARGET=_blank>OpenHelix</A> 
		is conducting two free webinars on the UCSC Genome Browser:
                <UL>
                <LI><A HREF="http://www.openhelix.com/cgi/webinarInfo.cgi?id=43"
		TARGET=_blank>UCSC Genome Browser: An Introduction</A> 
		(Thursday, May 17, 1pm EDT) -- designed for new users
                of the UCSC Genome Browser and those who want to improve their 
                skills at basic navigation and display.
                <LI><A HREF="http://www.openhelix.com/cgi/webinarInfo.cgi?id=44"
		TARGET=_blank>UCSC Table Browser and Custom Tracks</A>
		(Thursday, May 24, at 1pm EDT) -- advanced Genome Browser 
		topics, including creating Custom Tracks and using the Table 
		Browser.
                </UL>
                <P>
                The webinars will last approximately 1hr 15min.
                <P>
                Seating is limited and                 
		<A HREF="http://www.openhelix.com/cgi/webinar.cgi"
                TARGET=_blank>registration is required</A>. Slides may be 
                downloaded from the <A HREF="http://www.openhelix.com/ucsc"
                TARGET=_blank>OpenHelix website</A> prior to the webinar.
                <P>
                OpenHelix provides training materials and programs on hundreds 
		of free, publicly accessible bioinformatics and genomics 
		resources. To browse their entire collection of UCSC genomics 
		tutorials, go to the OpenHelix 
		<A HREF="http://www.openhelix.com/ucsc" TARGET=_blank>UCSC 
		training page</A>.
                <P>
                <HR>
            
		

	   <!-- start archives -->
	
		<FONT FACE="courier" SIZE="3"><B>2 May 2012 - OpenHelix
                Releases Updated Genome Browser Tutorials: </B></FONT>
		<A HREF="http://www.openhelix.com" TARGET=_blank>OpenHelix</A>
                has released updated versions of two Genome Browser tutorial
                suites.
		<A HREF="goldenPath/newsarch.html#050212">Read more</A>.
		<P>
                <FONT FACE="courier" SIZE="3"><B>18 April 2012 - GRC Planning Phase 
		for GRCh38 Human Reference Assembly: </B></FONT>
		The <A HREF="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/"
		TARGET=_blank>Genome Reference Consortium</A> (GRC) plans to
		update the human reference assembly to GRCh38 in the summer of 
		2013. 
                <A HREF="goldenPath/newsarch.html#041812">Read more</A>.
		<P>
==> <a href="goldenPath/newsarch.html">News Archives</a>

                    </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
