<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome browser, genome assembly, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, fruitfly, fugu, comparative genomics">
	<TITLE>UCSC Genome Bioinformatics Home</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">

</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<!--HOTLINKS BAR-->
    <TR><TD COLSPAN=2 HEIGHT=40>
    <TABLE BGCOLOR="000000" CELLPADDING="1" CELLSPACING="1" WIDTH="99%" HEIGHT="27">
        <TR BGCOLOR="2636D1"><TD VALIGN="middle">
        <TABLE BORDER=0 CELLSPACING=0 CELLPADDING=0 BGCOLOR="2636D1" HEIGHT="24">
    	    <TR><TD VALIGN="middle"><FONT color="#89A1DE">&nbsp; 
 	    <A HREF="../cgi-bin/hgGateway" class="topbar">
			Genome Browser</a> &nbsp; - &nbsp; 
 	    <A HREF="../cgi-bin/hgNear" class="topbar">
			Family Browser</a> &nbsp; - &nbsp; 
	    <A HREF="../cgi-bin/hgBlat" class="topbar">
			Blat</a> &nbsp; - &nbsp; 
	    <A HREF="../cgi-bin/hgText" class="topbar">
			Table Browser</a> &nbsp; - &nbsp; 
	    <A HREF="/FAQ/" class="topbar">
			FAQ</a> &nbsp; - &nbsp; 
	    <A HREF="goldenPath/help/hgTracksHelp.html" class="topbar">
			Help</a> &nbsp; </FONT></TD>
        </TR></TABLE>
    </TD></TR></TABLE>
    </TD></TR>

    <TR HEIGHT="100%" VALIGN="top">
    <TD ROWSPAN="3" VALIGN="top" HEIGHT="100%"> 
    <!-- outer table is to align leftbar & content boxes at top of page -->
    <TABLE WIDTH="100%" BORDER=0 CELLPADDING=0 CELLSPACING=0 MARGIN=0>
        <TR><TD><IMG SRC="images/DOT.gif" ALT="" HEIGHT="7"></TD></TR>
        <TR HEIGHT="100%"><TD HEIGHT="100%">
        <!--outer table is for black border-->
        <TABLE WIDTH="100%" BGCOLOR="#000000" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>
            <TABLE WIDTH="100%" BORDER=0 CELLSPACING="0" CELLPADDING=10>
                <TR HEIGHT="100%">
	        <TD BGCOLOR="738ED3" VALIGN="top" HEIGHT="100%" > 
	        <A CLASS="leftbar" HREF="../cgi-bin/hgGateway">
			Genome Browser</A><HR>
	        <A CLASS="leftbar" HREF="../cgi-bin/hgNear">
			Family Browser</A><HR>
	        <A CLASS="leftbar" HREF="../cgi-bin/hgBlat">
			Blat</A><HR>
	        <A CLASS="leftbar" HREF="../cgi-bin/hgText">
			Tables</A><HR>
	        <A CLASS="leftbar" HREF="downloads.html">
			Downloads</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/releaseLog.html">
			Release Log</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/customTracks/custTracks.html">
			Custom Tracks</A><HR>
	        <A CLASS="leftbar" HREF="mirror.html">
			Mirrors</A><HR>
  	        <A CLASS="leftbar" HREF="http://genome-archive.cse.ucsc.edu">
			Archives</A><HR>
	        <A CLASS="leftbar" HREF="goldenPath/credits.html">
			Credits</A><HR>
	        <A CLASS="leftbar" HREF="cite.html">
			Cite Us</A><HR>
	        <A CLASS="leftbar" HREF="license/">
			Licenses</A><HR>
	        <A CLASS="leftbar" HREF="jobs.html">
			Jobs</A><HR>
  	        <A CLASS="leftbar" HREF="contacts.html">
			Contact Us</A><HR>
	        <BR>
	        <IMG SRC="images/DOT.gif" ALT="" HEIGHT=1200 WIDTH=1>
   	    </TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
    </TD>

<!--Content Tables-->  
    <TD VALIGN="top">
    <TABLE CELLPADDING=5 BORDER=0><TR><TD>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; About the UCSC Genome Bioinformatics Site</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
		    This site contains the reference sequence for the human and <em>C. elegans</em> 
		    genomes and working drafts for the mouse, rat, Fugu, <EM>Drosophila</EM>, 
		    <em>C. briggsae</em>, and SARS genomes.  It also contains the
		    CFTR (cystic fibrosis) region in 13 species.  
		    <P>
		    We encourage you to explore these sequences with our tools. The Genome 
		    Browser zooms and scrolls over chromosomes, showing the work of annotators 
		    worldwide.  The Family Browser shows expression, homology and other 
		    information on groups of genes that can be related in many ways.  
		    The Table Browser provides convenient access to the underlying database.  
		    Blat quickly maps your sequence to the genome.
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
		    <!--cheap trick to get background on row to be continuous-->
		    <TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
  		        <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
		        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
		    </TD></TR></TABLE>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>

	            <P>
		    <FONT FACE="courier" SIZE=3><B>Oct. 31, 2003</B></FONT><BR>
		    We have added the <EM>Drosophila melanogaster</EM> (fruitfly) 
		    assembly to the growing collection of genomes available in 
		    the UCSC Genome Browser and Blat servers.  Release 3.1 
		    (Jan. 2003) of the <EM>Drosophila</EM> annotated genome 
		    sequence was provided by the
		    <a HREF="http://www.fruitfly.org/" TARGET=_blank>Berkeley 
		    Drosophila Genome Project</a> (BDGP). The 116.8 Mb euchromatic 
		    sequence - which is virtually gap-free and of high accuracy - 
		    contains six euchromatic chromosome arms represented by 13 
		    scaffolds with a total of 37 sequence gaps. The sequence quality 
		    of this release has an estimated error rate of less than one 
		    in 100,000 base pairs in the unique portion of the sequence, 
		    and less than one in 10,000 base pairs in the repetitive portion. 
		    The Release 3.1 sequence was reannotated using the 
		    <a HREF="http://www.fruitfly.org/annot/apollo/" TARGET=_blank>Apollo 
		    Genome Annotation and Curation Tool</a>.   We also provide data
		    comparing the genome of <EM>D. melanogaster</EM> with that of 
		    <EM>D. pseudoobscura</EM>.
		    <P>
		    The fruitfly, one of the first organisms to be used in systematic
		    scientific investigations, has been the subject of intensive study 
		    in genetics for nearly a century and remains a major model organism 
		    in biomedical research, population biology and evolution.  
	 	    We are pleased to add the fruitfly to the roster of assemblies 
		    available on our site.
	            <P>
		    Downloads of the <EM>Drosophila</EM> data and annotations may be
		    obtained from the UCSC Genome Browser 
		    <A HREF ="ftp://genome.cse.ucsc.edu/goldenPath/dm1">ftp site</A>.
	            <P> We'd like to thank BDGP and the 
		    <A HREF = "http://flybase.bio.indiana.edu">Flybase Consortium</A>
		    (Harvard, U. Cambridge, U. Indiana, UC Berkeley and EBI)
                    for providing the sequence, assembly, and analysis of this
		    genome. We'd also like to acknowledge the members of the 
		    UCSC Genome Bioinformatics group who contributed to this 
		    release: Angie Hinrichs (lead engineer), Heather Trumbower, 
		    Robert Kuhn, Donna Karolchik, and Jim Kent and the system
		    administrators Jorge Garcia, Patrick Gavin and Paul Tatarsky.
	            <P>
                    <FONT FACE="courier" SIZE=3><B>Oct. 17, 2003</B></FONT><BR>
		    Daily and weekly incremental updates of mRNA, RefSeq, 
		    and EST data are now in place for several of the UCSC Genome 
		    Browser assemblies. Data sets that are updated incrementally 
		    from GenBank include the latest human (hg16), mouse (mm3), 
		    and rat (rn3). Fugu (fr1) will soon be added to the list, 
		    with others to follow.
		    <P>
		    Previously, these tables were updated only when we loaded a 
		    new genome assembly into the Genome Browser or made a major 
		    revision to a table. By updating the data on a nightly basis, 
		    we are able to provide researchers with the most current 
		    versions available in GenBank. All new genome assemblies 
		    released after this date will incorporate the incremental 
		    update technology.
		    <P>
		    Data are updated on the following schedule:<BR>
		    - native and xeno mRNA and refSeq tracks - updated daily at 
		    4:30 p.m. Pacific Time (weekdays), early Saturday morning 
		    (weekends)<BR>
		    - EST data - updated once per week on Saturday morning<BR>
		    - downloadable data files - updated weekly on Sunday morning<BR>
		    - outdated sequences - removed once per quarter<BR>
		    <P>
		    Mirror sites are not required to migrate to an incremental 
		    update process, and should not experience problems as a 
		    result of this upgrade. Mirror site questions should be 
		    addressed to 
		    <A HREF="genome-mirror@soe.ucsc.edu">genome-mirror@soe.ucsc.edu</A>.
		    <P>
		    We'd like to acknowledge the hard work of Mark Diekhans in the
		    implementation of this new feature, and thank the QA and 
	  	    sysadmin teams (particularly Paul Tatarsky) for their 
		    support in this release.

	            <P>
                    <FONT FACE="courier" SIZE=3><B>Oct. 17, 2003</B></FONT><BR>
		    The UCSC <A HREF="http://genome.ucsc.edu/cgi-bin/hgText">Table
		    Browser</A> is an excellent tool for retrieving and searching 
		    the data underlying the Genome Browser. We've recently added 
		    some new features to the Table Browser to make it even 
		    easier to query and download data.
		    <P>
		    Many of our users have requested a batch query utility that 
		    will allow them to paste in or upload a list of terms on 
		    which to search. You can now do this by clicking the "Item 
		    name/accession" button, then uploading a list of search 
		    terms by selecting the "Paste in" or "Upload" option. Note 
		    that the Paste option supports wildcards, but the Upload 
		    option does not.
		    <P>
		    We've also added 2 new lists of searchable tables/tracks. 
		    The Browser tracks list contains the names of annotation 
		    tracks in the currently selected assembly as they appear in 
		    the Genome Browser. This list is useful if you don't know 
		    the name of the underlying database table that contains the 
		    data in which you're interested. The Custom tracks list 
		    contains the names of all custom annotation tables currently 
		    loaded into the Genome Browser for the given assembly. This 
		    includes tracks that have been created/loaded by the user as 
		    well as custom annotations created via the Table Browser.
		    <P>
		    If you have feedback or questions about the Table 
		    Browser, please send us email at 
		    <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>.
		    <P>
                    <FONT FACE="courier" SIZE=3><B>Sep. 29, 2003</B></FONT><BR>
		    We are pleased to announce the release of the 
		    <A HREF="http://genome.ucsc.edu/cgi-bin/hgNear/" TARGET=_blank>UCSC 
		    Gene 
		    Family Browser</A>.  This major new addition to our website is 
		    a useful tool for collecting information on groups of genes 
		    that may be related in many different ways. The 
		    Browser provides information on gene expression, 
		    protein homology (both within and across species), GO terms, 
		    and Pfam domains, and cross links to many other databases. 
		    <P>
		    To access the Family Browser, click the link on the top
		    menu on this page. The first time you use the Browser, the
		    application will display a brief overview of the tool and
		    information for starting and configuring it. To read a more
		    detailed description of the Browser, see the Family Browser
		    <A HREF="/goldenPath/help/hgNearHelp" TARGET=_blank>User's 
		    Guide</A>.
		    <P>
		    We are always interested in hearing feedback
		    about the tools on our site. If you have comments or questions
		    about the Family Browser, please email us at 
		    <A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>.

    	            </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
	    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%"><TR><TD>
	            <A NAME="About"></A><FONT SIZE="4"><B>&nbsp; Conditions of Use</B></FONT>
	        </TD></TR></TABLE>
	
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD><TD>
	            <P>
                    The Genome Browser and 
                    the data it displays are freely available for academic, 
                    nonprofit, and personal use with a few exceptions. Check the 
                    README.txt file in the download directory of a particular 
                    assembly to view the use restrictions specific to that 
                    release. Program-driven use is limited to a maximum of one 
                    hit every 10 seconds and no more than 5,000 hits per day.  
	            </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>

        <BR>
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	        
                    <FONT SIZE="4" ><A name="Tech"></A><B>&nbsp; Technical Information About the Assembled Sequence</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="goldenPath/gbdDescriptions.html"><FONT SIZE="4">Description of the annotation database</FONT></A>
	            <LI><A HREF="goldenPath/datorg.html"><FONT SIZE="4">Data organization and format</FONT></A>
	            <LI><A HREF="goldenPath/term.html"><FONT SIZE="4">Terminology</FONT></A>
                    </UL>
                    </TD>
                    <TD WIDTH=15>&nbsp;</TD>
                 </TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
        <BR>
    	
        <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>	
            <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>	
	        <TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%">
                    <TR><TD>
	            <FONT SIZE="4" ><A NAME="Other"></A><B>&nbsp; Other Links</B></FONT>
	        </TD></TR></TABLE>
	        <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD>
                    <UL>
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/Genome/"><FONT SIZE="4">Human Genome Project at UCSC</FONT></A><FONT SIZE="4"></FONT>
	            <LI><A HREF="http://www.cse.ucsc.edu/centers/cbe/"><FONT SIZE="4">Center for Biomolecular Science and Engineering Home Page</FONT></A>
                    </UL>
                </TD><TD WIDTH=15>
                </TD></TR></TABLE>
	    <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
</TD></TR></TABLE>

</BODY></HTML>

