<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                         <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD>
                  <TD>
                  <P>
                  To receive announcements of new genome 
                  assembly releases, new software features, updates and 
                  training seminars by email, subscribe to the
                  <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
                  TARGET=_blank>genome-announce</A> mailing list.</P>

            <!-- start news -->
                <font face="courier" size="3"><b>20 December 2012 - 
		28 New Vertebrate Assemblies!</b></font>
		<p>
		Over the past several weeks, we have released 28 new vertebrate assemblies on the 
		public Genome Browser website, featuring 22 new species and 6 assembly updates. 
		These assemblies were added to support the 60-species Conservation track on the
		latest mouse assembly (mm10/GRCm38). Several of these species were originally
		sequenced and assembled for the Mammalian Genome Project (Lindblad-Toh 
		<em>et al</em>., <em>Nature</em> 2011)*. 
		<P>
		<b><em>Primates:</em></b>
		<UL>
		<LI> <b>Baboon</b> (<em>Papio hamadryas</em>) <b>papHam1</b> &ndash; 
		Pham_1.0 (Nov. 2008) from the Baylor College of Medicine HGSC 
		
		<LI> <b>Bushbaby</b> (<em>Otolemur garnettii</em>) <b>otoGar3</b> &ndash;  
		OtoGar3 (Mar. 2011) from the Broad Institute 
		
		<LI> <b>Chimpanzee</b> (<em>Pan troglodytes</em>) <b>panTro4</b> &ndash; 
		Build 2.1.4 (Feb. 2011) from the Chimpanzee Genome Sequencing Consortium 

		<LI> <b>Gibbon</b> (<em>Nomascus leucogenys</em>) <b> nomLeu2</b> &ndash; 
		Nleu1.1 (Jun. 2011) from the Gibbon Genome Sequencing Consortium 
		
		<LI> <b>Mouse lemur</b> (<em>Microcebus murinus</em>) <b>micMur1</b> &ndash; 
		MicMur1.0 (Jul. 2007) from the Broad Institute 
		
		<LI> <b>Rhesus macaque</b> (<em>Macaca mulatta</em>) <b>rheMac3</b> &ndash; 
		CR_1.0 (Oct. 2010) from the Beijing Genomics Institute 
		
		<LI> <b>Squirrel monkey</b> (<em>Saimiri boliviensis boliviensis</em>) 
		<b>saiBol1</b> &ndash; SaiBol1.0 (Oct. 2011) from the Broad Institute 
		
		<LI> <b>Tarsier</b> (<em>Tarsius syrichta</em>) <b>tarSyr1</b> &ndash; 
		Tarsyr1.0 (Aug. 2008) from WUSTL and the Broad Institute 
		</UL>
		</p>
		<p>
		<b><em>Other mammals:</em></b>
		<UL>
		<LI> <b>Alpaca</b> (<em>Vicugna pacos</em>) <b>vicPac1</b> &ndash; 
		VicPac1.0 (Jul. 2008) from the Broad Institute 
		
		<LI> <b>Armadillo</b> (<em>Dasypus novemcinctus</em>) <b>dasNov3</b> &ndash; 
		DasNov3 (Dec. 2011) from the Broad Institute 
		
		<LI> <b>Cat</b> (<em>Felis catus</em>) <b>felCat5</b> &ndash; 
		Felis_catus-6.2 (Sep. 2011) from the International Cat Genome Sequencing Consortium
		
		<LI> <b>Dolphin</b> (<em>Tursiops truncatus</em>) <b>turTru2</b> &ndash; 
		Ttru_1.4 (Oct. 2011) from the  Baylor College of Medicine HGSC
		
		<LI> <b>Hedgehog</b> (<em>Erinaceus europaeus</em>) <b>eriEur1</b> &ndash; 
		Draft_v1 (Jun. 2006) from the Broad Institute 
		
		<LI> <b>Kangaroo rat</b> (<em>Dipodomys ordii</em>) <b>dipOrd1</b> &ndash; 
		DipOrd1.0 (Jul. 2008) from Baylor College of Medicine HGSC and the Broad Institute 
		
		<LI> <b>Manatee</b> (<em>Trichechus manatus latirostris</em>) <b>triMan1</b> &ndash;
		TriManLat1.0 (Oct. 2011) from the Broad Institute 
		
		<LI> <b>Megabat</b> (<em>Pteropus vampyrus</em>) <b>pteVam1</b> &ndash; 
		PteVap1.0 (Jul. 2008) from the Broad Institute 
		
		<LI> <b>Naked mole rat</b> (<em>Heterocephalus glaber</em>) <b>hetGla2</b> &ndash; 
		HetGla_female_1.0 (Jan. 2012) from the Broad Institute 
		
		<LI> <b>Pig</b> (<em>Sus scrofa</em>) <b>susScr3</b> &ndash; 
		Sscrofa10.2 (Aug. 2011) from the Swine Genome Sequencing Consortium 
		
		<LI> <b>Pika</b> (<em>Ochotona princeps</em>) <b>ochPri2</b> &ndash; 
		OchPri2 (Jul. 2008) from the Broad Institute 
		
		<LI> <b>Rock hyrax</b> (<em>Procavia capensis</em>) <b>proCap1</b> &ndash; 
		Procap1.0 (Jul. 2008) from the Baylor College of Medicine HGSC 
		
		<LI> <b>Shrew</b> (<em>Sorex araneus</em>) <b>sorAra1</b> &ndash; 
		SorAra1.0 (Jun. 2006) from the Broad Institute 
		
		<LI> <b>Sloth</b> (<em>Choloepus hoffmanni</em>) <b>choHof1</b> &ndash; 
		ChoHof1.0 (Jul. 2008) from the Broad Institute 

		<LI> <b>Squirrel</b> (<em>Spermophilus tridecemlineatus</em>) <b>speTri2</b> &ndash;
		SpeTri2.0 (Nov. 2011) from the Broad Institute 

		<LI> <b>Tree shrew</b> (<em>Tupaia belangeri</em>) <b>tupBel1</b> &ndash; 
		Tupbel1.0 (Dec. 2006) from the Broad Institute 
		</UL>
		</p>
		<p>
		<b><em>Other vertebrates:</em></b>
		<UL>
		<LI> <B>Atlantic cod</b> (<em>Gadus morhua</em>) <b>gadMor1</b> &ndash; 
		GadMor_May2010 (May 2010) from Genofisk 
		
		<LI> <B>Budgerigar</b> (<em>Melopsittacus undulatus</em>) <b>melUnd1</b> &ndash; 
		v6.3 (Sep. 2011) from WUSTL 
		
		<LI> <B>Coelacanth</b> (<em>Latimeria chalumnae</em>) <b>latCha1</b> &ndash; 
		LatCha1 (Aug. 2011) from the Broad Institute 
		
		<LI> <B>Nile tilapia</b> (<em>Oreochromis niloticus</em>) <b>oreNil2</b> &ndash; 
		OreNil1.1 (Jan. 2011) from the Broad Institute 
		</UL>
		<p>
		Our sincere thanks to the many organizations who contributed to the release 
		of these datasets. We'd also like to acknowledge the Genome Browser
		staff who worked on this release: Hiram Clawson, Chin Li, Brian Raney,
		Brooke Rhead, Steve Heitner, Pauline Fujita, Luvina Guruvadoo, Greg Roe, and
		Donna Karolchik. Please see the Genome Browser 
		<a href="goldenPath/credits.html">credits page</a> for complete acknowledgment 
		information and data use restrictions. All datasets may be downloaded from our
		<a href="ftp://hgdownload.cse.ucsc.edu/goldenPath/">ftp server</a> or 
		<a href="http://hgdownload.cse.ucsc.edu/downloads.html">downloads</a> page.
		</p>
		<p>
		We regret that we don't have the resources to support the many other 
		genomics research communities who would like us to visualize their data 
		in our browser. However, we can provide you with information for setting up 
		your own browser or track hub to display your data, as well as links to 
		other visualization resources on the web that are specific to your 
		genome of interest. Feel free to contact our 
		<a href="mailto:genome@soe.ucsc.edu">mail list</a> for more information.
		</p>
		<p>
		*Lindblad-Toh K <em>et al</em>. 
		<a href="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		target="_blank">A high-resolution map of human evolutionary constraint using 29 
		mammals</a>. <em>Nature</em>. 2011 Oct 12;478(7370):476-82.
		</p>
		<p>
		<hr>

                <font face="courier" size="3"><b>14 December 2012 -
                Reduced Coverage During the Holidays</b></font>
                <p>
		The UCSC campus will be closed for the holidays from December 22, 2012 through 
		January 1, 2013. The Genome Browser will still be available; 
		however, our backup systems will be offline for two days, December 27 and 28. 
		During this time if we experience any issues, associated fixes are likely to be 
		delayed. The help desk will also be minimally staffed during this period. We will 
		be monitoring the mailing lists for urgent messages, but the bulk of the questions 
		will go unanswered until early January.
		</p>
		<hr>

	    <!-- start archives -->
                <font face="courier" size="3"><b>07 December 2012 -
                dbSNP 137 Available for hg19: </b></font>
                We are pleased to announce the release of four tracks derived
                from dbSNP build 137, available on the human assembly (GRCh37/hg19).
                dbSNP build 137 is available at NCBI. The new tracks contain
                additional annotation data not included in previous dbSNP tracks,
                with corresponding coloring and filtering options in the Genome
                Browser.
		<A HREF="goldenPath/newsarch.html#120712">Read more</A>.
                </p>
		<p>
	    	<font face="courier" size="3"><b>27 November 2012 -
		New UCSC Genes Track Released for Mouse: </b></font>
		We're happy to announce the release of a new UCSC Genes 
		track for the GRCm38/mm10 mouse Genome Browser.
		<A HREF="goldenPath/newsarch.html#112712">Read more</A>.
		</p>
==> <a href="goldenPath/newsarch.html">News Archives</a>

                    </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
