<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

		    <P>
		    <FONT FACE="courier" SIZE="3"><B>29 October 2008 - Medaka Genome Browser Updated</B></FONT><BR>
		    <P>
		    We have updated the initial UCSC Medaka Genome Browser 
		    (oryLat1) to correct an error with chrUn in which the gap 
		    relationships between the contigs within their 
		    ultracontigs were incorrect. This error was introduced by
		    UCSC during the browser assembly process. The fixed 
		    replacement browser, oryLat2 (Oct. 2005), is based on the 
		    same v1.0 assembly produced by the National Institute of 
		    Genetics (NIG) and the University of Tokyo, Japan. 
		    <P>
<!--
		    <P>
		    <FONT FACE="courier" SIZE="3"><B>18 August 2008 - Updated Horse Genome Browser Available</B></FONT><BR>
		    <P>
		    The Sep. 2007 EquCab2 release of the horse genome (Equus 
		    caballus) is now available in the UCSC Genome Browser. This 
		    assembly, UCSC version equCab2, was produced by the 
		    <A HREF="http://www.broad.mit.edu/" TARGET=_blank>Broad 
		    Institute</A>. </P>
		    <P>
		    The horse draft genome has been sequenced to 6.8X coverage. 
		    Approximately 84% of the sequence has been anchored to 
		    chromosomes, which include autosomes 1-31 and sex 
		    chromosome X. Unanchored contigs that could not be 
		    localized to a chromosome have been concatenated into the 
		    virtual chromosome &quot;chrUn&quot;, separated by gaps of 
		    1,000 bp. The mitochondrial sequence is also available in
		    the Genome Browser as the virtual chromosome 
		    &quot;chrM&quot;. 
		    <P>
		    The UCSC Genome Browser team has identified an apparent
		    assembly error in the Broad EquCab2 release: a duplicated
		    25.9 Mb region from chr6 is appended to the distal end of 
		    chr27 (visible as scaffold_34 in the Scaffolds track). We 
		    have notified the Broad Institute of this problem, which 
		    likely will be corrected in their next assembly release. 
		    For more details about the current assembly, see 
		    the Broad Institute 
		    <A HREF="http://www.broad.mit.edu/mammals/horse/"
		    TARGET=_blank>Horse Genome Project</A> page. </P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available from the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/equCab2/">FTP server</A> 
		    or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#horse">Downloads</A> 
		    page. These data have 
		    <A HREF="goldenPath/credits.html#horse_use">specific 
		    conditions for use</A>. </P>
		    <P>
                    The UCSC Horse Genome Browser was produced by Larry Meyer,
		    Hiram Clawson, Pauline Fujita, Brooke Rhead, and Donna 
		    Karolchik. See the 
		    <A HREF="goldenPath/credits.html#horse_credits">Credits</A> 
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. </P>
-->
		    <HR>
		    <FONT FACE="courier" SIZE="3"><B>17 October 2008 - Quality Assurance Engineer Job Opening:</B></FONT>
		    The UCSC Genome browser group has re-opened a search for a 
		    Quality Assurance engineer: Biological Database Testing and
		    User Support Technician. Applicants should have a strong 
		    background in biology and computation, and excellent skills 
		    in scientific writing/editing. For details see the 
		    <A HREF="http://www.cbse.ucsc.edu/jobs/jobs_staff.shtml#bdt"
		    TARGET=_blank>job listing</A>.
		    <P>
		    <FONT FACE="courier" SIZE="3"><B>17 September 2008 - Seven Assemblies Archived:</B></FONT>
		    We have archived seven older assemblies of the human
		    (hg15), mouse (mm5, mm6), rat (rn2), <em>C. elegans</em>
		    (ce1), zebrafish (danRer2), and cow (bosTau1) genomes.
                    <A HREF="goldenPath/newsarch.html#091708">Read more</A>.</P>
                        </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
