<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="http://www.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

<!--
		    <P>
		    <FONT FACE="courier" SIZE="3"><B>18 August 2008 - Updated Horse Genome Browser Available</B></FONT><BR>
		    <P>
		    The Sep. 2007 EquCab2 release of the horse genome (Equus 
		    caballus) is now available in the UCSC Genome Browser. This 
		    assembly, UCSC version equCab2, was produced by the 
		    <A HREF="http://www.broad.mit.edu/" TARGET=_blank>Broad 
		    Institute</A>. </P>
		    <P>
		    The horse draft genome has been sequenced to 6.8X coverage. 
		    Approximately 84% of the sequence has been anchored to 
		    chromosomes, which include autosomes 1-31 and sex 
		    chromosome X. Unanchored contigs that could not be 
		    localized to a chromosome have been concatenated into the 
		    virtual chromosome &quot;chrUn&quot;, separated by gaps of 
		    1,000 bp. The mitochondrial sequence is also available in
		    the Genome Browser as the virtual chromosome 
		    &quot;chrM&quot;. 
		    <P>
		    The UCSC Genome Browser team has identified an apparent
		    assembly error in the Broad EquCab2 release: a duplicated
		    25.9 Mb region from chr6 is appended to the distal end of 
		    chr27 (visible as scaffold_34 in the Scaffolds track). We 
		    have notified the Broad Institute of this problem, which 
		    likely will be corrected in their next assembly release. 
		    For more details about the current assembly, see 
		    the Broad Institute 
		    <A HREF="http://www.broad.mit.edu/mammals/horse/"
		    TARGET=_blank>Horse Genome Project</A> page. </P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available from the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/equCab2/">FTP server</A> 
		    or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#horse">Downloads</A> 
		    page. These data have 
		    <A HREF="credits.html#horse_use">specific 
		    conditions for use</A>. </P>
		    <P>
                    The UCSC Horse Genome Browser was produced by Larry Meyer,
		    Hiram Clawson, Pauline Fujita, Brooke Rhead, and Donna 
		    Karolchik. See the 
		    <A HREF="credits.html#horse_credits">Credits</A> 
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. </P>
		    <P>
		    <HR>
-->
		    <P>
		    <FONT FACE="courier" SIZE="3"><B>22 July 2008 - Updated <em>C. elegans</em> Browser Released</B></FONT><BR>
		    <P>
		    Culminating our recent updates of the nematode browsers on
		    our site, we've released a Genome Browser for the latest
		    <em>C. elegans</em> assembly based on sequence version WS190
		    deposited into WormBase in May 2008. This new assembly
		    (UCSC version ce6) was produced jointly by the Wellcome
		    Trust Sanger Institute in Hinxton, England, and the Genome 
		    Sequencing Center at Washington University in St. Louis 
		    (WUSTL) School of Medicine, MO, USA. The ce6 browser 
		    includes a Conservation annotation and pairwise alignment 
		    tracks comparing <em>C. elegans</em> sequence with that of 
		    five other worms.</P>
		    <P>
		    Bulk downloads of the sequence and annotation data are 
		    available via the Genome Browser 
		    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/ce6">FTP
		    server</A> or 
		    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#c_elegans">Downloads</A> 
		    page. Please review the WUSTL
		    <A HREF="http://genome.wustl.edu/data.cgi"
		    TARGET=_blank>data use policy</A> for usage restrictions
		    and citation information. 
		    </P>
		    <P>
		    We'd like to thank WUSTL and the Sanger Institue for 
		    providing the sequence data 
		    for this assembly. The UCSC ce6 browser was produced by 
		    Hiram Clawson, Kayla Smith, Pauline Fujita, Brooke Rhead,
		    and Donna Karolchik. See the Genome Browser
		    <A HREF="goldenPath/credits.html#worm_credits">Credits</A> 
		    page for a detailed list of the organizations and 
		    individuals who contributed to this release. </P>
		    <P>
		    <HR>
		    <FONT FACE="courier" SIZE="3"><B>26 June 2008 - New Worm Genome Available: </B></FONT>
		    Along with the set of worm browser updates that we're 
		    currently releasing, we've added a new nematode to the 
		    collection: <em>Caenorhabditis japonica</em>. 
                    <A HREF="goldenPath/newsarch.html#062608">Read more</A>.</P>
		    <FONT FACE="courier" SIZE="3"><B>20 June 2008 - Two Worm Updates Released: </B></FONT>
		    We've updated our browsers for the <em>C. 
		    remanei</em> and <em>C. brenneri</em> nematode genomes. 
                    <A HREF="goldenPath/newsarch.html#062008">Read more</A>.</P>
                        </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
