<!DOCTYPE html>
<!--#set var="TITLE" value="Genome Browser Staff" -->
<!--#set var="ROOT" value="." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<!--#include virtual="$ROOT/redmineWidget.html" -->

<h1>UCSC Genome Browser Store</h1>

<p>
<style="color: red;">All products offered are free for personal and non-profit academic research use.</span> Commercial use requires purchase of a license with setup fee and annual payment. A non-profit organization in service to a for-profit organization must purchase a commercial license, even if the latter already possesses a commercial product license.

During checkout, non-profit academic users will be given the option to declare that they are not using the products commercially and qualify for a free license.

</p>


<table class="table">
	<tr>
		<td>
			<h2>Genome Browser in a Box (GBiB)</h2>
		</td>
	</tr>
	<tr>
		<td>

 <div class="product-details">Genome Browser in a Box (GBiB) is a small, virtual machine version of the UCSC Genome Browser that can be run on your own laptop or desktop computer. Annotation data is loaded on demand through the Internet from UCSC or can be downloaded to your machine for faster access. Once GBiB is installed, you use a web browser to access the virtual server running on your own machine and add personal data files from your local hard disk.  A more detailed description of this tool and how to use it is available at <a href="https://genome.ucsc.edu/goldenpath/help/gbib.html" target="_blank">https://genome.ucsc.edu/goldenpath/help/gbib.html</a>.</div>
      
      
        
          <div class="product-eula">
          
            <a href="https://genome-store.ucsc.edu/media/eula/2019/10/14/UCSC_GB_EULA.pdf" target="_blank">Terms and conditions</a>
          
          </div>
        
      
      
      <div class="product-price">
        
          <strong>One time setup fee:</strong> $1000<br>
        
        
          
            <strong>Cost per year:</strong> $1100<br>
          
        
        <strong><span style="color:red">Cost for non-profit academic research use:</span></strong> Free<br>
      </div>
      
        <div class="product-version">
          
          <div class="product-size">
            <strong>Size:</strong> 10.8 GB
          </div>
          
          <div class="product-requirements">
            <strong>System Requirements:</strong><br>
            GBiB will run on most modern PCs and major operating systems that meet these basic requirements:
<ul>
<li>The computer must support virtualization (common for most PCs sold after 2010).</li>
<li>A compatible version of the <a href="https://www.virtualbox.org/" target="_blank">VirtualBox Software</a> must be installed. This software is free to use in many situations. See the VirtualBox wiki for <a href="https://www.virtualbox.org/wiki/Licensing_FAQ">licensing terms and conditions</a> and <a href="https://www.virtualbox.org/wiki/Downloads">installation</a> instructions. We have tested and confirmed that GBiB works using version 4.3.6 or above. Note that you must have administrator rights for your computer to install VirtualBox.</li>
<li>The computer disk must have at least 25 GB of free space (more if you plan to mirror many tracks).</li>
<li>Your network firewall must allow connections on the following ports:
<ul>
<li>Port 3306, used by MySQL. Without this, only mirrored tracks are shown.</li>
<li>TCP port 873, used by rsync. Without this, you cannot download track data.</li>
</ul></li>
<li> The GBiB product includes executable versions of <a href="http://www.linux.org/" target="_blank">LINUX</a>, <a href="http://httpd.apache.org" target="_blank">Apache</a>, and <a href="http://dev.mysql.com" target="_blank">MySQL</a>. Please follow the links to determine if you have the correct license for each of these open source programs</li>
<li>The Online Mendelian Inheritance in Man (<a href="http://www.omim.org/help/agreement" target="_blank">OMIM</a>) content in the GBiB is copyrighted and owned by Johns Hopkins University. Use of the copyrighted material solely within GBiB is permitted. Redistribution or repackaging of the copyrighted material requires a license from Johns Hopkins University.</li>
</ul>
          </div>
          
          
          <div class="product-includes">
            <strong>Product Includes:</strong><br>
            <ul>
<li>The Genome Browser in a Box setup program. This program:</li>
<ul>
<li>downloads the Genome Browser CGIs, HTML, and Javascript files</li>
<li>configures the machine to download supporting data files from UCSC</li>
<li>(optionally) downloads the UCSC command line tools to itself</li>
<li>automatically updates software (including non-gb software) and data</li>
</ul>
<li>Pre-downloaded version of the hg19 database (not the full set but a majority of tables).</li>
<li>Pre installed versions of Apache and MySQL.</li>
</ul>
          </div>
          
          
          <div class="product-not_includes">
            <strong>Product does not Include:</strong><br>
            <ul>
<li>The UCSC Genome Browser source code.</li>
<li>Virtual Machine software required to run GBiB.</li>
</ul>
          </div>
          
        </div>
		</td>
	</tr>
</table>





<table class="table">
	<tr>
		<td>
			<h1>Genome Browser source code</h2>
		</td>
	</tr>
	<tr>
		<td>

<div class="product-details">The Genome Browser source code consists of all of the software files necessary to build and compile a full mirror site of the Genome Browser on your own server. Additionally, it builds more than 200 bioinformatics command-line utilities. You must also obtain and install Apache, MySQL and other software packages (see ‘system requirements’).</div>
      
      
        
          <div class="product-eula">
          
            <a href="https://genome-store.ucsc.edu/media/eula/2019/10/14/UCSC_GB_EULA.pdf" target="_blank">Terms and conditions</a>
          
          </div>
        
      
      
      <div class="product-price">
        
          <strong>One time setup fee:</strong> $6000<br>
        
        
          <strong>Cost per year per user:</strong> $1000 (5 users minimum)<br>
        
        <strong><span style="color:red">Cost for non-profit academic research use:</span></strong> Free<br>
      </div>
      
        <div class="product-version">
          
          <div class="product-size">
            <strong>Size:</strong> 110 MB
          </div>
          
          <div class="product-requirements">
            <strong>System Requirements:</strong><br>
            The following software packages are required for a full install of the Genome Browser. Please visit each link to determine if you are required to purchase or sign a license:
<ul>
<li>A multiple-CPU machine with a Linux/Ubuntu/CentOS/Unix/MacOSX operating system</li>
<li>Apache2.x - <a href="http://httpd.apache.org" target="_blank">http web server</a></li>
<li><a href="http://www.gnu.org/software/gcc" target="_blank">gnu gcc, C code development system</a></li>
<li><a href="http://www.gnu.org/software/make" target="_blank">gnu make</a></li>
<li><a href="http://dev.mysql.com" target="_blank">MySQL development system and libraries</a></li>
<li><a href="http://www.libpng.org" target="_blank">libpng runtime and development packages</a></li>
<li><a href="http://www.openssl.org" target="_blank">libssl runtime and development packages</a></li>
</ul>
          </div>
          
          
          <div class="product-includes">
            <strong>Product Includes:</strong><br>
            <ul>
<li>A complete set of the cgi-bin files, HTML files, Javascript files, and database files needed to run a stock version of the UCSC Genome Browser.</li>
<li>A complete set of source files required to build the UCSC Genome Browser, including .c and .h files, Javascript files, HTML files, makefiles, include files, libraries, database files, and instructions for building the UCSC Genome Browser from source.</li>
<li>Installation instructions, including system requirements.</li>
<li>A complete set of online user documentation (HTML format).</li>
</ul>
          </div>
          
          
          <div class="product-not_includes">
            <strong>Product does not Include:</strong><br>
            <ul>
<li>Source or executables for the Blat Search tool.</li>
<li>Genomic sequence or annotation data (this data may be freely downloaded from the UCSC Genome Browser website. (http://genome.ucsc.edu/downloads.html) or from many other publicly available sources on the internet, with restrictions outlined in the README files that accompany the data).</li>
<li>The supporting software required to build, install and run the UCSC Genome Browser (e.g., C compilation environment, Apache server, MariaDB).</li>
</ul>
          </div>
          
        </div>


		</td>
	</tr>
</table>





<table class="table">
	<tr>
		<td>
			<h2>LiftOver program</h2>
		</td>
	</tr>
	<tr>
		<td>
<div class="product-details">The LiftOver program can be used to convert coordinate ranges between genome assemblies. This is a command-line tool, and supports forward/reverse conversions, batch conversions, and conversions between species. The LiftOver program requires a UCSC-generated over.chain file as input. Pre-generated files are available for selected assemblies from the <a href="http://hgdownload.cse.ucsc.edu/downloads.html" target="_blank">downloads page</a>.</div>
      
      
        
          <div class="product-eula">
          
            <a href="https://genome-store.ucsc.edu/media/eula/2019/10/14/UCSC_GB_EULA.pdf" target="_blank">Terms and conditions</a>
          
          </div>
        
      
      
      <div class="product-price">
        
        
          
            <strong>Cost per year:</strong> $1000<br>
          
        
        <strong><span style="color:red">Cost for non-profit academic research use:</span></strong> Free<br>
      </div>
      
        <div class="product-version">
          <h4>LiftOver command-line program (LINUX 64-bit)</h4>
          <div class="product-size">
            <strong>Size:</strong> 23.4 MB
          </div>
          
          
          <div class="product-includes">
            <strong>Product Includes:</strong><br>
            <ul>
<li>Pre-compiled LiftOver standalone command line tool for LINUX or MacOSX.</li>
</ul>
          </div>
          
          
          <div class="product-not_includes">
            <strong>Product does not Include:</strong><br>
            <ul>
<li>The UCSC Genome Browser source code.</li>
<li>The over.chain data files.</li>
</ul>
          </div>
          
        </div>
      
        <div class="product-version">
          <h4>LiftOver command-line program (Mac OSX 64-bit)</h4>
          <div class="product-size">
            <strong>Size:</strong> 9.35 MB
          </div>
          
          
          <div class="product-includes">
            <strong>Product Includes:</strong><br>
            <ul>
<li>Pre-compiled LiftOver standalone command line tool for LINUX or MacOSX.</li>
</ul>
          </div>
          
          
          <div class="product-not_includes">
            <strong>Product does not Include:</strong><br>
            <ul>
<li>The UCSC Genome Browser source code.</li>
<li>The over.chain data files.</li>
</ul>
          </div>
          
        </div>
		</td>
	</tr>
</table>






<table class="table">
	<tr>
		<td>
			<h2>Genome Browser in the Cloud (GBiC)</h2>
		</td>
	</tr>
	<tr>
		<td>
<div class="product-details">Genome Browser in the Cloud (GBiC) is a convenient program that automates the setup of a UCSC Genome Browser mirror, including the installation and setup of MySQL (or MariaDB) and Apache servers. The program downloads and configures MySQL and Apache, then downloads the UCSC Genome Browser software to /usr/local/apache. The program can also be used to mirror full or partial assembly databases, keep up-to-date with the Genome Browser software, remove temporary files, and install the Kent command line utilities. 

You can use the GBiC program to install a Genome Browser mirror on any Linux based operating system, including your server, desktop/laptop, or cloud instance. The program overwrites the default Apache install location, and thus is intended to be run on machines/cloud instances that don’t already act as a web server. A more detailed description of this tool and how to use it is available from the <a href="https://genome.ucsc.edu/goldenpath/help/gbic.html" target="_blank">GBIC User Guide</a>. If you have questions about the script please send an email to the publicly archived Genome Browser Mirror mailing list:  <a href="mailto:genome-mirror@soe.ucsc.edu" target="_blank">genome-mirror@soe.ucsc.edu</a>.</div>
      
      
        
          <div class="product-eula">
          
            <a href="https://genome-store.ucsc.edu/media/eula/2019/10/14/UCSC_GB_EULA.pdf" target="_blank">Terms and conditions</a>
          
          </div>
        
      
      
      <div class="product-price">
        
          <strong>One time setup fee:</strong> $2000<br>
        
        
          
            <strong>Cost per year:</strong> $1000<br>
          
        
        <strong><span style="color:red">Cost for non-profit academic research use:</span></strong> Free<br>
      </div>
      
        <div class="product-version">
          
          <div class="product-size">
            <strong>Size:</strong> 68.4 KB
          </div>
          
          <div class="product-requirements">
            <strong>System Requirements:</strong><br>
            The GBiC program will run on most Debian or RedHat based Linux operating systems, however there are a few additional considerations:
<ul>
<li>Your network firewall must allow connections on the following ports:
<ul>
<li>Port 3306, used by MySQL/MariaDB. Without this, only locally <a href="http://genome.ucsc.edu/goldenPath/help/gbic.html#the-gbic-commands" target="_blank">mirrored tracks</a> are shown.</li>
<li>TCP port 873/9000, used by rsync/<a href="https://github.com/LabAdvComp/UDR" target="_blank">UDR</a>. Without this, you cannot download the software or track data.</li>
</ul></li>
</li>The GBiC product installs <a href="http://httpd.apache.org" target="_blank">Apache</a> and <a href="http://dev.mysql.com" target="_blank">MySQL</a>/<a href="https://mariadb.org/" target="_blank">MariaDB</a>. Please follow the links to determine if you have the correct license for each of these open source programs.</li>
<li>The tool has been tested on Ubuntu 14 & 16 LTS, CentOS 6 & 7, and Fedora 20. Other Linux distributions have not been tested yet and may not work correctly. </li>
<li>The utility has been tested on virtual machines in Amazon’s EC2 (Centos 6 and Ubuntu 14) and Microsoft’s Azure (Ubuntu). Other cloud providers will likely work, but have not been tested.</li>
<li>The minimum amount of disk space needed to load all data from UCSC on-the-fly is about 5GB, although temporary files will accumulate quickly after normal usage and take up more space. In order to fully mirror the hg19 assembly database locally, approximately 10TB of disk space is needed, however most other organism databases are significantly smaller -- in the 5-10GB range -- plus around 40GB for associated GenBank data. This GenBank data will only be downloaded once. </li>
<li>Around 4GB of memory is required to display all of chromosome 1 on hg19. Most other assemblies will require significantly less memory.</li>
</ul>

          </div>
          
          
          <div class="product-includes">
            <strong>Product Includes:</strong><br>
            <ul>
<li>The Genome Browser in the Cloud setup program. This program:</li>
<ul>
<li>downloads the Genome Browser CGIs</li>
<li>installs Apache, MySQL (or MariaDB), and Ghostscript</li>
<li>configures the machine to download supporting data files from UCSC</li>
<li>(optionally) downloads the UCSC command line tools to itself </li>
</ul>
</ul>
          </div>
          
          
          <div class="product-not_includes">
            <strong>Product does not Include:</strong><br>
            <ul>
<li>The UCSC Genome Browser source code.</li>
</ul>
          </div>
          
        </div>
		</td>
	</tr>
</table>




Please address all questions and comments about the UCSC Genome Browser Store to genome-browser-store-group@ucsc.edu.






<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->
