<!DOCTYPE html>
<!--#set var="TITLE" value="Other Tools - UCSC Genome Browser" -->
<!--#set var="ROOT" value="." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<h1>Other Tools from the UCSC Genome Browser</h1>
<p>
The UCSC Genome Browser team develops and updates the following main tools:
the <a href="../cgi-bin/hgTracks">Genome Browser</a>,
<a href="../cgi-bin/hgBlat">BLAT</a>, <a href="../cgi-bin/hgPcr">In-Silico PCR</a>, 
<a href="../cgi-bin/hgTables">Table Browser</a>, and <a href="../cgi-bin/hgLiftOver">LiftOver</a>.
We maintain the following less-used tools: <a href="../cgi-bin/hgNear">Gene Sorter</a>,
<a href="../cgi-bin/hgGenome">Genome Graphs</a>, and
<a href="../cgi-bin/hgIntegrator">Data Integrator</a>.
These are available from the &quot;Tools&quot; dropdown menu at the top of the site.</p>

<p>
We also offer command-line utilities for many file conversions and basic bioinformatics functions.
You can learn more and download these utilities through the 
<a href="https://hgdownload.gi.ucsc.edu/downloads.html#utilities_downloads">utilities section</a>
of our downloads page.</p>

<p>
The following tools and utilities created by the UCSC Genome Browser Group are also available 
for public use:</p>
<ul>
   <li>
   <strong><a href="../cgi-bin/hgGenome">Genome Graphs</a></strong> -
   Allows for the display of any number of annotations as genome-wide density graphs. Useful
   for exploratory data analysis, publication images, and specific data such as linkage studies.</li>
   <li>
   <strong><a href="../cgi-bin/hgNear">Gene Sorter</a></strong> -
   Displays a sorted table of genes that are related to one another. The relationship can be one 
   of several types, including protein-level homology, similarity of gene expression profiles, 
   or genomic proximity.</li>
   <li>
   <strong><a href="../cgi-bin/hgPhyloPlace">UShER</a> (for SARS-CoV-2 only)</strong> - 
   allows placement of SARS-CoV-2 sequences onto existing phylogenetic trees for tracing strains
   and transmission events.</li>
   <li>
   <strong><a title="Explore gene/protein interactions and pathways" 
   href="../cgi-bin/hgGeneGraph">Gene Interactions</a></strong> - 
   displays a gene interaction and pathway graph based on data from curated protein-interaction 
   databases and text mining of PubMed abstracts.</li>   
   <li>
   <strong><a id="visiGeneMenuLink" href="../cgi-bin/hgVisiGene">VisiGene</a></strong> -
   displays high-quality images of in-situ mouse and frogs with gene mRNA hybridization 
   probes and transcription factors.</li>
   <li>
  <strong><a href="http://users.soe.ucsc.edu/~kent/dnaDust/dnadust.html">DNA Duster</a></strong> - 
  removes formatting characters and other non-sequence-related characters from an input sequence. 
  Offers several configuration options for the output format, including translated protein.</li>
  <li>
  <strong><a href="http://users.soe.ucsc.edu/~kent/protDust/protDust.html">Protein 
  Duster</a></strong> -  removes formatting characters and other non-sequence-related characters 
  from an input sequence. Offers several configuration options for the output format.</li>
  <li>
  <strong><a href="cgi-bin/phyloPng">Phylogenetic Tree PNG Maker</a></strong> - creates a PNG image 
  from phylogenetic tree specification given; several configuration options for branch 
  lengths, normalized lengths, branch labels, legend, etc.</li>
  <li>
  <strong><a href="http://hgdownload.gi.ucsc.edu/downloads.html#source_downloads">Executable and 
  Source Code Downloads</a></strong> - executable and source code downloads of the Genome Browser, 
  Blat and liftOver.</li>
</ul>
</p>
<p>
The following tools and utilities created by outside groups may be helpful when working with our
file formats and the genome annotation databases that we provide. References to these tools are
provided for the benefit of our users. We are unable to support the use of externally developed
tools; if you have questions or problems, please contact the developers of the tool directly.
<ul>
  <li><strong><a href="https://bedops.readthedocs.io/en/latest/">BEDOPS</a></strong> -
      A highly scalable and easily-parallelizable genome analysis toolkit
  </li>
  <li><strong><a href="https://bedtools.readthedocs.io/en/latest/">bedtools</a></strong> -
      A swiss-army knife of tools for a wide-range of genomics analysis tasks
  </li>
  <li><strong><a href="https://github.com/CRG-Barcelona/bwtool/wiki">bwtool</a></strong> -
      A command-line utility for bigWig files
  </li>
  <li><strong><a href="http://crossmap.sourceforge.net/">CrossMap</a></strong> -
      A program for genome coordinate conversion between different assemblies
  </li>
  <li><strong><a href="https://pypi.org/project/cruzdb/">CruzDb</a></strong> -
      Python package to load genome annotations from our servers
  </li>
  <li><strong><a href="https://g-onramp.org/">G-OnRamp</a></strong> -
      GEP + Galaxy (G-OnRamp) tool to create assembly hub browsers for eukaryotic genomes
  </li>
  <li><strong><a href="https://github.com/dpryan79/libBigWig">libBigWig</a></strong> - 
      A C library to read bigWig files, without a dependency on our source code
  </li>
  <li><strong><a href="https://github.com/Gaius-Augustus/MakeHub">MakeHub</a></strong> -
      Python tool to build assembly hub files for new genomes
  </li>
  <li><strong><a href="https://bioconductor.org/packages/release/bioc/html/rtracklayer.html">RTracklayer</a></strong> -
      R package to import genome annotations from our databases
  </li>
  <li><strong><a href="https://daler.github.io/trackhub/">trackhub</a></strong> -
      Python package to manage files in our trackDb format
  </li>
  <li><strong><a href="https://pypi.org/project/twobitreader/">twobitreader</a></strong> - Python package to open
      2bit genome sequence files
  </li>
  <li><strong><a href="https://pypi.org/project/ucsc-genomes-downloader/">ucsc-genomes-download</a></strong> -
      Python package to download genome sequences from our servers
  </li>
  <li><strong><a href="https://github.com/Ensembl/WiggleTools">WiggleTools</a></strong> -
      C++ Unix command-line tool to work with bigWig files: combine, merge, scale, aggregate and many other operations
  </li>
</ul>
</p>
<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->
