<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=185>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Choloepus%20hoffmanni"
    TARGET=_blank>
    <IMG SRC="/images/Choloepus_hoffmanni.jpg" WIDTH=*** HEIGHT=*** ALT="Hoffmanns two-fingered sloth"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Choloepus hoffmanni</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of
      (<A HREF="http://animaldiversity.ummz.umich.edu/site/index.html"
      TARGET=_blank>Dr. Tanya Dewey</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The February 2008 <em>Choloepus hoffmanni</em> draft assembly 
(WUSTL Choloepus_hoffmanni_1.0) was produced by the
    <A HREF="http://genome.wustl.edu/genome.cgi?GENOME=Choloepus%20hoffmanni"
    TARGET=_blank>WUSTL Genome sequencing center</A>.

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
Hoffmanns two-fingered sloth genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>chrZ</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of chromosome Z</TD>
    </TR>
    <TR><TD VALIGN=Top>chr3:1-1000000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of chr 3</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr3:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr 3 that spans 2000 bases, starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>BX950328</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number BX950328</TD>
    </TR>
    <TR><TD VALIGN=Top>BU335953</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		BU335953 on chr Z</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>pseudogene mRNA</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>zhang</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zhang</TD>
    </TR>
    <TR><TD VALIGN=Top>Taguchi,T.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author T. Taguchi</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Taguchi T</em>, internally it
	uses the format <em>Taguchi,T.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
Description from:
<A HREF="ftp://genome.wustl.edu/pub/organism/Other_Vertebrates/Choloepus_hoffmanni/assembly/Choloepus_hoffmanni-1.0/ASSEMBLY"
TARGET=_blank>WUSTL GSC</A><BR>
The two-toed sloth, Choloepus hoffmanni, was sequenced as part of an
NHGRI initiative to expand sequencing of mammalian genomes. The
C. hoffmanni genome was sequenced to ~2X whole genome coverage with
various sized plasmids. The combined sequence reads were assembled
using the PCAP software (Genome Res. 13(9):2164-70 2003) and filtered
for all known non-sloth sequence contaminants. This first draft
assembly is referred to as Choloepus_hoffmanni_1.0.  Funding
for the sequence characterization of the sloth genome was provided by
the National Human Genome Research Institute (NHGRI), National
Institutes of Health (NIH).
</P>
<P>
This assembly is 352,568 contigs, 2,677,018,039 bp.  Fifty percent
of the sequence is contained in 3,547 contigs of size 31,241 bp (N50)
or greater.  The largest contig is 131,429 bp, the smallest 60 bp.
</P>


<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/choHof1/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#hoffmanns_two-fingered sloth">Downloads</A> page. 

These data have
<A HREF="../goldenPath/credits.html#hoffmanns_two-fingered sloth_use">specific 
conditions for use</A>. 

The Hoffmanns two-fingered sloth browser annotation tracks were generated
by UCSC and collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#hoffmanns_two-fingered sloth_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
