</P>
<H2>Description</H2>
<P>
The aim of the <a href="https://www.gencodegenes.org/" target="_blank">GENCODE
Genes project</a> (Harrow <em>et al.</em>, 2006) is to produce a set of
highly accurate annotations of evidence-based gene features on the human reference genome.
This includes the identification of all protein-coding loci with associated
alternative splice variants, non-coding with transcript evidence in the public
databases (NCBI/EMBL/DDBJ) and pseudogenes. A high quality set of gene
structures is necessary for many research studies such as comparative or
evolutionary analyses, or for experimental design and interpretation of the
results.</p>
<p>
The GENCODE Genes tracks display the high-quality manual annotations merged
with evidence-based automated annotations across the entire
human genome. The GENCODE gene set presents a full merge
between HAVANA manual annotation and Ensembl automatic annotation.
Priority is given to the manually curated HAVANA annotation using predicted
Ensembl annotations when there are no corresponding manual annotations. With
each release, there is an increase in the number of annotations that have undergone
manual curation.
This annotation was carried out on the GRCm38 (mm10) genome assembly.
</p>
<p>Experimental verification details are given in each descriptions for each
track.</p>

<p>
For more information on the different gene tracks, see our <a target=_blank 
href="/FAQ/FAQgenes.html">Genes FAQ</a>.</p>

<h2>Display Conventions</h2>
<P>
These are multi-view composite tracks that contain differing data sets
(<EM>views</EM>).  Instructions for configuring multi-view tracks are
<A HREF="/goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
Only some subtracks are shown by default. The user can select which subtracks
are displayed via the display controls on the track details pages.
Further details on display conventions and data interpretation are available in the track descriptions.</p>

<h2>Release Notes</h2>
<p>
<span style="font-weight: bold;">GENCODE version M37</span>
corresponds to Ensembl 114.
</p>
<p>
<span style="font-weight: bold;">GENCODE version M36</span>
corresponds to Ensembl 113.
</p>
<p>
<span style="font-weight: bold;">GENCODE version M35</span>
corresponds to Ensembl 112.
</p>
<p>
<span style="font-weight: bold;">GENCODE version M34</span>
corresponds to Ensembl 111.
</p>
<p>
<span style="font-weight: bold;">GENCODE version M33</span>
corresponds to Ensembl 110.
</p>
<p>
<span style="font-weight: bold;">GENCODE version M32</span>
corresponds to Ensembl 109.
</p>
<p>
<span style="font-weight: bold;">GENCODE version M31</span>
corresponds to Ensembl 108.
</p>
<p>
<span style="font-weight: bold;">GENCODE version M30</span>
corresponds to Ensembl 107.
</p>
<p>
<span style="font-weight: bold;">GENCODE version M29</span>
corresponds to Ensembl 106.
</p>
<p>
<span style="font-weight: bold;">GENCODE version M28</span>
corresponds to Ensembl 105.
</p>
<p>
<span style="font-weight: bold;">GENCODE version M27</span>
corresponds to Ensembl 104.
</p>
<p>
<span style="font-weight: bold;">GENCODE version M26</span>
corresponds to Ensembl 103.
</p>
<p>
See also: <a href="https://www.gencodegenes.org/mouse/releases.html" 
target="_blank">The GENCODE Project Release History</a>.
</p>
<!--#insert file="../../wgEncodeGencodeCredits1.shared.html"-->
